// @azure/arm-cdn/dist-esm/src/models/parameters.js
/*
 * Copyright (c) Microsoft Corporation.
 * Licensed under the MIT License.
 *
 * Code generated by Microsoft (R) AutoRest Code Generator.
 * Changes may cause incorrect behavior and will be lost if the code is regenerated.
 */
import { CheckEndpointNameAvailabilityInput as CheckEndpointNameAvailabilityInputMapper, CheckNameAvailabilityInput as CheckNameAvailabilityInputMapper, ValidateProbeInput as ValidateProbeInputMapper, CheckHostNameAvailabilityInput as CheckHostNameAvailabilityInputMapper, ValidateSecretInput as ValidateSecretInputMapper, ProfileUpgradeParameters as ProfileUpgradeParametersMapper, AFDDomain as AFDDomainMapper, AFDDomainUpdateParameters as AFDDomainUpdateParametersMapper, AFDEndpoint as AFDEndpointMapper, AFDEndpointUpdateParameters as AFDEndpointUpdateParametersMapper, AfdPurgeParameters as AfdPurgeParametersMapper, ValidateCustomDomainInput as ValidateCustomDomainInputMapper, AFDOriginGroup as AFDOriginGroupMapper, AFDOriginGroupUpdateParameters as AFDOriginGroupUpdateParametersMapper, AFDOrigin as AFDOriginMapper, AFDOriginUpdateParameters as AFDOriginUpdateParametersMapper, Route as RouteMapper, RouteUpdateParameters as RouteUpdateParametersMapper, Rule as RuleMapper, RuleUpdateParameters as RuleUpdateParametersMapper, SecurityPolicy as SecurityPolicyMapper, SecurityPolicyUpdateParameters as SecurityPolicyUpdateParametersMapper, Secret as SecretMapper, Profile as ProfileMapper, ProfileUpdateParameters as ProfileUpdateParametersMapper, CanMigrateParameters as CanMigrateParametersMapper, MigrationParameters as MigrationParametersMapper, Endpoint as EndpointMapper, EndpointUpdateParameters as EndpointUpdateParametersMapper, PurgeParameters as PurgeParametersMapper, LoadParameters as LoadParametersMapper, Origin as OriginMapper, OriginUpdateParameters as OriginUpdateParametersMapper, OriginGroup as OriginGroupMapper, OriginGroupUpdateParameters as OriginGroupUpdateParametersMapper, CustomDomainParameters as CustomDomainParametersMapper, CustomDomainHttpsParameters as CustomDomainHttpsParametersMapper, CdnWebApplicationFirewallPolicy as CdnWebApplicationFirewallPolicyMapper, CdnWebApplicationFirewallPolicyPatchParameters as CdnWebApplicationFirewallPolicyPatchParametersMapper, } from "../models/mappers";
export const contentType = {
    parameterPath: ["options", "contentType"],
    mapper: {
        defaultValue: "application/json",
        isConstant: true,
        serializedName: "Content-Type",
        type: {
            name: "String",
        },
    },
};
export const checkEndpointNameAvailabilityInput = {
    parameterPath: "checkEndpointNameAvailabilityInput",
    mapper: CheckEndpointNameAvailabilityInputMapper,
};
export const accept = {
    parameterPath: "accept",
    mapper: {
        defaultValue: "application/json",
        isConstant: true,
        serializedName: "Accept",
        type: {
            name: "String",
        },
    },
};
export const $host = {
    parameterPath: "$host",
    mapper: {
        serializedName: "$host",
        required: true,
        type: {
            name: "String",
        },
    },
    skipEncoding: true,
};
export const subscriptionId = {
    parameterPath: "subscriptionId",
    mapper: {
        serializedName: "subscriptionId",
        required: true,
        type: {
            name: "String",
        },
    },
};
export const resourceGroupName = {
    parameterPath: "resourceGroupName",
    mapper: {
        constraints: {
            Pattern: new RegExp("^[-\\w\\._\\(\\)]+$"),
            MaxLength: 90,
            MinLength: 1,
        },
        serializedName: "resourceGroupName",
        required: true,
        type: {
            name: "String",
        },
    },
};
export const apiVersion = {
    parameterPath: "apiVersion",
    mapper: {
        defaultValue: "2024-02-01",
        isConstant: true,
        serializedName: "api-version",
        type: {
            name: "String",
        },
    },
};
export const checkNameAvailabilityInput = {
    parameterPath: "checkNameAvailabilityInput",
    mapper: CheckNameAvailabilityInputMapper,
};
export const validateProbeInput = {
    parameterPath: "validateProbeInput",
    mapper: ValidateProbeInputMapper,
};
export const profileName = {
    parameterPath: "profileName",
    mapper: {
        constraints: {
            Pattern: new RegExp("^[a-zA-Z0-9]+(-*[a-zA-Z0-9])*$"),
            MaxLength: 260,
            MinLength: 1,
        },
        serializedName: "profileName",
        required: true,
        type: {
            name: "String",
        },
    },
};
export const profileName1 = {
    parameterPath: "profileName",
    mapper: {
        serializedName: "profileName",
        required: true,
        type: {
            name: "String",
        },
    },
};
export const checkHostNameAvailabilityInput = {
    parameterPath: "checkHostNameAvailabilityInput",
    mapper: CheckHostNameAvailabilityInputMapper,
};
export const validateSecretInput = {
    parameterPath: "validateSecretInput",
    mapper: ValidateSecretInputMapper,
};
export const profileUpgradeParameters = {
    parameterPath: "profileUpgradeParameters",
    mapper: ProfileUpgradeParametersMapper,
};
export const nextLink = {
    parameterPath: "nextLink",
    mapper: {
        serializedName: "nextLink",
        required: true,
        type: {
            name: "String",
        },
    },
    skipEncoding: true,
};
export const customDomainName = {
    parameterPath: "customDomainName",
    mapper: {
        serializedName: "customDomainName",
        required: true,
        type: {
            name: "String",
        },
    },
};
export const customDomain = {
    parameterPath: "customDomain",
    mapper: AFDDomainMapper,
};
export const customDomainUpdateProperties = {
    parameterPath: "customDomainUpdateProperties",
    mapper: AFDDomainUpdateParametersMapper,
};
export const endpointName = {
    parameterPath: "endpointName",
    mapper: {
        serializedName: "endpointName",
        required: true,
        type: {
            name: "String",
        },
    },
};
export const endpoint = {
    parameterPath: "endpoint",
    mapper: AFDEndpointMapper,
};
export const endpointUpdateProperties = {
    parameterPath: "endpointUpdateProperties",
    mapper: AFDEndpointUpdateParametersMapper,
};
export const contents = {
    parameterPath: "contents",
    mapper: AfdPurgeParametersMapper,
};
export const customDomainProperties = {
    parameterPath: "customDomainProperties",
    mapper: ValidateCustomDomainInputMapper,
};
export const originGroupName = {
    parameterPath: "originGroupName",
    mapper: {
        serializedName: "originGroupName",
        required: true,
        type: {
            name: "String",
        },
    },
};
export const originGroup = {
    parameterPath: "originGroup",
    mapper: AFDOriginGroupMapper,
};
export const originGroupUpdateProperties = {
    parameterPath: "originGroupUpdateProperties",
    mapper: AFDOriginGroupUpdateParametersMapper,
};
export const originName = {
    parameterPath: "originName",
    mapper: {
        serializedName: "originName",
        required: true,
        type: {
            name: "String",
        },
    },
};
export const origin = {
    parameterPath: "origin",
    mapper: AFDOriginMapper,
};
export const originUpdateProperties = {
    parameterPath: "originUpdateProperties",
    mapper: AFDOriginUpdateParametersMapper,
};
export const routeName = {
    parameterPath: "routeName",
    mapper: {
        serializedName: "routeName",
        required: true,
        type: {
            name: "String",
        },
    },
};
export const route = {
    parameterPath: "route",
    mapper: RouteMapper,
};
export const routeUpdateProperties = {
    parameterPath: "routeUpdateProperties",
    mapper: RouteUpdateParametersMapper,
};
export const ruleSetName = {
    parameterPath: "ruleSetName",
    mapper: {
        serializedName: "ruleSetName",
        required: true,
        type: {
            name: "String",
        },
    },
};
export const ruleName = {
    parameterPath: "ruleName",
    mapper: {
        serializedName: "ruleName",
        required: true,
        type: {
            name: "String",
        },
    },
};
export const rule = {
    parameterPath: "rule",
    mapper: RuleMapper,
};
export const ruleUpdateProperties = {
    parameterPath: "ruleUpdateProperties",
    mapper: RuleUpdateParametersMapper,
};
export const securityPolicyName = {
    parameterPath: "securityPolicyName",
    mapper: {
        serializedName: "securityPolicyName",
        required: true,
        type: {
            name: "String",
        },
    },
};
export const securityPolicy = {
    parameterPath: "securityPolicy",
    mapper: SecurityPolicyMapper,
};
export const securityPolicyUpdateProperties = {
    parameterPath: "securityPolicyUpdateProperties",
    mapper: SecurityPolicyUpdateParametersMapper,
};
export const secretName = {
    parameterPath: "secretName",
    mapper: {
        serializedName: "secretName",
        required: true,
        type: {
            name: "String",
        },
    },
};
export const secret = {
    parameterPath: "secret",
    mapper: SecretMapper,
};
export const metrics = {
    parameterPath: "metrics",
    mapper: {
        serializedName: "metrics",
        required: true,
        type: {
            name: "Sequence",
            element: {
                type: {
                    name: "String",
                },
            },
        },
    },
    collectionFormat: "Multi",
};
export const dateTimeBegin = {
    parameterPath: "dateTimeBegin",
    mapper: {
        serializedName: "dateTimeBegin",
        required: true,
        type: {
            name: "DateTime",
        },
    },
};
export const dateTimeEnd = {
    parameterPath: "dateTimeEnd",
    mapper: {
        serializedName: "dateTimeEnd",
        required: true,
        type: {
            name: "DateTime",
        },
    },
};
export const granularity = {
    parameterPath: "granularity",
    mapper: {
        serializedName: "granularity",
        required: true,
        type: {
            name: "String",
        },
    },
};
export const groupBy = {
    parameterPath: ["options", "groupBy"],
    mapper: {
        serializedName: "groupBy",
        type: {
            name: "Sequence",
            element: {
                type: {
                    name: "String",
                },
            },
        },
    },
    collectionFormat: "Multi",
};
export const continents = {
    parameterPath: ["options", "continents"],
    mapper: {
        serializedName: "continents",
        type: {
            name: "Sequence",
            element: {
                type: {
                    name: "String",
                },
            },
        },
    },
    collectionFormat: "Multi",
};
export const countryOrRegions = {
    parameterPath: ["options", "countryOrRegions"],
    mapper: {
        serializedName: "countryOrRegions",
        type: {
            name: "Sequence",
            element: {
                type: {
                    name: "String",
                },
            },
        },
    },
    collectionFormat: "Multi",
};
export const customDomains = {
    parameterPath: "customDomains",
    mapper: {
        serializedName: "customDomains",
        required: true,
        type: {
            name: "Sequence",
            element: {
                type: {
                    name: "String",
                },
            },
        },
    },
    collectionFormat: "Multi",
};
export const protocols = {
    parameterPath: "protocols",
    mapper: {
        serializedName: "protocols",
        required: true,
        type: {
            name: "Sequence",
            element: {
                type: {
                    name: "String",
                },
            },
        },
    },
    collectionFormat: "Multi",
};
export const rankings = {
    parameterPath: "rankings",
    mapper: {
        serializedName: "rankings",
        required: true,
        type: {
            name: "Sequence",
            element: {
                type: {
                    name: "String",
                },
            },
        },
    },
    collectionFormat: "Multi",
};
export const metrics1 = {
    parameterPath: "metrics",
    mapper: {
        serializedName: "metrics",
        required: true,
        type: {
            name: "Sequence",
            element: {
                type: {
                    name: "String",
                },
            },
        },
    },
    collectionFormat: "Multi",
};
export const maxRanking = {
    parameterPath: "maxRanking",
    mapper: {
        serializedName: "maxRanking",
        required: true,
        type: {
            name: "Number",
        },
    },
};
export const customDomains1 = {
    parameterPath: ["options", "customDomains"],
    mapper: {
        serializedName: "customDomains",
        type: {
            name: "Sequence",
            element: {
                type: {
                    name: "String",
                },
            },
        },
    },
    collectionFormat: "Multi",
};
export const metrics2 = {
    parameterPath: "metrics",
    mapper: {
        serializedName: "metrics",
        required: true,
        type: {
            name: "Sequence",
            element: {
                type: {
                    name: "String",
                },
            },
        },
    },
    collectionFormat: "Multi",
};
export const granularity1 = {
    parameterPath: "granularity",
    mapper: {
        serializedName: "granularity",
        required: true,
        type: {
            name: "String",
        },
    },
};
export const actions = {
    parameterPath: ["options", "actions"],
    mapper: {
        serializedName: "actions",
        type: {
            name: "Sequence",
            element: {
                type: {
                    name: "String",
                },
            },
        },
    },
    collectionFormat: "Multi",
};
export const groupBy1 = {
    parameterPath: ["options", "groupBy"],
    mapper: {
        serializedName: "groupBy",
        type: {
            name: "Sequence",
            element: {
                type: {
                    name: "String",
                },
            },
        },
    },
    collectionFormat: "Multi",
};
export const ruleTypes = {
    parameterPath: ["options", "ruleTypes"],
    mapper: {
        serializedName: "ruleTypes",
        type: {
            name: "Sequence",
            element: {
                type: {
                    name: "String",
                },
            },
        },
    },
    collectionFormat: "Multi",
};
export const rankings1 = {
    parameterPath: "rankings",
    mapper: {
        serializedName: "rankings",
        required: true,
        type: {
            name: "Sequence",
            element: {
                type: {
                    name: "String",
                },
            },
        },
    },
    collectionFormat: "Multi",
};
export const profile = {
    parameterPath: "profile",
    mapper: ProfileMapper,
};
export const profileUpdateParameters = {
    parameterPath: "profileUpdateParameters",
    mapper: ProfileUpdateParametersMapper,
};
export const canMigrateParameters = {
    parameterPath: "canMigrateParameters",
    mapper: CanMigrateParametersMapper,
};
export const migrationParameters = {
    parameterPath: "migrationParameters",
    mapper: MigrationParametersMapper,
};
export const endpoint1 = {
    parameterPath: "endpoint",
    mapper: EndpointMapper,
};
export const endpointUpdateProperties1 = {
    parameterPath: "endpointUpdateProperties",
    mapper: EndpointUpdateParametersMapper,
};
export const contentFilePaths = {
    parameterPath: "contentFilePaths",
    mapper: PurgeParametersMapper,
};
export const contentFilePaths1 = {
    parameterPath: "contentFilePaths",
    mapper: LoadParametersMapper,
};
export const origin1 = {
    parameterPath: "origin",
    mapper: OriginMapper,
};
export const originUpdateProperties1 = {
    parameterPath: "originUpdateProperties",
    mapper: OriginUpdateParametersMapper,
};
export const originGroup1 = {
    parameterPath: "originGroup",
    mapper: OriginGroupMapper,
};
export const originGroupUpdateProperties1 = {
    parameterPath: "originGroupUpdateProperties",
    mapper: OriginGroupUpdateParametersMapper,
};
export const customDomainProperties1 = {
    parameterPath: "customDomainProperties",
    mapper: CustomDomainParametersMapper,
};
export const customDomainHttpsParameters = {
    parameterPath: ["options", "customDomainHttpsParameters"],
    mapper: CustomDomainHttpsParametersMapper,
};
export const resourceGroupName1 = {
    parameterPath: "resourceGroupName",
    mapper: {
        constraints: {
            Pattern: new RegExp("^[a-zA-Z0-9_\\-\\(\\)\\.]*[^\\.]$"),
            MaxLength: 80,
            MinLength: 1,
        },
        serializedName: "resourceGroupName",
        required: true,
        type: {
            name: "String",
        },
    },
};
export const policyName = {
    parameterPath: "policyName",
    mapper: {
        constraints: {
            MaxLength: 128,
        },
        serializedName: "policyName",
        required: true,
        type: {
            name: "String",
        },
    },
};
export const cdnWebApplicationFirewallPolicy = {
    parameterPath: "cdnWebApplicationFirewallPolicy",
    mapper: CdnWebApplicationFirewallPolicyMapper,
};
export const cdnWebApplicationFirewallPolicyPatchParameters = {
    parameterPath: "cdnWebApplicationFirewallPolicyPatchParameters",
    mapper: CdnWebApplicationFirewallPolicyPatchParametersMapper,
};
