// @azure/arm-policy/dist-esm/src/models/index.js
/*
 * Copyright (c) Microsoft Corporation.
 * Licensed under the MIT License.
 *
 * Code generated by Microsoft (R) AutoRest Code Generator.
 * Changes may cause incorrect behavior and will be lost if the code is regenerated.
 */
/** Known values of {@link AliasPathTokenType} that the service accepts. */
export var KnownAliasPathTokenType;
(function (KnownAliasPathTokenType) {
    /** The token type is not specified. */
    KnownAliasPathTokenType["NotSpecified"] = "NotSpecified";
    /** The token type can be anything. */
    KnownAliasPathTokenType["Any"] = "Any";
    /** The token type is string. */
    KnownAliasPathTokenType["String"] = "String";
    /** The token type is object. */
    KnownAliasPathTokenType["Object"] = "Object";
    /** The token type is array. */
    KnownAliasPathTokenType["Array"] = "Array";
    /** The token type is integer. */
    KnownAliasPathTokenType["Integer"] = "Integer";
    /** The token type is number. */
    KnownAliasPathTokenType["Number"] = "Number";
    /** The token type is boolean. */
    KnownAliasPathTokenType["Boolean"] = "Boolean";
})(KnownAliasPathTokenType || (KnownAliasPathTokenType = {}));
/** Known values of {@link AliasPathAttributes} that the service accepts. */
export var KnownAliasPathAttributes;
(function (KnownAliasPathAttributes) {
    /** The token that the alias path is referring to has no attributes. */
    KnownAliasPathAttributes["None"] = "None";
    /** The token that the alias path is referring to is modifiable by policies with 'modify' effect. */
    KnownAliasPathAttributes["Modifiable"] = "Modifiable";
})(KnownAliasPathAttributes || (KnownAliasPathAttributes = {}));
/** Known values of {@link EnforcementMode} that the service accepts. */
export var KnownEnforcementMode;
(function (KnownEnforcementMode) {
    /** The policy effect is enforced during resource creation or update. */
    KnownEnforcementMode["Default"] = "Default";
    /** The policy effect is not enforced during resource creation or update. */
    KnownEnforcementMode["DoNotEnforce"] = "DoNotEnforce";
})(KnownEnforcementMode || (KnownEnforcementMode = {}));
/** Known values of {@link CreatedByType} that the service accepts. */
export var KnownCreatedByType;
(function (KnownCreatedByType) {
    /** User */
    KnownCreatedByType["User"] = "User";
    /** Application */
    KnownCreatedByType["Application"] = "Application";
    /** ManagedIdentity */
    KnownCreatedByType["ManagedIdentity"] = "ManagedIdentity";
    /** Key */
    KnownCreatedByType["Key"] = "Key";
})(KnownCreatedByType || (KnownCreatedByType = {}));
/** Known values of {@link PolicyType} that the service accepts. */
export var KnownPolicyType;
(function (KnownPolicyType) {
    /** NotSpecified */
    KnownPolicyType["NotSpecified"] = "NotSpecified";
    /** BuiltIn */
    KnownPolicyType["BuiltIn"] = "BuiltIn";
    /** Custom */
    KnownPolicyType["Custom"] = "Custom";
    /** Static */
    KnownPolicyType["Static"] = "Static";
})(KnownPolicyType || (KnownPolicyType = {}));
/** Known values of {@link ParameterType} that the service accepts. */
export var KnownParameterType;
(function (KnownParameterType) {
    /** String */
    KnownParameterType["String"] = "String";
    /** Array */
    KnownParameterType["Array"] = "Array";
    /** Object */
    KnownParameterType["Object"] = "Object";
    /** Boolean */
    KnownParameterType["Boolean"] = "Boolean";
    /** Integer */
    KnownParameterType["Integer"] = "Integer";
    /** Float */
    KnownParameterType["Float"] = "Float";
    /** DateTime */
    KnownParameterType["DateTime"] = "DateTime";
})(KnownParameterType || (KnownParameterType = {}));
/** Known values of {@link ExemptionCategory} that the service accepts. */
export var KnownExemptionCategory;
(function (KnownExemptionCategory) {
    /** This category of exemptions usually means the scope is not applicable for the policy. */
    KnownExemptionCategory["Waiver"] = "Waiver";
    /** This category of exemptions usually means the mitigation actions have been applied to the scope. */
    KnownExemptionCategory["Mitigated"] = "Mitigated";
})(KnownExemptionCategory || (KnownExemptionCategory = {}));
