// @aws-sdk/client-mediastore/dist-cjs/index.js
"use strict";
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __name = (target, value) => __defProp(target, "name", { value, configurable: true });
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// src/index.ts
var src_exports = {};
__export(src_exports, {
  ContainerInUseException: () => ContainerInUseException,
  ContainerLevelMetrics: () => ContainerLevelMetrics,
  ContainerNotFoundException: () => ContainerNotFoundException,
  ContainerStatus: () => ContainerStatus,
  CorsPolicyNotFoundException: () => CorsPolicyNotFoundException,
  CreateContainerCommand: () => CreateContainerCommand,
  DeleteContainerCommand: () => DeleteContainerCommand,
  DeleteContainerPolicyCommand: () => DeleteContainerPolicyCommand,
  DeleteCorsPolicyCommand: () => DeleteCorsPolicyCommand,
  DeleteLifecyclePolicyCommand: () => DeleteLifecyclePolicyCommand,
  DeleteMetricPolicyCommand: () => DeleteMetricPolicyCommand,
  DescribeContainerCommand: () => DescribeContainerCommand,
  GetContainerPolicyCommand: () => GetContainerPolicyCommand,
  GetCorsPolicyCommand: () => GetCorsPolicyCommand,
  GetLifecyclePolicyCommand: () => GetLifecyclePolicyCommand,
  GetMetricPolicyCommand: () => GetMetricPolicyCommand,
  InternalServerError: () => InternalServerError,
  LimitExceededException: () => LimitExceededException,
  ListContainersCommand: () => ListContainersCommand,
  ListTagsForResourceCommand: () => ListTagsForResourceCommand,
  MediaStore: () => MediaStore,
  MediaStoreClient: () => MediaStoreClient,
  MediaStoreServiceException: () => MediaStoreServiceException,
  MethodName: () => MethodName,
  PolicyNotFoundException: () => PolicyNotFoundException,
  PutContainerPolicyCommand: () => PutContainerPolicyCommand,
  PutCorsPolicyCommand: () => PutCorsPolicyCommand,
  PutLifecyclePolicyCommand: () => PutLifecyclePolicyCommand,
  PutMetricPolicyCommand: () => PutMetricPolicyCommand,
  StartAccessLoggingCommand: () => StartAccessLoggingCommand,
  StopAccessLoggingCommand: () => StopAccessLoggingCommand,
  TagResourceCommand: () => TagResourceCommand,
  UntagResourceCommand: () => UntagResourceCommand,
  __Client: () => import_smithy_client.Client,
  paginateListContainers: () => paginateListContainers
});
module.exports = __toCommonJS(src_exports);

// src/MediaStoreClient.ts
var import_middleware_host_header = require("@aws-sdk/middleware-host-header");
var import_middleware_logger = require("@aws-sdk/middleware-logger");
var import_middleware_recursion_detection = require("@aws-sdk/middleware-recursion-detection");
var import_middleware_user_agent = require("@aws-sdk/middleware-user-agent");
var import_config_resolver = require("@smithy/config-resolver");
var import_core = require("@smithy/core");
var import_middleware_content_length = require("@smithy/middleware-content-length");
var import_middleware_endpoint = require("@smithy/middleware-endpoint");
var import_middleware_retry = require("@smithy/middleware-retry");

var import_httpAuthSchemeProvider = require("./auth/httpAuthSchemeProvider");

// src/endpoint/EndpointParameters.ts
var resolveClientEndpointParameters = /* @__PURE__ */ __name((options) => {
  return {
    ...options,
    useDualstackEndpoint: options.useDualstackEndpoint ?? false,
    useFipsEndpoint: options.useFipsEndpoint ?? false,
    defaultSigningName: "mediastore"
  };
}, "resolveClientEndpointParameters");
var commonParams = {
  UseFIPS: { type: "builtInParams", name: "useFipsEndpoint" },
  Endpoint: { type: "builtInParams", name: "endpoint" },
  Region: { type: "builtInParams", name: "region" },
  UseDualStack: { type: "builtInParams", name: "useDualstackEndpoint" }
};

// src/MediaStoreClient.ts
var import_runtimeConfig = require("././runtimeConfig");

// src/runtimeExtensions.ts
var import_region_config_resolver = require("@aws-sdk/region-config-resolver");
var import_protocol_http = require("@smithy/protocol-http");
var import_smithy_client = require("@smithy/smithy-client");

// src/auth/httpAuthExtensionConfiguration.ts
var getHttpAuthExtensionConfiguration = /* @__PURE__ */ __name((runtimeConfig) => {
  const _httpAuthSchemes = runtimeConfig.httpAuthSchemes;
  let _httpAuthSchemeProvider = runtimeConfig.httpAuthSchemeProvider;
  let _credentials = runtimeConfig.credentials;
  return {
    setHttpAuthScheme(httpAuthScheme) {
      const index = _httpAuthSchemes.findIndex((scheme) => scheme.schemeId === httpAuthScheme.schemeId);
      if (index === -1) {
        _httpAuthSchemes.push(httpAuthScheme);
      } else {
        _httpAuthSchemes.splice(index, 1, httpAuthScheme);
      }
    },
    httpAuthSchemes() {
      return _httpAuthSchemes;
    },
    setHttpAuthSchemeProvider(httpAuthSchemeProvider) {
      _httpAuthSchemeProvider = httpAuthSchemeProvider;
    },
    httpAuthSchemeProvider() {
      return _httpAuthSchemeProvider;
    },
    setCredentials(credentials) {
      _credentials = credentials;
    },
    credentials() {
      return _credentials;
    }
  };
}, "getHttpAuthExtensionConfiguration");
var resolveHttpAuthRuntimeConfig = /* @__PURE__ */ __name((config) => {
  return {
    httpAuthSchemes: config.httpAuthSchemes(),
    httpAuthSchemeProvider: config.httpAuthSchemeProvider(),
    credentials: config.credentials()
  };
}, "resolveHttpAuthRuntimeConfig");

// src/runtimeExtensions.ts
var asPartial = /* @__PURE__ */ __name((t) => t, "asPartial");
var resolveRuntimeExtensions = /* @__PURE__ */ __name((runtimeConfig, extensions) => {
  const extensionConfiguration = {
    ...asPartial((0, import_region_config_resolver.getAwsRegionExtensionConfiguration)(runtimeConfig)),
    ...asPartial((0, import_smithy_client.getDefaultExtensionConfiguration)(runtimeConfig)),
    ...asPartial((0, import_protocol_http.getHttpHandlerExtensionConfiguration)(runtimeConfig)),
    ...asPartial(getHttpAuthExtensionConfiguration(runtimeConfig))
  };
  extensions.forEach((extension) => extension.configure(extensionConfiguration));
  return {
    ...runtimeConfig,
    ...(0, import_region_config_resolver.resolveAwsRegionExtensionConfiguration)(extensionConfiguration),
    ...(0, import_smithy_client.resolveDefaultRuntimeConfig)(extensionConfiguration),
    ...(0, import_protocol_http.resolveHttpHandlerRuntimeConfig)(extensionConfiguration),
    ...resolveHttpAuthRuntimeConfig(extensionConfiguration)
  };
}, "resolveRuntimeExtensions");

// src/MediaStoreClient.ts
var _MediaStoreClient = class _MediaStoreClient extends import_smithy_client.Client {
  constructor(...[configuration]) {
    const _config_0 = (0, import_runtimeConfig.getRuntimeConfig)(configuration || {});
    const _config_1 = resolveClientEndpointParameters(_config_0);
    const _config_2 = (0, import_config_resolver.resolveRegionConfig)(_config_1);
    const _config_3 = (0, import_middleware_endpoint.resolveEndpointConfig)(_config_2);
    const _config_4 = (0, import_middleware_retry.resolveRetryConfig)(_config_3);
    const _config_5 = (0, import_middleware_host_header.resolveHostHeaderConfig)(_config_4);
    const _config_6 = (0, import_middleware_user_agent.resolveUserAgentConfig)(_config_5);
    const _config_7 = (0, import_httpAuthSchemeProvider.resolveHttpAuthSchemeConfig)(_config_6);
    const _config_8 = resolveRuntimeExtensions(_config_7, (configuration == null ? void 0 : configuration.extensions) || []);
    super(_config_8);
    this.config = _config_8;
    this.middlewareStack.use((0, import_middleware_retry.getRetryPlugin)(this.config));
    this.middlewareStack.use((0, import_middleware_content_length.getContentLengthPlugin)(this.config));
    this.middlewareStack.use((0, import_middleware_host_header.getHostHeaderPlugin)(this.config));
    this.middlewareStack.use((0, import_middleware_logger.getLoggerPlugin)(this.config));
    this.middlewareStack.use((0, import_middleware_recursion_detection.getRecursionDetectionPlugin)(this.config));
    this.middlewareStack.use((0, import_middleware_user_agent.getUserAgentPlugin)(this.config));
    this.middlewareStack.use(
      (0, import_core.getHttpAuthSchemeEndpointRuleSetPlugin)(this.config, {
        httpAuthSchemeParametersProvider: this.getDefaultHttpAuthSchemeParametersProvider(),
        identityProviderConfigProvider: this.getIdentityProviderConfigProvider()
      })
    );
    this.middlewareStack.use((0, import_core.getHttpSigningPlugin)(this.config));
  }
  /**
   * Destroy underlying resources, like sockets. It's usually not necessary to do this.
   * However in Node.js, it's best to explicitly shut down the client's agent when it is no longer needed.
   * Otherwise, sockets might stay open for quite a long time before the server terminates them.
   */
  destroy() {
    super.destroy();
  }
  getDefaultHttpAuthSchemeParametersProvider() {
    return import_httpAuthSchemeProvider.defaultMediaStoreHttpAuthSchemeParametersProvider;
  }
  getIdentityProviderConfigProvider() {
    return async (config) => new import_core.DefaultIdentityProviderConfig({
      "aws.auth#sigv4": config.credentials
    });
  }
};
__name(_MediaStoreClient, "MediaStoreClient");
var MediaStoreClient = _MediaStoreClient;

// src/MediaStore.ts


// src/commands/CreateContainerCommand.ts

var import_middleware_serde = require("@smithy/middleware-serde");

var import_types = require("@smithy/types");

// src/protocols/Aws_json1_1.ts
var import_core2 = require("@aws-sdk/core");



// src/models/MediaStoreServiceException.ts

var _MediaStoreServiceException = class _MediaStoreServiceException extends import_smithy_client.ServiceException {
  /**
   * @internal
   */
  constructor(options) {
    super(options);
    Object.setPrototypeOf(this, _MediaStoreServiceException.prototype);
  }
};
__name(_MediaStoreServiceException, "MediaStoreServiceException");
var MediaStoreServiceException = _MediaStoreServiceException;

// src/models/models_0.ts
var MethodName = {
  DELETE: "DELETE",
  GET: "GET",
  HEAD: "HEAD",
  PUT: "PUT"
};
var ContainerStatus = {
  ACTIVE: "ACTIVE",
  CREATING: "CREATING",
  DELETING: "DELETING"
};
var _ContainerInUseException = class _ContainerInUseException extends MediaStoreServiceException {
  /**
   * @internal
   */
  constructor(opts) {
    super({
      name: "ContainerInUseException",
      $fault: "client",
      ...opts
    });
    this.name = "ContainerInUseException";
    this.$fault = "client";
    Object.setPrototypeOf(this, _ContainerInUseException.prototype);
    this.Message = opts.Message;
  }
};
__name(_ContainerInUseException, "ContainerInUseException");
var ContainerInUseException = _ContainerInUseException;
var ContainerLevelMetrics = {
  DISABLED: "DISABLED",
  ENABLED: "ENABLED"
};
var _ContainerNotFoundException = class _ContainerNotFoundException extends MediaStoreServiceException {
  /**
   * @internal
   */
  constructor(opts) {
    super({
      name: "ContainerNotFoundException",
      $fault: "client",
      ...opts
    });
    this.name = "ContainerNotFoundException";
    this.$fault = "client";
    Object.setPrototypeOf(this, _ContainerNotFoundException.prototype);
    this.Message = opts.Message;
  }
};
__name(_ContainerNotFoundException, "ContainerNotFoundException");
var ContainerNotFoundException = _ContainerNotFoundException;
var _CorsPolicyNotFoundException = class _CorsPolicyNotFoundException extends MediaStoreServiceException {
  /**
   * @internal
   */
  constructor(opts) {
    super({
      name: "CorsPolicyNotFoundException",
      $fault: "client",
      ...opts
    });
    this.name = "CorsPolicyNotFoundException";
    this.$fault = "client";
    Object.setPrototypeOf(this, _CorsPolicyNotFoundException.prototype);
    this.Message = opts.Message;
  }
};
__name(_CorsPolicyNotFoundException, "CorsPolicyNotFoundException");
var CorsPolicyNotFoundException = _CorsPolicyNotFoundException;
var _InternalServerError = class _InternalServerError extends MediaStoreServiceException {
  /**
   * @internal
   */
  constructor(opts) {
    super({
      name: "InternalServerError",
      $fault: "server",
      ...opts
    });
    this.name = "InternalServerError";
    this.$fault = "server";
    Object.setPrototypeOf(this, _InternalServerError.prototype);
    this.Message = opts.Message;
  }
};
__name(_InternalServerError, "InternalServerError");
var InternalServerError = _InternalServerError;
var _LimitExceededException = class _LimitExceededException extends MediaStoreServiceException {
  /**
   * @internal
   */
  constructor(opts) {
    super({
      name: "LimitExceededException",
      $fault: "client",
      ...opts
    });
    this.name = "LimitExceededException";
    this.$fault = "client";
    Object.setPrototypeOf(this, _LimitExceededException.prototype);
    this.Message = opts.Message;
  }
};
__name(_LimitExceededException, "LimitExceededException");
var LimitExceededException = _LimitExceededException;
var _PolicyNotFoundException = class _PolicyNotFoundException extends MediaStoreServiceException {
  /**
   * @internal
   */
  constructor(opts) {
    super({
      name: "PolicyNotFoundException",
      $fault: "client",
      ...opts
    });
    this.name = "PolicyNotFoundException";
    this.$fault = "client";
    Object.setPrototypeOf(this, _PolicyNotFoundException.prototype);
    this.Message = opts.Message;
  }
};
__name(_PolicyNotFoundException, "PolicyNotFoundException");
var PolicyNotFoundException = _PolicyNotFoundException;

// src/protocols/Aws_json1_1.ts
var se_CreateContainerCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("CreateContainer");
  let body;
  body = JSON.stringify((0, import_smithy_client._json)(input));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_CreateContainerCommand");
var se_DeleteContainerCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("DeleteContainer");
  let body;
  body = JSON.stringify((0, import_smithy_client._json)(input));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_DeleteContainerCommand");
var se_DeleteContainerPolicyCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("DeleteContainerPolicy");
  let body;
  body = JSON.stringify((0, import_smithy_client._json)(input));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_DeleteContainerPolicyCommand");
var se_DeleteCorsPolicyCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("DeleteCorsPolicy");
  let body;
  body = JSON.stringify((0, import_smithy_client._json)(input));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_DeleteCorsPolicyCommand");
var se_DeleteLifecyclePolicyCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("DeleteLifecyclePolicy");
  let body;
  body = JSON.stringify((0, import_smithy_client._json)(input));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_DeleteLifecyclePolicyCommand");
var se_DeleteMetricPolicyCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("DeleteMetricPolicy");
  let body;
  body = JSON.stringify((0, import_smithy_client._json)(input));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_DeleteMetricPolicyCommand");
var se_DescribeContainerCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("DescribeContainer");
  let body;
  body = JSON.stringify((0, import_smithy_client._json)(input));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_DescribeContainerCommand");
var se_GetContainerPolicyCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("GetContainerPolicy");
  let body;
  body = JSON.stringify((0, import_smithy_client._json)(input));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_GetContainerPolicyCommand");
var se_GetCorsPolicyCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("GetCorsPolicy");
  let body;
  body = JSON.stringify((0, import_smithy_client._json)(input));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_GetCorsPolicyCommand");
var se_GetLifecyclePolicyCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("GetLifecyclePolicy");
  let body;
  body = JSON.stringify((0, import_smithy_client._json)(input));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_GetLifecyclePolicyCommand");
var se_GetMetricPolicyCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("GetMetricPolicy");
  let body;
  body = JSON.stringify((0, import_smithy_client._json)(input));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_GetMetricPolicyCommand");
var se_ListContainersCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("ListContainers");
  let body;
  body = JSON.stringify((0, import_smithy_client._json)(input));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_ListContainersCommand");
var se_ListTagsForResourceCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("ListTagsForResource");
  let body;
  body = JSON.stringify((0, import_smithy_client._json)(input));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_ListTagsForResourceCommand");
var se_PutContainerPolicyCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("PutContainerPolicy");
  let body;
  body = JSON.stringify((0, import_smithy_client._json)(input));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_PutContainerPolicyCommand");
var se_PutCorsPolicyCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("PutCorsPolicy");
  let body;
  body = JSON.stringify((0, import_smithy_client._json)(input));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_PutCorsPolicyCommand");
var se_PutLifecyclePolicyCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("PutLifecyclePolicy");
  let body;
  body = JSON.stringify((0, import_smithy_client._json)(input));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_PutLifecyclePolicyCommand");
var se_PutMetricPolicyCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("PutMetricPolicy");
  let body;
  body = JSON.stringify((0, import_smithy_client._json)(input));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_PutMetricPolicyCommand");
var se_StartAccessLoggingCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("StartAccessLogging");
  let body;
  body = JSON.stringify((0, import_smithy_client._json)(input));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_StartAccessLoggingCommand");
var se_StopAccessLoggingCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("StopAccessLogging");
  let body;
  body = JSON.stringify((0, import_smithy_client._json)(input));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_StopAccessLoggingCommand");
var se_TagResourceCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("TagResource");
  let body;
  body = JSON.stringify((0, import_smithy_client._json)(input));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_TagResourceCommand");
var se_UntagResourceCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("UntagResource");
  let body;
  body = JSON.stringify((0, import_smithy_client._json)(input));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_UntagResourceCommand");
var de_CreateContainerCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const data = await (0, import_core2.parseJsonBody)(output.body, context);
  let contents = {};
  contents = de_CreateContainerOutput(data, context);
  const response = {
    $metadata: deserializeMetadata(output),
    ...contents
  };
  return response;
}, "de_CreateContainerCommand");
var de_DeleteContainerCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const data = await (0, import_core2.parseJsonBody)(output.body, context);
  let contents = {};
  contents = (0, import_smithy_client._json)(data);
  const response = {
    $metadata: deserializeMetadata(output),
    ...contents
  };
  return response;
}, "de_DeleteContainerCommand");
var de_DeleteContainerPolicyCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const data = await (0, import_core2.parseJsonBody)(output.body, context);
  let contents = {};
  contents = (0, import_smithy_client._json)(data);
  const response = {
    $metadata: deserializeMetadata(output),
    ...contents
  };
  return response;
}, "de_DeleteContainerPolicyCommand");
var de_DeleteCorsPolicyCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const data = await (0, import_core2.parseJsonBody)(output.body, context);
  let contents = {};
  contents = (0, import_smithy_client._json)(data);
  const response = {
    $metadata: deserializeMetadata(output),
    ...contents
  };
  return response;
}, "de_DeleteCorsPolicyCommand");
var de_DeleteLifecyclePolicyCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const data = await (0, import_core2.parseJsonBody)(output.body, context);
  let contents = {};
  contents = (0, import_smithy_client._json)(data);
  const response = {
    $metadata: deserializeMetadata(output),
    ...contents
  };
  return response;
}, "de_DeleteLifecyclePolicyCommand");
var de_DeleteMetricPolicyCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const data = await (0, import_core2.parseJsonBody)(output.body, context);
  let contents = {};
  contents = (0, import_smithy_client._json)(data);
  const response = {
    $metadata: deserializeMetadata(output),
    ...contents
  };
  return response;
}, "de_DeleteMetricPolicyCommand");
var de_DescribeContainerCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const data = await (0, import_core2.parseJsonBody)(output.body, context);
  let contents = {};
  contents = de_DescribeContainerOutput(data, context);
  const response = {
    $metadata: deserializeMetadata(output),
    ...contents
  };
  return response;
}, "de_DescribeContainerCommand");
var de_GetContainerPolicyCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const data = await (0, import_core2.parseJsonBody)(output.body, context);
  let contents = {};
  contents = (0, import_smithy_client._json)(data);
  const response = {
    $metadata: deserializeMetadata(output),
    ...contents
  };
  return response;
}, "de_GetContainerPolicyCommand");
var de_GetCorsPolicyCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const data = await (0, import_core2.parseJsonBody)(output.body, context);
  let contents = {};
  contents = (0, import_smithy_client._json)(data);
  const response = {
    $metadata: deserializeMetadata(output),
    ...contents
  };
  return response;
}, "de_GetCorsPolicyCommand");
var de_GetLifecyclePolicyCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const data = await (0, import_core2.parseJsonBody)(output.body, context);
  let contents = {};
  contents = (0, import_smithy_client._json)(data);
  const response = {
    $metadata: deserializeMetadata(output),
    ...contents
  };
  return response;
}, "de_GetLifecyclePolicyCommand");
var de_GetMetricPolicyCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const data = await (0, import_core2.parseJsonBody)(output.body, context);
  let contents = {};
  contents = (0, import_smithy_client._json)(data);
  const response = {
    $metadata: deserializeMetadata(output),
    ...contents
  };
  return response;
}, "de_GetMetricPolicyCommand");
var de_ListContainersCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const data = await (0, import_core2.parseJsonBody)(output.body, context);
  let contents = {};
  contents = de_ListContainersOutput(data, context);
  const response = {
    $metadata: deserializeMetadata(output),
    ...contents
  };
  return response;
}, "de_ListContainersCommand");
var de_ListTagsForResourceCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const data = await (0, import_core2.parseJsonBody)(output.body, context);
  let contents = {};
  contents = (0, import_smithy_client._json)(data);
  const response = {
    $metadata: deserializeMetadata(output),
    ...contents
  };
  return response;
}, "de_ListTagsForResourceCommand");
var de_PutContainerPolicyCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const data = await (0, import_core2.parseJsonBody)(output.body, context);
  let contents = {};
  contents = (0, import_smithy_client._json)(data);
  const response = {
    $metadata: deserializeMetadata(output),
    ...contents
  };
  return response;
}, "de_PutContainerPolicyCommand");
var de_PutCorsPolicyCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const data = await (0, import_core2.parseJsonBody)(output.body, context);
  let contents = {};
  contents = (0, import_smithy_client._json)(data);
  const response = {
    $metadata: deserializeMetadata(output),
    ...contents
  };
  return response;
}, "de_PutCorsPolicyCommand");
var de_PutLifecyclePolicyCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const data = await (0, import_core2.parseJsonBody)(output.body, context);
  let contents = {};
  contents = (0, import_smithy_client._json)(data);
  const response = {
    $metadata: deserializeMetadata(output),
    ...contents
  };
  return response;
}, "de_PutLifecyclePolicyCommand");
var de_PutMetricPolicyCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const data = await (0, import_core2.parseJsonBody)(output.body, context);
  let contents = {};
  contents = (0, import_smithy_client._json)(data);
  const response = {
    $metadata: deserializeMetadata(output),
    ...contents
  };
  return response;
}, "de_PutMetricPolicyCommand");
var de_StartAccessLoggingCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const data = await (0, import_core2.parseJsonBody)(output.body, context);
  let contents = {};
  contents = (0, import_smithy_client._json)(data);
  const response = {
    $metadata: deserializeMetadata(output),
    ...contents
  };
  return response;
}, "de_StartAccessLoggingCommand");
var de_StopAccessLoggingCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const data = await (0, import_core2.parseJsonBody)(output.body, context);
  let contents = {};
  contents = (0, import_smithy_client._json)(data);
  const response = {
    $metadata: deserializeMetadata(output),
    ...contents
  };
  return response;
}, "de_StopAccessLoggingCommand");
var de_TagResourceCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const data = await (0, import_core2.parseJsonBody)(output.body, context);
  let contents = {};
  contents = (0, import_smithy_client._json)(data);
  const response = {
    $metadata: deserializeMetadata(output),
    ...contents
  };
  return response;
}, "de_TagResourceCommand");
var de_UntagResourceCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const data = await (0, import_core2.parseJsonBody)(output.body, context);
  let contents = {};
  contents = (0, import_smithy_client._json)(data);
  const response = {
    $metadata: deserializeMetadata(output),
    ...contents
  };
  return response;
}, "de_UntagResourceCommand");
var de_CommandError = /* @__PURE__ */ __name(async (output, context) => {
  const parsedOutput = {
    ...output,
    body: await (0, import_core2.parseJsonErrorBody)(output.body, context)
  };
  const errorCode = (0, import_core2.loadRestJsonErrorCode)(output, parsedOutput.body);
  switch (errorCode) {
    case "ContainerInUseException":
    case "com.amazonaws.mediastore#ContainerInUseException":
      throw await de_ContainerInUseExceptionRes(parsedOutput, context);
    case "InternalServerError":
    case "com.amazonaws.mediastore#InternalServerError":
      throw await de_InternalServerErrorRes(parsedOutput, context);
    case "LimitExceededException":
    case "com.amazonaws.mediastore#LimitExceededException":
      throw await de_LimitExceededExceptionRes(parsedOutput, context);
    case "ContainerNotFoundException":
    case "com.amazonaws.mediastore#ContainerNotFoundException":
      throw await de_ContainerNotFoundExceptionRes(parsedOutput, context);
    case "PolicyNotFoundException":
    case "com.amazonaws.mediastore#PolicyNotFoundException":
      throw await de_PolicyNotFoundExceptionRes(parsedOutput, context);
    case "CorsPolicyNotFoundException":
    case "com.amazonaws.mediastore#CorsPolicyNotFoundException":
      throw await de_CorsPolicyNotFoundExceptionRes(parsedOutput, context);
    default:
      const parsedBody = parsedOutput.body;
      return throwDefaultError({
        output,
        parsedBody,
        errorCode
      });
  }
}, "de_CommandError");
var de_ContainerInUseExceptionRes = /* @__PURE__ */ __name(async (parsedOutput, context) => {
  const body = parsedOutput.body;
  const deserialized = (0, import_smithy_client._json)(body);
  const exception = new ContainerInUseException({
    $metadata: deserializeMetadata(parsedOutput),
    ...deserialized
  });
  return (0, import_smithy_client.decorateServiceException)(exception, body);
}, "de_ContainerInUseExceptionRes");
var de_ContainerNotFoundExceptionRes = /* @__PURE__ */ __name(async (parsedOutput, context) => {
  const body = parsedOutput.body;
  const deserialized = (0, import_smithy_client._json)(body);
  const exception = new ContainerNotFoundException({
    $metadata: deserializeMetadata(parsedOutput),
    ...deserialized
  });
  return (0, import_smithy_client.decorateServiceException)(exception, body);
}, "de_ContainerNotFoundExceptionRes");
var de_CorsPolicyNotFoundExceptionRes = /* @__PURE__ */ __name(async (parsedOutput, context) => {
  const body = parsedOutput.body;
  const deserialized = (0, import_smithy_client._json)(body);
  const exception = new CorsPolicyNotFoundException({
    $metadata: deserializeMetadata(parsedOutput),
    ...deserialized
  });
  return (0, import_smithy_client.decorateServiceException)(exception, body);
}, "de_CorsPolicyNotFoundExceptionRes");
var de_InternalServerErrorRes = /* @__PURE__ */ __name(async (parsedOutput, context) => {
  const body = parsedOutput.body;
  const deserialized = (0, import_smithy_client._json)(body);
  const exception = new InternalServerError({
    $metadata: deserializeMetadata(parsedOutput),
    ...deserialized
  });
  return (0, import_smithy_client.decorateServiceException)(exception, body);
}, "de_InternalServerErrorRes");
var de_LimitExceededExceptionRes = /* @__PURE__ */ __name(async (parsedOutput, context) => {
  const body = parsedOutput.body;
  const deserialized = (0, import_smithy_client._json)(body);
  const exception = new LimitExceededException({
    $metadata: deserializeMetadata(parsedOutput),
    ...deserialized
  });
  return (0, import_smithy_client.decorateServiceException)(exception, body);
}, "de_LimitExceededExceptionRes");
var de_PolicyNotFoundExceptionRes = /* @__PURE__ */ __name(async (parsedOutput, context) => {
  const body = parsedOutput.body;
  const deserialized = (0, import_smithy_client._json)(body);
  const exception = new PolicyNotFoundException({
    $metadata: deserializeMetadata(parsedOutput),
    ...deserialized
  });
  return (0, import_smithy_client.decorateServiceException)(exception, body);
}, "de_PolicyNotFoundExceptionRes");
var de_Container = /* @__PURE__ */ __name((output, context) => {
  return (0, import_smithy_client.take)(output, {
    ARN: import_smithy_client.expectString,
    AccessLoggingEnabled: import_smithy_client.expectBoolean,
    CreationTime: (_) => (0, import_smithy_client.expectNonNull)((0, import_smithy_client.parseEpochTimestamp)((0, import_smithy_client.expectNumber)(_))),
    Endpoint: import_smithy_client.expectString,
    Name: import_smithy_client.expectString,
    Status: import_smithy_client.expectString
  });
}, "de_Container");
var de_ContainerList = /* @__PURE__ */ __name((output, context) => {
  const retVal = (output || []).filter((e) => e != null).map((entry) => {
    return de_Container(entry, context);
  });
  return retVal;
}, "de_ContainerList");
var de_CreateContainerOutput = /* @__PURE__ */ __name((output, context) => {
  return (0, import_smithy_client.take)(output, {
    Container: (_) => de_Container(_, context)
  });
}, "de_CreateContainerOutput");
var de_DescribeContainerOutput = /* @__PURE__ */ __name((output, context) => {
  return (0, import_smithy_client.take)(output, {
    Container: (_) => de_Container(_, context)
  });
}, "de_DescribeContainerOutput");
var de_ListContainersOutput = /* @__PURE__ */ __name((output, context) => {
  return (0, import_smithy_client.take)(output, {
    Containers: (_) => de_ContainerList(_, context),
    NextToken: import_smithy_client.expectString
  });
}, "de_ListContainersOutput");
var deserializeMetadata = /* @__PURE__ */ __name((output) => ({
  httpStatusCode: output.statusCode,
  requestId: output.headers["x-amzn-requestid"] ?? output.headers["x-amzn-request-id"] ?? output.headers["x-amz-request-id"],
  extendedRequestId: output.headers["x-amz-id-2"],
  cfId: output.headers["x-amz-cf-id"]
}), "deserializeMetadata");
var throwDefaultError = (0, import_smithy_client.withBaseException)(MediaStoreServiceException);
var buildHttpRpcRequest = /* @__PURE__ */ __name(async (context, headers, path, resolvedHostname, body) => {
  const { hostname, protocol = "https", port, path: basePath } = await context.endpoint();
  const contents = {
    protocol,
    hostname,
    port,
    method: "POST",
    path: basePath.endsWith("/") ? basePath.slice(0, -1) + path : basePath + path,
    headers
  };
  if (resolvedHostname !== void 0) {
    contents.hostname = resolvedHostname;
  }
  if (body !== void 0) {
    contents.body = body;
  }
  return new import_protocol_http.HttpRequest(contents);
}, "buildHttpRpcRequest");
function sharedHeaders(operation) {
  return {
    "content-type": "application/x-amz-json-1.1",
    "x-amz-target": `MediaStore_20170901.${operation}`
  };
}
__name(sharedHeaders, "sharedHeaders");

// src/commands/CreateContainerCommand.ts
var _CreateContainerCommand = class _CreateContainerCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("MediaStore_20170901", "CreateContainer", {}).n("MediaStoreClient", "CreateContainerCommand").f(void 0, void 0).ser(se_CreateContainerCommand).de(de_CreateContainerCommand).build() {
};
__name(_CreateContainerCommand, "CreateContainerCommand");
var CreateContainerCommand = _CreateContainerCommand;

// src/commands/DeleteContainerCommand.ts




var _DeleteContainerCommand = class _DeleteContainerCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("MediaStore_20170901", "DeleteContainer", {}).n("MediaStoreClient", "DeleteContainerCommand").f(void 0, void 0).ser(se_DeleteContainerCommand).de(de_DeleteContainerCommand).build() {
};
__name(_DeleteContainerCommand, "DeleteContainerCommand");
var DeleteContainerCommand = _DeleteContainerCommand;

// src/commands/DeleteContainerPolicyCommand.ts




var _DeleteContainerPolicyCommand = class _DeleteContainerPolicyCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("MediaStore_20170901", "DeleteContainerPolicy", {}).n("MediaStoreClient", "DeleteContainerPolicyCommand").f(void 0, void 0).ser(se_DeleteContainerPolicyCommand).de(de_DeleteContainerPolicyCommand).build() {
};
__name(_DeleteContainerPolicyCommand, "DeleteContainerPolicyCommand");
var DeleteContainerPolicyCommand = _DeleteContainerPolicyCommand;

// src/commands/DeleteCorsPolicyCommand.ts




var _DeleteCorsPolicyCommand = class _DeleteCorsPolicyCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("MediaStore_20170901", "DeleteCorsPolicy", {}).n("MediaStoreClient", "DeleteCorsPolicyCommand").f(void 0, void 0).ser(se_DeleteCorsPolicyCommand).de(de_DeleteCorsPolicyCommand).build() {
};
__name(_DeleteCorsPolicyCommand, "DeleteCorsPolicyCommand");
var DeleteCorsPolicyCommand = _DeleteCorsPolicyCommand;

// src/commands/DeleteLifecyclePolicyCommand.ts




var _DeleteLifecyclePolicyCommand = class _DeleteLifecyclePolicyCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("MediaStore_20170901", "DeleteLifecyclePolicy", {}).n("MediaStoreClient", "DeleteLifecyclePolicyCommand").f(void 0, void 0).ser(se_DeleteLifecyclePolicyCommand).de(de_DeleteLifecyclePolicyCommand).build() {
};
__name(_DeleteLifecyclePolicyCommand, "DeleteLifecyclePolicyCommand");
var DeleteLifecyclePolicyCommand = _DeleteLifecyclePolicyCommand;

// src/commands/DeleteMetricPolicyCommand.ts




var _DeleteMetricPolicyCommand = class _DeleteMetricPolicyCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("MediaStore_20170901", "DeleteMetricPolicy", {}).n("MediaStoreClient", "DeleteMetricPolicyCommand").f(void 0, void 0).ser(se_DeleteMetricPolicyCommand).de(de_DeleteMetricPolicyCommand).build() {
};
__name(_DeleteMetricPolicyCommand, "DeleteMetricPolicyCommand");
var DeleteMetricPolicyCommand = _DeleteMetricPolicyCommand;

// src/commands/DescribeContainerCommand.ts




var _DescribeContainerCommand = class _DescribeContainerCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("MediaStore_20170901", "DescribeContainer", {}).n("MediaStoreClient", "DescribeContainerCommand").f(void 0, void 0).ser(se_DescribeContainerCommand).de(de_DescribeContainerCommand).build() {
};
__name(_DescribeContainerCommand, "DescribeContainerCommand");
var DescribeContainerCommand = _DescribeContainerCommand;

// src/commands/GetContainerPolicyCommand.ts




var _GetContainerPolicyCommand = class _GetContainerPolicyCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("MediaStore_20170901", "GetContainerPolicy", {}).n("MediaStoreClient", "GetContainerPolicyCommand").f(void 0, void 0).ser(se_GetContainerPolicyCommand).de(de_GetContainerPolicyCommand).build() {
};
__name(_GetContainerPolicyCommand, "GetContainerPolicyCommand");
var GetContainerPolicyCommand = _GetContainerPolicyCommand;

// src/commands/GetCorsPolicyCommand.ts




var _GetCorsPolicyCommand = class _GetCorsPolicyCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("MediaStore_20170901", "GetCorsPolicy", {}).n("MediaStoreClient", "GetCorsPolicyCommand").f(void 0, void 0).ser(se_GetCorsPolicyCommand).de(de_GetCorsPolicyCommand).build() {
};
__name(_GetCorsPolicyCommand, "GetCorsPolicyCommand");
var GetCorsPolicyCommand = _GetCorsPolicyCommand;

// src/commands/GetLifecyclePolicyCommand.ts




var _GetLifecyclePolicyCommand = class _GetLifecyclePolicyCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("MediaStore_20170901", "GetLifecyclePolicy", {}).n("MediaStoreClient", "GetLifecyclePolicyCommand").f(void 0, void 0).ser(se_GetLifecyclePolicyCommand).de(de_GetLifecyclePolicyCommand).build() {
};
__name(_GetLifecyclePolicyCommand, "GetLifecyclePolicyCommand");
var GetLifecyclePolicyCommand = _GetLifecyclePolicyCommand;

// src/commands/GetMetricPolicyCommand.ts




var _GetMetricPolicyCommand = class _GetMetricPolicyCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("MediaStore_20170901", "GetMetricPolicy", {}).n("MediaStoreClient", "GetMetricPolicyCommand").f(void 0, void 0).ser(se_GetMetricPolicyCommand).de(de_GetMetricPolicyCommand).build() {
};
__name(_GetMetricPolicyCommand, "GetMetricPolicyCommand");
var GetMetricPolicyCommand = _GetMetricPolicyCommand;

// src/commands/ListContainersCommand.ts




var _ListContainersCommand = class _ListContainersCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("MediaStore_20170901", "ListContainers", {}).n("MediaStoreClient", "ListContainersCommand").f(void 0, void 0).ser(se_ListContainersCommand).de(de_ListContainersCommand).build() {
};
__name(_ListContainersCommand, "ListContainersCommand");
var ListContainersCommand = _ListContainersCommand;

// src/commands/ListTagsForResourceCommand.ts




var _ListTagsForResourceCommand = class _ListTagsForResourceCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("MediaStore_20170901", "ListTagsForResource", {}).n("MediaStoreClient", "ListTagsForResourceCommand").f(void 0, void 0).ser(se_ListTagsForResourceCommand).de(de_ListTagsForResourceCommand).build() {
};
__name(_ListTagsForResourceCommand, "ListTagsForResourceCommand");
var ListTagsForResourceCommand = _ListTagsForResourceCommand;

// src/commands/PutContainerPolicyCommand.ts




var _PutContainerPolicyCommand = class _PutContainerPolicyCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("MediaStore_20170901", "PutContainerPolicy", {}).n("MediaStoreClient", "PutContainerPolicyCommand").f(void 0, void 0).ser(se_PutContainerPolicyCommand).de(de_PutContainerPolicyCommand).build() {
};
__name(_PutContainerPolicyCommand, "PutContainerPolicyCommand");
var PutContainerPolicyCommand = _PutContainerPolicyCommand;

// src/commands/PutCorsPolicyCommand.ts




var _PutCorsPolicyCommand = class _PutCorsPolicyCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("MediaStore_20170901", "PutCorsPolicy", {}).n("MediaStoreClient", "PutCorsPolicyCommand").f(void 0, void 0).ser(se_PutCorsPolicyCommand).de(de_PutCorsPolicyCommand).build() {
};
__name(_PutCorsPolicyCommand, "PutCorsPolicyCommand");
var PutCorsPolicyCommand = _PutCorsPolicyCommand;

// src/commands/PutLifecyclePolicyCommand.ts




var _PutLifecyclePolicyCommand = class _PutLifecyclePolicyCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("MediaStore_20170901", "PutLifecyclePolicy", {}).n("MediaStoreClient", "PutLifecyclePolicyCommand").f(void 0, void 0).ser(se_PutLifecyclePolicyCommand).de(de_PutLifecyclePolicyCommand).build() {
};
__name(_PutLifecyclePolicyCommand, "PutLifecyclePolicyCommand");
var PutLifecyclePolicyCommand = _PutLifecyclePolicyCommand;

// src/commands/PutMetricPolicyCommand.ts




var _PutMetricPolicyCommand = class _PutMetricPolicyCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("MediaStore_20170901", "PutMetricPolicy", {}).n("MediaStoreClient", "PutMetricPolicyCommand").f(void 0, void 0).ser(se_PutMetricPolicyCommand).de(de_PutMetricPolicyCommand).build() {
};
__name(_PutMetricPolicyCommand, "PutMetricPolicyCommand");
var PutMetricPolicyCommand = _PutMetricPolicyCommand;

// src/commands/StartAccessLoggingCommand.ts




var _StartAccessLoggingCommand = class _StartAccessLoggingCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("MediaStore_20170901", "StartAccessLogging", {}).n("MediaStoreClient", "StartAccessLoggingCommand").f(void 0, void 0).ser(se_StartAccessLoggingCommand).de(de_StartAccessLoggingCommand).build() {
};
__name(_StartAccessLoggingCommand, "StartAccessLoggingCommand");
var StartAccessLoggingCommand = _StartAccessLoggingCommand;

// src/commands/StopAccessLoggingCommand.ts




var _StopAccessLoggingCommand = class _StopAccessLoggingCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("MediaStore_20170901", "StopAccessLogging", {}).n("MediaStoreClient", "StopAccessLoggingCommand").f(void 0, void 0).ser(se_StopAccessLoggingCommand).de(de_StopAccessLoggingCommand).build() {
};
__name(_StopAccessLoggingCommand, "StopAccessLoggingCommand");
var StopAccessLoggingCommand = _StopAccessLoggingCommand;

// src/commands/TagResourceCommand.ts




var _TagResourceCommand = class _TagResourceCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("MediaStore_20170901", "TagResource", {}).n("MediaStoreClient", "TagResourceCommand").f(void 0, void 0).ser(se_TagResourceCommand).de(de_TagResourceCommand).build() {
};
__name(_TagResourceCommand, "TagResourceCommand");
var TagResourceCommand = _TagResourceCommand;

// src/commands/UntagResourceCommand.ts




var _UntagResourceCommand = class _UntagResourceCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("MediaStore_20170901", "UntagResource", {}).n("MediaStoreClient", "UntagResourceCommand").f(void 0, void 0).ser(se_UntagResourceCommand).de(de_UntagResourceCommand).build() {
};
__name(_UntagResourceCommand, "UntagResourceCommand");
var UntagResourceCommand = _UntagResourceCommand;

// src/MediaStore.ts
var commands = {
  CreateContainerCommand,
  DeleteContainerCommand,
  DeleteContainerPolicyCommand,
  DeleteCorsPolicyCommand,
  DeleteLifecyclePolicyCommand,
  DeleteMetricPolicyCommand,
  DescribeContainerCommand,
  GetContainerPolicyCommand,
  GetCorsPolicyCommand,
  GetLifecyclePolicyCommand,
  GetMetricPolicyCommand,
  ListContainersCommand,
  ListTagsForResourceCommand,
  PutContainerPolicyCommand,
  PutCorsPolicyCommand,
  PutLifecyclePolicyCommand,
  PutMetricPolicyCommand,
  StartAccessLoggingCommand,
  StopAccessLoggingCommand,
  TagResourceCommand,
  UntagResourceCommand
};
var _MediaStore = class _MediaStore extends MediaStoreClient {
};
__name(_MediaStore, "MediaStore");
var MediaStore = _MediaStore;
(0, import_smithy_client.createAggregatedClient)(commands, MediaStore);

// src/pagination/ListContainersPaginator.ts

var paginateListContainers = (0, import_core.createPaginator)(MediaStoreClient, ListContainersCommand, "NextToken", "NextToken", "MaxResults");
// Annotate the CommonJS export names for ESM import in node:

0 && (module.exports = {
  MediaStoreServiceException,
  __Client,
  MediaStoreClient,
  MediaStore,
  $Command,
  CreateContainerCommand,
  DeleteContainerCommand,
  DeleteContainerPolicyCommand,
  DeleteCorsPolicyCommand,
  DeleteLifecyclePolicyCommand,
  DeleteMetricPolicyCommand,
  DescribeContainerCommand,
  GetContainerPolicyCommand,
  GetCorsPolicyCommand,
  GetLifecyclePolicyCommand,
  GetMetricPolicyCommand,
  ListContainersCommand,
  ListTagsForResourceCommand,
  PutContainerPolicyCommand,
  PutCorsPolicyCommand,
  PutLifecyclePolicyCommand,
  PutMetricPolicyCommand,
  StartAccessLoggingCommand,
  StopAccessLoggingCommand,
  TagResourceCommand,
  UntagResourceCommand,
  paginateListContainers,
  MethodName,
  ContainerStatus,
  ContainerInUseException,
  ContainerLevelMetrics,
  ContainerNotFoundException,
  CorsPolicyNotFoundException,
  InternalServerError,
  LimitExceededException,
  PolicyNotFoundException
});
