// @azure/arm-recoveryservices-siterecovery/dist-esm/src/models/index.js
/*
 * Copyright (c) Microsoft Corporation.
 * Licensed under the MIT License.
 *
 * Code generated by Microsoft (R) AutoRest Code Generator.
 * Changes may cause incorrect behavior and will be lost if the code is regenerated.
 */
/** Known values of {@link HealthErrorCustomerResolvability} that the service accepts. */
export var KnownHealthErrorCustomerResolvability;
(function (KnownHealthErrorCustomerResolvability) {
    /** Allowed */
    KnownHealthErrorCustomerResolvability["Allowed"] = "Allowed";
    /** NotAllowed */
    KnownHealthErrorCustomerResolvability["NotAllowed"] = "NotAllowed";
})(KnownHealthErrorCustomerResolvability || (KnownHealthErrorCustomerResolvability = {}));
/** Known values of {@link MigrationState} that the service accepts. */
export var KnownMigrationState;
(function (KnownMigrationState) {
    /** None */
    KnownMigrationState["None"] = "None";
    /** EnableMigrationInProgress */
    KnownMigrationState["EnableMigrationInProgress"] = "EnableMigrationInProgress";
    /** EnableMigrationFailed */
    KnownMigrationState["EnableMigrationFailed"] = "EnableMigrationFailed";
    /** DisableMigrationInProgress */
    KnownMigrationState["DisableMigrationInProgress"] = "DisableMigrationInProgress";
    /** DisableMigrationFailed */
    KnownMigrationState["DisableMigrationFailed"] = "DisableMigrationFailed";
    /** InitialSeedingInProgress */
    KnownMigrationState["InitialSeedingInProgress"] = "InitialSeedingInProgress";
    /** InitialSeedingFailed */
    KnownMigrationState["InitialSeedingFailed"] = "InitialSeedingFailed";
    /** Replicating */
    KnownMigrationState["Replicating"] = "Replicating";
    /** MigrationInProgress */
    KnownMigrationState["MigrationInProgress"] = "MigrationInProgress";
    /** MigrationSucceeded */
    KnownMigrationState["MigrationSucceeded"] = "MigrationSucceeded";
    /** MigrationFailed */
    KnownMigrationState["MigrationFailed"] = "MigrationFailed";
    /** ResumeInProgress */
    KnownMigrationState["ResumeInProgress"] = "ResumeInProgress";
    /** ResumeInitiated */
    KnownMigrationState["ResumeInitiated"] = "ResumeInitiated";
    /** SuspendingProtection */
    KnownMigrationState["SuspendingProtection"] = "SuspendingProtection";
    /** ProtectionSuspended */
    KnownMigrationState["ProtectionSuspended"] = "ProtectionSuspended";
    /** MigrationCompletedWithInformation */
    KnownMigrationState["MigrationCompletedWithInformation"] = "MigrationCompletedWithInformation";
    /** MigrationPartiallySucceeded */
    KnownMigrationState["MigrationPartiallySucceeded"] = "MigrationPartiallySucceeded";
})(KnownMigrationState || (KnownMigrationState = {}));
/** Known values of {@link TestMigrationState} that the service accepts. */
export var KnownTestMigrationState;
(function (KnownTestMigrationState) {
    /** None */
    KnownTestMigrationState["None"] = "None";
    /** TestMigrationInProgress */
    KnownTestMigrationState["TestMigrationInProgress"] = "TestMigrationInProgress";
    /** TestMigrationSucceeded */
    KnownTestMigrationState["TestMigrationSucceeded"] = "TestMigrationSucceeded";
    /** TestMigrationFailed */
    KnownTestMigrationState["TestMigrationFailed"] = "TestMigrationFailed";
    /** TestMigrationCleanupInProgress */
    KnownTestMigrationState["TestMigrationCleanupInProgress"] = "TestMigrationCleanupInProgress";
    /** TestMigrationCompletedWithInformation */
    KnownTestMigrationState["TestMigrationCompletedWithInformation"] = "TestMigrationCompletedWithInformation";
    /** TestMigrationPartiallySucceeded */
    KnownTestMigrationState["TestMigrationPartiallySucceeded"] = "TestMigrationPartiallySucceeded";
})(KnownTestMigrationState || (KnownTestMigrationState = {}));
/** Known values of {@link ProtectionHealth} that the service accepts. */
export var KnownProtectionHealth;
(function (KnownProtectionHealth) {
    /** None */
    KnownProtectionHealth["None"] = "None";
    /** Normal */
    KnownProtectionHealth["Normal"] = "Normal";
    /** Warning */
    KnownProtectionHealth["Warning"] = "Warning";
    /** Critical */
    KnownProtectionHealth["Critical"] = "Critical";
})(KnownProtectionHealth || (KnownProtectionHealth = {}));
/** Known values of {@link MigrationItemOperation} that the service accepts. */
export var KnownMigrationItemOperation;
(function (KnownMigrationItemOperation) {
    /** DisableMigration */
    KnownMigrationItemOperation["DisableMigration"] = "DisableMigration";
    /** TestMigrate */
    KnownMigrationItemOperation["TestMigrate"] = "TestMigrate";
    /** TestMigrateCleanup */
    KnownMigrationItemOperation["TestMigrateCleanup"] = "TestMigrateCleanup";
    /** Migrate */
    KnownMigrationItemOperation["Migrate"] = "Migrate";
    /** StartResync */
    KnownMigrationItemOperation["StartResync"] = "StartResync";
    /** PauseReplication */
    KnownMigrationItemOperation["PauseReplication"] = "PauseReplication";
    /** ResumeReplication */
    KnownMigrationItemOperation["ResumeReplication"] = "ResumeReplication";
})(KnownMigrationItemOperation || (KnownMigrationItemOperation = {}));
/** Known values of {@link MigrationRecoveryPointType} that the service accepts. */
export var KnownMigrationRecoveryPointType;
(function (KnownMigrationRecoveryPointType) {
    /** NotSpecified */
    KnownMigrationRecoveryPointType["NotSpecified"] = "NotSpecified";
    /** ApplicationConsistent */
    KnownMigrationRecoveryPointType["ApplicationConsistent"] = "ApplicationConsistent";
    /** CrashConsistent */
    KnownMigrationRecoveryPointType["CrashConsistent"] = "CrashConsistent";
})(KnownMigrationRecoveryPointType || (KnownMigrationRecoveryPointType = {}));
/** Known values of {@link LicenseType} that the service accepts. */
export var KnownLicenseType;
(function (KnownLicenseType) {
    /** NotSpecified */
    KnownLicenseType["NotSpecified"] = "NotSpecified";
    /** NoLicenseType */
    KnownLicenseType["NoLicenseType"] = "NoLicenseType";
    /** WindowsServer */
    KnownLicenseType["WindowsServer"] = "WindowsServer";
})(KnownLicenseType || (KnownLicenseType = {}));
/** Known values of {@link DisableProtectionReason} that the service accepts. */
export var KnownDisableProtectionReason;
(function (KnownDisableProtectionReason) {
    /** NotSpecified */
    KnownDisableProtectionReason["NotSpecified"] = "NotSpecified";
    /** MigrationComplete */
    KnownDisableProtectionReason["MigrationComplete"] = "MigrationComplete";
})(KnownDisableProtectionReason || (KnownDisableProtectionReason = {}));
/** Known values of {@link AgentVersionStatus} that the service accepts. */
export var KnownAgentVersionStatus;
(function (KnownAgentVersionStatus) {
    /** Supported */
    KnownAgentVersionStatus["Supported"] = "Supported";
    /** NotSupported */
    KnownAgentVersionStatus["NotSupported"] = "NotSupported";
    /** Deprecated */
    KnownAgentVersionStatus["Deprecated"] = "Deprecated";
    /** UpdateRequired */
    KnownAgentVersionStatus["UpdateRequired"] = "UpdateRequired";
    /** SecurityUpdateRequired */
    KnownAgentVersionStatus["SecurityUpdateRequired"] = "SecurityUpdateRequired";
})(KnownAgentVersionStatus || (KnownAgentVersionStatus = {}));
/** Known values of {@link ExportJobOutputSerializationType} that the service accepts. */
export var KnownExportJobOutputSerializationType;
(function (KnownExportJobOutputSerializationType) {
    /** Json */
    KnownExportJobOutputSerializationType["Json"] = "Json";
    /** Xml */
    KnownExportJobOutputSerializationType["Xml"] = "Xml";
    /** Excel */
    KnownExportJobOutputSerializationType["Excel"] = "Excel";
})(KnownExportJobOutputSerializationType || (KnownExportJobOutputSerializationType = {}));
/** Known values of {@link RecoveryPlanGroupType} that the service accepts. */
export var KnownRecoveryPlanGroupType;
(function (KnownRecoveryPlanGroupType) {
    /** Shutdown */
    KnownRecoveryPlanGroupType["Shutdown"] = "Shutdown";
    /** Boot */
    KnownRecoveryPlanGroupType["Boot"] = "Boot";
    /** Failover */
    KnownRecoveryPlanGroupType["Failover"] = "Failover";
})(KnownRecoveryPlanGroupType || (KnownRecoveryPlanGroupType = {}));
/** Known values of {@link ReplicationProtectedItemOperation} that the service accepts. */
export var KnownReplicationProtectedItemOperation;
(function (KnownReplicationProtectedItemOperation) {
    /** ReverseReplicate */
    KnownReplicationProtectedItemOperation["ReverseReplicate"] = "ReverseReplicate";
    /** Commit */
    KnownReplicationProtectedItemOperation["Commit"] = "Commit";
    /** PlannedFailover */
    KnownReplicationProtectedItemOperation["PlannedFailover"] = "PlannedFailover";
    /** UnplannedFailover */
    KnownReplicationProtectedItemOperation["UnplannedFailover"] = "UnplannedFailover";
    /** DisableProtection */
    KnownReplicationProtectedItemOperation["DisableProtection"] = "DisableProtection";
    /** TestFailover */
    KnownReplicationProtectedItemOperation["TestFailover"] = "TestFailover";
    /** TestFailoverCleanup */
    KnownReplicationProtectedItemOperation["TestFailoverCleanup"] = "TestFailoverCleanup";
    /** Failback */
    KnownReplicationProtectedItemOperation["Failback"] = "Failback";
    /** FinalizeFailback */
    KnownReplicationProtectedItemOperation["FinalizeFailback"] = "FinalizeFailback";
    /** CancelFailover */
    KnownReplicationProtectedItemOperation["CancelFailover"] = "CancelFailover";
    /** ChangePit */
    KnownReplicationProtectedItemOperation["ChangePit"] = "ChangePit";
    /** RepairReplication */
    KnownReplicationProtectedItemOperation["RepairReplication"] = "RepairReplication";
    /** SwitchProtection */
    KnownReplicationProtectedItemOperation["SwitchProtection"] = "SwitchProtection";
    /** CompleteMigration */
    KnownReplicationProtectedItemOperation["CompleteMigration"] = "CompleteMigration";
})(KnownReplicationProtectedItemOperation || (KnownReplicationProtectedItemOperation = {}));
/** Known values of {@link PossibleOperationsDirections} that the service accepts. */
export var KnownPossibleOperationsDirections;
(function (KnownPossibleOperationsDirections) {
    /** PrimaryToRecovery */
    KnownPossibleOperationsDirections["PrimaryToRecovery"] = "PrimaryToRecovery";
    /** RecoveryToPrimary */
    KnownPossibleOperationsDirections["RecoveryToPrimary"] = "RecoveryToPrimary";
})(KnownPossibleOperationsDirections || (KnownPossibleOperationsDirections = {}));
/** Known values of {@link FailoverDeploymentModel} that the service accepts. */
export var KnownFailoverDeploymentModel;
(function (KnownFailoverDeploymentModel) {
    /** NotApplicable */
    KnownFailoverDeploymentModel["NotApplicable"] = "NotApplicable";
    /** Classic */
    KnownFailoverDeploymentModel["Classic"] = "Classic";
    /** ResourceManager */
    KnownFailoverDeploymentModel["ResourceManager"] = "ResourceManager";
})(KnownFailoverDeploymentModel || (KnownFailoverDeploymentModel = {}));
/** Known values of {@link SourceSiteOperations} that the service accepts. */
export var KnownSourceSiteOperations;
(function (KnownSourceSiteOperations) {
    /** Required */
    KnownSourceSiteOperations["Required"] = "Required";
    /** NotRequired */
    KnownSourceSiteOperations["NotRequired"] = "NotRequired";
})(KnownSourceSiteOperations || (KnownSourceSiteOperations = {}));
/** Known values of {@link HealthErrorCategory} that the service accepts. */
export var KnownHealthErrorCategory;
(function (KnownHealthErrorCategory) {
    /** None */
    KnownHealthErrorCategory["None"] = "None";
    /** Replication */
    KnownHealthErrorCategory["Replication"] = "Replication";
    /** TestFailover */
    KnownHealthErrorCategory["TestFailover"] = "TestFailover";
    /** Configuration */
    KnownHealthErrorCategory["Configuration"] = "Configuration";
    /** FabricInfrastructure */
    KnownHealthErrorCategory["FabricInfrastructure"] = "FabricInfrastructure";
    /** VersionExpiry */
    KnownHealthErrorCategory["VersionExpiry"] = "VersionExpiry";
    /** AgentAutoUpdateInfra */
    KnownHealthErrorCategory["AgentAutoUpdateInfra"] = "AgentAutoUpdateInfra";
    /** AgentAutoUpdateArtifactDeleted */
    KnownHealthErrorCategory["AgentAutoUpdateArtifactDeleted"] = "AgentAutoUpdateArtifactDeleted";
    /** AgentAutoUpdateRunAsAccount */
    KnownHealthErrorCategory["AgentAutoUpdateRunAsAccount"] = "AgentAutoUpdateRunAsAccount";
    /** AgentAutoUpdateRunAsAccountExpiry */
    KnownHealthErrorCategory["AgentAutoUpdateRunAsAccountExpiry"] = "AgentAutoUpdateRunAsAccountExpiry";
    /** AgentAutoUpdateRunAsAccountExpired */
    KnownHealthErrorCategory["AgentAutoUpdateRunAsAccountExpired"] = "AgentAutoUpdateRunAsAccountExpired";
})(KnownHealthErrorCategory || (KnownHealthErrorCategory = {}));
/** Known values of {@link Severity} that the service accepts. */
export var KnownSeverity;
(function (KnownSeverity) {
    /** None */
    KnownSeverity["None"] = "NONE";
    /** Warning */
    KnownSeverity["Warning"] = "Warning";
    /** Error */
    KnownSeverity["Error"] = "Error";
    /** Info */
    KnownSeverity["Info"] = "Info";
})(KnownSeverity || (KnownSeverity = {}));
/** Known values of {@link AgentAutoUpdateStatus} that the service accepts. */
export var KnownAgentAutoUpdateStatus;
(function (KnownAgentAutoUpdateStatus) {
    /** Disabled */
    KnownAgentAutoUpdateStatus["Disabled"] = "Disabled";
    /** Enabled */
    KnownAgentAutoUpdateStatus["Enabled"] = "Enabled";
})(KnownAgentAutoUpdateStatus || (KnownAgentAutoUpdateStatus = {}));
/** Known values of {@link AutomationAccountAuthenticationType} that the service accepts. */
export var KnownAutomationAccountAuthenticationType;
(function (KnownAutomationAccountAuthenticationType) {
    /** RunAsAccount */
    KnownAutomationAccountAuthenticationType["RunAsAccount"] = "RunAsAccount";
    /** SystemAssignedIdentity */
    KnownAutomationAccountAuthenticationType["SystemAssignedIdentity"] = "SystemAssignedIdentity";
})(KnownAutomationAccountAuthenticationType || (KnownAutomationAccountAuthenticationType = {}));
/** Known values of {@link A2ARecoveryAvailabilityType} that the service accepts. */
export var KnownA2ARecoveryAvailabilityType;
(function (KnownA2ARecoveryAvailabilityType) {
    /** Single */
    KnownA2ARecoveryAvailabilityType["Single"] = "Single";
    /** AvailabilitySet */
    KnownA2ARecoveryAvailabilityType["AvailabilitySet"] = "AvailabilitySet";
    /** AvailabilityZone */
    KnownA2ARecoveryAvailabilityType["AvailabilityZone"] = "AvailabilityZone";
})(KnownA2ARecoveryAvailabilityType || (KnownA2ARecoveryAvailabilityType = {}));
/** Known values of {@link AutoProtectionOfDataDisk} that the service accepts. */
export var KnownAutoProtectionOfDataDisk;
(function (KnownAutoProtectionOfDataDisk) {
    /** Disabled */
    KnownAutoProtectionOfDataDisk["Disabled"] = "Disabled";
    /** Enabled */
    KnownAutoProtectionOfDataDisk["Enabled"] = "Enabled";
})(KnownAutoProtectionOfDataDisk || (KnownAutoProtectionOfDataDisk = {}));
/** Known values of {@link ExtendedLocationType} that the service accepts. */
export var KnownExtendedLocationType;
(function (KnownExtendedLocationType) {
    /** EdgeZone */
    KnownExtendedLocationType["EdgeZone"] = "EdgeZone";
})(KnownExtendedLocationType || (KnownExtendedLocationType = {}));
/** Known values of {@link SetMultiVmSyncStatus} that the service accepts. */
export var KnownSetMultiVmSyncStatus;
(function (KnownSetMultiVmSyncStatus) {
    /** Enable */
    KnownSetMultiVmSyncStatus["Enable"] = "Enable";
    /** Disable */
    KnownSetMultiVmSyncStatus["Disable"] = "Disable";
})(KnownSetMultiVmSyncStatus || (KnownSetMultiVmSyncStatus = {}));
/** Known values of {@link RecoveryPointSyncType} that the service accepts. */
export var KnownRecoveryPointSyncType;
(function (KnownRecoveryPointSyncType) {
    /** MultiVmSyncRecoveryPoint */
    KnownRecoveryPointSyncType["MultiVmSyncRecoveryPoint"] = "MultiVmSyncRecoveryPoint";
    /** PerVmRecoveryPoint */
    KnownRecoveryPointSyncType["PerVmRecoveryPoint"] = "PerVmRecoveryPoint";
})(KnownRecoveryPointSyncType || (KnownRecoveryPointSyncType = {}));
/** Known values of {@link MultiVmGroupCreateOption} that the service accepts. */
export var KnownMultiVmGroupCreateOption;
(function (KnownMultiVmGroupCreateOption) {
    /** AutoCreated */
    KnownMultiVmGroupCreateOption["AutoCreated"] = "AutoCreated";
    /** UserSpecified */
    KnownMultiVmGroupCreateOption["UserSpecified"] = "UserSpecified";
})(KnownMultiVmGroupCreateOption || (KnownMultiVmGroupCreateOption = {}));
/** Known values of {@link VmEncryptionType} that the service accepts. */
export var KnownVmEncryptionType;
(function (KnownVmEncryptionType) {
    /** NotEncrypted */
    KnownVmEncryptionType["NotEncrypted"] = "NotEncrypted";
    /** OnePassEncrypted */
    KnownVmEncryptionType["OnePassEncrypted"] = "OnePassEncrypted";
    /** TwoPassEncrypted */
    KnownVmEncryptionType["TwoPassEncrypted"] = "TwoPassEncrypted";
})(KnownVmEncryptionType || (KnownVmEncryptionType = {}));
/** Known values of {@link ChurnOptionSelected} that the service accepts. */
export var KnownChurnOptionSelected;
(function (KnownChurnOptionSelected) {
    /** Normal */
    KnownChurnOptionSelected["Normal"] = "Normal";
    /** High */
    KnownChurnOptionSelected["High"] = "High";
})(KnownChurnOptionSelected || (KnownChurnOptionSelected = {}));
/** Known values of {@link DiskAccountType} that the service accepts. */
export var KnownDiskAccountType;
(function (KnownDiskAccountType) {
    /** StandardLRS */
    KnownDiskAccountType["StandardLRS"] = "Standard_LRS";
    /** PremiumLRS */
    KnownDiskAccountType["PremiumLRS"] = "Premium_LRS";
    /** StandardSSDLRS */
    KnownDiskAccountType["StandardSSDLRS"] = "StandardSSD_LRS";
})(KnownDiskAccountType || (KnownDiskAccountType = {}));
/** Known values of {@link SqlServerLicenseType} that the service accepts. */
export var KnownSqlServerLicenseType;
(function (KnownSqlServerLicenseType) {
    /** NotSpecified */
    KnownSqlServerLicenseType["NotSpecified"] = "NotSpecified";
    /** NoLicenseType */
    KnownSqlServerLicenseType["NoLicenseType"] = "NoLicenseType";
    /** Payg */
    KnownSqlServerLicenseType["Payg"] = "PAYG";
    /** Ahub */
    KnownSqlServerLicenseType["Ahub"] = "AHUB";
})(KnownSqlServerLicenseType || (KnownSqlServerLicenseType = {}));
/** Known values of {@link PresenceStatus} that the service accepts. */
export var KnownPresenceStatus;
(function (KnownPresenceStatus) {
    /** Unknown */
    KnownPresenceStatus["Unknown"] = "Unknown";
    /** Present */
    KnownPresenceStatus["Present"] = "Present";
    /** NotPresent */
    KnownPresenceStatus["NotPresent"] = "NotPresent";
})(KnownPresenceStatus || (KnownPresenceStatus = {}));
/** Known values of {@link RcmComponentStatus} that the service accepts. */
export var KnownRcmComponentStatus;
(function (KnownRcmComponentStatus) {
    /** Healthy */
    KnownRcmComponentStatus["Healthy"] = "Healthy";
    /** Warning */
    KnownRcmComponentStatus["Warning"] = "Warning";
    /** Critical */
    KnownRcmComponentStatus["Critical"] = "Critical";
    /** Unknown */
    KnownRcmComponentStatus["Unknown"] = "Unknown";
})(KnownRcmComponentStatus || (KnownRcmComponentStatus = {}));
/** Known values of {@link AgentUpgradeBlockedReason} that the service accepts. */
export var KnownAgentUpgradeBlockedReason;
(function (KnownAgentUpgradeBlockedReason) {
    /** AlreadyOnLatestVersion */
    KnownAgentUpgradeBlockedReason["AlreadyOnLatestVersion"] = "AlreadyOnLatestVersion";
    /** RebootRequired */
    KnownAgentUpgradeBlockedReason["RebootRequired"] = "RebootRequired";
    /** AgentNoHeartbeat */
    KnownAgentUpgradeBlockedReason["AgentNoHeartbeat"] = "AgentNoHeartbeat";
    /** RcmProxyNoHeartbeat */
    KnownAgentUpgradeBlockedReason["RcmProxyNoHeartbeat"] = "RcmProxyNoHeartbeat";
    /** ProcessServerNoHeartbeat */
    KnownAgentUpgradeBlockedReason["ProcessServerNoHeartbeat"] = "ProcessServerNoHeartbeat";
    /** IncompatibleApplianceVersion */
    KnownAgentUpgradeBlockedReason["IncompatibleApplianceVersion"] = "IncompatibleApplianceVersion";
    /** NotProtected */
    KnownAgentUpgradeBlockedReason["NotProtected"] = "NotProtected";
    /** UnsupportedProtectionScenario */
    KnownAgentUpgradeBlockedReason["UnsupportedProtectionScenario"] = "UnsupportedProtectionScenario";
    /** DistroIsNotReported */
    KnownAgentUpgradeBlockedReason["DistroIsNotReported"] = "DistroIsNotReported";
    /** DistroNotSupportedForUpgrade */
    KnownAgentUpgradeBlockedReason["DistroNotSupportedForUpgrade"] = "DistroNotSupportedForUpgrade";
    /** MissingUpgradePath */
    KnownAgentUpgradeBlockedReason["MissingUpgradePath"] = "MissingUpgradePath";
    /** InvalidAgentVersion */
    KnownAgentUpgradeBlockedReason["InvalidAgentVersion"] = "InvalidAgentVersion";
    /** InvalidDriverVersion */
    KnownAgentUpgradeBlockedReason["InvalidDriverVersion"] = "InvalidDriverVersion";
    /** Unknown */
    KnownAgentUpgradeBlockedReason["Unknown"] = "Unknown";
})(KnownAgentUpgradeBlockedReason || (KnownAgentUpgradeBlockedReason = {}));
/** Known values of {@link InMageRcmFailbackRecoveryPointType} that the service accepts. */
export var KnownInMageRcmFailbackRecoveryPointType;
(function (KnownInMageRcmFailbackRecoveryPointType) {
    /** ApplicationConsistent */
    KnownInMageRcmFailbackRecoveryPointType["ApplicationConsistent"] = "ApplicationConsistent";
    /** CrashConsistent */
    KnownInMageRcmFailbackRecoveryPointType["CrashConsistent"] = "CrashConsistent";
})(KnownInMageRcmFailbackRecoveryPointType || (KnownInMageRcmFailbackRecoveryPointType = {}));
/** Known values of {@link DiskReplicationProgressHealth} that the service accepts. */
export var KnownDiskReplicationProgressHealth;
(function (KnownDiskReplicationProgressHealth) {
    /** None */
    KnownDiskReplicationProgressHealth["None"] = "None";
    /** InProgress */
    KnownDiskReplicationProgressHealth["InProgress"] = "InProgress";
    /** SlowProgress */
    KnownDiskReplicationProgressHealth["SlowProgress"] = "SlowProgress";
    /** NoProgress */
    KnownDiskReplicationProgressHealth["NoProgress"] = "NoProgress";
    /** Queued */
    KnownDiskReplicationProgressHealth["Queued"] = "Queued";
})(KnownDiskReplicationProgressHealth || (KnownDiskReplicationProgressHealth = {}));
/** Known values of {@link VmReplicationProgressHealth} that the service accepts. */
export var KnownVmReplicationProgressHealth;
(function (KnownVmReplicationProgressHealth) {
    /** None */
    KnownVmReplicationProgressHealth["None"] = "None";
    /** InProgress */
    KnownVmReplicationProgressHealth["InProgress"] = "InProgress";
    /** SlowProgress */
    KnownVmReplicationProgressHealth["SlowProgress"] = "SlowProgress";
    /** NoProgress */
    KnownVmReplicationProgressHealth["NoProgress"] = "NoProgress";
})(KnownVmReplicationProgressHealth || (KnownVmReplicationProgressHealth = {}));
/** Known values of {@link ResyncState} that the service accepts. */
export var KnownResyncState;
(function (KnownResyncState) {
    /** None */
    KnownResyncState["None"] = "None";
    /** PreparedForResynchronization */
    KnownResyncState["PreparedForResynchronization"] = "PreparedForResynchronization";
    /** StartedResynchronization */
    KnownResyncState["StartedResynchronization"] = "StartedResynchronization";
})(KnownResyncState || (KnownResyncState = {}));
/** Known values of {@link PlannedFailoverStatus} that the service accepts. */
export var KnownPlannedFailoverStatus;
(function (KnownPlannedFailoverStatus) {
    /** Succeeded */
    KnownPlannedFailoverStatus["Succeeded"] = "Succeeded";
    /** Failed */
    KnownPlannedFailoverStatus["Failed"] = "Failed";
    /** Cancelled */
    KnownPlannedFailoverStatus["Cancelled"] = "Cancelled";
    /** Unknown */
    KnownPlannedFailoverStatus["Unknown"] = "Unknown";
})(KnownPlannedFailoverStatus || (KnownPlannedFailoverStatus = {}));
/** Known values of {@link EthernetAddressType} that the service accepts. */
export var KnownEthernetAddressType;
(function (KnownEthernetAddressType) {
    /** Dynamic */
    KnownEthernetAddressType["Dynamic"] = "Dynamic";
    /** Static */
    KnownEthernetAddressType["Static"] = "Static";
})(KnownEthernetAddressType || (KnownEthernetAddressType = {}));
/** Known values of {@link MobilityAgentUpgradeState} that the service accepts. */
export var KnownMobilityAgentUpgradeState;
(function (KnownMobilityAgentUpgradeState) {
    /** None */
    KnownMobilityAgentUpgradeState["None"] = "None";
    /** Started */
    KnownMobilityAgentUpgradeState["Started"] = "Started";
    /** Completed */
    KnownMobilityAgentUpgradeState["Completed"] = "Completed";
    /** Commit */
    KnownMobilityAgentUpgradeState["Commit"] = "Commit";
})(KnownMobilityAgentUpgradeState || (KnownMobilityAgentUpgradeState = {}));
/** Known values of {@link RecoveryPointType} that the service accepts. */
export var KnownRecoveryPointType;
(function (KnownRecoveryPointType) {
    /** LatestTime */
    KnownRecoveryPointType["LatestTime"] = "LatestTime";
    /** LatestTag */
    KnownRecoveryPointType["LatestTag"] = "LatestTag";
    /** Custom */
    KnownRecoveryPointType["Custom"] = "Custom";
})(KnownRecoveryPointType || (KnownRecoveryPointType = {}));
/** Known values of {@link A2ARpRecoveryPointType} that the service accepts. */
export var KnownA2ARpRecoveryPointType;
(function (KnownA2ARpRecoveryPointType) {
    /** Latest */
    KnownA2ARpRecoveryPointType["Latest"] = "Latest";
    /** LatestApplicationConsistent */
    KnownA2ARpRecoveryPointType["LatestApplicationConsistent"] = "LatestApplicationConsistent";
    /** LatestCrashConsistent */
    KnownA2ARpRecoveryPointType["LatestCrashConsistent"] = "LatestCrashConsistent";
    /** LatestProcessed */
    KnownA2ARpRecoveryPointType["LatestProcessed"] = "LatestProcessed";
})(KnownA2ARpRecoveryPointType || (KnownA2ARpRecoveryPointType = {}));
/** Known values of {@link MultiVmSyncPointOption} that the service accepts. */
export var KnownMultiVmSyncPointOption;
(function (KnownMultiVmSyncPointOption) {
    /** UseMultiVmSyncRecoveryPoint */
    KnownMultiVmSyncPointOption["UseMultiVmSyncRecoveryPoint"] = "UseMultiVmSyncRecoveryPoint";
    /** UsePerVmRecoveryPoint */
    KnownMultiVmSyncPointOption["UsePerVmRecoveryPoint"] = "UsePerVmRecoveryPoint";
})(KnownMultiVmSyncPointOption || (KnownMultiVmSyncPointOption = {}));
/** Known values of {@link RecoveryPlanActionLocation} that the service accepts. */
export var KnownRecoveryPlanActionLocation;
(function (KnownRecoveryPlanActionLocation) {
    /** Primary */
    KnownRecoveryPlanActionLocation["Primary"] = "Primary";
    /** Recovery */
    KnownRecoveryPlanActionLocation["Recovery"] = "Recovery";
})(KnownRecoveryPlanActionLocation || (KnownRecoveryPlanActionLocation = {}));
/** Known values of {@link DataSyncStatus} that the service accepts. */
export var KnownDataSyncStatus;
(function (KnownDataSyncStatus) {
    /** ForDownTime */
    KnownDataSyncStatus["ForDownTime"] = "ForDownTime";
    /** ForSynchronization */
    KnownDataSyncStatus["ForSynchronization"] = "ForSynchronization";
})(KnownDataSyncStatus || (KnownDataSyncStatus = {}));
/** Known values of {@link AlternateLocationRecoveryOption} that the service accepts. */
export var KnownAlternateLocationRecoveryOption;
(function (KnownAlternateLocationRecoveryOption) {
    /** CreateVmIfNotFound */
    KnownAlternateLocationRecoveryOption["CreateVmIfNotFound"] = "CreateVmIfNotFound";
    /** NoAction */
    KnownAlternateLocationRecoveryOption["NoAction"] = "NoAction";
})(KnownAlternateLocationRecoveryOption || (KnownAlternateLocationRecoveryOption = {}));
/** Known values of {@link HyperVReplicaAzureRpRecoveryPointType} that the service accepts. */
export var KnownHyperVReplicaAzureRpRecoveryPointType;
(function (KnownHyperVReplicaAzureRpRecoveryPointType) {
    /** Latest */
    KnownHyperVReplicaAzureRpRecoveryPointType["Latest"] = "Latest";
    /** LatestApplicationConsistent */
    KnownHyperVReplicaAzureRpRecoveryPointType["LatestApplicationConsistent"] = "LatestApplicationConsistent";
    /** LatestProcessed */
    KnownHyperVReplicaAzureRpRecoveryPointType["LatestProcessed"] = "LatestProcessed";
})(KnownHyperVReplicaAzureRpRecoveryPointType || (KnownHyperVReplicaAzureRpRecoveryPointType = {}));
/** Known values of {@link InMageV2RpRecoveryPointType} that the service accepts. */
export var KnownInMageV2RpRecoveryPointType;
(function (KnownInMageV2RpRecoveryPointType) {
    /** Latest */
    KnownInMageV2RpRecoveryPointType["Latest"] = "Latest";
    /** LatestApplicationConsistent */
    KnownInMageV2RpRecoveryPointType["LatestApplicationConsistent"] = "LatestApplicationConsistent";
    /** LatestCrashConsistent */
    KnownInMageV2RpRecoveryPointType["LatestCrashConsistent"] = "LatestCrashConsistent";
    /** LatestProcessed */
    KnownInMageV2RpRecoveryPointType["LatestProcessed"] = "LatestProcessed";
})(KnownInMageV2RpRecoveryPointType || (KnownInMageV2RpRecoveryPointType = {}));
/** Known values of {@link RpInMageRecoveryPointType} that the service accepts. */
export var KnownRpInMageRecoveryPointType;
(function (KnownRpInMageRecoveryPointType) {
    /** LatestTime */
    KnownRpInMageRecoveryPointType["LatestTime"] = "LatestTime";
    /** LatestTag */
    KnownRpInMageRecoveryPointType["LatestTag"] = "LatestTag";
    /** Custom */
    KnownRpInMageRecoveryPointType["Custom"] = "Custom";
})(KnownRpInMageRecoveryPointType || (KnownRpInMageRecoveryPointType = {}));
/** Known values of {@link RecoveryPlanPointType} that the service accepts. */
export var KnownRecoveryPlanPointType;
(function (KnownRecoveryPlanPointType) {
    /** Latest */
    KnownRecoveryPlanPointType["Latest"] = "Latest";
    /** LatestApplicationConsistent */
    KnownRecoveryPlanPointType["LatestApplicationConsistent"] = "LatestApplicationConsistent";
    /** LatestCrashConsistent */
    KnownRecoveryPlanPointType["LatestCrashConsistent"] = "LatestCrashConsistent";
    /** LatestProcessed */
    KnownRecoveryPlanPointType["LatestProcessed"] = "LatestProcessed";
})(KnownRecoveryPlanPointType || (KnownRecoveryPlanPointType = {}));
/** Known values of {@link SecurityType} that the service accepts. */
export var KnownSecurityType;
(function (KnownSecurityType) {
    /** None */
    KnownSecurityType["None"] = "None";
    /** TrustedLaunch */
    KnownSecurityType["TrustedLaunch"] = "TrustedLaunch";
    /** ConfidentialVM */
    KnownSecurityType["ConfidentialVM"] = "ConfidentialVM";
})(KnownSecurityType || (KnownSecurityType = {}));
