// @azure/arm-recoveryservices-siterecovery/dist-esm/src/models/parameters.js
/*
 * Copyright (c) Microsoft Corporation.
 * Licensed under the MIT License.
 *
 * Code generated by Microsoft (R) AutoRest Code Generator.
 * Changes may cause incorrect behavior and will be lost if the code is regenerated.
 */
import { ConfigureAlertRequest as ConfigureAlertRequestMapper, FabricCreationInput as FabricCreationInputMapper, FailoverProcessServerRequest as FailoverProcessServerRequestMapper, RenewCertificateInput as RenewCertificateInputMapper, CreateNetworkMappingInput as CreateNetworkMappingInputMapper, UpdateNetworkMappingInput as UpdateNetworkMappingInputMapper, CreateProtectionContainerInput as CreateProtectionContainerInputMapper, DiscoverProtectableItemRequest as DiscoverProtectableItemRequestMapper, SwitchProtectionInput as SwitchProtectionInputMapper, EnableMigrationInput as EnableMigrationInputMapper, UpdateMigrationItemInput as UpdateMigrationItemInputMapper, MigrateInput as MigrateInputMapper, PauseReplicationInput as PauseReplicationInputMapper, ResumeReplicationInput as ResumeReplicationInputMapper, ResyncInput as ResyncInputMapper, TestMigrateInput as TestMigrateInputMapper, TestMigrateCleanupInput as TestMigrateCleanupInputMapper, EnableProtectionInput as EnableProtectionInputMapper, UpdateReplicationProtectedItemInput as UpdateReplicationProtectedItemInputMapper, AddDisksInput as AddDisksInputMapper, ApplyRecoveryPointInput as ApplyRecoveryPointInputMapper, PlannedFailoverInput as PlannedFailoverInputMapper, DisableProtectionInput as DisableProtectionInputMapper, RemoveDisksInput as RemoveDisksInputMapper, ReverseReplicationInput as ReverseReplicationInputMapper, ResolveHealthInput as ResolveHealthInputMapper, SwitchProviderInput as SwitchProviderInputMapper, TestFailoverInput as TestFailoverInputMapper, TestFailoverCleanupInput as TestFailoverCleanupInputMapper, UnplannedFailoverInput as UnplannedFailoverInputMapper, UpdateApplianceForReplicationProtectedItemInput as UpdateApplianceForReplicationProtectedItemInputMapper, UpdateMobilityServiceRequest as UpdateMobilityServiceRequestMapper, CreateProtectionContainerMappingInput as CreateProtectionContainerMappingInputMapper, UpdateProtectionContainerMappingInput as UpdateProtectionContainerMappingInputMapper, RemoveProtectionContainerMappingInput as RemoveProtectionContainerMappingInputMapper, AddRecoveryServicesProviderInput as AddRecoveryServicesProviderInputMapper, StorageClassificationMappingInput as StorageClassificationMappingInputMapper, AddVCenterRequest as AddVCenterRequestMapper, UpdateVCenterRequest as UpdateVCenterRequestMapper, ResumeJobParams as ResumeJobParamsMapper, JobQueryParameter as JobQueryParameterMapper, CreatePolicyInput as CreatePolicyInputMapper, UpdatePolicyInput as UpdatePolicyInputMapper, CreateProtectionIntentInput as CreateProtectionIntentInputMapper, CreateRecoveryPlanInput as CreateRecoveryPlanInputMapper, UpdateRecoveryPlanInput as UpdateRecoveryPlanInputMapper, RecoveryPlanPlannedFailoverInput as RecoveryPlanPlannedFailoverInputMapper, RecoveryPlanTestFailoverInput as RecoveryPlanTestFailoverInputMapper, RecoveryPlanTestFailoverCleanupInput as RecoveryPlanTestFailoverCleanupInputMapper, RecoveryPlanUnplannedFailoverInput as RecoveryPlanUnplannedFailoverInputMapper, VaultSettingCreationInput as VaultSettingCreationInputMapper } from "../models/mappers";
export const accept = {
    parameterPath: "accept",
    mapper: {
        defaultValue: "application/json",
        isConstant: true,
        serializedName: "Accept",
        type: {
            name: "String"
        }
    }
};
export const $host = {
    parameterPath: "$host",
    mapper: {
        serializedName: "$host",
        required: true,
        type: {
            name: "String"
        }
    },
    skipEncoding: true
};
export const apiVersion = {
    parameterPath: "apiVersion",
    mapper: {
        defaultValue: "2023-06-01",
        isConstant: true,
        serializedName: "api-version",
        type: {
            name: "String"
        }
    }
};
export const resourceGroupName = {
    parameterPath: "resourceGroupName",
    mapper: {
        serializedName: "resourceGroupName",
        required: true,
        type: {
            name: "String"
        }
    }
};
export const subscriptionId = {
    parameterPath: "subscriptionId",
    mapper: {
        serializedName: "subscriptionId",
        required: true,
        type: {
            name: "String"
        }
    }
};
export const nextLink = {
    parameterPath: "nextLink",
    mapper: {
        serializedName: "nextLink",
        required: true,
        type: {
            name: "String"
        }
    },
    skipEncoding: true
};
export const resourceName = {
    parameterPath: "resourceName",
    mapper: {
        serializedName: "resourceName",
        required: true,
        type: {
            name: "String"
        }
    }
};
export const alertSettingName = {
    parameterPath: "alertSettingName",
    mapper: {
        serializedName: "alertSettingName",
        required: true,
        type: {
            name: "String"
        }
    }
};
export const contentType = {
    parameterPath: ["options", "contentType"],
    mapper: {
        defaultValue: "application/json",
        isConstant: true,
        serializedName: "Content-Type",
        type: {
            name: "String"
        }
    }
};
export const request = {
    parameterPath: "request",
    mapper: ConfigureAlertRequestMapper
};
export const filter = {
    parameterPath: ["options", "filter"],
    mapper: {
        serializedName: "$filter",
        type: {
            name: "String"
        }
    }
};
export const virtualMachineName = {
    parameterPath: "virtualMachineName",
    mapper: {
        serializedName: "virtualMachineName",
        required: true,
        type: {
            name: "String"
        }
    }
};
export const eventName = {
    parameterPath: "eventName",
    mapper: {
        serializedName: "eventName",
        required: true,
        type: {
            name: "String"
        }
    }
};
export const fabricName = {
    parameterPath: "fabricName",
    mapper: {
        serializedName: "fabricName",
        required: true,
        type: {
            name: "String"
        }
    }
};
export const input = {
    parameterPath: "input",
    mapper: FabricCreationInputMapper
};
export const failoverProcessServerRequest = {
    parameterPath: "failoverProcessServerRequest",
    mapper: FailoverProcessServerRequestMapper
};
export const renewCertificate = {
    parameterPath: "renewCertificate",
    mapper: RenewCertificateInputMapper
};
export const logicalNetworkName = {
    parameterPath: "logicalNetworkName",
    mapper: {
        serializedName: "logicalNetworkName",
        required: true,
        type: {
            name: "String"
        }
    }
};
export const networkName = {
    parameterPath: "networkName",
    mapper: {
        serializedName: "networkName",
        required: true,
        type: {
            name: "String"
        }
    }
};
export const networkMappingName = {
    parameterPath: "networkMappingName",
    mapper: {
        serializedName: "networkMappingName",
        required: true,
        type: {
            name: "String"
        }
    }
};
export const input1 = {
    parameterPath: "input",
    mapper: CreateNetworkMappingInputMapper
};
export const input2 = {
    parameterPath: "input",
    mapper: UpdateNetworkMappingInputMapper
};
export const protectionContainerName = {
    parameterPath: "protectionContainerName",
    mapper: {
        serializedName: "protectionContainerName",
        required: true,
        type: {
            name: "String"
        }
    }
};
export const creationInput = {
    parameterPath: "creationInput",
    mapper: CreateProtectionContainerInputMapper
};
export const discoverProtectableItemRequest = {
    parameterPath: "discoverProtectableItemRequest",
    mapper: DiscoverProtectableItemRequestMapper
};
export const switchInput = {
    parameterPath: "switchInput",
    mapper: SwitchProtectionInputMapper
};
export const skipToken = {
    parameterPath: ["options", "skipToken"],
    mapper: {
        serializedName: "skipToken",
        type: {
            name: "String"
        }
    }
};
export const takeToken = {
    parameterPath: ["options", "takeToken"],
    mapper: {
        serializedName: "takeToken",
        type: {
            name: "String"
        }
    }
};
export const migrationItemName = {
    parameterPath: "migrationItemName",
    mapper: {
        serializedName: "migrationItemName",
        required: true,
        type: {
            name: "String"
        }
    }
};
export const input3 = {
    parameterPath: "input",
    mapper: EnableMigrationInputMapper
};
export const deleteOption = {
    parameterPath: ["options", "deleteOption"],
    mapper: {
        serializedName: "deleteOption",
        type: {
            name: "String"
        }
    }
};
export const input4 = {
    parameterPath: "input",
    mapper: UpdateMigrationItemInputMapper
};
export const migrateInput = {
    parameterPath: "migrateInput",
    mapper: MigrateInputMapper
};
export const pauseReplicationInput = {
    parameterPath: "pauseReplicationInput",
    mapper: PauseReplicationInputMapper
};
export const resumeReplicationInput = {
    parameterPath: "resumeReplicationInput",
    mapper: ResumeReplicationInputMapper
};
export const input5 = {
    parameterPath: "input",
    mapper: ResyncInputMapper
};
export const testMigrateInput = {
    parameterPath: "testMigrateInput",
    mapper: TestMigrateInputMapper
};
export const testMigrateCleanupInput = {
    parameterPath: "testMigrateCleanupInput",
    mapper: TestMigrateCleanupInputMapper
};
export const migrationRecoveryPointName = {
    parameterPath: "migrationRecoveryPointName",
    mapper: {
        serializedName: "migrationRecoveryPointName",
        required: true,
        type: {
            name: "String"
        }
    }
};
export const take = {
    parameterPath: ["options", "take"],
    mapper: {
        serializedName: "$take",
        type: {
            name: "String"
        }
    }
};
export const skipToken1 = {
    parameterPath: ["options", "skipToken"],
    mapper: {
        serializedName: "$skipToken",
        type: {
            name: "String"
        }
    }
};
export const protectableItemName = {
    parameterPath: "protectableItemName",
    mapper: {
        serializedName: "protectableItemName",
        required: true,
        type: {
            name: "String"
        }
    }
};
export const replicatedProtectedItemName = {
    parameterPath: "replicatedProtectedItemName",
    mapper: {
        serializedName: "replicatedProtectedItemName",
        required: true,
        type: {
            name: "String"
        }
    }
};
export const input6 = {
    parameterPath: "input",
    mapper: EnableProtectionInputMapper
};
export const updateProtectionInput = {
    parameterPath: "updateProtectionInput",
    mapper: UpdateReplicationProtectedItemInputMapper
};
export const addDisksInput = {
    parameterPath: "addDisksInput",
    mapper: AddDisksInputMapper
};
export const applyRecoveryPointInput = {
    parameterPath: "applyRecoveryPointInput",
    mapper: ApplyRecoveryPointInputMapper
};
export const failoverInput = {
    parameterPath: "failoverInput",
    mapper: PlannedFailoverInputMapper
};
export const disableProtectionInput = {
    parameterPath: "disableProtectionInput",
    mapper: DisableProtectionInputMapper
};
export const removeDisksInput = {
    parameterPath: "removeDisksInput",
    mapper: RemoveDisksInputMapper
};
export const reprotectInput = {
    parameterPath: "reprotectInput",
    mapper: ReverseReplicationInputMapper
};
export const resolveHealthInput = {
    parameterPath: "resolveHealthInput",
    mapper: ResolveHealthInputMapper
};
export const switchProviderInput = {
    parameterPath: "switchProviderInput",
    mapper: SwitchProviderInputMapper
};
export const testfailoverInput = {
    parameterPath: "testfailoverInput",
    mapper: TestFailoverInputMapper
};
export const cleanupInput = {
    parameterPath: "cleanupInput",
    mapper: TestFailoverCleanupInputMapper
};
export const failoverInput1 = {
    parameterPath: "failoverInput",
    mapper: UnplannedFailoverInputMapper
};
export const applianceUpdateInput = {
    parameterPath: "applianceUpdateInput",
    mapper: UpdateApplianceForReplicationProtectedItemInputMapper
};
export const updateMobilityServiceRequest = {
    parameterPath: "updateMobilityServiceRequest",
    mapper: UpdateMobilityServiceRequestMapper
};
export const recoveryPointName = {
    parameterPath: "recoveryPointName",
    mapper: {
        serializedName: "recoveryPointName",
        required: true,
        type: {
            name: "String"
        }
    }
};
export const mappingName = {
    parameterPath: "mappingName",
    mapper: {
        serializedName: "mappingName",
        required: true,
        type: {
            name: "String"
        }
    }
};
export const creationInput1 = {
    parameterPath: "creationInput",
    mapper: CreateProtectionContainerMappingInputMapper
};
export const updateInput = {
    parameterPath: "updateInput",
    mapper: UpdateProtectionContainerMappingInputMapper
};
export const removalInput = {
    parameterPath: "removalInput",
    mapper: RemoveProtectionContainerMappingInputMapper
};
export const providerName = {
    parameterPath: "providerName",
    mapper: {
        serializedName: "providerName",
        required: true,
        type: {
            name: "String"
        }
    }
};
export const addProviderInput = {
    parameterPath: "addProviderInput",
    mapper: AddRecoveryServicesProviderInputMapper
};
export const storageClassificationName = {
    parameterPath: "storageClassificationName",
    mapper: {
        serializedName: "storageClassificationName",
        required: true,
        type: {
            name: "String"
        }
    }
};
export const storageClassificationMappingName = {
    parameterPath: "storageClassificationMappingName",
    mapper: {
        serializedName: "storageClassificationMappingName",
        required: true,
        type: {
            name: "String"
        }
    }
};
export const pairingInput = {
    parameterPath: "pairingInput",
    mapper: StorageClassificationMappingInputMapper
};
export const vcenterName = {
    parameterPath: "vcenterName",
    mapper: {
        serializedName: "vcenterName",
        required: true,
        type: {
            name: "String"
        }
    }
};
export const addVCenterRequest = {
    parameterPath: "addVCenterRequest",
    mapper: AddVCenterRequestMapper
};
export const updateVCenterRequest = {
    parameterPath: "updateVCenterRequest",
    mapper: UpdateVCenterRequestMapper
};
export const jobName = {
    parameterPath: "jobName",
    mapper: {
        serializedName: "jobName",
        required: true,
        type: {
            name: "String"
        }
    }
};
export const resumeJobParams = {
    parameterPath: "resumeJobParams",
    mapper: ResumeJobParamsMapper
};
export const jobQueryParameter = {
    parameterPath: "jobQueryParameter",
    mapper: JobQueryParameterMapper
};
export const policyName = {
    parameterPath: "policyName",
    mapper: {
        serializedName: "policyName",
        required: true,
        type: {
            name: "String"
        }
    }
};
export const input7 = {
    parameterPath: "input",
    mapper: CreatePolicyInputMapper
};
export const input8 = {
    parameterPath: "input",
    mapper: UpdatePolicyInputMapper
};
export const intentObjectName = {
    parameterPath: "intentObjectName",
    mapper: {
        serializedName: "intentObjectName",
        required: true,
        type: {
            name: "String"
        }
    }
};
export const input9 = {
    parameterPath: "input",
    mapper: CreateProtectionIntentInputMapper
};
export const recoveryPlanName = {
    parameterPath: "recoveryPlanName",
    mapper: {
        serializedName: "recoveryPlanName",
        required: true,
        type: {
            name: "String"
        }
    }
};
export const input10 = {
    parameterPath: "input",
    mapper: CreateRecoveryPlanInputMapper
};
export const input11 = {
    parameterPath: "input",
    mapper: UpdateRecoveryPlanInputMapper
};
export const input12 = {
    parameterPath: "input",
    mapper: RecoveryPlanPlannedFailoverInputMapper
};
export const input13 = {
    parameterPath: "input",
    mapper: RecoveryPlanTestFailoverInputMapper
};
export const input14 = {
    parameterPath: "input",
    mapper: RecoveryPlanTestFailoverCleanupInputMapper
};
export const input15 = {
    parameterPath: "input",
    mapper: RecoveryPlanUnplannedFailoverInputMapper
};
export const instanceType = {
    parameterPath: ["options", "instanceType"],
    mapper: {
        serializedName: "instanceType",
        type: {
            name: "String"
        }
    }
};
export const vaultSettingName = {
    parameterPath: "vaultSettingName",
    mapper: {
        serializedName: "vaultSettingName",
        required: true,
        type: {
            name: "String"
        }
    }
};
export const input16 = {
    parameterPath: "input",
    mapper: VaultSettingCreationInputMapper
};
