// @azure/arm-frontdoor/dist-esm/src/models/index.js
/*
 * Copyright (c) Microsoft Corporation.
 * Licensed under the MIT License.
 *
 * Code generated by Microsoft (R) AutoRest Code Generator.
 * Changes may cause incorrect behavior and will be lost if the code is regenerated.
 */
/** Known values of {@link PolicyEnabledState} that the service accepts. */
export var KnownPolicyEnabledState;
(function (KnownPolicyEnabledState) {
    /** Disabled */
    KnownPolicyEnabledState["Disabled"] = "Disabled";
    /** Enabled */
    KnownPolicyEnabledState["Enabled"] = "Enabled";
})(KnownPolicyEnabledState || (KnownPolicyEnabledState = {}));
/** Known values of {@link PolicyMode} that the service accepts. */
export var KnownPolicyMode;
(function (KnownPolicyMode) {
    /** Prevention */
    KnownPolicyMode["Prevention"] = "Prevention";
    /** Detection */
    KnownPolicyMode["Detection"] = "Detection";
})(KnownPolicyMode || (KnownPolicyMode = {}));
/** Known values of {@link PolicyRequestBodyCheck} that the service accepts. */
export var KnownPolicyRequestBodyCheck;
(function (KnownPolicyRequestBodyCheck) {
    /** Disabled */
    KnownPolicyRequestBodyCheck["Disabled"] = "Disabled";
    /** Enabled */
    KnownPolicyRequestBodyCheck["Enabled"] = "Enabled";
})(KnownPolicyRequestBodyCheck || (KnownPolicyRequestBodyCheck = {}));
/** Known values of {@link WebApplicationFirewallScrubbingState} that the service accepts. */
export var KnownWebApplicationFirewallScrubbingState;
(function (KnownWebApplicationFirewallScrubbingState) {
    /** Enabled */
    KnownWebApplicationFirewallScrubbingState["Enabled"] = "Enabled";
    /** Disabled */
    KnownWebApplicationFirewallScrubbingState["Disabled"] = "Disabled";
})(KnownWebApplicationFirewallScrubbingState || (KnownWebApplicationFirewallScrubbingState = {}));
/** Known values of {@link ScrubbingRuleEntryMatchVariable} that the service accepts. */
export var KnownScrubbingRuleEntryMatchVariable;
(function (KnownScrubbingRuleEntryMatchVariable) {
    /** RequestIPAddress */
    KnownScrubbingRuleEntryMatchVariable["RequestIPAddress"] = "RequestIPAddress";
    /** RequestUri */
    KnownScrubbingRuleEntryMatchVariable["RequestUri"] = "RequestUri";
    /** QueryStringArgNames */
    KnownScrubbingRuleEntryMatchVariable["QueryStringArgNames"] = "QueryStringArgNames";
    /** RequestHeaderNames */
    KnownScrubbingRuleEntryMatchVariable["RequestHeaderNames"] = "RequestHeaderNames";
    /** RequestCookieNames */
    KnownScrubbingRuleEntryMatchVariable["RequestCookieNames"] = "RequestCookieNames";
    /** RequestBodyPostArgNames */
    KnownScrubbingRuleEntryMatchVariable["RequestBodyPostArgNames"] = "RequestBodyPostArgNames";
    /** RequestBodyJsonArgNames */
    KnownScrubbingRuleEntryMatchVariable["RequestBodyJsonArgNames"] = "RequestBodyJsonArgNames";
})(KnownScrubbingRuleEntryMatchVariable || (KnownScrubbingRuleEntryMatchVariable = {}));
/** Known values of {@link ScrubbingRuleEntryMatchOperator} that the service accepts. */
export var KnownScrubbingRuleEntryMatchOperator;
(function (KnownScrubbingRuleEntryMatchOperator) {
    /** EqualsAny */
    KnownScrubbingRuleEntryMatchOperator["EqualsAny"] = "EqualsAny";
    /** Equals */
    KnownScrubbingRuleEntryMatchOperator["Equals"] = "Equals";
})(KnownScrubbingRuleEntryMatchOperator || (KnownScrubbingRuleEntryMatchOperator = {}));
/** Known values of {@link ScrubbingRuleEntryState} that the service accepts. */
export var KnownScrubbingRuleEntryState;
(function (KnownScrubbingRuleEntryState) {
    /** Enabled */
    KnownScrubbingRuleEntryState["Enabled"] = "Enabled";
    /** Disabled */
    KnownScrubbingRuleEntryState["Disabled"] = "Disabled";
})(KnownScrubbingRuleEntryState || (KnownScrubbingRuleEntryState = {}));
/** Known values of {@link CustomRuleEnabledState} that the service accepts. */
export var KnownCustomRuleEnabledState;
(function (KnownCustomRuleEnabledState) {
    /** Disabled */
    KnownCustomRuleEnabledState["Disabled"] = "Disabled";
    /** Enabled */
    KnownCustomRuleEnabledState["Enabled"] = "Enabled";
})(KnownCustomRuleEnabledState || (KnownCustomRuleEnabledState = {}));
/** Known values of {@link RuleType} that the service accepts. */
export var KnownRuleType;
(function (KnownRuleType) {
    /** MatchRule */
    KnownRuleType["MatchRule"] = "MatchRule";
    /** RateLimitRule */
    KnownRuleType["RateLimitRule"] = "RateLimitRule";
})(KnownRuleType || (KnownRuleType = {}));
/** Known values of {@link VariableName} that the service accepts. */
export var KnownVariableName;
(function (KnownVariableName) {
    /** SocketAddr */
    KnownVariableName["SocketAddr"] = "SocketAddr";
    /** GeoLocation */
    KnownVariableName["GeoLocation"] = "GeoLocation";
    /** None */
    KnownVariableName["None"] = "None";
})(KnownVariableName || (KnownVariableName = {}));
/** Known values of {@link MatchVariable} that the service accepts. */
export var KnownMatchVariable;
(function (KnownMatchVariable) {
    /** RemoteAddr */
    KnownMatchVariable["RemoteAddr"] = "RemoteAddr";
    /** RequestMethod */
    KnownMatchVariable["RequestMethod"] = "RequestMethod";
    /** QueryString */
    KnownMatchVariable["QueryString"] = "QueryString";
    /** PostArgs */
    KnownMatchVariable["PostArgs"] = "PostArgs";
    /** RequestUri */
    KnownMatchVariable["RequestUri"] = "RequestUri";
    /** RequestHeader */
    KnownMatchVariable["RequestHeader"] = "RequestHeader";
    /** RequestBody */
    KnownMatchVariable["RequestBody"] = "RequestBody";
    /** Cookies */
    KnownMatchVariable["Cookies"] = "Cookies";
    /** SocketAddr */
    KnownMatchVariable["SocketAddr"] = "SocketAddr";
})(KnownMatchVariable || (KnownMatchVariable = {}));
/** Known values of {@link Operator} that the service accepts. */
export var KnownOperator;
(function (KnownOperator) {
    /** Any */
    KnownOperator["Any"] = "Any";
    /** IPMatch */
    KnownOperator["IPMatch"] = "IPMatch";
    /** GeoMatch */
    KnownOperator["GeoMatch"] = "GeoMatch";
    /** Equal */
    KnownOperator["Equal"] = "Equal";
    /** Contains */
    KnownOperator["Contains"] = "Contains";
    /** LessThan */
    KnownOperator["LessThan"] = "LessThan";
    /** GreaterThan */
    KnownOperator["GreaterThan"] = "GreaterThan";
    /** LessThanOrEqual */
    KnownOperator["LessThanOrEqual"] = "LessThanOrEqual";
    /** GreaterThanOrEqual */
    KnownOperator["GreaterThanOrEqual"] = "GreaterThanOrEqual";
    /** BeginsWith */
    KnownOperator["BeginsWith"] = "BeginsWith";
    /** EndsWith */
    KnownOperator["EndsWith"] = "EndsWith";
    /** RegEx */
    KnownOperator["RegEx"] = "RegEx";
})(KnownOperator || (KnownOperator = {}));
/** Known values of {@link TransformType} that the service accepts. */
export var KnownTransformType;
(function (KnownTransformType) {
    /** Lowercase */
    KnownTransformType["Lowercase"] = "Lowercase";
    /** Uppercase */
    KnownTransformType["Uppercase"] = "Uppercase";
    /** Trim */
    KnownTransformType["Trim"] = "Trim";
    /** UrlDecode */
    KnownTransformType["UrlDecode"] = "UrlDecode";
    /** UrlEncode */
    KnownTransformType["UrlEncode"] = "UrlEncode";
    /** RemoveNulls */
    KnownTransformType["RemoveNulls"] = "RemoveNulls";
})(KnownTransformType || (KnownTransformType = {}));
/** Known values of {@link ActionType} that the service accepts. */
export var KnownActionType;
(function (KnownActionType) {
    /** Allow */
    KnownActionType["Allow"] = "Allow";
    /** Block */
    KnownActionType["Block"] = "Block";
    /** Log */
    KnownActionType["Log"] = "Log";
    /** Redirect */
    KnownActionType["Redirect"] = "Redirect";
    /** AnomalyScoring */
    KnownActionType["AnomalyScoring"] = "AnomalyScoring";
    /** JSChallenge */
    KnownActionType["JSChallenge"] = "JSChallenge";
})(KnownActionType || (KnownActionType = {}));
/** Known values of {@link ManagedRuleSetActionType} that the service accepts. */
export var KnownManagedRuleSetActionType;
(function (KnownManagedRuleSetActionType) {
    /** Block */
    KnownManagedRuleSetActionType["Block"] = "Block";
    /** Log */
    KnownManagedRuleSetActionType["Log"] = "Log";
    /** Redirect */
    KnownManagedRuleSetActionType["Redirect"] = "Redirect";
})(KnownManagedRuleSetActionType || (KnownManagedRuleSetActionType = {}));
/** Known values of {@link ManagedRuleExclusionMatchVariable} that the service accepts. */
export var KnownManagedRuleExclusionMatchVariable;
(function (KnownManagedRuleExclusionMatchVariable) {
    /** RequestHeaderNames */
    KnownManagedRuleExclusionMatchVariable["RequestHeaderNames"] = "RequestHeaderNames";
    /** RequestCookieNames */
    KnownManagedRuleExclusionMatchVariable["RequestCookieNames"] = "RequestCookieNames";
    /** QueryStringArgNames */
    KnownManagedRuleExclusionMatchVariable["QueryStringArgNames"] = "QueryStringArgNames";
    /** RequestBodyPostArgNames */
    KnownManagedRuleExclusionMatchVariable["RequestBodyPostArgNames"] = "RequestBodyPostArgNames";
    /** RequestBodyJsonArgNames */
    KnownManagedRuleExclusionMatchVariable["RequestBodyJsonArgNames"] = "RequestBodyJsonArgNames";
})(KnownManagedRuleExclusionMatchVariable || (KnownManagedRuleExclusionMatchVariable = {}));
/** Known values of {@link ManagedRuleExclusionSelectorMatchOperator} that the service accepts. */
export var KnownManagedRuleExclusionSelectorMatchOperator;
(function (KnownManagedRuleExclusionSelectorMatchOperator) {
    /** Equals */
    KnownManagedRuleExclusionSelectorMatchOperator["Equals"] = "Equals";
    /** Contains */
    KnownManagedRuleExclusionSelectorMatchOperator["Contains"] = "Contains";
    /** StartsWith */
    KnownManagedRuleExclusionSelectorMatchOperator["StartsWith"] = "StartsWith";
    /** EndsWith */
    KnownManagedRuleExclusionSelectorMatchOperator["EndsWith"] = "EndsWith";
    /** EqualsAny */
    KnownManagedRuleExclusionSelectorMatchOperator["EqualsAny"] = "EqualsAny";
})(KnownManagedRuleExclusionSelectorMatchOperator || (KnownManagedRuleExclusionSelectorMatchOperator = {}));
/** Known values of {@link ManagedRuleEnabledState} that the service accepts. */
export var KnownManagedRuleEnabledState;
(function (KnownManagedRuleEnabledState) {
    /** Disabled */
    KnownManagedRuleEnabledState["Disabled"] = "Disabled";
    /** Enabled */
    KnownManagedRuleEnabledState["Enabled"] = "Enabled";
})(KnownManagedRuleEnabledState || (KnownManagedRuleEnabledState = {}));
/** Known values of {@link PolicyResourceState} that the service accepts. */
export var KnownPolicyResourceState;
(function (KnownPolicyResourceState) {
    /** Creating */
    KnownPolicyResourceState["Creating"] = "Creating";
    /** Enabling */
    KnownPolicyResourceState["Enabling"] = "Enabling";
    /** Enabled */
    KnownPolicyResourceState["Enabled"] = "Enabled";
    /** Disabling */
    KnownPolicyResourceState["Disabling"] = "Disabling";
    /** Disabled */
    KnownPolicyResourceState["Disabled"] = "Disabled";
    /** Deleting */
    KnownPolicyResourceState["Deleting"] = "Deleting";
})(KnownPolicyResourceState || (KnownPolicyResourceState = {}));
/** Known values of {@link SkuName} that the service accepts. */
export var KnownSkuName;
(function (KnownSkuName) {
    /** ClassicAzureFrontDoor */
    KnownSkuName["ClassicAzureFrontDoor"] = "Classic_AzureFrontDoor";
    /** StandardAzureFrontDoor */
    KnownSkuName["StandardAzureFrontDoor"] = "Standard_AzureFrontDoor";
    /** PremiumAzureFrontDoor */
    KnownSkuName["PremiumAzureFrontDoor"] = "Premium_AzureFrontDoor";
})(KnownSkuName || (KnownSkuName = {}));
/** Known values of {@link Availability} that the service accepts. */
export var KnownAvailability;
(function (KnownAvailability) {
    /** Available */
    KnownAvailability["Available"] = "Available";
    /** Unavailable */
    KnownAvailability["Unavailable"] = "Unavailable";
})(KnownAvailability || (KnownAvailability = {}));
/** Known values of {@link FrontDoorResourceState} that the service accepts. */
export var KnownFrontDoorResourceState;
(function (KnownFrontDoorResourceState) {
    /** Creating */
    KnownFrontDoorResourceState["Creating"] = "Creating";
    /** Enabling */
    KnownFrontDoorResourceState["Enabling"] = "Enabling";
    /** Enabled */
    KnownFrontDoorResourceState["Enabled"] = "Enabled";
    /** Disabling */
    KnownFrontDoorResourceState["Disabling"] = "Disabling";
    /** Disabled */
    KnownFrontDoorResourceState["Disabled"] = "Disabled";
    /** Deleting */
    KnownFrontDoorResourceState["Deleting"] = "Deleting";
    /** Migrating */
    KnownFrontDoorResourceState["Migrating"] = "Migrating";
    /** Migrated */
    KnownFrontDoorResourceState["Migrated"] = "Migrated";
})(KnownFrontDoorResourceState || (KnownFrontDoorResourceState = {}));
/** Known values of {@link HeaderActionType} that the service accepts. */
export var KnownHeaderActionType;
(function (KnownHeaderActionType) {
    /** Append */
    KnownHeaderActionType["Append"] = "Append";
    /** Delete */
    KnownHeaderActionType["Delete"] = "Delete";
    /** Overwrite */
    KnownHeaderActionType["Overwrite"] = "Overwrite";
})(KnownHeaderActionType || (KnownHeaderActionType = {}));
/** Known values of {@link RulesEngineMatchVariable} that the service accepts. */
export var KnownRulesEngineMatchVariable;
(function (KnownRulesEngineMatchVariable) {
    /** IsMobile */
    KnownRulesEngineMatchVariable["IsMobile"] = "IsMobile";
    /** RemoteAddr */
    KnownRulesEngineMatchVariable["RemoteAddr"] = "RemoteAddr";
    /** RequestMethod */
    KnownRulesEngineMatchVariable["RequestMethod"] = "RequestMethod";
    /** QueryString */
    KnownRulesEngineMatchVariable["QueryString"] = "QueryString";
    /** PostArgs */
    KnownRulesEngineMatchVariable["PostArgs"] = "PostArgs";
    /** RequestUri */
    KnownRulesEngineMatchVariable["RequestUri"] = "RequestUri";
    /** RequestPath */
    KnownRulesEngineMatchVariable["RequestPath"] = "RequestPath";
    /** RequestFilename */
    KnownRulesEngineMatchVariable["RequestFilename"] = "RequestFilename";
    /** RequestFilenameExtension */
    KnownRulesEngineMatchVariable["RequestFilenameExtension"] = "RequestFilenameExtension";
    /** RequestHeader */
    KnownRulesEngineMatchVariable["RequestHeader"] = "RequestHeader";
    /** RequestBody */
    KnownRulesEngineMatchVariable["RequestBody"] = "RequestBody";
    /** RequestScheme */
    KnownRulesEngineMatchVariable["RequestScheme"] = "RequestScheme";
})(KnownRulesEngineMatchVariable || (KnownRulesEngineMatchVariable = {}));
/** Known values of {@link RulesEngineOperator} that the service accepts. */
export var KnownRulesEngineOperator;
(function (KnownRulesEngineOperator) {
    /** Any */
    KnownRulesEngineOperator["Any"] = "Any";
    /** IPMatch */
    KnownRulesEngineOperator["IPMatch"] = "IPMatch";
    /** GeoMatch */
    KnownRulesEngineOperator["GeoMatch"] = "GeoMatch";
    /** Equal */
    KnownRulesEngineOperator["Equal"] = "Equal";
    /** Contains */
    KnownRulesEngineOperator["Contains"] = "Contains";
    /** LessThan */
    KnownRulesEngineOperator["LessThan"] = "LessThan";
    /** GreaterThan */
    KnownRulesEngineOperator["GreaterThan"] = "GreaterThan";
    /** LessThanOrEqual */
    KnownRulesEngineOperator["LessThanOrEqual"] = "LessThanOrEqual";
    /** GreaterThanOrEqual */
    KnownRulesEngineOperator["GreaterThanOrEqual"] = "GreaterThanOrEqual";
    /** BeginsWith */
    KnownRulesEngineOperator["BeginsWith"] = "BeginsWith";
    /** EndsWith */
    KnownRulesEngineOperator["EndsWith"] = "EndsWith";
})(KnownRulesEngineOperator || (KnownRulesEngineOperator = {}));
/** Known values of {@link Transform} that the service accepts. */
export var KnownTransform;
(function (KnownTransform) {
    /** Lowercase */
    KnownTransform["Lowercase"] = "Lowercase";
    /** Uppercase */
    KnownTransform["Uppercase"] = "Uppercase";
    /** Trim */
    KnownTransform["Trim"] = "Trim";
    /** UrlDecode */
    KnownTransform["UrlDecode"] = "UrlDecode";
    /** UrlEncode */
    KnownTransform["UrlEncode"] = "UrlEncode";
    /** RemoveNulls */
    KnownTransform["RemoveNulls"] = "RemoveNulls";
})(KnownTransform || (KnownTransform = {}));
/** Known values of {@link MatchProcessingBehavior} that the service accepts. */
export var KnownMatchProcessingBehavior;
(function (KnownMatchProcessingBehavior) {
    /** Continue */
    KnownMatchProcessingBehavior["Continue"] = "Continue";
    /** Stop */
    KnownMatchProcessingBehavior["Stop"] = "Stop";
})(KnownMatchProcessingBehavior || (KnownMatchProcessingBehavior = {}));
/** Known values of {@link FrontDoorProtocol} that the service accepts. */
export var KnownFrontDoorProtocol;
(function (KnownFrontDoorProtocol) {
    /** Http */
    KnownFrontDoorProtocol["Http"] = "Http";
    /** Https */
    KnownFrontDoorProtocol["Https"] = "Https";
})(KnownFrontDoorProtocol || (KnownFrontDoorProtocol = {}));
/** Known values of {@link RoutingRuleEnabledState} that the service accepts. */
export var KnownRoutingRuleEnabledState;
(function (KnownRoutingRuleEnabledState) {
    /** Enabled */
    KnownRoutingRuleEnabledState["Enabled"] = "Enabled";
    /** Disabled */
    KnownRoutingRuleEnabledState["Disabled"] = "Disabled";
})(KnownRoutingRuleEnabledState || (KnownRoutingRuleEnabledState = {}));
/** Known values of {@link FrontDoorHealthProbeMethod} that the service accepts. */
export var KnownFrontDoorHealthProbeMethod;
(function (KnownFrontDoorHealthProbeMethod) {
    /** GET */
    KnownFrontDoorHealthProbeMethod["GET"] = "GET";
    /** Head */
    KnownFrontDoorHealthProbeMethod["Head"] = "HEAD";
})(KnownFrontDoorHealthProbeMethod || (KnownFrontDoorHealthProbeMethod = {}));
/** Known values of {@link HealthProbeEnabled} that the service accepts. */
export var KnownHealthProbeEnabled;
(function (KnownHealthProbeEnabled) {
    /** Enabled */
    KnownHealthProbeEnabled["Enabled"] = "Enabled";
    /** Disabled */
    KnownHealthProbeEnabled["Disabled"] = "Disabled";
})(KnownHealthProbeEnabled || (KnownHealthProbeEnabled = {}));
/** Known values of {@link PrivateEndpointStatus} that the service accepts. */
export var KnownPrivateEndpointStatus;
(function (KnownPrivateEndpointStatus) {
    /** Pending */
    KnownPrivateEndpointStatus["Pending"] = "Pending";
    /** Approved */
    KnownPrivateEndpointStatus["Approved"] = "Approved";
    /** Rejected */
    KnownPrivateEndpointStatus["Rejected"] = "Rejected";
    /** Disconnected */
    KnownPrivateEndpointStatus["Disconnected"] = "Disconnected";
    /** Timeout */
    KnownPrivateEndpointStatus["Timeout"] = "Timeout";
})(KnownPrivateEndpointStatus || (KnownPrivateEndpointStatus = {}));
/** Known values of {@link BackendEnabledState} that the service accepts. */
export var KnownBackendEnabledState;
(function (KnownBackendEnabledState) {
    /** Enabled */
    KnownBackendEnabledState["Enabled"] = "Enabled";
    /** Disabled */
    KnownBackendEnabledState["Disabled"] = "Disabled";
})(KnownBackendEnabledState || (KnownBackendEnabledState = {}));
/** Known values of {@link CustomHttpsProvisioningState} that the service accepts. */
export var KnownCustomHttpsProvisioningState;
(function (KnownCustomHttpsProvisioningState) {
    /** Enabling */
    KnownCustomHttpsProvisioningState["Enabling"] = "Enabling";
    /** Enabled */
    KnownCustomHttpsProvisioningState["Enabled"] = "Enabled";
    /** Disabling */
    KnownCustomHttpsProvisioningState["Disabling"] = "Disabling";
    /** Disabled */
    KnownCustomHttpsProvisioningState["Disabled"] = "Disabled";
    /** Failed */
    KnownCustomHttpsProvisioningState["Failed"] = "Failed";
})(KnownCustomHttpsProvisioningState || (KnownCustomHttpsProvisioningState = {}));
/** Known values of {@link CustomHttpsProvisioningSubstate} that the service accepts. */
export var KnownCustomHttpsProvisioningSubstate;
(function (KnownCustomHttpsProvisioningSubstate) {
    /** SubmittingDomainControlValidationRequest */
    KnownCustomHttpsProvisioningSubstate["SubmittingDomainControlValidationRequest"] = "SubmittingDomainControlValidationRequest";
    /** PendingDomainControlValidationREquestApproval */
    KnownCustomHttpsProvisioningSubstate["PendingDomainControlValidationREquestApproval"] = "PendingDomainControlValidationREquestApproval";
    /** DomainControlValidationRequestApproved */
    KnownCustomHttpsProvisioningSubstate["DomainControlValidationRequestApproved"] = "DomainControlValidationRequestApproved";
    /** DomainControlValidationRequestRejected */
    KnownCustomHttpsProvisioningSubstate["DomainControlValidationRequestRejected"] = "DomainControlValidationRequestRejected";
    /** DomainControlValidationRequestTimedOut */
    KnownCustomHttpsProvisioningSubstate["DomainControlValidationRequestTimedOut"] = "DomainControlValidationRequestTimedOut";
    /** IssuingCertificate */
    KnownCustomHttpsProvisioningSubstate["IssuingCertificate"] = "IssuingCertificate";
    /** DeployingCertificate */
    KnownCustomHttpsProvisioningSubstate["DeployingCertificate"] = "DeployingCertificate";
    /** CertificateDeployed */
    KnownCustomHttpsProvisioningSubstate["CertificateDeployed"] = "CertificateDeployed";
    /** DeletingCertificate */
    KnownCustomHttpsProvisioningSubstate["DeletingCertificate"] = "DeletingCertificate";
    /** CertificateDeleted */
    KnownCustomHttpsProvisioningSubstate["CertificateDeleted"] = "CertificateDeleted";
})(KnownCustomHttpsProvisioningSubstate || (KnownCustomHttpsProvisioningSubstate = {}));
/** Known values of {@link FrontDoorCertificateSource} that the service accepts. */
export var KnownFrontDoorCertificateSource;
(function (KnownFrontDoorCertificateSource) {
    /** AzureKeyVault */
    KnownFrontDoorCertificateSource["AzureKeyVault"] = "AzureKeyVault";
    /** FrontDoor */
    KnownFrontDoorCertificateSource["FrontDoor"] = "FrontDoor";
})(KnownFrontDoorCertificateSource || (KnownFrontDoorCertificateSource = {}));
/** Known values of {@link FrontDoorTlsProtocolType} that the service accepts. */
export var KnownFrontDoorTlsProtocolType;
(function (KnownFrontDoorTlsProtocolType) {
    /** ServerNameIndication */
    KnownFrontDoorTlsProtocolType["ServerNameIndication"] = "ServerNameIndication";
})(KnownFrontDoorTlsProtocolType || (KnownFrontDoorTlsProtocolType = {}));
/** Known values of {@link MinimumTLSVersion} that the service accepts. */
export var KnownMinimumTLSVersion;
(function (KnownMinimumTLSVersion) {
    /** One0 */
    KnownMinimumTLSVersion["One0"] = "1.0";
    /** One2 */
    KnownMinimumTLSVersion["One2"] = "1.2";
})(KnownMinimumTLSVersion || (KnownMinimumTLSVersion = {}));
/** Known values of {@link FrontDoorCertificateType} that the service accepts. */
export var KnownFrontDoorCertificateType;
(function (KnownFrontDoorCertificateType) {
    /** Dedicated */
    KnownFrontDoorCertificateType["Dedicated"] = "Dedicated";
})(KnownFrontDoorCertificateType || (KnownFrontDoorCertificateType = {}));
/** Known values of {@link SessionAffinityEnabledState} that the service accepts. */
export var KnownSessionAffinityEnabledState;
(function (KnownSessionAffinityEnabledState) {
    /** Enabled */
    KnownSessionAffinityEnabledState["Enabled"] = "Enabled";
    /** Disabled */
    KnownSessionAffinityEnabledState["Disabled"] = "Disabled";
})(KnownSessionAffinityEnabledState || (KnownSessionAffinityEnabledState = {}));
/** Known values of {@link EnforceCertificateNameCheckEnabledState} that the service accepts. */
export var KnownEnforceCertificateNameCheckEnabledState;
(function (KnownEnforceCertificateNameCheckEnabledState) {
    /** Enabled */
    KnownEnforceCertificateNameCheckEnabledState["Enabled"] = "Enabled";
    /** Disabled */
    KnownEnforceCertificateNameCheckEnabledState["Disabled"] = "Disabled";
})(KnownEnforceCertificateNameCheckEnabledState || (KnownEnforceCertificateNameCheckEnabledState = {}));
/** Known values of {@link FrontDoorEnabledState} that the service accepts. */
export var KnownFrontDoorEnabledState;
(function (KnownFrontDoorEnabledState) {
    /** Enabled */
    KnownFrontDoorEnabledState["Enabled"] = "Enabled";
    /** Disabled */
    KnownFrontDoorEnabledState["Disabled"] = "Disabled";
})(KnownFrontDoorEnabledState || (KnownFrontDoorEnabledState = {}));
/** Known values of {@link NetworkExperimentResourceState} that the service accepts. */
export var KnownNetworkExperimentResourceState;
(function (KnownNetworkExperimentResourceState) {
    /** Creating */
    KnownNetworkExperimentResourceState["Creating"] = "Creating";
    /** Enabling */
    KnownNetworkExperimentResourceState["Enabling"] = "Enabling";
    /** Enabled */
    KnownNetworkExperimentResourceState["Enabled"] = "Enabled";
    /** Disabling */
    KnownNetworkExperimentResourceState["Disabling"] = "Disabling";
    /** Disabled */
    KnownNetworkExperimentResourceState["Disabled"] = "Disabled";
    /** Deleting */
    KnownNetworkExperimentResourceState["Deleting"] = "Deleting";
})(KnownNetworkExperimentResourceState || (KnownNetworkExperimentResourceState = {}));
/** Known values of {@link State} that the service accepts. */
export var KnownState;
(function (KnownState) {
    /** Enabled */
    KnownState["Enabled"] = "Enabled";
    /** Disabled */
    KnownState["Disabled"] = "Disabled";
})(KnownState || (KnownState = {}));
/** Known values of {@link EndpointType} that the service accepts. */
export var KnownEndpointType;
(function (KnownEndpointType) {
    /** AFD */
    KnownEndpointType["AFD"] = "AFD";
    /** AzureRegion */
    KnownEndpointType["AzureRegion"] = "AzureRegion";
    /** CDN */
    KnownEndpointType["CDN"] = "CDN";
    /** ATM */
    KnownEndpointType["ATM"] = "ATM";
})(KnownEndpointType || (KnownEndpointType = {}));
/** Known values of {@link LatencyScorecardAggregationInterval} that the service accepts. */
export var KnownLatencyScorecardAggregationInterval;
(function (KnownLatencyScorecardAggregationInterval) {
    /** Daily */
    KnownLatencyScorecardAggregationInterval["Daily"] = "Daily";
    /** Weekly */
    KnownLatencyScorecardAggregationInterval["Weekly"] = "Weekly";
    /** Monthly */
    KnownLatencyScorecardAggregationInterval["Monthly"] = "Monthly";
})(KnownLatencyScorecardAggregationInterval || (KnownLatencyScorecardAggregationInterval = {}));
/** Known values of {@link TimeseriesAggregationInterval} that the service accepts. */
export var KnownTimeseriesAggregationInterval;
(function (KnownTimeseriesAggregationInterval) {
    /** Hourly */
    KnownTimeseriesAggregationInterval["Hourly"] = "Hourly";
    /** Daily */
    KnownTimeseriesAggregationInterval["Daily"] = "Daily";
})(KnownTimeseriesAggregationInterval || (KnownTimeseriesAggregationInterval = {}));
/** Known values of {@link TimeseriesType} that the service accepts. */
export var KnownTimeseriesType;
(function (KnownTimeseriesType) {
    /** MeasurementCounts */
    KnownTimeseriesType["MeasurementCounts"] = "MeasurementCounts";
    /** LatencyP50 */
    KnownTimeseriesType["LatencyP50"] = "LatencyP50";
    /** LatencyP75 */
    KnownTimeseriesType["LatencyP75"] = "LatencyP75";
    /** LatencyP95 */
    KnownTimeseriesType["LatencyP95"] = "LatencyP95";
})(KnownTimeseriesType || (KnownTimeseriesType = {}));
/** Known values of {@link AggregationInterval} that the service accepts. */
export var KnownAggregationInterval;
(function (KnownAggregationInterval) {
    /** Hourly */
    KnownAggregationInterval["Hourly"] = "Hourly";
    /** Daily */
    KnownAggregationInterval["Daily"] = "Daily";
})(KnownAggregationInterval || (KnownAggregationInterval = {}));
/** Known values of {@link NetworkOperationStatus} that the service accepts. */
export var KnownNetworkOperationStatus;
(function (KnownNetworkOperationStatus) {
    /** InProgress */
    KnownNetworkOperationStatus["InProgress"] = "InProgress";
    /** Succeeded */
    KnownNetworkOperationStatus["Succeeded"] = "Succeeded";
    /** Failed */
    KnownNetworkOperationStatus["Failed"] = "Failed";
})(KnownNetworkOperationStatus || (KnownNetworkOperationStatus = {}));
/** Known values of {@link FrontDoorForwardingProtocol} that the service accepts. */
export var KnownFrontDoorForwardingProtocol;
(function (KnownFrontDoorForwardingProtocol) {
    /** HttpOnly */
    KnownFrontDoorForwardingProtocol["HttpOnly"] = "HttpOnly";
    /** HttpsOnly */
    KnownFrontDoorForwardingProtocol["HttpsOnly"] = "HttpsOnly";
    /** MatchRequest */
    KnownFrontDoorForwardingProtocol["MatchRequest"] = "MatchRequest";
})(KnownFrontDoorForwardingProtocol || (KnownFrontDoorForwardingProtocol = {}));
/** Known values of {@link FrontDoorQuery} that the service accepts. */
export var KnownFrontDoorQuery;
(function (KnownFrontDoorQuery) {
    /** StripNone */
    KnownFrontDoorQuery["StripNone"] = "StripNone";
    /** StripAll */
    KnownFrontDoorQuery["StripAll"] = "StripAll";
    /** StripOnly */
    KnownFrontDoorQuery["StripOnly"] = "StripOnly";
    /** StripAllExcept */
    KnownFrontDoorQuery["StripAllExcept"] = "StripAllExcept";
})(KnownFrontDoorQuery || (KnownFrontDoorQuery = {}));
/** Known values of {@link DynamicCompressionEnabled} that the service accepts. */
export var KnownDynamicCompressionEnabled;
(function (KnownDynamicCompressionEnabled) {
    /** Enabled */
    KnownDynamicCompressionEnabled["Enabled"] = "Enabled";
    /** Disabled */
    KnownDynamicCompressionEnabled["Disabled"] = "Disabled";
})(KnownDynamicCompressionEnabled || (KnownDynamicCompressionEnabled = {}));
/** Known values of {@link FrontDoorRedirectType} that the service accepts. */
export var KnownFrontDoorRedirectType;
(function (KnownFrontDoorRedirectType) {
    /** Moved */
    KnownFrontDoorRedirectType["Moved"] = "Moved";
    /** Found */
    KnownFrontDoorRedirectType["Found"] = "Found";
    /** TemporaryRedirect */
    KnownFrontDoorRedirectType["TemporaryRedirect"] = "TemporaryRedirect";
    /** PermanentRedirect */
    KnownFrontDoorRedirectType["PermanentRedirect"] = "PermanentRedirect";
})(KnownFrontDoorRedirectType || (KnownFrontDoorRedirectType = {}));
/** Known values of {@link FrontDoorRedirectProtocol} that the service accepts. */
export var KnownFrontDoorRedirectProtocol;
(function (KnownFrontDoorRedirectProtocol) {
    /** HttpOnly */
    KnownFrontDoorRedirectProtocol["HttpOnly"] = "HttpOnly";
    /** HttpsOnly */
    KnownFrontDoorRedirectProtocol["HttpsOnly"] = "HttpsOnly";
    /** MatchRequest */
    KnownFrontDoorRedirectProtocol["MatchRequest"] = "MatchRequest";
})(KnownFrontDoorRedirectProtocol || (KnownFrontDoorRedirectProtocol = {}));
