// @azure/arm-features/dist-esm/test/features_example.spec.js
/*
 * Copyright (c) Microsoft Corporation.
 * Licensed under the MIT License.
 *
 * Code generated by Microsoft (R) AutoRest Code Generator.
 * Changes may cause incorrect behavior and will be lost if the code is regenerated.
 */
import { __asyncValues, __awaiter } from "tslib";
import { env, Recorder, isPlaybackMode, } from "@azure-tools/test-recorder";
import { createTestCredential } from "@azure-tools/test-credential";
import { assert } from "chai";
import { FeatureClient } from "../src/featureClient";
const replaceableVariables = {
    AZURE_CLIENT_ID: "azure_client_id",
    AZURE_CLIENT_SECRET: "azure_client_secret",
    AZURE_TENANT_ID: "88888888-8888-8888-8888-888888888888",
    SUBSCRIPTION_ID: "azure_subscription_id"
};
const recorderOptions = {
    envSetupForPlayback: replaceableVariables
};
export const testPollingOptions = {
    updateIntervalInMs: isPlaybackMode() ? 0 : undefined,
};
describe("Features test", () => {
    let recorder;
    let subscriptionId;
    let client;
    let location;
    let resourceGroup;
    beforeEach(function () {
        return __awaiter(this, void 0, void 0, function* () {
            recorder = new Recorder(this.currentTest);
            yield recorder.start(recorderOptions);
            subscriptionId = env.SUBSCRIPTION_ID || '';
            // This is an example of how the environment variables are used
            const credential = createTestCredential();
            client = new FeatureClient(credential, subscriptionId, recorder.configureClientOptions({}));
            location = "eastus";
            resourceGroup = "myjstest";
        });
    });
    afterEach(function () {
        return __awaiter(this, void 0, void 0, function* () {
            yield recorder.stop();
        });
    });
    it("features listall test", function () {
        var e_1, _a;
        return __awaiter(this, void 0, void 0, function* () {
            const arrayList = [];
            try {
                for (var _b = __asyncValues(client.features.listAll()), _c; _c = yield _b.next(), !_c.done;) {
                    const item = _c.value;
                    arrayList.push(item);
                }
            }
            catch (e_1_1) { e_1 = { error: e_1_1 }; }
            finally {
                try {
                    if (_c && !_c.done && (_a = _b.return)) yield _a.call(_b);
                }
                finally { if (e_1) throw e_1.error; }
            }
            assert.notEqual(arrayList.length, 0);
        });
    });
    it("features list test", function () {
        var e_2, _a;
        return __awaiter(this, void 0, void 0, function* () {
            const arrayList = [];
            try {
                for (var _b = __asyncValues(client.features.list("Microsoft.Compute")), _c; _c = yield _b.next(), !_c.done;) {
                    const item = _c.value;
                    arrayList.push(item);
                }
            }
            catch (e_2_1) { e_2 = { error: e_2_1 }; }
            finally {
                try {
                    if (_c && !_c.done && (_a = _b.return)) yield _a.call(_b);
                }
                finally { if (e_2) throw e_2.error; }
            }
            assert.notEqual(arrayList.length, 0);
        });
    });
    it("features get test", function () {
        var e_3, _a;
        return __awaiter(this, void 0, void 0, function* () {
            const arrayList = new Array();
            try {
                for (var _b = __asyncValues(client.features.list("Microsoft.Compute")), _c; _c = yield _b.next(), !_c.done;) {
                    const item = _c.value;
                    arrayList.push(item);
                }
            }
            catch (e_3_1) { e_3 = { error: e_3_1 }; }
            finally {
                try {
                    if (_c && !_c.done && (_a = _b.return)) yield _a.call(_b);
                }
                finally { if (e_3) throw e_3.error; }
            }
            const featureName = arrayList[0].name.split("/")[1];
            const feature = yield client.features.get("Microsoft.Compute", featureName);
            assert.equal(feature.name, "Microsoft.Compute/" + featureName);
        });
    });
});
