// @azure/arm-cdn/dist-esm/src/cdnManagementClient.js
/*
 * Copyright (c) Microsoft Corporation.
 * Licensed under the MIT License.
 *
 * Code generated by Microsoft (R) AutoRest Code Generator.
 * Changes may cause incorrect behavior and will be lost if the code is regenerated.
 */
import { __awaiter } from "tslib";
import * as coreClient from "@azure/core-client";
import * as coreRestPipeline from "@azure/core-rest-pipeline";
import { AfdProfilesImpl, AfdCustomDomainsImpl, AfdEndpointsImpl, AfdOriginGroupsImpl, AfdOriginsImpl, RoutesImpl, RuleSetsImpl, RulesImpl, SecurityPoliciesImpl, SecretsImpl, LogAnalyticsImpl, ProfilesImpl, EndpointsImpl, OriginsImpl, OriginGroupsImpl, CustomDomainsImpl, ResourceUsageOperationsImpl, OperationsImpl, EdgeNodesImpl, PoliciesImpl, ManagedRuleSetsImpl, } from "./operations";
import * as Parameters from "./models/parameters";
import * as Mappers from "./models/mappers";
export class CdnManagementClient extends coreClient.ServiceClient {
    constructor(credentials, subscriptionIdOrOptions, options) {
        var _a, _b, _c;
        if (credentials === undefined) {
            throw new Error("'credentials' cannot be null");
        }
        let subscriptionId;
        if (typeof subscriptionIdOrOptions === "string") {
            subscriptionId = subscriptionIdOrOptions;
        }
        else if (typeof subscriptionIdOrOptions === "object") {
            options = subscriptionIdOrOptions;
        }
        // Initializing default values for options
        if (!options) {
            options = {};
        }
        const defaults = {
            requestContentType: "application/json; charset=utf-8",
            credential: credentials,
        };
        const packageDetails = `azsdk-js-arm-cdn/9.1.0`;
        const userAgentPrefix = options.userAgentOptions && options.userAgentOptions.userAgentPrefix
            ? `${options.userAgentOptions.userAgentPrefix} ${packageDetails}`
            : `${packageDetails}`;
        const optionsWithDefaults = Object.assign(Object.assign(Object.assign({}, defaults), options), { userAgentOptions: {
                userAgentPrefix,
            }, endpoint: (_b = (_a = options.endpoint) !== null && _a !== void 0 ? _a : options.baseUri) !== null && _b !== void 0 ? _b : "https://management.azure.com" });
        super(optionsWithDefaults);
        let bearerTokenAuthenticationPolicyFound = false;
        if ((options === null || options === void 0 ? void 0 : options.pipeline) && options.pipeline.getOrderedPolicies().length > 0) {
            const pipelinePolicies = options.pipeline.getOrderedPolicies();
            bearerTokenAuthenticationPolicyFound = pipelinePolicies.some((pipelinePolicy) => pipelinePolicy.name ===
                coreRestPipeline.bearerTokenAuthenticationPolicyName);
        }
        if (!options ||
            !options.pipeline ||
            options.pipeline.getOrderedPolicies().length == 0 ||
            !bearerTokenAuthenticationPolicyFound) {
            this.pipeline.removePolicy({
                name: coreRestPipeline.bearerTokenAuthenticationPolicyName,
            });
            this.pipeline.addPolicy(coreRestPipeline.bearerTokenAuthenticationPolicy({
                credential: credentials,
                scopes: (_c = optionsWithDefaults.credentialScopes) !== null && _c !== void 0 ? _c : `${optionsWithDefaults.endpoint}/.default`,
                challengeCallbacks: {
                    authorizeRequestOnChallenge: coreClient.authorizeRequestOnClaimChallenge,
                },
            }));
        }
        // Parameter assignments
        this.subscriptionId = subscriptionId;
        // Assigning values to Constant parameters
        this.$host = options.$host || "https://management.azure.com";
        this.apiVersion = options.apiVersion || "2024-02-01";
        this.afdProfiles = new AfdProfilesImpl(this);
        this.afdCustomDomains = new AfdCustomDomainsImpl(this);
        this.afdEndpoints = new AfdEndpointsImpl(this);
        this.afdOriginGroups = new AfdOriginGroupsImpl(this);
        this.afdOrigins = new AfdOriginsImpl(this);
        this.routes = new RoutesImpl(this);
        this.ruleSets = new RuleSetsImpl(this);
        this.rules = new RulesImpl(this);
        this.securityPolicies = new SecurityPoliciesImpl(this);
        this.secrets = new SecretsImpl(this);
        this.logAnalytics = new LogAnalyticsImpl(this);
        this.profiles = new ProfilesImpl(this);
        this.endpoints = new EndpointsImpl(this);
        this.origins = new OriginsImpl(this);
        this.originGroups = new OriginGroupsImpl(this);
        this.customDomains = new CustomDomainsImpl(this);
        this.resourceUsageOperations = new ResourceUsageOperationsImpl(this);
        this.operations = new OperationsImpl(this);
        this.edgeNodes = new EdgeNodesImpl(this);
        this.policies = new PoliciesImpl(this);
        this.managedRuleSets = new ManagedRuleSetsImpl(this);
        this.addCustomApiVersionPolicy(options.apiVersion);
    }
    /** A function that adds a policy that sets the api-version (or equivalent) to reflect the library version. */
    addCustomApiVersionPolicy(apiVersion) {
        if (!apiVersion) {
            return;
        }
        const apiVersionPolicy = {
            name: "CustomApiVersionPolicy",
            sendRequest(request, next) {
                return __awaiter(this, void 0, void 0, function* () {
                    const param = request.url.split("?");
                    if (param.length > 1) {
                        const newParams = param[1].split("&").map((item) => {
                            if (item.indexOf("api-version") > -1) {
                                return "api-version=" + apiVersion;
                            }
                            else {
                                return item;
                            }
                        });
                        request.url = param[0] + "?" + newParams.join("&");
                    }
                    return next(request);
                });
            },
        };
        this.pipeline.addPolicy(apiVersionPolicy);
    }
    /**
     * Check the availability of a resource name. This is needed for resources where name is globally
     * unique, such as a afdx endpoint.
     * @param resourceGroupName Name of the Resource group within the Azure subscription.
     * @param checkEndpointNameAvailabilityInput Input to check.
     * @param options The options parameters.
     */
    checkEndpointNameAvailability(resourceGroupName, checkEndpointNameAvailabilityInput, options) {
        return this.sendOperationRequest({ resourceGroupName, checkEndpointNameAvailabilityInput, options }, checkEndpointNameAvailabilityOperationSpec);
    }
    /**
     * Check the availability of a resource name. This is needed for resources where name is globally
     * unique, such as a CDN endpoint.
     * @param checkNameAvailabilityInput Input to check.
     * @param options The options parameters.
     */
    checkNameAvailability(checkNameAvailabilityInput, options) {
        return this.sendOperationRequest({ checkNameAvailabilityInput, options }, checkNameAvailabilityOperationSpec);
    }
    /**
     * Check the availability of a resource name. This is needed for resources where name is globally
     * unique, such as a CDN endpoint.
     * @param checkNameAvailabilityInput Input to check.
     * @param options The options parameters.
     */
    checkNameAvailabilityWithSubscription(checkNameAvailabilityInput, options) {
        return this.sendOperationRequest({ checkNameAvailabilityInput, options }, checkNameAvailabilityWithSubscriptionOperationSpec);
    }
    /**
     * Check if the probe path is a valid path and the file can be accessed. Probe path is the path to a
     * file hosted on the origin server to help accelerate the delivery of dynamic content via the CDN
     * endpoint. This path is relative to the origin path specified in the endpoint configuration.
     * @param validateProbeInput Input to check.
     * @param options The options parameters.
     */
    validateProbe(validateProbeInput, options) {
        return this.sendOperationRequest({ validateProbeInput, options }, validateProbeOperationSpec);
    }
}
// Operation Specifications
const serializer = coreClient.createSerializer(Mappers, /* isXml */ false);
const checkEndpointNameAvailabilityOperationSpec = {
    path: "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Cdn/checkEndpointNameAvailability",
    httpMethod: "POST",
    responses: {
        200: {
            bodyMapper: Mappers.CheckEndpointNameAvailabilityOutput,
        },
        default: {
            bodyMapper: Mappers.AfdErrorResponse,
        },
    },
    requestBody: Parameters.checkEndpointNameAvailabilityInput,
    queryParameters: [Parameters.apiVersion],
    urlParameters: [
        Parameters.$host,
        Parameters.subscriptionId,
        Parameters.resourceGroupName,
    ],
    headerParameters: [Parameters.contentType, Parameters.accept],
    mediaType: "json",
    serializer,
};
const checkNameAvailabilityOperationSpec = {
    path: "/providers/Microsoft.Cdn/checkNameAvailability",
    httpMethod: "POST",
    responses: {
        200: {
            bodyMapper: Mappers.CheckNameAvailabilityOutput,
        },
        default: {
            bodyMapper: Mappers.ErrorResponse,
        },
    },
    requestBody: Parameters.checkNameAvailabilityInput,
    queryParameters: [Parameters.apiVersion],
    urlParameters: [Parameters.$host],
    headerParameters: [Parameters.contentType, Parameters.accept],
    mediaType: "json",
    serializer,
};
const checkNameAvailabilityWithSubscriptionOperationSpec = {
    path: "/subscriptions/{subscriptionId}/providers/Microsoft.Cdn/checkNameAvailability",
    httpMethod: "POST",
    responses: {
        200: {
            bodyMapper: Mappers.CheckNameAvailabilityOutput,
        },
        default: {
            bodyMapper: Mappers.ErrorResponse,
        },
    },
    requestBody: Parameters.checkNameAvailabilityInput,
    queryParameters: [Parameters.apiVersion],
    urlParameters: [Parameters.$host, Parameters.subscriptionId],
    headerParameters: [Parameters.contentType, Parameters.accept],
    mediaType: "json",
    serializer,
};
const validateProbeOperationSpec = {
    path: "/subscriptions/{subscriptionId}/providers/Microsoft.Cdn/validateProbe",
    httpMethod: "POST",
    responses: {
        200: {
            bodyMapper: Mappers.ValidateProbeOutput,
        },
        default: {
            bodyMapper: Mappers.ErrorResponse,
        },
    },
    requestBody: Parameters.validateProbeInput,
    queryParameters: [Parameters.apiVersion],
    urlParameters: [Parameters.$host, Parameters.subscriptionId],
    headerParameters: [Parameters.contentType, Parameters.accept],
    mediaType: "json",
    serializer,
};
