// @azure/arm-maps/dist-esm/src/models/index.js
/*
 * Copyright (c) Microsoft Corporation.
 * Licensed under the MIT License.
 *
 * Code generated by Microsoft (R) AutoRest Code Generator.
 * Changes may cause incorrect behavior and will be lost if the code is regenerated.
 */
/** Known values of {@link Name} that the service accepts. */
export var KnownName;
(function (KnownName) {
    /** S0 */
    KnownName["S0"] = "S0";
    /** S1 */
    KnownName["S1"] = "S1";
    /** G2 */
    KnownName["G2"] = "G2";
})(KnownName || (KnownName = {}));
/** Known values of {@link Kind} that the service accepts. */
export var KnownKind;
(function (KnownKind) {
    /** Gen1 */
    KnownKind["Gen1"] = "Gen1";
    /** Gen2 */
    KnownKind["Gen2"] = "Gen2";
})(KnownKind || (KnownKind = {}));
/** Known values of {@link CreatedByType} that the service accepts. */
export var KnownCreatedByType;
(function (KnownCreatedByType) {
    /** User */
    KnownCreatedByType["User"] = "User";
    /** Application */
    KnownCreatedByType["Application"] = "Application";
    /** ManagedIdentity */
    KnownCreatedByType["ManagedIdentity"] = "ManagedIdentity";
    /** Key */
    KnownCreatedByType["Key"] = "Key";
})(KnownCreatedByType || (KnownCreatedByType = {}));
/** Known values of {@link ManagedServiceIdentityType} that the service accepts. */
export var KnownManagedServiceIdentityType;
(function (KnownManagedServiceIdentityType) {
    /** None */
    KnownManagedServiceIdentityType["None"] = "None";
    /** SystemAssigned */
    KnownManagedServiceIdentityType["SystemAssigned"] = "SystemAssigned";
    /** UserAssigned */
    KnownManagedServiceIdentityType["UserAssigned"] = "UserAssigned";
    /** SystemAssignedUserAssigned */
    KnownManagedServiceIdentityType["SystemAssignedUserAssigned"] = "SystemAssigned, UserAssigned";
})(KnownManagedServiceIdentityType || (KnownManagedServiceIdentityType = {}));
/** Known values of {@link InfrastructureEncryption} that the service accepts. */
export var KnownInfrastructureEncryption;
(function (KnownInfrastructureEncryption) {
    /** Enabled */
    KnownInfrastructureEncryption["Enabled"] = "enabled";
    /** Disabled */
    KnownInfrastructureEncryption["Disabled"] = "disabled";
})(KnownInfrastructureEncryption || (KnownInfrastructureEncryption = {}));
/** Known values of {@link IdentityType} that the service accepts. */
export var KnownIdentityType;
(function (KnownIdentityType) {
    /** SystemAssignedIdentity */
    KnownIdentityType["SystemAssignedIdentity"] = "systemAssignedIdentity";
    /** UserAssignedIdentity */
    KnownIdentityType["UserAssignedIdentity"] = "userAssignedIdentity";
    /** DelegatedResourceIdentity */
    KnownIdentityType["DelegatedResourceIdentity"] = "delegatedResourceIdentity";
})(KnownIdentityType || (KnownIdentityType = {}));
/** Known values of {@link SigningKey} that the service accepts. */
export var KnownSigningKey;
(function (KnownSigningKey) {
    /** PrimaryKey */
    KnownSigningKey["PrimaryKey"] = "primaryKey";
    /** SecondaryKey */
    KnownSigningKey["SecondaryKey"] = "secondaryKey";
    /** ManagedIdentity */
    KnownSigningKey["ManagedIdentity"] = "managedIdentity";
})(KnownSigningKey || (KnownSigningKey = {}));
/** Known values of {@link KeyType} that the service accepts. */
export var KnownKeyType;
(function (KnownKeyType) {
    /** Primary */
    KnownKeyType["Primary"] = "primary";
    /** Secondary */
    KnownKeyType["Secondary"] = "secondary";
})(KnownKeyType || (KnownKeyType = {}));
