// @azure/arm-eventgrid/dist-esm/src/operations/channels.js
/*
 * Copyright (c) Microsoft Corporation.
 * Licensed under the MIT License.
 *
 * Code generated by Microsoft (R) AutoRest Code Generator.
 * Changes may cause incorrect behavior and will be lost if the code is regenerated.
 */
import { __asyncDelegator, __asyncGenerator, __asyncValues, __await, __awaiter } from "tslib";
import { setContinuationToken } from "../pagingHelper";
import * as coreClient from "@azure/core-client";
import * as Mappers from "../models/mappers";
import * as Parameters from "../models/parameters";
import { LroEngine } from "@azure/core-lro";
import { LroImpl } from "../lroImpl";
/// <reference lib="esnext.asynciterable" />
/** Class containing Channels operations. */
export class ChannelsImpl {
    /**
     * Initialize a new instance of the class Channels class.
     * @param client Reference to the service client
     */
    constructor(client) {
        this.client = client;
    }
    /**
     * List all the channels in a partner namespace.
     * @param resourceGroupName The name of the resource group within the partners subscription.
     * @param partnerNamespaceName Name of the partner namespace.
     * @param options The options parameters.
     */
    listByPartnerNamespace(resourceGroupName, partnerNamespaceName, options) {
        const iter = this.listByPartnerNamespacePagingAll(resourceGroupName, partnerNamespaceName, options);
        return {
            next() {
                return iter.next();
            },
            [Symbol.asyncIterator]() {
                return this;
            },
            byPage: (settings) => {
                if (settings === null || settings === void 0 ? void 0 : settings.maxPageSize) {
                    throw new Error("maxPageSize is not supported by this operation.");
                }
                return this.listByPartnerNamespacePagingPage(resourceGroupName, partnerNamespaceName, options, settings);
            }
        };
    }
    listByPartnerNamespacePagingPage(resourceGroupName, partnerNamespaceName, options, settings) {
        return __asyncGenerator(this, arguments, function* listByPartnerNamespacePagingPage_1() {
            let result;
            let continuationToken = settings === null || settings === void 0 ? void 0 : settings.continuationToken;
            if (!continuationToken) {
                result = yield __await(this._listByPartnerNamespace(resourceGroupName, partnerNamespaceName, options));
                let page = result.value || [];
                continuationToken = result.nextLink;
                setContinuationToken(page, continuationToken);
                yield yield __await(page);
            }
            while (continuationToken) {
                result = yield __await(this._listByPartnerNamespaceNext(resourceGroupName, partnerNamespaceName, continuationToken, options));
                continuationToken = result.nextLink;
                let page = result.value || [];
                setContinuationToken(page, continuationToken);
                yield yield __await(page);
            }
        });
    }
    listByPartnerNamespacePagingAll(resourceGroupName, partnerNamespaceName, options) {
        return __asyncGenerator(this, arguments, function* listByPartnerNamespacePagingAll_1() {
            var e_1, _a;
            try {
                for (var _b = __asyncValues(this.listByPartnerNamespacePagingPage(resourceGroupName, partnerNamespaceName, options)), _c; _c = yield __await(_b.next()), !_c.done;) {
                    const page = _c.value;
                    yield __await(yield* __asyncDelegator(__asyncValues(page)));
                }
            }
            catch (e_1_1) { e_1 = { error: e_1_1 }; }
            finally {
                try {
                    if (_c && !_c.done && (_a = _b.return)) yield __await(_a.call(_b));
                }
                finally { if (e_1) throw e_1.error; }
            }
        });
    }
    /**
     * Get properties of a channel.
     * @param resourceGroupName The name of the resource group within the partners subscription.
     * @param partnerNamespaceName Name of the partner namespace.
     * @param channelName Name of the channel.
     * @param options The options parameters.
     */
    get(resourceGroupName, partnerNamespaceName, channelName, options) {
        return this.client.sendOperationRequest({ resourceGroupName, partnerNamespaceName, channelName, options }, getOperationSpec);
    }
    /**
     * Synchronously creates or updates a new channel with the specified parameters.
     * @param resourceGroupName The name of the resource group within the partners subscription.
     * @param partnerNamespaceName Name of the partner namespace.
     * @param channelName Name of the channel.
     * @param channelInfo Channel information.
     * @param options The options parameters.
     */
    createOrUpdate(resourceGroupName, partnerNamespaceName, channelName, channelInfo, options) {
        return this.client.sendOperationRequest({
            resourceGroupName,
            partnerNamespaceName,
            channelName,
            channelInfo,
            options
        }, createOrUpdateOperationSpec);
    }
    /**
     * Delete an existing channel.
     * @param resourceGroupName The name of the resource group within the partners subscription.
     * @param partnerNamespaceName Name of the partner namespace.
     * @param channelName Name of the channel.
     * @param options The options parameters.
     */
    beginDelete(resourceGroupName, partnerNamespaceName, channelName, options) {
        return __awaiter(this, void 0, void 0, function* () {
            const directSendOperation = (args, spec) => __awaiter(this, void 0, void 0, function* () {
                return this.client.sendOperationRequest(args, spec);
            });
            const sendOperation = (args, spec) => __awaiter(this, void 0, void 0, function* () {
                var _a;
                let currentRawResponse = undefined;
                const providedCallback = (_a = args.options) === null || _a === void 0 ? void 0 : _a.onResponse;
                const callback = (rawResponse, flatResponse) => {
                    currentRawResponse = rawResponse;
                    providedCallback === null || providedCallback === void 0 ? void 0 : providedCallback(rawResponse, flatResponse);
                };
                const updatedArgs = Object.assign(Object.assign({}, args), { options: Object.assign(Object.assign({}, args.options), { onResponse: callback }) });
                const flatResponse = yield directSendOperation(updatedArgs, spec);
                return {
                    flatResponse,
                    rawResponse: {
                        statusCode: currentRawResponse.status,
                        body: currentRawResponse.parsedBody,
                        headers: currentRawResponse.headers.toJSON()
                    }
                };
            });
            const lro = new LroImpl(sendOperation, { resourceGroupName, partnerNamespaceName, channelName, options }, deleteOperationSpec);
            const poller = new LroEngine(lro, {
                resumeFrom: options === null || options === void 0 ? void 0 : options.resumeFrom,
                intervalInMs: options === null || options === void 0 ? void 0 : options.updateIntervalInMs
            });
            yield poller.poll();
            return poller;
        });
    }
    /**
     * Delete an existing channel.
     * @param resourceGroupName The name of the resource group within the partners subscription.
     * @param partnerNamespaceName Name of the partner namespace.
     * @param channelName Name of the channel.
     * @param options The options parameters.
     */
    beginDeleteAndWait(resourceGroupName, partnerNamespaceName, channelName, options) {
        return __awaiter(this, void 0, void 0, function* () {
            const poller = yield this.beginDelete(resourceGroupName, partnerNamespaceName, channelName, options);
            return poller.pollUntilDone();
        });
    }
    /**
     * Synchronously updates a channel with the specified parameters.
     * @param resourceGroupName The name of the resource group within the partners subscription.
     * @param partnerNamespaceName Name of the partner namespace.
     * @param channelName Name of the channel.
     * @param channelUpdateParameters Channel update information.
     * @param options The options parameters.
     */
    update(resourceGroupName, partnerNamespaceName, channelName, channelUpdateParameters, options) {
        return this.client.sendOperationRequest({
            resourceGroupName,
            partnerNamespaceName,
            channelName,
            channelUpdateParameters,
            options
        }, updateOperationSpec);
    }
    /**
     * List all the channels in a partner namespace.
     * @param resourceGroupName The name of the resource group within the partners subscription.
     * @param partnerNamespaceName Name of the partner namespace.
     * @param options The options parameters.
     */
    _listByPartnerNamespace(resourceGroupName, partnerNamespaceName, options) {
        return this.client.sendOperationRequest({ resourceGroupName, partnerNamespaceName, options }, listByPartnerNamespaceOperationSpec);
    }
    /**
     * Get the full endpoint URL of a partner destination channel.
     * @param resourceGroupName The name of the resource group within the partners subscription.
     * @param partnerNamespaceName Name of the partner namespace.
     * @param channelName Name of the Channel.
     * @param options The options parameters.
     */
    getFullUrl(resourceGroupName, partnerNamespaceName, channelName, options) {
        return this.client.sendOperationRequest({ resourceGroupName, partnerNamespaceName, channelName, options }, getFullUrlOperationSpec);
    }
    /**
     * ListByPartnerNamespaceNext
     * @param resourceGroupName The name of the resource group within the partners subscription.
     * @param partnerNamespaceName Name of the partner namespace.
     * @param nextLink The nextLink from the previous successful call to the ListByPartnerNamespace method.
     * @param options The options parameters.
     */
    _listByPartnerNamespaceNext(resourceGroupName, partnerNamespaceName, nextLink, options) {
        return this.client.sendOperationRequest({ resourceGroupName, partnerNamespaceName, nextLink, options }, listByPartnerNamespaceNextOperationSpec);
    }
}
// Operation Specifications
const serializer = coreClient.createSerializer(Mappers, /* isXml */ false);
const getOperationSpec = {
    path: "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.EventGrid/partnerNamespaces/{partnerNamespaceName}/channels/{channelName}",
    httpMethod: "GET",
    responses: {
        200: {
            bodyMapper: Mappers.Channel
        },
        default: {}
    },
    queryParameters: [Parameters.apiVersion],
    urlParameters: [
        Parameters.$host,
        Parameters.subscriptionId,
        Parameters.resourceGroupName,
        Parameters.partnerNamespaceName,
        Parameters.channelName
    ],
    headerParameters: [Parameters.accept],
    serializer
};
const createOrUpdateOperationSpec = {
    path: "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.EventGrid/partnerNamespaces/{partnerNamespaceName}/channels/{channelName}",
    httpMethod: "PUT",
    responses: {
        200: {
            bodyMapper: Mappers.Channel
        },
        201: {
            bodyMapper: Mappers.Channel
        },
        default: {}
    },
    requestBody: Parameters.channelInfo,
    queryParameters: [Parameters.apiVersion],
    urlParameters: [
        Parameters.$host,
        Parameters.subscriptionId,
        Parameters.resourceGroupName,
        Parameters.partnerNamespaceName,
        Parameters.channelName
    ],
    headerParameters: [Parameters.accept, Parameters.contentType],
    mediaType: "json",
    serializer
};
const deleteOperationSpec = {
    path: "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.EventGrid/partnerNamespaces/{partnerNamespaceName}/channels/{channelName}",
    httpMethod: "DELETE",
    responses: { 200: {}, 201: {}, 202: {}, 204: {}, default: {} },
    queryParameters: [Parameters.apiVersion],
    urlParameters: [
        Parameters.$host,
        Parameters.subscriptionId,
        Parameters.resourceGroupName,
        Parameters.partnerNamespaceName,
        Parameters.channelName
    ],
    serializer
};
const updateOperationSpec = {
    path: "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.EventGrid/partnerNamespaces/{partnerNamespaceName}/channels/{channelName}",
    httpMethod: "PATCH",
    responses: { 200: {}, default: {} },
    requestBody: Parameters.channelUpdateParameters,
    queryParameters: [Parameters.apiVersion],
    urlParameters: [
        Parameters.$host,
        Parameters.subscriptionId,
        Parameters.resourceGroupName,
        Parameters.partnerNamespaceName,
        Parameters.channelName
    ],
    headerParameters: [Parameters.contentType],
    mediaType: "json",
    serializer
};
const listByPartnerNamespaceOperationSpec = {
    path: "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.EventGrid/partnerNamespaces/{partnerNamespaceName}/channels",
    httpMethod: "GET",
    responses: {
        200: {
            bodyMapper: Mappers.ChannelsListResult
        },
        default: {}
    },
    queryParameters: [Parameters.apiVersion, Parameters.filter, Parameters.top],
    urlParameters: [
        Parameters.$host,
        Parameters.subscriptionId,
        Parameters.resourceGroupName,
        Parameters.partnerNamespaceName
    ],
    headerParameters: [Parameters.accept],
    serializer
};
const getFullUrlOperationSpec = {
    path: "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.EventGrid/partnerNamespaces/{partnerNamespaceName}/channels/{channelName}/getFullUrl",
    httpMethod: "POST",
    responses: {
        200: {
            bodyMapper: Mappers.EventSubscriptionFullUrl
        },
        default: {}
    },
    queryParameters: [Parameters.apiVersion],
    urlParameters: [
        Parameters.$host,
        Parameters.subscriptionId,
        Parameters.resourceGroupName,
        Parameters.partnerNamespaceName,
        Parameters.channelName
    ],
    headerParameters: [Parameters.accept],
    serializer
};
const listByPartnerNamespaceNextOperationSpec = {
    path: "{nextLink}",
    httpMethod: "GET",
    responses: {
        200: {
            bodyMapper: Mappers.ChannelsListResult
        },
        default: {}
    },
    urlParameters: [
        Parameters.$host,
        Parameters.subscriptionId,
        Parameters.resourceGroupName,
        Parameters.partnerNamespaceName,
        Parameters.nextLink
    ],
    headerParameters: [Parameters.accept],
    serializer
};
