// @azure/arm-eventgrid/dist-esm/src/operations/topicTypes.js
/*
 * Copyright (c) Microsoft Corporation.
 * Licensed under the MIT License.
 *
 * Code generated by Microsoft (R) AutoRest Code Generator.
 * Changes may cause incorrect behavior and will be lost if the code is regenerated.
 */
import { __asyncDelegator, __asyncGenerator, __asyncValues, __await } from "tslib";
import * as coreClient from "@azure/core-client";
import * as Mappers from "../models/mappers";
import * as Parameters from "../models/parameters";
/// <reference lib="esnext.asynciterable" />
/** Class containing TopicTypes operations. */
export class TopicTypesImpl {
    /**
     * Initialize a new instance of the class TopicTypes class.
     * @param client Reference to the service client
     */
    constructor(client) {
        this.client = client;
    }
    /**
     * List all registered topic types.
     * @param options The options parameters.
     */
    list(options) {
        const iter = this.listPagingAll(options);
        return {
            next() {
                return iter.next();
            },
            [Symbol.asyncIterator]() {
                return this;
            },
            byPage: (settings) => {
                if (settings === null || settings === void 0 ? void 0 : settings.maxPageSize) {
                    throw new Error("maxPageSize is not supported by this operation.");
                }
                return this.listPagingPage(options, settings);
            }
        };
    }
    listPagingPage(options, _settings) {
        return __asyncGenerator(this, arguments, function* listPagingPage_1() {
            let result;
            result = yield __await(this._list(options));
            yield yield __await(result.value || []);
        });
    }
    listPagingAll(options) {
        return __asyncGenerator(this, arguments, function* listPagingAll_1() {
            var e_1, _a;
            try {
                for (var _b = __asyncValues(this.listPagingPage(options)), _c; _c = yield __await(_b.next()), !_c.done;) {
                    const page = _c.value;
                    yield __await(yield* __asyncDelegator(__asyncValues(page)));
                }
            }
            catch (e_1_1) { e_1 = { error: e_1_1 }; }
            finally {
                try {
                    if (_c && !_c.done && (_a = _b.return)) yield __await(_a.call(_b));
                }
                finally { if (e_1) throw e_1.error; }
            }
        });
    }
    /**
     * List event types for a topic type.
     * @param topicTypeName Name of the topic type.
     * @param options The options parameters.
     */
    listEventTypes(topicTypeName, options) {
        const iter = this.listEventTypesPagingAll(topicTypeName, options);
        return {
            next() {
                return iter.next();
            },
            [Symbol.asyncIterator]() {
                return this;
            },
            byPage: (settings) => {
                if (settings === null || settings === void 0 ? void 0 : settings.maxPageSize) {
                    throw new Error("maxPageSize is not supported by this operation.");
                }
                return this.listEventTypesPagingPage(topicTypeName, options, settings);
            }
        };
    }
    listEventTypesPagingPage(topicTypeName, options, _settings) {
        return __asyncGenerator(this, arguments, function* listEventTypesPagingPage_1() {
            let result;
            result = yield __await(this._listEventTypes(topicTypeName, options));
            yield yield __await(result.value || []);
        });
    }
    listEventTypesPagingAll(topicTypeName, options) {
        return __asyncGenerator(this, arguments, function* listEventTypesPagingAll_1() {
            var e_2, _a;
            try {
                for (var _b = __asyncValues(this.listEventTypesPagingPage(topicTypeName, options)), _c; _c = yield __await(_b.next()), !_c.done;) {
                    const page = _c.value;
                    yield __await(yield* __asyncDelegator(__asyncValues(page)));
                }
            }
            catch (e_2_1) { e_2 = { error: e_2_1 }; }
            finally {
                try {
                    if (_c && !_c.done && (_a = _b.return)) yield __await(_a.call(_b));
                }
                finally { if (e_2) throw e_2.error; }
            }
        });
    }
    /**
     * List all registered topic types.
     * @param options The options parameters.
     */
    _list(options) {
        return this.client.sendOperationRequest({ options }, listOperationSpec);
    }
    /**
     * Get information about a topic type.
     * @param topicTypeName Name of the topic type.
     * @param options The options parameters.
     */
    get(topicTypeName, options) {
        return this.client.sendOperationRequest({ topicTypeName, options }, getOperationSpec);
    }
    /**
     * List event types for a topic type.
     * @param topicTypeName Name of the topic type.
     * @param options The options parameters.
     */
    _listEventTypes(topicTypeName, options) {
        return this.client.sendOperationRequest({ topicTypeName, options }, listEventTypesOperationSpec);
    }
}
// Operation Specifications
const serializer = coreClient.createSerializer(Mappers, /* isXml */ false);
const listOperationSpec = {
    path: "/providers/Microsoft.EventGrid/topicTypes",
    httpMethod: "GET",
    responses: {
        200: {
            bodyMapper: Mappers.TopicTypesListResult
        },
        default: {}
    },
    queryParameters: [Parameters.apiVersion],
    urlParameters: [Parameters.$host],
    headerParameters: [Parameters.accept],
    serializer
};
const getOperationSpec = {
    path: "/providers/Microsoft.EventGrid/topicTypes/{topicTypeName}",
    httpMethod: "GET",
    responses: {
        200: {
            bodyMapper: Mappers.TopicTypeInfo
        },
        default: {}
    },
    queryParameters: [Parameters.apiVersion],
    urlParameters: [Parameters.$host, Parameters.topicTypeName],
    headerParameters: [Parameters.accept],
    serializer
};
const listEventTypesOperationSpec = {
    path: "/providers/Microsoft.EventGrid/topicTypes/{topicTypeName}/eventTypes",
    httpMethod: "GET",
    responses: {
        200: {
            bodyMapper: Mappers.EventTypesListResult
        },
        default: {}
    },
    queryParameters: [Parameters.apiVersion],
    urlParameters: [Parameters.$host, Parameters.topicTypeName],
    headerParameters: [Parameters.accept],
    serializer
};
