// @aws-sdk/client-migrationhubstrategy/dist-cjs/index.js
"use strict";
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __name = (target, value) => __defProp(target, "name", { value, configurable: true });
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// src/index.ts
var src_exports = {};
__export(src_exports, {
  AccessDeniedException: () => AccessDeniedException,
  AnalysisStatusUnion: () => AnalysisStatusUnion,
  AnalysisType: () => AnalysisType,
  AnalyzerNameUnion: () => AnalyzerNameUnion,
  AntipatternReportStatus: () => AntipatternReportStatus,
  AppType: () => AppType,
  AppUnitErrorCategory: () => AppUnitErrorCategory,
  ApplicationComponentCriteria: () => ApplicationComponentCriteria,
  ApplicationMode: () => ApplicationMode,
  AssessmentDataSourceType: () => AssessmentDataSourceType,
  AssessmentStatus: () => AssessmentStatus,
  AuthType: () => AuthType,
  AwsManagedTargetDestination: () => AwsManagedTargetDestination,
  BinaryAnalyzerName: () => BinaryAnalyzerName,
  CollectorHealth: () => CollectorHealth,
  Condition: () => Condition,
  ConflictException: () => ConflictException,
  DataSourceType: () => DataSourceType,
  DatabaseManagementPreference: () => DatabaseManagementPreference,
  DatabaseMigrationPreference: () => DatabaseMigrationPreference,
  DependencyException: () => DependencyException,
  GetApplicationComponentDetailsCommand: () => GetApplicationComponentDetailsCommand,
  GetApplicationComponentStrategiesCommand: () => GetApplicationComponentStrategiesCommand,
  GetAssessmentCommand: () => GetAssessmentCommand,
  GetImportFileTaskCommand: () => GetImportFileTaskCommand,
  GetLatestAssessmentIdCommand: () => GetLatestAssessmentIdCommand,
  GetPortfolioPreferencesCommand: () => GetPortfolioPreferencesCommand,
  GetPortfolioSummaryCommand: () => GetPortfolioSummaryCommand,
  GetRecommendationReportDetailsCommand: () => GetRecommendationReportDetailsCommand,
  GetServerDetailsCommand: () => GetServerDetailsCommand,
  GetServerStrategiesCommand: () => GetServerStrategiesCommand,
  GroupName: () => GroupName,
  HeterogeneousTargetDatabaseEngine: () => HeterogeneousTargetDatabaseEngine,
  HomogeneousTargetDatabaseEngine: () => HomogeneousTargetDatabaseEngine,
  ImportFileTaskStatus: () => ImportFileTaskStatus,
  InclusionStatus: () => InclusionStatus,
  InternalServerException: () => InternalServerException,
  ListAnalyzableServersCommand: () => ListAnalyzableServersCommand,
  ListApplicationComponentsCommand: () => ListApplicationComponentsCommand,
  ListCollectorsCommand: () => ListCollectorsCommand,
  ListImportFileTaskCommand: () => ListImportFileTaskCommand,
  ListServersCommand: () => ListServersCommand,
  ManagementPreference: () => ManagementPreference,
  MigrationHubStrategy: () => MigrationHubStrategy,
  MigrationHubStrategyClient: () => MigrationHubStrategyClient,
  MigrationHubStrategyServiceException: () => MigrationHubStrategyServiceException,
  NoPreferenceTargetDestination: () => NoPreferenceTargetDestination,
  OSType: () => OSType,
  OutputFormat: () => OutputFormat,
  PipelineType: () => PipelineType,
  PutPortfolioPreferencesCommand: () => PutPortfolioPreferencesCommand,
  RecommendationReportStatus: () => RecommendationReportStatus,
  ResourceNotFoundException: () => ResourceNotFoundException,
  ResourceSubType: () => ResourceSubType,
  RunTimeAnalyzerName: () => RunTimeAnalyzerName,
  RunTimeAssessmentStatus: () => RunTimeAssessmentStatus,
  RuntimeAnalysisStatus: () => RuntimeAnalysisStatus,
  SelfManageTargetDestination: () => SelfManageTargetDestination,
  ServerCriteria: () => ServerCriteria,
  ServerErrorCategory: () => ServerErrorCategory,
  ServerOsType: () => ServerOsType,
  ServiceLinkedRoleLockClientException: () => ServiceLinkedRoleLockClientException,
  ServiceQuotaExceededException: () => ServiceQuotaExceededException,
  Severity: () => Severity,
  SortOrder: () => SortOrder,
  SourceCodeAnalyzerName: () => SourceCodeAnalyzerName,
  SrcCodeOrDbAnalysisStatus: () => SrcCodeOrDbAnalysisStatus,
  StartAssessmentCommand: () => StartAssessmentCommand,
  StartImportFileTaskCommand: () => StartImportFileTaskCommand,
  StartRecommendationReportGenerationCommand: () => StartRecommendationReportGenerationCommand,
  StopAssessmentCommand: () => StopAssessmentCommand,
  Strategy: () => Strategy,
  StrategyRecommendation: () => StrategyRecommendation,
  TargetDatabaseEngine: () => TargetDatabaseEngine,
  TargetDestination: () => TargetDestination,
  ThrottlingException: () => ThrottlingException,
  TransformationToolName: () => TransformationToolName,
  UpdateApplicationComponentConfigCommand: () => UpdateApplicationComponentConfigCommand,
  UpdateApplicationComponentConfigRequestFilterSensitiveLog: () => UpdateApplicationComponentConfigRequestFilterSensitiveLog,
  UpdateServerConfigCommand: () => UpdateServerConfigCommand,
  ValidationException: () => ValidationException,
  VersionControl: () => VersionControl,
  VersionControlType: () => VersionControlType,
  __Client: () => import_smithy_client.Client,
  paginateGetServerDetails: () => paginateGetServerDetails,
  paginateListAnalyzableServers: () => paginateListAnalyzableServers,
  paginateListApplicationComponents: () => paginateListApplicationComponents,
  paginateListCollectors: () => paginateListCollectors,
  paginateListImportFileTask: () => paginateListImportFileTask,
  paginateListServers: () => paginateListServers
});
module.exports = __toCommonJS(src_exports);

// src/MigrationHubStrategyClient.ts
var import_middleware_host_header = require("@aws-sdk/middleware-host-header");
var import_middleware_logger = require("@aws-sdk/middleware-logger");
var import_middleware_recursion_detection = require("@aws-sdk/middleware-recursion-detection");
var import_middleware_user_agent = require("@aws-sdk/middleware-user-agent");
var import_config_resolver = require("@smithy/config-resolver");
var import_core = require("@smithy/core");
var import_middleware_content_length = require("@smithy/middleware-content-length");
var import_middleware_endpoint = require("@smithy/middleware-endpoint");
var import_middleware_retry = require("@smithy/middleware-retry");

var import_httpAuthSchemeProvider = require("./auth/httpAuthSchemeProvider");

// src/endpoint/EndpointParameters.ts
var resolveClientEndpointParameters = /* @__PURE__ */ __name((options) => {
  return {
    ...options,
    useDualstackEndpoint: options.useDualstackEndpoint ?? false,
    useFipsEndpoint: options.useFipsEndpoint ?? false,
    defaultSigningName: "migrationhub-strategy"
  };
}, "resolveClientEndpointParameters");
var commonParams = {
  UseFIPS: { type: "builtInParams", name: "useFipsEndpoint" },
  Endpoint: { type: "builtInParams", name: "endpoint" },
  Region: { type: "builtInParams", name: "region" },
  UseDualStack: { type: "builtInParams", name: "useDualstackEndpoint" }
};

// src/MigrationHubStrategyClient.ts
var import_runtimeConfig = require("././runtimeConfig");

// src/runtimeExtensions.ts
var import_region_config_resolver = require("@aws-sdk/region-config-resolver");
var import_protocol_http = require("@smithy/protocol-http");
var import_smithy_client = require("@smithy/smithy-client");

// src/auth/httpAuthExtensionConfiguration.ts
var getHttpAuthExtensionConfiguration = /* @__PURE__ */ __name((runtimeConfig) => {
  const _httpAuthSchemes = runtimeConfig.httpAuthSchemes;
  let _httpAuthSchemeProvider = runtimeConfig.httpAuthSchemeProvider;
  let _credentials = runtimeConfig.credentials;
  return {
    setHttpAuthScheme(httpAuthScheme) {
      const index = _httpAuthSchemes.findIndex((scheme) => scheme.schemeId === httpAuthScheme.schemeId);
      if (index === -1) {
        _httpAuthSchemes.push(httpAuthScheme);
      } else {
        _httpAuthSchemes.splice(index, 1, httpAuthScheme);
      }
    },
    httpAuthSchemes() {
      return _httpAuthSchemes;
    },
    setHttpAuthSchemeProvider(httpAuthSchemeProvider) {
      _httpAuthSchemeProvider = httpAuthSchemeProvider;
    },
    httpAuthSchemeProvider() {
      return _httpAuthSchemeProvider;
    },
    setCredentials(credentials) {
      _credentials = credentials;
    },
    credentials() {
      return _credentials;
    }
  };
}, "getHttpAuthExtensionConfiguration");
var resolveHttpAuthRuntimeConfig = /* @__PURE__ */ __name((config) => {
  return {
    httpAuthSchemes: config.httpAuthSchemes(),
    httpAuthSchemeProvider: config.httpAuthSchemeProvider(),
    credentials: config.credentials()
  };
}, "resolveHttpAuthRuntimeConfig");

// src/runtimeExtensions.ts
var asPartial = /* @__PURE__ */ __name((t) => t, "asPartial");
var resolveRuntimeExtensions = /* @__PURE__ */ __name((runtimeConfig, extensions) => {
  const extensionConfiguration = {
    ...asPartial((0, import_region_config_resolver.getAwsRegionExtensionConfiguration)(runtimeConfig)),
    ...asPartial((0, import_smithy_client.getDefaultExtensionConfiguration)(runtimeConfig)),
    ...asPartial((0, import_protocol_http.getHttpHandlerExtensionConfiguration)(runtimeConfig)),
    ...asPartial(getHttpAuthExtensionConfiguration(runtimeConfig))
  };
  extensions.forEach((extension) => extension.configure(extensionConfiguration));
  return {
    ...runtimeConfig,
    ...(0, import_region_config_resolver.resolveAwsRegionExtensionConfiguration)(extensionConfiguration),
    ...(0, import_smithy_client.resolveDefaultRuntimeConfig)(extensionConfiguration),
    ...(0, import_protocol_http.resolveHttpHandlerRuntimeConfig)(extensionConfiguration),
    ...resolveHttpAuthRuntimeConfig(extensionConfiguration)
  };
}, "resolveRuntimeExtensions");

// src/MigrationHubStrategyClient.ts
var _MigrationHubStrategyClient = class _MigrationHubStrategyClient extends import_smithy_client.Client {
  constructor(...[configuration]) {
    const _config_0 = (0, import_runtimeConfig.getRuntimeConfig)(configuration || {});
    const _config_1 = resolveClientEndpointParameters(_config_0);
    const _config_2 = (0, import_config_resolver.resolveRegionConfig)(_config_1);
    const _config_3 = (0, import_middleware_endpoint.resolveEndpointConfig)(_config_2);
    const _config_4 = (0, import_middleware_retry.resolveRetryConfig)(_config_3);
    const _config_5 = (0, import_middleware_host_header.resolveHostHeaderConfig)(_config_4);
    const _config_6 = (0, import_middleware_user_agent.resolveUserAgentConfig)(_config_5);
    const _config_7 = (0, import_httpAuthSchemeProvider.resolveHttpAuthSchemeConfig)(_config_6);
    const _config_8 = resolveRuntimeExtensions(_config_7, (configuration == null ? void 0 : configuration.extensions) || []);
    super(_config_8);
    this.config = _config_8;
    this.middlewareStack.use((0, import_middleware_retry.getRetryPlugin)(this.config));
    this.middlewareStack.use((0, import_middleware_content_length.getContentLengthPlugin)(this.config));
    this.middlewareStack.use((0, import_middleware_host_header.getHostHeaderPlugin)(this.config));
    this.middlewareStack.use((0, import_middleware_logger.getLoggerPlugin)(this.config));
    this.middlewareStack.use((0, import_middleware_recursion_detection.getRecursionDetectionPlugin)(this.config));
    this.middlewareStack.use((0, import_middleware_user_agent.getUserAgentPlugin)(this.config));
    this.middlewareStack.use(
      (0, import_core.getHttpAuthSchemeEndpointRuleSetPlugin)(this.config, {
        httpAuthSchemeParametersProvider: this.getDefaultHttpAuthSchemeParametersProvider(),
        identityProviderConfigProvider: this.getIdentityProviderConfigProvider()
      })
    );
    this.middlewareStack.use((0, import_core.getHttpSigningPlugin)(this.config));
  }
  /**
   * Destroy underlying resources, like sockets. It's usually not necessary to do this.
   * However in Node.js, it's best to explicitly shut down the client's agent when it is no longer needed.
   * Otherwise, sockets might stay open for quite a long time before the server terminates them.
   */
  destroy() {
    super.destroy();
  }
  getDefaultHttpAuthSchemeParametersProvider() {
    return import_httpAuthSchemeProvider.defaultMigrationHubStrategyHttpAuthSchemeParametersProvider;
  }
  getIdentityProviderConfigProvider() {
    return async (config) => new import_core.DefaultIdentityProviderConfig({
      "aws.auth#sigv4": config.credentials
    });
  }
};
__name(_MigrationHubStrategyClient, "MigrationHubStrategyClient");
var MigrationHubStrategyClient = _MigrationHubStrategyClient;

// src/MigrationHubStrategy.ts


// src/commands/GetApplicationComponentDetailsCommand.ts

var import_middleware_serde = require("@smithy/middleware-serde");

var import_types = require("@smithy/types");

// src/protocols/Aws_restJson1.ts



// src/models/MigrationHubStrategyServiceException.ts

var _MigrationHubStrategyServiceException = class _MigrationHubStrategyServiceException extends import_smithy_client.ServiceException {
  /**
   * @internal
   */
  constructor(options) {
    super(options);
    Object.setPrototypeOf(this, _MigrationHubStrategyServiceException.prototype);
  }
};
__name(_MigrationHubStrategyServiceException, "MigrationHubStrategyServiceException");
var MigrationHubStrategyServiceException = _MigrationHubStrategyServiceException;

// src/models/models_0.ts

var _AccessDeniedException = class _AccessDeniedException extends MigrationHubStrategyServiceException {
  /**
   * @internal
   */
  constructor(opts) {
    super({
      name: "AccessDeniedException",
      $fault: "client",
      ...opts
    });
    this.name = "AccessDeniedException";
    this.$fault = "client";
    Object.setPrototypeOf(this, _AccessDeniedException.prototype);
  }
};
__name(_AccessDeniedException, "AccessDeniedException");
var AccessDeniedException = _AccessDeniedException;
var RuntimeAnalysisStatus = {
  ANALYSIS_FAILED: "ANALYSIS_FAILED",
  ANALYSIS_STARTED: "ANALYSIS_STARTED",
  ANALYSIS_SUCCESS: "ANALYSIS_SUCCESS",
  ANALYSIS_TO_BE_SCHEDULED: "ANALYSIS_TO_BE_SCHEDULED"
};
var SrcCodeOrDbAnalysisStatus = {
  ANALYSIS_FAILED: "ANALYSIS_FAILED",
  ANALYSIS_PARTIAL_SUCCESS: "ANALYSIS_PARTIAL_SUCCESS",
  ANALYSIS_STARTED: "ANALYSIS_STARTED",
  ANALYSIS_SUCCESS: "ANALYSIS_SUCCESS",
  ANALYSIS_TO_BE_SCHEDULED: "ANALYSIS_TO_BE_SCHEDULED",
  CONFIGURED: "CONFIGURED",
  UNCONFIGURED: "UNCONFIGURED"
};
var AnalysisStatusUnion;
((AnalysisStatusUnion2) => {
  AnalysisStatusUnion2.visit = /* @__PURE__ */ __name((value, visitor) => {
    if (value.runtimeAnalysisStatus !== void 0)
      return visitor.runtimeAnalysisStatus(value.runtimeAnalysisStatus);
    if (value.srcCodeOrDbAnalysisStatus !== void 0)
      return visitor.srcCodeOrDbAnalysisStatus(value.srcCodeOrDbAnalysisStatus);
    return visitor._(value.$unknown[0], value.$unknown[1]);
  }, "visit");
})(AnalysisStatusUnion || (AnalysisStatusUnion = {}));
var AnalysisType = {
  BINARY_ANALYSIS: "BINARY_ANALYSIS",
  DATABASE_ANALYSIS: "DATABASE_ANALYSIS",
  RUNTIME_ANALYSIS: "RUNTIME_ANALYSIS",
  SOURCE_CODE_ANALYSIS: "SOURCE_CODE_ANALYSIS"
};
var BinaryAnalyzerName = {
  BYTECODE_ANALYZER: "BYTECODE_ANALYZER",
  DLL_ANALYZER: "DLL_ANALYZER"
};
var RunTimeAnalyzerName = {
  A2C_ANALYZER: "A2C_ANALYZER",
  DATABASE_ANALYZER: "DATABASE_ANALYZER",
  EMP_PA_ANALYZER: "EMP_PA_ANALYZER",
  REHOST_ANALYZER: "REHOST_ANALYZER",
  SCT_ANALYZER: "SCT_ANALYZER"
};
var SourceCodeAnalyzerName = {
  BYTECODE_ANALYZER: "BYTECODE_ANALYZER",
  CSHARP_ANALYZER: "CSHARP_ANALYZER",
  JAVA_ANALYZER: "JAVA_ANALYZER",
  PORTING_ASSISTANT: "PORTING_ASSISTANT"
};
var AnalyzerNameUnion;
((AnalyzerNameUnion2) => {
  AnalyzerNameUnion2.visit = /* @__PURE__ */ __name((value, visitor) => {
    if (value.binaryAnalyzerName !== void 0)
      return visitor.binaryAnalyzerName(value.binaryAnalyzerName);
    if (value.runTimeAnalyzerName !== void 0)
      return visitor.runTimeAnalyzerName(value.runTimeAnalyzerName);
    if (value.sourceCodeAnalyzerName !== void 0)
      return visitor.sourceCodeAnalyzerName(value.sourceCodeAnalyzerName);
    return visitor._(value.$unknown[0], value.$unknown[1]);
  }, "visit");
})(AnalyzerNameUnion || (AnalyzerNameUnion = {}));
var AntipatternReportStatus = {
  FAILED: "FAILED",
  IN_PROGRESS: "IN_PROGRESS",
  SUCCESS: "SUCCESS"
};
var Severity = {
  HIGH: "HIGH",
  LOW: "LOW",
  MEDIUM: "MEDIUM"
};
var ApplicationComponentCriteria = {
  ANALYSIS_STATUS: "ANALYSIS_STATUS",
  APP_NAME: "APP_NAME",
  APP_TYPE: "APP_TYPE",
  DESTINATION: "DESTINATION",
  ERROR_CATEGORY: "ERROR_CATEGORY",
  NOT_DEFINED: "NOT_DEFINED",
  SERVER_ID: "SERVER_ID",
  STRATEGY: "STRATEGY"
};
var AppType = {
  IIS: "IIS",
  cassandra: "Cassandra",
  db2: "DB2",
  dotNetFramework: "DotNetFramework",
  dotnet: "Dotnet",
  dotnetcore: "DotnetCore",
  java: "Java",
  jboss: "JBoss",
  mariadb: "Maria DB",
  mongodb: "Mongo DB",
  mysql: "MySQL",
  oracle: "Oracle",
  other: "Other",
  postgresqlserver: "PostgreSQLServer",
  spring: "Spring",
  sqlServer: "SQLServer",
  sybase: "Sybase",
  tomcat: "Tomcat",
  unknown: "Unknown",
  visualbasic: "Visual Basic",
  weblogic: "Oracle WebLogic",
  websphere: "IBM WebSphere"
};
var AppUnitErrorCategory = {
  CONNECTIVITY_ERROR: "CONNECTIVITY_ERROR",
  CREDENTIAL_ERROR: "CREDENTIAL_ERROR",
  OTHER_ERROR: "OTHER_ERROR",
  PERMISSION_ERROR: "PERMISSION_ERROR",
  UNSUPPORTED_ERROR: "UNSUPPORTED_ERROR"
};
var InclusionStatus = {
  EXCLUDE_FROM_RECOMMENDATION: "excludeFromAssessment",
  INCLUDE_IN_RECOMMENDATION: "includeInAssessment"
};
var Strategy = {
  REFACTOR: "Refactor",
  REHOST: "Rehost",
  RELOCATE: "Relocate",
  REPLATFORM: "Replatform",
  REPURCHASE: "Repurchase",
  RETAIN: "Retain",
  RETIREMENT: "Retirement"
};
var TargetDestination = {
  AMAZON_DOCUMENTDB: "Amazon DocumentDB",
  AMAZON_DYNAMODB: "Amazon DynamoDB",
  AMAZON_ELASTIC_CLOUD_COMPUTE: "Amazon Elastic Cloud Compute (EC2)",
  AMAZON_ELASTIC_CONTAINER_SERVICE: "Amazon Elastic Container Service (ECS)",
  AMAZON_ELASTIC_KUBERNETES_SERVICE: "Amazon Elastic Kubernetes Service (EKS)",
  AMAZON_RDS: "Amazon Relational Database Service",
  AMAZON_RDS_MYSQL: "Amazon Relational Database Service on MySQL",
  AMAZON_RDS_POSTGRESQL: "Amazon Relational Database Service on PostgreSQL",
  AURORA_MYSQL: "Aurora MySQL",
  AURORA_POSTGRESQL: "Aurora PostgreSQL",
  AWS_ELASTIC_BEANSTALK: "AWS Elastic BeanStalk",
  AWS_FARGATE: "AWS Fargate",
  BABELFISH_AURORA_POSTGRESQL: "Babelfish for Aurora PostgreSQL",
  NONE_SPECIFIED: "None specified"
};
var TransformationToolName = {
  APP2CONTAINER: "App2Container",
  DMS: "Database Migration Service",
  EMP: "End of Support Migration",
  IN_PLACE_OS_UPGRADE: "In Place Operating System Upgrade",
  MGN: "Application Migration Service",
  NATIVE_SQL: "Native SQL Server Backup/Restore",
  PORTING_ASSISTANT: "Porting Assistant For .NET",
  SCT: "Schema Conversion Tool",
  STRATEGY_RECOMMENDATION_SUPPORT: "Strategy Recommendation Support",
  WWAMA: "Windows Web Application Migration Assistant"
};
var ResourceSubType = {
  DATABASE: "Database",
  DATABASE_PROCESS: "DatabaseProcess",
  PROCESS: "Process"
};
var StrategyRecommendation = {
  NOT_RECOMMENDED: "notRecommended",
  POTENTIAL: "potential",
  RECOMMENDED: "recommended",
  VIABLE_OPTION: "viableOption"
};
var ApplicationMode = {
  ALL: "ALL",
  KNOWN: "KNOWN",
  UNKNOWN: "UNKNOWN"
};
var AwsManagedTargetDestination = {
  AWS_ELASTIC_BEANSTALK: "AWS Elastic BeanStalk",
  AWS_FARGATE: "AWS Fargate",
  NONE_SPECIFIED: "None specified"
};
var NoPreferenceTargetDestination = {
  AMAZON_ELASTIC_CLOUD_COMPUTE: "Amazon Elastic Cloud Compute (EC2)",
  AMAZON_ELASTIC_CONTAINER_SERVICE: "Amazon Elastic Container Service (ECS)",
  AMAZON_ELASTIC_KUBERNETES_SERVICE: "Amazon Elastic Kubernetes Service (EKS)",
  AWS_ELASTIC_BEANSTALK: "AWS Elastic BeanStalk",
  AWS_FARGATE: "AWS Fargate",
  NONE_SPECIFIED: "None specified"
};
var SelfManageTargetDestination = {
  AMAZON_ELASTIC_CLOUD_COMPUTE: "Amazon Elastic Cloud Compute (EC2)",
  AMAZON_ELASTIC_CONTAINER_SERVICE: "Amazon Elastic Container Service (ECS)",
  AMAZON_ELASTIC_KUBERNETES_SERVICE: "Amazon Elastic Kubernetes Service (EKS)",
  NONE_SPECIFIED: "None specified"
};
var ManagementPreference;
((ManagementPreference3) => {
  ManagementPreference3.visit = /* @__PURE__ */ __name((value, visitor) => {
    if (value.awsManagedResources !== void 0)
      return visitor.awsManagedResources(value.awsManagedResources);
    if (value.selfManageResources !== void 0)
      return visitor.selfManageResources(value.selfManageResources);
    if (value.noPreference !== void 0)
      return visitor.noPreference(value.noPreference);
    return visitor._(value.$unknown[0], value.$unknown[1]);
  }, "visit");
})(ManagementPreference || (ManagementPreference = {}));
var AssessmentDataSourceType = {
  ADS: "ApplicationDiscoveryService",
  MANUAL_IMPORT: "ManualImport",
  SR_COLLECTOR: "StrategyRecommendationsApplicationDataCollector"
};
var AssessmentStatus = {
  COMPLETE: "COMPLETE",
  FAILED: "FAILED",
  IN_PROGRESS: "IN_PROGRESS",
  STOPPED: "STOPPED"
};
var RunTimeAssessmentStatus = {
  DC_FAILED: "dataCollectionTaskFailed",
  DC_PARTIAL_SUCCESS: "dataCollectionTaskPartialSuccess",
  DC_REQ_SENT: "dataCollectionTaskScheduled",
  DC_STARTED: "dataCollectionTaskStarted",
  DC_STOPPED: "dataCollectionTaskStopped",
  DC_SUCCESS: "dataCollectionTaskSuccess",
  DC_TO_BE_SCHEDULED: "dataCollectionTaskToBeScheduled"
};
var ServerOsType = {
  amazonLinux: "AmazonLinux",
  endOfSupportWindowsServer: "EndOfSupportWindowsServer",
  other: "Other",
  redhat: "Redhat",
  windowsServer: "WindowsServer"
};
var Condition = {
  CONTAINS: "CONTAINS",
  EQUALS: "EQUALS",
  NOT_CONTAINS: "NOT_CONTAINS",
  NOT_EQUALS: "NOT_EQUALS"
};
var AuthType = {
  CERT: "CERT",
  NTLM: "NTLM",
  SSH: "SSH"
};
var _InternalServerException = class _InternalServerException extends MigrationHubStrategyServiceException {
  /**
   * @internal
   */
  constructor(opts) {
    super({
      name: "InternalServerException",
      $fault: "server",
      ...opts
    });
    this.name = "InternalServerException";
    this.$fault = "server";
    Object.setPrototypeOf(this, _InternalServerException.prototype);
  }
};
__name(_InternalServerException, "InternalServerException");
var InternalServerException = _InternalServerException;
var _ResourceNotFoundException = class _ResourceNotFoundException extends MigrationHubStrategyServiceException {
  /**
   * @internal
   */
  constructor(opts) {
    super({
      name: "ResourceNotFoundException",
      $fault: "client",
      ...opts
    });
    this.name = "ResourceNotFoundException";
    this.$fault = "client";
    Object.setPrototypeOf(this, _ResourceNotFoundException.prototype);
  }
};
__name(_ResourceNotFoundException, "ResourceNotFoundException");
var ResourceNotFoundException = _ResourceNotFoundException;
var _ThrottlingException = class _ThrottlingException extends MigrationHubStrategyServiceException {
  /**
   * @internal
   */
  constructor(opts) {
    super({
      name: "ThrottlingException",
      $fault: "client",
      ...opts
    });
    this.name = "ThrottlingException";
    this.$fault = "client";
    Object.setPrototypeOf(this, _ThrottlingException.prototype);
  }
};
__name(_ThrottlingException, "ThrottlingException");
var ThrottlingException = _ThrottlingException;
var ImportFileTaskStatus = {
  DELETE_FAILED: "DeleteFailed",
  DELETE_IN_PROGRESS: "DeleteInProgress",
  DELETE_PARTIAL_SUCCESS: "DeletePartialSuccess",
  DELETE_SUCCESS: "DeleteSuccess",
  IMPORT_FAILED: "ImportFailed",
  IMPORT_IN_PROGRESS: "ImportInProgress",
  IMPORT_PARTIAL_SUCCESS: "ImportPartialSuccess",
  IMPORT_SUCCESS: "ImportSuccess"
};
var _ValidationException = class _ValidationException extends MigrationHubStrategyServiceException {
  /**
   * @internal
   */
  constructor(opts) {
    super({
      name: "ValidationException",
      $fault: "client",
      ...opts
    });
    this.name = "ValidationException";
    this.$fault = "client";
    Object.setPrototypeOf(this, _ValidationException.prototype);
  }
};
__name(_ValidationException, "ValidationException");
var ValidationException = _ValidationException;
var _DependencyException = class _DependencyException extends MigrationHubStrategyServiceException {
  /**
   * @internal
   */
  constructor(opts) {
    super({
      name: "DependencyException",
      $fault: "server",
      ...opts
    });
    this.name = "DependencyException";
    this.$fault = "server";
    Object.setPrototypeOf(this, _DependencyException.prototype);
  }
};
__name(_DependencyException, "DependencyException");
var DependencyException = _DependencyException;
var DatabaseManagementPreference = {
  AWS_MANAGED: "AWS-managed",
  NO_PREFERENCE: "No preference",
  SELF_MANAGE: "Self-manage"
};
var HeterogeneousTargetDatabaseEngine = {
  AMAZON_AURORA: "Amazon Aurora",
  AWS_POSTGRESQL: "AWS PostgreSQL",
  DB2_LUW: "Db2 LUW",
  MARIA_DB: "MariaDB",
  MICROSOFT_SQL_SERVER: "Microsoft SQL Server",
  MONGO_DB: "MongoDB",
  MYSQL: "MySQL",
  NONE_SPECIFIED: "None specified",
  ORACLE_DATABASE: "Oracle Database",
  SAP: "SAP"
};
var HomogeneousTargetDatabaseEngine = {
  NONE_SPECIFIED: "None specified"
};
var TargetDatabaseEngine = {
  AMAZON_AURORA: "Amazon Aurora",
  AWS_POSTGRESQL: "AWS PostgreSQL",
  DB2_LUW: "Db2 LUW",
  MARIA_DB: "MariaDB",
  MICROSOFT_SQL_SERVER: "Microsoft SQL Server",
  MONGO_DB: "MongoDB",
  MYSQL: "MySQL",
  NONE_SPECIFIED: "None specified",
  ORACLE_DATABASE: "Oracle Database",
  SAP: "SAP"
};
var DatabaseMigrationPreference;
((DatabaseMigrationPreference3) => {
  DatabaseMigrationPreference3.visit = /* @__PURE__ */ __name((value, visitor) => {
    if (value.heterogeneous !== void 0)
      return visitor.heterogeneous(value.heterogeneous);
    if (value.homogeneous !== void 0)
      return visitor.homogeneous(value.homogeneous);
    if (value.noPreference !== void 0)
      return visitor.noPreference(value.noPreference);
    return visitor._(value.$unknown[0], value.$unknown[1]);
  }, "visit");
})(DatabaseMigrationPreference || (DatabaseMigrationPreference = {}));
var RecommendationReportStatus = {
  FAILED: "FAILED",
  IN_PROGRESS: "IN_PROGRESS",
  SUCCESS: "SUCCESS"
};
var ServerErrorCategory = {
  ARCHITECTURE_ERROR: "ARCHITECTURE_ERROR",
  CONNECTIVITY_ERROR: "CONNECTIVITY_ERROR",
  CREDENTIAL_ERROR: "CREDENTIAL_ERROR",
  OTHER_ERROR: "OTHER_ERROR",
  PERMISSION_ERROR: "PERMISSION_ERROR"
};
var OSType = {
  LINUX: "LINUX",
  WINDOWS: "WINDOWS"
};
var SortOrder = {
  ASC: "ASC",
  DESC: "DESC"
};
var GroupName = {
  EXTERNAL_ID: "ExternalId",
  EXTERNAL_SOURCE_TYPE: "ExternalSourceType"
};
var _ServiceLinkedRoleLockClientException = class _ServiceLinkedRoleLockClientException extends MigrationHubStrategyServiceException {
  /**
   * @internal
   */
  constructor(opts) {
    super({
      name: "ServiceLinkedRoleLockClientException",
      $fault: "client",
      ...opts
    });
    this.name = "ServiceLinkedRoleLockClientException";
    this.$fault = "client";
    Object.setPrototypeOf(this, _ServiceLinkedRoleLockClientException.prototype);
  }
};
__name(_ServiceLinkedRoleLockClientException, "ServiceLinkedRoleLockClientException");
var ServiceLinkedRoleLockClientException = _ServiceLinkedRoleLockClientException;
var CollectorHealth = {
  COLLECTOR_HEALTHY: "COLLECTOR_HEALTHY",
  COLLECTOR_UNHEALTHY: "COLLECTOR_UNHEALTHY"
};
var PipelineType = {
  AZURE_DEVOPS: "AZURE_DEVOPS"
};
var VersionControlType = {
  AZURE_DEVOPS_GIT: "AZURE_DEVOPS_GIT",
  GITHUB: "GITHUB",
  GITHUB_ENTERPRISE: "GITHUB_ENTERPRISE"
};
var ServerCriteria = {
  ANALYSIS_STATUS: "ANALYSIS_STATUS",
  DESTINATION: "DESTINATION",
  ERROR_CATEGORY: "ERROR_CATEGORY",
  NOT_DEFINED: "NOT_DEFINED",
  OS_NAME: "OS_NAME",
  SERVER_ID: "SERVER_ID",
  STRATEGY: "STRATEGY"
};
var _ConflictException = class _ConflictException extends MigrationHubStrategyServiceException {
  /**
   * @internal
   */
  constructor(opts) {
    super({
      name: "ConflictException",
      $fault: "client",
      ...opts
    });
    this.name = "ConflictException";
    this.$fault = "client";
    Object.setPrototypeOf(this, _ConflictException.prototype);
  }
};
__name(_ConflictException, "ConflictException");
var ConflictException = _ConflictException;
var _ServiceQuotaExceededException = class _ServiceQuotaExceededException extends MigrationHubStrategyServiceException {
  /**
   * @internal
   */
  constructor(opts) {
    super({
      name: "ServiceQuotaExceededException",
      $fault: "client",
      ...opts
    });
    this.name = "ServiceQuotaExceededException";
    this.$fault = "client";
    Object.setPrototypeOf(this, _ServiceQuotaExceededException.prototype);
  }
};
__name(_ServiceQuotaExceededException, "ServiceQuotaExceededException");
var ServiceQuotaExceededException = _ServiceQuotaExceededException;
var DataSourceType = {
  ADS: "ApplicationDiscoveryService",
  IMPORT: "Import",
  MHSR_COLLECTOR: "StrategyRecommendationsApplicationDataCollector",
  MPA: "MPA"
};
var OutputFormat = {
  Excel: "Excel",
  Json: "Json"
};
var VersionControl = {
  AZURE_DEVOPS_GIT: "AZURE_DEVOPS_GIT",
  GITHUB: "GITHUB",
  GITHUB_ENTERPRISE: "GITHUB_ENTERPRISE"
};
var UpdateApplicationComponentConfigRequestFilterSensitiveLog = /* @__PURE__ */ __name((obj) => ({
  ...obj,
  ...obj.secretsManagerKey && { secretsManagerKey: import_smithy_client.SENSITIVE_STRING }
}), "UpdateApplicationComponentConfigRequestFilterSensitiveLog");

// src/protocols/Aws_restJson1.ts
var se_GetApplicationComponentDetailsCommand = /* @__PURE__ */ __name(async (input, context) => {
  const b = (0, import_core.requestBuilder)(input, context);
  const headers = {};
  b.bp("/get-applicationcomponent-details/{applicationComponentId}");
  b.p("applicationComponentId", () => input.applicationComponentId, "{applicationComponentId}", false);
  let body;
  b.m("GET").h(headers).b(body);
  return b.build();
}, "se_GetApplicationComponentDetailsCommand");
var se_GetApplicationComponentStrategiesCommand = /* @__PURE__ */ __name(async (input, context) => {
  const b = (0, import_core.requestBuilder)(input, context);
  const headers = {};
  b.bp("/get-applicationcomponent-strategies/{applicationComponentId}");
  b.p("applicationComponentId", () => input.applicationComponentId, "{applicationComponentId}", false);
  let body;
  b.m("GET").h(headers).b(body);
  return b.build();
}, "se_GetApplicationComponentStrategiesCommand");
var se_GetAssessmentCommand = /* @__PURE__ */ __name(async (input, context) => {
  const b = (0, import_core.requestBuilder)(input, context);
  const headers = {};
  b.bp("/get-assessment/{id}");
  b.p("id", () => input.id, "{id}", false);
  let body;
  b.m("GET").h(headers).b(body);
  return b.build();
}, "se_GetAssessmentCommand");
var se_GetImportFileTaskCommand = /* @__PURE__ */ __name(async (input, context) => {
  const b = (0, import_core.requestBuilder)(input, context);
  const headers = {};
  b.bp("/get-import-file-task/{id}");
  b.p("id", () => input.id, "{id}", false);
  let body;
  b.m("GET").h(headers).b(body);
  return b.build();
}, "se_GetImportFileTaskCommand");
var se_GetLatestAssessmentIdCommand = /* @__PURE__ */ __name(async (input, context) => {
  const b = (0, import_core.requestBuilder)(input, context);
  const headers = {
    "content-type": "application/json"
  };
  b.bp("/get-latest-assessment-id");
  let body;
  body = "";
  b.m("GET").h(headers).b(body);
  return b.build();
}, "se_GetLatestAssessmentIdCommand");
var se_GetPortfolioPreferencesCommand = /* @__PURE__ */ __name(async (input, context) => {
  const b = (0, import_core.requestBuilder)(input, context);
  const headers = {
    "content-type": "application/json"
  };
  b.bp("/get-portfolio-preferences");
  let body;
  body = "";
  b.m("GET").h(headers).b(body);
  return b.build();
}, "se_GetPortfolioPreferencesCommand");
var se_GetPortfolioSummaryCommand = /* @__PURE__ */ __name(async (input, context) => {
  const b = (0, import_core.requestBuilder)(input, context);
  const headers = {
    "content-type": "application/json"
  };
  b.bp("/get-portfolio-summary");
  let body;
  body = "";
  b.m("GET").h(headers).b(body);
  return b.build();
}, "se_GetPortfolioSummaryCommand");
var se_GetRecommendationReportDetailsCommand = /* @__PURE__ */ __name(async (input, context) => {
  const b = (0, import_core.requestBuilder)(input, context);
  const headers = {};
  b.bp("/get-recommendation-report-details/{id}");
  b.p("id", () => input.id, "{id}", false);
  let body;
  b.m("GET").h(headers).b(body);
  return b.build();
}, "se_GetRecommendationReportDetailsCommand");
var se_GetServerDetailsCommand = /* @__PURE__ */ __name(async (input, context) => {
  const b = (0, import_core.requestBuilder)(input, context);
  const headers = {};
  b.bp("/get-server-details/{serverId}");
  b.p("serverId", () => input.serverId, "{serverId}", false);
  const query = (0, import_smithy_client.map)({
    [_nT]: [, input[_nT]],
    [_mR]: [() => input.maxResults !== void 0, () => input[_mR].toString()]
  });
  let body;
  b.m("GET").h(headers).q(query).b(body);
  return b.build();
}, "se_GetServerDetailsCommand");
var se_GetServerStrategiesCommand = /* @__PURE__ */ __name(async (input, context) => {
  const b = (0, import_core.requestBuilder)(input, context);
  const headers = {};
  b.bp("/get-server-strategies/{serverId}");
  b.p("serverId", () => input.serverId, "{serverId}", false);
  let body;
  b.m("GET").h(headers).b(body);
  return b.build();
}, "se_GetServerStrategiesCommand");
var se_ListAnalyzableServersCommand = /* @__PURE__ */ __name(async (input, context) => {
  const b = (0, import_core.requestBuilder)(input, context);
  const headers = {
    "content-type": "application/json"
  };
  b.bp("/list-analyzable-servers");
  let body;
  body = JSON.stringify(
    (0, import_smithy_client.take)(input, {
      maxResults: [],
      nextToken: [],
      sort: []
    })
  );
  b.m("POST").h(headers).b(body);
  return b.build();
}, "se_ListAnalyzableServersCommand");
var se_ListApplicationComponentsCommand = /* @__PURE__ */ __name(async (input, context) => {
  const b = (0, import_core.requestBuilder)(input, context);
  const headers = {
    "content-type": "application/json"
  };
  b.bp("/list-applicationcomponents");
  let body;
  body = JSON.stringify(
    (0, import_smithy_client.take)(input, {
      applicationComponentCriteria: [],
      filterValue: [],
      groupIdFilter: (_) => (0, import_smithy_client._json)(_),
      maxResults: [],
      nextToken: [],
      sort: []
    })
  );
  b.m("POST").h(headers).b(body);
  return b.build();
}, "se_ListApplicationComponentsCommand");
var se_ListCollectorsCommand = /* @__PURE__ */ __name(async (input, context) => {
  const b = (0, import_core.requestBuilder)(input, context);
  const headers = {};
  b.bp("/list-collectors");
  const query = (0, import_smithy_client.map)({
    [_nT]: [, input[_nT]],
    [_mR]: [() => input.maxResults !== void 0, () => input[_mR].toString()]
  });
  let body;
  b.m("GET").h(headers).q(query).b(body);
  return b.build();
}, "se_ListCollectorsCommand");
var se_ListImportFileTaskCommand = /* @__PURE__ */ __name(async (input, context) => {
  const b = (0, import_core.requestBuilder)(input, context);
  const headers = {};
  b.bp("/list-import-file-task");
  const query = (0, import_smithy_client.map)({
    [_nT]: [, input[_nT]],
    [_mR]: [() => input.maxResults !== void 0, () => input[_mR].toString()]
  });
  let body;
  b.m("GET").h(headers).q(query).b(body);
  return b.build();
}, "se_ListImportFileTaskCommand");
var se_ListServersCommand = /* @__PURE__ */ __name(async (input, context) => {
  const b = (0, import_core.requestBuilder)(input, context);
  const headers = {
    "content-type": "application/json"
  };
  b.bp("/list-servers");
  let body;
  body = JSON.stringify(
    (0, import_smithy_client.take)(input, {
      filterValue: [],
      groupIdFilter: (_) => (0, import_smithy_client._json)(_),
      maxResults: [],
      nextToken: [],
      serverCriteria: [],
      sort: []
    })
  );
  b.m("POST").h(headers).b(body);
  return b.build();
}, "se_ListServersCommand");
var se_PutPortfolioPreferencesCommand = /* @__PURE__ */ __name(async (input, context) => {
  const b = (0, import_core.requestBuilder)(input, context);
  const headers = {
    "content-type": "application/json"
  };
  b.bp("/put-portfolio-preferences");
  let body;
  body = JSON.stringify(
    (0, import_smithy_client.take)(input, {
      applicationMode: [],
      applicationPreferences: (_) => (0, import_smithy_client._json)(_),
      databasePreferences: (_) => (0, import_smithy_client._json)(_),
      prioritizeBusinessGoals: (_) => (0, import_smithy_client._json)(_)
    })
  );
  b.m("POST").h(headers).b(body);
  return b.build();
}, "se_PutPortfolioPreferencesCommand");
var se_StartAssessmentCommand = /* @__PURE__ */ __name(async (input, context) => {
  const b = (0, import_core.requestBuilder)(input, context);
  const headers = {
    "content-type": "application/json"
  };
  b.bp("/start-assessment");
  let body;
  body = JSON.stringify(
    (0, import_smithy_client.take)(input, {
      assessmentDataSourceType: [],
      assessmentTargets: (_) => (0, import_smithy_client._json)(_),
      s3bucketForAnalysisData: [],
      s3bucketForReportData: []
    })
  );
  b.m("POST").h(headers).b(body);
  return b.build();
}, "se_StartAssessmentCommand");
var se_StartImportFileTaskCommand = /* @__PURE__ */ __name(async (input, context) => {
  const b = (0, import_core.requestBuilder)(input, context);
  const headers = {
    "content-type": "application/json"
  };
  b.bp("/start-import-file-task");
  let body;
  body = JSON.stringify(
    (0, import_smithy_client.take)(input, {
      S3Bucket: [],
      dataSourceType: [],
      groupId: (_) => (0, import_smithy_client._json)(_),
      name: [],
      s3bucketForReportData: [],
      s3key: []
    })
  );
  b.m("POST").h(headers).b(body);
  return b.build();
}, "se_StartImportFileTaskCommand");
var se_StartRecommendationReportGenerationCommand = /* @__PURE__ */ __name(async (input, context) => {
  const b = (0, import_core.requestBuilder)(input, context);
  const headers = {
    "content-type": "application/json"
  };
  b.bp("/start-recommendation-report-generation");
  let body;
  body = JSON.stringify(
    (0, import_smithy_client.take)(input, {
      groupIdFilter: (_) => (0, import_smithy_client._json)(_),
      outputFormat: []
    })
  );
  b.m("POST").h(headers).b(body);
  return b.build();
}, "se_StartRecommendationReportGenerationCommand");
var se_StopAssessmentCommand = /* @__PURE__ */ __name(async (input, context) => {
  const b = (0, import_core.requestBuilder)(input, context);
  const headers = {
    "content-type": "application/json"
  };
  b.bp("/stop-assessment");
  let body;
  body = JSON.stringify(
    (0, import_smithy_client.take)(input, {
      assessmentId: []
    })
  );
  b.m("POST").h(headers).b(body);
  return b.build();
}, "se_StopAssessmentCommand");
var se_UpdateApplicationComponentConfigCommand = /* @__PURE__ */ __name(async (input, context) => {
  const b = (0, import_core.requestBuilder)(input, context);
  const headers = {
    "content-type": "application/json"
  };
  b.bp("/update-applicationcomponent-config");
  let body;
  body = JSON.stringify(
    (0, import_smithy_client.take)(input, {
      appType: [],
      applicationComponentId: [],
      configureOnly: [],
      inclusionStatus: [],
      secretsManagerKey: [],
      sourceCodeList: (_) => (0, import_smithy_client._json)(_),
      strategyOption: (_) => (0, import_smithy_client._json)(_)
    })
  );
  b.m("POST").h(headers).b(body);
  return b.build();
}, "se_UpdateApplicationComponentConfigCommand");
var se_UpdateServerConfigCommand = /* @__PURE__ */ __name(async (input, context) => {
  const b = (0, import_core.requestBuilder)(input, context);
  const headers = {
    "content-type": "application/json"
  };
  b.bp("/update-server-config");
  let body;
  body = JSON.stringify(
    (0, import_smithy_client.take)(input, {
      serverId: [],
      strategyOption: (_) => (0, import_smithy_client._json)(_)
    })
  );
  b.m("POST").h(headers).b(body);
  return b.build();
}, "se_UpdateServerConfigCommand");
var de_GetApplicationComponentDetailsCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode !== 200 && output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const contents = (0, import_smithy_client.map)({
    $metadata: deserializeMetadata(output)
  });
  const data = (0, import_smithy_client.expectNonNull)((0, import_smithy_client.expectObject)(await parseBody(output.body, context)), "body");
  const doc = (0, import_smithy_client.take)(data, {
    applicationComponentDetail: (_) => de_ApplicationComponentDetail(_, context),
    associatedApplications: import_smithy_client._json,
    associatedServerIds: import_smithy_client._json,
    moreApplicationResource: import_smithy_client.expectBoolean
  });
  Object.assign(contents, doc);
  return contents;
}, "de_GetApplicationComponentDetailsCommand");
var de_GetApplicationComponentStrategiesCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode !== 200 && output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const contents = (0, import_smithy_client.map)({
    $metadata: deserializeMetadata(output)
  });
  const data = (0, import_smithy_client.expectNonNull)((0, import_smithy_client.expectObject)(await parseBody(output.body, context)), "body");
  const doc = (0, import_smithy_client.take)(data, {
    applicationComponentStrategies: import_smithy_client._json
  });
  Object.assign(contents, doc);
  return contents;
}, "de_GetApplicationComponentStrategiesCommand");
var de_GetAssessmentCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode !== 200 && output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const contents = (0, import_smithy_client.map)({
    $metadata: deserializeMetadata(output)
  });
  const data = (0, import_smithy_client.expectNonNull)((0, import_smithy_client.expectObject)(await parseBody(output.body, context)), "body");
  const doc = (0, import_smithy_client.take)(data, {
    assessmentTargets: import_smithy_client._json,
    dataCollectionDetails: (_) => de_DataCollectionDetails(_, context),
    id: import_smithy_client.expectString
  });
  Object.assign(contents, doc);
  return contents;
}, "de_GetAssessmentCommand");
var de_GetImportFileTaskCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode !== 200 && output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const contents = (0, import_smithy_client.map)({
    $metadata: deserializeMetadata(output)
  });
  const data = (0, import_smithy_client.expectNonNull)((0, import_smithy_client.expectObject)(await parseBody(output.body, context)), "body");
  const doc = (0, import_smithy_client.take)(data, {
    completionTime: (_) => (0, import_smithy_client.expectNonNull)((0, import_smithy_client.parseEpochTimestamp)((0, import_smithy_client.expectNumber)(_))),
    id: import_smithy_client.expectString,
    importName: import_smithy_client.expectString,
    inputS3Bucket: import_smithy_client.expectString,
    inputS3Key: import_smithy_client.expectString,
    numberOfRecordsFailed: import_smithy_client.expectInt32,
    numberOfRecordsSuccess: import_smithy_client.expectInt32,
    startTime: (_) => (0, import_smithy_client.expectNonNull)((0, import_smithy_client.parseEpochTimestamp)((0, import_smithy_client.expectNumber)(_))),
    status: import_smithy_client.expectString,
    statusReportS3Bucket: import_smithy_client.expectString,
    statusReportS3Key: import_smithy_client.expectString
  });
  Object.assign(contents, doc);
  return contents;
}, "de_GetImportFileTaskCommand");
var de_GetLatestAssessmentIdCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode !== 200 && output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const contents = (0, import_smithy_client.map)({
    $metadata: deserializeMetadata(output)
  });
  const data = (0, import_smithy_client.expectNonNull)((0, import_smithy_client.expectObject)(await parseBody(output.body, context)), "body");
  const doc = (0, import_smithy_client.take)(data, {
    id: import_smithy_client.expectString
  });
  Object.assign(contents, doc);
  return contents;
}, "de_GetLatestAssessmentIdCommand");
var de_GetPortfolioPreferencesCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode !== 200 && output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const contents = (0, import_smithy_client.map)({
    $metadata: deserializeMetadata(output)
  });
  const data = (0, import_smithy_client.expectNonNull)((0, import_smithy_client.expectObject)(await parseBody(output.body, context)), "body");
  const doc = (0, import_smithy_client.take)(data, {
    applicationMode: import_smithy_client.expectString,
    applicationPreferences: import_smithy_client._json,
    databasePreferences: import_smithy_client._json,
    prioritizeBusinessGoals: import_smithy_client._json
  });
  Object.assign(contents, doc);
  return contents;
}, "de_GetPortfolioPreferencesCommand");
var de_GetPortfolioSummaryCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode !== 200 && output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const contents = (0, import_smithy_client.map)({
    $metadata: deserializeMetadata(output)
  });
  const data = (0, import_smithy_client.expectNonNull)((0, import_smithy_client.expectObject)(await parseBody(output.body, context)), "body");
  const doc = (0, import_smithy_client.take)(data, {
    assessmentSummary: (_) => de_AssessmentSummary(_, context)
  });
  Object.assign(contents, doc);
  return contents;
}, "de_GetPortfolioSummaryCommand");
var de_GetRecommendationReportDetailsCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode !== 200 && output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const contents = (0, import_smithy_client.map)({
    $metadata: deserializeMetadata(output)
  });
  const data = (0, import_smithy_client.expectNonNull)((0, import_smithy_client.expectObject)(await parseBody(output.body, context)), "body");
  const doc = (0, import_smithy_client.take)(data, {
    id: import_smithy_client.expectString,
    recommendationReportDetails: (_) => de_RecommendationReportDetails(_, context)
  });
  Object.assign(contents, doc);
  return contents;
}, "de_GetRecommendationReportDetailsCommand");
var de_GetServerDetailsCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode !== 200 && output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const contents = (0, import_smithy_client.map)({
    $metadata: deserializeMetadata(output)
  });
  const data = (0, import_smithy_client.expectNonNull)((0, import_smithy_client.expectObject)(await parseBody(output.body, context)), "body");
  const doc = (0, import_smithy_client.take)(data, {
    associatedApplications: import_smithy_client._json,
    nextToken: import_smithy_client.expectString,
    serverDetail: (_) => de_ServerDetail(_, context)
  });
  Object.assign(contents, doc);
  return contents;
}, "de_GetServerDetailsCommand");
var de_GetServerStrategiesCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode !== 200 && output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const contents = (0, import_smithy_client.map)({
    $metadata: deserializeMetadata(output)
  });
  const data = (0, import_smithy_client.expectNonNull)((0, import_smithy_client.expectObject)(await parseBody(output.body, context)), "body");
  const doc = (0, import_smithy_client.take)(data, {
    serverStrategies: import_smithy_client._json
  });
  Object.assign(contents, doc);
  return contents;
}, "de_GetServerStrategiesCommand");
var de_ListAnalyzableServersCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode !== 200 && output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const contents = (0, import_smithy_client.map)({
    $metadata: deserializeMetadata(output)
  });
  const data = (0, import_smithy_client.expectNonNull)((0, import_smithy_client.expectObject)(await parseBody(output.body, context)), "body");
  const doc = (0, import_smithy_client.take)(data, {
    analyzableServers: import_smithy_client._json,
    nextToken: import_smithy_client.expectString
  });
  Object.assign(contents, doc);
  return contents;
}, "de_ListAnalyzableServersCommand");
var de_ListApplicationComponentsCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode !== 200 && output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const contents = (0, import_smithy_client.map)({
    $metadata: deserializeMetadata(output)
  });
  const data = (0, import_smithy_client.expectNonNull)((0, import_smithy_client.expectObject)(await parseBody(output.body, context)), "body");
  const doc = (0, import_smithy_client.take)(data, {
    applicationComponentInfos: (_) => de_ApplicationComponentDetails(_, context),
    nextToken: import_smithy_client.expectString
  });
  Object.assign(contents, doc);
  return contents;
}, "de_ListApplicationComponentsCommand");
var de_ListCollectorsCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode !== 200 && output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const contents = (0, import_smithy_client.map)({
    $metadata: deserializeMetadata(output)
  });
  const data = (0, import_smithy_client.expectNonNull)((0, import_smithy_client.expectObject)(await parseBody(output.body, context)), "body");
  const doc = (0, import_smithy_client.take)(data, {
    Collectors: import_smithy_client._json,
    nextToken: import_smithy_client.expectString
  });
  Object.assign(contents, doc);
  return contents;
}, "de_ListCollectorsCommand");
var de_ListImportFileTaskCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode !== 200 && output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const contents = (0, import_smithy_client.map)({
    $metadata: deserializeMetadata(output)
  });
  const data = (0, import_smithy_client.expectNonNull)((0, import_smithy_client.expectObject)(await parseBody(output.body, context)), "body");
  const doc = (0, import_smithy_client.take)(data, {
    nextToken: import_smithy_client.expectString,
    taskInfos: (_) => de_ListImportFileTaskInformation(_, context)
  });
  Object.assign(contents, doc);
  return contents;
}, "de_ListImportFileTaskCommand");
var de_ListServersCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode !== 200 && output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const contents = (0, import_smithy_client.map)({
    $metadata: deserializeMetadata(output)
  });
  const data = (0, import_smithy_client.expectNonNull)((0, import_smithy_client.expectObject)(await parseBody(output.body, context)), "body");
  const doc = (0, import_smithy_client.take)(data, {
    nextToken: import_smithy_client.expectString,
    serverInfos: (_) => de_ServerDetails(_, context)
  });
  Object.assign(contents, doc);
  return contents;
}, "de_ListServersCommand");
var de_PutPortfolioPreferencesCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode !== 200 && output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const contents = (0, import_smithy_client.map)({
    $metadata: deserializeMetadata(output)
  });
  await (0, import_smithy_client.collectBody)(output.body, context);
  return contents;
}, "de_PutPortfolioPreferencesCommand");
var de_StartAssessmentCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode !== 200 && output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const contents = (0, import_smithy_client.map)({
    $metadata: deserializeMetadata(output)
  });
  const data = (0, import_smithy_client.expectNonNull)((0, import_smithy_client.expectObject)(await parseBody(output.body, context)), "body");
  const doc = (0, import_smithy_client.take)(data, {
    assessmentId: import_smithy_client.expectString
  });
  Object.assign(contents, doc);
  return contents;
}, "de_StartAssessmentCommand");
var de_StartImportFileTaskCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode !== 200 && output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const contents = (0, import_smithy_client.map)({
    $metadata: deserializeMetadata(output)
  });
  const data = (0, import_smithy_client.expectNonNull)((0, import_smithy_client.expectObject)(await parseBody(output.body, context)), "body");
  const doc = (0, import_smithy_client.take)(data, {
    id: import_smithy_client.expectString
  });
  Object.assign(contents, doc);
  return contents;
}, "de_StartImportFileTaskCommand");
var de_StartRecommendationReportGenerationCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode !== 200 && output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const contents = (0, import_smithy_client.map)({
    $metadata: deserializeMetadata(output)
  });
  const data = (0, import_smithy_client.expectNonNull)((0, import_smithy_client.expectObject)(await parseBody(output.body, context)), "body");
  const doc = (0, import_smithy_client.take)(data, {
    id: import_smithy_client.expectString
  });
  Object.assign(contents, doc);
  return contents;
}, "de_StartRecommendationReportGenerationCommand");
var de_StopAssessmentCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode !== 200 && output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const contents = (0, import_smithy_client.map)({
    $metadata: deserializeMetadata(output)
  });
  await (0, import_smithy_client.collectBody)(output.body, context);
  return contents;
}, "de_StopAssessmentCommand");
var de_UpdateApplicationComponentConfigCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode !== 200 && output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const contents = (0, import_smithy_client.map)({
    $metadata: deserializeMetadata(output)
  });
  await (0, import_smithy_client.collectBody)(output.body, context);
  return contents;
}, "de_UpdateApplicationComponentConfigCommand");
var de_UpdateServerConfigCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode !== 200 && output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const contents = (0, import_smithy_client.map)({
    $metadata: deserializeMetadata(output)
  });
  await (0, import_smithy_client.collectBody)(output.body, context);
  return contents;
}, "de_UpdateServerConfigCommand");
var de_CommandError = /* @__PURE__ */ __name(async (output, context) => {
  const parsedOutput = {
    ...output,
    body: await parseErrorBody(output.body, context)
  };
  const errorCode = loadRestJsonErrorCode(output, parsedOutput.body);
  switch (errorCode) {
    case "InternalServerException":
    case "com.amazonaws.migrationhubstrategy#InternalServerException":
      throw await de_InternalServerExceptionRes(parsedOutput, context);
    case "ResourceNotFoundException":
    case "com.amazonaws.migrationhubstrategy#ResourceNotFoundException":
      throw await de_ResourceNotFoundExceptionRes(parsedOutput, context);
    case "ThrottlingException":
    case "com.amazonaws.migrationhubstrategy#ThrottlingException":
      throw await de_ThrottlingExceptionRes(parsedOutput, context);
    case "AccessDeniedException":
    case "com.amazonaws.migrationhubstrategy#AccessDeniedException":
      throw await de_AccessDeniedExceptionRes(parsedOutput, context);
    case "ValidationException":
    case "com.amazonaws.migrationhubstrategy#ValidationException":
      throw await de_ValidationExceptionRes(parsedOutput, context);
    case "DependencyException":
    case "com.amazonaws.migrationhubstrategy#DependencyException":
      throw await de_DependencyExceptionRes(parsedOutput, context);
    case "ServiceLinkedRoleLockClientException":
    case "com.amazonaws.migrationhubstrategy#ServiceLinkedRoleLockClientException":
      throw await de_ServiceLinkedRoleLockClientExceptionRes(parsedOutput, context);
    case "ConflictException":
    case "com.amazonaws.migrationhubstrategy#ConflictException":
      throw await de_ConflictExceptionRes(parsedOutput, context);
    case "ServiceQuotaExceededException":
    case "com.amazonaws.migrationhubstrategy#ServiceQuotaExceededException":
      throw await de_ServiceQuotaExceededExceptionRes(parsedOutput, context);
    default:
      const parsedBody = parsedOutput.body;
      return throwDefaultError({
        output,
        parsedBody,
        errorCode
      });
  }
}, "de_CommandError");
var throwDefaultError = (0, import_smithy_client.withBaseException)(MigrationHubStrategyServiceException);
var de_AccessDeniedExceptionRes = /* @__PURE__ */ __name(async (parsedOutput, context) => {
  const contents = (0, import_smithy_client.map)({});
  const data = parsedOutput.body;
  const doc = (0, import_smithy_client.take)(data, {
    message: import_smithy_client.expectString
  });
  Object.assign(contents, doc);
  const exception = new AccessDeniedException({
    $metadata: deserializeMetadata(parsedOutput),
    ...contents
  });
  return (0, import_smithy_client.decorateServiceException)(exception, parsedOutput.body);
}, "de_AccessDeniedExceptionRes");
var de_ConflictExceptionRes = /* @__PURE__ */ __name(async (parsedOutput, context) => {
  const contents = (0, import_smithy_client.map)({});
  const data = parsedOutput.body;
  const doc = (0, import_smithy_client.take)(data, {
    message: import_smithy_client.expectString
  });
  Object.assign(contents, doc);
  const exception = new ConflictException({
    $metadata: deserializeMetadata(parsedOutput),
    ...contents
  });
  return (0, import_smithy_client.decorateServiceException)(exception, parsedOutput.body);
}, "de_ConflictExceptionRes");
var de_DependencyExceptionRes = /* @__PURE__ */ __name(async (parsedOutput, context) => {
  const contents = (0, import_smithy_client.map)({});
  const data = parsedOutput.body;
  const doc = (0, import_smithy_client.take)(data, {
    message: import_smithy_client.expectString
  });
  Object.assign(contents, doc);
  const exception = new DependencyException({
    $metadata: deserializeMetadata(parsedOutput),
    ...contents
  });
  return (0, import_smithy_client.decorateServiceException)(exception, parsedOutput.body);
}, "de_DependencyExceptionRes");
var de_InternalServerExceptionRes = /* @__PURE__ */ __name(async (parsedOutput, context) => {
  const contents = (0, import_smithy_client.map)({});
  const data = parsedOutput.body;
  const doc = (0, import_smithy_client.take)(data, {
    message: import_smithy_client.expectString
  });
  Object.assign(contents, doc);
  const exception = new InternalServerException({
    $metadata: deserializeMetadata(parsedOutput),
    ...contents
  });
  return (0, import_smithy_client.decorateServiceException)(exception, parsedOutput.body);
}, "de_InternalServerExceptionRes");
var de_ResourceNotFoundExceptionRes = /* @__PURE__ */ __name(async (parsedOutput, context) => {
  const contents = (0, import_smithy_client.map)({});
  const data = parsedOutput.body;
  const doc = (0, import_smithy_client.take)(data, {
    message: import_smithy_client.expectString
  });
  Object.assign(contents, doc);
  const exception = new ResourceNotFoundException({
    $metadata: deserializeMetadata(parsedOutput),
    ...contents
  });
  return (0, import_smithy_client.decorateServiceException)(exception, parsedOutput.body);
}, "de_ResourceNotFoundExceptionRes");
var de_ServiceLinkedRoleLockClientExceptionRes = /* @__PURE__ */ __name(async (parsedOutput, context) => {
  const contents = (0, import_smithy_client.map)({});
  const data = parsedOutput.body;
  const doc = (0, import_smithy_client.take)(data, {
    message: import_smithy_client.expectString
  });
  Object.assign(contents, doc);
  const exception = new ServiceLinkedRoleLockClientException({
    $metadata: deserializeMetadata(parsedOutput),
    ...contents
  });
  return (0, import_smithy_client.decorateServiceException)(exception, parsedOutput.body);
}, "de_ServiceLinkedRoleLockClientExceptionRes");
var de_ServiceQuotaExceededExceptionRes = /* @__PURE__ */ __name(async (parsedOutput, context) => {
  const contents = (0, import_smithy_client.map)({});
  const data = parsedOutput.body;
  const doc = (0, import_smithy_client.take)(data, {
    message: import_smithy_client.expectString
  });
  Object.assign(contents, doc);
  const exception = new ServiceQuotaExceededException({
    $metadata: deserializeMetadata(parsedOutput),
    ...contents
  });
  return (0, import_smithy_client.decorateServiceException)(exception, parsedOutput.body);
}, "de_ServiceQuotaExceededExceptionRes");
var de_ThrottlingExceptionRes = /* @__PURE__ */ __name(async (parsedOutput, context) => {
  const contents = (0, import_smithy_client.map)({});
  const data = parsedOutput.body;
  const doc = (0, import_smithy_client.take)(data, {
    message: import_smithy_client.expectString
  });
  Object.assign(contents, doc);
  const exception = new ThrottlingException({
    $metadata: deserializeMetadata(parsedOutput),
    ...contents
  });
  return (0, import_smithy_client.decorateServiceException)(exception, parsedOutput.body);
}, "de_ThrottlingExceptionRes");
var de_ValidationExceptionRes = /* @__PURE__ */ __name(async (parsedOutput, context) => {
  const contents = (0, import_smithy_client.map)({});
  const data = parsedOutput.body;
  const doc = (0, import_smithy_client.take)(data, {
    message: import_smithy_client.expectString
  });
  Object.assign(contents, doc);
  const exception = new ValidationException({
    $metadata: deserializeMetadata(parsedOutput),
    ...contents
  });
  return (0, import_smithy_client.decorateServiceException)(exception, parsedOutput.body);
}, "de_ValidationExceptionRes");
var de_ApplicationComponentDetail = /* @__PURE__ */ __name((output, context) => {
  return (0, import_smithy_client.take)(output, {
    analysisStatus: import_smithy_client.expectString,
    antipatternReportS3Object: import_smithy_client._json,
    antipatternReportStatus: import_smithy_client.expectString,
    antipatternReportStatusMessage: import_smithy_client.expectString,
    appType: import_smithy_client.expectString,
    appUnitError: import_smithy_client._json,
    associatedServerId: import_smithy_client.expectString,
    databaseConfigDetail: import_smithy_client._json,
    id: import_smithy_client.expectString,
    inclusionStatus: import_smithy_client.expectString,
    lastAnalyzedTimestamp: (_) => (0, import_smithy_client.expectNonNull)((0, import_smithy_client.parseEpochTimestamp)((0, import_smithy_client.expectNumber)(_))),
    listAntipatternSeveritySummary: import_smithy_client._json,
    moreServerAssociationExists: import_smithy_client.expectBoolean,
    name: import_smithy_client.expectString,
    osDriver: import_smithy_client.expectString,
    osVersion: import_smithy_client.expectString,
    recommendationSet: import_smithy_client._json,
    resourceSubType: import_smithy_client.expectString,
    resultList: import_smithy_client._json,
    runtimeStatus: import_smithy_client.expectString,
    runtimeStatusMessage: import_smithy_client.expectString,
    sourceCodeRepositories: import_smithy_client._json,
    statusMessage: import_smithy_client.expectString
  });
}, "de_ApplicationComponentDetail");
var de_ApplicationComponentDetails = /* @__PURE__ */ __name((output, context) => {
  const retVal = (output || []).filter((e) => e != null).map((entry) => {
    return de_ApplicationComponentDetail(entry, context);
  });
  return retVal;
}, "de_ApplicationComponentDetails");
var de_AssessmentSummary = /* @__PURE__ */ __name((output, context) => {
  return (0, import_smithy_client.take)(output, {
    antipatternReportS3Object: import_smithy_client._json,
    antipatternReportStatus: import_smithy_client.expectString,
    antipatternReportStatusMessage: import_smithy_client.expectString,
    lastAnalyzedTimestamp: (_) => (0, import_smithy_client.expectNonNull)((0, import_smithy_client.parseEpochTimestamp)((0, import_smithy_client.expectNumber)(_))),
    listAntipatternSeveritySummary: import_smithy_client._json,
    listApplicationComponentStatusSummary: import_smithy_client._json,
    listApplicationComponentStrategySummary: import_smithy_client._json,
    listApplicationComponentSummary: import_smithy_client._json,
    listServerStatusSummary: import_smithy_client._json,
    listServerStrategySummary: import_smithy_client._json,
    listServerSummary: import_smithy_client._json
  });
}, "de_AssessmentSummary");
var de_DataCollectionDetails = /* @__PURE__ */ __name((output, context) => {
  return (0, import_smithy_client.take)(output, {
    completionTime: (_) => (0, import_smithy_client.expectNonNull)((0, import_smithy_client.parseEpochTimestamp)((0, import_smithy_client.expectNumber)(_))),
    failed: import_smithy_client.expectInt32,
    inProgress: import_smithy_client.expectInt32,
    servers: import_smithy_client.expectInt32,
    startTime: (_) => (0, import_smithy_client.expectNonNull)((0, import_smithy_client.parseEpochTimestamp)((0, import_smithy_client.expectNumber)(_))),
    status: import_smithy_client.expectString,
    statusMessage: import_smithy_client.expectString,
    success: import_smithy_client.expectInt32
  });
}, "de_DataCollectionDetails");
var de_ImportFileTaskInformation = /* @__PURE__ */ __name((output, context) => {
  return (0, import_smithy_client.take)(output, {
    completionTime: (_) => (0, import_smithy_client.expectNonNull)((0, import_smithy_client.parseEpochTimestamp)((0, import_smithy_client.expectNumber)(_))),
    id: import_smithy_client.expectString,
    importName: import_smithy_client.expectString,
    inputS3Bucket: import_smithy_client.expectString,
    inputS3Key: import_smithy_client.expectString,
    numberOfRecordsFailed: import_smithy_client.expectInt32,
    numberOfRecordsSuccess: import_smithy_client.expectInt32,
    startTime: (_) => (0, import_smithy_client.expectNonNull)((0, import_smithy_client.parseEpochTimestamp)((0, import_smithy_client.expectNumber)(_))),
    status: import_smithy_client.expectString,
    statusReportS3Bucket: import_smithy_client.expectString,
    statusReportS3Key: import_smithy_client.expectString
  });
}, "de_ImportFileTaskInformation");
var de_ListImportFileTaskInformation = /* @__PURE__ */ __name((output, context) => {
  const retVal = (output || []).filter((e) => e != null).map((entry) => {
    return de_ImportFileTaskInformation(entry, context);
  });
  return retVal;
}, "de_ListImportFileTaskInformation");
var de_RecommendationReportDetails = /* @__PURE__ */ __name((output, context) => {
  return (0, import_smithy_client.take)(output, {
    completionTime: (_) => (0, import_smithy_client.expectNonNull)((0, import_smithy_client.parseEpochTimestamp)((0, import_smithy_client.expectNumber)(_))),
    s3Bucket: import_smithy_client.expectString,
    s3Keys: import_smithy_client._json,
    startTime: (_) => (0, import_smithy_client.expectNonNull)((0, import_smithy_client.parseEpochTimestamp)((0, import_smithy_client.expectNumber)(_))),
    status: import_smithy_client.expectString,
    statusMessage: import_smithy_client.expectString
  });
}, "de_RecommendationReportDetails");
var de_ServerDetail = /* @__PURE__ */ __name((output, context) => {
  return (0, import_smithy_client.take)(output, {
    antipatternReportS3Object: import_smithy_client._json,
    antipatternReportStatus: import_smithy_client.expectString,
    antipatternReportStatusMessage: import_smithy_client.expectString,
    applicationComponentStrategySummary: import_smithy_client._json,
    dataCollectionStatus: import_smithy_client.expectString,
    id: import_smithy_client.expectString,
    lastAnalyzedTimestamp: (_) => (0, import_smithy_client.expectNonNull)((0, import_smithy_client.parseEpochTimestamp)((0, import_smithy_client.expectNumber)(_))),
    listAntipatternSeveritySummary: import_smithy_client._json,
    name: import_smithy_client.expectString,
    recommendationSet: import_smithy_client._json,
    serverError: import_smithy_client._json,
    serverType: import_smithy_client.expectString,
    statusMessage: import_smithy_client.expectString,
    systemInfo: import_smithy_client._json
  });
}, "de_ServerDetail");
var de_ServerDetails = /* @__PURE__ */ __name((output, context) => {
  const retVal = (output || []).filter((e) => e != null).map((entry) => {
    return de_ServerDetail(entry, context);
  });
  return retVal;
}, "de_ServerDetails");
var deserializeMetadata = /* @__PURE__ */ __name((output) => ({
  httpStatusCode: output.statusCode,
  requestId: output.headers["x-amzn-requestid"] ?? output.headers["x-amzn-request-id"] ?? output.headers["x-amz-request-id"],
  extendedRequestId: output.headers["x-amz-id-2"],
  cfId: output.headers["x-amz-cf-id"]
}), "deserializeMetadata");
var collectBodyString = /* @__PURE__ */ __name((streamBody, context) => (0, import_smithy_client.collectBody)(streamBody, context).then((body) => context.utf8Encoder(body)), "collectBodyString");
var _mR = "maxResults";
var _nT = "nextToken";
var parseBody = /* @__PURE__ */ __name((streamBody, context) => collectBodyString(streamBody, context).then((encoded) => {
  if (encoded.length) {
    return JSON.parse(encoded);
  }
  return {};
}), "parseBody");
var parseErrorBody = /* @__PURE__ */ __name(async (errorBody, context) => {
  const value = await parseBody(errorBody, context);
  value.message = value.message ?? value.Message;
  return value;
}, "parseErrorBody");
var loadRestJsonErrorCode = /* @__PURE__ */ __name((output, data) => {
  const findKey = /* @__PURE__ */ __name((object, key) => Object.keys(object).find((k) => k.toLowerCase() === key.toLowerCase()), "findKey");
  const sanitizeErrorCode = /* @__PURE__ */ __name((rawValue) => {
    let cleanValue = rawValue;
    if (typeof cleanValue === "number") {
      cleanValue = cleanValue.toString();
    }
    if (cleanValue.indexOf(",") >= 0) {
      cleanValue = cleanValue.split(",")[0];
    }
    if (cleanValue.indexOf(":") >= 0) {
      cleanValue = cleanValue.split(":")[0];
    }
    if (cleanValue.indexOf("#") >= 0) {
      cleanValue = cleanValue.split("#")[1];
    }
    return cleanValue;
  }, "sanitizeErrorCode");
  const headerKey = findKey(output.headers, "x-amzn-errortype");
  if (headerKey !== void 0) {
    return sanitizeErrorCode(output.headers[headerKey]);
  }
  if (data.code !== void 0) {
    return sanitizeErrorCode(data.code);
  }
  if (data["__type"] !== void 0) {
    return sanitizeErrorCode(data["__type"]);
  }
}, "loadRestJsonErrorCode");

// src/commands/GetApplicationComponentDetailsCommand.ts
var _GetApplicationComponentDetailsCommand = class _GetApplicationComponentDetailsCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AWSMigrationHubStrategyRecommendation", "GetApplicationComponentDetails", {}).n("MigrationHubStrategyClient", "GetApplicationComponentDetailsCommand").f(void 0, void 0).ser(se_GetApplicationComponentDetailsCommand).de(de_GetApplicationComponentDetailsCommand).build() {
};
__name(_GetApplicationComponentDetailsCommand, "GetApplicationComponentDetailsCommand");
var GetApplicationComponentDetailsCommand = _GetApplicationComponentDetailsCommand;

// src/commands/GetApplicationComponentStrategiesCommand.ts




var _GetApplicationComponentStrategiesCommand = class _GetApplicationComponentStrategiesCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AWSMigrationHubStrategyRecommendation", "GetApplicationComponentStrategies", {}).n("MigrationHubStrategyClient", "GetApplicationComponentStrategiesCommand").f(void 0, void 0).ser(se_GetApplicationComponentStrategiesCommand).de(de_GetApplicationComponentStrategiesCommand).build() {
};
__name(_GetApplicationComponentStrategiesCommand, "GetApplicationComponentStrategiesCommand");
var GetApplicationComponentStrategiesCommand = _GetApplicationComponentStrategiesCommand;

// src/commands/GetAssessmentCommand.ts




var _GetAssessmentCommand = class _GetAssessmentCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AWSMigrationHubStrategyRecommendation", "GetAssessment", {}).n("MigrationHubStrategyClient", "GetAssessmentCommand").f(void 0, void 0).ser(se_GetAssessmentCommand).de(de_GetAssessmentCommand).build() {
};
__name(_GetAssessmentCommand, "GetAssessmentCommand");
var GetAssessmentCommand = _GetAssessmentCommand;

// src/commands/GetImportFileTaskCommand.ts




var _GetImportFileTaskCommand = class _GetImportFileTaskCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AWSMigrationHubStrategyRecommendation", "GetImportFileTask", {}).n("MigrationHubStrategyClient", "GetImportFileTaskCommand").f(void 0, void 0).ser(se_GetImportFileTaskCommand).de(de_GetImportFileTaskCommand).build() {
};
__name(_GetImportFileTaskCommand, "GetImportFileTaskCommand");
var GetImportFileTaskCommand = _GetImportFileTaskCommand;

// src/commands/GetLatestAssessmentIdCommand.ts




var _GetLatestAssessmentIdCommand = class _GetLatestAssessmentIdCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AWSMigrationHubStrategyRecommendation", "GetLatestAssessmentId", {}).n("MigrationHubStrategyClient", "GetLatestAssessmentIdCommand").f(void 0, void 0).ser(se_GetLatestAssessmentIdCommand).de(de_GetLatestAssessmentIdCommand).build() {
};
__name(_GetLatestAssessmentIdCommand, "GetLatestAssessmentIdCommand");
var GetLatestAssessmentIdCommand = _GetLatestAssessmentIdCommand;

// src/commands/GetPortfolioPreferencesCommand.ts




var _GetPortfolioPreferencesCommand = class _GetPortfolioPreferencesCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AWSMigrationHubStrategyRecommendation", "GetPortfolioPreferences", {}).n("MigrationHubStrategyClient", "GetPortfolioPreferencesCommand").f(void 0, void 0).ser(se_GetPortfolioPreferencesCommand).de(de_GetPortfolioPreferencesCommand).build() {
};
__name(_GetPortfolioPreferencesCommand, "GetPortfolioPreferencesCommand");
var GetPortfolioPreferencesCommand = _GetPortfolioPreferencesCommand;

// src/commands/GetPortfolioSummaryCommand.ts




var _GetPortfolioSummaryCommand = class _GetPortfolioSummaryCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AWSMigrationHubStrategyRecommendation", "GetPortfolioSummary", {}).n("MigrationHubStrategyClient", "GetPortfolioSummaryCommand").f(void 0, void 0).ser(se_GetPortfolioSummaryCommand).de(de_GetPortfolioSummaryCommand).build() {
};
__name(_GetPortfolioSummaryCommand, "GetPortfolioSummaryCommand");
var GetPortfolioSummaryCommand = _GetPortfolioSummaryCommand;

// src/commands/GetRecommendationReportDetailsCommand.ts




var _GetRecommendationReportDetailsCommand = class _GetRecommendationReportDetailsCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AWSMigrationHubStrategyRecommendation", "GetRecommendationReportDetails", {}).n("MigrationHubStrategyClient", "GetRecommendationReportDetailsCommand").f(void 0, void 0).ser(se_GetRecommendationReportDetailsCommand).de(de_GetRecommendationReportDetailsCommand).build() {
};
__name(_GetRecommendationReportDetailsCommand, "GetRecommendationReportDetailsCommand");
var GetRecommendationReportDetailsCommand = _GetRecommendationReportDetailsCommand;

// src/commands/GetServerDetailsCommand.ts




var _GetServerDetailsCommand = class _GetServerDetailsCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AWSMigrationHubStrategyRecommendation", "GetServerDetails", {}).n("MigrationHubStrategyClient", "GetServerDetailsCommand").f(void 0, void 0).ser(se_GetServerDetailsCommand).de(de_GetServerDetailsCommand).build() {
};
__name(_GetServerDetailsCommand, "GetServerDetailsCommand");
var GetServerDetailsCommand = _GetServerDetailsCommand;

// src/commands/GetServerStrategiesCommand.ts




var _GetServerStrategiesCommand = class _GetServerStrategiesCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AWSMigrationHubStrategyRecommendation", "GetServerStrategies", {}).n("MigrationHubStrategyClient", "GetServerStrategiesCommand").f(void 0, void 0).ser(se_GetServerStrategiesCommand).de(de_GetServerStrategiesCommand).build() {
};
__name(_GetServerStrategiesCommand, "GetServerStrategiesCommand");
var GetServerStrategiesCommand = _GetServerStrategiesCommand;

// src/commands/ListAnalyzableServersCommand.ts




var _ListAnalyzableServersCommand = class _ListAnalyzableServersCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AWSMigrationHubStrategyRecommendation", "ListAnalyzableServers", {}).n("MigrationHubStrategyClient", "ListAnalyzableServersCommand").f(void 0, void 0).ser(se_ListAnalyzableServersCommand).de(de_ListAnalyzableServersCommand).build() {
};
__name(_ListAnalyzableServersCommand, "ListAnalyzableServersCommand");
var ListAnalyzableServersCommand = _ListAnalyzableServersCommand;

// src/commands/ListApplicationComponentsCommand.ts




var _ListApplicationComponentsCommand = class _ListApplicationComponentsCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AWSMigrationHubStrategyRecommendation", "ListApplicationComponents", {}).n("MigrationHubStrategyClient", "ListApplicationComponentsCommand").f(void 0, void 0).ser(se_ListApplicationComponentsCommand).de(de_ListApplicationComponentsCommand).build() {
};
__name(_ListApplicationComponentsCommand, "ListApplicationComponentsCommand");
var ListApplicationComponentsCommand = _ListApplicationComponentsCommand;

// src/commands/ListCollectorsCommand.ts




var _ListCollectorsCommand = class _ListCollectorsCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AWSMigrationHubStrategyRecommendation", "ListCollectors", {}).n("MigrationHubStrategyClient", "ListCollectorsCommand").f(void 0, void 0).ser(se_ListCollectorsCommand).de(de_ListCollectorsCommand).build() {
};
__name(_ListCollectorsCommand, "ListCollectorsCommand");
var ListCollectorsCommand = _ListCollectorsCommand;

// src/commands/ListImportFileTaskCommand.ts




var _ListImportFileTaskCommand = class _ListImportFileTaskCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AWSMigrationHubStrategyRecommendation", "ListImportFileTask", {}).n("MigrationHubStrategyClient", "ListImportFileTaskCommand").f(void 0, void 0).ser(se_ListImportFileTaskCommand).de(de_ListImportFileTaskCommand).build() {
};
__name(_ListImportFileTaskCommand, "ListImportFileTaskCommand");
var ListImportFileTaskCommand = _ListImportFileTaskCommand;

// src/commands/ListServersCommand.ts




var _ListServersCommand = class _ListServersCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AWSMigrationHubStrategyRecommendation", "ListServers", {}).n("MigrationHubStrategyClient", "ListServersCommand").f(void 0, void 0).ser(se_ListServersCommand).de(de_ListServersCommand).build() {
};
__name(_ListServersCommand, "ListServersCommand");
var ListServersCommand = _ListServersCommand;

// src/commands/PutPortfolioPreferencesCommand.ts




var _PutPortfolioPreferencesCommand = class _PutPortfolioPreferencesCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AWSMigrationHubStrategyRecommendation", "PutPortfolioPreferences", {}).n("MigrationHubStrategyClient", "PutPortfolioPreferencesCommand").f(void 0, void 0).ser(se_PutPortfolioPreferencesCommand).de(de_PutPortfolioPreferencesCommand).build() {
};
__name(_PutPortfolioPreferencesCommand, "PutPortfolioPreferencesCommand");
var PutPortfolioPreferencesCommand = _PutPortfolioPreferencesCommand;

// src/commands/StartAssessmentCommand.ts




var _StartAssessmentCommand = class _StartAssessmentCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AWSMigrationHubStrategyRecommendation", "StartAssessment", {}).n("MigrationHubStrategyClient", "StartAssessmentCommand").f(void 0, void 0).ser(se_StartAssessmentCommand).de(de_StartAssessmentCommand).build() {
};
__name(_StartAssessmentCommand, "StartAssessmentCommand");
var StartAssessmentCommand = _StartAssessmentCommand;

// src/commands/StartImportFileTaskCommand.ts




var _StartImportFileTaskCommand = class _StartImportFileTaskCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AWSMigrationHubStrategyRecommendation", "StartImportFileTask", {}).n("MigrationHubStrategyClient", "StartImportFileTaskCommand").f(void 0, void 0).ser(se_StartImportFileTaskCommand).de(de_StartImportFileTaskCommand).build() {
};
__name(_StartImportFileTaskCommand, "StartImportFileTaskCommand");
var StartImportFileTaskCommand = _StartImportFileTaskCommand;

// src/commands/StartRecommendationReportGenerationCommand.ts




var _StartRecommendationReportGenerationCommand = class _StartRecommendationReportGenerationCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AWSMigrationHubStrategyRecommendation", "StartRecommendationReportGeneration", {}).n("MigrationHubStrategyClient", "StartRecommendationReportGenerationCommand").f(void 0, void 0).ser(se_StartRecommendationReportGenerationCommand).de(de_StartRecommendationReportGenerationCommand).build() {
};
__name(_StartRecommendationReportGenerationCommand, "StartRecommendationReportGenerationCommand");
var StartRecommendationReportGenerationCommand = _StartRecommendationReportGenerationCommand;

// src/commands/StopAssessmentCommand.ts




var _StopAssessmentCommand = class _StopAssessmentCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AWSMigrationHubStrategyRecommendation", "StopAssessment", {}).n("MigrationHubStrategyClient", "StopAssessmentCommand").f(void 0, void 0).ser(se_StopAssessmentCommand).de(de_StopAssessmentCommand).build() {
};
__name(_StopAssessmentCommand, "StopAssessmentCommand");
var StopAssessmentCommand = _StopAssessmentCommand;

// src/commands/UpdateApplicationComponentConfigCommand.ts




var _UpdateApplicationComponentConfigCommand = class _UpdateApplicationComponentConfigCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AWSMigrationHubStrategyRecommendation", "UpdateApplicationComponentConfig", {}).n("MigrationHubStrategyClient", "UpdateApplicationComponentConfigCommand").f(UpdateApplicationComponentConfigRequestFilterSensitiveLog, void 0).ser(se_UpdateApplicationComponentConfigCommand).de(de_UpdateApplicationComponentConfigCommand).build() {
};
__name(_UpdateApplicationComponentConfigCommand, "UpdateApplicationComponentConfigCommand");
var UpdateApplicationComponentConfigCommand = _UpdateApplicationComponentConfigCommand;

// src/commands/UpdateServerConfigCommand.ts




var _UpdateServerConfigCommand = class _UpdateServerConfigCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AWSMigrationHubStrategyRecommendation", "UpdateServerConfig", {}).n("MigrationHubStrategyClient", "UpdateServerConfigCommand").f(void 0, void 0).ser(se_UpdateServerConfigCommand).de(de_UpdateServerConfigCommand).build() {
};
__name(_UpdateServerConfigCommand, "UpdateServerConfigCommand");
var UpdateServerConfigCommand = _UpdateServerConfigCommand;

// src/MigrationHubStrategy.ts
var commands = {
  GetApplicationComponentDetailsCommand,
  GetApplicationComponentStrategiesCommand,
  GetAssessmentCommand,
  GetImportFileTaskCommand,
  GetLatestAssessmentIdCommand,
  GetPortfolioPreferencesCommand,
  GetPortfolioSummaryCommand,
  GetRecommendationReportDetailsCommand,
  GetServerDetailsCommand,
  GetServerStrategiesCommand,
  ListAnalyzableServersCommand,
  ListApplicationComponentsCommand,
  ListCollectorsCommand,
  ListImportFileTaskCommand,
  ListServersCommand,
  PutPortfolioPreferencesCommand,
  StartAssessmentCommand,
  StartImportFileTaskCommand,
  StartRecommendationReportGenerationCommand,
  StopAssessmentCommand,
  UpdateApplicationComponentConfigCommand,
  UpdateServerConfigCommand
};
var _MigrationHubStrategy = class _MigrationHubStrategy extends MigrationHubStrategyClient {
};
__name(_MigrationHubStrategy, "MigrationHubStrategy");
var MigrationHubStrategy = _MigrationHubStrategy;
(0, import_smithy_client.createAggregatedClient)(commands, MigrationHubStrategy);

// src/pagination/GetServerDetailsPaginator.ts

var paginateGetServerDetails = (0, import_core.createPaginator)(MigrationHubStrategyClient, GetServerDetailsCommand, "nextToken", "nextToken", "maxResults");

// src/pagination/ListAnalyzableServersPaginator.ts

var paginateListAnalyzableServers = (0, import_core.createPaginator)(MigrationHubStrategyClient, ListAnalyzableServersCommand, "nextToken", "nextToken", "maxResults");

// src/pagination/ListApplicationComponentsPaginator.ts

var paginateListApplicationComponents = (0, import_core.createPaginator)(MigrationHubStrategyClient, ListApplicationComponentsCommand, "nextToken", "nextToken", "maxResults");

// src/pagination/ListCollectorsPaginator.ts

var paginateListCollectors = (0, import_core.createPaginator)(MigrationHubStrategyClient, ListCollectorsCommand, "nextToken", "nextToken", "maxResults");

// src/pagination/ListImportFileTaskPaginator.ts

var paginateListImportFileTask = (0, import_core.createPaginator)(MigrationHubStrategyClient, ListImportFileTaskCommand, "nextToken", "nextToken", "maxResults");

// src/pagination/ListServersPaginator.ts

var paginateListServers = (0, import_core.createPaginator)(MigrationHubStrategyClient, ListServersCommand, "nextToken", "nextToken", "maxResults");

// src/index.ts
var import_util_endpoints = require("@aws-sdk/util-endpoints");
// Annotate the CommonJS export names for ESM import in node:

0 && (module.exports = {
  MigrationHubStrategyServiceException,
  __Client,
  MigrationHubStrategyClient,
  MigrationHubStrategy,
  $Command,
  GetApplicationComponentDetailsCommand,
  GetApplicationComponentStrategiesCommand,
  GetAssessmentCommand,
  GetImportFileTaskCommand,
  GetLatestAssessmentIdCommand,
  GetPortfolioPreferencesCommand,
  GetPortfolioSummaryCommand,
  GetRecommendationReportDetailsCommand,
  GetServerDetailsCommand,
  GetServerStrategiesCommand,
  ListAnalyzableServersCommand,
  ListApplicationComponentsCommand,
  ListCollectorsCommand,
  ListImportFileTaskCommand,
  ListServersCommand,
  PutPortfolioPreferencesCommand,
  StartAssessmentCommand,
  StartImportFileTaskCommand,
  StartRecommendationReportGenerationCommand,
  StopAssessmentCommand,
  UpdateApplicationComponentConfigCommand,
  UpdateServerConfigCommand,
  paginateGetServerDetails,
  paginateListAnalyzableServers,
  paginateListApplicationComponents,
  paginateListCollectors,
  paginateListImportFileTask,
  paginateListServers,
  AccessDeniedException,
  RuntimeAnalysisStatus,
  SrcCodeOrDbAnalysisStatus,
  AnalysisStatusUnion,
  AnalysisType,
  BinaryAnalyzerName,
  RunTimeAnalyzerName,
  SourceCodeAnalyzerName,
  AnalyzerNameUnion,
  AntipatternReportStatus,
  Severity,
  ApplicationComponentCriteria,
  AppType,
  AppUnitErrorCategory,
  InclusionStatus,
  Strategy,
  TargetDestination,
  TransformationToolName,
  ResourceSubType,
  StrategyRecommendation,
  ApplicationMode,
  AwsManagedTargetDestination,
  NoPreferenceTargetDestination,
  SelfManageTargetDestination,
  ManagementPreference,
  AssessmentDataSourceType,
  AssessmentStatus,
  RunTimeAssessmentStatus,
  ServerOsType,
  Condition,
  AuthType,
  InternalServerException,
  ResourceNotFoundException,
  ThrottlingException,
  ImportFileTaskStatus,
  ValidationException,
  DependencyException,
  DatabaseManagementPreference,
  HeterogeneousTargetDatabaseEngine,
  HomogeneousTargetDatabaseEngine,
  TargetDatabaseEngine,
  DatabaseMigrationPreference,
  RecommendationReportStatus,
  ServerErrorCategory,
  OSType,
  SortOrder,
  GroupName,
  ServiceLinkedRoleLockClientException,
  CollectorHealth,
  PipelineType,
  VersionControlType,
  ServerCriteria,
  ConflictException,
  ServiceQuotaExceededException,
  DataSourceType,
  OutputFormat,
  VersionControl,
  UpdateApplicationComponentConfigRequestFilterSensitiveLog
});
