// @azure/arm-managementpartner/dist-esm/src/models/index.js
/*
 * Copyright (c) Microsoft Corporation.
 * Licensed under the MIT License.
 *
 * Code generated by Microsoft (R) AutoRest Code Generator.
 * Changes may cause incorrect behavior and will be lost if the code is regenerated.
 */
/** Known values of {@link ManagementPartnerState} that the service accepts. */
export var KnownManagementPartnerState;
(function (KnownManagementPartnerState) {
    /** Active */
    KnownManagementPartnerState["Active"] = "Active";
    /** Deleted */
    KnownManagementPartnerState["Deleted"] = "Deleted";
})(KnownManagementPartnerState || (KnownManagementPartnerState = {}));
/** Known values of {@link ErrorResponseCode} that the service accepts. */
export var KnownErrorResponseCode;
(function (KnownErrorResponseCode) {
    /** NotFound */
    KnownErrorResponseCode["NotFound"] = "NotFound";
    /** Conflict */
    KnownErrorResponseCode["Conflict"] = "Conflict";
    /** BadRequest */
    KnownErrorResponseCode["BadRequest"] = "BadRequest";
})(KnownErrorResponseCode || (KnownErrorResponseCode = {}));
