// @aws-sdk/client-support/dist-es/models/models_0.js
import { SupportServiceException as __BaseException } from "./SupportServiceException";
export class AttachmentLimitExceeded extends __BaseException {
    constructor(opts) {
        super({
            name: "AttachmentLimitExceeded",
            $fault: "client",
            ...opts,
        });
        this.name = "AttachmentLimitExceeded";
        this.$fault = "client";
        Object.setPrototypeOf(this, AttachmentLimitExceeded.prototype);
    }
}
export class AttachmentSetExpired extends __BaseException {
    constructor(opts) {
        super({
            name: "AttachmentSetExpired",
            $fault: "client",
            ...opts,
        });
        this.name = "AttachmentSetExpired";
        this.$fault = "client";
        Object.setPrototypeOf(this, AttachmentSetExpired.prototype);
    }
}
export class AttachmentSetIdNotFound extends __BaseException {
    constructor(opts) {
        super({
            name: "AttachmentSetIdNotFound",
            $fault: "client",
            ...opts,
        });
        this.name = "AttachmentSetIdNotFound";
        this.$fault = "client";
        Object.setPrototypeOf(this, AttachmentSetIdNotFound.prototype);
    }
}
export class AttachmentSetSizeLimitExceeded extends __BaseException {
    constructor(opts) {
        super({
            name: "AttachmentSetSizeLimitExceeded",
            $fault: "client",
            ...opts,
        });
        this.name = "AttachmentSetSizeLimitExceeded";
        this.$fault = "client";
        Object.setPrototypeOf(this, AttachmentSetSizeLimitExceeded.prototype);
    }
}
export class InternalServerError extends __BaseException {
    constructor(opts) {
        super({
            name: "InternalServerError",
            $fault: "server",
            ...opts,
        });
        this.name = "InternalServerError";
        this.$fault = "server";
        Object.setPrototypeOf(this, InternalServerError.prototype);
    }
}
export class CaseIdNotFound extends __BaseException {
    constructor(opts) {
        super({
            name: "CaseIdNotFound",
            $fault: "client",
            ...opts,
        });
        this.name = "CaseIdNotFound";
        this.$fault = "client";
        Object.setPrototypeOf(this, CaseIdNotFound.prototype);
    }
}
export class AttachmentIdNotFound extends __BaseException {
    constructor(opts) {
        super({
            name: "AttachmentIdNotFound",
            $fault: "client",
            ...opts,
        });
        this.name = "AttachmentIdNotFound";
        this.$fault = "client";
        Object.setPrototypeOf(this, AttachmentIdNotFound.prototype);
    }
}
export class CaseCreationLimitExceeded extends __BaseException {
    constructor(opts) {
        super({
            name: "CaseCreationLimitExceeded",
            $fault: "client",
            ...opts,
        });
        this.name = "CaseCreationLimitExceeded";
        this.$fault = "client";
        Object.setPrototypeOf(this, CaseCreationLimitExceeded.prototype);
    }
}
export class DescribeAttachmentLimitExceeded extends __BaseException {
    constructor(opts) {
        super({
            name: "DescribeAttachmentLimitExceeded",
            $fault: "client",
            ...opts,
        });
        this.name = "DescribeAttachmentLimitExceeded";
        this.$fault = "client";
        Object.setPrototypeOf(this, DescribeAttachmentLimitExceeded.prototype);
    }
}
export class ThrottlingException extends __BaseException {
    constructor(opts) {
        super({
            name: "ThrottlingException",
            $fault: "client",
            ...opts,
        });
        this.name = "ThrottlingException";
        this.$fault = "client";
        Object.setPrototypeOf(this, ThrottlingException.prototype);
    }
}
