// @azure/arm-resourcemover/dist-esm/src/models/index.js
/*
 * Copyright (c) Microsoft Corporation.
 * Licensed under the MIT License.
 *
 * Code generated by Microsoft (R) AutoRest Code Generator.
 * Changes may cause incorrect behavior and will be lost if the code is regenerated.
 */
/** Known values of {@link ResourceIdentityType} that the service accepts. */
export var KnownResourceIdentityType;
(function (KnownResourceIdentityType) {
    /** None */
    KnownResourceIdentityType["None"] = "None";
    /** SystemAssigned */
    KnownResourceIdentityType["SystemAssigned"] = "SystemAssigned";
    /** UserAssigned */
    KnownResourceIdentityType["UserAssigned"] = "UserAssigned";
})(KnownResourceIdentityType || (KnownResourceIdentityType = {}));
/** Known values of {@link ProvisioningState} that the service accepts. */
export var KnownProvisioningState;
(function (KnownProvisioningState) {
    /** Succeeded */
    KnownProvisioningState["Succeeded"] = "Succeeded";
    /** Updating */
    KnownProvisioningState["Updating"] = "Updating";
    /** Creating */
    KnownProvisioningState["Creating"] = "Creating";
    /** Failed */
    KnownProvisioningState["Failed"] = "Failed";
})(KnownProvisioningState || (KnownProvisioningState = {}));
/** Known values of {@link MoveType} that the service accepts. */
export var KnownMoveType;
(function (KnownMoveType) {
    /** RegionToRegion */
    KnownMoveType["RegionToRegion"] = "RegionToRegion";
    /** RegionToZone */
    KnownMoveType["RegionToZone"] = "RegionToZone";
})(KnownMoveType || (KnownMoveType = {}));
/** Known values of {@link CreatedByType} that the service accepts. */
export var KnownCreatedByType;
(function (KnownCreatedByType) {
    /** User */
    KnownCreatedByType["User"] = "User";
    /** Application */
    KnownCreatedByType["Application"] = "Application";
    /** ManagedIdentity */
    KnownCreatedByType["ManagedIdentity"] = "ManagedIdentity";
    /** Key */
    KnownCreatedByType["Key"] = "Key";
})(KnownCreatedByType || (KnownCreatedByType = {}));
/** Known values of {@link MoveResourceInputType} that the service accepts. */
export var KnownMoveResourceInputType;
(function (KnownMoveResourceInputType) {
    /** MoveResourceId */
    KnownMoveResourceInputType["MoveResourceId"] = "MoveResourceId";
    /** MoveResourceSourceId */
    KnownMoveResourceInputType["MoveResourceSourceId"] = "MoveResourceSourceId";
})(KnownMoveResourceInputType || (KnownMoveResourceInputType = {}));
/** Known values of {@link MoveState} that the service accepts. */
export var KnownMoveState;
(function (KnownMoveState) {
    /** AssignmentPending */
    KnownMoveState["AssignmentPending"] = "AssignmentPending";
    /** PreparePending */
    KnownMoveState["PreparePending"] = "PreparePending";
    /** PrepareInProgress */
    KnownMoveState["PrepareInProgress"] = "PrepareInProgress";
    /** PrepareFailed */
    KnownMoveState["PrepareFailed"] = "PrepareFailed";
    /** MovePending */
    KnownMoveState["MovePending"] = "MovePending";
    /** MoveInProgress */
    KnownMoveState["MoveInProgress"] = "MoveInProgress";
    /** MoveFailed */
    KnownMoveState["MoveFailed"] = "MoveFailed";
    /** DiscardInProgress */
    KnownMoveState["DiscardInProgress"] = "DiscardInProgress";
    /** DiscardFailed */
    KnownMoveState["DiscardFailed"] = "DiscardFailed";
    /** CommitPending */
    KnownMoveState["CommitPending"] = "CommitPending";
    /** CommitInProgress */
    KnownMoveState["CommitInProgress"] = "CommitInProgress";
    /** CommitFailed */
    KnownMoveState["CommitFailed"] = "CommitFailed";
    /** Committed */
    KnownMoveState["Committed"] = "Committed";
    /** DeleteSourcePending */
    KnownMoveState["DeleteSourcePending"] = "DeleteSourcePending";
    /** ResourceMoveCompleted */
    KnownMoveState["ResourceMoveCompleted"] = "ResourceMoveCompleted";
})(KnownMoveState || (KnownMoveState = {}));
/** Known values of {@link JobName} that the service accepts. */
export var KnownJobName;
(function (KnownJobName) {
    /** InitialSync */
    KnownJobName["InitialSync"] = "InitialSync";
})(KnownJobName || (KnownJobName = {}));
/** Known values of {@link ResolutionType} that the service accepts. */
export var KnownResolutionType;
(function (KnownResolutionType) {
    /** Manual */
    KnownResolutionType["Manual"] = "Manual";
    /** Automatic */
    KnownResolutionType["Automatic"] = "Automatic";
})(KnownResolutionType || (KnownResolutionType = {}));
/** Known values of {@link DependencyType} that the service accepts. */
export var KnownDependencyType;
(function (KnownDependencyType) {
    /** RequiredForPrepare */
    KnownDependencyType["RequiredForPrepare"] = "RequiredForPrepare";
    /** RequiredForMove */
    KnownDependencyType["RequiredForMove"] = "RequiredForMove";
})(KnownDependencyType || (KnownDependencyType = {}));
/** Known values of {@link DependencyLevel} that the service accepts. */
export var KnownDependencyLevel;
(function (KnownDependencyLevel) {
    /** Direct */
    KnownDependencyLevel["Direct"] = "Direct";
    /** Descendant */
    KnownDependencyLevel["Descendant"] = "Descendant";
})(KnownDependencyLevel || (KnownDependencyLevel = {}));
/** Known values of {@link TargetAvailabilityZone} that the service accepts. */
export var KnownTargetAvailabilityZone;
(function (KnownTargetAvailabilityZone) {
    /** One */
    KnownTargetAvailabilityZone["One"] = "1";
    /** Two */
    KnownTargetAvailabilityZone["Two"] = "2";
    /** Three */
    KnownTargetAvailabilityZone["Three"] = "3";
    /** NA */
    KnownTargetAvailabilityZone["NA"] = "NA";
})(KnownTargetAvailabilityZone || (KnownTargetAvailabilityZone = {}));
/** Known values of {@link ZoneRedundant} that the service accepts. */
export var KnownZoneRedundant;
(function (KnownZoneRedundant) {
    /** Enable */
    KnownZoneRedundant["Enable"] = "Enable";
    /** Disable */
    KnownZoneRedundant["Disable"] = "Disable";
})(KnownZoneRedundant || (KnownZoneRedundant = {}));
