// @azure/arm-dashboard/dist-esm/src/models/index.js
/*
 * Copyright (c) Microsoft Corporation.
 * Licensed under the MIT License.
 *
 * Code generated by Microsoft (R) AutoRest Code Generator.
 * Changes may cause incorrect behavior and will be lost if the code is regenerated.
 */
/** Known values of {@link Origin} that the service accepts. */
export var KnownOrigin;
(function (KnownOrigin) {
    /** User */
    KnownOrigin["User"] = "user";
    /** System */
    KnownOrigin["System"] = "system";
    /** UserSystem */
    KnownOrigin["UserSystem"] = "user,system";
})(KnownOrigin || (KnownOrigin = {}));
/** Known values of {@link ActionType} that the service accepts. */
export var KnownActionType;
(function (KnownActionType) {
    /** Internal */
    KnownActionType["Internal"] = "Internal";
})(KnownActionType || (KnownActionType = {}));
/** Known values of {@link ProvisioningState} that the service accepts. */
export var KnownProvisioningState;
(function (KnownProvisioningState) {
    /** Accepted */
    KnownProvisioningState["Accepted"] = "Accepted";
    /** Creating */
    KnownProvisioningState["Creating"] = "Creating";
    /** Updating */
    KnownProvisioningState["Updating"] = "Updating";
    /** Deleting */
    KnownProvisioningState["Deleting"] = "Deleting";
    /** Succeeded */
    KnownProvisioningState["Succeeded"] = "Succeeded";
    /** Failed */
    KnownProvisioningState["Failed"] = "Failed";
    /** Canceled */
    KnownProvisioningState["Canceled"] = "Canceled";
    /** Deleted */
    KnownProvisioningState["Deleted"] = "Deleted";
    /** NotSpecified */
    KnownProvisioningState["NotSpecified"] = "NotSpecified";
})(KnownProvisioningState || (KnownProvisioningState = {}));
/** Known values of {@link PublicNetworkAccess} that the service accepts. */
export var KnownPublicNetworkAccess;
(function (KnownPublicNetworkAccess) {
    /** Enabled */
    KnownPublicNetworkAccess["Enabled"] = "Enabled";
    /** Disabled */
    KnownPublicNetworkAccess["Disabled"] = "Disabled";
})(KnownPublicNetworkAccess || (KnownPublicNetworkAccess = {}));
/** Known values of {@link ZoneRedundancy} that the service accepts. */
export var KnownZoneRedundancy;
(function (KnownZoneRedundancy) {
    /** Disabled */
    KnownZoneRedundancy["Disabled"] = "Disabled";
    /** Enabled */
    KnownZoneRedundancy["Enabled"] = "Enabled";
})(KnownZoneRedundancy || (KnownZoneRedundancy = {}));
/** Known values of {@link ApiKey} that the service accepts. */
export var KnownApiKey;
(function (KnownApiKey) {
    /** Disabled */
    KnownApiKey["Disabled"] = "Disabled";
    /** Enabled */
    KnownApiKey["Enabled"] = "Enabled";
})(KnownApiKey || (KnownApiKey = {}));
/** Known values of {@link DeterministicOutboundIP} that the service accepts. */
export var KnownDeterministicOutboundIP;
(function (KnownDeterministicOutboundIP) {
    /** Disabled */
    KnownDeterministicOutboundIP["Disabled"] = "Disabled";
    /** Enabled */
    KnownDeterministicOutboundIP["Enabled"] = "Enabled";
})(KnownDeterministicOutboundIP || (KnownDeterministicOutboundIP = {}));
/** Known values of {@link PrivateEndpointServiceConnectionStatus} that the service accepts. */
export var KnownPrivateEndpointServiceConnectionStatus;
(function (KnownPrivateEndpointServiceConnectionStatus) {
    /** Pending */
    KnownPrivateEndpointServiceConnectionStatus["Pending"] = "Pending";
    /** Approved */
    KnownPrivateEndpointServiceConnectionStatus["Approved"] = "Approved";
    /** Rejected */
    KnownPrivateEndpointServiceConnectionStatus["Rejected"] = "Rejected";
})(KnownPrivateEndpointServiceConnectionStatus || (KnownPrivateEndpointServiceConnectionStatus = {}));
/** Known values of {@link PrivateEndpointConnectionProvisioningState} that the service accepts. */
export var KnownPrivateEndpointConnectionProvisioningState;
(function (KnownPrivateEndpointConnectionProvisioningState) {
    /** Succeeded */
    KnownPrivateEndpointConnectionProvisioningState["Succeeded"] = "Succeeded";
    /** Creating */
    KnownPrivateEndpointConnectionProvisioningState["Creating"] = "Creating";
    /** Deleting */
    KnownPrivateEndpointConnectionProvisioningState["Deleting"] = "Deleting";
    /** Failed */
    KnownPrivateEndpointConnectionProvisioningState["Failed"] = "Failed";
})(KnownPrivateEndpointConnectionProvisioningState || (KnownPrivateEndpointConnectionProvisioningState = {}));
/** Known values of {@link CreatedByType} that the service accepts. */
export var KnownCreatedByType;
(function (KnownCreatedByType) {
    /** User */
    KnownCreatedByType["User"] = "User";
    /** Application */
    KnownCreatedByType["Application"] = "Application";
    /** ManagedIdentity */
    KnownCreatedByType["ManagedIdentity"] = "ManagedIdentity";
    /** Key */
    KnownCreatedByType["Key"] = "Key";
})(KnownCreatedByType || (KnownCreatedByType = {}));
/** Known values of {@link AutoGeneratedDomainNameLabelScope} that the service accepts. */
export var KnownAutoGeneratedDomainNameLabelScope;
(function (KnownAutoGeneratedDomainNameLabelScope) {
    /** TenantReuse */
    KnownAutoGeneratedDomainNameLabelScope["TenantReuse"] = "TenantReuse";
})(KnownAutoGeneratedDomainNameLabelScope || (KnownAutoGeneratedDomainNameLabelScope = {}));
/** Known values of {@link MarketplaceAutoRenew} that the service accepts. */
export var KnownMarketplaceAutoRenew;
(function (KnownMarketplaceAutoRenew) {
    /** Disabled */
    KnownMarketplaceAutoRenew["Disabled"] = "Disabled";
    /** Enabled */
    KnownMarketplaceAutoRenew["Enabled"] = "Enabled";
})(KnownMarketplaceAutoRenew || (KnownMarketplaceAutoRenew = {}));
/** Known values of {@link StartTLSPolicy} that the service accepts. */
export var KnownStartTLSPolicy;
(function (KnownStartTLSPolicy) {
    /** OpportunisticStartTLS */
    KnownStartTLSPolicy["OpportunisticStartTLS"] = "OpportunisticStartTLS";
    /** MandatoryStartTLS */
    KnownStartTLSPolicy["MandatoryStartTLS"] = "MandatoryStartTLS";
    /** NoStartTLS */
    KnownStartTLSPolicy["NoStartTLS"] = "NoStartTLS";
})(KnownStartTLSPolicy || (KnownStartTLSPolicy = {}));
/** Known values of {@link ManagedServiceIdentityType} that the service accepts. */
export var KnownManagedServiceIdentityType;
(function (KnownManagedServiceIdentityType) {
    /** None */
    KnownManagedServiceIdentityType["None"] = "None";
    /** SystemAssigned */
    KnownManagedServiceIdentityType["SystemAssigned"] = "SystemAssigned";
    /** UserAssigned */
    KnownManagedServiceIdentityType["UserAssigned"] = "UserAssigned";
    /** SystemAssignedUserAssigned */
    KnownManagedServiceIdentityType["SystemAssignedUserAssigned"] = "SystemAssigned,UserAssigned";
})(KnownManagedServiceIdentityType || (KnownManagedServiceIdentityType = {}));
/** Known values of {@link AvailablePromotion} that the service accepts. */
export var KnownAvailablePromotion;
(function (KnownAvailablePromotion) {
    /** None */
    KnownAvailablePromotion["None"] = "None";
    /** FreeTrial */
    KnownAvailablePromotion["FreeTrial"] = "FreeTrial";
})(KnownAvailablePromotion || (KnownAvailablePromotion = {}));
/** Known values of {@link ManagedPrivateEndpointConnectionStatus} that the service accepts. */
export var KnownManagedPrivateEndpointConnectionStatus;
(function (KnownManagedPrivateEndpointConnectionStatus) {
    /** Pending */
    KnownManagedPrivateEndpointConnectionStatus["Pending"] = "Pending";
    /** Approved */
    KnownManagedPrivateEndpointConnectionStatus["Approved"] = "Approved";
    /** Rejected */
    KnownManagedPrivateEndpointConnectionStatus["Rejected"] = "Rejected";
    /** Disconnected */
    KnownManagedPrivateEndpointConnectionStatus["Disconnected"] = "Disconnected";
})(KnownManagedPrivateEndpointConnectionStatus || (KnownManagedPrivateEndpointConnectionStatus = {}));
