// @azure/arm-advisor/dist-esm/src/models/index.js
/*
 * Copyright (c) Microsoft Corporation.
 * Licensed under the MIT License.
 *
 * Code generated by Microsoft (R) AutoRest Code Generator.
 * Changes may cause incorrect behavior and will be lost if the code is regenerated.
 */
/** Known values of {@link Scenario} that the service accepts. */
export var KnownScenario;
(function (KnownScenario) {
    /** Alerts */
    KnownScenario["Alerts"] = "Alerts";
})(KnownScenario || (KnownScenario = {}));
/** Known values of {@link CpuThreshold} that the service accepts. */
export var KnownCpuThreshold;
(function (KnownCpuThreshold) {
    /** Five */
    KnownCpuThreshold["Five"] = "5";
    /** Ten */
    KnownCpuThreshold["Ten"] = "10";
    /** Fifteen */
    KnownCpuThreshold["Fifteen"] = "15";
    /** Twenty */
    KnownCpuThreshold["Twenty"] = "20";
})(KnownCpuThreshold || (KnownCpuThreshold = {}));
/** Known values of {@link Category} that the service accepts. */
export var KnownCategory;
(function (KnownCategory) {
    /** HighAvailability */
    KnownCategory["HighAvailability"] = "HighAvailability";
    /** Security */
    KnownCategory["Security"] = "Security";
    /** Performance */
    KnownCategory["Performance"] = "Performance";
    /** Cost */
    KnownCategory["Cost"] = "Cost";
    /** OperationalExcellence */
    KnownCategory["OperationalExcellence"] = "OperationalExcellence";
})(KnownCategory || (KnownCategory = {}));
/** Known values of {@link DigestConfigState} that the service accepts. */
export var KnownDigestConfigState;
(function (KnownDigestConfigState) {
    /** Active */
    KnownDigestConfigState["Active"] = "Active";
    /** Disabled */
    KnownDigestConfigState["Disabled"] = "Disabled";
})(KnownDigestConfigState || (KnownDigestConfigState = {}));
/** Known values of {@link ConfigurationName} that the service accepts. */
export var KnownConfigurationName;
(function (KnownConfigurationName) {
    /** Default */
    KnownConfigurationName["Default"] = "default";
})(KnownConfigurationName || (KnownConfigurationName = {}));
/** Known values of {@link Impact} that the service accepts. */
export var KnownImpact;
(function (KnownImpact) {
    /** High */
    KnownImpact["High"] = "High";
    /** Medium */
    KnownImpact["Medium"] = "Medium";
    /** Low */
    KnownImpact["Low"] = "Low";
})(KnownImpact || (KnownImpact = {}));
/** Known values of {@link Risk} that the service accepts. */
export var KnownRisk;
(function (KnownRisk) {
    /** Error */
    KnownRisk["Error"] = "Error";
    /** Warning */
    KnownRisk["Warning"] = "Warning";
    /** None */
    KnownRisk["None"] = "None";
})(KnownRisk || (KnownRisk = {}));
