// @azure/arm-communication/dist-esm/test/communication_examples.js
/*
 * Copyright (c) Microsoft Corporation.
 * Licensed under the MIT License.
 *
 * Code generated by Microsoft (R) AutoRest Code Generator.
 * Changes may cause incorrect behavior and will be lost if the code is regenerated.
 */
import { __asyncValues, __awaiter } from "tslib";
import { env, Recorder, isPlaybackMode, } from "@azure-tools/test-recorder";
import { createTestCredential } from "@azure-tools/test-credential";
import { assert } from "chai";
import { CommunicationServiceManagementClient } from "../src/communicationServiceManagementClient";
const replaceableVariables = {
    AZURE_CLIENT_ID: "azure_client_id",
    AZURE_CLIENT_SECRET: "azure_client_secret",
    AZURE_TENANT_ID: "88888888-8888-8888-8888-888888888888",
    SUBSCRIPTION_ID: "88888888-8888-8888-8888-888888888888",
    COMMUNICATION_AZURE_AUTHORITY_HOST: "COMMUNICATION_AZURE_AUTHORITY_HOST",
    COMMUNICATION_RESOURCE_MANAGER_URL: "COMMUNICATION_RESOURCE_MANAGER_URL"
};
const recorderOptions = {
    envSetupForPlayback: replaceableVariables
};
export const testPollingOptions = {
    updateIntervalInMs: isPlaybackMode() ? 0 : undefined,
};
describe("CommunicationService test", () => {
    let recorder;
    let subscriptionId;
    let client;
    let location;
    let resourceGroup;
    let communicationServiceName;
    beforeEach(function () {
        return __awaiter(this, void 0, void 0, function* () {
            recorder = new Recorder(this.currentTest);
            yield recorder.start(recorderOptions);
            subscriptionId = env.SUBSCRIPTION_ID || '';
            // This is an example of how the environment variables are used
            const credential = createTestCredential();
            client = new CommunicationServiceManagementClient(credential, subscriptionId, recorder.configureClientOptions({}));
            location = "global";
            resourceGroup = "myjstest";
            communicationServiceName = "mycommunicationServicexxx";
        });
    });
    afterEach(function () {
        return __awaiter(this, void 0, void 0, function* () {
            yield recorder.stop();
        });
    });
    it("communicationService create test", function () {
        return __awaiter(this, void 0, void 0, function* () {
            const res = yield client.communicationServices.beginCreateOrUpdateAndWait(resourceGroup, communicationServiceName, { location: location, dataLocation: "UnitedStates" });
            assert.notEqual(res.id, undefined);
        });
    });
    it("communicationService get test", function () {
        return __awaiter(this, void 0, void 0, function* () {
            const res = yield client.communicationServices.get(resourceGroup, communicationServiceName);
            assert.equal(res.name, communicationServiceName);
        });
    });
    it("communicationService list test", function () {
        var e_1, _a;
        return __awaiter(this, void 0, void 0, function* () {
            const resArray = new Array();
            try {
                for (var _b = __asyncValues(client.communicationServices.listByResourceGroup(resourceGroup)), _c; _c = yield _b.next(), !_c.done;) {
                    let item = _c.value;
                    resArray.push(item);
                }
            }
            catch (e_1_1) { e_1 = { error: e_1_1 }; }
            finally {
                try {
                    if (_c && !_c.done && (_a = _b.return)) yield _a.call(_b);
                }
                finally { if (e_1) throw e_1.error; }
            }
            assert.equal(resArray.length, 1);
        });
    });
    it("communicationService delete test", function () {
        var e_2, _a;
        return __awaiter(this, void 0, void 0, function* () {
            const res = yield client.communicationServices.beginDeleteAndWait(resourceGroup, communicationServiceName, testPollingOptions);
            const resArray = new Array();
            try {
                for (var _b = __asyncValues(client.communicationServices.listByResourceGroup(resourceGroup)), _c; _c = yield _b.next(), !_c.done;) {
                    let item = _c.value;
                    resArray.push(item);
                }
            }
            catch (e_2_1) { e_2 = { error: e_2_1 }; }
            finally {
                try {
                    if (_c && !_c.done && (_a = _b.return)) yield _a.call(_b);
                }
                finally { if (e_2) throw e_2.error; }
            }
            assert.equal(resArray.length, 0);
        });
    });
});
