// @azure/arm-monitor/dist-esm/src/models/index.js
/*
 * Copyright (c) Microsoft Corporation.
 * Licensed under the MIT License.
 *
 * Code generated by Microsoft (R) AutoRest Code Generator.
 * Changes may cause incorrect behavior and will be lost if the code is regenerated.
 */
/** Known values of {@link ScaleRuleMetricDimensionOperationType} that the service accepts. */
export var KnownScaleRuleMetricDimensionOperationType;
(function (KnownScaleRuleMetricDimensionOperationType) {
    KnownScaleRuleMetricDimensionOperationType["Equals"] = "Equals";
    KnownScaleRuleMetricDimensionOperationType["NotEquals"] = "NotEquals";
})(KnownScaleRuleMetricDimensionOperationType || (KnownScaleRuleMetricDimensionOperationType = {}));
/** Known values of {@link MetricClass} that the service accepts. */
export var KnownMetricClass;
(function (KnownMetricClass) {
    KnownMetricClass["Availability"] = "Availability";
    KnownMetricClass["Transactions"] = "Transactions";
    KnownMetricClass["Errors"] = "Errors";
    KnownMetricClass["Latency"] = "Latency";
    KnownMetricClass["Saturation"] = "Saturation";
})(KnownMetricClass || (KnownMetricClass = {}));
/** Known values of {@link MetricUnit} that the service accepts. */
export var KnownMetricUnit;
(function (KnownMetricUnit) {
    KnownMetricUnit["Count"] = "Count";
    KnownMetricUnit["Bytes"] = "Bytes";
    KnownMetricUnit["Seconds"] = "Seconds";
    KnownMetricUnit["CountPerSecond"] = "CountPerSecond";
    KnownMetricUnit["BytesPerSecond"] = "BytesPerSecond";
    KnownMetricUnit["Percent"] = "Percent";
    KnownMetricUnit["MilliSeconds"] = "MilliSeconds";
    KnownMetricUnit["ByteSeconds"] = "ByteSeconds";
    KnownMetricUnit["Unspecified"] = "Unspecified";
    KnownMetricUnit["Cores"] = "Cores";
    KnownMetricUnit["MilliCores"] = "MilliCores";
    KnownMetricUnit["NanoCores"] = "NanoCores";
    KnownMetricUnit["BitsPerSecond"] = "BitsPerSecond";
})(KnownMetricUnit || (KnownMetricUnit = {}));
/** Known values of {@link BaselineSensitivity} that the service accepts. */
export var KnownBaselineSensitivity;
(function (KnownBaselineSensitivity) {
    KnownBaselineSensitivity["Low"] = "Low";
    KnownBaselineSensitivity["Medium"] = "Medium";
    KnownBaselineSensitivity["High"] = "High";
})(KnownBaselineSensitivity || (KnownBaselineSensitivity = {}));
/** Known values of {@link Odatatype} that the service accepts. */
export var KnownOdatatype;
(function (KnownOdatatype) {
    KnownOdatatype["MicrosoftAzureMonitorSingleResourceMultipleMetricCriteria"] = "Microsoft.Azure.Monitor.SingleResourceMultipleMetricCriteria";
    KnownOdatatype["MicrosoftAzureMonitorMultipleResourceMultipleMetricCriteria"] = "Microsoft.Azure.Monitor.MultipleResourceMultipleMetricCriteria";
    KnownOdatatype["MicrosoftAzureMonitorWebtestLocationAvailabilityCriteria"] = "Microsoft.Azure.Monitor.WebtestLocationAvailabilityCriteria";
})(KnownOdatatype || (KnownOdatatype = {}));
/** Known values of {@link Enabled} that the service accepts. */
export var KnownEnabled;
(function (KnownEnabled) {
    KnownEnabled["True"] = "true";
    KnownEnabled["False"] = "false";
})(KnownEnabled || (KnownEnabled = {}));
/** Known values of {@link ProvisioningState} that the service accepts. */
export var KnownProvisioningState;
(function (KnownProvisioningState) {
    KnownProvisioningState["Succeeded"] = "Succeeded";
    KnownProvisioningState["Deploying"] = "Deploying";
    KnownProvisioningState["Canceled"] = "Canceled";
    KnownProvisioningState["Failed"] = "Failed";
})(KnownProvisioningState || (KnownProvisioningState = {}));
/** Known values of {@link QueryType} that the service accepts. */
export var KnownQueryType;
(function (KnownQueryType) {
    KnownQueryType["ResultCount"] = "ResultCount";
})(KnownQueryType || (KnownQueryType = {}));
/** Known values of {@link NamespaceClassification} that the service accepts. */
export var KnownNamespaceClassification;
(function (KnownNamespaceClassification) {
    KnownNamespaceClassification["Platform"] = "Platform";
    KnownNamespaceClassification["Custom"] = "Custom";
    KnownNamespaceClassification["Qos"] = "Qos";
})(KnownNamespaceClassification || (KnownNamespaceClassification = {}));
/** Known values of {@link OnboardingStatus} that the service accepts. */
export var KnownOnboardingStatus;
(function (KnownOnboardingStatus) {
    KnownOnboardingStatus["Onboarded"] = "onboarded";
    KnownOnboardingStatus["NotOnboarded"] = "notOnboarded";
    KnownOnboardingStatus["Unknown"] = "unknown";
})(KnownOnboardingStatus || (KnownOnboardingStatus = {}));
/** Known values of {@link DataStatus} that the service accepts. */
export var KnownDataStatus;
(function (KnownDataStatus) {
    KnownDataStatus["Present"] = "present";
    KnownDataStatus["NotPresent"] = "notPresent";
})(KnownDataStatus || (KnownDataStatus = {}));
/** Known values of {@link KnownPublicNetworkAccessOptions} that the service accepts. */
export var KnownKnownPublicNetworkAccessOptions;
(function (KnownKnownPublicNetworkAccessOptions) {
    KnownKnownPublicNetworkAccessOptions["Enabled"] = "Enabled";
    KnownKnownPublicNetworkAccessOptions["Disabled"] = "Disabled";
})(KnownKnownPublicNetworkAccessOptions || (KnownKnownPublicNetworkAccessOptions = {}));
/** Known values of {@link KnownDataCollectionEndpointProvisioningState} that the service accepts. */
export var KnownKnownDataCollectionEndpointProvisioningState;
(function (KnownKnownDataCollectionEndpointProvisioningState) {
    KnownKnownDataCollectionEndpointProvisioningState["Creating"] = "Creating";
    KnownKnownDataCollectionEndpointProvisioningState["Updating"] = "Updating";
    KnownKnownDataCollectionEndpointProvisioningState["Deleting"] = "Deleting";
    KnownKnownDataCollectionEndpointProvisioningState["Succeeded"] = "Succeeded";
    KnownKnownDataCollectionEndpointProvisioningState["Failed"] = "Failed";
})(KnownKnownDataCollectionEndpointProvisioningState || (KnownKnownDataCollectionEndpointProvisioningState = {}));
/** Known values of {@link KnownDataCollectionEndpointResourceKind} that the service accepts. */
export var KnownKnownDataCollectionEndpointResourceKind;
(function (KnownKnownDataCollectionEndpointResourceKind) {
    KnownKnownDataCollectionEndpointResourceKind["Linux"] = "Linux";
    KnownKnownDataCollectionEndpointResourceKind["Windows"] = "Windows";
})(KnownKnownDataCollectionEndpointResourceKind || (KnownKnownDataCollectionEndpointResourceKind = {}));
/** Known values of {@link CreatedByType} that the service accepts. */
export var KnownCreatedByType;
(function (KnownCreatedByType) {
    KnownCreatedByType["User"] = "User";
    KnownCreatedByType["Application"] = "Application";
    KnownCreatedByType["ManagedIdentity"] = "ManagedIdentity";
    KnownCreatedByType["Key"] = "Key";
})(KnownCreatedByType || (KnownCreatedByType = {}));
/** Known values of {@link KnownDataCollectionRuleAssociationProvisioningState} that the service accepts. */
export var KnownKnownDataCollectionRuleAssociationProvisioningState;
(function (KnownKnownDataCollectionRuleAssociationProvisioningState) {
    KnownKnownDataCollectionRuleAssociationProvisioningState["Creating"] = "Creating";
    KnownKnownDataCollectionRuleAssociationProvisioningState["Updating"] = "Updating";
    KnownKnownDataCollectionRuleAssociationProvisioningState["Deleting"] = "Deleting";
    KnownKnownDataCollectionRuleAssociationProvisioningState["Succeeded"] = "Succeeded";
    KnownKnownDataCollectionRuleAssociationProvisioningState["Failed"] = "Failed";
})(KnownKnownDataCollectionRuleAssociationProvisioningState || (KnownKnownDataCollectionRuleAssociationProvisioningState = {}));
/** Known values of {@link KnownPerfCounterDataSourceStreams} that the service accepts. */
export var KnownKnownPerfCounterDataSourceStreams;
(function (KnownKnownPerfCounterDataSourceStreams) {
    KnownKnownPerfCounterDataSourceStreams["MicrosoftPerf"] = "Microsoft-Perf";
    KnownKnownPerfCounterDataSourceStreams["MicrosoftInsightsMetrics"] = "Microsoft-InsightsMetrics";
})(KnownKnownPerfCounterDataSourceStreams || (KnownKnownPerfCounterDataSourceStreams = {}));
/** Known values of {@link KnownWindowsEventLogDataSourceStreams} that the service accepts. */
export var KnownKnownWindowsEventLogDataSourceStreams;
(function (KnownKnownWindowsEventLogDataSourceStreams) {
    KnownKnownWindowsEventLogDataSourceStreams["MicrosoftWindowsEvent"] = "Microsoft-WindowsEvent";
    KnownKnownWindowsEventLogDataSourceStreams["MicrosoftEvent"] = "Microsoft-Event";
})(KnownKnownWindowsEventLogDataSourceStreams || (KnownKnownWindowsEventLogDataSourceStreams = {}));
/** Known values of {@link KnownSyslogDataSourceStreams} that the service accepts. */
export var KnownKnownSyslogDataSourceStreams;
(function (KnownKnownSyslogDataSourceStreams) {
    KnownKnownSyslogDataSourceStreams["MicrosoftSyslog"] = "Microsoft-Syslog";
})(KnownKnownSyslogDataSourceStreams || (KnownKnownSyslogDataSourceStreams = {}));
/** Known values of {@link KnownSyslogDataSourceFacilityNames} that the service accepts. */
export var KnownKnownSyslogDataSourceFacilityNames;
(function (KnownKnownSyslogDataSourceFacilityNames) {
    KnownKnownSyslogDataSourceFacilityNames["Auth"] = "auth";
    KnownKnownSyslogDataSourceFacilityNames["Authpriv"] = "authpriv";
    KnownKnownSyslogDataSourceFacilityNames["Cron"] = "cron";
    KnownKnownSyslogDataSourceFacilityNames["Daemon"] = "daemon";
    KnownKnownSyslogDataSourceFacilityNames["Kern"] = "kern";
    KnownKnownSyslogDataSourceFacilityNames["Lpr"] = "lpr";
    KnownKnownSyslogDataSourceFacilityNames["Mail"] = "mail";
    KnownKnownSyslogDataSourceFacilityNames["Mark"] = "mark";
    KnownKnownSyslogDataSourceFacilityNames["News"] = "news";
    KnownKnownSyslogDataSourceFacilityNames["Syslog"] = "syslog";
    KnownKnownSyslogDataSourceFacilityNames["User"] = "user";
    KnownKnownSyslogDataSourceFacilityNames["Uucp"] = "uucp";
    KnownKnownSyslogDataSourceFacilityNames["Local0"] = "local0";
    KnownKnownSyslogDataSourceFacilityNames["Local1"] = "local1";
    KnownKnownSyslogDataSourceFacilityNames["Local2"] = "local2";
    KnownKnownSyslogDataSourceFacilityNames["Local3"] = "local3";
    KnownKnownSyslogDataSourceFacilityNames["Local4"] = "local4";
    KnownKnownSyslogDataSourceFacilityNames["Local5"] = "local5";
    KnownKnownSyslogDataSourceFacilityNames["Local6"] = "local6";
    KnownKnownSyslogDataSourceFacilityNames["Local7"] = "local7";
    KnownKnownSyslogDataSourceFacilityNames["Asterisk"] = "*";
})(KnownKnownSyslogDataSourceFacilityNames || (KnownKnownSyslogDataSourceFacilityNames = {}));
/** Known values of {@link KnownSyslogDataSourceLogLevels} that the service accepts. */
export var KnownKnownSyslogDataSourceLogLevels;
(function (KnownKnownSyslogDataSourceLogLevels) {
    KnownKnownSyslogDataSourceLogLevels["Debug"] = "Debug";
    KnownKnownSyslogDataSourceLogLevels["Info"] = "Info";
    KnownKnownSyslogDataSourceLogLevels["Notice"] = "Notice";
    KnownKnownSyslogDataSourceLogLevels["Warning"] = "Warning";
    KnownKnownSyslogDataSourceLogLevels["Error"] = "Error";
    KnownKnownSyslogDataSourceLogLevels["Critical"] = "Critical";
    KnownKnownSyslogDataSourceLogLevels["Alert"] = "Alert";
    KnownKnownSyslogDataSourceLogLevels["Emergency"] = "Emergency";
    KnownKnownSyslogDataSourceLogLevels["Asterisk"] = "*";
})(KnownKnownSyslogDataSourceLogLevels || (KnownKnownSyslogDataSourceLogLevels = {}));
/** Known values of {@link KnownExtensionDataSourceStreams} that the service accepts. */
export var KnownKnownExtensionDataSourceStreams;
(function (KnownKnownExtensionDataSourceStreams) {
    KnownKnownExtensionDataSourceStreams["MicrosoftEvent"] = "Microsoft-Event";
    KnownKnownExtensionDataSourceStreams["MicrosoftInsightsMetrics"] = "Microsoft-InsightsMetrics";
    KnownKnownExtensionDataSourceStreams["MicrosoftPerf"] = "Microsoft-Perf";
    KnownKnownExtensionDataSourceStreams["MicrosoftSyslog"] = "Microsoft-Syslog";
    KnownKnownExtensionDataSourceStreams["MicrosoftWindowsEvent"] = "Microsoft-WindowsEvent";
})(KnownKnownExtensionDataSourceStreams || (KnownKnownExtensionDataSourceStreams = {}));
/** Known values of {@link KnownDataFlowStreams} that the service accepts. */
export var KnownKnownDataFlowStreams;
(function (KnownKnownDataFlowStreams) {
    KnownKnownDataFlowStreams["MicrosoftEvent"] = "Microsoft-Event";
    KnownKnownDataFlowStreams["MicrosoftInsightsMetrics"] = "Microsoft-InsightsMetrics";
    KnownKnownDataFlowStreams["MicrosoftPerf"] = "Microsoft-Perf";
    KnownKnownDataFlowStreams["MicrosoftSyslog"] = "Microsoft-Syslog";
    KnownKnownDataFlowStreams["MicrosoftWindowsEvent"] = "Microsoft-WindowsEvent";
})(KnownKnownDataFlowStreams || (KnownKnownDataFlowStreams = {}));
/** Known values of {@link KnownDataCollectionRuleProvisioningState} that the service accepts. */
export var KnownKnownDataCollectionRuleProvisioningState;
(function (KnownKnownDataCollectionRuleProvisioningState) {
    KnownKnownDataCollectionRuleProvisioningState["Creating"] = "Creating";
    KnownKnownDataCollectionRuleProvisioningState["Updating"] = "Updating";
    KnownKnownDataCollectionRuleProvisioningState["Deleting"] = "Deleting";
    KnownKnownDataCollectionRuleProvisioningState["Succeeded"] = "Succeeded";
    KnownKnownDataCollectionRuleProvisioningState["Failed"] = "Failed";
})(KnownKnownDataCollectionRuleProvisioningState || (KnownKnownDataCollectionRuleProvisioningState = {}));
/** Known values of {@link KnownDataCollectionRuleResourceKind} that the service accepts. */
export var KnownKnownDataCollectionRuleResourceKind;
(function (KnownKnownDataCollectionRuleResourceKind) {
    KnownKnownDataCollectionRuleResourceKind["Linux"] = "Linux";
    KnownKnownDataCollectionRuleResourceKind["Windows"] = "Windows";
})(KnownKnownDataCollectionRuleResourceKind || (KnownKnownDataCollectionRuleResourceKind = {}));
/** Known values of {@link Operator} that the service accepts. */
export var KnownOperator;
(function (KnownOperator) {
    KnownOperator["Equals"] = "Equals";
    KnownOperator["GreaterThan"] = "GreaterThan";
    KnownOperator["GreaterThanOrEqual"] = "GreaterThanOrEqual";
    KnownOperator["LessThan"] = "LessThan";
    KnownOperator["LessThanOrEqual"] = "LessThanOrEqual";
    KnownOperator["Include"] = "Include";
})(KnownOperator || (KnownOperator = {}));
/** Known values of {@link CriterionType} that the service accepts. */
export var KnownCriterionType;
(function (KnownCriterionType) {
    KnownCriterionType["StaticThresholdCriterion"] = "StaticThresholdCriterion";
    KnownCriterionType["DynamicThresholdCriterion"] = "DynamicThresholdCriterion";
})(KnownCriterionType || (KnownCriterionType = {}));
/** Known values of {@link AggregationTypeEnum} that the service accepts. */
export var KnownAggregationTypeEnum;
(function (KnownAggregationTypeEnum) {
    KnownAggregationTypeEnum["Average"] = "Average";
    KnownAggregationTypeEnum["Count"] = "Count";
    KnownAggregationTypeEnum["Minimum"] = "Minimum";
    KnownAggregationTypeEnum["Maximum"] = "Maximum";
    KnownAggregationTypeEnum["Total"] = "Total";
})(KnownAggregationTypeEnum || (KnownAggregationTypeEnum = {}));
/** Known values of {@link DynamicThresholdOperator} that the service accepts. */
export var KnownDynamicThresholdOperator;
(function (KnownDynamicThresholdOperator) {
    KnownDynamicThresholdOperator["GreaterThan"] = "GreaterThan";
    KnownDynamicThresholdOperator["LessThan"] = "LessThan";
    KnownDynamicThresholdOperator["GreaterOrLessThan"] = "GreaterOrLessThan";
})(KnownDynamicThresholdOperator || (KnownDynamicThresholdOperator = {}));
/** Known values of {@link DynamicThresholdSensitivity} that the service accepts. */
export var KnownDynamicThresholdSensitivity;
(function (KnownDynamicThresholdSensitivity) {
    KnownDynamicThresholdSensitivity["Low"] = "Low";
    KnownDynamicThresholdSensitivity["Medium"] = "Medium";
    KnownDynamicThresholdSensitivity["High"] = "High";
})(KnownDynamicThresholdSensitivity || (KnownDynamicThresholdSensitivity = {}));
/** Known values of {@link ConditionalOperator} that the service accepts. */
export var KnownConditionalOperator;
(function (KnownConditionalOperator) {
    KnownConditionalOperator["GreaterThanOrEqual"] = "GreaterThanOrEqual";
    KnownConditionalOperator["LessThanOrEqual"] = "LessThanOrEqual";
    KnownConditionalOperator["GreaterThan"] = "GreaterThan";
    KnownConditionalOperator["LessThan"] = "LessThan";
    KnownConditionalOperator["Equal"] = "Equal";
})(KnownConditionalOperator || (KnownConditionalOperator = {}));
/** Known values of {@link MetricTriggerType} that the service accepts. */
export var KnownMetricTriggerType;
(function (KnownMetricTriggerType) {
    KnownMetricTriggerType["Consecutive"] = "Consecutive";
    KnownMetricTriggerType["Total"] = "Total";
})(KnownMetricTriggerType || (KnownMetricTriggerType = {}));
/** Known values of {@link AlertSeverity} that the service accepts. */
export var KnownAlertSeverity;
(function (KnownAlertSeverity) {
    KnownAlertSeverity["Zero"] = "0";
    KnownAlertSeverity["One"] = "1";
    KnownAlertSeverity["Two"] = "2";
    KnownAlertSeverity["Three"] = "3";
    KnownAlertSeverity["Four"] = "4";
})(KnownAlertSeverity || (KnownAlertSeverity = {}));
