// @azure/arm-privatedns/dist-esm/src/models/index.js
/*
 * Copyright (c) Microsoft Corporation.
 * Licensed under the MIT License.
 *
 * Code generated by Microsoft (R) AutoRest Code Generator.
 * Changes may cause incorrect behavior and will be lost if the code is regenerated.
 */
/** Known values of {@link ProvisioningState} that the service accepts. */
export var KnownProvisioningState;
(function (KnownProvisioningState) {
    /** Creating */
    KnownProvisioningState["Creating"] = "Creating";
    /** Updating */
    KnownProvisioningState["Updating"] = "Updating";
    /** Deleting */
    KnownProvisioningState["Deleting"] = "Deleting";
    /** Succeeded */
    KnownProvisioningState["Succeeded"] = "Succeeded";
    /** Failed */
    KnownProvisioningState["Failed"] = "Failed";
    /** Canceled */
    KnownProvisioningState["Canceled"] = "Canceled";
})(KnownProvisioningState || (KnownProvisioningState = {}));
/** Known values of {@link VirtualNetworkLinkState} that the service accepts. */
export var KnownVirtualNetworkLinkState;
(function (KnownVirtualNetworkLinkState) {
    /** InProgress */
    KnownVirtualNetworkLinkState["InProgress"] = "InProgress";
    /** Completed */
    KnownVirtualNetworkLinkState["Completed"] = "Completed";
})(KnownVirtualNetworkLinkState || (KnownVirtualNetworkLinkState = {}));
