// @azure/arm-botservice/dist-esm/test/botservice_examples.js
/*
 * Copyright (c) Microsoft Corporation.
 * Licensed under the MIT License.
 *
 * Code generated by Microsoft (R) AutoRest Code Generator.
 * Changes may cause incorrect behavior and will be lost if the code is regenerated.
 */
import { __asyncValues, __awaiter } from "tslib";
import { env, Recorder, isPlaybackMode, } from "@azure-tools/test-recorder";
import { createTestCredential } from "@azure-tools/test-credential";
import { assert } from "chai";
import { AzureBotService } from "../src/azureBotService";
const replaceableVariables = {
    AZURE_CLIENT_ID: "azure_client_id",
    AZURE_CLIENT_SECRET: "azure_client_secret",
    AZURE_TENANT_ID: "88888888-8888-8888-8888-888888888888",
    SUBSCRIPTION_ID: "azure_subscription_id"
};
const recorderOptions = {
    envSetupForPlayback: replaceableVariables
};
export const testPollingOptions = {
    updateIntervalInMs: isPlaybackMode() ? 0 : undefined,
};
describe("Botservice test", () => {
    let recorder;
    let client;
    let subscriptionId;
    let location;
    let resourceGroup;
    let botresourceName;
    beforeEach(function () {
        return __awaiter(this, void 0, void 0, function* () {
            recorder = new Recorder(this.currentTest);
            yield recorder.start(recorderOptions);
            subscriptionId = env.SUBSCRIPTION_ID || '';
            // This is an example of how the environment variables are used
            const credential = createTestCredential();
            client = new AzureBotService(credential, subscriptionId, recorder.configureClientOptions({}));
            location = "global";
            resourceGroup = "myjstest";
            botresourceName = "mybotxxx";
        });
    });
    afterEach(function () {
        return __awaiter(this, void 0, void 0, function* () {
            yield recorder.stop();
        });
    });
    it("bots create test", function () {
        return __awaiter(this, void 0, void 0, function* () {
            const res = yield client.bots.create(resourceGroup, botresourceName, {
                sku: {
                    name: "S1"
                },
                etag: "etag1",
                tags: {
                    tag1: "value1",
                    tag2: "value2"
                },
                kind: "sdk",
                properties: {
                    description: "The description of the bot",
                    developerAppInsightsApiKey: "w24iw5ocbhcig71su7ibaj63hey5ieaozeuwdv2r",
                    developerAppInsightKey: "59513bad-10a7-4d41-b4d0-b1c34c6af52a",
                    developerAppInsightsApplicationId: "cf03484e-3fdb-4b5e-9ad7-94bde32e5a2b",
                    displayName: "this is a test bot",
                    endpoint: "https://bing.com/messages/",
                    msaAppId: "41a220b9-6571-4f0b-bbd2-43f1c1d82f51"
                },
                location: "global"
            });
            assert.equal(res.name, botresourceName);
        });
    });
    it("bots get test", function () {
        return __awaiter(this, void 0, void 0, function* () {
            const res = yield client.bots.get(resourceGroup, botresourceName);
            assert.equal(res.name, botresourceName);
        });
    });
    it("MsTeamsChannel create test", function () {
        return __awaiter(this, void 0, void 0, function* () {
            const res = yield client.channels.create(resourceGroup, botresourceName, "MsTeamsChannel", {
                location,
                properties: {
                    channelName: "MsTeamsChannel",
                    properties: {
                        isEnabled: true,
                    }
                }
            });
            assert.equal(res.name, "mybotxxx/MsTeamsChannel");
        });
    });
    it("MsTeamsChannel get test", function () {
        return __awaiter(this, void 0, void 0, function* () {
            const res = yield client.channels.get(resourceGroup, botresourceName, "MsTeamsChannel");
            assert.equal(res.name, "mybotxxx/MsTeamsChannel");
        });
    });
    it("MsTeamsChannel list test", function () {
        var e_1, _a;
        return __awaiter(this, void 0, void 0, function* () {
            const resArray = new Array();
            try {
                for (var _b = __asyncValues(client.channels.listByResourceGroup(resourceGroup, botresourceName)), _c; _c = yield _b.next(), !_c.done;) {
                    let item = _c.value;
                    resArray.push(item);
                }
            }
            catch (e_1_1) { e_1 = { error: e_1_1 }; }
            finally {
                try {
                    if (_c && !_c.done && (_a = _b.return)) yield _a.call(_b);
                }
                finally { if (e_1) throw e_1.error; }
            }
            assert.equal(resArray.length, 3); //bot has 2 basic channels
        });
    });
    it("MsTeamsChannel delete test", function () {
        var e_2, _a;
        return __awaiter(this, void 0, void 0, function* () {
            const res = yield client.channels.delete(resourceGroup, botresourceName, "MsTeamsChannel");
            const resArray = new Array();
            try {
                for (var _b = __asyncValues(client.channels.listByResourceGroup(resourceGroup, botresourceName)), _c; _c = yield _b.next(), !_c.done;) {
                    let item = _c.value;
                    resArray.push(item);
                }
            }
            catch (e_2_1) { e_2 = { error: e_2_1 }; }
            finally {
                try {
                    if (_c && !_c.done && (_a = _b.return)) yield _a.call(_b);
                }
                finally { if (e_2) throw e_2.error; }
            }
            assert.equal(resArray.length, 2); //bot has 2 basic channels
        });
    });
    it("bots delete test", function () {
        var e_3, _a;
        return __awaiter(this, void 0, void 0, function* () {
            const res = yield client.bots.delete(resourceGroup, botresourceName);
            const resArray = new Array();
            try {
                for (var _b = __asyncValues(client.bots.list()), _c; _c = yield _b.next(), !_c.done;) {
                    let item = _c.value;
                    resArray.push(item);
                }
            }
            catch (e_3_1) { e_3 = { error: e_3_1 }; }
            finally {
                try {
                    if (_c && !_c.done && (_a = _b.return)) yield _a.call(_b);
                }
                finally { if (e_3) throw e_3.error; }
            }
            assert.equal(resArray.length, 0);
        });
    });
});
