// @azure/arm-securityinsight/dist-esm/test/securityinsights_examples.spec.js
/*
 * Copyright (c) Microsoft Corporation.
 * Licensed under the MIT License.
 *
 * Code generated by Microsoft (R) AutoRest Code Generator.
 * Changes may cause incorrect behavior and will be lost if the code is regenerated.
 */
import { __asyncValues, __awaiter } from "tslib";
import { env, Recorder, isPlaybackMode, } from "@azure-tools/test-recorder";
import { createTestCredential } from "@azure-tools/test-credential";
import { assert } from "chai";
import { SecurityInsights } from "../src/securityInsights";
const replaceableVariables = {
    AZURE_CLIENT_ID: "azure_client_id",
    AZURE_CLIENT_SECRET: "azure_client_secret",
    AZURE_TENANT_ID: "88888888-8888-8888-8888-888888888888",
    SUBSCRIPTION_ID: "azure_subscription_id"
};
const recorderOptions = {
    envSetupForPlayback: replaceableVariables
};
export const testPollingOptions = {
    updateIntervalInMs: isPlaybackMode() ? 0 : undefined,
};
describe("securityinsight test", () => {
    let recorder;
    let subscriptionId;
    let client;
    let location;
    let resourceGroup;
    let workspaceName;
    let sentinelOnboardingStateName;
    beforeEach(function () {
        return __awaiter(this, void 0, void 0, function* () {
            recorder = new Recorder(this.currentTest);
            yield recorder.start(recorderOptions);
            subscriptionId = env.SUBSCRIPTION_ID || '';
            // This is an example of how the environment variables are used
            const credential = createTestCredential();
            client = new SecurityInsights(credential, subscriptionId, recorder.configureClientOptions({}));
            location = "eastus";
            resourceGroup = "myjstest";
            workspaceName = "myWorkspace";
            sentinelOnboardingStateName = "default";
        });
    });
    afterEach(function () {
        return __awaiter(this, void 0, void 0, function* () {
            yield recorder.stop();
        });
    });
    it("sentinel create test", function () {
        return __awaiter(this, void 0, void 0, function* () {
            const sentinelOnboardingStateParameter = {
                customerManagedKey: false
            };
            const options = {
                sentinelOnboardingStateParameter
            };
            const res = yield client.sentinelOnboardingStates.create(resourceGroup, workspaceName, sentinelOnboardingStateName, options);
            assert.equal(res.name, sentinelOnboardingStateName);
        });
    });
    it("sentinel get test", function () {
        return __awaiter(this, void 0, void 0, function* () {
            const res = yield client.sentinelOnboardingStates.get(resourceGroup, workspaceName, sentinelOnboardingStateName);
            assert.equal(res.name, sentinelOnboardingStateName);
        });
    });
    it("sentinel list test", function () {
        var e_1, _a;
        return __awaiter(this, void 0, void 0, function* () {
            const resArray = new Array();
            const result = yield client.sentinelOnboardingStates.list(resourceGroup, workspaceName);
            try {
                for (var _b = __asyncValues(result.value), _c; _c = yield _b.next(), !_c.done;) {
                    let item = _c.value;
                    resArray.push(item);
                }
            }
            catch (e_1_1) { e_1 = { error: e_1_1 }; }
            finally {
                try {
                    if (_c && !_c.done && (_a = _b.return)) yield _a.call(_b);
                }
                finally { if (e_1) throw e_1.error; }
            }
            assert.equal(resArray.length, 1);
        });
    });
    it("sentinel delete test", function () {
        return __awaiter(this, void 0, void 0, function* () {
            const resArray = new Array();
            const res = yield client.sentinelOnboardingStates.delete(resourceGroup, workspaceName, sentinelOnboardingStateName);
        });
    });
});
