// @azure/arm-support/dist-esm/src/operations/communications.js
/*
 * Copyright (c) Microsoft Corporation.
 * Licensed under the MIT License.
 *
 * Code generated by Microsoft (R) AutoRest Code Generator.
 * Changes may cause incorrect behavior and will be lost if the code is regenerated.
 */
import { __asyncDelegator, __asyncGenerator, __asyncValues, __await, __awaiter } from "tslib";
import { setContinuationToken } from "../pagingHelper";
import * as coreClient from "@azure/core-client";
import * as Mappers from "../models/mappers";
import * as Parameters from "../models/parameters";
import { LroEngine } from "@azure/core-lro";
import { LroImpl } from "../lroImpl";
/// <reference lib="esnext.asynciterable" />
/** Class containing Communications operations. */
export class CommunicationsImpl {
    /**
     * Initialize a new instance of the class Communications class.
     * @param client Reference to the service client
     */
    constructor(client) {
        this.client = client;
    }
    /**
     * Lists all communications (attachments not included) for a support ticket. <br/></br> You can also
     * filter support ticket communications by _CreatedDate_ or _CommunicationType_ using the $filter
     * parameter. The only type of communication supported today is _Web_. Output will be a paged result
     * with _nextLink_, using which you can retrieve the next set of Communication results.
     * <br/><br/>Support ticket data is available for 18 months after ticket creation. If a ticket was
     * created more than 18 months ago, a request for data might cause an error.
     * @param supportTicketName Support ticket name.
     * @param options The options parameters.
     */
    list(supportTicketName, options) {
        const iter = this.listPagingAll(supportTicketName, options);
        return {
            next() {
                return iter.next();
            },
            [Symbol.asyncIterator]() {
                return this;
            },
            byPage: (settings) => {
                if (settings === null || settings === void 0 ? void 0 : settings.maxPageSize) {
                    throw new Error("maxPageSize is not supported by this operation.");
                }
                return this.listPagingPage(supportTicketName, options, settings);
            }
        };
    }
    listPagingPage(supportTicketName, options, settings) {
        return __asyncGenerator(this, arguments, function* listPagingPage_1() {
            let result;
            let continuationToken = settings === null || settings === void 0 ? void 0 : settings.continuationToken;
            if (!continuationToken) {
                result = yield __await(this._list(supportTicketName, options));
                let page = result.value || [];
                continuationToken = result.nextLink;
                setContinuationToken(page, continuationToken);
                yield yield __await(page);
            }
            while (continuationToken) {
                result = yield __await(this._listNext(supportTicketName, continuationToken, options));
                continuationToken = result.nextLink;
                let page = result.value || [];
                setContinuationToken(page, continuationToken);
                yield yield __await(page);
            }
        });
    }
    listPagingAll(supportTicketName, options) {
        return __asyncGenerator(this, arguments, function* listPagingAll_1() {
            var e_1, _a;
            try {
                for (var _b = __asyncValues(this.listPagingPage(supportTicketName, options)), _c; _c = yield __await(_b.next()), !_c.done;) {
                    const page = _c.value;
                    yield __await(yield* __asyncDelegator(__asyncValues(page)));
                }
            }
            catch (e_1_1) { e_1 = { error: e_1_1 }; }
            finally {
                try {
                    if (_c && !_c.done && (_a = _b.return)) yield __await(_a.call(_b));
                }
                finally { if (e_1) throw e_1.error; }
            }
        });
    }
    /**
     * Check the availability of a resource name. This API should be used to check the uniqueness of the
     * name for adding a new communication to the support ticket.
     * @param supportTicketName Support ticket name.
     * @param checkNameAvailabilityInput Input to check.
     * @param options The options parameters.
     */
    checkNameAvailability(supportTicketName, checkNameAvailabilityInput, options) {
        return this.client.sendOperationRequest({ supportTicketName, checkNameAvailabilityInput, options }, checkNameAvailabilityOperationSpec);
    }
    /**
     * Lists all communications (attachments not included) for a support ticket. <br/></br> You can also
     * filter support ticket communications by _CreatedDate_ or _CommunicationType_ using the $filter
     * parameter. The only type of communication supported today is _Web_. Output will be a paged result
     * with _nextLink_, using which you can retrieve the next set of Communication results.
     * <br/><br/>Support ticket data is available for 18 months after ticket creation. If a ticket was
     * created more than 18 months ago, a request for data might cause an error.
     * @param supportTicketName Support ticket name.
     * @param options The options parameters.
     */
    _list(supportTicketName, options) {
        return this.client.sendOperationRequest({ supportTicketName, options }, listOperationSpec);
    }
    /**
     * Returns communication details for a support ticket.
     * @param supportTicketName Support ticket name.
     * @param communicationName Communication name.
     * @param options The options parameters.
     */
    get(supportTicketName, communicationName, options) {
        return this.client.sendOperationRequest({ supportTicketName, communicationName, options }, getOperationSpec);
    }
    /**
     * Adds a new customer communication to an Azure support ticket.
     * @param supportTicketName Support ticket name.
     * @param communicationName Communication name.
     * @param createCommunicationParameters Communication object.
     * @param options The options parameters.
     */
    beginCreate(supportTicketName, communicationName, createCommunicationParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            const directSendOperation = (args, spec) => __awaiter(this, void 0, void 0, function* () {
                return this.client.sendOperationRequest(args, spec);
            });
            const sendOperation = (args, spec) => __awaiter(this, void 0, void 0, function* () {
                var _a;
                let currentRawResponse = undefined;
                const providedCallback = (_a = args.options) === null || _a === void 0 ? void 0 : _a.onResponse;
                const callback = (rawResponse, flatResponse) => {
                    currentRawResponse = rawResponse;
                    providedCallback === null || providedCallback === void 0 ? void 0 : providedCallback(rawResponse, flatResponse);
                };
                const updatedArgs = Object.assign(Object.assign({}, args), { options: Object.assign(Object.assign({}, args.options), { onResponse: callback }) });
                const flatResponse = yield directSendOperation(updatedArgs, spec);
                return {
                    flatResponse,
                    rawResponse: {
                        statusCode: currentRawResponse.status,
                        body: currentRawResponse.parsedBody,
                        headers: currentRawResponse.headers.toJSON()
                    }
                };
            });
            const lro = new LroImpl(sendOperation, {
                supportTicketName,
                communicationName,
                createCommunicationParameters,
                options
            }, createOperationSpec);
            const poller = new LroEngine(lro, {
                resumeFrom: options === null || options === void 0 ? void 0 : options.resumeFrom,
                intervalInMs: options === null || options === void 0 ? void 0 : options.updateIntervalInMs,
                lroResourceLocationConfig: "azure-async-operation"
            });
            yield poller.poll();
            return poller;
        });
    }
    /**
     * Adds a new customer communication to an Azure support ticket.
     * @param supportTicketName Support ticket name.
     * @param communicationName Communication name.
     * @param createCommunicationParameters Communication object.
     * @param options The options parameters.
     */
    beginCreateAndWait(supportTicketName, communicationName, createCommunicationParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            const poller = yield this.beginCreate(supportTicketName, communicationName, createCommunicationParameters, options);
            return poller.pollUntilDone();
        });
    }
    /**
     * ListNext
     * @param supportTicketName Support ticket name.
     * @param nextLink The nextLink from the previous successful call to the List method.
     * @param options The options parameters.
     */
    _listNext(supportTicketName, nextLink, options) {
        return this.client.sendOperationRequest({ supportTicketName, nextLink, options }, listNextOperationSpec);
    }
}
// Operation Specifications
const serializer = coreClient.createSerializer(Mappers, /* isXml */ false);
const checkNameAvailabilityOperationSpec = {
    path: "/subscriptions/{subscriptionId}/providers/Microsoft.Support/supportTickets/{supportTicketName}/checkNameAvailability",
    httpMethod: "POST",
    responses: {
        200: {
            bodyMapper: Mappers.CheckNameAvailabilityOutput
        },
        default: {
            bodyMapper: Mappers.ExceptionResponse
        }
    },
    requestBody: Parameters.checkNameAvailabilityInput,
    queryParameters: [Parameters.apiVersion],
    urlParameters: [
        Parameters.$host,
        Parameters.subscriptionId,
        Parameters.supportTicketName
    ],
    headerParameters: [Parameters.accept, Parameters.contentType],
    mediaType: "json",
    serializer
};
const listOperationSpec = {
    path: "/subscriptions/{subscriptionId}/providers/Microsoft.Support/supportTickets/{supportTicketName}/communications",
    httpMethod: "GET",
    responses: {
        200: {
            bodyMapper: Mappers.CommunicationsListResult
        },
        default: {
            bodyMapper: Mappers.ExceptionResponse
        }
    },
    queryParameters: [Parameters.apiVersion, Parameters.top, Parameters.filter],
    urlParameters: [
        Parameters.$host,
        Parameters.subscriptionId,
        Parameters.supportTicketName
    ],
    headerParameters: [Parameters.accept],
    serializer
};
const getOperationSpec = {
    path: "/subscriptions/{subscriptionId}/providers/Microsoft.Support/supportTickets/{supportTicketName}/communications/{communicationName}",
    httpMethod: "GET",
    responses: {
        200: {
            bodyMapper: Mappers.CommunicationDetails
        },
        default: {
            bodyMapper: Mappers.ExceptionResponse
        }
    },
    queryParameters: [Parameters.apiVersion],
    urlParameters: [
        Parameters.$host,
        Parameters.subscriptionId,
        Parameters.supportTicketName,
        Parameters.communicationName
    ],
    headerParameters: [Parameters.accept],
    serializer
};
const createOperationSpec = {
    path: "/subscriptions/{subscriptionId}/providers/Microsoft.Support/supportTickets/{supportTicketName}/communications/{communicationName}",
    httpMethod: "PUT",
    responses: {
        200: {
            bodyMapper: Mappers.CommunicationDetails
        },
        201: {
            bodyMapper: Mappers.CommunicationDetails
        },
        202: {
            bodyMapper: Mappers.CommunicationDetails
        },
        204: {
            bodyMapper: Mappers.CommunicationDetails
        },
        default: {
            bodyMapper: Mappers.ExceptionResponse
        }
    },
    requestBody: Parameters.createCommunicationParameters,
    queryParameters: [Parameters.apiVersion],
    urlParameters: [
        Parameters.$host,
        Parameters.subscriptionId,
        Parameters.supportTicketName,
        Parameters.communicationName
    ],
    headerParameters: [Parameters.accept, Parameters.contentType],
    mediaType: "json",
    serializer
};
const listNextOperationSpec = {
    path: "{nextLink}",
    httpMethod: "GET",
    responses: {
        200: {
            bodyMapper: Mappers.CommunicationsListResult
        },
        default: {
            bodyMapper: Mappers.ExceptionResponse
        }
    },
    urlParameters: [
        Parameters.$host,
        Parameters.subscriptionId,
        Parameters.supportTicketName,
        Parameters.nextLink
    ],
    headerParameters: [Parameters.accept],
    serializer
};
