// @azure/arm-datalake-analytics/esm/models/parameters.js
/*
 * Copyright (c) Microsoft Corporation.
 * Licensed under the MIT License.
 *
 * Code generated by Microsoft (R) AutoRest Code Generator.
 * Changes may cause incorrect behavior and will be lost if the code is
 * regenerated.
 */
export var acceptLanguage = {
    parameterPath: "acceptLanguage",
    mapper: {
        serializedName: "accept-language",
        defaultValue: 'en-US',
        type: {
            name: "String"
        }
    }
};
export var accountName = {
    parameterPath: "accountName",
    mapper: {
        required: true,
        serializedName: "accountName",
        type: {
            name: "String"
        }
    }
};
export var apiVersion = {
    parameterPath: "apiVersion",
    mapper: {
        required: true,
        serializedName: "api-version",
        type: {
            name: "String"
        }
    }
};
export var computePolicyName = {
    parameterPath: "computePolicyName",
    mapper: {
        required: true,
        serializedName: "computePolicyName",
        type: {
            name: "String"
        }
    }
};
export var containerName = {
    parameterPath: "containerName",
    mapper: {
        required: true,
        serializedName: "containerName",
        type: {
            name: "String"
        }
    }
};
export var count = {
    parameterPath: [
        "options",
        "count"
    ],
    mapper: {
        serializedName: "$count",
        type: {
            name: "Boolean"
        }
    }
};
export var dataLakeStoreAccountName = {
    parameterPath: "dataLakeStoreAccountName",
    mapper: {
        required: true,
        serializedName: "dataLakeStoreAccountName",
        type: {
            name: "String"
        }
    }
};
export var filter = {
    parameterPath: [
        "options",
        "filter"
    ],
    mapper: {
        serializedName: "$filter",
        type: {
            name: "String"
        }
    }
};
export var firewallRuleName = {
    parameterPath: "firewallRuleName",
    mapper: {
        required: true,
        serializedName: "firewallRuleName",
        type: {
            name: "String"
        }
    }
};
export var location = {
    parameterPath: "location",
    mapper: {
        required: true,
        serializedName: "location",
        type: {
            name: "String"
        }
    }
};
export var nextPageLink = {
    parameterPath: "nextPageLink",
    mapper: {
        required: true,
        serializedName: "nextLink",
        type: {
            name: "String"
        }
    },
    skipEncoding: true
};
export var orderby = {
    parameterPath: [
        "options",
        "orderby"
    ],
    mapper: {
        serializedName: "$orderby",
        type: {
            name: "String"
        }
    }
};
export var resourceGroupName = {
    parameterPath: "resourceGroupName",
    mapper: {
        required: true,
        serializedName: "resourceGroupName",
        type: {
            name: "String"
        }
    }
};
export var select = {
    parameterPath: [
        "options",
        "select"
    ],
    mapper: {
        serializedName: "$select",
        type: {
            name: "String"
        }
    }
};
export var skip = {
    parameterPath: [
        "options",
        "skip"
    ],
    mapper: {
        serializedName: "$skip",
        constraints: {
            InclusiveMinimum: 1
        },
        type: {
            name: "Number"
        }
    }
};
export var storageAccountName = {
    parameterPath: "storageAccountName",
    mapper: {
        required: true,
        serializedName: "storageAccountName",
        type: {
            name: "String"
        }
    }
};
export var subscriptionId = {
    parameterPath: "subscriptionId",
    mapper: {
        required: true,
        serializedName: "subscriptionId",
        type: {
            name: "String"
        }
    }
};
export var top = {
    parameterPath: [
        "options",
        "top"
    ],
    mapper: {
        serializedName: "$top",
        constraints: {
            InclusiveMinimum: 1
        },
        type: {
            name: "Number"
        }
    }
};
