// @azure/arm-reservations/dist-esm/test/reservations_examples.spec.js
/*
 * Copyright (c) Microsoft Corporation.
 * Licensed under the MIT License.
 *
 * Code generated by Microsoft (R) AutoRest Code Generator.
 * Changes may cause incorrect behavior and will be lost if the code is regenerated.
 */
import { __asyncValues, __awaiter } from "tslib";
import { env, Recorder, isPlaybackMode, } from "@azure-tools/test-recorder";
import { createTestCredential } from "@azure-tools/test-credential";
import { assert } from "chai";
import { AzureReservationAPI } from "../src/azureReservationAPI";
const replaceableVariables = {
    AZURE_CLIENT_ID: "azure_client_id",
    AZURE_CLIENT_SECRET: "azure_client_secret",
    AZURE_TENANT_ID: "88888888-8888-8888-8888-888888888888",
    SUBSCRIPTION_ID: "azure_subscription_id"
};
const recorderOptions = {
    envSetupForPlayback: replaceableVariables
};
export const testPollingOptions = {
    updateIntervalInMs: isPlaybackMode() ? 0 : undefined,
};
describe("Datafactory test", () => {
    let recorder;
    let subscriptionId;
    let client;
    let location;
    let resourceGroup;
    let providerId;
    let resourceName;
    let createQuotaRequest;
    beforeEach(function () {
        return __awaiter(this, void 0, void 0, function* () {
            recorder = new Recorder(this.currentTest);
            yield recorder.start(recorderOptions);
            subscriptionId = env.SUBSCRIPTION_ID || '';
            // This is an example of how the environment variables are used
            const credential = createTestCredential();
            client = new AzureReservationAPI(credential, recorder.configureClientOptions({}));
            location = "eastus";
            resourceGroup = "myjstest";
            providerId = "Microsoft.Compute";
            resourceName = "standardFSv2Family";
            createQuotaRequest = {
                properties: {
                    name: { value: "standardFSv2Family" },
                    limit: 200,
                    unit: "Count"
                }
            };
        });
    });
    afterEach(function () {
        return __awaiter(this, void 0, void 0, function* () {
            yield recorder.stop();
        });
    });
    // it("quota create test", async function () {
    //   const res = await client.quota.beginCreateOrUpdateAndWait(
    //     subscriptionId,
    //     providerId,
    //     location,
    //     resourceName,
    //     createQuotaRequest
    //   );
    //   assert.equal(res.name, resourceName);
    // });
    it("quota get test", function () {
        return __awaiter(this, void 0, void 0, function* () {
            const res = yield client.quota.get(subscriptionId, providerId, location, resourceName);
            assert.equal(res.name, resourceName);
        });
    });
    it("quota list test", function () {
        var e_1, _a;
        return __awaiter(this, void 0, void 0, function* () {
            const resArray = new Array();
            try {
                for (var _b = __asyncValues(client.quota.list(subscriptionId, providerId, location)), _c; _c = yield _b.next(), !_c.done;) {
                    let item = _c.value;
                    resArray.push(item);
                }
            }
            catch (e_1_1) { e_1 = { error: e_1_1 }; }
            finally {
                try {
                    if (_c && !_c.done && (_a = _b.return)) yield _a.call(_b);
                }
                finally { if (e_1) throw e_1.error; }
            }
            assert.equal(resArray.length, 108);
        });
    });
});
