// @azure/arm-recoveryservices-siterecovery/dist-esm/src/operations/migrationRecoveryPoints.js
/*
 * Copyright (c) Microsoft Corporation.
 * Licensed under the MIT License.
 *
 * Code generated by Microsoft (R) AutoRest Code Generator.
 * Changes may cause incorrect behavior and will be lost if the code is regenerated.
 */
import { __asyncDelegator, __asyncGenerator, __asyncValues, __await } from "tslib";
import { setContinuationToken } from "../pagingHelper";
import * as coreClient from "@azure/core-client";
import * as Mappers from "../models/mappers";
import * as Parameters from "../models/parameters";
/// <reference lib="esnext.asynciterable" />
/** Class containing MigrationRecoveryPoints operations. */
export class MigrationRecoveryPointsImpl {
    /**
     * Initialize a new instance of the class MigrationRecoveryPoints class.
     * @param client Reference to the service client
     */
    constructor(client) {
        this.client = client;
    }
    /**
     * Gets the recovery points for a migration item.
     * @param resourceName The name of the recovery services vault.
     * @param resourceGroupName The name of the resource group where the recovery services vault is
     *                          present.
     * @param fabricName Fabric unique name.
     * @param protectionContainerName Protection container name.
     * @param migrationItemName Migration item name.
     * @param options The options parameters.
     */
    listByReplicationMigrationItems(resourceName, resourceGroupName, fabricName, protectionContainerName, migrationItemName, options) {
        const iter = this.listByReplicationMigrationItemsPagingAll(resourceName, resourceGroupName, fabricName, protectionContainerName, migrationItemName, options);
        return {
            next() {
                return iter.next();
            },
            [Symbol.asyncIterator]() {
                return this;
            },
            byPage: (settings) => {
                if (settings === null || settings === void 0 ? void 0 : settings.maxPageSize) {
                    throw new Error("maxPageSize is not supported by this operation.");
                }
                return this.listByReplicationMigrationItemsPagingPage(resourceName, resourceGroupName, fabricName, protectionContainerName, migrationItemName, options, settings);
            }
        };
    }
    listByReplicationMigrationItemsPagingPage(resourceName, resourceGroupName, fabricName, protectionContainerName, migrationItemName, options, settings) {
        return __asyncGenerator(this, arguments, function* listByReplicationMigrationItemsPagingPage_1() {
            let result;
            let continuationToken = settings === null || settings === void 0 ? void 0 : settings.continuationToken;
            if (!continuationToken) {
                result = yield __await(this._listByReplicationMigrationItems(resourceName, resourceGroupName, fabricName, protectionContainerName, migrationItemName, options));
                let page = result.value || [];
                continuationToken = result.nextLink;
                setContinuationToken(page, continuationToken);
                yield yield __await(page);
            }
            while (continuationToken) {
                result = yield __await(this._listByReplicationMigrationItemsNext(resourceName, resourceGroupName, fabricName, protectionContainerName, migrationItemName, continuationToken, options));
                continuationToken = result.nextLink;
                let page = result.value || [];
                setContinuationToken(page, continuationToken);
                yield yield __await(page);
            }
        });
    }
    listByReplicationMigrationItemsPagingAll(resourceName, resourceGroupName, fabricName, protectionContainerName, migrationItemName, options) {
        return __asyncGenerator(this, arguments, function* listByReplicationMigrationItemsPagingAll_1() {
            var _a, e_1, _b, _c;
            try {
                for (var _d = true, _e = __asyncValues(this.listByReplicationMigrationItemsPagingPage(resourceName, resourceGroupName, fabricName, protectionContainerName, migrationItemName, options)), _f; _f = yield __await(_e.next()), _a = _f.done, !_a;) {
                    _c = _f.value;
                    _d = false;
                    try {
                        const page = _c;
                        yield __await(yield* __asyncDelegator(__asyncValues(page)));
                    }
                    finally {
                        _d = true;
                    }
                }
            }
            catch (e_1_1) { e_1 = { error: e_1_1 }; }
            finally {
                try {
                    if (!_d && !_a && (_b = _e.return)) yield __await(_b.call(_e));
                }
                finally { if (e_1) throw e_1.error; }
            }
        });
    }
    /**
     * Gets the recovery points for a migration item.
     * @param resourceName The name of the recovery services vault.
     * @param resourceGroupName The name of the resource group where the recovery services vault is
     *                          present.
     * @param fabricName Fabric unique name.
     * @param protectionContainerName Protection container name.
     * @param migrationItemName Migration item name.
     * @param options The options parameters.
     */
    _listByReplicationMigrationItems(resourceName, resourceGroupName, fabricName, protectionContainerName, migrationItemName, options) {
        return this.client.sendOperationRequest({
            resourceName,
            resourceGroupName,
            fabricName,
            protectionContainerName,
            migrationItemName,
            options
        }, listByReplicationMigrationItemsOperationSpec);
    }
    /**
     * Gets a recovery point for a migration item.
     * @param resourceName The name of the recovery services vault.
     * @param resourceGroupName The name of the resource group where the recovery services vault is
     *                          present.
     * @param fabricName Fabric unique name.
     * @param protectionContainerName Protection container name.
     * @param migrationItemName Migration item name.
     * @param migrationRecoveryPointName The migration recovery point name.
     * @param options The options parameters.
     */
    get(resourceName, resourceGroupName, fabricName, protectionContainerName, migrationItemName, migrationRecoveryPointName, options) {
        return this.client.sendOperationRequest({
            resourceName,
            resourceGroupName,
            fabricName,
            protectionContainerName,
            migrationItemName,
            migrationRecoveryPointName,
            options
        }, getOperationSpec);
    }
    /**
     * ListByReplicationMigrationItemsNext
     * @param resourceName The name of the recovery services vault.
     * @param resourceGroupName The name of the resource group where the recovery services vault is
     *                          present.
     * @param fabricName Fabric unique name.
     * @param protectionContainerName Protection container name.
     * @param migrationItemName Migration item name.
     * @param nextLink The nextLink from the previous successful call to the
     *                 ListByReplicationMigrationItems method.
     * @param options The options parameters.
     */
    _listByReplicationMigrationItemsNext(resourceName, resourceGroupName, fabricName, protectionContainerName, migrationItemName, nextLink, options) {
        return this.client.sendOperationRequest({
            resourceName,
            resourceGroupName,
            fabricName,
            protectionContainerName,
            migrationItemName,
            nextLink,
            options
        }, listByReplicationMigrationItemsNextOperationSpec);
    }
}
// Operation Specifications
const serializer = coreClient.createSerializer(Mappers, /* isXml */ false);
const listByReplicationMigrationItemsOperationSpec = {
    path: "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.RecoveryServices/vaults/{resourceName}/replicationFabrics/{fabricName}/replicationProtectionContainers/{protectionContainerName}/replicationMigrationItems/{migrationItemName}/migrationRecoveryPoints",
    httpMethod: "GET",
    responses: {
        200: {
            bodyMapper: Mappers.MigrationRecoveryPointCollection
        }
    },
    queryParameters: [Parameters.apiVersion],
    urlParameters: [
        Parameters.$host,
        Parameters.resourceGroupName,
        Parameters.subscriptionId,
        Parameters.resourceName,
        Parameters.fabricName,
        Parameters.protectionContainerName,
        Parameters.migrationItemName
    ],
    headerParameters: [Parameters.accept],
    serializer
};
const getOperationSpec = {
    path: "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.RecoveryServices/vaults/{resourceName}/replicationFabrics/{fabricName}/replicationProtectionContainers/{protectionContainerName}/replicationMigrationItems/{migrationItemName}/migrationRecoveryPoints/{migrationRecoveryPointName}",
    httpMethod: "GET",
    responses: {
        200: {
            bodyMapper: Mappers.MigrationRecoveryPoint
        }
    },
    queryParameters: [Parameters.apiVersion],
    urlParameters: [
        Parameters.$host,
        Parameters.resourceGroupName,
        Parameters.subscriptionId,
        Parameters.resourceName,
        Parameters.fabricName,
        Parameters.protectionContainerName,
        Parameters.migrationItemName,
        Parameters.migrationRecoveryPointName
    ],
    headerParameters: [Parameters.accept],
    serializer
};
const listByReplicationMigrationItemsNextOperationSpec = {
    path: "{nextLink}",
    httpMethod: "GET",
    responses: {
        200: {
            bodyMapper: Mappers.MigrationRecoveryPointCollection
        }
    },
    urlParameters: [
        Parameters.$host,
        Parameters.resourceGroupName,
        Parameters.subscriptionId,
        Parameters.nextLink,
        Parameters.resourceName,
        Parameters.fabricName,
        Parameters.protectionContainerName,
        Parameters.migrationItemName
    ],
    headerParameters: [Parameters.accept],
    serializer
};
