// @azure/arm-recoveryservices-siterecovery/dist-esm/src/operations/replicationEligibilityResultsOperations.js
/*
 * Copyright (c) Microsoft Corporation.
 * Licensed under the MIT License.
 *
 * Code generated by Microsoft (R) AutoRest Code Generator.
 * Changes may cause incorrect behavior and will be lost if the code is regenerated.
 */
import * as coreClient from "@azure/core-client";
import * as Mappers from "../models/mappers";
import * as Parameters from "../models/parameters";
/** Class containing ReplicationEligibilityResultsOperations operations. */
export class ReplicationEligibilityResultsOperationsImpl {
    /**
     * Initialize a new instance of the class ReplicationEligibilityResultsOperations class.
     * @param client Reference to the service client
     */
    constructor(client) {
        this.client = client;
    }
    /**
     * Validates whether a given VM can be protected or not in which case returns list of errors.
     * @param resourceGroupName The name of the resource group where the recovery services vault is
     *                          present.
     * @param virtualMachineName Virtual Machine name.
     * @param options The options parameters.
     */
    list(resourceGroupName, virtualMachineName, options) {
        return this.client.sendOperationRequest({ resourceGroupName, virtualMachineName, options }, listOperationSpec);
    }
    /**
     * Validates whether a given VM can be protected or not in which case returns list of errors.
     * @param resourceGroupName The name of the resource group where the recovery services vault is
     *                          present.
     * @param virtualMachineName Virtual Machine name.
     * @param options The options parameters.
     */
    get(resourceGroupName, virtualMachineName, options) {
        return this.client.sendOperationRequest({ resourceGroupName, virtualMachineName, options }, getOperationSpec);
    }
}
// Operation Specifications
const serializer = coreClient.createSerializer(Mappers, /* isXml */ false);
const listOperationSpec = {
    path: "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Compute/virtualMachines/{virtualMachineName}/providers/Microsoft.RecoveryServices/replicationEligibilityResults",
    httpMethod: "GET",
    responses: {
        200: {
            bodyMapper: Mappers.ReplicationEligibilityResultsCollection
        }
    },
    queryParameters: [Parameters.apiVersion],
    urlParameters: [
        Parameters.$host,
        Parameters.resourceGroupName,
        Parameters.subscriptionId,
        Parameters.virtualMachineName
    ],
    headerParameters: [Parameters.accept],
    serializer
};
const getOperationSpec = {
    path: "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Compute/virtualMachines/{virtualMachineName}/providers/Microsoft.RecoveryServices/replicationEligibilityResults/default",
    httpMethod: "GET",
    responses: {
        200: {
            bodyMapper: Mappers.ReplicationEligibilityResults
        }
    },
    queryParameters: [Parameters.apiVersion],
    urlParameters: [
        Parameters.$host,
        Parameters.resourceGroupName,
        Parameters.subscriptionId,
        Parameters.virtualMachineName
    ],
    headerParameters: [Parameters.accept],
    serializer
};
