// @azure/arm-recoveryservices-siterecovery/dist-esm/src/operations/replicationLogicalNetworks.js
/*
 * Copyright (c) Microsoft Corporation.
 * Licensed under the MIT License.
 *
 * Code generated by Microsoft (R) AutoRest Code Generator.
 * Changes may cause incorrect behavior and will be lost if the code is regenerated.
 */
import { __asyncDelegator, __asyncGenerator, __asyncValues, __await } from "tslib";
import { setContinuationToken } from "../pagingHelper";
import * as coreClient from "@azure/core-client";
import * as Mappers from "../models/mappers";
import * as Parameters from "../models/parameters";
/// <reference lib="esnext.asynciterable" />
/** Class containing ReplicationLogicalNetworks operations. */
export class ReplicationLogicalNetworksImpl {
    /**
     * Initialize a new instance of the class ReplicationLogicalNetworks class.
     * @param client Reference to the service client
     */
    constructor(client) {
        this.client = client;
    }
    /**
     * Lists all the logical networks of the Azure Site Recovery fabric.
     * @param resourceName The name of the recovery services vault.
     * @param resourceGroupName The name of the resource group where the recovery services vault is
     *                          present.
     * @param fabricName Server Id.
     * @param options The options parameters.
     */
    listByReplicationFabrics(resourceName, resourceGroupName, fabricName, options) {
        const iter = this.listByReplicationFabricsPagingAll(resourceName, resourceGroupName, fabricName, options);
        return {
            next() {
                return iter.next();
            },
            [Symbol.asyncIterator]() {
                return this;
            },
            byPage: (settings) => {
                if (settings === null || settings === void 0 ? void 0 : settings.maxPageSize) {
                    throw new Error("maxPageSize is not supported by this operation.");
                }
                return this.listByReplicationFabricsPagingPage(resourceName, resourceGroupName, fabricName, options, settings);
            }
        };
    }
    listByReplicationFabricsPagingPage(resourceName, resourceGroupName, fabricName, options, settings) {
        return __asyncGenerator(this, arguments, function* listByReplicationFabricsPagingPage_1() {
            let result;
            let continuationToken = settings === null || settings === void 0 ? void 0 : settings.continuationToken;
            if (!continuationToken) {
                result = yield __await(this._listByReplicationFabrics(resourceName, resourceGroupName, fabricName, options));
                let page = result.value || [];
                continuationToken = result.nextLink;
                setContinuationToken(page, continuationToken);
                yield yield __await(page);
            }
            while (continuationToken) {
                result = yield __await(this._listByReplicationFabricsNext(resourceName, resourceGroupName, fabricName, continuationToken, options));
                continuationToken = result.nextLink;
                let page = result.value || [];
                setContinuationToken(page, continuationToken);
                yield yield __await(page);
            }
        });
    }
    listByReplicationFabricsPagingAll(resourceName, resourceGroupName, fabricName, options) {
        return __asyncGenerator(this, arguments, function* listByReplicationFabricsPagingAll_1() {
            var _a, e_1, _b, _c;
            try {
                for (var _d = true, _e = __asyncValues(this.listByReplicationFabricsPagingPage(resourceName, resourceGroupName, fabricName, options)), _f; _f = yield __await(_e.next()), _a = _f.done, !_a;) {
                    _c = _f.value;
                    _d = false;
                    try {
                        const page = _c;
                        yield __await(yield* __asyncDelegator(__asyncValues(page)));
                    }
                    finally {
                        _d = true;
                    }
                }
            }
            catch (e_1_1) { e_1 = { error: e_1_1 }; }
            finally {
                try {
                    if (!_d && !_a && (_b = _e.return)) yield __await(_b.call(_e));
                }
                finally { if (e_1) throw e_1.error; }
            }
        });
    }
    /**
     * Lists all the logical networks of the Azure Site Recovery fabric.
     * @param resourceName The name of the recovery services vault.
     * @param resourceGroupName The name of the resource group where the recovery services vault is
     *                          present.
     * @param fabricName Server Id.
     * @param options The options parameters.
     */
    _listByReplicationFabrics(resourceName, resourceGroupName, fabricName, options) {
        return this.client.sendOperationRequest({ resourceName, resourceGroupName, fabricName, options }, listByReplicationFabricsOperationSpec);
    }
    /**
     * Gets the details of a logical network.
     * @param resourceName The name of the recovery services vault.
     * @param resourceGroupName The name of the resource group where the recovery services vault is
     *                          present.
     * @param fabricName Server Id.
     * @param logicalNetworkName Logical network name.
     * @param options The options parameters.
     */
    get(resourceName, resourceGroupName, fabricName, logicalNetworkName, options) {
        return this.client.sendOperationRequest({
            resourceName,
            resourceGroupName,
            fabricName,
            logicalNetworkName,
            options
        }, getOperationSpec);
    }
    /**
     * ListByReplicationFabricsNext
     * @param resourceName The name of the recovery services vault.
     * @param resourceGroupName The name of the resource group where the recovery services vault is
     *                          present.
     * @param fabricName Server Id.
     * @param nextLink The nextLink from the previous successful call to the ListByReplicationFabrics
     *                 method.
     * @param options The options parameters.
     */
    _listByReplicationFabricsNext(resourceName, resourceGroupName, fabricName, nextLink, options) {
        return this.client.sendOperationRequest({ resourceName, resourceGroupName, fabricName, nextLink, options }, listByReplicationFabricsNextOperationSpec);
    }
}
// Operation Specifications
const serializer = coreClient.createSerializer(Mappers, /* isXml */ false);
const listByReplicationFabricsOperationSpec = {
    path: "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.RecoveryServices/vaults/{resourceName}/replicationFabrics/{fabricName}/replicationLogicalNetworks",
    httpMethod: "GET",
    responses: {
        200: {
            bodyMapper: Mappers.LogicalNetworkCollection
        }
    },
    queryParameters: [Parameters.apiVersion],
    urlParameters: [
        Parameters.$host,
        Parameters.resourceGroupName,
        Parameters.subscriptionId,
        Parameters.resourceName,
        Parameters.fabricName
    ],
    headerParameters: [Parameters.accept],
    serializer
};
const getOperationSpec = {
    path: "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.RecoveryServices/vaults/{resourceName}/replicationFabrics/{fabricName}/replicationLogicalNetworks/{logicalNetworkName}",
    httpMethod: "GET",
    responses: {
        200: {
            bodyMapper: Mappers.LogicalNetwork
        }
    },
    queryParameters: [Parameters.apiVersion],
    urlParameters: [
        Parameters.$host,
        Parameters.resourceGroupName,
        Parameters.subscriptionId,
        Parameters.resourceName,
        Parameters.fabricName,
        Parameters.logicalNetworkName
    ],
    headerParameters: [Parameters.accept],
    serializer
};
const listByReplicationFabricsNextOperationSpec = {
    path: "{nextLink}",
    httpMethod: "GET",
    responses: {
        200: {
            bodyMapper: Mappers.LogicalNetworkCollection
        }
    },
    urlParameters: [
        Parameters.$host,
        Parameters.resourceGroupName,
        Parameters.subscriptionId,
        Parameters.nextLink,
        Parameters.resourceName,
        Parameters.fabricName
    ],
    headerParameters: [Parameters.accept],
    serializer
};
