// @azure/arm-purview/dist-esm/src/models/index.js
/*
 * Copyright (c) Microsoft Corporation.
 * Licensed under the MIT License.
 *
 * Code generated by Microsoft (R) AutoRest Code Generator.
 * Changes may cause incorrect behavior and will be lost if the code is regenerated.
 */
/** Known values of {@link Status} that the service accepts. */
export var KnownStatus;
(function (KnownStatus) {
    /** Unknown */
    KnownStatus["Unknown"] = "Unknown";
    /** Pending */
    KnownStatus["Pending"] = "Pending";
    /** Approved */
    KnownStatus["Approved"] = "Approved";
    /** Rejected */
    KnownStatus["Rejected"] = "Rejected";
    /** Disconnected */
    KnownStatus["Disconnected"] = "Disconnected";
})(KnownStatus || (KnownStatus = {}));
/** Known values of {@link ProvisioningState} that the service accepts. */
export var KnownProvisioningState;
(function (KnownProvisioningState) {
    /** Unknown */
    KnownProvisioningState["Unknown"] = "Unknown";
    /** Creating */
    KnownProvisioningState["Creating"] = "Creating";
    /** Moving */
    KnownProvisioningState["Moving"] = "Moving";
    /** Deleting */
    KnownProvisioningState["Deleting"] = "Deleting";
    /** SoftDeleting */
    KnownProvisioningState["SoftDeleting"] = "SoftDeleting";
    /** SoftDeleted */
    KnownProvisioningState["SoftDeleted"] = "SoftDeleted";
    /** Failed */
    KnownProvisioningState["Failed"] = "Failed";
    /** Succeeded */
    KnownProvisioningState["Succeeded"] = "Succeeded";
    /** Canceled */
    KnownProvisioningState["Canceled"] = "Canceled";
})(KnownProvisioningState || (KnownProvisioningState = {}));
/** Known values of {@link PublicNetworkAccess} that the service accepts. */
export var KnownPublicNetworkAccess;
(function (KnownPublicNetworkAccess) {
    /** NotSpecified */
    KnownPublicNetworkAccess["NotSpecified"] = "NotSpecified";
    /** Enabled */
    KnownPublicNetworkAccess["Enabled"] = "Enabled";
    /** Disabled */
    KnownPublicNetworkAccess["Disabled"] = "Disabled";
})(KnownPublicNetworkAccess || (KnownPublicNetworkAccess = {}));
/** Known values of {@link Name} that the service accepts. */
export var KnownName;
(function (KnownName) {
    /** Standard */
    KnownName["Standard"] = "Standard";
})(KnownName || (KnownName = {}));
/** Known values of {@link Type} that the service accepts. */
export var KnownType;
(function (KnownType) {
    /** None */
    KnownType["None"] = "None";
    /** SystemAssigned */
    KnownType["SystemAssigned"] = "SystemAssigned";
    /** UserAssigned */
    KnownType["UserAssigned"] = "UserAssigned";
})(KnownType || (KnownType = {}));
/** Known values of {@link CreatedByType} that the service accepts. */
export var KnownCreatedByType;
(function (KnownCreatedByType) {
    /** User */
    KnownCreatedByType["User"] = "User";
    /** Application */
    KnownCreatedByType["Application"] = "Application";
    /** ManagedIdentity */
    KnownCreatedByType["ManagedIdentity"] = "ManagedIdentity";
    /** Key */
    KnownCreatedByType["Key"] = "Key";
})(KnownCreatedByType || (KnownCreatedByType = {}));
/** Known values of {@link LastModifiedByType} that the service accepts. */
export var KnownLastModifiedByType;
(function (KnownLastModifiedByType) {
    /** User */
    KnownLastModifiedByType["User"] = "User";
    /** Application */
    KnownLastModifiedByType["Application"] = "Application";
    /** ManagedIdentity */
    KnownLastModifiedByType["ManagedIdentity"] = "ManagedIdentity";
    /** Key */
    KnownLastModifiedByType["Key"] = "Key";
})(KnownLastModifiedByType || (KnownLastModifiedByType = {}));
/** Known values of {@link ScopeType} that the service accepts. */
export var KnownScopeType;
(function (KnownScopeType) {
    /** Tenant */
    KnownScopeType["Tenant"] = "Tenant";
    /** Subscription */
    KnownScopeType["Subscription"] = "Subscription";
})(KnownScopeType || (KnownScopeType = {}));
/** Known values of {@link Reason} that the service accepts. */
export var KnownReason;
(function (KnownReason) {
    /** Invalid */
    KnownReason["Invalid"] = "Invalid";
    /** AlreadyExists */
    KnownReason["AlreadyExists"] = "AlreadyExists";
})(KnownReason || (KnownReason = {}));
