// @aws-sdk/client-marketplace-metering/dist-cjs/index.js
"use strict";
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __name = (target, value) => __defProp(target, "name", { value, configurable: true });
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// src/index.ts
var src_exports = {};
__export(src_exports, {
  BatchMeterUsageCommand: () => BatchMeterUsageCommand,
  CustomerNotEntitledException: () => CustomerNotEntitledException,
  DisabledApiException: () => DisabledApiException,
  DuplicateRequestException: () => DuplicateRequestException,
  ExpiredTokenException: () => ExpiredTokenException,
  InternalServiceErrorException: () => InternalServiceErrorException,
  InvalidCustomerIdentifierException: () => InvalidCustomerIdentifierException,
  InvalidEndpointRegionException: () => InvalidEndpointRegionException,
  InvalidProductCodeException: () => InvalidProductCodeException,
  InvalidPublicKeyVersionException: () => InvalidPublicKeyVersionException,
  InvalidRegionException: () => InvalidRegionException,
  InvalidTagException: () => InvalidTagException,
  InvalidTokenException: () => InvalidTokenException,
  InvalidUsageAllocationsException: () => InvalidUsageAllocationsException,
  InvalidUsageDimensionException: () => InvalidUsageDimensionException,
  MarketplaceMetering: () => MarketplaceMetering,
  MarketplaceMeteringClient: () => MarketplaceMeteringClient,
  MarketplaceMeteringServiceException: () => MarketplaceMeteringServiceException,
  MeterUsageCommand: () => MeterUsageCommand,
  PlatformNotSupportedException: () => PlatformNotSupportedException,
  RegisterUsageCommand: () => RegisterUsageCommand,
  ResolveCustomerCommand: () => ResolveCustomerCommand,
  ThrottlingException: () => ThrottlingException,
  TimestampOutOfBoundsException: () => TimestampOutOfBoundsException,
  UsageRecordResultStatus: () => UsageRecordResultStatus,
  __Client: () => import_smithy_client.Client
});
module.exports = __toCommonJS(src_exports);

// src/MarketplaceMeteringClient.ts
var import_middleware_host_header = require("@aws-sdk/middleware-host-header");
var import_middleware_logger = require("@aws-sdk/middleware-logger");
var import_middleware_recursion_detection = require("@aws-sdk/middleware-recursion-detection");
var import_middleware_user_agent = require("@aws-sdk/middleware-user-agent");
var import_config_resolver = require("@smithy/config-resolver");
var import_core = require("@smithy/core");
var import_middleware_content_length = require("@smithy/middleware-content-length");
var import_middleware_endpoint = require("@smithy/middleware-endpoint");
var import_middleware_retry = require("@smithy/middleware-retry");

var import_httpAuthSchemeProvider = require("./auth/httpAuthSchemeProvider");

// src/endpoint/EndpointParameters.ts
var resolveClientEndpointParameters = /* @__PURE__ */ __name((options) => {
  return {
    ...options,
    useDualstackEndpoint: options.useDualstackEndpoint ?? false,
    useFipsEndpoint: options.useFipsEndpoint ?? false,
    defaultSigningName: "aws-marketplace"
  };
}, "resolveClientEndpointParameters");
var commonParams = {
  UseFIPS: { type: "builtInParams", name: "useFipsEndpoint" },
  Endpoint: { type: "builtInParams", name: "endpoint" },
  Region: { type: "builtInParams", name: "region" },
  UseDualStack: { type: "builtInParams", name: "useDualstackEndpoint" }
};

// src/MarketplaceMeteringClient.ts
var import_runtimeConfig = require("././runtimeConfig");

// src/runtimeExtensions.ts
var import_region_config_resolver = require("@aws-sdk/region-config-resolver");
var import_protocol_http = require("@smithy/protocol-http");
var import_smithy_client = require("@smithy/smithy-client");

// src/auth/httpAuthExtensionConfiguration.ts
var getHttpAuthExtensionConfiguration = /* @__PURE__ */ __name((runtimeConfig) => {
  const _httpAuthSchemes = runtimeConfig.httpAuthSchemes;
  let _httpAuthSchemeProvider = runtimeConfig.httpAuthSchemeProvider;
  let _credentials = runtimeConfig.credentials;
  return {
    setHttpAuthScheme(httpAuthScheme) {
      const index = _httpAuthSchemes.findIndex((scheme) => scheme.schemeId === httpAuthScheme.schemeId);
      if (index === -1) {
        _httpAuthSchemes.push(httpAuthScheme);
      } else {
        _httpAuthSchemes.splice(index, 1, httpAuthScheme);
      }
    },
    httpAuthSchemes() {
      return _httpAuthSchemes;
    },
    setHttpAuthSchemeProvider(httpAuthSchemeProvider) {
      _httpAuthSchemeProvider = httpAuthSchemeProvider;
    },
    httpAuthSchemeProvider() {
      return _httpAuthSchemeProvider;
    },
    setCredentials(credentials) {
      _credentials = credentials;
    },
    credentials() {
      return _credentials;
    }
  };
}, "getHttpAuthExtensionConfiguration");
var resolveHttpAuthRuntimeConfig = /* @__PURE__ */ __name((config) => {
  return {
    httpAuthSchemes: config.httpAuthSchemes(),
    httpAuthSchemeProvider: config.httpAuthSchemeProvider(),
    credentials: config.credentials()
  };
}, "resolveHttpAuthRuntimeConfig");

// src/runtimeExtensions.ts
var asPartial = /* @__PURE__ */ __name((t) => t, "asPartial");
var resolveRuntimeExtensions = /* @__PURE__ */ __name((runtimeConfig, extensions) => {
  const extensionConfiguration = {
    ...asPartial((0, import_region_config_resolver.getAwsRegionExtensionConfiguration)(runtimeConfig)),
    ...asPartial((0, import_smithy_client.getDefaultExtensionConfiguration)(runtimeConfig)),
    ...asPartial((0, import_protocol_http.getHttpHandlerExtensionConfiguration)(runtimeConfig)),
    ...asPartial(getHttpAuthExtensionConfiguration(runtimeConfig))
  };
  extensions.forEach((extension) => extension.configure(extensionConfiguration));
  return {
    ...runtimeConfig,
    ...(0, import_region_config_resolver.resolveAwsRegionExtensionConfiguration)(extensionConfiguration),
    ...(0, import_smithy_client.resolveDefaultRuntimeConfig)(extensionConfiguration),
    ...(0, import_protocol_http.resolveHttpHandlerRuntimeConfig)(extensionConfiguration),
    ...resolveHttpAuthRuntimeConfig(extensionConfiguration)
  };
}, "resolveRuntimeExtensions");

// src/MarketplaceMeteringClient.ts
var _MarketplaceMeteringClient = class _MarketplaceMeteringClient extends import_smithy_client.Client {
  constructor(...[configuration]) {
    const _config_0 = (0, import_runtimeConfig.getRuntimeConfig)(configuration || {});
    const _config_1 = resolveClientEndpointParameters(_config_0);
    const _config_2 = (0, import_config_resolver.resolveRegionConfig)(_config_1);
    const _config_3 = (0, import_middleware_endpoint.resolveEndpointConfig)(_config_2);
    const _config_4 = (0, import_middleware_retry.resolveRetryConfig)(_config_3);
    const _config_5 = (0, import_middleware_host_header.resolveHostHeaderConfig)(_config_4);
    const _config_6 = (0, import_middleware_user_agent.resolveUserAgentConfig)(_config_5);
    const _config_7 = (0, import_httpAuthSchemeProvider.resolveHttpAuthSchemeConfig)(_config_6);
    const _config_8 = resolveRuntimeExtensions(_config_7, (configuration == null ? void 0 : configuration.extensions) || []);
    super(_config_8);
    this.config = _config_8;
    this.middlewareStack.use((0, import_middleware_retry.getRetryPlugin)(this.config));
    this.middlewareStack.use((0, import_middleware_content_length.getContentLengthPlugin)(this.config));
    this.middlewareStack.use((0, import_middleware_host_header.getHostHeaderPlugin)(this.config));
    this.middlewareStack.use((0, import_middleware_logger.getLoggerPlugin)(this.config));
    this.middlewareStack.use((0, import_middleware_recursion_detection.getRecursionDetectionPlugin)(this.config));
    this.middlewareStack.use((0, import_middleware_user_agent.getUserAgentPlugin)(this.config));
    this.middlewareStack.use(
      (0, import_core.getHttpAuthSchemeEndpointRuleSetPlugin)(this.config, {
        httpAuthSchemeParametersProvider: this.getDefaultHttpAuthSchemeParametersProvider(),
        identityProviderConfigProvider: this.getIdentityProviderConfigProvider()
      })
    );
    this.middlewareStack.use((0, import_core.getHttpSigningPlugin)(this.config));
  }
  /**
   * Destroy underlying resources, like sockets. It's usually not necessary to do this.
   * However in Node.js, it's best to explicitly shut down the client's agent when it is no longer needed.
   * Otherwise, sockets might stay open for quite a long time before the server terminates them.
   */
  destroy() {
    super.destroy();
  }
  getDefaultHttpAuthSchemeParametersProvider() {
    return import_httpAuthSchemeProvider.defaultMarketplaceMeteringHttpAuthSchemeParametersProvider;
  }
  getIdentityProviderConfigProvider() {
    return async (config) => new import_core.DefaultIdentityProviderConfig({
      "aws.auth#sigv4": config.credentials
    });
  }
};
__name(_MarketplaceMeteringClient, "MarketplaceMeteringClient");
var MarketplaceMeteringClient = _MarketplaceMeteringClient;

// src/MarketplaceMetering.ts


// src/commands/BatchMeterUsageCommand.ts

var import_middleware_serde = require("@smithy/middleware-serde");

var import_types = require("@smithy/types");

// src/protocols/Aws_json1_1.ts



// src/models/MarketplaceMeteringServiceException.ts

var _MarketplaceMeteringServiceException = class _MarketplaceMeteringServiceException extends import_smithy_client.ServiceException {
  /**
   * @internal
   */
  constructor(options) {
    super(options);
    Object.setPrototypeOf(this, _MarketplaceMeteringServiceException.prototype);
  }
};
__name(_MarketplaceMeteringServiceException, "MarketplaceMeteringServiceException");
var MarketplaceMeteringServiceException = _MarketplaceMeteringServiceException;

// src/models/models_0.ts
var UsageRecordResultStatus = {
  CUSTOMER_NOT_SUBSCRIBED: "CustomerNotSubscribed",
  DUPLICATE_RECORD: "DuplicateRecord",
  SUCCESS: "Success"
};
var _DisabledApiException = class _DisabledApiException extends MarketplaceMeteringServiceException {
  /**
   * @internal
   */
  constructor(opts) {
    super({
      name: "DisabledApiException",
      $fault: "client",
      ...opts
    });
    this.name = "DisabledApiException";
    this.$fault = "client";
    Object.setPrototypeOf(this, _DisabledApiException.prototype);
  }
};
__name(_DisabledApiException, "DisabledApiException");
var DisabledApiException = _DisabledApiException;
var _InternalServiceErrorException = class _InternalServiceErrorException extends MarketplaceMeteringServiceException {
  /**
   * @internal
   */
  constructor(opts) {
    super({
      name: "InternalServiceErrorException",
      $fault: "server",
      ...opts
    });
    this.name = "InternalServiceErrorException";
    this.$fault = "server";
    Object.setPrototypeOf(this, _InternalServiceErrorException.prototype);
  }
};
__name(_InternalServiceErrorException, "InternalServiceErrorException");
var InternalServiceErrorException = _InternalServiceErrorException;
var _InvalidCustomerIdentifierException = class _InvalidCustomerIdentifierException extends MarketplaceMeteringServiceException {
  /**
   * @internal
   */
  constructor(opts) {
    super({
      name: "InvalidCustomerIdentifierException",
      $fault: "client",
      ...opts
    });
    this.name = "InvalidCustomerIdentifierException";
    this.$fault = "client";
    Object.setPrototypeOf(this, _InvalidCustomerIdentifierException.prototype);
  }
};
__name(_InvalidCustomerIdentifierException, "InvalidCustomerIdentifierException");
var InvalidCustomerIdentifierException = _InvalidCustomerIdentifierException;
var _InvalidProductCodeException = class _InvalidProductCodeException extends MarketplaceMeteringServiceException {
  /**
   * @internal
   */
  constructor(opts) {
    super({
      name: "InvalidProductCodeException",
      $fault: "client",
      ...opts
    });
    this.name = "InvalidProductCodeException";
    this.$fault = "client";
    Object.setPrototypeOf(this, _InvalidProductCodeException.prototype);
  }
};
__name(_InvalidProductCodeException, "InvalidProductCodeException");
var InvalidProductCodeException = _InvalidProductCodeException;
var _InvalidTagException = class _InvalidTagException extends MarketplaceMeteringServiceException {
  /**
   * @internal
   */
  constructor(opts) {
    super({
      name: "InvalidTagException",
      $fault: "client",
      ...opts
    });
    this.name = "InvalidTagException";
    this.$fault = "client";
    Object.setPrototypeOf(this, _InvalidTagException.prototype);
  }
};
__name(_InvalidTagException, "InvalidTagException");
var InvalidTagException = _InvalidTagException;
var _InvalidUsageAllocationsException = class _InvalidUsageAllocationsException extends MarketplaceMeteringServiceException {
  /**
   * @internal
   */
  constructor(opts) {
    super({
      name: "InvalidUsageAllocationsException",
      $fault: "client",
      ...opts
    });
    this.name = "InvalidUsageAllocationsException";
    this.$fault = "client";
    Object.setPrototypeOf(this, _InvalidUsageAllocationsException.prototype);
  }
};
__name(_InvalidUsageAllocationsException, "InvalidUsageAllocationsException");
var InvalidUsageAllocationsException = _InvalidUsageAllocationsException;
var _InvalidUsageDimensionException = class _InvalidUsageDimensionException extends MarketplaceMeteringServiceException {
  /**
   * @internal
   */
  constructor(opts) {
    super({
      name: "InvalidUsageDimensionException",
      $fault: "client",
      ...opts
    });
    this.name = "InvalidUsageDimensionException";
    this.$fault = "client";
    Object.setPrototypeOf(this, _InvalidUsageDimensionException.prototype);
  }
};
__name(_InvalidUsageDimensionException, "InvalidUsageDimensionException");
var InvalidUsageDimensionException = _InvalidUsageDimensionException;
var _ThrottlingException = class _ThrottlingException extends MarketplaceMeteringServiceException {
  /**
   * @internal
   */
  constructor(opts) {
    super({
      name: "ThrottlingException",
      $fault: "client",
      ...opts
    });
    this.name = "ThrottlingException";
    this.$fault = "client";
    Object.setPrototypeOf(this, _ThrottlingException.prototype);
  }
};
__name(_ThrottlingException, "ThrottlingException");
var ThrottlingException = _ThrottlingException;
var _TimestampOutOfBoundsException = class _TimestampOutOfBoundsException extends MarketplaceMeteringServiceException {
  /**
   * @internal
   */
  constructor(opts) {
    super({
      name: "TimestampOutOfBoundsException",
      $fault: "client",
      ...opts
    });
    this.name = "TimestampOutOfBoundsException";
    this.$fault = "client";
    Object.setPrototypeOf(this, _TimestampOutOfBoundsException.prototype);
  }
};
__name(_TimestampOutOfBoundsException, "TimestampOutOfBoundsException");
var TimestampOutOfBoundsException = _TimestampOutOfBoundsException;
var _CustomerNotEntitledException = class _CustomerNotEntitledException extends MarketplaceMeteringServiceException {
  /**
   * @internal
   */
  constructor(opts) {
    super({
      name: "CustomerNotEntitledException",
      $fault: "client",
      ...opts
    });
    this.name = "CustomerNotEntitledException";
    this.$fault = "client";
    Object.setPrototypeOf(this, _CustomerNotEntitledException.prototype);
  }
};
__name(_CustomerNotEntitledException, "CustomerNotEntitledException");
var CustomerNotEntitledException = _CustomerNotEntitledException;
var _DuplicateRequestException = class _DuplicateRequestException extends MarketplaceMeteringServiceException {
  /**
   * @internal
   */
  constructor(opts) {
    super({
      name: "DuplicateRequestException",
      $fault: "client",
      ...opts
    });
    this.name = "DuplicateRequestException";
    this.$fault = "client";
    Object.setPrototypeOf(this, _DuplicateRequestException.prototype);
  }
};
__name(_DuplicateRequestException, "DuplicateRequestException");
var DuplicateRequestException = _DuplicateRequestException;
var _InvalidEndpointRegionException = class _InvalidEndpointRegionException extends MarketplaceMeteringServiceException {
  /**
   * @internal
   */
  constructor(opts) {
    super({
      name: "InvalidEndpointRegionException",
      $fault: "client",
      ...opts
    });
    this.name = "InvalidEndpointRegionException";
    this.$fault = "client";
    Object.setPrototypeOf(this, _InvalidEndpointRegionException.prototype);
  }
};
__name(_InvalidEndpointRegionException, "InvalidEndpointRegionException");
var InvalidEndpointRegionException = _InvalidEndpointRegionException;
var _InvalidPublicKeyVersionException = class _InvalidPublicKeyVersionException extends MarketplaceMeteringServiceException {
  /**
   * @internal
   */
  constructor(opts) {
    super({
      name: "InvalidPublicKeyVersionException",
      $fault: "client",
      ...opts
    });
    this.name = "InvalidPublicKeyVersionException";
    this.$fault = "client";
    Object.setPrototypeOf(this, _InvalidPublicKeyVersionException.prototype);
  }
};
__name(_InvalidPublicKeyVersionException, "InvalidPublicKeyVersionException");
var InvalidPublicKeyVersionException = _InvalidPublicKeyVersionException;
var _InvalidRegionException = class _InvalidRegionException extends MarketplaceMeteringServiceException {
  /**
   * @internal
   */
  constructor(opts) {
    super({
      name: "InvalidRegionException",
      $fault: "client",
      ...opts
    });
    this.name = "InvalidRegionException";
    this.$fault = "client";
    Object.setPrototypeOf(this, _InvalidRegionException.prototype);
  }
};
__name(_InvalidRegionException, "InvalidRegionException");
var InvalidRegionException = _InvalidRegionException;
var _PlatformNotSupportedException = class _PlatformNotSupportedException extends MarketplaceMeteringServiceException {
  /**
   * @internal
   */
  constructor(opts) {
    super({
      name: "PlatformNotSupportedException",
      $fault: "client",
      ...opts
    });
    this.name = "PlatformNotSupportedException";
    this.$fault = "client";
    Object.setPrototypeOf(this, _PlatformNotSupportedException.prototype);
  }
};
__name(_PlatformNotSupportedException, "PlatformNotSupportedException");
var PlatformNotSupportedException = _PlatformNotSupportedException;
var _ExpiredTokenException = class _ExpiredTokenException extends MarketplaceMeteringServiceException {
  /**
   * @internal
   */
  constructor(opts) {
    super({
      name: "ExpiredTokenException",
      $fault: "client",
      ...opts
    });
    this.name = "ExpiredTokenException";
    this.$fault = "client";
    Object.setPrototypeOf(this, _ExpiredTokenException.prototype);
  }
};
__name(_ExpiredTokenException, "ExpiredTokenException");
var ExpiredTokenException = _ExpiredTokenException;
var _InvalidTokenException = class _InvalidTokenException extends MarketplaceMeteringServiceException {
  /**
   * @internal
   */
  constructor(opts) {
    super({
      name: "InvalidTokenException",
      $fault: "client",
      ...opts
    });
    this.name = "InvalidTokenException";
    this.$fault = "client";
    Object.setPrototypeOf(this, _InvalidTokenException.prototype);
  }
};
__name(_InvalidTokenException, "InvalidTokenException");
var InvalidTokenException = _InvalidTokenException;

// src/protocols/Aws_json1_1.ts
var se_BatchMeterUsageCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("BatchMeterUsage");
  let body;
  body = JSON.stringify(se_BatchMeterUsageRequest(input, context));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_BatchMeterUsageCommand");
var se_MeterUsageCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("MeterUsage");
  let body;
  body = JSON.stringify(se_MeterUsageRequest(input, context));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_MeterUsageCommand");
var se_RegisterUsageCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("RegisterUsage");
  let body;
  body = JSON.stringify((0, import_smithy_client._json)(input));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_RegisterUsageCommand");
var se_ResolveCustomerCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("ResolveCustomer");
  let body;
  body = JSON.stringify((0, import_smithy_client._json)(input));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_ResolveCustomerCommand");
var de_BatchMeterUsageCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const data = await parseBody(output.body, context);
  let contents = {};
  contents = de_BatchMeterUsageResult(data, context);
  const response = {
    $metadata: deserializeMetadata(output),
    ...contents
  };
  return response;
}, "de_BatchMeterUsageCommand");
var de_MeterUsageCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const data = await parseBody(output.body, context);
  let contents = {};
  contents = (0, import_smithy_client._json)(data);
  const response = {
    $metadata: deserializeMetadata(output),
    ...contents
  };
  return response;
}, "de_MeterUsageCommand");
var de_RegisterUsageCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const data = await parseBody(output.body, context);
  let contents = {};
  contents = de_RegisterUsageResult(data, context);
  const response = {
    $metadata: deserializeMetadata(output),
    ...contents
  };
  return response;
}, "de_RegisterUsageCommand");
var de_ResolveCustomerCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const data = await parseBody(output.body, context);
  let contents = {};
  contents = (0, import_smithy_client._json)(data);
  const response = {
    $metadata: deserializeMetadata(output),
    ...contents
  };
  return response;
}, "de_ResolveCustomerCommand");
var de_CommandError = /* @__PURE__ */ __name(async (output, context) => {
  const parsedOutput = {
    ...output,
    body: await parseErrorBody(output.body, context)
  };
  const errorCode = loadRestJsonErrorCode(output, parsedOutput.body);
  switch (errorCode) {
    case "DisabledApiException":
    case "com.amazonaws.marketplacemetering#DisabledApiException":
      throw await de_DisabledApiExceptionRes(parsedOutput, context);
    case "InternalServiceErrorException":
    case "com.amazonaws.marketplacemetering#InternalServiceErrorException":
      throw await de_InternalServiceErrorExceptionRes(parsedOutput, context);
    case "InvalidCustomerIdentifierException":
    case "com.amazonaws.marketplacemetering#InvalidCustomerIdentifierException":
      throw await de_InvalidCustomerIdentifierExceptionRes(parsedOutput, context);
    case "InvalidProductCodeException":
    case "com.amazonaws.marketplacemetering#InvalidProductCodeException":
      throw await de_InvalidProductCodeExceptionRes(parsedOutput, context);
    case "InvalidTagException":
    case "com.amazonaws.marketplacemetering#InvalidTagException":
      throw await de_InvalidTagExceptionRes(parsedOutput, context);
    case "InvalidUsageAllocationsException":
    case "com.amazonaws.marketplacemetering#InvalidUsageAllocationsException":
      throw await de_InvalidUsageAllocationsExceptionRes(parsedOutput, context);
    case "InvalidUsageDimensionException":
    case "com.amazonaws.marketplacemetering#InvalidUsageDimensionException":
      throw await de_InvalidUsageDimensionExceptionRes(parsedOutput, context);
    case "ThrottlingException":
    case "com.amazonaws.marketplacemetering#ThrottlingException":
      throw await de_ThrottlingExceptionRes(parsedOutput, context);
    case "TimestampOutOfBoundsException":
    case "com.amazonaws.marketplacemetering#TimestampOutOfBoundsException":
      throw await de_TimestampOutOfBoundsExceptionRes(parsedOutput, context);
    case "CustomerNotEntitledException":
    case "com.amazonaws.marketplacemetering#CustomerNotEntitledException":
      throw await de_CustomerNotEntitledExceptionRes(parsedOutput, context);
    case "DuplicateRequestException":
    case "com.amazonaws.marketplacemetering#DuplicateRequestException":
      throw await de_DuplicateRequestExceptionRes(parsedOutput, context);
    case "InvalidEndpointRegionException":
    case "com.amazonaws.marketplacemetering#InvalidEndpointRegionException":
      throw await de_InvalidEndpointRegionExceptionRes(parsedOutput, context);
    case "InvalidPublicKeyVersionException":
    case "com.amazonaws.marketplacemetering#InvalidPublicKeyVersionException":
      throw await de_InvalidPublicKeyVersionExceptionRes(parsedOutput, context);
    case "InvalidRegionException":
    case "com.amazonaws.marketplacemetering#InvalidRegionException":
      throw await de_InvalidRegionExceptionRes(parsedOutput, context);
    case "PlatformNotSupportedException":
    case "com.amazonaws.marketplacemetering#PlatformNotSupportedException":
      throw await de_PlatformNotSupportedExceptionRes(parsedOutput, context);
    case "ExpiredTokenException":
    case "com.amazonaws.marketplacemetering#ExpiredTokenException":
      throw await de_ExpiredTokenExceptionRes(parsedOutput, context);
    case "InvalidTokenException":
    case "com.amazonaws.marketplacemetering#InvalidTokenException":
      throw await de_InvalidTokenExceptionRes(parsedOutput, context);
    default:
      const parsedBody = parsedOutput.body;
      return throwDefaultError({
        output,
        parsedBody,
        errorCode
      });
  }
}, "de_CommandError");
var de_CustomerNotEntitledExceptionRes = /* @__PURE__ */ __name(async (parsedOutput, context) => {
  const body = parsedOutput.body;
  const deserialized = (0, import_smithy_client._json)(body);
  const exception = new CustomerNotEntitledException({
    $metadata: deserializeMetadata(parsedOutput),
    ...deserialized
  });
  return (0, import_smithy_client.decorateServiceException)(exception, body);
}, "de_CustomerNotEntitledExceptionRes");
var de_DisabledApiExceptionRes = /* @__PURE__ */ __name(async (parsedOutput, context) => {
  const body = parsedOutput.body;
  const deserialized = (0, import_smithy_client._json)(body);
  const exception = new DisabledApiException({
    $metadata: deserializeMetadata(parsedOutput),
    ...deserialized
  });
  return (0, import_smithy_client.decorateServiceException)(exception, body);
}, "de_DisabledApiExceptionRes");
var de_DuplicateRequestExceptionRes = /* @__PURE__ */ __name(async (parsedOutput, context) => {
  const body = parsedOutput.body;
  const deserialized = (0, import_smithy_client._json)(body);
  const exception = new DuplicateRequestException({
    $metadata: deserializeMetadata(parsedOutput),
    ...deserialized
  });
  return (0, import_smithy_client.decorateServiceException)(exception, body);
}, "de_DuplicateRequestExceptionRes");
var de_ExpiredTokenExceptionRes = /* @__PURE__ */ __name(async (parsedOutput, context) => {
  const body = parsedOutput.body;
  const deserialized = (0, import_smithy_client._json)(body);
  const exception = new ExpiredTokenException({
    $metadata: deserializeMetadata(parsedOutput),
    ...deserialized
  });
  return (0, import_smithy_client.decorateServiceException)(exception, body);
}, "de_ExpiredTokenExceptionRes");
var de_InternalServiceErrorExceptionRes = /* @__PURE__ */ __name(async (parsedOutput, context) => {
  const body = parsedOutput.body;
  const deserialized = (0, import_smithy_client._json)(body);
  const exception = new InternalServiceErrorException({
    $metadata: deserializeMetadata(parsedOutput),
    ...deserialized
  });
  return (0, import_smithy_client.decorateServiceException)(exception, body);
}, "de_InternalServiceErrorExceptionRes");
var de_InvalidCustomerIdentifierExceptionRes = /* @__PURE__ */ __name(async (parsedOutput, context) => {
  const body = parsedOutput.body;
  const deserialized = (0, import_smithy_client._json)(body);
  const exception = new InvalidCustomerIdentifierException({
    $metadata: deserializeMetadata(parsedOutput),
    ...deserialized
  });
  return (0, import_smithy_client.decorateServiceException)(exception, body);
}, "de_InvalidCustomerIdentifierExceptionRes");
var de_InvalidEndpointRegionExceptionRes = /* @__PURE__ */ __name(async (parsedOutput, context) => {
  const body = parsedOutput.body;
  const deserialized = (0, import_smithy_client._json)(body);
  const exception = new InvalidEndpointRegionException({
    $metadata: deserializeMetadata(parsedOutput),
    ...deserialized
  });
  return (0, import_smithy_client.decorateServiceException)(exception, body);
}, "de_InvalidEndpointRegionExceptionRes");
var de_InvalidProductCodeExceptionRes = /* @__PURE__ */ __name(async (parsedOutput, context) => {
  const body = parsedOutput.body;
  const deserialized = (0, import_smithy_client._json)(body);
  const exception = new InvalidProductCodeException({
    $metadata: deserializeMetadata(parsedOutput),
    ...deserialized
  });
  return (0, import_smithy_client.decorateServiceException)(exception, body);
}, "de_InvalidProductCodeExceptionRes");
var de_InvalidPublicKeyVersionExceptionRes = /* @__PURE__ */ __name(async (parsedOutput, context) => {
  const body = parsedOutput.body;
  const deserialized = (0, import_smithy_client._json)(body);
  const exception = new InvalidPublicKeyVersionException({
    $metadata: deserializeMetadata(parsedOutput),
    ...deserialized
  });
  return (0, import_smithy_client.decorateServiceException)(exception, body);
}, "de_InvalidPublicKeyVersionExceptionRes");
var de_InvalidRegionExceptionRes = /* @__PURE__ */ __name(async (parsedOutput, context) => {
  const body = parsedOutput.body;
  const deserialized = (0, import_smithy_client._json)(body);
  const exception = new InvalidRegionException({
    $metadata: deserializeMetadata(parsedOutput),
    ...deserialized
  });
  return (0, import_smithy_client.decorateServiceException)(exception, body);
}, "de_InvalidRegionExceptionRes");
var de_InvalidTagExceptionRes = /* @__PURE__ */ __name(async (parsedOutput, context) => {
  const body = parsedOutput.body;
  const deserialized = (0, import_smithy_client._json)(body);
  const exception = new InvalidTagException({
    $metadata: deserializeMetadata(parsedOutput),
    ...deserialized
  });
  return (0, import_smithy_client.decorateServiceException)(exception, body);
}, "de_InvalidTagExceptionRes");
var de_InvalidTokenExceptionRes = /* @__PURE__ */ __name(async (parsedOutput, context) => {
  const body = parsedOutput.body;
  const deserialized = (0, import_smithy_client._json)(body);
  const exception = new InvalidTokenException({
    $metadata: deserializeMetadata(parsedOutput),
    ...deserialized
  });
  return (0, import_smithy_client.decorateServiceException)(exception, body);
}, "de_InvalidTokenExceptionRes");
var de_InvalidUsageAllocationsExceptionRes = /* @__PURE__ */ __name(async (parsedOutput, context) => {
  const body = parsedOutput.body;
  const deserialized = (0, import_smithy_client._json)(body);
  const exception = new InvalidUsageAllocationsException({
    $metadata: deserializeMetadata(parsedOutput),
    ...deserialized
  });
  return (0, import_smithy_client.decorateServiceException)(exception, body);
}, "de_InvalidUsageAllocationsExceptionRes");
var de_InvalidUsageDimensionExceptionRes = /* @__PURE__ */ __name(async (parsedOutput, context) => {
  const body = parsedOutput.body;
  const deserialized = (0, import_smithy_client._json)(body);
  const exception = new InvalidUsageDimensionException({
    $metadata: deserializeMetadata(parsedOutput),
    ...deserialized
  });
  return (0, import_smithy_client.decorateServiceException)(exception, body);
}, "de_InvalidUsageDimensionExceptionRes");
var de_PlatformNotSupportedExceptionRes = /* @__PURE__ */ __name(async (parsedOutput, context) => {
  const body = parsedOutput.body;
  const deserialized = (0, import_smithy_client._json)(body);
  const exception = new PlatformNotSupportedException({
    $metadata: deserializeMetadata(parsedOutput),
    ...deserialized
  });
  return (0, import_smithy_client.decorateServiceException)(exception, body);
}, "de_PlatformNotSupportedExceptionRes");
var de_ThrottlingExceptionRes = /* @__PURE__ */ __name(async (parsedOutput, context) => {
  const body = parsedOutput.body;
  const deserialized = (0, import_smithy_client._json)(body);
  const exception = new ThrottlingException({
    $metadata: deserializeMetadata(parsedOutput),
    ...deserialized
  });
  return (0, import_smithy_client.decorateServiceException)(exception, body);
}, "de_ThrottlingExceptionRes");
var de_TimestampOutOfBoundsExceptionRes = /* @__PURE__ */ __name(async (parsedOutput, context) => {
  const body = parsedOutput.body;
  const deserialized = (0, import_smithy_client._json)(body);
  const exception = new TimestampOutOfBoundsException({
    $metadata: deserializeMetadata(parsedOutput),
    ...deserialized
  });
  return (0, import_smithy_client.decorateServiceException)(exception, body);
}, "de_TimestampOutOfBoundsExceptionRes");
var se_BatchMeterUsageRequest = /* @__PURE__ */ __name((input, context) => {
  return (0, import_smithy_client.take)(input, {
    ProductCode: [],
    UsageRecords: (_) => se_UsageRecordList(_, context)
  });
}, "se_BatchMeterUsageRequest");
var se_MeterUsageRequest = /* @__PURE__ */ __name((input, context) => {
  return (0, import_smithy_client.take)(input, {
    DryRun: [],
    ProductCode: [],
    Timestamp: (_) => Math.round(_.getTime() / 1e3),
    UsageAllocations: import_smithy_client._json,
    UsageDimension: [],
    UsageQuantity: []
  });
}, "se_MeterUsageRequest");
var se_UsageRecord = /* @__PURE__ */ __name((input, context) => {
  return (0, import_smithy_client.take)(input, {
    CustomerIdentifier: [],
    Dimension: [],
    Quantity: [],
    Timestamp: (_) => Math.round(_.getTime() / 1e3),
    UsageAllocations: import_smithy_client._json
  });
}, "se_UsageRecord");
var se_UsageRecordList = /* @__PURE__ */ __name((input, context) => {
  return input.filter((e) => e != null).map((entry) => {
    return se_UsageRecord(entry, context);
  });
}, "se_UsageRecordList");
var de_BatchMeterUsageResult = /* @__PURE__ */ __name((output, context) => {
  return (0, import_smithy_client.take)(output, {
    Results: (_) => de_UsageRecordResultList(_, context),
    UnprocessedRecords: (_) => de_UsageRecordList(_, context)
  });
}, "de_BatchMeterUsageResult");
var de_RegisterUsageResult = /* @__PURE__ */ __name((output, context) => {
  return (0, import_smithy_client.take)(output, {
    PublicKeyRotationTimestamp: (_) => (0, import_smithy_client.expectNonNull)((0, import_smithy_client.parseEpochTimestamp)((0, import_smithy_client.expectNumber)(_))),
    Signature: import_smithy_client.expectString
  });
}, "de_RegisterUsageResult");
var de_UsageRecord = /* @__PURE__ */ __name((output, context) => {
  return (0, import_smithy_client.take)(output, {
    CustomerIdentifier: import_smithy_client.expectString,
    Dimension: import_smithy_client.expectString,
    Quantity: import_smithy_client.expectInt32,
    Timestamp: (_) => (0, import_smithy_client.expectNonNull)((0, import_smithy_client.parseEpochTimestamp)((0, import_smithy_client.expectNumber)(_))),
    UsageAllocations: import_smithy_client._json
  });
}, "de_UsageRecord");
var de_UsageRecordList = /* @__PURE__ */ __name((output, context) => {
  const retVal = (output || []).filter((e) => e != null).map((entry) => {
    return de_UsageRecord(entry, context);
  });
  return retVal;
}, "de_UsageRecordList");
var de_UsageRecordResult = /* @__PURE__ */ __name((output, context) => {
  return (0, import_smithy_client.take)(output, {
    MeteringRecordId: import_smithy_client.expectString,
    Status: import_smithy_client.expectString,
    UsageRecord: (_) => de_UsageRecord(_, context)
  });
}, "de_UsageRecordResult");
var de_UsageRecordResultList = /* @__PURE__ */ __name((output, context) => {
  const retVal = (output || []).filter((e) => e != null).map((entry) => {
    return de_UsageRecordResult(entry, context);
  });
  return retVal;
}, "de_UsageRecordResultList");
var deserializeMetadata = /* @__PURE__ */ __name((output) => ({
  httpStatusCode: output.statusCode,
  requestId: output.headers["x-amzn-requestid"] ?? output.headers["x-amzn-request-id"] ?? output.headers["x-amz-request-id"],
  extendedRequestId: output.headers["x-amz-id-2"],
  cfId: output.headers["x-amz-cf-id"]
}), "deserializeMetadata");
var collectBodyString = /* @__PURE__ */ __name((streamBody, context) => (0, import_smithy_client.collectBody)(streamBody, context).then((body) => context.utf8Encoder(body)), "collectBodyString");
var throwDefaultError = (0, import_smithy_client.withBaseException)(MarketplaceMeteringServiceException);
var buildHttpRpcRequest = /* @__PURE__ */ __name(async (context, headers, path, resolvedHostname, body) => {
  const { hostname, protocol = "https", port, path: basePath } = await context.endpoint();
  const contents = {
    protocol,
    hostname,
    port,
    method: "POST",
    path: basePath.endsWith("/") ? basePath.slice(0, -1) + path : basePath + path,
    headers
  };
  if (resolvedHostname !== void 0) {
    contents.hostname = resolvedHostname;
  }
  if (body !== void 0) {
    contents.body = body;
  }
  return new import_protocol_http.HttpRequest(contents);
}, "buildHttpRpcRequest");
function sharedHeaders(operation) {
  return {
    "content-type": "application/x-amz-json-1.1",
    "x-amz-target": `AWSMPMeteringService.${operation}`
  };
}
__name(sharedHeaders, "sharedHeaders");
var parseBody = /* @__PURE__ */ __name((streamBody, context) => collectBodyString(streamBody, context).then((encoded) => {
  if (encoded.length) {
    return JSON.parse(encoded);
  }
  return {};
}), "parseBody");
var parseErrorBody = /* @__PURE__ */ __name(async (errorBody, context) => {
  const value = await parseBody(errorBody, context);
  value.message = value.message ?? value.Message;
  return value;
}, "parseErrorBody");
var loadRestJsonErrorCode = /* @__PURE__ */ __name((output, data) => {
  const findKey = /* @__PURE__ */ __name((object, key) => Object.keys(object).find((k) => k.toLowerCase() === key.toLowerCase()), "findKey");
  const sanitizeErrorCode = /* @__PURE__ */ __name((rawValue) => {
    let cleanValue = rawValue;
    if (typeof cleanValue === "number") {
      cleanValue = cleanValue.toString();
    }
    if (cleanValue.indexOf(",") >= 0) {
      cleanValue = cleanValue.split(",")[0];
    }
    if (cleanValue.indexOf(":") >= 0) {
      cleanValue = cleanValue.split(":")[0];
    }
    if (cleanValue.indexOf("#") >= 0) {
      cleanValue = cleanValue.split("#")[1];
    }
    return cleanValue;
  }, "sanitizeErrorCode");
  const headerKey = findKey(output.headers, "x-amzn-errortype");
  if (headerKey !== void 0) {
    return sanitizeErrorCode(output.headers[headerKey]);
  }
  if (data.code !== void 0) {
    return sanitizeErrorCode(data.code);
  }
  if (data["__type"] !== void 0) {
    return sanitizeErrorCode(data["__type"]);
  }
}, "loadRestJsonErrorCode");

// src/commands/BatchMeterUsageCommand.ts
var _BatchMeterUsageCommand = class _BatchMeterUsageCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AWSMPMeteringService", "BatchMeterUsage", {}).n("MarketplaceMeteringClient", "BatchMeterUsageCommand").f(void 0, void 0).ser(se_BatchMeterUsageCommand).de(de_BatchMeterUsageCommand).build() {
};
__name(_BatchMeterUsageCommand, "BatchMeterUsageCommand");
var BatchMeterUsageCommand = _BatchMeterUsageCommand;

// src/commands/MeterUsageCommand.ts




var _MeterUsageCommand = class _MeterUsageCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AWSMPMeteringService", "MeterUsage", {}).n("MarketplaceMeteringClient", "MeterUsageCommand").f(void 0, void 0).ser(se_MeterUsageCommand).de(de_MeterUsageCommand).build() {
};
__name(_MeterUsageCommand, "MeterUsageCommand");
var MeterUsageCommand = _MeterUsageCommand;

// src/commands/RegisterUsageCommand.ts




var _RegisterUsageCommand = class _RegisterUsageCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AWSMPMeteringService", "RegisterUsage", {}).n("MarketplaceMeteringClient", "RegisterUsageCommand").f(void 0, void 0).ser(se_RegisterUsageCommand).de(de_RegisterUsageCommand).build() {
};
__name(_RegisterUsageCommand, "RegisterUsageCommand");
var RegisterUsageCommand = _RegisterUsageCommand;

// src/commands/ResolveCustomerCommand.ts




var _ResolveCustomerCommand = class _ResolveCustomerCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AWSMPMeteringService", "ResolveCustomer", {}).n("MarketplaceMeteringClient", "ResolveCustomerCommand").f(void 0, void 0).ser(se_ResolveCustomerCommand).de(de_ResolveCustomerCommand).build() {
};
__name(_ResolveCustomerCommand, "ResolveCustomerCommand");
var ResolveCustomerCommand = _ResolveCustomerCommand;

// src/MarketplaceMetering.ts
var commands = {
  BatchMeterUsageCommand,
  MeterUsageCommand,
  RegisterUsageCommand,
  ResolveCustomerCommand
};
var _MarketplaceMetering = class _MarketplaceMetering extends MarketplaceMeteringClient {
};
__name(_MarketplaceMetering, "MarketplaceMetering");
var MarketplaceMetering = _MarketplaceMetering;
(0, import_smithy_client.createAggregatedClient)(commands, MarketplaceMetering);

// src/index.ts
var import_util_endpoints = require("@aws-sdk/util-endpoints");
// Annotate the CommonJS export names for ESM import in node:

0 && (module.exports = {
  MarketplaceMeteringServiceException,
  __Client,
  MarketplaceMeteringClient,
  MarketplaceMetering,
  $Command,
  BatchMeterUsageCommand,
  MeterUsageCommand,
  RegisterUsageCommand,
  ResolveCustomerCommand,
  UsageRecordResultStatus,
  DisabledApiException,
  InternalServiceErrorException,
  InvalidCustomerIdentifierException,
  InvalidProductCodeException,
  InvalidTagException,
  InvalidUsageAllocationsException,
  InvalidUsageDimensionException,
  ThrottlingException,
  TimestampOutOfBoundsException,
  CustomerNotEntitledException,
  DuplicateRequestException,
  InvalidEndpointRegionException,
  InvalidPublicKeyVersionException,
  InvalidRegionException,
  PlatformNotSupportedException,
  ExpiredTokenException,
  InvalidTokenException
});
