// @azure/arm-recoveryservicesbackup/dist-esm/src/models/parameters.js
/*
 * Copyright (c) Microsoft Corporation.
 * Licensed under the MIT License.
 *
 * Code generated by Microsoft (R) AutoRest Code Generator.
 * Changes may cause incorrect behavior and will be lost if the code is regenerated.
 */
import { BackupResourceConfigResource as BackupResourceConfigResourceMapper, PreValidateEnableBackupRequest as PreValidateEnableBackupRequestMapper, ProtectionIntentResource as ProtectionIntentResourceMapper, BackupStatusRequest as BackupStatusRequestMapper, FeatureSupportRequest as FeatureSupportRequestMapper, BackupResourceVaultConfigResource as BackupResourceVaultConfigResourceMapper, BackupResourceEncryptionConfigResource as BackupResourceEncryptionConfigResourceMapper, PrivateEndpointConnectionResource as PrivateEndpointConnectionResourceMapper, PrepareDataMoveRequest as PrepareDataMoveRequestMapper, TriggerDataMoveRequest as TriggerDataMoveRequestMapper, MoveRPAcrossTiersRequest as MoveRPAcrossTiersRequestMapper, ProtectedItemResource as ProtectedItemResourceMapper, RestoreRequestResource as RestoreRequestResourceMapper, ProtectionPolicyResource as ProtectionPolicyResourceMapper, ValidateOperationRequestResource as ValidateOperationRequestResourceMapper, ProtectionContainerResource as ProtectionContainerResourceMapper, BackupRequestResource as BackupRequestResourceMapper, ILRRequestResource as ILRRequestResourceMapper, SecurityPinBase as SecurityPinBaseMapper, ListRecoveryPointsRecommendedForMoveRequest as ListRecoveryPointsRecommendedForMoveRequestMapper, ResourceGuardProxyBaseResource as ResourceGuardProxyBaseResourceMapper, UnlockDeleteRequest as UnlockDeleteRequestMapper, FetchTieringCostInfoRequest as FetchTieringCostInfoRequestMapper } from "../models/mappers";
export const accept = {
    parameterPath: "accept",
    mapper: {
        defaultValue: "application/json",
        isConstant: true,
        serializedName: "Accept",
        type: {
            name: "String"
        }
    }
};
export const $host = {
    parameterPath: "$host",
    mapper: {
        serializedName: "$host",
        required: true,
        type: {
            name: "String"
        }
    },
    skipEncoding: true
};
export const apiVersion = {
    parameterPath: "apiVersion",
    mapper: {
        defaultValue: "2023-06-01",
        isConstant: true,
        serializedName: "api-version",
        type: {
            name: "String"
        }
    }
};
export const vaultName = {
    parameterPath: "vaultName",
    mapper: {
        serializedName: "vaultName",
        required: true,
        type: {
            name: "String"
        }
    }
};
export const resourceGroupName = {
    parameterPath: "resourceGroupName",
    mapper: {
        serializedName: "resourceGroupName",
        required: true,
        type: {
            name: "String"
        }
    }
};
export const subscriptionId = {
    parameterPath: "subscriptionId",
    mapper: {
        serializedName: "subscriptionId",
        required: true,
        type: {
            name: "String"
        }
    }
};
export const contentType = {
    parameterPath: ["options", "contentType"],
    mapper: {
        defaultValue: "application/json",
        isConstant: true,
        serializedName: "Content-Type",
        type: {
            name: "String"
        }
    }
};
export const parameters = {
    parameterPath: "parameters",
    mapper: BackupResourceConfigResourceMapper
};
export const parameters1 = {
    parameterPath: "parameters",
    mapper: PreValidateEnableBackupRequestMapper
};
export const azureRegion = {
    parameterPath: "azureRegion",
    mapper: {
        serializedName: "azureRegion",
        required: true,
        type: {
            name: "String"
        }
    }
};
export const fabricName = {
    parameterPath: "fabricName",
    mapper: {
        serializedName: "fabricName",
        required: true,
        type: {
            name: "String"
        }
    }
};
export const intentObjectName = {
    parameterPath: "intentObjectName",
    mapper: {
        serializedName: "intentObjectName",
        required: true,
        type: {
            name: "String"
        }
    }
};
export const parameters2 = {
    parameterPath: "parameters",
    mapper: ProtectionIntentResourceMapper
};
export const parameters3 = {
    parameterPath: "parameters",
    mapper: BackupStatusRequestMapper
};
export const parameters4 = {
    parameterPath: "parameters",
    mapper: FeatureSupportRequestMapper
};
export const filter = {
    parameterPath: ["options", "filter"],
    mapper: {
        serializedName: "$filter",
        type: {
            name: "String"
        }
    }
};
export const skipToken = {
    parameterPath: ["options", "skipToken"],
    mapper: {
        serializedName: "$skipToken",
        type: {
            name: "String"
        }
    }
};
export const nextLink = {
    parameterPath: "nextLink",
    mapper: {
        serializedName: "nextLink",
        required: true,
        type: {
            name: "String"
        }
    },
    skipEncoding: true
};
export const parameters5 = {
    parameterPath: "parameters",
    mapper: BackupResourceVaultConfigResourceMapper
};
export const parameters6 = {
    parameterPath: "parameters",
    mapper: BackupResourceEncryptionConfigResourceMapper
};
export const privateEndpointConnectionName = {
    parameterPath: "privateEndpointConnectionName",
    mapper: {
        serializedName: "privateEndpointConnectionName",
        required: true,
        type: {
            name: "String"
        }
    }
};
export const parameters7 = {
    parameterPath: "parameters",
    mapper: PrivateEndpointConnectionResourceMapper
};
export const operationId = {
    parameterPath: "operationId",
    mapper: {
        serializedName: "operationId",
        required: true,
        type: {
            name: "String"
        }
    }
};
export const parameters8 = {
    parameterPath: "parameters",
    mapper: PrepareDataMoveRequestMapper
};
export const parameters9 = {
    parameterPath: "parameters",
    mapper: TriggerDataMoveRequestMapper
};
export const parameters10 = {
    parameterPath: "parameters",
    mapper: MoveRPAcrossTiersRequestMapper
};
export const containerName = {
    parameterPath: "containerName",
    mapper: {
        serializedName: "containerName",
        required: true,
        type: {
            name: "String"
        }
    }
};
export const protectedItemName = {
    parameterPath: "protectedItemName",
    mapper: {
        serializedName: "protectedItemName",
        required: true,
        type: {
            name: "String"
        }
    }
};
export const recoveryPointId = {
    parameterPath: "recoveryPointId",
    mapper: {
        serializedName: "recoveryPointId",
        required: true,
        type: {
            name: "String"
        }
    }
};
export const parameters11 = {
    parameterPath: "parameters",
    mapper: ProtectedItemResourceMapper
};
export const parameters12 = {
    parameterPath: "parameters",
    mapper: RestoreRequestResourceMapper
};
export const policyName = {
    parameterPath: "policyName",
    mapper: {
        serializedName: "policyName",
        required: true,
        type: {
            name: "String"
        }
    }
};
export const parameters13 = {
    parameterPath: "parameters",
    mapper: ProtectionPolicyResourceMapper
};
export const jobName = {
    parameterPath: "jobName",
    mapper: {
        serializedName: "jobName",
        required: true,
        type: {
            name: "String"
        }
    }
};
export const parameters14 = {
    parameterPath: "parameters",
    mapper: ValidateOperationRequestResourceMapper
};
export const backupEngineName = {
    parameterPath: "backupEngineName",
    mapper: {
        serializedName: "backupEngineName",
        required: true,
        type: {
            name: "String"
        }
    }
};
export const parameters15 = {
    parameterPath: "parameters",
    mapper: ProtectionContainerResourceMapper
};
export const parameters16 = {
    parameterPath: "parameters",
    mapper: BackupRequestResourceMapper
};
export const parameters17 = {
    parameterPath: "parameters",
    mapper: ILRRequestResourceMapper
};
export const parameters18 = {
    parameterPath: ["options", "parameters"],
    mapper: SecurityPinBaseMapper
};
export const parameters19 = {
    parameterPath: "parameters",
    mapper: ListRecoveryPointsRecommendedForMoveRequestMapper
};
export const resourceGuardProxyName = {
    parameterPath: "resourceGuardProxyName",
    mapper: {
        serializedName: "resourceGuardProxyName",
        required: true,
        type: {
            name: "String"
        }
    }
};
export const parameters20 = {
    parameterPath: "parameters",
    mapper: ResourceGuardProxyBaseResourceMapper
};
export const parameters21 = {
    parameterPath: "parameters",
    mapper: UnlockDeleteRequestMapper
};
export const parameters22 = {
    parameterPath: "parameters",
    mapper: FetchTieringCostInfoRequestMapper
};
export const vaultName1 = {
    parameterPath: "vaultName",
    mapper: {
        constraints: {
            Pattern: new RegExp("^[A-Za-z][-A-Za-z0-9]*[A-Za-z0-9]$"),
            MaxLength: 50,
            MinLength: 2
        },
        serializedName: "vaultName",
        required: true,
        type: {
            name: "String"
        }
    }
};
