// @azure/arm-managedapplications/dist-esm/src/models/index.js
/*
 * Copyright (c) Microsoft Corporation.
 * Licensed under the MIT License.
 *
 * Code generated by Microsoft (R) AutoRest Code Generator.
 * Changes may cause incorrect behavior and will be lost if the code is regenerated.
 */
/** Known values of {@link Origin} that the service accepts. */
export var KnownOrigin;
(function (KnownOrigin) {
    /** User */
    KnownOrigin["User"] = "user";
    /** System */
    KnownOrigin["System"] = "system";
    /** UserSystem */
    KnownOrigin["UserSystem"] = "user,system";
})(KnownOrigin || (KnownOrigin = {}));
/** Known values of {@link ActionType} that the service accepts. */
export var KnownActionType;
(function (KnownActionType) {
    /** Internal */
    KnownActionType["Internal"] = "Internal";
})(KnownActionType || (KnownActionType = {}));
/** Known values of {@link ProvisioningState} that the service accepts. */
export var KnownProvisioningState;
(function (KnownProvisioningState) {
    /** NotSpecified */
    KnownProvisioningState["NotSpecified"] = "NotSpecified";
    /** Accepted */
    KnownProvisioningState["Accepted"] = "Accepted";
    /** Running */
    KnownProvisioningState["Running"] = "Running";
    /** Deleting */
    KnownProvisioningState["Deleting"] = "Deleting";
    /** Deleted */
    KnownProvisioningState["Deleted"] = "Deleted";
    /** Canceled */
    KnownProvisioningState["Canceled"] = "Canceled";
    /** Failed */
    KnownProvisioningState["Failed"] = "Failed";
    /** Succeeded */
    KnownProvisioningState["Succeeded"] = "Succeeded";
    /** Updating */
    KnownProvisioningState["Updating"] = "Updating";
})(KnownProvisioningState || (KnownProvisioningState = {}));
/** Known values of {@link JitApprovalMode} that the service accepts. */
export var KnownJitApprovalMode;
(function (KnownJitApprovalMode) {
    /** NotSpecified */
    KnownJitApprovalMode["NotSpecified"] = "NotSpecified";
    /** AutoApprove */
    KnownJitApprovalMode["AutoApprove"] = "AutoApprove";
    /** ManualApprove */
    KnownJitApprovalMode["ManualApprove"] = "ManualApprove";
})(KnownJitApprovalMode || (KnownJitApprovalMode = {}));
/** Known values of {@link JitApproverType} that the service accepts. */
export var KnownJitApproverType;
(function (KnownJitApproverType) {
    /** User */
    KnownJitApproverType["User"] = "user";
    /** Group */
    KnownJitApproverType["Group"] = "group";
})(KnownJitApproverType || (KnownJitApproverType = {}));
/** Known values of {@link ApplicationManagementMode} that the service accepts. */
export var KnownApplicationManagementMode;
(function (KnownApplicationManagementMode) {
    /** NotSpecified */
    KnownApplicationManagementMode["NotSpecified"] = "NotSpecified";
    /** Unmanaged */
    KnownApplicationManagementMode["Unmanaged"] = "Unmanaged";
    /** Managed */
    KnownApplicationManagementMode["Managed"] = "Managed";
})(KnownApplicationManagementMode || (KnownApplicationManagementMode = {}));
/** Known values of {@link ApplicationArtifactName} that the service accepts. */
export var KnownApplicationArtifactName;
(function (KnownApplicationArtifactName) {
    /** NotSpecified */
    KnownApplicationArtifactName["NotSpecified"] = "NotSpecified";
    /** ViewDefinition */
    KnownApplicationArtifactName["ViewDefinition"] = "ViewDefinition";
    /** Authorizations */
    KnownApplicationArtifactName["Authorizations"] = "Authorizations";
    /** CustomRoleDefinition */
    KnownApplicationArtifactName["CustomRoleDefinition"] = "CustomRoleDefinition";
})(KnownApplicationArtifactName || (KnownApplicationArtifactName = {}));
/** Known values of {@link CreatedByType} that the service accepts. */
export var KnownCreatedByType;
(function (KnownCreatedByType) {
    /** User */
    KnownCreatedByType["User"] = "User";
    /** Application */
    KnownCreatedByType["Application"] = "Application";
    /** ManagedIdentity */
    KnownCreatedByType["ManagedIdentity"] = "ManagedIdentity";
    /** Key */
    KnownCreatedByType["Key"] = "Key";
})(KnownCreatedByType || (KnownCreatedByType = {}));
/** Known values of {@link ApplicationDefinitionArtifactName} that the service accepts. */
export var KnownApplicationDefinitionArtifactName;
(function (KnownApplicationDefinitionArtifactName) {
    /** NotSpecified */
    KnownApplicationDefinitionArtifactName["NotSpecified"] = "NotSpecified";
    /** ApplicationResourceTemplate */
    KnownApplicationDefinitionArtifactName["ApplicationResourceTemplate"] = "ApplicationResourceTemplate";
    /** CreateUiDefinition */
    KnownApplicationDefinitionArtifactName["CreateUiDefinition"] = "CreateUiDefinition";
    /** MainTemplateParameters */
    KnownApplicationDefinitionArtifactName["MainTemplateParameters"] = "MainTemplateParameters";
})(KnownApplicationDefinitionArtifactName || (KnownApplicationDefinitionArtifactName = {}));
/** Known values of {@link DeploymentMode} that the service accepts. */
export var KnownDeploymentMode;
(function (KnownDeploymentMode) {
    /** NotSpecified */
    KnownDeploymentMode["NotSpecified"] = "NotSpecified";
    /** Incremental */
    KnownDeploymentMode["Incremental"] = "Incremental";
    /** Complete */
    KnownDeploymentMode["Complete"] = "Complete";
})(KnownDeploymentMode || (KnownDeploymentMode = {}));
/** Known values of {@link JitSchedulingType} that the service accepts. */
export var KnownJitSchedulingType;
(function (KnownJitSchedulingType) {
    /** NotSpecified */
    KnownJitSchedulingType["NotSpecified"] = "NotSpecified";
    /** Once */
    KnownJitSchedulingType["Once"] = "Once";
    /** Recurring */
    KnownJitSchedulingType["Recurring"] = "Recurring";
})(KnownJitSchedulingType || (KnownJitSchedulingType = {}));
/** Known values of {@link JitRequestState} that the service accepts. */
export var KnownJitRequestState;
(function (KnownJitRequestState) {
    /** NotSpecified */
    KnownJitRequestState["NotSpecified"] = "NotSpecified";
    /** Pending */
    KnownJitRequestState["Pending"] = "Pending";
    /** Approved */
    KnownJitRequestState["Approved"] = "Approved";
    /** Denied */
    KnownJitRequestState["Denied"] = "Denied";
    /** Failed */
    KnownJitRequestState["Failed"] = "Failed";
    /** Canceled */
    KnownJitRequestState["Canceled"] = "Canceled";
    /** Expired */
    KnownJitRequestState["Expired"] = "Expired";
    /** Timeout */
    KnownJitRequestState["Timeout"] = "Timeout";
})(KnownJitRequestState || (KnownJitRequestState = {}));
/** Known values of {@link Status} that the service accepts. */
export var KnownStatus;
(function (KnownStatus) {
    /** NotSpecified */
    KnownStatus["NotSpecified"] = "NotSpecified";
    /** Elevate */
    KnownStatus["Elevate"] = "Elevate";
    /** Remove */
    KnownStatus["Remove"] = "Remove";
})(KnownStatus || (KnownStatus = {}));
/** Known values of {@link Substatus} that the service accepts. */
export var KnownSubstatus;
(function (KnownSubstatus) {
    /** NotSpecified */
    KnownSubstatus["NotSpecified"] = "NotSpecified";
    /** Approved */
    KnownSubstatus["Approved"] = "Approved";
    /** Denied */
    KnownSubstatus["Denied"] = "Denied";
    /** Failed */
    KnownSubstatus["Failed"] = "Failed";
    /** Expired */
    KnownSubstatus["Expired"] = "Expired";
    /** Timeout */
    KnownSubstatus["Timeout"] = "Timeout";
})(KnownSubstatus || (KnownSubstatus = {}));
