// @azure/arm-managedapplications/dist-esm/src/models/parameters.js
/*
 * Copyright (c) Microsoft Corporation.
 * Licensed under the MIT License.
 *
 * Code generated by Microsoft (R) AutoRest Code Generator.
 * Changes may cause incorrect behavior and will be lost if the code is regenerated.
 */
import { Application as ApplicationMapper, ApplicationPatchable as ApplicationPatchableMapper, UpdateAccessDefinition as UpdateAccessDefinitionMapper, ListTokenRequest as ListTokenRequestMapper, ApplicationDefinition as ApplicationDefinitionMapper, ApplicationDefinitionPatchable as ApplicationDefinitionPatchableMapper, JitRequestDefinition as JitRequestDefinitionMapper, JitRequestPatchable as JitRequestPatchableMapper } from "../models/mappers";
export const accept = {
    parameterPath: "accept",
    mapper: {
        defaultValue: "application/json",
        isConstant: true,
        serializedName: "Accept",
        type: {
            name: "String"
        }
    }
};
export const $host = {
    parameterPath: "$host",
    mapper: {
        serializedName: "$host",
        required: true,
        type: {
            name: "String"
        }
    },
    skipEncoding: true
};
export const apiVersion = {
    parameterPath: "apiVersion",
    mapper: {
        defaultValue: "2021-07-01",
        isConstant: true,
        serializedName: "api-version",
        type: {
            name: "String"
        }
    }
};
export const nextLink = {
    parameterPath: "nextLink",
    mapper: {
        serializedName: "nextLink",
        required: true,
        type: {
            name: "String"
        }
    },
    skipEncoding: true
};
export const subscriptionId = {
    parameterPath: "subscriptionId",
    mapper: {
        constraints: {
            MinLength: 1
        },
        serializedName: "subscriptionId",
        required: true,
        type: {
            name: "String"
        }
    }
};
export const resourceGroupName = {
    parameterPath: "resourceGroupName",
    mapper: {
        constraints: {
            MaxLength: 90,
            MinLength: 1
        },
        serializedName: "resourceGroupName",
        required: true,
        type: {
            name: "String"
        }
    }
};
export const applicationName = {
    parameterPath: "applicationName",
    mapper: {
        constraints: {
            MaxLength: 64,
            MinLength: 3
        },
        serializedName: "applicationName",
        required: true,
        type: {
            name: "String"
        }
    }
};
export const contentType = {
    parameterPath: ["options", "contentType"],
    mapper: {
        defaultValue: "application/json",
        isConstant: true,
        serializedName: "Content-Type",
        type: {
            name: "String"
        }
    }
};
export const parameters = {
    parameterPath: "parameters",
    mapper: ApplicationMapper
};
export const parameters1 = {
    parameterPath: ["options", "parameters"],
    mapper: ApplicationPatchableMapper
};
export const applicationId = {
    parameterPath: "applicationId",
    mapper: {
        serializedName: "applicationId",
        required: true,
        type: {
            name: "String"
        }
    },
    skipEncoding: true
};
export const parameters2 = {
    parameterPath: "parameters",
    mapper: UpdateAccessDefinitionMapper
};
export const parameters3 = {
    parameterPath: "parameters",
    mapper: ListTokenRequestMapper
};
export const applicationDefinitionName = {
    parameterPath: "applicationDefinitionName",
    mapper: {
        constraints: {
            MaxLength: 64,
            MinLength: 3
        },
        serializedName: "applicationDefinitionName",
        required: true,
        type: {
            name: "String"
        }
    }
};
export const parameters4 = {
    parameterPath: "parameters",
    mapper: ApplicationDefinitionMapper
};
export const parameters5 = {
    parameterPath: "parameters",
    mapper: ApplicationDefinitionPatchableMapper
};
export const resourceGroupName1 = {
    parameterPath: "resourceGroupName",
    mapper: {
        constraints: {
            Pattern: new RegExp("^[-\\w\\._\\(\\)]+$"),
            MaxLength: 90,
            MinLength: 1
        },
        serializedName: "resourceGroupName",
        required: true,
        type: {
            name: "String"
        }
    }
};
export const jitRequestName = {
    parameterPath: "jitRequestName",
    mapper: {
        serializedName: "jitRequestName",
        required: true,
        type: {
            name: "String"
        }
    }
};
export const parameters6 = {
    parameterPath: "parameters",
    mapper: JitRequestDefinitionMapper
};
export const parameters7 = {
    parameterPath: "parameters",
    mapper: JitRequestPatchableMapper
};
