// @azure/arm-containerregistry/dist-esm/src/models/index.js
/*
 * Copyright (c) Microsoft Corporation.
 * Licensed under the MIT License.
 *
 * Code generated by Microsoft (R) AutoRest Code Generator.
 * Changes may cause incorrect behavior and will be lost if the code is regenerated.
 */
/** Known values of {@link ImportMode} that the service accepts. */
export var KnownImportMode;
(function (KnownImportMode) {
    /** NoForce */
    KnownImportMode["NoForce"] = "NoForce";
    /** Force */
    KnownImportMode["Force"] = "Force";
})(KnownImportMode || (KnownImportMode = {}));
/** Known values of {@link ConnectionStatus} that the service accepts. */
export var KnownConnectionStatus;
(function (KnownConnectionStatus) {
    /** Approved */
    KnownConnectionStatus["Approved"] = "Approved";
    /** Pending */
    KnownConnectionStatus["Pending"] = "Pending";
    /** Rejected */
    KnownConnectionStatus["Rejected"] = "Rejected";
    /** Disconnected */
    KnownConnectionStatus["Disconnected"] = "Disconnected";
})(KnownConnectionStatus || (KnownConnectionStatus = {}));
/** Known values of {@link ActionsRequired} that the service accepts. */
export var KnownActionsRequired;
(function (KnownActionsRequired) {
    /** None */
    KnownActionsRequired["None"] = "None";
    /** Recreate */
    KnownActionsRequired["Recreate"] = "Recreate";
})(KnownActionsRequired || (KnownActionsRequired = {}));
/** Known values of {@link ProvisioningState} that the service accepts. */
export var KnownProvisioningState;
(function (KnownProvisioningState) {
    /** Creating */
    KnownProvisioningState["Creating"] = "Creating";
    /** Updating */
    KnownProvisioningState["Updating"] = "Updating";
    /** Deleting */
    KnownProvisioningState["Deleting"] = "Deleting";
    /** Succeeded */
    KnownProvisioningState["Succeeded"] = "Succeeded";
    /** Failed */
    KnownProvisioningState["Failed"] = "Failed";
    /** Canceled */
    KnownProvisioningState["Canceled"] = "Canceled";
})(KnownProvisioningState || (KnownProvisioningState = {}));
/** Known values of {@link CreatedByType} that the service accepts. */
export var KnownCreatedByType;
(function (KnownCreatedByType) {
    /** User */
    KnownCreatedByType["User"] = "User";
    /** Application */
    KnownCreatedByType["Application"] = "Application";
    /** ManagedIdentity */
    KnownCreatedByType["ManagedIdentity"] = "ManagedIdentity";
    /** Key */
    KnownCreatedByType["Key"] = "Key";
})(KnownCreatedByType || (KnownCreatedByType = {}));
/** Known values of {@link LastModifiedByType} that the service accepts. */
export var KnownLastModifiedByType;
(function (KnownLastModifiedByType) {
    /** User */
    KnownLastModifiedByType["User"] = "User";
    /** Application */
    KnownLastModifiedByType["Application"] = "Application";
    /** ManagedIdentity */
    KnownLastModifiedByType["ManagedIdentity"] = "ManagedIdentity";
    /** Key */
    KnownLastModifiedByType["Key"] = "Key";
})(KnownLastModifiedByType || (KnownLastModifiedByType = {}));
/** Known values of {@link SkuName} that the service accepts. */
export var KnownSkuName;
(function (KnownSkuName) {
    /** Classic */
    KnownSkuName["Classic"] = "Classic";
    /** Basic */
    KnownSkuName["Basic"] = "Basic";
    /** Standard */
    KnownSkuName["Standard"] = "Standard";
    /** Premium */
    KnownSkuName["Premium"] = "Premium";
})(KnownSkuName || (KnownSkuName = {}));
/** Known values of {@link SkuTier} that the service accepts. */
export var KnownSkuTier;
(function (KnownSkuTier) {
    /** Classic */
    KnownSkuTier["Classic"] = "Classic";
    /** Basic */
    KnownSkuTier["Basic"] = "Basic";
    /** Standard */
    KnownSkuTier["Standard"] = "Standard";
    /** Premium */
    KnownSkuTier["Premium"] = "Premium";
})(KnownSkuTier || (KnownSkuTier = {}));
/** Known values of {@link DefaultAction} that the service accepts. */
export var KnownDefaultAction;
(function (KnownDefaultAction) {
    /** Allow */
    KnownDefaultAction["Allow"] = "Allow";
    /** Deny */
    KnownDefaultAction["Deny"] = "Deny";
})(KnownDefaultAction || (KnownDefaultAction = {}));
/** Known values of {@link Action} that the service accepts. */
export var KnownAction;
(function (KnownAction) {
    /** Allow */
    KnownAction["Allow"] = "Allow";
})(KnownAction || (KnownAction = {}));
/** Known values of {@link PolicyStatus} that the service accepts. */
export var KnownPolicyStatus;
(function (KnownPolicyStatus) {
    /** Enabled */
    KnownPolicyStatus["Enabled"] = "enabled";
    /** Disabled */
    KnownPolicyStatus["Disabled"] = "disabled";
})(KnownPolicyStatus || (KnownPolicyStatus = {}));
/** Known values of {@link TrustPolicyType} that the service accepts. */
export var KnownTrustPolicyType;
(function (KnownTrustPolicyType) {
    /** Notary */
    KnownTrustPolicyType["Notary"] = "Notary";
})(KnownTrustPolicyType || (KnownTrustPolicyType = {}));
/** Known values of {@link ExportPolicyStatus} that the service accepts. */
export var KnownExportPolicyStatus;
(function (KnownExportPolicyStatus) {
    /** Enabled */
    KnownExportPolicyStatus["Enabled"] = "enabled";
    /** Disabled */
    KnownExportPolicyStatus["Disabled"] = "disabled";
})(KnownExportPolicyStatus || (KnownExportPolicyStatus = {}));
/** Known values of {@link EncryptionStatus} that the service accepts. */
export var KnownEncryptionStatus;
(function (KnownEncryptionStatus) {
    /** Enabled */
    KnownEncryptionStatus["Enabled"] = "enabled";
    /** Disabled */
    KnownEncryptionStatus["Disabled"] = "disabled";
})(KnownEncryptionStatus || (KnownEncryptionStatus = {}));
/** Known values of {@link PublicNetworkAccess} that the service accepts. */
export var KnownPublicNetworkAccess;
(function (KnownPublicNetworkAccess) {
    /** Enabled */
    KnownPublicNetworkAccess["Enabled"] = "Enabled";
    /** Disabled */
    KnownPublicNetworkAccess["Disabled"] = "Disabled";
})(KnownPublicNetworkAccess || (KnownPublicNetworkAccess = {}));
/** Known values of {@link NetworkRuleBypassOptions} that the service accepts. */
export var KnownNetworkRuleBypassOptions;
(function (KnownNetworkRuleBypassOptions) {
    /** AzureServices */
    KnownNetworkRuleBypassOptions["AzureServices"] = "AzureServices";
    /** None */
    KnownNetworkRuleBypassOptions["None"] = "None";
})(KnownNetworkRuleBypassOptions || (KnownNetworkRuleBypassOptions = {}));
/** Known values of {@link ZoneRedundancy} that the service accepts. */
export var KnownZoneRedundancy;
(function (KnownZoneRedundancy) {
    /** Enabled */
    KnownZoneRedundancy["Enabled"] = "Enabled";
    /** Disabled */
    KnownZoneRedundancy["Disabled"] = "Disabled";
})(KnownZoneRedundancy || (KnownZoneRedundancy = {}));
/** Known values of {@link RegistryUsageUnit} that the service accepts. */
export var KnownRegistryUsageUnit;
(function (KnownRegistryUsageUnit) {
    /** Count */
    KnownRegistryUsageUnit["Count"] = "Count";
    /** Bytes */
    KnownRegistryUsageUnit["Bytes"] = "Bytes";
})(KnownRegistryUsageUnit || (KnownRegistryUsageUnit = {}));
/** Known values of {@link TokenCertificateName} that the service accepts. */
export var KnownTokenCertificateName;
(function (KnownTokenCertificateName) {
    /** Certificate1 */
    KnownTokenCertificateName["Certificate1"] = "certificate1";
    /** Certificate2 */
    KnownTokenCertificateName["Certificate2"] = "certificate2";
})(KnownTokenCertificateName || (KnownTokenCertificateName = {}));
/** Known values of {@link TokenPasswordName} that the service accepts. */
export var KnownTokenPasswordName;
(function (KnownTokenPasswordName) {
    /** Password1 */
    KnownTokenPasswordName["Password1"] = "password1";
    /** Password2 */
    KnownTokenPasswordName["Password2"] = "password2";
})(KnownTokenPasswordName || (KnownTokenPasswordName = {}));
/** Known values of {@link TokenStatus} that the service accepts. */
export var KnownTokenStatus;
(function (KnownTokenStatus) {
    /** Enabled */
    KnownTokenStatus["Enabled"] = "enabled";
    /** Disabled */
    KnownTokenStatus["Disabled"] = "disabled";
})(KnownTokenStatus || (KnownTokenStatus = {}));
/** Known values of {@link WebhookStatus} that the service accepts. */
export var KnownWebhookStatus;
(function (KnownWebhookStatus) {
    /** Enabled */
    KnownWebhookStatus["Enabled"] = "enabled";
    /** Disabled */
    KnownWebhookStatus["Disabled"] = "disabled";
})(KnownWebhookStatus || (KnownWebhookStatus = {}));
/** Known values of {@link WebhookAction} that the service accepts. */
export var KnownWebhookAction;
(function (KnownWebhookAction) {
    /** Push */
    KnownWebhookAction["Push"] = "push";
    /** Delete */
    KnownWebhookAction["Delete"] = "delete";
    /** Quarantine */
    KnownWebhookAction["Quarantine"] = "quarantine";
    /** ChartPush */
    KnownWebhookAction["ChartPush"] = "chart_push";
    /** ChartDelete */
    KnownWebhookAction["ChartDelete"] = "chart_delete";
})(KnownWebhookAction || (KnownWebhookAction = {}));
/** Known values of {@link OS} that the service accepts. */
export var KnownOS;
(function (KnownOS) {
    /** Windows */
    KnownOS["Windows"] = "Windows";
    /** Linux */
    KnownOS["Linux"] = "Linux";
})(KnownOS || (KnownOS = {}));
/** Known values of {@link RunStatus} that the service accepts. */
export var KnownRunStatus;
(function (KnownRunStatus) {
    /** Queued */
    KnownRunStatus["Queued"] = "Queued";
    /** Started */
    KnownRunStatus["Started"] = "Started";
    /** Running */
    KnownRunStatus["Running"] = "Running";
    /** Succeeded */
    KnownRunStatus["Succeeded"] = "Succeeded";
    /** Failed */
    KnownRunStatus["Failed"] = "Failed";
    /** Canceled */
    KnownRunStatus["Canceled"] = "Canceled";
    /** Error */
    KnownRunStatus["Error"] = "Error";
    /** Timeout */
    KnownRunStatus["Timeout"] = "Timeout";
})(KnownRunStatus || (KnownRunStatus = {}));
/** Known values of {@link RunType} that the service accepts. */
export var KnownRunType;
(function (KnownRunType) {
    /** QuickBuild */
    KnownRunType["QuickBuild"] = "QuickBuild";
    /** QuickRun */
    KnownRunType["QuickRun"] = "QuickRun";
    /** AutoBuild */
    KnownRunType["AutoBuild"] = "AutoBuild";
    /** AutoRun */
    KnownRunType["AutoRun"] = "AutoRun";
})(KnownRunType || (KnownRunType = {}));
/** Known values of {@link Architecture} that the service accepts. */
export var KnownArchitecture;
(function (KnownArchitecture) {
    /** Amd64 */
    KnownArchitecture["Amd64"] = "amd64";
    /** X86 */
    KnownArchitecture["X86"] = "x86";
    /** ThreeHundredEightySix */
    KnownArchitecture["ThreeHundredEightySix"] = "386";
    /** Arm */
    KnownArchitecture["Arm"] = "arm";
    /** Arm64 */
    KnownArchitecture["Arm64"] = "arm64";
})(KnownArchitecture || (KnownArchitecture = {}));
/** Known values of {@link Variant} that the service accepts. */
export var KnownVariant;
(function (KnownVariant) {
    /** V6 */
    KnownVariant["V6"] = "v6";
    /** V7 */
    KnownVariant["V7"] = "v7";
    /** V8 */
    KnownVariant["V8"] = "v8";
})(KnownVariant || (KnownVariant = {}));
/** Known values of {@link TaskStatus} that the service accepts. */
export var KnownTaskStatus;
(function (KnownTaskStatus) {
    /** Disabled */
    KnownTaskStatus["Disabled"] = "Disabled";
    /** Enabled */
    KnownTaskStatus["Enabled"] = "Enabled";
})(KnownTaskStatus || (KnownTaskStatus = {}));
/** Known values of {@link StepType} that the service accepts. */
export var KnownStepType;
(function (KnownStepType) {
    /** Docker */
    KnownStepType["Docker"] = "Docker";
    /** FileTask */
    KnownStepType["FileTask"] = "FileTask";
    /** EncodedTask */
    KnownStepType["EncodedTask"] = "EncodedTask";
})(KnownStepType || (KnownStepType = {}));
/** Known values of {@link BaseImageDependencyType} that the service accepts. */
export var KnownBaseImageDependencyType;
(function (KnownBaseImageDependencyType) {
    /** BuildTime */
    KnownBaseImageDependencyType["BuildTime"] = "BuildTime";
    /** RunTime */
    KnownBaseImageDependencyType["RunTime"] = "RunTime";
})(KnownBaseImageDependencyType || (KnownBaseImageDependencyType = {}));
/** Known values of {@link TriggerStatus} that the service accepts. */
export var KnownTriggerStatus;
(function (KnownTriggerStatus) {
    /** Disabled */
    KnownTriggerStatus["Disabled"] = "Disabled";
    /** Enabled */
    KnownTriggerStatus["Enabled"] = "Enabled";
})(KnownTriggerStatus || (KnownTriggerStatus = {}));
/** Known values of {@link SourceControlType} that the service accepts. */
export var KnownSourceControlType;
(function (KnownSourceControlType) {
    /** Github */
    KnownSourceControlType["Github"] = "Github";
    /** VisualStudioTeamService */
    KnownSourceControlType["VisualStudioTeamService"] = "VisualStudioTeamService";
})(KnownSourceControlType || (KnownSourceControlType = {}));
/** Known values of {@link TokenType} that the service accepts. */
export var KnownTokenType;
(function (KnownTokenType) {
    /** PAT */
    KnownTokenType["PAT"] = "PAT";
    /** OAuth */
    KnownTokenType["OAuth"] = "OAuth";
})(KnownTokenType || (KnownTokenType = {}));
/** Known values of {@link SourceTriggerEvent} that the service accepts. */
export var KnownSourceTriggerEvent;
(function (KnownSourceTriggerEvent) {
    /** Commit */
    KnownSourceTriggerEvent["Commit"] = "commit";
    /** Pullrequest */
    KnownSourceTriggerEvent["Pullrequest"] = "pullrequest";
})(KnownSourceTriggerEvent || (KnownSourceTriggerEvent = {}));
/** Known values of {@link BaseImageTriggerType} that the service accepts. */
export var KnownBaseImageTriggerType;
(function (KnownBaseImageTriggerType) {
    /** All */
    KnownBaseImageTriggerType["All"] = "All";
    /** Runtime */
    KnownBaseImageTriggerType["Runtime"] = "Runtime";
})(KnownBaseImageTriggerType || (KnownBaseImageTriggerType = {}));
/** Known values of {@link UpdateTriggerPayloadType} that the service accepts. */
export var KnownUpdateTriggerPayloadType;
(function (KnownUpdateTriggerPayloadType) {
    /** Default */
    KnownUpdateTriggerPayloadType["Default"] = "Default";
    /** Token */
    KnownUpdateTriggerPayloadType["Token"] = "Token";
})(KnownUpdateTriggerPayloadType || (KnownUpdateTriggerPayloadType = {}));
/** Known values of {@link SourceRegistryLoginMode} that the service accepts. */
export var KnownSourceRegistryLoginMode;
(function (KnownSourceRegistryLoginMode) {
    /** None */
    KnownSourceRegistryLoginMode["None"] = "None";
    /** Default */
    KnownSourceRegistryLoginMode["Default"] = "Default";
})(KnownSourceRegistryLoginMode || (KnownSourceRegistryLoginMode = {}));
/** Known values of {@link SecretObjectType} that the service accepts. */
export var KnownSecretObjectType;
(function (KnownSecretObjectType) {
    /** Opaque */
    KnownSecretObjectType["Opaque"] = "Opaque";
    /** Vaultsecret */
    KnownSecretObjectType["Vaultsecret"] = "Vaultsecret";
})(KnownSecretObjectType || (KnownSecretObjectType = {}));
