// @azure/arm-iotcentral/dist-esm/src/models/index.js
/*
 * Copyright (c) Microsoft Corporation.
 * Licensed under the MIT License.
 *
 * Code generated by Microsoft (R) AutoRest Code Generator.
 * Changes may cause incorrect behavior and will be lost if the code is regenerated.
 */
/** Known values of {@link AppState} that the service accepts. */
export var KnownAppState;
(function (KnownAppState) {
    KnownAppState["Created"] = "created";
    KnownAppState["Suspended"] = "suspended";
})(KnownAppState || (KnownAppState = {}));
/** Known values of {@link AppSku} that the service accepts. */
export var KnownAppSku;
(function (KnownAppSku) {
    KnownAppSku["ST0"] = "ST0";
    KnownAppSku["ST1"] = "ST1";
    KnownAppSku["ST2"] = "ST2";
})(KnownAppSku || (KnownAppSku = {}));
/** Known values of {@link SystemAssignedServiceIdentityType} that the service accepts. */
export var KnownSystemAssignedServiceIdentityType;
(function (KnownSystemAssignedServiceIdentityType) {
    KnownSystemAssignedServiceIdentityType["None"] = "None";
    KnownSystemAssignedServiceIdentityType["SystemAssigned"] = "SystemAssigned";
})(KnownSystemAssignedServiceIdentityType || (KnownSystemAssignedServiceIdentityType = {}));
