// @azure/arm-cosmosdb/dist-esm/src/operations/collection.js
/*
 * Copyright (c) Microsoft Corporation.
 * Licensed under the MIT License.
 *
 * Code generated by Microsoft (R) AutoRest Code Generator.
 * Changes may cause incorrect behavior and will be lost if the code is regenerated.
 */
import { __asyncDelegator, __asyncGenerator, __asyncValues, __await } from "tslib";
import * as coreClient from "@azure/core-client";
import * as Mappers from "../models/mappers";
import * as Parameters from "../models/parameters";
/// <reference lib="esnext.asynciterable" />
/** Class containing Collection operations. */
export class CollectionImpl {
    /**
     * Initialize a new instance of the class Collection class.
     * @param client Reference to the service client
     */
    constructor(client) {
        this.client = client;
    }
    /**
     * Retrieves the metrics determined by the given filter for the given database account and collection.
     * @param resourceGroupName The name of the resource group. The name is case insensitive.
     * @param accountName Cosmos DB database account name.
     * @param databaseRid Cosmos DB database rid.
     * @param collectionRid Cosmos DB collection rid.
     * @param filter An OData filter expression that describes a subset of metrics to return. The
     *               parameters that can be filtered are name.value (name of the metric, can have an or of multiple
     *               names), startTime, endTime, and timeGrain. The supported operator is eq.
     * @param options The options parameters.
     */
    listMetrics(resourceGroupName, accountName, databaseRid, collectionRid, filter, options) {
        const iter = this.listMetricsPagingAll(resourceGroupName, accountName, databaseRid, collectionRid, filter, options);
        return {
            next() {
                return iter.next();
            },
            [Symbol.asyncIterator]() {
                return this;
            },
            byPage: (settings) => {
                if (settings === null || settings === void 0 ? void 0 : settings.maxPageSize) {
                    throw new Error("maxPageSize is not supported by this operation.");
                }
                return this.listMetricsPagingPage(resourceGroupName, accountName, databaseRid, collectionRid, filter, options, settings);
            }
        };
    }
    listMetricsPagingPage(resourceGroupName, accountName, databaseRid, collectionRid, filter, options, _settings) {
        return __asyncGenerator(this, arguments, function* listMetricsPagingPage_1() {
            let result;
            result = yield __await(this._listMetrics(resourceGroupName, accountName, databaseRid, collectionRid, filter, options));
            yield yield __await(result.value || []);
        });
    }
    listMetricsPagingAll(resourceGroupName, accountName, databaseRid, collectionRid, filter, options) {
        return __asyncGenerator(this, arguments, function* listMetricsPagingAll_1() {
            var _a, e_1, _b, _c;
            try {
                for (var _d = true, _e = __asyncValues(this.listMetricsPagingPage(resourceGroupName, accountName, databaseRid, collectionRid, filter, options)), _f; _f = yield __await(_e.next()), _a = _f.done, !_a;) {
                    _c = _f.value;
                    _d = false;
                    try {
                        const page = _c;
                        yield __await(yield* __asyncDelegator(__asyncValues(page)));
                    }
                    finally {
                        _d = true;
                    }
                }
            }
            catch (e_1_1) { e_1 = { error: e_1_1 }; }
            finally {
                try {
                    if (!_d && !_a && (_b = _e.return)) yield __await(_b.call(_e));
                }
                finally { if (e_1) throw e_1.error; }
            }
        });
    }
    /**
     * Retrieves the usages (most recent storage data) for the given collection.
     * @param resourceGroupName The name of the resource group. The name is case insensitive.
     * @param accountName Cosmos DB database account name.
     * @param databaseRid Cosmos DB database rid.
     * @param collectionRid Cosmos DB collection rid.
     * @param options The options parameters.
     */
    listUsages(resourceGroupName, accountName, databaseRid, collectionRid, options) {
        const iter = this.listUsagesPagingAll(resourceGroupName, accountName, databaseRid, collectionRid, options);
        return {
            next() {
                return iter.next();
            },
            [Symbol.asyncIterator]() {
                return this;
            },
            byPage: (settings) => {
                if (settings === null || settings === void 0 ? void 0 : settings.maxPageSize) {
                    throw new Error("maxPageSize is not supported by this operation.");
                }
                return this.listUsagesPagingPage(resourceGroupName, accountName, databaseRid, collectionRid, options, settings);
            }
        };
    }
    listUsagesPagingPage(resourceGroupName, accountName, databaseRid, collectionRid, options, _settings) {
        return __asyncGenerator(this, arguments, function* listUsagesPagingPage_1() {
            let result;
            result = yield __await(this._listUsages(resourceGroupName, accountName, databaseRid, collectionRid, options));
            yield yield __await(result.value || []);
        });
    }
    listUsagesPagingAll(resourceGroupName, accountName, databaseRid, collectionRid, options) {
        return __asyncGenerator(this, arguments, function* listUsagesPagingAll_1() {
            var _a, e_2, _b, _c;
            try {
                for (var _d = true, _e = __asyncValues(this.listUsagesPagingPage(resourceGroupName, accountName, databaseRid, collectionRid, options)), _f; _f = yield __await(_e.next()), _a = _f.done, !_a;) {
                    _c = _f.value;
                    _d = false;
                    try {
                        const page = _c;
                        yield __await(yield* __asyncDelegator(__asyncValues(page)));
                    }
                    finally {
                        _d = true;
                    }
                }
            }
            catch (e_2_1) { e_2 = { error: e_2_1 }; }
            finally {
                try {
                    if (!_d && !_a && (_b = _e.return)) yield __await(_b.call(_e));
                }
                finally { if (e_2) throw e_2.error; }
            }
        });
    }
    /**
     * Retrieves metric definitions for the given collection.
     * @param resourceGroupName The name of the resource group. The name is case insensitive.
     * @param accountName Cosmos DB database account name.
     * @param databaseRid Cosmos DB database rid.
     * @param collectionRid Cosmos DB collection rid.
     * @param options The options parameters.
     */
    listMetricDefinitions(resourceGroupName, accountName, databaseRid, collectionRid, options) {
        const iter = this.listMetricDefinitionsPagingAll(resourceGroupName, accountName, databaseRid, collectionRid, options);
        return {
            next() {
                return iter.next();
            },
            [Symbol.asyncIterator]() {
                return this;
            },
            byPage: (settings) => {
                if (settings === null || settings === void 0 ? void 0 : settings.maxPageSize) {
                    throw new Error("maxPageSize is not supported by this operation.");
                }
                return this.listMetricDefinitionsPagingPage(resourceGroupName, accountName, databaseRid, collectionRid, options, settings);
            }
        };
    }
    listMetricDefinitionsPagingPage(resourceGroupName, accountName, databaseRid, collectionRid, options, _settings) {
        return __asyncGenerator(this, arguments, function* listMetricDefinitionsPagingPage_1() {
            let result;
            result = yield __await(this._listMetricDefinitions(resourceGroupName, accountName, databaseRid, collectionRid, options));
            yield yield __await(result.value || []);
        });
    }
    listMetricDefinitionsPagingAll(resourceGroupName, accountName, databaseRid, collectionRid, options) {
        return __asyncGenerator(this, arguments, function* listMetricDefinitionsPagingAll_1() {
            var _a, e_3, _b, _c;
            try {
                for (var _d = true, _e = __asyncValues(this.listMetricDefinitionsPagingPage(resourceGroupName, accountName, databaseRid, collectionRid, options)), _f; _f = yield __await(_e.next()), _a = _f.done, !_a;) {
                    _c = _f.value;
                    _d = false;
                    try {
                        const page = _c;
                        yield __await(yield* __asyncDelegator(__asyncValues(page)));
                    }
                    finally {
                        _d = true;
                    }
                }
            }
            catch (e_3_1) { e_3 = { error: e_3_1 }; }
            finally {
                try {
                    if (!_d && !_a && (_b = _e.return)) yield __await(_b.call(_e));
                }
                finally { if (e_3) throw e_3.error; }
            }
        });
    }
    /**
     * Retrieves the metrics determined by the given filter for the given database account and collection.
     * @param resourceGroupName The name of the resource group. The name is case insensitive.
     * @param accountName Cosmos DB database account name.
     * @param databaseRid Cosmos DB database rid.
     * @param collectionRid Cosmos DB collection rid.
     * @param filter An OData filter expression that describes a subset of metrics to return. The
     *               parameters that can be filtered are name.value (name of the metric, can have an or of multiple
     *               names), startTime, endTime, and timeGrain. The supported operator is eq.
     * @param options The options parameters.
     */
    _listMetrics(resourceGroupName, accountName, databaseRid, collectionRid, filter, options) {
        return this.client.sendOperationRequest({
            resourceGroupName,
            accountName,
            databaseRid,
            collectionRid,
            filter,
            options
        }, listMetricsOperationSpec);
    }
    /**
     * Retrieves the usages (most recent storage data) for the given collection.
     * @param resourceGroupName The name of the resource group. The name is case insensitive.
     * @param accountName Cosmos DB database account name.
     * @param databaseRid Cosmos DB database rid.
     * @param collectionRid Cosmos DB collection rid.
     * @param options The options parameters.
     */
    _listUsages(resourceGroupName, accountName, databaseRid, collectionRid, options) {
        return this.client.sendOperationRequest({ resourceGroupName, accountName, databaseRid, collectionRid, options }, listUsagesOperationSpec);
    }
    /**
     * Retrieves metric definitions for the given collection.
     * @param resourceGroupName The name of the resource group. The name is case insensitive.
     * @param accountName Cosmos DB database account name.
     * @param databaseRid Cosmos DB database rid.
     * @param collectionRid Cosmos DB collection rid.
     * @param options The options parameters.
     */
    _listMetricDefinitions(resourceGroupName, accountName, databaseRid, collectionRid, options) {
        return this.client.sendOperationRequest({ resourceGroupName, accountName, databaseRid, collectionRid, options }, listMetricDefinitionsOperationSpec);
    }
}
// Operation Specifications
const serializer = coreClient.createSerializer(Mappers, /* isXml */ false);
const listMetricsOperationSpec = {
    path: "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.DocumentDB/databaseAccounts/{accountName}/databases/{databaseRid}/collections/{collectionRid}/metrics",
    httpMethod: "GET",
    responses: {
        200: {
            bodyMapper: Mappers.MetricListResult
        }
    },
    queryParameters: [Parameters.apiVersion, Parameters.filter],
    urlParameters: [
        Parameters.$host,
        Parameters.subscriptionId,
        Parameters.resourceGroupName,
        Parameters.accountName,
        Parameters.databaseRid,
        Parameters.collectionRid
    ],
    headerParameters: [Parameters.accept],
    serializer
};
const listUsagesOperationSpec = {
    path: "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.DocumentDB/databaseAccounts/{accountName}/databases/{databaseRid}/collections/{collectionRid}/usages",
    httpMethod: "GET",
    responses: {
        200: {
            bodyMapper: Mappers.UsagesResult
        }
    },
    queryParameters: [Parameters.apiVersion, Parameters.filter1],
    urlParameters: [
        Parameters.$host,
        Parameters.subscriptionId,
        Parameters.resourceGroupName,
        Parameters.accountName,
        Parameters.databaseRid,
        Parameters.collectionRid
    ],
    headerParameters: [Parameters.accept],
    serializer
};
const listMetricDefinitionsOperationSpec = {
    path: "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.DocumentDB/databaseAccounts/{accountName}/databases/{databaseRid}/collections/{collectionRid}/metricDefinitions",
    httpMethod: "GET",
    responses: {
        200: {
            bodyMapper: Mappers.MetricDefinitionsListResult
        }
    },
    queryParameters: [Parameters.apiVersion],
    urlParameters: [
        Parameters.$host,
        Parameters.subscriptionId,
        Parameters.resourceGroupName,
        Parameters.accountName,
        Parameters.databaseRid,
        Parameters.collectionRid
    ],
    headerParameters: [Parameters.accept],
    serializer
};
