// @azure/arm-commerce/esm/models/parameters.js
/*
 * Copyright (c) Microsoft Corporation.
 * Licensed under the MIT License.
 *
 * Code generated by Microsoft (R) AutoRest Code Generator.
 * Changes may cause incorrect behavior and will be lost if the code is
 * regenerated.
 */
export var acceptLanguage = {
    parameterPath: "acceptLanguage",
    mapper: {
        serializedName: "accept-language",
        defaultValue: 'en-US',
        type: {
            name: "String"
        }
    }
};
export var aggregationGranularity = {
    parameterPath: [
        "options",
        "aggregationGranularity"
    ],
    mapper: {
        serializedName: "aggregationGranularity",
        defaultValue: 'Daily',
        type: {
            name: "Enum",
            allowedValues: [
                "Daily",
                "Hourly"
            ]
        }
    }
};
export var apiVersion = {
    parameterPath: "apiVersion",
    mapper: {
        required: true,
        serializedName: "api-version",
        type: {
            name: "String"
        }
    }
};
export var continuationToken = {
    parameterPath: [
        "options",
        "continuationToken"
    ],
    mapper: {
        serializedName: "continuationToken",
        type: {
            name: "String"
        }
    }
};
export var filter = {
    parameterPath: "filter",
    mapper: {
        required: true,
        serializedName: "$filter",
        type: {
            name: "String"
        }
    }
};
export var nextPageLink = {
    parameterPath: "nextPageLink",
    mapper: {
        required: true,
        serializedName: "nextLink",
        type: {
            name: "String"
        }
    },
    skipEncoding: true
};
export var reportedEndTime = {
    parameterPath: "reportedEndTime",
    mapper: {
        required: true,
        serializedName: "reportedEndTime",
        type: {
            name: "DateTime"
        }
    }
};
export var reportedStartTime = {
    parameterPath: "reportedStartTime",
    mapper: {
        required: true,
        serializedName: "reportedStartTime",
        type: {
            name: "DateTime"
        }
    }
};
export var showDetails = {
    parameterPath: [
        "options",
        "showDetails"
    ],
    mapper: {
        serializedName: "showDetails",
        type: {
            name: "Boolean"
        }
    }
};
export var subscriptionId = {
    parameterPath: "subscriptionId",
    mapper: {
        required: true,
        serializedName: "subscriptionId",
        type: {
            name: "String"
        }
    }
};
