// @azure/arm-avs/dist-esm/src/operations/scriptCmdlets.js
/*
 * Copyright (c) Microsoft Corporation.
 * Licensed under the MIT License.
 *
 * Code generated by Microsoft (R) AutoRest Code Generator.
 * Changes may cause incorrect behavior and will be lost if the code is regenerated.
 */
import { __asyncDelegator, __asyncGenerator, __asyncValues, __await } from "tslib";
import { setContinuationToken } from "../pagingHelper";
import * as coreClient from "@azure/core-client";
import * as Mappers from "../models/mappers";
import * as Parameters from "../models/parameters";
/// <reference lib="esnext.asynciterable" />
/** Class containing ScriptCmdlets operations. */
export class ScriptCmdletsImpl {
    /**
     * Initialize a new instance of the class ScriptCmdlets class.
     * @param client Reference to the service client
     */
    constructor(client) {
        this.client = client;
    }
    /**
     * List script cmdlet resources available for a private cloud to create a script execution resource on
     * a private cloud
     * @param resourceGroupName The name of the resource group. The name is case insensitive.
     * @param privateCloudName Name of the private cloud
     * @param scriptPackageName Name of the script package in the private cloud
     * @param options The options parameters.
     */
    list(resourceGroupName, privateCloudName, scriptPackageName, options) {
        const iter = this.listPagingAll(resourceGroupName, privateCloudName, scriptPackageName, options);
        return {
            next() {
                return iter.next();
            },
            [Symbol.asyncIterator]() {
                return this;
            },
            byPage: (settings) => {
                if (settings === null || settings === void 0 ? void 0 : settings.maxPageSize) {
                    throw new Error("maxPageSize is not supported by this operation.");
                }
                return this.listPagingPage(resourceGroupName, privateCloudName, scriptPackageName, options, settings);
            }
        };
    }
    listPagingPage(resourceGroupName, privateCloudName, scriptPackageName, options, settings) {
        return __asyncGenerator(this, arguments, function* listPagingPage_1() {
            let result;
            let continuationToken = settings === null || settings === void 0 ? void 0 : settings.continuationToken;
            if (!continuationToken) {
                result = yield __await(this._list(resourceGroupName, privateCloudName, scriptPackageName, options));
                let page = result.value || [];
                continuationToken = result.nextLink;
                setContinuationToken(page, continuationToken);
                yield yield __await(page);
            }
            while (continuationToken) {
                result = yield __await(this._listNext(resourceGroupName, privateCloudName, scriptPackageName, continuationToken, options));
                continuationToken = result.nextLink;
                let page = result.value || [];
                setContinuationToken(page, continuationToken);
                yield yield __await(page);
            }
        });
    }
    listPagingAll(resourceGroupName, privateCloudName, scriptPackageName, options) {
        return __asyncGenerator(this, arguments, function* listPagingAll_1() {
            var _a, e_1, _b, _c;
            try {
                for (var _d = true, _e = __asyncValues(this.listPagingPage(resourceGroupName, privateCloudName, scriptPackageName, options)), _f; _f = yield __await(_e.next()), _a = _f.done, !_a;) {
                    _c = _f.value;
                    _d = false;
                    try {
                        const page = _c;
                        yield __await(yield* __asyncDelegator(__asyncValues(page)));
                    }
                    finally {
                        _d = true;
                    }
                }
            }
            catch (e_1_1) { e_1 = { error: e_1_1 }; }
            finally {
                try {
                    if (!_d && !_a && (_b = _e.return)) yield __await(_b.call(_e));
                }
                finally { if (e_1) throw e_1.error; }
            }
        });
    }
    /**
     * List script cmdlet resources available for a private cloud to create a script execution resource on
     * a private cloud
     * @param resourceGroupName The name of the resource group. The name is case insensitive.
     * @param privateCloudName Name of the private cloud
     * @param scriptPackageName Name of the script package in the private cloud
     * @param options The options parameters.
     */
    _list(resourceGroupName, privateCloudName, scriptPackageName, options) {
        return this.client.sendOperationRequest({ resourceGroupName, privateCloudName, scriptPackageName, options }, listOperationSpec);
    }
    /**
     * Return information about a script cmdlet resource in a specific package on a private cloud
     * @param resourceGroupName The name of the resource group. The name is case insensitive.
     * @param privateCloudName Name of the private cloud
     * @param scriptPackageName Name of the script package in the private cloud
     * @param scriptCmdletName Name of the script cmdlet resource in the script package in the private
     *                         cloud
     * @param options The options parameters.
     */
    get(resourceGroupName, privateCloudName, scriptPackageName, scriptCmdletName, options) {
        return this.client.sendOperationRequest({
            resourceGroupName,
            privateCloudName,
            scriptPackageName,
            scriptCmdletName,
            options
        }, getOperationSpec);
    }
    /**
     * ListNext
     * @param resourceGroupName The name of the resource group. The name is case insensitive.
     * @param privateCloudName Name of the private cloud
     * @param scriptPackageName Name of the script package in the private cloud
     * @param nextLink The nextLink from the previous successful call to the List method.
     * @param options The options parameters.
     */
    _listNext(resourceGroupName, privateCloudName, scriptPackageName, nextLink, options) {
        return this.client.sendOperationRequest({
            resourceGroupName,
            privateCloudName,
            scriptPackageName,
            nextLink,
            options
        }, listNextOperationSpec);
    }
}
// Operation Specifications
const serializer = coreClient.createSerializer(Mappers, /* isXml */ false);
const listOperationSpec = {
    path: "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.AVS/privateClouds/{privateCloudName}/scriptPackages/{scriptPackageName}/scriptCmdlets",
    httpMethod: "GET",
    responses: {
        200: {
            bodyMapper: Mappers.ScriptCmdletsList
        },
        default: {
            bodyMapper: Mappers.ErrorResponse
        }
    },
    queryParameters: [Parameters.apiVersion],
    urlParameters: [
        Parameters.$host,
        Parameters.subscriptionId,
        Parameters.resourceGroupName,
        Parameters.privateCloudName,
        Parameters.scriptPackageName
    ],
    headerParameters: [Parameters.accept],
    serializer
};
const getOperationSpec = {
    path: "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.AVS/privateClouds/{privateCloudName}/scriptPackages/{scriptPackageName}/scriptCmdlets/{scriptCmdletName}",
    httpMethod: "GET",
    responses: {
        200: {
            bodyMapper: Mappers.ScriptCmdlet
        },
        default: {
            bodyMapper: Mappers.ErrorResponse
        }
    },
    queryParameters: [Parameters.apiVersion],
    urlParameters: [
        Parameters.$host,
        Parameters.subscriptionId,
        Parameters.resourceGroupName,
        Parameters.privateCloudName,
        Parameters.scriptPackageName,
        Parameters.scriptCmdletName
    ],
    headerParameters: [Parameters.accept],
    serializer
};
const listNextOperationSpec = {
    path: "{nextLink}",
    httpMethod: "GET",
    responses: {
        200: {
            bodyMapper: Mappers.ScriptCmdletsList
        },
        default: {
            bodyMapper: Mappers.ErrorResponse
        }
    },
    urlParameters: [
        Parameters.$host,
        Parameters.nextLink,
        Parameters.subscriptionId,
        Parameters.resourceGroupName,
        Parameters.privateCloudName,
        Parameters.scriptPackageName
    ],
    headerParameters: [Parameters.accept],
    serializer
};
