// @azure/arm-peering/dist-esm/src/operations/prefixes.js
/*
 * Copyright (c) Microsoft Corporation.
 * Licensed under the MIT License.
 *
 * Code generated by Microsoft (R) AutoRest Code Generator.
 * Changes may cause incorrect behavior and will be lost if the code is regenerated.
 */
import { __asyncDelegator, __asyncGenerator, __asyncValues, __await } from "tslib";
import { setContinuationToken } from "../pagingHelper";
import * as coreClient from "@azure/core-client";
import * as Mappers from "../models/mappers";
import * as Parameters from "../models/parameters";
/// <reference lib="esnext.asynciterable" />
/** Class containing Prefixes operations. */
export class PrefixesImpl {
    /**
     * Initialize a new instance of the class Prefixes class.
     * @param client Reference to the service client
     */
    constructor(client) {
        this.client = client;
    }
    /**
     * Lists all prefixes under the given subscription, resource group and peering service.
     * @param resourceGroupName The name of the resource group.
     * @param peeringServiceName The name of the peering service.
     * @param options The options parameters.
     */
    listByPeeringService(resourceGroupName, peeringServiceName, options) {
        const iter = this.listByPeeringServicePagingAll(resourceGroupName, peeringServiceName, options);
        return {
            next() {
                return iter.next();
            },
            [Symbol.asyncIterator]() {
                return this;
            },
            byPage: (settings) => {
                if (settings === null || settings === void 0 ? void 0 : settings.maxPageSize) {
                    throw new Error("maxPageSize is not supported by this operation.");
                }
                return this.listByPeeringServicePagingPage(resourceGroupName, peeringServiceName, options, settings);
            }
        };
    }
    listByPeeringServicePagingPage(resourceGroupName, peeringServiceName, options, settings) {
        return __asyncGenerator(this, arguments, function* listByPeeringServicePagingPage_1() {
            let result;
            let continuationToken = settings === null || settings === void 0 ? void 0 : settings.continuationToken;
            if (!continuationToken) {
                result = yield __await(this._listByPeeringService(resourceGroupName, peeringServiceName, options));
                let page = result.value || [];
                continuationToken = result.nextLink;
                setContinuationToken(page, continuationToken);
                yield yield __await(page);
            }
            while (continuationToken) {
                result = yield __await(this._listByPeeringServiceNext(resourceGroupName, peeringServiceName, continuationToken, options));
                continuationToken = result.nextLink;
                let page = result.value || [];
                setContinuationToken(page, continuationToken);
                yield yield __await(page);
            }
        });
    }
    listByPeeringServicePagingAll(resourceGroupName, peeringServiceName, options) {
        return __asyncGenerator(this, arguments, function* listByPeeringServicePagingAll_1() {
            var e_1, _a;
            try {
                for (var _b = __asyncValues(this.listByPeeringServicePagingPage(resourceGroupName, peeringServiceName, options)), _c; _c = yield __await(_b.next()), !_c.done;) {
                    const page = _c.value;
                    yield __await(yield* __asyncDelegator(__asyncValues(page)));
                }
            }
            catch (e_1_1) { e_1 = { error: e_1_1 }; }
            finally {
                try {
                    if (_c && !_c.done && (_a = _b.return)) yield __await(_a.call(_b));
                }
                finally { if (e_1) throw e_1.error; }
            }
        });
    }
    /**
     * Gets an existing prefix with the specified name under the given subscription, resource group and
     * peering service.
     * @param resourceGroupName The name of the resource group.
     * @param peeringServiceName The name of the peering service.
     * @param prefixName The name of the prefix.
     * @param options The options parameters.
     */
    get(resourceGroupName, peeringServiceName, prefixName, options) {
        return this.client.sendOperationRequest({ resourceGroupName, peeringServiceName, prefixName, options }, getOperationSpec);
    }
    /**
     * Creates a new prefix with the specified name under the given subscription, resource group and
     * peering service.
     * @param resourceGroupName The name of the resource group.
     * @param peeringServiceName The name of the peering service.
     * @param prefixName The name of the prefix.
     * @param peeringServicePrefix The properties needed to create a prefix.
     * @param options The options parameters.
     */
    createOrUpdate(resourceGroupName, peeringServiceName, prefixName, peeringServicePrefix, options) {
        return this.client.sendOperationRequest({
            resourceGroupName,
            peeringServiceName,
            prefixName,
            peeringServicePrefix,
            options
        }, createOrUpdateOperationSpec);
    }
    /**
     * Deletes an existing prefix with the specified name under the given subscription, resource group and
     * peering service.
     * @param resourceGroupName The name of the resource group.
     * @param peeringServiceName The name of the peering service.
     * @param prefixName The name of the prefix.
     * @param options The options parameters.
     */
    delete(resourceGroupName, peeringServiceName, prefixName, options) {
        return this.client.sendOperationRequest({ resourceGroupName, peeringServiceName, prefixName, options }, deleteOperationSpec);
    }
    /**
     * Lists all prefixes under the given subscription, resource group and peering service.
     * @param resourceGroupName The name of the resource group.
     * @param peeringServiceName The name of the peering service.
     * @param options The options parameters.
     */
    _listByPeeringService(resourceGroupName, peeringServiceName, options) {
        return this.client.sendOperationRequest({ resourceGroupName, peeringServiceName, options }, listByPeeringServiceOperationSpec);
    }
    /**
     * ListByPeeringServiceNext
     * @param resourceGroupName The name of the resource group.
     * @param peeringServiceName The name of the peering service.
     * @param nextLink The nextLink from the previous successful call to the ListByPeeringService method.
     * @param options The options parameters.
     */
    _listByPeeringServiceNext(resourceGroupName, peeringServiceName, nextLink, options) {
        return this.client.sendOperationRequest({ resourceGroupName, peeringServiceName, nextLink, options }, listByPeeringServiceNextOperationSpec);
    }
}
// Operation Specifications
const serializer = coreClient.createSerializer(Mappers, /* isXml */ false);
const getOperationSpec = {
    path: "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Peering/peeringServices/{peeringServiceName}/prefixes/{prefixName}",
    httpMethod: "GET",
    responses: {
        200: {
            bodyMapper: Mappers.PeeringServicePrefix
        },
        default: {
            bodyMapper: Mappers.ErrorResponse
        }
    },
    queryParameters: [Parameters.apiVersion, Parameters.expand],
    urlParameters: [
        Parameters.$host,
        Parameters.subscriptionId,
        Parameters.resourceGroupName,
        Parameters.peeringServiceName,
        Parameters.prefixName
    ],
    headerParameters: [Parameters.accept],
    serializer
};
const createOrUpdateOperationSpec = {
    path: "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Peering/peeringServices/{peeringServiceName}/prefixes/{prefixName}",
    httpMethod: "PUT",
    responses: {
        200: {
            bodyMapper: Mappers.PeeringServicePrefix
        },
        201: {
            bodyMapper: Mappers.PeeringServicePrefix
        },
        default: {
            bodyMapper: Mappers.ErrorResponse
        }
    },
    requestBody: Parameters.peeringServicePrefix,
    queryParameters: [Parameters.apiVersion],
    urlParameters: [
        Parameters.$host,
        Parameters.subscriptionId,
        Parameters.resourceGroupName,
        Parameters.peeringServiceName,
        Parameters.prefixName
    ],
    headerParameters: [Parameters.accept, Parameters.contentType],
    mediaType: "json",
    serializer
};
const deleteOperationSpec = {
    path: "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Peering/peeringServices/{peeringServiceName}/prefixes/{prefixName}",
    httpMethod: "DELETE",
    responses: {
        200: {},
        204: {},
        default: {
            bodyMapper: Mappers.ErrorResponse
        }
    },
    queryParameters: [Parameters.apiVersion],
    urlParameters: [
        Parameters.$host,
        Parameters.subscriptionId,
        Parameters.resourceGroupName,
        Parameters.peeringServiceName,
        Parameters.prefixName
    ],
    headerParameters: [Parameters.accept],
    serializer
};
const listByPeeringServiceOperationSpec = {
    path: "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Peering/peeringServices/{peeringServiceName}/prefixes",
    httpMethod: "GET",
    responses: {
        200: {
            bodyMapper: Mappers.PeeringServicePrefixListResult
        },
        default: {
            bodyMapper: Mappers.ErrorResponse
        }
    },
    queryParameters: [Parameters.apiVersion, Parameters.expand],
    urlParameters: [
        Parameters.$host,
        Parameters.subscriptionId,
        Parameters.resourceGroupName,
        Parameters.peeringServiceName
    ],
    headerParameters: [Parameters.accept],
    serializer
};
const listByPeeringServiceNextOperationSpec = {
    path: "{nextLink}",
    httpMethod: "GET",
    responses: {
        200: {
            bodyMapper: Mappers.PeeringServicePrefixListResult
        },
        default: {
            bodyMapper: Mappers.ErrorResponse
        }
    },
    queryParameters: [Parameters.apiVersion, Parameters.expand],
    urlParameters: [
        Parameters.$host,
        Parameters.subscriptionId,
        Parameters.nextLink,
        Parameters.resourceGroupName,
        Parameters.peeringServiceName
    ],
    headerParameters: [Parameters.accept],
    serializer
};
