// @azure/arm-peering/dist-esm/src/operations/registeredAsns.js
/*
 * Copyright (c) Microsoft Corporation.
 * Licensed under the MIT License.
 *
 * Code generated by Microsoft (R) AutoRest Code Generator.
 * Changes may cause incorrect behavior and will be lost if the code is regenerated.
 */
import { __asyncDelegator, __asyncGenerator, __asyncValues, __await } from "tslib";
import { setContinuationToken } from "../pagingHelper";
import * as coreClient from "@azure/core-client";
import * as Mappers from "../models/mappers";
import * as Parameters from "../models/parameters";
/// <reference lib="esnext.asynciterable" />
/** Class containing RegisteredAsns operations. */
export class RegisteredAsnsImpl {
    /**
     * Initialize a new instance of the class RegisteredAsns class.
     * @param client Reference to the service client
     */
    constructor(client) {
        this.client = client;
    }
    /**
     * Lists all registered ASNs under the given subscription, resource group and peering.
     * @param resourceGroupName The name of the resource group.
     * @param peeringName The name of the peering.
     * @param options The options parameters.
     */
    listByPeering(resourceGroupName, peeringName, options) {
        const iter = this.listByPeeringPagingAll(resourceGroupName, peeringName, options);
        return {
            next() {
                return iter.next();
            },
            [Symbol.asyncIterator]() {
                return this;
            },
            byPage: (settings) => {
                if (settings === null || settings === void 0 ? void 0 : settings.maxPageSize) {
                    throw new Error("maxPageSize is not supported by this operation.");
                }
                return this.listByPeeringPagingPage(resourceGroupName, peeringName, options, settings);
            }
        };
    }
    listByPeeringPagingPage(resourceGroupName, peeringName, options, settings) {
        return __asyncGenerator(this, arguments, function* listByPeeringPagingPage_1() {
            let result;
            let continuationToken = settings === null || settings === void 0 ? void 0 : settings.continuationToken;
            if (!continuationToken) {
                result = yield __await(this._listByPeering(resourceGroupName, peeringName, options));
                let page = result.value || [];
                continuationToken = result.nextLink;
                setContinuationToken(page, continuationToken);
                yield yield __await(page);
            }
            while (continuationToken) {
                result = yield __await(this._listByPeeringNext(resourceGroupName, peeringName, continuationToken, options));
                continuationToken = result.nextLink;
                let page = result.value || [];
                setContinuationToken(page, continuationToken);
                yield yield __await(page);
            }
        });
    }
    listByPeeringPagingAll(resourceGroupName, peeringName, options) {
        return __asyncGenerator(this, arguments, function* listByPeeringPagingAll_1() {
            var e_1, _a;
            try {
                for (var _b = __asyncValues(this.listByPeeringPagingPage(resourceGroupName, peeringName, options)), _c; _c = yield __await(_b.next()), !_c.done;) {
                    const page = _c.value;
                    yield __await(yield* __asyncDelegator(__asyncValues(page)));
                }
            }
            catch (e_1_1) { e_1 = { error: e_1_1 }; }
            finally {
                try {
                    if (_c && !_c.done && (_a = _b.return)) yield __await(_a.call(_b));
                }
                finally { if (e_1) throw e_1.error; }
            }
        });
    }
    /**
     * Gets an existing registered ASN with the specified name under the given subscription, resource group
     * and peering.
     * @param resourceGroupName The name of the resource group.
     * @param peeringName The name of the peering.
     * @param registeredAsnName The name of the registered ASN.
     * @param options The options parameters.
     */
    get(resourceGroupName, peeringName, registeredAsnName, options) {
        return this.client.sendOperationRequest({ resourceGroupName, peeringName, registeredAsnName, options }, getOperationSpec);
    }
    /**
     * Creates a new registered ASN with the specified name under the given subscription, resource group
     * and peering.
     * @param resourceGroupName The name of the resource group.
     * @param peeringName The name of the peering.
     * @param registeredAsnName The name of the ASN.
     * @param registeredAsn The properties needed to create a registered ASN.
     * @param options The options parameters.
     */
    createOrUpdate(resourceGroupName, peeringName, registeredAsnName, registeredAsn, options) {
        return this.client.sendOperationRequest({
            resourceGroupName,
            peeringName,
            registeredAsnName,
            registeredAsn,
            options
        }, createOrUpdateOperationSpec);
    }
    /**
     * Deletes an existing registered ASN with the specified name under the given subscription, resource
     * group and peering.
     * @param resourceGroupName The name of the resource group.
     * @param peeringName The name of the peering.
     * @param registeredAsnName The name of the registered ASN.
     * @param options The options parameters.
     */
    delete(resourceGroupName, peeringName, registeredAsnName, options) {
        return this.client.sendOperationRequest({ resourceGroupName, peeringName, registeredAsnName, options }, deleteOperationSpec);
    }
    /**
     * Lists all registered ASNs under the given subscription, resource group and peering.
     * @param resourceGroupName The name of the resource group.
     * @param peeringName The name of the peering.
     * @param options The options parameters.
     */
    _listByPeering(resourceGroupName, peeringName, options) {
        return this.client.sendOperationRequest({ resourceGroupName, peeringName, options }, listByPeeringOperationSpec);
    }
    /**
     * ListByPeeringNext
     * @param resourceGroupName The name of the resource group.
     * @param peeringName The name of the peering.
     * @param nextLink The nextLink from the previous successful call to the ListByPeering method.
     * @param options The options parameters.
     */
    _listByPeeringNext(resourceGroupName, peeringName, nextLink, options) {
        return this.client.sendOperationRequest({ resourceGroupName, peeringName, nextLink, options }, listByPeeringNextOperationSpec);
    }
}
// Operation Specifications
const serializer = coreClient.createSerializer(Mappers, /* isXml */ false);
const getOperationSpec = {
    path: "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Peering/peerings/{peeringName}/registeredAsns/{registeredAsnName}",
    httpMethod: "GET",
    responses: {
        200: {
            bodyMapper: Mappers.PeeringRegisteredAsn
        },
        default: {
            bodyMapper: Mappers.ErrorResponse
        }
    },
    queryParameters: [Parameters.apiVersion],
    urlParameters: [
        Parameters.$host,
        Parameters.subscriptionId,
        Parameters.resourceGroupName,
        Parameters.peeringName,
        Parameters.registeredAsnName
    ],
    headerParameters: [Parameters.accept],
    serializer
};
const createOrUpdateOperationSpec = {
    path: "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Peering/peerings/{peeringName}/registeredAsns/{registeredAsnName}",
    httpMethod: "PUT",
    responses: {
        200: {
            bodyMapper: Mappers.PeeringRegisteredAsn
        },
        201: {
            bodyMapper: Mappers.PeeringRegisteredAsn
        },
        default: {
            bodyMapper: Mappers.ErrorResponse
        }
    },
    requestBody: Parameters.registeredAsn,
    queryParameters: [Parameters.apiVersion],
    urlParameters: [
        Parameters.$host,
        Parameters.subscriptionId,
        Parameters.resourceGroupName,
        Parameters.peeringName,
        Parameters.registeredAsnName
    ],
    headerParameters: [Parameters.accept, Parameters.contentType],
    mediaType: "json",
    serializer
};
const deleteOperationSpec = {
    path: "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Peering/peerings/{peeringName}/registeredAsns/{registeredAsnName}",
    httpMethod: "DELETE",
    responses: {
        200: {},
        204: {},
        default: {
            bodyMapper: Mappers.ErrorResponse
        }
    },
    queryParameters: [Parameters.apiVersion],
    urlParameters: [
        Parameters.$host,
        Parameters.subscriptionId,
        Parameters.resourceGroupName,
        Parameters.peeringName,
        Parameters.registeredAsnName
    ],
    headerParameters: [Parameters.accept],
    serializer
};
const listByPeeringOperationSpec = {
    path: "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Peering/peerings/{peeringName}/registeredAsns",
    httpMethod: "GET",
    responses: {
        200: {
            bodyMapper: Mappers.PeeringRegisteredAsnListResult
        },
        default: {
            bodyMapper: Mappers.ErrorResponse
        }
    },
    queryParameters: [Parameters.apiVersion],
    urlParameters: [
        Parameters.$host,
        Parameters.subscriptionId,
        Parameters.resourceGroupName,
        Parameters.peeringName
    ],
    headerParameters: [Parameters.accept],
    serializer
};
const listByPeeringNextOperationSpec = {
    path: "{nextLink}",
    httpMethod: "GET",
    responses: {
        200: {
            bodyMapper: Mappers.PeeringRegisteredAsnListResult
        },
        default: {
            bodyMapper: Mappers.ErrorResponse
        }
    },
    queryParameters: [Parameters.apiVersion],
    urlParameters: [
        Parameters.$host,
        Parameters.subscriptionId,
        Parameters.nextLink,
        Parameters.resourceGroupName,
        Parameters.peeringName
    ],
    headerParameters: [Parameters.accept],
    serializer
};
