// @azure/arm-devtestlabs/dist-esm/src/models/parameters.js
/*
 * Copyright (c) Microsoft Corporation.
 * Licensed under the MIT License.
 *
 * Code generated by Microsoft (R) AutoRest Code Generator.
 * Changes may cause incorrect behavior and will be lost if the code is regenerated.
 */
import { Lab as LabMapper, LabFragment as LabFragmentMapper, LabVirtualMachineCreationParameter as LabVirtualMachineCreationParameterMapper, ExportResourceUsageParameters as ExportResourceUsageParametersMapper, GenerateUploadUriParameter as GenerateUploadUriParameterMapper, ImportLabVirtualMachineRequest as ImportLabVirtualMachineRequestMapper, Schedule as ScheduleMapper, ScheduleFragment as ScheduleFragmentMapper, RetargetScheduleProperties as RetargetSchedulePropertiesMapper, ArtifactSource as ArtifactSourceMapper, ArtifactSourceFragment as ArtifactSourceFragmentMapper, GenerateArmTemplateRequest as GenerateArmTemplateRequestMapper, LabCost as LabCostMapper, CustomImage as CustomImageMapper, CustomImageFragment as CustomImageFragmentMapper, Formula as FormulaMapper, FormulaFragment as FormulaFragmentMapper, NotificationChannel as NotificationChannelMapper, NotificationChannelFragment as NotificationChannelFragmentMapper, NotifyParameters as NotifyParametersMapper, EvaluatePoliciesRequest as EvaluatePoliciesRequestMapper, Policy as PolicyMapper, PolicyFragment as PolicyFragmentMapper, ServiceRunner as ServiceRunnerMapper, User as UserMapper, UserFragment as UserFragmentMapper, Disk as DiskMapper, DiskFragment as DiskFragmentMapper, AttachDiskProperties as AttachDiskPropertiesMapper, DetachDiskProperties as DetachDiskPropertiesMapper, DtlEnvironment as DtlEnvironmentMapper, DtlEnvironmentFragment as DtlEnvironmentFragmentMapper, Secret as SecretMapper, SecretFragment as SecretFragmentMapper, ServiceFabric as ServiceFabricMapper, ServiceFabricFragment as ServiceFabricFragmentMapper, LabVirtualMachine as LabVirtualMachineMapper, LabVirtualMachineFragment as LabVirtualMachineFragmentMapper, DataDiskProperties as DataDiskPropertiesMapper, ApplyArtifactsRequest as ApplyArtifactsRequestMapper, DetachDataDiskProperties as DetachDataDiskPropertiesMapper, ResizeLabVirtualMachineProperties as ResizeLabVirtualMachinePropertiesMapper, VirtualNetwork as VirtualNetworkMapper, VirtualNetworkFragment as VirtualNetworkFragmentMapper } from "../models/mappers";
export const accept = {
    parameterPath: "accept",
    mapper: {
        defaultValue: "application/json",
        isConstant: true,
        serializedName: "Accept",
        type: {
            name: "String"
        }
    }
};
export const $host = {
    parameterPath: "$host",
    mapper: {
        serializedName: "$host",
        required: true,
        type: {
            name: "String"
        }
    },
    skipEncoding: true
};
export const apiVersion = {
    parameterPath: "apiVersion",
    mapper: {
        defaultValue: "2018-09-15",
        isConstant: true,
        serializedName: "api-version",
        type: {
            name: "String"
        }
    }
};
export const nextLink = {
    parameterPath: "nextLink",
    mapper: {
        serializedName: "nextLink",
        required: true,
        type: {
            name: "String"
        }
    },
    skipEncoding: true
};
export const subscriptionId = {
    parameterPath: "subscriptionId",
    mapper: {
        serializedName: "subscriptionId",
        required: true,
        type: {
            name: "String"
        }
    }
};
export const expand = {
    parameterPath: ["options", "expand"],
    mapper: {
        serializedName: "$expand",
        type: {
            name: "String"
        }
    }
};
export const filter = {
    parameterPath: ["options", "filter"],
    mapper: {
        serializedName: "$filter",
        type: {
            name: "String"
        }
    }
};
export const top = {
    parameterPath: ["options", "top"],
    mapper: {
        serializedName: "$top",
        type: {
            name: "Number"
        }
    }
};
export const orderby = {
    parameterPath: ["options", "orderby"],
    mapper: {
        serializedName: "$orderby",
        type: {
            name: "String"
        }
    }
};
export const resourceGroupName = {
    parameterPath: "resourceGroupName",
    mapper: {
        serializedName: "resourceGroupName",
        required: true,
        type: {
            name: "String"
        }
    }
};
export const name = {
    parameterPath: "name",
    mapper: {
        serializedName: "name",
        required: true,
        type: {
            name: "String"
        }
    }
};
export const contentType = {
    parameterPath: ["options", "contentType"],
    mapper: {
        defaultValue: "application/json",
        isConstant: true,
        serializedName: "Content-Type",
        type: {
            name: "String"
        }
    }
};
export const lab = {
    parameterPath: "lab",
    mapper: LabMapper
};
export const lab1 = {
    parameterPath: "lab",
    mapper: LabFragmentMapper
};
export const labVirtualMachineCreationParameter = {
    parameterPath: "labVirtualMachineCreationParameter",
    mapper: LabVirtualMachineCreationParameterMapper
};
export const exportResourceUsageParameters = {
    parameterPath: "exportResourceUsageParameters",
    mapper: ExportResourceUsageParametersMapper
};
export const generateUploadUriParameter = {
    parameterPath: "generateUploadUriParameter",
    mapper: GenerateUploadUriParameterMapper
};
export const importLabVirtualMachineRequest = {
    parameterPath: "importLabVirtualMachineRequest",
    mapper: ImportLabVirtualMachineRequestMapper
};
export const locationName = {
    parameterPath: "locationName",
    mapper: {
        serializedName: "locationName",
        required: true,
        type: {
            name: "String"
        }
    }
};
export const schedule = {
    parameterPath: "schedule",
    mapper: ScheduleMapper
};
export const schedule1 = {
    parameterPath: "schedule",
    mapper: ScheduleFragmentMapper
};
export const retargetScheduleProperties = {
    parameterPath: "retargetScheduleProperties",
    mapper: RetargetSchedulePropertiesMapper
};
export const labName = {
    parameterPath: "labName",
    mapper: {
        serializedName: "labName",
        required: true,
        type: {
            name: "String"
        }
    }
};
export const artifactSource = {
    parameterPath: "artifactSource",
    mapper: ArtifactSourceMapper
};
export const artifactSource1 = {
    parameterPath: "artifactSource",
    mapper: ArtifactSourceFragmentMapper
};
export const artifactSourceName = {
    parameterPath: "artifactSourceName",
    mapper: {
        serializedName: "artifactSourceName",
        required: true,
        type: {
            name: "String"
        }
    }
};
export const generateArmTemplateRequest = {
    parameterPath: "generateArmTemplateRequest",
    mapper: GenerateArmTemplateRequestMapper
};
export const labCost = {
    parameterPath: "labCost",
    mapper: LabCostMapper
};
export const customImage = {
    parameterPath: "customImage",
    mapper: CustomImageMapper
};
export const customImage1 = {
    parameterPath: "customImage",
    mapper: CustomImageFragmentMapper
};
export const formula = {
    parameterPath: "formula",
    mapper: FormulaMapper
};
export const formula1 = {
    parameterPath: "formula",
    mapper: FormulaFragmentMapper
};
export const notificationChannel = {
    parameterPath: "notificationChannel",
    mapper: NotificationChannelMapper
};
export const notificationChannel1 = {
    parameterPath: "notificationChannel",
    mapper: NotificationChannelFragmentMapper
};
export const notifyParameters = {
    parameterPath: "notifyParameters",
    mapper: NotifyParametersMapper
};
export const evaluatePoliciesRequest = {
    parameterPath: "evaluatePoliciesRequest",
    mapper: EvaluatePoliciesRequestMapper
};
export const policySetName = {
    parameterPath: "policySetName",
    mapper: {
        serializedName: "policySetName",
        required: true,
        type: {
            name: "String"
        }
    }
};
export const policy = {
    parameterPath: "policy",
    mapper: PolicyMapper
};
export const policy1 = {
    parameterPath: "policy",
    mapper: PolicyFragmentMapper
};
export const serviceRunner = {
    parameterPath: "serviceRunner",
    mapper: ServiceRunnerMapper
};
export const user = {
    parameterPath: "user",
    mapper: UserMapper
};
export const user1 = {
    parameterPath: "user",
    mapper: UserFragmentMapper
};
export const userName = {
    parameterPath: "userName",
    mapper: {
        serializedName: "userName",
        required: true,
        type: {
            name: "String"
        }
    }
};
export const disk = {
    parameterPath: "disk",
    mapper: DiskMapper
};
export const disk1 = {
    parameterPath: "disk",
    mapper: DiskFragmentMapper
};
export const attachDiskProperties = {
    parameterPath: "attachDiskProperties",
    mapper: AttachDiskPropertiesMapper
};
export const detachDiskProperties = {
    parameterPath: "detachDiskProperties",
    mapper: DetachDiskPropertiesMapper
};
export const dtlEnvironment = {
    parameterPath: "dtlEnvironment",
    mapper: DtlEnvironmentMapper
};
export const dtlEnvironment1 = {
    parameterPath: "dtlEnvironment",
    mapper: DtlEnvironmentFragmentMapper
};
export const secret = {
    parameterPath: "secret",
    mapper: SecretMapper
};
export const secret1 = {
    parameterPath: "secret",
    mapper: SecretFragmentMapper
};
export const serviceFabric = {
    parameterPath: "serviceFabric",
    mapper: ServiceFabricMapper
};
export const serviceFabric1 = {
    parameterPath: "serviceFabric",
    mapper: ServiceFabricFragmentMapper
};
export const serviceFabricName = {
    parameterPath: "serviceFabricName",
    mapper: {
        serializedName: "serviceFabricName",
        required: true,
        type: {
            name: "String"
        }
    }
};
export const labVirtualMachine = {
    parameterPath: "labVirtualMachine",
    mapper: LabVirtualMachineMapper
};
export const labVirtualMachine1 = {
    parameterPath: "labVirtualMachine",
    mapper: LabVirtualMachineFragmentMapper
};
export const dataDiskProperties = {
    parameterPath: "dataDiskProperties",
    mapper: DataDiskPropertiesMapper
};
export const applyArtifactsRequest = {
    parameterPath: "applyArtifactsRequest",
    mapper: ApplyArtifactsRequestMapper
};
export const detachDataDiskProperties = {
    parameterPath: "detachDataDiskProperties",
    mapper: DetachDataDiskPropertiesMapper
};
export const resizeLabVirtualMachineProperties = {
    parameterPath: "resizeLabVirtualMachineProperties",
    mapper: ResizeLabVirtualMachinePropertiesMapper
};
export const virtualMachineName = {
    parameterPath: "virtualMachineName",
    mapper: {
        serializedName: "virtualMachineName",
        required: true,
        type: {
            name: "String"
        }
    }
};
export const virtualNetwork = {
    parameterPath: "virtualNetwork",
    mapper: VirtualNetworkMapper
};
export const virtualNetwork1 = {
    parameterPath: "virtualNetwork",
    mapper: VirtualNetworkFragmentMapper
};
