// @azure/arm-labservices/dist-esm/src/operations/images.js
/*
 * Copyright (c) Microsoft Corporation.
 * Licensed under the MIT License.
 *
 * Code generated by Microsoft (R) AutoRest Code Generator.
 * Changes may cause incorrect behavior and will be lost if the code is regenerated.
 */
import { __asyncDelegator, __asyncGenerator, __asyncValues, __await } from "tslib";
import { setContinuationToken } from "../pagingHelper";
import * as coreClient from "@azure/core-client";
import * as Mappers from "../models/mappers";
import * as Parameters from "../models/parameters";
/// <reference lib="esnext.asynciterable" />
/** Class containing Images operations. */
export class ImagesImpl {
    /**
     * Initialize a new instance of the class Images class.
     * @param client Reference to the service client
     */
    constructor(client) {
        this.client = client;
    }
    /**
     * Gets all images from galleries attached to a lab plan.
     * @param resourceGroupName The name of the resource group. The name is case insensitive.
     * @param labPlanName The name of the lab plan that uniquely identifies it within containing resource
     *                    group. Used in resource URIs and in UI.
     * @param options The options parameters.
     */
    listByLabPlan(resourceGroupName, labPlanName, options) {
        const iter = this.listByLabPlanPagingAll(resourceGroupName, labPlanName, options);
        return {
            next() {
                return iter.next();
            },
            [Symbol.asyncIterator]() {
                return this;
            },
            byPage: (settings) => {
                if (settings === null || settings === void 0 ? void 0 : settings.maxPageSize) {
                    throw new Error("maxPageSize is not supported by this operation.");
                }
                return this.listByLabPlanPagingPage(resourceGroupName, labPlanName, options, settings);
            }
        };
    }
    listByLabPlanPagingPage(resourceGroupName, labPlanName, options, settings) {
        return __asyncGenerator(this, arguments, function* listByLabPlanPagingPage_1() {
            let result;
            let continuationToken = settings === null || settings === void 0 ? void 0 : settings.continuationToken;
            if (!continuationToken) {
                result = yield __await(this._listByLabPlan(resourceGroupName, labPlanName, options));
                let page = result.value || [];
                continuationToken = result.nextLink;
                setContinuationToken(page, continuationToken);
                yield yield __await(page);
            }
            while (continuationToken) {
                result = yield __await(this._listByLabPlanNext(resourceGroupName, labPlanName, continuationToken, options));
                continuationToken = result.nextLink;
                let page = result.value || [];
                setContinuationToken(page, continuationToken);
                yield yield __await(page);
            }
        });
    }
    listByLabPlanPagingAll(resourceGroupName, labPlanName, options) {
        return __asyncGenerator(this, arguments, function* listByLabPlanPagingAll_1() {
            var e_1, _a;
            try {
                for (var _b = __asyncValues(this.listByLabPlanPagingPage(resourceGroupName, labPlanName, options)), _c; _c = yield __await(_b.next()), !_c.done;) {
                    const page = _c.value;
                    yield __await(yield* __asyncDelegator(__asyncValues(page)));
                }
            }
            catch (e_1_1) { e_1 = { error: e_1_1 }; }
            finally {
                try {
                    if (_c && !_c.done && (_a = _b.return)) yield __await(_a.call(_b));
                }
                finally { if (e_1) throw e_1.error; }
            }
        });
    }
    /**
     * Gets all images from galleries attached to a lab plan.
     * @param resourceGroupName The name of the resource group. The name is case insensitive.
     * @param labPlanName The name of the lab plan that uniquely identifies it within containing resource
     *                    group. Used in resource URIs and in UI.
     * @param options The options parameters.
     */
    _listByLabPlan(resourceGroupName, labPlanName, options) {
        return this.client.sendOperationRequest({ resourceGroupName, labPlanName, options }, listByLabPlanOperationSpec);
    }
    /**
     * Gets an image resource.
     * @param resourceGroupName The name of the resource group. The name is case insensitive.
     * @param labPlanName The name of the lab plan that uniquely identifies it within containing resource
     *                    group. Used in resource URIs and in UI.
     * @param imageName The image name.
     * @param options The options parameters.
     */
    get(resourceGroupName, labPlanName, imageName, options) {
        return this.client.sendOperationRequest({ resourceGroupName, labPlanName, imageName, options }, getOperationSpec);
    }
    /**
     * Updates an image resource via PUT. Creating new resources via PUT will not function.
     * @param resourceGroupName The name of the resource group. The name is case insensitive.
     * @param labPlanName The name of the lab plan that uniquely identifies it within containing resource
     *                    group. Used in resource URIs and in UI.
     * @param imageName The image name.
     * @param body The request body.
     * @param options The options parameters.
     */
    createOrUpdate(resourceGroupName, labPlanName, imageName, body, options) {
        return this.client.sendOperationRequest({ resourceGroupName, labPlanName, imageName, body, options }, createOrUpdateOperationSpec);
    }
    /**
     * Updates an image resource.
     * @param resourceGroupName The name of the resource group. The name is case insensitive.
     * @param labPlanName The name of the lab plan that uniquely identifies it within containing resource
     *                    group. Used in resource URIs and in UI.
     * @param imageName The image name.
     * @param body The request body.
     * @param options The options parameters.
     */
    update(resourceGroupName, labPlanName, imageName, body, options) {
        return this.client.sendOperationRequest({ resourceGroupName, labPlanName, imageName, body, options }, updateOperationSpec);
    }
    /**
     * ListByLabPlanNext
     * @param resourceGroupName The name of the resource group. The name is case insensitive.
     * @param labPlanName The name of the lab plan that uniquely identifies it within containing resource
     *                    group. Used in resource URIs and in UI.
     * @param nextLink The nextLink from the previous successful call to the ListByLabPlan method.
     * @param options The options parameters.
     */
    _listByLabPlanNext(resourceGroupName, labPlanName, nextLink, options) {
        return this.client.sendOperationRequest({ resourceGroupName, labPlanName, nextLink, options }, listByLabPlanNextOperationSpec);
    }
}
// Operation Specifications
const serializer = coreClient.createSerializer(Mappers, /* isXml */ false);
const listByLabPlanOperationSpec = {
    path: "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.LabServices/labPlans/{labPlanName}/images",
    httpMethod: "GET",
    responses: {
        200: {
            bodyMapper: Mappers.PagedImages
        },
        default: {
            bodyMapper: Mappers.ErrorResponse
        }
    },
    queryParameters: [Parameters.apiVersion, Parameters.filter],
    urlParameters: [
        Parameters.$host,
        Parameters.subscriptionId,
        Parameters.resourceGroupName,
        Parameters.labPlanName
    ],
    headerParameters: [Parameters.accept],
    serializer
};
const getOperationSpec = {
    path: "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.LabServices/labPlans/{labPlanName}/images/{imageName}",
    httpMethod: "GET",
    responses: {
        200: {
            bodyMapper: Mappers.Image
        },
        default: {
            bodyMapper: Mappers.ErrorResponse
        }
    },
    queryParameters: [Parameters.apiVersion],
    urlParameters: [
        Parameters.$host,
        Parameters.subscriptionId,
        Parameters.resourceGroupName,
        Parameters.labPlanName,
        Parameters.imageName
    ],
    headerParameters: [Parameters.accept],
    serializer
};
const createOrUpdateOperationSpec = {
    path: "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.LabServices/labPlans/{labPlanName}/images/{imageName}",
    httpMethod: "PUT",
    responses: {
        200: {
            bodyMapper: Mappers.Image
        },
        default: {
            bodyMapper: Mappers.ErrorResponse
        }
    },
    requestBody: Parameters.body,
    queryParameters: [Parameters.apiVersion],
    urlParameters: [
        Parameters.$host,
        Parameters.subscriptionId,
        Parameters.resourceGroupName,
        Parameters.labPlanName,
        Parameters.imageName
    ],
    headerParameters: [Parameters.accept, Parameters.contentType],
    mediaType: "json",
    serializer
};
const updateOperationSpec = {
    path: "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.LabServices/labPlans/{labPlanName}/images/{imageName}",
    httpMethod: "PATCH",
    responses: {
        200: {
            bodyMapper: Mappers.Image
        },
        default: {
            bodyMapper: Mappers.ErrorResponse
        }
    },
    requestBody: Parameters.body1,
    queryParameters: [Parameters.apiVersion],
    urlParameters: [
        Parameters.$host,
        Parameters.subscriptionId,
        Parameters.resourceGroupName,
        Parameters.labPlanName,
        Parameters.imageName
    ],
    headerParameters: [Parameters.accept, Parameters.contentType],
    mediaType: "json",
    serializer
};
const listByLabPlanNextOperationSpec = {
    path: "{nextLink}",
    httpMethod: "GET",
    responses: {
        200: {
            bodyMapper: Mappers.PagedImages
        },
        default: {
            bodyMapper: Mappers.ErrorResponse
        }
    },
    urlParameters: [
        Parameters.$host,
        Parameters.subscriptionId,
        Parameters.resourceGroupName,
        Parameters.labPlanName,
        Parameters.nextLink
    ],
    headerParameters: [Parameters.accept],
    serializer
};
