// @azure/arm-labservices/dist-esm/src/operations/schedules.js
/*
 * Copyright (c) Microsoft Corporation.
 * Licensed under the MIT License.
 *
 * Code generated by Microsoft (R) AutoRest Code Generator.
 * Changes may cause incorrect behavior and will be lost if the code is regenerated.
 */
import { __asyncDelegator, __asyncGenerator, __asyncValues, __await, __awaiter } from "tslib";
import { setContinuationToken } from "../pagingHelper";
import * as coreClient from "@azure/core-client";
import * as Mappers from "../models/mappers";
import * as Parameters from "../models/parameters";
import { LroEngine } from "@azure/core-lro";
import { LroImpl } from "../lroImpl";
/// <reference lib="esnext.asynciterable" />
/** Class containing Schedules operations. */
export class SchedulesImpl {
    /**
     * Initialize a new instance of the class Schedules class.
     * @param client Reference to the service client
     */
    constructor(client) {
        this.client = client;
    }
    /**
     * Returns a list of all schedules for a lab.
     * @param resourceGroupName The name of the resource group. The name is case insensitive.
     * @param labName The name of the lab that uniquely identifies it within containing lab plan. Used in
     *                resource URIs.
     * @param options The options parameters.
     */
    listByLab(resourceGroupName, labName, options) {
        const iter = this.listByLabPagingAll(resourceGroupName, labName, options);
        return {
            next() {
                return iter.next();
            },
            [Symbol.asyncIterator]() {
                return this;
            },
            byPage: (settings) => {
                if (settings === null || settings === void 0 ? void 0 : settings.maxPageSize) {
                    throw new Error("maxPageSize is not supported by this operation.");
                }
                return this.listByLabPagingPage(resourceGroupName, labName, options, settings);
            }
        };
    }
    listByLabPagingPage(resourceGroupName, labName, options, settings) {
        return __asyncGenerator(this, arguments, function* listByLabPagingPage_1() {
            let result;
            let continuationToken = settings === null || settings === void 0 ? void 0 : settings.continuationToken;
            if (!continuationToken) {
                result = yield __await(this._listByLab(resourceGroupName, labName, options));
                let page = result.value || [];
                continuationToken = result.nextLink;
                setContinuationToken(page, continuationToken);
                yield yield __await(page);
            }
            while (continuationToken) {
                result = yield __await(this._listByLabNext(resourceGroupName, labName, continuationToken, options));
                continuationToken = result.nextLink;
                let page = result.value || [];
                setContinuationToken(page, continuationToken);
                yield yield __await(page);
            }
        });
    }
    listByLabPagingAll(resourceGroupName, labName, options) {
        return __asyncGenerator(this, arguments, function* listByLabPagingAll_1() {
            var e_1, _a;
            try {
                for (var _b = __asyncValues(this.listByLabPagingPage(resourceGroupName, labName, options)), _c; _c = yield __await(_b.next()), !_c.done;) {
                    const page = _c.value;
                    yield __await(yield* __asyncDelegator(__asyncValues(page)));
                }
            }
            catch (e_1_1) { e_1 = { error: e_1_1 }; }
            finally {
                try {
                    if (_c && !_c.done && (_a = _b.return)) yield __await(_a.call(_b));
                }
                finally { if (e_1) throw e_1.error; }
            }
        });
    }
    /**
     * Returns a list of all schedules for a lab.
     * @param resourceGroupName The name of the resource group. The name is case insensitive.
     * @param labName The name of the lab that uniquely identifies it within containing lab plan. Used in
     *                resource URIs.
     * @param options The options parameters.
     */
    _listByLab(resourceGroupName, labName, options) {
        return this.client.sendOperationRequest({ resourceGroupName, labName, options }, listByLabOperationSpec);
    }
    /**
     * Returns the properties of a lab Schedule.
     * @param resourceGroupName The name of the resource group. The name is case insensitive.
     * @param labName The name of the lab that uniquely identifies it within containing lab plan. Used in
     *                resource URIs.
     * @param scheduleName The name of the schedule that uniquely identifies it within containing lab. Used
     *                     in resource URIs.
     * @param options The options parameters.
     */
    get(resourceGroupName, labName, scheduleName, options) {
        return this.client.sendOperationRequest({ resourceGroupName, labName, scheduleName, options }, getOperationSpec);
    }
    /**
     * Operation to create or update a lab schedule.
     * @param resourceGroupName The name of the resource group. The name is case insensitive.
     * @param labName The name of the lab that uniquely identifies it within containing lab plan. Used in
     *                resource URIs.
     * @param scheduleName The name of the schedule that uniquely identifies it within containing lab. Used
     *                     in resource URIs.
     * @param body The request body.
     * @param options The options parameters.
     */
    createOrUpdate(resourceGroupName, labName, scheduleName, body, options) {
        return this.client.sendOperationRequest({ resourceGroupName, labName, scheduleName, body, options }, createOrUpdateOperationSpec);
    }
    /**
     * Operation to update a lab schedule.
     * @param resourceGroupName The name of the resource group. The name is case insensitive.
     * @param labName The name of the lab that uniquely identifies it within containing lab plan. Used in
     *                resource URIs.
     * @param scheduleName The name of the schedule that uniquely identifies it within containing lab. Used
     *                     in resource URIs.
     * @param body The request body.
     * @param options The options parameters.
     */
    update(resourceGroupName, labName, scheduleName, body, options) {
        return this.client.sendOperationRequest({ resourceGroupName, labName, scheduleName, body, options }, updateOperationSpec);
    }
    /**
     * Operation to delete a schedule resource.
     * @param resourceGroupName The name of the resource group. The name is case insensitive.
     * @param labName The name of the lab that uniquely identifies it within containing lab plan. Used in
     *                resource URIs.
     * @param scheduleName The name of the schedule that uniquely identifies it within containing lab. Used
     *                     in resource URIs.
     * @param options The options parameters.
     */
    beginDelete(resourceGroupName, labName, scheduleName, options) {
        return __awaiter(this, void 0, void 0, function* () {
            const directSendOperation = (args, spec) => __awaiter(this, void 0, void 0, function* () {
                return this.client.sendOperationRequest(args, spec);
            });
            const sendOperation = (args, spec) => __awaiter(this, void 0, void 0, function* () {
                var _a;
                let currentRawResponse = undefined;
                const providedCallback = (_a = args.options) === null || _a === void 0 ? void 0 : _a.onResponse;
                const callback = (rawResponse, flatResponse) => {
                    currentRawResponse = rawResponse;
                    providedCallback === null || providedCallback === void 0 ? void 0 : providedCallback(rawResponse, flatResponse);
                };
                const updatedArgs = Object.assign(Object.assign({}, args), { options: Object.assign(Object.assign({}, args.options), { onResponse: callback }) });
                const flatResponse = yield directSendOperation(updatedArgs, spec);
                return {
                    flatResponse,
                    rawResponse: {
                        statusCode: currentRawResponse.status,
                        body: currentRawResponse.parsedBody,
                        headers: currentRawResponse.headers.toJSON()
                    }
                };
            });
            const lro = new LroImpl(sendOperation, { resourceGroupName, labName, scheduleName, options }, deleteOperationSpec);
            const poller = new LroEngine(lro, {
                resumeFrom: options === null || options === void 0 ? void 0 : options.resumeFrom,
                intervalInMs: options === null || options === void 0 ? void 0 : options.updateIntervalInMs,
                lroResourceLocationConfig: "location"
            });
            yield poller.poll();
            return poller;
        });
    }
    /**
     * Operation to delete a schedule resource.
     * @param resourceGroupName The name of the resource group. The name is case insensitive.
     * @param labName The name of the lab that uniquely identifies it within containing lab plan. Used in
     *                resource URIs.
     * @param scheduleName The name of the schedule that uniquely identifies it within containing lab. Used
     *                     in resource URIs.
     * @param options The options parameters.
     */
    beginDeleteAndWait(resourceGroupName, labName, scheduleName, options) {
        return __awaiter(this, void 0, void 0, function* () {
            const poller = yield this.beginDelete(resourceGroupName, labName, scheduleName, options);
            return poller.pollUntilDone();
        });
    }
    /**
     * ListByLabNext
     * @param resourceGroupName The name of the resource group. The name is case insensitive.
     * @param labName The name of the lab that uniquely identifies it within containing lab plan. Used in
     *                resource URIs.
     * @param nextLink The nextLink from the previous successful call to the ListByLab method.
     * @param options The options parameters.
     */
    _listByLabNext(resourceGroupName, labName, nextLink, options) {
        return this.client.sendOperationRequest({ resourceGroupName, labName, nextLink, options }, listByLabNextOperationSpec);
    }
}
// Operation Specifications
const serializer = coreClient.createSerializer(Mappers, /* isXml */ false);
const listByLabOperationSpec = {
    path: "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.LabServices/labs/{labName}/schedules",
    httpMethod: "GET",
    responses: {
        200: {
            bodyMapper: Mappers.PagedSchedules
        },
        default: {
            bodyMapper: Mappers.ErrorResponse
        }
    },
    queryParameters: [Parameters.apiVersion, Parameters.filter],
    urlParameters: [
        Parameters.$host,
        Parameters.subscriptionId,
        Parameters.resourceGroupName,
        Parameters.labName
    ],
    headerParameters: [Parameters.accept],
    serializer
};
const getOperationSpec = {
    path: "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.LabServices/labs/{labName}/schedules/{scheduleName}",
    httpMethod: "GET",
    responses: {
        200: {
            bodyMapper: Mappers.Schedule
        },
        default: {
            bodyMapper: Mappers.ErrorResponse
        }
    },
    queryParameters: [Parameters.apiVersion],
    urlParameters: [
        Parameters.$host,
        Parameters.subscriptionId,
        Parameters.resourceGroupName,
        Parameters.labName,
        Parameters.scheduleName
    ],
    headerParameters: [Parameters.accept],
    serializer
};
const createOrUpdateOperationSpec = {
    path: "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.LabServices/labs/{labName}/schedules/{scheduleName}",
    httpMethod: "PUT",
    responses: {
        200: {
            bodyMapper: Mappers.Schedule
        },
        201: {
            bodyMapper: Mappers.Schedule
        },
        default: {
            bodyMapper: Mappers.ErrorResponse
        }
    },
    requestBody: Parameters.body7,
    queryParameters: [Parameters.apiVersion],
    urlParameters: [
        Parameters.$host,
        Parameters.subscriptionId,
        Parameters.resourceGroupName,
        Parameters.labName,
        Parameters.scheduleName
    ],
    headerParameters: [Parameters.accept, Parameters.contentType],
    mediaType: "json",
    serializer
};
const updateOperationSpec = {
    path: "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.LabServices/labs/{labName}/schedules/{scheduleName}",
    httpMethod: "PATCH",
    responses: {
        200: {
            bodyMapper: Mappers.Schedule
        },
        default: {
            bodyMapper: Mappers.ErrorResponse
        }
    },
    requestBody: Parameters.body8,
    queryParameters: [Parameters.apiVersion],
    urlParameters: [
        Parameters.$host,
        Parameters.subscriptionId,
        Parameters.resourceGroupName,
        Parameters.labName,
        Parameters.scheduleName
    ],
    headerParameters: [Parameters.accept, Parameters.contentType],
    mediaType: "json",
    serializer
};
const deleteOperationSpec = {
    path: "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.LabServices/labs/{labName}/schedules/{scheduleName}",
    httpMethod: "DELETE",
    responses: {
        200: {},
        201: {},
        202: {},
        204: {},
        default: {
            bodyMapper: Mappers.ErrorResponse
        }
    },
    queryParameters: [Parameters.apiVersion],
    urlParameters: [
        Parameters.$host,
        Parameters.subscriptionId,
        Parameters.resourceGroupName,
        Parameters.labName,
        Parameters.scheduleName
    ],
    headerParameters: [Parameters.accept],
    serializer
};
const listByLabNextOperationSpec = {
    path: "{nextLink}",
    httpMethod: "GET",
    responses: {
        200: {
            bodyMapper: Mappers.PagedSchedules
        },
        default: {
            bodyMapper: Mappers.ErrorResponse
        }
    },
    urlParameters: [
        Parameters.$host,
        Parameters.subscriptionId,
        Parameters.resourceGroupName,
        Parameters.nextLink,
        Parameters.labName
    ],
    headerParameters: [Parameters.accept],
    serializer
};
