// @azure/arm-managementgroups/dist-esm/src/models/index.js
/*
 * Copyright (c) Microsoft Corporation.
 * Licensed under the MIT License.
 *
 * Code generated by Microsoft (R) AutoRest Code Generator.
 * Changes may cause incorrect behavior and will be lost if the code is regenerated.
 */
/** Known values of {@link Enum0} that the service accepts. */
export var KnownEnum0;
(function (KnownEnum0) {
    /** Children */
    KnownEnum0["Children"] = "children";
    /** Path */
    KnownEnum0["Path"] = "path";
    /** Ancestors */
    KnownEnum0["Ancestors"] = "ancestors";
})(KnownEnum0 || (KnownEnum0 = {}));
/** Known values of {@link ManagementGroupChildType} that the service accepts. */
export var KnownManagementGroupChildType;
(function (KnownManagementGroupChildType) {
    /** MicrosoftManagementManagementGroups */
    KnownManagementGroupChildType["MicrosoftManagementManagementGroups"] = "Microsoft.Management/managementGroups";
    /** Subscriptions */
    KnownManagementGroupChildType["Subscriptions"] = "/subscriptions";
})(KnownManagementGroupChildType || (KnownManagementGroupChildType = {}));
/** Known values of {@link Enum2} that the service accepts. */
export var KnownEnum2;
(function (KnownEnum2) {
    /** AllowedParents */
    KnownEnum2["AllowedParents"] = "AllowedParents";
    /** AllowedChildren */
    KnownEnum2["AllowedChildren"] = "AllowedChildren";
    /** ParentAndFirstLevelChildren */
    KnownEnum2["ParentAndFirstLevelChildren"] = "ParentAndFirstLevelChildren";
    /** ParentOnly */
    KnownEnum2["ParentOnly"] = "ParentOnly";
    /** ChildrenOnly */
    KnownEnum2["ChildrenOnly"] = "ChildrenOnly";
})(KnownEnum2 || (KnownEnum2 = {}));
/** Known values of {@link Enum3} that the service accepts. */
export var KnownEnum3;
(function (KnownEnum3) {
    /** FullHierarchy */
    KnownEnum3["FullHierarchy"] = "FullHierarchy";
    /** GroupsOnly */
    KnownEnum3["GroupsOnly"] = "GroupsOnly";
    /** SubscriptionsOnly */
    KnownEnum3["SubscriptionsOnly"] = "SubscriptionsOnly";
    /** Audit */
    KnownEnum3["Audit"] = "Audit";
})(KnownEnum3 || (KnownEnum3 = {}));
/** Known values of {@link Permissions} that the service accepts. */
export var KnownPermissions;
(function (KnownPermissions) {
    /** Noaccess */
    KnownPermissions["Noaccess"] = "noaccess";
    /** View */
    KnownPermissions["View"] = "view";
    /** Edit */
    KnownPermissions["Edit"] = "edit";
    /** Delete */
    KnownPermissions["Delete"] = "delete";
})(KnownPermissions || (KnownPermissions = {}));
