// @azure/arm-dnsresolver/dist-esm/src/models/index.js
/*
 * Copyright (c) Microsoft Corporation.
 * Licensed under the MIT License.
 *
 * Code generated by Microsoft (R) AutoRest Code Generator.
 * Changes may cause incorrect behavior and will be lost if the code is regenerated.
 */
/** Known values of {@link DnsResolverState} that the service accepts. */
export var KnownDnsResolverState;
(function (KnownDnsResolverState) {
    /** Connected */
    KnownDnsResolverState["Connected"] = "Connected";
    /** Disconnected */
    KnownDnsResolverState["Disconnected"] = "Disconnected";
})(KnownDnsResolverState || (KnownDnsResolverState = {}));
/** Known values of {@link ProvisioningState} that the service accepts. */
export var KnownProvisioningState;
(function (KnownProvisioningState) {
    /** Creating */
    KnownProvisioningState["Creating"] = "Creating";
    /** Updating */
    KnownProvisioningState["Updating"] = "Updating";
    /** Deleting */
    KnownProvisioningState["Deleting"] = "Deleting";
    /** Succeeded */
    KnownProvisioningState["Succeeded"] = "Succeeded";
    /** Failed */
    KnownProvisioningState["Failed"] = "Failed";
    /** Canceled */
    KnownProvisioningState["Canceled"] = "Canceled";
})(KnownProvisioningState || (KnownProvisioningState = {}));
/** Known values of {@link CreatedByType} that the service accepts. */
export var KnownCreatedByType;
(function (KnownCreatedByType) {
    /** User */
    KnownCreatedByType["User"] = "User";
    /** Application */
    KnownCreatedByType["Application"] = "Application";
    /** ManagedIdentity */
    KnownCreatedByType["ManagedIdentity"] = "ManagedIdentity";
    /** Key */
    KnownCreatedByType["Key"] = "Key";
})(KnownCreatedByType || (KnownCreatedByType = {}));
/** Known values of {@link IpAllocationMethod} that the service accepts. */
export var KnownIpAllocationMethod;
(function (KnownIpAllocationMethod) {
    /** Static */
    KnownIpAllocationMethod["Static"] = "Static";
    /** Dynamic */
    KnownIpAllocationMethod["Dynamic"] = "Dynamic";
})(KnownIpAllocationMethod || (KnownIpAllocationMethod = {}));
/** Known values of {@link ForwardingRuleState} that the service accepts. */
export var KnownForwardingRuleState;
(function (KnownForwardingRuleState) {
    /** Enabled */
    KnownForwardingRuleState["Enabled"] = "Enabled";
    /** Disabled */
    KnownForwardingRuleState["Disabled"] = "Disabled";
})(KnownForwardingRuleState || (KnownForwardingRuleState = {}));
