// @azure/arm-servicefabricmesh/esm/models/mappers.js
/*
 * Copyright (c) Microsoft Corporation. All rights reserved.
 * Licensed under the MIT License. See License.txt in the project root for
 * license information.
 *
 * Code generated by Microsoft (R) AutoRest Code Generator.
 * Changes may cause incorrect behavior and will be lost if the code is
 * regenerated.
 */
import { __assign } from "tslib";
import { CloudErrorMapper, BaseResourceMapper } from "@azure/ms-rest-azure-js";
export var CloudError = CloudErrorMapper;
export var BaseResource = BaseResourceMapper;
export var AvailableOperationDisplay = {
    serializedName: "AvailableOperationDisplay",
    type: {
        name: "Composite",
        className: "AvailableOperationDisplay",
        modelProperties: {
            provider: {
                serializedName: "provider",
                type: {
                    name: "String"
                }
            },
            resource: {
                serializedName: "resource",
                type: {
                    name: "String"
                }
            },
            operation: {
                serializedName: "operation",
                type: {
                    name: "String"
                }
            },
            description: {
                serializedName: "description",
                type: {
                    name: "String"
                }
            }
        }
    }
};
export var ErrorDetailsModel = {
    serializedName: "ErrorDetailsModel",
    type: {
        name: "Composite",
        className: "ErrorDetailsModel",
        modelProperties: {
            code: {
                required: true,
                serializedName: "code",
                type: {
                    name: "String"
                }
            },
            message: {
                required: true,
                serializedName: "message",
                type: {
                    name: "String"
                }
            }
        }
    }
};
export var ErrorErrorModel = {
    serializedName: "ErrorErrorModel",
    type: {
        name: "Composite",
        className: "ErrorErrorModel",
        modelProperties: {
            code: {
                required: true,
                serializedName: "code",
                type: {
                    name: "String"
                }
            },
            message: {
                serializedName: "message",
                type: {
                    name: "String"
                }
            },
            innerError: {
                serializedName: "innerError",
                type: {
                    name: "String"
                }
            },
            details: {
                serializedName: "details",
                type: {
                    name: "Sequence",
                    element: {
                        type: {
                            name: "Composite",
                            className: "ErrorDetailsModel"
                        }
                    }
                }
            }
        }
    }
};
export var ErrorModel = {
    serializedName: "ErrorModel",
    type: {
        name: "Composite",
        className: "ErrorModel",
        modelProperties: {
            error: {
                required: true,
                serializedName: "error",
                type: {
                    name: "Composite",
                    className: "ErrorErrorModel"
                }
            }
        }
    }
};
export var OperationResult = {
    serializedName: "OperationResult",
    type: {
        name: "Composite",
        className: "OperationResult",
        modelProperties: {
            name: {
                serializedName: "name",
                type: {
                    name: "String"
                }
            },
            display: {
                serializedName: "display",
                type: {
                    name: "Composite",
                    className: "AvailableOperationDisplay"
                }
            },
            origin: {
                serializedName: "origin",
                type: {
                    name: "String"
                }
            },
            nextLink: {
                serializedName: "nextLink",
                type: {
                    name: "String"
                }
            }
        }
    }
};
export var ProvisionedResourceProperties = {
    serializedName: "ProvisionedResourceProperties",
    type: {
        name: "Composite",
        className: "ProvisionedResourceProperties",
        modelProperties: {
            provisioningState: {
                readOnly: true,
                serializedName: "provisioningState",
                type: {
                    name: "String"
                }
            }
        }
    }
};
export var Resource = {
    serializedName: "Resource",
    type: {
        name: "Composite",
        className: "Resource",
        modelProperties: {
            id: {
                readOnly: true,
                serializedName: "id",
                type: {
                    name: "String"
                }
            },
            name: {
                readOnly: true,
                serializedName: "name",
                type: {
                    name: "String"
                }
            },
            type: {
                readOnly: true,
                serializedName: "type",
                type: {
                    name: "String"
                }
            }
        }
    }
};
export var ProxyResource = {
    serializedName: "ProxyResource",
    type: {
        name: "Composite",
        className: "ProxyResource",
        modelProperties: __assign({}, Resource.type.modelProperties)
    }
};
export var ManagedProxyResource = {
    serializedName: "ManagedProxyResource",
    type: {
        name: "Composite",
        className: "ManagedProxyResource",
        modelProperties: {
            id: {
                readOnly: true,
                serializedName: "id",
                type: {
                    name: "String"
                }
            },
            name: {
                serializedName: "name",
                type: {
                    name: "String"
                }
            },
            type: {
                readOnly: true,
                serializedName: "type",
                type: {
                    name: "String"
                }
            }
        }
    }
};
export var TrackedResource = {
    serializedName: "TrackedResource",
    type: {
        name: "Composite",
        className: "TrackedResource",
        modelProperties: __assign(__assign({}, Resource.type.modelProperties), { tags: {
                serializedName: "tags",
                type: {
                    name: "Dictionary",
                    value: {
                        type: {
                            name: "String"
                        }
                    }
                }
            }, location: {
                required: true,
                serializedName: "location",
                type: {
                    name: "String"
                }
            } })
    }
};
export var SecretResourcePropertiesBase = {
    serializedName: "SecretResourcePropertiesBase",
    type: {
        name: "Composite",
        polymorphicDiscriminator: {
            serializedName: "kind",
            clientName: "kind"
        },
        uberParent: "SecretResourcePropertiesBase",
        className: "SecretResourcePropertiesBase",
        modelProperties: __assign(__assign({}, ProvisionedResourceProperties.type.modelProperties), { kind: {
                required: true,
                serializedName: "kind",
                type: {
                    name: "String"
                }
            } })
    }
};
export var SecretResourceProperties = {
    serializedName: "SecretResourceProperties",
    type: {
        name: "Composite",
        className: "SecretResourceProperties",
        modelProperties: __assign(__assign({}, SecretResourcePropertiesBase.type.modelProperties), { description: {
                serializedName: "description",
                type: {
                    name: "String"
                }
            }, status: {
                readOnly: true,
                serializedName: "status",
                type: {
                    name: "String"
                }
            }, statusDetails: {
                readOnly: true,
                serializedName: "statusDetails",
                type: {
                    name: "String"
                }
            }, contentType: {
                serializedName: "contentType",
                type: {
                    name: "String"
                }
            } })
    }
};
export var InlinedValueSecretResourceProperties = {
    serializedName: "inlinedValue",
    type: {
        name: "Composite",
        className: "InlinedValueSecretResourceProperties",
        modelProperties: __assign({}, SecretResourceProperties.type.modelProperties)
    }
};
export var SecretResourceDescription = {
    serializedName: "SecretResourceDescription",
    type: {
        name: "Composite",
        className: "SecretResourceDescription",
        modelProperties: __assign(__assign({}, TrackedResource.type.modelProperties), { properties: {
                required: true,
                serializedName: "properties",
                type: {
                    name: "Composite",
                    className: "SecretResourceProperties"
                }
            } })
    }
};
export var SecretValue = {
    serializedName: "SecretValue",
    type: {
        name: "Composite",
        className: "SecretValue",
        modelProperties: {
            value: {
                serializedName: "value",
                type: {
                    name: "String"
                }
            }
        }
    }
};
export var SecretValueProperties = {
    serializedName: "SecretValueProperties",
    type: {
        name: "Composite",
        className: "SecretValueProperties",
        modelProperties: {
            value: {
                serializedName: "value",
                type: {
                    name: "String"
                }
            }
        }
    }
};
export var SecretValueResourceDescription = {
    serializedName: "SecretValueResourceDescription",
    type: {
        name: "Composite",
        className: "SecretValueResourceDescription",
        modelProperties: __assign(__assign({}, TrackedResource.type.modelProperties), { provisioningState: {
                readOnly: true,
                serializedName: "properties.provisioningState",
                type: {
                    name: "String"
                }
            }, value: {
                serializedName: "properties.value",
                type: {
                    name: "String"
                }
            } })
    }
};
export var VolumeProviderParametersAzureFile = {
    serializedName: "VolumeProviderParametersAzureFile",
    type: {
        name: "Composite",
        className: "VolumeProviderParametersAzureFile",
        modelProperties: {
            accountName: {
                required: true,
                serializedName: "accountName",
                type: {
                    name: "String"
                }
            },
            accountKey: {
                serializedName: "accountKey",
                type: {
                    name: "String"
                }
            },
            shareName: {
                required: true,
                serializedName: "shareName",
                type: {
                    name: "String"
                }
            }
        }
    }
};
export var VolumeProperties = {
    serializedName: "VolumeProperties",
    type: {
        name: "Composite",
        className: "VolumeProperties",
        modelProperties: {
            description: {
                serializedName: "description",
                type: {
                    name: "String"
                }
            },
            status: {
                readOnly: true,
                serializedName: "status",
                type: {
                    name: "String"
                }
            },
            statusDetails: {
                readOnly: true,
                serializedName: "statusDetails",
                type: {
                    name: "String"
                }
            },
            provider: {
                required: true,
                isConstant: true,
                serializedName: "provider",
                defaultValue: 'SFAzureFile',
                type: {
                    name: "String"
                }
            },
            azureFileParameters: {
                serializedName: "azureFileParameters",
                type: {
                    name: "Composite",
                    className: "VolumeProviderParametersAzureFile"
                }
            }
        }
    }
};
export var VolumeReference = {
    serializedName: "VolumeReference",
    type: {
        name: "Composite",
        className: "VolumeReference",
        modelProperties: {
            name: {
                required: true,
                serializedName: "name",
                type: {
                    name: "String"
                }
            },
            readOnly: {
                serializedName: "readOnly",
                type: {
                    name: "Boolean"
                }
            },
            destinationPath: {
                required: true,
                serializedName: "destinationPath",
                type: {
                    name: "String"
                }
            }
        }
    }
};
export var ApplicationScopedVolumeCreationParameters = {
    serializedName: "ApplicationScopedVolumeCreationParameters",
    type: {
        name: "Composite",
        polymorphicDiscriminator: {
            serializedName: "kind",
            clientName: "kind"
        },
        uberParent: "ApplicationScopedVolumeCreationParameters",
        className: "ApplicationScopedVolumeCreationParameters",
        modelProperties: {
            description: {
                serializedName: "description",
                type: {
                    name: "String"
                }
            },
            kind: {
                required: true,
                serializedName: "kind",
                type: {
                    name: "String"
                }
            }
        }
    }
};
export var ApplicationScopedVolume = {
    serializedName: "ApplicationScopedVolume",
    type: {
        name: "Composite",
        className: "ApplicationScopedVolume",
        modelProperties: __assign(__assign({}, VolumeReference.type.modelProperties), { creationParameters: {
                required: true,
                serializedName: "creationParameters",
                type: {
                    name: "Composite",
                    className: "ApplicationScopedVolumeCreationParameters"
                }
            } })
    }
};
export var ApplicationScopedVolumeCreationParametersServiceFabricVolumeDisk = {
    serializedName: "ServiceFabricVolumeDisk",
    type: {
        name: "Composite",
        polymorphicDiscriminator: ApplicationScopedVolumeCreationParameters.type.polymorphicDiscriminator,
        uberParent: "ApplicationScopedVolumeCreationParameters",
        className: "ApplicationScopedVolumeCreationParametersServiceFabricVolumeDisk",
        modelProperties: __assign(__assign({}, ApplicationScopedVolumeCreationParameters.type.modelProperties), { sizeDisk: {
                required: true,
                serializedName: "sizeDisk",
                type: {
                    name: "String"
                }
            } })
    }
};
export var VolumeResourceDescription = {
    serializedName: "VolumeResourceDescription",
    type: {
        name: "Composite",
        className: "VolumeResourceDescription",
        modelProperties: __assign(__assign({}, TrackedResource.type.modelProperties), { provisioningState: {
                readOnly: true,
                serializedName: "properties.provisioningState",
                type: {
                    name: "String"
                }
            }, description: {
                serializedName: "properties.description",
                type: {
                    name: "String"
                }
            }, status: {
                readOnly: true,
                serializedName: "properties.status",
                type: {
                    name: "String"
                }
            }, statusDetails: {
                readOnly: true,
                serializedName: "properties.statusDetails",
                type: {
                    name: "String"
                }
            }, provider: {
                required: true,
                isConstant: true,
                serializedName: "properties.provider",
                defaultValue: 'SFAzureFile',
                type: {
                    name: "String"
                }
            }, azureFileParameters: {
                serializedName: "properties.azureFileParameters",
                type: {
                    name: "Composite",
                    className: "VolumeProviderParametersAzureFile"
                }
            } })
    }
};
export var NetworkResourcePropertiesBase = {
    serializedName: "NetworkResourcePropertiesBase",
    type: {
        name: "Composite",
        polymorphicDiscriminator: {
            serializedName: "kind",
            clientName: "kind"
        },
        uberParent: "NetworkResourcePropertiesBase",
        className: "NetworkResourcePropertiesBase",
        modelProperties: __assign(__assign({}, ProvisionedResourceProperties.type.modelProperties), { kind: {
                required: true,
                serializedName: "kind",
                type: {
                    name: "String"
                }
            } })
    }
};
export var NetworkResourceProperties = {
    serializedName: "NetworkResourceProperties",
    type: {
        name: "Composite",
        className: "NetworkResourceProperties",
        modelProperties: __assign(__assign({}, NetworkResourcePropertiesBase.type.modelProperties), { description: {
                serializedName: "description",
                type: {
                    name: "String"
                }
            }, status: {
                readOnly: true,
                serializedName: "status",
                type: {
                    name: "String"
                }
            }, statusDetails: {
                readOnly: true,
                serializedName: "statusDetails",
                type: {
                    name: "String"
                }
            } })
    }
};
export var LocalNetworkResourceProperties = {
    serializedName: "Local",
    type: {
        name: "Composite",
        className: "LocalNetworkResourceProperties",
        modelProperties: __assign(__assign({}, NetworkResourceProperties.type.modelProperties), { networkAddressPrefix: {
                serializedName: "networkAddressPrefix",
                type: {
                    name: "String"
                }
            } })
    }
};
export var EndpointRef = {
    serializedName: "EndpointRef",
    type: {
        name: "Composite",
        className: "EndpointRef",
        modelProperties: {
            name: {
                serializedName: "name",
                type: {
                    name: "String"
                }
            }
        }
    }
};
export var NetworkRef = {
    serializedName: "NetworkRef",
    type: {
        name: "Composite",
        className: "NetworkRef",
        modelProperties: {
            name: {
                serializedName: "name",
                type: {
                    name: "String"
                }
            },
            endpointRefs: {
                serializedName: "endpointRefs",
                type: {
                    name: "Sequence",
                    element: {
                        type: {
                            name: "Composite",
                            className: "EndpointRef"
                        }
                    }
                }
            }
        }
    }
};
export var NetworkResourceDescription = {
    serializedName: "NetworkResourceDescription",
    type: {
        name: "Composite",
        className: "NetworkResourceDescription",
        modelProperties: __assign(__assign({}, TrackedResource.type.modelProperties), { properties: {
                required: true,
                serializedName: "properties",
                type: {
                    name: "Composite",
                    className: "NetworkResourceProperties"
                }
            } })
    }
};
export var GatewayDestination = {
    serializedName: "GatewayDestination",
    type: {
        name: "Composite",
        className: "GatewayDestination",
        modelProperties: {
            applicationName: {
                required: true,
                serializedName: "applicationName",
                type: {
                    name: "String"
                }
            },
            serviceName: {
                required: true,
                serializedName: "serviceName",
                type: {
                    name: "String"
                }
            },
            endpointName: {
                required: true,
                serializedName: "endpointName",
                type: {
                    name: "String"
                }
            }
        }
    }
};
export var TcpConfig = {
    serializedName: "TcpConfig",
    type: {
        name: "Composite",
        className: "TcpConfig",
        modelProperties: {
            name: {
                required: true,
                serializedName: "name",
                type: {
                    name: "String"
                }
            },
            port: {
                required: true,
                serializedName: "port",
                type: {
                    name: "Number"
                }
            },
            destination: {
                required: true,
                serializedName: "destination",
                type: {
                    name: "Composite",
                    className: "GatewayDestination"
                }
            }
        }
    }
};
export var HttpRouteMatchPath = {
    serializedName: "HttpRouteMatchPath",
    type: {
        name: "Composite",
        className: "HttpRouteMatchPath",
        modelProperties: {
            value: {
                required: true,
                serializedName: "value",
                type: {
                    name: "String"
                }
            },
            rewrite: {
                serializedName: "rewrite",
                type: {
                    name: "String"
                }
            },
            type: {
                required: true,
                isConstant: true,
                serializedName: "type",
                defaultValue: 'prefix',
                type: {
                    name: "String"
                }
            }
        }
    }
};
export var HttpRouteMatchHeader = {
    serializedName: "HttpRouteMatchHeader",
    type: {
        name: "Composite",
        className: "HttpRouteMatchHeader",
        modelProperties: {
            name: {
                required: true,
                serializedName: "name",
                type: {
                    name: "String"
                }
            },
            value: {
                serializedName: "value",
                type: {
                    name: "String"
                }
            },
            type: {
                serializedName: "type",
                type: {
                    name: "String"
                }
            }
        }
    }
};
export var HttpRouteMatchRule = {
    serializedName: "HttpRouteMatchRule",
    type: {
        name: "Composite",
        className: "HttpRouteMatchRule",
        modelProperties: {
            path: {
                required: true,
                serializedName: "path",
                defaultValue: {},
                type: {
                    name: "Composite",
                    className: "HttpRouteMatchPath"
                }
            },
            headers: {
                serializedName: "headers",
                type: {
                    name: "Sequence",
                    element: {
                        type: {
                            name: "Composite",
                            className: "HttpRouteMatchHeader"
                        }
                    }
                }
            }
        }
    }
};
export var HttpRouteConfig = {
    serializedName: "HttpRouteConfig",
    type: {
        name: "Composite",
        className: "HttpRouteConfig",
        modelProperties: {
            name: {
                required: true,
                serializedName: "name",
                type: {
                    name: "String"
                }
            },
            match: {
                required: true,
                serializedName: "match",
                defaultValue: {},
                type: {
                    name: "Composite",
                    className: "HttpRouteMatchRule"
                }
            },
            destination: {
                required: true,
                serializedName: "destination",
                type: {
                    name: "Composite",
                    className: "GatewayDestination"
                }
            }
        }
    }
};
export var HttpHostConfig = {
    serializedName: "HttpHostConfig",
    type: {
        name: "Composite",
        className: "HttpHostConfig",
        modelProperties: {
            name: {
                required: true,
                serializedName: "name",
                type: {
                    name: "String"
                }
            },
            routes: {
                required: true,
                serializedName: "routes",
                type: {
                    name: "Sequence",
                    element: {
                        type: {
                            name: "Composite",
                            className: "HttpRouteConfig"
                        }
                    }
                }
            }
        }
    }
};
export var HttpConfig = {
    serializedName: "HttpConfig",
    type: {
        name: "Composite",
        className: "HttpConfig",
        modelProperties: {
            name: {
                required: true,
                serializedName: "name",
                type: {
                    name: "String"
                }
            },
            port: {
                required: true,
                serializedName: "port",
                type: {
                    name: "Number"
                }
            },
            hosts: {
                required: true,
                serializedName: "hosts",
                type: {
                    name: "Sequence",
                    element: {
                        type: {
                            name: "Composite",
                            className: "HttpHostConfig"
                        }
                    }
                }
            }
        }
    }
};
export var GatewayProperties = {
    serializedName: "GatewayProperties",
    type: {
        name: "Composite",
        className: "GatewayProperties",
        modelProperties: {
            description: {
                serializedName: "description",
                type: {
                    name: "String"
                }
            },
            sourceNetwork: {
                required: true,
                serializedName: "sourceNetwork",
                type: {
                    name: "Composite",
                    className: "NetworkRef"
                }
            },
            destinationNetwork: {
                required: true,
                serializedName: "destinationNetwork",
                type: {
                    name: "Composite",
                    className: "NetworkRef"
                }
            },
            tcp: {
                serializedName: "tcp",
                type: {
                    name: "Sequence",
                    element: {
                        type: {
                            name: "Composite",
                            className: "TcpConfig"
                        }
                    }
                }
            },
            http: {
                serializedName: "http",
                type: {
                    name: "Sequence",
                    element: {
                        type: {
                            name: "Composite",
                            className: "HttpConfig"
                        }
                    }
                }
            },
            status: {
                readOnly: true,
                serializedName: "status",
                type: {
                    name: "String"
                }
            },
            statusDetails: {
                readOnly: true,
                serializedName: "statusDetails",
                type: {
                    name: "String"
                }
            },
            ipAddress: {
                readOnly: true,
                serializedName: "ipAddress",
                type: {
                    name: "String"
                }
            }
        }
    }
};
export var GatewayResourceDescription = {
    serializedName: "GatewayResourceDescription",
    type: {
        name: "Composite",
        className: "GatewayResourceDescription",
        modelProperties: __assign(__assign({}, TrackedResource.type.modelProperties), { provisioningState: {
                readOnly: true,
                serializedName: "properties.provisioningState",
                type: {
                    name: "String"
                }
            }, description: {
                serializedName: "properties.description",
                type: {
                    name: "String"
                }
            }, sourceNetwork: {
                required: true,
                serializedName: "properties.sourceNetwork",
                type: {
                    name: "Composite",
                    className: "NetworkRef"
                }
            }, destinationNetwork: {
                required: true,
                serializedName: "properties.destinationNetwork",
                type: {
                    name: "Composite",
                    className: "NetworkRef"
                }
            }, tcp: {
                serializedName: "properties.tcp",
                type: {
                    name: "Sequence",
                    element: {
                        type: {
                            name: "Composite",
                            className: "TcpConfig"
                        }
                    }
                }
            }, http: {
                serializedName: "properties.http",
                type: {
                    name: "Sequence",
                    element: {
                        type: {
                            name: "Composite",
                            className: "HttpConfig"
                        }
                    }
                }
            }, status: {
                readOnly: true,
                serializedName: "properties.status",
                type: {
                    name: "String"
                }
            }, statusDetails: {
                readOnly: true,
                serializedName: "properties.statusDetails",
                type: {
                    name: "String"
                }
            }, ipAddress: {
                readOnly: true,
                serializedName: "properties.ipAddress",
                type: {
                    name: "String"
                }
            } })
    }
};
export var ImageRegistryCredential = {
    serializedName: "ImageRegistryCredential",
    type: {
        name: "Composite",
        className: "ImageRegistryCredential",
        modelProperties: {
            server: {
                required: true,
                serializedName: "server",
                type: {
                    name: "String"
                }
            },
            username: {
                required: true,
                serializedName: "username",
                type: {
                    name: "String"
                }
            },
            password: {
                serializedName: "password",
                type: {
                    name: "String"
                }
            }
        }
    }
};
export var EnvironmentVariable = {
    serializedName: "EnvironmentVariable",
    type: {
        name: "Composite",
        className: "EnvironmentVariable",
        modelProperties: {
            name: {
                serializedName: "name",
                type: {
                    name: "String"
                }
            },
            value: {
                serializedName: "value",
                type: {
                    name: "String"
                }
            }
        }
    }
};
export var Setting = {
    serializedName: "Setting",
    type: {
        name: "Composite",
        className: "Setting",
        modelProperties: {
            name: {
                serializedName: "name",
                type: {
                    name: "String"
                }
            },
            value: {
                serializedName: "value",
                type: {
                    name: "String"
                }
            }
        }
    }
};
export var ContainerLabel = {
    serializedName: "ContainerLabel",
    type: {
        name: "Composite",
        className: "ContainerLabel",
        modelProperties: {
            name: {
                required: true,
                serializedName: "name",
                type: {
                    name: "String"
                }
            },
            value: {
                required: true,
                serializedName: "value",
                type: {
                    name: "String"
                }
            }
        }
    }
};
export var EndpointProperties = {
    serializedName: "EndpointProperties",
    type: {
        name: "Composite",
        className: "EndpointProperties",
        modelProperties: {
            name: {
                required: true,
                serializedName: "name",
                type: {
                    name: "String"
                }
            },
            port: {
                serializedName: "port",
                type: {
                    name: "Number"
                }
            }
        }
    }
};
export var ResourceRequests = {
    serializedName: "ResourceRequests",
    type: {
        name: "Composite",
        className: "ResourceRequests",
        modelProperties: {
            memoryInGB: {
                required: true,
                serializedName: "memoryInGB",
                type: {
                    name: "Number"
                }
            },
            cpu: {
                required: true,
                serializedName: "cpu",
                type: {
                    name: "Number"
                }
            }
        }
    }
};
export var ResourceLimits = {
    serializedName: "ResourceLimits",
    type: {
        name: "Composite",
        className: "ResourceLimits",
        modelProperties: {
            memoryInGB: {
                serializedName: "memoryInGB",
                type: {
                    name: "Number"
                }
            },
            cpu: {
                serializedName: "cpu",
                type: {
                    name: "Number"
                }
            }
        }
    }
};
export var ResourceRequirements = {
    serializedName: "ResourceRequirements",
    type: {
        name: "Composite",
        className: "ResourceRequirements",
        modelProperties: {
            requests: {
                required: true,
                serializedName: "requests",
                type: {
                    name: "Composite",
                    className: "ResourceRequests"
                }
            },
            limits: {
                serializedName: "limits",
                type: {
                    name: "Composite",
                    className: "ResourceLimits"
                }
            }
        }
    }
};
export var DiagnosticsRef = {
    serializedName: "DiagnosticsRef",
    type: {
        name: "Composite",
        className: "DiagnosticsRef",
        modelProperties: {
            enabled: {
                serializedName: "enabled",
                type: {
                    name: "Boolean"
                }
            },
            sinkRefs: {
                serializedName: "sinkRefs",
                type: {
                    name: "Sequence",
                    element: {
                        type: {
                            name: "String"
                        }
                    }
                }
            }
        }
    }
};
export var ReliableCollectionsRef = {
    serializedName: "ReliableCollectionsRef",
    type: {
        name: "Composite",
        className: "ReliableCollectionsRef",
        modelProperties: {
            name: {
                required: true,
                serializedName: "name",
                type: {
                    name: "String"
                }
            },
            doNotPersistState: {
                serializedName: "doNotPersistState",
                type: {
                    name: "Boolean"
                }
            }
        }
    }
};
export var ContainerState = {
    serializedName: "ContainerState",
    type: {
        name: "Composite",
        className: "ContainerState",
        modelProperties: {
            state: {
                serializedName: "state",
                type: {
                    name: "String"
                }
            },
            startTime: {
                serializedName: "startTime",
                type: {
                    name: "DateTime"
                }
            },
            exitCode: {
                serializedName: "exitCode",
                type: {
                    name: "String"
                }
            },
            finishTime: {
                serializedName: "finishTime",
                type: {
                    name: "DateTime"
                }
            },
            detailStatus: {
                serializedName: "detailStatus",
                type: {
                    name: "String"
                }
            }
        }
    }
};
export var ContainerEvent = {
    serializedName: "ContainerEvent",
    type: {
        name: "Composite",
        className: "ContainerEvent",
        modelProperties: {
            name: {
                serializedName: "name",
                type: {
                    name: "String"
                }
            },
            count: {
                serializedName: "count",
                type: {
                    name: "Number"
                }
            },
            firstTimestamp: {
                serializedName: "firstTimestamp",
                type: {
                    name: "String"
                }
            },
            lastTimestamp: {
                serializedName: "lastTimestamp",
                type: {
                    name: "String"
                }
            },
            message: {
                serializedName: "message",
                type: {
                    name: "String"
                }
            },
            type: {
                serializedName: "type",
                type: {
                    name: "String"
                }
            }
        }
    }
};
export var ContainerInstanceView = {
    serializedName: "ContainerInstanceView",
    type: {
        name: "Composite",
        className: "ContainerInstanceView",
        modelProperties: {
            restartCount: {
                serializedName: "restartCount",
                type: {
                    name: "Number"
                }
            },
            currentState: {
                serializedName: "currentState",
                type: {
                    name: "Composite",
                    className: "ContainerState"
                }
            },
            previousState: {
                serializedName: "previousState",
                type: {
                    name: "Composite",
                    className: "ContainerState"
                }
            },
            events: {
                serializedName: "events",
                type: {
                    name: "Sequence",
                    element: {
                        type: {
                            name: "Composite",
                            className: "ContainerEvent"
                        }
                    }
                }
            }
        }
    }
};
export var ContainerCodePackageProperties = {
    serializedName: "ContainerCodePackageProperties",
    type: {
        name: "Composite",
        className: "ContainerCodePackageProperties",
        modelProperties: {
            name: {
                required: true,
                serializedName: "name",
                type: {
                    name: "String"
                }
            },
            image: {
                required: true,
                serializedName: "image",
                type: {
                    name: "String"
                }
            },
            imageRegistryCredential: {
                serializedName: "imageRegistryCredential",
                type: {
                    name: "Composite",
                    className: "ImageRegistryCredential"
                }
            },
            entrypoint: {
                serializedName: "entrypoint",
                type: {
                    name: "String"
                }
            },
            commands: {
                serializedName: "commands",
                type: {
                    name: "Sequence",
                    element: {
                        type: {
                            name: "String"
                        }
                    }
                }
            },
            environmentVariables: {
                serializedName: "environmentVariables",
                type: {
                    name: "Sequence",
                    element: {
                        type: {
                            name: "Composite",
                            className: "EnvironmentVariable"
                        }
                    }
                }
            },
            settings: {
                serializedName: "settings",
                type: {
                    name: "Sequence",
                    element: {
                        type: {
                            name: "Composite",
                            className: "Setting"
                        }
                    }
                }
            },
            labels: {
                serializedName: "labels",
                type: {
                    name: "Sequence",
                    element: {
                        type: {
                            name: "Composite",
                            className: "ContainerLabel"
                        }
                    }
                }
            },
            endpoints: {
                serializedName: "endpoints",
                type: {
                    name: "Sequence",
                    element: {
                        type: {
                            name: "Composite",
                            className: "EndpointProperties"
                        }
                    }
                }
            },
            resources: {
                required: true,
                serializedName: "resources",
                type: {
                    name: "Composite",
                    className: "ResourceRequirements"
                }
            },
            volumeRefs: {
                serializedName: "volumeRefs",
                type: {
                    name: "Sequence",
                    element: {
                        type: {
                            name: "Composite",
                            className: "VolumeReference"
                        }
                    }
                }
            },
            volumes: {
                serializedName: "volumes",
                type: {
                    name: "Sequence",
                    element: {
                        type: {
                            name: "Composite",
                            className: "ApplicationScopedVolume"
                        }
                    }
                }
            },
            diagnostics: {
                serializedName: "diagnostics",
                type: {
                    name: "Composite",
                    className: "DiagnosticsRef"
                }
            },
            reliableCollectionsRefs: {
                serializedName: "reliableCollectionsRefs",
                type: {
                    name: "Sequence",
                    element: {
                        type: {
                            name: "Composite",
                            className: "ReliableCollectionsRef"
                        }
                    }
                }
            },
            instanceView: {
                readOnly: true,
                serializedName: "instanceView",
                type: {
                    name: "Composite",
                    className: "ContainerInstanceView"
                }
            }
        }
    }
};
export var AutoScalingTrigger = {
    serializedName: "AutoScalingTrigger",
    type: {
        name: "Composite",
        polymorphicDiscriminator: {
            serializedName: "kind",
            clientName: "kind"
        },
        uberParent: "AutoScalingTrigger",
        className: "AutoScalingTrigger",
        modelProperties: {
            kind: {
                required: true,
                serializedName: "kind",
                type: {
                    name: "String"
                }
            }
        }
    }
};
export var AutoScalingMechanism = {
    serializedName: "AutoScalingMechanism",
    type: {
        name: "Composite",
        polymorphicDiscriminator: {
            serializedName: "kind",
            clientName: "kind"
        },
        uberParent: "AutoScalingMechanism",
        className: "AutoScalingMechanism",
        modelProperties: {
            kind: {
                required: true,
                serializedName: "kind",
                type: {
                    name: "String"
                }
            }
        }
    }
};
export var AutoScalingPolicy = {
    serializedName: "AutoScalingPolicy",
    type: {
        name: "Composite",
        className: "AutoScalingPolicy",
        modelProperties: {
            name: {
                required: true,
                serializedName: "name",
                type: {
                    name: "String"
                }
            },
            trigger: {
                required: true,
                serializedName: "trigger",
                type: {
                    name: "Composite",
                    className: "AutoScalingTrigger"
                }
            },
            mechanism: {
                required: true,
                serializedName: "mechanism",
                type: {
                    name: "Composite",
                    className: "AutoScalingMechanism"
                }
            }
        }
    }
};
export var ServiceResourceDescription = {
    serializedName: "ServiceResourceDescription",
    type: {
        name: "Composite",
        className: "ServiceResourceDescription",
        modelProperties: __assign(__assign({}, ManagedProxyResource.type.modelProperties), { provisioningState: {
                readOnly: true,
                serializedName: "properties.provisioningState",
                type: {
                    name: "String"
                }
            }, osType: {
                required: true,
                serializedName: "properties.osType",
                type: {
                    name: "String"
                }
            }, codePackages: {
                required: true,
                serializedName: "properties.codePackages",
                type: {
                    name: "Sequence",
                    element: {
                        type: {
                            name: "Composite",
                            className: "ContainerCodePackageProperties"
                        }
                    }
                }
            }, networkRefs: {
                serializedName: "properties.networkRefs",
                type: {
                    name: "Sequence",
                    element: {
                        type: {
                            name: "Composite",
                            className: "NetworkRef"
                        }
                    }
                }
            }, diagnostics: {
                serializedName: "properties.diagnostics",
                type: {
                    name: "Composite",
                    className: "DiagnosticsRef"
                }
            }, description: {
                serializedName: "properties.description",
                type: {
                    name: "String"
                }
            }, replicaCount: {
                serializedName: "properties.replicaCount",
                type: {
                    name: "Number"
                }
            }, autoScalingPolicies: {
                serializedName: "properties.autoScalingPolicies",
                type: {
                    name: "Sequence",
                    element: {
                        type: {
                            name: "Composite",
                            className: "AutoScalingPolicy"
                        }
                    }
                }
            }, status: {
                readOnly: true,
                serializedName: "properties.status",
                type: {
                    name: "String"
                }
            }, statusDetails: {
                readOnly: true,
                serializedName: "properties.statusDetails",
                type: {
                    name: "String"
                }
            }, healthState: {
                readOnly: true,
                serializedName: "properties.healthState",
                type: {
                    name: "String"
                }
            }, unhealthyEvaluation: {
                readOnly: true,
                serializedName: "properties.unhealthyEvaluation",
                type: {
                    name: "String"
                }
            } })
    }
};
export var DiagnosticsSinkProperties = {
    serializedName: "DiagnosticsSinkProperties",
    type: {
        name: "Composite",
        polymorphicDiscriminator: {
            serializedName: "kind",
            clientName: "kind"
        },
        uberParent: "DiagnosticsSinkProperties",
        className: "DiagnosticsSinkProperties",
        modelProperties: {
            name: {
                serializedName: "name",
                type: {
                    name: "String"
                }
            },
            description: {
                serializedName: "description",
                type: {
                    name: "String"
                }
            },
            kind: {
                required: true,
                serializedName: "kind",
                type: {
                    name: "String"
                }
            }
        }
    }
};
export var DiagnosticsDescription = {
    serializedName: "DiagnosticsDescription",
    type: {
        name: "Composite",
        className: "DiagnosticsDescription",
        modelProperties: {
            sinks: {
                serializedName: "sinks",
                type: {
                    name: "Sequence",
                    element: {
                        type: {
                            name: "Composite",
                            className: "DiagnosticsSinkProperties"
                        }
                    }
                }
            },
            enabled: {
                serializedName: "enabled",
                type: {
                    name: "Boolean"
                }
            },
            defaultSinkRefs: {
                serializedName: "defaultSinkRefs",
                type: {
                    name: "Sequence",
                    element: {
                        type: {
                            name: "String"
                        }
                    }
                }
            }
        }
    }
};
export var ApplicationProperties = {
    serializedName: "ApplicationProperties",
    type: {
        name: "Composite",
        className: "ApplicationProperties",
        modelProperties: {
            description: {
                serializedName: "description",
                type: {
                    name: "String"
                }
            },
            services: {
                serializedName: "services",
                type: {
                    name: "Sequence",
                    element: {
                        type: {
                            name: "Composite",
                            className: "ServiceResourceDescription"
                        }
                    }
                }
            },
            diagnostics: {
                serializedName: "diagnostics",
                type: {
                    name: "Composite",
                    className: "DiagnosticsDescription"
                }
            },
            debugParams: {
                serializedName: "debugParams",
                type: {
                    name: "String"
                }
            },
            serviceNames: {
                readOnly: true,
                serializedName: "serviceNames",
                type: {
                    name: "Sequence",
                    element: {
                        type: {
                            name: "String"
                        }
                    }
                }
            },
            status: {
                readOnly: true,
                serializedName: "status",
                type: {
                    name: "String"
                }
            },
            statusDetails: {
                readOnly: true,
                serializedName: "statusDetails",
                type: {
                    name: "String"
                }
            },
            healthState: {
                readOnly: true,
                serializedName: "healthState",
                type: {
                    name: "String"
                }
            },
            unhealthyEvaluation: {
                readOnly: true,
                serializedName: "unhealthyEvaluation",
                type: {
                    name: "String"
                }
            }
        }
    }
};
export var AzureInternalMonitoringPipelineSinkDescription = {
    serializedName: "AzureInternalMonitoringPipeline",
    type: {
        name: "Composite",
        polymorphicDiscriminator: DiagnosticsSinkProperties.type.polymorphicDiscriminator,
        uberParent: "DiagnosticsSinkProperties",
        className: "AzureInternalMonitoringPipelineSinkDescription",
        modelProperties: __assign(__assign({}, DiagnosticsSinkProperties.type.modelProperties), { accountName: {
                serializedName: "accountName",
                type: {
                    name: "String"
                }
            }, namespace: {
                serializedName: "namespace",
                type: {
                    name: "String"
                }
            }, maConfigUrl: {
                serializedName: "maConfigUrl",
                type: {
                    name: "String"
                }
            }, fluentdConfigUrl: {
                serializedName: "fluentdConfigUrl",
                type: {
                    name: "Object"
                }
            }, autoKeyConfigUrl: {
                serializedName: "autoKeyConfigUrl",
                type: {
                    name: "String"
                }
            } })
    }
};
export var ApplicationResourceDescription = {
    serializedName: "ApplicationResourceDescription",
    type: {
        name: "Composite",
        className: "ApplicationResourceDescription",
        modelProperties: __assign(__assign({}, TrackedResource.type.modelProperties), { provisioningState: {
                readOnly: true,
                serializedName: "properties.provisioningState",
                type: {
                    name: "String"
                }
            }, description: {
                serializedName: "properties.description",
                type: {
                    name: "String"
                }
            }, services: {
                serializedName: "properties.services",
                type: {
                    name: "Sequence",
                    element: {
                        type: {
                            name: "Composite",
                            className: "ServiceResourceDescription"
                        }
                    }
                }
            }, diagnostics: {
                serializedName: "properties.diagnostics",
                type: {
                    name: "Composite",
                    className: "DiagnosticsDescription"
                }
            }, debugParams: {
                serializedName: "properties.debugParams",
                type: {
                    name: "String"
                }
            }, serviceNames: {
                readOnly: true,
                serializedName: "properties.serviceNames",
                type: {
                    name: "Sequence",
                    element: {
                        type: {
                            name: "String"
                        }
                    }
                }
            }, status: {
                readOnly: true,
                serializedName: "properties.status",
                type: {
                    name: "String"
                }
            }, statusDetails: {
                readOnly: true,
                serializedName: "properties.statusDetails",
                type: {
                    name: "String"
                }
            }, healthState: {
                readOnly: true,
                serializedName: "properties.healthState",
                type: {
                    name: "String"
                }
            }, unhealthyEvaluation: {
                readOnly: true,
                serializedName: "properties.unhealthyEvaluation",
                type: {
                    name: "String"
                }
            } })
    }
};
export var AddRemoveReplicaScalingMechanism = {
    serializedName: "AddRemoveReplica",
    type: {
        name: "Composite",
        polymorphicDiscriminator: AutoScalingMechanism.type.polymorphicDiscriminator,
        uberParent: "AutoScalingMechanism",
        className: "AddRemoveReplicaScalingMechanism",
        modelProperties: __assign(__assign({}, AutoScalingMechanism.type.modelProperties), { minCount: {
                required: true,
                serializedName: "minCount",
                type: {
                    name: "Number"
                }
            }, maxCount: {
                required: true,
                serializedName: "maxCount",
                type: {
                    name: "Number"
                }
            }, scaleIncrement: {
                required: true,
                serializedName: "scaleIncrement",
                type: {
                    name: "Number"
                }
            } })
    }
};
export var AutoScalingMetric = {
    serializedName: "AutoScalingMetric",
    type: {
        name: "Composite",
        polymorphicDiscriminator: {
            serializedName: "kind",
            clientName: "kind"
        },
        uberParent: "AutoScalingMetric",
        className: "AutoScalingMetric",
        modelProperties: {
            kind: {
                required: true,
                serializedName: "kind",
                type: {
                    name: "String"
                }
            }
        }
    }
};
export var AutoScalingResourceMetric = {
    serializedName: "Resource",
    type: {
        name: "Composite",
        polymorphicDiscriminator: AutoScalingMetric.type.polymorphicDiscriminator,
        uberParent: "AutoScalingMetric",
        className: "AutoScalingResourceMetric",
        modelProperties: __assign(__assign({}, AutoScalingMetric.type.modelProperties), { name: {
                required: true,
                serializedName: "name",
                type: {
                    name: "String"
                }
            } })
    }
};
export var ServiceProperties = {
    serializedName: "ServiceProperties",
    type: {
        name: "Composite",
        className: "ServiceProperties",
        modelProperties: {
            description: {
                serializedName: "description",
                type: {
                    name: "String"
                }
            },
            replicaCount: {
                serializedName: "replicaCount",
                type: {
                    name: "Number"
                }
            },
            autoScalingPolicies: {
                serializedName: "autoScalingPolicies",
                type: {
                    name: "Sequence",
                    element: {
                        type: {
                            name: "Composite",
                            className: "AutoScalingPolicy"
                        }
                    }
                }
            },
            status: {
                readOnly: true,
                serializedName: "status",
                type: {
                    name: "String"
                }
            },
            statusDetails: {
                readOnly: true,
                serializedName: "statusDetails",
                type: {
                    name: "String"
                }
            },
            healthState: {
                readOnly: true,
                serializedName: "healthState",
                type: {
                    name: "String"
                }
            },
            unhealthyEvaluation: {
                readOnly: true,
                serializedName: "unhealthyEvaluation",
                type: {
                    name: "String"
                }
            }
        }
    }
};
export var ServiceReplicaProperties = {
    serializedName: "ServiceReplicaProperties",
    type: {
        name: "Composite",
        className: "ServiceReplicaProperties",
        modelProperties: {
            osType: {
                required: true,
                serializedName: "osType",
                type: {
                    name: "String"
                }
            },
            codePackages: {
                required: true,
                serializedName: "codePackages",
                type: {
                    name: "Sequence",
                    element: {
                        type: {
                            name: "Composite",
                            className: "ContainerCodePackageProperties"
                        }
                    }
                }
            },
            networkRefs: {
                serializedName: "networkRefs",
                type: {
                    name: "Sequence",
                    element: {
                        type: {
                            name: "Composite",
                            className: "NetworkRef"
                        }
                    }
                }
            },
            diagnostics: {
                serializedName: "diagnostics",
                type: {
                    name: "Composite",
                    className: "DiagnosticsRef"
                }
            }
        }
    }
};
export var ServiceReplicaDescription = {
    serializedName: "ServiceReplicaDescription",
    type: {
        name: "Composite",
        className: "ServiceReplicaDescription",
        modelProperties: __assign(__assign({}, ServiceReplicaProperties.type.modelProperties), { replicaName: {
                required: true,
                serializedName: "replicaName",
                type: {
                    name: "String"
                }
            } })
    }
};
export var AverageLoadScalingTrigger = {
    serializedName: "AverageLoad",
    type: {
        name: "Composite",
        polymorphicDiscriminator: AutoScalingTrigger.type.polymorphicDiscriminator,
        uberParent: "AutoScalingTrigger",
        className: "AverageLoadScalingTrigger",
        modelProperties: __assign(__assign({}, AutoScalingTrigger.type.modelProperties), { metric: {
                required: true,
                serializedName: "metric",
                type: {
                    name: "Composite",
                    className: "AutoScalingMetric"
                }
            }, lowerLoadThreshold: {
                required: true,
                serializedName: "lowerLoadThreshold",
                type: {
                    name: "Number"
                }
            }, upperLoadThreshold: {
                required: true,
                serializedName: "upperLoadThreshold",
                type: {
                    name: "Number"
                }
            }, scaleIntervalInSeconds: {
                required: true,
                serializedName: "scaleIntervalInSeconds",
                constraints: {
                    InclusiveMinimum: 60
                },
                type: {
                    name: "Number"
                }
            } })
    }
};
export var ContainerLogs = {
    serializedName: "ContainerLogs",
    type: {
        name: "Composite",
        className: "ContainerLogs",
        modelProperties: {
            content: {
                serializedName: "content",
                type: {
                    name: "String"
                }
            }
        }
    }
};
export var OperationListResult = {
    serializedName: "OperationListResult",
    type: {
        name: "Composite",
        className: "OperationListResult",
        modelProperties: {
            value: {
                serializedName: "",
                type: {
                    name: "Sequence",
                    element: {
                        type: {
                            name: "Composite",
                            className: "OperationResult"
                        }
                    }
                }
            },
            nextLink: {
                readOnly: true,
                serializedName: "nextLink",
                type: {
                    name: "String"
                }
            }
        }
    }
};
export var SecretResourceDescriptionList = {
    serializedName: "SecretResourceDescriptionList",
    type: {
        name: "Composite",
        className: "SecretResourceDescriptionList",
        modelProperties: {
            value: {
                serializedName: "",
                type: {
                    name: "Sequence",
                    element: {
                        type: {
                            name: "Composite",
                            className: "SecretResourceDescription"
                        }
                    }
                }
            },
            nextLink: {
                serializedName: "nextLink",
                type: {
                    name: "String"
                }
            }
        }
    }
};
export var SecretValueResourceDescriptionList = {
    serializedName: "SecretValueResourceDescriptionList",
    type: {
        name: "Composite",
        className: "SecretValueResourceDescriptionList",
        modelProperties: {
            value: {
                serializedName: "",
                type: {
                    name: "Sequence",
                    element: {
                        type: {
                            name: "Composite",
                            className: "SecretValueResourceDescription"
                        }
                    }
                }
            },
            nextLink: {
                serializedName: "nextLink",
                type: {
                    name: "String"
                }
            }
        }
    }
};
export var VolumeResourceDescriptionList = {
    serializedName: "VolumeResourceDescriptionList",
    type: {
        name: "Composite",
        className: "VolumeResourceDescriptionList",
        modelProperties: {
            value: {
                serializedName: "",
                type: {
                    name: "Sequence",
                    element: {
                        type: {
                            name: "Composite",
                            className: "VolumeResourceDescription"
                        }
                    }
                }
            },
            nextLink: {
                serializedName: "nextLink",
                type: {
                    name: "String"
                }
            }
        }
    }
};
export var NetworkResourceDescriptionList = {
    serializedName: "NetworkResourceDescriptionList",
    type: {
        name: "Composite",
        className: "NetworkResourceDescriptionList",
        modelProperties: {
            value: {
                serializedName: "",
                type: {
                    name: "Sequence",
                    element: {
                        type: {
                            name: "Composite",
                            className: "NetworkResourceDescription"
                        }
                    }
                }
            },
            nextLink: {
                serializedName: "nextLink",
                type: {
                    name: "String"
                }
            }
        }
    }
};
export var GatewayResourceDescriptionList = {
    serializedName: "GatewayResourceDescriptionList",
    type: {
        name: "Composite",
        className: "GatewayResourceDescriptionList",
        modelProperties: {
            value: {
                serializedName: "",
                type: {
                    name: "Sequence",
                    element: {
                        type: {
                            name: "Composite",
                            className: "GatewayResourceDescription"
                        }
                    }
                }
            },
            nextLink: {
                serializedName: "nextLink",
                type: {
                    name: "String"
                }
            }
        }
    }
};
export var ApplicationResourceDescriptionList = {
    serializedName: "ApplicationResourceDescriptionList",
    type: {
        name: "Composite",
        className: "ApplicationResourceDescriptionList",
        modelProperties: {
            value: {
                serializedName: "",
                type: {
                    name: "Sequence",
                    element: {
                        type: {
                            name: "Composite",
                            className: "ApplicationResourceDescription"
                        }
                    }
                }
            },
            nextLink: {
                serializedName: "nextLink",
                type: {
                    name: "String"
                }
            }
        }
    }
};
export var ServiceResourceDescriptionList = {
    serializedName: "ServiceResourceDescriptionList",
    type: {
        name: "Composite",
        className: "ServiceResourceDescriptionList",
        modelProperties: {
            value: {
                serializedName: "",
                type: {
                    name: "Sequence",
                    element: {
                        type: {
                            name: "Composite",
                            className: "ServiceResourceDescription"
                        }
                    }
                }
            },
            nextLink: {
                serializedName: "nextLink",
                type: {
                    name: "String"
                }
            }
        }
    }
};
export var ServiceReplicaDescriptionList = {
    serializedName: "ServiceReplicaDescriptionList",
    type: {
        name: "Composite",
        className: "ServiceReplicaDescriptionList",
        modelProperties: {
            value: {
                serializedName: "",
                type: {
                    name: "Sequence",
                    element: {
                        type: {
                            name: "Composite",
                            className: "ServiceReplicaDescription"
                        }
                    }
                }
            },
            nextLink: {
                serializedName: "nextLink",
                type: {
                    name: "String"
                }
            }
        }
    }
};
export var discriminators = {
    'ProvisionedResourceProperties.SecretResourceProperties': SecretResourceProperties,
    'ProvisionedResourceProperties.inlinedValue': InlinedValueSecretResourceProperties,
    'ProvisionedResourceProperties.SecretResourcePropertiesBase': SecretResourcePropertiesBase,
    'ApplicationScopedVolumeCreationParameters': ApplicationScopedVolumeCreationParameters,
    'ApplicationScopedVolumeCreationParameters.ServiceFabricVolumeDisk': ApplicationScopedVolumeCreationParametersServiceFabricVolumeDisk,
    'ProvisionedResourceProperties.NetworkResourceProperties': NetworkResourceProperties,
    'ProvisionedResourceProperties.Local': LocalNetworkResourceProperties,
    'ProvisionedResourceProperties.NetworkResourcePropertiesBase': NetworkResourcePropertiesBase,
    'AutoScalingTrigger': AutoScalingTrigger,
    'AutoScalingMechanism': AutoScalingMechanism,
    'DiagnosticsSinkProperties': DiagnosticsSinkProperties,
    'DiagnosticsSinkProperties.AzureInternalMonitoringPipeline': AzureInternalMonitoringPipelineSinkDescription,
    'AutoScalingMechanism.AddRemoveReplica': AddRemoveReplicaScalingMechanism,
    'AutoScalingMetric': AutoScalingMetric,
    'AutoScalingMetric.Resource': AutoScalingResourceMetric,
    'AutoScalingTrigger.AverageLoad': AverageLoadScalingTrigger
};
