// @azure/arm-redisenterprisecache/dist-esm/src/models/index.js
/*
 * Copyright (c) Microsoft Corporation.
 * Licensed under the MIT License.
 *
 * Code generated by Microsoft (R) AutoRest Code Generator.
 * Changes may cause incorrect behavior and will be lost if the code is regenerated.
 */
/** Known values of {@link Origin} that the service accepts. */
export var KnownOrigin;
(function (KnownOrigin) {
    /** User */
    KnownOrigin["User"] = "user";
    /** System */
    KnownOrigin["System"] = "system";
    /** UserSystem */
    KnownOrigin["UserSystem"] = "user,system";
})(KnownOrigin || (KnownOrigin = {}));
/** Known values of {@link ActionType} that the service accepts. */
export var KnownActionType;
(function (KnownActionType) {
    /** Internal */
    KnownActionType["Internal"] = "Internal";
})(KnownActionType || (KnownActionType = {}));
/** Known values of {@link SkuName} that the service accepts. */
export var KnownSkuName;
(function (KnownSkuName) {
    /** EnterpriseE10 */
    KnownSkuName["EnterpriseE10"] = "Enterprise_E10";
    /** EnterpriseE20 */
    KnownSkuName["EnterpriseE20"] = "Enterprise_E20";
    /** EnterpriseE50 */
    KnownSkuName["EnterpriseE50"] = "Enterprise_E50";
    /** EnterpriseE100 */
    KnownSkuName["EnterpriseE100"] = "Enterprise_E100";
    /** EnterpriseFlashF300 */
    KnownSkuName["EnterpriseFlashF300"] = "EnterpriseFlash_F300";
    /** EnterpriseFlashF700 */
    KnownSkuName["EnterpriseFlashF700"] = "EnterpriseFlash_F700";
    /** EnterpriseFlashF1500 */
    KnownSkuName["EnterpriseFlashF1500"] = "EnterpriseFlash_F1500";
})(KnownSkuName || (KnownSkuName = {}));
/** Known values of {@link TlsVersion} that the service accepts. */
export var KnownTlsVersion;
(function (KnownTlsVersion) {
    /** One0 */
    KnownTlsVersion["One0"] = "1.0";
    /** One1 */
    KnownTlsVersion["One1"] = "1.1";
    /** One2 */
    KnownTlsVersion["One2"] = "1.2";
})(KnownTlsVersion || (KnownTlsVersion = {}));
/** Known values of {@link ProvisioningState} that the service accepts. */
export var KnownProvisioningState;
(function (KnownProvisioningState) {
    /** Succeeded */
    KnownProvisioningState["Succeeded"] = "Succeeded";
    /** Failed */
    KnownProvisioningState["Failed"] = "Failed";
    /** Canceled */
    KnownProvisioningState["Canceled"] = "Canceled";
    /** Creating */
    KnownProvisioningState["Creating"] = "Creating";
    /** Updating */
    KnownProvisioningState["Updating"] = "Updating";
    /** Deleting */
    KnownProvisioningState["Deleting"] = "Deleting";
})(KnownProvisioningState || (KnownProvisioningState = {}));
/** Known values of {@link ResourceState} that the service accepts. */
export var KnownResourceState;
(function (KnownResourceState) {
    /** Running */
    KnownResourceState["Running"] = "Running";
    /** Creating */
    KnownResourceState["Creating"] = "Creating";
    /** CreateFailed */
    KnownResourceState["CreateFailed"] = "CreateFailed";
    /** Updating */
    KnownResourceState["Updating"] = "Updating";
    /** UpdateFailed */
    KnownResourceState["UpdateFailed"] = "UpdateFailed";
    /** Deleting */
    KnownResourceState["Deleting"] = "Deleting";
    /** DeleteFailed */
    KnownResourceState["DeleteFailed"] = "DeleteFailed";
    /** Enabling */
    KnownResourceState["Enabling"] = "Enabling";
    /** EnableFailed */
    KnownResourceState["EnableFailed"] = "EnableFailed";
    /** Disabling */
    KnownResourceState["Disabling"] = "Disabling";
    /** DisableFailed */
    KnownResourceState["DisableFailed"] = "DisableFailed";
    /** Disabled */
    KnownResourceState["Disabled"] = "Disabled";
})(KnownResourceState || (KnownResourceState = {}));
/** Known values of {@link PrivateEndpointServiceConnectionStatus} that the service accepts. */
export var KnownPrivateEndpointServiceConnectionStatus;
(function (KnownPrivateEndpointServiceConnectionStatus) {
    /** Pending */
    KnownPrivateEndpointServiceConnectionStatus["Pending"] = "Pending";
    /** Approved */
    KnownPrivateEndpointServiceConnectionStatus["Approved"] = "Approved";
    /** Rejected */
    KnownPrivateEndpointServiceConnectionStatus["Rejected"] = "Rejected";
})(KnownPrivateEndpointServiceConnectionStatus || (KnownPrivateEndpointServiceConnectionStatus = {}));
/** Known values of {@link PrivateEndpointConnectionProvisioningState} that the service accepts. */
export var KnownPrivateEndpointConnectionProvisioningState;
(function (KnownPrivateEndpointConnectionProvisioningState) {
    /** Succeeded */
    KnownPrivateEndpointConnectionProvisioningState["Succeeded"] = "Succeeded";
    /** Creating */
    KnownPrivateEndpointConnectionProvisioningState["Creating"] = "Creating";
    /** Deleting */
    KnownPrivateEndpointConnectionProvisioningState["Deleting"] = "Deleting";
    /** Failed */
    KnownPrivateEndpointConnectionProvisioningState["Failed"] = "Failed";
})(KnownPrivateEndpointConnectionProvisioningState || (KnownPrivateEndpointConnectionProvisioningState = {}));
/** Known values of {@link Protocol} that the service accepts. */
export var KnownProtocol;
(function (KnownProtocol) {
    /** Encrypted */
    KnownProtocol["Encrypted"] = "Encrypted";
    /** Plaintext */
    KnownProtocol["Plaintext"] = "Plaintext";
})(KnownProtocol || (KnownProtocol = {}));
/** Known values of {@link ClusteringPolicy} that the service accepts. */
export var KnownClusteringPolicy;
(function (KnownClusteringPolicy) {
    /** EnterpriseCluster */
    KnownClusteringPolicy["EnterpriseCluster"] = "EnterpriseCluster";
    /** OSSCluster */
    KnownClusteringPolicy["OSSCluster"] = "OSSCluster";
})(KnownClusteringPolicy || (KnownClusteringPolicy = {}));
/** Known values of {@link EvictionPolicy} that the service accepts. */
export var KnownEvictionPolicy;
(function (KnownEvictionPolicy) {
    /** AllKeysLFU */
    KnownEvictionPolicy["AllKeysLFU"] = "AllKeysLFU";
    /** AllKeysLRU */
    KnownEvictionPolicy["AllKeysLRU"] = "AllKeysLRU";
    /** AllKeysRandom */
    KnownEvictionPolicy["AllKeysRandom"] = "AllKeysRandom";
    /** VolatileLRU */
    KnownEvictionPolicy["VolatileLRU"] = "VolatileLRU";
    /** VolatileLFU */
    KnownEvictionPolicy["VolatileLFU"] = "VolatileLFU";
    /** VolatileTTL */
    KnownEvictionPolicy["VolatileTTL"] = "VolatileTTL";
    /** VolatileRandom */
    KnownEvictionPolicy["VolatileRandom"] = "VolatileRandom";
    /** NoEviction */
    KnownEvictionPolicy["NoEviction"] = "NoEviction";
})(KnownEvictionPolicy || (KnownEvictionPolicy = {}));
/** Known values of {@link AofFrequency} that the service accepts. */
export var KnownAofFrequency;
(function (KnownAofFrequency) {
    /** OneS */
    KnownAofFrequency["OneS"] = "1s";
    /** Always */
    KnownAofFrequency["Always"] = "always";
})(KnownAofFrequency || (KnownAofFrequency = {}));
/** Known values of {@link RdbFrequency} that the service accepts. */
export var KnownRdbFrequency;
(function (KnownRdbFrequency) {
    /** OneH */
    KnownRdbFrequency["OneH"] = "1h";
    /** SixH */
    KnownRdbFrequency["SixH"] = "6h";
    /** TwelveH */
    KnownRdbFrequency["TwelveH"] = "12h";
})(KnownRdbFrequency || (KnownRdbFrequency = {}));
/** Known values of {@link LinkState} that the service accepts. */
export var KnownLinkState;
(function (KnownLinkState) {
    /** Linked */
    KnownLinkState["Linked"] = "Linked";
    /** Linking */
    KnownLinkState["Linking"] = "Linking";
    /** Unlinking */
    KnownLinkState["Unlinking"] = "Unlinking";
    /** LinkFailed */
    KnownLinkState["LinkFailed"] = "LinkFailed";
    /** UnlinkFailed */
    KnownLinkState["UnlinkFailed"] = "UnlinkFailed";
})(KnownLinkState || (KnownLinkState = {}));
