// @azure/arm-app/dist-esm/src/models/index.js
/*
 * Copyright (c) Microsoft Corporation.
 * Licensed under the MIT License.
 *
 * Code generated by Microsoft (R) AutoRest Code Generator.
 * Changes may cause incorrect behavior and will be lost if the code is regenerated.
 */
/** Known values of {@link ManagedServiceIdentityType} that the service accepts. */
export var KnownManagedServiceIdentityType;
(function (KnownManagedServiceIdentityType) {
    KnownManagedServiceIdentityType["None"] = "None";
    KnownManagedServiceIdentityType["SystemAssigned"] = "SystemAssigned";
    KnownManagedServiceIdentityType["UserAssigned"] = "UserAssigned";
    KnownManagedServiceIdentityType["SystemAssignedUserAssigned"] = "SystemAssigned,UserAssigned";
})(KnownManagedServiceIdentityType || (KnownManagedServiceIdentityType = {}));
/** Known values of {@link ContainerAppProvisioningState} that the service accepts. */
export var KnownContainerAppProvisioningState;
(function (KnownContainerAppProvisioningState) {
    KnownContainerAppProvisioningState["InProgress"] = "InProgress";
    KnownContainerAppProvisioningState["Succeeded"] = "Succeeded";
    KnownContainerAppProvisioningState["Failed"] = "Failed";
    KnownContainerAppProvisioningState["Canceled"] = "Canceled";
})(KnownContainerAppProvisioningState || (KnownContainerAppProvisioningState = {}));
/** Known values of {@link ActiveRevisionsMode} that the service accepts. */
export var KnownActiveRevisionsMode;
(function (KnownActiveRevisionsMode) {
    KnownActiveRevisionsMode["Multiple"] = "multiple";
    KnownActiveRevisionsMode["Single"] = "single";
})(KnownActiveRevisionsMode || (KnownActiveRevisionsMode = {}));
/** Known values of {@link IngressTransportMethod} that the service accepts. */
export var KnownIngressTransportMethod;
(function (KnownIngressTransportMethod) {
    KnownIngressTransportMethod["Auto"] = "auto";
    KnownIngressTransportMethod["Http"] = "http";
    KnownIngressTransportMethod["Http2"] = "http2";
})(KnownIngressTransportMethod || (KnownIngressTransportMethod = {}));
/** Known values of {@link BindingType} that the service accepts. */
export var KnownBindingType;
(function (KnownBindingType) {
    KnownBindingType["Disabled"] = "Disabled";
    KnownBindingType["SniEnabled"] = "SniEnabled";
})(KnownBindingType || (KnownBindingType = {}));
/** Known values of {@link AppProtocol} that the service accepts. */
export var KnownAppProtocol;
(function (KnownAppProtocol) {
    KnownAppProtocol["Http"] = "http";
    KnownAppProtocol["Grpc"] = "grpc";
})(KnownAppProtocol || (KnownAppProtocol = {}));
/** Known values of {@link Type} that the service accepts. */
export var KnownType;
(function (KnownType) {
    KnownType["Liveness"] = "liveness";
    KnownType["Readiness"] = "readiness";
    KnownType["Startup"] = "startup";
})(KnownType || (KnownType = {}));
/** Known values of {@link StorageType} that the service accepts. */
export var KnownStorageType;
(function (KnownStorageType) {
    KnownStorageType["AzureFile"] = "AzureFile";
    KnownStorageType["EmptyDir"] = "EmptyDir";
})(KnownStorageType || (KnownStorageType = {}));
/** Known values of {@link CreatedByType} that the service accepts. */
export var KnownCreatedByType;
(function (KnownCreatedByType) {
    KnownCreatedByType["User"] = "User";
    KnownCreatedByType["Application"] = "Application";
    KnownCreatedByType["ManagedIdentity"] = "ManagedIdentity";
    KnownCreatedByType["Key"] = "Key";
})(KnownCreatedByType || (KnownCreatedByType = {}));
/** Known values of {@link RevisionHealthState} that the service accepts. */
export var KnownRevisionHealthState;
(function (KnownRevisionHealthState) {
    KnownRevisionHealthState["Healthy"] = "Healthy";
    KnownRevisionHealthState["Unhealthy"] = "Unhealthy";
    KnownRevisionHealthState["None"] = "None";
})(KnownRevisionHealthState || (KnownRevisionHealthState = {}));
/** Known values of {@link RevisionProvisioningState} that the service accepts. */
export var KnownRevisionProvisioningState;
(function (KnownRevisionProvisioningState) {
    KnownRevisionProvisioningState["Provisioning"] = "Provisioning";
    KnownRevisionProvisioningState["Provisioned"] = "Provisioned";
    KnownRevisionProvisioningState["Failed"] = "Failed";
    KnownRevisionProvisioningState["Deprovisioning"] = "Deprovisioning";
    KnownRevisionProvisioningState["Deprovisioned"] = "Deprovisioned";
})(KnownRevisionProvisioningState || (KnownRevisionProvisioningState = {}));
/** Known values of {@link EnvironmentProvisioningState} that the service accepts. */
export var KnownEnvironmentProvisioningState;
(function (KnownEnvironmentProvisioningState) {
    KnownEnvironmentProvisioningState["Succeeded"] = "Succeeded";
    KnownEnvironmentProvisioningState["Failed"] = "Failed";
    KnownEnvironmentProvisioningState["Canceled"] = "Canceled";
    KnownEnvironmentProvisioningState["Waiting"] = "Waiting";
    KnownEnvironmentProvisioningState["InitializationInProgress"] = "InitializationInProgress";
    KnownEnvironmentProvisioningState["InfrastructureSetupInProgress"] = "InfrastructureSetupInProgress";
    KnownEnvironmentProvisioningState["InfrastructureSetupComplete"] = "InfrastructureSetupComplete";
    KnownEnvironmentProvisioningState["ScheduledForDelete"] = "ScheduledForDelete";
    KnownEnvironmentProvisioningState["UpgradeRequested"] = "UpgradeRequested";
    KnownEnvironmentProvisioningState["UpgradeFailed"] = "UpgradeFailed";
})(KnownEnvironmentProvisioningState || (KnownEnvironmentProvisioningState = {}));
/** Known values of {@link CertificateProvisioningState} that the service accepts. */
export var KnownCertificateProvisioningState;
(function (KnownCertificateProvisioningState) {
    KnownCertificateProvisioningState["Succeeded"] = "Succeeded";
    KnownCertificateProvisioningState["Failed"] = "Failed";
    KnownCertificateProvisioningState["Canceled"] = "Canceled";
    KnownCertificateProvisioningState["DeleteFailed"] = "DeleteFailed";
    KnownCertificateProvisioningState["Pending"] = "Pending";
})(KnownCertificateProvisioningState || (KnownCertificateProvisioningState = {}));
/** Known values of {@link SourceControlOperationState} that the service accepts. */
export var KnownSourceControlOperationState;
(function (KnownSourceControlOperationState) {
    KnownSourceControlOperationState["InProgress"] = "InProgress";
    KnownSourceControlOperationState["Succeeded"] = "Succeeded";
    KnownSourceControlOperationState["Failed"] = "Failed";
    KnownSourceControlOperationState["Canceled"] = "Canceled";
})(KnownSourceControlOperationState || (KnownSourceControlOperationState = {}));
/** Known values of {@link AccessMode} that the service accepts. */
export var KnownAccessMode;
(function (KnownAccessMode) {
    KnownAccessMode["ReadOnly"] = "ReadOnly";
    KnownAccessMode["ReadWrite"] = "ReadWrite";
})(KnownAccessMode || (KnownAccessMode = {}));
