// @google-cloud/org-policy/build/src/v2/org_policy_client.js
"use strict";
// Copyright 2024 Google LLC
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     https://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.
//
// ** This file is automatically generated by gapic-generator-typescript. **
// ** https://github.com/googleapis/gapic-generator-typescript **
// ** All changes to this file may be overwritten. **
Object.defineProperty(exports, "__esModule", { value: true });
exports.OrgPolicyClient = void 0;
const jsonProtos = require("../../protos/protos.json");
/**
 * Client JSON configuration object, loaded from
 * `src/v2/org_policy_client_config.json`.
 * This file defines retry strategy and timeouts for all API methods in this library.
 */
const gapicConfig = require("./org_policy_client_config.json");
const version = require('../../../package.json').version;
/**
 *  An interface for managing organization policies.
 *
 *  The Organization Policy Service provides a simple mechanism for
 *  organizations to restrict the allowed configurations across their entire
 *  resource hierarchy.
 *
 *  You can use a policy to configure restrictions on resources. For
 *  example, you can enforce a policy that restricts which Google
 *  Cloud APIs can be activated in a certain part of your resource
 *  hierarchy, or prevents serial port access to VM instances in a
 *  particular folder.
 *
 *  Policies are inherited down through the resource hierarchy. A policy
 *  applied to a parent resource automatically applies to all its child resources
 *  unless overridden with a policy lower in the hierarchy.
 *
 *  A constraint defines an aspect of a resource's configuration that can be
 *  controlled by an organization's policy administrator. Policies are a
 *  collection of constraints that defines their allowable configuration on a
 *  particular resource and its child resources.
 * @class
 * @memberof v2
 */
class OrgPolicyClient {
    /**
     * Construct an instance of OrgPolicyClient.
     *
     * @param {object} [options] - The configuration object.
     * The options accepted by the constructor are described in detail
     * in [this document](https://github.com/googleapis/gax-nodejs/blob/main/client-libraries.md#creating-the-client-instance).
     * The common options are:
     * @param {object} [options.credentials] - Credentials object.
     * @param {string} [options.credentials.client_email]
     * @param {string} [options.credentials.private_key]
     * @param {string} [options.email] - Account email address. Required when
     *     using a .pem or .p12 keyFilename.
     * @param {string} [options.keyFilename] - Full path to the a .json, .pem, or
     *     .p12 key downloaded from the Google Developers Console. If you provide
     *     a path to a JSON file, the projectId option below is not necessary.
     *     NOTE: .pem and .p12 require you to specify options.email as well.
     * @param {number} [options.port] - The port on which to connect to
     *     the remote host.
     * @param {string} [options.projectId] - The project ID from the Google
     *     Developer's Console, e.g. 'grape-spaceship-123'. We will also check
     *     the environment variable GCLOUD_PROJECT for your project ID. If your
     *     app is running in an environment which supports
     *     {@link https://developers.google.com/identity/protocols/application-default-credentials Application Default Credentials},
     *     your project ID will be detected automatically.
     * @param {string} [options.apiEndpoint] - The domain name of the
     *     API remote host.
     * @param {gax.ClientConfig} [options.clientConfig] - Client configuration override.
     *     Follows the structure of {@link gapicConfig}.
     * @param {boolean} [options.fallback] - Use HTTP/1.1 REST mode.
     *     For more information, please check the
     *     {@link https://github.com/googleapis/gax-nodejs/blob/main/client-libraries.md#http11-rest-api-mode documentation}.
     * @param {gax} [gaxInstance]: loaded instance of `google-gax`. Useful if you
     *     need to avoid loading the default gRPC version and want to use the fallback
     *     HTTP implementation. Load only fallback version and pass it to the constructor:
     *     ```
     *     const gax = require('google-gax/build/src/fallback'); // avoids loading google-gax with gRPC
     *     const client = new OrgPolicyClient({fallback: true}, gax);
     *     ```
     */
    constructor(opts, gaxInstance) {
        var _a, _b, _c, _d, _e;
        this._terminated = false;
        this.descriptors = {
            page: {},
            stream: {},
            longrunning: {},
            batching: {},
        };
        // Ensure that options include all the required fields.
        const staticMembers = this.constructor;
        if ((opts === null || opts === void 0 ? void 0 : opts.universe_domain) &&
            (opts === null || opts === void 0 ? void 0 : opts.universeDomain) &&
            (opts === null || opts === void 0 ? void 0 : opts.universe_domain) !== (opts === null || opts === void 0 ? void 0 : opts.universeDomain)) {
            throw new Error('Please set either universe_domain or universeDomain, but not both.');
        }
        const universeDomainEnvVar = typeof process === 'object' && typeof process.env === 'object'
            ? process.env['GOOGLE_CLOUD_UNIVERSE_DOMAIN']
            : undefined;
        this._universeDomain =
            (_c = (_b = (_a = opts === null || opts === void 0 ? void 0 : opts.universeDomain) !== null && _a !== void 0 ? _a : opts === null || opts === void 0 ? void 0 : opts.universe_domain) !== null && _b !== void 0 ? _b : universeDomainEnvVar) !== null && _c !== void 0 ? _c : 'googleapis.com';
        this._servicePath = 'orgpolicy.' + this._universeDomain;
        const servicePath = (opts === null || opts === void 0 ? void 0 : opts.servicePath) || (opts === null || opts === void 0 ? void 0 : opts.apiEndpoint) || this._servicePath;
        this._providedCustomServicePath = !!((opts === null || opts === void 0 ? void 0 : opts.servicePath) || (opts === null || opts === void 0 ? void 0 : opts.apiEndpoint));
        const port = (opts === null || opts === void 0 ? void 0 : opts.port) || staticMembers.port;
        const clientConfig = (_d = opts === null || opts === void 0 ? void 0 : opts.clientConfig) !== null && _d !== void 0 ? _d : {};
        const fallback = (_e = opts === null || opts === void 0 ? void 0 : opts.fallback) !== null && _e !== void 0 ? _e : (typeof window !== 'undefined' && typeof (window === null || window === void 0 ? void 0 : window.fetch) === 'function');
        opts = Object.assign({ servicePath, port, clientConfig, fallback }, opts);
        // Request numeric enum values if REST transport is used.
        opts.numericEnums = true;
        // If scopes are unset in options and we're connecting to a non-default endpoint, set scopes just in case.
        if (servicePath !== this._servicePath && !('scopes' in opts)) {
            opts['scopes'] = staticMembers.scopes;
        }
        // Load google-gax module synchronously if needed
        if (!gaxInstance) {
            gaxInstance = require('google-gax');
        }
        // Choose either gRPC or proto-over-HTTP implementation of google-gax.
        this._gaxModule = opts.fallback ? gaxInstance.fallback : gaxInstance;
        // Create a `gaxGrpc` object, with any grpc-specific options sent to the client.
        this._gaxGrpc = new this._gaxModule.GrpcClient(opts);
        // Save options to use in initialize() method.
        this._opts = opts;
        // Save the auth object to the client, for use by other methods.
        this.auth = this._gaxGrpc.auth;
        // Set useJWTAccessWithScope on the auth object.
        this.auth.useJWTAccessWithScope = true;
        // Set defaultServicePath on the auth object.
        this.auth.defaultServicePath = this._servicePath;
        // Set the default scopes in auth client if needed.
        if (servicePath === this._servicePath) {
            this.auth.defaultScopes = staticMembers.scopes;
        }
        // Determine the client header string.
        const clientHeader = [`gax/${this._gaxModule.version}`, `gapic/${version}`];
        if (typeof process === 'object' && 'versions' in process) {
            clientHeader.push(`gl-node/${process.versions.node}`);
        }
        else {
            clientHeader.push(`gl-web/${this._gaxModule.version}`);
        }
        if (!opts.fallback) {
            clientHeader.push(`grpc/${this._gaxGrpc.grpcVersion}`);
        }
        else {
            clientHeader.push(`rest/${this._gaxGrpc.grpcVersion}`);
        }
        if (opts.libName && opts.libVersion) {
            clientHeader.push(`${opts.libName}/${opts.libVersion}`);
        }
        // Load the applicable protos.
        this._protos = this._gaxGrpc.loadProtoJSON(jsonProtos);
        // This API contains "path templates"; forward-slash-separated
        // identifiers to uniquely identify resources within the API.
        // Create useful helper objects for these.
        this.pathTemplates = {
            customConstraintPathTemplate: new this._gaxModule.PathTemplate('organizations/{organization}/customConstraints/{custom_constraint}'),
            folderConstraintPathTemplate: new this._gaxModule.PathTemplate('folders/{folder}/constraints/{constraint}'),
            folderPolicyPathTemplate: new this._gaxModule.PathTemplate('folders/{folder}/policies/{policy}'),
            organizationPathTemplate: new this._gaxModule.PathTemplate('organizations/{organization}'),
            organizationConstraintPathTemplate: new this._gaxModule.PathTemplate('organizations/{organization}/constraints/{constraint}'),
            organizationPolicyPathTemplate: new this._gaxModule.PathTemplate('organizations/{organization}/policies/{policy}'),
            projectPathTemplate: new this._gaxModule.PathTemplate('projects/{project}'),
            projectConstraintPathTemplate: new this._gaxModule.PathTemplate('projects/{project}/constraints/{constraint}'),
            projectPolicyPathTemplate: new this._gaxModule.PathTemplate('projects/{project}/policies/{policy}'),
        };
        // Some of the methods on this service return "paged" results,
        // (e.g. 50 results at a time, with tokens to get subsequent
        // pages). Denote the keys used for pagination and results.
        this.descriptors.page = {
            listConstraints: new this._gaxModule.PageDescriptor('pageToken', 'nextPageToken', 'constraints'),
            listPolicies: new this._gaxModule.PageDescriptor('pageToken', 'nextPageToken', 'policies'),
            listCustomConstraints: new this._gaxModule.PageDescriptor('pageToken', 'nextPageToken', 'customConstraints'),
        };
        // Put together the default options sent with requests.
        this._defaults = this._gaxGrpc.constructSettings('google.cloud.orgpolicy.v2.OrgPolicy', gapicConfig, opts.clientConfig || {}, { 'x-goog-api-client': clientHeader.join(' ') });
        // Set up a dictionary of "inner API calls"; the core implementation
        // of calling the API is handled in `google-gax`, with this code
        // merely providing the destination and request information.
        this.innerApiCalls = {};
        // Add a warn function to the client constructor so it can be easily tested.
        this.warn = this._gaxModule.warn;
    }
    /**
     * Initialize the client.
     * Performs asynchronous operations (such as authentication) and prepares the client.
     * This function will be called automatically when any class method is called for the
     * first time, but if you need to initialize it before calling an actual method,
     * feel free to call initialize() directly.
     *
     * You can await on this method if you want to make sure the client is initialized.
     *
     * @returns {Promise} A promise that resolves to an authenticated service stub.
     */
    initialize() {
        // If the client stub promise is already initialized, return immediately.
        if (this.orgPolicyStub) {
            return this.orgPolicyStub;
        }
        // Put together the "service stub" for
        // google.cloud.orgpolicy.v2.OrgPolicy.
        this.orgPolicyStub = this._gaxGrpc.createStub(this._opts.fallback
            ? this._protos.lookupService('google.cloud.orgpolicy.v2.OrgPolicy')
            : // eslint-disable-next-line @typescript-eslint/no-explicit-any
                this._protos.google.cloud.orgpolicy.v2.OrgPolicy, this._opts, this._providedCustomServicePath);
        // Iterate over each of the methods that the service provides
        // and create an API call method for each.
        const orgPolicyStubMethods = [
            'listConstraints',
            'listPolicies',
            'getPolicy',
            'getEffectivePolicy',
            'createPolicy',
            'updatePolicy',
            'deletePolicy',
            'createCustomConstraint',
            'updateCustomConstraint',
            'getCustomConstraint',
            'listCustomConstraints',
            'deleteCustomConstraint',
        ];
        for (const methodName of orgPolicyStubMethods) {
            const callPromise = this.orgPolicyStub.then(stub => (...args) => {
                if (this._terminated) {
                    return Promise.reject('The client has already been closed.');
                }
                const func = stub[methodName];
                return func.apply(stub, args);
            }, (err) => () => {
                throw err;
            });
            const descriptor = this.descriptors.page[methodName] || undefined;
            const apiCall = this._gaxModule.createApiCall(callPromise, this._defaults[methodName], descriptor, this._opts.fallback);
            this.innerApiCalls[methodName] = apiCall;
        }
        return this.orgPolicyStub;
    }
    /**
     * The DNS address for this API service.
     * @deprecated Use the apiEndpoint method of the client instance.
     * @returns {string} The DNS address for this service.
     */
    static get servicePath() {
        if (typeof process === 'object' &&
            typeof process.emitWarning === 'function') {
            process.emitWarning('Static servicePath is deprecated, please use the instance method instead.', 'DeprecationWarning');
        }
        return 'orgpolicy.googleapis.com';
    }
    /**
     * The DNS address for this API service - same as servicePath.
     * @deprecated Use the apiEndpoint method of the client instance.
     * @returns {string} The DNS address for this service.
     */
    static get apiEndpoint() {
        if (typeof process === 'object' &&
            typeof process.emitWarning === 'function') {
            process.emitWarning('Static apiEndpoint is deprecated, please use the instance method instead.', 'DeprecationWarning');
        }
        return 'orgpolicy.googleapis.com';
    }
    /**
     * The DNS address for this API service.
     * @returns {string} The DNS address for this service.
     */
    get apiEndpoint() {
        return this._servicePath;
    }
    get universeDomain() {
        return this._universeDomain;
    }
    /**
     * The port for this API service.
     * @returns {number} The default port for this service.
     */
    static get port() {
        return 443;
    }
    /**
     * The scopes needed to make gRPC calls for every method defined
     * in this service.
     * @returns {string[]} List of default scopes.
     */
    static get scopes() {
        return ['https://www.googleapis.com/auth/cloud-platform'];
    }
    /**
     * Return the project ID used by this class.
     * @returns {Promise} A promise that resolves to string containing the project ID.
     */
    getProjectId(callback) {
        if (callback) {
            this.auth.getProjectId(callback);
            return;
        }
        return this.auth.getProjectId();
    }
    getPolicy(request, optionsOrCallback, callback) {
        var _a;
        request = request || {};
        let options;
        if (typeof optionsOrCallback === 'function' && callback === undefined) {
            callback = optionsOrCallback;
            options = {};
        }
        else {
            options = optionsOrCallback;
        }
        options = options || {};
        options.otherArgs = options.otherArgs || {};
        options.otherArgs.headers = options.otherArgs.headers || {};
        options.otherArgs.headers['x-goog-request-params'] =
            this._gaxModule.routingHeader.fromParams({
                name: (_a = request.name) !== null && _a !== void 0 ? _a : '',
            });
        this.initialize();
        return this.innerApiCalls.getPolicy(request, options, callback);
    }
    getEffectivePolicy(request, optionsOrCallback, callback) {
        var _a;
        request = request || {};
        let options;
        if (typeof optionsOrCallback === 'function' && callback === undefined) {
            callback = optionsOrCallback;
            options = {};
        }
        else {
            options = optionsOrCallback;
        }
        options = options || {};
        options.otherArgs = options.otherArgs || {};
        options.otherArgs.headers = options.otherArgs.headers || {};
        options.otherArgs.headers['x-goog-request-params'] =
            this._gaxModule.routingHeader.fromParams({
                name: (_a = request.name) !== null && _a !== void 0 ? _a : '',
            });
        this.initialize();
        return this.innerApiCalls.getEffectivePolicy(request, options, callback);
    }
    createPolicy(request, optionsOrCallback, callback) {
        var _a;
        request = request || {};
        let options;
        if (typeof optionsOrCallback === 'function' && callback === undefined) {
            callback = optionsOrCallback;
            options = {};
        }
        else {
            options = optionsOrCallback;
        }
        options = options || {};
        options.otherArgs = options.otherArgs || {};
        options.otherArgs.headers = options.otherArgs.headers || {};
        options.otherArgs.headers['x-goog-request-params'] =
            this._gaxModule.routingHeader.fromParams({
                parent: (_a = request.parent) !== null && _a !== void 0 ? _a : '',
            });
        this.initialize();
        return this.innerApiCalls.createPolicy(request, options, callback);
    }
    updatePolicy(request, optionsOrCallback, callback) {
        var _a;
        request = request || {};
        let options;
        if (typeof optionsOrCallback === 'function' && callback === undefined) {
            callback = optionsOrCallback;
            options = {};
        }
        else {
            options = optionsOrCallback;
        }
        options = options || {};
        options.otherArgs = options.otherArgs || {};
        options.otherArgs.headers = options.otherArgs.headers || {};
        options.otherArgs.headers['x-goog-request-params'] =
            this._gaxModule.routingHeader.fromParams({
                'policy.name': (_a = request.policy.name) !== null && _a !== void 0 ? _a : '',
            });
        this.initialize();
        return this.innerApiCalls.updatePolicy(request, options, callback);
    }
    deletePolicy(request, optionsOrCallback, callback) {
        var _a;
        request = request || {};
        let options;
        if (typeof optionsOrCallback === 'function' && callback === undefined) {
            callback = optionsOrCallback;
            options = {};
        }
        else {
            options = optionsOrCallback;
        }
        options = options || {};
        options.otherArgs = options.otherArgs || {};
        options.otherArgs.headers = options.otherArgs.headers || {};
        options.otherArgs.headers['x-goog-request-params'] =
            this._gaxModule.routingHeader.fromParams({
                name: (_a = request.name) !== null && _a !== void 0 ? _a : '',
            });
        this.initialize();
        return this.innerApiCalls.deletePolicy(request, options, callback);
    }
    createCustomConstraint(request, optionsOrCallback, callback) {
        var _a;
        request = request || {};
        let options;
        if (typeof optionsOrCallback === 'function' && callback === undefined) {
            callback = optionsOrCallback;
            options = {};
        }
        else {
            options = optionsOrCallback;
        }
        options = options || {};
        options.otherArgs = options.otherArgs || {};
        options.otherArgs.headers = options.otherArgs.headers || {};
        options.otherArgs.headers['x-goog-request-params'] =
            this._gaxModule.routingHeader.fromParams({
                parent: (_a = request.parent) !== null && _a !== void 0 ? _a : '',
            });
        this.initialize();
        return this.innerApiCalls.createCustomConstraint(request, options, callback);
    }
    updateCustomConstraint(request, optionsOrCallback, callback) {
        var _a;
        request = request || {};
        let options;
        if (typeof optionsOrCallback === 'function' && callback === undefined) {
            callback = optionsOrCallback;
            options = {};
        }
        else {
            options = optionsOrCallback;
        }
        options = options || {};
        options.otherArgs = options.otherArgs || {};
        options.otherArgs.headers = options.otherArgs.headers || {};
        options.otherArgs.headers['x-goog-request-params'] =
            this._gaxModule.routingHeader.fromParams({
                'custom_constraint.name': (_a = request.customConstraint.name) !== null && _a !== void 0 ? _a : '',
            });
        this.initialize();
        return this.innerApiCalls.updateCustomConstraint(request, options, callback);
    }
    getCustomConstraint(request, optionsOrCallback, callback) {
        var _a;
        request = request || {};
        let options;
        if (typeof optionsOrCallback === 'function' && callback === undefined) {
            callback = optionsOrCallback;
            options = {};
        }
        else {
            options = optionsOrCallback;
        }
        options = options || {};
        options.otherArgs = options.otherArgs || {};
        options.otherArgs.headers = options.otherArgs.headers || {};
        options.otherArgs.headers['x-goog-request-params'] =
            this._gaxModule.routingHeader.fromParams({
                name: (_a = request.name) !== null && _a !== void 0 ? _a : '',
            });
        this.initialize();
        return this.innerApiCalls.getCustomConstraint(request, options, callback);
    }
    deleteCustomConstraint(request, optionsOrCallback, callback) {
        var _a;
        request = request || {};
        let options;
        if (typeof optionsOrCallback === 'function' && callback === undefined) {
            callback = optionsOrCallback;
            options = {};
        }
        else {
            options = optionsOrCallback;
        }
        options = options || {};
        options.otherArgs = options.otherArgs || {};
        options.otherArgs.headers = options.otherArgs.headers || {};
        options.otherArgs.headers['x-goog-request-params'] =
            this._gaxModule.routingHeader.fromParams({
                name: (_a = request.name) !== null && _a !== void 0 ? _a : '',
            });
        this.initialize();
        return this.innerApiCalls.deleteCustomConstraint(request, options, callback);
    }
    listConstraints(request, optionsOrCallback, callback) {
        var _a;
        request = request || {};
        let options;
        if (typeof optionsOrCallback === 'function' && callback === undefined) {
            callback = optionsOrCallback;
            options = {};
        }
        else {
            options = optionsOrCallback;
        }
        options = options || {};
        options.otherArgs = options.otherArgs || {};
        options.otherArgs.headers = options.otherArgs.headers || {};
        options.otherArgs.headers['x-goog-request-params'] =
            this._gaxModule.routingHeader.fromParams({
                parent: (_a = request.parent) !== null && _a !== void 0 ? _a : '',
            });
        this.initialize();
        return this.innerApiCalls.listConstraints(request, options, callback);
    }
    /**
     * Equivalent to `method.name.toCamelCase()`, but returns a NodeJS Stream object.
     * @param {Object} request
     *   The request object that will be sent.
     * @param {string} request.parent
     *   Required. The Google Cloud resource that parents the constraint. Must be in
     *   one of the following forms:
     *
     *   * `projects/{project_number}`
     *   * `projects/{project_id}`
     *   * `folders/{folder_id}`
     *   * `organizations/{organization_id}`
     * @param {number} request.pageSize
     *   Size of the pages to be returned. This is currently unsupported and will
     *   be ignored. The server may at any point start using this field to limit
     *   page size.
     * @param {string} request.pageToken
     *   Page token used to retrieve the next page. This is currently unsupported
     *   and will be ignored. The server may at any point start using this field.
     * @param {object} [options]
     *   Call options. See {@link https://googleapis.dev/nodejs/google-gax/latest/interfaces/CallOptions.html|CallOptions} for more details.
     * @returns {Stream}
     *   An object stream which emits an object representing {@link protos.google.cloud.orgpolicy.v2.Constraint|Constraint} on 'data' event.
     *   The client library will perform auto-pagination by default: it will call the API as many
     *   times as needed. Note that it can affect your quota.
     *   We recommend using `listConstraintsAsync()`
     *   method described below for async iteration which you can stop as needed.
     *   Please see the {@link https://github.com/googleapis/gax-nodejs/blob/master/client-libraries.md#auto-pagination | documentation }
     *   for more details and examples.
     */
    listConstraintsStream(request, options) {
        var _a;
        request = request || {};
        options = options || {};
        options.otherArgs = options.otherArgs || {};
        options.otherArgs.headers = options.otherArgs.headers || {};
        options.otherArgs.headers['x-goog-request-params'] =
            this._gaxModule.routingHeader.fromParams({
                parent: (_a = request.parent) !== null && _a !== void 0 ? _a : '',
            });
        const defaultCallSettings = this._defaults['listConstraints'];
        const callSettings = defaultCallSettings.merge(options);
        this.initialize();
        return this.descriptors.page.listConstraints.createStream(this.innerApiCalls.listConstraints, request, callSettings);
    }
    /**
     * Equivalent to `listConstraints`, but returns an iterable object.
     *
     * `for`-`await`-`of` syntax is used with the iterable to get response elements on-demand.
     * @param {Object} request
     *   The request object that will be sent.
     * @param {string} request.parent
     *   Required. The Google Cloud resource that parents the constraint. Must be in
     *   one of the following forms:
     *
     *   * `projects/{project_number}`
     *   * `projects/{project_id}`
     *   * `folders/{folder_id}`
     *   * `organizations/{organization_id}`
     * @param {number} request.pageSize
     *   Size of the pages to be returned. This is currently unsupported and will
     *   be ignored. The server may at any point start using this field to limit
     *   page size.
     * @param {string} request.pageToken
     *   Page token used to retrieve the next page. This is currently unsupported
     *   and will be ignored. The server may at any point start using this field.
     * @param {object} [options]
     *   Call options. See {@link https://googleapis.dev/nodejs/google-gax/latest/interfaces/CallOptions.html|CallOptions} for more details.
     * @returns {Object}
     *   An iterable Object that allows {@link https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Iteration_protocols | async iteration }.
     *   When you iterate the returned iterable, each element will be an object representing
     *   {@link protos.google.cloud.orgpolicy.v2.Constraint|Constraint}. The API will be called under the hood as needed, once per the page,
     *   so you can stop the iteration when you don't need more results.
     *   Please see the {@link https://github.com/googleapis/gax-nodejs/blob/master/client-libraries.md#auto-pagination | documentation }
     *   for more details and examples.
     * @example <caption>include:samples/generated/v2/org_policy.list_constraints.js</caption>
     * region_tag:orgpolicy_v2_generated_OrgPolicy_ListConstraints_async
     */
    listConstraintsAsync(request, options) {
        var _a;
        request = request || {};
        options = options || {};
        options.otherArgs = options.otherArgs || {};
        options.otherArgs.headers = options.otherArgs.headers || {};
        options.otherArgs.headers['x-goog-request-params'] =
            this._gaxModule.routingHeader.fromParams({
                parent: (_a = request.parent) !== null && _a !== void 0 ? _a : '',
            });
        const defaultCallSettings = this._defaults['listConstraints'];
        const callSettings = defaultCallSettings.merge(options);
        this.initialize();
        return this.descriptors.page.listConstraints.asyncIterate(this.innerApiCalls['listConstraints'], request, callSettings);
    }
    listPolicies(request, optionsOrCallback, callback) {
        var _a;
        request = request || {};
        let options;
        if (typeof optionsOrCallback === 'function' && callback === undefined) {
            callback = optionsOrCallback;
            options = {};
        }
        else {
            options = optionsOrCallback;
        }
        options = options || {};
        options.otherArgs = options.otherArgs || {};
        options.otherArgs.headers = options.otherArgs.headers || {};
        options.otherArgs.headers['x-goog-request-params'] =
            this._gaxModule.routingHeader.fromParams({
                parent: (_a = request.parent) !== null && _a !== void 0 ? _a : '',
            });
        this.initialize();
        return this.innerApiCalls.listPolicies(request, options, callback);
    }
    /**
     * Equivalent to `method.name.toCamelCase()`, but returns a NodeJS Stream object.
     * @param {Object} request
     *   The request object that will be sent.
     * @param {string} request.parent
     *   Required. The target Google Cloud resource that parents the set of
     *   constraints and policies that will be returned from this call. Must be in
     *   one of the following forms:
     *
     *   * `projects/{project_number}`
     *   * `projects/{project_id}`
     *   * `folders/{folder_id}`
     *   * `organizations/{organization_id}`
     * @param {number} request.pageSize
     *   Size of the pages to be returned. This is currently unsupported and will
     *   be ignored. The server may at any point start using this field to limit
     *   page size.
     * @param {string} request.pageToken
     *   Page token used to retrieve the next page. This is currently unsupported
     *   and will be ignored. The server may at any point start using this field.
     * @param {object} [options]
     *   Call options. See {@link https://googleapis.dev/nodejs/google-gax/latest/interfaces/CallOptions.html|CallOptions} for more details.
     * @returns {Stream}
     *   An object stream which emits an object representing {@link protos.google.cloud.orgpolicy.v2.Policy|Policy} on 'data' event.
     *   The client library will perform auto-pagination by default: it will call the API as many
     *   times as needed. Note that it can affect your quota.
     *   We recommend using `listPoliciesAsync()`
     *   method described below for async iteration which you can stop as needed.
     *   Please see the {@link https://github.com/googleapis/gax-nodejs/blob/master/client-libraries.md#auto-pagination | documentation }
     *   for more details and examples.
     */
    listPoliciesStream(request, options) {
        var _a;
        request = request || {};
        options = options || {};
        options.otherArgs = options.otherArgs || {};
        options.otherArgs.headers = options.otherArgs.headers || {};
        options.otherArgs.headers['x-goog-request-params'] =
            this._gaxModule.routingHeader.fromParams({
                parent: (_a = request.parent) !== null && _a !== void 0 ? _a : '',
            });
        const defaultCallSettings = this._defaults['listPolicies'];
        const callSettings = defaultCallSettings.merge(options);
        this.initialize();
        return this.descriptors.page.listPolicies.createStream(this.innerApiCalls.listPolicies, request, callSettings);
    }
    /**
     * Equivalent to `listPolicies`, but returns an iterable object.
     *
     * `for`-`await`-`of` syntax is used with the iterable to get response elements on-demand.
     * @param {Object} request
     *   The request object that will be sent.
     * @param {string} request.parent
     *   Required. The target Google Cloud resource that parents the set of
     *   constraints and policies that will be returned from this call. Must be in
     *   one of the following forms:
     *
     *   * `projects/{project_number}`
     *   * `projects/{project_id}`
     *   * `folders/{folder_id}`
     *   * `organizations/{organization_id}`
     * @param {number} request.pageSize
     *   Size of the pages to be returned. This is currently unsupported and will
     *   be ignored. The server may at any point start using this field to limit
     *   page size.
     * @param {string} request.pageToken
     *   Page token used to retrieve the next page. This is currently unsupported
     *   and will be ignored. The server may at any point start using this field.
     * @param {object} [options]
     *   Call options. See {@link https://googleapis.dev/nodejs/google-gax/latest/interfaces/CallOptions.html|CallOptions} for more details.
     * @returns {Object}
     *   An iterable Object that allows {@link https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Iteration_protocols | async iteration }.
     *   When you iterate the returned iterable, each element will be an object representing
     *   {@link protos.google.cloud.orgpolicy.v2.Policy|Policy}. The API will be called under the hood as needed, once per the page,
     *   so you can stop the iteration when you don't need more results.
     *   Please see the {@link https://github.com/googleapis/gax-nodejs/blob/master/client-libraries.md#auto-pagination | documentation }
     *   for more details and examples.
     * @example <caption>include:samples/generated/v2/org_policy.list_policies.js</caption>
     * region_tag:orgpolicy_v2_generated_OrgPolicy_ListPolicies_async
     */
    listPoliciesAsync(request, options) {
        var _a;
        request = request || {};
        options = options || {};
        options.otherArgs = options.otherArgs || {};
        options.otherArgs.headers = options.otherArgs.headers || {};
        options.otherArgs.headers['x-goog-request-params'] =
            this._gaxModule.routingHeader.fromParams({
                parent: (_a = request.parent) !== null && _a !== void 0 ? _a : '',
            });
        const defaultCallSettings = this._defaults['listPolicies'];
        const callSettings = defaultCallSettings.merge(options);
        this.initialize();
        return this.descriptors.page.listPolicies.asyncIterate(this.innerApiCalls['listPolicies'], request, callSettings);
    }
    listCustomConstraints(request, optionsOrCallback, callback) {
        var _a;
        request = request || {};
        let options;
        if (typeof optionsOrCallback === 'function' && callback === undefined) {
            callback = optionsOrCallback;
            options = {};
        }
        else {
            options = optionsOrCallback;
        }
        options = options || {};
        options.otherArgs = options.otherArgs || {};
        options.otherArgs.headers = options.otherArgs.headers || {};
        options.otherArgs.headers['x-goog-request-params'] =
            this._gaxModule.routingHeader.fromParams({
                parent: (_a = request.parent) !== null && _a !== void 0 ? _a : '',
            });
        this.initialize();
        return this.innerApiCalls.listCustomConstraints(request, options, callback);
    }
    /**
     * Equivalent to `method.name.toCamelCase()`, but returns a NodeJS Stream object.
     * @param {Object} request
     *   The request object that will be sent.
     * @param {string} request.parent
     *   Required. The target Google Cloud resource that parents the set of custom
     *   constraints that will be returned from this call. Must be in one of the
     *   following forms:
     *
     *   * `organizations/{organization_id}`
     * @param {number} request.pageSize
     *   Size of the pages to be returned. This is currently unsupported and will
     *   be ignored. The server may at any point start using this field to limit
     *   page size.
     * @param {string} request.pageToken
     *   Page token used to retrieve the next page. This is currently unsupported
     *   and will be ignored. The server may at any point start using this field.
     * @param {object} [options]
     *   Call options. See {@link https://googleapis.dev/nodejs/google-gax/latest/interfaces/CallOptions.html|CallOptions} for more details.
     * @returns {Stream}
     *   An object stream which emits an object representing {@link protos.google.cloud.orgpolicy.v2.CustomConstraint|CustomConstraint} on 'data' event.
     *   The client library will perform auto-pagination by default: it will call the API as many
     *   times as needed. Note that it can affect your quota.
     *   We recommend using `listCustomConstraintsAsync()`
     *   method described below for async iteration which you can stop as needed.
     *   Please see the {@link https://github.com/googleapis/gax-nodejs/blob/master/client-libraries.md#auto-pagination | documentation }
     *   for more details and examples.
     */
    listCustomConstraintsStream(request, options) {
        var _a;
        request = request || {};
        options = options || {};
        options.otherArgs = options.otherArgs || {};
        options.otherArgs.headers = options.otherArgs.headers || {};
        options.otherArgs.headers['x-goog-request-params'] =
            this._gaxModule.routingHeader.fromParams({
                parent: (_a = request.parent) !== null && _a !== void 0 ? _a : '',
            });
        const defaultCallSettings = this._defaults['listCustomConstraints'];
        const callSettings = defaultCallSettings.merge(options);
        this.initialize();
        return this.descriptors.page.listCustomConstraints.createStream(this.innerApiCalls.listCustomConstraints, request, callSettings);
    }
    /**
     * Equivalent to `listCustomConstraints`, but returns an iterable object.
     *
     * `for`-`await`-`of` syntax is used with the iterable to get response elements on-demand.
     * @param {Object} request
     *   The request object that will be sent.
     * @param {string} request.parent
     *   Required. The target Google Cloud resource that parents the set of custom
     *   constraints that will be returned from this call. Must be in one of the
     *   following forms:
     *
     *   * `organizations/{organization_id}`
     * @param {number} request.pageSize
     *   Size of the pages to be returned. This is currently unsupported and will
     *   be ignored. The server may at any point start using this field to limit
     *   page size.
     * @param {string} request.pageToken
     *   Page token used to retrieve the next page. This is currently unsupported
     *   and will be ignored. The server may at any point start using this field.
     * @param {object} [options]
     *   Call options. See {@link https://googleapis.dev/nodejs/google-gax/latest/interfaces/CallOptions.html|CallOptions} for more details.
     * @returns {Object}
     *   An iterable Object that allows {@link https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Iteration_protocols | async iteration }.
     *   When you iterate the returned iterable, each element will be an object representing
     *   {@link protos.google.cloud.orgpolicy.v2.CustomConstraint|CustomConstraint}. The API will be called under the hood as needed, once per the page,
     *   so you can stop the iteration when you don't need more results.
     *   Please see the {@link https://github.com/googleapis/gax-nodejs/blob/master/client-libraries.md#auto-pagination | documentation }
     *   for more details and examples.
     * @example <caption>include:samples/generated/v2/org_policy.list_custom_constraints.js</caption>
     * region_tag:orgpolicy_v2_generated_OrgPolicy_ListCustomConstraints_async
     */
    listCustomConstraintsAsync(request, options) {
        var _a;
        request = request || {};
        options = options || {};
        options.otherArgs = options.otherArgs || {};
        options.otherArgs.headers = options.otherArgs.headers || {};
        options.otherArgs.headers['x-goog-request-params'] =
            this._gaxModule.routingHeader.fromParams({
                parent: (_a = request.parent) !== null && _a !== void 0 ? _a : '',
            });
        const defaultCallSettings = this._defaults['listCustomConstraints'];
        const callSettings = defaultCallSettings.merge(options);
        this.initialize();
        return this.descriptors.page.listCustomConstraints.asyncIterate(this.innerApiCalls['listCustomConstraints'], request, callSettings);
    }
    // --------------------
    // -- Path templates --
    // --------------------
    /**
     * Return a fully-qualified customConstraint resource name string.
     *
     * @param {string} organization
     * @param {string} custom_constraint
     * @returns {string} Resource name string.
     */
    customConstraintPath(organization, customConstraint) {
        return this.pathTemplates.customConstraintPathTemplate.render({
            organization: organization,
            custom_constraint: customConstraint,
        });
    }
    /**
     * Parse the organization from CustomConstraint resource.
     *
     * @param {string} customConstraintName
     *   A fully-qualified path representing CustomConstraint resource.
     * @returns {string} A string representing the organization.
     */
    matchOrganizationFromCustomConstraintName(customConstraintName) {
        return this.pathTemplates.customConstraintPathTemplate.match(customConstraintName).organization;
    }
    /**
     * Parse the custom_constraint from CustomConstraint resource.
     *
     * @param {string} customConstraintName
     *   A fully-qualified path representing CustomConstraint resource.
     * @returns {string} A string representing the custom_constraint.
     */
    matchCustomConstraintFromCustomConstraintName(customConstraintName) {
        return this.pathTemplates.customConstraintPathTemplate.match(customConstraintName).custom_constraint;
    }
    /**
     * Return a fully-qualified folderConstraint resource name string.
     *
     * @param {string} folder
     * @param {string} constraint
     * @returns {string} Resource name string.
     */
    folderConstraintPath(folder, constraint) {
        return this.pathTemplates.folderConstraintPathTemplate.render({
            folder: folder,
            constraint: constraint,
        });
    }
    /**
     * Parse the folder from FolderConstraint resource.
     *
     * @param {string} folderConstraintName
     *   A fully-qualified path representing folder_constraint resource.
     * @returns {string} A string representing the folder.
     */
    matchFolderFromFolderConstraintName(folderConstraintName) {
        return this.pathTemplates.folderConstraintPathTemplate.match(folderConstraintName).folder;
    }
    /**
     * Parse the constraint from FolderConstraint resource.
     *
     * @param {string} folderConstraintName
     *   A fully-qualified path representing folder_constraint resource.
     * @returns {string} A string representing the constraint.
     */
    matchConstraintFromFolderConstraintName(folderConstraintName) {
        return this.pathTemplates.folderConstraintPathTemplate.match(folderConstraintName).constraint;
    }
    /**
     * Return a fully-qualified folderPolicy resource name string.
     *
     * @param {string} folder
     * @param {string} policy
     * @returns {string} Resource name string.
     */
    folderPolicyPath(folder, policy) {
        return this.pathTemplates.folderPolicyPathTemplate.render({
            folder: folder,
            policy: policy,
        });
    }
    /**
     * Parse the folder from FolderPolicy resource.
     *
     * @param {string} folderPolicyName
     *   A fully-qualified path representing folder_policy resource.
     * @returns {string} A string representing the folder.
     */
    matchFolderFromFolderPolicyName(folderPolicyName) {
        return this.pathTemplates.folderPolicyPathTemplate.match(folderPolicyName)
            .folder;
    }
    /**
     * Parse the policy from FolderPolicy resource.
     *
     * @param {string} folderPolicyName
     *   A fully-qualified path representing folder_policy resource.
     * @returns {string} A string representing the policy.
     */
    matchPolicyFromFolderPolicyName(folderPolicyName) {
        return this.pathTemplates.folderPolicyPathTemplate.match(folderPolicyName)
            .policy;
    }
    /**
     * Return a fully-qualified organization resource name string.
     *
     * @param {string} organization
     * @returns {string} Resource name string.
     */
    organizationPath(organization) {
        return this.pathTemplates.organizationPathTemplate.render({
            organization: organization,
        });
    }
    /**
     * Parse the organization from Organization resource.
     *
     * @param {string} organizationName
     *   A fully-qualified path representing Organization resource.
     * @returns {string} A string representing the organization.
     */
    matchOrganizationFromOrganizationName(organizationName) {
        return this.pathTemplates.organizationPathTemplate.match(organizationName)
            .organization;
    }
    /**
     * Return a fully-qualified organizationConstraint resource name string.
     *
     * @param {string} organization
     * @param {string} constraint
     * @returns {string} Resource name string.
     */
    organizationConstraintPath(organization, constraint) {
        return this.pathTemplates.organizationConstraintPathTemplate.render({
            organization: organization,
            constraint: constraint,
        });
    }
    /**
     * Parse the organization from OrganizationConstraint resource.
     *
     * @param {string} organizationConstraintName
     *   A fully-qualified path representing organization_constraint resource.
     * @returns {string} A string representing the organization.
     */
    matchOrganizationFromOrganizationConstraintName(organizationConstraintName) {
        return this.pathTemplates.organizationConstraintPathTemplate.match(organizationConstraintName).organization;
    }
    /**
     * Parse the constraint from OrganizationConstraint resource.
     *
     * @param {string} organizationConstraintName
     *   A fully-qualified path representing organization_constraint resource.
     * @returns {string} A string representing the constraint.
     */
    matchConstraintFromOrganizationConstraintName(organizationConstraintName) {
        return this.pathTemplates.organizationConstraintPathTemplate.match(organizationConstraintName).constraint;
    }
    /**
     * Return a fully-qualified organizationPolicy resource name string.
     *
     * @param {string} organization
     * @param {string} policy
     * @returns {string} Resource name string.
     */
    organizationPolicyPath(organization, policy) {
        return this.pathTemplates.organizationPolicyPathTemplate.render({
            organization: organization,
            policy: policy,
        });
    }
    /**
     * Parse the organization from OrganizationPolicy resource.
     *
     * @param {string} organizationPolicyName
     *   A fully-qualified path representing organization_policy resource.
     * @returns {string} A string representing the organization.
     */
    matchOrganizationFromOrganizationPolicyName(organizationPolicyName) {
        return this.pathTemplates.organizationPolicyPathTemplate.match(organizationPolicyName).organization;
    }
    /**
     * Parse the policy from OrganizationPolicy resource.
     *
     * @param {string} organizationPolicyName
     *   A fully-qualified path representing organization_policy resource.
     * @returns {string} A string representing the policy.
     */
    matchPolicyFromOrganizationPolicyName(organizationPolicyName) {
        return this.pathTemplates.organizationPolicyPathTemplate.match(organizationPolicyName).policy;
    }
    /**
     * Return a fully-qualified project resource name string.
     *
     * @param {string} project
     * @returns {string} Resource name string.
     */
    projectPath(project) {
        return this.pathTemplates.projectPathTemplate.render({
            project: project,
        });
    }
    /**
     * Parse the project from Project resource.
     *
     * @param {string} projectName
     *   A fully-qualified path representing Project resource.
     * @returns {string} A string representing the project.
     */
    matchProjectFromProjectName(projectName) {
        return this.pathTemplates.projectPathTemplate.match(projectName).project;
    }
    /**
     * Return a fully-qualified projectConstraint resource name string.
     *
     * @param {string} project
     * @param {string} constraint
     * @returns {string} Resource name string.
     */
    projectConstraintPath(project, constraint) {
        return this.pathTemplates.projectConstraintPathTemplate.render({
            project: project,
            constraint: constraint,
        });
    }
    /**
     * Parse the project from ProjectConstraint resource.
     *
     * @param {string} projectConstraintName
     *   A fully-qualified path representing project_constraint resource.
     * @returns {string} A string representing the project.
     */
    matchProjectFromProjectConstraintName(projectConstraintName) {
        return this.pathTemplates.projectConstraintPathTemplate.match(projectConstraintName).project;
    }
    /**
     * Parse the constraint from ProjectConstraint resource.
     *
     * @param {string} projectConstraintName
     *   A fully-qualified path representing project_constraint resource.
     * @returns {string} A string representing the constraint.
     */
    matchConstraintFromProjectConstraintName(projectConstraintName) {
        return this.pathTemplates.projectConstraintPathTemplate.match(projectConstraintName).constraint;
    }
    /**
     * Return a fully-qualified projectPolicy resource name string.
     *
     * @param {string} project
     * @param {string} policy
     * @returns {string} Resource name string.
     */
    projectPolicyPath(project, policy) {
        return this.pathTemplates.projectPolicyPathTemplate.render({
            project: project,
            policy: policy,
        });
    }
    /**
     * Parse the project from ProjectPolicy resource.
     *
     * @param {string} projectPolicyName
     *   A fully-qualified path representing project_policy resource.
     * @returns {string} A string representing the project.
     */
    matchProjectFromProjectPolicyName(projectPolicyName) {
        return this.pathTemplates.projectPolicyPathTemplate.match(projectPolicyName)
            .project;
    }
    /**
     * Parse the policy from ProjectPolicy resource.
     *
     * @param {string} projectPolicyName
     *   A fully-qualified path representing project_policy resource.
     * @returns {string} A string representing the policy.
     */
    matchPolicyFromProjectPolicyName(projectPolicyName) {
        return this.pathTemplates.projectPolicyPathTemplate.match(projectPolicyName)
            .policy;
    }
    /**
     * Terminate the gRPC channel and close the client.
     *
     * The client will no longer be usable and all future behavior is undefined.
     * @returns {Promise} A promise that resolves when the client is closed.
     */
    close() {
        if (this.orgPolicyStub && !this._terminated) {
            return this.orgPolicyStub.then(stub => {
                this._terminated = true;
                stub.close();
            });
        }
        return Promise.resolve();
    }
}
exports.OrgPolicyClient = OrgPolicyClient;
