// @azure/arm-eventgrid/dist-esm/test/eventgrid_examples.js
/*
 * Copyright (c) Microsoft Corporation.
 * Licensed under the MIT License.
 *
 * Code generated by Microsoft (R) AutoRest Code Generator.
 * Changes may cause incorrect behavior and will be lost if the code is regenerated.
 */
import { __asyncValues, __awaiter } from "tslib";
import { env, Recorder, isPlaybackMode, } from "@azure-tools/test-recorder";
import { createTestCredential } from "@azure-tools/test-credential";
import { assert } from "chai";
import { EventGridManagementClient } from "../src/eventGridManagementClient";
const replaceableVariables = {
    AZURE_CLIENT_ID: "azure_client_id",
    AZURE_CLIENT_SECRET: "azure_client_secret",
    AZURE_TENANT_ID: "88888888-8888-8888-8888-888888888888",
    SUBSCRIPTION_ID: "azure_subscription_id"
};
const recorderOptions = {
    envSetupForPlayback: replaceableVariables
};
export const testPollingOptions = {
    updateIntervalInMs: isPlaybackMode() ? 0 : undefined,
};
describe("Eventgrid test", () => {
    let recorder;
    let subscriptionId;
    let client;
    let location;
    let resourceGroupName;
    let topicName;
    let domainName;
    beforeEach(function () {
        return __awaiter(this, void 0, void 0, function* () {
            recorder = new Recorder(this.currentTest);
            yield recorder.start(recorderOptions);
            subscriptionId = env.SUBSCRIPTION_ID || '';
            // This is an example of how the environment variables are used
            const credential = createTestCredential();
            client = new EventGridManagementClient(credential, subscriptionId, recorder.configureClientOptions({}));
            location = "eastus2";
            resourceGroupName = "myjstest";
            topicName = "mytopicxxx";
            domainName = "mydomainxxx";
        });
    });
    afterEach(function () {
        return __awaiter(this, void 0, void 0, function* () {
            yield recorder.stop();
        });
    });
    it("topics create test", function () {
        return __awaiter(this, void 0, void 0, function* () {
            const res = yield client.topics.beginCreateOrUpdateAndWait(resourceGroupName, topicName, { location: "eastus2" }, testPollingOptions);
            assert.equal(res.name, topicName);
        });
    });
    it("topics listByResourceGroup test", function () {
        var e_1, _a;
        return __awaiter(this, void 0, void 0, function* () {
            const resArray = new Array();
            try {
                for (var _b = __asyncValues(client.topics.listByResourceGroup(resourceGroupName)), _c; _c = yield _b.next(), !_c.done;) {
                    let item = _c.value;
                    resArray.push(item);
                }
            }
            catch (e_1_1) { e_1 = { error: e_1_1 }; }
            finally {
                try {
                    if (_c && !_c.done && (_a = _b.return)) yield _a.call(_b);
                }
                finally { if (e_1) throw e_1.error; }
            }
            assert.equal(resArray.length, 1);
        });
    });
    it("topics delete test", function () {
        var e_2, _a;
        return __awaiter(this, void 0, void 0, function* () {
            const res = yield client.topics.beginDeleteAndWait(resourceGroupName, topicName, testPollingOptions);
            const resArray = new Array();
            try {
                for (var _b = __asyncValues(client.topics.listByResourceGroup(resourceGroupName)), _c; _c = yield _b.next(), !_c.done;) {
                    let item = _c.value;
                    resArray.push(item);
                }
            }
            catch (e_2_1) { e_2 = { error: e_2_1 }; }
            finally {
                try {
                    if (_c && !_c.done && (_a = _b.return)) yield _a.call(_b);
                }
                finally { if (e_2) throw e_2.error; }
            }
            assert.equal(resArray.length, 0);
        });
    });
    it("domains create test", function () {
        return __awaiter(this, void 0, void 0, function* () {
            const res = yield client.domains.beginCreateOrUpdateAndWait(resourceGroupName, domainName, { location: location }, testPollingOptions);
            assert.equal(res.name, domainName);
        });
    });
    it("domains update test", function () {
        return __awaiter(this, void 0, void 0, function* () {
            const res = yield client.domains.beginUpdateAndWait(resourceGroupName, domainName, {
                tags: {
                    tag1: "value1",
                    tag2: "value2"
                }
            }, testPollingOptions);
            //It's void response
        });
    });
    it("domains get test", function () {
        return __awaiter(this, void 0, void 0, function* () {
            const res = yield client.domains.get(resourceGroupName, domainName);
            assert.equal(res.name, domainName);
        });
    });
    it("domains listByResourceGroup test", function () {
        var e_3, _a;
        return __awaiter(this, void 0, void 0, function* () {
            const resArray = new Array();
            try {
                for (var _b = __asyncValues(client.domains.listByResourceGroup(resourceGroupName)), _c; _c = yield _b.next(), !_c.done;) {
                    let item = _c.value;
                    resArray.push(item);
                }
            }
            catch (e_3_1) { e_3 = { error: e_3_1 }; }
            finally {
                try {
                    if (_c && !_c.done && (_a = _b.return)) yield _a.call(_b);
                }
                finally { if (e_3) throw e_3.error; }
            }
            assert.equal(resArray.length, 1);
        });
    });
    it("domains delete test", function () {
        var e_4, _a;
        return __awaiter(this, void 0, void 0, function* () {
            const res = yield client.domains.beginDeleteAndWait(resourceGroupName, domainName, testPollingOptions);
            const resArray = new Array();
            try {
                for (var _b = __asyncValues(client.domains.listByResourceGroup(resourceGroupName)), _c; _c = yield _b.next(), !_c.done;) {
                    let item = _c.value;
                    resArray.push(item);
                }
            }
            catch (e_4_1) { e_4 = { error: e_4_1 }; }
            finally {
                try {
                    if (_c && !_c.done && (_a = _b.return)) yield _a.call(_b);
                }
                finally { if (e_4) throw e_4.error; }
            }
            assert.equal(resArray.length, 0);
        });
    });
});
