// @azure/arm-desktopvirtualization/dist-esm/src/operations/msixImages.js
/*
 * Copyright (c) Microsoft Corporation.
 * Licensed under the MIT License.
 *
 * Code generated by Microsoft (R) AutoRest Code Generator.
 * Changes may cause incorrect behavior and will be lost if the code is regenerated.
 */
import { __asyncDelegator, __asyncGenerator, __asyncValues, __await } from "tslib";
import { setContinuationToken } from "../pagingHelper";
import * as coreClient from "@azure/core-client";
import * as Mappers from "../models/mappers";
import * as Parameters from "../models/parameters";
/// <reference lib="esnext.asynciterable" />
/** Class containing MsixImages operations. */
export class MsixImagesImpl {
    /**
     * Initialize a new instance of the class MsixImages class.
     * @param client Reference to the service client
     */
    constructor(client) {
        this.client = client;
    }
    /**
     * Expands and Lists MSIX packages in an Image, given the Image Path.
     * @param resourceGroupName The name of the resource group. The name is case insensitive.
     * @param hostPoolName The name of the host pool within the specified resource group
     * @param msixImageURI Object containing URI to MSIX Image
     * @param options The options parameters.
     */
    listExpand(resourceGroupName, hostPoolName, msixImageURI, options) {
        const iter = this.expandPagingAll(resourceGroupName, hostPoolName, msixImageURI, options);
        return {
            next() {
                return iter.next();
            },
            [Symbol.asyncIterator]() {
                return this;
            },
            byPage: (settings) => {
                if (settings === null || settings === void 0 ? void 0 : settings.maxPageSize) {
                    throw new Error("maxPageSize is not supported by this operation.");
                }
                return this.expandPagingPage(resourceGroupName, hostPoolName, msixImageURI, options, settings);
            }
        };
    }
    expandPagingPage(resourceGroupName, hostPoolName, msixImageURI, options, settings) {
        return __asyncGenerator(this, arguments, function* expandPagingPage_1() {
            let result;
            let continuationToken = settings === null || settings === void 0 ? void 0 : settings.continuationToken;
            if (!continuationToken) {
                result = yield __await(this._expand(resourceGroupName, hostPoolName, msixImageURI, options));
                let page = result.value || [];
                continuationToken = result.nextLink;
                setContinuationToken(page, continuationToken);
                yield yield __await(page);
            }
            while (continuationToken) {
                result = yield __await(this._expandNext(resourceGroupName, hostPoolName, msixImageURI, continuationToken, options));
                continuationToken = result.nextLink;
                let page = result.value || [];
                setContinuationToken(page, continuationToken);
                yield yield __await(page);
            }
        });
    }
    expandPagingAll(resourceGroupName, hostPoolName, msixImageURI, options) {
        return __asyncGenerator(this, arguments, function* expandPagingAll_1() {
            var _a, e_1, _b, _c;
            try {
                for (var _d = true, _e = __asyncValues(this.expandPagingPage(resourceGroupName, hostPoolName, msixImageURI, options)), _f; _f = yield __await(_e.next()), _a = _f.done, !_a;) {
                    _c = _f.value;
                    _d = false;
                    try {
                        const page = _c;
                        yield __await(yield* __asyncDelegator(__asyncValues(page)));
                    }
                    finally {
                        _d = true;
                    }
                }
            }
            catch (e_1_1) { e_1 = { error: e_1_1 }; }
            finally {
                try {
                    if (!_d && !_a && (_b = _e.return)) yield __await(_b.call(_e));
                }
                finally { if (e_1) throw e_1.error; }
            }
        });
    }
    /**
     * Expands and Lists MSIX packages in an Image, given the Image Path.
     * @param resourceGroupName The name of the resource group. The name is case insensitive.
     * @param hostPoolName The name of the host pool within the specified resource group
     * @param msixImageURI Object containing URI to MSIX Image
     * @param options The options parameters.
     */
    _expand(resourceGroupName, hostPoolName, msixImageURI, options) {
        return this.client.sendOperationRequest({ resourceGroupName, hostPoolName, msixImageURI, options }, expandOperationSpec);
    }
    /**
     * ExpandNext
     * @param resourceGroupName The name of the resource group. The name is case insensitive.
     * @param hostPoolName The name of the host pool within the specified resource group
     * @param msixImageURI Object containing URI to MSIX Image
     * @param nextLink The nextLink from the previous successful call to the Expand method.
     * @param options The options parameters.
     */
    _expandNext(resourceGroupName, hostPoolName, msixImageURI, nextLink, options) {
        return this.client.sendOperationRequest({ resourceGroupName, hostPoolName, msixImageURI, nextLink, options }, expandNextOperationSpec);
    }
}
// Operation Specifications
const serializer = coreClient.createSerializer(Mappers, /* isXml */ false);
const expandOperationSpec = {
    path: "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.DesktopVirtualization/hostPools/{hostPoolName}/expandMsixImage",
    httpMethod: "POST",
    responses: {
        200: {
            bodyMapper: Mappers.ExpandMsixImageList
        },
        default: {
            bodyMapper: Mappers.CloudError
        }
    },
    requestBody: Parameters.msixImageURI,
    queryParameters: [Parameters.apiVersion],
    urlParameters: [
        Parameters.$host,
        Parameters.subscriptionId,
        Parameters.resourceGroupName,
        Parameters.hostPoolName
    ],
    headerParameters: [Parameters.accept, Parameters.contentType],
    mediaType: "json",
    serializer
};
const expandNextOperationSpec = {
    path: "{nextLink}",
    httpMethod: "GET",
    responses: {
        200: {
            bodyMapper: Mappers.ExpandMsixImageList
        },
        default: {
            bodyMapper: Mappers.CloudError
        }
    },
    urlParameters: [
        Parameters.$host,
        Parameters.nextLink,
        Parameters.subscriptionId,
        Parameters.resourceGroupName,
        Parameters.hostPoolName
    ],
    headerParameters: [Parameters.accept, Parameters.contentType],
    mediaType: "json",
    serializer
};
