// @azure/arm-desktopvirtualization/dist-esm/src/operations/scalingPlanPersonalSchedules.js
/*
 * Copyright (c) Microsoft Corporation.
 * Licensed under the MIT License.
 *
 * Code generated by Microsoft (R) AutoRest Code Generator.
 * Changes may cause incorrect behavior and will be lost if the code is regenerated.
 */
import { __asyncDelegator, __asyncGenerator, __asyncValues, __await } from "tslib";
import { setContinuationToken } from "../pagingHelper";
import * as coreClient from "@azure/core-client";
import * as Mappers from "../models/mappers";
import * as Parameters from "../models/parameters";
/// <reference lib="esnext.asynciterable" />
/** Class containing ScalingPlanPersonalSchedules operations. */
export class ScalingPlanPersonalSchedulesImpl {
    /**
     * Initialize a new instance of the class ScalingPlanPersonalSchedules class.
     * @param client Reference to the service client
     */
    constructor(client) {
        this.client = client;
    }
    /**
     * List ScalingPlanPersonalSchedules.
     * @param resourceGroupName The name of the resource group. The name is case insensitive.
     * @param scalingPlanName The name of the scaling plan.
     * @param options The options parameters.
     */
    list(resourceGroupName, scalingPlanName, options) {
        const iter = this.listPagingAll(resourceGroupName, scalingPlanName, options);
        return {
            next() {
                return iter.next();
            },
            [Symbol.asyncIterator]() {
                return this;
            },
            byPage: (settings) => {
                if (settings === null || settings === void 0 ? void 0 : settings.maxPageSize) {
                    throw new Error("maxPageSize is not supported by this operation.");
                }
                return this.listPagingPage(resourceGroupName, scalingPlanName, options, settings);
            }
        };
    }
    listPagingPage(resourceGroupName, scalingPlanName, options, settings) {
        return __asyncGenerator(this, arguments, function* listPagingPage_1() {
            let result;
            let continuationToken = settings === null || settings === void 0 ? void 0 : settings.continuationToken;
            if (!continuationToken) {
                result = yield __await(this._list(resourceGroupName, scalingPlanName, options));
                let page = result.value || [];
                continuationToken = result.nextLink;
                setContinuationToken(page, continuationToken);
                yield yield __await(page);
            }
            while (continuationToken) {
                result = yield __await(this._listNext(resourceGroupName, scalingPlanName, continuationToken, options));
                continuationToken = result.nextLink;
                let page = result.value || [];
                setContinuationToken(page, continuationToken);
                yield yield __await(page);
            }
        });
    }
    listPagingAll(resourceGroupName, scalingPlanName, options) {
        return __asyncGenerator(this, arguments, function* listPagingAll_1() {
            var _a, e_1, _b, _c;
            try {
                for (var _d = true, _e = __asyncValues(this.listPagingPage(resourceGroupName, scalingPlanName, options)), _f; _f = yield __await(_e.next()), _a = _f.done, !_a;) {
                    _c = _f.value;
                    _d = false;
                    try {
                        const page = _c;
                        yield __await(yield* __asyncDelegator(__asyncValues(page)));
                    }
                    finally {
                        _d = true;
                    }
                }
            }
            catch (e_1_1) { e_1 = { error: e_1_1 }; }
            finally {
                try {
                    if (!_d && !_a && (_b = _e.return)) yield __await(_b.call(_e));
                }
                finally { if (e_1) throw e_1.error; }
            }
        });
    }
    /**
     * Get a ScalingPlanPersonalSchedule.
     * @param resourceGroupName The name of the resource group. The name is case insensitive.
     * @param scalingPlanName The name of the scaling plan.
     * @param scalingPlanScheduleName The name of the ScalingPlanSchedule
     * @param options The options parameters.
     */
    get(resourceGroupName, scalingPlanName, scalingPlanScheduleName, options) {
        return this.client.sendOperationRequest({ resourceGroupName, scalingPlanName, scalingPlanScheduleName, options }, getOperationSpec);
    }
    /**
     * Create or update a ScalingPlanPersonalSchedule.
     * @param resourceGroupName The name of the resource group. The name is case insensitive.
     * @param scalingPlanName The name of the scaling plan.
     * @param scalingPlanScheduleName The name of the ScalingPlanSchedule
     * @param scalingPlanSchedule Object containing ScalingPlanPersonalSchedule definitions.
     * @param options The options parameters.
     */
    create(resourceGroupName, scalingPlanName, scalingPlanScheduleName, scalingPlanSchedule, options) {
        return this.client.sendOperationRequest({
            resourceGroupName,
            scalingPlanName,
            scalingPlanScheduleName,
            scalingPlanSchedule,
            options
        }, createOperationSpec);
    }
    /**
     * Remove a ScalingPlanPersonalSchedule.
     * @param resourceGroupName The name of the resource group. The name is case insensitive.
     * @param scalingPlanName The name of the scaling plan.
     * @param scalingPlanScheduleName The name of the ScalingPlanSchedule
     * @param options The options parameters.
     */
    delete(resourceGroupName, scalingPlanName, scalingPlanScheduleName, options) {
        return this.client.sendOperationRequest({ resourceGroupName, scalingPlanName, scalingPlanScheduleName, options }, deleteOperationSpec);
    }
    /**
     * Update a ScalingPlanPersonalSchedule.
     * @param resourceGroupName The name of the resource group. The name is case insensitive.
     * @param scalingPlanName The name of the scaling plan.
     * @param scalingPlanScheduleName The name of the ScalingPlanSchedule
     * @param options The options parameters.
     */
    update(resourceGroupName, scalingPlanName, scalingPlanScheduleName, options) {
        return this.client.sendOperationRequest({ resourceGroupName, scalingPlanName, scalingPlanScheduleName, options }, updateOperationSpec);
    }
    /**
     * List ScalingPlanPersonalSchedules.
     * @param resourceGroupName The name of the resource group. The name is case insensitive.
     * @param scalingPlanName The name of the scaling plan.
     * @param options The options parameters.
     */
    _list(resourceGroupName, scalingPlanName, options) {
        return this.client.sendOperationRequest({ resourceGroupName, scalingPlanName, options }, listOperationSpec);
    }
    /**
     * ListNext
     * @param resourceGroupName The name of the resource group. The name is case insensitive.
     * @param scalingPlanName The name of the scaling plan.
     * @param nextLink The nextLink from the previous successful call to the List method.
     * @param options The options parameters.
     */
    _listNext(resourceGroupName, scalingPlanName, nextLink, options) {
        return this.client.sendOperationRequest({ resourceGroupName, scalingPlanName, nextLink, options }, listNextOperationSpec);
    }
}
// Operation Specifications
const serializer = coreClient.createSerializer(Mappers, /* isXml */ false);
const getOperationSpec = {
    path: "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.DesktopVirtualization/scalingPlans/{scalingPlanName}/personalSchedules/{scalingPlanScheduleName}",
    httpMethod: "GET",
    responses: {
        200: {
            bodyMapper: Mappers.ScalingPlanPersonalSchedule
        },
        default: {
            bodyMapper: Mappers.CloudError
        }
    },
    queryParameters: [Parameters.apiVersion],
    urlParameters: [
        Parameters.$host,
        Parameters.subscriptionId,
        Parameters.resourceGroupName,
        Parameters.scalingPlanName,
        Parameters.scalingPlanScheduleName
    ],
    headerParameters: [Parameters.accept],
    serializer
};
const createOperationSpec = {
    path: "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.DesktopVirtualization/scalingPlans/{scalingPlanName}/personalSchedules/{scalingPlanScheduleName}",
    httpMethod: "PUT",
    responses: {
        200: {
            bodyMapper: Mappers.ScalingPlanPersonalSchedule
        },
        201: {
            bodyMapper: Mappers.ScalingPlanPersonalSchedule
        },
        default: {
            bodyMapper: Mappers.CloudError
        }
    },
    requestBody: Parameters.scalingPlanSchedule2,
    queryParameters: [Parameters.apiVersion],
    urlParameters: [
        Parameters.$host,
        Parameters.subscriptionId,
        Parameters.resourceGroupName,
        Parameters.scalingPlanName,
        Parameters.scalingPlanScheduleName
    ],
    headerParameters: [Parameters.accept, Parameters.contentType],
    mediaType: "json",
    serializer
};
const deleteOperationSpec = {
    path: "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.DesktopVirtualization/scalingPlans/{scalingPlanName}/personalSchedules/{scalingPlanScheduleName}",
    httpMethod: "DELETE",
    responses: {
        200: {},
        204: {},
        default: {
            bodyMapper: Mappers.CloudError
        }
    },
    queryParameters: [Parameters.apiVersion],
    urlParameters: [
        Parameters.$host,
        Parameters.subscriptionId,
        Parameters.resourceGroupName,
        Parameters.scalingPlanName,
        Parameters.scalingPlanScheduleName
    ],
    headerParameters: [Parameters.accept],
    serializer
};
const updateOperationSpec = {
    path: "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.DesktopVirtualization/scalingPlans/{scalingPlanName}/personalSchedules/{scalingPlanScheduleName}",
    httpMethod: "PATCH",
    responses: {
        200: {
            bodyMapper: Mappers.ScalingPlanPersonalSchedule
        },
        default: {
            bodyMapper: Mappers.CloudError
        }
    },
    requestBody: Parameters.scalingPlanSchedule3,
    queryParameters: [Parameters.apiVersion],
    urlParameters: [
        Parameters.$host,
        Parameters.subscriptionId,
        Parameters.resourceGroupName,
        Parameters.scalingPlanName,
        Parameters.scalingPlanScheduleName
    ],
    headerParameters: [Parameters.accept, Parameters.contentType],
    mediaType: "json",
    serializer
};
const listOperationSpec = {
    path: "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.DesktopVirtualization/scalingPlans/{scalingPlanName}/personalSchedules",
    httpMethod: "GET",
    responses: {
        200: {
            bodyMapper: Mappers.ScalingPlanPersonalScheduleList
        },
        default: {
            bodyMapper: Mappers.CloudError
        }
    },
    queryParameters: [
        Parameters.apiVersion,
        Parameters.pageSize,
        Parameters.isDescending,
        Parameters.initialSkip
    ],
    urlParameters: [
        Parameters.$host,
        Parameters.subscriptionId,
        Parameters.resourceGroupName,
        Parameters.scalingPlanName
    ],
    headerParameters: [Parameters.accept],
    serializer
};
const listNextOperationSpec = {
    path: "{nextLink}",
    httpMethod: "GET",
    responses: {
        200: {
            bodyMapper: Mappers.ScalingPlanPersonalScheduleList
        },
        default: {
            bodyMapper: Mappers.CloudError
        }
    },
    urlParameters: [
        Parameters.$host,
        Parameters.nextLink,
        Parameters.subscriptionId,
        Parameters.resourceGroupName,
        Parameters.scalingPlanName
    ],
    headerParameters: [Parameters.accept],
    serializer
};
