// @azure/arm-servicebus/dist-esm/src/models/parameters.js
/*
 * Copyright (c) Microsoft Corporation.
 * Licensed under the MIT License.
 *
 * Code generated by Microsoft (R) AutoRest Code Generator.
 * Changes may cause incorrect behavior and will be lost if the code is regenerated.
 */
import { SBNamespace as SBNamespaceMapper, SBNamespaceUpdateParameters as SBNamespaceUpdateParametersMapper, NetworkRuleSet as NetworkRuleSetMapper, SBAuthorizationRule as SBAuthorizationRuleMapper, RegenerateAccessKeyParameters as RegenerateAccessKeyParametersMapper, CheckNameAvailability as CheckNameAvailabilityMapper, PrivateEndpointConnection as PrivateEndpointConnectionMapper, ArmDisasterRecovery as ArmDisasterRecoveryMapper, FailoverProperties as FailoverPropertiesMapper, MigrationConfigProperties as MigrationConfigPropertiesMapper, SBQueue as SBQueueMapper, SBTopic as SBTopicMapper, Rule as RuleMapper, SBSubscription as SBSubscriptionMapper } from "../models/mappers";
export const accept = {
    parameterPath: "accept",
    mapper: {
        defaultValue: "application/json",
        isConstant: true,
        serializedName: "Accept",
        type: {
            name: "String"
        }
    }
};
export const $host = {
    parameterPath: "$host",
    mapper: {
        serializedName: "$host",
        required: true,
        type: {
            name: "String"
        }
    },
    skipEncoding: true
};
export const apiVersion = {
    parameterPath: "apiVersion",
    mapper: {
        defaultValue: "2021-11-01",
        isConstant: true,
        serializedName: "api-version",
        type: {
            name: "String"
        }
    }
};
export const subscriptionId = {
    parameterPath: "subscriptionId",
    mapper: {
        serializedName: "subscriptionId",
        required: true,
        type: {
            name: "String"
        }
    }
};
export const resourceGroupName = {
    parameterPath: "resourceGroupName",
    mapper: {
        constraints: {
            MaxLength: 90,
            MinLength: 1
        },
        serializedName: "resourceGroupName",
        required: true,
        type: {
            name: "String"
        }
    }
};
export const contentType = {
    parameterPath: ["options", "contentType"],
    mapper: {
        defaultValue: "application/json",
        isConstant: true,
        serializedName: "Content-Type",
        type: {
            name: "String"
        }
    }
};
export const parameters = {
    parameterPath: "parameters",
    mapper: SBNamespaceMapper
};
export const namespaceName = {
    parameterPath: "namespaceName",
    mapper: {
        serializedName: "namespaceName",
        required: true,
        type: {
            name: "String"
        }
    }
};
export const namespaceName1 = {
    parameterPath: "namespaceName",
    mapper: {
        constraints: {
            MaxLength: 50,
            MinLength: 6
        },
        serializedName: "namespaceName",
        required: true,
        type: {
            name: "String"
        }
    }
};
export const parameters1 = {
    parameterPath: "parameters",
    mapper: SBNamespaceUpdateParametersMapper
};
export const parameters2 = {
    parameterPath: "parameters",
    mapper: NetworkRuleSetMapper
};
export const parameters3 = {
    parameterPath: "parameters",
    mapper: SBAuthorizationRuleMapper
};
export const authorizationRuleName = {
    parameterPath: "authorizationRuleName",
    mapper: {
        constraints: {
            MaxLength: 50,
            MinLength: 1
        },
        serializedName: "authorizationRuleName",
        required: true,
        type: {
            name: "String"
        }
    }
};
export const parameters4 = {
    parameterPath: "parameters",
    mapper: RegenerateAccessKeyParametersMapper
};
export const parameters5 = {
    parameterPath: "parameters",
    mapper: CheckNameAvailabilityMapper
};
export const nextLink = {
    parameterPath: "nextLink",
    mapper: {
        serializedName: "nextLink",
        required: true,
        type: {
            name: "String"
        }
    },
    skipEncoding: true
};
export const parameters6 = {
    parameterPath: "parameters",
    mapper: PrivateEndpointConnectionMapper
};
export const privateEndpointConnectionName = {
    parameterPath: "privateEndpointConnectionName",
    mapper: {
        serializedName: "privateEndpointConnectionName",
        required: true,
        type: {
            name: "String"
        }
    }
};
export const parameters7 = {
    parameterPath: "parameters",
    mapper: ArmDisasterRecoveryMapper
};
export const alias = {
    parameterPath: "alias",
    mapper: {
        constraints: {
            MaxLength: 50,
            MinLength: 1
        },
        serializedName: "alias",
        required: true,
        type: {
            name: "String"
        }
    }
};
export const parameters8 = {
    parameterPath: ["options", "parameters"],
    mapper: FailoverPropertiesMapper
};
export const parameters9 = {
    parameterPath: "parameters",
    mapper: MigrationConfigPropertiesMapper
};
export const configName = {
    parameterPath: "configName",
    mapper: {
        serializedName: "configName",
        required: true,
        type: {
            name: "String"
        }
    }
};
export const queueName = {
    parameterPath: "queueName",
    mapper: {
        constraints: {
            MinLength: 1
        },
        serializedName: "queueName",
        required: true,
        type: {
            name: "String"
        }
    }
};
export const skip = {
    parameterPath: ["options", "skip"],
    mapper: {
        constraints: {
            InclusiveMaximum: 1000,
            InclusiveMinimum: 0
        },
        serializedName: "$skip",
        type: {
            name: "Number"
        }
    }
};
export const top = {
    parameterPath: ["options", "top"],
    mapper: {
        constraints: {
            InclusiveMaximum: 1000,
            InclusiveMinimum: 1
        },
        serializedName: "$top",
        type: {
            name: "Number"
        }
    }
};
export const parameters10 = {
    parameterPath: "parameters",
    mapper: SBQueueMapper
};
export const topicName = {
    parameterPath: "topicName",
    mapper: {
        constraints: {
            MinLength: 1
        },
        serializedName: "topicName",
        required: true,
        type: {
            name: "String"
        }
    }
};
export const parameters11 = {
    parameterPath: "parameters",
    mapper: SBTopicMapper
};
export const subscriptionName = {
    parameterPath: "subscriptionName",
    mapper: {
        constraints: {
            MaxLength: 50,
            MinLength: 1
        },
        serializedName: "subscriptionName",
        required: true,
        type: {
            name: "String"
        }
    }
};
export const parameters12 = {
    parameterPath: "parameters",
    mapper: RuleMapper
};
export const ruleName = {
    parameterPath: "ruleName",
    mapper: {
        constraints: {
            MaxLength: 50,
            MinLength: 1
        },
        serializedName: "ruleName",
        required: true,
        type: {
            name: "String"
        }
    }
};
export const parameters13 = {
    parameterPath: "parameters",
    mapper: SBSubscriptionMapper
};
