// @azure/arm-batch/dist-esm/src/operations/location.js
/*
 * Copyright (c) Microsoft Corporation.
 * Licensed under the MIT License.
 *
 * Code generated by Microsoft (R) AutoRest Code Generator.
 * Changes may cause incorrect behavior and will be lost if the code is regenerated.
 */
import { __asyncDelegator, __asyncGenerator, __asyncValues, __await } from "tslib";
import { setContinuationToken } from "../pagingHelper";
import * as coreClient from "@azure/core-client";
import * as Mappers from "../models/mappers";
import * as Parameters from "../models/parameters";
/// <reference lib="esnext.asynciterable" />
/** Class containing Location operations. */
export class LocationImpl {
    /**
     * Initialize a new instance of the class Location class.
     * @param client Reference to the service client
     */
    constructor(client) {
        this.client = client;
    }
    /**
     * Gets the list of Batch supported Virtual Machine VM sizes available at the given location.
     * @param locationName The region for which to retrieve Batch service supported SKUs.
     * @param options The options parameters.
     */
    listSupportedVirtualMachineSkus(locationName, options) {
        const iter = this.listSupportedVirtualMachineSkusPagingAll(locationName, options);
        return {
            next() {
                return iter.next();
            },
            [Symbol.asyncIterator]() {
                return this;
            },
            byPage: (settings) => {
                if (settings === null || settings === void 0 ? void 0 : settings.maxPageSize) {
                    throw new Error("maxPageSize is not supported by this operation.");
                }
                return this.listSupportedVirtualMachineSkusPagingPage(locationName, options, settings);
            },
        };
    }
    listSupportedVirtualMachineSkusPagingPage(locationName, options, settings) {
        return __asyncGenerator(this, arguments, function* listSupportedVirtualMachineSkusPagingPage_1() {
            let result;
            let continuationToken = settings === null || settings === void 0 ? void 0 : settings.continuationToken;
            if (!continuationToken) {
                result = yield __await(this._listSupportedVirtualMachineSkus(locationName, options));
                let page = result.value || [];
                continuationToken = result.nextLink;
                setContinuationToken(page, continuationToken);
                yield yield __await(page);
            }
            while (continuationToken) {
                result = yield __await(this._listSupportedVirtualMachineSkusNext(locationName, continuationToken, options));
                continuationToken = result.nextLink;
                let page = result.value || [];
                setContinuationToken(page, continuationToken);
                yield yield __await(page);
            }
        });
    }
    listSupportedVirtualMachineSkusPagingAll(locationName, options) {
        return __asyncGenerator(this, arguments, function* listSupportedVirtualMachineSkusPagingAll_1() {
            var _a, e_1, _b, _c;
            try {
                for (var _d = true, _e = __asyncValues(this.listSupportedVirtualMachineSkusPagingPage(locationName, options)), _f; _f = yield __await(_e.next()), _a = _f.done, !_a; _d = true) {
                    _c = _f.value;
                    _d = false;
                    const page = _c;
                    yield __await(yield* __asyncDelegator(__asyncValues(page)));
                }
            }
            catch (e_1_1) { e_1 = { error: e_1_1 }; }
            finally {
                try {
                    if (!_d && !_a && (_b = _e.return)) yield __await(_b.call(_e));
                }
                finally { if (e_1) throw e_1.error; }
            }
        });
    }
    /**
     * Gets the list of Batch supported Cloud Service VM sizes available at the given location.
     * @param locationName The region for which to retrieve Batch service supported SKUs.
     * @param options The options parameters.
     */
    listSupportedCloudServiceSkus(locationName, options) {
        const iter = this.listSupportedCloudServiceSkusPagingAll(locationName, options);
        return {
            next() {
                return iter.next();
            },
            [Symbol.asyncIterator]() {
                return this;
            },
            byPage: (settings) => {
                if (settings === null || settings === void 0 ? void 0 : settings.maxPageSize) {
                    throw new Error("maxPageSize is not supported by this operation.");
                }
                return this.listSupportedCloudServiceSkusPagingPage(locationName, options, settings);
            },
        };
    }
    listSupportedCloudServiceSkusPagingPage(locationName, options, settings) {
        return __asyncGenerator(this, arguments, function* listSupportedCloudServiceSkusPagingPage_1() {
            let result;
            let continuationToken = settings === null || settings === void 0 ? void 0 : settings.continuationToken;
            if (!continuationToken) {
                result = yield __await(this._listSupportedCloudServiceSkus(locationName, options));
                let page = result.value || [];
                continuationToken = result.nextLink;
                setContinuationToken(page, continuationToken);
                yield yield __await(page);
            }
            while (continuationToken) {
                result = yield __await(this._listSupportedCloudServiceSkusNext(locationName, continuationToken, options));
                continuationToken = result.nextLink;
                let page = result.value || [];
                setContinuationToken(page, continuationToken);
                yield yield __await(page);
            }
        });
    }
    listSupportedCloudServiceSkusPagingAll(locationName, options) {
        return __asyncGenerator(this, arguments, function* listSupportedCloudServiceSkusPagingAll_1() {
            var _a, e_2, _b, _c;
            try {
                for (var _d = true, _e = __asyncValues(this.listSupportedCloudServiceSkusPagingPage(locationName, options)), _f; _f = yield __await(_e.next()), _a = _f.done, !_a; _d = true) {
                    _c = _f.value;
                    _d = false;
                    const page = _c;
                    yield __await(yield* __asyncDelegator(__asyncValues(page)));
                }
            }
            catch (e_2_1) { e_2 = { error: e_2_1 }; }
            finally {
                try {
                    if (!_d && !_a && (_b = _e.return)) yield __await(_b.call(_e));
                }
                finally { if (e_2) throw e_2.error; }
            }
        });
    }
    /**
     * Gets the Batch service quotas for the specified subscription at the given location.
     * @param locationName The region for which to retrieve Batch service quotas.
     * @param options The options parameters.
     */
    getQuotas(locationName, options) {
        return this.client.sendOperationRequest({ locationName, options }, getQuotasOperationSpec);
    }
    /**
     * Gets the list of Batch supported Virtual Machine VM sizes available at the given location.
     * @param locationName The region for which to retrieve Batch service supported SKUs.
     * @param options The options parameters.
     */
    _listSupportedVirtualMachineSkus(locationName, options) {
        return this.client.sendOperationRequest({ locationName, options }, listSupportedVirtualMachineSkusOperationSpec);
    }
    /**
     * Gets the list of Batch supported Cloud Service VM sizes available at the given location.
     * @param locationName The region for which to retrieve Batch service supported SKUs.
     * @param options The options parameters.
     */
    _listSupportedCloudServiceSkus(locationName, options) {
        return this.client.sendOperationRequest({ locationName, options }, listSupportedCloudServiceSkusOperationSpec);
    }
    /**
     * Checks whether the Batch account name is available in the specified region.
     * @param locationName The desired region for the name check.
     * @param parameters Properties needed to check the availability of a name.
     * @param options The options parameters.
     */
    checkNameAvailability(locationName, parameters, options) {
        return this.client.sendOperationRequest({ locationName, parameters, options }, checkNameAvailabilityOperationSpec);
    }
    /**
     * ListSupportedVirtualMachineSkusNext
     * @param locationName The region for which to retrieve Batch service supported SKUs.
     * @param nextLink The nextLink from the previous successful call to the
     *                 ListSupportedVirtualMachineSkus method.
     * @param options The options parameters.
     */
    _listSupportedVirtualMachineSkusNext(locationName, nextLink, options) {
        return this.client.sendOperationRequest({ locationName, nextLink, options }, listSupportedVirtualMachineSkusNextOperationSpec);
    }
    /**
     * ListSupportedCloudServiceSkusNext
     * @param locationName The region for which to retrieve Batch service supported SKUs.
     * @param nextLink The nextLink from the previous successful call to the ListSupportedCloudServiceSkus
     *                 method.
     * @param options The options parameters.
     */
    _listSupportedCloudServiceSkusNext(locationName, nextLink, options) {
        return this.client.sendOperationRequest({ locationName, nextLink, options }, listSupportedCloudServiceSkusNextOperationSpec);
    }
}
// Operation Specifications
const serializer = coreClient.createSerializer(Mappers, /* isXml */ false);
const getQuotasOperationSpec = {
    path: "/subscriptions/{subscriptionId}/providers/Microsoft.Batch/locations/{locationName}/quotas",
    httpMethod: "GET",
    responses: {
        200: {
            bodyMapper: Mappers.BatchLocationQuota,
        },
        default: {
            bodyMapper: Mappers.CloudError,
        },
    },
    queryParameters: [Parameters.apiVersion],
    urlParameters: [
        Parameters.$host,
        Parameters.subscriptionId,
        Parameters.locationName,
    ],
    headerParameters: [Parameters.accept],
    serializer,
};
const listSupportedVirtualMachineSkusOperationSpec = {
    path: "/subscriptions/{subscriptionId}/providers/Microsoft.Batch/locations/{locationName}/virtualMachineSkus",
    httpMethod: "GET",
    responses: {
        200: {
            bodyMapper: Mappers.SupportedSkusResult,
        },
        default: {
            bodyMapper: Mappers.CloudError,
        },
    },
    queryParameters: [
        Parameters.apiVersion,
        Parameters.maxresults,
        Parameters.filter,
    ],
    urlParameters: [
        Parameters.$host,
        Parameters.subscriptionId,
        Parameters.locationName,
    ],
    headerParameters: [Parameters.accept],
    serializer,
};
const listSupportedCloudServiceSkusOperationSpec = {
    path: "/subscriptions/{subscriptionId}/providers/Microsoft.Batch/locations/{locationName}/cloudServiceSkus",
    httpMethod: "GET",
    responses: {
        200: {
            bodyMapper: Mappers.SupportedSkusResult,
        },
        default: {
            bodyMapper: Mappers.CloudError,
        },
    },
    queryParameters: [
        Parameters.apiVersion,
        Parameters.maxresults,
        Parameters.filter,
    ],
    urlParameters: [
        Parameters.$host,
        Parameters.subscriptionId,
        Parameters.locationName,
    ],
    headerParameters: [Parameters.accept],
    serializer,
};
const checkNameAvailabilityOperationSpec = {
    path: "/subscriptions/{subscriptionId}/providers/Microsoft.Batch/locations/{locationName}/checkNameAvailability",
    httpMethod: "POST",
    responses: {
        200: {
            bodyMapper: Mappers.CheckNameAvailabilityResult,
        },
        default: {
            bodyMapper: Mappers.CloudError,
        },
    },
    requestBody: Parameters.parameters7,
    queryParameters: [Parameters.apiVersion],
    urlParameters: [
        Parameters.$host,
        Parameters.subscriptionId,
        Parameters.locationName,
    ],
    headerParameters: [Parameters.contentType, Parameters.accept],
    mediaType: "json",
    serializer,
};
const listSupportedVirtualMachineSkusNextOperationSpec = {
    path: "{nextLink}",
    httpMethod: "GET",
    responses: {
        200: {
            bodyMapper: Mappers.SupportedSkusResult,
        },
        default: {
            bodyMapper: Mappers.CloudError,
        },
    },
    urlParameters: [
        Parameters.$host,
        Parameters.subscriptionId,
        Parameters.nextLink,
        Parameters.locationName,
    ],
    headerParameters: [Parameters.accept],
    serializer,
};
const listSupportedCloudServiceSkusNextOperationSpec = {
    path: "{nextLink}",
    httpMethod: "GET",
    responses: {
        200: {
            bodyMapper: Mappers.SupportedSkusResult,
        },
        default: {
            bodyMapper: Mappers.CloudError,
        },
    },
    urlParameters: [
        Parameters.$host,
        Parameters.subscriptionId,
        Parameters.nextLink,
        Parameters.locationName,
    ],
    headerParameters: [Parameters.accept],
    serializer,
};
