// @azure/arm-billing/dist-esm/src/billingManagementClient.js
/*
 * Copyright (c) Microsoft Corporation.
 * Licensed under the MIT License.
 *
 * Code generated by Microsoft (R) AutoRest Code Generator.
 * Changes may cause incorrect behavior and will be lost if the code is regenerated.
 */
import * as coreClient from "@azure/core-client";
import * as coreRestPipeline from "@azure/core-rest-pipeline";
import { BillingAccountsImpl, AddressImpl, AvailableBalancesImpl, InstructionsImpl, BillingProfilesImpl, CustomersImpl, InvoiceSectionsImpl, BillingPermissionsImpl, BillingSubscriptionsImpl, ProductsImpl, InvoicesImpl, TransactionsImpl, PoliciesImpl, BillingPropertyOperationsImpl, OperationsImpl, BillingRoleDefinitionsImpl, BillingRoleAssignmentsImpl, AgreementsImpl, ReservationsImpl, EnrollmentAccountsImpl, BillingPeriodsImpl } from "./operations";
export class BillingManagementClient extends coreClient.ServiceClient {
    /**
     * Initializes a new instance of the BillingManagementClient class.
     * @param credentials Subscription credentials which uniquely identify client subscription.
     * @param subscriptionId The ID that uniquely identifies an Azure subscription.
     * @param options The parameter options
     */
    constructor(credentials, subscriptionId, options) {
        var _a, _b, _c;
        if (credentials === undefined) {
            throw new Error("'credentials' cannot be null");
        }
        if (subscriptionId === undefined) {
            throw new Error("'subscriptionId' cannot be null");
        }
        // Initializing default values for options
        if (!options) {
            options = {};
        }
        const defaults = {
            requestContentType: "application/json; charset=utf-8",
            credential: credentials
        };
        const packageDetails = `azsdk-js-arm-billing/4.1.0`;
        const userAgentPrefix = options.userAgentOptions && options.userAgentOptions.userAgentPrefix
            ? `${options.userAgentOptions.userAgentPrefix} ${packageDetails}`
            : `${packageDetails}`;
        const optionsWithDefaults = Object.assign(Object.assign(Object.assign({}, defaults), options), { userAgentOptions: {
                userAgentPrefix
            }, endpoint: (_b = (_a = options.endpoint) !== null && _a !== void 0 ? _a : options.baseUri) !== null && _b !== void 0 ? _b : "https://management.azure.com" });
        super(optionsWithDefaults);
        let bearerTokenAuthenticationPolicyFound = false;
        if ((options === null || options === void 0 ? void 0 : options.pipeline) && options.pipeline.getOrderedPolicies().length > 0) {
            const pipelinePolicies = options.pipeline.getOrderedPolicies();
            bearerTokenAuthenticationPolicyFound = pipelinePolicies.some((pipelinePolicy) => pipelinePolicy.name ===
                coreRestPipeline.bearerTokenAuthenticationPolicyName);
        }
        if (!options ||
            !options.pipeline ||
            options.pipeline.getOrderedPolicies().length == 0 ||
            !bearerTokenAuthenticationPolicyFound) {
            this.pipeline.removePolicy({
                name: coreRestPipeline.bearerTokenAuthenticationPolicyName
            });
            this.pipeline.addPolicy(coreRestPipeline.bearerTokenAuthenticationPolicy({
                credential: credentials,
                scopes: (_c = optionsWithDefaults.credentialScopes) !== null && _c !== void 0 ? _c : `${optionsWithDefaults.endpoint}/.default`,
                challengeCallbacks: {
                    authorizeRequestOnChallenge: coreClient.authorizeRequestOnClaimChallenge
                }
            }));
        }
        // Parameter assignments
        this.subscriptionId = subscriptionId;
        // Assigning values to Constant parameters
        this.$host = options.$host || "https://management.azure.com";
        this.billingAccounts = new BillingAccountsImpl(this);
        this.address = new AddressImpl(this);
        this.availableBalances = new AvailableBalancesImpl(this);
        this.instructions = new InstructionsImpl(this);
        this.billingProfiles = new BillingProfilesImpl(this);
        this.customers = new CustomersImpl(this);
        this.invoiceSections = new InvoiceSectionsImpl(this);
        this.billingPermissions = new BillingPermissionsImpl(this);
        this.billingSubscriptions = new BillingSubscriptionsImpl(this);
        this.products = new ProductsImpl(this);
        this.invoices = new InvoicesImpl(this);
        this.transactions = new TransactionsImpl(this);
        this.policies = new PoliciesImpl(this);
        this.billingPropertyOperations = new BillingPropertyOperationsImpl(this);
        this.operations = new OperationsImpl(this);
        this.billingRoleDefinitions = new BillingRoleDefinitionsImpl(this);
        this.billingRoleAssignments = new BillingRoleAssignmentsImpl(this);
        this.agreements = new AgreementsImpl(this);
        this.reservations = new ReservationsImpl(this);
        this.enrollmentAccounts = new EnrollmentAccountsImpl(this);
        this.billingPeriods = new BillingPeriodsImpl(this);
    }
}
