// @aws-sdk/client-codestar/dist-cjs/index.js
"use strict";
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __name = (target, value) => __defProp(target, "name", { value, configurable: true });
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// src/index.ts
var src_exports = {};
__export(src_exports, {
  AssociateTeamMemberCommand: () => AssociateTeamMemberCommand,
  CodeDestinationFilterSensitiveLog: () => CodeDestinationFilterSensitiveLog,
  CodeFilterSensitiveLog: () => CodeFilterSensitiveLog,
  CodeStar: () => CodeStar,
  CodeStarClient: () => CodeStarClient,
  CodeStarServiceException: () => CodeStarServiceException,
  ConcurrentModificationException: () => ConcurrentModificationException,
  CreateProjectCommand: () => CreateProjectCommand,
  CreateProjectRequestFilterSensitiveLog: () => CreateProjectRequestFilterSensitiveLog,
  CreateUserProfileCommand: () => CreateUserProfileCommand,
  CreateUserProfileRequestFilterSensitiveLog: () => CreateUserProfileRequestFilterSensitiveLog,
  CreateUserProfileResultFilterSensitiveLog: () => CreateUserProfileResultFilterSensitiveLog,
  DeleteProjectCommand: () => DeleteProjectCommand,
  DeleteUserProfileCommand: () => DeleteUserProfileCommand,
  DescribeProjectCommand: () => DescribeProjectCommand,
  DescribeProjectResultFilterSensitiveLog: () => DescribeProjectResultFilterSensitiveLog,
  DescribeUserProfileCommand: () => DescribeUserProfileCommand,
  DescribeUserProfileResultFilterSensitiveLog: () => DescribeUserProfileResultFilterSensitiveLog,
  DisassociateTeamMemberCommand: () => DisassociateTeamMemberCommand,
  GitHubCodeDestinationFilterSensitiveLog: () => GitHubCodeDestinationFilterSensitiveLog,
  InvalidNextTokenException: () => InvalidNextTokenException,
  InvalidServiceRoleException: () => InvalidServiceRoleException,
  LimitExceededException: () => LimitExceededException,
  ListProjectsCommand: () => ListProjectsCommand,
  ListResourcesCommand: () => ListResourcesCommand,
  ListTagsForProjectCommand: () => ListTagsForProjectCommand,
  ListTeamMembersCommand: () => ListTeamMembersCommand,
  ListUserProfilesCommand: () => ListUserProfilesCommand,
  ListUserProfilesResultFilterSensitiveLog: () => ListUserProfilesResultFilterSensitiveLog,
  ProjectAlreadyExistsException: () => ProjectAlreadyExistsException,
  ProjectConfigurationException: () => ProjectConfigurationException,
  ProjectCreationFailedException: () => ProjectCreationFailedException,
  ProjectNotFoundException: () => ProjectNotFoundException,
  TagProjectCommand: () => TagProjectCommand,
  TeamMemberAlreadyAssociatedException: () => TeamMemberAlreadyAssociatedException,
  TeamMemberNotFoundException: () => TeamMemberNotFoundException,
  ToolchainFilterSensitiveLog: () => ToolchainFilterSensitiveLog,
  UntagProjectCommand: () => UntagProjectCommand,
  UpdateProjectCommand: () => UpdateProjectCommand,
  UpdateProjectRequestFilterSensitiveLog: () => UpdateProjectRequestFilterSensitiveLog,
  UpdateTeamMemberCommand: () => UpdateTeamMemberCommand,
  UpdateUserProfileCommand: () => UpdateUserProfileCommand,
  UpdateUserProfileRequestFilterSensitiveLog: () => UpdateUserProfileRequestFilterSensitiveLog,
  UpdateUserProfileResultFilterSensitiveLog: () => UpdateUserProfileResultFilterSensitiveLog,
  UserProfileAlreadyExistsException: () => UserProfileAlreadyExistsException,
  UserProfileNotFoundException: () => UserProfileNotFoundException,
  UserProfileSummaryFilterSensitiveLog: () => UserProfileSummaryFilterSensitiveLog,
  ValidationException: () => ValidationException,
  __Client: () => import_smithy_client.Client
});
module.exports = __toCommonJS(src_exports);

// src/CodeStarClient.ts
var import_middleware_host_header = require("@aws-sdk/middleware-host-header");
var import_middleware_logger = require("@aws-sdk/middleware-logger");
var import_middleware_recursion_detection = require("@aws-sdk/middleware-recursion-detection");
var import_middleware_user_agent = require("@aws-sdk/middleware-user-agent");
var import_config_resolver = require("@smithy/config-resolver");
var import_core = require("@smithy/core");
var import_middleware_content_length = require("@smithy/middleware-content-length");
var import_middleware_endpoint = require("@smithy/middleware-endpoint");
var import_middleware_retry = require("@smithy/middleware-retry");

var import_httpAuthSchemeProvider = require("./auth/httpAuthSchemeProvider");

// src/endpoint/EndpointParameters.ts
var resolveClientEndpointParameters = /* @__PURE__ */ __name((options) => {
  return {
    ...options,
    useDualstackEndpoint: options.useDualstackEndpoint ?? false,
    useFipsEndpoint: options.useFipsEndpoint ?? false,
    defaultSigningName: "codestar"
  };
}, "resolveClientEndpointParameters");
var commonParams = {
  UseFIPS: { type: "builtInParams", name: "useFipsEndpoint" },
  Endpoint: { type: "builtInParams", name: "endpoint" },
  Region: { type: "builtInParams", name: "region" },
  UseDualStack: { type: "builtInParams", name: "useDualstackEndpoint" }
};

// src/CodeStarClient.ts
var import_runtimeConfig = require("././runtimeConfig");

// src/runtimeExtensions.ts
var import_region_config_resolver = require("@aws-sdk/region-config-resolver");
var import_protocol_http = require("@smithy/protocol-http");
var import_smithy_client = require("@smithy/smithy-client");

// src/auth/httpAuthExtensionConfiguration.ts
var getHttpAuthExtensionConfiguration = /* @__PURE__ */ __name((runtimeConfig) => {
  const _httpAuthSchemes = runtimeConfig.httpAuthSchemes;
  let _httpAuthSchemeProvider = runtimeConfig.httpAuthSchemeProvider;
  let _credentials = runtimeConfig.credentials;
  return {
    setHttpAuthScheme(httpAuthScheme) {
      const index = _httpAuthSchemes.findIndex((scheme) => scheme.schemeId === httpAuthScheme.schemeId);
      if (index === -1) {
        _httpAuthSchemes.push(httpAuthScheme);
      } else {
        _httpAuthSchemes.splice(index, 1, httpAuthScheme);
      }
    },
    httpAuthSchemes() {
      return _httpAuthSchemes;
    },
    setHttpAuthSchemeProvider(httpAuthSchemeProvider) {
      _httpAuthSchemeProvider = httpAuthSchemeProvider;
    },
    httpAuthSchemeProvider() {
      return _httpAuthSchemeProvider;
    },
    setCredentials(credentials) {
      _credentials = credentials;
    },
    credentials() {
      return _credentials;
    }
  };
}, "getHttpAuthExtensionConfiguration");
var resolveHttpAuthRuntimeConfig = /* @__PURE__ */ __name((config) => {
  return {
    httpAuthSchemes: config.httpAuthSchemes(),
    httpAuthSchemeProvider: config.httpAuthSchemeProvider(),
    credentials: config.credentials()
  };
}, "resolveHttpAuthRuntimeConfig");

// src/runtimeExtensions.ts
var asPartial = /* @__PURE__ */ __name((t) => t, "asPartial");
var resolveRuntimeExtensions = /* @__PURE__ */ __name((runtimeConfig, extensions) => {
  const extensionConfiguration = {
    ...asPartial((0, import_region_config_resolver.getAwsRegionExtensionConfiguration)(runtimeConfig)),
    ...asPartial((0, import_smithy_client.getDefaultExtensionConfiguration)(runtimeConfig)),
    ...asPartial((0, import_protocol_http.getHttpHandlerExtensionConfiguration)(runtimeConfig)),
    ...asPartial(getHttpAuthExtensionConfiguration(runtimeConfig))
  };
  extensions.forEach((extension) => extension.configure(extensionConfiguration));
  return {
    ...runtimeConfig,
    ...(0, import_region_config_resolver.resolveAwsRegionExtensionConfiguration)(extensionConfiguration),
    ...(0, import_smithy_client.resolveDefaultRuntimeConfig)(extensionConfiguration),
    ...(0, import_protocol_http.resolveHttpHandlerRuntimeConfig)(extensionConfiguration),
    ...resolveHttpAuthRuntimeConfig(extensionConfiguration)
  };
}, "resolveRuntimeExtensions");

// src/CodeStarClient.ts
var _CodeStarClient = class _CodeStarClient extends import_smithy_client.Client {
  constructor(...[configuration]) {
    const _config_0 = (0, import_runtimeConfig.getRuntimeConfig)(configuration || {});
    const _config_1 = resolveClientEndpointParameters(_config_0);
    const _config_2 = (0, import_config_resolver.resolveRegionConfig)(_config_1);
    const _config_3 = (0, import_middleware_endpoint.resolveEndpointConfig)(_config_2);
    const _config_4 = (0, import_middleware_retry.resolveRetryConfig)(_config_3);
    const _config_5 = (0, import_middleware_host_header.resolveHostHeaderConfig)(_config_4);
    const _config_6 = (0, import_middleware_user_agent.resolveUserAgentConfig)(_config_5);
    const _config_7 = (0, import_httpAuthSchemeProvider.resolveHttpAuthSchemeConfig)(_config_6);
    const _config_8 = resolveRuntimeExtensions(_config_7, (configuration == null ? void 0 : configuration.extensions) || []);
    super(_config_8);
    this.config = _config_8;
    this.middlewareStack.use((0, import_middleware_retry.getRetryPlugin)(this.config));
    this.middlewareStack.use((0, import_middleware_content_length.getContentLengthPlugin)(this.config));
    this.middlewareStack.use((0, import_middleware_host_header.getHostHeaderPlugin)(this.config));
    this.middlewareStack.use((0, import_middleware_logger.getLoggerPlugin)(this.config));
    this.middlewareStack.use((0, import_middleware_recursion_detection.getRecursionDetectionPlugin)(this.config));
    this.middlewareStack.use((0, import_middleware_user_agent.getUserAgentPlugin)(this.config));
    this.middlewareStack.use(
      (0, import_core.getHttpAuthSchemeEndpointRuleSetPlugin)(this.config, {
        httpAuthSchemeParametersProvider: this.getDefaultHttpAuthSchemeParametersProvider(),
        identityProviderConfigProvider: this.getIdentityProviderConfigProvider()
      })
    );
    this.middlewareStack.use((0, import_core.getHttpSigningPlugin)(this.config));
  }
  /**
   * Destroy underlying resources, like sockets. It's usually not necessary to do this.
   * However in Node.js, it's best to explicitly shut down the client's agent when it is no longer needed.
   * Otherwise, sockets might stay open for quite a long time before the server terminates them.
   */
  destroy() {
    super.destroy();
  }
  getDefaultHttpAuthSchemeParametersProvider() {
    return import_httpAuthSchemeProvider.defaultCodeStarHttpAuthSchemeParametersProvider;
  }
  getIdentityProviderConfigProvider() {
    return async (config) => new import_core.DefaultIdentityProviderConfig({
      "aws.auth#sigv4": config.credentials
    });
  }
};
__name(_CodeStarClient, "CodeStarClient");
var CodeStarClient = _CodeStarClient;

// src/CodeStar.ts


// src/commands/AssociateTeamMemberCommand.ts

var import_middleware_serde = require("@smithy/middleware-serde");

var import_types = require("@smithy/types");

// src/protocols/Aws_json1_1.ts
var import_core2 = require("@aws-sdk/core");



// src/models/CodeStarServiceException.ts

var _CodeStarServiceException = class _CodeStarServiceException extends import_smithy_client.ServiceException {
  /**
   * @internal
   */
  constructor(options) {
    super(options);
    Object.setPrototypeOf(this, _CodeStarServiceException.prototype);
  }
};
__name(_CodeStarServiceException, "CodeStarServiceException");
var CodeStarServiceException = _CodeStarServiceException;

// src/models/models_0.ts

var _ConcurrentModificationException = class _ConcurrentModificationException extends CodeStarServiceException {
  /**
   * @internal
   */
  constructor(opts) {
    super({
      name: "ConcurrentModificationException",
      $fault: "server",
      ...opts
    });
    this.name = "ConcurrentModificationException";
    this.$fault = "server";
    Object.setPrototypeOf(this, _ConcurrentModificationException.prototype);
  }
};
__name(_ConcurrentModificationException, "ConcurrentModificationException");
var ConcurrentModificationException = _ConcurrentModificationException;
var _InvalidServiceRoleException = class _InvalidServiceRoleException extends CodeStarServiceException {
  /**
   * @internal
   */
  constructor(opts) {
    super({
      name: "InvalidServiceRoleException",
      $fault: "client",
      ...opts
    });
    this.name = "InvalidServiceRoleException";
    this.$fault = "client";
    Object.setPrototypeOf(this, _InvalidServiceRoleException.prototype);
  }
};
__name(_InvalidServiceRoleException, "InvalidServiceRoleException");
var InvalidServiceRoleException = _InvalidServiceRoleException;
var _LimitExceededException = class _LimitExceededException extends CodeStarServiceException {
  /**
   * @internal
   */
  constructor(opts) {
    super({
      name: "LimitExceededException",
      $fault: "client",
      ...opts
    });
    this.name = "LimitExceededException";
    this.$fault = "client";
    Object.setPrototypeOf(this, _LimitExceededException.prototype);
  }
};
__name(_LimitExceededException, "LimitExceededException");
var LimitExceededException = _LimitExceededException;
var _ProjectConfigurationException = class _ProjectConfigurationException extends CodeStarServiceException {
  /**
   * @internal
   */
  constructor(opts) {
    super({
      name: "ProjectConfigurationException",
      $fault: "client",
      ...opts
    });
    this.name = "ProjectConfigurationException";
    this.$fault = "client";
    Object.setPrototypeOf(this, _ProjectConfigurationException.prototype);
  }
};
__name(_ProjectConfigurationException, "ProjectConfigurationException");
var ProjectConfigurationException = _ProjectConfigurationException;
var _ProjectNotFoundException = class _ProjectNotFoundException extends CodeStarServiceException {
  /**
   * @internal
   */
  constructor(opts) {
    super({
      name: "ProjectNotFoundException",
      $fault: "client",
      ...opts
    });
    this.name = "ProjectNotFoundException";
    this.$fault = "client";
    Object.setPrototypeOf(this, _ProjectNotFoundException.prototype);
  }
};
__name(_ProjectNotFoundException, "ProjectNotFoundException");
var ProjectNotFoundException = _ProjectNotFoundException;
var _TeamMemberAlreadyAssociatedException = class _TeamMemberAlreadyAssociatedException extends CodeStarServiceException {
  /**
   * @internal
   */
  constructor(opts) {
    super({
      name: "TeamMemberAlreadyAssociatedException",
      $fault: "client",
      ...opts
    });
    this.name = "TeamMemberAlreadyAssociatedException";
    this.$fault = "client";
    Object.setPrototypeOf(this, _TeamMemberAlreadyAssociatedException.prototype);
  }
};
__name(_TeamMemberAlreadyAssociatedException, "TeamMemberAlreadyAssociatedException");
var TeamMemberAlreadyAssociatedException = _TeamMemberAlreadyAssociatedException;
var _ValidationException = class _ValidationException extends CodeStarServiceException {
  /**
   * @internal
   */
  constructor(opts) {
    super({
      name: "ValidationException",
      $fault: "client",
      ...opts
    });
    this.name = "ValidationException";
    this.$fault = "client";
    Object.setPrototypeOf(this, _ValidationException.prototype);
  }
};
__name(_ValidationException, "ValidationException");
var ValidationException = _ValidationException;
var _ProjectAlreadyExistsException = class _ProjectAlreadyExistsException extends CodeStarServiceException {
  /**
   * @internal
   */
  constructor(opts) {
    super({
      name: "ProjectAlreadyExistsException",
      $fault: "client",
      ...opts
    });
    this.name = "ProjectAlreadyExistsException";
    this.$fault = "client";
    Object.setPrototypeOf(this, _ProjectAlreadyExistsException.prototype);
  }
};
__name(_ProjectAlreadyExistsException, "ProjectAlreadyExistsException");
var ProjectAlreadyExistsException = _ProjectAlreadyExistsException;
var _ProjectCreationFailedException = class _ProjectCreationFailedException extends CodeStarServiceException {
  /**
   * @internal
   */
  constructor(opts) {
    super({
      name: "ProjectCreationFailedException",
      $fault: "client",
      ...opts
    });
    this.name = "ProjectCreationFailedException";
    this.$fault = "client";
    Object.setPrototypeOf(this, _ProjectCreationFailedException.prototype);
  }
};
__name(_ProjectCreationFailedException, "ProjectCreationFailedException");
var ProjectCreationFailedException = _ProjectCreationFailedException;
var _UserProfileAlreadyExistsException = class _UserProfileAlreadyExistsException extends CodeStarServiceException {
  /**
   * @internal
   */
  constructor(opts) {
    super({
      name: "UserProfileAlreadyExistsException",
      $fault: "client",
      ...opts
    });
    this.name = "UserProfileAlreadyExistsException";
    this.$fault = "client";
    Object.setPrototypeOf(this, _UserProfileAlreadyExistsException.prototype);
  }
};
__name(_UserProfileAlreadyExistsException, "UserProfileAlreadyExistsException");
var UserProfileAlreadyExistsException = _UserProfileAlreadyExistsException;
var _UserProfileNotFoundException = class _UserProfileNotFoundException extends CodeStarServiceException {
  /**
   * @internal
   */
  constructor(opts) {
    super({
      name: "UserProfileNotFoundException",
      $fault: "client",
      ...opts
    });
    this.name = "UserProfileNotFoundException";
    this.$fault = "client";
    Object.setPrototypeOf(this, _UserProfileNotFoundException.prototype);
  }
};
__name(_UserProfileNotFoundException, "UserProfileNotFoundException");
var UserProfileNotFoundException = _UserProfileNotFoundException;
var _InvalidNextTokenException = class _InvalidNextTokenException extends CodeStarServiceException {
  /**
   * @internal
   */
  constructor(opts) {
    super({
      name: "InvalidNextTokenException",
      $fault: "client",
      ...opts
    });
    this.name = "InvalidNextTokenException";
    this.$fault = "client";
    Object.setPrototypeOf(this, _InvalidNextTokenException.prototype);
  }
};
__name(_InvalidNextTokenException, "InvalidNextTokenException");
var InvalidNextTokenException = _InvalidNextTokenException;
var _TeamMemberNotFoundException = class _TeamMemberNotFoundException extends CodeStarServiceException {
  /**
   * @internal
   */
  constructor(opts) {
    super({
      name: "TeamMemberNotFoundException",
      $fault: "client",
      ...opts
    });
    this.name = "TeamMemberNotFoundException";
    this.$fault = "client";
    Object.setPrototypeOf(this, _TeamMemberNotFoundException.prototype);
  }
};
__name(_TeamMemberNotFoundException, "TeamMemberNotFoundException");
var TeamMemberNotFoundException = _TeamMemberNotFoundException;
var GitHubCodeDestinationFilterSensitiveLog = /* @__PURE__ */ __name((obj) => ({
  ...obj,
  ...obj.token && { token: import_smithy_client.SENSITIVE_STRING }
}), "GitHubCodeDestinationFilterSensitiveLog");
var CodeDestinationFilterSensitiveLog = /* @__PURE__ */ __name((obj) => ({
  ...obj,
  ...obj.gitHub && { gitHub: GitHubCodeDestinationFilterSensitiveLog(obj.gitHub) }
}), "CodeDestinationFilterSensitiveLog");
var CodeFilterSensitiveLog = /* @__PURE__ */ __name((obj) => ({
  ...obj,
  ...obj.destination && { destination: CodeDestinationFilterSensitiveLog(obj.destination) }
}), "CodeFilterSensitiveLog");
var ToolchainFilterSensitiveLog = /* @__PURE__ */ __name((obj) => ({
  ...obj,
  ...obj.stackParameters && { stackParameters: import_smithy_client.SENSITIVE_STRING }
}), "ToolchainFilterSensitiveLog");
var CreateProjectRequestFilterSensitiveLog = /* @__PURE__ */ __name((obj) => ({
  ...obj,
  ...obj.name && { name: import_smithy_client.SENSITIVE_STRING },
  ...obj.description && { description: import_smithy_client.SENSITIVE_STRING },
  ...obj.sourceCode && { sourceCode: obj.sourceCode.map((item) => CodeFilterSensitiveLog(item)) },
  ...obj.toolchain && { toolchain: ToolchainFilterSensitiveLog(obj.toolchain) }
}), "CreateProjectRequestFilterSensitiveLog");
var CreateUserProfileRequestFilterSensitiveLog = /* @__PURE__ */ __name((obj) => ({
  ...obj,
  ...obj.displayName && { displayName: import_smithy_client.SENSITIVE_STRING },
  ...obj.emailAddress && { emailAddress: import_smithy_client.SENSITIVE_STRING }
}), "CreateUserProfileRequestFilterSensitiveLog");
var CreateUserProfileResultFilterSensitiveLog = /* @__PURE__ */ __name((obj) => ({
  ...obj,
  ...obj.displayName && { displayName: import_smithy_client.SENSITIVE_STRING },
  ...obj.emailAddress && { emailAddress: import_smithy_client.SENSITIVE_STRING }
}), "CreateUserProfileResultFilterSensitiveLog");
var DescribeProjectResultFilterSensitiveLog = /* @__PURE__ */ __name((obj) => ({
  ...obj,
  ...obj.name && { name: import_smithy_client.SENSITIVE_STRING },
  ...obj.description && { description: import_smithy_client.SENSITIVE_STRING }
}), "DescribeProjectResultFilterSensitiveLog");
var DescribeUserProfileResultFilterSensitiveLog = /* @__PURE__ */ __name((obj) => ({
  ...obj,
  ...obj.displayName && { displayName: import_smithy_client.SENSITIVE_STRING },
  ...obj.emailAddress && { emailAddress: import_smithy_client.SENSITIVE_STRING }
}), "DescribeUserProfileResultFilterSensitiveLog");
var UserProfileSummaryFilterSensitiveLog = /* @__PURE__ */ __name((obj) => ({
  ...obj,
  ...obj.displayName && { displayName: import_smithy_client.SENSITIVE_STRING },
  ...obj.emailAddress && { emailAddress: import_smithy_client.SENSITIVE_STRING }
}), "UserProfileSummaryFilterSensitiveLog");
var ListUserProfilesResultFilterSensitiveLog = /* @__PURE__ */ __name((obj) => ({
  ...obj,
  ...obj.userProfiles && { userProfiles: obj.userProfiles.map((item) => UserProfileSummaryFilterSensitiveLog(item)) }
}), "ListUserProfilesResultFilterSensitiveLog");
var UpdateProjectRequestFilterSensitiveLog = /* @__PURE__ */ __name((obj) => ({
  ...obj,
  ...obj.name && { name: import_smithy_client.SENSITIVE_STRING },
  ...obj.description && { description: import_smithy_client.SENSITIVE_STRING }
}), "UpdateProjectRequestFilterSensitiveLog");
var UpdateUserProfileRequestFilterSensitiveLog = /* @__PURE__ */ __name((obj) => ({
  ...obj,
  ...obj.displayName && { displayName: import_smithy_client.SENSITIVE_STRING },
  ...obj.emailAddress && { emailAddress: import_smithy_client.SENSITIVE_STRING }
}), "UpdateUserProfileRequestFilterSensitiveLog");
var UpdateUserProfileResultFilterSensitiveLog = /* @__PURE__ */ __name((obj) => ({
  ...obj,
  ...obj.displayName && { displayName: import_smithy_client.SENSITIVE_STRING },
  ...obj.emailAddress && { emailAddress: import_smithy_client.SENSITIVE_STRING }
}), "UpdateUserProfileResultFilterSensitiveLog");

// src/protocols/Aws_json1_1.ts
var se_AssociateTeamMemberCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("AssociateTeamMember");
  let body;
  body = JSON.stringify((0, import_smithy_client._json)(input));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_AssociateTeamMemberCommand");
var se_CreateProjectCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("CreateProject");
  let body;
  body = JSON.stringify((0, import_smithy_client._json)(input));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_CreateProjectCommand");
var se_CreateUserProfileCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("CreateUserProfile");
  let body;
  body = JSON.stringify((0, import_smithy_client._json)(input));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_CreateUserProfileCommand");
var se_DeleteProjectCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("DeleteProject");
  let body;
  body = JSON.stringify((0, import_smithy_client._json)(input));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_DeleteProjectCommand");
var se_DeleteUserProfileCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("DeleteUserProfile");
  let body;
  body = JSON.stringify((0, import_smithy_client._json)(input));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_DeleteUserProfileCommand");
var se_DescribeProjectCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("DescribeProject");
  let body;
  body = JSON.stringify((0, import_smithy_client._json)(input));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_DescribeProjectCommand");
var se_DescribeUserProfileCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("DescribeUserProfile");
  let body;
  body = JSON.stringify((0, import_smithy_client._json)(input));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_DescribeUserProfileCommand");
var se_DisassociateTeamMemberCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("DisassociateTeamMember");
  let body;
  body = JSON.stringify((0, import_smithy_client._json)(input));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_DisassociateTeamMemberCommand");
var se_ListProjectsCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("ListProjects");
  let body;
  body = JSON.stringify((0, import_smithy_client._json)(input));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_ListProjectsCommand");
var se_ListResourcesCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("ListResources");
  let body;
  body = JSON.stringify((0, import_smithy_client._json)(input));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_ListResourcesCommand");
var se_ListTagsForProjectCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("ListTagsForProject");
  let body;
  body = JSON.stringify((0, import_smithy_client._json)(input));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_ListTagsForProjectCommand");
var se_ListTeamMembersCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("ListTeamMembers");
  let body;
  body = JSON.stringify((0, import_smithy_client._json)(input));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_ListTeamMembersCommand");
var se_ListUserProfilesCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("ListUserProfiles");
  let body;
  body = JSON.stringify((0, import_smithy_client._json)(input));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_ListUserProfilesCommand");
var se_TagProjectCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("TagProject");
  let body;
  body = JSON.stringify((0, import_smithy_client._json)(input));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_TagProjectCommand");
var se_UntagProjectCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("UntagProject");
  let body;
  body = JSON.stringify((0, import_smithy_client._json)(input));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_UntagProjectCommand");
var se_UpdateProjectCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("UpdateProject");
  let body;
  body = JSON.stringify((0, import_smithy_client._json)(input));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_UpdateProjectCommand");
var se_UpdateTeamMemberCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("UpdateTeamMember");
  let body;
  body = JSON.stringify((0, import_smithy_client._json)(input));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_UpdateTeamMemberCommand");
var se_UpdateUserProfileCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("UpdateUserProfile");
  let body;
  body = JSON.stringify((0, import_smithy_client._json)(input));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_UpdateUserProfileCommand");
var de_AssociateTeamMemberCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const data = await (0, import_core2.parseJsonBody)(output.body, context);
  let contents = {};
  contents = (0, import_smithy_client._json)(data);
  const response = {
    $metadata: deserializeMetadata(output),
    ...contents
  };
  return response;
}, "de_AssociateTeamMemberCommand");
var de_CreateProjectCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const data = await (0, import_core2.parseJsonBody)(output.body, context);
  let contents = {};
  contents = (0, import_smithy_client._json)(data);
  const response = {
    $metadata: deserializeMetadata(output),
    ...contents
  };
  return response;
}, "de_CreateProjectCommand");
var de_CreateUserProfileCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const data = await (0, import_core2.parseJsonBody)(output.body, context);
  let contents = {};
  contents = de_CreateUserProfileResult(data, context);
  const response = {
    $metadata: deserializeMetadata(output),
    ...contents
  };
  return response;
}, "de_CreateUserProfileCommand");
var de_DeleteProjectCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const data = await (0, import_core2.parseJsonBody)(output.body, context);
  let contents = {};
  contents = (0, import_smithy_client._json)(data);
  const response = {
    $metadata: deserializeMetadata(output),
    ...contents
  };
  return response;
}, "de_DeleteProjectCommand");
var de_DeleteUserProfileCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const data = await (0, import_core2.parseJsonBody)(output.body, context);
  let contents = {};
  contents = (0, import_smithy_client._json)(data);
  const response = {
    $metadata: deserializeMetadata(output),
    ...contents
  };
  return response;
}, "de_DeleteUserProfileCommand");
var de_DescribeProjectCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const data = await (0, import_core2.parseJsonBody)(output.body, context);
  let contents = {};
  contents = de_DescribeProjectResult(data, context);
  const response = {
    $metadata: deserializeMetadata(output),
    ...contents
  };
  return response;
}, "de_DescribeProjectCommand");
var de_DescribeUserProfileCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const data = await (0, import_core2.parseJsonBody)(output.body, context);
  let contents = {};
  contents = de_DescribeUserProfileResult(data, context);
  const response = {
    $metadata: deserializeMetadata(output),
    ...contents
  };
  return response;
}, "de_DescribeUserProfileCommand");
var de_DisassociateTeamMemberCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const data = await (0, import_core2.parseJsonBody)(output.body, context);
  let contents = {};
  contents = (0, import_smithy_client._json)(data);
  const response = {
    $metadata: deserializeMetadata(output),
    ...contents
  };
  return response;
}, "de_DisassociateTeamMemberCommand");
var de_ListProjectsCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const data = await (0, import_core2.parseJsonBody)(output.body, context);
  let contents = {};
  contents = (0, import_smithy_client._json)(data);
  const response = {
    $metadata: deserializeMetadata(output),
    ...contents
  };
  return response;
}, "de_ListProjectsCommand");
var de_ListResourcesCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const data = await (0, import_core2.parseJsonBody)(output.body, context);
  let contents = {};
  contents = (0, import_smithy_client._json)(data);
  const response = {
    $metadata: deserializeMetadata(output),
    ...contents
  };
  return response;
}, "de_ListResourcesCommand");
var de_ListTagsForProjectCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const data = await (0, import_core2.parseJsonBody)(output.body, context);
  let contents = {};
  contents = (0, import_smithy_client._json)(data);
  const response = {
    $metadata: deserializeMetadata(output),
    ...contents
  };
  return response;
}, "de_ListTagsForProjectCommand");
var de_ListTeamMembersCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const data = await (0, import_core2.parseJsonBody)(output.body, context);
  let contents = {};
  contents = (0, import_smithy_client._json)(data);
  const response = {
    $metadata: deserializeMetadata(output),
    ...contents
  };
  return response;
}, "de_ListTeamMembersCommand");
var de_ListUserProfilesCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const data = await (0, import_core2.parseJsonBody)(output.body, context);
  let contents = {};
  contents = (0, import_smithy_client._json)(data);
  const response = {
    $metadata: deserializeMetadata(output),
    ...contents
  };
  return response;
}, "de_ListUserProfilesCommand");
var de_TagProjectCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const data = await (0, import_core2.parseJsonBody)(output.body, context);
  let contents = {};
  contents = (0, import_smithy_client._json)(data);
  const response = {
    $metadata: deserializeMetadata(output),
    ...contents
  };
  return response;
}, "de_TagProjectCommand");
var de_UntagProjectCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const data = await (0, import_core2.parseJsonBody)(output.body, context);
  let contents = {};
  contents = (0, import_smithy_client._json)(data);
  const response = {
    $metadata: deserializeMetadata(output),
    ...contents
  };
  return response;
}, "de_UntagProjectCommand");
var de_UpdateProjectCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const data = await (0, import_core2.parseJsonBody)(output.body, context);
  let contents = {};
  contents = (0, import_smithy_client._json)(data);
  const response = {
    $metadata: deserializeMetadata(output),
    ...contents
  };
  return response;
}, "de_UpdateProjectCommand");
var de_UpdateTeamMemberCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const data = await (0, import_core2.parseJsonBody)(output.body, context);
  let contents = {};
  contents = (0, import_smithy_client._json)(data);
  const response = {
    $metadata: deserializeMetadata(output),
    ...contents
  };
  return response;
}, "de_UpdateTeamMemberCommand");
var de_UpdateUserProfileCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const data = await (0, import_core2.parseJsonBody)(output.body, context);
  let contents = {};
  contents = de_UpdateUserProfileResult(data, context);
  const response = {
    $metadata: deserializeMetadata(output),
    ...contents
  };
  return response;
}, "de_UpdateUserProfileCommand");
var de_CommandError = /* @__PURE__ */ __name(async (output, context) => {
  const parsedOutput = {
    ...output,
    body: await (0, import_core2.parseJsonErrorBody)(output.body, context)
  };
  const errorCode = (0, import_core2.loadRestJsonErrorCode)(output, parsedOutput.body);
  switch (errorCode) {
    case "ConcurrentModificationException":
    case "com.amazonaws.codestar#ConcurrentModificationException":
      throw await de_ConcurrentModificationExceptionRes(parsedOutput, context);
    case "InvalidServiceRoleException":
    case "com.amazonaws.codestar#InvalidServiceRoleException":
      throw await de_InvalidServiceRoleExceptionRes(parsedOutput, context);
    case "LimitExceededException":
    case "com.amazonaws.codestar#LimitExceededException":
      throw await de_LimitExceededExceptionRes(parsedOutput, context);
    case "ProjectConfigurationException":
    case "com.amazonaws.codestar#ProjectConfigurationException":
      throw await de_ProjectConfigurationExceptionRes(parsedOutput, context);
    case "ProjectNotFoundException":
    case "com.amazonaws.codestar#ProjectNotFoundException":
      throw await de_ProjectNotFoundExceptionRes(parsedOutput, context);
    case "TeamMemberAlreadyAssociatedException":
    case "com.amazonaws.codestar#TeamMemberAlreadyAssociatedException":
      throw await de_TeamMemberAlreadyAssociatedExceptionRes(parsedOutput, context);
    case "ValidationException":
    case "com.amazonaws.codestar#ValidationException":
      throw await de_ValidationExceptionRes(parsedOutput, context);
    case "ProjectAlreadyExistsException":
    case "com.amazonaws.codestar#ProjectAlreadyExistsException":
      throw await de_ProjectAlreadyExistsExceptionRes(parsedOutput, context);
    case "ProjectCreationFailedException":
    case "com.amazonaws.codestar#ProjectCreationFailedException":
      throw await de_ProjectCreationFailedExceptionRes(parsedOutput, context);
    case "UserProfileAlreadyExistsException":
    case "com.amazonaws.codestar#UserProfileAlreadyExistsException":
      throw await de_UserProfileAlreadyExistsExceptionRes(parsedOutput, context);
    case "UserProfileNotFoundException":
    case "com.amazonaws.codestar#UserProfileNotFoundException":
      throw await de_UserProfileNotFoundExceptionRes(parsedOutput, context);
    case "InvalidNextTokenException":
    case "com.amazonaws.codestar#InvalidNextTokenException":
      throw await de_InvalidNextTokenExceptionRes(parsedOutput, context);
    case "TeamMemberNotFoundException":
    case "com.amazonaws.codestar#TeamMemberNotFoundException":
      throw await de_TeamMemberNotFoundExceptionRes(parsedOutput, context);
    default:
      const parsedBody = parsedOutput.body;
      return throwDefaultError({
        output,
        parsedBody,
        errorCode
      });
  }
}, "de_CommandError");
var de_ConcurrentModificationExceptionRes = /* @__PURE__ */ __name(async (parsedOutput, context) => {
  const body = parsedOutput.body;
  const deserialized = (0, import_smithy_client._json)(body);
  const exception = new ConcurrentModificationException({
    $metadata: deserializeMetadata(parsedOutput),
    ...deserialized
  });
  return (0, import_smithy_client.decorateServiceException)(exception, body);
}, "de_ConcurrentModificationExceptionRes");
var de_InvalidNextTokenExceptionRes = /* @__PURE__ */ __name(async (parsedOutput, context) => {
  const body = parsedOutput.body;
  const deserialized = (0, import_smithy_client._json)(body);
  const exception = new InvalidNextTokenException({
    $metadata: deserializeMetadata(parsedOutput),
    ...deserialized
  });
  return (0, import_smithy_client.decorateServiceException)(exception, body);
}, "de_InvalidNextTokenExceptionRes");
var de_InvalidServiceRoleExceptionRes = /* @__PURE__ */ __name(async (parsedOutput, context) => {
  const body = parsedOutput.body;
  const deserialized = (0, import_smithy_client._json)(body);
  const exception = new InvalidServiceRoleException({
    $metadata: deserializeMetadata(parsedOutput),
    ...deserialized
  });
  return (0, import_smithy_client.decorateServiceException)(exception, body);
}, "de_InvalidServiceRoleExceptionRes");
var de_LimitExceededExceptionRes = /* @__PURE__ */ __name(async (parsedOutput, context) => {
  const body = parsedOutput.body;
  const deserialized = (0, import_smithy_client._json)(body);
  const exception = new LimitExceededException({
    $metadata: deserializeMetadata(parsedOutput),
    ...deserialized
  });
  return (0, import_smithy_client.decorateServiceException)(exception, body);
}, "de_LimitExceededExceptionRes");
var de_ProjectAlreadyExistsExceptionRes = /* @__PURE__ */ __name(async (parsedOutput, context) => {
  const body = parsedOutput.body;
  const deserialized = (0, import_smithy_client._json)(body);
  const exception = new ProjectAlreadyExistsException({
    $metadata: deserializeMetadata(parsedOutput),
    ...deserialized
  });
  return (0, import_smithy_client.decorateServiceException)(exception, body);
}, "de_ProjectAlreadyExistsExceptionRes");
var de_ProjectConfigurationExceptionRes = /* @__PURE__ */ __name(async (parsedOutput, context) => {
  const body = parsedOutput.body;
  const deserialized = (0, import_smithy_client._json)(body);
  const exception = new ProjectConfigurationException({
    $metadata: deserializeMetadata(parsedOutput),
    ...deserialized
  });
  return (0, import_smithy_client.decorateServiceException)(exception, body);
}, "de_ProjectConfigurationExceptionRes");
var de_ProjectCreationFailedExceptionRes = /* @__PURE__ */ __name(async (parsedOutput, context) => {
  const body = parsedOutput.body;
  const deserialized = (0, import_smithy_client._json)(body);
  const exception = new ProjectCreationFailedException({
    $metadata: deserializeMetadata(parsedOutput),
    ...deserialized
  });
  return (0, import_smithy_client.decorateServiceException)(exception, body);
}, "de_ProjectCreationFailedExceptionRes");
var de_ProjectNotFoundExceptionRes = /* @__PURE__ */ __name(async (parsedOutput, context) => {
  const body = parsedOutput.body;
  const deserialized = (0, import_smithy_client._json)(body);
  const exception = new ProjectNotFoundException({
    $metadata: deserializeMetadata(parsedOutput),
    ...deserialized
  });
  return (0, import_smithy_client.decorateServiceException)(exception, body);
}, "de_ProjectNotFoundExceptionRes");
var de_TeamMemberAlreadyAssociatedExceptionRes = /* @__PURE__ */ __name(async (parsedOutput, context) => {
  const body = parsedOutput.body;
  const deserialized = (0, import_smithy_client._json)(body);
  const exception = new TeamMemberAlreadyAssociatedException({
    $metadata: deserializeMetadata(parsedOutput),
    ...deserialized
  });
  return (0, import_smithy_client.decorateServiceException)(exception, body);
}, "de_TeamMemberAlreadyAssociatedExceptionRes");
var de_TeamMemberNotFoundExceptionRes = /* @__PURE__ */ __name(async (parsedOutput, context) => {
  const body = parsedOutput.body;
  const deserialized = (0, import_smithy_client._json)(body);
  const exception = new TeamMemberNotFoundException({
    $metadata: deserializeMetadata(parsedOutput),
    ...deserialized
  });
  return (0, import_smithy_client.decorateServiceException)(exception, body);
}, "de_TeamMemberNotFoundExceptionRes");
var de_UserProfileAlreadyExistsExceptionRes = /* @__PURE__ */ __name(async (parsedOutput, context) => {
  const body = parsedOutput.body;
  const deserialized = (0, import_smithy_client._json)(body);
  const exception = new UserProfileAlreadyExistsException({
    $metadata: deserializeMetadata(parsedOutput),
    ...deserialized
  });
  return (0, import_smithy_client.decorateServiceException)(exception, body);
}, "de_UserProfileAlreadyExistsExceptionRes");
var de_UserProfileNotFoundExceptionRes = /* @__PURE__ */ __name(async (parsedOutput, context) => {
  const body = parsedOutput.body;
  const deserialized = (0, import_smithy_client._json)(body);
  const exception = new UserProfileNotFoundException({
    $metadata: deserializeMetadata(parsedOutput),
    ...deserialized
  });
  return (0, import_smithy_client.decorateServiceException)(exception, body);
}, "de_UserProfileNotFoundExceptionRes");
var de_ValidationExceptionRes = /* @__PURE__ */ __name(async (parsedOutput, context) => {
  const body = parsedOutput.body;
  const deserialized = (0, import_smithy_client._json)(body);
  const exception = new ValidationException({
    $metadata: deserializeMetadata(parsedOutput),
    ...deserialized
  });
  return (0, import_smithy_client.decorateServiceException)(exception, body);
}, "de_ValidationExceptionRes");
var de_CreateUserProfileResult = /* @__PURE__ */ __name((output, context) => {
  return (0, import_smithy_client.take)(output, {
    createdTimestamp: (_) => (0, import_smithy_client.expectNonNull)((0, import_smithy_client.parseEpochTimestamp)((0, import_smithy_client.expectNumber)(_))),
    displayName: import_smithy_client.expectString,
    emailAddress: import_smithy_client.expectString,
    lastModifiedTimestamp: (_) => (0, import_smithy_client.expectNonNull)((0, import_smithy_client.parseEpochTimestamp)((0, import_smithy_client.expectNumber)(_))),
    sshPublicKey: import_smithy_client.expectString,
    userArn: import_smithy_client.expectString
  });
}, "de_CreateUserProfileResult");
var de_DescribeProjectResult = /* @__PURE__ */ __name((output, context) => {
  return (0, import_smithy_client.take)(output, {
    arn: import_smithy_client.expectString,
    clientRequestToken: import_smithy_client.expectString,
    createdTimeStamp: (_) => (0, import_smithy_client.expectNonNull)((0, import_smithy_client.parseEpochTimestamp)((0, import_smithy_client.expectNumber)(_))),
    description: import_smithy_client.expectString,
    id: import_smithy_client.expectString,
    name: import_smithy_client.expectString,
    projectTemplateId: import_smithy_client.expectString,
    stackId: import_smithy_client.expectString,
    status: import_smithy_client._json
  });
}, "de_DescribeProjectResult");
var de_DescribeUserProfileResult = /* @__PURE__ */ __name((output, context) => {
  return (0, import_smithy_client.take)(output, {
    createdTimestamp: (_) => (0, import_smithy_client.expectNonNull)((0, import_smithy_client.parseEpochTimestamp)((0, import_smithy_client.expectNumber)(_))),
    displayName: import_smithy_client.expectString,
    emailAddress: import_smithy_client.expectString,
    lastModifiedTimestamp: (_) => (0, import_smithy_client.expectNonNull)((0, import_smithy_client.parseEpochTimestamp)((0, import_smithy_client.expectNumber)(_))),
    sshPublicKey: import_smithy_client.expectString,
    userArn: import_smithy_client.expectString
  });
}, "de_DescribeUserProfileResult");
var de_UpdateUserProfileResult = /* @__PURE__ */ __name((output, context) => {
  return (0, import_smithy_client.take)(output, {
    createdTimestamp: (_) => (0, import_smithy_client.expectNonNull)((0, import_smithy_client.parseEpochTimestamp)((0, import_smithy_client.expectNumber)(_))),
    displayName: import_smithy_client.expectString,
    emailAddress: import_smithy_client.expectString,
    lastModifiedTimestamp: (_) => (0, import_smithy_client.expectNonNull)((0, import_smithy_client.parseEpochTimestamp)((0, import_smithy_client.expectNumber)(_))),
    sshPublicKey: import_smithy_client.expectString,
    userArn: import_smithy_client.expectString
  });
}, "de_UpdateUserProfileResult");
var deserializeMetadata = /* @__PURE__ */ __name((output) => ({
  httpStatusCode: output.statusCode,
  requestId: output.headers["x-amzn-requestid"] ?? output.headers["x-amzn-request-id"] ?? output.headers["x-amz-request-id"],
  extendedRequestId: output.headers["x-amz-id-2"],
  cfId: output.headers["x-amz-cf-id"]
}), "deserializeMetadata");
var throwDefaultError = (0, import_smithy_client.withBaseException)(CodeStarServiceException);
var buildHttpRpcRequest = /* @__PURE__ */ __name(async (context, headers, path, resolvedHostname, body) => {
  const { hostname, protocol = "https", port, path: basePath } = await context.endpoint();
  const contents = {
    protocol,
    hostname,
    port,
    method: "POST",
    path: basePath.endsWith("/") ? basePath.slice(0, -1) + path : basePath + path,
    headers
  };
  if (resolvedHostname !== void 0) {
    contents.hostname = resolvedHostname;
  }
  if (body !== void 0) {
    contents.body = body;
  }
  return new import_protocol_http.HttpRequest(contents);
}, "buildHttpRpcRequest");
function sharedHeaders(operation) {
  return {
    "content-type": "application/x-amz-json-1.1",
    "x-amz-target": `CodeStar_20170419.${operation}`
  };
}
__name(sharedHeaders, "sharedHeaders");

// src/commands/AssociateTeamMemberCommand.ts
var _AssociateTeamMemberCommand = class _AssociateTeamMemberCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("CodeStar_20170419", "AssociateTeamMember", {}).n("CodeStarClient", "AssociateTeamMemberCommand").f(void 0, void 0).ser(se_AssociateTeamMemberCommand).de(de_AssociateTeamMemberCommand).build() {
};
__name(_AssociateTeamMemberCommand, "AssociateTeamMemberCommand");
var AssociateTeamMemberCommand = _AssociateTeamMemberCommand;

// src/commands/CreateProjectCommand.ts




var _CreateProjectCommand = class _CreateProjectCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("CodeStar_20170419", "CreateProject", {}).n("CodeStarClient", "CreateProjectCommand").f(CreateProjectRequestFilterSensitiveLog, void 0).ser(se_CreateProjectCommand).de(de_CreateProjectCommand).build() {
};
__name(_CreateProjectCommand, "CreateProjectCommand");
var CreateProjectCommand = _CreateProjectCommand;

// src/commands/CreateUserProfileCommand.ts




var _CreateUserProfileCommand = class _CreateUserProfileCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("CodeStar_20170419", "CreateUserProfile", {}).n("CodeStarClient", "CreateUserProfileCommand").f(CreateUserProfileRequestFilterSensitiveLog, CreateUserProfileResultFilterSensitiveLog).ser(se_CreateUserProfileCommand).de(de_CreateUserProfileCommand).build() {
};
__name(_CreateUserProfileCommand, "CreateUserProfileCommand");
var CreateUserProfileCommand = _CreateUserProfileCommand;

// src/commands/DeleteProjectCommand.ts




var _DeleteProjectCommand = class _DeleteProjectCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("CodeStar_20170419", "DeleteProject", {}).n("CodeStarClient", "DeleteProjectCommand").f(void 0, void 0).ser(se_DeleteProjectCommand).de(de_DeleteProjectCommand).build() {
};
__name(_DeleteProjectCommand, "DeleteProjectCommand");
var DeleteProjectCommand = _DeleteProjectCommand;

// src/commands/DeleteUserProfileCommand.ts




var _DeleteUserProfileCommand = class _DeleteUserProfileCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("CodeStar_20170419", "DeleteUserProfile", {}).n("CodeStarClient", "DeleteUserProfileCommand").f(void 0, void 0).ser(se_DeleteUserProfileCommand).de(de_DeleteUserProfileCommand).build() {
};
__name(_DeleteUserProfileCommand, "DeleteUserProfileCommand");
var DeleteUserProfileCommand = _DeleteUserProfileCommand;

// src/commands/DescribeProjectCommand.ts




var _DescribeProjectCommand = class _DescribeProjectCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("CodeStar_20170419", "DescribeProject", {}).n("CodeStarClient", "DescribeProjectCommand").f(void 0, DescribeProjectResultFilterSensitiveLog).ser(se_DescribeProjectCommand).de(de_DescribeProjectCommand).build() {
};
__name(_DescribeProjectCommand, "DescribeProjectCommand");
var DescribeProjectCommand = _DescribeProjectCommand;

// src/commands/DescribeUserProfileCommand.ts




var _DescribeUserProfileCommand = class _DescribeUserProfileCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("CodeStar_20170419", "DescribeUserProfile", {}).n("CodeStarClient", "DescribeUserProfileCommand").f(void 0, DescribeUserProfileResultFilterSensitiveLog).ser(se_DescribeUserProfileCommand).de(de_DescribeUserProfileCommand).build() {
};
__name(_DescribeUserProfileCommand, "DescribeUserProfileCommand");
var DescribeUserProfileCommand = _DescribeUserProfileCommand;

// src/commands/DisassociateTeamMemberCommand.ts




var _DisassociateTeamMemberCommand = class _DisassociateTeamMemberCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("CodeStar_20170419", "DisassociateTeamMember", {}).n("CodeStarClient", "DisassociateTeamMemberCommand").f(void 0, void 0).ser(se_DisassociateTeamMemberCommand).de(de_DisassociateTeamMemberCommand).build() {
};
__name(_DisassociateTeamMemberCommand, "DisassociateTeamMemberCommand");
var DisassociateTeamMemberCommand = _DisassociateTeamMemberCommand;

// src/commands/ListProjectsCommand.ts




var _ListProjectsCommand = class _ListProjectsCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("CodeStar_20170419", "ListProjects", {}).n("CodeStarClient", "ListProjectsCommand").f(void 0, void 0).ser(se_ListProjectsCommand).de(de_ListProjectsCommand).build() {
};
__name(_ListProjectsCommand, "ListProjectsCommand");
var ListProjectsCommand = _ListProjectsCommand;

// src/commands/ListResourcesCommand.ts




var _ListResourcesCommand = class _ListResourcesCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("CodeStar_20170419", "ListResources", {}).n("CodeStarClient", "ListResourcesCommand").f(void 0, void 0).ser(se_ListResourcesCommand).de(de_ListResourcesCommand).build() {
};
__name(_ListResourcesCommand, "ListResourcesCommand");
var ListResourcesCommand = _ListResourcesCommand;

// src/commands/ListTagsForProjectCommand.ts




var _ListTagsForProjectCommand = class _ListTagsForProjectCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("CodeStar_20170419", "ListTagsForProject", {}).n("CodeStarClient", "ListTagsForProjectCommand").f(void 0, void 0).ser(se_ListTagsForProjectCommand).de(de_ListTagsForProjectCommand).build() {
};
__name(_ListTagsForProjectCommand, "ListTagsForProjectCommand");
var ListTagsForProjectCommand = _ListTagsForProjectCommand;

// src/commands/ListTeamMembersCommand.ts




var _ListTeamMembersCommand = class _ListTeamMembersCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("CodeStar_20170419", "ListTeamMembers", {}).n("CodeStarClient", "ListTeamMembersCommand").f(void 0, void 0).ser(se_ListTeamMembersCommand).de(de_ListTeamMembersCommand).build() {
};
__name(_ListTeamMembersCommand, "ListTeamMembersCommand");
var ListTeamMembersCommand = _ListTeamMembersCommand;

// src/commands/ListUserProfilesCommand.ts




var _ListUserProfilesCommand = class _ListUserProfilesCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("CodeStar_20170419", "ListUserProfiles", {}).n("CodeStarClient", "ListUserProfilesCommand").f(void 0, ListUserProfilesResultFilterSensitiveLog).ser(se_ListUserProfilesCommand).de(de_ListUserProfilesCommand).build() {
};
__name(_ListUserProfilesCommand, "ListUserProfilesCommand");
var ListUserProfilesCommand = _ListUserProfilesCommand;

// src/commands/TagProjectCommand.ts




var _TagProjectCommand = class _TagProjectCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("CodeStar_20170419", "TagProject", {}).n("CodeStarClient", "TagProjectCommand").f(void 0, void 0).ser(se_TagProjectCommand).de(de_TagProjectCommand).build() {
};
__name(_TagProjectCommand, "TagProjectCommand");
var TagProjectCommand = _TagProjectCommand;

// src/commands/UntagProjectCommand.ts




var _UntagProjectCommand = class _UntagProjectCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("CodeStar_20170419", "UntagProject", {}).n("CodeStarClient", "UntagProjectCommand").f(void 0, void 0).ser(se_UntagProjectCommand).de(de_UntagProjectCommand).build() {
};
__name(_UntagProjectCommand, "UntagProjectCommand");
var UntagProjectCommand = _UntagProjectCommand;

// src/commands/UpdateProjectCommand.ts




var _UpdateProjectCommand = class _UpdateProjectCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("CodeStar_20170419", "UpdateProject", {}).n("CodeStarClient", "UpdateProjectCommand").f(UpdateProjectRequestFilterSensitiveLog, void 0).ser(se_UpdateProjectCommand).de(de_UpdateProjectCommand).build() {
};
__name(_UpdateProjectCommand, "UpdateProjectCommand");
var UpdateProjectCommand = _UpdateProjectCommand;

// src/commands/UpdateTeamMemberCommand.ts




var _UpdateTeamMemberCommand = class _UpdateTeamMemberCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("CodeStar_20170419", "UpdateTeamMember", {}).n("CodeStarClient", "UpdateTeamMemberCommand").f(void 0, void 0).ser(se_UpdateTeamMemberCommand).de(de_UpdateTeamMemberCommand).build() {
};
__name(_UpdateTeamMemberCommand, "UpdateTeamMemberCommand");
var UpdateTeamMemberCommand = _UpdateTeamMemberCommand;

// src/commands/UpdateUserProfileCommand.ts




var _UpdateUserProfileCommand = class _UpdateUserProfileCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("CodeStar_20170419", "UpdateUserProfile", {}).n("CodeStarClient", "UpdateUserProfileCommand").f(UpdateUserProfileRequestFilterSensitiveLog, UpdateUserProfileResultFilterSensitiveLog).ser(se_UpdateUserProfileCommand).de(de_UpdateUserProfileCommand).build() {
};
__name(_UpdateUserProfileCommand, "UpdateUserProfileCommand");
var UpdateUserProfileCommand = _UpdateUserProfileCommand;

// src/CodeStar.ts
var commands = {
  AssociateTeamMemberCommand,
  CreateProjectCommand,
  CreateUserProfileCommand,
  DeleteProjectCommand,
  DeleteUserProfileCommand,
  DescribeProjectCommand,
  DescribeUserProfileCommand,
  DisassociateTeamMemberCommand,
  ListProjectsCommand,
  ListResourcesCommand,
  ListTagsForProjectCommand,
  ListTeamMembersCommand,
  ListUserProfilesCommand,
  TagProjectCommand,
  UntagProjectCommand,
  UpdateProjectCommand,
  UpdateTeamMemberCommand,
  UpdateUserProfileCommand
};
var _CodeStar = class _CodeStar extends CodeStarClient {
};
__name(_CodeStar, "CodeStar");
var CodeStar = _CodeStar;
(0, import_smithy_client.createAggregatedClient)(commands, CodeStar);
// Annotate the CommonJS export names for ESM import in node:

0 && (module.exports = {
  CodeStarServiceException,
  __Client,
  CodeStarClient,
  CodeStar,
  $Command,
  AssociateTeamMemberCommand,
  CreateProjectCommand,
  CreateUserProfileCommand,
  DeleteProjectCommand,
  DeleteUserProfileCommand,
  DescribeProjectCommand,
  DescribeUserProfileCommand,
  DisassociateTeamMemberCommand,
  ListProjectsCommand,
  ListResourcesCommand,
  ListTagsForProjectCommand,
  ListTeamMembersCommand,
  ListUserProfilesCommand,
  TagProjectCommand,
  UntagProjectCommand,
  UpdateProjectCommand,
  UpdateTeamMemberCommand,
  UpdateUserProfileCommand,
  ConcurrentModificationException,
  InvalidServiceRoleException,
  LimitExceededException,
  ProjectConfigurationException,
  ProjectNotFoundException,
  TeamMemberAlreadyAssociatedException,
  ValidationException,
  ProjectAlreadyExistsException,
  ProjectCreationFailedException,
  UserProfileAlreadyExistsException,
  UserProfileNotFoundException,
  InvalidNextTokenException,
  TeamMemberNotFoundException,
  GitHubCodeDestinationFilterSensitiveLog,
  CodeDestinationFilterSensitiveLog,
  CodeFilterSensitiveLog,
  ToolchainFilterSensitiveLog,
  CreateProjectRequestFilterSensitiveLog,
  CreateUserProfileRequestFilterSensitiveLog,
  CreateUserProfileResultFilterSensitiveLog,
  DescribeProjectResultFilterSensitiveLog,
  DescribeUserProfileResultFilterSensitiveLog,
  UserProfileSummaryFilterSensitiveLog,
  ListUserProfilesResultFilterSensitiveLog,
  UpdateProjectRequestFilterSensitiveLog,
  UpdateUserProfileRequestFilterSensitiveLog,
  UpdateUserProfileResultFilterSensitiveLog
});
