// @azure/arm-signalr/dist-esm/src/models/index.js
/*
 * Copyright (c) Microsoft Corporation.
 * Licensed under the MIT License.
 *
 * Code generated by Microsoft (R) AutoRest Code Generator.
 * Changes may cause incorrect behavior and will be lost if the code is regenerated.
 */
/** Known values of {@link SignalRSkuTier} that the service accepts. */
export var KnownSignalRSkuTier;
(function (KnownSignalRSkuTier) {
    /** Free */
    KnownSignalRSkuTier["Free"] = "Free";
    /** Basic */
    KnownSignalRSkuTier["Basic"] = "Basic";
    /** Standard */
    KnownSignalRSkuTier["Standard"] = "Standard";
    /** Premium */
    KnownSignalRSkuTier["Premium"] = "Premium";
})(KnownSignalRSkuTier || (KnownSignalRSkuTier = {}));
/** Known values of {@link ProvisioningState} that the service accepts. */
export var KnownProvisioningState;
(function (KnownProvisioningState) {
    /** Unknown */
    KnownProvisioningState["Unknown"] = "Unknown";
    /** Succeeded */
    KnownProvisioningState["Succeeded"] = "Succeeded";
    /** Failed */
    KnownProvisioningState["Failed"] = "Failed";
    /** Canceled */
    KnownProvisioningState["Canceled"] = "Canceled";
    /** Running */
    KnownProvisioningState["Running"] = "Running";
    /** Creating */
    KnownProvisioningState["Creating"] = "Creating";
    /** Updating */
    KnownProvisioningState["Updating"] = "Updating";
    /** Deleting */
    KnownProvisioningState["Deleting"] = "Deleting";
    /** Moving */
    KnownProvisioningState["Moving"] = "Moving";
})(KnownProvisioningState || (KnownProvisioningState = {}));
/** Known values of {@link CreatedByType} that the service accepts. */
export var KnownCreatedByType;
(function (KnownCreatedByType) {
    /** User */
    KnownCreatedByType["User"] = "User";
    /** Application */
    KnownCreatedByType["Application"] = "Application";
    /** ManagedIdentity */
    KnownCreatedByType["ManagedIdentity"] = "ManagedIdentity";
    /** Key */
    KnownCreatedByType["Key"] = "Key";
})(KnownCreatedByType || (KnownCreatedByType = {}));
/** Known values of {@link PrivateLinkServiceConnectionStatus} that the service accepts. */
export var KnownPrivateLinkServiceConnectionStatus;
(function (KnownPrivateLinkServiceConnectionStatus) {
    /** Pending */
    KnownPrivateLinkServiceConnectionStatus["Pending"] = "Pending";
    /** Approved */
    KnownPrivateLinkServiceConnectionStatus["Approved"] = "Approved";
    /** Rejected */
    KnownPrivateLinkServiceConnectionStatus["Rejected"] = "Rejected";
    /** Disconnected */
    KnownPrivateLinkServiceConnectionStatus["Disconnected"] = "Disconnected";
})(KnownPrivateLinkServiceConnectionStatus || (KnownPrivateLinkServiceConnectionStatus = {}));
/** Known values of {@link SharedPrivateLinkResourceStatus} that the service accepts. */
export var KnownSharedPrivateLinkResourceStatus;
(function (KnownSharedPrivateLinkResourceStatus) {
    /** Pending */
    KnownSharedPrivateLinkResourceStatus["Pending"] = "Pending";
    /** Approved */
    KnownSharedPrivateLinkResourceStatus["Approved"] = "Approved";
    /** Rejected */
    KnownSharedPrivateLinkResourceStatus["Rejected"] = "Rejected";
    /** Disconnected */
    KnownSharedPrivateLinkResourceStatus["Disconnected"] = "Disconnected";
    /** Timeout */
    KnownSharedPrivateLinkResourceStatus["Timeout"] = "Timeout";
})(KnownSharedPrivateLinkResourceStatus || (KnownSharedPrivateLinkResourceStatus = {}));
/** Known values of {@link FeatureFlags} that the service accepts. */
export var KnownFeatureFlags;
(function (KnownFeatureFlags) {
    /** ServiceMode */
    KnownFeatureFlags["ServiceMode"] = "ServiceMode";
    /** EnableConnectivityLogs */
    KnownFeatureFlags["EnableConnectivityLogs"] = "EnableConnectivityLogs";
    /** EnableMessagingLogs */
    KnownFeatureFlags["EnableMessagingLogs"] = "EnableMessagingLogs";
    /** EnableLiveTrace */
    KnownFeatureFlags["EnableLiveTrace"] = "EnableLiveTrace";
})(KnownFeatureFlags || (KnownFeatureFlags = {}));
/** Known values of {@link UpstreamAuthType} that the service accepts. */
export var KnownUpstreamAuthType;
(function (KnownUpstreamAuthType) {
    /** None */
    KnownUpstreamAuthType["None"] = "None";
    /** ManagedIdentity */
    KnownUpstreamAuthType["ManagedIdentity"] = "ManagedIdentity";
})(KnownUpstreamAuthType || (KnownUpstreamAuthType = {}));
/** Known values of {@link ACLAction} that the service accepts. */
export var KnownACLAction;
(function (KnownACLAction) {
    /** Allow */
    KnownACLAction["Allow"] = "Allow";
    /** Deny */
    KnownACLAction["Deny"] = "Deny";
})(KnownACLAction || (KnownACLAction = {}));
/** Known values of {@link SignalRRequestType} that the service accepts. */
export var KnownSignalRRequestType;
(function (KnownSignalRRequestType) {
    /** ClientConnection */
    KnownSignalRRequestType["ClientConnection"] = "ClientConnection";
    /** ServerConnection */
    KnownSignalRRequestType["ServerConnection"] = "ServerConnection";
    /** Restapi */
    KnownSignalRRequestType["Restapi"] = "RESTAPI";
    /** Trace */
    KnownSignalRRequestType["Trace"] = "Trace";
})(KnownSignalRRequestType || (KnownSignalRRequestType = {}));
/** Known values of {@link ServiceKind} that the service accepts. */
export var KnownServiceKind;
(function (KnownServiceKind) {
    /** SignalR */
    KnownServiceKind["SignalR"] = "SignalR";
    /** RawWebSockets */
    KnownServiceKind["RawWebSockets"] = "RawWebSockets";
})(KnownServiceKind || (KnownServiceKind = {}));
/** Known values of {@link ManagedIdentityType} that the service accepts. */
export var KnownManagedIdentityType;
(function (KnownManagedIdentityType) {
    /** None */
    KnownManagedIdentityType["None"] = "None";
    /** SystemAssigned */
    KnownManagedIdentityType["SystemAssigned"] = "SystemAssigned";
    /** UserAssigned */
    KnownManagedIdentityType["UserAssigned"] = "UserAssigned";
})(KnownManagedIdentityType || (KnownManagedIdentityType = {}));
/** Known values of {@link KeyType} that the service accepts. */
export var KnownKeyType;
(function (KnownKeyType) {
    /** Primary */
    KnownKeyType["Primary"] = "Primary";
    /** Secondary */
    KnownKeyType["Secondary"] = "Secondary";
    /** Salt */
    KnownKeyType["Salt"] = "Salt";
})(KnownKeyType || (KnownKeyType = {}));
/** Known values of {@link ScaleType} that the service accepts. */
export var KnownScaleType;
(function (KnownScaleType) {
    /** None */
    KnownScaleType["None"] = "None";
    /** Manual */
    KnownScaleType["Manual"] = "Manual";
    /** Automatic */
    KnownScaleType["Automatic"] = "Automatic";
})(KnownScaleType || (KnownScaleType = {}));
