// @azure/arm-synapse/dist-esm/src/models/parameters.js
/*
 * Copyright (c) Microsoft Corporation.
 * Licensed under the MIT License.
 *
 * Code generated by Microsoft (R) AutoRest Code Generator.
 * Changes may cause incorrect behavior and will be lost if the code is regenerated.
 */
import { AzureADOnlyAuthentication as AzureADOnlyAuthenticationMapper, CheckNameAvailabilityRequest as CheckNameAvailabilityRequestMapper, IpFirewallRuleInfo as IpFirewallRuleInfoMapper, ReplaceAllIpFirewallRulesRequest as ReplaceAllIpFirewallRulesRequestMapper, Key as KeyMapper, PrivateEndpointConnection as PrivateEndpointConnectionMapper, PrivateLinkHubPatchInfo as PrivateLinkHubPatchInfoMapper, PrivateLinkHub as PrivateLinkHubMapper, SqlPoolPatchInfo as SqlPoolPatchInfoMapper, SqlPool as SqlPoolMapper, ResourceMoveDefinition as ResourceMoveDefinitionMapper, MetadataSyncConfig as MetadataSyncConfigMapper, GeoBackupPolicy as GeoBackupPolicyMapper, CreateSqlPoolRestorePointDefinition as CreateSqlPoolRestorePointDefinitionMapper, MaintenanceWindows as MaintenanceWindowsMapper, TransparentDataEncryption as TransparentDataEncryptionMapper, SqlPoolBlobAuditingPolicy as SqlPoolBlobAuditingPolicyMapper, SensitivityLabelUpdateList as SensitivityLabelUpdateListMapper, SensitivityLabel as SensitivityLabelMapper, RecommendedSensitivityLabelUpdateList as RecommendedSensitivityLabelUpdateListMapper, SqlPoolVulnerabilityAssessment as SqlPoolVulnerabilityAssessmentMapper, SqlPoolSecurityAlertPolicy as SqlPoolSecurityAlertPolicyMapper, SqlPoolVulnerabilityAssessmentRuleBaseline as SqlPoolVulnerabilityAssessmentRuleBaselineMapper, ExtendedSqlPoolBlobAuditingPolicy as ExtendedSqlPoolBlobAuditingPolicyMapper, DataMaskingPolicy as DataMaskingPolicyMapper, DataMaskingRule as DataMaskingRuleMapper, WorkloadGroup as WorkloadGroupMapper, WorkloadClassifier as WorkloadClassifierMapper, ServerBlobAuditingPolicy as ServerBlobAuditingPolicyMapper, ExtendedServerBlobAuditingPolicy as ExtendedServerBlobAuditingPolicyMapper, ServerSecurityAlertPolicy as ServerSecurityAlertPolicyMapper, ServerVulnerabilityAssessment as ServerVulnerabilityAssessmentMapper, EncryptionProtector as EncryptionProtectorMapper, WorkspacePatchInfo as WorkspacePatchInfoMapper, Workspace as WorkspaceMapper, WorkspaceAadAdminInfo as WorkspaceAadAdminInfoMapper, ManagedIdentitySqlControlSettingsModel as ManagedIdentitySqlControlSettingsModelMapper, BigDataPoolPatchInfo as BigDataPoolPatchInfoMapper, BigDataPoolResourceInfo as BigDataPoolResourceInfoMapper, UpdateIntegrationRuntimeRequest as UpdateIntegrationRuntimeRequestMapper, IntegrationRuntimeResource as IntegrationRuntimeResourceMapper, GetSsisObjectMetadataRequest as GetSsisObjectMetadataRequestMapper, UpdateIntegrationRuntimeNodeRequest as UpdateIntegrationRuntimeNodeRequestMapper, IntegrationRuntimeRegenerateKeyParameters as IntegrationRuntimeRegenerateKeyParametersMapper, KustoPoolCheckNameRequest as KustoPoolCheckNameRequestMapper, KustoPool as KustoPoolMapper, KustoPoolUpdate as KustoPoolUpdateMapper, LanguageExtensionsList as LanguageExtensionsListMapper, FollowerDatabaseDefinition as FollowerDatabaseDefinitionMapper, DatabaseCheckNameRequest as DatabaseCheckNameRequestMapper, AttachedDatabaseConfiguration as AttachedDatabaseConfigurationMapper, Database as DatabaseMapper, DataConnectionCheckNameRequest as DataConnectionCheckNameRequestMapper, DataConnectionValidation as DataConnectionValidationMapper, DataConnection as DataConnectionMapper, ClusterPrincipalAssignmentCheckNameRequest as ClusterPrincipalAssignmentCheckNameRequestMapper, ClusterPrincipalAssignment as ClusterPrincipalAssignmentMapper, DatabasePrincipalAssignmentCheckNameRequest as DatabasePrincipalAssignmentCheckNameRequestMapper, DatabasePrincipalAssignment as DatabasePrincipalAssignmentMapper } from "../models/mappers";
export const accept = {
    parameterPath: "accept",
    mapper: {
        defaultValue: "application/json",
        isConstant: true,
        serializedName: "Accept",
        type: {
            name: "String"
        }
    }
};
export const $host = {
    parameterPath: "$host",
    mapper: {
        serializedName: "$host",
        required: true,
        type: {
            name: "String"
        }
    },
    skipEncoding: true
};
export const apiVersion = {
    parameterPath: "apiVersion",
    mapper: {
        defaultValue: "2021-06-01",
        isConstant: true,
        serializedName: "api-version",
        type: {
            name: "String"
        }
    }
};
export const subscriptionId = {
    parameterPath: "subscriptionId",
    mapper: {
        constraints: {
            MinLength: 1
        },
        serializedName: "subscriptionId",
        required: true,
        type: {
            name: "String"
        }
    }
};
export const resourceGroupName = {
    parameterPath: "resourceGroupName",
    mapper: {
        constraints: {
            MaxLength: 90,
            MinLength: 1
        },
        serializedName: "resourceGroupName",
        required: true,
        type: {
            name: "String"
        }
    }
};
export const workspaceName = {
    parameterPath: "workspaceName",
    mapper: {
        serializedName: "workspaceName",
        required: true,
        type: {
            name: "String"
        }
    }
};
export const azureADOnlyAuthenticationName = {
    parameterPath: "azureADOnlyAuthenticationName",
    mapper: {
        serializedName: "azureADOnlyAuthenticationName",
        required: true,
        type: {
            name: "String"
        }
    }
};
export const contentType = {
    parameterPath: ["options", "contentType"],
    mapper: {
        defaultValue: "application/json",
        isConstant: true,
        serializedName: "Content-Type",
        type: {
            name: "String"
        }
    }
};
export const azureADOnlyAuthenticationInfo = {
    parameterPath: "azureADOnlyAuthenticationInfo",
    mapper: AzureADOnlyAuthenticationMapper
};
export const nextLink = {
    parameterPath: "nextLink",
    mapper: {
        serializedName: "nextLink",
        required: true,
        type: {
            name: "String"
        }
    },
    skipEncoding: true
};
export const request = {
    parameterPath: "request",
    mapper: CheckNameAvailabilityRequestMapper
};
export const operationId = {
    parameterPath: "operationId",
    mapper: {
        serializedName: "operationId",
        required: true,
        type: {
            name: "String"
        }
    }
};
export const ipFirewallRuleInfo = {
    parameterPath: "ipFirewallRuleInfo",
    mapper: IpFirewallRuleInfoMapper
};
export const ruleName = {
    parameterPath: "ruleName",
    mapper: {
        serializedName: "ruleName",
        required: true,
        type: {
            name: "String"
        }
    }
};
export const request1 = {
    parameterPath: "request",
    mapper: ReplaceAllIpFirewallRulesRequestMapper
};
export const keyName = {
    parameterPath: "keyName",
    mapper: {
        serializedName: "keyName",
        required: true,
        type: {
            name: "String"
        }
    }
};
export const keyProperties = {
    parameterPath: "keyProperties",
    mapper: KeyMapper
};
export const privateEndpointConnectionName = {
    parameterPath: "privateEndpointConnectionName",
    mapper: {
        serializedName: "privateEndpointConnectionName",
        required: true,
        type: {
            name: "String"
        }
    }
};
export const request2 = {
    parameterPath: "request",
    mapper: PrivateEndpointConnectionMapper
};
export const privateLinkResourceName = {
    parameterPath: "privateLinkResourceName",
    mapper: {
        serializedName: "privateLinkResourceName",
        required: true,
        type: {
            name: "String"
        }
    }
};
export const privateLinkHubName = {
    parameterPath: "privateLinkHubName",
    mapper: {
        serializedName: "privateLinkHubName",
        required: true,
        type: {
            name: "String"
        }
    }
};
export const privateLinkHubPatchInfo = {
    parameterPath: "privateLinkHubPatchInfo",
    mapper: PrivateLinkHubPatchInfoMapper
};
export const privateLinkHubInfo = {
    parameterPath: "privateLinkHubInfo",
    mapper: PrivateLinkHubMapper
};
export const sqlPoolName = {
    parameterPath: "sqlPoolName",
    mapper: {
        serializedName: "sqlPoolName",
        required: true,
        type: {
            name: "String"
        }
    }
};
export const sqlPoolInfo = {
    parameterPath: "sqlPoolInfo",
    mapper: SqlPoolPatchInfoMapper
};
export const sqlPoolInfo1 = {
    parameterPath: "sqlPoolInfo",
    mapper: SqlPoolMapper
};
export const parameters = {
    parameterPath: "parameters",
    mapper: ResourceMoveDefinitionMapper
};
export const metadataSyncConfiguration = {
    parameterPath: "metadataSyncConfiguration",
    mapper: MetadataSyncConfigMapper
};
export const parameters1 = {
    parameterPath: "parameters",
    mapper: GeoBackupPolicyMapper
};
export const geoBackupPolicyName = {
    parameterPath: "geoBackupPolicyName",
    mapper: {
        serializedName: "geoBackupPolicyName",
        required: true,
        type: {
            name: "String"
        }
    }
};
export const dataWarehouseUserActivityName = {
    parameterPath: "dataWarehouseUserActivityName",
    mapper: {
        serializedName: "dataWarehouseUserActivityName",
        required: true,
        type: {
            name: "String"
        }
    }
};
export const parameters2 = {
    parameterPath: "parameters",
    mapper: CreateSqlPoolRestorePointDefinitionMapper
};
export const restorePointName = {
    parameterPath: "restorePointName",
    mapper: {
        serializedName: "restorePointName",
        required: true,
        type: {
            name: "String"
        }
    }
};
export const linkId = {
    parameterPath: "linkId",
    mapper: {
        serializedName: "linkId",
        required: true,
        type: {
            name: "String"
        }
    }
};
export const maintenanceWindowName = {
    parameterPath: "maintenanceWindowName",
    mapper: {
        serializedName: "maintenanceWindowName",
        required: true,
        type: {
            name: "String"
        }
    }
};
export const parameters3 = {
    parameterPath: "parameters",
    mapper: MaintenanceWindowsMapper
};
export const maintenanceWindowOptionsName = {
    parameterPath: "maintenanceWindowOptionsName",
    mapper: {
        serializedName: "maintenanceWindowOptionsName",
        required: true,
        type: {
            name: "String"
        }
    }
};
export const transparentDataEncryptionName = {
    parameterPath: "transparentDataEncryptionName",
    mapper: {
        serializedName: "transparentDataEncryptionName",
        required: true,
        type: {
            name: "String"
        }
    }
};
export const parameters4 = {
    parameterPath: "parameters",
    mapper: TransparentDataEncryptionMapper
};
export const blobAuditingPolicyName = {
    parameterPath: "blobAuditingPolicyName",
    mapper: {
        defaultValue: "default",
        isConstant: true,
        serializedName: "blobAuditingPolicyName",
        type: {
            name: "String"
        }
    }
};
export const parameters5 = {
    parameterPath: "parameters",
    mapper: SqlPoolBlobAuditingPolicyMapper
};
export const filter = {
    parameterPath: ["options", "filter"],
    mapper: {
        serializedName: "$filter",
        type: {
            name: "String"
        }
    }
};
export const parameters6 = {
    parameterPath: "parameters",
    mapper: SensitivityLabelUpdateListMapper
};
export const includeDisabledRecommendations = {
    parameterPath: ["options", "includeDisabledRecommendations"],
    mapper: {
        serializedName: "includeDisabledRecommendations",
        type: {
            name: "Boolean"
        }
    }
};
export const skipToken = {
    parameterPath: ["options", "skipToken"],
    mapper: {
        serializedName: "$skipToken",
        type: {
            name: "String"
        }
    }
};
export const parameters7 = {
    parameterPath: "parameters",
    mapper: SensitivityLabelMapper
};
export const schemaName = {
    parameterPath: "schemaName",
    mapper: {
        serializedName: "schemaName",
        required: true,
        type: {
            name: "String"
        }
    }
};
export const tableName = {
    parameterPath: "tableName",
    mapper: {
        serializedName: "tableName",
        required: true,
        type: {
            name: "String"
        }
    }
};
export const columnName = {
    parameterPath: "columnName",
    mapper: {
        serializedName: "columnName",
        required: true,
        type: {
            name: "String"
        }
    }
};
export const sensitivityLabelSource = {
    parameterPath: "sensitivityLabelSource",
    mapper: {
        defaultValue: "current",
        isConstant: true,
        serializedName: "sensitivityLabelSource",
        type: {
            name: "String"
        }
    }
};
export const sensitivityLabelSource1 = {
    parameterPath: "sensitivityLabelSource",
    mapper: {
        serializedName: "sensitivityLabelSource",
        required: true,
        type: {
            name: "Enum",
            allowedValues: ["current", "recommended"]
        }
    }
};
export const sensitivityLabelSource2 = {
    parameterPath: "sensitivityLabelSource",
    mapper: {
        defaultValue: "recommended",
        isConstant: true,
        serializedName: "sensitivityLabelSource",
        type: {
            name: "String"
        }
    }
};
export const parameters8 = {
    parameterPath: "parameters",
    mapper: RecommendedSensitivityLabelUpdateListMapper
};
export const connectionPolicyName = {
    parameterPath: "connectionPolicyName",
    mapper: {
        serializedName: "connectionPolicyName",
        required: true,
        type: {
            name: "String"
        }
    }
};
export const vulnerabilityAssessmentName = {
    parameterPath: "vulnerabilityAssessmentName",
    mapper: {
        serializedName: "vulnerabilityAssessmentName",
        required: true,
        type: {
            name: "String"
        }
    }
};
export const parameters9 = {
    parameterPath: "parameters",
    mapper: SqlPoolVulnerabilityAssessmentMapper
};
export const scanId = {
    parameterPath: "scanId",
    mapper: {
        serializedName: "scanId",
        required: true,
        type: {
            name: "String"
        }
    }
};
export const securityAlertPolicyName = {
    parameterPath: "securityAlertPolicyName",
    mapper: {
        serializedName: "securityAlertPolicyName",
        required: true,
        type: {
            name: "String"
        }
    }
};
export const parameters10 = {
    parameterPath: "parameters",
    mapper: SqlPoolSecurityAlertPolicyMapper
};
export const parameters11 = {
    parameterPath: "parameters",
    mapper: SqlPoolVulnerabilityAssessmentRuleBaselineMapper
};
export const ruleId = {
    parameterPath: "ruleId",
    mapper: {
        serializedName: "ruleId",
        required: true,
        type: {
            name: "String"
        }
    }
};
export const baselineName = {
    parameterPath: "baselineName",
    mapper: {
        serializedName: "baselineName",
        required: true,
        type: {
            name: "Enum",
            allowedValues: ["master", "default"]
        }
    }
};
export const parameters12 = {
    parameterPath: "parameters",
    mapper: ExtendedSqlPoolBlobAuditingPolicyMapper
};
export const parameters13 = {
    parameterPath: "parameters",
    mapper: DataMaskingPolicyMapper
};
export const dataMaskingPolicyName = {
    parameterPath: "dataMaskingPolicyName",
    mapper: {
        defaultValue: "Default",
        isConstant: true,
        serializedName: "dataMaskingPolicyName",
        type: {
            name: "String"
        }
    }
};
export const parameters14 = {
    parameterPath: "parameters",
    mapper: DataMaskingRuleMapper
};
export const dataMaskingRuleName = {
    parameterPath: "dataMaskingRuleName",
    mapper: {
        serializedName: "dataMaskingRuleName",
        required: true,
        type: {
            name: "String"
        }
    }
};
export const workloadGroupName = {
    parameterPath: "workloadGroupName",
    mapper: {
        serializedName: "workloadGroupName",
        required: true,
        type: {
            name: "String"
        }
    }
};
export const parameters15 = {
    parameterPath: "parameters",
    mapper: WorkloadGroupMapper
};
export const workloadClassifierName = {
    parameterPath: "workloadClassifierName",
    mapper: {
        serializedName: "workloadClassifierName",
        required: true,
        type: {
            name: "String"
        }
    }
};
export const parameters16 = {
    parameterPath: "parameters",
    mapper: WorkloadClassifierMapper
};
export const blobAuditingPolicyName1 = {
    parameterPath: "blobAuditingPolicyName",
    mapper: {
        serializedName: "blobAuditingPolicyName",
        required: true,
        type: {
            name: "String"
        }
    }
};
export const parameters17 = {
    parameterPath: "parameters",
    mapper: ServerBlobAuditingPolicyMapper
};
export const parameters18 = {
    parameterPath: "parameters",
    mapper: ExtendedServerBlobAuditingPolicyMapper
};
export const securityAlertPolicyName1 = {
    parameterPath: "securityAlertPolicyName",
    mapper: {
        serializedName: "securityAlertPolicyName",
        required: true,
        type: {
            name: "String"
        }
    }
};
export const parameters19 = {
    parameterPath: "parameters",
    mapper: ServerSecurityAlertPolicyMapper
};
export const parameters20 = {
    parameterPath: "parameters",
    mapper: ServerVulnerabilityAssessmentMapper
};
export const encryptionProtectorName = {
    parameterPath: "encryptionProtectorName",
    mapper: {
        serializedName: "encryptionProtectorName",
        required: true,
        type: {
            name: "String"
        }
    }
};
export const parameters21 = {
    parameterPath: "parameters",
    mapper: EncryptionProtectorMapper
};
export const workspacePatchInfo = {
    parameterPath: "workspacePatchInfo",
    mapper: WorkspacePatchInfoMapper
};
export const workspaceInfo = {
    parameterPath: "workspaceInfo",
    mapper: WorkspaceMapper
};
export const aadAdminInfo = {
    parameterPath: "aadAdminInfo",
    mapper: WorkspaceAadAdminInfoMapper
};
export const managedIdentitySqlControlSettings = {
    parameterPath: "managedIdentitySqlControlSettings",
    mapper: ManagedIdentitySqlControlSettingsModelMapper
};
export const restorableDroppedSqlPoolId = {
    parameterPath: "restorableDroppedSqlPoolId",
    mapper: {
        serializedName: "restorableDroppedSqlPoolId",
        required: true,
        type: {
            name: "String"
        }
    }
};
export const apiVersion1 = {
    parameterPath: "apiVersion",
    mapper: {
        defaultValue: "2021-06-01-preview",
        isConstant: true,
        serializedName: "api-version",
        type: {
            name: "String"
        }
    }
};
export const bigDataPoolName = {
    parameterPath: "bigDataPoolName",
    mapper: {
        serializedName: "bigDataPoolName",
        required: true,
        type: {
            name: "String"
        }
    }
};
export const bigDataPoolPatchInfo = {
    parameterPath: "bigDataPoolPatchInfo",
    mapper: BigDataPoolPatchInfoMapper
};
export const bigDataPoolInfo = {
    parameterPath: "bigDataPoolInfo",
    mapper: BigDataPoolResourceInfoMapper
};
export const force = {
    parameterPath: ["options", "force"],
    mapper: {
        defaultValue: false,
        serializedName: "force",
        type: {
            name: "Boolean"
        }
    }
};
export const libraryName = {
    parameterPath: "libraryName",
    mapper: {
        serializedName: "libraryName",
        required: true,
        type: {
            name: "String"
        }
    }
};
export const updateIntegrationRuntimeRequest = {
    parameterPath: "updateIntegrationRuntimeRequest",
    mapper: UpdateIntegrationRuntimeRequestMapper
};
export const integrationRuntimeName = {
    parameterPath: "integrationRuntimeName",
    mapper: {
        serializedName: "integrationRuntimeName",
        required: true,
        type: {
            name: "String"
        }
    }
};
export const ifNoneMatch = {
    parameterPath: ["options", "ifNoneMatch"],
    mapper: {
        serializedName: "If-None-Match",
        type: {
            name: "String"
        }
    }
};
export const integrationRuntime = {
    parameterPath: "integrationRuntime",
    mapper: IntegrationRuntimeResourceMapper
};
export const ifMatch = {
    parameterPath: ["options", "ifMatch"],
    mapper: {
        serializedName: "If-Match",
        type: {
            name: "String"
        }
    }
};
export const nodeName = {
    parameterPath: "nodeName",
    mapper: {
        serializedName: "nodeName",
        required: true,
        type: {
            name: "String"
        }
    }
};
export const getMetadataRequest = {
    parameterPath: ["options", "getMetadataRequest"],
    mapper: GetSsisObjectMetadataRequestMapper
};
export const updateIntegrationRuntimeNodeRequest = {
    parameterPath: "updateIntegrationRuntimeNodeRequest",
    mapper: UpdateIntegrationRuntimeNodeRequestMapper
};
export const regenerateKeyParameters = {
    parameterPath: "regenerateKeyParameters",
    mapper: IntegrationRuntimeRegenerateKeyParametersMapper
};
export const sparkConfigurationName = {
    parameterPath: "sparkConfigurationName",
    mapper: {
        serializedName: "sparkConfigurationName",
        required: true,
        type: {
            name: "String"
        }
    }
};
export const kustoPoolName = {
    parameterPath: "kustoPoolName",
    mapper: KustoPoolCheckNameRequestMapper
};
export const location = {
    parameterPath: "location",
    mapper: {
        constraints: {
            MinLength: 1
        },
        serializedName: "location",
        required: true,
        type: {
            name: "String"
        }
    }
};
export const kustoPoolName1 = {
    parameterPath: "kustoPoolName",
    mapper: {
        serializedName: "kustoPoolName",
        required: true,
        type: {
            name: "String"
        }
    }
};
export const parameters22 = {
    parameterPath: "parameters",
    mapper: KustoPoolMapper
};
export const parameters23 = {
    parameterPath: "parameters",
    mapper: KustoPoolUpdateMapper
};
export const languageExtensionsToAdd = {
    parameterPath: "languageExtensionsToAdd",
    mapper: LanguageExtensionsListMapper
};
export const languageExtensionsToRemove = {
    parameterPath: "languageExtensionsToRemove",
    mapper: LanguageExtensionsListMapper
};
export const followerDatabaseToRemove = {
    parameterPath: "followerDatabaseToRemove",
    mapper: FollowerDatabaseDefinitionMapper
};
export const resourceName = {
    parameterPath: "resourceName",
    mapper: DatabaseCheckNameRequestMapper
};
export const attachedDatabaseConfigurationName = {
    parameterPath: "attachedDatabaseConfigurationName",
    mapper: {
        serializedName: "attachedDatabaseConfigurationName",
        required: true,
        type: {
            name: "String"
        }
    }
};
export const parameters24 = {
    parameterPath: "parameters",
    mapper: AttachedDatabaseConfigurationMapper
};
export const databaseName = {
    parameterPath: "databaseName",
    mapper: {
        serializedName: "databaseName",
        required: true,
        type: {
            name: "String"
        }
    }
};
export const parameters25 = {
    parameterPath: "parameters",
    mapper: DatabaseMapper
};
export const dataConnectionName = {
    parameterPath: "dataConnectionName",
    mapper: DataConnectionCheckNameRequestMapper
};
export const parameters26 = {
    parameterPath: "parameters",
    mapper: DataConnectionValidationMapper
};
export const dataConnectionName1 = {
    parameterPath: "dataConnectionName",
    mapper: {
        serializedName: "dataConnectionName",
        required: true,
        type: {
            name: "String"
        }
    }
};
export const parameters27 = {
    parameterPath: "parameters",
    mapper: DataConnectionMapper
};
export const principalAssignmentName = {
    parameterPath: "principalAssignmentName",
    mapper: ClusterPrincipalAssignmentCheckNameRequestMapper
};
export const principalAssignmentName1 = {
    parameterPath: "principalAssignmentName",
    mapper: {
        serializedName: "principalAssignmentName",
        required: true,
        type: {
            name: "String"
        }
    }
};
export const parameters28 = {
    parameterPath: "parameters",
    mapper: ClusterPrincipalAssignmentMapper
};
export const principalAssignmentName2 = {
    parameterPath: "principalAssignmentName",
    mapper: DatabasePrincipalAssignmentCheckNameRequestMapper
};
export const parameters29 = {
    parameterPath: "parameters",
    mapper: DatabasePrincipalAssignmentMapper
};
