// @aws-sdk/client-grafana/dist-es/models/models_0.js
import { SENSITIVE_STRING, } from "@smithy/smithy-client";
import { GrafanaServiceException as __BaseException } from "./GrafanaServiceException";
export class AccessDeniedException extends __BaseException {
    constructor(opts) {
        super({
            name: "AccessDeniedException",
            $fault: "client",
            ...opts,
        });
        this.name = "AccessDeniedException";
        this.$fault = "client";
        Object.setPrototypeOf(this, AccessDeniedException.prototype);
    }
}
export const AccountAccessType = {
    CURRENT_ACCOUNT: "CURRENT_ACCOUNT",
    ORGANIZATION: "ORGANIZATION",
};
export class ConflictException extends __BaseException {
    constructor(opts) {
        super({
            name: "ConflictException",
            $fault: "client",
            ...opts,
        });
        this.name = "ConflictException";
        this.$fault = "client";
        Object.setPrototypeOf(this, ConflictException.prototype);
        this.resourceId = opts.resourceId;
        this.resourceType = opts.resourceType;
    }
}
export class InternalServerException extends __BaseException {
    constructor(opts) {
        super({
            name: "InternalServerException",
            $fault: "server",
            ...opts,
        });
        this.name = "InternalServerException";
        this.$fault = "server";
        this.$retryable = {};
        Object.setPrototypeOf(this, InternalServerException.prototype);
        this.retryAfterSeconds = opts.retryAfterSeconds;
    }
}
export class ResourceNotFoundException extends __BaseException {
    constructor(opts) {
        super({
            name: "ResourceNotFoundException",
            $fault: "client",
            ...opts,
        });
        this.name = "ResourceNotFoundException";
        this.$fault = "client";
        Object.setPrototypeOf(this, ResourceNotFoundException.prototype);
        this.resourceId = opts.resourceId;
        this.resourceType = opts.resourceType;
    }
}
export class ServiceQuotaExceededException extends __BaseException {
    constructor(opts) {
        super({
            name: "ServiceQuotaExceededException",
            $fault: "client",
            ...opts,
        });
        this.name = "ServiceQuotaExceededException";
        this.$fault = "client";
        Object.setPrototypeOf(this, ServiceQuotaExceededException.prototype);
        this.resourceId = opts.resourceId;
        this.resourceType = opts.resourceType;
        this.serviceCode = opts.serviceCode;
        this.quotaCode = opts.quotaCode;
    }
}
export class ThrottlingException extends __BaseException {
    constructor(opts) {
        super({
            name: "ThrottlingException",
            $fault: "client",
            ...opts,
        });
        this.name = "ThrottlingException";
        this.$fault = "client";
        this.$retryable = {};
        Object.setPrototypeOf(this, ThrottlingException.prototype);
        this.serviceCode = opts.serviceCode;
        this.quotaCode = opts.quotaCode;
        this.retryAfterSeconds = opts.retryAfterSeconds;
    }
}
export const ValidationExceptionReason = {
    CANNOT_PARSE: "CANNOT_PARSE",
    FIELD_VALIDATION_FAILED: "FIELD_VALIDATION_FAILED",
    OTHER: "OTHER",
    UNKNOWN_OPERATION: "UNKNOWN_OPERATION",
};
export class ValidationException extends __BaseException {
    constructor(opts) {
        super({
            name: "ValidationException",
            $fault: "client",
            ...opts,
        });
        this.name = "ValidationException";
        this.$fault = "client";
        Object.setPrototypeOf(this, ValidationException.prototype);
        this.reason = opts.reason;
        this.fieldList = opts.fieldList;
    }
}
export const LicenseType = {
    ENTERPRISE: "ENTERPRISE",
    ENTERPRISE_FREE_TRIAL: "ENTERPRISE_FREE_TRIAL",
};
export const AuthenticationProviderTypes = {
    AWS_SSO: "AWS_SSO",
    SAML: "SAML",
};
export const SamlConfigurationStatus = {
    CONFIGURED: "CONFIGURED",
    NOT_CONFIGURED: "NOT_CONFIGURED",
};
export const DataSourceType = {
    AMAZON_OPENSEARCH_SERVICE: "AMAZON_OPENSEARCH_SERVICE",
    ATHENA: "ATHENA",
    CLOUDWATCH: "CLOUDWATCH",
    PROMETHEUS: "PROMETHEUS",
    REDSHIFT: "REDSHIFT",
    SITEWISE: "SITEWISE",
    TIMESTREAM: "TIMESTREAM",
    TWINMAKER: "TWINMAKER",
    XRAY: "XRAY",
};
export const NotificationDestinationType = {
    SNS: "SNS",
};
export const PermissionType = {
    CUSTOMER_MANAGED: "CUSTOMER_MANAGED",
    SERVICE_MANAGED: "SERVICE_MANAGED",
};
export const WorkspaceStatus = {
    ACTIVE: "ACTIVE",
    CREATING: "CREATING",
    CREATION_FAILED: "CREATION_FAILED",
    DELETING: "DELETING",
    DELETION_FAILED: "DELETION_FAILED",
    FAILED: "FAILED",
    LICENSE_REMOVAL_FAILED: "LICENSE_REMOVAL_FAILED",
    UPDATE_FAILED: "UPDATE_FAILED",
    UPDATING: "UPDATING",
    UPGRADE_FAILED: "UPGRADE_FAILED",
    UPGRADING: "UPGRADING",
    VERSION_UPDATE_FAILED: "VERSION_UPDATE_FAILED",
    VERSION_UPDATING: "VERSION_UPDATING",
};
export var IdpMetadata;
(function (IdpMetadata) {
    IdpMetadata.visit = (value, visitor) => {
        if (value.url !== undefined)
            return visitor.url(value.url);
        if (value.xml !== undefined)
            return visitor.xml(value.xml);
        return visitor._(value.$unknown[0], value.$unknown[1]);
    };
})(IdpMetadata || (IdpMetadata = {}));
export const UserType = {
    SSO_GROUP: "SSO_GROUP",
    SSO_USER: "SSO_USER",
};
export const Role = {
    ADMIN: "ADMIN",
    EDITOR: "EDITOR",
    VIEWER: "VIEWER",
};
export const UpdateAction = {
    ADD: "ADD",
    REVOKE: "REVOKE",
};
export const CreateWorkspaceApiKeyResponseFilterSensitiveLog = (obj) => ({
    ...obj,
    ...(obj.key && { key: SENSITIVE_STRING }),
});
export const WorkspaceDescriptionFilterSensitiveLog = (obj) => ({
    ...obj,
    ...(obj.description && { description: SENSITIVE_STRING }),
    ...(obj.name && { name: SENSITIVE_STRING }),
    ...(obj.organizationRoleName && { organizationRoleName: SENSITIVE_STRING }),
    ...(obj.organizationalUnits && { organizationalUnits: SENSITIVE_STRING }),
    ...(obj.workspaceRoleArn && { workspaceRoleArn: SENSITIVE_STRING }),
});
export const AssociateLicenseResponseFilterSensitiveLog = (obj) => ({
    ...obj,
    ...(obj.workspace && { workspace: WorkspaceDescriptionFilterSensitiveLog(obj.workspace) }),
});
export const RoleValuesFilterSensitiveLog = (obj) => ({
    ...obj,
    ...(obj.editor && { editor: SENSITIVE_STRING }),
    ...(obj.admin && { admin: SENSITIVE_STRING }),
});
export const SamlConfigurationFilterSensitiveLog = (obj) => ({
    ...obj,
    ...(obj.idpMetadata && { idpMetadata: obj.idpMetadata }),
    ...(obj.roleValues && { roleValues: RoleValuesFilterSensitiveLog(obj.roleValues) }),
});
export const SamlAuthenticationFilterSensitiveLog = (obj) => ({
    ...obj,
    ...(obj.configuration && { configuration: SamlConfigurationFilterSensitiveLog(obj.configuration) }),
});
export const AuthenticationDescriptionFilterSensitiveLog = (obj) => ({
    ...obj,
    ...(obj.saml && { saml: SamlAuthenticationFilterSensitiveLog(obj.saml) }),
});
export const DescribeWorkspaceAuthenticationResponseFilterSensitiveLog = (obj) => ({
    ...obj,
    ...(obj.authentication && { authentication: AuthenticationDescriptionFilterSensitiveLog(obj.authentication) }),
});
export const UpdateWorkspaceAuthenticationRequestFilterSensitiveLog = (obj) => ({
    ...obj,
    ...(obj.samlConfiguration && { samlConfiguration: SamlConfigurationFilterSensitiveLog(obj.samlConfiguration) }),
});
export const UpdateWorkspaceAuthenticationResponseFilterSensitiveLog = (obj) => ({
    ...obj,
    ...(obj.authentication && { authentication: AuthenticationDescriptionFilterSensitiveLog(obj.authentication) }),
});
export const DisassociateLicenseResponseFilterSensitiveLog = (obj) => ({
    ...obj,
    ...(obj.workspace && { workspace: WorkspaceDescriptionFilterSensitiveLog(obj.workspace) }),
});
export const CreateWorkspaceRequestFilterSensitiveLog = (obj) => ({
    ...obj,
    ...(obj.organizationRoleName && { organizationRoleName: SENSITIVE_STRING }),
    ...(obj.workspaceDescription && { workspaceDescription: SENSITIVE_STRING }),
    ...(obj.workspaceName && { workspaceName: SENSITIVE_STRING }),
    ...(obj.workspaceOrganizationalUnits && { workspaceOrganizationalUnits: SENSITIVE_STRING }),
    ...(obj.workspaceRoleArn && { workspaceRoleArn: SENSITIVE_STRING }),
});
export const CreateWorkspaceResponseFilterSensitiveLog = (obj) => ({
    ...obj,
    ...(obj.workspace && { workspace: WorkspaceDescriptionFilterSensitiveLog(obj.workspace) }),
});
export const DeleteWorkspaceResponseFilterSensitiveLog = (obj) => ({
    ...obj,
    ...(obj.workspace && { workspace: WorkspaceDescriptionFilterSensitiveLog(obj.workspace) }),
});
export const DescribeWorkspaceResponseFilterSensitiveLog = (obj) => ({
    ...obj,
    ...(obj.workspace && { workspace: WorkspaceDescriptionFilterSensitiveLog(obj.workspace) }),
});
export const WorkspaceSummaryFilterSensitiveLog = (obj) => ({
    ...obj,
    ...(obj.description && { description: SENSITIVE_STRING }),
    ...(obj.name && { name: SENSITIVE_STRING }),
});
export const ListWorkspacesResponseFilterSensitiveLog = (obj) => ({
    ...obj,
    ...(obj.workspaces && { workspaces: obj.workspaces.map((item) => WorkspaceSummaryFilterSensitiveLog(item)) }),
});
export const UpdateWorkspaceRequestFilterSensitiveLog = (obj) => ({
    ...obj,
    ...(obj.organizationRoleName && { organizationRoleName: SENSITIVE_STRING }),
    ...(obj.workspaceDescription && { workspaceDescription: SENSITIVE_STRING }),
    ...(obj.workspaceName && { workspaceName: SENSITIVE_STRING }),
    ...(obj.workspaceOrganizationalUnits && { workspaceOrganizationalUnits: SENSITIVE_STRING }),
    ...(obj.workspaceRoleArn && { workspaceRoleArn: SENSITIVE_STRING }),
});
export const UpdateWorkspaceResponseFilterSensitiveLog = (obj) => ({
    ...obj,
    ...(obj.workspace && { workspace: WorkspaceDescriptionFilterSensitiveLog(obj.workspace) }),
});
