// @azure/arm-eventgrid/dist-esm/src/models/index.js
/*
 * Copyright (c) Microsoft Corporation.
 * Licensed under the MIT License.
 *
 * Code generated by Microsoft (R) AutoRest Code Generator.
 * Changes may cause incorrect behavior and will be lost if the code is regenerated.
 */
/** Known values of {@link ChannelType} that the service accepts. */
export var KnownChannelType;
(function (KnownChannelType) {
    /** PartnerTopic */
    KnownChannelType["PartnerTopic"] = "PartnerTopic";
})(KnownChannelType || (KnownChannelType = {}));
/** Known values of {@link EventDefinitionKind} that the service accepts. */
export var KnownEventDefinitionKind;
(function (KnownEventDefinitionKind) {
    /** Inline */
    KnownEventDefinitionKind["Inline"] = "Inline";
})(KnownEventDefinitionKind || (KnownEventDefinitionKind = {}));
/** Known values of {@link ChannelProvisioningState} that the service accepts. */
export var KnownChannelProvisioningState;
(function (KnownChannelProvisioningState) {
    /** Creating */
    KnownChannelProvisioningState["Creating"] = "Creating";
    /** Updating */
    KnownChannelProvisioningState["Updating"] = "Updating";
    /** Deleting */
    KnownChannelProvisioningState["Deleting"] = "Deleting";
    /** Succeeded */
    KnownChannelProvisioningState["Succeeded"] = "Succeeded";
    /** Canceled */
    KnownChannelProvisioningState["Canceled"] = "Canceled";
    /** Failed */
    KnownChannelProvisioningState["Failed"] = "Failed";
    /** IdleDueToMirroredPartnerTopicDeletion */
    KnownChannelProvisioningState["IdleDueToMirroredPartnerTopicDeletion"] = "IdleDueToMirroredPartnerTopicDeletion";
})(KnownChannelProvisioningState || (KnownChannelProvisioningState = {}));
/** Known values of {@link ReadinessState} that the service accepts. */
export var KnownReadinessState;
(function (KnownReadinessState) {
    /** NeverActivated */
    KnownReadinessState["NeverActivated"] = "NeverActivated";
    /** Activated */
    KnownReadinessState["Activated"] = "Activated";
})(KnownReadinessState || (KnownReadinessState = {}));
/** Known values of {@link CreatedByType} that the service accepts. */
export var KnownCreatedByType;
(function (KnownCreatedByType) {
    /** User */
    KnownCreatedByType["User"] = "User";
    /** Application */
    KnownCreatedByType["Application"] = "Application";
    /** ManagedIdentity */
    KnownCreatedByType["ManagedIdentity"] = "ManagedIdentity";
    /** Key */
    KnownCreatedByType["Key"] = "Key";
})(KnownCreatedByType || (KnownCreatedByType = {}));
/** Known values of {@link PersistedConnectionStatus} that the service accepts. */
export var KnownPersistedConnectionStatus;
(function (KnownPersistedConnectionStatus) {
    /** Pending */
    KnownPersistedConnectionStatus["Pending"] = "Pending";
    /** Approved */
    KnownPersistedConnectionStatus["Approved"] = "Approved";
    /** Rejected */
    KnownPersistedConnectionStatus["Rejected"] = "Rejected";
    /** Disconnected */
    KnownPersistedConnectionStatus["Disconnected"] = "Disconnected";
})(KnownPersistedConnectionStatus || (KnownPersistedConnectionStatus = {}));
/** Known values of {@link ResourceProvisioningState} that the service accepts. */
export var KnownResourceProvisioningState;
(function (KnownResourceProvisioningState) {
    /** Creating */
    KnownResourceProvisioningState["Creating"] = "Creating";
    /** Updating */
    KnownResourceProvisioningState["Updating"] = "Updating";
    /** Deleting */
    KnownResourceProvisioningState["Deleting"] = "Deleting";
    /** Succeeded */
    KnownResourceProvisioningState["Succeeded"] = "Succeeded";
    /** Canceled */
    KnownResourceProvisioningState["Canceled"] = "Canceled";
    /** Failed */
    KnownResourceProvisioningState["Failed"] = "Failed";
})(KnownResourceProvisioningState || (KnownResourceProvisioningState = {}));
/** Known values of {@link DomainProvisioningState} that the service accepts. */
export var KnownDomainProvisioningState;
(function (KnownDomainProvisioningState) {
    /** Creating */
    KnownDomainProvisioningState["Creating"] = "Creating";
    /** Updating */
    KnownDomainProvisioningState["Updating"] = "Updating";
    /** Deleting */
    KnownDomainProvisioningState["Deleting"] = "Deleting";
    /** Succeeded */
    KnownDomainProvisioningState["Succeeded"] = "Succeeded";
    /** Canceled */
    KnownDomainProvisioningState["Canceled"] = "Canceled";
    /** Failed */
    KnownDomainProvisioningState["Failed"] = "Failed";
})(KnownDomainProvisioningState || (KnownDomainProvisioningState = {}));
/** Known values of {@link InputSchema} that the service accepts. */
export var KnownInputSchema;
(function (KnownInputSchema) {
    /** EventGridSchema */
    KnownInputSchema["EventGridSchema"] = "EventGridSchema";
    /** CustomEventSchema */
    KnownInputSchema["CustomEventSchema"] = "CustomEventSchema";
    /** CloudEventSchemaV10 */
    KnownInputSchema["CloudEventSchemaV10"] = "CloudEventSchemaV1_0";
})(KnownInputSchema || (KnownInputSchema = {}));
/** Known values of {@link InputSchemaMappingType} that the service accepts. */
export var KnownInputSchemaMappingType;
(function (KnownInputSchemaMappingType) {
    /** Json */
    KnownInputSchemaMappingType["Json"] = "Json";
})(KnownInputSchemaMappingType || (KnownInputSchemaMappingType = {}));
/** Known values of {@link PublicNetworkAccess} that the service accepts. */
export var KnownPublicNetworkAccess;
(function (KnownPublicNetworkAccess) {
    /** Enabled */
    KnownPublicNetworkAccess["Enabled"] = "Enabled";
    /** Disabled */
    KnownPublicNetworkAccess["Disabled"] = "Disabled";
})(KnownPublicNetworkAccess || (KnownPublicNetworkAccess = {}));
/** Known values of {@link IpActionType} that the service accepts. */
export var KnownIpActionType;
(function (KnownIpActionType) {
    /** Allow */
    KnownIpActionType["Allow"] = "Allow";
})(KnownIpActionType || (KnownIpActionType = {}));
/** Known values of {@link DataResidencyBoundary} that the service accepts. */
export var KnownDataResidencyBoundary;
(function (KnownDataResidencyBoundary) {
    /** WithinGeopair */
    KnownDataResidencyBoundary["WithinGeopair"] = "WithinGeopair";
    /** WithinRegion */
    KnownDataResidencyBoundary["WithinRegion"] = "WithinRegion";
})(KnownDataResidencyBoundary || (KnownDataResidencyBoundary = {}));
/** Known values of {@link IdentityType} that the service accepts. */
export var KnownIdentityType;
(function (KnownIdentityType) {
    /** None */
    KnownIdentityType["None"] = "None";
    /** SystemAssigned */
    KnownIdentityType["SystemAssigned"] = "SystemAssigned";
    /** UserAssigned */
    KnownIdentityType["UserAssigned"] = "UserAssigned";
    /** SystemAssignedUserAssigned */
    KnownIdentityType["SystemAssignedUserAssigned"] = "SystemAssigned, UserAssigned";
})(KnownIdentityType || (KnownIdentityType = {}));
/** Known values of {@link DomainTopicProvisioningState} that the service accepts. */
export var KnownDomainTopicProvisioningState;
(function (KnownDomainTopicProvisioningState) {
    /** Creating */
    KnownDomainTopicProvisioningState["Creating"] = "Creating";
    /** Updating */
    KnownDomainTopicProvisioningState["Updating"] = "Updating";
    /** Deleting */
    KnownDomainTopicProvisioningState["Deleting"] = "Deleting";
    /** Succeeded */
    KnownDomainTopicProvisioningState["Succeeded"] = "Succeeded";
    /** Canceled */
    KnownDomainTopicProvisioningState["Canceled"] = "Canceled";
    /** Failed */
    KnownDomainTopicProvisioningState["Failed"] = "Failed";
})(KnownDomainTopicProvisioningState || (KnownDomainTopicProvisioningState = {}));
/** Known values of {@link DeliveryAttributeMappingType} that the service accepts. */
export var KnownDeliveryAttributeMappingType;
(function (KnownDeliveryAttributeMappingType) {
    /** Static */
    KnownDeliveryAttributeMappingType["Static"] = "Static";
    /** Dynamic */
    KnownDeliveryAttributeMappingType["Dynamic"] = "Dynamic";
})(KnownDeliveryAttributeMappingType || (KnownDeliveryAttributeMappingType = {}));
/** Known values of {@link EventSubscriptionProvisioningState} that the service accepts. */
export var KnownEventSubscriptionProvisioningState;
(function (KnownEventSubscriptionProvisioningState) {
    /** Creating */
    KnownEventSubscriptionProvisioningState["Creating"] = "Creating";
    /** Updating */
    KnownEventSubscriptionProvisioningState["Updating"] = "Updating";
    /** Deleting */
    KnownEventSubscriptionProvisioningState["Deleting"] = "Deleting";
    /** Succeeded */
    KnownEventSubscriptionProvisioningState["Succeeded"] = "Succeeded";
    /** Canceled */
    KnownEventSubscriptionProvisioningState["Canceled"] = "Canceled";
    /** Failed */
    KnownEventSubscriptionProvisioningState["Failed"] = "Failed";
    /** AwaitingManualAction */
    KnownEventSubscriptionProvisioningState["AwaitingManualAction"] = "AwaitingManualAction";
})(KnownEventSubscriptionProvisioningState || (KnownEventSubscriptionProvisioningState = {}));
/** Known values of {@link EndpointType} that the service accepts. */
export var KnownEndpointType;
(function (KnownEndpointType) {
    /** WebHook */
    KnownEndpointType["WebHook"] = "WebHook";
    /** EventHub */
    KnownEndpointType["EventHub"] = "EventHub";
    /** StorageQueue */
    KnownEndpointType["StorageQueue"] = "StorageQueue";
    /** HybridConnection */
    KnownEndpointType["HybridConnection"] = "HybridConnection";
    /** ServiceBusQueue */
    KnownEndpointType["ServiceBusQueue"] = "ServiceBusQueue";
    /** ServiceBusTopic */
    KnownEndpointType["ServiceBusTopic"] = "ServiceBusTopic";
    /** AzureFunction */
    KnownEndpointType["AzureFunction"] = "AzureFunction";
})(KnownEndpointType || (KnownEndpointType = {}));
/** Known values of {@link EventSubscriptionIdentityType} that the service accepts. */
export var KnownEventSubscriptionIdentityType;
(function (KnownEventSubscriptionIdentityType) {
    /** SystemAssigned */
    KnownEventSubscriptionIdentityType["SystemAssigned"] = "SystemAssigned";
    /** UserAssigned */
    KnownEventSubscriptionIdentityType["UserAssigned"] = "UserAssigned";
})(KnownEventSubscriptionIdentityType || (KnownEventSubscriptionIdentityType = {}));
/** Known values of {@link AdvancedFilterOperatorType} that the service accepts. */
export var KnownAdvancedFilterOperatorType;
(function (KnownAdvancedFilterOperatorType) {
    /** NumberIn */
    KnownAdvancedFilterOperatorType["NumberIn"] = "NumberIn";
    /** NumberNotIn */
    KnownAdvancedFilterOperatorType["NumberNotIn"] = "NumberNotIn";
    /** NumberLessThan */
    KnownAdvancedFilterOperatorType["NumberLessThan"] = "NumberLessThan";
    /** NumberGreaterThan */
    KnownAdvancedFilterOperatorType["NumberGreaterThan"] = "NumberGreaterThan";
    /** NumberLessThanOrEquals */
    KnownAdvancedFilterOperatorType["NumberLessThanOrEquals"] = "NumberLessThanOrEquals";
    /** NumberGreaterThanOrEquals */
    KnownAdvancedFilterOperatorType["NumberGreaterThanOrEquals"] = "NumberGreaterThanOrEquals";
    /** BoolEquals */
    KnownAdvancedFilterOperatorType["BoolEquals"] = "BoolEquals";
    /** StringIn */
    KnownAdvancedFilterOperatorType["StringIn"] = "StringIn";
    /** StringNotIn */
    KnownAdvancedFilterOperatorType["StringNotIn"] = "StringNotIn";
    /** StringBeginsWith */
    KnownAdvancedFilterOperatorType["StringBeginsWith"] = "StringBeginsWith";
    /** StringEndsWith */
    KnownAdvancedFilterOperatorType["StringEndsWith"] = "StringEndsWith";
    /** StringContains */
    KnownAdvancedFilterOperatorType["StringContains"] = "StringContains";
    /** NumberInRange */
    KnownAdvancedFilterOperatorType["NumberInRange"] = "NumberInRange";
    /** NumberNotInRange */
    KnownAdvancedFilterOperatorType["NumberNotInRange"] = "NumberNotInRange";
    /** StringNotBeginsWith */
    KnownAdvancedFilterOperatorType["StringNotBeginsWith"] = "StringNotBeginsWith";
    /** StringNotEndsWith */
    KnownAdvancedFilterOperatorType["StringNotEndsWith"] = "StringNotEndsWith";
    /** StringNotContains */
    KnownAdvancedFilterOperatorType["StringNotContains"] = "StringNotContains";
    /** IsNullOrUndefined */
    KnownAdvancedFilterOperatorType["IsNullOrUndefined"] = "IsNullOrUndefined";
    /** IsNotNull */
    KnownAdvancedFilterOperatorType["IsNotNull"] = "IsNotNull";
})(KnownAdvancedFilterOperatorType || (KnownAdvancedFilterOperatorType = {}));
/** Known values of {@link EventDeliverySchema} that the service accepts. */
export var KnownEventDeliverySchema;
(function (KnownEventDeliverySchema) {
    /** EventGridSchema */
    KnownEventDeliverySchema["EventGridSchema"] = "EventGridSchema";
    /** CustomInputSchema */
    KnownEventDeliverySchema["CustomInputSchema"] = "CustomInputSchema";
    /** CloudEventSchemaV10 */
    KnownEventDeliverySchema["CloudEventSchemaV10"] = "CloudEventSchemaV1_0";
})(KnownEventDeliverySchema || (KnownEventDeliverySchema = {}));
/** Known values of {@link DeadLetterEndPointType} that the service accepts. */
export var KnownDeadLetterEndPointType;
(function (KnownDeadLetterEndPointType) {
    /** StorageBlob */
    KnownDeadLetterEndPointType["StorageBlob"] = "StorageBlob";
})(KnownDeadLetterEndPointType || (KnownDeadLetterEndPointType = {}));
/** Known values of {@link TopicProvisioningState} that the service accepts. */
export var KnownTopicProvisioningState;
(function (KnownTopicProvisioningState) {
    /** Creating */
    KnownTopicProvisioningState["Creating"] = "Creating";
    /** Updating */
    KnownTopicProvisioningState["Updating"] = "Updating";
    /** Deleting */
    KnownTopicProvisioningState["Deleting"] = "Deleting";
    /** Succeeded */
    KnownTopicProvisioningState["Succeeded"] = "Succeeded";
    /** Canceled */
    KnownTopicProvisioningState["Canceled"] = "Canceled";
    /** Failed */
    KnownTopicProvisioningState["Failed"] = "Failed";
})(KnownTopicProvisioningState || (KnownTopicProvisioningState = {}));
/** Known values of {@link PartnerConfigurationProvisioningState} that the service accepts. */
export var KnownPartnerConfigurationProvisioningState;
(function (KnownPartnerConfigurationProvisioningState) {
    /** Creating */
    KnownPartnerConfigurationProvisioningState["Creating"] = "Creating";
    /** Updating */
    KnownPartnerConfigurationProvisioningState["Updating"] = "Updating";
    /** Deleting */
    KnownPartnerConfigurationProvisioningState["Deleting"] = "Deleting";
    /** Succeeded */
    KnownPartnerConfigurationProvisioningState["Succeeded"] = "Succeeded";
    /** Canceled */
    KnownPartnerConfigurationProvisioningState["Canceled"] = "Canceled";
    /** Failed */
    KnownPartnerConfigurationProvisioningState["Failed"] = "Failed";
})(KnownPartnerConfigurationProvisioningState || (KnownPartnerConfigurationProvisioningState = {}));
/** Known values of {@link PartnerNamespaceProvisioningState} that the service accepts. */
export var KnownPartnerNamespaceProvisioningState;
(function (KnownPartnerNamespaceProvisioningState) {
    /** Creating */
    KnownPartnerNamespaceProvisioningState["Creating"] = "Creating";
    /** Updating */
    KnownPartnerNamespaceProvisioningState["Updating"] = "Updating";
    /** Deleting */
    KnownPartnerNamespaceProvisioningState["Deleting"] = "Deleting";
    /** Succeeded */
    KnownPartnerNamespaceProvisioningState["Succeeded"] = "Succeeded";
    /** Canceled */
    KnownPartnerNamespaceProvisioningState["Canceled"] = "Canceled";
    /** Failed */
    KnownPartnerNamespaceProvisioningState["Failed"] = "Failed";
})(KnownPartnerNamespaceProvisioningState || (KnownPartnerNamespaceProvisioningState = {}));
/** Known values of {@link PartnerTopicRoutingMode} that the service accepts. */
export var KnownPartnerTopicRoutingMode;
(function (KnownPartnerTopicRoutingMode) {
    /** SourceEventAttribute */
    KnownPartnerTopicRoutingMode["SourceEventAttribute"] = "SourceEventAttribute";
    /** ChannelNameHeader */
    KnownPartnerTopicRoutingMode["ChannelNameHeader"] = "ChannelNameHeader";
})(KnownPartnerTopicRoutingMode || (KnownPartnerTopicRoutingMode = {}));
/** Known values of {@link PartnerRegistrationProvisioningState} that the service accepts. */
export var KnownPartnerRegistrationProvisioningState;
(function (KnownPartnerRegistrationProvisioningState) {
    /** Creating */
    KnownPartnerRegistrationProvisioningState["Creating"] = "Creating";
    /** Updating */
    KnownPartnerRegistrationProvisioningState["Updating"] = "Updating";
    /** Deleting */
    KnownPartnerRegistrationProvisioningState["Deleting"] = "Deleting";
    /** Succeeded */
    KnownPartnerRegistrationProvisioningState["Succeeded"] = "Succeeded";
    /** Canceled */
    KnownPartnerRegistrationProvisioningState["Canceled"] = "Canceled";
    /** Failed */
    KnownPartnerRegistrationProvisioningState["Failed"] = "Failed";
})(KnownPartnerRegistrationProvisioningState || (KnownPartnerRegistrationProvisioningState = {}));
/** Known values of {@link PartnerTopicProvisioningState} that the service accepts. */
export var KnownPartnerTopicProvisioningState;
(function (KnownPartnerTopicProvisioningState) {
    /** Creating */
    KnownPartnerTopicProvisioningState["Creating"] = "Creating";
    /** Updating */
    KnownPartnerTopicProvisioningState["Updating"] = "Updating";
    /** Deleting */
    KnownPartnerTopicProvisioningState["Deleting"] = "Deleting";
    /** Succeeded */
    KnownPartnerTopicProvisioningState["Succeeded"] = "Succeeded";
    /** Canceled */
    KnownPartnerTopicProvisioningState["Canceled"] = "Canceled";
    /** Failed */
    KnownPartnerTopicProvisioningState["Failed"] = "Failed";
    /** IdleDueToMirroredChannelResourceDeletion */
    KnownPartnerTopicProvisioningState["IdleDueToMirroredChannelResourceDeletion"] = "IdleDueToMirroredChannelResourceDeletion";
})(KnownPartnerTopicProvisioningState || (KnownPartnerTopicProvisioningState = {}));
/** Known values of {@link PartnerTopicActivationState} that the service accepts. */
export var KnownPartnerTopicActivationState;
(function (KnownPartnerTopicActivationState) {
    /** NeverActivated */
    KnownPartnerTopicActivationState["NeverActivated"] = "NeverActivated";
    /** Activated */
    KnownPartnerTopicActivationState["Activated"] = "Activated";
    /** Deactivated */
    KnownPartnerTopicActivationState["Deactivated"] = "Deactivated";
})(KnownPartnerTopicActivationState || (KnownPartnerTopicActivationState = {}));
/** Known values of {@link PrivateEndpointConnectionsParentType} that the service accepts. */
export var KnownPrivateEndpointConnectionsParentType;
(function (KnownPrivateEndpointConnectionsParentType) {
    /** Topics */
    KnownPrivateEndpointConnectionsParentType["Topics"] = "topics";
    /** Domains */
    KnownPrivateEndpointConnectionsParentType["Domains"] = "domains";
    /** PartnerNamespaces */
    KnownPrivateEndpointConnectionsParentType["PartnerNamespaces"] = "partnerNamespaces";
})(KnownPrivateEndpointConnectionsParentType || (KnownPrivateEndpointConnectionsParentType = {}));
/** Known values of {@link ResourceRegionType} that the service accepts. */
export var KnownResourceRegionType;
(function (KnownResourceRegionType) {
    /** RegionalResource */
    KnownResourceRegionType["RegionalResource"] = "RegionalResource";
    /** GlobalResource */
    KnownResourceRegionType["GlobalResource"] = "GlobalResource";
})(KnownResourceRegionType || (KnownResourceRegionType = {}));
/** Known values of {@link TopicTypeProvisioningState} that the service accepts. */
export var KnownTopicTypeProvisioningState;
(function (KnownTopicTypeProvisioningState) {
    /** Creating */
    KnownTopicTypeProvisioningState["Creating"] = "Creating";
    /** Updating */
    KnownTopicTypeProvisioningState["Updating"] = "Updating";
    /** Deleting */
    KnownTopicTypeProvisioningState["Deleting"] = "Deleting";
    /** Succeeded */
    KnownTopicTypeProvisioningState["Succeeded"] = "Succeeded";
    /** Canceled */
    KnownTopicTypeProvisioningState["Canceled"] = "Canceled";
    /** Failed */
    KnownTopicTypeProvisioningState["Failed"] = "Failed";
})(KnownTopicTypeProvisioningState || (KnownTopicTypeProvisioningState = {}));
/** Known values of {@link TopicTypeSourceScope} that the service accepts. */
export var KnownTopicTypeSourceScope;
(function (KnownTopicTypeSourceScope) {
    /** Resource */
    KnownTopicTypeSourceScope["Resource"] = "Resource";
    /** ResourceGroup */
    KnownTopicTypeSourceScope["ResourceGroup"] = "ResourceGroup";
    /** AzureSubscription */
    KnownTopicTypeSourceScope["AzureSubscription"] = "AzureSubscription";
    /** ManagementGroup */
    KnownTopicTypeSourceScope["ManagementGroup"] = "ManagementGroup";
})(KnownTopicTypeSourceScope || (KnownTopicTypeSourceScope = {}));
/** Known values of {@link VerifiedPartnerProvisioningState} that the service accepts. */
export var KnownVerifiedPartnerProvisioningState;
(function (KnownVerifiedPartnerProvisioningState) {
    /** Creating */
    KnownVerifiedPartnerProvisioningState["Creating"] = "Creating";
    /** Updating */
    KnownVerifiedPartnerProvisioningState["Updating"] = "Updating";
    /** Deleting */
    KnownVerifiedPartnerProvisioningState["Deleting"] = "Deleting";
    /** Succeeded */
    KnownVerifiedPartnerProvisioningState["Succeeded"] = "Succeeded";
    /** Canceled */
    KnownVerifiedPartnerProvisioningState["Canceled"] = "Canceled";
    /** Failed */
    KnownVerifiedPartnerProvisioningState["Failed"] = "Failed";
})(KnownVerifiedPartnerProvisioningState || (KnownVerifiedPartnerProvisioningState = {}));
