// @aws-sdk/client-sms/dist-cjs/index.js
"use strict";
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __name = (target, value) => __defProp(target, "name", { value, configurable: true });
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// src/index.ts
var src_exports = {};
__export(src_exports, {
  AppLaunchConfigurationStatus: () => AppLaunchConfigurationStatus,
  AppLaunchStatus: () => AppLaunchStatus,
  AppReplicationConfigurationStatus: () => AppReplicationConfigurationStatus,
  AppReplicationStatus: () => AppReplicationStatus,
  AppStatus: () => AppStatus,
  AppValidationStrategy: () => AppValidationStrategy,
  ConnectorCapability: () => ConnectorCapability,
  ConnectorStatus: () => ConnectorStatus,
  CreateAppCommand: () => CreateAppCommand,
  CreateReplicationJobCommand: () => CreateReplicationJobCommand,
  DeleteAppCommand: () => DeleteAppCommand,
  DeleteAppLaunchConfigurationCommand: () => DeleteAppLaunchConfigurationCommand,
  DeleteAppReplicationConfigurationCommand: () => DeleteAppReplicationConfigurationCommand,
  DeleteAppValidationConfigurationCommand: () => DeleteAppValidationConfigurationCommand,
  DeleteReplicationJobCommand: () => DeleteReplicationJobCommand,
  DeleteServerCatalogCommand: () => DeleteServerCatalogCommand,
  DisassociateConnectorCommand: () => DisassociateConnectorCommand,
  DryRunOperationException: () => DryRunOperationException,
  GenerateChangeSetCommand: () => GenerateChangeSetCommand,
  GenerateTemplateCommand: () => GenerateTemplateCommand,
  GetAppCommand: () => GetAppCommand,
  GetAppLaunchConfigurationCommand: () => GetAppLaunchConfigurationCommand,
  GetAppReplicationConfigurationCommand: () => GetAppReplicationConfigurationCommand,
  GetAppValidationConfigurationCommand: () => GetAppValidationConfigurationCommand,
  GetAppValidationOutputCommand: () => GetAppValidationOutputCommand,
  GetConnectorsCommand: () => GetConnectorsCommand,
  GetReplicationJobsCommand: () => GetReplicationJobsCommand,
  GetReplicationRunsCommand: () => GetReplicationRunsCommand,
  GetServersCommand: () => GetServersCommand,
  ImportAppCatalogCommand: () => ImportAppCatalogCommand,
  ImportServerCatalogCommand: () => ImportServerCatalogCommand,
  InternalError: () => InternalError,
  InvalidParameterException: () => InvalidParameterException,
  LaunchAppCommand: () => LaunchAppCommand,
  LicenseType: () => LicenseType,
  ListAppsCommand: () => ListAppsCommand,
  MissingRequiredParameterException: () => MissingRequiredParameterException,
  NoConnectorsAvailableException: () => NoConnectorsAvailableException,
  NotifyAppValidationOutputCommand: () => NotifyAppValidationOutputCommand,
  OperationNotPermittedException: () => OperationNotPermittedException,
  OutputFormat: () => OutputFormat,
  PutAppLaunchConfigurationCommand: () => PutAppLaunchConfigurationCommand,
  PutAppReplicationConfigurationCommand: () => PutAppReplicationConfigurationCommand,
  PutAppValidationConfigurationCommand: () => PutAppValidationConfigurationCommand,
  ReplicationJobAlreadyExistsException: () => ReplicationJobAlreadyExistsException,
  ReplicationJobNotFoundException: () => ReplicationJobNotFoundException,
  ReplicationJobState: () => ReplicationJobState,
  ReplicationRunLimitExceededException: () => ReplicationRunLimitExceededException,
  ReplicationRunState: () => ReplicationRunState,
  ReplicationRunType: () => ReplicationRunType,
  SMS: () => SMS,
  SMSClient: () => SMSClient,
  SMSServiceException: () => SMSServiceException,
  ScriptType: () => ScriptType,
  ServerCannotBeReplicatedException: () => ServerCannotBeReplicatedException,
  ServerCatalogStatus: () => ServerCatalogStatus,
  ServerType: () => ServerType,
  ServerValidationStrategy: () => ServerValidationStrategy,
  StartAppReplicationCommand: () => StartAppReplicationCommand,
  StartOnDemandAppReplicationCommand: () => StartOnDemandAppReplicationCommand,
  StartOnDemandReplicationRunCommand: () => StartOnDemandReplicationRunCommand,
  StopAppReplicationCommand: () => StopAppReplicationCommand,
  TemporarilyUnavailableException: () => TemporarilyUnavailableException,
  TerminateAppCommand: () => TerminateAppCommand,
  UnauthorizedOperationException: () => UnauthorizedOperationException,
  UpdateAppCommand: () => UpdateAppCommand,
  UpdateReplicationJobCommand: () => UpdateReplicationJobCommand,
  ValidationStatus: () => ValidationStatus,
  VmManagerType: () => VmManagerType,
  __Client: () => import_smithy_client.Client,
  paginateGetConnectors: () => paginateGetConnectors,
  paginateGetReplicationJobs: () => paginateGetReplicationJobs,
  paginateGetReplicationRuns: () => paginateGetReplicationRuns,
  paginateGetServers: () => paginateGetServers
});
module.exports = __toCommonJS(src_exports);

// src/SMSClient.ts
var import_middleware_host_header = require("@aws-sdk/middleware-host-header");
var import_middleware_logger = require("@aws-sdk/middleware-logger");
var import_middleware_recursion_detection = require("@aws-sdk/middleware-recursion-detection");
var import_middleware_user_agent = require("@aws-sdk/middleware-user-agent");
var import_config_resolver = require("@smithy/config-resolver");
var import_core = require("@smithy/core");
var import_middleware_content_length = require("@smithy/middleware-content-length");
var import_middleware_endpoint = require("@smithy/middleware-endpoint");
var import_middleware_retry = require("@smithy/middleware-retry");

var import_httpAuthSchemeProvider = require("./auth/httpAuthSchemeProvider");

// src/endpoint/EndpointParameters.ts
var resolveClientEndpointParameters = /* @__PURE__ */ __name((options) => {
  return {
    ...options,
    useDualstackEndpoint: options.useDualstackEndpoint ?? false,
    useFipsEndpoint: options.useFipsEndpoint ?? false,
    defaultSigningName: "sms"
  };
}, "resolveClientEndpointParameters");
var commonParams = {
  UseFIPS: { type: "builtInParams", name: "useFipsEndpoint" },
  Endpoint: { type: "builtInParams", name: "endpoint" },
  Region: { type: "builtInParams", name: "region" },
  UseDualStack: { type: "builtInParams", name: "useDualstackEndpoint" }
};

// src/SMSClient.ts
var import_runtimeConfig = require("././runtimeConfig");

// src/runtimeExtensions.ts
var import_region_config_resolver = require("@aws-sdk/region-config-resolver");
var import_protocol_http = require("@smithy/protocol-http");
var import_smithy_client = require("@smithy/smithy-client");

// src/auth/httpAuthExtensionConfiguration.ts
var getHttpAuthExtensionConfiguration = /* @__PURE__ */ __name((runtimeConfig) => {
  const _httpAuthSchemes = runtimeConfig.httpAuthSchemes;
  let _httpAuthSchemeProvider = runtimeConfig.httpAuthSchemeProvider;
  let _credentials = runtimeConfig.credentials;
  return {
    setHttpAuthScheme(httpAuthScheme) {
      const index = _httpAuthSchemes.findIndex((scheme) => scheme.schemeId === httpAuthScheme.schemeId);
      if (index === -1) {
        _httpAuthSchemes.push(httpAuthScheme);
      } else {
        _httpAuthSchemes.splice(index, 1, httpAuthScheme);
      }
    },
    httpAuthSchemes() {
      return _httpAuthSchemes;
    },
    setHttpAuthSchemeProvider(httpAuthSchemeProvider) {
      _httpAuthSchemeProvider = httpAuthSchemeProvider;
    },
    httpAuthSchemeProvider() {
      return _httpAuthSchemeProvider;
    },
    setCredentials(credentials) {
      _credentials = credentials;
    },
    credentials() {
      return _credentials;
    }
  };
}, "getHttpAuthExtensionConfiguration");
var resolveHttpAuthRuntimeConfig = /* @__PURE__ */ __name((config) => {
  return {
    httpAuthSchemes: config.httpAuthSchemes(),
    httpAuthSchemeProvider: config.httpAuthSchemeProvider(),
    credentials: config.credentials()
  };
}, "resolveHttpAuthRuntimeConfig");

// src/runtimeExtensions.ts
var asPartial = /* @__PURE__ */ __name((t) => t, "asPartial");
var resolveRuntimeExtensions = /* @__PURE__ */ __name((runtimeConfig, extensions) => {
  const extensionConfiguration = {
    ...asPartial((0, import_region_config_resolver.getAwsRegionExtensionConfiguration)(runtimeConfig)),
    ...asPartial((0, import_smithy_client.getDefaultExtensionConfiguration)(runtimeConfig)),
    ...asPartial((0, import_protocol_http.getHttpHandlerExtensionConfiguration)(runtimeConfig)),
    ...asPartial(getHttpAuthExtensionConfiguration(runtimeConfig))
  };
  extensions.forEach((extension) => extension.configure(extensionConfiguration));
  return {
    ...runtimeConfig,
    ...(0, import_region_config_resolver.resolveAwsRegionExtensionConfiguration)(extensionConfiguration),
    ...(0, import_smithy_client.resolveDefaultRuntimeConfig)(extensionConfiguration),
    ...(0, import_protocol_http.resolveHttpHandlerRuntimeConfig)(extensionConfiguration),
    ...resolveHttpAuthRuntimeConfig(extensionConfiguration)
  };
}, "resolveRuntimeExtensions");

// src/SMSClient.ts
var _SMSClient = class _SMSClient extends import_smithy_client.Client {
  constructor(...[configuration]) {
    const _config_0 = (0, import_runtimeConfig.getRuntimeConfig)(configuration || {});
    const _config_1 = resolveClientEndpointParameters(_config_0);
    const _config_2 = (0, import_config_resolver.resolveRegionConfig)(_config_1);
    const _config_3 = (0, import_middleware_endpoint.resolveEndpointConfig)(_config_2);
    const _config_4 = (0, import_middleware_retry.resolveRetryConfig)(_config_3);
    const _config_5 = (0, import_middleware_host_header.resolveHostHeaderConfig)(_config_4);
    const _config_6 = (0, import_middleware_user_agent.resolveUserAgentConfig)(_config_5);
    const _config_7 = (0, import_httpAuthSchemeProvider.resolveHttpAuthSchemeConfig)(_config_6);
    const _config_8 = resolveRuntimeExtensions(_config_7, (configuration == null ? void 0 : configuration.extensions) || []);
    super(_config_8);
    this.config = _config_8;
    this.middlewareStack.use((0, import_middleware_retry.getRetryPlugin)(this.config));
    this.middlewareStack.use((0, import_middleware_content_length.getContentLengthPlugin)(this.config));
    this.middlewareStack.use((0, import_middleware_host_header.getHostHeaderPlugin)(this.config));
    this.middlewareStack.use((0, import_middleware_logger.getLoggerPlugin)(this.config));
    this.middlewareStack.use((0, import_middleware_recursion_detection.getRecursionDetectionPlugin)(this.config));
    this.middlewareStack.use((0, import_middleware_user_agent.getUserAgentPlugin)(this.config));
    this.middlewareStack.use(
      (0, import_core.getHttpAuthSchemeEndpointRuleSetPlugin)(this.config, {
        httpAuthSchemeParametersProvider: this.getDefaultHttpAuthSchemeParametersProvider(),
        identityProviderConfigProvider: this.getIdentityProviderConfigProvider()
      })
    );
    this.middlewareStack.use((0, import_core.getHttpSigningPlugin)(this.config));
  }
  /**
   * Destroy underlying resources, like sockets. It's usually not necessary to do this.
   * However in Node.js, it's best to explicitly shut down the client's agent when it is no longer needed.
   * Otherwise, sockets might stay open for quite a long time before the server terminates them.
   */
  destroy() {
    super.destroy();
  }
  getDefaultHttpAuthSchemeParametersProvider() {
    return import_httpAuthSchemeProvider.defaultSMSHttpAuthSchemeParametersProvider;
  }
  getIdentityProviderConfigProvider() {
    return async (config) => new import_core.DefaultIdentityProviderConfig({
      "aws.auth#sigv4": config.credentials
    });
  }
};
__name(_SMSClient, "SMSClient");
var SMSClient = _SMSClient;

// src/SMS.ts


// src/commands/CreateAppCommand.ts

var import_middleware_serde = require("@smithy/middleware-serde");

var import_types = require("@smithy/types");

// src/protocols/Aws_json1_1.ts
var import_core2 = require("@aws-sdk/core");



// src/models/SMSServiceException.ts

var _SMSServiceException = class _SMSServiceException extends import_smithy_client.ServiceException {
  /**
   * @internal
   */
  constructor(options) {
    super(options);
    Object.setPrototypeOf(this, _SMSServiceException.prototype);
  }
};
__name(_SMSServiceException, "SMSServiceException");
var SMSServiceException = _SMSServiceException;

// src/models/models_0.ts
var AppLaunchConfigurationStatus = {
  Configured: "CONFIGURED",
  NotConfigured: "NOT_CONFIGURED"
};
var AppLaunchStatus = {
  ConfigurationInProgress: "CONFIGURATION_IN_PROGRESS",
  ConfigurationInvalid: "CONFIGURATION_INVALID",
  DeltaLaunchFailed: "DELTA_LAUNCH_FAILED",
  DeltaLaunchInProgress: "DELTA_LAUNCH_IN_PROGRESS",
  LaunchFailed: "LAUNCH_FAILED",
  LaunchInProgress: "LAUNCH_IN_PROGRESS",
  LaunchPending: "LAUNCH_PENDING",
  Launched: "LAUNCHED",
  PartiallyLaunched: "PARTIALLY_LAUNCHED",
  ReadyForConfiguration: "READY_FOR_CONFIGURATION",
  ReadyForLaunch: "READY_FOR_LAUNCH",
  TerminateFailed: "TERMINATE_FAILED",
  TerminateInProgress: "TERMINATE_IN_PROGRESS",
  Terminated: "TERMINATED",
  ValidationInProgress: "VALIDATION_IN_PROGRESS"
};
var AppReplicationConfigurationStatus = {
  Configured: "CONFIGURED",
  NotConfigured: "NOT_CONFIGURED"
};
var AppReplicationStatus = {
  ConfigurationInProgress: "CONFIGURATION_IN_PROGRESS",
  ConfigurationInvalid: "CONFIGURATION_INVALID",
  DeltaReplicated: "DELTA_REPLICATED",
  DeltaReplicationFailed: "DELTA_REPLICATION_FAILED",
  DeltaReplicationInProgress: "DELTA_REPLICATION_IN_PROGRESS",
  PartiallyReplicated: "PARTIALLY_REPLICATED",
  ReadyForConfiguration: "READY_FOR_CONFIGURATION",
  ReadyForReplication: "READY_FOR_REPLICATION",
  Replicated: "REPLICATED",
  ReplicationFailed: "REPLICATION_FAILED",
  ReplicationInProgress: "REPLICATION_IN_PROGRESS",
  ReplicationPending: "REPLICATION_PENDING",
  ReplicationStopFailed: "REPLICATION_STOP_FAILED",
  ReplicationStopped: "REPLICATION_STOPPED",
  ReplicationStopping: "REPLICATION_STOPPING",
  ValidationInProgress: "VALIDATION_IN_PROGRESS"
};
var AppStatus = {
  Active: "ACTIVE",
  Creating: "CREATING",
  DELETE_FAILED: "DELETE_FAILED",
  Deleted: "DELETED",
  Deleting: "DELETING",
  Updating: "UPDATING"
};
var AppValidationStrategy = {
  SSM: "SSM"
};
var ScriptType = {
  POWERSHELL_SCRIPT: "POWERSHELL_SCRIPT",
  SHELL_SCRIPT: "SHELL_SCRIPT"
};
var ServerType = {
  VirtualMachine: "VIRTUAL_MACHINE"
};
var VmManagerType = {
  hyperVManager: "HYPERV-MANAGER",
  scvmm: "SCVMM",
  vSphere: "VSPHERE"
};
var _InternalError = class _InternalError extends SMSServiceException {
  /**
   * @internal
   */
  constructor(opts) {
    super({
      name: "InternalError",
      $fault: "server",
      ...opts
    });
    this.name = "InternalError";
    this.$fault = "server";
    Object.setPrototypeOf(this, _InternalError.prototype);
  }
};
__name(_InternalError, "InternalError");
var InternalError = _InternalError;
var _InvalidParameterException = class _InvalidParameterException extends SMSServiceException {
  /**
   * @internal
   */
  constructor(opts) {
    super({
      name: "InvalidParameterException",
      $fault: "client",
      ...opts
    });
    this.name = "InvalidParameterException";
    this.$fault = "client";
    Object.setPrototypeOf(this, _InvalidParameterException.prototype);
  }
};
__name(_InvalidParameterException, "InvalidParameterException");
var InvalidParameterException = _InvalidParameterException;
var _MissingRequiredParameterException = class _MissingRequiredParameterException extends SMSServiceException {
  /**
   * @internal
   */
  constructor(opts) {
    super({
      name: "MissingRequiredParameterException",
      $fault: "client",
      ...opts
    });
    this.name = "MissingRequiredParameterException";
    this.$fault = "client";
    Object.setPrototypeOf(this, _MissingRequiredParameterException.prototype);
  }
};
__name(_MissingRequiredParameterException, "MissingRequiredParameterException");
var MissingRequiredParameterException = _MissingRequiredParameterException;
var _OperationNotPermittedException = class _OperationNotPermittedException extends SMSServiceException {
  /**
   * @internal
   */
  constructor(opts) {
    super({
      name: "OperationNotPermittedException",
      $fault: "client",
      ...opts
    });
    this.name = "OperationNotPermittedException";
    this.$fault = "client";
    Object.setPrototypeOf(this, _OperationNotPermittedException.prototype);
  }
};
__name(_OperationNotPermittedException, "OperationNotPermittedException");
var OperationNotPermittedException = _OperationNotPermittedException;
var _UnauthorizedOperationException = class _UnauthorizedOperationException extends SMSServiceException {
  /**
   * @internal
   */
  constructor(opts) {
    super({
      name: "UnauthorizedOperationException",
      $fault: "client",
      ...opts
    });
    this.name = "UnauthorizedOperationException";
    this.$fault = "client";
    Object.setPrototypeOf(this, _UnauthorizedOperationException.prototype);
  }
};
__name(_UnauthorizedOperationException, "UnauthorizedOperationException");
var UnauthorizedOperationException = _UnauthorizedOperationException;
var LicenseType = {
  AWS: "AWS",
  BYOL: "BYOL"
};
var _NoConnectorsAvailableException = class _NoConnectorsAvailableException extends SMSServiceException {
  /**
   * @internal
   */
  constructor(opts) {
    super({
      name: "NoConnectorsAvailableException",
      $fault: "client",
      ...opts
    });
    this.name = "NoConnectorsAvailableException";
    this.$fault = "client";
    Object.setPrototypeOf(this, _NoConnectorsAvailableException.prototype);
  }
};
__name(_NoConnectorsAvailableException, "NoConnectorsAvailableException");
var NoConnectorsAvailableException = _NoConnectorsAvailableException;
var _ReplicationJobAlreadyExistsException = class _ReplicationJobAlreadyExistsException extends SMSServiceException {
  /**
   * @internal
   */
  constructor(opts) {
    super({
      name: "ReplicationJobAlreadyExistsException",
      $fault: "client",
      ...opts
    });
    this.name = "ReplicationJobAlreadyExistsException";
    this.$fault = "client";
    Object.setPrototypeOf(this, _ReplicationJobAlreadyExistsException.prototype);
  }
};
__name(_ReplicationJobAlreadyExistsException, "ReplicationJobAlreadyExistsException");
var ReplicationJobAlreadyExistsException = _ReplicationJobAlreadyExistsException;
var _ServerCannotBeReplicatedException = class _ServerCannotBeReplicatedException extends SMSServiceException {
  /**
   * @internal
   */
  constructor(opts) {
    super({
      name: "ServerCannotBeReplicatedException",
      $fault: "client",
      ...opts
    });
    this.name = "ServerCannotBeReplicatedException";
    this.$fault = "client";
    Object.setPrototypeOf(this, _ServerCannotBeReplicatedException.prototype);
  }
};
__name(_ServerCannotBeReplicatedException, "ServerCannotBeReplicatedException");
var ServerCannotBeReplicatedException = _ServerCannotBeReplicatedException;
var _TemporarilyUnavailableException = class _TemporarilyUnavailableException extends SMSServiceException {
  /**
   * @internal
   */
  constructor(opts) {
    super({
      name: "TemporarilyUnavailableException",
      $fault: "server",
      ...opts
    });
    this.name = "TemporarilyUnavailableException";
    this.$fault = "server";
    Object.setPrototypeOf(this, _TemporarilyUnavailableException.prototype);
  }
};
__name(_TemporarilyUnavailableException, "TemporarilyUnavailableException");
var TemporarilyUnavailableException = _TemporarilyUnavailableException;
var _ReplicationJobNotFoundException = class _ReplicationJobNotFoundException extends SMSServiceException {
  /**
   * @internal
   */
  constructor(opts) {
    super({
      name: "ReplicationJobNotFoundException",
      $fault: "client",
      ...opts
    });
    this.name = "ReplicationJobNotFoundException";
    this.$fault = "client";
    Object.setPrototypeOf(this, _ReplicationJobNotFoundException.prototype);
  }
};
__name(_ReplicationJobNotFoundException, "ReplicationJobNotFoundException");
var ReplicationJobNotFoundException = _ReplicationJobNotFoundException;
var OutputFormat = {
  JSON: "JSON",
  YAML: "YAML"
};
var ServerValidationStrategy = {
  USERDATA: "USERDATA"
};
var ValidationStatus = {
  Failed: "FAILED",
  InProgress: "IN_PROGRESS",
  Pending: "PENDING",
  ReadyForValidation: "READY_FOR_VALIDATION",
  Succeeded: "SUCCEEDED"
};
var ConnectorCapability = {
  hyperVManager: "HYPERV-MANAGER",
  scvmm: "SCVMM",
  smsOptimized: "SMS_OPTIMIZED",
  snapshotBatching: "SNAPSHOT_BATCHING",
  vSphere: "VSPHERE"
};
var ConnectorStatus = {
  Healthy: "HEALTHY",
  Unhealthy: "UNHEALTHY"
};
var ReplicationRunState = {
  Active: "ACTIVE",
  Completed: "COMPLETED",
  Deleted: "DELETED",
  Deleting: "DELETING",
  Failed: "FAILED",
  Missed: "MISSED",
  Pending: "PENDING"
};
var ReplicationRunType = {
  Automatic: "AUTOMATIC",
  OnDemand: "ON_DEMAND"
};
var ReplicationJobState = {
  Active: "ACTIVE",
  Completed: "COMPLETED",
  Deleted: "DELETED",
  Deleting: "DELETING",
  Failed: "FAILED",
  Failing: "FAILING",
  PausedOnFailure: "PAUSED_ON_FAILURE",
  Pending: "PENDING"
};
var ServerCatalogStatus = {
  Available: "AVAILABLE",
  Deleted: "DELETED",
  Expired: "EXPIRED",
  Importing: "IMPORTING",
  NotImported: "NOT_IMPORTED"
};
var _DryRunOperationException = class _DryRunOperationException extends SMSServiceException {
  /**
   * @internal
   */
  constructor(opts) {
    super({
      name: "DryRunOperationException",
      $fault: "client",
      ...opts
    });
    this.name = "DryRunOperationException";
    this.$fault = "client";
    Object.setPrototypeOf(this, _DryRunOperationException.prototype);
  }
};
__name(_DryRunOperationException, "DryRunOperationException");
var DryRunOperationException = _DryRunOperationException;
var _ReplicationRunLimitExceededException = class _ReplicationRunLimitExceededException extends SMSServiceException {
  /**
   * @internal
   */
  constructor(opts) {
    super({
      name: "ReplicationRunLimitExceededException",
      $fault: "client",
      ...opts
    });
    this.name = "ReplicationRunLimitExceededException";
    this.$fault = "client";
    Object.setPrototypeOf(this, _ReplicationRunLimitExceededException.prototype);
  }
};
__name(_ReplicationRunLimitExceededException, "ReplicationRunLimitExceededException");
var ReplicationRunLimitExceededException = _ReplicationRunLimitExceededException;

// src/protocols/Aws_json1_1.ts
var se_CreateAppCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("CreateApp");
  let body;
  body = JSON.stringify((0, import_smithy_client._json)(input));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_CreateAppCommand");
var se_CreateReplicationJobCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("CreateReplicationJob");
  let body;
  body = JSON.stringify(se_CreateReplicationJobRequest(input, context));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_CreateReplicationJobCommand");
var se_DeleteAppCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("DeleteApp");
  let body;
  body = JSON.stringify((0, import_smithy_client._json)(input));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_DeleteAppCommand");
var se_DeleteAppLaunchConfigurationCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("DeleteAppLaunchConfiguration");
  let body;
  body = JSON.stringify((0, import_smithy_client._json)(input));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_DeleteAppLaunchConfigurationCommand");
var se_DeleteAppReplicationConfigurationCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("DeleteAppReplicationConfiguration");
  let body;
  body = JSON.stringify((0, import_smithy_client._json)(input));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_DeleteAppReplicationConfigurationCommand");
var se_DeleteAppValidationConfigurationCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("DeleteAppValidationConfiguration");
  let body;
  body = JSON.stringify((0, import_smithy_client._json)(input));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_DeleteAppValidationConfigurationCommand");
var se_DeleteReplicationJobCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("DeleteReplicationJob");
  let body;
  body = JSON.stringify((0, import_smithy_client._json)(input));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_DeleteReplicationJobCommand");
var se_DeleteServerCatalogCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("DeleteServerCatalog");
  let body;
  body = JSON.stringify((0, import_smithy_client._json)(input));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_DeleteServerCatalogCommand");
var se_DisassociateConnectorCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("DisassociateConnector");
  let body;
  body = JSON.stringify((0, import_smithy_client._json)(input));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_DisassociateConnectorCommand");
var se_GenerateChangeSetCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("GenerateChangeSet");
  let body;
  body = JSON.stringify((0, import_smithy_client._json)(input));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_GenerateChangeSetCommand");
var se_GenerateTemplateCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("GenerateTemplate");
  let body;
  body = JSON.stringify((0, import_smithy_client._json)(input));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_GenerateTemplateCommand");
var se_GetAppCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("GetApp");
  let body;
  body = JSON.stringify((0, import_smithy_client._json)(input));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_GetAppCommand");
var se_GetAppLaunchConfigurationCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("GetAppLaunchConfiguration");
  let body;
  body = JSON.stringify((0, import_smithy_client._json)(input));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_GetAppLaunchConfigurationCommand");
var se_GetAppReplicationConfigurationCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("GetAppReplicationConfiguration");
  let body;
  body = JSON.stringify((0, import_smithy_client._json)(input));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_GetAppReplicationConfigurationCommand");
var se_GetAppValidationConfigurationCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("GetAppValidationConfiguration");
  let body;
  body = JSON.stringify((0, import_smithy_client._json)(input));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_GetAppValidationConfigurationCommand");
var se_GetAppValidationOutputCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("GetAppValidationOutput");
  let body;
  body = JSON.stringify((0, import_smithy_client._json)(input));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_GetAppValidationOutputCommand");
var se_GetConnectorsCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("GetConnectors");
  let body;
  body = JSON.stringify((0, import_smithy_client._json)(input));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_GetConnectorsCommand");
var se_GetReplicationJobsCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("GetReplicationJobs");
  let body;
  body = JSON.stringify((0, import_smithy_client._json)(input));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_GetReplicationJobsCommand");
var se_GetReplicationRunsCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("GetReplicationRuns");
  let body;
  body = JSON.stringify((0, import_smithy_client._json)(input));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_GetReplicationRunsCommand");
var se_GetServersCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("GetServers");
  let body;
  body = JSON.stringify((0, import_smithy_client._json)(input));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_GetServersCommand");
var se_ImportAppCatalogCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("ImportAppCatalog");
  let body;
  body = JSON.stringify((0, import_smithy_client._json)(input));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_ImportAppCatalogCommand");
var se_ImportServerCatalogCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("ImportServerCatalog");
  let body;
  body = JSON.stringify((0, import_smithy_client._json)(input));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_ImportServerCatalogCommand");
var se_LaunchAppCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("LaunchApp");
  let body;
  body = JSON.stringify((0, import_smithy_client._json)(input));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_LaunchAppCommand");
var se_ListAppsCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("ListApps");
  let body;
  body = JSON.stringify((0, import_smithy_client._json)(input));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_ListAppsCommand");
var se_NotifyAppValidationOutputCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("NotifyAppValidationOutput");
  let body;
  body = JSON.stringify((0, import_smithy_client._json)(input));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_NotifyAppValidationOutputCommand");
var se_PutAppLaunchConfigurationCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("PutAppLaunchConfiguration");
  let body;
  body = JSON.stringify((0, import_smithy_client._json)(input));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_PutAppLaunchConfigurationCommand");
var se_PutAppReplicationConfigurationCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("PutAppReplicationConfiguration");
  let body;
  body = JSON.stringify(se_PutAppReplicationConfigurationRequest(input, context));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_PutAppReplicationConfigurationCommand");
var se_PutAppValidationConfigurationCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("PutAppValidationConfiguration");
  let body;
  body = JSON.stringify((0, import_smithy_client._json)(input));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_PutAppValidationConfigurationCommand");
var se_StartAppReplicationCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("StartAppReplication");
  let body;
  body = JSON.stringify((0, import_smithy_client._json)(input));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_StartAppReplicationCommand");
var se_StartOnDemandAppReplicationCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("StartOnDemandAppReplication");
  let body;
  body = JSON.stringify((0, import_smithy_client._json)(input));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_StartOnDemandAppReplicationCommand");
var se_StartOnDemandReplicationRunCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("StartOnDemandReplicationRun");
  let body;
  body = JSON.stringify((0, import_smithy_client._json)(input));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_StartOnDemandReplicationRunCommand");
var se_StopAppReplicationCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("StopAppReplication");
  let body;
  body = JSON.stringify((0, import_smithy_client._json)(input));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_StopAppReplicationCommand");
var se_TerminateAppCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("TerminateApp");
  let body;
  body = JSON.stringify((0, import_smithy_client._json)(input));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_TerminateAppCommand");
var se_UpdateAppCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("UpdateApp");
  let body;
  body = JSON.stringify((0, import_smithy_client._json)(input));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_UpdateAppCommand");
var se_UpdateReplicationJobCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("UpdateReplicationJob");
  let body;
  body = JSON.stringify(se_UpdateReplicationJobRequest(input, context));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_UpdateReplicationJobCommand");
var de_CreateAppCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const data = await (0, import_core2.parseJsonBody)(output.body, context);
  let contents = {};
  contents = de_CreateAppResponse(data, context);
  const response = {
    $metadata: deserializeMetadata(output),
    ...contents
  };
  return response;
}, "de_CreateAppCommand");
var de_CreateReplicationJobCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const data = await (0, import_core2.parseJsonBody)(output.body, context);
  let contents = {};
  contents = (0, import_smithy_client._json)(data);
  const response = {
    $metadata: deserializeMetadata(output),
    ...contents
  };
  return response;
}, "de_CreateReplicationJobCommand");
var de_DeleteAppCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const data = await (0, import_core2.parseJsonBody)(output.body, context);
  let contents = {};
  contents = (0, import_smithy_client._json)(data);
  const response = {
    $metadata: deserializeMetadata(output),
    ...contents
  };
  return response;
}, "de_DeleteAppCommand");
var de_DeleteAppLaunchConfigurationCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const data = await (0, import_core2.parseJsonBody)(output.body, context);
  let contents = {};
  contents = (0, import_smithy_client._json)(data);
  const response = {
    $metadata: deserializeMetadata(output),
    ...contents
  };
  return response;
}, "de_DeleteAppLaunchConfigurationCommand");
var de_DeleteAppReplicationConfigurationCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const data = await (0, import_core2.parseJsonBody)(output.body, context);
  let contents = {};
  contents = (0, import_smithy_client._json)(data);
  const response = {
    $metadata: deserializeMetadata(output),
    ...contents
  };
  return response;
}, "de_DeleteAppReplicationConfigurationCommand");
var de_DeleteAppValidationConfigurationCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const data = await (0, import_core2.parseJsonBody)(output.body, context);
  let contents = {};
  contents = (0, import_smithy_client._json)(data);
  const response = {
    $metadata: deserializeMetadata(output),
    ...contents
  };
  return response;
}, "de_DeleteAppValidationConfigurationCommand");
var de_DeleteReplicationJobCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const data = await (0, import_core2.parseJsonBody)(output.body, context);
  let contents = {};
  contents = (0, import_smithy_client._json)(data);
  const response = {
    $metadata: deserializeMetadata(output),
    ...contents
  };
  return response;
}, "de_DeleteReplicationJobCommand");
var de_DeleteServerCatalogCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const data = await (0, import_core2.parseJsonBody)(output.body, context);
  let contents = {};
  contents = (0, import_smithy_client._json)(data);
  const response = {
    $metadata: deserializeMetadata(output),
    ...contents
  };
  return response;
}, "de_DeleteServerCatalogCommand");
var de_DisassociateConnectorCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const data = await (0, import_core2.parseJsonBody)(output.body, context);
  let contents = {};
  contents = (0, import_smithy_client._json)(data);
  const response = {
    $metadata: deserializeMetadata(output),
    ...contents
  };
  return response;
}, "de_DisassociateConnectorCommand");
var de_GenerateChangeSetCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const data = await (0, import_core2.parseJsonBody)(output.body, context);
  let contents = {};
  contents = (0, import_smithy_client._json)(data);
  const response = {
    $metadata: deserializeMetadata(output),
    ...contents
  };
  return response;
}, "de_GenerateChangeSetCommand");
var de_GenerateTemplateCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const data = await (0, import_core2.parseJsonBody)(output.body, context);
  let contents = {};
  contents = (0, import_smithy_client._json)(data);
  const response = {
    $metadata: deserializeMetadata(output),
    ...contents
  };
  return response;
}, "de_GenerateTemplateCommand");
var de_GetAppCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const data = await (0, import_core2.parseJsonBody)(output.body, context);
  let contents = {};
  contents = de_GetAppResponse(data, context);
  const response = {
    $metadata: deserializeMetadata(output),
    ...contents
  };
  return response;
}, "de_GetAppCommand");
var de_GetAppLaunchConfigurationCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const data = await (0, import_core2.parseJsonBody)(output.body, context);
  let contents = {};
  contents = (0, import_smithy_client._json)(data);
  const response = {
    $metadata: deserializeMetadata(output),
    ...contents
  };
  return response;
}, "de_GetAppLaunchConfigurationCommand");
var de_GetAppReplicationConfigurationCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const data = await (0, import_core2.parseJsonBody)(output.body, context);
  let contents = {};
  contents = de_GetAppReplicationConfigurationResponse(data, context);
  const response = {
    $metadata: deserializeMetadata(output),
    ...contents
  };
  return response;
}, "de_GetAppReplicationConfigurationCommand");
var de_GetAppValidationConfigurationCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const data = await (0, import_core2.parseJsonBody)(output.body, context);
  let contents = {};
  contents = (0, import_smithy_client._json)(data);
  const response = {
    $metadata: deserializeMetadata(output),
    ...contents
  };
  return response;
}, "de_GetAppValidationConfigurationCommand");
var de_GetAppValidationOutputCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const data = await (0, import_core2.parseJsonBody)(output.body, context);
  let contents = {};
  contents = de_GetAppValidationOutputResponse(data, context);
  const response = {
    $metadata: deserializeMetadata(output),
    ...contents
  };
  return response;
}, "de_GetAppValidationOutputCommand");
var de_GetConnectorsCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const data = await (0, import_core2.parseJsonBody)(output.body, context);
  let contents = {};
  contents = de_GetConnectorsResponse(data, context);
  const response = {
    $metadata: deserializeMetadata(output),
    ...contents
  };
  return response;
}, "de_GetConnectorsCommand");
var de_GetReplicationJobsCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const data = await (0, import_core2.parseJsonBody)(output.body, context);
  let contents = {};
  contents = de_GetReplicationJobsResponse(data, context);
  const response = {
    $metadata: deserializeMetadata(output),
    ...contents
  };
  return response;
}, "de_GetReplicationJobsCommand");
var de_GetReplicationRunsCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const data = await (0, import_core2.parseJsonBody)(output.body, context);
  let contents = {};
  contents = de_GetReplicationRunsResponse(data, context);
  const response = {
    $metadata: deserializeMetadata(output),
    ...contents
  };
  return response;
}, "de_GetReplicationRunsCommand");
var de_GetServersCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const data = await (0, import_core2.parseJsonBody)(output.body, context);
  let contents = {};
  contents = de_GetServersResponse(data, context);
  const response = {
    $metadata: deserializeMetadata(output),
    ...contents
  };
  return response;
}, "de_GetServersCommand");
var de_ImportAppCatalogCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const data = await (0, import_core2.parseJsonBody)(output.body, context);
  let contents = {};
  contents = (0, import_smithy_client._json)(data);
  const response = {
    $metadata: deserializeMetadata(output),
    ...contents
  };
  return response;
}, "de_ImportAppCatalogCommand");
var de_ImportServerCatalogCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const data = await (0, import_core2.parseJsonBody)(output.body, context);
  let contents = {};
  contents = (0, import_smithy_client._json)(data);
  const response = {
    $metadata: deserializeMetadata(output),
    ...contents
  };
  return response;
}, "de_ImportServerCatalogCommand");
var de_LaunchAppCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const data = await (0, import_core2.parseJsonBody)(output.body, context);
  let contents = {};
  contents = (0, import_smithy_client._json)(data);
  const response = {
    $metadata: deserializeMetadata(output),
    ...contents
  };
  return response;
}, "de_LaunchAppCommand");
var de_ListAppsCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const data = await (0, import_core2.parseJsonBody)(output.body, context);
  let contents = {};
  contents = de_ListAppsResponse(data, context);
  const response = {
    $metadata: deserializeMetadata(output),
    ...contents
  };
  return response;
}, "de_ListAppsCommand");
var de_NotifyAppValidationOutputCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const data = await (0, import_core2.parseJsonBody)(output.body, context);
  let contents = {};
  contents = (0, import_smithy_client._json)(data);
  const response = {
    $metadata: deserializeMetadata(output),
    ...contents
  };
  return response;
}, "de_NotifyAppValidationOutputCommand");
var de_PutAppLaunchConfigurationCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const data = await (0, import_core2.parseJsonBody)(output.body, context);
  let contents = {};
  contents = (0, import_smithy_client._json)(data);
  const response = {
    $metadata: deserializeMetadata(output),
    ...contents
  };
  return response;
}, "de_PutAppLaunchConfigurationCommand");
var de_PutAppReplicationConfigurationCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const data = await (0, import_core2.parseJsonBody)(output.body, context);
  let contents = {};
  contents = (0, import_smithy_client._json)(data);
  const response = {
    $metadata: deserializeMetadata(output),
    ...contents
  };
  return response;
}, "de_PutAppReplicationConfigurationCommand");
var de_PutAppValidationConfigurationCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const data = await (0, import_core2.parseJsonBody)(output.body, context);
  let contents = {};
  contents = (0, import_smithy_client._json)(data);
  const response = {
    $metadata: deserializeMetadata(output),
    ...contents
  };
  return response;
}, "de_PutAppValidationConfigurationCommand");
var de_StartAppReplicationCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const data = await (0, import_core2.parseJsonBody)(output.body, context);
  let contents = {};
  contents = (0, import_smithy_client._json)(data);
  const response = {
    $metadata: deserializeMetadata(output),
    ...contents
  };
  return response;
}, "de_StartAppReplicationCommand");
var de_StartOnDemandAppReplicationCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const data = await (0, import_core2.parseJsonBody)(output.body, context);
  let contents = {};
  contents = (0, import_smithy_client._json)(data);
  const response = {
    $metadata: deserializeMetadata(output),
    ...contents
  };
  return response;
}, "de_StartOnDemandAppReplicationCommand");
var de_StartOnDemandReplicationRunCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const data = await (0, import_core2.parseJsonBody)(output.body, context);
  let contents = {};
  contents = (0, import_smithy_client._json)(data);
  const response = {
    $metadata: deserializeMetadata(output),
    ...contents
  };
  return response;
}, "de_StartOnDemandReplicationRunCommand");
var de_StopAppReplicationCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const data = await (0, import_core2.parseJsonBody)(output.body, context);
  let contents = {};
  contents = (0, import_smithy_client._json)(data);
  const response = {
    $metadata: deserializeMetadata(output),
    ...contents
  };
  return response;
}, "de_StopAppReplicationCommand");
var de_TerminateAppCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const data = await (0, import_core2.parseJsonBody)(output.body, context);
  let contents = {};
  contents = (0, import_smithy_client._json)(data);
  const response = {
    $metadata: deserializeMetadata(output),
    ...contents
  };
  return response;
}, "de_TerminateAppCommand");
var de_UpdateAppCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const data = await (0, import_core2.parseJsonBody)(output.body, context);
  let contents = {};
  contents = de_UpdateAppResponse(data, context);
  const response = {
    $metadata: deserializeMetadata(output),
    ...contents
  };
  return response;
}, "de_UpdateAppCommand");
var de_UpdateReplicationJobCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const data = await (0, import_core2.parseJsonBody)(output.body, context);
  let contents = {};
  contents = (0, import_smithy_client._json)(data);
  const response = {
    $metadata: deserializeMetadata(output),
    ...contents
  };
  return response;
}, "de_UpdateReplicationJobCommand");
var de_CommandError = /* @__PURE__ */ __name(async (output, context) => {
  const parsedOutput = {
    ...output,
    body: await (0, import_core2.parseJsonErrorBody)(output.body, context)
  };
  const errorCode = (0, import_core2.loadRestJsonErrorCode)(output, parsedOutput.body);
  switch (errorCode) {
    case "InternalError":
    case "com.amazonaws.sms#InternalError":
      throw await de_InternalErrorRes(parsedOutput, context);
    case "InvalidParameterException":
    case "com.amazonaws.sms#InvalidParameterException":
      throw await de_InvalidParameterExceptionRes(parsedOutput, context);
    case "MissingRequiredParameterException":
    case "com.amazonaws.sms#MissingRequiredParameterException":
      throw await de_MissingRequiredParameterExceptionRes(parsedOutput, context);
    case "OperationNotPermittedException":
    case "com.amazonaws.sms#OperationNotPermittedException":
      throw await de_OperationNotPermittedExceptionRes(parsedOutput, context);
    case "UnauthorizedOperationException":
    case "com.amazonaws.sms#UnauthorizedOperationException":
      throw await de_UnauthorizedOperationExceptionRes(parsedOutput, context);
    case "NoConnectorsAvailableException":
    case "com.amazonaws.sms#NoConnectorsAvailableException":
      throw await de_NoConnectorsAvailableExceptionRes(parsedOutput, context);
    case "ReplicationJobAlreadyExistsException":
    case "com.amazonaws.sms#ReplicationJobAlreadyExistsException":
      throw await de_ReplicationJobAlreadyExistsExceptionRes(parsedOutput, context);
    case "ServerCannotBeReplicatedException":
    case "com.amazonaws.sms#ServerCannotBeReplicatedException":
      throw await de_ServerCannotBeReplicatedExceptionRes(parsedOutput, context);
    case "TemporarilyUnavailableException":
    case "com.amazonaws.sms#TemporarilyUnavailableException":
      throw await de_TemporarilyUnavailableExceptionRes(parsedOutput, context);
    case "ReplicationJobNotFoundException":
    case "com.amazonaws.sms#ReplicationJobNotFoundException":
      throw await de_ReplicationJobNotFoundExceptionRes(parsedOutput, context);
    case "DryRunOperationException":
    case "com.amazonaws.sms#DryRunOperationException":
      throw await de_DryRunOperationExceptionRes(parsedOutput, context);
    case "ReplicationRunLimitExceededException":
    case "com.amazonaws.sms#ReplicationRunLimitExceededException":
      throw await de_ReplicationRunLimitExceededExceptionRes(parsedOutput, context);
    default:
      const parsedBody = parsedOutput.body;
      return throwDefaultError({
        output,
        parsedBody,
        errorCode
      });
  }
}, "de_CommandError");
var de_DryRunOperationExceptionRes = /* @__PURE__ */ __name(async (parsedOutput, context) => {
  const body = parsedOutput.body;
  const deserialized = (0, import_smithy_client._json)(body);
  const exception = new DryRunOperationException({
    $metadata: deserializeMetadata(parsedOutput),
    ...deserialized
  });
  return (0, import_smithy_client.decorateServiceException)(exception, body);
}, "de_DryRunOperationExceptionRes");
var de_InternalErrorRes = /* @__PURE__ */ __name(async (parsedOutput, context) => {
  const body = parsedOutput.body;
  const deserialized = (0, import_smithy_client._json)(body);
  const exception = new InternalError({
    $metadata: deserializeMetadata(parsedOutput),
    ...deserialized
  });
  return (0, import_smithy_client.decorateServiceException)(exception, body);
}, "de_InternalErrorRes");
var de_InvalidParameterExceptionRes = /* @__PURE__ */ __name(async (parsedOutput, context) => {
  const body = parsedOutput.body;
  const deserialized = (0, import_smithy_client._json)(body);
  const exception = new InvalidParameterException({
    $metadata: deserializeMetadata(parsedOutput),
    ...deserialized
  });
  return (0, import_smithy_client.decorateServiceException)(exception, body);
}, "de_InvalidParameterExceptionRes");
var de_MissingRequiredParameterExceptionRes = /* @__PURE__ */ __name(async (parsedOutput, context) => {
  const body = parsedOutput.body;
  const deserialized = (0, import_smithy_client._json)(body);
  const exception = new MissingRequiredParameterException({
    $metadata: deserializeMetadata(parsedOutput),
    ...deserialized
  });
  return (0, import_smithy_client.decorateServiceException)(exception, body);
}, "de_MissingRequiredParameterExceptionRes");
var de_NoConnectorsAvailableExceptionRes = /* @__PURE__ */ __name(async (parsedOutput, context) => {
  const body = parsedOutput.body;
  const deserialized = (0, import_smithy_client._json)(body);
  const exception = new NoConnectorsAvailableException({
    $metadata: deserializeMetadata(parsedOutput),
    ...deserialized
  });
  return (0, import_smithy_client.decorateServiceException)(exception, body);
}, "de_NoConnectorsAvailableExceptionRes");
var de_OperationNotPermittedExceptionRes = /* @__PURE__ */ __name(async (parsedOutput, context) => {
  const body = parsedOutput.body;
  const deserialized = (0, import_smithy_client._json)(body);
  const exception = new OperationNotPermittedException({
    $metadata: deserializeMetadata(parsedOutput),
    ...deserialized
  });
  return (0, import_smithy_client.decorateServiceException)(exception, body);
}, "de_OperationNotPermittedExceptionRes");
var de_ReplicationJobAlreadyExistsExceptionRes = /* @__PURE__ */ __name(async (parsedOutput, context) => {
  const body = parsedOutput.body;
  const deserialized = (0, import_smithy_client._json)(body);
  const exception = new ReplicationJobAlreadyExistsException({
    $metadata: deserializeMetadata(parsedOutput),
    ...deserialized
  });
  return (0, import_smithy_client.decorateServiceException)(exception, body);
}, "de_ReplicationJobAlreadyExistsExceptionRes");
var de_ReplicationJobNotFoundExceptionRes = /* @__PURE__ */ __name(async (parsedOutput, context) => {
  const body = parsedOutput.body;
  const deserialized = (0, import_smithy_client._json)(body);
  const exception = new ReplicationJobNotFoundException({
    $metadata: deserializeMetadata(parsedOutput),
    ...deserialized
  });
  return (0, import_smithy_client.decorateServiceException)(exception, body);
}, "de_ReplicationJobNotFoundExceptionRes");
var de_ReplicationRunLimitExceededExceptionRes = /* @__PURE__ */ __name(async (parsedOutput, context) => {
  const body = parsedOutput.body;
  const deserialized = (0, import_smithy_client._json)(body);
  const exception = new ReplicationRunLimitExceededException({
    $metadata: deserializeMetadata(parsedOutput),
    ...deserialized
  });
  return (0, import_smithy_client.decorateServiceException)(exception, body);
}, "de_ReplicationRunLimitExceededExceptionRes");
var de_ServerCannotBeReplicatedExceptionRes = /* @__PURE__ */ __name(async (parsedOutput, context) => {
  const body = parsedOutput.body;
  const deserialized = (0, import_smithy_client._json)(body);
  const exception = new ServerCannotBeReplicatedException({
    $metadata: deserializeMetadata(parsedOutput),
    ...deserialized
  });
  return (0, import_smithy_client.decorateServiceException)(exception, body);
}, "de_ServerCannotBeReplicatedExceptionRes");
var de_TemporarilyUnavailableExceptionRes = /* @__PURE__ */ __name(async (parsedOutput, context) => {
  const body = parsedOutput.body;
  const deserialized = (0, import_smithy_client._json)(body);
  const exception = new TemporarilyUnavailableException({
    $metadata: deserializeMetadata(parsedOutput),
    ...deserialized
  });
  return (0, import_smithy_client.decorateServiceException)(exception, body);
}, "de_TemporarilyUnavailableExceptionRes");
var de_UnauthorizedOperationExceptionRes = /* @__PURE__ */ __name(async (parsedOutput, context) => {
  const body = parsedOutput.body;
  const deserialized = (0, import_smithy_client._json)(body);
  const exception = new UnauthorizedOperationException({
    $metadata: deserializeMetadata(parsedOutput),
    ...deserialized
  });
  return (0, import_smithy_client.decorateServiceException)(exception, body);
}, "de_UnauthorizedOperationExceptionRes");
var se_CreateReplicationJobRequest = /* @__PURE__ */ __name((input, context) => {
  return (0, import_smithy_client.take)(input, {
    description: [],
    encrypted: [],
    frequency: [],
    kmsKeyId: [],
    licenseType: [],
    numberOfRecentAmisToKeep: [],
    roleName: [],
    runOnce: [],
    seedReplicationTime: (_) => Math.round(_.getTime() / 1e3),
    serverId: []
  });
}, "se_CreateReplicationJobRequest");
var se_PutAppReplicationConfigurationRequest = /* @__PURE__ */ __name((input, context) => {
  return (0, import_smithy_client.take)(input, {
    appId: [],
    serverGroupReplicationConfigurations: (_) => se_ServerGroupReplicationConfigurations(_, context)
  });
}, "se_PutAppReplicationConfigurationRequest");
var se_ServerGroupReplicationConfiguration = /* @__PURE__ */ __name((input, context) => {
  return (0, import_smithy_client.take)(input, {
    serverGroupId: [],
    serverReplicationConfigurations: (_) => se_ServerReplicationConfigurations(_, context)
  });
}, "se_ServerGroupReplicationConfiguration");
var se_ServerGroupReplicationConfigurations = /* @__PURE__ */ __name((input, context) => {
  return input.filter((e) => e != null).map((entry) => {
    return se_ServerGroupReplicationConfiguration(entry, context);
  });
}, "se_ServerGroupReplicationConfigurations");
var se_ServerReplicationConfiguration = /* @__PURE__ */ __name((input, context) => {
  return (0, import_smithy_client.take)(input, {
    server: import_smithy_client._json,
    serverReplicationParameters: (_) => se_ServerReplicationParameters(_, context)
  });
}, "se_ServerReplicationConfiguration");
var se_ServerReplicationConfigurations = /* @__PURE__ */ __name((input, context) => {
  return input.filter((e) => e != null).map((entry) => {
    return se_ServerReplicationConfiguration(entry, context);
  });
}, "se_ServerReplicationConfigurations");
var se_ServerReplicationParameters = /* @__PURE__ */ __name((input, context) => {
  return (0, import_smithy_client.take)(input, {
    encrypted: [],
    frequency: [],
    kmsKeyId: [],
    licenseType: [],
    numberOfRecentAmisToKeep: [],
    runOnce: [],
    seedTime: (_) => Math.round(_.getTime() / 1e3)
  });
}, "se_ServerReplicationParameters");
var se_UpdateReplicationJobRequest = /* @__PURE__ */ __name((input, context) => {
  return (0, import_smithy_client.take)(input, {
    description: [],
    encrypted: [],
    frequency: [],
    kmsKeyId: [],
    licenseType: [],
    nextReplicationRunStartTime: (_) => Math.round(_.getTime() / 1e3),
    numberOfRecentAmisToKeep: [],
    replicationJobId: [],
    roleName: []
  });
}, "se_UpdateReplicationJobRequest");
var de_Apps = /* @__PURE__ */ __name((output, context) => {
  const retVal = (output || []).filter((e) => e != null).map((entry) => {
    return de_AppSummary(entry, context);
  });
  return retVal;
}, "de_Apps");
var de_AppSummary = /* @__PURE__ */ __name((output, context) => {
  return (0, import_smithy_client.take)(output, {
    appId: import_smithy_client.expectString,
    creationTime: (_) => (0, import_smithy_client.expectNonNull)((0, import_smithy_client.parseEpochTimestamp)((0, import_smithy_client.expectNumber)(_))),
    description: import_smithy_client.expectString,
    importedAppId: import_smithy_client.expectString,
    lastModified: (_) => (0, import_smithy_client.expectNonNull)((0, import_smithy_client.parseEpochTimestamp)((0, import_smithy_client.expectNumber)(_))),
    latestReplicationTime: (_) => (0, import_smithy_client.expectNonNull)((0, import_smithy_client.parseEpochTimestamp)((0, import_smithy_client.expectNumber)(_))),
    launchConfigurationStatus: import_smithy_client.expectString,
    launchDetails: (_) => de_LaunchDetails(_, context),
    launchStatus: import_smithy_client.expectString,
    launchStatusMessage: import_smithy_client.expectString,
    name: import_smithy_client.expectString,
    replicationConfigurationStatus: import_smithy_client.expectString,
    replicationStatus: import_smithy_client.expectString,
    replicationStatusMessage: import_smithy_client.expectString,
    roleName: import_smithy_client.expectString,
    status: import_smithy_client.expectString,
    statusMessage: import_smithy_client.expectString,
    totalServerGroups: import_smithy_client.expectInt32,
    totalServers: import_smithy_client.expectInt32
  });
}, "de_AppSummary");
var de_Connector = /* @__PURE__ */ __name((output, context) => {
  return (0, import_smithy_client.take)(output, {
    associatedOn: (_) => (0, import_smithy_client.expectNonNull)((0, import_smithy_client.parseEpochTimestamp)((0, import_smithy_client.expectNumber)(_))),
    capabilityList: import_smithy_client._json,
    connectorId: import_smithy_client.expectString,
    ipAddress: import_smithy_client.expectString,
    macAddress: import_smithy_client.expectString,
    status: import_smithy_client.expectString,
    version: import_smithy_client.expectString,
    vmManagerId: import_smithy_client.expectString,
    vmManagerName: import_smithy_client.expectString,
    vmManagerType: import_smithy_client.expectString
  });
}, "de_Connector");
var de_ConnectorList = /* @__PURE__ */ __name((output, context) => {
  const retVal = (output || []).filter((e) => e != null).map((entry) => {
    return de_Connector(entry, context);
  });
  return retVal;
}, "de_ConnectorList");
var de_CreateAppResponse = /* @__PURE__ */ __name((output, context) => {
  return (0, import_smithy_client.take)(output, {
    appSummary: (_) => de_AppSummary(_, context),
    serverGroups: import_smithy_client._json,
    tags: import_smithy_client._json
  });
}, "de_CreateAppResponse");
var de_GetAppReplicationConfigurationResponse = /* @__PURE__ */ __name((output, context) => {
  return (0, import_smithy_client.take)(output, {
    serverGroupReplicationConfigurations: (_) => de_ServerGroupReplicationConfigurations(_, context)
  });
}, "de_GetAppReplicationConfigurationResponse");
var de_GetAppResponse = /* @__PURE__ */ __name((output, context) => {
  return (0, import_smithy_client.take)(output, {
    appSummary: (_) => de_AppSummary(_, context),
    serverGroups: import_smithy_client._json,
    tags: import_smithy_client._json
  });
}, "de_GetAppResponse");
var de_GetAppValidationOutputResponse = /* @__PURE__ */ __name((output, context) => {
  return (0, import_smithy_client.take)(output, {
    validationOutputList: (_) => de_ValidationOutputList(_, context)
  });
}, "de_GetAppValidationOutputResponse");
var de_GetConnectorsResponse = /* @__PURE__ */ __name((output, context) => {
  return (0, import_smithy_client.take)(output, {
    connectorList: (_) => de_ConnectorList(_, context),
    nextToken: import_smithy_client.expectString
  });
}, "de_GetConnectorsResponse");
var de_GetReplicationJobsResponse = /* @__PURE__ */ __name((output, context) => {
  return (0, import_smithy_client.take)(output, {
    nextToken: import_smithy_client.expectString,
    replicationJobList: (_) => de_ReplicationJobList(_, context)
  });
}, "de_GetReplicationJobsResponse");
var de_GetReplicationRunsResponse = /* @__PURE__ */ __name((output, context) => {
  return (0, import_smithy_client.take)(output, {
    nextToken: import_smithy_client.expectString,
    replicationJob: (_) => de_ReplicationJob(_, context),
    replicationRunList: (_) => de_ReplicationRunList(_, context)
  });
}, "de_GetReplicationRunsResponse");
var de_GetServersResponse = /* @__PURE__ */ __name((output, context) => {
  return (0, import_smithy_client.take)(output, {
    lastModifiedOn: (_) => (0, import_smithy_client.expectNonNull)((0, import_smithy_client.parseEpochTimestamp)((0, import_smithy_client.expectNumber)(_))),
    nextToken: import_smithy_client.expectString,
    serverCatalogStatus: import_smithy_client.expectString,
    serverList: import_smithy_client._json
  });
}, "de_GetServersResponse");
var de_LaunchDetails = /* @__PURE__ */ __name((output, context) => {
  return (0, import_smithy_client.take)(output, {
    latestLaunchTime: (_) => (0, import_smithy_client.expectNonNull)((0, import_smithy_client.parseEpochTimestamp)((0, import_smithy_client.expectNumber)(_))),
    stackId: import_smithy_client.expectString,
    stackName: import_smithy_client.expectString
  });
}, "de_LaunchDetails");
var de_ListAppsResponse = /* @__PURE__ */ __name((output, context) => {
  return (0, import_smithy_client.take)(output, {
    apps: (_) => de_Apps(_, context),
    nextToken: import_smithy_client.expectString
  });
}, "de_ListAppsResponse");
var de_ReplicationJob = /* @__PURE__ */ __name((output, context) => {
  return (0, import_smithy_client.take)(output, {
    description: import_smithy_client.expectString,
    encrypted: import_smithy_client.expectBoolean,
    frequency: import_smithy_client.expectInt32,
    kmsKeyId: import_smithy_client.expectString,
    latestAmiId: import_smithy_client.expectString,
    licenseType: import_smithy_client.expectString,
    nextReplicationRunStartTime: (_) => (0, import_smithy_client.expectNonNull)((0, import_smithy_client.parseEpochTimestamp)((0, import_smithy_client.expectNumber)(_))),
    numberOfRecentAmisToKeep: import_smithy_client.expectInt32,
    replicationJobId: import_smithy_client.expectString,
    replicationRunList: (_) => de_ReplicationRunList(_, context),
    roleName: import_smithy_client.expectString,
    runOnce: import_smithy_client.expectBoolean,
    seedReplicationTime: (_) => (0, import_smithy_client.expectNonNull)((0, import_smithy_client.parseEpochTimestamp)((0, import_smithy_client.expectNumber)(_))),
    serverId: import_smithy_client.expectString,
    serverType: import_smithy_client.expectString,
    state: import_smithy_client.expectString,
    statusMessage: import_smithy_client.expectString,
    vmServer: import_smithy_client._json
  });
}, "de_ReplicationJob");
var de_ReplicationJobList = /* @__PURE__ */ __name((output, context) => {
  const retVal = (output || []).filter((e) => e != null).map((entry) => {
    return de_ReplicationJob(entry, context);
  });
  return retVal;
}, "de_ReplicationJobList");
var de_ReplicationRun = /* @__PURE__ */ __name((output, context) => {
  return (0, import_smithy_client.take)(output, {
    amiId: import_smithy_client.expectString,
    completedTime: (_) => (0, import_smithy_client.expectNonNull)((0, import_smithy_client.parseEpochTimestamp)((0, import_smithy_client.expectNumber)(_))),
    description: import_smithy_client.expectString,
    encrypted: import_smithy_client.expectBoolean,
    kmsKeyId: import_smithy_client.expectString,
    replicationRunId: import_smithy_client.expectString,
    scheduledStartTime: (_) => (0, import_smithy_client.expectNonNull)((0, import_smithy_client.parseEpochTimestamp)((0, import_smithy_client.expectNumber)(_))),
    stageDetails: import_smithy_client._json,
    state: import_smithy_client.expectString,
    statusMessage: import_smithy_client.expectString,
    type: import_smithy_client.expectString
  });
}, "de_ReplicationRun");
var de_ReplicationRunList = /* @__PURE__ */ __name((output, context) => {
  const retVal = (output || []).filter((e) => e != null).map((entry) => {
    return de_ReplicationRun(entry, context);
  });
  return retVal;
}, "de_ReplicationRunList");
var de_ServerGroupReplicationConfiguration = /* @__PURE__ */ __name((output, context) => {
  return (0, import_smithy_client.take)(output, {
    serverGroupId: import_smithy_client.expectString,
    serverReplicationConfigurations: (_) => de_ServerReplicationConfigurations(_, context)
  });
}, "de_ServerGroupReplicationConfiguration");
var de_ServerGroupReplicationConfigurations = /* @__PURE__ */ __name((output, context) => {
  const retVal = (output || []).filter((e) => e != null).map((entry) => {
    return de_ServerGroupReplicationConfiguration(entry, context);
  });
  return retVal;
}, "de_ServerGroupReplicationConfigurations");
var de_ServerReplicationConfiguration = /* @__PURE__ */ __name((output, context) => {
  return (0, import_smithy_client.take)(output, {
    server: import_smithy_client._json,
    serverReplicationParameters: (_) => de_ServerReplicationParameters(_, context)
  });
}, "de_ServerReplicationConfiguration");
var de_ServerReplicationConfigurations = /* @__PURE__ */ __name((output, context) => {
  const retVal = (output || []).filter((e) => e != null).map((entry) => {
    return de_ServerReplicationConfiguration(entry, context);
  });
  return retVal;
}, "de_ServerReplicationConfigurations");
var de_ServerReplicationParameters = /* @__PURE__ */ __name((output, context) => {
  return (0, import_smithy_client.take)(output, {
    encrypted: import_smithy_client.expectBoolean,
    frequency: import_smithy_client.expectInt32,
    kmsKeyId: import_smithy_client.expectString,
    licenseType: import_smithy_client.expectString,
    numberOfRecentAmisToKeep: import_smithy_client.expectInt32,
    runOnce: import_smithy_client.expectBoolean,
    seedTime: (_) => (0, import_smithy_client.expectNonNull)((0, import_smithy_client.parseEpochTimestamp)((0, import_smithy_client.expectNumber)(_)))
  });
}, "de_ServerReplicationParameters");
var de_UpdateAppResponse = /* @__PURE__ */ __name((output, context) => {
  return (0, import_smithy_client.take)(output, {
    appSummary: (_) => de_AppSummary(_, context),
    serverGroups: import_smithy_client._json,
    tags: import_smithy_client._json
  });
}, "de_UpdateAppResponse");
var de_ValidationOutput = /* @__PURE__ */ __name((output, context) => {
  return (0, import_smithy_client.take)(output, {
    appValidationOutput: import_smithy_client._json,
    latestValidationTime: (_) => (0, import_smithy_client.expectNonNull)((0, import_smithy_client.parseEpochTimestamp)((0, import_smithy_client.expectNumber)(_))),
    name: import_smithy_client.expectString,
    serverValidationOutput: import_smithy_client._json,
    status: import_smithy_client.expectString,
    statusMessage: import_smithy_client.expectString,
    validationId: import_smithy_client.expectString
  });
}, "de_ValidationOutput");
var de_ValidationOutputList = /* @__PURE__ */ __name((output, context) => {
  const retVal = (output || []).filter((e) => e != null).map((entry) => {
    return de_ValidationOutput(entry, context);
  });
  return retVal;
}, "de_ValidationOutputList");
var deserializeMetadata = /* @__PURE__ */ __name((output) => ({
  httpStatusCode: output.statusCode,
  requestId: output.headers["x-amzn-requestid"] ?? output.headers["x-amzn-request-id"] ?? output.headers["x-amz-request-id"],
  extendedRequestId: output.headers["x-amz-id-2"],
  cfId: output.headers["x-amz-cf-id"]
}), "deserializeMetadata");
var throwDefaultError = (0, import_smithy_client.withBaseException)(SMSServiceException);
var buildHttpRpcRequest = /* @__PURE__ */ __name(async (context, headers, path, resolvedHostname, body) => {
  const { hostname, protocol = "https", port, path: basePath } = await context.endpoint();
  const contents = {
    protocol,
    hostname,
    port,
    method: "POST",
    path: basePath.endsWith("/") ? basePath.slice(0, -1) + path : basePath + path,
    headers
  };
  if (resolvedHostname !== void 0) {
    contents.hostname = resolvedHostname;
  }
  if (body !== void 0) {
    contents.body = body;
  }
  return new import_protocol_http.HttpRequest(contents);
}, "buildHttpRpcRequest");
function sharedHeaders(operation) {
  return {
    "content-type": "application/x-amz-json-1.1",
    "x-amz-target": `AWSServerMigrationService_V2016_10_24.${operation}`
  };
}
__name(sharedHeaders, "sharedHeaders");

// src/commands/CreateAppCommand.ts
var _CreateAppCommand = class _CreateAppCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AWSServerMigrationService_V2016_10_24", "CreateApp", {}).n("SMSClient", "CreateAppCommand").f(void 0, void 0).ser(se_CreateAppCommand).de(de_CreateAppCommand).build() {
};
__name(_CreateAppCommand, "CreateAppCommand");
var CreateAppCommand = _CreateAppCommand;

// src/commands/CreateReplicationJobCommand.ts




var _CreateReplicationJobCommand = class _CreateReplicationJobCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AWSServerMigrationService_V2016_10_24", "CreateReplicationJob", {}).n("SMSClient", "CreateReplicationJobCommand").f(void 0, void 0).ser(se_CreateReplicationJobCommand).de(de_CreateReplicationJobCommand).build() {
};
__name(_CreateReplicationJobCommand, "CreateReplicationJobCommand");
var CreateReplicationJobCommand = _CreateReplicationJobCommand;

// src/commands/DeleteAppCommand.ts




var _DeleteAppCommand = class _DeleteAppCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AWSServerMigrationService_V2016_10_24", "DeleteApp", {}).n("SMSClient", "DeleteAppCommand").f(void 0, void 0).ser(se_DeleteAppCommand).de(de_DeleteAppCommand).build() {
};
__name(_DeleteAppCommand, "DeleteAppCommand");
var DeleteAppCommand = _DeleteAppCommand;

// src/commands/DeleteAppLaunchConfigurationCommand.ts




var _DeleteAppLaunchConfigurationCommand = class _DeleteAppLaunchConfigurationCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AWSServerMigrationService_V2016_10_24", "DeleteAppLaunchConfiguration", {}).n("SMSClient", "DeleteAppLaunchConfigurationCommand").f(void 0, void 0).ser(se_DeleteAppLaunchConfigurationCommand).de(de_DeleteAppLaunchConfigurationCommand).build() {
};
__name(_DeleteAppLaunchConfigurationCommand, "DeleteAppLaunchConfigurationCommand");
var DeleteAppLaunchConfigurationCommand = _DeleteAppLaunchConfigurationCommand;

// src/commands/DeleteAppReplicationConfigurationCommand.ts




var _DeleteAppReplicationConfigurationCommand = class _DeleteAppReplicationConfigurationCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AWSServerMigrationService_V2016_10_24", "DeleteAppReplicationConfiguration", {}).n("SMSClient", "DeleteAppReplicationConfigurationCommand").f(void 0, void 0).ser(se_DeleteAppReplicationConfigurationCommand).de(de_DeleteAppReplicationConfigurationCommand).build() {
};
__name(_DeleteAppReplicationConfigurationCommand, "DeleteAppReplicationConfigurationCommand");
var DeleteAppReplicationConfigurationCommand = _DeleteAppReplicationConfigurationCommand;

// src/commands/DeleteAppValidationConfigurationCommand.ts




var _DeleteAppValidationConfigurationCommand = class _DeleteAppValidationConfigurationCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AWSServerMigrationService_V2016_10_24", "DeleteAppValidationConfiguration", {}).n("SMSClient", "DeleteAppValidationConfigurationCommand").f(void 0, void 0).ser(se_DeleteAppValidationConfigurationCommand).de(de_DeleteAppValidationConfigurationCommand).build() {
};
__name(_DeleteAppValidationConfigurationCommand, "DeleteAppValidationConfigurationCommand");
var DeleteAppValidationConfigurationCommand = _DeleteAppValidationConfigurationCommand;

// src/commands/DeleteReplicationJobCommand.ts




var _DeleteReplicationJobCommand = class _DeleteReplicationJobCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AWSServerMigrationService_V2016_10_24", "DeleteReplicationJob", {}).n("SMSClient", "DeleteReplicationJobCommand").f(void 0, void 0).ser(se_DeleteReplicationJobCommand).de(de_DeleteReplicationJobCommand).build() {
};
__name(_DeleteReplicationJobCommand, "DeleteReplicationJobCommand");
var DeleteReplicationJobCommand = _DeleteReplicationJobCommand;

// src/commands/DeleteServerCatalogCommand.ts




var _DeleteServerCatalogCommand = class _DeleteServerCatalogCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AWSServerMigrationService_V2016_10_24", "DeleteServerCatalog", {}).n("SMSClient", "DeleteServerCatalogCommand").f(void 0, void 0).ser(se_DeleteServerCatalogCommand).de(de_DeleteServerCatalogCommand).build() {
};
__name(_DeleteServerCatalogCommand, "DeleteServerCatalogCommand");
var DeleteServerCatalogCommand = _DeleteServerCatalogCommand;

// src/commands/DisassociateConnectorCommand.ts




var _DisassociateConnectorCommand = class _DisassociateConnectorCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AWSServerMigrationService_V2016_10_24", "DisassociateConnector", {}).n("SMSClient", "DisassociateConnectorCommand").f(void 0, void 0).ser(se_DisassociateConnectorCommand).de(de_DisassociateConnectorCommand).build() {
};
__name(_DisassociateConnectorCommand, "DisassociateConnectorCommand");
var DisassociateConnectorCommand = _DisassociateConnectorCommand;

// src/commands/GenerateChangeSetCommand.ts




var _GenerateChangeSetCommand = class _GenerateChangeSetCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AWSServerMigrationService_V2016_10_24", "GenerateChangeSet", {}).n("SMSClient", "GenerateChangeSetCommand").f(void 0, void 0).ser(se_GenerateChangeSetCommand).de(de_GenerateChangeSetCommand).build() {
};
__name(_GenerateChangeSetCommand, "GenerateChangeSetCommand");
var GenerateChangeSetCommand = _GenerateChangeSetCommand;

// src/commands/GenerateTemplateCommand.ts




var _GenerateTemplateCommand = class _GenerateTemplateCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AWSServerMigrationService_V2016_10_24", "GenerateTemplate", {}).n("SMSClient", "GenerateTemplateCommand").f(void 0, void 0).ser(se_GenerateTemplateCommand).de(de_GenerateTemplateCommand).build() {
};
__name(_GenerateTemplateCommand, "GenerateTemplateCommand");
var GenerateTemplateCommand = _GenerateTemplateCommand;

// src/commands/GetAppCommand.ts




var _GetAppCommand = class _GetAppCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AWSServerMigrationService_V2016_10_24", "GetApp", {}).n("SMSClient", "GetAppCommand").f(void 0, void 0).ser(se_GetAppCommand).de(de_GetAppCommand).build() {
};
__name(_GetAppCommand, "GetAppCommand");
var GetAppCommand = _GetAppCommand;

// src/commands/GetAppLaunchConfigurationCommand.ts




var _GetAppLaunchConfigurationCommand = class _GetAppLaunchConfigurationCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AWSServerMigrationService_V2016_10_24", "GetAppLaunchConfiguration", {}).n("SMSClient", "GetAppLaunchConfigurationCommand").f(void 0, void 0).ser(se_GetAppLaunchConfigurationCommand).de(de_GetAppLaunchConfigurationCommand).build() {
};
__name(_GetAppLaunchConfigurationCommand, "GetAppLaunchConfigurationCommand");
var GetAppLaunchConfigurationCommand = _GetAppLaunchConfigurationCommand;

// src/commands/GetAppReplicationConfigurationCommand.ts




var _GetAppReplicationConfigurationCommand = class _GetAppReplicationConfigurationCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AWSServerMigrationService_V2016_10_24", "GetAppReplicationConfiguration", {}).n("SMSClient", "GetAppReplicationConfigurationCommand").f(void 0, void 0).ser(se_GetAppReplicationConfigurationCommand).de(de_GetAppReplicationConfigurationCommand).build() {
};
__name(_GetAppReplicationConfigurationCommand, "GetAppReplicationConfigurationCommand");
var GetAppReplicationConfigurationCommand = _GetAppReplicationConfigurationCommand;

// src/commands/GetAppValidationConfigurationCommand.ts




var _GetAppValidationConfigurationCommand = class _GetAppValidationConfigurationCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AWSServerMigrationService_V2016_10_24", "GetAppValidationConfiguration", {}).n("SMSClient", "GetAppValidationConfigurationCommand").f(void 0, void 0).ser(se_GetAppValidationConfigurationCommand).de(de_GetAppValidationConfigurationCommand).build() {
};
__name(_GetAppValidationConfigurationCommand, "GetAppValidationConfigurationCommand");
var GetAppValidationConfigurationCommand = _GetAppValidationConfigurationCommand;

// src/commands/GetAppValidationOutputCommand.ts




var _GetAppValidationOutputCommand = class _GetAppValidationOutputCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AWSServerMigrationService_V2016_10_24", "GetAppValidationOutput", {}).n("SMSClient", "GetAppValidationOutputCommand").f(void 0, void 0).ser(se_GetAppValidationOutputCommand).de(de_GetAppValidationOutputCommand).build() {
};
__name(_GetAppValidationOutputCommand, "GetAppValidationOutputCommand");
var GetAppValidationOutputCommand = _GetAppValidationOutputCommand;

// src/commands/GetConnectorsCommand.ts




var _GetConnectorsCommand = class _GetConnectorsCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AWSServerMigrationService_V2016_10_24", "GetConnectors", {}).n("SMSClient", "GetConnectorsCommand").f(void 0, void 0).ser(se_GetConnectorsCommand).de(de_GetConnectorsCommand).build() {
};
__name(_GetConnectorsCommand, "GetConnectorsCommand");
var GetConnectorsCommand = _GetConnectorsCommand;

// src/commands/GetReplicationJobsCommand.ts




var _GetReplicationJobsCommand = class _GetReplicationJobsCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AWSServerMigrationService_V2016_10_24", "GetReplicationJobs", {}).n("SMSClient", "GetReplicationJobsCommand").f(void 0, void 0).ser(se_GetReplicationJobsCommand).de(de_GetReplicationJobsCommand).build() {
};
__name(_GetReplicationJobsCommand, "GetReplicationJobsCommand");
var GetReplicationJobsCommand = _GetReplicationJobsCommand;

// src/commands/GetReplicationRunsCommand.ts




var _GetReplicationRunsCommand = class _GetReplicationRunsCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AWSServerMigrationService_V2016_10_24", "GetReplicationRuns", {}).n("SMSClient", "GetReplicationRunsCommand").f(void 0, void 0).ser(se_GetReplicationRunsCommand).de(de_GetReplicationRunsCommand).build() {
};
__name(_GetReplicationRunsCommand, "GetReplicationRunsCommand");
var GetReplicationRunsCommand = _GetReplicationRunsCommand;

// src/commands/GetServersCommand.ts




var _GetServersCommand = class _GetServersCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AWSServerMigrationService_V2016_10_24", "GetServers", {}).n("SMSClient", "GetServersCommand").f(void 0, void 0).ser(se_GetServersCommand).de(de_GetServersCommand).build() {
};
__name(_GetServersCommand, "GetServersCommand");
var GetServersCommand = _GetServersCommand;

// src/commands/ImportAppCatalogCommand.ts




var _ImportAppCatalogCommand = class _ImportAppCatalogCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AWSServerMigrationService_V2016_10_24", "ImportAppCatalog", {}).n("SMSClient", "ImportAppCatalogCommand").f(void 0, void 0).ser(se_ImportAppCatalogCommand).de(de_ImportAppCatalogCommand).build() {
};
__name(_ImportAppCatalogCommand, "ImportAppCatalogCommand");
var ImportAppCatalogCommand = _ImportAppCatalogCommand;

// src/commands/ImportServerCatalogCommand.ts




var _ImportServerCatalogCommand = class _ImportServerCatalogCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AWSServerMigrationService_V2016_10_24", "ImportServerCatalog", {}).n("SMSClient", "ImportServerCatalogCommand").f(void 0, void 0).ser(se_ImportServerCatalogCommand).de(de_ImportServerCatalogCommand).build() {
};
__name(_ImportServerCatalogCommand, "ImportServerCatalogCommand");
var ImportServerCatalogCommand = _ImportServerCatalogCommand;

// src/commands/LaunchAppCommand.ts




var _LaunchAppCommand = class _LaunchAppCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AWSServerMigrationService_V2016_10_24", "LaunchApp", {}).n("SMSClient", "LaunchAppCommand").f(void 0, void 0).ser(se_LaunchAppCommand).de(de_LaunchAppCommand).build() {
};
__name(_LaunchAppCommand, "LaunchAppCommand");
var LaunchAppCommand = _LaunchAppCommand;

// src/commands/ListAppsCommand.ts




var _ListAppsCommand = class _ListAppsCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AWSServerMigrationService_V2016_10_24", "ListApps", {}).n("SMSClient", "ListAppsCommand").f(void 0, void 0).ser(se_ListAppsCommand).de(de_ListAppsCommand).build() {
};
__name(_ListAppsCommand, "ListAppsCommand");
var ListAppsCommand = _ListAppsCommand;

// src/commands/NotifyAppValidationOutputCommand.ts




var _NotifyAppValidationOutputCommand = class _NotifyAppValidationOutputCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AWSServerMigrationService_V2016_10_24", "NotifyAppValidationOutput", {}).n("SMSClient", "NotifyAppValidationOutputCommand").f(void 0, void 0).ser(se_NotifyAppValidationOutputCommand).de(de_NotifyAppValidationOutputCommand).build() {
};
__name(_NotifyAppValidationOutputCommand, "NotifyAppValidationOutputCommand");
var NotifyAppValidationOutputCommand = _NotifyAppValidationOutputCommand;

// src/commands/PutAppLaunchConfigurationCommand.ts




var _PutAppLaunchConfigurationCommand = class _PutAppLaunchConfigurationCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AWSServerMigrationService_V2016_10_24", "PutAppLaunchConfiguration", {}).n("SMSClient", "PutAppLaunchConfigurationCommand").f(void 0, void 0).ser(se_PutAppLaunchConfigurationCommand).de(de_PutAppLaunchConfigurationCommand).build() {
};
__name(_PutAppLaunchConfigurationCommand, "PutAppLaunchConfigurationCommand");
var PutAppLaunchConfigurationCommand = _PutAppLaunchConfigurationCommand;

// src/commands/PutAppReplicationConfigurationCommand.ts




var _PutAppReplicationConfigurationCommand = class _PutAppReplicationConfigurationCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AWSServerMigrationService_V2016_10_24", "PutAppReplicationConfiguration", {}).n("SMSClient", "PutAppReplicationConfigurationCommand").f(void 0, void 0).ser(se_PutAppReplicationConfigurationCommand).de(de_PutAppReplicationConfigurationCommand).build() {
};
__name(_PutAppReplicationConfigurationCommand, "PutAppReplicationConfigurationCommand");
var PutAppReplicationConfigurationCommand = _PutAppReplicationConfigurationCommand;

// src/commands/PutAppValidationConfigurationCommand.ts




var _PutAppValidationConfigurationCommand = class _PutAppValidationConfigurationCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AWSServerMigrationService_V2016_10_24", "PutAppValidationConfiguration", {}).n("SMSClient", "PutAppValidationConfigurationCommand").f(void 0, void 0).ser(se_PutAppValidationConfigurationCommand).de(de_PutAppValidationConfigurationCommand).build() {
};
__name(_PutAppValidationConfigurationCommand, "PutAppValidationConfigurationCommand");
var PutAppValidationConfigurationCommand = _PutAppValidationConfigurationCommand;

// src/commands/StartAppReplicationCommand.ts




var _StartAppReplicationCommand = class _StartAppReplicationCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AWSServerMigrationService_V2016_10_24", "StartAppReplication", {}).n("SMSClient", "StartAppReplicationCommand").f(void 0, void 0).ser(se_StartAppReplicationCommand).de(de_StartAppReplicationCommand).build() {
};
__name(_StartAppReplicationCommand, "StartAppReplicationCommand");
var StartAppReplicationCommand = _StartAppReplicationCommand;

// src/commands/StartOnDemandAppReplicationCommand.ts




var _StartOnDemandAppReplicationCommand = class _StartOnDemandAppReplicationCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AWSServerMigrationService_V2016_10_24", "StartOnDemandAppReplication", {}).n("SMSClient", "StartOnDemandAppReplicationCommand").f(void 0, void 0).ser(se_StartOnDemandAppReplicationCommand).de(de_StartOnDemandAppReplicationCommand).build() {
};
__name(_StartOnDemandAppReplicationCommand, "StartOnDemandAppReplicationCommand");
var StartOnDemandAppReplicationCommand = _StartOnDemandAppReplicationCommand;

// src/commands/StartOnDemandReplicationRunCommand.ts




var _StartOnDemandReplicationRunCommand = class _StartOnDemandReplicationRunCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AWSServerMigrationService_V2016_10_24", "StartOnDemandReplicationRun", {}).n("SMSClient", "StartOnDemandReplicationRunCommand").f(void 0, void 0).ser(se_StartOnDemandReplicationRunCommand).de(de_StartOnDemandReplicationRunCommand).build() {
};
__name(_StartOnDemandReplicationRunCommand, "StartOnDemandReplicationRunCommand");
var StartOnDemandReplicationRunCommand = _StartOnDemandReplicationRunCommand;

// src/commands/StopAppReplicationCommand.ts




var _StopAppReplicationCommand = class _StopAppReplicationCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AWSServerMigrationService_V2016_10_24", "StopAppReplication", {}).n("SMSClient", "StopAppReplicationCommand").f(void 0, void 0).ser(se_StopAppReplicationCommand).de(de_StopAppReplicationCommand).build() {
};
__name(_StopAppReplicationCommand, "StopAppReplicationCommand");
var StopAppReplicationCommand = _StopAppReplicationCommand;

// src/commands/TerminateAppCommand.ts




var _TerminateAppCommand = class _TerminateAppCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AWSServerMigrationService_V2016_10_24", "TerminateApp", {}).n("SMSClient", "TerminateAppCommand").f(void 0, void 0).ser(se_TerminateAppCommand).de(de_TerminateAppCommand).build() {
};
__name(_TerminateAppCommand, "TerminateAppCommand");
var TerminateAppCommand = _TerminateAppCommand;

// src/commands/UpdateAppCommand.ts




var _UpdateAppCommand = class _UpdateAppCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AWSServerMigrationService_V2016_10_24", "UpdateApp", {}).n("SMSClient", "UpdateAppCommand").f(void 0, void 0).ser(se_UpdateAppCommand).de(de_UpdateAppCommand).build() {
};
__name(_UpdateAppCommand, "UpdateAppCommand");
var UpdateAppCommand = _UpdateAppCommand;

// src/commands/UpdateReplicationJobCommand.ts




var _UpdateReplicationJobCommand = class _UpdateReplicationJobCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AWSServerMigrationService_V2016_10_24", "UpdateReplicationJob", {}).n("SMSClient", "UpdateReplicationJobCommand").f(void 0, void 0).ser(se_UpdateReplicationJobCommand).de(de_UpdateReplicationJobCommand).build() {
};
__name(_UpdateReplicationJobCommand, "UpdateReplicationJobCommand");
var UpdateReplicationJobCommand = _UpdateReplicationJobCommand;

// src/SMS.ts
var commands = {
  CreateAppCommand,
  CreateReplicationJobCommand,
  DeleteAppCommand,
  DeleteAppLaunchConfigurationCommand,
  DeleteAppReplicationConfigurationCommand,
  DeleteAppValidationConfigurationCommand,
  DeleteReplicationJobCommand,
  DeleteServerCatalogCommand,
  DisassociateConnectorCommand,
  GenerateChangeSetCommand,
  GenerateTemplateCommand,
  GetAppCommand,
  GetAppLaunchConfigurationCommand,
  GetAppReplicationConfigurationCommand,
  GetAppValidationConfigurationCommand,
  GetAppValidationOutputCommand,
  GetConnectorsCommand,
  GetReplicationJobsCommand,
  GetReplicationRunsCommand,
  GetServersCommand,
  ImportAppCatalogCommand,
  ImportServerCatalogCommand,
  LaunchAppCommand,
  ListAppsCommand,
  NotifyAppValidationOutputCommand,
  PutAppLaunchConfigurationCommand,
  PutAppReplicationConfigurationCommand,
  PutAppValidationConfigurationCommand,
  StartAppReplicationCommand,
  StartOnDemandAppReplicationCommand,
  StartOnDemandReplicationRunCommand,
  StopAppReplicationCommand,
  TerminateAppCommand,
  UpdateAppCommand,
  UpdateReplicationJobCommand
};
var _SMS = class _SMS extends SMSClient {
};
__name(_SMS, "SMS");
var SMS = _SMS;
(0, import_smithy_client.createAggregatedClient)(commands, SMS);

// src/pagination/GetConnectorsPaginator.ts

var paginateGetConnectors = (0, import_core.createPaginator)(SMSClient, GetConnectorsCommand, "nextToken", "nextToken", "maxResults");

// src/pagination/GetReplicationJobsPaginator.ts

var paginateGetReplicationJobs = (0, import_core.createPaginator)(SMSClient, GetReplicationJobsCommand, "nextToken", "nextToken", "maxResults");

// src/pagination/GetReplicationRunsPaginator.ts

var paginateGetReplicationRuns = (0, import_core.createPaginator)(SMSClient, GetReplicationRunsCommand, "nextToken", "nextToken", "maxResults");

// src/pagination/GetServersPaginator.ts

var paginateGetServers = (0, import_core.createPaginator)(SMSClient, GetServersCommand, "nextToken", "nextToken", "maxResults");
// Annotate the CommonJS export names for ESM import in node:

0 && (module.exports = {
  SMSServiceException,
  __Client,
  SMSClient,
  SMS,
  $Command,
  CreateAppCommand,
  CreateReplicationJobCommand,
  DeleteAppCommand,
  DeleteAppLaunchConfigurationCommand,
  DeleteAppReplicationConfigurationCommand,
  DeleteAppValidationConfigurationCommand,
  DeleteReplicationJobCommand,
  DeleteServerCatalogCommand,
  DisassociateConnectorCommand,
  GenerateChangeSetCommand,
  GenerateTemplateCommand,
  GetAppCommand,
  GetAppLaunchConfigurationCommand,
  GetAppReplicationConfigurationCommand,
  GetAppValidationConfigurationCommand,
  GetAppValidationOutputCommand,
  GetConnectorsCommand,
  GetReplicationJobsCommand,
  GetReplicationRunsCommand,
  GetServersCommand,
  ImportAppCatalogCommand,
  ImportServerCatalogCommand,
  LaunchAppCommand,
  ListAppsCommand,
  NotifyAppValidationOutputCommand,
  PutAppLaunchConfigurationCommand,
  PutAppReplicationConfigurationCommand,
  PutAppValidationConfigurationCommand,
  StartAppReplicationCommand,
  StartOnDemandAppReplicationCommand,
  StartOnDemandReplicationRunCommand,
  StopAppReplicationCommand,
  TerminateAppCommand,
  UpdateAppCommand,
  UpdateReplicationJobCommand,
  paginateGetConnectors,
  paginateGetReplicationJobs,
  paginateGetReplicationRuns,
  paginateGetServers,
  AppLaunchConfigurationStatus,
  AppLaunchStatus,
  AppReplicationConfigurationStatus,
  AppReplicationStatus,
  AppStatus,
  AppValidationStrategy,
  ScriptType,
  ServerType,
  VmManagerType,
  InternalError,
  InvalidParameterException,
  MissingRequiredParameterException,
  OperationNotPermittedException,
  UnauthorizedOperationException,
  LicenseType,
  NoConnectorsAvailableException,
  ReplicationJobAlreadyExistsException,
  ServerCannotBeReplicatedException,
  TemporarilyUnavailableException,
  ReplicationJobNotFoundException,
  OutputFormat,
  ServerValidationStrategy,
  ValidationStatus,
  ConnectorCapability,
  ConnectorStatus,
  ReplicationRunState,
  ReplicationRunType,
  ReplicationJobState,
  ServerCatalogStatus,
  DryRunOperationException,
  ReplicationRunLimitExceededException
});
