// @azure/arm-relay/dist-esm/src/models/parameters.js
/*
 * Copyright (c) Microsoft Corporation.
 * Licensed under the MIT License.
 *
 * Code generated by Microsoft (R) AutoRest Code Generator.
 * Changes may cause incorrect behavior and will be lost if the code is regenerated.
 */
import { CheckNameAvailability as CheckNameAvailabilityMapper, RelayNamespace as RelayNamespaceMapper, RelayUpdateParameters as RelayUpdateParametersMapper, AuthorizationRule as AuthorizationRuleMapper, RegenerateAccessKeyParameters as RegenerateAccessKeyParametersMapper, HybridConnection as HybridConnectionMapper, WcfRelay as WcfRelayMapper } from "../models/mappers";
export const accept = {
    parameterPath: "accept",
    mapper: {
        defaultValue: "application/json",
        isConstant: true,
        serializedName: "Accept",
        type: {
            name: "String"
        }
    }
};
export const $host = {
    parameterPath: "$host",
    mapper: {
        serializedName: "$host",
        required: true,
        type: {
            name: "String"
        }
    },
    skipEncoding: true
};
export const apiVersion = {
    parameterPath: "apiVersion",
    mapper: {
        defaultValue: "2017-04-01",
        isConstant: true,
        serializedName: "api-version",
        type: {
            name: "String"
        }
    }
};
export const nextLink = {
    parameterPath: "nextLink",
    mapper: {
        serializedName: "nextLink",
        required: true,
        type: {
            name: "String"
        }
    },
    skipEncoding: true
};
export const contentType = {
    parameterPath: ["options", "contentType"],
    mapper: {
        defaultValue: "application/json",
        isConstant: true,
        serializedName: "Content-Type",
        type: {
            name: "String"
        }
    }
};
export const parameters = {
    parameterPath: "parameters",
    mapper: CheckNameAvailabilityMapper
};
export const subscriptionId = {
    parameterPath: "subscriptionId",
    mapper: {
        serializedName: "subscriptionId",
        required: true,
        type: {
            name: "String"
        }
    }
};
export const resourceGroupName = {
    parameterPath: "resourceGroupName",
    mapper: {
        constraints: {
            MaxLength: 90,
            MinLength: 1
        },
        serializedName: "resourceGroupName",
        required: true,
        type: {
            name: "String"
        }
    }
};
export const parameters1 = {
    parameterPath: "parameters",
    mapper: RelayNamespaceMapper
};
export const namespaceName = {
    parameterPath: "namespaceName",
    mapper: {
        constraints: {
            MaxLength: 50,
            MinLength: 6
        },
        serializedName: "namespaceName",
        required: true,
        type: {
            name: "String"
        }
    }
};
export const parameters2 = {
    parameterPath: "parameters",
    mapper: RelayUpdateParametersMapper
};
export const parameters3 = {
    parameterPath: "parameters",
    mapper: AuthorizationRuleMapper
};
export const authorizationRuleName = {
    parameterPath: "authorizationRuleName",
    mapper: {
        constraints: {
            MinLength: 1
        },
        serializedName: "authorizationRuleName",
        required: true,
        type: {
            name: "String"
        }
    }
};
export const parameters4 = {
    parameterPath: "parameters",
    mapper: RegenerateAccessKeyParametersMapper
};
export const parameters5 = {
    parameterPath: "parameters",
    mapper: HybridConnectionMapper
};
export const hybridConnectionName = {
    parameterPath: "hybridConnectionName",
    mapper: {
        constraints: {
            MinLength: 1
        },
        serializedName: "hybridConnectionName",
        required: true,
        type: {
            name: "String"
        }
    }
};
export const parameters6 = {
    parameterPath: "parameters",
    mapper: WcfRelayMapper
};
export const relayName = {
    parameterPath: "relayName",
    mapper: {
        constraints: {
            MinLength: 1
        },
        serializedName: "relayName",
        required: true,
        type: {
            name: "String"
        }
    }
};
