// @azure/arm-cognitiveservices/dist-esm/src/models/mappers.js
/*
 * Copyright (c) Microsoft Corporation.
 * Licensed under the MIT License.
 *
 * Code generated by Microsoft (R) AutoRest Code Generator.
 * Changes may cause incorrect behavior and will be lost if the code is regenerated.
 */
export const Sku = {
    type: {
        name: "Composite",
        className: "Sku",
        modelProperties: {
            name: {
                serializedName: "name",
                required: true,
                type: {
                    name: "String"
                }
            },
            tier: {
                serializedName: "tier",
                type: {
                    name: "String"
                }
            },
            size: {
                serializedName: "size",
                type: {
                    name: "String"
                }
            },
            family: {
                serializedName: "family",
                type: {
                    name: "String"
                }
            },
            capacity: {
                serializedName: "capacity",
                type: {
                    name: "Number"
                }
            }
        }
    }
};
export const Identity = {
    type: {
        name: "Composite",
        className: "Identity",
        modelProperties: {
            type: {
                serializedName: "type",
                type: {
                    name: "Enum",
                    allowedValues: [
                        "None",
                        "SystemAssigned",
                        "UserAssigned",
                        "SystemAssigned, UserAssigned"
                    ]
                }
            },
            tenantId: {
                serializedName: "tenantId",
                readOnly: true,
                type: {
                    name: "String"
                }
            },
            principalId: {
                serializedName: "principalId",
                readOnly: true,
                type: {
                    name: "String"
                }
            },
            userAssignedIdentities: {
                serializedName: "userAssignedIdentities",
                type: {
                    name: "Dictionary",
                    value: {
                        type: { name: "Composite", className: "UserAssignedIdentity" }
                    }
                }
            }
        }
    }
};
export const UserAssignedIdentity = {
    type: {
        name: "Composite",
        className: "UserAssignedIdentity",
        modelProperties: {
            principalId: {
                serializedName: "principalId",
                readOnly: true,
                type: {
                    name: "String"
                }
            },
            clientId: {
                serializedName: "clientId",
                readOnly: true,
                type: {
                    name: "String"
                }
            }
        }
    }
};
export const SystemData = {
    type: {
        name: "Composite",
        className: "SystemData",
        modelProperties: {
            createdBy: {
                serializedName: "createdBy",
                type: {
                    name: "String"
                }
            },
            createdByType: {
                serializedName: "createdByType",
                type: {
                    name: "String"
                }
            },
            createdAt: {
                serializedName: "createdAt",
                type: {
                    name: "DateTime"
                }
            },
            lastModifiedBy: {
                serializedName: "lastModifiedBy",
                type: {
                    name: "String"
                }
            },
            lastModifiedByType: {
                serializedName: "lastModifiedByType",
                type: {
                    name: "String"
                }
            },
            lastModifiedAt: {
                serializedName: "lastModifiedAt",
                type: {
                    name: "DateTime"
                }
            }
        }
    }
};
export const AccountProperties = {
    type: {
        name: "Composite",
        className: "AccountProperties",
        modelProperties: {
            provisioningState: {
                serializedName: "provisioningState",
                readOnly: true,
                type: {
                    name: "String"
                }
            },
            endpoint: {
                serializedName: "endpoint",
                readOnly: true,
                type: {
                    name: "String"
                }
            },
            internalId: {
                serializedName: "internalId",
                readOnly: true,
                type: {
                    name: "String"
                }
            },
            capabilities: {
                serializedName: "capabilities",
                readOnly: true,
                type: {
                    name: "Sequence",
                    element: {
                        type: {
                            name: "Composite",
                            className: "SkuCapability"
                        }
                    }
                }
            },
            isMigrated: {
                serializedName: "isMigrated",
                readOnly: true,
                type: {
                    name: "Boolean"
                }
            },
            migrationToken: {
                serializedName: "migrationToken",
                type: {
                    name: "String"
                }
            },
            skuChangeInfo: {
                serializedName: "skuChangeInfo",
                type: {
                    name: "Composite",
                    className: "SkuChangeInfo"
                }
            },
            customSubDomainName: {
                serializedName: "customSubDomainName",
                type: {
                    name: "String"
                }
            },
            networkAcls: {
                serializedName: "networkAcls",
                type: {
                    name: "Composite",
                    className: "NetworkRuleSet"
                }
            },
            encryption: {
                serializedName: "encryption",
                type: {
                    name: "Composite",
                    className: "Encryption"
                }
            },
            userOwnedStorage: {
                serializedName: "userOwnedStorage",
                type: {
                    name: "Sequence",
                    element: {
                        type: {
                            name: "Composite",
                            className: "UserOwnedStorage"
                        }
                    }
                }
            },
            privateEndpointConnections: {
                serializedName: "privateEndpointConnections",
                readOnly: true,
                type: {
                    name: "Sequence",
                    element: {
                        type: {
                            name: "Composite",
                            className: "PrivateEndpointConnection"
                        }
                    }
                }
            },
            publicNetworkAccess: {
                serializedName: "publicNetworkAccess",
                type: {
                    name: "String"
                }
            },
            apiProperties: {
                serializedName: "apiProperties",
                type: {
                    name: "Composite",
                    className: "ApiProperties"
                }
            },
            dateCreated: {
                serializedName: "dateCreated",
                readOnly: true,
                type: {
                    name: "String"
                }
            },
            callRateLimit: {
                serializedName: "callRateLimit",
                type: {
                    name: "Composite",
                    className: "CallRateLimit"
                }
            },
            dynamicThrottlingEnabled: {
                serializedName: "dynamicThrottlingEnabled",
                type: {
                    name: "Boolean"
                }
            },
            quotaLimit: {
                serializedName: "quotaLimit",
                type: {
                    name: "Composite",
                    className: "QuotaLimit"
                }
            },
            restrictOutboundNetworkAccess: {
                serializedName: "restrictOutboundNetworkAccess",
                type: {
                    name: "Boolean"
                }
            },
            allowedFqdnList: {
                serializedName: "allowedFqdnList",
                type: {
                    name: "Sequence",
                    element: {
                        type: {
                            name: "String"
                        }
                    }
                }
            },
            disableLocalAuth: {
                serializedName: "disableLocalAuth",
                type: {
                    name: "Boolean"
                }
            },
            endpoints: {
                serializedName: "endpoints",
                readOnly: true,
                type: {
                    name: "Dictionary",
                    value: { type: { name: "String" } }
                }
            },
            restore: {
                serializedName: "restore",
                type: {
                    name: "Boolean"
                }
            },
            deletionDate: {
                serializedName: "deletionDate",
                readOnly: true,
                type: {
                    name: "String"
                }
            },
            scheduledPurgeDate: {
                serializedName: "scheduledPurgeDate",
                readOnly: true,
                type: {
                    name: "String"
                }
            },
            locations: {
                serializedName: "locations",
                type: {
                    name: "Composite",
                    className: "MultiRegionSettings"
                }
            },
            commitmentPlanAssociations: {
                serializedName: "commitmentPlanAssociations",
                readOnly: true,
                type: {
                    name: "Sequence",
                    element: {
                        type: {
                            name: "Composite",
                            className: "CommitmentPlanAssociation"
                        }
                    }
                }
            },
            abusePenalty: {
                serializedName: "abusePenalty",
                type: {
                    name: "Composite",
                    className: "AbusePenalty"
                }
            }
        }
    }
};
export const SkuCapability = {
    type: {
        name: "Composite",
        className: "SkuCapability",
        modelProperties: {
            name: {
                serializedName: "name",
                type: {
                    name: "String"
                }
            },
            value: {
                serializedName: "value",
                type: {
                    name: "String"
                }
            }
        }
    }
};
export const SkuChangeInfo = {
    type: {
        name: "Composite",
        className: "SkuChangeInfo",
        modelProperties: {
            countOfDowngrades: {
                serializedName: "countOfDowngrades",
                type: {
                    name: "Number"
                }
            },
            countOfUpgradesAfterDowngrades: {
                serializedName: "countOfUpgradesAfterDowngrades",
                type: {
                    name: "Number"
                }
            },
            lastChangeDate: {
                serializedName: "lastChangeDate",
                type: {
                    name: "String"
                }
            }
        }
    }
};
export const NetworkRuleSet = {
    type: {
        name: "Composite",
        className: "NetworkRuleSet",
        modelProperties: {
            defaultAction: {
                serializedName: "defaultAction",
                type: {
                    name: "String"
                }
            },
            ipRules: {
                serializedName: "ipRules",
                type: {
                    name: "Sequence",
                    element: {
                        type: {
                            name: "Composite",
                            className: "IpRule"
                        }
                    }
                }
            },
            virtualNetworkRules: {
                serializedName: "virtualNetworkRules",
                type: {
                    name: "Sequence",
                    element: {
                        type: {
                            name: "Composite",
                            className: "VirtualNetworkRule"
                        }
                    }
                }
            }
        }
    }
};
export const IpRule = {
    type: {
        name: "Composite",
        className: "IpRule",
        modelProperties: {
            value: {
                serializedName: "value",
                required: true,
                type: {
                    name: "String"
                }
            }
        }
    }
};
export const VirtualNetworkRule = {
    type: {
        name: "Composite",
        className: "VirtualNetworkRule",
        modelProperties: {
            id: {
                serializedName: "id",
                required: true,
                type: {
                    name: "String"
                }
            },
            state: {
                serializedName: "state",
                type: {
                    name: "String"
                }
            },
            ignoreMissingVnetServiceEndpoint: {
                serializedName: "ignoreMissingVnetServiceEndpoint",
                type: {
                    name: "Boolean"
                }
            }
        }
    }
};
export const Encryption = {
    type: {
        name: "Composite",
        className: "Encryption",
        modelProperties: {
            keyVaultProperties: {
                serializedName: "keyVaultProperties",
                type: {
                    name: "Composite",
                    className: "KeyVaultProperties"
                }
            },
            keySource: {
                defaultValue: "Microsoft.KeyVault",
                serializedName: "keySource",
                type: {
                    name: "String"
                }
            }
        }
    }
};
export const KeyVaultProperties = {
    type: {
        name: "Composite",
        className: "KeyVaultProperties",
        modelProperties: {
            keyName: {
                serializedName: "keyName",
                type: {
                    name: "String"
                }
            },
            keyVersion: {
                serializedName: "keyVersion",
                type: {
                    name: "String"
                }
            },
            keyVaultUri: {
                serializedName: "keyVaultUri",
                type: {
                    name: "String"
                }
            },
            identityClientId: {
                serializedName: "identityClientId",
                type: {
                    name: "String"
                }
            }
        }
    }
};
export const UserOwnedStorage = {
    type: {
        name: "Composite",
        className: "UserOwnedStorage",
        modelProperties: {
            resourceId: {
                serializedName: "resourceId",
                type: {
                    name: "String"
                }
            },
            identityClientId: {
                serializedName: "identityClientId",
                type: {
                    name: "String"
                }
            }
        }
    }
};
export const PrivateEndpointConnectionProperties = {
    type: {
        name: "Composite",
        className: "PrivateEndpointConnectionProperties",
        modelProperties: {
            privateEndpoint: {
                serializedName: "privateEndpoint",
                type: {
                    name: "Composite",
                    className: "PrivateEndpoint"
                }
            },
            privateLinkServiceConnectionState: {
                serializedName: "privateLinkServiceConnectionState",
                type: {
                    name: "Composite",
                    className: "PrivateLinkServiceConnectionState"
                }
            },
            provisioningState: {
                serializedName: "provisioningState",
                readOnly: true,
                type: {
                    name: "String"
                }
            },
            groupIds: {
                serializedName: "groupIds",
                type: {
                    name: "Sequence",
                    element: {
                        type: {
                            name: "String"
                        }
                    }
                }
            }
        }
    }
};
export const PrivateEndpoint = {
    type: {
        name: "Composite",
        className: "PrivateEndpoint",
        modelProperties: {
            id: {
                serializedName: "id",
                readOnly: true,
                type: {
                    name: "String"
                }
            }
        }
    }
};
export const PrivateLinkServiceConnectionState = {
    type: {
        name: "Composite",
        className: "PrivateLinkServiceConnectionState",
        modelProperties: {
            status: {
                serializedName: "status",
                type: {
                    name: "String"
                }
            },
            description: {
                serializedName: "description",
                type: {
                    name: "String"
                }
            },
            actionsRequired: {
                serializedName: "actionsRequired",
                type: {
                    name: "String"
                }
            }
        }
    }
};
export const Resource = {
    type: {
        name: "Composite",
        className: "Resource",
        modelProperties: {
            id: {
                serializedName: "id",
                readOnly: true,
                type: {
                    name: "String"
                }
            },
            name: {
                serializedName: "name",
                readOnly: true,
                type: {
                    name: "String"
                }
            },
            type: {
                serializedName: "type",
                readOnly: true,
                type: {
                    name: "String"
                }
            }
        }
    }
};
export const ApiProperties = {
    type: {
        name: "Composite",
        className: "ApiProperties",
        additionalProperties: { type: { name: "Object" } },
        modelProperties: {
            qnaRuntimeEndpoint: {
                serializedName: "qnaRuntimeEndpoint",
                type: {
                    name: "String"
                }
            },
            qnaAzureSearchEndpointKey: {
                serializedName: "qnaAzureSearchEndpointKey",
                type: {
                    name: "String"
                }
            },
            qnaAzureSearchEndpointId: {
                serializedName: "qnaAzureSearchEndpointId",
                type: {
                    name: "String"
                }
            },
            statisticsEnabled: {
                serializedName: "statisticsEnabled",
                type: {
                    name: "Boolean"
                }
            },
            eventHubConnectionString: {
                constraints: {
                    Pattern: new RegExp("^( *)Endpoint=sb:\\/\\/(.*);( *)SharedAccessKeyName=(.*);( *)SharedAccessKey=(.*)$"),
                    MaxLength: 1000
                },
                serializedName: "eventHubConnectionString",
                type: {
                    name: "String"
                }
            },
            storageAccountConnectionString: {
                constraints: {
                    Pattern: new RegExp("^(( *)DefaultEndpointsProtocol=(http|https)( *);( *))?AccountName=(.*)AccountKey=(.*)EndpointSuffix=(.*)$"),
                    MaxLength: 1000
                },
                serializedName: "storageAccountConnectionString",
                type: {
                    name: "String"
                }
            },
            aadClientId: {
                constraints: {
                    MaxLength: 500
                },
                serializedName: "aadClientId",
                type: {
                    name: "String"
                }
            },
            aadTenantId: {
                constraints: {
                    MaxLength: 500
                },
                serializedName: "aadTenantId",
                type: {
                    name: "String"
                }
            },
            superUser: {
                constraints: {
                    MaxLength: 500
                },
                serializedName: "superUser",
                type: {
                    name: "String"
                }
            },
            websiteName: {
                constraints: {
                    MaxLength: 500
                },
                serializedName: "websiteName",
                type: {
                    name: "String"
                }
            }
        }
    }
};
export const CallRateLimit = {
    type: {
        name: "Composite",
        className: "CallRateLimit",
        modelProperties: {
            count: {
                serializedName: "count",
                type: {
                    name: "Number"
                }
            },
            renewalPeriod: {
                serializedName: "renewalPeriod",
                type: {
                    name: "Number"
                }
            },
            rules: {
                serializedName: "rules",
                type: {
                    name: "Sequence",
                    element: {
                        type: {
                            name: "Composite",
                            className: "ThrottlingRule"
                        }
                    }
                }
            }
        }
    }
};
export const ThrottlingRule = {
    type: {
        name: "Composite",
        className: "ThrottlingRule",
        modelProperties: {
            key: {
                serializedName: "key",
                type: {
                    name: "String"
                }
            },
            renewalPeriod: {
                serializedName: "renewalPeriod",
                type: {
                    name: "Number"
                }
            },
            count: {
                serializedName: "count",
                type: {
                    name: "Number"
                }
            },
            minCount: {
                serializedName: "minCount",
                type: {
                    name: "Number"
                }
            },
            dynamicThrottlingEnabled: {
                serializedName: "dynamicThrottlingEnabled",
                type: {
                    name: "Boolean"
                }
            },
            matchPatterns: {
                serializedName: "matchPatterns",
                type: {
                    name: "Sequence",
                    element: {
                        type: {
                            name: "Composite",
                            className: "RequestMatchPattern"
                        }
                    }
                }
            }
        }
    }
};
export const RequestMatchPattern = {
    type: {
        name: "Composite",
        className: "RequestMatchPattern",
        modelProperties: {
            path: {
                serializedName: "path",
                type: {
                    name: "String"
                }
            },
            method: {
                serializedName: "method",
                type: {
                    name: "String"
                }
            }
        }
    }
};
export const QuotaLimit = {
    type: {
        name: "Composite",
        className: "QuotaLimit",
        modelProperties: {
            count: {
                serializedName: "count",
                type: {
                    name: "Number"
                }
            },
            renewalPeriod: {
                serializedName: "renewalPeriod",
                type: {
                    name: "Number"
                }
            },
            rules: {
                serializedName: "rules",
                type: {
                    name: "Sequence",
                    element: {
                        type: {
                            name: "Composite",
                            className: "ThrottlingRule"
                        }
                    }
                }
            }
        }
    }
};
export const MultiRegionSettings = {
    type: {
        name: "Composite",
        className: "MultiRegionSettings",
        modelProperties: {
            routingMethod: {
                serializedName: "routingMethod",
                type: {
                    name: "String"
                }
            },
            regions: {
                serializedName: "regions",
                type: {
                    name: "Sequence",
                    element: {
                        type: {
                            name: "Composite",
                            className: "RegionSetting"
                        }
                    }
                }
            }
        }
    }
};
export const RegionSetting = {
    type: {
        name: "Composite",
        className: "RegionSetting",
        modelProperties: {
            name: {
                serializedName: "name",
                type: {
                    name: "String"
                }
            },
            value: {
                serializedName: "value",
                type: {
                    name: "Number"
                }
            },
            customsubdomain: {
                serializedName: "customsubdomain",
                type: {
                    name: "String"
                }
            }
        }
    }
};
export const CommitmentPlanAssociation = {
    type: {
        name: "Composite",
        className: "CommitmentPlanAssociation",
        modelProperties: {
            commitmentPlanId: {
                serializedName: "commitmentPlanId",
                type: {
                    name: "String"
                }
            },
            commitmentPlanLocation: {
                serializedName: "commitmentPlanLocation",
                type: {
                    name: "String"
                }
            }
        }
    }
};
export const AbusePenalty = {
    type: {
        name: "Composite",
        className: "AbusePenalty",
        modelProperties: {
            action: {
                serializedName: "action",
                type: {
                    name: "String"
                }
            },
            rateLimitPercentage: {
                serializedName: "rateLimitPercentage",
                type: {
                    name: "Number"
                }
            },
            expiration: {
                serializedName: "expiration",
                type: {
                    name: "DateTime"
                }
            }
        }
    }
};
export const ErrorResponse = {
    type: {
        name: "Composite",
        className: "ErrorResponse",
        modelProperties: {
            error: {
                serializedName: "error",
                type: {
                    name: "Composite",
                    className: "ErrorDetail"
                }
            }
        }
    }
};
export const ErrorDetail = {
    type: {
        name: "Composite",
        className: "ErrorDetail",
        modelProperties: {
            code: {
                serializedName: "code",
                readOnly: true,
                type: {
                    name: "String"
                }
            },
            message: {
                serializedName: "message",
                readOnly: true,
                type: {
                    name: "String"
                }
            },
            target: {
                serializedName: "target",
                readOnly: true,
                type: {
                    name: "String"
                }
            },
            details: {
                serializedName: "details",
                readOnly: true,
                type: {
                    name: "Sequence",
                    element: {
                        type: {
                            name: "Composite",
                            className: "ErrorDetail"
                        }
                    }
                }
            },
            additionalInfo: {
                serializedName: "additionalInfo",
                readOnly: true,
                type: {
                    name: "Sequence",
                    element: {
                        type: {
                            name: "Composite",
                            className: "ErrorAdditionalInfo"
                        }
                    }
                }
            }
        }
    }
};
export const ErrorAdditionalInfo = {
    type: {
        name: "Composite",
        className: "ErrorAdditionalInfo",
        modelProperties: {
            type: {
                serializedName: "type",
                readOnly: true,
                type: {
                    name: "String"
                }
            },
            info: {
                serializedName: "info",
                readOnly: true,
                type: {
                    name: "Dictionary",
                    value: { type: { name: "any" } }
                }
            }
        }
    }
};
export const AccountListResult = {
    type: {
        name: "Composite",
        className: "AccountListResult",
        modelProperties: {
            nextLink: {
                serializedName: "nextLink",
                type: {
                    name: "String"
                }
            },
            value: {
                serializedName: "value",
                readOnly: true,
                type: {
                    name: "Sequence",
                    element: {
                        type: {
                            name: "Composite",
                            className: "Account"
                        }
                    }
                }
            }
        }
    }
};
export const ApiKeys = {
    type: {
        name: "Composite",
        className: "ApiKeys",
        modelProperties: {
            key1: {
                serializedName: "key1",
                type: {
                    name: "String"
                }
            },
            key2: {
                serializedName: "key2",
                type: {
                    name: "String"
                }
            }
        }
    }
};
export const RegenerateKeyParameters = {
    type: {
        name: "Composite",
        className: "RegenerateKeyParameters",
        modelProperties: {
            keyName: {
                serializedName: "keyName",
                required: true,
                type: {
                    name: "Enum",
                    allowedValues: ["Key1", "Key2"]
                }
            }
        }
    }
};
export const ResourceSkuListResult = {
    type: {
        name: "Composite",
        className: "ResourceSkuListResult",
        modelProperties: {
            value: {
                serializedName: "value",
                required: true,
                type: {
                    name: "Sequence",
                    element: {
                        type: {
                            name: "Composite",
                            className: "ResourceSku"
                        }
                    }
                }
            },
            nextLink: {
                serializedName: "nextLink",
                type: {
                    name: "String"
                }
            }
        }
    }
};
export const ResourceSku = {
    type: {
        name: "Composite",
        className: "ResourceSku",
        modelProperties: {
            resourceType: {
                serializedName: "resourceType",
                type: {
                    name: "String"
                }
            },
            name: {
                serializedName: "name",
                type: {
                    name: "String"
                }
            },
            tier: {
                serializedName: "tier",
                type: {
                    name: "String"
                }
            },
            kind: {
                serializedName: "kind",
                type: {
                    name: "String"
                }
            },
            locations: {
                serializedName: "locations",
                type: {
                    name: "Sequence",
                    element: {
                        type: {
                            name: "String"
                        }
                    }
                }
            },
            restrictions: {
                serializedName: "restrictions",
                type: {
                    name: "Sequence",
                    element: {
                        type: {
                            name: "Composite",
                            className: "ResourceSkuRestrictions"
                        }
                    }
                }
            }
        }
    }
};
export const ResourceSkuRestrictions = {
    type: {
        name: "Composite",
        className: "ResourceSkuRestrictions",
        modelProperties: {
            type: {
                serializedName: "type",
                type: {
                    name: "Enum",
                    allowedValues: ["Location", "Zone"]
                }
            },
            values: {
                serializedName: "values",
                type: {
                    name: "Sequence",
                    element: {
                        type: {
                            name: "String"
                        }
                    }
                }
            },
            restrictionInfo: {
                serializedName: "restrictionInfo",
                type: {
                    name: "Composite",
                    className: "ResourceSkuRestrictionInfo"
                }
            },
            reasonCode: {
                serializedName: "reasonCode",
                type: {
                    name: "String"
                }
            }
        }
    }
};
export const ResourceSkuRestrictionInfo = {
    type: {
        name: "Composite",
        className: "ResourceSkuRestrictionInfo",
        modelProperties: {
            locations: {
                serializedName: "locations",
                type: {
                    name: "Sequence",
                    element: {
                        type: {
                            name: "String"
                        }
                    }
                }
            },
            zones: {
                serializedName: "zones",
                type: {
                    name: "Sequence",
                    element: {
                        type: {
                            name: "String"
                        }
                    }
                }
            }
        }
    }
};
export const AccountSkuListResult = {
    type: {
        name: "Composite",
        className: "AccountSkuListResult",
        modelProperties: {
            value: {
                serializedName: "value",
                type: {
                    name: "Sequence",
                    element: {
                        type: {
                            name: "Composite",
                            className: "AccountSku"
                        }
                    }
                }
            }
        }
    }
};
export const AccountSku = {
    type: {
        name: "Composite",
        className: "AccountSku",
        modelProperties: {
            resourceType: {
                serializedName: "resourceType",
                type: {
                    name: "String"
                }
            },
            sku: {
                serializedName: "sku",
                type: {
                    name: "Composite",
                    className: "Sku"
                }
            }
        }
    }
};
export const UsageListResult = {
    type: {
        name: "Composite",
        className: "UsageListResult",
        modelProperties: {
            nextLink: {
                serializedName: "nextLink",
                type: {
                    name: "String"
                }
            },
            value: {
                serializedName: "value",
                type: {
                    name: "Sequence",
                    element: {
                        type: {
                            name: "Composite",
                            className: "Usage"
                        }
                    }
                }
            }
        }
    }
};
export const Usage = {
    type: {
        name: "Composite",
        className: "Usage",
        modelProperties: {
            unit: {
                serializedName: "unit",
                type: {
                    name: "String"
                }
            },
            name: {
                serializedName: "name",
                type: {
                    name: "Composite",
                    className: "MetricName"
                }
            },
            quotaPeriod: {
                serializedName: "quotaPeriod",
                type: {
                    name: "String"
                }
            },
            limit: {
                serializedName: "limit",
                type: {
                    name: "Number"
                }
            },
            currentValue: {
                serializedName: "currentValue",
                type: {
                    name: "Number"
                }
            },
            nextResetTime: {
                serializedName: "nextResetTime",
                type: {
                    name: "String"
                }
            },
            status: {
                serializedName: "status",
                type: {
                    name: "String"
                }
            }
        }
    }
};
export const MetricName = {
    type: {
        name: "Composite",
        className: "MetricName",
        modelProperties: {
            value: {
                serializedName: "value",
                type: {
                    name: "String"
                }
            },
            localizedValue: {
                serializedName: "localizedValue",
                type: {
                    name: "String"
                }
            }
        }
    }
};
export const AccountModelListResult = {
    type: {
        name: "Composite",
        className: "AccountModelListResult",
        modelProperties: {
            nextLink: {
                serializedName: "nextLink",
                type: {
                    name: "String"
                }
            },
            value: {
                serializedName: "value",
                type: {
                    name: "Sequence",
                    element: {
                        type: {
                            name: "Composite",
                            className: "AccountModel"
                        }
                    }
                }
            }
        }
    }
};
export const DeploymentModel = {
    type: {
        name: "Composite",
        className: "DeploymentModel",
        modelProperties: {
            format: {
                serializedName: "format",
                type: {
                    name: "String"
                }
            },
            name: {
                serializedName: "name",
                type: {
                    name: "String"
                }
            },
            version: {
                serializedName: "version",
                type: {
                    name: "String"
                }
            },
            source: {
                serializedName: "source",
                type: {
                    name: "String"
                }
            },
            callRateLimit: {
                serializedName: "callRateLimit",
                type: {
                    name: "Composite",
                    className: "CallRateLimit"
                }
            }
        }
    }
};
export const ModelSku = {
    type: {
        name: "Composite",
        className: "ModelSku",
        modelProperties: {
            name: {
                serializedName: "name",
                type: {
                    name: "String"
                }
            },
            usageName: {
                serializedName: "usageName",
                type: {
                    name: "String"
                }
            },
            deprecationDate: {
                serializedName: "deprecationDate",
                type: {
                    name: "DateTime"
                }
            },
            capacity: {
                serializedName: "capacity",
                type: {
                    name: "Composite",
                    className: "CapacityConfig"
                }
            },
            rateLimits: {
                serializedName: "rateLimits",
                type: {
                    name: "Sequence",
                    element: {
                        type: {
                            name: "Composite",
                            className: "CallRateLimit"
                        }
                    }
                }
            }
        }
    }
};
export const CapacityConfig = {
    type: {
        name: "Composite",
        className: "CapacityConfig",
        modelProperties: {
            minimum: {
                serializedName: "minimum",
                type: {
                    name: "Number"
                }
            },
            maximum: {
                serializedName: "maximum",
                type: {
                    name: "Number"
                }
            },
            step: {
                serializedName: "step",
                type: {
                    name: "Number"
                }
            },
            default: {
                serializedName: "default",
                type: {
                    name: "Number"
                }
            }
        }
    }
};
export const ModelDeprecationInfo = {
    type: {
        name: "Composite",
        className: "ModelDeprecationInfo",
        modelProperties: {
            fineTune: {
                serializedName: "fineTune",
                type: {
                    name: "String"
                }
            },
            inference: {
                serializedName: "inference",
                type: {
                    name: "String"
                }
            }
        }
    }
};
export const OperationListResult = {
    type: {
        name: "Composite",
        className: "OperationListResult",
        modelProperties: {
            value: {
                serializedName: "value",
                readOnly: true,
                type: {
                    name: "Sequence",
                    element: {
                        type: {
                            name: "Composite",
                            className: "Operation"
                        }
                    }
                }
            },
            nextLink: {
                serializedName: "nextLink",
                readOnly: true,
                type: {
                    name: "String"
                }
            }
        }
    }
};
export const Operation = {
    type: {
        name: "Composite",
        className: "Operation",
        modelProperties: {
            name: {
                serializedName: "name",
                readOnly: true,
                type: {
                    name: "String"
                }
            },
            isDataAction: {
                serializedName: "isDataAction",
                readOnly: true,
                type: {
                    name: "Boolean"
                }
            },
            display: {
                serializedName: "display",
                type: {
                    name: "Composite",
                    className: "OperationDisplay"
                }
            },
            origin: {
                serializedName: "origin",
                readOnly: true,
                type: {
                    name: "String"
                }
            },
            actionType: {
                serializedName: "actionType",
                readOnly: true,
                type: {
                    name: "String"
                }
            }
        }
    }
};
export const OperationDisplay = {
    type: {
        name: "Composite",
        className: "OperationDisplay",
        modelProperties: {
            provider: {
                serializedName: "provider",
                readOnly: true,
                type: {
                    name: "String"
                }
            },
            resource: {
                serializedName: "resource",
                readOnly: true,
                type: {
                    name: "String"
                }
            },
            operation: {
                serializedName: "operation",
                readOnly: true,
                type: {
                    name: "String"
                }
            },
            description: {
                serializedName: "description",
                readOnly: true,
                type: {
                    name: "String"
                }
            }
        }
    }
};
export const CheckSkuAvailabilityParameter = {
    type: {
        name: "Composite",
        className: "CheckSkuAvailabilityParameter",
        modelProperties: {
            skus: {
                serializedName: "skus",
                required: true,
                type: {
                    name: "Sequence",
                    element: {
                        type: {
                            name: "String"
                        }
                    }
                }
            },
            kind: {
                serializedName: "kind",
                required: true,
                type: {
                    name: "String"
                }
            },
            typeParam: {
                serializedName: "type",
                required: true,
                type: {
                    name: "String"
                }
            }
        }
    }
};
export const SkuAvailabilityListResult = {
    type: {
        name: "Composite",
        className: "SkuAvailabilityListResult",
        modelProperties: {
            value: {
                serializedName: "value",
                type: {
                    name: "Sequence",
                    element: {
                        type: {
                            name: "Composite",
                            className: "SkuAvailability"
                        }
                    }
                }
            }
        }
    }
};
export const SkuAvailability = {
    type: {
        name: "Composite",
        className: "SkuAvailability",
        modelProperties: {
            kind: {
                serializedName: "kind",
                type: {
                    name: "String"
                }
            },
            type: {
                serializedName: "type",
                type: {
                    name: "String"
                }
            },
            skuName: {
                serializedName: "skuName",
                type: {
                    name: "String"
                }
            },
            skuAvailable: {
                serializedName: "skuAvailable",
                type: {
                    name: "Boolean"
                }
            },
            reason: {
                serializedName: "reason",
                type: {
                    name: "String"
                }
            },
            message: {
                serializedName: "message",
                type: {
                    name: "String"
                }
            }
        }
    }
};
export const CommitmentTierListResult = {
    type: {
        name: "Composite",
        className: "CommitmentTierListResult",
        modelProperties: {
            nextLink: {
                serializedName: "nextLink",
                type: {
                    name: "String"
                }
            },
            value: {
                serializedName: "value",
                readOnly: true,
                type: {
                    name: "Sequence",
                    element: {
                        type: {
                            name: "Composite",
                            className: "CommitmentTier"
                        }
                    }
                }
            }
        }
    }
};
export const CommitmentTier = {
    type: {
        name: "Composite",
        className: "CommitmentTier",
        modelProperties: {
            kind: {
                serializedName: "kind",
                type: {
                    name: "String"
                }
            },
            skuName: {
                serializedName: "skuName",
                type: {
                    name: "String"
                }
            },
            hostingModel: {
                serializedName: "hostingModel",
                type: {
                    name: "String"
                }
            },
            planType: {
                serializedName: "planType",
                type: {
                    name: "String"
                }
            },
            tier: {
                serializedName: "tier",
                type: {
                    name: "String"
                }
            },
            maxCount: {
                serializedName: "maxCount",
                type: {
                    name: "Number"
                }
            },
            quota: {
                serializedName: "quota",
                type: {
                    name: "Composite",
                    className: "CommitmentQuota"
                }
            },
            cost: {
                serializedName: "cost",
                type: {
                    name: "Composite",
                    className: "CommitmentCost"
                }
            }
        }
    }
};
export const CommitmentQuota = {
    type: {
        name: "Composite",
        className: "CommitmentQuota",
        modelProperties: {
            quantity: {
                serializedName: "quantity",
                type: {
                    name: "Number"
                }
            },
            unit: {
                serializedName: "unit",
                type: {
                    name: "String"
                }
            }
        }
    }
};
export const CommitmentCost = {
    type: {
        name: "Composite",
        className: "CommitmentCost",
        modelProperties: {
            commitmentMeterId: {
                serializedName: "commitmentMeterId",
                type: {
                    name: "String"
                }
            },
            overageMeterId: {
                serializedName: "overageMeterId",
                type: {
                    name: "String"
                }
            }
        }
    }
};
export const ModelListResult = {
    type: {
        name: "Composite",
        className: "ModelListResult",
        modelProperties: {
            nextLink: {
                serializedName: "nextLink",
                type: {
                    name: "String"
                }
            },
            value: {
                serializedName: "value",
                type: {
                    name: "Sequence",
                    element: {
                        type: {
                            name: "Composite",
                            className: "Model"
                        }
                    }
                }
            }
        }
    }
};
export const Model = {
    type: {
        name: "Composite",
        className: "Model",
        modelProperties: {
            model: {
                serializedName: "model",
                type: {
                    name: "Composite",
                    className: "AccountModel"
                }
            },
            kind: {
                serializedName: "kind",
                type: {
                    name: "String"
                }
            },
            skuName: {
                serializedName: "skuName",
                type: {
                    name: "String"
                }
            }
        }
    }
};
export const CheckDomainAvailabilityParameter = {
    type: {
        name: "Composite",
        className: "CheckDomainAvailabilityParameter",
        modelProperties: {
            subdomainName: {
                serializedName: "subdomainName",
                required: true,
                type: {
                    name: "String"
                }
            },
            typeParam: {
                serializedName: "type",
                required: true,
                type: {
                    name: "String"
                }
            },
            kind: {
                serializedName: "kind",
                type: {
                    name: "String"
                }
            }
        }
    }
};
export const DomainAvailability = {
    type: {
        name: "Composite",
        className: "DomainAvailability",
        modelProperties: {
            isSubdomainAvailable: {
                serializedName: "isSubdomainAvailable",
                type: {
                    name: "Boolean"
                }
            },
            reason: {
                serializedName: "reason",
                type: {
                    name: "String"
                }
            },
            subdomainName: {
                serializedName: "subdomainName",
                type: {
                    name: "String"
                }
            },
            type: {
                serializedName: "type",
                type: {
                    name: "String"
                }
            },
            kind: {
                serializedName: "kind",
                type: {
                    name: "String"
                }
            }
        }
    }
};
export const PrivateEndpointConnectionListResult = {
    type: {
        name: "Composite",
        className: "PrivateEndpointConnectionListResult",
        modelProperties: {
            value: {
                serializedName: "value",
                type: {
                    name: "Sequence",
                    element: {
                        type: {
                            name: "Composite",
                            className: "PrivateEndpointConnection"
                        }
                    }
                }
            }
        }
    }
};
export const PrivateLinkResourceListResult = {
    type: {
        name: "Composite",
        className: "PrivateLinkResourceListResult",
        modelProperties: {
            value: {
                serializedName: "value",
                type: {
                    name: "Sequence",
                    element: {
                        type: {
                            name: "Composite",
                            className: "PrivateLinkResource"
                        }
                    }
                }
            }
        }
    }
};
export const PrivateLinkResourceProperties = {
    type: {
        name: "Composite",
        className: "PrivateLinkResourceProperties",
        modelProperties: {
            groupId: {
                serializedName: "groupId",
                readOnly: true,
                type: {
                    name: "String"
                }
            },
            requiredMembers: {
                serializedName: "requiredMembers",
                readOnly: true,
                type: {
                    name: "Sequence",
                    element: {
                        type: {
                            name: "String"
                        }
                    }
                }
            },
            requiredZoneNames: {
                serializedName: "requiredZoneNames",
                type: {
                    name: "Sequence",
                    element: {
                        type: {
                            name: "String"
                        }
                    }
                }
            },
            displayName: {
                serializedName: "displayName",
                readOnly: true,
                type: {
                    name: "String"
                }
            }
        }
    }
};
export const DeploymentListResult = {
    type: {
        name: "Composite",
        className: "DeploymentListResult",
        modelProperties: {
            nextLink: {
                serializedName: "nextLink",
                type: {
                    name: "String"
                }
            },
            value: {
                serializedName: "value",
                readOnly: true,
                type: {
                    name: "Sequence",
                    element: {
                        type: {
                            name: "Composite",
                            className: "Deployment"
                        }
                    }
                }
            }
        }
    }
};
export const DeploymentProperties = {
    type: {
        name: "Composite",
        className: "DeploymentProperties",
        modelProperties: {
            provisioningState: {
                serializedName: "provisioningState",
                readOnly: true,
                type: {
                    name: "String"
                }
            },
            model: {
                serializedName: "model",
                type: {
                    name: "Composite",
                    className: "DeploymentModel"
                }
            },
            scaleSettings: {
                serializedName: "scaleSettings",
                type: {
                    name: "Composite",
                    className: "DeploymentScaleSettings"
                }
            },
            capabilities: {
                serializedName: "capabilities",
                readOnly: true,
                type: {
                    name: "Dictionary",
                    value: { type: { name: "String" } }
                }
            },
            raiPolicyName: {
                serializedName: "raiPolicyName",
                type: {
                    name: "String"
                }
            },
            callRateLimit: {
                serializedName: "callRateLimit",
                type: {
                    name: "Composite",
                    className: "CallRateLimit"
                }
            },
            rateLimits: {
                serializedName: "rateLimits",
                readOnly: true,
                type: {
                    name: "Sequence",
                    element: {
                        type: {
                            name: "Composite",
                            className: "ThrottlingRule"
                        }
                    }
                }
            },
            versionUpgradeOption: {
                serializedName: "versionUpgradeOption",
                type: {
                    name: "String"
                }
            }
        }
    }
};
export const DeploymentScaleSettings = {
    type: {
        name: "Composite",
        className: "DeploymentScaleSettings",
        modelProperties: {
            scaleType: {
                serializedName: "scaleType",
                type: {
                    name: "String"
                }
            },
            capacity: {
                serializedName: "capacity",
                type: {
                    name: "Number"
                }
            },
            activeCapacity: {
                serializedName: "activeCapacity",
                readOnly: true,
                type: {
                    name: "Number"
                }
            }
        }
    }
};
export const CommitmentPlanListResult = {
    type: {
        name: "Composite",
        className: "CommitmentPlanListResult",
        modelProperties: {
            nextLink: {
                serializedName: "nextLink",
                type: {
                    name: "String"
                }
            },
            value: {
                serializedName: "value",
                readOnly: true,
                type: {
                    name: "Sequence",
                    element: {
                        type: {
                            name: "Composite",
                            className: "CommitmentPlan"
                        }
                    }
                }
            }
        }
    }
};
export const CommitmentPlanProperties = {
    type: {
        name: "Composite",
        className: "CommitmentPlanProperties",
        modelProperties: {
            provisioningState: {
                serializedName: "provisioningState",
                readOnly: true,
                type: {
                    name: "String"
                }
            },
            commitmentPlanGuid: {
                serializedName: "commitmentPlanGuid",
                type: {
                    name: "String"
                }
            },
            hostingModel: {
                serializedName: "hostingModel",
                type: {
                    name: "String"
                }
            },
            planType: {
                serializedName: "planType",
                type: {
                    name: "String"
                }
            },
            current: {
                serializedName: "current",
                type: {
                    name: "Composite",
                    className: "CommitmentPeriod"
                }
            },
            autoRenew: {
                serializedName: "autoRenew",
                type: {
                    name: "Boolean"
                }
            },
            next: {
                serializedName: "next",
                type: {
                    name: "Composite",
                    className: "CommitmentPeriod"
                }
            },
            last: {
                serializedName: "last",
                type: {
                    name: "Composite",
                    className: "CommitmentPeriod"
                }
            },
            provisioningIssues: {
                serializedName: "provisioningIssues",
                readOnly: true,
                type: {
                    name: "Sequence",
                    element: {
                        type: {
                            name: "String"
                        }
                    }
                }
            }
        }
    }
};
export const CommitmentPeriod = {
    type: {
        name: "Composite",
        className: "CommitmentPeriod",
        modelProperties: {
            tier: {
                serializedName: "tier",
                type: {
                    name: "String"
                }
            },
            count: {
                serializedName: "count",
                type: {
                    name: "Number"
                }
            },
            quota: {
                serializedName: "quota",
                type: {
                    name: "Composite",
                    className: "CommitmentQuota"
                }
            },
            startDate: {
                serializedName: "startDate",
                readOnly: true,
                type: {
                    name: "String"
                }
            },
            endDate: {
                serializedName: "endDate",
                readOnly: true,
                type: {
                    name: "String"
                }
            }
        }
    }
};
export const PatchResourceTags = {
    type: {
        name: "Composite",
        className: "PatchResourceTags",
        modelProperties: {
            tags: {
                serializedName: "tags",
                type: {
                    name: "Dictionary",
                    value: { type: { name: "String" } }
                }
            }
        }
    }
};
export const CommitmentPlanAccountAssociationListResult = {
    type: {
        name: "Composite",
        className: "CommitmentPlanAccountAssociationListResult",
        modelProperties: {
            nextLink: {
                serializedName: "nextLink",
                type: {
                    name: "String"
                }
            },
            value: {
                serializedName: "value",
                readOnly: true,
                type: {
                    name: "Sequence",
                    element: {
                        type: {
                            name: "Composite",
                            className: "CommitmentPlanAccountAssociation"
                        }
                    }
                }
            }
        }
    }
};
export const AzureEntityResource = {
    type: {
        name: "Composite",
        className: "AzureEntityResource",
        modelProperties: Object.assign(Object.assign({}, Resource.type.modelProperties), { etag: {
                serializedName: "etag",
                readOnly: true,
                type: {
                    name: "String"
                }
            } })
    }
};
export const PrivateLinkResource = {
    type: {
        name: "Composite",
        className: "PrivateLinkResource",
        modelProperties: Object.assign(Object.assign({}, Resource.type.modelProperties), { properties: {
                serializedName: "properties",
                type: {
                    name: "Composite",
                    className: "PrivateLinkResourceProperties"
                }
            } })
    }
};
export const ProxyResource = {
    type: {
        name: "Composite",
        className: "ProxyResource",
        modelProperties: Object.assign({}, Resource.type.modelProperties)
    }
};
export const AccountModel = {
    type: {
        name: "Composite",
        className: "AccountModel",
        modelProperties: Object.assign(Object.assign({}, DeploymentModel.type.modelProperties), { baseModel: {
                serializedName: "baseModel",
                type: {
                    name: "Composite",
                    className: "DeploymentModel"
                }
            }, isDefaultVersion: {
                serializedName: "isDefaultVersion",
                type: {
                    name: "Boolean"
                }
            }, skus: {
                serializedName: "skus",
                type: {
                    name: "Sequence",
                    element: {
                        type: {
                            name: "Composite",
                            className: "ModelSku"
                        }
                    }
                }
            }, maxCapacity: {
                serializedName: "maxCapacity",
                type: {
                    name: "Number"
                }
            }, capabilities: {
                serializedName: "capabilities",
                type: {
                    name: "Dictionary",
                    value: { type: { name: "String" } }
                }
            }, finetuneCapabilities: {
                serializedName: "finetuneCapabilities",
                type: {
                    name: "Dictionary",
                    value: { type: { name: "String" } }
                }
            }, deprecation: {
                serializedName: "deprecation",
                type: {
                    name: "Composite",
                    className: "ModelDeprecationInfo"
                }
            }, lifecycleStatus: {
                serializedName: "lifecycleStatus",
                type: {
                    name: "String"
                }
            }, systemData: {
                serializedName: "systemData",
                type: {
                    name: "Composite",
                    className: "SystemData"
                }
            } })
    }
};
export const PatchResourceTagsAndSku = {
    type: {
        name: "Composite",
        className: "PatchResourceTagsAndSku",
        modelProperties: Object.assign(Object.assign({}, PatchResourceTags.type.modelProperties), { sku: {
                serializedName: "sku",
                type: {
                    name: "Composite",
                    className: "Sku"
                }
            } })
    }
};
export const PrivateEndpointConnection = {
    type: {
        name: "Composite",
        className: "PrivateEndpointConnection",
        modelProperties: Object.assign(Object.assign({}, AzureEntityResource.type.modelProperties), { properties: {
                serializedName: "properties",
                type: {
                    name: "Composite",
                    className: "PrivateEndpointConnectionProperties"
                }
            }, systemData: {
                serializedName: "systemData",
                type: {
                    name: "Composite",
                    className: "SystemData"
                }
            }, location: {
                serializedName: "location",
                type: {
                    name: "String"
                }
            } })
    }
};
export const Account = {
    type: {
        name: "Composite",
        className: "Account",
        modelProperties: Object.assign(Object.assign({}, AzureEntityResource.type.modelProperties), { kind: {
                serializedName: "kind",
                type: {
                    name: "String"
                }
            }, sku: {
                serializedName: "sku",
                type: {
                    name: "Composite",
                    className: "Sku"
                }
            }, identity: {
                serializedName: "identity",
                type: {
                    name: "Composite",
                    className: "Identity"
                }
            }, systemData: {
                serializedName: "systemData",
                type: {
                    name: "Composite",
                    className: "SystemData"
                }
            }, tags: {
                serializedName: "tags",
                type: {
                    name: "Dictionary",
                    value: { type: { name: "String" } }
                }
            }, location: {
                serializedName: "location",
                type: {
                    name: "String"
                }
            }, properties: {
                serializedName: "properties",
                type: {
                    name: "Composite",
                    className: "AccountProperties"
                }
            } })
    }
};
export const Deployment = {
    type: {
        name: "Composite",
        className: "Deployment",
        modelProperties: Object.assign(Object.assign({}, ProxyResource.type.modelProperties), { sku: {
                serializedName: "sku",
                type: {
                    name: "Composite",
                    className: "Sku"
                }
            }, systemData: {
                serializedName: "systemData",
                type: {
                    name: "Composite",
                    className: "SystemData"
                }
            }, etag: {
                serializedName: "etag",
                readOnly: true,
                type: {
                    name: "String"
                }
            }, properties: {
                serializedName: "properties",
                type: {
                    name: "Composite",
                    className: "DeploymentProperties"
                }
            } })
    }
};
export const CommitmentPlan = {
    type: {
        name: "Composite",
        className: "CommitmentPlan",
        modelProperties: Object.assign(Object.assign({}, ProxyResource.type.modelProperties), { systemData: {
                serializedName: "systemData",
                type: {
                    name: "Composite",
                    className: "SystemData"
                }
            }, etag: {
                serializedName: "etag",
                readOnly: true,
                type: {
                    name: "String"
                }
            }, kind: {
                serializedName: "kind",
                type: {
                    name: "String"
                }
            }, sku: {
                serializedName: "sku",
                type: {
                    name: "Composite",
                    className: "Sku"
                }
            }, tags: {
                serializedName: "tags",
                type: {
                    name: "Dictionary",
                    value: { type: { name: "String" } }
                }
            }, location: {
                serializedName: "location",
                type: {
                    name: "String"
                }
            }, properties: {
                serializedName: "properties",
                type: {
                    name: "Composite",
                    className: "CommitmentPlanProperties"
                }
            } })
    }
};
export const CommitmentPlanAccountAssociation = {
    type: {
        name: "Composite",
        className: "CommitmentPlanAccountAssociation",
        modelProperties: Object.assign(Object.assign({}, ProxyResource.type.modelProperties), { systemData: {
                serializedName: "systemData",
                type: {
                    name: "Composite",
                    className: "SystemData"
                }
            }, etag: {
                serializedName: "etag",
                readOnly: true,
                type: {
                    name: "String"
                }
            }, accountId: {
                serializedName: "properties.accountId",
                type: {
                    name: "String"
                }
            } })
    }
};
export const CommitmentPlansUpdatePlanHeaders = {
    type: {
        name: "Composite",
        className: "CommitmentPlansUpdatePlanHeaders",
        modelProperties: {
            location: {
                serializedName: "location",
                type: {
                    name: "String"
                }
            }
        }
    }
};
export const CommitmentPlansDeletePlanHeaders = {
    type: {
        name: "Composite",
        className: "CommitmentPlansDeletePlanHeaders",
        modelProperties: {
            location: {
                serializedName: "location",
                type: {
                    name: "String"
                }
            }
        }
    }
};
export const CommitmentPlansDeleteAssociationHeaders = {
    type: {
        name: "Composite",
        className: "CommitmentPlansDeleteAssociationHeaders",
        modelProperties: {
            location: {
                serializedName: "location",
                type: {
                    name: "String"
                }
            }
        }
    }
};
