// @aws-sdk/client-mturk/dist-es/models/models_0.js
import { MTurkServiceException as __BaseException } from "./MTurkServiceException";
export class RequestError extends __BaseException {
    constructor(opts) {
        super({
            name: "RequestError",
            $fault: "client",
            ...opts,
        });
        this.name = "RequestError";
        this.$fault = "client";
        Object.setPrototypeOf(this, RequestError.prototype);
        this.Message = opts.Message;
        this.TurkErrorCode = opts.TurkErrorCode;
    }
}
export class ServiceFault extends __BaseException {
    constructor(opts) {
        super({
            name: "ServiceFault",
            $fault: "server",
            ...opts,
        });
        this.name = "ServiceFault";
        this.$fault = "server";
        Object.setPrototypeOf(this, ServiceFault.prototype);
        this.Message = opts.Message;
        this.TurkErrorCode = opts.TurkErrorCode;
    }
}
export const AssignmentStatus = {
    Approved: "Approved",
    Rejected: "Rejected",
    Submitted: "Submitted",
};
export const Comparator = {
    DoesNotExist: "DoesNotExist",
    EqualTo: "EqualTo",
    Exists: "Exists",
    GreaterThan: "GreaterThan",
    GreaterThanOrEqualTo: "GreaterThanOrEqualTo",
    In: "In",
    LessThan: "LessThan",
    LessThanOrEqualTo: "LessThanOrEqualTo",
    NotEqualTo: "NotEqualTo",
    NotIn: "NotIn",
};
export const HITAccessActions = {
    Accept: "Accept",
    DiscoverPreviewAndAccept: "DiscoverPreviewAndAccept",
    PreviewAndAccept: "PreviewAndAccept",
};
export const HITReviewStatus = {
    MarkedForReview: "MarkedForReview",
    NotReviewed: "NotReviewed",
    ReviewedAppropriate: "ReviewedAppropriate",
    ReviewedInappropriate: "ReviewedInappropriate",
};
export const HITStatus = {
    Assignable: "Assignable",
    Disposed: "Disposed",
    Reviewable: "Reviewable",
    Reviewing: "Reviewing",
    Unassignable: "Unassignable",
};
export const QualificationTypeStatus = {
    Active: "Active",
    Inactive: "Inactive",
};
export const EventType = {
    AssignmentAbandoned: "AssignmentAbandoned",
    AssignmentAccepted: "AssignmentAccepted",
    AssignmentApproved: "AssignmentApproved",
    AssignmentRejected: "AssignmentRejected",
    AssignmentReturned: "AssignmentReturned",
    AssignmentSubmitted: "AssignmentSubmitted",
    HITCreated: "HITCreated",
    HITDisposed: "HITDisposed",
    HITExpired: "HITExpired",
    HITExtended: "HITExtended",
    HITReviewable: "HITReviewable",
    Ping: "Ping",
};
export const QualificationStatus = {
    Granted: "Granted",
    Revoked: "Revoked",
};
export const ReviewableHITStatus = {
    Reviewable: "Reviewable",
    Reviewing: "Reviewing",
};
export const ReviewPolicyLevel = {
    Assignment: "Assignment",
    HIT: "HIT",
};
export const ReviewActionStatus = {
    Cancelled: "Cancelled",
    Failed: "Failed",
    Intended: "Intended",
    Succeeded: "Succeeded",
};
export const NotifyWorkersFailureCode = {
    HardFailure: "HardFailure",
    SoftFailure: "SoftFailure",
};
export const NotificationTransport = {
    Email: "Email",
    SNS: "SNS",
    SQS: "SQS",
};
