// @azure/arm-consumption/dist-esm/src/operations/lotsOperations.js
/*
 * Copyright (c) Microsoft Corporation.
 * Licensed under the MIT License.
 *
 * Code generated by Microsoft (R) AutoRest Code Generator.
 * Changes may cause incorrect behavior and will be lost if the code is regenerated.
 */
import { __asyncDelegator, __asyncGenerator, __asyncValues, __await } from "tslib";
import { setContinuationToken } from "../pagingHelper";
import * as coreClient from "@azure/core-client";
import * as Mappers from "../models/mappers";
import * as Parameters from "../models/parameters";
/// <reference lib="esnext.asynciterable" />
/** Class containing LotsOperations operations. */
export class LotsOperationsImpl {
    /**
     * Initialize a new instance of the class LotsOperations class.
     * @param client Reference to the service client
     */
    constructor(client) {
        this.client = client;
    }
    /**
     * Lists all Azure credits and Microsoft Azure consumption commitments for a billing account or a
     * billing profile. Microsoft Azure consumption commitments are only supported for the billing account
     * scope.
     * @param billingAccountId BillingAccount ID
     * @param billingProfileId Azure Billing Profile ID.
     * @param options The options parameters.
     */
    listByBillingProfile(billingAccountId, billingProfileId, options) {
        const iter = this.listByBillingProfilePagingAll(billingAccountId, billingProfileId, options);
        return {
            next() {
                return iter.next();
            },
            [Symbol.asyncIterator]() {
                return this;
            },
            byPage: (settings) => {
                if (settings === null || settings === void 0 ? void 0 : settings.maxPageSize) {
                    throw new Error("maxPageSize is not supported by this operation.");
                }
                return this.listByBillingProfilePagingPage(billingAccountId, billingProfileId, options, settings);
            }
        };
    }
    listByBillingProfilePagingPage(billingAccountId, billingProfileId, options, settings) {
        return __asyncGenerator(this, arguments, function* listByBillingProfilePagingPage_1() {
            let result;
            let continuationToken = settings === null || settings === void 0 ? void 0 : settings.continuationToken;
            if (!continuationToken) {
                result = yield __await(this._listByBillingProfile(billingAccountId, billingProfileId, options));
                let page = result.value || [];
                continuationToken = result.nextLink;
                setContinuationToken(page, continuationToken);
                yield yield __await(page);
            }
            while (continuationToken) {
                result = yield __await(this._listByBillingProfileNext(billingAccountId, billingProfileId, continuationToken, options));
                continuationToken = result.nextLink;
                let page = result.value || [];
                setContinuationToken(page, continuationToken);
                yield yield __await(page);
            }
        });
    }
    listByBillingProfilePagingAll(billingAccountId, billingProfileId, options) {
        return __asyncGenerator(this, arguments, function* listByBillingProfilePagingAll_1() {
            var e_1, _a;
            try {
                for (var _b = __asyncValues(this.listByBillingProfilePagingPage(billingAccountId, billingProfileId, options)), _c; _c = yield __await(_b.next()), !_c.done;) {
                    const page = _c.value;
                    yield __await(yield* __asyncDelegator(__asyncValues(page)));
                }
            }
            catch (e_1_1) { e_1 = { error: e_1_1 }; }
            finally {
                try {
                    if (_c && !_c.done && (_a = _b.return)) yield __await(_a.call(_b));
                }
                finally { if (e_1) throw e_1.error; }
            }
        });
    }
    /**
     * Lists all Azure credits and Microsoft Azure consumption commitments for a billing account or a
     * billing profile. Microsoft Azure consumption commitments are only supported for the billing account
     * scope.
     * @param billingAccountId BillingAccount ID
     * @param options The options parameters.
     */
    listByBillingAccount(billingAccountId, options) {
        const iter = this.listByBillingAccountPagingAll(billingAccountId, options);
        return {
            next() {
                return iter.next();
            },
            [Symbol.asyncIterator]() {
                return this;
            },
            byPage: (settings) => {
                if (settings === null || settings === void 0 ? void 0 : settings.maxPageSize) {
                    throw new Error("maxPageSize is not supported by this operation.");
                }
                return this.listByBillingAccountPagingPage(billingAccountId, options, settings);
            }
        };
    }
    listByBillingAccountPagingPage(billingAccountId, options, settings) {
        return __asyncGenerator(this, arguments, function* listByBillingAccountPagingPage_1() {
            let result;
            let continuationToken = settings === null || settings === void 0 ? void 0 : settings.continuationToken;
            if (!continuationToken) {
                result = yield __await(this._listByBillingAccount(billingAccountId, options));
                let page = result.value || [];
                continuationToken = result.nextLink;
                setContinuationToken(page, continuationToken);
                yield yield __await(page);
            }
            while (continuationToken) {
                result = yield __await(this._listByBillingAccountNext(billingAccountId, continuationToken, options));
                continuationToken = result.nextLink;
                let page = result.value || [];
                setContinuationToken(page, continuationToken);
                yield yield __await(page);
            }
        });
    }
    listByBillingAccountPagingAll(billingAccountId, options) {
        return __asyncGenerator(this, arguments, function* listByBillingAccountPagingAll_1() {
            var e_2, _a;
            try {
                for (var _b = __asyncValues(this.listByBillingAccountPagingPage(billingAccountId, options)), _c; _c = yield __await(_b.next()), !_c.done;) {
                    const page = _c.value;
                    yield __await(yield* __asyncDelegator(__asyncValues(page)));
                }
            }
            catch (e_2_1) { e_2 = { error: e_2_1 }; }
            finally {
                try {
                    if (_c && !_c.done && (_a = _b.return)) yield __await(_a.call(_b));
                }
                finally { if (e_2) throw e_2.error; }
            }
        });
    }
    /**
     * Lists all Azure credits and Microsoft Azure consumption commitments for a billing account or a
     * billing profile. Microsoft Azure consumption commitments are only supported for the billing account
     * scope.
     * @param billingAccountId BillingAccount ID
     * @param billingProfileId Azure Billing Profile ID.
     * @param options The options parameters.
     */
    _listByBillingProfile(billingAccountId, billingProfileId, options) {
        return this.client.sendOperationRequest({ billingAccountId, billingProfileId, options }, listByBillingProfileOperationSpec);
    }
    /**
     * Lists all Azure credits and Microsoft Azure consumption commitments for a billing account or a
     * billing profile. Microsoft Azure consumption commitments are only supported for the billing account
     * scope.
     * @param billingAccountId BillingAccount ID
     * @param options The options parameters.
     */
    _listByBillingAccount(billingAccountId, options) {
        return this.client.sendOperationRequest({ billingAccountId, options }, listByBillingAccountOperationSpec);
    }
    /**
     * ListByBillingProfileNext
     * @param billingAccountId BillingAccount ID
     * @param billingProfileId Azure Billing Profile ID.
     * @param nextLink The nextLink from the previous successful call to the ListByBillingProfile method.
     * @param options The options parameters.
     */
    _listByBillingProfileNext(billingAccountId, billingProfileId, nextLink, options) {
        return this.client.sendOperationRequest({ billingAccountId, billingProfileId, nextLink, options }, listByBillingProfileNextOperationSpec);
    }
    /**
     * ListByBillingAccountNext
     * @param billingAccountId BillingAccount ID
     * @param nextLink The nextLink from the previous successful call to the ListByBillingAccount method.
     * @param options The options parameters.
     */
    _listByBillingAccountNext(billingAccountId, nextLink, options) {
        return this.client.sendOperationRequest({ billingAccountId, nextLink, options }, listByBillingAccountNextOperationSpec);
    }
}
// Operation Specifications
const serializer = coreClient.createSerializer(Mappers, /* isXml */ false);
const listByBillingProfileOperationSpec = {
    path: "/providers/Microsoft.Billing/billingAccounts/{billingAccountId}/billingProfiles/{billingProfileId}/providers/Microsoft.Consumption/lots",
    httpMethod: "GET",
    responses: {
        200: {
            bodyMapper: Mappers.Lots
        },
        default: {
            bodyMapper: Mappers.ErrorResponse
        }
    },
    queryParameters: [Parameters.apiVersion],
    urlParameters: [
        Parameters.$host,
        Parameters.billingAccountId,
        Parameters.billingProfileId
    ],
    headerParameters: [Parameters.accept],
    serializer
};
const listByBillingAccountOperationSpec = {
    path: "/providers/Microsoft.Billing/billingAccounts/{billingAccountId}/providers/Microsoft.Consumption/lots",
    httpMethod: "GET",
    responses: {
        200: {
            bodyMapper: Mappers.Lots
        },
        default: {
            bodyMapper: Mappers.ErrorResponse
        }
    },
    queryParameters: [Parameters.filter, Parameters.apiVersion],
    urlParameters: [Parameters.$host, Parameters.billingAccountId],
    headerParameters: [Parameters.accept],
    serializer
};
const listByBillingProfileNextOperationSpec = {
    path: "{nextLink}",
    httpMethod: "GET",
    responses: {
        200: {
            bodyMapper: Mappers.Lots
        },
        default: {
            bodyMapper: Mappers.ErrorResponse
        }
    },
    urlParameters: [
        Parameters.$host,
        Parameters.nextLink,
        Parameters.billingAccountId,
        Parameters.billingProfileId
    ],
    headerParameters: [Parameters.accept],
    serializer
};
const listByBillingAccountNextOperationSpec = {
    path: "{nextLink}",
    httpMethod: "GET",
    responses: {
        200: {
            bodyMapper: Mappers.Lots
        },
        default: {
            bodyMapper: Mappers.ErrorResponse
        }
    },
    urlParameters: [
        Parameters.$host,
        Parameters.nextLink,
        Parameters.billingAccountId
    ],
    headerParameters: [Parameters.accept],
    serializer
};
