// @azure/arm-consumption/dist-esm/src/operations/reservationsSummaries.js
/*
 * Copyright (c) Microsoft Corporation.
 * Licensed under the MIT License.
 *
 * Code generated by Microsoft (R) AutoRest Code Generator.
 * Changes may cause incorrect behavior and will be lost if the code is regenerated.
 */
import { __asyncDelegator, __asyncGenerator, __asyncValues, __await } from "tslib";
import { setContinuationToken } from "../pagingHelper";
import * as coreClient from "@azure/core-client";
import * as Mappers from "../models/mappers";
import * as Parameters from "../models/parameters";
/// <reference lib="esnext.asynciterable" />
/** Class containing ReservationsSummaries operations. */
export class ReservationsSummariesImpl {
    /**
     * Initialize a new instance of the class ReservationsSummaries class.
     * @param client Reference to the service client
     */
    constructor(client) {
        this.client = client;
    }
    /**
     * Lists the reservations summaries for daily or monthly grain.
     * @param reservationOrderId Order Id of the reservation
     * @param grain Can be daily or monthly
     * @param options The options parameters.
     */
    listByReservationOrder(reservationOrderId, grain, options) {
        const iter = this.listByReservationOrderPagingAll(reservationOrderId, grain, options);
        return {
            next() {
                return iter.next();
            },
            [Symbol.asyncIterator]() {
                return this;
            },
            byPage: (settings) => {
                if (settings === null || settings === void 0 ? void 0 : settings.maxPageSize) {
                    throw new Error("maxPageSize is not supported by this operation.");
                }
                return this.listByReservationOrderPagingPage(reservationOrderId, grain, options, settings);
            }
        };
    }
    listByReservationOrderPagingPage(reservationOrderId, grain, options, settings) {
        return __asyncGenerator(this, arguments, function* listByReservationOrderPagingPage_1() {
            let result;
            let continuationToken = settings === null || settings === void 0 ? void 0 : settings.continuationToken;
            if (!continuationToken) {
                result = yield __await(this._listByReservationOrder(reservationOrderId, grain, options));
                let page = result.value || [];
                continuationToken = result.nextLink;
                setContinuationToken(page, continuationToken);
                yield yield __await(page);
            }
            while (continuationToken) {
                result = yield __await(this._listByReservationOrderNext(reservationOrderId, continuationToken, options));
                continuationToken = result.nextLink;
                let page = result.value || [];
                setContinuationToken(page, continuationToken);
                yield yield __await(page);
            }
        });
    }
    listByReservationOrderPagingAll(reservationOrderId, grain, options) {
        return __asyncGenerator(this, arguments, function* listByReservationOrderPagingAll_1() {
            var e_1, _a;
            try {
                for (var _b = __asyncValues(this.listByReservationOrderPagingPage(reservationOrderId, grain, options)), _c; _c = yield __await(_b.next()), !_c.done;) {
                    const page = _c.value;
                    yield __await(yield* __asyncDelegator(__asyncValues(page)));
                }
            }
            catch (e_1_1) { e_1 = { error: e_1_1 }; }
            finally {
                try {
                    if (_c && !_c.done && (_a = _b.return)) yield __await(_a.call(_b));
                }
                finally { if (e_1) throw e_1.error; }
            }
        });
    }
    /**
     * Lists the reservations summaries for daily or monthly grain.
     * @param reservationOrderId Order Id of the reservation
     * @param reservationId Id of the reservation
     * @param grain Can be daily or monthly
     * @param options The options parameters.
     */
    listByReservationOrderAndReservation(reservationOrderId, reservationId, grain, options) {
        const iter = this.listByReservationOrderAndReservationPagingAll(reservationOrderId, reservationId, grain, options);
        return {
            next() {
                return iter.next();
            },
            [Symbol.asyncIterator]() {
                return this;
            },
            byPage: (settings) => {
                if (settings === null || settings === void 0 ? void 0 : settings.maxPageSize) {
                    throw new Error("maxPageSize is not supported by this operation.");
                }
                return this.listByReservationOrderAndReservationPagingPage(reservationOrderId, reservationId, grain, options, settings);
            }
        };
    }
    listByReservationOrderAndReservationPagingPage(reservationOrderId, reservationId, grain, options, settings) {
        return __asyncGenerator(this, arguments, function* listByReservationOrderAndReservationPagingPage_1() {
            let result;
            let continuationToken = settings === null || settings === void 0 ? void 0 : settings.continuationToken;
            if (!continuationToken) {
                result = yield __await(this._listByReservationOrderAndReservation(reservationOrderId, reservationId, grain, options));
                let page = result.value || [];
                continuationToken = result.nextLink;
                setContinuationToken(page, continuationToken);
                yield yield __await(page);
            }
            while (continuationToken) {
                result = yield __await(this._listByReservationOrderAndReservationNext(reservationOrderId, reservationId, continuationToken, options));
                continuationToken = result.nextLink;
                let page = result.value || [];
                setContinuationToken(page, continuationToken);
                yield yield __await(page);
            }
        });
    }
    listByReservationOrderAndReservationPagingAll(reservationOrderId, reservationId, grain, options) {
        return __asyncGenerator(this, arguments, function* listByReservationOrderAndReservationPagingAll_1() {
            var e_2, _a;
            try {
                for (var _b = __asyncValues(this.listByReservationOrderAndReservationPagingPage(reservationOrderId, reservationId, grain, options)), _c; _c = yield __await(_b.next()), !_c.done;) {
                    const page = _c.value;
                    yield __await(yield* __asyncDelegator(__asyncValues(page)));
                }
            }
            catch (e_2_1) { e_2 = { error: e_2_1 }; }
            finally {
                try {
                    if (_c && !_c.done && (_a = _b.return)) yield __await(_a.call(_b));
                }
                finally { if (e_2) throw e_2.error; }
            }
        });
    }
    /**
     * Lists the reservations summaries for the defined scope daily or monthly grain.
     * @param scope The scope associated with reservations summaries operations. This includes
     *              '/providers/Microsoft.Billing/billingAccounts/{billingAccountId}' for BillingAccount scope (legacy),
     *              and
     *              '/providers/Microsoft.Billing/billingAccounts/{billingAccountId}/billingProfiles/{billingProfileId}'
     *              for BillingProfile scope (modern).
     * @param grain Can be daily or monthly
     * @param options The options parameters.
     */
    list(scope, grain, options) {
        const iter = this.listPagingAll(scope, grain, options);
        return {
            next() {
                return iter.next();
            },
            [Symbol.asyncIterator]() {
                return this;
            },
            byPage: (settings) => {
                if (settings === null || settings === void 0 ? void 0 : settings.maxPageSize) {
                    throw new Error("maxPageSize is not supported by this operation.");
                }
                return this.listPagingPage(scope, grain, options, settings);
            }
        };
    }
    listPagingPage(scope, grain, options, settings) {
        return __asyncGenerator(this, arguments, function* listPagingPage_1() {
            let result;
            let continuationToken = settings === null || settings === void 0 ? void 0 : settings.continuationToken;
            if (!continuationToken) {
                result = yield __await(this._list(scope, grain, options));
                let page = result.value || [];
                continuationToken = result.nextLink;
                setContinuationToken(page, continuationToken);
                yield yield __await(page);
            }
            while (continuationToken) {
                result = yield __await(this._listNext(scope, continuationToken, options));
                continuationToken = result.nextLink;
                let page = result.value || [];
                setContinuationToken(page, continuationToken);
                yield yield __await(page);
            }
        });
    }
    listPagingAll(scope, grain, options) {
        return __asyncGenerator(this, arguments, function* listPagingAll_1() {
            var e_3, _a;
            try {
                for (var _b = __asyncValues(this.listPagingPage(scope, grain, options)), _c; _c = yield __await(_b.next()), !_c.done;) {
                    const page = _c.value;
                    yield __await(yield* __asyncDelegator(__asyncValues(page)));
                }
            }
            catch (e_3_1) { e_3 = { error: e_3_1 }; }
            finally {
                try {
                    if (_c && !_c.done && (_a = _b.return)) yield __await(_a.call(_b));
                }
                finally { if (e_3) throw e_3.error; }
            }
        });
    }
    /**
     * Lists the reservations summaries for daily or monthly grain.
     * @param reservationOrderId Order Id of the reservation
     * @param grain Can be daily or monthly
     * @param options The options parameters.
     */
    _listByReservationOrder(reservationOrderId, grain, options) {
        return this.client.sendOperationRequest({ reservationOrderId, grain, options }, listByReservationOrderOperationSpec);
    }
    /**
     * Lists the reservations summaries for daily or monthly grain.
     * @param reservationOrderId Order Id of the reservation
     * @param reservationId Id of the reservation
     * @param grain Can be daily or monthly
     * @param options The options parameters.
     */
    _listByReservationOrderAndReservation(reservationOrderId, reservationId, grain, options) {
        return this.client.sendOperationRequest({ reservationOrderId, reservationId, grain, options }, listByReservationOrderAndReservationOperationSpec);
    }
    /**
     * Lists the reservations summaries for the defined scope daily or monthly grain.
     * @param scope The scope associated with reservations summaries operations. This includes
     *              '/providers/Microsoft.Billing/billingAccounts/{billingAccountId}' for BillingAccount scope (legacy),
     *              and
     *              '/providers/Microsoft.Billing/billingAccounts/{billingAccountId}/billingProfiles/{billingProfileId}'
     *              for BillingProfile scope (modern).
     * @param grain Can be daily or monthly
     * @param options The options parameters.
     */
    _list(scope, grain, options) {
        return this.client.sendOperationRequest({ scope, grain, options }, listOperationSpec);
    }
    /**
     * ListByReservationOrderNext
     * @param reservationOrderId Order Id of the reservation
     * @param nextLink The nextLink from the previous successful call to the ListByReservationOrder method.
     * @param options The options parameters.
     */
    _listByReservationOrderNext(reservationOrderId, nextLink, options) {
        return this.client.sendOperationRequest({ reservationOrderId, nextLink, options }, listByReservationOrderNextOperationSpec);
    }
    /**
     * ListByReservationOrderAndReservationNext
     * @param reservationOrderId Order Id of the reservation
     * @param reservationId Id of the reservation
     * @param nextLink The nextLink from the previous successful call to the
     *                 ListByReservationOrderAndReservation method.
     * @param options The options parameters.
     */
    _listByReservationOrderAndReservationNext(reservationOrderId, reservationId, nextLink, options) {
        return this.client.sendOperationRequest({ reservationOrderId, reservationId, nextLink, options }, listByReservationOrderAndReservationNextOperationSpec);
    }
    /**
     * ListNext
     * @param scope The scope associated with reservations summaries operations. This includes
     *              '/providers/Microsoft.Billing/billingAccounts/{billingAccountId}' for BillingAccount scope (legacy),
     *              and
     *              '/providers/Microsoft.Billing/billingAccounts/{billingAccountId}/billingProfiles/{billingProfileId}'
     *              for BillingProfile scope (modern).
     * @param nextLink The nextLink from the previous successful call to the List method.
     * @param options The options parameters.
     */
    _listNext(scope, nextLink, options) {
        return this.client.sendOperationRequest({ scope, nextLink, options }, listNextOperationSpec);
    }
}
// Operation Specifications
const serializer = coreClient.createSerializer(Mappers, /* isXml */ false);
const listByReservationOrderOperationSpec = {
    path: "/providers/Microsoft.Capacity/reservationorders/{reservationOrderId}/providers/Microsoft.Consumption/reservationSummaries",
    httpMethod: "GET",
    responses: {
        200: {
            bodyMapper: Mappers.ReservationSummariesListResult
        },
        default: {
            bodyMapper: Mappers.ErrorResponse
        }
    },
    queryParameters: [Parameters.filter, Parameters.apiVersion, Parameters.grain],
    urlParameters: [Parameters.$host, Parameters.reservationOrderId],
    headerParameters: [Parameters.accept],
    serializer
};
const listByReservationOrderAndReservationOperationSpec = {
    path: "/providers/Microsoft.Capacity/reservationorders/{reservationOrderId}/reservations/{reservationId}/providers/Microsoft.Consumption/reservationSummaries",
    httpMethod: "GET",
    responses: {
        200: {
            bodyMapper: Mappers.ReservationSummariesListResult
        },
        default: {
            bodyMapper: Mappers.ErrorResponse
        }
    },
    queryParameters: [Parameters.filter, Parameters.apiVersion, Parameters.grain],
    urlParameters: [
        Parameters.$host,
        Parameters.reservationOrderId,
        Parameters.reservationId
    ],
    headerParameters: [Parameters.accept],
    serializer
};
const listOperationSpec = {
    path: "/{scope}/providers/Microsoft.Consumption/reservationSummaries",
    httpMethod: "GET",
    responses: {
        200: {
            bodyMapper: Mappers.ReservationSummariesListResult
        },
        default: {
            bodyMapper: Mappers.ErrorResponse
        }
    },
    queryParameters: [
        Parameters.filter,
        Parameters.apiVersion,
        Parameters.startDate,
        Parameters.endDate,
        Parameters.grain,
        Parameters.reservationId1,
        Parameters.reservationOrderId1
    ],
    urlParameters: [Parameters.$host, Parameters.scope],
    headerParameters: [Parameters.accept],
    serializer
};
const listByReservationOrderNextOperationSpec = {
    path: "{nextLink}",
    httpMethod: "GET",
    responses: {
        200: {
            bodyMapper: Mappers.ReservationSummariesListResult
        },
        default: {
            bodyMapper: Mappers.ErrorResponse
        }
    },
    urlParameters: [
        Parameters.$host,
        Parameters.nextLink,
        Parameters.reservationOrderId
    ],
    headerParameters: [Parameters.accept],
    serializer
};
const listByReservationOrderAndReservationNextOperationSpec = {
    path: "{nextLink}",
    httpMethod: "GET",
    responses: {
        200: {
            bodyMapper: Mappers.ReservationSummariesListResult
        },
        default: {
            bodyMapper: Mappers.ErrorResponse
        }
    },
    urlParameters: [
        Parameters.$host,
        Parameters.nextLink,
        Parameters.reservationOrderId,
        Parameters.reservationId
    ],
    headerParameters: [Parameters.accept],
    serializer
};
const listNextOperationSpec = {
    path: "{nextLink}",
    httpMethod: "GET",
    responses: {
        200: {
            bodyMapper: Mappers.ReservationSummariesListResult
        },
        default: {
            bodyMapper: Mappers.ErrorResponse
        }
    },
    urlParameters: [Parameters.$host, Parameters.scope, Parameters.nextLink],
    headerParameters: [Parameters.accept],
    serializer
};
