// @azure/arm-cognitiveservices/dist-esm/test/cognitiveservices_congitiveservice.spec.js
/*
 * Copyright (c) Microsoft Corporation.
 * Licensed under the MIT License.
 */
import { __asyncValues, __awaiter } from "tslib";
import { env, Recorder, isPlaybackMode, } from "@azure-tools/test-recorder";
import { createTestCredential } from "@azure-tools/test-credential";
import { assert } from "chai";
import { CognitiveServicesManagementClient } from "../src/cognitiveServicesManagementClient";
const replaceableVariables = {
    AZURE_CLIENT_ID: "azure_client_id",
    AZURE_CLIENT_SECRET: "azure_client_secret",
    AZURE_TENANT_ID: "88888888-8888-8888-8888-888888888888",
    SUBSCRIPTION_ID: "azure_subscription_id"
};
const recorderOptions = {
    envSetupForPlayback: replaceableVariables
};
export const testPollingOptions = {
    updateIntervalInMs: isPlaybackMode() ? 0 : undefined,
};
describe("CognitiveServices test", () => {
    let recorder;
    let subscriptionId;
    let client;
    let location;
    let resourceGroup;
    let accountName;
    beforeEach(function () {
        return __awaiter(this, void 0, void 0, function* () {
            recorder = new Recorder(this.currentTest);
            yield recorder.start(recorderOptions);
            subscriptionId = env.SUBSCRIPTION_ID || '';
            // This is an example of how the environment variables are used
            const credential = createTestCredential();
            client = new CognitiveServicesManagementClient(credential, subscriptionId, recorder.configureClientOptions({}));
            location = "eastus";
            resourceGroup = "myjstest";
            accountName = "myaccountxxxx1";
        });
    });
    afterEach(function () {
        return __awaiter(this, void 0, void 0, function* () {
            yield recorder.stop();
        });
    });
    it("accounts create test", function () {
        return __awaiter(this, void 0, void 0, function* () {
            const res = yield client.accounts.beginCreateAndWait(resourceGroup, accountName, {
                location: location,
                kind: "CognitiveServices",
                sku: {
                    name: "S0"
                },
                identity: {
                    type: "SystemAssigned"
                }
            });
            assert.equal(res.name, accountName);
        });
    });
    it("accounts get test", function () {
        return __awaiter(this, void 0, void 0, function* () {
            const res = yield client.accounts.get(resourceGroup, accountName);
            assert.equal(res.name, accountName);
        });
    });
    it("accounts list test", function () {
        var _a, e_1, _b, _c;
        return __awaiter(this, void 0, void 0, function* () {
            const resArray = new Array();
            try {
                for (var _d = true, _e = __asyncValues(client.accounts.listByResourceGroup(resourceGroup)), _f; _f = yield _e.next(), _a = _f.done, !_a;) {
                    _c = _f.value;
                    _d = false;
                    try {
                        let item = _c;
                        resArray.push(item);
                    }
                    finally {
                        _d = true;
                    }
                }
            }
            catch (e_1_1) { e_1 = { error: e_1_1 }; }
            finally {
                try {
                    if (!_d && !_a && (_b = _e.return)) yield _b.call(_e);
                }
                finally { if (e_1) throw e_1.error; }
            }
            assert.equal(resArray.length, 1);
        });
    });
    it("accounts regenerateKey test", function () {
        return __awaiter(this, void 0, void 0, function* () {
            const res = yield client.accounts.regenerateKey(resourceGroup, accountName, "Key2");
            assert.notEqual(res.key2, "");
        });
    });
    it("accounts update test", function () {
        return __awaiter(this, void 0, void 0, function* () {
            const res = yield client.accounts.beginUpdateAndWait(resourceGroup, accountName, { tags: { tag1: "value1" } });
            assert.equal(res.type, "Microsoft.CognitiveServices/accounts");
        });
    });
    it("accounts delete test", function () {
        var _a, e_2, _b, _c;
        return __awaiter(this, void 0, void 0, function* () {
            const res = yield client.accounts.beginDeleteAndWait(resourceGroup, accountName);
            const resArray = new Array();
            try {
                for (var _d = true, _e = __asyncValues(client.accounts.listByResourceGroup(resourceGroup)), _f; _f = yield _e.next(), _a = _f.done, !_a;) {
                    _c = _f.value;
                    _d = false;
                    try {
                        let item = _c;
                        resArray.push(item);
                    }
                    finally {
                        _d = true;
                    }
                }
            }
            catch (e_2_1) { e_2 = { error: e_2_1 }; }
            finally {
                try {
                    if (!_d && !_a && (_b = _e.return)) yield _b.call(_e);
                }
                finally { if (e_2) throw e_2.error; }
            }
            assert.equal(resArray.length, 0);
        });
    });
    it("operations list test", function () {
        var _a, e_3, _b, _c;
        return __awaiter(this, void 0, void 0, function* () {
            const resArray = new Array();
            try {
                for (var _d = true, _e = __asyncValues(client.operations.list()), _f; _f = yield _e.next(), _a = _f.done, !_a;) {
                    _c = _f.value;
                    _d = false;
                    try {
                        let item = _c;
                        resArray.push(item);
                    }
                    finally {
                        _d = true;
                    }
                }
            }
            catch (e_3_1) { e_3 = { error: e_3_1 }; }
            finally {
                try {
                    if (!_d && !_a && (_b = _e.return)) yield _b.call(_e);
                }
                finally { if (e_3) throw e_3.error; }
            }
        });
    });
});
