// @azure/arm-datacatalog/dist-esm/src/operations/aDCCatalogs.js
/*
 * Copyright (c) Microsoft Corporation.
 * Licensed under the MIT License.
 *
 * Code generated by Microsoft (R) AutoRest Code Generator.
 * Changes may cause incorrect behavior and will be lost if the code is regenerated.
 */
import { __awaiter } from "tslib";
import * as coreClient from "@azure/core-client";
import * as Mappers from "../models/mappers";
import * as Parameters from "../models/parameters";
import { LroEngine } from "@azure/core-lro";
import { LroImpl } from "../lroImpl";
/** Class containing ADCCatalogs operations. */
export class ADCCatalogsImpl {
    /**
     * Initialize a new instance of the class ADCCatalogs class.
     * @param client Reference to the service client
     */
    constructor(client) {
        this.client = client;
    }
    /**
     * The List catalogs in Resource Group operation lists all the Azure Data Catalogs available under the
     * given resource group.
     * @param resourceGroupName The name of the resource group within the user's subscription. The name is
     *                          case insensitive.
     * @param options The options parameters.
     */
    listtByResourceGroup(resourceGroupName, options) {
        return this.client.sendOperationRequest({ resourceGroupName, options }, listtByResourceGroupOperationSpec);
    }
    /**
     * The Create Azure Data Catalog service operation creates a new data catalog service with the
     * specified parameters. If the specific service already exists, then any patchable properties will be
     * updated and any immutable properties will remain unchanged.
     * @param resourceGroupName The name of the resource group within the user's subscription. The name is
     *                          case insensitive.
     * @param catalogName The name of the data catalog in the specified subscription and resource group.
     * @param properties Properties supplied to the Create or Update a data catalog.
     * @param options The options parameters.
     */
    createOrUpdate(resourceGroupName, catalogName, properties, options) {
        return this.client.sendOperationRequest({ resourceGroupName, catalogName, properties, options }, createOrUpdateOperationSpec);
    }
    /**
     * The Get Azure Data Catalog Service operation retrieves a json representation of the data catalog.
     * @param resourceGroupName The name of the resource group within the user's subscription. The name is
     *                          case insensitive.
     * @param catalogName The name of the data catalog in the specified subscription and resource group.
     * @param options The options parameters.
     */
    get(resourceGroupName, catalogName, options) {
        return this.client.sendOperationRequest({ resourceGroupName, catalogName, options }, getOperationSpec);
    }
    /**
     * The Delete Azure Data Catalog Service operation deletes an existing data catalog.
     * @param resourceGroupName The name of the resource group within the user's subscription. The name is
     *                          case insensitive.
     * @param catalogName The name of the data catalog in the specified subscription and resource group.
     * @param options The options parameters.
     */
    beginDelete(resourceGroupName, catalogName, options) {
        return __awaiter(this, void 0, void 0, function* () {
            const directSendOperation = (args, spec) => __awaiter(this, void 0, void 0, function* () {
                return this.client.sendOperationRequest(args, spec);
            });
            const sendOperation = (args, spec) => __awaiter(this, void 0, void 0, function* () {
                var _a;
                let currentRawResponse = undefined;
                const providedCallback = (_a = args.options) === null || _a === void 0 ? void 0 : _a.onResponse;
                const callback = (rawResponse, flatResponse) => {
                    currentRawResponse = rawResponse;
                    providedCallback === null || providedCallback === void 0 ? void 0 : providedCallback(rawResponse, flatResponse);
                };
                const updatedArgs = Object.assign(Object.assign({}, args), { options: Object.assign(Object.assign({}, args.options), { onResponse: callback }) });
                const flatResponse = yield directSendOperation(updatedArgs, spec);
                return {
                    flatResponse,
                    rawResponse: {
                        statusCode: currentRawResponse.status,
                        body: currentRawResponse.parsedBody,
                        headers: currentRawResponse.headers.toJSON()
                    }
                };
            });
            const lro = new LroImpl(sendOperation, { resourceGroupName, catalogName, options }, deleteOperationSpec);
            const poller = new LroEngine(lro, {
                resumeFrom: options === null || options === void 0 ? void 0 : options.resumeFrom,
                intervalInMs: options === null || options === void 0 ? void 0 : options.updateIntervalInMs
            });
            yield poller.poll();
            return poller;
        });
    }
    /**
     * The Delete Azure Data Catalog Service operation deletes an existing data catalog.
     * @param resourceGroupName The name of the resource group within the user's subscription. The name is
     *                          case insensitive.
     * @param catalogName The name of the data catalog in the specified subscription and resource group.
     * @param options The options parameters.
     */
    beginDeleteAndWait(resourceGroupName, catalogName, options) {
        return __awaiter(this, void 0, void 0, function* () {
            const poller = yield this.beginDelete(resourceGroupName, catalogName, options);
            return poller.pollUntilDone();
        });
    }
    /**
     * The Update Azure Data Catalog Service operation can be used to update the existing deployment. The
     * update call only supports the properties listed in the PATCH body.
     * @param resourceGroupName The name of the resource group within the user's subscription. The name is
     *                          case insensitive.
     * @param catalogName The name of the data catalog in the specified subscription and resource group.
     * @param properties Properties supplied to the Update a data catalog.
     * @param options The options parameters.
     */
    update(resourceGroupName, catalogName, properties, options) {
        return this.client.sendOperationRequest({ resourceGroupName, catalogName, properties, options }, updateOperationSpec);
    }
}
// Operation Specifications
const serializer = coreClient.createSerializer(Mappers, /* isXml */ false);
const listtByResourceGroupOperationSpec = {
    path: "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.DataCatalog/catalogs",
    httpMethod: "GET",
    responses: {
        200: {
            bodyMapper: Mappers.ADCCatalogsListResult
        }
    },
    queryParameters: [Parameters.apiVersion],
    urlParameters: [
        Parameters.$host,
        Parameters.subscriptionId,
        Parameters.resourceGroupName
    ],
    headerParameters: [Parameters.accept],
    serializer
};
const createOrUpdateOperationSpec = {
    path: "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.DataCatalog/catalogs/{catalogName}",
    httpMethod: "PUT",
    responses: {
        200: {
            bodyMapper: Mappers.ADCCatalog
        },
        201: {
            bodyMapper: Mappers.ADCCatalog
        }
    },
    requestBody: Parameters.properties,
    queryParameters: [Parameters.apiVersion],
    urlParameters: [
        Parameters.$host,
        Parameters.subscriptionId,
        Parameters.resourceGroupName,
        Parameters.catalogName
    ],
    headerParameters: [Parameters.accept, Parameters.contentType],
    mediaType: "json",
    serializer
};
const getOperationSpec = {
    path: "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.DataCatalog/catalogs/{catalogName}",
    httpMethod: "GET",
    responses: {
        200: {
            bodyMapper: Mappers.ADCCatalog
        }
    },
    queryParameters: [Parameters.apiVersion],
    urlParameters: [
        Parameters.$host,
        Parameters.subscriptionId,
        Parameters.resourceGroupName,
        Parameters.catalogName
    ],
    headerParameters: [Parameters.accept],
    serializer
};
const deleteOperationSpec = {
    path: "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.DataCatalog/catalogs/{catalogName}",
    httpMethod: "DELETE",
    responses: { 200: {}, 201: {}, 202: {}, 204: {} },
    queryParameters: [Parameters.apiVersion],
    urlParameters: [
        Parameters.$host,
        Parameters.subscriptionId,
        Parameters.resourceGroupName,
        Parameters.catalogName
    ],
    serializer
};
const updateOperationSpec = {
    path: "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.DataCatalog/catalogs/{catalogName}",
    httpMethod: "PATCH",
    responses: {
        200: {
            bodyMapper: Mappers.ADCCatalog
        }
    },
    requestBody: Parameters.properties,
    queryParameters: [Parameters.apiVersion],
    urlParameters: [
        Parameters.$host,
        Parameters.subscriptionId,
        Parameters.resourceGroupName,
        Parameters.catalogName
    ],
    headerParameters: [Parameters.accept, Parameters.contentType],
    mediaType: "json",
    serializer
};
