// @azure/arm-postgresql/dist-esm/src/models/index.js
/*
 * Copyright (c) Microsoft Corporation.
 * Licensed under the MIT License.
 *
 * Code generated by Microsoft (R) AutoRest Code Generator.
 * Changes may cause incorrect behavior and will be lost if the code is regenerated.
 */
/** Known values of {@link IdentityType} that the service accepts. */
export var KnownIdentityType;
(function (KnownIdentityType) {
    /** SystemAssigned */
    KnownIdentityType["SystemAssigned"] = "SystemAssigned";
})(KnownIdentityType || (KnownIdentityType = {}));
/** Known values of {@link SkuTier} that the service accepts. */
export var KnownSkuTier;
(function (KnownSkuTier) {
    /** Basic */
    KnownSkuTier["Basic"] = "Basic";
    /** GeneralPurpose */
    KnownSkuTier["GeneralPurpose"] = "GeneralPurpose";
    /** MemoryOptimized */
    KnownSkuTier["MemoryOptimized"] = "MemoryOptimized";
})(KnownSkuTier || (KnownSkuTier = {}));
/** Known values of {@link ServerVersion} that the service accepts. */
export var KnownServerVersion;
(function (KnownServerVersion) {
    /** Nine5 */
    KnownServerVersion["Nine5"] = "9.5";
    /** Nine6 */
    KnownServerVersion["Nine6"] = "9.6";
    /** Ten */
    KnownServerVersion["Ten"] = "10";
    /** Ten0 */
    KnownServerVersion["Ten0"] = "10.0";
    /** Ten2 */
    KnownServerVersion["Ten2"] = "10.2";
    /** Eleven */
    KnownServerVersion["Eleven"] = "11";
})(KnownServerVersion || (KnownServerVersion = {}));
/** Known values of {@link MinimalTlsVersionEnum} that the service accepts. */
export var KnownMinimalTlsVersionEnum;
(function (KnownMinimalTlsVersionEnum) {
    /** TLS10 */
    KnownMinimalTlsVersionEnum["TLS10"] = "TLS1_0";
    /** TLS11 */
    KnownMinimalTlsVersionEnum["TLS11"] = "TLS1_1";
    /** TLS12 */
    KnownMinimalTlsVersionEnum["TLS12"] = "TLS1_2";
    /** TLSEnforcementDisabled */
    KnownMinimalTlsVersionEnum["TLSEnforcementDisabled"] = "TLSEnforcementDisabled";
})(KnownMinimalTlsVersionEnum || (KnownMinimalTlsVersionEnum = {}));
/** Known values of {@link InfrastructureEncryption} that the service accepts. */
export var KnownInfrastructureEncryption;
(function (KnownInfrastructureEncryption) {
    /** Default value for single layer of encryption for data at rest. */
    KnownInfrastructureEncryption["Enabled"] = "Enabled";
    /** Additional (2nd) layer of encryption for data at rest */
    KnownInfrastructureEncryption["Disabled"] = "Disabled";
})(KnownInfrastructureEncryption || (KnownInfrastructureEncryption = {}));
/** Known values of {@link PublicNetworkAccessEnum} that the service accepts. */
export var KnownPublicNetworkAccessEnum;
(function (KnownPublicNetworkAccessEnum) {
    /** Enabled */
    KnownPublicNetworkAccessEnum["Enabled"] = "Enabled";
    /** Disabled */
    KnownPublicNetworkAccessEnum["Disabled"] = "Disabled";
})(KnownPublicNetworkAccessEnum || (KnownPublicNetworkAccessEnum = {}));
/** Known values of {@link GeoRedundantBackup} that the service accepts. */
export var KnownGeoRedundantBackup;
(function (KnownGeoRedundantBackup) {
    /** Enabled */
    KnownGeoRedundantBackup["Enabled"] = "Enabled";
    /** Disabled */
    KnownGeoRedundantBackup["Disabled"] = "Disabled";
})(KnownGeoRedundantBackup || (KnownGeoRedundantBackup = {}));
/** Known values of {@link StorageAutogrow} that the service accepts. */
export var KnownStorageAutogrow;
(function (KnownStorageAutogrow) {
    /** Enabled */
    KnownStorageAutogrow["Enabled"] = "Enabled";
    /** Disabled */
    KnownStorageAutogrow["Disabled"] = "Disabled";
})(KnownStorageAutogrow || (KnownStorageAutogrow = {}));
/** Known values of {@link CreateMode} that the service accepts. */
export var KnownCreateMode;
(function (KnownCreateMode) {
    /** Default */
    KnownCreateMode["Default"] = "Default";
    /** PointInTimeRestore */
    KnownCreateMode["PointInTimeRestore"] = "PointInTimeRestore";
    /** GeoRestore */
    KnownCreateMode["GeoRestore"] = "GeoRestore";
    /** Replica */
    KnownCreateMode["Replica"] = "Replica";
})(KnownCreateMode || (KnownCreateMode = {}));
/** Known values of {@link ServerState} that the service accepts. */
export var KnownServerState;
(function (KnownServerState) {
    /** Ready */
    KnownServerState["Ready"] = "Ready";
    /** Dropping */
    KnownServerState["Dropping"] = "Dropping";
    /** Disabled */
    KnownServerState["Disabled"] = "Disabled";
    /** Inaccessible */
    KnownServerState["Inaccessible"] = "Inaccessible";
})(KnownServerState || (KnownServerState = {}));
/** Known values of {@link PrivateLinkServiceConnectionStateStatus} that the service accepts. */
export var KnownPrivateLinkServiceConnectionStateStatus;
(function (KnownPrivateLinkServiceConnectionStateStatus) {
    /** Approved */
    KnownPrivateLinkServiceConnectionStateStatus["Approved"] = "Approved";
    /** Pending */
    KnownPrivateLinkServiceConnectionStateStatus["Pending"] = "Pending";
    /** Rejected */
    KnownPrivateLinkServiceConnectionStateStatus["Rejected"] = "Rejected";
    /** Disconnected */
    KnownPrivateLinkServiceConnectionStateStatus["Disconnected"] = "Disconnected";
})(KnownPrivateLinkServiceConnectionStateStatus || (KnownPrivateLinkServiceConnectionStateStatus = {}));
/** Known values of {@link PrivateLinkServiceConnectionStateActionsRequire} that the service accepts. */
export var KnownPrivateLinkServiceConnectionStateActionsRequire;
(function (KnownPrivateLinkServiceConnectionStateActionsRequire) {
    /** None */
    KnownPrivateLinkServiceConnectionStateActionsRequire["None"] = "None";
})(KnownPrivateLinkServiceConnectionStateActionsRequire || (KnownPrivateLinkServiceConnectionStateActionsRequire = {}));
/** Known values of {@link PrivateEndpointProvisioningState} that the service accepts. */
export var KnownPrivateEndpointProvisioningState;
(function (KnownPrivateEndpointProvisioningState) {
    /** Approving */
    KnownPrivateEndpointProvisioningState["Approving"] = "Approving";
    /** Ready */
    KnownPrivateEndpointProvisioningState["Ready"] = "Ready";
    /** Dropping */
    KnownPrivateEndpointProvisioningState["Dropping"] = "Dropping";
    /** Failed */
    KnownPrivateEndpointProvisioningState["Failed"] = "Failed";
    /** Rejecting */
    KnownPrivateEndpointProvisioningState["Rejecting"] = "Rejecting";
})(KnownPrivateEndpointProvisioningState || (KnownPrivateEndpointProvisioningState = {}));
/** Known values of {@link VirtualNetworkRuleState} that the service accepts. */
export var KnownVirtualNetworkRuleState;
(function (KnownVirtualNetworkRuleState) {
    /** Initializing */
    KnownVirtualNetworkRuleState["Initializing"] = "Initializing";
    /** InProgress */
    KnownVirtualNetworkRuleState["InProgress"] = "InProgress";
    /** Ready */
    KnownVirtualNetworkRuleState["Ready"] = "Ready";
    /** Deleting */
    KnownVirtualNetworkRuleState["Deleting"] = "Deleting";
    /** Unknown */
    KnownVirtualNetworkRuleState["Unknown"] = "Unknown";
})(KnownVirtualNetworkRuleState || (KnownVirtualNetworkRuleState = {}));
/** Known values of {@link OperationOrigin} that the service accepts. */
export var KnownOperationOrigin;
(function (KnownOperationOrigin) {
    /** NotSpecified */
    KnownOperationOrigin["NotSpecified"] = "NotSpecified";
    /** User */
    KnownOperationOrigin["User"] = "user";
    /** System */
    KnownOperationOrigin["System"] = "system";
})(KnownOperationOrigin || (KnownOperationOrigin = {}));
/** Known values of {@link SecurityAlertPolicyName} that the service accepts. */
export var KnownSecurityAlertPolicyName;
(function (KnownSecurityAlertPolicyName) {
    /** Default */
    KnownSecurityAlertPolicyName["Default"] = "Default";
})(KnownSecurityAlertPolicyName || (KnownSecurityAlertPolicyName = {}));
/** Known values of {@link ServerKeyType} that the service accepts. */
export var KnownServerKeyType;
(function (KnownServerKeyType) {
    /** AzureKeyVault */
    KnownServerKeyType["AzureKeyVault"] = "AzureKeyVault";
})(KnownServerKeyType || (KnownServerKeyType = {}));
