// @azure/arm-databoxedge/dist-esm/src/operations/alerts.js
/*
 * Copyright (c) Microsoft Corporation.
 * Licensed under the MIT License.
 *
 * Code generated by Microsoft (R) AutoRest Code Generator.
 * Changes may cause incorrect behavior and will be lost if the code is regenerated.
 */
import { __asyncDelegator, __asyncGenerator, __asyncValues, __await } from "tslib";
import { setContinuationToken } from "../pagingHelper";
import * as coreClient from "@azure/core-client";
import * as Mappers from "../models/mappers";
import * as Parameters from "../models/parameters";
/// <reference lib="esnext.asynciterable" />
/** Class containing Alerts operations. */
export class AlertsImpl {
    /**
     * Initialize a new instance of the class Alerts class.
     * @param client Reference to the service client
     */
    constructor(client) {
        this.client = client;
    }
    /**
     * Gets all the alerts for a Data Box Edge/Data Box Gateway device.
     * @param deviceName The device name.
     * @param resourceGroupName The resource group name.
     * @param options The options parameters.
     */
    listByDataBoxEdgeDevice(deviceName, resourceGroupName, options) {
        const iter = this.listByDataBoxEdgeDevicePagingAll(deviceName, resourceGroupName, options);
        return {
            next() {
                return iter.next();
            },
            [Symbol.asyncIterator]() {
                return this;
            },
            byPage: (settings) => {
                if (settings === null || settings === void 0 ? void 0 : settings.maxPageSize) {
                    throw new Error("maxPageSize is not supported by this operation.");
                }
                return this.listByDataBoxEdgeDevicePagingPage(deviceName, resourceGroupName, options, settings);
            }
        };
    }
    listByDataBoxEdgeDevicePagingPage(deviceName, resourceGroupName, options, settings) {
        return __asyncGenerator(this, arguments, function* listByDataBoxEdgeDevicePagingPage_1() {
            let result;
            let continuationToken = settings === null || settings === void 0 ? void 0 : settings.continuationToken;
            if (!continuationToken) {
                result = yield __await(this._listByDataBoxEdgeDevice(deviceName, resourceGroupName, options));
                let page = result.value || [];
                continuationToken = result.nextLink;
                setContinuationToken(page, continuationToken);
                yield yield __await(page);
            }
            while (continuationToken) {
                result = yield __await(this._listByDataBoxEdgeDeviceNext(deviceName, resourceGroupName, continuationToken, options));
                continuationToken = result.nextLink;
                let page = result.value || [];
                setContinuationToken(page, continuationToken);
                yield yield __await(page);
            }
        });
    }
    listByDataBoxEdgeDevicePagingAll(deviceName, resourceGroupName, options) {
        return __asyncGenerator(this, arguments, function* listByDataBoxEdgeDevicePagingAll_1() {
            var e_1, _a;
            try {
                for (var _b = __asyncValues(this.listByDataBoxEdgeDevicePagingPage(deviceName, resourceGroupName, options)), _c; _c = yield __await(_b.next()), !_c.done;) {
                    const page = _c.value;
                    yield __await(yield* __asyncDelegator(__asyncValues(page)));
                }
            }
            catch (e_1_1) { e_1 = { error: e_1_1 }; }
            finally {
                try {
                    if (_c && !_c.done && (_a = _b.return)) yield __await(_a.call(_b));
                }
                finally { if (e_1) throw e_1.error; }
            }
        });
    }
    /**
     * Gets all the alerts for a Data Box Edge/Data Box Gateway device.
     * @param deviceName The device name.
     * @param resourceGroupName The resource group name.
     * @param options The options parameters.
     */
    _listByDataBoxEdgeDevice(deviceName, resourceGroupName, options) {
        return this.client.sendOperationRequest({ deviceName, resourceGroupName, options }, listByDataBoxEdgeDeviceOperationSpec);
    }
    /**
     * Gets an alert by name.
     * @param deviceName The device name.
     * @param name The alert name.
     * @param resourceGroupName The resource group name.
     * @param options The options parameters.
     */
    get(deviceName, name, resourceGroupName, options) {
        return this.client.sendOperationRequest({ deviceName, name, resourceGroupName, options }, getOperationSpec);
    }
    /**
     * ListByDataBoxEdgeDeviceNext
     * @param deviceName The device name.
     * @param resourceGroupName The resource group name.
     * @param nextLink The nextLink from the previous successful call to the ListByDataBoxEdgeDevice
     *                 method.
     * @param options The options parameters.
     */
    _listByDataBoxEdgeDeviceNext(deviceName, resourceGroupName, nextLink, options) {
        return this.client.sendOperationRequest({ deviceName, resourceGroupName, nextLink, options }, listByDataBoxEdgeDeviceNextOperationSpec);
    }
}
// Operation Specifications
const serializer = coreClient.createSerializer(Mappers, /* isXml */ false);
const listByDataBoxEdgeDeviceOperationSpec = {
    path: "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.DataBoxEdge/dataBoxEdgeDevices/{deviceName}/alerts",
    httpMethod: "GET",
    responses: {
        200: {
            bodyMapper: Mappers.AlertList
        },
        default: {
            bodyMapper: Mappers.CloudError
        }
    },
    queryParameters: [Parameters.apiVersion],
    urlParameters: [
        Parameters.$host,
        Parameters.subscriptionId,
        Parameters.resourceGroupName,
        Parameters.deviceName
    ],
    headerParameters: [Parameters.accept],
    serializer
};
const getOperationSpec = {
    path: "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.DataBoxEdge/dataBoxEdgeDevices/{deviceName}/alerts/{name}",
    httpMethod: "GET",
    responses: {
        200: {
            bodyMapper: Mappers.Alert
        },
        default: {
            bodyMapper: Mappers.CloudError
        }
    },
    queryParameters: [Parameters.apiVersion],
    urlParameters: [
        Parameters.$host,
        Parameters.subscriptionId,
        Parameters.resourceGroupName,
        Parameters.deviceName,
        Parameters.name
    ],
    headerParameters: [Parameters.accept],
    serializer
};
const listByDataBoxEdgeDeviceNextOperationSpec = {
    path: "{nextLink}",
    httpMethod: "GET",
    responses: {
        200: {
            bodyMapper: Mappers.AlertList
        },
        default: {
            bodyMapper: Mappers.CloudError
        }
    },
    queryParameters: [Parameters.apiVersion],
    urlParameters: [
        Parameters.$host,
        Parameters.nextLink,
        Parameters.subscriptionId,
        Parameters.resourceGroupName,
        Parameters.deviceName
    ],
    headerParameters: [Parameters.accept],
    serializer
};
