// @azure/arm-databoxedge/dist-esm/src/operations/diagnosticSettings.js
/*
 * Copyright (c) Microsoft Corporation.
 * Licensed under the MIT License.
 *
 * Code generated by Microsoft (R) AutoRest Code Generator.
 * Changes may cause incorrect behavior and will be lost if the code is regenerated.
 */
import { __awaiter } from "tslib";
import * as coreClient from "@azure/core-client";
import * as Mappers from "../models/mappers";
import * as Parameters from "../models/parameters";
import { LroEngine } from "@azure/core-lro";
import { LroImpl } from "../lroImpl";
/** Class containing DiagnosticSettings operations. */
export class DiagnosticSettingsImpl {
    /**
     * Initialize a new instance of the class DiagnosticSettings class.
     * @param client Reference to the service client
     */
    constructor(client) {
        this.client = client;
    }
    /**
     * Gets the proactive log collection settings of the specified Data Box Edge/Data Box Gateway device.
     * @param deviceName The device name.
     * @param resourceGroupName The resource group name.
     * @param options The options parameters.
     */
    getDiagnosticProactiveLogCollectionSettings(deviceName, resourceGroupName, options) {
        return this.client.sendOperationRequest({ deviceName, resourceGroupName, options }, getDiagnosticProactiveLogCollectionSettingsOperationSpec);
    }
    /**
     * Updates the proactive log collection settings on a Data Box Edge/Data Box Gateway device.
     * @param deviceName The device name.
     * @param resourceGroupName The resource group name.
     * @param proactiveLogCollectionSettings The proactive log collection settings.
     * @param options The options parameters.
     */
    beginUpdateDiagnosticProactiveLogCollectionSettings(deviceName, resourceGroupName, proactiveLogCollectionSettings, options) {
        return __awaiter(this, void 0, void 0, function* () {
            const directSendOperation = (args, spec) => __awaiter(this, void 0, void 0, function* () {
                return this.client.sendOperationRequest(args, spec);
            });
            const sendOperation = (args, spec) => __awaiter(this, void 0, void 0, function* () {
                var _a;
                let currentRawResponse = undefined;
                const providedCallback = (_a = args.options) === null || _a === void 0 ? void 0 : _a.onResponse;
                const callback = (rawResponse, flatResponse) => {
                    currentRawResponse = rawResponse;
                    providedCallback === null || providedCallback === void 0 ? void 0 : providedCallback(rawResponse, flatResponse);
                };
                const updatedArgs = Object.assign(Object.assign({}, args), { options: Object.assign(Object.assign({}, args.options), { onResponse: callback }) });
                const flatResponse = yield directSendOperation(updatedArgs, spec);
                return {
                    flatResponse,
                    rawResponse: {
                        statusCode: currentRawResponse.status,
                        body: currentRawResponse.parsedBody,
                        headers: currentRawResponse.headers.toJSON()
                    }
                };
            });
            const lro = new LroImpl(sendOperation, {
                deviceName,
                resourceGroupName,
                proactiveLogCollectionSettings,
                options
            }, updateDiagnosticProactiveLogCollectionSettingsOperationSpec);
            const poller = new LroEngine(lro, {
                resumeFrom: options === null || options === void 0 ? void 0 : options.resumeFrom,
                intervalInMs: options === null || options === void 0 ? void 0 : options.updateIntervalInMs
            });
            yield poller.poll();
            return poller;
        });
    }
    /**
     * Updates the proactive log collection settings on a Data Box Edge/Data Box Gateway device.
     * @param deviceName The device name.
     * @param resourceGroupName The resource group name.
     * @param proactiveLogCollectionSettings The proactive log collection settings.
     * @param options The options parameters.
     */
    beginUpdateDiagnosticProactiveLogCollectionSettingsAndWait(deviceName, resourceGroupName, proactiveLogCollectionSettings, options) {
        return __awaiter(this, void 0, void 0, function* () {
            const poller = yield this.beginUpdateDiagnosticProactiveLogCollectionSettings(deviceName, resourceGroupName, proactiveLogCollectionSettings, options);
            return poller.pollUntilDone();
        });
    }
    /**
     * Gets the diagnostic remote support settings of the specified Data Box Edge/Data Box Gateway device.
     * @param deviceName The device name.
     * @param resourceGroupName The resource group name.
     * @param options The options parameters.
     */
    getDiagnosticRemoteSupportSettings(deviceName, resourceGroupName, options) {
        return this.client.sendOperationRequest({ deviceName, resourceGroupName, options }, getDiagnosticRemoteSupportSettingsOperationSpec);
    }
    /**
     * Updates the diagnostic remote support settings on a Data Box Edge/Data Box Gateway device.
     * @param deviceName The device name.
     * @param resourceGroupName The resource group name.
     * @param diagnosticRemoteSupportSettings The diagnostic remote support settings.
     * @param options The options parameters.
     */
    beginUpdateDiagnosticRemoteSupportSettings(deviceName, resourceGroupName, diagnosticRemoteSupportSettings, options) {
        return __awaiter(this, void 0, void 0, function* () {
            const directSendOperation = (args, spec) => __awaiter(this, void 0, void 0, function* () {
                return this.client.sendOperationRequest(args, spec);
            });
            const sendOperation = (args, spec) => __awaiter(this, void 0, void 0, function* () {
                var _a;
                let currentRawResponse = undefined;
                const providedCallback = (_a = args.options) === null || _a === void 0 ? void 0 : _a.onResponse;
                const callback = (rawResponse, flatResponse) => {
                    currentRawResponse = rawResponse;
                    providedCallback === null || providedCallback === void 0 ? void 0 : providedCallback(rawResponse, flatResponse);
                };
                const updatedArgs = Object.assign(Object.assign({}, args), { options: Object.assign(Object.assign({}, args.options), { onResponse: callback }) });
                const flatResponse = yield directSendOperation(updatedArgs, spec);
                return {
                    flatResponse,
                    rawResponse: {
                        statusCode: currentRawResponse.status,
                        body: currentRawResponse.parsedBody,
                        headers: currentRawResponse.headers.toJSON()
                    }
                };
            });
            const lro = new LroImpl(sendOperation, {
                deviceName,
                resourceGroupName,
                diagnosticRemoteSupportSettings,
                options
            }, updateDiagnosticRemoteSupportSettingsOperationSpec);
            const poller = new LroEngine(lro, {
                resumeFrom: options === null || options === void 0 ? void 0 : options.resumeFrom,
                intervalInMs: options === null || options === void 0 ? void 0 : options.updateIntervalInMs
            });
            yield poller.poll();
            return poller;
        });
    }
    /**
     * Updates the diagnostic remote support settings on a Data Box Edge/Data Box Gateway device.
     * @param deviceName The device name.
     * @param resourceGroupName The resource group name.
     * @param diagnosticRemoteSupportSettings The diagnostic remote support settings.
     * @param options The options parameters.
     */
    beginUpdateDiagnosticRemoteSupportSettingsAndWait(deviceName, resourceGroupName, diagnosticRemoteSupportSettings, options) {
        return __awaiter(this, void 0, void 0, function* () {
            const poller = yield this.beginUpdateDiagnosticRemoteSupportSettings(deviceName, resourceGroupName, diagnosticRemoteSupportSettings, options);
            return poller.pollUntilDone();
        });
    }
}
// Operation Specifications
const serializer = coreClient.createSerializer(Mappers, /* isXml */ false);
const getDiagnosticProactiveLogCollectionSettingsOperationSpec = {
    path: "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.DataBoxEdge/dataBoxEdgeDevices/{deviceName}/diagnosticProactiveLogCollectionSettings/default",
    httpMethod: "GET",
    responses: {
        200: {
            bodyMapper: Mappers.DiagnosticProactiveLogCollectionSettings
        },
        default: {
            bodyMapper: Mappers.CloudError
        }
    },
    queryParameters: [Parameters.apiVersion],
    urlParameters: [
        Parameters.$host,
        Parameters.subscriptionId,
        Parameters.resourceGroupName,
        Parameters.deviceName
    ],
    headerParameters: [Parameters.accept],
    serializer
};
const updateDiagnosticProactiveLogCollectionSettingsOperationSpec = {
    path: "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.DataBoxEdge/dataBoxEdgeDevices/{deviceName}/diagnosticProactiveLogCollectionSettings/default",
    httpMethod: "PUT",
    responses: {
        200: {
            bodyMapper: Mappers.DiagnosticProactiveLogCollectionSettings
        },
        201: {
            bodyMapper: Mappers.DiagnosticProactiveLogCollectionSettings
        },
        202: {
            bodyMapper: Mappers.DiagnosticProactiveLogCollectionSettings
        },
        204: {
            bodyMapper: Mappers.DiagnosticProactiveLogCollectionSettings
        },
        default: {
            bodyMapper: Mappers.CloudError
        }
    },
    requestBody: Parameters.proactiveLogCollectionSettings,
    queryParameters: [Parameters.apiVersion],
    urlParameters: [
        Parameters.$host,
        Parameters.subscriptionId,
        Parameters.resourceGroupName,
        Parameters.deviceName
    ],
    headerParameters: [Parameters.accept, Parameters.contentType],
    mediaType: "json",
    serializer
};
const getDiagnosticRemoteSupportSettingsOperationSpec = {
    path: "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.DataBoxEdge/dataBoxEdgeDevices/{deviceName}/diagnosticRemoteSupportSettings/default",
    httpMethod: "GET",
    responses: {
        200: {
            bodyMapper: Mappers.DiagnosticRemoteSupportSettings
        },
        default: {
            bodyMapper: Mappers.CloudError
        }
    },
    queryParameters: [Parameters.apiVersion],
    urlParameters: [
        Parameters.$host,
        Parameters.subscriptionId,
        Parameters.resourceGroupName,
        Parameters.deviceName
    ],
    headerParameters: [Parameters.accept],
    serializer
};
const updateDiagnosticRemoteSupportSettingsOperationSpec = {
    path: "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.DataBoxEdge/dataBoxEdgeDevices/{deviceName}/diagnosticRemoteSupportSettings/default",
    httpMethod: "PUT",
    responses: {
        200: {
            bodyMapper: Mappers.DiagnosticRemoteSupportSettings
        },
        201: {
            bodyMapper: Mappers.DiagnosticRemoteSupportSettings
        },
        202: {
            bodyMapper: Mappers.DiagnosticRemoteSupportSettings
        },
        204: {
            bodyMapper: Mappers.DiagnosticRemoteSupportSettings
        },
        default: {
            bodyMapper: Mappers.CloudError
        }
    },
    requestBody: Parameters.diagnosticRemoteSupportSettings,
    queryParameters: [Parameters.apiVersion],
    urlParameters: [
        Parameters.$host,
        Parameters.subscriptionId,
        Parameters.resourceGroupName,
        Parameters.deviceName
    ],
    headerParameters: [Parameters.accept, Parameters.contentType],
    mediaType: "json",
    serializer
};
