// @azure/arm-databoxedge/dist-esm/src/operations/operationsStatus.js
/*
 * Copyright (c) Microsoft Corporation.
 * Licensed under the MIT License.
 *
 * Code generated by Microsoft (R) AutoRest Code Generator.
 * Changes may cause incorrect behavior and will be lost if the code is regenerated.
 */
import * as coreClient from "@azure/core-client";
import * as Mappers from "../models/mappers";
import * as Parameters from "../models/parameters";
/** Class containing OperationsStatus operations. */
export class OperationsStatusImpl {
    /**
     * Initialize a new instance of the class OperationsStatus class.
     * @param client Reference to the service client
     */
    constructor(client) {
        this.client = client;
    }
    /**
     * Gets the details of a specified job on a Data Box Edge/Data Box Gateway device.
     * @param deviceName The device name.
     * @param name The job name.
     * @param resourceGroupName The resource group name.
     * @param options The options parameters.
     */
    get(deviceName, name, resourceGroupName, options) {
        return this.client.sendOperationRequest({ deviceName, name, resourceGroupName, options }, getOperationSpec);
    }
}
// Operation Specifications
const serializer = coreClient.createSerializer(Mappers, /* isXml */ false);
const getOperationSpec = {
    path: "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.DataBoxEdge/dataBoxEdgeDevices/{deviceName}/operationsStatus/{name}",
    httpMethod: "GET",
    responses: {
        200: {
            bodyMapper: Mappers.Job
        },
        default: {
            bodyMapper: Mappers.CloudError
        }
    },
    queryParameters: [Parameters.apiVersion],
    urlParameters: [
        Parameters.$host,
        Parameters.subscriptionId,
        Parameters.resourceGroupName,
        Parameters.deviceName,
        Parameters.name
    ],
    headerParameters: [Parameters.accept],
    serializer
};
