// @azure/arm-features/dist-esm/src/models/index.js
/*
 * Copyright (c) Microsoft Corporation.
 * Licensed under the MIT License.
 *
 * Code generated by Microsoft (R) AutoRest Code Generator.
 * Changes may cause incorrect behavior and will be lost if the code is regenerated.
 */
/** Known values of {@link SubscriptionFeatureRegistrationState} that the service accepts. */
export var KnownSubscriptionFeatureRegistrationState;
(function (KnownSubscriptionFeatureRegistrationState) {
    /** NotSpecified */
    KnownSubscriptionFeatureRegistrationState["NotSpecified"] = "NotSpecified";
    /** NotRegistered */
    KnownSubscriptionFeatureRegistrationState["NotRegistered"] = "NotRegistered";
    /** Pending */
    KnownSubscriptionFeatureRegistrationState["Pending"] = "Pending";
    /** Registering */
    KnownSubscriptionFeatureRegistrationState["Registering"] = "Registering";
    /** Registered */
    KnownSubscriptionFeatureRegistrationState["Registered"] = "Registered";
    /** Unregistering */
    KnownSubscriptionFeatureRegistrationState["Unregistering"] = "Unregistering";
    /** Unregistered */
    KnownSubscriptionFeatureRegistrationState["Unregistered"] = "Unregistered";
})(KnownSubscriptionFeatureRegistrationState || (KnownSubscriptionFeatureRegistrationState = {}));
/** Known values of {@link SubscriptionFeatureRegistrationApprovalType} that the service accepts. */
export var KnownSubscriptionFeatureRegistrationApprovalType;
(function (KnownSubscriptionFeatureRegistrationApprovalType) {
    /** NotSpecified */
    KnownSubscriptionFeatureRegistrationApprovalType["NotSpecified"] = "NotSpecified";
    /** ApprovalRequired */
    KnownSubscriptionFeatureRegistrationApprovalType["ApprovalRequired"] = "ApprovalRequired";
    /** AutoApproval */
    KnownSubscriptionFeatureRegistrationApprovalType["AutoApproval"] = "AutoApproval";
})(KnownSubscriptionFeatureRegistrationApprovalType || (KnownSubscriptionFeatureRegistrationApprovalType = {}));
