// @azure/arm-hybridcompute/dist-esm/src/models/index.js
/*
 * Copyright (c) Microsoft Corporation.
 * Licensed under the MIT License.
 *
 * Code generated by Microsoft (R) AutoRest Code Generator.
 * Changes may cause incorrect behavior and will be lost if the code is regenerated.
 */
/** Known values of {@link InstanceViewTypes} that the service accepts. */
export var KnownInstanceViewTypes;
(function (KnownInstanceViewTypes) {
    KnownInstanceViewTypes["InstanceView"] = "instanceView";
})(KnownInstanceViewTypes || (KnownInstanceViewTypes = {}));
/** Known values of {@link StatusTypes} that the service accepts. */
export var KnownStatusTypes;
(function (KnownStatusTypes) {
    KnownStatusTypes["Connected"] = "Connected";
    KnownStatusTypes["Disconnected"] = "Disconnected";
    KnownStatusTypes["Error"] = "Error";
})(KnownStatusTypes || (KnownStatusTypes = {}));
/** Known values of {@link StatusLevelTypes} that the service accepts. */
export var KnownStatusLevelTypes;
(function (KnownStatusLevelTypes) {
    KnownStatusLevelTypes["Info"] = "Info";
    KnownStatusLevelTypes["Warning"] = "Warning";
    KnownStatusLevelTypes["Error"] = "Error";
})(KnownStatusLevelTypes || (KnownStatusLevelTypes = {}));
/** Known values of {@link CreatedByType} that the service accepts. */
export var KnownCreatedByType;
(function (KnownCreatedByType) {
    KnownCreatedByType["User"] = "User";
    KnownCreatedByType["Application"] = "Application";
    KnownCreatedByType["ManagedIdentity"] = "ManagedIdentity";
    KnownCreatedByType["Key"] = "Key";
})(KnownCreatedByType || (KnownCreatedByType = {}));
/** Known values of {@link PublicNetworkAccessType} that the service accepts. */
export var KnownPublicNetworkAccessType;
(function (KnownPublicNetworkAccessType) {
    /** Allows Azure Arc agents to communicate with Azure Arc services over both public (internet) and private endpoints. */
    KnownPublicNetworkAccessType["Enabled"] = "Enabled";
    /** Does not allow Azure Arc agents to communicate with Azure Arc services over public (internet) endpoints. The agents must use the private link. */
    KnownPublicNetworkAccessType["Disabled"] = "Disabled";
})(KnownPublicNetworkAccessType || (KnownPublicNetworkAccessType = {}));
