// @azure/arm-recoveryservices/dist-esm/test/recoveryservices_examples.js
/*
 * Copyright (c) Microsoft Corporation.
 * Licensed under the MIT License.
 *
 * Code generated by Microsoft (R) AutoRest Code Generator.
 * Changes may cause incorrect behavior and will be lost if the code is regenerated.
 */
import { __asyncValues, __awaiter } from "tslib";
import { env, Recorder, isPlaybackMode, } from "@azure-tools/test-recorder";
import { createTestCredential } from "@azure-tools/test-credential";
import { assert } from "chai";
import { RecoveryServicesClient } from "../src/recoveryServicesClient";
const replaceableVariables = {
    AZURE_CLIENT_ID: "azure_client_id",
    AZURE_CLIENT_SECRET: "azure_client_secret",
    AZURE_TENANT_ID: "88888888-8888-8888-8888-888888888888",
    SUBSCRIPTION_ID: "azure_subscription_id"
};
const recorderOptions = {
    envSetupForPlayback: replaceableVariables
};
export const testPollingOptions = {
    updateIntervalInMs: isPlaybackMode() ? 0 : undefined,
};
describe("Recoveryservices test", () => {
    let recorder;
    let subscriptionId;
    let client;
    let location;
    let resourceGroup;
    let vaultsName;
    beforeEach(function () {
        return __awaiter(this, void 0, void 0, function* () {
            recorder = new Recorder(this.currentTest);
            yield recorder.start(recorderOptions);
            subscriptionId = env.SUBSCRIPTION_ID || '';
            // This is an example of how the environment variables are used
            const credential = createTestCredential();
            client = new RecoveryServicesClient(credential, subscriptionId, recorder.configureClientOptions({}));
            location = "eastus";
            resourceGroup = "myjstest";
            vaultsName = "myvaultxxx";
        });
    });
    afterEach(function () {
        return __awaiter(this, void 0, void 0, function* () {
            yield recorder.stop();
        });
    });
    it("vaults create test", function () {
        return __awaiter(this, void 0, void 0, function* () {
            const res = yield client.vaults.beginCreateOrUpdateAndWait(resourceGroup, vaultsName, {
                location: location,
                properties: { publicNetworkAccess: "Enabled" },
                sku: {
                    name: "Standard"
                },
                identity: {
                    type: "SystemAssigned"
                }
            }, testPollingOptions);
            assert.equal(res.name, vaultsName);
        });
    });
    it("vaults get test", function () {
        return __awaiter(this, void 0, void 0, function* () {
            const res = yield client.vaults.get(resourceGroup, vaultsName);
            assert.equal(res.name, vaultsName);
        });
    });
    it("vaults list test", function () {
        var _a, e_1, _b, _c;
        return __awaiter(this, void 0, void 0, function* () {
            const resArray = new Array();
            try {
                for (var _d = true, _e = __asyncValues(client.vaults.listByResourceGroup(resourceGroup)), _f; _f = yield _e.next(), _a = _f.done, !_a;) {
                    _c = _f.value;
                    _d = false;
                    try {
                        let item = _c;
                        resArray.push(item);
                    }
                    finally {
                        _d = true;
                    }
                }
            }
            catch (e_1_1) { e_1 = { error: e_1_1 }; }
            finally {
                try {
                    if (!_d && !_a && (_b = _e.return)) yield _b.call(_e);
                }
                finally { if (e_1) throw e_1.error; }
            }
            assert.equal(resArray.length, 1);
        });
    });
    it("vaultExtendedInfo create test", function () {
        return __awaiter(this, void 0, void 0, function* () {
            const res = yield client.vaultExtendedInfo.createOrUpdate(resourceGroup, vaultsName, {
                algorithm: "None"
            });
            assert.equal(res.algorithm, "None");
        });
    });
    it("vaultExtendedInfo get test", function () {
        return __awaiter(this, void 0, void 0, function* () {
            const res = yield client.vaultExtendedInfo.get(resourceGroup, vaultsName);
            assert.equal(res.type, "Microsoft.RecoveryServices/vaults/extendedInformation");
        });
    });
    it("vaults delete test", function () {
        var _a, e_2, _b, _c;
        return __awaiter(this, void 0, void 0, function* () {
            const res = yield client.vaults.delete(resourceGroup, vaultsName);
            const resArray = new Array();
            try {
                for (var _d = true, _e = __asyncValues(client.vaults.listByResourceGroup(resourceGroup)), _f; _f = yield _e.next(), _a = _f.done, !_a;) {
                    _c = _f.value;
                    _d = false;
                    try {
                        let item = _c;
                        resArray.push(item);
                    }
                    finally {
                        _d = true;
                    }
                }
            }
            catch (e_2_1) { e_2 = { error: e_2_1 }; }
            finally {
                try {
                    if (!_d && !_a && (_b = _e.return)) yield _b.call(_e);
                }
                finally { if (e_2) throw e_2.error; }
            }
            assert.equal(resArray.length, 0);
        });
    });
});
