// @aws-sdk/client-chime-sdk-identity/dist-cjs/index.js
"use strict";
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __name = (target, value) => __defProp(target, "name", { value, configurable: true });
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// src/index.ts
var src_exports = {};
__export(src_exports, {
  AllowMessages: () => AllowMessages,
  AppInstanceAdminFilterSensitiveLog: () => AppInstanceAdminFilterSensitiveLog,
  AppInstanceAdminSummaryFilterSensitiveLog: () => AppInstanceAdminSummaryFilterSensitiveLog,
  AppInstanceBotFilterSensitiveLog: () => AppInstanceBotFilterSensitiveLog,
  AppInstanceBotSummaryFilterSensitiveLog: () => AppInstanceBotSummaryFilterSensitiveLog,
  AppInstanceFilterSensitiveLog: () => AppInstanceFilterSensitiveLog,
  AppInstanceSummaryFilterSensitiveLog: () => AppInstanceSummaryFilterSensitiveLog,
  AppInstanceUserEndpointFilterSensitiveLog: () => AppInstanceUserEndpointFilterSensitiveLog,
  AppInstanceUserEndpointSummaryFilterSensitiveLog: () => AppInstanceUserEndpointSummaryFilterSensitiveLog,
  AppInstanceUserEndpointType: () => AppInstanceUserEndpointType,
  AppInstanceUserFilterSensitiveLog: () => AppInstanceUserFilterSensitiveLog,
  AppInstanceUserSummaryFilterSensitiveLog: () => AppInstanceUserSummaryFilterSensitiveLog,
  BadRequestException: () => BadRequestException,
  ChimeSDKIdentity: () => ChimeSDKIdentity,
  ChimeSDKIdentityClient: () => ChimeSDKIdentityClient,
  ChimeSDKIdentityServiceException: () => ChimeSDKIdentityServiceException,
  ConflictException: () => ConflictException,
  CreateAppInstanceAdminCommand: () => CreateAppInstanceAdminCommand,
  CreateAppInstanceAdminResponseFilterSensitiveLog: () => CreateAppInstanceAdminResponseFilterSensitiveLog,
  CreateAppInstanceBotCommand: () => CreateAppInstanceBotCommand,
  CreateAppInstanceBotRequestFilterSensitiveLog: () => CreateAppInstanceBotRequestFilterSensitiveLog,
  CreateAppInstanceCommand: () => CreateAppInstanceCommand,
  CreateAppInstanceRequestFilterSensitiveLog: () => CreateAppInstanceRequestFilterSensitiveLog,
  CreateAppInstanceUserCommand: () => CreateAppInstanceUserCommand,
  CreateAppInstanceUserRequestFilterSensitiveLog: () => CreateAppInstanceUserRequestFilterSensitiveLog,
  DeleteAppInstanceAdminCommand: () => DeleteAppInstanceAdminCommand,
  DeleteAppInstanceBotCommand: () => DeleteAppInstanceBotCommand,
  DeleteAppInstanceCommand: () => DeleteAppInstanceCommand,
  DeleteAppInstanceUserCommand: () => DeleteAppInstanceUserCommand,
  DeregisterAppInstanceUserEndpointCommand: () => DeregisterAppInstanceUserEndpointCommand,
  DescribeAppInstanceAdminCommand: () => DescribeAppInstanceAdminCommand,
  DescribeAppInstanceAdminResponseFilterSensitiveLog: () => DescribeAppInstanceAdminResponseFilterSensitiveLog,
  DescribeAppInstanceBotCommand: () => DescribeAppInstanceBotCommand,
  DescribeAppInstanceBotResponseFilterSensitiveLog: () => DescribeAppInstanceBotResponseFilterSensitiveLog,
  DescribeAppInstanceCommand: () => DescribeAppInstanceCommand,
  DescribeAppInstanceResponseFilterSensitiveLog: () => DescribeAppInstanceResponseFilterSensitiveLog,
  DescribeAppInstanceUserCommand: () => DescribeAppInstanceUserCommand,
  DescribeAppInstanceUserEndpointCommand: () => DescribeAppInstanceUserEndpointCommand,
  DescribeAppInstanceUserEndpointResponseFilterSensitiveLog: () => DescribeAppInstanceUserEndpointResponseFilterSensitiveLog,
  DescribeAppInstanceUserResponseFilterSensitiveLog: () => DescribeAppInstanceUserResponseFilterSensitiveLog,
  EndpointAttributesFilterSensitiveLog: () => EndpointAttributesFilterSensitiveLog,
  EndpointStatus: () => EndpointStatus,
  EndpointStatusReason: () => EndpointStatusReason,
  ErrorCode: () => ErrorCode,
  ExpirationCriterion: () => ExpirationCriterion,
  ForbiddenException: () => ForbiddenException,
  GetAppInstanceRetentionSettingsCommand: () => GetAppInstanceRetentionSettingsCommand,
  IdentityFilterSensitiveLog: () => IdentityFilterSensitiveLog,
  ListAppInstanceAdminsCommand: () => ListAppInstanceAdminsCommand,
  ListAppInstanceAdminsRequestFilterSensitiveLog: () => ListAppInstanceAdminsRequestFilterSensitiveLog,
  ListAppInstanceAdminsResponseFilterSensitiveLog: () => ListAppInstanceAdminsResponseFilterSensitiveLog,
  ListAppInstanceBotsCommand: () => ListAppInstanceBotsCommand,
  ListAppInstanceBotsRequestFilterSensitiveLog: () => ListAppInstanceBotsRequestFilterSensitiveLog,
  ListAppInstanceBotsResponseFilterSensitiveLog: () => ListAppInstanceBotsResponseFilterSensitiveLog,
  ListAppInstanceUserEndpointsCommand: () => ListAppInstanceUserEndpointsCommand,
  ListAppInstanceUserEndpointsRequestFilterSensitiveLog: () => ListAppInstanceUserEndpointsRequestFilterSensitiveLog,
  ListAppInstanceUserEndpointsResponseFilterSensitiveLog: () => ListAppInstanceUserEndpointsResponseFilterSensitiveLog,
  ListAppInstanceUsersCommand: () => ListAppInstanceUsersCommand,
  ListAppInstanceUsersRequestFilterSensitiveLog: () => ListAppInstanceUsersRequestFilterSensitiveLog,
  ListAppInstanceUsersResponseFilterSensitiveLog: () => ListAppInstanceUsersResponseFilterSensitiveLog,
  ListAppInstancesCommand: () => ListAppInstancesCommand,
  ListAppInstancesRequestFilterSensitiveLog: () => ListAppInstancesRequestFilterSensitiveLog,
  ListAppInstancesResponseFilterSensitiveLog: () => ListAppInstancesResponseFilterSensitiveLog,
  ListTagsForResourceCommand: () => ListTagsForResourceCommand,
  ListTagsForResourceResponseFilterSensitiveLog: () => ListTagsForResourceResponseFilterSensitiveLog,
  NotFoundException: () => NotFoundException,
  PutAppInstanceRetentionSettingsCommand: () => PutAppInstanceRetentionSettingsCommand,
  PutAppInstanceUserExpirationSettingsCommand: () => PutAppInstanceUserExpirationSettingsCommand,
  RegisterAppInstanceUserEndpointCommand: () => RegisterAppInstanceUserEndpointCommand,
  RegisterAppInstanceUserEndpointRequestFilterSensitiveLog: () => RegisterAppInstanceUserEndpointRequestFilterSensitiveLog,
  ResourceLimitExceededException: () => ResourceLimitExceededException,
  RespondsTo: () => RespondsTo,
  ServiceFailureException: () => ServiceFailureException,
  ServiceUnavailableException: () => ServiceUnavailableException,
  StandardMessages: () => StandardMessages,
  TagFilterSensitiveLog: () => TagFilterSensitiveLog,
  TagResourceCommand: () => TagResourceCommand,
  TagResourceRequestFilterSensitiveLog: () => TagResourceRequestFilterSensitiveLog,
  TargetedMessages: () => TargetedMessages,
  ThrottledClientException: () => ThrottledClientException,
  UnauthorizedClientException: () => UnauthorizedClientException,
  UntagResourceCommand: () => UntagResourceCommand,
  UntagResourceRequestFilterSensitiveLog: () => UntagResourceRequestFilterSensitiveLog,
  UpdateAppInstanceBotCommand: () => UpdateAppInstanceBotCommand,
  UpdateAppInstanceBotRequestFilterSensitiveLog: () => UpdateAppInstanceBotRequestFilterSensitiveLog,
  UpdateAppInstanceCommand: () => UpdateAppInstanceCommand,
  UpdateAppInstanceRequestFilterSensitiveLog: () => UpdateAppInstanceRequestFilterSensitiveLog,
  UpdateAppInstanceUserCommand: () => UpdateAppInstanceUserCommand,
  UpdateAppInstanceUserEndpointCommand: () => UpdateAppInstanceUserEndpointCommand,
  UpdateAppInstanceUserEndpointRequestFilterSensitiveLog: () => UpdateAppInstanceUserEndpointRequestFilterSensitiveLog,
  UpdateAppInstanceUserRequestFilterSensitiveLog: () => UpdateAppInstanceUserRequestFilterSensitiveLog,
  __Client: () => import_smithy_client.Client,
  paginateListAppInstanceAdmins: () => paginateListAppInstanceAdmins,
  paginateListAppInstanceBots: () => paginateListAppInstanceBots,
  paginateListAppInstanceUserEndpoints: () => paginateListAppInstanceUserEndpoints,
  paginateListAppInstanceUsers: () => paginateListAppInstanceUsers,
  paginateListAppInstances: () => paginateListAppInstances
});
module.exports = __toCommonJS(src_exports);

// src/ChimeSDKIdentityClient.ts
var import_middleware_host_header = require("@aws-sdk/middleware-host-header");
var import_middleware_logger = require("@aws-sdk/middleware-logger");
var import_middleware_recursion_detection = require("@aws-sdk/middleware-recursion-detection");
var import_middleware_user_agent = require("@aws-sdk/middleware-user-agent");
var import_config_resolver = require("@smithy/config-resolver");
var import_core = require("@smithy/core");
var import_middleware_content_length = require("@smithy/middleware-content-length");
var import_middleware_endpoint = require("@smithy/middleware-endpoint");
var import_middleware_retry = require("@smithy/middleware-retry");

var import_httpAuthSchemeProvider = require("./auth/httpAuthSchemeProvider");

// src/endpoint/EndpointParameters.ts
var resolveClientEndpointParameters = /* @__PURE__ */ __name((options) => {
  return {
    ...options,
    useDualstackEndpoint: options.useDualstackEndpoint ?? false,
    useFipsEndpoint: options.useFipsEndpoint ?? false,
    defaultSigningName: "chime"
  };
}, "resolveClientEndpointParameters");
var commonParams = {
  UseFIPS: { type: "builtInParams", name: "useFipsEndpoint" },
  Endpoint: { type: "builtInParams", name: "endpoint" },
  Region: { type: "builtInParams", name: "region" },
  UseDualStack: { type: "builtInParams", name: "useDualstackEndpoint" }
};

// src/ChimeSDKIdentityClient.ts
var import_runtimeConfig = require("././runtimeConfig");

// src/runtimeExtensions.ts
var import_region_config_resolver = require("@aws-sdk/region-config-resolver");
var import_protocol_http = require("@smithy/protocol-http");
var import_smithy_client = require("@smithy/smithy-client");

// src/auth/httpAuthExtensionConfiguration.ts
var getHttpAuthExtensionConfiguration = /* @__PURE__ */ __name((runtimeConfig) => {
  const _httpAuthSchemes = runtimeConfig.httpAuthSchemes;
  let _httpAuthSchemeProvider = runtimeConfig.httpAuthSchemeProvider;
  let _credentials = runtimeConfig.credentials;
  return {
    setHttpAuthScheme(httpAuthScheme) {
      const index = _httpAuthSchemes.findIndex((scheme) => scheme.schemeId === httpAuthScheme.schemeId);
      if (index === -1) {
        _httpAuthSchemes.push(httpAuthScheme);
      } else {
        _httpAuthSchemes.splice(index, 1, httpAuthScheme);
      }
    },
    httpAuthSchemes() {
      return _httpAuthSchemes;
    },
    setHttpAuthSchemeProvider(httpAuthSchemeProvider) {
      _httpAuthSchemeProvider = httpAuthSchemeProvider;
    },
    httpAuthSchemeProvider() {
      return _httpAuthSchemeProvider;
    },
    setCredentials(credentials) {
      _credentials = credentials;
    },
    credentials() {
      return _credentials;
    }
  };
}, "getHttpAuthExtensionConfiguration");
var resolveHttpAuthRuntimeConfig = /* @__PURE__ */ __name((config) => {
  return {
    httpAuthSchemes: config.httpAuthSchemes(),
    httpAuthSchemeProvider: config.httpAuthSchemeProvider(),
    credentials: config.credentials()
  };
}, "resolveHttpAuthRuntimeConfig");

// src/runtimeExtensions.ts
var asPartial = /* @__PURE__ */ __name((t) => t, "asPartial");
var resolveRuntimeExtensions = /* @__PURE__ */ __name((runtimeConfig, extensions) => {
  const extensionConfiguration = {
    ...asPartial((0, import_region_config_resolver.getAwsRegionExtensionConfiguration)(runtimeConfig)),
    ...asPartial((0, import_smithy_client.getDefaultExtensionConfiguration)(runtimeConfig)),
    ...asPartial((0, import_protocol_http.getHttpHandlerExtensionConfiguration)(runtimeConfig)),
    ...asPartial(getHttpAuthExtensionConfiguration(runtimeConfig))
  };
  extensions.forEach((extension) => extension.configure(extensionConfiguration));
  return {
    ...runtimeConfig,
    ...(0, import_region_config_resolver.resolveAwsRegionExtensionConfiguration)(extensionConfiguration),
    ...(0, import_smithy_client.resolveDefaultRuntimeConfig)(extensionConfiguration),
    ...(0, import_protocol_http.resolveHttpHandlerRuntimeConfig)(extensionConfiguration),
    ...resolveHttpAuthRuntimeConfig(extensionConfiguration)
  };
}, "resolveRuntimeExtensions");

// src/ChimeSDKIdentityClient.ts
var _ChimeSDKIdentityClient = class _ChimeSDKIdentityClient extends import_smithy_client.Client {
  constructor(...[configuration]) {
    const _config_0 = (0, import_runtimeConfig.getRuntimeConfig)(configuration || {});
    const _config_1 = resolveClientEndpointParameters(_config_0);
    const _config_2 = (0, import_config_resolver.resolveRegionConfig)(_config_1);
    const _config_3 = (0, import_middleware_endpoint.resolveEndpointConfig)(_config_2);
    const _config_4 = (0, import_middleware_retry.resolveRetryConfig)(_config_3);
    const _config_5 = (0, import_middleware_host_header.resolveHostHeaderConfig)(_config_4);
    const _config_6 = (0, import_middleware_user_agent.resolveUserAgentConfig)(_config_5);
    const _config_7 = (0, import_httpAuthSchemeProvider.resolveHttpAuthSchemeConfig)(_config_6);
    const _config_8 = resolveRuntimeExtensions(_config_7, (configuration == null ? void 0 : configuration.extensions) || []);
    super(_config_8);
    this.config = _config_8;
    this.middlewareStack.use((0, import_middleware_retry.getRetryPlugin)(this.config));
    this.middlewareStack.use((0, import_middleware_content_length.getContentLengthPlugin)(this.config));
    this.middlewareStack.use((0, import_middleware_host_header.getHostHeaderPlugin)(this.config));
    this.middlewareStack.use((0, import_middleware_logger.getLoggerPlugin)(this.config));
    this.middlewareStack.use((0, import_middleware_recursion_detection.getRecursionDetectionPlugin)(this.config));
    this.middlewareStack.use((0, import_middleware_user_agent.getUserAgentPlugin)(this.config));
    this.middlewareStack.use(
      (0, import_core.getHttpAuthSchemeEndpointRuleSetPlugin)(this.config, {
        httpAuthSchemeParametersProvider: this.getDefaultHttpAuthSchemeParametersProvider(),
        identityProviderConfigProvider: this.getIdentityProviderConfigProvider()
      })
    );
    this.middlewareStack.use((0, import_core.getHttpSigningPlugin)(this.config));
  }
  /**
   * Destroy underlying resources, like sockets. It's usually not necessary to do this.
   * However in Node.js, it's best to explicitly shut down the client's agent when it is no longer needed.
   * Otherwise, sockets might stay open for quite a long time before the server terminates them.
   */
  destroy() {
    super.destroy();
  }
  getDefaultHttpAuthSchemeParametersProvider() {
    return import_httpAuthSchemeProvider.defaultChimeSDKIdentityHttpAuthSchemeParametersProvider;
  }
  getIdentityProviderConfigProvider() {
    return async (config) => new import_core.DefaultIdentityProviderConfig({
      "aws.auth#sigv4": config.credentials
    });
  }
};
__name(_ChimeSDKIdentityClient, "ChimeSDKIdentityClient");
var ChimeSDKIdentityClient = _ChimeSDKIdentityClient;

// src/ChimeSDKIdentity.ts


// src/commands/CreateAppInstanceAdminCommand.ts

var import_middleware_serde = require("@smithy/middleware-serde");

var import_types = require("@smithy/types");

// src/models/models_0.ts


// src/models/ChimeSDKIdentityServiceException.ts

var _ChimeSDKIdentityServiceException = class _ChimeSDKIdentityServiceException extends import_smithy_client.ServiceException {
  /**
   * @internal
   */
  constructor(options) {
    super(options);
    Object.setPrototypeOf(this, _ChimeSDKIdentityServiceException.prototype);
  }
};
__name(_ChimeSDKIdentityServiceException, "ChimeSDKIdentityServiceException");
var ChimeSDKIdentityServiceException = _ChimeSDKIdentityServiceException;

// src/models/models_0.ts
var AllowMessages = {
  ALL: "ALL",
  NONE: "NONE"
};
var StandardMessages = {
  ALL: "ALL",
  AUTO: "AUTO",
  MENTIONS: "MENTIONS",
  NONE: "NONE"
};
var TargetedMessages = {
  ALL: "ALL",
  NONE: "NONE"
};
var RespondsTo = {
  STANDARD_MESSAGES: "STANDARD_MESSAGES"
};
var ExpirationCriterion = {
  CREATED_TIMESTAMP: "CREATED_TIMESTAMP"
};
var EndpointStatus = {
  ACTIVE: "ACTIVE",
  INACTIVE: "INACTIVE"
};
var EndpointStatusReason = {
  INVALID_DEVICE_TOKEN: "INVALID_DEVICE_TOKEN",
  INVALID_PINPOINT_ARN: "INVALID_PINPOINT_ARN"
};
var AppInstanceUserEndpointType = {
  APNS: "APNS",
  APNS_SANDBOX: "APNS_SANDBOX",
  GCM: "GCM"
};
var ErrorCode = {
  AccessDenied: "AccessDenied",
  BadRequest: "BadRequest",
  Conflict: "Conflict",
  Forbidden: "Forbidden",
  NotFound: "NotFound",
  PhoneNumberAssociationsExist: "PhoneNumberAssociationsExist",
  PreconditionFailed: "PreconditionFailed",
  ResourceLimitExceeded: "ResourceLimitExceeded",
  ServiceFailure: "ServiceFailure",
  ServiceUnavailable: "ServiceUnavailable",
  Throttled: "Throttled",
  Throttling: "Throttling",
  Unauthorized: "Unauthorized",
  Unprocessable: "Unprocessable",
  VoiceConnectorGroupAssociationsExist: "VoiceConnectorGroupAssociationsExist"
};
var _BadRequestException = class _BadRequestException extends ChimeSDKIdentityServiceException {
  /**
   * @internal
   */
  constructor(opts) {
    super({
      name: "BadRequestException",
      $fault: "client",
      ...opts
    });
    this.name = "BadRequestException";
    this.$fault = "client";
    Object.setPrototypeOf(this, _BadRequestException.prototype);
    this.Code = opts.Code;
    this.Message = opts.Message;
  }
};
__name(_BadRequestException, "BadRequestException");
var BadRequestException = _BadRequestException;
var _ConflictException = class _ConflictException extends ChimeSDKIdentityServiceException {
  /**
   * @internal
   */
  constructor(opts) {
    super({
      name: "ConflictException",
      $fault: "client",
      ...opts
    });
    this.name = "ConflictException";
    this.$fault = "client";
    Object.setPrototypeOf(this, _ConflictException.prototype);
    this.Code = opts.Code;
    this.Message = opts.Message;
  }
};
__name(_ConflictException, "ConflictException");
var ConflictException = _ConflictException;
var _ForbiddenException = class _ForbiddenException extends ChimeSDKIdentityServiceException {
  /**
   * @internal
   */
  constructor(opts) {
    super({
      name: "ForbiddenException",
      $fault: "client",
      ...opts
    });
    this.name = "ForbiddenException";
    this.$fault = "client";
    Object.setPrototypeOf(this, _ForbiddenException.prototype);
    this.Code = opts.Code;
    this.Message = opts.Message;
  }
};
__name(_ForbiddenException, "ForbiddenException");
var ForbiddenException = _ForbiddenException;
var _ResourceLimitExceededException = class _ResourceLimitExceededException extends ChimeSDKIdentityServiceException {
  /**
   * @internal
   */
  constructor(opts) {
    super({
      name: "ResourceLimitExceededException",
      $fault: "client",
      ...opts
    });
    this.name = "ResourceLimitExceededException";
    this.$fault = "client";
    Object.setPrototypeOf(this, _ResourceLimitExceededException.prototype);
    this.Code = opts.Code;
    this.Message = opts.Message;
  }
};
__name(_ResourceLimitExceededException, "ResourceLimitExceededException");
var ResourceLimitExceededException = _ResourceLimitExceededException;
var _ServiceFailureException = class _ServiceFailureException extends ChimeSDKIdentityServiceException {
  /**
   * @internal
   */
  constructor(opts) {
    super({
      name: "ServiceFailureException",
      $fault: "server",
      ...opts
    });
    this.name = "ServiceFailureException";
    this.$fault = "server";
    Object.setPrototypeOf(this, _ServiceFailureException.prototype);
    this.Code = opts.Code;
    this.Message = opts.Message;
  }
};
__name(_ServiceFailureException, "ServiceFailureException");
var ServiceFailureException = _ServiceFailureException;
var _ServiceUnavailableException = class _ServiceUnavailableException extends ChimeSDKIdentityServiceException {
  /**
   * @internal
   */
  constructor(opts) {
    super({
      name: "ServiceUnavailableException",
      $fault: "server",
      ...opts
    });
    this.name = "ServiceUnavailableException";
    this.$fault = "server";
    Object.setPrototypeOf(this, _ServiceUnavailableException.prototype);
    this.Code = opts.Code;
    this.Message = opts.Message;
  }
};
__name(_ServiceUnavailableException, "ServiceUnavailableException");
var ServiceUnavailableException = _ServiceUnavailableException;
var _ThrottledClientException = class _ThrottledClientException extends ChimeSDKIdentityServiceException {
  /**
   * @internal
   */
  constructor(opts) {
    super({
      name: "ThrottledClientException",
      $fault: "client",
      ...opts
    });
    this.name = "ThrottledClientException";
    this.$fault = "client";
    Object.setPrototypeOf(this, _ThrottledClientException.prototype);
    this.Code = opts.Code;
    this.Message = opts.Message;
  }
};
__name(_ThrottledClientException, "ThrottledClientException");
var ThrottledClientException = _ThrottledClientException;
var _UnauthorizedClientException = class _UnauthorizedClientException extends ChimeSDKIdentityServiceException {
  /**
   * @internal
   */
  constructor(opts) {
    super({
      name: "UnauthorizedClientException",
      $fault: "client",
      ...opts
    });
    this.name = "UnauthorizedClientException";
    this.$fault = "client";
    Object.setPrototypeOf(this, _UnauthorizedClientException.prototype);
    this.Code = opts.Code;
    this.Message = opts.Message;
  }
};
__name(_UnauthorizedClientException, "UnauthorizedClientException");
var UnauthorizedClientException = _UnauthorizedClientException;
var _NotFoundException = class _NotFoundException extends ChimeSDKIdentityServiceException {
  /**
   * @internal
   */
  constructor(opts) {
    super({
      name: "NotFoundException",
      $fault: "client",
      ...opts
    });
    this.name = "NotFoundException";
    this.$fault = "client";
    Object.setPrototypeOf(this, _NotFoundException.prototype);
    this.Code = opts.Code;
    this.Message = opts.Message;
  }
};
__name(_NotFoundException, "NotFoundException");
var NotFoundException = _NotFoundException;
var AppInstanceFilterSensitiveLog = /* @__PURE__ */ __name((obj) => ({
  ...obj,
  ...obj.Name && { Name: import_smithy_client.SENSITIVE_STRING },
  ...obj.Metadata && { Metadata: import_smithy_client.SENSITIVE_STRING }
}), "AppInstanceFilterSensitiveLog");
var IdentityFilterSensitiveLog = /* @__PURE__ */ __name((obj) => ({
  ...obj,
  ...obj.Name && { Name: import_smithy_client.SENSITIVE_STRING }
}), "IdentityFilterSensitiveLog");
var AppInstanceAdminFilterSensitiveLog = /* @__PURE__ */ __name((obj) => ({
  ...obj,
  ...obj.Admin && { Admin: IdentityFilterSensitiveLog(obj.Admin) }
}), "AppInstanceAdminFilterSensitiveLog");
var AppInstanceAdminSummaryFilterSensitiveLog = /* @__PURE__ */ __name((obj) => ({
  ...obj,
  ...obj.Admin && { Admin: IdentityFilterSensitiveLog(obj.Admin) }
}), "AppInstanceAdminSummaryFilterSensitiveLog");
var AppInstanceBotFilterSensitiveLog = /* @__PURE__ */ __name((obj) => ({
  ...obj,
  ...obj.Name && { Name: import_smithy_client.SENSITIVE_STRING },
  ...obj.Metadata && { Metadata: import_smithy_client.SENSITIVE_STRING }
}), "AppInstanceBotFilterSensitiveLog");
var AppInstanceBotSummaryFilterSensitiveLog = /* @__PURE__ */ __name((obj) => ({
  ...obj,
  ...obj.Name && { Name: import_smithy_client.SENSITIVE_STRING },
  ...obj.Metadata && { Metadata: import_smithy_client.SENSITIVE_STRING }
}), "AppInstanceBotSummaryFilterSensitiveLog");
var AppInstanceSummaryFilterSensitiveLog = /* @__PURE__ */ __name((obj) => ({
  ...obj,
  ...obj.Name && { Name: import_smithy_client.SENSITIVE_STRING },
  ...obj.Metadata && { Metadata: import_smithy_client.SENSITIVE_STRING }
}), "AppInstanceSummaryFilterSensitiveLog");
var AppInstanceUserFilterSensitiveLog = /* @__PURE__ */ __name((obj) => ({
  ...obj,
  ...obj.Name && { Name: import_smithy_client.SENSITIVE_STRING },
  ...obj.Metadata && { Metadata: import_smithy_client.SENSITIVE_STRING }
}), "AppInstanceUserFilterSensitiveLog");
var EndpointAttributesFilterSensitiveLog = /* @__PURE__ */ __name((obj) => ({
  ...obj,
  ...obj.DeviceToken && { DeviceToken: import_smithy_client.SENSITIVE_STRING },
  ...obj.VoipDeviceToken && { VoipDeviceToken: import_smithy_client.SENSITIVE_STRING }
}), "EndpointAttributesFilterSensitiveLog");
var AppInstanceUserEndpointFilterSensitiveLog = /* @__PURE__ */ __name((obj) => ({
  ...obj,
  ...obj.Name && { Name: import_smithy_client.SENSITIVE_STRING },
  ...obj.EndpointAttributes && { EndpointAttributes: EndpointAttributesFilterSensitiveLog(obj.EndpointAttributes) }
}), "AppInstanceUserEndpointFilterSensitiveLog");
var AppInstanceUserEndpointSummaryFilterSensitiveLog = /* @__PURE__ */ __name((obj) => ({
  ...obj,
  ...obj.Name && { Name: import_smithy_client.SENSITIVE_STRING }
}), "AppInstanceUserEndpointSummaryFilterSensitiveLog");
var AppInstanceUserSummaryFilterSensitiveLog = /* @__PURE__ */ __name((obj) => ({
  ...obj,
  ...obj.Name && { Name: import_smithy_client.SENSITIVE_STRING },
  ...obj.Metadata && { Metadata: import_smithy_client.SENSITIVE_STRING }
}), "AppInstanceUserSummaryFilterSensitiveLog");
var TagFilterSensitiveLog = /* @__PURE__ */ __name((obj) => ({
  ...obj,
  ...obj.Key && { Key: import_smithy_client.SENSITIVE_STRING },
  ...obj.Value && { Value: import_smithy_client.SENSITIVE_STRING }
}), "TagFilterSensitiveLog");
var CreateAppInstanceRequestFilterSensitiveLog = /* @__PURE__ */ __name((obj) => ({
  ...obj,
  ...obj.Name && { Name: import_smithy_client.SENSITIVE_STRING },
  ...obj.Metadata && { Metadata: import_smithy_client.SENSITIVE_STRING },
  ...obj.Tags && { Tags: obj.Tags.map((item) => TagFilterSensitiveLog(item)) }
}), "CreateAppInstanceRequestFilterSensitiveLog");
var CreateAppInstanceAdminResponseFilterSensitiveLog = /* @__PURE__ */ __name((obj) => ({
  ...obj,
  ...obj.AppInstanceAdmin && { AppInstanceAdmin: IdentityFilterSensitiveLog(obj.AppInstanceAdmin) }
}), "CreateAppInstanceAdminResponseFilterSensitiveLog");
var CreateAppInstanceBotRequestFilterSensitiveLog = /* @__PURE__ */ __name((obj) => ({
  ...obj,
  ...obj.Name && { Name: import_smithy_client.SENSITIVE_STRING },
  ...obj.Metadata && { Metadata: import_smithy_client.SENSITIVE_STRING },
  ...obj.Tags && { Tags: obj.Tags.map((item) => TagFilterSensitiveLog(item)) }
}), "CreateAppInstanceBotRequestFilterSensitiveLog");
var CreateAppInstanceUserRequestFilterSensitiveLog = /* @__PURE__ */ __name((obj) => ({
  ...obj,
  ...obj.AppInstanceUserId && { AppInstanceUserId: import_smithy_client.SENSITIVE_STRING },
  ...obj.Name && { Name: import_smithy_client.SENSITIVE_STRING },
  ...obj.Metadata && { Metadata: import_smithy_client.SENSITIVE_STRING },
  ...obj.Tags && { Tags: obj.Tags.map((item) => TagFilterSensitiveLog(item)) }
}), "CreateAppInstanceUserRequestFilterSensitiveLog");
var DescribeAppInstanceResponseFilterSensitiveLog = /* @__PURE__ */ __name((obj) => ({
  ...obj,
  ...obj.AppInstance && { AppInstance: AppInstanceFilterSensitiveLog(obj.AppInstance) }
}), "DescribeAppInstanceResponseFilterSensitiveLog");
var DescribeAppInstanceAdminResponseFilterSensitiveLog = /* @__PURE__ */ __name((obj) => ({
  ...obj,
  ...obj.AppInstanceAdmin && { AppInstanceAdmin: AppInstanceAdminFilterSensitiveLog(obj.AppInstanceAdmin) }
}), "DescribeAppInstanceAdminResponseFilterSensitiveLog");
var DescribeAppInstanceBotResponseFilterSensitiveLog = /* @__PURE__ */ __name((obj) => ({
  ...obj,
  ...obj.AppInstanceBot && { AppInstanceBot: AppInstanceBotFilterSensitiveLog(obj.AppInstanceBot) }
}), "DescribeAppInstanceBotResponseFilterSensitiveLog");
var DescribeAppInstanceUserResponseFilterSensitiveLog = /* @__PURE__ */ __name((obj) => ({
  ...obj,
  ...obj.AppInstanceUser && { AppInstanceUser: AppInstanceUserFilterSensitiveLog(obj.AppInstanceUser) }
}), "DescribeAppInstanceUserResponseFilterSensitiveLog");
var DescribeAppInstanceUserEndpointResponseFilterSensitiveLog = /* @__PURE__ */ __name((obj) => ({
  ...obj,
  ...obj.AppInstanceUserEndpoint && {
    AppInstanceUserEndpoint: AppInstanceUserEndpointFilterSensitiveLog(obj.AppInstanceUserEndpoint)
  }
}), "DescribeAppInstanceUserEndpointResponseFilterSensitiveLog");
var ListAppInstanceAdminsRequestFilterSensitiveLog = /* @__PURE__ */ __name((obj) => ({
  ...obj,
  ...obj.NextToken && { NextToken: import_smithy_client.SENSITIVE_STRING }
}), "ListAppInstanceAdminsRequestFilterSensitiveLog");
var ListAppInstanceAdminsResponseFilterSensitiveLog = /* @__PURE__ */ __name((obj) => ({
  ...obj,
  ...obj.AppInstanceAdmins && {
    AppInstanceAdmins: obj.AppInstanceAdmins.map((item) => AppInstanceAdminSummaryFilterSensitiveLog(item))
  },
  ...obj.NextToken && { NextToken: import_smithy_client.SENSITIVE_STRING }
}), "ListAppInstanceAdminsResponseFilterSensitiveLog");
var ListAppInstanceBotsRequestFilterSensitiveLog = /* @__PURE__ */ __name((obj) => ({
  ...obj,
  ...obj.NextToken && { NextToken: import_smithy_client.SENSITIVE_STRING }
}), "ListAppInstanceBotsRequestFilterSensitiveLog");
var ListAppInstanceBotsResponseFilterSensitiveLog = /* @__PURE__ */ __name((obj) => ({
  ...obj,
  ...obj.AppInstanceBots && {
    AppInstanceBots: obj.AppInstanceBots.map((item) => AppInstanceBotSummaryFilterSensitiveLog(item))
  },
  ...obj.NextToken && { NextToken: import_smithy_client.SENSITIVE_STRING }
}), "ListAppInstanceBotsResponseFilterSensitiveLog");
var ListAppInstancesRequestFilterSensitiveLog = /* @__PURE__ */ __name((obj) => ({
  ...obj,
  ...obj.NextToken && { NextToken: import_smithy_client.SENSITIVE_STRING }
}), "ListAppInstancesRequestFilterSensitiveLog");
var ListAppInstancesResponseFilterSensitiveLog = /* @__PURE__ */ __name((obj) => ({
  ...obj,
  ...obj.AppInstances && { AppInstances: obj.AppInstances.map((item) => AppInstanceSummaryFilterSensitiveLog(item)) },
  ...obj.NextToken && { NextToken: import_smithy_client.SENSITIVE_STRING }
}), "ListAppInstancesResponseFilterSensitiveLog");
var ListAppInstanceUserEndpointsRequestFilterSensitiveLog = /* @__PURE__ */ __name((obj) => ({
  ...obj,
  ...obj.AppInstanceUserArn && { AppInstanceUserArn: import_smithy_client.SENSITIVE_STRING },
  ...obj.NextToken && { NextToken: import_smithy_client.SENSITIVE_STRING }
}), "ListAppInstanceUserEndpointsRequestFilterSensitiveLog");
var ListAppInstanceUserEndpointsResponseFilterSensitiveLog = /* @__PURE__ */ __name((obj) => ({
  ...obj,
  ...obj.AppInstanceUserEndpoints && {
    AppInstanceUserEndpoints: obj.AppInstanceUserEndpoints.map(
      (item) => AppInstanceUserEndpointSummaryFilterSensitiveLog(item)
    )
  },
  ...obj.NextToken && { NextToken: import_smithy_client.SENSITIVE_STRING }
}), "ListAppInstanceUserEndpointsResponseFilterSensitiveLog");
var ListAppInstanceUsersRequestFilterSensitiveLog = /* @__PURE__ */ __name((obj) => ({
  ...obj,
  ...obj.NextToken && { NextToken: import_smithy_client.SENSITIVE_STRING }
}), "ListAppInstanceUsersRequestFilterSensitiveLog");
var ListAppInstanceUsersResponseFilterSensitiveLog = /* @__PURE__ */ __name((obj) => ({
  ...obj,
  ...obj.AppInstanceUsers && {
    AppInstanceUsers: obj.AppInstanceUsers.map((item) => AppInstanceUserSummaryFilterSensitiveLog(item))
  },
  ...obj.NextToken && { NextToken: import_smithy_client.SENSITIVE_STRING }
}), "ListAppInstanceUsersResponseFilterSensitiveLog");
var ListTagsForResourceResponseFilterSensitiveLog = /* @__PURE__ */ __name((obj) => ({
  ...obj,
  ...obj.Tags && { Tags: obj.Tags.map((item) => TagFilterSensitiveLog(item)) }
}), "ListTagsForResourceResponseFilterSensitiveLog");
var RegisterAppInstanceUserEndpointRequestFilterSensitiveLog = /* @__PURE__ */ __name((obj) => ({
  ...obj,
  ...obj.AppInstanceUserArn && { AppInstanceUserArn: import_smithy_client.SENSITIVE_STRING },
  ...obj.Name && { Name: import_smithy_client.SENSITIVE_STRING },
  ...obj.EndpointAttributes && { EndpointAttributes: EndpointAttributesFilterSensitiveLog(obj.EndpointAttributes) }
}), "RegisterAppInstanceUserEndpointRequestFilterSensitiveLog");
var TagResourceRequestFilterSensitiveLog = /* @__PURE__ */ __name((obj) => ({
  ...obj,
  ...obj.Tags && { Tags: obj.Tags.map((item) => TagFilterSensitiveLog(item)) }
}), "TagResourceRequestFilterSensitiveLog");
var UntagResourceRequestFilterSensitiveLog = /* @__PURE__ */ __name((obj) => ({
  ...obj,
  ...obj.TagKeys && { TagKeys: import_smithy_client.SENSITIVE_STRING }
}), "UntagResourceRequestFilterSensitiveLog");
var UpdateAppInstanceRequestFilterSensitiveLog = /* @__PURE__ */ __name((obj) => ({
  ...obj,
  ...obj.Name && { Name: import_smithy_client.SENSITIVE_STRING },
  ...obj.Metadata && { Metadata: import_smithy_client.SENSITIVE_STRING }
}), "UpdateAppInstanceRequestFilterSensitiveLog");
var UpdateAppInstanceBotRequestFilterSensitiveLog = /* @__PURE__ */ __name((obj) => ({
  ...obj,
  ...obj.Name && { Name: import_smithy_client.SENSITIVE_STRING },
  ...obj.Metadata && { Metadata: import_smithy_client.SENSITIVE_STRING }
}), "UpdateAppInstanceBotRequestFilterSensitiveLog");
var UpdateAppInstanceUserRequestFilterSensitiveLog = /* @__PURE__ */ __name((obj) => ({
  ...obj,
  ...obj.Name && { Name: import_smithy_client.SENSITIVE_STRING },
  ...obj.Metadata && { Metadata: import_smithy_client.SENSITIVE_STRING }
}), "UpdateAppInstanceUserRequestFilterSensitiveLog");
var UpdateAppInstanceUserEndpointRequestFilterSensitiveLog = /* @__PURE__ */ __name((obj) => ({
  ...obj,
  ...obj.Name && { Name: import_smithy_client.SENSITIVE_STRING }
}), "UpdateAppInstanceUserEndpointRequestFilterSensitiveLog");

// src/protocols/Aws_restJson1.ts
var import_core2 = require("@aws-sdk/core");


var import_uuid = require("uuid");
var se_CreateAppInstanceCommand = /* @__PURE__ */ __name(async (input, context) => {
  const b = (0, import_core.requestBuilder)(input, context);
  const headers = {
    "content-type": "application/json"
  };
  b.bp("/app-instances");
  let body;
  body = JSON.stringify(
    (0, import_smithy_client.take)(input, {
      ClientRequestToken: [true, (_) => _ ?? (0, import_uuid.v4)()],
      Metadata: [],
      Name: [],
      Tags: (_) => (0, import_smithy_client._json)(_)
    })
  );
  b.m("POST").h(headers).b(body);
  return b.build();
}, "se_CreateAppInstanceCommand");
var se_CreateAppInstanceAdminCommand = /* @__PURE__ */ __name(async (input, context) => {
  const b = (0, import_core.requestBuilder)(input, context);
  const headers = {
    "content-type": "application/json"
  };
  b.bp("/app-instances/{AppInstanceArn}/admins");
  b.p("AppInstanceArn", () => input.AppInstanceArn, "{AppInstanceArn}", false);
  let body;
  body = JSON.stringify(
    (0, import_smithy_client.take)(input, {
      AppInstanceAdminArn: []
    })
  );
  b.m("POST").h(headers).b(body);
  return b.build();
}, "se_CreateAppInstanceAdminCommand");
var se_CreateAppInstanceBotCommand = /* @__PURE__ */ __name(async (input, context) => {
  const b = (0, import_core.requestBuilder)(input, context);
  const headers = {
    "content-type": "application/json"
  };
  b.bp("/app-instance-bots");
  let body;
  body = JSON.stringify(
    (0, import_smithy_client.take)(input, {
      AppInstanceArn: [],
      ClientRequestToken: [true, (_) => _ ?? (0, import_uuid.v4)()],
      Configuration: (_) => (0, import_smithy_client._json)(_),
      Metadata: [],
      Name: [],
      Tags: (_) => (0, import_smithy_client._json)(_)
    })
  );
  b.m("POST").h(headers).b(body);
  return b.build();
}, "se_CreateAppInstanceBotCommand");
var se_CreateAppInstanceUserCommand = /* @__PURE__ */ __name(async (input, context) => {
  const b = (0, import_core.requestBuilder)(input, context);
  const headers = {
    "content-type": "application/json"
  };
  b.bp("/app-instance-users");
  let body;
  body = JSON.stringify(
    (0, import_smithy_client.take)(input, {
      AppInstanceArn: [],
      AppInstanceUserId: [],
      ClientRequestToken: [true, (_) => _ ?? (0, import_uuid.v4)()],
      ExpirationSettings: (_) => (0, import_smithy_client._json)(_),
      Metadata: [],
      Name: [],
      Tags: (_) => (0, import_smithy_client._json)(_)
    })
  );
  b.m("POST").h(headers).b(body);
  return b.build();
}, "se_CreateAppInstanceUserCommand");
var se_DeleteAppInstanceCommand = /* @__PURE__ */ __name(async (input, context) => {
  const b = (0, import_core.requestBuilder)(input, context);
  const headers = {};
  b.bp("/app-instances/{AppInstanceArn}");
  b.p("AppInstanceArn", () => input.AppInstanceArn, "{AppInstanceArn}", false);
  let body;
  b.m("DELETE").h(headers).b(body);
  return b.build();
}, "se_DeleteAppInstanceCommand");
var se_DeleteAppInstanceAdminCommand = /* @__PURE__ */ __name(async (input, context) => {
  const b = (0, import_core.requestBuilder)(input, context);
  const headers = {};
  b.bp("/app-instances/{AppInstanceArn}/admins/{AppInstanceAdminArn}");
  b.p("AppInstanceAdminArn", () => input.AppInstanceAdminArn, "{AppInstanceAdminArn}", false);
  b.p("AppInstanceArn", () => input.AppInstanceArn, "{AppInstanceArn}", false);
  let body;
  b.m("DELETE").h(headers).b(body);
  return b.build();
}, "se_DeleteAppInstanceAdminCommand");
var se_DeleteAppInstanceBotCommand = /* @__PURE__ */ __name(async (input, context) => {
  const b = (0, import_core.requestBuilder)(input, context);
  const headers = {};
  b.bp("/app-instance-bots/{AppInstanceBotArn}");
  b.p("AppInstanceBotArn", () => input.AppInstanceBotArn, "{AppInstanceBotArn}", false);
  let body;
  b.m("DELETE").h(headers).b(body);
  return b.build();
}, "se_DeleteAppInstanceBotCommand");
var se_DeleteAppInstanceUserCommand = /* @__PURE__ */ __name(async (input, context) => {
  const b = (0, import_core.requestBuilder)(input, context);
  const headers = {};
  b.bp("/app-instance-users/{AppInstanceUserArn}");
  b.p("AppInstanceUserArn", () => input.AppInstanceUserArn, "{AppInstanceUserArn}", false);
  let body;
  b.m("DELETE").h(headers).b(body);
  return b.build();
}, "se_DeleteAppInstanceUserCommand");
var se_DeregisterAppInstanceUserEndpointCommand = /* @__PURE__ */ __name(async (input, context) => {
  const b = (0, import_core.requestBuilder)(input, context);
  const headers = {};
  b.bp("/app-instance-users/{AppInstanceUserArn}/endpoints/{EndpointId}");
  b.p("AppInstanceUserArn", () => input.AppInstanceUserArn, "{AppInstanceUserArn}", false);
  b.p("EndpointId", () => input.EndpointId, "{EndpointId}", false);
  let body;
  b.m("DELETE").h(headers).b(body);
  return b.build();
}, "se_DeregisterAppInstanceUserEndpointCommand");
var se_DescribeAppInstanceCommand = /* @__PURE__ */ __name(async (input, context) => {
  const b = (0, import_core.requestBuilder)(input, context);
  const headers = {};
  b.bp("/app-instances/{AppInstanceArn}");
  b.p("AppInstanceArn", () => input.AppInstanceArn, "{AppInstanceArn}", false);
  let body;
  b.m("GET").h(headers).b(body);
  return b.build();
}, "se_DescribeAppInstanceCommand");
var se_DescribeAppInstanceAdminCommand = /* @__PURE__ */ __name(async (input, context) => {
  const b = (0, import_core.requestBuilder)(input, context);
  const headers = {};
  b.bp("/app-instances/{AppInstanceArn}/admins/{AppInstanceAdminArn}");
  b.p("AppInstanceAdminArn", () => input.AppInstanceAdminArn, "{AppInstanceAdminArn}", false);
  b.p("AppInstanceArn", () => input.AppInstanceArn, "{AppInstanceArn}", false);
  let body;
  b.m("GET").h(headers).b(body);
  return b.build();
}, "se_DescribeAppInstanceAdminCommand");
var se_DescribeAppInstanceBotCommand = /* @__PURE__ */ __name(async (input, context) => {
  const b = (0, import_core.requestBuilder)(input, context);
  const headers = {};
  b.bp("/app-instance-bots/{AppInstanceBotArn}");
  b.p("AppInstanceBotArn", () => input.AppInstanceBotArn, "{AppInstanceBotArn}", false);
  let body;
  b.m("GET").h(headers).b(body);
  return b.build();
}, "se_DescribeAppInstanceBotCommand");
var se_DescribeAppInstanceUserCommand = /* @__PURE__ */ __name(async (input, context) => {
  const b = (0, import_core.requestBuilder)(input, context);
  const headers = {};
  b.bp("/app-instance-users/{AppInstanceUserArn}");
  b.p("AppInstanceUserArn", () => input.AppInstanceUserArn, "{AppInstanceUserArn}", false);
  let body;
  b.m("GET").h(headers).b(body);
  return b.build();
}, "se_DescribeAppInstanceUserCommand");
var se_DescribeAppInstanceUserEndpointCommand = /* @__PURE__ */ __name(async (input, context) => {
  const b = (0, import_core.requestBuilder)(input, context);
  const headers = {};
  b.bp("/app-instance-users/{AppInstanceUserArn}/endpoints/{EndpointId}");
  b.p("AppInstanceUserArn", () => input.AppInstanceUserArn, "{AppInstanceUserArn}", false);
  b.p("EndpointId", () => input.EndpointId, "{EndpointId}", false);
  let body;
  b.m("GET").h(headers).b(body);
  return b.build();
}, "se_DescribeAppInstanceUserEndpointCommand");
var se_GetAppInstanceRetentionSettingsCommand = /* @__PURE__ */ __name(async (input, context) => {
  const b = (0, import_core.requestBuilder)(input, context);
  const headers = {};
  b.bp("/app-instances/{AppInstanceArn}/retention-settings");
  b.p("AppInstanceArn", () => input.AppInstanceArn, "{AppInstanceArn}", false);
  let body;
  b.m("GET").h(headers).b(body);
  return b.build();
}, "se_GetAppInstanceRetentionSettingsCommand");
var se_ListAppInstanceAdminsCommand = /* @__PURE__ */ __name(async (input, context) => {
  const b = (0, import_core.requestBuilder)(input, context);
  const headers = {};
  b.bp("/app-instances/{AppInstanceArn}/admins");
  b.p("AppInstanceArn", () => input.AppInstanceArn, "{AppInstanceArn}", false);
  const query = (0, import_smithy_client.map)({
    [_mr]: [() => input.MaxResults !== void 0, () => input[_MR].toString()],
    [_nt]: [, input[_NT]]
  });
  let body;
  b.m("GET").h(headers).q(query).b(body);
  return b.build();
}, "se_ListAppInstanceAdminsCommand");
var se_ListAppInstanceBotsCommand = /* @__PURE__ */ __name(async (input, context) => {
  const b = (0, import_core.requestBuilder)(input, context);
  const headers = {};
  b.bp("/app-instance-bots");
  const query = (0, import_smithy_client.map)({
    [_aia]: [, (0, import_smithy_client.expectNonNull)(input[_AIA], `AppInstanceArn`)],
    [_mr]: [() => input.MaxResults !== void 0, () => input[_MR].toString()],
    [_nt]: [, input[_NT]]
  });
  let body;
  b.m("GET").h(headers).q(query).b(body);
  return b.build();
}, "se_ListAppInstanceBotsCommand");
var se_ListAppInstancesCommand = /* @__PURE__ */ __name(async (input, context) => {
  const b = (0, import_core.requestBuilder)(input, context);
  const headers = {};
  b.bp("/app-instances");
  const query = (0, import_smithy_client.map)({
    [_mr]: [() => input.MaxResults !== void 0, () => input[_MR].toString()],
    [_nt]: [, input[_NT]]
  });
  let body;
  b.m("GET").h(headers).q(query).b(body);
  return b.build();
}, "se_ListAppInstancesCommand");
var se_ListAppInstanceUserEndpointsCommand = /* @__PURE__ */ __name(async (input, context) => {
  const b = (0, import_core.requestBuilder)(input, context);
  const headers = {};
  b.bp("/app-instance-users/{AppInstanceUserArn}/endpoints");
  b.p("AppInstanceUserArn", () => input.AppInstanceUserArn, "{AppInstanceUserArn}", false);
  const query = (0, import_smithy_client.map)({
    [_mr]: [() => input.MaxResults !== void 0, () => input[_MR].toString()],
    [_nt]: [, input[_NT]]
  });
  let body;
  b.m("GET").h(headers).q(query).b(body);
  return b.build();
}, "se_ListAppInstanceUserEndpointsCommand");
var se_ListAppInstanceUsersCommand = /* @__PURE__ */ __name(async (input, context) => {
  const b = (0, import_core.requestBuilder)(input, context);
  const headers = {};
  b.bp("/app-instance-users");
  const query = (0, import_smithy_client.map)({
    [_aia]: [, (0, import_smithy_client.expectNonNull)(input[_AIA], `AppInstanceArn`)],
    [_mr]: [() => input.MaxResults !== void 0, () => input[_MR].toString()],
    [_nt]: [, input[_NT]]
  });
  let body;
  b.m("GET").h(headers).q(query).b(body);
  return b.build();
}, "se_ListAppInstanceUsersCommand");
var se_ListTagsForResourceCommand = /* @__PURE__ */ __name(async (input, context) => {
  const b = (0, import_core.requestBuilder)(input, context);
  const headers = {};
  b.bp("/tags");
  const query = (0, import_smithy_client.map)({
    [_a]: [, (0, import_smithy_client.expectNonNull)(input[_RARN], `ResourceARN`)]
  });
  let body;
  b.m("GET").h(headers).q(query).b(body);
  return b.build();
}, "se_ListTagsForResourceCommand");
var se_PutAppInstanceRetentionSettingsCommand = /* @__PURE__ */ __name(async (input, context) => {
  const b = (0, import_core.requestBuilder)(input, context);
  const headers = {
    "content-type": "application/json"
  };
  b.bp("/app-instances/{AppInstanceArn}/retention-settings");
  b.p("AppInstanceArn", () => input.AppInstanceArn, "{AppInstanceArn}", false);
  let body;
  body = JSON.stringify(
    (0, import_smithy_client.take)(input, {
      AppInstanceRetentionSettings: (_) => (0, import_smithy_client._json)(_)
    })
  );
  b.m("PUT").h(headers).b(body);
  return b.build();
}, "se_PutAppInstanceRetentionSettingsCommand");
var se_PutAppInstanceUserExpirationSettingsCommand = /* @__PURE__ */ __name(async (input, context) => {
  const b = (0, import_core.requestBuilder)(input, context);
  const headers = {
    "content-type": "application/json"
  };
  b.bp("/app-instance-users/{AppInstanceUserArn}/expiration-settings");
  b.p("AppInstanceUserArn", () => input.AppInstanceUserArn, "{AppInstanceUserArn}", false);
  let body;
  body = JSON.stringify(
    (0, import_smithy_client.take)(input, {
      ExpirationSettings: (_) => (0, import_smithy_client._json)(_)
    })
  );
  b.m("PUT").h(headers).b(body);
  return b.build();
}, "se_PutAppInstanceUserExpirationSettingsCommand");
var se_RegisterAppInstanceUserEndpointCommand = /* @__PURE__ */ __name(async (input, context) => {
  const b = (0, import_core.requestBuilder)(input, context);
  const headers = {
    "content-type": "application/json"
  };
  b.bp("/app-instance-users/{AppInstanceUserArn}/endpoints");
  b.p("AppInstanceUserArn", () => input.AppInstanceUserArn, "{AppInstanceUserArn}", false);
  let body;
  body = JSON.stringify(
    (0, import_smithy_client.take)(input, {
      AllowMessages: [],
      ClientRequestToken: [true, (_) => _ ?? (0, import_uuid.v4)()],
      EndpointAttributes: (_) => (0, import_smithy_client._json)(_),
      Name: [],
      ResourceArn: [],
      Type: []
    })
  );
  b.m("POST").h(headers).b(body);
  return b.build();
}, "se_RegisterAppInstanceUserEndpointCommand");
var se_TagResourceCommand = /* @__PURE__ */ __name(async (input, context) => {
  const b = (0, import_core.requestBuilder)(input, context);
  const headers = {
    "content-type": "application/json"
  };
  b.bp("/tags");
  const query = (0, import_smithy_client.map)({
    [_o]: [, "tag-resource"]
  });
  let body;
  body = JSON.stringify(
    (0, import_smithy_client.take)(input, {
      ResourceARN: [],
      Tags: (_) => (0, import_smithy_client._json)(_)
    })
  );
  b.m("POST").h(headers).q(query).b(body);
  return b.build();
}, "se_TagResourceCommand");
var se_UntagResourceCommand = /* @__PURE__ */ __name(async (input, context) => {
  const b = (0, import_core.requestBuilder)(input, context);
  const headers = {
    "content-type": "application/json"
  };
  b.bp("/tags");
  const query = (0, import_smithy_client.map)({
    [_o]: [, "untag-resource"]
  });
  let body;
  body = JSON.stringify(
    (0, import_smithy_client.take)(input, {
      ResourceARN: [],
      TagKeys: (_) => (0, import_smithy_client._json)(_)
    })
  );
  b.m("POST").h(headers).q(query).b(body);
  return b.build();
}, "se_UntagResourceCommand");
var se_UpdateAppInstanceCommand = /* @__PURE__ */ __name(async (input, context) => {
  const b = (0, import_core.requestBuilder)(input, context);
  const headers = {
    "content-type": "application/json"
  };
  b.bp("/app-instances/{AppInstanceArn}");
  b.p("AppInstanceArn", () => input.AppInstanceArn, "{AppInstanceArn}", false);
  let body;
  body = JSON.stringify(
    (0, import_smithy_client.take)(input, {
      Metadata: [],
      Name: []
    })
  );
  b.m("PUT").h(headers).b(body);
  return b.build();
}, "se_UpdateAppInstanceCommand");
var se_UpdateAppInstanceBotCommand = /* @__PURE__ */ __name(async (input, context) => {
  const b = (0, import_core.requestBuilder)(input, context);
  const headers = {
    "content-type": "application/json"
  };
  b.bp("/app-instance-bots/{AppInstanceBotArn}");
  b.p("AppInstanceBotArn", () => input.AppInstanceBotArn, "{AppInstanceBotArn}", false);
  let body;
  body = JSON.stringify(
    (0, import_smithy_client.take)(input, {
      Configuration: (_) => (0, import_smithy_client._json)(_),
      Metadata: [],
      Name: []
    })
  );
  b.m("PUT").h(headers).b(body);
  return b.build();
}, "se_UpdateAppInstanceBotCommand");
var se_UpdateAppInstanceUserCommand = /* @__PURE__ */ __name(async (input, context) => {
  const b = (0, import_core.requestBuilder)(input, context);
  const headers = {
    "content-type": "application/json"
  };
  b.bp("/app-instance-users/{AppInstanceUserArn}");
  b.p("AppInstanceUserArn", () => input.AppInstanceUserArn, "{AppInstanceUserArn}", false);
  let body;
  body = JSON.stringify(
    (0, import_smithy_client.take)(input, {
      Metadata: [],
      Name: []
    })
  );
  b.m("PUT").h(headers).b(body);
  return b.build();
}, "se_UpdateAppInstanceUserCommand");
var se_UpdateAppInstanceUserEndpointCommand = /* @__PURE__ */ __name(async (input, context) => {
  const b = (0, import_core.requestBuilder)(input, context);
  const headers = {
    "content-type": "application/json"
  };
  b.bp("/app-instance-users/{AppInstanceUserArn}/endpoints/{EndpointId}");
  b.p("AppInstanceUserArn", () => input.AppInstanceUserArn, "{AppInstanceUserArn}", false);
  b.p("EndpointId", () => input.EndpointId, "{EndpointId}", false);
  let body;
  body = JSON.stringify(
    (0, import_smithy_client.take)(input, {
      AllowMessages: [],
      Name: []
    })
  );
  b.m("PUT").h(headers).b(body);
  return b.build();
}, "se_UpdateAppInstanceUserEndpointCommand");
var de_CreateAppInstanceCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode !== 201 && output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const contents = (0, import_smithy_client.map)({
    $metadata: deserializeMetadata(output)
  });
  const data = (0, import_smithy_client.expectNonNull)((0, import_smithy_client.expectObject)(await (0, import_core2.parseJsonBody)(output.body, context)), "body");
  const doc = (0, import_smithy_client.take)(data, {
    AppInstanceArn: import_smithy_client.expectString
  });
  Object.assign(contents, doc);
  return contents;
}, "de_CreateAppInstanceCommand");
var de_CreateAppInstanceAdminCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode !== 201 && output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const contents = (0, import_smithy_client.map)({
    $metadata: deserializeMetadata(output)
  });
  const data = (0, import_smithy_client.expectNonNull)((0, import_smithy_client.expectObject)(await (0, import_core2.parseJsonBody)(output.body, context)), "body");
  const doc = (0, import_smithy_client.take)(data, {
    AppInstanceAdmin: import_smithy_client._json,
    AppInstanceArn: import_smithy_client.expectString
  });
  Object.assign(contents, doc);
  return contents;
}, "de_CreateAppInstanceAdminCommand");
var de_CreateAppInstanceBotCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode !== 201 && output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const contents = (0, import_smithy_client.map)({
    $metadata: deserializeMetadata(output)
  });
  const data = (0, import_smithy_client.expectNonNull)((0, import_smithy_client.expectObject)(await (0, import_core2.parseJsonBody)(output.body, context)), "body");
  const doc = (0, import_smithy_client.take)(data, {
    AppInstanceBotArn: import_smithy_client.expectString
  });
  Object.assign(contents, doc);
  return contents;
}, "de_CreateAppInstanceBotCommand");
var de_CreateAppInstanceUserCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode !== 201 && output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const contents = (0, import_smithy_client.map)({
    $metadata: deserializeMetadata(output)
  });
  const data = (0, import_smithy_client.expectNonNull)((0, import_smithy_client.expectObject)(await (0, import_core2.parseJsonBody)(output.body, context)), "body");
  const doc = (0, import_smithy_client.take)(data, {
    AppInstanceUserArn: import_smithy_client.expectString
  });
  Object.assign(contents, doc);
  return contents;
}, "de_CreateAppInstanceUserCommand");
var de_DeleteAppInstanceCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode !== 204 && output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const contents = (0, import_smithy_client.map)({
    $metadata: deserializeMetadata(output)
  });
  await (0, import_smithy_client.collectBody)(output.body, context);
  return contents;
}, "de_DeleteAppInstanceCommand");
var de_DeleteAppInstanceAdminCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode !== 204 && output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const contents = (0, import_smithy_client.map)({
    $metadata: deserializeMetadata(output)
  });
  await (0, import_smithy_client.collectBody)(output.body, context);
  return contents;
}, "de_DeleteAppInstanceAdminCommand");
var de_DeleteAppInstanceBotCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode !== 204 && output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const contents = (0, import_smithy_client.map)({
    $metadata: deserializeMetadata(output)
  });
  await (0, import_smithy_client.collectBody)(output.body, context);
  return contents;
}, "de_DeleteAppInstanceBotCommand");
var de_DeleteAppInstanceUserCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode !== 204 && output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const contents = (0, import_smithy_client.map)({
    $metadata: deserializeMetadata(output)
  });
  await (0, import_smithy_client.collectBody)(output.body, context);
  return contents;
}, "de_DeleteAppInstanceUserCommand");
var de_DeregisterAppInstanceUserEndpointCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode !== 204 && output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const contents = (0, import_smithy_client.map)({
    $metadata: deserializeMetadata(output)
  });
  await (0, import_smithy_client.collectBody)(output.body, context);
  return contents;
}, "de_DeregisterAppInstanceUserEndpointCommand");
var de_DescribeAppInstanceCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode !== 200 && output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const contents = (0, import_smithy_client.map)({
    $metadata: deserializeMetadata(output)
  });
  const data = (0, import_smithy_client.expectNonNull)((0, import_smithy_client.expectObject)(await (0, import_core2.parseJsonBody)(output.body, context)), "body");
  const doc = (0, import_smithy_client.take)(data, {
    AppInstance: (_) => de_AppInstance(_, context)
  });
  Object.assign(contents, doc);
  return contents;
}, "de_DescribeAppInstanceCommand");
var de_DescribeAppInstanceAdminCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode !== 200 && output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const contents = (0, import_smithy_client.map)({
    $metadata: deserializeMetadata(output)
  });
  const data = (0, import_smithy_client.expectNonNull)((0, import_smithy_client.expectObject)(await (0, import_core2.parseJsonBody)(output.body, context)), "body");
  const doc = (0, import_smithy_client.take)(data, {
    AppInstanceAdmin: (_) => de_AppInstanceAdmin(_, context)
  });
  Object.assign(contents, doc);
  return contents;
}, "de_DescribeAppInstanceAdminCommand");
var de_DescribeAppInstanceBotCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode !== 200 && output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const contents = (0, import_smithy_client.map)({
    $metadata: deserializeMetadata(output)
  });
  const data = (0, import_smithy_client.expectNonNull)((0, import_smithy_client.expectObject)(await (0, import_core2.parseJsonBody)(output.body, context)), "body");
  const doc = (0, import_smithy_client.take)(data, {
    AppInstanceBot: (_) => de_AppInstanceBot(_, context)
  });
  Object.assign(contents, doc);
  return contents;
}, "de_DescribeAppInstanceBotCommand");
var de_DescribeAppInstanceUserCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode !== 200 && output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const contents = (0, import_smithy_client.map)({
    $metadata: deserializeMetadata(output)
  });
  const data = (0, import_smithy_client.expectNonNull)((0, import_smithy_client.expectObject)(await (0, import_core2.parseJsonBody)(output.body, context)), "body");
  const doc = (0, import_smithy_client.take)(data, {
    AppInstanceUser: (_) => de_AppInstanceUser(_, context)
  });
  Object.assign(contents, doc);
  return contents;
}, "de_DescribeAppInstanceUserCommand");
var de_DescribeAppInstanceUserEndpointCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode !== 200 && output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const contents = (0, import_smithy_client.map)({
    $metadata: deserializeMetadata(output)
  });
  const data = (0, import_smithy_client.expectNonNull)((0, import_smithy_client.expectObject)(await (0, import_core2.parseJsonBody)(output.body, context)), "body");
  const doc = (0, import_smithy_client.take)(data, {
    AppInstanceUserEndpoint: (_) => de_AppInstanceUserEndpoint(_, context)
  });
  Object.assign(contents, doc);
  return contents;
}, "de_DescribeAppInstanceUserEndpointCommand");
var de_GetAppInstanceRetentionSettingsCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode !== 200 && output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const contents = (0, import_smithy_client.map)({
    $metadata: deserializeMetadata(output)
  });
  const data = (0, import_smithy_client.expectNonNull)((0, import_smithy_client.expectObject)(await (0, import_core2.parseJsonBody)(output.body, context)), "body");
  const doc = (0, import_smithy_client.take)(data, {
    AppInstanceRetentionSettings: import_smithy_client._json,
    InitiateDeletionTimestamp: (_) => (0, import_smithy_client.expectNonNull)((0, import_smithy_client.parseEpochTimestamp)((0, import_smithy_client.expectNumber)(_)))
  });
  Object.assign(contents, doc);
  return contents;
}, "de_GetAppInstanceRetentionSettingsCommand");
var de_ListAppInstanceAdminsCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode !== 200 && output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const contents = (0, import_smithy_client.map)({
    $metadata: deserializeMetadata(output)
  });
  const data = (0, import_smithy_client.expectNonNull)((0, import_smithy_client.expectObject)(await (0, import_core2.parseJsonBody)(output.body, context)), "body");
  const doc = (0, import_smithy_client.take)(data, {
    AppInstanceAdmins: import_smithy_client._json,
    AppInstanceArn: import_smithy_client.expectString,
    NextToken: import_smithy_client.expectString
  });
  Object.assign(contents, doc);
  return contents;
}, "de_ListAppInstanceAdminsCommand");
var de_ListAppInstanceBotsCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode !== 200 && output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const contents = (0, import_smithy_client.map)({
    $metadata: deserializeMetadata(output)
  });
  const data = (0, import_smithy_client.expectNonNull)((0, import_smithy_client.expectObject)(await (0, import_core2.parseJsonBody)(output.body, context)), "body");
  const doc = (0, import_smithy_client.take)(data, {
    AppInstanceArn: import_smithy_client.expectString,
    AppInstanceBots: import_smithy_client._json,
    NextToken: import_smithy_client.expectString
  });
  Object.assign(contents, doc);
  return contents;
}, "de_ListAppInstanceBotsCommand");
var de_ListAppInstancesCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode !== 200 && output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const contents = (0, import_smithy_client.map)({
    $metadata: deserializeMetadata(output)
  });
  const data = (0, import_smithy_client.expectNonNull)((0, import_smithy_client.expectObject)(await (0, import_core2.parseJsonBody)(output.body, context)), "body");
  const doc = (0, import_smithy_client.take)(data, {
    AppInstances: import_smithy_client._json,
    NextToken: import_smithy_client.expectString
  });
  Object.assign(contents, doc);
  return contents;
}, "de_ListAppInstancesCommand");
var de_ListAppInstanceUserEndpointsCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode !== 200 && output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const contents = (0, import_smithy_client.map)({
    $metadata: deserializeMetadata(output)
  });
  const data = (0, import_smithy_client.expectNonNull)((0, import_smithy_client.expectObject)(await (0, import_core2.parseJsonBody)(output.body, context)), "body");
  const doc = (0, import_smithy_client.take)(data, {
    AppInstanceUserEndpoints: import_smithy_client._json,
    NextToken: import_smithy_client.expectString
  });
  Object.assign(contents, doc);
  return contents;
}, "de_ListAppInstanceUserEndpointsCommand");
var de_ListAppInstanceUsersCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode !== 200 && output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const contents = (0, import_smithy_client.map)({
    $metadata: deserializeMetadata(output)
  });
  const data = (0, import_smithy_client.expectNonNull)((0, import_smithy_client.expectObject)(await (0, import_core2.parseJsonBody)(output.body, context)), "body");
  const doc = (0, import_smithy_client.take)(data, {
    AppInstanceArn: import_smithy_client.expectString,
    AppInstanceUsers: import_smithy_client._json,
    NextToken: import_smithy_client.expectString
  });
  Object.assign(contents, doc);
  return contents;
}, "de_ListAppInstanceUsersCommand");
var de_ListTagsForResourceCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode !== 200 && output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const contents = (0, import_smithy_client.map)({
    $metadata: deserializeMetadata(output)
  });
  const data = (0, import_smithy_client.expectNonNull)((0, import_smithy_client.expectObject)(await (0, import_core2.parseJsonBody)(output.body, context)), "body");
  const doc = (0, import_smithy_client.take)(data, {
    Tags: import_smithy_client._json
  });
  Object.assign(contents, doc);
  return contents;
}, "de_ListTagsForResourceCommand");
var de_PutAppInstanceRetentionSettingsCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode !== 200 && output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const contents = (0, import_smithy_client.map)({
    $metadata: deserializeMetadata(output)
  });
  const data = (0, import_smithy_client.expectNonNull)((0, import_smithy_client.expectObject)(await (0, import_core2.parseJsonBody)(output.body, context)), "body");
  const doc = (0, import_smithy_client.take)(data, {
    AppInstanceRetentionSettings: import_smithy_client._json,
    InitiateDeletionTimestamp: (_) => (0, import_smithy_client.expectNonNull)((0, import_smithy_client.parseEpochTimestamp)((0, import_smithy_client.expectNumber)(_)))
  });
  Object.assign(contents, doc);
  return contents;
}, "de_PutAppInstanceRetentionSettingsCommand");
var de_PutAppInstanceUserExpirationSettingsCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode !== 200 && output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const contents = (0, import_smithy_client.map)({
    $metadata: deserializeMetadata(output)
  });
  const data = (0, import_smithy_client.expectNonNull)((0, import_smithy_client.expectObject)(await (0, import_core2.parseJsonBody)(output.body, context)), "body");
  const doc = (0, import_smithy_client.take)(data, {
    AppInstanceUserArn: import_smithy_client.expectString,
    ExpirationSettings: import_smithy_client._json
  });
  Object.assign(contents, doc);
  return contents;
}, "de_PutAppInstanceUserExpirationSettingsCommand");
var de_RegisterAppInstanceUserEndpointCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode !== 201 && output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const contents = (0, import_smithy_client.map)({
    $metadata: deserializeMetadata(output)
  });
  const data = (0, import_smithy_client.expectNonNull)((0, import_smithy_client.expectObject)(await (0, import_core2.parseJsonBody)(output.body, context)), "body");
  const doc = (0, import_smithy_client.take)(data, {
    AppInstanceUserArn: import_smithy_client.expectString,
    EndpointId: import_smithy_client.expectString
  });
  Object.assign(contents, doc);
  return contents;
}, "de_RegisterAppInstanceUserEndpointCommand");
var de_TagResourceCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode !== 204 && output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const contents = (0, import_smithy_client.map)({
    $metadata: deserializeMetadata(output)
  });
  await (0, import_smithy_client.collectBody)(output.body, context);
  return contents;
}, "de_TagResourceCommand");
var de_UntagResourceCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode !== 204 && output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const contents = (0, import_smithy_client.map)({
    $metadata: deserializeMetadata(output)
  });
  await (0, import_smithy_client.collectBody)(output.body, context);
  return contents;
}, "de_UntagResourceCommand");
var de_UpdateAppInstanceCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode !== 200 && output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const contents = (0, import_smithy_client.map)({
    $metadata: deserializeMetadata(output)
  });
  const data = (0, import_smithy_client.expectNonNull)((0, import_smithy_client.expectObject)(await (0, import_core2.parseJsonBody)(output.body, context)), "body");
  const doc = (0, import_smithy_client.take)(data, {
    AppInstanceArn: import_smithy_client.expectString
  });
  Object.assign(contents, doc);
  return contents;
}, "de_UpdateAppInstanceCommand");
var de_UpdateAppInstanceBotCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode !== 200 && output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const contents = (0, import_smithy_client.map)({
    $metadata: deserializeMetadata(output)
  });
  const data = (0, import_smithy_client.expectNonNull)((0, import_smithy_client.expectObject)(await (0, import_core2.parseJsonBody)(output.body, context)), "body");
  const doc = (0, import_smithy_client.take)(data, {
    AppInstanceBotArn: import_smithy_client.expectString
  });
  Object.assign(contents, doc);
  return contents;
}, "de_UpdateAppInstanceBotCommand");
var de_UpdateAppInstanceUserCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode !== 200 && output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const contents = (0, import_smithy_client.map)({
    $metadata: deserializeMetadata(output)
  });
  const data = (0, import_smithy_client.expectNonNull)((0, import_smithy_client.expectObject)(await (0, import_core2.parseJsonBody)(output.body, context)), "body");
  const doc = (0, import_smithy_client.take)(data, {
    AppInstanceUserArn: import_smithy_client.expectString
  });
  Object.assign(contents, doc);
  return contents;
}, "de_UpdateAppInstanceUserCommand");
var de_UpdateAppInstanceUserEndpointCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode !== 200 && output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const contents = (0, import_smithy_client.map)({
    $metadata: deserializeMetadata(output)
  });
  const data = (0, import_smithy_client.expectNonNull)((0, import_smithy_client.expectObject)(await (0, import_core2.parseJsonBody)(output.body, context)), "body");
  const doc = (0, import_smithy_client.take)(data, {
    AppInstanceUserArn: import_smithy_client.expectString,
    EndpointId: import_smithy_client.expectString
  });
  Object.assign(contents, doc);
  return contents;
}, "de_UpdateAppInstanceUserEndpointCommand");
var de_CommandError = /* @__PURE__ */ __name(async (output, context) => {
  const parsedOutput = {
    ...output,
    body: await (0, import_core2.parseJsonErrorBody)(output.body, context)
  };
  const errorCode = (0, import_core2.loadRestJsonErrorCode)(output, parsedOutput.body);
  switch (errorCode) {
    case "BadRequestException":
    case "com.amazonaws.chimesdkidentity#BadRequestException":
      throw await de_BadRequestExceptionRes(parsedOutput, context);
    case "ConflictException":
    case "com.amazonaws.chimesdkidentity#ConflictException":
      throw await de_ConflictExceptionRes(parsedOutput, context);
    case "ForbiddenException":
    case "com.amazonaws.chimesdkidentity#ForbiddenException":
      throw await de_ForbiddenExceptionRes(parsedOutput, context);
    case "ResourceLimitExceededException":
    case "com.amazonaws.chimesdkidentity#ResourceLimitExceededException":
      throw await de_ResourceLimitExceededExceptionRes(parsedOutput, context);
    case "ServiceFailureException":
    case "com.amazonaws.chimesdkidentity#ServiceFailureException":
      throw await de_ServiceFailureExceptionRes(parsedOutput, context);
    case "ServiceUnavailableException":
    case "com.amazonaws.chimesdkidentity#ServiceUnavailableException":
      throw await de_ServiceUnavailableExceptionRes(parsedOutput, context);
    case "ThrottledClientException":
    case "com.amazonaws.chimesdkidentity#ThrottledClientException":
      throw await de_ThrottledClientExceptionRes(parsedOutput, context);
    case "UnauthorizedClientException":
    case "com.amazonaws.chimesdkidentity#UnauthorizedClientException":
      throw await de_UnauthorizedClientExceptionRes(parsedOutput, context);
    case "NotFoundException":
    case "com.amazonaws.chimesdkidentity#NotFoundException":
      throw await de_NotFoundExceptionRes(parsedOutput, context);
    default:
      const parsedBody = parsedOutput.body;
      return throwDefaultError({
        output,
        parsedBody,
        errorCode
      });
  }
}, "de_CommandError");
var throwDefaultError = (0, import_smithy_client.withBaseException)(ChimeSDKIdentityServiceException);
var de_BadRequestExceptionRes = /* @__PURE__ */ __name(async (parsedOutput, context) => {
  const contents = (0, import_smithy_client.map)({});
  const data = parsedOutput.body;
  const doc = (0, import_smithy_client.take)(data, {
    Code: import_smithy_client.expectString,
    Message: import_smithy_client.expectString
  });
  Object.assign(contents, doc);
  const exception = new BadRequestException({
    $metadata: deserializeMetadata(parsedOutput),
    ...contents
  });
  return (0, import_smithy_client.decorateServiceException)(exception, parsedOutput.body);
}, "de_BadRequestExceptionRes");
var de_ConflictExceptionRes = /* @__PURE__ */ __name(async (parsedOutput, context) => {
  const contents = (0, import_smithy_client.map)({});
  const data = parsedOutput.body;
  const doc = (0, import_smithy_client.take)(data, {
    Code: import_smithy_client.expectString,
    Message: import_smithy_client.expectString
  });
  Object.assign(contents, doc);
  const exception = new ConflictException({
    $metadata: deserializeMetadata(parsedOutput),
    ...contents
  });
  return (0, import_smithy_client.decorateServiceException)(exception, parsedOutput.body);
}, "de_ConflictExceptionRes");
var de_ForbiddenExceptionRes = /* @__PURE__ */ __name(async (parsedOutput, context) => {
  const contents = (0, import_smithy_client.map)({});
  const data = parsedOutput.body;
  const doc = (0, import_smithy_client.take)(data, {
    Code: import_smithy_client.expectString,
    Message: import_smithy_client.expectString
  });
  Object.assign(contents, doc);
  const exception = new ForbiddenException({
    $metadata: deserializeMetadata(parsedOutput),
    ...contents
  });
  return (0, import_smithy_client.decorateServiceException)(exception, parsedOutput.body);
}, "de_ForbiddenExceptionRes");
var de_NotFoundExceptionRes = /* @__PURE__ */ __name(async (parsedOutput, context) => {
  const contents = (0, import_smithy_client.map)({});
  const data = parsedOutput.body;
  const doc = (0, import_smithy_client.take)(data, {
    Code: import_smithy_client.expectString,
    Message: import_smithy_client.expectString
  });
  Object.assign(contents, doc);
  const exception = new NotFoundException({
    $metadata: deserializeMetadata(parsedOutput),
    ...contents
  });
  return (0, import_smithy_client.decorateServiceException)(exception, parsedOutput.body);
}, "de_NotFoundExceptionRes");
var de_ResourceLimitExceededExceptionRes = /* @__PURE__ */ __name(async (parsedOutput, context) => {
  const contents = (0, import_smithy_client.map)({});
  const data = parsedOutput.body;
  const doc = (0, import_smithy_client.take)(data, {
    Code: import_smithy_client.expectString,
    Message: import_smithy_client.expectString
  });
  Object.assign(contents, doc);
  const exception = new ResourceLimitExceededException({
    $metadata: deserializeMetadata(parsedOutput),
    ...contents
  });
  return (0, import_smithy_client.decorateServiceException)(exception, parsedOutput.body);
}, "de_ResourceLimitExceededExceptionRes");
var de_ServiceFailureExceptionRes = /* @__PURE__ */ __name(async (parsedOutput, context) => {
  const contents = (0, import_smithy_client.map)({});
  const data = parsedOutput.body;
  const doc = (0, import_smithy_client.take)(data, {
    Code: import_smithy_client.expectString,
    Message: import_smithy_client.expectString
  });
  Object.assign(contents, doc);
  const exception = new ServiceFailureException({
    $metadata: deserializeMetadata(parsedOutput),
    ...contents
  });
  return (0, import_smithy_client.decorateServiceException)(exception, parsedOutput.body);
}, "de_ServiceFailureExceptionRes");
var de_ServiceUnavailableExceptionRes = /* @__PURE__ */ __name(async (parsedOutput, context) => {
  const contents = (0, import_smithy_client.map)({});
  const data = parsedOutput.body;
  const doc = (0, import_smithy_client.take)(data, {
    Code: import_smithy_client.expectString,
    Message: import_smithy_client.expectString
  });
  Object.assign(contents, doc);
  const exception = new ServiceUnavailableException({
    $metadata: deserializeMetadata(parsedOutput),
    ...contents
  });
  return (0, import_smithy_client.decorateServiceException)(exception, parsedOutput.body);
}, "de_ServiceUnavailableExceptionRes");
var de_ThrottledClientExceptionRes = /* @__PURE__ */ __name(async (parsedOutput, context) => {
  const contents = (0, import_smithy_client.map)({});
  const data = parsedOutput.body;
  const doc = (0, import_smithy_client.take)(data, {
    Code: import_smithy_client.expectString,
    Message: import_smithy_client.expectString
  });
  Object.assign(contents, doc);
  const exception = new ThrottledClientException({
    $metadata: deserializeMetadata(parsedOutput),
    ...contents
  });
  return (0, import_smithy_client.decorateServiceException)(exception, parsedOutput.body);
}, "de_ThrottledClientExceptionRes");
var de_UnauthorizedClientExceptionRes = /* @__PURE__ */ __name(async (parsedOutput, context) => {
  const contents = (0, import_smithy_client.map)({});
  const data = parsedOutput.body;
  const doc = (0, import_smithy_client.take)(data, {
    Code: import_smithy_client.expectString,
    Message: import_smithy_client.expectString
  });
  Object.assign(contents, doc);
  const exception = new UnauthorizedClientException({
    $metadata: deserializeMetadata(parsedOutput),
    ...contents
  });
  return (0, import_smithy_client.decorateServiceException)(exception, parsedOutput.body);
}, "de_UnauthorizedClientExceptionRes");
var de_AppInstance = /* @__PURE__ */ __name((output, context) => {
  return (0, import_smithy_client.take)(output, {
    AppInstanceArn: import_smithy_client.expectString,
    CreatedTimestamp: (_) => (0, import_smithy_client.expectNonNull)((0, import_smithy_client.parseEpochTimestamp)((0, import_smithy_client.expectNumber)(_))),
    LastUpdatedTimestamp: (_) => (0, import_smithy_client.expectNonNull)((0, import_smithy_client.parseEpochTimestamp)((0, import_smithy_client.expectNumber)(_))),
    Metadata: import_smithy_client.expectString,
    Name: import_smithy_client.expectString
  });
}, "de_AppInstance");
var de_AppInstanceAdmin = /* @__PURE__ */ __name((output, context) => {
  return (0, import_smithy_client.take)(output, {
    Admin: import_smithy_client._json,
    AppInstanceArn: import_smithy_client.expectString,
    CreatedTimestamp: (_) => (0, import_smithy_client.expectNonNull)((0, import_smithy_client.parseEpochTimestamp)((0, import_smithy_client.expectNumber)(_)))
  });
}, "de_AppInstanceAdmin");
var de_AppInstanceBot = /* @__PURE__ */ __name((output, context) => {
  return (0, import_smithy_client.take)(output, {
    AppInstanceBotArn: import_smithy_client.expectString,
    Configuration: import_smithy_client._json,
    CreatedTimestamp: (_) => (0, import_smithy_client.expectNonNull)((0, import_smithy_client.parseEpochTimestamp)((0, import_smithy_client.expectNumber)(_))),
    LastUpdatedTimestamp: (_) => (0, import_smithy_client.expectNonNull)((0, import_smithy_client.parseEpochTimestamp)((0, import_smithy_client.expectNumber)(_))),
    Metadata: import_smithy_client.expectString,
    Name: import_smithy_client.expectString
  });
}, "de_AppInstanceBot");
var de_AppInstanceUser = /* @__PURE__ */ __name((output, context) => {
  return (0, import_smithy_client.take)(output, {
    AppInstanceUserArn: import_smithy_client.expectString,
    CreatedTimestamp: (_) => (0, import_smithy_client.expectNonNull)((0, import_smithy_client.parseEpochTimestamp)((0, import_smithy_client.expectNumber)(_))),
    ExpirationSettings: import_smithy_client._json,
    LastUpdatedTimestamp: (_) => (0, import_smithy_client.expectNonNull)((0, import_smithy_client.parseEpochTimestamp)((0, import_smithy_client.expectNumber)(_))),
    Metadata: import_smithy_client.expectString,
    Name: import_smithy_client.expectString
  });
}, "de_AppInstanceUser");
var de_AppInstanceUserEndpoint = /* @__PURE__ */ __name((output, context) => {
  return (0, import_smithy_client.take)(output, {
    AllowMessages: import_smithy_client.expectString,
    AppInstanceUserArn: import_smithy_client.expectString,
    CreatedTimestamp: (_) => (0, import_smithy_client.expectNonNull)((0, import_smithy_client.parseEpochTimestamp)((0, import_smithy_client.expectNumber)(_))),
    EndpointAttributes: import_smithy_client._json,
    EndpointId: import_smithy_client.expectString,
    EndpointState: import_smithy_client._json,
    LastUpdatedTimestamp: (_) => (0, import_smithy_client.expectNonNull)((0, import_smithy_client.parseEpochTimestamp)((0, import_smithy_client.expectNumber)(_))),
    Name: import_smithy_client.expectString,
    ResourceArn: import_smithy_client.expectString,
    Type: import_smithy_client.expectString
  });
}, "de_AppInstanceUserEndpoint");
var deserializeMetadata = /* @__PURE__ */ __name((output) => ({
  httpStatusCode: output.statusCode,
  requestId: output.headers["x-amzn-requestid"] ?? output.headers["x-amzn-request-id"] ?? output.headers["x-amz-request-id"],
  extendedRequestId: output.headers["x-amz-id-2"],
  cfId: output.headers["x-amz-cf-id"]
}), "deserializeMetadata");
var _AIA = "AppInstanceArn";
var _MR = "MaxResults";
var _NT = "NextToken";
var _RARN = "ResourceARN";
var _a = "arn";
var _aia = "app-instance-arn";
var _mr = "max-results";
var _nt = "next-token";
var _o = "operation";

// src/commands/CreateAppInstanceAdminCommand.ts
var _CreateAppInstanceAdminCommand = class _CreateAppInstanceAdminCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("ChimeIdentityService", "CreateAppInstanceAdmin", {}).n("ChimeSDKIdentityClient", "CreateAppInstanceAdminCommand").f(void 0, CreateAppInstanceAdminResponseFilterSensitiveLog).ser(se_CreateAppInstanceAdminCommand).de(de_CreateAppInstanceAdminCommand).build() {
};
__name(_CreateAppInstanceAdminCommand, "CreateAppInstanceAdminCommand");
var CreateAppInstanceAdminCommand = _CreateAppInstanceAdminCommand;

// src/commands/CreateAppInstanceBotCommand.ts




var _CreateAppInstanceBotCommand = class _CreateAppInstanceBotCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("ChimeIdentityService", "CreateAppInstanceBot", {}).n("ChimeSDKIdentityClient", "CreateAppInstanceBotCommand").f(CreateAppInstanceBotRequestFilterSensitiveLog, void 0).ser(se_CreateAppInstanceBotCommand).de(de_CreateAppInstanceBotCommand).build() {
};
__name(_CreateAppInstanceBotCommand, "CreateAppInstanceBotCommand");
var CreateAppInstanceBotCommand = _CreateAppInstanceBotCommand;

// src/commands/CreateAppInstanceCommand.ts




var _CreateAppInstanceCommand = class _CreateAppInstanceCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("ChimeIdentityService", "CreateAppInstance", {}).n("ChimeSDKIdentityClient", "CreateAppInstanceCommand").f(CreateAppInstanceRequestFilterSensitiveLog, void 0).ser(se_CreateAppInstanceCommand).de(de_CreateAppInstanceCommand).build() {
};
__name(_CreateAppInstanceCommand, "CreateAppInstanceCommand");
var CreateAppInstanceCommand = _CreateAppInstanceCommand;

// src/commands/CreateAppInstanceUserCommand.ts




var _CreateAppInstanceUserCommand = class _CreateAppInstanceUserCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("ChimeIdentityService", "CreateAppInstanceUser", {}).n("ChimeSDKIdentityClient", "CreateAppInstanceUserCommand").f(CreateAppInstanceUserRequestFilterSensitiveLog, void 0).ser(se_CreateAppInstanceUserCommand).de(de_CreateAppInstanceUserCommand).build() {
};
__name(_CreateAppInstanceUserCommand, "CreateAppInstanceUserCommand");
var CreateAppInstanceUserCommand = _CreateAppInstanceUserCommand;

// src/commands/DeleteAppInstanceAdminCommand.ts




var _DeleteAppInstanceAdminCommand = class _DeleteAppInstanceAdminCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("ChimeIdentityService", "DeleteAppInstanceAdmin", {}).n("ChimeSDKIdentityClient", "DeleteAppInstanceAdminCommand").f(void 0, void 0).ser(se_DeleteAppInstanceAdminCommand).de(de_DeleteAppInstanceAdminCommand).build() {
};
__name(_DeleteAppInstanceAdminCommand, "DeleteAppInstanceAdminCommand");
var DeleteAppInstanceAdminCommand = _DeleteAppInstanceAdminCommand;

// src/commands/DeleteAppInstanceBotCommand.ts




var _DeleteAppInstanceBotCommand = class _DeleteAppInstanceBotCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("ChimeIdentityService", "DeleteAppInstanceBot", {}).n("ChimeSDKIdentityClient", "DeleteAppInstanceBotCommand").f(void 0, void 0).ser(se_DeleteAppInstanceBotCommand).de(de_DeleteAppInstanceBotCommand).build() {
};
__name(_DeleteAppInstanceBotCommand, "DeleteAppInstanceBotCommand");
var DeleteAppInstanceBotCommand = _DeleteAppInstanceBotCommand;

// src/commands/DeleteAppInstanceCommand.ts




var _DeleteAppInstanceCommand = class _DeleteAppInstanceCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("ChimeIdentityService", "DeleteAppInstance", {}).n("ChimeSDKIdentityClient", "DeleteAppInstanceCommand").f(void 0, void 0).ser(se_DeleteAppInstanceCommand).de(de_DeleteAppInstanceCommand).build() {
};
__name(_DeleteAppInstanceCommand, "DeleteAppInstanceCommand");
var DeleteAppInstanceCommand = _DeleteAppInstanceCommand;

// src/commands/DeleteAppInstanceUserCommand.ts




var _DeleteAppInstanceUserCommand = class _DeleteAppInstanceUserCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("ChimeIdentityService", "DeleteAppInstanceUser", {}).n("ChimeSDKIdentityClient", "DeleteAppInstanceUserCommand").f(void 0, void 0).ser(se_DeleteAppInstanceUserCommand).de(de_DeleteAppInstanceUserCommand).build() {
};
__name(_DeleteAppInstanceUserCommand, "DeleteAppInstanceUserCommand");
var DeleteAppInstanceUserCommand = _DeleteAppInstanceUserCommand;

// src/commands/DeregisterAppInstanceUserEndpointCommand.ts




var _DeregisterAppInstanceUserEndpointCommand = class _DeregisterAppInstanceUserEndpointCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("ChimeIdentityService", "DeregisterAppInstanceUserEndpoint", {}).n("ChimeSDKIdentityClient", "DeregisterAppInstanceUserEndpointCommand").f(void 0, void 0).ser(se_DeregisterAppInstanceUserEndpointCommand).de(de_DeregisterAppInstanceUserEndpointCommand).build() {
};
__name(_DeregisterAppInstanceUserEndpointCommand, "DeregisterAppInstanceUserEndpointCommand");
var DeregisterAppInstanceUserEndpointCommand = _DeregisterAppInstanceUserEndpointCommand;

// src/commands/DescribeAppInstanceAdminCommand.ts




var _DescribeAppInstanceAdminCommand = class _DescribeAppInstanceAdminCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("ChimeIdentityService", "DescribeAppInstanceAdmin", {}).n("ChimeSDKIdentityClient", "DescribeAppInstanceAdminCommand").f(void 0, DescribeAppInstanceAdminResponseFilterSensitiveLog).ser(se_DescribeAppInstanceAdminCommand).de(de_DescribeAppInstanceAdminCommand).build() {
};
__name(_DescribeAppInstanceAdminCommand, "DescribeAppInstanceAdminCommand");
var DescribeAppInstanceAdminCommand = _DescribeAppInstanceAdminCommand;

// src/commands/DescribeAppInstanceBotCommand.ts




var _DescribeAppInstanceBotCommand = class _DescribeAppInstanceBotCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("ChimeIdentityService", "DescribeAppInstanceBot", {}).n("ChimeSDKIdentityClient", "DescribeAppInstanceBotCommand").f(void 0, DescribeAppInstanceBotResponseFilterSensitiveLog).ser(se_DescribeAppInstanceBotCommand).de(de_DescribeAppInstanceBotCommand).build() {
};
__name(_DescribeAppInstanceBotCommand, "DescribeAppInstanceBotCommand");
var DescribeAppInstanceBotCommand = _DescribeAppInstanceBotCommand;

// src/commands/DescribeAppInstanceCommand.ts




var _DescribeAppInstanceCommand = class _DescribeAppInstanceCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("ChimeIdentityService", "DescribeAppInstance", {}).n("ChimeSDKIdentityClient", "DescribeAppInstanceCommand").f(void 0, DescribeAppInstanceResponseFilterSensitiveLog).ser(se_DescribeAppInstanceCommand).de(de_DescribeAppInstanceCommand).build() {
};
__name(_DescribeAppInstanceCommand, "DescribeAppInstanceCommand");
var DescribeAppInstanceCommand = _DescribeAppInstanceCommand;

// src/commands/DescribeAppInstanceUserCommand.ts




var _DescribeAppInstanceUserCommand = class _DescribeAppInstanceUserCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("ChimeIdentityService", "DescribeAppInstanceUser", {}).n("ChimeSDKIdentityClient", "DescribeAppInstanceUserCommand").f(void 0, DescribeAppInstanceUserResponseFilterSensitiveLog).ser(se_DescribeAppInstanceUserCommand).de(de_DescribeAppInstanceUserCommand).build() {
};
__name(_DescribeAppInstanceUserCommand, "DescribeAppInstanceUserCommand");
var DescribeAppInstanceUserCommand = _DescribeAppInstanceUserCommand;

// src/commands/DescribeAppInstanceUserEndpointCommand.ts




var _DescribeAppInstanceUserEndpointCommand = class _DescribeAppInstanceUserEndpointCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("ChimeIdentityService", "DescribeAppInstanceUserEndpoint", {}).n("ChimeSDKIdentityClient", "DescribeAppInstanceUserEndpointCommand").f(void 0, DescribeAppInstanceUserEndpointResponseFilterSensitiveLog).ser(se_DescribeAppInstanceUserEndpointCommand).de(de_DescribeAppInstanceUserEndpointCommand).build() {
};
__name(_DescribeAppInstanceUserEndpointCommand, "DescribeAppInstanceUserEndpointCommand");
var DescribeAppInstanceUserEndpointCommand = _DescribeAppInstanceUserEndpointCommand;

// src/commands/GetAppInstanceRetentionSettingsCommand.ts




var _GetAppInstanceRetentionSettingsCommand = class _GetAppInstanceRetentionSettingsCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("ChimeIdentityService", "GetAppInstanceRetentionSettings", {}).n("ChimeSDKIdentityClient", "GetAppInstanceRetentionSettingsCommand").f(void 0, void 0).ser(se_GetAppInstanceRetentionSettingsCommand).de(de_GetAppInstanceRetentionSettingsCommand).build() {
};
__name(_GetAppInstanceRetentionSettingsCommand, "GetAppInstanceRetentionSettingsCommand");
var GetAppInstanceRetentionSettingsCommand = _GetAppInstanceRetentionSettingsCommand;

// src/commands/ListAppInstanceAdminsCommand.ts




var _ListAppInstanceAdminsCommand = class _ListAppInstanceAdminsCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("ChimeIdentityService", "ListAppInstanceAdmins", {}).n("ChimeSDKIdentityClient", "ListAppInstanceAdminsCommand").f(ListAppInstanceAdminsRequestFilterSensitiveLog, ListAppInstanceAdminsResponseFilterSensitiveLog).ser(se_ListAppInstanceAdminsCommand).de(de_ListAppInstanceAdminsCommand).build() {
};
__name(_ListAppInstanceAdminsCommand, "ListAppInstanceAdminsCommand");
var ListAppInstanceAdminsCommand = _ListAppInstanceAdminsCommand;

// src/commands/ListAppInstanceBotsCommand.ts




var _ListAppInstanceBotsCommand = class _ListAppInstanceBotsCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("ChimeIdentityService", "ListAppInstanceBots", {}).n("ChimeSDKIdentityClient", "ListAppInstanceBotsCommand").f(ListAppInstanceBotsRequestFilterSensitiveLog, ListAppInstanceBotsResponseFilterSensitiveLog).ser(se_ListAppInstanceBotsCommand).de(de_ListAppInstanceBotsCommand).build() {
};
__name(_ListAppInstanceBotsCommand, "ListAppInstanceBotsCommand");
var ListAppInstanceBotsCommand = _ListAppInstanceBotsCommand;

// src/commands/ListAppInstancesCommand.ts




var _ListAppInstancesCommand = class _ListAppInstancesCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("ChimeIdentityService", "ListAppInstances", {}).n("ChimeSDKIdentityClient", "ListAppInstancesCommand").f(ListAppInstancesRequestFilterSensitiveLog, ListAppInstancesResponseFilterSensitiveLog).ser(se_ListAppInstancesCommand).de(de_ListAppInstancesCommand).build() {
};
__name(_ListAppInstancesCommand, "ListAppInstancesCommand");
var ListAppInstancesCommand = _ListAppInstancesCommand;

// src/commands/ListAppInstanceUserEndpointsCommand.ts




var _ListAppInstanceUserEndpointsCommand = class _ListAppInstanceUserEndpointsCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("ChimeIdentityService", "ListAppInstanceUserEndpoints", {}).n("ChimeSDKIdentityClient", "ListAppInstanceUserEndpointsCommand").f(ListAppInstanceUserEndpointsRequestFilterSensitiveLog, ListAppInstanceUserEndpointsResponseFilterSensitiveLog).ser(se_ListAppInstanceUserEndpointsCommand).de(de_ListAppInstanceUserEndpointsCommand).build() {
};
__name(_ListAppInstanceUserEndpointsCommand, "ListAppInstanceUserEndpointsCommand");
var ListAppInstanceUserEndpointsCommand = _ListAppInstanceUserEndpointsCommand;

// src/commands/ListAppInstanceUsersCommand.ts




var _ListAppInstanceUsersCommand = class _ListAppInstanceUsersCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("ChimeIdentityService", "ListAppInstanceUsers", {}).n("ChimeSDKIdentityClient", "ListAppInstanceUsersCommand").f(ListAppInstanceUsersRequestFilterSensitiveLog, ListAppInstanceUsersResponseFilterSensitiveLog).ser(se_ListAppInstanceUsersCommand).de(de_ListAppInstanceUsersCommand).build() {
};
__name(_ListAppInstanceUsersCommand, "ListAppInstanceUsersCommand");
var ListAppInstanceUsersCommand = _ListAppInstanceUsersCommand;

// src/commands/ListTagsForResourceCommand.ts




var _ListTagsForResourceCommand = class _ListTagsForResourceCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("ChimeIdentityService", "ListTagsForResource", {}).n("ChimeSDKIdentityClient", "ListTagsForResourceCommand").f(void 0, ListTagsForResourceResponseFilterSensitiveLog).ser(se_ListTagsForResourceCommand).de(de_ListTagsForResourceCommand).build() {
};
__name(_ListTagsForResourceCommand, "ListTagsForResourceCommand");
var ListTagsForResourceCommand = _ListTagsForResourceCommand;

// src/commands/PutAppInstanceRetentionSettingsCommand.ts




var _PutAppInstanceRetentionSettingsCommand = class _PutAppInstanceRetentionSettingsCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("ChimeIdentityService", "PutAppInstanceRetentionSettings", {}).n("ChimeSDKIdentityClient", "PutAppInstanceRetentionSettingsCommand").f(void 0, void 0).ser(se_PutAppInstanceRetentionSettingsCommand).de(de_PutAppInstanceRetentionSettingsCommand).build() {
};
__name(_PutAppInstanceRetentionSettingsCommand, "PutAppInstanceRetentionSettingsCommand");
var PutAppInstanceRetentionSettingsCommand = _PutAppInstanceRetentionSettingsCommand;

// src/commands/PutAppInstanceUserExpirationSettingsCommand.ts




var _PutAppInstanceUserExpirationSettingsCommand = class _PutAppInstanceUserExpirationSettingsCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("ChimeIdentityService", "PutAppInstanceUserExpirationSettings", {}).n("ChimeSDKIdentityClient", "PutAppInstanceUserExpirationSettingsCommand").f(void 0, void 0).ser(se_PutAppInstanceUserExpirationSettingsCommand).de(de_PutAppInstanceUserExpirationSettingsCommand).build() {
};
__name(_PutAppInstanceUserExpirationSettingsCommand, "PutAppInstanceUserExpirationSettingsCommand");
var PutAppInstanceUserExpirationSettingsCommand = _PutAppInstanceUserExpirationSettingsCommand;

// src/commands/RegisterAppInstanceUserEndpointCommand.ts




var _RegisterAppInstanceUserEndpointCommand = class _RegisterAppInstanceUserEndpointCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("ChimeIdentityService", "RegisterAppInstanceUserEndpoint", {}).n("ChimeSDKIdentityClient", "RegisterAppInstanceUserEndpointCommand").f(RegisterAppInstanceUserEndpointRequestFilterSensitiveLog, void 0).ser(se_RegisterAppInstanceUserEndpointCommand).de(de_RegisterAppInstanceUserEndpointCommand).build() {
};
__name(_RegisterAppInstanceUserEndpointCommand, "RegisterAppInstanceUserEndpointCommand");
var RegisterAppInstanceUserEndpointCommand = _RegisterAppInstanceUserEndpointCommand;

// src/commands/TagResourceCommand.ts




var _TagResourceCommand = class _TagResourceCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("ChimeIdentityService", "TagResource", {}).n("ChimeSDKIdentityClient", "TagResourceCommand").f(TagResourceRequestFilterSensitiveLog, void 0).ser(se_TagResourceCommand).de(de_TagResourceCommand).build() {
};
__name(_TagResourceCommand, "TagResourceCommand");
var TagResourceCommand = _TagResourceCommand;

// src/commands/UntagResourceCommand.ts




var _UntagResourceCommand = class _UntagResourceCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("ChimeIdentityService", "UntagResource", {}).n("ChimeSDKIdentityClient", "UntagResourceCommand").f(UntagResourceRequestFilterSensitiveLog, void 0).ser(se_UntagResourceCommand).de(de_UntagResourceCommand).build() {
};
__name(_UntagResourceCommand, "UntagResourceCommand");
var UntagResourceCommand = _UntagResourceCommand;

// src/commands/UpdateAppInstanceBotCommand.ts




var _UpdateAppInstanceBotCommand = class _UpdateAppInstanceBotCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("ChimeIdentityService", "UpdateAppInstanceBot", {}).n("ChimeSDKIdentityClient", "UpdateAppInstanceBotCommand").f(UpdateAppInstanceBotRequestFilterSensitiveLog, void 0).ser(se_UpdateAppInstanceBotCommand).de(de_UpdateAppInstanceBotCommand).build() {
};
__name(_UpdateAppInstanceBotCommand, "UpdateAppInstanceBotCommand");
var UpdateAppInstanceBotCommand = _UpdateAppInstanceBotCommand;

// src/commands/UpdateAppInstanceCommand.ts




var _UpdateAppInstanceCommand = class _UpdateAppInstanceCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("ChimeIdentityService", "UpdateAppInstance", {}).n("ChimeSDKIdentityClient", "UpdateAppInstanceCommand").f(UpdateAppInstanceRequestFilterSensitiveLog, void 0).ser(se_UpdateAppInstanceCommand).de(de_UpdateAppInstanceCommand).build() {
};
__name(_UpdateAppInstanceCommand, "UpdateAppInstanceCommand");
var UpdateAppInstanceCommand = _UpdateAppInstanceCommand;

// src/commands/UpdateAppInstanceUserCommand.ts




var _UpdateAppInstanceUserCommand = class _UpdateAppInstanceUserCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("ChimeIdentityService", "UpdateAppInstanceUser", {}).n("ChimeSDKIdentityClient", "UpdateAppInstanceUserCommand").f(UpdateAppInstanceUserRequestFilterSensitiveLog, void 0).ser(se_UpdateAppInstanceUserCommand).de(de_UpdateAppInstanceUserCommand).build() {
};
__name(_UpdateAppInstanceUserCommand, "UpdateAppInstanceUserCommand");
var UpdateAppInstanceUserCommand = _UpdateAppInstanceUserCommand;

// src/commands/UpdateAppInstanceUserEndpointCommand.ts




var _UpdateAppInstanceUserEndpointCommand = class _UpdateAppInstanceUserEndpointCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("ChimeIdentityService", "UpdateAppInstanceUserEndpoint", {}).n("ChimeSDKIdentityClient", "UpdateAppInstanceUserEndpointCommand").f(UpdateAppInstanceUserEndpointRequestFilterSensitiveLog, void 0).ser(se_UpdateAppInstanceUserEndpointCommand).de(de_UpdateAppInstanceUserEndpointCommand).build() {
};
__name(_UpdateAppInstanceUserEndpointCommand, "UpdateAppInstanceUserEndpointCommand");
var UpdateAppInstanceUserEndpointCommand = _UpdateAppInstanceUserEndpointCommand;

// src/ChimeSDKIdentity.ts
var commands = {
  CreateAppInstanceCommand,
  CreateAppInstanceAdminCommand,
  CreateAppInstanceBotCommand,
  CreateAppInstanceUserCommand,
  DeleteAppInstanceCommand,
  DeleteAppInstanceAdminCommand,
  DeleteAppInstanceBotCommand,
  DeleteAppInstanceUserCommand,
  DeregisterAppInstanceUserEndpointCommand,
  DescribeAppInstanceCommand,
  DescribeAppInstanceAdminCommand,
  DescribeAppInstanceBotCommand,
  DescribeAppInstanceUserCommand,
  DescribeAppInstanceUserEndpointCommand,
  GetAppInstanceRetentionSettingsCommand,
  ListAppInstanceAdminsCommand,
  ListAppInstanceBotsCommand,
  ListAppInstancesCommand,
  ListAppInstanceUserEndpointsCommand,
  ListAppInstanceUsersCommand,
  ListTagsForResourceCommand,
  PutAppInstanceRetentionSettingsCommand,
  PutAppInstanceUserExpirationSettingsCommand,
  RegisterAppInstanceUserEndpointCommand,
  TagResourceCommand,
  UntagResourceCommand,
  UpdateAppInstanceCommand,
  UpdateAppInstanceBotCommand,
  UpdateAppInstanceUserCommand,
  UpdateAppInstanceUserEndpointCommand
};
var _ChimeSDKIdentity = class _ChimeSDKIdentity extends ChimeSDKIdentityClient {
};
__name(_ChimeSDKIdentity, "ChimeSDKIdentity");
var ChimeSDKIdentity = _ChimeSDKIdentity;
(0, import_smithy_client.createAggregatedClient)(commands, ChimeSDKIdentity);

// src/pagination/ListAppInstanceAdminsPaginator.ts

var paginateListAppInstanceAdmins = (0, import_core.createPaginator)(ChimeSDKIdentityClient, ListAppInstanceAdminsCommand, "NextToken", "NextToken", "MaxResults");

// src/pagination/ListAppInstanceBotsPaginator.ts

var paginateListAppInstanceBots = (0, import_core.createPaginator)(ChimeSDKIdentityClient, ListAppInstanceBotsCommand, "NextToken", "NextToken", "MaxResults");

// src/pagination/ListAppInstanceUserEndpointsPaginator.ts

var paginateListAppInstanceUserEndpoints = (0, import_core.createPaginator)(ChimeSDKIdentityClient, ListAppInstanceUserEndpointsCommand, "NextToken", "NextToken", "MaxResults");

// src/pagination/ListAppInstanceUsersPaginator.ts

var paginateListAppInstanceUsers = (0, import_core.createPaginator)(ChimeSDKIdentityClient, ListAppInstanceUsersCommand, "NextToken", "NextToken", "MaxResults");

// src/pagination/ListAppInstancesPaginator.ts

var paginateListAppInstances = (0, import_core.createPaginator)(ChimeSDKIdentityClient, ListAppInstancesCommand, "NextToken", "NextToken", "MaxResults");
// Annotate the CommonJS export names for ESM import in node:

0 && (module.exports = {
  ChimeSDKIdentityServiceException,
  __Client,
  ChimeSDKIdentityClient,
  ChimeSDKIdentity,
  $Command,
  CreateAppInstanceAdminCommand,
  CreateAppInstanceBotCommand,
  CreateAppInstanceCommand,
  CreateAppInstanceUserCommand,
  DeleteAppInstanceAdminCommand,
  DeleteAppInstanceBotCommand,
  DeleteAppInstanceCommand,
  DeleteAppInstanceUserCommand,
  DeregisterAppInstanceUserEndpointCommand,
  DescribeAppInstanceAdminCommand,
  DescribeAppInstanceBotCommand,
  DescribeAppInstanceCommand,
  DescribeAppInstanceUserCommand,
  DescribeAppInstanceUserEndpointCommand,
  GetAppInstanceRetentionSettingsCommand,
  ListAppInstanceAdminsCommand,
  ListAppInstanceBotsCommand,
  ListAppInstanceUserEndpointsCommand,
  ListAppInstanceUsersCommand,
  ListAppInstancesCommand,
  ListTagsForResourceCommand,
  PutAppInstanceRetentionSettingsCommand,
  PutAppInstanceUserExpirationSettingsCommand,
  RegisterAppInstanceUserEndpointCommand,
  TagResourceCommand,
  UntagResourceCommand,
  UpdateAppInstanceBotCommand,
  UpdateAppInstanceCommand,
  UpdateAppInstanceUserCommand,
  UpdateAppInstanceUserEndpointCommand,
  paginateListAppInstanceAdmins,
  paginateListAppInstanceBots,
  paginateListAppInstanceUserEndpoints,
  paginateListAppInstanceUsers,
  paginateListAppInstances,
  AllowMessages,
  StandardMessages,
  TargetedMessages,
  RespondsTo,
  ExpirationCriterion,
  EndpointStatus,
  EndpointStatusReason,
  AppInstanceUserEndpointType,
  ErrorCode,
  BadRequestException,
  ConflictException,
  ForbiddenException,
  ResourceLimitExceededException,
  ServiceFailureException,
  ServiceUnavailableException,
  ThrottledClientException,
  UnauthorizedClientException,
  NotFoundException,
  AppInstanceFilterSensitiveLog,
  IdentityFilterSensitiveLog,
  AppInstanceAdminFilterSensitiveLog,
  AppInstanceAdminSummaryFilterSensitiveLog,
  AppInstanceBotFilterSensitiveLog,
  AppInstanceBotSummaryFilterSensitiveLog,
  AppInstanceSummaryFilterSensitiveLog,
  AppInstanceUserFilterSensitiveLog,
  EndpointAttributesFilterSensitiveLog,
  AppInstanceUserEndpointFilterSensitiveLog,
  AppInstanceUserEndpointSummaryFilterSensitiveLog,
  AppInstanceUserSummaryFilterSensitiveLog,
  TagFilterSensitiveLog,
  CreateAppInstanceRequestFilterSensitiveLog,
  CreateAppInstanceAdminResponseFilterSensitiveLog,
  CreateAppInstanceBotRequestFilterSensitiveLog,
  CreateAppInstanceUserRequestFilterSensitiveLog,
  DescribeAppInstanceResponseFilterSensitiveLog,
  DescribeAppInstanceAdminResponseFilterSensitiveLog,
  DescribeAppInstanceBotResponseFilterSensitiveLog,
  DescribeAppInstanceUserResponseFilterSensitiveLog,
  DescribeAppInstanceUserEndpointResponseFilterSensitiveLog,
  ListAppInstanceAdminsRequestFilterSensitiveLog,
  ListAppInstanceAdminsResponseFilterSensitiveLog,
  ListAppInstanceBotsRequestFilterSensitiveLog,
  ListAppInstanceBotsResponseFilterSensitiveLog,
  ListAppInstancesRequestFilterSensitiveLog,
  ListAppInstancesResponseFilterSensitiveLog,
  ListAppInstanceUserEndpointsRequestFilterSensitiveLog,
  ListAppInstanceUserEndpointsResponseFilterSensitiveLog,
  ListAppInstanceUsersRequestFilterSensitiveLog,
  ListAppInstanceUsersResponseFilterSensitiveLog,
  ListTagsForResourceResponseFilterSensitiveLog,
  RegisterAppInstanceUserEndpointRequestFilterSensitiveLog,
  TagResourceRequestFilterSensitiveLog,
  UntagResourceRequestFilterSensitiveLog,
  UpdateAppInstanceRequestFilterSensitiveLog,
  UpdateAppInstanceBotRequestFilterSensitiveLog,
  UpdateAppInstanceUserRequestFilterSensitiveLog,
  UpdateAppInstanceUserEndpointRequestFilterSensitiveLog
});
