// @azure/arm-cosmosdb/dist-esm/src/models/parameters.js
/*
 * Copyright (c) Microsoft Corporation.
 * Licensed under the MIT License.
 *
 * Code generated by Microsoft (R) AutoRest Code Generator.
 * Changes may cause incorrect behavior and will be lost if the code is regenerated.
 */
import { DatabaseAccountUpdateParameters as DatabaseAccountUpdateParametersMapper, DatabaseAccountCreateUpdateParameters as DatabaseAccountCreateUpdateParametersMapper, FailoverPolicies as FailoverPoliciesMapper, RegionForOnlineOffline as RegionForOnlineOfflineMapper, DatabaseAccountRegenerateKeyParameters as DatabaseAccountRegenerateKeyParametersMapper, SqlDatabaseCreateUpdateParameters as SqlDatabaseCreateUpdateParametersMapper, ThroughputSettingsUpdateParameters as ThroughputSettingsUpdateParametersMapper, SqlContainerCreateUpdateParameters as SqlContainerCreateUpdateParametersMapper, ClientEncryptionKeyCreateUpdateParameters as ClientEncryptionKeyCreateUpdateParametersMapper, SqlStoredProcedureCreateUpdateParameters as SqlStoredProcedureCreateUpdateParametersMapper, SqlUserDefinedFunctionCreateUpdateParameters as SqlUserDefinedFunctionCreateUpdateParametersMapper, SqlTriggerCreateUpdateParameters as SqlTriggerCreateUpdateParametersMapper, SqlRoleDefinitionCreateUpdateParameters as SqlRoleDefinitionCreateUpdateParametersMapper, SqlRoleAssignmentCreateUpdateParameters as SqlRoleAssignmentCreateUpdateParametersMapper, ContinuousBackupRestoreLocation as ContinuousBackupRestoreLocationMapper, MongoDBDatabaseCreateUpdateParameters as MongoDBDatabaseCreateUpdateParametersMapper, MongoDBCollectionCreateUpdateParameters as MongoDBCollectionCreateUpdateParametersMapper, MongoRoleDefinitionCreateUpdateParameters as MongoRoleDefinitionCreateUpdateParametersMapper, MongoUserDefinitionCreateUpdateParameters as MongoUserDefinitionCreateUpdateParametersMapper, TableCreateUpdateParameters as TableCreateUpdateParametersMapper, CassandraKeyspaceCreateUpdateParameters as CassandraKeyspaceCreateUpdateParametersMapper, CassandraTableCreateUpdateParameters as CassandraTableCreateUpdateParametersMapper, GremlinDatabaseCreateUpdateParameters as GremlinDatabaseCreateUpdateParametersMapper, GremlinGraphCreateUpdateParameters as GremlinGraphCreateUpdateParametersMapper, ClusterResource as ClusterResourceMapper, CommandPostBody as CommandPostBodyMapper, DataCenterResource as DataCenterResourceMapper, NotebookWorkspaceCreateUpdateParameters as NotebookWorkspaceCreateUpdateParametersMapper, PrivateEndpointConnection as PrivateEndpointConnectionMapper, ServiceResourceCreateUpdateParameters as ServiceResourceCreateUpdateParametersMapper } from "../models/mappers";
export const accept = {
    parameterPath: "accept",
    mapper: {
        defaultValue: "application/json",
        isConstant: true,
        serializedName: "Accept",
        type: {
            name: "String"
        }
    }
};
export const $host = {
    parameterPath: "$host",
    mapper: {
        serializedName: "$host",
        required: true,
        type: {
            name: "String"
        }
    },
    skipEncoding: true
};
export const subscriptionId = {
    parameterPath: "subscriptionId",
    mapper: {
        constraints: {
            MinLength: 1
        },
        serializedName: "subscriptionId",
        required: true,
        type: {
            name: "String"
        }
    }
};
export const resourceGroupName = {
    parameterPath: "resourceGroupName",
    mapper: {
        constraints: {
            MaxLength: 90,
            MinLength: 1
        },
        serializedName: "resourceGroupName",
        required: true,
        type: {
            name: "String"
        }
    }
};
export const accountName = {
    parameterPath: "accountName",
    mapper: {
        constraints: {
            Pattern: new RegExp("^[a-z0-9]+(-[a-z0-9]+)*"),
            MaxLength: 50,
            MinLength: 3
        },
        serializedName: "accountName",
        required: true,
        type: {
            name: "String"
        }
    }
};
export const apiVersion = {
    parameterPath: "apiVersion",
    mapper: {
        defaultValue: "2023-11-15",
        isConstant: true,
        serializedName: "api-version",
        type: {
            name: "String"
        }
    }
};
export const contentType = {
    parameterPath: ["options", "contentType"],
    mapper: {
        defaultValue: "application/json",
        isConstant: true,
        serializedName: "Content-Type",
        type: {
            name: "String"
        }
    }
};
export const updateParameters = {
    parameterPath: "updateParameters",
    mapper: DatabaseAccountUpdateParametersMapper
};
export const createUpdateParameters = {
    parameterPath: "createUpdateParameters",
    mapper: DatabaseAccountCreateUpdateParametersMapper
};
export const failoverParameters = {
    parameterPath: "failoverParameters",
    mapper: FailoverPoliciesMapper
};
export const regionParameterForOffline = {
    parameterPath: "regionParameterForOffline",
    mapper: RegionForOnlineOfflineMapper
};
export const regionParameterForOnline = {
    parameterPath: "regionParameterForOnline",
    mapper: RegionForOnlineOfflineMapper
};
export const keyToRegenerate = {
    parameterPath: "keyToRegenerate",
    mapper: DatabaseAccountRegenerateKeyParametersMapper
};
export const filter = {
    parameterPath: "filter",
    mapper: {
        serializedName: "$filter",
        required: true,
        type: {
            name: "String"
        }
    }
};
export const filter1 = {
    parameterPath: ["options", "filter"],
    mapper: {
        serializedName: "$filter",
        type: {
            name: "String"
        }
    }
};
export const nextLink = {
    parameterPath: "nextLink",
    mapper: {
        serializedName: "nextLink",
        required: true,
        type: {
            name: "String"
        }
    },
    skipEncoding: true
};
export const databaseRid = {
    parameterPath: "databaseRid",
    mapper: {
        serializedName: "databaseRid",
        required: true,
        type: {
            name: "String"
        }
    }
};
export const collectionRid = {
    parameterPath: "collectionRid",
    mapper: {
        serializedName: "collectionRid",
        required: true,
        type: {
            name: "String"
        }
    }
};
export const region = {
    parameterPath: "region",
    mapper: {
        serializedName: "region",
        required: true,
        type: {
            name: "String"
        }
    }
};
export const sourceRegion = {
    parameterPath: "sourceRegion",
    mapper: {
        serializedName: "sourceRegion",
        required: true,
        type: {
            name: "String"
        }
    }
};
export const targetRegion = {
    parameterPath: "targetRegion",
    mapper: {
        serializedName: "targetRegion",
        required: true,
        type: {
            name: "String"
        }
    }
};
export const partitionKeyRangeId = {
    parameterPath: "partitionKeyRangeId",
    mapper: {
        serializedName: "partitionKeyRangeId",
        required: true,
        type: {
            name: "String"
        }
    }
};
export const databaseName = {
    parameterPath: "databaseName",
    mapper: {
        serializedName: "databaseName",
        required: true,
        type: {
            name: "String"
        }
    }
};
export const createUpdateSqlDatabaseParameters = {
    parameterPath: "createUpdateSqlDatabaseParameters",
    mapper: SqlDatabaseCreateUpdateParametersMapper
};
export const updateThroughputParameters = {
    parameterPath: "updateThroughputParameters",
    mapper: ThroughputSettingsUpdateParametersMapper
};
export const containerName = {
    parameterPath: "containerName",
    mapper: {
        serializedName: "containerName",
        required: true,
        type: {
            name: "String"
        }
    }
};
export const createUpdateSqlContainerParameters = {
    parameterPath: "createUpdateSqlContainerParameters",
    mapper: SqlContainerCreateUpdateParametersMapper
};
export const clientEncryptionKeyName = {
    parameterPath: "clientEncryptionKeyName",
    mapper: {
        serializedName: "clientEncryptionKeyName",
        required: true,
        type: {
            name: "String"
        }
    }
};
export const createUpdateClientEncryptionKeyParameters = {
    parameterPath: "createUpdateClientEncryptionKeyParameters",
    mapper: ClientEncryptionKeyCreateUpdateParametersMapper
};
export const storedProcedureName = {
    parameterPath: "storedProcedureName",
    mapper: {
        serializedName: "storedProcedureName",
        required: true,
        type: {
            name: "String"
        }
    }
};
export const createUpdateSqlStoredProcedureParameters = {
    parameterPath: "createUpdateSqlStoredProcedureParameters",
    mapper: SqlStoredProcedureCreateUpdateParametersMapper
};
export const userDefinedFunctionName = {
    parameterPath: "userDefinedFunctionName",
    mapper: {
        serializedName: "userDefinedFunctionName",
        required: true,
        type: {
            name: "String"
        }
    }
};
export const createUpdateSqlUserDefinedFunctionParameters = {
    parameterPath: "createUpdateSqlUserDefinedFunctionParameters",
    mapper: SqlUserDefinedFunctionCreateUpdateParametersMapper
};
export const triggerName = {
    parameterPath: "triggerName",
    mapper: {
        serializedName: "triggerName",
        required: true,
        type: {
            name: "String"
        }
    }
};
export const createUpdateSqlTriggerParameters = {
    parameterPath: "createUpdateSqlTriggerParameters",
    mapper: SqlTriggerCreateUpdateParametersMapper
};
export const roleDefinitionId = {
    parameterPath: "roleDefinitionId",
    mapper: {
        serializedName: "roleDefinitionId",
        required: true,
        type: {
            name: "String"
        }
    }
};
export const createUpdateSqlRoleDefinitionParameters = {
    parameterPath: "createUpdateSqlRoleDefinitionParameters",
    mapper: SqlRoleDefinitionCreateUpdateParametersMapper
};
export const roleAssignmentId = {
    parameterPath: "roleAssignmentId",
    mapper: {
        serializedName: "roleAssignmentId",
        required: true,
        type: {
            name: "String"
        }
    }
};
export const createUpdateSqlRoleAssignmentParameters = {
    parameterPath: "createUpdateSqlRoleAssignmentParameters",
    mapper: SqlRoleAssignmentCreateUpdateParametersMapper
};
export const location = {
    parameterPath: "location",
    mapper: ContinuousBackupRestoreLocationMapper
};
export const createUpdateMongoDBDatabaseParameters = {
    parameterPath: "createUpdateMongoDBDatabaseParameters",
    mapper: MongoDBDatabaseCreateUpdateParametersMapper
};
export const collectionName = {
    parameterPath: "collectionName",
    mapper: {
        serializedName: "collectionName",
        required: true,
        type: {
            name: "String"
        }
    }
};
export const createUpdateMongoDBCollectionParameters = {
    parameterPath: "createUpdateMongoDBCollectionParameters",
    mapper: MongoDBCollectionCreateUpdateParametersMapper
};
export const mongoRoleDefinitionId = {
    parameterPath: "mongoRoleDefinitionId",
    mapper: {
        serializedName: "mongoRoleDefinitionId",
        required: true,
        type: {
            name: "String"
        }
    }
};
export const createUpdateMongoRoleDefinitionParameters = {
    parameterPath: "createUpdateMongoRoleDefinitionParameters",
    mapper: MongoRoleDefinitionCreateUpdateParametersMapper
};
export const mongoUserDefinitionId = {
    parameterPath: "mongoUserDefinitionId",
    mapper: {
        serializedName: "mongoUserDefinitionId",
        required: true,
        type: {
            name: "String"
        }
    }
};
export const createUpdateMongoUserDefinitionParameters = {
    parameterPath: "createUpdateMongoUserDefinitionParameters",
    mapper: MongoUserDefinitionCreateUpdateParametersMapper
};
export const tableName = {
    parameterPath: "tableName",
    mapper: {
        serializedName: "tableName",
        required: true,
        type: {
            name: "String"
        }
    }
};
export const createUpdateTableParameters = {
    parameterPath: "createUpdateTableParameters",
    mapper: TableCreateUpdateParametersMapper
};
export const keyspaceName = {
    parameterPath: "keyspaceName",
    mapper: {
        serializedName: "keyspaceName",
        required: true,
        type: {
            name: "String"
        }
    }
};
export const createUpdateCassandraKeyspaceParameters = {
    parameterPath: "createUpdateCassandraKeyspaceParameters",
    mapper: CassandraKeyspaceCreateUpdateParametersMapper
};
export const createUpdateCassandraTableParameters = {
    parameterPath: "createUpdateCassandraTableParameters",
    mapper: CassandraTableCreateUpdateParametersMapper
};
export const createUpdateGremlinDatabaseParameters = {
    parameterPath: "createUpdateGremlinDatabaseParameters",
    mapper: GremlinDatabaseCreateUpdateParametersMapper
};
export const graphName = {
    parameterPath: "graphName",
    mapper: {
        serializedName: "graphName",
        required: true,
        type: {
            name: "String"
        }
    }
};
export const createUpdateGremlinGraphParameters = {
    parameterPath: "createUpdateGremlinGraphParameters",
    mapper: GremlinGraphCreateUpdateParametersMapper
};
export const location1 = {
    parameterPath: "location",
    mapper: {
        serializedName: "location",
        required: true,
        type: {
            name: "String"
        }
    }
};
export const clusterName = {
    parameterPath: "clusterName",
    mapper: {
        constraints: {
            Pattern: new RegExp("^[a-zA-Z0-9]+(-[a-zA-Z0-9]+)*$"),
            MaxLength: 100,
            MinLength: 1
        },
        serializedName: "clusterName",
        required: true,
        type: {
            name: "String"
        }
    }
};
export const body = {
    parameterPath: "body",
    mapper: ClusterResourceMapper
};
export const body1 = {
    parameterPath: "body",
    mapper: CommandPostBodyMapper
};
export const dataCenterName = {
    parameterPath: "dataCenterName",
    mapper: {
        constraints: {
            Pattern: new RegExp("^[a-zA-Z0-9]+(-[a-zA-Z0-9]+)*$"),
            MaxLength: 100,
            MinLength: 1
        },
        serializedName: "dataCenterName",
        required: true,
        type: {
            name: "String"
        }
    }
};
export const body2 = {
    parameterPath: "body",
    mapper: DataCenterResourceMapper
};
export const notebookWorkspaceName = {
    parameterPath: "notebookWorkspaceName",
    mapper: {
        serializedName: "notebookWorkspaceName",
        required: true,
        type: {
            name: "String"
        }
    }
};
export const notebookCreateUpdateParameters = {
    parameterPath: "notebookCreateUpdateParameters",
    mapper: NotebookWorkspaceCreateUpdateParametersMapper
};
export const privateEndpointConnectionName = {
    parameterPath: "privateEndpointConnectionName",
    mapper: {
        serializedName: "privateEndpointConnectionName",
        required: true,
        type: {
            name: "String"
        }
    }
};
export const parameters = {
    parameterPath: "parameters",
    mapper: PrivateEndpointConnectionMapper
};
export const groupName = {
    parameterPath: "groupName",
    mapper: {
        serializedName: "groupName",
        required: true,
        type: {
            name: "String"
        }
    }
};
export const instanceId = {
    parameterPath: "instanceId",
    mapper: {
        serializedName: "instanceId",
        required: true,
        type: {
            name: "String"
        }
    }
};
export const restorableSqlDatabaseRid = {
    parameterPath: ["options", "restorableSqlDatabaseRid"],
    mapper: {
        serializedName: "restorableSqlDatabaseRid",
        type: {
            name: "String"
        }
    }
};
export const startTime = {
    parameterPath: ["options", "startTime"],
    mapper: {
        serializedName: "startTime",
        type: {
            name: "String"
        }
    }
};
export const endTime = {
    parameterPath: ["options", "endTime"],
    mapper: {
        serializedName: "endTime",
        type: {
            name: "String"
        }
    }
};
export const restoreLocation = {
    parameterPath: ["options", "restoreLocation"],
    mapper: {
        serializedName: "restoreLocation",
        type: {
            name: "String"
        }
    }
};
export const restoreTimestampInUtc = {
    parameterPath: ["options", "restoreTimestampInUtc"],
    mapper: {
        serializedName: "restoreTimestampInUtc",
        type: {
            name: "String"
        }
    }
};
export const restorableMongodbDatabaseRid = {
    parameterPath: ["options", "restorableMongodbDatabaseRid"],
    mapper: {
        serializedName: "restorableMongodbDatabaseRid",
        type: {
            name: "String"
        }
    }
};
export const restorableGremlinDatabaseRid = {
    parameterPath: ["options", "restorableGremlinDatabaseRid"],
    mapper: {
        serializedName: "restorableGremlinDatabaseRid",
        type: {
            name: "String"
        }
    }
};
export const createUpdateParameters1 = {
    parameterPath: "createUpdateParameters",
    mapper: ServiceResourceCreateUpdateParametersMapper
};
export const serviceName = {
    parameterPath: "serviceName",
    mapper: {
        constraints: {
            MaxLength: 50,
            MinLength: 3
        },
        serializedName: "serviceName",
        required: true,
        type: {
            name: "String"
        }
    }
};
