// @azure/arm-kubernetesconfiguration/dist-esm/src/models/index.js
/*
 * Copyright (c) Microsoft Corporation.
 * Licensed under the MIT License.
 *
 * Code generated by Microsoft (R) AutoRest Code Generator.
 * Changes may cause incorrect behavior and will be lost if the code is regenerated.
 */
/** Known values of {@link ProvisioningState} that the service accepts. */
export var KnownProvisioningState;
(function (KnownProvisioningState) {
    /** Succeeded */
    KnownProvisioningState["Succeeded"] = "Succeeded";
    /** Failed */
    KnownProvisioningState["Failed"] = "Failed";
    /** Canceled */
    KnownProvisioningState["Canceled"] = "Canceled";
    /** Creating */
    KnownProvisioningState["Creating"] = "Creating";
    /** Updating */
    KnownProvisioningState["Updating"] = "Updating";
    /** Deleting */
    KnownProvisioningState["Deleting"] = "Deleting";
})(KnownProvisioningState || (KnownProvisioningState = {}));
/** Known values of {@link LevelType} that the service accepts. */
export var KnownLevelType;
(function (KnownLevelType) {
    /** Error */
    KnownLevelType["Error"] = "Error";
    /** Warning */
    KnownLevelType["Warning"] = "Warning";
    /** Information */
    KnownLevelType["Information"] = "Information";
})(KnownLevelType || (KnownLevelType = {}));
/** Known values of {@link CreatedByType} that the service accepts. */
export var KnownCreatedByType;
(function (KnownCreatedByType) {
    /** User */
    KnownCreatedByType["User"] = "User";
    /** Application */
    KnownCreatedByType["Application"] = "Application";
    /** ManagedIdentity */
    KnownCreatedByType["ManagedIdentity"] = "ManagedIdentity";
    /** Key */
    KnownCreatedByType["Key"] = "Key";
})(KnownCreatedByType || (KnownCreatedByType = {}));
/** Known values of {@link ScopeType} that the service accepts. */
export var KnownScopeType;
(function (KnownScopeType) {
    /** Cluster */
    KnownScopeType["Cluster"] = "cluster";
    /** Namespace */
    KnownScopeType["Namespace"] = "namespace";
})(KnownScopeType || (KnownScopeType = {}));
/** Known values of {@link SourceKindType} that the service accepts. */
export var KnownSourceKindType;
(function (KnownSourceKindType) {
    /** GitRepository */
    KnownSourceKindType["GitRepository"] = "GitRepository";
    /** Bucket */
    KnownSourceKindType["Bucket"] = "Bucket";
    /** AzureBlob */
    KnownSourceKindType["AzureBlob"] = "AzureBlob";
})(KnownSourceKindType || (KnownSourceKindType = {}));
/** Known values of {@link FluxComplianceState} that the service accepts. */
export var KnownFluxComplianceState;
(function (KnownFluxComplianceState) {
    /** Compliant */
    KnownFluxComplianceState["Compliant"] = "Compliant";
    /** NonCompliant */
    KnownFluxComplianceState["NonCompliant"] = "Non-Compliant";
    /** Pending */
    KnownFluxComplianceState["Pending"] = "Pending";
    /** Suspended */
    KnownFluxComplianceState["Suspended"] = "Suspended";
    /** Unknown */
    KnownFluxComplianceState["Unknown"] = "Unknown";
})(KnownFluxComplianceState || (KnownFluxComplianceState = {}));
/** Known values of {@link OperatorType} that the service accepts. */
export var KnownOperatorType;
(function (KnownOperatorType) {
    /** Flux */
    KnownOperatorType["Flux"] = "Flux";
})(KnownOperatorType || (KnownOperatorType = {}));
/** Known values of {@link OperatorScopeType} that the service accepts. */
export var KnownOperatorScopeType;
(function (KnownOperatorScopeType) {
    /** Cluster */
    KnownOperatorScopeType["Cluster"] = "cluster";
    /** Namespace */
    KnownOperatorScopeType["Namespace"] = "namespace";
})(KnownOperatorScopeType || (KnownOperatorScopeType = {}));
/** Known values of {@link ProvisioningStateType} that the service accepts. */
export var KnownProvisioningStateType;
(function (KnownProvisioningStateType) {
    /** Accepted */
    KnownProvisioningStateType["Accepted"] = "Accepted";
    /** Deleting */
    KnownProvisioningStateType["Deleting"] = "Deleting";
    /** Running */
    KnownProvisioningStateType["Running"] = "Running";
    /** Succeeded */
    KnownProvisioningStateType["Succeeded"] = "Succeeded";
    /** Failed */
    KnownProvisioningStateType["Failed"] = "Failed";
})(KnownProvisioningStateType || (KnownProvisioningStateType = {}));
/** Known values of {@link ComplianceStateType} that the service accepts. */
export var KnownComplianceStateType;
(function (KnownComplianceStateType) {
    /** Pending */
    KnownComplianceStateType["Pending"] = "Pending";
    /** Compliant */
    KnownComplianceStateType["Compliant"] = "Compliant";
    /** Noncompliant */
    KnownComplianceStateType["Noncompliant"] = "Noncompliant";
    /** Installed */
    KnownComplianceStateType["Installed"] = "Installed";
    /** Failed */
    KnownComplianceStateType["Failed"] = "Failed";
})(KnownComplianceStateType || (KnownComplianceStateType = {}));
/** Known values of {@link MessageLevelType} that the service accepts. */
export var KnownMessageLevelType;
(function (KnownMessageLevelType) {
    /** Error */
    KnownMessageLevelType["Error"] = "Error";
    /** Warning */
    KnownMessageLevelType["Warning"] = "Warning";
    /** Information */
    KnownMessageLevelType["Information"] = "Information";
})(KnownMessageLevelType || (KnownMessageLevelType = {}));
/** Known values of {@link KustomizationValidationType} that the service accepts. */
export var KnownKustomizationValidationType;
(function (KnownKustomizationValidationType) {
    /** None */
    KnownKustomizationValidationType["None"] = "none";
    /** Client */
    KnownKustomizationValidationType["Client"] = "client";
    /** Server */
    KnownKustomizationValidationType["Server"] = "server";
})(KnownKustomizationValidationType || (KnownKustomizationValidationType = {}));
