// @azure/arm-servicefabricmesh/esm/operations/codePackage.js
/*
 * Copyright (c) Microsoft Corporation. All rights reserved.
 * Licensed under the MIT License. See License.txt in the project root for
 * license information.
 *
 * Code generated by Microsoft (R) AutoRest Code Generator.
 * Changes may cause incorrect behavior and will be lost if the code is
 * regenerated.
 */
import * as msRest from "@azure/ms-rest-js";
import * as Mappers from "../models/codePackageMappers";
import * as Parameters from "../models/parameters";
/** Class representing a CodePackage. */
var CodePackage = /** @class */ (function () {
    /**
     * Create a CodePackage.
     * @param {ServiceFabricMeshManagementClientContext} client Reference to the service client.
     */
    function CodePackage(client) {
        this.client = client;
    }
    CodePackage.prototype.getContainerLogs = function (resourceGroupName, applicationResourceName, serviceResourceName, replicaName, codePackageName, options, callback) {
        return this.client.sendOperationRequest({
            resourceGroupName: resourceGroupName,
            applicationResourceName: applicationResourceName,
            serviceResourceName: serviceResourceName,
            replicaName: replicaName,
            codePackageName: codePackageName,
            options: options
        }, getContainerLogsOperationSpec, callback);
    };
    return CodePackage;
}());
export { CodePackage };
// Operation Specifications
var serializer = new msRest.Serializer(Mappers);
var getContainerLogsOperationSpec = {
    httpMethod: "GET",
    path: "subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.ServiceFabricMesh/applications/{applicationResourceName}/services/{serviceResourceName}/replicas/{replicaName}/codePackages/{codePackageName}/logs",
    urlParameters: [
        Parameters.subscriptionId,
        Parameters.resourceGroupName,
        Parameters.applicationResourceName,
        Parameters.serviceResourceName,
        Parameters.replicaName,
        Parameters.codePackageName
    ],
    queryParameters: [
        Parameters.apiVersion,
        Parameters.tail
    ],
    headerParameters: [
        Parameters.acceptLanguage
    ],
    responses: {
        200: {
            bodyMapper: Mappers.ContainerLogs
        },
        default: {
            bodyMapper: Mappers.ErrorModel
        }
    },
    serializer: serializer
};
