// @azure/arm-servicefabricmesh/esm/operations/secretValueOperations.js
/*
 * Copyright (c) Microsoft Corporation. All rights reserved.
 * Licensed under the MIT License. See License.txt in the project root for
 * license information.
 *
 * Code generated by Microsoft (R) AutoRest Code Generator.
 * Changes may cause incorrect behavior and will be lost if the code is
 * regenerated.
 */
import { __assign } from "tslib";
import * as msRest from "@azure/ms-rest-js";
import * as Mappers from "../models/secretValueOperationsMappers";
import * as Parameters from "../models/parameters";
/** Class representing a SecretValueOperations. */
var SecretValueOperations = /** @class */ (function () {
    /**
     * Create a SecretValueOperations.
     * @param {ServiceFabricMeshManagementClientContext} client Reference to the service client.
     */
    function SecretValueOperations(client) {
        this.client = client;
    }
    SecretValueOperations.prototype.create = function (resourceGroupName, secretResourceName, secretValueResourceName, secretValueResourceDescription, options, callback) {
        return this.client.sendOperationRequest({
            resourceGroupName: resourceGroupName,
            secretResourceName: secretResourceName,
            secretValueResourceName: secretValueResourceName,
            secretValueResourceDescription: secretValueResourceDescription,
            options: options
        }, createOperationSpec, callback);
    };
    SecretValueOperations.prototype.get = function (resourceGroupName, secretResourceName, secretValueResourceName, options, callback) {
        return this.client.sendOperationRequest({
            resourceGroupName: resourceGroupName,
            secretResourceName: secretResourceName,
            secretValueResourceName: secretValueResourceName,
            options: options
        }, getOperationSpec, callback);
    };
    SecretValueOperations.prototype.deleteMethod = function (resourceGroupName, secretResourceName, secretValueResourceName, options, callback) {
        return this.client.sendOperationRequest({
            resourceGroupName: resourceGroupName,
            secretResourceName: secretResourceName,
            secretValueResourceName: secretValueResourceName,
            options: options
        }, deleteMethodOperationSpec, callback);
    };
    SecretValueOperations.prototype.list = function (resourceGroupName, secretResourceName, options, callback) {
        return this.client.sendOperationRequest({
            resourceGroupName: resourceGroupName,
            secretResourceName: secretResourceName,
            options: options
        }, listOperationSpec, callback);
    };
    SecretValueOperations.prototype.listValue = function (resourceGroupName, secretResourceName, secretValueResourceName, options, callback) {
        return this.client.sendOperationRequest({
            resourceGroupName: resourceGroupName,
            secretResourceName: secretResourceName,
            secretValueResourceName: secretValueResourceName,
            options: options
        }, listValueOperationSpec, callback);
    };
    SecretValueOperations.prototype.listNext = function (nextPageLink, options, callback) {
        return this.client.sendOperationRequest({
            nextPageLink: nextPageLink,
            options: options
        }, listNextOperationSpec, callback);
    };
    return SecretValueOperations;
}());
export { SecretValueOperations };
// Operation Specifications
var serializer = new msRest.Serializer(Mappers);
var createOperationSpec = {
    httpMethod: "PUT",
    path: "subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.ServiceFabricMesh/secrets/{secretResourceName}/values/{secretValueResourceName}",
    urlParameters: [
        Parameters.subscriptionId,
        Parameters.resourceGroupName,
        Parameters.secretResourceName,
        Parameters.secretValueResourceName
    ],
    queryParameters: [
        Parameters.apiVersion
    ],
    headerParameters: [
        Parameters.acceptLanguage
    ],
    requestBody: {
        parameterPath: "secretValueResourceDescription",
        mapper: __assign(__assign({}, Mappers.SecretValueResourceDescription), { required: true })
    },
    responses: {
        200: {
            bodyMapper: Mappers.SecretValueResourceDescription
        },
        201: {
            bodyMapper: Mappers.SecretValueResourceDescription
        },
        202: {},
        default: {
            bodyMapper: Mappers.ErrorModel
        }
    },
    serializer: serializer
};
var getOperationSpec = {
    httpMethod: "GET",
    path: "subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.ServiceFabricMesh/secrets/{secretResourceName}/values/{secretValueResourceName}",
    urlParameters: [
        Parameters.subscriptionId,
        Parameters.resourceGroupName,
        Parameters.secretResourceName,
        Parameters.secretValueResourceName
    ],
    queryParameters: [
        Parameters.apiVersion
    ],
    headerParameters: [
        Parameters.acceptLanguage
    ],
    responses: {
        200: {
            bodyMapper: Mappers.SecretValueResourceDescription
        },
        default: {
            bodyMapper: Mappers.ErrorModel
        }
    },
    serializer: serializer
};
var deleteMethodOperationSpec = {
    httpMethod: "DELETE",
    path: "subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.ServiceFabricMesh/secrets/{secretResourceName}/values/{secretValueResourceName}",
    urlParameters: [
        Parameters.subscriptionId,
        Parameters.resourceGroupName,
        Parameters.secretResourceName,
        Parameters.secretValueResourceName
    ],
    queryParameters: [
        Parameters.apiVersion
    ],
    headerParameters: [
        Parameters.acceptLanguage
    ],
    responses: {
        200: {},
        202: {},
        204: {},
        default: {
            bodyMapper: Mappers.ErrorModel
        }
    },
    serializer: serializer
};
var listOperationSpec = {
    httpMethod: "GET",
    path: "subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.ServiceFabricMesh/secrets/{secretResourceName}/values",
    urlParameters: [
        Parameters.subscriptionId,
        Parameters.resourceGroupName,
        Parameters.secretResourceName
    ],
    queryParameters: [
        Parameters.apiVersion
    ],
    headerParameters: [
        Parameters.acceptLanguage
    ],
    responses: {
        200: {
            bodyMapper: Mappers.SecretValueResourceDescriptionList
        },
        default: {
            bodyMapper: Mappers.ErrorModel
        }
    },
    serializer: serializer
};
var listValueOperationSpec = {
    httpMethod: "POST",
    path: "subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.ServiceFabricMesh/secrets/{secretResourceName}/values/{secretValueResourceName}/list_value",
    urlParameters: [
        Parameters.subscriptionId,
        Parameters.resourceGroupName,
        Parameters.secretResourceName,
        Parameters.secretValueResourceName
    ],
    queryParameters: [
        Parameters.apiVersion
    ],
    headerParameters: [
        Parameters.acceptLanguage
    ],
    responses: {
        200: {
            bodyMapper: Mappers.SecretValue
        },
        default: {
            bodyMapper: Mappers.ErrorModel
        }
    },
    serializer: serializer
};
var listNextOperationSpec = {
    httpMethod: "GET",
    baseUrl: "https://management.azure.com",
    path: "{nextLink}",
    urlParameters: [
        Parameters.nextPageLink
    ],
    headerParameters: [
        Parameters.acceptLanguage
    ],
    responses: {
        200: {
            bodyMapper: Mappers.SecretValueResourceDescriptionList
        },
        default: {
            bodyMapper: Mappers.ErrorModel
        }
    },
    serializer: serializer
};
