// @aws-sdk/client-databrew/dist-cjs/index.js
"use strict";
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __name = (target, value) => __defProp(target, "name", { value, configurable: true });
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// src/index.ts
var src_exports = {};
__export(src_exports, {
  AccessDeniedException: () => AccessDeniedException,
  AnalyticsMode: () => AnalyticsMode,
  BatchDeleteRecipeVersionCommand: () => BatchDeleteRecipeVersionCommand,
  CompressionFormat: () => CompressionFormat,
  ConflictException: () => ConflictException,
  CreateDatasetCommand: () => CreateDatasetCommand,
  CreateProfileJobCommand: () => CreateProfileJobCommand,
  CreateProjectCommand: () => CreateProjectCommand,
  CreateRecipeCommand: () => CreateRecipeCommand,
  CreateRecipeJobCommand: () => CreateRecipeJobCommand,
  CreateRulesetCommand: () => CreateRulesetCommand,
  CreateScheduleCommand: () => CreateScheduleCommand,
  DataBrew: () => DataBrew,
  DataBrewClient: () => DataBrewClient,
  DataBrewServiceException: () => DataBrewServiceException,
  DatabaseOutputMode: () => DatabaseOutputMode,
  DeleteDatasetCommand: () => DeleteDatasetCommand,
  DeleteJobCommand: () => DeleteJobCommand,
  DeleteProjectCommand: () => DeleteProjectCommand,
  DeleteRecipeVersionCommand: () => DeleteRecipeVersionCommand,
  DeleteRulesetCommand: () => DeleteRulesetCommand,
  DeleteScheduleCommand: () => DeleteScheduleCommand,
  DescribeDatasetCommand: () => DescribeDatasetCommand,
  DescribeJobCommand: () => DescribeJobCommand,
  DescribeJobRunCommand: () => DescribeJobRunCommand,
  DescribeProjectCommand: () => DescribeProjectCommand,
  DescribeRecipeCommand: () => DescribeRecipeCommand,
  DescribeRulesetCommand: () => DescribeRulesetCommand,
  DescribeScheduleCommand: () => DescribeScheduleCommand,
  EncryptionMode: () => EncryptionMode,
  InputFormat: () => InputFormat,
  InternalServerException: () => InternalServerException,
  JobRunState: () => JobRunState,
  JobType: () => JobType,
  ListDatasetsCommand: () => ListDatasetsCommand,
  ListJobRunsCommand: () => ListJobRunsCommand,
  ListJobsCommand: () => ListJobsCommand,
  ListProjectsCommand: () => ListProjectsCommand,
  ListRecipeVersionsCommand: () => ListRecipeVersionsCommand,
  ListRecipesCommand: () => ListRecipesCommand,
  ListRulesetsCommand: () => ListRulesetsCommand,
  ListSchedulesCommand: () => ListSchedulesCommand,
  ListTagsForResourceCommand: () => ListTagsForResourceCommand,
  LogSubscription: () => LogSubscription,
  Order: () => Order,
  OrderedBy: () => OrderedBy,
  OutputFormat: () => OutputFormat,
  ParameterType: () => ParameterType,
  PublishRecipeCommand: () => PublishRecipeCommand,
  ResourceNotFoundException: () => ResourceNotFoundException,
  SampleMode: () => SampleMode,
  SampleType: () => SampleType,
  SendProjectSessionActionCommand: () => SendProjectSessionActionCommand,
  SendProjectSessionActionRequestFilterSensitiveLog: () => SendProjectSessionActionRequestFilterSensitiveLog,
  ServiceQuotaExceededException: () => ServiceQuotaExceededException,
  SessionStatus: () => SessionStatus,
  Source: () => Source,
  StartJobRunCommand: () => StartJobRunCommand,
  StartProjectSessionCommand: () => StartProjectSessionCommand,
  StartProjectSessionResponseFilterSensitiveLog: () => StartProjectSessionResponseFilterSensitiveLog,
  StopJobRunCommand: () => StopJobRunCommand,
  TagResourceCommand: () => TagResourceCommand,
  ThresholdType: () => ThresholdType,
  ThresholdUnit: () => ThresholdUnit,
  UntagResourceCommand: () => UntagResourceCommand,
  UpdateDatasetCommand: () => UpdateDatasetCommand,
  UpdateProfileJobCommand: () => UpdateProfileJobCommand,
  UpdateProjectCommand: () => UpdateProjectCommand,
  UpdateRecipeCommand: () => UpdateRecipeCommand,
  UpdateRecipeJobCommand: () => UpdateRecipeJobCommand,
  UpdateRulesetCommand: () => UpdateRulesetCommand,
  UpdateScheduleCommand: () => UpdateScheduleCommand,
  ValidationException: () => ValidationException,
  ValidationMode: () => ValidationMode,
  __Client: () => import_smithy_client.Client,
  paginateListDatasets: () => paginateListDatasets,
  paginateListJobRuns: () => paginateListJobRuns,
  paginateListJobs: () => paginateListJobs,
  paginateListProjects: () => paginateListProjects,
  paginateListRecipeVersions: () => paginateListRecipeVersions,
  paginateListRecipes: () => paginateListRecipes,
  paginateListRulesets: () => paginateListRulesets,
  paginateListSchedules: () => paginateListSchedules
});
module.exports = __toCommonJS(src_exports);

// src/DataBrewClient.ts
var import_middleware_host_header = require("@aws-sdk/middleware-host-header");
var import_middleware_logger = require("@aws-sdk/middleware-logger");
var import_middleware_recursion_detection = require("@aws-sdk/middleware-recursion-detection");
var import_middleware_user_agent = require("@aws-sdk/middleware-user-agent");
var import_config_resolver = require("@smithy/config-resolver");
var import_core = require("@smithy/core");
var import_middleware_content_length = require("@smithy/middleware-content-length");
var import_middleware_endpoint = require("@smithy/middleware-endpoint");
var import_middleware_retry = require("@smithy/middleware-retry");

var import_httpAuthSchemeProvider = require("./auth/httpAuthSchemeProvider");

// src/endpoint/EndpointParameters.ts
var resolveClientEndpointParameters = /* @__PURE__ */ __name((options) => {
  return {
    ...options,
    useDualstackEndpoint: options.useDualstackEndpoint ?? false,
    useFipsEndpoint: options.useFipsEndpoint ?? false,
    defaultSigningName: "databrew"
  };
}, "resolveClientEndpointParameters");
var commonParams = {
  UseFIPS: { type: "builtInParams", name: "useFipsEndpoint" },
  Endpoint: { type: "builtInParams", name: "endpoint" },
  Region: { type: "builtInParams", name: "region" },
  UseDualStack: { type: "builtInParams", name: "useDualstackEndpoint" }
};

// src/DataBrewClient.ts
var import_runtimeConfig = require("././runtimeConfig");

// src/runtimeExtensions.ts
var import_region_config_resolver = require("@aws-sdk/region-config-resolver");
var import_protocol_http = require("@smithy/protocol-http");
var import_smithy_client = require("@smithy/smithy-client");

// src/auth/httpAuthExtensionConfiguration.ts
var getHttpAuthExtensionConfiguration = /* @__PURE__ */ __name((runtimeConfig) => {
  const _httpAuthSchemes = runtimeConfig.httpAuthSchemes;
  let _httpAuthSchemeProvider = runtimeConfig.httpAuthSchemeProvider;
  let _credentials = runtimeConfig.credentials;
  return {
    setHttpAuthScheme(httpAuthScheme) {
      const index = _httpAuthSchemes.findIndex((scheme) => scheme.schemeId === httpAuthScheme.schemeId);
      if (index === -1) {
        _httpAuthSchemes.push(httpAuthScheme);
      } else {
        _httpAuthSchemes.splice(index, 1, httpAuthScheme);
      }
    },
    httpAuthSchemes() {
      return _httpAuthSchemes;
    },
    setHttpAuthSchemeProvider(httpAuthSchemeProvider) {
      _httpAuthSchemeProvider = httpAuthSchemeProvider;
    },
    httpAuthSchemeProvider() {
      return _httpAuthSchemeProvider;
    },
    setCredentials(credentials) {
      _credentials = credentials;
    },
    credentials() {
      return _credentials;
    }
  };
}, "getHttpAuthExtensionConfiguration");
var resolveHttpAuthRuntimeConfig = /* @__PURE__ */ __name((config) => {
  return {
    httpAuthSchemes: config.httpAuthSchemes(),
    httpAuthSchemeProvider: config.httpAuthSchemeProvider(),
    credentials: config.credentials()
  };
}, "resolveHttpAuthRuntimeConfig");

// src/runtimeExtensions.ts
var asPartial = /* @__PURE__ */ __name((t) => t, "asPartial");
var resolveRuntimeExtensions = /* @__PURE__ */ __name((runtimeConfig, extensions) => {
  const extensionConfiguration = {
    ...asPartial((0, import_region_config_resolver.getAwsRegionExtensionConfiguration)(runtimeConfig)),
    ...asPartial((0, import_smithy_client.getDefaultExtensionConfiguration)(runtimeConfig)),
    ...asPartial((0, import_protocol_http.getHttpHandlerExtensionConfiguration)(runtimeConfig)),
    ...asPartial(getHttpAuthExtensionConfiguration(runtimeConfig))
  };
  extensions.forEach((extension) => extension.configure(extensionConfiguration));
  return {
    ...runtimeConfig,
    ...(0, import_region_config_resolver.resolveAwsRegionExtensionConfiguration)(extensionConfiguration),
    ...(0, import_smithy_client.resolveDefaultRuntimeConfig)(extensionConfiguration),
    ...(0, import_protocol_http.resolveHttpHandlerRuntimeConfig)(extensionConfiguration),
    ...resolveHttpAuthRuntimeConfig(extensionConfiguration)
  };
}, "resolveRuntimeExtensions");

// src/DataBrewClient.ts
var _DataBrewClient = class _DataBrewClient extends import_smithy_client.Client {
  constructor(...[configuration]) {
    const _config_0 = (0, import_runtimeConfig.getRuntimeConfig)(configuration || {});
    const _config_1 = resolveClientEndpointParameters(_config_0);
    const _config_2 = (0, import_config_resolver.resolveRegionConfig)(_config_1);
    const _config_3 = (0, import_middleware_endpoint.resolveEndpointConfig)(_config_2);
    const _config_4 = (0, import_middleware_retry.resolveRetryConfig)(_config_3);
    const _config_5 = (0, import_middleware_host_header.resolveHostHeaderConfig)(_config_4);
    const _config_6 = (0, import_middleware_user_agent.resolveUserAgentConfig)(_config_5);
    const _config_7 = (0, import_httpAuthSchemeProvider.resolveHttpAuthSchemeConfig)(_config_6);
    const _config_8 = resolveRuntimeExtensions(_config_7, (configuration == null ? void 0 : configuration.extensions) || []);
    super(_config_8);
    this.config = _config_8;
    this.middlewareStack.use((0, import_middleware_retry.getRetryPlugin)(this.config));
    this.middlewareStack.use((0, import_middleware_content_length.getContentLengthPlugin)(this.config));
    this.middlewareStack.use((0, import_middleware_host_header.getHostHeaderPlugin)(this.config));
    this.middlewareStack.use((0, import_middleware_logger.getLoggerPlugin)(this.config));
    this.middlewareStack.use((0, import_middleware_recursion_detection.getRecursionDetectionPlugin)(this.config));
    this.middlewareStack.use((0, import_middleware_user_agent.getUserAgentPlugin)(this.config));
    this.middlewareStack.use((0, import_middleware_retry.getOmitRetryHeadersPlugin)(this.config));
    this.middlewareStack.use(
      (0, import_core.getHttpAuthSchemeEndpointRuleSetPlugin)(this.config, {
        httpAuthSchemeParametersProvider: this.getDefaultHttpAuthSchemeParametersProvider(),
        identityProviderConfigProvider: this.getIdentityProviderConfigProvider()
      })
    );
    this.middlewareStack.use((0, import_core.getHttpSigningPlugin)(this.config));
  }
  /**
   * Destroy underlying resources, like sockets. It's usually not necessary to do this.
   * However in Node.js, it's best to explicitly shut down the client's agent when it is no longer needed.
   * Otherwise, sockets might stay open for quite a long time before the server terminates them.
   */
  destroy() {
    super.destroy();
  }
  getDefaultHttpAuthSchemeParametersProvider() {
    return import_httpAuthSchemeProvider.defaultDataBrewHttpAuthSchemeParametersProvider;
  }
  getIdentityProviderConfigProvider() {
    return async (config) => new import_core.DefaultIdentityProviderConfig({
      "aws.auth#sigv4": config.credentials
    });
  }
};
__name(_DataBrewClient, "DataBrewClient");
var DataBrewClient = _DataBrewClient;

// src/DataBrew.ts


// src/commands/BatchDeleteRecipeVersionCommand.ts

var import_middleware_serde = require("@smithy/middleware-serde");

var import_types = require("@smithy/types");

// src/protocols/Aws_restJson1.ts
var import_core2 = require("@aws-sdk/core");



// src/models/DataBrewServiceException.ts

var _DataBrewServiceException = class _DataBrewServiceException extends import_smithy_client.ServiceException {
  /**
   * @internal
   */
  constructor(options) {
    super(options);
    Object.setPrototypeOf(this, _DataBrewServiceException.prototype);
  }
};
__name(_DataBrewServiceException, "DataBrewServiceException");
var DataBrewServiceException = _DataBrewServiceException;

// src/models/models_0.ts

var _AccessDeniedException = class _AccessDeniedException extends DataBrewServiceException {
  /**
   * @internal
   */
  constructor(opts) {
    super({
      name: "AccessDeniedException",
      $fault: "client",
      ...opts
    });
    this.name = "AccessDeniedException";
    this.$fault = "client";
    Object.setPrototypeOf(this, _AccessDeniedException.prototype);
    this.Message = opts.Message;
  }
};
__name(_AccessDeniedException, "AccessDeniedException");
var AccessDeniedException = _AccessDeniedException;
var AnalyticsMode = {
  DISABLE: "DISABLE",
  ENABLE: "ENABLE"
};
var _ConflictException = class _ConflictException extends DataBrewServiceException {
  /**
   * @internal
   */
  constructor(opts) {
    super({
      name: "ConflictException",
      $fault: "client",
      ...opts
    });
    this.name = "ConflictException";
    this.$fault = "client";
    Object.setPrototypeOf(this, _ConflictException.prototype);
    this.Message = opts.Message;
  }
};
__name(_ConflictException, "ConflictException");
var ConflictException = _ConflictException;
var _ResourceNotFoundException = class _ResourceNotFoundException extends DataBrewServiceException {
  /**
   * @internal
   */
  constructor(opts) {
    super({
      name: "ResourceNotFoundException",
      $fault: "client",
      ...opts
    });
    this.name = "ResourceNotFoundException";
    this.$fault = "client";
    Object.setPrototypeOf(this, _ResourceNotFoundException.prototype);
    this.Message = opts.Message;
  }
};
__name(_ResourceNotFoundException, "ResourceNotFoundException");
var ResourceNotFoundException = _ResourceNotFoundException;
var _ValidationException = class _ValidationException extends DataBrewServiceException {
  /**
   * @internal
   */
  constructor(opts) {
    super({
      name: "ValidationException",
      $fault: "client",
      ...opts
    });
    this.name = "ValidationException";
    this.$fault = "client";
    Object.setPrototypeOf(this, _ValidationException.prototype);
    this.Message = opts.Message;
  }
};
__name(_ValidationException, "ValidationException");
var ValidationException = _ValidationException;
var InputFormat = {
  CSV: "CSV",
  EXCEL: "EXCEL",
  JSON: "JSON",
  ORC: "ORC",
  PARQUET: "PARQUET"
};
var Order = {
  ASCENDING: "ASCENDING",
  DESCENDING: "DESCENDING"
};
var OrderedBy = {
  LAST_MODIFIED_DATE: "LAST_MODIFIED_DATE"
};
var ParameterType = {
  Datetime: "Datetime",
  Number: "Number",
  String: "String"
};
var _ServiceQuotaExceededException = class _ServiceQuotaExceededException extends DataBrewServiceException {
  /**
   * @internal
   */
  constructor(opts) {
    super({
      name: "ServiceQuotaExceededException",
      $fault: "client",
      ...opts
    });
    this.name = "ServiceQuotaExceededException";
    this.$fault = "client";
    Object.setPrototypeOf(this, _ServiceQuotaExceededException.prototype);
    this.Message = opts.Message;
  }
};
__name(_ServiceQuotaExceededException, "ServiceQuotaExceededException");
var ServiceQuotaExceededException = _ServiceQuotaExceededException;
var EncryptionMode = {
  SSEKMS: "SSE-KMS",
  SSES3: "SSE-S3"
};
var SampleMode = {
  CUSTOM_ROWS: "CUSTOM_ROWS",
  FULL_DATASET: "FULL_DATASET"
};
var LogSubscription = {
  DISABLE: "DISABLE",
  ENABLE: "ENABLE"
};
var ValidationMode = {
  CHECK_ALL: "CHECK_ALL"
};
var SampleType = {
  FIRST_N: "FIRST_N",
  LAST_N: "LAST_N",
  RANDOM: "RANDOM"
};
var _InternalServerException = class _InternalServerException extends DataBrewServiceException {
  /**
   * @internal
   */
  constructor(opts) {
    super({
      name: "InternalServerException",
      $fault: "server",
      ...opts
    });
    this.name = "InternalServerException";
    this.$fault = "server";
    Object.setPrototypeOf(this, _InternalServerException.prototype);
    this.Message = opts.Message;
  }
};
__name(_InternalServerException, "InternalServerException");
var InternalServerException = _InternalServerException;
var DatabaseOutputMode = {
  NEW_TABLE: "NEW_TABLE"
};
var CompressionFormat = {
  BROTLI: "BROTLI",
  BZIP2: "BZIP2",
  DEFLATE: "DEFLATE",
  GZIP: "GZIP",
  LZ4: "LZ4",
  LZO: "LZO",
  SNAPPY: "SNAPPY",
  ZLIB: "ZLIB",
  ZSTD: "ZSTD"
};
var OutputFormat = {
  AVRO: "AVRO",
  CSV: "CSV",
  GLUEPARQUET: "GLUEPARQUET",
  JSON: "JSON",
  ORC: "ORC",
  PARQUET: "PARQUET",
  TABLEAUHYPER: "TABLEAUHYPER",
  XML: "XML"
};
var ThresholdType = {
  GREATER_THAN: "GREATER_THAN",
  GREATER_THAN_OR_EQUAL: "GREATER_THAN_OR_EQUAL",
  LESS_THAN: "LESS_THAN",
  LESS_THAN_OR_EQUAL: "LESS_THAN_OR_EQUAL"
};
var ThresholdUnit = {
  COUNT: "COUNT",
  PERCENTAGE: "PERCENTAGE"
};
var Source = {
  DATABASE: "DATABASE",
  DATACATALOG: "DATA-CATALOG",
  S3: "S3"
};
var JobType = {
  PROFILE: "PROFILE",
  RECIPE: "RECIPE"
};
var JobRunState = {
  FAILED: "FAILED",
  RUNNING: "RUNNING",
  STARTING: "STARTING",
  STOPPED: "STOPPED",
  STOPPING: "STOPPING",
  SUCCEEDED: "SUCCEEDED",
  TIMEOUT: "TIMEOUT"
};
var SessionStatus = {
  ASSIGNED: "ASSIGNED",
  FAILED: "FAILED",
  INITIALIZING: "INITIALIZING",
  PROVISIONING: "PROVISIONING",
  READY: "READY",
  RECYCLING: "RECYCLING",
  ROTATING: "ROTATING",
  TERMINATED: "TERMINATED",
  TERMINATING: "TERMINATING",
  UPDATING: "UPDATING"
};
var SendProjectSessionActionRequestFilterSensitiveLog = /* @__PURE__ */ __name((obj) => ({
  ...obj,
  ...obj.ClientSessionId && { ClientSessionId: import_smithy_client.SENSITIVE_STRING }
}), "SendProjectSessionActionRequestFilterSensitiveLog");
var StartProjectSessionResponseFilterSensitiveLog = /* @__PURE__ */ __name((obj) => ({
  ...obj,
  ...obj.ClientSessionId && { ClientSessionId: import_smithy_client.SENSITIVE_STRING }
}), "StartProjectSessionResponseFilterSensitiveLog");

// src/protocols/Aws_restJson1.ts
var se_BatchDeleteRecipeVersionCommand = /* @__PURE__ */ __name(async (input, context) => {
  const b = (0, import_core.requestBuilder)(input, context);
  const headers = {
    "content-type": "application/json"
  };
  b.bp("/recipes/{Name}/batchDeleteRecipeVersion");
  b.p("Name", () => input.Name, "{Name}", false);
  let body;
  body = JSON.stringify(
    (0, import_smithy_client.take)(input, {
      RecipeVersions: (_) => (0, import_smithy_client._json)(_)
    })
  );
  b.m("POST").h(headers).b(body);
  return b.build();
}, "se_BatchDeleteRecipeVersionCommand");
var se_CreateDatasetCommand = /* @__PURE__ */ __name(async (input, context) => {
  const b = (0, import_core.requestBuilder)(input, context);
  const headers = {
    "content-type": "application/json"
  };
  b.bp("/datasets");
  let body;
  body = JSON.stringify(
    (0, import_smithy_client.take)(input, {
      Format: [],
      FormatOptions: (_) => (0, import_smithy_client._json)(_),
      Input: (_) => (0, import_smithy_client._json)(_),
      Name: [],
      PathOptions: (_) => (0, import_smithy_client._json)(_),
      Tags: (_) => (0, import_smithy_client._json)(_)
    })
  );
  b.m("POST").h(headers).b(body);
  return b.build();
}, "se_CreateDatasetCommand");
var se_CreateProfileJobCommand = /* @__PURE__ */ __name(async (input, context) => {
  const b = (0, import_core.requestBuilder)(input, context);
  const headers = {
    "content-type": "application/json"
  };
  b.bp("/profileJobs");
  let body;
  body = JSON.stringify(
    (0, import_smithy_client.take)(input, {
      Configuration: (_) => (0, import_smithy_client._json)(_),
      DatasetName: [],
      EncryptionKeyArn: [],
      EncryptionMode: [],
      JobSample: (_) => (0, import_smithy_client._json)(_),
      LogSubscription: [],
      MaxCapacity: [],
      MaxRetries: [],
      Name: [],
      OutputLocation: (_) => (0, import_smithy_client._json)(_),
      RoleArn: [],
      Tags: (_) => (0, import_smithy_client._json)(_),
      Timeout: [],
      ValidationConfigurations: (_) => (0, import_smithy_client._json)(_)
    })
  );
  b.m("POST").h(headers).b(body);
  return b.build();
}, "se_CreateProfileJobCommand");
var se_CreateProjectCommand = /* @__PURE__ */ __name(async (input, context) => {
  const b = (0, import_core.requestBuilder)(input, context);
  const headers = {
    "content-type": "application/json"
  };
  b.bp("/projects");
  let body;
  body = JSON.stringify(
    (0, import_smithy_client.take)(input, {
      DatasetName: [],
      Name: [],
      RecipeName: [],
      RoleArn: [],
      Sample: (_) => (0, import_smithy_client._json)(_),
      Tags: (_) => (0, import_smithy_client._json)(_)
    })
  );
  b.m("POST").h(headers).b(body);
  return b.build();
}, "se_CreateProjectCommand");
var se_CreateRecipeCommand = /* @__PURE__ */ __name(async (input, context) => {
  const b = (0, import_core.requestBuilder)(input, context);
  const headers = {
    "content-type": "application/json"
  };
  b.bp("/recipes");
  let body;
  body = JSON.stringify(
    (0, import_smithy_client.take)(input, {
      Description: [],
      Name: [],
      Steps: (_) => (0, import_smithy_client._json)(_),
      Tags: (_) => (0, import_smithy_client._json)(_)
    })
  );
  b.m("POST").h(headers).b(body);
  return b.build();
}, "se_CreateRecipeCommand");
var se_CreateRecipeJobCommand = /* @__PURE__ */ __name(async (input, context) => {
  const b = (0, import_core.requestBuilder)(input, context);
  const headers = {
    "content-type": "application/json"
  };
  b.bp("/recipeJobs");
  let body;
  body = JSON.stringify(
    (0, import_smithy_client.take)(input, {
      DataCatalogOutputs: (_) => (0, import_smithy_client._json)(_),
      DatabaseOutputs: (_) => (0, import_smithy_client._json)(_),
      DatasetName: [],
      EncryptionKeyArn: [],
      EncryptionMode: [],
      LogSubscription: [],
      MaxCapacity: [],
      MaxRetries: [],
      Name: [],
      Outputs: (_) => (0, import_smithy_client._json)(_),
      ProjectName: [],
      RecipeReference: (_) => (0, import_smithy_client._json)(_),
      RoleArn: [],
      Tags: (_) => (0, import_smithy_client._json)(_),
      Timeout: []
    })
  );
  b.m("POST").h(headers).b(body);
  return b.build();
}, "se_CreateRecipeJobCommand");
var se_CreateRulesetCommand = /* @__PURE__ */ __name(async (input, context) => {
  const b = (0, import_core.requestBuilder)(input, context);
  const headers = {
    "content-type": "application/json"
  };
  b.bp("/rulesets");
  let body;
  body = JSON.stringify(
    (0, import_smithy_client.take)(input, {
      Description: [],
      Name: [],
      Rules: (_) => se_RuleList(_, context),
      Tags: (_) => (0, import_smithy_client._json)(_),
      TargetArn: []
    })
  );
  b.m("POST").h(headers).b(body);
  return b.build();
}, "se_CreateRulesetCommand");
var se_CreateScheduleCommand = /* @__PURE__ */ __name(async (input, context) => {
  const b = (0, import_core.requestBuilder)(input, context);
  const headers = {
    "content-type": "application/json"
  };
  b.bp("/schedules");
  let body;
  body = JSON.stringify(
    (0, import_smithy_client.take)(input, {
      CronExpression: [],
      JobNames: (_) => (0, import_smithy_client._json)(_),
      Name: [],
      Tags: (_) => (0, import_smithy_client._json)(_)
    })
  );
  b.m("POST").h(headers).b(body);
  return b.build();
}, "se_CreateScheduleCommand");
var se_DeleteDatasetCommand = /* @__PURE__ */ __name(async (input, context) => {
  const b = (0, import_core.requestBuilder)(input, context);
  const headers = {};
  b.bp("/datasets/{Name}");
  b.p("Name", () => input.Name, "{Name}", false);
  let body;
  b.m("DELETE").h(headers).b(body);
  return b.build();
}, "se_DeleteDatasetCommand");
var se_DeleteJobCommand = /* @__PURE__ */ __name(async (input, context) => {
  const b = (0, import_core.requestBuilder)(input, context);
  const headers = {};
  b.bp("/jobs/{Name}");
  b.p("Name", () => input.Name, "{Name}", false);
  let body;
  b.m("DELETE").h(headers).b(body);
  return b.build();
}, "se_DeleteJobCommand");
var se_DeleteProjectCommand = /* @__PURE__ */ __name(async (input, context) => {
  const b = (0, import_core.requestBuilder)(input, context);
  const headers = {};
  b.bp("/projects/{Name}");
  b.p("Name", () => input.Name, "{Name}", false);
  let body;
  b.m("DELETE").h(headers).b(body);
  return b.build();
}, "se_DeleteProjectCommand");
var se_DeleteRecipeVersionCommand = /* @__PURE__ */ __name(async (input, context) => {
  const b = (0, import_core.requestBuilder)(input, context);
  const headers = {};
  b.bp("/recipes/{Name}/recipeVersion/{RecipeVersion}");
  b.p("Name", () => input.Name, "{Name}", false);
  b.p("RecipeVersion", () => input.RecipeVersion, "{RecipeVersion}", false);
  let body;
  b.m("DELETE").h(headers).b(body);
  return b.build();
}, "se_DeleteRecipeVersionCommand");
var se_DeleteRulesetCommand = /* @__PURE__ */ __name(async (input, context) => {
  const b = (0, import_core.requestBuilder)(input, context);
  const headers = {};
  b.bp("/rulesets/{Name}");
  b.p("Name", () => input.Name, "{Name}", false);
  let body;
  b.m("DELETE").h(headers).b(body);
  return b.build();
}, "se_DeleteRulesetCommand");
var se_DeleteScheduleCommand = /* @__PURE__ */ __name(async (input, context) => {
  const b = (0, import_core.requestBuilder)(input, context);
  const headers = {};
  b.bp("/schedules/{Name}");
  b.p("Name", () => input.Name, "{Name}", false);
  let body;
  b.m("DELETE").h(headers).b(body);
  return b.build();
}, "se_DeleteScheduleCommand");
var se_DescribeDatasetCommand = /* @__PURE__ */ __name(async (input, context) => {
  const b = (0, import_core.requestBuilder)(input, context);
  const headers = {};
  b.bp("/datasets/{Name}");
  b.p("Name", () => input.Name, "{Name}", false);
  let body;
  b.m("GET").h(headers).b(body);
  return b.build();
}, "se_DescribeDatasetCommand");
var se_DescribeJobCommand = /* @__PURE__ */ __name(async (input, context) => {
  const b = (0, import_core.requestBuilder)(input, context);
  const headers = {};
  b.bp("/jobs/{Name}");
  b.p("Name", () => input.Name, "{Name}", false);
  let body;
  b.m("GET").h(headers).b(body);
  return b.build();
}, "se_DescribeJobCommand");
var se_DescribeJobRunCommand = /* @__PURE__ */ __name(async (input, context) => {
  const b = (0, import_core.requestBuilder)(input, context);
  const headers = {};
  b.bp("/jobs/{Name}/jobRun/{RunId}");
  b.p("Name", () => input.Name, "{Name}", false);
  b.p("RunId", () => input.RunId, "{RunId}", false);
  let body;
  b.m("GET").h(headers).b(body);
  return b.build();
}, "se_DescribeJobRunCommand");
var se_DescribeProjectCommand = /* @__PURE__ */ __name(async (input, context) => {
  const b = (0, import_core.requestBuilder)(input, context);
  const headers = {};
  b.bp("/projects/{Name}");
  b.p("Name", () => input.Name, "{Name}", false);
  let body;
  b.m("GET").h(headers).b(body);
  return b.build();
}, "se_DescribeProjectCommand");
var se_DescribeRecipeCommand = /* @__PURE__ */ __name(async (input, context) => {
  const b = (0, import_core.requestBuilder)(input, context);
  const headers = {};
  b.bp("/recipes/{Name}");
  b.p("Name", () => input.Name, "{Name}", false);
  const query = (0, import_smithy_client.map)({
    [_rV]: [, input[_RV]]
  });
  let body;
  b.m("GET").h(headers).q(query).b(body);
  return b.build();
}, "se_DescribeRecipeCommand");
var se_DescribeRulesetCommand = /* @__PURE__ */ __name(async (input, context) => {
  const b = (0, import_core.requestBuilder)(input, context);
  const headers = {};
  b.bp("/rulesets/{Name}");
  b.p("Name", () => input.Name, "{Name}", false);
  let body;
  b.m("GET").h(headers).b(body);
  return b.build();
}, "se_DescribeRulesetCommand");
var se_DescribeScheduleCommand = /* @__PURE__ */ __name(async (input, context) => {
  const b = (0, import_core.requestBuilder)(input, context);
  const headers = {};
  b.bp("/schedules/{Name}");
  b.p("Name", () => input.Name, "{Name}", false);
  let body;
  b.m("GET").h(headers).b(body);
  return b.build();
}, "se_DescribeScheduleCommand");
var se_ListDatasetsCommand = /* @__PURE__ */ __name(async (input, context) => {
  const b = (0, import_core.requestBuilder)(input, context);
  const headers = {};
  b.bp("/datasets");
  const query = (0, import_smithy_client.map)({
    [_mR]: [() => input.MaxResults !== void 0, () => input[_MR].toString()],
    [_nT]: [, input[_NT]]
  });
  let body;
  b.m("GET").h(headers).q(query).b(body);
  return b.build();
}, "se_ListDatasetsCommand");
var se_ListJobRunsCommand = /* @__PURE__ */ __name(async (input, context) => {
  const b = (0, import_core.requestBuilder)(input, context);
  const headers = {};
  b.bp("/jobs/{Name}/jobRuns");
  b.p("Name", () => input.Name, "{Name}", false);
  const query = (0, import_smithy_client.map)({
    [_mR]: [() => input.MaxResults !== void 0, () => input[_MR].toString()],
    [_nT]: [, input[_NT]]
  });
  let body;
  b.m("GET").h(headers).q(query).b(body);
  return b.build();
}, "se_ListJobRunsCommand");
var se_ListJobsCommand = /* @__PURE__ */ __name(async (input, context) => {
  const b = (0, import_core.requestBuilder)(input, context);
  const headers = {};
  b.bp("/jobs");
  const query = (0, import_smithy_client.map)({
    [_dN]: [, input[_DN]],
    [_mR]: [() => input.MaxResults !== void 0, () => input[_MR].toString()],
    [_nT]: [, input[_NT]],
    [_pN]: [, input[_PN]]
  });
  let body;
  b.m("GET").h(headers).q(query).b(body);
  return b.build();
}, "se_ListJobsCommand");
var se_ListProjectsCommand = /* @__PURE__ */ __name(async (input, context) => {
  const b = (0, import_core.requestBuilder)(input, context);
  const headers = {};
  b.bp("/projects");
  const query = (0, import_smithy_client.map)({
    [_nT]: [, input[_NT]],
    [_mR]: [() => input.MaxResults !== void 0, () => input[_MR].toString()]
  });
  let body;
  b.m("GET").h(headers).q(query).b(body);
  return b.build();
}, "se_ListProjectsCommand");
var se_ListRecipesCommand = /* @__PURE__ */ __name(async (input, context) => {
  const b = (0, import_core.requestBuilder)(input, context);
  const headers = {};
  b.bp("/recipes");
  const query = (0, import_smithy_client.map)({
    [_mR]: [() => input.MaxResults !== void 0, () => input[_MR].toString()],
    [_nT]: [, input[_NT]],
    [_rV]: [, input[_RV]]
  });
  let body;
  b.m("GET").h(headers).q(query).b(body);
  return b.build();
}, "se_ListRecipesCommand");
var se_ListRecipeVersionsCommand = /* @__PURE__ */ __name(async (input, context) => {
  const b = (0, import_core.requestBuilder)(input, context);
  const headers = {};
  b.bp("/recipeVersions");
  const query = (0, import_smithy_client.map)({
    [_mR]: [() => input.MaxResults !== void 0, () => input[_MR].toString()],
    [_nT]: [, input[_NT]],
    [_n]: [, (0, import_smithy_client.expectNonNull)(input[_N], `Name`)]
  });
  let body;
  b.m("GET").h(headers).q(query).b(body);
  return b.build();
}, "se_ListRecipeVersionsCommand");
var se_ListRulesetsCommand = /* @__PURE__ */ __name(async (input, context) => {
  const b = (0, import_core.requestBuilder)(input, context);
  const headers = {};
  b.bp("/rulesets");
  const query = (0, import_smithy_client.map)({
    [_tA]: [, input[_TA]],
    [_mR]: [() => input.MaxResults !== void 0, () => input[_MR].toString()],
    [_nT]: [, input[_NT]]
  });
  let body;
  b.m("GET").h(headers).q(query).b(body);
  return b.build();
}, "se_ListRulesetsCommand");
var se_ListSchedulesCommand = /* @__PURE__ */ __name(async (input, context) => {
  const b = (0, import_core.requestBuilder)(input, context);
  const headers = {};
  b.bp("/schedules");
  const query = (0, import_smithy_client.map)({
    [_jN]: [, input[_JN]],
    [_mR]: [() => input.MaxResults !== void 0, () => input[_MR].toString()],
    [_nT]: [, input[_NT]]
  });
  let body;
  b.m("GET").h(headers).q(query).b(body);
  return b.build();
}, "se_ListSchedulesCommand");
var se_ListTagsForResourceCommand = /* @__PURE__ */ __name(async (input, context) => {
  const b = (0, import_core.requestBuilder)(input, context);
  const headers = {};
  b.bp("/tags/{ResourceArn}");
  b.p("ResourceArn", () => input.ResourceArn, "{ResourceArn}", false);
  let body;
  b.m("GET").h(headers).b(body);
  return b.build();
}, "se_ListTagsForResourceCommand");
var se_PublishRecipeCommand = /* @__PURE__ */ __name(async (input, context) => {
  const b = (0, import_core.requestBuilder)(input, context);
  const headers = {
    "content-type": "application/json"
  };
  b.bp("/recipes/{Name}/publishRecipe");
  b.p("Name", () => input.Name, "{Name}", false);
  let body;
  body = JSON.stringify(
    (0, import_smithy_client.take)(input, {
      Description: []
    })
  );
  b.m("POST").h(headers).b(body);
  return b.build();
}, "se_PublishRecipeCommand");
var se_SendProjectSessionActionCommand = /* @__PURE__ */ __name(async (input, context) => {
  const b = (0, import_core.requestBuilder)(input, context);
  const headers = {
    "content-type": "application/json"
  };
  b.bp("/projects/{Name}/sendProjectSessionAction");
  b.p("Name", () => input.Name, "{Name}", false);
  let body;
  body = JSON.stringify(
    (0, import_smithy_client.take)(input, {
      ClientSessionId: [],
      Preview: [],
      RecipeStep: (_) => (0, import_smithy_client._json)(_),
      StepIndex: [],
      ViewFrame: (_) => (0, import_smithy_client._json)(_)
    })
  );
  b.m("PUT").h(headers).b(body);
  return b.build();
}, "se_SendProjectSessionActionCommand");
var se_StartJobRunCommand = /* @__PURE__ */ __name(async (input, context) => {
  const b = (0, import_core.requestBuilder)(input, context);
  const headers = {};
  b.bp("/jobs/{Name}/startJobRun");
  b.p("Name", () => input.Name, "{Name}", false);
  let body;
  b.m("POST").h(headers).b(body);
  return b.build();
}, "se_StartJobRunCommand");
var se_StartProjectSessionCommand = /* @__PURE__ */ __name(async (input, context) => {
  const b = (0, import_core.requestBuilder)(input, context);
  const headers = {
    "content-type": "application/json"
  };
  b.bp("/projects/{Name}/startProjectSession");
  b.p("Name", () => input.Name, "{Name}", false);
  let body;
  body = JSON.stringify(
    (0, import_smithy_client.take)(input, {
      AssumeControl: []
    })
  );
  b.m("PUT").h(headers).b(body);
  return b.build();
}, "se_StartProjectSessionCommand");
var se_StopJobRunCommand = /* @__PURE__ */ __name(async (input, context) => {
  const b = (0, import_core.requestBuilder)(input, context);
  const headers = {};
  b.bp("/jobs/{Name}/jobRun/{RunId}/stopJobRun");
  b.p("Name", () => input.Name, "{Name}", false);
  b.p("RunId", () => input.RunId, "{RunId}", false);
  let body;
  b.m("POST").h(headers).b(body);
  return b.build();
}, "se_StopJobRunCommand");
var se_TagResourceCommand = /* @__PURE__ */ __name(async (input, context) => {
  const b = (0, import_core.requestBuilder)(input, context);
  const headers = {
    "content-type": "application/json"
  };
  b.bp("/tags/{ResourceArn}");
  b.p("ResourceArn", () => input.ResourceArn, "{ResourceArn}", false);
  let body;
  body = JSON.stringify(
    (0, import_smithy_client.take)(input, {
      Tags: (_) => (0, import_smithy_client._json)(_)
    })
  );
  b.m("POST").h(headers).b(body);
  return b.build();
}, "se_TagResourceCommand");
var se_UntagResourceCommand = /* @__PURE__ */ __name(async (input, context) => {
  const b = (0, import_core.requestBuilder)(input, context);
  const headers = {};
  b.bp("/tags/{ResourceArn}");
  b.p("ResourceArn", () => input.ResourceArn, "{ResourceArn}", false);
  const query = (0, import_smithy_client.map)({
    [_tK]: [
      (0, import_smithy_client.expectNonNull)(input.TagKeys, `TagKeys`) != null,
      () => (input[_TK] || []).map((_entry) => _entry)
    ]
  });
  let body;
  b.m("DELETE").h(headers).q(query).b(body);
  return b.build();
}, "se_UntagResourceCommand");
var se_UpdateDatasetCommand = /* @__PURE__ */ __name(async (input, context) => {
  const b = (0, import_core.requestBuilder)(input, context);
  const headers = {
    "content-type": "application/json"
  };
  b.bp("/datasets/{Name}");
  b.p("Name", () => input.Name, "{Name}", false);
  let body;
  body = JSON.stringify(
    (0, import_smithy_client.take)(input, {
      Format: [],
      FormatOptions: (_) => (0, import_smithy_client._json)(_),
      Input: (_) => (0, import_smithy_client._json)(_),
      PathOptions: (_) => (0, import_smithy_client._json)(_)
    })
  );
  b.m("PUT").h(headers).b(body);
  return b.build();
}, "se_UpdateDatasetCommand");
var se_UpdateProfileJobCommand = /* @__PURE__ */ __name(async (input, context) => {
  const b = (0, import_core.requestBuilder)(input, context);
  const headers = {
    "content-type": "application/json"
  };
  b.bp("/profileJobs/{Name}");
  b.p("Name", () => input.Name, "{Name}", false);
  let body;
  body = JSON.stringify(
    (0, import_smithy_client.take)(input, {
      Configuration: (_) => (0, import_smithy_client._json)(_),
      EncryptionKeyArn: [],
      EncryptionMode: [],
      JobSample: (_) => (0, import_smithy_client._json)(_),
      LogSubscription: [],
      MaxCapacity: [],
      MaxRetries: [],
      OutputLocation: (_) => (0, import_smithy_client._json)(_),
      RoleArn: [],
      Timeout: [],
      ValidationConfigurations: (_) => (0, import_smithy_client._json)(_)
    })
  );
  b.m("PUT").h(headers).b(body);
  return b.build();
}, "se_UpdateProfileJobCommand");
var se_UpdateProjectCommand = /* @__PURE__ */ __name(async (input, context) => {
  const b = (0, import_core.requestBuilder)(input, context);
  const headers = {
    "content-type": "application/json"
  };
  b.bp("/projects/{Name}");
  b.p("Name", () => input.Name, "{Name}", false);
  let body;
  body = JSON.stringify(
    (0, import_smithy_client.take)(input, {
      RoleArn: [],
      Sample: (_) => (0, import_smithy_client._json)(_)
    })
  );
  b.m("PUT").h(headers).b(body);
  return b.build();
}, "se_UpdateProjectCommand");
var se_UpdateRecipeCommand = /* @__PURE__ */ __name(async (input, context) => {
  const b = (0, import_core.requestBuilder)(input, context);
  const headers = {
    "content-type": "application/json"
  };
  b.bp("/recipes/{Name}");
  b.p("Name", () => input.Name, "{Name}", false);
  let body;
  body = JSON.stringify(
    (0, import_smithy_client.take)(input, {
      Description: [],
      Steps: (_) => (0, import_smithy_client._json)(_)
    })
  );
  b.m("PUT").h(headers).b(body);
  return b.build();
}, "se_UpdateRecipeCommand");
var se_UpdateRecipeJobCommand = /* @__PURE__ */ __name(async (input, context) => {
  const b = (0, import_core.requestBuilder)(input, context);
  const headers = {
    "content-type": "application/json"
  };
  b.bp("/recipeJobs/{Name}");
  b.p("Name", () => input.Name, "{Name}", false);
  let body;
  body = JSON.stringify(
    (0, import_smithy_client.take)(input, {
      DataCatalogOutputs: (_) => (0, import_smithy_client._json)(_),
      DatabaseOutputs: (_) => (0, import_smithy_client._json)(_),
      EncryptionKeyArn: [],
      EncryptionMode: [],
      LogSubscription: [],
      MaxCapacity: [],
      MaxRetries: [],
      Outputs: (_) => (0, import_smithy_client._json)(_),
      RoleArn: [],
      Timeout: []
    })
  );
  b.m("PUT").h(headers).b(body);
  return b.build();
}, "se_UpdateRecipeJobCommand");
var se_UpdateRulesetCommand = /* @__PURE__ */ __name(async (input, context) => {
  const b = (0, import_core.requestBuilder)(input, context);
  const headers = {
    "content-type": "application/json"
  };
  b.bp("/rulesets/{Name}");
  b.p("Name", () => input.Name, "{Name}", false);
  let body;
  body = JSON.stringify(
    (0, import_smithy_client.take)(input, {
      Description: [],
      Rules: (_) => se_RuleList(_, context)
    })
  );
  b.m("PUT").h(headers).b(body);
  return b.build();
}, "se_UpdateRulesetCommand");
var se_UpdateScheduleCommand = /* @__PURE__ */ __name(async (input, context) => {
  const b = (0, import_core.requestBuilder)(input, context);
  const headers = {
    "content-type": "application/json"
  };
  b.bp("/schedules/{Name}");
  b.p("Name", () => input.Name, "{Name}", false);
  let body;
  body = JSON.stringify(
    (0, import_smithy_client.take)(input, {
      CronExpression: [],
      JobNames: (_) => (0, import_smithy_client._json)(_)
    })
  );
  b.m("PUT").h(headers).b(body);
  return b.build();
}, "se_UpdateScheduleCommand");
var de_BatchDeleteRecipeVersionCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode !== 200 && output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const contents = (0, import_smithy_client.map)({
    $metadata: deserializeMetadata(output)
  });
  const data = (0, import_smithy_client.expectNonNull)((0, import_smithy_client.expectObject)(await (0, import_core2.parseJsonBody)(output.body, context)), "body");
  const doc = (0, import_smithy_client.take)(data, {
    Errors: import_smithy_client._json,
    Name: import_smithy_client.expectString
  });
  Object.assign(contents, doc);
  return contents;
}, "de_BatchDeleteRecipeVersionCommand");
var de_CreateDatasetCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode !== 200 && output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const contents = (0, import_smithy_client.map)({
    $metadata: deserializeMetadata(output)
  });
  const data = (0, import_smithy_client.expectNonNull)((0, import_smithy_client.expectObject)(await (0, import_core2.parseJsonBody)(output.body, context)), "body");
  const doc = (0, import_smithy_client.take)(data, {
    Name: import_smithy_client.expectString
  });
  Object.assign(contents, doc);
  return contents;
}, "de_CreateDatasetCommand");
var de_CreateProfileJobCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode !== 200 && output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const contents = (0, import_smithy_client.map)({
    $metadata: deserializeMetadata(output)
  });
  const data = (0, import_smithy_client.expectNonNull)((0, import_smithy_client.expectObject)(await (0, import_core2.parseJsonBody)(output.body, context)), "body");
  const doc = (0, import_smithy_client.take)(data, {
    Name: import_smithy_client.expectString
  });
  Object.assign(contents, doc);
  return contents;
}, "de_CreateProfileJobCommand");
var de_CreateProjectCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode !== 200 && output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const contents = (0, import_smithy_client.map)({
    $metadata: deserializeMetadata(output)
  });
  const data = (0, import_smithy_client.expectNonNull)((0, import_smithy_client.expectObject)(await (0, import_core2.parseJsonBody)(output.body, context)), "body");
  const doc = (0, import_smithy_client.take)(data, {
    Name: import_smithy_client.expectString
  });
  Object.assign(contents, doc);
  return contents;
}, "de_CreateProjectCommand");
var de_CreateRecipeCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode !== 200 && output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const contents = (0, import_smithy_client.map)({
    $metadata: deserializeMetadata(output)
  });
  const data = (0, import_smithy_client.expectNonNull)((0, import_smithy_client.expectObject)(await (0, import_core2.parseJsonBody)(output.body, context)), "body");
  const doc = (0, import_smithy_client.take)(data, {
    Name: import_smithy_client.expectString
  });
  Object.assign(contents, doc);
  return contents;
}, "de_CreateRecipeCommand");
var de_CreateRecipeJobCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode !== 200 && output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const contents = (0, import_smithy_client.map)({
    $metadata: deserializeMetadata(output)
  });
  const data = (0, import_smithy_client.expectNonNull)((0, import_smithy_client.expectObject)(await (0, import_core2.parseJsonBody)(output.body, context)), "body");
  const doc = (0, import_smithy_client.take)(data, {
    Name: import_smithy_client.expectString
  });
  Object.assign(contents, doc);
  return contents;
}, "de_CreateRecipeJobCommand");
var de_CreateRulesetCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode !== 200 && output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const contents = (0, import_smithy_client.map)({
    $metadata: deserializeMetadata(output)
  });
  const data = (0, import_smithy_client.expectNonNull)((0, import_smithy_client.expectObject)(await (0, import_core2.parseJsonBody)(output.body, context)), "body");
  const doc = (0, import_smithy_client.take)(data, {
    Name: import_smithy_client.expectString
  });
  Object.assign(contents, doc);
  return contents;
}, "de_CreateRulesetCommand");
var de_CreateScheduleCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode !== 200 && output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const contents = (0, import_smithy_client.map)({
    $metadata: deserializeMetadata(output)
  });
  const data = (0, import_smithy_client.expectNonNull)((0, import_smithy_client.expectObject)(await (0, import_core2.parseJsonBody)(output.body, context)), "body");
  const doc = (0, import_smithy_client.take)(data, {
    Name: import_smithy_client.expectString
  });
  Object.assign(contents, doc);
  return contents;
}, "de_CreateScheduleCommand");
var de_DeleteDatasetCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode !== 200 && output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const contents = (0, import_smithy_client.map)({
    $metadata: deserializeMetadata(output)
  });
  const data = (0, import_smithy_client.expectNonNull)((0, import_smithy_client.expectObject)(await (0, import_core2.parseJsonBody)(output.body, context)), "body");
  const doc = (0, import_smithy_client.take)(data, {
    Name: import_smithy_client.expectString
  });
  Object.assign(contents, doc);
  return contents;
}, "de_DeleteDatasetCommand");
var de_DeleteJobCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode !== 200 && output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const contents = (0, import_smithy_client.map)({
    $metadata: deserializeMetadata(output)
  });
  const data = (0, import_smithy_client.expectNonNull)((0, import_smithy_client.expectObject)(await (0, import_core2.parseJsonBody)(output.body, context)), "body");
  const doc = (0, import_smithy_client.take)(data, {
    Name: import_smithy_client.expectString
  });
  Object.assign(contents, doc);
  return contents;
}, "de_DeleteJobCommand");
var de_DeleteProjectCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode !== 200 && output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const contents = (0, import_smithy_client.map)({
    $metadata: deserializeMetadata(output)
  });
  const data = (0, import_smithy_client.expectNonNull)((0, import_smithy_client.expectObject)(await (0, import_core2.parseJsonBody)(output.body, context)), "body");
  const doc = (0, import_smithy_client.take)(data, {
    Name: import_smithy_client.expectString
  });
  Object.assign(contents, doc);
  return contents;
}, "de_DeleteProjectCommand");
var de_DeleteRecipeVersionCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode !== 200 && output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const contents = (0, import_smithy_client.map)({
    $metadata: deserializeMetadata(output)
  });
  const data = (0, import_smithy_client.expectNonNull)((0, import_smithy_client.expectObject)(await (0, import_core2.parseJsonBody)(output.body, context)), "body");
  const doc = (0, import_smithy_client.take)(data, {
    Name: import_smithy_client.expectString,
    RecipeVersion: import_smithy_client.expectString
  });
  Object.assign(contents, doc);
  return contents;
}, "de_DeleteRecipeVersionCommand");
var de_DeleteRulesetCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode !== 200 && output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const contents = (0, import_smithy_client.map)({
    $metadata: deserializeMetadata(output)
  });
  const data = (0, import_smithy_client.expectNonNull)((0, import_smithy_client.expectObject)(await (0, import_core2.parseJsonBody)(output.body, context)), "body");
  const doc = (0, import_smithy_client.take)(data, {
    Name: import_smithy_client.expectString
  });
  Object.assign(contents, doc);
  return contents;
}, "de_DeleteRulesetCommand");
var de_DeleteScheduleCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode !== 200 && output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const contents = (0, import_smithy_client.map)({
    $metadata: deserializeMetadata(output)
  });
  const data = (0, import_smithy_client.expectNonNull)((0, import_smithy_client.expectObject)(await (0, import_core2.parseJsonBody)(output.body, context)), "body");
  const doc = (0, import_smithy_client.take)(data, {
    Name: import_smithy_client.expectString
  });
  Object.assign(contents, doc);
  return contents;
}, "de_DeleteScheduleCommand");
var de_DescribeDatasetCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode !== 200 && output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const contents = (0, import_smithy_client.map)({
    $metadata: deserializeMetadata(output)
  });
  const data = (0, import_smithy_client.expectNonNull)((0, import_smithy_client.expectObject)(await (0, import_core2.parseJsonBody)(output.body, context)), "body");
  const doc = (0, import_smithy_client.take)(data, {
    CreateDate: (_) => (0, import_smithy_client.expectNonNull)((0, import_smithy_client.parseEpochTimestamp)((0, import_smithy_client.expectNumber)(_))),
    CreatedBy: import_smithy_client.expectString,
    Format: import_smithy_client.expectString,
    FormatOptions: import_smithy_client._json,
    Input: import_smithy_client._json,
    LastModifiedBy: import_smithy_client.expectString,
    LastModifiedDate: (_) => (0, import_smithy_client.expectNonNull)((0, import_smithy_client.parseEpochTimestamp)((0, import_smithy_client.expectNumber)(_))),
    Name: import_smithy_client.expectString,
    PathOptions: import_smithy_client._json,
    ResourceArn: import_smithy_client.expectString,
    Source: import_smithy_client.expectString,
    Tags: import_smithy_client._json
  });
  Object.assign(contents, doc);
  return contents;
}, "de_DescribeDatasetCommand");
var de_DescribeJobCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode !== 200 && output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const contents = (0, import_smithy_client.map)({
    $metadata: deserializeMetadata(output)
  });
  const data = (0, import_smithy_client.expectNonNull)((0, import_smithy_client.expectObject)(await (0, import_core2.parseJsonBody)(output.body, context)), "body");
  const doc = (0, import_smithy_client.take)(data, {
    CreateDate: (_) => (0, import_smithy_client.expectNonNull)((0, import_smithy_client.parseEpochTimestamp)((0, import_smithy_client.expectNumber)(_))),
    CreatedBy: import_smithy_client.expectString,
    DataCatalogOutputs: import_smithy_client._json,
    DatabaseOutputs: import_smithy_client._json,
    DatasetName: import_smithy_client.expectString,
    EncryptionKeyArn: import_smithy_client.expectString,
    EncryptionMode: import_smithy_client.expectString,
    JobSample: import_smithy_client._json,
    LastModifiedBy: import_smithy_client.expectString,
    LastModifiedDate: (_) => (0, import_smithy_client.expectNonNull)((0, import_smithy_client.parseEpochTimestamp)((0, import_smithy_client.expectNumber)(_))),
    LogSubscription: import_smithy_client.expectString,
    MaxCapacity: import_smithy_client.expectInt32,
    MaxRetries: import_smithy_client.expectInt32,
    Name: import_smithy_client.expectString,
    Outputs: import_smithy_client._json,
    ProfileConfiguration: import_smithy_client._json,
    ProjectName: import_smithy_client.expectString,
    RecipeReference: import_smithy_client._json,
    ResourceArn: import_smithy_client.expectString,
    RoleArn: import_smithy_client.expectString,
    Tags: import_smithy_client._json,
    Timeout: import_smithy_client.expectInt32,
    Type: import_smithy_client.expectString,
    ValidationConfigurations: import_smithy_client._json
  });
  Object.assign(contents, doc);
  return contents;
}, "de_DescribeJobCommand");
var de_DescribeJobRunCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode !== 200 && output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const contents = (0, import_smithy_client.map)({
    $metadata: deserializeMetadata(output)
  });
  const data = (0, import_smithy_client.expectNonNull)((0, import_smithy_client.expectObject)(await (0, import_core2.parseJsonBody)(output.body, context)), "body");
  const doc = (0, import_smithy_client.take)(data, {
    Attempt: import_smithy_client.expectInt32,
    CompletedOn: (_) => (0, import_smithy_client.expectNonNull)((0, import_smithy_client.parseEpochTimestamp)((0, import_smithy_client.expectNumber)(_))),
    DataCatalogOutputs: import_smithy_client._json,
    DatabaseOutputs: import_smithy_client._json,
    DatasetName: import_smithy_client.expectString,
    ErrorMessage: import_smithy_client.expectString,
    ExecutionTime: import_smithy_client.expectInt32,
    JobName: import_smithy_client.expectString,
    JobSample: import_smithy_client._json,
    LogGroupName: import_smithy_client.expectString,
    LogSubscription: import_smithy_client.expectString,
    Outputs: import_smithy_client._json,
    ProfileConfiguration: import_smithy_client._json,
    RecipeReference: import_smithy_client._json,
    RunId: import_smithy_client.expectString,
    StartedBy: import_smithy_client.expectString,
    StartedOn: (_) => (0, import_smithy_client.expectNonNull)((0, import_smithy_client.parseEpochTimestamp)((0, import_smithy_client.expectNumber)(_))),
    State: import_smithy_client.expectString,
    ValidationConfigurations: import_smithy_client._json
  });
  Object.assign(contents, doc);
  return contents;
}, "de_DescribeJobRunCommand");
var de_DescribeProjectCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode !== 200 && output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const contents = (0, import_smithy_client.map)({
    $metadata: deserializeMetadata(output)
  });
  const data = (0, import_smithy_client.expectNonNull)((0, import_smithy_client.expectObject)(await (0, import_core2.parseJsonBody)(output.body, context)), "body");
  const doc = (0, import_smithy_client.take)(data, {
    CreateDate: (_) => (0, import_smithy_client.expectNonNull)((0, import_smithy_client.parseEpochTimestamp)((0, import_smithy_client.expectNumber)(_))),
    CreatedBy: import_smithy_client.expectString,
    DatasetName: import_smithy_client.expectString,
    LastModifiedBy: import_smithy_client.expectString,
    LastModifiedDate: (_) => (0, import_smithy_client.expectNonNull)((0, import_smithy_client.parseEpochTimestamp)((0, import_smithy_client.expectNumber)(_))),
    Name: import_smithy_client.expectString,
    OpenDate: (_) => (0, import_smithy_client.expectNonNull)((0, import_smithy_client.parseEpochTimestamp)((0, import_smithy_client.expectNumber)(_))),
    OpenedBy: import_smithy_client.expectString,
    RecipeName: import_smithy_client.expectString,
    ResourceArn: import_smithy_client.expectString,
    RoleArn: import_smithy_client.expectString,
    Sample: import_smithy_client._json,
    SessionStatus: import_smithy_client.expectString,
    Tags: import_smithy_client._json
  });
  Object.assign(contents, doc);
  return contents;
}, "de_DescribeProjectCommand");
var de_DescribeRecipeCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode !== 200 && output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const contents = (0, import_smithy_client.map)({
    $metadata: deserializeMetadata(output)
  });
  const data = (0, import_smithy_client.expectNonNull)((0, import_smithy_client.expectObject)(await (0, import_core2.parseJsonBody)(output.body, context)), "body");
  const doc = (0, import_smithy_client.take)(data, {
    CreateDate: (_) => (0, import_smithy_client.expectNonNull)((0, import_smithy_client.parseEpochTimestamp)((0, import_smithy_client.expectNumber)(_))),
    CreatedBy: import_smithy_client.expectString,
    Description: import_smithy_client.expectString,
    LastModifiedBy: import_smithy_client.expectString,
    LastModifiedDate: (_) => (0, import_smithy_client.expectNonNull)((0, import_smithy_client.parseEpochTimestamp)((0, import_smithy_client.expectNumber)(_))),
    Name: import_smithy_client.expectString,
    ProjectName: import_smithy_client.expectString,
    PublishedBy: import_smithy_client.expectString,
    PublishedDate: (_) => (0, import_smithy_client.expectNonNull)((0, import_smithy_client.parseEpochTimestamp)((0, import_smithy_client.expectNumber)(_))),
    RecipeVersion: import_smithy_client.expectString,
    ResourceArn: import_smithy_client.expectString,
    Steps: import_smithy_client._json,
    Tags: import_smithy_client._json
  });
  Object.assign(contents, doc);
  return contents;
}, "de_DescribeRecipeCommand");
var de_DescribeRulesetCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode !== 200 && output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const contents = (0, import_smithy_client.map)({
    $metadata: deserializeMetadata(output)
  });
  const data = (0, import_smithy_client.expectNonNull)((0, import_smithy_client.expectObject)(await (0, import_core2.parseJsonBody)(output.body, context)), "body");
  const doc = (0, import_smithy_client.take)(data, {
    CreateDate: (_) => (0, import_smithy_client.expectNonNull)((0, import_smithy_client.parseEpochTimestamp)((0, import_smithy_client.expectNumber)(_))),
    CreatedBy: import_smithy_client.expectString,
    Description: import_smithy_client.expectString,
    LastModifiedBy: import_smithy_client.expectString,
    LastModifiedDate: (_) => (0, import_smithy_client.expectNonNull)((0, import_smithy_client.parseEpochTimestamp)((0, import_smithy_client.expectNumber)(_))),
    Name: import_smithy_client.expectString,
    ResourceArn: import_smithy_client.expectString,
    Rules: (_) => de_RuleList(_, context),
    Tags: import_smithy_client._json,
    TargetArn: import_smithy_client.expectString
  });
  Object.assign(contents, doc);
  return contents;
}, "de_DescribeRulesetCommand");
var de_DescribeScheduleCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode !== 200 && output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const contents = (0, import_smithy_client.map)({
    $metadata: deserializeMetadata(output)
  });
  const data = (0, import_smithy_client.expectNonNull)((0, import_smithy_client.expectObject)(await (0, import_core2.parseJsonBody)(output.body, context)), "body");
  const doc = (0, import_smithy_client.take)(data, {
    CreateDate: (_) => (0, import_smithy_client.expectNonNull)((0, import_smithy_client.parseEpochTimestamp)((0, import_smithy_client.expectNumber)(_))),
    CreatedBy: import_smithy_client.expectString,
    CronExpression: import_smithy_client.expectString,
    JobNames: import_smithy_client._json,
    LastModifiedBy: import_smithy_client.expectString,
    LastModifiedDate: (_) => (0, import_smithy_client.expectNonNull)((0, import_smithy_client.parseEpochTimestamp)((0, import_smithy_client.expectNumber)(_))),
    Name: import_smithy_client.expectString,
    ResourceArn: import_smithy_client.expectString,
    Tags: import_smithy_client._json
  });
  Object.assign(contents, doc);
  return contents;
}, "de_DescribeScheduleCommand");
var de_ListDatasetsCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode !== 200 && output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const contents = (0, import_smithy_client.map)({
    $metadata: deserializeMetadata(output)
  });
  const data = (0, import_smithy_client.expectNonNull)((0, import_smithy_client.expectObject)(await (0, import_core2.parseJsonBody)(output.body, context)), "body");
  const doc = (0, import_smithy_client.take)(data, {
    Datasets: (_) => de_DatasetList(_, context),
    NextToken: import_smithy_client.expectString
  });
  Object.assign(contents, doc);
  return contents;
}, "de_ListDatasetsCommand");
var de_ListJobRunsCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode !== 200 && output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const contents = (0, import_smithy_client.map)({
    $metadata: deserializeMetadata(output)
  });
  const data = (0, import_smithy_client.expectNonNull)((0, import_smithy_client.expectObject)(await (0, import_core2.parseJsonBody)(output.body, context)), "body");
  const doc = (0, import_smithy_client.take)(data, {
    JobRuns: (_) => de_JobRunList(_, context),
    NextToken: import_smithy_client.expectString
  });
  Object.assign(contents, doc);
  return contents;
}, "de_ListJobRunsCommand");
var de_ListJobsCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode !== 200 && output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const contents = (0, import_smithy_client.map)({
    $metadata: deserializeMetadata(output)
  });
  const data = (0, import_smithy_client.expectNonNull)((0, import_smithy_client.expectObject)(await (0, import_core2.parseJsonBody)(output.body, context)), "body");
  const doc = (0, import_smithy_client.take)(data, {
    Jobs: (_) => de_JobList(_, context),
    NextToken: import_smithy_client.expectString
  });
  Object.assign(contents, doc);
  return contents;
}, "de_ListJobsCommand");
var de_ListProjectsCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode !== 200 && output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const contents = (0, import_smithy_client.map)({
    $metadata: deserializeMetadata(output)
  });
  const data = (0, import_smithy_client.expectNonNull)((0, import_smithy_client.expectObject)(await (0, import_core2.parseJsonBody)(output.body, context)), "body");
  const doc = (0, import_smithy_client.take)(data, {
    NextToken: import_smithy_client.expectString,
    Projects: (_) => de_ProjectList(_, context)
  });
  Object.assign(contents, doc);
  return contents;
}, "de_ListProjectsCommand");
var de_ListRecipesCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode !== 200 && output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const contents = (0, import_smithy_client.map)({
    $metadata: deserializeMetadata(output)
  });
  const data = (0, import_smithy_client.expectNonNull)((0, import_smithy_client.expectObject)(await (0, import_core2.parseJsonBody)(output.body, context)), "body");
  const doc = (0, import_smithy_client.take)(data, {
    NextToken: import_smithy_client.expectString,
    Recipes: (_) => de_RecipeList(_, context)
  });
  Object.assign(contents, doc);
  return contents;
}, "de_ListRecipesCommand");
var de_ListRecipeVersionsCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode !== 200 && output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const contents = (0, import_smithy_client.map)({
    $metadata: deserializeMetadata(output)
  });
  const data = (0, import_smithy_client.expectNonNull)((0, import_smithy_client.expectObject)(await (0, import_core2.parseJsonBody)(output.body, context)), "body");
  const doc = (0, import_smithy_client.take)(data, {
    NextToken: import_smithy_client.expectString,
    Recipes: (_) => de_RecipeList(_, context)
  });
  Object.assign(contents, doc);
  return contents;
}, "de_ListRecipeVersionsCommand");
var de_ListRulesetsCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode !== 200 && output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const contents = (0, import_smithy_client.map)({
    $metadata: deserializeMetadata(output)
  });
  const data = (0, import_smithy_client.expectNonNull)((0, import_smithy_client.expectObject)(await (0, import_core2.parseJsonBody)(output.body, context)), "body");
  const doc = (0, import_smithy_client.take)(data, {
    NextToken: import_smithy_client.expectString,
    Rulesets: (_) => de_RulesetItemList(_, context)
  });
  Object.assign(contents, doc);
  return contents;
}, "de_ListRulesetsCommand");
var de_ListSchedulesCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode !== 200 && output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const contents = (0, import_smithy_client.map)({
    $metadata: deserializeMetadata(output)
  });
  const data = (0, import_smithy_client.expectNonNull)((0, import_smithy_client.expectObject)(await (0, import_core2.parseJsonBody)(output.body, context)), "body");
  const doc = (0, import_smithy_client.take)(data, {
    NextToken: import_smithy_client.expectString,
    Schedules: (_) => de_ScheduleList(_, context)
  });
  Object.assign(contents, doc);
  return contents;
}, "de_ListSchedulesCommand");
var de_ListTagsForResourceCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode !== 200 && output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const contents = (0, import_smithy_client.map)({
    $metadata: deserializeMetadata(output)
  });
  const data = (0, import_smithy_client.expectNonNull)((0, import_smithy_client.expectObject)(await (0, import_core2.parseJsonBody)(output.body, context)), "body");
  const doc = (0, import_smithy_client.take)(data, {
    Tags: import_smithy_client._json
  });
  Object.assign(contents, doc);
  return contents;
}, "de_ListTagsForResourceCommand");
var de_PublishRecipeCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode !== 200 && output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const contents = (0, import_smithy_client.map)({
    $metadata: deserializeMetadata(output)
  });
  const data = (0, import_smithy_client.expectNonNull)((0, import_smithy_client.expectObject)(await (0, import_core2.parseJsonBody)(output.body, context)), "body");
  const doc = (0, import_smithy_client.take)(data, {
    Name: import_smithy_client.expectString
  });
  Object.assign(contents, doc);
  return contents;
}, "de_PublishRecipeCommand");
var de_SendProjectSessionActionCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode !== 200 && output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const contents = (0, import_smithy_client.map)({
    $metadata: deserializeMetadata(output)
  });
  const data = (0, import_smithy_client.expectNonNull)((0, import_smithy_client.expectObject)(await (0, import_core2.parseJsonBody)(output.body, context)), "body");
  const doc = (0, import_smithy_client.take)(data, {
    ActionId: import_smithy_client.expectInt32,
    Name: import_smithy_client.expectString,
    Result: import_smithy_client.expectString
  });
  Object.assign(contents, doc);
  return contents;
}, "de_SendProjectSessionActionCommand");
var de_StartJobRunCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode !== 200 && output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const contents = (0, import_smithy_client.map)({
    $metadata: deserializeMetadata(output)
  });
  const data = (0, import_smithy_client.expectNonNull)((0, import_smithy_client.expectObject)(await (0, import_core2.parseJsonBody)(output.body, context)), "body");
  const doc = (0, import_smithy_client.take)(data, {
    RunId: import_smithy_client.expectString
  });
  Object.assign(contents, doc);
  return contents;
}, "de_StartJobRunCommand");
var de_StartProjectSessionCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode !== 200 && output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const contents = (0, import_smithy_client.map)({
    $metadata: deserializeMetadata(output)
  });
  const data = (0, import_smithy_client.expectNonNull)((0, import_smithy_client.expectObject)(await (0, import_core2.parseJsonBody)(output.body, context)), "body");
  const doc = (0, import_smithy_client.take)(data, {
    ClientSessionId: import_smithy_client.expectString,
    Name: import_smithy_client.expectString
  });
  Object.assign(contents, doc);
  return contents;
}, "de_StartProjectSessionCommand");
var de_StopJobRunCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode !== 200 && output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const contents = (0, import_smithy_client.map)({
    $metadata: deserializeMetadata(output)
  });
  const data = (0, import_smithy_client.expectNonNull)((0, import_smithy_client.expectObject)(await (0, import_core2.parseJsonBody)(output.body, context)), "body");
  const doc = (0, import_smithy_client.take)(data, {
    RunId: import_smithy_client.expectString
  });
  Object.assign(contents, doc);
  return contents;
}, "de_StopJobRunCommand");
var de_TagResourceCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode !== 200 && output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const contents = (0, import_smithy_client.map)({
    $metadata: deserializeMetadata(output)
  });
  await (0, import_smithy_client.collectBody)(output.body, context);
  return contents;
}, "de_TagResourceCommand");
var de_UntagResourceCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode !== 200 && output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const contents = (0, import_smithy_client.map)({
    $metadata: deserializeMetadata(output)
  });
  await (0, import_smithy_client.collectBody)(output.body, context);
  return contents;
}, "de_UntagResourceCommand");
var de_UpdateDatasetCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode !== 200 && output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const contents = (0, import_smithy_client.map)({
    $metadata: deserializeMetadata(output)
  });
  const data = (0, import_smithy_client.expectNonNull)((0, import_smithy_client.expectObject)(await (0, import_core2.parseJsonBody)(output.body, context)), "body");
  const doc = (0, import_smithy_client.take)(data, {
    Name: import_smithy_client.expectString
  });
  Object.assign(contents, doc);
  return contents;
}, "de_UpdateDatasetCommand");
var de_UpdateProfileJobCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode !== 200 && output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const contents = (0, import_smithy_client.map)({
    $metadata: deserializeMetadata(output)
  });
  const data = (0, import_smithy_client.expectNonNull)((0, import_smithy_client.expectObject)(await (0, import_core2.parseJsonBody)(output.body, context)), "body");
  const doc = (0, import_smithy_client.take)(data, {
    Name: import_smithy_client.expectString
  });
  Object.assign(contents, doc);
  return contents;
}, "de_UpdateProfileJobCommand");
var de_UpdateProjectCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode !== 200 && output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const contents = (0, import_smithy_client.map)({
    $metadata: deserializeMetadata(output)
  });
  const data = (0, import_smithy_client.expectNonNull)((0, import_smithy_client.expectObject)(await (0, import_core2.parseJsonBody)(output.body, context)), "body");
  const doc = (0, import_smithy_client.take)(data, {
    LastModifiedDate: (_) => (0, import_smithy_client.expectNonNull)((0, import_smithy_client.parseEpochTimestamp)((0, import_smithy_client.expectNumber)(_))),
    Name: import_smithy_client.expectString
  });
  Object.assign(contents, doc);
  return contents;
}, "de_UpdateProjectCommand");
var de_UpdateRecipeCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode !== 200 && output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const contents = (0, import_smithy_client.map)({
    $metadata: deserializeMetadata(output)
  });
  const data = (0, import_smithy_client.expectNonNull)((0, import_smithy_client.expectObject)(await (0, import_core2.parseJsonBody)(output.body, context)), "body");
  const doc = (0, import_smithy_client.take)(data, {
    Name: import_smithy_client.expectString
  });
  Object.assign(contents, doc);
  return contents;
}, "de_UpdateRecipeCommand");
var de_UpdateRecipeJobCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode !== 200 && output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const contents = (0, import_smithy_client.map)({
    $metadata: deserializeMetadata(output)
  });
  const data = (0, import_smithy_client.expectNonNull)((0, import_smithy_client.expectObject)(await (0, import_core2.parseJsonBody)(output.body, context)), "body");
  const doc = (0, import_smithy_client.take)(data, {
    Name: import_smithy_client.expectString
  });
  Object.assign(contents, doc);
  return contents;
}, "de_UpdateRecipeJobCommand");
var de_UpdateRulesetCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode !== 200 && output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const contents = (0, import_smithy_client.map)({
    $metadata: deserializeMetadata(output)
  });
  const data = (0, import_smithy_client.expectNonNull)((0, import_smithy_client.expectObject)(await (0, import_core2.parseJsonBody)(output.body, context)), "body");
  const doc = (0, import_smithy_client.take)(data, {
    Name: import_smithy_client.expectString
  });
  Object.assign(contents, doc);
  return contents;
}, "de_UpdateRulesetCommand");
var de_UpdateScheduleCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode !== 200 && output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const contents = (0, import_smithy_client.map)({
    $metadata: deserializeMetadata(output)
  });
  const data = (0, import_smithy_client.expectNonNull)((0, import_smithy_client.expectObject)(await (0, import_core2.parseJsonBody)(output.body, context)), "body");
  const doc = (0, import_smithy_client.take)(data, {
    Name: import_smithy_client.expectString
  });
  Object.assign(contents, doc);
  return contents;
}, "de_UpdateScheduleCommand");
var de_CommandError = /* @__PURE__ */ __name(async (output, context) => {
  const parsedOutput = {
    ...output,
    body: await (0, import_core2.parseJsonErrorBody)(output.body, context)
  };
  const errorCode = (0, import_core2.loadRestJsonErrorCode)(output, parsedOutput.body);
  switch (errorCode) {
    case "ConflictException":
    case "com.amazonaws.databrew#ConflictException":
      throw await de_ConflictExceptionRes(parsedOutput, context);
    case "ResourceNotFoundException":
    case "com.amazonaws.databrew#ResourceNotFoundException":
      throw await de_ResourceNotFoundExceptionRes(parsedOutput, context);
    case "ValidationException":
    case "com.amazonaws.databrew#ValidationException":
      throw await de_ValidationExceptionRes(parsedOutput, context);
    case "AccessDeniedException":
    case "com.amazonaws.databrew#AccessDeniedException":
      throw await de_AccessDeniedExceptionRes(parsedOutput, context);
    case "ServiceQuotaExceededException":
    case "com.amazonaws.databrew#ServiceQuotaExceededException":
      throw await de_ServiceQuotaExceededExceptionRes(parsedOutput, context);
    case "InternalServerException":
    case "com.amazonaws.databrew#InternalServerException":
      throw await de_InternalServerExceptionRes(parsedOutput, context);
    default:
      const parsedBody = parsedOutput.body;
      return throwDefaultError({
        output,
        parsedBody,
        errorCode
      });
  }
}, "de_CommandError");
var throwDefaultError = (0, import_smithy_client.withBaseException)(DataBrewServiceException);
var de_AccessDeniedExceptionRes = /* @__PURE__ */ __name(async (parsedOutput, context) => {
  const contents = (0, import_smithy_client.map)({});
  const data = parsedOutput.body;
  const doc = (0, import_smithy_client.take)(data, {
    Message: import_smithy_client.expectString
  });
  Object.assign(contents, doc);
  const exception = new AccessDeniedException({
    $metadata: deserializeMetadata(parsedOutput),
    ...contents
  });
  return (0, import_smithy_client.decorateServiceException)(exception, parsedOutput.body);
}, "de_AccessDeniedExceptionRes");
var de_ConflictExceptionRes = /* @__PURE__ */ __name(async (parsedOutput, context) => {
  const contents = (0, import_smithy_client.map)({});
  const data = parsedOutput.body;
  const doc = (0, import_smithy_client.take)(data, {
    Message: import_smithy_client.expectString
  });
  Object.assign(contents, doc);
  const exception = new ConflictException({
    $metadata: deserializeMetadata(parsedOutput),
    ...contents
  });
  return (0, import_smithy_client.decorateServiceException)(exception, parsedOutput.body);
}, "de_ConflictExceptionRes");
var de_InternalServerExceptionRes = /* @__PURE__ */ __name(async (parsedOutput, context) => {
  const contents = (0, import_smithy_client.map)({});
  const data = parsedOutput.body;
  const doc = (0, import_smithy_client.take)(data, {
    Message: import_smithy_client.expectString
  });
  Object.assign(contents, doc);
  const exception = new InternalServerException({
    $metadata: deserializeMetadata(parsedOutput),
    ...contents
  });
  return (0, import_smithy_client.decorateServiceException)(exception, parsedOutput.body);
}, "de_InternalServerExceptionRes");
var de_ResourceNotFoundExceptionRes = /* @__PURE__ */ __name(async (parsedOutput, context) => {
  const contents = (0, import_smithy_client.map)({});
  const data = parsedOutput.body;
  const doc = (0, import_smithy_client.take)(data, {
    Message: import_smithy_client.expectString
  });
  Object.assign(contents, doc);
  const exception = new ResourceNotFoundException({
    $metadata: deserializeMetadata(parsedOutput),
    ...contents
  });
  return (0, import_smithy_client.decorateServiceException)(exception, parsedOutput.body);
}, "de_ResourceNotFoundExceptionRes");
var de_ServiceQuotaExceededExceptionRes = /* @__PURE__ */ __name(async (parsedOutput, context) => {
  const contents = (0, import_smithy_client.map)({});
  const data = parsedOutput.body;
  const doc = (0, import_smithy_client.take)(data, {
    Message: import_smithy_client.expectString
  });
  Object.assign(contents, doc);
  const exception = new ServiceQuotaExceededException({
    $metadata: deserializeMetadata(parsedOutput),
    ...contents
  });
  return (0, import_smithy_client.decorateServiceException)(exception, parsedOutput.body);
}, "de_ServiceQuotaExceededExceptionRes");
var de_ValidationExceptionRes = /* @__PURE__ */ __name(async (parsedOutput, context) => {
  const contents = (0, import_smithy_client.map)({});
  const data = parsedOutput.body;
  const doc = (0, import_smithy_client.take)(data, {
    Message: import_smithy_client.expectString
  });
  Object.assign(contents, doc);
  const exception = new ValidationException({
    $metadata: deserializeMetadata(parsedOutput),
    ...contents
  });
  return (0, import_smithy_client.decorateServiceException)(exception, parsedOutput.body);
}, "de_ValidationExceptionRes");
var se_Rule = /* @__PURE__ */ __name((input, context) => {
  return (0, import_smithy_client.take)(input, {
    CheckExpression: [],
    ColumnSelectors: import_smithy_client._json,
    Disabled: [],
    Name: [],
    SubstitutionMap: import_smithy_client._json,
    Threshold: (_) => se_Threshold(_, context)
  });
}, "se_Rule");
var se_RuleList = /* @__PURE__ */ __name((input, context) => {
  return input.filter((e) => e != null).map((entry) => {
    return se_Rule(entry, context);
  });
}, "se_RuleList");
var se_Threshold = /* @__PURE__ */ __name((input, context) => {
  return (0, import_smithy_client.take)(input, {
    Type: [],
    Unit: [],
    Value: import_smithy_client.serializeFloat
  });
}, "se_Threshold");
var de_Dataset = /* @__PURE__ */ __name((output, context) => {
  return (0, import_smithy_client.take)(output, {
    AccountId: import_smithy_client.expectString,
    CreateDate: (_) => (0, import_smithy_client.expectNonNull)((0, import_smithy_client.parseEpochTimestamp)((0, import_smithy_client.expectNumber)(_))),
    CreatedBy: import_smithy_client.expectString,
    Format: import_smithy_client.expectString,
    FormatOptions: import_smithy_client._json,
    Input: import_smithy_client._json,
    LastModifiedBy: import_smithy_client.expectString,
    LastModifiedDate: (_) => (0, import_smithy_client.expectNonNull)((0, import_smithy_client.parseEpochTimestamp)((0, import_smithy_client.expectNumber)(_))),
    Name: import_smithy_client.expectString,
    PathOptions: import_smithy_client._json,
    ResourceArn: import_smithy_client.expectString,
    Source: import_smithy_client.expectString,
    Tags: import_smithy_client._json
  });
}, "de_Dataset");
var de_DatasetList = /* @__PURE__ */ __name((output, context) => {
  const retVal = (output || []).filter((e) => e != null).map((entry) => {
    return de_Dataset(entry, context);
  });
  return retVal;
}, "de_DatasetList");
var de_Job = /* @__PURE__ */ __name((output, context) => {
  return (0, import_smithy_client.take)(output, {
    AccountId: import_smithy_client.expectString,
    CreateDate: (_) => (0, import_smithy_client.expectNonNull)((0, import_smithy_client.parseEpochTimestamp)((0, import_smithy_client.expectNumber)(_))),
    CreatedBy: import_smithy_client.expectString,
    DataCatalogOutputs: import_smithy_client._json,
    DatabaseOutputs: import_smithy_client._json,
    DatasetName: import_smithy_client.expectString,
    EncryptionKeyArn: import_smithy_client.expectString,
    EncryptionMode: import_smithy_client.expectString,
    JobSample: import_smithy_client._json,
    LastModifiedBy: import_smithy_client.expectString,
    LastModifiedDate: (_) => (0, import_smithy_client.expectNonNull)((0, import_smithy_client.parseEpochTimestamp)((0, import_smithy_client.expectNumber)(_))),
    LogSubscription: import_smithy_client.expectString,
    MaxCapacity: import_smithy_client.expectInt32,
    MaxRetries: import_smithy_client.expectInt32,
    Name: import_smithy_client.expectString,
    Outputs: import_smithy_client._json,
    ProjectName: import_smithy_client.expectString,
    RecipeReference: import_smithy_client._json,
    ResourceArn: import_smithy_client.expectString,
    RoleArn: import_smithy_client.expectString,
    Tags: import_smithy_client._json,
    Timeout: import_smithy_client.expectInt32,
    Type: import_smithy_client.expectString,
    ValidationConfigurations: import_smithy_client._json
  });
}, "de_Job");
var de_JobList = /* @__PURE__ */ __name((output, context) => {
  const retVal = (output || []).filter((e) => e != null).map((entry) => {
    return de_Job(entry, context);
  });
  return retVal;
}, "de_JobList");
var de_JobRun = /* @__PURE__ */ __name((output, context) => {
  return (0, import_smithy_client.take)(output, {
    Attempt: import_smithy_client.expectInt32,
    CompletedOn: (_) => (0, import_smithy_client.expectNonNull)((0, import_smithy_client.parseEpochTimestamp)((0, import_smithy_client.expectNumber)(_))),
    DataCatalogOutputs: import_smithy_client._json,
    DatabaseOutputs: import_smithy_client._json,
    DatasetName: import_smithy_client.expectString,
    ErrorMessage: import_smithy_client.expectString,
    ExecutionTime: import_smithy_client.expectInt32,
    JobName: import_smithy_client.expectString,
    JobSample: import_smithy_client._json,
    LogGroupName: import_smithy_client.expectString,
    LogSubscription: import_smithy_client.expectString,
    Outputs: import_smithy_client._json,
    RecipeReference: import_smithy_client._json,
    RunId: import_smithy_client.expectString,
    StartedBy: import_smithy_client.expectString,
    StartedOn: (_) => (0, import_smithy_client.expectNonNull)((0, import_smithy_client.parseEpochTimestamp)((0, import_smithy_client.expectNumber)(_))),
    State: import_smithy_client.expectString,
    ValidationConfigurations: import_smithy_client._json
  });
}, "de_JobRun");
var de_JobRunList = /* @__PURE__ */ __name((output, context) => {
  const retVal = (output || []).filter((e) => e != null).map((entry) => {
    return de_JobRun(entry, context);
  });
  return retVal;
}, "de_JobRunList");
var de_Project = /* @__PURE__ */ __name((output, context) => {
  return (0, import_smithy_client.take)(output, {
    AccountId: import_smithy_client.expectString,
    CreateDate: (_) => (0, import_smithy_client.expectNonNull)((0, import_smithy_client.parseEpochTimestamp)((0, import_smithy_client.expectNumber)(_))),
    CreatedBy: import_smithy_client.expectString,
    DatasetName: import_smithy_client.expectString,
    LastModifiedBy: import_smithy_client.expectString,
    LastModifiedDate: (_) => (0, import_smithy_client.expectNonNull)((0, import_smithy_client.parseEpochTimestamp)((0, import_smithy_client.expectNumber)(_))),
    Name: import_smithy_client.expectString,
    OpenDate: (_) => (0, import_smithy_client.expectNonNull)((0, import_smithy_client.parseEpochTimestamp)((0, import_smithy_client.expectNumber)(_))),
    OpenedBy: import_smithy_client.expectString,
    RecipeName: import_smithy_client.expectString,
    ResourceArn: import_smithy_client.expectString,
    RoleArn: import_smithy_client.expectString,
    Sample: import_smithy_client._json,
    Tags: import_smithy_client._json
  });
}, "de_Project");
var de_ProjectList = /* @__PURE__ */ __name((output, context) => {
  const retVal = (output || []).filter((e) => e != null).map((entry) => {
    return de_Project(entry, context);
  });
  return retVal;
}, "de_ProjectList");
var de_Recipe = /* @__PURE__ */ __name((output, context) => {
  return (0, import_smithy_client.take)(output, {
    CreateDate: (_) => (0, import_smithy_client.expectNonNull)((0, import_smithy_client.parseEpochTimestamp)((0, import_smithy_client.expectNumber)(_))),
    CreatedBy: import_smithy_client.expectString,
    Description: import_smithy_client.expectString,
    LastModifiedBy: import_smithy_client.expectString,
    LastModifiedDate: (_) => (0, import_smithy_client.expectNonNull)((0, import_smithy_client.parseEpochTimestamp)((0, import_smithy_client.expectNumber)(_))),
    Name: import_smithy_client.expectString,
    ProjectName: import_smithy_client.expectString,
    PublishedBy: import_smithy_client.expectString,
    PublishedDate: (_) => (0, import_smithy_client.expectNonNull)((0, import_smithy_client.parseEpochTimestamp)((0, import_smithy_client.expectNumber)(_))),
    RecipeVersion: import_smithy_client.expectString,
    ResourceArn: import_smithy_client.expectString,
    Steps: import_smithy_client._json,
    Tags: import_smithy_client._json
  });
}, "de_Recipe");
var de_RecipeList = /* @__PURE__ */ __name((output, context) => {
  const retVal = (output || []).filter((e) => e != null).map((entry) => {
    return de_Recipe(entry, context);
  });
  return retVal;
}, "de_RecipeList");
var de_Rule = /* @__PURE__ */ __name((output, context) => {
  return (0, import_smithy_client.take)(output, {
    CheckExpression: import_smithy_client.expectString,
    ColumnSelectors: import_smithy_client._json,
    Disabled: import_smithy_client.expectBoolean,
    Name: import_smithy_client.expectString,
    SubstitutionMap: import_smithy_client._json,
    Threshold: (_) => de_Threshold(_, context)
  });
}, "de_Rule");
var de_RuleList = /* @__PURE__ */ __name((output, context) => {
  const retVal = (output || []).filter((e) => e != null).map((entry) => {
    return de_Rule(entry, context);
  });
  return retVal;
}, "de_RuleList");
var de_RulesetItem = /* @__PURE__ */ __name((output, context) => {
  return (0, import_smithy_client.take)(output, {
    AccountId: import_smithy_client.expectString,
    CreateDate: (_) => (0, import_smithy_client.expectNonNull)((0, import_smithy_client.parseEpochTimestamp)((0, import_smithy_client.expectNumber)(_))),
    CreatedBy: import_smithy_client.expectString,
    Description: import_smithy_client.expectString,
    LastModifiedBy: import_smithy_client.expectString,
    LastModifiedDate: (_) => (0, import_smithy_client.expectNonNull)((0, import_smithy_client.parseEpochTimestamp)((0, import_smithy_client.expectNumber)(_))),
    Name: import_smithy_client.expectString,
    ResourceArn: import_smithy_client.expectString,
    RuleCount: import_smithy_client.expectInt32,
    Tags: import_smithy_client._json,
    TargetArn: import_smithy_client.expectString
  });
}, "de_RulesetItem");
var de_RulesetItemList = /* @__PURE__ */ __name((output, context) => {
  const retVal = (output || []).filter((e) => e != null).map((entry) => {
    return de_RulesetItem(entry, context);
  });
  return retVal;
}, "de_RulesetItemList");
var de_Schedule = /* @__PURE__ */ __name((output, context) => {
  return (0, import_smithy_client.take)(output, {
    AccountId: import_smithy_client.expectString,
    CreateDate: (_) => (0, import_smithy_client.expectNonNull)((0, import_smithy_client.parseEpochTimestamp)((0, import_smithy_client.expectNumber)(_))),
    CreatedBy: import_smithy_client.expectString,
    CronExpression: import_smithy_client.expectString,
    JobNames: import_smithy_client._json,
    LastModifiedBy: import_smithy_client.expectString,
    LastModifiedDate: (_) => (0, import_smithy_client.expectNonNull)((0, import_smithy_client.parseEpochTimestamp)((0, import_smithy_client.expectNumber)(_))),
    Name: import_smithy_client.expectString,
    ResourceArn: import_smithy_client.expectString,
    Tags: import_smithy_client._json
  });
}, "de_Schedule");
var de_ScheduleList = /* @__PURE__ */ __name((output, context) => {
  const retVal = (output || []).filter((e) => e != null).map((entry) => {
    return de_Schedule(entry, context);
  });
  return retVal;
}, "de_ScheduleList");
var de_Threshold = /* @__PURE__ */ __name((output, context) => {
  return (0, import_smithy_client.take)(output, {
    Type: import_smithy_client.expectString,
    Unit: import_smithy_client.expectString,
    Value: import_smithy_client.limitedParseDouble
  });
}, "de_Threshold");
var deserializeMetadata = /* @__PURE__ */ __name((output) => ({
  httpStatusCode: output.statusCode,
  requestId: output.headers["x-amzn-requestid"] ?? output.headers["x-amzn-request-id"] ?? output.headers["x-amz-request-id"],
  extendedRequestId: output.headers["x-amz-id-2"],
  cfId: output.headers["x-amz-cf-id"]
}), "deserializeMetadata");
var _DN = "DatasetName";
var _JN = "JobName";
var _MR = "MaxResults";
var _N = "Name";
var _NT = "NextToken";
var _PN = "ProjectName";
var _RV = "RecipeVersion";
var _TA = "TargetArn";
var _TK = "TagKeys";
var _dN = "datasetName";
var _jN = "jobName";
var _mR = "maxResults";
var _n = "name";
var _nT = "nextToken";
var _pN = "projectName";
var _rV = "recipeVersion";
var _tA = "targetArn";
var _tK = "tagKeys";

// src/commands/BatchDeleteRecipeVersionCommand.ts
var _BatchDeleteRecipeVersionCommand = class _BatchDeleteRecipeVersionCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AWSGlueDataBrew", "BatchDeleteRecipeVersion", {}).n("DataBrewClient", "BatchDeleteRecipeVersionCommand").f(void 0, void 0).ser(se_BatchDeleteRecipeVersionCommand).de(de_BatchDeleteRecipeVersionCommand).build() {
};
__name(_BatchDeleteRecipeVersionCommand, "BatchDeleteRecipeVersionCommand");
var BatchDeleteRecipeVersionCommand = _BatchDeleteRecipeVersionCommand;

// src/commands/CreateDatasetCommand.ts




var _CreateDatasetCommand = class _CreateDatasetCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AWSGlueDataBrew", "CreateDataset", {}).n("DataBrewClient", "CreateDatasetCommand").f(void 0, void 0).ser(se_CreateDatasetCommand).de(de_CreateDatasetCommand).build() {
};
__name(_CreateDatasetCommand, "CreateDatasetCommand");
var CreateDatasetCommand = _CreateDatasetCommand;

// src/commands/CreateProfileJobCommand.ts




var _CreateProfileJobCommand = class _CreateProfileJobCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AWSGlueDataBrew", "CreateProfileJob", {}).n("DataBrewClient", "CreateProfileJobCommand").f(void 0, void 0).ser(se_CreateProfileJobCommand).de(de_CreateProfileJobCommand).build() {
};
__name(_CreateProfileJobCommand, "CreateProfileJobCommand");
var CreateProfileJobCommand = _CreateProfileJobCommand;

// src/commands/CreateProjectCommand.ts




var _CreateProjectCommand = class _CreateProjectCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AWSGlueDataBrew", "CreateProject", {}).n("DataBrewClient", "CreateProjectCommand").f(void 0, void 0).ser(se_CreateProjectCommand).de(de_CreateProjectCommand).build() {
};
__name(_CreateProjectCommand, "CreateProjectCommand");
var CreateProjectCommand = _CreateProjectCommand;

// src/commands/CreateRecipeCommand.ts




var _CreateRecipeCommand = class _CreateRecipeCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AWSGlueDataBrew", "CreateRecipe", {}).n("DataBrewClient", "CreateRecipeCommand").f(void 0, void 0).ser(se_CreateRecipeCommand).de(de_CreateRecipeCommand).build() {
};
__name(_CreateRecipeCommand, "CreateRecipeCommand");
var CreateRecipeCommand = _CreateRecipeCommand;

// src/commands/CreateRecipeJobCommand.ts




var _CreateRecipeJobCommand = class _CreateRecipeJobCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AWSGlueDataBrew", "CreateRecipeJob", {}).n("DataBrewClient", "CreateRecipeJobCommand").f(void 0, void 0).ser(se_CreateRecipeJobCommand).de(de_CreateRecipeJobCommand).build() {
};
__name(_CreateRecipeJobCommand, "CreateRecipeJobCommand");
var CreateRecipeJobCommand = _CreateRecipeJobCommand;

// src/commands/CreateRulesetCommand.ts




var _CreateRulesetCommand = class _CreateRulesetCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AWSGlueDataBrew", "CreateRuleset", {}).n("DataBrewClient", "CreateRulesetCommand").f(void 0, void 0).ser(se_CreateRulesetCommand).de(de_CreateRulesetCommand).build() {
};
__name(_CreateRulesetCommand, "CreateRulesetCommand");
var CreateRulesetCommand = _CreateRulesetCommand;

// src/commands/CreateScheduleCommand.ts




var _CreateScheduleCommand = class _CreateScheduleCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AWSGlueDataBrew", "CreateSchedule", {}).n("DataBrewClient", "CreateScheduleCommand").f(void 0, void 0).ser(se_CreateScheduleCommand).de(de_CreateScheduleCommand).build() {
};
__name(_CreateScheduleCommand, "CreateScheduleCommand");
var CreateScheduleCommand = _CreateScheduleCommand;

// src/commands/DeleteDatasetCommand.ts




var _DeleteDatasetCommand = class _DeleteDatasetCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AWSGlueDataBrew", "DeleteDataset", {}).n("DataBrewClient", "DeleteDatasetCommand").f(void 0, void 0).ser(se_DeleteDatasetCommand).de(de_DeleteDatasetCommand).build() {
};
__name(_DeleteDatasetCommand, "DeleteDatasetCommand");
var DeleteDatasetCommand = _DeleteDatasetCommand;

// src/commands/DeleteJobCommand.ts




var _DeleteJobCommand = class _DeleteJobCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AWSGlueDataBrew", "DeleteJob", {}).n("DataBrewClient", "DeleteJobCommand").f(void 0, void 0).ser(se_DeleteJobCommand).de(de_DeleteJobCommand).build() {
};
__name(_DeleteJobCommand, "DeleteJobCommand");
var DeleteJobCommand = _DeleteJobCommand;

// src/commands/DeleteProjectCommand.ts




var _DeleteProjectCommand = class _DeleteProjectCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AWSGlueDataBrew", "DeleteProject", {}).n("DataBrewClient", "DeleteProjectCommand").f(void 0, void 0).ser(se_DeleteProjectCommand).de(de_DeleteProjectCommand).build() {
};
__name(_DeleteProjectCommand, "DeleteProjectCommand");
var DeleteProjectCommand = _DeleteProjectCommand;

// src/commands/DeleteRecipeVersionCommand.ts




var _DeleteRecipeVersionCommand = class _DeleteRecipeVersionCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AWSGlueDataBrew", "DeleteRecipeVersion", {}).n("DataBrewClient", "DeleteRecipeVersionCommand").f(void 0, void 0).ser(se_DeleteRecipeVersionCommand).de(de_DeleteRecipeVersionCommand).build() {
};
__name(_DeleteRecipeVersionCommand, "DeleteRecipeVersionCommand");
var DeleteRecipeVersionCommand = _DeleteRecipeVersionCommand;

// src/commands/DeleteRulesetCommand.ts




var _DeleteRulesetCommand = class _DeleteRulesetCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AWSGlueDataBrew", "DeleteRuleset", {}).n("DataBrewClient", "DeleteRulesetCommand").f(void 0, void 0).ser(se_DeleteRulesetCommand).de(de_DeleteRulesetCommand).build() {
};
__name(_DeleteRulesetCommand, "DeleteRulesetCommand");
var DeleteRulesetCommand = _DeleteRulesetCommand;

// src/commands/DeleteScheduleCommand.ts




var _DeleteScheduleCommand = class _DeleteScheduleCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AWSGlueDataBrew", "DeleteSchedule", {}).n("DataBrewClient", "DeleteScheduleCommand").f(void 0, void 0).ser(se_DeleteScheduleCommand).de(de_DeleteScheduleCommand).build() {
};
__name(_DeleteScheduleCommand, "DeleteScheduleCommand");
var DeleteScheduleCommand = _DeleteScheduleCommand;

// src/commands/DescribeDatasetCommand.ts




var _DescribeDatasetCommand = class _DescribeDatasetCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AWSGlueDataBrew", "DescribeDataset", {}).n("DataBrewClient", "DescribeDatasetCommand").f(void 0, void 0).ser(se_DescribeDatasetCommand).de(de_DescribeDatasetCommand).build() {
};
__name(_DescribeDatasetCommand, "DescribeDatasetCommand");
var DescribeDatasetCommand = _DescribeDatasetCommand;

// src/commands/DescribeJobCommand.ts




var _DescribeJobCommand = class _DescribeJobCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AWSGlueDataBrew", "DescribeJob", {}).n("DataBrewClient", "DescribeJobCommand").f(void 0, void 0).ser(se_DescribeJobCommand).de(de_DescribeJobCommand).build() {
};
__name(_DescribeJobCommand, "DescribeJobCommand");
var DescribeJobCommand = _DescribeJobCommand;

// src/commands/DescribeJobRunCommand.ts




var _DescribeJobRunCommand = class _DescribeJobRunCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AWSGlueDataBrew", "DescribeJobRun", {}).n("DataBrewClient", "DescribeJobRunCommand").f(void 0, void 0).ser(se_DescribeJobRunCommand).de(de_DescribeJobRunCommand).build() {
};
__name(_DescribeJobRunCommand, "DescribeJobRunCommand");
var DescribeJobRunCommand = _DescribeJobRunCommand;

// src/commands/DescribeProjectCommand.ts




var _DescribeProjectCommand = class _DescribeProjectCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AWSGlueDataBrew", "DescribeProject", {}).n("DataBrewClient", "DescribeProjectCommand").f(void 0, void 0).ser(se_DescribeProjectCommand).de(de_DescribeProjectCommand).build() {
};
__name(_DescribeProjectCommand, "DescribeProjectCommand");
var DescribeProjectCommand = _DescribeProjectCommand;

// src/commands/DescribeRecipeCommand.ts




var _DescribeRecipeCommand = class _DescribeRecipeCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AWSGlueDataBrew", "DescribeRecipe", {}).n("DataBrewClient", "DescribeRecipeCommand").f(void 0, void 0).ser(se_DescribeRecipeCommand).de(de_DescribeRecipeCommand).build() {
};
__name(_DescribeRecipeCommand, "DescribeRecipeCommand");
var DescribeRecipeCommand = _DescribeRecipeCommand;

// src/commands/DescribeRulesetCommand.ts




var _DescribeRulesetCommand = class _DescribeRulesetCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AWSGlueDataBrew", "DescribeRuleset", {}).n("DataBrewClient", "DescribeRulesetCommand").f(void 0, void 0).ser(se_DescribeRulesetCommand).de(de_DescribeRulesetCommand).build() {
};
__name(_DescribeRulesetCommand, "DescribeRulesetCommand");
var DescribeRulesetCommand = _DescribeRulesetCommand;

// src/commands/DescribeScheduleCommand.ts




var _DescribeScheduleCommand = class _DescribeScheduleCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AWSGlueDataBrew", "DescribeSchedule", {}).n("DataBrewClient", "DescribeScheduleCommand").f(void 0, void 0).ser(se_DescribeScheduleCommand).de(de_DescribeScheduleCommand).build() {
};
__name(_DescribeScheduleCommand, "DescribeScheduleCommand");
var DescribeScheduleCommand = _DescribeScheduleCommand;

// src/commands/ListDatasetsCommand.ts




var _ListDatasetsCommand = class _ListDatasetsCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AWSGlueDataBrew", "ListDatasets", {}).n("DataBrewClient", "ListDatasetsCommand").f(void 0, void 0).ser(se_ListDatasetsCommand).de(de_ListDatasetsCommand).build() {
};
__name(_ListDatasetsCommand, "ListDatasetsCommand");
var ListDatasetsCommand = _ListDatasetsCommand;

// src/commands/ListJobRunsCommand.ts




var _ListJobRunsCommand = class _ListJobRunsCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AWSGlueDataBrew", "ListJobRuns", {}).n("DataBrewClient", "ListJobRunsCommand").f(void 0, void 0).ser(se_ListJobRunsCommand).de(de_ListJobRunsCommand).build() {
};
__name(_ListJobRunsCommand, "ListJobRunsCommand");
var ListJobRunsCommand = _ListJobRunsCommand;

// src/commands/ListJobsCommand.ts




var _ListJobsCommand = class _ListJobsCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AWSGlueDataBrew", "ListJobs", {}).n("DataBrewClient", "ListJobsCommand").f(void 0, void 0).ser(se_ListJobsCommand).de(de_ListJobsCommand).build() {
};
__name(_ListJobsCommand, "ListJobsCommand");
var ListJobsCommand = _ListJobsCommand;

// src/commands/ListProjectsCommand.ts




var _ListProjectsCommand = class _ListProjectsCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AWSGlueDataBrew", "ListProjects", {}).n("DataBrewClient", "ListProjectsCommand").f(void 0, void 0).ser(se_ListProjectsCommand).de(de_ListProjectsCommand).build() {
};
__name(_ListProjectsCommand, "ListProjectsCommand");
var ListProjectsCommand = _ListProjectsCommand;

// src/commands/ListRecipesCommand.ts




var _ListRecipesCommand = class _ListRecipesCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AWSGlueDataBrew", "ListRecipes", {}).n("DataBrewClient", "ListRecipesCommand").f(void 0, void 0).ser(se_ListRecipesCommand).de(de_ListRecipesCommand).build() {
};
__name(_ListRecipesCommand, "ListRecipesCommand");
var ListRecipesCommand = _ListRecipesCommand;

// src/commands/ListRecipeVersionsCommand.ts




var _ListRecipeVersionsCommand = class _ListRecipeVersionsCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AWSGlueDataBrew", "ListRecipeVersions", {}).n("DataBrewClient", "ListRecipeVersionsCommand").f(void 0, void 0).ser(se_ListRecipeVersionsCommand).de(de_ListRecipeVersionsCommand).build() {
};
__name(_ListRecipeVersionsCommand, "ListRecipeVersionsCommand");
var ListRecipeVersionsCommand = _ListRecipeVersionsCommand;

// src/commands/ListRulesetsCommand.ts




var _ListRulesetsCommand = class _ListRulesetsCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AWSGlueDataBrew", "ListRulesets", {}).n("DataBrewClient", "ListRulesetsCommand").f(void 0, void 0).ser(se_ListRulesetsCommand).de(de_ListRulesetsCommand).build() {
};
__name(_ListRulesetsCommand, "ListRulesetsCommand");
var ListRulesetsCommand = _ListRulesetsCommand;

// src/commands/ListSchedulesCommand.ts




var _ListSchedulesCommand = class _ListSchedulesCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AWSGlueDataBrew", "ListSchedules", {}).n("DataBrewClient", "ListSchedulesCommand").f(void 0, void 0).ser(se_ListSchedulesCommand).de(de_ListSchedulesCommand).build() {
};
__name(_ListSchedulesCommand, "ListSchedulesCommand");
var ListSchedulesCommand = _ListSchedulesCommand;

// src/commands/ListTagsForResourceCommand.ts




var _ListTagsForResourceCommand = class _ListTagsForResourceCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AWSGlueDataBrew", "ListTagsForResource", {}).n("DataBrewClient", "ListTagsForResourceCommand").f(void 0, void 0).ser(se_ListTagsForResourceCommand).de(de_ListTagsForResourceCommand).build() {
};
__name(_ListTagsForResourceCommand, "ListTagsForResourceCommand");
var ListTagsForResourceCommand = _ListTagsForResourceCommand;

// src/commands/PublishRecipeCommand.ts




var _PublishRecipeCommand = class _PublishRecipeCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AWSGlueDataBrew", "PublishRecipe", {}).n("DataBrewClient", "PublishRecipeCommand").f(void 0, void 0).ser(se_PublishRecipeCommand).de(de_PublishRecipeCommand).build() {
};
__name(_PublishRecipeCommand, "PublishRecipeCommand");
var PublishRecipeCommand = _PublishRecipeCommand;

// src/commands/SendProjectSessionActionCommand.ts




var _SendProjectSessionActionCommand = class _SendProjectSessionActionCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AWSGlueDataBrew", "SendProjectSessionAction", {}).n("DataBrewClient", "SendProjectSessionActionCommand").f(SendProjectSessionActionRequestFilterSensitiveLog, void 0).ser(se_SendProjectSessionActionCommand).de(de_SendProjectSessionActionCommand).build() {
};
__name(_SendProjectSessionActionCommand, "SendProjectSessionActionCommand");
var SendProjectSessionActionCommand = _SendProjectSessionActionCommand;

// src/commands/StartJobRunCommand.ts




var _StartJobRunCommand = class _StartJobRunCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AWSGlueDataBrew", "StartJobRun", {}).n("DataBrewClient", "StartJobRunCommand").f(void 0, void 0).ser(se_StartJobRunCommand).de(de_StartJobRunCommand).build() {
};
__name(_StartJobRunCommand, "StartJobRunCommand");
var StartJobRunCommand = _StartJobRunCommand;

// src/commands/StartProjectSessionCommand.ts




var _StartProjectSessionCommand = class _StartProjectSessionCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AWSGlueDataBrew", "StartProjectSession", {}).n("DataBrewClient", "StartProjectSessionCommand").f(void 0, StartProjectSessionResponseFilterSensitiveLog).ser(se_StartProjectSessionCommand).de(de_StartProjectSessionCommand).build() {
};
__name(_StartProjectSessionCommand, "StartProjectSessionCommand");
var StartProjectSessionCommand = _StartProjectSessionCommand;

// src/commands/StopJobRunCommand.ts




var _StopJobRunCommand = class _StopJobRunCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AWSGlueDataBrew", "StopJobRun", {}).n("DataBrewClient", "StopJobRunCommand").f(void 0, void 0).ser(se_StopJobRunCommand).de(de_StopJobRunCommand).build() {
};
__name(_StopJobRunCommand, "StopJobRunCommand");
var StopJobRunCommand = _StopJobRunCommand;

// src/commands/TagResourceCommand.ts




var _TagResourceCommand = class _TagResourceCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AWSGlueDataBrew", "TagResource", {}).n("DataBrewClient", "TagResourceCommand").f(void 0, void 0).ser(se_TagResourceCommand).de(de_TagResourceCommand).build() {
};
__name(_TagResourceCommand, "TagResourceCommand");
var TagResourceCommand = _TagResourceCommand;

// src/commands/UntagResourceCommand.ts




var _UntagResourceCommand = class _UntagResourceCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AWSGlueDataBrew", "UntagResource", {}).n("DataBrewClient", "UntagResourceCommand").f(void 0, void 0).ser(se_UntagResourceCommand).de(de_UntagResourceCommand).build() {
};
__name(_UntagResourceCommand, "UntagResourceCommand");
var UntagResourceCommand = _UntagResourceCommand;

// src/commands/UpdateDatasetCommand.ts




var _UpdateDatasetCommand = class _UpdateDatasetCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AWSGlueDataBrew", "UpdateDataset", {}).n("DataBrewClient", "UpdateDatasetCommand").f(void 0, void 0).ser(se_UpdateDatasetCommand).de(de_UpdateDatasetCommand).build() {
};
__name(_UpdateDatasetCommand, "UpdateDatasetCommand");
var UpdateDatasetCommand = _UpdateDatasetCommand;

// src/commands/UpdateProfileJobCommand.ts




var _UpdateProfileJobCommand = class _UpdateProfileJobCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AWSGlueDataBrew", "UpdateProfileJob", {}).n("DataBrewClient", "UpdateProfileJobCommand").f(void 0, void 0).ser(se_UpdateProfileJobCommand).de(de_UpdateProfileJobCommand).build() {
};
__name(_UpdateProfileJobCommand, "UpdateProfileJobCommand");
var UpdateProfileJobCommand = _UpdateProfileJobCommand;

// src/commands/UpdateProjectCommand.ts




var _UpdateProjectCommand = class _UpdateProjectCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AWSGlueDataBrew", "UpdateProject", {}).n("DataBrewClient", "UpdateProjectCommand").f(void 0, void 0).ser(se_UpdateProjectCommand).de(de_UpdateProjectCommand).build() {
};
__name(_UpdateProjectCommand, "UpdateProjectCommand");
var UpdateProjectCommand = _UpdateProjectCommand;

// src/commands/UpdateRecipeCommand.ts




var _UpdateRecipeCommand = class _UpdateRecipeCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AWSGlueDataBrew", "UpdateRecipe", {}).n("DataBrewClient", "UpdateRecipeCommand").f(void 0, void 0).ser(se_UpdateRecipeCommand).de(de_UpdateRecipeCommand).build() {
};
__name(_UpdateRecipeCommand, "UpdateRecipeCommand");
var UpdateRecipeCommand = _UpdateRecipeCommand;

// src/commands/UpdateRecipeJobCommand.ts




var _UpdateRecipeJobCommand = class _UpdateRecipeJobCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AWSGlueDataBrew", "UpdateRecipeJob", {}).n("DataBrewClient", "UpdateRecipeJobCommand").f(void 0, void 0).ser(se_UpdateRecipeJobCommand).de(de_UpdateRecipeJobCommand).build() {
};
__name(_UpdateRecipeJobCommand, "UpdateRecipeJobCommand");
var UpdateRecipeJobCommand = _UpdateRecipeJobCommand;

// src/commands/UpdateRulesetCommand.ts




var _UpdateRulesetCommand = class _UpdateRulesetCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AWSGlueDataBrew", "UpdateRuleset", {}).n("DataBrewClient", "UpdateRulesetCommand").f(void 0, void 0).ser(se_UpdateRulesetCommand).de(de_UpdateRulesetCommand).build() {
};
__name(_UpdateRulesetCommand, "UpdateRulesetCommand");
var UpdateRulesetCommand = _UpdateRulesetCommand;

// src/commands/UpdateScheduleCommand.ts




var _UpdateScheduleCommand = class _UpdateScheduleCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AWSGlueDataBrew", "UpdateSchedule", {}).n("DataBrewClient", "UpdateScheduleCommand").f(void 0, void 0).ser(se_UpdateScheduleCommand).de(de_UpdateScheduleCommand).build() {
};
__name(_UpdateScheduleCommand, "UpdateScheduleCommand");
var UpdateScheduleCommand = _UpdateScheduleCommand;

// src/DataBrew.ts
var commands = {
  BatchDeleteRecipeVersionCommand,
  CreateDatasetCommand,
  CreateProfileJobCommand,
  CreateProjectCommand,
  CreateRecipeCommand,
  CreateRecipeJobCommand,
  CreateRulesetCommand,
  CreateScheduleCommand,
  DeleteDatasetCommand,
  DeleteJobCommand,
  DeleteProjectCommand,
  DeleteRecipeVersionCommand,
  DeleteRulesetCommand,
  DeleteScheduleCommand,
  DescribeDatasetCommand,
  DescribeJobCommand,
  DescribeJobRunCommand,
  DescribeProjectCommand,
  DescribeRecipeCommand,
  DescribeRulesetCommand,
  DescribeScheduleCommand,
  ListDatasetsCommand,
  ListJobRunsCommand,
  ListJobsCommand,
  ListProjectsCommand,
  ListRecipesCommand,
  ListRecipeVersionsCommand,
  ListRulesetsCommand,
  ListSchedulesCommand,
  ListTagsForResourceCommand,
  PublishRecipeCommand,
  SendProjectSessionActionCommand,
  StartJobRunCommand,
  StartProjectSessionCommand,
  StopJobRunCommand,
  TagResourceCommand,
  UntagResourceCommand,
  UpdateDatasetCommand,
  UpdateProfileJobCommand,
  UpdateProjectCommand,
  UpdateRecipeCommand,
  UpdateRecipeJobCommand,
  UpdateRulesetCommand,
  UpdateScheduleCommand
};
var _DataBrew = class _DataBrew extends DataBrewClient {
};
__name(_DataBrew, "DataBrew");
var DataBrew = _DataBrew;
(0, import_smithy_client.createAggregatedClient)(commands, DataBrew);

// src/pagination/ListDatasetsPaginator.ts

var paginateListDatasets = (0, import_core.createPaginator)(DataBrewClient, ListDatasetsCommand, "NextToken", "NextToken", "MaxResults");

// src/pagination/ListJobRunsPaginator.ts

var paginateListJobRuns = (0, import_core.createPaginator)(DataBrewClient, ListJobRunsCommand, "NextToken", "NextToken", "MaxResults");

// src/pagination/ListJobsPaginator.ts

var paginateListJobs = (0, import_core.createPaginator)(DataBrewClient, ListJobsCommand, "NextToken", "NextToken", "MaxResults");

// src/pagination/ListProjectsPaginator.ts

var paginateListProjects = (0, import_core.createPaginator)(DataBrewClient, ListProjectsCommand, "NextToken", "NextToken", "MaxResults");

// src/pagination/ListRecipeVersionsPaginator.ts

var paginateListRecipeVersions = (0, import_core.createPaginator)(DataBrewClient, ListRecipeVersionsCommand, "NextToken", "NextToken", "MaxResults");

// src/pagination/ListRecipesPaginator.ts

var paginateListRecipes = (0, import_core.createPaginator)(DataBrewClient, ListRecipesCommand, "NextToken", "NextToken", "MaxResults");

// src/pagination/ListRulesetsPaginator.ts

var paginateListRulesets = (0, import_core.createPaginator)(DataBrewClient, ListRulesetsCommand, "NextToken", "NextToken", "MaxResults");

// src/pagination/ListSchedulesPaginator.ts

var paginateListSchedules = (0, import_core.createPaginator)(DataBrewClient, ListSchedulesCommand, "NextToken", "NextToken", "MaxResults");
// Annotate the CommonJS export names for ESM import in node:

0 && (module.exports = {
  DataBrewServiceException,
  __Client,
  DataBrewClient,
  DataBrew,
  $Command,
  BatchDeleteRecipeVersionCommand,
  CreateDatasetCommand,
  CreateProfileJobCommand,
  CreateProjectCommand,
  CreateRecipeCommand,
  CreateRecipeJobCommand,
  CreateRulesetCommand,
  CreateScheduleCommand,
  DeleteDatasetCommand,
  DeleteJobCommand,
  DeleteProjectCommand,
  DeleteRecipeVersionCommand,
  DeleteRulesetCommand,
  DeleteScheduleCommand,
  DescribeDatasetCommand,
  DescribeJobCommand,
  DescribeJobRunCommand,
  DescribeProjectCommand,
  DescribeRecipeCommand,
  DescribeRulesetCommand,
  DescribeScheduleCommand,
  ListDatasetsCommand,
  ListJobRunsCommand,
  ListJobsCommand,
  ListProjectsCommand,
  ListRecipeVersionsCommand,
  ListRecipesCommand,
  ListRulesetsCommand,
  ListSchedulesCommand,
  ListTagsForResourceCommand,
  PublishRecipeCommand,
  SendProjectSessionActionCommand,
  StartJobRunCommand,
  StartProjectSessionCommand,
  StopJobRunCommand,
  TagResourceCommand,
  UntagResourceCommand,
  UpdateDatasetCommand,
  UpdateProfileJobCommand,
  UpdateProjectCommand,
  UpdateRecipeCommand,
  UpdateRecipeJobCommand,
  UpdateRulesetCommand,
  UpdateScheduleCommand,
  paginateListDatasets,
  paginateListJobRuns,
  paginateListJobs,
  paginateListProjects,
  paginateListRecipeVersions,
  paginateListRecipes,
  paginateListRulesets,
  paginateListSchedules,
  AccessDeniedException,
  AnalyticsMode,
  ConflictException,
  ResourceNotFoundException,
  ValidationException,
  InputFormat,
  Order,
  OrderedBy,
  ParameterType,
  ServiceQuotaExceededException,
  EncryptionMode,
  SampleMode,
  LogSubscription,
  ValidationMode,
  SampleType,
  InternalServerException,
  DatabaseOutputMode,
  CompressionFormat,
  OutputFormat,
  ThresholdType,
  ThresholdUnit,
  Source,
  JobType,
  JobRunState,
  SessionStatus,
  SendProjectSessionActionRequestFilterSensitiveLog,
  StartProjectSessionResponseFilterSensitiveLog
});
