// @azure/arm-machinelearning/dist-esm/src/azureMachineLearningWorkspaces.js
/*
 * Copyright (c) Microsoft Corporation.
 * Licensed under the MIT License.
 *
 * Code generated by Microsoft (R) AutoRest Code Generator.
 * Changes may cause incorrect behavior and will be lost if the code is regenerated.
 */
import { __awaiter } from "tslib";
import * as coreClient from "@azure/core-client";
import * as coreRestPipeline from "@azure/core-rest-pipeline";
import { OperationsImpl, WorkspacesImpl, UsagesImpl, VirtualMachineSizesImpl, QuotasImpl, ComputeOperationsImpl, PrivateEndpointConnectionsImpl, PrivateLinkResourcesImpl, WorkspaceConnectionsImpl, BatchEndpointsImpl, BatchDeploymentsImpl, CodeContainersImpl, CodeVersionsImpl, ComponentContainersImpl, ComponentVersionsImpl, DataContainersImpl, DataVersionsImpl, DatastoresImpl, EnvironmentContainersImpl, EnvironmentVersionsImpl, JobsImpl, ModelContainersImpl, ModelVersionsImpl, OnlineEndpointsImpl, OnlineDeploymentsImpl, SchedulesImpl, WorkspaceFeaturesImpl } from "./operations";
export class AzureMachineLearningWorkspaces extends coreClient.ServiceClient {
    /**
     * Initializes a new instance of the AzureMachineLearningWorkspaces class.
     * @param credentials Subscription credentials which uniquely identify client subscription.
     * @param subscriptionId The ID of the target subscription.
     * @param options The parameter options
     */
    constructor(credentials, subscriptionId, options) {
        var _a, _b, _c;
        if (credentials === undefined) {
            throw new Error("'credentials' cannot be null");
        }
        if (subscriptionId === undefined) {
            throw new Error("'subscriptionId' cannot be null");
        }
        // Initializing default values for options
        if (!options) {
            options = {};
        }
        const defaults = {
            requestContentType: "application/json; charset=utf-8",
            credential: credentials
        };
        const packageDetails = `azsdk-js-arm-machinelearning/2.1.1`;
        const userAgentPrefix = options.userAgentOptions && options.userAgentOptions.userAgentPrefix
            ? `${options.userAgentOptions.userAgentPrefix} ${packageDetails}`
            : `${packageDetails}`;
        const optionsWithDefaults = Object.assign(Object.assign(Object.assign({}, defaults), options), { userAgentOptions: {
                userAgentPrefix
            }, endpoint: (_b = (_a = options.endpoint) !== null && _a !== void 0 ? _a : options.baseUri) !== null && _b !== void 0 ? _b : "https://management.azure.com" });
        super(optionsWithDefaults);
        let bearerTokenAuthenticationPolicyFound = false;
        if ((options === null || options === void 0 ? void 0 : options.pipeline) && options.pipeline.getOrderedPolicies().length > 0) {
            const pipelinePolicies = options.pipeline.getOrderedPolicies();
            bearerTokenAuthenticationPolicyFound = pipelinePolicies.some((pipelinePolicy) => pipelinePolicy.name ===
                coreRestPipeline.bearerTokenAuthenticationPolicyName);
        }
        if (!options ||
            !options.pipeline ||
            options.pipeline.getOrderedPolicies().length == 0 ||
            !bearerTokenAuthenticationPolicyFound) {
            this.pipeline.removePolicy({
                name: coreRestPipeline.bearerTokenAuthenticationPolicyName
            });
            this.pipeline.addPolicy(coreRestPipeline.bearerTokenAuthenticationPolicy({
                credential: credentials,
                scopes: (_c = optionsWithDefaults.credentialScopes) !== null && _c !== void 0 ? _c : `${optionsWithDefaults.endpoint}/.default`,
                challengeCallbacks: {
                    authorizeRequestOnChallenge: coreClient.authorizeRequestOnClaimChallenge
                }
            }));
        }
        // Parameter assignments
        this.subscriptionId = subscriptionId;
        // Assigning values to Constant parameters
        this.$host = options.$host || "https://management.azure.com";
        this.apiVersion = options.apiVersion || "2022-10-01";
        this.operations = new OperationsImpl(this);
        this.workspaces = new WorkspacesImpl(this);
        this.usages = new UsagesImpl(this);
        this.virtualMachineSizes = new VirtualMachineSizesImpl(this);
        this.quotas = new QuotasImpl(this);
        this.computeOperations = new ComputeOperationsImpl(this);
        this.privateEndpointConnections = new PrivateEndpointConnectionsImpl(this);
        this.privateLinkResources = new PrivateLinkResourcesImpl(this);
        this.workspaceConnections = new WorkspaceConnectionsImpl(this);
        this.batchEndpoints = new BatchEndpointsImpl(this);
        this.batchDeployments = new BatchDeploymentsImpl(this);
        this.codeContainers = new CodeContainersImpl(this);
        this.codeVersions = new CodeVersionsImpl(this);
        this.componentContainers = new ComponentContainersImpl(this);
        this.componentVersions = new ComponentVersionsImpl(this);
        this.dataContainers = new DataContainersImpl(this);
        this.dataVersions = new DataVersionsImpl(this);
        this.datastores = new DatastoresImpl(this);
        this.environmentContainers = new EnvironmentContainersImpl(this);
        this.environmentVersions = new EnvironmentVersionsImpl(this);
        this.jobs = new JobsImpl(this);
        this.modelContainers = new ModelContainersImpl(this);
        this.modelVersions = new ModelVersionsImpl(this);
        this.onlineEndpoints = new OnlineEndpointsImpl(this);
        this.onlineDeployments = new OnlineDeploymentsImpl(this);
        this.schedules = new SchedulesImpl(this);
        this.workspaceFeatures = new WorkspaceFeaturesImpl(this);
        this.addCustomApiVersionPolicy(options.apiVersion);
    }
    /** A function that adds a policy that sets the api-version (or equivalent) to reflect the library version. */
    addCustomApiVersionPolicy(apiVersion) {
        if (!apiVersion) {
            return;
        }
        const apiVersionPolicy = {
            name: "CustomApiVersionPolicy",
            sendRequest(request, next) {
                return __awaiter(this, void 0, void 0, function* () {
                    const param = request.url.split("?");
                    if (param.length > 1) {
                        const newParams = param[1].split("&").map((item) => {
                            if (item.indexOf("api-version") > -1) {
                                return "api-version=" + apiVersion;
                            }
                            else {
                                return item;
                            }
                        });
                        request.url = param[0] + "?" + newParams.join("&");
                    }
                    return next(request);
                });
            }
        };
        this.pipeline.addPolicy(apiVersionPolicy);
    }
}
