// @azure/arm-containerinstance/dist-esm/src/models/index.js
/*
 * Copyright (c) Microsoft Corporation.
 * Licensed under the MIT License.
 *
 * Code generated by Microsoft (R) AutoRest Code Generator.
 * Changes may cause incorrect behavior and will be lost if the code is regenerated.
 */
/** Known values of {@link ContainerNetworkProtocol} that the service accepts. */
export var KnownContainerNetworkProtocol;
(function (KnownContainerNetworkProtocol) {
    /** TCP */
    KnownContainerNetworkProtocol["TCP"] = "TCP";
    /** UDP */
    KnownContainerNetworkProtocol["UDP"] = "UDP";
})(KnownContainerNetworkProtocol || (KnownContainerNetworkProtocol = {}));
/** Known values of {@link GpuSku} that the service accepts. */
export var KnownGpuSku;
(function (KnownGpuSku) {
    /** K80 */
    KnownGpuSku["K80"] = "K80";
    /** P100 */
    KnownGpuSku["P100"] = "P100";
    /** V100 */
    KnownGpuSku["V100"] = "V100";
})(KnownGpuSku || (KnownGpuSku = {}));
/** Known values of {@link Scheme} that the service accepts. */
export var KnownScheme;
(function (KnownScheme) {
    /** Http */
    KnownScheme["Http"] = "http";
    /** Https */
    KnownScheme["Https"] = "https";
})(KnownScheme || (KnownScheme = {}));
/** Known values of {@link ContainerGroupRestartPolicy} that the service accepts. */
export var KnownContainerGroupRestartPolicy;
(function (KnownContainerGroupRestartPolicy) {
    /** Always */
    KnownContainerGroupRestartPolicy["Always"] = "Always";
    /** OnFailure */
    KnownContainerGroupRestartPolicy["OnFailure"] = "OnFailure";
    /** Never */
    KnownContainerGroupRestartPolicy["Never"] = "Never";
})(KnownContainerGroupRestartPolicy || (KnownContainerGroupRestartPolicy = {}));
/** Known values of {@link ContainerGroupNetworkProtocol} that the service accepts. */
export var KnownContainerGroupNetworkProtocol;
(function (KnownContainerGroupNetworkProtocol) {
    /** TCP */
    KnownContainerGroupNetworkProtocol["TCP"] = "TCP";
    /** UDP */
    KnownContainerGroupNetworkProtocol["UDP"] = "UDP";
})(KnownContainerGroupNetworkProtocol || (KnownContainerGroupNetworkProtocol = {}));
/** Known values of {@link ContainerGroupIpAddressType} that the service accepts. */
export var KnownContainerGroupIpAddressType;
(function (KnownContainerGroupIpAddressType) {
    /** Public */
    KnownContainerGroupIpAddressType["Public"] = "Public";
    /** Private */
    KnownContainerGroupIpAddressType["Private"] = "Private";
})(KnownContainerGroupIpAddressType || (KnownContainerGroupIpAddressType = {}));
/** Known values of {@link DnsNameLabelReusePolicy} that the service accepts. */
export var KnownDnsNameLabelReusePolicy;
(function (KnownDnsNameLabelReusePolicy) {
    /** Unsecure */
    KnownDnsNameLabelReusePolicy["Unsecure"] = "Unsecure";
    /** TenantReuse */
    KnownDnsNameLabelReusePolicy["TenantReuse"] = "TenantReuse";
    /** SubscriptionReuse */
    KnownDnsNameLabelReusePolicy["SubscriptionReuse"] = "SubscriptionReuse";
    /** ResourceGroupReuse */
    KnownDnsNameLabelReusePolicy["ResourceGroupReuse"] = "ResourceGroupReuse";
    /** Noreuse */
    KnownDnsNameLabelReusePolicy["Noreuse"] = "Noreuse";
})(KnownDnsNameLabelReusePolicy || (KnownDnsNameLabelReusePolicy = {}));
/** Known values of {@link OperatingSystemTypes} that the service accepts. */
export var KnownOperatingSystemTypes;
(function (KnownOperatingSystemTypes) {
    /** Windows */
    KnownOperatingSystemTypes["Windows"] = "Windows";
    /** Linux */
    KnownOperatingSystemTypes["Linux"] = "Linux";
})(KnownOperatingSystemTypes || (KnownOperatingSystemTypes = {}));
/** Known values of {@link LogAnalyticsLogType} that the service accepts. */
export var KnownLogAnalyticsLogType;
(function (KnownLogAnalyticsLogType) {
    /** ContainerInsights */
    KnownLogAnalyticsLogType["ContainerInsights"] = "ContainerInsights";
    /** ContainerInstanceLogs */
    KnownLogAnalyticsLogType["ContainerInstanceLogs"] = "ContainerInstanceLogs";
})(KnownLogAnalyticsLogType || (KnownLogAnalyticsLogType = {}));
/** Known values of {@link ContainerGroupSku} that the service accepts. */
export var KnownContainerGroupSku;
(function (KnownContainerGroupSku) {
    /** Standard */
    KnownContainerGroupSku["Standard"] = "Standard";
    /** Dedicated */
    KnownContainerGroupSku["Dedicated"] = "Dedicated";
    /** Confidential */
    KnownContainerGroupSku["Confidential"] = "Confidential";
})(KnownContainerGroupSku || (KnownContainerGroupSku = {}));
/** Known values of {@link ContainerGroupPriority} that the service accepts. */
export var KnownContainerGroupPriority;
(function (KnownContainerGroupPriority) {
    /** Regular */
    KnownContainerGroupPriority["Regular"] = "Regular";
    /** Spot */
    KnownContainerGroupPriority["Spot"] = "Spot";
})(KnownContainerGroupPriority || (KnownContainerGroupPriority = {}));
/** Known values of {@link ContainerInstanceOperationsOrigin} that the service accepts. */
export var KnownContainerInstanceOperationsOrigin;
(function (KnownContainerInstanceOperationsOrigin) {
    /** User */
    KnownContainerInstanceOperationsOrigin["User"] = "User";
    /** System */
    KnownContainerInstanceOperationsOrigin["System"] = "System";
})(KnownContainerInstanceOperationsOrigin || (KnownContainerInstanceOperationsOrigin = {}));
