// @azure/arm-containerinstance/dist-esm/src/models/parameters.js
/*
 * Copyright (c) Microsoft Corporation.
 * Licensed under the MIT License.
 *
 * Code generated by Microsoft (R) AutoRest Code Generator.
 * Changes may cause incorrect behavior and will be lost if the code is regenerated.
 */
import { ContainerGroup as ContainerGroupMapper, Resource as ResourceMapper, ContainerExecRequest as ContainerExecRequestMapper } from "../models/mappers";
export const accept = {
    parameterPath: "accept",
    mapper: {
        defaultValue: "application/json",
        isConstant: true,
        serializedName: "Accept",
        type: {
            name: "String"
        }
    }
};
export const $host = {
    parameterPath: "$host",
    mapper: {
        serializedName: "$host",
        required: true,
        type: {
            name: "String"
        }
    },
    skipEncoding: true
};
export const subscriptionId = {
    parameterPath: "subscriptionId",
    mapper: {
        serializedName: "subscriptionId",
        required: true,
        type: {
            name: "String"
        }
    }
};
export const apiVersion = {
    parameterPath: "apiVersion",
    mapper: {
        defaultValue: "2023-05-01",
        isConstant: true,
        serializedName: "api-version",
        type: {
            name: "String"
        }
    }
};
export const resourceGroupName = {
    parameterPath: "resourceGroupName",
    mapper: {
        serializedName: "resourceGroupName",
        required: true,
        type: {
            name: "String"
        }
    }
};
export const containerGroupName = {
    parameterPath: "containerGroupName",
    mapper: {
        serializedName: "containerGroupName",
        required: true,
        type: {
            name: "String"
        }
    }
};
export const contentType = {
    parameterPath: ["options", "contentType"],
    mapper: {
        defaultValue: "application/json",
        isConstant: true,
        serializedName: "Content-Type",
        type: {
            name: "String"
        }
    }
};
export const containerGroup = {
    parameterPath: "containerGroup",
    mapper: ContainerGroupMapper
};
export const resource = {
    parameterPath: "resource",
    mapper: ResourceMapper
};
export const nextLink = {
    parameterPath: "nextLink",
    mapper: {
        serializedName: "nextLink",
        required: true,
        type: {
            name: "String"
        }
    },
    skipEncoding: true
};
export const location = {
    parameterPath: "location",
    mapper: {
        serializedName: "location",
        required: true,
        type: {
            name: "String"
        }
    }
};
export const containerName = {
    parameterPath: "containerName",
    mapper: {
        serializedName: "containerName",
        required: true,
        type: {
            name: "String"
        }
    }
};
export const tail = {
    parameterPath: ["options", "tail"],
    mapper: {
        serializedName: "tail",
        type: {
            name: "Number"
        }
    }
};
export const timestamps = {
    parameterPath: ["options", "timestamps"],
    mapper: {
        serializedName: "timestamps",
        type: {
            name: "Boolean"
        }
    }
};
export const containerExecRequest = {
    parameterPath: "containerExecRequest",
    mapper: ContainerExecRequestMapper
};
export const virtualNetworkName = {
    parameterPath: "virtualNetworkName",
    mapper: {
        serializedName: "virtualNetworkName",
        required: true,
        type: {
            name: "String"
        }
    }
};
export const subnetName = {
    parameterPath: "subnetName",
    mapper: {
        serializedName: "subnetName",
        required: true,
        type: {
            name: "String"
        }
    }
};
