// @aws-sdk/client-workdocs/dist-cjs/index.js
"use strict";
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __name = (target, value) => __defProp(target, "name", { value, configurable: true });
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// src/index.ts
var src_exports = {};
__export(src_exports, {
  AbortDocumentVersionUploadCommand: () => AbortDocumentVersionUploadCommand,
  AbortDocumentVersionUploadRequestFilterSensitiveLog: () => AbortDocumentVersionUploadRequestFilterSensitiveLog,
  ActivateUserCommand: () => ActivateUserCommand,
  ActivateUserRequestFilterSensitiveLog: () => ActivateUserRequestFilterSensitiveLog,
  ActivateUserResponseFilterSensitiveLog: () => ActivateUserResponseFilterSensitiveLog,
  ActivityFilterSensitiveLog: () => ActivityFilterSensitiveLog,
  ActivityType: () => ActivityType,
  AddResourcePermissionsCommand: () => AddResourcePermissionsCommand,
  AddResourcePermissionsRequestFilterSensitiveLog: () => AddResourcePermissionsRequestFilterSensitiveLog,
  AddResourcePermissionsResponseFilterSensitiveLog: () => AddResourcePermissionsResponseFilterSensitiveLog,
  AdditionalResponseFieldType: () => AdditionalResponseFieldType,
  BooleanEnumType: () => BooleanEnumType,
  CommentFilterSensitiveLog: () => CommentFilterSensitiveLog,
  CommentMetadataFilterSensitiveLog: () => CommentMetadataFilterSensitiveLog,
  CommentStatusType: () => CommentStatusType,
  CommentVisibilityType: () => CommentVisibilityType,
  ConcurrentModificationException: () => ConcurrentModificationException,
  ConflictingOperationException: () => ConflictingOperationException,
  ContentCategoryType: () => ContentCategoryType,
  CreateCommentCommand: () => CreateCommentCommand,
  CreateCommentRequestFilterSensitiveLog: () => CreateCommentRequestFilterSensitiveLog,
  CreateCommentResponseFilterSensitiveLog: () => CreateCommentResponseFilterSensitiveLog,
  CreateCustomMetadataCommand: () => CreateCustomMetadataCommand,
  CreateCustomMetadataRequestFilterSensitiveLog: () => CreateCustomMetadataRequestFilterSensitiveLog,
  CreateFolderCommand: () => CreateFolderCommand,
  CreateFolderRequestFilterSensitiveLog: () => CreateFolderRequestFilterSensitiveLog,
  CreateFolderResponseFilterSensitiveLog: () => CreateFolderResponseFilterSensitiveLog,
  CreateLabelsCommand: () => CreateLabelsCommand,
  CreateLabelsRequestFilterSensitiveLog: () => CreateLabelsRequestFilterSensitiveLog,
  CreateNotificationSubscriptionCommand: () => CreateNotificationSubscriptionCommand,
  CreateUserCommand: () => CreateUserCommand,
  CreateUserRequestFilterSensitiveLog: () => CreateUserRequestFilterSensitiveLog,
  CreateUserResponseFilterSensitiveLog: () => CreateUserResponseFilterSensitiveLog,
  CustomMetadataLimitExceededException: () => CustomMetadataLimitExceededException,
  DeactivateUserCommand: () => DeactivateUserCommand,
  DeactivateUserRequestFilterSensitiveLog: () => DeactivateUserRequestFilterSensitiveLog,
  DeactivatingLastSystemUserException: () => DeactivatingLastSystemUserException,
  DeleteCommentCommand: () => DeleteCommentCommand,
  DeleteCommentRequestFilterSensitiveLog: () => DeleteCommentRequestFilterSensitiveLog,
  DeleteCustomMetadataCommand: () => DeleteCustomMetadataCommand,
  DeleteCustomMetadataRequestFilterSensitiveLog: () => DeleteCustomMetadataRequestFilterSensitiveLog,
  DeleteDocumentCommand: () => DeleteDocumentCommand,
  DeleteDocumentRequestFilterSensitiveLog: () => DeleteDocumentRequestFilterSensitiveLog,
  DeleteDocumentVersionCommand: () => DeleteDocumentVersionCommand,
  DeleteDocumentVersionRequestFilterSensitiveLog: () => DeleteDocumentVersionRequestFilterSensitiveLog,
  DeleteFolderCommand: () => DeleteFolderCommand,
  DeleteFolderContentsCommand: () => DeleteFolderContentsCommand,
  DeleteFolderContentsRequestFilterSensitiveLog: () => DeleteFolderContentsRequestFilterSensitiveLog,
  DeleteFolderRequestFilterSensitiveLog: () => DeleteFolderRequestFilterSensitiveLog,
  DeleteLabelsCommand: () => DeleteLabelsCommand,
  DeleteLabelsRequestFilterSensitiveLog: () => DeleteLabelsRequestFilterSensitiveLog,
  DeleteNotificationSubscriptionCommand: () => DeleteNotificationSubscriptionCommand,
  DeleteUserCommand: () => DeleteUserCommand,
  DeleteUserRequestFilterSensitiveLog: () => DeleteUserRequestFilterSensitiveLog,
  DescribeActivitiesCommand: () => DescribeActivitiesCommand,
  DescribeActivitiesRequestFilterSensitiveLog: () => DescribeActivitiesRequestFilterSensitiveLog,
  DescribeActivitiesResponseFilterSensitiveLog: () => DescribeActivitiesResponseFilterSensitiveLog,
  DescribeCommentsCommand: () => DescribeCommentsCommand,
  DescribeCommentsRequestFilterSensitiveLog: () => DescribeCommentsRequestFilterSensitiveLog,
  DescribeCommentsResponseFilterSensitiveLog: () => DescribeCommentsResponseFilterSensitiveLog,
  DescribeDocumentVersionsCommand: () => DescribeDocumentVersionsCommand,
  DescribeDocumentVersionsRequestFilterSensitiveLog: () => DescribeDocumentVersionsRequestFilterSensitiveLog,
  DescribeDocumentVersionsResponseFilterSensitiveLog: () => DescribeDocumentVersionsResponseFilterSensitiveLog,
  DescribeFolderContentsCommand: () => DescribeFolderContentsCommand,
  DescribeFolderContentsRequestFilterSensitiveLog: () => DescribeFolderContentsRequestFilterSensitiveLog,
  DescribeFolderContentsResponseFilterSensitiveLog: () => DescribeFolderContentsResponseFilterSensitiveLog,
  DescribeGroupsCommand: () => DescribeGroupsCommand,
  DescribeGroupsRequestFilterSensitiveLog: () => DescribeGroupsRequestFilterSensitiveLog,
  DescribeNotificationSubscriptionsCommand: () => DescribeNotificationSubscriptionsCommand,
  DescribeResourcePermissionsCommand: () => DescribeResourcePermissionsCommand,
  DescribeResourcePermissionsRequestFilterSensitiveLog: () => DescribeResourcePermissionsRequestFilterSensitiveLog,
  DescribeRootFoldersCommand: () => DescribeRootFoldersCommand,
  DescribeRootFoldersRequestFilterSensitiveLog: () => DescribeRootFoldersRequestFilterSensitiveLog,
  DescribeRootFoldersResponseFilterSensitiveLog: () => DescribeRootFoldersResponseFilterSensitiveLog,
  DescribeUsersCommand: () => DescribeUsersCommand,
  DescribeUsersRequestFilterSensitiveLog: () => DescribeUsersRequestFilterSensitiveLog,
  DescribeUsersResponseFilterSensitiveLog: () => DescribeUsersResponseFilterSensitiveLog,
  DocumentLockedForCommentsException: () => DocumentLockedForCommentsException,
  DocumentMetadataFilterSensitiveLog: () => DocumentMetadataFilterSensitiveLog,
  DocumentSourceType: () => DocumentSourceType,
  DocumentStatusType: () => DocumentStatusType,
  DocumentThumbnailType: () => DocumentThumbnailType,
  DocumentVersionMetadataFilterSensitiveLog: () => DocumentVersionMetadataFilterSensitiveLog,
  DocumentVersionStatus: () => DocumentVersionStatus,
  DraftUploadOutOfSyncException: () => DraftUploadOutOfSyncException,
  EntityAlreadyExistsException: () => EntityAlreadyExistsException,
  EntityNotExistsException: () => EntityNotExistsException,
  FailedDependencyException: () => FailedDependencyException,
  FolderContentType: () => FolderContentType,
  FolderMetadataFilterSensitiveLog: () => FolderMetadataFilterSensitiveLog,
  GetCurrentUserCommand: () => GetCurrentUserCommand,
  GetCurrentUserRequestFilterSensitiveLog: () => GetCurrentUserRequestFilterSensitiveLog,
  GetCurrentUserResponseFilterSensitiveLog: () => GetCurrentUserResponseFilterSensitiveLog,
  GetDocumentCommand: () => GetDocumentCommand,
  GetDocumentPathCommand: () => GetDocumentPathCommand,
  GetDocumentPathRequestFilterSensitiveLog: () => GetDocumentPathRequestFilterSensitiveLog,
  GetDocumentPathResponseFilterSensitiveLog: () => GetDocumentPathResponseFilterSensitiveLog,
  GetDocumentRequestFilterSensitiveLog: () => GetDocumentRequestFilterSensitiveLog,
  GetDocumentResponseFilterSensitiveLog: () => GetDocumentResponseFilterSensitiveLog,
  GetDocumentVersionCommand: () => GetDocumentVersionCommand,
  GetDocumentVersionRequestFilterSensitiveLog: () => GetDocumentVersionRequestFilterSensitiveLog,
  GetDocumentVersionResponseFilterSensitiveLog: () => GetDocumentVersionResponseFilterSensitiveLog,
  GetFolderCommand: () => GetFolderCommand,
  GetFolderPathCommand: () => GetFolderPathCommand,
  GetFolderPathRequestFilterSensitiveLog: () => GetFolderPathRequestFilterSensitiveLog,
  GetFolderPathResponseFilterSensitiveLog: () => GetFolderPathResponseFilterSensitiveLog,
  GetFolderRequestFilterSensitiveLog: () => GetFolderRequestFilterSensitiveLog,
  GetFolderResponseFilterSensitiveLog: () => GetFolderResponseFilterSensitiveLog,
  GetResourcesCommand: () => GetResourcesCommand,
  GetResourcesRequestFilterSensitiveLog: () => GetResourcesRequestFilterSensitiveLog,
  GetResourcesResponseFilterSensitiveLog: () => GetResourcesResponseFilterSensitiveLog,
  IllegalUserStateException: () => IllegalUserStateException,
  InitiateDocumentVersionUploadCommand: () => InitiateDocumentVersionUploadCommand,
  InitiateDocumentVersionUploadRequestFilterSensitiveLog: () => InitiateDocumentVersionUploadRequestFilterSensitiveLog,
  InitiateDocumentVersionUploadResponseFilterSensitiveLog: () => InitiateDocumentVersionUploadResponseFilterSensitiveLog,
  InvalidArgumentException: () => InvalidArgumentException,
  InvalidCommentOperationException: () => InvalidCommentOperationException,
  InvalidOperationException: () => InvalidOperationException,
  InvalidPasswordException: () => InvalidPasswordException,
  LanguageCodeType: () => LanguageCodeType,
  LimitExceededException: () => LimitExceededException,
  LocaleType: () => LocaleType,
  NotificationOptionsFilterSensitiveLog: () => NotificationOptionsFilterSensitiveLog,
  OrderByFieldType: () => OrderByFieldType,
  OrderType: () => OrderType,
  ParticipantsFilterSensitiveLog: () => ParticipantsFilterSensitiveLog,
  PrincipalRoleType: () => PrincipalRoleType,
  PrincipalType: () => PrincipalType,
  ProhibitedStateException: () => ProhibitedStateException,
  RemoveAllResourcePermissionsCommand: () => RemoveAllResourcePermissionsCommand,
  RemoveAllResourcePermissionsRequestFilterSensitiveLog: () => RemoveAllResourcePermissionsRequestFilterSensitiveLog,
  RemoveResourcePermissionCommand: () => RemoveResourcePermissionCommand,
  RemoveResourcePermissionRequestFilterSensitiveLog: () => RemoveResourcePermissionRequestFilterSensitiveLog,
  RequestedEntityTooLargeException: () => RequestedEntityTooLargeException,
  ResourceAlreadyCheckedOutException: () => ResourceAlreadyCheckedOutException,
  ResourceCollectionType: () => ResourceCollectionType,
  ResourceMetadataFilterSensitiveLog: () => ResourceMetadataFilterSensitiveLog,
  ResourcePathComponentFilterSensitiveLog: () => ResourcePathComponentFilterSensitiveLog,
  ResourcePathFilterSensitiveLog: () => ResourcePathFilterSensitiveLog,
  ResourceSortType: () => ResourceSortType,
  ResourceStateType: () => ResourceStateType,
  ResourceType: () => ResourceType,
  ResponseItemFilterSensitiveLog: () => ResponseItemFilterSensitiveLog,
  ResponseItemType: () => ResponseItemType,
  RestoreDocumentVersionsCommand: () => RestoreDocumentVersionsCommand,
  RestoreDocumentVersionsRequestFilterSensitiveLog: () => RestoreDocumentVersionsRequestFilterSensitiveLog,
  RolePermissionType: () => RolePermissionType,
  RoleType: () => RoleType,
  SearchCollectionType: () => SearchCollectionType,
  SearchQueryScopeType: () => SearchQueryScopeType,
  SearchResourceType: () => SearchResourceType,
  SearchResourcesCommand: () => SearchResourcesCommand,
  SearchResourcesRequestFilterSensitiveLog: () => SearchResourcesRequestFilterSensitiveLog,
  SearchResourcesResponseFilterSensitiveLog: () => SearchResourcesResponseFilterSensitiveLog,
  ServiceUnavailableException: () => ServiceUnavailableException,
  ShareResultFilterSensitiveLog: () => ShareResultFilterSensitiveLog,
  ShareStatusType: () => ShareStatusType,
  SortOrder: () => SortOrder,
  StorageLimitExceededException: () => StorageLimitExceededException,
  StorageLimitWillExceedException: () => StorageLimitWillExceedException,
  StorageType: () => StorageType,
  SubscriptionProtocolType: () => SubscriptionProtocolType,
  SubscriptionType: () => SubscriptionType,
  TooManyLabelsException: () => TooManyLabelsException,
  TooManySubscriptionsException: () => TooManySubscriptionsException,
  UnauthorizedOperationException: () => UnauthorizedOperationException,
  UnauthorizedResourceAccessException: () => UnauthorizedResourceAccessException,
  UpdateDocumentCommand: () => UpdateDocumentCommand,
  UpdateDocumentRequestFilterSensitiveLog: () => UpdateDocumentRequestFilterSensitiveLog,
  UpdateDocumentVersionCommand: () => UpdateDocumentVersionCommand,
  UpdateDocumentVersionRequestFilterSensitiveLog: () => UpdateDocumentVersionRequestFilterSensitiveLog,
  UpdateFolderCommand: () => UpdateFolderCommand,
  UpdateFolderRequestFilterSensitiveLog: () => UpdateFolderRequestFilterSensitiveLog,
  UpdateUserCommand: () => UpdateUserCommand,
  UpdateUserRequestFilterSensitiveLog: () => UpdateUserRequestFilterSensitiveLog,
  UpdateUserResponseFilterSensitiveLog: () => UpdateUserResponseFilterSensitiveLog,
  UploadMetadataFilterSensitiveLog: () => UploadMetadataFilterSensitiveLog,
  UserFilterSensitiveLog: () => UserFilterSensitiveLog,
  UserFilterType: () => UserFilterType,
  UserMetadataFilterSensitiveLog: () => UserMetadataFilterSensitiveLog,
  UserSortType: () => UserSortType,
  UserStatusType: () => UserStatusType,
  UserType: () => UserType,
  WorkDocs: () => WorkDocs,
  WorkDocsClient: () => WorkDocsClient,
  WorkDocsServiceException: () => WorkDocsServiceException,
  __Client: () => import_smithy_client.Client,
  paginateDescribeActivities: () => paginateDescribeActivities,
  paginateDescribeComments: () => paginateDescribeComments,
  paginateDescribeDocumentVersions: () => paginateDescribeDocumentVersions,
  paginateDescribeFolderContents: () => paginateDescribeFolderContents,
  paginateDescribeGroups: () => paginateDescribeGroups,
  paginateDescribeNotificationSubscriptions: () => paginateDescribeNotificationSubscriptions,
  paginateDescribeResourcePermissions: () => paginateDescribeResourcePermissions,
  paginateDescribeRootFolders: () => paginateDescribeRootFolders,
  paginateDescribeUsers: () => paginateDescribeUsers,
  paginateSearchResources: () => paginateSearchResources
});
module.exports = __toCommonJS(src_exports);

// src/WorkDocsClient.ts
var import_middleware_host_header = require("@aws-sdk/middleware-host-header");
var import_middleware_logger = require("@aws-sdk/middleware-logger");
var import_middleware_recursion_detection = require("@aws-sdk/middleware-recursion-detection");
var import_middleware_user_agent = require("@aws-sdk/middleware-user-agent");
var import_config_resolver = require("@smithy/config-resolver");
var import_core = require("@smithy/core");
var import_middleware_content_length = require("@smithy/middleware-content-length");
var import_middleware_endpoint = require("@smithy/middleware-endpoint");
var import_middleware_retry = require("@smithy/middleware-retry");

var import_httpAuthSchemeProvider = require("./auth/httpAuthSchemeProvider");

// src/endpoint/EndpointParameters.ts
var resolveClientEndpointParameters = /* @__PURE__ */ __name((options) => {
  return {
    ...options,
    useDualstackEndpoint: options.useDualstackEndpoint ?? false,
    useFipsEndpoint: options.useFipsEndpoint ?? false,
    defaultSigningName: "workdocs"
  };
}, "resolveClientEndpointParameters");
var commonParams = {
  UseFIPS: { type: "builtInParams", name: "useFipsEndpoint" },
  Endpoint: { type: "builtInParams", name: "endpoint" },
  Region: { type: "builtInParams", name: "region" },
  UseDualStack: { type: "builtInParams", name: "useDualstackEndpoint" }
};

// src/WorkDocsClient.ts
var import_runtimeConfig = require("././runtimeConfig");

// src/runtimeExtensions.ts
var import_region_config_resolver = require("@aws-sdk/region-config-resolver");
var import_protocol_http = require("@smithy/protocol-http");
var import_smithy_client = require("@smithy/smithy-client");

// src/auth/httpAuthExtensionConfiguration.ts
var getHttpAuthExtensionConfiguration = /* @__PURE__ */ __name((runtimeConfig) => {
  const _httpAuthSchemes = runtimeConfig.httpAuthSchemes;
  let _httpAuthSchemeProvider = runtimeConfig.httpAuthSchemeProvider;
  let _credentials = runtimeConfig.credentials;
  return {
    setHttpAuthScheme(httpAuthScheme) {
      const index = _httpAuthSchemes.findIndex((scheme) => scheme.schemeId === httpAuthScheme.schemeId);
      if (index === -1) {
        _httpAuthSchemes.push(httpAuthScheme);
      } else {
        _httpAuthSchemes.splice(index, 1, httpAuthScheme);
      }
    },
    httpAuthSchemes() {
      return _httpAuthSchemes;
    },
    setHttpAuthSchemeProvider(httpAuthSchemeProvider) {
      _httpAuthSchemeProvider = httpAuthSchemeProvider;
    },
    httpAuthSchemeProvider() {
      return _httpAuthSchemeProvider;
    },
    setCredentials(credentials) {
      _credentials = credentials;
    },
    credentials() {
      return _credentials;
    }
  };
}, "getHttpAuthExtensionConfiguration");
var resolveHttpAuthRuntimeConfig = /* @__PURE__ */ __name((config) => {
  return {
    httpAuthSchemes: config.httpAuthSchemes(),
    httpAuthSchemeProvider: config.httpAuthSchemeProvider(),
    credentials: config.credentials()
  };
}, "resolveHttpAuthRuntimeConfig");

// src/runtimeExtensions.ts
var asPartial = /* @__PURE__ */ __name((t) => t, "asPartial");
var resolveRuntimeExtensions = /* @__PURE__ */ __name((runtimeConfig, extensions) => {
  const extensionConfiguration = {
    ...asPartial((0, import_region_config_resolver.getAwsRegionExtensionConfiguration)(runtimeConfig)),
    ...asPartial((0, import_smithy_client.getDefaultExtensionConfiguration)(runtimeConfig)),
    ...asPartial((0, import_protocol_http.getHttpHandlerExtensionConfiguration)(runtimeConfig)),
    ...asPartial(getHttpAuthExtensionConfiguration(runtimeConfig))
  };
  extensions.forEach((extension) => extension.configure(extensionConfiguration));
  return {
    ...runtimeConfig,
    ...(0, import_region_config_resolver.resolveAwsRegionExtensionConfiguration)(extensionConfiguration),
    ...(0, import_smithy_client.resolveDefaultRuntimeConfig)(extensionConfiguration),
    ...(0, import_protocol_http.resolveHttpHandlerRuntimeConfig)(extensionConfiguration),
    ...resolveHttpAuthRuntimeConfig(extensionConfiguration)
  };
}, "resolveRuntimeExtensions");

// src/WorkDocsClient.ts
var _WorkDocsClient = class _WorkDocsClient extends import_smithy_client.Client {
  constructor(...[configuration]) {
    const _config_0 = (0, import_runtimeConfig.getRuntimeConfig)(configuration || {});
    const _config_1 = resolveClientEndpointParameters(_config_0);
    const _config_2 = (0, import_config_resolver.resolveRegionConfig)(_config_1);
    const _config_3 = (0, import_middleware_endpoint.resolveEndpointConfig)(_config_2);
    const _config_4 = (0, import_middleware_retry.resolveRetryConfig)(_config_3);
    const _config_5 = (0, import_middleware_host_header.resolveHostHeaderConfig)(_config_4);
    const _config_6 = (0, import_middleware_user_agent.resolveUserAgentConfig)(_config_5);
    const _config_7 = (0, import_httpAuthSchemeProvider.resolveHttpAuthSchemeConfig)(_config_6);
    const _config_8 = resolveRuntimeExtensions(_config_7, (configuration == null ? void 0 : configuration.extensions) || []);
    super(_config_8);
    this.config = _config_8;
    this.middlewareStack.use((0, import_middleware_retry.getRetryPlugin)(this.config));
    this.middlewareStack.use((0, import_middleware_content_length.getContentLengthPlugin)(this.config));
    this.middlewareStack.use((0, import_middleware_host_header.getHostHeaderPlugin)(this.config));
    this.middlewareStack.use((0, import_middleware_logger.getLoggerPlugin)(this.config));
    this.middlewareStack.use((0, import_middleware_recursion_detection.getRecursionDetectionPlugin)(this.config));
    this.middlewareStack.use((0, import_middleware_user_agent.getUserAgentPlugin)(this.config));
    this.middlewareStack.use(
      (0, import_core.getHttpAuthSchemeEndpointRuleSetPlugin)(this.config, {
        httpAuthSchemeParametersProvider: this.getDefaultHttpAuthSchemeParametersProvider(),
        identityProviderConfigProvider: this.getIdentityProviderConfigProvider()
      })
    );
    this.middlewareStack.use((0, import_core.getHttpSigningPlugin)(this.config));
  }
  /**
   * Destroy underlying resources, like sockets. It's usually not necessary to do this.
   * However in Node.js, it's best to explicitly shut down the client's agent when it is no longer needed.
   * Otherwise, sockets might stay open for quite a long time before the server terminates them.
   */
  destroy() {
    super.destroy();
  }
  getDefaultHttpAuthSchemeParametersProvider() {
    return import_httpAuthSchemeProvider.defaultWorkDocsHttpAuthSchemeParametersProvider;
  }
  getIdentityProviderConfigProvider() {
    return async (config) => new import_core.DefaultIdentityProviderConfig({
      "aws.auth#sigv4": config.credentials
    });
  }
};
__name(_WorkDocsClient, "WorkDocsClient");
var WorkDocsClient = _WorkDocsClient;

// src/WorkDocs.ts


// src/commands/AbortDocumentVersionUploadCommand.ts

var import_middleware_serde = require("@smithy/middleware-serde");

var import_types = require("@smithy/types");

// src/models/models_0.ts


// src/models/WorkDocsServiceException.ts

var _WorkDocsServiceException = class _WorkDocsServiceException extends import_smithy_client.ServiceException {
  /**
   * @internal
   */
  constructor(options) {
    super(options);
    Object.setPrototypeOf(this, _WorkDocsServiceException.prototype);
  }
};
__name(_WorkDocsServiceException, "WorkDocsServiceException");
var WorkDocsServiceException = _WorkDocsServiceException;

// src/models/models_0.ts
var _ConcurrentModificationException = class _ConcurrentModificationException extends WorkDocsServiceException {
  /**
   * @internal
   */
  constructor(opts) {
    super({
      name: "ConcurrentModificationException",
      $fault: "client",
      ...opts
    });
    this.name = "ConcurrentModificationException";
    this.$fault = "client";
    Object.setPrototypeOf(this, _ConcurrentModificationException.prototype);
    this.Message = opts.Message;
  }
};
__name(_ConcurrentModificationException, "ConcurrentModificationException");
var ConcurrentModificationException = _ConcurrentModificationException;
var _EntityNotExistsException = class _EntityNotExistsException extends WorkDocsServiceException {
  /**
   * @internal
   */
  constructor(opts) {
    super({
      name: "EntityNotExistsException",
      $fault: "client",
      ...opts
    });
    this.name = "EntityNotExistsException";
    this.$fault = "client";
    Object.setPrototypeOf(this, _EntityNotExistsException.prototype);
    this.Message = opts.Message;
    this.EntityIds = opts.EntityIds;
  }
};
__name(_EntityNotExistsException, "EntityNotExistsException");
var EntityNotExistsException = _EntityNotExistsException;
var _FailedDependencyException = class _FailedDependencyException extends WorkDocsServiceException {
  /**
   * @internal
   */
  constructor(opts) {
    super({
      name: "FailedDependencyException",
      $fault: "client",
      ...opts
    });
    this.name = "FailedDependencyException";
    this.$fault = "client";
    Object.setPrototypeOf(this, _FailedDependencyException.prototype);
    this.Message = opts.Message;
  }
};
__name(_FailedDependencyException, "FailedDependencyException");
var FailedDependencyException = _FailedDependencyException;
var _ProhibitedStateException = class _ProhibitedStateException extends WorkDocsServiceException {
  /**
   * @internal
   */
  constructor(opts) {
    super({
      name: "ProhibitedStateException",
      $fault: "client",
      ...opts
    });
    this.name = "ProhibitedStateException";
    this.$fault = "client";
    Object.setPrototypeOf(this, _ProhibitedStateException.prototype);
    this.Message = opts.Message;
  }
};
__name(_ProhibitedStateException, "ProhibitedStateException");
var ProhibitedStateException = _ProhibitedStateException;
var _ServiceUnavailableException = class _ServiceUnavailableException extends WorkDocsServiceException {
  /**
   * @internal
   */
  constructor(opts) {
    super({
      name: "ServiceUnavailableException",
      $fault: "server",
      ...opts
    });
    this.name = "ServiceUnavailableException";
    this.$fault = "server";
    Object.setPrototypeOf(this, _ServiceUnavailableException.prototype);
    this.Message = opts.Message;
  }
};
__name(_ServiceUnavailableException, "ServiceUnavailableException");
var ServiceUnavailableException = _ServiceUnavailableException;
var _UnauthorizedOperationException = class _UnauthorizedOperationException extends WorkDocsServiceException {
  /**
   * @internal
   */
  constructor(opts) {
    super({
      name: "UnauthorizedOperationException",
      $fault: "client",
      ...opts
    });
    this.name = "UnauthorizedOperationException";
    this.$fault = "client";
    Object.setPrototypeOf(this, _UnauthorizedOperationException.prototype);
    this.Message = opts.Message;
    this.Code = opts.Code;
  }
};
__name(_UnauthorizedOperationException, "UnauthorizedOperationException");
var UnauthorizedOperationException = _UnauthorizedOperationException;
var _UnauthorizedResourceAccessException = class _UnauthorizedResourceAccessException extends WorkDocsServiceException {
  /**
   * @internal
   */
  constructor(opts) {
    super({
      name: "UnauthorizedResourceAccessException",
      $fault: "client",
      ...opts
    });
    this.name = "UnauthorizedResourceAccessException";
    this.$fault = "client";
    Object.setPrototypeOf(this, _UnauthorizedResourceAccessException.prototype);
    this.Message = opts.Message;
  }
};
__name(_UnauthorizedResourceAccessException, "UnauthorizedResourceAccessException");
var UnauthorizedResourceAccessException = _UnauthorizedResourceAccessException;
var LocaleType = {
  DE: "de",
  DEFAULT: "default",
  EN: "en",
  ES: "es",
  FR: "fr",
  JA: "ja",
  KO: "ko",
  PT_BR: "pt_BR",
  RU: "ru",
  ZH_CN: "zh_CN",
  ZH_TW: "zh_TW"
};
var UserStatusType = {
  ACTIVE: "ACTIVE",
  INACTIVE: "INACTIVE",
  PENDING: "PENDING"
};
var StorageType = {
  QUOTA: "QUOTA",
  UNLIMITED: "UNLIMITED"
};
var UserType = {
  ADMIN: "ADMIN",
  MINIMALUSER: "MINIMALUSER",
  POWERUSER: "POWERUSER",
  USER: "USER",
  WORKSPACESUSER: "WORKSPACESUSER"
};
var CommentStatusType = {
  DELETED: "DELETED",
  DRAFT: "DRAFT",
  PUBLISHED: "PUBLISHED"
};
var ResourceType = {
  DOCUMENT: "DOCUMENT",
  FOLDER: "FOLDER"
};
var ActivityType = {
  DOCUMENT_ANNOTATION_ADDED: "DOCUMENT_ANNOTATION_ADDED",
  DOCUMENT_ANNOTATION_DELETED: "DOCUMENT_ANNOTATION_DELETED",
  DOCUMENT_CHECKED_IN: "DOCUMENT_CHECKED_IN",
  DOCUMENT_CHECKED_OUT: "DOCUMENT_CHECKED_OUT",
  DOCUMENT_COMMENT_ADDED: "DOCUMENT_COMMENT_ADDED",
  DOCUMENT_COMMENT_DELETED: "DOCUMENT_COMMENT_DELETED",
  DOCUMENT_MOVED: "DOCUMENT_MOVED",
  DOCUMENT_RECYCLED: "DOCUMENT_RECYCLED",
  DOCUMENT_RENAMED: "DOCUMENT_RENAMED",
  DOCUMENT_RESTORED: "DOCUMENT_RESTORED",
  DOCUMENT_REVERTED: "DOCUMENT_REVERTED",
  DOCUMENT_SHAREABLE_LINK_CREATED: "DOCUMENT_SHAREABLE_LINK_CREATED",
  DOCUMENT_SHAREABLE_LINK_PERMISSION_CHANGED: "DOCUMENT_SHAREABLE_LINK_PERMISSION_CHANGED",
  DOCUMENT_SHAREABLE_LINK_REMOVED: "DOCUMENT_SHAREABLE_LINK_REMOVED",
  DOCUMENT_SHARED: "DOCUMENT_SHARED",
  DOCUMENT_SHARE_PERMISSION_CHANGED: "DOCUMENT_SHARE_PERMISSION_CHANGED",
  DOCUMENT_UNSHARED: "DOCUMENT_UNSHARED",
  DOCUMENT_VERSION_DELETED: "DOCUMENT_VERSION_DELETED",
  DOCUMENT_VERSION_DOWNLOADED: "DOCUMENT_VERSION_DOWNLOADED",
  DOCUMENT_VERSION_UPLOADED: "DOCUMENT_VERSION_UPLOADED",
  DOCUMENT_VERSION_VIEWED: "DOCUMENT_VERSION_VIEWED",
  FOLDER_CREATED: "FOLDER_CREATED",
  FOLDER_DELETED: "FOLDER_DELETED",
  FOLDER_MOVED: "FOLDER_MOVED",
  FOLDER_RECYCLED: "FOLDER_RECYCLED",
  FOLDER_RENAMED: "FOLDER_RENAMED",
  FOLDER_RESTORED: "FOLDER_RESTORED",
  FOLDER_SHAREABLE_LINK_CREATED: "FOLDER_SHAREABLE_LINK_CREATED",
  FOLDER_SHAREABLE_LINK_PERMISSION_CHANGED: "FOLDER_SHAREABLE_LINK_PERMISSION_CHANGED",
  FOLDER_SHAREABLE_LINK_REMOVED: "FOLDER_SHAREABLE_LINK_REMOVED",
  FOLDER_SHARED: "FOLDER_SHARED",
  FOLDER_SHARE_PERMISSION_CHANGED: "FOLDER_SHARE_PERMISSION_CHANGED",
  FOLDER_UNSHARED: "FOLDER_UNSHARED"
};
var AdditionalResponseFieldType = {
  WEBURL: "WEBURL"
};
var RoleType = {
  CONTRIBUTOR: "CONTRIBUTOR",
  COOWNER: "COOWNER",
  OWNER: "OWNER",
  VIEWER: "VIEWER"
};
var PrincipalType = {
  ANONYMOUS: "ANONYMOUS",
  GROUP: "GROUP",
  INVITE: "INVITE",
  ORGANIZATION: "ORGANIZATION",
  USER: "USER"
};
var ShareStatusType = {
  FAILURE: "FAILURE",
  SUCCESS: "SUCCESS"
};
var CommentVisibilityType = {
  PRIVATE: "PRIVATE",
  PUBLIC: "PUBLIC"
};
var _DocumentLockedForCommentsException = class _DocumentLockedForCommentsException extends WorkDocsServiceException {
  /**
   * @internal
   */
  constructor(opts) {
    super({
      name: "DocumentLockedForCommentsException",
      $fault: "client",
      ...opts
    });
    this.name = "DocumentLockedForCommentsException";
    this.$fault = "client";
    Object.setPrototypeOf(this, _DocumentLockedForCommentsException.prototype);
    this.Message = opts.Message;
  }
};
__name(_DocumentLockedForCommentsException, "DocumentLockedForCommentsException");
var DocumentLockedForCommentsException = _DocumentLockedForCommentsException;
var _InvalidCommentOperationException = class _InvalidCommentOperationException extends WorkDocsServiceException {
  /**
   * @internal
   */
  constructor(opts) {
    super({
      name: "InvalidCommentOperationException",
      $fault: "client",
      ...opts
    });
    this.name = "InvalidCommentOperationException";
    this.$fault = "client";
    Object.setPrototypeOf(this, _InvalidCommentOperationException.prototype);
    this.Message = opts.Message;
  }
};
__name(_InvalidCommentOperationException, "InvalidCommentOperationException");
var InvalidCommentOperationException = _InvalidCommentOperationException;
var _CustomMetadataLimitExceededException = class _CustomMetadataLimitExceededException extends WorkDocsServiceException {
  /**
   * @internal
   */
  constructor(opts) {
    super({
      name: "CustomMetadataLimitExceededException",
      $fault: "client",
      ...opts
    });
    this.name = "CustomMetadataLimitExceededException";
    this.$fault = "client";
    Object.setPrototypeOf(this, _CustomMetadataLimitExceededException.prototype);
    this.Message = opts.Message;
  }
};
__name(_CustomMetadataLimitExceededException, "CustomMetadataLimitExceededException");
var CustomMetadataLimitExceededException = _CustomMetadataLimitExceededException;
var _ConflictingOperationException = class _ConflictingOperationException extends WorkDocsServiceException {
  /**
   * @internal
   */
  constructor(opts) {
    super({
      name: "ConflictingOperationException",
      $fault: "client",
      ...opts
    });
    this.name = "ConflictingOperationException";
    this.$fault = "client";
    Object.setPrototypeOf(this, _ConflictingOperationException.prototype);
    this.Message = opts.Message;
  }
};
__name(_ConflictingOperationException, "ConflictingOperationException");
var ConflictingOperationException = _ConflictingOperationException;
var ResourceStateType = {
  ACTIVE: "ACTIVE",
  RECYCLED: "RECYCLED",
  RECYCLING: "RECYCLING",
  RESTORING: "RESTORING"
};
var _EntityAlreadyExistsException = class _EntityAlreadyExistsException extends WorkDocsServiceException {
  /**
   * @internal
   */
  constructor(opts) {
    super({
      name: "EntityAlreadyExistsException",
      $fault: "client",
      ...opts
    });
    this.name = "EntityAlreadyExistsException";
    this.$fault = "client";
    Object.setPrototypeOf(this, _EntityAlreadyExistsException.prototype);
    this.Message = opts.Message;
  }
};
__name(_EntityAlreadyExistsException, "EntityAlreadyExistsException");
var EntityAlreadyExistsException = _EntityAlreadyExistsException;
var _LimitExceededException = class _LimitExceededException extends WorkDocsServiceException {
  /**
   * @internal
   */
  constructor(opts) {
    super({
      name: "LimitExceededException",
      $fault: "client",
      ...opts
    });
    this.name = "LimitExceededException";
    this.$fault = "client";
    Object.setPrototypeOf(this, _LimitExceededException.prototype);
    this.Message = opts.Message;
  }
};
__name(_LimitExceededException, "LimitExceededException");
var LimitExceededException = _LimitExceededException;
var _TooManyLabelsException = class _TooManyLabelsException extends WorkDocsServiceException {
  /**
   * @internal
   */
  constructor(opts) {
    super({
      name: "TooManyLabelsException",
      $fault: "client",
      ...opts
    });
    this.name = "TooManyLabelsException";
    this.$fault = "client";
    Object.setPrototypeOf(this, _TooManyLabelsException.prototype);
    this.Message = opts.Message;
  }
};
__name(_TooManyLabelsException, "TooManyLabelsException");
var TooManyLabelsException = _TooManyLabelsException;
var SubscriptionProtocolType = {
  HTTPS: "HTTPS",
  SQS: "SQS"
};
var SubscriptionType = {
  ALL: "ALL"
};
var _InvalidArgumentException = class _InvalidArgumentException extends WorkDocsServiceException {
  /**
   * @internal
   */
  constructor(opts) {
    super({
      name: "InvalidArgumentException",
      $fault: "client",
      ...opts
    });
    this.name = "InvalidArgumentException";
    this.$fault = "client";
    Object.setPrototypeOf(this, _InvalidArgumentException.prototype);
    this.Message = opts.Message;
  }
};
__name(_InvalidArgumentException, "InvalidArgumentException");
var InvalidArgumentException = _InvalidArgumentException;
var _TooManySubscriptionsException = class _TooManySubscriptionsException extends WorkDocsServiceException {
  /**
   * @internal
   */
  constructor(opts) {
    super({
      name: "TooManySubscriptionsException",
      $fault: "client",
      ...opts
    });
    this.name = "TooManySubscriptionsException";
    this.$fault = "client";
    Object.setPrototypeOf(this, _TooManySubscriptionsException.prototype);
    this.Message = opts.Message;
  }
};
__name(_TooManySubscriptionsException, "TooManySubscriptionsException");
var TooManySubscriptionsException = _TooManySubscriptionsException;
var _InvalidOperationException = class _InvalidOperationException extends WorkDocsServiceException {
  /**
   * @internal
   */
  constructor(opts) {
    super({
      name: "InvalidOperationException",
      $fault: "client",
      ...opts
    });
    this.name = "InvalidOperationException";
    this.$fault = "client";
    Object.setPrototypeOf(this, _InvalidOperationException.prototype);
    this.Message = opts.Message;
  }
};
__name(_InvalidOperationException, "InvalidOperationException");
var InvalidOperationException = _InvalidOperationException;
var DocumentSourceType = {
  ORIGINAL: "ORIGINAL",
  WITH_COMMENTS: "WITH_COMMENTS"
};
var DocumentStatusType = {
  ACTIVE: "ACTIVE",
  INITIALIZED: "INITIALIZED"
};
var DocumentThumbnailType = {
  LARGE: "LARGE",
  SMALL: "SMALL",
  SMALL_HQ: "SMALL_HQ"
};
var _InvalidPasswordException = class _InvalidPasswordException extends WorkDocsServiceException {
  /**
   * @internal
   */
  constructor(opts) {
    super({
      name: "InvalidPasswordException",
      $fault: "client",
      ...opts
    });
    this.name = "InvalidPasswordException";
    this.$fault = "client";
    Object.setPrototypeOf(this, _InvalidPasswordException.prototype);
    this.Message = opts.Message;
  }
};
__name(_InvalidPasswordException, "InvalidPasswordException");
var InvalidPasswordException = _InvalidPasswordException;
var OrderType = {
  ASCENDING: "ASCENDING",
  DESCENDING: "DESCENDING"
};
var ResourceSortType = {
  DATE: "DATE",
  NAME: "NAME"
};
var FolderContentType = {
  ALL: "ALL",
  DOCUMENT: "DOCUMENT",
  FOLDER: "FOLDER"
};
var RolePermissionType = {
  DIRECT: "DIRECT",
  INHERITED: "INHERITED"
};
var UserFilterType = {
  ACTIVE_PENDING: "ACTIVE_PENDING",
  ALL: "ALL"
};
var UserSortType = {
  FULL_NAME: "FULL_NAME",
  STORAGE_LIMIT: "STORAGE_LIMIT",
  STORAGE_USED: "STORAGE_USED",
  USER_NAME: "USER_NAME",
  USER_STATUS: "USER_STATUS"
};
var _RequestedEntityTooLargeException = class _RequestedEntityTooLargeException extends WorkDocsServiceException {
  /**
   * @internal
   */
  constructor(opts) {
    super({
      name: "RequestedEntityTooLargeException",
      $fault: "client",
      ...opts
    });
    this.name = "RequestedEntityTooLargeException";
    this.$fault = "client";
    Object.setPrototypeOf(this, _RequestedEntityTooLargeException.prototype);
    this.Message = opts.Message;
  }
};
__name(_RequestedEntityTooLargeException, "RequestedEntityTooLargeException");
var RequestedEntityTooLargeException = _RequestedEntityTooLargeException;
var ResourceCollectionType = {
  SHARED_WITH_ME: "SHARED_WITH_ME"
};
var _DraftUploadOutOfSyncException = class _DraftUploadOutOfSyncException extends WorkDocsServiceException {
  /**
   * @internal
   */
  constructor(opts) {
    super({
      name: "DraftUploadOutOfSyncException",
      $fault: "client",
      ...opts
    });
    this.name = "DraftUploadOutOfSyncException";
    this.$fault = "client";
    Object.setPrototypeOf(this, _DraftUploadOutOfSyncException.prototype);
    this.Message = opts.Message;
  }
};
__name(_DraftUploadOutOfSyncException, "DraftUploadOutOfSyncException");
var DraftUploadOutOfSyncException = _DraftUploadOutOfSyncException;
var _ResourceAlreadyCheckedOutException = class _ResourceAlreadyCheckedOutException extends WorkDocsServiceException {
  /**
   * @internal
   */
  constructor(opts) {
    super({
      name: "ResourceAlreadyCheckedOutException",
      $fault: "client",
      ...opts
    });
    this.name = "ResourceAlreadyCheckedOutException";
    this.$fault = "client";
    Object.setPrototypeOf(this, _ResourceAlreadyCheckedOutException.prototype);
    this.Message = opts.Message;
  }
};
__name(_ResourceAlreadyCheckedOutException, "ResourceAlreadyCheckedOutException");
var ResourceAlreadyCheckedOutException = _ResourceAlreadyCheckedOutException;
var _StorageLimitExceededException = class _StorageLimitExceededException extends WorkDocsServiceException {
  /**
   * @internal
   */
  constructor(opts) {
    super({
      name: "StorageLimitExceededException",
      $fault: "client",
      ...opts
    });
    this.name = "StorageLimitExceededException";
    this.$fault = "client";
    Object.setPrototypeOf(this, _StorageLimitExceededException.prototype);
    this.Message = opts.Message;
  }
};
__name(_StorageLimitExceededException, "StorageLimitExceededException");
var StorageLimitExceededException = _StorageLimitExceededException;
var _StorageLimitWillExceedException = class _StorageLimitWillExceedException extends WorkDocsServiceException {
  /**
   * @internal
   */
  constructor(opts) {
    super({
      name: "StorageLimitWillExceedException",
      $fault: "client",
      ...opts
    });
    this.name = "StorageLimitWillExceedException";
    this.$fault = "client";
    Object.setPrototypeOf(this, _StorageLimitWillExceedException.prototype);
    this.Message = opts.Message;
  }
};
__name(_StorageLimitWillExceedException, "StorageLimitWillExceedException");
var StorageLimitWillExceedException = _StorageLimitWillExceedException;
var ContentCategoryType = {
  AUDIO: "AUDIO",
  DOCUMENT: "DOCUMENT",
  IMAGE: "IMAGE",
  OTHER: "OTHER",
  PDF: "PDF",
  PRESENTATION: "PRESENTATION",
  SOURCE_CODE: "SOURCE_CODE",
  SPREADSHEET: "SPREADSHEET",
  VIDEO: "VIDEO"
};
var PrincipalRoleType = {
  CONTRIBUTOR: "CONTRIBUTOR",
  COOWNER: "COOWNER",
  OWNER: "OWNER",
  VIEWER: "VIEWER"
};
var SearchResourceType = {
  COMMENT: "COMMENT",
  DOCUMENT: "DOCUMENT",
  DOCUMENT_VERSION: "DOCUMENT_VERSION",
  FOLDER: "FOLDER"
};
var SearchCollectionType = {
  OWNED: "OWNED",
  SHARED_WITH_ME: "SHARED_WITH_ME"
};
var LanguageCodeType = {
  AR: "AR",
  BG: "BG",
  BN: "BN",
  CS: "CS",
  DA: "DA",
  DE: "DE",
  DEFAULT: "DEFAULT",
  EL: "EL",
  EN: "EN",
  ES: "ES",
  FA: "FA",
  FI: "FI",
  FR: "FR",
  HI: "HI",
  HU: "HU",
  ID: "ID",
  IT: "IT",
  JA: "JA",
  KO: "KO",
  LT: "LT",
  LV: "LV",
  NL: "NL",
  NO: "NO",
  PT: "PT",
  RO: "RO",
  RU: "RU",
  SV: "SV",
  SW: "SW",
  TH: "TH",
  TR: "TR",
  ZH: "ZH"
};
var OrderByFieldType = {
  CREATED_TIMESTAMP: "CREATED_TIMESTAMP",
  MODIFIED_TIMESTAMP: "MODIFIED_TIMESTAMP",
  NAME: "NAME",
  RELEVANCE: "RELEVANCE",
  SIZE: "SIZE"
};
var SortOrder = {
  ASC: "ASC",
  DESC: "DESC"
};
var SearchQueryScopeType = {
  CONTENT: "CONTENT",
  NAME: "NAME"
};
var ResponseItemType = {
  COMMENT: "COMMENT",
  DOCUMENT: "DOCUMENT",
  DOCUMENT_VERSION: "DOCUMENT_VERSION",
  FOLDER: "FOLDER"
};
var DocumentVersionStatus = {
  ACTIVE: "ACTIVE"
};
var _DeactivatingLastSystemUserException = class _DeactivatingLastSystemUserException extends WorkDocsServiceException {
  /**
   * @internal
   */
  constructor(opts) {
    super({
      name: "DeactivatingLastSystemUserException",
      $fault: "client",
      ...opts
    });
    this.name = "DeactivatingLastSystemUserException";
    this.$fault = "client";
    Object.setPrototypeOf(this, _DeactivatingLastSystemUserException.prototype);
    this.Message = opts.Message;
    this.Code = opts.Code;
  }
};
__name(_DeactivatingLastSystemUserException, "DeactivatingLastSystemUserException");
var DeactivatingLastSystemUserException = _DeactivatingLastSystemUserException;
var _IllegalUserStateException = class _IllegalUserStateException extends WorkDocsServiceException {
  /**
   * @internal
   */
  constructor(opts) {
    super({
      name: "IllegalUserStateException",
      $fault: "client",
      ...opts
    });
    this.name = "IllegalUserStateException";
    this.$fault = "client";
    Object.setPrototypeOf(this, _IllegalUserStateException.prototype);
    this.Message = opts.Message;
  }
};
__name(_IllegalUserStateException, "IllegalUserStateException");
var IllegalUserStateException = _IllegalUserStateException;
var BooleanEnumType = {
  FALSE: "FALSE",
  TRUE: "TRUE"
};
var AbortDocumentVersionUploadRequestFilterSensitiveLog = /* @__PURE__ */ __name((obj) => ({
  ...obj,
  ...obj.AuthenticationToken && { AuthenticationToken: import_smithy_client.SENSITIVE_STRING }
}), "AbortDocumentVersionUploadRequestFilterSensitiveLog");
var ActivateUserRequestFilterSensitiveLog = /* @__PURE__ */ __name((obj) => ({
  ...obj,
  ...obj.AuthenticationToken && { AuthenticationToken: import_smithy_client.SENSITIVE_STRING }
}), "ActivateUserRequestFilterSensitiveLog");
var UserFilterSensitiveLog = /* @__PURE__ */ __name((obj) => ({
  ...obj,
  ...obj.Username && { Username: import_smithy_client.SENSITIVE_STRING },
  ...obj.EmailAddress && { EmailAddress: import_smithy_client.SENSITIVE_STRING },
  ...obj.GivenName && { GivenName: import_smithy_client.SENSITIVE_STRING },
  ...obj.Surname && { Surname: import_smithy_client.SENSITIVE_STRING }
}), "UserFilterSensitiveLog");
var ActivateUserResponseFilterSensitiveLog = /* @__PURE__ */ __name((obj) => ({
  ...obj,
  ...obj.User && { User: UserFilterSensitiveLog(obj.User) }
}), "ActivateUserResponseFilterSensitiveLog");
var CommentMetadataFilterSensitiveLog = /* @__PURE__ */ __name((obj) => ({
  ...obj,
  ...obj.Contributor && { Contributor: UserFilterSensitiveLog(obj.Contributor) }
}), "CommentMetadataFilterSensitiveLog");
var UserMetadataFilterSensitiveLog = /* @__PURE__ */ __name((obj) => ({
  ...obj,
  ...obj.Username && { Username: import_smithy_client.SENSITIVE_STRING },
  ...obj.GivenName && { GivenName: import_smithy_client.SENSITIVE_STRING },
  ...obj.Surname && { Surname: import_smithy_client.SENSITIVE_STRING },
  ...obj.EmailAddress && { EmailAddress: import_smithy_client.SENSITIVE_STRING }
}), "UserMetadataFilterSensitiveLog");
var ResourceMetadataFilterSensitiveLog = /* @__PURE__ */ __name((obj) => ({
  ...obj,
  ...obj.Name && { Name: import_smithy_client.SENSITIVE_STRING },
  ...obj.OriginalName && { OriginalName: import_smithy_client.SENSITIVE_STRING },
  ...obj.Owner && { Owner: UserMetadataFilterSensitiveLog(obj.Owner) }
}), "ResourceMetadataFilterSensitiveLog");
var ParticipantsFilterSensitiveLog = /* @__PURE__ */ __name((obj) => ({
  ...obj,
  ...obj.Users && { Users: obj.Users.map((item) => UserMetadataFilterSensitiveLog(item)) }
}), "ParticipantsFilterSensitiveLog");
var ActivityFilterSensitiveLog = /* @__PURE__ */ __name((obj) => ({
  ...obj,
  ...obj.Initiator && { Initiator: UserMetadataFilterSensitiveLog(obj.Initiator) },
  ...obj.Participants && { Participants: ParticipantsFilterSensitiveLog(obj.Participants) },
  ...obj.ResourceMetadata && { ResourceMetadata: ResourceMetadataFilterSensitiveLog(obj.ResourceMetadata) },
  ...obj.OriginalParent && { OriginalParent: ResourceMetadataFilterSensitiveLog(obj.OriginalParent) },
  ...obj.CommentMetadata && { CommentMetadata: CommentMetadataFilterSensitiveLog(obj.CommentMetadata) }
}), "ActivityFilterSensitiveLog");
var NotificationOptionsFilterSensitiveLog = /* @__PURE__ */ __name((obj) => ({
  ...obj,
  ...obj.EmailMessage && { EmailMessage: import_smithy_client.SENSITIVE_STRING }
}), "NotificationOptionsFilterSensitiveLog");
var AddResourcePermissionsRequestFilterSensitiveLog = /* @__PURE__ */ __name((obj) => ({
  ...obj,
  ...obj.AuthenticationToken && { AuthenticationToken: import_smithy_client.SENSITIVE_STRING },
  ...obj.NotificationOptions && {
    NotificationOptions: NotificationOptionsFilterSensitiveLog(obj.NotificationOptions)
  }
}), "AddResourcePermissionsRequestFilterSensitiveLog");
var ShareResultFilterSensitiveLog = /* @__PURE__ */ __name((obj) => ({
  ...obj,
  ...obj.StatusMessage && { StatusMessage: import_smithy_client.SENSITIVE_STRING }
}), "ShareResultFilterSensitiveLog");
var AddResourcePermissionsResponseFilterSensitiveLog = /* @__PURE__ */ __name((obj) => ({
  ...obj,
  ...obj.ShareResults && { ShareResults: obj.ShareResults.map((item) => ShareResultFilterSensitiveLog(item)) }
}), "AddResourcePermissionsResponseFilterSensitiveLog");
var CreateCommentRequestFilterSensitiveLog = /* @__PURE__ */ __name((obj) => ({
  ...obj,
  ...obj.AuthenticationToken && { AuthenticationToken: import_smithy_client.SENSITIVE_STRING },
  ...obj.Text && { Text: import_smithy_client.SENSITIVE_STRING }
}), "CreateCommentRequestFilterSensitiveLog");
var CommentFilterSensitiveLog = /* @__PURE__ */ __name((obj) => ({
  ...obj,
  ...obj.Text && { Text: import_smithy_client.SENSITIVE_STRING },
  ...obj.Contributor && { Contributor: UserFilterSensitiveLog(obj.Contributor) }
}), "CommentFilterSensitiveLog");
var CreateCommentResponseFilterSensitiveLog = /* @__PURE__ */ __name((obj) => ({
  ...obj,
  ...obj.Comment && { Comment: CommentFilterSensitiveLog(obj.Comment) }
}), "CreateCommentResponseFilterSensitiveLog");
var CreateCustomMetadataRequestFilterSensitiveLog = /* @__PURE__ */ __name((obj) => ({
  ...obj,
  ...obj.AuthenticationToken && { AuthenticationToken: import_smithy_client.SENSITIVE_STRING }
}), "CreateCustomMetadataRequestFilterSensitiveLog");
var CreateFolderRequestFilterSensitiveLog = /* @__PURE__ */ __name((obj) => ({
  ...obj,
  ...obj.AuthenticationToken && { AuthenticationToken: import_smithy_client.SENSITIVE_STRING },
  ...obj.Name && { Name: import_smithy_client.SENSITIVE_STRING }
}), "CreateFolderRequestFilterSensitiveLog");
var FolderMetadataFilterSensitiveLog = /* @__PURE__ */ __name((obj) => ({
  ...obj,
  ...obj.Name && { Name: import_smithy_client.SENSITIVE_STRING }
}), "FolderMetadataFilterSensitiveLog");
var CreateFolderResponseFilterSensitiveLog = /* @__PURE__ */ __name((obj) => ({
  ...obj,
  ...obj.Metadata && { Metadata: FolderMetadataFilterSensitiveLog(obj.Metadata) }
}), "CreateFolderResponseFilterSensitiveLog");
var CreateLabelsRequestFilterSensitiveLog = /* @__PURE__ */ __name((obj) => ({
  ...obj,
  ...obj.AuthenticationToken && { AuthenticationToken: import_smithy_client.SENSITIVE_STRING }
}), "CreateLabelsRequestFilterSensitiveLog");
var CreateUserRequestFilterSensitiveLog = /* @__PURE__ */ __name((obj) => ({
  ...obj,
  ...obj.Username && { Username: import_smithy_client.SENSITIVE_STRING },
  ...obj.EmailAddress && { EmailAddress: import_smithy_client.SENSITIVE_STRING },
  ...obj.GivenName && { GivenName: import_smithy_client.SENSITIVE_STRING },
  ...obj.Surname && { Surname: import_smithy_client.SENSITIVE_STRING },
  ...obj.Password && { Password: import_smithy_client.SENSITIVE_STRING },
  ...obj.AuthenticationToken && { AuthenticationToken: import_smithy_client.SENSITIVE_STRING }
}), "CreateUserRequestFilterSensitiveLog");
var CreateUserResponseFilterSensitiveLog = /* @__PURE__ */ __name((obj) => ({
  ...obj,
  ...obj.User && { User: UserFilterSensitiveLog(obj.User) }
}), "CreateUserResponseFilterSensitiveLog");
var DeactivateUserRequestFilterSensitiveLog = /* @__PURE__ */ __name((obj) => ({
  ...obj,
  ...obj.AuthenticationToken && { AuthenticationToken: import_smithy_client.SENSITIVE_STRING }
}), "DeactivateUserRequestFilterSensitiveLog");
var DeleteCommentRequestFilterSensitiveLog = /* @__PURE__ */ __name((obj) => ({
  ...obj,
  ...obj.AuthenticationToken && { AuthenticationToken: import_smithy_client.SENSITIVE_STRING }
}), "DeleteCommentRequestFilterSensitiveLog");
var DeleteCustomMetadataRequestFilterSensitiveLog = /* @__PURE__ */ __name((obj) => ({
  ...obj,
  ...obj.AuthenticationToken && { AuthenticationToken: import_smithy_client.SENSITIVE_STRING }
}), "DeleteCustomMetadataRequestFilterSensitiveLog");
var DeleteDocumentRequestFilterSensitiveLog = /* @__PURE__ */ __name((obj) => ({
  ...obj,
  ...obj.AuthenticationToken && { AuthenticationToken: import_smithy_client.SENSITIVE_STRING }
}), "DeleteDocumentRequestFilterSensitiveLog");
var DeleteDocumentVersionRequestFilterSensitiveLog = /* @__PURE__ */ __name((obj) => ({
  ...obj,
  ...obj.AuthenticationToken && { AuthenticationToken: import_smithy_client.SENSITIVE_STRING }
}), "DeleteDocumentVersionRequestFilterSensitiveLog");
var DeleteFolderRequestFilterSensitiveLog = /* @__PURE__ */ __name((obj) => ({
  ...obj,
  ...obj.AuthenticationToken && { AuthenticationToken: import_smithy_client.SENSITIVE_STRING }
}), "DeleteFolderRequestFilterSensitiveLog");
var DeleteFolderContentsRequestFilterSensitiveLog = /* @__PURE__ */ __name((obj) => ({
  ...obj,
  ...obj.AuthenticationToken && { AuthenticationToken: import_smithy_client.SENSITIVE_STRING }
}), "DeleteFolderContentsRequestFilterSensitiveLog");
var DeleteLabelsRequestFilterSensitiveLog = /* @__PURE__ */ __name((obj) => ({
  ...obj,
  ...obj.AuthenticationToken && { AuthenticationToken: import_smithy_client.SENSITIVE_STRING }
}), "DeleteLabelsRequestFilterSensitiveLog");
var DeleteUserRequestFilterSensitiveLog = /* @__PURE__ */ __name((obj) => ({
  ...obj,
  ...obj.AuthenticationToken && { AuthenticationToken: import_smithy_client.SENSITIVE_STRING }
}), "DeleteUserRequestFilterSensitiveLog");
var DescribeActivitiesRequestFilterSensitiveLog = /* @__PURE__ */ __name((obj) => ({
  ...obj,
  ...obj.AuthenticationToken && { AuthenticationToken: import_smithy_client.SENSITIVE_STRING }
}), "DescribeActivitiesRequestFilterSensitiveLog");
var DescribeActivitiesResponseFilterSensitiveLog = /* @__PURE__ */ __name((obj) => ({
  ...obj,
  ...obj.UserActivities && { UserActivities: obj.UserActivities.map((item) => ActivityFilterSensitiveLog(item)) }
}), "DescribeActivitiesResponseFilterSensitiveLog");
var DescribeCommentsRequestFilterSensitiveLog = /* @__PURE__ */ __name((obj) => ({
  ...obj,
  ...obj.AuthenticationToken && { AuthenticationToken: import_smithy_client.SENSITIVE_STRING }
}), "DescribeCommentsRequestFilterSensitiveLog");
var DescribeCommentsResponseFilterSensitiveLog = /* @__PURE__ */ __name((obj) => ({
  ...obj,
  ...obj.Comments && { Comments: obj.Comments.map((item) => CommentFilterSensitiveLog(item)) }
}), "DescribeCommentsResponseFilterSensitiveLog");
var DescribeDocumentVersionsRequestFilterSensitiveLog = /* @__PURE__ */ __name((obj) => ({
  ...obj,
  ...obj.AuthenticationToken && { AuthenticationToken: import_smithy_client.SENSITIVE_STRING }
}), "DescribeDocumentVersionsRequestFilterSensitiveLog");
var DocumentVersionMetadataFilterSensitiveLog = /* @__PURE__ */ __name((obj) => ({
  ...obj,
  ...obj.Name && { Name: import_smithy_client.SENSITIVE_STRING },
  ...obj.Thumbnail && { Thumbnail: import_smithy_client.SENSITIVE_STRING },
  ...obj.Source && { Source: import_smithy_client.SENSITIVE_STRING }
}), "DocumentVersionMetadataFilterSensitiveLog");
var DescribeDocumentVersionsResponseFilterSensitiveLog = /* @__PURE__ */ __name((obj) => ({
  ...obj,
  ...obj.DocumentVersions && {
    DocumentVersions: obj.DocumentVersions.map((item) => DocumentVersionMetadataFilterSensitiveLog(item))
  }
}), "DescribeDocumentVersionsResponseFilterSensitiveLog");
var DescribeFolderContentsRequestFilterSensitiveLog = /* @__PURE__ */ __name((obj) => ({
  ...obj,
  ...obj.AuthenticationToken && { AuthenticationToken: import_smithy_client.SENSITIVE_STRING }
}), "DescribeFolderContentsRequestFilterSensitiveLog");
var DocumentMetadataFilterSensitiveLog = /* @__PURE__ */ __name((obj) => ({
  ...obj,
  ...obj.LatestVersionMetadata && {
    LatestVersionMetadata: DocumentVersionMetadataFilterSensitiveLog(obj.LatestVersionMetadata)
  }
}), "DocumentMetadataFilterSensitiveLog");
var DescribeFolderContentsResponseFilterSensitiveLog = /* @__PURE__ */ __name((obj) => ({
  ...obj,
  ...obj.Folders && { Folders: obj.Folders.map((item) => FolderMetadataFilterSensitiveLog(item)) },
  ...obj.Documents && { Documents: obj.Documents.map((item) => DocumentMetadataFilterSensitiveLog(item)) }
}), "DescribeFolderContentsResponseFilterSensitiveLog");
var DescribeGroupsRequestFilterSensitiveLog = /* @__PURE__ */ __name((obj) => ({
  ...obj,
  ...obj.AuthenticationToken && { AuthenticationToken: import_smithy_client.SENSITIVE_STRING },
  ...obj.SearchQuery && { SearchQuery: import_smithy_client.SENSITIVE_STRING }
}), "DescribeGroupsRequestFilterSensitiveLog");
var DescribeResourcePermissionsRequestFilterSensitiveLog = /* @__PURE__ */ __name((obj) => ({
  ...obj,
  ...obj.AuthenticationToken && { AuthenticationToken: import_smithy_client.SENSITIVE_STRING }
}), "DescribeResourcePermissionsRequestFilterSensitiveLog");
var DescribeRootFoldersRequestFilterSensitiveLog = /* @__PURE__ */ __name((obj) => ({
  ...obj,
  ...obj.AuthenticationToken && { AuthenticationToken: import_smithy_client.SENSITIVE_STRING }
}), "DescribeRootFoldersRequestFilterSensitiveLog");
var DescribeRootFoldersResponseFilterSensitiveLog = /* @__PURE__ */ __name((obj) => ({
  ...obj,
  ...obj.Folders && { Folders: obj.Folders.map((item) => FolderMetadataFilterSensitiveLog(item)) }
}), "DescribeRootFoldersResponseFilterSensitiveLog");
var DescribeUsersRequestFilterSensitiveLog = /* @__PURE__ */ __name((obj) => ({
  ...obj,
  ...obj.AuthenticationToken && { AuthenticationToken: import_smithy_client.SENSITIVE_STRING },
  ...obj.Query && { Query: import_smithy_client.SENSITIVE_STRING }
}), "DescribeUsersRequestFilterSensitiveLog");
var DescribeUsersResponseFilterSensitiveLog = /* @__PURE__ */ __name((obj) => ({
  ...obj,
  ...obj.Users && { Users: obj.Users.map((item) => UserFilterSensitiveLog(item)) }
}), "DescribeUsersResponseFilterSensitiveLog");
var GetCurrentUserRequestFilterSensitiveLog = /* @__PURE__ */ __name((obj) => ({
  ...obj,
  ...obj.AuthenticationToken && { AuthenticationToken: import_smithy_client.SENSITIVE_STRING }
}), "GetCurrentUserRequestFilterSensitiveLog");
var GetCurrentUserResponseFilterSensitiveLog = /* @__PURE__ */ __name((obj) => ({
  ...obj,
  ...obj.User && { User: UserFilterSensitiveLog(obj.User) }
}), "GetCurrentUserResponseFilterSensitiveLog");
var GetDocumentRequestFilterSensitiveLog = /* @__PURE__ */ __name((obj) => ({
  ...obj,
  ...obj.AuthenticationToken && { AuthenticationToken: import_smithy_client.SENSITIVE_STRING }
}), "GetDocumentRequestFilterSensitiveLog");
var GetDocumentResponseFilterSensitiveLog = /* @__PURE__ */ __name((obj) => ({
  ...obj,
  ...obj.Metadata && { Metadata: DocumentMetadataFilterSensitiveLog(obj.Metadata) }
}), "GetDocumentResponseFilterSensitiveLog");
var GetDocumentPathRequestFilterSensitiveLog = /* @__PURE__ */ __name((obj) => ({
  ...obj,
  ...obj.AuthenticationToken && { AuthenticationToken: import_smithy_client.SENSITIVE_STRING }
}), "GetDocumentPathRequestFilterSensitiveLog");
var ResourcePathComponentFilterSensitiveLog = /* @__PURE__ */ __name((obj) => ({
  ...obj,
  ...obj.Name && { Name: import_smithy_client.SENSITIVE_STRING }
}), "ResourcePathComponentFilterSensitiveLog");
var ResourcePathFilterSensitiveLog = /* @__PURE__ */ __name((obj) => ({
  ...obj,
  ...obj.Components && { Components: obj.Components.map((item) => ResourcePathComponentFilterSensitiveLog(item)) }
}), "ResourcePathFilterSensitiveLog");
var GetDocumentPathResponseFilterSensitiveLog = /* @__PURE__ */ __name((obj) => ({
  ...obj,
  ...obj.Path && { Path: ResourcePathFilterSensitiveLog(obj.Path) }
}), "GetDocumentPathResponseFilterSensitiveLog");
var GetDocumentVersionRequestFilterSensitiveLog = /* @__PURE__ */ __name((obj) => ({
  ...obj,
  ...obj.AuthenticationToken && { AuthenticationToken: import_smithy_client.SENSITIVE_STRING }
}), "GetDocumentVersionRequestFilterSensitiveLog");
var GetDocumentVersionResponseFilterSensitiveLog = /* @__PURE__ */ __name((obj) => ({
  ...obj,
  ...obj.Metadata && { Metadata: DocumentVersionMetadataFilterSensitiveLog(obj.Metadata) }
}), "GetDocumentVersionResponseFilterSensitiveLog");
var GetFolderRequestFilterSensitiveLog = /* @__PURE__ */ __name((obj) => ({
  ...obj,
  ...obj.AuthenticationToken && { AuthenticationToken: import_smithy_client.SENSITIVE_STRING }
}), "GetFolderRequestFilterSensitiveLog");
var GetFolderResponseFilterSensitiveLog = /* @__PURE__ */ __name((obj) => ({
  ...obj,
  ...obj.Metadata && { Metadata: FolderMetadataFilterSensitiveLog(obj.Metadata) }
}), "GetFolderResponseFilterSensitiveLog");
var GetFolderPathRequestFilterSensitiveLog = /* @__PURE__ */ __name((obj) => ({
  ...obj,
  ...obj.AuthenticationToken && { AuthenticationToken: import_smithy_client.SENSITIVE_STRING }
}), "GetFolderPathRequestFilterSensitiveLog");
var GetFolderPathResponseFilterSensitiveLog = /* @__PURE__ */ __name((obj) => ({
  ...obj,
  ...obj.Path && { Path: ResourcePathFilterSensitiveLog(obj.Path) }
}), "GetFolderPathResponseFilterSensitiveLog");
var GetResourcesRequestFilterSensitiveLog = /* @__PURE__ */ __name((obj) => ({
  ...obj,
  ...obj.AuthenticationToken && { AuthenticationToken: import_smithy_client.SENSITIVE_STRING }
}), "GetResourcesRequestFilterSensitiveLog");
var GetResourcesResponseFilterSensitiveLog = /* @__PURE__ */ __name((obj) => ({
  ...obj,
  ...obj.Folders && { Folders: obj.Folders.map((item) => FolderMetadataFilterSensitiveLog(item)) },
  ...obj.Documents && { Documents: obj.Documents.map((item) => DocumentMetadataFilterSensitiveLog(item)) }
}), "GetResourcesResponseFilterSensitiveLog");
var InitiateDocumentVersionUploadRequestFilterSensitiveLog = /* @__PURE__ */ __name((obj) => ({
  ...obj,
  ...obj.AuthenticationToken && { AuthenticationToken: import_smithy_client.SENSITIVE_STRING },
  ...obj.Name && { Name: import_smithy_client.SENSITIVE_STRING }
}), "InitiateDocumentVersionUploadRequestFilterSensitiveLog");
var UploadMetadataFilterSensitiveLog = /* @__PURE__ */ __name((obj) => ({
  ...obj,
  ...obj.UploadUrl && { UploadUrl: import_smithy_client.SENSITIVE_STRING }
}), "UploadMetadataFilterSensitiveLog");
var InitiateDocumentVersionUploadResponseFilterSensitiveLog = /* @__PURE__ */ __name((obj) => ({
  ...obj,
  ...obj.Metadata && { Metadata: DocumentMetadataFilterSensitiveLog(obj.Metadata) },
  ...obj.UploadMetadata && { UploadMetadata: UploadMetadataFilterSensitiveLog(obj.UploadMetadata) }
}), "InitiateDocumentVersionUploadResponseFilterSensitiveLog");
var RemoveAllResourcePermissionsRequestFilterSensitiveLog = /* @__PURE__ */ __name((obj) => ({
  ...obj,
  ...obj.AuthenticationToken && { AuthenticationToken: import_smithy_client.SENSITIVE_STRING }
}), "RemoveAllResourcePermissionsRequestFilterSensitiveLog");
var RemoveResourcePermissionRequestFilterSensitiveLog = /* @__PURE__ */ __name((obj) => ({
  ...obj,
  ...obj.AuthenticationToken && { AuthenticationToken: import_smithy_client.SENSITIVE_STRING }
}), "RemoveResourcePermissionRequestFilterSensitiveLog");
var RestoreDocumentVersionsRequestFilterSensitiveLog = /* @__PURE__ */ __name((obj) => ({
  ...obj,
  ...obj.AuthenticationToken && { AuthenticationToken: import_smithy_client.SENSITIVE_STRING }
}), "RestoreDocumentVersionsRequestFilterSensitiveLog");
var SearchResourcesRequestFilterSensitiveLog = /* @__PURE__ */ __name((obj) => ({
  ...obj,
  ...obj.AuthenticationToken && { AuthenticationToken: import_smithy_client.SENSITIVE_STRING },
  ...obj.QueryText && { QueryText: import_smithy_client.SENSITIVE_STRING }
}), "SearchResourcesRequestFilterSensitiveLog");
var ResponseItemFilterSensitiveLog = /* @__PURE__ */ __name((obj) => ({
  ...obj,
  ...obj.WebUrl && { WebUrl: import_smithy_client.SENSITIVE_STRING },
  ...obj.DocumentMetadata && { DocumentMetadata: DocumentMetadataFilterSensitiveLog(obj.DocumentMetadata) },
  ...obj.FolderMetadata && { FolderMetadata: FolderMetadataFilterSensitiveLog(obj.FolderMetadata) },
  ...obj.CommentMetadata && { CommentMetadata: CommentMetadataFilterSensitiveLog(obj.CommentMetadata) },
  ...obj.DocumentVersionMetadata && {
    DocumentVersionMetadata: DocumentVersionMetadataFilterSensitiveLog(obj.DocumentVersionMetadata)
  }
}), "ResponseItemFilterSensitiveLog");
var SearchResourcesResponseFilterSensitiveLog = /* @__PURE__ */ __name((obj) => ({
  ...obj,
  ...obj.Items && { Items: obj.Items.map((item) => ResponseItemFilterSensitiveLog(item)) }
}), "SearchResourcesResponseFilterSensitiveLog");
var UpdateDocumentRequestFilterSensitiveLog = /* @__PURE__ */ __name((obj) => ({
  ...obj,
  ...obj.AuthenticationToken && { AuthenticationToken: import_smithy_client.SENSITIVE_STRING },
  ...obj.Name && { Name: import_smithy_client.SENSITIVE_STRING }
}), "UpdateDocumentRequestFilterSensitiveLog");
var UpdateDocumentVersionRequestFilterSensitiveLog = /* @__PURE__ */ __name((obj) => ({
  ...obj,
  ...obj.AuthenticationToken && { AuthenticationToken: import_smithy_client.SENSITIVE_STRING }
}), "UpdateDocumentVersionRequestFilterSensitiveLog");
var UpdateFolderRequestFilterSensitiveLog = /* @__PURE__ */ __name((obj) => ({
  ...obj,
  ...obj.AuthenticationToken && { AuthenticationToken: import_smithy_client.SENSITIVE_STRING },
  ...obj.Name && { Name: import_smithy_client.SENSITIVE_STRING }
}), "UpdateFolderRequestFilterSensitiveLog");
var UpdateUserRequestFilterSensitiveLog = /* @__PURE__ */ __name((obj) => ({
  ...obj,
  ...obj.AuthenticationToken && { AuthenticationToken: import_smithy_client.SENSITIVE_STRING },
  ...obj.GivenName && { GivenName: import_smithy_client.SENSITIVE_STRING },
  ...obj.Surname && { Surname: import_smithy_client.SENSITIVE_STRING }
}), "UpdateUserRequestFilterSensitiveLog");
var UpdateUserResponseFilterSensitiveLog = /* @__PURE__ */ __name((obj) => ({
  ...obj,
  ...obj.User && { User: UserFilterSensitiveLog(obj.User) }
}), "UpdateUserResponseFilterSensitiveLog");

// src/protocols/Aws_restJson1.ts
var import_core2 = require("@aws-sdk/core");


var se_AbortDocumentVersionUploadCommand = /* @__PURE__ */ __name(async (input, context) => {
  const b = (0, import_core.requestBuilder)(input, context);
  const headers = (0, import_smithy_client.map)({}, isSerializableHeaderValue, {
    [_a]: input[_AT]
  });
  b.bp("/api/v1/documents/{DocumentId}/versions/{VersionId}");
  b.p("DocumentId", () => input.DocumentId, "{DocumentId}", false);
  b.p("VersionId", () => input.VersionId, "{VersionId}", false);
  let body;
  b.m("DELETE").h(headers).b(body);
  return b.build();
}, "se_AbortDocumentVersionUploadCommand");
var se_ActivateUserCommand = /* @__PURE__ */ __name(async (input, context) => {
  const b = (0, import_core.requestBuilder)(input, context);
  const headers = (0, import_smithy_client.map)({}, isSerializableHeaderValue, {
    [_a]: input[_AT]
  });
  b.bp("/api/v1/users/{UserId}/activation");
  b.p("UserId", () => input.UserId, "{UserId}", false);
  let body;
  b.m("POST").h(headers).b(body);
  return b.build();
}, "se_ActivateUserCommand");
var se_AddResourcePermissionsCommand = /* @__PURE__ */ __name(async (input, context) => {
  const b = (0, import_core.requestBuilder)(input, context);
  const headers = (0, import_smithy_client.map)({}, isSerializableHeaderValue, {
    "content-type": "application/json",
    [_a]: input[_AT]
  });
  b.bp("/api/v1/resources/{ResourceId}/permissions");
  b.p("ResourceId", () => input.ResourceId, "{ResourceId}", false);
  let body;
  body = JSON.stringify(
    (0, import_smithy_client.take)(input, {
      NotificationOptions: (_) => (0, import_smithy_client._json)(_),
      Principals: (_) => (0, import_smithy_client._json)(_)
    })
  );
  b.m("POST").h(headers).b(body);
  return b.build();
}, "se_AddResourcePermissionsCommand");
var se_CreateCommentCommand = /* @__PURE__ */ __name(async (input, context) => {
  const b = (0, import_core.requestBuilder)(input, context);
  const headers = (0, import_smithy_client.map)({}, isSerializableHeaderValue, {
    "content-type": "application/json",
    [_a]: input[_AT]
  });
  b.bp("/api/v1/documents/{DocumentId}/versions/{VersionId}/comment");
  b.p("DocumentId", () => input.DocumentId, "{DocumentId}", false);
  b.p("VersionId", () => input.VersionId, "{VersionId}", false);
  let body;
  body = JSON.stringify(
    (0, import_smithy_client.take)(input, {
      NotifyCollaborators: [],
      ParentId: [],
      Text: [],
      ThreadId: [],
      Visibility: []
    })
  );
  b.m("POST").h(headers).b(body);
  return b.build();
}, "se_CreateCommentCommand");
var se_CreateCustomMetadataCommand = /* @__PURE__ */ __name(async (input, context) => {
  const b = (0, import_core.requestBuilder)(input, context);
  const headers = (0, import_smithy_client.map)({}, isSerializableHeaderValue, {
    "content-type": "application/json",
    [_a]: input[_AT]
  });
  b.bp("/api/v1/resources/{ResourceId}/customMetadata");
  b.p("ResourceId", () => input.ResourceId, "{ResourceId}", false);
  const query = (0, import_smithy_client.map)({
    [_v]: [, input[_VI]]
  });
  let body;
  body = JSON.stringify(
    (0, import_smithy_client.take)(input, {
      CustomMetadata: (_) => (0, import_smithy_client._json)(_)
    })
  );
  b.m("PUT").h(headers).q(query).b(body);
  return b.build();
}, "se_CreateCustomMetadataCommand");
var se_CreateFolderCommand = /* @__PURE__ */ __name(async (input, context) => {
  const b = (0, import_core.requestBuilder)(input, context);
  const headers = (0, import_smithy_client.map)({}, isSerializableHeaderValue, {
    "content-type": "application/json",
    [_a]: input[_AT]
  });
  b.bp("/api/v1/folders");
  let body;
  body = JSON.stringify(
    (0, import_smithy_client.take)(input, {
      Name: [],
      ParentFolderId: []
    })
  );
  b.m("POST").h(headers).b(body);
  return b.build();
}, "se_CreateFolderCommand");
var se_CreateLabelsCommand = /* @__PURE__ */ __name(async (input, context) => {
  const b = (0, import_core.requestBuilder)(input, context);
  const headers = (0, import_smithy_client.map)({}, isSerializableHeaderValue, {
    "content-type": "application/json",
    [_a]: input[_AT]
  });
  b.bp("/api/v1/resources/{ResourceId}/labels");
  b.p("ResourceId", () => input.ResourceId, "{ResourceId}", false);
  let body;
  body = JSON.stringify(
    (0, import_smithy_client.take)(input, {
      Labels: (_) => (0, import_smithy_client._json)(_)
    })
  );
  b.m("PUT").h(headers).b(body);
  return b.build();
}, "se_CreateLabelsCommand");
var se_CreateNotificationSubscriptionCommand = /* @__PURE__ */ __name(async (input, context) => {
  const b = (0, import_core.requestBuilder)(input, context);
  const headers = {
    "content-type": "application/json"
  };
  b.bp("/api/v1/organizations/{OrganizationId}/subscriptions");
  b.p("OrganizationId", () => input.OrganizationId, "{OrganizationId}", false);
  let body;
  body = JSON.stringify(
    (0, import_smithy_client.take)(input, {
      Endpoint: [],
      Protocol: [],
      SubscriptionType: []
    })
  );
  b.m("POST").h(headers).b(body);
  return b.build();
}, "se_CreateNotificationSubscriptionCommand");
var se_CreateUserCommand = /* @__PURE__ */ __name(async (input, context) => {
  const b = (0, import_core.requestBuilder)(input, context);
  const headers = (0, import_smithy_client.map)({}, isSerializableHeaderValue, {
    "content-type": "application/json",
    [_a]: input[_AT]
  });
  b.bp("/api/v1/users");
  let body;
  body = JSON.stringify(
    (0, import_smithy_client.take)(input, {
      EmailAddress: [],
      GivenName: [],
      OrganizationId: [],
      Password: [],
      StorageRule: (_) => (0, import_smithy_client._json)(_),
      Surname: [],
      TimeZoneId: [],
      Username: []
    })
  );
  b.m("POST").h(headers).b(body);
  return b.build();
}, "se_CreateUserCommand");
var se_DeactivateUserCommand = /* @__PURE__ */ __name(async (input, context) => {
  const b = (0, import_core.requestBuilder)(input, context);
  const headers = (0, import_smithy_client.map)({}, isSerializableHeaderValue, {
    [_a]: input[_AT]
  });
  b.bp("/api/v1/users/{UserId}/activation");
  b.p("UserId", () => input.UserId, "{UserId}", false);
  let body;
  b.m("DELETE").h(headers).b(body);
  return b.build();
}, "se_DeactivateUserCommand");
var se_DeleteCommentCommand = /* @__PURE__ */ __name(async (input, context) => {
  const b = (0, import_core.requestBuilder)(input, context);
  const headers = (0, import_smithy_client.map)({}, isSerializableHeaderValue, {
    [_a]: input[_AT]
  });
  b.bp("/api/v1/documents/{DocumentId}/versions/{VersionId}/comment/{CommentId}");
  b.p("DocumentId", () => input.DocumentId, "{DocumentId}", false);
  b.p("VersionId", () => input.VersionId, "{VersionId}", false);
  b.p("CommentId", () => input.CommentId, "{CommentId}", false);
  let body;
  b.m("DELETE").h(headers).b(body);
  return b.build();
}, "se_DeleteCommentCommand");
var se_DeleteCustomMetadataCommand = /* @__PURE__ */ __name(async (input, context) => {
  const b = (0, import_core.requestBuilder)(input, context);
  const headers = (0, import_smithy_client.map)({}, isSerializableHeaderValue, {
    [_a]: input[_AT]
  });
  b.bp("/api/v1/resources/{ResourceId}/customMetadata");
  b.p("ResourceId", () => input.ResourceId, "{ResourceId}", false);
  const query = (0, import_smithy_client.map)({
    [_vI]: [, input[_VI]],
    [_k]: [() => input.Keys !== void 0, () => (input[_K] || []).map((_entry) => _entry)],
    [_dA]: [() => input.DeleteAll !== void 0, () => input[_DA].toString()]
  });
  let body;
  b.m("DELETE").h(headers).q(query).b(body);
  return b.build();
}, "se_DeleteCustomMetadataCommand");
var se_DeleteDocumentCommand = /* @__PURE__ */ __name(async (input, context) => {
  const b = (0, import_core.requestBuilder)(input, context);
  const headers = (0, import_smithy_client.map)({}, isSerializableHeaderValue, {
    [_a]: input[_AT]
  });
  b.bp("/api/v1/documents/{DocumentId}");
  b.p("DocumentId", () => input.DocumentId, "{DocumentId}", false);
  let body;
  b.m("DELETE").h(headers).b(body);
  return b.build();
}, "se_DeleteDocumentCommand");
var se_DeleteDocumentVersionCommand = /* @__PURE__ */ __name(async (input, context) => {
  const b = (0, import_core.requestBuilder)(input, context);
  const headers = (0, import_smithy_client.map)({}, isSerializableHeaderValue, {
    [_a]: input[_AT]
  });
  b.bp("/api/v1/documentVersions/{DocumentId}/versions/{VersionId}");
  b.p("DocumentId", () => input.DocumentId, "{DocumentId}", false);
  b.p("VersionId", () => input.VersionId, "{VersionId}", false);
  const query = (0, import_smithy_client.map)({
    [_dPV]: [(0, import_smithy_client.expectNonNull)(input.DeletePriorVersions, `DeletePriorVersions`) != null, () => input[_DPV].toString()]
  });
  let body;
  b.m("DELETE").h(headers).q(query).b(body);
  return b.build();
}, "se_DeleteDocumentVersionCommand");
var se_DeleteFolderCommand = /* @__PURE__ */ __name(async (input, context) => {
  const b = (0, import_core.requestBuilder)(input, context);
  const headers = (0, import_smithy_client.map)({}, isSerializableHeaderValue, {
    [_a]: input[_AT]
  });
  b.bp("/api/v1/folders/{FolderId}");
  b.p("FolderId", () => input.FolderId, "{FolderId}", false);
  let body;
  b.m("DELETE").h(headers).b(body);
  return b.build();
}, "se_DeleteFolderCommand");
var se_DeleteFolderContentsCommand = /* @__PURE__ */ __name(async (input, context) => {
  const b = (0, import_core.requestBuilder)(input, context);
  const headers = (0, import_smithy_client.map)({}, isSerializableHeaderValue, {
    [_a]: input[_AT]
  });
  b.bp("/api/v1/folders/{FolderId}/contents");
  b.p("FolderId", () => input.FolderId, "{FolderId}", false);
  let body;
  b.m("DELETE").h(headers).b(body);
  return b.build();
}, "se_DeleteFolderContentsCommand");
var se_DeleteLabelsCommand = /* @__PURE__ */ __name(async (input, context) => {
  const b = (0, import_core.requestBuilder)(input, context);
  const headers = (0, import_smithy_client.map)({}, isSerializableHeaderValue, {
    [_a]: input[_AT]
  });
  b.bp("/api/v1/resources/{ResourceId}/labels");
  b.p("ResourceId", () => input.ResourceId, "{ResourceId}", false);
  const query = (0, import_smithy_client.map)({
    [_l]: [() => input.Labels !== void 0, () => (input[_L] || []).map((_entry) => _entry)],
    [_dA]: [() => input.DeleteAll !== void 0, () => input[_DA].toString()]
  });
  let body;
  b.m("DELETE").h(headers).q(query).b(body);
  return b.build();
}, "se_DeleteLabelsCommand");
var se_DeleteNotificationSubscriptionCommand = /* @__PURE__ */ __name(async (input, context) => {
  const b = (0, import_core.requestBuilder)(input, context);
  const headers = {};
  b.bp("/api/v1/organizations/{OrganizationId}/subscriptions/{SubscriptionId}");
  b.p("SubscriptionId", () => input.SubscriptionId, "{SubscriptionId}", false);
  b.p("OrganizationId", () => input.OrganizationId, "{OrganizationId}", false);
  let body;
  b.m("DELETE").h(headers).b(body);
  return b.build();
}, "se_DeleteNotificationSubscriptionCommand");
var se_DeleteUserCommand = /* @__PURE__ */ __name(async (input, context) => {
  const b = (0, import_core.requestBuilder)(input, context);
  const headers = (0, import_smithy_client.map)({}, isSerializableHeaderValue, {
    [_a]: input[_AT]
  });
  b.bp("/api/v1/users/{UserId}");
  b.p("UserId", () => input.UserId, "{UserId}", false);
  let body;
  b.m("DELETE").h(headers).b(body);
  return b.build();
}, "se_DeleteUserCommand");
var se_DescribeActivitiesCommand = /* @__PURE__ */ __name(async (input, context) => {
  const b = (0, import_core.requestBuilder)(input, context);
  const headers = (0, import_smithy_client.map)({}, isSerializableHeaderValue, {
    [_a]: input[_AT]
  });
  b.bp("/api/v1/activities");
  const query = (0, import_smithy_client.map)({
    [_sT]: [() => input.StartTime !== void 0, () => (input[_ST].toISOString().split(".")[0] + "Z").toString()],
    [_eT]: [() => input.EndTime !== void 0, () => (input[_ET].toISOString().split(".")[0] + "Z").toString()],
    [_oI]: [, input[_OI]],
    [_aT]: [, input[_ATc]],
    [_rI]: [, input[_RI]],
    [_uI]: [, input[_UI]],
    [_iIA]: [() => input.IncludeIndirectActivities !== void 0, () => input[_IIA].toString()],
    [_li]: [() => input.Limit !== void 0, () => input[_Li].toString()],
    [_m]: [, input[_M]]
  });
  let body;
  b.m("GET").h(headers).q(query).b(body);
  return b.build();
}, "se_DescribeActivitiesCommand");
var se_DescribeCommentsCommand = /* @__PURE__ */ __name(async (input, context) => {
  const b = (0, import_core.requestBuilder)(input, context);
  const headers = (0, import_smithy_client.map)({}, isSerializableHeaderValue, {
    [_a]: input[_AT]
  });
  b.bp("/api/v1/documents/{DocumentId}/versions/{VersionId}/comments");
  b.p("DocumentId", () => input.DocumentId, "{DocumentId}", false);
  b.p("VersionId", () => input.VersionId, "{VersionId}", false);
  const query = (0, import_smithy_client.map)({
    [_li]: [() => input.Limit !== void 0, () => input[_Li].toString()],
    [_m]: [, input[_M]]
  });
  let body;
  b.m("GET").h(headers).q(query).b(body);
  return b.build();
}, "se_DescribeCommentsCommand");
var se_DescribeDocumentVersionsCommand = /* @__PURE__ */ __name(async (input, context) => {
  const b = (0, import_core.requestBuilder)(input, context);
  const headers = (0, import_smithy_client.map)({}, isSerializableHeaderValue, {
    [_a]: input[_AT]
  });
  b.bp("/api/v1/documents/{DocumentId}/versions");
  b.p("DocumentId", () => input.DocumentId, "{DocumentId}", false);
  const query = (0, import_smithy_client.map)({
    [_m]: [, input[_M]],
    [_li]: [() => input.Limit !== void 0, () => input[_Li].toString()],
    [_i]: [, input[_I]],
    [_f]: [, input[_F]]
  });
  let body;
  b.m("GET").h(headers).q(query).b(body);
  return b.build();
}, "se_DescribeDocumentVersionsCommand");
var se_DescribeFolderContentsCommand = /* @__PURE__ */ __name(async (input, context) => {
  const b = (0, import_core.requestBuilder)(input, context);
  const headers = (0, import_smithy_client.map)({}, isSerializableHeaderValue, {
    [_a]: input[_AT]
  });
  b.bp("/api/v1/folders/{FolderId}/contents");
  b.p("FolderId", () => input.FolderId, "{FolderId}", false);
  const query = (0, import_smithy_client.map)({
    [_s]: [, input[_S]],
    [_o]: [, input[_O]],
    [_li]: [() => input.Limit !== void 0, () => input[_Li].toString()],
    [_m]: [, input[_M]],
    [_t]: [, input[_T]],
    [_i]: [, input[_I]]
  });
  let body;
  b.m("GET").h(headers).q(query).b(body);
  return b.build();
}, "se_DescribeFolderContentsCommand");
var se_DescribeGroupsCommand = /* @__PURE__ */ __name(async (input, context) => {
  const b = (0, import_core.requestBuilder)(input, context);
  const headers = (0, import_smithy_client.map)({}, isSerializableHeaderValue, {
    [_a]: input[_AT]
  });
  b.bp("/api/v1/groups");
  const query = (0, import_smithy_client.map)({
    [_sQ]: [, (0, import_smithy_client.expectNonNull)(input[_SQ], `SearchQuery`)],
    [_oI]: [, input[_OI]],
    [_m]: [, input[_M]],
    [_li]: [() => input.Limit !== void 0, () => input[_Li].toString()]
  });
  let body;
  b.m("GET").h(headers).q(query).b(body);
  return b.build();
}, "se_DescribeGroupsCommand");
var se_DescribeNotificationSubscriptionsCommand = /* @__PURE__ */ __name(async (input, context) => {
  const b = (0, import_core.requestBuilder)(input, context);
  const headers = {};
  b.bp("/api/v1/organizations/{OrganizationId}/subscriptions");
  b.p("OrganizationId", () => input.OrganizationId, "{OrganizationId}", false);
  const query = (0, import_smithy_client.map)({
    [_m]: [, input[_M]],
    [_li]: [() => input.Limit !== void 0, () => input[_Li].toString()]
  });
  let body;
  b.m("GET").h(headers).q(query).b(body);
  return b.build();
}, "se_DescribeNotificationSubscriptionsCommand");
var se_DescribeResourcePermissionsCommand = /* @__PURE__ */ __name(async (input, context) => {
  const b = (0, import_core.requestBuilder)(input, context);
  const headers = (0, import_smithy_client.map)({}, isSerializableHeaderValue, {
    [_a]: input[_AT]
  });
  b.bp("/api/v1/resources/{ResourceId}/permissions");
  b.p("ResourceId", () => input.ResourceId, "{ResourceId}", false);
  const query = (0, import_smithy_client.map)({
    [_pI]: [, input[_PI]],
    [_li]: [() => input.Limit !== void 0, () => input[_Li].toString()],
    [_m]: [, input[_M]]
  });
  let body;
  b.m("GET").h(headers).q(query).b(body);
  return b.build();
}, "se_DescribeResourcePermissionsCommand");
var se_DescribeRootFoldersCommand = /* @__PURE__ */ __name(async (input, context) => {
  const b = (0, import_core.requestBuilder)(input, context);
  const headers = (0, import_smithy_client.map)({}, isSerializableHeaderValue, {
    [_a]: input[_AT]
  });
  b.bp("/api/v1/me/root");
  const query = (0, import_smithy_client.map)({
    [_li]: [() => input.Limit !== void 0, () => input[_Li].toString()],
    [_m]: [, input[_M]]
  });
  let body;
  b.m("GET").h(headers).q(query).b(body);
  return b.build();
}, "se_DescribeRootFoldersCommand");
var se_DescribeUsersCommand = /* @__PURE__ */ __name(async (input, context) => {
  const b = (0, import_core.requestBuilder)(input, context);
  const headers = (0, import_smithy_client.map)({}, isSerializableHeaderValue, {
    [_a]: input[_AT]
  });
  b.bp("/api/v1/users");
  const query = (0, import_smithy_client.map)({
    [_oI]: [, input[_OI]],
    [_uIs]: [, input[_UIs]],
    [_q]: [, input[_Q]],
    [_i]: [, input[_I]],
    [_o]: [, input[_O]],
    [_s]: [, input[_S]],
    [_m]: [, input[_M]],
    [_li]: [() => input.Limit !== void 0, () => input[_Li].toString()],
    [_f]: [, input[_F]]
  });
  let body;
  b.m("GET").h(headers).q(query).b(body);
  return b.build();
}, "se_DescribeUsersCommand");
var se_GetCurrentUserCommand = /* @__PURE__ */ __name(async (input, context) => {
  const b = (0, import_core.requestBuilder)(input, context);
  const headers = (0, import_smithy_client.map)({}, isSerializableHeaderValue, {
    [_a]: input[_AT]
  });
  b.bp("/api/v1/me");
  let body;
  b.m("GET").h(headers).b(body);
  return b.build();
}, "se_GetCurrentUserCommand");
var se_GetDocumentCommand = /* @__PURE__ */ __name(async (input, context) => {
  const b = (0, import_core.requestBuilder)(input, context);
  const headers = (0, import_smithy_client.map)({}, isSerializableHeaderValue, {
    [_a]: input[_AT]
  });
  b.bp("/api/v1/documents/{DocumentId}");
  b.p("DocumentId", () => input.DocumentId, "{DocumentId}", false);
  const query = (0, import_smithy_client.map)({
    [_iCM]: [() => input.IncludeCustomMetadata !== void 0, () => input[_ICM].toString()]
  });
  let body;
  b.m("GET").h(headers).q(query).b(body);
  return b.build();
}, "se_GetDocumentCommand");
var se_GetDocumentPathCommand = /* @__PURE__ */ __name(async (input, context) => {
  const b = (0, import_core.requestBuilder)(input, context);
  const headers = (0, import_smithy_client.map)({}, isSerializableHeaderValue, {
    [_a]: input[_AT]
  });
  b.bp("/api/v1/documents/{DocumentId}/path");
  b.p("DocumentId", () => input.DocumentId, "{DocumentId}", false);
  const query = (0, import_smithy_client.map)({
    [_li]: [() => input.Limit !== void 0, () => input[_Li].toString()],
    [_f]: [, input[_F]],
    [_m]: [, input[_M]]
  });
  let body;
  b.m("GET").h(headers).q(query).b(body);
  return b.build();
}, "se_GetDocumentPathCommand");
var se_GetDocumentVersionCommand = /* @__PURE__ */ __name(async (input, context) => {
  const b = (0, import_core.requestBuilder)(input, context);
  const headers = (0, import_smithy_client.map)({}, isSerializableHeaderValue, {
    [_a]: input[_AT]
  });
  b.bp("/api/v1/documents/{DocumentId}/versions/{VersionId}");
  b.p("DocumentId", () => input.DocumentId, "{DocumentId}", false);
  b.p("VersionId", () => input.VersionId, "{VersionId}", false);
  const query = (0, import_smithy_client.map)({
    [_f]: [, input[_F]],
    [_iCM]: [() => input.IncludeCustomMetadata !== void 0, () => input[_ICM].toString()]
  });
  let body;
  b.m("GET").h(headers).q(query).b(body);
  return b.build();
}, "se_GetDocumentVersionCommand");
var se_GetFolderCommand = /* @__PURE__ */ __name(async (input, context) => {
  const b = (0, import_core.requestBuilder)(input, context);
  const headers = (0, import_smithy_client.map)({}, isSerializableHeaderValue, {
    [_a]: input[_AT]
  });
  b.bp("/api/v1/folders/{FolderId}");
  b.p("FolderId", () => input.FolderId, "{FolderId}", false);
  const query = (0, import_smithy_client.map)({
    [_iCM]: [() => input.IncludeCustomMetadata !== void 0, () => input[_ICM].toString()]
  });
  let body;
  b.m("GET").h(headers).q(query).b(body);
  return b.build();
}, "se_GetFolderCommand");
var se_GetFolderPathCommand = /* @__PURE__ */ __name(async (input, context) => {
  const b = (0, import_core.requestBuilder)(input, context);
  const headers = (0, import_smithy_client.map)({}, isSerializableHeaderValue, {
    [_a]: input[_AT]
  });
  b.bp("/api/v1/folders/{FolderId}/path");
  b.p("FolderId", () => input.FolderId, "{FolderId}", false);
  const query = (0, import_smithy_client.map)({
    [_li]: [() => input.Limit !== void 0, () => input[_Li].toString()],
    [_f]: [, input[_F]],
    [_m]: [, input[_M]]
  });
  let body;
  b.m("GET").h(headers).q(query).b(body);
  return b.build();
}, "se_GetFolderPathCommand");
var se_GetResourcesCommand = /* @__PURE__ */ __name(async (input, context) => {
  const b = (0, import_core.requestBuilder)(input, context);
  const headers = (0, import_smithy_client.map)({}, isSerializableHeaderValue, {
    [_a]: input[_AT]
  });
  b.bp("/api/v1/resources");
  const query = (0, import_smithy_client.map)({
    [_uI]: [, input[_UI]],
    [_cT]: [, input[_CT]],
    [_li]: [() => input.Limit !== void 0, () => input[_Li].toString()],
    [_m]: [, input[_M]]
  });
  let body;
  b.m("GET").h(headers).q(query).b(body);
  return b.build();
}, "se_GetResourcesCommand");
var se_InitiateDocumentVersionUploadCommand = /* @__PURE__ */ __name(async (input, context) => {
  const b = (0, import_core.requestBuilder)(input, context);
  const headers = (0, import_smithy_client.map)({}, isSerializableHeaderValue, {
    "content-type": "application/json",
    [_a]: input[_AT]
  });
  b.bp("/api/v1/documents");
  let body;
  body = JSON.stringify(
    (0, import_smithy_client.take)(input, {
      ContentCreatedTimestamp: (_) => Math.round(_.getTime() / 1e3),
      ContentModifiedTimestamp: (_) => Math.round(_.getTime() / 1e3),
      ContentType: [],
      DocumentSizeInBytes: [],
      Id: [],
      Name: [],
      ParentFolderId: []
    })
  );
  b.m("POST").h(headers).b(body);
  return b.build();
}, "se_InitiateDocumentVersionUploadCommand");
var se_RemoveAllResourcePermissionsCommand = /* @__PURE__ */ __name(async (input, context) => {
  const b = (0, import_core.requestBuilder)(input, context);
  const headers = (0, import_smithy_client.map)({}, isSerializableHeaderValue, {
    [_a]: input[_AT]
  });
  b.bp("/api/v1/resources/{ResourceId}/permissions");
  b.p("ResourceId", () => input.ResourceId, "{ResourceId}", false);
  let body;
  b.m("DELETE").h(headers).b(body);
  return b.build();
}, "se_RemoveAllResourcePermissionsCommand");
var se_RemoveResourcePermissionCommand = /* @__PURE__ */ __name(async (input, context) => {
  const b = (0, import_core.requestBuilder)(input, context);
  const headers = (0, import_smithy_client.map)({}, isSerializableHeaderValue, {
    [_a]: input[_AT]
  });
  b.bp("/api/v1/resources/{ResourceId}/permissions/{PrincipalId}");
  b.p("ResourceId", () => input.ResourceId, "{ResourceId}", false);
  b.p("PrincipalId", () => input.PrincipalId, "{PrincipalId}", false);
  const query = (0, import_smithy_client.map)({
    [_t]: [, input[_PT]]
  });
  let body;
  b.m("DELETE").h(headers).q(query).b(body);
  return b.build();
}, "se_RemoveResourcePermissionCommand");
var se_RestoreDocumentVersionsCommand = /* @__PURE__ */ __name(async (input, context) => {
  const b = (0, import_core.requestBuilder)(input, context);
  const headers = (0, import_smithy_client.map)({}, isSerializableHeaderValue, {
    [_a]: input[_AT]
  });
  b.bp("/api/v1/documentVersions/restore/{DocumentId}");
  b.p("DocumentId", () => input.DocumentId, "{DocumentId}", false);
  let body;
  b.m("POST").h(headers).b(body);
  return b.build();
}, "se_RestoreDocumentVersionsCommand");
var se_SearchResourcesCommand = /* @__PURE__ */ __name(async (input, context) => {
  const b = (0, import_core.requestBuilder)(input, context);
  const headers = (0, import_smithy_client.map)({}, isSerializableHeaderValue, {
    "content-type": "application/json",
    [_a]: input[_AT]
  });
  b.bp("/api/v1/search");
  let body;
  body = JSON.stringify(
    (0, import_smithy_client.take)(input, {
      AdditionalResponseFields: (_) => (0, import_smithy_client._json)(_),
      Filters: (_) => se_Filters(_, context),
      Limit: [],
      Marker: [],
      OrderBy: (_) => (0, import_smithy_client._json)(_),
      OrganizationId: [],
      QueryScopes: (_) => (0, import_smithy_client._json)(_),
      QueryText: []
    })
  );
  b.m("POST").h(headers).b(body);
  return b.build();
}, "se_SearchResourcesCommand");
var se_UpdateDocumentCommand = /* @__PURE__ */ __name(async (input, context) => {
  const b = (0, import_core.requestBuilder)(input, context);
  const headers = (0, import_smithy_client.map)({}, isSerializableHeaderValue, {
    "content-type": "application/json",
    [_a]: input[_AT]
  });
  b.bp("/api/v1/documents/{DocumentId}");
  b.p("DocumentId", () => input.DocumentId, "{DocumentId}", false);
  let body;
  body = JSON.stringify(
    (0, import_smithy_client.take)(input, {
      Name: [],
      ParentFolderId: [],
      ResourceState: []
    })
  );
  b.m("PATCH").h(headers).b(body);
  return b.build();
}, "se_UpdateDocumentCommand");
var se_UpdateDocumentVersionCommand = /* @__PURE__ */ __name(async (input, context) => {
  const b = (0, import_core.requestBuilder)(input, context);
  const headers = (0, import_smithy_client.map)({}, isSerializableHeaderValue, {
    "content-type": "application/json",
    [_a]: input[_AT]
  });
  b.bp("/api/v1/documents/{DocumentId}/versions/{VersionId}");
  b.p("DocumentId", () => input.DocumentId, "{DocumentId}", false);
  b.p("VersionId", () => input.VersionId, "{VersionId}", false);
  let body;
  body = JSON.stringify(
    (0, import_smithy_client.take)(input, {
      VersionStatus: []
    })
  );
  b.m("PATCH").h(headers).b(body);
  return b.build();
}, "se_UpdateDocumentVersionCommand");
var se_UpdateFolderCommand = /* @__PURE__ */ __name(async (input, context) => {
  const b = (0, import_core.requestBuilder)(input, context);
  const headers = (0, import_smithy_client.map)({}, isSerializableHeaderValue, {
    "content-type": "application/json",
    [_a]: input[_AT]
  });
  b.bp("/api/v1/folders/{FolderId}");
  b.p("FolderId", () => input.FolderId, "{FolderId}", false);
  let body;
  body = JSON.stringify(
    (0, import_smithy_client.take)(input, {
      Name: [],
      ParentFolderId: [],
      ResourceState: []
    })
  );
  b.m("PATCH").h(headers).b(body);
  return b.build();
}, "se_UpdateFolderCommand");
var se_UpdateUserCommand = /* @__PURE__ */ __name(async (input, context) => {
  const b = (0, import_core.requestBuilder)(input, context);
  const headers = (0, import_smithy_client.map)({}, isSerializableHeaderValue, {
    "content-type": "application/json",
    [_a]: input[_AT]
  });
  b.bp("/api/v1/users/{UserId}");
  b.p("UserId", () => input.UserId, "{UserId}", false);
  let body;
  body = JSON.stringify(
    (0, import_smithy_client.take)(input, {
      GivenName: [],
      GrantPoweruserPrivileges: [],
      Locale: [],
      StorageRule: (_) => (0, import_smithy_client._json)(_),
      Surname: [],
      TimeZoneId: [],
      Type: []
    })
  );
  b.m("PATCH").h(headers).b(body);
  return b.build();
}, "se_UpdateUserCommand");
var de_AbortDocumentVersionUploadCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode !== 204 && output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const contents = (0, import_smithy_client.map)({
    $metadata: deserializeMetadata(output)
  });
  await (0, import_smithy_client.collectBody)(output.body, context);
  return contents;
}, "de_AbortDocumentVersionUploadCommand");
var de_ActivateUserCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode !== 200 && output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const contents = (0, import_smithy_client.map)({
    $metadata: deserializeMetadata(output)
  });
  const data = (0, import_smithy_client.expectNonNull)((0, import_smithy_client.expectObject)(await (0, import_core2.parseJsonBody)(output.body, context)), "body");
  const doc = (0, import_smithy_client.take)(data, {
    User: (_) => de_User(_, context)
  });
  Object.assign(contents, doc);
  return contents;
}, "de_ActivateUserCommand");
var de_AddResourcePermissionsCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode !== 201 && output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const contents = (0, import_smithy_client.map)({
    $metadata: deserializeMetadata(output)
  });
  const data = (0, import_smithy_client.expectNonNull)((0, import_smithy_client.expectObject)(await (0, import_core2.parseJsonBody)(output.body, context)), "body");
  const doc = (0, import_smithy_client.take)(data, {
    ShareResults: import_smithy_client._json
  });
  Object.assign(contents, doc);
  return contents;
}, "de_AddResourcePermissionsCommand");
var de_CreateCommentCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode !== 201 && output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const contents = (0, import_smithy_client.map)({
    $metadata: deserializeMetadata(output)
  });
  const data = (0, import_smithy_client.expectNonNull)((0, import_smithy_client.expectObject)(await (0, import_core2.parseJsonBody)(output.body, context)), "body");
  const doc = (0, import_smithy_client.take)(data, {
    Comment: (_) => de_Comment(_, context)
  });
  Object.assign(contents, doc);
  return contents;
}, "de_CreateCommentCommand");
var de_CreateCustomMetadataCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode !== 200 && output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const contents = (0, import_smithy_client.map)({
    $metadata: deserializeMetadata(output)
  });
  await (0, import_smithy_client.collectBody)(output.body, context);
  return contents;
}, "de_CreateCustomMetadataCommand");
var de_CreateFolderCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode !== 201 && output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const contents = (0, import_smithy_client.map)({
    $metadata: deserializeMetadata(output)
  });
  const data = (0, import_smithy_client.expectNonNull)((0, import_smithy_client.expectObject)(await (0, import_core2.parseJsonBody)(output.body, context)), "body");
  const doc = (0, import_smithy_client.take)(data, {
    Metadata: (_) => de_FolderMetadata(_, context)
  });
  Object.assign(contents, doc);
  return contents;
}, "de_CreateFolderCommand");
var de_CreateLabelsCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode !== 200 && output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const contents = (0, import_smithy_client.map)({
    $metadata: deserializeMetadata(output)
  });
  await (0, import_smithy_client.collectBody)(output.body, context);
  return contents;
}, "de_CreateLabelsCommand");
var de_CreateNotificationSubscriptionCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode !== 200 && output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const contents = (0, import_smithy_client.map)({
    $metadata: deserializeMetadata(output)
  });
  const data = (0, import_smithy_client.expectNonNull)((0, import_smithy_client.expectObject)(await (0, import_core2.parseJsonBody)(output.body, context)), "body");
  const doc = (0, import_smithy_client.take)(data, {
    Subscription: import_smithy_client._json
  });
  Object.assign(contents, doc);
  return contents;
}, "de_CreateNotificationSubscriptionCommand");
var de_CreateUserCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode !== 201 && output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const contents = (0, import_smithy_client.map)({
    $metadata: deserializeMetadata(output)
  });
  const data = (0, import_smithy_client.expectNonNull)((0, import_smithy_client.expectObject)(await (0, import_core2.parseJsonBody)(output.body, context)), "body");
  const doc = (0, import_smithy_client.take)(data, {
    User: (_) => de_User(_, context)
  });
  Object.assign(contents, doc);
  return contents;
}, "de_CreateUserCommand");
var de_DeactivateUserCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode !== 204 && output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const contents = (0, import_smithy_client.map)({
    $metadata: deserializeMetadata(output)
  });
  await (0, import_smithy_client.collectBody)(output.body, context);
  return contents;
}, "de_DeactivateUserCommand");
var de_DeleteCommentCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode !== 204 && output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const contents = (0, import_smithy_client.map)({
    $metadata: deserializeMetadata(output)
  });
  await (0, import_smithy_client.collectBody)(output.body, context);
  return contents;
}, "de_DeleteCommentCommand");
var de_DeleteCustomMetadataCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode !== 200 && output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const contents = (0, import_smithy_client.map)({
    $metadata: deserializeMetadata(output)
  });
  await (0, import_smithy_client.collectBody)(output.body, context);
  return contents;
}, "de_DeleteCustomMetadataCommand");
var de_DeleteDocumentCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode !== 204 && output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const contents = (0, import_smithy_client.map)({
    $metadata: deserializeMetadata(output)
  });
  await (0, import_smithy_client.collectBody)(output.body, context);
  return contents;
}, "de_DeleteDocumentCommand");
var de_DeleteDocumentVersionCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode !== 204 && output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const contents = (0, import_smithy_client.map)({
    $metadata: deserializeMetadata(output)
  });
  await (0, import_smithy_client.collectBody)(output.body, context);
  return contents;
}, "de_DeleteDocumentVersionCommand");
var de_DeleteFolderCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode !== 204 && output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const contents = (0, import_smithy_client.map)({
    $metadata: deserializeMetadata(output)
  });
  await (0, import_smithy_client.collectBody)(output.body, context);
  return contents;
}, "de_DeleteFolderCommand");
var de_DeleteFolderContentsCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode !== 204 && output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const contents = (0, import_smithy_client.map)({
    $metadata: deserializeMetadata(output)
  });
  await (0, import_smithy_client.collectBody)(output.body, context);
  return contents;
}, "de_DeleteFolderContentsCommand");
var de_DeleteLabelsCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode !== 200 && output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const contents = (0, import_smithy_client.map)({
    $metadata: deserializeMetadata(output)
  });
  await (0, import_smithy_client.collectBody)(output.body, context);
  return contents;
}, "de_DeleteLabelsCommand");
var de_DeleteNotificationSubscriptionCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode !== 200 && output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const contents = (0, import_smithy_client.map)({
    $metadata: deserializeMetadata(output)
  });
  await (0, import_smithy_client.collectBody)(output.body, context);
  return contents;
}, "de_DeleteNotificationSubscriptionCommand");
var de_DeleteUserCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode !== 204 && output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const contents = (0, import_smithy_client.map)({
    $metadata: deserializeMetadata(output)
  });
  await (0, import_smithy_client.collectBody)(output.body, context);
  return contents;
}, "de_DeleteUserCommand");
var de_DescribeActivitiesCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode !== 200 && output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const contents = (0, import_smithy_client.map)({
    $metadata: deserializeMetadata(output)
  });
  const data = (0, import_smithy_client.expectNonNull)((0, import_smithy_client.expectObject)(await (0, import_core2.parseJsonBody)(output.body, context)), "body");
  const doc = (0, import_smithy_client.take)(data, {
    Marker: import_smithy_client.expectString,
    UserActivities: (_) => de_UserActivities(_, context)
  });
  Object.assign(contents, doc);
  return contents;
}, "de_DescribeActivitiesCommand");
var de_DescribeCommentsCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode !== 200 && output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const contents = (0, import_smithy_client.map)({
    $metadata: deserializeMetadata(output)
  });
  const data = (0, import_smithy_client.expectNonNull)((0, import_smithy_client.expectObject)(await (0, import_core2.parseJsonBody)(output.body, context)), "body");
  const doc = (0, import_smithy_client.take)(data, {
    Comments: (_) => de_CommentList(_, context),
    Marker: import_smithy_client.expectString
  });
  Object.assign(contents, doc);
  return contents;
}, "de_DescribeCommentsCommand");
var de_DescribeDocumentVersionsCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode !== 200 && output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const contents = (0, import_smithy_client.map)({
    $metadata: deserializeMetadata(output)
  });
  const data = (0, import_smithy_client.expectNonNull)((0, import_smithy_client.expectObject)(await (0, import_core2.parseJsonBody)(output.body, context)), "body");
  const doc = (0, import_smithy_client.take)(data, {
    DocumentVersions: (_) => de_DocumentVersionMetadataList(_, context),
    Marker: import_smithy_client.expectString
  });
  Object.assign(contents, doc);
  return contents;
}, "de_DescribeDocumentVersionsCommand");
var de_DescribeFolderContentsCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode !== 200 && output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const contents = (0, import_smithy_client.map)({
    $metadata: deserializeMetadata(output)
  });
  const data = (0, import_smithy_client.expectNonNull)((0, import_smithy_client.expectObject)(await (0, import_core2.parseJsonBody)(output.body, context)), "body");
  const doc = (0, import_smithy_client.take)(data, {
    Documents: (_) => de_DocumentMetadataList(_, context),
    Folders: (_) => de_FolderMetadataList(_, context),
    Marker: import_smithy_client.expectString
  });
  Object.assign(contents, doc);
  return contents;
}, "de_DescribeFolderContentsCommand");
var de_DescribeGroupsCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode !== 200 && output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const contents = (0, import_smithy_client.map)({
    $metadata: deserializeMetadata(output)
  });
  const data = (0, import_smithy_client.expectNonNull)((0, import_smithy_client.expectObject)(await (0, import_core2.parseJsonBody)(output.body, context)), "body");
  const doc = (0, import_smithy_client.take)(data, {
    Groups: import_smithy_client._json,
    Marker: import_smithy_client.expectString
  });
  Object.assign(contents, doc);
  return contents;
}, "de_DescribeGroupsCommand");
var de_DescribeNotificationSubscriptionsCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode !== 200 && output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const contents = (0, import_smithy_client.map)({
    $metadata: deserializeMetadata(output)
  });
  const data = (0, import_smithy_client.expectNonNull)((0, import_smithy_client.expectObject)(await (0, import_core2.parseJsonBody)(output.body, context)), "body");
  const doc = (0, import_smithy_client.take)(data, {
    Marker: import_smithy_client.expectString,
    Subscriptions: import_smithy_client._json
  });
  Object.assign(contents, doc);
  return contents;
}, "de_DescribeNotificationSubscriptionsCommand");
var de_DescribeResourcePermissionsCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode !== 200 && output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const contents = (0, import_smithy_client.map)({
    $metadata: deserializeMetadata(output)
  });
  const data = (0, import_smithy_client.expectNonNull)((0, import_smithy_client.expectObject)(await (0, import_core2.parseJsonBody)(output.body, context)), "body");
  const doc = (0, import_smithy_client.take)(data, {
    Marker: import_smithy_client.expectString,
    Principals: import_smithy_client._json
  });
  Object.assign(contents, doc);
  return contents;
}, "de_DescribeResourcePermissionsCommand");
var de_DescribeRootFoldersCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode !== 200 && output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const contents = (0, import_smithy_client.map)({
    $metadata: deserializeMetadata(output)
  });
  const data = (0, import_smithy_client.expectNonNull)((0, import_smithy_client.expectObject)(await (0, import_core2.parseJsonBody)(output.body, context)), "body");
  const doc = (0, import_smithy_client.take)(data, {
    Folders: (_) => de_FolderMetadataList(_, context),
    Marker: import_smithy_client.expectString
  });
  Object.assign(contents, doc);
  return contents;
}, "de_DescribeRootFoldersCommand");
var de_DescribeUsersCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode !== 200 && output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const contents = (0, import_smithy_client.map)({
    $metadata: deserializeMetadata(output)
  });
  const data = (0, import_smithy_client.expectNonNull)((0, import_smithy_client.expectObject)(await (0, import_core2.parseJsonBody)(output.body, context)), "body");
  const doc = (0, import_smithy_client.take)(data, {
    Marker: import_smithy_client.expectString,
    TotalNumberOfUsers: import_smithy_client.expectLong,
    Users: (_) => de_OrganizationUserList(_, context)
  });
  Object.assign(contents, doc);
  return contents;
}, "de_DescribeUsersCommand");
var de_GetCurrentUserCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode !== 200 && output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const contents = (0, import_smithy_client.map)({
    $metadata: deserializeMetadata(output)
  });
  const data = (0, import_smithy_client.expectNonNull)((0, import_smithy_client.expectObject)(await (0, import_core2.parseJsonBody)(output.body, context)), "body");
  const doc = (0, import_smithy_client.take)(data, {
    User: (_) => de_User(_, context)
  });
  Object.assign(contents, doc);
  return contents;
}, "de_GetCurrentUserCommand");
var de_GetDocumentCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode !== 200 && output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const contents = (0, import_smithy_client.map)({
    $metadata: deserializeMetadata(output)
  });
  const data = (0, import_smithy_client.expectNonNull)((0, import_smithy_client.expectObject)(await (0, import_core2.parseJsonBody)(output.body, context)), "body");
  const doc = (0, import_smithy_client.take)(data, {
    CustomMetadata: import_smithy_client._json,
    Metadata: (_) => de_DocumentMetadata(_, context)
  });
  Object.assign(contents, doc);
  return contents;
}, "de_GetDocumentCommand");
var de_GetDocumentPathCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode !== 200 && output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const contents = (0, import_smithy_client.map)({
    $metadata: deserializeMetadata(output)
  });
  const data = (0, import_smithy_client.expectNonNull)((0, import_smithy_client.expectObject)(await (0, import_core2.parseJsonBody)(output.body, context)), "body");
  const doc = (0, import_smithy_client.take)(data, {
    Path: import_smithy_client._json
  });
  Object.assign(contents, doc);
  return contents;
}, "de_GetDocumentPathCommand");
var de_GetDocumentVersionCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode !== 200 && output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const contents = (0, import_smithy_client.map)({
    $metadata: deserializeMetadata(output)
  });
  const data = (0, import_smithy_client.expectNonNull)((0, import_smithy_client.expectObject)(await (0, import_core2.parseJsonBody)(output.body, context)), "body");
  const doc = (0, import_smithy_client.take)(data, {
    CustomMetadata: import_smithy_client._json,
    Metadata: (_) => de_DocumentVersionMetadata(_, context)
  });
  Object.assign(contents, doc);
  return contents;
}, "de_GetDocumentVersionCommand");
var de_GetFolderCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode !== 200 && output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const contents = (0, import_smithy_client.map)({
    $metadata: deserializeMetadata(output)
  });
  const data = (0, import_smithy_client.expectNonNull)((0, import_smithy_client.expectObject)(await (0, import_core2.parseJsonBody)(output.body, context)), "body");
  const doc = (0, import_smithy_client.take)(data, {
    CustomMetadata: import_smithy_client._json,
    Metadata: (_) => de_FolderMetadata(_, context)
  });
  Object.assign(contents, doc);
  return contents;
}, "de_GetFolderCommand");
var de_GetFolderPathCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode !== 200 && output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const contents = (0, import_smithy_client.map)({
    $metadata: deserializeMetadata(output)
  });
  const data = (0, import_smithy_client.expectNonNull)((0, import_smithy_client.expectObject)(await (0, import_core2.parseJsonBody)(output.body, context)), "body");
  const doc = (0, import_smithy_client.take)(data, {
    Path: import_smithy_client._json
  });
  Object.assign(contents, doc);
  return contents;
}, "de_GetFolderPathCommand");
var de_GetResourcesCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode !== 200 && output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const contents = (0, import_smithy_client.map)({
    $metadata: deserializeMetadata(output)
  });
  const data = (0, import_smithy_client.expectNonNull)((0, import_smithy_client.expectObject)(await (0, import_core2.parseJsonBody)(output.body, context)), "body");
  const doc = (0, import_smithy_client.take)(data, {
    Documents: (_) => de_DocumentMetadataList(_, context),
    Folders: (_) => de_FolderMetadataList(_, context),
    Marker: import_smithy_client.expectString
  });
  Object.assign(contents, doc);
  return contents;
}, "de_GetResourcesCommand");
var de_InitiateDocumentVersionUploadCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode !== 201 && output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const contents = (0, import_smithy_client.map)({
    $metadata: deserializeMetadata(output)
  });
  const data = (0, import_smithy_client.expectNonNull)((0, import_smithy_client.expectObject)(await (0, import_core2.parseJsonBody)(output.body, context)), "body");
  const doc = (0, import_smithy_client.take)(data, {
    Metadata: (_) => de_DocumentMetadata(_, context),
    UploadMetadata: import_smithy_client._json
  });
  Object.assign(contents, doc);
  return contents;
}, "de_InitiateDocumentVersionUploadCommand");
var de_RemoveAllResourcePermissionsCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode !== 204 && output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const contents = (0, import_smithy_client.map)({
    $metadata: deserializeMetadata(output)
  });
  await (0, import_smithy_client.collectBody)(output.body, context);
  return contents;
}, "de_RemoveAllResourcePermissionsCommand");
var de_RemoveResourcePermissionCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode !== 204 && output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const contents = (0, import_smithy_client.map)({
    $metadata: deserializeMetadata(output)
  });
  await (0, import_smithy_client.collectBody)(output.body, context);
  return contents;
}, "de_RemoveResourcePermissionCommand");
var de_RestoreDocumentVersionsCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode !== 204 && output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const contents = (0, import_smithy_client.map)({
    $metadata: deserializeMetadata(output)
  });
  await (0, import_smithy_client.collectBody)(output.body, context);
  return contents;
}, "de_RestoreDocumentVersionsCommand");
var de_SearchResourcesCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode !== 200 && output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const contents = (0, import_smithy_client.map)({
    $metadata: deserializeMetadata(output)
  });
  const data = (0, import_smithy_client.expectNonNull)((0, import_smithy_client.expectObject)(await (0, import_core2.parseJsonBody)(output.body, context)), "body");
  const doc = (0, import_smithy_client.take)(data, {
    Items: (_) => de_ResponseItemsList(_, context),
    Marker: import_smithy_client.expectString
  });
  Object.assign(contents, doc);
  return contents;
}, "de_SearchResourcesCommand");
var de_UpdateDocumentCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode !== 200 && output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const contents = (0, import_smithy_client.map)({
    $metadata: deserializeMetadata(output)
  });
  await (0, import_smithy_client.collectBody)(output.body, context);
  return contents;
}, "de_UpdateDocumentCommand");
var de_UpdateDocumentVersionCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode !== 200 && output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const contents = (0, import_smithy_client.map)({
    $metadata: deserializeMetadata(output)
  });
  await (0, import_smithy_client.collectBody)(output.body, context);
  return contents;
}, "de_UpdateDocumentVersionCommand");
var de_UpdateFolderCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode !== 200 && output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const contents = (0, import_smithy_client.map)({
    $metadata: deserializeMetadata(output)
  });
  await (0, import_smithy_client.collectBody)(output.body, context);
  return contents;
}, "de_UpdateFolderCommand");
var de_UpdateUserCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode !== 200 && output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const contents = (0, import_smithy_client.map)({
    $metadata: deserializeMetadata(output)
  });
  const data = (0, import_smithy_client.expectNonNull)((0, import_smithy_client.expectObject)(await (0, import_core2.parseJsonBody)(output.body, context)), "body");
  const doc = (0, import_smithy_client.take)(data, {
    User: (_) => de_User(_, context)
  });
  Object.assign(contents, doc);
  return contents;
}, "de_UpdateUserCommand");
var de_CommandError = /* @__PURE__ */ __name(async (output, context) => {
  const parsedOutput = {
    ...output,
    body: await (0, import_core2.parseJsonErrorBody)(output.body, context)
  };
  const errorCode = (0, import_core2.loadRestJsonErrorCode)(output, parsedOutput.body);
  switch (errorCode) {
    case "ConcurrentModificationException":
    case "com.amazonaws.workdocs#ConcurrentModificationException":
      throw await de_ConcurrentModificationExceptionRes(parsedOutput, context);
    case "EntityNotExistsException":
    case "com.amazonaws.workdocs#EntityNotExistsException":
      throw await de_EntityNotExistsExceptionRes(parsedOutput, context);
    case "FailedDependencyException":
    case "com.amazonaws.workdocs#FailedDependencyException":
      throw await de_FailedDependencyExceptionRes(parsedOutput, context);
    case "ProhibitedStateException":
    case "com.amazonaws.workdocs#ProhibitedStateException":
      throw await de_ProhibitedStateExceptionRes(parsedOutput, context);
    case "ServiceUnavailableException":
    case "com.amazonaws.workdocs#ServiceUnavailableException":
      throw await de_ServiceUnavailableExceptionRes(parsedOutput, context);
    case "UnauthorizedOperationException":
    case "com.amazonaws.workdocs#UnauthorizedOperationException":
      throw await de_UnauthorizedOperationExceptionRes(parsedOutput, context);
    case "UnauthorizedResourceAccessException":
    case "com.amazonaws.workdocs#UnauthorizedResourceAccessException":
      throw await de_UnauthorizedResourceAccessExceptionRes(parsedOutput, context);
    case "DocumentLockedForCommentsException":
    case "com.amazonaws.workdocs#DocumentLockedForCommentsException":
      throw await de_DocumentLockedForCommentsExceptionRes(parsedOutput, context);
    case "InvalidCommentOperationException":
    case "com.amazonaws.workdocs#InvalidCommentOperationException":
      throw await de_InvalidCommentOperationExceptionRes(parsedOutput, context);
    case "CustomMetadataLimitExceededException":
    case "com.amazonaws.workdocs#CustomMetadataLimitExceededException":
      throw await de_CustomMetadataLimitExceededExceptionRes(parsedOutput, context);
    case "ConflictingOperationException":
    case "com.amazonaws.workdocs#ConflictingOperationException":
      throw await de_ConflictingOperationExceptionRes(parsedOutput, context);
    case "EntityAlreadyExistsException":
    case "com.amazonaws.workdocs#EntityAlreadyExistsException":
      throw await de_EntityAlreadyExistsExceptionRes(parsedOutput, context);
    case "LimitExceededException":
    case "com.amazonaws.workdocs#LimitExceededException":
      throw await de_LimitExceededExceptionRes(parsedOutput, context);
    case "TooManyLabelsException":
    case "com.amazonaws.workdocs#TooManyLabelsException":
      throw await de_TooManyLabelsExceptionRes(parsedOutput, context);
    case "InvalidArgumentException":
    case "com.amazonaws.workdocs#InvalidArgumentException":
      throw await de_InvalidArgumentExceptionRes(parsedOutput, context);
    case "TooManySubscriptionsException":
    case "com.amazonaws.workdocs#TooManySubscriptionsException":
      throw await de_TooManySubscriptionsExceptionRes(parsedOutput, context);
    case "InvalidOperationException":
    case "com.amazonaws.workdocs#InvalidOperationException":
      throw await de_InvalidOperationExceptionRes(parsedOutput, context);
    case "InvalidPasswordException":
    case "com.amazonaws.workdocs#InvalidPasswordException":
      throw await de_InvalidPasswordExceptionRes(parsedOutput, context);
    case "RequestedEntityTooLargeException":
    case "com.amazonaws.workdocs#RequestedEntityTooLargeException":
      throw await de_RequestedEntityTooLargeExceptionRes(parsedOutput, context);
    case "DraftUploadOutOfSyncException":
    case "com.amazonaws.workdocs#DraftUploadOutOfSyncException":
      throw await de_DraftUploadOutOfSyncExceptionRes(parsedOutput, context);
    case "ResourceAlreadyCheckedOutException":
    case "com.amazonaws.workdocs#ResourceAlreadyCheckedOutException":
      throw await de_ResourceAlreadyCheckedOutExceptionRes(parsedOutput, context);
    case "StorageLimitExceededException":
    case "com.amazonaws.workdocs#StorageLimitExceededException":
      throw await de_StorageLimitExceededExceptionRes(parsedOutput, context);
    case "StorageLimitWillExceedException":
    case "com.amazonaws.workdocs#StorageLimitWillExceedException":
      throw await de_StorageLimitWillExceedExceptionRes(parsedOutput, context);
    case "DeactivatingLastSystemUserException":
    case "com.amazonaws.workdocs#DeactivatingLastSystemUserException":
      throw await de_DeactivatingLastSystemUserExceptionRes(parsedOutput, context);
    case "IllegalUserStateException":
    case "com.amazonaws.workdocs#IllegalUserStateException":
      throw await de_IllegalUserStateExceptionRes(parsedOutput, context);
    default:
      const parsedBody = parsedOutput.body;
      return throwDefaultError({
        output,
        parsedBody,
        errorCode
      });
  }
}, "de_CommandError");
var throwDefaultError = (0, import_smithy_client.withBaseException)(WorkDocsServiceException);
var de_ConcurrentModificationExceptionRes = /* @__PURE__ */ __name(async (parsedOutput, context) => {
  const contents = (0, import_smithy_client.map)({});
  const data = parsedOutput.body;
  const doc = (0, import_smithy_client.take)(data, {
    Message: import_smithy_client.expectString
  });
  Object.assign(contents, doc);
  const exception = new ConcurrentModificationException({
    $metadata: deserializeMetadata(parsedOutput),
    ...contents
  });
  return (0, import_smithy_client.decorateServiceException)(exception, parsedOutput.body);
}, "de_ConcurrentModificationExceptionRes");
var de_ConflictingOperationExceptionRes = /* @__PURE__ */ __name(async (parsedOutput, context) => {
  const contents = (0, import_smithy_client.map)({});
  const data = parsedOutput.body;
  const doc = (0, import_smithy_client.take)(data, {
    Message: import_smithy_client.expectString
  });
  Object.assign(contents, doc);
  const exception = new ConflictingOperationException({
    $metadata: deserializeMetadata(parsedOutput),
    ...contents
  });
  return (0, import_smithy_client.decorateServiceException)(exception, parsedOutput.body);
}, "de_ConflictingOperationExceptionRes");
var de_CustomMetadataLimitExceededExceptionRes = /* @__PURE__ */ __name(async (parsedOutput, context) => {
  const contents = (0, import_smithy_client.map)({});
  const data = parsedOutput.body;
  const doc = (0, import_smithy_client.take)(data, {
    Message: import_smithy_client.expectString
  });
  Object.assign(contents, doc);
  const exception = new CustomMetadataLimitExceededException({
    $metadata: deserializeMetadata(parsedOutput),
    ...contents
  });
  return (0, import_smithy_client.decorateServiceException)(exception, parsedOutput.body);
}, "de_CustomMetadataLimitExceededExceptionRes");
var de_DeactivatingLastSystemUserExceptionRes = /* @__PURE__ */ __name(async (parsedOutput, context) => {
  const contents = (0, import_smithy_client.map)({});
  const data = parsedOutput.body;
  const doc = (0, import_smithy_client.take)(data, {
    Code: import_smithy_client.expectString,
    Message: import_smithy_client.expectString
  });
  Object.assign(contents, doc);
  const exception = new DeactivatingLastSystemUserException({
    $metadata: deserializeMetadata(parsedOutput),
    ...contents
  });
  return (0, import_smithy_client.decorateServiceException)(exception, parsedOutput.body);
}, "de_DeactivatingLastSystemUserExceptionRes");
var de_DocumentLockedForCommentsExceptionRes = /* @__PURE__ */ __name(async (parsedOutput, context) => {
  const contents = (0, import_smithy_client.map)({});
  const data = parsedOutput.body;
  const doc = (0, import_smithy_client.take)(data, {
    Message: import_smithy_client.expectString
  });
  Object.assign(contents, doc);
  const exception = new DocumentLockedForCommentsException({
    $metadata: deserializeMetadata(parsedOutput),
    ...contents
  });
  return (0, import_smithy_client.decorateServiceException)(exception, parsedOutput.body);
}, "de_DocumentLockedForCommentsExceptionRes");
var de_DraftUploadOutOfSyncExceptionRes = /* @__PURE__ */ __name(async (parsedOutput, context) => {
  const contents = (0, import_smithy_client.map)({});
  const data = parsedOutput.body;
  const doc = (0, import_smithy_client.take)(data, {
    Message: import_smithy_client.expectString
  });
  Object.assign(contents, doc);
  const exception = new DraftUploadOutOfSyncException({
    $metadata: deserializeMetadata(parsedOutput),
    ...contents
  });
  return (0, import_smithy_client.decorateServiceException)(exception, parsedOutput.body);
}, "de_DraftUploadOutOfSyncExceptionRes");
var de_EntityAlreadyExistsExceptionRes = /* @__PURE__ */ __name(async (parsedOutput, context) => {
  const contents = (0, import_smithy_client.map)({});
  const data = parsedOutput.body;
  const doc = (0, import_smithy_client.take)(data, {
    Message: import_smithy_client.expectString
  });
  Object.assign(contents, doc);
  const exception = new EntityAlreadyExistsException({
    $metadata: deserializeMetadata(parsedOutput),
    ...contents
  });
  return (0, import_smithy_client.decorateServiceException)(exception, parsedOutput.body);
}, "de_EntityAlreadyExistsExceptionRes");
var de_EntityNotExistsExceptionRes = /* @__PURE__ */ __name(async (parsedOutput, context) => {
  const contents = (0, import_smithy_client.map)({});
  const data = parsedOutput.body;
  const doc = (0, import_smithy_client.take)(data, {
    EntityIds: import_smithy_client._json,
    Message: import_smithy_client.expectString
  });
  Object.assign(contents, doc);
  const exception = new EntityNotExistsException({
    $metadata: deserializeMetadata(parsedOutput),
    ...contents
  });
  return (0, import_smithy_client.decorateServiceException)(exception, parsedOutput.body);
}, "de_EntityNotExistsExceptionRes");
var de_FailedDependencyExceptionRes = /* @__PURE__ */ __name(async (parsedOutput, context) => {
  const contents = (0, import_smithy_client.map)({});
  const data = parsedOutput.body;
  const doc = (0, import_smithy_client.take)(data, {
    Message: import_smithy_client.expectString
  });
  Object.assign(contents, doc);
  const exception = new FailedDependencyException({
    $metadata: deserializeMetadata(parsedOutput),
    ...contents
  });
  return (0, import_smithy_client.decorateServiceException)(exception, parsedOutput.body);
}, "de_FailedDependencyExceptionRes");
var de_IllegalUserStateExceptionRes = /* @__PURE__ */ __name(async (parsedOutput, context) => {
  const contents = (0, import_smithy_client.map)({});
  const data = parsedOutput.body;
  const doc = (0, import_smithy_client.take)(data, {
    Message: import_smithy_client.expectString
  });
  Object.assign(contents, doc);
  const exception = new IllegalUserStateException({
    $metadata: deserializeMetadata(parsedOutput),
    ...contents
  });
  return (0, import_smithy_client.decorateServiceException)(exception, parsedOutput.body);
}, "de_IllegalUserStateExceptionRes");
var de_InvalidArgumentExceptionRes = /* @__PURE__ */ __name(async (parsedOutput, context) => {
  const contents = (0, import_smithy_client.map)({});
  const data = parsedOutput.body;
  const doc = (0, import_smithy_client.take)(data, {
    Message: import_smithy_client.expectString
  });
  Object.assign(contents, doc);
  const exception = new InvalidArgumentException({
    $metadata: deserializeMetadata(parsedOutput),
    ...contents
  });
  return (0, import_smithy_client.decorateServiceException)(exception, parsedOutput.body);
}, "de_InvalidArgumentExceptionRes");
var de_InvalidCommentOperationExceptionRes = /* @__PURE__ */ __name(async (parsedOutput, context) => {
  const contents = (0, import_smithy_client.map)({});
  const data = parsedOutput.body;
  const doc = (0, import_smithy_client.take)(data, {
    Message: import_smithy_client.expectString
  });
  Object.assign(contents, doc);
  const exception = new InvalidCommentOperationException({
    $metadata: deserializeMetadata(parsedOutput),
    ...contents
  });
  return (0, import_smithy_client.decorateServiceException)(exception, parsedOutput.body);
}, "de_InvalidCommentOperationExceptionRes");
var de_InvalidOperationExceptionRes = /* @__PURE__ */ __name(async (parsedOutput, context) => {
  const contents = (0, import_smithy_client.map)({});
  const data = parsedOutput.body;
  const doc = (0, import_smithy_client.take)(data, {
    Message: import_smithy_client.expectString
  });
  Object.assign(contents, doc);
  const exception = new InvalidOperationException({
    $metadata: deserializeMetadata(parsedOutput),
    ...contents
  });
  return (0, import_smithy_client.decorateServiceException)(exception, parsedOutput.body);
}, "de_InvalidOperationExceptionRes");
var de_InvalidPasswordExceptionRes = /* @__PURE__ */ __name(async (parsedOutput, context) => {
  const contents = (0, import_smithy_client.map)({});
  const data = parsedOutput.body;
  const doc = (0, import_smithy_client.take)(data, {
    Message: import_smithy_client.expectString
  });
  Object.assign(contents, doc);
  const exception = new InvalidPasswordException({
    $metadata: deserializeMetadata(parsedOutput),
    ...contents
  });
  return (0, import_smithy_client.decorateServiceException)(exception, parsedOutput.body);
}, "de_InvalidPasswordExceptionRes");
var de_LimitExceededExceptionRes = /* @__PURE__ */ __name(async (parsedOutput, context) => {
  const contents = (0, import_smithy_client.map)({});
  const data = parsedOutput.body;
  const doc = (0, import_smithy_client.take)(data, {
    Message: import_smithy_client.expectString
  });
  Object.assign(contents, doc);
  const exception = new LimitExceededException({
    $metadata: deserializeMetadata(parsedOutput),
    ...contents
  });
  return (0, import_smithy_client.decorateServiceException)(exception, parsedOutput.body);
}, "de_LimitExceededExceptionRes");
var de_ProhibitedStateExceptionRes = /* @__PURE__ */ __name(async (parsedOutput, context) => {
  const contents = (0, import_smithy_client.map)({});
  const data = parsedOutput.body;
  const doc = (0, import_smithy_client.take)(data, {
    Message: import_smithy_client.expectString
  });
  Object.assign(contents, doc);
  const exception = new ProhibitedStateException({
    $metadata: deserializeMetadata(parsedOutput),
    ...contents
  });
  return (0, import_smithy_client.decorateServiceException)(exception, parsedOutput.body);
}, "de_ProhibitedStateExceptionRes");
var de_RequestedEntityTooLargeExceptionRes = /* @__PURE__ */ __name(async (parsedOutput, context) => {
  const contents = (0, import_smithy_client.map)({});
  const data = parsedOutput.body;
  const doc = (0, import_smithy_client.take)(data, {
    Message: import_smithy_client.expectString
  });
  Object.assign(contents, doc);
  const exception = new RequestedEntityTooLargeException({
    $metadata: deserializeMetadata(parsedOutput),
    ...contents
  });
  return (0, import_smithy_client.decorateServiceException)(exception, parsedOutput.body);
}, "de_RequestedEntityTooLargeExceptionRes");
var de_ResourceAlreadyCheckedOutExceptionRes = /* @__PURE__ */ __name(async (parsedOutput, context) => {
  const contents = (0, import_smithy_client.map)({});
  const data = parsedOutput.body;
  const doc = (0, import_smithy_client.take)(data, {
    Message: import_smithy_client.expectString
  });
  Object.assign(contents, doc);
  const exception = new ResourceAlreadyCheckedOutException({
    $metadata: deserializeMetadata(parsedOutput),
    ...contents
  });
  return (0, import_smithy_client.decorateServiceException)(exception, parsedOutput.body);
}, "de_ResourceAlreadyCheckedOutExceptionRes");
var de_ServiceUnavailableExceptionRes = /* @__PURE__ */ __name(async (parsedOutput, context) => {
  const contents = (0, import_smithy_client.map)({});
  const data = parsedOutput.body;
  const doc = (0, import_smithy_client.take)(data, {
    Message: import_smithy_client.expectString
  });
  Object.assign(contents, doc);
  const exception = new ServiceUnavailableException({
    $metadata: deserializeMetadata(parsedOutput),
    ...contents
  });
  return (0, import_smithy_client.decorateServiceException)(exception, parsedOutput.body);
}, "de_ServiceUnavailableExceptionRes");
var de_StorageLimitExceededExceptionRes = /* @__PURE__ */ __name(async (parsedOutput, context) => {
  const contents = (0, import_smithy_client.map)({});
  const data = parsedOutput.body;
  const doc = (0, import_smithy_client.take)(data, {
    Message: import_smithy_client.expectString
  });
  Object.assign(contents, doc);
  const exception = new StorageLimitExceededException({
    $metadata: deserializeMetadata(parsedOutput),
    ...contents
  });
  return (0, import_smithy_client.decorateServiceException)(exception, parsedOutput.body);
}, "de_StorageLimitExceededExceptionRes");
var de_StorageLimitWillExceedExceptionRes = /* @__PURE__ */ __name(async (parsedOutput, context) => {
  const contents = (0, import_smithy_client.map)({});
  const data = parsedOutput.body;
  const doc = (0, import_smithy_client.take)(data, {
    Message: import_smithy_client.expectString
  });
  Object.assign(contents, doc);
  const exception = new StorageLimitWillExceedException({
    $metadata: deserializeMetadata(parsedOutput),
    ...contents
  });
  return (0, import_smithy_client.decorateServiceException)(exception, parsedOutput.body);
}, "de_StorageLimitWillExceedExceptionRes");
var de_TooManyLabelsExceptionRes = /* @__PURE__ */ __name(async (parsedOutput, context) => {
  const contents = (0, import_smithy_client.map)({});
  const data = parsedOutput.body;
  const doc = (0, import_smithy_client.take)(data, {
    Message: import_smithy_client.expectString
  });
  Object.assign(contents, doc);
  const exception = new TooManyLabelsException({
    $metadata: deserializeMetadata(parsedOutput),
    ...contents
  });
  return (0, import_smithy_client.decorateServiceException)(exception, parsedOutput.body);
}, "de_TooManyLabelsExceptionRes");
var de_TooManySubscriptionsExceptionRes = /* @__PURE__ */ __name(async (parsedOutput, context) => {
  const contents = (0, import_smithy_client.map)({});
  const data = parsedOutput.body;
  const doc = (0, import_smithy_client.take)(data, {
    Message: import_smithy_client.expectString
  });
  Object.assign(contents, doc);
  const exception = new TooManySubscriptionsException({
    $metadata: deserializeMetadata(parsedOutput),
    ...contents
  });
  return (0, import_smithy_client.decorateServiceException)(exception, parsedOutput.body);
}, "de_TooManySubscriptionsExceptionRes");
var de_UnauthorizedOperationExceptionRes = /* @__PURE__ */ __name(async (parsedOutput, context) => {
  const contents = (0, import_smithy_client.map)({});
  const data = parsedOutput.body;
  const doc = (0, import_smithy_client.take)(data, {
    Code: import_smithy_client.expectString,
    Message: import_smithy_client.expectString
  });
  Object.assign(contents, doc);
  const exception = new UnauthorizedOperationException({
    $metadata: deserializeMetadata(parsedOutput),
    ...contents
  });
  return (0, import_smithy_client.decorateServiceException)(exception, parsedOutput.body);
}, "de_UnauthorizedOperationExceptionRes");
var de_UnauthorizedResourceAccessExceptionRes = /* @__PURE__ */ __name(async (parsedOutput, context) => {
  const contents = (0, import_smithy_client.map)({});
  const data = parsedOutput.body;
  const doc = (0, import_smithy_client.take)(data, {
    Message: import_smithy_client.expectString
  });
  Object.assign(contents, doc);
  const exception = new UnauthorizedResourceAccessException({
    $metadata: deserializeMetadata(parsedOutput),
    ...contents
  });
  return (0, import_smithy_client.decorateServiceException)(exception, parsedOutput.body);
}, "de_UnauthorizedResourceAccessExceptionRes");
var se_DateRangeType = /* @__PURE__ */ __name((input, context) => {
  return (0, import_smithy_client.take)(input, {
    EndValue: (_) => Math.round(_.getTime() / 1e3),
    StartValue: (_) => Math.round(_.getTime() / 1e3)
  });
}, "se_DateRangeType");
var se_Filters = /* @__PURE__ */ __name((input, context) => {
  return (0, import_smithy_client.take)(input, {
    AncestorIds: import_smithy_client._json,
    ContentCategories: import_smithy_client._json,
    CreatedRange: (_) => se_DateRangeType(_, context),
    Labels: import_smithy_client._json,
    ModifiedRange: (_) => se_DateRangeType(_, context),
    Principals: import_smithy_client._json,
    ResourceTypes: import_smithy_client._json,
    SearchCollectionTypes: import_smithy_client._json,
    SizeRange: import_smithy_client._json,
    TextLocales: import_smithy_client._json
  });
}, "se_Filters");
var de_Activity = /* @__PURE__ */ __name((output, context) => {
  return (0, import_smithy_client.take)(output, {
    CommentMetadata: (_) => de_CommentMetadata(_, context),
    Initiator: import_smithy_client._json,
    IsIndirectActivity: import_smithy_client.expectBoolean,
    OrganizationId: import_smithy_client.expectString,
    OriginalParent: import_smithy_client._json,
    Participants: import_smithy_client._json,
    ResourceMetadata: import_smithy_client._json,
    TimeStamp: (_) => (0, import_smithy_client.expectNonNull)((0, import_smithy_client.parseEpochTimestamp)((0, import_smithy_client.expectNumber)(_))),
    Type: import_smithy_client.expectString
  });
}, "de_Activity");
var de_Comment = /* @__PURE__ */ __name((output, context) => {
  return (0, import_smithy_client.take)(output, {
    CommentId: import_smithy_client.expectString,
    Contributor: (_) => de_User(_, context),
    CreatedTimestamp: (_) => (0, import_smithy_client.expectNonNull)((0, import_smithy_client.parseEpochTimestamp)((0, import_smithy_client.expectNumber)(_))),
    ParentId: import_smithy_client.expectString,
    RecipientId: import_smithy_client.expectString,
    Status: import_smithy_client.expectString,
    Text: import_smithy_client.expectString,
    ThreadId: import_smithy_client.expectString,
    Visibility: import_smithy_client.expectString
  });
}, "de_Comment");
var de_CommentList = /* @__PURE__ */ __name((output, context) => {
  const retVal = (output || []).filter((e) => e != null).map((entry) => {
    return de_Comment(entry, context);
  });
  return retVal;
}, "de_CommentList");
var de_CommentMetadata = /* @__PURE__ */ __name((output, context) => {
  return (0, import_smithy_client.take)(output, {
    CommentId: import_smithy_client.expectString,
    CommentStatus: import_smithy_client.expectString,
    Contributor: (_) => de_User(_, context),
    ContributorId: import_smithy_client.expectString,
    CreatedTimestamp: (_) => (0, import_smithy_client.expectNonNull)((0, import_smithy_client.parseEpochTimestamp)((0, import_smithy_client.expectNumber)(_))),
    RecipientId: import_smithy_client.expectString
  });
}, "de_CommentMetadata");
var de_DocumentMetadata = /* @__PURE__ */ __name((output, context) => {
  return (0, import_smithy_client.take)(output, {
    CreatedTimestamp: (_) => (0, import_smithy_client.expectNonNull)((0, import_smithy_client.parseEpochTimestamp)((0, import_smithy_client.expectNumber)(_))),
    CreatorId: import_smithy_client.expectString,
    Id: import_smithy_client.expectString,
    Labels: import_smithy_client._json,
    LatestVersionMetadata: (_) => de_DocumentVersionMetadata(_, context),
    ModifiedTimestamp: (_) => (0, import_smithy_client.expectNonNull)((0, import_smithy_client.parseEpochTimestamp)((0, import_smithy_client.expectNumber)(_))),
    ParentFolderId: import_smithy_client.expectString,
    ResourceState: import_smithy_client.expectString
  });
}, "de_DocumentMetadata");
var de_DocumentMetadataList = /* @__PURE__ */ __name((output, context) => {
  const retVal = (output || []).filter((e) => e != null).map((entry) => {
    return de_DocumentMetadata(entry, context);
  });
  return retVal;
}, "de_DocumentMetadataList");
var de_DocumentVersionMetadata = /* @__PURE__ */ __name((output, context) => {
  return (0, import_smithy_client.take)(output, {
    ContentCreatedTimestamp: (_) => (0, import_smithy_client.expectNonNull)((0, import_smithy_client.parseEpochTimestamp)((0, import_smithy_client.expectNumber)(_))),
    ContentModifiedTimestamp: (_) => (0, import_smithy_client.expectNonNull)((0, import_smithy_client.parseEpochTimestamp)((0, import_smithy_client.expectNumber)(_))),
    ContentType: import_smithy_client.expectString,
    CreatedTimestamp: (_) => (0, import_smithy_client.expectNonNull)((0, import_smithy_client.parseEpochTimestamp)((0, import_smithy_client.expectNumber)(_))),
    CreatorId: import_smithy_client.expectString,
    Id: import_smithy_client.expectString,
    ModifiedTimestamp: (_) => (0, import_smithy_client.expectNonNull)((0, import_smithy_client.parseEpochTimestamp)((0, import_smithy_client.expectNumber)(_))),
    Name: import_smithy_client.expectString,
    Signature: import_smithy_client.expectString,
    Size: import_smithy_client.expectLong,
    Source: import_smithy_client._json,
    Status: import_smithy_client.expectString,
    Thumbnail: import_smithy_client._json
  });
}, "de_DocumentVersionMetadata");
var de_DocumentVersionMetadataList = /* @__PURE__ */ __name((output, context) => {
  const retVal = (output || []).filter((e) => e != null).map((entry) => {
    return de_DocumentVersionMetadata(entry, context);
  });
  return retVal;
}, "de_DocumentVersionMetadataList");
var de_FolderMetadata = /* @__PURE__ */ __name((output, context) => {
  return (0, import_smithy_client.take)(output, {
    CreatedTimestamp: (_) => (0, import_smithy_client.expectNonNull)((0, import_smithy_client.parseEpochTimestamp)((0, import_smithy_client.expectNumber)(_))),
    CreatorId: import_smithy_client.expectString,
    Id: import_smithy_client.expectString,
    Labels: import_smithy_client._json,
    LatestVersionSize: import_smithy_client.expectLong,
    ModifiedTimestamp: (_) => (0, import_smithy_client.expectNonNull)((0, import_smithy_client.parseEpochTimestamp)((0, import_smithy_client.expectNumber)(_))),
    Name: import_smithy_client.expectString,
    ParentFolderId: import_smithy_client.expectString,
    ResourceState: import_smithy_client.expectString,
    Signature: import_smithy_client.expectString,
    Size: import_smithy_client.expectLong
  });
}, "de_FolderMetadata");
var de_FolderMetadataList = /* @__PURE__ */ __name((output, context) => {
  const retVal = (output || []).filter((e) => e != null).map((entry) => {
    return de_FolderMetadata(entry, context);
  });
  return retVal;
}, "de_FolderMetadataList");
var de_OrganizationUserList = /* @__PURE__ */ __name((output, context) => {
  const retVal = (output || []).filter((e) => e != null).map((entry) => {
    return de_User(entry, context);
  });
  return retVal;
}, "de_OrganizationUserList");
var de_ResponseItem = /* @__PURE__ */ __name((output, context) => {
  return (0, import_smithy_client.take)(output, {
    CommentMetadata: (_) => de_CommentMetadata(_, context),
    DocumentMetadata: (_) => de_DocumentMetadata(_, context),
    DocumentVersionMetadata: (_) => de_DocumentVersionMetadata(_, context),
    FolderMetadata: (_) => de_FolderMetadata(_, context),
    ResourceType: import_smithy_client.expectString,
    WebUrl: import_smithy_client.expectString
  });
}, "de_ResponseItem");
var de_ResponseItemsList = /* @__PURE__ */ __name((output, context) => {
  const retVal = (output || []).filter((e) => e != null).map((entry) => {
    return de_ResponseItem(entry, context);
  });
  return retVal;
}, "de_ResponseItemsList");
var de_User = /* @__PURE__ */ __name((output, context) => {
  return (0, import_smithy_client.take)(output, {
    CreatedTimestamp: (_) => (0, import_smithy_client.expectNonNull)((0, import_smithy_client.parseEpochTimestamp)((0, import_smithy_client.expectNumber)(_))),
    EmailAddress: import_smithy_client.expectString,
    GivenName: import_smithy_client.expectString,
    Id: import_smithy_client.expectString,
    Locale: import_smithy_client.expectString,
    ModifiedTimestamp: (_) => (0, import_smithy_client.expectNonNull)((0, import_smithy_client.parseEpochTimestamp)((0, import_smithy_client.expectNumber)(_))),
    OrganizationId: import_smithy_client.expectString,
    RecycleBinFolderId: import_smithy_client.expectString,
    RootFolderId: import_smithy_client.expectString,
    Status: import_smithy_client.expectString,
    Storage: import_smithy_client._json,
    Surname: import_smithy_client.expectString,
    TimeZoneId: import_smithy_client.expectString,
    Type: import_smithy_client.expectString,
    Username: import_smithy_client.expectString
  });
}, "de_User");
var de_UserActivities = /* @__PURE__ */ __name((output, context) => {
  const retVal = (output || []).filter((e) => e != null).map((entry) => {
    return de_Activity(entry, context);
  });
  return retVal;
}, "de_UserActivities");
var deserializeMetadata = /* @__PURE__ */ __name((output) => ({
  httpStatusCode: output.statusCode,
  requestId: output.headers["x-amzn-requestid"] ?? output.headers["x-amzn-request-id"] ?? output.headers["x-amz-request-id"],
  extendedRequestId: output.headers["x-amz-id-2"],
  cfId: output.headers["x-amz-cf-id"]
}), "deserializeMetadata");
var isSerializableHeaderValue = /* @__PURE__ */ __name((value) => value !== void 0 && value !== null && value !== "" && (!Object.getOwnPropertyNames(value).includes("length") || value.length != 0) && (!Object.getOwnPropertyNames(value).includes("size") || value.size != 0), "isSerializableHeaderValue");
var _AT = "AuthenticationToken";
var _ATc = "ActivityTypes";
var _CT = "CollectionType";
var _DA = "DeleteAll";
var _DPV = "DeletePriorVersions";
var _ET = "EndTime";
var _F = "Fields";
var _I = "Include";
var _ICM = "IncludeCustomMetadata";
var _IIA = "IncludeIndirectActivities";
var _K = "Keys";
var _L = "Labels";
var _Li = "Limit";
var _M = "Marker";
var _O = "Order";
var _OI = "OrganizationId";
var _PI = "PrincipalId";
var _PT = "PrincipalType";
var _Q = "Query";
var _RI = "ResourceId";
var _S = "Sort";
var _SQ = "SearchQuery";
var _ST = "StartTime";
var _T = "Type";
var _UI = "UserId";
var _UIs = "UserIds";
var _VI = "VersionId";
var _a = "authentication";
var _aT = "activityTypes";
var _cT = "collectionType";
var _dA = "deleteAll";
var _dPV = "deletePriorVersions";
var _eT = "endTime";
var _f = "fields";
var _i = "include";
var _iCM = "includeCustomMetadata";
var _iIA = "includeIndirectActivities";
var _k = "keys";
var _l = "labels";
var _li = "limit";
var _m = "marker";
var _o = "order";
var _oI = "organizationId";
var _pI = "principalId";
var _q = "query";
var _rI = "resourceId";
var _s = "sort";
var _sQ = "searchQuery";
var _sT = "startTime";
var _t = "type";
var _uI = "userId";
var _uIs = "userIds";
var _v = "versionid";
var _vI = "versionId";

// src/commands/AbortDocumentVersionUploadCommand.ts
var _AbortDocumentVersionUploadCommand = class _AbortDocumentVersionUploadCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AWSGorillaBoyService", "AbortDocumentVersionUpload", {}).n("WorkDocsClient", "AbortDocumentVersionUploadCommand").f(AbortDocumentVersionUploadRequestFilterSensitiveLog, void 0).ser(se_AbortDocumentVersionUploadCommand).de(de_AbortDocumentVersionUploadCommand).build() {
};
__name(_AbortDocumentVersionUploadCommand, "AbortDocumentVersionUploadCommand");
var AbortDocumentVersionUploadCommand = _AbortDocumentVersionUploadCommand;

// src/commands/ActivateUserCommand.ts




var _ActivateUserCommand = class _ActivateUserCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AWSGorillaBoyService", "ActivateUser", {}).n("WorkDocsClient", "ActivateUserCommand").f(ActivateUserRequestFilterSensitiveLog, ActivateUserResponseFilterSensitiveLog).ser(se_ActivateUserCommand).de(de_ActivateUserCommand).build() {
};
__name(_ActivateUserCommand, "ActivateUserCommand");
var ActivateUserCommand = _ActivateUserCommand;

// src/commands/AddResourcePermissionsCommand.ts




var _AddResourcePermissionsCommand = class _AddResourcePermissionsCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AWSGorillaBoyService", "AddResourcePermissions", {}).n("WorkDocsClient", "AddResourcePermissionsCommand").f(AddResourcePermissionsRequestFilterSensitiveLog, AddResourcePermissionsResponseFilterSensitiveLog).ser(se_AddResourcePermissionsCommand).de(de_AddResourcePermissionsCommand).build() {
};
__name(_AddResourcePermissionsCommand, "AddResourcePermissionsCommand");
var AddResourcePermissionsCommand = _AddResourcePermissionsCommand;

// src/commands/CreateCommentCommand.ts




var _CreateCommentCommand = class _CreateCommentCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AWSGorillaBoyService", "CreateComment", {}).n("WorkDocsClient", "CreateCommentCommand").f(CreateCommentRequestFilterSensitiveLog, CreateCommentResponseFilterSensitiveLog).ser(se_CreateCommentCommand).de(de_CreateCommentCommand).build() {
};
__name(_CreateCommentCommand, "CreateCommentCommand");
var CreateCommentCommand = _CreateCommentCommand;

// src/commands/CreateCustomMetadataCommand.ts




var _CreateCustomMetadataCommand = class _CreateCustomMetadataCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AWSGorillaBoyService", "CreateCustomMetadata", {}).n("WorkDocsClient", "CreateCustomMetadataCommand").f(CreateCustomMetadataRequestFilterSensitiveLog, void 0).ser(se_CreateCustomMetadataCommand).de(de_CreateCustomMetadataCommand).build() {
};
__name(_CreateCustomMetadataCommand, "CreateCustomMetadataCommand");
var CreateCustomMetadataCommand = _CreateCustomMetadataCommand;

// src/commands/CreateFolderCommand.ts




var _CreateFolderCommand = class _CreateFolderCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AWSGorillaBoyService", "CreateFolder", {}).n("WorkDocsClient", "CreateFolderCommand").f(CreateFolderRequestFilterSensitiveLog, CreateFolderResponseFilterSensitiveLog).ser(se_CreateFolderCommand).de(de_CreateFolderCommand).build() {
};
__name(_CreateFolderCommand, "CreateFolderCommand");
var CreateFolderCommand = _CreateFolderCommand;

// src/commands/CreateLabelsCommand.ts




var _CreateLabelsCommand = class _CreateLabelsCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AWSGorillaBoyService", "CreateLabels", {}).n("WorkDocsClient", "CreateLabelsCommand").f(CreateLabelsRequestFilterSensitiveLog, void 0).ser(se_CreateLabelsCommand).de(de_CreateLabelsCommand).build() {
};
__name(_CreateLabelsCommand, "CreateLabelsCommand");
var CreateLabelsCommand = _CreateLabelsCommand;

// src/commands/CreateNotificationSubscriptionCommand.ts




var _CreateNotificationSubscriptionCommand = class _CreateNotificationSubscriptionCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AWSGorillaBoyService", "CreateNotificationSubscription", {}).n("WorkDocsClient", "CreateNotificationSubscriptionCommand").f(void 0, void 0).ser(se_CreateNotificationSubscriptionCommand).de(de_CreateNotificationSubscriptionCommand).build() {
};
__name(_CreateNotificationSubscriptionCommand, "CreateNotificationSubscriptionCommand");
var CreateNotificationSubscriptionCommand = _CreateNotificationSubscriptionCommand;

// src/commands/CreateUserCommand.ts




var _CreateUserCommand = class _CreateUserCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AWSGorillaBoyService", "CreateUser", {}).n("WorkDocsClient", "CreateUserCommand").f(CreateUserRequestFilterSensitiveLog, CreateUserResponseFilterSensitiveLog).ser(se_CreateUserCommand).de(de_CreateUserCommand).build() {
};
__name(_CreateUserCommand, "CreateUserCommand");
var CreateUserCommand = _CreateUserCommand;

// src/commands/DeactivateUserCommand.ts




var _DeactivateUserCommand = class _DeactivateUserCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AWSGorillaBoyService", "DeactivateUser", {}).n("WorkDocsClient", "DeactivateUserCommand").f(DeactivateUserRequestFilterSensitiveLog, void 0).ser(se_DeactivateUserCommand).de(de_DeactivateUserCommand).build() {
};
__name(_DeactivateUserCommand, "DeactivateUserCommand");
var DeactivateUserCommand = _DeactivateUserCommand;

// src/commands/DeleteCommentCommand.ts




var _DeleteCommentCommand = class _DeleteCommentCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AWSGorillaBoyService", "DeleteComment", {}).n("WorkDocsClient", "DeleteCommentCommand").f(DeleteCommentRequestFilterSensitiveLog, void 0).ser(se_DeleteCommentCommand).de(de_DeleteCommentCommand).build() {
};
__name(_DeleteCommentCommand, "DeleteCommentCommand");
var DeleteCommentCommand = _DeleteCommentCommand;

// src/commands/DeleteCustomMetadataCommand.ts




var _DeleteCustomMetadataCommand = class _DeleteCustomMetadataCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AWSGorillaBoyService", "DeleteCustomMetadata", {}).n("WorkDocsClient", "DeleteCustomMetadataCommand").f(DeleteCustomMetadataRequestFilterSensitiveLog, void 0).ser(se_DeleteCustomMetadataCommand).de(de_DeleteCustomMetadataCommand).build() {
};
__name(_DeleteCustomMetadataCommand, "DeleteCustomMetadataCommand");
var DeleteCustomMetadataCommand = _DeleteCustomMetadataCommand;

// src/commands/DeleteDocumentCommand.ts




var _DeleteDocumentCommand = class _DeleteDocumentCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AWSGorillaBoyService", "DeleteDocument", {}).n("WorkDocsClient", "DeleteDocumentCommand").f(DeleteDocumentRequestFilterSensitiveLog, void 0).ser(se_DeleteDocumentCommand).de(de_DeleteDocumentCommand).build() {
};
__name(_DeleteDocumentCommand, "DeleteDocumentCommand");
var DeleteDocumentCommand = _DeleteDocumentCommand;

// src/commands/DeleteDocumentVersionCommand.ts




var _DeleteDocumentVersionCommand = class _DeleteDocumentVersionCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AWSGorillaBoyService", "DeleteDocumentVersion", {}).n("WorkDocsClient", "DeleteDocumentVersionCommand").f(DeleteDocumentVersionRequestFilterSensitiveLog, void 0).ser(se_DeleteDocumentVersionCommand).de(de_DeleteDocumentVersionCommand).build() {
};
__name(_DeleteDocumentVersionCommand, "DeleteDocumentVersionCommand");
var DeleteDocumentVersionCommand = _DeleteDocumentVersionCommand;

// src/commands/DeleteFolderCommand.ts




var _DeleteFolderCommand = class _DeleteFolderCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AWSGorillaBoyService", "DeleteFolder", {}).n("WorkDocsClient", "DeleteFolderCommand").f(DeleteFolderRequestFilterSensitiveLog, void 0).ser(se_DeleteFolderCommand).de(de_DeleteFolderCommand).build() {
};
__name(_DeleteFolderCommand, "DeleteFolderCommand");
var DeleteFolderCommand = _DeleteFolderCommand;

// src/commands/DeleteFolderContentsCommand.ts




var _DeleteFolderContentsCommand = class _DeleteFolderContentsCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AWSGorillaBoyService", "DeleteFolderContents", {}).n("WorkDocsClient", "DeleteFolderContentsCommand").f(DeleteFolderContentsRequestFilterSensitiveLog, void 0).ser(se_DeleteFolderContentsCommand).de(de_DeleteFolderContentsCommand).build() {
};
__name(_DeleteFolderContentsCommand, "DeleteFolderContentsCommand");
var DeleteFolderContentsCommand = _DeleteFolderContentsCommand;

// src/commands/DeleteLabelsCommand.ts




var _DeleteLabelsCommand = class _DeleteLabelsCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AWSGorillaBoyService", "DeleteLabels", {}).n("WorkDocsClient", "DeleteLabelsCommand").f(DeleteLabelsRequestFilterSensitiveLog, void 0).ser(se_DeleteLabelsCommand).de(de_DeleteLabelsCommand).build() {
};
__name(_DeleteLabelsCommand, "DeleteLabelsCommand");
var DeleteLabelsCommand = _DeleteLabelsCommand;

// src/commands/DeleteNotificationSubscriptionCommand.ts




var _DeleteNotificationSubscriptionCommand = class _DeleteNotificationSubscriptionCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AWSGorillaBoyService", "DeleteNotificationSubscription", {}).n("WorkDocsClient", "DeleteNotificationSubscriptionCommand").f(void 0, void 0).ser(se_DeleteNotificationSubscriptionCommand).de(de_DeleteNotificationSubscriptionCommand).build() {
};
__name(_DeleteNotificationSubscriptionCommand, "DeleteNotificationSubscriptionCommand");
var DeleteNotificationSubscriptionCommand = _DeleteNotificationSubscriptionCommand;

// src/commands/DeleteUserCommand.ts




var _DeleteUserCommand = class _DeleteUserCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AWSGorillaBoyService", "DeleteUser", {}).n("WorkDocsClient", "DeleteUserCommand").f(DeleteUserRequestFilterSensitiveLog, void 0).ser(se_DeleteUserCommand).de(de_DeleteUserCommand).build() {
};
__name(_DeleteUserCommand, "DeleteUserCommand");
var DeleteUserCommand = _DeleteUserCommand;

// src/commands/DescribeActivitiesCommand.ts




var _DescribeActivitiesCommand = class _DescribeActivitiesCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AWSGorillaBoyService", "DescribeActivities", {}).n("WorkDocsClient", "DescribeActivitiesCommand").f(DescribeActivitiesRequestFilterSensitiveLog, DescribeActivitiesResponseFilterSensitiveLog).ser(se_DescribeActivitiesCommand).de(de_DescribeActivitiesCommand).build() {
};
__name(_DescribeActivitiesCommand, "DescribeActivitiesCommand");
var DescribeActivitiesCommand = _DescribeActivitiesCommand;

// src/commands/DescribeCommentsCommand.ts




var _DescribeCommentsCommand = class _DescribeCommentsCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AWSGorillaBoyService", "DescribeComments", {}).n("WorkDocsClient", "DescribeCommentsCommand").f(DescribeCommentsRequestFilterSensitiveLog, DescribeCommentsResponseFilterSensitiveLog).ser(se_DescribeCommentsCommand).de(de_DescribeCommentsCommand).build() {
};
__name(_DescribeCommentsCommand, "DescribeCommentsCommand");
var DescribeCommentsCommand = _DescribeCommentsCommand;

// src/commands/DescribeDocumentVersionsCommand.ts




var _DescribeDocumentVersionsCommand = class _DescribeDocumentVersionsCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AWSGorillaBoyService", "DescribeDocumentVersions", {}).n("WorkDocsClient", "DescribeDocumentVersionsCommand").f(DescribeDocumentVersionsRequestFilterSensitiveLog, DescribeDocumentVersionsResponseFilterSensitiveLog).ser(se_DescribeDocumentVersionsCommand).de(de_DescribeDocumentVersionsCommand).build() {
};
__name(_DescribeDocumentVersionsCommand, "DescribeDocumentVersionsCommand");
var DescribeDocumentVersionsCommand = _DescribeDocumentVersionsCommand;

// src/commands/DescribeFolderContentsCommand.ts




var _DescribeFolderContentsCommand = class _DescribeFolderContentsCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AWSGorillaBoyService", "DescribeFolderContents", {}).n("WorkDocsClient", "DescribeFolderContentsCommand").f(DescribeFolderContentsRequestFilterSensitiveLog, DescribeFolderContentsResponseFilterSensitiveLog).ser(se_DescribeFolderContentsCommand).de(de_DescribeFolderContentsCommand).build() {
};
__name(_DescribeFolderContentsCommand, "DescribeFolderContentsCommand");
var DescribeFolderContentsCommand = _DescribeFolderContentsCommand;

// src/commands/DescribeGroupsCommand.ts




var _DescribeGroupsCommand = class _DescribeGroupsCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AWSGorillaBoyService", "DescribeGroups", {}).n("WorkDocsClient", "DescribeGroupsCommand").f(DescribeGroupsRequestFilterSensitiveLog, void 0).ser(se_DescribeGroupsCommand).de(de_DescribeGroupsCommand).build() {
};
__name(_DescribeGroupsCommand, "DescribeGroupsCommand");
var DescribeGroupsCommand = _DescribeGroupsCommand;

// src/commands/DescribeNotificationSubscriptionsCommand.ts




var _DescribeNotificationSubscriptionsCommand = class _DescribeNotificationSubscriptionsCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AWSGorillaBoyService", "DescribeNotificationSubscriptions", {}).n("WorkDocsClient", "DescribeNotificationSubscriptionsCommand").f(void 0, void 0).ser(se_DescribeNotificationSubscriptionsCommand).de(de_DescribeNotificationSubscriptionsCommand).build() {
};
__name(_DescribeNotificationSubscriptionsCommand, "DescribeNotificationSubscriptionsCommand");
var DescribeNotificationSubscriptionsCommand = _DescribeNotificationSubscriptionsCommand;

// src/commands/DescribeResourcePermissionsCommand.ts




var _DescribeResourcePermissionsCommand = class _DescribeResourcePermissionsCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AWSGorillaBoyService", "DescribeResourcePermissions", {}).n("WorkDocsClient", "DescribeResourcePermissionsCommand").f(DescribeResourcePermissionsRequestFilterSensitiveLog, void 0).ser(se_DescribeResourcePermissionsCommand).de(de_DescribeResourcePermissionsCommand).build() {
};
__name(_DescribeResourcePermissionsCommand, "DescribeResourcePermissionsCommand");
var DescribeResourcePermissionsCommand = _DescribeResourcePermissionsCommand;

// src/commands/DescribeRootFoldersCommand.ts




var _DescribeRootFoldersCommand = class _DescribeRootFoldersCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AWSGorillaBoyService", "DescribeRootFolders", {}).n("WorkDocsClient", "DescribeRootFoldersCommand").f(DescribeRootFoldersRequestFilterSensitiveLog, DescribeRootFoldersResponseFilterSensitiveLog).ser(se_DescribeRootFoldersCommand).de(de_DescribeRootFoldersCommand).build() {
};
__name(_DescribeRootFoldersCommand, "DescribeRootFoldersCommand");
var DescribeRootFoldersCommand = _DescribeRootFoldersCommand;

// src/commands/DescribeUsersCommand.ts




var _DescribeUsersCommand = class _DescribeUsersCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AWSGorillaBoyService", "DescribeUsers", {}).n("WorkDocsClient", "DescribeUsersCommand").f(DescribeUsersRequestFilterSensitiveLog, DescribeUsersResponseFilterSensitiveLog).ser(se_DescribeUsersCommand).de(de_DescribeUsersCommand).build() {
};
__name(_DescribeUsersCommand, "DescribeUsersCommand");
var DescribeUsersCommand = _DescribeUsersCommand;

// src/commands/GetCurrentUserCommand.ts




var _GetCurrentUserCommand = class _GetCurrentUserCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AWSGorillaBoyService", "GetCurrentUser", {}).n("WorkDocsClient", "GetCurrentUserCommand").f(GetCurrentUserRequestFilterSensitiveLog, GetCurrentUserResponseFilterSensitiveLog).ser(se_GetCurrentUserCommand).de(de_GetCurrentUserCommand).build() {
};
__name(_GetCurrentUserCommand, "GetCurrentUserCommand");
var GetCurrentUserCommand = _GetCurrentUserCommand;

// src/commands/GetDocumentCommand.ts




var _GetDocumentCommand = class _GetDocumentCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AWSGorillaBoyService", "GetDocument", {}).n("WorkDocsClient", "GetDocumentCommand").f(GetDocumentRequestFilterSensitiveLog, GetDocumentResponseFilterSensitiveLog).ser(se_GetDocumentCommand).de(de_GetDocumentCommand).build() {
};
__name(_GetDocumentCommand, "GetDocumentCommand");
var GetDocumentCommand = _GetDocumentCommand;

// src/commands/GetDocumentPathCommand.ts




var _GetDocumentPathCommand = class _GetDocumentPathCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AWSGorillaBoyService", "GetDocumentPath", {}).n("WorkDocsClient", "GetDocumentPathCommand").f(GetDocumentPathRequestFilterSensitiveLog, GetDocumentPathResponseFilterSensitiveLog).ser(se_GetDocumentPathCommand).de(de_GetDocumentPathCommand).build() {
};
__name(_GetDocumentPathCommand, "GetDocumentPathCommand");
var GetDocumentPathCommand = _GetDocumentPathCommand;

// src/commands/GetDocumentVersionCommand.ts




var _GetDocumentVersionCommand = class _GetDocumentVersionCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AWSGorillaBoyService", "GetDocumentVersion", {}).n("WorkDocsClient", "GetDocumentVersionCommand").f(GetDocumentVersionRequestFilterSensitiveLog, GetDocumentVersionResponseFilterSensitiveLog).ser(se_GetDocumentVersionCommand).de(de_GetDocumentVersionCommand).build() {
};
__name(_GetDocumentVersionCommand, "GetDocumentVersionCommand");
var GetDocumentVersionCommand = _GetDocumentVersionCommand;

// src/commands/GetFolderCommand.ts




var _GetFolderCommand = class _GetFolderCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AWSGorillaBoyService", "GetFolder", {}).n("WorkDocsClient", "GetFolderCommand").f(GetFolderRequestFilterSensitiveLog, GetFolderResponseFilterSensitiveLog).ser(se_GetFolderCommand).de(de_GetFolderCommand).build() {
};
__name(_GetFolderCommand, "GetFolderCommand");
var GetFolderCommand = _GetFolderCommand;

// src/commands/GetFolderPathCommand.ts




var _GetFolderPathCommand = class _GetFolderPathCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AWSGorillaBoyService", "GetFolderPath", {}).n("WorkDocsClient", "GetFolderPathCommand").f(GetFolderPathRequestFilterSensitiveLog, GetFolderPathResponseFilterSensitiveLog).ser(se_GetFolderPathCommand).de(de_GetFolderPathCommand).build() {
};
__name(_GetFolderPathCommand, "GetFolderPathCommand");
var GetFolderPathCommand = _GetFolderPathCommand;

// src/commands/GetResourcesCommand.ts




var _GetResourcesCommand = class _GetResourcesCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AWSGorillaBoyService", "GetResources", {}).n("WorkDocsClient", "GetResourcesCommand").f(GetResourcesRequestFilterSensitiveLog, GetResourcesResponseFilterSensitiveLog).ser(se_GetResourcesCommand).de(de_GetResourcesCommand).build() {
};
__name(_GetResourcesCommand, "GetResourcesCommand");
var GetResourcesCommand = _GetResourcesCommand;

// src/commands/InitiateDocumentVersionUploadCommand.ts




var _InitiateDocumentVersionUploadCommand = class _InitiateDocumentVersionUploadCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AWSGorillaBoyService", "InitiateDocumentVersionUpload", {}).n("WorkDocsClient", "InitiateDocumentVersionUploadCommand").f(InitiateDocumentVersionUploadRequestFilterSensitiveLog, InitiateDocumentVersionUploadResponseFilterSensitiveLog).ser(se_InitiateDocumentVersionUploadCommand).de(de_InitiateDocumentVersionUploadCommand).build() {
};
__name(_InitiateDocumentVersionUploadCommand, "InitiateDocumentVersionUploadCommand");
var InitiateDocumentVersionUploadCommand = _InitiateDocumentVersionUploadCommand;

// src/commands/RemoveAllResourcePermissionsCommand.ts




var _RemoveAllResourcePermissionsCommand = class _RemoveAllResourcePermissionsCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AWSGorillaBoyService", "RemoveAllResourcePermissions", {}).n("WorkDocsClient", "RemoveAllResourcePermissionsCommand").f(RemoveAllResourcePermissionsRequestFilterSensitiveLog, void 0).ser(se_RemoveAllResourcePermissionsCommand).de(de_RemoveAllResourcePermissionsCommand).build() {
};
__name(_RemoveAllResourcePermissionsCommand, "RemoveAllResourcePermissionsCommand");
var RemoveAllResourcePermissionsCommand = _RemoveAllResourcePermissionsCommand;

// src/commands/RemoveResourcePermissionCommand.ts




var _RemoveResourcePermissionCommand = class _RemoveResourcePermissionCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AWSGorillaBoyService", "RemoveResourcePermission", {}).n("WorkDocsClient", "RemoveResourcePermissionCommand").f(RemoveResourcePermissionRequestFilterSensitiveLog, void 0).ser(se_RemoveResourcePermissionCommand).de(de_RemoveResourcePermissionCommand).build() {
};
__name(_RemoveResourcePermissionCommand, "RemoveResourcePermissionCommand");
var RemoveResourcePermissionCommand = _RemoveResourcePermissionCommand;

// src/commands/RestoreDocumentVersionsCommand.ts




var _RestoreDocumentVersionsCommand = class _RestoreDocumentVersionsCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AWSGorillaBoyService", "RestoreDocumentVersions", {}).n("WorkDocsClient", "RestoreDocumentVersionsCommand").f(RestoreDocumentVersionsRequestFilterSensitiveLog, void 0).ser(se_RestoreDocumentVersionsCommand).de(de_RestoreDocumentVersionsCommand).build() {
};
__name(_RestoreDocumentVersionsCommand, "RestoreDocumentVersionsCommand");
var RestoreDocumentVersionsCommand = _RestoreDocumentVersionsCommand;

// src/commands/SearchResourcesCommand.ts




var _SearchResourcesCommand = class _SearchResourcesCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AWSGorillaBoyService", "SearchResources", {}).n("WorkDocsClient", "SearchResourcesCommand").f(SearchResourcesRequestFilterSensitiveLog, SearchResourcesResponseFilterSensitiveLog).ser(se_SearchResourcesCommand).de(de_SearchResourcesCommand).build() {
};
__name(_SearchResourcesCommand, "SearchResourcesCommand");
var SearchResourcesCommand = _SearchResourcesCommand;

// src/commands/UpdateDocumentCommand.ts




var _UpdateDocumentCommand = class _UpdateDocumentCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AWSGorillaBoyService", "UpdateDocument", {}).n("WorkDocsClient", "UpdateDocumentCommand").f(UpdateDocumentRequestFilterSensitiveLog, void 0).ser(se_UpdateDocumentCommand).de(de_UpdateDocumentCommand).build() {
};
__name(_UpdateDocumentCommand, "UpdateDocumentCommand");
var UpdateDocumentCommand = _UpdateDocumentCommand;

// src/commands/UpdateDocumentVersionCommand.ts




var _UpdateDocumentVersionCommand = class _UpdateDocumentVersionCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AWSGorillaBoyService", "UpdateDocumentVersion", {}).n("WorkDocsClient", "UpdateDocumentVersionCommand").f(UpdateDocumentVersionRequestFilterSensitiveLog, void 0).ser(se_UpdateDocumentVersionCommand).de(de_UpdateDocumentVersionCommand).build() {
};
__name(_UpdateDocumentVersionCommand, "UpdateDocumentVersionCommand");
var UpdateDocumentVersionCommand = _UpdateDocumentVersionCommand;

// src/commands/UpdateFolderCommand.ts




var _UpdateFolderCommand = class _UpdateFolderCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AWSGorillaBoyService", "UpdateFolder", {}).n("WorkDocsClient", "UpdateFolderCommand").f(UpdateFolderRequestFilterSensitiveLog, void 0).ser(se_UpdateFolderCommand).de(de_UpdateFolderCommand).build() {
};
__name(_UpdateFolderCommand, "UpdateFolderCommand");
var UpdateFolderCommand = _UpdateFolderCommand;

// src/commands/UpdateUserCommand.ts




var _UpdateUserCommand = class _UpdateUserCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AWSGorillaBoyService", "UpdateUser", {}).n("WorkDocsClient", "UpdateUserCommand").f(UpdateUserRequestFilterSensitiveLog, UpdateUserResponseFilterSensitiveLog).ser(se_UpdateUserCommand).de(de_UpdateUserCommand).build() {
};
__name(_UpdateUserCommand, "UpdateUserCommand");
var UpdateUserCommand = _UpdateUserCommand;

// src/WorkDocs.ts
var commands = {
  AbortDocumentVersionUploadCommand,
  ActivateUserCommand,
  AddResourcePermissionsCommand,
  CreateCommentCommand,
  CreateCustomMetadataCommand,
  CreateFolderCommand,
  CreateLabelsCommand,
  CreateNotificationSubscriptionCommand,
  CreateUserCommand,
  DeactivateUserCommand,
  DeleteCommentCommand,
  DeleteCustomMetadataCommand,
  DeleteDocumentCommand,
  DeleteDocumentVersionCommand,
  DeleteFolderCommand,
  DeleteFolderContentsCommand,
  DeleteLabelsCommand,
  DeleteNotificationSubscriptionCommand,
  DeleteUserCommand,
  DescribeActivitiesCommand,
  DescribeCommentsCommand,
  DescribeDocumentVersionsCommand,
  DescribeFolderContentsCommand,
  DescribeGroupsCommand,
  DescribeNotificationSubscriptionsCommand,
  DescribeResourcePermissionsCommand,
  DescribeRootFoldersCommand,
  DescribeUsersCommand,
  GetCurrentUserCommand,
  GetDocumentCommand,
  GetDocumentPathCommand,
  GetDocumentVersionCommand,
  GetFolderCommand,
  GetFolderPathCommand,
  GetResourcesCommand,
  InitiateDocumentVersionUploadCommand,
  RemoveAllResourcePermissionsCommand,
  RemoveResourcePermissionCommand,
  RestoreDocumentVersionsCommand,
  SearchResourcesCommand,
  UpdateDocumentCommand,
  UpdateDocumentVersionCommand,
  UpdateFolderCommand,
  UpdateUserCommand
};
var _WorkDocs = class _WorkDocs extends WorkDocsClient {
};
__name(_WorkDocs, "WorkDocs");
var WorkDocs = _WorkDocs;
(0, import_smithy_client.createAggregatedClient)(commands, WorkDocs);

// src/pagination/DescribeActivitiesPaginator.ts

var paginateDescribeActivities = (0, import_core.createPaginator)(WorkDocsClient, DescribeActivitiesCommand, "Marker", "Marker", "Limit");

// src/pagination/DescribeCommentsPaginator.ts

var paginateDescribeComments = (0, import_core.createPaginator)(WorkDocsClient, DescribeCommentsCommand, "Marker", "Marker", "Limit");

// src/pagination/DescribeDocumentVersionsPaginator.ts

var paginateDescribeDocumentVersions = (0, import_core.createPaginator)(WorkDocsClient, DescribeDocumentVersionsCommand, "Marker", "Marker", "Limit");

// src/pagination/DescribeFolderContentsPaginator.ts

var paginateDescribeFolderContents = (0, import_core.createPaginator)(WorkDocsClient, DescribeFolderContentsCommand, "Marker", "Marker", "Limit");

// src/pagination/DescribeGroupsPaginator.ts

var paginateDescribeGroups = (0, import_core.createPaginator)(WorkDocsClient, DescribeGroupsCommand, "Marker", "Marker", "Limit");

// src/pagination/DescribeNotificationSubscriptionsPaginator.ts

var paginateDescribeNotificationSubscriptions = (0, import_core.createPaginator)(WorkDocsClient, DescribeNotificationSubscriptionsCommand, "Marker", "Marker", "Limit");

// src/pagination/DescribeResourcePermissionsPaginator.ts

var paginateDescribeResourcePermissions = (0, import_core.createPaginator)(WorkDocsClient, DescribeResourcePermissionsCommand, "Marker", "Marker", "Limit");

// src/pagination/DescribeRootFoldersPaginator.ts

var paginateDescribeRootFolders = (0, import_core.createPaginator)(WorkDocsClient, DescribeRootFoldersCommand, "Marker", "Marker", "Limit");

// src/pagination/DescribeUsersPaginator.ts

var paginateDescribeUsers = (0, import_core.createPaginator)(WorkDocsClient, DescribeUsersCommand, "Marker", "Marker", "Limit");

// src/pagination/SearchResourcesPaginator.ts

var paginateSearchResources = (0, import_core.createPaginator)(WorkDocsClient, SearchResourcesCommand, "Marker", "Marker", "Limit");
// Annotate the CommonJS export names for ESM import in node:

0 && (module.exports = {
  WorkDocsServiceException,
  __Client,
  WorkDocsClient,
  WorkDocs,
  $Command,
  AbortDocumentVersionUploadCommand,
  ActivateUserCommand,
  AddResourcePermissionsCommand,
  CreateCommentCommand,
  CreateCustomMetadataCommand,
  CreateFolderCommand,
  CreateLabelsCommand,
  CreateNotificationSubscriptionCommand,
  CreateUserCommand,
  DeactivateUserCommand,
  DeleteCommentCommand,
  DeleteCustomMetadataCommand,
  DeleteDocumentCommand,
  DeleteDocumentVersionCommand,
  DeleteFolderCommand,
  DeleteFolderContentsCommand,
  DeleteLabelsCommand,
  DeleteNotificationSubscriptionCommand,
  DeleteUserCommand,
  DescribeActivitiesCommand,
  DescribeCommentsCommand,
  DescribeDocumentVersionsCommand,
  DescribeFolderContentsCommand,
  DescribeGroupsCommand,
  DescribeNotificationSubscriptionsCommand,
  DescribeResourcePermissionsCommand,
  DescribeRootFoldersCommand,
  DescribeUsersCommand,
  GetCurrentUserCommand,
  GetDocumentCommand,
  GetDocumentPathCommand,
  GetDocumentVersionCommand,
  GetFolderCommand,
  GetFolderPathCommand,
  GetResourcesCommand,
  InitiateDocumentVersionUploadCommand,
  RemoveAllResourcePermissionsCommand,
  RemoveResourcePermissionCommand,
  RestoreDocumentVersionsCommand,
  SearchResourcesCommand,
  UpdateDocumentCommand,
  UpdateDocumentVersionCommand,
  UpdateFolderCommand,
  UpdateUserCommand,
  paginateDescribeActivities,
  paginateDescribeComments,
  paginateDescribeDocumentVersions,
  paginateDescribeFolderContents,
  paginateDescribeGroups,
  paginateDescribeNotificationSubscriptions,
  paginateDescribeResourcePermissions,
  paginateDescribeRootFolders,
  paginateDescribeUsers,
  paginateSearchResources,
  ConcurrentModificationException,
  EntityNotExistsException,
  FailedDependencyException,
  ProhibitedStateException,
  ServiceUnavailableException,
  UnauthorizedOperationException,
  UnauthorizedResourceAccessException,
  LocaleType,
  UserStatusType,
  StorageType,
  UserType,
  CommentStatusType,
  ResourceType,
  ActivityType,
  AdditionalResponseFieldType,
  RoleType,
  PrincipalType,
  ShareStatusType,
  CommentVisibilityType,
  DocumentLockedForCommentsException,
  InvalidCommentOperationException,
  CustomMetadataLimitExceededException,
  ConflictingOperationException,
  ResourceStateType,
  EntityAlreadyExistsException,
  LimitExceededException,
  TooManyLabelsException,
  SubscriptionProtocolType,
  SubscriptionType,
  InvalidArgumentException,
  TooManySubscriptionsException,
  InvalidOperationException,
  DocumentSourceType,
  DocumentStatusType,
  DocumentThumbnailType,
  InvalidPasswordException,
  OrderType,
  ResourceSortType,
  FolderContentType,
  RolePermissionType,
  UserFilterType,
  UserSortType,
  RequestedEntityTooLargeException,
  ResourceCollectionType,
  DraftUploadOutOfSyncException,
  ResourceAlreadyCheckedOutException,
  StorageLimitExceededException,
  StorageLimitWillExceedException,
  ContentCategoryType,
  PrincipalRoleType,
  SearchResourceType,
  SearchCollectionType,
  LanguageCodeType,
  OrderByFieldType,
  SortOrder,
  SearchQueryScopeType,
  ResponseItemType,
  DocumentVersionStatus,
  DeactivatingLastSystemUserException,
  IllegalUserStateException,
  BooleanEnumType,
  AbortDocumentVersionUploadRequestFilterSensitiveLog,
  ActivateUserRequestFilterSensitiveLog,
  UserFilterSensitiveLog,
  ActivateUserResponseFilterSensitiveLog,
  CommentMetadataFilterSensitiveLog,
  UserMetadataFilterSensitiveLog,
  ResourceMetadataFilterSensitiveLog,
  ParticipantsFilterSensitiveLog,
  ActivityFilterSensitiveLog,
  NotificationOptionsFilterSensitiveLog,
  AddResourcePermissionsRequestFilterSensitiveLog,
  ShareResultFilterSensitiveLog,
  AddResourcePermissionsResponseFilterSensitiveLog,
  CreateCommentRequestFilterSensitiveLog,
  CommentFilterSensitiveLog,
  CreateCommentResponseFilterSensitiveLog,
  CreateCustomMetadataRequestFilterSensitiveLog,
  CreateFolderRequestFilterSensitiveLog,
  FolderMetadataFilterSensitiveLog,
  CreateFolderResponseFilterSensitiveLog,
  CreateLabelsRequestFilterSensitiveLog,
  CreateUserRequestFilterSensitiveLog,
  CreateUserResponseFilterSensitiveLog,
  DeactivateUserRequestFilterSensitiveLog,
  DeleteCommentRequestFilterSensitiveLog,
  DeleteCustomMetadataRequestFilterSensitiveLog,
  DeleteDocumentRequestFilterSensitiveLog,
  DeleteDocumentVersionRequestFilterSensitiveLog,
  DeleteFolderRequestFilterSensitiveLog,
  DeleteFolderContentsRequestFilterSensitiveLog,
  DeleteLabelsRequestFilterSensitiveLog,
  DeleteUserRequestFilterSensitiveLog,
  DescribeActivitiesRequestFilterSensitiveLog,
  DescribeActivitiesResponseFilterSensitiveLog,
  DescribeCommentsRequestFilterSensitiveLog,
  DescribeCommentsResponseFilterSensitiveLog,
  DescribeDocumentVersionsRequestFilterSensitiveLog,
  DocumentVersionMetadataFilterSensitiveLog,
  DescribeDocumentVersionsResponseFilterSensitiveLog,
  DescribeFolderContentsRequestFilterSensitiveLog,
  DocumentMetadataFilterSensitiveLog,
  DescribeFolderContentsResponseFilterSensitiveLog,
  DescribeGroupsRequestFilterSensitiveLog,
  DescribeResourcePermissionsRequestFilterSensitiveLog,
  DescribeRootFoldersRequestFilterSensitiveLog,
  DescribeRootFoldersResponseFilterSensitiveLog,
  DescribeUsersRequestFilterSensitiveLog,
  DescribeUsersResponseFilterSensitiveLog,
  GetCurrentUserRequestFilterSensitiveLog,
  GetCurrentUserResponseFilterSensitiveLog,
  GetDocumentRequestFilterSensitiveLog,
  GetDocumentResponseFilterSensitiveLog,
  GetDocumentPathRequestFilterSensitiveLog,
  ResourcePathComponentFilterSensitiveLog,
  ResourcePathFilterSensitiveLog,
  GetDocumentPathResponseFilterSensitiveLog,
  GetDocumentVersionRequestFilterSensitiveLog,
  GetDocumentVersionResponseFilterSensitiveLog,
  GetFolderRequestFilterSensitiveLog,
  GetFolderResponseFilterSensitiveLog,
  GetFolderPathRequestFilterSensitiveLog,
  GetFolderPathResponseFilterSensitiveLog,
  GetResourcesRequestFilterSensitiveLog,
  GetResourcesResponseFilterSensitiveLog,
  InitiateDocumentVersionUploadRequestFilterSensitiveLog,
  UploadMetadataFilterSensitiveLog,
  InitiateDocumentVersionUploadResponseFilterSensitiveLog,
  RemoveAllResourcePermissionsRequestFilterSensitiveLog,
  RemoveResourcePermissionRequestFilterSensitiveLog,
  RestoreDocumentVersionsRequestFilterSensitiveLog,
  SearchResourcesRequestFilterSensitiveLog,
  ResponseItemFilterSensitiveLog,
  SearchResourcesResponseFilterSensitiveLog,
  UpdateDocumentRequestFilterSensitiveLog,
  UpdateDocumentVersionRequestFilterSensitiveLog,
  UpdateFolderRequestFilterSensitiveLog,
  UpdateUserRequestFilterSensitiveLog,
  UpdateUserResponseFilterSensitiveLog
});
