// @aws-sdk/client-billingconductor/dist-cjs/index.js
"use strict";
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __name = (target, value) => __defProp(target, "name", { value, configurable: true });
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// src/index.ts
var src_exports = {};
__export(src_exports, {
  AccessDeniedException: () => AccessDeniedException,
  AccountAssociationsListElementFilterSensitiveLog: () => AccountAssociationsListElementFilterSensitiveLog,
  AssociateAccountsCommand: () => AssociateAccountsCommand,
  AssociatePricingRulesCommand: () => AssociatePricingRulesCommand,
  AssociateResourceErrorReason: () => AssociateResourceErrorReason,
  BatchAssociateResourcesToCustomLineItemCommand: () => BatchAssociateResourcesToCustomLineItemCommand,
  BatchDisassociateResourcesFromCustomLineItemCommand: () => BatchDisassociateResourcesFromCustomLineItemCommand,
  BillingGroupListElementFilterSensitiveLog: () => BillingGroupListElementFilterSensitiveLog,
  BillingGroupStatus: () => BillingGroupStatus,
  Billingconductor: () => Billingconductor,
  BillingconductorClient: () => BillingconductorClient,
  BillingconductorServiceException: () => BillingconductorServiceException,
  ConflictException: () => ConflictException,
  ConflictExceptionReason: () => ConflictExceptionReason,
  CreateBillingGroupCommand: () => CreateBillingGroupCommand,
  CreateBillingGroupInputFilterSensitiveLog: () => CreateBillingGroupInputFilterSensitiveLog,
  CreateCustomLineItemCommand: () => CreateCustomLineItemCommand,
  CreateCustomLineItemInputFilterSensitiveLog: () => CreateCustomLineItemInputFilterSensitiveLog,
  CreatePricingPlanCommand: () => CreatePricingPlanCommand,
  CreatePricingPlanInputFilterSensitiveLog: () => CreatePricingPlanInputFilterSensitiveLog,
  CreatePricingRuleCommand: () => CreatePricingRuleCommand,
  CreatePricingRuleInputFilterSensitiveLog: () => CreatePricingRuleInputFilterSensitiveLog,
  CurrencyCode: () => CurrencyCode,
  CustomLineItemListElementFilterSensitiveLog: () => CustomLineItemListElementFilterSensitiveLog,
  CustomLineItemRelationship: () => CustomLineItemRelationship,
  CustomLineItemType: () => CustomLineItemType,
  CustomLineItemVersionListElementFilterSensitiveLog: () => CustomLineItemVersionListElementFilterSensitiveLog,
  DeleteBillingGroupCommand: () => DeleteBillingGroupCommand,
  DeleteCustomLineItemCommand: () => DeleteCustomLineItemCommand,
  DeletePricingPlanCommand: () => DeletePricingPlanCommand,
  DeletePricingRuleCommand: () => DeletePricingRuleCommand,
  DisassociateAccountsCommand: () => DisassociateAccountsCommand,
  DisassociatePricingRulesCommand: () => DisassociatePricingRulesCommand,
  GetBillingGroupCostReportCommand: () => GetBillingGroupCostReportCommand,
  GroupByAttributeName: () => GroupByAttributeName,
  InternalServerException: () => InternalServerException,
  LineItemFilterAttributeName: () => LineItemFilterAttributeName,
  LineItemFilterValue: () => LineItemFilterValue,
  ListAccountAssociationsCommand: () => ListAccountAssociationsCommand,
  ListAccountAssociationsOutputFilterSensitiveLog: () => ListAccountAssociationsOutputFilterSensitiveLog,
  ListBillingGroupCostReportsCommand: () => ListBillingGroupCostReportsCommand,
  ListBillingGroupsCommand: () => ListBillingGroupsCommand,
  ListBillingGroupsOutputFilterSensitiveLog: () => ListBillingGroupsOutputFilterSensitiveLog,
  ListCustomLineItemVersionsCommand: () => ListCustomLineItemVersionsCommand,
  ListCustomLineItemVersionsOutputFilterSensitiveLog: () => ListCustomLineItemVersionsOutputFilterSensitiveLog,
  ListCustomLineItemsCommand: () => ListCustomLineItemsCommand,
  ListCustomLineItemsFilterFilterSensitiveLog: () => ListCustomLineItemsFilterFilterSensitiveLog,
  ListCustomLineItemsInputFilterSensitiveLog: () => ListCustomLineItemsInputFilterSensitiveLog,
  ListCustomLineItemsOutputFilterSensitiveLog: () => ListCustomLineItemsOutputFilterSensitiveLog,
  ListPricingPlansAssociatedWithPricingRuleCommand: () => ListPricingPlansAssociatedWithPricingRuleCommand,
  ListPricingPlansCommand: () => ListPricingPlansCommand,
  ListPricingPlansOutputFilterSensitiveLog: () => ListPricingPlansOutputFilterSensitiveLog,
  ListPricingRulesAssociatedToPricingPlanCommand: () => ListPricingRulesAssociatedToPricingPlanCommand,
  ListPricingRulesCommand: () => ListPricingRulesCommand,
  ListPricingRulesOutputFilterSensitiveLog: () => ListPricingRulesOutputFilterSensitiveLog,
  ListResourcesAssociatedToCustomLineItemCommand: () => ListResourcesAssociatedToCustomLineItemCommand,
  ListTagsForResourceCommand: () => ListTagsForResourceCommand,
  MatchOption: () => MatchOption,
  PricingPlanListElementFilterSensitiveLog: () => PricingPlanListElementFilterSensitiveLog,
  PricingRuleListElementFilterSensitiveLog: () => PricingRuleListElementFilterSensitiveLog,
  PricingRuleScope: () => PricingRuleScope,
  PricingRuleType: () => PricingRuleType,
  ResourceNotFoundException: () => ResourceNotFoundException,
  ServiceLimitExceededException: () => ServiceLimitExceededException,
  TagResourceCommand: () => TagResourceCommand,
  ThrottlingException: () => ThrottlingException,
  UntagResourceCommand: () => UntagResourceCommand,
  UpdateBillingGroupCommand: () => UpdateBillingGroupCommand,
  UpdateBillingGroupInputFilterSensitiveLog: () => UpdateBillingGroupInputFilterSensitiveLog,
  UpdateBillingGroupOutputFilterSensitiveLog: () => UpdateBillingGroupOutputFilterSensitiveLog,
  UpdateCustomLineItemCommand: () => UpdateCustomLineItemCommand,
  UpdateCustomLineItemInputFilterSensitiveLog: () => UpdateCustomLineItemInputFilterSensitiveLog,
  UpdateCustomLineItemOutputFilterSensitiveLog: () => UpdateCustomLineItemOutputFilterSensitiveLog,
  UpdatePricingPlanCommand: () => UpdatePricingPlanCommand,
  UpdatePricingPlanInputFilterSensitiveLog: () => UpdatePricingPlanInputFilterSensitiveLog,
  UpdatePricingPlanOutputFilterSensitiveLog: () => UpdatePricingPlanOutputFilterSensitiveLog,
  UpdatePricingRuleCommand: () => UpdatePricingRuleCommand,
  UpdatePricingRuleInputFilterSensitiveLog: () => UpdatePricingRuleInputFilterSensitiveLog,
  UpdatePricingRuleOutputFilterSensitiveLog: () => UpdatePricingRuleOutputFilterSensitiveLog,
  ValidationException: () => ValidationException,
  ValidationExceptionReason: () => ValidationExceptionReason,
  __Client: () => import_smithy_client.Client,
  paginateGetBillingGroupCostReport: () => paginateGetBillingGroupCostReport,
  paginateListAccountAssociations: () => paginateListAccountAssociations,
  paginateListBillingGroupCostReports: () => paginateListBillingGroupCostReports,
  paginateListBillingGroups: () => paginateListBillingGroups,
  paginateListCustomLineItemVersions: () => paginateListCustomLineItemVersions,
  paginateListCustomLineItems: () => paginateListCustomLineItems,
  paginateListPricingPlans: () => paginateListPricingPlans,
  paginateListPricingPlansAssociatedWithPricingRule: () => paginateListPricingPlansAssociatedWithPricingRule,
  paginateListPricingRules: () => paginateListPricingRules,
  paginateListPricingRulesAssociatedToPricingPlan: () => paginateListPricingRulesAssociatedToPricingPlan,
  paginateListResourcesAssociatedToCustomLineItem: () => paginateListResourcesAssociatedToCustomLineItem
});
module.exports = __toCommonJS(src_exports);

// src/BillingconductorClient.ts
var import_middleware_host_header = require("@aws-sdk/middleware-host-header");
var import_middleware_logger = require("@aws-sdk/middleware-logger");
var import_middleware_recursion_detection = require("@aws-sdk/middleware-recursion-detection");
var import_middleware_user_agent = require("@aws-sdk/middleware-user-agent");
var import_config_resolver = require("@smithy/config-resolver");
var import_core = require("@smithy/core");
var import_middleware_content_length = require("@smithy/middleware-content-length");
var import_middleware_endpoint = require("@smithy/middleware-endpoint");
var import_middleware_retry = require("@smithy/middleware-retry");

var import_httpAuthSchemeProvider = require("./auth/httpAuthSchemeProvider");

// src/endpoint/EndpointParameters.ts
var resolveClientEndpointParameters = /* @__PURE__ */ __name((options) => {
  return {
    ...options,
    useDualstackEndpoint: options.useDualstackEndpoint ?? false,
    useFipsEndpoint: options.useFipsEndpoint ?? false,
    defaultSigningName: "billingconductor"
  };
}, "resolveClientEndpointParameters");
var commonParams = {
  UseFIPS: { type: "builtInParams", name: "useFipsEndpoint" },
  Endpoint: { type: "builtInParams", name: "endpoint" },
  Region: { type: "builtInParams", name: "region" },
  UseDualStack: { type: "builtInParams", name: "useDualstackEndpoint" }
};

// src/BillingconductorClient.ts
var import_runtimeConfig = require("././runtimeConfig");

// src/runtimeExtensions.ts
var import_region_config_resolver = require("@aws-sdk/region-config-resolver");
var import_protocol_http = require("@smithy/protocol-http");
var import_smithy_client = require("@smithy/smithy-client");

// src/auth/httpAuthExtensionConfiguration.ts
var getHttpAuthExtensionConfiguration = /* @__PURE__ */ __name((runtimeConfig) => {
  const _httpAuthSchemes = runtimeConfig.httpAuthSchemes;
  let _httpAuthSchemeProvider = runtimeConfig.httpAuthSchemeProvider;
  let _credentials = runtimeConfig.credentials;
  return {
    setHttpAuthScheme(httpAuthScheme) {
      const index = _httpAuthSchemes.findIndex((scheme) => scheme.schemeId === httpAuthScheme.schemeId);
      if (index === -1) {
        _httpAuthSchemes.push(httpAuthScheme);
      } else {
        _httpAuthSchemes.splice(index, 1, httpAuthScheme);
      }
    },
    httpAuthSchemes() {
      return _httpAuthSchemes;
    },
    setHttpAuthSchemeProvider(httpAuthSchemeProvider) {
      _httpAuthSchemeProvider = httpAuthSchemeProvider;
    },
    httpAuthSchemeProvider() {
      return _httpAuthSchemeProvider;
    },
    setCredentials(credentials) {
      _credentials = credentials;
    },
    credentials() {
      return _credentials;
    }
  };
}, "getHttpAuthExtensionConfiguration");
var resolveHttpAuthRuntimeConfig = /* @__PURE__ */ __name((config) => {
  return {
    httpAuthSchemes: config.httpAuthSchemes(),
    httpAuthSchemeProvider: config.httpAuthSchemeProvider(),
    credentials: config.credentials()
  };
}, "resolveHttpAuthRuntimeConfig");

// src/runtimeExtensions.ts
var asPartial = /* @__PURE__ */ __name((t) => t, "asPartial");
var resolveRuntimeExtensions = /* @__PURE__ */ __name((runtimeConfig, extensions) => {
  const extensionConfiguration = {
    ...asPartial((0, import_region_config_resolver.getAwsRegionExtensionConfiguration)(runtimeConfig)),
    ...asPartial((0, import_smithy_client.getDefaultExtensionConfiguration)(runtimeConfig)),
    ...asPartial((0, import_protocol_http.getHttpHandlerExtensionConfiguration)(runtimeConfig)),
    ...asPartial(getHttpAuthExtensionConfiguration(runtimeConfig))
  };
  extensions.forEach((extension) => extension.configure(extensionConfiguration));
  return {
    ...runtimeConfig,
    ...(0, import_region_config_resolver.resolveAwsRegionExtensionConfiguration)(extensionConfiguration),
    ...(0, import_smithy_client.resolveDefaultRuntimeConfig)(extensionConfiguration),
    ...(0, import_protocol_http.resolveHttpHandlerRuntimeConfig)(extensionConfiguration),
    ...resolveHttpAuthRuntimeConfig(extensionConfiguration)
  };
}, "resolveRuntimeExtensions");

// src/BillingconductorClient.ts
var _BillingconductorClient = class _BillingconductorClient extends import_smithy_client.Client {
  constructor(...[configuration]) {
    const _config_0 = (0, import_runtimeConfig.getRuntimeConfig)(configuration || {});
    const _config_1 = resolveClientEndpointParameters(_config_0);
    const _config_2 = (0, import_config_resolver.resolveRegionConfig)(_config_1);
    const _config_3 = (0, import_middleware_endpoint.resolveEndpointConfig)(_config_2);
    const _config_4 = (0, import_middleware_retry.resolveRetryConfig)(_config_3);
    const _config_5 = (0, import_middleware_host_header.resolveHostHeaderConfig)(_config_4);
    const _config_6 = (0, import_middleware_user_agent.resolveUserAgentConfig)(_config_5);
    const _config_7 = (0, import_httpAuthSchemeProvider.resolveHttpAuthSchemeConfig)(_config_6);
    const _config_8 = resolveRuntimeExtensions(_config_7, (configuration == null ? void 0 : configuration.extensions) || []);
    super(_config_8);
    this.config = _config_8;
    this.middlewareStack.use((0, import_middleware_retry.getRetryPlugin)(this.config));
    this.middlewareStack.use((0, import_middleware_content_length.getContentLengthPlugin)(this.config));
    this.middlewareStack.use((0, import_middleware_host_header.getHostHeaderPlugin)(this.config));
    this.middlewareStack.use((0, import_middleware_logger.getLoggerPlugin)(this.config));
    this.middlewareStack.use((0, import_middleware_recursion_detection.getRecursionDetectionPlugin)(this.config));
    this.middlewareStack.use((0, import_middleware_user_agent.getUserAgentPlugin)(this.config));
    this.middlewareStack.use(
      (0, import_core.getHttpAuthSchemeEndpointRuleSetPlugin)(this.config, {
        httpAuthSchemeParametersProvider: this.getDefaultHttpAuthSchemeParametersProvider(),
        identityProviderConfigProvider: this.getIdentityProviderConfigProvider()
      })
    );
    this.middlewareStack.use((0, import_core.getHttpSigningPlugin)(this.config));
  }
  /**
   * Destroy underlying resources, like sockets. It's usually not necessary to do this.
   * However in Node.js, it's best to explicitly shut down the client's agent when it is no longer needed.
   * Otherwise, sockets might stay open for quite a long time before the server terminates them.
   */
  destroy() {
    super.destroy();
  }
  getDefaultHttpAuthSchemeParametersProvider() {
    return import_httpAuthSchemeProvider.defaultBillingconductorHttpAuthSchemeParametersProvider;
  }
  getIdentityProviderConfigProvider() {
    return async (config) => new import_core.DefaultIdentityProviderConfig({
      "aws.auth#sigv4": config.credentials
    });
  }
};
__name(_BillingconductorClient, "BillingconductorClient");
var BillingconductorClient = _BillingconductorClient;

// src/Billingconductor.ts


// src/commands/AssociateAccountsCommand.ts

var import_middleware_serde = require("@smithy/middleware-serde");

var import_types = require("@smithy/types");

// src/protocols/Aws_restJson1.ts
var import_core2 = require("@aws-sdk/core");



// src/models/BillingconductorServiceException.ts

var _BillingconductorServiceException = class _BillingconductorServiceException extends import_smithy_client.ServiceException {
  /**
   * @internal
   */
  constructor(options) {
    super(options);
    Object.setPrototypeOf(this, _BillingconductorServiceException.prototype);
  }
};
__name(_BillingconductorServiceException, "BillingconductorServiceException");
var BillingconductorServiceException = _BillingconductorServiceException;

// src/models/models_0.ts

var _AccessDeniedException = class _AccessDeniedException extends BillingconductorServiceException {
  /**
   * @internal
   */
  constructor(opts) {
    super({
      name: "AccessDeniedException",
      $fault: "client",
      ...opts
    });
    this.name = "AccessDeniedException";
    this.$fault = "client";
    Object.setPrototypeOf(this, _AccessDeniedException.prototype);
    this.Message = opts.Message;
  }
};
__name(_AccessDeniedException, "AccessDeniedException");
var AccessDeniedException = _AccessDeniedException;
var ConflictExceptionReason = {
  PRICING_PLAN_ATTACHED_TO_BILLING_GROUP_DELETE_CONFLICT: "PRICING_PLAN_ATTACHED_TO_BILLING_GROUP_DELETE_CONFLICT",
  PRICING_RULE_ATTACHED_TO_PRICING_PLAN_DELETE_CONFLICT: "PRICING_RULE_ATTACHED_TO_PRICING_PLAN_DELETE_CONFLICT",
  PRICING_RULE_IN_PRICING_PLAN_CONFLICT: "PRICING_RULE_IN_PRICING_PLAN_CONFLICT",
  RESOURCE_NAME_CONFLICT: "RESOURCE_NAME_CONFLICT",
  WRITE_CONFLICT_RETRY: "WRITE_CONFLICT_RETRY"
};
var _ConflictException = class _ConflictException extends BillingconductorServiceException {
  /**
   * @internal
   */
  constructor(opts) {
    super({
      name: "ConflictException",
      $fault: "client",
      ...opts
    });
    this.name = "ConflictException";
    this.$fault = "client";
    Object.setPrototypeOf(this, _ConflictException.prototype);
    this.Message = opts.Message;
    this.ResourceId = opts.ResourceId;
    this.ResourceType = opts.ResourceType;
    this.Reason = opts.Reason;
  }
};
__name(_ConflictException, "ConflictException");
var ConflictException = _ConflictException;
var _InternalServerException = class _InternalServerException extends BillingconductorServiceException {
  /**
   * @internal
   */
  constructor(opts) {
    super({
      name: "InternalServerException",
      $fault: "server",
      ...opts
    });
    this.name = "InternalServerException";
    this.$fault = "server";
    Object.setPrototypeOf(this, _InternalServerException.prototype);
    this.Message = opts.Message;
    this.RetryAfterSeconds = opts.RetryAfterSeconds;
  }
};
__name(_InternalServerException, "InternalServerException");
var InternalServerException = _InternalServerException;
var _ResourceNotFoundException = class _ResourceNotFoundException extends BillingconductorServiceException {
  /**
   * @internal
   */
  constructor(opts) {
    super({
      name: "ResourceNotFoundException",
      $fault: "client",
      ...opts
    });
    this.name = "ResourceNotFoundException";
    this.$fault = "client";
    Object.setPrototypeOf(this, _ResourceNotFoundException.prototype);
    this.Message = opts.Message;
    this.ResourceId = opts.ResourceId;
    this.ResourceType = opts.ResourceType;
  }
};
__name(_ResourceNotFoundException, "ResourceNotFoundException");
var ResourceNotFoundException = _ResourceNotFoundException;
var _ServiceLimitExceededException = class _ServiceLimitExceededException extends BillingconductorServiceException {
  /**
   * @internal
   */
  constructor(opts) {
    super({
      name: "ServiceLimitExceededException",
      $fault: "client",
      ...opts
    });
    this.name = "ServiceLimitExceededException";
    this.$fault = "client";
    Object.setPrototypeOf(this, _ServiceLimitExceededException.prototype);
    this.Message = opts.Message;
    this.ResourceId = opts.ResourceId;
    this.ResourceType = opts.ResourceType;
    this.LimitCode = opts.LimitCode;
    this.ServiceCode = opts.ServiceCode;
  }
};
__name(_ServiceLimitExceededException, "ServiceLimitExceededException");
var ServiceLimitExceededException = _ServiceLimitExceededException;
var _ThrottlingException = class _ThrottlingException extends BillingconductorServiceException {
  /**
   * @internal
   */
  constructor(opts) {
    super({
      name: "ThrottlingException",
      $fault: "client",
      ...opts
    });
    this.name = "ThrottlingException";
    this.$fault = "client";
    Object.setPrototypeOf(this, _ThrottlingException.prototype);
    this.Message = opts.Message;
    this.RetryAfterSeconds = opts.RetryAfterSeconds;
  }
};
__name(_ThrottlingException, "ThrottlingException");
var ThrottlingException = _ThrottlingException;
var ValidationExceptionReason = {
  ACCOUNTS_ALREADY_ASSOCIATED: "ACCOUNTS_ALREADY_ASSOCIATED",
  ACCOUNTS_NOT_ASSOCIATED: "ACCOUNTS_NOT_ASSOCIATED",
  CANNOT_DELETE_AUTO_ASSOCIATE_BILLING_GROUP: "CANNOT_DELETE_AUTO_ASSOCIATE_BILLING_GROUP",
  CANNOT_PARSE: "CANNOT_PARSE",
  CUSTOM_LINE_ITEM_ASSOCIATION_EXISTS: "CUSTOM_LINE_ITEM_ASSOCIATION_EXISTS",
  DUPLICATE_ACCOUNT: "DUPLICATE_ACCOUNT",
  DUPLICATE_PRICINGRULE_ARNS: "DUPLICATE_PRICINGRULE_ARNS",
  FIELD_VALIDATION_FAILED: "FIELD_VALIDATION_FAILED",
  ILLEGAL_ACCOUNTS: "ILLEGAL_ACCOUNTS",
  ILLEGAL_ACCOUNT_ID: "ILLEGAL_ACCOUNT_ID",
  ILLEGAL_BILLING_ENTITY: "ILLEGAL_BILLING_ENTITY",
  ILLEGAL_BILLING_PERIOD: "ILLEGAL_BILLING_PERIOD",
  ILLEGAL_BILLING_PERIOD_RANGE: "ILLEGAL_BILLING_PERIOD_RANGE",
  ILLEGAL_CHARGE_DETAILS: "ILLEGAL_CHARGE_DETAILS",
  ILLEGAL_CHILD_ASSOCIATE_RESOURCE: "ILLEGAL_CHILD_ASSOCIATE_RESOURCE",
  ILLEGAL_CUSTOMLINEITEM: "ILLEGAL_CUSTOMLINEITEM",
  ILLEGAL_CUSTOMLINEITEM_MODIFICATION: "ILLEGAL_CUSTOMLINEITEM_MODIFICATION",
  ILLEGAL_CUSTOMLINEITEM_UPDATE: "ILLEGAL_CUSTOMLINEITEM_UPDATE",
  ILLEGAL_ENDED_BILLINGGROUP: "ILLEGAL_ENDED_BILLINGGROUP",
  ILLEGAL_EXPRESSION: "ILLEGAL_EXPRESSION",
  ILLEGAL_MODIFIER_PERCENTAGE: "ILLEGAL_MODIFIER_PERCENTAGE",
  ILLEGAL_OPERATION: "ILLEGAL_OPERATION",
  ILLEGAL_PRIMARY_ACCOUNT: "ILLEGAL_PRIMARY_ACCOUNT",
  ILLEGAL_RESOURCE_ARNS: "ILLEGAL_RESOURCE_ARNS",
  ILLEGAL_SCOPE: "ILLEGAL_SCOPE",
  ILLEGAL_SERVICE: "ILLEGAL_SERVICE",
  ILLEGAL_TIERING_INPUT: "ILLEGAL_TIERING_INPUT",
  ILLEGAL_TYPE: "ILLEGAL_TYPE",
  ILLEGAL_UPDATE_CHARGE_DETAILS: "ILLEGAL_UPDATE_CHARGE_DETAILS",
  ILLEGAL_USAGE_TYPE: "ILLEGAL_USAGE_TYPE",
  INVALID_ARN: "INVALID_ARN",
  INVALID_BILLINGVIEW_ARN: "INVALID_BILLINGVIEW_ARN",
  INVALID_BILLING_GROUP: "INVALID_BILLING_GROUP",
  INVALID_BILLING_GROUP_STATUS: "INVALID_BILLING_GROUP_STATUS",
  INVALID_BILLING_PERIOD_FOR_OPERATION: "INVALID_BILLING_PERIOD_FOR_OPERATION",
  INVALID_FILTER: "INVALID_FILTER",
  INVALID_SKU_COMBO: "INVALID_SKU_COMBO",
  INVALID_TIME_RANGE: "INVALID_TIME_RANGE",
  MISMATCHED_BILLINGGROUP_ARN: "MISMATCHED_BILLINGGROUP_ARN",
  MISMATCHED_BILLINGVIEW_ARN: "MISMATCHED_BILLINGVIEW_ARN",
  MISMATCHED_CUSTOMLINEITEM_ARN: "MISMATCHED_CUSTOMLINEITEM_ARN",
  MISMATCHED_PRICINGPLAN_ARN: "MISMATCHED_PRICINGPLAN_ARN",
  MISMATCHED_PRICINGRULE_ARN: "MISMATCHED_PRICINGRULE_ARN",
  MISSING_BILLINGGROUP: "MISSING_BILLINGGROUP",
  MISSING_CUSTOMLINEITEM: "MISSING_CUSTOMLINEITEM",
  MISSING_LINKED_ACCOUNT_IDS: "MISSING_LINKED_ACCOUNT_IDS",
  MISSING_PRICINGPLAN: "MISSING_PRICINGPLAN",
  MISSING_PRICING_PLAN_ARN: "MISSING_PRICING_PLAN_ARN",
  MULTIPLE_LINKED_ACCOUNT_IDS: "MULTIPLE_LINKED_ACCOUNT_IDS",
  MULTIPLE_PRICING_PLAN_ARN: "MULTIPLE_PRICING_PLAN_ARN",
  OTHER: "OTHER",
  PRICINGRULES_ALREADY_ASSOCIATED: "PRICINGRULES_ALREADY_ASSOCIATED",
  PRICINGRULES_NOT_ASSOCIATED: "PRICINGRULES_NOT_ASSOCIATED",
  PRICINGRULES_NOT_EXIST: "PRICINGRULES_NOT_EXIST",
  PRIMARY_CANNOT_DISASSOCIATE: "PRIMARY_CANNOT_DISASSOCIATE",
  PRIMARY_NOT_ASSOCIATED: "PRIMARY_NOT_ASSOCIATED",
  TOO_MANY_ACCOUNTS_IN_REQUEST: "TOO_MANY_ACCOUNTS_IN_REQUEST",
  TOO_MANY_AUTO_ASSOCIATE_BILLING_GROUPS: "TOO_MANY_AUTO_ASSOCIATE_BILLING_GROUPS",
  TOO_MANY_CUSTOMLINEITEMS_IN_REQUEST: "TOO_MANY_CUSTOMLINEITEMS_IN_REQUEST",
  UNKNOWN_OPERATION: "UNKNOWN_OPERATION"
};
var _ValidationException = class _ValidationException extends BillingconductorServiceException {
  /**
   * @internal
   */
  constructor(opts) {
    super({
      name: "ValidationException",
      $fault: "client",
      ...opts
    });
    this.name = "ValidationException";
    this.$fault = "client";
    Object.setPrototypeOf(this, _ValidationException.prototype);
    this.Message = opts.Message;
    this.Reason = opts.Reason;
    this.Fields = opts.Fields;
  }
};
__name(_ValidationException, "ValidationException");
var ValidationException = _ValidationException;
var AssociateResourceErrorReason = {
  ILLEGAL_CUSTOMLINEITEM: "ILLEGAL_CUSTOMLINEITEM",
  INTERNAL_SERVER_EXCEPTION: "INTERNAL_SERVER_EXCEPTION",
  INVALID_ARN: "INVALID_ARN",
  INVALID_BILLING_PERIOD_RANGE: "INVALID_BILLING_PERIOD_RANGE",
  SERVICE_LIMIT_EXCEEDED: "SERVICE_LIMIT_EXCEEDED"
};
var BillingGroupStatus = {
  ACTIVE: "ACTIVE",
  PRIMARY_ACCOUNT_MISSING: "PRIMARY_ACCOUNT_MISSING"
};
var LineItemFilterAttributeName = {
  LINE_ITEM_TYPE: "LINE_ITEM_TYPE"
};
var MatchOption = {
  NOT_EQUAL: "NOT_EQUAL"
};
var LineItemFilterValue = {
  SAVINGS_PLAN_NEGATION: "SAVINGS_PLAN_NEGATION"
};
var CustomLineItemType = {
  CREDIT: "CREDIT",
  FEE: "FEE"
};
var CurrencyCode = {
  CNY: "CNY",
  USD: "USD"
};
var CustomLineItemRelationship = {
  CHILD: "CHILD",
  PARENT: "PARENT"
};
var GroupByAttributeName = {
  BILLING_PERIOD: "BILLING_PERIOD",
  PRODUCT_NAME: "PRODUCT_NAME"
};
var PricingRuleScope = {
  BILLING_ENTITY: "BILLING_ENTITY",
  GLOBAL: "GLOBAL",
  SERVICE: "SERVICE",
  SKU: "SKU"
};
var PricingRuleType = {
  DISCOUNT: "DISCOUNT",
  MARKUP: "MARKUP",
  TIERING: "TIERING"
};
var AccountAssociationsListElementFilterSensitiveLog = /* @__PURE__ */ __name((obj) => ({
  ...obj,
  ...obj.AccountName && { AccountName: import_smithy_client.SENSITIVE_STRING },
  ...obj.AccountEmail && { AccountEmail: import_smithy_client.SENSITIVE_STRING }
}), "AccountAssociationsListElementFilterSensitiveLog");
var CreateBillingGroupInputFilterSensitiveLog = /* @__PURE__ */ __name((obj) => ({
  ...obj,
  ...obj.Name && { Name: import_smithy_client.SENSITIVE_STRING },
  ...obj.Description && { Description: import_smithy_client.SENSITIVE_STRING }
}), "CreateBillingGroupInputFilterSensitiveLog");
var BillingGroupListElementFilterSensitiveLog = /* @__PURE__ */ __name((obj) => ({
  ...obj,
  ...obj.Name && { Name: import_smithy_client.SENSITIVE_STRING },
  ...obj.Description && { Description: import_smithy_client.SENSITIVE_STRING }
}), "BillingGroupListElementFilterSensitiveLog");
var ListBillingGroupsOutputFilterSensitiveLog = /* @__PURE__ */ __name((obj) => ({
  ...obj,
  ...obj.BillingGroups && {
    BillingGroups: obj.BillingGroups.map((item) => BillingGroupListElementFilterSensitiveLog(item))
  }
}), "ListBillingGroupsOutputFilterSensitiveLog");
var UpdateBillingGroupInputFilterSensitiveLog = /* @__PURE__ */ __name((obj) => ({
  ...obj,
  ...obj.Name && { Name: import_smithy_client.SENSITIVE_STRING },
  ...obj.Description && { Description: import_smithy_client.SENSITIVE_STRING }
}), "UpdateBillingGroupInputFilterSensitiveLog");
var UpdateBillingGroupOutputFilterSensitiveLog = /* @__PURE__ */ __name((obj) => ({
  ...obj,
  ...obj.Name && { Name: import_smithy_client.SENSITIVE_STRING },
  ...obj.Description && { Description: import_smithy_client.SENSITIVE_STRING }
}), "UpdateBillingGroupOutputFilterSensitiveLog");
var CreateCustomLineItemInputFilterSensitiveLog = /* @__PURE__ */ __name((obj) => ({
  ...obj,
  ...obj.Name && { Name: import_smithy_client.SENSITIVE_STRING },
  ...obj.Description && { Description: import_smithy_client.SENSITIVE_STRING }
}), "CreateCustomLineItemInputFilterSensitiveLog");
var ListCustomLineItemsFilterFilterSensitiveLog = /* @__PURE__ */ __name((obj) => ({
  ...obj,
  ...obj.Names && { Names: import_smithy_client.SENSITIVE_STRING }
}), "ListCustomLineItemsFilterFilterSensitiveLog");
var ListCustomLineItemsInputFilterSensitiveLog = /* @__PURE__ */ __name((obj) => ({
  ...obj,
  ...obj.Filters && { Filters: ListCustomLineItemsFilterFilterSensitiveLog(obj.Filters) }
}), "ListCustomLineItemsInputFilterSensitiveLog");
var CustomLineItemListElementFilterSensitiveLog = /* @__PURE__ */ __name((obj) => ({
  ...obj,
  ...obj.Name && { Name: import_smithy_client.SENSITIVE_STRING },
  ...obj.Description && { Description: import_smithy_client.SENSITIVE_STRING }
}), "CustomLineItemListElementFilterSensitiveLog");
var ListCustomLineItemsOutputFilterSensitiveLog = /* @__PURE__ */ __name((obj) => ({
  ...obj,
  ...obj.CustomLineItems && {
    CustomLineItems: obj.CustomLineItems.map((item) => CustomLineItemListElementFilterSensitiveLog(item))
  }
}), "ListCustomLineItemsOutputFilterSensitiveLog");
var CustomLineItemVersionListElementFilterSensitiveLog = /* @__PURE__ */ __name((obj) => ({
  ...obj,
  ...obj.Name && { Name: import_smithy_client.SENSITIVE_STRING },
  ...obj.Description && { Description: import_smithy_client.SENSITIVE_STRING }
}), "CustomLineItemVersionListElementFilterSensitiveLog");
var ListCustomLineItemVersionsOutputFilterSensitiveLog = /* @__PURE__ */ __name((obj) => ({
  ...obj,
  ...obj.CustomLineItemVersions && {
    CustomLineItemVersions: obj.CustomLineItemVersions.map(
      (item) => CustomLineItemVersionListElementFilterSensitiveLog(item)
    )
  }
}), "ListCustomLineItemVersionsOutputFilterSensitiveLog");
var UpdateCustomLineItemInputFilterSensitiveLog = /* @__PURE__ */ __name((obj) => ({
  ...obj,
  ...obj.Name && { Name: import_smithy_client.SENSITIVE_STRING },
  ...obj.Description && { Description: import_smithy_client.SENSITIVE_STRING }
}), "UpdateCustomLineItemInputFilterSensitiveLog");
var UpdateCustomLineItemOutputFilterSensitiveLog = /* @__PURE__ */ __name((obj) => ({
  ...obj,
  ...obj.Name && { Name: import_smithy_client.SENSITIVE_STRING },
  ...obj.Description && { Description: import_smithy_client.SENSITIVE_STRING }
}), "UpdateCustomLineItemOutputFilterSensitiveLog");
var ListAccountAssociationsOutputFilterSensitiveLog = /* @__PURE__ */ __name((obj) => ({
  ...obj,
  ...obj.LinkedAccounts && {
    LinkedAccounts: obj.LinkedAccounts.map((item) => AccountAssociationsListElementFilterSensitiveLog(item))
  }
}), "ListAccountAssociationsOutputFilterSensitiveLog");
var CreatePricingPlanInputFilterSensitiveLog = /* @__PURE__ */ __name((obj) => ({
  ...obj,
  ...obj.Name && { Name: import_smithy_client.SENSITIVE_STRING },
  ...obj.Description && { Description: import_smithy_client.SENSITIVE_STRING }
}), "CreatePricingPlanInputFilterSensitiveLog");
var PricingPlanListElementFilterSensitiveLog = /* @__PURE__ */ __name((obj) => ({
  ...obj,
  ...obj.Name && { Name: import_smithy_client.SENSITIVE_STRING },
  ...obj.Description && { Description: import_smithy_client.SENSITIVE_STRING }
}), "PricingPlanListElementFilterSensitiveLog");
var ListPricingPlansOutputFilterSensitiveLog = /* @__PURE__ */ __name((obj) => ({
  ...obj,
  ...obj.PricingPlans && {
    PricingPlans: obj.PricingPlans.map((item) => PricingPlanListElementFilterSensitiveLog(item))
  }
}), "ListPricingPlansOutputFilterSensitiveLog");
var UpdatePricingPlanInputFilterSensitiveLog = /* @__PURE__ */ __name((obj) => ({
  ...obj,
  ...obj.Name && { Name: import_smithy_client.SENSITIVE_STRING },
  ...obj.Description && { Description: import_smithy_client.SENSITIVE_STRING }
}), "UpdatePricingPlanInputFilterSensitiveLog");
var UpdatePricingPlanOutputFilterSensitiveLog = /* @__PURE__ */ __name((obj) => ({
  ...obj,
  ...obj.Name && { Name: import_smithy_client.SENSITIVE_STRING },
  ...obj.Description && { Description: import_smithy_client.SENSITIVE_STRING }
}), "UpdatePricingPlanOutputFilterSensitiveLog");
var CreatePricingRuleInputFilterSensitiveLog = /* @__PURE__ */ __name((obj) => ({
  ...obj,
  ...obj.Name && { Name: import_smithy_client.SENSITIVE_STRING },
  ...obj.Description && { Description: import_smithy_client.SENSITIVE_STRING }
}), "CreatePricingRuleInputFilterSensitiveLog");
var PricingRuleListElementFilterSensitiveLog = /* @__PURE__ */ __name((obj) => ({
  ...obj,
  ...obj.Name && { Name: import_smithy_client.SENSITIVE_STRING },
  ...obj.Description && { Description: import_smithy_client.SENSITIVE_STRING }
}), "PricingRuleListElementFilterSensitiveLog");
var ListPricingRulesOutputFilterSensitiveLog = /* @__PURE__ */ __name((obj) => ({
  ...obj,
  ...obj.PricingRules && {
    PricingRules: obj.PricingRules.map((item) => PricingRuleListElementFilterSensitiveLog(item))
  }
}), "ListPricingRulesOutputFilterSensitiveLog");
var UpdatePricingRuleInputFilterSensitiveLog = /* @__PURE__ */ __name((obj) => ({
  ...obj,
  ...obj.Name && { Name: import_smithy_client.SENSITIVE_STRING },
  ...obj.Description && { Description: import_smithy_client.SENSITIVE_STRING }
}), "UpdatePricingRuleInputFilterSensitiveLog");
var UpdatePricingRuleOutputFilterSensitiveLog = /* @__PURE__ */ __name((obj) => ({
  ...obj,
  ...obj.Name && { Name: import_smithy_client.SENSITIVE_STRING },
  ...obj.Description && { Description: import_smithy_client.SENSITIVE_STRING }
}), "UpdatePricingRuleOutputFilterSensitiveLog");

// src/protocols/Aws_restJson1.ts
var se_AssociateAccountsCommand = /* @__PURE__ */ __name(async (input, context) => {
  const b = (0, import_core.requestBuilder)(input, context);
  const headers = {
    "content-type": "application/json"
  };
  b.bp("/associate-accounts");
  let body;
  body = JSON.stringify(
    (0, import_smithy_client.take)(input, {
      AccountIds: (_) => (0, import_smithy_client._json)(_),
      Arn: []
    })
  );
  b.m("POST").h(headers).b(body);
  return b.build();
}, "se_AssociateAccountsCommand");
var se_AssociatePricingRulesCommand = /* @__PURE__ */ __name(async (input, context) => {
  const b = (0, import_core.requestBuilder)(input, context);
  const headers = {
    "content-type": "application/json"
  };
  b.bp("/associate-pricing-rules");
  let body;
  body = JSON.stringify(
    (0, import_smithy_client.take)(input, {
      Arn: [],
      PricingRuleArns: (_) => (0, import_smithy_client._json)(_)
    })
  );
  b.m("PUT").h(headers).b(body);
  return b.build();
}, "se_AssociatePricingRulesCommand");
var se_BatchAssociateResourcesToCustomLineItemCommand = /* @__PURE__ */ __name(async (input, context) => {
  const b = (0, import_core.requestBuilder)(input, context);
  const headers = {
    "content-type": "application/json"
  };
  b.bp("/batch-associate-resources-to-custom-line-item");
  let body;
  body = JSON.stringify(
    (0, import_smithy_client.take)(input, {
      BillingPeriodRange: (_) => (0, import_smithy_client._json)(_),
      ResourceArns: (_) => (0, import_smithy_client._json)(_),
      TargetArn: []
    })
  );
  b.m("PUT").h(headers).b(body);
  return b.build();
}, "se_BatchAssociateResourcesToCustomLineItemCommand");
var se_BatchDisassociateResourcesFromCustomLineItemCommand = /* @__PURE__ */ __name(async (input, context) => {
  const b = (0, import_core.requestBuilder)(input, context);
  const headers = {
    "content-type": "application/json"
  };
  b.bp("/batch-disassociate-resources-from-custom-line-item");
  let body;
  body = JSON.stringify(
    (0, import_smithy_client.take)(input, {
      BillingPeriodRange: (_) => (0, import_smithy_client._json)(_),
      ResourceArns: (_) => (0, import_smithy_client._json)(_),
      TargetArn: []
    })
  );
  b.m("PUT").h(headers).b(body);
  return b.build();
}, "se_BatchDisassociateResourcesFromCustomLineItemCommand");
var se_CreateBillingGroupCommand = /* @__PURE__ */ __name(async (input, context) => {
  const b = (0, import_core.requestBuilder)(input, context);
  const headers = (0, import_smithy_client.map)({}, isSerializableHeaderValue, {
    "content-type": "application/json",
    [_xact]: input[_CT]
  });
  b.bp("/create-billing-group");
  let body;
  body = JSON.stringify(
    (0, import_smithy_client.take)(input, {
      AccountGrouping: (_) => (0, import_smithy_client._json)(_),
      ComputationPreference: (_) => (0, import_smithy_client._json)(_),
      Description: [],
      Name: [],
      PrimaryAccountId: [],
      Tags: (_) => (0, import_smithy_client._json)(_)
    })
  );
  b.m("POST").h(headers).b(body);
  return b.build();
}, "se_CreateBillingGroupCommand");
var se_CreateCustomLineItemCommand = /* @__PURE__ */ __name(async (input, context) => {
  const b = (0, import_core.requestBuilder)(input, context);
  const headers = (0, import_smithy_client.map)({}, isSerializableHeaderValue, {
    "content-type": "application/json",
    [_xact]: input[_CT]
  });
  b.bp("/create-custom-line-item");
  let body;
  body = JSON.stringify(
    (0, import_smithy_client.take)(input, {
      AccountId: [],
      BillingGroupArn: [],
      BillingPeriodRange: (_) => (0, import_smithy_client._json)(_),
      ChargeDetails: (_) => se_CustomLineItemChargeDetails(_, context),
      Description: [],
      Name: [],
      Tags: (_) => (0, import_smithy_client._json)(_)
    })
  );
  b.m("POST").h(headers).b(body);
  return b.build();
}, "se_CreateCustomLineItemCommand");
var se_CreatePricingPlanCommand = /* @__PURE__ */ __name(async (input, context) => {
  const b = (0, import_core.requestBuilder)(input, context);
  const headers = (0, import_smithy_client.map)({}, isSerializableHeaderValue, {
    "content-type": "application/json",
    [_xact]: input[_CT]
  });
  b.bp("/create-pricing-plan");
  let body;
  body = JSON.stringify(
    (0, import_smithy_client.take)(input, {
      Description: [],
      Name: [],
      PricingRuleArns: (_) => (0, import_smithy_client._json)(_),
      Tags: (_) => (0, import_smithy_client._json)(_)
    })
  );
  b.m("POST").h(headers).b(body);
  return b.build();
}, "se_CreatePricingPlanCommand");
var se_CreatePricingRuleCommand = /* @__PURE__ */ __name(async (input, context) => {
  const b = (0, import_core.requestBuilder)(input, context);
  const headers = (0, import_smithy_client.map)({}, isSerializableHeaderValue, {
    "content-type": "application/json",
    [_xact]: input[_CT]
  });
  b.bp("/create-pricing-rule");
  let body;
  body = JSON.stringify(
    (0, import_smithy_client.take)(input, {
      BillingEntity: [],
      Description: [],
      ModifierPercentage: (_) => (0, import_smithy_client.serializeFloat)(_),
      Name: [],
      Operation: [],
      Scope: [],
      Service: [],
      Tags: (_) => (0, import_smithy_client._json)(_),
      Tiering: (_) => (0, import_smithy_client._json)(_),
      Type: [],
      UsageType: []
    })
  );
  b.m("POST").h(headers).b(body);
  return b.build();
}, "se_CreatePricingRuleCommand");
var se_DeleteBillingGroupCommand = /* @__PURE__ */ __name(async (input, context) => {
  const b = (0, import_core.requestBuilder)(input, context);
  const headers = {
    "content-type": "application/json"
  };
  b.bp("/delete-billing-group");
  let body;
  body = JSON.stringify(
    (0, import_smithy_client.take)(input, {
      Arn: []
    })
  );
  b.m("POST").h(headers).b(body);
  return b.build();
}, "se_DeleteBillingGroupCommand");
var se_DeleteCustomLineItemCommand = /* @__PURE__ */ __name(async (input, context) => {
  const b = (0, import_core.requestBuilder)(input, context);
  const headers = {
    "content-type": "application/json"
  };
  b.bp("/delete-custom-line-item");
  let body;
  body = JSON.stringify(
    (0, import_smithy_client.take)(input, {
      Arn: [],
      BillingPeriodRange: (_) => (0, import_smithy_client._json)(_)
    })
  );
  b.m("POST").h(headers).b(body);
  return b.build();
}, "se_DeleteCustomLineItemCommand");
var se_DeletePricingPlanCommand = /* @__PURE__ */ __name(async (input, context) => {
  const b = (0, import_core.requestBuilder)(input, context);
  const headers = {
    "content-type": "application/json"
  };
  b.bp("/delete-pricing-plan");
  let body;
  body = JSON.stringify(
    (0, import_smithy_client.take)(input, {
      Arn: []
    })
  );
  b.m("POST").h(headers).b(body);
  return b.build();
}, "se_DeletePricingPlanCommand");
var se_DeletePricingRuleCommand = /* @__PURE__ */ __name(async (input, context) => {
  const b = (0, import_core.requestBuilder)(input, context);
  const headers = {
    "content-type": "application/json"
  };
  b.bp("/delete-pricing-rule");
  let body;
  body = JSON.stringify(
    (0, import_smithy_client.take)(input, {
      Arn: []
    })
  );
  b.m("POST").h(headers).b(body);
  return b.build();
}, "se_DeletePricingRuleCommand");
var se_DisassociateAccountsCommand = /* @__PURE__ */ __name(async (input, context) => {
  const b = (0, import_core.requestBuilder)(input, context);
  const headers = {
    "content-type": "application/json"
  };
  b.bp("/disassociate-accounts");
  let body;
  body = JSON.stringify(
    (0, import_smithy_client.take)(input, {
      AccountIds: (_) => (0, import_smithy_client._json)(_),
      Arn: []
    })
  );
  b.m("POST").h(headers).b(body);
  return b.build();
}, "se_DisassociateAccountsCommand");
var se_DisassociatePricingRulesCommand = /* @__PURE__ */ __name(async (input, context) => {
  const b = (0, import_core.requestBuilder)(input, context);
  const headers = {
    "content-type": "application/json"
  };
  b.bp("/disassociate-pricing-rules");
  let body;
  body = JSON.stringify(
    (0, import_smithy_client.take)(input, {
      Arn: [],
      PricingRuleArns: (_) => (0, import_smithy_client._json)(_)
    })
  );
  b.m("PUT").h(headers).b(body);
  return b.build();
}, "se_DisassociatePricingRulesCommand");
var se_GetBillingGroupCostReportCommand = /* @__PURE__ */ __name(async (input, context) => {
  const b = (0, import_core.requestBuilder)(input, context);
  const headers = {
    "content-type": "application/json"
  };
  b.bp("/get-billing-group-cost-report");
  let body;
  body = JSON.stringify(
    (0, import_smithy_client.take)(input, {
      Arn: [],
      BillingPeriodRange: (_) => (0, import_smithy_client._json)(_),
      GroupBy: (_) => (0, import_smithy_client._json)(_),
      MaxResults: [],
      NextToken: []
    })
  );
  b.m("POST").h(headers).b(body);
  return b.build();
}, "se_GetBillingGroupCostReportCommand");
var se_ListAccountAssociationsCommand = /* @__PURE__ */ __name(async (input, context) => {
  const b = (0, import_core.requestBuilder)(input, context);
  const headers = {
    "content-type": "application/json"
  };
  b.bp("/list-account-associations");
  let body;
  body = JSON.stringify(
    (0, import_smithy_client.take)(input, {
      BillingPeriod: [],
      Filters: (_) => (0, import_smithy_client._json)(_),
      NextToken: []
    })
  );
  b.m("POST").h(headers).b(body);
  return b.build();
}, "se_ListAccountAssociationsCommand");
var se_ListBillingGroupCostReportsCommand = /* @__PURE__ */ __name(async (input, context) => {
  const b = (0, import_core.requestBuilder)(input, context);
  const headers = {
    "content-type": "application/json"
  };
  b.bp("/list-billing-group-cost-reports");
  let body;
  body = JSON.stringify(
    (0, import_smithy_client.take)(input, {
      BillingPeriod: [],
      Filters: (_) => (0, import_smithy_client._json)(_),
      MaxResults: [],
      NextToken: []
    })
  );
  b.m("POST").h(headers).b(body);
  return b.build();
}, "se_ListBillingGroupCostReportsCommand");
var se_ListBillingGroupsCommand = /* @__PURE__ */ __name(async (input, context) => {
  const b = (0, import_core.requestBuilder)(input, context);
  const headers = {
    "content-type": "application/json"
  };
  b.bp("/list-billing-groups");
  let body;
  body = JSON.stringify(
    (0, import_smithy_client.take)(input, {
      BillingPeriod: [],
      Filters: (_) => (0, import_smithy_client._json)(_),
      MaxResults: [],
      NextToken: []
    })
  );
  b.m("POST").h(headers).b(body);
  return b.build();
}, "se_ListBillingGroupsCommand");
var se_ListCustomLineItemsCommand = /* @__PURE__ */ __name(async (input, context) => {
  const b = (0, import_core.requestBuilder)(input, context);
  const headers = {
    "content-type": "application/json"
  };
  b.bp("/list-custom-line-items");
  let body;
  body = JSON.stringify(
    (0, import_smithy_client.take)(input, {
      BillingPeriod: [],
      Filters: (_) => (0, import_smithy_client._json)(_),
      MaxResults: [],
      NextToken: []
    })
  );
  b.m("POST").h(headers).b(body);
  return b.build();
}, "se_ListCustomLineItemsCommand");
var se_ListCustomLineItemVersionsCommand = /* @__PURE__ */ __name(async (input, context) => {
  const b = (0, import_core.requestBuilder)(input, context);
  const headers = {
    "content-type": "application/json"
  };
  b.bp("/list-custom-line-item-versions");
  let body;
  body = JSON.stringify(
    (0, import_smithy_client.take)(input, {
      Arn: [],
      Filters: (_) => (0, import_smithy_client._json)(_),
      MaxResults: [],
      NextToken: []
    })
  );
  b.m("POST").h(headers).b(body);
  return b.build();
}, "se_ListCustomLineItemVersionsCommand");
var se_ListPricingPlansCommand = /* @__PURE__ */ __name(async (input, context) => {
  const b = (0, import_core.requestBuilder)(input, context);
  const headers = {
    "content-type": "application/json"
  };
  b.bp("/list-pricing-plans");
  let body;
  body = JSON.stringify(
    (0, import_smithy_client.take)(input, {
      BillingPeriod: [],
      Filters: (_) => (0, import_smithy_client._json)(_),
      MaxResults: [],
      NextToken: []
    })
  );
  b.m("POST").h(headers).b(body);
  return b.build();
}, "se_ListPricingPlansCommand");
var se_ListPricingPlansAssociatedWithPricingRuleCommand = /* @__PURE__ */ __name(async (input, context) => {
  const b = (0, import_core.requestBuilder)(input, context);
  const headers = {
    "content-type": "application/json"
  };
  b.bp("/list-pricing-plans-associated-with-pricing-rule");
  let body;
  body = JSON.stringify(
    (0, import_smithy_client.take)(input, {
      BillingPeriod: [],
      MaxResults: [],
      NextToken: [],
      PricingRuleArn: []
    })
  );
  b.m("POST").h(headers).b(body);
  return b.build();
}, "se_ListPricingPlansAssociatedWithPricingRuleCommand");
var se_ListPricingRulesCommand = /* @__PURE__ */ __name(async (input, context) => {
  const b = (0, import_core.requestBuilder)(input, context);
  const headers = {
    "content-type": "application/json"
  };
  b.bp("/list-pricing-rules");
  let body;
  body = JSON.stringify(
    (0, import_smithy_client.take)(input, {
      BillingPeriod: [],
      Filters: (_) => (0, import_smithy_client._json)(_),
      MaxResults: [],
      NextToken: []
    })
  );
  b.m("POST").h(headers).b(body);
  return b.build();
}, "se_ListPricingRulesCommand");
var se_ListPricingRulesAssociatedToPricingPlanCommand = /* @__PURE__ */ __name(async (input, context) => {
  const b = (0, import_core.requestBuilder)(input, context);
  const headers = {
    "content-type": "application/json"
  };
  b.bp("/list-pricing-rules-associated-to-pricing-plan");
  let body;
  body = JSON.stringify(
    (0, import_smithy_client.take)(input, {
      BillingPeriod: [],
      MaxResults: [],
      NextToken: [],
      PricingPlanArn: []
    })
  );
  b.m("POST").h(headers).b(body);
  return b.build();
}, "se_ListPricingRulesAssociatedToPricingPlanCommand");
var se_ListResourcesAssociatedToCustomLineItemCommand = /* @__PURE__ */ __name(async (input, context) => {
  const b = (0, import_core.requestBuilder)(input, context);
  const headers = {
    "content-type": "application/json"
  };
  b.bp("/list-resources-associated-to-custom-line-item");
  let body;
  body = JSON.stringify(
    (0, import_smithy_client.take)(input, {
      Arn: [],
      BillingPeriod: [],
      Filters: (_) => (0, import_smithy_client._json)(_),
      MaxResults: [],
      NextToken: []
    })
  );
  b.m("POST").h(headers).b(body);
  return b.build();
}, "se_ListResourcesAssociatedToCustomLineItemCommand");
var se_ListTagsForResourceCommand = /* @__PURE__ */ __name(async (input, context) => {
  const b = (0, import_core.requestBuilder)(input, context);
  const headers = {};
  b.bp("/tags/{ResourceArn}");
  b.p("ResourceArn", () => input.ResourceArn, "{ResourceArn}", false);
  let body;
  b.m("GET").h(headers).b(body);
  return b.build();
}, "se_ListTagsForResourceCommand");
var se_TagResourceCommand = /* @__PURE__ */ __name(async (input, context) => {
  const b = (0, import_core.requestBuilder)(input, context);
  const headers = {
    "content-type": "application/json"
  };
  b.bp("/tags/{ResourceArn}");
  b.p("ResourceArn", () => input.ResourceArn, "{ResourceArn}", false);
  let body;
  body = JSON.stringify(
    (0, import_smithy_client.take)(input, {
      Tags: (_) => (0, import_smithy_client._json)(_)
    })
  );
  b.m("POST").h(headers).b(body);
  return b.build();
}, "se_TagResourceCommand");
var se_UntagResourceCommand = /* @__PURE__ */ __name(async (input, context) => {
  const b = (0, import_core.requestBuilder)(input, context);
  const headers = {};
  b.bp("/tags/{ResourceArn}");
  b.p("ResourceArn", () => input.ResourceArn, "{ResourceArn}", false);
  const query = (0, import_smithy_client.map)({
    [_tK]: [
      (0, import_smithy_client.expectNonNull)(input.TagKeys, `TagKeys`) != null,
      () => (input[_TK] || []).map((_entry) => _entry)
    ]
  });
  let body;
  b.m("DELETE").h(headers).q(query).b(body);
  return b.build();
}, "se_UntagResourceCommand");
var se_UpdateBillingGroupCommand = /* @__PURE__ */ __name(async (input, context) => {
  const b = (0, import_core.requestBuilder)(input, context);
  const headers = {
    "content-type": "application/json"
  };
  b.bp("/update-billing-group");
  let body;
  body = JSON.stringify(
    (0, import_smithy_client.take)(input, {
      AccountGrouping: (_) => (0, import_smithy_client._json)(_),
      Arn: [],
      ComputationPreference: (_) => (0, import_smithy_client._json)(_),
      Description: [],
      Name: [],
      Status: []
    })
  );
  b.m("POST").h(headers).b(body);
  return b.build();
}, "se_UpdateBillingGroupCommand");
var se_UpdateCustomLineItemCommand = /* @__PURE__ */ __name(async (input, context) => {
  const b = (0, import_core.requestBuilder)(input, context);
  const headers = {
    "content-type": "application/json"
  };
  b.bp("/update-custom-line-item");
  let body;
  body = JSON.stringify(
    (0, import_smithy_client.take)(input, {
      Arn: [],
      BillingPeriodRange: (_) => (0, import_smithy_client._json)(_),
      ChargeDetails: (_) => se_UpdateCustomLineItemChargeDetails(_, context),
      Description: [],
      Name: []
    })
  );
  b.m("POST").h(headers).b(body);
  return b.build();
}, "se_UpdateCustomLineItemCommand");
var se_UpdatePricingPlanCommand = /* @__PURE__ */ __name(async (input, context) => {
  const b = (0, import_core.requestBuilder)(input, context);
  const headers = {
    "content-type": "application/json"
  };
  b.bp("/update-pricing-plan");
  let body;
  body = JSON.stringify(
    (0, import_smithy_client.take)(input, {
      Arn: [],
      Description: [],
      Name: []
    })
  );
  b.m("PUT").h(headers).b(body);
  return b.build();
}, "se_UpdatePricingPlanCommand");
var se_UpdatePricingRuleCommand = /* @__PURE__ */ __name(async (input, context) => {
  const b = (0, import_core.requestBuilder)(input, context);
  const headers = {
    "content-type": "application/json"
  };
  b.bp("/update-pricing-rule");
  let body;
  body = JSON.stringify(
    (0, import_smithy_client.take)(input, {
      Arn: [],
      Description: [],
      ModifierPercentage: (_) => (0, import_smithy_client.serializeFloat)(_),
      Name: [],
      Tiering: (_) => (0, import_smithy_client._json)(_),
      Type: []
    })
  );
  b.m("PUT").h(headers).b(body);
  return b.build();
}, "se_UpdatePricingRuleCommand");
var de_AssociateAccountsCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode !== 200 && output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const contents = (0, import_smithy_client.map)({
    $metadata: deserializeMetadata(output)
  });
  const data = (0, import_smithy_client.expectNonNull)((0, import_smithy_client.expectObject)(await (0, import_core2.parseJsonBody)(output.body, context)), "body");
  const doc = (0, import_smithy_client.take)(data, {
    Arn: import_smithy_client.expectString
  });
  Object.assign(contents, doc);
  return contents;
}, "de_AssociateAccountsCommand");
var de_AssociatePricingRulesCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode !== 200 && output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const contents = (0, import_smithy_client.map)({
    $metadata: deserializeMetadata(output)
  });
  const data = (0, import_smithy_client.expectNonNull)((0, import_smithy_client.expectObject)(await (0, import_core2.parseJsonBody)(output.body, context)), "body");
  const doc = (0, import_smithy_client.take)(data, {
    Arn: import_smithy_client.expectString
  });
  Object.assign(contents, doc);
  return contents;
}, "de_AssociatePricingRulesCommand");
var de_BatchAssociateResourcesToCustomLineItemCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode !== 200 && output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const contents = (0, import_smithy_client.map)({
    $metadata: deserializeMetadata(output)
  });
  const data = (0, import_smithy_client.expectNonNull)((0, import_smithy_client.expectObject)(await (0, import_core2.parseJsonBody)(output.body, context)), "body");
  const doc = (0, import_smithy_client.take)(data, {
    FailedAssociatedResources: import_smithy_client._json,
    SuccessfullyAssociatedResources: import_smithy_client._json
  });
  Object.assign(contents, doc);
  return contents;
}, "de_BatchAssociateResourcesToCustomLineItemCommand");
var de_BatchDisassociateResourcesFromCustomLineItemCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode !== 200 && output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const contents = (0, import_smithy_client.map)({
    $metadata: deserializeMetadata(output)
  });
  const data = (0, import_smithy_client.expectNonNull)((0, import_smithy_client.expectObject)(await (0, import_core2.parseJsonBody)(output.body, context)), "body");
  const doc = (0, import_smithy_client.take)(data, {
    FailedDisassociatedResources: import_smithy_client._json,
    SuccessfullyDisassociatedResources: import_smithy_client._json
  });
  Object.assign(contents, doc);
  return contents;
}, "de_BatchDisassociateResourcesFromCustomLineItemCommand");
var de_CreateBillingGroupCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode !== 200 && output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const contents = (0, import_smithy_client.map)({
    $metadata: deserializeMetadata(output)
  });
  const data = (0, import_smithy_client.expectNonNull)((0, import_smithy_client.expectObject)(await (0, import_core2.parseJsonBody)(output.body, context)), "body");
  const doc = (0, import_smithy_client.take)(data, {
    Arn: import_smithy_client.expectString
  });
  Object.assign(contents, doc);
  return contents;
}, "de_CreateBillingGroupCommand");
var de_CreateCustomLineItemCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode !== 200 && output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const contents = (0, import_smithy_client.map)({
    $metadata: deserializeMetadata(output)
  });
  const data = (0, import_smithy_client.expectNonNull)((0, import_smithy_client.expectObject)(await (0, import_core2.parseJsonBody)(output.body, context)), "body");
  const doc = (0, import_smithy_client.take)(data, {
    Arn: import_smithy_client.expectString
  });
  Object.assign(contents, doc);
  return contents;
}, "de_CreateCustomLineItemCommand");
var de_CreatePricingPlanCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode !== 200 && output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const contents = (0, import_smithy_client.map)({
    $metadata: deserializeMetadata(output)
  });
  const data = (0, import_smithy_client.expectNonNull)((0, import_smithy_client.expectObject)(await (0, import_core2.parseJsonBody)(output.body, context)), "body");
  const doc = (0, import_smithy_client.take)(data, {
    Arn: import_smithy_client.expectString
  });
  Object.assign(contents, doc);
  return contents;
}, "de_CreatePricingPlanCommand");
var de_CreatePricingRuleCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode !== 200 && output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const contents = (0, import_smithy_client.map)({
    $metadata: deserializeMetadata(output)
  });
  const data = (0, import_smithy_client.expectNonNull)((0, import_smithy_client.expectObject)(await (0, import_core2.parseJsonBody)(output.body, context)), "body");
  const doc = (0, import_smithy_client.take)(data, {
    Arn: import_smithy_client.expectString
  });
  Object.assign(contents, doc);
  return contents;
}, "de_CreatePricingRuleCommand");
var de_DeleteBillingGroupCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode !== 200 && output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const contents = (0, import_smithy_client.map)({
    $metadata: deserializeMetadata(output)
  });
  const data = (0, import_smithy_client.expectNonNull)((0, import_smithy_client.expectObject)(await (0, import_core2.parseJsonBody)(output.body, context)), "body");
  const doc = (0, import_smithy_client.take)(data, {
    Arn: import_smithy_client.expectString
  });
  Object.assign(contents, doc);
  return contents;
}, "de_DeleteBillingGroupCommand");
var de_DeleteCustomLineItemCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode !== 200 && output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const contents = (0, import_smithy_client.map)({
    $metadata: deserializeMetadata(output)
  });
  const data = (0, import_smithy_client.expectNonNull)((0, import_smithy_client.expectObject)(await (0, import_core2.parseJsonBody)(output.body, context)), "body");
  const doc = (0, import_smithy_client.take)(data, {
    Arn: import_smithy_client.expectString
  });
  Object.assign(contents, doc);
  return contents;
}, "de_DeleteCustomLineItemCommand");
var de_DeletePricingPlanCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode !== 200 && output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const contents = (0, import_smithy_client.map)({
    $metadata: deserializeMetadata(output)
  });
  const data = (0, import_smithy_client.expectNonNull)((0, import_smithy_client.expectObject)(await (0, import_core2.parseJsonBody)(output.body, context)), "body");
  const doc = (0, import_smithy_client.take)(data, {
    Arn: import_smithy_client.expectString
  });
  Object.assign(contents, doc);
  return contents;
}, "de_DeletePricingPlanCommand");
var de_DeletePricingRuleCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode !== 200 && output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const contents = (0, import_smithy_client.map)({
    $metadata: deserializeMetadata(output)
  });
  const data = (0, import_smithy_client.expectNonNull)((0, import_smithy_client.expectObject)(await (0, import_core2.parseJsonBody)(output.body, context)), "body");
  const doc = (0, import_smithy_client.take)(data, {
    Arn: import_smithy_client.expectString
  });
  Object.assign(contents, doc);
  return contents;
}, "de_DeletePricingRuleCommand");
var de_DisassociateAccountsCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode !== 200 && output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const contents = (0, import_smithy_client.map)({
    $metadata: deserializeMetadata(output)
  });
  const data = (0, import_smithy_client.expectNonNull)((0, import_smithy_client.expectObject)(await (0, import_core2.parseJsonBody)(output.body, context)), "body");
  const doc = (0, import_smithy_client.take)(data, {
    Arn: import_smithy_client.expectString
  });
  Object.assign(contents, doc);
  return contents;
}, "de_DisassociateAccountsCommand");
var de_DisassociatePricingRulesCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode !== 200 && output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const contents = (0, import_smithy_client.map)({
    $metadata: deserializeMetadata(output)
  });
  const data = (0, import_smithy_client.expectNonNull)((0, import_smithy_client.expectObject)(await (0, import_core2.parseJsonBody)(output.body, context)), "body");
  const doc = (0, import_smithy_client.take)(data, {
    Arn: import_smithy_client.expectString
  });
  Object.assign(contents, doc);
  return contents;
}, "de_DisassociatePricingRulesCommand");
var de_GetBillingGroupCostReportCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode !== 200 && output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const contents = (0, import_smithy_client.map)({
    $metadata: deserializeMetadata(output)
  });
  const data = (0, import_smithy_client.expectNonNull)((0, import_smithy_client.expectObject)(await (0, import_core2.parseJsonBody)(output.body, context)), "body");
  const doc = (0, import_smithy_client.take)(data, {
    BillingGroupCostReportResults: import_smithy_client._json,
    NextToken: import_smithy_client.expectString
  });
  Object.assign(contents, doc);
  return contents;
}, "de_GetBillingGroupCostReportCommand");
var de_ListAccountAssociationsCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode !== 200 && output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const contents = (0, import_smithy_client.map)({
    $metadata: deserializeMetadata(output)
  });
  const data = (0, import_smithy_client.expectNonNull)((0, import_smithy_client.expectObject)(await (0, import_core2.parseJsonBody)(output.body, context)), "body");
  const doc = (0, import_smithy_client.take)(data, {
    LinkedAccounts: import_smithy_client._json,
    NextToken: import_smithy_client.expectString
  });
  Object.assign(contents, doc);
  return contents;
}, "de_ListAccountAssociationsCommand");
var de_ListBillingGroupCostReportsCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode !== 200 && output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const contents = (0, import_smithy_client.map)({
    $metadata: deserializeMetadata(output)
  });
  const data = (0, import_smithy_client.expectNonNull)((0, import_smithy_client.expectObject)(await (0, import_core2.parseJsonBody)(output.body, context)), "body");
  const doc = (0, import_smithy_client.take)(data, {
    BillingGroupCostReports: import_smithy_client._json,
    NextToken: import_smithy_client.expectString
  });
  Object.assign(contents, doc);
  return contents;
}, "de_ListBillingGroupCostReportsCommand");
var de_ListBillingGroupsCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode !== 200 && output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const contents = (0, import_smithy_client.map)({
    $metadata: deserializeMetadata(output)
  });
  const data = (0, import_smithy_client.expectNonNull)((0, import_smithy_client.expectObject)(await (0, import_core2.parseJsonBody)(output.body, context)), "body");
  const doc = (0, import_smithy_client.take)(data, {
    BillingGroups: import_smithy_client._json,
    NextToken: import_smithy_client.expectString
  });
  Object.assign(contents, doc);
  return contents;
}, "de_ListBillingGroupsCommand");
var de_ListCustomLineItemsCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode !== 200 && output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const contents = (0, import_smithy_client.map)({
    $metadata: deserializeMetadata(output)
  });
  const data = (0, import_smithy_client.expectNonNull)((0, import_smithy_client.expectObject)(await (0, import_core2.parseJsonBody)(output.body, context)), "body");
  const doc = (0, import_smithy_client.take)(data, {
    CustomLineItems: (_) => de_CustomLineItemList(_, context),
    NextToken: import_smithy_client.expectString
  });
  Object.assign(contents, doc);
  return contents;
}, "de_ListCustomLineItemsCommand");
var de_ListCustomLineItemVersionsCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode !== 200 && output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const contents = (0, import_smithy_client.map)({
    $metadata: deserializeMetadata(output)
  });
  const data = (0, import_smithy_client.expectNonNull)((0, import_smithy_client.expectObject)(await (0, import_core2.parseJsonBody)(output.body, context)), "body");
  const doc = (0, import_smithy_client.take)(data, {
    CustomLineItemVersions: (_) => de_CustomLineItemVersionList(_, context),
    NextToken: import_smithy_client.expectString
  });
  Object.assign(contents, doc);
  return contents;
}, "de_ListCustomLineItemVersionsCommand");
var de_ListPricingPlansCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode !== 200 && output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const contents = (0, import_smithy_client.map)({
    $metadata: deserializeMetadata(output)
  });
  const data = (0, import_smithy_client.expectNonNull)((0, import_smithy_client.expectObject)(await (0, import_core2.parseJsonBody)(output.body, context)), "body");
  const doc = (0, import_smithy_client.take)(data, {
    BillingPeriod: import_smithy_client.expectString,
    NextToken: import_smithy_client.expectString,
    PricingPlans: import_smithy_client._json
  });
  Object.assign(contents, doc);
  return contents;
}, "de_ListPricingPlansCommand");
var de_ListPricingPlansAssociatedWithPricingRuleCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode !== 200 && output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const contents = (0, import_smithy_client.map)({
    $metadata: deserializeMetadata(output)
  });
  const data = (0, import_smithy_client.expectNonNull)((0, import_smithy_client.expectObject)(await (0, import_core2.parseJsonBody)(output.body, context)), "body");
  const doc = (0, import_smithy_client.take)(data, {
    BillingPeriod: import_smithy_client.expectString,
    NextToken: import_smithy_client.expectString,
    PricingPlanArns: import_smithy_client._json,
    PricingRuleArn: import_smithy_client.expectString
  });
  Object.assign(contents, doc);
  return contents;
}, "de_ListPricingPlansAssociatedWithPricingRuleCommand");
var de_ListPricingRulesCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode !== 200 && output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const contents = (0, import_smithy_client.map)({
    $metadata: deserializeMetadata(output)
  });
  const data = (0, import_smithy_client.expectNonNull)((0, import_smithy_client.expectObject)(await (0, import_core2.parseJsonBody)(output.body, context)), "body");
  const doc = (0, import_smithy_client.take)(data, {
    BillingPeriod: import_smithy_client.expectString,
    NextToken: import_smithy_client.expectString,
    PricingRules: (_) => de_PricingRuleList(_, context)
  });
  Object.assign(contents, doc);
  return contents;
}, "de_ListPricingRulesCommand");
var de_ListPricingRulesAssociatedToPricingPlanCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode !== 200 && output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const contents = (0, import_smithy_client.map)({
    $metadata: deserializeMetadata(output)
  });
  const data = (0, import_smithy_client.expectNonNull)((0, import_smithy_client.expectObject)(await (0, import_core2.parseJsonBody)(output.body, context)), "body");
  const doc = (0, import_smithy_client.take)(data, {
    BillingPeriod: import_smithy_client.expectString,
    NextToken: import_smithy_client.expectString,
    PricingPlanArn: import_smithy_client.expectString,
    PricingRuleArns: import_smithy_client._json
  });
  Object.assign(contents, doc);
  return contents;
}, "de_ListPricingRulesAssociatedToPricingPlanCommand");
var de_ListResourcesAssociatedToCustomLineItemCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode !== 200 && output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const contents = (0, import_smithy_client.map)({
    $metadata: deserializeMetadata(output)
  });
  const data = (0, import_smithy_client.expectNonNull)((0, import_smithy_client.expectObject)(await (0, import_core2.parseJsonBody)(output.body, context)), "body");
  const doc = (0, import_smithy_client.take)(data, {
    Arn: import_smithy_client.expectString,
    AssociatedResources: import_smithy_client._json,
    NextToken: import_smithy_client.expectString
  });
  Object.assign(contents, doc);
  return contents;
}, "de_ListResourcesAssociatedToCustomLineItemCommand");
var de_ListTagsForResourceCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode !== 204 && output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const contents = (0, import_smithy_client.map)({
    $metadata: deserializeMetadata(output)
  });
  const data = (0, import_smithy_client.expectNonNull)((0, import_smithy_client.expectObject)(await (0, import_core2.parseJsonBody)(output.body, context)), "body");
  const doc = (0, import_smithy_client.take)(data, {
    Tags: import_smithy_client._json
  });
  Object.assign(contents, doc);
  return contents;
}, "de_ListTagsForResourceCommand");
var de_TagResourceCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode !== 204 && output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const contents = (0, import_smithy_client.map)({
    $metadata: deserializeMetadata(output)
  });
  await (0, import_smithy_client.collectBody)(output.body, context);
  return contents;
}, "de_TagResourceCommand");
var de_UntagResourceCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode !== 204 && output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const contents = (0, import_smithy_client.map)({
    $metadata: deserializeMetadata(output)
  });
  await (0, import_smithy_client.collectBody)(output.body, context);
  return contents;
}, "de_UntagResourceCommand");
var de_UpdateBillingGroupCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode !== 200 && output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const contents = (0, import_smithy_client.map)({
    $metadata: deserializeMetadata(output)
  });
  const data = (0, import_smithy_client.expectNonNull)((0, import_smithy_client.expectObject)(await (0, import_core2.parseJsonBody)(output.body, context)), "body");
  const doc = (0, import_smithy_client.take)(data, {
    AccountGrouping: import_smithy_client._json,
    Arn: import_smithy_client.expectString,
    Description: import_smithy_client.expectString,
    LastModifiedTime: import_smithy_client.expectLong,
    Name: import_smithy_client.expectString,
    PricingPlanArn: import_smithy_client.expectString,
    PrimaryAccountId: import_smithy_client.expectString,
    Size: import_smithy_client.expectLong,
    Status: import_smithy_client.expectString,
    StatusReason: import_smithy_client.expectString
  });
  Object.assign(contents, doc);
  return contents;
}, "de_UpdateBillingGroupCommand");
var de_UpdateCustomLineItemCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode !== 200 && output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const contents = (0, import_smithy_client.map)({
    $metadata: deserializeMetadata(output)
  });
  const data = (0, import_smithy_client.expectNonNull)((0, import_smithy_client.expectObject)(await (0, import_core2.parseJsonBody)(output.body, context)), "body");
  const doc = (0, import_smithy_client.take)(data, {
    Arn: import_smithy_client.expectString,
    AssociationSize: import_smithy_client.expectLong,
    BillingGroupArn: import_smithy_client.expectString,
    ChargeDetails: (_) => de_ListCustomLineItemChargeDetails(_, context),
    Description: import_smithy_client.expectString,
    LastModifiedTime: import_smithy_client.expectLong,
    Name: import_smithy_client.expectString
  });
  Object.assign(contents, doc);
  return contents;
}, "de_UpdateCustomLineItemCommand");
var de_UpdatePricingPlanCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode !== 200 && output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const contents = (0, import_smithy_client.map)({
    $metadata: deserializeMetadata(output)
  });
  const data = (0, import_smithy_client.expectNonNull)((0, import_smithy_client.expectObject)(await (0, import_core2.parseJsonBody)(output.body, context)), "body");
  const doc = (0, import_smithy_client.take)(data, {
    Arn: import_smithy_client.expectString,
    Description: import_smithy_client.expectString,
    LastModifiedTime: import_smithy_client.expectLong,
    Name: import_smithy_client.expectString,
    Size: import_smithy_client.expectLong
  });
  Object.assign(contents, doc);
  return contents;
}, "de_UpdatePricingPlanCommand");
var de_UpdatePricingRuleCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode !== 200 && output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const contents = (0, import_smithy_client.map)({
    $metadata: deserializeMetadata(output)
  });
  const data = (0, import_smithy_client.expectNonNull)((0, import_smithy_client.expectObject)(await (0, import_core2.parseJsonBody)(output.body, context)), "body");
  const doc = (0, import_smithy_client.take)(data, {
    Arn: import_smithy_client.expectString,
    AssociatedPricingPlanCount: import_smithy_client.expectLong,
    BillingEntity: import_smithy_client.expectString,
    Description: import_smithy_client.expectString,
    LastModifiedTime: import_smithy_client.expectLong,
    ModifierPercentage: import_smithy_client.limitedParseDouble,
    Name: import_smithy_client.expectString,
    Operation: import_smithy_client.expectString,
    Scope: import_smithy_client.expectString,
    Service: import_smithy_client.expectString,
    Tiering: import_smithy_client._json,
    Type: import_smithy_client.expectString,
    UsageType: import_smithy_client.expectString
  });
  Object.assign(contents, doc);
  return contents;
}, "de_UpdatePricingRuleCommand");
var de_CommandError = /* @__PURE__ */ __name(async (output, context) => {
  const parsedOutput = {
    ...output,
    body: await (0, import_core2.parseJsonErrorBody)(output.body, context)
  };
  const errorCode = (0, import_core2.loadRestJsonErrorCode)(output, parsedOutput.body);
  switch (errorCode) {
    case "AccessDeniedException":
    case "com.amazonaws.billingconductor#AccessDeniedException":
      throw await de_AccessDeniedExceptionRes(parsedOutput, context);
    case "ConflictException":
    case "com.amazonaws.billingconductor#ConflictException":
      throw await de_ConflictExceptionRes(parsedOutput, context);
    case "InternalServerException":
    case "com.amazonaws.billingconductor#InternalServerException":
      throw await de_InternalServerExceptionRes(parsedOutput, context);
    case "ResourceNotFoundException":
    case "com.amazonaws.billingconductor#ResourceNotFoundException":
      throw await de_ResourceNotFoundExceptionRes(parsedOutput, context);
    case "ServiceLimitExceededException":
    case "com.amazonaws.billingconductor#ServiceLimitExceededException":
      throw await de_ServiceLimitExceededExceptionRes(parsedOutput, context);
    case "ThrottlingException":
    case "com.amazonaws.billingconductor#ThrottlingException":
      throw await de_ThrottlingExceptionRes(parsedOutput, context);
    case "ValidationException":
    case "com.amazonaws.billingconductor#ValidationException":
      throw await de_ValidationExceptionRes(parsedOutput, context);
    default:
      const parsedBody = parsedOutput.body;
      return throwDefaultError({
        output,
        parsedBody,
        errorCode
      });
  }
}, "de_CommandError");
var throwDefaultError = (0, import_smithy_client.withBaseException)(BillingconductorServiceException);
var de_AccessDeniedExceptionRes = /* @__PURE__ */ __name(async (parsedOutput, context) => {
  const contents = (0, import_smithy_client.map)({});
  const data = parsedOutput.body;
  const doc = (0, import_smithy_client.take)(data, {
    Message: import_smithy_client.expectString
  });
  Object.assign(contents, doc);
  const exception = new AccessDeniedException({
    $metadata: deserializeMetadata(parsedOutput),
    ...contents
  });
  return (0, import_smithy_client.decorateServiceException)(exception, parsedOutput.body);
}, "de_AccessDeniedExceptionRes");
var de_ConflictExceptionRes = /* @__PURE__ */ __name(async (parsedOutput, context) => {
  const contents = (0, import_smithy_client.map)({});
  const data = parsedOutput.body;
  const doc = (0, import_smithy_client.take)(data, {
    Message: import_smithy_client.expectString,
    Reason: import_smithy_client.expectString,
    ResourceId: import_smithy_client.expectString,
    ResourceType: import_smithy_client.expectString
  });
  Object.assign(contents, doc);
  const exception = new ConflictException({
    $metadata: deserializeMetadata(parsedOutput),
    ...contents
  });
  return (0, import_smithy_client.decorateServiceException)(exception, parsedOutput.body);
}, "de_ConflictExceptionRes");
var de_InternalServerExceptionRes = /* @__PURE__ */ __name(async (parsedOutput, context) => {
  const contents = (0, import_smithy_client.map)({
    [_RAS]: [() => void 0 !== parsedOutput.headers[_ra], () => (0, import_smithy_client.strictParseInt32)(parsedOutput.headers[_ra])]
  });
  const data = parsedOutput.body;
  const doc = (0, import_smithy_client.take)(data, {
    Message: import_smithy_client.expectString
  });
  Object.assign(contents, doc);
  const exception = new InternalServerException({
    $metadata: deserializeMetadata(parsedOutput),
    ...contents
  });
  return (0, import_smithy_client.decorateServiceException)(exception, parsedOutput.body);
}, "de_InternalServerExceptionRes");
var de_ResourceNotFoundExceptionRes = /* @__PURE__ */ __name(async (parsedOutput, context) => {
  const contents = (0, import_smithy_client.map)({});
  const data = parsedOutput.body;
  const doc = (0, import_smithy_client.take)(data, {
    Message: import_smithy_client.expectString,
    ResourceId: import_smithy_client.expectString,
    ResourceType: import_smithy_client.expectString
  });
  Object.assign(contents, doc);
  const exception = new ResourceNotFoundException({
    $metadata: deserializeMetadata(parsedOutput),
    ...contents
  });
  return (0, import_smithy_client.decorateServiceException)(exception, parsedOutput.body);
}, "de_ResourceNotFoundExceptionRes");
var de_ServiceLimitExceededExceptionRes = /* @__PURE__ */ __name(async (parsedOutput, context) => {
  const contents = (0, import_smithy_client.map)({});
  const data = parsedOutput.body;
  const doc = (0, import_smithy_client.take)(data, {
    LimitCode: import_smithy_client.expectString,
    Message: import_smithy_client.expectString,
    ResourceId: import_smithy_client.expectString,
    ResourceType: import_smithy_client.expectString,
    ServiceCode: import_smithy_client.expectString
  });
  Object.assign(contents, doc);
  const exception = new ServiceLimitExceededException({
    $metadata: deserializeMetadata(parsedOutput),
    ...contents
  });
  return (0, import_smithy_client.decorateServiceException)(exception, parsedOutput.body);
}, "de_ServiceLimitExceededExceptionRes");
var de_ThrottlingExceptionRes = /* @__PURE__ */ __name(async (parsedOutput, context) => {
  const contents = (0, import_smithy_client.map)({
    [_RAS]: [() => void 0 !== parsedOutput.headers[_ra], () => (0, import_smithy_client.strictParseInt32)(parsedOutput.headers[_ra])]
  });
  const data = parsedOutput.body;
  const doc = (0, import_smithy_client.take)(data, {
    Message: import_smithy_client.expectString
  });
  Object.assign(contents, doc);
  const exception = new ThrottlingException({
    $metadata: deserializeMetadata(parsedOutput),
    ...contents
  });
  return (0, import_smithy_client.decorateServiceException)(exception, parsedOutput.body);
}, "de_ThrottlingExceptionRes");
var de_ValidationExceptionRes = /* @__PURE__ */ __name(async (parsedOutput, context) => {
  const contents = (0, import_smithy_client.map)({});
  const data = parsedOutput.body;
  const doc = (0, import_smithy_client.take)(data, {
    Fields: import_smithy_client._json,
    Message: import_smithy_client.expectString,
    Reason: import_smithy_client.expectString
  });
  Object.assign(contents, doc);
  const exception = new ValidationException({
    $metadata: deserializeMetadata(parsedOutput),
    ...contents
  });
  return (0, import_smithy_client.decorateServiceException)(exception, parsedOutput.body);
}, "de_ValidationExceptionRes");
var se_CustomLineItemChargeDetails = /* @__PURE__ */ __name((input, context) => {
  return (0, import_smithy_client.take)(input, {
    Flat: (_) => se_CustomLineItemFlatChargeDetails(_, context),
    LineItemFilters: import_smithy_client._json,
    Percentage: (_) => se_CustomLineItemPercentageChargeDetails(_, context),
    Type: []
  });
}, "se_CustomLineItemChargeDetails");
var se_CustomLineItemFlatChargeDetails = /* @__PURE__ */ __name((input, context) => {
  return (0, import_smithy_client.take)(input, {
    ChargeValue: import_smithy_client.serializeFloat
  });
}, "se_CustomLineItemFlatChargeDetails");
var se_CustomLineItemPercentageChargeDetails = /* @__PURE__ */ __name((input, context) => {
  return (0, import_smithy_client.take)(input, {
    AssociatedValues: import_smithy_client._json,
    PercentageValue: import_smithy_client.serializeFloat
  });
}, "se_CustomLineItemPercentageChargeDetails");
var se_UpdateCustomLineItemChargeDetails = /* @__PURE__ */ __name((input, context) => {
  return (0, import_smithy_client.take)(input, {
    Flat: (_) => se_UpdateCustomLineItemFlatChargeDetails(_, context),
    LineItemFilters: import_smithy_client._json,
    Percentage: (_) => se_UpdateCustomLineItemPercentageChargeDetails(_, context)
  });
}, "se_UpdateCustomLineItemChargeDetails");
var se_UpdateCustomLineItemFlatChargeDetails = /* @__PURE__ */ __name((input, context) => {
  return (0, import_smithy_client.take)(input, {
    ChargeValue: import_smithy_client.serializeFloat
  });
}, "se_UpdateCustomLineItemFlatChargeDetails");
var se_UpdateCustomLineItemPercentageChargeDetails = /* @__PURE__ */ __name((input, context) => {
  return (0, import_smithy_client.take)(input, {
    PercentageValue: import_smithy_client.serializeFloat
  });
}, "se_UpdateCustomLineItemPercentageChargeDetails");
var de_CustomLineItemList = /* @__PURE__ */ __name((output, context) => {
  const retVal = (output || []).filter((e) => e != null).map((entry) => {
    return de_CustomLineItemListElement(entry, context);
  });
  return retVal;
}, "de_CustomLineItemList");
var de_CustomLineItemListElement = /* @__PURE__ */ __name((output, context) => {
  return (0, import_smithy_client.take)(output, {
    AccountId: import_smithy_client.expectString,
    Arn: import_smithy_client.expectString,
    AssociationSize: import_smithy_client.expectLong,
    BillingGroupArn: import_smithy_client.expectString,
    ChargeDetails: (_) => de_ListCustomLineItemChargeDetails(_, context),
    CreationTime: import_smithy_client.expectLong,
    CurrencyCode: import_smithy_client.expectString,
    Description: import_smithy_client.expectString,
    LastModifiedTime: import_smithy_client.expectLong,
    Name: import_smithy_client.expectString,
    ProductCode: import_smithy_client.expectString
  });
}, "de_CustomLineItemListElement");
var de_CustomLineItemVersionList = /* @__PURE__ */ __name((output, context) => {
  const retVal = (output || []).filter((e) => e != null).map((entry) => {
    return de_CustomLineItemVersionListElement(entry, context);
  });
  return retVal;
}, "de_CustomLineItemVersionList");
var de_CustomLineItemVersionListElement = /* @__PURE__ */ __name((output, context) => {
  return (0, import_smithy_client.take)(output, {
    AccountId: import_smithy_client.expectString,
    Arn: import_smithy_client.expectString,
    AssociationSize: import_smithy_client.expectLong,
    BillingGroupArn: import_smithy_client.expectString,
    ChargeDetails: (_) => de_ListCustomLineItemChargeDetails(_, context),
    CreationTime: import_smithy_client.expectLong,
    CurrencyCode: import_smithy_client.expectString,
    Description: import_smithy_client.expectString,
    EndBillingPeriod: import_smithy_client.expectString,
    LastModifiedTime: import_smithy_client.expectLong,
    Name: import_smithy_client.expectString,
    ProductCode: import_smithy_client.expectString,
    StartBillingPeriod: import_smithy_client.expectString,
    StartTime: import_smithy_client.expectLong
  });
}, "de_CustomLineItemVersionListElement");
var de_ListCustomLineItemChargeDetails = /* @__PURE__ */ __name((output, context) => {
  return (0, import_smithy_client.take)(output, {
    Flat: (_) => de_ListCustomLineItemFlatChargeDetails(_, context),
    LineItemFilters: import_smithy_client._json,
    Percentage: (_) => de_ListCustomLineItemPercentageChargeDetails(_, context),
    Type: import_smithy_client.expectString
  });
}, "de_ListCustomLineItemChargeDetails");
var de_ListCustomLineItemFlatChargeDetails = /* @__PURE__ */ __name((output, context) => {
  return (0, import_smithy_client.take)(output, {
    ChargeValue: import_smithy_client.limitedParseDouble
  });
}, "de_ListCustomLineItemFlatChargeDetails");
var de_ListCustomLineItemPercentageChargeDetails = /* @__PURE__ */ __name((output, context) => {
  return (0, import_smithy_client.take)(output, {
    PercentageValue: import_smithy_client.limitedParseDouble
  });
}, "de_ListCustomLineItemPercentageChargeDetails");
var de_PricingRuleList = /* @__PURE__ */ __name((output, context) => {
  const retVal = (output || []).filter((e) => e != null).map((entry) => {
    return de_PricingRuleListElement(entry, context);
  });
  return retVal;
}, "de_PricingRuleList");
var de_PricingRuleListElement = /* @__PURE__ */ __name((output, context) => {
  return (0, import_smithy_client.take)(output, {
    Arn: import_smithy_client.expectString,
    AssociatedPricingPlanCount: import_smithy_client.expectLong,
    BillingEntity: import_smithy_client.expectString,
    CreationTime: import_smithy_client.expectLong,
    Description: import_smithy_client.expectString,
    LastModifiedTime: import_smithy_client.expectLong,
    ModifierPercentage: import_smithy_client.limitedParseDouble,
    Name: import_smithy_client.expectString,
    Operation: import_smithy_client.expectString,
    Scope: import_smithy_client.expectString,
    Service: import_smithy_client.expectString,
    Tiering: import_smithy_client._json,
    Type: import_smithy_client.expectString,
    UsageType: import_smithy_client.expectString
  });
}, "de_PricingRuleListElement");
var deserializeMetadata = /* @__PURE__ */ __name((output) => ({
  httpStatusCode: output.statusCode,
  requestId: output.headers["x-amzn-requestid"] ?? output.headers["x-amzn-request-id"] ?? output.headers["x-amz-request-id"],
  extendedRequestId: output.headers["x-amz-id-2"],
  cfId: output.headers["x-amz-cf-id"]
}), "deserializeMetadata");
var isSerializableHeaderValue = /* @__PURE__ */ __name((value) => value !== void 0 && value !== null && value !== "" && (!Object.getOwnPropertyNames(value).includes("length") || value.length != 0) && (!Object.getOwnPropertyNames(value).includes("size") || value.size != 0), "isSerializableHeaderValue");
var _CT = "ClientToken";
var _RAS = "RetryAfterSeconds";
var _TK = "TagKeys";
var _ra = "retry-after";
var _tK = "tagKeys";
var _xact = "x-amzn-client-token";

// src/commands/AssociateAccountsCommand.ts
var _AssociateAccountsCommand = class _AssociateAccountsCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AWSBillingConductor", "AssociateAccounts", {}).n("BillingconductorClient", "AssociateAccountsCommand").f(void 0, void 0).ser(se_AssociateAccountsCommand).de(de_AssociateAccountsCommand).build() {
};
__name(_AssociateAccountsCommand, "AssociateAccountsCommand");
var AssociateAccountsCommand = _AssociateAccountsCommand;

// src/commands/AssociatePricingRulesCommand.ts




var _AssociatePricingRulesCommand = class _AssociatePricingRulesCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AWSBillingConductor", "AssociatePricingRules", {}).n("BillingconductorClient", "AssociatePricingRulesCommand").f(void 0, void 0).ser(se_AssociatePricingRulesCommand).de(de_AssociatePricingRulesCommand).build() {
};
__name(_AssociatePricingRulesCommand, "AssociatePricingRulesCommand");
var AssociatePricingRulesCommand = _AssociatePricingRulesCommand;

// src/commands/BatchAssociateResourcesToCustomLineItemCommand.ts




var _BatchAssociateResourcesToCustomLineItemCommand = class _BatchAssociateResourcesToCustomLineItemCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AWSBillingConductor", "BatchAssociateResourcesToCustomLineItem", {}).n("BillingconductorClient", "BatchAssociateResourcesToCustomLineItemCommand").f(void 0, void 0).ser(se_BatchAssociateResourcesToCustomLineItemCommand).de(de_BatchAssociateResourcesToCustomLineItemCommand).build() {
};
__name(_BatchAssociateResourcesToCustomLineItemCommand, "BatchAssociateResourcesToCustomLineItemCommand");
var BatchAssociateResourcesToCustomLineItemCommand = _BatchAssociateResourcesToCustomLineItemCommand;

// src/commands/BatchDisassociateResourcesFromCustomLineItemCommand.ts




var _BatchDisassociateResourcesFromCustomLineItemCommand = class _BatchDisassociateResourcesFromCustomLineItemCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AWSBillingConductor", "BatchDisassociateResourcesFromCustomLineItem", {}).n("BillingconductorClient", "BatchDisassociateResourcesFromCustomLineItemCommand").f(void 0, void 0).ser(se_BatchDisassociateResourcesFromCustomLineItemCommand).de(de_BatchDisassociateResourcesFromCustomLineItemCommand).build() {
};
__name(_BatchDisassociateResourcesFromCustomLineItemCommand, "BatchDisassociateResourcesFromCustomLineItemCommand");
var BatchDisassociateResourcesFromCustomLineItemCommand = _BatchDisassociateResourcesFromCustomLineItemCommand;

// src/commands/CreateBillingGroupCommand.ts




var _CreateBillingGroupCommand = class _CreateBillingGroupCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AWSBillingConductor", "CreateBillingGroup", {}).n("BillingconductorClient", "CreateBillingGroupCommand").f(CreateBillingGroupInputFilterSensitiveLog, void 0).ser(se_CreateBillingGroupCommand).de(de_CreateBillingGroupCommand).build() {
};
__name(_CreateBillingGroupCommand, "CreateBillingGroupCommand");
var CreateBillingGroupCommand = _CreateBillingGroupCommand;

// src/commands/CreateCustomLineItemCommand.ts




var _CreateCustomLineItemCommand = class _CreateCustomLineItemCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AWSBillingConductor", "CreateCustomLineItem", {}).n("BillingconductorClient", "CreateCustomLineItemCommand").f(CreateCustomLineItemInputFilterSensitiveLog, void 0).ser(se_CreateCustomLineItemCommand).de(de_CreateCustomLineItemCommand).build() {
};
__name(_CreateCustomLineItemCommand, "CreateCustomLineItemCommand");
var CreateCustomLineItemCommand = _CreateCustomLineItemCommand;

// src/commands/CreatePricingPlanCommand.ts




var _CreatePricingPlanCommand = class _CreatePricingPlanCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AWSBillingConductor", "CreatePricingPlan", {}).n("BillingconductorClient", "CreatePricingPlanCommand").f(CreatePricingPlanInputFilterSensitiveLog, void 0).ser(se_CreatePricingPlanCommand).de(de_CreatePricingPlanCommand).build() {
};
__name(_CreatePricingPlanCommand, "CreatePricingPlanCommand");
var CreatePricingPlanCommand = _CreatePricingPlanCommand;

// src/commands/CreatePricingRuleCommand.ts




var _CreatePricingRuleCommand = class _CreatePricingRuleCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AWSBillingConductor", "CreatePricingRule", {}).n("BillingconductorClient", "CreatePricingRuleCommand").f(CreatePricingRuleInputFilterSensitiveLog, void 0).ser(se_CreatePricingRuleCommand).de(de_CreatePricingRuleCommand).build() {
};
__name(_CreatePricingRuleCommand, "CreatePricingRuleCommand");
var CreatePricingRuleCommand = _CreatePricingRuleCommand;

// src/commands/DeleteBillingGroupCommand.ts




var _DeleteBillingGroupCommand = class _DeleteBillingGroupCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AWSBillingConductor", "DeleteBillingGroup", {}).n("BillingconductorClient", "DeleteBillingGroupCommand").f(void 0, void 0).ser(se_DeleteBillingGroupCommand).de(de_DeleteBillingGroupCommand).build() {
};
__name(_DeleteBillingGroupCommand, "DeleteBillingGroupCommand");
var DeleteBillingGroupCommand = _DeleteBillingGroupCommand;

// src/commands/DeleteCustomLineItemCommand.ts




var _DeleteCustomLineItemCommand = class _DeleteCustomLineItemCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AWSBillingConductor", "DeleteCustomLineItem", {}).n("BillingconductorClient", "DeleteCustomLineItemCommand").f(void 0, void 0).ser(se_DeleteCustomLineItemCommand).de(de_DeleteCustomLineItemCommand).build() {
};
__name(_DeleteCustomLineItemCommand, "DeleteCustomLineItemCommand");
var DeleteCustomLineItemCommand = _DeleteCustomLineItemCommand;

// src/commands/DeletePricingPlanCommand.ts




var _DeletePricingPlanCommand = class _DeletePricingPlanCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AWSBillingConductor", "DeletePricingPlan", {}).n("BillingconductorClient", "DeletePricingPlanCommand").f(void 0, void 0).ser(se_DeletePricingPlanCommand).de(de_DeletePricingPlanCommand).build() {
};
__name(_DeletePricingPlanCommand, "DeletePricingPlanCommand");
var DeletePricingPlanCommand = _DeletePricingPlanCommand;

// src/commands/DeletePricingRuleCommand.ts




var _DeletePricingRuleCommand = class _DeletePricingRuleCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AWSBillingConductor", "DeletePricingRule", {}).n("BillingconductorClient", "DeletePricingRuleCommand").f(void 0, void 0).ser(se_DeletePricingRuleCommand).de(de_DeletePricingRuleCommand).build() {
};
__name(_DeletePricingRuleCommand, "DeletePricingRuleCommand");
var DeletePricingRuleCommand = _DeletePricingRuleCommand;

// src/commands/DisassociateAccountsCommand.ts




var _DisassociateAccountsCommand = class _DisassociateAccountsCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AWSBillingConductor", "DisassociateAccounts", {}).n("BillingconductorClient", "DisassociateAccountsCommand").f(void 0, void 0).ser(se_DisassociateAccountsCommand).de(de_DisassociateAccountsCommand).build() {
};
__name(_DisassociateAccountsCommand, "DisassociateAccountsCommand");
var DisassociateAccountsCommand = _DisassociateAccountsCommand;

// src/commands/DisassociatePricingRulesCommand.ts




var _DisassociatePricingRulesCommand = class _DisassociatePricingRulesCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AWSBillingConductor", "DisassociatePricingRules", {}).n("BillingconductorClient", "DisassociatePricingRulesCommand").f(void 0, void 0).ser(se_DisassociatePricingRulesCommand).de(de_DisassociatePricingRulesCommand).build() {
};
__name(_DisassociatePricingRulesCommand, "DisassociatePricingRulesCommand");
var DisassociatePricingRulesCommand = _DisassociatePricingRulesCommand;

// src/commands/GetBillingGroupCostReportCommand.ts




var _GetBillingGroupCostReportCommand = class _GetBillingGroupCostReportCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AWSBillingConductor", "GetBillingGroupCostReport", {}).n("BillingconductorClient", "GetBillingGroupCostReportCommand").f(void 0, void 0).ser(se_GetBillingGroupCostReportCommand).de(de_GetBillingGroupCostReportCommand).build() {
};
__name(_GetBillingGroupCostReportCommand, "GetBillingGroupCostReportCommand");
var GetBillingGroupCostReportCommand = _GetBillingGroupCostReportCommand;

// src/commands/ListAccountAssociationsCommand.ts




var _ListAccountAssociationsCommand = class _ListAccountAssociationsCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AWSBillingConductor", "ListAccountAssociations", {}).n("BillingconductorClient", "ListAccountAssociationsCommand").f(void 0, ListAccountAssociationsOutputFilterSensitiveLog).ser(se_ListAccountAssociationsCommand).de(de_ListAccountAssociationsCommand).build() {
};
__name(_ListAccountAssociationsCommand, "ListAccountAssociationsCommand");
var ListAccountAssociationsCommand = _ListAccountAssociationsCommand;

// src/commands/ListBillingGroupCostReportsCommand.ts




var _ListBillingGroupCostReportsCommand = class _ListBillingGroupCostReportsCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AWSBillingConductor", "ListBillingGroupCostReports", {}).n("BillingconductorClient", "ListBillingGroupCostReportsCommand").f(void 0, void 0).ser(se_ListBillingGroupCostReportsCommand).de(de_ListBillingGroupCostReportsCommand).build() {
};
__name(_ListBillingGroupCostReportsCommand, "ListBillingGroupCostReportsCommand");
var ListBillingGroupCostReportsCommand = _ListBillingGroupCostReportsCommand;

// src/commands/ListBillingGroupsCommand.ts




var _ListBillingGroupsCommand = class _ListBillingGroupsCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AWSBillingConductor", "ListBillingGroups", {}).n("BillingconductorClient", "ListBillingGroupsCommand").f(void 0, ListBillingGroupsOutputFilterSensitiveLog).ser(se_ListBillingGroupsCommand).de(de_ListBillingGroupsCommand).build() {
};
__name(_ListBillingGroupsCommand, "ListBillingGroupsCommand");
var ListBillingGroupsCommand = _ListBillingGroupsCommand;

// src/commands/ListCustomLineItemsCommand.ts




var _ListCustomLineItemsCommand = class _ListCustomLineItemsCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AWSBillingConductor", "ListCustomLineItems", {}).n("BillingconductorClient", "ListCustomLineItemsCommand").f(ListCustomLineItemsInputFilterSensitiveLog, ListCustomLineItemsOutputFilterSensitiveLog).ser(se_ListCustomLineItemsCommand).de(de_ListCustomLineItemsCommand).build() {
};
__name(_ListCustomLineItemsCommand, "ListCustomLineItemsCommand");
var ListCustomLineItemsCommand = _ListCustomLineItemsCommand;

// src/commands/ListCustomLineItemVersionsCommand.ts




var _ListCustomLineItemVersionsCommand = class _ListCustomLineItemVersionsCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AWSBillingConductor", "ListCustomLineItemVersions", {}).n("BillingconductorClient", "ListCustomLineItemVersionsCommand").f(void 0, ListCustomLineItemVersionsOutputFilterSensitiveLog).ser(se_ListCustomLineItemVersionsCommand).de(de_ListCustomLineItemVersionsCommand).build() {
};
__name(_ListCustomLineItemVersionsCommand, "ListCustomLineItemVersionsCommand");
var ListCustomLineItemVersionsCommand = _ListCustomLineItemVersionsCommand;

// src/commands/ListPricingPlansAssociatedWithPricingRuleCommand.ts




var _ListPricingPlansAssociatedWithPricingRuleCommand = class _ListPricingPlansAssociatedWithPricingRuleCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AWSBillingConductor", "ListPricingPlansAssociatedWithPricingRule", {}).n("BillingconductorClient", "ListPricingPlansAssociatedWithPricingRuleCommand").f(void 0, void 0).ser(se_ListPricingPlansAssociatedWithPricingRuleCommand).de(de_ListPricingPlansAssociatedWithPricingRuleCommand).build() {
};
__name(_ListPricingPlansAssociatedWithPricingRuleCommand, "ListPricingPlansAssociatedWithPricingRuleCommand");
var ListPricingPlansAssociatedWithPricingRuleCommand = _ListPricingPlansAssociatedWithPricingRuleCommand;

// src/commands/ListPricingPlansCommand.ts




var _ListPricingPlansCommand = class _ListPricingPlansCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AWSBillingConductor", "ListPricingPlans", {}).n("BillingconductorClient", "ListPricingPlansCommand").f(void 0, ListPricingPlansOutputFilterSensitiveLog).ser(se_ListPricingPlansCommand).de(de_ListPricingPlansCommand).build() {
};
__name(_ListPricingPlansCommand, "ListPricingPlansCommand");
var ListPricingPlansCommand = _ListPricingPlansCommand;

// src/commands/ListPricingRulesAssociatedToPricingPlanCommand.ts




var _ListPricingRulesAssociatedToPricingPlanCommand = class _ListPricingRulesAssociatedToPricingPlanCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AWSBillingConductor", "ListPricingRulesAssociatedToPricingPlan", {}).n("BillingconductorClient", "ListPricingRulesAssociatedToPricingPlanCommand").f(void 0, void 0).ser(se_ListPricingRulesAssociatedToPricingPlanCommand).de(de_ListPricingRulesAssociatedToPricingPlanCommand).build() {
};
__name(_ListPricingRulesAssociatedToPricingPlanCommand, "ListPricingRulesAssociatedToPricingPlanCommand");
var ListPricingRulesAssociatedToPricingPlanCommand = _ListPricingRulesAssociatedToPricingPlanCommand;

// src/commands/ListPricingRulesCommand.ts




var _ListPricingRulesCommand = class _ListPricingRulesCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AWSBillingConductor", "ListPricingRules", {}).n("BillingconductorClient", "ListPricingRulesCommand").f(void 0, ListPricingRulesOutputFilterSensitiveLog).ser(se_ListPricingRulesCommand).de(de_ListPricingRulesCommand).build() {
};
__name(_ListPricingRulesCommand, "ListPricingRulesCommand");
var ListPricingRulesCommand = _ListPricingRulesCommand;

// src/commands/ListResourcesAssociatedToCustomLineItemCommand.ts




var _ListResourcesAssociatedToCustomLineItemCommand = class _ListResourcesAssociatedToCustomLineItemCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AWSBillingConductor", "ListResourcesAssociatedToCustomLineItem", {}).n("BillingconductorClient", "ListResourcesAssociatedToCustomLineItemCommand").f(void 0, void 0).ser(se_ListResourcesAssociatedToCustomLineItemCommand).de(de_ListResourcesAssociatedToCustomLineItemCommand).build() {
};
__name(_ListResourcesAssociatedToCustomLineItemCommand, "ListResourcesAssociatedToCustomLineItemCommand");
var ListResourcesAssociatedToCustomLineItemCommand = _ListResourcesAssociatedToCustomLineItemCommand;

// src/commands/ListTagsForResourceCommand.ts




var _ListTagsForResourceCommand = class _ListTagsForResourceCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AWSBillingConductor", "ListTagsForResource", {}).n("BillingconductorClient", "ListTagsForResourceCommand").f(void 0, void 0).ser(se_ListTagsForResourceCommand).de(de_ListTagsForResourceCommand).build() {
};
__name(_ListTagsForResourceCommand, "ListTagsForResourceCommand");
var ListTagsForResourceCommand = _ListTagsForResourceCommand;

// src/commands/TagResourceCommand.ts




var _TagResourceCommand = class _TagResourceCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AWSBillingConductor", "TagResource", {}).n("BillingconductorClient", "TagResourceCommand").f(void 0, void 0).ser(se_TagResourceCommand).de(de_TagResourceCommand).build() {
};
__name(_TagResourceCommand, "TagResourceCommand");
var TagResourceCommand = _TagResourceCommand;

// src/commands/UntagResourceCommand.ts




var _UntagResourceCommand = class _UntagResourceCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AWSBillingConductor", "UntagResource", {}).n("BillingconductorClient", "UntagResourceCommand").f(void 0, void 0).ser(se_UntagResourceCommand).de(de_UntagResourceCommand).build() {
};
__name(_UntagResourceCommand, "UntagResourceCommand");
var UntagResourceCommand = _UntagResourceCommand;

// src/commands/UpdateBillingGroupCommand.ts




var _UpdateBillingGroupCommand = class _UpdateBillingGroupCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AWSBillingConductor", "UpdateBillingGroup", {}).n("BillingconductorClient", "UpdateBillingGroupCommand").f(UpdateBillingGroupInputFilterSensitiveLog, UpdateBillingGroupOutputFilterSensitiveLog).ser(se_UpdateBillingGroupCommand).de(de_UpdateBillingGroupCommand).build() {
};
__name(_UpdateBillingGroupCommand, "UpdateBillingGroupCommand");
var UpdateBillingGroupCommand = _UpdateBillingGroupCommand;

// src/commands/UpdateCustomLineItemCommand.ts




var _UpdateCustomLineItemCommand = class _UpdateCustomLineItemCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AWSBillingConductor", "UpdateCustomLineItem", {}).n("BillingconductorClient", "UpdateCustomLineItemCommand").f(UpdateCustomLineItemInputFilterSensitiveLog, UpdateCustomLineItemOutputFilterSensitiveLog).ser(se_UpdateCustomLineItemCommand).de(de_UpdateCustomLineItemCommand).build() {
};
__name(_UpdateCustomLineItemCommand, "UpdateCustomLineItemCommand");
var UpdateCustomLineItemCommand = _UpdateCustomLineItemCommand;

// src/commands/UpdatePricingPlanCommand.ts




var _UpdatePricingPlanCommand = class _UpdatePricingPlanCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AWSBillingConductor", "UpdatePricingPlan", {}).n("BillingconductorClient", "UpdatePricingPlanCommand").f(UpdatePricingPlanInputFilterSensitiveLog, UpdatePricingPlanOutputFilterSensitiveLog).ser(se_UpdatePricingPlanCommand).de(de_UpdatePricingPlanCommand).build() {
};
__name(_UpdatePricingPlanCommand, "UpdatePricingPlanCommand");
var UpdatePricingPlanCommand = _UpdatePricingPlanCommand;

// src/commands/UpdatePricingRuleCommand.ts




var _UpdatePricingRuleCommand = class _UpdatePricingRuleCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AWSBillingConductor", "UpdatePricingRule", {}).n("BillingconductorClient", "UpdatePricingRuleCommand").f(UpdatePricingRuleInputFilterSensitiveLog, UpdatePricingRuleOutputFilterSensitiveLog).ser(se_UpdatePricingRuleCommand).de(de_UpdatePricingRuleCommand).build() {
};
__name(_UpdatePricingRuleCommand, "UpdatePricingRuleCommand");
var UpdatePricingRuleCommand = _UpdatePricingRuleCommand;

// src/Billingconductor.ts
var commands = {
  AssociateAccountsCommand,
  AssociatePricingRulesCommand,
  BatchAssociateResourcesToCustomLineItemCommand,
  BatchDisassociateResourcesFromCustomLineItemCommand,
  CreateBillingGroupCommand,
  CreateCustomLineItemCommand,
  CreatePricingPlanCommand,
  CreatePricingRuleCommand,
  DeleteBillingGroupCommand,
  DeleteCustomLineItemCommand,
  DeletePricingPlanCommand,
  DeletePricingRuleCommand,
  DisassociateAccountsCommand,
  DisassociatePricingRulesCommand,
  GetBillingGroupCostReportCommand,
  ListAccountAssociationsCommand,
  ListBillingGroupCostReportsCommand,
  ListBillingGroupsCommand,
  ListCustomLineItemsCommand,
  ListCustomLineItemVersionsCommand,
  ListPricingPlansCommand,
  ListPricingPlansAssociatedWithPricingRuleCommand,
  ListPricingRulesCommand,
  ListPricingRulesAssociatedToPricingPlanCommand,
  ListResourcesAssociatedToCustomLineItemCommand,
  ListTagsForResourceCommand,
  TagResourceCommand,
  UntagResourceCommand,
  UpdateBillingGroupCommand,
  UpdateCustomLineItemCommand,
  UpdatePricingPlanCommand,
  UpdatePricingRuleCommand
};
var _Billingconductor = class _Billingconductor extends BillingconductorClient {
};
__name(_Billingconductor, "Billingconductor");
var Billingconductor = _Billingconductor;
(0, import_smithy_client.createAggregatedClient)(commands, Billingconductor);

// src/pagination/GetBillingGroupCostReportPaginator.ts

var paginateGetBillingGroupCostReport = (0, import_core.createPaginator)(BillingconductorClient, GetBillingGroupCostReportCommand, "NextToken", "NextToken", "MaxResults");

// src/pagination/ListAccountAssociationsPaginator.ts

var paginateListAccountAssociations = (0, import_core.createPaginator)(BillingconductorClient, ListAccountAssociationsCommand, "NextToken", "NextToken", "");

// src/pagination/ListBillingGroupCostReportsPaginator.ts

var paginateListBillingGroupCostReports = (0, import_core.createPaginator)(BillingconductorClient, ListBillingGroupCostReportsCommand, "NextToken", "NextToken", "MaxResults");

// src/pagination/ListBillingGroupsPaginator.ts

var paginateListBillingGroups = (0, import_core.createPaginator)(BillingconductorClient, ListBillingGroupsCommand, "NextToken", "NextToken", "MaxResults");

// src/pagination/ListCustomLineItemVersionsPaginator.ts

var paginateListCustomLineItemVersions = (0, import_core.createPaginator)(BillingconductorClient, ListCustomLineItemVersionsCommand, "NextToken", "NextToken", "MaxResults");

// src/pagination/ListCustomLineItemsPaginator.ts

var paginateListCustomLineItems = (0, import_core.createPaginator)(BillingconductorClient, ListCustomLineItemsCommand, "NextToken", "NextToken", "MaxResults");

// src/pagination/ListPricingPlansAssociatedWithPricingRulePaginator.ts

var paginateListPricingPlansAssociatedWithPricingRule = (0, import_core.createPaginator)(BillingconductorClient, ListPricingPlansAssociatedWithPricingRuleCommand, "NextToken", "NextToken", "MaxResults");

// src/pagination/ListPricingPlansPaginator.ts

var paginateListPricingPlans = (0, import_core.createPaginator)(BillingconductorClient, ListPricingPlansCommand, "NextToken", "NextToken", "MaxResults");

// src/pagination/ListPricingRulesAssociatedToPricingPlanPaginator.ts

var paginateListPricingRulesAssociatedToPricingPlan = (0, import_core.createPaginator)(BillingconductorClient, ListPricingRulesAssociatedToPricingPlanCommand, "NextToken", "NextToken", "MaxResults");

// src/pagination/ListPricingRulesPaginator.ts

var paginateListPricingRules = (0, import_core.createPaginator)(BillingconductorClient, ListPricingRulesCommand, "NextToken", "NextToken", "MaxResults");

// src/pagination/ListResourcesAssociatedToCustomLineItemPaginator.ts

var paginateListResourcesAssociatedToCustomLineItem = (0, import_core.createPaginator)(BillingconductorClient, ListResourcesAssociatedToCustomLineItemCommand, "NextToken", "NextToken", "MaxResults");
// Annotate the CommonJS export names for ESM import in node:

0 && (module.exports = {
  BillingconductorServiceException,
  __Client,
  BillingconductorClient,
  Billingconductor,
  $Command,
  AssociateAccountsCommand,
  AssociatePricingRulesCommand,
  BatchAssociateResourcesToCustomLineItemCommand,
  BatchDisassociateResourcesFromCustomLineItemCommand,
  CreateBillingGroupCommand,
  CreateCustomLineItemCommand,
  CreatePricingPlanCommand,
  CreatePricingRuleCommand,
  DeleteBillingGroupCommand,
  DeleteCustomLineItemCommand,
  DeletePricingPlanCommand,
  DeletePricingRuleCommand,
  DisassociateAccountsCommand,
  DisassociatePricingRulesCommand,
  GetBillingGroupCostReportCommand,
  ListAccountAssociationsCommand,
  ListBillingGroupCostReportsCommand,
  ListBillingGroupsCommand,
  ListCustomLineItemVersionsCommand,
  ListCustomLineItemsCommand,
  ListPricingPlansAssociatedWithPricingRuleCommand,
  ListPricingPlansCommand,
  ListPricingRulesAssociatedToPricingPlanCommand,
  ListPricingRulesCommand,
  ListResourcesAssociatedToCustomLineItemCommand,
  ListTagsForResourceCommand,
  TagResourceCommand,
  UntagResourceCommand,
  UpdateBillingGroupCommand,
  UpdateCustomLineItemCommand,
  UpdatePricingPlanCommand,
  UpdatePricingRuleCommand,
  paginateGetBillingGroupCostReport,
  paginateListAccountAssociations,
  paginateListBillingGroupCostReports,
  paginateListBillingGroups,
  paginateListCustomLineItemVersions,
  paginateListCustomLineItems,
  paginateListPricingPlansAssociatedWithPricingRule,
  paginateListPricingPlans,
  paginateListPricingRulesAssociatedToPricingPlan,
  paginateListPricingRules,
  paginateListResourcesAssociatedToCustomLineItem,
  AccessDeniedException,
  ConflictExceptionReason,
  ConflictException,
  InternalServerException,
  ResourceNotFoundException,
  ServiceLimitExceededException,
  ThrottlingException,
  ValidationExceptionReason,
  ValidationException,
  AssociateResourceErrorReason,
  BillingGroupStatus,
  LineItemFilterAttributeName,
  MatchOption,
  LineItemFilterValue,
  CustomLineItemType,
  CurrencyCode,
  CustomLineItemRelationship,
  GroupByAttributeName,
  PricingRuleScope,
  PricingRuleType,
  AccountAssociationsListElementFilterSensitiveLog,
  CreateBillingGroupInputFilterSensitiveLog,
  BillingGroupListElementFilterSensitiveLog,
  ListBillingGroupsOutputFilterSensitiveLog,
  UpdateBillingGroupInputFilterSensitiveLog,
  UpdateBillingGroupOutputFilterSensitiveLog,
  CreateCustomLineItemInputFilterSensitiveLog,
  ListCustomLineItemsFilterFilterSensitiveLog,
  ListCustomLineItemsInputFilterSensitiveLog,
  CustomLineItemListElementFilterSensitiveLog,
  ListCustomLineItemsOutputFilterSensitiveLog,
  CustomLineItemVersionListElementFilterSensitiveLog,
  ListCustomLineItemVersionsOutputFilterSensitiveLog,
  UpdateCustomLineItemInputFilterSensitiveLog,
  UpdateCustomLineItemOutputFilterSensitiveLog,
  ListAccountAssociationsOutputFilterSensitiveLog,
  CreatePricingPlanInputFilterSensitiveLog,
  PricingPlanListElementFilterSensitiveLog,
  ListPricingPlansOutputFilterSensitiveLog,
  UpdatePricingPlanInputFilterSensitiveLog,
  UpdatePricingPlanOutputFilterSensitiveLog,
  CreatePricingRuleInputFilterSensitiveLog,
  PricingRuleListElementFilterSensitiveLog,
  ListPricingRulesOutputFilterSensitiveLog,
  UpdatePricingRuleInputFilterSensitiveLog,
  UpdatePricingRuleOutputFilterSensitiveLog
});
