// @azure/arm-trafficmanager/dist-esm/src/models/index.js
/*
 * Copyright (c) Microsoft Corporation.
 * Licensed under the MIT License.
 *
 * Code generated by Microsoft (R) AutoRest Code Generator.
 * Changes may cause incorrect behavior and will be lost if the code is regenerated.
 */
/** Known values of {@link EndpointStatus} that the service accepts. */
export var KnownEndpointStatus;
(function (KnownEndpointStatus) {
    /** Enabled */
    KnownEndpointStatus["Enabled"] = "Enabled";
    /** Disabled */
    KnownEndpointStatus["Disabled"] = "Disabled";
})(KnownEndpointStatus || (KnownEndpointStatus = {}));
/** Known values of {@link EndpointMonitorStatus} that the service accepts. */
export var KnownEndpointMonitorStatus;
(function (KnownEndpointMonitorStatus) {
    /** CheckingEndpoint */
    KnownEndpointMonitorStatus["CheckingEndpoint"] = "CheckingEndpoint";
    /** Online */
    KnownEndpointMonitorStatus["Online"] = "Online";
    /** Degraded */
    KnownEndpointMonitorStatus["Degraded"] = "Degraded";
    /** Disabled */
    KnownEndpointMonitorStatus["Disabled"] = "Disabled";
    /** Inactive */
    KnownEndpointMonitorStatus["Inactive"] = "Inactive";
    /** Stopped */
    KnownEndpointMonitorStatus["Stopped"] = "Stopped";
    /** Unmonitored */
    KnownEndpointMonitorStatus["Unmonitored"] = "Unmonitored";
})(KnownEndpointMonitorStatus || (KnownEndpointMonitorStatus = {}));
/** Known values of {@link AlwaysServe} that the service accepts. */
export var KnownAlwaysServe;
(function (KnownAlwaysServe) {
    /** Enabled */
    KnownAlwaysServe["Enabled"] = "Enabled";
    /** Disabled */
    KnownAlwaysServe["Disabled"] = "Disabled";
})(KnownAlwaysServe || (KnownAlwaysServe = {}));
/** Known values of {@link ProfileStatus} that the service accepts. */
export var KnownProfileStatus;
(function (KnownProfileStatus) {
    /** Enabled */
    KnownProfileStatus["Enabled"] = "Enabled";
    /** Disabled */
    KnownProfileStatus["Disabled"] = "Disabled";
})(KnownProfileStatus || (KnownProfileStatus = {}));
/** Known values of {@link TrafficRoutingMethod} that the service accepts. */
export var KnownTrafficRoutingMethod;
(function (KnownTrafficRoutingMethod) {
    /** Performance */
    KnownTrafficRoutingMethod["Performance"] = "Performance";
    /** Priority */
    KnownTrafficRoutingMethod["Priority"] = "Priority";
    /** Weighted */
    KnownTrafficRoutingMethod["Weighted"] = "Weighted";
    /** Geographic */
    KnownTrafficRoutingMethod["Geographic"] = "Geographic";
    /** MultiValue */
    KnownTrafficRoutingMethod["MultiValue"] = "MultiValue";
    /** Subnet */
    KnownTrafficRoutingMethod["Subnet"] = "Subnet";
})(KnownTrafficRoutingMethod || (KnownTrafficRoutingMethod = {}));
/** Known values of {@link ProfileMonitorStatus} that the service accepts. */
export var KnownProfileMonitorStatus;
(function (KnownProfileMonitorStatus) {
    /** CheckingEndpoints */
    KnownProfileMonitorStatus["CheckingEndpoints"] = "CheckingEndpoints";
    /** Online */
    KnownProfileMonitorStatus["Online"] = "Online";
    /** Degraded */
    KnownProfileMonitorStatus["Degraded"] = "Degraded";
    /** Disabled */
    KnownProfileMonitorStatus["Disabled"] = "Disabled";
    /** Inactive */
    KnownProfileMonitorStatus["Inactive"] = "Inactive";
})(KnownProfileMonitorStatus || (KnownProfileMonitorStatus = {}));
/** Known values of {@link MonitorProtocol} that the service accepts. */
export var KnownMonitorProtocol;
(function (KnownMonitorProtocol) {
    /** Http */
    KnownMonitorProtocol["Http"] = "HTTP";
    /** Https */
    KnownMonitorProtocol["Https"] = "HTTPS";
    /** TCP */
    KnownMonitorProtocol["TCP"] = "TCP";
})(KnownMonitorProtocol || (KnownMonitorProtocol = {}));
/** Known values of {@link TrafficViewEnrollmentStatus} that the service accepts. */
export var KnownTrafficViewEnrollmentStatus;
(function (KnownTrafficViewEnrollmentStatus) {
    /** Enabled */
    KnownTrafficViewEnrollmentStatus["Enabled"] = "Enabled";
    /** Disabled */
    KnownTrafficViewEnrollmentStatus["Disabled"] = "Disabled";
})(KnownTrafficViewEnrollmentStatus || (KnownTrafficViewEnrollmentStatus = {}));
/** Known values of {@link AllowedEndpointRecordType} that the service accepts. */
export var KnownAllowedEndpointRecordType;
(function (KnownAllowedEndpointRecordType) {
    /** DomainName */
    KnownAllowedEndpointRecordType["DomainName"] = "DomainName";
    /** IPv4Address */
    KnownAllowedEndpointRecordType["IPv4Address"] = "IPv4Address";
    /** IPv6Address */
    KnownAllowedEndpointRecordType["IPv6Address"] = "IPv6Address";
    /** Any */
    KnownAllowedEndpointRecordType["Any"] = "Any";
})(KnownAllowedEndpointRecordType || (KnownAllowedEndpointRecordType = {}));
