// @azure/arm-datadog/dist-esm/src/models/index.js
/*
 * Copyright (c) Microsoft Corporation.
 * Licensed under the MIT License.
 *
 * Code generated by Microsoft (R) AutoRest Code Generator.
 * Changes may cause incorrect behavior and will be lost if the code is regenerated.
 */
/** Known values of {@link CreatedByType} that the service accepts. */
export var KnownCreatedByType;
(function (KnownCreatedByType) {
    /** User */
    KnownCreatedByType["User"] = "User";
    /** Application */
    KnownCreatedByType["Application"] = "Application";
    /** ManagedIdentity */
    KnownCreatedByType["ManagedIdentity"] = "ManagedIdentity";
    /** Key */
    KnownCreatedByType["Key"] = "Key";
})(KnownCreatedByType || (KnownCreatedByType = {}));
/** Known values of {@link ProvisioningState} that the service accepts. */
export var KnownProvisioningState;
(function (KnownProvisioningState) {
    /** Accepted */
    KnownProvisioningState["Accepted"] = "Accepted";
    /** Creating */
    KnownProvisioningState["Creating"] = "Creating";
    /** Updating */
    KnownProvisioningState["Updating"] = "Updating";
    /** Deleting */
    KnownProvisioningState["Deleting"] = "Deleting";
    /** Succeeded */
    KnownProvisioningState["Succeeded"] = "Succeeded";
    /** Failed */
    KnownProvisioningState["Failed"] = "Failed";
    /** Canceled */
    KnownProvisioningState["Canceled"] = "Canceled";
    /** Deleted */
    KnownProvisioningState["Deleted"] = "Deleted";
    /** NotSpecified */
    KnownProvisioningState["NotSpecified"] = "NotSpecified";
})(KnownProvisioningState || (KnownProvisioningState = {}));
/** Known values of {@link MonitoringStatus} that the service accepts. */
export var KnownMonitoringStatus;
(function (KnownMonitoringStatus) {
    /** Enabled */
    KnownMonitoringStatus["Enabled"] = "Enabled";
    /** Disabled */
    KnownMonitoringStatus["Disabled"] = "Disabled";
})(KnownMonitoringStatus || (KnownMonitoringStatus = {}));
/** Known values of {@link MarketplaceSubscriptionStatus} that the service accepts. */
export var KnownMarketplaceSubscriptionStatus;
(function (KnownMarketplaceSubscriptionStatus) {
    /** Provisioning */
    KnownMarketplaceSubscriptionStatus["Provisioning"] = "Provisioning";
    /** Active */
    KnownMarketplaceSubscriptionStatus["Active"] = "Active";
    /** Suspended */
    KnownMarketplaceSubscriptionStatus["Suspended"] = "Suspended";
    /** Unsubscribed */
    KnownMarketplaceSubscriptionStatus["Unsubscribed"] = "Unsubscribed";
})(KnownMarketplaceSubscriptionStatus || (KnownMarketplaceSubscriptionStatus = {}));
/** Known values of {@link LiftrResourceCategories} that the service accepts. */
export var KnownLiftrResourceCategories;
(function (KnownLiftrResourceCategories) {
    /** Unknown */
    KnownLiftrResourceCategories["Unknown"] = "Unknown";
    /** MonitorLogs */
    KnownLiftrResourceCategories["MonitorLogs"] = "MonitorLogs";
})(KnownLiftrResourceCategories || (KnownLiftrResourceCategories = {}));
/** Known values of {@link ManagedIdentityTypes} that the service accepts. */
export var KnownManagedIdentityTypes;
(function (KnownManagedIdentityTypes) {
    /** SystemAssigned */
    KnownManagedIdentityTypes["SystemAssigned"] = "SystemAssigned";
    /** UserAssigned */
    KnownManagedIdentityTypes["UserAssigned"] = "UserAssigned";
})(KnownManagedIdentityTypes || (KnownManagedIdentityTypes = {}));
/** Known values of {@link TagAction} that the service accepts. */
export var KnownTagAction;
(function (KnownTagAction) {
    /** Include */
    KnownTagAction["Include"] = "Include";
    /** Exclude */
    KnownTagAction["Exclude"] = "Exclude";
})(KnownTagAction || (KnownTagAction = {}));
/** Known values of {@link SingleSignOnStates} that the service accepts. */
export var KnownSingleSignOnStates;
(function (KnownSingleSignOnStates) {
    /** Initial */
    KnownSingleSignOnStates["Initial"] = "Initial";
    /** Enable */
    KnownSingleSignOnStates["Enable"] = "Enable";
    /** Disable */
    KnownSingleSignOnStates["Disable"] = "Disable";
    /** Existing */
    KnownSingleSignOnStates["Existing"] = "Existing";
})(KnownSingleSignOnStates || (KnownSingleSignOnStates = {}));
/** Known values of {@link Operation} that the service accepts. */
export var KnownOperation;
(function (KnownOperation) {
    /** AddBegin */
    KnownOperation["AddBegin"] = "AddBegin";
    /** AddComplete */
    KnownOperation["AddComplete"] = "AddComplete";
    /** DeleteBegin */
    KnownOperation["DeleteBegin"] = "DeleteBegin";
    /** DeleteComplete */
    KnownOperation["DeleteComplete"] = "DeleteComplete";
    /** Active */
    KnownOperation["Active"] = "Active";
})(KnownOperation || (KnownOperation = {}));
/** Known values of {@link Status} that the service accepts. */
export var KnownStatus;
(function (KnownStatus) {
    /** InProgress */
    KnownStatus["InProgress"] = "InProgress";
    /** Active */
    KnownStatus["Active"] = "Active";
    /** Failed */
    KnownStatus["Failed"] = "Failed";
    /** Deleting */
    KnownStatus["Deleting"] = "Deleting";
})(KnownStatus || (KnownStatus = {}));
