// @azure/arm-iothub/dist-esm/src/models/index.js
/*
 * Copyright (c) Microsoft Corporation.
 * Licensed under the MIT License.
 *
 * Code generated by Microsoft (R) AutoRest Code Generator.
 * Changes may cause incorrect behavior and will be lost if the code is regenerated.
 */
/** Known values of {@link PublicNetworkAccess} that the service accepts. */
export var KnownPublicNetworkAccess;
(function (KnownPublicNetworkAccess) {
    /** Enabled */
    KnownPublicNetworkAccess["Enabled"] = "Enabled";
    /** Disabled */
    KnownPublicNetworkAccess["Disabled"] = "Disabled";
})(KnownPublicNetworkAccess || (KnownPublicNetworkAccess = {}));
/** Known values of {@link DefaultAction} that the service accepts. */
export var KnownDefaultAction;
(function (KnownDefaultAction) {
    /** Deny */
    KnownDefaultAction["Deny"] = "Deny";
    /** Allow */
    KnownDefaultAction["Allow"] = "Allow";
})(KnownDefaultAction || (KnownDefaultAction = {}));
/** Known values of {@link NetworkRuleIPAction} that the service accepts. */
export var KnownNetworkRuleIPAction;
(function (KnownNetworkRuleIPAction) {
    /** Allow */
    KnownNetworkRuleIPAction["Allow"] = "Allow";
})(KnownNetworkRuleIPAction || (KnownNetworkRuleIPAction = {}));
/** Known values of {@link PrivateLinkServiceConnectionStatus} that the service accepts. */
export var KnownPrivateLinkServiceConnectionStatus;
(function (KnownPrivateLinkServiceConnectionStatus) {
    /** Pending */
    KnownPrivateLinkServiceConnectionStatus["Pending"] = "Pending";
    /** Approved */
    KnownPrivateLinkServiceConnectionStatus["Approved"] = "Approved";
    /** Rejected */
    KnownPrivateLinkServiceConnectionStatus["Rejected"] = "Rejected";
    /** Disconnected */
    KnownPrivateLinkServiceConnectionStatus["Disconnected"] = "Disconnected";
})(KnownPrivateLinkServiceConnectionStatus || (KnownPrivateLinkServiceConnectionStatus = {}));
/** Known values of {@link AuthenticationType} that the service accepts. */
export var KnownAuthenticationType;
(function (KnownAuthenticationType) {
    /** KeyBased */
    KnownAuthenticationType["KeyBased"] = "keyBased";
    /** IdentityBased */
    KnownAuthenticationType["IdentityBased"] = "identityBased";
})(KnownAuthenticationType || (KnownAuthenticationType = {}));
/** Known values of {@link RoutingStorageContainerPropertiesEncoding} that the service accepts. */
export var KnownRoutingStorageContainerPropertiesEncoding;
(function (KnownRoutingStorageContainerPropertiesEncoding) {
    /** Avro */
    KnownRoutingStorageContainerPropertiesEncoding["Avro"] = "Avro";
    /** AvroDeflate */
    KnownRoutingStorageContainerPropertiesEncoding["AvroDeflate"] = "AvroDeflate";
    /** Json */
    KnownRoutingStorageContainerPropertiesEncoding["Json"] = "JSON";
})(KnownRoutingStorageContainerPropertiesEncoding || (KnownRoutingStorageContainerPropertiesEncoding = {}));
/** Known values of {@link RoutingSource} that the service accepts. */
export var KnownRoutingSource;
(function (KnownRoutingSource) {
    /** Invalid */
    KnownRoutingSource["Invalid"] = "Invalid";
    /** DeviceMessages */
    KnownRoutingSource["DeviceMessages"] = "DeviceMessages";
    /** TwinChangeEvents */
    KnownRoutingSource["TwinChangeEvents"] = "TwinChangeEvents";
    /** DeviceLifecycleEvents */
    KnownRoutingSource["DeviceLifecycleEvents"] = "DeviceLifecycleEvents";
    /** DeviceJobLifecycleEvents */
    KnownRoutingSource["DeviceJobLifecycleEvents"] = "DeviceJobLifecycleEvents";
    /** DeviceConnectionStateEvents */
    KnownRoutingSource["DeviceConnectionStateEvents"] = "DeviceConnectionStateEvents";
})(KnownRoutingSource || (KnownRoutingSource = {}));
/** Known values of {@link Capabilities} that the service accepts. */
export var KnownCapabilities;
(function (KnownCapabilities) {
    /** None */
    KnownCapabilities["None"] = "None";
    /** DeviceManagement */
    KnownCapabilities["DeviceManagement"] = "DeviceManagement";
})(KnownCapabilities || (KnownCapabilities = {}));
/** Known values of {@link IotHubReplicaRoleType} that the service accepts. */
export var KnownIotHubReplicaRoleType;
(function (KnownIotHubReplicaRoleType) {
    /** Primary */
    KnownIotHubReplicaRoleType["Primary"] = "primary";
    /** Secondary */
    KnownIotHubReplicaRoleType["Secondary"] = "secondary";
})(KnownIotHubReplicaRoleType || (KnownIotHubReplicaRoleType = {}));
/** Known values of {@link IotHubSku} that the service accepts. */
export var KnownIotHubSku;
(function (KnownIotHubSku) {
    /** F1 */
    KnownIotHubSku["F1"] = "F1";
    /** S1 */
    KnownIotHubSku["S1"] = "S1";
    /** S2 */
    KnownIotHubSku["S2"] = "S2";
    /** S3 */
    KnownIotHubSku["S3"] = "S3";
    /** B1 */
    KnownIotHubSku["B1"] = "B1";
    /** B2 */
    KnownIotHubSku["B2"] = "B2";
    /** B3 */
    KnownIotHubSku["B3"] = "B3";
})(KnownIotHubSku || (KnownIotHubSku = {}));
/** Known values of {@link CreatedByType} that the service accepts. */
export var KnownCreatedByType;
(function (KnownCreatedByType) {
    /** User */
    KnownCreatedByType["User"] = "User";
    /** Application */
    KnownCreatedByType["Application"] = "Application";
    /** ManagedIdentity */
    KnownCreatedByType["ManagedIdentity"] = "ManagedIdentity";
    /** Key */
    KnownCreatedByType["Key"] = "Key";
})(KnownCreatedByType || (KnownCreatedByType = {}));
/** Known values of {@link JobType} that the service accepts. */
export var KnownJobType;
(function (KnownJobType) {
    /** Unknown */
    KnownJobType["Unknown"] = "unknown";
    /** Export */
    KnownJobType["Export"] = "export";
    /** Import */
    KnownJobType["Import"] = "import";
    /** Backup */
    KnownJobType["Backup"] = "backup";
    /** ReadDeviceProperties */
    KnownJobType["ReadDeviceProperties"] = "readDeviceProperties";
    /** WriteDeviceProperties */
    KnownJobType["WriteDeviceProperties"] = "writeDeviceProperties";
    /** UpdateDeviceConfiguration */
    KnownJobType["UpdateDeviceConfiguration"] = "updateDeviceConfiguration";
    /** RebootDevice */
    KnownJobType["RebootDevice"] = "rebootDevice";
    /** FactoryResetDevice */
    KnownJobType["FactoryResetDevice"] = "factoryResetDevice";
    /** FirmwareUpdate */
    KnownJobType["FirmwareUpdate"] = "firmwareUpdate";
})(KnownJobType || (KnownJobType = {}));
/** Known values of {@link EndpointHealthStatus} that the service accepts. */
export var KnownEndpointHealthStatus;
(function (KnownEndpointHealthStatus) {
    /** Unknown */
    KnownEndpointHealthStatus["Unknown"] = "unknown";
    /** Healthy */
    KnownEndpointHealthStatus["Healthy"] = "healthy";
    /** Degraded */
    KnownEndpointHealthStatus["Degraded"] = "degraded";
    /** Unhealthy */
    KnownEndpointHealthStatus["Unhealthy"] = "unhealthy";
    /** Dead */
    KnownEndpointHealthStatus["Dead"] = "dead";
})(KnownEndpointHealthStatus || (KnownEndpointHealthStatus = {}));
/** Known values of {@link TestResultStatus} that the service accepts. */
export var KnownTestResultStatus;
(function (KnownTestResultStatus) {
    /** Undefined */
    KnownTestResultStatus["Undefined"] = "undefined";
    /** False */
    KnownTestResultStatus["False"] = "false";
    /** True */
    KnownTestResultStatus["True"] = "true";
})(KnownTestResultStatus || (KnownTestResultStatus = {}));
/** Known values of {@link RouteErrorSeverity} that the service accepts. */
export var KnownRouteErrorSeverity;
(function (KnownRouteErrorSeverity) {
    /** Error */
    KnownRouteErrorSeverity["Error"] = "error";
    /** Warning */
    KnownRouteErrorSeverity["Warning"] = "warning";
})(KnownRouteErrorSeverity || (KnownRouteErrorSeverity = {}));
