// @azure/arm-reservations/dist-esm/src/models/parameters.js
/*
 * Copyright (c) Microsoft Corporation.
 * Licensed under the MIT License.
 *
 * Code generated by Microsoft (R) AutoRest Code Generator.
 * Changes may cause incorrect behavior and will be lost if the code is regenerated.
 */
import { AvailableScopeRequest as AvailableScopeRequestMapper, SplitRequest as SplitRequestMapper, MergeRequest as MergeRequestMapper, Patch as PatchMapper, PurchaseRequest as PurchaseRequestMapper, ChangeDirectoryRequest as ChangeDirectoryRequestMapper, CalculateRefundRequest as CalculateRefundRequestMapper, RefundRequest as RefundRequestMapper, CalculateExchangeRequest as CalculateExchangeRequestMapper, ExchangeRequest as ExchangeRequestMapper, CurrentQuotaLimitBase as CurrentQuotaLimitBaseMapper } from "../models/mappers";
export const contentType = {
    parameterPath: ["options", "contentType"],
    mapper: {
        defaultValue: "application/json",
        isConstant: true,
        serializedName: "Content-Type",
        type: {
            name: "String"
        }
    }
};
export const body = {
    parameterPath: "body",
    mapper: AvailableScopeRequestMapper
};
export const accept = {
    parameterPath: "accept",
    mapper: {
        defaultValue: "application/json",
        isConstant: true,
        serializedName: "Accept",
        type: {
            name: "String"
        }
    }
};
export const $host = {
    parameterPath: "$host",
    mapper: {
        serializedName: "$host",
        required: true,
        type: {
            name: "String"
        }
    },
    skipEncoding: true
};
export const reservationOrderId = {
    parameterPath: "reservationOrderId",
    mapper: {
        serializedName: "reservationOrderId",
        required: true,
        type: {
            name: "String"
        }
    }
};
export const reservationId = {
    parameterPath: "reservationId",
    mapper: {
        serializedName: "reservationId",
        required: true,
        type: {
            name: "String"
        }
    }
};
export const apiVersion = {
    parameterPath: "apiVersion",
    mapper: {
        defaultValue: "2022-11-01",
        isConstant: true,
        serializedName: "api-version",
        type: {
            name: "String"
        }
    }
};
export const body1 = {
    parameterPath: "body",
    mapper: SplitRequestMapper
};
export const body2 = {
    parameterPath: "body",
    mapper: MergeRequestMapper
};
export const expand = {
    parameterPath: ["options", "expand"],
    mapper: {
        serializedName: "$expand",
        type: {
            name: "String"
        }
    }
};
export const parameters = {
    parameterPath: "parameters",
    mapper: PatchMapper
};
export const filter = {
    parameterPath: ["options", "filter"],
    mapper: {
        serializedName: "$filter",
        type: {
            name: "String"
        }
    }
};
export const orderby = {
    parameterPath: ["options", "orderby"],
    mapper: {
        serializedName: "$orderby",
        type: {
            name: "String"
        }
    }
};
export const refreshSummary = {
    parameterPath: ["options", "refreshSummary"],
    mapper: {
        serializedName: "refreshSummary",
        type: {
            name: "String"
        }
    }
};
export const skiptoken = {
    parameterPath: ["options", "skiptoken"],
    mapper: {
        serializedName: "$skiptoken",
        type: {
            name: "Number"
        }
    }
};
export const selectedState = {
    parameterPath: ["options", "selectedState"],
    mapper: {
        serializedName: "selectedState",
        type: {
            name: "String"
        }
    }
};
export const take = {
    parameterPath: ["options", "take"],
    mapper: {
        serializedName: "take",
        type: {
            name: "Number"
        }
    }
};
export const nextLink = {
    parameterPath: "nextLink",
    mapper: {
        serializedName: "nextLink",
        required: true,
        type: {
            name: "String"
        }
    },
    skipEncoding: true
};
export const subscriptionId = {
    parameterPath: "subscriptionId",
    mapper: {
        serializedName: "subscriptionId",
        required: true,
        type: {
            name: "String"
        }
    }
};
export const reservedResourceType = {
    parameterPath: ["options", "reservedResourceType"],
    mapper: {
        serializedName: "reservedResourceType",
        type: {
            name: "String"
        }
    }
};
export const location = {
    parameterPath: ["options", "location"],
    mapper: {
        serializedName: "location",
        type: {
            name: "String"
        }
    }
};
export const publisherId = {
    parameterPath: ["options", "publisherId"],
    mapper: {
        serializedName: "publisherId",
        type: {
            name: "String"
        }
    }
};
export const offerId = {
    parameterPath: ["options", "offerId"],
    mapper: {
        serializedName: "offerId",
        type: {
            name: "String"
        }
    }
};
export const planId = {
    parameterPath: ["options", "planId"],
    mapper: {
        serializedName: "planId",
        type: {
            name: "String"
        }
    }
};
export const skip = {
    parameterPath: ["options", "skip"],
    mapper: {
        serializedName: "$skip",
        type: {
            name: "Number"
        }
    }
};
export const take1 = {
    parameterPath: ["options", "take"],
    mapper: {
        serializedName: "$take",
        type: {
            name: "Number"
        }
    }
};
export const body3 = {
    parameterPath: "body",
    mapper: PurchaseRequestMapper
};
export const body4 = {
    parameterPath: "body",
    mapper: ChangeDirectoryRequestMapper
};
export const body5 = {
    parameterPath: "body",
    mapper: CalculateRefundRequestMapper
};
export const body6 = {
    parameterPath: "body",
    mapper: RefundRequestMapper
};
export const body7 = {
    parameterPath: "body",
    mapper: CalculateExchangeRequestMapper
};
export const body8 = {
    parameterPath: "body",
    mapper: ExchangeRequestMapper
};
export const providerId = {
    parameterPath: "providerId",
    mapper: {
        serializedName: "providerId",
        required: true,
        type: {
            name: "String"
        }
    }
};
export const location1 = {
    parameterPath: "location",
    mapper: {
        serializedName: "location",
        required: true,
        type: {
            name: "String"
        }
    }
};
export const apiVersion1 = {
    parameterPath: "apiVersion",
    mapper: {
        defaultValue: "2020-10-25",
        isConstant: true,
        serializedName: "api-version",
        type: {
            name: "String"
        }
    }
};
export const resourceName = {
    parameterPath: "resourceName",
    mapper: {
        serializedName: "resourceName",
        required: true,
        type: {
            name: "String"
        }
    }
};
export const createQuotaRequest = {
    parameterPath: "createQuotaRequest",
    mapper: CurrentQuotaLimitBaseMapper
};
export const id = {
    parameterPath: "id",
    mapper: {
        serializedName: "id",
        required: true,
        type: {
            name: "String"
        }
    }
};
export const top = {
    parameterPath: ["options", "top"],
    mapper: {
        constraints: {
            InclusiveMinimum: 1
        },
        serializedName: "$top",
        type: {
            name: "Number"
        }
    }
};
export const skiptoken1 = {
    parameterPath: ["options", "skiptoken"],
    mapper: {
        serializedName: "$skiptoken",
        type: {
            name: "String"
        }
    }
};
