// @azure/arm-orbital/dist-esm/src/operations/availableGroundStations.js
/*
 * Copyright (c) Microsoft Corporation.
 * Licensed under the MIT License.
 *
 * Code generated by Microsoft (R) AutoRest Code Generator.
 * Changes may cause incorrect behavior and will be lost if the code is regenerated.
 */
import { __asyncDelegator, __asyncGenerator, __asyncValues, __await } from "tslib";
import { setContinuationToken } from "../pagingHelper";
import * as coreClient from "@azure/core-client";
import * as Mappers from "../models/mappers";
import * as Parameters from "../models/parameters";
/// <reference lib="esnext.asynciterable" />
/** Class containing AvailableGroundStations operations. */
export class AvailableGroundStationsImpl {
    /**
     * Initialize a new instance of the class AvailableGroundStations class.
     * @param client Reference to the service client
     */
    constructor(client) {
        this.client = client;
    }
    /**
     * Returns list of available ground stations.
     * @param capability Ground Station Capability.
     * @param options The options parameters.
     */
    listByCapability(capability, options) {
        const iter = this.listByCapabilityPagingAll(capability, options);
        return {
            next() {
                return iter.next();
            },
            [Symbol.asyncIterator]() {
                return this;
            },
            byPage: (settings) => {
                if (settings === null || settings === void 0 ? void 0 : settings.maxPageSize) {
                    throw new Error("maxPageSize is not supported by this operation.");
                }
                return this.listByCapabilityPagingPage(capability, options, settings);
            }
        };
    }
    listByCapabilityPagingPage(capability, options, settings) {
        return __asyncGenerator(this, arguments, function* listByCapabilityPagingPage_1() {
            let result;
            let continuationToken = settings === null || settings === void 0 ? void 0 : settings.continuationToken;
            if (!continuationToken) {
                result = yield __await(this._listByCapability(capability, options));
                let page = result.value || [];
                continuationToken = result.nextLink;
                setContinuationToken(page, continuationToken);
                yield yield __await(page);
            }
            while (continuationToken) {
                result = yield __await(this._listByCapabilityNext(continuationToken, options));
                continuationToken = result.nextLink;
                let page = result.value || [];
                setContinuationToken(page, continuationToken);
                yield yield __await(page);
            }
        });
    }
    listByCapabilityPagingAll(capability, options) {
        return __asyncGenerator(this, arguments, function* listByCapabilityPagingAll_1() {
            var e_1, _a;
            try {
                for (var _b = __asyncValues(this.listByCapabilityPagingPage(capability, options)), _c; _c = yield __await(_b.next()), !_c.done;) {
                    const page = _c.value;
                    yield __await(yield* __asyncDelegator(__asyncValues(page)));
                }
            }
            catch (e_1_1) { e_1 = { error: e_1_1 }; }
            finally {
                try {
                    if (_c && !_c.done && (_a = _b.return)) yield __await(_a.call(_b));
                }
                finally { if (e_1) throw e_1.error; }
            }
        });
    }
    /**
     * Returns list of available ground stations.
     * @param capability Ground Station Capability.
     * @param options The options parameters.
     */
    _listByCapability(capability, options) {
        return this.client.sendOperationRequest({ capability, options }, listByCapabilityOperationSpec);
    }
    /**
     * ListByCapabilityNext
     * @param nextLink The nextLink from the previous successful call to the ListByCapability method.
     * @param options The options parameters.
     */
    _listByCapabilityNext(nextLink, options) {
        return this.client.sendOperationRequest({ nextLink, options }, listByCapabilityNextOperationSpec);
    }
}
// Operation Specifications
const serializer = coreClient.createSerializer(Mappers, /* isXml */ false);
const listByCapabilityOperationSpec = {
    path: "/subscriptions/{subscriptionId}/providers/Microsoft.Orbital/availableGroundStations",
    httpMethod: "GET",
    responses: {
        200: {
            bodyMapper: Mappers.AvailableGroundStationListResult
        },
        default: {
            bodyMapper: Mappers.ErrorResponse
        }
    },
    queryParameters: [Parameters.apiVersion, Parameters.capability],
    urlParameters: [Parameters.$host, Parameters.subscriptionId],
    headerParameters: [Parameters.accept],
    serializer
};
const listByCapabilityNextOperationSpec = {
    path: "{nextLink}",
    httpMethod: "GET",
    responses: {
        200: {
            bodyMapper: Mappers.AvailableGroundStationListResult
        },
        default: {
            bodyMapper: Mappers.ErrorResponse
        }
    },
    urlParameters: [
        Parameters.$host,
        Parameters.subscriptionId,
        Parameters.nextLink
    ],
    headerParameters: [Parameters.accept],
    serializer
};
