// @azure/arm-billing/dist-esm/src/models/index.js
/*
 * Copyright (c) Microsoft Corporation.
 * Licensed under the MIT License.
 *
 * Code generated by Microsoft (R) AutoRest Code Generator.
 * Changes may cause incorrect behavior and will be lost if the code is regenerated.
 */
/** Known values of {@link AgreementType} that the service accepts. */
export var KnownAgreementType;
(function (KnownAgreementType) {
    /** MicrosoftCustomerAgreement */
    KnownAgreementType["MicrosoftCustomerAgreement"] = "MicrosoftCustomerAgreement";
    /** EnterpriseAgreement */
    KnownAgreementType["EnterpriseAgreement"] = "EnterpriseAgreement";
    /** MicrosoftOnlineServicesProgram */
    KnownAgreementType["MicrosoftOnlineServicesProgram"] = "MicrosoftOnlineServicesProgram";
    /** MicrosoftPartnerAgreement */
    KnownAgreementType["MicrosoftPartnerAgreement"] = "MicrosoftPartnerAgreement";
})(KnownAgreementType || (KnownAgreementType = {}));
/** Known values of {@link AccountType} that the service accepts. */
export var KnownAccountType;
(function (KnownAccountType) {
    /** Enterprise */
    KnownAccountType["Enterprise"] = "Enterprise";
    /** Individual */
    KnownAccountType["Individual"] = "Individual";
    /** Partner */
    KnownAccountType["Partner"] = "Partner";
})(KnownAccountType || (KnownAccountType = {}));
/** Known values of {@link AccountStatus} that the service accepts. */
export var KnownAccountStatus;
(function (KnownAccountStatus) {
    /** Active */
    KnownAccountStatus["Active"] = "Active";
    /** Deleted */
    KnownAccountStatus["Deleted"] = "Deleted";
    /** Disabled */
    KnownAccountStatus["Disabled"] = "Disabled";
    /** Expired */
    KnownAccountStatus["Expired"] = "Expired";
    /** Transferred */
    KnownAccountStatus["Transferred"] = "Transferred";
    /** Extended */
    KnownAccountStatus["Extended"] = "Extended";
    /** Terminated */
    KnownAccountStatus["Terminated"] = "Terminated";
})(KnownAccountStatus || (KnownAccountStatus = {}));
/** Known values of {@link BillingRelationshipType} that the service accepts. */
export var KnownBillingRelationshipType;
(function (KnownBillingRelationshipType) {
    /** Direct */
    KnownBillingRelationshipType["Direct"] = "Direct";
    /** IndirectCustomer */
    KnownBillingRelationshipType["IndirectCustomer"] = "IndirectCustomer";
    /** IndirectPartner */
    KnownBillingRelationshipType["IndirectPartner"] = "IndirectPartner";
    /** CSPPartner */
    KnownBillingRelationshipType["CSPPartner"] = "CSPPartner";
})(KnownBillingRelationshipType || (KnownBillingRelationshipType = {}));
/** Known values of {@link InvoiceSectionState} that the service accepts. */
export var KnownInvoiceSectionState;
(function (KnownInvoiceSectionState) {
    /** Active */
    KnownInvoiceSectionState["Active"] = "Active";
    /** Restricted */
    KnownInvoiceSectionState["Restricted"] = "Restricted";
})(KnownInvoiceSectionState || (KnownInvoiceSectionState = {}));
/** Known values of {@link TargetCloud} that the service accepts. */
export var KnownTargetCloud;
(function (KnownTargetCloud) {
    /** USGov */
    KnownTargetCloud["USGov"] = "USGov";
    /** USNat */
    KnownTargetCloud["USNat"] = "USNat";
    /** USSec */
    KnownTargetCloud["USSec"] = "USSec";
})(KnownTargetCloud || (KnownTargetCloud = {}));
/** Known values of {@link BillingProfileStatus} that the service accepts. */
export var KnownBillingProfileStatus;
(function (KnownBillingProfileStatus) {
    /** Active */
    KnownBillingProfileStatus["Active"] = "Active";
    /** Disabled */
    KnownBillingProfileStatus["Disabled"] = "Disabled";
    /** Warned */
    KnownBillingProfileStatus["Warned"] = "Warned";
})(KnownBillingProfileStatus || (KnownBillingProfileStatus = {}));
/** Known values of {@link StatusReasonCode} that the service accepts. */
export var KnownStatusReasonCode;
(function (KnownStatusReasonCode) {
    /** PastDue */
    KnownStatusReasonCode["PastDue"] = "PastDue";
    /** SpendingLimitReached */
    KnownStatusReasonCode["SpendingLimitReached"] = "SpendingLimitReached";
    /** SpendingLimitExpired */
    KnownStatusReasonCode["SpendingLimitExpired"] = "SpendingLimitExpired";
})(KnownStatusReasonCode || (KnownStatusReasonCode = {}));
/** Known values of {@link SpendingLimit} that the service accepts. */
export var KnownSpendingLimit;
(function (KnownSpendingLimit) {
    /** Off */
    KnownSpendingLimit["Off"] = "Off";
    /** On */
    KnownSpendingLimit["On"] = "On";
})(KnownSpendingLimit || (KnownSpendingLimit = {}));
/** Known values of {@link StatusReasonCodeForBillingProfile} that the service accepts. */
export var KnownStatusReasonCodeForBillingProfile;
(function (KnownStatusReasonCodeForBillingProfile) {
    /** PastDue */
    KnownStatusReasonCodeForBillingProfile["PastDue"] = "PastDue";
    /** SpendingLimitReached */
    KnownStatusReasonCodeForBillingProfile["SpendingLimitReached"] = "SpendingLimitReached";
    /** SpendingLimitExpired */
    KnownStatusReasonCodeForBillingProfile["SpendingLimitExpired"] = "SpendingLimitExpired";
})(KnownStatusReasonCodeForBillingProfile || (KnownStatusReasonCodeForBillingProfile = {}));
/** Known values of {@link SpendingLimitForBillingProfile} that the service accepts. */
export var KnownSpendingLimitForBillingProfile;
(function (KnownSpendingLimitForBillingProfile) {
    /** Off */
    KnownSpendingLimitForBillingProfile["Off"] = "Off";
    /** On */
    KnownSpendingLimitForBillingProfile["On"] = "On";
})(KnownSpendingLimitForBillingProfile || (KnownSpendingLimitForBillingProfile = {}));
/** Known values of {@link AddressValidationStatus} that the service accepts. */
export var KnownAddressValidationStatus;
(function (KnownAddressValidationStatus) {
    /** Valid */
    KnownAddressValidationStatus["Valid"] = "Valid";
    /** Invalid */
    KnownAddressValidationStatus["Invalid"] = "Invalid";
})(KnownAddressValidationStatus || (KnownAddressValidationStatus = {}));
/** Known values of {@link BillingSubscriptionStatusType} that the service accepts. */
export var KnownBillingSubscriptionStatusType;
(function (KnownBillingSubscriptionStatusType) {
    /** Active */
    KnownBillingSubscriptionStatusType["Active"] = "Active";
    /** Inactive */
    KnownBillingSubscriptionStatusType["Inactive"] = "Inactive";
    /** Abandoned */
    KnownBillingSubscriptionStatusType["Abandoned"] = "Abandoned";
    /** Deleted */
    KnownBillingSubscriptionStatusType["Deleted"] = "Deleted";
    /** Warning */
    KnownBillingSubscriptionStatusType["Warning"] = "Warning";
})(KnownBillingSubscriptionStatusType || (KnownBillingSubscriptionStatusType = {}));
/** Known values of {@link AutoRenew} that the service accepts. */
export var KnownAutoRenew;
(function (KnownAutoRenew) {
    /** Off */
    KnownAutoRenew["Off"] = "Off";
    /** On */
    KnownAutoRenew["On"] = "On";
})(KnownAutoRenew || (KnownAutoRenew = {}));
/** Known values of {@link ProductStatusType} that the service accepts. */
export var KnownProductStatusType;
(function (KnownProductStatusType) {
    /** Active */
    KnownProductStatusType["Active"] = "Active";
    /** Inactive */
    KnownProductStatusType["Inactive"] = "Inactive";
    /** PastDue */
    KnownProductStatusType["PastDue"] = "PastDue";
    /** Expiring */
    KnownProductStatusType["Expiring"] = "Expiring";
    /** Expired */
    KnownProductStatusType["Expired"] = "Expired";
    /** Disabled */
    KnownProductStatusType["Disabled"] = "Disabled";
    /** Cancelled */
    KnownProductStatusType["Cancelled"] = "Cancelled";
    /** AutoRenew */
    KnownProductStatusType["AutoRenew"] = "AutoRenew";
})(KnownProductStatusType || (KnownProductStatusType = {}));
/** Known values of {@link BillingFrequency} that the service accepts. */
export var KnownBillingFrequency;
(function (KnownBillingFrequency) {
    /** OneTime */
    KnownBillingFrequency["OneTime"] = "OneTime";
    /** Monthly */
    KnownBillingFrequency["Monthly"] = "Monthly";
    /** UsageBased */
    KnownBillingFrequency["UsageBased"] = "UsageBased";
})(KnownBillingFrequency || (KnownBillingFrequency = {}));
/** Known values of {@link InvoiceStatus} that the service accepts. */
export var KnownInvoiceStatus;
(function (KnownInvoiceStatus) {
    /** Due */
    KnownInvoiceStatus["Due"] = "Due";
    /** OverDue */
    KnownInvoiceStatus["OverDue"] = "OverDue";
    /** Paid */
    KnownInvoiceStatus["Paid"] = "Paid";
    /** Void */
    KnownInvoiceStatus["Void"] = "Void";
})(KnownInvoiceStatus || (KnownInvoiceStatus = {}));
/** Known values of {@link InvoiceType} that the service accepts. */
export var KnownInvoiceType;
(function (KnownInvoiceType) {
    /** AzureService */
    KnownInvoiceType["AzureService"] = "AzureService";
    /** AzureMarketplace */
    KnownInvoiceType["AzureMarketplace"] = "AzureMarketplace";
    /** AzureSupport */
    KnownInvoiceType["AzureSupport"] = "AzureSupport";
})(KnownInvoiceType || (KnownInvoiceType = {}));
/** Known values of {@link DocumentType} that the service accepts. */
export var KnownDocumentType;
(function (KnownDocumentType) {
    /** Invoice */
    KnownDocumentType["Invoice"] = "Invoice";
    /** VoidNote */
    KnownDocumentType["VoidNote"] = "VoidNote";
    /** TaxReceipt */
    KnownDocumentType["TaxReceipt"] = "TaxReceipt";
    /** CreditNote */
    KnownDocumentType["CreditNote"] = "CreditNote";
})(KnownDocumentType || (KnownDocumentType = {}));
/** Known values of {@link DocumentSource} that the service accepts. */
export var KnownDocumentSource;
(function (KnownDocumentSource) {
    /** DRS */
    KnownDocumentSource["DRS"] = "DRS";
    /** ENF */
    KnownDocumentSource["ENF"] = "ENF";
})(KnownDocumentSource || (KnownDocumentSource = {}));
/** Known values of {@link PaymentMethodFamily} that the service accepts. */
export var KnownPaymentMethodFamily;
(function (KnownPaymentMethodFamily) {
    /** Credits */
    KnownPaymentMethodFamily["Credits"] = "Credits";
    /** CheckWire */
    KnownPaymentMethodFamily["CheckWire"] = "CheckWire";
    /** CreditCard */
    KnownPaymentMethodFamily["CreditCard"] = "CreditCard";
    /** None */
    KnownPaymentMethodFamily["None"] = "None";
})(KnownPaymentMethodFamily || (KnownPaymentMethodFamily = {}));
/** Known values of {@link InvoiceDocumentType} that the service accepts. */
export var KnownInvoiceDocumentType;
(function (KnownInvoiceDocumentType) {
    /** Invoice */
    KnownInvoiceDocumentType["Invoice"] = "Invoice";
    /** CreditNote */
    KnownInvoiceDocumentType["CreditNote"] = "CreditNote";
})(KnownInvoiceDocumentType || (KnownInvoiceDocumentType = {}));
/** Known values of {@link SubscriptionTransferValidationErrorCode} that the service accepts. */
export var KnownSubscriptionTransferValidationErrorCode;
(function (KnownSubscriptionTransferValidationErrorCode) {
    /** BillingAccountInactive */
    KnownSubscriptionTransferValidationErrorCode["BillingAccountInactive"] = "BillingAccountInactive";
    /** CrossBillingAccountNotAllowed */
    KnownSubscriptionTransferValidationErrorCode["CrossBillingAccountNotAllowed"] = "CrossBillingAccountNotAllowed";
    /** DestinationBillingProfileInactive */
    KnownSubscriptionTransferValidationErrorCode["DestinationBillingProfileInactive"] = "DestinationBillingProfileInactive";
    /** DestinationBillingProfileNotFound */
    KnownSubscriptionTransferValidationErrorCode["DestinationBillingProfileNotFound"] = "DestinationBillingProfileNotFound";
    /** DestinationBillingProfilePastDue */
    KnownSubscriptionTransferValidationErrorCode["DestinationBillingProfilePastDue"] = "DestinationBillingProfilePastDue";
    /** DestinationInvoiceSectionInactive */
    KnownSubscriptionTransferValidationErrorCode["DestinationInvoiceSectionInactive"] = "DestinationInvoiceSectionInactive";
    /** DestinationInvoiceSectionNotFound */
    KnownSubscriptionTransferValidationErrorCode["DestinationInvoiceSectionNotFound"] = "DestinationInvoiceSectionNotFound";
    /** InsufficientPermissionOnDestination */
    KnownSubscriptionTransferValidationErrorCode["InsufficientPermissionOnDestination"] = "InsufficientPermissionOnDestination";
    /** InsufficientPermissionOnSource */
    KnownSubscriptionTransferValidationErrorCode["InsufficientPermissionOnSource"] = "InsufficientPermissionOnSource";
    /** InvalidDestination */
    KnownSubscriptionTransferValidationErrorCode["InvalidDestination"] = "InvalidDestination";
    /** InvalidSource */
    KnownSubscriptionTransferValidationErrorCode["InvalidSource"] = "InvalidSource";
    /** MarketplaceNotEnabledOnDestination */
    KnownSubscriptionTransferValidationErrorCode["MarketplaceNotEnabledOnDestination"] = "MarketplaceNotEnabledOnDestination";
    /** NotAvailableForDestinationMarket */
    KnownSubscriptionTransferValidationErrorCode["NotAvailableForDestinationMarket"] = "NotAvailableForDestinationMarket";
    /** ProductInactive */
    KnownSubscriptionTransferValidationErrorCode["ProductInactive"] = "ProductInactive";
    /** ProductNotFound */
    KnownSubscriptionTransferValidationErrorCode["ProductNotFound"] = "ProductNotFound";
    /** ProductTypeNotSupported */
    KnownSubscriptionTransferValidationErrorCode["ProductTypeNotSupported"] = "ProductTypeNotSupported";
    /** SourceBillingProfilePastDue */
    KnownSubscriptionTransferValidationErrorCode["SourceBillingProfilePastDue"] = "SourceBillingProfilePastDue";
    /** SourceInvoiceSectionInactive */
    KnownSubscriptionTransferValidationErrorCode["SourceInvoiceSectionInactive"] = "SourceInvoiceSectionInactive";
    /** SubscriptionNotActive */
    KnownSubscriptionTransferValidationErrorCode["SubscriptionNotActive"] = "SubscriptionNotActive";
    /** SubscriptionTypeNotSupported */
    KnownSubscriptionTransferValidationErrorCode["SubscriptionTypeNotSupported"] = "SubscriptionTypeNotSupported";
})(KnownSubscriptionTransferValidationErrorCode || (KnownSubscriptionTransferValidationErrorCode = {}));
/** Known values of {@link ProductTransferValidationErrorCode} that the service accepts. */
export var KnownProductTransferValidationErrorCode;
(function (KnownProductTransferValidationErrorCode) {
    /** InvalidSource */
    KnownProductTransferValidationErrorCode["InvalidSource"] = "InvalidSource";
    /** ProductNotActive */
    KnownProductTransferValidationErrorCode["ProductNotActive"] = "ProductNotActive";
    /** InsufficientPermissionOnSource */
    KnownProductTransferValidationErrorCode["InsufficientPermissionOnSource"] = "InsufficientPermissionOnSource";
    /** InsufficientPermissionOnDestination */
    KnownProductTransferValidationErrorCode["InsufficientPermissionOnDestination"] = "InsufficientPermissionOnDestination";
    /** DestinationBillingProfilePastDue */
    KnownProductTransferValidationErrorCode["DestinationBillingProfilePastDue"] = "DestinationBillingProfilePastDue";
    /** ProductTypeNotSupported */
    KnownProductTransferValidationErrorCode["ProductTypeNotSupported"] = "ProductTypeNotSupported";
    /** CrossBillingAccountNotAllowed */
    KnownProductTransferValidationErrorCode["CrossBillingAccountNotAllowed"] = "CrossBillingAccountNotAllowed";
    /** NotAvailableForDestinationMarket */
    KnownProductTransferValidationErrorCode["NotAvailableForDestinationMarket"] = "NotAvailableForDestinationMarket";
    /** OneTimePurchaseProductTransferNotAllowed */
    KnownProductTransferValidationErrorCode["OneTimePurchaseProductTransferNotAllowed"] = "OneTimePurchaseProductTransferNotAllowed";
})(KnownProductTransferValidationErrorCode || (KnownProductTransferValidationErrorCode = {}));
/** Known values of {@link TransactionTypeKind} that the service accepts. */
export var KnownTransactionTypeKind;
(function (KnownTransactionTypeKind) {
    /** All */
    KnownTransactionTypeKind["All"] = "all";
    /** Reservation */
    KnownTransactionTypeKind["Reservation"] = "reservation";
})(KnownTransactionTypeKind || (KnownTransactionTypeKind = {}));
/** Known values of {@link ReservationType} that the service accepts. */
export var KnownReservationType;
(function (KnownReservationType) {
    /** Purchase */
    KnownReservationType["Purchase"] = "Purchase";
    /** UsageCharge */
    KnownReservationType["UsageCharge"] = "Usage Charge";
})(KnownReservationType || (KnownReservationType = {}));
/** Known values of {@link MarketplacePurchasesPolicy} that the service accepts. */
export var KnownMarketplacePurchasesPolicy;
(function (KnownMarketplacePurchasesPolicy) {
    /** AllAllowed */
    KnownMarketplacePurchasesPolicy["AllAllowed"] = "AllAllowed";
    /** OnlyFreeAllowed */
    KnownMarketplacePurchasesPolicy["OnlyFreeAllowed"] = "OnlyFreeAllowed";
    /** NotAllowed */
    KnownMarketplacePurchasesPolicy["NotAllowed"] = "NotAllowed";
})(KnownMarketplacePurchasesPolicy || (KnownMarketplacePurchasesPolicy = {}));
/** Known values of {@link ReservationPurchasesPolicy} that the service accepts. */
export var KnownReservationPurchasesPolicy;
(function (KnownReservationPurchasesPolicy) {
    /** Allowed */
    KnownReservationPurchasesPolicy["Allowed"] = "Allowed";
    /** NotAllowed */
    KnownReservationPurchasesPolicy["NotAllowed"] = "NotAllowed";
})(KnownReservationPurchasesPolicy || (KnownReservationPurchasesPolicy = {}));
/** Known values of {@link ViewChargesPolicy} that the service accepts. */
export var KnownViewChargesPolicy;
(function (KnownViewChargesPolicy) {
    /** Allowed */
    KnownViewChargesPolicy["Allowed"] = "Allowed";
    /** NotAllowed */
    KnownViewChargesPolicy["NotAllowed"] = "NotAllowed";
})(KnownViewChargesPolicy || (KnownViewChargesPolicy = {}));
/** Known values of {@link ViewCharges} that the service accepts. */
export var KnownViewCharges;
(function (KnownViewCharges) {
    /** Allowed */
    KnownViewCharges["Allowed"] = "Allowed";
    /** NotAllowed */
    KnownViewCharges["NotAllowed"] = "NotAllowed";
})(KnownViewCharges || (KnownViewCharges = {}));
/** Known values of {@link BillingProfileStatusReasonCode} that the service accepts. */
export var KnownBillingProfileStatusReasonCode;
(function (KnownBillingProfileStatusReasonCode) {
    /** PastDue */
    KnownBillingProfileStatusReasonCode["PastDue"] = "PastDue";
    /** SpendingLimitReached */
    KnownBillingProfileStatusReasonCode["SpendingLimitReached"] = "SpendingLimitReached";
    /** SpendingLimitExpired */
    KnownBillingProfileStatusReasonCode["SpendingLimitExpired"] = "SpendingLimitExpired";
})(KnownBillingProfileStatusReasonCode || (KnownBillingProfileStatusReasonCode = {}));
/** Known values of {@link BillingProfileSpendingLimit} that the service accepts. */
export var KnownBillingProfileSpendingLimit;
(function (KnownBillingProfileSpendingLimit) {
    /** Off */
    KnownBillingProfileSpendingLimit["Off"] = "Off";
    /** On */
    KnownBillingProfileSpendingLimit["On"] = "On";
})(KnownBillingProfileSpendingLimit || (KnownBillingProfileSpendingLimit = {}));
/** Known values of {@link Category} that the service accepts. */
export var KnownCategory;
(function (KnownCategory) {
    /** MicrosoftCustomerAgreement */
    KnownCategory["MicrosoftCustomerAgreement"] = "MicrosoftCustomerAgreement";
    /** AffiliatePurchaseTerms */
    KnownCategory["AffiliatePurchaseTerms"] = "AffiliatePurchaseTerms";
    /** Other */
    KnownCategory["Other"] = "Other";
})(KnownCategory || (KnownCategory = {}));
/** Known values of {@link AcceptanceMode} that the service accepts. */
export var KnownAcceptanceMode;
(function (KnownAcceptanceMode) {
    /** ClickToAccept */
    KnownAcceptanceMode["ClickToAccept"] = "ClickToAccept";
    /** ESignEmbedded */
    KnownAcceptanceMode["ESignEmbedded"] = "ESignEmbedded";
    /** ESignOffline */
    KnownAcceptanceMode["ESignOffline"] = "ESignOffline";
})(KnownAcceptanceMode || (KnownAcceptanceMode = {}));
