// @azure/arm-orbital/dist-esm/src/models/index.js
/*
 * Copyright (c) Microsoft Corporation.
 * Licensed under the MIT License.
 *
 * Code generated by Microsoft (R) AutoRest Code Generator.
 * Changes may cause incorrect behavior and will be lost if the code is regenerated.
 */
/** Known values of {@link Origin} that the service accepts. */
export var KnownOrigin;
(function (KnownOrigin) {
    /** User */
    KnownOrigin["User"] = "user";
    /** System */
    KnownOrigin["System"] = "system";
    /** UserSystem */
    KnownOrigin["UserSystem"] = "user,system";
})(KnownOrigin || (KnownOrigin = {}));
/** Known values of {@link ActionType} that the service accepts. */
export var KnownActionType;
(function (KnownActionType) {
    /** Internal */
    KnownActionType["Internal"] = "Internal";
})(KnownActionType || (KnownActionType = {}));
/** Known values of {@link ProvisioningState} that the service accepts. */
export var KnownProvisioningState;
(function (KnownProvisioningState) {
    /** Creating */
    KnownProvisioningState["Creating"] = "creating";
    /** Succeeded */
    KnownProvisioningState["Succeeded"] = "succeeded";
    /** Failed */
    KnownProvisioningState["Failed"] = "failed";
    /** Canceled */
    KnownProvisioningState["Canceled"] = "canceled";
    /** Updating */
    KnownProvisioningState["Updating"] = "updating";
    /** Deleting */
    KnownProvisioningState["Deleting"] = "deleting";
})(KnownProvisioningState || (KnownProvisioningState = {}));
/** Known values of {@link SpacecraftsPropertiesProvisioningState} that the service accepts. */
export var KnownSpacecraftsPropertiesProvisioningState;
(function (KnownSpacecraftsPropertiesProvisioningState) {
    /** Creating */
    KnownSpacecraftsPropertiesProvisioningState["Creating"] = "creating";
    /** Succeeded */
    KnownSpacecraftsPropertiesProvisioningState["Succeeded"] = "succeeded";
    /** Failed */
    KnownSpacecraftsPropertiesProvisioningState["Failed"] = "failed";
    /** Canceled */
    KnownSpacecraftsPropertiesProvisioningState["Canceled"] = "canceled";
    /** Updating */
    KnownSpacecraftsPropertiesProvisioningState["Updating"] = "updating";
    /** Deleting */
    KnownSpacecraftsPropertiesProvisioningState["Deleting"] = "deleting";
})(KnownSpacecraftsPropertiesProvisioningState || (KnownSpacecraftsPropertiesProvisioningState = {}));
/** Known values of {@link Direction} that the service accepts. */
export var KnownDirection;
(function (KnownDirection) {
    /** Uplink */
    KnownDirection["Uplink"] = "Uplink";
    /** Downlink */
    KnownDirection["Downlink"] = "Downlink";
})(KnownDirection || (KnownDirection = {}));
/** Known values of {@link Polarization} that the service accepts. */
export var KnownPolarization;
(function (KnownPolarization) {
    /** Rhcp */
    KnownPolarization["Rhcp"] = "RHCP";
    /** Lhcp */
    KnownPolarization["Lhcp"] = "LHCP";
    /** LinearVertical */
    KnownPolarization["LinearVertical"] = "linearVertical";
    /** LinearHorizontal */
    KnownPolarization["LinearHorizontal"] = "linearHorizontal";
})(KnownPolarization || (KnownPolarization = {}));
/** Known values of {@link CreatedByType} that the service accepts. */
export var KnownCreatedByType;
(function (KnownCreatedByType) {
    /** User */
    KnownCreatedByType["User"] = "User";
    /** Application */
    KnownCreatedByType["Application"] = "Application";
    /** ManagedIdentity */
    KnownCreatedByType["ManagedIdentity"] = "ManagedIdentity";
    /** Key */
    KnownCreatedByType["Key"] = "Key";
})(KnownCreatedByType || (KnownCreatedByType = {}));
/** Known values of {@link ContactsPropertiesProvisioningState} that the service accepts. */
export var KnownContactsPropertiesProvisioningState;
(function (KnownContactsPropertiesProvisioningState) {
    /** Creating */
    KnownContactsPropertiesProvisioningState["Creating"] = "creating";
    /** Succeeded */
    KnownContactsPropertiesProvisioningState["Succeeded"] = "succeeded";
    /** Failed */
    KnownContactsPropertiesProvisioningState["Failed"] = "failed";
    /** Canceled */
    KnownContactsPropertiesProvisioningState["Canceled"] = "canceled";
    /** Updating */
    KnownContactsPropertiesProvisioningState["Updating"] = "updating";
    /** Deleting */
    KnownContactsPropertiesProvisioningState["Deleting"] = "deleting";
})(KnownContactsPropertiesProvisioningState || (KnownContactsPropertiesProvisioningState = {}));
/** Known values of {@link ContactsStatus} that the service accepts. */
export var KnownContactsStatus;
(function (KnownContactsStatus) {
    /** Scheduled */
    KnownContactsStatus["Scheduled"] = "scheduled";
    /** Cancelled */
    KnownContactsStatus["Cancelled"] = "cancelled";
    /** Succeeded */
    KnownContactsStatus["Succeeded"] = "succeeded";
    /** Failed */
    KnownContactsStatus["Failed"] = "failed";
    /** ProviderCancelled */
    KnownContactsStatus["ProviderCancelled"] = "providerCancelled";
})(KnownContactsStatus || (KnownContactsStatus = {}));
/** Known values of {@link ContactProfilesPropertiesProvisioningState} that the service accepts. */
export var KnownContactProfilesPropertiesProvisioningState;
(function (KnownContactProfilesPropertiesProvisioningState) {
    /** Creating */
    KnownContactProfilesPropertiesProvisioningState["Creating"] = "creating";
    /** Succeeded */
    KnownContactProfilesPropertiesProvisioningState["Succeeded"] = "succeeded";
    /** Failed */
    KnownContactProfilesPropertiesProvisioningState["Failed"] = "failed";
    /** Canceled */
    KnownContactProfilesPropertiesProvisioningState["Canceled"] = "canceled";
    /** Updating */
    KnownContactProfilesPropertiesProvisioningState["Updating"] = "updating";
    /** Deleting */
    KnownContactProfilesPropertiesProvisioningState["Deleting"] = "deleting";
})(KnownContactProfilesPropertiesProvisioningState || (KnownContactProfilesPropertiesProvisioningState = {}));
/** Known values of {@link Protocol} that the service accepts. */
export var KnownProtocol;
(function (KnownProtocol) {
    /** TCP */
    KnownProtocol["TCP"] = "TCP";
    /** UDP */
    KnownProtocol["UDP"] = "UDP";
})(KnownProtocol || (KnownProtocol = {}));
/** Known values of {@link CapabilityParameter} that the service accepts. */
export var KnownCapabilityParameter;
(function (KnownCapabilityParameter) {
    /** EarthObservation */
    KnownCapabilityParameter["EarthObservation"] = "EarthObservation";
    /** Communication */
    KnownCapabilityParameter["Communication"] = "Communication";
})(KnownCapabilityParameter || (KnownCapabilityParameter = {}));
/** Known values of {@link ReleaseMode} that the service accepts. */
export var KnownReleaseMode;
(function (KnownReleaseMode) {
    /** Preview */
    KnownReleaseMode["Preview"] = "Preview";
    /** GA */
    KnownReleaseMode["GA"] = "GA";
})(KnownReleaseMode || (KnownReleaseMode = {}));
/** Known values of {@link Status} that the service accepts. */
export var KnownStatus;
(function (KnownStatus) {
    /** Succeeded */
    KnownStatus["Succeeded"] = "Succeeded";
    /** Canceled */
    KnownStatus["Canceled"] = "Canceled";
    /** Failed */
    KnownStatus["Failed"] = "Failed";
    /** Running */
    KnownStatus["Running"] = "Running";
})(KnownStatus || (KnownStatus = {}));
/** Known values of {@link Capability} that the service accepts. */
export var KnownCapability;
(function (KnownCapability) {
    /** EarthObservation */
    KnownCapability["EarthObservation"] = "EarthObservation";
    /** Communication */
    KnownCapability["Communication"] = "Communication";
})(KnownCapability || (KnownCapability = {}));
