// @azure/arm-iothub/dist-esm/src/operations/resourceProviderCommon.js
/*
 * Copyright (c) Microsoft Corporation.
 * Licensed under the MIT License.
 *
 * Code generated by Microsoft (R) AutoRest Code Generator.
 * Changes may cause incorrect behavior and will be lost if the code is regenerated.
 */
import * as coreClient from "@azure/core-client";
import * as Mappers from "../models/mappers";
import * as Parameters from "../models/parameters";
/** Class containing ResourceProviderCommon operations. */
export class ResourceProviderCommonImpl {
    /**
     * Initialize a new instance of the class ResourceProviderCommon class.
     * @param client Reference to the service client
     */
    constructor(client) {
        this.client = client;
    }
    /**
     * Get the number of free and paid iot hubs in the subscription
     * @param options The options parameters.
     */
    getSubscriptionQuota(options) {
        return this.client.sendOperationRequest({ options }, getSubscriptionQuotaOperationSpec);
    }
}
// Operation Specifications
const serializer = coreClient.createSerializer(Mappers, /* isXml */ false);
const getSubscriptionQuotaOperationSpec = {
    path: "/subscriptions/{subscriptionId}/providers/Microsoft.Devices/usages",
    httpMethod: "GET",
    responses: {
        200: {
            bodyMapper: Mappers.UserSubscriptionQuotaListResult
        },
        default: {
            bodyMapper: Mappers.ErrorDetails
        }
    },
    queryParameters: [Parameters.apiVersion],
    urlParameters: [Parameters.$host, Parameters.subscriptionId],
    headerParameters: [Parameters.accept],
    serializer
};
