// @azure/arm-analysisservices/dist-esm/src/models/index.js
/*
 * Copyright (c) Microsoft Corporation.
 * Licensed under the MIT License.
 *
 * Code generated by Microsoft (R) AutoRest Code Generator.
 * Changes may cause incorrect behavior and will be lost if the code is regenerated.
 */
/** Known values of {@link State} that the service accepts. */
export var KnownState;
(function (KnownState) {
    /** Deleting */
    KnownState["Deleting"] = "Deleting";
    /** Succeeded */
    KnownState["Succeeded"] = "Succeeded";
    /** Failed */
    KnownState["Failed"] = "Failed";
    /** Paused */
    KnownState["Paused"] = "Paused";
    /** Suspended */
    KnownState["Suspended"] = "Suspended";
    /** Provisioning */
    KnownState["Provisioning"] = "Provisioning";
    /** Updating */
    KnownState["Updating"] = "Updating";
    /** Suspending */
    KnownState["Suspending"] = "Suspending";
    /** Pausing */
    KnownState["Pausing"] = "Pausing";
    /** Resuming */
    KnownState["Resuming"] = "Resuming";
    /** Preparing */
    KnownState["Preparing"] = "Preparing";
    /** Scaling */
    KnownState["Scaling"] = "Scaling";
})(KnownState || (KnownState = {}));
/** Known values of {@link ProvisioningState} that the service accepts. */
export var KnownProvisioningState;
(function (KnownProvisioningState) {
    /** Deleting */
    KnownProvisioningState["Deleting"] = "Deleting";
    /** Succeeded */
    KnownProvisioningState["Succeeded"] = "Succeeded";
    /** Failed */
    KnownProvisioningState["Failed"] = "Failed";
    /** Paused */
    KnownProvisioningState["Paused"] = "Paused";
    /** Suspended */
    KnownProvisioningState["Suspended"] = "Suspended";
    /** Provisioning */
    KnownProvisioningState["Provisioning"] = "Provisioning";
    /** Updating */
    KnownProvisioningState["Updating"] = "Updating";
    /** Suspending */
    KnownProvisioningState["Suspending"] = "Suspending";
    /** Pausing */
    KnownProvisioningState["Pausing"] = "Pausing";
    /** Resuming */
    KnownProvisioningState["Resuming"] = "Resuming";
    /** Preparing */
    KnownProvisioningState["Preparing"] = "Preparing";
    /** Scaling */
    KnownProvisioningState["Scaling"] = "Scaling";
})(KnownProvisioningState || (KnownProvisioningState = {}));
/** Known values of {@link SkuTier} that the service accepts. */
export var KnownSkuTier;
(function (KnownSkuTier) {
    /** Development */
    KnownSkuTier["Development"] = "Development";
    /** Basic */
    KnownSkuTier["Basic"] = "Basic";
    /** Standard */
    KnownSkuTier["Standard"] = "Standard";
})(KnownSkuTier || (KnownSkuTier = {}));
