// @azure/arm-confluent/dist-esm/src/models/index.js
/*
 * Copyright (c) Microsoft Corporation.
 * Licensed under the MIT License.
 *
 * Code generated by Microsoft (R) AutoRest Code Generator.
 * Changes may cause incorrect behavior and will be lost if the code is regenerated.
 */
/** Known values of {@link CreatedByType} that the service accepts. */
export var KnownCreatedByType;
(function (KnownCreatedByType) {
    /** User */
    KnownCreatedByType["User"] = "User";
    /** Application */
    KnownCreatedByType["Application"] = "Application";
    /** ManagedIdentity */
    KnownCreatedByType["ManagedIdentity"] = "ManagedIdentity";
    /** Key */
    KnownCreatedByType["Key"] = "Key";
})(KnownCreatedByType || (KnownCreatedByType = {}));
/** Known values of {@link ProvisionState} that the service accepts. */
export var KnownProvisionState;
(function (KnownProvisionState) {
    /** Accepted */
    KnownProvisionState["Accepted"] = "Accepted";
    /** Creating */
    KnownProvisionState["Creating"] = "Creating";
    /** Updating */
    KnownProvisionState["Updating"] = "Updating";
    /** Deleting */
    KnownProvisionState["Deleting"] = "Deleting";
    /** Succeeded */
    KnownProvisionState["Succeeded"] = "Succeeded";
    /** Failed */
    KnownProvisionState["Failed"] = "Failed";
    /** Canceled */
    KnownProvisionState["Canceled"] = "Canceled";
    /** Deleted */
    KnownProvisionState["Deleted"] = "Deleted";
    /** NotSpecified */
    KnownProvisionState["NotSpecified"] = "NotSpecified";
})(KnownProvisionState || (KnownProvisionState = {}));
/** Known values of {@link SaaSOfferStatus} that the service accepts. */
export var KnownSaaSOfferStatus;
(function (KnownSaaSOfferStatus) {
    /** Started */
    KnownSaaSOfferStatus["Started"] = "Started";
    /** PendingFulfillmentStart */
    KnownSaaSOfferStatus["PendingFulfillmentStart"] = "PendingFulfillmentStart";
    /** InProgress */
    KnownSaaSOfferStatus["InProgress"] = "InProgress";
    /** Subscribed */
    KnownSaaSOfferStatus["Subscribed"] = "Subscribed";
    /** Suspended */
    KnownSaaSOfferStatus["Suspended"] = "Suspended";
    /** Reinstated */
    KnownSaaSOfferStatus["Reinstated"] = "Reinstated";
    /** Succeeded */
    KnownSaaSOfferStatus["Succeeded"] = "Succeeded";
    /** Failed */
    KnownSaaSOfferStatus["Failed"] = "Failed";
    /** Unsubscribed */
    KnownSaaSOfferStatus["Unsubscribed"] = "Unsubscribed";
    /** Updating */
    KnownSaaSOfferStatus["Updating"] = "Updating";
})(KnownSaaSOfferStatus || (KnownSaaSOfferStatus = {}));
