// @azure/arm-confluent/dist-esm/src/models/parameters.js
/*
 * Copyright (c) Microsoft Corporation.
 * Licensed under the MIT License.
 *
 * Code generated by Microsoft (R) AutoRest Code Generator.
 * Changes may cause incorrect behavior and will be lost if the code is regenerated.
 */
import { ConfluentAgreementResource as ConfluentAgreementResourceMapper, OrganizationResource as OrganizationResourceMapper, OrganizationResourceUpdate as OrganizationResourceUpdateMapper, ListAccessRequestModel as ListAccessRequestModelMapper, CreateAPIKeyModel as CreateAPIKeyModelMapper, AccessInviteUserAccountModel as AccessInviteUserAccountModelMapper, AccessCreateRoleBindingRequestModel as AccessCreateRoleBindingRequestModelMapper, } from "../models/mappers";
export const accept = {
    parameterPath: "accept",
    mapper: {
        defaultValue: "application/json",
        isConstant: true,
        serializedName: "Accept",
        type: {
            name: "String",
        },
    },
};
export const $host = {
    parameterPath: "$host",
    mapper: {
        serializedName: "$host",
        required: true,
        type: {
            name: "String",
        },
    },
    skipEncoding: true,
};
export const apiVersion = {
    parameterPath: "apiVersion",
    mapper: {
        defaultValue: "2024-02-13",
        isConstant: true,
        serializedName: "api-version",
        type: {
            name: "String",
        },
    },
};
export const subscriptionId = {
    parameterPath: "subscriptionId",
    mapper: {
        serializedName: "subscriptionId",
        required: true,
        type: {
            name: "String",
        },
    },
};
export const contentType = {
    parameterPath: ["options", "contentType"],
    mapper: {
        defaultValue: "application/json",
        isConstant: true,
        serializedName: "Content-Type",
        type: {
            name: "String",
        },
    },
};
export const body = {
    parameterPath: ["options", "body"],
    mapper: ConfluentAgreementResourceMapper,
};
export const nextLink = {
    parameterPath: "nextLink",
    mapper: {
        serializedName: "nextLink",
        required: true,
        type: {
            name: "String",
        },
    },
    skipEncoding: true,
};
export const resourceGroupName = {
    parameterPath: "resourceGroupName",
    mapper: {
        serializedName: "resourceGroupName",
        required: true,
        type: {
            name: "String",
        },
    },
};
export const organizationName = {
    parameterPath: "organizationName",
    mapper: {
        serializedName: "organizationName",
        required: true,
        type: {
            name: "String",
        },
    },
};
export const body1 = {
    parameterPath: ["options", "body"],
    mapper: OrganizationResourceMapper,
};
export const body2 = {
    parameterPath: ["options", "body"],
    mapper: OrganizationResourceUpdateMapper,
};
export const resourceGroupName1 = {
    parameterPath: "resourceGroupName",
    mapper: {
        constraints: {
            MaxLength: 90,
            MinLength: 1,
        },
        serializedName: "resourceGroupName",
        required: true,
        type: {
            name: "String",
        },
    },
};
export const pageSize = {
    parameterPath: ["options", "pageSize"],
    mapper: {
        serializedName: "pageSize",
        type: {
            name: "Number",
        },
    },
};
export const pageToken = {
    parameterPath: ["options", "pageToken"],
    mapper: {
        serializedName: "pageToken",
        type: {
            name: "String",
        },
    },
};
export const environmentId = {
    parameterPath: "environmentId",
    mapper: {
        serializedName: "environmentId",
        required: true,
        type: {
            name: "String",
        },
    },
};
export const body3 = {
    parameterPath: "body",
    mapper: ListAccessRequestModelMapper,
};
export const body4 = {
    parameterPath: "body",
    mapper: CreateAPIKeyModelMapper,
};
export const clusterId = {
    parameterPath: "clusterId",
    mapper: {
        serializedName: "clusterId",
        required: true,
        type: {
            name: "String",
        },
    },
};
export const apiKeyId = {
    parameterPath: "apiKeyId",
    mapper: {
        serializedName: "apiKeyId",
        required: true,
        type: {
            name: "String",
        },
    },
};
export const body5 = {
    parameterPath: "body",
    mapper: OrganizationResourceMapper,
};
export const body6 = {
    parameterPath: "body",
    mapper: AccessInviteUserAccountModelMapper,
};
export const body7 = {
    parameterPath: "body",
    mapper: AccessCreateRoleBindingRequestModelMapper,
};
export const roleBindingId = {
    parameterPath: "roleBindingId",
    mapper: {
        serializedName: "roleBindingId",
        required: true,
        type: {
            name: "String",
        },
    },
};
