// @azure/arm-support/dist-esm/src/models/index.js
/*
 * Copyright (c) Microsoft Corporation.
 * Licensed under the MIT License.
 *
 * Code generated by Microsoft (R) AutoRest Code Generator.
 * Changes may cause incorrect behavior and will be lost if the code is regenerated.
 */
/** Known values of {@link SeverityLevel} that the service accepts. */
export var KnownSeverityLevel;
(function (KnownSeverityLevel) {
    /** Minimal */
    KnownSeverityLevel["Minimal"] = "minimal";
    /** Moderate */
    KnownSeverityLevel["Moderate"] = "moderate";
    /** Critical */
    KnownSeverityLevel["Critical"] = "critical";
    /** Highestcriticalimpact */
    KnownSeverityLevel["Highestcriticalimpact"] = "highestcriticalimpact";
})(KnownSeverityLevel || (KnownSeverityLevel = {}));
/** Known values of {@link PreferredContactMethod} that the service accepts. */
export var KnownPreferredContactMethod;
(function (KnownPreferredContactMethod) {
    /** Email */
    KnownPreferredContactMethod["Email"] = "email";
    /** Phone */
    KnownPreferredContactMethod["Phone"] = "phone";
})(KnownPreferredContactMethod || (KnownPreferredContactMethod = {}));
/** Known values of {@link Status} that the service accepts. */
export var KnownStatus;
(function (KnownStatus) {
    /** Open */
    KnownStatus["Open"] = "open";
    /** Closed */
    KnownStatus["Closed"] = "closed";
})(KnownStatus || (KnownStatus = {}));
/** Known values of {@link CommunicationType} that the service accepts. */
export var KnownCommunicationType;
(function (KnownCommunicationType) {
    /** Web */
    KnownCommunicationType["Web"] = "web";
    /** Phone */
    KnownCommunicationType["Phone"] = "phone";
})(KnownCommunicationType || (KnownCommunicationType = {}));
/** Known values of {@link CommunicationDirection} that the service accepts. */
export var KnownCommunicationDirection;
(function (KnownCommunicationDirection) {
    /** Inbound */
    KnownCommunicationDirection["Inbound"] = "inbound";
    /** Outbound */
    KnownCommunicationDirection["Outbound"] = "outbound";
})(KnownCommunicationDirection || (KnownCommunicationDirection = {}));
