// @azure/arm-batchai/esm/models/parameters.js
/*
 * Copyright (c) Microsoft Corporation. All rights reserved.
 * Licensed under the MIT License. See License.txt in the project root for
 * license information.
 *
 * Code generated by Microsoft (R) AutoRest Code Generator.
 * Changes may cause incorrect behavior and will be lost if the code is
 * regenerated.
 */
export var acceptLanguage = {
    parameterPath: "acceptLanguage",
    mapper: {
        serializedName: "accept-language",
        defaultValue: 'en-US',
        type: {
            name: "String"
        }
    }
};
export var apiVersion = {
    parameterPath: "apiVersion",
    mapper: {
        required: true,
        serializedName: "api-version",
        type: {
            name: "String"
        }
    }
};
export var clusterName = {
    parameterPath: "clusterName",
    mapper: {
        required: true,
        serializedName: "clusterName",
        constraints: {
            MaxLength: 64,
            MinLength: 1,
            Pattern: /^[-\w_]+$/
        },
        type: {
            name: "String"
        }
    }
};
export var directory = {
    parameterPath: [
        "jobsListOutputFilesOptions",
        "directory"
    ],
    mapper: {
        serializedName: "directory",
        defaultValue: '.',
        type: {
            name: "String"
        }
    }
};
export var experimentName = {
    parameterPath: "experimentName",
    mapper: {
        required: true,
        serializedName: "experimentName",
        constraints: {
            MaxLength: 64,
            MinLength: 1,
            Pattern: /^[-\w_]+$/
        },
        type: {
            name: "String"
        }
    }
};
export var fileServerName = {
    parameterPath: "fileServerName",
    mapper: {
        required: true,
        serializedName: "fileServerName",
        constraints: {
            MaxLength: 64,
            MinLength: 1,
            Pattern: /^[-\w_]+$/
        },
        type: {
            name: "String"
        }
    }
};
export var jobName = {
    parameterPath: "jobName",
    mapper: {
        required: true,
        serializedName: "jobName",
        constraints: {
            MaxLength: 64,
            MinLength: 1,
            Pattern: /^[-\w_]+$/
        },
        type: {
            name: "String"
        }
    }
};
export var linkexpiryinminutes = {
    parameterPath: [
        "jobsListOutputFilesOptions",
        "linkexpiryinminutes"
    ],
    mapper: {
        serializedName: "linkexpiryinminutes",
        defaultValue: 60,
        constraints: {
            InclusiveMaximum: 600,
            InclusiveMinimum: 5
        },
        type: {
            name: "Number"
        }
    }
};
export var location = {
    parameterPath: "location",
    mapper: {
        required: true,
        serializedName: "location",
        constraints: {
            Pattern: /^[-\w\._]+$/
        },
        type: {
            name: "String"
        }
    }
};
export var maxResults0 = {
    parameterPath: [
        "options",
        "workspacesListOptions",
        "maxResults"
    ],
    mapper: {
        serializedName: "maxresults",
        defaultValue: 1000,
        constraints: {
            InclusiveMaximum: 1000,
            InclusiveMinimum: 1
        },
        type: {
            name: "Number"
        }
    }
};
export var maxResults1 = {
    parameterPath: [
        "options",
        "workspacesListByResourceGroupOptions",
        "maxResults"
    ],
    mapper: {
        serializedName: "maxresults",
        defaultValue: 1000,
        constraints: {
            InclusiveMaximum: 1000,
            InclusiveMinimum: 1
        },
        type: {
            name: "Number"
        }
    }
};
export var maxResults2 = {
    parameterPath: [
        "options",
        "experimentsListByWorkspaceOptions",
        "maxResults"
    ],
    mapper: {
        serializedName: "maxresults",
        defaultValue: 1000,
        constraints: {
            InclusiveMaximum: 1000,
            InclusiveMinimum: 1
        },
        type: {
            name: "Number"
        }
    }
};
export var maxResults3 = {
    parameterPath: [
        "options",
        "jobsListByExperimentOptions",
        "maxResults"
    ],
    mapper: {
        serializedName: "maxresults",
        defaultValue: 1000,
        constraints: {
            InclusiveMaximum: 1000,
            InclusiveMinimum: 1
        },
        type: {
            name: "Number"
        }
    }
};
export var maxResults4 = {
    parameterPath: [
        "jobsListOutputFilesOptions",
        "maxResults"
    ],
    mapper: {
        serializedName: "maxresults",
        defaultValue: 1000,
        constraints: {
            InclusiveMaximum: 1000,
            InclusiveMinimum: 1
        },
        type: {
            name: "Number"
        }
    }
};
export var maxResults5 = {
    parameterPath: [
        "options",
        "fileServersListByWorkspaceOptions",
        "maxResults"
    ],
    mapper: {
        serializedName: "maxresults",
        defaultValue: 1000,
        constraints: {
            InclusiveMaximum: 1000,
            InclusiveMinimum: 1
        },
        type: {
            name: "Number"
        }
    }
};
export var maxResults6 = {
    parameterPath: [
        "options",
        "clustersListByWorkspaceOptions",
        "maxResults"
    ],
    mapper: {
        serializedName: "maxresults",
        defaultValue: 1000,
        constraints: {
            InclusiveMaximum: 1000,
            InclusiveMinimum: 1
        },
        type: {
            name: "Number"
        }
    }
};
export var nextPageLink = {
    parameterPath: "nextPageLink",
    mapper: {
        required: true,
        serializedName: "nextLink",
        type: {
            name: "String"
        }
    },
    skipEncoding: true
};
export var outputdirectoryid = {
    parameterPath: [
        "jobsListOutputFilesOptions",
        "outputdirectoryid"
    ],
    mapper: {
        required: true,
        serializedName: "outputdirectoryid",
        type: {
            name: "String"
        }
    }
};
export var resourceGroupName = {
    parameterPath: "resourceGroupName",
    mapper: {
        required: true,
        serializedName: "resourceGroupName",
        constraints: {
            Pattern: /^[-\w\._]+$/
        },
        type: {
            name: "String"
        }
    }
};
export var subscriptionId = {
    parameterPath: "subscriptionId",
    mapper: {
        required: true,
        serializedName: "subscriptionId",
        type: {
            name: "String"
        }
    }
};
export var workspaceName = {
    parameterPath: "workspaceName",
    mapper: {
        required: true,
        serializedName: "workspaceName",
        constraints: {
            MaxLength: 64,
            MinLength: 1,
            Pattern: /^[-\w_]+$/
        },
        type: {
            name: "String"
        }
    }
};
