// @azure/arm-scvmm/dist-esm/src/models/index.js
/*
 * Copyright (c) Microsoft Corporation.
 * Licensed under the MIT License.
 *
 * Code generated by Microsoft (R) AutoRest Code Generator.
 * Changes may cause incorrect behavior and will be lost if the code is regenerated.
 */
/** Known values of {@link CreatedByType} that the service accepts. */
export var KnownCreatedByType;
(function (KnownCreatedByType) {
    /** User */
    KnownCreatedByType["User"] = "User";
    /** Application */
    KnownCreatedByType["Application"] = "Application";
    /** ManagedIdentity */
    KnownCreatedByType["ManagedIdentity"] = "ManagedIdentity";
    /** Key */
    KnownCreatedByType["Key"] = "Key";
})(KnownCreatedByType || (KnownCreatedByType = {}));
/** Known values of {@link OsType} that the service accepts. */
export var KnownOsType;
(function (KnownOsType) {
    /** Windows */
    KnownOsType["Windows"] = "Windows";
    /** Linux */
    KnownOsType["Linux"] = "Linux";
    /** Other */
    KnownOsType["Other"] = "Other";
})(KnownOsType || (KnownOsType = {}));
/** Known values of {@link LimitCpuForMigration} that the service accepts. */
export var KnownLimitCpuForMigration;
(function (KnownLimitCpuForMigration) {
    /** False */
    KnownLimitCpuForMigration["False"] = "false";
    /** True */
    KnownLimitCpuForMigration["True"] = "true";
})(KnownLimitCpuForMigration || (KnownLimitCpuForMigration = {}));
/** Known values of {@link DynamicMemoryEnabled} that the service accepts. */
export var KnownDynamicMemoryEnabled;
(function (KnownDynamicMemoryEnabled) {
    /** False */
    KnownDynamicMemoryEnabled["False"] = "false";
    /** True */
    KnownDynamicMemoryEnabled["True"] = "true";
})(KnownDynamicMemoryEnabled || (KnownDynamicMemoryEnabled = {}));
/** Known values of {@link AllocationMethod} that the service accepts. */
export var KnownAllocationMethod;
(function (KnownAllocationMethod) {
    /** Dynamic */
    KnownAllocationMethod["Dynamic"] = "Dynamic";
    /** Static */
    KnownAllocationMethod["Static"] = "Static";
})(KnownAllocationMethod || (KnownAllocationMethod = {}));
/** Known values of {@link CreateDiffDisk} that the service accepts. */
export var KnownCreateDiffDisk;
(function (KnownCreateDiffDisk) {
    /** False */
    KnownCreateDiffDisk["False"] = "false";
    /** True */
    KnownCreateDiffDisk["True"] = "true";
})(KnownCreateDiffDisk || (KnownCreateDiffDisk = {}));
/** Known values of {@link IsCustomizable} that the service accepts. */
export var KnownIsCustomizable;
(function (KnownIsCustomizable) {
    /** False */
    KnownIsCustomizable["False"] = "false";
    /** True */
    KnownIsCustomizable["True"] = "true";
})(KnownIsCustomizable || (KnownIsCustomizable = {}));
/** Known values of {@link InventoryType} that the service accepts. */
export var KnownInventoryType;
(function (KnownInventoryType) {
    /** Cloud */
    KnownInventoryType["Cloud"] = "Cloud";
    /** VirtualNetwork */
    KnownInventoryType["VirtualNetwork"] = "VirtualNetwork";
    /** VirtualMachineTemplate */
    KnownInventoryType["VirtualMachineTemplate"] = "VirtualMachineTemplate";
    /** VirtualMachine */
    KnownInventoryType["VirtualMachine"] = "VirtualMachine";
})(KnownInventoryType || (KnownInventoryType = {}));
