// @azure/arm-scvmm/dist-esm/src/models/mappers.js
/*
 * Copyright (c) Microsoft Corporation.
 * Licensed under the MIT License.
 *
 * Code generated by Microsoft (R) AutoRest Code Generator.
 * Changes may cause incorrect behavior and will be lost if the code is regenerated.
 */
export const VMMServer = {
    type: {
        name: "Composite",
        className: "VMMServer",
        modelProperties: {
            id: {
                serializedName: "id",
                readOnly: true,
                type: {
                    name: "String"
                }
            },
            name: {
                serializedName: "name",
                readOnly: true,
                type: {
                    name: "String"
                }
            },
            type: {
                serializedName: "type",
                readOnly: true,
                type: {
                    name: "String"
                }
            },
            location: {
                serializedName: "location",
                required: true,
                type: {
                    name: "String"
                }
            },
            tags: {
                serializedName: "tags",
                type: {
                    name: "Dictionary",
                    value: { type: { name: "String" } }
                }
            },
            systemData: {
                serializedName: "systemData",
                type: {
                    name: "Composite",
                    className: "SystemData"
                }
            },
            extendedLocation: {
                serializedName: "extendedLocation",
                type: {
                    name: "Composite",
                    className: "ExtendedLocation"
                }
            },
            credentials: {
                serializedName: "properties.credentials",
                type: {
                    name: "Composite",
                    className: "VMMServerPropertiesCredentials"
                }
            },
            fqdn: {
                constraints: {
                    MinLength: 1
                },
                serializedName: "properties.fqdn",
                required: true,
                type: {
                    name: "String"
                }
            },
            port: {
                constraints: {
                    InclusiveMaximum: 65535,
                    InclusiveMinimum: 1
                },
                serializedName: "properties.port",
                type: {
                    name: "Number"
                }
            },
            connectionStatus: {
                serializedName: "properties.connectionStatus",
                readOnly: true,
                type: {
                    name: "String"
                }
            },
            errorMessage: {
                serializedName: "properties.errorMessage",
                readOnly: true,
                type: {
                    name: "String"
                }
            },
            uuid: {
                serializedName: "properties.uuid",
                readOnly: true,
                type: {
                    name: "String"
                }
            },
            version: {
                serializedName: "properties.version",
                readOnly: true,
                type: {
                    name: "String"
                }
            },
            provisioningState: {
                serializedName: "properties.provisioningState",
                readOnly: true,
                type: {
                    name: "String"
                }
            }
        }
    }
};
export const VMMServerPropertiesCredentials = {
    type: {
        name: "Composite",
        className: "VMMServerPropertiesCredentials",
        modelProperties: {
            username: {
                serializedName: "username",
                type: {
                    name: "String"
                }
            },
            password: {
                serializedName: "password",
                type: {
                    name: "String"
                }
            }
        }
    }
};
export const SystemData = {
    type: {
        name: "Composite",
        className: "SystemData",
        modelProperties: {
            createdBy: {
                serializedName: "createdBy",
                type: {
                    name: "String"
                }
            },
            createdByType: {
                serializedName: "createdByType",
                type: {
                    name: "String"
                }
            },
            createdAt: {
                serializedName: "createdAt",
                type: {
                    name: "DateTime"
                }
            },
            lastModifiedBy: {
                serializedName: "lastModifiedBy",
                type: {
                    name: "String"
                }
            },
            lastModifiedByType: {
                serializedName: "lastModifiedByType",
                type: {
                    name: "String"
                }
            },
            lastModifiedAt: {
                serializedName: "lastModifiedAt",
                type: {
                    name: "DateTime"
                }
            }
        }
    }
};
export const ExtendedLocation = {
    type: {
        name: "Composite",
        className: "ExtendedLocation",
        modelProperties: {
            type: {
                serializedName: "type",
                type: {
                    name: "String"
                }
            },
            name: {
                serializedName: "name",
                type: {
                    name: "String"
                }
            }
        }
    }
};
export const ErrorResponse = {
    type: {
        name: "Composite",
        className: "ErrorResponse",
        modelProperties: {
            error: {
                serializedName: "error",
                type: {
                    name: "Composite",
                    className: "ErrorDefinition"
                }
            }
        }
    }
};
export const ErrorDefinition = {
    type: {
        name: "Composite",
        className: "ErrorDefinition",
        modelProperties: {
            code: {
                serializedName: "code",
                readOnly: true,
                type: {
                    name: "String"
                }
            },
            message: {
                serializedName: "message",
                readOnly: true,
                type: {
                    name: "String"
                }
            },
            details: {
                serializedName: "details",
                readOnly: true,
                type: {
                    name: "Sequence",
                    element: {
                        type: {
                            name: "Composite",
                            className: "ErrorDefinition"
                        }
                    }
                }
            }
        }
    }
};
export const ResourcePatch = {
    type: {
        name: "Composite",
        className: "ResourcePatch",
        modelProperties: {
            tags: {
                serializedName: "tags",
                type: {
                    name: "Dictionary",
                    value: { type: { name: "String" } }
                }
            }
        }
    }
};
export const VMMServerListResult = {
    type: {
        name: "Composite",
        className: "VMMServerListResult",
        modelProperties: {
            value: {
                serializedName: "value",
                type: {
                    name: "Sequence",
                    element: {
                        type: {
                            name: "Composite",
                            className: "VMMServer"
                        }
                    }
                }
            },
            nextLink: {
                serializedName: "nextLink",
                type: {
                    name: "String"
                }
            }
        }
    }
};
export const ResourceProviderOperationList = {
    type: {
        name: "Composite",
        className: "ResourceProviderOperationList",
        modelProperties: {
            value: {
                serializedName: "value",
                type: {
                    name: "Sequence",
                    element: {
                        type: {
                            name: "Composite",
                            className: "ResourceProviderOperation"
                        }
                    }
                }
            },
            nextLink: {
                serializedName: "nextLink",
                type: {
                    name: "String"
                }
            }
        }
    }
};
export const ResourceProviderOperation = {
    type: {
        name: "Composite",
        className: "ResourceProviderOperation",
        modelProperties: {
            isDataAction: {
                serializedName: "isDataAction",
                type: {
                    name: "String"
                }
            },
            name: {
                serializedName: "name",
                type: {
                    name: "String"
                }
            },
            display: {
                serializedName: "display",
                type: {
                    name: "Composite",
                    className: "ResourceProviderOperationDisplay"
                }
            }
        }
    }
};
export const ResourceProviderOperationDisplay = {
    type: {
        name: "Composite",
        className: "ResourceProviderOperationDisplay",
        modelProperties: {
            provider: {
                serializedName: "provider",
                type: {
                    name: "String"
                }
            },
            resource: {
                serializedName: "resource",
                type: {
                    name: "String"
                }
            },
            operation: {
                serializedName: "operation",
                type: {
                    name: "String"
                }
            },
            description: {
                serializedName: "description",
                type: {
                    name: "String"
                }
            }
        }
    }
};
export const Cloud = {
    type: {
        name: "Composite",
        className: "Cloud",
        modelProperties: {
            id: {
                serializedName: "id",
                readOnly: true,
                type: {
                    name: "String"
                }
            },
            name: {
                serializedName: "name",
                readOnly: true,
                type: {
                    name: "String"
                }
            },
            type: {
                serializedName: "type",
                readOnly: true,
                type: {
                    name: "String"
                }
            },
            location: {
                serializedName: "location",
                required: true,
                type: {
                    name: "String"
                }
            },
            tags: {
                serializedName: "tags",
                type: {
                    name: "Dictionary",
                    value: { type: { name: "String" } }
                }
            },
            systemData: {
                serializedName: "systemData",
                type: {
                    name: "Composite",
                    className: "SystemData"
                }
            },
            extendedLocation: {
                serializedName: "extendedLocation",
                type: {
                    name: "Composite",
                    className: "ExtendedLocation"
                }
            },
            inventoryItemId: {
                serializedName: "properties.inventoryItemId",
                type: {
                    name: "String"
                }
            },
            uuid: {
                constraints: {
                    MinLength: 1
                },
                serializedName: "properties.uuid",
                type: {
                    name: "String"
                }
            },
            vmmServerId: {
                serializedName: "properties.vmmServerId",
                type: {
                    name: "String"
                }
            },
            cloudName: {
                serializedName: "properties.cloudName",
                readOnly: true,
                type: {
                    name: "String"
                }
            },
            cloudCapacity: {
                serializedName: "properties.cloudCapacity",
                type: {
                    name: "Composite",
                    className: "CloudCapacity"
                }
            },
            storageQoSPolicies: {
                serializedName: "properties.storageQoSPolicies",
                readOnly: true,
                type: {
                    name: "Sequence",
                    element: {
                        type: {
                            name: "Composite",
                            className: "StorageQoSPolicy"
                        }
                    }
                }
            },
            provisioningState: {
                serializedName: "properties.provisioningState",
                readOnly: true,
                type: {
                    name: "String"
                }
            }
        }
    }
};
export const CloudCapacity = {
    type: {
        name: "Composite",
        className: "CloudCapacity",
        modelProperties: {
            cpuCount: {
                serializedName: "cpuCount",
                type: {
                    name: "Number"
                }
            },
            memoryMB: {
                serializedName: "memoryMB",
                type: {
                    name: "Number"
                }
            },
            vmCount: {
                serializedName: "vmCount",
                type: {
                    name: "Number"
                }
            }
        }
    }
};
export const StorageQoSPolicy = {
    type: {
        name: "Composite",
        className: "StorageQoSPolicy",
        modelProperties: {
            name: {
                serializedName: "name",
                type: {
                    name: "String"
                }
            },
            id: {
                serializedName: "id",
                type: {
                    name: "String"
                }
            },
            iopsMaximum: {
                serializedName: "iopsMaximum",
                type: {
                    name: "Number"
                }
            },
            iopsMinimum: {
                serializedName: "iopsMinimum",
                type: {
                    name: "Number"
                }
            },
            bandwidthLimit: {
                serializedName: "bandwidthLimit",
                type: {
                    name: "Number"
                }
            },
            policyId: {
                serializedName: "policyId",
                type: {
                    name: "String"
                }
            }
        }
    }
};
export const CloudListResult = {
    type: {
        name: "Composite",
        className: "CloudListResult",
        modelProperties: {
            value: {
                serializedName: "value",
                type: {
                    name: "Sequence",
                    element: {
                        type: {
                            name: "Composite",
                            className: "Cloud"
                        }
                    }
                }
            },
            nextLink: {
                serializedName: "nextLink",
                type: {
                    name: "String"
                }
            }
        }
    }
};
export const VirtualNetwork = {
    type: {
        name: "Composite",
        className: "VirtualNetwork",
        modelProperties: {
            id: {
                serializedName: "id",
                readOnly: true,
                type: {
                    name: "String"
                }
            },
            name: {
                serializedName: "name",
                readOnly: true,
                type: {
                    name: "String"
                }
            },
            type: {
                serializedName: "type",
                readOnly: true,
                type: {
                    name: "String"
                }
            },
            location: {
                serializedName: "location",
                required: true,
                type: {
                    name: "String"
                }
            },
            tags: {
                serializedName: "tags",
                type: {
                    name: "Dictionary",
                    value: { type: { name: "String" } }
                }
            },
            systemData: {
                serializedName: "systemData",
                type: {
                    name: "Composite",
                    className: "SystemData"
                }
            },
            extendedLocation: {
                serializedName: "extendedLocation",
                type: {
                    name: "Composite",
                    className: "ExtendedLocation"
                }
            },
            inventoryItemId: {
                serializedName: "properties.inventoryItemId",
                type: {
                    name: "String"
                }
            },
            uuid: {
                constraints: {
                    MinLength: 1
                },
                serializedName: "properties.uuid",
                type: {
                    name: "String"
                }
            },
            vmmServerId: {
                serializedName: "properties.vmmServerId",
                type: {
                    name: "String"
                }
            },
            networkName: {
                serializedName: "properties.networkName",
                readOnly: true,
                type: {
                    name: "String"
                }
            },
            provisioningState: {
                serializedName: "properties.provisioningState",
                readOnly: true,
                type: {
                    name: "String"
                }
            }
        }
    }
};
export const VirtualNetworkListResult = {
    type: {
        name: "Composite",
        className: "VirtualNetworkListResult",
        modelProperties: {
            value: {
                serializedName: "value",
                type: {
                    name: "Sequence",
                    element: {
                        type: {
                            name: "Composite",
                            className: "VirtualNetwork"
                        }
                    }
                }
            },
            nextLink: {
                serializedName: "nextLink",
                type: {
                    name: "String"
                }
            }
        }
    }
};
export const VirtualMachine = {
    type: {
        name: "Composite",
        className: "VirtualMachine",
        modelProperties: {
            id: {
                serializedName: "id",
                readOnly: true,
                type: {
                    name: "String"
                }
            },
            name: {
                serializedName: "name",
                readOnly: true,
                type: {
                    name: "String"
                }
            },
            type: {
                serializedName: "type",
                readOnly: true,
                type: {
                    name: "String"
                }
            },
            location: {
                serializedName: "location",
                required: true,
                type: {
                    name: "String"
                }
            },
            tags: {
                serializedName: "tags",
                type: {
                    name: "Dictionary",
                    value: { type: { name: "String" } }
                }
            },
            systemData: {
                serializedName: "systemData",
                type: {
                    name: "Composite",
                    className: "SystemData"
                }
            },
            extendedLocation: {
                serializedName: "extendedLocation",
                type: {
                    name: "Composite",
                    className: "ExtendedLocation"
                }
            },
            inventoryItemId: {
                serializedName: "properties.inventoryItemId",
                type: {
                    name: "String"
                }
            },
            vmmServerId: {
                serializedName: "properties.vmmServerId",
                type: {
                    name: "String"
                }
            },
            cloudId: {
                serializedName: "properties.cloudId",
                type: {
                    name: "String"
                }
            },
            templateId: {
                serializedName: "properties.templateId",
                type: {
                    name: "String"
                }
            },
            checkpointType: {
                serializedName: "properties.checkpointType",
                type: {
                    name: "String"
                }
            },
            checkpoints: {
                serializedName: "properties.checkpoints",
                type: {
                    name: "Sequence",
                    element: {
                        type: {
                            name: "Composite",
                            className: "Checkpoint"
                        }
                    }
                }
            },
            availabilitySets: {
                serializedName: "properties.availabilitySets",
                type: {
                    name: "Sequence",
                    element: {
                        type: {
                            name: "Composite",
                            className: "AvailabilitySetListItem"
                        }
                    }
                }
            },
            osProfile: {
                serializedName: "properties.osProfile",
                type: {
                    name: "Composite",
                    className: "OsProfile"
                }
            },
            hardwareProfile: {
                serializedName: "properties.hardwareProfile",
                type: {
                    name: "Composite",
                    className: "HardwareProfile"
                }
            },
            networkProfile: {
                serializedName: "properties.networkProfile",
                type: {
                    name: "Composite",
                    className: "NetworkProfile"
                }
            },
            storageProfile: {
                serializedName: "properties.storageProfile",
                type: {
                    name: "Composite",
                    className: "StorageProfile"
                }
            },
            vmName: {
                constraints: {
                    MinLength: 1
                },
                serializedName: "properties.vmName",
                type: {
                    name: "String"
                }
            },
            uuid: {
                serializedName: "properties.uuid",
                type: {
                    name: "String"
                }
            },
            generation: {
                serializedName: "properties.generation",
                type: {
                    name: "Number"
                }
            },
            powerState: {
                serializedName: "properties.powerState",
                readOnly: true,
                type: {
                    name: "String"
                }
            },
            provisioningState: {
                serializedName: "properties.provisioningState",
                readOnly: true,
                type: {
                    name: "String"
                }
            }
        }
    }
};
export const Checkpoint = {
    type: {
        name: "Composite",
        className: "Checkpoint",
        modelProperties: {
            parentCheckpointID: {
                serializedName: "parentCheckpointID",
                type: {
                    name: "String"
                }
            },
            checkpointID: {
                serializedName: "checkpointID",
                type: {
                    name: "String"
                }
            },
            name: {
                serializedName: "name",
                type: {
                    name: "String"
                }
            },
            description: {
                serializedName: "description",
                type: {
                    name: "String"
                }
            }
        }
    }
};
export const AvailabilitySetListItem = {
    type: {
        name: "Composite",
        className: "AvailabilitySetListItem",
        modelProperties: {
            id: {
                serializedName: "id",
                type: {
                    name: "String"
                }
            },
            name: {
                serializedName: "name",
                type: {
                    name: "String"
                }
            }
        }
    }
};
export const OsProfile = {
    type: {
        name: "Composite",
        className: "OsProfile",
        modelProperties: {
            adminPassword: {
                serializedName: "adminPassword",
                type: {
                    name: "String"
                }
            },
            computerName: {
                serializedName: "computerName",
                type: {
                    name: "String"
                }
            },
            osType: {
                serializedName: "osType",
                readOnly: true,
                type: {
                    name: "String"
                }
            },
            osName: {
                serializedName: "osName",
                readOnly: true,
                type: {
                    name: "String"
                }
            }
        }
    }
};
export const HardwareProfile = {
    type: {
        name: "Composite",
        className: "HardwareProfile",
        modelProperties: {
            memoryMB: {
                serializedName: "memoryMB",
                type: {
                    name: "Number"
                }
            },
            cpuCount: {
                serializedName: "cpuCount",
                type: {
                    name: "Number"
                }
            },
            limitCpuForMigration: {
                serializedName: "limitCpuForMigration",
                type: {
                    name: "String"
                }
            },
            dynamicMemoryEnabled: {
                serializedName: "dynamicMemoryEnabled",
                type: {
                    name: "String"
                }
            },
            dynamicMemoryMaxMB: {
                serializedName: "dynamicMemoryMaxMB",
                type: {
                    name: "Number"
                }
            },
            dynamicMemoryMinMB: {
                serializedName: "dynamicMemoryMinMB",
                type: {
                    name: "Number"
                }
            },
            isHighlyAvailable: {
                serializedName: "isHighlyAvailable",
                type: {
                    name: "String"
                }
            }
        }
    }
};
export const NetworkProfile = {
    type: {
        name: "Composite",
        className: "NetworkProfile",
        modelProperties: {
            networkInterfaces: {
                serializedName: "networkInterfaces",
                type: {
                    name: "Sequence",
                    element: {
                        type: {
                            name: "Composite",
                            className: "NetworkInterfaces"
                        }
                    }
                }
            }
        }
    }
};
export const NetworkInterfaces = {
    type: {
        name: "Composite",
        className: "NetworkInterfaces",
        modelProperties: {
            name: {
                serializedName: "name",
                type: {
                    name: "String"
                }
            },
            displayName: {
                serializedName: "displayName",
                readOnly: true,
                type: {
                    name: "String"
                }
            },
            ipv4Addresses: {
                serializedName: "ipv4Addresses",
                readOnly: true,
                type: {
                    name: "Sequence",
                    element: {
                        type: {
                            name: "String"
                        }
                    }
                }
            },
            ipv6Addresses: {
                serializedName: "ipv6Addresses",
                readOnly: true,
                type: {
                    name: "Sequence",
                    element: {
                        type: {
                            name: "String"
                        }
                    }
                }
            },
            macAddress: {
                serializedName: "macAddress",
                type: {
                    name: "String"
                }
            },
            virtualNetworkId: {
                serializedName: "virtualNetworkId",
                type: {
                    name: "String"
                }
            },
            networkName: {
                serializedName: "networkName",
                readOnly: true,
                type: {
                    name: "String"
                }
            },
            ipv4AddressType: {
                serializedName: "ipv4AddressType",
                type: {
                    name: "String"
                }
            },
            ipv6AddressType: {
                serializedName: "ipv6AddressType",
                type: {
                    name: "String"
                }
            },
            macAddressType: {
                serializedName: "macAddressType",
                type: {
                    name: "String"
                }
            },
            nicId: {
                serializedName: "nicId",
                type: {
                    name: "String"
                }
            }
        }
    }
};
export const StorageProfile = {
    type: {
        name: "Composite",
        className: "StorageProfile",
        modelProperties: {
            disks: {
                serializedName: "disks",
                type: {
                    name: "Sequence",
                    element: {
                        type: {
                            name: "Composite",
                            className: "VirtualDisk"
                        }
                    }
                }
            }
        }
    }
};
export const VirtualDisk = {
    type: {
        name: "Composite",
        className: "VirtualDisk",
        modelProperties: {
            name: {
                serializedName: "name",
                type: {
                    name: "String"
                }
            },
            displayName: {
                serializedName: "displayName",
                readOnly: true,
                type: {
                    name: "String"
                }
            },
            diskId: {
                serializedName: "diskId",
                type: {
                    name: "String"
                }
            },
            diskSizeGB: {
                serializedName: "diskSizeGB",
                type: {
                    name: "Number"
                }
            },
            maxDiskSizeGB: {
                serializedName: "maxDiskSizeGB",
                readOnly: true,
                type: {
                    name: "Number"
                }
            },
            bus: {
                serializedName: "bus",
                type: {
                    name: "Number"
                }
            },
            lun: {
                serializedName: "lun",
                type: {
                    name: "Number"
                }
            },
            busType: {
                serializedName: "busType",
                type: {
                    name: "String"
                }
            },
            vhdType: {
                serializedName: "vhdType",
                type: {
                    name: "String"
                }
            },
            volumeType: {
                serializedName: "volumeType",
                readOnly: true,
                type: {
                    name: "String"
                }
            },
            vhdFormatType: {
                serializedName: "vhdFormatType",
                readOnly: true,
                type: {
                    name: "String"
                }
            },
            templateDiskId: {
                serializedName: "templateDiskId",
                type: {
                    name: "String"
                }
            },
            storageQoSPolicy: {
                serializedName: "storageQoSPolicy",
                type: {
                    name: "Composite",
                    className: "StorageQoSPolicyDetails"
                }
            },
            createDiffDisk: {
                serializedName: "createDiffDisk",
                type: {
                    name: "String"
                }
            }
        }
    }
};
export const StorageQoSPolicyDetails = {
    type: {
        name: "Composite",
        className: "StorageQoSPolicyDetails",
        modelProperties: {
            name: {
                serializedName: "name",
                type: {
                    name: "String"
                }
            },
            id: {
                serializedName: "id",
                type: {
                    name: "String"
                }
            }
        }
    }
};
export const VirtualMachineUpdate = {
    type: {
        name: "Composite",
        className: "VirtualMachineUpdate",
        modelProperties: {
            properties: {
                serializedName: "properties",
                type: {
                    name: "Composite",
                    className: "VirtualMachineUpdateProperties"
                }
            },
            tags: {
                serializedName: "tags",
                type: {
                    name: "Dictionary",
                    value: { type: { name: "String" } }
                }
            }
        }
    }
};
export const VirtualMachineUpdateProperties = {
    type: {
        name: "Composite",
        className: "VirtualMachineUpdateProperties",
        modelProperties: {
            hardwareProfile: {
                serializedName: "hardwareProfile",
                type: {
                    name: "Composite",
                    className: "HardwareProfileUpdate"
                }
            },
            storageProfile: {
                serializedName: "storageProfile",
                type: {
                    name: "Composite",
                    className: "StorageProfileUpdate"
                }
            },
            networkProfile: {
                serializedName: "networkProfile",
                type: {
                    name: "Composite",
                    className: "NetworkProfileUpdate"
                }
            },
            availabilitySets: {
                serializedName: "availabilitySets",
                type: {
                    name: "Sequence",
                    element: {
                        type: {
                            name: "Composite",
                            className: "AvailabilitySetListItem"
                        }
                    }
                }
            }
        }
    }
};
export const HardwareProfileUpdate = {
    type: {
        name: "Composite",
        className: "HardwareProfileUpdate",
        modelProperties: {
            memoryMB: {
                serializedName: "memoryMB",
                type: {
                    name: "Number"
                }
            },
            cpuCount: {
                serializedName: "cpuCount",
                type: {
                    name: "Number"
                }
            },
            limitCpuForMigration: {
                serializedName: "limitCpuForMigration",
                type: {
                    name: "String"
                }
            },
            dynamicMemoryEnabled: {
                serializedName: "dynamicMemoryEnabled",
                type: {
                    name: "String"
                }
            },
            dynamicMemoryMaxMB: {
                serializedName: "dynamicMemoryMaxMB",
                type: {
                    name: "Number"
                }
            },
            dynamicMemoryMinMB: {
                serializedName: "dynamicMemoryMinMB",
                type: {
                    name: "Number"
                }
            }
        }
    }
};
export const StorageProfileUpdate = {
    type: {
        name: "Composite",
        className: "StorageProfileUpdate",
        modelProperties: {
            disks: {
                serializedName: "disks",
                type: {
                    name: "Sequence",
                    element: {
                        type: {
                            name: "Composite",
                            className: "VirtualDiskUpdate"
                        }
                    }
                }
            }
        }
    }
};
export const VirtualDiskUpdate = {
    type: {
        name: "Composite",
        className: "VirtualDiskUpdate",
        modelProperties: {
            name: {
                serializedName: "name",
                type: {
                    name: "String"
                }
            },
            diskId: {
                serializedName: "diskId",
                type: {
                    name: "String"
                }
            },
            diskSizeGB: {
                serializedName: "diskSizeGB",
                type: {
                    name: "Number"
                }
            },
            bus: {
                serializedName: "bus",
                type: {
                    name: "Number"
                }
            },
            lun: {
                serializedName: "lun",
                type: {
                    name: "Number"
                }
            },
            busType: {
                serializedName: "busType",
                type: {
                    name: "String"
                }
            },
            vhdType: {
                serializedName: "vhdType",
                type: {
                    name: "String"
                }
            },
            storageQoSPolicy: {
                serializedName: "storageQoSPolicy",
                type: {
                    name: "Composite",
                    className: "StorageQoSPolicyDetails"
                }
            }
        }
    }
};
export const NetworkProfileUpdate = {
    type: {
        name: "Composite",
        className: "NetworkProfileUpdate",
        modelProperties: {
            networkInterfaces: {
                serializedName: "networkInterfaces",
                type: {
                    name: "Sequence",
                    element: {
                        type: {
                            name: "Composite",
                            className: "NetworkInterfacesUpdate"
                        }
                    }
                }
            }
        }
    }
};
export const NetworkInterfacesUpdate = {
    type: {
        name: "Composite",
        className: "NetworkInterfacesUpdate",
        modelProperties: {
            name: {
                serializedName: "name",
                type: {
                    name: "String"
                }
            },
            macAddress: {
                serializedName: "macAddress",
                type: {
                    name: "String"
                }
            },
            virtualNetworkId: {
                serializedName: "virtualNetworkId",
                type: {
                    name: "String"
                }
            },
            ipv4AddressType: {
                serializedName: "ipv4AddressType",
                type: {
                    name: "String"
                }
            },
            ipv6AddressType: {
                serializedName: "ipv6AddressType",
                type: {
                    name: "String"
                }
            },
            macAddressType: {
                serializedName: "macAddressType",
                type: {
                    name: "String"
                }
            },
            nicId: {
                serializedName: "nicId",
                type: {
                    name: "String"
                }
            }
        }
    }
};
export const StopVirtualMachineOptions = {
    type: {
        name: "Composite",
        className: "StopVirtualMachineOptions",
        modelProperties: {
            skipShutdown: {
                defaultValue: false,
                serializedName: "skipShutdown",
                type: {
                    name: "Boolean"
                }
            }
        }
    }
};
export const VirtualMachineCreateCheckpoint = {
    type: {
        name: "Composite",
        className: "VirtualMachineCreateCheckpoint",
        modelProperties: {
            name: {
                serializedName: "name",
                type: {
                    name: "String"
                }
            },
            description: {
                serializedName: "description",
                type: {
                    name: "String"
                }
            }
        }
    }
};
export const VirtualMachineDeleteCheckpoint = {
    type: {
        name: "Composite",
        className: "VirtualMachineDeleteCheckpoint",
        modelProperties: {
            id: {
                serializedName: "id",
                type: {
                    name: "String"
                }
            }
        }
    }
};
export const VirtualMachineRestoreCheckpoint = {
    type: {
        name: "Composite",
        className: "VirtualMachineRestoreCheckpoint",
        modelProperties: {
            id: {
                serializedName: "id",
                type: {
                    name: "String"
                }
            }
        }
    }
};
export const VirtualMachineListResult = {
    type: {
        name: "Composite",
        className: "VirtualMachineListResult",
        modelProperties: {
            value: {
                serializedName: "value",
                type: {
                    name: "Sequence",
                    element: {
                        type: {
                            name: "Composite",
                            className: "VirtualMachine"
                        }
                    }
                }
            },
            nextLink: {
                serializedName: "nextLink",
                type: {
                    name: "String"
                }
            }
        }
    }
};
export const VirtualMachineTemplate = {
    type: {
        name: "Composite",
        className: "VirtualMachineTemplate",
        modelProperties: {
            id: {
                serializedName: "id",
                readOnly: true,
                type: {
                    name: "String"
                }
            },
            name: {
                serializedName: "name",
                readOnly: true,
                type: {
                    name: "String"
                }
            },
            type: {
                serializedName: "type",
                readOnly: true,
                type: {
                    name: "String"
                }
            },
            location: {
                serializedName: "location",
                required: true,
                type: {
                    name: "String"
                }
            },
            tags: {
                serializedName: "tags",
                type: {
                    name: "Dictionary",
                    value: { type: { name: "String" } }
                }
            },
            systemData: {
                serializedName: "systemData",
                type: {
                    name: "Composite",
                    className: "SystemData"
                }
            },
            extendedLocation: {
                serializedName: "extendedLocation",
                type: {
                    name: "Composite",
                    className: "ExtendedLocation"
                }
            },
            inventoryItemId: {
                serializedName: "properties.inventoryItemId",
                type: {
                    name: "String"
                }
            },
            uuid: {
                constraints: {
                    MinLength: 1
                },
                serializedName: "properties.uuid",
                type: {
                    name: "String"
                }
            },
            vmmServerId: {
                serializedName: "properties.vmmServerId",
                type: {
                    name: "String"
                }
            },
            osType: {
                serializedName: "properties.osType",
                readOnly: true,
                type: {
                    name: "String"
                }
            },
            osName: {
                serializedName: "properties.osName",
                readOnly: true,
                type: {
                    name: "String"
                }
            },
            computerName: {
                serializedName: "properties.computerName",
                readOnly: true,
                type: {
                    name: "String"
                }
            },
            memoryMB: {
                serializedName: "properties.memoryMB",
                readOnly: true,
                type: {
                    name: "Number"
                }
            },
            cpuCount: {
                serializedName: "properties.cpuCount",
                readOnly: true,
                type: {
                    name: "Number"
                }
            },
            limitCpuForMigration: {
                serializedName: "properties.limitCpuForMigration",
                readOnly: true,
                type: {
                    name: "String"
                }
            },
            dynamicMemoryEnabled: {
                serializedName: "properties.dynamicMemoryEnabled",
                readOnly: true,
                type: {
                    name: "String"
                }
            },
            isCustomizable: {
                serializedName: "properties.isCustomizable",
                readOnly: true,
                type: {
                    name: "String"
                }
            },
            dynamicMemoryMaxMB: {
                serializedName: "properties.dynamicMemoryMaxMB",
                readOnly: true,
                type: {
                    name: "Number"
                }
            },
            dynamicMemoryMinMB: {
                serializedName: "properties.dynamicMemoryMinMB",
                readOnly: true,
                type: {
                    name: "Number"
                }
            },
            isHighlyAvailable: {
                serializedName: "properties.isHighlyAvailable",
                readOnly: true,
                type: {
                    name: "String"
                }
            },
            generation: {
                serializedName: "properties.generation",
                readOnly: true,
                type: {
                    name: "Number"
                }
            },
            networkInterfaces: {
                serializedName: "properties.networkInterfaces",
                readOnly: true,
                type: {
                    name: "Sequence",
                    element: {
                        type: {
                            name: "Composite",
                            className: "NetworkInterfaces"
                        }
                    }
                }
            },
            disks: {
                serializedName: "properties.disks",
                readOnly: true,
                type: {
                    name: "Sequence",
                    element: {
                        type: {
                            name: "Composite",
                            className: "VirtualDisk"
                        }
                    }
                }
            },
            provisioningState: {
                serializedName: "properties.provisioningState",
                readOnly: true,
                type: {
                    name: "String"
                }
            }
        }
    }
};
export const VirtualMachineTemplateListResult = {
    type: {
        name: "Composite",
        className: "VirtualMachineTemplateListResult",
        modelProperties: {
            value: {
                serializedName: "value",
                type: {
                    name: "Sequence",
                    element: {
                        type: {
                            name: "Composite",
                            className: "VirtualMachineTemplate"
                        }
                    }
                }
            },
            nextLink: {
                serializedName: "nextLink",
                type: {
                    name: "String"
                }
            }
        }
    }
};
export const AvailabilitySet = {
    type: {
        name: "Composite",
        className: "AvailabilitySet",
        modelProperties: {
            id: {
                serializedName: "id",
                readOnly: true,
                type: {
                    name: "String"
                }
            },
            name: {
                serializedName: "name",
                readOnly: true,
                type: {
                    name: "String"
                }
            },
            type: {
                serializedName: "type",
                readOnly: true,
                type: {
                    name: "String"
                }
            },
            location: {
                serializedName: "location",
                type: {
                    name: "String"
                }
            },
            tags: {
                serializedName: "tags",
                type: {
                    name: "Dictionary",
                    value: { type: { name: "String" } }
                }
            },
            systemData: {
                serializedName: "systemData",
                type: {
                    name: "Composite",
                    className: "SystemData"
                }
            },
            extendedLocation: {
                serializedName: "extendedLocation",
                type: {
                    name: "Composite",
                    className: "ExtendedLocation"
                }
            },
            availabilitySetName: {
                constraints: {
                    MinLength: 1
                },
                serializedName: "properties.availabilitySetName",
                type: {
                    name: "String"
                }
            },
            vmmServerId: {
                serializedName: "properties.vmmServerId",
                type: {
                    name: "String"
                }
            },
            provisioningState: {
                serializedName: "properties.provisioningState",
                readOnly: true,
                type: {
                    name: "String"
                }
            }
        }
    }
};
export const AvailabilitySetListResult = {
    type: {
        name: "Composite",
        className: "AvailabilitySetListResult",
        modelProperties: {
            value: {
                serializedName: "value",
                type: {
                    name: "Sequence",
                    element: {
                        type: {
                            name: "Composite",
                            className: "AvailabilitySet"
                        }
                    }
                }
            },
            nextLink: {
                serializedName: "nextLink",
                type: {
                    name: "String"
                }
            }
        }
    }
};
export const InventoryItemProperties = {
    type: {
        name: "Composite",
        className: "InventoryItemProperties",
        uberParent: "InventoryItemProperties",
        polymorphicDiscriminator: {
            serializedName: "inventoryType",
            clientName: "inventoryType"
        },
        modelProperties: {
            inventoryType: {
                serializedName: "inventoryType",
                required: true,
                type: {
                    name: "String"
                }
            },
            managedResourceId: {
                serializedName: "managedResourceId",
                readOnly: true,
                type: {
                    name: "String"
                }
            },
            uuid: {
                serializedName: "uuid",
                readOnly: true,
                type: {
                    name: "String"
                }
            },
            inventoryItemName: {
                serializedName: "inventoryItemName",
                readOnly: true,
                type: {
                    name: "String"
                }
            },
            provisioningState: {
                serializedName: "provisioningState",
                readOnly: true,
                type: {
                    name: "String"
                }
            }
        }
    }
};
export const Resource = {
    type: {
        name: "Composite",
        className: "Resource",
        modelProperties: {
            id: {
                serializedName: "id",
                readOnly: true,
                type: {
                    name: "String"
                }
            },
            name: {
                serializedName: "name",
                readOnly: true,
                type: {
                    name: "String"
                }
            },
            type: {
                serializedName: "type",
                readOnly: true,
                type: {
                    name: "String"
                }
            }
        }
    }
};
export const InventoryItemsList = {
    type: {
        name: "Composite",
        className: "InventoryItemsList",
        modelProperties: {
            nextLink: {
                serializedName: "nextLink",
                type: {
                    name: "String"
                }
            },
            value: {
                serializedName: "value",
                required: true,
                type: {
                    name: "Sequence",
                    element: {
                        type: {
                            name: "Composite",
                            className: "InventoryItem"
                        }
                    }
                }
            }
        }
    }
};
export const InventoryItemDetails = {
    type: {
        name: "Composite",
        className: "InventoryItemDetails",
        modelProperties: {
            inventoryItemId: {
                serializedName: "inventoryItemId",
                type: {
                    name: "String"
                }
            },
            inventoryItemName: {
                serializedName: "inventoryItemName",
                type: {
                    name: "String"
                }
            }
        }
    }
};
export const CloudInventoryItem = {
    serializedName: "Cloud",
    type: {
        name: "Composite",
        className: "CloudInventoryItem",
        uberParent: "InventoryItemProperties",
        polymorphicDiscriminator: InventoryItemProperties.type.polymorphicDiscriminator,
        modelProperties: Object.assign({}, InventoryItemProperties.type.modelProperties)
    }
};
export const VirtualNetworkInventoryItem = {
    serializedName: "VirtualNetwork",
    type: {
        name: "Composite",
        className: "VirtualNetworkInventoryItem",
        uberParent: "InventoryItemProperties",
        polymorphicDiscriminator: InventoryItemProperties.type.polymorphicDiscriminator,
        modelProperties: Object.assign({}, InventoryItemProperties.type.modelProperties)
    }
};
export const VirtualMachineTemplateInventoryItem = {
    serializedName: "VirtualMachineTemplate",
    type: {
        name: "Composite",
        className: "VirtualMachineTemplateInventoryItem",
        uberParent: "InventoryItemProperties",
        polymorphicDiscriminator: InventoryItemProperties.type.polymorphicDiscriminator,
        modelProperties: Object.assign(Object.assign({}, InventoryItemProperties.type.modelProperties), { cpuCount: {
                serializedName: "cpuCount",
                readOnly: true,
                type: {
                    name: "Number"
                }
            }, memoryMB: {
                serializedName: "memoryMB",
                readOnly: true,
                type: {
                    name: "Number"
                }
            }, osType: {
                serializedName: "osType",
                readOnly: true,
                type: {
                    name: "String"
                }
            }, osName: {
                serializedName: "osName",
                readOnly: true,
                type: {
                    name: "String"
                }
            } })
    }
};
export const VirtualMachineInventoryItem = {
    serializedName: "VirtualMachine",
    type: {
        name: "Composite",
        className: "VirtualMachineInventoryItem",
        uberParent: "InventoryItemProperties",
        polymorphicDiscriminator: InventoryItemProperties.type.polymorphicDiscriminator,
        modelProperties: Object.assign(Object.assign({}, InventoryItemProperties.type.modelProperties), { osType: {
                serializedName: "osType",
                readOnly: true,
                type: {
                    name: "String"
                }
            }, osName: {
                serializedName: "osName",
                readOnly: true,
                type: {
                    name: "String"
                }
            }, powerState: {
                serializedName: "powerState",
                readOnly: true,
                type: {
                    name: "String"
                }
            }, ipAddresses: {
                serializedName: "ipAddresses",
                type: {
                    name: "Sequence",
                    element: {
                        type: {
                            name: "String"
                        }
                    }
                }
            }, cloud: {
                serializedName: "cloud",
                type: {
                    name: "Composite",
                    className: "InventoryItemDetails"
                }
            } })
    }
};
export const ProxyResource = {
    type: {
        name: "Composite",
        className: "ProxyResource",
        modelProperties: Object.assign({}, Resource.type.modelProperties)
    }
};
export const InventoryItem = {
    type: {
        name: "Composite",
        className: "InventoryItem",
        modelProperties: Object.assign(Object.assign({}, ProxyResource.type.modelProperties), { systemData: {
                serializedName: "systemData",
                type: {
                    name: "Composite",
                    className: "SystemData"
                }
            }, kind: {
                serializedName: "kind",
                type: {
                    name: "String"
                }
            }, inventoryType: {
                serializedName: "properties.inventoryType",
                required: true,
                type: {
                    name: "String"
                }
            }, managedResourceId: {
                serializedName: "properties.managedResourceId",
                readOnly: true,
                type: {
                    name: "String"
                }
            }, uuid: {
                serializedName: "properties.uuid",
                readOnly: true,
                type: {
                    name: "String"
                }
            }, inventoryItemName: {
                serializedName: "properties.inventoryItemName",
                readOnly: true,
                type: {
                    name: "String"
                }
            }, provisioningState: {
                serializedName: "properties.provisioningState",
                readOnly: true,
                type: {
                    name: "String"
                }
            } })
    }
};
export const AvailabilitySetsCreateOrUpdateHeaders = {
    type: {
        name: "Composite",
        className: "AvailabilitySetsCreateOrUpdateHeaders",
        modelProperties: {
            azureAsyncOperation: {
                serializedName: "azure-asyncoperation",
                type: {
                    name: "String"
                }
            }
        }
    }
};
export const AvailabilitySetsDeleteHeaders = {
    type: {
        name: "Composite",
        className: "AvailabilitySetsDeleteHeaders",
        modelProperties: {
            azureAsyncOperation: {
                serializedName: "azure-asyncoperation",
                type: {
                    name: "String"
                }
            }
        }
    }
};
export let discriminators = {
    InventoryItemProperties: InventoryItemProperties,
    "InventoryItemProperties.Cloud": CloudInventoryItem,
    "InventoryItemProperties.VirtualNetwork": VirtualNetworkInventoryItem,
    "InventoryItemProperties.VirtualMachineTemplate": VirtualMachineTemplateInventoryItem,
    "InventoryItemProperties.VirtualMachine": VirtualMachineInventoryItem
};
