// @azure/arm-azurestackhci/dist-esm/src/models/mappers.js
/*
 * Copyright (c) Microsoft Corporation.
 * Licensed under the MIT License.
 *
 * Code generated by Microsoft (R) AutoRest Code Generator.
 * Changes may cause incorrect behavior and will be lost if the code is regenerated.
 */
export const ArcSettingList = {
    type: {
        name: "Composite",
        className: "ArcSettingList",
        modelProperties: {
            value: {
                serializedName: "value",
                readOnly: true,
                type: {
                    name: "Sequence",
                    element: {
                        type: {
                            name: "Composite",
                            className: "ArcSetting"
                        }
                    }
                }
            },
            nextLink: {
                serializedName: "nextLink",
                readOnly: true,
                type: {
                    name: "String"
                }
            }
        }
    }
};
export const PerNodeState = {
    type: {
        name: "Composite",
        className: "PerNodeState",
        modelProperties: {
            name: {
                serializedName: "name",
                readOnly: true,
                type: {
                    name: "String"
                }
            },
            arcInstance: {
                serializedName: "arcInstance",
                readOnly: true,
                type: {
                    name: "String"
                }
            },
            state: {
                serializedName: "state",
                readOnly: true,
                type: {
                    name: "String"
                }
            }
        }
    }
};
export const Resource = {
    type: {
        name: "Composite",
        className: "Resource",
        modelProperties: {
            id: {
                serializedName: "id",
                readOnly: true,
                type: {
                    name: "String"
                }
            },
            name: {
                serializedName: "name",
                readOnly: true,
                type: {
                    name: "String"
                }
            },
            type: {
                serializedName: "type",
                readOnly: true,
                type: {
                    name: "String"
                }
            }
        }
    }
};
export const ErrorResponse = {
    type: {
        name: "Composite",
        className: "ErrorResponse",
        modelProperties: {
            error: {
                serializedName: "error",
                type: {
                    name: "Composite",
                    className: "ErrorDetail"
                }
            }
        }
    }
};
export const ErrorDetail = {
    type: {
        name: "Composite",
        className: "ErrorDetail",
        modelProperties: {
            code: {
                serializedName: "code",
                readOnly: true,
                type: {
                    name: "String"
                }
            },
            message: {
                serializedName: "message",
                readOnly: true,
                type: {
                    name: "String"
                }
            },
            target: {
                serializedName: "target",
                readOnly: true,
                type: {
                    name: "String"
                }
            },
            details: {
                serializedName: "details",
                readOnly: true,
                type: {
                    name: "Sequence",
                    element: {
                        type: {
                            name: "Composite",
                            className: "ErrorDetail"
                        }
                    }
                }
            },
            additionalInfo: {
                serializedName: "additionalInfo",
                readOnly: true,
                type: {
                    name: "Sequence",
                    element: {
                        type: {
                            name: "Composite",
                            className: "ErrorAdditionalInfo"
                        }
                    }
                }
            }
        }
    }
};
export const ErrorAdditionalInfo = {
    type: {
        name: "Composite",
        className: "ErrorAdditionalInfo",
        modelProperties: {
            type: {
                serializedName: "type",
                readOnly: true,
                type: {
                    name: "String"
                }
            },
            info: {
                serializedName: "info",
                readOnly: true,
                type: {
                    name: "Dictionary",
                    value: { type: { name: "any" } }
                }
            }
        }
    }
};
export const ArcSettingsPatch = {
    type: {
        name: "Composite",
        className: "ArcSettingsPatch",
        modelProperties: {
            tags: {
                serializedName: "tags",
                type: {
                    name: "Dictionary",
                    value: { type: { name: "String" } }
                }
            },
            connectivityProperties: {
                serializedName: "properties.connectivityProperties",
                type: {
                    name: "Dictionary",
                    value: { type: { name: "any" } }
                }
            }
        }
    }
};
export const PasswordCredential = {
    type: {
        name: "Composite",
        className: "PasswordCredential",
        modelProperties: {
            secretText: {
                serializedName: "secretText",
                type: {
                    name: "String"
                }
            },
            keyId: {
                serializedName: "keyId",
                type: {
                    name: "String"
                }
            },
            startDateTime: {
                serializedName: "startDateTime",
                type: {
                    name: "DateTime"
                }
            },
            endDateTime: {
                serializedName: "endDateTime",
                type: {
                    name: "DateTime"
                }
            }
        }
    }
};
export const ArcIdentityResponse = {
    type: {
        name: "Composite",
        className: "ArcIdentityResponse",
        modelProperties: {
            arcApplicationClientId: {
                serializedName: "properties.arcApplicationClientId",
                type: {
                    name: "String"
                }
            },
            arcApplicationTenantId: {
                serializedName: "properties.arcApplicationTenantId",
                type: {
                    name: "String"
                }
            },
            arcServicePrincipalObjectId: {
                serializedName: "properties.arcServicePrincipalObjectId",
                type: {
                    name: "String"
                }
            },
            arcApplicationObjectId: {
                serializedName: "properties.arcApplicationObjectId",
                type: {
                    name: "String"
                }
            }
        }
    }
};
export const ClusterList = {
    type: {
        name: "Composite",
        className: "ClusterList",
        modelProperties: {
            value: {
                serializedName: "value",
                type: {
                    name: "Sequence",
                    element: {
                        type: {
                            name: "Composite",
                            className: "Cluster"
                        }
                    }
                }
            },
            nextLink: {
                serializedName: "nextLink",
                readOnly: true,
                type: {
                    name: "String"
                }
            }
        }
    }
};
export const ClusterDesiredProperties = {
    type: {
        name: "Composite",
        className: "ClusterDesiredProperties",
        modelProperties: {
            windowsServerSubscription: {
                serializedName: "windowsServerSubscription",
                type: {
                    name: "String"
                }
            },
            diagnosticLevel: {
                serializedName: "diagnosticLevel",
                type: {
                    name: "String"
                }
            }
        }
    }
};
export const ClusterReportedProperties = {
    type: {
        name: "Composite",
        className: "ClusterReportedProperties",
        modelProperties: {
            clusterName: {
                serializedName: "clusterName",
                readOnly: true,
                type: {
                    name: "String"
                }
            },
            clusterId: {
                serializedName: "clusterId",
                readOnly: true,
                type: {
                    name: "String"
                }
            },
            clusterVersion: {
                serializedName: "clusterVersion",
                readOnly: true,
                type: {
                    name: "String"
                }
            },
            nodes: {
                serializedName: "nodes",
                readOnly: true,
                type: {
                    name: "Sequence",
                    element: {
                        type: {
                            name: "Composite",
                            className: "ClusterNode"
                        }
                    }
                }
            },
            lastUpdated: {
                serializedName: "lastUpdated",
                readOnly: true,
                type: {
                    name: "DateTime"
                }
            },
            imdsAttestation: {
                serializedName: "imdsAttestation",
                readOnly: true,
                type: {
                    name: "String"
                }
            },
            diagnosticLevel: {
                serializedName: "diagnosticLevel",
                type: {
                    name: "String"
                }
            }
        }
    }
};
export const ClusterNode = {
    type: {
        name: "Composite",
        className: "ClusterNode",
        modelProperties: {
            name: {
                serializedName: "name",
                readOnly: true,
                type: {
                    name: "String"
                }
            },
            id: {
                serializedName: "id",
                readOnly: true,
                type: {
                    name: "Number"
                }
            },
            windowsServerSubscription: {
                serializedName: "windowsServerSubscription",
                readOnly: true,
                type: {
                    name: "String"
                }
            },
            manufacturer: {
                serializedName: "manufacturer",
                readOnly: true,
                type: {
                    name: "String"
                }
            },
            model: {
                serializedName: "model",
                readOnly: true,
                type: {
                    name: "String"
                }
            },
            osName: {
                serializedName: "osName",
                readOnly: true,
                type: {
                    name: "String"
                }
            },
            osVersion: {
                serializedName: "osVersion",
                readOnly: true,
                type: {
                    name: "String"
                }
            },
            serialNumber: {
                serializedName: "serialNumber",
                readOnly: true,
                type: {
                    name: "String"
                }
            },
            coreCount: {
                serializedName: "coreCount",
                readOnly: true,
                type: {
                    name: "Number"
                }
            },
            memoryInGiB: {
                serializedName: "memoryInGiB",
                readOnly: true,
                type: {
                    name: "Number"
                }
            }
        }
    }
};
export const ClusterPatch = {
    type: {
        name: "Composite",
        className: "ClusterPatch",
        modelProperties: {
            tags: {
                serializedName: "tags",
                type: {
                    name: "Dictionary",
                    value: { type: { name: "String" } }
                }
            },
            cloudManagementEndpoint: {
                serializedName: "properties.cloudManagementEndpoint",
                type: {
                    name: "String"
                }
            },
            aadClientId: {
                serializedName: "properties.aadClientId",
                type: {
                    name: "String"
                }
            },
            aadTenantId: {
                serializedName: "properties.aadTenantId",
                type: {
                    name: "String"
                }
            },
            desiredProperties: {
                serializedName: "properties.desiredProperties",
                type: {
                    name: "Composite",
                    className: "ClusterDesiredProperties"
                }
            }
        }
    }
};
export const UploadCertificateRequest = {
    type: {
        name: "Composite",
        className: "UploadCertificateRequest",
        modelProperties: {
            properties: {
                serializedName: "properties",
                type: {
                    name: "Composite",
                    className: "RawCertificateData"
                }
            }
        }
    }
};
export const RawCertificateData = {
    type: {
        name: "Composite",
        className: "RawCertificateData",
        modelProperties: {
            certificates: {
                serializedName: "certificates",
                type: {
                    name: "Sequence",
                    element: {
                        type: {
                            name: "String"
                        }
                    }
                }
            }
        }
    }
};
export const ClusterIdentityResponse = {
    type: {
        name: "Composite",
        className: "ClusterIdentityResponse",
        modelProperties: {
            aadClientId: {
                serializedName: "properties.aadClientId",
                type: {
                    name: "String"
                }
            },
            aadTenantId: {
                serializedName: "properties.aadTenantId",
                type: {
                    name: "String"
                }
            },
            aadServicePrincipalObjectId: {
                serializedName: "properties.aadServicePrincipalObjectId",
                type: {
                    name: "String"
                }
            },
            aadApplicationObjectId: {
                serializedName: "properties.aadApplicationObjectId",
                type: {
                    name: "String"
                }
            }
        }
    }
};
export const ExtensionList = {
    type: {
        name: "Composite",
        className: "ExtensionList",
        modelProperties: {
            value: {
                serializedName: "value",
                readOnly: true,
                type: {
                    name: "Sequence",
                    element: {
                        type: {
                            name: "Composite",
                            className: "Extension"
                        }
                    }
                }
            },
            nextLink: {
                serializedName: "nextLink",
                readOnly: true,
                type: {
                    name: "String"
                }
            }
        }
    }
};
export const PerNodeExtensionState = {
    type: {
        name: "Composite",
        className: "PerNodeExtensionState",
        modelProperties: {
            name: {
                serializedName: "name",
                readOnly: true,
                type: {
                    name: "String"
                }
            },
            extension: {
                serializedName: "extension",
                readOnly: true,
                type: {
                    name: "String"
                }
            },
            state: {
                serializedName: "state",
                readOnly: true,
                type: {
                    name: "String"
                }
            }
        }
    }
};
export const OperationListResult = {
    type: {
        name: "Composite",
        className: "OperationListResult",
        modelProperties: {
            value: {
                serializedName: "value",
                readOnly: true,
                type: {
                    name: "Sequence",
                    element: {
                        type: {
                            name: "Composite",
                            className: "Operation"
                        }
                    }
                }
            },
            nextLink: {
                serializedName: "nextLink",
                readOnly: true,
                type: {
                    name: "String"
                }
            }
        }
    }
};
export const Operation = {
    type: {
        name: "Composite",
        className: "Operation",
        modelProperties: {
            name: {
                serializedName: "name",
                readOnly: true,
                type: {
                    name: "String"
                }
            },
            isDataAction: {
                serializedName: "isDataAction",
                readOnly: true,
                type: {
                    name: "Boolean"
                }
            },
            display: {
                serializedName: "display",
                type: {
                    name: "Composite",
                    className: "OperationDisplay"
                }
            },
            origin: {
                serializedName: "origin",
                readOnly: true,
                type: {
                    name: "String"
                }
            },
            actionType: {
                serializedName: "actionType",
                readOnly: true,
                type: {
                    name: "String"
                }
            }
        }
    }
};
export const OperationDisplay = {
    type: {
        name: "Composite",
        className: "OperationDisplay",
        modelProperties: {
            provider: {
                serializedName: "provider",
                readOnly: true,
                type: {
                    name: "String"
                }
            },
            resource: {
                serializedName: "resource",
                readOnly: true,
                type: {
                    name: "String"
                }
            },
            operation: {
                serializedName: "operation",
                readOnly: true,
                type: {
                    name: "String"
                }
            },
            description: {
                serializedName: "description",
                readOnly: true,
                type: {
                    name: "String"
                }
            }
        }
    }
};
export const ArcConnectivityProperties = {
    type: {
        name: "Composite",
        className: "ArcConnectivityProperties",
        modelProperties: {
            enabled: {
                serializedName: "enabled",
                type: {
                    name: "Boolean"
                }
            }
        }
    }
};
export const ProxyResource = {
    type: {
        name: "Composite",
        className: "ProxyResource",
        modelProperties: Object.assign({}, Resource.type.modelProperties)
    }
};
export const TrackedResource = {
    type: {
        name: "Composite",
        className: "TrackedResource",
        modelProperties: Object.assign(Object.assign({}, Resource.type.modelProperties), { tags: {
                serializedName: "tags",
                type: {
                    name: "Dictionary",
                    value: { type: { name: "String" } }
                }
            }, location: {
                serializedName: "location",
                required: true,
                type: {
                    name: "String"
                }
            } })
    }
};
export const ArcSetting = {
    type: {
        name: "Composite",
        className: "ArcSetting",
        modelProperties: Object.assign(Object.assign({}, ProxyResource.type.modelProperties), { provisioningState: {
                serializedName: "properties.provisioningState",
                readOnly: true,
                type: {
                    name: "String"
                }
            }, arcInstanceResourceGroup: {
                serializedName: "properties.arcInstanceResourceGroup",
                type: {
                    name: "String"
                }
            }, arcApplicationClientId: {
                serializedName: "properties.arcApplicationClientId",
                type: {
                    name: "String"
                }
            }, arcApplicationTenantId: {
                serializedName: "properties.arcApplicationTenantId",
                type: {
                    name: "String"
                }
            }, arcServicePrincipalObjectId: {
                serializedName: "properties.arcServicePrincipalObjectId",
                type: {
                    name: "String"
                }
            }, arcApplicationObjectId: {
                serializedName: "properties.arcApplicationObjectId",
                type: {
                    name: "String"
                }
            }, aggregateState: {
                serializedName: "properties.aggregateState",
                readOnly: true,
                type: {
                    name: "String"
                }
            }, perNodeDetails: {
                serializedName: "properties.perNodeDetails",
                readOnly: true,
                type: {
                    name: "Sequence",
                    element: {
                        type: {
                            name: "Composite",
                            className: "PerNodeState"
                        }
                    }
                }
            }, connectivityProperties: {
                serializedName: "properties.connectivityProperties",
                type: {
                    name: "Dictionary",
                    value: { type: { name: "any" } }
                }
            }, createdBy: {
                serializedName: "systemData.createdBy",
                type: {
                    name: "String"
                }
            }, createdByType: {
                serializedName: "systemData.createdByType",
                type: {
                    name: "String"
                }
            }, createdAt: {
                serializedName: "systemData.createdAt",
                type: {
                    name: "DateTime"
                }
            }, lastModifiedBy: {
                serializedName: "systemData.lastModifiedBy",
                type: {
                    name: "String"
                }
            }, lastModifiedByType: {
                serializedName: "systemData.lastModifiedByType",
                type: {
                    name: "String"
                }
            }, lastModifiedAt: {
                serializedName: "systemData.lastModifiedAt",
                type: {
                    name: "DateTime"
                }
            } })
    }
};
export const Extension = {
    type: {
        name: "Composite",
        className: "Extension",
        modelProperties: Object.assign(Object.assign({}, ProxyResource.type.modelProperties), { provisioningState: {
                serializedName: "properties.provisioningState",
                readOnly: true,
                type: {
                    name: "String"
                }
            }, aggregateState: {
                serializedName: "properties.aggregateState",
                readOnly: true,
                type: {
                    name: "String"
                }
            }, perNodeExtensionDetails: {
                serializedName: "properties.perNodeExtensionDetails",
                readOnly: true,
                type: {
                    name: "Sequence",
                    element: {
                        type: {
                            name: "Composite",
                            className: "PerNodeExtensionState"
                        }
                    }
                }
            }, forceUpdateTag: {
                serializedName: "properties.extensionParameters.forceUpdateTag",
                type: {
                    name: "String"
                }
            }, publisher: {
                serializedName: "properties.extensionParameters.publisher",
                type: {
                    name: "String"
                }
            }, typePropertiesExtensionParametersType: {
                serializedName: "properties.extensionParameters.type",
                type: {
                    name: "String"
                }
            }, typeHandlerVersion: {
                serializedName: "properties.extensionParameters.typeHandlerVersion",
                type: {
                    name: "String"
                }
            }, autoUpgradeMinorVersion: {
                serializedName: "properties.extensionParameters.autoUpgradeMinorVersion",
                type: {
                    name: "Boolean"
                }
            }, settings: {
                serializedName: "properties.extensionParameters.settings",
                type: {
                    name: "Dictionary",
                    value: { type: { name: "any" } }
                }
            }, protectedSettings: {
                serializedName: "properties.extensionParameters.protectedSettings",
                type: {
                    name: "Dictionary",
                    value: { type: { name: "any" } }
                }
            }, createdBy: {
                serializedName: "systemData.createdBy",
                type: {
                    name: "String"
                }
            }, createdByType: {
                serializedName: "systemData.createdByType",
                type: {
                    name: "String"
                }
            }, createdAt: {
                serializedName: "systemData.createdAt",
                type: {
                    name: "DateTime"
                }
            }, lastModifiedBy: {
                serializedName: "systemData.lastModifiedBy",
                type: {
                    name: "String"
                }
            }, lastModifiedByType: {
                serializedName: "systemData.lastModifiedByType",
                type: {
                    name: "String"
                }
            }, lastModifiedAt: {
                serializedName: "systemData.lastModifiedAt",
                type: {
                    name: "DateTime"
                }
            } })
    }
};
export const Cluster = {
    type: {
        name: "Composite",
        className: "Cluster",
        modelProperties: Object.assign(Object.assign({}, TrackedResource.type.modelProperties), { provisioningState: {
                serializedName: "properties.provisioningState",
                readOnly: true,
                type: {
                    name: "String"
                }
            }, status: {
                serializedName: "properties.status",
                readOnly: true,
                type: {
                    name: "String"
                }
            }, cloudId: {
                serializedName: "properties.cloudId",
                readOnly: true,
                type: {
                    name: "String"
                }
            }, cloudManagementEndpoint: {
                serializedName: "properties.cloudManagementEndpoint",
                type: {
                    name: "String"
                }
            }, aadClientId: {
                serializedName: "properties.aadClientId",
                type: {
                    name: "String"
                }
            }, aadTenantId: {
                serializedName: "properties.aadTenantId",
                type: {
                    name: "String"
                }
            }, aadApplicationObjectId: {
                serializedName: "properties.aadApplicationObjectId",
                type: {
                    name: "String"
                }
            }, aadServicePrincipalObjectId: {
                serializedName: "properties.aadServicePrincipalObjectId",
                type: {
                    name: "String"
                }
            }, desiredProperties: {
                serializedName: "properties.desiredProperties",
                type: {
                    name: "Composite",
                    className: "ClusterDesiredProperties"
                }
            }, reportedProperties: {
                serializedName: "properties.reportedProperties",
                type: {
                    name: "Composite",
                    className: "ClusterReportedProperties"
                }
            }, trialDaysRemaining: {
                serializedName: "properties.trialDaysRemaining",
                readOnly: true,
                type: {
                    name: "Number"
                }
            }, billingModel: {
                serializedName: "properties.billingModel",
                readOnly: true,
                type: {
                    name: "String"
                }
            }, registrationTimestamp: {
                serializedName: "properties.registrationTimestamp",
                readOnly: true,
                type: {
                    name: "DateTime"
                }
            }, lastSyncTimestamp: {
                serializedName: "properties.lastSyncTimestamp",
                readOnly: true,
                type: {
                    name: "DateTime"
                }
            }, lastBillingTimestamp: {
                serializedName: "properties.lastBillingTimestamp",
                readOnly: true,
                type: {
                    name: "DateTime"
                }
            }, serviceEndpoint: {
                serializedName: "properties.serviceEndpoint",
                readOnly: true,
                type: {
                    name: "String"
                }
            }, createdBy: {
                serializedName: "systemData.createdBy",
                type: {
                    name: "String"
                }
            }, createdByType: {
                serializedName: "systemData.createdByType",
                type: {
                    name: "String"
                }
            }, createdAt: {
                serializedName: "systemData.createdAt",
                type: {
                    name: "DateTime"
                }
            }, lastModifiedBy: {
                serializedName: "systemData.lastModifiedBy",
                type: {
                    name: "String"
                }
            }, lastModifiedByType: {
                serializedName: "systemData.lastModifiedByType",
                type: {
                    name: "String"
                }
            }, lastModifiedAt: {
                serializedName: "systemData.lastModifiedAt",
                type: {
                    name: "DateTime"
                }
            } })
    }
};
