// @azure/arm-healthcareapis/dist-esm/src/models/index.js
/*
 * Copyright (c) Microsoft Corporation.
 * Licensed under the MIT License.
 *
 * Code generated by Microsoft (R) AutoRest Code Generator.
 * Changes may cause incorrect behavior and will be lost if the code is regenerated.
 */
/** Known values of {@link ProvisioningState} that the service accepts. */
export var KnownProvisioningState;
(function (KnownProvisioningState) {
    /** Deleting */
    KnownProvisioningState["Deleting"] = "Deleting";
    /** Succeeded */
    KnownProvisioningState["Succeeded"] = "Succeeded";
    /** Creating */
    KnownProvisioningState["Creating"] = "Creating";
    /** Accepted */
    KnownProvisioningState["Accepted"] = "Accepted";
    /** Verifying */
    KnownProvisioningState["Verifying"] = "Verifying";
    /** Updating */
    KnownProvisioningState["Updating"] = "Updating";
    /** Failed */
    KnownProvisioningState["Failed"] = "Failed";
    /** Canceled */
    KnownProvisioningState["Canceled"] = "Canceled";
    /** Deprovisioned */
    KnownProvisioningState["Deprovisioned"] = "Deprovisioned";
    /** Moving */
    KnownProvisioningState["Moving"] = "Moving";
    /** Suspended */
    KnownProvisioningState["Suspended"] = "Suspended";
    /** Warned */
    KnownProvisioningState["Warned"] = "Warned";
    /** SystemMaintenance */
    KnownProvisioningState["SystemMaintenance"] = "SystemMaintenance";
})(KnownProvisioningState || (KnownProvisioningState = {}));
/** Known values of {@link PrivateEndpointServiceConnectionStatus} that the service accepts. */
export var KnownPrivateEndpointServiceConnectionStatus;
(function (KnownPrivateEndpointServiceConnectionStatus) {
    /** Pending */
    KnownPrivateEndpointServiceConnectionStatus["Pending"] = "Pending";
    /** Approved */
    KnownPrivateEndpointServiceConnectionStatus["Approved"] = "Approved";
    /** Rejected */
    KnownPrivateEndpointServiceConnectionStatus["Rejected"] = "Rejected";
})(KnownPrivateEndpointServiceConnectionStatus || (KnownPrivateEndpointServiceConnectionStatus = {}));
/** Known values of {@link PrivateEndpointConnectionProvisioningState} that the service accepts. */
export var KnownPrivateEndpointConnectionProvisioningState;
(function (KnownPrivateEndpointConnectionProvisioningState) {
    /** Succeeded */
    KnownPrivateEndpointConnectionProvisioningState["Succeeded"] = "Succeeded";
    /** Creating */
    KnownPrivateEndpointConnectionProvisioningState["Creating"] = "Creating";
    /** Deleting */
    KnownPrivateEndpointConnectionProvisioningState["Deleting"] = "Deleting";
    /** Failed */
    KnownPrivateEndpointConnectionProvisioningState["Failed"] = "Failed";
})(KnownPrivateEndpointConnectionProvisioningState || (KnownPrivateEndpointConnectionProvisioningState = {}));
/** Known values of {@link PublicNetworkAccess} that the service accepts. */
export var KnownPublicNetworkAccess;
(function (KnownPublicNetworkAccess) {
    /** Enabled */
    KnownPublicNetworkAccess["Enabled"] = "Enabled";
    /** Disabled */
    KnownPublicNetworkAccess["Disabled"] = "Disabled";
})(KnownPublicNetworkAccess || (KnownPublicNetworkAccess = {}));
/** Known values of {@link CreatedByType} that the service accepts. */
export var KnownCreatedByType;
(function (KnownCreatedByType) {
    /** User */
    KnownCreatedByType["User"] = "User";
    /** Application */
    KnownCreatedByType["Application"] = "Application";
    /** ManagedIdentity */
    KnownCreatedByType["ManagedIdentity"] = "ManagedIdentity";
    /** Key */
    KnownCreatedByType["Key"] = "Key";
})(KnownCreatedByType || (KnownCreatedByType = {}));
/** Known values of {@link ManagedServiceIdentityType} that the service accepts. */
export var KnownManagedServiceIdentityType;
(function (KnownManagedServiceIdentityType) {
    /** SystemAssigned */
    KnownManagedServiceIdentityType["SystemAssigned"] = "SystemAssigned";
    /** None */
    KnownManagedServiceIdentityType["None"] = "None";
})(KnownManagedServiceIdentityType || (KnownManagedServiceIdentityType = {}));
/** Known values of {@link ServiceEventState} that the service accepts. */
export var KnownServiceEventState;
(function (KnownServiceEventState) {
    /** Disabled */
    KnownServiceEventState["Disabled"] = "Disabled";
    /** Enabled */
    KnownServiceEventState["Enabled"] = "Enabled";
    /** Updating */
    KnownServiceEventState["Updating"] = "Updating";
})(KnownServiceEventState || (KnownServiceEventState = {}));
/** Known values of {@link ServiceManagedIdentityType} that the service accepts. */
export var KnownServiceManagedIdentityType;
(function (KnownServiceManagedIdentityType) {
    /** None */
    KnownServiceManagedIdentityType["None"] = "None";
    /** SystemAssigned */
    KnownServiceManagedIdentityType["SystemAssigned"] = "SystemAssigned";
    /** UserAssigned */
    KnownServiceManagedIdentityType["UserAssigned"] = "UserAssigned";
    /** SystemAssignedUserAssigned */
    KnownServiceManagedIdentityType["SystemAssignedUserAssigned"] = "SystemAssigned,UserAssigned";
})(KnownServiceManagedIdentityType || (KnownServiceManagedIdentityType = {}));
/** Known values of {@link IotIdentityResolutionType} that the service accepts. */
export var KnownIotIdentityResolutionType;
(function (KnownIotIdentityResolutionType) {
    /** Create */
    KnownIotIdentityResolutionType["Create"] = "Create";
    /** Lookup */
    KnownIotIdentityResolutionType["Lookup"] = "Lookup";
})(KnownIotIdentityResolutionType || (KnownIotIdentityResolutionType = {}));
/** Known values of {@link FhirServiceKind} that the service accepts. */
export var KnownFhirServiceKind;
(function (KnownFhirServiceKind) {
    /** FhirStu3 */
    KnownFhirServiceKind["FhirStu3"] = "fhir-Stu3";
    /** FhirR4 */
    KnownFhirServiceKind["FhirR4"] = "fhir-R4";
})(KnownFhirServiceKind || (KnownFhirServiceKind = {}));
/** Known values of {@link FhirResourceVersionPolicy} that the service accepts. */
export var KnownFhirResourceVersionPolicy;
(function (KnownFhirResourceVersionPolicy) {
    /** NoVersion */
    KnownFhirResourceVersionPolicy["NoVersion"] = "no-version";
    /** Versioned */
    KnownFhirResourceVersionPolicy["Versioned"] = "versioned";
    /** VersionedUpdate */
    KnownFhirResourceVersionPolicy["VersionedUpdate"] = "versioned-update";
})(KnownFhirResourceVersionPolicy || (KnownFhirResourceVersionPolicy = {}));
/** Known values of {@link ActionType} that the service accepts. */
export var KnownActionType;
(function (KnownActionType) {
    /** Internal */
    KnownActionType["Internal"] = "Internal";
})(KnownActionType || (KnownActionType = {}));
/** Known values of {@link OperationResultStatus} that the service accepts. */
export var KnownOperationResultStatus;
(function (KnownOperationResultStatus) {
    /** Canceled */
    KnownOperationResultStatus["Canceled"] = "Canceled";
    /** Succeeded */
    KnownOperationResultStatus["Succeeded"] = "Succeeded";
    /** Failed */
    KnownOperationResultStatus["Failed"] = "Failed";
    /** Requested */
    KnownOperationResultStatus["Requested"] = "Requested";
    /** Running */
    KnownOperationResultStatus["Running"] = "Running";
})(KnownOperationResultStatus || (KnownOperationResultStatus = {}));
