// @azure/arm-powerbiembedded/dist-esm/src/models/index.js
/*
 * Copyright (c) Microsoft Corporation.
 * Licensed under the MIT License.
 *
 * Code generated by Microsoft (R) AutoRest Code Generator.
 * Changes may cause incorrect behavior and will be lost if the code is regenerated.
 */
/** Known values of {@link AzureSkuName} that the service accepts. */
export var KnownAzureSkuName;
(function (KnownAzureSkuName) {
    /** S1 */
    KnownAzureSkuName["S1"] = "S1";
})(KnownAzureSkuName || (KnownAzureSkuName = {}));
/** Known values of {@link AzureSkuTier} that the service accepts. */
export var KnownAzureSkuTier;
(function (KnownAzureSkuTier) {
    /** Standard */
    KnownAzureSkuTier["Standard"] = "Standard";
})(KnownAzureSkuTier || (KnownAzureSkuTier = {}));
/** Known values of {@link CheckNameReason} that the service accepts. */
export var KnownCheckNameReason;
(function (KnownCheckNameReason) {
    /** Unavailable */
    KnownCheckNameReason["Unavailable"] = "Unavailable";
    /** Invalid */
    KnownCheckNameReason["Invalid"] = "Invalid";
})(KnownCheckNameReason || (KnownCheckNameReason = {}));
