// @azure/arm-powerbiembedded/dist-esm/src/models/parameters.js
/*
 * Copyright (c) Microsoft Corporation.
 * Licensed under the MIT License.
 *
 * Code generated by Microsoft (R) AutoRest Code Generator.
 * Changes may cause incorrect behavior and will be lost if the code is regenerated.
 */
import { CreateWorkspaceCollectionRequest as CreateWorkspaceCollectionRequestMapper, UpdateWorkspaceCollectionRequest as UpdateWorkspaceCollectionRequestMapper, CheckNameRequest as CheckNameRequestMapper, WorkspaceCollectionAccessKey as WorkspaceCollectionAccessKeyMapper, MigrateWorkspaceCollectionRequest as MigrateWorkspaceCollectionRequestMapper } from "../models/mappers";
export const accept = {
    parameterPath: "accept",
    mapper: {
        defaultValue: "application/json",
        isConstant: true,
        serializedName: "Accept",
        type: {
            name: "String"
        }
    }
};
export const $host = {
    parameterPath: "$host",
    mapper: {
        serializedName: "$host",
        required: true,
        type: {
            name: "String"
        }
    },
    skipEncoding: true
};
export const subscriptionId = {
    parameterPath: "subscriptionId",
    mapper: {
        serializedName: "subscriptionId",
        required: true,
        type: {
            name: "String"
        }
    }
};
export const resourceGroupName = {
    parameterPath: "resourceGroupName",
    mapper: {
        serializedName: "resourceGroupName",
        required: true,
        type: {
            name: "String"
        }
    }
};
export const workspaceCollectionName = {
    parameterPath: "workspaceCollectionName",
    mapper: {
        serializedName: "workspaceCollectionName",
        required: true,
        type: {
            name: "String"
        }
    }
};
export const apiVersion = {
    parameterPath: "apiVersion",
    mapper: {
        defaultValue: "2016-01-29",
        isConstant: true,
        serializedName: "api-version",
        type: {
            name: "String"
        }
    }
};
export const contentType = {
    parameterPath: ["options", "contentType"],
    mapper: {
        defaultValue: "application/json",
        isConstant: true,
        serializedName: "Content-Type",
        type: {
            name: "String"
        }
    }
};
export const body = {
    parameterPath: "body",
    mapper: CreateWorkspaceCollectionRequestMapper
};
export const body1 = {
    parameterPath: "body",
    mapper: UpdateWorkspaceCollectionRequestMapper
};
export const body2 = {
    parameterPath: "body",
    mapper: CheckNameRequestMapper
};
export const location = {
    parameterPath: "location",
    mapper: {
        serializedName: "location",
        required: true,
        type: {
            name: "String"
        }
    }
};
export const body3 = {
    parameterPath: "body",
    mapper: WorkspaceCollectionAccessKeyMapper
};
export const body4 = {
    parameterPath: "body",
    mapper: MigrateWorkspaceCollectionRequestMapper
};
