// @azure/arm-azureadexternalidentities/dist-esm/src/models/index.js
/*
 * Copyright (c) Microsoft Corporation.
 * Licensed under the MIT License.
 *
 * Code generated by Microsoft (R) AutoRest Code Generator.
 * Changes may cause incorrect behavior and will be lost if the code is regenerated.
 */
/** Known values of {@link NameAvailabilityReasonType} that the service accepts. */
export var KnownNameAvailabilityReasonType;
(function (KnownNameAvailabilityReasonType) {
    /** The name is already in use and is therefore unavailable. */
    KnownNameAvailabilityReasonType["AlreadyExists"] = "AlreadyExists";
    /** The name provided does not match the resource provider’s naming requirements (incorrect length, unsupported characters, etc.). */
    KnownNameAvailabilityReasonType["Invalid"] = "Invalid";
})(KnownNameAvailabilityReasonType || (KnownNameAvailabilityReasonType = {}));
/** Known values of {@link TypeValue} that the service accepts. */
export var KnownTypeValue;
(function (KnownTypeValue) {
    /** The resource type for Azure AD B2C tenant resource. */
    KnownTypeValue["MicrosoftAzureActiveDirectoryB2CDirectories"] = "Microsoft.AzureActiveDirectory/b2cDirectories";
})(KnownTypeValue || (KnownTypeValue = {}));
/** Known values of {@link B2CResourceSKUName} that the service accepts. */
export var KnownB2CResourceSKUName;
(function (KnownB2CResourceSKUName) {
    /** Azure AD B2C usage is billed to a linked Azure subscription and uses a monthly active users (MAU) billing model. */
    KnownB2CResourceSKUName["Standard"] = "Standard";
    /** Azure AD B2C usage is billed to a linked Azure subscription and uses number of authentications based billing. */
    KnownB2CResourceSKUName["PremiumP1"] = "PremiumP1";
    /** Azure AD B2C usage is billed to a linked Azure subscription and uses number of authentications based billing. */
    KnownB2CResourceSKUName["PremiumP2"] = "PremiumP2";
})(KnownB2CResourceSKUName || (KnownB2CResourceSKUName = {}));
/** Known values of {@link B2CResourceSKUTier} that the service accepts. */
export var KnownB2CResourceSKUTier;
(function (KnownB2CResourceSKUTier) {
    /** The SKU tier used for all Azure AD B2C tenants. */
    KnownB2CResourceSKUTier["A0"] = "A0";
})(KnownB2CResourceSKUTier || (KnownB2CResourceSKUTier = {}));
/** Known values of {@link BillingType} that the service accepts. */
export var KnownBillingType;
(function (KnownBillingType) {
    /** Azure AD B2C usage is billed to a linked Azure subscription and uses a monthly active users (MAU) billing model. */
    KnownBillingType["MAU"] = "MAU";
    /** Azure AD B2C usage is billed to a linked Azure subscription and uses number of authentications based billing. */
    KnownBillingType["Auths"] = "Auths";
})(KnownBillingType || (KnownBillingType = {}));
/** Known values of {@link CreatedByType} that the service accepts. */
export var KnownCreatedByType;
(function (KnownCreatedByType) {
    /** User */
    KnownCreatedByType["User"] = "User";
    /** Application */
    KnownCreatedByType["Application"] = "Application";
    /** ManagedIdentity */
    KnownCreatedByType["ManagedIdentity"] = "ManagedIdentity";
    /** Key */
    KnownCreatedByType["Key"] = "Key";
})(KnownCreatedByType || (KnownCreatedByType = {}));
