// @azure/arm-imagebuilder/dist-esm/src/models/index.js
/*
 * Copyright (c) Microsoft Corporation.
 * Licensed under the MIT License.
 *
 * Code generated by Microsoft (R) AutoRest Code Generator.
 * Changes may cause incorrect behavior and will be lost if the code is regenerated.
 */
/** Known values of {@link OnBuildError} that the service accepts. */
export var KnownOnBuildError;
(function (KnownOnBuildError) {
    /** Cleanup */
    KnownOnBuildError["Cleanup"] = "cleanup";
    /** Abort */
    KnownOnBuildError["Abort"] = "abort";
})(KnownOnBuildError || (KnownOnBuildError = {}));
/** Known values of {@link ProvisioningErrorCode} that the service accepts. */
export var KnownProvisioningErrorCode;
(function (KnownProvisioningErrorCode) {
    /** BadSourceType */
    KnownProvisioningErrorCode["BadSourceType"] = "BadSourceType";
    /** BadPIRSource */
    KnownProvisioningErrorCode["BadPIRSource"] = "BadPIRSource";
    /** BadManagedImageSource */
    KnownProvisioningErrorCode["BadManagedImageSource"] = "BadManagedImageSource";
    /** BadSharedImageVersionSource */
    KnownProvisioningErrorCode["BadSharedImageVersionSource"] = "BadSharedImageVersionSource";
    /** BadCustomizerType */
    KnownProvisioningErrorCode["BadCustomizerType"] = "BadCustomizerType";
    /** UnsupportedCustomizerType */
    KnownProvisioningErrorCode["UnsupportedCustomizerType"] = "UnsupportedCustomizerType";
    /** NoCustomizerScript */
    KnownProvisioningErrorCode["NoCustomizerScript"] = "NoCustomizerScript";
    /** BadValidatorType */
    KnownProvisioningErrorCode["BadValidatorType"] = "BadValidatorType";
    /** UnsupportedValidatorType */
    KnownProvisioningErrorCode["UnsupportedValidatorType"] = "UnsupportedValidatorType";
    /** NoValidatorScript */
    KnownProvisioningErrorCode["NoValidatorScript"] = "NoValidatorScript";
    /** BadDistributeType */
    KnownProvisioningErrorCode["BadDistributeType"] = "BadDistributeType";
    /** BadSharedImageDistribute */
    KnownProvisioningErrorCode["BadSharedImageDistribute"] = "BadSharedImageDistribute";
    /** BadStagingResourceGroup */
    KnownProvisioningErrorCode["BadStagingResourceGroup"] = "BadStagingResourceGroup";
    /** ServerError */
    KnownProvisioningErrorCode["ServerError"] = "ServerError";
    /** Other */
    KnownProvisioningErrorCode["Other"] = "Other";
})(KnownProvisioningErrorCode || (KnownProvisioningErrorCode = {}));
/** Known values of {@link CreatedByType} that the service accepts. */
export var KnownCreatedByType;
(function (KnownCreatedByType) {
    /** User */
    KnownCreatedByType["User"] = "User";
    /** Application */
    KnownCreatedByType["Application"] = "Application";
    /** ManagedIdentity */
    KnownCreatedByType["ManagedIdentity"] = "ManagedIdentity";
    /** Key */
    KnownCreatedByType["Key"] = "Key";
})(KnownCreatedByType || (KnownCreatedByType = {}));
/** Known values of {@link SharedImageStorageAccountType} that the service accepts. */
export var KnownSharedImageStorageAccountType;
(function (KnownSharedImageStorageAccountType) {
    /** StandardLRS */
    KnownSharedImageStorageAccountType["StandardLRS"] = "Standard_LRS";
    /** StandardZRS */
    KnownSharedImageStorageAccountType["StandardZRS"] = "Standard_ZRS";
    /** PremiumLRS */
    KnownSharedImageStorageAccountType["PremiumLRS"] = "Premium_LRS";
})(KnownSharedImageStorageAccountType || (KnownSharedImageStorageAccountType = {}));
