// @aws-sdk/client-fms/dist-cjs/index.js
"use strict";
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __name = (target, value) => __defProp(target, "name", { value, configurable: true });
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// src/index.ts
var src_exports = {};
__export(src_exports, {
  AccountRoleStatus: () => AccountRoleStatus,
  AssociateAdminAccountCommand: () => AssociateAdminAccountCommand,
  AssociateThirdPartyFirewallCommand: () => AssociateThirdPartyFirewallCommand,
  BatchAssociateResourceCommand: () => BatchAssociateResourceCommand,
  BatchDisassociateResourceCommand: () => BatchDisassociateResourceCommand,
  CustomerPolicyScopeIdType: () => CustomerPolicyScopeIdType,
  CustomerPolicyStatus: () => CustomerPolicyStatus,
  DeleteAppsListCommand: () => DeleteAppsListCommand,
  DeleteNotificationChannelCommand: () => DeleteNotificationChannelCommand,
  DeletePolicyCommand: () => DeletePolicyCommand,
  DeleteProtocolsListCommand: () => DeleteProtocolsListCommand,
  DeleteResourceSetCommand: () => DeleteResourceSetCommand,
  DependentServiceName: () => DependentServiceName,
  DestinationType: () => DestinationType,
  DisassociateAdminAccountCommand: () => DisassociateAdminAccountCommand,
  DisassociateThirdPartyFirewallCommand: () => DisassociateThirdPartyFirewallCommand,
  FMS: () => FMS,
  FMSClient: () => FMSClient,
  FMSServiceException: () => FMSServiceException,
  FailedItemReason: () => FailedItemReason,
  FirewallDeploymentModel: () => FirewallDeploymentModel,
  GetAdminAccountCommand: () => GetAdminAccountCommand,
  GetAdminScopeCommand: () => GetAdminScopeCommand,
  GetAppsListCommand: () => GetAppsListCommand,
  GetComplianceDetailCommand: () => GetComplianceDetailCommand,
  GetNotificationChannelCommand: () => GetNotificationChannelCommand,
  GetPolicyCommand: () => GetPolicyCommand,
  GetProtectionStatusCommand: () => GetProtectionStatusCommand,
  GetProtocolsListCommand: () => GetProtocolsListCommand,
  GetResourceSetCommand: () => GetResourceSetCommand,
  GetThirdPartyFirewallAssociationStatusCommand: () => GetThirdPartyFirewallAssociationStatusCommand,
  GetViolationDetailsCommand: () => GetViolationDetailsCommand,
  InternalErrorException: () => InternalErrorException,
  InvalidInputException: () => InvalidInputException,
  InvalidOperationException: () => InvalidOperationException,
  InvalidTypeException: () => InvalidTypeException,
  LimitExceededException: () => LimitExceededException,
  ListAdminAccountsForOrganizationCommand: () => ListAdminAccountsForOrganizationCommand,
  ListAdminsManagingAccountCommand: () => ListAdminsManagingAccountCommand,
  ListAppsListsCommand: () => ListAppsListsCommand,
  ListComplianceStatusCommand: () => ListComplianceStatusCommand,
  ListDiscoveredResourcesCommand: () => ListDiscoveredResourcesCommand,
  ListMemberAccountsCommand: () => ListMemberAccountsCommand,
  ListPoliciesCommand: () => ListPoliciesCommand,
  ListProtocolsListsCommand: () => ListProtocolsListsCommand,
  ListResourceSetResourcesCommand: () => ListResourceSetResourcesCommand,
  ListResourceSetsCommand: () => ListResourceSetsCommand,
  ListTagsForResourceCommand: () => ListTagsForResourceCommand,
  ListThirdPartyFirewallFirewallPoliciesCommand: () => ListThirdPartyFirewallFirewallPoliciesCommand,
  MarketplaceSubscriptionOnboardingStatus: () => MarketplaceSubscriptionOnboardingStatus,
  NetworkFirewallOverrideAction: () => NetworkFirewallOverrideAction,
  OrganizationStatus: () => OrganizationStatus,
  PolicyComplianceStatusType: () => PolicyComplianceStatusType,
  PutAdminAccountCommand: () => PutAdminAccountCommand,
  PutAppsListCommand: () => PutAppsListCommand,
  PutNotificationChannelCommand: () => PutNotificationChannelCommand,
  PutPolicyCommand: () => PutPolicyCommand,
  PutProtocolsListCommand: () => PutProtocolsListCommand,
  PutResourceSetCommand: () => PutResourceSetCommand,
  RemediationActionType: () => RemediationActionType,
  ResourceNotFoundException: () => ResourceNotFoundException,
  ResourceSetStatus: () => ResourceSetStatus,
  RuleOrder: () => RuleOrder,
  SecurityServiceType: () => SecurityServiceType,
  TagResourceCommand: () => TagResourceCommand,
  TargetType: () => TargetType,
  ThirdPartyFirewall: () => ThirdPartyFirewall,
  ThirdPartyFirewallAssociationStatus: () => ThirdPartyFirewallAssociationStatus,
  UntagResourceCommand: () => UntagResourceCommand,
  ViolationReason: () => ViolationReason,
  __Client: () => import_smithy_client.Client,
  paginateListAdminAccountsForOrganization: () => paginateListAdminAccountsForOrganization,
  paginateListAdminsManagingAccount: () => paginateListAdminsManagingAccount,
  paginateListAppsLists: () => paginateListAppsLists,
  paginateListComplianceStatus: () => paginateListComplianceStatus,
  paginateListMemberAccounts: () => paginateListMemberAccounts,
  paginateListPolicies: () => paginateListPolicies,
  paginateListProtocolsLists: () => paginateListProtocolsLists,
  paginateListThirdPartyFirewallFirewallPolicies: () => paginateListThirdPartyFirewallFirewallPolicies
});
module.exports = __toCommonJS(src_exports);

// src/FMSClient.ts
var import_middleware_host_header = require("@aws-sdk/middleware-host-header");
var import_middleware_logger = require("@aws-sdk/middleware-logger");
var import_middleware_recursion_detection = require("@aws-sdk/middleware-recursion-detection");
var import_middleware_user_agent = require("@aws-sdk/middleware-user-agent");
var import_config_resolver = require("@smithy/config-resolver");
var import_core = require("@smithy/core");
var import_middleware_content_length = require("@smithy/middleware-content-length");
var import_middleware_endpoint = require("@smithy/middleware-endpoint");
var import_middleware_retry = require("@smithy/middleware-retry");

var import_httpAuthSchemeProvider = require("./auth/httpAuthSchemeProvider");

// src/endpoint/EndpointParameters.ts
var resolveClientEndpointParameters = /* @__PURE__ */ __name((options) => {
  return {
    ...options,
    useDualstackEndpoint: options.useDualstackEndpoint ?? false,
    useFipsEndpoint: options.useFipsEndpoint ?? false,
    defaultSigningName: "fms"
  };
}, "resolveClientEndpointParameters");
var commonParams = {
  UseFIPS: { type: "builtInParams", name: "useFipsEndpoint" },
  Endpoint: { type: "builtInParams", name: "endpoint" },
  Region: { type: "builtInParams", name: "region" },
  UseDualStack: { type: "builtInParams", name: "useDualstackEndpoint" }
};

// src/FMSClient.ts
var import_runtimeConfig = require("././runtimeConfig");

// src/runtimeExtensions.ts
var import_region_config_resolver = require("@aws-sdk/region-config-resolver");
var import_protocol_http = require("@smithy/protocol-http");
var import_smithy_client = require("@smithy/smithy-client");

// src/auth/httpAuthExtensionConfiguration.ts
var getHttpAuthExtensionConfiguration = /* @__PURE__ */ __name((runtimeConfig) => {
  const _httpAuthSchemes = runtimeConfig.httpAuthSchemes;
  let _httpAuthSchemeProvider = runtimeConfig.httpAuthSchemeProvider;
  let _credentials = runtimeConfig.credentials;
  return {
    setHttpAuthScheme(httpAuthScheme) {
      const index = _httpAuthSchemes.findIndex((scheme) => scheme.schemeId === httpAuthScheme.schemeId);
      if (index === -1) {
        _httpAuthSchemes.push(httpAuthScheme);
      } else {
        _httpAuthSchemes.splice(index, 1, httpAuthScheme);
      }
    },
    httpAuthSchemes() {
      return _httpAuthSchemes;
    },
    setHttpAuthSchemeProvider(httpAuthSchemeProvider) {
      _httpAuthSchemeProvider = httpAuthSchemeProvider;
    },
    httpAuthSchemeProvider() {
      return _httpAuthSchemeProvider;
    },
    setCredentials(credentials) {
      _credentials = credentials;
    },
    credentials() {
      return _credentials;
    }
  };
}, "getHttpAuthExtensionConfiguration");
var resolveHttpAuthRuntimeConfig = /* @__PURE__ */ __name((config) => {
  return {
    httpAuthSchemes: config.httpAuthSchemes(),
    httpAuthSchemeProvider: config.httpAuthSchemeProvider(),
    credentials: config.credentials()
  };
}, "resolveHttpAuthRuntimeConfig");

// src/runtimeExtensions.ts
var asPartial = /* @__PURE__ */ __name((t) => t, "asPartial");
var resolveRuntimeExtensions = /* @__PURE__ */ __name((runtimeConfig, extensions) => {
  const extensionConfiguration = {
    ...asPartial((0, import_region_config_resolver.getAwsRegionExtensionConfiguration)(runtimeConfig)),
    ...asPartial((0, import_smithy_client.getDefaultExtensionConfiguration)(runtimeConfig)),
    ...asPartial((0, import_protocol_http.getHttpHandlerExtensionConfiguration)(runtimeConfig)),
    ...asPartial(getHttpAuthExtensionConfiguration(runtimeConfig))
  };
  extensions.forEach((extension) => extension.configure(extensionConfiguration));
  return {
    ...runtimeConfig,
    ...(0, import_region_config_resolver.resolveAwsRegionExtensionConfiguration)(extensionConfiguration),
    ...(0, import_smithy_client.resolveDefaultRuntimeConfig)(extensionConfiguration),
    ...(0, import_protocol_http.resolveHttpHandlerRuntimeConfig)(extensionConfiguration),
    ...resolveHttpAuthRuntimeConfig(extensionConfiguration)
  };
}, "resolveRuntimeExtensions");

// src/FMSClient.ts
var _FMSClient = class _FMSClient extends import_smithy_client.Client {
  constructor(...[configuration]) {
    const _config_0 = (0, import_runtimeConfig.getRuntimeConfig)(configuration || {});
    const _config_1 = resolveClientEndpointParameters(_config_0);
    const _config_2 = (0, import_config_resolver.resolveRegionConfig)(_config_1);
    const _config_3 = (0, import_middleware_endpoint.resolveEndpointConfig)(_config_2);
    const _config_4 = (0, import_middleware_retry.resolveRetryConfig)(_config_3);
    const _config_5 = (0, import_middleware_host_header.resolveHostHeaderConfig)(_config_4);
    const _config_6 = (0, import_middleware_user_agent.resolveUserAgentConfig)(_config_5);
    const _config_7 = (0, import_httpAuthSchemeProvider.resolveHttpAuthSchemeConfig)(_config_6);
    const _config_8 = resolveRuntimeExtensions(_config_7, (configuration == null ? void 0 : configuration.extensions) || []);
    super(_config_8);
    this.config = _config_8;
    this.middlewareStack.use((0, import_middleware_retry.getRetryPlugin)(this.config));
    this.middlewareStack.use((0, import_middleware_content_length.getContentLengthPlugin)(this.config));
    this.middlewareStack.use((0, import_middleware_host_header.getHostHeaderPlugin)(this.config));
    this.middlewareStack.use((0, import_middleware_logger.getLoggerPlugin)(this.config));
    this.middlewareStack.use((0, import_middleware_recursion_detection.getRecursionDetectionPlugin)(this.config));
    this.middlewareStack.use((0, import_middleware_user_agent.getUserAgentPlugin)(this.config));
    this.middlewareStack.use(
      (0, import_core.getHttpAuthSchemeEndpointRuleSetPlugin)(this.config, {
        httpAuthSchemeParametersProvider: this.getDefaultHttpAuthSchemeParametersProvider(),
        identityProviderConfigProvider: this.getIdentityProviderConfigProvider()
      })
    );
    this.middlewareStack.use((0, import_core.getHttpSigningPlugin)(this.config));
  }
  /**
   * Destroy underlying resources, like sockets. It's usually not necessary to do this.
   * However in Node.js, it's best to explicitly shut down the client's agent when it is no longer needed.
   * Otherwise, sockets might stay open for quite a long time before the server terminates them.
   */
  destroy() {
    super.destroy();
  }
  getDefaultHttpAuthSchemeParametersProvider() {
    return import_httpAuthSchemeProvider.defaultFMSHttpAuthSchemeParametersProvider;
  }
  getIdentityProviderConfigProvider() {
    return async (config) => new import_core.DefaultIdentityProviderConfig({
      "aws.auth#sigv4": config.credentials
    });
  }
};
__name(_FMSClient, "FMSClient");
var FMSClient = _FMSClient;

// src/FMS.ts


// src/commands/AssociateAdminAccountCommand.ts

var import_middleware_serde = require("@smithy/middleware-serde");

var import_types = require("@smithy/types");

// src/protocols/Aws_json1_1.ts
var import_core2 = require("@aws-sdk/core");



// src/models/FMSServiceException.ts

var _FMSServiceException = class _FMSServiceException extends import_smithy_client.ServiceException {
  /**
   * @internal
   */
  constructor(options) {
    super(options);
    Object.setPrototypeOf(this, _FMSServiceException.prototype);
  }
};
__name(_FMSServiceException, "FMSServiceException");
var FMSServiceException = _FMSServiceException;

// src/models/models_0.ts
var AccountRoleStatus = {
  Creating: "CREATING",
  Deleted: "DELETED",
  Deleting: "DELETING",
  PendingDeletion: "PENDING_DELETION",
  Ready: "READY"
};
var OrganizationStatus = {
  Offboarding: "OFFBOARDING",
  OffboardingComplete: "OFFBOARDING_COMPLETE",
  Onboarding: "ONBOARDING",
  OnboardingComplete: "ONBOARDING_COMPLETE"
};
var SecurityServiceType = {
  DNS_FIREWALL: "DNS_FIREWALL",
  IMPORT_NETWORK_FIREWALL: "IMPORT_NETWORK_FIREWALL",
  NETWORK_FIREWALL: "NETWORK_FIREWALL",
  SECURITY_GROUPS_COMMON: "SECURITY_GROUPS_COMMON",
  SECURITY_GROUPS_CONTENT_AUDIT: "SECURITY_GROUPS_CONTENT_AUDIT",
  SECURITY_GROUPS_USAGE_AUDIT: "SECURITY_GROUPS_USAGE_AUDIT",
  SHIELD_ADVANCED: "SHIELD_ADVANCED",
  THIRD_PARTY_FIREWALL: "THIRD_PARTY_FIREWALL",
  WAF: "WAF",
  WAFV2: "WAFV2"
};
var _InternalErrorException = class _InternalErrorException extends FMSServiceException {
  /**
   * @internal
   */
  constructor(opts) {
    super({
      name: "InternalErrorException",
      $fault: "client",
      ...opts
    });
    this.name = "InternalErrorException";
    this.$fault = "client";
    Object.setPrototypeOf(this, _InternalErrorException.prototype);
    this.Message = opts.Message;
  }
};
__name(_InternalErrorException, "InternalErrorException");
var InternalErrorException = _InternalErrorException;
var _InvalidInputException = class _InvalidInputException extends FMSServiceException {
  /**
   * @internal
   */
  constructor(opts) {
    super({
      name: "InvalidInputException",
      $fault: "client",
      ...opts
    });
    this.name = "InvalidInputException";
    this.$fault = "client";
    Object.setPrototypeOf(this, _InvalidInputException.prototype);
    this.Message = opts.Message;
  }
};
__name(_InvalidInputException, "InvalidInputException");
var InvalidInputException = _InvalidInputException;
var _InvalidOperationException = class _InvalidOperationException extends FMSServiceException {
  /**
   * @internal
   */
  constructor(opts) {
    super({
      name: "InvalidOperationException",
      $fault: "client",
      ...opts
    });
    this.name = "InvalidOperationException";
    this.$fault = "client";
    Object.setPrototypeOf(this, _InvalidOperationException.prototype);
    this.Message = opts.Message;
  }
};
__name(_InvalidOperationException, "InvalidOperationException");
var InvalidOperationException = _InvalidOperationException;
var _LimitExceededException = class _LimitExceededException extends FMSServiceException {
  /**
   * @internal
   */
  constructor(opts) {
    super({
      name: "LimitExceededException",
      $fault: "client",
      ...opts
    });
    this.name = "LimitExceededException";
    this.$fault = "client";
    Object.setPrototypeOf(this, _LimitExceededException.prototype);
    this.Message = opts.Message;
  }
};
__name(_LimitExceededException, "LimitExceededException");
var LimitExceededException = _LimitExceededException;
var _ResourceNotFoundException = class _ResourceNotFoundException extends FMSServiceException {
  /**
   * @internal
   */
  constructor(opts) {
    super({
      name: "ResourceNotFoundException",
      $fault: "client",
      ...opts
    });
    this.name = "ResourceNotFoundException";
    this.$fault = "client";
    Object.setPrototypeOf(this, _ResourceNotFoundException.prototype);
    this.Message = opts.Message;
  }
};
__name(_ResourceNotFoundException, "ResourceNotFoundException");
var ResourceNotFoundException = _ResourceNotFoundException;
var ThirdPartyFirewall = {
  FORTIGATE_CLOUD_NATIVE_FIREWALL: "FORTIGATE_CLOUD_NATIVE_FIREWALL",
  PALO_ALTO_NETWORKS_CLOUD_NGFW: "PALO_ALTO_NETWORKS_CLOUD_NGFW"
};
var ThirdPartyFirewallAssociationStatus = {
  NOT_EXIST: "NOT_EXIST",
  OFFBOARDING: "OFFBOARDING",
  OFFBOARD_COMPLETE: "OFFBOARD_COMPLETE",
  ONBOARDING: "ONBOARDING",
  ONBOARD_COMPLETE: "ONBOARD_COMPLETE"
};
var FailedItemReason = {
  NotValidAccountId: "NOT_VALID_ACCOUNT_ID",
  NotValidArn: "NOT_VALID_ARN",
  NotValidPartition: "NOT_VALID_PARTITION",
  NotValidRegion: "NOT_VALID_REGION",
  NotValidResourceType: "NOT_VALID_RESOURCE_TYPE",
  NotValidService: "NOT_VALID_SERVICE"
};
var DependentServiceName = {
  AWSConfig: "AWSCONFIG",
  AWSShieldAdvanced: "AWSSHIELD_ADVANCED",
  AWSVirtualPrivateCloud: "AWSVPC",
  AWSWAF: "AWSWAF"
};
var ViolationReason = {
  BlackHoleRouteDetected: "BLACK_HOLE_ROUTE_DETECTED",
  BlackHoleRouteDetectedInFirewallSubnet: "BLACK_HOLE_ROUTE_DETECTED_IN_FIREWALL_SUBNET",
  FMSCreatedSecurityGroupEdited: "FMS_CREATED_SECURITY_GROUP_EDITED",
  FirewallSubnetIsOutOfScope: "FIREWALL_SUBNET_IS_OUT_OF_SCOPE",
  FirewallSubnetMissingExpectedRoute: "FIREWALL_SUBNET_MISSING_EXPECTED_ROUTE",
  FirewallSubnetMissingVPCEndpoint: "FIREWALL_SUBNET_MISSING_VPCE_ENDPOINT",
  InternetGatewayMissingExpectedRoute: "INTERNET_GATEWAY_MISSING_EXPECTED_ROUTE",
  InternetTrafficNotInspected: "INTERNET_TRAFFIC_NOT_INSPECTED",
  InvalidRouteConfiguration: "INVALID_ROUTE_CONFIGURATION",
  MissingExpectedRouteTable: "MISSING_EXPECTED_ROUTE_TABLE",
  MissingFirewall: "MISSING_FIREWALL",
  MissingFirewallSubnetInAZ: "MISSING_FIREWALL_SUBNET_IN_AZ",
  MissingTargetGateway: "MISSING_TARGET_GATEWAY",
  NetworkFirewallPolicyModified: "NETWORK_FIREWALL_POLICY_MODIFIED",
  ResourceIncorrectWebAcl: "RESOURCE_INCORRECT_WEB_ACL",
  ResourceMissingDnsFirewall: "RESOURCE_MISSING_DNS_FIREWALL",
  ResourceMissingSecurityGroup: "RESOURCE_MISSING_SECURITY_GROUP",
  ResourceMissingShieldProtection: "RESOURCE_MISSING_SHIELD_PROTECTION",
  ResourceMissingWebAcl: "RESOURCE_MISSING_WEB_ACL",
  ResourceMissingWebaclOrShieldProtection: "RESOURCE_MISSING_WEB_ACL_OR_SHIELD_PROTECTION",
  ResourceViolatesAuditSecurityGroup: "RESOURCE_VIOLATES_AUDIT_SECURITY_GROUP",
  RouteHasOutOfScopeEndpoint: "ROUTE_HAS_OUT_OF_SCOPE_ENDPOINT",
  SecurityGroupRedundant: "SECURITY_GROUP_REDUNDANT",
  SecurityGroupUnused: "SECURITY_GROUP_UNUSED",
  TrafficInspectionCrossesAZBoundary: "TRAFFIC_INSPECTION_CROSSES_AZ_BOUNDARY",
  UnexpectedFirewallRoutes: "UNEXPECTED_FIREWALL_ROUTES",
  UnexpectedTargetGatewayRoutes: "UNEXPECTED_TARGET_GATEWAY_ROUTES",
  WebAclMissingRuleGroup: "WEB_ACL_MISSING_RULE_GROUP"
};
var CustomerPolicyScopeIdType = {
  ACCOUNT: "ACCOUNT",
  ORG_UNIT: "ORG_UNIT"
};
var CustomerPolicyStatus = {
  ACTIVE: "ACTIVE",
  OUT_OF_ADMIN_SCOPE: "OUT_OF_ADMIN_SCOPE"
};
var FirewallDeploymentModel = {
  CENTRALIZED: "CENTRALIZED",
  DISTRIBUTED: "DISTRIBUTED"
};
var _InvalidTypeException = class _InvalidTypeException extends FMSServiceException {
  /**
   * @internal
   */
  constructor(opts) {
    super({
      name: "InvalidTypeException",
      $fault: "client",
      ...opts
    });
    this.name = "InvalidTypeException";
    this.$fault = "client";
    Object.setPrototypeOf(this, _InvalidTypeException.prototype);
    this.Message = opts.Message;
  }
};
__name(_InvalidTypeException, "InvalidTypeException");
var InvalidTypeException = _InvalidTypeException;
var ResourceSetStatus = {
  ACTIVE: "ACTIVE",
  OUT_OF_ADMIN_SCOPE: "OUT_OF_ADMIN_SCOPE"
};
var MarketplaceSubscriptionOnboardingStatus = {
  COMPLETE: "COMPLETE",
  NOT_COMPLETE: "NOT_COMPLETE",
  NO_SUBSCRIPTION: "NO_SUBSCRIPTION"
};
var RemediationActionType = {
  Modify: "MODIFY",
  Remove: "REMOVE"
};
var DestinationType = {
  IPV4: "IPV4",
  IPV6: "IPV6",
  PrefixList: "PREFIX_LIST"
};
var TargetType = {
  CarrierGateway: "CARRIER_GATEWAY",
  EgressOnlyInternetGateway: "EGRESS_ONLY_INTERNET_GATEWAY",
  Gateway: "GATEWAY",
  Instance: "INSTANCE",
  LocalGateway: "LOCAL_GATEWAY",
  NatGateway: "NAT_GATEWAY",
  NetworkInterface: "NETWORK_INTERFACE",
  TransitGateway: "TRANSIT_GATEWAY",
  VPCEndpoint: "VPC_ENDPOINT",
  VPCPeeringConnection: "VPC_PEERING_CONNECTION"
};
var RuleOrder = {
  DEFAULT_ACTION_ORDER: "DEFAULT_ACTION_ORDER",
  STRICT_ORDER: "STRICT_ORDER"
};
var NetworkFirewallOverrideAction = {
  DROP_TO_ALERT: "DROP_TO_ALERT"
};
var PolicyComplianceStatusType = {
  Compliant: "COMPLIANT",
  NonCompliant: "NON_COMPLIANT"
};

// src/protocols/Aws_json1_1.ts
var se_AssociateAdminAccountCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("AssociateAdminAccount");
  let body;
  body = JSON.stringify((0, import_smithy_client._json)(input));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_AssociateAdminAccountCommand");
var se_AssociateThirdPartyFirewallCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("AssociateThirdPartyFirewall");
  let body;
  body = JSON.stringify((0, import_smithy_client._json)(input));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_AssociateThirdPartyFirewallCommand");
var se_BatchAssociateResourceCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("BatchAssociateResource");
  let body;
  body = JSON.stringify((0, import_smithy_client._json)(input));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_BatchAssociateResourceCommand");
var se_BatchDisassociateResourceCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("BatchDisassociateResource");
  let body;
  body = JSON.stringify((0, import_smithy_client._json)(input));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_BatchDisassociateResourceCommand");
var se_DeleteAppsListCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("DeleteAppsList");
  let body;
  body = JSON.stringify((0, import_smithy_client._json)(input));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_DeleteAppsListCommand");
var se_DeleteNotificationChannelCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("DeleteNotificationChannel");
  let body;
  body = JSON.stringify((0, import_smithy_client._json)(input));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_DeleteNotificationChannelCommand");
var se_DeletePolicyCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("DeletePolicy");
  let body;
  body = JSON.stringify((0, import_smithy_client._json)(input));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_DeletePolicyCommand");
var se_DeleteProtocolsListCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("DeleteProtocolsList");
  let body;
  body = JSON.stringify((0, import_smithy_client._json)(input));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_DeleteProtocolsListCommand");
var se_DeleteResourceSetCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("DeleteResourceSet");
  let body;
  body = JSON.stringify((0, import_smithy_client._json)(input));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_DeleteResourceSetCommand");
var se_DisassociateAdminAccountCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("DisassociateAdminAccount");
  let body;
  body = JSON.stringify((0, import_smithy_client._json)(input));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_DisassociateAdminAccountCommand");
var se_DisassociateThirdPartyFirewallCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("DisassociateThirdPartyFirewall");
  let body;
  body = JSON.stringify((0, import_smithy_client._json)(input));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_DisassociateThirdPartyFirewallCommand");
var se_GetAdminAccountCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("GetAdminAccount");
  let body;
  body = JSON.stringify((0, import_smithy_client._json)(input));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_GetAdminAccountCommand");
var se_GetAdminScopeCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("GetAdminScope");
  let body;
  body = JSON.stringify((0, import_smithy_client._json)(input));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_GetAdminScopeCommand");
var se_GetAppsListCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("GetAppsList");
  let body;
  body = JSON.stringify((0, import_smithy_client._json)(input));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_GetAppsListCommand");
var se_GetComplianceDetailCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("GetComplianceDetail");
  let body;
  body = JSON.stringify((0, import_smithy_client._json)(input));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_GetComplianceDetailCommand");
var se_GetNotificationChannelCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("GetNotificationChannel");
  let body;
  body = JSON.stringify((0, import_smithy_client._json)(input));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_GetNotificationChannelCommand");
var se_GetPolicyCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("GetPolicy");
  let body;
  body = JSON.stringify((0, import_smithy_client._json)(input));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_GetPolicyCommand");
var se_GetProtectionStatusCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("GetProtectionStatus");
  let body;
  body = JSON.stringify(se_GetProtectionStatusRequest(input, context));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_GetProtectionStatusCommand");
var se_GetProtocolsListCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("GetProtocolsList");
  let body;
  body = JSON.stringify((0, import_smithy_client._json)(input));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_GetProtocolsListCommand");
var se_GetResourceSetCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("GetResourceSet");
  let body;
  body = JSON.stringify((0, import_smithy_client._json)(input));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_GetResourceSetCommand");
var se_GetThirdPartyFirewallAssociationStatusCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("GetThirdPartyFirewallAssociationStatus");
  let body;
  body = JSON.stringify((0, import_smithy_client._json)(input));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_GetThirdPartyFirewallAssociationStatusCommand");
var se_GetViolationDetailsCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("GetViolationDetails");
  let body;
  body = JSON.stringify((0, import_smithy_client._json)(input));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_GetViolationDetailsCommand");
var se_ListAdminAccountsForOrganizationCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("ListAdminAccountsForOrganization");
  let body;
  body = JSON.stringify((0, import_smithy_client._json)(input));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_ListAdminAccountsForOrganizationCommand");
var se_ListAdminsManagingAccountCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("ListAdminsManagingAccount");
  let body;
  body = JSON.stringify((0, import_smithy_client._json)(input));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_ListAdminsManagingAccountCommand");
var se_ListAppsListsCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("ListAppsLists");
  let body;
  body = JSON.stringify((0, import_smithy_client._json)(input));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_ListAppsListsCommand");
var se_ListComplianceStatusCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("ListComplianceStatus");
  let body;
  body = JSON.stringify((0, import_smithy_client._json)(input));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_ListComplianceStatusCommand");
var se_ListDiscoveredResourcesCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("ListDiscoveredResources");
  let body;
  body = JSON.stringify((0, import_smithy_client._json)(input));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_ListDiscoveredResourcesCommand");
var se_ListMemberAccountsCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("ListMemberAccounts");
  let body;
  body = JSON.stringify((0, import_smithy_client._json)(input));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_ListMemberAccountsCommand");
var se_ListPoliciesCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("ListPolicies");
  let body;
  body = JSON.stringify((0, import_smithy_client._json)(input));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_ListPoliciesCommand");
var se_ListProtocolsListsCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("ListProtocolsLists");
  let body;
  body = JSON.stringify((0, import_smithy_client._json)(input));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_ListProtocolsListsCommand");
var se_ListResourceSetResourcesCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("ListResourceSetResources");
  let body;
  body = JSON.stringify((0, import_smithy_client._json)(input));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_ListResourceSetResourcesCommand");
var se_ListResourceSetsCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("ListResourceSets");
  let body;
  body = JSON.stringify((0, import_smithy_client._json)(input));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_ListResourceSetsCommand");
var se_ListTagsForResourceCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("ListTagsForResource");
  let body;
  body = JSON.stringify((0, import_smithy_client._json)(input));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_ListTagsForResourceCommand");
var se_ListThirdPartyFirewallFirewallPoliciesCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("ListThirdPartyFirewallFirewallPolicies");
  let body;
  body = JSON.stringify((0, import_smithy_client._json)(input));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_ListThirdPartyFirewallFirewallPoliciesCommand");
var se_PutAdminAccountCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("PutAdminAccount");
  let body;
  body = JSON.stringify((0, import_smithy_client._json)(input));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_PutAdminAccountCommand");
var se_PutAppsListCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("PutAppsList");
  let body;
  body = JSON.stringify(se_PutAppsListRequest(input, context));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_PutAppsListCommand");
var se_PutNotificationChannelCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("PutNotificationChannel");
  let body;
  body = JSON.stringify((0, import_smithy_client._json)(input));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_PutNotificationChannelCommand");
var se_PutPolicyCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("PutPolicy");
  let body;
  body = JSON.stringify((0, import_smithy_client._json)(input));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_PutPolicyCommand");
var se_PutProtocolsListCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("PutProtocolsList");
  let body;
  body = JSON.stringify(se_PutProtocolsListRequest(input, context));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_PutProtocolsListCommand");
var se_PutResourceSetCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("PutResourceSet");
  let body;
  body = JSON.stringify(se_PutResourceSetRequest(input, context));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_PutResourceSetCommand");
var se_TagResourceCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("TagResource");
  let body;
  body = JSON.stringify((0, import_smithy_client._json)(input));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_TagResourceCommand");
var se_UntagResourceCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("UntagResource");
  let body;
  body = JSON.stringify((0, import_smithy_client._json)(input));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_UntagResourceCommand");
var de_AssociateAdminAccountCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  await (0, import_smithy_client.collectBody)(output.body, context);
  const response = {
    $metadata: deserializeMetadata(output)
  };
  return response;
}, "de_AssociateAdminAccountCommand");
var de_AssociateThirdPartyFirewallCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const data = await (0, import_core2.parseJsonBody)(output.body, context);
  let contents = {};
  contents = (0, import_smithy_client._json)(data);
  const response = {
    $metadata: deserializeMetadata(output),
    ...contents
  };
  return response;
}, "de_AssociateThirdPartyFirewallCommand");
var de_BatchAssociateResourceCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const data = await (0, import_core2.parseJsonBody)(output.body, context);
  let contents = {};
  contents = (0, import_smithy_client._json)(data);
  const response = {
    $metadata: deserializeMetadata(output),
    ...contents
  };
  return response;
}, "de_BatchAssociateResourceCommand");
var de_BatchDisassociateResourceCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const data = await (0, import_core2.parseJsonBody)(output.body, context);
  let contents = {};
  contents = (0, import_smithy_client._json)(data);
  const response = {
    $metadata: deserializeMetadata(output),
    ...contents
  };
  return response;
}, "de_BatchDisassociateResourceCommand");
var de_DeleteAppsListCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  await (0, import_smithy_client.collectBody)(output.body, context);
  const response = {
    $metadata: deserializeMetadata(output)
  };
  return response;
}, "de_DeleteAppsListCommand");
var de_DeleteNotificationChannelCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  await (0, import_smithy_client.collectBody)(output.body, context);
  const response = {
    $metadata: deserializeMetadata(output)
  };
  return response;
}, "de_DeleteNotificationChannelCommand");
var de_DeletePolicyCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  await (0, import_smithy_client.collectBody)(output.body, context);
  const response = {
    $metadata: deserializeMetadata(output)
  };
  return response;
}, "de_DeletePolicyCommand");
var de_DeleteProtocolsListCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  await (0, import_smithy_client.collectBody)(output.body, context);
  const response = {
    $metadata: deserializeMetadata(output)
  };
  return response;
}, "de_DeleteProtocolsListCommand");
var de_DeleteResourceSetCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  await (0, import_smithy_client.collectBody)(output.body, context);
  const response = {
    $metadata: deserializeMetadata(output)
  };
  return response;
}, "de_DeleteResourceSetCommand");
var de_DisassociateAdminAccountCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  await (0, import_smithy_client.collectBody)(output.body, context);
  const response = {
    $metadata: deserializeMetadata(output)
  };
  return response;
}, "de_DisassociateAdminAccountCommand");
var de_DisassociateThirdPartyFirewallCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const data = await (0, import_core2.parseJsonBody)(output.body, context);
  let contents = {};
  contents = (0, import_smithy_client._json)(data);
  const response = {
    $metadata: deserializeMetadata(output),
    ...contents
  };
  return response;
}, "de_DisassociateThirdPartyFirewallCommand");
var de_GetAdminAccountCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const data = await (0, import_core2.parseJsonBody)(output.body, context);
  let contents = {};
  contents = (0, import_smithy_client._json)(data);
  const response = {
    $metadata: deserializeMetadata(output),
    ...contents
  };
  return response;
}, "de_GetAdminAccountCommand");
var de_GetAdminScopeCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const data = await (0, import_core2.parseJsonBody)(output.body, context);
  let contents = {};
  contents = (0, import_smithy_client._json)(data);
  const response = {
    $metadata: deserializeMetadata(output),
    ...contents
  };
  return response;
}, "de_GetAdminScopeCommand");
var de_GetAppsListCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const data = await (0, import_core2.parseJsonBody)(output.body, context);
  let contents = {};
  contents = de_GetAppsListResponse(data, context);
  const response = {
    $metadata: deserializeMetadata(output),
    ...contents
  };
  return response;
}, "de_GetAppsListCommand");
var de_GetComplianceDetailCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const data = await (0, import_core2.parseJsonBody)(output.body, context);
  let contents = {};
  contents = de_GetComplianceDetailResponse(data, context);
  const response = {
    $metadata: deserializeMetadata(output),
    ...contents
  };
  return response;
}, "de_GetComplianceDetailCommand");
var de_GetNotificationChannelCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const data = await (0, import_core2.parseJsonBody)(output.body, context);
  let contents = {};
  contents = (0, import_smithy_client._json)(data);
  const response = {
    $metadata: deserializeMetadata(output),
    ...contents
  };
  return response;
}, "de_GetNotificationChannelCommand");
var de_GetPolicyCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const data = await (0, import_core2.parseJsonBody)(output.body, context);
  let contents = {};
  contents = (0, import_smithy_client._json)(data);
  const response = {
    $metadata: deserializeMetadata(output),
    ...contents
  };
  return response;
}, "de_GetPolicyCommand");
var de_GetProtectionStatusCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const data = await (0, import_core2.parseJsonBody)(output.body, context);
  let contents = {};
  contents = (0, import_smithy_client._json)(data);
  const response = {
    $metadata: deserializeMetadata(output),
    ...contents
  };
  return response;
}, "de_GetProtectionStatusCommand");
var de_GetProtocolsListCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const data = await (0, import_core2.parseJsonBody)(output.body, context);
  let contents = {};
  contents = de_GetProtocolsListResponse(data, context);
  const response = {
    $metadata: deserializeMetadata(output),
    ...contents
  };
  return response;
}, "de_GetProtocolsListCommand");
var de_GetResourceSetCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const data = await (0, import_core2.parseJsonBody)(output.body, context);
  let contents = {};
  contents = de_GetResourceSetResponse(data, context);
  const response = {
    $metadata: deserializeMetadata(output),
    ...contents
  };
  return response;
}, "de_GetResourceSetCommand");
var de_GetThirdPartyFirewallAssociationStatusCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const data = await (0, import_core2.parseJsonBody)(output.body, context);
  let contents = {};
  contents = (0, import_smithy_client._json)(data);
  const response = {
    $metadata: deserializeMetadata(output),
    ...contents
  };
  return response;
}, "de_GetThirdPartyFirewallAssociationStatusCommand");
var de_GetViolationDetailsCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const data = await (0, import_core2.parseJsonBody)(output.body, context);
  let contents = {};
  contents = de_GetViolationDetailsResponse(data, context);
  const response = {
    $metadata: deserializeMetadata(output),
    ...contents
  };
  return response;
}, "de_GetViolationDetailsCommand");
var de_ListAdminAccountsForOrganizationCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const data = await (0, import_core2.parseJsonBody)(output.body, context);
  let contents = {};
  contents = (0, import_smithy_client._json)(data);
  const response = {
    $metadata: deserializeMetadata(output),
    ...contents
  };
  return response;
}, "de_ListAdminAccountsForOrganizationCommand");
var de_ListAdminsManagingAccountCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const data = await (0, import_core2.parseJsonBody)(output.body, context);
  let contents = {};
  contents = (0, import_smithy_client._json)(data);
  const response = {
    $metadata: deserializeMetadata(output),
    ...contents
  };
  return response;
}, "de_ListAdminsManagingAccountCommand");
var de_ListAppsListsCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const data = await (0, import_core2.parseJsonBody)(output.body, context);
  let contents = {};
  contents = (0, import_smithy_client._json)(data);
  const response = {
    $metadata: deserializeMetadata(output),
    ...contents
  };
  return response;
}, "de_ListAppsListsCommand");
var de_ListComplianceStatusCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const data = await (0, import_core2.parseJsonBody)(output.body, context);
  let contents = {};
  contents = de_ListComplianceStatusResponse(data, context);
  const response = {
    $metadata: deserializeMetadata(output),
    ...contents
  };
  return response;
}, "de_ListComplianceStatusCommand");
var de_ListDiscoveredResourcesCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const data = await (0, import_core2.parseJsonBody)(output.body, context);
  let contents = {};
  contents = (0, import_smithy_client._json)(data);
  const response = {
    $metadata: deserializeMetadata(output),
    ...contents
  };
  return response;
}, "de_ListDiscoveredResourcesCommand");
var de_ListMemberAccountsCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const data = await (0, import_core2.parseJsonBody)(output.body, context);
  let contents = {};
  contents = (0, import_smithy_client._json)(data);
  const response = {
    $metadata: deserializeMetadata(output),
    ...contents
  };
  return response;
}, "de_ListMemberAccountsCommand");
var de_ListPoliciesCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const data = await (0, import_core2.parseJsonBody)(output.body, context);
  let contents = {};
  contents = (0, import_smithy_client._json)(data);
  const response = {
    $metadata: deserializeMetadata(output),
    ...contents
  };
  return response;
}, "de_ListPoliciesCommand");
var de_ListProtocolsListsCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const data = await (0, import_core2.parseJsonBody)(output.body, context);
  let contents = {};
  contents = (0, import_smithy_client._json)(data);
  const response = {
    $metadata: deserializeMetadata(output),
    ...contents
  };
  return response;
}, "de_ListProtocolsListsCommand");
var de_ListResourceSetResourcesCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const data = await (0, import_core2.parseJsonBody)(output.body, context);
  let contents = {};
  contents = (0, import_smithy_client._json)(data);
  const response = {
    $metadata: deserializeMetadata(output),
    ...contents
  };
  return response;
}, "de_ListResourceSetResourcesCommand");
var de_ListResourceSetsCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const data = await (0, import_core2.parseJsonBody)(output.body, context);
  let contents = {};
  contents = de_ListResourceSetsResponse(data, context);
  const response = {
    $metadata: deserializeMetadata(output),
    ...contents
  };
  return response;
}, "de_ListResourceSetsCommand");
var de_ListTagsForResourceCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const data = await (0, import_core2.parseJsonBody)(output.body, context);
  let contents = {};
  contents = (0, import_smithy_client._json)(data);
  const response = {
    $metadata: deserializeMetadata(output),
    ...contents
  };
  return response;
}, "de_ListTagsForResourceCommand");
var de_ListThirdPartyFirewallFirewallPoliciesCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const data = await (0, import_core2.parseJsonBody)(output.body, context);
  let contents = {};
  contents = (0, import_smithy_client._json)(data);
  const response = {
    $metadata: deserializeMetadata(output),
    ...contents
  };
  return response;
}, "de_ListThirdPartyFirewallFirewallPoliciesCommand");
var de_PutAdminAccountCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  await (0, import_smithy_client.collectBody)(output.body, context);
  const response = {
    $metadata: deserializeMetadata(output)
  };
  return response;
}, "de_PutAdminAccountCommand");
var de_PutAppsListCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const data = await (0, import_core2.parseJsonBody)(output.body, context);
  let contents = {};
  contents = de_PutAppsListResponse(data, context);
  const response = {
    $metadata: deserializeMetadata(output),
    ...contents
  };
  return response;
}, "de_PutAppsListCommand");
var de_PutNotificationChannelCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  await (0, import_smithy_client.collectBody)(output.body, context);
  const response = {
    $metadata: deserializeMetadata(output)
  };
  return response;
}, "de_PutNotificationChannelCommand");
var de_PutPolicyCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const data = await (0, import_core2.parseJsonBody)(output.body, context);
  let contents = {};
  contents = (0, import_smithy_client._json)(data);
  const response = {
    $metadata: deserializeMetadata(output),
    ...contents
  };
  return response;
}, "de_PutPolicyCommand");
var de_PutProtocolsListCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const data = await (0, import_core2.parseJsonBody)(output.body, context);
  let contents = {};
  contents = de_PutProtocolsListResponse(data, context);
  const response = {
    $metadata: deserializeMetadata(output),
    ...contents
  };
  return response;
}, "de_PutProtocolsListCommand");
var de_PutResourceSetCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const data = await (0, import_core2.parseJsonBody)(output.body, context);
  let contents = {};
  contents = de_PutResourceSetResponse(data, context);
  const response = {
    $metadata: deserializeMetadata(output),
    ...contents
  };
  return response;
}, "de_PutResourceSetCommand");
var de_TagResourceCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const data = await (0, import_core2.parseJsonBody)(output.body, context);
  let contents = {};
  contents = (0, import_smithy_client._json)(data);
  const response = {
    $metadata: deserializeMetadata(output),
    ...contents
  };
  return response;
}, "de_TagResourceCommand");
var de_UntagResourceCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const data = await (0, import_core2.parseJsonBody)(output.body, context);
  let contents = {};
  contents = (0, import_smithy_client._json)(data);
  const response = {
    $metadata: deserializeMetadata(output),
    ...contents
  };
  return response;
}, "de_UntagResourceCommand");
var de_CommandError = /* @__PURE__ */ __name(async (output, context) => {
  const parsedOutput = {
    ...output,
    body: await (0, import_core2.parseJsonErrorBody)(output.body, context)
  };
  const errorCode = (0, import_core2.loadRestJsonErrorCode)(output, parsedOutput.body);
  switch (errorCode) {
    case "InternalErrorException":
    case "com.amazonaws.fms#InternalErrorException":
      throw await de_InternalErrorExceptionRes(parsedOutput, context);
    case "InvalidInputException":
    case "com.amazonaws.fms#InvalidInputException":
      throw await de_InvalidInputExceptionRes(parsedOutput, context);
    case "InvalidOperationException":
    case "com.amazonaws.fms#InvalidOperationException":
      throw await de_InvalidOperationExceptionRes(parsedOutput, context);
    case "LimitExceededException":
    case "com.amazonaws.fms#LimitExceededException":
      throw await de_LimitExceededExceptionRes(parsedOutput, context);
    case "ResourceNotFoundException":
    case "com.amazonaws.fms#ResourceNotFoundException":
      throw await de_ResourceNotFoundExceptionRes(parsedOutput, context);
    case "InvalidTypeException":
    case "com.amazonaws.fms#InvalidTypeException":
      throw await de_InvalidTypeExceptionRes(parsedOutput, context);
    default:
      const parsedBody = parsedOutput.body;
      return throwDefaultError({
        output,
        parsedBody,
        errorCode
      });
  }
}, "de_CommandError");
var de_InternalErrorExceptionRes = /* @__PURE__ */ __name(async (parsedOutput, context) => {
  const body = parsedOutput.body;
  const deserialized = (0, import_smithy_client._json)(body);
  const exception = new InternalErrorException({
    $metadata: deserializeMetadata(parsedOutput),
    ...deserialized
  });
  return (0, import_smithy_client.decorateServiceException)(exception, body);
}, "de_InternalErrorExceptionRes");
var de_InvalidInputExceptionRes = /* @__PURE__ */ __name(async (parsedOutput, context) => {
  const body = parsedOutput.body;
  const deserialized = (0, import_smithy_client._json)(body);
  const exception = new InvalidInputException({
    $metadata: deserializeMetadata(parsedOutput),
    ...deserialized
  });
  return (0, import_smithy_client.decorateServiceException)(exception, body);
}, "de_InvalidInputExceptionRes");
var de_InvalidOperationExceptionRes = /* @__PURE__ */ __name(async (parsedOutput, context) => {
  const body = parsedOutput.body;
  const deserialized = (0, import_smithy_client._json)(body);
  const exception = new InvalidOperationException({
    $metadata: deserializeMetadata(parsedOutput),
    ...deserialized
  });
  return (0, import_smithy_client.decorateServiceException)(exception, body);
}, "de_InvalidOperationExceptionRes");
var de_InvalidTypeExceptionRes = /* @__PURE__ */ __name(async (parsedOutput, context) => {
  const body = parsedOutput.body;
  const deserialized = (0, import_smithy_client._json)(body);
  const exception = new InvalidTypeException({
    $metadata: deserializeMetadata(parsedOutput),
    ...deserialized
  });
  return (0, import_smithy_client.decorateServiceException)(exception, body);
}, "de_InvalidTypeExceptionRes");
var de_LimitExceededExceptionRes = /* @__PURE__ */ __name(async (parsedOutput, context) => {
  const body = parsedOutput.body;
  const deserialized = (0, import_smithy_client._json)(body);
  const exception = new LimitExceededException({
    $metadata: deserializeMetadata(parsedOutput),
    ...deserialized
  });
  return (0, import_smithy_client.decorateServiceException)(exception, body);
}, "de_LimitExceededExceptionRes");
var de_ResourceNotFoundExceptionRes = /* @__PURE__ */ __name(async (parsedOutput, context) => {
  const body = parsedOutput.body;
  const deserialized = (0, import_smithy_client._json)(body);
  const exception = new ResourceNotFoundException({
    $metadata: deserializeMetadata(parsedOutput),
    ...deserialized
  });
  return (0, import_smithy_client.decorateServiceException)(exception, body);
}, "de_ResourceNotFoundExceptionRes");
var se_AppsListData = /* @__PURE__ */ __name((input, context) => {
  return (0, import_smithy_client.take)(input, {
    AppsList: import_smithy_client._json,
    CreateTime: (_) => Math.round(_.getTime() / 1e3),
    LastUpdateTime: (_) => Math.round(_.getTime() / 1e3),
    ListId: [],
    ListName: [],
    ListUpdateToken: [],
    PreviousAppsList: import_smithy_client._json
  });
}, "se_AppsListData");
var se_GetProtectionStatusRequest = /* @__PURE__ */ __name((input, context) => {
  return (0, import_smithy_client.take)(input, {
    EndTime: (_) => Math.round(_.getTime() / 1e3),
    MaxResults: [],
    MemberAccountId: [],
    NextToken: [],
    PolicyId: [],
    StartTime: (_) => Math.round(_.getTime() / 1e3)
  });
}, "se_GetProtectionStatusRequest");
var se_ProtocolsListData = /* @__PURE__ */ __name((input, context) => {
  return (0, import_smithy_client.take)(input, {
    CreateTime: (_) => Math.round(_.getTime() / 1e3),
    LastUpdateTime: (_) => Math.round(_.getTime() / 1e3),
    ListId: [],
    ListName: [],
    ListUpdateToken: [],
    PreviousProtocolsList: import_smithy_client._json,
    ProtocolsList: import_smithy_client._json
  });
}, "se_ProtocolsListData");
var se_PutAppsListRequest = /* @__PURE__ */ __name((input, context) => {
  return (0, import_smithy_client.take)(input, {
    AppsList: (_) => se_AppsListData(_, context),
    TagList: import_smithy_client._json
  });
}, "se_PutAppsListRequest");
var se_PutProtocolsListRequest = /* @__PURE__ */ __name((input, context) => {
  return (0, import_smithy_client.take)(input, {
    ProtocolsList: (_) => se_ProtocolsListData(_, context),
    TagList: import_smithy_client._json
  });
}, "se_PutProtocolsListRequest");
var se_PutResourceSetRequest = /* @__PURE__ */ __name((input, context) => {
  return (0, import_smithy_client.take)(input, {
    ResourceSet: (_) => se_ResourceSet(_, context),
    TagList: import_smithy_client._json
  });
}, "se_PutResourceSetRequest");
var se_ResourceSet = /* @__PURE__ */ __name((input, context) => {
  return (0, import_smithy_client.take)(input, {
    Description: [],
    Id: [],
    LastUpdateTime: (_) => Math.round(_.getTime() / 1e3),
    Name: [],
    ResourceSetStatus: [],
    ResourceTypeList: import_smithy_client._json,
    UpdateToken: []
  });
}, "se_ResourceSet");
var de_AppsListData = /* @__PURE__ */ __name((output, context) => {
  return (0, import_smithy_client.take)(output, {
    AppsList: import_smithy_client._json,
    CreateTime: (_) => (0, import_smithy_client.expectNonNull)((0, import_smithy_client.parseEpochTimestamp)((0, import_smithy_client.expectNumber)(_))),
    LastUpdateTime: (_) => (0, import_smithy_client.expectNonNull)((0, import_smithy_client.parseEpochTimestamp)((0, import_smithy_client.expectNumber)(_))),
    ListId: import_smithy_client.expectString,
    ListName: import_smithy_client.expectString,
    ListUpdateToken: import_smithy_client.expectString,
    PreviousAppsList: import_smithy_client._json
  });
}, "de_AppsListData");
var de_GetAppsListResponse = /* @__PURE__ */ __name((output, context) => {
  return (0, import_smithy_client.take)(output, {
    AppsList: (_) => de_AppsListData(_, context),
    AppsListArn: import_smithy_client.expectString
  });
}, "de_GetAppsListResponse");
var de_GetComplianceDetailResponse = /* @__PURE__ */ __name((output, context) => {
  return (0, import_smithy_client.take)(output, {
    PolicyComplianceDetail: (_) => de_PolicyComplianceDetail(_, context)
  });
}, "de_GetComplianceDetailResponse");
var de_GetProtocolsListResponse = /* @__PURE__ */ __name((output, context) => {
  return (0, import_smithy_client.take)(output, {
    ProtocolsList: (_) => de_ProtocolsListData(_, context),
    ProtocolsListArn: import_smithy_client.expectString
  });
}, "de_GetProtocolsListResponse");
var de_GetResourceSetResponse = /* @__PURE__ */ __name((output, context) => {
  return (0, import_smithy_client.take)(output, {
    ResourceSet: (_) => de_ResourceSet(_, context),
    ResourceSetArn: import_smithy_client.expectString
  });
}, "de_GetResourceSetResponse");
var de_GetViolationDetailsResponse = /* @__PURE__ */ __name((output, context) => {
  return (0, import_smithy_client.take)(output, {
    ViolationDetail: (_) => de_ViolationDetail(_, context)
  });
}, "de_GetViolationDetailsResponse");
var de_ListComplianceStatusResponse = /* @__PURE__ */ __name((output, context) => {
  return (0, import_smithy_client.take)(output, {
    NextToken: import_smithy_client.expectString,
    PolicyComplianceStatusList: (_) => de_PolicyComplianceStatusList(_, context)
  });
}, "de_ListComplianceStatusResponse");
var de_ListResourceSetsResponse = /* @__PURE__ */ __name((output, context) => {
  return (0, import_smithy_client.take)(output, {
    NextToken: import_smithy_client.expectString,
    ResourceSets: (_) => de_ResourceSetSummaryList(_, context)
  });
}, "de_ListResourceSetsResponse");
var de_PolicyComplianceDetail = /* @__PURE__ */ __name((output, context) => {
  return (0, import_smithy_client.take)(output, {
    EvaluationLimitExceeded: import_smithy_client.expectBoolean,
    ExpiredAt: (_) => (0, import_smithy_client.expectNonNull)((0, import_smithy_client.parseEpochTimestamp)((0, import_smithy_client.expectNumber)(_))),
    IssueInfoMap: import_smithy_client._json,
    MemberAccount: import_smithy_client.expectString,
    PolicyId: import_smithy_client.expectString,
    PolicyOwner: import_smithy_client.expectString,
    Violators: import_smithy_client._json
  });
}, "de_PolicyComplianceDetail");
var de_PolicyComplianceStatus = /* @__PURE__ */ __name((output, context) => {
  return (0, import_smithy_client.take)(output, {
    EvaluationResults: import_smithy_client._json,
    IssueInfoMap: import_smithy_client._json,
    LastUpdated: (_) => (0, import_smithy_client.expectNonNull)((0, import_smithy_client.parseEpochTimestamp)((0, import_smithy_client.expectNumber)(_))),
    MemberAccount: import_smithy_client.expectString,
    PolicyId: import_smithy_client.expectString,
    PolicyName: import_smithy_client.expectString,
    PolicyOwner: import_smithy_client.expectString
  });
}, "de_PolicyComplianceStatus");
var de_PolicyComplianceStatusList = /* @__PURE__ */ __name((output, context) => {
  const retVal = (output || []).filter((e) => e != null).map((entry) => {
    return de_PolicyComplianceStatus(entry, context);
  });
  return retVal;
}, "de_PolicyComplianceStatusList");
var de_ProtocolsListData = /* @__PURE__ */ __name((output, context) => {
  return (0, import_smithy_client.take)(output, {
    CreateTime: (_) => (0, import_smithy_client.expectNonNull)((0, import_smithy_client.parseEpochTimestamp)((0, import_smithy_client.expectNumber)(_))),
    LastUpdateTime: (_) => (0, import_smithy_client.expectNonNull)((0, import_smithy_client.parseEpochTimestamp)((0, import_smithy_client.expectNumber)(_))),
    ListId: import_smithy_client.expectString,
    ListName: import_smithy_client.expectString,
    ListUpdateToken: import_smithy_client.expectString,
    PreviousProtocolsList: import_smithy_client._json,
    ProtocolsList: import_smithy_client._json
  });
}, "de_ProtocolsListData");
var de_PutAppsListResponse = /* @__PURE__ */ __name((output, context) => {
  return (0, import_smithy_client.take)(output, {
    AppsList: (_) => de_AppsListData(_, context),
    AppsListArn: import_smithy_client.expectString
  });
}, "de_PutAppsListResponse");
var de_PutProtocolsListResponse = /* @__PURE__ */ __name((output, context) => {
  return (0, import_smithy_client.take)(output, {
    ProtocolsList: (_) => de_ProtocolsListData(_, context),
    ProtocolsListArn: import_smithy_client.expectString
  });
}, "de_PutProtocolsListResponse");
var de_PutResourceSetResponse = /* @__PURE__ */ __name((output, context) => {
  return (0, import_smithy_client.take)(output, {
    ResourceSet: (_) => de_ResourceSet(_, context),
    ResourceSetArn: import_smithy_client.expectString
  });
}, "de_PutResourceSetResponse");
var de_ResourceSet = /* @__PURE__ */ __name((output, context) => {
  return (0, import_smithy_client.take)(output, {
    Description: import_smithy_client.expectString,
    Id: import_smithy_client.expectString,
    LastUpdateTime: (_) => (0, import_smithy_client.expectNonNull)((0, import_smithy_client.parseEpochTimestamp)((0, import_smithy_client.expectNumber)(_))),
    Name: import_smithy_client.expectString,
    ResourceSetStatus: import_smithy_client.expectString,
    ResourceTypeList: import_smithy_client._json,
    UpdateToken: import_smithy_client.expectString
  });
}, "de_ResourceSet");
var de_ResourceSetSummary = /* @__PURE__ */ __name((output, context) => {
  return (0, import_smithy_client.take)(output, {
    Description: import_smithy_client.expectString,
    Id: import_smithy_client.expectString,
    LastUpdateTime: (_) => (0, import_smithy_client.expectNonNull)((0, import_smithy_client.parseEpochTimestamp)((0, import_smithy_client.expectNumber)(_))),
    Name: import_smithy_client.expectString,
    ResourceSetStatus: import_smithy_client.expectString
  });
}, "de_ResourceSetSummary");
var de_ResourceSetSummaryList = /* @__PURE__ */ __name((output, context) => {
  const retVal = (output || []).filter((e) => e != null).map((entry) => {
    return de_ResourceSetSummary(entry, context);
  });
  return retVal;
}, "de_ResourceSetSummaryList");
var de_ViolationDetail = /* @__PURE__ */ __name((output, context) => {
  return (0, import_smithy_client.take)(output, {
    MemberAccount: import_smithy_client.expectString,
    PolicyId: import_smithy_client.expectString,
    ResourceDescription: import_smithy_client.expectString,
    ResourceId: import_smithy_client.expectString,
    ResourceTags: import_smithy_client._json,
    ResourceType: import_smithy_client.expectString,
    ResourceViolations: import_smithy_client._json
  });
}, "de_ViolationDetail");
var deserializeMetadata = /* @__PURE__ */ __name((output) => ({
  httpStatusCode: output.statusCode,
  requestId: output.headers["x-amzn-requestid"] ?? output.headers["x-amzn-request-id"] ?? output.headers["x-amz-request-id"],
  extendedRequestId: output.headers["x-amz-id-2"],
  cfId: output.headers["x-amz-cf-id"]
}), "deserializeMetadata");
var throwDefaultError = (0, import_smithy_client.withBaseException)(FMSServiceException);
var buildHttpRpcRequest = /* @__PURE__ */ __name(async (context, headers, path, resolvedHostname, body) => {
  const { hostname, protocol = "https", port, path: basePath } = await context.endpoint();
  const contents = {
    protocol,
    hostname,
    port,
    method: "POST",
    path: basePath.endsWith("/") ? basePath.slice(0, -1) + path : basePath + path,
    headers
  };
  if (resolvedHostname !== void 0) {
    contents.hostname = resolvedHostname;
  }
  if (body !== void 0) {
    contents.body = body;
  }
  return new import_protocol_http.HttpRequest(contents);
}, "buildHttpRpcRequest");
function sharedHeaders(operation) {
  return {
    "content-type": "application/x-amz-json-1.1",
    "x-amz-target": `AWSFMS_20180101.${operation}`
  };
}
__name(sharedHeaders, "sharedHeaders");

// src/commands/AssociateAdminAccountCommand.ts
var _AssociateAdminAccountCommand = class _AssociateAdminAccountCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AWSFMS_20180101", "AssociateAdminAccount", {}).n("FMSClient", "AssociateAdminAccountCommand").f(void 0, void 0).ser(se_AssociateAdminAccountCommand).de(de_AssociateAdminAccountCommand).build() {
};
__name(_AssociateAdminAccountCommand, "AssociateAdminAccountCommand");
var AssociateAdminAccountCommand = _AssociateAdminAccountCommand;

// src/commands/AssociateThirdPartyFirewallCommand.ts




var _AssociateThirdPartyFirewallCommand = class _AssociateThirdPartyFirewallCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AWSFMS_20180101", "AssociateThirdPartyFirewall", {}).n("FMSClient", "AssociateThirdPartyFirewallCommand").f(void 0, void 0).ser(se_AssociateThirdPartyFirewallCommand).de(de_AssociateThirdPartyFirewallCommand).build() {
};
__name(_AssociateThirdPartyFirewallCommand, "AssociateThirdPartyFirewallCommand");
var AssociateThirdPartyFirewallCommand = _AssociateThirdPartyFirewallCommand;

// src/commands/BatchAssociateResourceCommand.ts




var _BatchAssociateResourceCommand = class _BatchAssociateResourceCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AWSFMS_20180101", "BatchAssociateResource", {}).n("FMSClient", "BatchAssociateResourceCommand").f(void 0, void 0).ser(se_BatchAssociateResourceCommand).de(de_BatchAssociateResourceCommand).build() {
};
__name(_BatchAssociateResourceCommand, "BatchAssociateResourceCommand");
var BatchAssociateResourceCommand = _BatchAssociateResourceCommand;

// src/commands/BatchDisassociateResourceCommand.ts




var _BatchDisassociateResourceCommand = class _BatchDisassociateResourceCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AWSFMS_20180101", "BatchDisassociateResource", {}).n("FMSClient", "BatchDisassociateResourceCommand").f(void 0, void 0).ser(se_BatchDisassociateResourceCommand).de(de_BatchDisassociateResourceCommand).build() {
};
__name(_BatchDisassociateResourceCommand, "BatchDisassociateResourceCommand");
var BatchDisassociateResourceCommand = _BatchDisassociateResourceCommand;

// src/commands/DeleteAppsListCommand.ts




var _DeleteAppsListCommand = class _DeleteAppsListCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AWSFMS_20180101", "DeleteAppsList", {}).n("FMSClient", "DeleteAppsListCommand").f(void 0, void 0).ser(se_DeleteAppsListCommand).de(de_DeleteAppsListCommand).build() {
};
__name(_DeleteAppsListCommand, "DeleteAppsListCommand");
var DeleteAppsListCommand = _DeleteAppsListCommand;

// src/commands/DeleteNotificationChannelCommand.ts




var _DeleteNotificationChannelCommand = class _DeleteNotificationChannelCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AWSFMS_20180101", "DeleteNotificationChannel", {}).n("FMSClient", "DeleteNotificationChannelCommand").f(void 0, void 0).ser(se_DeleteNotificationChannelCommand).de(de_DeleteNotificationChannelCommand).build() {
};
__name(_DeleteNotificationChannelCommand, "DeleteNotificationChannelCommand");
var DeleteNotificationChannelCommand = _DeleteNotificationChannelCommand;

// src/commands/DeletePolicyCommand.ts




var _DeletePolicyCommand = class _DeletePolicyCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AWSFMS_20180101", "DeletePolicy", {}).n("FMSClient", "DeletePolicyCommand").f(void 0, void 0).ser(se_DeletePolicyCommand).de(de_DeletePolicyCommand).build() {
};
__name(_DeletePolicyCommand, "DeletePolicyCommand");
var DeletePolicyCommand = _DeletePolicyCommand;

// src/commands/DeleteProtocolsListCommand.ts




var _DeleteProtocolsListCommand = class _DeleteProtocolsListCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AWSFMS_20180101", "DeleteProtocolsList", {}).n("FMSClient", "DeleteProtocolsListCommand").f(void 0, void 0).ser(se_DeleteProtocolsListCommand).de(de_DeleteProtocolsListCommand).build() {
};
__name(_DeleteProtocolsListCommand, "DeleteProtocolsListCommand");
var DeleteProtocolsListCommand = _DeleteProtocolsListCommand;

// src/commands/DeleteResourceSetCommand.ts




var _DeleteResourceSetCommand = class _DeleteResourceSetCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AWSFMS_20180101", "DeleteResourceSet", {}).n("FMSClient", "DeleteResourceSetCommand").f(void 0, void 0).ser(se_DeleteResourceSetCommand).de(de_DeleteResourceSetCommand).build() {
};
__name(_DeleteResourceSetCommand, "DeleteResourceSetCommand");
var DeleteResourceSetCommand = _DeleteResourceSetCommand;

// src/commands/DisassociateAdminAccountCommand.ts




var _DisassociateAdminAccountCommand = class _DisassociateAdminAccountCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AWSFMS_20180101", "DisassociateAdminAccount", {}).n("FMSClient", "DisassociateAdminAccountCommand").f(void 0, void 0).ser(se_DisassociateAdminAccountCommand).de(de_DisassociateAdminAccountCommand).build() {
};
__name(_DisassociateAdminAccountCommand, "DisassociateAdminAccountCommand");
var DisassociateAdminAccountCommand = _DisassociateAdminAccountCommand;

// src/commands/DisassociateThirdPartyFirewallCommand.ts




var _DisassociateThirdPartyFirewallCommand = class _DisassociateThirdPartyFirewallCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AWSFMS_20180101", "DisassociateThirdPartyFirewall", {}).n("FMSClient", "DisassociateThirdPartyFirewallCommand").f(void 0, void 0).ser(se_DisassociateThirdPartyFirewallCommand).de(de_DisassociateThirdPartyFirewallCommand).build() {
};
__name(_DisassociateThirdPartyFirewallCommand, "DisassociateThirdPartyFirewallCommand");
var DisassociateThirdPartyFirewallCommand = _DisassociateThirdPartyFirewallCommand;

// src/commands/GetAdminAccountCommand.ts




var _GetAdminAccountCommand = class _GetAdminAccountCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AWSFMS_20180101", "GetAdminAccount", {}).n("FMSClient", "GetAdminAccountCommand").f(void 0, void 0).ser(se_GetAdminAccountCommand).de(de_GetAdminAccountCommand).build() {
};
__name(_GetAdminAccountCommand, "GetAdminAccountCommand");
var GetAdminAccountCommand = _GetAdminAccountCommand;

// src/commands/GetAdminScopeCommand.ts




var _GetAdminScopeCommand = class _GetAdminScopeCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AWSFMS_20180101", "GetAdminScope", {}).n("FMSClient", "GetAdminScopeCommand").f(void 0, void 0).ser(se_GetAdminScopeCommand).de(de_GetAdminScopeCommand).build() {
};
__name(_GetAdminScopeCommand, "GetAdminScopeCommand");
var GetAdminScopeCommand = _GetAdminScopeCommand;

// src/commands/GetAppsListCommand.ts




var _GetAppsListCommand = class _GetAppsListCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AWSFMS_20180101", "GetAppsList", {}).n("FMSClient", "GetAppsListCommand").f(void 0, void 0).ser(se_GetAppsListCommand).de(de_GetAppsListCommand).build() {
};
__name(_GetAppsListCommand, "GetAppsListCommand");
var GetAppsListCommand = _GetAppsListCommand;

// src/commands/GetComplianceDetailCommand.ts




var _GetComplianceDetailCommand = class _GetComplianceDetailCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AWSFMS_20180101", "GetComplianceDetail", {}).n("FMSClient", "GetComplianceDetailCommand").f(void 0, void 0).ser(se_GetComplianceDetailCommand).de(de_GetComplianceDetailCommand).build() {
};
__name(_GetComplianceDetailCommand, "GetComplianceDetailCommand");
var GetComplianceDetailCommand = _GetComplianceDetailCommand;

// src/commands/GetNotificationChannelCommand.ts




var _GetNotificationChannelCommand = class _GetNotificationChannelCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AWSFMS_20180101", "GetNotificationChannel", {}).n("FMSClient", "GetNotificationChannelCommand").f(void 0, void 0).ser(se_GetNotificationChannelCommand).de(de_GetNotificationChannelCommand).build() {
};
__name(_GetNotificationChannelCommand, "GetNotificationChannelCommand");
var GetNotificationChannelCommand = _GetNotificationChannelCommand;

// src/commands/GetPolicyCommand.ts




var _GetPolicyCommand = class _GetPolicyCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AWSFMS_20180101", "GetPolicy", {}).n("FMSClient", "GetPolicyCommand").f(void 0, void 0).ser(se_GetPolicyCommand).de(de_GetPolicyCommand).build() {
};
__name(_GetPolicyCommand, "GetPolicyCommand");
var GetPolicyCommand = _GetPolicyCommand;

// src/commands/GetProtectionStatusCommand.ts




var _GetProtectionStatusCommand = class _GetProtectionStatusCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AWSFMS_20180101", "GetProtectionStatus", {}).n("FMSClient", "GetProtectionStatusCommand").f(void 0, void 0).ser(se_GetProtectionStatusCommand).de(de_GetProtectionStatusCommand).build() {
};
__name(_GetProtectionStatusCommand, "GetProtectionStatusCommand");
var GetProtectionStatusCommand = _GetProtectionStatusCommand;

// src/commands/GetProtocolsListCommand.ts




var _GetProtocolsListCommand = class _GetProtocolsListCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AWSFMS_20180101", "GetProtocolsList", {}).n("FMSClient", "GetProtocolsListCommand").f(void 0, void 0).ser(se_GetProtocolsListCommand).de(de_GetProtocolsListCommand).build() {
};
__name(_GetProtocolsListCommand, "GetProtocolsListCommand");
var GetProtocolsListCommand = _GetProtocolsListCommand;

// src/commands/GetResourceSetCommand.ts




var _GetResourceSetCommand = class _GetResourceSetCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AWSFMS_20180101", "GetResourceSet", {}).n("FMSClient", "GetResourceSetCommand").f(void 0, void 0).ser(se_GetResourceSetCommand).de(de_GetResourceSetCommand).build() {
};
__name(_GetResourceSetCommand, "GetResourceSetCommand");
var GetResourceSetCommand = _GetResourceSetCommand;

// src/commands/GetThirdPartyFirewallAssociationStatusCommand.ts




var _GetThirdPartyFirewallAssociationStatusCommand = class _GetThirdPartyFirewallAssociationStatusCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AWSFMS_20180101", "GetThirdPartyFirewallAssociationStatus", {}).n("FMSClient", "GetThirdPartyFirewallAssociationStatusCommand").f(void 0, void 0).ser(se_GetThirdPartyFirewallAssociationStatusCommand).de(de_GetThirdPartyFirewallAssociationStatusCommand).build() {
};
__name(_GetThirdPartyFirewallAssociationStatusCommand, "GetThirdPartyFirewallAssociationStatusCommand");
var GetThirdPartyFirewallAssociationStatusCommand = _GetThirdPartyFirewallAssociationStatusCommand;

// src/commands/GetViolationDetailsCommand.ts




var _GetViolationDetailsCommand = class _GetViolationDetailsCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AWSFMS_20180101", "GetViolationDetails", {}).n("FMSClient", "GetViolationDetailsCommand").f(void 0, void 0).ser(se_GetViolationDetailsCommand).de(de_GetViolationDetailsCommand).build() {
};
__name(_GetViolationDetailsCommand, "GetViolationDetailsCommand");
var GetViolationDetailsCommand = _GetViolationDetailsCommand;

// src/commands/ListAdminAccountsForOrganizationCommand.ts




var _ListAdminAccountsForOrganizationCommand = class _ListAdminAccountsForOrganizationCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AWSFMS_20180101", "ListAdminAccountsForOrganization", {}).n("FMSClient", "ListAdminAccountsForOrganizationCommand").f(void 0, void 0).ser(se_ListAdminAccountsForOrganizationCommand).de(de_ListAdminAccountsForOrganizationCommand).build() {
};
__name(_ListAdminAccountsForOrganizationCommand, "ListAdminAccountsForOrganizationCommand");
var ListAdminAccountsForOrganizationCommand = _ListAdminAccountsForOrganizationCommand;

// src/commands/ListAdminsManagingAccountCommand.ts




var _ListAdminsManagingAccountCommand = class _ListAdminsManagingAccountCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AWSFMS_20180101", "ListAdminsManagingAccount", {}).n("FMSClient", "ListAdminsManagingAccountCommand").f(void 0, void 0).ser(se_ListAdminsManagingAccountCommand).de(de_ListAdminsManagingAccountCommand).build() {
};
__name(_ListAdminsManagingAccountCommand, "ListAdminsManagingAccountCommand");
var ListAdminsManagingAccountCommand = _ListAdminsManagingAccountCommand;

// src/commands/ListAppsListsCommand.ts




var _ListAppsListsCommand = class _ListAppsListsCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AWSFMS_20180101", "ListAppsLists", {}).n("FMSClient", "ListAppsListsCommand").f(void 0, void 0).ser(se_ListAppsListsCommand).de(de_ListAppsListsCommand).build() {
};
__name(_ListAppsListsCommand, "ListAppsListsCommand");
var ListAppsListsCommand = _ListAppsListsCommand;

// src/commands/ListComplianceStatusCommand.ts




var _ListComplianceStatusCommand = class _ListComplianceStatusCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AWSFMS_20180101", "ListComplianceStatus", {}).n("FMSClient", "ListComplianceStatusCommand").f(void 0, void 0).ser(se_ListComplianceStatusCommand).de(de_ListComplianceStatusCommand).build() {
};
__name(_ListComplianceStatusCommand, "ListComplianceStatusCommand");
var ListComplianceStatusCommand = _ListComplianceStatusCommand;

// src/commands/ListDiscoveredResourcesCommand.ts




var _ListDiscoveredResourcesCommand = class _ListDiscoveredResourcesCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AWSFMS_20180101", "ListDiscoveredResources", {}).n("FMSClient", "ListDiscoveredResourcesCommand").f(void 0, void 0).ser(se_ListDiscoveredResourcesCommand).de(de_ListDiscoveredResourcesCommand).build() {
};
__name(_ListDiscoveredResourcesCommand, "ListDiscoveredResourcesCommand");
var ListDiscoveredResourcesCommand = _ListDiscoveredResourcesCommand;

// src/commands/ListMemberAccountsCommand.ts




var _ListMemberAccountsCommand = class _ListMemberAccountsCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AWSFMS_20180101", "ListMemberAccounts", {}).n("FMSClient", "ListMemberAccountsCommand").f(void 0, void 0).ser(se_ListMemberAccountsCommand).de(de_ListMemberAccountsCommand).build() {
};
__name(_ListMemberAccountsCommand, "ListMemberAccountsCommand");
var ListMemberAccountsCommand = _ListMemberAccountsCommand;

// src/commands/ListPoliciesCommand.ts




var _ListPoliciesCommand = class _ListPoliciesCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AWSFMS_20180101", "ListPolicies", {}).n("FMSClient", "ListPoliciesCommand").f(void 0, void 0).ser(se_ListPoliciesCommand).de(de_ListPoliciesCommand).build() {
};
__name(_ListPoliciesCommand, "ListPoliciesCommand");
var ListPoliciesCommand = _ListPoliciesCommand;

// src/commands/ListProtocolsListsCommand.ts




var _ListProtocolsListsCommand = class _ListProtocolsListsCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AWSFMS_20180101", "ListProtocolsLists", {}).n("FMSClient", "ListProtocolsListsCommand").f(void 0, void 0).ser(se_ListProtocolsListsCommand).de(de_ListProtocolsListsCommand).build() {
};
__name(_ListProtocolsListsCommand, "ListProtocolsListsCommand");
var ListProtocolsListsCommand = _ListProtocolsListsCommand;

// src/commands/ListResourceSetResourcesCommand.ts




var _ListResourceSetResourcesCommand = class _ListResourceSetResourcesCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AWSFMS_20180101", "ListResourceSetResources", {}).n("FMSClient", "ListResourceSetResourcesCommand").f(void 0, void 0).ser(se_ListResourceSetResourcesCommand).de(de_ListResourceSetResourcesCommand).build() {
};
__name(_ListResourceSetResourcesCommand, "ListResourceSetResourcesCommand");
var ListResourceSetResourcesCommand = _ListResourceSetResourcesCommand;

// src/commands/ListResourceSetsCommand.ts




var _ListResourceSetsCommand = class _ListResourceSetsCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AWSFMS_20180101", "ListResourceSets", {}).n("FMSClient", "ListResourceSetsCommand").f(void 0, void 0).ser(se_ListResourceSetsCommand).de(de_ListResourceSetsCommand).build() {
};
__name(_ListResourceSetsCommand, "ListResourceSetsCommand");
var ListResourceSetsCommand = _ListResourceSetsCommand;

// src/commands/ListTagsForResourceCommand.ts




var _ListTagsForResourceCommand = class _ListTagsForResourceCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AWSFMS_20180101", "ListTagsForResource", {}).n("FMSClient", "ListTagsForResourceCommand").f(void 0, void 0).ser(se_ListTagsForResourceCommand).de(de_ListTagsForResourceCommand).build() {
};
__name(_ListTagsForResourceCommand, "ListTagsForResourceCommand");
var ListTagsForResourceCommand = _ListTagsForResourceCommand;

// src/commands/ListThirdPartyFirewallFirewallPoliciesCommand.ts




var _ListThirdPartyFirewallFirewallPoliciesCommand = class _ListThirdPartyFirewallFirewallPoliciesCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AWSFMS_20180101", "ListThirdPartyFirewallFirewallPolicies", {}).n("FMSClient", "ListThirdPartyFirewallFirewallPoliciesCommand").f(void 0, void 0).ser(se_ListThirdPartyFirewallFirewallPoliciesCommand).de(de_ListThirdPartyFirewallFirewallPoliciesCommand).build() {
};
__name(_ListThirdPartyFirewallFirewallPoliciesCommand, "ListThirdPartyFirewallFirewallPoliciesCommand");
var ListThirdPartyFirewallFirewallPoliciesCommand = _ListThirdPartyFirewallFirewallPoliciesCommand;

// src/commands/PutAdminAccountCommand.ts




var _PutAdminAccountCommand = class _PutAdminAccountCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AWSFMS_20180101", "PutAdminAccount", {}).n("FMSClient", "PutAdminAccountCommand").f(void 0, void 0).ser(se_PutAdminAccountCommand).de(de_PutAdminAccountCommand).build() {
};
__name(_PutAdminAccountCommand, "PutAdminAccountCommand");
var PutAdminAccountCommand = _PutAdminAccountCommand;

// src/commands/PutAppsListCommand.ts




var _PutAppsListCommand = class _PutAppsListCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AWSFMS_20180101", "PutAppsList", {}).n("FMSClient", "PutAppsListCommand").f(void 0, void 0).ser(se_PutAppsListCommand).de(de_PutAppsListCommand).build() {
};
__name(_PutAppsListCommand, "PutAppsListCommand");
var PutAppsListCommand = _PutAppsListCommand;

// src/commands/PutNotificationChannelCommand.ts




var _PutNotificationChannelCommand = class _PutNotificationChannelCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AWSFMS_20180101", "PutNotificationChannel", {}).n("FMSClient", "PutNotificationChannelCommand").f(void 0, void 0).ser(se_PutNotificationChannelCommand).de(de_PutNotificationChannelCommand).build() {
};
__name(_PutNotificationChannelCommand, "PutNotificationChannelCommand");
var PutNotificationChannelCommand = _PutNotificationChannelCommand;

// src/commands/PutPolicyCommand.ts




var _PutPolicyCommand = class _PutPolicyCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AWSFMS_20180101", "PutPolicy", {}).n("FMSClient", "PutPolicyCommand").f(void 0, void 0).ser(se_PutPolicyCommand).de(de_PutPolicyCommand).build() {
};
__name(_PutPolicyCommand, "PutPolicyCommand");
var PutPolicyCommand = _PutPolicyCommand;

// src/commands/PutProtocolsListCommand.ts




var _PutProtocolsListCommand = class _PutProtocolsListCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AWSFMS_20180101", "PutProtocolsList", {}).n("FMSClient", "PutProtocolsListCommand").f(void 0, void 0).ser(se_PutProtocolsListCommand).de(de_PutProtocolsListCommand).build() {
};
__name(_PutProtocolsListCommand, "PutProtocolsListCommand");
var PutProtocolsListCommand = _PutProtocolsListCommand;

// src/commands/PutResourceSetCommand.ts




var _PutResourceSetCommand = class _PutResourceSetCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AWSFMS_20180101", "PutResourceSet", {}).n("FMSClient", "PutResourceSetCommand").f(void 0, void 0).ser(se_PutResourceSetCommand).de(de_PutResourceSetCommand).build() {
};
__name(_PutResourceSetCommand, "PutResourceSetCommand");
var PutResourceSetCommand = _PutResourceSetCommand;

// src/commands/TagResourceCommand.ts




var _TagResourceCommand = class _TagResourceCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AWSFMS_20180101", "TagResource", {}).n("FMSClient", "TagResourceCommand").f(void 0, void 0).ser(se_TagResourceCommand).de(de_TagResourceCommand).build() {
};
__name(_TagResourceCommand, "TagResourceCommand");
var TagResourceCommand = _TagResourceCommand;

// src/commands/UntagResourceCommand.ts




var _UntagResourceCommand = class _UntagResourceCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AWSFMS_20180101", "UntagResource", {}).n("FMSClient", "UntagResourceCommand").f(void 0, void 0).ser(se_UntagResourceCommand).de(de_UntagResourceCommand).build() {
};
__name(_UntagResourceCommand, "UntagResourceCommand");
var UntagResourceCommand = _UntagResourceCommand;

// src/FMS.ts
var commands = {
  AssociateAdminAccountCommand,
  AssociateThirdPartyFirewallCommand,
  BatchAssociateResourceCommand,
  BatchDisassociateResourceCommand,
  DeleteAppsListCommand,
  DeleteNotificationChannelCommand,
  DeletePolicyCommand,
  DeleteProtocolsListCommand,
  DeleteResourceSetCommand,
  DisassociateAdminAccountCommand,
  DisassociateThirdPartyFirewallCommand,
  GetAdminAccountCommand,
  GetAdminScopeCommand,
  GetAppsListCommand,
  GetComplianceDetailCommand,
  GetNotificationChannelCommand,
  GetPolicyCommand,
  GetProtectionStatusCommand,
  GetProtocolsListCommand,
  GetResourceSetCommand,
  GetThirdPartyFirewallAssociationStatusCommand,
  GetViolationDetailsCommand,
  ListAdminAccountsForOrganizationCommand,
  ListAdminsManagingAccountCommand,
  ListAppsListsCommand,
  ListComplianceStatusCommand,
  ListDiscoveredResourcesCommand,
  ListMemberAccountsCommand,
  ListPoliciesCommand,
  ListProtocolsListsCommand,
  ListResourceSetResourcesCommand,
  ListResourceSetsCommand,
  ListTagsForResourceCommand,
  ListThirdPartyFirewallFirewallPoliciesCommand,
  PutAdminAccountCommand,
  PutAppsListCommand,
  PutNotificationChannelCommand,
  PutPolicyCommand,
  PutProtocolsListCommand,
  PutResourceSetCommand,
  TagResourceCommand,
  UntagResourceCommand
};
var _FMS = class _FMS extends FMSClient {
};
__name(_FMS, "FMS");
var FMS = _FMS;
(0, import_smithy_client.createAggregatedClient)(commands, FMS);

// src/pagination/ListAdminAccountsForOrganizationPaginator.ts

var paginateListAdminAccountsForOrganization = (0, import_core.createPaginator)(FMSClient, ListAdminAccountsForOrganizationCommand, "NextToken", "NextToken", "MaxResults");

// src/pagination/ListAdminsManagingAccountPaginator.ts

var paginateListAdminsManagingAccount = (0, import_core.createPaginator)(FMSClient, ListAdminsManagingAccountCommand, "NextToken", "NextToken", "MaxResults");

// src/pagination/ListAppsListsPaginator.ts

var paginateListAppsLists = (0, import_core.createPaginator)(FMSClient, ListAppsListsCommand, "NextToken", "NextToken", "MaxResults");

// src/pagination/ListComplianceStatusPaginator.ts

var paginateListComplianceStatus = (0, import_core.createPaginator)(FMSClient, ListComplianceStatusCommand, "NextToken", "NextToken", "MaxResults");

// src/pagination/ListMemberAccountsPaginator.ts

var paginateListMemberAccounts = (0, import_core.createPaginator)(FMSClient, ListMemberAccountsCommand, "NextToken", "NextToken", "MaxResults");

// src/pagination/ListPoliciesPaginator.ts

var paginateListPolicies = (0, import_core.createPaginator)(FMSClient, ListPoliciesCommand, "NextToken", "NextToken", "MaxResults");

// src/pagination/ListProtocolsListsPaginator.ts

var paginateListProtocolsLists = (0, import_core.createPaginator)(FMSClient, ListProtocolsListsCommand, "NextToken", "NextToken", "MaxResults");

// src/pagination/ListThirdPartyFirewallFirewallPoliciesPaginator.ts

var paginateListThirdPartyFirewallFirewallPolicies = (0, import_core.createPaginator)(FMSClient, ListThirdPartyFirewallFirewallPoliciesCommand, "NextToken", "NextToken", "MaxResults");
// Annotate the CommonJS export names for ESM import in node:

0 && (module.exports = {
  FMSServiceException,
  __Client,
  FMSClient,
  FMS,
  $Command,
  AssociateAdminAccountCommand,
  AssociateThirdPartyFirewallCommand,
  BatchAssociateResourceCommand,
  BatchDisassociateResourceCommand,
  DeleteAppsListCommand,
  DeleteNotificationChannelCommand,
  DeletePolicyCommand,
  DeleteProtocolsListCommand,
  DeleteResourceSetCommand,
  DisassociateAdminAccountCommand,
  DisassociateThirdPartyFirewallCommand,
  GetAdminAccountCommand,
  GetAdminScopeCommand,
  GetAppsListCommand,
  GetComplianceDetailCommand,
  GetNotificationChannelCommand,
  GetPolicyCommand,
  GetProtectionStatusCommand,
  GetProtocolsListCommand,
  GetResourceSetCommand,
  GetThirdPartyFirewallAssociationStatusCommand,
  GetViolationDetailsCommand,
  ListAdminAccountsForOrganizationCommand,
  ListAdminsManagingAccountCommand,
  ListAppsListsCommand,
  ListComplianceStatusCommand,
  ListDiscoveredResourcesCommand,
  ListMemberAccountsCommand,
  ListPoliciesCommand,
  ListProtocolsListsCommand,
  ListResourceSetResourcesCommand,
  ListResourceSetsCommand,
  ListTagsForResourceCommand,
  ListThirdPartyFirewallFirewallPoliciesCommand,
  PutAdminAccountCommand,
  PutAppsListCommand,
  PutNotificationChannelCommand,
  PutPolicyCommand,
  PutProtocolsListCommand,
  PutResourceSetCommand,
  TagResourceCommand,
  UntagResourceCommand,
  paginateListAdminAccountsForOrganization,
  paginateListAdminsManagingAccount,
  paginateListAppsLists,
  paginateListComplianceStatus,
  paginateListMemberAccounts,
  paginateListPolicies,
  paginateListProtocolsLists,
  paginateListThirdPartyFirewallFirewallPolicies,
  AccountRoleStatus,
  OrganizationStatus,
  SecurityServiceType,
  InternalErrorException,
  InvalidInputException,
  InvalidOperationException,
  LimitExceededException,
  ResourceNotFoundException,
  ThirdPartyFirewall,
  ThirdPartyFirewallAssociationStatus,
  FailedItemReason,
  DependentServiceName,
  ViolationReason,
  CustomerPolicyScopeIdType,
  CustomerPolicyStatus,
  FirewallDeploymentModel,
  InvalidTypeException,
  ResourceSetStatus,
  MarketplaceSubscriptionOnboardingStatus,
  RemediationActionType,
  DestinationType,
  TargetType,
  RuleOrder,
  NetworkFirewallOverrideAction,
  PolicyComplianceStatusType
});
