// @azure/arm-peering/dist-esm/src/models/index.js
/*
 * Copyright (c) Microsoft Corporation.
 * Licensed under the MIT License.
 *
 * Code generated by Microsoft (R) AutoRest Code Generator.
 * Changes may cause incorrect behavior and will be lost if the code is regenerated.
 */
/** Known values of {@link Enum0} that the service accepts. */
export var KnownEnum0;
(function (KnownEnum0) {
    /** Available */
    KnownEnum0["Available"] = "Available";
    /** Unavailable */
    KnownEnum0["Unavailable"] = "Unavailable";
})(KnownEnum0 || (KnownEnum0 = {}));
/** Known values of {@link LegacyPeeringsKind} that the service accepts. */
export var KnownLegacyPeeringsKind;
(function (KnownLegacyPeeringsKind) {
    /** Direct */
    KnownLegacyPeeringsKind["Direct"] = "Direct";
    /** Exchange */
    KnownLegacyPeeringsKind["Exchange"] = "Exchange";
})(KnownLegacyPeeringsKind || (KnownLegacyPeeringsKind = {}));
/** Known values of {@link Tier} that the service accepts. */
export var KnownTier;
(function (KnownTier) {
    /** Basic */
    KnownTier["Basic"] = "Basic";
    /** Premium */
    KnownTier["Premium"] = "Premium";
})(KnownTier || (KnownTier = {}));
/** Known values of {@link Family} that the service accepts. */
export var KnownFamily;
(function (KnownFamily) {
    /** Direct */
    KnownFamily["Direct"] = "Direct";
    /** Exchange */
    KnownFamily["Exchange"] = "Exchange";
})(KnownFamily || (KnownFamily = {}));
/** Known values of {@link Size} that the service accepts. */
export var KnownSize;
(function (KnownSize) {
    /** Free */
    KnownSize["Free"] = "Free";
    /** Metered */
    KnownSize["Metered"] = "Metered";
    /** Unlimited */
    KnownSize["Unlimited"] = "Unlimited";
})(KnownSize || (KnownSize = {}));
/** Known values of {@link Kind} that the service accepts. */
export var KnownKind;
(function (KnownKind) {
    /** Direct */
    KnownKind["Direct"] = "Direct";
    /** Exchange */
    KnownKind["Exchange"] = "Exchange";
})(KnownKind || (KnownKind = {}));
/** Known values of {@link SessionAddressProvider} that the service accepts. */
export var KnownSessionAddressProvider;
(function (KnownSessionAddressProvider) {
    /** Microsoft */
    KnownSessionAddressProvider["Microsoft"] = "Microsoft";
    /** Peer */
    KnownSessionAddressProvider["Peer"] = "Peer";
})(KnownSessionAddressProvider || (KnownSessionAddressProvider = {}));
/** Known values of {@link ConnectionState} that the service accepts. */
export var KnownConnectionState;
(function (KnownConnectionState) {
    /** None */
    KnownConnectionState["None"] = "None";
    /** PendingApproval */
    KnownConnectionState["PendingApproval"] = "PendingApproval";
    /** Approved */
    KnownConnectionState["Approved"] = "Approved";
    /** ProvisioningStarted */
    KnownConnectionState["ProvisioningStarted"] = "ProvisioningStarted";
    /** ProvisioningFailed */
    KnownConnectionState["ProvisioningFailed"] = "ProvisioningFailed";
    /** ProvisioningCompleted */
    KnownConnectionState["ProvisioningCompleted"] = "ProvisioningCompleted";
    /** Validating */
    KnownConnectionState["Validating"] = "Validating";
    /** Active */
    KnownConnectionState["Active"] = "Active";
})(KnownConnectionState || (KnownConnectionState = {}));
/** Known values of {@link SessionStateV4} that the service accepts. */
export var KnownSessionStateV4;
(function (KnownSessionStateV4) {
    /** None */
    KnownSessionStateV4["None"] = "None";
    /** Idle */
    KnownSessionStateV4["Idle"] = "Idle";
    /** Connect */
    KnownSessionStateV4["Connect"] = "Connect";
    /** Active */
    KnownSessionStateV4["Active"] = "Active";
    /** OpenSent */
    KnownSessionStateV4["OpenSent"] = "OpenSent";
    /** OpenConfirm */
    KnownSessionStateV4["OpenConfirm"] = "OpenConfirm";
    /** OpenReceived */
    KnownSessionStateV4["OpenReceived"] = "OpenReceived";
    /** Established */
    KnownSessionStateV4["Established"] = "Established";
    /** PendingAdd */
    KnownSessionStateV4["PendingAdd"] = "PendingAdd";
    /** PendingUpdate */
    KnownSessionStateV4["PendingUpdate"] = "PendingUpdate";
    /** PendingRemove */
    KnownSessionStateV4["PendingRemove"] = "PendingRemove";
})(KnownSessionStateV4 || (KnownSessionStateV4 = {}));
/** Known values of {@link SessionStateV6} that the service accepts. */
export var KnownSessionStateV6;
(function (KnownSessionStateV6) {
    /** None */
    KnownSessionStateV6["None"] = "None";
    /** Idle */
    KnownSessionStateV6["Idle"] = "Idle";
    /** Connect */
    KnownSessionStateV6["Connect"] = "Connect";
    /** Active */
    KnownSessionStateV6["Active"] = "Active";
    /** OpenSent */
    KnownSessionStateV6["OpenSent"] = "OpenSent";
    /** OpenConfirm */
    KnownSessionStateV6["OpenConfirm"] = "OpenConfirm";
    /** OpenReceived */
    KnownSessionStateV6["OpenReceived"] = "OpenReceived";
    /** Established */
    KnownSessionStateV6["Established"] = "Established";
    /** PendingAdd */
    KnownSessionStateV6["PendingAdd"] = "PendingAdd";
    /** PendingUpdate */
    KnownSessionStateV6["PendingUpdate"] = "PendingUpdate";
    /** PendingRemove */
    KnownSessionStateV6["PendingRemove"] = "PendingRemove";
})(KnownSessionStateV6 || (KnownSessionStateV6 = {}));
/** Known values of {@link DirectPeeringType} that the service accepts. */
export var KnownDirectPeeringType;
(function (KnownDirectPeeringType) {
    /** Edge */
    KnownDirectPeeringType["Edge"] = "Edge";
    /** Transit */
    KnownDirectPeeringType["Transit"] = "Transit";
    /** Cdn */
    KnownDirectPeeringType["Cdn"] = "Cdn";
    /** Internal */
    KnownDirectPeeringType["Internal"] = "Internal";
    /** Ix */
    KnownDirectPeeringType["Ix"] = "Ix";
    /** IxRs */
    KnownDirectPeeringType["IxRs"] = "IxRs";
    /** Voice */
    KnownDirectPeeringType["Voice"] = "Voice";
})(KnownDirectPeeringType || (KnownDirectPeeringType = {}));
/** Known values of {@link ProvisioningState} that the service accepts. */
export var KnownProvisioningState;
(function (KnownProvisioningState) {
    /** Succeeded */
    KnownProvisioningState["Succeeded"] = "Succeeded";
    /** Updating */
    KnownProvisioningState["Updating"] = "Updating";
    /** Deleting */
    KnownProvisioningState["Deleting"] = "Deleting";
    /** Failed */
    KnownProvisioningState["Failed"] = "Failed";
})(KnownProvisioningState || (KnownProvisioningState = {}));
/** Known values of {@link LookingGlassCommand} that the service accepts. */
export var KnownLookingGlassCommand;
(function (KnownLookingGlassCommand) {
    /** Traceroute */
    KnownLookingGlassCommand["Traceroute"] = "Traceroute";
    /** Ping */
    KnownLookingGlassCommand["Ping"] = "Ping";
    /** BgpRoute */
    KnownLookingGlassCommand["BgpRoute"] = "BgpRoute";
})(KnownLookingGlassCommand || (KnownLookingGlassCommand = {}));
/** Known values of {@link LookingGlassSourceType} that the service accepts. */
export var KnownLookingGlassSourceType;
(function (KnownLookingGlassSourceType) {
    /** EdgeSite */
    KnownLookingGlassSourceType["EdgeSite"] = "EdgeSite";
    /** AzureRegion */
    KnownLookingGlassSourceType["AzureRegion"] = "AzureRegion";
})(KnownLookingGlassSourceType || (KnownLookingGlassSourceType = {}));
/** Known values of {@link Command} that the service accepts. */
export var KnownCommand;
(function (KnownCommand) {
    /** Traceroute */
    KnownCommand["Traceroute"] = "Traceroute";
    /** Ping */
    KnownCommand["Ping"] = "Ping";
    /** BgpRoute */
    KnownCommand["BgpRoute"] = "BgpRoute";
})(KnownCommand || (KnownCommand = {}));
/** Known values of {@link Role} that the service accepts. */
export var KnownRole;
(function (KnownRole) {
    /** Noc */
    KnownRole["Noc"] = "Noc";
    /** Policy */
    KnownRole["Policy"] = "Policy";
    /** Technical */
    KnownRole["Technical"] = "Technical";
    /** Service */
    KnownRole["Service"] = "Service";
    /** Escalation */
    KnownRole["Escalation"] = "Escalation";
    /** Other */
    KnownRole["Other"] = "Other";
})(KnownRole || (KnownRole = {}));
/** Known values of {@link ValidationState} that the service accepts. */
export var KnownValidationState;
(function (KnownValidationState) {
    /** None */
    KnownValidationState["None"] = "None";
    /** Pending */
    KnownValidationState["Pending"] = "Pending";
    /** Approved */
    KnownValidationState["Approved"] = "Approved";
    /** Failed */
    KnownValidationState["Failed"] = "Failed";
})(KnownValidationState || (KnownValidationState = {}));
/** Known values of {@link PeeringLocationsKind} that the service accepts. */
export var KnownPeeringLocationsKind;
(function (KnownPeeringLocationsKind) {
    /** Direct */
    KnownPeeringLocationsKind["Direct"] = "Direct";
    /** Exchange */
    KnownPeeringLocationsKind["Exchange"] = "Exchange";
})(KnownPeeringLocationsKind || (KnownPeeringLocationsKind = {}));
/** Known values of {@link PeeringLocationsDirectPeeringType} that the service accepts. */
export var KnownPeeringLocationsDirectPeeringType;
(function (KnownPeeringLocationsDirectPeeringType) {
    /** Edge */
    KnownPeeringLocationsDirectPeeringType["Edge"] = "Edge";
    /** Transit */
    KnownPeeringLocationsDirectPeeringType["Transit"] = "Transit";
    /** Cdn */
    KnownPeeringLocationsDirectPeeringType["Cdn"] = "Cdn";
    /** Internal */
    KnownPeeringLocationsDirectPeeringType["Internal"] = "Internal";
    /** Ix */
    KnownPeeringLocationsDirectPeeringType["Ix"] = "Ix";
    /** IxRs */
    KnownPeeringLocationsDirectPeeringType["IxRs"] = "IxRs";
    /** Voice */
    KnownPeeringLocationsDirectPeeringType["Voice"] = "Voice";
})(KnownPeeringLocationsDirectPeeringType || (KnownPeeringLocationsDirectPeeringType = {}));
/** Known values of {@link PrefixValidationState} that the service accepts. */
export var KnownPrefixValidationState;
(function (KnownPrefixValidationState) {
    /** None */
    KnownPrefixValidationState["None"] = "None";
    /** Invalid */
    KnownPrefixValidationState["Invalid"] = "Invalid";
    /** Verified */
    KnownPrefixValidationState["Verified"] = "Verified";
    /** Failed */
    KnownPrefixValidationState["Failed"] = "Failed";
    /** Pending */
    KnownPrefixValidationState["Pending"] = "Pending";
    /** Warning */
    KnownPrefixValidationState["Warning"] = "Warning";
    /** Unknown */
    KnownPrefixValidationState["Unknown"] = "Unknown";
})(KnownPrefixValidationState || (KnownPrefixValidationState = {}));
/** Known values of {@link LearnedType} that the service accepts. */
export var KnownLearnedType;
(function (KnownLearnedType) {
    /** None */
    KnownLearnedType["None"] = "None";
    /** ViaServiceProvider */
    KnownLearnedType["ViaServiceProvider"] = "ViaServiceProvider";
    /** ViaSession */
    KnownLearnedType["ViaSession"] = "ViaSession";
})(KnownLearnedType || (KnownLearnedType = {}));
