// @azure/arm-apimanagement/dist-esm/src/operations/authorization.js
/*
 * Copyright (c) Microsoft Corporation.
 * Licensed under the MIT License.
 *
 * Code generated by Microsoft (R) AutoRest Code Generator.
 * Changes may cause incorrect behavior and will be lost if the code is regenerated.
 */
import { __asyncDelegator, __asyncGenerator, __asyncValues, __await } from "tslib";
import { setContinuationToken } from "../pagingHelper";
import * as coreClient from "@azure/core-client";
import * as Mappers from "../models/mappers";
import * as Parameters from "../models/parameters";
/// <reference lib="esnext.asynciterable" />
/** Class containing Authorization operations. */
export class AuthorizationImpl {
    /**
     * Initialize a new instance of the class Authorization class.
     * @param client Reference to the service client
     */
    constructor(client) {
        this.client = client;
    }
    /**
     * Lists a collection of authorization providers defined within a authorization provider.
     * @param resourceGroupName The name of the resource group. The name is case insensitive.
     * @param serviceName The name of the API Management service.
     * @param authorizationProviderId Identifier of the authorization provider.
     * @param options The options parameters.
     */
    listByAuthorizationProvider(resourceGroupName, serviceName, authorizationProviderId, options) {
        const iter = this.listByAuthorizationProviderPagingAll(resourceGroupName, serviceName, authorizationProviderId, options);
        return {
            next() {
                return iter.next();
            },
            [Symbol.asyncIterator]() {
                return this;
            },
            byPage: (settings) => {
                if (settings === null || settings === void 0 ? void 0 : settings.maxPageSize) {
                    throw new Error("maxPageSize is not supported by this operation.");
                }
                return this.listByAuthorizationProviderPagingPage(resourceGroupName, serviceName, authorizationProviderId, options, settings);
            }
        };
    }
    listByAuthorizationProviderPagingPage(resourceGroupName, serviceName, authorizationProviderId, options, settings) {
        return __asyncGenerator(this, arguments, function* listByAuthorizationProviderPagingPage_1() {
            let result;
            let continuationToken = settings === null || settings === void 0 ? void 0 : settings.continuationToken;
            if (!continuationToken) {
                result = yield __await(this._listByAuthorizationProvider(resourceGroupName, serviceName, authorizationProviderId, options));
                let page = result.value || [];
                continuationToken = result.nextLink;
                setContinuationToken(page, continuationToken);
                yield yield __await(page);
            }
            while (continuationToken) {
                result = yield __await(this._listByAuthorizationProviderNext(resourceGroupName, serviceName, authorizationProviderId, continuationToken, options));
                continuationToken = result.nextLink;
                let page = result.value || [];
                setContinuationToken(page, continuationToken);
                yield yield __await(page);
            }
        });
    }
    listByAuthorizationProviderPagingAll(resourceGroupName, serviceName, authorizationProviderId, options) {
        return __asyncGenerator(this, arguments, function* listByAuthorizationProviderPagingAll_1() {
            var _a, e_1, _b, _c;
            try {
                for (var _d = true, _e = __asyncValues(this.listByAuthorizationProviderPagingPage(resourceGroupName, serviceName, authorizationProviderId, options)), _f; _f = yield __await(_e.next()), _a = _f.done, !_a;) {
                    _c = _f.value;
                    _d = false;
                    try {
                        const page = _c;
                        yield __await(yield* __asyncDelegator(__asyncValues(page)));
                    }
                    finally {
                        _d = true;
                    }
                }
            }
            catch (e_1_1) { e_1 = { error: e_1_1 }; }
            finally {
                try {
                    if (!_d && !_a && (_b = _e.return)) yield __await(_b.call(_e));
                }
                finally { if (e_1) throw e_1.error; }
            }
        });
    }
    /**
     * Lists a collection of authorization providers defined within a authorization provider.
     * @param resourceGroupName The name of the resource group. The name is case insensitive.
     * @param serviceName The name of the API Management service.
     * @param authorizationProviderId Identifier of the authorization provider.
     * @param options The options parameters.
     */
    _listByAuthorizationProvider(resourceGroupName, serviceName, authorizationProviderId, options) {
        return this.client.sendOperationRequest({ resourceGroupName, serviceName, authorizationProviderId, options }, listByAuthorizationProviderOperationSpec);
    }
    /**
     * Gets the details of the authorization specified by its identifier.
     * @param resourceGroupName The name of the resource group. The name is case insensitive.
     * @param serviceName The name of the API Management service.
     * @param authorizationProviderId Identifier of the authorization provider.
     * @param authorizationId Identifier of the authorization.
     * @param options The options parameters.
     */
    get(resourceGroupName, serviceName, authorizationProviderId, authorizationId, options) {
        return this.client.sendOperationRequest({
            resourceGroupName,
            serviceName,
            authorizationProviderId,
            authorizationId,
            options
        }, getOperationSpec);
    }
    /**
     * Creates or updates authorization.
     * @param resourceGroupName The name of the resource group. The name is case insensitive.
     * @param serviceName The name of the API Management service.
     * @param authorizationProviderId Identifier of the authorization provider.
     * @param authorizationId Identifier of the authorization.
     * @param parameters Create parameters.
     * @param options The options parameters.
     */
    createOrUpdate(resourceGroupName, serviceName, authorizationProviderId, authorizationId, parameters, options) {
        return this.client.sendOperationRequest({
            resourceGroupName,
            serviceName,
            authorizationProviderId,
            authorizationId,
            parameters,
            options
        }, createOrUpdateOperationSpec);
    }
    /**
     * Deletes specific Authorization from the Authorization provider.
     * @param resourceGroupName The name of the resource group. The name is case insensitive.
     * @param serviceName The name of the API Management service.
     * @param authorizationProviderId Identifier of the authorization provider.
     * @param authorizationId Identifier of the authorization.
     * @param ifMatch ETag of the Entity. ETag should match the current entity state from the header
     *                response of the GET request or it should be * for unconditional update.
     * @param options The options parameters.
     */
    delete(resourceGroupName, serviceName, authorizationProviderId, authorizationId, ifMatch, options) {
        return this.client.sendOperationRequest({
            resourceGroupName,
            serviceName,
            authorizationProviderId,
            authorizationId,
            ifMatch,
            options
        }, deleteOperationSpec);
    }
    /**
     * Confirm valid consent code to suppress Authorizations anti-phishing page.
     * @param resourceGroupName The name of the resource group. The name is case insensitive.
     * @param serviceName The name of the API Management service.
     * @param authorizationProviderId Identifier of the authorization provider.
     * @param authorizationId Identifier of the authorization.
     * @param parameters Create parameters.
     * @param options The options parameters.
     */
    confirmConsentCode(resourceGroupName, serviceName, authorizationProviderId, authorizationId, parameters, options) {
        return this.client.sendOperationRequest({
            resourceGroupName,
            serviceName,
            authorizationProviderId,
            authorizationId,
            parameters,
            options
        }, confirmConsentCodeOperationSpec);
    }
    /**
     * ListByAuthorizationProviderNext
     * @param resourceGroupName The name of the resource group. The name is case insensitive.
     * @param serviceName The name of the API Management service.
     * @param authorizationProviderId Identifier of the authorization provider.
     * @param nextLink The nextLink from the previous successful call to the ListByAuthorizationProvider
     *                 method.
     * @param options The options parameters.
     */
    _listByAuthorizationProviderNext(resourceGroupName, serviceName, authorizationProviderId, nextLink, options) {
        return this.client.sendOperationRequest({
            resourceGroupName,
            serviceName,
            authorizationProviderId,
            nextLink,
            options
        }, listByAuthorizationProviderNextOperationSpec);
    }
}
// Operation Specifications
const serializer = coreClient.createSerializer(Mappers, /* isXml */ false);
const listByAuthorizationProviderOperationSpec = {
    path: "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.ApiManagement/service/{serviceName}/authorizationProviders/{authorizationProviderId}/authorizations",
    httpMethod: "GET",
    responses: {
        200: {
            bodyMapper: Mappers.AuthorizationCollection
        },
        default: {
            bodyMapper: Mappers.ErrorResponse
        }
    },
    queryParameters: [
        Parameters.filter,
        Parameters.top,
        Parameters.skip,
        Parameters.apiVersion
    ],
    urlParameters: [
        Parameters.$host,
        Parameters.resourceGroupName,
        Parameters.serviceName,
        Parameters.subscriptionId,
        Parameters.authorizationProviderId
    ],
    headerParameters: [Parameters.accept],
    serializer
};
const getOperationSpec = {
    path: "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.ApiManagement/service/{serviceName}/authorizationProviders/{authorizationProviderId}/authorizations/{authorizationId}",
    httpMethod: "GET",
    responses: {
        200: {
            bodyMapper: Mappers.AuthorizationContract,
            headersMapper: Mappers.AuthorizationGetHeaders
        },
        default: {
            bodyMapper: Mappers.ErrorResponse
        }
    },
    queryParameters: [Parameters.apiVersion],
    urlParameters: [
        Parameters.$host,
        Parameters.resourceGroupName,
        Parameters.serviceName,
        Parameters.subscriptionId,
        Parameters.authorizationProviderId,
        Parameters.authorizationId
    ],
    headerParameters: [Parameters.accept],
    serializer
};
const createOrUpdateOperationSpec = {
    path: "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.ApiManagement/service/{serviceName}/authorizationProviders/{authorizationProviderId}/authorizations/{authorizationId}",
    httpMethod: "PUT",
    responses: {
        200: {
            bodyMapper: Mappers.AuthorizationContract,
            headersMapper: Mappers.AuthorizationCreateOrUpdateHeaders
        },
        201: {
            bodyMapper: Mappers.AuthorizationContract,
            headersMapper: Mappers.AuthorizationCreateOrUpdateHeaders
        },
        default: {
            bodyMapper: Mappers.ErrorResponse
        }
    },
    requestBody: Parameters.parameters23,
    queryParameters: [Parameters.apiVersion],
    urlParameters: [
        Parameters.$host,
        Parameters.resourceGroupName,
        Parameters.serviceName,
        Parameters.subscriptionId,
        Parameters.authorizationProviderId,
        Parameters.authorizationId
    ],
    headerParameters: [
        Parameters.accept,
        Parameters.contentType,
        Parameters.ifMatch
    ],
    mediaType: "json",
    serializer
};
const deleteOperationSpec = {
    path: "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.ApiManagement/service/{serviceName}/authorizationProviders/{authorizationProviderId}/authorizations/{authorizationId}",
    httpMethod: "DELETE",
    responses: {
        200: {},
        204: {},
        default: {
            bodyMapper: Mappers.ErrorResponse
        }
    },
    queryParameters: [Parameters.apiVersion],
    urlParameters: [
        Parameters.$host,
        Parameters.resourceGroupName,
        Parameters.serviceName,
        Parameters.subscriptionId,
        Parameters.authorizationProviderId,
        Parameters.authorizationId
    ],
    headerParameters: [Parameters.accept, Parameters.ifMatch1],
    serializer
};
const confirmConsentCodeOperationSpec = {
    path: "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.ApiManagement/service/{serviceName}/authorizationProviders/{authorizationProviderId}/authorizations/{authorizationId}/confirmConsentCode",
    httpMethod: "POST",
    responses: {
        200: {
            headersMapper: Mappers.AuthorizationConfirmConsentCodeHeaders
        },
        default: {
            bodyMapper: Mappers.ErrorResponse
        }
    },
    requestBody: Parameters.parameters24,
    queryParameters: [Parameters.apiVersion],
    urlParameters: [
        Parameters.$host,
        Parameters.resourceGroupName,
        Parameters.serviceName,
        Parameters.subscriptionId,
        Parameters.authorizationProviderId,
        Parameters.authorizationId
    ],
    headerParameters: [Parameters.accept, Parameters.contentType],
    mediaType: "json",
    serializer
};
const listByAuthorizationProviderNextOperationSpec = {
    path: "{nextLink}",
    httpMethod: "GET",
    responses: {
        200: {
            bodyMapper: Mappers.AuthorizationCollection
        },
        default: {
            bodyMapper: Mappers.ErrorResponse
        }
    },
    urlParameters: [
        Parameters.$host,
        Parameters.resourceGroupName,
        Parameters.serviceName,
        Parameters.subscriptionId,
        Parameters.nextLink,
        Parameters.authorizationProviderId
    ],
    headerParameters: [Parameters.accept],
    serializer
};
