// @azure/arm-apimanagement/dist-esm/src/operations/authorizationAccessPolicy.js
/*
 * Copyright (c) Microsoft Corporation.
 * Licensed under the MIT License.
 *
 * Code generated by Microsoft (R) AutoRest Code Generator.
 * Changes may cause incorrect behavior and will be lost if the code is regenerated.
 */
import { __asyncDelegator, __asyncGenerator, __asyncValues, __await } from "tslib";
import { setContinuationToken } from "../pagingHelper";
import * as coreClient from "@azure/core-client";
import * as Mappers from "../models/mappers";
import * as Parameters from "../models/parameters";
/// <reference lib="esnext.asynciterable" />
/** Class containing AuthorizationAccessPolicy operations. */
export class AuthorizationAccessPolicyImpl {
    /**
     * Initialize a new instance of the class AuthorizationAccessPolicy class.
     * @param client Reference to the service client
     */
    constructor(client) {
        this.client = client;
    }
    /**
     * Lists a collection of authorization access policy defined within a authorization.
     * @param resourceGroupName The name of the resource group. The name is case insensitive.
     * @param serviceName The name of the API Management service.
     * @param authorizationProviderId Identifier of the authorization provider.
     * @param authorizationId Identifier of the authorization.
     * @param options The options parameters.
     */
    listByAuthorization(resourceGroupName, serviceName, authorizationProviderId, authorizationId, options) {
        const iter = this.listByAuthorizationPagingAll(resourceGroupName, serviceName, authorizationProviderId, authorizationId, options);
        return {
            next() {
                return iter.next();
            },
            [Symbol.asyncIterator]() {
                return this;
            },
            byPage: (settings) => {
                if (settings === null || settings === void 0 ? void 0 : settings.maxPageSize) {
                    throw new Error("maxPageSize is not supported by this operation.");
                }
                return this.listByAuthorizationPagingPage(resourceGroupName, serviceName, authorizationProviderId, authorizationId, options, settings);
            }
        };
    }
    listByAuthorizationPagingPage(resourceGroupName, serviceName, authorizationProviderId, authorizationId, options, settings) {
        return __asyncGenerator(this, arguments, function* listByAuthorizationPagingPage_1() {
            let result;
            let continuationToken = settings === null || settings === void 0 ? void 0 : settings.continuationToken;
            if (!continuationToken) {
                result = yield __await(this._listByAuthorization(resourceGroupName, serviceName, authorizationProviderId, authorizationId, options));
                let page = result.value || [];
                continuationToken = result.nextLink;
                setContinuationToken(page, continuationToken);
                yield yield __await(page);
            }
            while (continuationToken) {
                result = yield __await(this._listByAuthorizationNext(resourceGroupName, serviceName, authorizationProviderId, authorizationId, continuationToken, options));
                continuationToken = result.nextLink;
                let page = result.value || [];
                setContinuationToken(page, continuationToken);
                yield yield __await(page);
            }
        });
    }
    listByAuthorizationPagingAll(resourceGroupName, serviceName, authorizationProviderId, authorizationId, options) {
        return __asyncGenerator(this, arguments, function* listByAuthorizationPagingAll_1() {
            var _a, e_1, _b, _c;
            try {
                for (var _d = true, _e = __asyncValues(this.listByAuthorizationPagingPage(resourceGroupName, serviceName, authorizationProviderId, authorizationId, options)), _f; _f = yield __await(_e.next()), _a = _f.done, !_a;) {
                    _c = _f.value;
                    _d = false;
                    try {
                        const page = _c;
                        yield __await(yield* __asyncDelegator(__asyncValues(page)));
                    }
                    finally {
                        _d = true;
                    }
                }
            }
            catch (e_1_1) { e_1 = { error: e_1_1 }; }
            finally {
                try {
                    if (!_d && !_a && (_b = _e.return)) yield __await(_b.call(_e));
                }
                finally { if (e_1) throw e_1.error; }
            }
        });
    }
    /**
     * Lists a collection of authorization access policy defined within a authorization.
     * @param resourceGroupName The name of the resource group. The name is case insensitive.
     * @param serviceName The name of the API Management service.
     * @param authorizationProviderId Identifier of the authorization provider.
     * @param authorizationId Identifier of the authorization.
     * @param options The options parameters.
     */
    _listByAuthorization(resourceGroupName, serviceName, authorizationProviderId, authorizationId, options) {
        return this.client.sendOperationRequest({
            resourceGroupName,
            serviceName,
            authorizationProviderId,
            authorizationId,
            options
        }, listByAuthorizationOperationSpec);
    }
    /**
     * Gets the details of the authorization access policy specified by its identifier.
     * @param resourceGroupName The name of the resource group. The name is case insensitive.
     * @param serviceName The name of the API Management service.
     * @param authorizationProviderId Identifier of the authorization provider.
     * @param authorizationId Identifier of the authorization.
     * @param authorizationAccessPolicyId Identifier of the authorization access policy.
     * @param options The options parameters.
     */
    get(resourceGroupName, serviceName, authorizationProviderId, authorizationId, authorizationAccessPolicyId, options) {
        return this.client.sendOperationRequest({
            resourceGroupName,
            serviceName,
            authorizationProviderId,
            authorizationId,
            authorizationAccessPolicyId,
            options
        }, getOperationSpec);
    }
    /**
     * Creates or updates Authorization Access Policy.
     * @param resourceGroupName The name of the resource group. The name is case insensitive.
     * @param serviceName The name of the API Management service.
     * @param authorizationProviderId Identifier of the authorization provider.
     * @param authorizationId Identifier of the authorization.
     * @param authorizationAccessPolicyId Identifier of the authorization access policy.
     * @param parameters Create parameters.
     * @param options The options parameters.
     */
    createOrUpdate(resourceGroupName, serviceName, authorizationProviderId, authorizationId, authorizationAccessPolicyId, parameters, options) {
        return this.client.sendOperationRequest({
            resourceGroupName,
            serviceName,
            authorizationProviderId,
            authorizationId,
            authorizationAccessPolicyId,
            parameters,
            options
        }, createOrUpdateOperationSpec);
    }
    /**
     * Deletes specific access policy from the Authorization.
     * @param resourceGroupName The name of the resource group. The name is case insensitive.
     * @param serviceName The name of the API Management service.
     * @param authorizationProviderId Identifier of the authorization provider.
     * @param authorizationId Identifier of the authorization.
     * @param authorizationAccessPolicyId Identifier of the authorization access policy.
     * @param ifMatch ETag of the Entity. ETag should match the current entity state from the header
     *                response of the GET request or it should be * for unconditional update.
     * @param options The options parameters.
     */
    delete(resourceGroupName, serviceName, authorizationProviderId, authorizationId, authorizationAccessPolicyId, ifMatch, options) {
        return this.client.sendOperationRequest({
            resourceGroupName,
            serviceName,
            authorizationProviderId,
            authorizationId,
            authorizationAccessPolicyId,
            ifMatch,
            options
        }, deleteOperationSpec);
    }
    /**
     * ListByAuthorizationNext
     * @param resourceGroupName The name of the resource group. The name is case insensitive.
     * @param serviceName The name of the API Management service.
     * @param authorizationProviderId Identifier of the authorization provider.
     * @param authorizationId Identifier of the authorization.
     * @param nextLink The nextLink from the previous successful call to the ListByAuthorization method.
     * @param options The options parameters.
     */
    _listByAuthorizationNext(resourceGroupName, serviceName, authorizationProviderId, authorizationId, nextLink, options) {
        return this.client.sendOperationRequest({
            resourceGroupName,
            serviceName,
            authorizationProviderId,
            authorizationId,
            nextLink,
            options
        }, listByAuthorizationNextOperationSpec);
    }
}
// Operation Specifications
const serializer = coreClient.createSerializer(Mappers, /* isXml */ false);
const listByAuthorizationOperationSpec = {
    path: "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.ApiManagement/service/{serviceName}/authorizationProviders/{authorizationProviderId}/authorizations/{authorizationId}/accessPolicies",
    httpMethod: "GET",
    responses: {
        200: {
            bodyMapper: Mappers.AuthorizationAccessPolicyCollection
        },
        default: {
            bodyMapper: Mappers.ErrorResponse
        }
    },
    queryParameters: [
        Parameters.filter,
        Parameters.top,
        Parameters.skip,
        Parameters.apiVersion
    ],
    urlParameters: [
        Parameters.$host,
        Parameters.resourceGroupName,
        Parameters.serviceName,
        Parameters.subscriptionId,
        Parameters.authorizationProviderId,
        Parameters.authorizationId
    ],
    headerParameters: [Parameters.accept],
    serializer
};
const getOperationSpec = {
    path: "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.ApiManagement/service/{serviceName}/authorizationProviders/{authorizationProviderId}/authorizations/{authorizationId}/accessPolicies/{authorizationAccessPolicyId}",
    httpMethod: "GET",
    responses: {
        200: {
            bodyMapper: Mappers.AuthorizationAccessPolicyContract,
            headersMapper: Mappers.AuthorizationAccessPolicyGetHeaders
        },
        default: {
            bodyMapper: Mappers.ErrorResponse
        }
    },
    queryParameters: [Parameters.apiVersion],
    urlParameters: [
        Parameters.$host,
        Parameters.resourceGroupName,
        Parameters.serviceName,
        Parameters.subscriptionId,
        Parameters.authorizationProviderId,
        Parameters.authorizationId,
        Parameters.authorizationAccessPolicyId
    ],
    headerParameters: [Parameters.accept],
    serializer
};
const createOrUpdateOperationSpec = {
    path: "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.ApiManagement/service/{serviceName}/authorizationProviders/{authorizationProviderId}/authorizations/{authorizationId}/accessPolicies/{authorizationAccessPolicyId}",
    httpMethod: "PUT",
    responses: {
        200: {
            bodyMapper: Mappers.AuthorizationAccessPolicyContract,
            headersMapper: Mappers.AuthorizationAccessPolicyCreateOrUpdateHeaders
        },
        201: {
            bodyMapper: Mappers.AuthorizationAccessPolicyContract,
            headersMapper: Mappers.AuthorizationAccessPolicyCreateOrUpdateHeaders
        },
        default: {
            bodyMapper: Mappers.ErrorResponse
        }
    },
    requestBody: Parameters.parameters26,
    queryParameters: [Parameters.apiVersion],
    urlParameters: [
        Parameters.$host,
        Parameters.resourceGroupName,
        Parameters.serviceName,
        Parameters.subscriptionId,
        Parameters.authorizationProviderId,
        Parameters.authorizationId,
        Parameters.authorizationAccessPolicyId
    ],
    headerParameters: [
        Parameters.accept,
        Parameters.contentType,
        Parameters.ifMatch
    ],
    mediaType: "json",
    serializer
};
const deleteOperationSpec = {
    path: "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.ApiManagement/service/{serviceName}/authorizationProviders/{authorizationProviderId}/authorizations/{authorizationId}/accessPolicies/{authorizationAccessPolicyId}",
    httpMethod: "DELETE",
    responses: {
        200: {},
        204: {},
        default: {
            bodyMapper: Mappers.ErrorResponse
        }
    },
    queryParameters: [Parameters.apiVersion],
    urlParameters: [
        Parameters.$host,
        Parameters.resourceGroupName,
        Parameters.serviceName,
        Parameters.subscriptionId,
        Parameters.authorizationProviderId,
        Parameters.authorizationId,
        Parameters.authorizationAccessPolicyId
    ],
    headerParameters: [Parameters.accept, Parameters.ifMatch1],
    serializer
};
const listByAuthorizationNextOperationSpec = {
    path: "{nextLink}",
    httpMethod: "GET",
    responses: {
        200: {
            bodyMapper: Mappers.AuthorizationAccessPolicyCollection
        },
        default: {
            bodyMapper: Mappers.ErrorResponse
        }
    },
    urlParameters: [
        Parameters.$host,
        Parameters.resourceGroupName,
        Parameters.serviceName,
        Parameters.subscriptionId,
        Parameters.nextLink,
        Parameters.authorizationProviderId,
        Parameters.authorizationId
    ],
    headerParameters: [Parameters.accept],
    serializer
};
