// @azure/arm-templatespecs/dist-esm/src/models/index.js
/*
 * Copyright (c) Microsoft Corporation.
 * Licensed under the MIT License.
 *
 * Code generated by Microsoft (R) AutoRest Code Generator.
 * Changes may cause incorrect behavior and will be lost if the code is regenerated.
 */
/** Known values of {@link CreatedByType} that the service accepts. */
export var KnownCreatedByType;
(function (KnownCreatedByType) {
    /** User */
    KnownCreatedByType["User"] = "User";
    /** Application */
    KnownCreatedByType["Application"] = "Application";
    /** ManagedIdentity */
    KnownCreatedByType["ManagedIdentity"] = "ManagedIdentity";
    /** Key */
    KnownCreatedByType["Key"] = "Key";
})(KnownCreatedByType || (KnownCreatedByType = {}));
/** Known values of {@link TemplateSpecExpandKind} that the service accepts. */
export var KnownTemplateSpecExpandKind;
(function (KnownTemplateSpecExpandKind) {
    /** Includes version information with the Template Spec. */
    KnownTemplateSpecExpandKind["Versions"] = "versions";
})(KnownTemplateSpecExpandKind || (KnownTemplateSpecExpandKind = {}));
