// @azure/arm-avs/dist-esm/src/models/index.js
/*
 * Copyright (c) Microsoft Corporation.
 * Licensed under the MIT License.
 *
 * Code generated by Microsoft (R) AutoRest Code Generator.
 * Changes may cause incorrect behavior and will be lost if the code is regenerated.
 */
/** Known values of {@link TrialStatus} that the service accepts. */
export var KnownTrialStatus;
(function (KnownTrialStatus) {
    /** TrialAvailable */
    KnownTrialStatus["TrialAvailable"] = "TrialAvailable";
    /** TrialUsed */
    KnownTrialStatus["TrialUsed"] = "TrialUsed";
    /** TrialDisabled */
    KnownTrialStatus["TrialDisabled"] = "TrialDisabled";
})(KnownTrialStatus || (KnownTrialStatus = {}));
/** Known values of {@link QuotaEnabled} that the service accepts. */
export var KnownQuotaEnabled;
(function (KnownQuotaEnabled) {
    /** Enabled */
    KnownQuotaEnabled["Enabled"] = "Enabled";
    /** Disabled */
    KnownQuotaEnabled["Disabled"] = "Disabled";
})(KnownQuotaEnabled || (KnownQuotaEnabled = {}));
/** Known values of {@link PrivateCloudProvisioningState} that the service accepts. */
export var KnownPrivateCloudProvisioningState;
(function (KnownPrivateCloudProvisioningState) {
    /** Succeeded */
    KnownPrivateCloudProvisioningState["Succeeded"] = "Succeeded";
    /** Failed */
    KnownPrivateCloudProvisioningState["Failed"] = "Failed";
    /** Cancelled */
    KnownPrivateCloudProvisioningState["Cancelled"] = "Cancelled";
    /** Pending */
    KnownPrivateCloudProvisioningState["Pending"] = "Pending";
    /** Building */
    KnownPrivateCloudProvisioningState["Building"] = "Building";
    /** Deleting */
    KnownPrivateCloudProvisioningState["Deleting"] = "Deleting";
    /** Updating */
    KnownPrivateCloudProvisioningState["Updating"] = "Updating";
    /** Canceled */
    KnownPrivateCloudProvisioningState["Canceled"] = "Canceled";
})(KnownPrivateCloudProvisioningState || (KnownPrivateCloudProvisioningState = {}));
/** Known values of {@link NsxPublicIpQuotaRaisedEnum} that the service accepts. */
export var KnownNsxPublicIpQuotaRaisedEnum;
(function (KnownNsxPublicIpQuotaRaisedEnum) {
    /** Enabled */
    KnownNsxPublicIpQuotaRaisedEnum["Enabled"] = "Enabled";
    /** Disabled */
    KnownNsxPublicIpQuotaRaisedEnum["Disabled"] = "Disabled";
})(KnownNsxPublicIpQuotaRaisedEnum || (KnownNsxPublicIpQuotaRaisedEnum = {}));
/** Known values of {@link ClusterProvisioningState} that the service accepts. */
export var KnownClusterProvisioningState;
(function (KnownClusterProvisioningState) {
    /** Succeeded */
    KnownClusterProvisioningState["Succeeded"] = "Succeeded";
    /** Failed */
    KnownClusterProvisioningState["Failed"] = "Failed";
    /** Cancelled */
    KnownClusterProvisioningState["Cancelled"] = "Cancelled";
    /** Deleting */
    KnownClusterProvisioningState["Deleting"] = "Deleting";
    /** Updating */
    KnownClusterProvisioningState["Updating"] = "Updating";
    /** Canceled */
    KnownClusterProvisioningState["Canceled"] = "Canceled";
})(KnownClusterProvisioningState || (KnownClusterProvisioningState = {}));
/** Known values of {@link InternetEnum} that the service accepts. */
export var KnownInternetEnum;
(function (KnownInternetEnum) {
    /** Enabled */
    KnownInternetEnum["Enabled"] = "Enabled";
    /** Disabled */
    KnownInternetEnum["Disabled"] = "Disabled";
})(KnownInternetEnum || (KnownInternetEnum = {}));
/** Known values of {@link SslEnum} that the service accepts. */
export var KnownSslEnum;
(function (KnownSslEnum) {
    /** Enabled */
    KnownSslEnum["Enabled"] = "Enabled";
    /** Disabled */
    KnownSslEnum["Disabled"] = "Disabled";
})(KnownSslEnum || (KnownSslEnum = {}));
/** Known values of {@link AvailabilityStrategy} that the service accepts. */
export var KnownAvailabilityStrategy;
(function (KnownAvailabilityStrategy) {
    /** SingleZone */
    KnownAvailabilityStrategy["SingleZone"] = "SingleZone";
    /** DualZone */
    KnownAvailabilityStrategy["DualZone"] = "DualZone";
})(KnownAvailabilityStrategy || (KnownAvailabilityStrategy = {}));
/** Known values of {@link EncryptionState} that the service accepts. */
export var KnownEncryptionState;
(function (KnownEncryptionState) {
    /** Enabled */
    KnownEncryptionState["Enabled"] = "Enabled";
    /** Disabled */
    KnownEncryptionState["Disabled"] = "Disabled";
})(KnownEncryptionState || (KnownEncryptionState = {}));
/** Known values of {@link EncryptionKeyStatus} that the service accepts. */
export var KnownEncryptionKeyStatus;
(function (KnownEncryptionKeyStatus) {
    /** Connected */
    KnownEncryptionKeyStatus["Connected"] = "Connected";
    /** AccessDenied */
    KnownEncryptionKeyStatus["AccessDenied"] = "AccessDenied";
})(KnownEncryptionKeyStatus || (KnownEncryptionKeyStatus = {}));
/** Known values of {@link EncryptionVersionType} that the service accepts. */
export var KnownEncryptionVersionType;
(function (KnownEncryptionVersionType) {
    /** Fixed */
    KnownEncryptionVersionType["Fixed"] = "Fixed";
    /** AutoDetected */
    KnownEncryptionVersionType["AutoDetected"] = "AutoDetected";
})(KnownEncryptionVersionType || (KnownEncryptionVersionType = {}));
/** Known values of {@link ResourceIdentityType} that the service accepts. */
export var KnownResourceIdentityType;
(function (KnownResourceIdentityType) {
    /** SystemAssigned */
    KnownResourceIdentityType["SystemAssigned"] = "SystemAssigned";
    /** None */
    KnownResourceIdentityType["None"] = "None";
})(KnownResourceIdentityType || (KnownResourceIdentityType = {}));
/** Known values of {@link DatastoreProvisioningState} that the service accepts. */
export var KnownDatastoreProvisioningState;
(function (KnownDatastoreProvisioningState) {
    /** Succeeded */
    KnownDatastoreProvisioningState["Succeeded"] = "Succeeded";
    /** Failed */
    KnownDatastoreProvisioningState["Failed"] = "Failed";
    /** Cancelled */
    KnownDatastoreProvisioningState["Cancelled"] = "Cancelled";
    /** Pending */
    KnownDatastoreProvisioningState["Pending"] = "Pending";
    /** Creating */
    KnownDatastoreProvisioningState["Creating"] = "Creating";
    /** Updating */
    KnownDatastoreProvisioningState["Updating"] = "Updating";
    /** Deleting */
    KnownDatastoreProvisioningState["Deleting"] = "Deleting";
    /** Canceled */
    KnownDatastoreProvisioningState["Canceled"] = "Canceled";
})(KnownDatastoreProvisioningState || (KnownDatastoreProvisioningState = {}));
/** Known values of {@link MountOptionEnum} that the service accepts. */
export var KnownMountOptionEnum;
(function (KnownMountOptionEnum) {
    /** Mount */
    KnownMountOptionEnum["Mount"] = "MOUNT";
    /** Attach */
    KnownMountOptionEnum["Attach"] = "ATTACH";
})(KnownMountOptionEnum || (KnownMountOptionEnum = {}));
/** Known values of {@link DatastoreStatus} that the service accepts. */
export var KnownDatastoreStatus;
(function (KnownDatastoreStatus) {
    /** Unknown */
    KnownDatastoreStatus["Unknown"] = "Unknown";
    /** Accessible */
    KnownDatastoreStatus["Accessible"] = "Accessible";
    /** Inaccessible */
    KnownDatastoreStatus["Inaccessible"] = "Inaccessible";
    /** Attached */
    KnownDatastoreStatus["Attached"] = "Attached";
    /** Detached */
    KnownDatastoreStatus["Detached"] = "Detached";
    /** LostCommunication */
    KnownDatastoreStatus["LostCommunication"] = "LostCommunication";
    /** DeadOrError */
    KnownDatastoreStatus["DeadOrError"] = "DeadOrError";
})(KnownDatastoreStatus || (KnownDatastoreStatus = {}));
/** Known values of {@link HcxEnterpriseSiteStatus} that the service accepts. */
export var KnownHcxEnterpriseSiteStatus;
(function (KnownHcxEnterpriseSiteStatus) {
    /** Available */
    KnownHcxEnterpriseSiteStatus["Available"] = "Available";
    /** Consumed */
    KnownHcxEnterpriseSiteStatus["Consumed"] = "Consumed";
    /** Deactivated */
    KnownHcxEnterpriseSiteStatus["Deactivated"] = "Deactivated";
    /** Deleted */
    KnownHcxEnterpriseSiteStatus["Deleted"] = "Deleted";
})(KnownHcxEnterpriseSiteStatus || (KnownHcxEnterpriseSiteStatus = {}));
/** Known values of {@link ExpressRouteAuthorizationProvisioningState} that the service accepts. */
export var KnownExpressRouteAuthorizationProvisioningState;
(function (KnownExpressRouteAuthorizationProvisioningState) {
    /** Succeeded */
    KnownExpressRouteAuthorizationProvisioningState["Succeeded"] = "Succeeded";
    /** Failed */
    KnownExpressRouteAuthorizationProvisioningState["Failed"] = "Failed";
    /** Updating */
    KnownExpressRouteAuthorizationProvisioningState["Updating"] = "Updating";
    /** Canceled */
    KnownExpressRouteAuthorizationProvisioningState["Canceled"] = "Canceled";
})(KnownExpressRouteAuthorizationProvisioningState || (KnownExpressRouteAuthorizationProvisioningState = {}));
/** Known values of {@link GlobalReachConnectionProvisioningState} that the service accepts. */
export var KnownGlobalReachConnectionProvisioningState;
(function (KnownGlobalReachConnectionProvisioningState) {
    /** Succeeded */
    KnownGlobalReachConnectionProvisioningState["Succeeded"] = "Succeeded";
    /** Failed */
    KnownGlobalReachConnectionProvisioningState["Failed"] = "Failed";
    /** Updating */
    KnownGlobalReachConnectionProvisioningState["Updating"] = "Updating";
    /** Canceled */
    KnownGlobalReachConnectionProvisioningState["Canceled"] = "Canceled";
})(KnownGlobalReachConnectionProvisioningState || (KnownGlobalReachConnectionProvisioningState = {}));
/** Known values of {@link GlobalReachConnectionStatus} that the service accepts. */
export var KnownGlobalReachConnectionStatus;
(function (KnownGlobalReachConnectionStatus) {
    /** Connected */
    KnownGlobalReachConnectionStatus["Connected"] = "Connected";
    /** Connecting */
    KnownGlobalReachConnectionStatus["Connecting"] = "Connecting";
    /** Disconnected */
    KnownGlobalReachConnectionStatus["Disconnected"] = "Disconnected";
})(KnownGlobalReachConnectionStatus || (KnownGlobalReachConnectionStatus = {}));
/** Known values of {@link WorkloadNetworkName} that the service accepts. */
export var KnownWorkloadNetworkName;
(function (KnownWorkloadNetworkName) {
    /** Default */
    KnownWorkloadNetworkName["Default"] = "default";
})(KnownWorkloadNetworkName || (KnownWorkloadNetworkName = {}));
/** Known values of {@link SegmentStatusEnum} that the service accepts. */
export var KnownSegmentStatusEnum;
(function (KnownSegmentStatusEnum) {
    /** Success */
    KnownSegmentStatusEnum["Success"] = "SUCCESS";
    /** Failure */
    KnownSegmentStatusEnum["Failure"] = "FAILURE";
})(KnownSegmentStatusEnum || (KnownSegmentStatusEnum = {}));
/** Known values of {@link WorkloadNetworkSegmentProvisioningState} that the service accepts. */
export var KnownWorkloadNetworkSegmentProvisioningState;
(function (KnownWorkloadNetworkSegmentProvisioningState) {
    /** Succeeded */
    KnownWorkloadNetworkSegmentProvisioningState["Succeeded"] = "Succeeded";
    /** Failed */
    KnownWorkloadNetworkSegmentProvisioningState["Failed"] = "Failed";
    /** Building */
    KnownWorkloadNetworkSegmentProvisioningState["Building"] = "Building";
    /** Deleting */
    KnownWorkloadNetworkSegmentProvisioningState["Deleting"] = "Deleting";
    /** Updating */
    KnownWorkloadNetworkSegmentProvisioningState["Updating"] = "Updating";
    /** Canceled */
    KnownWorkloadNetworkSegmentProvisioningState["Canceled"] = "Canceled";
})(KnownWorkloadNetworkSegmentProvisioningState || (KnownWorkloadNetworkSegmentProvisioningState = {}));
/** Known values of {@link DhcpTypeEnum} that the service accepts. */
export var KnownDhcpTypeEnum;
(function (KnownDhcpTypeEnum) {
    /** Server */
    KnownDhcpTypeEnum["Server"] = "SERVER";
    /** Relay */
    KnownDhcpTypeEnum["Relay"] = "RELAY";
})(KnownDhcpTypeEnum || (KnownDhcpTypeEnum = {}));
/** Known values of {@link WorkloadNetworkDhcpProvisioningState} that the service accepts. */
export var KnownWorkloadNetworkDhcpProvisioningState;
(function (KnownWorkloadNetworkDhcpProvisioningState) {
    /** Succeeded */
    KnownWorkloadNetworkDhcpProvisioningState["Succeeded"] = "Succeeded";
    /** Failed */
    KnownWorkloadNetworkDhcpProvisioningState["Failed"] = "Failed";
    /** Building */
    KnownWorkloadNetworkDhcpProvisioningState["Building"] = "Building";
    /** Deleting */
    KnownWorkloadNetworkDhcpProvisioningState["Deleting"] = "Deleting";
    /** Updating */
    KnownWorkloadNetworkDhcpProvisioningState["Updating"] = "Updating";
    /** Canceled */
    KnownWorkloadNetworkDhcpProvisioningState["Canceled"] = "Canceled";
})(KnownWorkloadNetworkDhcpProvisioningState || (KnownWorkloadNetworkDhcpProvisioningState = {}));
/** Known values of {@link PortMirroringDirectionEnum} that the service accepts. */
export var KnownPortMirroringDirectionEnum;
(function (KnownPortMirroringDirectionEnum) {
    /** Ingress */
    KnownPortMirroringDirectionEnum["Ingress"] = "INGRESS";
    /** Egress */
    KnownPortMirroringDirectionEnum["Egress"] = "EGRESS";
    /** Bidirectional */
    KnownPortMirroringDirectionEnum["Bidirectional"] = "BIDIRECTIONAL";
})(KnownPortMirroringDirectionEnum || (KnownPortMirroringDirectionEnum = {}));
/** Known values of {@link PortMirroringStatusEnum} that the service accepts. */
export var KnownPortMirroringStatusEnum;
(function (KnownPortMirroringStatusEnum) {
    /** Success */
    KnownPortMirroringStatusEnum["Success"] = "SUCCESS";
    /** Failure */
    KnownPortMirroringStatusEnum["Failure"] = "FAILURE";
})(KnownPortMirroringStatusEnum || (KnownPortMirroringStatusEnum = {}));
/** Known values of {@link WorkloadNetworkPortMirroringProvisioningState} that the service accepts. */
export var KnownWorkloadNetworkPortMirroringProvisioningState;
(function (KnownWorkloadNetworkPortMirroringProvisioningState) {
    /** Succeeded */
    KnownWorkloadNetworkPortMirroringProvisioningState["Succeeded"] = "Succeeded";
    /** Failed */
    KnownWorkloadNetworkPortMirroringProvisioningState["Failed"] = "Failed";
    /** Building */
    KnownWorkloadNetworkPortMirroringProvisioningState["Building"] = "Building";
    /** Deleting */
    KnownWorkloadNetworkPortMirroringProvisioningState["Deleting"] = "Deleting";
    /** Updating */
    KnownWorkloadNetworkPortMirroringProvisioningState["Updating"] = "Updating";
    /** Canceled */
    KnownWorkloadNetworkPortMirroringProvisioningState["Canceled"] = "Canceled";
})(KnownWorkloadNetworkPortMirroringProvisioningState || (KnownWorkloadNetworkPortMirroringProvisioningState = {}));
/** Known values of {@link VMGroupStatusEnum} that the service accepts. */
export var KnownVMGroupStatusEnum;
(function (KnownVMGroupStatusEnum) {
    /** Success */
    KnownVMGroupStatusEnum["Success"] = "SUCCESS";
    /** Failure */
    KnownVMGroupStatusEnum["Failure"] = "FAILURE";
})(KnownVMGroupStatusEnum || (KnownVMGroupStatusEnum = {}));
/** Known values of {@link WorkloadNetworkVMGroupProvisioningState} that the service accepts. */
export var KnownWorkloadNetworkVMGroupProvisioningState;
(function (KnownWorkloadNetworkVMGroupProvisioningState) {
    /** Succeeded */
    KnownWorkloadNetworkVMGroupProvisioningState["Succeeded"] = "Succeeded";
    /** Failed */
    KnownWorkloadNetworkVMGroupProvisioningState["Failed"] = "Failed";
    /** Building */
    KnownWorkloadNetworkVMGroupProvisioningState["Building"] = "Building";
    /** Deleting */
    KnownWorkloadNetworkVMGroupProvisioningState["Deleting"] = "Deleting";
    /** Updating */
    KnownWorkloadNetworkVMGroupProvisioningState["Updating"] = "Updating";
    /** Canceled */
    KnownWorkloadNetworkVMGroupProvisioningState["Canceled"] = "Canceled";
})(KnownWorkloadNetworkVMGroupProvisioningState || (KnownWorkloadNetworkVMGroupProvisioningState = {}));
/** Known values of {@link VMTypeEnum} that the service accepts. */
export var KnownVMTypeEnum;
(function (KnownVMTypeEnum) {
    /** Regular */
    KnownVMTypeEnum["Regular"] = "REGULAR";
    /** Edge */
    KnownVMTypeEnum["Edge"] = "EDGE";
    /** Service */
    KnownVMTypeEnum["Service"] = "SERVICE";
})(KnownVMTypeEnum || (KnownVMTypeEnum = {}));
/** Known values of {@link DnsServiceLogLevelEnum} that the service accepts. */
export var KnownDnsServiceLogLevelEnum;
(function (KnownDnsServiceLogLevelEnum) {
    /** Debug */
    KnownDnsServiceLogLevelEnum["Debug"] = "DEBUG";
    /** Info */
    KnownDnsServiceLogLevelEnum["Info"] = "INFO";
    /** Warning */
    KnownDnsServiceLogLevelEnum["Warning"] = "WARNING";
    /** Error */
    KnownDnsServiceLogLevelEnum["Error"] = "ERROR";
    /** Fatal */
    KnownDnsServiceLogLevelEnum["Fatal"] = "FATAL";
})(KnownDnsServiceLogLevelEnum || (KnownDnsServiceLogLevelEnum = {}));
/** Known values of {@link DnsServiceStatusEnum} that the service accepts. */
export var KnownDnsServiceStatusEnum;
(function (KnownDnsServiceStatusEnum) {
    /** Success */
    KnownDnsServiceStatusEnum["Success"] = "SUCCESS";
    /** Failure */
    KnownDnsServiceStatusEnum["Failure"] = "FAILURE";
})(KnownDnsServiceStatusEnum || (KnownDnsServiceStatusEnum = {}));
/** Known values of {@link WorkloadNetworkDnsServiceProvisioningState} that the service accepts. */
export var KnownWorkloadNetworkDnsServiceProvisioningState;
(function (KnownWorkloadNetworkDnsServiceProvisioningState) {
    /** Succeeded */
    KnownWorkloadNetworkDnsServiceProvisioningState["Succeeded"] = "Succeeded";
    /** Failed */
    KnownWorkloadNetworkDnsServiceProvisioningState["Failed"] = "Failed";
    /** Building */
    KnownWorkloadNetworkDnsServiceProvisioningState["Building"] = "Building";
    /** Deleting */
    KnownWorkloadNetworkDnsServiceProvisioningState["Deleting"] = "Deleting";
    /** Updating */
    KnownWorkloadNetworkDnsServiceProvisioningState["Updating"] = "Updating";
    /** Canceled */
    KnownWorkloadNetworkDnsServiceProvisioningState["Canceled"] = "Canceled";
})(KnownWorkloadNetworkDnsServiceProvisioningState || (KnownWorkloadNetworkDnsServiceProvisioningState = {}));
/** Known values of {@link WorkloadNetworkDnsZoneProvisioningState} that the service accepts. */
export var KnownWorkloadNetworkDnsZoneProvisioningState;
(function (KnownWorkloadNetworkDnsZoneProvisioningState) {
    /** Succeeded */
    KnownWorkloadNetworkDnsZoneProvisioningState["Succeeded"] = "Succeeded";
    /** Failed */
    KnownWorkloadNetworkDnsZoneProvisioningState["Failed"] = "Failed";
    /** Building */
    KnownWorkloadNetworkDnsZoneProvisioningState["Building"] = "Building";
    /** Deleting */
    KnownWorkloadNetworkDnsZoneProvisioningState["Deleting"] = "Deleting";
    /** Updating */
    KnownWorkloadNetworkDnsZoneProvisioningState["Updating"] = "Updating";
    /** Canceled */
    KnownWorkloadNetworkDnsZoneProvisioningState["Canceled"] = "Canceled";
})(KnownWorkloadNetworkDnsZoneProvisioningState || (KnownWorkloadNetworkDnsZoneProvisioningState = {}));
/** Known values of {@link WorkloadNetworkPublicIPProvisioningState} that the service accepts. */
export var KnownWorkloadNetworkPublicIPProvisioningState;
(function (KnownWorkloadNetworkPublicIPProvisioningState) {
    /** Succeeded */
    KnownWorkloadNetworkPublicIPProvisioningState["Succeeded"] = "Succeeded";
    /** Failed */
    KnownWorkloadNetworkPublicIPProvisioningState["Failed"] = "Failed";
    /** Building */
    KnownWorkloadNetworkPublicIPProvisioningState["Building"] = "Building";
    /** Deleting */
    KnownWorkloadNetworkPublicIPProvisioningState["Deleting"] = "Deleting";
    /** Updating */
    KnownWorkloadNetworkPublicIPProvisioningState["Updating"] = "Updating";
    /** Canceled */
    KnownWorkloadNetworkPublicIPProvisioningState["Canceled"] = "Canceled";
})(KnownWorkloadNetworkPublicIPProvisioningState || (KnownWorkloadNetworkPublicIPProvisioningState = {}));
/** Known values of {@link CloudLinkStatus} that the service accepts. */
export var KnownCloudLinkStatus;
(function (KnownCloudLinkStatus) {
    /** Active */
    KnownCloudLinkStatus["Active"] = "Active";
    /** Building */
    KnownCloudLinkStatus["Building"] = "Building";
    /** Deleting */
    KnownCloudLinkStatus["Deleting"] = "Deleting";
    /** Failed */
    KnownCloudLinkStatus["Failed"] = "Failed";
    /** Disconnected */
    KnownCloudLinkStatus["Disconnected"] = "Disconnected";
})(KnownCloudLinkStatus || (KnownCloudLinkStatus = {}));
/** Known values of {@link AddonType} that the service accepts. */
export var KnownAddonType;
(function (KnownAddonType) {
    /** SRM */
    KnownAddonType["SRM"] = "SRM";
    /** VR */
    KnownAddonType["VR"] = "VR";
    /** HCX */
    KnownAddonType["HCX"] = "HCX";
    /** Arc */
    KnownAddonType["Arc"] = "Arc";
})(KnownAddonType || (KnownAddonType = {}));
/** Known values of {@link AddonProvisioningState} that the service accepts. */
export var KnownAddonProvisioningState;
(function (KnownAddonProvisioningState) {
    /** Succeeded */
    KnownAddonProvisioningState["Succeeded"] = "Succeeded";
    /** Failed */
    KnownAddonProvisioningState["Failed"] = "Failed";
    /** Cancelled */
    KnownAddonProvisioningState["Cancelled"] = "Cancelled";
    /** Building */
    KnownAddonProvisioningState["Building"] = "Building";
    /** Deleting */
    KnownAddonProvisioningState["Deleting"] = "Deleting";
    /** Updating */
    KnownAddonProvisioningState["Updating"] = "Updating";
    /** Canceled */
    KnownAddonProvisioningState["Canceled"] = "Canceled";
})(KnownAddonProvisioningState || (KnownAddonProvisioningState = {}));
/** Known values of {@link VirtualMachineRestrictMovementState} that the service accepts. */
export var KnownVirtualMachineRestrictMovementState;
(function (KnownVirtualMachineRestrictMovementState) {
    /** Enabled */
    KnownVirtualMachineRestrictMovementState["Enabled"] = "Enabled";
    /** Disabled */
    KnownVirtualMachineRestrictMovementState["Disabled"] = "Disabled";
})(KnownVirtualMachineRestrictMovementState || (KnownVirtualMachineRestrictMovementState = {}));
/** Known values of {@link PlacementPolicyType} that the service accepts. */
export var KnownPlacementPolicyType;
(function (KnownPlacementPolicyType) {
    /** VmVm */
    KnownPlacementPolicyType["VmVm"] = "VmVm";
    /** VmHost */
    KnownPlacementPolicyType["VmHost"] = "VmHost";
})(KnownPlacementPolicyType || (KnownPlacementPolicyType = {}));
/** Known values of {@link PlacementPolicyState} that the service accepts. */
export var KnownPlacementPolicyState;
(function (KnownPlacementPolicyState) {
    /** Enabled */
    KnownPlacementPolicyState["Enabled"] = "Enabled";
    /** Disabled */
    KnownPlacementPolicyState["Disabled"] = "Disabled";
})(KnownPlacementPolicyState || (KnownPlacementPolicyState = {}));
/** Known values of {@link PlacementPolicyProvisioningState} that the service accepts. */
export var KnownPlacementPolicyProvisioningState;
(function (KnownPlacementPolicyProvisioningState) {
    /** Succeeded */
    KnownPlacementPolicyProvisioningState["Succeeded"] = "Succeeded";
    /** Failed */
    KnownPlacementPolicyProvisioningState["Failed"] = "Failed";
    /** Building */
    KnownPlacementPolicyProvisioningState["Building"] = "Building";
    /** Deleting */
    KnownPlacementPolicyProvisioningState["Deleting"] = "Deleting";
    /** Updating */
    KnownPlacementPolicyProvisioningState["Updating"] = "Updating";
    /** Canceled */
    KnownPlacementPolicyProvisioningState["Canceled"] = "Canceled";
})(KnownPlacementPolicyProvisioningState || (KnownPlacementPolicyProvisioningState = {}));
/** Known values of {@link AffinityStrength} that the service accepts. */
export var KnownAffinityStrength;
(function (KnownAffinityStrength) {
    /** Should */
    KnownAffinityStrength["Should"] = "Should";
    /** Must */
    KnownAffinityStrength["Must"] = "Must";
})(KnownAffinityStrength || (KnownAffinityStrength = {}));
/** Known values of {@link AzureHybridBenefitType} that the service accepts. */
export var KnownAzureHybridBenefitType;
(function (KnownAzureHybridBenefitType) {
    /** SqlHost */
    KnownAzureHybridBenefitType["SqlHost"] = "SqlHost";
    /** None */
    KnownAzureHybridBenefitType["None"] = "None";
})(KnownAzureHybridBenefitType || (KnownAzureHybridBenefitType = {}));
/** Known values of {@link ScriptParameterTypes} that the service accepts. */
export var KnownScriptParameterTypes;
(function (KnownScriptParameterTypes) {
    /** String */
    KnownScriptParameterTypes["String"] = "String";
    /** SecureString */
    KnownScriptParameterTypes["SecureString"] = "SecureString";
    /** Credential */
    KnownScriptParameterTypes["Credential"] = "Credential";
    /** Int */
    KnownScriptParameterTypes["Int"] = "Int";
    /** Bool */
    KnownScriptParameterTypes["Bool"] = "Bool";
    /** Float */
    KnownScriptParameterTypes["Float"] = "Float";
})(KnownScriptParameterTypes || (KnownScriptParameterTypes = {}));
/** Known values of {@link VisibilityParameterEnum} that the service accepts. */
export var KnownVisibilityParameterEnum;
(function (KnownVisibilityParameterEnum) {
    /** Visible */
    KnownVisibilityParameterEnum["Visible"] = "Visible";
    /** Hidden */
    KnownVisibilityParameterEnum["Hidden"] = "Hidden";
})(KnownVisibilityParameterEnum || (KnownVisibilityParameterEnum = {}));
/** Known values of {@link OptionalParamEnum} that the service accepts. */
export var KnownOptionalParamEnum;
(function (KnownOptionalParamEnum) {
    /** Optional */
    KnownOptionalParamEnum["Optional"] = "Optional";
    /** Required */
    KnownOptionalParamEnum["Required"] = "Required";
})(KnownOptionalParamEnum || (KnownOptionalParamEnum = {}));
/** Known values of {@link ScriptExecutionParameterType} that the service accepts. */
export var KnownScriptExecutionParameterType;
(function (KnownScriptExecutionParameterType) {
    /** Value */
    KnownScriptExecutionParameterType["Value"] = "Value";
    /** SecureValue */
    KnownScriptExecutionParameterType["SecureValue"] = "SecureValue";
    /** Credential */
    KnownScriptExecutionParameterType["Credential"] = "Credential";
})(KnownScriptExecutionParameterType || (KnownScriptExecutionParameterType = {}));
/** Known values of {@link ScriptExecutionProvisioningState} that the service accepts. */
export var KnownScriptExecutionProvisioningState;
(function (KnownScriptExecutionProvisioningState) {
    /** Pending */
    KnownScriptExecutionProvisioningState["Pending"] = "Pending";
    /** Running */
    KnownScriptExecutionProvisioningState["Running"] = "Running";
    /** Succeeded */
    KnownScriptExecutionProvisioningState["Succeeded"] = "Succeeded";
    /** Failed */
    KnownScriptExecutionProvisioningState["Failed"] = "Failed";
    /** Cancelling */
    KnownScriptExecutionProvisioningState["Cancelling"] = "Cancelling";
    /** Cancelled */
    KnownScriptExecutionProvisioningState["Cancelled"] = "Cancelled";
    /** Deleting */
    KnownScriptExecutionProvisioningState["Deleting"] = "Deleting";
    /** Canceled */
    KnownScriptExecutionProvisioningState["Canceled"] = "Canceled";
})(KnownScriptExecutionProvisioningState || (KnownScriptExecutionProvisioningState = {}));
/** Known values of {@link ScriptOutputStreamType} that the service accepts. */
export var KnownScriptOutputStreamType;
(function (KnownScriptOutputStreamType) {
    /** Information */
    KnownScriptOutputStreamType["Information"] = "Information";
    /** Warning */
    KnownScriptOutputStreamType["Warning"] = "Warning";
    /** Output */
    KnownScriptOutputStreamType["Output"] = "Output";
    /** Error */
    KnownScriptOutputStreamType["Error"] = "Error";
})(KnownScriptOutputStreamType || (KnownScriptOutputStreamType = {}));
/** Known values of {@link AffinityType} that the service accepts. */
export var KnownAffinityType;
(function (KnownAffinityType) {
    /** Affinity */
    KnownAffinityType["Affinity"] = "Affinity";
    /** AntiAffinity */
    KnownAffinityType["AntiAffinity"] = "AntiAffinity";
})(KnownAffinityType || (KnownAffinityType = {}));
