// @azure/arm-avs/dist-esm/src/models/parameters.js
/*
 * Copyright (c) Microsoft Corporation.
 * Licensed under the MIT License.
 *
 * Code generated by Microsoft (R) AutoRest Code Generator.
 * Changes may cause incorrect behavior and will be lost if the code is regenerated.
 */
import { Sku as SkuMapper, PrivateCloud as PrivateCloudMapper, PrivateCloudUpdate as PrivateCloudUpdateMapper, Cluster as ClusterMapper, ClusterUpdate as ClusterUpdateMapper, Datastore as DatastoreMapper, HcxEnterpriseSite as HcxEnterpriseSiteMapper, ExpressRouteAuthorization as ExpressRouteAuthorizationMapper, GlobalReachConnection as GlobalReachConnectionMapper, WorkloadNetworkSegment as WorkloadNetworkSegmentMapper, WorkloadNetworkDhcp as WorkloadNetworkDhcpMapper, WorkloadNetworkPortMirroring as WorkloadNetworkPortMirroringMapper, WorkloadNetworkVMGroup as WorkloadNetworkVMGroupMapper, WorkloadNetworkDnsService as WorkloadNetworkDnsServiceMapper, WorkloadNetworkDnsZone as WorkloadNetworkDnsZoneMapper, WorkloadNetworkPublicIP as WorkloadNetworkPublicIPMapper, CloudLink as CloudLinkMapper, Addon as AddonMapper, VirtualMachineRestrictMovement as VirtualMachineRestrictMovementMapper, PlacementPolicy as PlacementPolicyMapper, PlacementPolicyUpdate as PlacementPolicyUpdateMapper, ScriptExecution as ScriptExecutionMapper } from "../models/mappers";
export const accept = {
    parameterPath: "accept",
    mapper: {
        defaultValue: "application/json",
        isConstant: true,
        serializedName: "Accept",
        type: {
            name: "String"
        }
    }
};
export const $host = {
    parameterPath: "$host",
    mapper: {
        serializedName: "$host",
        required: true,
        type: {
            name: "String"
        }
    },
    skipEncoding: true
};
export const apiVersion = {
    parameterPath: "apiVersion",
    mapper: {
        defaultValue: "2023-03-01",
        isConstant: true,
        serializedName: "api-version",
        type: {
            name: "String"
        }
    }
};
export const nextLink = {
    parameterPath: "nextLink",
    mapper: {
        serializedName: "nextLink",
        required: true,
        type: {
            name: "String"
        }
    },
    skipEncoding: true
};
export const contentType = {
    parameterPath: ["options", "contentType"],
    mapper: {
        defaultValue: "application/json",
        isConstant: true,
        serializedName: "Content-Type",
        type: {
            name: "String"
        }
    }
};
export const sku = {
    parameterPath: ["options", "sku"],
    mapper: SkuMapper
};
export const subscriptionId = {
    parameterPath: "subscriptionId",
    mapper: {
        constraints: {
            MinLength: 1
        },
        serializedName: "subscriptionId",
        required: true,
        type: {
            name: "String"
        }
    }
};
export const location = {
    parameterPath: "location",
    mapper: {
        serializedName: "location",
        required: true,
        type: {
            name: "String"
        }
    }
};
export const resourceGroupName = {
    parameterPath: "resourceGroupName",
    mapper: {
        constraints: {
            MaxLength: 90,
            MinLength: 1
        },
        serializedName: "resourceGroupName",
        required: true,
        type: {
            name: "String"
        }
    }
};
export const privateCloudName = {
    parameterPath: "privateCloudName",
    mapper: {
        constraints: {
            Pattern: new RegExp("^[-\\w\\._]+$")
        },
        serializedName: "privateCloudName",
        required: true,
        type: {
            name: "String"
        }
    }
};
export const privateCloud = {
    parameterPath: "privateCloud",
    mapper: PrivateCloudMapper
};
export const privateCloudUpdate = {
    parameterPath: "privateCloudUpdate",
    mapper: PrivateCloudUpdateMapper
};
export const clusterName = {
    parameterPath: "clusterName",
    mapper: {
        constraints: {
            Pattern: new RegExp("^[-\\w\\._]+$")
        },
        serializedName: "clusterName",
        required: true,
        type: {
            name: "String"
        }
    }
};
export const cluster = {
    parameterPath: "cluster",
    mapper: ClusterMapper
};
export const privateCloudName1 = {
    parameterPath: "privateCloudName",
    mapper: {
        serializedName: "privateCloudName",
        required: true,
        type: {
            name: "String"
        }
    }
};
export const clusterUpdate = {
    parameterPath: "clusterUpdate",
    mapper: ClusterUpdateMapper
};
export const datastoreName = {
    parameterPath: "datastoreName",
    mapper: {
        constraints: {
            Pattern: new RegExp("^[-\\w\\._]+$")
        },
        serializedName: "datastoreName",
        required: true,
        type: {
            name: "String"
        }
    }
};
export const datastore = {
    parameterPath: "datastore",
    mapper: DatastoreMapper
};
export const hcxEnterpriseSiteName = {
    parameterPath: "hcxEnterpriseSiteName",
    mapper: {
        constraints: {
            Pattern: new RegExp("^[-\\w\\._]+$")
        },
        serializedName: "hcxEnterpriseSiteName",
        required: true,
        type: {
            name: "String"
        }
    }
};
export const hcxEnterpriseSite = {
    parameterPath: "hcxEnterpriseSite",
    mapper: HcxEnterpriseSiteMapper
};
export const authorizationName = {
    parameterPath: "authorizationName",
    mapper: {
        constraints: {
            Pattern: new RegExp("^[-\\w\\._]+$")
        },
        serializedName: "authorizationName",
        required: true,
        type: {
            name: "String"
        }
    }
};
export const authorization = {
    parameterPath: "authorization",
    mapper: ExpressRouteAuthorizationMapper
};
export const globalReachConnectionName = {
    parameterPath: "globalReachConnectionName",
    mapper: {
        constraints: {
            Pattern: new RegExp("^[-\\w\\._]+$")
        },
        serializedName: "globalReachConnectionName",
        required: true,
        type: {
            name: "String"
        }
    }
};
export const globalReachConnection = {
    parameterPath: "globalReachConnection",
    mapper: GlobalReachConnectionMapper
};
export const workloadNetworkName = {
    parameterPath: "workloadNetworkName",
    mapper: {
        serializedName: "workloadNetworkName",
        required: true,
        type: {
            name: "String"
        }
    }
};
export const segmentId = {
    parameterPath: "segmentId",
    mapper: {
        serializedName: "segmentId",
        required: true,
        type: {
            name: "String"
        }
    }
};
export const workloadNetworkSegment = {
    parameterPath: "workloadNetworkSegment",
    mapper: WorkloadNetworkSegmentMapper
};
export const dhcpId = {
    parameterPath: "dhcpId",
    mapper: {
        serializedName: "dhcpId",
        required: true,
        type: {
            name: "String"
        }
    }
};
export const workloadNetworkDhcp = {
    parameterPath: "workloadNetworkDhcp",
    mapper: WorkloadNetworkDhcpMapper
};
export const gatewayId = {
    parameterPath: "gatewayId",
    mapper: {
        serializedName: "gatewayId",
        required: true,
        type: {
            name: "String"
        }
    }
};
export const portMirroringId = {
    parameterPath: "portMirroringId",
    mapper: {
        serializedName: "portMirroringId",
        required: true,
        type: {
            name: "String"
        }
    }
};
export const workloadNetworkPortMirroring = {
    parameterPath: "workloadNetworkPortMirroring",
    mapper: WorkloadNetworkPortMirroringMapper
};
export const vmGroupId = {
    parameterPath: "vmGroupId",
    mapper: {
        serializedName: "vmGroupId",
        required: true,
        type: {
            name: "String"
        }
    }
};
export const workloadNetworkVMGroup = {
    parameterPath: "workloadNetworkVMGroup",
    mapper: WorkloadNetworkVMGroupMapper
};
export const virtualMachineId = {
    parameterPath: "virtualMachineId",
    mapper: {
        serializedName: "virtualMachineId",
        required: true,
        type: {
            name: "String"
        }
    }
};
export const dnsServiceId = {
    parameterPath: "dnsServiceId",
    mapper: {
        serializedName: "dnsServiceId",
        required: true,
        type: {
            name: "String"
        }
    }
};
export const workloadNetworkDnsService = {
    parameterPath: "workloadNetworkDnsService",
    mapper: WorkloadNetworkDnsServiceMapper
};
export const dnsZoneId = {
    parameterPath: "dnsZoneId",
    mapper: {
        serializedName: "dnsZoneId",
        required: true,
        type: {
            name: "String"
        }
    }
};
export const workloadNetworkDnsZone = {
    parameterPath: "workloadNetworkDnsZone",
    mapper: WorkloadNetworkDnsZoneMapper
};
export const publicIPId = {
    parameterPath: "publicIPId",
    mapper: {
        serializedName: "publicIPId",
        required: true,
        type: {
            name: "String"
        }
    }
};
export const workloadNetworkPublicIP = {
    parameterPath: "workloadNetworkPublicIP",
    mapper: WorkloadNetworkPublicIPMapper
};
export const cloudLinkName = {
    parameterPath: "cloudLinkName",
    mapper: {
        constraints: {
            Pattern: new RegExp("^[-\\w\\._]+$")
        },
        serializedName: "cloudLinkName",
        required: true,
        type: {
            name: "String"
        }
    }
};
export const cloudLink = {
    parameterPath: "cloudLink",
    mapper: CloudLinkMapper
};
export const addonName = {
    parameterPath: "addonName",
    mapper: {
        constraints: {
            Pattern: new RegExp("^[-\\w\\._]+$")
        },
        serializedName: "addonName",
        required: true,
        type: {
            name: "String"
        }
    }
};
export const addon = {
    parameterPath: "addon",
    mapper: AddonMapper
};
export const restrictMovement = {
    parameterPath: "restrictMovement",
    mapper: VirtualMachineRestrictMovementMapper
};
export const placementPolicyName = {
    parameterPath: "placementPolicyName",
    mapper: {
        constraints: {
            Pattern: new RegExp("^[-\\w\\._]+$")
        },
        serializedName: "placementPolicyName",
        required: true,
        type: {
            name: "String"
        }
    }
};
export const placementPolicy = {
    parameterPath: "placementPolicy",
    mapper: PlacementPolicyMapper
};
export const placementPolicyUpdate = {
    parameterPath: "placementPolicyUpdate",
    mapper: PlacementPolicyUpdateMapper
};
export const scriptPackageName = {
    parameterPath: "scriptPackageName",
    mapper: {
        constraints: {
            Pattern: new RegExp("^[-\\w\\._@]+$")
        },
        serializedName: "scriptPackageName",
        required: true,
        type: {
            name: "String"
        }
    }
};
export const scriptCmdletName = {
    parameterPath: "scriptCmdletName",
    mapper: {
        constraints: {
            Pattern: new RegExp("^[-\\w\\._]+$")
        },
        serializedName: "scriptCmdletName",
        required: true,
        type: {
            name: "String"
        }
    }
};
export const scriptExecutionName = {
    parameterPath: "scriptExecutionName",
    mapper: {
        constraints: {
            Pattern: new RegExp("^[-\\w\\._]+$")
        },
        serializedName: "scriptExecutionName",
        required: true,
        type: {
            name: "String"
        }
    }
};
export const scriptExecution = {
    parameterPath: "scriptExecution",
    mapper: ScriptExecutionMapper
};
export const scriptOutputStreamType = {
    parameterPath: ["options", "scriptOutputStreamType"],
    mapper: {
        serializedName: "scriptOutputStreamType",
        type: {
            name: "Sequence",
            element: {
                type: {
                    name: "String"
                }
            }
        }
    }
};
