// @azure/arm-operations/esm/models/mappers.js
/*
 * Copyright (c) Microsoft Corporation.
 * Licensed under the MIT License.
 *
 * Code generated by Microsoft (R) AutoRest Code Generator.
 * Changes may cause incorrect behavior and will be lost if the code is regenerated.
 */
import { CloudErrorMapper, BaseResourceMapper } from "@azure/ms-rest-azure-js";
export var CloudError = CloudErrorMapper;
export var BaseResource = BaseResourceMapper;
export var OperationDisplay = {
    serializedName: "Operation_display",
    type: {
        name: "Composite",
        className: "OperationDisplay",
        modelProperties: {
            provider: {
                serializedName: "provider",
                type: {
                    name: "String"
                }
            },
            resource: {
                serializedName: "resource",
                type: {
                    name: "String"
                }
            },
            operation: {
                serializedName: "operation",
                type: {
                    name: "String"
                }
            }
        }
    }
};
export var Operation = {
    serializedName: "Operation",
    type: {
        name: "Composite",
        className: "Operation",
        modelProperties: {
            name: {
                serializedName: "name",
                type: {
                    name: "String"
                }
            },
            display: {
                serializedName: "display",
                type: {
                    name: "Composite",
                    className: "OperationDisplay"
                }
            }
        }
    }
};
export var SolutionProperties = {
    serializedName: "SolutionProperties",
    type: {
        name: "Composite",
        className: "SolutionProperties",
        modelProperties: {
            workspaceResourceId: {
                required: true,
                serializedName: "workspaceResourceId",
                type: {
                    name: "String"
                }
            },
            provisioningState: {
                readOnly: true,
                serializedName: "provisioningState",
                type: {
                    name: "String"
                }
            },
            containedResources: {
                serializedName: "containedResources",
                type: {
                    name: "Sequence",
                    element: {
                        type: {
                            name: "String"
                        }
                    }
                }
            },
            referencedResources: {
                serializedName: "referencedResources",
                type: {
                    name: "Sequence",
                    element: {
                        type: {
                            name: "String"
                        }
                    }
                }
            }
        }
    }
};
export var ManagementAssociationProperties = {
    serializedName: "ManagementAssociationProperties",
    type: {
        name: "Composite",
        className: "ManagementAssociationProperties",
        modelProperties: {
            applicationId: {
                required: true,
                serializedName: "applicationId",
                type: {
                    name: "String"
                }
            }
        }
    }
};
export var ArmTemplateParameter = {
    serializedName: "ArmTemplateParameter",
    type: {
        name: "Composite",
        className: "ArmTemplateParameter",
        modelProperties: {
            name: {
                serializedName: "name",
                type: {
                    name: "String"
                }
            },
            value: {
                serializedName: "value",
                type: {
                    name: "String"
                }
            }
        }
    }
};
export var ManagementConfigurationProperties = {
    serializedName: "ManagementConfigurationProperties",
    type: {
        name: "Composite",
        className: "ManagementConfigurationProperties",
        modelProperties: {
            applicationId: {
                serializedName: "applicationId",
                type: {
                    name: "String"
                }
            },
            parentResourceType: {
                required: true,
                serializedName: "parentResourceType",
                type: {
                    name: "String"
                }
            },
            parameters: {
                required: true,
                serializedName: "parameters",
                type: {
                    name: "Sequence",
                    element: {
                        type: {
                            name: "Composite",
                            className: "ArmTemplateParameter"
                        }
                    }
                }
            },
            provisioningState: {
                readOnly: true,
                serializedName: "provisioningState",
                type: {
                    name: "String"
                }
            },
            template: {
                required: true,
                serializedName: "template",
                type: {
                    name: "Object"
                }
            }
        }
    }
};
export var SolutionPlan = {
    serializedName: "SolutionPlan",
    type: {
        name: "Composite",
        className: "SolutionPlan",
        modelProperties: {
            name: {
                serializedName: "name",
                type: {
                    name: "String"
                }
            },
            publisher: {
                serializedName: "publisher",
                type: {
                    name: "String"
                }
            },
            promotionCode: {
                serializedName: "promotionCode",
                type: {
                    name: "String"
                }
            },
            product: {
                serializedName: "product",
                type: {
                    name: "String"
                }
            }
        }
    }
};
export var Solution = {
    serializedName: "Solution",
    type: {
        name: "Composite",
        className: "Solution",
        modelProperties: {
            id: {
                readOnly: true,
                serializedName: "id",
                type: {
                    name: "String"
                }
            },
            name: {
                readOnly: true,
                serializedName: "name",
                type: {
                    name: "String"
                }
            },
            type: {
                readOnly: true,
                serializedName: "type",
                type: {
                    name: "String"
                }
            },
            location: {
                serializedName: "location",
                type: {
                    name: "String"
                }
            },
            tags: {
                serializedName: "tags",
                type: {
                    name: "Dictionary",
                    value: {
                        type: {
                            name: "String"
                        }
                    }
                }
            },
            plan: {
                serializedName: "plan",
                type: {
                    name: "Composite",
                    className: "SolutionPlan"
                }
            },
            properties: {
                serializedName: "properties",
                type: {
                    name: "Composite",
                    className: "SolutionProperties"
                }
            }
        }
    }
};
export var SolutionPropertiesList = {
    serializedName: "SolutionPropertiesList",
    type: {
        name: "Composite",
        className: "SolutionPropertiesList",
        modelProperties: {
            value: {
                serializedName: "value",
                type: {
                    name: "Sequence",
                    element: {
                        type: {
                            name: "Composite",
                            className: "Solution"
                        }
                    }
                }
            }
        }
    }
};
export var ManagementAssociation = {
    serializedName: "ManagementAssociation",
    type: {
        name: "Composite",
        className: "ManagementAssociation",
        modelProperties: {
            id: {
                readOnly: true,
                serializedName: "id",
                type: {
                    name: "String"
                }
            },
            name: {
                readOnly: true,
                serializedName: "name",
                type: {
                    name: "String"
                }
            },
            type: {
                readOnly: true,
                serializedName: "type",
                type: {
                    name: "String"
                }
            },
            location: {
                serializedName: "location",
                type: {
                    name: "String"
                }
            },
            properties: {
                serializedName: "properties",
                type: {
                    name: "Composite",
                    className: "ManagementAssociationProperties"
                }
            }
        }
    }
};
export var ManagementAssociationPropertiesList = {
    serializedName: "ManagementAssociationPropertiesList",
    type: {
        name: "Composite",
        className: "ManagementAssociationPropertiesList",
        modelProperties: {
            value: {
                serializedName: "value",
                type: {
                    name: "Sequence",
                    element: {
                        type: {
                            name: "Composite",
                            className: "ManagementAssociation"
                        }
                    }
                }
            }
        }
    }
};
export var ManagementConfiguration = {
    serializedName: "ManagementConfiguration",
    type: {
        name: "Composite",
        className: "ManagementConfiguration",
        modelProperties: {
            id: {
                readOnly: true,
                serializedName: "id",
                type: {
                    name: "String"
                }
            },
            name: {
                readOnly: true,
                serializedName: "name",
                type: {
                    name: "String"
                }
            },
            type: {
                readOnly: true,
                serializedName: "type",
                type: {
                    name: "String"
                }
            },
            location: {
                serializedName: "location",
                type: {
                    name: "String"
                }
            },
            properties: {
                serializedName: "properties",
                type: {
                    name: "Composite",
                    className: "ManagementConfigurationProperties"
                }
            }
        }
    }
};
export var ManagementConfigurationPropertiesList = {
    serializedName: "ManagementConfigurationPropertiesList",
    type: {
        name: "Composite",
        className: "ManagementConfigurationPropertiesList",
        modelProperties: {
            value: {
                serializedName: "value",
                type: {
                    name: "Sequence",
                    element: {
                        type: {
                            name: "Composite",
                            className: "ManagementConfiguration"
                        }
                    }
                }
            }
        }
    }
};
export var SolutionPatch = {
    serializedName: "SolutionPatch",
    type: {
        name: "Composite",
        className: "SolutionPatch",
        modelProperties: {
            tags: {
                serializedName: "tags",
                type: {
                    name: "Dictionary",
                    value: {
                        type: {
                            name: "String"
                        }
                    }
                }
            }
        }
    }
};
export var CodeMessageErrorError = {
    serializedName: "CodeMessageError_error",
    type: {
        name: "Composite",
        className: "CodeMessageErrorError",
        modelProperties: {
            code: {
                serializedName: "code",
                type: {
                    name: "String"
                }
            },
            message: {
                serializedName: "message",
                type: {
                    name: "String"
                }
            }
        }
    }
};
export var CodeMessageError = {
    serializedName: "CodeMessageError",
    type: {
        name: "Composite",
        className: "CodeMessageError",
        modelProperties: {
            error: {
                serializedName: "error",
                type: {
                    name: "Composite",
                    className: "CodeMessageErrorError"
                }
            }
        }
    }
};
export var OperationListResult = {
    serializedName: "OperationListResult",
    type: {
        name: "Composite",
        className: "OperationListResult",
        modelProperties: {
            value: {
                serializedName: "",
                type: {
                    name: "Sequence",
                    element: {
                        type: {
                            name: "Composite",
                            className: "Operation"
                        }
                    }
                }
            }
        }
    }
};
