// @azure/arm-operations/esm/models/parameters.js
/*
 * Copyright (c) Microsoft Corporation.
 * Licensed under the MIT License.
 *
 * Code generated by Microsoft (R) AutoRest Code Generator.
 * Changes may cause incorrect behavior and will be lost if the code is
 * regenerated.
 */
export var acceptLanguage = {
    parameterPath: "acceptLanguage",
    mapper: {
        serializedName: "accept-language",
        defaultValue: 'en-US',
        type: {
            name: "String"
        }
    }
};
export var apiVersion = {
    parameterPath: "apiVersion",
    mapper: {
        required: true,
        serializedName: "api-version",
        type: {
            name: "String"
        }
    }
};
export var managementAssociationName = {
    parameterPath: "managementAssociationName",
    mapper: {
        required: true,
        serializedName: "managementAssociationName",
        type: {
            name: "String"
        }
    }
};
export var managementConfigurationName = {
    parameterPath: "managementConfigurationName",
    mapper: {
        required: true,
        serializedName: "managementConfigurationName",
        type: {
            name: "String"
        }
    }
};
export var providerName = {
    parameterPath: "providerName",
    mapper: {
        required: true,
        serializedName: "providerName",
        type: {
            name: "String"
        }
    }
};
export var resourceGroupName = {
    parameterPath: "resourceGroupName",
    mapper: {
        required: true,
        serializedName: "resourceGroupName",
        constraints: {
            MaxLength: 90,
            MinLength: 1,
            Pattern: /^[-\w\._\(\)]+$/
        },
        type: {
            name: "String"
        }
    }
};
export var resourceName = {
    parameterPath: "resourceName",
    mapper: {
        required: true,
        serializedName: "resourceName",
        type: {
            name: "String"
        }
    }
};
export var resourceType = {
    parameterPath: "resourceType",
    mapper: {
        required: true,
        serializedName: "resourceType",
        type: {
            name: "String"
        }
    }
};
export var solutionName = {
    parameterPath: "solutionName",
    mapper: {
        required: true,
        serializedName: "solutionName",
        type: {
            name: "String"
        }
    }
};
export var subscriptionId = {
    parameterPath: "subscriptionId",
    mapper: {
        required: true,
        serializedName: "subscriptionId",
        type: {
            name: "String"
        }
    }
};
