// @google-cloud/retail/build/src/v2/index.js
"use strict";
// Copyright 2023 Google LLC
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     https://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.
//
// ** This file is automatically generated by gapic-generator-typescript. **
// ** https://github.com/googleapis/gapic-generator-typescript **
// ** All changes to this file may be overwritten. **
Object.defineProperty(exports, "__esModule", { value: true });
exports.UserEventServiceClient = exports.ServingConfigServiceClient = exports.SearchServiceClient = exports.ProductServiceClient = exports.PredictionServiceClient = exports.ModelServiceClient = exports.ControlServiceClient = exports.CompletionServiceClient = exports.CatalogServiceClient = void 0;
var catalog_service_client_1 = require("./catalog_service_client");
Object.defineProperty(exports, "CatalogServiceClient", { enumerable: true, get: function () { return catalog_service_client_1.CatalogServiceClient; } });
var completion_service_client_1 = require("./completion_service_client");
Object.defineProperty(exports, "CompletionServiceClient", { enumerable: true, get: function () { return completion_service_client_1.CompletionServiceClient; } });
var control_service_client_1 = require("./control_service_client");
Object.defineProperty(exports, "ControlServiceClient", { enumerable: true, get: function () { return control_service_client_1.ControlServiceClient; } });
var model_service_client_1 = require("./model_service_client");
Object.defineProperty(exports, "ModelServiceClient", { enumerable: true, get: function () { return model_service_client_1.ModelServiceClient; } });
var prediction_service_client_1 = require("./prediction_service_client");
Object.defineProperty(exports, "PredictionServiceClient", { enumerable: true, get: function () { return prediction_service_client_1.PredictionServiceClient; } });
var product_service_client_1 = require("./product_service_client");
Object.defineProperty(exports, "ProductServiceClient", { enumerable: true, get: function () { return product_service_client_1.ProductServiceClient; } });
var search_service_client_1 = require("./search_service_client");
Object.defineProperty(exports, "SearchServiceClient", { enumerable: true, get: function () { return search_service_client_1.SearchServiceClient; } });
var serving_config_service_client_1 = require("./serving_config_service_client");
Object.defineProperty(exports, "ServingConfigServiceClient", { enumerable: true, get: function () { return serving_config_service_client_1.ServingConfigServiceClient; } });
var user_event_service_client_1 = require("./user_event_service_client");
Object.defineProperty(exports, "UserEventServiceClient", { enumerable: true, get: function () { return user_event_service_client_1.UserEventServiceClient; } });
