// @aws-sdk/client-forecast/dist-cjs/index.js
"use strict";
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __name = (target, value) => __defProp(target, "name", { value, configurable: true });
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// src/index.ts
var src_exports = {};
__export(src_exports, {
  AttributeType: () => AttributeType,
  AutoMLOverrideStrategy: () => AutoMLOverrideStrategy,
  Condition: () => Condition,
  CreateAutoPredictorCommand: () => CreateAutoPredictorCommand,
  CreateAutoPredictorRequestFilterSensitiveLog: () => CreateAutoPredictorRequestFilterSensitiveLog,
  CreateDatasetCommand: () => CreateDatasetCommand,
  CreateDatasetGroupCommand: () => CreateDatasetGroupCommand,
  CreateDatasetGroupRequestFilterSensitiveLog: () => CreateDatasetGroupRequestFilterSensitiveLog,
  CreateDatasetImportJobCommand: () => CreateDatasetImportJobCommand,
  CreateDatasetImportJobRequestFilterSensitiveLog: () => CreateDatasetImportJobRequestFilterSensitiveLog,
  CreateDatasetRequestFilterSensitiveLog: () => CreateDatasetRequestFilterSensitiveLog,
  CreateExplainabilityCommand: () => CreateExplainabilityCommand,
  CreateExplainabilityExportCommand: () => CreateExplainabilityExportCommand,
  CreateExplainabilityExportRequestFilterSensitiveLog: () => CreateExplainabilityExportRequestFilterSensitiveLog,
  CreateExplainabilityRequestFilterSensitiveLog: () => CreateExplainabilityRequestFilterSensitiveLog,
  CreateForecastCommand: () => CreateForecastCommand,
  CreateForecastExportJobCommand: () => CreateForecastExportJobCommand,
  CreateForecastExportJobRequestFilterSensitiveLog: () => CreateForecastExportJobRequestFilterSensitiveLog,
  CreateForecastRequestFilterSensitiveLog: () => CreateForecastRequestFilterSensitiveLog,
  CreateMonitorCommand: () => CreateMonitorCommand,
  CreateMonitorRequestFilterSensitiveLog: () => CreateMonitorRequestFilterSensitiveLog,
  CreatePredictorBacktestExportJobCommand: () => CreatePredictorBacktestExportJobCommand,
  CreatePredictorBacktestExportJobRequestFilterSensitiveLog: () => CreatePredictorBacktestExportJobRequestFilterSensitiveLog,
  CreatePredictorCommand: () => CreatePredictorCommand,
  CreatePredictorRequestFilterSensitiveLog: () => CreatePredictorRequestFilterSensitiveLog,
  CreateWhatIfAnalysisCommand: () => CreateWhatIfAnalysisCommand,
  CreateWhatIfAnalysisRequestFilterSensitiveLog: () => CreateWhatIfAnalysisRequestFilterSensitiveLog,
  CreateWhatIfForecastCommand: () => CreateWhatIfForecastCommand,
  CreateWhatIfForecastExportCommand: () => CreateWhatIfForecastExportCommand,
  CreateWhatIfForecastExportRequestFilterSensitiveLog: () => CreateWhatIfForecastExportRequestFilterSensitiveLog,
  CreateWhatIfForecastRequestFilterSensitiveLog: () => CreateWhatIfForecastRequestFilterSensitiveLog,
  DatasetType: () => DatasetType,
  DayOfWeek: () => DayOfWeek,
  DeleteDatasetCommand: () => DeleteDatasetCommand,
  DeleteDatasetGroupCommand: () => DeleteDatasetGroupCommand,
  DeleteDatasetImportJobCommand: () => DeleteDatasetImportJobCommand,
  DeleteExplainabilityCommand: () => DeleteExplainabilityCommand,
  DeleteExplainabilityExportCommand: () => DeleteExplainabilityExportCommand,
  DeleteForecastCommand: () => DeleteForecastCommand,
  DeleteForecastExportJobCommand: () => DeleteForecastExportJobCommand,
  DeleteMonitorCommand: () => DeleteMonitorCommand,
  DeletePredictorBacktestExportJobCommand: () => DeletePredictorBacktestExportJobCommand,
  DeletePredictorCommand: () => DeletePredictorCommand,
  DeleteResourceTreeCommand: () => DeleteResourceTreeCommand,
  DeleteWhatIfAnalysisCommand: () => DeleteWhatIfAnalysisCommand,
  DeleteWhatIfForecastCommand: () => DeleteWhatIfForecastCommand,
  DeleteWhatIfForecastExportCommand: () => DeleteWhatIfForecastExportCommand,
  DescribeAutoPredictorCommand: () => DescribeAutoPredictorCommand,
  DescribeDatasetCommand: () => DescribeDatasetCommand,
  DescribeDatasetGroupCommand: () => DescribeDatasetGroupCommand,
  DescribeDatasetImportJobCommand: () => DescribeDatasetImportJobCommand,
  DescribeExplainabilityCommand: () => DescribeExplainabilityCommand,
  DescribeExplainabilityExportCommand: () => DescribeExplainabilityExportCommand,
  DescribeForecastCommand: () => DescribeForecastCommand,
  DescribeForecastExportJobCommand: () => DescribeForecastExportJobCommand,
  DescribeMonitorCommand: () => DescribeMonitorCommand,
  DescribePredictorBacktestExportJobCommand: () => DescribePredictorBacktestExportJobCommand,
  DescribePredictorCommand: () => DescribePredictorCommand,
  DescribeWhatIfAnalysisCommand: () => DescribeWhatIfAnalysisCommand,
  DescribeWhatIfForecastCommand: () => DescribeWhatIfForecastCommand,
  DescribeWhatIfForecastExportCommand: () => DescribeWhatIfForecastExportCommand,
  Domain: () => Domain,
  EvaluationType: () => EvaluationType,
  FeaturizationMethodName: () => FeaturizationMethodName,
  FilterConditionString: () => FilterConditionString,
  Forecast: () => Forecast,
  ForecastClient: () => ForecastClient,
  ForecastServiceException: () => ForecastServiceException,
  GetAccuracyMetricsCommand: () => GetAccuracyMetricsCommand,
  ImportMode: () => ImportMode,
  InvalidInputException: () => InvalidInputException,
  InvalidNextTokenException: () => InvalidNextTokenException,
  LimitExceededException: () => LimitExceededException,
  ListDatasetGroupsCommand: () => ListDatasetGroupsCommand,
  ListDatasetImportJobsCommand: () => ListDatasetImportJobsCommand,
  ListDatasetsCommand: () => ListDatasetsCommand,
  ListExplainabilitiesCommand: () => ListExplainabilitiesCommand,
  ListExplainabilityExportsCommand: () => ListExplainabilityExportsCommand,
  ListForecastExportJobsCommand: () => ListForecastExportJobsCommand,
  ListForecastsCommand: () => ListForecastsCommand,
  ListMonitorEvaluationsCommand: () => ListMonitorEvaluationsCommand,
  ListMonitorsCommand: () => ListMonitorsCommand,
  ListPredictorBacktestExportJobsCommand: () => ListPredictorBacktestExportJobsCommand,
  ListPredictorsCommand: () => ListPredictorsCommand,
  ListTagsForResourceCommand: () => ListTagsForResourceCommand,
  ListTagsForResourceResponseFilterSensitiveLog: () => ListTagsForResourceResponseFilterSensitiveLog,
  ListWhatIfAnalysesCommand: () => ListWhatIfAnalysesCommand,
  ListWhatIfForecastExportsCommand: () => ListWhatIfForecastExportsCommand,
  ListWhatIfForecastsCommand: () => ListWhatIfForecastsCommand,
  Month: () => Month,
  Operation: () => Operation,
  OptimizationMetric: () => OptimizationMetric,
  ResourceAlreadyExistsException: () => ResourceAlreadyExistsException,
  ResourceInUseException: () => ResourceInUseException,
  ResourceNotFoundException: () => ResourceNotFoundException,
  ResumeResourceCommand: () => ResumeResourceCommand,
  ScalingType: () => ScalingType,
  State: () => State,
  StopResourceCommand: () => StopResourceCommand,
  TagFilterSensitiveLog: () => TagFilterSensitiveLog,
  TagResourceCommand: () => TagResourceCommand,
  TagResourceRequestFilterSensitiveLog: () => TagResourceRequestFilterSensitiveLog,
  TimePointGranularity: () => TimePointGranularity,
  TimeSeriesGranularity: () => TimeSeriesGranularity,
  UntagResourceCommand: () => UntagResourceCommand,
  UntagResourceRequestFilterSensitiveLog: () => UntagResourceRequestFilterSensitiveLog,
  UpdateDatasetGroupCommand: () => UpdateDatasetGroupCommand,
  __Client: () => import_smithy_client.Client,
  paginateListDatasetGroups: () => paginateListDatasetGroups,
  paginateListDatasetImportJobs: () => paginateListDatasetImportJobs,
  paginateListDatasets: () => paginateListDatasets,
  paginateListExplainabilities: () => paginateListExplainabilities,
  paginateListExplainabilityExports: () => paginateListExplainabilityExports,
  paginateListForecastExportJobs: () => paginateListForecastExportJobs,
  paginateListForecasts: () => paginateListForecasts,
  paginateListMonitorEvaluations: () => paginateListMonitorEvaluations,
  paginateListMonitors: () => paginateListMonitors,
  paginateListPredictorBacktestExportJobs: () => paginateListPredictorBacktestExportJobs,
  paginateListPredictors: () => paginateListPredictors,
  paginateListWhatIfAnalyses: () => paginateListWhatIfAnalyses,
  paginateListWhatIfForecastExports: () => paginateListWhatIfForecastExports,
  paginateListWhatIfForecasts: () => paginateListWhatIfForecasts
});
module.exports = __toCommonJS(src_exports);

// src/ForecastClient.ts
var import_middleware_host_header = require("@aws-sdk/middleware-host-header");
var import_middleware_logger = require("@aws-sdk/middleware-logger");
var import_middleware_recursion_detection = require("@aws-sdk/middleware-recursion-detection");
var import_middleware_user_agent = require("@aws-sdk/middleware-user-agent");
var import_config_resolver = require("@smithy/config-resolver");
var import_core = require("@smithy/core");
var import_middleware_content_length = require("@smithy/middleware-content-length");
var import_middleware_endpoint = require("@smithy/middleware-endpoint");
var import_middleware_retry = require("@smithy/middleware-retry");

var import_httpAuthSchemeProvider = require("./auth/httpAuthSchemeProvider");

// src/endpoint/EndpointParameters.ts
var resolveClientEndpointParameters = /* @__PURE__ */ __name((options) => {
  return {
    ...options,
    useDualstackEndpoint: options.useDualstackEndpoint ?? false,
    useFipsEndpoint: options.useFipsEndpoint ?? false,
    defaultSigningName: "forecast"
  };
}, "resolveClientEndpointParameters");
var commonParams = {
  UseFIPS: { type: "builtInParams", name: "useFipsEndpoint" },
  Endpoint: { type: "builtInParams", name: "endpoint" },
  Region: { type: "builtInParams", name: "region" },
  UseDualStack: { type: "builtInParams", name: "useDualstackEndpoint" }
};

// src/ForecastClient.ts
var import_runtimeConfig = require("././runtimeConfig");

// src/runtimeExtensions.ts
var import_region_config_resolver = require("@aws-sdk/region-config-resolver");
var import_protocol_http = require("@smithy/protocol-http");
var import_smithy_client = require("@smithy/smithy-client");

// src/auth/httpAuthExtensionConfiguration.ts
var getHttpAuthExtensionConfiguration = /* @__PURE__ */ __name((runtimeConfig) => {
  const _httpAuthSchemes = runtimeConfig.httpAuthSchemes;
  let _httpAuthSchemeProvider = runtimeConfig.httpAuthSchemeProvider;
  let _credentials = runtimeConfig.credentials;
  return {
    setHttpAuthScheme(httpAuthScheme) {
      const index = _httpAuthSchemes.findIndex((scheme) => scheme.schemeId === httpAuthScheme.schemeId);
      if (index === -1) {
        _httpAuthSchemes.push(httpAuthScheme);
      } else {
        _httpAuthSchemes.splice(index, 1, httpAuthScheme);
      }
    },
    httpAuthSchemes() {
      return _httpAuthSchemes;
    },
    setHttpAuthSchemeProvider(httpAuthSchemeProvider) {
      _httpAuthSchemeProvider = httpAuthSchemeProvider;
    },
    httpAuthSchemeProvider() {
      return _httpAuthSchemeProvider;
    },
    setCredentials(credentials) {
      _credentials = credentials;
    },
    credentials() {
      return _credentials;
    }
  };
}, "getHttpAuthExtensionConfiguration");
var resolveHttpAuthRuntimeConfig = /* @__PURE__ */ __name((config) => {
  return {
    httpAuthSchemes: config.httpAuthSchemes(),
    httpAuthSchemeProvider: config.httpAuthSchemeProvider(),
    credentials: config.credentials()
  };
}, "resolveHttpAuthRuntimeConfig");

// src/runtimeExtensions.ts
var asPartial = /* @__PURE__ */ __name((t) => t, "asPartial");
var resolveRuntimeExtensions = /* @__PURE__ */ __name((runtimeConfig, extensions) => {
  const extensionConfiguration = {
    ...asPartial((0, import_region_config_resolver.getAwsRegionExtensionConfiguration)(runtimeConfig)),
    ...asPartial((0, import_smithy_client.getDefaultExtensionConfiguration)(runtimeConfig)),
    ...asPartial((0, import_protocol_http.getHttpHandlerExtensionConfiguration)(runtimeConfig)),
    ...asPartial(getHttpAuthExtensionConfiguration(runtimeConfig))
  };
  extensions.forEach((extension) => extension.configure(extensionConfiguration));
  return {
    ...runtimeConfig,
    ...(0, import_region_config_resolver.resolveAwsRegionExtensionConfiguration)(extensionConfiguration),
    ...(0, import_smithy_client.resolveDefaultRuntimeConfig)(extensionConfiguration),
    ...(0, import_protocol_http.resolveHttpHandlerRuntimeConfig)(extensionConfiguration),
    ...resolveHttpAuthRuntimeConfig(extensionConfiguration)
  };
}, "resolveRuntimeExtensions");

// src/ForecastClient.ts
var _ForecastClient = class _ForecastClient extends import_smithy_client.Client {
  constructor(...[configuration]) {
    const _config_0 = (0, import_runtimeConfig.getRuntimeConfig)(configuration || {});
    const _config_1 = resolveClientEndpointParameters(_config_0);
    const _config_2 = (0, import_config_resolver.resolveRegionConfig)(_config_1);
    const _config_3 = (0, import_middleware_endpoint.resolveEndpointConfig)(_config_2);
    const _config_4 = (0, import_middleware_retry.resolveRetryConfig)(_config_3);
    const _config_5 = (0, import_middleware_host_header.resolveHostHeaderConfig)(_config_4);
    const _config_6 = (0, import_middleware_user_agent.resolveUserAgentConfig)(_config_5);
    const _config_7 = (0, import_httpAuthSchemeProvider.resolveHttpAuthSchemeConfig)(_config_6);
    const _config_8 = resolveRuntimeExtensions(_config_7, (configuration == null ? void 0 : configuration.extensions) || []);
    super(_config_8);
    this.config = _config_8;
    this.middlewareStack.use((0, import_middleware_retry.getRetryPlugin)(this.config));
    this.middlewareStack.use((0, import_middleware_content_length.getContentLengthPlugin)(this.config));
    this.middlewareStack.use((0, import_middleware_host_header.getHostHeaderPlugin)(this.config));
    this.middlewareStack.use((0, import_middleware_logger.getLoggerPlugin)(this.config));
    this.middlewareStack.use((0, import_middleware_recursion_detection.getRecursionDetectionPlugin)(this.config));
    this.middlewareStack.use((0, import_middleware_user_agent.getUserAgentPlugin)(this.config));
    this.middlewareStack.use(
      (0, import_core.getHttpAuthSchemeEndpointRuleSetPlugin)(this.config, {
        httpAuthSchemeParametersProvider: this.getDefaultHttpAuthSchemeParametersProvider(),
        identityProviderConfigProvider: this.getIdentityProviderConfigProvider()
      })
    );
    this.middlewareStack.use((0, import_core.getHttpSigningPlugin)(this.config));
  }
  /**
   * Destroy underlying resources, like sockets. It's usually not necessary to do this.
   * However in Node.js, it's best to explicitly shut down the client's agent when it is no longer needed.
   * Otherwise, sockets might stay open for quite a long time before the server terminates them.
   */
  destroy() {
    super.destroy();
  }
  getDefaultHttpAuthSchemeParametersProvider() {
    return import_httpAuthSchemeProvider.defaultForecastHttpAuthSchemeParametersProvider;
  }
  getIdentityProviderConfigProvider() {
    return async (config) => new import_core.DefaultIdentityProviderConfig({
      "aws.auth#sigv4": config.credentials
    });
  }
};
__name(_ForecastClient, "ForecastClient");
var ForecastClient = _ForecastClient;

// src/Forecast.ts


// src/commands/CreateAutoPredictorCommand.ts

var import_middleware_serde = require("@smithy/middleware-serde");

var import_types = require("@smithy/types");

// src/models/models_0.ts


// src/models/ForecastServiceException.ts

var _ForecastServiceException = class _ForecastServiceException extends import_smithy_client.ServiceException {
  /**
   * @internal
   */
  constructor(options) {
    super(options);
    Object.setPrototypeOf(this, _ForecastServiceException.prototype);
  }
};
__name(_ForecastServiceException, "ForecastServiceException");
var ForecastServiceException = _ForecastServiceException;

// src/models/models_0.ts
var Operation = {
  ADD: "ADD",
  DIVIDE: "DIVIDE",
  MULTIPLY: "MULTIPLY",
  SUBTRACT: "SUBTRACT"
};
var OptimizationMetric = {
  AverageWeightedQuantileLoss: "AverageWeightedQuantileLoss",
  MAPE: "MAPE",
  MASE: "MASE",
  RMSE: "RMSE",
  WAPE: "WAPE"
};
var DayOfWeek = {
  FRIDAY: "FRIDAY",
  MONDAY: "MONDAY",
  SATURDAY: "SATURDAY",
  SUNDAY: "SUNDAY",
  THURSDAY: "THURSDAY",
  TUESDAY: "TUESDAY",
  WEDNESDAY: "WEDNESDAY"
};
var Month = {
  APRIL: "APRIL",
  AUGUST: "AUGUST",
  DECEMBER: "DECEMBER",
  FEBRUARY: "FEBRUARY",
  JANUARY: "JANUARY",
  JULY: "JULY",
  JUNE: "JUNE",
  MARCH: "MARCH",
  MAY: "MAY",
  NOVEMBER: "NOVEMBER",
  OCTOBER: "OCTOBER",
  SEPTEMBER: "SEPTEMBER"
};
var _InvalidInputException = class _InvalidInputException extends ForecastServiceException {
  /**
   * @internal
   */
  constructor(opts) {
    super({
      name: "InvalidInputException",
      $fault: "client",
      ...opts
    });
    this.name = "InvalidInputException";
    this.$fault = "client";
    Object.setPrototypeOf(this, _InvalidInputException.prototype);
    this.Message = opts.Message;
  }
};
__name(_InvalidInputException, "InvalidInputException");
var InvalidInputException = _InvalidInputException;
var _LimitExceededException = class _LimitExceededException extends ForecastServiceException {
  /**
   * @internal
   */
  constructor(opts) {
    super({
      name: "LimitExceededException",
      $fault: "client",
      ...opts
    });
    this.name = "LimitExceededException";
    this.$fault = "client";
    Object.setPrototypeOf(this, _LimitExceededException.prototype);
    this.Message = opts.Message;
  }
};
__name(_LimitExceededException, "LimitExceededException");
var LimitExceededException = _LimitExceededException;
var _ResourceAlreadyExistsException = class _ResourceAlreadyExistsException extends ForecastServiceException {
  /**
   * @internal
   */
  constructor(opts) {
    super({
      name: "ResourceAlreadyExistsException",
      $fault: "client",
      ...opts
    });
    this.name = "ResourceAlreadyExistsException";
    this.$fault = "client";
    Object.setPrototypeOf(this, _ResourceAlreadyExistsException.prototype);
    this.Message = opts.Message;
  }
};
__name(_ResourceAlreadyExistsException, "ResourceAlreadyExistsException");
var ResourceAlreadyExistsException = _ResourceAlreadyExistsException;
var _ResourceInUseException = class _ResourceInUseException extends ForecastServiceException {
  /**
   * @internal
   */
  constructor(opts) {
    super({
      name: "ResourceInUseException",
      $fault: "client",
      ...opts
    });
    this.name = "ResourceInUseException";
    this.$fault = "client";
    Object.setPrototypeOf(this, _ResourceInUseException.prototype);
    this.Message = opts.Message;
  }
};
__name(_ResourceInUseException, "ResourceInUseException");
var ResourceInUseException = _ResourceInUseException;
var _ResourceNotFoundException = class _ResourceNotFoundException extends ForecastServiceException {
  /**
   * @internal
   */
  constructor(opts) {
    super({
      name: "ResourceNotFoundException",
      $fault: "client",
      ...opts
    });
    this.name = "ResourceNotFoundException";
    this.$fault = "client";
    Object.setPrototypeOf(this, _ResourceNotFoundException.prototype);
    this.Message = opts.Message;
  }
};
__name(_ResourceNotFoundException, "ResourceNotFoundException");
var ResourceNotFoundException = _ResourceNotFoundException;
var DatasetType = {
  ITEM_METADATA: "ITEM_METADATA",
  RELATED_TIME_SERIES: "RELATED_TIME_SERIES",
  TARGET_TIME_SERIES: "TARGET_TIME_SERIES"
};
var Domain = {
  CUSTOM: "CUSTOM",
  EC2_CAPACITY: "EC2_CAPACITY",
  INVENTORY_PLANNING: "INVENTORY_PLANNING",
  METRICS: "METRICS",
  RETAIL: "RETAIL",
  WEB_TRAFFIC: "WEB_TRAFFIC",
  WORK_FORCE: "WORK_FORCE"
};
var AttributeType = {
  FLOAT: "float",
  GEOLOCATION: "geolocation",
  INTEGER: "integer",
  STRING: "string",
  TIMESTAMP: "timestamp"
};
var ImportMode = {
  FULL: "FULL",
  INCREMENTAL: "INCREMENTAL"
};
var TimePointGranularity = {
  ALL: "ALL",
  SPECIFIC: "SPECIFIC"
};
var TimeSeriesGranularity = {
  ALL: "ALL",
  SPECIFIC: "SPECIFIC"
};
var AutoMLOverrideStrategy = {
  AccuracyOptimized: "AccuracyOptimized",
  LatencyOptimized: "LatencyOptimized"
};
var FeaturizationMethodName = {
  filling: "filling"
};
var ScalingType = {
  Auto: "Auto",
  Linear: "Linear",
  Logarithmic: "Logarithmic",
  ReverseLogarithmic: "ReverseLogarithmic"
};
var Condition = {
  EQUALS: "EQUALS",
  GREATER_THAN: "GREATER_THAN",
  LESS_THAN: "LESS_THAN",
  NOT_EQUALS: "NOT_EQUALS"
};
var State = {
  Active: "Active",
  Deleted: "Deleted"
};
var EvaluationType = {
  COMPUTED: "COMPUTED",
  SUMMARY: "SUMMARY"
};
var _InvalidNextTokenException = class _InvalidNextTokenException extends ForecastServiceException {
  /**
   * @internal
   */
  constructor(opts) {
    super({
      name: "InvalidNextTokenException",
      $fault: "client",
      ...opts
    });
    this.name = "InvalidNextTokenException";
    this.$fault = "client";
    Object.setPrototypeOf(this, _InvalidNextTokenException.prototype);
    this.Message = opts.Message;
  }
};
__name(_InvalidNextTokenException, "InvalidNextTokenException");
var InvalidNextTokenException = _InvalidNextTokenException;
var FilterConditionString = {
  IS: "IS",
  IS_NOT: "IS_NOT"
};
var TagFilterSensitiveLog = /* @__PURE__ */ __name((obj) => ({
  ...obj,
  ...obj.Key && { Key: import_smithy_client.SENSITIVE_STRING },
  ...obj.Value && { Value: import_smithy_client.SENSITIVE_STRING }
}), "TagFilterSensitiveLog");
var CreateAutoPredictorRequestFilterSensitiveLog = /* @__PURE__ */ __name((obj) => ({
  ...obj,
  ...obj.Tags && { Tags: obj.Tags.map((item) => TagFilterSensitiveLog(item)) }
}), "CreateAutoPredictorRequestFilterSensitiveLog");
var CreateDatasetRequestFilterSensitiveLog = /* @__PURE__ */ __name((obj) => ({
  ...obj,
  ...obj.Tags && { Tags: obj.Tags.map((item) => TagFilterSensitiveLog(item)) }
}), "CreateDatasetRequestFilterSensitiveLog");
var CreateDatasetGroupRequestFilterSensitiveLog = /* @__PURE__ */ __name((obj) => ({
  ...obj,
  ...obj.Tags && { Tags: obj.Tags.map((item) => TagFilterSensitiveLog(item)) }
}), "CreateDatasetGroupRequestFilterSensitiveLog");
var CreateDatasetImportJobRequestFilterSensitiveLog = /* @__PURE__ */ __name((obj) => ({
  ...obj,
  ...obj.Tags && { Tags: obj.Tags.map((item) => TagFilterSensitiveLog(item)) }
}), "CreateDatasetImportJobRequestFilterSensitiveLog");
var CreateExplainabilityRequestFilterSensitiveLog = /* @__PURE__ */ __name((obj) => ({
  ...obj,
  ...obj.Tags && { Tags: obj.Tags.map((item) => TagFilterSensitiveLog(item)) }
}), "CreateExplainabilityRequestFilterSensitiveLog");
var CreateExplainabilityExportRequestFilterSensitiveLog = /* @__PURE__ */ __name((obj) => ({
  ...obj,
  ...obj.Tags && { Tags: obj.Tags.map((item) => TagFilterSensitiveLog(item)) }
}), "CreateExplainabilityExportRequestFilterSensitiveLog");
var CreateForecastRequestFilterSensitiveLog = /* @__PURE__ */ __name((obj) => ({
  ...obj,
  ...obj.Tags && { Tags: obj.Tags.map((item) => TagFilterSensitiveLog(item)) }
}), "CreateForecastRequestFilterSensitiveLog");
var CreateForecastExportJobRequestFilterSensitiveLog = /* @__PURE__ */ __name((obj) => ({
  ...obj,
  ...obj.Tags && { Tags: obj.Tags.map((item) => TagFilterSensitiveLog(item)) }
}), "CreateForecastExportJobRequestFilterSensitiveLog");
var CreateMonitorRequestFilterSensitiveLog = /* @__PURE__ */ __name((obj) => ({
  ...obj,
  ...obj.Tags && { Tags: obj.Tags.map((item) => TagFilterSensitiveLog(item)) }
}), "CreateMonitorRequestFilterSensitiveLog");
var CreatePredictorRequestFilterSensitiveLog = /* @__PURE__ */ __name((obj) => ({
  ...obj,
  ...obj.Tags && { Tags: obj.Tags.map((item) => TagFilterSensitiveLog(item)) }
}), "CreatePredictorRequestFilterSensitiveLog");
var CreatePredictorBacktestExportJobRequestFilterSensitiveLog = /* @__PURE__ */ __name((obj) => ({
  ...obj,
  ...obj.Tags && { Tags: obj.Tags.map((item) => TagFilterSensitiveLog(item)) }
}), "CreatePredictorBacktestExportJobRequestFilterSensitiveLog");
var CreateWhatIfAnalysisRequestFilterSensitiveLog = /* @__PURE__ */ __name((obj) => ({
  ...obj,
  ...obj.Tags && { Tags: obj.Tags.map((item) => TagFilterSensitiveLog(item)) }
}), "CreateWhatIfAnalysisRequestFilterSensitiveLog");
var CreateWhatIfForecastRequestFilterSensitiveLog = /* @__PURE__ */ __name((obj) => ({
  ...obj,
  ...obj.Tags && { Tags: obj.Tags.map((item) => TagFilterSensitiveLog(item)) }
}), "CreateWhatIfForecastRequestFilterSensitiveLog");
var CreateWhatIfForecastExportRequestFilterSensitiveLog = /* @__PURE__ */ __name((obj) => ({
  ...obj,
  ...obj.Tags && { Tags: obj.Tags.map((item) => TagFilterSensitiveLog(item)) }
}), "CreateWhatIfForecastExportRequestFilterSensitiveLog");
var ListTagsForResourceResponseFilterSensitiveLog = /* @__PURE__ */ __name((obj) => ({
  ...obj,
  ...obj.Tags && { Tags: obj.Tags.map((item) => TagFilterSensitiveLog(item)) }
}), "ListTagsForResourceResponseFilterSensitiveLog");
var TagResourceRequestFilterSensitiveLog = /* @__PURE__ */ __name((obj) => ({
  ...obj,
  ...obj.Tags && { Tags: obj.Tags.map((item) => TagFilterSensitiveLog(item)) }
}), "TagResourceRequestFilterSensitiveLog");
var UntagResourceRequestFilterSensitiveLog = /* @__PURE__ */ __name((obj) => ({
  ...obj,
  ...obj.TagKeys && { TagKeys: import_smithy_client.SENSITIVE_STRING }
}), "UntagResourceRequestFilterSensitiveLog");

// src/protocols/Aws_json1_1.ts
var import_core2 = require("@aws-sdk/core");


var se_CreateAutoPredictorCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("CreateAutoPredictor");
  let body;
  body = JSON.stringify((0, import_smithy_client._json)(input));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_CreateAutoPredictorCommand");
var se_CreateDatasetCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("CreateDataset");
  let body;
  body = JSON.stringify((0, import_smithy_client._json)(input));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_CreateDatasetCommand");
var se_CreateDatasetGroupCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("CreateDatasetGroup");
  let body;
  body = JSON.stringify((0, import_smithy_client._json)(input));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_CreateDatasetGroupCommand");
var se_CreateDatasetImportJobCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("CreateDatasetImportJob");
  let body;
  body = JSON.stringify((0, import_smithy_client._json)(input));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_CreateDatasetImportJobCommand");
var se_CreateExplainabilityCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("CreateExplainability");
  let body;
  body = JSON.stringify((0, import_smithy_client._json)(input));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_CreateExplainabilityCommand");
var se_CreateExplainabilityExportCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("CreateExplainabilityExport");
  let body;
  body = JSON.stringify((0, import_smithy_client._json)(input));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_CreateExplainabilityExportCommand");
var se_CreateForecastCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("CreateForecast");
  let body;
  body = JSON.stringify((0, import_smithy_client._json)(input));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_CreateForecastCommand");
var se_CreateForecastExportJobCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("CreateForecastExportJob");
  let body;
  body = JSON.stringify((0, import_smithy_client._json)(input));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_CreateForecastExportJobCommand");
var se_CreateMonitorCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("CreateMonitor");
  let body;
  body = JSON.stringify((0, import_smithy_client._json)(input));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_CreateMonitorCommand");
var se_CreatePredictorCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("CreatePredictor");
  let body;
  body = JSON.stringify(se_CreatePredictorRequest(input, context));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_CreatePredictorCommand");
var se_CreatePredictorBacktestExportJobCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("CreatePredictorBacktestExportJob");
  let body;
  body = JSON.stringify((0, import_smithy_client._json)(input));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_CreatePredictorBacktestExportJobCommand");
var se_CreateWhatIfAnalysisCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("CreateWhatIfAnalysis");
  let body;
  body = JSON.stringify((0, import_smithy_client._json)(input));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_CreateWhatIfAnalysisCommand");
var se_CreateWhatIfForecastCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("CreateWhatIfForecast");
  let body;
  body = JSON.stringify(se_CreateWhatIfForecastRequest(input, context));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_CreateWhatIfForecastCommand");
var se_CreateWhatIfForecastExportCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("CreateWhatIfForecastExport");
  let body;
  body = JSON.stringify((0, import_smithy_client._json)(input));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_CreateWhatIfForecastExportCommand");
var se_DeleteDatasetCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("DeleteDataset");
  let body;
  body = JSON.stringify((0, import_smithy_client._json)(input));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_DeleteDatasetCommand");
var se_DeleteDatasetGroupCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("DeleteDatasetGroup");
  let body;
  body = JSON.stringify((0, import_smithy_client._json)(input));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_DeleteDatasetGroupCommand");
var se_DeleteDatasetImportJobCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("DeleteDatasetImportJob");
  let body;
  body = JSON.stringify((0, import_smithy_client._json)(input));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_DeleteDatasetImportJobCommand");
var se_DeleteExplainabilityCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("DeleteExplainability");
  let body;
  body = JSON.stringify((0, import_smithy_client._json)(input));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_DeleteExplainabilityCommand");
var se_DeleteExplainabilityExportCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("DeleteExplainabilityExport");
  let body;
  body = JSON.stringify((0, import_smithy_client._json)(input));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_DeleteExplainabilityExportCommand");
var se_DeleteForecastCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("DeleteForecast");
  let body;
  body = JSON.stringify((0, import_smithy_client._json)(input));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_DeleteForecastCommand");
var se_DeleteForecastExportJobCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("DeleteForecastExportJob");
  let body;
  body = JSON.stringify((0, import_smithy_client._json)(input));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_DeleteForecastExportJobCommand");
var se_DeleteMonitorCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("DeleteMonitor");
  let body;
  body = JSON.stringify((0, import_smithy_client._json)(input));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_DeleteMonitorCommand");
var se_DeletePredictorCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("DeletePredictor");
  let body;
  body = JSON.stringify((0, import_smithy_client._json)(input));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_DeletePredictorCommand");
var se_DeletePredictorBacktestExportJobCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("DeletePredictorBacktestExportJob");
  let body;
  body = JSON.stringify((0, import_smithy_client._json)(input));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_DeletePredictorBacktestExportJobCommand");
var se_DeleteResourceTreeCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("DeleteResourceTree");
  let body;
  body = JSON.stringify((0, import_smithy_client._json)(input));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_DeleteResourceTreeCommand");
var se_DeleteWhatIfAnalysisCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("DeleteWhatIfAnalysis");
  let body;
  body = JSON.stringify((0, import_smithy_client._json)(input));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_DeleteWhatIfAnalysisCommand");
var se_DeleteWhatIfForecastCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("DeleteWhatIfForecast");
  let body;
  body = JSON.stringify((0, import_smithy_client._json)(input));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_DeleteWhatIfForecastCommand");
var se_DeleteWhatIfForecastExportCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("DeleteWhatIfForecastExport");
  let body;
  body = JSON.stringify((0, import_smithy_client._json)(input));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_DeleteWhatIfForecastExportCommand");
var se_DescribeAutoPredictorCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("DescribeAutoPredictor");
  let body;
  body = JSON.stringify((0, import_smithy_client._json)(input));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_DescribeAutoPredictorCommand");
var se_DescribeDatasetCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("DescribeDataset");
  let body;
  body = JSON.stringify((0, import_smithy_client._json)(input));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_DescribeDatasetCommand");
var se_DescribeDatasetGroupCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("DescribeDatasetGroup");
  let body;
  body = JSON.stringify((0, import_smithy_client._json)(input));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_DescribeDatasetGroupCommand");
var se_DescribeDatasetImportJobCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("DescribeDatasetImportJob");
  let body;
  body = JSON.stringify((0, import_smithy_client._json)(input));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_DescribeDatasetImportJobCommand");
var se_DescribeExplainabilityCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("DescribeExplainability");
  let body;
  body = JSON.stringify((0, import_smithy_client._json)(input));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_DescribeExplainabilityCommand");
var se_DescribeExplainabilityExportCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("DescribeExplainabilityExport");
  let body;
  body = JSON.stringify((0, import_smithy_client._json)(input));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_DescribeExplainabilityExportCommand");
var se_DescribeForecastCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("DescribeForecast");
  let body;
  body = JSON.stringify((0, import_smithy_client._json)(input));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_DescribeForecastCommand");
var se_DescribeForecastExportJobCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("DescribeForecastExportJob");
  let body;
  body = JSON.stringify((0, import_smithy_client._json)(input));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_DescribeForecastExportJobCommand");
var se_DescribeMonitorCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("DescribeMonitor");
  let body;
  body = JSON.stringify((0, import_smithy_client._json)(input));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_DescribeMonitorCommand");
var se_DescribePredictorCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("DescribePredictor");
  let body;
  body = JSON.stringify((0, import_smithy_client._json)(input));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_DescribePredictorCommand");
var se_DescribePredictorBacktestExportJobCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("DescribePredictorBacktestExportJob");
  let body;
  body = JSON.stringify((0, import_smithy_client._json)(input));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_DescribePredictorBacktestExportJobCommand");
var se_DescribeWhatIfAnalysisCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("DescribeWhatIfAnalysis");
  let body;
  body = JSON.stringify((0, import_smithy_client._json)(input));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_DescribeWhatIfAnalysisCommand");
var se_DescribeWhatIfForecastCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("DescribeWhatIfForecast");
  let body;
  body = JSON.stringify((0, import_smithy_client._json)(input));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_DescribeWhatIfForecastCommand");
var se_DescribeWhatIfForecastExportCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("DescribeWhatIfForecastExport");
  let body;
  body = JSON.stringify((0, import_smithy_client._json)(input));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_DescribeWhatIfForecastExportCommand");
var se_GetAccuracyMetricsCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("GetAccuracyMetrics");
  let body;
  body = JSON.stringify((0, import_smithy_client._json)(input));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_GetAccuracyMetricsCommand");
var se_ListDatasetGroupsCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("ListDatasetGroups");
  let body;
  body = JSON.stringify((0, import_smithy_client._json)(input));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_ListDatasetGroupsCommand");
var se_ListDatasetImportJobsCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("ListDatasetImportJobs");
  let body;
  body = JSON.stringify((0, import_smithy_client._json)(input));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_ListDatasetImportJobsCommand");
var se_ListDatasetsCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("ListDatasets");
  let body;
  body = JSON.stringify((0, import_smithy_client._json)(input));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_ListDatasetsCommand");
var se_ListExplainabilitiesCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("ListExplainabilities");
  let body;
  body = JSON.stringify((0, import_smithy_client._json)(input));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_ListExplainabilitiesCommand");
var se_ListExplainabilityExportsCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("ListExplainabilityExports");
  let body;
  body = JSON.stringify((0, import_smithy_client._json)(input));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_ListExplainabilityExportsCommand");
var se_ListForecastExportJobsCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("ListForecastExportJobs");
  let body;
  body = JSON.stringify((0, import_smithy_client._json)(input));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_ListForecastExportJobsCommand");
var se_ListForecastsCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("ListForecasts");
  let body;
  body = JSON.stringify((0, import_smithy_client._json)(input));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_ListForecastsCommand");
var se_ListMonitorEvaluationsCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("ListMonitorEvaluations");
  let body;
  body = JSON.stringify((0, import_smithy_client._json)(input));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_ListMonitorEvaluationsCommand");
var se_ListMonitorsCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("ListMonitors");
  let body;
  body = JSON.stringify((0, import_smithy_client._json)(input));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_ListMonitorsCommand");
var se_ListPredictorBacktestExportJobsCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("ListPredictorBacktestExportJobs");
  let body;
  body = JSON.stringify((0, import_smithy_client._json)(input));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_ListPredictorBacktestExportJobsCommand");
var se_ListPredictorsCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("ListPredictors");
  let body;
  body = JSON.stringify((0, import_smithy_client._json)(input));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_ListPredictorsCommand");
var se_ListTagsForResourceCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("ListTagsForResource");
  let body;
  body = JSON.stringify((0, import_smithy_client._json)(input));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_ListTagsForResourceCommand");
var se_ListWhatIfAnalysesCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("ListWhatIfAnalyses");
  let body;
  body = JSON.stringify((0, import_smithy_client._json)(input));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_ListWhatIfAnalysesCommand");
var se_ListWhatIfForecastExportsCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("ListWhatIfForecastExports");
  let body;
  body = JSON.stringify((0, import_smithy_client._json)(input));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_ListWhatIfForecastExportsCommand");
var se_ListWhatIfForecastsCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("ListWhatIfForecasts");
  let body;
  body = JSON.stringify((0, import_smithy_client._json)(input));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_ListWhatIfForecastsCommand");
var se_ResumeResourceCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("ResumeResource");
  let body;
  body = JSON.stringify((0, import_smithy_client._json)(input));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_ResumeResourceCommand");
var se_StopResourceCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("StopResource");
  let body;
  body = JSON.stringify((0, import_smithy_client._json)(input));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_StopResourceCommand");
var se_TagResourceCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("TagResource");
  let body;
  body = JSON.stringify((0, import_smithy_client._json)(input));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_TagResourceCommand");
var se_UntagResourceCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("UntagResource");
  let body;
  body = JSON.stringify((0, import_smithy_client._json)(input));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_UntagResourceCommand");
var se_UpdateDatasetGroupCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("UpdateDatasetGroup");
  let body;
  body = JSON.stringify((0, import_smithy_client._json)(input));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_UpdateDatasetGroupCommand");
var de_CreateAutoPredictorCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const data = await (0, import_core2.parseJsonBody)(output.body, context);
  let contents = {};
  contents = (0, import_smithy_client._json)(data);
  const response = {
    $metadata: deserializeMetadata(output),
    ...contents
  };
  return response;
}, "de_CreateAutoPredictorCommand");
var de_CreateDatasetCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const data = await (0, import_core2.parseJsonBody)(output.body, context);
  let contents = {};
  contents = (0, import_smithy_client._json)(data);
  const response = {
    $metadata: deserializeMetadata(output),
    ...contents
  };
  return response;
}, "de_CreateDatasetCommand");
var de_CreateDatasetGroupCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const data = await (0, import_core2.parseJsonBody)(output.body, context);
  let contents = {};
  contents = (0, import_smithy_client._json)(data);
  const response = {
    $metadata: deserializeMetadata(output),
    ...contents
  };
  return response;
}, "de_CreateDatasetGroupCommand");
var de_CreateDatasetImportJobCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const data = await (0, import_core2.parseJsonBody)(output.body, context);
  let contents = {};
  contents = (0, import_smithy_client._json)(data);
  const response = {
    $metadata: deserializeMetadata(output),
    ...contents
  };
  return response;
}, "de_CreateDatasetImportJobCommand");
var de_CreateExplainabilityCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const data = await (0, import_core2.parseJsonBody)(output.body, context);
  let contents = {};
  contents = (0, import_smithy_client._json)(data);
  const response = {
    $metadata: deserializeMetadata(output),
    ...contents
  };
  return response;
}, "de_CreateExplainabilityCommand");
var de_CreateExplainabilityExportCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const data = await (0, import_core2.parseJsonBody)(output.body, context);
  let contents = {};
  contents = (0, import_smithy_client._json)(data);
  const response = {
    $metadata: deserializeMetadata(output),
    ...contents
  };
  return response;
}, "de_CreateExplainabilityExportCommand");
var de_CreateForecastCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const data = await (0, import_core2.parseJsonBody)(output.body, context);
  let contents = {};
  contents = (0, import_smithy_client._json)(data);
  const response = {
    $metadata: deserializeMetadata(output),
    ...contents
  };
  return response;
}, "de_CreateForecastCommand");
var de_CreateForecastExportJobCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const data = await (0, import_core2.parseJsonBody)(output.body, context);
  let contents = {};
  contents = (0, import_smithy_client._json)(data);
  const response = {
    $metadata: deserializeMetadata(output),
    ...contents
  };
  return response;
}, "de_CreateForecastExportJobCommand");
var de_CreateMonitorCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const data = await (0, import_core2.parseJsonBody)(output.body, context);
  let contents = {};
  contents = (0, import_smithy_client._json)(data);
  const response = {
    $metadata: deserializeMetadata(output),
    ...contents
  };
  return response;
}, "de_CreateMonitorCommand");
var de_CreatePredictorCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const data = await (0, import_core2.parseJsonBody)(output.body, context);
  let contents = {};
  contents = (0, import_smithy_client._json)(data);
  const response = {
    $metadata: deserializeMetadata(output),
    ...contents
  };
  return response;
}, "de_CreatePredictorCommand");
var de_CreatePredictorBacktestExportJobCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const data = await (0, import_core2.parseJsonBody)(output.body, context);
  let contents = {};
  contents = (0, import_smithy_client._json)(data);
  const response = {
    $metadata: deserializeMetadata(output),
    ...contents
  };
  return response;
}, "de_CreatePredictorBacktestExportJobCommand");
var de_CreateWhatIfAnalysisCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const data = await (0, import_core2.parseJsonBody)(output.body, context);
  let contents = {};
  contents = (0, import_smithy_client._json)(data);
  const response = {
    $metadata: deserializeMetadata(output),
    ...contents
  };
  return response;
}, "de_CreateWhatIfAnalysisCommand");
var de_CreateWhatIfForecastCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const data = await (0, import_core2.parseJsonBody)(output.body, context);
  let contents = {};
  contents = (0, import_smithy_client._json)(data);
  const response = {
    $metadata: deserializeMetadata(output),
    ...contents
  };
  return response;
}, "de_CreateWhatIfForecastCommand");
var de_CreateWhatIfForecastExportCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const data = await (0, import_core2.parseJsonBody)(output.body, context);
  let contents = {};
  contents = (0, import_smithy_client._json)(data);
  const response = {
    $metadata: deserializeMetadata(output),
    ...contents
  };
  return response;
}, "de_CreateWhatIfForecastExportCommand");
var de_DeleteDatasetCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  await (0, import_smithy_client.collectBody)(output.body, context);
  const response = {
    $metadata: deserializeMetadata(output)
  };
  return response;
}, "de_DeleteDatasetCommand");
var de_DeleteDatasetGroupCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  await (0, import_smithy_client.collectBody)(output.body, context);
  const response = {
    $metadata: deserializeMetadata(output)
  };
  return response;
}, "de_DeleteDatasetGroupCommand");
var de_DeleteDatasetImportJobCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  await (0, import_smithy_client.collectBody)(output.body, context);
  const response = {
    $metadata: deserializeMetadata(output)
  };
  return response;
}, "de_DeleteDatasetImportJobCommand");
var de_DeleteExplainabilityCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  await (0, import_smithy_client.collectBody)(output.body, context);
  const response = {
    $metadata: deserializeMetadata(output)
  };
  return response;
}, "de_DeleteExplainabilityCommand");
var de_DeleteExplainabilityExportCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  await (0, import_smithy_client.collectBody)(output.body, context);
  const response = {
    $metadata: deserializeMetadata(output)
  };
  return response;
}, "de_DeleteExplainabilityExportCommand");
var de_DeleteForecastCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  await (0, import_smithy_client.collectBody)(output.body, context);
  const response = {
    $metadata: deserializeMetadata(output)
  };
  return response;
}, "de_DeleteForecastCommand");
var de_DeleteForecastExportJobCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  await (0, import_smithy_client.collectBody)(output.body, context);
  const response = {
    $metadata: deserializeMetadata(output)
  };
  return response;
}, "de_DeleteForecastExportJobCommand");
var de_DeleteMonitorCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  await (0, import_smithy_client.collectBody)(output.body, context);
  const response = {
    $metadata: deserializeMetadata(output)
  };
  return response;
}, "de_DeleteMonitorCommand");
var de_DeletePredictorCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  await (0, import_smithy_client.collectBody)(output.body, context);
  const response = {
    $metadata: deserializeMetadata(output)
  };
  return response;
}, "de_DeletePredictorCommand");
var de_DeletePredictorBacktestExportJobCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  await (0, import_smithy_client.collectBody)(output.body, context);
  const response = {
    $metadata: deserializeMetadata(output)
  };
  return response;
}, "de_DeletePredictorBacktestExportJobCommand");
var de_DeleteResourceTreeCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  await (0, import_smithy_client.collectBody)(output.body, context);
  const response = {
    $metadata: deserializeMetadata(output)
  };
  return response;
}, "de_DeleteResourceTreeCommand");
var de_DeleteWhatIfAnalysisCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  await (0, import_smithy_client.collectBody)(output.body, context);
  const response = {
    $metadata: deserializeMetadata(output)
  };
  return response;
}, "de_DeleteWhatIfAnalysisCommand");
var de_DeleteWhatIfForecastCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  await (0, import_smithy_client.collectBody)(output.body, context);
  const response = {
    $metadata: deserializeMetadata(output)
  };
  return response;
}, "de_DeleteWhatIfForecastCommand");
var de_DeleteWhatIfForecastExportCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  await (0, import_smithy_client.collectBody)(output.body, context);
  const response = {
    $metadata: deserializeMetadata(output)
  };
  return response;
}, "de_DeleteWhatIfForecastExportCommand");
var de_DescribeAutoPredictorCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const data = await (0, import_core2.parseJsonBody)(output.body, context);
  let contents = {};
  contents = de_DescribeAutoPredictorResponse(data, context);
  const response = {
    $metadata: deserializeMetadata(output),
    ...contents
  };
  return response;
}, "de_DescribeAutoPredictorCommand");
var de_DescribeDatasetCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const data = await (0, import_core2.parseJsonBody)(output.body, context);
  let contents = {};
  contents = de_DescribeDatasetResponse(data, context);
  const response = {
    $metadata: deserializeMetadata(output),
    ...contents
  };
  return response;
}, "de_DescribeDatasetCommand");
var de_DescribeDatasetGroupCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const data = await (0, import_core2.parseJsonBody)(output.body, context);
  let contents = {};
  contents = de_DescribeDatasetGroupResponse(data, context);
  const response = {
    $metadata: deserializeMetadata(output),
    ...contents
  };
  return response;
}, "de_DescribeDatasetGroupCommand");
var de_DescribeDatasetImportJobCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const data = await (0, import_core2.parseJsonBody)(output.body, context);
  let contents = {};
  contents = de_DescribeDatasetImportJobResponse(data, context);
  const response = {
    $metadata: deserializeMetadata(output),
    ...contents
  };
  return response;
}, "de_DescribeDatasetImportJobCommand");
var de_DescribeExplainabilityCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const data = await (0, import_core2.parseJsonBody)(output.body, context);
  let contents = {};
  contents = de_DescribeExplainabilityResponse(data, context);
  const response = {
    $metadata: deserializeMetadata(output),
    ...contents
  };
  return response;
}, "de_DescribeExplainabilityCommand");
var de_DescribeExplainabilityExportCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const data = await (0, import_core2.parseJsonBody)(output.body, context);
  let contents = {};
  contents = de_DescribeExplainabilityExportResponse(data, context);
  const response = {
    $metadata: deserializeMetadata(output),
    ...contents
  };
  return response;
}, "de_DescribeExplainabilityExportCommand");
var de_DescribeForecastCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const data = await (0, import_core2.parseJsonBody)(output.body, context);
  let contents = {};
  contents = de_DescribeForecastResponse(data, context);
  const response = {
    $metadata: deserializeMetadata(output),
    ...contents
  };
  return response;
}, "de_DescribeForecastCommand");
var de_DescribeForecastExportJobCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const data = await (0, import_core2.parseJsonBody)(output.body, context);
  let contents = {};
  contents = de_DescribeForecastExportJobResponse(data, context);
  const response = {
    $metadata: deserializeMetadata(output),
    ...contents
  };
  return response;
}, "de_DescribeForecastExportJobCommand");
var de_DescribeMonitorCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const data = await (0, import_core2.parseJsonBody)(output.body, context);
  let contents = {};
  contents = de_DescribeMonitorResponse(data, context);
  const response = {
    $metadata: deserializeMetadata(output),
    ...contents
  };
  return response;
}, "de_DescribeMonitorCommand");
var de_DescribePredictorCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const data = await (0, import_core2.parseJsonBody)(output.body, context);
  let contents = {};
  contents = de_DescribePredictorResponse(data, context);
  const response = {
    $metadata: deserializeMetadata(output),
    ...contents
  };
  return response;
}, "de_DescribePredictorCommand");
var de_DescribePredictorBacktestExportJobCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const data = await (0, import_core2.parseJsonBody)(output.body, context);
  let contents = {};
  contents = de_DescribePredictorBacktestExportJobResponse(data, context);
  const response = {
    $metadata: deserializeMetadata(output),
    ...contents
  };
  return response;
}, "de_DescribePredictorBacktestExportJobCommand");
var de_DescribeWhatIfAnalysisCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const data = await (0, import_core2.parseJsonBody)(output.body, context);
  let contents = {};
  contents = de_DescribeWhatIfAnalysisResponse(data, context);
  const response = {
    $metadata: deserializeMetadata(output),
    ...contents
  };
  return response;
}, "de_DescribeWhatIfAnalysisCommand");
var de_DescribeWhatIfForecastCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const data = await (0, import_core2.parseJsonBody)(output.body, context);
  let contents = {};
  contents = de_DescribeWhatIfForecastResponse(data, context);
  const response = {
    $metadata: deserializeMetadata(output),
    ...contents
  };
  return response;
}, "de_DescribeWhatIfForecastCommand");
var de_DescribeWhatIfForecastExportCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const data = await (0, import_core2.parseJsonBody)(output.body, context);
  let contents = {};
  contents = de_DescribeWhatIfForecastExportResponse(data, context);
  const response = {
    $metadata: deserializeMetadata(output),
    ...contents
  };
  return response;
}, "de_DescribeWhatIfForecastExportCommand");
var de_GetAccuracyMetricsCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const data = await (0, import_core2.parseJsonBody)(output.body, context);
  let contents = {};
  contents = de_GetAccuracyMetricsResponse(data, context);
  const response = {
    $metadata: deserializeMetadata(output),
    ...contents
  };
  return response;
}, "de_GetAccuracyMetricsCommand");
var de_ListDatasetGroupsCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const data = await (0, import_core2.parseJsonBody)(output.body, context);
  let contents = {};
  contents = de_ListDatasetGroupsResponse(data, context);
  const response = {
    $metadata: deserializeMetadata(output),
    ...contents
  };
  return response;
}, "de_ListDatasetGroupsCommand");
var de_ListDatasetImportJobsCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const data = await (0, import_core2.parseJsonBody)(output.body, context);
  let contents = {};
  contents = de_ListDatasetImportJobsResponse(data, context);
  const response = {
    $metadata: deserializeMetadata(output),
    ...contents
  };
  return response;
}, "de_ListDatasetImportJobsCommand");
var de_ListDatasetsCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const data = await (0, import_core2.parseJsonBody)(output.body, context);
  let contents = {};
  contents = de_ListDatasetsResponse(data, context);
  const response = {
    $metadata: deserializeMetadata(output),
    ...contents
  };
  return response;
}, "de_ListDatasetsCommand");
var de_ListExplainabilitiesCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const data = await (0, import_core2.parseJsonBody)(output.body, context);
  let contents = {};
  contents = de_ListExplainabilitiesResponse(data, context);
  const response = {
    $metadata: deserializeMetadata(output),
    ...contents
  };
  return response;
}, "de_ListExplainabilitiesCommand");
var de_ListExplainabilityExportsCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const data = await (0, import_core2.parseJsonBody)(output.body, context);
  let contents = {};
  contents = de_ListExplainabilityExportsResponse(data, context);
  const response = {
    $metadata: deserializeMetadata(output),
    ...contents
  };
  return response;
}, "de_ListExplainabilityExportsCommand");
var de_ListForecastExportJobsCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const data = await (0, import_core2.parseJsonBody)(output.body, context);
  let contents = {};
  contents = de_ListForecastExportJobsResponse(data, context);
  const response = {
    $metadata: deserializeMetadata(output),
    ...contents
  };
  return response;
}, "de_ListForecastExportJobsCommand");
var de_ListForecastsCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const data = await (0, import_core2.parseJsonBody)(output.body, context);
  let contents = {};
  contents = de_ListForecastsResponse(data, context);
  const response = {
    $metadata: deserializeMetadata(output),
    ...contents
  };
  return response;
}, "de_ListForecastsCommand");
var de_ListMonitorEvaluationsCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const data = await (0, import_core2.parseJsonBody)(output.body, context);
  let contents = {};
  contents = de_ListMonitorEvaluationsResponse(data, context);
  const response = {
    $metadata: deserializeMetadata(output),
    ...contents
  };
  return response;
}, "de_ListMonitorEvaluationsCommand");
var de_ListMonitorsCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const data = await (0, import_core2.parseJsonBody)(output.body, context);
  let contents = {};
  contents = de_ListMonitorsResponse(data, context);
  const response = {
    $metadata: deserializeMetadata(output),
    ...contents
  };
  return response;
}, "de_ListMonitorsCommand");
var de_ListPredictorBacktestExportJobsCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const data = await (0, import_core2.parseJsonBody)(output.body, context);
  let contents = {};
  contents = de_ListPredictorBacktestExportJobsResponse(data, context);
  const response = {
    $metadata: deserializeMetadata(output),
    ...contents
  };
  return response;
}, "de_ListPredictorBacktestExportJobsCommand");
var de_ListPredictorsCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const data = await (0, import_core2.parseJsonBody)(output.body, context);
  let contents = {};
  contents = de_ListPredictorsResponse(data, context);
  const response = {
    $metadata: deserializeMetadata(output),
    ...contents
  };
  return response;
}, "de_ListPredictorsCommand");
var de_ListTagsForResourceCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const data = await (0, import_core2.parseJsonBody)(output.body, context);
  let contents = {};
  contents = (0, import_smithy_client._json)(data);
  const response = {
    $metadata: deserializeMetadata(output),
    ...contents
  };
  return response;
}, "de_ListTagsForResourceCommand");
var de_ListWhatIfAnalysesCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const data = await (0, import_core2.parseJsonBody)(output.body, context);
  let contents = {};
  contents = de_ListWhatIfAnalysesResponse(data, context);
  const response = {
    $metadata: deserializeMetadata(output),
    ...contents
  };
  return response;
}, "de_ListWhatIfAnalysesCommand");
var de_ListWhatIfForecastExportsCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const data = await (0, import_core2.parseJsonBody)(output.body, context);
  let contents = {};
  contents = de_ListWhatIfForecastExportsResponse(data, context);
  const response = {
    $metadata: deserializeMetadata(output),
    ...contents
  };
  return response;
}, "de_ListWhatIfForecastExportsCommand");
var de_ListWhatIfForecastsCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const data = await (0, import_core2.parseJsonBody)(output.body, context);
  let contents = {};
  contents = de_ListWhatIfForecastsResponse(data, context);
  const response = {
    $metadata: deserializeMetadata(output),
    ...contents
  };
  return response;
}, "de_ListWhatIfForecastsCommand");
var de_ResumeResourceCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  await (0, import_smithy_client.collectBody)(output.body, context);
  const response = {
    $metadata: deserializeMetadata(output)
  };
  return response;
}, "de_ResumeResourceCommand");
var de_StopResourceCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  await (0, import_smithy_client.collectBody)(output.body, context);
  const response = {
    $metadata: deserializeMetadata(output)
  };
  return response;
}, "de_StopResourceCommand");
var de_TagResourceCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const data = await (0, import_core2.parseJsonBody)(output.body, context);
  let contents = {};
  contents = (0, import_smithy_client._json)(data);
  const response = {
    $metadata: deserializeMetadata(output),
    ...contents
  };
  return response;
}, "de_TagResourceCommand");
var de_UntagResourceCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const data = await (0, import_core2.parseJsonBody)(output.body, context);
  let contents = {};
  contents = (0, import_smithy_client._json)(data);
  const response = {
    $metadata: deserializeMetadata(output),
    ...contents
  };
  return response;
}, "de_UntagResourceCommand");
var de_UpdateDatasetGroupCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const data = await (0, import_core2.parseJsonBody)(output.body, context);
  let contents = {};
  contents = (0, import_smithy_client._json)(data);
  const response = {
    $metadata: deserializeMetadata(output),
    ...contents
  };
  return response;
}, "de_UpdateDatasetGroupCommand");
var de_CommandError = /* @__PURE__ */ __name(async (output, context) => {
  const parsedOutput = {
    ...output,
    body: await (0, import_core2.parseJsonErrorBody)(output.body, context)
  };
  const errorCode = (0, import_core2.loadRestJsonErrorCode)(output, parsedOutput.body);
  switch (errorCode) {
    case "InvalidInputException":
    case "com.amazonaws.forecast#InvalidInputException":
      throw await de_InvalidInputExceptionRes(parsedOutput, context);
    case "LimitExceededException":
    case "com.amazonaws.forecast#LimitExceededException":
      throw await de_LimitExceededExceptionRes(parsedOutput, context);
    case "ResourceAlreadyExistsException":
    case "com.amazonaws.forecast#ResourceAlreadyExistsException":
      throw await de_ResourceAlreadyExistsExceptionRes(parsedOutput, context);
    case "ResourceInUseException":
    case "com.amazonaws.forecast#ResourceInUseException":
      throw await de_ResourceInUseExceptionRes(parsedOutput, context);
    case "ResourceNotFoundException":
    case "com.amazonaws.forecast#ResourceNotFoundException":
      throw await de_ResourceNotFoundExceptionRes(parsedOutput, context);
    case "InvalidNextTokenException":
    case "com.amazonaws.forecast#InvalidNextTokenException":
      throw await de_InvalidNextTokenExceptionRes(parsedOutput, context);
    default:
      const parsedBody = parsedOutput.body;
      return throwDefaultError({
        output,
        parsedBody,
        errorCode
      });
  }
}, "de_CommandError");
var de_InvalidInputExceptionRes = /* @__PURE__ */ __name(async (parsedOutput, context) => {
  const body = parsedOutput.body;
  const deserialized = (0, import_smithy_client._json)(body);
  const exception = new InvalidInputException({
    $metadata: deserializeMetadata(parsedOutput),
    ...deserialized
  });
  return (0, import_smithy_client.decorateServiceException)(exception, body);
}, "de_InvalidInputExceptionRes");
var de_InvalidNextTokenExceptionRes = /* @__PURE__ */ __name(async (parsedOutput, context) => {
  const body = parsedOutput.body;
  const deserialized = (0, import_smithy_client._json)(body);
  const exception = new InvalidNextTokenException({
    $metadata: deserializeMetadata(parsedOutput),
    ...deserialized
  });
  return (0, import_smithy_client.decorateServiceException)(exception, body);
}, "de_InvalidNextTokenExceptionRes");
var de_LimitExceededExceptionRes = /* @__PURE__ */ __name(async (parsedOutput, context) => {
  const body = parsedOutput.body;
  const deserialized = (0, import_smithy_client._json)(body);
  const exception = new LimitExceededException({
    $metadata: deserializeMetadata(parsedOutput),
    ...deserialized
  });
  return (0, import_smithy_client.decorateServiceException)(exception, body);
}, "de_LimitExceededExceptionRes");
var de_ResourceAlreadyExistsExceptionRes = /* @__PURE__ */ __name(async (parsedOutput, context) => {
  const body = parsedOutput.body;
  const deserialized = (0, import_smithy_client._json)(body);
  const exception = new ResourceAlreadyExistsException({
    $metadata: deserializeMetadata(parsedOutput),
    ...deserialized
  });
  return (0, import_smithy_client.decorateServiceException)(exception, body);
}, "de_ResourceAlreadyExistsExceptionRes");
var de_ResourceInUseExceptionRes = /* @__PURE__ */ __name(async (parsedOutput, context) => {
  const body = parsedOutput.body;
  const deserialized = (0, import_smithy_client._json)(body);
  const exception = new ResourceInUseException({
    $metadata: deserializeMetadata(parsedOutput),
    ...deserialized
  });
  return (0, import_smithy_client.decorateServiceException)(exception, body);
}, "de_ResourceInUseExceptionRes");
var de_ResourceNotFoundExceptionRes = /* @__PURE__ */ __name(async (parsedOutput, context) => {
  const body = parsedOutput.body;
  const deserialized = (0, import_smithy_client._json)(body);
  const exception = new ResourceNotFoundException({
    $metadata: deserializeMetadata(parsedOutput),
    ...deserialized
  });
  return (0, import_smithy_client.decorateServiceException)(exception, body);
}, "de_ResourceNotFoundExceptionRes");
var se_Action = /* @__PURE__ */ __name((input, context) => {
  return (0, import_smithy_client.take)(input, {
    AttributeName: [],
    Operation: [],
    Value: import_smithy_client.serializeFloat
  });
}, "se_Action");
var se_ContinuousParameterRange = /* @__PURE__ */ __name((input, context) => {
  return (0, import_smithy_client.take)(input, {
    MaxValue: import_smithy_client.serializeFloat,
    MinValue: import_smithy_client.serializeFloat,
    Name: [],
    ScalingType: []
  });
}, "se_ContinuousParameterRange");
var se_ContinuousParameterRanges = /* @__PURE__ */ __name((input, context) => {
  return input.filter((e) => e != null).map((entry) => {
    return se_ContinuousParameterRange(entry, context);
  });
}, "se_ContinuousParameterRanges");
var se_CreatePredictorRequest = /* @__PURE__ */ __name((input, context) => {
  return (0, import_smithy_client.take)(input, {
    AlgorithmArn: [],
    AutoMLOverrideStrategy: [],
    EncryptionConfig: import_smithy_client._json,
    EvaluationParameters: import_smithy_client._json,
    FeaturizationConfig: import_smithy_client._json,
    ForecastHorizon: [],
    ForecastTypes: import_smithy_client._json,
    HPOConfig: (_) => se_HyperParameterTuningJobConfig(_, context),
    InputDataConfig: import_smithy_client._json,
    OptimizationMetric: [],
    PerformAutoML: [],
    PerformHPO: [],
    PredictorName: [],
    Tags: import_smithy_client._json,
    TrainingParameters: import_smithy_client._json
  });
}, "se_CreatePredictorRequest");
var se_CreateWhatIfForecastRequest = /* @__PURE__ */ __name((input, context) => {
  return (0, import_smithy_client.take)(input, {
    Tags: import_smithy_client._json,
    TimeSeriesReplacementsDataSource: import_smithy_client._json,
    TimeSeriesTransformations: (_) => se_TimeSeriesTransformations(_, context),
    WhatIfAnalysisArn: [],
    WhatIfForecastName: []
  });
}, "se_CreateWhatIfForecastRequest");
var se_HyperParameterTuningJobConfig = /* @__PURE__ */ __name((input, context) => {
  return (0, import_smithy_client.take)(input, {
    ParameterRanges: (_) => se_ParameterRanges(_, context)
  });
}, "se_HyperParameterTuningJobConfig");
var se_ParameterRanges = /* @__PURE__ */ __name((input, context) => {
  return (0, import_smithy_client.take)(input, {
    CategoricalParameterRanges: import_smithy_client._json,
    ContinuousParameterRanges: (_) => se_ContinuousParameterRanges(_, context),
    IntegerParameterRanges: import_smithy_client._json
  });
}, "se_ParameterRanges");
var se_TimeSeriesTransformation = /* @__PURE__ */ __name((input, context) => {
  return (0, import_smithy_client.take)(input, {
    Action: (_) => se_Action(_, context),
    TimeSeriesConditions: import_smithy_client._json
  });
}, "se_TimeSeriesTransformation");
var se_TimeSeriesTransformations = /* @__PURE__ */ __name((input, context) => {
  return input.filter((e) => e != null).map((entry) => {
    return se_TimeSeriesTransformation(entry, context);
  });
}, "se_TimeSeriesTransformations");
var de_Action = /* @__PURE__ */ __name((output, context) => {
  return (0, import_smithy_client.take)(output, {
    AttributeName: import_smithy_client.expectString,
    Operation: import_smithy_client.expectString,
    Value: import_smithy_client.limitedParseDouble
  });
}, "de_Action");
var de_Baseline = /* @__PURE__ */ __name((output, context) => {
  return (0, import_smithy_client.take)(output, {
    PredictorBaseline: (_) => de_PredictorBaseline(_, context)
  });
}, "de_Baseline");
var de_BaselineMetric = /* @__PURE__ */ __name((output, context) => {
  return (0, import_smithy_client.take)(output, {
    Name: import_smithy_client.expectString,
    Value: import_smithy_client.limitedParseDouble
  });
}, "de_BaselineMetric");
var de_BaselineMetrics = /* @__PURE__ */ __name((output, context) => {
  const retVal = (output || []).filter((e) => e != null).map((entry) => {
    return de_BaselineMetric(entry, context);
  });
  return retVal;
}, "de_BaselineMetrics");
var de_ContinuousParameterRange = /* @__PURE__ */ __name((output, context) => {
  return (0, import_smithy_client.take)(output, {
    MaxValue: import_smithy_client.limitedParseDouble,
    MinValue: import_smithy_client.limitedParseDouble,
    Name: import_smithy_client.expectString,
    ScalingType: import_smithy_client.expectString
  });
}, "de_ContinuousParameterRange");
var de_ContinuousParameterRanges = /* @__PURE__ */ __name((output, context) => {
  const retVal = (output || []).filter((e) => e != null).map((entry) => {
    return de_ContinuousParameterRange(entry, context);
  });
  return retVal;
}, "de_ContinuousParameterRanges");
var de_DatasetGroups = /* @__PURE__ */ __name((output, context) => {
  const retVal = (output || []).filter((e) => e != null).map((entry) => {
    return de_DatasetGroupSummary(entry, context);
  });
  return retVal;
}, "de_DatasetGroups");
var de_DatasetGroupSummary = /* @__PURE__ */ __name((output, context) => {
  return (0, import_smithy_client.take)(output, {
    CreationTime: (_) => (0, import_smithy_client.expectNonNull)((0, import_smithy_client.parseEpochTimestamp)((0, import_smithy_client.expectNumber)(_))),
    DatasetGroupArn: import_smithy_client.expectString,
    DatasetGroupName: import_smithy_client.expectString,
    LastModificationTime: (_) => (0, import_smithy_client.expectNonNull)((0, import_smithy_client.parseEpochTimestamp)((0, import_smithy_client.expectNumber)(_)))
  });
}, "de_DatasetGroupSummary");
var de_DatasetImportJobs = /* @__PURE__ */ __name((output, context) => {
  const retVal = (output || []).filter((e) => e != null).map((entry) => {
    return de_DatasetImportJobSummary(entry, context);
  });
  return retVal;
}, "de_DatasetImportJobs");
var de_DatasetImportJobSummary = /* @__PURE__ */ __name((output, context) => {
  return (0, import_smithy_client.take)(output, {
    CreationTime: (_) => (0, import_smithy_client.expectNonNull)((0, import_smithy_client.parseEpochTimestamp)((0, import_smithy_client.expectNumber)(_))),
    DataSource: import_smithy_client._json,
    DatasetImportJobArn: import_smithy_client.expectString,
    DatasetImportJobName: import_smithy_client.expectString,
    ImportMode: import_smithy_client.expectString,
    LastModificationTime: (_) => (0, import_smithy_client.expectNonNull)((0, import_smithy_client.parseEpochTimestamp)((0, import_smithy_client.expectNumber)(_))),
    Message: import_smithy_client.expectString,
    Status: import_smithy_client.expectString
  });
}, "de_DatasetImportJobSummary");
var de_Datasets = /* @__PURE__ */ __name((output, context) => {
  const retVal = (output || []).filter((e) => e != null).map((entry) => {
    return de_DatasetSummary(entry, context);
  });
  return retVal;
}, "de_Datasets");
var de_DatasetSummary = /* @__PURE__ */ __name((output, context) => {
  return (0, import_smithy_client.take)(output, {
    CreationTime: (_) => (0, import_smithy_client.expectNonNull)((0, import_smithy_client.parseEpochTimestamp)((0, import_smithy_client.expectNumber)(_))),
    DatasetArn: import_smithy_client.expectString,
    DatasetName: import_smithy_client.expectString,
    DatasetType: import_smithy_client.expectString,
    Domain: import_smithy_client.expectString,
    LastModificationTime: (_) => (0, import_smithy_client.expectNonNull)((0, import_smithy_client.parseEpochTimestamp)((0, import_smithy_client.expectNumber)(_)))
  });
}, "de_DatasetSummary");
var de_DescribeAutoPredictorResponse = /* @__PURE__ */ __name((output, context) => {
  return (0, import_smithy_client.take)(output, {
    CreationTime: (_) => (0, import_smithy_client.expectNonNull)((0, import_smithy_client.parseEpochTimestamp)((0, import_smithy_client.expectNumber)(_))),
    DataConfig: import_smithy_client._json,
    DatasetImportJobArns: import_smithy_client._json,
    EncryptionConfig: import_smithy_client._json,
    EstimatedTimeRemainingInMinutes: import_smithy_client.expectLong,
    ExplainabilityInfo: import_smithy_client._json,
    ForecastDimensions: import_smithy_client._json,
    ForecastFrequency: import_smithy_client.expectString,
    ForecastHorizon: import_smithy_client.expectInt32,
    ForecastTypes: import_smithy_client._json,
    LastModificationTime: (_) => (0, import_smithy_client.expectNonNull)((0, import_smithy_client.parseEpochTimestamp)((0, import_smithy_client.expectNumber)(_))),
    Message: import_smithy_client.expectString,
    MonitorInfo: import_smithy_client._json,
    OptimizationMetric: import_smithy_client.expectString,
    PredictorArn: import_smithy_client.expectString,
    PredictorName: import_smithy_client.expectString,
    ReferencePredictorSummary: import_smithy_client._json,
    Status: import_smithy_client.expectString,
    TimeAlignmentBoundary: import_smithy_client._json
  });
}, "de_DescribeAutoPredictorResponse");
var de_DescribeDatasetGroupResponse = /* @__PURE__ */ __name((output, context) => {
  return (0, import_smithy_client.take)(output, {
    CreationTime: (_) => (0, import_smithy_client.expectNonNull)((0, import_smithy_client.parseEpochTimestamp)((0, import_smithy_client.expectNumber)(_))),
    DatasetArns: import_smithy_client._json,
    DatasetGroupArn: import_smithy_client.expectString,
    DatasetGroupName: import_smithy_client.expectString,
    Domain: import_smithy_client.expectString,
    LastModificationTime: (_) => (0, import_smithy_client.expectNonNull)((0, import_smithy_client.parseEpochTimestamp)((0, import_smithy_client.expectNumber)(_))),
    Status: import_smithy_client.expectString
  });
}, "de_DescribeDatasetGroupResponse");
var de_DescribeDatasetImportJobResponse = /* @__PURE__ */ __name((output, context) => {
  return (0, import_smithy_client.take)(output, {
    CreationTime: (_) => (0, import_smithy_client.expectNonNull)((0, import_smithy_client.parseEpochTimestamp)((0, import_smithy_client.expectNumber)(_))),
    DataSize: import_smithy_client.limitedParseDouble,
    DataSource: import_smithy_client._json,
    DatasetArn: import_smithy_client.expectString,
    DatasetImportJobArn: import_smithy_client.expectString,
    DatasetImportJobName: import_smithy_client.expectString,
    EstimatedTimeRemainingInMinutes: import_smithy_client.expectLong,
    FieldStatistics: (_) => de_FieldStatistics(_, context),
    Format: import_smithy_client.expectString,
    GeolocationFormat: import_smithy_client.expectString,
    ImportMode: import_smithy_client.expectString,
    LastModificationTime: (_) => (0, import_smithy_client.expectNonNull)((0, import_smithy_client.parseEpochTimestamp)((0, import_smithy_client.expectNumber)(_))),
    Message: import_smithy_client.expectString,
    Status: import_smithy_client.expectString,
    TimeZone: import_smithy_client.expectString,
    TimestampFormat: import_smithy_client.expectString,
    UseGeolocationForTimeZone: import_smithy_client.expectBoolean
  });
}, "de_DescribeDatasetImportJobResponse");
var de_DescribeDatasetResponse = /* @__PURE__ */ __name((output, context) => {
  return (0, import_smithy_client.take)(output, {
    CreationTime: (_) => (0, import_smithy_client.expectNonNull)((0, import_smithy_client.parseEpochTimestamp)((0, import_smithy_client.expectNumber)(_))),
    DataFrequency: import_smithy_client.expectString,
    DatasetArn: import_smithy_client.expectString,
    DatasetName: import_smithy_client.expectString,
    DatasetType: import_smithy_client.expectString,
    Domain: import_smithy_client.expectString,
    EncryptionConfig: import_smithy_client._json,
    LastModificationTime: (_) => (0, import_smithy_client.expectNonNull)((0, import_smithy_client.parseEpochTimestamp)((0, import_smithy_client.expectNumber)(_))),
    Schema: import_smithy_client._json,
    Status: import_smithy_client.expectString
  });
}, "de_DescribeDatasetResponse");
var de_DescribeExplainabilityExportResponse = /* @__PURE__ */ __name((output, context) => {
  return (0, import_smithy_client.take)(output, {
    CreationTime: (_) => (0, import_smithy_client.expectNonNull)((0, import_smithy_client.parseEpochTimestamp)((0, import_smithy_client.expectNumber)(_))),
    Destination: import_smithy_client._json,
    ExplainabilityArn: import_smithy_client.expectString,
    ExplainabilityExportArn: import_smithy_client.expectString,
    ExplainabilityExportName: import_smithy_client.expectString,
    Format: import_smithy_client.expectString,
    LastModificationTime: (_) => (0, import_smithy_client.expectNonNull)((0, import_smithy_client.parseEpochTimestamp)((0, import_smithy_client.expectNumber)(_))),
    Message: import_smithy_client.expectString,
    Status: import_smithy_client.expectString
  });
}, "de_DescribeExplainabilityExportResponse");
var de_DescribeExplainabilityResponse = /* @__PURE__ */ __name((output, context) => {
  return (0, import_smithy_client.take)(output, {
    CreationTime: (_) => (0, import_smithy_client.expectNonNull)((0, import_smithy_client.parseEpochTimestamp)((0, import_smithy_client.expectNumber)(_))),
    DataSource: import_smithy_client._json,
    EnableVisualization: import_smithy_client.expectBoolean,
    EndDateTime: import_smithy_client.expectString,
    EstimatedTimeRemainingInMinutes: import_smithy_client.expectLong,
    ExplainabilityArn: import_smithy_client.expectString,
    ExplainabilityConfig: import_smithy_client._json,
    ExplainabilityName: import_smithy_client.expectString,
    LastModificationTime: (_) => (0, import_smithy_client.expectNonNull)((0, import_smithy_client.parseEpochTimestamp)((0, import_smithy_client.expectNumber)(_))),
    Message: import_smithy_client.expectString,
    ResourceArn: import_smithy_client.expectString,
    Schema: import_smithy_client._json,
    StartDateTime: import_smithy_client.expectString,
    Status: import_smithy_client.expectString
  });
}, "de_DescribeExplainabilityResponse");
var de_DescribeForecastExportJobResponse = /* @__PURE__ */ __name((output, context) => {
  return (0, import_smithy_client.take)(output, {
    CreationTime: (_) => (0, import_smithy_client.expectNonNull)((0, import_smithy_client.parseEpochTimestamp)((0, import_smithy_client.expectNumber)(_))),
    Destination: import_smithy_client._json,
    ForecastArn: import_smithy_client.expectString,
    ForecastExportJobArn: import_smithy_client.expectString,
    ForecastExportJobName: import_smithy_client.expectString,
    Format: import_smithy_client.expectString,
    LastModificationTime: (_) => (0, import_smithy_client.expectNonNull)((0, import_smithy_client.parseEpochTimestamp)((0, import_smithy_client.expectNumber)(_))),
    Message: import_smithy_client.expectString,
    Status: import_smithy_client.expectString
  });
}, "de_DescribeForecastExportJobResponse");
var de_DescribeForecastResponse = /* @__PURE__ */ __name((output, context) => {
  return (0, import_smithy_client.take)(output, {
    CreationTime: (_) => (0, import_smithy_client.expectNonNull)((0, import_smithy_client.parseEpochTimestamp)((0, import_smithy_client.expectNumber)(_))),
    DatasetGroupArn: import_smithy_client.expectString,
    EstimatedTimeRemainingInMinutes: import_smithy_client.expectLong,
    ForecastArn: import_smithy_client.expectString,
    ForecastName: import_smithy_client.expectString,
    ForecastTypes: import_smithy_client._json,
    LastModificationTime: (_) => (0, import_smithy_client.expectNonNull)((0, import_smithy_client.parseEpochTimestamp)((0, import_smithy_client.expectNumber)(_))),
    Message: import_smithy_client.expectString,
    PredictorArn: import_smithy_client.expectString,
    Status: import_smithy_client.expectString,
    TimeSeriesSelector: import_smithy_client._json
  });
}, "de_DescribeForecastResponse");
var de_DescribeMonitorResponse = /* @__PURE__ */ __name((output, context) => {
  return (0, import_smithy_client.take)(output, {
    Baseline: (_) => de_Baseline(_, context),
    CreationTime: (_) => (0, import_smithy_client.expectNonNull)((0, import_smithy_client.parseEpochTimestamp)((0, import_smithy_client.expectNumber)(_))),
    EstimatedEvaluationTimeRemainingInMinutes: import_smithy_client.expectLong,
    LastEvaluationState: import_smithy_client.expectString,
    LastEvaluationTime: (_) => (0, import_smithy_client.expectNonNull)((0, import_smithy_client.parseEpochTimestamp)((0, import_smithy_client.expectNumber)(_))),
    LastModificationTime: (_) => (0, import_smithy_client.expectNonNull)((0, import_smithy_client.parseEpochTimestamp)((0, import_smithy_client.expectNumber)(_))),
    Message: import_smithy_client.expectString,
    MonitorArn: import_smithy_client.expectString,
    MonitorName: import_smithy_client.expectString,
    ResourceArn: import_smithy_client.expectString,
    Status: import_smithy_client.expectString
  });
}, "de_DescribeMonitorResponse");
var de_DescribePredictorBacktestExportJobResponse = /* @__PURE__ */ __name((output, context) => {
  return (0, import_smithy_client.take)(output, {
    CreationTime: (_) => (0, import_smithy_client.expectNonNull)((0, import_smithy_client.parseEpochTimestamp)((0, import_smithy_client.expectNumber)(_))),
    Destination: import_smithy_client._json,
    Format: import_smithy_client.expectString,
    LastModificationTime: (_) => (0, import_smithy_client.expectNonNull)((0, import_smithy_client.parseEpochTimestamp)((0, import_smithy_client.expectNumber)(_))),
    Message: import_smithy_client.expectString,
    PredictorArn: import_smithy_client.expectString,
    PredictorBacktestExportJobArn: import_smithy_client.expectString,
    PredictorBacktestExportJobName: import_smithy_client.expectString,
    Status: import_smithy_client.expectString
  });
}, "de_DescribePredictorBacktestExportJobResponse");
var de_DescribePredictorResponse = /* @__PURE__ */ __name((output, context) => {
  return (0, import_smithy_client.take)(output, {
    AlgorithmArn: import_smithy_client.expectString,
    AutoMLAlgorithmArns: import_smithy_client._json,
    AutoMLOverrideStrategy: import_smithy_client.expectString,
    CreationTime: (_) => (0, import_smithy_client.expectNonNull)((0, import_smithy_client.parseEpochTimestamp)((0, import_smithy_client.expectNumber)(_))),
    DatasetImportJobArns: import_smithy_client._json,
    EncryptionConfig: import_smithy_client._json,
    EstimatedTimeRemainingInMinutes: import_smithy_client.expectLong,
    EvaluationParameters: import_smithy_client._json,
    FeaturizationConfig: import_smithy_client._json,
    ForecastHorizon: import_smithy_client.expectInt32,
    ForecastTypes: import_smithy_client._json,
    HPOConfig: (_) => de_HyperParameterTuningJobConfig(_, context),
    InputDataConfig: import_smithy_client._json,
    IsAutoPredictor: import_smithy_client.expectBoolean,
    LastModificationTime: (_) => (0, import_smithy_client.expectNonNull)((0, import_smithy_client.parseEpochTimestamp)((0, import_smithy_client.expectNumber)(_))),
    Message: import_smithy_client.expectString,
    OptimizationMetric: import_smithy_client.expectString,
    PerformAutoML: import_smithy_client.expectBoolean,
    PerformHPO: import_smithy_client.expectBoolean,
    PredictorArn: import_smithy_client.expectString,
    PredictorExecutionDetails: (_) => de_PredictorExecutionDetails(_, context),
    PredictorName: import_smithy_client.expectString,
    Status: import_smithy_client.expectString,
    TrainingParameters: import_smithy_client._json
  });
}, "de_DescribePredictorResponse");
var de_DescribeWhatIfAnalysisResponse = /* @__PURE__ */ __name((output, context) => {
  return (0, import_smithy_client.take)(output, {
    CreationTime: (_) => (0, import_smithy_client.expectNonNull)((0, import_smithy_client.parseEpochTimestamp)((0, import_smithy_client.expectNumber)(_))),
    EstimatedTimeRemainingInMinutes: import_smithy_client.expectLong,
    ForecastArn: import_smithy_client.expectString,
    LastModificationTime: (_) => (0, import_smithy_client.expectNonNull)((0, import_smithy_client.parseEpochTimestamp)((0, import_smithy_client.expectNumber)(_))),
    Message: import_smithy_client.expectString,
    Status: import_smithy_client.expectString,
    TimeSeriesSelector: import_smithy_client._json,
    WhatIfAnalysisArn: import_smithy_client.expectString,
    WhatIfAnalysisName: import_smithy_client.expectString
  });
}, "de_DescribeWhatIfAnalysisResponse");
var de_DescribeWhatIfForecastExportResponse = /* @__PURE__ */ __name((output, context) => {
  return (0, import_smithy_client.take)(output, {
    CreationTime: (_) => (0, import_smithy_client.expectNonNull)((0, import_smithy_client.parseEpochTimestamp)((0, import_smithy_client.expectNumber)(_))),
    Destination: import_smithy_client._json,
    EstimatedTimeRemainingInMinutes: import_smithy_client.expectLong,
    Format: import_smithy_client.expectString,
    LastModificationTime: (_) => (0, import_smithy_client.expectNonNull)((0, import_smithy_client.parseEpochTimestamp)((0, import_smithy_client.expectNumber)(_))),
    Message: import_smithy_client.expectString,
    Status: import_smithy_client.expectString,
    WhatIfForecastArns: import_smithy_client._json,
    WhatIfForecastExportArn: import_smithy_client.expectString,
    WhatIfForecastExportName: import_smithy_client.expectString
  });
}, "de_DescribeWhatIfForecastExportResponse");
var de_DescribeWhatIfForecastResponse = /* @__PURE__ */ __name((output, context) => {
  return (0, import_smithy_client.take)(output, {
    CreationTime: (_) => (0, import_smithy_client.expectNonNull)((0, import_smithy_client.parseEpochTimestamp)((0, import_smithy_client.expectNumber)(_))),
    EstimatedTimeRemainingInMinutes: import_smithy_client.expectLong,
    ForecastTypes: import_smithy_client._json,
    LastModificationTime: (_) => (0, import_smithy_client.expectNonNull)((0, import_smithy_client.parseEpochTimestamp)((0, import_smithy_client.expectNumber)(_))),
    Message: import_smithy_client.expectString,
    Status: import_smithy_client.expectString,
    TimeSeriesReplacementsDataSource: import_smithy_client._json,
    TimeSeriesTransformations: (_) => de_TimeSeriesTransformations(_, context),
    WhatIfAnalysisArn: import_smithy_client.expectString,
    WhatIfForecastArn: import_smithy_client.expectString,
    WhatIfForecastName: import_smithy_client.expectString
  });
}, "de_DescribeWhatIfForecastResponse");
var de_ErrorMetric = /* @__PURE__ */ __name((output, context) => {
  return (0, import_smithy_client.take)(output, {
    ForecastType: import_smithy_client.expectString,
    MAPE: import_smithy_client.limitedParseDouble,
    MASE: import_smithy_client.limitedParseDouble,
    RMSE: import_smithy_client.limitedParseDouble,
    WAPE: import_smithy_client.limitedParseDouble
  });
}, "de_ErrorMetric");
var de_ErrorMetrics = /* @__PURE__ */ __name((output, context) => {
  const retVal = (output || []).filter((e) => e != null).map((entry) => {
    return de_ErrorMetric(entry, context);
  });
  return retVal;
}, "de_ErrorMetrics");
var de_EvaluationResult = /* @__PURE__ */ __name((output, context) => {
  return (0, import_smithy_client.take)(output, {
    AlgorithmArn: import_smithy_client.expectString,
    TestWindows: (_) => de_TestWindows(_, context)
  });
}, "de_EvaluationResult");
var de_Explainabilities = /* @__PURE__ */ __name((output, context) => {
  const retVal = (output || []).filter((e) => e != null).map((entry) => {
    return de_ExplainabilitySummary(entry, context);
  });
  return retVal;
}, "de_Explainabilities");
var de_ExplainabilityExports = /* @__PURE__ */ __name((output, context) => {
  const retVal = (output || []).filter((e) => e != null).map((entry) => {
    return de_ExplainabilityExportSummary(entry, context);
  });
  return retVal;
}, "de_ExplainabilityExports");
var de_ExplainabilityExportSummary = /* @__PURE__ */ __name((output, context) => {
  return (0, import_smithy_client.take)(output, {
    CreationTime: (_) => (0, import_smithy_client.expectNonNull)((0, import_smithy_client.parseEpochTimestamp)((0, import_smithy_client.expectNumber)(_))),
    Destination: import_smithy_client._json,
    ExplainabilityExportArn: import_smithy_client.expectString,
    ExplainabilityExportName: import_smithy_client.expectString,
    LastModificationTime: (_) => (0, import_smithy_client.expectNonNull)((0, import_smithy_client.parseEpochTimestamp)((0, import_smithy_client.expectNumber)(_))),
    Message: import_smithy_client.expectString,
    Status: import_smithy_client.expectString
  });
}, "de_ExplainabilityExportSummary");
var de_ExplainabilitySummary = /* @__PURE__ */ __name((output, context) => {
  return (0, import_smithy_client.take)(output, {
    CreationTime: (_) => (0, import_smithy_client.expectNonNull)((0, import_smithy_client.parseEpochTimestamp)((0, import_smithy_client.expectNumber)(_))),
    ExplainabilityArn: import_smithy_client.expectString,
    ExplainabilityConfig: import_smithy_client._json,
    ExplainabilityName: import_smithy_client.expectString,
    LastModificationTime: (_) => (0, import_smithy_client.expectNonNull)((0, import_smithy_client.parseEpochTimestamp)((0, import_smithy_client.expectNumber)(_))),
    Message: import_smithy_client.expectString,
    ResourceArn: import_smithy_client.expectString,
    Status: import_smithy_client.expectString
  });
}, "de_ExplainabilitySummary");
var de_FieldStatistics = /* @__PURE__ */ __name((output, context) => {
  return Object.entries(output).reduce((acc, [key, value]) => {
    if (value === null) {
      return acc;
    }
    acc[key] = de_Statistics(value, context);
    return acc;
  }, {});
}, "de_FieldStatistics");
var de_ForecastExportJobs = /* @__PURE__ */ __name((output, context) => {
  const retVal = (output || []).filter((e) => e != null).map((entry) => {
    return de_ForecastExportJobSummary(entry, context);
  });
  return retVal;
}, "de_ForecastExportJobs");
var de_ForecastExportJobSummary = /* @__PURE__ */ __name((output, context) => {
  return (0, import_smithy_client.take)(output, {
    CreationTime: (_) => (0, import_smithy_client.expectNonNull)((0, import_smithy_client.parseEpochTimestamp)((0, import_smithy_client.expectNumber)(_))),
    Destination: import_smithy_client._json,
    ForecastExportJobArn: import_smithy_client.expectString,
    ForecastExportJobName: import_smithy_client.expectString,
    LastModificationTime: (_) => (0, import_smithy_client.expectNonNull)((0, import_smithy_client.parseEpochTimestamp)((0, import_smithy_client.expectNumber)(_))),
    Message: import_smithy_client.expectString,
    Status: import_smithy_client.expectString
  });
}, "de_ForecastExportJobSummary");
var de_Forecasts = /* @__PURE__ */ __name((output, context) => {
  const retVal = (output || []).filter((e) => e != null).map((entry) => {
    return de_ForecastSummary(entry, context);
  });
  return retVal;
}, "de_Forecasts");
var de_ForecastSummary = /* @__PURE__ */ __name((output, context) => {
  return (0, import_smithy_client.take)(output, {
    CreatedUsingAutoPredictor: import_smithy_client.expectBoolean,
    CreationTime: (_) => (0, import_smithy_client.expectNonNull)((0, import_smithy_client.parseEpochTimestamp)((0, import_smithy_client.expectNumber)(_))),
    DatasetGroupArn: import_smithy_client.expectString,
    ForecastArn: import_smithy_client.expectString,
    ForecastName: import_smithy_client.expectString,
    LastModificationTime: (_) => (0, import_smithy_client.expectNonNull)((0, import_smithy_client.parseEpochTimestamp)((0, import_smithy_client.expectNumber)(_))),
    Message: import_smithy_client.expectString,
    PredictorArn: import_smithy_client.expectString,
    Status: import_smithy_client.expectString
  });
}, "de_ForecastSummary");
var de_GetAccuracyMetricsResponse = /* @__PURE__ */ __name((output, context) => {
  return (0, import_smithy_client.take)(output, {
    AutoMLOverrideStrategy: import_smithy_client.expectString,
    IsAutoPredictor: import_smithy_client.expectBoolean,
    OptimizationMetric: import_smithy_client.expectString,
    PredictorEvaluationResults: (_) => de_PredictorEvaluationResults(_, context)
  });
}, "de_GetAccuracyMetricsResponse");
var de_HyperParameterTuningJobConfig = /* @__PURE__ */ __name((output, context) => {
  return (0, import_smithy_client.take)(output, {
    ParameterRanges: (_) => de_ParameterRanges(_, context)
  });
}, "de_HyperParameterTuningJobConfig");
var de_ListDatasetGroupsResponse = /* @__PURE__ */ __name((output, context) => {
  return (0, import_smithy_client.take)(output, {
    DatasetGroups: (_) => de_DatasetGroups(_, context),
    NextToken: import_smithy_client.expectString
  });
}, "de_ListDatasetGroupsResponse");
var de_ListDatasetImportJobsResponse = /* @__PURE__ */ __name((output, context) => {
  return (0, import_smithy_client.take)(output, {
    DatasetImportJobs: (_) => de_DatasetImportJobs(_, context),
    NextToken: import_smithy_client.expectString
  });
}, "de_ListDatasetImportJobsResponse");
var de_ListDatasetsResponse = /* @__PURE__ */ __name((output, context) => {
  return (0, import_smithy_client.take)(output, {
    Datasets: (_) => de_Datasets(_, context),
    NextToken: import_smithy_client.expectString
  });
}, "de_ListDatasetsResponse");
var de_ListExplainabilitiesResponse = /* @__PURE__ */ __name((output, context) => {
  return (0, import_smithy_client.take)(output, {
    Explainabilities: (_) => de_Explainabilities(_, context),
    NextToken: import_smithy_client.expectString
  });
}, "de_ListExplainabilitiesResponse");
var de_ListExplainabilityExportsResponse = /* @__PURE__ */ __name((output, context) => {
  return (0, import_smithy_client.take)(output, {
    ExplainabilityExports: (_) => de_ExplainabilityExports(_, context),
    NextToken: import_smithy_client.expectString
  });
}, "de_ListExplainabilityExportsResponse");
var de_ListForecastExportJobsResponse = /* @__PURE__ */ __name((output, context) => {
  return (0, import_smithy_client.take)(output, {
    ForecastExportJobs: (_) => de_ForecastExportJobs(_, context),
    NextToken: import_smithy_client.expectString
  });
}, "de_ListForecastExportJobsResponse");
var de_ListForecastsResponse = /* @__PURE__ */ __name((output, context) => {
  return (0, import_smithy_client.take)(output, {
    Forecasts: (_) => de_Forecasts(_, context),
    NextToken: import_smithy_client.expectString
  });
}, "de_ListForecastsResponse");
var de_ListMonitorEvaluationsResponse = /* @__PURE__ */ __name((output, context) => {
  return (0, import_smithy_client.take)(output, {
    NextToken: import_smithy_client.expectString,
    PredictorMonitorEvaluations: (_) => de_PredictorMonitorEvaluations(_, context)
  });
}, "de_ListMonitorEvaluationsResponse");
var de_ListMonitorsResponse = /* @__PURE__ */ __name((output, context) => {
  return (0, import_smithy_client.take)(output, {
    Monitors: (_) => de_Monitors(_, context),
    NextToken: import_smithy_client.expectString
  });
}, "de_ListMonitorsResponse");
var de_ListPredictorBacktestExportJobsResponse = /* @__PURE__ */ __name((output, context) => {
  return (0, import_smithy_client.take)(output, {
    NextToken: import_smithy_client.expectString,
    PredictorBacktestExportJobs: (_) => de_PredictorBacktestExportJobs(_, context)
  });
}, "de_ListPredictorBacktestExportJobsResponse");
var de_ListPredictorsResponse = /* @__PURE__ */ __name((output, context) => {
  return (0, import_smithy_client.take)(output, {
    NextToken: import_smithy_client.expectString,
    Predictors: (_) => de_Predictors(_, context)
  });
}, "de_ListPredictorsResponse");
var de_ListWhatIfAnalysesResponse = /* @__PURE__ */ __name((output, context) => {
  return (0, import_smithy_client.take)(output, {
    NextToken: import_smithy_client.expectString,
    WhatIfAnalyses: (_) => de_WhatIfAnalyses(_, context)
  });
}, "de_ListWhatIfAnalysesResponse");
var de_ListWhatIfForecastExportsResponse = /* @__PURE__ */ __name((output, context) => {
  return (0, import_smithy_client.take)(output, {
    NextToken: import_smithy_client.expectString,
    WhatIfForecastExports: (_) => de_WhatIfForecastExports(_, context)
  });
}, "de_ListWhatIfForecastExportsResponse");
var de_ListWhatIfForecastsResponse = /* @__PURE__ */ __name((output, context) => {
  return (0, import_smithy_client.take)(output, {
    NextToken: import_smithy_client.expectString,
    WhatIfForecasts: (_) => de_WhatIfForecasts(_, context)
  });
}, "de_ListWhatIfForecastsResponse");
var de_MetricResult = /* @__PURE__ */ __name((output, context) => {
  return (0, import_smithy_client.take)(output, {
    MetricName: import_smithy_client.expectString,
    MetricValue: import_smithy_client.limitedParseDouble
  });
}, "de_MetricResult");
var de_MetricResults = /* @__PURE__ */ __name((output, context) => {
  const retVal = (output || []).filter((e) => e != null).map((entry) => {
    return de_MetricResult(entry, context);
  });
  return retVal;
}, "de_MetricResults");
var de_Metrics = /* @__PURE__ */ __name((output, context) => {
  return (0, import_smithy_client.take)(output, {
    AverageWeightedQuantileLoss: import_smithy_client.limitedParseDouble,
    ErrorMetrics: (_) => de_ErrorMetrics(_, context),
    RMSE: import_smithy_client.limitedParseDouble,
    WeightedQuantileLosses: (_) => de_WeightedQuantileLosses(_, context)
  });
}, "de_Metrics");
var de_Monitors = /* @__PURE__ */ __name((output, context) => {
  const retVal = (output || []).filter((e) => e != null).map((entry) => {
    return de_MonitorSummary(entry, context);
  });
  return retVal;
}, "de_Monitors");
var de_MonitorSummary = /* @__PURE__ */ __name((output, context) => {
  return (0, import_smithy_client.take)(output, {
    CreationTime: (_) => (0, import_smithy_client.expectNonNull)((0, import_smithy_client.parseEpochTimestamp)((0, import_smithy_client.expectNumber)(_))),
    LastModificationTime: (_) => (0, import_smithy_client.expectNonNull)((0, import_smithy_client.parseEpochTimestamp)((0, import_smithy_client.expectNumber)(_))),
    MonitorArn: import_smithy_client.expectString,
    MonitorName: import_smithy_client.expectString,
    ResourceArn: import_smithy_client.expectString,
    Status: import_smithy_client.expectString
  });
}, "de_MonitorSummary");
var de_ParameterRanges = /* @__PURE__ */ __name((output, context) => {
  return (0, import_smithy_client.take)(output, {
    CategoricalParameterRanges: import_smithy_client._json,
    ContinuousParameterRanges: (_) => de_ContinuousParameterRanges(_, context),
    IntegerParameterRanges: import_smithy_client._json
  });
}, "de_ParameterRanges");
var de_PredictorBacktestExportJobs = /* @__PURE__ */ __name((output, context) => {
  const retVal = (output || []).filter((e) => e != null).map((entry) => {
    return de_PredictorBacktestExportJobSummary(entry, context);
  });
  return retVal;
}, "de_PredictorBacktestExportJobs");
var de_PredictorBacktestExportJobSummary = /* @__PURE__ */ __name((output, context) => {
  return (0, import_smithy_client.take)(output, {
    CreationTime: (_) => (0, import_smithy_client.expectNonNull)((0, import_smithy_client.parseEpochTimestamp)((0, import_smithy_client.expectNumber)(_))),
    Destination: import_smithy_client._json,
    LastModificationTime: (_) => (0, import_smithy_client.expectNonNull)((0, import_smithy_client.parseEpochTimestamp)((0, import_smithy_client.expectNumber)(_))),
    Message: import_smithy_client.expectString,
    PredictorBacktestExportJobArn: import_smithy_client.expectString,
    PredictorBacktestExportJobName: import_smithy_client.expectString,
    Status: import_smithy_client.expectString
  });
}, "de_PredictorBacktestExportJobSummary");
var de_PredictorBaseline = /* @__PURE__ */ __name((output, context) => {
  return (0, import_smithy_client.take)(output, {
    BaselineMetrics: (_) => de_BaselineMetrics(_, context)
  });
}, "de_PredictorBaseline");
var de_PredictorEvaluationResults = /* @__PURE__ */ __name((output, context) => {
  const retVal = (output || []).filter((e) => e != null).map((entry) => {
    return de_EvaluationResult(entry, context);
  });
  return retVal;
}, "de_PredictorEvaluationResults");
var de_PredictorEvent = /* @__PURE__ */ __name((output, context) => {
  return (0, import_smithy_client.take)(output, {
    Datetime: (_) => (0, import_smithy_client.expectNonNull)((0, import_smithy_client.parseEpochTimestamp)((0, import_smithy_client.expectNumber)(_))),
    Detail: import_smithy_client.expectString
  });
}, "de_PredictorEvent");
var de_PredictorExecution = /* @__PURE__ */ __name((output, context) => {
  return (0, import_smithy_client.take)(output, {
    AlgorithmArn: import_smithy_client.expectString,
    TestWindows: (_) => de_TestWindowDetails(_, context)
  });
}, "de_PredictorExecution");
var de_PredictorExecutionDetails = /* @__PURE__ */ __name((output, context) => {
  return (0, import_smithy_client.take)(output, {
    PredictorExecutions: (_) => de_PredictorExecutions(_, context)
  });
}, "de_PredictorExecutionDetails");
var de_PredictorExecutions = /* @__PURE__ */ __name((output, context) => {
  const retVal = (output || []).filter((e) => e != null).map((entry) => {
    return de_PredictorExecution(entry, context);
  });
  return retVal;
}, "de_PredictorExecutions");
var de_PredictorMonitorEvaluation = /* @__PURE__ */ __name((output, context) => {
  return (0, import_smithy_client.take)(output, {
    EvaluationState: import_smithy_client.expectString,
    EvaluationTime: (_) => (0, import_smithy_client.expectNonNull)((0, import_smithy_client.parseEpochTimestamp)((0, import_smithy_client.expectNumber)(_))),
    Message: import_smithy_client.expectString,
    MetricResults: (_) => de_MetricResults(_, context),
    MonitorArn: import_smithy_client.expectString,
    MonitorDataSource: import_smithy_client._json,
    NumItemsEvaluated: import_smithy_client.expectLong,
    PredictorEvent: (_) => de_PredictorEvent(_, context),
    ResourceArn: import_smithy_client.expectString,
    WindowEndDatetime: (_) => (0, import_smithy_client.expectNonNull)((0, import_smithy_client.parseEpochTimestamp)((0, import_smithy_client.expectNumber)(_))),
    WindowStartDatetime: (_) => (0, import_smithy_client.expectNonNull)((0, import_smithy_client.parseEpochTimestamp)((0, import_smithy_client.expectNumber)(_)))
  });
}, "de_PredictorMonitorEvaluation");
var de_PredictorMonitorEvaluations = /* @__PURE__ */ __name((output, context) => {
  const retVal = (output || []).filter((e) => e != null).map((entry) => {
    return de_PredictorMonitorEvaluation(entry, context);
  });
  return retVal;
}, "de_PredictorMonitorEvaluations");
var de_Predictors = /* @__PURE__ */ __name((output, context) => {
  const retVal = (output || []).filter((e) => e != null).map((entry) => {
    return de_PredictorSummary(entry, context);
  });
  return retVal;
}, "de_Predictors");
var de_PredictorSummary = /* @__PURE__ */ __name((output, context) => {
  return (0, import_smithy_client.take)(output, {
    CreationTime: (_) => (0, import_smithy_client.expectNonNull)((0, import_smithy_client.parseEpochTimestamp)((0, import_smithy_client.expectNumber)(_))),
    DatasetGroupArn: import_smithy_client.expectString,
    IsAutoPredictor: import_smithy_client.expectBoolean,
    LastModificationTime: (_) => (0, import_smithy_client.expectNonNull)((0, import_smithy_client.parseEpochTimestamp)((0, import_smithy_client.expectNumber)(_))),
    Message: import_smithy_client.expectString,
    PredictorArn: import_smithy_client.expectString,
    PredictorName: import_smithy_client.expectString,
    ReferencePredictorSummary: import_smithy_client._json,
    Status: import_smithy_client.expectString
  });
}, "de_PredictorSummary");
var de_Statistics = /* @__PURE__ */ __name((output, context) => {
  return (0, import_smithy_client.take)(output, {
    Avg: import_smithy_client.limitedParseDouble,
    Count: import_smithy_client.expectInt32,
    CountDistinct: import_smithy_client.expectInt32,
    CountDistinctLong: import_smithy_client.expectLong,
    CountLong: import_smithy_client.expectLong,
    CountNan: import_smithy_client.expectInt32,
    CountNanLong: import_smithy_client.expectLong,
    CountNull: import_smithy_client.expectInt32,
    CountNullLong: import_smithy_client.expectLong,
    Max: import_smithy_client.expectString,
    Min: import_smithy_client.expectString,
    Stddev: import_smithy_client.limitedParseDouble
  });
}, "de_Statistics");
var de_TestWindowDetails = /* @__PURE__ */ __name((output, context) => {
  const retVal = (output || []).filter((e) => e != null).map((entry) => {
    return de_TestWindowSummary(entry, context);
  });
  return retVal;
}, "de_TestWindowDetails");
var de_TestWindows = /* @__PURE__ */ __name((output, context) => {
  const retVal = (output || []).filter((e) => e != null).map((entry) => {
    return de_WindowSummary(entry, context);
  });
  return retVal;
}, "de_TestWindows");
var de_TestWindowSummary = /* @__PURE__ */ __name((output, context) => {
  return (0, import_smithy_client.take)(output, {
    Message: import_smithy_client.expectString,
    Status: import_smithy_client.expectString,
    TestWindowEnd: (_) => (0, import_smithy_client.expectNonNull)((0, import_smithy_client.parseEpochTimestamp)((0, import_smithy_client.expectNumber)(_))),
    TestWindowStart: (_) => (0, import_smithy_client.expectNonNull)((0, import_smithy_client.parseEpochTimestamp)((0, import_smithy_client.expectNumber)(_)))
  });
}, "de_TestWindowSummary");
var de_TimeSeriesTransformation = /* @__PURE__ */ __name((output, context) => {
  return (0, import_smithy_client.take)(output, {
    Action: (_) => de_Action(_, context),
    TimeSeriesConditions: import_smithy_client._json
  });
}, "de_TimeSeriesTransformation");
var de_TimeSeriesTransformations = /* @__PURE__ */ __name((output, context) => {
  const retVal = (output || []).filter((e) => e != null).map((entry) => {
    return de_TimeSeriesTransformation(entry, context);
  });
  return retVal;
}, "de_TimeSeriesTransformations");
var de_WeightedQuantileLoss = /* @__PURE__ */ __name((output, context) => {
  return (0, import_smithy_client.take)(output, {
    LossValue: import_smithy_client.limitedParseDouble,
    Quantile: import_smithy_client.limitedParseDouble
  });
}, "de_WeightedQuantileLoss");
var de_WeightedQuantileLosses = /* @__PURE__ */ __name((output, context) => {
  const retVal = (output || []).filter((e) => e != null).map((entry) => {
    return de_WeightedQuantileLoss(entry, context);
  });
  return retVal;
}, "de_WeightedQuantileLosses");
var de_WhatIfAnalyses = /* @__PURE__ */ __name((output, context) => {
  const retVal = (output || []).filter((e) => e != null).map((entry) => {
    return de_WhatIfAnalysisSummary(entry, context);
  });
  return retVal;
}, "de_WhatIfAnalyses");
var de_WhatIfAnalysisSummary = /* @__PURE__ */ __name((output, context) => {
  return (0, import_smithy_client.take)(output, {
    CreationTime: (_) => (0, import_smithy_client.expectNonNull)((0, import_smithy_client.parseEpochTimestamp)((0, import_smithy_client.expectNumber)(_))),
    ForecastArn: import_smithy_client.expectString,
    LastModificationTime: (_) => (0, import_smithy_client.expectNonNull)((0, import_smithy_client.parseEpochTimestamp)((0, import_smithy_client.expectNumber)(_))),
    Message: import_smithy_client.expectString,
    Status: import_smithy_client.expectString,
    WhatIfAnalysisArn: import_smithy_client.expectString,
    WhatIfAnalysisName: import_smithy_client.expectString
  });
}, "de_WhatIfAnalysisSummary");
var de_WhatIfForecastExports = /* @__PURE__ */ __name((output, context) => {
  const retVal = (output || []).filter((e) => e != null).map((entry) => {
    return de_WhatIfForecastExportSummary(entry, context);
  });
  return retVal;
}, "de_WhatIfForecastExports");
var de_WhatIfForecastExportSummary = /* @__PURE__ */ __name((output, context) => {
  return (0, import_smithy_client.take)(output, {
    CreationTime: (_) => (0, import_smithy_client.expectNonNull)((0, import_smithy_client.parseEpochTimestamp)((0, import_smithy_client.expectNumber)(_))),
    Destination: import_smithy_client._json,
    LastModificationTime: (_) => (0, import_smithy_client.expectNonNull)((0, import_smithy_client.parseEpochTimestamp)((0, import_smithy_client.expectNumber)(_))),
    Message: import_smithy_client.expectString,
    Status: import_smithy_client.expectString,
    WhatIfForecastArns: import_smithy_client._json,
    WhatIfForecastExportArn: import_smithy_client.expectString,
    WhatIfForecastExportName: import_smithy_client.expectString
  });
}, "de_WhatIfForecastExportSummary");
var de_WhatIfForecasts = /* @__PURE__ */ __name((output, context) => {
  const retVal = (output || []).filter((e) => e != null).map((entry) => {
    return de_WhatIfForecastSummary(entry, context);
  });
  return retVal;
}, "de_WhatIfForecasts");
var de_WhatIfForecastSummary = /* @__PURE__ */ __name((output, context) => {
  return (0, import_smithy_client.take)(output, {
    CreationTime: (_) => (0, import_smithy_client.expectNonNull)((0, import_smithy_client.parseEpochTimestamp)((0, import_smithy_client.expectNumber)(_))),
    LastModificationTime: (_) => (0, import_smithy_client.expectNonNull)((0, import_smithy_client.parseEpochTimestamp)((0, import_smithy_client.expectNumber)(_))),
    Message: import_smithy_client.expectString,
    Status: import_smithy_client.expectString,
    WhatIfAnalysisArn: import_smithy_client.expectString,
    WhatIfForecastArn: import_smithy_client.expectString,
    WhatIfForecastName: import_smithy_client.expectString
  });
}, "de_WhatIfForecastSummary");
var de_WindowSummary = /* @__PURE__ */ __name((output, context) => {
  return (0, import_smithy_client.take)(output, {
    EvaluationType: import_smithy_client.expectString,
    ItemCount: import_smithy_client.expectInt32,
    Metrics: (_) => de_Metrics(_, context),
    TestWindowEnd: (_) => (0, import_smithy_client.expectNonNull)((0, import_smithy_client.parseEpochTimestamp)((0, import_smithy_client.expectNumber)(_))),
    TestWindowStart: (_) => (0, import_smithy_client.expectNonNull)((0, import_smithy_client.parseEpochTimestamp)((0, import_smithy_client.expectNumber)(_)))
  });
}, "de_WindowSummary");
var deserializeMetadata = /* @__PURE__ */ __name((output) => ({
  httpStatusCode: output.statusCode,
  requestId: output.headers["x-amzn-requestid"] ?? output.headers["x-amzn-request-id"] ?? output.headers["x-amz-request-id"],
  extendedRequestId: output.headers["x-amz-id-2"],
  cfId: output.headers["x-amz-cf-id"]
}), "deserializeMetadata");
var throwDefaultError = (0, import_smithy_client.withBaseException)(ForecastServiceException);
var buildHttpRpcRequest = /* @__PURE__ */ __name(async (context, headers, path, resolvedHostname, body) => {
  const { hostname, protocol = "https", port, path: basePath } = await context.endpoint();
  const contents = {
    protocol,
    hostname,
    port,
    method: "POST",
    path: basePath.endsWith("/") ? basePath.slice(0, -1) + path : basePath + path,
    headers
  };
  if (resolvedHostname !== void 0) {
    contents.hostname = resolvedHostname;
  }
  if (body !== void 0) {
    contents.body = body;
  }
  return new import_protocol_http.HttpRequest(contents);
}, "buildHttpRpcRequest");
function sharedHeaders(operation) {
  return {
    "content-type": "application/x-amz-json-1.1",
    "x-amz-target": `AmazonForecast.${operation}`
  };
}
__name(sharedHeaders, "sharedHeaders");

// src/commands/CreateAutoPredictorCommand.ts
var _CreateAutoPredictorCommand = class _CreateAutoPredictorCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AmazonForecast", "CreateAutoPredictor", {}).n("ForecastClient", "CreateAutoPredictorCommand").f(CreateAutoPredictorRequestFilterSensitiveLog, void 0).ser(se_CreateAutoPredictorCommand).de(de_CreateAutoPredictorCommand).build() {
};
__name(_CreateAutoPredictorCommand, "CreateAutoPredictorCommand");
var CreateAutoPredictorCommand = _CreateAutoPredictorCommand;

// src/commands/CreateDatasetCommand.ts




var _CreateDatasetCommand = class _CreateDatasetCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AmazonForecast", "CreateDataset", {}).n("ForecastClient", "CreateDatasetCommand").f(CreateDatasetRequestFilterSensitiveLog, void 0).ser(se_CreateDatasetCommand).de(de_CreateDatasetCommand).build() {
};
__name(_CreateDatasetCommand, "CreateDatasetCommand");
var CreateDatasetCommand = _CreateDatasetCommand;

// src/commands/CreateDatasetGroupCommand.ts




var _CreateDatasetGroupCommand = class _CreateDatasetGroupCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AmazonForecast", "CreateDatasetGroup", {}).n("ForecastClient", "CreateDatasetGroupCommand").f(CreateDatasetGroupRequestFilterSensitiveLog, void 0).ser(se_CreateDatasetGroupCommand).de(de_CreateDatasetGroupCommand).build() {
};
__name(_CreateDatasetGroupCommand, "CreateDatasetGroupCommand");
var CreateDatasetGroupCommand = _CreateDatasetGroupCommand;

// src/commands/CreateDatasetImportJobCommand.ts




var _CreateDatasetImportJobCommand = class _CreateDatasetImportJobCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AmazonForecast", "CreateDatasetImportJob", {}).n("ForecastClient", "CreateDatasetImportJobCommand").f(CreateDatasetImportJobRequestFilterSensitiveLog, void 0).ser(se_CreateDatasetImportJobCommand).de(de_CreateDatasetImportJobCommand).build() {
};
__name(_CreateDatasetImportJobCommand, "CreateDatasetImportJobCommand");
var CreateDatasetImportJobCommand = _CreateDatasetImportJobCommand;

// src/commands/CreateExplainabilityCommand.ts




var _CreateExplainabilityCommand = class _CreateExplainabilityCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AmazonForecast", "CreateExplainability", {}).n("ForecastClient", "CreateExplainabilityCommand").f(CreateExplainabilityRequestFilterSensitiveLog, void 0).ser(se_CreateExplainabilityCommand).de(de_CreateExplainabilityCommand).build() {
};
__name(_CreateExplainabilityCommand, "CreateExplainabilityCommand");
var CreateExplainabilityCommand = _CreateExplainabilityCommand;

// src/commands/CreateExplainabilityExportCommand.ts




var _CreateExplainabilityExportCommand = class _CreateExplainabilityExportCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AmazonForecast", "CreateExplainabilityExport", {}).n("ForecastClient", "CreateExplainabilityExportCommand").f(CreateExplainabilityExportRequestFilterSensitiveLog, void 0).ser(se_CreateExplainabilityExportCommand).de(de_CreateExplainabilityExportCommand).build() {
};
__name(_CreateExplainabilityExportCommand, "CreateExplainabilityExportCommand");
var CreateExplainabilityExportCommand = _CreateExplainabilityExportCommand;

// src/commands/CreateForecastCommand.ts




var _CreateForecastCommand = class _CreateForecastCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AmazonForecast", "CreateForecast", {}).n("ForecastClient", "CreateForecastCommand").f(CreateForecastRequestFilterSensitiveLog, void 0).ser(se_CreateForecastCommand).de(de_CreateForecastCommand).build() {
};
__name(_CreateForecastCommand, "CreateForecastCommand");
var CreateForecastCommand = _CreateForecastCommand;

// src/commands/CreateForecastExportJobCommand.ts




var _CreateForecastExportJobCommand = class _CreateForecastExportJobCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AmazonForecast", "CreateForecastExportJob", {}).n("ForecastClient", "CreateForecastExportJobCommand").f(CreateForecastExportJobRequestFilterSensitiveLog, void 0).ser(se_CreateForecastExportJobCommand).de(de_CreateForecastExportJobCommand).build() {
};
__name(_CreateForecastExportJobCommand, "CreateForecastExportJobCommand");
var CreateForecastExportJobCommand = _CreateForecastExportJobCommand;

// src/commands/CreateMonitorCommand.ts




var _CreateMonitorCommand = class _CreateMonitorCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AmazonForecast", "CreateMonitor", {}).n("ForecastClient", "CreateMonitorCommand").f(CreateMonitorRequestFilterSensitiveLog, void 0).ser(se_CreateMonitorCommand).de(de_CreateMonitorCommand).build() {
};
__name(_CreateMonitorCommand, "CreateMonitorCommand");
var CreateMonitorCommand = _CreateMonitorCommand;

// src/commands/CreatePredictorBacktestExportJobCommand.ts




var _CreatePredictorBacktestExportJobCommand = class _CreatePredictorBacktestExportJobCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AmazonForecast", "CreatePredictorBacktestExportJob", {}).n("ForecastClient", "CreatePredictorBacktestExportJobCommand").f(CreatePredictorBacktestExportJobRequestFilterSensitiveLog, void 0).ser(se_CreatePredictorBacktestExportJobCommand).de(de_CreatePredictorBacktestExportJobCommand).build() {
};
__name(_CreatePredictorBacktestExportJobCommand, "CreatePredictorBacktestExportJobCommand");
var CreatePredictorBacktestExportJobCommand = _CreatePredictorBacktestExportJobCommand;

// src/commands/CreatePredictorCommand.ts




var _CreatePredictorCommand = class _CreatePredictorCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AmazonForecast", "CreatePredictor", {}).n("ForecastClient", "CreatePredictorCommand").f(CreatePredictorRequestFilterSensitiveLog, void 0).ser(se_CreatePredictorCommand).de(de_CreatePredictorCommand).build() {
};
__name(_CreatePredictorCommand, "CreatePredictorCommand");
var CreatePredictorCommand = _CreatePredictorCommand;

// src/commands/CreateWhatIfAnalysisCommand.ts




var _CreateWhatIfAnalysisCommand = class _CreateWhatIfAnalysisCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AmazonForecast", "CreateWhatIfAnalysis", {}).n("ForecastClient", "CreateWhatIfAnalysisCommand").f(CreateWhatIfAnalysisRequestFilterSensitiveLog, void 0).ser(se_CreateWhatIfAnalysisCommand).de(de_CreateWhatIfAnalysisCommand).build() {
};
__name(_CreateWhatIfAnalysisCommand, "CreateWhatIfAnalysisCommand");
var CreateWhatIfAnalysisCommand = _CreateWhatIfAnalysisCommand;

// src/commands/CreateWhatIfForecastCommand.ts




var _CreateWhatIfForecastCommand = class _CreateWhatIfForecastCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AmazonForecast", "CreateWhatIfForecast", {}).n("ForecastClient", "CreateWhatIfForecastCommand").f(CreateWhatIfForecastRequestFilterSensitiveLog, void 0).ser(se_CreateWhatIfForecastCommand).de(de_CreateWhatIfForecastCommand).build() {
};
__name(_CreateWhatIfForecastCommand, "CreateWhatIfForecastCommand");
var CreateWhatIfForecastCommand = _CreateWhatIfForecastCommand;

// src/commands/CreateWhatIfForecastExportCommand.ts




var _CreateWhatIfForecastExportCommand = class _CreateWhatIfForecastExportCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AmazonForecast", "CreateWhatIfForecastExport", {}).n("ForecastClient", "CreateWhatIfForecastExportCommand").f(CreateWhatIfForecastExportRequestFilterSensitiveLog, void 0).ser(se_CreateWhatIfForecastExportCommand).de(de_CreateWhatIfForecastExportCommand).build() {
};
__name(_CreateWhatIfForecastExportCommand, "CreateWhatIfForecastExportCommand");
var CreateWhatIfForecastExportCommand = _CreateWhatIfForecastExportCommand;

// src/commands/DeleteDatasetCommand.ts




var _DeleteDatasetCommand = class _DeleteDatasetCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AmazonForecast", "DeleteDataset", {}).n("ForecastClient", "DeleteDatasetCommand").f(void 0, void 0).ser(se_DeleteDatasetCommand).de(de_DeleteDatasetCommand).build() {
};
__name(_DeleteDatasetCommand, "DeleteDatasetCommand");
var DeleteDatasetCommand = _DeleteDatasetCommand;

// src/commands/DeleteDatasetGroupCommand.ts




var _DeleteDatasetGroupCommand = class _DeleteDatasetGroupCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AmazonForecast", "DeleteDatasetGroup", {}).n("ForecastClient", "DeleteDatasetGroupCommand").f(void 0, void 0).ser(se_DeleteDatasetGroupCommand).de(de_DeleteDatasetGroupCommand).build() {
};
__name(_DeleteDatasetGroupCommand, "DeleteDatasetGroupCommand");
var DeleteDatasetGroupCommand = _DeleteDatasetGroupCommand;

// src/commands/DeleteDatasetImportJobCommand.ts




var _DeleteDatasetImportJobCommand = class _DeleteDatasetImportJobCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AmazonForecast", "DeleteDatasetImportJob", {}).n("ForecastClient", "DeleteDatasetImportJobCommand").f(void 0, void 0).ser(se_DeleteDatasetImportJobCommand).de(de_DeleteDatasetImportJobCommand).build() {
};
__name(_DeleteDatasetImportJobCommand, "DeleteDatasetImportJobCommand");
var DeleteDatasetImportJobCommand = _DeleteDatasetImportJobCommand;

// src/commands/DeleteExplainabilityCommand.ts




var _DeleteExplainabilityCommand = class _DeleteExplainabilityCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AmazonForecast", "DeleteExplainability", {}).n("ForecastClient", "DeleteExplainabilityCommand").f(void 0, void 0).ser(se_DeleteExplainabilityCommand).de(de_DeleteExplainabilityCommand).build() {
};
__name(_DeleteExplainabilityCommand, "DeleteExplainabilityCommand");
var DeleteExplainabilityCommand = _DeleteExplainabilityCommand;

// src/commands/DeleteExplainabilityExportCommand.ts




var _DeleteExplainabilityExportCommand = class _DeleteExplainabilityExportCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AmazonForecast", "DeleteExplainabilityExport", {}).n("ForecastClient", "DeleteExplainabilityExportCommand").f(void 0, void 0).ser(se_DeleteExplainabilityExportCommand).de(de_DeleteExplainabilityExportCommand).build() {
};
__name(_DeleteExplainabilityExportCommand, "DeleteExplainabilityExportCommand");
var DeleteExplainabilityExportCommand = _DeleteExplainabilityExportCommand;

// src/commands/DeleteForecastCommand.ts




var _DeleteForecastCommand = class _DeleteForecastCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AmazonForecast", "DeleteForecast", {}).n("ForecastClient", "DeleteForecastCommand").f(void 0, void 0).ser(se_DeleteForecastCommand).de(de_DeleteForecastCommand).build() {
};
__name(_DeleteForecastCommand, "DeleteForecastCommand");
var DeleteForecastCommand = _DeleteForecastCommand;

// src/commands/DeleteForecastExportJobCommand.ts




var _DeleteForecastExportJobCommand = class _DeleteForecastExportJobCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AmazonForecast", "DeleteForecastExportJob", {}).n("ForecastClient", "DeleteForecastExportJobCommand").f(void 0, void 0).ser(se_DeleteForecastExportJobCommand).de(de_DeleteForecastExportJobCommand).build() {
};
__name(_DeleteForecastExportJobCommand, "DeleteForecastExportJobCommand");
var DeleteForecastExportJobCommand = _DeleteForecastExportJobCommand;

// src/commands/DeleteMonitorCommand.ts




var _DeleteMonitorCommand = class _DeleteMonitorCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AmazonForecast", "DeleteMonitor", {}).n("ForecastClient", "DeleteMonitorCommand").f(void 0, void 0).ser(se_DeleteMonitorCommand).de(de_DeleteMonitorCommand).build() {
};
__name(_DeleteMonitorCommand, "DeleteMonitorCommand");
var DeleteMonitorCommand = _DeleteMonitorCommand;

// src/commands/DeletePredictorBacktestExportJobCommand.ts




var _DeletePredictorBacktestExportJobCommand = class _DeletePredictorBacktestExportJobCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AmazonForecast", "DeletePredictorBacktestExportJob", {}).n("ForecastClient", "DeletePredictorBacktestExportJobCommand").f(void 0, void 0).ser(se_DeletePredictorBacktestExportJobCommand).de(de_DeletePredictorBacktestExportJobCommand).build() {
};
__name(_DeletePredictorBacktestExportJobCommand, "DeletePredictorBacktestExportJobCommand");
var DeletePredictorBacktestExportJobCommand = _DeletePredictorBacktestExportJobCommand;

// src/commands/DeletePredictorCommand.ts




var _DeletePredictorCommand = class _DeletePredictorCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AmazonForecast", "DeletePredictor", {}).n("ForecastClient", "DeletePredictorCommand").f(void 0, void 0).ser(se_DeletePredictorCommand).de(de_DeletePredictorCommand).build() {
};
__name(_DeletePredictorCommand, "DeletePredictorCommand");
var DeletePredictorCommand = _DeletePredictorCommand;

// src/commands/DeleteResourceTreeCommand.ts




var _DeleteResourceTreeCommand = class _DeleteResourceTreeCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AmazonForecast", "DeleteResourceTree", {}).n("ForecastClient", "DeleteResourceTreeCommand").f(void 0, void 0).ser(se_DeleteResourceTreeCommand).de(de_DeleteResourceTreeCommand).build() {
};
__name(_DeleteResourceTreeCommand, "DeleteResourceTreeCommand");
var DeleteResourceTreeCommand = _DeleteResourceTreeCommand;

// src/commands/DeleteWhatIfAnalysisCommand.ts




var _DeleteWhatIfAnalysisCommand = class _DeleteWhatIfAnalysisCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AmazonForecast", "DeleteWhatIfAnalysis", {}).n("ForecastClient", "DeleteWhatIfAnalysisCommand").f(void 0, void 0).ser(se_DeleteWhatIfAnalysisCommand).de(de_DeleteWhatIfAnalysisCommand).build() {
};
__name(_DeleteWhatIfAnalysisCommand, "DeleteWhatIfAnalysisCommand");
var DeleteWhatIfAnalysisCommand = _DeleteWhatIfAnalysisCommand;

// src/commands/DeleteWhatIfForecastCommand.ts




var _DeleteWhatIfForecastCommand = class _DeleteWhatIfForecastCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AmazonForecast", "DeleteWhatIfForecast", {}).n("ForecastClient", "DeleteWhatIfForecastCommand").f(void 0, void 0).ser(se_DeleteWhatIfForecastCommand).de(de_DeleteWhatIfForecastCommand).build() {
};
__name(_DeleteWhatIfForecastCommand, "DeleteWhatIfForecastCommand");
var DeleteWhatIfForecastCommand = _DeleteWhatIfForecastCommand;

// src/commands/DeleteWhatIfForecastExportCommand.ts




var _DeleteWhatIfForecastExportCommand = class _DeleteWhatIfForecastExportCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AmazonForecast", "DeleteWhatIfForecastExport", {}).n("ForecastClient", "DeleteWhatIfForecastExportCommand").f(void 0, void 0).ser(se_DeleteWhatIfForecastExportCommand).de(de_DeleteWhatIfForecastExportCommand).build() {
};
__name(_DeleteWhatIfForecastExportCommand, "DeleteWhatIfForecastExportCommand");
var DeleteWhatIfForecastExportCommand = _DeleteWhatIfForecastExportCommand;

// src/commands/DescribeAutoPredictorCommand.ts




var _DescribeAutoPredictorCommand = class _DescribeAutoPredictorCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AmazonForecast", "DescribeAutoPredictor", {}).n("ForecastClient", "DescribeAutoPredictorCommand").f(void 0, void 0).ser(se_DescribeAutoPredictorCommand).de(de_DescribeAutoPredictorCommand).build() {
};
__name(_DescribeAutoPredictorCommand, "DescribeAutoPredictorCommand");
var DescribeAutoPredictorCommand = _DescribeAutoPredictorCommand;

// src/commands/DescribeDatasetCommand.ts




var _DescribeDatasetCommand = class _DescribeDatasetCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AmazonForecast", "DescribeDataset", {}).n("ForecastClient", "DescribeDatasetCommand").f(void 0, void 0).ser(se_DescribeDatasetCommand).de(de_DescribeDatasetCommand).build() {
};
__name(_DescribeDatasetCommand, "DescribeDatasetCommand");
var DescribeDatasetCommand = _DescribeDatasetCommand;

// src/commands/DescribeDatasetGroupCommand.ts




var _DescribeDatasetGroupCommand = class _DescribeDatasetGroupCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AmazonForecast", "DescribeDatasetGroup", {}).n("ForecastClient", "DescribeDatasetGroupCommand").f(void 0, void 0).ser(se_DescribeDatasetGroupCommand).de(de_DescribeDatasetGroupCommand).build() {
};
__name(_DescribeDatasetGroupCommand, "DescribeDatasetGroupCommand");
var DescribeDatasetGroupCommand = _DescribeDatasetGroupCommand;

// src/commands/DescribeDatasetImportJobCommand.ts




var _DescribeDatasetImportJobCommand = class _DescribeDatasetImportJobCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AmazonForecast", "DescribeDatasetImportJob", {}).n("ForecastClient", "DescribeDatasetImportJobCommand").f(void 0, void 0).ser(se_DescribeDatasetImportJobCommand).de(de_DescribeDatasetImportJobCommand).build() {
};
__name(_DescribeDatasetImportJobCommand, "DescribeDatasetImportJobCommand");
var DescribeDatasetImportJobCommand = _DescribeDatasetImportJobCommand;

// src/commands/DescribeExplainabilityCommand.ts




var _DescribeExplainabilityCommand = class _DescribeExplainabilityCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AmazonForecast", "DescribeExplainability", {}).n("ForecastClient", "DescribeExplainabilityCommand").f(void 0, void 0).ser(se_DescribeExplainabilityCommand).de(de_DescribeExplainabilityCommand).build() {
};
__name(_DescribeExplainabilityCommand, "DescribeExplainabilityCommand");
var DescribeExplainabilityCommand = _DescribeExplainabilityCommand;

// src/commands/DescribeExplainabilityExportCommand.ts




var _DescribeExplainabilityExportCommand = class _DescribeExplainabilityExportCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AmazonForecast", "DescribeExplainabilityExport", {}).n("ForecastClient", "DescribeExplainabilityExportCommand").f(void 0, void 0).ser(se_DescribeExplainabilityExportCommand).de(de_DescribeExplainabilityExportCommand).build() {
};
__name(_DescribeExplainabilityExportCommand, "DescribeExplainabilityExportCommand");
var DescribeExplainabilityExportCommand = _DescribeExplainabilityExportCommand;

// src/commands/DescribeForecastCommand.ts




var _DescribeForecastCommand = class _DescribeForecastCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AmazonForecast", "DescribeForecast", {}).n("ForecastClient", "DescribeForecastCommand").f(void 0, void 0).ser(se_DescribeForecastCommand).de(de_DescribeForecastCommand).build() {
};
__name(_DescribeForecastCommand, "DescribeForecastCommand");
var DescribeForecastCommand = _DescribeForecastCommand;

// src/commands/DescribeForecastExportJobCommand.ts




var _DescribeForecastExportJobCommand = class _DescribeForecastExportJobCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AmazonForecast", "DescribeForecastExportJob", {}).n("ForecastClient", "DescribeForecastExportJobCommand").f(void 0, void 0).ser(se_DescribeForecastExportJobCommand).de(de_DescribeForecastExportJobCommand).build() {
};
__name(_DescribeForecastExportJobCommand, "DescribeForecastExportJobCommand");
var DescribeForecastExportJobCommand = _DescribeForecastExportJobCommand;

// src/commands/DescribeMonitorCommand.ts




var _DescribeMonitorCommand = class _DescribeMonitorCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AmazonForecast", "DescribeMonitor", {}).n("ForecastClient", "DescribeMonitorCommand").f(void 0, void 0).ser(se_DescribeMonitorCommand).de(de_DescribeMonitorCommand).build() {
};
__name(_DescribeMonitorCommand, "DescribeMonitorCommand");
var DescribeMonitorCommand = _DescribeMonitorCommand;

// src/commands/DescribePredictorBacktestExportJobCommand.ts




var _DescribePredictorBacktestExportJobCommand = class _DescribePredictorBacktestExportJobCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AmazonForecast", "DescribePredictorBacktestExportJob", {}).n("ForecastClient", "DescribePredictorBacktestExportJobCommand").f(void 0, void 0).ser(se_DescribePredictorBacktestExportJobCommand).de(de_DescribePredictorBacktestExportJobCommand).build() {
};
__name(_DescribePredictorBacktestExportJobCommand, "DescribePredictorBacktestExportJobCommand");
var DescribePredictorBacktestExportJobCommand = _DescribePredictorBacktestExportJobCommand;

// src/commands/DescribePredictorCommand.ts




var _DescribePredictorCommand = class _DescribePredictorCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AmazonForecast", "DescribePredictor", {}).n("ForecastClient", "DescribePredictorCommand").f(void 0, void 0).ser(se_DescribePredictorCommand).de(de_DescribePredictorCommand).build() {
};
__name(_DescribePredictorCommand, "DescribePredictorCommand");
var DescribePredictorCommand = _DescribePredictorCommand;

// src/commands/DescribeWhatIfAnalysisCommand.ts




var _DescribeWhatIfAnalysisCommand = class _DescribeWhatIfAnalysisCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AmazonForecast", "DescribeWhatIfAnalysis", {}).n("ForecastClient", "DescribeWhatIfAnalysisCommand").f(void 0, void 0).ser(se_DescribeWhatIfAnalysisCommand).de(de_DescribeWhatIfAnalysisCommand).build() {
};
__name(_DescribeWhatIfAnalysisCommand, "DescribeWhatIfAnalysisCommand");
var DescribeWhatIfAnalysisCommand = _DescribeWhatIfAnalysisCommand;

// src/commands/DescribeWhatIfForecastCommand.ts




var _DescribeWhatIfForecastCommand = class _DescribeWhatIfForecastCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AmazonForecast", "DescribeWhatIfForecast", {}).n("ForecastClient", "DescribeWhatIfForecastCommand").f(void 0, void 0).ser(se_DescribeWhatIfForecastCommand).de(de_DescribeWhatIfForecastCommand).build() {
};
__name(_DescribeWhatIfForecastCommand, "DescribeWhatIfForecastCommand");
var DescribeWhatIfForecastCommand = _DescribeWhatIfForecastCommand;

// src/commands/DescribeWhatIfForecastExportCommand.ts




var _DescribeWhatIfForecastExportCommand = class _DescribeWhatIfForecastExportCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AmazonForecast", "DescribeWhatIfForecastExport", {}).n("ForecastClient", "DescribeWhatIfForecastExportCommand").f(void 0, void 0).ser(se_DescribeWhatIfForecastExportCommand).de(de_DescribeWhatIfForecastExportCommand).build() {
};
__name(_DescribeWhatIfForecastExportCommand, "DescribeWhatIfForecastExportCommand");
var DescribeWhatIfForecastExportCommand = _DescribeWhatIfForecastExportCommand;

// src/commands/GetAccuracyMetricsCommand.ts




var _GetAccuracyMetricsCommand = class _GetAccuracyMetricsCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AmazonForecast", "GetAccuracyMetrics", {}).n("ForecastClient", "GetAccuracyMetricsCommand").f(void 0, void 0).ser(se_GetAccuracyMetricsCommand).de(de_GetAccuracyMetricsCommand).build() {
};
__name(_GetAccuracyMetricsCommand, "GetAccuracyMetricsCommand");
var GetAccuracyMetricsCommand = _GetAccuracyMetricsCommand;

// src/commands/ListDatasetGroupsCommand.ts




var _ListDatasetGroupsCommand = class _ListDatasetGroupsCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AmazonForecast", "ListDatasetGroups", {}).n("ForecastClient", "ListDatasetGroupsCommand").f(void 0, void 0).ser(se_ListDatasetGroupsCommand).de(de_ListDatasetGroupsCommand).build() {
};
__name(_ListDatasetGroupsCommand, "ListDatasetGroupsCommand");
var ListDatasetGroupsCommand = _ListDatasetGroupsCommand;

// src/commands/ListDatasetImportJobsCommand.ts




var _ListDatasetImportJobsCommand = class _ListDatasetImportJobsCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AmazonForecast", "ListDatasetImportJobs", {}).n("ForecastClient", "ListDatasetImportJobsCommand").f(void 0, void 0).ser(se_ListDatasetImportJobsCommand).de(de_ListDatasetImportJobsCommand).build() {
};
__name(_ListDatasetImportJobsCommand, "ListDatasetImportJobsCommand");
var ListDatasetImportJobsCommand = _ListDatasetImportJobsCommand;

// src/commands/ListDatasetsCommand.ts




var _ListDatasetsCommand = class _ListDatasetsCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AmazonForecast", "ListDatasets", {}).n("ForecastClient", "ListDatasetsCommand").f(void 0, void 0).ser(se_ListDatasetsCommand).de(de_ListDatasetsCommand).build() {
};
__name(_ListDatasetsCommand, "ListDatasetsCommand");
var ListDatasetsCommand = _ListDatasetsCommand;

// src/commands/ListExplainabilitiesCommand.ts




var _ListExplainabilitiesCommand = class _ListExplainabilitiesCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AmazonForecast", "ListExplainabilities", {}).n("ForecastClient", "ListExplainabilitiesCommand").f(void 0, void 0).ser(se_ListExplainabilitiesCommand).de(de_ListExplainabilitiesCommand).build() {
};
__name(_ListExplainabilitiesCommand, "ListExplainabilitiesCommand");
var ListExplainabilitiesCommand = _ListExplainabilitiesCommand;

// src/commands/ListExplainabilityExportsCommand.ts




var _ListExplainabilityExportsCommand = class _ListExplainabilityExportsCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AmazonForecast", "ListExplainabilityExports", {}).n("ForecastClient", "ListExplainabilityExportsCommand").f(void 0, void 0).ser(se_ListExplainabilityExportsCommand).de(de_ListExplainabilityExportsCommand).build() {
};
__name(_ListExplainabilityExportsCommand, "ListExplainabilityExportsCommand");
var ListExplainabilityExportsCommand = _ListExplainabilityExportsCommand;

// src/commands/ListForecastExportJobsCommand.ts




var _ListForecastExportJobsCommand = class _ListForecastExportJobsCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AmazonForecast", "ListForecastExportJobs", {}).n("ForecastClient", "ListForecastExportJobsCommand").f(void 0, void 0).ser(se_ListForecastExportJobsCommand).de(de_ListForecastExportJobsCommand).build() {
};
__name(_ListForecastExportJobsCommand, "ListForecastExportJobsCommand");
var ListForecastExportJobsCommand = _ListForecastExportJobsCommand;

// src/commands/ListForecastsCommand.ts




var _ListForecastsCommand = class _ListForecastsCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AmazonForecast", "ListForecasts", {}).n("ForecastClient", "ListForecastsCommand").f(void 0, void 0).ser(se_ListForecastsCommand).de(de_ListForecastsCommand).build() {
};
__name(_ListForecastsCommand, "ListForecastsCommand");
var ListForecastsCommand = _ListForecastsCommand;

// src/commands/ListMonitorEvaluationsCommand.ts




var _ListMonitorEvaluationsCommand = class _ListMonitorEvaluationsCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AmazonForecast", "ListMonitorEvaluations", {}).n("ForecastClient", "ListMonitorEvaluationsCommand").f(void 0, void 0).ser(se_ListMonitorEvaluationsCommand).de(de_ListMonitorEvaluationsCommand).build() {
};
__name(_ListMonitorEvaluationsCommand, "ListMonitorEvaluationsCommand");
var ListMonitorEvaluationsCommand = _ListMonitorEvaluationsCommand;

// src/commands/ListMonitorsCommand.ts




var _ListMonitorsCommand = class _ListMonitorsCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AmazonForecast", "ListMonitors", {}).n("ForecastClient", "ListMonitorsCommand").f(void 0, void 0).ser(se_ListMonitorsCommand).de(de_ListMonitorsCommand).build() {
};
__name(_ListMonitorsCommand, "ListMonitorsCommand");
var ListMonitorsCommand = _ListMonitorsCommand;

// src/commands/ListPredictorBacktestExportJobsCommand.ts




var _ListPredictorBacktestExportJobsCommand = class _ListPredictorBacktestExportJobsCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AmazonForecast", "ListPredictorBacktestExportJobs", {}).n("ForecastClient", "ListPredictorBacktestExportJobsCommand").f(void 0, void 0).ser(se_ListPredictorBacktestExportJobsCommand).de(de_ListPredictorBacktestExportJobsCommand).build() {
};
__name(_ListPredictorBacktestExportJobsCommand, "ListPredictorBacktestExportJobsCommand");
var ListPredictorBacktestExportJobsCommand = _ListPredictorBacktestExportJobsCommand;

// src/commands/ListPredictorsCommand.ts




var _ListPredictorsCommand = class _ListPredictorsCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AmazonForecast", "ListPredictors", {}).n("ForecastClient", "ListPredictorsCommand").f(void 0, void 0).ser(se_ListPredictorsCommand).de(de_ListPredictorsCommand).build() {
};
__name(_ListPredictorsCommand, "ListPredictorsCommand");
var ListPredictorsCommand = _ListPredictorsCommand;

// src/commands/ListTagsForResourceCommand.ts




var _ListTagsForResourceCommand = class _ListTagsForResourceCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AmazonForecast", "ListTagsForResource", {}).n("ForecastClient", "ListTagsForResourceCommand").f(void 0, ListTagsForResourceResponseFilterSensitiveLog).ser(se_ListTagsForResourceCommand).de(de_ListTagsForResourceCommand).build() {
};
__name(_ListTagsForResourceCommand, "ListTagsForResourceCommand");
var ListTagsForResourceCommand = _ListTagsForResourceCommand;

// src/commands/ListWhatIfAnalysesCommand.ts




var _ListWhatIfAnalysesCommand = class _ListWhatIfAnalysesCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AmazonForecast", "ListWhatIfAnalyses", {}).n("ForecastClient", "ListWhatIfAnalysesCommand").f(void 0, void 0).ser(se_ListWhatIfAnalysesCommand).de(de_ListWhatIfAnalysesCommand).build() {
};
__name(_ListWhatIfAnalysesCommand, "ListWhatIfAnalysesCommand");
var ListWhatIfAnalysesCommand = _ListWhatIfAnalysesCommand;

// src/commands/ListWhatIfForecastExportsCommand.ts




var _ListWhatIfForecastExportsCommand = class _ListWhatIfForecastExportsCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AmazonForecast", "ListWhatIfForecastExports", {}).n("ForecastClient", "ListWhatIfForecastExportsCommand").f(void 0, void 0).ser(se_ListWhatIfForecastExportsCommand).de(de_ListWhatIfForecastExportsCommand).build() {
};
__name(_ListWhatIfForecastExportsCommand, "ListWhatIfForecastExportsCommand");
var ListWhatIfForecastExportsCommand = _ListWhatIfForecastExportsCommand;

// src/commands/ListWhatIfForecastsCommand.ts




var _ListWhatIfForecastsCommand = class _ListWhatIfForecastsCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AmazonForecast", "ListWhatIfForecasts", {}).n("ForecastClient", "ListWhatIfForecastsCommand").f(void 0, void 0).ser(se_ListWhatIfForecastsCommand).de(de_ListWhatIfForecastsCommand).build() {
};
__name(_ListWhatIfForecastsCommand, "ListWhatIfForecastsCommand");
var ListWhatIfForecastsCommand = _ListWhatIfForecastsCommand;

// src/commands/ResumeResourceCommand.ts




var _ResumeResourceCommand = class _ResumeResourceCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AmazonForecast", "ResumeResource", {}).n("ForecastClient", "ResumeResourceCommand").f(void 0, void 0).ser(se_ResumeResourceCommand).de(de_ResumeResourceCommand).build() {
};
__name(_ResumeResourceCommand, "ResumeResourceCommand");
var ResumeResourceCommand = _ResumeResourceCommand;

// src/commands/StopResourceCommand.ts




var _StopResourceCommand = class _StopResourceCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AmazonForecast", "StopResource", {}).n("ForecastClient", "StopResourceCommand").f(void 0, void 0).ser(se_StopResourceCommand).de(de_StopResourceCommand).build() {
};
__name(_StopResourceCommand, "StopResourceCommand");
var StopResourceCommand = _StopResourceCommand;

// src/commands/TagResourceCommand.ts




var _TagResourceCommand = class _TagResourceCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AmazonForecast", "TagResource", {}).n("ForecastClient", "TagResourceCommand").f(TagResourceRequestFilterSensitiveLog, void 0).ser(se_TagResourceCommand).de(de_TagResourceCommand).build() {
};
__name(_TagResourceCommand, "TagResourceCommand");
var TagResourceCommand = _TagResourceCommand;

// src/commands/UntagResourceCommand.ts




var _UntagResourceCommand = class _UntagResourceCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AmazonForecast", "UntagResource", {}).n("ForecastClient", "UntagResourceCommand").f(UntagResourceRequestFilterSensitiveLog, void 0).ser(se_UntagResourceCommand).de(de_UntagResourceCommand).build() {
};
__name(_UntagResourceCommand, "UntagResourceCommand");
var UntagResourceCommand = _UntagResourceCommand;

// src/commands/UpdateDatasetGroupCommand.ts




var _UpdateDatasetGroupCommand = class _UpdateDatasetGroupCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AmazonForecast", "UpdateDatasetGroup", {}).n("ForecastClient", "UpdateDatasetGroupCommand").f(void 0, void 0).ser(se_UpdateDatasetGroupCommand).de(de_UpdateDatasetGroupCommand).build() {
};
__name(_UpdateDatasetGroupCommand, "UpdateDatasetGroupCommand");
var UpdateDatasetGroupCommand = _UpdateDatasetGroupCommand;

// src/Forecast.ts
var commands = {
  CreateAutoPredictorCommand,
  CreateDatasetCommand,
  CreateDatasetGroupCommand,
  CreateDatasetImportJobCommand,
  CreateExplainabilityCommand,
  CreateExplainabilityExportCommand,
  CreateForecastCommand,
  CreateForecastExportJobCommand,
  CreateMonitorCommand,
  CreatePredictorCommand,
  CreatePredictorBacktestExportJobCommand,
  CreateWhatIfAnalysisCommand,
  CreateWhatIfForecastCommand,
  CreateWhatIfForecastExportCommand,
  DeleteDatasetCommand,
  DeleteDatasetGroupCommand,
  DeleteDatasetImportJobCommand,
  DeleteExplainabilityCommand,
  DeleteExplainabilityExportCommand,
  DeleteForecastCommand,
  DeleteForecastExportJobCommand,
  DeleteMonitorCommand,
  DeletePredictorCommand,
  DeletePredictorBacktestExportJobCommand,
  DeleteResourceTreeCommand,
  DeleteWhatIfAnalysisCommand,
  DeleteWhatIfForecastCommand,
  DeleteWhatIfForecastExportCommand,
  DescribeAutoPredictorCommand,
  DescribeDatasetCommand,
  DescribeDatasetGroupCommand,
  DescribeDatasetImportJobCommand,
  DescribeExplainabilityCommand,
  DescribeExplainabilityExportCommand,
  DescribeForecastCommand,
  DescribeForecastExportJobCommand,
  DescribeMonitorCommand,
  DescribePredictorCommand,
  DescribePredictorBacktestExportJobCommand,
  DescribeWhatIfAnalysisCommand,
  DescribeWhatIfForecastCommand,
  DescribeWhatIfForecastExportCommand,
  GetAccuracyMetricsCommand,
  ListDatasetGroupsCommand,
  ListDatasetImportJobsCommand,
  ListDatasetsCommand,
  ListExplainabilitiesCommand,
  ListExplainabilityExportsCommand,
  ListForecastExportJobsCommand,
  ListForecastsCommand,
  ListMonitorEvaluationsCommand,
  ListMonitorsCommand,
  ListPredictorBacktestExportJobsCommand,
  ListPredictorsCommand,
  ListTagsForResourceCommand,
  ListWhatIfAnalysesCommand,
  ListWhatIfForecastExportsCommand,
  ListWhatIfForecastsCommand,
  ResumeResourceCommand,
  StopResourceCommand,
  TagResourceCommand,
  UntagResourceCommand,
  UpdateDatasetGroupCommand
};
var _Forecast = class _Forecast extends ForecastClient {
};
__name(_Forecast, "Forecast");
var Forecast = _Forecast;
(0, import_smithy_client.createAggregatedClient)(commands, Forecast);

// src/pagination/ListDatasetGroupsPaginator.ts

var paginateListDatasetGroups = (0, import_core.createPaginator)(ForecastClient, ListDatasetGroupsCommand, "NextToken", "NextToken", "MaxResults");

// src/pagination/ListDatasetImportJobsPaginator.ts

var paginateListDatasetImportJobs = (0, import_core.createPaginator)(ForecastClient, ListDatasetImportJobsCommand, "NextToken", "NextToken", "MaxResults");

// src/pagination/ListDatasetsPaginator.ts

var paginateListDatasets = (0, import_core.createPaginator)(ForecastClient, ListDatasetsCommand, "NextToken", "NextToken", "MaxResults");

// src/pagination/ListExplainabilitiesPaginator.ts

var paginateListExplainabilities = (0, import_core.createPaginator)(ForecastClient, ListExplainabilitiesCommand, "NextToken", "NextToken", "MaxResults");

// src/pagination/ListExplainabilityExportsPaginator.ts

var paginateListExplainabilityExports = (0, import_core.createPaginator)(ForecastClient, ListExplainabilityExportsCommand, "NextToken", "NextToken", "MaxResults");

// src/pagination/ListForecastExportJobsPaginator.ts

var paginateListForecastExportJobs = (0, import_core.createPaginator)(ForecastClient, ListForecastExportJobsCommand, "NextToken", "NextToken", "MaxResults");

// src/pagination/ListForecastsPaginator.ts

var paginateListForecasts = (0, import_core.createPaginator)(ForecastClient, ListForecastsCommand, "NextToken", "NextToken", "MaxResults");

// src/pagination/ListMonitorEvaluationsPaginator.ts

var paginateListMonitorEvaluations = (0, import_core.createPaginator)(ForecastClient, ListMonitorEvaluationsCommand, "NextToken", "NextToken", "MaxResults");

// src/pagination/ListMonitorsPaginator.ts

var paginateListMonitors = (0, import_core.createPaginator)(ForecastClient, ListMonitorsCommand, "NextToken", "NextToken", "MaxResults");

// src/pagination/ListPredictorBacktestExportJobsPaginator.ts

var paginateListPredictorBacktestExportJobs = (0, import_core.createPaginator)(ForecastClient, ListPredictorBacktestExportJobsCommand, "NextToken", "NextToken", "MaxResults");

// src/pagination/ListPredictorsPaginator.ts

var paginateListPredictors = (0, import_core.createPaginator)(ForecastClient, ListPredictorsCommand, "NextToken", "NextToken", "MaxResults");

// src/pagination/ListWhatIfAnalysesPaginator.ts

var paginateListWhatIfAnalyses = (0, import_core.createPaginator)(ForecastClient, ListWhatIfAnalysesCommand, "NextToken", "NextToken", "MaxResults");

// src/pagination/ListWhatIfForecastExportsPaginator.ts

var paginateListWhatIfForecastExports = (0, import_core.createPaginator)(ForecastClient, ListWhatIfForecastExportsCommand, "NextToken", "NextToken", "MaxResults");

// src/pagination/ListWhatIfForecastsPaginator.ts

var paginateListWhatIfForecasts = (0, import_core.createPaginator)(ForecastClient, ListWhatIfForecastsCommand, "NextToken", "NextToken", "MaxResults");
// Annotate the CommonJS export names for ESM import in node:

0 && (module.exports = {
  ForecastServiceException,
  __Client,
  ForecastClient,
  Forecast,
  $Command,
  CreateAutoPredictorCommand,
  CreateDatasetCommand,
  CreateDatasetGroupCommand,
  CreateDatasetImportJobCommand,
  CreateExplainabilityCommand,
  CreateExplainabilityExportCommand,
  CreateForecastCommand,
  CreateForecastExportJobCommand,
  CreateMonitorCommand,
  CreatePredictorBacktestExportJobCommand,
  CreatePredictorCommand,
  CreateWhatIfAnalysisCommand,
  CreateWhatIfForecastCommand,
  CreateWhatIfForecastExportCommand,
  DeleteDatasetCommand,
  DeleteDatasetGroupCommand,
  DeleteDatasetImportJobCommand,
  DeleteExplainabilityCommand,
  DeleteExplainabilityExportCommand,
  DeleteForecastCommand,
  DeleteForecastExportJobCommand,
  DeleteMonitorCommand,
  DeletePredictorBacktestExportJobCommand,
  DeletePredictorCommand,
  DeleteResourceTreeCommand,
  DeleteWhatIfAnalysisCommand,
  DeleteWhatIfForecastCommand,
  DeleteWhatIfForecastExportCommand,
  DescribeAutoPredictorCommand,
  DescribeDatasetCommand,
  DescribeDatasetGroupCommand,
  DescribeDatasetImportJobCommand,
  DescribeExplainabilityCommand,
  DescribeExplainabilityExportCommand,
  DescribeForecastCommand,
  DescribeForecastExportJobCommand,
  DescribeMonitorCommand,
  DescribePredictorBacktestExportJobCommand,
  DescribePredictorCommand,
  DescribeWhatIfAnalysisCommand,
  DescribeWhatIfForecastCommand,
  DescribeWhatIfForecastExportCommand,
  GetAccuracyMetricsCommand,
  ListDatasetGroupsCommand,
  ListDatasetImportJobsCommand,
  ListDatasetsCommand,
  ListExplainabilitiesCommand,
  ListExplainabilityExportsCommand,
  ListForecastExportJobsCommand,
  ListForecastsCommand,
  ListMonitorEvaluationsCommand,
  ListMonitorsCommand,
  ListPredictorBacktestExportJobsCommand,
  ListPredictorsCommand,
  ListTagsForResourceCommand,
  ListWhatIfAnalysesCommand,
  ListWhatIfForecastExportsCommand,
  ListWhatIfForecastsCommand,
  ResumeResourceCommand,
  StopResourceCommand,
  TagResourceCommand,
  UntagResourceCommand,
  UpdateDatasetGroupCommand,
  paginateListDatasetGroups,
  paginateListDatasetImportJobs,
  paginateListDatasets,
  paginateListExplainabilities,
  paginateListExplainabilityExports,
  paginateListForecastExportJobs,
  paginateListForecasts,
  paginateListMonitorEvaluations,
  paginateListMonitors,
  paginateListPredictorBacktestExportJobs,
  paginateListPredictors,
  paginateListWhatIfAnalyses,
  paginateListWhatIfForecastExports,
  paginateListWhatIfForecasts,
  Operation,
  OptimizationMetric,
  DayOfWeek,
  Month,
  InvalidInputException,
  LimitExceededException,
  ResourceAlreadyExistsException,
  ResourceInUseException,
  ResourceNotFoundException,
  DatasetType,
  Domain,
  AttributeType,
  ImportMode,
  TimePointGranularity,
  TimeSeriesGranularity,
  AutoMLOverrideStrategy,
  FeaturizationMethodName,
  ScalingType,
  Condition,
  State,
  EvaluationType,
  InvalidNextTokenException,
  FilterConditionString,
  TagFilterSensitiveLog,
  CreateAutoPredictorRequestFilterSensitiveLog,
  CreateDatasetRequestFilterSensitiveLog,
  CreateDatasetGroupRequestFilterSensitiveLog,
  CreateDatasetImportJobRequestFilterSensitiveLog,
  CreateExplainabilityRequestFilterSensitiveLog,
  CreateExplainabilityExportRequestFilterSensitiveLog,
  CreateForecastRequestFilterSensitiveLog,
  CreateForecastExportJobRequestFilterSensitiveLog,
  CreateMonitorRequestFilterSensitiveLog,
  CreatePredictorRequestFilterSensitiveLog,
  CreatePredictorBacktestExportJobRequestFilterSensitiveLog,
  CreateWhatIfAnalysisRequestFilterSensitiveLog,
  CreateWhatIfForecastRequestFilterSensitiveLog,
  CreateWhatIfForecastExportRequestFilterSensitiveLog,
  ListTagsForResourceResponseFilterSensitiveLog,
  TagResourceRequestFilterSensitiveLog,
  UntagResourceRequestFilterSensitiveLog
});
