// @azure/arm-botservice/dist-esm/src/models/index.js
/*
 * Copyright (c) Microsoft Corporation.
 * Licensed under the MIT License.
 *
 * Code generated by Microsoft (R) AutoRest Code Generator.
 * Changes may cause incorrect behavior and will be lost if the code is regenerated.
 */
/** Known values of {@link MsaAppType} that the service accepts. */
export var KnownMsaAppType;
(function (KnownMsaAppType) {
    /** UserAssignedMSI */
    KnownMsaAppType["UserAssignedMSI"] = "UserAssignedMSI";
    /** SingleTenant */
    KnownMsaAppType["SingleTenant"] = "SingleTenant";
    /** MultiTenant */
    KnownMsaAppType["MultiTenant"] = "MultiTenant";
})(KnownMsaAppType || (KnownMsaAppType = {}));
/** Known values of {@link PublicNetworkAccess} that the service accepts. */
export var KnownPublicNetworkAccess;
(function (KnownPublicNetworkAccess) {
    /** Enabled */
    KnownPublicNetworkAccess["Enabled"] = "Enabled";
    /** Disabled */
    KnownPublicNetworkAccess["Disabled"] = "Disabled";
})(KnownPublicNetworkAccess || (KnownPublicNetworkAccess = {}));
/** Known values of {@link PrivateEndpointServiceConnectionStatus} that the service accepts. */
export var KnownPrivateEndpointServiceConnectionStatus;
(function (KnownPrivateEndpointServiceConnectionStatus) {
    /** Pending */
    KnownPrivateEndpointServiceConnectionStatus["Pending"] = "Pending";
    /** Approved */
    KnownPrivateEndpointServiceConnectionStatus["Approved"] = "Approved";
    /** Rejected */
    KnownPrivateEndpointServiceConnectionStatus["Rejected"] = "Rejected";
})(KnownPrivateEndpointServiceConnectionStatus || (KnownPrivateEndpointServiceConnectionStatus = {}));
/** Known values of {@link PrivateEndpointConnectionProvisioningState} that the service accepts. */
export var KnownPrivateEndpointConnectionProvisioningState;
(function (KnownPrivateEndpointConnectionProvisioningState) {
    /** Succeeded */
    KnownPrivateEndpointConnectionProvisioningState["Succeeded"] = "Succeeded";
    /** Creating */
    KnownPrivateEndpointConnectionProvisioningState["Creating"] = "Creating";
    /** Deleting */
    KnownPrivateEndpointConnectionProvisioningState["Deleting"] = "Deleting";
    /** Failed */
    KnownPrivateEndpointConnectionProvisioningState["Failed"] = "Failed";
})(KnownPrivateEndpointConnectionProvisioningState || (KnownPrivateEndpointConnectionProvisioningState = {}));
/** Known values of {@link SkuName} that the service accepts. */
export var KnownSkuName;
(function (KnownSkuName) {
    /** F0 */
    KnownSkuName["F0"] = "F0";
    /** S1 */
    KnownSkuName["S1"] = "S1";
})(KnownSkuName || (KnownSkuName = {}));
/** Known values of {@link SkuTier} that the service accepts. */
export var KnownSkuTier;
(function (KnownSkuTier) {
    /** Free */
    KnownSkuTier["Free"] = "Free";
    /** Standard */
    KnownSkuTier["Standard"] = "Standard";
})(KnownSkuTier || (KnownSkuTier = {}));
/** Known values of {@link Kind} that the service accepts. */
export var KnownKind;
(function (KnownKind) {
    /** Sdk */
    KnownKind["Sdk"] = "sdk";
    /** Designer */
    KnownKind["Designer"] = "designer";
    /** Bot */
    KnownKind["Bot"] = "bot";
    /** Function */
    KnownKind["Function"] = "function";
    /** Azurebot */
    KnownKind["Azurebot"] = "azurebot";
})(KnownKind || (KnownKind = {}));
/** Known values of {@link OperationResultStatus} that the service accepts. */
export var KnownOperationResultStatus;
(function (KnownOperationResultStatus) {
    /** Canceled */
    KnownOperationResultStatus["Canceled"] = "Canceled";
    /** Succeeded */
    KnownOperationResultStatus["Succeeded"] = "Succeeded";
    /** Failed */
    KnownOperationResultStatus["Failed"] = "Failed";
    /** Requested */
    KnownOperationResultStatus["Requested"] = "Requested";
    /** Running */
    KnownOperationResultStatus["Running"] = "Running";
})(KnownOperationResultStatus || (KnownOperationResultStatus = {}));
