// @azure/arm-reservations/dist-esm/src/azureReservationAPI.js
/*
 * Copyright (c) Microsoft Corporation.
 * Licensed under the MIT License.
 *
 * Code generated by Microsoft (R) AutoRest Code Generator.
 * Changes may cause incorrect behavior and will be lost if the code is regenerated.
 */
import { __asyncDelegator, __asyncGenerator, __asyncValues, __await } from "tslib";
import * as coreClient from "@azure/core-client";
import * as coreRestPipeline from "@azure/core-rest-pipeline";
import { setContinuationToken } from "./pagingHelper";
import { ReservationImpl, ReservationOrderImpl, OperationImpl, CalculateRefundImpl, ReturnImpl, CalculateExchangeImpl, ExchangeImpl, QuotaImpl, QuotaRequestStatusImpl } from "./operations";
import * as Parameters from "./models/parameters";
import * as Mappers from "./models/mappers";
/// <reference lib="esnext.asynciterable" />
export class AzureReservationAPI extends coreClient.ServiceClient {
    /**
     * Initializes a new instance of the AzureReservationAPI class.
     * @param credentials Subscription credentials which uniquely identify client subscription.
     * @param options The parameter options
     */
    constructor(credentials, options) {
        var _a, _b, _c;
        if (credentials === undefined) {
            throw new Error("'credentials' cannot be null");
        }
        // Initializing default values for options
        if (!options) {
            options = {};
        }
        const defaults = {
            requestContentType: "application/json; charset=utf-8",
            credential: credentials
        };
        const packageDetails = `azsdk-js-arm-reservations/9.0.0`;
        const userAgentPrefix = options.userAgentOptions && options.userAgentOptions.userAgentPrefix
            ? `${options.userAgentOptions.userAgentPrefix} ${packageDetails}`
            : `${packageDetails}`;
        const optionsWithDefaults = Object.assign(Object.assign(Object.assign({}, defaults), options), { userAgentOptions: {
                userAgentPrefix
            }, endpoint: (_b = (_a = options.endpoint) !== null && _a !== void 0 ? _a : options.baseUri) !== null && _b !== void 0 ? _b : "https://management.azure.com" });
        super(optionsWithDefaults);
        let bearerTokenAuthenticationPolicyFound = false;
        if ((options === null || options === void 0 ? void 0 : options.pipeline) && options.pipeline.getOrderedPolicies().length > 0) {
            const pipelinePolicies = options.pipeline.getOrderedPolicies();
            bearerTokenAuthenticationPolicyFound = pipelinePolicies.some((pipelinePolicy) => pipelinePolicy.name ===
                coreRestPipeline.bearerTokenAuthenticationPolicyName);
        }
        if (!options ||
            !options.pipeline ||
            options.pipeline.getOrderedPolicies().length == 0 ||
            !bearerTokenAuthenticationPolicyFound) {
            this.pipeline.removePolicy({
                name: coreRestPipeline.bearerTokenAuthenticationPolicyName
            });
            this.pipeline.addPolicy(coreRestPipeline.bearerTokenAuthenticationPolicy({
                credential: credentials,
                scopes: (_c = optionsWithDefaults.credentialScopes) !== null && _c !== void 0 ? _c : `${optionsWithDefaults.endpoint}/.default`,
                challengeCallbacks: {
                    authorizeRequestOnChallenge: coreClient.authorizeRequestOnClaimChallenge
                }
            }));
        }
        // Assigning values to Constant parameters
        this.$host = options.$host || "https://management.azure.com";
        this.reservation = new ReservationImpl(this);
        this.reservationOrder = new ReservationOrderImpl(this);
        this.operation = new OperationImpl(this);
        this.calculateRefund = new CalculateRefundImpl(this);
        this.return = new ReturnImpl(this);
        this.calculateExchange = new CalculateExchangeImpl(this);
        this.exchange = new ExchangeImpl(this);
        this.quota = new QuotaImpl(this);
        this.quotaRequestStatus = new QuotaRequestStatusImpl(this);
    }
    /**
     * Get the regions and skus that are available for RI purchase for the specified Azure subscription.
     * @param subscriptionId Id of the subscription
     * @param options The options parameters.
     */
    listCatalog(subscriptionId, options) {
        const iter = this.getCatalogPagingAll(subscriptionId, options);
        return {
            next() {
                return iter.next();
            },
            [Symbol.asyncIterator]() {
                return this;
            },
            byPage: (settings) => {
                if (settings === null || settings === void 0 ? void 0 : settings.maxPageSize) {
                    throw new Error("maxPageSize is not supported by this operation.");
                }
                return this.getCatalogPagingPage(subscriptionId, options, settings);
            }
        };
    }
    getCatalogPagingPage(subscriptionId, options, settings) {
        return __asyncGenerator(this, arguments, function* getCatalogPagingPage_1() {
            let result;
            let continuationToken = settings === null || settings === void 0 ? void 0 : settings.continuationToken;
            if (!continuationToken) {
                result = yield __await(this._getCatalog(subscriptionId, options));
                let page = result.value || [];
                continuationToken = result.nextLink;
                setContinuationToken(page, continuationToken);
                yield yield __await(page);
            }
            while (continuationToken) {
                result = yield __await(this._getCatalogNext(subscriptionId, continuationToken, options));
                continuationToken = result.nextLink;
                let page = result.value || [];
                setContinuationToken(page, continuationToken);
                yield yield __await(page);
            }
        });
    }
    getCatalogPagingAll(subscriptionId, options) {
        return __asyncGenerator(this, arguments, function* getCatalogPagingAll_1() {
            var e_1, _a;
            try {
                for (var _b = __asyncValues(this.getCatalogPagingPage(subscriptionId, options)), _c; _c = yield __await(_b.next()), !_c.done;) {
                    const page = _c.value;
                    yield __await(yield* __asyncDelegator(__asyncValues(page)));
                }
            }
            catch (e_1_1) { e_1 = { error: e_1_1 }; }
            finally {
                try {
                    if (_c && !_c.done && (_a = _b.return)) yield __await(_a.call(_b));
                }
                finally { if (e_1) throw e_1.error; }
            }
        });
    }
    /**
     * Get the regions and skus that are available for RI purchase for the specified Azure subscription.
     * @param subscriptionId Id of the subscription
     * @param options The options parameters.
     */
    _getCatalog(subscriptionId, options) {
        return this.sendOperationRequest({ subscriptionId, options }, getCatalogOperationSpec);
    }
    /**
     * Get applicable `Reservation`s that are applied to this subscription or a resource group under this
     * subscription.
     * @param subscriptionId Id of the subscription
     * @param options The options parameters.
     */
    getAppliedReservationList(subscriptionId, options) {
        return this.sendOperationRequest({ subscriptionId, options }, getAppliedReservationListOperationSpec);
    }
    /**
     * GetCatalogNext
     * @param subscriptionId Id of the subscription
     * @param nextLink The nextLink from the previous successful call to the GetCatalog method.
     * @param options The options parameters.
     */
    _getCatalogNext(subscriptionId, nextLink, options) {
        return this.sendOperationRequest({ subscriptionId, nextLink, options }, getCatalogNextOperationSpec);
    }
}
// Operation Specifications
const serializer = coreClient.createSerializer(Mappers, /* isXml */ false);
const getCatalogOperationSpec = {
    path: "/subscriptions/{subscriptionId}/providers/Microsoft.Capacity/catalogs",
    httpMethod: "GET",
    responses: {
        200: {
            bodyMapper: Mappers.CatalogsResult
        },
        default: {
            bodyMapper: Mappers.ErrorModel
        }
    },
    queryParameters: [
        Parameters.apiVersion,
        Parameters.filter,
        Parameters.reservedResourceType,
        Parameters.location,
        Parameters.publisherId,
        Parameters.offerId,
        Parameters.planId,
        Parameters.skip,
        Parameters.take1
    ],
    urlParameters: [Parameters.$host, Parameters.subscriptionId],
    headerParameters: [Parameters.accept],
    serializer
};
const getAppliedReservationListOperationSpec = {
    path: "/subscriptions/{subscriptionId}/providers/Microsoft.Capacity/appliedReservations",
    httpMethod: "GET",
    responses: {
        200: {
            bodyMapper: Mappers.AppliedReservations
        },
        default: {
            bodyMapper: Mappers.ErrorModel
        }
    },
    queryParameters: [Parameters.apiVersion],
    urlParameters: [Parameters.$host, Parameters.subscriptionId],
    headerParameters: [Parameters.accept],
    serializer
};
const getCatalogNextOperationSpec = {
    path: "{nextLink}",
    httpMethod: "GET",
    responses: {
        200: {
            bodyMapper: Mappers.CatalogsResult
        },
        default: {
            bodyMapper: Mappers.ErrorModel
        }
    },
    urlParameters: [
        Parameters.$host,
        Parameters.nextLink,
        Parameters.subscriptionId
    ],
    headerParameters: [Parameters.accept],
    serializer
};
