// @azure/arm-network/dist-esm/test/network_examples.js
/*
 * Copyright (c) Microsoft Corporation.
 * Licensed under the MIT License.
 *
 * Code generated by Microsoft (R) AutoRest Code Generator.
 * Changes may cause incorrect behavior and will be lost if the code is regenerated.
 */
import { __asyncValues, __awaiter } from "tslib";
import { env, Recorder, isPlaybackMode, } from "@azure-tools/test-recorder";
import { createTestCredential } from "@azure-tools/test-credential";
import { assert } from "chai";
import { NetworkManagementClient } from "../src/networkManagementClient";
const replaceableVariables = {
    AZURE_CLIENT_ID: "azure_client_id",
    AZURE_CLIENT_SECRET: "azure_client_secret",
    AZURE_TENANT_ID: "88888888-8888-8888-8888-888888888888",
    SUBSCRIPTION_ID: "azure_subscription_id"
};
const recorderOptions = {
    envSetupForPlayback: replaceableVariables
};
export const testPollingOptions = {
    updateIntervalInMs: isPlaybackMode() ? 0 : undefined,
};
describe("Network test", () => {
    let recorder;
    let subscriptionId;
    let client;
    let location;
    let resourceGroupName;
    let virtualNetworkName;
    let subnetName;
    let ipGroupName;
    beforeEach(function () {
        return __awaiter(this, void 0, void 0, function* () {
            recorder = new Recorder(this.currentTest);
            yield recorder.start(recorderOptions);
            subscriptionId = env.SUBSCRIPTION_ID || '';
            // This is an example of how the environment variables are used
            const credential = createTestCredential();
            client = new NetworkManagementClient(credential, subscriptionId, recorder.configureClientOptions({}));
            location = "eastus";
            resourceGroupName = "myjstest";
            virtualNetworkName = "virtualnetworkzzz";
            subnetName = "subnetzzz";
            ipGroupName = "ipgroupyyy";
        });
    });
    afterEach(function () {
        return __awaiter(this, void 0, void 0, function* () {
            yield recorder.stop();
        });
    });
    it("virtualNetworks create test", function () {
        return __awaiter(this, void 0, void 0, function* () {
            const res = yield client.virtualNetworks.beginCreateOrUpdateAndWait(resourceGroupName, virtualNetworkName, {
                addressSpace: {
                    addressPrefixes: ["10.0.0.0/16"],
                },
                location: "eastus",
            }, testPollingOptions);
        });
    });
    it("subnets create test", function () {
        return __awaiter(this, void 0, void 0, function* () {
            const res = yield client.subnets.beginCreateOrUpdateAndWait(resourceGroupName, virtualNetworkName, subnetName, { addressPrefix: "10.0.0.0/24" }, testPollingOptions);
        });
    });
    it("ipGroups create test", function () {
        return __awaiter(this, void 0, void 0, function* () {
            const res = yield client.ipGroups.beginCreateOrUpdateAndWait(resourceGroupName, virtualNetworkName, {
                tags: {
                    key1: "value1",
                },
                location: "eastus",
                ipAddresses: ["13.64.39.16/32", "40.74.146.80/31", "40.74.147.32/28"],
            }, testPollingOptions);
        });
    });
    it("virtualNetworks get test", function () {
        return __awaiter(this, void 0, void 0, function* () {
            const res = yield client.virtualNetworks.get(resourceGroupName, virtualNetworkName);
            assert.equal(res.name, virtualNetworkName);
        });
    });
    it("subnets get test", function () {
        return __awaiter(this, void 0, void 0, function* () {
            const res = yield client.subnets.get(resourceGroupName, virtualNetworkName, subnetName);
            assert.equal(res.name, subnetName);
        });
    });
    it("ipGroups get test", function () {
        return __awaiter(this, void 0, void 0, function* () {
            const res = yield client.ipGroups.get(resourceGroupName, virtualNetworkName);
            assert.equal(res.type, "Microsoft.Network/IpGroups");
        });
    });
    it("virtualNetworks list test", function () {
        var e_1, _a;
        return __awaiter(this, void 0, void 0, function* () {
            const resArray = new Array();
            try {
                for (var _b = __asyncValues(client.virtualNetworks.list(resourceGroupName)), _c; _c = yield _b.next(), !_c.done;) {
                    let item = _c.value;
                    resArray.push(item);
                }
            }
            catch (e_1_1) { e_1 = { error: e_1_1 }; }
            finally {
                try {
                    if (_c && !_c.done && (_a = _b.return)) yield _a.call(_b);
                }
                finally { if (e_1) throw e_1.error; }
            }
            ;
            assert.equal(resArray.length, 1);
        });
    });
    it("subnets list test", function () {
        var e_2, _a;
        return __awaiter(this, void 0, void 0, function* () {
            const resArray = new Array();
            try {
                for (var _b = __asyncValues(client.subnets.list(resourceGroupName, virtualNetworkName)), _c; _c = yield _b.next(), !_c.done;) {
                    let item = _c.value;
                    resArray.push(item);
                }
            }
            catch (e_2_1) { e_2 = { error: e_2_1 }; }
            finally {
                try {
                    if (_c && !_c.done && (_a = _b.return)) yield _a.call(_b);
                }
                finally { if (e_2) throw e_2.error; }
            }
            ;
            assert.equal(resArray.length, 1);
        });
    });
    it("ipGroups list test", function () {
        var e_3, _a;
        return __awaiter(this, void 0, void 0, function* () {
            const resArray = new Array();
            try {
                for (var _b = __asyncValues(client.ipGroups.listByResourceGroup(resourceGroupName)), _c; _c = yield _b.next(), !_c.done;) {
                    let item = _c.value;
                    resArray.push(item);
                }
            }
            catch (e_3_1) { e_3 = { error: e_3_1 }; }
            finally {
                try {
                    if (_c && !_c.done && (_a = _b.return)) yield _a.call(_b);
                }
                finally { if (e_3) throw e_3.error; }
            }
            ;
            assert.equal(resArray.length, 1);
        });
    });
    it("virtualNetworks updatetags test", function () {
        return __awaiter(this, void 0, void 0, function* () {
            const res = yield client.virtualNetworks.updateTags(resourceGroupName, virtualNetworkName, { tags: { tag1: "value1", tag2: "value2" } });
            assert.equal(res.name, virtualNetworkName);
        });
    });
    it("ipGroups beginDeleteAndWait test", function () {
        var e_4, _a;
        return __awaiter(this, void 0, void 0, function* () {
            const res = yield client.ipGroups.beginDeleteAndWait(resourceGroupName, virtualNetworkName, testPollingOptions);
            const resArray = new Array();
            try {
                for (var _b = __asyncValues(client.ipGroups.listByResourceGroup(resourceGroupName)), _c; _c = yield _b.next(), !_c.done;) {
                    let item = _c.value;
                    resArray.push(item);
                }
            }
            catch (e_4_1) { e_4 = { error: e_4_1 }; }
            finally {
                try {
                    if (_c && !_c.done && (_a = _b.return)) yield _a.call(_b);
                }
                finally { if (e_4) throw e_4.error; }
            }
            ;
            assert.equal(resArray.length, 0);
        });
    });
    it("subnets beginDeleteAndWait test", function () {
        var e_5, _a;
        return __awaiter(this, void 0, void 0, function* () {
            const res = yield client.subnets.beginDeleteAndWait(resourceGroupName, virtualNetworkName, subnetName, testPollingOptions);
            const resArray = new Array();
            try {
                for (var _b = __asyncValues(client.subnets.list(resourceGroupName, virtualNetworkName)), _c; _c = yield _b.next(), !_c.done;) {
                    let item = _c.value;
                    resArray.push(item);
                }
            }
            catch (e_5_1) { e_5 = { error: e_5_1 }; }
            finally {
                try {
                    if (_c && !_c.done && (_a = _b.return)) yield _a.call(_b);
                }
                finally { if (e_5) throw e_5.error; }
            }
            ;
            assert.equal(resArray.length, 0);
        });
    });
    it("virtualNetworks beginDeleteAndWait test", function () {
        var e_6, _a;
        return __awaiter(this, void 0, void 0, function* () {
            const res = yield client.virtualNetworks.beginDeleteAndWait(resourceGroupName, virtualNetworkName, testPollingOptions);
            const resArray = new Array();
            try {
                for (var _b = __asyncValues(client.virtualNetworks.list(resourceGroupName)), _c; _c = yield _b.next(), !_c.done;) {
                    let item = _c.value;
                    resArray.push(item);
                }
            }
            catch (e_6_1) { e_6 = { error: e_6_1 }; }
            finally {
                try {
                    if (_c && !_c.done && (_a = _b.return)) yield _a.call(_b);
                }
                finally { if (e_6) throw e_6.error; }
            }
            ;
            assert.equal(resArray.length, 0);
        });
    });
});
