// @azure/arm-recoveryservicesbackup/dist-esm/src/operations/backupStatus.js
/*
 * Copyright (c) Microsoft Corporation.
 * Licensed under the MIT License.
 *
 * Code generated by Microsoft (R) AutoRest Code Generator.
 * Changes may cause incorrect behavior and will be lost if the code is regenerated.
 */
import * as coreClient from "@azure/core-client";
import * as Mappers from "../models/mappers";
import * as Parameters from "../models/parameters";
/** Class containing BackupStatus operations. */
export class BackupStatusImpl {
    /**
     * Initialize a new instance of the class BackupStatus class.
     * @param client Reference to the service client
     */
    constructor(client) {
        this.client = client;
    }
    /**
     * Get the container backup status
     * @param azureRegion Azure region to hit Api
     * @param parameters Container Backup Status Request
     * @param options The options parameters.
     */
    get(azureRegion, parameters, options) {
        return this.client.sendOperationRequest({ azureRegion, parameters, options }, getOperationSpec);
    }
}
// Operation Specifications
const serializer = coreClient.createSerializer(Mappers, /* isXml */ false);
const getOperationSpec = {
    path: "/Subscriptions/{subscriptionId}/providers/Microsoft.RecoveryServices/locations/{azureRegion}/backupStatus",
    httpMethod: "POST",
    responses: {
        200: {
            bodyMapper: Mappers.BackupStatusResponse
        }
    },
    requestBody: Parameters.parameters3,
    queryParameters: [Parameters.apiVersion],
    urlParameters: [
        Parameters.$host,
        Parameters.subscriptionId,
        Parameters.azureRegion
    ],
    headerParameters: [Parameters.accept, Parameters.contentType],
    mediaType: "json",
    serializer
};
