// @azure/arm-recoveryservicesbackup/dist-esm/src/operations/privateEndpointOperations.js
/*
 * Copyright (c) Microsoft Corporation.
 * Licensed under the MIT License.
 *
 * Code generated by Microsoft (R) AutoRest Code Generator.
 * Changes may cause incorrect behavior and will be lost if the code is regenerated.
 */
import * as coreClient from "@azure/core-client";
import * as Mappers from "../models/mappers";
import * as Parameters from "../models/parameters";
/** Class containing PrivateEndpointOperations operations. */
export class PrivateEndpointOperationsImpl {
    /**
     * Initialize a new instance of the class PrivateEndpointOperations class.
     * @param client Reference to the service client
     */
    constructor(client) {
        this.client = client;
    }
    /**
     * Gets the operation status for a private endpoint connection.
     * @param vaultName The name of the recovery services vault.
     * @param resourceGroupName The name of the resource group where the recovery services vault is
     *                          present.
     * @param privateEndpointConnectionName The name of the private endpoint connection.
     * @param operationId Operation id
     * @param options The options parameters.
     */
    getOperationStatus(vaultName, resourceGroupName, privateEndpointConnectionName, operationId, options) {
        return this.client.sendOperationRequest({
            vaultName,
            resourceGroupName,
            privateEndpointConnectionName,
            operationId,
            options
        }, getOperationStatusOperationSpec);
    }
}
// Operation Specifications
const serializer = coreClient.createSerializer(Mappers, /* isXml */ false);
const getOperationStatusOperationSpec = {
    path: "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.RecoveryServices/vaults/{vaultName}/privateEndpointConnections/{privateEndpointConnectionName}/operationsStatus/{operationId}",
    httpMethod: "GET",
    responses: {
        200: {
            bodyMapper: Mappers.OperationStatus
        },
        default: {
            bodyMapper: Mappers.NewErrorResponse
        }
    },
    queryParameters: [Parameters.apiVersion],
    urlParameters: [
        Parameters.$host,
        Parameters.vaultName,
        Parameters.resourceGroupName,
        Parameters.subscriptionId,
        Parameters.privateEndpointConnectionName,
        Parameters.operationId
    ],
    headerParameters: [Parameters.accept],
    serializer
};
