// @aws-sdk/client-codepipeline/dist-cjs/index.js
"use strict";
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __name = (target, value) => __defProp(target, "name", { value, configurable: true });
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// src/index.ts
var src_exports = {};
__export(src_exports, {
  AWSSessionCredentialsFilterSensitiveLog: () => AWSSessionCredentialsFilterSensitiveLog,
  AcknowledgeJobCommand: () => AcknowledgeJobCommand,
  AcknowledgeThirdPartyJobCommand: () => AcknowledgeThirdPartyJobCommand,
  ActionCategory: () => ActionCategory,
  ActionConfigurationPropertyType: () => ActionConfigurationPropertyType,
  ActionExecutionStatus: () => ActionExecutionStatus,
  ActionNotFoundException: () => ActionNotFoundException,
  ActionOwner: () => ActionOwner,
  ActionTypeNotFoundException: () => ActionTypeNotFoundException,
  ApprovalAlreadyCompletedException: () => ApprovalAlreadyCompletedException,
  ApprovalStatus: () => ApprovalStatus,
  ArtifactLocationType: () => ArtifactLocationType,
  ArtifactStoreType: () => ArtifactStoreType,
  BlockerType: () => BlockerType,
  CodePipeline: () => CodePipeline,
  CodePipelineClient: () => CodePipelineClient,
  CodePipelineServiceException: () => CodePipelineServiceException,
  ConcurrentModificationException: () => ConcurrentModificationException,
  ConcurrentPipelineExecutionsLimitExceededException: () => ConcurrentPipelineExecutionsLimitExceededException,
  ConflictException: () => ConflictException,
  CreateCustomActionTypeCommand: () => CreateCustomActionTypeCommand,
  CreatePipelineCommand: () => CreatePipelineCommand,
  DeleteCustomActionTypeCommand: () => DeleteCustomActionTypeCommand,
  DeletePipelineCommand: () => DeletePipelineCommand,
  DeleteWebhookCommand: () => DeleteWebhookCommand,
  DeregisterWebhookWithThirdPartyCommand: () => DeregisterWebhookWithThirdPartyCommand,
  DisableStageTransitionCommand: () => DisableStageTransitionCommand,
  DuplicatedStopRequestException: () => DuplicatedStopRequestException,
  EnableStageTransitionCommand: () => EnableStageTransitionCommand,
  EncryptionKeyType: () => EncryptionKeyType,
  ExecutionMode: () => ExecutionMode,
  ExecutorType: () => ExecutorType,
  FailureType: () => FailureType,
  GetActionTypeCommand: () => GetActionTypeCommand,
  GetJobDetailsCommand: () => GetJobDetailsCommand,
  GetJobDetailsOutputFilterSensitiveLog: () => GetJobDetailsOutputFilterSensitiveLog,
  GetPipelineCommand: () => GetPipelineCommand,
  GetPipelineExecutionCommand: () => GetPipelineExecutionCommand,
  GetPipelineStateCommand: () => GetPipelineStateCommand,
  GetThirdPartyJobDetailsCommand: () => GetThirdPartyJobDetailsCommand,
  GetThirdPartyJobDetailsOutputFilterSensitiveLog: () => GetThirdPartyJobDetailsOutputFilterSensitiveLog,
  GitPullRequestEventType: () => GitPullRequestEventType,
  InvalidActionDeclarationException: () => InvalidActionDeclarationException,
  InvalidApprovalTokenException: () => InvalidApprovalTokenException,
  InvalidArnException: () => InvalidArnException,
  InvalidBlockerDeclarationException: () => InvalidBlockerDeclarationException,
  InvalidClientTokenException: () => InvalidClientTokenException,
  InvalidJobException: () => InvalidJobException,
  InvalidJobStateException: () => InvalidJobStateException,
  InvalidNextTokenException: () => InvalidNextTokenException,
  InvalidNonceException: () => InvalidNonceException,
  InvalidStageDeclarationException: () => InvalidStageDeclarationException,
  InvalidStructureException: () => InvalidStructureException,
  InvalidTagsException: () => InvalidTagsException,
  InvalidWebhookAuthenticationParametersException: () => InvalidWebhookAuthenticationParametersException,
  InvalidWebhookFilterPatternException: () => InvalidWebhookFilterPatternException,
  JobDataFilterSensitiveLog: () => JobDataFilterSensitiveLog,
  JobDetailsFilterSensitiveLog: () => JobDetailsFilterSensitiveLog,
  JobFilterSensitiveLog: () => JobFilterSensitiveLog,
  JobNotFoundException: () => JobNotFoundException,
  JobStatus: () => JobStatus,
  LimitExceededException: () => LimitExceededException,
  ListActionExecutionsCommand: () => ListActionExecutionsCommand,
  ListActionTypesCommand: () => ListActionTypesCommand,
  ListPipelineExecutionsCommand: () => ListPipelineExecutionsCommand,
  ListPipelinesCommand: () => ListPipelinesCommand,
  ListTagsForResourceCommand: () => ListTagsForResourceCommand,
  ListWebhooksCommand: () => ListWebhooksCommand,
  NotLatestPipelineExecutionException: () => NotLatestPipelineExecutionException,
  OutputVariablesSizeExceededException: () => OutputVariablesSizeExceededException,
  PipelineExecutionNotFoundException: () => PipelineExecutionNotFoundException,
  PipelineExecutionNotStoppableException: () => PipelineExecutionNotStoppableException,
  PipelineExecutionStatus: () => PipelineExecutionStatus,
  PipelineNameInUseException: () => PipelineNameInUseException,
  PipelineNotFoundException: () => PipelineNotFoundException,
  PipelineTriggerProviderType: () => PipelineTriggerProviderType,
  PipelineType: () => PipelineType,
  PipelineVersionNotFoundException: () => PipelineVersionNotFoundException,
  PollForJobsCommand: () => PollForJobsCommand,
  PollForJobsOutputFilterSensitiveLog: () => PollForJobsOutputFilterSensitiveLog,
  PollForThirdPartyJobsCommand: () => PollForThirdPartyJobsCommand,
  PutActionRevisionCommand: () => PutActionRevisionCommand,
  PutApprovalResultCommand: () => PutApprovalResultCommand,
  PutJobFailureResultCommand: () => PutJobFailureResultCommand,
  PutJobSuccessResultCommand: () => PutJobSuccessResultCommand,
  PutThirdPartyJobFailureResultCommand: () => PutThirdPartyJobFailureResultCommand,
  PutThirdPartyJobSuccessResultCommand: () => PutThirdPartyJobSuccessResultCommand,
  PutWebhookCommand: () => PutWebhookCommand,
  RegisterWebhookWithThirdPartyCommand: () => RegisterWebhookWithThirdPartyCommand,
  RequestFailedException: () => RequestFailedException,
  ResourceNotFoundException: () => ResourceNotFoundException,
  RetryStageExecutionCommand: () => RetryStageExecutionCommand,
  SourceRevisionType: () => SourceRevisionType,
  StageExecutionStatus: () => StageExecutionStatus,
  StageNotFoundException: () => StageNotFoundException,
  StageNotRetryableException: () => StageNotRetryableException,
  StageRetryMode: () => StageRetryMode,
  StageTransitionType: () => StageTransitionType,
  StartPipelineExecutionCommand: () => StartPipelineExecutionCommand,
  StartTimeRange: () => StartTimeRange,
  StopPipelineExecutionCommand: () => StopPipelineExecutionCommand,
  TagResourceCommand: () => TagResourceCommand,
  ThirdPartyJobDataFilterSensitiveLog: () => ThirdPartyJobDataFilterSensitiveLog,
  ThirdPartyJobDetailsFilterSensitiveLog: () => ThirdPartyJobDetailsFilterSensitiveLog,
  TooManyTagsException: () => TooManyTagsException,
  TriggerType: () => TriggerType,
  UntagResourceCommand: () => UntagResourceCommand,
  UpdateActionTypeCommand: () => UpdateActionTypeCommand,
  UpdatePipelineCommand: () => UpdatePipelineCommand,
  ValidationException: () => ValidationException,
  WebhookAuthenticationType: () => WebhookAuthenticationType,
  WebhookNotFoundException: () => WebhookNotFoundException,
  __Client: () => import_smithy_client.Client,
  paginateListActionExecutions: () => paginateListActionExecutions,
  paginateListActionTypes: () => paginateListActionTypes,
  paginateListPipelineExecutions: () => paginateListPipelineExecutions,
  paginateListPipelines: () => paginateListPipelines,
  paginateListTagsForResource: () => paginateListTagsForResource,
  paginateListWebhooks: () => paginateListWebhooks
});
module.exports = __toCommonJS(src_exports);

// src/CodePipelineClient.ts
var import_middleware_host_header = require("@aws-sdk/middleware-host-header");
var import_middleware_logger = require("@aws-sdk/middleware-logger");
var import_middleware_recursion_detection = require("@aws-sdk/middleware-recursion-detection");
var import_middleware_user_agent = require("@aws-sdk/middleware-user-agent");
var import_config_resolver = require("@smithy/config-resolver");
var import_core = require("@smithy/core");
var import_middleware_content_length = require("@smithy/middleware-content-length");
var import_middleware_endpoint = require("@smithy/middleware-endpoint");
var import_middleware_retry = require("@smithy/middleware-retry");

var import_httpAuthSchemeProvider = require("./auth/httpAuthSchemeProvider");

// src/endpoint/EndpointParameters.ts
var resolveClientEndpointParameters = /* @__PURE__ */ __name((options) => {
  return {
    ...options,
    useDualstackEndpoint: options.useDualstackEndpoint ?? false,
    useFipsEndpoint: options.useFipsEndpoint ?? false,
    defaultSigningName: "codepipeline"
  };
}, "resolveClientEndpointParameters");
var commonParams = {
  UseFIPS: { type: "builtInParams", name: "useFipsEndpoint" },
  Endpoint: { type: "builtInParams", name: "endpoint" },
  Region: { type: "builtInParams", name: "region" },
  UseDualStack: { type: "builtInParams", name: "useDualstackEndpoint" }
};

// src/CodePipelineClient.ts
var import_runtimeConfig = require("././runtimeConfig");

// src/runtimeExtensions.ts
var import_region_config_resolver = require("@aws-sdk/region-config-resolver");
var import_protocol_http = require("@smithy/protocol-http");
var import_smithy_client = require("@smithy/smithy-client");

// src/auth/httpAuthExtensionConfiguration.ts
var getHttpAuthExtensionConfiguration = /* @__PURE__ */ __name((runtimeConfig) => {
  const _httpAuthSchemes = runtimeConfig.httpAuthSchemes;
  let _httpAuthSchemeProvider = runtimeConfig.httpAuthSchemeProvider;
  let _credentials = runtimeConfig.credentials;
  return {
    setHttpAuthScheme(httpAuthScheme) {
      const index = _httpAuthSchemes.findIndex((scheme) => scheme.schemeId === httpAuthScheme.schemeId);
      if (index === -1) {
        _httpAuthSchemes.push(httpAuthScheme);
      } else {
        _httpAuthSchemes.splice(index, 1, httpAuthScheme);
      }
    },
    httpAuthSchemes() {
      return _httpAuthSchemes;
    },
    setHttpAuthSchemeProvider(httpAuthSchemeProvider) {
      _httpAuthSchemeProvider = httpAuthSchemeProvider;
    },
    httpAuthSchemeProvider() {
      return _httpAuthSchemeProvider;
    },
    setCredentials(credentials) {
      _credentials = credentials;
    },
    credentials() {
      return _credentials;
    }
  };
}, "getHttpAuthExtensionConfiguration");
var resolveHttpAuthRuntimeConfig = /* @__PURE__ */ __name((config) => {
  return {
    httpAuthSchemes: config.httpAuthSchemes(),
    httpAuthSchemeProvider: config.httpAuthSchemeProvider(),
    credentials: config.credentials()
  };
}, "resolveHttpAuthRuntimeConfig");

// src/runtimeExtensions.ts
var asPartial = /* @__PURE__ */ __name((t) => t, "asPartial");
var resolveRuntimeExtensions = /* @__PURE__ */ __name((runtimeConfig, extensions) => {
  const extensionConfiguration = {
    ...asPartial((0, import_region_config_resolver.getAwsRegionExtensionConfiguration)(runtimeConfig)),
    ...asPartial((0, import_smithy_client.getDefaultExtensionConfiguration)(runtimeConfig)),
    ...asPartial((0, import_protocol_http.getHttpHandlerExtensionConfiguration)(runtimeConfig)),
    ...asPartial(getHttpAuthExtensionConfiguration(runtimeConfig))
  };
  extensions.forEach((extension) => extension.configure(extensionConfiguration));
  return {
    ...runtimeConfig,
    ...(0, import_region_config_resolver.resolveAwsRegionExtensionConfiguration)(extensionConfiguration),
    ...(0, import_smithy_client.resolveDefaultRuntimeConfig)(extensionConfiguration),
    ...(0, import_protocol_http.resolveHttpHandlerRuntimeConfig)(extensionConfiguration),
    ...resolveHttpAuthRuntimeConfig(extensionConfiguration)
  };
}, "resolveRuntimeExtensions");

// src/CodePipelineClient.ts
var _CodePipelineClient = class _CodePipelineClient extends import_smithy_client.Client {
  constructor(...[configuration]) {
    const _config_0 = (0, import_runtimeConfig.getRuntimeConfig)(configuration || {});
    const _config_1 = resolveClientEndpointParameters(_config_0);
    const _config_2 = (0, import_config_resolver.resolveRegionConfig)(_config_1);
    const _config_3 = (0, import_middleware_endpoint.resolveEndpointConfig)(_config_2);
    const _config_4 = (0, import_middleware_retry.resolveRetryConfig)(_config_3);
    const _config_5 = (0, import_middleware_host_header.resolveHostHeaderConfig)(_config_4);
    const _config_6 = (0, import_middleware_user_agent.resolveUserAgentConfig)(_config_5);
    const _config_7 = (0, import_httpAuthSchemeProvider.resolveHttpAuthSchemeConfig)(_config_6);
    const _config_8 = resolveRuntimeExtensions(_config_7, (configuration == null ? void 0 : configuration.extensions) || []);
    super(_config_8);
    this.config = _config_8;
    this.middlewareStack.use((0, import_middleware_retry.getRetryPlugin)(this.config));
    this.middlewareStack.use((0, import_middleware_content_length.getContentLengthPlugin)(this.config));
    this.middlewareStack.use((0, import_middleware_host_header.getHostHeaderPlugin)(this.config));
    this.middlewareStack.use((0, import_middleware_logger.getLoggerPlugin)(this.config));
    this.middlewareStack.use((0, import_middleware_recursion_detection.getRecursionDetectionPlugin)(this.config));
    this.middlewareStack.use((0, import_middleware_user_agent.getUserAgentPlugin)(this.config));
    this.middlewareStack.use(
      (0, import_core.getHttpAuthSchemeEndpointRuleSetPlugin)(this.config, {
        httpAuthSchemeParametersProvider: this.getDefaultHttpAuthSchemeParametersProvider(),
        identityProviderConfigProvider: this.getIdentityProviderConfigProvider()
      })
    );
    this.middlewareStack.use((0, import_core.getHttpSigningPlugin)(this.config));
  }
  /**
   * Destroy underlying resources, like sockets. It's usually not necessary to do this.
   * However in Node.js, it's best to explicitly shut down the client's agent when it is no longer needed.
   * Otherwise, sockets might stay open for quite a long time before the server terminates them.
   */
  destroy() {
    super.destroy();
  }
  getDefaultHttpAuthSchemeParametersProvider() {
    return import_httpAuthSchemeProvider.defaultCodePipelineHttpAuthSchemeParametersProvider;
  }
  getIdentityProviderConfigProvider() {
    return async (config) => new import_core.DefaultIdentityProviderConfig({
      "aws.auth#sigv4": config.credentials
    });
  }
};
__name(_CodePipelineClient, "CodePipelineClient");
var CodePipelineClient = _CodePipelineClient;

// src/CodePipeline.ts


// src/commands/AcknowledgeJobCommand.ts

var import_middleware_serde = require("@smithy/middleware-serde");

var import_types = require("@smithy/types");

// src/protocols/Aws_json1_1.ts


var import_uuid = require("uuid");

// src/models/CodePipelineServiceException.ts

var _CodePipelineServiceException = class _CodePipelineServiceException extends import_smithy_client.ServiceException {
  /**
   * @internal
   */
  constructor(options) {
    super(options);
    Object.setPrototypeOf(this, _CodePipelineServiceException.prototype);
  }
};
__name(_CodePipelineServiceException, "CodePipelineServiceException");
var CodePipelineServiceException = _CodePipelineServiceException;

// src/models/models_0.ts

var JobStatus = {
  Created: "Created",
  Dispatched: "Dispatched",
  Failed: "Failed",
  InProgress: "InProgress",
  Queued: "Queued",
  Succeeded: "Succeeded",
  TimedOut: "TimedOut"
};
var _InvalidNonceException = class _InvalidNonceException extends CodePipelineServiceException {
  /**
   * @internal
   */
  constructor(opts) {
    super({
      name: "InvalidNonceException",
      $fault: "client",
      ...opts
    });
    this.name = "InvalidNonceException";
    this.$fault = "client";
    Object.setPrototypeOf(this, _InvalidNonceException.prototype);
  }
};
__name(_InvalidNonceException, "InvalidNonceException");
var InvalidNonceException = _InvalidNonceException;
var _JobNotFoundException = class _JobNotFoundException extends CodePipelineServiceException {
  /**
   * @internal
   */
  constructor(opts) {
    super({
      name: "JobNotFoundException",
      $fault: "client",
      ...opts
    });
    this.name = "JobNotFoundException";
    this.$fault = "client";
    Object.setPrototypeOf(this, _JobNotFoundException.prototype);
  }
};
__name(_JobNotFoundException, "JobNotFoundException");
var JobNotFoundException = _JobNotFoundException;
var _ValidationException = class _ValidationException extends CodePipelineServiceException {
  /**
   * @internal
   */
  constructor(opts) {
    super({
      name: "ValidationException",
      $fault: "client",
      ...opts
    });
    this.name = "ValidationException";
    this.$fault = "client";
    Object.setPrototypeOf(this, _ValidationException.prototype);
  }
};
__name(_ValidationException, "ValidationException");
var ValidationException = _ValidationException;
var _InvalidClientTokenException = class _InvalidClientTokenException extends CodePipelineServiceException {
  /**
   * @internal
   */
  constructor(opts) {
    super({
      name: "InvalidClientTokenException",
      $fault: "client",
      ...opts
    });
    this.name = "InvalidClientTokenException";
    this.$fault = "client";
    Object.setPrototypeOf(this, _InvalidClientTokenException.prototype);
  }
};
__name(_InvalidClientTokenException, "InvalidClientTokenException");
var InvalidClientTokenException = _InvalidClientTokenException;
var ActionCategory = {
  Approval: "Approval",
  Build: "Build",
  Deploy: "Deploy",
  Invoke: "Invoke",
  Source: "Source",
  Test: "Test"
};
var ActionConfigurationPropertyType = {
  Boolean: "Boolean",
  Number: "Number",
  String: "String"
};
var ActionOwner = {
  AWS: "AWS",
  Custom: "Custom",
  ThirdParty: "ThirdParty"
};
var ActionExecutionStatus = {
  Abandoned: "Abandoned",
  Failed: "Failed",
  InProgress: "InProgress",
  Succeeded: "Succeeded"
};
var StartTimeRange = {
  All: "All",
  Latest: "Latest"
};
var _ActionNotFoundException = class _ActionNotFoundException extends CodePipelineServiceException {
  /**
   * @internal
   */
  constructor(opts) {
    super({
      name: "ActionNotFoundException",
      $fault: "client",
      ...opts
    });
    this.name = "ActionNotFoundException";
    this.$fault = "client";
    Object.setPrototypeOf(this, _ActionNotFoundException.prototype);
  }
};
__name(_ActionNotFoundException, "ActionNotFoundException");
var ActionNotFoundException = _ActionNotFoundException;
var ExecutorType = {
  JobWorker: "JobWorker",
  Lambda: "Lambda"
};
var _ActionTypeNotFoundException = class _ActionTypeNotFoundException extends CodePipelineServiceException {
  /**
   * @internal
   */
  constructor(opts) {
    super({
      name: "ActionTypeNotFoundException",
      $fault: "client",
      ...opts
    });
    this.name = "ActionTypeNotFoundException";
    this.$fault = "client";
    Object.setPrototypeOf(this, _ActionTypeNotFoundException.prototype);
  }
};
__name(_ActionTypeNotFoundException, "ActionTypeNotFoundException");
var ActionTypeNotFoundException = _ActionTypeNotFoundException;
var _ApprovalAlreadyCompletedException = class _ApprovalAlreadyCompletedException extends CodePipelineServiceException {
  /**
   * @internal
   */
  constructor(opts) {
    super({
      name: "ApprovalAlreadyCompletedException",
      $fault: "client",
      ...opts
    });
    this.name = "ApprovalAlreadyCompletedException";
    this.$fault = "client";
    Object.setPrototypeOf(this, _ApprovalAlreadyCompletedException.prototype);
  }
};
__name(_ApprovalAlreadyCompletedException, "ApprovalAlreadyCompletedException");
var ApprovalAlreadyCompletedException = _ApprovalAlreadyCompletedException;
var ApprovalStatus = {
  Approved: "Approved",
  Rejected: "Rejected"
};
var ArtifactLocationType = {
  S3: "S3"
};
var EncryptionKeyType = {
  KMS: "KMS"
};
var ArtifactStoreType = {
  S3: "S3"
};
var BlockerType = {
  Schedule: "Schedule"
};
var _ConcurrentModificationException = class _ConcurrentModificationException extends CodePipelineServiceException {
  /**
   * @internal
   */
  constructor(opts) {
    super({
      name: "ConcurrentModificationException",
      $fault: "client",
      ...opts
    });
    this.name = "ConcurrentModificationException";
    this.$fault = "client";
    Object.setPrototypeOf(this, _ConcurrentModificationException.prototype);
  }
};
__name(_ConcurrentModificationException, "ConcurrentModificationException");
var ConcurrentModificationException = _ConcurrentModificationException;
var _InvalidTagsException = class _InvalidTagsException extends CodePipelineServiceException {
  /**
   * @internal
   */
  constructor(opts) {
    super({
      name: "InvalidTagsException",
      $fault: "client",
      ...opts
    });
    this.name = "InvalidTagsException";
    this.$fault = "client";
    Object.setPrototypeOf(this, _InvalidTagsException.prototype);
  }
};
__name(_InvalidTagsException, "InvalidTagsException");
var InvalidTagsException = _InvalidTagsException;
var _LimitExceededException = class _LimitExceededException extends CodePipelineServiceException {
  /**
   * @internal
   */
  constructor(opts) {
    super({
      name: "LimitExceededException",
      $fault: "client",
      ...opts
    });
    this.name = "LimitExceededException";
    this.$fault = "client";
    Object.setPrototypeOf(this, _LimitExceededException.prototype);
  }
};
__name(_LimitExceededException, "LimitExceededException");
var LimitExceededException = _LimitExceededException;
var _TooManyTagsException = class _TooManyTagsException extends CodePipelineServiceException {
  /**
   * @internal
   */
  constructor(opts) {
    super({
      name: "TooManyTagsException",
      $fault: "client",
      ...opts
    });
    this.name = "TooManyTagsException";
    this.$fault = "client";
    Object.setPrototypeOf(this, _TooManyTagsException.prototype);
  }
};
__name(_TooManyTagsException, "TooManyTagsException");
var TooManyTagsException = _TooManyTagsException;
var ExecutionMode = {
  PARALLEL: "PARALLEL",
  QUEUED: "QUEUED",
  SUPERSEDED: "SUPERSEDED"
};
var PipelineType = {
  V1: "V1",
  V2: "V2"
};
var GitPullRequestEventType = {
  CLOSED: "CLOSED",
  OPEN: "OPEN",
  UPDATED: "UPDATED"
};
var PipelineTriggerProviderType = {
  CodeStarSourceConnection: "CodeStarSourceConnection"
};
var _InvalidActionDeclarationException = class _InvalidActionDeclarationException extends CodePipelineServiceException {
  /**
   * @internal
   */
  constructor(opts) {
    super({
      name: "InvalidActionDeclarationException",
      $fault: "client",
      ...opts
    });
    this.name = "InvalidActionDeclarationException";
    this.$fault = "client";
    Object.setPrototypeOf(this, _InvalidActionDeclarationException.prototype);
  }
};
__name(_InvalidActionDeclarationException, "InvalidActionDeclarationException");
var InvalidActionDeclarationException = _InvalidActionDeclarationException;
var _InvalidBlockerDeclarationException = class _InvalidBlockerDeclarationException extends CodePipelineServiceException {
  /**
   * @internal
   */
  constructor(opts) {
    super({
      name: "InvalidBlockerDeclarationException",
      $fault: "client",
      ...opts
    });
    this.name = "InvalidBlockerDeclarationException";
    this.$fault = "client";
    Object.setPrototypeOf(this, _InvalidBlockerDeclarationException.prototype);
  }
};
__name(_InvalidBlockerDeclarationException, "InvalidBlockerDeclarationException");
var InvalidBlockerDeclarationException = _InvalidBlockerDeclarationException;
var _InvalidStageDeclarationException = class _InvalidStageDeclarationException extends CodePipelineServiceException {
  /**
   * @internal
   */
  constructor(opts) {
    super({
      name: "InvalidStageDeclarationException",
      $fault: "client",
      ...opts
    });
    this.name = "InvalidStageDeclarationException";
    this.$fault = "client";
    Object.setPrototypeOf(this, _InvalidStageDeclarationException.prototype);
  }
};
__name(_InvalidStageDeclarationException, "InvalidStageDeclarationException");
var InvalidStageDeclarationException = _InvalidStageDeclarationException;
var _InvalidStructureException = class _InvalidStructureException extends CodePipelineServiceException {
  /**
   * @internal
   */
  constructor(opts) {
    super({
      name: "InvalidStructureException",
      $fault: "client",
      ...opts
    });
    this.name = "InvalidStructureException";
    this.$fault = "client";
    Object.setPrototypeOf(this, _InvalidStructureException.prototype);
  }
};
__name(_InvalidStructureException, "InvalidStructureException");
var InvalidStructureException = _InvalidStructureException;
var _PipelineNameInUseException = class _PipelineNameInUseException extends CodePipelineServiceException {
  /**
   * @internal
   */
  constructor(opts) {
    super({
      name: "PipelineNameInUseException",
      $fault: "client",
      ...opts
    });
    this.name = "PipelineNameInUseException";
    this.$fault = "client";
    Object.setPrototypeOf(this, _PipelineNameInUseException.prototype);
  }
};
__name(_PipelineNameInUseException, "PipelineNameInUseException");
var PipelineNameInUseException = _PipelineNameInUseException;
var _WebhookNotFoundException = class _WebhookNotFoundException extends CodePipelineServiceException {
  /**
   * @internal
   */
  constructor(opts) {
    super({
      name: "WebhookNotFoundException",
      $fault: "client",
      ...opts
    });
    this.name = "WebhookNotFoundException";
    this.$fault = "client";
    Object.setPrototypeOf(this, _WebhookNotFoundException.prototype);
  }
};
__name(_WebhookNotFoundException, "WebhookNotFoundException");
var WebhookNotFoundException = _WebhookNotFoundException;
var StageTransitionType = {
  Inbound: "Inbound",
  Outbound: "Outbound"
};
var _PipelineNotFoundException = class _PipelineNotFoundException extends CodePipelineServiceException {
  /**
   * @internal
   */
  constructor(opts) {
    super({
      name: "PipelineNotFoundException",
      $fault: "client",
      ...opts
    });
    this.name = "PipelineNotFoundException";
    this.$fault = "client";
    Object.setPrototypeOf(this, _PipelineNotFoundException.prototype);
  }
};
__name(_PipelineNotFoundException, "PipelineNotFoundException");
var PipelineNotFoundException = _PipelineNotFoundException;
var _StageNotFoundException = class _StageNotFoundException extends CodePipelineServiceException {
  /**
   * @internal
   */
  constructor(opts) {
    super({
      name: "StageNotFoundException",
      $fault: "client",
      ...opts
    });
    this.name = "StageNotFoundException";
    this.$fault = "client";
    Object.setPrototypeOf(this, _StageNotFoundException.prototype);
  }
};
__name(_StageNotFoundException, "StageNotFoundException");
var StageNotFoundException = _StageNotFoundException;
var _PipelineVersionNotFoundException = class _PipelineVersionNotFoundException extends CodePipelineServiceException {
  /**
   * @internal
   */
  constructor(opts) {
    super({
      name: "PipelineVersionNotFoundException",
      $fault: "client",
      ...opts
    });
    this.name = "PipelineVersionNotFoundException";
    this.$fault = "client";
    Object.setPrototypeOf(this, _PipelineVersionNotFoundException.prototype);
  }
};
__name(_PipelineVersionNotFoundException, "PipelineVersionNotFoundException");
var PipelineVersionNotFoundException = _PipelineVersionNotFoundException;
var PipelineExecutionStatus = {
  Cancelled: "Cancelled",
  Failed: "Failed",
  InProgress: "InProgress",
  Stopped: "Stopped",
  Stopping: "Stopping",
  Succeeded: "Succeeded",
  Superseded: "Superseded"
};
var TriggerType = {
  CloudWatchEvent: "CloudWatchEvent",
  CreatePipeline: "CreatePipeline",
  PollForSourceChanges: "PollForSourceChanges",
  PutActionRevision: "PutActionRevision",
  StartPipelineExecution: "StartPipelineExecution",
  Webhook: "Webhook",
  WebhookV2: "WebhookV2"
};
var _PipelineExecutionNotFoundException = class _PipelineExecutionNotFoundException extends CodePipelineServiceException {
  /**
   * @internal
   */
  constructor(opts) {
    super({
      name: "PipelineExecutionNotFoundException",
      $fault: "client",
      ...opts
    });
    this.name = "PipelineExecutionNotFoundException";
    this.$fault = "client";
    Object.setPrototypeOf(this, _PipelineExecutionNotFoundException.prototype);
  }
};
__name(_PipelineExecutionNotFoundException, "PipelineExecutionNotFoundException");
var PipelineExecutionNotFoundException = _PipelineExecutionNotFoundException;
var StageExecutionStatus = {
  Cancelled: "Cancelled",
  Failed: "Failed",
  InProgress: "InProgress",
  Stopped: "Stopped",
  Stopping: "Stopping",
  Succeeded: "Succeeded"
};
var _InvalidJobException = class _InvalidJobException extends CodePipelineServiceException {
  /**
   * @internal
   */
  constructor(opts) {
    super({
      name: "InvalidJobException",
      $fault: "client",
      ...opts
    });
    this.name = "InvalidJobException";
    this.$fault = "client";
    Object.setPrototypeOf(this, _InvalidJobException.prototype);
  }
};
__name(_InvalidJobException, "InvalidJobException");
var InvalidJobException = _InvalidJobException;
var _InvalidNextTokenException = class _InvalidNextTokenException extends CodePipelineServiceException {
  /**
   * @internal
   */
  constructor(opts) {
    super({
      name: "InvalidNextTokenException",
      $fault: "client",
      ...opts
    });
    this.name = "InvalidNextTokenException";
    this.$fault = "client";
    Object.setPrototypeOf(this, _InvalidNextTokenException.prototype);
  }
};
__name(_InvalidNextTokenException, "InvalidNextTokenException");
var InvalidNextTokenException = _InvalidNextTokenException;
var _InvalidArnException = class _InvalidArnException extends CodePipelineServiceException {
  /**
   * @internal
   */
  constructor(opts) {
    super({
      name: "InvalidArnException",
      $fault: "client",
      ...opts
    });
    this.name = "InvalidArnException";
    this.$fault = "client";
    Object.setPrototypeOf(this, _InvalidArnException.prototype);
  }
};
__name(_InvalidArnException, "InvalidArnException");
var InvalidArnException = _InvalidArnException;
var _ResourceNotFoundException = class _ResourceNotFoundException extends CodePipelineServiceException {
  /**
   * @internal
   */
  constructor(opts) {
    super({
      name: "ResourceNotFoundException",
      $fault: "client",
      ...opts
    });
    this.name = "ResourceNotFoundException";
    this.$fault = "client";
    Object.setPrototypeOf(this, _ResourceNotFoundException.prototype);
  }
};
__name(_ResourceNotFoundException, "ResourceNotFoundException");
var ResourceNotFoundException = _ResourceNotFoundException;
var WebhookAuthenticationType = {
  GITHUB_HMAC: "GITHUB_HMAC",
  IP: "IP",
  UNAUTHENTICATED: "UNAUTHENTICATED"
};
var _InvalidApprovalTokenException = class _InvalidApprovalTokenException extends CodePipelineServiceException {
  /**
   * @internal
   */
  constructor(opts) {
    super({
      name: "InvalidApprovalTokenException",
      $fault: "client",
      ...opts
    });
    this.name = "InvalidApprovalTokenException";
    this.$fault = "client";
    Object.setPrototypeOf(this, _InvalidApprovalTokenException.prototype);
  }
};
__name(_InvalidApprovalTokenException, "InvalidApprovalTokenException");
var InvalidApprovalTokenException = _InvalidApprovalTokenException;
var _InvalidJobStateException = class _InvalidJobStateException extends CodePipelineServiceException {
  /**
   * @internal
   */
  constructor(opts) {
    super({
      name: "InvalidJobStateException",
      $fault: "client",
      ...opts
    });
    this.name = "InvalidJobStateException";
    this.$fault = "client";
    Object.setPrototypeOf(this, _InvalidJobStateException.prototype);
  }
};
__name(_InvalidJobStateException, "InvalidJobStateException");
var InvalidJobStateException = _InvalidJobStateException;
var FailureType = {
  ConfigurationError: "ConfigurationError",
  JobFailed: "JobFailed",
  PermissionError: "PermissionError",
  RevisionOutOfSync: "RevisionOutOfSync",
  RevisionUnavailable: "RevisionUnavailable",
  SystemUnavailable: "SystemUnavailable"
};
var _OutputVariablesSizeExceededException = class _OutputVariablesSizeExceededException extends CodePipelineServiceException {
  /**
   * @internal
   */
  constructor(opts) {
    super({
      name: "OutputVariablesSizeExceededException",
      $fault: "client",
      ...opts
    });
    this.name = "OutputVariablesSizeExceededException";
    this.$fault = "client";
    Object.setPrototypeOf(this, _OutputVariablesSizeExceededException.prototype);
  }
};
__name(_OutputVariablesSizeExceededException, "OutputVariablesSizeExceededException");
var OutputVariablesSizeExceededException = _OutputVariablesSizeExceededException;
var _InvalidWebhookAuthenticationParametersException = class _InvalidWebhookAuthenticationParametersException extends CodePipelineServiceException {
  /**
   * @internal
   */
  constructor(opts) {
    super({
      name: "InvalidWebhookAuthenticationParametersException",
      $fault: "client",
      ...opts
    });
    this.name = "InvalidWebhookAuthenticationParametersException";
    this.$fault = "client";
    Object.setPrototypeOf(this, _InvalidWebhookAuthenticationParametersException.prototype);
  }
};
__name(_InvalidWebhookAuthenticationParametersException, "InvalidWebhookAuthenticationParametersException");
var InvalidWebhookAuthenticationParametersException = _InvalidWebhookAuthenticationParametersException;
var _InvalidWebhookFilterPatternException = class _InvalidWebhookFilterPatternException extends CodePipelineServiceException {
  /**
   * @internal
   */
  constructor(opts) {
    super({
      name: "InvalidWebhookFilterPatternException",
      $fault: "client",
      ...opts
    });
    this.name = "InvalidWebhookFilterPatternException";
    this.$fault = "client";
    Object.setPrototypeOf(this, _InvalidWebhookFilterPatternException.prototype);
  }
};
__name(_InvalidWebhookFilterPatternException, "InvalidWebhookFilterPatternException");
var InvalidWebhookFilterPatternException = _InvalidWebhookFilterPatternException;
var _ConflictException = class _ConflictException extends CodePipelineServiceException {
  /**
   * @internal
   */
  constructor(opts) {
    super({
      name: "ConflictException",
      $fault: "client",
      ...opts
    });
    this.name = "ConflictException";
    this.$fault = "client";
    Object.setPrototypeOf(this, _ConflictException.prototype);
  }
};
__name(_ConflictException, "ConflictException");
var ConflictException = _ConflictException;
var _NotLatestPipelineExecutionException = class _NotLatestPipelineExecutionException extends CodePipelineServiceException {
  /**
   * @internal
   */
  constructor(opts) {
    super({
      name: "NotLatestPipelineExecutionException",
      $fault: "client",
      ...opts
    });
    this.name = "NotLatestPipelineExecutionException";
    this.$fault = "client";
    Object.setPrototypeOf(this, _NotLatestPipelineExecutionException.prototype);
  }
};
__name(_NotLatestPipelineExecutionException, "NotLatestPipelineExecutionException");
var NotLatestPipelineExecutionException = _NotLatestPipelineExecutionException;
var StageRetryMode = {
  ALL_ACTIONS: "ALL_ACTIONS",
  FAILED_ACTIONS: "FAILED_ACTIONS"
};
var _StageNotRetryableException = class _StageNotRetryableException extends CodePipelineServiceException {
  /**
   * @internal
   */
  constructor(opts) {
    super({
      name: "StageNotRetryableException",
      $fault: "client",
      ...opts
    });
    this.name = "StageNotRetryableException";
    this.$fault = "client";
    Object.setPrototypeOf(this, _StageNotRetryableException.prototype);
  }
};
__name(_StageNotRetryableException, "StageNotRetryableException");
var StageNotRetryableException = _StageNotRetryableException;
var _ConcurrentPipelineExecutionsLimitExceededException = class _ConcurrentPipelineExecutionsLimitExceededException extends CodePipelineServiceException {
  /**
   * @internal
   */
  constructor(opts) {
    super({
      name: "ConcurrentPipelineExecutionsLimitExceededException",
      $fault: "client",
      ...opts
    });
    this.name = "ConcurrentPipelineExecutionsLimitExceededException";
    this.$fault = "client";
    Object.setPrototypeOf(this, _ConcurrentPipelineExecutionsLimitExceededException.prototype);
  }
};
__name(_ConcurrentPipelineExecutionsLimitExceededException, "ConcurrentPipelineExecutionsLimitExceededException");
var ConcurrentPipelineExecutionsLimitExceededException = _ConcurrentPipelineExecutionsLimitExceededException;
var SourceRevisionType = {
  COMMIT_ID: "COMMIT_ID",
  IMAGE_DIGEST: "IMAGE_DIGEST",
  S3_OBJECT_VERSION_ID: "S3_OBJECT_VERSION_ID"
};
var _DuplicatedStopRequestException = class _DuplicatedStopRequestException extends CodePipelineServiceException {
  /**
   * @internal
   */
  constructor(opts) {
    super({
      name: "DuplicatedStopRequestException",
      $fault: "client",
      ...opts
    });
    this.name = "DuplicatedStopRequestException";
    this.$fault = "client";
    Object.setPrototypeOf(this, _DuplicatedStopRequestException.prototype);
  }
};
__name(_DuplicatedStopRequestException, "DuplicatedStopRequestException");
var DuplicatedStopRequestException = _DuplicatedStopRequestException;
var _PipelineExecutionNotStoppableException = class _PipelineExecutionNotStoppableException extends CodePipelineServiceException {
  /**
   * @internal
   */
  constructor(opts) {
    super({
      name: "PipelineExecutionNotStoppableException",
      $fault: "client",
      ...opts
    });
    this.name = "PipelineExecutionNotStoppableException";
    this.$fault = "client";
    Object.setPrototypeOf(this, _PipelineExecutionNotStoppableException.prototype);
  }
};
__name(_PipelineExecutionNotStoppableException, "PipelineExecutionNotStoppableException");
var PipelineExecutionNotStoppableException = _PipelineExecutionNotStoppableException;
var _RequestFailedException = class _RequestFailedException extends CodePipelineServiceException {
  /**
   * @internal
   */
  constructor(opts) {
    super({
      name: "RequestFailedException",
      $fault: "client",
      ...opts
    });
    this.name = "RequestFailedException";
    this.$fault = "client";
    Object.setPrototypeOf(this, _RequestFailedException.prototype);
  }
};
__name(_RequestFailedException, "RequestFailedException");
var RequestFailedException = _RequestFailedException;
var AWSSessionCredentialsFilterSensitiveLog = /* @__PURE__ */ __name((obj) => ({
  ...obj,
  ...obj.accessKeyId && { accessKeyId: import_smithy_client.SENSITIVE_STRING },
  ...obj.secretAccessKey && { secretAccessKey: import_smithy_client.SENSITIVE_STRING },
  ...obj.sessionToken && { sessionToken: import_smithy_client.SENSITIVE_STRING }
}), "AWSSessionCredentialsFilterSensitiveLog");
var JobDataFilterSensitiveLog = /* @__PURE__ */ __name((obj) => ({
  ...obj,
  ...obj.artifactCredentials && { artifactCredentials: import_smithy_client.SENSITIVE_STRING }
}), "JobDataFilterSensitiveLog");
var JobDetailsFilterSensitiveLog = /* @__PURE__ */ __name((obj) => ({
  ...obj,
  ...obj.data && { data: JobDataFilterSensitiveLog(obj.data) }
}), "JobDetailsFilterSensitiveLog");
var GetJobDetailsOutputFilterSensitiveLog = /* @__PURE__ */ __name((obj) => ({
  ...obj,
  ...obj.jobDetails && { jobDetails: JobDetailsFilterSensitiveLog(obj.jobDetails) }
}), "GetJobDetailsOutputFilterSensitiveLog");
var ThirdPartyJobDataFilterSensitiveLog = /* @__PURE__ */ __name((obj) => ({
  ...obj,
  ...obj.artifactCredentials && { artifactCredentials: import_smithy_client.SENSITIVE_STRING }
}), "ThirdPartyJobDataFilterSensitiveLog");
var ThirdPartyJobDetailsFilterSensitiveLog = /* @__PURE__ */ __name((obj) => ({
  ...obj,
  ...obj.data && { data: ThirdPartyJobDataFilterSensitiveLog(obj.data) }
}), "ThirdPartyJobDetailsFilterSensitiveLog");
var GetThirdPartyJobDetailsOutputFilterSensitiveLog = /* @__PURE__ */ __name((obj) => ({
  ...obj,
  ...obj.jobDetails && { jobDetails: ThirdPartyJobDetailsFilterSensitiveLog(obj.jobDetails) }
}), "GetThirdPartyJobDetailsOutputFilterSensitiveLog");
var JobFilterSensitiveLog = /* @__PURE__ */ __name((obj) => ({
  ...obj,
  ...obj.data && { data: JobDataFilterSensitiveLog(obj.data) }
}), "JobFilterSensitiveLog");
var PollForJobsOutputFilterSensitiveLog = /* @__PURE__ */ __name((obj) => ({
  ...obj,
  ...obj.jobs && { jobs: obj.jobs.map((item) => JobFilterSensitiveLog(item)) }
}), "PollForJobsOutputFilterSensitiveLog");

// src/protocols/Aws_json1_1.ts
var se_AcknowledgeJobCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("AcknowledgeJob");
  let body;
  body = JSON.stringify((0, import_smithy_client._json)(input));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_AcknowledgeJobCommand");
var se_AcknowledgeThirdPartyJobCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("AcknowledgeThirdPartyJob");
  let body;
  body = JSON.stringify((0, import_smithy_client._json)(input));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_AcknowledgeThirdPartyJobCommand");
var se_CreateCustomActionTypeCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("CreateCustomActionType");
  let body;
  body = JSON.stringify((0, import_smithy_client._json)(input));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_CreateCustomActionTypeCommand");
var se_CreatePipelineCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("CreatePipeline");
  let body;
  body = JSON.stringify((0, import_smithy_client._json)(input));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_CreatePipelineCommand");
var se_DeleteCustomActionTypeCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("DeleteCustomActionType");
  let body;
  body = JSON.stringify((0, import_smithy_client._json)(input));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_DeleteCustomActionTypeCommand");
var se_DeletePipelineCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("DeletePipeline");
  let body;
  body = JSON.stringify((0, import_smithy_client._json)(input));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_DeletePipelineCommand");
var se_DeleteWebhookCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("DeleteWebhook");
  let body;
  body = JSON.stringify((0, import_smithy_client._json)(input));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_DeleteWebhookCommand");
var se_DeregisterWebhookWithThirdPartyCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("DeregisterWebhookWithThirdParty");
  let body;
  body = JSON.stringify((0, import_smithy_client._json)(input));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_DeregisterWebhookWithThirdPartyCommand");
var se_DisableStageTransitionCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("DisableStageTransition");
  let body;
  body = JSON.stringify((0, import_smithy_client._json)(input));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_DisableStageTransitionCommand");
var se_EnableStageTransitionCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("EnableStageTransition");
  let body;
  body = JSON.stringify((0, import_smithy_client._json)(input));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_EnableStageTransitionCommand");
var se_GetActionTypeCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("GetActionType");
  let body;
  body = JSON.stringify((0, import_smithy_client._json)(input));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_GetActionTypeCommand");
var se_GetJobDetailsCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("GetJobDetails");
  let body;
  body = JSON.stringify((0, import_smithy_client._json)(input));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_GetJobDetailsCommand");
var se_GetPipelineCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("GetPipeline");
  let body;
  body = JSON.stringify((0, import_smithy_client._json)(input));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_GetPipelineCommand");
var se_GetPipelineExecutionCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("GetPipelineExecution");
  let body;
  body = JSON.stringify((0, import_smithy_client._json)(input));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_GetPipelineExecutionCommand");
var se_GetPipelineStateCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("GetPipelineState");
  let body;
  body = JSON.stringify((0, import_smithy_client._json)(input));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_GetPipelineStateCommand");
var se_GetThirdPartyJobDetailsCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("GetThirdPartyJobDetails");
  let body;
  body = JSON.stringify((0, import_smithy_client._json)(input));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_GetThirdPartyJobDetailsCommand");
var se_ListActionExecutionsCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("ListActionExecutions");
  let body;
  body = JSON.stringify((0, import_smithy_client._json)(input));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_ListActionExecutionsCommand");
var se_ListActionTypesCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("ListActionTypes");
  let body;
  body = JSON.stringify((0, import_smithy_client._json)(input));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_ListActionTypesCommand");
var se_ListPipelineExecutionsCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("ListPipelineExecutions");
  let body;
  body = JSON.stringify((0, import_smithy_client._json)(input));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_ListPipelineExecutionsCommand");
var se_ListPipelinesCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("ListPipelines");
  let body;
  body = JSON.stringify((0, import_smithy_client._json)(input));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_ListPipelinesCommand");
var se_ListTagsForResourceCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("ListTagsForResource");
  let body;
  body = JSON.stringify((0, import_smithy_client._json)(input));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_ListTagsForResourceCommand");
var se_ListWebhooksCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("ListWebhooks");
  let body;
  body = JSON.stringify((0, import_smithy_client._json)(input));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_ListWebhooksCommand");
var se_PollForJobsCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("PollForJobs");
  let body;
  body = JSON.stringify((0, import_smithy_client._json)(input));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_PollForJobsCommand");
var se_PollForThirdPartyJobsCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("PollForThirdPartyJobs");
  let body;
  body = JSON.stringify((0, import_smithy_client._json)(input));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_PollForThirdPartyJobsCommand");
var se_PutActionRevisionCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("PutActionRevision");
  let body;
  body = JSON.stringify(se_PutActionRevisionInput(input, context));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_PutActionRevisionCommand");
var se_PutApprovalResultCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("PutApprovalResult");
  let body;
  body = JSON.stringify((0, import_smithy_client._json)(input));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_PutApprovalResultCommand");
var se_PutJobFailureResultCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("PutJobFailureResult");
  let body;
  body = JSON.stringify((0, import_smithy_client._json)(input));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_PutJobFailureResultCommand");
var se_PutJobSuccessResultCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("PutJobSuccessResult");
  let body;
  body = JSON.stringify(se_PutJobSuccessResultInput(input, context));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_PutJobSuccessResultCommand");
var se_PutThirdPartyJobFailureResultCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("PutThirdPartyJobFailureResult");
  let body;
  body = JSON.stringify((0, import_smithy_client._json)(input));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_PutThirdPartyJobFailureResultCommand");
var se_PutThirdPartyJobSuccessResultCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("PutThirdPartyJobSuccessResult");
  let body;
  body = JSON.stringify(se_PutThirdPartyJobSuccessResultInput(input, context));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_PutThirdPartyJobSuccessResultCommand");
var se_PutWebhookCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("PutWebhook");
  let body;
  body = JSON.stringify((0, import_smithy_client._json)(input));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_PutWebhookCommand");
var se_RegisterWebhookWithThirdPartyCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("RegisterWebhookWithThirdParty");
  let body;
  body = JSON.stringify((0, import_smithy_client._json)(input));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_RegisterWebhookWithThirdPartyCommand");
var se_RetryStageExecutionCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("RetryStageExecution");
  let body;
  body = JSON.stringify((0, import_smithy_client._json)(input));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_RetryStageExecutionCommand");
var se_StartPipelineExecutionCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("StartPipelineExecution");
  let body;
  body = JSON.stringify(se_StartPipelineExecutionInput(input, context));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_StartPipelineExecutionCommand");
var se_StopPipelineExecutionCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("StopPipelineExecution");
  let body;
  body = JSON.stringify((0, import_smithy_client._json)(input));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_StopPipelineExecutionCommand");
var se_TagResourceCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("TagResource");
  let body;
  body = JSON.stringify((0, import_smithy_client._json)(input));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_TagResourceCommand");
var se_UntagResourceCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("UntagResource");
  let body;
  body = JSON.stringify((0, import_smithy_client._json)(input));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_UntagResourceCommand");
var se_UpdateActionTypeCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("UpdateActionType");
  let body;
  body = JSON.stringify((0, import_smithy_client._json)(input));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_UpdateActionTypeCommand");
var se_UpdatePipelineCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("UpdatePipeline");
  let body;
  body = JSON.stringify((0, import_smithy_client._json)(input));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_UpdatePipelineCommand");
var de_AcknowledgeJobCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const data = await parseBody(output.body, context);
  let contents = {};
  contents = (0, import_smithy_client._json)(data);
  const response = {
    $metadata: deserializeMetadata(output),
    ...contents
  };
  return response;
}, "de_AcknowledgeJobCommand");
var de_AcknowledgeThirdPartyJobCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const data = await parseBody(output.body, context);
  let contents = {};
  contents = (0, import_smithy_client._json)(data);
  const response = {
    $metadata: deserializeMetadata(output),
    ...contents
  };
  return response;
}, "de_AcknowledgeThirdPartyJobCommand");
var de_CreateCustomActionTypeCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const data = await parseBody(output.body, context);
  let contents = {};
  contents = (0, import_smithy_client._json)(data);
  const response = {
    $metadata: deserializeMetadata(output),
    ...contents
  };
  return response;
}, "de_CreateCustomActionTypeCommand");
var de_CreatePipelineCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const data = await parseBody(output.body, context);
  let contents = {};
  contents = (0, import_smithy_client._json)(data);
  const response = {
    $metadata: deserializeMetadata(output),
    ...contents
  };
  return response;
}, "de_CreatePipelineCommand");
var de_DeleteCustomActionTypeCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  await (0, import_smithy_client.collectBody)(output.body, context);
  const response = {
    $metadata: deserializeMetadata(output)
  };
  return response;
}, "de_DeleteCustomActionTypeCommand");
var de_DeletePipelineCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  await (0, import_smithy_client.collectBody)(output.body, context);
  const response = {
    $metadata: deserializeMetadata(output)
  };
  return response;
}, "de_DeletePipelineCommand");
var de_DeleteWebhookCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const data = await parseBody(output.body, context);
  let contents = {};
  contents = (0, import_smithy_client._json)(data);
  const response = {
    $metadata: deserializeMetadata(output),
    ...contents
  };
  return response;
}, "de_DeleteWebhookCommand");
var de_DeregisterWebhookWithThirdPartyCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const data = await parseBody(output.body, context);
  let contents = {};
  contents = (0, import_smithy_client._json)(data);
  const response = {
    $metadata: deserializeMetadata(output),
    ...contents
  };
  return response;
}, "de_DeregisterWebhookWithThirdPartyCommand");
var de_DisableStageTransitionCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  await (0, import_smithy_client.collectBody)(output.body, context);
  const response = {
    $metadata: deserializeMetadata(output)
  };
  return response;
}, "de_DisableStageTransitionCommand");
var de_EnableStageTransitionCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  await (0, import_smithy_client.collectBody)(output.body, context);
  const response = {
    $metadata: deserializeMetadata(output)
  };
  return response;
}, "de_EnableStageTransitionCommand");
var de_GetActionTypeCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const data = await parseBody(output.body, context);
  let contents = {};
  contents = (0, import_smithy_client._json)(data);
  const response = {
    $metadata: deserializeMetadata(output),
    ...contents
  };
  return response;
}, "de_GetActionTypeCommand");
var de_GetJobDetailsCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const data = await parseBody(output.body, context);
  let contents = {};
  contents = (0, import_smithy_client._json)(data);
  const response = {
    $metadata: deserializeMetadata(output),
    ...contents
  };
  return response;
}, "de_GetJobDetailsCommand");
var de_GetPipelineCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const data = await parseBody(output.body, context);
  let contents = {};
  contents = de_GetPipelineOutput(data, context);
  const response = {
    $metadata: deserializeMetadata(output),
    ...contents
  };
  return response;
}, "de_GetPipelineCommand");
var de_GetPipelineExecutionCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const data = await parseBody(output.body, context);
  let contents = {};
  contents = de_GetPipelineExecutionOutput(data, context);
  const response = {
    $metadata: deserializeMetadata(output),
    ...contents
  };
  return response;
}, "de_GetPipelineExecutionCommand");
var de_GetPipelineStateCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const data = await parseBody(output.body, context);
  let contents = {};
  contents = de_GetPipelineStateOutput(data, context);
  const response = {
    $metadata: deserializeMetadata(output),
    ...contents
  };
  return response;
}, "de_GetPipelineStateCommand");
var de_GetThirdPartyJobDetailsCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const data = await parseBody(output.body, context);
  let contents = {};
  contents = (0, import_smithy_client._json)(data);
  const response = {
    $metadata: deserializeMetadata(output),
    ...contents
  };
  return response;
}, "de_GetThirdPartyJobDetailsCommand");
var de_ListActionExecutionsCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const data = await parseBody(output.body, context);
  let contents = {};
  contents = de_ListActionExecutionsOutput(data, context);
  const response = {
    $metadata: deserializeMetadata(output),
    ...contents
  };
  return response;
}, "de_ListActionExecutionsCommand");
var de_ListActionTypesCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const data = await parseBody(output.body, context);
  let contents = {};
  contents = (0, import_smithy_client._json)(data);
  const response = {
    $metadata: deserializeMetadata(output),
    ...contents
  };
  return response;
}, "de_ListActionTypesCommand");
var de_ListPipelineExecutionsCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const data = await parseBody(output.body, context);
  let contents = {};
  contents = de_ListPipelineExecutionsOutput(data, context);
  const response = {
    $metadata: deserializeMetadata(output),
    ...contents
  };
  return response;
}, "de_ListPipelineExecutionsCommand");
var de_ListPipelinesCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const data = await parseBody(output.body, context);
  let contents = {};
  contents = de_ListPipelinesOutput(data, context);
  const response = {
    $metadata: deserializeMetadata(output),
    ...contents
  };
  return response;
}, "de_ListPipelinesCommand");
var de_ListTagsForResourceCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const data = await parseBody(output.body, context);
  let contents = {};
  contents = (0, import_smithy_client._json)(data);
  const response = {
    $metadata: deserializeMetadata(output),
    ...contents
  };
  return response;
}, "de_ListTagsForResourceCommand");
var de_ListWebhooksCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const data = await parseBody(output.body, context);
  let contents = {};
  contents = de_ListWebhooksOutput(data, context);
  const response = {
    $metadata: deserializeMetadata(output),
    ...contents
  };
  return response;
}, "de_ListWebhooksCommand");
var de_PollForJobsCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const data = await parseBody(output.body, context);
  let contents = {};
  contents = (0, import_smithy_client._json)(data);
  const response = {
    $metadata: deserializeMetadata(output),
    ...contents
  };
  return response;
}, "de_PollForJobsCommand");
var de_PollForThirdPartyJobsCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const data = await parseBody(output.body, context);
  let contents = {};
  contents = (0, import_smithy_client._json)(data);
  const response = {
    $metadata: deserializeMetadata(output),
    ...contents
  };
  return response;
}, "de_PollForThirdPartyJobsCommand");
var de_PutActionRevisionCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const data = await parseBody(output.body, context);
  let contents = {};
  contents = (0, import_smithy_client._json)(data);
  const response = {
    $metadata: deserializeMetadata(output),
    ...contents
  };
  return response;
}, "de_PutActionRevisionCommand");
var de_PutApprovalResultCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const data = await parseBody(output.body, context);
  let contents = {};
  contents = de_PutApprovalResultOutput(data, context);
  const response = {
    $metadata: deserializeMetadata(output),
    ...contents
  };
  return response;
}, "de_PutApprovalResultCommand");
var de_PutJobFailureResultCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  await (0, import_smithy_client.collectBody)(output.body, context);
  const response = {
    $metadata: deserializeMetadata(output)
  };
  return response;
}, "de_PutJobFailureResultCommand");
var de_PutJobSuccessResultCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  await (0, import_smithy_client.collectBody)(output.body, context);
  const response = {
    $metadata: deserializeMetadata(output)
  };
  return response;
}, "de_PutJobSuccessResultCommand");
var de_PutThirdPartyJobFailureResultCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  await (0, import_smithy_client.collectBody)(output.body, context);
  const response = {
    $metadata: deserializeMetadata(output)
  };
  return response;
}, "de_PutThirdPartyJobFailureResultCommand");
var de_PutThirdPartyJobSuccessResultCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  await (0, import_smithy_client.collectBody)(output.body, context);
  const response = {
    $metadata: deserializeMetadata(output)
  };
  return response;
}, "de_PutThirdPartyJobSuccessResultCommand");
var de_PutWebhookCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const data = await parseBody(output.body, context);
  let contents = {};
  contents = de_PutWebhookOutput(data, context);
  const response = {
    $metadata: deserializeMetadata(output),
    ...contents
  };
  return response;
}, "de_PutWebhookCommand");
var de_RegisterWebhookWithThirdPartyCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const data = await parseBody(output.body, context);
  let contents = {};
  contents = (0, import_smithy_client._json)(data);
  const response = {
    $metadata: deserializeMetadata(output),
    ...contents
  };
  return response;
}, "de_RegisterWebhookWithThirdPartyCommand");
var de_RetryStageExecutionCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const data = await parseBody(output.body, context);
  let contents = {};
  contents = (0, import_smithy_client._json)(data);
  const response = {
    $metadata: deserializeMetadata(output),
    ...contents
  };
  return response;
}, "de_RetryStageExecutionCommand");
var de_StartPipelineExecutionCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const data = await parseBody(output.body, context);
  let contents = {};
  contents = (0, import_smithy_client._json)(data);
  const response = {
    $metadata: deserializeMetadata(output),
    ...contents
  };
  return response;
}, "de_StartPipelineExecutionCommand");
var de_StopPipelineExecutionCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const data = await parseBody(output.body, context);
  let contents = {};
  contents = (0, import_smithy_client._json)(data);
  const response = {
    $metadata: deserializeMetadata(output),
    ...contents
  };
  return response;
}, "de_StopPipelineExecutionCommand");
var de_TagResourceCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const data = await parseBody(output.body, context);
  let contents = {};
  contents = (0, import_smithy_client._json)(data);
  const response = {
    $metadata: deserializeMetadata(output),
    ...contents
  };
  return response;
}, "de_TagResourceCommand");
var de_UntagResourceCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const data = await parseBody(output.body, context);
  let contents = {};
  contents = (0, import_smithy_client._json)(data);
  const response = {
    $metadata: deserializeMetadata(output),
    ...contents
  };
  return response;
}, "de_UntagResourceCommand");
var de_UpdateActionTypeCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  await (0, import_smithy_client.collectBody)(output.body, context);
  const response = {
    $metadata: deserializeMetadata(output)
  };
  return response;
}, "de_UpdateActionTypeCommand");
var de_UpdatePipelineCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const data = await parseBody(output.body, context);
  let contents = {};
  contents = (0, import_smithy_client._json)(data);
  const response = {
    $metadata: deserializeMetadata(output),
    ...contents
  };
  return response;
}, "de_UpdatePipelineCommand");
var de_CommandError = /* @__PURE__ */ __name(async (output, context) => {
  const parsedOutput = {
    ...output,
    body: await parseErrorBody(output.body, context)
  };
  const errorCode = loadRestJsonErrorCode(output, parsedOutput.body);
  switch (errorCode) {
    case "InvalidNonceException":
    case "com.amazonaws.codepipeline#InvalidNonceException":
      throw await de_InvalidNonceExceptionRes(parsedOutput, context);
    case "JobNotFoundException":
    case "com.amazonaws.codepipeline#JobNotFoundException":
      throw await de_JobNotFoundExceptionRes(parsedOutput, context);
    case "ValidationException":
    case "com.amazonaws.codepipeline#ValidationException":
      throw await de_ValidationExceptionRes(parsedOutput, context);
    case "InvalidClientTokenException":
    case "com.amazonaws.codepipeline#InvalidClientTokenException":
      throw await de_InvalidClientTokenExceptionRes(parsedOutput, context);
    case "ConcurrentModificationException":
    case "com.amazonaws.codepipeline#ConcurrentModificationException":
      throw await de_ConcurrentModificationExceptionRes(parsedOutput, context);
    case "InvalidTagsException":
    case "com.amazonaws.codepipeline#InvalidTagsException":
      throw await de_InvalidTagsExceptionRes(parsedOutput, context);
    case "LimitExceededException":
    case "com.amazonaws.codepipeline#LimitExceededException":
      throw await de_LimitExceededExceptionRes(parsedOutput, context);
    case "TooManyTagsException":
    case "com.amazonaws.codepipeline#TooManyTagsException":
      throw await de_TooManyTagsExceptionRes(parsedOutput, context);
    case "InvalidActionDeclarationException":
    case "com.amazonaws.codepipeline#InvalidActionDeclarationException":
      throw await de_InvalidActionDeclarationExceptionRes(parsedOutput, context);
    case "InvalidBlockerDeclarationException":
    case "com.amazonaws.codepipeline#InvalidBlockerDeclarationException":
      throw await de_InvalidBlockerDeclarationExceptionRes(parsedOutput, context);
    case "InvalidStageDeclarationException":
    case "com.amazonaws.codepipeline#InvalidStageDeclarationException":
      throw await de_InvalidStageDeclarationExceptionRes(parsedOutput, context);
    case "InvalidStructureException":
    case "com.amazonaws.codepipeline#InvalidStructureException":
      throw await de_InvalidStructureExceptionRes(parsedOutput, context);
    case "PipelineNameInUseException":
    case "com.amazonaws.codepipeline#PipelineNameInUseException":
      throw await de_PipelineNameInUseExceptionRes(parsedOutput, context);
    case "WebhookNotFoundException":
    case "com.amazonaws.codepipeline#WebhookNotFoundException":
      throw await de_WebhookNotFoundExceptionRes(parsedOutput, context);
    case "PipelineNotFoundException":
    case "com.amazonaws.codepipeline#PipelineNotFoundException":
      throw await de_PipelineNotFoundExceptionRes(parsedOutput, context);
    case "StageNotFoundException":
    case "com.amazonaws.codepipeline#StageNotFoundException":
      throw await de_StageNotFoundExceptionRes(parsedOutput, context);
    case "ActionTypeNotFoundException":
    case "com.amazonaws.codepipeline#ActionTypeNotFoundException":
      throw await de_ActionTypeNotFoundExceptionRes(parsedOutput, context);
    case "PipelineVersionNotFoundException":
    case "com.amazonaws.codepipeline#PipelineVersionNotFoundException":
      throw await de_PipelineVersionNotFoundExceptionRes(parsedOutput, context);
    case "PipelineExecutionNotFoundException":
    case "com.amazonaws.codepipeline#PipelineExecutionNotFoundException":
      throw await de_PipelineExecutionNotFoundExceptionRes(parsedOutput, context);
    case "InvalidJobException":
    case "com.amazonaws.codepipeline#InvalidJobException":
      throw await de_InvalidJobExceptionRes(parsedOutput, context);
    case "InvalidNextTokenException":
    case "com.amazonaws.codepipeline#InvalidNextTokenException":
      throw await de_InvalidNextTokenExceptionRes(parsedOutput, context);
    case "InvalidArnException":
    case "com.amazonaws.codepipeline#InvalidArnException":
      throw await de_InvalidArnExceptionRes(parsedOutput, context);
    case "ResourceNotFoundException":
    case "com.amazonaws.codepipeline#ResourceNotFoundException":
      throw await de_ResourceNotFoundExceptionRes(parsedOutput, context);
    case "ActionNotFoundException":
    case "com.amazonaws.codepipeline#ActionNotFoundException":
      throw await de_ActionNotFoundExceptionRes(parsedOutput, context);
    case "ApprovalAlreadyCompletedException":
    case "com.amazonaws.codepipeline#ApprovalAlreadyCompletedException":
      throw await de_ApprovalAlreadyCompletedExceptionRes(parsedOutput, context);
    case "InvalidApprovalTokenException":
    case "com.amazonaws.codepipeline#InvalidApprovalTokenException":
      throw await de_InvalidApprovalTokenExceptionRes(parsedOutput, context);
    case "InvalidJobStateException":
    case "com.amazonaws.codepipeline#InvalidJobStateException":
      throw await de_InvalidJobStateExceptionRes(parsedOutput, context);
    case "OutputVariablesSizeExceededException":
    case "com.amazonaws.codepipeline#OutputVariablesSizeExceededException":
      throw await de_OutputVariablesSizeExceededExceptionRes(parsedOutput, context);
    case "InvalidWebhookAuthenticationParametersException":
    case "com.amazonaws.codepipeline#InvalidWebhookAuthenticationParametersException":
      throw await de_InvalidWebhookAuthenticationParametersExceptionRes(parsedOutput, context);
    case "InvalidWebhookFilterPatternException":
    case "com.amazonaws.codepipeline#InvalidWebhookFilterPatternException":
      throw await de_InvalidWebhookFilterPatternExceptionRes(parsedOutput, context);
    case "ConflictException":
    case "com.amazonaws.codepipeline#ConflictException":
      throw await de_ConflictExceptionRes(parsedOutput, context);
    case "NotLatestPipelineExecutionException":
    case "com.amazonaws.codepipeline#NotLatestPipelineExecutionException":
      throw await de_NotLatestPipelineExecutionExceptionRes(parsedOutput, context);
    case "StageNotRetryableException":
    case "com.amazonaws.codepipeline#StageNotRetryableException":
      throw await de_StageNotRetryableExceptionRes(parsedOutput, context);
    case "ConcurrentPipelineExecutionsLimitExceededException":
    case "com.amazonaws.codepipeline#ConcurrentPipelineExecutionsLimitExceededException":
      throw await de_ConcurrentPipelineExecutionsLimitExceededExceptionRes(parsedOutput, context);
    case "DuplicatedStopRequestException":
    case "com.amazonaws.codepipeline#DuplicatedStopRequestException":
      throw await de_DuplicatedStopRequestExceptionRes(parsedOutput, context);
    case "PipelineExecutionNotStoppableException":
    case "com.amazonaws.codepipeline#PipelineExecutionNotStoppableException":
      throw await de_PipelineExecutionNotStoppableExceptionRes(parsedOutput, context);
    case "RequestFailedException":
    case "com.amazonaws.codepipeline#RequestFailedException":
      throw await de_RequestFailedExceptionRes(parsedOutput, context);
    default:
      const parsedBody = parsedOutput.body;
      return throwDefaultError({
        output,
        parsedBody,
        errorCode
      });
  }
}, "de_CommandError");
var de_ActionNotFoundExceptionRes = /* @__PURE__ */ __name(async (parsedOutput, context) => {
  const body = parsedOutput.body;
  const deserialized = (0, import_smithy_client._json)(body);
  const exception = new ActionNotFoundException({
    $metadata: deserializeMetadata(parsedOutput),
    ...deserialized
  });
  return (0, import_smithy_client.decorateServiceException)(exception, body);
}, "de_ActionNotFoundExceptionRes");
var de_ActionTypeNotFoundExceptionRes = /* @__PURE__ */ __name(async (parsedOutput, context) => {
  const body = parsedOutput.body;
  const deserialized = (0, import_smithy_client._json)(body);
  const exception = new ActionTypeNotFoundException({
    $metadata: deserializeMetadata(parsedOutput),
    ...deserialized
  });
  return (0, import_smithy_client.decorateServiceException)(exception, body);
}, "de_ActionTypeNotFoundExceptionRes");
var de_ApprovalAlreadyCompletedExceptionRes = /* @__PURE__ */ __name(async (parsedOutput, context) => {
  const body = parsedOutput.body;
  const deserialized = (0, import_smithy_client._json)(body);
  const exception = new ApprovalAlreadyCompletedException({
    $metadata: deserializeMetadata(parsedOutput),
    ...deserialized
  });
  return (0, import_smithy_client.decorateServiceException)(exception, body);
}, "de_ApprovalAlreadyCompletedExceptionRes");
var de_ConcurrentModificationExceptionRes = /* @__PURE__ */ __name(async (parsedOutput, context) => {
  const body = parsedOutput.body;
  const deserialized = (0, import_smithy_client._json)(body);
  const exception = new ConcurrentModificationException({
    $metadata: deserializeMetadata(parsedOutput),
    ...deserialized
  });
  return (0, import_smithy_client.decorateServiceException)(exception, body);
}, "de_ConcurrentModificationExceptionRes");
var de_ConcurrentPipelineExecutionsLimitExceededExceptionRes = /* @__PURE__ */ __name(async (parsedOutput, context) => {
  const body = parsedOutput.body;
  const deserialized = (0, import_smithy_client._json)(body);
  const exception = new ConcurrentPipelineExecutionsLimitExceededException({
    $metadata: deserializeMetadata(parsedOutput),
    ...deserialized
  });
  return (0, import_smithy_client.decorateServiceException)(exception, body);
}, "de_ConcurrentPipelineExecutionsLimitExceededExceptionRes");
var de_ConflictExceptionRes = /* @__PURE__ */ __name(async (parsedOutput, context) => {
  const body = parsedOutput.body;
  const deserialized = (0, import_smithy_client._json)(body);
  const exception = new ConflictException({
    $metadata: deserializeMetadata(parsedOutput),
    ...deserialized
  });
  return (0, import_smithy_client.decorateServiceException)(exception, body);
}, "de_ConflictExceptionRes");
var de_DuplicatedStopRequestExceptionRes = /* @__PURE__ */ __name(async (parsedOutput, context) => {
  const body = parsedOutput.body;
  const deserialized = (0, import_smithy_client._json)(body);
  const exception = new DuplicatedStopRequestException({
    $metadata: deserializeMetadata(parsedOutput),
    ...deserialized
  });
  return (0, import_smithy_client.decorateServiceException)(exception, body);
}, "de_DuplicatedStopRequestExceptionRes");
var de_InvalidActionDeclarationExceptionRes = /* @__PURE__ */ __name(async (parsedOutput, context) => {
  const body = parsedOutput.body;
  const deserialized = (0, import_smithy_client._json)(body);
  const exception = new InvalidActionDeclarationException({
    $metadata: deserializeMetadata(parsedOutput),
    ...deserialized
  });
  return (0, import_smithy_client.decorateServiceException)(exception, body);
}, "de_InvalidActionDeclarationExceptionRes");
var de_InvalidApprovalTokenExceptionRes = /* @__PURE__ */ __name(async (parsedOutput, context) => {
  const body = parsedOutput.body;
  const deserialized = (0, import_smithy_client._json)(body);
  const exception = new InvalidApprovalTokenException({
    $metadata: deserializeMetadata(parsedOutput),
    ...deserialized
  });
  return (0, import_smithy_client.decorateServiceException)(exception, body);
}, "de_InvalidApprovalTokenExceptionRes");
var de_InvalidArnExceptionRes = /* @__PURE__ */ __name(async (parsedOutput, context) => {
  const body = parsedOutput.body;
  const deserialized = (0, import_smithy_client._json)(body);
  const exception = new InvalidArnException({
    $metadata: deserializeMetadata(parsedOutput),
    ...deserialized
  });
  return (0, import_smithy_client.decorateServiceException)(exception, body);
}, "de_InvalidArnExceptionRes");
var de_InvalidBlockerDeclarationExceptionRes = /* @__PURE__ */ __name(async (parsedOutput, context) => {
  const body = parsedOutput.body;
  const deserialized = (0, import_smithy_client._json)(body);
  const exception = new InvalidBlockerDeclarationException({
    $metadata: deserializeMetadata(parsedOutput),
    ...deserialized
  });
  return (0, import_smithy_client.decorateServiceException)(exception, body);
}, "de_InvalidBlockerDeclarationExceptionRes");
var de_InvalidClientTokenExceptionRes = /* @__PURE__ */ __name(async (parsedOutput, context) => {
  const body = parsedOutput.body;
  const deserialized = (0, import_smithy_client._json)(body);
  const exception = new InvalidClientTokenException({
    $metadata: deserializeMetadata(parsedOutput),
    ...deserialized
  });
  return (0, import_smithy_client.decorateServiceException)(exception, body);
}, "de_InvalidClientTokenExceptionRes");
var de_InvalidJobExceptionRes = /* @__PURE__ */ __name(async (parsedOutput, context) => {
  const body = parsedOutput.body;
  const deserialized = (0, import_smithy_client._json)(body);
  const exception = new InvalidJobException({
    $metadata: deserializeMetadata(parsedOutput),
    ...deserialized
  });
  return (0, import_smithy_client.decorateServiceException)(exception, body);
}, "de_InvalidJobExceptionRes");
var de_InvalidJobStateExceptionRes = /* @__PURE__ */ __name(async (parsedOutput, context) => {
  const body = parsedOutput.body;
  const deserialized = (0, import_smithy_client._json)(body);
  const exception = new InvalidJobStateException({
    $metadata: deserializeMetadata(parsedOutput),
    ...deserialized
  });
  return (0, import_smithy_client.decorateServiceException)(exception, body);
}, "de_InvalidJobStateExceptionRes");
var de_InvalidNextTokenExceptionRes = /* @__PURE__ */ __name(async (parsedOutput, context) => {
  const body = parsedOutput.body;
  const deserialized = (0, import_smithy_client._json)(body);
  const exception = new InvalidNextTokenException({
    $metadata: deserializeMetadata(parsedOutput),
    ...deserialized
  });
  return (0, import_smithy_client.decorateServiceException)(exception, body);
}, "de_InvalidNextTokenExceptionRes");
var de_InvalidNonceExceptionRes = /* @__PURE__ */ __name(async (parsedOutput, context) => {
  const body = parsedOutput.body;
  const deserialized = (0, import_smithy_client._json)(body);
  const exception = new InvalidNonceException({
    $metadata: deserializeMetadata(parsedOutput),
    ...deserialized
  });
  return (0, import_smithy_client.decorateServiceException)(exception, body);
}, "de_InvalidNonceExceptionRes");
var de_InvalidStageDeclarationExceptionRes = /* @__PURE__ */ __name(async (parsedOutput, context) => {
  const body = parsedOutput.body;
  const deserialized = (0, import_smithy_client._json)(body);
  const exception = new InvalidStageDeclarationException({
    $metadata: deserializeMetadata(parsedOutput),
    ...deserialized
  });
  return (0, import_smithy_client.decorateServiceException)(exception, body);
}, "de_InvalidStageDeclarationExceptionRes");
var de_InvalidStructureExceptionRes = /* @__PURE__ */ __name(async (parsedOutput, context) => {
  const body = parsedOutput.body;
  const deserialized = (0, import_smithy_client._json)(body);
  const exception = new InvalidStructureException({
    $metadata: deserializeMetadata(parsedOutput),
    ...deserialized
  });
  return (0, import_smithy_client.decorateServiceException)(exception, body);
}, "de_InvalidStructureExceptionRes");
var de_InvalidTagsExceptionRes = /* @__PURE__ */ __name(async (parsedOutput, context) => {
  const body = parsedOutput.body;
  const deserialized = (0, import_smithy_client._json)(body);
  const exception = new InvalidTagsException({
    $metadata: deserializeMetadata(parsedOutput),
    ...deserialized
  });
  return (0, import_smithy_client.decorateServiceException)(exception, body);
}, "de_InvalidTagsExceptionRes");
var de_InvalidWebhookAuthenticationParametersExceptionRes = /* @__PURE__ */ __name(async (parsedOutput, context) => {
  const body = parsedOutput.body;
  const deserialized = (0, import_smithy_client._json)(body);
  const exception = new InvalidWebhookAuthenticationParametersException({
    $metadata: deserializeMetadata(parsedOutput),
    ...deserialized
  });
  return (0, import_smithy_client.decorateServiceException)(exception, body);
}, "de_InvalidWebhookAuthenticationParametersExceptionRes");
var de_InvalidWebhookFilterPatternExceptionRes = /* @__PURE__ */ __name(async (parsedOutput, context) => {
  const body = parsedOutput.body;
  const deserialized = (0, import_smithy_client._json)(body);
  const exception = new InvalidWebhookFilterPatternException({
    $metadata: deserializeMetadata(parsedOutput),
    ...deserialized
  });
  return (0, import_smithy_client.decorateServiceException)(exception, body);
}, "de_InvalidWebhookFilterPatternExceptionRes");
var de_JobNotFoundExceptionRes = /* @__PURE__ */ __name(async (parsedOutput, context) => {
  const body = parsedOutput.body;
  const deserialized = (0, import_smithy_client._json)(body);
  const exception = new JobNotFoundException({
    $metadata: deserializeMetadata(parsedOutput),
    ...deserialized
  });
  return (0, import_smithy_client.decorateServiceException)(exception, body);
}, "de_JobNotFoundExceptionRes");
var de_LimitExceededExceptionRes = /* @__PURE__ */ __name(async (parsedOutput, context) => {
  const body = parsedOutput.body;
  const deserialized = (0, import_smithy_client._json)(body);
  const exception = new LimitExceededException({
    $metadata: deserializeMetadata(parsedOutput),
    ...deserialized
  });
  return (0, import_smithy_client.decorateServiceException)(exception, body);
}, "de_LimitExceededExceptionRes");
var de_NotLatestPipelineExecutionExceptionRes = /* @__PURE__ */ __name(async (parsedOutput, context) => {
  const body = parsedOutput.body;
  const deserialized = (0, import_smithy_client._json)(body);
  const exception = new NotLatestPipelineExecutionException({
    $metadata: deserializeMetadata(parsedOutput),
    ...deserialized
  });
  return (0, import_smithy_client.decorateServiceException)(exception, body);
}, "de_NotLatestPipelineExecutionExceptionRes");
var de_OutputVariablesSizeExceededExceptionRes = /* @__PURE__ */ __name(async (parsedOutput, context) => {
  const body = parsedOutput.body;
  const deserialized = (0, import_smithy_client._json)(body);
  const exception = new OutputVariablesSizeExceededException({
    $metadata: deserializeMetadata(parsedOutput),
    ...deserialized
  });
  return (0, import_smithy_client.decorateServiceException)(exception, body);
}, "de_OutputVariablesSizeExceededExceptionRes");
var de_PipelineExecutionNotFoundExceptionRes = /* @__PURE__ */ __name(async (parsedOutput, context) => {
  const body = parsedOutput.body;
  const deserialized = (0, import_smithy_client._json)(body);
  const exception = new PipelineExecutionNotFoundException({
    $metadata: deserializeMetadata(parsedOutput),
    ...deserialized
  });
  return (0, import_smithy_client.decorateServiceException)(exception, body);
}, "de_PipelineExecutionNotFoundExceptionRes");
var de_PipelineExecutionNotStoppableExceptionRes = /* @__PURE__ */ __name(async (parsedOutput, context) => {
  const body = parsedOutput.body;
  const deserialized = (0, import_smithy_client._json)(body);
  const exception = new PipelineExecutionNotStoppableException({
    $metadata: deserializeMetadata(parsedOutput),
    ...deserialized
  });
  return (0, import_smithy_client.decorateServiceException)(exception, body);
}, "de_PipelineExecutionNotStoppableExceptionRes");
var de_PipelineNameInUseExceptionRes = /* @__PURE__ */ __name(async (parsedOutput, context) => {
  const body = parsedOutput.body;
  const deserialized = (0, import_smithy_client._json)(body);
  const exception = new PipelineNameInUseException({
    $metadata: deserializeMetadata(parsedOutput),
    ...deserialized
  });
  return (0, import_smithy_client.decorateServiceException)(exception, body);
}, "de_PipelineNameInUseExceptionRes");
var de_PipelineNotFoundExceptionRes = /* @__PURE__ */ __name(async (parsedOutput, context) => {
  const body = parsedOutput.body;
  const deserialized = (0, import_smithy_client._json)(body);
  const exception = new PipelineNotFoundException({
    $metadata: deserializeMetadata(parsedOutput),
    ...deserialized
  });
  return (0, import_smithy_client.decorateServiceException)(exception, body);
}, "de_PipelineNotFoundExceptionRes");
var de_PipelineVersionNotFoundExceptionRes = /* @__PURE__ */ __name(async (parsedOutput, context) => {
  const body = parsedOutput.body;
  const deserialized = (0, import_smithy_client._json)(body);
  const exception = new PipelineVersionNotFoundException({
    $metadata: deserializeMetadata(parsedOutput),
    ...deserialized
  });
  return (0, import_smithy_client.decorateServiceException)(exception, body);
}, "de_PipelineVersionNotFoundExceptionRes");
var de_RequestFailedExceptionRes = /* @__PURE__ */ __name(async (parsedOutput, context) => {
  const body = parsedOutput.body;
  const deserialized = (0, import_smithy_client._json)(body);
  const exception = new RequestFailedException({
    $metadata: deserializeMetadata(parsedOutput),
    ...deserialized
  });
  return (0, import_smithy_client.decorateServiceException)(exception, body);
}, "de_RequestFailedExceptionRes");
var de_ResourceNotFoundExceptionRes = /* @__PURE__ */ __name(async (parsedOutput, context) => {
  const body = parsedOutput.body;
  const deserialized = (0, import_smithy_client._json)(body);
  const exception = new ResourceNotFoundException({
    $metadata: deserializeMetadata(parsedOutput),
    ...deserialized
  });
  return (0, import_smithy_client.decorateServiceException)(exception, body);
}, "de_ResourceNotFoundExceptionRes");
var de_StageNotFoundExceptionRes = /* @__PURE__ */ __name(async (parsedOutput, context) => {
  const body = parsedOutput.body;
  const deserialized = (0, import_smithy_client._json)(body);
  const exception = new StageNotFoundException({
    $metadata: deserializeMetadata(parsedOutput),
    ...deserialized
  });
  return (0, import_smithy_client.decorateServiceException)(exception, body);
}, "de_StageNotFoundExceptionRes");
var de_StageNotRetryableExceptionRes = /* @__PURE__ */ __name(async (parsedOutput, context) => {
  const body = parsedOutput.body;
  const deserialized = (0, import_smithy_client._json)(body);
  const exception = new StageNotRetryableException({
    $metadata: deserializeMetadata(parsedOutput),
    ...deserialized
  });
  return (0, import_smithy_client.decorateServiceException)(exception, body);
}, "de_StageNotRetryableExceptionRes");
var de_TooManyTagsExceptionRes = /* @__PURE__ */ __name(async (parsedOutput, context) => {
  const body = parsedOutput.body;
  const deserialized = (0, import_smithy_client._json)(body);
  const exception = new TooManyTagsException({
    $metadata: deserializeMetadata(parsedOutput),
    ...deserialized
  });
  return (0, import_smithy_client.decorateServiceException)(exception, body);
}, "de_TooManyTagsExceptionRes");
var de_ValidationExceptionRes = /* @__PURE__ */ __name(async (parsedOutput, context) => {
  const body = parsedOutput.body;
  const deserialized = (0, import_smithy_client._json)(body);
  const exception = new ValidationException({
    $metadata: deserializeMetadata(parsedOutput),
    ...deserialized
  });
  return (0, import_smithy_client.decorateServiceException)(exception, body);
}, "de_ValidationExceptionRes");
var de_WebhookNotFoundExceptionRes = /* @__PURE__ */ __name(async (parsedOutput, context) => {
  const body = parsedOutput.body;
  const deserialized = (0, import_smithy_client._json)(body);
  const exception = new WebhookNotFoundException({
    $metadata: deserializeMetadata(parsedOutput),
    ...deserialized
  });
  return (0, import_smithy_client.decorateServiceException)(exception, body);
}, "de_WebhookNotFoundExceptionRes");
var se_ActionRevision = /* @__PURE__ */ __name((input, context) => {
  return (0, import_smithy_client.take)(input, {
    created: (_) => Math.round(_.getTime() / 1e3),
    revisionChangeId: [],
    revisionId: []
  });
}, "se_ActionRevision");
var se_CurrentRevision = /* @__PURE__ */ __name((input, context) => {
  return (0, import_smithy_client.take)(input, {
    changeIdentifier: [],
    created: (_) => Math.round(_.getTime() / 1e3),
    revision: [],
    revisionSummary: []
  });
}, "se_CurrentRevision");
var se_PutActionRevisionInput = /* @__PURE__ */ __name((input, context) => {
  return (0, import_smithy_client.take)(input, {
    actionName: [],
    actionRevision: (_) => se_ActionRevision(_, context),
    pipelineName: [],
    stageName: []
  });
}, "se_PutActionRevisionInput");
var se_PutJobSuccessResultInput = /* @__PURE__ */ __name((input, context) => {
  return (0, import_smithy_client.take)(input, {
    continuationToken: [],
    currentRevision: (_) => se_CurrentRevision(_, context),
    executionDetails: import_smithy_client._json,
    jobId: [],
    outputVariables: import_smithy_client._json
  });
}, "se_PutJobSuccessResultInput");
var se_PutThirdPartyJobSuccessResultInput = /* @__PURE__ */ __name((input, context) => {
  return (0, import_smithy_client.take)(input, {
    clientToken: [],
    continuationToken: [],
    currentRevision: (_) => se_CurrentRevision(_, context),
    executionDetails: import_smithy_client._json,
    jobId: []
  });
}, "se_PutThirdPartyJobSuccessResultInput");
var se_StartPipelineExecutionInput = /* @__PURE__ */ __name((input, context) => {
  return (0, import_smithy_client.take)(input, {
    clientRequestToken: [true, (_) => _ ?? (0, import_uuid.v4)()],
    name: [],
    sourceRevisions: import_smithy_client._json,
    variables: import_smithy_client._json
  });
}, "se_StartPipelineExecutionInput");
var de_ActionExecution = /* @__PURE__ */ __name((output, context) => {
  return (0, import_smithy_client.take)(output, {
    actionExecutionId: import_smithy_client.expectString,
    errorDetails: import_smithy_client._json,
    externalExecutionId: import_smithy_client.expectString,
    externalExecutionUrl: import_smithy_client.expectString,
    lastStatusChange: (_) => (0, import_smithy_client.expectNonNull)((0, import_smithy_client.parseEpochTimestamp)((0, import_smithy_client.expectNumber)(_))),
    lastUpdatedBy: import_smithy_client.expectString,
    percentComplete: import_smithy_client.expectInt32,
    status: import_smithy_client.expectString,
    summary: import_smithy_client.expectString,
    token: import_smithy_client.expectString
  });
}, "de_ActionExecution");
var de_ActionExecutionDetail = /* @__PURE__ */ __name((output, context) => {
  return (0, import_smithy_client.take)(output, {
    actionExecutionId: import_smithy_client.expectString,
    actionName: import_smithy_client.expectString,
    input: import_smithy_client._json,
    lastUpdateTime: (_) => (0, import_smithy_client.expectNonNull)((0, import_smithy_client.parseEpochTimestamp)((0, import_smithy_client.expectNumber)(_))),
    output: import_smithy_client._json,
    pipelineExecutionId: import_smithy_client.expectString,
    pipelineVersion: import_smithy_client.expectInt32,
    stageName: import_smithy_client.expectString,
    startTime: (_) => (0, import_smithy_client.expectNonNull)((0, import_smithy_client.parseEpochTimestamp)((0, import_smithy_client.expectNumber)(_))),
    status: import_smithy_client.expectString,
    updatedBy: import_smithy_client.expectString
  });
}, "de_ActionExecutionDetail");
var de_ActionExecutionDetailList = /* @__PURE__ */ __name((output, context) => {
  const retVal = (output || []).filter((e) => e != null).map((entry) => {
    return de_ActionExecutionDetail(entry, context);
  });
  return retVal;
}, "de_ActionExecutionDetailList");
var de_ActionRevision = /* @__PURE__ */ __name((output, context) => {
  return (0, import_smithy_client.take)(output, {
    created: (_) => (0, import_smithy_client.expectNonNull)((0, import_smithy_client.parseEpochTimestamp)((0, import_smithy_client.expectNumber)(_))),
    revisionChangeId: import_smithy_client.expectString,
    revisionId: import_smithy_client.expectString
  });
}, "de_ActionRevision");
var de_ActionState = /* @__PURE__ */ __name((output, context) => {
  return (0, import_smithy_client.take)(output, {
    actionName: import_smithy_client.expectString,
    currentRevision: (_) => de_ActionRevision(_, context),
    entityUrl: import_smithy_client.expectString,
    latestExecution: (_) => de_ActionExecution(_, context),
    revisionUrl: import_smithy_client.expectString
  });
}, "de_ActionState");
var de_ActionStateList = /* @__PURE__ */ __name((output, context) => {
  const retVal = (output || []).filter((e) => e != null).map((entry) => {
    return de_ActionState(entry, context);
  });
  return retVal;
}, "de_ActionStateList");
var de_ArtifactRevision = /* @__PURE__ */ __name((output, context) => {
  return (0, import_smithy_client.take)(output, {
    created: (_) => (0, import_smithy_client.expectNonNull)((0, import_smithy_client.parseEpochTimestamp)((0, import_smithy_client.expectNumber)(_))),
    name: import_smithy_client.expectString,
    revisionChangeIdentifier: import_smithy_client.expectString,
    revisionId: import_smithy_client.expectString,
    revisionSummary: import_smithy_client.expectString,
    revisionUrl: import_smithy_client.expectString
  });
}, "de_ArtifactRevision");
var de_ArtifactRevisionList = /* @__PURE__ */ __name((output, context) => {
  const retVal = (output || []).filter((e) => e != null).map((entry) => {
    return de_ArtifactRevision(entry, context);
  });
  return retVal;
}, "de_ArtifactRevisionList");
var de_GetPipelineExecutionOutput = /* @__PURE__ */ __name((output, context) => {
  return (0, import_smithy_client.take)(output, {
    pipelineExecution: (_) => de_PipelineExecution(_, context)
  });
}, "de_GetPipelineExecutionOutput");
var de_GetPipelineOutput = /* @__PURE__ */ __name((output, context) => {
  return (0, import_smithy_client.take)(output, {
    metadata: (_) => de_PipelineMetadata(_, context),
    pipeline: import_smithy_client._json
  });
}, "de_GetPipelineOutput");
var de_GetPipelineStateOutput = /* @__PURE__ */ __name((output, context) => {
  return (0, import_smithy_client.take)(output, {
    created: (_) => (0, import_smithy_client.expectNonNull)((0, import_smithy_client.parseEpochTimestamp)((0, import_smithy_client.expectNumber)(_))),
    pipelineName: import_smithy_client.expectString,
    pipelineVersion: import_smithy_client.expectInt32,
    stageStates: (_) => de_StageStateList(_, context),
    updated: (_) => (0, import_smithy_client.expectNonNull)((0, import_smithy_client.parseEpochTimestamp)((0, import_smithy_client.expectNumber)(_)))
  });
}, "de_GetPipelineStateOutput");
var de_ListActionExecutionsOutput = /* @__PURE__ */ __name((output, context) => {
  return (0, import_smithy_client.take)(output, {
    actionExecutionDetails: (_) => de_ActionExecutionDetailList(_, context),
    nextToken: import_smithy_client.expectString
  });
}, "de_ListActionExecutionsOutput");
var de_ListPipelineExecutionsOutput = /* @__PURE__ */ __name((output, context) => {
  return (0, import_smithy_client.take)(output, {
    nextToken: import_smithy_client.expectString,
    pipelineExecutionSummaries: (_) => de_PipelineExecutionSummaryList(_, context)
  });
}, "de_ListPipelineExecutionsOutput");
var de_ListPipelinesOutput = /* @__PURE__ */ __name((output, context) => {
  return (0, import_smithy_client.take)(output, {
    nextToken: import_smithy_client.expectString,
    pipelines: (_) => de_PipelineList(_, context)
  });
}, "de_ListPipelinesOutput");
var de_ListWebhookItem = /* @__PURE__ */ __name((output, context) => {
  return (0, import_smithy_client.take)(output, {
    arn: import_smithy_client.expectString,
    definition: import_smithy_client._json,
    errorCode: import_smithy_client.expectString,
    errorMessage: import_smithy_client.expectString,
    lastTriggered: (_) => (0, import_smithy_client.expectNonNull)((0, import_smithy_client.parseEpochTimestamp)((0, import_smithy_client.expectNumber)(_))),
    tags: import_smithy_client._json,
    url: import_smithy_client.expectString
  });
}, "de_ListWebhookItem");
var de_ListWebhooksOutput = /* @__PURE__ */ __name((output, context) => {
  return (0, import_smithy_client.take)(output, {
    NextToken: import_smithy_client.expectString,
    webhooks: (_) => de_WebhookList(_, context)
  });
}, "de_ListWebhooksOutput");
var de_PipelineExecution = /* @__PURE__ */ __name((output, context) => {
  return (0, import_smithy_client.take)(output, {
    artifactRevisions: (_) => de_ArtifactRevisionList(_, context),
    executionMode: import_smithy_client.expectString,
    pipelineExecutionId: import_smithy_client.expectString,
    pipelineName: import_smithy_client.expectString,
    pipelineVersion: import_smithy_client.expectInt32,
    status: import_smithy_client.expectString,
    statusSummary: import_smithy_client.expectString,
    trigger: import_smithy_client._json,
    variables: import_smithy_client._json
  });
}, "de_PipelineExecution");
var de_PipelineExecutionSummary = /* @__PURE__ */ __name((output, context) => {
  return (0, import_smithy_client.take)(output, {
    executionMode: import_smithy_client.expectString,
    lastUpdateTime: (_) => (0, import_smithy_client.expectNonNull)((0, import_smithy_client.parseEpochTimestamp)((0, import_smithy_client.expectNumber)(_))),
    pipelineExecutionId: import_smithy_client.expectString,
    sourceRevisions: import_smithy_client._json,
    startTime: (_) => (0, import_smithy_client.expectNonNull)((0, import_smithy_client.parseEpochTimestamp)((0, import_smithy_client.expectNumber)(_))),
    status: import_smithy_client.expectString,
    stopTrigger: import_smithy_client._json,
    trigger: import_smithy_client._json
  });
}, "de_PipelineExecutionSummary");
var de_PipelineExecutionSummaryList = /* @__PURE__ */ __name((output, context) => {
  const retVal = (output || []).filter((e) => e != null).map((entry) => {
    return de_PipelineExecutionSummary(entry, context);
  });
  return retVal;
}, "de_PipelineExecutionSummaryList");
var de_PipelineList = /* @__PURE__ */ __name((output, context) => {
  const retVal = (output || []).filter((e) => e != null).map((entry) => {
    return de_PipelineSummary(entry, context);
  });
  return retVal;
}, "de_PipelineList");
var de_PipelineMetadata = /* @__PURE__ */ __name((output, context) => {
  return (0, import_smithy_client.take)(output, {
    created: (_) => (0, import_smithy_client.expectNonNull)((0, import_smithy_client.parseEpochTimestamp)((0, import_smithy_client.expectNumber)(_))),
    pipelineArn: import_smithy_client.expectString,
    pollingDisabledAt: (_) => (0, import_smithy_client.expectNonNull)((0, import_smithy_client.parseEpochTimestamp)((0, import_smithy_client.expectNumber)(_))),
    updated: (_) => (0, import_smithy_client.expectNonNull)((0, import_smithy_client.parseEpochTimestamp)((0, import_smithy_client.expectNumber)(_)))
  });
}, "de_PipelineMetadata");
var de_PipelineSummary = /* @__PURE__ */ __name((output, context) => {
  return (0, import_smithy_client.take)(output, {
    created: (_) => (0, import_smithy_client.expectNonNull)((0, import_smithy_client.parseEpochTimestamp)((0, import_smithy_client.expectNumber)(_))),
    executionMode: import_smithy_client.expectString,
    name: import_smithy_client.expectString,
    pipelineType: import_smithy_client.expectString,
    updated: (_) => (0, import_smithy_client.expectNonNull)((0, import_smithy_client.parseEpochTimestamp)((0, import_smithy_client.expectNumber)(_))),
    version: import_smithy_client.expectInt32
  });
}, "de_PipelineSummary");
var de_PutApprovalResultOutput = /* @__PURE__ */ __name((output, context) => {
  return (0, import_smithy_client.take)(output, {
    approvedAt: (_) => (0, import_smithy_client.expectNonNull)((0, import_smithy_client.parseEpochTimestamp)((0, import_smithy_client.expectNumber)(_)))
  });
}, "de_PutApprovalResultOutput");
var de_PutWebhookOutput = /* @__PURE__ */ __name((output, context) => {
  return (0, import_smithy_client.take)(output, {
    webhook: (_) => de_ListWebhookItem(_, context)
  });
}, "de_PutWebhookOutput");
var de_StageState = /* @__PURE__ */ __name((output, context) => {
  return (0, import_smithy_client.take)(output, {
    actionStates: (_) => de_ActionStateList(_, context),
    inboundExecution: import_smithy_client._json,
    inboundExecutions: import_smithy_client._json,
    inboundTransitionState: (_) => de_TransitionState(_, context),
    latestExecution: import_smithy_client._json,
    stageName: import_smithy_client.expectString
  });
}, "de_StageState");
var de_StageStateList = /* @__PURE__ */ __name((output, context) => {
  const retVal = (output || []).filter((e) => e != null).map((entry) => {
    return de_StageState(entry, context);
  });
  return retVal;
}, "de_StageStateList");
var de_TransitionState = /* @__PURE__ */ __name((output, context) => {
  return (0, import_smithy_client.take)(output, {
    disabledReason: import_smithy_client.expectString,
    enabled: import_smithy_client.expectBoolean,
    lastChangedAt: (_) => (0, import_smithy_client.expectNonNull)((0, import_smithy_client.parseEpochTimestamp)((0, import_smithy_client.expectNumber)(_))),
    lastChangedBy: import_smithy_client.expectString
  });
}, "de_TransitionState");
var de_WebhookList = /* @__PURE__ */ __name((output, context) => {
  const retVal = (output || []).filter((e) => e != null).map((entry) => {
    return de_ListWebhookItem(entry, context);
  });
  return retVal;
}, "de_WebhookList");
var deserializeMetadata = /* @__PURE__ */ __name((output) => ({
  httpStatusCode: output.statusCode,
  requestId: output.headers["x-amzn-requestid"] ?? output.headers["x-amzn-request-id"] ?? output.headers["x-amz-request-id"],
  extendedRequestId: output.headers["x-amz-id-2"],
  cfId: output.headers["x-amz-cf-id"]
}), "deserializeMetadata");
var collectBodyString = /* @__PURE__ */ __name((streamBody, context) => (0, import_smithy_client.collectBody)(streamBody, context).then((body) => context.utf8Encoder(body)), "collectBodyString");
var throwDefaultError = (0, import_smithy_client.withBaseException)(CodePipelineServiceException);
var buildHttpRpcRequest = /* @__PURE__ */ __name(async (context, headers, path, resolvedHostname, body) => {
  const { hostname, protocol = "https", port, path: basePath } = await context.endpoint();
  const contents = {
    protocol,
    hostname,
    port,
    method: "POST",
    path: basePath.endsWith("/") ? basePath.slice(0, -1) + path : basePath + path,
    headers
  };
  if (resolvedHostname !== void 0) {
    contents.hostname = resolvedHostname;
  }
  if (body !== void 0) {
    contents.body = body;
  }
  return new import_protocol_http.HttpRequest(contents);
}, "buildHttpRpcRequest");
function sharedHeaders(operation) {
  return {
    "content-type": "application/x-amz-json-1.1",
    "x-amz-target": `CodePipeline_20150709.${operation}`
  };
}
__name(sharedHeaders, "sharedHeaders");
var parseBody = /* @__PURE__ */ __name((streamBody, context) => collectBodyString(streamBody, context).then((encoded) => {
  if (encoded.length) {
    return JSON.parse(encoded);
  }
  return {};
}), "parseBody");
var parseErrorBody = /* @__PURE__ */ __name(async (errorBody, context) => {
  const value = await parseBody(errorBody, context);
  value.message = value.message ?? value.Message;
  return value;
}, "parseErrorBody");
var loadRestJsonErrorCode = /* @__PURE__ */ __name((output, data) => {
  const findKey = /* @__PURE__ */ __name((object, key) => Object.keys(object).find((k) => k.toLowerCase() === key.toLowerCase()), "findKey");
  const sanitizeErrorCode = /* @__PURE__ */ __name((rawValue) => {
    let cleanValue = rawValue;
    if (typeof cleanValue === "number") {
      cleanValue = cleanValue.toString();
    }
    if (cleanValue.indexOf(",") >= 0) {
      cleanValue = cleanValue.split(",")[0];
    }
    if (cleanValue.indexOf(":") >= 0) {
      cleanValue = cleanValue.split(":")[0];
    }
    if (cleanValue.indexOf("#") >= 0) {
      cleanValue = cleanValue.split("#")[1];
    }
    return cleanValue;
  }, "sanitizeErrorCode");
  const headerKey = findKey(output.headers, "x-amzn-errortype");
  if (headerKey !== void 0) {
    return sanitizeErrorCode(output.headers[headerKey]);
  }
  if (data.code !== void 0) {
    return sanitizeErrorCode(data.code);
  }
  if (data["__type"] !== void 0) {
    return sanitizeErrorCode(data["__type"]);
  }
}, "loadRestJsonErrorCode");

// src/commands/AcknowledgeJobCommand.ts
var _AcknowledgeJobCommand = class _AcknowledgeJobCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("CodePipeline_20150709", "AcknowledgeJob", {}).n("CodePipelineClient", "AcknowledgeJobCommand").f(void 0, void 0).ser(se_AcknowledgeJobCommand).de(de_AcknowledgeJobCommand).build() {
};
__name(_AcknowledgeJobCommand, "AcknowledgeJobCommand");
var AcknowledgeJobCommand = _AcknowledgeJobCommand;

// src/commands/AcknowledgeThirdPartyJobCommand.ts




var _AcknowledgeThirdPartyJobCommand = class _AcknowledgeThirdPartyJobCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("CodePipeline_20150709", "AcknowledgeThirdPartyJob", {}).n("CodePipelineClient", "AcknowledgeThirdPartyJobCommand").f(void 0, void 0).ser(se_AcknowledgeThirdPartyJobCommand).de(de_AcknowledgeThirdPartyJobCommand).build() {
};
__name(_AcknowledgeThirdPartyJobCommand, "AcknowledgeThirdPartyJobCommand");
var AcknowledgeThirdPartyJobCommand = _AcknowledgeThirdPartyJobCommand;

// src/commands/CreateCustomActionTypeCommand.ts




var _CreateCustomActionTypeCommand = class _CreateCustomActionTypeCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("CodePipeline_20150709", "CreateCustomActionType", {}).n("CodePipelineClient", "CreateCustomActionTypeCommand").f(void 0, void 0).ser(se_CreateCustomActionTypeCommand).de(de_CreateCustomActionTypeCommand).build() {
};
__name(_CreateCustomActionTypeCommand, "CreateCustomActionTypeCommand");
var CreateCustomActionTypeCommand = _CreateCustomActionTypeCommand;

// src/commands/CreatePipelineCommand.ts




var _CreatePipelineCommand = class _CreatePipelineCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("CodePipeline_20150709", "CreatePipeline", {}).n("CodePipelineClient", "CreatePipelineCommand").f(void 0, void 0).ser(se_CreatePipelineCommand).de(de_CreatePipelineCommand).build() {
};
__name(_CreatePipelineCommand, "CreatePipelineCommand");
var CreatePipelineCommand = _CreatePipelineCommand;

// src/commands/DeleteCustomActionTypeCommand.ts




var _DeleteCustomActionTypeCommand = class _DeleteCustomActionTypeCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("CodePipeline_20150709", "DeleteCustomActionType", {}).n("CodePipelineClient", "DeleteCustomActionTypeCommand").f(void 0, void 0).ser(se_DeleteCustomActionTypeCommand).de(de_DeleteCustomActionTypeCommand).build() {
};
__name(_DeleteCustomActionTypeCommand, "DeleteCustomActionTypeCommand");
var DeleteCustomActionTypeCommand = _DeleteCustomActionTypeCommand;

// src/commands/DeletePipelineCommand.ts




var _DeletePipelineCommand = class _DeletePipelineCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("CodePipeline_20150709", "DeletePipeline", {}).n("CodePipelineClient", "DeletePipelineCommand").f(void 0, void 0).ser(se_DeletePipelineCommand).de(de_DeletePipelineCommand).build() {
};
__name(_DeletePipelineCommand, "DeletePipelineCommand");
var DeletePipelineCommand = _DeletePipelineCommand;

// src/commands/DeleteWebhookCommand.ts




var _DeleteWebhookCommand = class _DeleteWebhookCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("CodePipeline_20150709", "DeleteWebhook", {}).n("CodePipelineClient", "DeleteWebhookCommand").f(void 0, void 0).ser(se_DeleteWebhookCommand).de(de_DeleteWebhookCommand).build() {
};
__name(_DeleteWebhookCommand, "DeleteWebhookCommand");
var DeleteWebhookCommand = _DeleteWebhookCommand;

// src/commands/DeregisterWebhookWithThirdPartyCommand.ts




var _DeregisterWebhookWithThirdPartyCommand = class _DeregisterWebhookWithThirdPartyCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("CodePipeline_20150709", "DeregisterWebhookWithThirdParty", {}).n("CodePipelineClient", "DeregisterWebhookWithThirdPartyCommand").f(void 0, void 0).ser(se_DeregisterWebhookWithThirdPartyCommand).de(de_DeregisterWebhookWithThirdPartyCommand).build() {
};
__name(_DeregisterWebhookWithThirdPartyCommand, "DeregisterWebhookWithThirdPartyCommand");
var DeregisterWebhookWithThirdPartyCommand = _DeregisterWebhookWithThirdPartyCommand;

// src/commands/DisableStageTransitionCommand.ts




var _DisableStageTransitionCommand = class _DisableStageTransitionCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("CodePipeline_20150709", "DisableStageTransition", {}).n("CodePipelineClient", "DisableStageTransitionCommand").f(void 0, void 0).ser(se_DisableStageTransitionCommand).de(de_DisableStageTransitionCommand).build() {
};
__name(_DisableStageTransitionCommand, "DisableStageTransitionCommand");
var DisableStageTransitionCommand = _DisableStageTransitionCommand;

// src/commands/EnableStageTransitionCommand.ts




var _EnableStageTransitionCommand = class _EnableStageTransitionCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("CodePipeline_20150709", "EnableStageTransition", {}).n("CodePipelineClient", "EnableStageTransitionCommand").f(void 0, void 0).ser(se_EnableStageTransitionCommand).de(de_EnableStageTransitionCommand).build() {
};
__name(_EnableStageTransitionCommand, "EnableStageTransitionCommand");
var EnableStageTransitionCommand = _EnableStageTransitionCommand;

// src/commands/GetActionTypeCommand.ts




var _GetActionTypeCommand = class _GetActionTypeCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("CodePipeline_20150709", "GetActionType", {}).n("CodePipelineClient", "GetActionTypeCommand").f(void 0, void 0).ser(se_GetActionTypeCommand).de(de_GetActionTypeCommand).build() {
};
__name(_GetActionTypeCommand, "GetActionTypeCommand");
var GetActionTypeCommand = _GetActionTypeCommand;

// src/commands/GetJobDetailsCommand.ts




var _GetJobDetailsCommand = class _GetJobDetailsCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("CodePipeline_20150709", "GetJobDetails", {}).n("CodePipelineClient", "GetJobDetailsCommand").f(void 0, GetJobDetailsOutputFilterSensitiveLog).ser(se_GetJobDetailsCommand).de(de_GetJobDetailsCommand).build() {
};
__name(_GetJobDetailsCommand, "GetJobDetailsCommand");
var GetJobDetailsCommand = _GetJobDetailsCommand;

// src/commands/GetPipelineCommand.ts




var _GetPipelineCommand = class _GetPipelineCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("CodePipeline_20150709", "GetPipeline", {}).n("CodePipelineClient", "GetPipelineCommand").f(void 0, void 0).ser(se_GetPipelineCommand).de(de_GetPipelineCommand).build() {
};
__name(_GetPipelineCommand, "GetPipelineCommand");
var GetPipelineCommand = _GetPipelineCommand;

// src/commands/GetPipelineExecutionCommand.ts




var _GetPipelineExecutionCommand = class _GetPipelineExecutionCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("CodePipeline_20150709", "GetPipelineExecution", {}).n("CodePipelineClient", "GetPipelineExecutionCommand").f(void 0, void 0).ser(se_GetPipelineExecutionCommand).de(de_GetPipelineExecutionCommand).build() {
};
__name(_GetPipelineExecutionCommand, "GetPipelineExecutionCommand");
var GetPipelineExecutionCommand = _GetPipelineExecutionCommand;

// src/commands/GetPipelineStateCommand.ts




var _GetPipelineStateCommand = class _GetPipelineStateCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("CodePipeline_20150709", "GetPipelineState", {}).n("CodePipelineClient", "GetPipelineStateCommand").f(void 0, void 0).ser(se_GetPipelineStateCommand).de(de_GetPipelineStateCommand).build() {
};
__name(_GetPipelineStateCommand, "GetPipelineStateCommand");
var GetPipelineStateCommand = _GetPipelineStateCommand;

// src/commands/GetThirdPartyJobDetailsCommand.ts




var _GetThirdPartyJobDetailsCommand = class _GetThirdPartyJobDetailsCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("CodePipeline_20150709", "GetThirdPartyJobDetails", {}).n("CodePipelineClient", "GetThirdPartyJobDetailsCommand").f(void 0, GetThirdPartyJobDetailsOutputFilterSensitiveLog).ser(se_GetThirdPartyJobDetailsCommand).de(de_GetThirdPartyJobDetailsCommand).build() {
};
__name(_GetThirdPartyJobDetailsCommand, "GetThirdPartyJobDetailsCommand");
var GetThirdPartyJobDetailsCommand = _GetThirdPartyJobDetailsCommand;

// src/commands/ListActionExecutionsCommand.ts




var _ListActionExecutionsCommand = class _ListActionExecutionsCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("CodePipeline_20150709", "ListActionExecutions", {}).n("CodePipelineClient", "ListActionExecutionsCommand").f(void 0, void 0).ser(se_ListActionExecutionsCommand).de(de_ListActionExecutionsCommand).build() {
};
__name(_ListActionExecutionsCommand, "ListActionExecutionsCommand");
var ListActionExecutionsCommand = _ListActionExecutionsCommand;

// src/commands/ListActionTypesCommand.ts




var _ListActionTypesCommand = class _ListActionTypesCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("CodePipeline_20150709", "ListActionTypes", {}).n("CodePipelineClient", "ListActionTypesCommand").f(void 0, void 0).ser(se_ListActionTypesCommand).de(de_ListActionTypesCommand).build() {
};
__name(_ListActionTypesCommand, "ListActionTypesCommand");
var ListActionTypesCommand = _ListActionTypesCommand;

// src/commands/ListPipelineExecutionsCommand.ts




var _ListPipelineExecutionsCommand = class _ListPipelineExecutionsCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("CodePipeline_20150709", "ListPipelineExecutions", {}).n("CodePipelineClient", "ListPipelineExecutionsCommand").f(void 0, void 0).ser(se_ListPipelineExecutionsCommand).de(de_ListPipelineExecutionsCommand).build() {
};
__name(_ListPipelineExecutionsCommand, "ListPipelineExecutionsCommand");
var ListPipelineExecutionsCommand = _ListPipelineExecutionsCommand;

// src/commands/ListPipelinesCommand.ts




var _ListPipelinesCommand = class _ListPipelinesCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("CodePipeline_20150709", "ListPipelines", {}).n("CodePipelineClient", "ListPipelinesCommand").f(void 0, void 0).ser(se_ListPipelinesCommand).de(de_ListPipelinesCommand).build() {
};
__name(_ListPipelinesCommand, "ListPipelinesCommand");
var ListPipelinesCommand = _ListPipelinesCommand;

// src/commands/ListTagsForResourceCommand.ts




var _ListTagsForResourceCommand = class _ListTagsForResourceCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("CodePipeline_20150709", "ListTagsForResource", {}).n("CodePipelineClient", "ListTagsForResourceCommand").f(void 0, void 0).ser(se_ListTagsForResourceCommand).de(de_ListTagsForResourceCommand).build() {
};
__name(_ListTagsForResourceCommand, "ListTagsForResourceCommand");
var ListTagsForResourceCommand = _ListTagsForResourceCommand;

// src/commands/ListWebhooksCommand.ts




var _ListWebhooksCommand = class _ListWebhooksCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("CodePipeline_20150709", "ListWebhooks", {}).n("CodePipelineClient", "ListWebhooksCommand").f(void 0, void 0).ser(se_ListWebhooksCommand).de(de_ListWebhooksCommand).build() {
};
__name(_ListWebhooksCommand, "ListWebhooksCommand");
var ListWebhooksCommand = _ListWebhooksCommand;

// src/commands/PollForJobsCommand.ts




var _PollForJobsCommand = class _PollForJobsCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("CodePipeline_20150709", "PollForJobs", {}).n("CodePipelineClient", "PollForJobsCommand").f(void 0, PollForJobsOutputFilterSensitiveLog).ser(se_PollForJobsCommand).de(de_PollForJobsCommand).build() {
};
__name(_PollForJobsCommand, "PollForJobsCommand");
var PollForJobsCommand = _PollForJobsCommand;

// src/commands/PollForThirdPartyJobsCommand.ts




var _PollForThirdPartyJobsCommand = class _PollForThirdPartyJobsCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("CodePipeline_20150709", "PollForThirdPartyJobs", {}).n("CodePipelineClient", "PollForThirdPartyJobsCommand").f(void 0, void 0).ser(se_PollForThirdPartyJobsCommand).de(de_PollForThirdPartyJobsCommand).build() {
};
__name(_PollForThirdPartyJobsCommand, "PollForThirdPartyJobsCommand");
var PollForThirdPartyJobsCommand = _PollForThirdPartyJobsCommand;

// src/commands/PutActionRevisionCommand.ts




var _PutActionRevisionCommand = class _PutActionRevisionCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("CodePipeline_20150709", "PutActionRevision", {}).n("CodePipelineClient", "PutActionRevisionCommand").f(void 0, void 0).ser(se_PutActionRevisionCommand).de(de_PutActionRevisionCommand).build() {
};
__name(_PutActionRevisionCommand, "PutActionRevisionCommand");
var PutActionRevisionCommand = _PutActionRevisionCommand;

// src/commands/PutApprovalResultCommand.ts




var _PutApprovalResultCommand = class _PutApprovalResultCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("CodePipeline_20150709", "PutApprovalResult", {}).n("CodePipelineClient", "PutApprovalResultCommand").f(void 0, void 0).ser(se_PutApprovalResultCommand).de(de_PutApprovalResultCommand).build() {
};
__name(_PutApprovalResultCommand, "PutApprovalResultCommand");
var PutApprovalResultCommand = _PutApprovalResultCommand;

// src/commands/PutJobFailureResultCommand.ts




var _PutJobFailureResultCommand = class _PutJobFailureResultCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("CodePipeline_20150709", "PutJobFailureResult", {}).n("CodePipelineClient", "PutJobFailureResultCommand").f(void 0, void 0).ser(se_PutJobFailureResultCommand).de(de_PutJobFailureResultCommand).build() {
};
__name(_PutJobFailureResultCommand, "PutJobFailureResultCommand");
var PutJobFailureResultCommand = _PutJobFailureResultCommand;

// src/commands/PutJobSuccessResultCommand.ts




var _PutJobSuccessResultCommand = class _PutJobSuccessResultCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("CodePipeline_20150709", "PutJobSuccessResult", {}).n("CodePipelineClient", "PutJobSuccessResultCommand").f(void 0, void 0).ser(se_PutJobSuccessResultCommand).de(de_PutJobSuccessResultCommand).build() {
};
__name(_PutJobSuccessResultCommand, "PutJobSuccessResultCommand");
var PutJobSuccessResultCommand = _PutJobSuccessResultCommand;

// src/commands/PutThirdPartyJobFailureResultCommand.ts




var _PutThirdPartyJobFailureResultCommand = class _PutThirdPartyJobFailureResultCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("CodePipeline_20150709", "PutThirdPartyJobFailureResult", {}).n("CodePipelineClient", "PutThirdPartyJobFailureResultCommand").f(void 0, void 0).ser(se_PutThirdPartyJobFailureResultCommand).de(de_PutThirdPartyJobFailureResultCommand).build() {
};
__name(_PutThirdPartyJobFailureResultCommand, "PutThirdPartyJobFailureResultCommand");
var PutThirdPartyJobFailureResultCommand = _PutThirdPartyJobFailureResultCommand;

// src/commands/PutThirdPartyJobSuccessResultCommand.ts




var _PutThirdPartyJobSuccessResultCommand = class _PutThirdPartyJobSuccessResultCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("CodePipeline_20150709", "PutThirdPartyJobSuccessResult", {}).n("CodePipelineClient", "PutThirdPartyJobSuccessResultCommand").f(void 0, void 0).ser(se_PutThirdPartyJobSuccessResultCommand).de(de_PutThirdPartyJobSuccessResultCommand).build() {
};
__name(_PutThirdPartyJobSuccessResultCommand, "PutThirdPartyJobSuccessResultCommand");
var PutThirdPartyJobSuccessResultCommand = _PutThirdPartyJobSuccessResultCommand;

// src/commands/PutWebhookCommand.ts




var _PutWebhookCommand = class _PutWebhookCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("CodePipeline_20150709", "PutWebhook", {}).n("CodePipelineClient", "PutWebhookCommand").f(void 0, void 0).ser(se_PutWebhookCommand).de(de_PutWebhookCommand).build() {
};
__name(_PutWebhookCommand, "PutWebhookCommand");
var PutWebhookCommand = _PutWebhookCommand;

// src/commands/RegisterWebhookWithThirdPartyCommand.ts




var _RegisterWebhookWithThirdPartyCommand = class _RegisterWebhookWithThirdPartyCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("CodePipeline_20150709", "RegisterWebhookWithThirdParty", {}).n("CodePipelineClient", "RegisterWebhookWithThirdPartyCommand").f(void 0, void 0).ser(se_RegisterWebhookWithThirdPartyCommand).de(de_RegisterWebhookWithThirdPartyCommand).build() {
};
__name(_RegisterWebhookWithThirdPartyCommand, "RegisterWebhookWithThirdPartyCommand");
var RegisterWebhookWithThirdPartyCommand = _RegisterWebhookWithThirdPartyCommand;

// src/commands/RetryStageExecutionCommand.ts




var _RetryStageExecutionCommand = class _RetryStageExecutionCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("CodePipeline_20150709", "RetryStageExecution", {}).n("CodePipelineClient", "RetryStageExecutionCommand").f(void 0, void 0).ser(se_RetryStageExecutionCommand).de(de_RetryStageExecutionCommand).build() {
};
__name(_RetryStageExecutionCommand, "RetryStageExecutionCommand");
var RetryStageExecutionCommand = _RetryStageExecutionCommand;

// src/commands/StartPipelineExecutionCommand.ts




var _StartPipelineExecutionCommand = class _StartPipelineExecutionCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("CodePipeline_20150709", "StartPipelineExecution", {}).n("CodePipelineClient", "StartPipelineExecutionCommand").f(void 0, void 0).ser(se_StartPipelineExecutionCommand).de(de_StartPipelineExecutionCommand).build() {
};
__name(_StartPipelineExecutionCommand, "StartPipelineExecutionCommand");
var StartPipelineExecutionCommand = _StartPipelineExecutionCommand;

// src/commands/StopPipelineExecutionCommand.ts




var _StopPipelineExecutionCommand = class _StopPipelineExecutionCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("CodePipeline_20150709", "StopPipelineExecution", {}).n("CodePipelineClient", "StopPipelineExecutionCommand").f(void 0, void 0).ser(se_StopPipelineExecutionCommand).de(de_StopPipelineExecutionCommand).build() {
};
__name(_StopPipelineExecutionCommand, "StopPipelineExecutionCommand");
var StopPipelineExecutionCommand = _StopPipelineExecutionCommand;

// src/commands/TagResourceCommand.ts




var _TagResourceCommand = class _TagResourceCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("CodePipeline_20150709", "TagResource", {}).n("CodePipelineClient", "TagResourceCommand").f(void 0, void 0).ser(se_TagResourceCommand).de(de_TagResourceCommand).build() {
};
__name(_TagResourceCommand, "TagResourceCommand");
var TagResourceCommand = _TagResourceCommand;

// src/commands/UntagResourceCommand.ts




var _UntagResourceCommand = class _UntagResourceCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("CodePipeline_20150709", "UntagResource", {}).n("CodePipelineClient", "UntagResourceCommand").f(void 0, void 0).ser(se_UntagResourceCommand).de(de_UntagResourceCommand).build() {
};
__name(_UntagResourceCommand, "UntagResourceCommand");
var UntagResourceCommand = _UntagResourceCommand;

// src/commands/UpdateActionTypeCommand.ts




var _UpdateActionTypeCommand = class _UpdateActionTypeCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("CodePipeline_20150709", "UpdateActionType", {}).n("CodePipelineClient", "UpdateActionTypeCommand").f(void 0, void 0).ser(se_UpdateActionTypeCommand).de(de_UpdateActionTypeCommand).build() {
};
__name(_UpdateActionTypeCommand, "UpdateActionTypeCommand");
var UpdateActionTypeCommand = _UpdateActionTypeCommand;

// src/commands/UpdatePipelineCommand.ts




var _UpdatePipelineCommand = class _UpdatePipelineCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("CodePipeline_20150709", "UpdatePipeline", {}).n("CodePipelineClient", "UpdatePipelineCommand").f(void 0, void 0).ser(se_UpdatePipelineCommand).de(de_UpdatePipelineCommand).build() {
};
__name(_UpdatePipelineCommand, "UpdatePipelineCommand");
var UpdatePipelineCommand = _UpdatePipelineCommand;

// src/CodePipeline.ts
var commands = {
  AcknowledgeJobCommand,
  AcknowledgeThirdPartyJobCommand,
  CreateCustomActionTypeCommand,
  CreatePipelineCommand,
  DeleteCustomActionTypeCommand,
  DeletePipelineCommand,
  DeleteWebhookCommand,
  DeregisterWebhookWithThirdPartyCommand,
  DisableStageTransitionCommand,
  EnableStageTransitionCommand,
  GetActionTypeCommand,
  GetJobDetailsCommand,
  GetPipelineCommand,
  GetPipelineExecutionCommand,
  GetPipelineStateCommand,
  GetThirdPartyJobDetailsCommand,
  ListActionExecutionsCommand,
  ListActionTypesCommand,
  ListPipelineExecutionsCommand,
  ListPipelinesCommand,
  ListTagsForResourceCommand,
  ListWebhooksCommand,
  PollForJobsCommand,
  PollForThirdPartyJobsCommand,
  PutActionRevisionCommand,
  PutApprovalResultCommand,
  PutJobFailureResultCommand,
  PutJobSuccessResultCommand,
  PutThirdPartyJobFailureResultCommand,
  PutThirdPartyJobSuccessResultCommand,
  PutWebhookCommand,
  RegisterWebhookWithThirdPartyCommand,
  RetryStageExecutionCommand,
  StartPipelineExecutionCommand,
  StopPipelineExecutionCommand,
  TagResourceCommand,
  UntagResourceCommand,
  UpdateActionTypeCommand,
  UpdatePipelineCommand
};
var _CodePipeline = class _CodePipeline extends CodePipelineClient {
};
__name(_CodePipeline, "CodePipeline");
var CodePipeline = _CodePipeline;
(0, import_smithy_client.createAggregatedClient)(commands, CodePipeline);

// src/pagination/ListActionExecutionsPaginator.ts

var paginateListActionExecutions = (0, import_core.createPaginator)(CodePipelineClient, ListActionExecutionsCommand, "nextToken", "nextToken", "maxResults");

// src/pagination/ListActionTypesPaginator.ts

var paginateListActionTypes = (0, import_core.createPaginator)(CodePipelineClient, ListActionTypesCommand, "nextToken", "nextToken", "");

// src/pagination/ListPipelineExecutionsPaginator.ts

var paginateListPipelineExecutions = (0, import_core.createPaginator)(CodePipelineClient, ListPipelineExecutionsCommand, "nextToken", "nextToken", "maxResults");

// src/pagination/ListPipelinesPaginator.ts

var paginateListPipelines = (0, import_core.createPaginator)(CodePipelineClient, ListPipelinesCommand, "nextToken", "nextToken", "maxResults");

// src/pagination/ListTagsForResourcePaginator.ts

var paginateListTagsForResource = (0, import_core.createPaginator)(CodePipelineClient, ListTagsForResourceCommand, "nextToken", "nextToken", "maxResults");

// src/pagination/ListWebhooksPaginator.ts

var paginateListWebhooks = (0, import_core.createPaginator)(CodePipelineClient, ListWebhooksCommand, "NextToken", "NextToken", "MaxResults");

// src/index.ts
var import_util_endpoints = require("@aws-sdk/util-endpoints");
// Annotate the CommonJS export names for ESM import in node:

0 && (module.exports = {
  CodePipelineServiceException,
  __Client,
  CodePipelineClient,
  CodePipeline,
  $Command,
  AcknowledgeJobCommand,
  AcknowledgeThirdPartyJobCommand,
  CreateCustomActionTypeCommand,
  CreatePipelineCommand,
  DeleteCustomActionTypeCommand,
  DeletePipelineCommand,
  DeleteWebhookCommand,
  DeregisterWebhookWithThirdPartyCommand,
  DisableStageTransitionCommand,
  EnableStageTransitionCommand,
  GetActionTypeCommand,
  GetJobDetailsCommand,
  GetPipelineCommand,
  GetPipelineExecutionCommand,
  GetPipelineStateCommand,
  GetThirdPartyJobDetailsCommand,
  ListActionExecutionsCommand,
  ListActionTypesCommand,
  ListPipelineExecutionsCommand,
  ListPipelinesCommand,
  ListTagsForResourceCommand,
  ListWebhooksCommand,
  PollForJobsCommand,
  PollForThirdPartyJobsCommand,
  PutActionRevisionCommand,
  PutApprovalResultCommand,
  PutJobFailureResultCommand,
  PutJobSuccessResultCommand,
  PutThirdPartyJobFailureResultCommand,
  PutThirdPartyJobSuccessResultCommand,
  PutWebhookCommand,
  RegisterWebhookWithThirdPartyCommand,
  RetryStageExecutionCommand,
  StartPipelineExecutionCommand,
  StopPipelineExecutionCommand,
  TagResourceCommand,
  UntagResourceCommand,
  UpdateActionTypeCommand,
  UpdatePipelineCommand,
  paginateListActionExecutions,
  paginateListActionTypes,
  paginateListPipelineExecutions,
  paginateListPipelines,
  paginateListTagsForResource,
  paginateListWebhooks,
  JobStatus,
  InvalidNonceException,
  JobNotFoundException,
  ValidationException,
  InvalidClientTokenException,
  ActionCategory,
  ActionConfigurationPropertyType,
  ActionOwner,
  ActionExecutionStatus,
  StartTimeRange,
  ActionNotFoundException,
  ExecutorType,
  ActionTypeNotFoundException,
  ApprovalAlreadyCompletedException,
  ApprovalStatus,
  ArtifactLocationType,
  EncryptionKeyType,
  ArtifactStoreType,
  BlockerType,
  ConcurrentModificationException,
  InvalidTagsException,
  LimitExceededException,
  TooManyTagsException,
  ExecutionMode,
  PipelineType,
  GitPullRequestEventType,
  PipelineTriggerProviderType,
  InvalidActionDeclarationException,
  InvalidBlockerDeclarationException,
  InvalidStageDeclarationException,
  InvalidStructureException,
  PipelineNameInUseException,
  WebhookNotFoundException,
  StageTransitionType,
  PipelineNotFoundException,
  StageNotFoundException,
  PipelineVersionNotFoundException,
  PipelineExecutionStatus,
  TriggerType,
  PipelineExecutionNotFoundException,
  StageExecutionStatus,
  InvalidJobException,
  InvalidNextTokenException,
  InvalidArnException,
  ResourceNotFoundException,
  WebhookAuthenticationType,
  InvalidApprovalTokenException,
  InvalidJobStateException,
  FailureType,
  OutputVariablesSizeExceededException,
  InvalidWebhookAuthenticationParametersException,
  InvalidWebhookFilterPatternException,
  ConflictException,
  NotLatestPipelineExecutionException,
  StageRetryMode,
  StageNotRetryableException,
  ConcurrentPipelineExecutionsLimitExceededException,
  SourceRevisionType,
  DuplicatedStopRequestException,
  PipelineExecutionNotStoppableException,
  RequestFailedException,
  AWSSessionCredentialsFilterSensitiveLog,
  JobDataFilterSensitiveLog,
  JobDetailsFilterSensitiveLog,
  GetJobDetailsOutputFilterSensitiveLog,
  ThirdPartyJobDataFilterSensitiveLog,
  ThirdPartyJobDetailsFilterSensitiveLog,
  GetThirdPartyJobDetailsOutputFilterSensitiveLog,
  JobFilterSensitiveLog,
  PollForJobsOutputFilterSensitiveLog
});
