// @azure/arm-mysql-flexible/dist-esm/src/models/index.js
/*
 * Copyright (c) Microsoft Corporation.
 * Licensed under the MIT License.
 *
 * Code generated by Microsoft (R) AutoRest Code Generator.
 * Changes may cause incorrect behavior and will be lost if the code is regenerated.
 */
/** Known values of {@link SkuTier} that the service accepts. */
export var KnownSkuTier;
(function (KnownSkuTier) {
    /** Burstable */
    KnownSkuTier["Burstable"] = "Burstable";
    /** GeneralPurpose */
    KnownSkuTier["GeneralPurpose"] = "GeneralPurpose";
    /** MemoryOptimized */
    KnownSkuTier["MemoryOptimized"] = "MemoryOptimized";
})(KnownSkuTier || (KnownSkuTier = {}));
/** Known values of {@link ServerVersion} that the service accepts. */
export var KnownServerVersion;
(function (KnownServerVersion) {
    /** Five7 */
    KnownServerVersion["Five7"] = "5.7";
    /** Eight021 */
    KnownServerVersion["Eight021"] = "8.0.21";
})(KnownServerVersion || (KnownServerVersion = {}));
/** Known values of {@link CreateMode} that the service accepts. */
export var KnownCreateMode;
(function (KnownCreateMode) {
    /** Default */
    KnownCreateMode["Default"] = "Default";
    /** PointInTimeRestore */
    KnownCreateMode["PointInTimeRestore"] = "PointInTimeRestore";
    /** Replica */
    KnownCreateMode["Replica"] = "Replica";
    /** GeoRestore */
    KnownCreateMode["GeoRestore"] = "GeoRestore";
})(KnownCreateMode || (KnownCreateMode = {}));
/** Known values of {@link ReplicationRole} that the service accepts. */
export var KnownReplicationRole;
(function (KnownReplicationRole) {
    /** None */
    KnownReplicationRole["None"] = "None";
    /** Source */
    KnownReplicationRole["Source"] = "Source";
    /** Replica */
    KnownReplicationRole["Replica"] = "Replica";
})(KnownReplicationRole || (KnownReplicationRole = {}));
/** Known values of {@link ServerState} that the service accepts. */
export var KnownServerState;
(function (KnownServerState) {
    /** Ready */
    KnownServerState["Ready"] = "Ready";
    /** Dropping */
    KnownServerState["Dropping"] = "Dropping";
    /** Disabled */
    KnownServerState["Disabled"] = "Disabled";
    /** Starting */
    KnownServerState["Starting"] = "Starting";
    /** Stopping */
    KnownServerState["Stopping"] = "Stopping";
    /** Stopped */
    KnownServerState["Stopped"] = "Stopped";
    /** Updating */
    KnownServerState["Updating"] = "Updating";
})(KnownServerState || (KnownServerState = {}));
/** Known values of {@link EnableStatusEnum} that the service accepts. */
export var KnownEnableStatusEnum;
(function (KnownEnableStatusEnum) {
    /** Enabled */
    KnownEnableStatusEnum["Enabled"] = "Enabled";
    /** Disabled */
    KnownEnableStatusEnum["Disabled"] = "Disabled";
})(KnownEnableStatusEnum || (KnownEnableStatusEnum = {}));
/** Known values of {@link HighAvailabilityMode} that the service accepts. */
export var KnownHighAvailabilityMode;
(function (KnownHighAvailabilityMode) {
    /** Disabled */
    KnownHighAvailabilityMode["Disabled"] = "Disabled";
    /** ZoneRedundant */
    KnownHighAvailabilityMode["ZoneRedundant"] = "ZoneRedundant";
    /** SameZone */
    KnownHighAvailabilityMode["SameZone"] = "SameZone";
})(KnownHighAvailabilityMode || (KnownHighAvailabilityMode = {}));
/** Known values of {@link HighAvailabilityState} that the service accepts. */
export var KnownHighAvailabilityState;
(function (KnownHighAvailabilityState) {
    /** NotEnabled */
    KnownHighAvailabilityState["NotEnabled"] = "NotEnabled";
    /** CreatingStandby */
    KnownHighAvailabilityState["CreatingStandby"] = "CreatingStandby";
    /** Healthy */
    KnownHighAvailabilityState["Healthy"] = "Healthy";
    /** FailingOver */
    KnownHighAvailabilityState["FailingOver"] = "FailingOver";
    /** RemovingStandby */
    KnownHighAvailabilityState["RemovingStandby"] = "RemovingStandby";
})(KnownHighAvailabilityState || (KnownHighAvailabilityState = {}));
/** Known values of {@link CreatedByType} that the service accepts. */
export var KnownCreatedByType;
(function (KnownCreatedByType) {
    /** User */
    KnownCreatedByType["User"] = "User";
    /** Application */
    KnownCreatedByType["Application"] = "Application";
    /** ManagedIdentity */
    KnownCreatedByType["ManagedIdentity"] = "ManagedIdentity";
    /** Key */
    KnownCreatedByType["Key"] = "Key";
})(KnownCreatedByType || (KnownCreatedByType = {}));
/** Known values of {@link ConfigurationSource} that the service accepts. */
export var KnownConfigurationSource;
(function (KnownConfigurationSource) {
    /** SystemDefault */
    KnownConfigurationSource["SystemDefault"] = "system-default";
    /** UserOverride */
    KnownConfigurationSource["UserOverride"] = "user-override";
})(KnownConfigurationSource || (KnownConfigurationSource = {}));
/** Known values of {@link IsReadOnly} that the service accepts. */
export var KnownIsReadOnly;
(function (KnownIsReadOnly) {
    /** True */
    KnownIsReadOnly["True"] = "True";
    /** False */
    KnownIsReadOnly["False"] = "False";
})(KnownIsReadOnly || (KnownIsReadOnly = {}));
/** Known values of {@link IsConfigPendingRestart} that the service accepts. */
export var KnownIsConfigPendingRestart;
(function (KnownIsConfigPendingRestart) {
    /** True */
    KnownIsConfigPendingRestart["True"] = "True";
    /** False */
    KnownIsConfigPendingRestart["False"] = "False";
})(KnownIsConfigPendingRestart || (KnownIsConfigPendingRestart = {}));
/** Known values of {@link IsDynamicConfig} that the service accepts. */
export var KnownIsDynamicConfig;
(function (KnownIsDynamicConfig) {
    /** True */
    KnownIsDynamicConfig["True"] = "True";
    /** False */
    KnownIsDynamicConfig["False"] = "False";
})(KnownIsDynamicConfig || (KnownIsDynamicConfig = {}));
