// @azure/arm-scvmm/dist-esm/src/operations/inventoryItems.js
/*
 * Copyright (c) Microsoft Corporation.
 * Licensed under the MIT License.
 *
 * Code generated by Microsoft (R) AutoRest Code Generator.
 * Changes may cause incorrect behavior and will be lost if the code is regenerated.
 */
import { __asyncDelegator, __asyncGenerator, __asyncValues, __await } from "tslib";
import { setContinuationToken } from "../pagingHelper";
import * as coreClient from "@azure/core-client";
import * as Mappers from "../models/mappers";
import * as Parameters from "../models/parameters";
/// <reference lib="esnext.asynciterable" />
/** Class containing InventoryItems operations. */
export class InventoryItemsImpl {
    /**
     * Initialize a new instance of the class InventoryItems class.
     * @param client Reference to the service client
     */
    constructor(client) {
        this.client = client;
    }
    /**
     * Returns the list of inventoryItems in the given VMMServer.
     * @param resourceGroupName The name of the resource group.
     * @param vmmServerName Name of the VMMServer.
     * @param options The options parameters.
     */
    listByVMMServer(resourceGroupName, vmmServerName, options) {
        const iter = this.listByVMMServerPagingAll(resourceGroupName, vmmServerName, options);
        return {
            next() {
                return iter.next();
            },
            [Symbol.asyncIterator]() {
                return this;
            },
            byPage: (settings) => {
                if (settings === null || settings === void 0 ? void 0 : settings.maxPageSize) {
                    throw new Error("maxPageSize is not supported by this operation.");
                }
                return this.listByVMMServerPagingPage(resourceGroupName, vmmServerName, options, settings);
            }
        };
    }
    listByVMMServerPagingPage(resourceGroupName, vmmServerName, options, settings) {
        return __asyncGenerator(this, arguments, function* listByVMMServerPagingPage_1() {
            let result;
            let continuationToken = settings === null || settings === void 0 ? void 0 : settings.continuationToken;
            if (!continuationToken) {
                result = yield __await(this._listByVMMServer(resourceGroupName, vmmServerName, options));
                let page = result.value || [];
                continuationToken = result.nextLink;
                setContinuationToken(page, continuationToken);
                yield yield __await(page);
            }
            while (continuationToken) {
                result = yield __await(this._listByVMMServerNext(resourceGroupName, vmmServerName, continuationToken, options));
                continuationToken = result.nextLink;
                let page = result.value || [];
                setContinuationToken(page, continuationToken);
                yield yield __await(page);
            }
        });
    }
    listByVMMServerPagingAll(resourceGroupName, vmmServerName, options) {
        return __asyncGenerator(this, arguments, function* listByVMMServerPagingAll_1() {
            var e_1, _a;
            try {
                for (var _b = __asyncValues(this.listByVMMServerPagingPage(resourceGroupName, vmmServerName, options)), _c; _c = yield __await(_b.next()), !_c.done;) {
                    const page = _c.value;
                    yield __await(yield* __asyncDelegator(__asyncValues(page)));
                }
            }
            catch (e_1_1) { e_1 = { error: e_1_1 }; }
            finally {
                try {
                    if (_c && !_c.done && (_a = _b.return)) yield __await(_a.call(_b));
                }
                finally { if (e_1) throw e_1.error; }
            }
        });
    }
    /**
     * Create Or Update InventoryItem.
     * @param resourceGroupName The name of the resource group.
     * @param vmmServerName Name of the VMMServer.
     * @param inventoryItemName Name of the inventoryItem.
     * @param options The options parameters.
     */
    create(resourceGroupName, vmmServerName, inventoryItemName, options) {
        return this.client.sendOperationRequest({ resourceGroupName, vmmServerName, inventoryItemName, options }, createOperationSpec);
    }
    /**
     * Shows an inventory item.
     * @param resourceGroupName The name of the resource group.
     * @param vmmServerName Name of the VMMServer.
     * @param inventoryItemName Name of the inventoryItem.
     * @param options The options parameters.
     */
    get(resourceGroupName, vmmServerName, inventoryItemName, options) {
        return this.client.sendOperationRequest({ resourceGroupName, vmmServerName, inventoryItemName, options }, getOperationSpec);
    }
    /**
     * Deletes an inventoryItem.
     * @param resourceGroupName The name of the resource group.
     * @param vmmServerName Name of the VMMServer.
     * @param inventoryItemName Name of the inventoryItem.
     * @param options The options parameters.
     */
    delete(resourceGroupName, vmmServerName, inventoryItemName, options) {
        return this.client.sendOperationRequest({ resourceGroupName, vmmServerName, inventoryItemName, options }, deleteOperationSpec);
    }
    /**
     * Returns the list of inventoryItems in the given VMMServer.
     * @param resourceGroupName The name of the resource group.
     * @param vmmServerName Name of the VMMServer.
     * @param options The options parameters.
     */
    _listByVMMServer(resourceGroupName, vmmServerName, options) {
        return this.client.sendOperationRequest({ resourceGroupName, vmmServerName, options }, listByVMMServerOperationSpec);
    }
    /**
     * ListByVMMServerNext
     * @param resourceGroupName The name of the resource group.
     * @param vmmServerName Name of the VMMServer.
     * @param nextLink The nextLink from the previous successful call to the ListByVMMServer method.
     * @param options The options parameters.
     */
    _listByVMMServerNext(resourceGroupName, vmmServerName, nextLink, options) {
        return this.client.sendOperationRequest({ resourceGroupName, vmmServerName, nextLink, options }, listByVMMServerNextOperationSpec);
    }
}
// Operation Specifications
const serializer = coreClient.createSerializer(Mappers, /* isXml */ false);
const createOperationSpec = {
    path: "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.ScVmm/vmmServers/{vmmServerName}/inventoryItems/{inventoryItemName}",
    httpMethod: "PUT",
    responses: {
        200: {
            bodyMapper: Mappers.InventoryItem
        },
        default: {
            bodyMapper: Mappers.ErrorResponse
        }
    },
    requestBody: Parameters.body12,
    queryParameters: [Parameters.apiVersion],
    urlParameters: [
        Parameters.$host,
        Parameters.subscriptionId,
        Parameters.resourceGroupName,
        Parameters.vmmServerName,
        Parameters.inventoryItemName
    ],
    headerParameters: [Parameters.accept, Parameters.contentType],
    mediaType: "json",
    serializer
};
const getOperationSpec = {
    path: "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.ScVmm/vmmServers/{vmmServerName}/inventoryItems/{inventoryItemName}",
    httpMethod: "GET",
    responses: {
        200: {
            bodyMapper: Mappers.InventoryItem
        },
        default: {
            bodyMapper: Mappers.ErrorResponse
        }
    },
    queryParameters: [Parameters.apiVersion],
    urlParameters: [
        Parameters.$host,
        Parameters.subscriptionId,
        Parameters.resourceGroupName,
        Parameters.vmmServerName,
        Parameters.inventoryItemName
    ],
    headerParameters: [Parameters.accept],
    serializer
};
const deleteOperationSpec = {
    path: "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.ScVmm/vmmServers/{vmmServerName}/inventoryItems/{inventoryItemName}",
    httpMethod: "DELETE",
    responses: {
        200: {},
        204: {},
        default: {
            bodyMapper: Mappers.ErrorResponse
        }
    },
    queryParameters: [Parameters.apiVersion],
    urlParameters: [
        Parameters.$host,
        Parameters.subscriptionId,
        Parameters.resourceGroupName,
        Parameters.vmmServerName,
        Parameters.inventoryItemName
    ],
    headerParameters: [Parameters.accept],
    serializer
};
const listByVMMServerOperationSpec = {
    path: "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.ScVmm/vmmServers/{vmmServerName}/inventoryItems",
    httpMethod: "GET",
    responses: {
        200: {
            bodyMapper: Mappers.InventoryItemsList
        },
        default: {
            bodyMapper: Mappers.ErrorResponse
        }
    },
    queryParameters: [Parameters.apiVersion],
    urlParameters: [
        Parameters.$host,
        Parameters.subscriptionId,
        Parameters.resourceGroupName,
        Parameters.vmmServerName
    ],
    headerParameters: [Parameters.accept],
    serializer
};
const listByVMMServerNextOperationSpec = {
    path: "{nextLink}",
    httpMethod: "GET",
    responses: {
        200: {
            bodyMapper: Mappers.InventoryItemsList
        },
        default: {
            bodyMapper: Mappers.ErrorResponse
        }
    },
    urlParameters: [
        Parameters.$host,
        Parameters.subscriptionId,
        Parameters.resourceGroupName,
        Parameters.vmmServerName,
        Parameters.nextLink
    ],
    headerParameters: [Parameters.accept],
    serializer
};
