// @azure/arm-automation/esm/operations/dscCompilationJobStream.js
/*
 * Copyright (c) Microsoft Corporation. All rights reserved.
 * Licensed under the MIT License. See License.txt in the project root for
 * license information.
 *
 * Code generated by Microsoft (R) AutoRest Code Generator.
 * Changes may cause incorrect behavior and will be lost if the code is
 * regenerated.
 */
import * as msRest from "@azure/ms-rest-js";
import * as Mappers from "../models/dscCompilationJobStreamMappers";
import * as Parameters from "../models/parameters";
/** Class representing a DscCompilationJobStream. */
var DscCompilationJobStream = /** @class */ (function () {
    /**
     * Create a DscCompilationJobStream.
     * @param {AutomationClientContext} client Reference to the service client.
     */
    function DscCompilationJobStream(client) {
        this.client = client;
    }
    DscCompilationJobStream.prototype.listByJob = function (resourceGroupName, automationAccountName, jobId, options, callback) {
        return this.client.sendOperationRequest({
            resourceGroupName: resourceGroupName,
            automationAccountName: automationAccountName,
            jobId: jobId,
            options: options
        }, listByJobOperationSpec, callback);
    };
    return DscCompilationJobStream;
}());
export { DscCompilationJobStream };
// Operation Specifications
var serializer = new msRest.Serializer(Mappers);
var listByJobOperationSpec = {
    httpMethod: "GET",
    path: "subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Automation/automationAccounts/{automationAccountName}/compilationjobs/{jobId}/streams",
    urlParameters: [
        Parameters.resourceGroupName,
        Parameters.automationAccountName,
        Parameters.jobId,
        Parameters.subscriptionId
    ],
    queryParameters: [
        Parameters.apiVersion2
    ],
    headerParameters: [
        Parameters.acceptLanguage
    ],
    responses: {
        200: {
            bodyMapper: Mappers.JobStreamListResult
        },
        default: {
            bodyMapper: Mappers.ErrorResponse
        }
    },
    serializer: serializer
};
