// @azure/arm-automation/esm/operations/fields.js
/*
 * Copyright (c) Microsoft Corporation. All rights reserved.
 * Licensed under the MIT License. See License.txt in the project root for
 * license information.
 *
 * Code generated by Microsoft (R) AutoRest Code Generator.
 * Changes may cause incorrect behavior and will be lost if the code is
 * regenerated.
 */
import * as msRest from "@azure/ms-rest-js";
import * as Mappers from "../models/fieldsMappers";
import * as Parameters from "../models/parameters";
/** Class representing a Fields. */
var Fields = /** @class */ (function () {
    /**
     * Create a Fields.
     * @param {AutomationClientContext} client Reference to the service client.
     */
    function Fields(client) {
        this.client = client;
    }
    Fields.prototype.listByType = function (resourceGroupName, automationAccountName, moduleName, typeName, options, callback) {
        return this.client.sendOperationRequest({
            resourceGroupName: resourceGroupName,
            automationAccountName: automationAccountName,
            moduleName: moduleName,
            typeName: typeName,
            options: options
        }, listByTypeOperationSpec, callback);
    };
    return Fields;
}());
export { Fields };
// Operation Specifications
var serializer = new msRest.Serializer(Mappers);
var listByTypeOperationSpec = {
    httpMethod: "GET",
    path: "subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Automation/automationAccounts/{automationAccountName}/modules/{moduleName}/types/{typeName}/fields",
    urlParameters: [
        Parameters.resourceGroupName,
        Parameters.automationAccountName,
        Parameters.moduleName,
        Parameters.typeName,
        Parameters.subscriptionId
    ],
    queryParameters: [
        Parameters.apiVersion0
    ],
    headerParameters: [
        Parameters.acceptLanguage
    ],
    responses: {
        200: {
            bodyMapper: Mappers.TypeFieldListResult
        },
        default: {
            bodyMapper: Mappers.ErrorResponse
        }
    },
    serializer: serializer
};
