// @azure/arm-automation/esm/operations/linkedWorkspaceOperations.js
/*
 * Copyright (c) Microsoft Corporation. All rights reserved.
 * Licensed under the MIT License. See License.txt in the project root for
 * license information.
 *
 * Code generated by Microsoft (R) AutoRest Code Generator.
 * Changes may cause incorrect behavior and will be lost if the code is
 * regenerated.
 */
import * as msRest from "@azure/ms-rest-js";
import * as Mappers from "../models/linkedWorkspaceOperationsMappers";
import * as Parameters from "../models/parameters";
/** Class representing a LinkedWorkspaceOperations. */
var LinkedWorkspaceOperations = /** @class */ (function () {
    /**
     * Create a LinkedWorkspaceOperations.
     * @param {AutomationClientContext} client Reference to the service client.
     */
    function LinkedWorkspaceOperations(client) {
        this.client = client;
    }
    LinkedWorkspaceOperations.prototype.get = function (resourceGroupName, automationAccountName, options, callback) {
        return this.client.sendOperationRequest({
            resourceGroupName: resourceGroupName,
            automationAccountName: automationAccountName,
            options: options
        }, getOperationSpec, callback);
    };
    return LinkedWorkspaceOperations;
}());
export { LinkedWorkspaceOperations };
// Operation Specifications
var serializer = new msRest.Serializer(Mappers);
var getOperationSpec = {
    httpMethod: "GET",
    path: "subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Automation/automationAccounts/{automationAccountName}/linkedWorkspace",
    urlParameters: [
        Parameters.resourceGroupName,
        Parameters.automationAccountName,
        Parameters.subscriptionId
    ],
    queryParameters: [
        Parameters.apiVersion0
    ],
    headerParameters: [
        Parameters.acceptLanguage
    ],
    responses: {
        200: {
            bodyMapper: Mappers.LinkedWorkspace
        },
        default: {
            bodyMapper: Mappers.ErrorResponse
        }
    },
    serializer: serializer
};
