// @azure/arm-automation/esm/operations/runbookDraftOperations.js
/*
 * Copyright (c) Microsoft Corporation. All rights reserved.
 * Licensed under the MIT License. See License.txt in the project root for
 * license information.
 *
 * Code generated by Microsoft (R) AutoRest Code Generator.
 * Changes may cause incorrect behavior and will be lost if the code is
 * regenerated.
 */
import * as msRest from "@azure/ms-rest-js";
import * as Mappers from "../models/runbookDraftOperationsMappers";
import * as Parameters from "../models/parameters";
/** Class representing a RunbookDraftOperations. */
var RunbookDraftOperations = /** @class */ (function () {
    /**
     * Create a RunbookDraftOperations.
     * @param {AutomationClientContext} client Reference to the service client.
     */
    function RunbookDraftOperations(client) {
        this.client = client;
    }
    RunbookDraftOperations.prototype.getContent = function (resourceGroupName, automationAccountName, runbookName, options, callback) {
        return this.client.sendOperationRequest({
            resourceGroupName: resourceGroupName,
            automationAccountName: automationAccountName,
            runbookName: runbookName,
            options: options
        }, getContentOperationSpec, callback);
    };
    /**
     * Replaces the runbook draft content.
     * @param resourceGroupName Name of an Azure Resource group.
     * @param automationAccountName The name of the automation account.
     * @param runbookName The runbook name.
     * @param runbookContent The runbook draft content.
     * @param [options] The optional parameters
     * @returns Promise<Models.RunbookDraftReplaceContentResponse>
     */
    RunbookDraftOperations.prototype.replaceContent = function (resourceGroupName, automationAccountName, runbookName, runbookContent, options) {
        return this.beginReplaceContent(resourceGroupName, automationAccountName, runbookName, runbookContent, options)
            .then(function (lroPoller) { return lroPoller.pollUntilFinished(); });
    };
    RunbookDraftOperations.prototype.get = function (resourceGroupName, automationAccountName, runbookName, options, callback) {
        return this.client.sendOperationRequest({
            resourceGroupName: resourceGroupName,
            automationAccountName: automationAccountName,
            runbookName: runbookName,
            options: options
        }, getOperationSpec, callback);
    };
    RunbookDraftOperations.prototype.undoEdit = function (resourceGroupName, automationAccountName, runbookName, options, callback) {
        return this.client.sendOperationRequest({
            resourceGroupName: resourceGroupName,
            automationAccountName: automationAccountName,
            runbookName: runbookName,
            options: options
        }, undoEditOperationSpec, callback);
    };
    /**
     * Replaces the runbook draft content.
     * @param resourceGroupName Name of an Azure Resource group.
     * @param automationAccountName The name of the automation account.
     * @param runbookName The runbook name.
     * @param runbookContent The runbook draft content.
     * @param [options] The optional parameters
     * @returns Promise<msRestAzure.LROPoller>
     */
    RunbookDraftOperations.prototype.beginReplaceContent = function (resourceGroupName, automationAccountName, runbookName, runbookContent, options) {
        return this.client.sendLRORequest({
            resourceGroupName: resourceGroupName,
            automationAccountName: automationAccountName,
            runbookName: runbookName,
            runbookContent: runbookContent,
            options: options
        }, beginReplaceContentOperationSpec, options);
    };
    return RunbookDraftOperations;
}());
export { RunbookDraftOperations };
// Operation Specifications
var serializer = new msRest.Serializer(Mappers);
var getContentOperationSpec = {
    httpMethod: "GET",
    path: "subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Automation/automationAccounts/{automationAccountName}/runbooks/{runbookName}/draft/content",
    urlParameters: [
        Parameters.subscriptionId,
        Parameters.resourceGroupName,
        Parameters.automationAccountName,
        Parameters.runbookName
    ],
    queryParameters: [
        Parameters.apiVersion3
    ],
    headerParameters: [
        Parameters.acceptLanguage
    ],
    responses: {
        200: {
            bodyMapper: {
                serializedName: "parsedResponse",
                type: {
                    name: "Stream"
                }
            }
        },
        default: {
            bodyMapper: Mappers.CloudError
        }
    },
    serializer: serializer
};
var getOperationSpec = {
    httpMethod: "GET",
    path: "subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Automation/automationAccounts/{automationAccountName}/runbooks/{runbookName}/draft",
    urlParameters: [
        Parameters.subscriptionId,
        Parameters.resourceGroupName,
        Parameters.automationAccountName,
        Parameters.runbookName
    ],
    queryParameters: [
        Parameters.apiVersion3
    ],
    headerParameters: [
        Parameters.acceptLanguage
    ],
    responses: {
        200: {
            bodyMapper: Mappers.RunbookDraft
        },
        default: {
            bodyMapper: Mappers.ErrorResponse
        }
    },
    serializer: serializer
};
var undoEditOperationSpec = {
    httpMethod: "POST",
    path: "subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Automation/automationAccounts/{automationAccountName}/runbooks/{runbookName}/draft/undoEdit",
    urlParameters: [
        Parameters.subscriptionId,
        Parameters.resourceGroupName,
        Parameters.automationAccountName,
        Parameters.runbookName
    ],
    queryParameters: [
        Parameters.apiVersion3
    ],
    headerParameters: [
        Parameters.acceptLanguage
    ],
    responses: {
        200: {
            bodyMapper: Mappers.RunbookDraftUndoEditResult
        },
        default: {
            bodyMapper: Mappers.ErrorResponse
        }
    },
    serializer: serializer
};
var beginReplaceContentOperationSpec = {
    httpMethod: "PUT",
    path: "subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Automation/automationAccounts/{automationAccountName}/runbooks/{runbookName}/draft/content",
    urlParameters: [
        Parameters.subscriptionId,
        Parameters.resourceGroupName,
        Parameters.automationAccountName,
        Parameters.runbookName
    ],
    queryParameters: [
        Parameters.apiVersion3
    ],
    headerParameters: [
        Parameters.acceptLanguage
    ],
    requestBody: {
        parameterPath: "runbookContent",
        mapper: {
            required: true,
            serializedName: "runbookContent",
            type: {
                name: "Stream"
            }
        }
    },
    contentType: "text/powershell",
    responses: {
        200: {
            bodyMapper: {
                serializedName: "parsedResponse",
                type: {
                    name: "Stream"
                }
            },
            headersMapper: Mappers.RunbookDraftReplaceContentHeaders
        },
        202: {
            headersMapper: Mappers.RunbookDraftReplaceContentHeaders
        },
        default: {
            bodyMapper: Mappers.ErrorResponse
        }
    },
    serializer: serializer
};
