// @azure/arm-automation/esm/operations/softwareUpdateConfigurationRuns.js
/*
 * Copyright (c) Microsoft Corporation. All rights reserved.
 * Licensed under the MIT License. See License.txt in the project root for
 * license information.
 *
 * Code generated by Microsoft (R) AutoRest Code Generator.
 * Changes may cause incorrect behavior and will be lost if the code is
 * regenerated.
 */
import * as msRest from "@azure/ms-rest-js";
import * as Mappers from "../models/softwareUpdateConfigurationRunsMappers";
import * as Parameters from "../models/parameters";
/** Class representing a SoftwareUpdateConfigurationRuns. */
var SoftwareUpdateConfigurationRuns = /** @class */ (function () {
    /**
     * Create a SoftwareUpdateConfigurationRuns.
     * @param {AutomationClientContext} client Reference to the service client.
     */
    function SoftwareUpdateConfigurationRuns(client) {
        this.client = client;
    }
    SoftwareUpdateConfigurationRuns.prototype.getById = function (resourceGroupName, automationAccountName, softwareUpdateConfigurationRunId, options, callback) {
        return this.client.sendOperationRequest({
            resourceGroupName: resourceGroupName,
            automationAccountName: automationAccountName,
            softwareUpdateConfigurationRunId: softwareUpdateConfigurationRunId,
            options: options
        }, getByIdOperationSpec, callback);
    };
    SoftwareUpdateConfigurationRuns.prototype.list = function (resourceGroupName, automationAccountName, options, callback) {
        return this.client.sendOperationRequest({
            resourceGroupName: resourceGroupName,
            automationAccountName: automationAccountName,
            options: options
        }, listOperationSpec, callback);
    };
    return SoftwareUpdateConfigurationRuns;
}());
export { SoftwareUpdateConfigurationRuns };
// Operation Specifications
var serializer = new msRest.Serializer(Mappers);
var getByIdOperationSpec = {
    httpMethod: "GET",
    path: "subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Automation/automationAccounts/{automationAccountName}/softwareUpdateConfigurationRuns/{softwareUpdateConfigurationRunId}",
    urlParameters: [
        Parameters.subscriptionId,
        Parameters.resourceGroupName,
        Parameters.automationAccountName,
        Parameters.softwareUpdateConfigurationRunId
    ],
    queryParameters: [
        Parameters.apiVersion1
    ],
    headerParameters: [
        Parameters.clientRequestId,
        Parameters.acceptLanguage
    ],
    responses: {
        200: {
            bodyMapper: Mappers.SoftwareUpdateConfigurationRun
        },
        default: {
            bodyMapper: Mappers.ErrorResponse
        }
    },
    serializer: serializer
};
var listOperationSpec = {
    httpMethod: "GET",
    path: "subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Automation/automationAccounts/{automationAccountName}/softwareUpdateConfigurationRuns",
    urlParameters: [
        Parameters.subscriptionId,
        Parameters.resourceGroupName,
        Parameters.automationAccountName
    ],
    queryParameters: [
        Parameters.apiVersion1,
        Parameters.filter,
        Parameters.skip1,
        Parameters.top1
    ],
    headerParameters: [
        Parameters.clientRequestId,
        Parameters.acceptLanguage
    ],
    responses: {
        200: {
            bodyMapper: Mappers.SoftwareUpdateConfigurationRunListResult
        },
        default: {
            bodyMapper: Mappers.ErrorResponse
        }
    },
    serializer: serializer
};
