// @azure/arm-automation/esm/operations/sourceControlSyncJobOperations.js
/*
 * Copyright (c) Microsoft Corporation. All rights reserved.
 * Licensed under the MIT License. See License.txt in the project root for
 * license information.
 *
 * Code generated by Microsoft (R) AutoRest Code Generator.
 * Changes may cause incorrect behavior and will be lost if the code is
 * regenerated.
 */
import { __assign } from "tslib";
import * as msRest from "@azure/ms-rest-js";
import * as Mappers from "../models/sourceControlSyncJobOperationsMappers";
import * as Parameters from "../models/parameters";
/** Class representing a SourceControlSyncJobOperations. */
var SourceControlSyncJobOperations = /** @class */ (function () {
    /**
     * Create a SourceControlSyncJobOperations.
     * @param {AutomationClientContext} client Reference to the service client.
     */
    function SourceControlSyncJobOperations(client) {
        this.client = client;
    }
    SourceControlSyncJobOperations.prototype.create = function (resourceGroupName, automationAccountName, sourceControlName, sourceControlSyncJobId, parameters, options, callback) {
        return this.client.sendOperationRequest({
            resourceGroupName: resourceGroupName,
            automationAccountName: automationAccountName,
            sourceControlName: sourceControlName,
            sourceControlSyncJobId: sourceControlSyncJobId,
            parameters: parameters,
            options: options
        }, createOperationSpec, callback);
    };
    SourceControlSyncJobOperations.prototype.get = function (resourceGroupName, automationAccountName, sourceControlName, sourceControlSyncJobId, options, callback) {
        return this.client.sendOperationRequest({
            resourceGroupName: resourceGroupName,
            automationAccountName: automationAccountName,
            sourceControlName: sourceControlName,
            sourceControlSyncJobId: sourceControlSyncJobId,
            options: options
        }, getOperationSpec, callback);
    };
    SourceControlSyncJobOperations.prototype.listByAutomationAccount = function (resourceGroupName, automationAccountName, sourceControlName, options, callback) {
        return this.client.sendOperationRequest({
            resourceGroupName: resourceGroupName,
            automationAccountName: automationAccountName,
            sourceControlName: sourceControlName,
            options: options
        }, listByAutomationAccountOperationSpec, callback);
    };
    SourceControlSyncJobOperations.prototype.listByAutomationAccountNext = function (nextPageLink, options, callback) {
        return this.client.sendOperationRequest({
            nextPageLink: nextPageLink,
            options: options
        }, listByAutomationAccountNextOperationSpec, callback);
    };
    return SourceControlSyncJobOperations;
}());
export { SourceControlSyncJobOperations };
// Operation Specifications
var serializer = new msRest.Serializer(Mappers);
var createOperationSpec = {
    httpMethod: "PUT",
    path: "subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Automation/automationAccounts/{automationAccountName}/sourceControls/{sourceControlName}/sourceControlSyncJobs/{sourceControlSyncJobId}",
    urlParameters: [
        Parameters.resourceGroupName,
        Parameters.automationAccountName,
        Parameters.sourceControlName,
        Parameters.sourceControlSyncJobId,
        Parameters.subscriptionId
    ],
    queryParameters: [
        Parameters.apiVersion1
    ],
    headerParameters: [
        Parameters.acceptLanguage
    ],
    requestBody: {
        parameterPath: "parameters",
        mapper: __assign(__assign({}, Mappers.SourceControlSyncJobCreateParameters), { required: true })
    },
    responses: {
        201: {
            bodyMapper: Mappers.SourceControlSyncJob
        },
        default: {
            bodyMapper: Mappers.ErrorResponse
        }
    },
    serializer: serializer
};
var getOperationSpec = {
    httpMethod: "GET",
    path: "subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Automation/automationAccounts/{automationAccountName}/sourceControls/{sourceControlName}/sourceControlSyncJobs/{sourceControlSyncJobId}",
    urlParameters: [
        Parameters.resourceGroupName,
        Parameters.automationAccountName,
        Parameters.sourceControlName,
        Parameters.sourceControlSyncJobId,
        Parameters.subscriptionId
    ],
    queryParameters: [
        Parameters.apiVersion1
    ],
    headerParameters: [
        Parameters.acceptLanguage
    ],
    responses: {
        200: {
            bodyMapper: Mappers.SourceControlSyncJobById
        },
        default: {
            bodyMapper: Mappers.ErrorResponse
        }
    },
    serializer: serializer
};
var listByAutomationAccountOperationSpec = {
    httpMethod: "GET",
    path: "subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Automation/automationAccounts/{automationAccountName}/sourceControls/{sourceControlName}/sourceControlSyncJobs",
    urlParameters: [
        Parameters.resourceGroupName,
        Parameters.automationAccountName,
        Parameters.sourceControlName,
        Parameters.subscriptionId
    ],
    queryParameters: [
        Parameters.filter,
        Parameters.apiVersion1
    ],
    headerParameters: [
        Parameters.acceptLanguage
    ],
    responses: {
        200: {
            bodyMapper: Mappers.SourceControlSyncJobListResult
        },
        default: {
            bodyMapper: Mappers.ErrorResponse
        }
    },
    serializer: serializer
};
var listByAutomationAccountNextOperationSpec = {
    httpMethod: "GET",
    baseUrl: "https://management.azure.com",
    path: "{nextLink}",
    urlParameters: [
        Parameters.nextPageLink
    ],
    headerParameters: [
        Parameters.acceptLanguage
    ],
    responses: {
        200: {
            bodyMapper: Mappers.SourceControlSyncJobListResult
        },
        default: {
            bodyMapper: Mappers.ErrorResponse
        }
    },
    serializer: serializer
};
