// @azure/arm-quota/dist-esm/src/models/index.js
/*
 * Copyright (c) Microsoft Corporation.
 * Licensed under the MIT License.
 *
 * Code generated by Microsoft (R) AutoRest Code Generator.
 * Changes may cause incorrect behavior and will be lost if the code is regenerated.
 */
/** Known values of {@link UsagesTypes} that the service accepts. */
export var KnownUsagesTypes;
(function (KnownUsagesTypes) {
    /** Individual */
    KnownUsagesTypes["Individual"] = "Individual";
    /** Combined */
    KnownUsagesTypes["Combined"] = "Combined";
})(KnownUsagesTypes || (KnownUsagesTypes = {}));
/** Known values of {@link LimitType} that the service accepts. */
export var KnownLimitType;
(function (KnownLimitType) {
    /** LimitValue */
    KnownLimitType["LimitValue"] = "LimitValue";
})(KnownLimitType || (KnownLimitType = {}));
/** Known values of {@link QuotaRequestState} that the service accepts. */
export var KnownQuotaRequestState;
(function (KnownQuotaRequestState) {
    /** Accepted */
    KnownQuotaRequestState["Accepted"] = "Accepted";
    /** Invalid */
    KnownQuotaRequestState["Invalid"] = "Invalid";
    /** Succeeded */
    KnownQuotaRequestState["Succeeded"] = "Succeeded";
    /** Failed */
    KnownQuotaRequestState["Failed"] = "Failed";
    /** InProgress */
    KnownQuotaRequestState["InProgress"] = "InProgress";
})(KnownQuotaRequestState || (KnownQuotaRequestState = {}));
/** Known values of {@link QuotaLimitTypes} that the service accepts. */
export var KnownQuotaLimitTypes;
(function (KnownQuotaLimitTypes) {
    /** Independent */
    KnownQuotaLimitTypes["Independent"] = "Independent";
    /** Shared */
    KnownQuotaLimitTypes["Shared"] = "Shared";
})(KnownQuotaLimitTypes || (KnownQuotaLimitTypes = {}));
