// @azure/arm-machinelearningexperimentation/esm/models/parameters.js
/*
 * Copyright (c) Microsoft Corporation. All rights reserved.
 * Licensed under the MIT License. See License.txt in the project root for
 * license information.
 *
 * Code generated by Microsoft (R) AutoRest Code Generator.
 * Changes may cause incorrect behavior and will be lost if the code is
 * regenerated.
 */
export var acceptLanguage = {
    parameterPath: "acceptLanguage",
    mapper: {
        serializedName: "accept-language",
        defaultValue: 'en-US',
        type: {
            name: "String"
        }
    }
};
export var accountName = {
    parameterPath: "accountName",
    mapper: {
        required: true,
        serializedName: "accountName",
        constraints: {
            MaxLength: 50,
            MinLength: 5,
            Pattern: /^[a-zA-Z0-9]*$/
        },
        type: {
            name: "String"
        }
    }
};
export var apiVersion = {
    parameterPath: "apiVersion",
    mapper: {
        required: true,
        serializedName: "api-version",
        type: {
            name: "String"
        }
    }
};
export var nextPageLink = {
    parameterPath: "nextPageLink",
    mapper: {
        required: true,
        serializedName: "nextLink",
        type: {
            name: "String"
        }
    },
    skipEncoding: true
};
export var projectName = {
    parameterPath: "projectName",
    mapper: {
        required: true,
        serializedName: "projectName",
        constraints: {
            MaxLength: 50,
            MinLength: 5,
            Pattern: /^[a-zA-Z0-9]*$/
        },
        type: {
            name: "String"
        }
    }
};
export var resourceGroupName = {
    parameterPath: "resourceGroupName",
    mapper: {
        required: true,
        serializedName: "resourceGroupName",
        type: {
            name: "String"
        }
    }
};
export var subscriptionId = {
    parameterPath: "subscriptionId",
    mapper: {
        required: true,
        serializedName: "subscriptionId",
        type: {
            name: "String"
        }
    }
};
export var workspaceName = {
    parameterPath: "workspaceName",
    mapper: {
        required: true,
        serializedName: "workspaceName",
        constraints: {
            MaxLength: 50,
            MinLength: 5,
            Pattern: /^[a-zA-Z0-9]*$/
        },
        type: {
            name: "String"
        }
    }
};
