// @azure/arm-education/dist-esm/src/models/index.js
/*
 * Copyright (c) Microsoft Corporation.
 * Licensed under the MIT License.
 *
 * Code generated by Microsoft (R) AutoRest Code Generator.
 * Changes may cause incorrect behavior and will be lost if the code is regenerated.
 */
/** Known values of {@link Origin} that the service accepts. */
export var KnownOrigin;
(function (KnownOrigin) {
    /** User */
    KnownOrigin["User"] = "user";
    /** System */
    KnownOrigin["System"] = "system";
    /** UserSystem */
    KnownOrigin["UserSystem"] = "user,system";
})(KnownOrigin || (KnownOrigin = {}));
/** Known values of {@link ActionType} that the service accepts. */
export var KnownActionType;
(function (KnownActionType) {
    /** Internal */
    KnownActionType["Internal"] = "Internal";
})(KnownActionType || (KnownActionType = {}));
/** Known values of {@link GrantType} that the service accepts. */
export var KnownGrantType;
(function (KnownGrantType) {
    /** Student */
    KnownGrantType["Student"] = "Student";
    /** Academic */
    KnownGrantType["Academic"] = "Academic";
})(KnownGrantType || (KnownGrantType = {}));
/** Known values of {@link GrantStatus} that the service accepts. */
export var KnownGrantStatus;
(function (KnownGrantStatus) {
    /** Active */
    KnownGrantStatus["Active"] = "Active";
    /** Inactive */
    KnownGrantStatus["Inactive"] = "Inactive";
})(KnownGrantStatus || (KnownGrantStatus = {}));
/** Known values of {@link CreatedByType} that the service accepts. */
export var KnownCreatedByType;
(function (KnownCreatedByType) {
    /** User */
    KnownCreatedByType["User"] = "User";
    /** Application */
    KnownCreatedByType["Application"] = "Application";
    /** ManagedIdentity */
    KnownCreatedByType["ManagedIdentity"] = "ManagedIdentity";
    /** Key */
    KnownCreatedByType["Key"] = "Key";
})(KnownCreatedByType || (KnownCreatedByType = {}));
/** Known values of {@link LabStatus} that the service accepts. */
export var KnownLabStatus;
(function (KnownLabStatus) {
    /** Active */
    KnownLabStatus["Active"] = "Active";
    /** Deleted */
    KnownLabStatus["Deleted"] = "Deleted";
    /** Pending */
    KnownLabStatus["Pending"] = "Pending";
})(KnownLabStatus || (KnownLabStatus = {}));
/** Known values of {@link JoinRequestStatus} that the service accepts. */
export var KnownJoinRequestStatus;
(function (KnownJoinRequestStatus) {
    /** Pending */
    KnownJoinRequestStatus["Pending"] = "Pending";
    /** Denied */
    KnownJoinRequestStatus["Denied"] = "Denied";
})(KnownJoinRequestStatus || (KnownJoinRequestStatus = {}));
/** Known values of {@link StudentRole} that the service accepts. */
export var KnownStudentRole;
(function (KnownStudentRole) {
    /** Student */
    KnownStudentRole["Student"] = "Student";
    /** Admin */
    KnownStudentRole["Admin"] = "Admin";
})(KnownStudentRole || (KnownStudentRole = {}));
/** Known values of {@link StudentLabStatus} that the service accepts. */
export var KnownStudentLabStatus;
(function (KnownStudentLabStatus) {
    /** Active */
    KnownStudentLabStatus["Active"] = "Active";
    /** Disabled */
    KnownStudentLabStatus["Disabled"] = "Disabled";
    /** Expired */
    KnownStudentLabStatus["Expired"] = "Expired";
    /** Pending */
    KnownStudentLabStatus["Pending"] = "Pending";
    /** Deleted */
    KnownStudentLabStatus["Deleted"] = "Deleted";
})(KnownStudentLabStatus || (KnownStudentLabStatus = {}));
