// @azure/arm-advisor/dist-esm/src/operations/suppressions.js
/*
 * Copyright (c) Microsoft Corporation.
 * Licensed under the MIT License.
 *
 * Code generated by Microsoft (R) AutoRest Code Generator.
 * Changes may cause incorrect behavior and will be lost if the code is regenerated.
 */
import { __asyncDelegator, __asyncGenerator, __asyncValues, __await } from "tslib";
import { setContinuationToken } from "../pagingHelper";
import * as coreClient from "@azure/core-client";
import * as Mappers from "../models/mappers";
import * as Parameters from "../models/parameters";
/// <reference lib="esnext.asynciterable" />
/** Class containing Suppressions operations. */
export class SuppressionsImpl {
    /**
     * Initialize a new instance of the class Suppressions class.
     * @param client Reference to the service client
     */
    constructor(client) {
        this.client = client;
    }
    /**
     * Retrieves the list of snoozed or dismissed suppressions for a subscription. The snoozed or dismissed
     * attribute of a recommendation is referred to as a suppression.
     * @param options The options parameters.
     */
    list(options) {
        const iter = this.listPagingAll(options);
        return {
            next() {
                return iter.next();
            },
            [Symbol.asyncIterator]() {
                return this;
            },
            byPage: (settings) => {
                if (settings === null || settings === void 0 ? void 0 : settings.maxPageSize) {
                    throw new Error("maxPageSize is not supported by this operation.");
                }
                return this.listPagingPage(options, settings);
            }
        };
    }
    listPagingPage(options, settings) {
        return __asyncGenerator(this, arguments, function* listPagingPage_1() {
            let result;
            let continuationToken = settings === null || settings === void 0 ? void 0 : settings.continuationToken;
            if (!continuationToken) {
                result = yield __await(this._list(options));
                let page = result.value || [];
                continuationToken = result.nextLink;
                setContinuationToken(page, continuationToken);
                yield yield __await(page);
            }
            while (continuationToken) {
                result = yield __await(this._listNext(continuationToken, options));
                continuationToken = result.nextLink;
                let page = result.value || [];
                setContinuationToken(page, continuationToken);
                yield yield __await(page);
            }
        });
    }
    listPagingAll(options) {
        return __asyncGenerator(this, arguments, function* listPagingAll_1() {
            var _a, e_1, _b, _c;
            try {
                for (var _d = true, _e = __asyncValues(this.listPagingPage(options)), _f; _f = yield __await(_e.next()), _a = _f.done, !_a;) {
                    _c = _f.value;
                    _d = false;
                    try {
                        const page = _c;
                        yield __await(yield* __asyncDelegator(__asyncValues(page)));
                    }
                    finally {
                        _d = true;
                    }
                }
            }
            catch (e_1_1) { e_1 = { error: e_1_1 }; }
            finally {
                try {
                    if (!_d && !_a && (_b = _e.return)) yield __await(_b.call(_e));
                }
                finally { if (e_1) throw e_1.error; }
            }
        });
    }
    /**
     * Obtains the details of a suppression.
     * @param resourceUri The fully qualified Azure Resource Manager identifier of the resource to which
     *                    the recommendation applies.
     * @param recommendationId The recommendation ID.
     * @param name The name of the suppression.
     * @param options The options parameters.
     */
    get(resourceUri, recommendationId, name, options) {
        return this.client.sendOperationRequest({ resourceUri, recommendationId, name, options }, getOperationSpec);
    }
    /**
     * Enables the snoozed or dismissed attribute of a recommendation. The snoozed or dismissed attribute
     * is referred to as a suppression. Use this API to create or update the snoozed or dismissed status of
     * a recommendation.
     * @param resourceUri The fully qualified Azure Resource Manager identifier of the resource to which
     *                    the recommendation applies.
     * @param recommendationId The recommendation ID.
     * @param name The name of the suppression.
     * @param suppressionContract The snoozed or dismissed attribute; for example, the snooze duration.
     * @param options The options parameters.
     */
    create(resourceUri, recommendationId, name, suppressionContract, options) {
        return this.client.sendOperationRequest({ resourceUri, recommendationId, name, suppressionContract, options }, createOperationSpec);
    }
    /**
     * Enables the activation of a snoozed or dismissed recommendation. The snoozed or dismissed attribute
     * of a recommendation is referred to as a suppression.
     * @param resourceUri The fully qualified Azure Resource Manager identifier of the resource to which
     *                    the recommendation applies.
     * @param recommendationId The recommendation ID.
     * @param name The name of the suppression.
     * @param options The options parameters.
     */
    delete(resourceUri, recommendationId, name, options) {
        return this.client.sendOperationRequest({ resourceUri, recommendationId, name, options }, deleteOperationSpec);
    }
    /**
     * Retrieves the list of snoozed or dismissed suppressions for a subscription. The snoozed or dismissed
     * attribute of a recommendation is referred to as a suppression.
     * @param options The options parameters.
     */
    _list(options) {
        return this.client.sendOperationRequest({ options }, listOperationSpec);
    }
    /**
     * ListNext
     * @param nextLink The nextLink from the previous successful call to the List method.
     * @param options The options parameters.
     */
    _listNext(nextLink, options) {
        return this.client.sendOperationRequest({ nextLink, options }, listNextOperationSpec);
    }
}
// Operation Specifications
const serializer = coreClient.createSerializer(Mappers, /* isXml */ false);
const getOperationSpec = {
    path: "/{resourceUri}/providers/Microsoft.Advisor/recommendations/{recommendationId}/suppressions/{name}",
    httpMethod: "GET",
    responses: {
        200: {
            bodyMapper: Mappers.SuppressionContract
        },
        404: {
            bodyMapper: Mappers.ArmErrorResponse,
            isError: true
        },
        default: {
            bodyMapper: Mappers.ArmErrorResponse
        }
    },
    queryParameters: [Parameters.apiVersion],
    urlParameters: [
        Parameters.$host,
        Parameters.name,
        Parameters.resourceUri,
        Parameters.recommendationId
    ],
    headerParameters: [Parameters.accept],
    serializer
};
const createOperationSpec = {
    path: "/{resourceUri}/providers/Microsoft.Advisor/recommendations/{recommendationId}/suppressions/{name}",
    httpMethod: "PUT",
    responses: {
        200: {
            bodyMapper: Mappers.SuppressionContract
        },
        404: {
            bodyMapper: Mappers.ArmErrorResponse,
            isError: true
        },
        default: {
            bodyMapper: Mappers.ArmErrorResponse
        }
    },
    requestBody: Parameters.suppressionContract,
    queryParameters: [Parameters.apiVersion],
    urlParameters: [
        Parameters.$host,
        Parameters.name,
        Parameters.resourceUri,
        Parameters.recommendationId
    ],
    headerParameters: [Parameters.accept, Parameters.contentType],
    mediaType: "json",
    serializer
};
const deleteOperationSpec = {
    path: "/{resourceUri}/providers/Microsoft.Advisor/recommendations/{recommendationId}/suppressions/{name}",
    httpMethod: "DELETE",
    responses: {
        204: {},
        default: {
            bodyMapper: Mappers.ArmErrorResponse
        }
    },
    queryParameters: [Parameters.apiVersion],
    urlParameters: [
        Parameters.$host,
        Parameters.name,
        Parameters.resourceUri,
        Parameters.recommendationId
    ],
    headerParameters: [Parameters.accept],
    serializer
};
const listOperationSpec = {
    path: "/subscriptions/{subscriptionId}/providers/Microsoft.Advisor/suppressions",
    httpMethod: "GET",
    responses: {
        200: {
            bodyMapper: Mappers.SuppressionContractListResult
        },
        default: {
            bodyMapper: Mappers.ArmErrorResponse
        }
    },
    queryParameters: [
        Parameters.apiVersion,
        Parameters.top,
        Parameters.skipToken
    ],
    urlParameters: [Parameters.$host, Parameters.subscriptionId],
    headerParameters: [Parameters.accept],
    serializer
};
const listNextOperationSpec = {
    path: "{nextLink}",
    httpMethod: "GET",
    responses: {
        200: {
            bodyMapper: Mappers.SuppressionContractListResult
        },
        default: {
            bodyMapper: Mappers.ArmErrorResponse
        }
    },
    urlParameters: [
        Parameters.$host,
        Parameters.nextLink,
        Parameters.subscriptionId
    ],
    headerParameters: [Parameters.accept],
    serializer
};
