// @azure/arm-reservations/dist-esm/src/operations/calculateRefund.js
/*
 * Copyright (c) Microsoft Corporation.
 * Licensed under the MIT License.
 *
 * Code generated by Microsoft (R) AutoRest Code Generator.
 * Changes may cause incorrect behavior and will be lost if the code is regenerated.
 */
import * as coreClient from "@azure/core-client";
import * as Mappers from "../models/mappers";
import * as Parameters from "../models/parameters";
/** Class containing CalculateRefund operations. */
export class CalculateRefundImpl {
    /**
     * Initialize a new instance of the class CalculateRefund class.
     * @param client Reference to the service client
     */
    constructor(client) {
        this.client = client;
    }
    /**
     * Calculate price for returning `Reservations` if there are no policy errors.
     *
     * @param reservationOrderId Order Id of the reservation
     * @param body Information needed for calculating refund of a reservation.
     * @param options The options parameters.
     */
    post(reservationOrderId, body, options) {
        return this.client.sendOperationRequest({ reservationOrderId, body, options }, postOperationSpec);
    }
}
// Operation Specifications
const serializer = coreClient.createSerializer(Mappers, /* isXml */ false);
const postOperationSpec = {
    path: "/providers/Microsoft.Capacity/reservationOrders/{reservationOrderId}/calculateRefund",
    httpMethod: "POST",
    responses: {
        200: {
            bodyMapper: Mappers.CalculateRefundResponse
        },
        default: {
            bodyMapper: Mappers.ErrorModel
        }
    },
    requestBody: Parameters.body5,
    queryParameters: [Parameters.apiVersion],
    urlParameters: [Parameters.$host, Parameters.reservationOrderId],
    headerParameters: [Parameters.contentType, Parameters.accept],
    mediaType: "json",
    serializer
};
