// @azure/arm-reservations/dist-esm/src/operations/quotaRequestStatus.js
/*
 * Copyright (c) Microsoft Corporation.
 * Licensed under the MIT License.
 *
 * Code generated by Microsoft (R) AutoRest Code Generator.
 * Changes may cause incorrect behavior and will be lost if the code is regenerated.
 */
import { __asyncDelegator, __asyncGenerator, __asyncValues, __await } from "tslib";
import { setContinuationToken } from "../pagingHelper";
import * as coreClient from "@azure/core-client";
import * as Mappers from "../models/mappers";
import * as Parameters from "../models/parameters";
/// <reference lib="esnext.asynciterable" />
/** Class containing QuotaRequestStatus operations. */
export class QuotaRequestStatusImpl {
    /**
     * Initialize a new instance of the class QuotaRequestStatus class.
     * @param client Reference to the service client
     */
    constructor(client) {
        this.client = client;
    }
    /**
     * For the specified Azure region (location), subscription, and resource provider, get the history of
     * the quota requests for the past year. To select specific quota requests, use the oData filter.
     * @param subscriptionId Azure subscription ID.
     * @param providerId Azure resource provider ID.
     * @param location Azure region.
     * @param options The options parameters.
     */
    list(subscriptionId, providerId, location, options) {
        const iter = this.listPagingAll(subscriptionId, providerId, location, options);
        return {
            next() {
                return iter.next();
            },
            [Symbol.asyncIterator]() {
                return this;
            },
            byPage: (settings) => {
                if (settings === null || settings === void 0 ? void 0 : settings.maxPageSize) {
                    throw new Error("maxPageSize is not supported by this operation.");
                }
                return this.listPagingPage(subscriptionId, providerId, location, options, settings);
            }
        };
    }
    listPagingPage(subscriptionId, providerId, location, options, settings) {
        return __asyncGenerator(this, arguments, function* listPagingPage_1() {
            let result;
            let continuationToken = settings === null || settings === void 0 ? void 0 : settings.continuationToken;
            if (!continuationToken) {
                result = yield __await(this._list(subscriptionId, providerId, location, options));
                let page = result.value || [];
                continuationToken = result.nextLink;
                setContinuationToken(page, continuationToken);
                yield yield __await(page);
            }
            while (continuationToken) {
                result = yield __await(this._listNext(subscriptionId, providerId, location, continuationToken, options));
                continuationToken = result.nextLink;
                let page = result.value || [];
                setContinuationToken(page, continuationToken);
                yield yield __await(page);
            }
        });
    }
    listPagingAll(subscriptionId, providerId, location, options) {
        return __asyncGenerator(this, arguments, function* listPagingAll_1() {
            var e_1, _a;
            try {
                for (var _b = __asyncValues(this.listPagingPage(subscriptionId, providerId, location, options)), _c; _c = yield __await(_b.next()), !_c.done;) {
                    const page = _c.value;
                    yield __await(yield* __asyncDelegator(__asyncValues(page)));
                }
            }
            catch (e_1_1) { e_1 = { error: e_1_1 }; }
            finally {
                try {
                    if (_c && !_c.done && (_a = _b.return)) yield __await(_a.call(_b));
                }
                finally { if (e_1) throw e_1.error; }
            }
        });
    }
    /**
     * For the specified Azure region (location), get the details and status of the quota request by the
     * quota request ID for the resources of the resource provider. The PUT request for the quota (service
     * limit) returns a response with the requestId parameter.
     * @param subscriptionId Azure subscription ID.
     * @param providerId Azure resource provider ID.
     * @param location Azure region.
     * @param id Quota Request ID.
     * @param options The options parameters.
     */
    get(subscriptionId, providerId, location, id, options) {
        return this.client.sendOperationRequest({ subscriptionId, providerId, location, id, options }, getOperationSpec);
    }
    /**
     * For the specified Azure region (location), subscription, and resource provider, get the history of
     * the quota requests for the past year. To select specific quota requests, use the oData filter.
     * @param subscriptionId Azure subscription ID.
     * @param providerId Azure resource provider ID.
     * @param location Azure region.
     * @param options The options parameters.
     */
    _list(subscriptionId, providerId, location, options) {
        return this.client.sendOperationRequest({ subscriptionId, providerId, location, options }, listOperationSpec);
    }
    /**
     * ListNext
     * @param subscriptionId Azure subscription ID.
     * @param providerId Azure resource provider ID.
     * @param location Azure region.
     * @param nextLink The nextLink from the previous successful call to the List method.
     * @param options The options parameters.
     */
    _listNext(subscriptionId, providerId, location, nextLink, options) {
        return this.client.sendOperationRequest({ subscriptionId, providerId, location, nextLink, options }, listNextOperationSpec);
    }
}
// Operation Specifications
const serializer = coreClient.createSerializer(Mappers, /* isXml */ false);
const getOperationSpec = {
    path: "/subscriptions/{subscriptionId}/providers/Microsoft.Capacity/resourceProviders/{providerId}/locations/{location}/serviceLimitsRequests/{id}",
    httpMethod: "GET",
    responses: {
        200: {
            bodyMapper: Mappers.QuotaRequestDetails
        },
        default: {
            bodyMapper: Mappers.ExceptionResponse
        }
    },
    queryParameters: [Parameters.apiVersion1],
    urlParameters: [
        Parameters.$host,
        Parameters.subscriptionId,
        Parameters.providerId,
        Parameters.location1,
        Parameters.id
    ],
    headerParameters: [Parameters.accept],
    serializer
};
const listOperationSpec = {
    path: "/subscriptions/{subscriptionId}/providers/Microsoft.Capacity/resourceProviders/{providerId}/locations/{location}/serviceLimitsRequests",
    httpMethod: "GET",
    responses: {
        200: {
            bodyMapper: Mappers.QuotaRequestDetailsList
        },
        default: {
            bodyMapper: Mappers.ExceptionResponse
        }
    },
    queryParameters: [
        Parameters.filter,
        Parameters.apiVersion1,
        Parameters.top,
        Parameters.skiptoken1
    ],
    urlParameters: [
        Parameters.$host,
        Parameters.subscriptionId,
        Parameters.providerId,
        Parameters.location1
    ],
    headerParameters: [Parameters.accept],
    serializer
};
const listNextOperationSpec = {
    path: "{nextLink}",
    httpMethod: "GET",
    responses: {
        200: {
            bodyMapper: Mappers.QuotaRequestDetailsList
        },
        default: {
            bodyMapper: Mappers.ExceptionResponse
        }
    },
    urlParameters: [
        Parameters.$host,
        Parameters.nextLink,
        Parameters.subscriptionId,
        Parameters.providerId,
        Parameters.location1
    ],
    headerParameters: [Parameters.accept],
    serializer
};
