// @azure/arm-reservations/dist-esm/src/operations/reservationOrder.js
/*
 * Copyright (c) Microsoft Corporation.
 * Licensed under the MIT License.
 *
 * Code generated by Microsoft (R) AutoRest Code Generator.
 * Changes may cause incorrect behavior and will be lost if the code is regenerated.
 */
import { __asyncDelegator, __asyncGenerator, __asyncValues, __await, __awaiter } from "tslib";
import { setContinuationToken } from "../pagingHelper";
import * as coreClient from "@azure/core-client";
import * as Mappers from "../models/mappers";
import * as Parameters from "../models/parameters";
import { createHttpPoller } from "@azure/core-lro";
import { createLroSpec } from "../lroImpl";
/// <reference lib="esnext.asynciterable" />
/** Class containing ReservationOrder operations. */
export class ReservationOrderImpl {
    /**
     * Initialize a new instance of the class ReservationOrder class.
     * @param client Reference to the service client
     */
    constructor(client) {
        this.client = client;
    }
    /**
     * List of all the `ReservationOrder`s that the user has access to in the current tenant.
     * @param options The options parameters.
     */
    list(options) {
        const iter = this.listPagingAll(options);
        return {
            next() {
                return iter.next();
            },
            [Symbol.asyncIterator]() {
                return this;
            },
            byPage: (settings) => {
                if (settings === null || settings === void 0 ? void 0 : settings.maxPageSize) {
                    throw new Error("maxPageSize is not supported by this operation.");
                }
                return this.listPagingPage(options, settings);
            }
        };
    }
    listPagingPage(options, settings) {
        return __asyncGenerator(this, arguments, function* listPagingPage_1() {
            let result;
            let continuationToken = settings === null || settings === void 0 ? void 0 : settings.continuationToken;
            if (!continuationToken) {
                result = yield __await(this._list(options));
                let page = result.value || [];
                continuationToken = result.nextLink;
                setContinuationToken(page, continuationToken);
                yield yield __await(page);
            }
            while (continuationToken) {
                result = yield __await(this._listNext(continuationToken, options));
                continuationToken = result.nextLink;
                let page = result.value || [];
                setContinuationToken(page, continuationToken);
                yield yield __await(page);
            }
        });
    }
    listPagingAll(options) {
        return __asyncGenerator(this, arguments, function* listPagingAll_1() {
            var e_1, _a;
            try {
                for (var _b = __asyncValues(this.listPagingPage(options)), _c; _c = yield __await(_b.next()), !_c.done;) {
                    const page = _c.value;
                    yield __await(yield* __asyncDelegator(__asyncValues(page)));
                }
            }
            catch (e_1_1) { e_1 = { error: e_1_1 }; }
            finally {
                try {
                    if (_c && !_c.done && (_a = _b.return)) yield __await(_a.call(_b));
                }
                finally { if (e_1) throw e_1.error; }
            }
        });
    }
    /**
     * Calculate price for placing a `ReservationOrder`.
     * @param body Information needed for calculate or purchase reservation
     * @param options The options parameters.
     */
    calculate(body, options) {
        return this.client.sendOperationRequest({ body, options }, calculateOperationSpec);
    }
    /**
     * List of all the `ReservationOrder`s that the user has access to in the current tenant.
     * @param options The options parameters.
     */
    _list(options) {
        return this.client.sendOperationRequest({ options }, listOperationSpec);
    }
    /**
     * Purchase `ReservationOrder` and create resource under the specified URI.
     * @param reservationOrderId Order Id of the reservation
     * @param body Information needed for calculate or purchase reservation
     * @param options The options parameters.
     */
    beginPurchase(reservationOrderId, body, options) {
        return __awaiter(this, void 0, void 0, function* () {
            const directSendOperation = (args, spec) => __awaiter(this, void 0, void 0, function* () {
                return this.client.sendOperationRequest(args, spec);
            });
            const sendOperationFn = (args, spec) => __awaiter(this, void 0, void 0, function* () {
                var _a;
                let currentRawResponse = undefined;
                const providedCallback = (_a = args.options) === null || _a === void 0 ? void 0 : _a.onResponse;
                const callback = (rawResponse, flatResponse) => {
                    currentRawResponse = rawResponse;
                    providedCallback === null || providedCallback === void 0 ? void 0 : providedCallback(rawResponse, flatResponse);
                };
                const updatedArgs = Object.assign(Object.assign({}, args), { options: Object.assign(Object.assign({}, args.options), { onResponse: callback }) });
                const flatResponse = yield directSendOperation(updatedArgs, spec);
                return {
                    flatResponse,
                    rawResponse: {
                        statusCode: currentRawResponse.status,
                        body: currentRawResponse.parsedBody,
                        headers: currentRawResponse.headers.toJSON()
                    }
                };
            });
            const lro = createLroSpec({
                sendOperationFn,
                args: { reservationOrderId, body, options },
                spec: purchaseOperationSpec
            });
            const poller = yield createHttpPoller(lro, {
                restoreFrom: options === null || options === void 0 ? void 0 : options.resumeFrom,
                intervalInMs: options === null || options === void 0 ? void 0 : options.updateIntervalInMs,
                resourceLocationConfig: "location"
            });
            yield poller.poll();
            return poller;
        });
    }
    /**
     * Purchase `ReservationOrder` and create resource under the specified URI.
     * @param reservationOrderId Order Id of the reservation
     * @param body Information needed for calculate or purchase reservation
     * @param options The options parameters.
     */
    beginPurchaseAndWait(reservationOrderId, body, options) {
        return __awaiter(this, void 0, void 0, function* () {
            const poller = yield this.beginPurchase(reservationOrderId, body, options);
            return poller.pollUntilDone();
        });
    }
    /**
     * Get the details of the `ReservationOrder`.
     * @param reservationOrderId Order Id of the reservation
     * @param options The options parameters.
     */
    get(reservationOrderId, options) {
        return this.client.sendOperationRequest({ reservationOrderId, options }, getOperationSpec);
    }
    /**
     * Change directory (tenant) of `ReservationOrder` and all `Reservation` under it to specified tenant
     * id
     * @param reservationOrderId Order Id of the reservation
     * @param body Information needed to change directory of reservation order
     * @param options The options parameters.
     */
    changeDirectory(reservationOrderId, body, options) {
        return this.client.sendOperationRequest({ reservationOrderId, body, options }, changeDirectoryOperationSpec);
    }
    /**
     * ListNext
     * @param nextLink The nextLink from the previous successful call to the List method.
     * @param options The options parameters.
     */
    _listNext(nextLink, options) {
        return this.client.sendOperationRequest({ nextLink, options }, listNextOperationSpec);
    }
}
// Operation Specifications
const serializer = coreClient.createSerializer(Mappers, /* isXml */ false);
const calculateOperationSpec = {
    path: "/providers/Microsoft.Capacity/calculatePrice",
    httpMethod: "POST",
    responses: {
        200: {
            bodyMapper: Mappers.CalculatePriceResponse
        },
        default: {
            bodyMapper: Mappers.ErrorModel
        }
    },
    requestBody: Parameters.body3,
    queryParameters: [Parameters.apiVersion],
    urlParameters: [Parameters.$host],
    headerParameters: [Parameters.contentType, Parameters.accept],
    mediaType: "json",
    serializer
};
const listOperationSpec = {
    path: "/providers/Microsoft.Capacity/reservationOrders",
    httpMethod: "GET",
    responses: {
        200: {
            bodyMapper: Mappers.ReservationOrderList
        },
        default: {
            bodyMapper: Mappers.ErrorModel
        }
    },
    queryParameters: [Parameters.apiVersion],
    urlParameters: [Parameters.$host],
    headerParameters: [Parameters.accept],
    serializer
};
const purchaseOperationSpec = {
    path: "/providers/Microsoft.Capacity/reservationOrders/{reservationOrderId}",
    httpMethod: "PUT",
    responses: {
        200: {
            bodyMapper: Mappers.ReservationOrderResponse
        },
        201: {
            bodyMapper: Mappers.ReservationOrderResponse
        },
        202: {
            bodyMapper: Mappers.ReservationOrderResponse
        },
        204: {
            bodyMapper: Mappers.ReservationOrderResponse
        },
        default: {
            bodyMapper: Mappers.ErrorModel
        }
    },
    requestBody: Parameters.body3,
    queryParameters: [Parameters.apiVersion],
    urlParameters: [Parameters.$host, Parameters.reservationOrderId],
    headerParameters: [Parameters.contentType, Parameters.accept],
    mediaType: "json",
    serializer
};
const getOperationSpec = {
    path: "/providers/Microsoft.Capacity/reservationOrders/{reservationOrderId}",
    httpMethod: "GET",
    responses: {
        200: {
            bodyMapper: Mappers.ReservationOrderResponse
        },
        default: {
            bodyMapper: Mappers.ErrorModel
        }
    },
    queryParameters: [Parameters.apiVersion, Parameters.expand],
    urlParameters: [Parameters.$host, Parameters.reservationOrderId],
    headerParameters: [Parameters.accept],
    serializer
};
const changeDirectoryOperationSpec = {
    path: "/providers/Microsoft.Capacity/reservationOrders/{reservationOrderId}/changeDirectory",
    httpMethod: "POST",
    responses: {
        200: {
            bodyMapper: Mappers.ChangeDirectoryResponse
        },
        default: {
            bodyMapper: Mappers.ErrorModel
        }
    },
    requestBody: Parameters.body4,
    queryParameters: [Parameters.apiVersion],
    urlParameters: [Parameters.$host, Parameters.reservationOrderId],
    headerParameters: [Parameters.contentType, Parameters.accept],
    mediaType: "json",
    serializer
};
const listNextOperationSpec = {
    path: "{nextLink}",
    httpMethod: "GET",
    responses: {
        200: {
            bodyMapper: Mappers.ReservationOrderList
        },
        default: {
            bodyMapper: Mappers.ErrorModel
        }
    },
    urlParameters: [Parameters.$host, Parameters.nextLink],
    headerParameters: [Parameters.accept],
    serializer
};
