// @azure/arm-reservations/dist-esm/src/operations/return.js
/*
 * Copyright (c) Microsoft Corporation.
 * Licensed under the MIT License.
 *
 * Code generated by Microsoft (R) AutoRest Code Generator.
 * Changes may cause incorrect behavior and will be lost if the code is regenerated.
 */
import { __awaiter } from "tslib";
import * as coreClient from "@azure/core-client";
import * as Mappers from "../models/mappers";
import * as Parameters from "../models/parameters";
import { createHttpPoller } from "@azure/core-lro";
import { createLroSpec } from "../lroImpl";
/** Class containing Return operations. */
export class ReturnImpl {
    /**
     * Initialize a new instance of the class Return class.
     * @param client Reference to the service client
     */
    constructor(client) {
        this.client = client;
    }
    /**
     * Return a reservation and get refund information.
     * @param reservationOrderId Order Id of the reservation
     * @param body Information needed for returning reservation.
     * @param options The options parameters.
     */
    beginPost(reservationOrderId, body, options) {
        return __awaiter(this, void 0, void 0, function* () {
            const directSendOperation = (args, spec) => __awaiter(this, void 0, void 0, function* () {
                return this.client.sendOperationRequest(args, spec);
            });
            const sendOperationFn = (args, spec) => __awaiter(this, void 0, void 0, function* () {
                var _a;
                let currentRawResponse = undefined;
                const providedCallback = (_a = args.options) === null || _a === void 0 ? void 0 : _a.onResponse;
                const callback = (rawResponse, flatResponse) => {
                    currentRawResponse = rawResponse;
                    providedCallback === null || providedCallback === void 0 ? void 0 : providedCallback(rawResponse, flatResponse);
                };
                const updatedArgs = Object.assign(Object.assign({}, args), { options: Object.assign(Object.assign({}, args.options), { onResponse: callback }) });
                const flatResponse = yield directSendOperation(updatedArgs, spec);
                return {
                    flatResponse,
                    rawResponse: {
                        statusCode: currentRawResponse.status,
                        body: currentRawResponse.parsedBody,
                        headers: currentRawResponse.headers.toJSON()
                    }
                };
            });
            const lro = createLroSpec({
                sendOperationFn,
                args: { reservationOrderId, body, options },
                spec: postOperationSpec
            });
            const poller = yield createHttpPoller(lro, {
                restoreFrom: options === null || options === void 0 ? void 0 : options.resumeFrom,
                intervalInMs: options === null || options === void 0 ? void 0 : options.updateIntervalInMs,
                resourceLocationConfig: "location"
            });
            yield poller.poll();
            return poller;
        });
    }
    /**
     * Return a reservation and get refund information.
     * @param reservationOrderId Order Id of the reservation
     * @param body Information needed for returning reservation.
     * @param options The options parameters.
     */
    beginPostAndWait(reservationOrderId, body, options) {
        return __awaiter(this, void 0, void 0, function* () {
            const poller = yield this.beginPost(reservationOrderId, body, options);
            return poller.pollUntilDone();
        });
    }
}
// Operation Specifications
const serializer = coreClient.createSerializer(Mappers, /* isXml */ false);
const postOperationSpec = {
    path: "/providers/Microsoft.Capacity/reservationOrders/{reservationOrderId}/return",
    httpMethod: "POST",
    responses: {
        200: {
            bodyMapper: Mappers.ReservationOrderResponse
        },
        201: {
            bodyMapper: Mappers.ReservationOrderResponse
        },
        202: {
            bodyMapper: Mappers.ReservationOrderResponse
        },
        204: {
            bodyMapper: Mappers.ReservationOrderResponse
        },
        default: {
            bodyMapper: Mappers.ErrorModel
        }
    },
    requestBody: Parameters.body6,
    queryParameters: [Parameters.apiVersion],
    urlParameters: [Parameters.$host, Parameters.reservationOrderId],
    headerParameters: [Parameters.contentType, Parameters.accept],
    mediaType: "json",
    serializer
};
