// @azure/arm-appconfiguration/dist-esm/src/models/index.js
/*
 * Copyright (c) Microsoft Corporation.
 * Licensed under the MIT License.
 *
 * Code generated by Microsoft (R) AutoRest Code Generator.
 * Changes may cause incorrect behavior and will be lost if the code is regenerated.
 */
/** Known values of {@link IdentityType} that the service accepts. */
export var KnownIdentityType;
(function (KnownIdentityType) {
    /** None */
    KnownIdentityType["None"] = "None";
    /** SystemAssigned */
    KnownIdentityType["SystemAssigned"] = "SystemAssigned";
    /** UserAssigned */
    KnownIdentityType["UserAssigned"] = "UserAssigned";
    /** SystemAssignedUserAssigned */
    KnownIdentityType["SystemAssignedUserAssigned"] = "SystemAssigned, UserAssigned";
})(KnownIdentityType || (KnownIdentityType = {}));
/** Known values of {@link ProvisioningState} that the service accepts. */
export var KnownProvisioningState;
(function (KnownProvisioningState) {
    /** Creating */
    KnownProvisioningState["Creating"] = "Creating";
    /** Updating */
    KnownProvisioningState["Updating"] = "Updating";
    /** Deleting */
    KnownProvisioningState["Deleting"] = "Deleting";
    /** Succeeded */
    KnownProvisioningState["Succeeded"] = "Succeeded";
    /** Failed */
    KnownProvisioningState["Failed"] = "Failed";
    /** Canceled */
    KnownProvisioningState["Canceled"] = "Canceled";
})(KnownProvisioningState || (KnownProvisioningState = {}));
/** Known values of {@link ConnectionStatus} that the service accepts. */
export var KnownConnectionStatus;
(function (KnownConnectionStatus) {
    /** Pending */
    KnownConnectionStatus["Pending"] = "Pending";
    /** Approved */
    KnownConnectionStatus["Approved"] = "Approved";
    /** Rejected */
    KnownConnectionStatus["Rejected"] = "Rejected";
    /** Disconnected */
    KnownConnectionStatus["Disconnected"] = "Disconnected";
})(KnownConnectionStatus || (KnownConnectionStatus = {}));
/** Known values of {@link ActionsRequired} that the service accepts. */
export var KnownActionsRequired;
(function (KnownActionsRequired) {
    /** None */
    KnownActionsRequired["None"] = "None";
    /** Recreate */
    KnownActionsRequired["Recreate"] = "Recreate";
})(KnownActionsRequired || (KnownActionsRequired = {}));
/** Known values of {@link PublicNetworkAccess} that the service accepts. */
export var KnownPublicNetworkAccess;
(function (KnownPublicNetworkAccess) {
    /** Enabled */
    KnownPublicNetworkAccess["Enabled"] = "Enabled";
    /** Disabled */
    KnownPublicNetworkAccess["Disabled"] = "Disabled";
})(KnownPublicNetworkAccess || (KnownPublicNetworkAccess = {}));
/** Known values of {@link CreatedByType} that the service accepts. */
export var KnownCreatedByType;
(function (KnownCreatedByType) {
    /** User */
    KnownCreatedByType["User"] = "User";
    /** Application */
    KnownCreatedByType["Application"] = "Application";
    /** ManagedIdentity */
    KnownCreatedByType["ManagedIdentity"] = "ManagedIdentity";
    /** Key */
    KnownCreatedByType["Key"] = "Key";
})(KnownCreatedByType || (KnownCreatedByType = {}));
/** Known values of {@link ConfigurationResourceType} that the service accepts. */
export var KnownConfigurationResourceType;
(function (KnownConfigurationResourceType) {
    /** MicrosoftAppConfigurationConfigurationStores */
    KnownConfigurationResourceType["MicrosoftAppConfigurationConfigurationStores"] = "Microsoft.AppConfiguration/configurationStores";
})(KnownConfigurationResourceType || (KnownConfigurationResourceType = {}));
/** Known values of {@link ReplicaProvisioningState} that the service accepts. */
export var KnownReplicaProvisioningState;
(function (KnownReplicaProvisioningState) {
    /** Creating */
    KnownReplicaProvisioningState["Creating"] = "Creating";
    /** Succeeded */
    KnownReplicaProvisioningState["Succeeded"] = "Succeeded";
    /** Deleting */
    KnownReplicaProvisioningState["Deleting"] = "Deleting";
    /** Failed */
    KnownReplicaProvisioningState["Failed"] = "Failed";
    /** Canceled */
    KnownReplicaProvisioningState["Canceled"] = "Canceled";
})(KnownReplicaProvisioningState || (KnownReplicaProvisioningState = {}));
