// @aws-sdk/client-personalize/dist-cjs/index.js
"use strict";
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __name = (target, value) => __defProp(target, "name", { value, configurable: true });
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// src/index.ts
var src_exports = {};
__export(src_exports, {
  BatchInferenceJobMode: () => BatchInferenceJobMode,
  CreateBatchInferenceJobCommand: () => CreateBatchInferenceJobCommand,
  CreateBatchSegmentJobCommand: () => CreateBatchSegmentJobCommand,
  CreateCampaignCommand: () => CreateCampaignCommand,
  CreateDatasetCommand: () => CreateDatasetCommand,
  CreateDatasetExportJobCommand: () => CreateDatasetExportJobCommand,
  CreateDatasetGroupCommand: () => CreateDatasetGroupCommand,
  CreateDatasetImportJobCommand: () => CreateDatasetImportJobCommand,
  CreateEventTrackerCommand: () => CreateEventTrackerCommand,
  CreateFilterCommand: () => CreateFilterCommand,
  CreateFilterRequestFilterSensitiveLog: () => CreateFilterRequestFilterSensitiveLog,
  CreateMetricAttributionCommand: () => CreateMetricAttributionCommand,
  CreateRecommenderCommand: () => CreateRecommenderCommand,
  CreateSchemaCommand: () => CreateSchemaCommand,
  CreateSolutionCommand: () => CreateSolutionCommand,
  CreateSolutionVersionCommand: () => CreateSolutionVersionCommand,
  DeleteCampaignCommand: () => DeleteCampaignCommand,
  DeleteDatasetCommand: () => DeleteDatasetCommand,
  DeleteDatasetGroupCommand: () => DeleteDatasetGroupCommand,
  DeleteEventTrackerCommand: () => DeleteEventTrackerCommand,
  DeleteFilterCommand: () => DeleteFilterCommand,
  DeleteMetricAttributionCommand: () => DeleteMetricAttributionCommand,
  DeleteRecommenderCommand: () => DeleteRecommenderCommand,
  DeleteSchemaCommand: () => DeleteSchemaCommand,
  DeleteSolutionCommand: () => DeleteSolutionCommand,
  DescribeAlgorithmCommand: () => DescribeAlgorithmCommand,
  DescribeBatchInferenceJobCommand: () => DescribeBatchInferenceJobCommand,
  DescribeBatchSegmentJobCommand: () => DescribeBatchSegmentJobCommand,
  DescribeCampaignCommand: () => DescribeCampaignCommand,
  DescribeDatasetCommand: () => DescribeDatasetCommand,
  DescribeDatasetExportJobCommand: () => DescribeDatasetExportJobCommand,
  DescribeDatasetGroupCommand: () => DescribeDatasetGroupCommand,
  DescribeDatasetImportJobCommand: () => DescribeDatasetImportJobCommand,
  DescribeEventTrackerCommand: () => DescribeEventTrackerCommand,
  DescribeFeatureTransformationCommand: () => DescribeFeatureTransformationCommand,
  DescribeFilterCommand: () => DescribeFilterCommand,
  DescribeFilterResponseFilterSensitiveLog: () => DescribeFilterResponseFilterSensitiveLog,
  DescribeMetricAttributionCommand: () => DescribeMetricAttributionCommand,
  DescribeRecipeCommand: () => DescribeRecipeCommand,
  DescribeRecommenderCommand: () => DescribeRecommenderCommand,
  DescribeSchemaCommand: () => DescribeSchemaCommand,
  DescribeSolutionCommand: () => DescribeSolutionCommand,
  DescribeSolutionVersionCommand: () => DescribeSolutionVersionCommand,
  Domain: () => Domain,
  FilterFilterSensitiveLog: () => FilterFilterSensitiveLog,
  GetSolutionMetricsCommand: () => GetSolutionMetricsCommand,
  ImportMode: () => ImportMode,
  IngestionMode: () => IngestionMode,
  InvalidInputException: () => InvalidInputException,
  InvalidNextTokenException: () => InvalidNextTokenException,
  LimitExceededException: () => LimitExceededException,
  ListBatchInferenceJobsCommand: () => ListBatchInferenceJobsCommand,
  ListBatchSegmentJobsCommand: () => ListBatchSegmentJobsCommand,
  ListCampaignsCommand: () => ListCampaignsCommand,
  ListDatasetExportJobsCommand: () => ListDatasetExportJobsCommand,
  ListDatasetGroupsCommand: () => ListDatasetGroupsCommand,
  ListDatasetImportJobsCommand: () => ListDatasetImportJobsCommand,
  ListDatasetsCommand: () => ListDatasetsCommand,
  ListEventTrackersCommand: () => ListEventTrackersCommand,
  ListFiltersCommand: () => ListFiltersCommand,
  ListMetricAttributionMetricsCommand: () => ListMetricAttributionMetricsCommand,
  ListMetricAttributionsCommand: () => ListMetricAttributionsCommand,
  ListRecipesCommand: () => ListRecipesCommand,
  ListRecommendersCommand: () => ListRecommendersCommand,
  ListSchemasCommand: () => ListSchemasCommand,
  ListSolutionVersionsCommand: () => ListSolutionVersionsCommand,
  ListSolutionsCommand: () => ListSolutionsCommand,
  ListTagsForResourceCommand: () => ListTagsForResourceCommand,
  ObjectiveSensitivity: () => ObjectiveSensitivity,
  Personalize: () => Personalize,
  PersonalizeClient: () => PersonalizeClient,
  PersonalizeServiceException: () => PersonalizeServiceException,
  RecipeProvider: () => RecipeProvider,
  ResourceAlreadyExistsException: () => ResourceAlreadyExistsException,
  ResourceInUseException: () => ResourceInUseException,
  ResourceNotFoundException: () => ResourceNotFoundException,
  StartRecommenderCommand: () => StartRecommenderCommand,
  StopRecommenderCommand: () => StopRecommenderCommand,
  StopSolutionVersionCreationCommand: () => StopSolutionVersionCreationCommand,
  TagResourceCommand: () => TagResourceCommand,
  TooManyTagKeysException: () => TooManyTagKeysException,
  TooManyTagsException: () => TooManyTagsException,
  TrainingMode: () => TrainingMode,
  TrainingType: () => TrainingType,
  UntagResourceCommand: () => UntagResourceCommand,
  UpdateCampaignCommand: () => UpdateCampaignCommand,
  UpdateDatasetCommand: () => UpdateDatasetCommand,
  UpdateMetricAttributionCommand: () => UpdateMetricAttributionCommand,
  UpdateRecommenderCommand: () => UpdateRecommenderCommand,
  __Client: () => import_smithy_client.Client,
  paginateListBatchInferenceJobs: () => paginateListBatchInferenceJobs,
  paginateListBatchSegmentJobs: () => paginateListBatchSegmentJobs,
  paginateListCampaigns: () => paginateListCampaigns,
  paginateListDatasetExportJobs: () => paginateListDatasetExportJobs,
  paginateListDatasetGroups: () => paginateListDatasetGroups,
  paginateListDatasetImportJobs: () => paginateListDatasetImportJobs,
  paginateListDatasets: () => paginateListDatasets,
  paginateListEventTrackers: () => paginateListEventTrackers,
  paginateListFilters: () => paginateListFilters,
  paginateListMetricAttributionMetrics: () => paginateListMetricAttributionMetrics,
  paginateListMetricAttributions: () => paginateListMetricAttributions,
  paginateListRecipes: () => paginateListRecipes,
  paginateListRecommenders: () => paginateListRecommenders,
  paginateListSchemas: () => paginateListSchemas,
  paginateListSolutionVersions: () => paginateListSolutionVersions,
  paginateListSolutions: () => paginateListSolutions
});
module.exports = __toCommonJS(src_exports);

// src/PersonalizeClient.ts
var import_middleware_host_header = require("@aws-sdk/middleware-host-header");
var import_middleware_logger = require("@aws-sdk/middleware-logger");
var import_middleware_recursion_detection = require("@aws-sdk/middleware-recursion-detection");
var import_middleware_user_agent = require("@aws-sdk/middleware-user-agent");
var import_config_resolver = require("@smithy/config-resolver");
var import_core = require("@smithy/core");
var import_middleware_content_length = require("@smithy/middleware-content-length");
var import_middleware_endpoint = require("@smithy/middleware-endpoint");
var import_middleware_retry = require("@smithy/middleware-retry");

var import_httpAuthSchemeProvider = require("./auth/httpAuthSchemeProvider");

// src/endpoint/EndpointParameters.ts
var resolveClientEndpointParameters = /* @__PURE__ */ __name((options) => {
  return {
    ...options,
    useDualstackEndpoint: options.useDualstackEndpoint ?? false,
    useFipsEndpoint: options.useFipsEndpoint ?? false,
    defaultSigningName: "personalize"
  };
}, "resolveClientEndpointParameters");
var commonParams = {
  UseFIPS: { type: "builtInParams", name: "useFipsEndpoint" },
  Endpoint: { type: "builtInParams", name: "endpoint" },
  Region: { type: "builtInParams", name: "region" },
  UseDualStack: { type: "builtInParams", name: "useDualstackEndpoint" }
};

// src/PersonalizeClient.ts
var import_runtimeConfig = require("././runtimeConfig");

// src/runtimeExtensions.ts
var import_region_config_resolver = require("@aws-sdk/region-config-resolver");
var import_protocol_http = require("@smithy/protocol-http");
var import_smithy_client = require("@smithy/smithy-client");

// src/auth/httpAuthExtensionConfiguration.ts
var getHttpAuthExtensionConfiguration = /* @__PURE__ */ __name((runtimeConfig) => {
  const _httpAuthSchemes = runtimeConfig.httpAuthSchemes;
  let _httpAuthSchemeProvider = runtimeConfig.httpAuthSchemeProvider;
  let _credentials = runtimeConfig.credentials;
  return {
    setHttpAuthScheme(httpAuthScheme) {
      const index = _httpAuthSchemes.findIndex((scheme) => scheme.schemeId === httpAuthScheme.schemeId);
      if (index === -1) {
        _httpAuthSchemes.push(httpAuthScheme);
      } else {
        _httpAuthSchemes.splice(index, 1, httpAuthScheme);
      }
    },
    httpAuthSchemes() {
      return _httpAuthSchemes;
    },
    setHttpAuthSchemeProvider(httpAuthSchemeProvider) {
      _httpAuthSchemeProvider = httpAuthSchemeProvider;
    },
    httpAuthSchemeProvider() {
      return _httpAuthSchemeProvider;
    },
    setCredentials(credentials) {
      _credentials = credentials;
    },
    credentials() {
      return _credentials;
    }
  };
}, "getHttpAuthExtensionConfiguration");
var resolveHttpAuthRuntimeConfig = /* @__PURE__ */ __name((config) => {
  return {
    httpAuthSchemes: config.httpAuthSchemes(),
    httpAuthSchemeProvider: config.httpAuthSchemeProvider(),
    credentials: config.credentials()
  };
}, "resolveHttpAuthRuntimeConfig");

// src/runtimeExtensions.ts
var asPartial = /* @__PURE__ */ __name((t) => t, "asPartial");
var resolveRuntimeExtensions = /* @__PURE__ */ __name((runtimeConfig, extensions) => {
  const extensionConfiguration = {
    ...asPartial((0, import_region_config_resolver.getAwsRegionExtensionConfiguration)(runtimeConfig)),
    ...asPartial((0, import_smithy_client.getDefaultExtensionConfiguration)(runtimeConfig)),
    ...asPartial((0, import_protocol_http.getHttpHandlerExtensionConfiguration)(runtimeConfig)),
    ...asPartial(getHttpAuthExtensionConfiguration(runtimeConfig))
  };
  extensions.forEach((extension) => extension.configure(extensionConfiguration));
  return {
    ...runtimeConfig,
    ...(0, import_region_config_resolver.resolveAwsRegionExtensionConfiguration)(extensionConfiguration),
    ...(0, import_smithy_client.resolveDefaultRuntimeConfig)(extensionConfiguration),
    ...(0, import_protocol_http.resolveHttpHandlerRuntimeConfig)(extensionConfiguration),
    ...resolveHttpAuthRuntimeConfig(extensionConfiguration)
  };
}, "resolveRuntimeExtensions");

// src/PersonalizeClient.ts
var _PersonalizeClient = class _PersonalizeClient extends import_smithy_client.Client {
  constructor(...[configuration]) {
    const _config_0 = (0, import_runtimeConfig.getRuntimeConfig)(configuration || {});
    const _config_1 = resolveClientEndpointParameters(_config_0);
    const _config_2 = (0, import_config_resolver.resolveRegionConfig)(_config_1);
    const _config_3 = (0, import_middleware_endpoint.resolveEndpointConfig)(_config_2);
    const _config_4 = (0, import_middleware_retry.resolveRetryConfig)(_config_3);
    const _config_5 = (0, import_middleware_host_header.resolveHostHeaderConfig)(_config_4);
    const _config_6 = (0, import_middleware_user_agent.resolveUserAgentConfig)(_config_5);
    const _config_7 = (0, import_httpAuthSchemeProvider.resolveHttpAuthSchemeConfig)(_config_6);
    const _config_8 = resolveRuntimeExtensions(_config_7, (configuration == null ? void 0 : configuration.extensions) || []);
    super(_config_8);
    this.config = _config_8;
    this.middlewareStack.use((0, import_middleware_retry.getRetryPlugin)(this.config));
    this.middlewareStack.use((0, import_middleware_content_length.getContentLengthPlugin)(this.config));
    this.middlewareStack.use((0, import_middleware_host_header.getHostHeaderPlugin)(this.config));
    this.middlewareStack.use((0, import_middleware_logger.getLoggerPlugin)(this.config));
    this.middlewareStack.use((0, import_middleware_recursion_detection.getRecursionDetectionPlugin)(this.config));
    this.middlewareStack.use((0, import_middleware_user_agent.getUserAgentPlugin)(this.config));
    this.middlewareStack.use(
      (0, import_core.getHttpAuthSchemeEndpointRuleSetPlugin)(this.config, {
        httpAuthSchemeParametersProvider: this.getDefaultHttpAuthSchemeParametersProvider(),
        identityProviderConfigProvider: this.getIdentityProviderConfigProvider()
      })
    );
    this.middlewareStack.use((0, import_core.getHttpSigningPlugin)(this.config));
  }
  /**
   * Destroy underlying resources, like sockets. It's usually not necessary to do this.
   * However in Node.js, it's best to explicitly shut down the client's agent when it is no longer needed.
   * Otherwise, sockets might stay open for quite a long time before the server terminates them.
   */
  destroy() {
    super.destroy();
  }
  getDefaultHttpAuthSchemeParametersProvider() {
    return import_httpAuthSchemeProvider.defaultPersonalizeHttpAuthSchemeParametersProvider;
  }
  getIdentityProviderConfigProvider() {
    return async (config) => new import_core.DefaultIdentityProviderConfig({
      "aws.auth#sigv4": config.credentials
    });
  }
};
__name(_PersonalizeClient, "PersonalizeClient");
var PersonalizeClient = _PersonalizeClient;

// src/Personalize.ts


// src/commands/CreateBatchInferenceJobCommand.ts

var import_middleware_serde = require("@smithy/middleware-serde");

var import_types = require("@smithy/types");

// src/protocols/Aws_json1_1.ts
var import_core2 = require("@aws-sdk/core");



// src/models/models_0.ts


// src/models/PersonalizeServiceException.ts

var _PersonalizeServiceException = class _PersonalizeServiceException extends import_smithy_client.ServiceException {
  /**
   * @internal
   */
  constructor(options) {
    super(options);
    Object.setPrototypeOf(this, _PersonalizeServiceException.prototype);
  }
};
__name(_PersonalizeServiceException, "PersonalizeServiceException");
var PersonalizeServiceException = _PersonalizeServiceException;

// src/models/models_0.ts
var BatchInferenceJobMode = {
  BATCH_INFERENCE: "BATCH_INFERENCE",
  THEME_GENERATION: "THEME_GENERATION"
};
var _InvalidInputException = class _InvalidInputException extends PersonalizeServiceException {
  /**
   * @internal
   */
  constructor(opts) {
    super({
      name: "InvalidInputException",
      $fault: "client",
      ...opts
    });
    this.name = "InvalidInputException";
    this.$fault = "client";
    Object.setPrototypeOf(this, _InvalidInputException.prototype);
  }
};
__name(_InvalidInputException, "InvalidInputException");
var InvalidInputException = _InvalidInputException;
var _LimitExceededException = class _LimitExceededException extends PersonalizeServiceException {
  /**
   * @internal
   */
  constructor(opts) {
    super({
      name: "LimitExceededException",
      $fault: "client",
      ...opts
    });
    this.name = "LimitExceededException";
    this.$fault = "client";
    Object.setPrototypeOf(this, _LimitExceededException.prototype);
  }
};
__name(_LimitExceededException, "LimitExceededException");
var LimitExceededException = _LimitExceededException;
var _ResourceAlreadyExistsException = class _ResourceAlreadyExistsException extends PersonalizeServiceException {
  /**
   * @internal
   */
  constructor(opts) {
    super({
      name: "ResourceAlreadyExistsException",
      $fault: "client",
      ...opts
    });
    this.name = "ResourceAlreadyExistsException";
    this.$fault = "client";
    Object.setPrototypeOf(this, _ResourceAlreadyExistsException.prototype);
  }
};
__name(_ResourceAlreadyExistsException, "ResourceAlreadyExistsException");
var ResourceAlreadyExistsException = _ResourceAlreadyExistsException;
var _ResourceInUseException = class _ResourceInUseException extends PersonalizeServiceException {
  /**
   * @internal
   */
  constructor(opts) {
    super({
      name: "ResourceInUseException",
      $fault: "client",
      ...opts
    });
    this.name = "ResourceInUseException";
    this.$fault = "client";
    Object.setPrototypeOf(this, _ResourceInUseException.prototype);
  }
};
__name(_ResourceInUseException, "ResourceInUseException");
var ResourceInUseException = _ResourceInUseException;
var _ResourceNotFoundException = class _ResourceNotFoundException extends PersonalizeServiceException {
  /**
   * @internal
   */
  constructor(opts) {
    super({
      name: "ResourceNotFoundException",
      $fault: "client",
      ...opts
    });
    this.name = "ResourceNotFoundException";
    this.$fault = "client";
    Object.setPrototypeOf(this, _ResourceNotFoundException.prototype);
  }
};
__name(_ResourceNotFoundException, "ResourceNotFoundException");
var ResourceNotFoundException = _ResourceNotFoundException;
var _TooManyTagsException = class _TooManyTagsException extends PersonalizeServiceException {
  /**
   * @internal
   */
  constructor(opts) {
    super({
      name: "TooManyTagsException",
      $fault: "client",
      ...opts
    });
    this.name = "TooManyTagsException";
    this.$fault = "client";
    Object.setPrototypeOf(this, _TooManyTagsException.prototype);
  }
};
__name(_TooManyTagsException, "TooManyTagsException");
var TooManyTagsException = _TooManyTagsException;
var IngestionMode = {
  ALL: "ALL",
  BULK: "BULK",
  PUT: "PUT"
};
var Domain = {
  ECOMMERCE: "ECOMMERCE",
  VIDEO_ON_DEMAND: "VIDEO_ON_DEMAND"
};
var ImportMode = {
  FULL: "FULL",
  INCREMENTAL: "INCREMENTAL"
};
var ObjectiveSensitivity = {
  HIGH: "HIGH",
  LOW: "LOW",
  MEDIUM: "MEDIUM",
  OFF: "OFF"
};
var TrainingMode = {
  AUTOTRAIN: "AUTOTRAIN",
  FULL: "FULL",
  UPDATE: "UPDATE"
};
var TrainingType = {
  AUTOMATIC: "AUTOMATIC",
  MANUAL: "MANUAL"
};
var _InvalidNextTokenException = class _InvalidNextTokenException extends PersonalizeServiceException {
  /**
   * @internal
   */
  constructor(opts) {
    super({
      name: "InvalidNextTokenException",
      $fault: "client",
      ...opts
    });
    this.name = "InvalidNextTokenException";
    this.$fault = "client";
    Object.setPrototypeOf(this, _InvalidNextTokenException.prototype);
  }
};
__name(_InvalidNextTokenException, "InvalidNextTokenException");
var InvalidNextTokenException = _InvalidNextTokenException;
var RecipeProvider = {
  SERVICE: "SERVICE"
};
var _TooManyTagKeysException = class _TooManyTagKeysException extends PersonalizeServiceException {
  /**
   * @internal
   */
  constructor(opts) {
    super({
      name: "TooManyTagKeysException",
      $fault: "client",
      ...opts
    });
    this.name = "TooManyTagKeysException";
    this.$fault = "client";
    Object.setPrototypeOf(this, _TooManyTagKeysException.prototype);
  }
};
__name(_TooManyTagKeysException, "TooManyTagKeysException");
var TooManyTagKeysException = _TooManyTagKeysException;
var CreateFilterRequestFilterSensitiveLog = /* @__PURE__ */ __name((obj) => ({
  ...obj,
  ...obj.filterExpression && { filterExpression: import_smithy_client.SENSITIVE_STRING }
}), "CreateFilterRequestFilterSensitiveLog");
var FilterFilterSensitiveLog = /* @__PURE__ */ __name((obj) => ({
  ...obj,
  ...obj.filterExpression && { filterExpression: import_smithy_client.SENSITIVE_STRING }
}), "FilterFilterSensitiveLog");
var DescribeFilterResponseFilterSensitiveLog = /* @__PURE__ */ __name((obj) => ({
  ...obj,
  ...obj.filter && { filter: FilterFilterSensitiveLog(obj.filter) }
}), "DescribeFilterResponseFilterSensitiveLog");

// src/protocols/Aws_json1_1.ts
var se_CreateBatchInferenceJobCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("CreateBatchInferenceJob");
  let body;
  body = JSON.stringify((0, import_smithy_client._json)(input));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_CreateBatchInferenceJobCommand");
var se_CreateBatchSegmentJobCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("CreateBatchSegmentJob");
  let body;
  body = JSON.stringify((0, import_smithy_client._json)(input));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_CreateBatchSegmentJobCommand");
var se_CreateCampaignCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("CreateCampaign");
  let body;
  body = JSON.stringify((0, import_smithy_client._json)(input));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_CreateCampaignCommand");
var se_CreateDatasetCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("CreateDataset");
  let body;
  body = JSON.stringify((0, import_smithy_client._json)(input));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_CreateDatasetCommand");
var se_CreateDatasetExportJobCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("CreateDatasetExportJob");
  let body;
  body = JSON.stringify((0, import_smithy_client._json)(input));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_CreateDatasetExportJobCommand");
var se_CreateDatasetGroupCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("CreateDatasetGroup");
  let body;
  body = JSON.stringify((0, import_smithy_client._json)(input));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_CreateDatasetGroupCommand");
var se_CreateDatasetImportJobCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("CreateDatasetImportJob");
  let body;
  body = JSON.stringify((0, import_smithy_client._json)(input));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_CreateDatasetImportJobCommand");
var se_CreateEventTrackerCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("CreateEventTracker");
  let body;
  body = JSON.stringify((0, import_smithy_client._json)(input));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_CreateEventTrackerCommand");
var se_CreateFilterCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("CreateFilter");
  let body;
  body = JSON.stringify((0, import_smithy_client._json)(input));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_CreateFilterCommand");
var se_CreateMetricAttributionCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("CreateMetricAttribution");
  let body;
  body = JSON.stringify((0, import_smithy_client._json)(input));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_CreateMetricAttributionCommand");
var se_CreateRecommenderCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("CreateRecommender");
  let body;
  body = JSON.stringify((0, import_smithy_client._json)(input));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_CreateRecommenderCommand");
var se_CreateSchemaCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("CreateSchema");
  let body;
  body = JSON.stringify((0, import_smithy_client._json)(input));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_CreateSchemaCommand");
var se_CreateSolutionCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("CreateSolution");
  let body;
  body = JSON.stringify(se_CreateSolutionRequest(input, context));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_CreateSolutionCommand");
var se_CreateSolutionVersionCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("CreateSolutionVersion");
  let body;
  body = JSON.stringify((0, import_smithy_client._json)(input));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_CreateSolutionVersionCommand");
var se_DeleteCampaignCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("DeleteCampaign");
  let body;
  body = JSON.stringify((0, import_smithy_client._json)(input));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_DeleteCampaignCommand");
var se_DeleteDatasetCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("DeleteDataset");
  let body;
  body = JSON.stringify((0, import_smithy_client._json)(input));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_DeleteDatasetCommand");
var se_DeleteDatasetGroupCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("DeleteDatasetGroup");
  let body;
  body = JSON.stringify((0, import_smithy_client._json)(input));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_DeleteDatasetGroupCommand");
var se_DeleteEventTrackerCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("DeleteEventTracker");
  let body;
  body = JSON.stringify((0, import_smithy_client._json)(input));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_DeleteEventTrackerCommand");
var se_DeleteFilterCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("DeleteFilter");
  let body;
  body = JSON.stringify((0, import_smithy_client._json)(input));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_DeleteFilterCommand");
var se_DeleteMetricAttributionCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("DeleteMetricAttribution");
  let body;
  body = JSON.stringify((0, import_smithy_client._json)(input));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_DeleteMetricAttributionCommand");
var se_DeleteRecommenderCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("DeleteRecommender");
  let body;
  body = JSON.stringify((0, import_smithy_client._json)(input));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_DeleteRecommenderCommand");
var se_DeleteSchemaCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("DeleteSchema");
  let body;
  body = JSON.stringify((0, import_smithy_client._json)(input));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_DeleteSchemaCommand");
var se_DeleteSolutionCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("DeleteSolution");
  let body;
  body = JSON.stringify((0, import_smithy_client._json)(input));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_DeleteSolutionCommand");
var se_DescribeAlgorithmCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("DescribeAlgorithm");
  let body;
  body = JSON.stringify((0, import_smithy_client._json)(input));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_DescribeAlgorithmCommand");
var se_DescribeBatchInferenceJobCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("DescribeBatchInferenceJob");
  let body;
  body = JSON.stringify((0, import_smithy_client._json)(input));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_DescribeBatchInferenceJobCommand");
var se_DescribeBatchSegmentJobCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("DescribeBatchSegmentJob");
  let body;
  body = JSON.stringify((0, import_smithy_client._json)(input));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_DescribeBatchSegmentJobCommand");
var se_DescribeCampaignCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("DescribeCampaign");
  let body;
  body = JSON.stringify((0, import_smithy_client._json)(input));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_DescribeCampaignCommand");
var se_DescribeDatasetCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("DescribeDataset");
  let body;
  body = JSON.stringify((0, import_smithy_client._json)(input));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_DescribeDatasetCommand");
var se_DescribeDatasetExportJobCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("DescribeDatasetExportJob");
  let body;
  body = JSON.stringify((0, import_smithy_client._json)(input));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_DescribeDatasetExportJobCommand");
var se_DescribeDatasetGroupCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("DescribeDatasetGroup");
  let body;
  body = JSON.stringify((0, import_smithy_client._json)(input));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_DescribeDatasetGroupCommand");
var se_DescribeDatasetImportJobCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("DescribeDatasetImportJob");
  let body;
  body = JSON.stringify((0, import_smithy_client._json)(input));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_DescribeDatasetImportJobCommand");
var se_DescribeEventTrackerCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("DescribeEventTracker");
  let body;
  body = JSON.stringify((0, import_smithy_client._json)(input));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_DescribeEventTrackerCommand");
var se_DescribeFeatureTransformationCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("DescribeFeatureTransformation");
  let body;
  body = JSON.stringify((0, import_smithy_client._json)(input));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_DescribeFeatureTransformationCommand");
var se_DescribeFilterCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("DescribeFilter");
  let body;
  body = JSON.stringify((0, import_smithy_client._json)(input));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_DescribeFilterCommand");
var se_DescribeMetricAttributionCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("DescribeMetricAttribution");
  let body;
  body = JSON.stringify((0, import_smithy_client._json)(input));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_DescribeMetricAttributionCommand");
var se_DescribeRecipeCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("DescribeRecipe");
  let body;
  body = JSON.stringify((0, import_smithy_client._json)(input));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_DescribeRecipeCommand");
var se_DescribeRecommenderCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("DescribeRecommender");
  let body;
  body = JSON.stringify((0, import_smithy_client._json)(input));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_DescribeRecommenderCommand");
var se_DescribeSchemaCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("DescribeSchema");
  let body;
  body = JSON.stringify((0, import_smithy_client._json)(input));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_DescribeSchemaCommand");
var se_DescribeSolutionCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("DescribeSolution");
  let body;
  body = JSON.stringify((0, import_smithy_client._json)(input));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_DescribeSolutionCommand");
var se_DescribeSolutionVersionCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("DescribeSolutionVersion");
  let body;
  body = JSON.stringify((0, import_smithy_client._json)(input));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_DescribeSolutionVersionCommand");
var se_GetSolutionMetricsCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("GetSolutionMetrics");
  let body;
  body = JSON.stringify((0, import_smithy_client._json)(input));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_GetSolutionMetricsCommand");
var se_ListBatchInferenceJobsCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("ListBatchInferenceJobs");
  let body;
  body = JSON.stringify((0, import_smithy_client._json)(input));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_ListBatchInferenceJobsCommand");
var se_ListBatchSegmentJobsCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("ListBatchSegmentJobs");
  let body;
  body = JSON.stringify((0, import_smithy_client._json)(input));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_ListBatchSegmentJobsCommand");
var se_ListCampaignsCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("ListCampaigns");
  let body;
  body = JSON.stringify((0, import_smithy_client._json)(input));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_ListCampaignsCommand");
var se_ListDatasetExportJobsCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("ListDatasetExportJobs");
  let body;
  body = JSON.stringify((0, import_smithy_client._json)(input));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_ListDatasetExportJobsCommand");
var se_ListDatasetGroupsCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("ListDatasetGroups");
  let body;
  body = JSON.stringify((0, import_smithy_client._json)(input));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_ListDatasetGroupsCommand");
var se_ListDatasetImportJobsCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("ListDatasetImportJobs");
  let body;
  body = JSON.stringify((0, import_smithy_client._json)(input));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_ListDatasetImportJobsCommand");
var se_ListDatasetsCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("ListDatasets");
  let body;
  body = JSON.stringify((0, import_smithy_client._json)(input));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_ListDatasetsCommand");
var se_ListEventTrackersCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("ListEventTrackers");
  let body;
  body = JSON.stringify((0, import_smithy_client._json)(input));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_ListEventTrackersCommand");
var se_ListFiltersCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("ListFilters");
  let body;
  body = JSON.stringify((0, import_smithy_client._json)(input));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_ListFiltersCommand");
var se_ListMetricAttributionMetricsCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("ListMetricAttributionMetrics");
  let body;
  body = JSON.stringify((0, import_smithy_client._json)(input));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_ListMetricAttributionMetricsCommand");
var se_ListMetricAttributionsCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("ListMetricAttributions");
  let body;
  body = JSON.stringify((0, import_smithy_client._json)(input));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_ListMetricAttributionsCommand");
var se_ListRecipesCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("ListRecipes");
  let body;
  body = JSON.stringify((0, import_smithy_client._json)(input));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_ListRecipesCommand");
var se_ListRecommendersCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("ListRecommenders");
  let body;
  body = JSON.stringify((0, import_smithy_client._json)(input));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_ListRecommendersCommand");
var se_ListSchemasCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("ListSchemas");
  let body;
  body = JSON.stringify((0, import_smithy_client._json)(input));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_ListSchemasCommand");
var se_ListSolutionsCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("ListSolutions");
  let body;
  body = JSON.stringify((0, import_smithy_client._json)(input));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_ListSolutionsCommand");
var se_ListSolutionVersionsCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("ListSolutionVersions");
  let body;
  body = JSON.stringify((0, import_smithy_client._json)(input));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_ListSolutionVersionsCommand");
var se_ListTagsForResourceCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("ListTagsForResource");
  let body;
  body = JSON.stringify((0, import_smithy_client._json)(input));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_ListTagsForResourceCommand");
var se_StartRecommenderCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("StartRecommender");
  let body;
  body = JSON.stringify((0, import_smithy_client._json)(input));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_StartRecommenderCommand");
var se_StopRecommenderCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("StopRecommender");
  let body;
  body = JSON.stringify((0, import_smithy_client._json)(input));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_StopRecommenderCommand");
var se_StopSolutionVersionCreationCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("StopSolutionVersionCreation");
  let body;
  body = JSON.stringify((0, import_smithy_client._json)(input));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_StopSolutionVersionCreationCommand");
var se_TagResourceCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("TagResource");
  let body;
  body = JSON.stringify((0, import_smithy_client._json)(input));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_TagResourceCommand");
var se_UntagResourceCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("UntagResource");
  let body;
  body = JSON.stringify((0, import_smithy_client._json)(input));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_UntagResourceCommand");
var se_UpdateCampaignCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("UpdateCampaign");
  let body;
  body = JSON.stringify((0, import_smithy_client._json)(input));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_UpdateCampaignCommand");
var se_UpdateDatasetCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("UpdateDataset");
  let body;
  body = JSON.stringify((0, import_smithy_client._json)(input));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_UpdateDatasetCommand");
var se_UpdateMetricAttributionCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("UpdateMetricAttribution");
  let body;
  body = JSON.stringify((0, import_smithy_client._json)(input));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_UpdateMetricAttributionCommand");
var se_UpdateRecommenderCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("UpdateRecommender");
  let body;
  body = JSON.stringify((0, import_smithy_client._json)(input));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_UpdateRecommenderCommand");
var de_CreateBatchInferenceJobCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const data = await (0, import_core2.parseJsonBody)(output.body, context);
  let contents = {};
  contents = (0, import_smithy_client._json)(data);
  const response = {
    $metadata: deserializeMetadata(output),
    ...contents
  };
  return response;
}, "de_CreateBatchInferenceJobCommand");
var de_CreateBatchSegmentJobCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const data = await (0, import_core2.parseJsonBody)(output.body, context);
  let contents = {};
  contents = (0, import_smithy_client._json)(data);
  const response = {
    $metadata: deserializeMetadata(output),
    ...contents
  };
  return response;
}, "de_CreateBatchSegmentJobCommand");
var de_CreateCampaignCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const data = await (0, import_core2.parseJsonBody)(output.body, context);
  let contents = {};
  contents = (0, import_smithy_client._json)(data);
  const response = {
    $metadata: deserializeMetadata(output),
    ...contents
  };
  return response;
}, "de_CreateCampaignCommand");
var de_CreateDatasetCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const data = await (0, import_core2.parseJsonBody)(output.body, context);
  let contents = {};
  contents = (0, import_smithy_client._json)(data);
  const response = {
    $metadata: deserializeMetadata(output),
    ...contents
  };
  return response;
}, "de_CreateDatasetCommand");
var de_CreateDatasetExportJobCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const data = await (0, import_core2.parseJsonBody)(output.body, context);
  let contents = {};
  contents = (0, import_smithy_client._json)(data);
  const response = {
    $metadata: deserializeMetadata(output),
    ...contents
  };
  return response;
}, "de_CreateDatasetExportJobCommand");
var de_CreateDatasetGroupCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const data = await (0, import_core2.parseJsonBody)(output.body, context);
  let contents = {};
  contents = (0, import_smithy_client._json)(data);
  const response = {
    $metadata: deserializeMetadata(output),
    ...contents
  };
  return response;
}, "de_CreateDatasetGroupCommand");
var de_CreateDatasetImportJobCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const data = await (0, import_core2.parseJsonBody)(output.body, context);
  let contents = {};
  contents = (0, import_smithy_client._json)(data);
  const response = {
    $metadata: deserializeMetadata(output),
    ...contents
  };
  return response;
}, "de_CreateDatasetImportJobCommand");
var de_CreateEventTrackerCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const data = await (0, import_core2.parseJsonBody)(output.body, context);
  let contents = {};
  contents = (0, import_smithy_client._json)(data);
  const response = {
    $metadata: deserializeMetadata(output),
    ...contents
  };
  return response;
}, "de_CreateEventTrackerCommand");
var de_CreateFilterCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const data = await (0, import_core2.parseJsonBody)(output.body, context);
  let contents = {};
  contents = (0, import_smithy_client._json)(data);
  const response = {
    $metadata: deserializeMetadata(output),
    ...contents
  };
  return response;
}, "de_CreateFilterCommand");
var de_CreateMetricAttributionCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const data = await (0, import_core2.parseJsonBody)(output.body, context);
  let contents = {};
  contents = (0, import_smithy_client._json)(data);
  const response = {
    $metadata: deserializeMetadata(output),
    ...contents
  };
  return response;
}, "de_CreateMetricAttributionCommand");
var de_CreateRecommenderCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const data = await (0, import_core2.parseJsonBody)(output.body, context);
  let contents = {};
  contents = (0, import_smithy_client._json)(data);
  const response = {
    $metadata: deserializeMetadata(output),
    ...contents
  };
  return response;
}, "de_CreateRecommenderCommand");
var de_CreateSchemaCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const data = await (0, import_core2.parseJsonBody)(output.body, context);
  let contents = {};
  contents = (0, import_smithy_client._json)(data);
  const response = {
    $metadata: deserializeMetadata(output),
    ...contents
  };
  return response;
}, "de_CreateSchemaCommand");
var de_CreateSolutionCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const data = await (0, import_core2.parseJsonBody)(output.body, context);
  let contents = {};
  contents = (0, import_smithy_client._json)(data);
  const response = {
    $metadata: deserializeMetadata(output),
    ...contents
  };
  return response;
}, "de_CreateSolutionCommand");
var de_CreateSolutionVersionCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const data = await (0, import_core2.parseJsonBody)(output.body, context);
  let contents = {};
  contents = (0, import_smithy_client._json)(data);
  const response = {
    $metadata: deserializeMetadata(output),
    ...contents
  };
  return response;
}, "de_CreateSolutionVersionCommand");
var de_DeleteCampaignCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  await (0, import_smithy_client.collectBody)(output.body, context);
  const response = {
    $metadata: deserializeMetadata(output)
  };
  return response;
}, "de_DeleteCampaignCommand");
var de_DeleteDatasetCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  await (0, import_smithy_client.collectBody)(output.body, context);
  const response = {
    $metadata: deserializeMetadata(output)
  };
  return response;
}, "de_DeleteDatasetCommand");
var de_DeleteDatasetGroupCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  await (0, import_smithy_client.collectBody)(output.body, context);
  const response = {
    $metadata: deserializeMetadata(output)
  };
  return response;
}, "de_DeleteDatasetGroupCommand");
var de_DeleteEventTrackerCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  await (0, import_smithy_client.collectBody)(output.body, context);
  const response = {
    $metadata: deserializeMetadata(output)
  };
  return response;
}, "de_DeleteEventTrackerCommand");
var de_DeleteFilterCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  await (0, import_smithy_client.collectBody)(output.body, context);
  const response = {
    $metadata: deserializeMetadata(output)
  };
  return response;
}, "de_DeleteFilterCommand");
var de_DeleteMetricAttributionCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  await (0, import_smithy_client.collectBody)(output.body, context);
  const response = {
    $metadata: deserializeMetadata(output)
  };
  return response;
}, "de_DeleteMetricAttributionCommand");
var de_DeleteRecommenderCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  await (0, import_smithy_client.collectBody)(output.body, context);
  const response = {
    $metadata: deserializeMetadata(output)
  };
  return response;
}, "de_DeleteRecommenderCommand");
var de_DeleteSchemaCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  await (0, import_smithy_client.collectBody)(output.body, context);
  const response = {
    $metadata: deserializeMetadata(output)
  };
  return response;
}, "de_DeleteSchemaCommand");
var de_DeleteSolutionCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  await (0, import_smithy_client.collectBody)(output.body, context);
  const response = {
    $metadata: deserializeMetadata(output)
  };
  return response;
}, "de_DeleteSolutionCommand");
var de_DescribeAlgorithmCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const data = await (0, import_core2.parseJsonBody)(output.body, context);
  let contents = {};
  contents = de_DescribeAlgorithmResponse(data, context);
  const response = {
    $metadata: deserializeMetadata(output),
    ...contents
  };
  return response;
}, "de_DescribeAlgorithmCommand");
var de_DescribeBatchInferenceJobCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const data = await (0, import_core2.parseJsonBody)(output.body, context);
  let contents = {};
  contents = de_DescribeBatchInferenceJobResponse(data, context);
  const response = {
    $metadata: deserializeMetadata(output),
    ...contents
  };
  return response;
}, "de_DescribeBatchInferenceJobCommand");
var de_DescribeBatchSegmentJobCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const data = await (0, import_core2.parseJsonBody)(output.body, context);
  let contents = {};
  contents = de_DescribeBatchSegmentJobResponse(data, context);
  const response = {
    $metadata: deserializeMetadata(output),
    ...contents
  };
  return response;
}, "de_DescribeBatchSegmentJobCommand");
var de_DescribeCampaignCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const data = await (0, import_core2.parseJsonBody)(output.body, context);
  let contents = {};
  contents = de_DescribeCampaignResponse(data, context);
  const response = {
    $metadata: deserializeMetadata(output),
    ...contents
  };
  return response;
}, "de_DescribeCampaignCommand");
var de_DescribeDatasetCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const data = await (0, import_core2.parseJsonBody)(output.body, context);
  let contents = {};
  contents = de_DescribeDatasetResponse(data, context);
  const response = {
    $metadata: deserializeMetadata(output),
    ...contents
  };
  return response;
}, "de_DescribeDatasetCommand");
var de_DescribeDatasetExportJobCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const data = await (0, import_core2.parseJsonBody)(output.body, context);
  let contents = {};
  contents = de_DescribeDatasetExportJobResponse(data, context);
  const response = {
    $metadata: deserializeMetadata(output),
    ...contents
  };
  return response;
}, "de_DescribeDatasetExportJobCommand");
var de_DescribeDatasetGroupCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const data = await (0, import_core2.parseJsonBody)(output.body, context);
  let contents = {};
  contents = de_DescribeDatasetGroupResponse(data, context);
  const response = {
    $metadata: deserializeMetadata(output),
    ...contents
  };
  return response;
}, "de_DescribeDatasetGroupCommand");
var de_DescribeDatasetImportJobCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const data = await (0, import_core2.parseJsonBody)(output.body, context);
  let contents = {};
  contents = de_DescribeDatasetImportJobResponse(data, context);
  const response = {
    $metadata: deserializeMetadata(output),
    ...contents
  };
  return response;
}, "de_DescribeDatasetImportJobCommand");
var de_DescribeEventTrackerCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const data = await (0, import_core2.parseJsonBody)(output.body, context);
  let contents = {};
  contents = de_DescribeEventTrackerResponse(data, context);
  const response = {
    $metadata: deserializeMetadata(output),
    ...contents
  };
  return response;
}, "de_DescribeEventTrackerCommand");
var de_DescribeFeatureTransformationCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const data = await (0, import_core2.parseJsonBody)(output.body, context);
  let contents = {};
  contents = de_DescribeFeatureTransformationResponse(data, context);
  const response = {
    $metadata: deserializeMetadata(output),
    ...contents
  };
  return response;
}, "de_DescribeFeatureTransformationCommand");
var de_DescribeFilterCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const data = await (0, import_core2.parseJsonBody)(output.body, context);
  let contents = {};
  contents = de_DescribeFilterResponse(data, context);
  const response = {
    $metadata: deserializeMetadata(output),
    ...contents
  };
  return response;
}, "de_DescribeFilterCommand");
var de_DescribeMetricAttributionCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const data = await (0, import_core2.parseJsonBody)(output.body, context);
  let contents = {};
  contents = de_DescribeMetricAttributionResponse(data, context);
  const response = {
    $metadata: deserializeMetadata(output),
    ...contents
  };
  return response;
}, "de_DescribeMetricAttributionCommand");
var de_DescribeRecipeCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const data = await (0, import_core2.parseJsonBody)(output.body, context);
  let contents = {};
  contents = de_DescribeRecipeResponse(data, context);
  const response = {
    $metadata: deserializeMetadata(output),
    ...contents
  };
  return response;
}, "de_DescribeRecipeCommand");
var de_DescribeRecommenderCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const data = await (0, import_core2.parseJsonBody)(output.body, context);
  let contents = {};
  contents = de_DescribeRecommenderResponse(data, context);
  const response = {
    $metadata: deserializeMetadata(output),
    ...contents
  };
  return response;
}, "de_DescribeRecommenderCommand");
var de_DescribeSchemaCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const data = await (0, import_core2.parseJsonBody)(output.body, context);
  let contents = {};
  contents = de_DescribeSchemaResponse(data, context);
  const response = {
    $metadata: deserializeMetadata(output),
    ...contents
  };
  return response;
}, "de_DescribeSchemaCommand");
var de_DescribeSolutionCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const data = await (0, import_core2.parseJsonBody)(output.body, context);
  let contents = {};
  contents = de_DescribeSolutionResponse(data, context);
  const response = {
    $metadata: deserializeMetadata(output),
    ...contents
  };
  return response;
}, "de_DescribeSolutionCommand");
var de_DescribeSolutionVersionCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const data = await (0, import_core2.parseJsonBody)(output.body, context);
  let contents = {};
  contents = de_DescribeSolutionVersionResponse(data, context);
  const response = {
    $metadata: deserializeMetadata(output),
    ...contents
  };
  return response;
}, "de_DescribeSolutionVersionCommand");
var de_GetSolutionMetricsCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const data = await (0, import_core2.parseJsonBody)(output.body, context);
  let contents = {};
  contents = de_GetSolutionMetricsResponse(data, context);
  const response = {
    $metadata: deserializeMetadata(output),
    ...contents
  };
  return response;
}, "de_GetSolutionMetricsCommand");
var de_ListBatchInferenceJobsCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const data = await (0, import_core2.parseJsonBody)(output.body, context);
  let contents = {};
  contents = de_ListBatchInferenceJobsResponse(data, context);
  const response = {
    $metadata: deserializeMetadata(output),
    ...contents
  };
  return response;
}, "de_ListBatchInferenceJobsCommand");
var de_ListBatchSegmentJobsCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const data = await (0, import_core2.parseJsonBody)(output.body, context);
  let contents = {};
  contents = de_ListBatchSegmentJobsResponse(data, context);
  const response = {
    $metadata: deserializeMetadata(output),
    ...contents
  };
  return response;
}, "de_ListBatchSegmentJobsCommand");
var de_ListCampaignsCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const data = await (0, import_core2.parseJsonBody)(output.body, context);
  let contents = {};
  contents = de_ListCampaignsResponse(data, context);
  const response = {
    $metadata: deserializeMetadata(output),
    ...contents
  };
  return response;
}, "de_ListCampaignsCommand");
var de_ListDatasetExportJobsCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const data = await (0, import_core2.parseJsonBody)(output.body, context);
  let contents = {};
  contents = de_ListDatasetExportJobsResponse(data, context);
  const response = {
    $metadata: deserializeMetadata(output),
    ...contents
  };
  return response;
}, "de_ListDatasetExportJobsCommand");
var de_ListDatasetGroupsCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const data = await (0, import_core2.parseJsonBody)(output.body, context);
  let contents = {};
  contents = de_ListDatasetGroupsResponse(data, context);
  const response = {
    $metadata: deserializeMetadata(output),
    ...contents
  };
  return response;
}, "de_ListDatasetGroupsCommand");
var de_ListDatasetImportJobsCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const data = await (0, import_core2.parseJsonBody)(output.body, context);
  let contents = {};
  contents = de_ListDatasetImportJobsResponse(data, context);
  const response = {
    $metadata: deserializeMetadata(output),
    ...contents
  };
  return response;
}, "de_ListDatasetImportJobsCommand");
var de_ListDatasetsCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const data = await (0, import_core2.parseJsonBody)(output.body, context);
  let contents = {};
  contents = de_ListDatasetsResponse(data, context);
  const response = {
    $metadata: deserializeMetadata(output),
    ...contents
  };
  return response;
}, "de_ListDatasetsCommand");
var de_ListEventTrackersCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const data = await (0, import_core2.parseJsonBody)(output.body, context);
  let contents = {};
  contents = de_ListEventTrackersResponse(data, context);
  const response = {
    $metadata: deserializeMetadata(output),
    ...contents
  };
  return response;
}, "de_ListEventTrackersCommand");
var de_ListFiltersCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const data = await (0, import_core2.parseJsonBody)(output.body, context);
  let contents = {};
  contents = de_ListFiltersResponse(data, context);
  const response = {
    $metadata: deserializeMetadata(output),
    ...contents
  };
  return response;
}, "de_ListFiltersCommand");
var de_ListMetricAttributionMetricsCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const data = await (0, import_core2.parseJsonBody)(output.body, context);
  let contents = {};
  contents = (0, import_smithy_client._json)(data);
  const response = {
    $metadata: deserializeMetadata(output),
    ...contents
  };
  return response;
}, "de_ListMetricAttributionMetricsCommand");
var de_ListMetricAttributionsCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const data = await (0, import_core2.parseJsonBody)(output.body, context);
  let contents = {};
  contents = de_ListMetricAttributionsResponse(data, context);
  const response = {
    $metadata: deserializeMetadata(output),
    ...contents
  };
  return response;
}, "de_ListMetricAttributionsCommand");
var de_ListRecipesCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const data = await (0, import_core2.parseJsonBody)(output.body, context);
  let contents = {};
  contents = de_ListRecipesResponse(data, context);
  const response = {
    $metadata: deserializeMetadata(output),
    ...contents
  };
  return response;
}, "de_ListRecipesCommand");
var de_ListRecommendersCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const data = await (0, import_core2.parseJsonBody)(output.body, context);
  let contents = {};
  contents = de_ListRecommendersResponse(data, context);
  const response = {
    $metadata: deserializeMetadata(output),
    ...contents
  };
  return response;
}, "de_ListRecommendersCommand");
var de_ListSchemasCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const data = await (0, import_core2.parseJsonBody)(output.body, context);
  let contents = {};
  contents = de_ListSchemasResponse(data, context);
  const response = {
    $metadata: deserializeMetadata(output),
    ...contents
  };
  return response;
}, "de_ListSchemasCommand");
var de_ListSolutionsCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const data = await (0, import_core2.parseJsonBody)(output.body, context);
  let contents = {};
  contents = de_ListSolutionsResponse(data, context);
  const response = {
    $metadata: deserializeMetadata(output),
    ...contents
  };
  return response;
}, "de_ListSolutionsCommand");
var de_ListSolutionVersionsCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const data = await (0, import_core2.parseJsonBody)(output.body, context);
  let contents = {};
  contents = de_ListSolutionVersionsResponse(data, context);
  const response = {
    $metadata: deserializeMetadata(output),
    ...contents
  };
  return response;
}, "de_ListSolutionVersionsCommand");
var de_ListTagsForResourceCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const data = await (0, import_core2.parseJsonBody)(output.body, context);
  let contents = {};
  contents = (0, import_smithy_client._json)(data);
  const response = {
    $metadata: deserializeMetadata(output),
    ...contents
  };
  return response;
}, "de_ListTagsForResourceCommand");
var de_StartRecommenderCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const data = await (0, import_core2.parseJsonBody)(output.body, context);
  let contents = {};
  contents = (0, import_smithy_client._json)(data);
  const response = {
    $metadata: deserializeMetadata(output),
    ...contents
  };
  return response;
}, "de_StartRecommenderCommand");
var de_StopRecommenderCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const data = await (0, import_core2.parseJsonBody)(output.body, context);
  let contents = {};
  contents = (0, import_smithy_client._json)(data);
  const response = {
    $metadata: deserializeMetadata(output),
    ...contents
  };
  return response;
}, "de_StopRecommenderCommand");
var de_StopSolutionVersionCreationCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  await (0, import_smithy_client.collectBody)(output.body, context);
  const response = {
    $metadata: deserializeMetadata(output)
  };
  return response;
}, "de_StopSolutionVersionCreationCommand");
var de_TagResourceCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const data = await (0, import_core2.parseJsonBody)(output.body, context);
  let contents = {};
  contents = (0, import_smithy_client._json)(data);
  const response = {
    $metadata: deserializeMetadata(output),
    ...contents
  };
  return response;
}, "de_TagResourceCommand");
var de_UntagResourceCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const data = await (0, import_core2.parseJsonBody)(output.body, context);
  let contents = {};
  contents = (0, import_smithy_client._json)(data);
  const response = {
    $metadata: deserializeMetadata(output),
    ...contents
  };
  return response;
}, "de_UntagResourceCommand");
var de_UpdateCampaignCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const data = await (0, import_core2.parseJsonBody)(output.body, context);
  let contents = {};
  contents = (0, import_smithy_client._json)(data);
  const response = {
    $metadata: deserializeMetadata(output),
    ...contents
  };
  return response;
}, "de_UpdateCampaignCommand");
var de_UpdateDatasetCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const data = await (0, import_core2.parseJsonBody)(output.body, context);
  let contents = {};
  contents = (0, import_smithy_client._json)(data);
  const response = {
    $metadata: deserializeMetadata(output),
    ...contents
  };
  return response;
}, "de_UpdateDatasetCommand");
var de_UpdateMetricAttributionCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const data = await (0, import_core2.parseJsonBody)(output.body, context);
  let contents = {};
  contents = (0, import_smithy_client._json)(data);
  const response = {
    $metadata: deserializeMetadata(output),
    ...contents
  };
  return response;
}, "de_UpdateMetricAttributionCommand");
var de_UpdateRecommenderCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const data = await (0, import_core2.parseJsonBody)(output.body, context);
  let contents = {};
  contents = (0, import_smithy_client._json)(data);
  const response = {
    $metadata: deserializeMetadata(output),
    ...contents
  };
  return response;
}, "de_UpdateRecommenderCommand");
var de_CommandError = /* @__PURE__ */ __name(async (output, context) => {
  const parsedOutput = {
    ...output,
    body: await (0, import_core2.parseJsonErrorBody)(output.body, context)
  };
  const errorCode = (0, import_core2.loadRestJsonErrorCode)(output, parsedOutput.body);
  switch (errorCode) {
    case "InvalidInputException":
    case "com.amazonaws.personalize#InvalidInputException":
      throw await de_InvalidInputExceptionRes(parsedOutput, context);
    case "LimitExceededException":
    case "com.amazonaws.personalize#LimitExceededException":
      throw await de_LimitExceededExceptionRes(parsedOutput, context);
    case "ResourceAlreadyExistsException":
    case "com.amazonaws.personalize#ResourceAlreadyExistsException":
      throw await de_ResourceAlreadyExistsExceptionRes(parsedOutput, context);
    case "ResourceInUseException":
    case "com.amazonaws.personalize#ResourceInUseException":
      throw await de_ResourceInUseExceptionRes(parsedOutput, context);
    case "ResourceNotFoundException":
    case "com.amazonaws.personalize#ResourceNotFoundException":
      throw await de_ResourceNotFoundExceptionRes(parsedOutput, context);
    case "TooManyTagsException":
    case "com.amazonaws.personalize#TooManyTagsException":
      throw await de_TooManyTagsExceptionRes(parsedOutput, context);
    case "InvalidNextTokenException":
    case "com.amazonaws.personalize#InvalidNextTokenException":
      throw await de_InvalidNextTokenExceptionRes(parsedOutput, context);
    case "TooManyTagKeysException":
    case "com.amazonaws.personalize#TooManyTagKeysException":
      throw await de_TooManyTagKeysExceptionRes(parsedOutput, context);
    default:
      const parsedBody = parsedOutput.body;
      return throwDefaultError({
        output,
        parsedBody,
        errorCode
      });
  }
}, "de_CommandError");
var de_InvalidInputExceptionRes = /* @__PURE__ */ __name(async (parsedOutput, context) => {
  const body = parsedOutput.body;
  const deserialized = (0, import_smithy_client._json)(body);
  const exception = new InvalidInputException({
    $metadata: deserializeMetadata(parsedOutput),
    ...deserialized
  });
  return (0, import_smithy_client.decorateServiceException)(exception, body);
}, "de_InvalidInputExceptionRes");
var de_InvalidNextTokenExceptionRes = /* @__PURE__ */ __name(async (parsedOutput, context) => {
  const body = parsedOutput.body;
  const deserialized = (0, import_smithy_client._json)(body);
  const exception = new InvalidNextTokenException({
    $metadata: deserializeMetadata(parsedOutput),
    ...deserialized
  });
  return (0, import_smithy_client.decorateServiceException)(exception, body);
}, "de_InvalidNextTokenExceptionRes");
var de_LimitExceededExceptionRes = /* @__PURE__ */ __name(async (parsedOutput, context) => {
  const body = parsedOutput.body;
  const deserialized = (0, import_smithy_client._json)(body);
  const exception = new LimitExceededException({
    $metadata: deserializeMetadata(parsedOutput),
    ...deserialized
  });
  return (0, import_smithy_client.decorateServiceException)(exception, body);
}, "de_LimitExceededExceptionRes");
var de_ResourceAlreadyExistsExceptionRes = /* @__PURE__ */ __name(async (parsedOutput, context) => {
  const body = parsedOutput.body;
  const deserialized = (0, import_smithy_client._json)(body);
  const exception = new ResourceAlreadyExistsException({
    $metadata: deserializeMetadata(parsedOutput),
    ...deserialized
  });
  return (0, import_smithy_client.decorateServiceException)(exception, body);
}, "de_ResourceAlreadyExistsExceptionRes");
var de_ResourceInUseExceptionRes = /* @__PURE__ */ __name(async (parsedOutput, context) => {
  const body = parsedOutput.body;
  const deserialized = (0, import_smithy_client._json)(body);
  const exception = new ResourceInUseException({
    $metadata: deserializeMetadata(parsedOutput),
    ...deserialized
  });
  return (0, import_smithy_client.decorateServiceException)(exception, body);
}, "de_ResourceInUseExceptionRes");
var de_ResourceNotFoundExceptionRes = /* @__PURE__ */ __name(async (parsedOutput, context) => {
  const body = parsedOutput.body;
  const deserialized = (0, import_smithy_client._json)(body);
  const exception = new ResourceNotFoundException({
    $metadata: deserializeMetadata(parsedOutput),
    ...deserialized
  });
  return (0, import_smithy_client.decorateServiceException)(exception, body);
}, "de_ResourceNotFoundExceptionRes");
var de_TooManyTagKeysExceptionRes = /* @__PURE__ */ __name(async (parsedOutput, context) => {
  const body = parsedOutput.body;
  const deserialized = (0, import_smithy_client._json)(body);
  const exception = new TooManyTagKeysException({
    $metadata: deserializeMetadata(parsedOutput),
    ...deserialized
  });
  return (0, import_smithy_client.decorateServiceException)(exception, body);
}, "de_TooManyTagKeysExceptionRes");
var de_TooManyTagsExceptionRes = /* @__PURE__ */ __name(async (parsedOutput, context) => {
  const body = parsedOutput.body;
  const deserialized = (0, import_smithy_client._json)(body);
  const exception = new TooManyTagsException({
    $metadata: deserializeMetadata(parsedOutput),
    ...deserialized
  });
  return (0, import_smithy_client.decorateServiceException)(exception, body);
}, "de_TooManyTagsExceptionRes");
var se_ContinuousHyperParameterRange = /* @__PURE__ */ __name((input, context) => {
  return (0, import_smithy_client.take)(input, {
    maxValue: import_smithy_client.serializeFloat,
    minValue: import_smithy_client.serializeFloat,
    name: []
  });
}, "se_ContinuousHyperParameterRange");
var se_ContinuousHyperParameterRanges = /* @__PURE__ */ __name((input, context) => {
  return input.filter((e) => e != null).map((entry) => {
    return se_ContinuousHyperParameterRange(entry, context);
  });
}, "se_ContinuousHyperParameterRanges");
var se_CreateSolutionRequest = /* @__PURE__ */ __name((input, context) => {
  return (0, import_smithy_client.take)(input, {
    datasetGroupArn: [],
    eventType: [],
    name: [],
    performAutoML: [],
    performAutoTraining: [],
    performHPO: [],
    recipeArn: [],
    solutionConfig: (_) => se_SolutionConfig(_, context),
    tags: import_smithy_client._json
  });
}, "se_CreateSolutionRequest");
var se_HPOConfig = /* @__PURE__ */ __name((input, context) => {
  return (0, import_smithy_client.take)(input, {
    algorithmHyperParameterRanges: (_) => se_HyperParameterRanges(_, context),
    hpoObjective: import_smithy_client._json,
    hpoResourceConfig: import_smithy_client._json
  });
}, "se_HPOConfig");
var se_HyperParameterRanges = /* @__PURE__ */ __name((input, context) => {
  return (0, import_smithy_client.take)(input, {
    categoricalHyperParameterRanges: import_smithy_client._json,
    continuousHyperParameterRanges: (_) => se_ContinuousHyperParameterRanges(_, context),
    integerHyperParameterRanges: import_smithy_client._json
  });
}, "se_HyperParameterRanges");
var se_SolutionConfig = /* @__PURE__ */ __name((input, context) => {
  return (0, import_smithy_client.take)(input, {
    algorithmHyperParameters: import_smithy_client._json,
    autoMLConfig: import_smithy_client._json,
    autoTrainingConfig: import_smithy_client._json,
    eventValueThreshold: [],
    featureTransformationParameters: import_smithy_client._json,
    hpoConfig: (_) => se_HPOConfig(_, context),
    optimizationObjective: import_smithy_client._json,
    trainingDataConfig: import_smithy_client._json
  });
}, "se_SolutionConfig");
var de_Algorithm = /* @__PURE__ */ __name((output, context) => {
  return (0, import_smithy_client.take)(output, {
    algorithmArn: import_smithy_client.expectString,
    algorithmImage: import_smithy_client._json,
    creationDateTime: (_) => (0, import_smithy_client.expectNonNull)((0, import_smithy_client.parseEpochTimestamp)((0, import_smithy_client.expectNumber)(_))),
    defaultHyperParameterRanges: (_) => de_DefaultHyperParameterRanges(_, context),
    defaultHyperParameters: import_smithy_client._json,
    defaultResourceConfig: import_smithy_client._json,
    lastUpdatedDateTime: (_) => (0, import_smithy_client.expectNonNull)((0, import_smithy_client.parseEpochTimestamp)((0, import_smithy_client.expectNumber)(_))),
    name: import_smithy_client.expectString,
    roleArn: import_smithy_client.expectString,
    trainingInputMode: import_smithy_client.expectString
  });
}, "de_Algorithm");
var de_BatchInferenceJob = /* @__PURE__ */ __name((output, context) => {
  return (0, import_smithy_client.take)(output, {
    batchInferenceJobArn: import_smithy_client.expectString,
    batchInferenceJobConfig: import_smithy_client._json,
    batchInferenceJobMode: import_smithy_client.expectString,
    creationDateTime: (_) => (0, import_smithy_client.expectNonNull)((0, import_smithy_client.parseEpochTimestamp)((0, import_smithy_client.expectNumber)(_))),
    failureReason: import_smithy_client.expectString,
    filterArn: import_smithy_client.expectString,
    jobInput: import_smithy_client._json,
    jobName: import_smithy_client.expectString,
    jobOutput: import_smithy_client._json,
    lastUpdatedDateTime: (_) => (0, import_smithy_client.expectNonNull)((0, import_smithy_client.parseEpochTimestamp)((0, import_smithy_client.expectNumber)(_))),
    numResults: import_smithy_client.expectInt32,
    roleArn: import_smithy_client.expectString,
    solutionVersionArn: import_smithy_client.expectString,
    status: import_smithy_client.expectString,
    themeGenerationConfig: import_smithy_client._json
  });
}, "de_BatchInferenceJob");
var de_BatchInferenceJobs = /* @__PURE__ */ __name((output, context) => {
  const retVal = (output || []).filter((e) => e != null).map((entry) => {
    return de_BatchInferenceJobSummary(entry, context);
  });
  return retVal;
}, "de_BatchInferenceJobs");
var de_BatchInferenceJobSummary = /* @__PURE__ */ __name((output, context) => {
  return (0, import_smithy_client.take)(output, {
    batchInferenceJobArn: import_smithy_client.expectString,
    batchInferenceJobMode: import_smithy_client.expectString,
    creationDateTime: (_) => (0, import_smithy_client.expectNonNull)((0, import_smithy_client.parseEpochTimestamp)((0, import_smithy_client.expectNumber)(_))),
    failureReason: import_smithy_client.expectString,
    jobName: import_smithy_client.expectString,
    lastUpdatedDateTime: (_) => (0, import_smithy_client.expectNonNull)((0, import_smithy_client.parseEpochTimestamp)((0, import_smithy_client.expectNumber)(_))),
    solutionVersionArn: import_smithy_client.expectString,
    status: import_smithy_client.expectString
  });
}, "de_BatchInferenceJobSummary");
var de_BatchSegmentJob = /* @__PURE__ */ __name((output, context) => {
  return (0, import_smithy_client.take)(output, {
    batchSegmentJobArn: import_smithy_client.expectString,
    creationDateTime: (_) => (0, import_smithy_client.expectNonNull)((0, import_smithy_client.parseEpochTimestamp)((0, import_smithy_client.expectNumber)(_))),
    failureReason: import_smithy_client.expectString,
    filterArn: import_smithy_client.expectString,
    jobInput: import_smithy_client._json,
    jobName: import_smithy_client.expectString,
    jobOutput: import_smithy_client._json,
    lastUpdatedDateTime: (_) => (0, import_smithy_client.expectNonNull)((0, import_smithy_client.parseEpochTimestamp)((0, import_smithy_client.expectNumber)(_))),
    numResults: import_smithy_client.expectInt32,
    roleArn: import_smithy_client.expectString,
    solutionVersionArn: import_smithy_client.expectString,
    status: import_smithy_client.expectString
  });
}, "de_BatchSegmentJob");
var de_BatchSegmentJobs = /* @__PURE__ */ __name((output, context) => {
  const retVal = (output || []).filter((e) => e != null).map((entry) => {
    return de_BatchSegmentJobSummary(entry, context);
  });
  return retVal;
}, "de_BatchSegmentJobs");
var de_BatchSegmentJobSummary = /* @__PURE__ */ __name((output, context) => {
  return (0, import_smithy_client.take)(output, {
    batchSegmentJobArn: import_smithy_client.expectString,
    creationDateTime: (_) => (0, import_smithy_client.expectNonNull)((0, import_smithy_client.parseEpochTimestamp)((0, import_smithy_client.expectNumber)(_))),
    failureReason: import_smithy_client.expectString,
    jobName: import_smithy_client.expectString,
    lastUpdatedDateTime: (_) => (0, import_smithy_client.expectNonNull)((0, import_smithy_client.parseEpochTimestamp)((0, import_smithy_client.expectNumber)(_))),
    solutionVersionArn: import_smithy_client.expectString,
    status: import_smithy_client.expectString
  });
}, "de_BatchSegmentJobSummary");
var de_Campaign = /* @__PURE__ */ __name((output, context) => {
  return (0, import_smithy_client.take)(output, {
    campaignArn: import_smithy_client.expectString,
    campaignConfig: import_smithy_client._json,
    creationDateTime: (_) => (0, import_smithy_client.expectNonNull)((0, import_smithy_client.parseEpochTimestamp)((0, import_smithy_client.expectNumber)(_))),
    failureReason: import_smithy_client.expectString,
    lastUpdatedDateTime: (_) => (0, import_smithy_client.expectNonNull)((0, import_smithy_client.parseEpochTimestamp)((0, import_smithy_client.expectNumber)(_))),
    latestCampaignUpdate: (_) => de_CampaignUpdateSummary(_, context),
    minProvisionedTPS: import_smithy_client.expectInt32,
    name: import_smithy_client.expectString,
    solutionVersionArn: import_smithy_client.expectString,
    status: import_smithy_client.expectString
  });
}, "de_Campaign");
var de_Campaigns = /* @__PURE__ */ __name((output, context) => {
  const retVal = (output || []).filter((e) => e != null).map((entry) => {
    return de_CampaignSummary(entry, context);
  });
  return retVal;
}, "de_Campaigns");
var de_CampaignSummary = /* @__PURE__ */ __name((output, context) => {
  return (0, import_smithy_client.take)(output, {
    campaignArn: import_smithy_client.expectString,
    creationDateTime: (_) => (0, import_smithy_client.expectNonNull)((0, import_smithy_client.parseEpochTimestamp)((0, import_smithy_client.expectNumber)(_))),
    failureReason: import_smithy_client.expectString,
    lastUpdatedDateTime: (_) => (0, import_smithy_client.expectNonNull)((0, import_smithy_client.parseEpochTimestamp)((0, import_smithy_client.expectNumber)(_))),
    name: import_smithy_client.expectString,
    status: import_smithy_client.expectString
  });
}, "de_CampaignSummary");
var de_CampaignUpdateSummary = /* @__PURE__ */ __name((output, context) => {
  return (0, import_smithy_client.take)(output, {
    campaignConfig: import_smithy_client._json,
    creationDateTime: (_) => (0, import_smithy_client.expectNonNull)((0, import_smithy_client.parseEpochTimestamp)((0, import_smithy_client.expectNumber)(_))),
    failureReason: import_smithy_client.expectString,
    lastUpdatedDateTime: (_) => (0, import_smithy_client.expectNonNull)((0, import_smithy_client.parseEpochTimestamp)((0, import_smithy_client.expectNumber)(_))),
    minProvisionedTPS: import_smithy_client.expectInt32,
    solutionVersionArn: import_smithy_client.expectString,
    status: import_smithy_client.expectString
  });
}, "de_CampaignUpdateSummary");
var de_ContinuousHyperParameterRange = /* @__PURE__ */ __name((output, context) => {
  return (0, import_smithy_client.take)(output, {
    maxValue: import_smithy_client.limitedParseDouble,
    minValue: import_smithy_client.limitedParseDouble,
    name: import_smithy_client.expectString
  });
}, "de_ContinuousHyperParameterRange");
var de_ContinuousHyperParameterRanges = /* @__PURE__ */ __name((output, context) => {
  const retVal = (output || []).filter((e) => e != null).map((entry) => {
    return de_ContinuousHyperParameterRange(entry, context);
  });
  return retVal;
}, "de_ContinuousHyperParameterRanges");
var de_Dataset = /* @__PURE__ */ __name((output, context) => {
  return (0, import_smithy_client.take)(output, {
    creationDateTime: (_) => (0, import_smithy_client.expectNonNull)((0, import_smithy_client.parseEpochTimestamp)((0, import_smithy_client.expectNumber)(_))),
    datasetArn: import_smithy_client.expectString,
    datasetGroupArn: import_smithy_client.expectString,
    datasetType: import_smithy_client.expectString,
    lastUpdatedDateTime: (_) => (0, import_smithy_client.expectNonNull)((0, import_smithy_client.parseEpochTimestamp)((0, import_smithy_client.expectNumber)(_))),
    latestDatasetUpdate: (_) => de_DatasetUpdateSummary(_, context),
    name: import_smithy_client.expectString,
    schemaArn: import_smithy_client.expectString,
    status: import_smithy_client.expectString,
    trackingId: import_smithy_client.expectString
  });
}, "de_Dataset");
var de_DatasetExportJob = /* @__PURE__ */ __name((output, context) => {
  return (0, import_smithy_client.take)(output, {
    creationDateTime: (_) => (0, import_smithy_client.expectNonNull)((0, import_smithy_client.parseEpochTimestamp)((0, import_smithy_client.expectNumber)(_))),
    datasetArn: import_smithy_client.expectString,
    datasetExportJobArn: import_smithy_client.expectString,
    failureReason: import_smithy_client.expectString,
    ingestionMode: import_smithy_client.expectString,
    jobName: import_smithy_client.expectString,
    jobOutput: import_smithy_client._json,
    lastUpdatedDateTime: (_) => (0, import_smithy_client.expectNonNull)((0, import_smithy_client.parseEpochTimestamp)((0, import_smithy_client.expectNumber)(_))),
    roleArn: import_smithy_client.expectString,
    status: import_smithy_client.expectString
  });
}, "de_DatasetExportJob");
var de_DatasetExportJobs = /* @__PURE__ */ __name((output, context) => {
  const retVal = (output || []).filter((e) => e != null).map((entry) => {
    return de_DatasetExportJobSummary(entry, context);
  });
  return retVal;
}, "de_DatasetExportJobs");
var de_DatasetExportJobSummary = /* @__PURE__ */ __name((output, context) => {
  return (0, import_smithy_client.take)(output, {
    creationDateTime: (_) => (0, import_smithy_client.expectNonNull)((0, import_smithy_client.parseEpochTimestamp)((0, import_smithy_client.expectNumber)(_))),
    datasetExportJobArn: import_smithy_client.expectString,
    failureReason: import_smithy_client.expectString,
    jobName: import_smithy_client.expectString,
    lastUpdatedDateTime: (_) => (0, import_smithy_client.expectNonNull)((0, import_smithy_client.parseEpochTimestamp)((0, import_smithy_client.expectNumber)(_))),
    status: import_smithy_client.expectString
  });
}, "de_DatasetExportJobSummary");
var de_DatasetGroup = /* @__PURE__ */ __name((output, context) => {
  return (0, import_smithy_client.take)(output, {
    creationDateTime: (_) => (0, import_smithy_client.expectNonNull)((0, import_smithy_client.parseEpochTimestamp)((0, import_smithy_client.expectNumber)(_))),
    datasetGroupArn: import_smithy_client.expectString,
    domain: import_smithy_client.expectString,
    failureReason: import_smithy_client.expectString,
    kmsKeyArn: import_smithy_client.expectString,
    lastUpdatedDateTime: (_) => (0, import_smithy_client.expectNonNull)((0, import_smithy_client.parseEpochTimestamp)((0, import_smithy_client.expectNumber)(_))),
    name: import_smithy_client.expectString,
    roleArn: import_smithy_client.expectString,
    status: import_smithy_client.expectString
  });
}, "de_DatasetGroup");
var de_DatasetGroups = /* @__PURE__ */ __name((output, context) => {
  const retVal = (output || []).filter((e) => e != null).map((entry) => {
    return de_DatasetGroupSummary(entry, context);
  });
  return retVal;
}, "de_DatasetGroups");
var de_DatasetGroupSummary = /* @__PURE__ */ __name((output, context) => {
  return (0, import_smithy_client.take)(output, {
    creationDateTime: (_) => (0, import_smithy_client.expectNonNull)((0, import_smithy_client.parseEpochTimestamp)((0, import_smithy_client.expectNumber)(_))),
    datasetGroupArn: import_smithy_client.expectString,
    domain: import_smithy_client.expectString,
    failureReason: import_smithy_client.expectString,
    lastUpdatedDateTime: (_) => (0, import_smithy_client.expectNonNull)((0, import_smithy_client.parseEpochTimestamp)((0, import_smithy_client.expectNumber)(_))),
    name: import_smithy_client.expectString,
    status: import_smithy_client.expectString
  });
}, "de_DatasetGroupSummary");
var de_DatasetImportJob = /* @__PURE__ */ __name((output, context) => {
  return (0, import_smithy_client.take)(output, {
    creationDateTime: (_) => (0, import_smithy_client.expectNonNull)((0, import_smithy_client.parseEpochTimestamp)((0, import_smithy_client.expectNumber)(_))),
    dataSource: import_smithy_client._json,
    datasetArn: import_smithy_client.expectString,
    datasetImportJobArn: import_smithy_client.expectString,
    failureReason: import_smithy_client.expectString,
    importMode: import_smithy_client.expectString,
    jobName: import_smithy_client.expectString,
    lastUpdatedDateTime: (_) => (0, import_smithy_client.expectNonNull)((0, import_smithy_client.parseEpochTimestamp)((0, import_smithy_client.expectNumber)(_))),
    publishAttributionMetricsToS3: import_smithy_client.expectBoolean,
    roleArn: import_smithy_client.expectString,
    status: import_smithy_client.expectString
  });
}, "de_DatasetImportJob");
var de_DatasetImportJobs = /* @__PURE__ */ __name((output, context) => {
  const retVal = (output || []).filter((e) => e != null).map((entry) => {
    return de_DatasetImportJobSummary(entry, context);
  });
  return retVal;
}, "de_DatasetImportJobs");
var de_DatasetImportJobSummary = /* @__PURE__ */ __name((output, context) => {
  return (0, import_smithy_client.take)(output, {
    creationDateTime: (_) => (0, import_smithy_client.expectNonNull)((0, import_smithy_client.parseEpochTimestamp)((0, import_smithy_client.expectNumber)(_))),
    datasetImportJobArn: import_smithy_client.expectString,
    failureReason: import_smithy_client.expectString,
    importMode: import_smithy_client.expectString,
    jobName: import_smithy_client.expectString,
    lastUpdatedDateTime: (_) => (0, import_smithy_client.expectNonNull)((0, import_smithy_client.parseEpochTimestamp)((0, import_smithy_client.expectNumber)(_))),
    status: import_smithy_client.expectString
  });
}, "de_DatasetImportJobSummary");
var de_Datasets = /* @__PURE__ */ __name((output, context) => {
  const retVal = (output || []).filter((e) => e != null).map((entry) => {
    return de_DatasetSummary(entry, context);
  });
  return retVal;
}, "de_Datasets");
var de_DatasetSchema = /* @__PURE__ */ __name((output, context) => {
  return (0, import_smithy_client.take)(output, {
    creationDateTime: (_) => (0, import_smithy_client.expectNonNull)((0, import_smithy_client.parseEpochTimestamp)((0, import_smithy_client.expectNumber)(_))),
    domain: import_smithy_client.expectString,
    lastUpdatedDateTime: (_) => (0, import_smithy_client.expectNonNull)((0, import_smithy_client.parseEpochTimestamp)((0, import_smithy_client.expectNumber)(_))),
    name: import_smithy_client.expectString,
    schema: import_smithy_client.expectString,
    schemaArn: import_smithy_client.expectString
  });
}, "de_DatasetSchema");
var de_DatasetSchemaSummary = /* @__PURE__ */ __name((output, context) => {
  return (0, import_smithy_client.take)(output, {
    creationDateTime: (_) => (0, import_smithy_client.expectNonNull)((0, import_smithy_client.parseEpochTimestamp)((0, import_smithy_client.expectNumber)(_))),
    domain: import_smithy_client.expectString,
    lastUpdatedDateTime: (_) => (0, import_smithy_client.expectNonNull)((0, import_smithy_client.parseEpochTimestamp)((0, import_smithy_client.expectNumber)(_))),
    name: import_smithy_client.expectString,
    schemaArn: import_smithy_client.expectString
  });
}, "de_DatasetSchemaSummary");
var de_DatasetSummary = /* @__PURE__ */ __name((output, context) => {
  return (0, import_smithy_client.take)(output, {
    creationDateTime: (_) => (0, import_smithy_client.expectNonNull)((0, import_smithy_client.parseEpochTimestamp)((0, import_smithy_client.expectNumber)(_))),
    datasetArn: import_smithy_client.expectString,
    datasetType: import_smithy_client.expectString,
    lastUpdatedDateTime: (_) => (0, import_smithy_client.expectNonNull)((0, import_smithy_client.parseEpochTimestamp)((0, import_smithy_client.expectNumber)(_))),
    name: import_smithy_client.expectString,
    status: import_smithy_client.expectString
  });
}, "de_DatasetSummary");
var de_DatasetUpdateSummary = /* @__PURE__ */ __name((output, context) => {
  return (0, import_smithy_client.take)(output, {
    creationDateTime: (_) => (0, import_smithy_client.expectNonNull)((0, import_smithy_client.parseEpochTimestamp)((0, import_smithy_client.expectNumber)(_))),
    failureReason: import_smithy_client.expectString,
    lastUpdatedDateTime: (_) => (0, import_smithy_client.expectNonNull)((0, import_smithy_client.parseEpochTimestamp)((0, import_smithy_client.expectNumber)(_))),
    schemaArn: import_smithy_client.expectString,
    status: import_smithy_client.expectString
  });
}, "de_DatasetUpdateSummary");
var de_DefaultContinuousHyperParameterRange = /* @__PURE__ */ __name((output, context) => {
  return (0, import_smithy_client.take)(output, {
    isTunable: import_smithy_client.expectBoolean,
    maxValue: import_smithy_client.limitedParseDouble,
    minValue: import_smithy_client.limitedParseDouble,
    name: import_smithy_client.expectString
  });
}, "de_DefaultContinuousHyperParameterRange");
var de_DefaultContinuousHyperParameterRanges = /* @__PURE__ */ __name((output, context) => {
  const retVal = (output || []).filter((e) => e != null).map((entry) => {
    return de_DefaultContinuousHyperParameterRange(entry, context);
  });
  return retVal;
}, "de_DefaultContinuousHyperParameterRanges");
var de_DefaultHyperParameterRanges = /* @__PURE__ */ __name((output, context) => {
  return (0, import_smithy_client.take)(output, {
    categoricalHyperParameterRanges: import_smithy_client._json,
    continuousHyperParameterRanges: (_) => de_DefaultContinuousHyperParameterRanges(_, context),
    integerHyperParameterRanges: import_smithy_client._json
  });
}, "de_DefaultHyperParameterRanges");
var de_DescribeAlgorithmResponse = /* @__PURE__ */ __name((output, context) => {
  return (0, import_smithy_client.take)(output, {
    algorithm: (_) => de_Algorithm(_, context)
  });
}, "de_DescribeAlgorithmResponse");
var de_DescribeBatchInferenceJobResponse = /* @__PURE__ */ __name((output, context) => {
  return (0, import_smithy_client.take)(output, {
    batchInferenceJob: (_) => de_BatchInferenceJob(_, context)
  });
}, "de_DescribeBatchInferenceJobResponse");
var de_DescribeBatchSegmentJobResponse = /* @__PURE__ */ __name((output, context) => {
  return (0, import_smithy_client.take)(output, {
    batchSegmentJob: (_) => de_BatchSegmentJob(_, context)
  });
}, "de_DescribeBatchSegmentJobResponse");
var de_DescribeCampaignResponse = /* @__PURE__ */ __name((output, context) => {
  return (0, import_smithy_client.take)(output, {
    campaign: (_) => de_Campaign(_, context)
  });
}, "de_DescribeCampaignResponse");
var de_DescribeDatasetExportJobResponse = /* @__PURE__ */ __name((output, context) => {
  return (0, import_smithy_client.take)(output, {
    datasetExportJob: (_) => de_DatasetExportJob(_, context)
  });
}, "de_DescribeDatasetExportJobResponse");
var de_DescribeDatasetGroupResponse = /* @__PURE__ */ __name((output, context) => {
  return (0, import_smithy_client.take)(output, {
    datasetGroup: (_) => de_DatasetGroup(_, context)
  });
}, "de_DescribeDatasetGroupResponse");
var de_DescribeDatasetImportJobResponse = /* @__PURE__ */ __name((output, context) => {
  return (0, import_smithy_client.take)(output, {
    datasetImportJob: (_) => de_DatasetImportJob(_, context)
  });
}, "de_DescribeDatasetImportJobResponse");
var de_DescribeDatasetResponse = /* @__PURE__ */ __name((output, context) => {
  return (0, import_smithy_client.take)(output, {
    dataset: (_) => de_Dataset(_, context)
  });
}, "de_DescribeDatasetResponse");
var de_DescribeEventTrackerResponse = /* @__PURE__ */ __name((output, context) => {
  return (0, import_smithy_client.take)(output, {
    eventTracker: (_) => de_EventTracker(_, context)
  });
}, "de_DescribeEventTrackerResponse");
var de_DescribeFeatureTransformationResponse = /* @__PURE__ */ __name((output, context) => {
  return (0, import_smithy_client.take)(output, {
    featureTransformation: (_) => de_FeatureTransformation(_, context)
  });
}, "de_DescribeFeatureTransformationResponse");
var de_DescribeFilterResponse = /* @__PURE__ */ __name((output, context) => {
  return (0, import_smithy_client.take)(output, {
    filter: (_) => de_Filter(_, context)
  });
}, "de_DescribeFilterResponse");
var de_DescribeMetricAttributionResponse = /* @__PURE__ */ __name((output, context) => {
  return (0, import_smithy_client.take)(output, {
    metricAttribution: (_) => de_MetricAttribution(_, context)
  });
}, "de_DescribeMetricAttributionResponse");
var de_DescribeRecipeResponse = /* @__PURE__ */ __name((output, context) => {
  return (0, import_smithy_client.take)(output, {
    recipe: (_) => de_Recipe(_, context)
  });
}, "de_DescribeRecipeResponse");
var de_DescribeRecommenderResponse = /* @__PURE__ */ __name((output, context) => {
  return (0, import_smithy_client.take)(output, {
    recommender: (_) => de_Recommender(_, context)
  });
}, "de_DescribeRecommenderResponse");
var de_DescribeSchemaResponse = /* @__PURE__ */ __name((output, context) => {
  return (0, import_smithy_client.take)(output, {
    schema: (_) => de_DatasetSchema(_, context)
  });
}, "de_DescribeSchemaResponse");
var de_DescribeSolutionResponse = /* @__PURE__ */ __name((output, context) => {
  return (0, import_smithy_client.take)(output, {
    solution: (_) => de_Solution(_, context)
  });
}, "de_DescribeSolutionResponse");
var de_DescribeSolutionVersionResponse = /* @__PURE__ */ __name((output, context) => {
  return (0, import_smithy_client.take)(output, {
    solutionVersion: (_) => de_SolutionVersion(_, context)
  });
}, "de_DescribeSolutionVersionResponse");
var de_EventTracker = /* @__PURE__ */ __name((output, context) => {
  return (0, import_smithy_client.take)(output, {
    accountId: import_smithy_client.expectString,
    creationDateTime: (_) => (0, import_smithy_client.expectNonNull)((0, import_smithy_client.parseEpochTimestamp)((0, import_smithy_client.expectNumber)(_))),
    datasetGroupArn: import_smithy_client.expectString,
    eventTrackerArn: import_smithy_client.expectString,
    lastUpdatedDateTime: (_) => (0, import_smithy_client.expectNonNull)((0, import_smithy_client.parseEpochTimestamp)((0, import_smithy_client.expectNumber)(_))),
    name: import_smithy_client.expectString,
    status: import_smithy_client.expectString,
    trackingId: import_smithy_client.expectString
  });
}, "de_EventTracker");
var de_EventTrackers = /* @__PURE__ */ __name((output, context) => {
  const retVal = (output || []).filter((e) => e != null).map((entry) => {
    return de_EventTrackerSummary(entry, context);
  });
  return retVal;
}, "de_EventTrackers");
var de_EventTrackerSummary = /* @__PURE__ */ __name((output, context) => {
  return (0, import_smithy_client.take)(output, {
    creationDateTime: (_) => (0, import_smithy_client.expectNonNull)((0, import_smithy_client.parseEpochTimestamp)((0, import_smithy_client.expectNumber)(_))),
    eventTrackerArn: import_smithy_client.expectString,
    lastUpdatedDateTime: (_) => (0, import_smithy_client.expectNonNull)((0, import_smithy_client.parseEpochTimestamp)((0, import_smithy_client.expectNumber)(_))),
    name: import_smithy_client.expectString,
    status: import_smithy_client.expectString
  });
}, "de_EventTrackerSummary");
var de_FeatureTransformation = /* @__PURE__ */ __name((output, context) => {
  return (0, import_smithy_client.take)(output, {
    creationDateTime: (_) => (0, import_smithy_client.expectNonNull)((0, import_smithy_client.parseEpochTimestamp)((0, import_smithy_client.expectNumber)(_))),
    defaultParameters: import_smithy_client._json,
    featureTransformationArn: import_smithy_client.expectString,
    lastUpdatedDateTime: (_) => (0, import_smithy_client.expectNonNull)((0, import_smithy_client.parseEpochTimestamp)((0, import_smithy_client.expectNumber)(_))),
    name: import_smithy_client.expectString,
    status: import_smithy_client.expectString
  });
}, "de_FeatureTransformation");
var de_Filter = /* @__PURE__ */ __name((output, context) => {
  return (0, import_smithy_client.take)(output, {
    creationDateTime: (_) => (0, import_smithy_client.expectNonNull)((0, import_smithy_client.parseEpochTimestamp)((0, import_smithy_client.expectNumber)(_))),
    datasetGroupArn: import_smithy_client.expectString,
    failureReason: import_smithy_client.expectString,
    filterArn: import_smithy_client.expectString,
    filterExpression: import_smithy_client.expectString,
    lastUpdatedDateTime: (_) => (0, import_smithy_client.expectNonNull)((0, import_smithy_client.parseEpochTimestamp)((0, import_smithy_client.expectNumber)(_))),
    name: import_smithy_client.expectString,
    status: import_smithy_client.expectString
  });
}, "de_Filter");
var de_Filters = /* @__PURE__ */ __name((output, context) => {
  const retVal = (output || []).filter((e) => e != null).map((entry) => {
    return de_FilterSummary(entry, context);
  });
  return retVal;
}, "de_Filters");
var de_FilterSummary = /* @__PURE__ */ __name((output, context) => {
  return (0, import_smithy_client.take)(output, {
    creationDateTime: (_) => (0, import_smithy_client.expectNonNull)((0, import_smithy_client.parseEpochTimestamp)((0, import_smithy_client.expectNumber)(_))),
    datasetGroupArn: import_smithy_client.expectString,
    failureReason: import_smithy_client.expectString,
    filterArn: import_smithy_client.expectString,
    lastUpdatedDateTime: (_) => (0, import_smithy_client.expectNonNull)((0, import_smithy_client.parseEpochTimestamp)((0, import_smithy_client.expectNumber)(_))),
    name: import_smithy_client.expectString,
    status: import_smithy_client.expectString
  });
}, "de_FilterSummary");
var de_GetSolutionMetricsResponse = /* @__PURE__ */ __name((output, context) => {
  return (0, import_smithy_client.take)(output, {
    metrics: (_) => de_Metrics(_, context),
    solutionVersionArn: import_smithy_client.expectString
  });
}, "de_GetSolutionMetricsResponse");
var de_HPOConfig = /* @__PURE__ */ __name((output, context) => {
  return (0, import_smithy_client.take)(output, {
    algorithmHyperParameterRanges: (_) => de_HyperParameterRanges(_, context),
    hpoObjective: import_smithy_client._json,
    hpoResourceConfig: import_smithy_client._json
  });
}, "de_HPOConfig");
var de_HyperParameterRanges = /* @__PURE__ */ __name((output, context) => {
  return (0, import_smithy_client.take)(output, {
    categoricalHyperParameterRanges: import_smithy_client._json,
    continuousHyperParameterRanges: (_) => de_ContinuousHyperParameterRanges(_, context),
    integerHyperParameterRanges: import_smithy_client._json
  });
}, "de_HyperParameterRanges");
var de_ListBatchInferenceJobsResponse = /* @__PURE__ */ __name((output, context) => {
  return (0, import_smithy_client.take)(output, {
    batchInferenceJobs: (_) => de_BatchInferenceJobs(_, context),
    nextToken: import_smithy_client.expectString
  });
}, "de_ListBatchInferenceJobsResponse");
var de_ListBatchSegmentJobsResponse = /* @__PURE__ */ __name((output, context) => {
  return (0, import_smithy_client.take)(output, {
    batchSegmentJobs: (_) => de_BatchSegmentJobs(_, context),
    nextToken: import_smithy_client.expectString
  });
}, "de_ListBatchSegmentJobsResponse");
var de_ListCampaignsResponse = /* @__PURE__ */ __name((output, context) => {
  return (0, import_smithy_client.take)(output, {
    campaigns: (_) => de_Campaigns(_, context),
    nextToken: import_smithy_client.expectString
  });
}, "de_ListCampaignsResponse");
var de_ListDatasetExportJobsResponse = /* @__PURE__ */ __name((output, context) => {
  return (0, import_smithy_client.take)(output, {
    datasetExportJobs: (_) => de_DatasetExportJobs(_, context),
    nextToken: import_smithy_client.expectString
  });
}, "de_ListDatasetExportJobsResponse");
var de_ListDatasetGroupsResponse = /* @__PURE__ */ __name((output, context) => {
  return (0, import_smithy_client.take)(output, {
    datasetGroups: (_) => de_DatasetGroups(_, context),
    nextToken: import_smithy_client.expectString
  });
}, "de_ListDatasetGroupsResponse");
var de_ListDatasetImportJobsResponse = /* @__PURE__ */ __name((output, context) => {
  return (0, import_smithy_client.take)(output, {
    datasetImportJobs: (_) => de_DatasetImportJobs(_, context),
    nextToken: import_smithy_client.expectString
  });
}, "de_ListDatasetImportJobsResponse");
var de_ListDatasetsResponse = /* @__PURE__ */ __name((output, context) => {
  return (0, import_smithy_client.take)(output, {
    datasets: (_) => de_Datasets(_, context),
    nextToken: import_smithy_client.expectString
  });
}, "de_ListDatasetsResponse");
var de_ListEventTrackersResponse = /* @__PURE__ */ __name((output, context) => {
  return (0, import_smithy_client.take)(output, {
    eventTrackers: (_) => de_EventTrackers(_, context),
    nextToken: import_smithy_client.expectString
  });
}, "de_ListEventTrackersResponse");
var de_ListFiltersResponse = /* @__PURE__ */ __name((output, context) => {
  return (0, import_smithy_client.take)(output, {
    Filters: (_) => de_Filters(_, context),
    nextToken: import_smithy_client.expectString
  });
}, "de_ListFiltersResponse");
var de_ListMetricAttributionsResponse = /* @__PURE__ */ __name((output, context) => {
  return (0, import_smithy_client.take)(output, {
    metricAttributions: (_) => de_MetricAttributions(_, context),
    nextToken: import_smithy_client.expectString
  });
}, "de_ListMetricAttributionsResponse");
var de_ListRecipesResponse = /* @__PURE__ */ __name((output, context) => {
  return (0, import_smithy_client.take)(output, {
    nextToken: import_smithy_client.expectString,
    recipes: (_) => de_Recipes(_, context)
  });
}, "de_ListRecipesResponse");
var de_ListRecommendersResponse = /* @__PURE__ */ __name((output, context) => {
  return (0, import_smithy_client.take)(output, {
    nextToken: import_smithy_client.expectString,
    recommenders: (_) => de_Recommenders(_, context)
  });
}, "de_ListRecommendersResponse");
var de_ListSchemasResponse = /* @__PURE__ */ __name((output, context) => {
  return (0, import_smithy_client.take)(output, {
    nextToken: import_smithy_client.expectString,
    schemas: (_) => de_Schemas(_, context)
  });
}, "de_ListSchemasResponse");
var de_ListSolutionsResponse = /* @__PURE__ */ __name((output, context) => {
  return (0, import_smithy_client.take)(output, {
    nextToken: import_smithy_client.expectString,
    solutions: (_) => de_Solutions(_, context)
  });
}, "de_ListSolutionsResponse");
var de_ListSolutionVersionsResponse = /* @__PURE__ */ __name((output, context) => {
  return (0, import_smithy_client.take)(output, {
    nextToken: import_smithy_client.expectString,
    solutionVersions: (_) => de_SolutionVersions(_, context)
  });
}, "de_ListSolutionVersionsResponse");
var de_MetricAttribution = /* @__PURE__ */ __name((output, context) => {
  return (0, import_smithy_client.take)(output, {
    creationDateTime: (_) => (0, import_smithy_client.expectNonNull)((0, import_smithy_client.parseEpochTimestamp)((0, import_smithy_client.expectNumber)(_))),
    datasetGroupArn: import_smithy_client.expectString,
    failureReason: import_smithy_client.expectString,
    lastUpdatedDateTime: (_) => (0, import_smithy_client.expectNonNull)((0, import_smithy_client.parseEpochTimestamp)((0, import_smithy_client.expectNumber)(_))),
    metricAttributionArn: import_smithy_client.expectString,
    metricsOutputConfig: import_smithy_client._json,
    name: import_smithy_client.expectString,
    status: import_smithy_client.expectString
  });
}, "de_MetricAttribution");
var de_MetricAttributions = /* @__PURE__ */ __name((output, context) => {
  const retVal = (output || []).filter((e) => e != null).map((entry) => {
    return de_MetricAttributionSummary(entry, context);
  });
  return retVal;
}, "de_MetricAttributions");
var de_MetricAttributionSummary = /* @__PURE__ */ __name((output, context) => {
  return (0, import_smithy_client.take)(output, {
    creationDateTime: (_) => (0, import_smithy_client.expectNonNull)((0, import_smithy_client.parseEpochTimestamp)((0, import_smithy_client.expectNumber)(_))),
    failureReason: import_smithy_client.expectString,
    lastUpdatedDateTime: (_) => (0, import_smithy_client.expectNonNull)((0, import_smithy_client.parseEpochTimestamp)((0, import_smithy_client.expectNumber)(_))),
    metricAttributionArn: import_smithy_client.expectString,
    name: import_smithy_client.expectString,
    status: import_smithy_client.expectString
  });
}, "de_MetricAttributionSummary");
var de_Metrics = /* @__PURE__ */ __name((output, context) => {
  return Object.entries(output).reduce((acc, [key, value]) => {
    if (value === null) {
      return acc;
    }
    acc[key] = (0, import_smithy_client.limitedParseDouble)(value);
    return acc;
  }, {});
}, "de_Metrics");
var de_Recipe = /* @__PURE__ */ __name((output, context) => {
  return (0, import_smithy_client.take)(output, {
    algorithmArn: import_smithy_client.expectString,
    creationDateTime: (_) => (0, import_smithy_client.expectNonNull)((0, import_smithy_client.parseEpochTimestamp)((0, import_smithy_client.expectNumber)(_))),
    description: import_smithy_client.expectString,
    featureTransformationArn: import_smithy_client.expectString,
    lastUpdatedDateTime: (_) => (0, import_smithy_client.expectNonNull)((0, import_smithy_client.parseEpochTimestamp)((0, import_smithy_client.expectNumber)(_))),
    name: import_smithy_client.expectString,
    recipeArn: import_smithy_client.expectString,
    recipeType: import_smithy_client.expectString,
    status: import_smithy_client.expectString
  });
}, "de_Recipe");
var de_Recipes = /* @__PURE__ */ __name((output, context) => {
  const retVal = (output || []).filter((e) => e != null).map((entry) => {
    return de_RecipeSummary(entry, context);
  });
  return retVal;
}, "de_Recipes");
var de_RecipeSummary = /* @__PURE__ */ __name((output, context) => {
  return (0, import_smithy_client.take)(output, {
    creationDateTime: (_) => (0, import_smithy_client.expectNonNull)((0, import_smithy_client.parseEpochTimestamp)((0, import_smithy_client.expectNumber)(_))),
    domain: import_smithy_client.expectString,
    lastUpdatedDateTime: (_) => (0, import_smithy_client.expectNonNull)((0, import_smithy_client.parseEpochTimestamp)((0, import_smithy_client.expectNumber)(_))),
    name: import_smithy_client.expectString,
    recipeArn: import_smithy_client.expectString,
    status: import_smithy_client.expectString
  });
}, "de_RecipeSummary");
var de_Recommender = /* @__PURE__ */ __name((output, context) => {
  return (0, import_smithy_client.take)(output, {
    creationDateTime: (_) => (0, import_smithy_client.expectNonNull)((0, import_smithy_client.parseEpochTimestamp)((0, import_smithy_client.expectNumber)(_))),
    datasetGroupArn: import_smithy_client.expectString,
    failureReason: import_smithy_client.expectString,
    lastUpdatedDateTime: (_) => (0, import_smithy_client.expectNonNull)((0, import_smithy_client.parseEpochTimestamp)((0, import_smithy_client.expectNumber)(_))),
    latestRecommenderUpdate: (_) => de_RecommenderUpdateSummary(_, context),
    modelMetrics: (_) => de_Metrics(_, context),
    name: import_smithy_client.expectString,
    recipeArn: import_smithy_client.expectString,
    recommenderArn: import_smithy_client.expectString,
    recommenderConfig: import_smithy_client._json,
    status: import_smithy_client.expectString
  });
}, "de_Recommender");
var de_Recommenders = /* @__PURE__ */ __name((output, context) => {
  const retVal = (output || []).filter((e) => e != null).map((entry) => {
    return de_RecommenderSummary(entry, context);
  });
  return retVal;
}, "de_Recommenders");
var de_RecommenderSummary = /* @__PURE__ */ __name((output, context) => {
  return (0, import_smithy_client.take)(output, {
    creationDateTime: (_) => (0, import_smithy_client.expectNonNull)((0, import_smithy_client.parseEpochTimestamp)((0, import_smithy_client.expectNumber)(_))),
    datasetGroupArn: import_smithy_client.expectString,
    lastUpdatedDateTime: (_) => (0, import_smithy_client.expectNonNull)((0, import_smithy_client.parseEpochTimestamp)((0, import_smithy_client.expectNumber)(_))),
    name: import_smithy_client.expectString,
    recipeArn: import_smithy_client.expectString,
    recommenderArn: import_smithy_client.expectString,
    recommenderConfig: import_smithy_client._json,
    status: import_smithy_client.expectString
  });
}, "de_RecommenderSummary");
var de_RecommenderUpdateSummary = /* @__PURE__ */ __name((output, context) => {
  return (0, import_smithy_client.take)(output, {
    creationDateTime: (_) => (0, import_smithy_client.expectNonNull)((0, import_smithy_client.parseEpochTimestamp)((0, import_smithy_client.expectNumber)(_))),
    failureReason: import_smithy_client.expectString,
    lastUpdatedDateTime: (_) => (0, import_smithy_client.expectNonNull)((0, import_smithy_client.parseEpochTimestamp)((0, import_smithy_client.expectNumber)(_))),
    recommenderConfig: import_smithy_client._json,
    status: import_smithy_client.expectString
  });
}, "de_RecommenderUpdateSummary");
var de_Schemas = /* @__PURE__ */ __name((output, context) => {
  const retVal = (output || []).filter((e) => e != null).map((entry) => {
    return de_DatasetSchemaSummary(entry, context);
  });
  return retVal;
}, "de_Schemas");
var de_Solution = /* @__PURE__ */ __name((output, context) => {
  return (0, import_smithy_client.take)(output, {
    autoMLResult: import_smithy_client._json,
    creationDateTime: (_) => (0, import_smithy_client.expectNonNull)((0, import_smithy_client.parseEpochTimestamp)((0, import_smithy_client.expectNumber)(_))),
    datasetGroupArn: import_smithy_client.expectString,
    eventType: import_smithy_client.expectString,
    lastUpdatedDateTime: (_) => (0, import_smithy_client.expectNonNull)((0, import_smithy_client.parseEpochTimestamp)((0, import_smithy_client.expectNumber)(_))),
    latestSolutionVersion: (_) => de_SolutionVersionSummary(_, context),
    name: import_smithy_client.expectString,
    performAutoML: import_smithy_client.expectBoolean,
    performAutoTraining: import_smithy_client.expectBoolean,
    performHPO: import_smithy_client.expectBoolean,
    recipeArn: import_smithy_client.expectString,
    solutionArn: import_smithy_client.expectString,
    solutionConfig: (_) => de_SolutionConfig(_, context),
    status: import_smithy_client.expectString
  });
}, "de_Solution");
var de_SolutionConfig = /* @__PURE__ */ __name((output, context) => {
  return (0, import_smithy_client.take)(output, {
    algorithmHyperParameters: import_smithy_client._json,
    autoMLConfig: import_smithy_client._json,
    autoTrainingConfig: import_smithy_client._json,
    eventValueThreshold: import_smithy_client.expectString,
    featureTransformationParameters: import_smithy_client._json,
    hpoConfig: (_) => de_HPOConfig(_, context),
    optimizationObjective: import_smithy_client._json,
    trainingDataConfig: import_smithy_client._json
  });
}, "de_SolutionConfig");
var de_Solutions = /* @__PURE__ */ __name((output, context) => {
  const retVal = (output || []).filter((e) => e != null).map((entry) => {
    return de_SolutionSummary(entry, context);
  });
  return retVal;
}, "de_Solutions");
var de_SolutionSummary = /* @__PURE__ */ __name((output, context) => {
  return (0, import_smithy_client.take)(output, {
    creationDateTime: (_) => (0, import_smithy_client.expectNonNull)((0, import_smithy_client.parseEpochTimestamp)((0, import_smithy_client.expectNumber)(_))),
    lastUpdatedDateTime: (_) => (0, import_smithy_client.expectNonNull)((0, import_smithy_client.parseEpochTimestamp)((0, import_smithy_client.expectNumber)(_))),
    name: import_smithy_client.expectString,
    recipeArn: import_smithy_client.expectString,
    solutionArn: import_smithy_client.expectString,
    status: import_smithy_client.expectString
  });
}, "de_SolutionSummary");
var de_SolutionVersion = /* @__PURE__ */ __name((output, context) => {
  return (0, import_smithy_client.take)(output, {
    creationDateTime: (_) => (0, import_smithy_client.expectNonNull)((0, import_smithy_client.parseEpochTimestamp)((0, import_smithy_client.expectNumber)(_))),
    datasetGroupArn: import_smithy_client.expectString,
    eventType: import_smithy_client.expectString,
    failureReason: import_smithy_client.expectString,
    lastUpdatedDateTime: (_) => (0, import_smithy_client.expectNonNull)((0, import_smithy_client.parseEpochTimestamp)((0, import_smithy_client.expectNumber)(_))),
    name: import_smithy_client.expectString,
    performAutoML: import_smithy_client.expectBoolean,
    performHPO: import_smithy_client.expectBoolean,
    recipeArn: import_smithy_client.expectString,
    solutionArn: import_smithy_client.expectString,
    solutionConfig: (_) => de_SolutionConfig(_, context),
    solutionVersionArn: import_smithy_client.expectString,
    status: import_smithy_client.expectString,
    trainingHours: import_smithy_client.limitedParseDouble,
    trainingMode: import_smithy_client.expectString,
    trainingType: import_smithy_client.expectString,
    tunedHPOParams: import_smithy_client._json
  });
}, "de_SolutionVersion");
var de_SolutionVersions = /* @__PURE__ */ __name((output, context) => {
  const retVal = (output || []).filter((e) => e != null).map((entry) => {
    return de_SolutionVersionSummary(entry, context);
  });
  return retVal;
}, "de_SolutionVersions");
var de_SolutionVersionSummary = /* @__PURE__ */ __name((output, context) => {
  return (0, import_smithy_client.take)(output, {
    creationDateTime: (_) => (0, import_smithy_client.expectNonNull)((0, import_smithy_client.parseEpochTimestamp)((0, import_smithy_client.expectNumber)(_))),
    failureReason: import_smithy_client.expectString,
    lastUpdatedDateTime: (_) => (0, import_smithy_client.expectNonNull)((0, import_smithy_client.parseEpochTimestamp)((0, import_smithy_client.expectNumber)(_))),
    solutionVersionArn: import_smithy_client.expectString,
    status: import_smithy_client.expectString,
    trainingMode: import_smithy_client.expectString,
    trainingType: import_smithy_client.expectString
  });
}, "de_SolutionVersionSummary");
var deserializeMetadata = /* @__PURE__ */ __name((output) => ({
  httpStatusCode: output.statusCode,
  requestId: output.headers["x-amzn-requestid"] ?? output.headers["x-amzn-request-id"] ?? output.headers["x-amz-request-id"],
  extendedRequestId: output.headers["x-amz-id-2"],
  cfId: output.headers["x-amz-cf-id"]
}), "deserializeMetadata");
var throwDefaultError = (0, import_smithy_client.withBaseException)(PersonalizeServiceException);
var buildHttpRpcRequest = /* @__PURE__ */ __name(async (context, headers, path, resolvedHostname, body) => {
  const { hostname, protocol = "https", port, path: basePath } = await context.endpoint();
  const contents = {
    protocol,
    hostname,
    port,
    method: "POST",
    path: basePath.endsWith("/") ? basePath.slice(0, -1) + path : basePath + path,
    headers
  };
  if (resolvedHostname !== void 0) {
    contents.hostname = resolvedHostname;
  }
  if (body !== void 0) {
    contents.body = body;
  }
  return new import_protocol_http.HttpRequest(contents);
}, "buildHttpRpcRequest");
function sharedHeaders(operation) {
  return {
    "content-type": "application/x-amz-json-1.1",
    "x-amz-target": `AmazonPersonalize.${operation}`
  };
}
__name(sharedHeaders, "sharedHeaders");

// src/commands/CreateBatchInferenceJobCommand.ts
var _CreateBatchInferenceJobCommand = class _CreateBatchInferenceJobCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AmazonPersonalize", "CreateBatchInferenceJob", {}).n("PersonalizeClient", "CreateBatchInferenceJobCommand").f(void 0, void 0).ser(se_CreateBatchInferenceJobCommand).de(de_CreateBatchInferenceJobCommand).build() {
};
__name(_CreateBatchInferenceJobCommand, "CreateBatchInferenceJobCommand");
var CreateBatchInferenceJobCommand = _CreateBatchInferenceJobCommand;

// src/commands/CreateBatchSegmentJobCommand.ts




var _CreateBatchSegmentJobCommand = class _CreateBatchSegmentJobCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AmazonPersonalize", "CreateBatchSegmentJob", {}).n("PersonalizeClient", "CreateBatchSegmentJobCommand").f(void 0, void 0).ser(se_CreateBatchSegmentJobCommand).de(de_CreateBatchSegmentJobCommand).build() {
};
__name(_CreateBatchSegmentJobCommand, "CreateBatchSegmentJobCommand");
var CreateBatchSegmentJobCommand = _CreateBatchSegmentJobCommand;

// src/commands/CreateCampaignCommand.ts




var _CreateCampaignCommand = class _CreateCampaignCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AmazonPersonalize", "CreateCampaign", {}).n("PersonalizeClient", "CreateCampaignCommand").f(void 0, void 0).ser(se_CreateCampaignCommand).de(de_CreateCampaignCommand).build() {
};
__name(_CreateCampaignCommand, "CreateCampaignCommand");
var CreateCampaignCommand = _CreateCampaignCommand;

// src/commands/CreateDatasetCommand.ts




var _CreateDatasetCommand = class _CreateDatasetCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AmazonPersonalize", "CreateDataset", {}).n("PersonalizeClient", "CreateDatasetCommand").f(void 0, void 0).ser(se_CreateDatasetCommand).de(de_CreateDatasetCommand).build() {
};
__name(_CreateDatasetCommand, "CreateDatasetCommand");
var CreateDatasetCommand = _CreateDatasetCommand;

// src/commands/CreateDatasetExportJobCommand.ts




var _CreateDatasetExportJobCommand = class _CreateDatasetExportJobCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AmazonPersonalize", "CreateDatasetExportJob", {}).n("PersonalizeClient", "CreateDatasetExportJobCommand").f(void 0, void 0).ser(se_CreateDatasetExportJobCommand).de(de_CreateDatasetExportJobCommand).build() {
};
__name(_CreateDatasetExportJobCommand, "CreateDatasetExportJobCommand");
var CreateDatasetExportJobCommand = _CreateDatasetExportJobCommand;

// src/commands/CreateDatasetGroupCommand.ts




var _CreateDatasetGroupCommand = class _CreateDatasetGroupCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AmazonPersonalize", "CreateDatasetGroup", {}).n("PersonalizeClient", "CreateDatasetGroupCommand").f(void 0, void 0).ser(se_CreateDatasetGroupCommand).de(de_CreateDatasetGroupCommand).build() {
};
__name(_CreateDatasetGroupCommand, "CreateDatasetGroupCommand");
var CreateDatasetGroupCommand = _CreateDatasetGroupCommand;

// src/commands/CreateDatasetImportJobCommand.ts




var _CreateDatasetImportJobCommand = class _CreateDatasetImportJobCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AmazonPersonalize", "CreateDatasetImportJob", {}).n("PersonalizeClient", "CreateDatasetImportJobCommand").f(void 0, void 0).ser(se_CreateDatasetImportJobCommand).de(de_CreateDatasetImportJobCommand).build() {
};
__name(_CreateDatasetImportJobCommand, "CreateDatasetImportJobCommand");
var CreateDatasetImportJobCommand = _CreateDatasetImportJobCommand;

// src/commands/CreateEventTrackerCommand.ts




var _CreateEventTrackerCommand = class _CreateEventTrackerCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AmazonPersonalize", "CreateEventTracker", {}).n("PersonalizeClient", "CreateEventTrackerCommand").f(void 0, void 0).ser(se_CreateEventTrackerCommand).de(de_CreateEventTrackerCommand).build() {
};
__name(_CreateEventTrackerCommand, "CreateEventTrackerCommand");
var CreateEventTrackerCommand = _CreateEventTrackerCommand;

// src/commands/CreateFilterCommand.ts




var _CreateFilterCommand = class _CreateFilterCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AmazonPersonalize", "CreateFilter", {}).n("PersonalizeClient", "CreateFilterCommand").f(CreateFilterRequestFilterSensitiveLog, void 0).ser(se_CreateFilterCommand).de(de_CreateFilterCommand).build() {
};
__name(_CreateFilterCommand, "CreateFilterCommand");
var CreateFilterCommand = _CreateFilterCommand;

// src/commands/CreateMetricAttributionCommand.ts




var _CreateMetricAttributionCommand = class _CreateMetricAttributionCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AmazonPersonalize", "CreateMetricAttribution", {}).n("PersonalizeClient", "CreateMetricAttributionCommand").f(void 0, void 0).ser(se_CreateMetricAttributionCommand).de(de_CreateMetricAttributionCommand).build() {
};
__name(_CreateMetricAttributionCommand, "CreateMetricAttributionCommand");
var CreateMetricAttributionCommand = _CreateMetricAttributionCommand;

// src/commands/CreateRecommenderCommand.ts




var _CreateRecommenderCommand = class _CreateRecommenderCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AmazonPersonalize", "CreateRecommender", {}).n("PersonalizeClient", "CreateRecommenderCommand").f(void 0, void 0).ser(se_CreateRecommenderCommand).de(de_CreateRecommenderCommand).build() {
};
__name(_CreateRecommenderCommand, "CreateRecommenderCommand");
var CreateRecommenderCommand = _CreateRecommenderCommand;

// src/commands/CreateSchemaCommand.ts




var _CreateSchemaCommand = class _CreateSchemaCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AmazonPersonalize", "CreateSchema", {}).n("PersonalizeClient", "CreateSchemaCommand").f(void 0, void 0).ser(se_CreateSchemaCommand).de(de_CreateSchemaCommand).build() {
};
__name(_CreateSchemaCommand, "CreateSchemaCommand");
var CreateSchemaCommand = _CreateSchemaCommand;

// src/commands/CreateSolutionCommand.ts




var _CreateSolutionCommand = class _CreateSolutionCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AmazonPersonalize", "CreateSolution", {}).n("PersonalizeClient", "CreateSolutionCommand").f(void 0, void 0).ser(se_CreateSolutionCommand).de(de_CreateSolutionCommand).build() {
};
__name(_CreateSolutionCommand, "CreateSolutionCommand");
var CreateSolutionCommand = _CreateSolutionCommand;

// src/commands/CreateSolutionVersionCommand.ts




var _CreateSolutionVersionCommand = class _CreateSolutionVersionCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AmazonPersonalize", "CreateSolutionVersion", {}).n("PersonalizeClient", "CreateSolutionVersionCommand").f(void 0, void 0).ser(se_CreateSolutionVersionCommand).de(de_CreateSolutionVersionCommand).build() {
};
__name(_CreateSolutionVersionCommand, "CreateSolutionVersionCommand");
var CreateSolutionVersionCommand = _CreateSolutionVersionCommand;

// src/commands/DeleteCampaignCommand.ts




var _DeleteCampaignCommand = class _DeleteCampaignCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AmazonPersonalize", "DeleteCampaign", {}).n("PersonalizeClient", "DeleteCampaignCommand").f(void 0, void 0).ser(se_DeleteCampaignCommand).de(de_DeleteCampaignCommand).build() {
};
__name(_DeleteCampaignCommand, "DeleteCampaignCommand");
var DeleteCampaignCommand = _DeleteCampaignCommand;

// src/commands/DeleteDatasetCommand.ts




var _DeleteDatasetCommand = class _DeleteDatasetCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AmazonPersonalize", "DeleteDataset", {}).n("PersonalizeClient", "DeleteDatasetCommand").f(void 0, void 0).ser(se_DeleteDatasetCommand).de(de_DeleteDatasetCommand).build() {
};
__name(_DeleteDatasetCommand, "DeleteDatasetCommand");
var DeleteDatasetCommand = _DeleteDatasetCommand;

// src/commands/DeleteDatasetGroupCommand.ts




var _DeleteDatasetGroupCommand = class _DeleteDatasetGroupCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AmazonPersonalize", "DeleteDatasetGroup", {}).n("PersonalizeClient", "DeleteDatasetGroupCommand").f(void 0, void 0).ser(se_DeleteDatasetGroupCommand).de(de_DeleteDatasetGroupCommand).build() {
};
__name(_DeleteDatasetGroupCommand, "DeleteDatasetGroupCommand");
var DeleteDatasetGroupCommand = _DeleteDatasetGroupCommand;

// src/commands/DeleteEventTrackerCommand.ts




var _DeleteEventTrackerCommand = class _DeleteEventTrackerCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AmazonPersonalize", "DeleteEventTracker", {}).n("PersonalizeClient", "DeleteEventTrackerCommand").f(void 0, void 0).ser(se_DeleteEventTrackerCommand).de(de_DeleteEventTrackerCommand).build() {
};
__name(_DeleteEventTrackerCommand, "DeleteEventTrackerCommand");
var DeleteEventTrackerCommand = _DeleteEventTrackerCommand;

// src/commands/DeleteFilterCommand.ts




var _DeleteFilterCommand = class _DeleteFilterCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AmazonPersonalize", "DeleteFilter", {}).n("PersonalizeClient", "DeleteFilterCommand").f(void 0, void 0).ser(se_DeleteFilterCommand).de(de_DeleteFilterCommand).build() {
};
__name(_DeleteFilterCommand, "DeleteFilterCommand");
var DeleteFilterCommand = _DeleteFilterCommand;

// src/commands/DeleteMetricAttributionCommand.ts




var _DeleteMetricAttributionCommand = class _DeleteMetricAttributionCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AmazonPersonalize", "DeleteMetricAttribution", {}).n("PersonalizeClient", "DeleteMetricAttributionCommand").f(void 0, void 0).ser(se_DeleteMetricAttributionCommand).de(de_DeleteMetricAttributionCommand).build() {
};
__name(_DeleteMetricAttributionCommand, "DeleteMetricAttributionCommand");
var DeleteMetricAttributionCommand = _DeleteMetricAttributionCommand;

// src/commands/DeleteRecommenderCommand.ts




var _DeleteRecommenderCommand = class _DeleteRecommenderCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AmazonPersonalize", "DeleteRecommender", {}).n("PersonalizeClient", "DeleteRecommenderCommand").f(void 0, void 0).ser(se_DeleteRecommenderCommand).de(de_DeleteRecommenderCommand).build() {
};
__name(_DeleteRecommenderCommand, "DeleteRecommenderCommand");
var DeleteRecommenderCommand = _DeleteRecommenderCommand;

// src/commands/DeleteSchemaCommand.ts




var _DeleteSchemaCommand = class _DeleteSchemaCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AmazonPersonalize", "DeleteSchema", {}).n("PersonalizeClient", "DeleteSchemaCommand").f(void 0, void 0).ser(se_DeleteSchemaCommand).de(de_DeleteSchemaCommand).build() {
};
__name(_DeleteSchemaCommand, "DeleteSchemaCommand");
var DeleteSchemaCommand = _DeleteSchemaCommand;

// src/commands/DeleteSolutionCommand.ts




var _DeleteSolutionCommand = class _DeleteSolutionCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AmazonPersonalize", "DeleteSolution", {}).n("PersonalizeClient", "DeleteSolutionCommand").f(void 0, void 0).ser(se_DeleteSolutionCommand).de(de_DeleteSolutionCommand).build() {
};
__name(_DeleteSolutionCommand, "DeleteSolutionCommand");
var DeleteSolutionCommand = _DeleteSolutionCommand;

// src/commands/DescribeAlgorithmCommand.ts




var _DescribeAlgorithmCommand = class _DescribeAlgorithmCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AmazonPersonalize", "DescribeAlgorithm", {}).n("PersonalizeClient", "DescribeAlgorithmCommand").f(void 0, void 0).ser(se_DescribeAlgorithmCommand).de(de_DescribeAlgorithmCommand).build() {
};
__name(_DescribeAlgorithmCommand, "DescribeAlgorithmCommand");
var DescribeAlgorithmCommand = _DescribeAlgorithmCommand;

// src/commands/DescribeBatchInferenceJobCommand.ts




var _DescribeBatchInferenceJobCommand = class _DescribeBatchInferenceJobCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AmazonPersonalize", "DescribeBatchInferenceJob", {}).n("PersonalizeClient", "DescribeBatchInferenceJobCommand").f(void 0, void 0).ser(se_DescribeBatchInferenceJobCommand).de(de_DescribeBatchInferenceJobCommand).build() {
};
__name(_DescribeBatchInferenceJobCommand, "DescribeBatchInferenceJobCommand");
var DescribeBatchInferenceJobCommand = _DescribeBatchInferenceJobCommand;

// src/commands/DescribeBatchSegmentJobCommand.ts




var _DescribeBatchSegmentJobCommand = class _DescribeBatchSegmentJobCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AmazonPersonalize", "DescribeBatchSegmentJob", {}).n("PersonalizeClient", "DescribeBatchSegmentJobCommand").f(void 0, void 0).ser(se_DescribeBatchSegmentJobCommand).de(de_DescribeBatchSegmentJobCommand).build() {
};
__name(_DescribeBatchSegmentJobCommand, "DescribeBatchSegmentJobCommand");
var DescribeBatchSegmentJobCommand = _DescribeBatchSegmentJobCommand;

// src/commands/DescribeCampaignCommand.ts




var _DescribeCampaignCommand = class _DescribeCampaignCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AmazonPersonalize", "DescribeCampaign", {}).n("PersonalizeClient", "DescribeCampaignCommand").f(void 0, void 0).ser(se_DescribeCampaignCommand).de(de_DescribeCampaignCommand).build() {
};
__name(_DescribeCampaignCommand, "DescribeCampaignCommand");
var DescribeCampaignCommand = _DescribeCampaignCommand;

// src/commands/DescribeDatasetCommand.ts




var _DescribeDatasetCommand = class _DescribeDatasetCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AmazonPersonalize", "DescribeDataset", {}).n("PersonalizeClient", "DescribeDatasetCommand").f(void 0, void 0).ser(se_DescribeDatasetCommand).de(de_DescribeDatasetCommand).build() {
};
__name(_DescribeDatasetCommand, "DescribeDatasetCommand");
var DescribeDatasetCommand = _DescribeDatasetCommand;

// src/commands/DescribeDatasetExportJobCommand.ts




var _DescribeDatasetExportJobCommand = class _DescribeDatasetExportJobCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AmazonPersonalize", "DescribeDatasetExportJob", {}).n("PersonalizeClient", "DescribeDatasetExportJobCommand").f(void 0, void 0).ser(se_DescribeDatasetExportJobCommand).de(de_DescribeDatasetExportJobCommand).build() {
};
__name(_DescribeDatasetExportJobCommand, "DescribeDatasetExportJobCommand");
var DescribeDatasetExportJobCommand = _DescribeDatasetExportJobCommand;

// src/commands/DescribeDatasetGroupCommand.ts




var _DescribeDatasetGroupCommand = class _DescribeDatasetGroupCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AmazonPersonalize", "DescribeDatasetGroup", {}).n("PersonalizeClient", "DescribeDatasetGroupCommand").f(void 0, void 0).ser(se_DescribeDatasetGroupCommand).de(de_DescribeDatasetGroupCommand).build() {
};
__name(_DescribeDatasetGroupCommand, "DescribeDatasetGroupCommand");
var DescribeDatasetGroupCommand = _DescribeDatasetGroupCommand;

// src/commands/DescribeDatasetImportJobCommand.ts




var _DescribeDatasetImportJobCommand = class _DescribeDatasetImportJobCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AmazonPersonalize", "DescribeDatasetImportJob", {}).n("PersonalizeClient", "DescribeDatasetImportJobCommand").f(void 0, void 0).ser(se_DescribeDatasetImportJobCommand).de(de_DescribeDatasetImportJobCommand).build() {
};
__name(_DescribeDatasetImportJobCommand, "DescribeDatasetImportJobCommand");
var DescribeDatasetImportJobCommand = _DescribeDatasetImportJobCommand;

// src/commands/DescribeEventTrackerCommand.ts




var _DescribeEventTrackerCommand = class _DescribeEventTrackerCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AmazonPersonalize", "DescribeEventTracker", {}).n("PersonalizeClient", "DescribeEventTrackerCommand").f(void 0, void 0).ser(se_DescribeEventTrackerCommand).de(de_DescribeEventTrackerCommand).build() {
};
__name(_DescribeEventTrackerCommand, "DescribeEventTrackerCommand");
var DescribeEventTrackerCommand = _DescribeEventTrackerCommand;

// src/commands/DescribeFeatureTransformationCommand.ts




var _DescribeFeatureTransformationCommand = class _DescribeFeatureTransformationCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AmazonPersonalize", "DescribeFeatureTransformation", {}).n("PersonalizeClient", "DescribeFeatureTransformationCommand").f(void 0, void 0).ser(se_DescribeFeatureTransformationCommand).de(de_DescribeFeatureTransformationCommand).build() {
};
__name(_DescribeFeatureTransformationCommand, "DescribeFeatureTransformationCommand");
var DescribeFeatureTransformationCommand = _DescribeFeatureTransformationCommand;

// src/commands/DescribeFilterCommand.ts




var _DescribeFilterCommand = class _DescribeFilterCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AmazonPersonalize", "DescribeFilter", {}).n("PersonalizeClient", "DescribeFilterCommand").f(void 0, DescribeFilterResponseFilterSensitiveLog).ser(se_DescribeFilterCommand).de(de_DescribeFilterCommand).build() {
};
__name(_DescribeFilterCommand, "DescribeFilterCommand");
var DescribeFilterCommand = _DescribeFilterCommand;

// src/commands/DescribeMetricAttributionCommand.ts




var _DescribeMetricAttributionCommand = class _DescribeMetricAttributionCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AmazonPersonalize", "DescribeMetricAttribution", {}).n("PersonalizeClient", "DescribeMetricAttributionCommand").f(void 0, void 0).ser(se_DescribeMetricAttributionCommand).de(de_DescribeMetricAttributionCommand).build() {
};
__name(_DescribeMetricAttributionCommand, "DescribeMetricAttributionCommand");
var DescribeMetricAttributionCommand = _DescribeMetricAttributionCommand;

// src/commands/DescribeRecipeCommand.ts




var _DescribeRecipeCommand = class _DescribeRecipeCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AmazonPersonalize", "DescribeRecipe", {}).n("PersonalizeClient", "DescribeRecipeCommand").f(void 0, void 0).ser(se_DescribeRecipeCommand).de(de_DescribeRecipeCommand).build() {
};
__name(_DescribeRecipeCommand, "DescribeRecipeCommand");
var DescribeRecipeCommand = _DescribeRecipeCommand;

// src/commands/DescribeRecommenderCommand.ts




var _DescribeRecommenderCommand = class _DescribeRecommenderCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AmazonPersonalize", "DescribeRecommender", {}).n("PersonalizeClient", "DescribeRecommenderCommand").f(void 0, void 0).ser(se_DescribeRecommenderCommand).de(de_DescribeRecommenderCommand).build() {
};
__name(_DescribeRecommenderCommand, "DescribeRecommenderCommand");
var DescribeRecommenderCommand = _DescribeRecommenderCommand;

// src/commands/DescribeSchemaCommand.ts




var _DescribeSchemaCommand = class _DescribeSchemaCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AmazonPersonalize", "DescribeSchema", {}).n("PersonalizeClient", "DescribeSchemaCommand").f(void 0, void 0).ser(se_DescribeSchemaCommand).de(de_DescribeSchemaCommand).build() {
};
__name(_DescribeSchemaCommand, "DescribeSchemaCommand");
var DescribeSchemaCommand = _DescribeSchemaCommand;

// src/commands/DescribeSolutionCommand.ts




var _DescribeSolutionCommand = class _DescribeSolutionCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AmazonPersonalize", "DescribeSolution", {}).n("PersonalizeClient", "DescribeSolutionCommand").f(void 0, void 0).ser(se_DescribeSolutionCommand).de(de_DescribeSolutionCommand).build() {
};
__name(_DescribeSolutionCommand, "DescribeSolutionCommand");
var DescribeSolutionCommand = _DescribeSolutionCommand;

// src/commands/DescribeSolutionVersionCommand.ts




var _DescribeSolutionVersionCommand = class _DescribeSolutionVersionCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AmazonPersonalize", "DescribeSolutionVersion", {}).n("PersonalizeClient", "DescribeSolutionVersionCommand").f(void 0, void 0).ser(se_DescribeSolutionVersionCommand).de(de_DescribeSolutionVersionCommand).build() {
};
__name(_DescribeSolutionVersionCommand, "DescribeSolutionVersionCommand");
var DescribeSolutionVersionCommand = _DescribeSolutionVersionCommand;

// src/commands/GetSolutionMetricsCommand.ts




var _GetSolutionMetricsCommand = class _GetSolutionMetricsCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AmazonPersonalize", "GetSolutionMetrics", {}).n("PersonalizeClient", "GetSolutionMetricsCommand").f(void 0, void 0).ser(se_GetSolutionMetricsCommand).de(de_GetSolutionMetricsCommand).build() {
};
__name(_GetSolutionMetricsCommand, "GetSolutionMetricsCommand");
var GetSolutionMetricsCommand = _GetSolutionMetricsCommand;

// src/commands/ListBatchInferenceJobsCommand.ts




var _ListBatchInferenceJobsCommand = class _ListBatchInferenceJobsCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AmazonPersonalize", "ListBatchInferenceJobs", {}).n("PersonalizeClient", "ListBatchInferenceJobsCommand").f(void 0, void 0).ser(se_ListBatchInferenceJobsCommand).de(de_ListBatchInferenceJobsCommand).build() {
};
__name(_ListBatchInferenceJobsCommand, "ListBatchInferenceJobsCommand");
var ListBatchInferenceJobsCommand = _ListBatchInferenceJobsCommand;

// src/commands/ListBatchSegmentJobsCommand.ts




var _ListBatchSegmentJobsCommand = class _ListBatchSegmentJobsCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AmazonPersonalize", "ListBatchSegmentJobs", {}).n("PersonalizeClient", "ListBatchSegmentJobsCommand").f(void 0, void 0).ser(se_ListBatchSegmentJobsCommand).de(de_ListBatchSegmentJobsCommand).build() {
};
__name(_ListBatchSegmentJobsCommand, "ListBatchSegmentJobsCommand");
var ListBatchSegmentJobsCommand = _ListBatchSegmentJobsCommand;

// src/commands/ListCampaignsCommand.ts




var _ListCampaignsCommand = class _ListCampaignsCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AmazonPersonalize", "ListCampaigns", {}).n("PersonalizeClient", "ListCampaignsCommand").f(void 0, void 0).ser(se_ListCampaignsCommand).de(de_ListCampaignsCommand).build() {
};
__name(_ListCampaignsCommand, "ListCampaignsCommand");
var ListCampaignsCommand = _ListCampaignsCommand;

// src/commands/ListDatasetExportJobsCommand.ts




var _ListDatasetExportJobsCommand = class _ListDatasetExportJobsCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AmazonPersonalize", "ListDatasetExportJobs", {}).n("PersonalizeClient", "ListDatasetExportJobsCommand").f(void 0, void 0).ser(se_ListDatasetExportJobsCommand).de(de_ListDatasetExportJobsCommand).build() {
};
__name(_ListDatasetExportJobsCommand, "ListDatasetExportJobsCommand");
var ListDatasetExportJobsCommand = _ListDatasetExportJobsCommand;

// src/commands/ListDatasetGroupsCommand.ts




var _ListDatasetGroupsCommand = class _ListDatasetGroupsCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AmazonPersonalize", "ListDatasetGroups", {}).n("PersonalizeClient", "ListDatasetGroupsCommand").f(void 0, void 0).ser(se_ListDatasetGroupsCommand).de(de_ListDatasetGroupsCommand).build() {
};
__name(_ListDatasetGroupsCommand, "ListDatasetGroupsCommand");
var ListDatasetGroupsCommand = _ListDatasetGroupsCommand;

// src/commands/ListDatasetImportJobsCommand.ts




var _ListDatasetImportJobsCommand = class _ListDatasetImportJobsCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AmazonPersonalize", "ListDatasetImportJobs", {}).n("PersonalizeClient", "ListDatasetImportJobsCommand").f(void 0, void 0).ser(se_ListDatasetImportJobsCommand).de(de_ListDatasetImportJobsCommand).build() {
};
__name(_ListDatasetImportJobsCommand, "ListDatasetImportJobsCommand");
var ListDatasetImportJobsCommand = _ListDatasetImportJobsCommand;

// src/commands/ListDatasetsCommand.ts




var _ListDatasetsCommand = class _ListDatasetsCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AmazonPersonalize", "ListDatasets", {}).n("PersonalizeClient", "ListDatasetsCommand").f(void 0, void 0).ser(se_ListDatasetsCommand).de(de_ListDatasetsCommand).build() {
};
__name(_ListDatasetsCommand, "ListDatasetsCommand");
var ListDatasetsCommand = _ListDatasetsCommand;

// src/commands/ListEventTrackersCommand.ts




var _ListEventTrackersCommand = class _ListEventTrackersCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AmazonPersonalize", "ListEventTrackers", {}).n("PersonalizeClient", "ListEventTrackersCommand").f(void 0, void 0).ser(se_ListEventTrackersCommand).de(de_ListEventTrackersCommand).build() {
};
__name(_ListEventTrackersCommand, "ListEventTrackersCommand");
var ListEventTrackersCommand = _ListEventTrackersCommand;

// src/commands/ListFiltersCommand.ts




var _ListFiltersCommand = class _ListFiltersCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AmazonPersonalize", "ListFilters", {}).n("PersonalizeClient", "ListFiltersCommand").f(void 0, void 0).ser(se_ListFiltersCommand).de(de_ListFiltersCommand).build() {
};
__name(_ListFiltersCommand, "ListFiltersCommand");
var ListFiltersCommand = _ListFiltersCommand;

// src/commands/ListMetricAttributionMetricsCommand.ts




var _ListMetricAttributionMetricsCommand = class _ListMetricAttributionMetricsCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AmazonPersonalize", "ListMetricAttributionMetrics", {}).n("PersonalizeClient", "ListMetricAttributionMetricsCommand").f(void 0, void 0).ser(se_ListMetricAttributionMetricsCommand).de(de_ListMetricAttributionMetricsCommand).build() {
};
__name(_ListMetricAttributionMetricsCommand, "ListMetricAttributionMetricsCommand");
var ListMetricAttributionMetricsCommand = _ListMetricAttributionMetricsCommand;

// src/commands/ListMetricAttributionsCommand.ts




var _ListMetricAttributionsCommand = class _ListMetricAttributionsCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AmazonPersonalize", "ListMetricAttributions", {}).n("PersonalizeClient", "ListMetricAttributionsCommand").f(void 0, void 0).ser(se_ListMetricAttributionsCommand).de(de_ListMetricAttributionsCommand).build() {
};
__name(_ListMetricAttributionsCommand, "ListMetricAttributionsCommand");
var ListMetricAttributionsCommand = _ListMetricAttributionsCommand;

// src/commands/ListRecipesCommand.ts




var _ListRecipesCommand = class _ListRecipesCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AmazonPersonalize", "ListRecipes", {}).n("PersonalizeClient", "ListRecipesCommand").f(void 0, void 0).ser(se_ListRecipesCommand).de(de_ListRecipesCommand).build() {
};
__name(_ListRecipesCommand, "ListRecipesCommand");
var ListRecipesCommand = _ListRecipesCommand;

// src/commands/ListRecommendersCommand.ts




var _ListRecommendersCommand = class _ListRecommendersCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AmazonPersonalize", "ListRecommenders", {}).n("PersonalizeClient", "ListRecommendersCommand").f(void 0, void 0).ser(se_ListRecommendersCommand).de(de_ListRecommendersCommand).build() {
};
__name(_ListRecommendersCommand, "ListRecommendersCommand");
var ListRecommendersCommand = _ListRecommendersCommand;

// src/commands/ListSchemasCommand.ts




var _ListSchemasCommand = class _ListSchemasCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AmazonPersonalize", "ListSchemas", {}).n("PersonalizeClient", "ListSchemasCommand").f(void 0, void 0).ser(se_ListSchemasCommand).de(de_ListSchemasCommand).build() {
};
__name(_ListSchemasCommand, "ListSchemasCommand");
var ListSchemasCommand = _ListSchemasCommand;

// src/commands/ListSolutionsCommand.ts




var _ListSolutionsCommand = class _ListSolutionsCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AmazonPersonalize", "ListSolutions", {}).n("PersonalizeClient", "ListSolutionsCommand").f(void 0, void 0).ser(se_ListSolutionsCommand).de(de_ListSolutionsCommand).build() {
};
__name(_ListSolutionsCommand, "ListSolutionsCommand");
var ListSolutionsCommand = _ListSolutionsCommand;

// src/commands/ListSolutionVersionsCommand.ts




var _ListSolutionVersionsCommand = class _ListSolutionVersionsCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AmazonPersonalize", "ListSolutionVersions", {}).n("PersonalizeClient", "ListSolutionVersionsCommand").f(void 0, void 0).ser(se_ListSolutionVersionsCommand).de(de_ListSolutionVersionsCommand).build() {
};
__name(_ListSolutionVersionsCommand, "ListSolutionVersionsCommand");
var ListSolutionVersionsCommand = _ListSolutionVersionsCommand;

// src/commands/ListTagsForResourceCommand.ts




var _ListTagsForResourceCommand = class _ListTagsForResourceCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AmazonPersonalize", "ListTagsForResource", {}).n("PersonalizeClient", "ListTagsForResourceCommand").f(void 0, void 0).ser(se_ListTagsForResourceCommand).de(de_ListTagsForResourceCommand).build() {
};
__name(_ListTagsForResourceCommand, "ListTagsForResourceCommand");
var ListTagsForResourceCommand = _ListTagsForResourceCommand;

// src/commands/StartRecommenderCommand.ts




var _StartRecommenderCommand = class _StartRecommenderCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AmazonPersonalize", "StartRecommender", {}).n("PersonalizeClient", "StartRecommenderCommand").f(void 0, void 0).ser(se_StartRecommenderCommand).de(de_StartRecommenderCommand).build() {
};
__name(_StartRecommenderCommand, "StartRecommenderCommand");
var StartRecommenderCommand = _StartRecommenderCommand;

// src/commands/StopRecommenderCommand.ts




var _StopRecommenderCommand = class _StopRecommenderCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AmazonPersonalize", "StopRecommender", {}).n("PersonalizeClient", "StopRecommenderCommand").f(void 0, void 0).ser(se_StopRecommenderCommand).de(de_StopRecommenderCommand).build() {
};
__name(_StopRecommenderCommand, "StopRecommenderCommand");
var StopRecommenderCommand = _StopRecommenderCommand;

// src/commands/StopSolutionVersionCreationCommand.ts




var _StopSolutionVersionCreationCommand = class _StopSolutionVersionCreationCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AmazonPersonalize", "StopSolutionVersionCreation", {}).n("PersonalizeClient", "StopSolutionVersionCreationCommand").f(void 0, void 0).ser(se_StopSolutionVersionCreationCommand).de(de_StopSolutionVersionCreationCommand).build() {
};
__name(_StopSolutionVersionCreationCommand, "StopSolutionVersionCreationCommand");
var StopSolutionVersionCreationCommand = _StopSolutionVersionCreationCommand;

// src/commands/TagResourceCommand.ts




var _TagResourceCommand = class _TagResourceCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AmazonPersonalize", "TagResource", {}).n("PersonalizeClient", "TagResourceCommand").f(void 0, void 0).ser(se_TagResourceCommand).de(de_TagResourceCommand).build() {
};
__name(_TagResourceCommand, "TagResourceCommand");
var TagResourceCommand = _TagResourceCommand;

// src/commands/UntagResourceCommand.ts




var _UntagResourceCommand = class _UntagResourceCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AmazonPersonalize", "UntagResource", {}).n("PersonalizeClient", "UntagResourceCommand").f(void 0, void 0).ser(se_UntagResourceCommand).de(de_UntagResourceCommand).build() {
};
__name(_UntagResourceCommand, "UntagResourceCommand");
var UntagResourceCommand = _UntagResourceCommand;

// src/commands/UpdateCampaignCommand.ts




var _UpdateCampaignCommand = class _UpdateCampaignCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AmazonPersonalize", "UpdateCampaign", {}).n("PersonalizeClient", "UpdateCampaignCommand").f(void 0, void 0).ser(se_UpdateCampaignCommand).de(de_UpdateCampaignCommand).build() {
};
__name(_UpdateCampaignCommand, "UpdateCampaignCommand");
var UpdateCampaignCommand = _UpdateCampaignCommand;

// src/commands/UpdateDatasetCommand.ts




var _UpdateDatasetCommand = class _UpdateDatasetCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AmazonPersonalize", "UpdateDataset", {}).n("PersonalizeClient", "UpdateDatasetCommand").f(void 0, void 0).ser(se_UpdateDatasetCommand).de(de_UpdateDatasetCommand).build() {
};
__name(_UpdateDatasetCommand, "UpdateDatasetCommand");
var UpdateDatasetCommand = _UpdateDatasetCommand;

// src/commands/UpdateMetricAttributionCommand.ts




var _UpdateMetricAttributionCommand = class _UpdateMetricAttributionCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AmazonPersonalize", "UpdateMetricAttribution", {}).n("PersonalizeClient", "UpdateMetricAttributionCommand").f(void 0, void 0).ser(se_UpdateMetricAttributionCommand).de(de_UpdateMetricAttributionCommand).build() {
};
__name(_UpdateMetricAttributionCommand, "UpdateMetricAttributionCommand");
var UpdateMetricAttributionCommand = _UpdateMetricAttributionCommand;

// src/commands/UpdateRecommenderCommand.ts




var _UpdateRecommenderCommand = class _UpdateRecommenderCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AmazonPersonalize", "UpdateRecommender", {}).n("PersonalizeClient", "UpdateRecommenderCommand").f(void 0, void 0).ser(se_UpdateRecommenderCommand).de(de_UpdateRecommenderCommand).build() {
};
__name(_UpdateRecommenderCommand, "UpdateRecommenderCommand");
var UpdateRecommenderCommand = _UpdateRecommenderCommand;

// src/Personalize.ts
var commands = {
  CreateBatchInferenceJobCommand,
  CreateBatchSegmentJobCommand,
  CreateCampaignCommand,
  CreateDatasetCommand,
  CreateDatasetExportJobCommand,
  CreateDatasetGroupCommand,
  CreateDatasetImportJobCommand,
  CreateEventTrackerCommand,
  CreateFilterCommand,
  CreateMetricAttributionCommand,
  CreateRecommenderCommand,
  CreateSchemaCommand,
  CreateSolutionCommand,
  CreateSolutionVersionCommand,
  DeleteCampaignCommand,
  DeleteDatasetCommand,
  DeleteDatasetGroupCommand,
  DeleteEventTrackerCommand,
  DeleteFilterCommand,
  DeleteMetricAttributionCommand,
  DeleteRecommenderCommand,
  DeleteSchemaCommand,
  DeleteSolutionCommand,
  DescribeAlgorithmCommand,
  DescribeBatchInferenceJobCommand,
  DescribeBatchSegmentJobCommand,
  DescribeCampaignCommand,
  DescribeDatasetCommand,
  DescribeDatasetExportJobCommand,
  DescribeDatasetGroupCommand,
  DescribeDatasetImportJobCommand,
  DescribeEventTrackerCommand,
  DescribeFeatureTransformationCommand,
  DescribeFilterCommand,
  DescribeMetricAttributionCommand,
  DescribeRecipeCommand,
  DescribeRecommenderCommand,
  DescribeSchemaCommand,
  DescribeSolutionCommand,
  DescribeSolutionVersionCommand,
  GetSolutionMetricsCommand,
  ListBatchInferenceJobsCommand,
  ListBatchSegmentJobsCommand,
  ListCampaignsCommand,
  ListDatasetExportJobsCommand,
  ListDatasetGroupsCommand,
  ListDatasetImportJobsCommand,
  ListDatasetsCommand,
  ListEventTrackersCommand,
  ListFiltersCommand,
  ListMetricAttributionMetricsCommand,
  ListMetricAttributionsCommand,
  ListRecipesCommand,
  ListRecommendersCommand,
  ListSchemasCommand,
  ListSolutionsCommand,
  ListSolutionVersionsCommand,
  ListTagsForResourceCommand,
  StartRecommenderCommand,
  StopRecommenderCommand,
  StopSolutionVersionCreationCommand,
  TagResourceCommand,
  UntagResourceCommand,
  UpdateCampaignCommand,
  UpdateDatasetCommand,
  UpdateMetricAttributionCommand,
  UpdateRecommenderCommand
};
var _Personalize = class _Personalize extends PersonalizeClient {
};
__name(_Personalize, "Personalize");
var Personalize = _Personalize;
(0, import_smithy_client.createAggregatedClient)(commands, Personalize);

// src/pagination/ListBatchInferenceJobsPaginator.ts

var paginateListBatchInferenceJobs = (0, import_core.createPaginator)(PersonalizeClient, ListBatchInferenceJobsCommand, "nextToken", "nextToken", "maxResults");

// src/pagination/ListBatchSegmentJobsPaginator.ts

var paginateListBatchSegmentJobs = (0, import_core.createPaginator)(PersonalizeClient, ListBatchSegmentJobsCommand, "nextToken", "nextToken", "maxResults");

// src/pagination/ListCampaignsPaginator.ts

var paginateListCampaigns = (0, import_core.createPaginator)(PersonalizeClient, ListCampaignsCommand, "nextToken", "nextToken", "maxResults");

// src/pagination/ListDatasetExportJobsPaginator.ts

var paginateListDatasetExportJobs = (0, import_core.createPaginator)(PersonalizeClient, ListDatasetExportJobsCommand, "nextToken", "nextToken", "maxResults");

// src/pagination/ListDatasetGroupsPaginator.ts

var paginateListDatasetGroups = (0, import_core.createPaginator)(PersonalizeClient, ListDatasetGroupsCommand, "nextToken", "nextToken", "maxResults");

// src/pagination/ListDatasetImportJobsPaginator.ts

var paginateListDatasetImportJobs = (0, import_core.createPaginator)(PersonalizeClient, ListDatasetImportJobsCommand, "nextToken", "nextToken", "maxResults");

// src/pagination/ListDatasetsPaginator.ts

var paginateListDatasets = (0, import_core.createPaginator)(PersonalizeClient, ListDatasetsCommand, "nextToken", "nextToken", "maxResults");

// src/pagination/ListEventTrackersPaginator.ts

var paginateListEventTrackers = (0, import_core.createPaginator)(PersonalizeClient, ListEventTrackersCommand, "nextToken", "nextToken", "maxResults");

// src/pagination/ListFiltersPaginator.ts

var paginateListFilters = (0, import_core.createPaginator)(PersonalizeClient, ListFiltersCommand, "nextToken", "nextToken", "maxResults");

// src/pagination/ListMetricAttributionMetricsPaginator.ts

var paginateListMetricAttributionMetrics = (0, import_core.createPaginator)(PersonalizeClient, ListMetricAttributionMetricsCommand, "nextToken", "nextToken", "maxResults");

// src/pagination/ListMetricAttributionsPaginator.ts

var paginateListMetricAttributions = (0, import_core.createPaginator)(PersonalizeClient, ListMetricAttributionsCommand, "nextToken", "nextToken", "maxResults");

// src/pagination/ListRecipesPaginator.ts

var paginateListRecipes = (0, import_core.createPaginator)(PersonalizeClient, ListRecipesCommand, "nextToken", "nextToken", "maxResults");

// src/pagination/ListRecommendersPaginator.ts

var paginateListRecommenders = (0, import_core.createPaginator)(PersonalizeClient, ListRecommendersCommand, "nextToken", "nextToken", "maxResults");

// src/pagination/ListSchemasPaginator.ts

var paginateListSchemas = (0, import_core.createPaginator)(PersonalizeClient, ListSchemasCommand, "nextToken", "nextToken", "maxResults");

// src/pagination/ListSolutionVersionsPaginator.ts

var paginateListSolutionVersions = (0, import_core.createPaginator)(PersonalizeClient, ListSolutionVersionsCommand, "nextToken", "nextToken", "maxResults");

// src/pagination/ListSolutionsPaginator.ts

var paginateListSolutions = (0, import_core.createPaginator)(PersonalizeClient, ListSolutionsCommand, "nextToken", "nextToken", "maxResults");
// Annotate the CommonJS export names for ESM import in node:

0 && (module.exports = {
  PersonalizeServiceException,
  __Client,
  PersonalizeClient,
  Personalize,
  $Command,
  CreateBatchInferenceJobCommand,
  CreateBatchSegmentJobCommand,
  CreateCampaignCommand,
  CreateDatasetCommand,
  CreateDatasetExportJobCommand,
  CreateDatasetGroupCommand,
  CreateDatasetImportJobCommand,
  CreateEventTrackerCommand,
  CreateFilterCommand,
  CreateMetricAttributionCommand,
  CreateRecommenderCommand,
  CreateSchemaCommand,
  CreateSolutionCommand,
  CreateSolutionVersionCommand,
  DeleteCampaignCommand,
  DeleteDatasetCommand,
  DeleteDatasetGroupCommand,
  DeleteEventTrackerCommand,
  DeleteFilterCommand,
  DeleteMetricAttributionCommand,
  DeleteRecommenderCommand,
  DeleteSchemaCommand,
  DeleteSolutionCommand,
  DescribeAlgorithmCommand,
  DescribeBatchInferenceJobCommand,
  DescribeBatchSegmentJobCommand,
  DescribeCampaignCommand,
  DescribeDatasetCommand,
  DescribeDatasetExportJobCommand,
  DescribeDatasetGroupCommand,
  DescribeDatasetImportJobCommand,
  DescribeEventTrackerCommand,
  DescribeFeatureTransformationCommand,
  DescribeFilterCommand,
  DescribeMetricAttributionCommand,
  DescribeRecipeCommand,
  DescribeRecommenderCommand,
  DescribeSchemaCommand,
  DescribeSolutionCommand,
  DescribeSolutionVersionCommand,
  GetSolutionMetricsCommand,
  ListBatchInferenceJobsCommand,
  ListBatchSegmentJobsCommand,
  ListCampaignsCommand,
  ListDatasetExportJobsCommand,
  ListDatasetGroupsCommand,
  ListDatasetImportJobsCommand,
  ListDatasetsCommand,
  ListEventTrackersCommand,
  ListFiltersCommand,
  ListMetricAttributionMetricsCommand,
  ListMetricAttributionsCommand,
  ListRecipesCommand,
  ListRecommendersCommand,
  ListSchemasCommand,
  ListSolutionVersionsCommand,
  ListSolutionsCommand,
  ListTagsForResourceCommand,
  StartRecommenderCommand,
  StopRecommenderCommand,
  StopSolutionVersionCreationCommand,
  TagResourceCommand,
  UntagResourceCommand,
  UpdateCampaignCommand,
  UpdateDatasetCommand,
  UpdateMetricAttributionCommand,
  UpdateRecommenderCommand,
  paginateListBatchInferenceJobs,
  paginateListBatchSegmentJobs,
  paginateListCampaigns,
  paginateListDatasetExportJobs,
  paginateListDatasetGroups,
  paginateListDatasetImportJobs,
  paginateListDatasets,
  paginateListEventTrackers,
  paginateListFilters,
  paginateListMetricAttributionMetrics,
  paginateListMetricAttributions,
  paginateListRecipes,
  paginateListRecommenders,
  paginateListSchemas,
  paginateListSolutionVersions,
  paginateListSolutions,
  BatchInferenceJobMode,
  InvalidInputException,
  LimitExceededException,
  ResourceAlreadyExistsException,
  ResourceInUseException,
  ResourceNotFoundException,
  TooManyTagsException,
  IngestionMode,
  Domain,
  ImportMode,
  ObjectiveSensitivity,
  TrainingMode,
  TrainingType,
  InvalidNextTokenException,
  RecipeProvider,
  TooManyTagKeysException,
  CreateFilterRequestFilterSensitiveLog,
  FilterFilterSensitiveLog,
  DescribeFilterResponseFilterSensitiveLog
});
