// @azure/arm-communication/dist-esm/src/models/index.js
/*
 * Copyright (c) Microsoft Corporation.
 * Licensed under the MIT License.
 *
 * Code generated by Microsoft (R) AutoRest Code Generator.
 * Changes may cause incorrect behavior and will be lost if the code is regenerated.
 */
/** Known values of {@link Origin} that the service accepts. */
export var KnownOrigin;
(function (KnownOrigin) {
    /** User */
    KnownOrigin["User"] = "user";
    /** System */
    KnownOrigin["System"] = "system";
    /** UserSystem */
    KnownOrigin["UserSystem"] = "user,system";
})(KnownOrigin || (KnownOrigin = {}));
/** Known values of {@link ActionType} that the service accepts. */
export var KnownActionType;
(function (KnownActionType) {
    /** Internal */
    KnownActionType["Internal"] = "Internal";
})(KnownActionType || (KnownActionType = {}));
/** Known values of {@link CheckNameAvailabilityReason} that the service accepts. */
export var KnownCheckNameAvailabilityReason;
(function (KnownCheckNameAvailabilityReason) {
    /** Invalid */
    KnownCheckNameAvailabilityReason["Invalid"] = "Invalid";
    /** AlreadyExists */
    KnownCheckNameAvailabilityReason["AlreadyExists"] = "AlreadyExists";
})(KnownCheckNameAvailabilityReason || (KnownCheckNameAvailabilityReason = {}));
/** Known values of {@link CommunicationServicesProvisioningState} that the service accepts. */
export var KnownCommunicationServicesProvisioningState;
(function (KnownCommunicationServicesProvisioningState) {
    /** Unknown */
    KnownCommunicationServicesProvisioningState["Unknown"] = "Unknown";
    /** Succeeded */
    KnownCommunicationServicesProvisioningState["Succeeded"] = "Succeeded";
    /** Failed */
    KnownCommunicationServicesProvisioningState["Failed"] = "Failed";
    /** Canceled */
    KnownCommunicationServicesProvisioningState["Canceled"] = "Canceled";
    /** Running */
    KnownCommunicationServicesProvisioningState["Running"] = "Running";
    /** Creating */
    KnownCommunicationServicesProvisioningState["Creating"] = "Creating";
    /** Updating */
    KnownCommunicationServicesProvisioningState["Updating"] = "Updating";
    /** Deleting */
    KnownCommunicationServicesProvisioningState["Deleting"] = "Deleting";
    /** Moving */
    KnownCommunicationServicesProvisioningState["Moving"] = "Moving";
})(KnownCommunicationServicesProvisioningState || (KnownCommunicationServicesProvisioningState = {}));
/** Known values of {@link ManagedServiceIdentityType} that the service accepts. */
export var KnownManagedServiceIdentityType;
(function (KnownManagedServiceIdentityType) {
    /** None */
    KnownManagedServiceIdentityType["None"] = "None";
    /** SystemAssigned */
    KnownManagedServiceIdentityType["SystemAssigned"] = "SystemAssigned";
    /** UserAssigned */
    KnownManagedServiceIdentityType["UserAssigned"] = "UserAssigned";
    /** SystemAssignedUserAssigned */
    KnownManagedServiceIdentityType["SystemAssignedUserAssigned"] = "SystemAssigned,UserAssigned";
})(KnownManagedServiceIdentityType || (KnownManagedServiceIdentityType = {}));
/** Known values of {@link CreatedByType} that the service accepts. */
export var KnownCreatedByType;
(function (KnownCreatedByType) {
    /** User */
    KnownCreatedByType["User"] = "User";
    /** Application */
    KnownCreatedByType["Application"] = "Application";
    /** ManagedIdentity */
    KnownCreatedByType["ManagedIdentity"] = "ManagedIdentity";
    /** Key */
    KnownCreatedByType["Key"] = "Key";
})(KnownCreatedByType || (KnownCreatedByType = {}));
/** Known values of {@link DomainsProvisioningState} that the service accepts. */
export var KnownDomainsProvisioningState;
(function (KnownDomainsProvisioningState) {
    /** Unknown */
    KnownDomainsProvisioningState["Unknown"] = "Unknown";
    /** Succeeded */
    KnownDomainsProvisioningState["Succeeded"] = "Succeeded";
    /** Failed */
    KnownDomainsProvisioningState["Failed"] = "Failed";
    /** Canceled */
    KnownDomainsProvisioningState["Canceled"] = "Canceled";
    /** Running */
    KnownDomainsProvisioningState["Running"] = "Running";
    /** Creating */
    KnownDomainsProvisioningState["Creating"] = "Creating";
    /** Updating */
    KnownDomainsProvisioningState["Updating"] = "Updating";
    /** Deleting */
    KnownDomainsProvisioningState["Deleting"] = "Deleting";
    /** Moving */
    KnownDomainsProvisioningState["Moving"] = "Moving";
})(KnownDomainsProvisioningState || (KnownDomainsProvisioningState = {}));
/** Known values of {@link DomainManagement} that the service accepts. */
export var KnownDomainManagement;
(function (KnownDomainManagement) {
    /** AzureManaged */
    KnownDomainManagement["AzureManaged"] = "AzureManaged";
    /** CustomerManaged */
    KnownDomainManagement["CustomerManaged"] = "CustomerManaged";
    /** CustomerManagedInExchangeOnline */
    KnownDomainManagement["CustomerManagedInExchangeOnline"] = "CustomerManagedInExchangeOnline";
})(KnownDomainManagement || (KnownDomainManagement = {}));
/** Known values of {@link VerificationStatus} that the service accepts. */
export var KnownVerificationStatus;
(function (KnownVerificationStatus) {
    /** NotStarted */
    KnownVerificationStatus["NotStarted"] = "NotStarted";
    /** VerificationRequested */
    KnownVerificationStatus["VerificationRequested"] = "VerificationRequested";
    /** VerificationInProgress */
    KnownVerificationStatus["VerificationInProgress"] = "VerificationInProgress";
    /** VerificationFailed */
    KnownVerificationStatus["VerificationFailed"] = "VerificationFailed";
    /** Verified */
    KnownVerificationStatus["Verified"] = "Verified";
    /** CancellationRequested */
    KnownVerificationStatus["CancellationRequested"] = "CancellationRequested";
})(KnownVerificationStatus || (KnownVerificationStatus = {}));
/** Known values of {@link UserEngagementTracking} that the service accepts. */
export var KnownUserEngagementTracking;
(function (KnownUserEngagementTracking) {
    /** Disabled */
    KnownUserEngagementTracking["Disabled"] = "Disabled";
    /** Enabled */
    KnownUserEngagementTracking["Enabled"] = "Enabled";
})(KnownUserEngagementTracking || (KnownUserEngagementTracking = {}));
/** Known values of {@link VerificationType} that the service accepts. */
export var KnownVerificationType;
(function (KnownVerificationType) {
    /** Domain */
    KnownVerificationType["Domain"] = "Domain";
    /** SPF */
    KnownVerificationType["SPF"] = "SPF";
    /** Dkim */
    KnownVerificationType["Dkim"] = "DKIM";
    /** Dkim2 */
    KnownVerificationType["Dkim2"] = "DKIM2";
    /** Dmarc */
    KnownVerificationType["Dmarc"] = "DMARC";
})(KnownVerificationType || (KnownVerificationType = {}));
/** Known values of {@link EmailServicesProvisioningState} that the service accepts. */
export var KnownEmailServicesProvisioningState;
(function (KnownEmailServicesProvisioningState) {
    /** Unknown */
    KnownEmailServicesProvisioningState["Unknown"] = "Unknown";
    /** Succeeded */
    KnownEmailServicesProvisioningState["Succeeded"] = "Succeeded";
    /** Failed */
    KnownEmailServicesProvisioningState["Failed"] = "Failed";
    /** Canceled */
    KnownEmailServicesProvisioningState["Canceled"] = "Canceled";
    /** Running */
    KnownEmailServicesProvisioningState["Running"] = "Running";
    /** Creating */
    KnownEmailServicesProvisioningState["Creating"] = "Creating";
    /** Updating */
    KnownEmailServicesProvisioningState["Updating"] = "Updating";
    /** Deleting */
    KnownEmailServicesProvisioningState["Deleting"] = "Deleting";
    /** Moving */
    KnownEmailServicesProvisioningState["Moving"] = "Moving";
})(KnownEmailServicesProvisioningState || (KnownEmailServicesProvisioningState = {}));
/** Known values of {@link ProvisioningState} that the service accepts. */
export var KnownProvisioningState;
(function (KnownProvisioningState) {
    /** Unknown */
    KnownProvisioningState["Unknown"] = "Unknown";
    /** Succeeded */
    KnownProvisioningState["Succeeded"] = "Succeeded";
    /** Failed */
    KnownProvisioningState["Failed"] = "Failed";
    /** Canceled */
    KnownProvisioningState["Canceled"] = "Canceled";
    /** Running */
    KnownProvisioningState["Running"] = "Running";
    /** Creating */
    KnownProvisioningState["Creating"] = "Creating";
    /** Updating */
    KnownProvisioningState["Updating"] = "Updating";
    /** Deleting */
    KnownProvisioningState["Deleting"] = "Deleting";
    /** Moving */
    KnownProvisioningState["Moving"] = "Moving";
})(KnownProvisioningState || (KnownProvisioningState = {}));
