// @aws-sdk/client-customer-profiles/dist-cjs/index.js
"use strict";
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __name = (target, value) => __defProp(target, "name", { value, configurable: true });
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// src/index.ts
var src_exports = {};
__export(src_exports, {
  AccessDeniedException: () => AccessDeniedException,
  AddProfileKeyCommand: () => AddProfileKeyCommand,
  AddressFilterSensitiveLog: () => AddressFilterSensitiveLog,
  AppflowIntegrationFilterSensitiveLog: () => AppflowIntegrationFilterSensitiveLog,
  AttributeDetailsFilterSensitiveLog: () => AttributeDetailsFilterSensitiveLog,
  AttributeMatchingModel: () => AttributeMatchingModel,
  BadRequestException: () => BadRequestException,
  ConditionsFilterSensitiveLog: () => ConditionsFilterSensitiveLog,
  ConflictResolvingModel: () => ConflictResolvingModel,
  CreateCalculatedAttributeDefinitionCommand: () => CreateCalculatedAttributeDefinitionCommand,
  CreateCalculatedAttributeDefinitionRequestFilterSensitiveLog: () => CreateCalculatedAttributeDefinitionRequestFilterSensitiveLog,
  CreateCalculatedAttributeDefinitionResponseFilterSensitiveLog: () => CreateCalculatedAttributeDefinitionResponseFilterSensitiveLog,
  CreateDomainCommand: () => CreateDomainCommand,
  CreateEventStreamCommand: () => CreateEventStreamCommand,
  CreateIntegrationWorkflowCommand: () => CreateIntegrationWorkflowCommand,
  CreateIntegrationWorkflowRequestFilterSensitiveLog: () => CreateIntegrationWorkflowRequestFilterSensitiveLog,
  CreateProfileCommand: () => CreateProfileCommand,
  CreateProfileRequestFilterSensitiveLog: () => CreateProfileRequestFilterSensitiveLog,
  CustomerProfiles: () => CustomerProfiles,
  CustomerProfilesClient: () => CustomerProfilesClient,
  CustomerProfilesServiceException: () => CustomerProfilesServiceException,
  DataPullMode: () => DataPullMode,
  DeleteCalculatedAttributeDefinitionCommand: () => DeleteCalculatedAttributeDefinitionCommand,
  DeleteDomainCommand: () => DeleteDomainCommand,
  DeleteEventStreamCommand: () => DeleteEventStreamCommand,
  DeleteIntegrationCommand: () => DeleteIntegrationCommand,
  DeleteProfileCommand: () => DeleteProfileCommand,
  DeleteProfileKeyCommand: () => DeleteProfileKeyCommand,
  DeleteProfileObjectCommand: () => DeleteProfileObjectCommand,
  DeleteProfileObjectTypeCommand: () => DeleteProfileObjectTypeCommand,
  DeleteWorkflowCommand: () => DeleteWorkflowCommand,
  DetectProfileObjectTypeCommand: () => DetectProfileObjectTypeCommand,
  DetectProfileObjectTypeRequestFilterSensitiveLog: () => DetectProfileObjectTypeRequestFilterSensitiveLog,
  DetectProfileObjectTypeResponseFilterSensitiveLog: () => DetectProfileObjectTypeResponseFilterSensitiveLog,
  DetectedProfileObjectTypeFilterSensitiveLog: () => DetectedProfileObjectTypeFilterSensitiveLog,
  EventStreamDestinationStatus: () => EventStreamDestinationStatus,
  EventStreamState: () => EventStreamState,
  FieldContentType: () => FieldContentType,
  FlowDefinitionFilterSensitiveLog: () => FlowDefinitionFilterSensitiveLog,
  Gender: () => Gender,
  GetAutoMergingPreviewCommand: () => GetAutoMergingPreviewCommand,
  GetCalculatedAttributeDefinitionCommand: () => GetCalculatedAttributeDefinitionCommand,
  GetCalculatedAttributeDefinitionResponseFilterSensitiveLog: () => GetCalculatedAttributeDefinitionResponseFilterSensitiveLog,
  GetCalculatedAttributeForProfileCommand: () => GetCalculatedAttributeForProfileCommand,
  GetDomainCommand: () => GetDomainCommand,
  GetEventStreamCommand: () => GetEventStreamCommand,
  GetIdentityResolutionJobCommand: () => GetIdentityResolutionJobCommand,
  GetIntegrationCommand: () => GetIntegrationCommand,
  GetMatchesCommand: () => GetMatchesCommand,
  GetProfileObjectTypeCommand: () => GetProfileObjectTypeCommand,
  GetProfileObjectTypeResponseFilterSensitiveLog: () => GetProfileObjectTypeResponseFilterSensitiveLog,
  GetProfileObjectTypeTemplateCommand: () => GetProfileObjectTypeTemplateCommand,
  GetProfileObjectTypeTemplateResponseFilterSensitiveLog: () => GetProfileObjectTypeTemplateResponseFilterSensitiveLog,
  GetSimilarProfilesCommand: () => GetSimilarProfilesCommand,
  GetWorkflowCommand: () => GetWorkflowCommand,
  GetWorkflowStepsCommand: () => GetWorkflowStepsCommand,
  IdentityResolutionJobStatus: () => IdentityResolutionJobStatus,
  IntegrationConfigFilterSensitiveLog: () => IntegrationConfigFilterSensitiveLog,
  InternalServerException: () => InternalServerException,
  JobScheduleDayOfTheWeek: () => JobScheduleDayOfTheWeek,
  ListAccountIntegrationsCommand: () => ListAccountIntegrationsCommand,
  ListCalculatedAttributeDefinitionItemFilterSensitiveLog: () => ListCalculatedAttributeDefinitionItemFilterSensitiveLog,
  ListCalculatedAttributeDefinitionsCommand: () => ListCalculatedAttributeDefinitionsCommand,
  ListCalculatedAttributeDefinitionsResponseFilterSensitiveLog: () => ListCalculatedAttributeDefinitionsResponseFilterSensitiveLog,
  ListCalculatedAttributesForProfileCommand: () => ListCalculatedAttributesForProfileCommand,
  ListDomainsCommand: () => ListDomainsCommand,
  ListEventStreamsCommand: () => ListEventStreamsCommand,
  ListIdentityResolutionJobsCommand: () => ListIdentityResolutionJobsCommand,
  ListIntegrationsCommand: () => ListIntegrationsCommand,
  ListProfileObjectTypeTemplatesCommand: () => ListProfileObjectTypeTemplatesCommand,
  ListProfileObjectTypesCommand: () => ListProfileObjectTypesCommand,
  ListProfileObjectTypesResponseFilterSensitiveLog: () => ListProfileObjectTypesResponseFilterSensitiveLog,
  ListProfileObjectsCommand: () => ListProfileObjectsCommand,
  ListProfileObjectsItemFilterSensitiveLog: () => ListProfileObjectsItemFilterSensitiveLog,
  ListProfileObjectsResponseFilterSensitiveLog: () => ListProfileObjectsResponseFilterSensitiveLog,
  ListRuleBasedMatchesCommand: () => ListRuleBasedMatchesCommand,
  ListTagsForResourceCommand: () => ListTagsForResourceCommand,
  ListWorkflowsCommand: () => ListWorkflowsCommand,
  LogicalOperator: () => LogicalOperator,
  MarketoConnectorOperator: () => MarketoConnectorOperator,
  MatchType: () => MatchType,
  MergeProfilesCommand: () => MergeProfilesCommand,
  Operator: () => Operator,
  OperatorPropertiesKeys: () => OperatorPropertiesKeys,
  PartyType: () => PartyType,
  ProfileFilterSensitiveLog: () => ProfileFilterSensitiveLog,
  PutIntegrationCommand: () => PutIntegrationCommand,
  PutIntegrationRequestFilterSensitiveLog: () => PutIntegrationRequestFilterSensitiveLog,
  PutProfileObjectCommand: () => PutProfileObjectCommand,
  PutProfileObjectRequestFilterSensitiveLog: () => PutProfileObjectRequestFilterSensitiveLog,
  PutProfileObjectTypeCommand: () => PutProfileObjectTypeCommand,
  PutProfileObjectTypeRequestFilterSensitiveLog: () => PutProfileObjectTypeRequestFilterSensitiveLog,
  PutProfileObjectTypeResponseFilterSensitiveLog: () => PutProfileObjectTypeResponseFilterSensitiveLog,
  ResourceNotFoundException: () => ResourceNotFoundException,
  RuleBasedMatchingStatus: () => RuleBasedMatchingStatus,
  S3ConnectorOperator: () => S3ConnectorOperator,
  SalesforceConnectorOperator: () => SalesforceConnectorOperator,
  SearchProfilesCommand: () => SearchProfilesCommand,
  SearchProfilesResponseFilterSensitiveLog: () => SearchProfilesResponseFilterSensitiveLog,
  ServiceNowConnectorOperator: () => ServiceNowConnectorOperator,
  SourceConnectorType: () => SourceConnectorType,
  StandardIdentifier: () => StandardIdentifier,
  Statistic: () => Statistic,
  Status: () => Status,
  TagResourceCommand: () => TagResourceCommand,
  TaskType: () => TaskType,
  ThrottlingException: () => ThrottlingException,
  TriggerType: () => TriggerType,
  Unit: () => Unit,
  UntagResourceCommand: () => UntagResourceCommand,
  UpdateAddressFilterSensitiveLog: () => UpdateAddressFilterSensitiveLog,
  UpdateCalculatedAttributeDefinitionCommand: () => UpdateCalculatedAttributeDefinitionCommand,
  UpdateCalculatedAttributeDefinitionRequestFilterSensitiveLog: () => UpdateCalculatedAttributeDefinitionRequestFilterSensitiveLog,
  UpdateCalculatedAttributeDefinitionResponseFilterSensitiveLog: () => UpdateCalculatedAttributeDefinitionResponseFilterSensitiveLog,
  UpdateDomainCommand: () => UpdateDomainCommand,
  UpdateProfileCommand: () => UpdateProfileCommand,
  UpdateProfileRequestFilterSensitiveLog: () => UpdateProfileRequestFilterSensitiveLog,
  WorkflowType: () => WorkflowType,
  ZendeskConnectorOperator: () => ZendeskConnectorOperator,
  __Client: () => import_smithy_client.Client,
  paginateListEventStreams: () => paginateListEventStreams
});
module.exports = __toCommonJS(src_exports);

// src/CustomerProfilesClient.ts
var import_middleware_host_header = require("@aws-sdk/middleware-host-header");
var import_middleware_logger = require("@aws-sdk/middleware-logger");
var import_middleware_recursion_detection = require("@aws-sdk/middleware-recursion-detection");
var import_middleware_user_agent = require("@aws-sdk/middleware-user-agent");
var import_config_resolver = require("@smithy/config-resolver");
var import_core = require("@smithy/core");
var import_middleware_content_length = require("@smithy/middleware-content-length");
var import_middleware_endpoint = require("@smithy/middleware-endpoint");
var import_middleware_retry = require("@smithy/middleware-retry");

var import_httpAuthSchemeProvider = require("./auth/httpAuthSchemeProvider");

// src/endpoint/EndpointParameters.ts
var resolveClientEndpointParameters = /* @__PURE__ */ __name((options) => {
  return {
    ...options,
    useDualstackEndpoint: options.useDualstackEndpoint ?? false,
    useFipsEndpoint: options.useFipsEndpoint ?? false,
    defaultSigningName: "profile"
  };
}, "resolveClientEndpointParameters");
var commonParams = {
  UseFIPS: { type: "builtInParams", name: "useFipsEndpoint" },
  Endpoint: { type: "builtInParams", name: "endpoint" },
  Region: { type: "builtInParams", name: "region" },
  UseDualStack: { type: "builtInParams", name: "useDualstackEndpoint" }
};

// src/CustomerProfilesClient.ts
var import_runtimeConfig = require("././runtimeConfig");

// src/runtimeExtensions.ts
var import_region_config_resolver = require("@aws-sdk/region-config-resolver");
var import_protocol_http = require("@smithy/protocol-http");
var import_smithy_client = require("@smithy/smithy-client");

// src/auth/httpAuthExtensionConfiguration.ts
var getHttpAuthExtensionConfiguration = /* @__PURE__ */ __name((runtimeConfig) => {
  const _httpAuthSchemes = runtimeConfig.httpAuthSchemes;
  let _httpAuthSchemeProvider = runtimeConfig.httpAuthSchemeProvider;
  let _credentials = runtimeConfig.credentials;
  return {
    setHttpAuthScheme(httpAuthScheme) {
      const index = _httpAuthSchemes.findIndex((scheme) => scheme.schemeId === httpAuthScheme.schemeId);
      if (index === -1) {
        _httpAuthSchemes.push(httpAuthScheme);
      } else {
        _httpAuthSchemes.splice(index, 1, httpAuthScheme);
      }
    },
    httpAuthSchemes() {
      return _httpAuthSchemes;
    },
    setHttpAuthSchemeProvider(httpAuthSchemeProvider) {
      _httpAuthSchemeProvider = httpAuthSchemeProvider;
    },
    httpAuthSchemeProvider() {
      return _httpAuthSchemeProvider;
    },
    setCredentials(credentials) {
      _credentials = credentials;
    },
    credentials() {
      return _credentials;
    }
  };
}, "getHttpAuthExtensionConfiguration");
var resolveHttpAuthRuntimeConfig = /* @__PURE__ */ __name((config) => {
  return {
    httpAuthSchemes: config.httpAuthSchemes(),
    httpAuthSchemeProvider: config.httpAuthSchemeProvider(),
    credentials: config.credentials()
  };
}, "resolveHttpAuthRuntimeConfig");

// src/runtimeExtensions.ts
var asPartial = /* @__PURE__ */ __name((t) => t, "asPartial");
var resolveRuntimeExtensions = /* @__PURE__ */ __name((runtimeConfig, extensions) => {
  const extensionConfiguration = {
    ...asPartial((0, import_region_config_resolver.getAwsRegionExtensionConfiguration)(runtimeConfig)),
    ...asPartial((0, import_smithy_client.getDefaultExtensionConfiguration)(runtimeConfig)),
    ...asPartial((0, import_protocol_http.getHttpHandlerExtensionConfiguration)(runtimeConfig)),
    ...asPartial(getHttpAuthExtensionConfiguration(runtimeConfig))
  };
  extensions.forEach((extension) => extension.configure(extensionConfiguration));
  return {
    ...runtimeConfig,
    ...(0, import_region_config_resolver.resolveAwsRegionExtensionConfiguration)(extensionConfiguration),
    ...(0, import_smithy_client.resolveDefaultRuntimeConfig)(extensionConfiguration),
    ...(0, import_protocol_http.resolveHttpHandlerRuntimeConfig)(extensionConfiguration),
    ...resolveHttpAuthRuntimeConfig(extensionConfiguration)
  };
}, "resolveRuntimeExtensions");

// src/CustomerProfilesClient.ts
var _CustomerProfilesClient = class _CustomerProfilesClient extends import_smithy_client.Client {
  constructor(...[configuration]) {
    const _config_0 = (0, import_runtimeConfig.getRuntimeConfig)(configuration || {});
    const _config_1 = resolveClientEndpointParameters(_config_0);
    const _config_2 = (0, import_config_resolver.resolveRegionConfig)(_config_1);
    const _config_3 = (0, import_middleware_endpoint.resolveEndpointConfig)(_config_2);
    const _config_4 = (0, import_middleware_retry.resolveRetryConfig)(_config_3);
    const _config_5 = (0, import_middleware_host_header.resolveHostHeaderConfig)(_config_4);
    const _config_6 = (0, import_middleware_user_agent.resolveUserAgentConfig)(_config_5);
    const _config_7 = (0, import_httpAuthSchemeProvider.resolveHttpAuthSchemeConfig)(_config_6);
    const _config_8 = resolveRuntimeExtensions(_config_7, (configuration == null ? void 0 : configuration.extensions) || []);
    super(_config_8);
    this.config = _config_8;
    this.middlewareStack.use((0, import_middleware_retry.getRetryPlugin)(this.config));
    this.middlewareStack.use((0, import_middleware_content_length.getContentLengthPlugin)(this.config));
    this.middlewareStack.use((0, import_middleware_host_header.getHostHeaderPlugin)(this.config));
    this.middlewareStack.use((0, import_middleware_logger.getLoggerPlugin)(this.config));
    this.middlewareStack.use((0, import_middleware_recursion_detection.getRecursionDetectionPlugin)(this.config));
    this.middlewareStack.use((0, import_middleware_user_agent.getUserAgentPlugin)(this.config));
    this.middlewareStack.use(
      (0, import_core.getHttpAuthSchemeEndpointRuleSetPlugin)(this.config, {
        httpAuthSchemeParametersProvider: this.getDefaultHttpAuthSchemeParametersProvider(),
        identityProviderConfigProvider: this.getIdentityProviderConfigProvider()
      })
    );
    this.middlewareStack.use((0, import_core.getHttpSigningPlugin)(this.config));
  }
  /**
   * Destroy underlying resources, like sockets. It's usually not necessary to do this.
   * However in Node.js, it's best to explicitly shut down the client's agent when it is no longer needed.
   * Otherwise, sockets might stay open for quite a long time before the server terminates them.
   */
  destroy() {
    super.destroy();
  }
  getDefaultHttpAuthSchemeParametersProvider() {
    return import_httpAuthSchemeProvider.defaultCustomerProfilesHttpAuthSchemeParametersProvider;
  }
  getIdentityProviderConfigProvider() {
    return async (config) => new import_core.DefaultIdentityProviderConfig({
      "aws.auth#sigv4": config.credentials
    });
  }
};
__name(_CustomerProfilesClient, "CustomerProfilesClient");
var CustomerProfilesClient = _CustomerProfilesClient;

// src/CustomerProfiles.ts


// src/commands/AddProfileKeyCommand.ts

var import_middleware_serde = require("@smithy/middleware-serde");

var import_types = require("@smithy/types");

// src/protocols/Aws_restJson1.ts



// src/models/CustomerProfilesServiceException.ts

var _CustomerProfilesServiceException = class _CustomerProfilesServiceException extends import_smithy_client.ServiceException {
  /**
   * @internal
   */
  constructor(options) {
    super(options);
    Object.setPrototypeOf(this, _CustomerProfilesServiceException.prototype);
  }
};
__name(_CustomerProfilesServiceException, "CustomerProfilesServiceException");
var CustomerProfilesServiceException = _CustomerProfilesServiceException;

// src/models/models_0.ts

var _AccessDeniedException = class _AccessDeniedException extends CustomerProfilesServiceException {
  /**
   * @internal
   */
  constructor(opts) {
    super({
      name: "AccessDeniedException",
      $fault: "client",
      ...opts
    });
    this.name = "AccessDeniedException";
    this.$fault = "client";
    Object.setPrototypeOf(this, _AccessDeniedException.prototype);
    this.Message = opts.Message;
  }
};
__name(_AccessDeniedException, "AccessDeniedException");
var AccessDeniedException = _AccessDeniedException;
var _BadRequestException = class _BadRequestException extends CustomerProfilesServiceException {
  /**
   * @internal
   */
  constructor(opts) {
    super({
      name: "BadRequestException",
      $fault: "client",
      ...opts
    });
    this.name = "BadRequestException";
    this.$fault = "client";
    Object.setPrototypeOf(this, _BadRequestException.prototype);
    this.Message = opts.Message;
  }
};
__name(_BadRequestException, "BadRequestException");
var BadRequestException = _BadRequestException;
var _InternalServerException = class _InternalServerException extends CustomerProfilesServiceException {
  /**
   * @internal
   */
  constructor(opts) {
    super({
      name: "InternalServerException",
      $fault: "server",
      ...opts
    });
    this.name = "InternalServerException";
    this.$fault = "server";
    Object.setPrototypeOf(this, _InternalServerException.prototype);
    this.Message = opts.Message;
  }
};
__name(_InternalServerException, "InternalServerException");
var InternalServerException = _InternalServerException;
var _ResourceNotFoundException = class _ResourceNotFoundException extends CustomerProfilesServiceException {
  /**
   * @internal
   */
  constructor(opts) {
    super({
      name: "ResourceNotFoundException",
      $fault: "client",
      ...opts
    });
    this.name = "ResourceNotFoundException";
    this.$fault = "client";
    Object.setPrototypeOf(this, _ResourceNotFoundException.prototype);
    this.Message = opts.Message;
  }
};
__name(_ResourceNotFoundException, "ResourceNotFoundException");
var ResourceNotFoundException = _ResourceNotFoundException;
var _ThrottlingException = class _ThrottlingException extends CustomerProfilesServiceException {
  /**
   * @internal
   */
  constructor(opts) {
    super({
      name: "ThrottlingException",
      $fault: "client",
      ...opts
    });
    this.name = "ThrottlingException";
    this.$fault = "client";
    Object.setPrototypeOf(this, _ThrottlingException.prototype);
    this.Message = opts.Message;
  }
};
__name(_ThrottlingException, "ThrottlingException");
var ThrottlingException = _ThrottlingException;
var SourceConnectorType = {
  MARKETO: "Marketo",
  S3: "S3",
  SALESFORCE: "Salesforce",
  SERVICENOW: "Servicenow",
  ZENDESK: "Zendesk"
};
var MarketoConnectorOperator = {
  ADDITION: "ADDITION",
  BETWEEN: "BETWEEN",
  DIVISION: "DIVISION",
  GREATER_THAN: "GREATER_THAN",
  LESS_THAN: "LESS_THAN",
  MASK_ALL: "MASK_ALL",
  MASK_FIRST_N: "MASK_FIRST_N",
  MASK_LAST_N: "MASK_LAST_N",
  MULTIPLICATION: "MULTIPLICATION",
  NO_OP: "NO_OP",
  PROJECTION: "PROJECTION",
  SUBTRACTION: "SUBTRACTION",
  VALIDATE_NON_NEGATIVE: "VALIDATE_NON_NEGATIVE",
  VALIDATE_NON_NULL: "VALIDATE_NON_NULL",
  VALIDATE_NON_ZERO: "VALIDATE_NON_ZERO",
  VALIDATE_NUMERIC: "VALIDATE_NUMERIC"
};
var S3ConnectorOperator = {
  ADDITION: "ADDITION",
  BETWEEN: "BETWEEN",
  DIVISION: "DIVISION",
  EQUAL_TO: "EQUAL_TO",
  GREATER_THAN: "GREATER_THAN",
  GREATER_THAN_OR_EQUAL_TO: "GREATER_THAN_OR_EQUAL_TO",
  LESS_THAN: "LESS_THAN",
  LESS_THAN_OR_EQUAL_TO: "LESS_THAN_OR_EQUAL_TO",
  MASK_ALL: "MASK_ALL",
  MASK_FIRST_N: "MASK_FIRST_N",
  MASK_LAST_N: "MASK_LAST_N",
  MULTIPLICATION: "MULTIPLICATION",
  NOT_EQUAL_TO: "NOT_EQUAL_TO",
  NO_OP: "NO_OP",
  PROJECTION: "PROJECTION",
  SUBTRACTION: "SUBTRACTION",
  VALIDATE_NON_NEGATIVE: "VALIDATE_NON_NEGATIVE",
  VALIDATE_NON_NULL: "VALIDATE_NON_NULL",
  VALIDATE_NON_ZERO: "VALIDATE_NON_ZERO",
  VALIDATE_NUMERIC: "VALIDATE_NUMERIC"
};
var SalesforceConnectorOperator = {
  ADDITION: "ADDITION",
  BETWEEN: "BETWEEN",
  CONTAINS: "CONTAINS",
  DIVISION: "DIVISION",
  EQUAL_TO: "EQUAL_TO",
  GREATER_THAN: "GREATER_THAN",
  GREATER_THAN_OR_EQUAL_TO: "GREATER_THAN_OR_EQUAL_TO",
  LESS_THAN: "LESS_THAN",
  LESS_THAN_OR_EQUAL_TO: "LESS_THAN_OR_EQUAL_TO",
  MASK_ALL: "MASK_ALL",
  MASK_FIRST_N: "MASK_FIRST_N",
  MASK_LAST_N: "MASK_LAST_N",
  MULTIPLICATION: "MULTIPLICATION",
  NOT_EQUAL_TO: "NOT_EQUAL_TO",
  NO_OP: "NO_OP",
  PROJECTION: "PROJECTION",
  SUBTRACTION: "SUBTRACTION",
  VALIDATE_NON_NEGATIVE: "VALIDATE_NON_NEGATIVE",
  VALIDATE_NON_NULL: "VALIDATE_NON_NULL",
  VALIDATE_NON_ZERO: "VALIDATE_NON_ZERO",
  VALIDATE_NUMERIC: "VALIDATE_NUMERIC"
};
var ServiceNowConnectorOperator = {
  ADDITION: "ADDITION",
  BETWEEN: "BETWEEN",
  CONTAINS: "CONTAINS",
  DIVISION: "DIVISION",
  EQUAL_TO: "EQUAL_TO",
  GREATER_THAN: "GREATER_THAN",
  GREATER_THAN_OR_EQUAL_TO: "GREATER_THAN_OR_EQUAL_TO",
  LESS_THAN: "LESS_THAN",
  LESS_THAN_OR_EQUAL_TO: "LESS_THAN_OR_EQUAL_TO",
  MASK_ALL: "MASK_ALL",
  MASK_FIRST_N: "MASK_FIRST_N",
  MASK_LAST_N: "MASK_LAST_N",
  MULTIPLICATION: "MULTIPLICATION",
  NOT_EQUAL_TO: "NOT_EQUAL_TO",
  NO_OP: "NO_OP",
  PROJECTION: "PROJECTION",
  SUBTRACTION: "SUBTRACTION",
  VALIDATE_NON_NEGATIVE: "VALIDATE_NON_NEGATIVE",
  VALIDATE_NON_NULL: "VALIDATE_NON_NULL",
  VALIDATE_NON_ZERO: "VALIDATE_NON_ZERO",
  VALIDATE_NUMERIC: "VALIDATE_NUMERIC"
};
var ZendeskConnectorOperator = {
  ADDITION: "ADDITION",
  DIVISION: "DIVISION",
  GREATER_THAN: "GREATER_THAN",
  MASK_ALL: "MASK_ALL",
  MASK_FIRST_N: "MASK_FIRST_N",
  MASK_LAST_N: "MASK_LAST_N",
  MULTIPLICATION: "MULTIPLICATION",
  NO_OP: "NO_OP",
  PROJECTION: "PROJECTION",
  SUBTRACTION: "SUBTRACTION",
  VALIDATE_NON_NEGATIVE: "VALIDATE_NON_NEGATIVE",
  VALIDATE_NON_NULL: "VALIDATE_NON_NULL",
  VALIDATE_NON_ZERO: "VALIDATE_NON_ZERO",
  VALIDATE_NUMERIC: "VALIDATE_NUMERIC"
};
var OperatorPropertiesKeys = {
  CONCAT_FORMAT: "CONCAT_FORMAT",
  DATA_TYPE: "DATA_TYPE",
  DESTINATION_DATA_TYPE: "DESTINATION_DATA_TYPE",
  LOWER_BOUND: "LOWER_BOUND",
  MASK_LENGTH: "MASK_LENGTH",
  MASK_VALUE: "MASK_VALUE",
  MATH_OPERATION_FIELDS_ORDER: "MATH_OPERATION_FIELDS_ORDER",
  SOURCE_DATA_TYPE: "SOURCE_DATA_TYPE",
  SUBFIELD_CATEGORY_MAP: "SUBFIELD_CATEGORY_MAP",
  TRUNCATE_LENGTH: "TRUNCATE_LENGTH",
  UPPER_BOUND: "UPPER_BOUND",
  VALIDATION_ACTION: "VALIDATION_ACTION",
  VALUE: "VALUE",
  VALUES: "VALUES"
};
var TaskType = {
  ARITHMETIC: "Arithmetic",
  FILTER: "Filter",
  MAP: "Map",
  MASK: "Mask",
  MERGE: "Merge",
  TRUNCATE: "Truncate",
  VALIDATE: "Validate"
};
var DataPullMode = {
  COMPLETE: "Complete",
  INCREMENTAL: "Incremental"
};
var TriggerType = {
  EVENT: "Event",
  ONDEMAND: "OnDemand",
  SCHEDULED: "Scheduled"
};
var Status = {
  CANCELLED: "CANCELLED",
  COMPLETE: "COMPLETE",
  FAILED: "FAILED",
  IN_PROGRESS: "IN_PROGRESS",
  NOT_STARTED: "NOT_STARTED",
  RETRY: "RETRY",
  SPLIT: "SPLIT"
};
var AttributeMatchingModel = {
  MANY_TO_MANY: "MANY_TO_MANY",
  ONE_TO_ONE: "ONE_TO_ONE"
};
var ConflictResolvingModel = {
  RECENCY: "RECENCY",
  SOURCE: "SOURCE"
};
var Unit = {
  DAYS: "DAYS"
};
var Operator = {
  EQUAL_TO: "EQUAL_TO",
  GREATER_THAN: "GREATER_THAN",
  LESS_THAN: "LESS_THAN",
  NOT_EQUAL_TO: "NOT_EQUAL_TO"
};
var Statistic = {
  AVERAGE: "AVERAGE",
  COUNT: "COUNT",
  FIRST_OCCURRENCE: "FIRST_OCCURRENCE",
  LAST_OCCURRENCE: "LAST_OCCURRENCE",
  MAXIMUM: "MAXIMUM",
  MAX_OCCURRENCE: "MAX_OCCURRENCE",
  MINIMUM: "MINIMUM",
  SUM: "SUM"
};
var JobScheduleDayOfTheWeek = {
  FRIDAY: "FRIDAY",
  MONDAY: "MONDAY",
  SATURDAY: "SATURDAY",
  SUNDAY: "SUNDAY",
  THURSDAY: "THURSDAY",
  TUESDAY: "TUESDAY",
  WEDNESDAY: "WEDNESDAY"
};
var RuleBasedMatchingStatus = {
  ACTIVE: "ACTIVE",
  IN_PROGRESS: "IN_PROGRESS",
  PENDING: "PENDING"
};
var WorkflowType = {
  APPFLOW_INTEGRATION: "APPFLOW_INTEGRATION"
};
var Gender = {
  FEMALE: "FEMALE",
  MALE: "MALE",
  UNSPECIFIED: "UNSPECIFIED"
};
var PartyType = {
  BUSINESS: "BUSINESS",
  INDIVIDUAL: "INDIVIDUAL",
  OTHER: "OTHER"
};
var FieldContentType = {
  EMAIL_ADDRESS: "EMAIL_ADDRESS",
  NAME: "NAME",
  NUMBER: "NUMBER",
  PHONE_NUMBER: "PHONE_NUMBER",
  STRING: "STRING"
};
var StandardIdentifier = {
  ASSET: "ASSET",
  CASE: "CASE",
  LOOKUP_ONLY: "LOOKUP_ONLY",
  NEW_ONLY: "NEW_ONLY",
  ORDER: "ORDER",
  PROFILE: "PROFILE",
  SECONDARY: "SECONDARY",
  UNIQUE: "UNIQUE"
};
var EventStreamDestinationStatus = {
  HEALTHY: "HEALTHY",
  UNHEALTHY: "UNHEALTHY"
};
var EventStreamState = {
  RUNNING: "RUNNING",
  STOPPED: "STOPPED"
};
var IdentityResolutionJobStatus = {
  COMPLETED: "COMPLETED",
  FAILED: "FAILED",
  FIND_MATCHING: "FIND_MATCHING",
  MERGING: "MERGING",
  PARTIAL_SUCCESS: "PARTIAL_SUCCESS",
  PENDING: "PENDING",
  PREPROCESSING: "PREPROCESSING"
};
var MatchType = {
  ML_BASED_MATCHING: "ML_BASED_MATCHING",
  RULE_BASED_MATCHING: "RULE_BASED_MATCHING"
};
var LogicalOperator = {
  AND: "AND",
  OR: "OR"
};
var AddressFilterSensitiveLog = /* @__PURE__ */ __name((obj) => ({
  ...obj
}), "AddressFilterSensitiveLog");
var FlowDefinitionFilterSensitiveLog = /* @__PURE__ */ __name((obj) => ({
  ...obj
}), "FlowDefinitionFilterSensitiveLog");
var AppflowIntegrationFilterSensitiveLog = /* @__PURE__ */ __name((obj) => ({
  ...obj,
  ...obj.FlowDefinition && { FlowDefinition: import_smithy_client.SENSITIVE_STRING }
}), "AppflowIntegrationFilterSensitiveLog");
var AttributeDetailsFilterSensitiveLog = /* @__PURE__ */ __name((obj) => ({
  ...obj
}), "AttributeDetailsFilterSensitiveLog");
var ListCalculatedAttributeDefinitionItemFilterSensitiveLog = /* @__PURE__ */ __name((obj) => ({
  ...obj,
  ...obj.Description && { Description: import_smithy_client.SENSITIVE_STRING }
}), "ListCalculatedAttributeDefinitionItemFilterSensitiveLog");
var ConditionsFilterSensitiveLog = /* @__PURE__ */ __name((obj) => ({
  ...obj
}), "ConditionsFilterSensitiveLog");
var CreateCalculatedAttributeDefinitionRequestFilterSensitiveLog = /* @__PURE__ */ __name((obj) => ({
  ...obj,
  ...obj.Description && { Description: import_smithy_client.SENSITIVE_STRING },
  ...obj.AttributeDetails && { AttributeDetails: import_smithy_client.SENSITIVE_STRING },
  ...obj.Conditions && { Conditions: import_smithy_client.SENSITIVE_STRING },
  ...obj.Statistic && { Statistic: import_smithy_client.SENSITIVE_STRING }
}), "CreateCalculatedAttributeDefinitionRequestFilterSensitiveLog");
var CreateCalculatedAttributeDefinitionResponseFilterSensitiveLog = /* @__PURE__ */ __name((obj) => ({
  ...obj,
  ...obj.Description && { Description: import_smithy_client.SENSITIVE_STRING },
  ...obj.AttributeDetails && { AttributeDetails: import_smithy_client.SENSITIVE_STRING },
  ...obj.Conditions && { Conditions: import_smithy_client.SENSITIVE_STRING },
  ...obj.Statistic && { Statistic: import_smithy_client.SENSITIVE_STRING }
}), "CreateCalculatedAttributeDefinitionResponseFilterSensitiveLog");
var IntegrationConfigFilterSensitiveLog = /* @__PURE__ */ __name((obj) => ({
  ...obj,
  ...obj.AppflowIntegration && { AppflowIntegration: AppflowIntegrationFilterSensitiveLog(obj.AppflowIntegration) }
}), "IntegrationConfigFilterSensitiveLog");
var CreateIntegrationWorkflowRequestFilterSensitiveLog = /* @__PURE__ */ __name((obj) => ({
  ...obj,
  ...obj.IntegrationConfig && { IntegrationConfig: IntegrationConfigFilterSensitiveLog(obj.IntegrationConfig) }
}), "CreateIntegrationWorkflowRequestFilterSensitiveLog");
var CreateProfileRequestFilterSensitiveLog = /* @__PURE__ */ __name((obj) => ({
  ...obj,
  ...obj.AccountNumber && { AccountNumber: import_smithy_client.SENSITIVE_STRING },
  ...obj.AdditionalInformation && { AdditionalInformation: import_smithy_client.SENSITIVE_STRING },
  ...obj.PartyType && { PartyType: import_smithy_client.SENSITIVE_STRING },
  ...obj.BusinessName && { BusinessName: import_smithy_client.SENSITIVE_STRING },
  ...obj.FirstName && { FirstName: import_smithy_client.SENSITIVE_STRING },
  ...obj.MiddleName && { MiddleName: import_smithy_client.SENSITIVE_STRING },
  ...obj.LastName && { LastName: import_smithy_client.SENSITIVE_STRING },
  ...obj.BirthDate && { BirthDate: import_smithy_client.SENSITIVE_STRING },
  ...obj.Gender && { Gender: import_smithy_client.SENSITIVE_STRING },
  ...obj.PhoneNumber && { PhoneNumber: import_smithy_client.SENSITIVE_STRING },
  ...obj.MobilePhoneNumber && { MobilePhoneNumber: import_smithy_client.SENSITIVE_STRING },
  ...obj.HomePhoneNumber && { HomePhoneNumber: import_smithy_client.SENSITIVE_STRING },
  ...obj.BusinessPhoneNumber && { BusinessPhoneNumber: import_smithy_client.SENSITIVE_STRING },
  ...obj.EmailAddress && { EmailAddress: import_smithy_client.SENSITIVE_STRING },
  ...obj.PersonalEmailAddress && { PersonalEmailAddress: import_smithy_client.SENSITIVE_STRING },
  ...obj.BusinessEmailAddress && { BusinessEmailAddress: import_smithy_client.SENSITIVE_STRING },
  ...obj.Address && { Address: import_smithy_client.SENSITIVE_STRING },
  ...obj.ShippingAddress && { ShippingAddress: import_smithy_client.SENSITIVE_STRING },
  ...obj.MailingAddress && { MailingAddress: import_smithy_client.SENSITIVE_STRING },
  ...obj.BillingAddress && { BillingAddress: import_smithy_client.SENSITIVE_STRING },
  ...obj.Attributes && { Attributes: import_smithy_client.SENSITIVE_STRING },
  ...obj.PartyTypeString && { PartyTypeString: import_smithy_client.SENSITIVE_STRING },
  ...obj.GenderString && { GenderString: import_smithy_client.SENSITIVE_STRING }
}), "CreateProfileRequestFilterSensitiveLog");
var DetectProfileObjectTypeRequestFilterSensitiveLog = /* @__PURE__ */ __name((obj) => ({
  ...obj,
  ...obj.Objects && { Objects: import_smithy_client.SENSITIVE_STRING }
}), "DetectProfileObjectTypeRequestFilterSensitiveLog");
var DetectedProfileObjectTypeFilterSensitiveLog = /* @__PURE__ */ __name((obj) => ({
  ...obj,
  ...obj.Fields && { Fields: import_smithy_client.SENSITIVE_STRING },
  ...obj.Keys && { Keys: import_smithy_client.SENSITIVE_STRING }
}), "DetectedProfileObjectTypeFilterSensitiveLog");
var DetectProfileObjectTypeResponseFilterSensitiveLog = /* @__PURE__ */ __name((obj) => ({
  ...obj,
  ...obj.DetectedProfileObjectTypes && {
    DetectedProfileObjectTypes: obj.DetectedProfileObjectTypes.map(
      (item) => DetectedProfileObjectTypeFilterSensitiveLog(item)
    )
  }
}), "DetectProfileObjectTypeResponseFilterSensitiveLog");
var GetCalculatedAttributeDefinitionResponseFilterSensitiveLog = /* @__PURE__ */ __name((obj) => ({
  ...obj,
  ...obj.Description && { Description: import_smithy_client.SENSITIVE_STRING },
  ...obj.Statistic && { Statistic: import_smithy_client.SENSITIVE_STRING },
  ...obj.Conditions && { Conditions: import_smithy_client.SENSITIVE_STRING },
  ...obj.AttributeDetails && { AttributeDetails: import_smithy_client.SENSITIVE_STRING }
}), "GetCalculatedAttributeDefinitionResponseFilterSensitiveLog");
var GetProfileObjectTypeResponseFilterSensitiveLog = /* @__PURE__ */ __name((obj) => ({
  ...obj,
  ...obj.Description && { Description: import_smithy_client.SENSITIVE_STRING },
  ...obj.Fields && { Fields: import_smithy_client.SENSITIVE_STRING },
  ...obj.Keys && { Keys: import_smithy_client.SENSITIVE_STRING }
}), "GetProfileObjectTypeResponseFilterSensitiveLog");
var GetProfileObjectTypeTemplateResponseFilterSensitiveLog = /* @__PURE__ */ __name((obj) => ({
  ...obj,
  ...obj.Fields && { Fields: import_smithy_client.SENSITIVE_STRING },
  ...obj.Keys && { Keys: import_smithy_client.SENSITIVE_STRING }
}), "GetProfileObjectTypeTemplateResponseFilterSensitiveLog");
var ListCalculatedAttributeDefinitionsResponseFilterSensitiveLog = /* @__PURE__ */ __name((obj) => ({
  ...obj,
  ...obj.Items && { Items: import_smithy_client.SENSITIVE_STRING }
}), "ListCalculatedAttributeDefinitionsResponseFilterSensitiveLog");
var ListProfileObjectsItemFilterSensitiveLog = /* @__PURE__ */ __name((obj) => ({
  ...obj,
  ...obj.Object && { Object: import_smithy_client.SENSITIVE_STRING }
}), "ListProfileObjectsItemFilterSensitiveLog");
var ListProfileObjectsResponseFilterSensitiveLog = /* @__PURE__ */ __name((obj) => ({
  ...obj,
  ...obj.Items && { Items: obj.Items.map((item) => ListProfileObjectsItemFilterSensitiveLog(item)) }
}), "ListProfileObjectsResponseFilterSensitiveLog");
var ListProfileObjectTypesResponseFilterSensitiveLog = /* @__PURE__ */ __name((obj) => ({
  ...obj,
  ...obj.Items && { Items: import_smithy_client.SENSITIVE_STRING }
}), "ListProfileObjectTypesResponseFilterSensitiveLog");
var PutIntegrationRequestFilterSensitiveLog = /* @__PURE__ */ __name((obj) => ({
  ...obj,
  ...obj.FlowDefinition && { FlowDefinition: import_smithy_client.SENSITIVE_STRING }
}), "PutIntegrationRequestFilterSensitiveLog");
var PutProfileObjectRequestFilterSensitiveLog = /* @__PURE__ */ __name((obj) => ({
  ...obj,
  ...obj.Object && { Object: import_smithy_client.SENSITIVE_STRING }
}), "PutProfileObjectRequestFilterSensitiveLog");
var PutProfileObjectTypeRequestFilterSensitiveLog = /* @__PURE__ */ __name((obj) => ({
  ...obj,
  ...obj.Description && { Description: import_smithy_client.SENSITIVE_STRING },
  ...obj.Fields && { Fields: import_smithy_client.SENSITIVE_STRING },
  ...obj.Keys && { Keys: import_smithy_client.SENSITIVE_STRING }
}), "PutProfileObjectTypeRequestFilterSensitiveLog");
var PutProfileObjectTypeResponseFilterSensitiveLog = /* @__PURE__ */ __name((obj) => ({
  ...obj,
  ...obj.Description && { Description: import_smithy_client.SENSITIVE_STRING },
  ...obj.Fields && { Fields: import_smithy_client.SENSITIVE_STRING },
  ...obj.Keys && { Keys: import_smithy_client.SENSITIVE_STRING }
}), "PutProfileObjectTypeResponseFilterSensitiveLog");
var ProfileFilterSensitiveLog = /* @__PURE__ */ __name((obj) => ({
  ...obj,
  ...obj.AccountNumber && { AccountNumber: import_smithy_client.SENSITIVE_STRING },
  ...obj.AdditionalInformation && { AdditionalInformation: import_smithy_client.SENSITIVE_STRING },
  ...obj.PartyType && { PartyType: import_smithy_client.SENSITIVE_STRING },
  ...obj.BusinessName && { BusinessName: import_smithy_client.SENSITIVE_STRING },
  ...obj.FirstName && { FirstName: import_smithy_client.SENSITIVE_STRING },
  ...obj.MiddleName && { MiddleName: import_smithy_client.SENSITIVE_STRING },
  ...obj.LastName && { LastName: import_smithy_client.SENSITIVE_STRING },
  ...obj.BirthDate && { BirthDate: import_smithy_client.SENSITIVE_STRING },
  ...obj.Gender && { Gender: import_smithy_client.SENSITIVE_STRING },
  ...obj.PhoneNumber && { PhoneNumber: import_smithy_client.SENSITIVE_STRING },
  ...obj.MobilePhoneNumber && { MobilePhoneNumber: import_smithy_client.SENSITIVE_STRING },
  ...obj.HomePhoneNumber && { HomePhoneNumber: import_smithy_client.SENSITIVE_STRING },
  ...obj.BusinessPhoneNumber && { BusinessPhoneNumber: import_smithy_client.SENSITIVE_STRING },
  ...obj.EmailAddress && { EmailAddress: import_smithy_client.SENSITIVE_STRING },
  ...obj.PersonalEmailAddress && { PersonalEmailAddress: import_smithy_client.SENSITIVE_STRING },
  ...obj.BusinessEmailAddress && { BusinessEmailAddress: import_smithy_client.SENSITIVE_STRING },
  ...obj.Address && { Address: import_smithy_client.SENSITIVE_STRING },
  ...obj.ShippingAddress && { ShippingAddress: import_smithy_client.SENSITIVE_STRING },
  ...obj.MailingAddress && { MailingAddress: import_smithy_client.SENSITIVE_STRING },
  ...obj.BillingAddress && { BillingAddress: import_smithy_client.SENSITIVE_STRING },
  ...obj.Attributes && { Attributes: import_smithy_client.SENSITIVE_STRING },
  ...obj.PartyTypeString && { PartyTypeString: import_smithy_client.SENSITIVE_STRING },
  ...obj.GenderString && { GenderString: import_smithy_client.SENSITIVE_STRING }
}), "ProfileFilterSensitiveLog");
var SearchProfilesResponseFilterSensitiveLog = /* @__PURE__ */ __name((obj) => ({
  ...obj,
  ...obj.Items && { Items: obj.Items.map((item) => ProfileFilterSensitiveLog(item)) }
}), "SearchProfilesResponseFilterSensitiveLog");
var UpdateCalculatedAttributeDefinitionRequestFilterSensitiveLog = /* @__PURE__ */ __name((obj) => ({
  ...obj,
  ...obj.Description && { Description: import_smithy_client.SENSITIVE_STRING },
  ...obj.Conditions && { Conditions: import_smithy_client.SENSITIVE_STRING }
}), "UpdateCalculatedAttributeDefinitionRequestFilterSensitiveLog");
var UpdateCalculatedAttributeDefinitionResponseFilterSensitiveLog = /* @__PURE__ */ __name((obj) => ({
  ...obj,
  ...obj.Description && { Description: import_smithy_client.SENSITIVE_STRING },
  ...obj.Statistic && { Statistic: import_smithy_client.SENSITIVE_STRING },
  ...obj.Conditions && { Conditions: import_smithy_client.SENSITIVE_STRING },
  ...obj.AttributeDetails && { AttributeDetails: import_smithy_client.SENSITIVE_STRING }
}), "UpdateCalculatedAttributeDefinitionResponseFilterSensitiveLog");
var UpdateAddressFilterSensitiveLog = /* @__PURE__ */ __name((obj) => ({
  ...obj
}), "UpdateAddressFilterSensitiveLog");
var UpdateProfileRequestFilterSensitiveLog = /* @__PURE__ */ __name((obj) => ({
  ...obj,
  ...obj.AdditionalInformation && { AdditionalInformation: import_smithy_client.SENSITIVE_STRING },
  ...obj.AccountNumber && { AccountNumber: import_smithy_client.SENSITIVE_STRING },
  ...obj.PartyType && { PartyType: import_smithy_client.SENSITIVE_STRING },
  ...obj.BusinessName && { BusinessName: import_smithy_client.SENSITIVE_STRING },
  ...obj.FirstName && { FirstName: import_smithy_client.SENSITIVE_STRING },
  ...obj.MiddleName && { MiddleName: import_smithy_client.SENSITIVE_STRING },
  ...obj.LastName && { LastName: import_smithy_client.SENSITIVE_STRING },
  ...obj.BirthDate && { BirthDate: import_smithy_client.SENSITIVE_STRING },
  ...obj.Gender && { Gender: import_smithy_client.SENSITIVE_STRING },
  ...obj.PhoneNumber && { PhoneNumber: import_smithy_client.SENSITIVE_STRING },
  ...obj.MobilePhoneNumber && { MobilePhoneNumber: import_smithy_client.SENSITIVE_STRING },
  ...obj.HomePhoneNumber && { HomePhoneNumber: import_smithy_client.SENSITIVE_STRING },
  ...obj.BusinessPhoneNumber && { BusinessPhoneNumber: import_smithy_client.SENSITIVE_STRING },
  ...obj.EmailAddress && { EmailAddress: import_smithy_client.SENSITIVE_STRING },
  ...obj.PersonalEmailAddress && { PersonalEmailAddress: import_smithy_client.SENSITIVE_STRING },
  ...obj.BusinessEmailAddress && { BusinessEmailAddress: import_smithy_client.SENSITIVE_STRING },
  ...obj.Address && { Address: import_smithy_client.SENSITIVE_STRING },
  ...obj.ShippingAddress && { ShippingAddress: import_smithy_client.SENSITIVE_STRING },
  ...obj.MailingAddress && { MailingAddress: import_smithy_client.SENSITIVE_STRING },
  ...obj.BillingAddress && { BillingAddress: import_smithy_client.SENSITIVE_STRING },
  ...obj.Attributes && { Attributes: import_smithy_client.SENSITIVE_STRING },
  ...obj.PartyTypeString && { PartyTypeString: import_smithy_client.SENSITIVE_STRING },
  ...obj.GenderString && { GenderString: import_smithy_client.SENSITIVE_STRING }
}), "UpdateProfileRequestFilterSensitiveLog");

// src/protocols/Aws_restJson1.ts
var se_AddProfileKeyCommand = /* @__PURE__ */ __name(async (input, context) => {
  const b = (0, import_core.requestBuilder)(input, context);
  const headers = {
    "content-type": "application/json"
  };
  b.bp("/domains/{DomainName}/profiles/keys");
  b.p("DomainName", () => input.DomainName, "{DomainName}", false);
  let body;
  body = JSON.stringify(
    (0, import_smithy_client.take)(input, {
      KeyName: [],
      ProfileId: [],
      Values: (_) => (0, import_smithy_client._json)(_)
    })
  );
  b.m("POST").h(headers).b(body);
  return b.build();
}, "se_AddProfileKeyCommand");
var se_CreateCalculatedAttributeDefinitionCommand = /* @__PURE__ */ __name(async (input, context) => {
  const b = (0, import_core.requestBuilder)(input, context);
  const headers = {
    "content-type": "application/json"
  };
  b.bp("/domains/{DomainName}/calculated-attributes/{CalculatedAttributeName}");
  b.p("DomainName", () => input.DomainName, "{DomainName}", false);
  b.p("CalculatedAttributeName", () => input.CalculatedAttributeName, "{CalculatedAttributeName}", false);
  let body;
  body = JSON.stringify(
    (0, import_smithy_client.take)(input, {
      AttributeDetails: (_) => (0, import_smithy_client._json)(_),
      Conditions: (_) => (0, import_smithy_client._json)(_),
      Description: [],
      DisplayName: [],
      Statistic: [],
      Tags: (_) => (0, import_smithy_client._json)(_)
    })
  );
  b.m("POST").h(headers).b(body);
  return b.build();
}, "se_CreateCalculatedAttributeDefinitionCommand");
var se_CreateDomainCommand = /* @__PURE__ */ __name(async (input, context) => {
  const b = (0, import_core.requestBuilder)(input, context);
  const headers = {
    "content-type": "application/json"
  };
  b.bp("/domains/{DomainName}");
  b.p("DomainName", () => input.DomainName, "{DomainName}", false);
  let body;
  body = JSON.stringify(
    (0, import_smithy_client.take)(input, {
      DeadLetterQueueUrl: [],
      DefaultEncryptionKey: [],
      DefaultExpirationDays: [],
      Matching: (_) => se_MatchingRequest(_, context),
      RuleBasedMatching: (_) => (0, import_smithy_client._json)(_),
      Tags: (_) => (0, import_smithy_client._json)(_)
    })
  );
  b.m("POST").h(headers).b(body);
  return b.build();
}, "se_CreateDomainCommand");
var se_CreateEventStreamCommand = /* @__PURE__ */ __name(async (input, context) => {
  const b = (0, import_core.requestBuilder)(input, context);
  const headers = {
    "content-type": "application/json"
  };
  b.bp("/domains/{DomainName}/event-streams/{EventStreamName}");
  b.p("DomainName", () => input.DomainName, "{DomainName}", false);
  b.p("EventStreamName", () => input.EventStreamName, "{EventStreamName}", false);
  let body;
  body = JSON.stringify(
    (0, import_smithy_client.take)(input, {
      Tags: (_) => (0, import_smithy_client._json)(_),
      Uri: []
    })
  );
  b.m("POST").h(headers).b(body);
  return b.build();
}, "se_CreateEventStreamCommand");
var se_CreateIntegrationWorkflowCommand = /* @__PURE__ */ __name(async (input, context) => {
  const b = (0, import_core.requestBuilder)(input, context);
  const headers = {
    "content-type": "application/json"
  };
  b.bp("/domains/{DomainName}/workflows/integrations");
  b.p("DomainName", () => input.DomainName, "{DomainName}", false);
  let body;
  body = JSON.stringify(
    (0, import_smithy_client.take)(input, {
      IntegrationConfig: (_) => se_IntegrationConfig(_, context),
      ObjectTypeName: [],
      RoleArn: [],
      Tags: (_) => (0, import_smithy_client._json)(_),
      WorkflowType: []
    })
  );
  b.m("POST").h(headers).b(body);
  return b.build();
}, "se_CreateIntegrationWorkflowCommand");
var se_CreateProfileCommand = /* @__PURE__ */ __name(async (input, context) => {
  const b = (0, import_core.requestBuilder)(input, context);
  const headers = {
    "content-type": "application/json"
  };
  b.bp("/domains/{DomainName}/profiles");
  b.p("DomainName", () => input.DomainName, "{DomainName}", false);
  let body;
  body = JSON.stringify(
    (0, import_smithy_client.take)(input, {
      AccountNumber: [],
      AdditionalInformation: [],
      Address: (_) => (0, import_smithy_client._json)(_),
      Attributes: (_) => (0, import_smithy_client._json)(_),
      BillingAddress: (_) => (0, import_smithy_client._json)(_),
      BirthDate: [],
      BusinessEmailAddress: [],
      BusinessName: [],
      BusinessPhoneNumber: [],
      EmailAddress: [],
      FirstName: [],
      Gender: [],
      GenderString: [],
      HomePhoneNumber: [],
      LastName: [],
      MailingAddress: (_) => (0, import_smithy_client._json)(_),
      MiddleName: [],
      MobilePhoneNumber: [],
      PartyType: [],
      PartyTypeString: [],
      PersonalEmailAddress: [],
      PhoneNumber: [],
      ShippingAddress: (_) => (0, import_smithy_client._json)(_)
    })
  );
  b.m("POST").h(headers).b(body);
  return b.build();
}, "se_CreateProfileCommand");
var se_DeleteCalculatedAttributeDefinitionCommand = /* @__PURE__ */ __name(async (input, context) => {
  const b = (0, import_core.requestBuilder)(input, context);
  const headers = {};
  b.bp("/domains/{DomainName}/calculated-attributes/{CalculatedAttributeName}");
  b.p("DomainName", () => input.DomainName, "{DomainName}", false);
  b.p("CalculatedAttributeName", () => input.CalculatedAttributeName, "{CalculatedAttributeName}", false);
  let body;
  b.m("DELETE").h(headers).b(body);
  return b.build();
}, "se_DeleteCalculatedAttributeDefinitionCommand");
var se_DeleteDomainCommand = /* @__PURE__ */ __name(async (input, context) => {
  const b = (0, import_core.requestBuilder)(input, context);
  const headers = {};
  b.bp("/domains/{DomainName}");
  b.p("DomainName", () => input.DomainName, "{DomainName}", false);
  let body;
  b.m("DELETE").h(headers).b(body);
  return b.build();
}, "se_DeleteDomainCommand");
var se_DeleteEventStreamCommand = /* @__PURE__ */ __name(async (input, context) => {
  const b = (0, import_core.requestBuilder)(input, context);
  const headers = {};
  b.bp("/domains/{DomainName}/event-streams/{EventStreamName}");
  b.p("DomainName", () => input.DomainName, "{DomainName}", false);
  b.p("EventStreamName", () => input.EventStreamName, "{EventStreamName}", false);
  let body;
  b.m("DELETE").h(headers).b(body);
  return b.build();
}, "se_DeleteEventStreamCommand");
var se_DeleteIntegrationCommand = /* @__PURE__ */ __name(async (input, context) => {
  const b = (0, import_core.requestBuilder)(input, context);
  const headers = {
    "content-type": "application/json"
  };
  b.bp("/domains/{DomainName}/integrations/delete");
  b.p("DomainName", () => input.DomainName, "{DomainName}", false);
  let body;
  body = JSON.stringify(
    (0, import_smithy_client.take)(input, {
      Uri: []
    })
  );
  b.m("POST").h(headers).b(body);
  return b.build();
}, "se_DeleteIntegrationCommand");
var se_DeleteProfileCommand = /* @__PURE__ */ __name(async (input, context) => {
  const b = (0, import_core.requestBuilder)(input, context);
  const headers = {
    "content-type": "application/json"
  };
  b.bp("/domains/{DomainName}/profiles/delete");
  b.p("DomainName", () => input.DomainName, "{DomainName}", false);
  let body;
  body = JSON.stringify(
    (0, import_smithy_client.take)(input, {
      ProfileId: []
    })
  );
  b.m("POST").h(headers).b(body);
  return b.build();
}, "se_DeleteProfileCommand");
var se_DeleteProfileKeyCommand = /* @__PURE__ */ __name(async (input, context) => {
  const b = (0, import_core.requestBuilder)(input, context);
  const headers = {
    "content-type": "application/json"
  };
  b.bp("/domains/{DomainName}/profiles/keys/delete");
  b.p("DomainName", () => input.DomainName, "{DomainName}", false);
  let body;
  body = JSON.stringify(
    (0, import_smithy_client.take)(input, {
      KeyName: [],
      ProfileId: [],
      Values: (_) => (0, import_smithy_client._json)(_)
    })
  );
  b.m("POST").h(headers).b(body);
  return b.build();
}, "se_DeleteProfileKeyCommand");
var se_DeleteProfileObjectCommand = /* @__PURE__ */ __name(async (input, context) => {
  const b = (0, import_core.requestBuilder)(input, context);
  const headers = {
    "content-type": "application/json"
  };
  b.bp("/domains/{DomainName}/profiles/objects/delete");
  b.p("DomainName", () => input.DomainName, "{DomainName}", false);
  let body;
  body = JSON.stringify(
    (0, import_smithy_client.take)(input, {
      ObjectTypeName: [],
      ProfileId: [],
      ProfileObjectUniqueKey: []
    })
  );
  b.m("POST").h(headers).b(body);
  return b.build();
}, "se_DeleteProfileObjectCommand");
var se_DeleteProfileObjectTypeCommand = /* @__PURE__ */ __name(async (input, context) => {
  const b = (0, import_core.requestBuilder)(input, context);
  const headers = {};
  b.bp("/domains/{DomainName}/object-types/{ObjectTypeName}");
  b.p("DomainName", () => input.DomainName, "{DomainName}", false);
  b.p("ObjectTypeName", () => input.ObjectTypeName, "{ObjectTypeName}", false);
  let body;
  b.m("DELETE").h(headers).b(body);
  return b.build();
}, "se_DeleteProfileObjectTypeCommand");
var se_DeleteWorkflowCommand = /* @__PURE__ */ __name(async (input, context) => {
  const b = (0, import_core.requestBuilder)(input, context);
  const headers = {};
  b.bp("/domains/{DomainName}/workflows/{WorkflowId}");
  b.p("DomainName", () => input.DomainName, "{DomainName}", false);
  b.p("WorkflowId", () => input.WorkflowId, "{WorkflowId}", false);
  let body;
  b.m("DELETE").h(headers).b(body);
  return b.build();
}, "se_DeleteWorkflowCommand");
var se_DetectProfileObjectTypeCommand = /* @__PURE__ */ __name(async (input, context) => {
  const b = (0, import_core.requestBuilder)(input, context);
  const headers = {
    "content-type": "application/json"
  };
  b.bp("/domains/{DomainName}/detect/object-types");
  b.p("DomainName", () => input.DomainName, "{DomainName}", false);
  let body;
  body = JSON.stringify(
    (0, import_smithy_client.take)(input, {
      Objects: (_) => (0, import_smithy_client._json)(_)
    })
  );
  b.m("POST").h(headers).b(body);
  return b.build();
}, "se_DetectProfileObjectTypeCommand");
var se_GetAutoMergingPreviewCommand = /* @__PURE__ */ __name(async (input, context) => {
  const b = (0, import_core.requestBuilder)(input, context);
  const headers = {
    "content-type": "application/json"
  };
  b.bp("/domains/{DomainName}/identity-resolution-jobs/auto-merging-preview");
  b.p("DomainName", () => input.DomainName, "{DomainName}", false);
  let body;
  body = JSON.stringify(
    (0, import_smithy_client.take)(input, {
      ConflictResolution: (_) => (0, import_smithy_client._json)(_),
      Consolidation: (_) => (0, import_smithy_client._json)(_),
      MinAllowedConfidenceScoreForMerging: (_) => (0, import_smithy_client.serializeFloat)(_)
    })
  );
  b.m("POST").h(headers).b(body);
  return b.build();
}, "se_GetAutoMergingPreviewCommand");
var se_GetCalculatedAttributeDefinitionCommand = /* @__PURE__ */ __name(async (input, context) => {
  const b = (0, import_core.requestBuilder)(input, context);
  const headers = {};
  b.bp("/domains/{DomainName}/calculated-attributes/{CalculatedAttributeName}");
  b.p("DomainName", () => input.DomainName, "{DomainName}", false);
  b.p("CalculatedAttributeName", () => input.CalculatedAttributeName, "{CalculatedAttributeName}", false);
  let body;
  b.m("GET").h(headers).b(body);
  return b.build();
}, "se_GetCalculatedAttributeDefinitionCommand");
var se_GetCalculatedAttributeForProfileCommand = /* @__PURE__ */ __name(async (input, context) => {
  const b = (0, import_core.requestBuilder)(input, context);
  const headers = {};
  b.bp("/domains/{DomainName}/profile/{ProfileId}/calculated-attributes/{CalculatedAttributeName}");
  b.p("DomainName", () => input.DomainName, "{DomainName}", false);
  b.p("ProfileId", () => input.ProfileId, "{ProfileId}", false);
  b.p("CalculatedAttributeName", () => input.CalculatedAttributeName, "{CalculatedAttributeName}", false);
  let body;
  b.m("GET").h(headers).b(body);
  return b.build();
}, "se_GetCalculatedAttributeForProfileCommand");
var se_GetDomainCommand = /* @__PURE__ */ __name(async (input, context) => {
  const b = (0, import_core.requestBuilder)(input, context);
  const headers = {};
  b.bp("/domains/{DomainName}");
  b.p("DomainName", () => input.DomainName, "{DomainName}", false);
  let body;
  b.m("GET").h(headers).b(body);
  return b.build();
}, "se_GetDomainCommand");
var se_GetEventStreamCommand = /* @__PURE__ */ __name(async (input, context) => {
  const b = (0, import_core.requestBuilder)(input, context);
  const headers = {};
  b.bp("/domains/{DomainName}/event-streams/{EventStreamName}");
  b.p("DomainName", () => input.DomainName, "{DomainName}", false);
  b.p("EventStreamName", () => input.EventStreamName, "{EventStreamName}", false);
  let body;
  b.m("GET").h(headers).b(body);
  return b.build();
}, "se_GetEventStreamCommand");
var se_GetIdentityResolutionJobCommand = /* @__PURE__ */ __name(async (input, context) => {
  const b = (0, import_core.requestBuilder)(input, context);
  const headers = {};
  b.bp("/domains/{DomainName}/identity-resolution-jobs/{JobId}");
  b.p("DomainName", () => input.DomainName, "{DomainName}", false);
  b.p("JobId", () => input.JobId, "{JobId}", false);
  let body;
  b.m("GET").h(headers).b(body);
  return b.build();
}, "se_GetIdentityResolutionJobCommand");
var se_GetIntegrationCommand = /* @__PURE__ */ __name(async (input, context) => {
  const b = (0, import_core.requestBuilder)(input, context);
  const headers = {
    "content-type": "application/json"
  };
  b.bp("/domains/{DomainName}/integrations");
  b.p("DomainName", () => input.DomainName, "{DomainName}", false);
  let body;
  body = JSON.stringify(
    (0, import_smithy_client.take)(input, {
      Uri: []
    })
  );
  b.m("POST").h(headers).b(body);
  return b.build();
}, "se_GetIntegrationCommand");
var se_GetMatchesCommand = /* @__PURE__ */ __name(async (input, context) => {
  const b = (0, import_core.requestBuilder)(input, context);
  const headers = {};
  b.bp("/domains/{DomainName}/matches");
  b.p("DomainName", () => input.DomainName, "{DomainName}", false);
  const query = (0, import_smithy_client.map)({
    [_nt]: [, input[_NT]],
    [_mr]: [() => input.MaxResults !== void 0, () => input[_MR].toString()]
  });
  let body;
  b.m("GET").h(headers).q(query).b(body);
  return b.build();
}, "se_GetMatchesCommand");
var se_GetProfileObjectTypeCommand = /* @__PURE__ */ __name(async (input, context) => {
  const b = (0, import_core.requestBuilder)(input, context);
  const headers = {};
  b.bp("/domains/{DomainName}/object-types/{ObjectTypeName}");
  b.p("DomainName", () => input.DomainName, "{DomainName}", false);
  b.p("ObjectTypeName", () => input.ObjectTypeName, "{ObjectTypeName}", false);
  let body;
  b.m("GET").h(headers).b(body);
  return b.build();
}, "se_GetProfileObjectTypeCommand");
var se_GetProfileObjectTypeTemplateCommand = /* @__PURE__ */ __name(async (input, context) => {
  const b = (0, import_core.requestBuilder)(input, context);
  const headers = {};
  b.bp("/templates/{TemplateId}");
  b.p("TemplateId", () => input.TemplateId, "{TemplateId}", false);
  let body;
  b.m("GET").h(headers).b(body);
  return b.build();
}, "se_GetProfileObjectTypeTemplateCommand");
var se_GetSimilarProfilesCommand = /* @__PURE__ */ __name(async (input, context) => {
  const b = (0, import_core.requestBuilder)(input, context);
  const headers = {
    "content-type": "application/json"
  };
  b.bp("/domains/{DomainName}/matches");
  b.p("DomainName", () => input.DomainName, "{DomainName}", false);
  const query = (0, import_smithy_client.map)({
    [_nt]: [, input[_NT]],
    [_mr]: [() => input.MaxResults !== void 0, () => input[_MR].toString()]
  });
  let body;
  body = JSON.stringify(
    (0, import_smithy_client.take)(input, {
      MatchType: [],
      SearchKey: [],
      SearchValue: []
    })
  );
  b.m("POST").h(headers).q(query).b(body);
  return b.build();
}, "se_GetSimilarProfilesCommand");
var se_GetWorkflowCommand = /* @__PURE__ */ __name(async (input, context) => {
  const b = (0, import_core.requestBuilder)(input, context);
  const headers = {};
  b.bp("/domains/{DomainName}/workflows/{WorkflowId}");
  b.p("DomainName", () => input.DomainName, "{DomainName}", false);
  b.p("WorkflowId", () => input.WorkflowId, "{WorkflowId}", false);
  let body;
  b.m("GET").h(headers).b(body);
  return b.build();
}, "se_GetWorkflowCommand");
var se_GetWorkflowStepsCommand = /* @__PURE__ */ __name(async (input, context) => {
  const b = (0, import_core.requestBuilder)(input, context);
  const headers = {};
  b.bp("/domains/{DomainName}/workflows/{WorkflowId}/steps");
  b.p("DomainName", () => input.DomainName, "{DomainName}", false);
  b.p("WorkflowId", () => input.WorkflowId, "{WorkflowId}", false);
  const query = (0, import_smithy_client.map)({
    [_nt]: [, input[_NT]],
    [_mr]: [() => input.MaxResults !== void 0, () => input[_MR].toString()]
  });
  let body;
  b.m("GET").h(headers).q(query).b(body);
  return b.build();
}, "se_GetWorkflowStepsCommand");
var se_ListAccountIntegrationsCommand = /* @__PURE__ */ __name(async (input, context) => {
  const b = (0, import_core.requestBuilder)(input, context);
  const headers = {
    "content-type": "application/json"
  };
  b.bp("/integrations");
  const query = (0, import_smithy_client.map)({
    [_nt]: [, input[_NT]],
    [_mr]: [() => input.MaxResults !== void 0, () => input[_MR].toString()],
    [_ih]: [() => input.IncludeHidden !== void 0, () => input[_IH].toString()]
  });
  let body;
  body = JSON.stringify(
    (0, import_smithy_client.take)(input, {
      Uri: []
    })
  );
  b.m("POST").h(headers).q(query).b(body);
  return b.build();
}, "se_ListAccountIntegrationsCommand");
var se_ListCalculatedAttributeDefinitionsCommand = /* @__PURE__ */ __name(async (input, context) => {
  const b = (0, import_core.requestBuilder)(input, context);
  const headers = {};
  b.bp("/domains/{DomainName}/calculated-attributes");
  b.p("DomainName", () => input.DomainName, "{DomainName}", false);
  const query = (0, import_smithy_client.map)({
    [_nt]: [, input[_NT]],
    [_mr]: [() => input.MaxResults !== void 0, () => input[_MR].toString()]
  });
  let body;
  b.m("GET").h(headers).q(query).b(body);
  return b.build();
}, "se_ListCalculatedAttributeDefinitionsCommand");
var se_ListCalculatedAttributesForProfileCommand = /* @__PURE__ */ __name(async (input, context) => {
  const b = (0, import_core.requestBuilder)(input, context);
  const headers = {};
  b.bp("/domains/{DomainName}/profile/{ProfileId}/calculated-attributes");
  b.p("DomainName", () => input.DomainName, "{DomainName}", false);
  b.p("ProfileId", () => input.ProfileId, "{ProfileId}", false);
  const query = (0, import_smithy_client.map)({
    [_nt]: [, input[_NT]],
    [_mr]: [() => input.MaxResults !== void 0, () => input[_MR].toString()]
  });
  let body;
  b.m("GET").h(headers).q(query).b(body);
  return b.build();
}, "se_ListCalculatedAttributesForProfileCommand");
var se_ListDomainsCommand = /* @__PURE__ */ __name(async (input, context) => {
  const b = (0, import_core.requestBuilder)(input, context);
  const headers = {};
  b.bp("/domains");
  const query = (0, import_smithy_client.map)({
    [_nt]: [, input[_NT]],
    [_mr]: [() => input.MaxResults !== void 0, () => input[_MR].toString()]
  });
  let body;
  b.m("GET").h(headers).q(query).b(body);
  return b.build();
}, "se_ListDomainsCommand");
var se_ListEventStreamsCommand = /* @__PURE__ */ __name(async (input, context) => {
  const b = (0, import_core.requestBuilder)(input, context);
  const headers = {};
  b.bp("/domains/{DomainName}/event-streams");
  b.p("DomainName", () => input.DomainName, "{DomainName}", false);
  const query = (0, import_smithy_client.map)({
    [_nt]: [, input[_NT]],
    [_mr]: [() => input.MaxResults !== void 0, () => input[_MR].toString()]
  });
  let body;
  b.m("GET").h(headers).q(query).b(body);
  return b.build();
}, "se_ListEventStreamsCommand");
var se_ListIdentityResolutionJobsCommand = /* @__PURE__ */ __name(async (input, context) => {
  const b = (0, import_core.requestBuilder)(input, context);
  const headers = {};
  b.bp("/domains/{DomainName}/identity-resolution-jobs");
  b.p("DomainName", () => input.DomainName, "{DomainName}", false);
  const query = (0, import_smithy_client.map)({
    [_nt]: [, input[_NT]],
    [_mr]: [() => input.MaxResults !== void 0, () => input[_MR].toString()]
  });
  let body;
  b.m("GET").h(headers).q(query).b(body);
  return b.build();
}, "se_ListIdentityResolutionJobsCommand");
var se_ListIntegrationsCommand = /* @__PURE__ */ __name(async (input, context) => {
  const b = (0, import_core.requestBuilder)(input, context);
  const headers = {};
  b.bp("/domains/{DomainName}/integrations");
  b.p("DomainName", () => input.DomainName, "{DomainName}", false);
  const query = (0, import_smithy_client.map)({
    [_nt]: [, input[_NT]],
    [_mr]: [() => input.MaxResults !== void 0, () => input[_MR].toString()],
    [_ih]: [() => input.IncludeHidden !== void 0, () => input[_IH].toString()]
  });
  let body;
  b.m("GET").h(headers).q(query).b(body);
  return b.build();
}, "se_ListIntegrationsCommand");
var se_ListProfileObjectsCommand = /* @__PURE__ */ __name(async (input, context) => {
  const b = (0, import_core.requestBuilder)(input, context);
  const headers = {
    "content-type": "application/json"
  };
  b.bp("/domains/{DomainName}/profiles/objects");
  b.p("DomainName", () => input.DomainName, "{DomainName}", false);
  const query = (0, import_smithy_client.map)({
    [_nt]: [, input[_NT]],
    [_mr]: [() => input.MaxResults !== void 0, () => input[_MR].toString()]
  });
  let body;
  body = JSON.stringify(
    (0, import_smithy_client.take)(input, {
      ObjectFilter: (_) => (0, import_smithy_client._json)(_),
      ObjectTypeName: [],
      ProfileId: []
    })
  );
  b.m("POST").h(headers).q(query).b(body);
  return b.build();
}, "se_ListProfileObjectsCommand");
var se_ListProfileObjectTypesCommand = /* @__PURE__ */ __name(async (input, context) => {
  const b = (0, import_core.requestBuilder)(input, context);
  const headers = {};
  b.bp("/domains/{DomainName}/object-types");
  b.p("DomainName", () => input.DomainName, "{DomainName}", false);
  const query = (0, import_smithy_client.map)({
    [_nt]: [, input[_NT]],
    [_mr]: [() => input.MaxResults !== void 0, () => input[_MR].toString()]
  });
  let body;
  b.m("GET").h(headers).q(query).b(body);
  return b.build();
}, "se_ListProfileObjectTypesCommand");
var se_ListProfileObjectTypeTemplatesCommand = /* @__PURE__ */ __name(async (input, context) => {
  const b = (0, import_core.requestBuilder)(input, context);
  const headers = {};
  b.bp("/templates");
  const query = (0, import_smithy_client.map)({
    [_nt]: [, input[_NT]],
    [_mr]: [() => input.MaxResults !== void 0, () => input[_MR].toString()]
  });
  let body;
  b.m("GET").h(headers).q(query).b(body);
  return b.build();
}, "se_ListProfileObjectTypeTemplatesCommand");
var se_ListRuleBasedMatchesCommand = /* @__PURE__ */ __name(async (input, context) => {
  const b = (0, import_core.requestBuilder)(input, context);
  const headers = {};
  b.bp("/domains/{DomainName}/profiles/ruleBasedMatches");
  b.p("DomainName", () => input.DomainName, "{DomainName}", false);
  const query = (0, import_smithy_client.map)({
    [_nt]: [, input[_NT]],
    [_mr]: [() => input.MaxResults !== void 0, () => input[_MR].toString()]
  });
  let body;
  b.m("GET").h(headers).q(query).b(body);
  return b.build();
}, "se_ListRuleBasedMatchesCommand");
var se_ListTagsForResourceCommand = /* @__PURE__ */ __name(async (input, context) => {
  const b = (0, import_core.requestBuilder)(input, context);
  const headers = {};
  b.bp("/tags/{resourceArn}");
  b.p("resourceArn", () => input.resourceArn, "{resourceArn}", false);
  let body;
  b.m("GET").h(headers).b(body);
  return b.build();
}, "se_ListTagsForResourceCommand");
var se_ListWorkflowsCommand = /* @__PURE__ */ __name(async (input, context) => {
  const b = (0, import_core.requestBuilder)(input, context);
  const headers = {
    "content-type": "application/json"
  };
  b.bp("/domains/{DomainName}/workflows");
  b.p("DomainName", () => input.DomainName, "{DomainName}", false);
  const query = (0, import_smithy_client.map)({
    [_nt]: [, input[_NT]],
    [_mr]: [() => input.MaxResults !== void 0, () => input[_MR].toString()]
  });
  let body;
  body = JSON.stringify(
    (0, import_smithy_client.take)(input, {
      QueryEndDate: (_) => Math.round(_.getTime() / 1e3),
      QueryStartDate: (_) => Math.round(_.getTime() / 1e3),
      Status: [],
      WorkflowType: []
    })
  );
  b.m("POST").h(headers).q(query).b(body);
  return b.build();
}, "se_ListWorkflowsCommand");
var se_MergeProfilesCommand = /* @__PURE__ */ __name(async (input, context) => {
  const b = (0, import_core.requestBuilder)(input, context);
  const headers = {
    "content-type": "application/json"
  };
  b.bp("/domains/{DomainName}/profiles/objects/merge");
  b.p("DomainName", () => input.DomainName, "{DomainName}", false);
  let body;
  body = JSON.stringify(
    (0, import_smithy_client.take)(input, {
      FieldSourceProfileIds: (_) => (0, import_smithy_client._json)(_),
      MainProfileId: [],
      ProfileIdsToBeMerged: (_) => (0, import_smithy_client._json)(_)
    })
  );
  b.m("POST").h(headers).b(body);
  return b.build();
}, "se_MergeProfilesCommand");
var se_PutIntegrationCommand = /* @__PURE__ */ __name(async (input, context) => {
  const b = (0, import_core.requestBuilder)(input, context);
  const headers = {
    "content-type": "application/json"
  };
  b.bp("/domains/{DomainName}/integrations");
  b.p("DomainName", () => input.DomainName, "{DomainName}", false);
  let body;
  body = JSON.stringify(
    (0, import_smithy_client.take)(input, {
      FlowDefinition: (_) => se_FlowDefinition(_, context),
      ObjectTypeName: [],
      ObjectTypeNames: (_) => (0, import_smithy_client._json)(_),
      Tags: (_) => (0, import_smithy_client._json)(_),
      Uri: []
    })
  );
  b.m("PUT").h(headers).b(body);
  return b.build();
}, "se_PutIntegrationCommand");
var se_PutProfileObjectCommand = /* @__PURE__ */ __name(async (input, context) => {
  const b = (0, import_core.requestBuilder)(input, context);
  const headers = {
    "content-type": "application/json"
  };
  b.bp("/domains/{DomainName}/profiles/objects");
  b.p("DomainName", () => input.DomainName, "{DomainName}", false);
  let body;
  body = JSON.stringify(
    (0, import_smithy_client.take)(input, {
      Object: [],
      ObjectTypeName: []
    })
  );
  b.m("PUT").h(headers).b(body);
  return b.build();
}, "se_PutProfileObjectCommand");
var se_PutProfileObjectTypeCommand = /* @__PURE__ */ __name(async (input, context) => {
  const b = (0, import_core.requestBuilder)(input, context);
  const headers = {
    "content-type": "application/json"
  };
  b.bp("/domains/{DomainName}/object-types/{ObjectTypeName}");
  b.p("DomainName", () => input.DomainName, "{DomainName}", false);
  b.p("ObjectTypeName", () => input.ObjectTypeName, "{ObjectTypeName}", false);
  let body;
  body = JSON.stringify(
    (0, import_smithy_client.take)(input, {
      AllowProfileCreation: [],
      Description: [],
      EncryptionKey: [],
      ExpirationDays: [],
      Fields: (_) => (0, import_smithy_client._json)(_),
      Keys: (_) => (0, import_smithy_client._json)(_),
      SourceLastUpdatedTimestampFormat: [],
      Tags: (_) => (0, import_smithy_client._json)(_),
      TemplateId: []
    })
  );
  b.m("PUT").h(headers).b(body);
  return b.build();
}, "se_PutProfileObjectTypeCommand");
var se_SearchProfilesCommand = /* @__PURE__ */ __name(async (input, context) => {
  const b = (0, import_core.requestBuilder)(input, context);
  const headers = {
    "content-type": "application/json"
  };
  b.bp("/domains/{DomainName}/profiles/search");
  b.p("DomainName", () => input.DomainName, "{DomainName}", false);
  const query = (0, import_smithy_client.map)({
    [_nt]: [, input[_NT]],
    [_mr]: [() => input.MaxResults !== void 0, () => input[_MR].toString()]
  });
  let body;
  body = JSON.stringify(
    (0, import_smithy_client.take)(input, {
      AdditionalSearchKeys: (_) => (0, import_smithy_client._json)(_),
      KeyName: [],
      LogicalOperator: [],
      Values: (_) => (0, import_smithy_client._json)(_)
    })
  );
  b.m("POST").h(headers).q(query).b(body);
  return b.build();
}, "se_SearchProfilesCommand");
var se_TagResourceCommand = /* @__PURE__ */ __name(async (input, context) => {
  const b = (0, import_core.requestBuilder)(input, context);
  const headers = {
    "content-type": "application/json"
  };
  b.bp("/tags/{resourceArn}");
  b.p("resourceArn", () => input.resourceArn, "{resourceArn}", false);
  let body;
  body = JSON.stringify(
    (0, import_smithy_client.take)(input, {
      tags: (_) => (0, import_smithy_client._json)(_)
    })
  );
  b.m("POST").h(headers).b(body);
  return b.build();
}, "se_TagResourceCommand");
var se_UntagResourceCommand = /* @__PURE__ */ __name(async (input, context) => {
  const b = (0, import_core.requestBuilder)(input, context);
  const headers = {};
  b.bp("/tags/{resourceArn}");
  b.p("resourceArn", () => input.resourceArn, "{resourceArn}", false);
  const query = (0, import_smithy_client.map)({
    [_tK]: [
      (0, import_smithy_client.expectNonNull)(input.tagKeys, `tagKeys`) != null,
      () => (input[_tK] || []).map((_entry) => _entry)
    ]
  });
  let body;
  b.m("DELETE").h(headers).q(query).b(body);
  return b.build();
}, "se_UntagResourceCommand");
var se_UpdateCalculatedAttributeDefinitionCommand = /* @__PURE__ */ __name(async (input, context) => {
  const b = (0, import_core.requestBuilder)(input, context);
  const headers = {
    "content-type": "application/json"
  };
  b.bp("/domains/{DomainName}/calculated-attributes/{CalculatedAttributeName}");
  b.p("DomainName", () => input.DomainName, "{DomainName}", false);
  b.p("CalculatedAttributeName", () => input.CalculatedAttributeName, "{CalculatedAttributeName}", false);
  let body;
  body = JSON.stringify(
    (0, import_smithy_client.take)(input, {
      Conditions: (_) => (0, import_smithy_client._json)(_),
      Description: [],
      DisplayName: []
    })
  );
  b.m("PUT").h(headers).b(body);
  return b.build();
}, "se_UpdateCalculatedAttributeDefinitionCommand");
var se_UpdateDomainCommand = /* @__PURE__ */ __name(async (input, context) => {
  const b = (0, import_core.requestBuilder)(input, context);
  const headers = {
    "content-type": "application/json"
  };
  b.bp("/domains/{DomainName}");
  b.p("DomainName", () => input.DomainName, "{DomainName}", false);
  let body;
  body = JSON.stringify(
    (0, import_smithy_client.take)(input, {
      DeadLetterQueueUrl: [],
      DefaultEncryptionKey: [],
      DefaultExpirationDays: [],
      Matching: (_) => se_MatchingRequest(_, context),
      RuleBasedMatching: (_) => (0, import_smithy_client._json)(_),
      Tags: (_) => (0, import_smithy_client._json)(_)
    })
  );
  b.m("PUT").h(headers).b(body);
  return b.build();
}, "se_UpdateDomainCommand");
var se_UpdateProfileCommand = /* @__PURE__ */ __name(async (input, context) => {
  const b = (0, import_core.requestBuilder)(input, context);
  const headers = {
    "content-type": "application/json"
  };
  b.bp("/domains/{DomainName}/profiles");
  b.p("DomainName", () => input.DomainName, "{DomainName}", false);
  let body;
  body = JSON.stringify(
    (0, import_smithy_client.take)(input, {
      AccountNumber: [],
      AdditionalInformation: [],
      Address: (_) => (0, import_smithy_client._json)(_),
      Attributes: (_) => (0, import_smithy_client._json)(_),
      BillingAddress: (_) => (0, import_smithy_client._json)(_),
      BirthDate: [],
      BusinessEmailAddress: [],
      BusinessName: [],
      BusinessPhoneNumber: [],
      EmailAddress: [],
      FirstName: [],
      Gender: [],
      GenderString: [],
      HomePhoneNumber: [],
      LastName: [],
      MailingAddress: (_) => (0, import_smithy_client._json)(_),
      MiddleName: [],
      MobilePhoneNumber: [],
      PartyType: [],
      PartyTypeString: [],
      PersonalEmailAddress: [],
      PhoneNumber: [],
      ProfileId: [],
      ShippingAddress: (_) => (0, import_smithy_client._json)(_)
    })
  );
  b.m("PUT").h(headers).b(body);
  return b.build();
}, "se_UpdateProfileCommand");
var de_AddProfileKeyCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode !== 200 && output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const contents = (0, import_smithy_client.map)({
    $metadata: deserializeMetadata(output)
  });
  const data = (0, import_smithy_client.expectNonNull)((0, import_smithy_client.expectObject)(await parseBody(output.body, context)), "body");
  const doc = (0, import_smithy_client.take)(data, {
    KeyName: import_smithy_client.expectString,
    Values: import_smithy_client._json
  });
  Object.assign(contents, doc);
  return contents;
}, "de_AddProfileKeyCommand");
var de_CreateCalculatedAttributeDefinitionCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode !== 200 && output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const contents = (0, import_smithy_client.map)({
    $metadata: deserializeMetadata(output)
  });
  const data = (0, import_smithy_client.expectNonNull)((0, import_smithy_client.expectObject)(await parseBody(output.body, context)), "body");
  const doc = (0, import_smithy_client.take)(data, {
    AttributeDetails: import_smithy_client._json,
    CalculatedAttributeName: import_smithy_client.expectString,
    Conditions: import_smithy_client._json,
    CreatedAt: (_) => (0, import_smithy_client.expectNonNull)((0, import_smithy_client.parseEpochTimestamp)((0, import_smithy_client.expectNumber)(_))),
    Description: import_smithy_client.expectString,
    DisplayName: import_smithy_client.expectString,
    LastUpdatedAt: (_) => (0, import_smithy_client.expectNonNull)((0, import_smithy_client.parseEpochTimestamp)((0, import_smithy_client.expectNumber)(_))),
    Statistic: import_smithy_client.expectString,
    Tags: import_smithy_client._json
  });
  Object.assign(contents, doc);
  return contents;
}, "de_CreateCalculatedAttributeDefinitionCommand");
var de_CreateDomainCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode !== 200 && output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const contents = (0, import_smithy_client.map)({
    $metadata: deserializeMetadata(output)
  });
  const data = (0, import_smithy_client.expectNonNull)((0, import_smithy_client.expectObject)(await parseBody(output.body, context)), "body");
  const doc = (0, import_smithy_client.take)(data, {
    CreatedAt: (_) => (0, import_smithy_client.expectNonNull)((0, import_smithy_client.parseEpochTimestamp)((0, import_smithy_client.expectNumber)(_))),
    DeadLetterQueueUrl: import_smithy_client.expectString,
    DefaultEncryptionKey: import_smithy_client.expectString,
    DefaultExpirationDays: import_smithy_client.expectInt32,
    DomainName: import_smithy_client.expectString,
    LastUpdatedAt: (_) => (0, import_smithy_client.expectNonNull)((0, import_smithy_client.parseEpochTimestamp)((0, import_smithy_client.expectNumber)(_))),
    Matching: (_) => de_MatchingResponse(_, context),
    RuleBasedMatching: import_smithy_client._json,
    Tags: import_smithy_client._json
  });
  Object.assign(contents, doc);
  return contents;
}, "de_CreateDomainCommand");
var de_CreateEventStreamCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode !== 200 && output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const contents = (0, import_smithy_client.map)({
    $metadata: deserializeMetadata(output)
  });
  const data = (0, import_smithy_client.expectNonNull)((0, import_smithy_client.expectObject)(await parseBody(output.body, context)), "body");
  const doc = (0, import_smithy_client.take)(data, {
    EventStreamArn: import_smithy_client.expectString,
    Tags: import_smithy_client._json
  });
  Object.assign(contents, doc);
  return contents;
}, "de_CreateEventStreamCommand");
var de_CreateIntegrationWorkflowCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode !== 200 && output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const contents = (0, import_smithy_client.map)({
    $metadata: deserializeMetadata(output)
  });
  const data = (0, import_smithy_client.expectNonNull)((0, import_smithy_client.expectObject)(await parseBody(output.body, context)), "body");
  const doc = (0, import_smithy_client.take)(data, {
    Message: import_smithy_client.expectString,
    WorkflowId: import_smithy_client.expectString
  });
  Object.assign(contents, doc);
  return contents;
}, "de_CreateIntegrationWorkflowCommand");
var de_CreateProfileCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode !== 200 && output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const contents = (0, import_smithy_client.map)({
    $metadata: deserializeMetadata(output)
  });
  const data = (0, import_smithy_client.expectNonNull)((0, import_smithy_client.expectObject)(await parseBody(output.body, context)), "body");
  const doc = (0, import_smithy_client.take)(data, {
    ProfileId: import_smithy_client.expectString
  });
  Object.assign(contents, doc);
  return contents;
}, "de_CreateProfileCommand");
var de_DeleteCalculatedAttributeDefinitionCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode !== 200 && output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const contents = (0, import_smithy_client.map)({
    $metadata: deserializeMetadata(output)
  });
  await (0, import_smithy_client.collectBody)(output.body, context);
  return contents;
}, "de_DeleteCalculatedAttributeDefinitionCommand");
var de_DeleteDomainCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode !== 200 && output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const contents = (0, import_smithy_client.map)({
    $metadata: deserializeMetadata(output)
  });
  const data = (0, import_smithy_client.expectNonNull)((0, import_smithy_client.expectObject)(await parseBody(output.body, context)), "body");
  const doc = (0, import_smithy_client.take)(data, {
    Message: import_smithy_client.expectString
  });
  Object.assign(contents, doc);
  return contents;
}, "de_DeleteDomainCommand");
var de_DeleteEventStreamCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode !== 200 && output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const contents = (0, import_smithy_client.map)({
    $metadata: deserializeMetadata(output)
  });
  await (0, import_smithy_client.collectBody)(output.body, context);
  return contents;
}, "de_DeleteEventStreamCommand");
var de_DeleteIntegrationCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode !== 200 && output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const contents = (0, import_smithy_client.map)({
    $metadata: deserializeMetadata(output)
  });
  const data = (0, import_smithy_client.expectNonNull)((0, import_smithy_client.expectObject)(await parseBody(output.body, context)), "body");
  const doc = (0, import_smithy_client.take)(data, {
    Message: import_smithy_client.expectString
  });
  Object.assign(contents, doc);
  return contents;
}, "de_DeleteIntegrationCommand");
var de_DeleteProfileCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode !== 200 && output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const contents = (0, import_smithy_client.map)({
    $metadata: deserializeMetadata(output)
  });
  const data = (0, import_smithy_client.expectNonNull)((0, import_smithy_client.expectObject)(await parseBody(output.body, context)), "body");
  const doc = (0, import_smithy_client.take)(data, {
    Message: import_smithy_client.expectString
  });
  Object.assign(contents, doc);
  return contents;
}, "de_DeleteProfileCommand");
var de_DeleteProfileKeyCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode !== 200 && output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const contents = (0, import_smithy_client.map)({
    $metadata: deserializeMetadata(output)
  });
  const data = (0, import_smithy_client.expectNonNull)((0, import_smithy_client.expectObject)(await parseBody(output.body, context)), "body");
  const doc = (0, import_smithy_client.take)(data, {
    Message: import_smithy_client.expectString
  });
  Object.assign(contents, doc);
  return contents;
}, "de_DeleteProfileKeyCommand");
var de_DeleteProfileObjectCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode !== 200 && output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const contents = (0, import_smithy_client.map)({
    $metadata: deserializeMetadata(output)
  });
  const data = (0, import_smithy_client.expectNonNull)((0, import_smithy_client.expectObject)(await parseBody(output.body, context)), "body");
  const doc = (0, import_smithy_client.take)(data, {
    Message: import_smithy_client.expectString
  });
  Object.assign(contents, doc);
  return contents;
}, "de_DeleteProfileObjectCommand");
var de_DeleteProfileObjectTypeCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode !== 200 && output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const contents = (0, import_smithy_client.map)({
    $metadata: deserializeMetadata(output)
  });
  const data = (0, import_smithy_client.expectNonNull)((0, import_smithy_client.expectObject)(await parseBody(output.body, context)), "body");
  const doc = (0, import_smithy_client.take)(data, {
    Message: import_smithy_client.expectString
  });
  Object.assign(contents, doc);
  return contents;
}, "de_DeleteProfileObjectTypeCommand");
var de_DeleteWorkflowCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode !== 200 && output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const contents = (0, import_smithy_client.map)({
    $metadata: deserializeMetadata(output)
  });
  await (0, import_smithy_client.collectBody)(output.body, context);
  return contents;
}, "de_DeleteWorkflowCommand");
var de_DetectProfileObjectTypeCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode !== 200 && output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const contents = (0, import_smithy_client.map)({
    $metadata: deserializeMetadata(output)
  });
  const data = (0, import_smithy_client.expectNonNull)((0, import_smithy_client.expectObject)(await parseBody(output.body, context)), "body");
  const doc = (0, import_smithy_client.take)(data, {
    DetectedProfileObjectTypes: import_smithy_client._json
  });
  Object.assign(contents, doc);
  return contents;
}, "de_DetectProfileObjectTypeCommand");
var de_GetAutoMergingPreviewCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode !== 200 && output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const contents = (0, import_smithy_client.map)({
    $metadata: deserializeMetadata(output)
  });
  const data = (0, import_smithy_client.expectNonNull)((0, import_smithy_client.expectObject)(await parseBody(output.body, context)), "body");
  const doc = (0, import_smithy_client.take)(data, {
    DomainName: import_smithy_client.expectString,
    NumberOfMatchesInSample: import_smithy_client.expectLong,
    NumberOfProfilesInSample: import_smithy_client.expectLong,
    NumberOfProfilesWillBeMerged: import_smithy_client.expectLong
  });
  Object.assign(contents, doc);
  return contents;
}, "de_GetAutoMergingPreviewCommand");
var de_GetCalculatedAttributeDefinitionCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode !== 200 && output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const contents = (0, import_smithy_client.map)({
    $metadata: deserializeMetadata(output)
  });
  const data = (0, import_smithy_client.expectNonNull)((0, import_smithy_client.expectObject)(await parseBody(output.body, context)), "body");
  const doc = (0, import_smithy_client.take)(data, {
    AttributeDetails: import_smithy_client._json,
    CalculatedAttributeName: import_smithy_client.expectString,
    Conditions: import_smithy_client._json,
    CreatedAt: (_) => (0, import_smithy_client.expectNonNull)((0, import_smithy_client.parseEpochTimestamp)((0, import_smithy_client.expectNumber)(_))),
    Description: import_smithy_client.expectString,
    DisplayName: import_smithy_client.expectString,
    LastUpdatedAt: (_) => (0, import_smithy_client.expectNonNull)((0, import_smithy_client.parseEpochTimestamp)((0, import_smithy_client.expectNumber)(_))),
    Statistic: import_smithy_client.expectString,
    Tags: import_smithy_client._json
  });
  Object.assign(contents, doc);
  return contents;
}, "de_GetCalculatedAttributeDefinitionCommand");
var de_GetCalculatedAttributeForProfileCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode !== 200 && output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const contents = (0, import_smithy_client.map)({
    $metadata: deserializeMetadata(output)
  });
  const data = (0, import_smithy_client.expectNonNull)((0, import_smithy_client.expectObject)(await parseBody(output.body, context)), "body");
  const doc = (0, import_smithy_client.take)(data, {
    CalculatedAttributeName: import_smithy_client.expectString,
    DisplayName: import_smithy_client.expectString,
    IsDataPartial: import_smithy_client.expectString,
    Value: import_smithy_client.expectString
  });
  Object.assign(contents, doc);
  return contents;
}, "de_GetCalculatedAttributeForProfileCommand");
var de_GetDomainCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode !== 200 && output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const contents = (0, import_smithy_client.map)({
    $metadata: deserializeMetadata(output)
  });
  const data = (0, import_smithy_client.expectNonNull)((0, import_smithy_client.expectObject)(await parseBody(output.body, context)), "body");
  const doc = (0, import_smithy_client.take)(data, {
    CreatedAt: (_) => (0, import_smithy_client.expectNonNull)((0, import_smithy_client.parseEpochTimestamp)((0, import_smithy_client.expectNumber)(_))),
    DeadLetterQueueUrl: import_smithy_client.expectString,
    DefaultEncryptionKey: import_smithy_client.expectString,
    DefaultExpirationDays: import_smithy_client.expectInt32,
    DomainName: import_smithy_client.expectString,
    LastUpdatedAt: (_) => (0, import_smithy_client.expectNonNull)((0, import_smithy_client.parseEpochTimestamp)((0, import_smithy_client.expectNumber)(_))),
    Matching: (_) => de_MatchingResponse(_, context),
    RuleBasedMatching: import_smithy_client._json,
    Stats: import_smithy_client._json,
    Tags: import_smithy_client._json
  });
  Object.assign(contents, doc);
  return contents;
}, "de_GetDomainCommand");
var de_GetEventStreamCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode !== 200 && output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const contents = (0, import_smithy_client.map)({
    $metadata: deserializeMetadata(output)
  });
  const data = (0, import_smithy_client.expectNonNull)((0, import_smithy_client.expectObject)(await parseBody(output.body, context)), "body");
  const doc = (0, import_smithy_client.take)(data, {
    CreatedAt: (_) => (0, import_smithy_client.expectNonNull)((0, import_smithy_client.parseEpochTimestamp)((0, import_smithy_client.expectNumber)(_))),
    DestinationDetails: (_) => de_EventStreamDestinationDetails(_, context),
    DomainName: import_smithy_client.expectString,
    EventStreamArn: import_smithy_client.expectString,
    State: import_smithy_client.expectString,
    StoppedSince: (_) => (0, import_smithy_client.expectNonNull)((0, import_smithy_client.parseEpochTimestamp)((0, import_smithy_client.expectNumber)(_))),
    Tags: import_smithy_client._json
  });
  Object.assign(contents, doc);
  return contents;
}, "de_GetEventStreamCommand");
var de_GetIdentityResolutionJobCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode !== 200 && output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const contents = (0, import_smithy_client.map)({
    $metadata: deserializeMetadata(output)
  });
  const data = (0, import_smithy_client.expectNonNull)((0, import_smithy_client.expectObject)(await parseBody(output.body, context)), "body");
  const doc = (0, import_smithy_client.take)(data, {
    AutoMerging: (_) => de_AutoMerging(_, context),
    DomainName: import_smithy_client.expectString,
    ExportingLocation: import_smithy_client._json,
    JobEndTime: (_) => (0, import_smithy_client.expectNonNull)((0, import_smithy_client.parseEpochTimestamp)((0, import_smithy_client.expectNumber)(_))),
    JobExpirationTime: (_) => (0, import_smithy_client.expectNonNull)((0, import_smithy_client.parseEpochTimestamp)((0, import_smithy_client.expectNumber)(_))),
    JobId: import_smithy_client.expectString,
    JobStartTime: (_) => (0, import_smithy_client.expectNonNull)((0, import_smithy_client.parseEpochTimestamp)((0, import_smithy_client.expectNumber)(_))),
    JobStats: import_smithy_client._json,
    LastUpdatedAt: (_) => (0, import_smithy_client.expectNonNull)((0, import_smithy_client.parseEpochTimestamp)((0, import_smithy_client.expectNumber)(_))),
    Message: import_smithy_client.expectString,
    Status: import_smithy_client.expectString
  });
  Object.assign(contents, doc);
  return contents;
}, "de_GetIdentityResolutionJobCommand");
var de_GetIntegrationCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode !== 200 && output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const contents = (0, import_smithy_client.map)({
    $metadata: deserializeMetadata(output)
  });
  const data = (0, import_smithy_client.expectNonNull)((0, import_smithy_client.expectObject)(await parseBody(output.body, context)), "body");
  const doc = (0, import_smithy_client.take)(data, {
    CreatedAt: (_) => (0, import_smithy_client.expectNonNull)((0, import_smithy_client.parseEpochTimestamp)((0, import_smithy_client.expectNumber)(_))),
    DomainName: import_smithy_client.expectString,
    IsUnstructured: import_smithy_client.expectBoolean,
    LastUpdatedAt: (_) => (0, import_smithy_client.expectNonNull)((0, import_smithy_client.parseEpochTimestamp)((0, import_smithy_client.expectNumber)(_))),
    ObjectTypeName: import_smithy_client.expectString,
    ObjectTypeNames: import_smithy_client._json,
    Tags: import_smithy_client._json,
    Uri: import_smithy_client.expectString,
    WorkflowId: import_smithy_client.expectString
  });
  Object.assign(contents, doc);
  return contents;
}, "de_GetIntegrationCommand");
var de_GetMatchesCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode !== 200 && output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const contents = (0, import_smithy_client.map)({
    $metadata: deserializeMetadata(output)
  });
  const data = (0, import_smithy_client.expectNonNull)((0, import_smithy_client.expectObject)(await parseBody(output.body, context)), "body");
  const doc = (0, import_smithy_client.take)(data, {
    MatchGenerationDate: (_) => (0, import_smithy_client.expectNonNull)((0, import_smithy_client.parseEpochTimestamp)((0, import_smithy_client.expectNumber)(_))),
    Matches: (_) => de_MatchesList(_, context),
    NextToken: import_smithy_client.expectString,
    PotentialMatches: import_smithy_client.expectInt32
  });
  Object.assign(contents, doc);
  return contents;
}, "de_GetMatchesCommand");
var de_GetProfileObjectTypeCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode !== 200 && output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const contents = (0, import_smithy_client.map)({
    $metadata: deserializeMetadata(output)
  });
  const data = (0, import_smithy_client.expectNonNull)((0, import_smithy_client.expectObject)(await parseBody(output.body, context)), "body");
  const doc = (0, import_smithy_client.take)(data, {
    AllowProfileCreation: import_smithy_client.expectBoolean,
    CreatedAt: (_) => (0, import_smithy_client.expectNonNull)((0, import_smithy_client.parseEpochTimestamp)((0, import_smithy_client.expectNumber)(_))),
    Description: import_smithy_client.expectString,
    EncryptionKey: import_smithy_client.expectString,
    ExpirationDays: import_smithy_client.expectInt32,
    Fields: import_smithy_client._json,
    Keys: import_smithy_client._json,
    LastUpdatedAt: (_) => (0, import_smithy_client.expectNonNull)((0, import_smithy_client.parseEpochTimestamp)((0, import_smithy_client.expectNumber)(_))),
    ObjectTypeName: import_smithy_client.expectString,
    SourceLastUpdatedTimestampFormat: import_smithy_client.expectString,
    Tags: import_smithy_client._json,
    TemplateId: import_smithy_client.expectString
  });
  Object.assign(contents, doc);
  return contents;
}, "de_GetProfileObjectTypeCommand");
var de_GetProfileObjectTypeTemplateCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode !== 200 && output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const contents = (0, import_smithy_client.map)({
    $metadata: deserializeMetadata(output)
  });
  const data = (0, import_smithy_client.expectNonNull)((0, import_smithy_client.expectObject)(await parseBody(output.body, context)), "body");
  const doc = (0, import_smithy_client.take)(data, {
    AllowProfileCreation: import_smithy_client.expectBoolean,
    Fields: import_smithy_client._json,
    Keys: import_smithy_client._json,
    SourceLastUpdatedTimestampFormat: import_smithy_client.expectString,
    SourceName: import_smithy_client.expectString,
    SourceObject: import_smithy_client.expectString,
    TemplateId: import_smithy_client.expectString
  });
  Object.assign(contents, doc);
  return contents;
}, "de_GetProfileObjectTypeTemplateCommand");
var de_GetSimilarProfilesCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode !== 200 && output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const contents = (0, import_smithy_client.map)({
    $metadata: deserializeMetadata(output)
  });
  const data = (0, import_smithy_client.expectNonNull)((0, import_smithy_client.expectObject)(await parseBody(output.body, context)), "body");
  const doc = (0, import_smithy_client.take)(data, {
    ConfidenceScore: import_smithy_client.limitedParseDouble,
    MatchId: import_smithy_client.expectString,
    MatchType: import_smithy_client.expectString,
    NextToken: import_smithy_client.expectString,
    ProfileIds: import_smithy_client._json,
    RuleLevel: import_smithy_client.expectInt32
  });
  Object.assign(contents, doc);
  return contents;
}, "de_GetSimilarProfilesCommand");
var de_GetWorkflowCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode !== 200 && output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const contents = (0, import_smithy_client.map)({
    $metadata: deserializeMetadata(output)
  });
  const data = (0, import_smithy_client.expectNonNull)((0, import_smithy_client.expectObject)(await parseBody(output.body, context)), "body");
  const doc = (0, import_smithy_client.take)(data, {
    Attributes: import_smithy_client._json,
    ErrorDescription: import_smithy_client.expectString,
    LastUpdatedAt: (_) => (0, import_smithy_client.expectNonNull)((0, import_smithy_client.parseEpochTimestamp)((0, import_smithy_client.expectNumber)(_))),
    Metrics: import_smithy_client._json,
    StartDate: (_) => (0, import_smithy_client.expectNonNull)((0, import_smithy_client.parseEpochTimestamp)((0, import_smithy_client.expectNumber)(_))),
    Status: import_smithy_client.expectString,
    WorkflowId: import_smithy_client.expectString,
    WorkflowType: import_smithy_client.expectString
  });
  Object.assign(contents, doc);
  return contents;
}, "de_GetWorkflowCommand");
var de_GetWorkflowStepsCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode !== 200 && output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const contents = (0, import_smithy_client.map)({
    $metadata: deserializeMetadata(output)
  });
  const data = (0, import_smithy_client.expectNonNull)((0, import_smithy_client.expectObject)(await parseBody(output.body, context)), "body");
  const doc = (0, import_smithy_client.take)(data, {
    Items: (_) => de_WorkflowStepsList(_, context),
    NextToken: import_smithy_client.expectString,
    WorkflowId: import_smithy_client.expectString,
    WorkflowType: import_smithy_client.expectString
  });
  Object.assign(contents, doc);
  return contents;
}, "de_GetWorkflowStepsCommand");
var de_ListAccountIntegrationsCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode !== 200 && output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const contents = (0, import_smithy_client.map)({
    $metadata: deserializeMetadata(output)
  });
  const data = (0, import_smithy_client.expectNonNull)((0, import_smithy_client.expectObject)(await parseBody(output.body, context)), "body");
  const doc = (0, import_smithy_client.take)(data, {
    Items: (_) => de_IntegrationList(_, context),
    NextToken: import_smithy_client.expectString
  });
  Object.assign(contents, doc);
  return contents;
}, "de_ListAccountIntegrationsCommand");
var de_ListCalculatedAttributeDefinitionsCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode !== 200 && output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const contents = (0, import_smithy_client.map)({
    $metadata: deserializeMetadata(output)
  });
  const data = (0, import_smithy_client.expectNonNull)((0, import_smithy_client.expectObject)(await parseBody(output.body, context)), "body");
  const doc = (0, import_smithy_client.take)(data, {
    Items: (_) => de_CalculatedAttributeDefinitionsList(_, context),
    NextToken: import_smithy_client.expectString
  });
  Object.assign(contents, doc);
  return contents;
}, "de_ListCalculatedAttributeDefinitionsCommand");
var de_ListCalculatedAttributesForProfileCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode !== 200 && output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const contents = (0, import_smithy_client.map)({
    $metadata: deserializeMetadata(output)
  });
  const data = (0, import_smithy_client.expectNonNull)((0, import_smithy_client.expectObject)(await parseBody(output.body, context)), "body");
  const doc = (0, import_smithy_client.take)(data, {
    Items: import_smithy_client._json,
    NextToken: import_smithy_client.expectString
  });
  Object.assign(contents, doc);
  return contents;
}, "de_ListCalculatedAttributesForProfileCommand");
var de_ListDomainsCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode !== 200 && output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const contents = (0, import_smithy_client.map)({
    $metadata: deserializeMetadata(output)
  });
  const data = (0, import_smithy_client.expectNonNull)((0, import_smithy_client.expectObject)(await parseBody(output.body, context)), "body");
  const doc = (0, import_smithy_client.take)(data, {
    Items: (_) => de_DomainList(_, context),
    NextToken: import_smithy_client.expectString
  });
  Object.assign(contents, doc);
  return contents;
}, "de_ListDomainsCommand");
var de_ListEventStreamsCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode !== 200 && output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const contents = (0, import_smithy_client.map)({
    $metadata: deserializeMetadata(output)
  });
  const data = (0, import_smithy_client.expectNonNull)((0, import_smithy_client.expectObject)(await parseBody(output.body, context)), "body");
  const doc = (0, import_smithy_client.take)(data, {
    Items: (_) => de_EventStreamSummaryList(_, context),
    NextToken: import_smithy_client.expectString
  });
  Object.assign(contents, doc);
  return contents;
}, "de_ListEventStreamsCommand");
var de_ListIdentityResolutionJobsCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode !== 200 && output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const contents = (0, import_smithy_client.map)({
    $metadata: deserializeMetadata(output)
  });
  const data = (0, import_smithy_client.expectNonNull)((0, import_smithy_client.expectObject)(await parseBody(output.body, context)), "body");
  const doc = (0, import_smithy_client.take)(data, {
    IdentityResolutionJobsList: (_) => de_IdentityResolutionJobsList(_, context),
    NextToken: import_smithy_client.expectString
  });
  Object.assign(contents, doc);
  return contents;
}, "de_ListIdentityResolutionJobsCommand");
var de_ListIntegrationsCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode !== 200 && output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const contents = (0, import_smithy_client.map)({
    $metadata: deserializeMetadata(output)
  });
  const data = (0, import_smithy_client.expectNonNull)((0, import_smithy_client.expectObject)(await parseBody(output.body, context)), "body");
  const doc = (0, import_smithy_client.take)(data, {
    Items: (_) => de_IntegrationList(_, context),
    NextToken: import_smithy_client.expectString
  });
  Object.assign(contents, doc);
  return contents;
}, "de_ListIntegrationsCommand");
var de_ListProfileObjectsCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode !== 200 && output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const contents = (0, import_smithy_client.map)({
    $metadata: deserializeMetadata(output)
  });
  const data = (0, import_smithy_client.expectNonNull)((0, import_smithy_client.expectObject)(await parseBody(output.body, context)), "body");
  const doc = (0, import_smithy_client.take)(data, {
    Items: import_smithy_client._json,
    NextToken: import_smithy_client.expectString
  });
  Object.assign(contents, doc);
  return contents;
}, "de_ListProfileObjectsCommand");
var de_ListProfileObjectTypesCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode !== 200 && output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const contents = (0, import_smithy_client.map)({
    $metadata: deserializeMetadata(output)
  });
  const data = (0, import_smithy_client.expectNonNull)((0, import_smithy_client.expectObject)(await parseBody(output.body, context)), "body");
  const doc = (0, import_smithy_client.take)(data, {
    Items: (_) => de_ProfileObjectTypeList(_, context),
    NextToken: import_smithy_client.expectString
  });
  Object.assign(contents, doc);
  return contents;
}, "de_ListProfileObjectTypesCommand");
var de_ListProfileObjectTypeTemplatesCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode !== 200 && output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const contents = (0, import_smithy_client.map)({
    $metadata: deserializeMetadata(output)
  });
  const data = (0, import_smithy_client.expectNonNull)((0, import_smithy_client.expectObject)(await parseBody(output.body, context)), "body");
  const doc = (0, import_smithy_client.take)(data, {
    Items: import_smithy_client._json,
    NextToken: import_smithy_client.expectString
  });
  Object.assign(contents, doc);
  return contents;
}, "de_ListProfileObjectTypeTemplatesCommand");
var de_ListRuleBasedMatchesCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode !== 200 && output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const contents = (0, import_smithy_client.map)({
    $metadata: deserializeMetadata(output)
  });
  const data = (0, import_smithy_client.expectNonNull)((0, import_smithy_client.expectObject)(await parseBody(output.body, context)), "body");
  const doc = (0, import_smithy_client.take)(data, {
    MatchIds: import_smithy_client._json,
    NextToken: import_smithy_client.expectString
  });
  Object.assign(contents, doc);
  return contents;
}, "de_ListRuleBasedMatchesCommand");
var de_ListTagsForResourceCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode !== 200 && output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const contents = (0, import_smithy_client.map)({
    $metadata: deserializeMetadata(output)
  });
  const data = (0, import_smithy_client.expectNonNull)((0, import_smithy_client.expectObject)(await parseBody(output.body, context)), "body");
  const doc = (0, import_smithy_client.take)(data, {
    tags: import_smithy_client._json
  });
  Object.assign(contents, doc);
  return contents;
}, "de_ListTagsForResourceCommand");
var de_ListWorkflowsCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode !== 200 && output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const contents = (0, import_smithy_client.map)({
    $metadata: deserializeMetadata(output)
  });
  const data = (0, import_smithy_client.expectNonNull)((0, import_smithy_client.expectObject)(await parseBody(output.body, context)), "body");
  const doc = (0, import_smithy_client.take)(data, {
    Items: (_) => de_WorkflowList(_, context),
    NextToken: import_smithy_client.expectString
  });
  Object.assign(contents, doc);
  return contents;
}, "de_ListWorkflowsCommand");
var de_MergeProfilesCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode !== 200 && output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const contents = (0, import_smithy_client.map)({
    $metadata: deserializeMetadata(output)
  });
  const data = (0, import_smithy_client.expectNonNull)((0, import_smithy_client.expectObject)(await parseBody(output.body, context)), "body");
  const doc = (0, import_smithy_client.take)(data, {
    Message: import_smithy_client.expectString
  });
  Object.assign(contents, doc);
  return contents;
}, "de_MergeProfilesCommand");
var de_PutIntegrationCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode !== 200 && output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const contents = (0, import_smithy_client.map)({
    $metadata: deserializeMetadata(output)
  });
  const data = (0, import_smithy_client.expectNonNull)((0, import_smithy_client.expectObject)(await parseBody(output.body, context)), "body");
  const doc = (0, import_smithy_client.take)(data, {
    CreatedAt: (_) => (0, import_smithy_client.expectNonNull)((0, import_smithy_client.parseEpochTimestamp)((0, import_smithy_client.expectNumber)(_))),
    DomainName: import_smithy_client.expectString,
    IsUnstructured: import_smithy_client.expectBoolean,
    LastUpdatedAt: (_) => (0, import_smithy_client.expectNonNull)((0, import_smithy_client.parseEpochTimestamp)((0, import_smithy_client.expectNumber)(_))),
    ObjectTypeName: import_smithy_client.expectString,
    ObjectTypeNames: import_smithy_client._json,
    Tags: import_smithy_client._json,
    Uri: import_smithy_client.expectString,
    WorkflowId: import_smithy_client.expectString
  });
  Object.assign(contents, doc);
  return contents;
}, "de_PutIntegrationCommand");
var de_PutProfileObjectCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode !== 200 && output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const contents = (0, import_smithy_client.map)({
    $metadata: deserializeMetadata(output)
  });
  const data = (0, import_smithy_client.expectNonNull)((0, import_smithy_client.expectObject)(await parseBody(output.body, context)), "body");
  const doc = (0, import_smithy_client.take)(data, {
    ProfileObjectUniqueKey: import_smithy_client.expectString
  });
  Object.assign(contents, doc);
  return contents;
}, "de_PutProfileObjectCommand");
var de_PutProfileObjectTypeCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode !== 200 && output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const contents = (0, import_smithy_client.map)({
    $metadata: deserializeMetadata(output)
  });
  const data = (0, import_smithy_client.expectNonNull)((0, import_smithy_client.expectObject)(await parseBody(output.body, context)), "body");
  const doc = (0, import_smithy_client.take)(data, {
    AllowProfileCreation: import_smithy_client.expectBoolean,
    CreatedAt: (_) => (0, import_smithy_client.expectNonNull)((0, import_smithy_client.parseEpochTimestamp)((0, import_smithy_client.expectNumber)(_))),
    Description: import_smithy_client.expectString,
    EncryptionKey: import_smithy_client.expectString,
    ExpirationDays: import_smithy_client.expectInt32,
    Fields: import_smithy_client._json,
    Keys: import_smithy_client._json,
    LastUpdatedAt: (_) => (0, import_smithy_client.expectNonNull)((0, import_smithy_client.parseEpochTimestamp)((0, import_smithy_client.expectNumber)(_))),
    ObjectTypeName: import_smithy_client.expectString,
    SourceLastUpdatedTimestampFormat: import_smithy_client.expectString,
    Tags: import_smithy_client._json,
    TemplateId: import_smithy_client.expectString
  });
  Object.assign(contents, doc);
  return contents;
}, "de_PutProfileObjectTypeCommand");
var de_SearchProfilesCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode !== 200 && output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const contents = (0, import_smithy_client.map)({
    $metadata: deserializeMetadata(output)
  });
  const data = (0, import_smithy_client.expectNonNull)((0, import_smithy_client.expectObject)(await parseBody(output.body, context)), "body");
  const doc = (0, import_smithy_client.take)(data, {
    Items: import_smithy_client._json,
    NextToken: import_smithy_client.expectString
  });
  Object.assign(contents, doc);
  return contents;
}, "de_SearchProfilesCommand");
var de_TagResourceCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode !== 200 && output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const contents = (0, import_smithy_client.map)({
    $metadata: deserializeMetadata(output)
  });
  await (0, import_smithy_client.collectBody)(output.body, context);
  return contents;
}, "de_TagResourceCommand");
var de_UntagResourceCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode !== 200 && output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const contents = (0, import_smithy_client.map)({
    $metadata: deserializeMetadata(output)
  });
  await (0, import_smithy_client.collectBody)(output.body, context);
  return contents;
}, "de_UntagResourceCommand");
var de_UpdateCalculatedAttributeDefinitionCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode !== 200 && output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const contents = (0, import_smithy_client.map)({
    $metadata: deserializeMetadata(output)
  });
  const data = (0, import_smithy_client.expectNonNull)((0, import_smithy_client.expectObject)(await parseBody(output.body, context)), "body");
  const doc = (0, import_smithy_client.take)(data, {
    AttributeDetails: import_smithy_client._json,
    CalculatedAttributeName: import_smithy_client.expectString,
    Conditions: import_smithy_client._json,
    CreatedAt: (_) => (0, import_smithy_client.expectNonNull)((0, import_smithy_client.parseEpochTimestamp)((0, import_smithy_client.expectNumber)(_))),
    Description: import_smithy_client.expectString,
    DisplayName: import_smithy_client.expectString,
    LastUpdatedAt: (_) => (0, import_smithy_client.expectNonNull)((0, import_smithy_client.parseEpochTimestamp)((0, import_smithy_client.expectNumber)(_))),
    Statistic: import_smithy_client.expectString,
    Tags: import_smithy_client._json
  });
  Object.assign(contents, doc);
  return contents;
}, "de_UpdateCalculatedAttributeDefinitionCommand");
var de_UpdateDomainCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode !== 200 && output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const contents = (0, import_smithy_client.map)({
    $metadata: deserializeMetadata(output)
  });
  const data = (0, import_smithy_client.expectNonNull)((0, import_smithy_client.expectObject)(await parseBody(output.body, context)), "body");
  const doc = (0, import_smithy_client.take)(data, {
    CreatedAt: (_) => (0, import_smithy_client.expectNonNull)((0, import_smithy_client.parseEpochTimestamp)((0, import_smithy_client.expectNumber)(_))),
    DeadLetterQueueUrl: import_smithy_client.expectString,
    DefaultEncryptionKey: import_smithy_client.expectString,
    DefaultExpirationDays: import_smithy_client.expectInt32,
    DomainName: import_smithy_client.expectString,
    LastUpdatedAt: (_) => (0, import_smithy_client.expectNonNull)((0, import_smithy_client.parseEpochTimestamp)((0, import_smithy_client.expectNumber)(_))),
    Matching: (_) => de_MatchingResponse(_, context),
    RuleBasedMatching: import_smithy_client._json,
    Tags: import_smithy_client._json
  });
  Object.assign(contents, doc);
  return contents;
}, "de_UpdateDomainCommand");
var de_UpdateProfileCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode !== 200 && output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const contents = (0, import_smithy_client.map)({
    $metadata: deserializeMetadata(output)
  });
  const data = (0, import_smithy_client.expectNonNull)((0, import_smithy_client.expectObject)(await parseBody(output.body, context)), "body");
  const doc = (0, import_smithy_client.take)(data, {
    ProfileId: import_smithy_client.expectString
  });
  Object.assign(contents, doc);
  return contents;
}, "de_UpdateProfileCommand");
var de_CommandError = /* @__PURE__ */ __name(async (output, context) => {
  const parsedOutput = {
    ...output,
    body: await parseErrorBody(output.body, context)
  };
  const errorCode = loadRestJsonErrorCode(output, parsedOutput.body);
  switch (errorCode) {
    case "AccessDeniedException":
    case "com.amazonaws.customerprofiles#AccessDeniedException":
      throw await de_AccessDeniedExceptionRes(parsedOutput, context);
    case "BadRequestException":
    case "com.amazonaws.customerprofiles#BadRequestException":
      throw await de_BadRequestExceptionRes(parsedOutput, context);
    case "InternalServerException":
    case "com.amazonaws.customerprofiles#InternalServerException":
      throw await de_InternalServerExceptionRes(parsedOutput, context);
    case "ResourceNotFoundException":
    case "com.amazonaws.customerprofiles#ResourceNotFoundException":
      throw await de_ResourceNotFoundExceptionRes(parsedOutput, context);
    case "ThrottlingException":
    case "com.amazonaws.customerprofiles#ThrottlingException":
      throw await de_ThrottlingExceptionRes(parsedOutput, context);
    default:
      const parsedBody = parsedOutput.body;
      return throwDefaultError({
        output,
        parsedBody,
        errorCode
      });
  }
}, "de_CommandError");
var throwDefaultError = (0, import_smithy_client.withBaseException)(CustomerProfilesServiceException);
var de_AccessDeniedExceptionRes = /* @__PURE__ */ __name(async (parsedOutput, context) => {
  const contents = (0, import_smithy_client.map)({});
  const data = parsedOutput.body;
  const doc = (0, import_smithy_client.take)(data, {
    Message: import_smithy_client.expectString
  });
  Object.assign(contents, doc);
  const exception = new AccessDeniedException({
    $metadata: deserializeMetadata(parsedOutput),
    ...contents
  });
  return (0, import_smithy_client.decorateServiceException)(exception, parsedOutput.body);
}, "de_AccessDeniedExceptionRes");
var de_BadRequestExceptionRes = /* @__PURE__ */ __name(async (parsedOutput, context) => {
  const contents = (0, import_smithy_client.map)({});
  const data = parsedOutput.body;
  const doc = (0, import_smithy_client.take)(data, {
    Message: import_smithy_client.expectString
  });
  Object.assign(contents, doc);
  const exception = new BadRequestException({
    $metadata: deserializeMetadata(parsedOutput),
    ...contents
  });
  return (0, import_smithy_client.decorateServiceException)(exception, parsedOutput.body);
}, "de_BadRequestExceptionRes");
var de_InternalServerExceptionRes = /* @__PURE__ */ __name(async (parsedOutput, context) => {
  const contents = (0, import_smithy_client.map)({});
  const data = parsedOutput.body;
  const doc = (0, import_smithy_client.take)(data, {
    Message: import_smithy_client.expectString
  });
  Object.assign(contents, doc);
  const exception = new InternalServerException({
    $metadata: deserializeMetadata(parsedOutput),
    ...contents
  });
  return (0, import_smithy_client.decorateServiceException)(exception, parsedOutput.body);
}, "de_InternalServerExceptionRes");
var de_ResourceNotFoundExceptionRes = /* @__PURE__ */ __name(async (parsedOutput, context) => {
  const contents = (0, import_smithy_client.map)({});
  const data = parsedOutput.body;
  const doc = (0, import_smithy_client.take)(data, {
    Message: import_smithy_client.expectString
  });
  Object.assign(contents, doc);
  const exception = new ResourceNotFoundException({
    $metadata: deserializeMetadata(parsedOutput),
    ...contents
  });
  return (0, import_smithy_client.decorateServiceException)(exception, parsedOutput.body);
}, "de_ResourceNotFoundExceptionRes");
var de_ThrottlingExceptionRes = /* @__PURE__ */ __name(async (parsedOutput, context) => {
  const contents = (0, import_smithy_client.map)({});
  const data = parsedOutput.body;
  const doc = (0, import_smithy_client.take)(data, {
    Message: import_smithy_client.expectString
  });
  Object.assign(contents, doc);
  const exception = new ThrottlingException({
    $metadata: deserializeMetadata(parsedOutput),
    ...contents
  });
  return (0, import_smithy_client.decorateServiceException)(exception, parsedOutput.body);
}, "de_ThrottlingExceptionRes");
var se_AppflowIntegration = /* @__PURE__ */ __name((input, context) => {
  return (0, import_smithy_client.take)(input, {
    Batches: (_) => se_Batches(_, context),
    FlowDefinition: (_) => se_FlowDefinition(_, context)
  });
}, "se_AppflowIntegration");
var se_AutoMerging = /* @__PURE__ */ __name((input, context) => {
  return (0, import_smithy_client.take)(input, {
    ConflictResolution: import_smithy_client._json,
    Consolidation: import_smithy_client._json,
    Enabled: [],
    MinAllowedConfidenceScoreForMerging: import_smithy_client.serializeFloat
  });
}, "se_AutoMerging");
var se_Batch = /* @__PURE__ */ __name((input, context) => {
  return (0, import_smithy_client.take)(input, {
    EndTime: (_) => Math.round(_.getTime() / 1e3),
    StartTime: (_) => Math.round(_.getTime() / 1e3)
  });
}, "se_Batch");
var se_Batches = /* @__PURE__ */ __name((input, context) => {
  return input.filter((e) => e != null).map((entry) => {
    return se_Batch(entry, context);
  });
}, "se_Batches");
var se_FlowDefinition = /* @__PURE__ */ __name((input, context) => {
  return (0, import_smithy_client.take)(input, {
    Description: [],
    FlowName: [],
    KmsArn: [],
    SourceFlowConfig: import_smithy_client._json,
    Tasks: import_smithy_client._json,
    TriggerConfig: (_) => se_TriggerConfig(_, context)
  });
}, "se_FlowDefinition");
var se_IntegrationConfig = /* @__PURE__ */ __name((input, context) => {
  return (0, import_smithy_client.take)(input, {
    AppflowIntegration: (_) => se_AppflowIntegration(_, context)
  });
}, "se_IntegrationConfig");
var se_MatchingRequest = /* @__PURE__ */ __name((input, context) => {
  return (0, import_smithy_client.take)(input, {
    AutoMerging: (_) => se_AutoMerging(_, context),
    Enabled: [],
    ExportingConfig: import_smithy_client._json,
    JobSchedule: import_smithy_client._json
  });
}, "se_MatchingRequest");
var se_ScheduledTriggerProperties = /* @__PURE__ */ __name((input, context) => {
  return (0, import_smithy_client.take)(input, {
    DataPullMode: [],
    FirstExecutionFrom: (_) => Math.round(_.getTime() / 1e3),
    ScheduleEndTime: (_) => Math.round(_.getTime() / 1e3),
    ScheduleExpression: [],
    ScheduleOffset: [],
    ScheduleStartTime: (_) => Math.round(_.getTime() / 1e3),
    Timezone: []
  });
}, "se_ScheduledTriggerProperties");
var se_TriggerConfig = /* @__PURE__ */ __name((input, context) => {
  return (0, import_smithy_client.take)(input, {
    TriggerProperties: (_) => se_TriggerProperties(_, context),
    TriggerType: []
  });
}, "se_TriggerConfig");
var se_TriggerProperties = /* @__PURE__ */ __name((input, context) => {
  return (0, import_smithy_client.take)(input, {
    Scheduled: (_) => se_ScheduledTriggerProperties(_, context)
  });
}, "se_TriggerProperties");
var de_AppflowIntegrationWorkflowStep = /* @__PURE__ */ __name((output, context) => {
  return (0, import_smithy_client.take)(output, {
    BatchRecordsEndTime: import_smithy_client.expectString,
    BatchRecordsStartTime: import_smithy_client.expectString,
    CreatedAt: (_) => (0, import_smithy_client.expectNonNull)((0, import_smithy_client.parseEpochTimestamp)((0, import_smithy_client.expectNumber)(_))),
    ExecutionMessage: import_smithy_client.expectString,
    FlowName: import_smithy_client.expectString,
    LastUpdatedAt: (_) => (0, import_smithy_client.expectNonNull)((0, import_smithy_client.parseEpochTimestamp)((0, import_smithy_client.expectNumber)(_))),
    RecordsProcessed: import_smithy_client.expectLong,
    Status: import_smithy_client.expectString
  });
}, "de_AppflowIntegrationWorkflowStep");
var de_AutoMerging = /* @__PURE__ */ __name((output, context) => {
  return (0, import_smithy_client.take)(output, {
    ConflictResolution: import_smithy_client._json,
    Consolidation: import_smithy_client._json,
    Enabled: import_smithy_client.expectBoolean,
    MinAllowedConfidenceScoreForMerging: import_smithy_client.limitedParseDouble
  });
}, "de_AutoMerging");
var de_CalculatedAttributeDefinitionsList = /* @__PURE__ */ __name((output, context) => {
  const retVal = (output || []).filter((e) => e != null).map((entry) => {
    return de_ListCalculatedAttributeDefinitionItem(entry, context);
  });
  return retVal;
}, "de_CalculatedAttributeDefinitionsList");
var de_DestinationSummary = /* @__PURE__ */ __name((output, context) => {
  return (0, import_smithy_client.take)(output, {
    Status: import_smithy_client.expectString,
    UnhealthySince: (_) => (0, import_smithy_client.expectNonNull)((0, import_smithy_client.parseEpochTimestamp)((0, import_smithy_client.expectNumber)(_))),
    Uri: import_smithy_client.expectString
  });
}, "de_DestinationSummary");
var de_DomainList = /* @__PURE__ */ __name((output, context) => {
  const retVal = (output || []).filter((e) => e != null).map((entry) => {
    return de_ListDomainItem(entry, context);
  });
  return retVal;
}, "de_DomainList");
var de_EventStreamDestinationDetails = /* @__PURE__ */ __name((output, context) => {
  return (0, import_smithy_client.take)(output, {
    Message: import_smithy_client.expectString,
    Status: import_smithy_client.expectString,
    UnhealthySince: (_) => (0, import_smithy_client.expectNonNull)((0, import_smithy_client.parseEpochTimestamp)((0, import_smithy_client.expectNumber)(_))),
    Uri: import_smithy_client.expectString
  });
}, "de_EventStreamDestinationDetails");
var de_EventStreamSummary = /* @__PURE__ */ __name((output, context) => {
  return (0, import_smithy_client.take)(output, {
    DestinationSummary: (_) => de_DestinationSummary(_, context),
    DomainName: import_smithy_client.expectString,
    EventStreamArn: import_smithy_client.expectString,
    EventStreamName: import_smithy_client.expectString,
    State: import_smithy_client.expectString,
    StoppedSince: (_) => (0, import_smithy_client.expectNonNull)((0, import_smithy_client.parseEpochTimestamp)((0, import_smithy_client.expectNumber)(_))),
    Tags: import_smithy_client._json
  });
}, "de_EventStreamSummary");
var de_EventStreamSummaryList = /* @__PURE__ */ __name((output, context) => {
  const retVal = (output || []).filter((e) => e != null).map((entry) => {
    return de_EventStreamSummary(entry, context);
  });
  return retVal;
}, "de_EventStreamSummaryList");
var de_IdentityResolutionJob = /* @__PURE__ */ __name((output, context) => {
  return (0, import_smithy_client.take)(output, {
    DomainName: import_smithy_client.expectString,
    ExportingLocation: import_smithy_client._json,
    JobEndTime: (_) => (0, import_smithy_client.expectNonNull)((0, import_smithy_client.parseEpochTimestamp)((0, import_smithy_client.expectNumber)(_))),
    JobId: import_smithy_client.expectString,
    JobStartTime: (_) => (0, import_smithy_client.expectNonNull)((0, import_smithy_client.parseEpochTimestamp)((0, import_smithy_client.expectNumber)(_))),
    JobStats: import_smithy_client._json,
    Message: import_smithy_client.expectString,
    Status: import_smithy_client.expectString
  });
}, "de_IdentityResolutionJob");
var de_IdentityResolutionJobsList = /* @__PURE__ */ __name((output, context) => {
  const retVal = (output || []).filter((e) => e != null).map((entry) => {
    return de_IdentityResolutionJob(entry, context);
  });
  return retVal;
}, "de_IdentityResolutionJobsList");
var de_IntegrationList = /* @__PURE__ */ __name((output, context) => {
  const retVal = (output || []).filter((e) => e != null).map((entry) => {
    return de_ListIntegrationItem(entry, context);
  });
  return retVal;
}, "de_IntegrationList");
var de_ListCalculatedAttributeDefinitionItem = /* @__PURE__ */ __name((output, context) => {
  return (0, import_smithy_client.take)(output, {
    CalculatedAttributeName: import_smithy_client.expectString,
    CreatedAt: (_) => (0, import_smithy_client.expectNonNull)((0, import_smithy_client.parseEpochTimestamp)((0, import_smithy_client.expectNumber)(_))),
    Description: import_smithy_client.expectString,
    DisplayName: import_smithy_client.expectString,
    LastUpdatedAt: (_) => (0, import_smithy_client.expectNonNull)((0, import_smithy_client.parseEpochTimestamp)((0, import_smithy_client.expectNumber)(_))),
    Tags: import_smithy_client._json
  });
}, "de_ListCalculatedAttributeDefinitionItem");
var de_ListDomainItem = /* @__PURE__ */ __name((output, context) => {
  return (0, import_smithy_client.take)(output, {
    CreatedAt: (_) => (0, import_smithy_client.expectNonNull)((0, import_smithy_client.parseEpochTimestamp)((0, import_smithy_client.expectNumber)(_))),
    DomainName: import_smithy_client.expectString,
    LastUpdatedAt: (_) => (0, import_smithy_client.expectNonNull)((0, import_smithy_client.parseEpochTimestamp)((0, import_smithy_client.expectNumber)(_))),
    Tags: import_smithy_client._json
  });
}, "de_ListDomainItem");
var de_ListIntegrationItem = /* @__PURE__ */ __name((output, context) => {
  return (0, import_smithy_client.take)(output, {
    CreatedAt: (_) => (0, import_smithy_client.expectNonNull)((0, import_smithy_client.parseEpochTimestamp)((0, import_smithy_client.expectNumber)(_))),
    DomainName: import_smithy_client.expectString,
    IsUnstructured: import_smithy_client.expectBoolean,
    LastUpdatedAt: (_) => (0, import_smithy_client.expectNonNull)((0, import_smithy_client.parseEpochTimestamp)((0, import_smithy_client.expectNumber)(_))),
    ObjectTypeName: import_smithy_client.expectString,
    ObjectTypeNames: import_smithy_client._json,
    Tags: import_smithy_client._json,
    Uri: import_smithy_client.expectString,
    WorkflowId: import_smithy_client.expectString
  });
}, "de_ListIntegrationItem");
var de_ListProfileObjectTypeItem = /* @__PURE__ */ __name((output, context) => {
  return (0, import_smithy_client.take)(output, {
    CreatedAt: (_) => (0, import_smithy_client.expectNonNull)((0, import_smithy_client.parseEpochTimestamp)((0, import_smithy_client.expectNumber)(_))),
    Description: import_smithy_client.expectString,
    LastUpdatedAt: (_) => (0, import_smithy_client.expectNonNull)((0, import_smithy_client.parseEpochTimestamp)((0, import_smithy_client.expectNumber)(_))),
    ObjectTypeName: import_smithy_client.expectString,
    Tags: import_smithy_client._json
  });
}, "de_ListProfileObjectTypeItem");
var de_ListWorkflowsItem = /* @__PURE__ */ __name((output, context) => {
  return (0, import_smithy_client.take)(output, {
    CreatedAt: (_) => (0, import_smithy_client.expectNonNull)((0, import_smithy_client.parseEpochTimestamp)((0, import_smithy_client.expectNumber)(_))),
    LastUpdatedAt: (_) => (0, import_smithy_client.expectNonNull)((0, import_smithy_client.parseEpochTimestamp)((0, import_smithy_client.expectNumber)(_))),
    Status: import_smithy_client.expectString,
    StatusDescription: import_smithy_client.expectString,
    WorkflowId: import_smithy_client.expectString,
    WorkflowType: import_smithy_client.expectString
  });
}, "de_ListWorkflowsItem");
var de_MatchesList = /* @__PURE__ */ __name((output, context) => {
  const retVal = (output || []).filter((e) => e != null).map((entry) => {
    return de_MatchItem(entry, context);
  });
  return retVal;
}, "de_MatchesList");
var de_MatchingResponse = /* @__PURE__ */ __name((output, context) => {
  return (0, import_smithy_client.take)(output, {
    AutoMerging: (_) => de_AutoMerging(_, context),
    Enabled: import_smithy_client.expectBoolean,
    ExportingConfig: import_smithy_client._json,
    JobSchedule: import_smithy_client._json
  });
}, "de_MatchingResponse");
var de_MatchItem = /* @__PURE__ */ __name((output, context) => {
  return (0, import_smithy_client.take)(output, {
    ConfidenceScore: import_smithy_client.limitedParseDouble,
    MatchId: import_smithy_client.expectString,
    ProfileIds: import_smithy_client._json
  });
}, "de_MatchItem");
var de_ProfileObjectTypeList = /* @__PURE__ */ __name((output, context) => {
  const retVal = (output || []).filter((e) => e != null).map((entry) => {
    return de_ListProfileObjectTypeItem(entry, context);
  });
  return retVal;
}, "de_ProfileObjectTypeList");
var de_WorkflowList = /* @__PURE__ */ __name((output, context) => {
  const retVal = (output || []).filter((e) => e != null).map((entry) => {
    return de_ListWorkflowsItem(entry, context);
  });
  return retVal;
}, "de_WorkflowList");
var de_WorkflowStepItem = /* @__PURE__ */ __name((output, context) => {
  return (0, import_smithy_client.take)(output, {
    AppflowIntegration: (_) => de_AppflowIntegrationWorkflowStep(_, context)
  });
}, "de_WorkflowStepItem");
var de_WorkflowStepsList = /* @__PURE__ */ __name((output, context) => {
  const retVal = (output || []).filter((e) => e != null).map((entry) => {
    return de_WorkflowStepItem(entry, context);
  });
  return retVal;
}, "de_WorkflowStepsList");
var deserializeMetadata = /* @__PURE__ */ __name((output) => ({
  httpStatusCode: output.statusCode,
  requestId: output.headers["x-amzn-requestid"] ?? output.headers["x-amzn-request-id"] ?? output.headers["x-amz-request-id"],
  extendedRequestId: output.headers["x-amz-id-2"],
  cfId: output.headers["x-amz-cf-id"]
}), "deserializeMetadata");
var collectBodyString = /* @__PURE__ */ __name((streamBody, context) => (0, import_smithy_client.collectBody)(streamBody, context).then((body) => context.utf8Encoder(body)), "collectBodyString");
var _IH = "IncludeHidden";
var _MR = "MaxResults";
var _NT = "NextToken";
var _ih = "include-hidden";
var _mr = "max-results";
var _nt = "next-token";
var _tK = "tagKeys";
var parseBody = /* @__PURE__ */ __name((streamBody, context) => collectBodyString(streamBody, context).then((encoded) => {
  if (encoded.length) {
    return JSON.parse(encoded);
  }
  return {};
}), "parseBody");
var parseErrorBody = /* @__PURE__ */ __name(async (errorBody, context) => {
  const value = await parseBody(errorBody, context);
  value.message = value.message ?? value.Message;
  return value;
}, "parseErrorBody");
var loadRestJsonErrorCode = /* @__PURE__ */ __name((output, data) => {
  const findKey = /* @__PURE__ */ __name((object, key) => Object.keys(object).find((k) => k.toLowerCase() === key.toLowerCase()), "findKey");
  const sanitizeErrorCode = /* @__PURE__ */ __name((rawValue) => {
    let cleanValue = rawValue;
    if (typeof cleanValue === "number") {
      cleanValue = cleanValue.toString();
    }
    if (cleanValue.indexOf(",") >= 0) {
      cleanValue = cleanValue.split(",")[0];
    }
    if (cleanValue.indexOf(":") >= 0) {
      cleanValue = cleanValue.split(":")[0];
    }
    if (cleanValue.indexOf("#") >= 0) {
      cleanValue = cleanValue.split("#")[1];
    }
    return cleanValue;
  }, "sanitizeErrorCode");
  const headerKey = findKey(output.headers, "x-amzn-errortype");
  if (headerKey !== void 0) {
    return sanitizeErrorCode(output.headers[headerKey]);
  }
  if (data.code !== void 0) {
    return sanitizeErrorCode(data.code);
  }
  if (data["__type"] !== void 0) {
    return sanitizeErrorCode(data["__type"]);
  }
}, "loadRestJsonErrorCode");

// src/commands/AddProfileKeyCommand.ts
var _AddProfileKeyCommand = class _AddProfileKeyCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("CustomerProfiles_20200815", "AddProfileKey", {}).n("CustomerProfilesClient", "AddProfileKeyCommand").f(void 0, void 0).ser(se_AddProfileKeyCommand).de(de_AddProfileKeyCommand).build() {
};
__name(_AddProfileKeyCommand, "AddProfileKeyCommand");
var AddProfileKeyCommand = _AddProfileKeyCommand;

// src/commands/CreateCalculatedAttributeDefinitionCommand.ts




var _CreateCalculatedAttributeDefinitionCommand = class _CreateCalculatedAttributeDefinitionCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("CustomerProfiles_20200815", "CreateCalculatedAttributeDefinition", {}).n("CustomerProfilesClient", "CreateCalculatedAttributeDefinitionCommand").f(
  CreateCalculatedAttributeDefinitionRequestFilterSensitiveLog,
  CreateCalculatedAttributeDefinitionResponseFilterSensitiveLog
).ser(se_CreateCalculatedAttributeDefinitionCommand).de(de_CreateCalculatedAttributeDefinitionCommand).build() {
};
__name(_CreateCalculatedAttributeDefinitionCommand, "CreateCalculatedAttributeDefinitionCommand");
var CreateCalculatedAttributeDefinitionCommand = _CreateCalculatedAttributeDefinitionCommand;

// src/commands/CreateDomainCommand.ts




var _CreateDomainCommand = class _CreateDomainCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("CustomerProfiles_20200815", "CreateDomain", {}).n("CustomerProfilesClient", "CreateDomainCommand").f(void 0, void 0).ser(se_CreateDomainCommand).de(de_CreateDomainCommand).build() {
};
__name(_CreateDomainCommand, "CreateDomainCommand");
var CreateDomainCommand = _CreateDomainCommand;

// src/commands/CreateEventStreamCommand.ts




var _CreateEventStreamCommand = class _CreateEventStreamCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("CustomerProfiles_20200815", "CreateEventStream", {}).n("CustomerProfilesClient", "CreateEventStreamCommand").f(void 0, void 0).ser(se_CreateEventStreamCommand).de(de_CreateEventStreamCommand).build() {
};
__name(_CreateEventStreamCommand, "CreateEventStreamCommand");
var CreateEventStreamCommand = _CreateEventStreamCommand;

// src/commands/CreateIntegrationWorkflowCommand.ts




var _CreateIntegrationWorkflowCommand = class _CreateIntegrationWorkflowCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("CustomerProfiles_20200815", "CreateIntegrationWorkflow", {}).n("CustomerProfilesClient", "CreateIntegrationWorkflowCommand").f(CreateIntegrationWorkflowRequestFilterSensitiveLog, void 0).ser(se_CreateIntegrationWorkflowCommand).de(de_CreateIntegrationWorkflowCommand).build() {
};
__name(_CreateIntegrationWorkflowCommand, "CreateIntegrationWorkflowCommand");
var CreateIntegrationWorkflowCommand = _CreateIntegrationWorkflowCommand;

// src/commands/CreateProfileCommand.ts




var _CreateProfileCommand = class _CreateProfileCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("CustomerProfiles_20200815", "CreateProfile", {}).n("CustomerProfilesClient", "CreateProfileCommand").f(CreateProfileRequestFilterSensitiveLog, void 0).ser(se_CreateProfileCommand).de(de_CreateProfileCommand).build() {
};
__name(_CreateProfileCommand, "CreateProfileCommand");
var CreateProfileCommand = _CreateProfileCommand;

// src/commands/DeleteCalculatedAttributeDefinitionCommand.ts




var _DeleteCalculatedAttributeDefinitionCommand = class _DeleteCalculatedAttributeDefinitionCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("CustomerProfiles_20200815", "DeleteCalculatedAttributeDefinition", {}).n("CustomerProfilesClient", "DeleteCalculatedAttributeDefinitionCommand").f(void 0, void 0).ser(se_DeleteCalculatedAttributeDefinitionCommand).de(de_DeleteCalculatedAttributeDefinitionCommand).build() {
};
__name(_DeleteCalculatedAttributeDefinitionCommand, "DeleteCalculatedAttributeDefinitionCommand");
var DeleteCalculatedAttributeDefinitionCommand = _DeleteCalculatedAttributeDefinitionCommand;

// src/commands/DeleteDomainCommand.ts




var _DeleteDomainCommand = class _DeleteDomainCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("CustomerProfiles_20200815", "DeleteDomain", {}).n("CustomerProfilesClient", "DeleteDomainCommand").f(void 0, void 0).ser(se_DeleteDomainCommand).de(de_DeleteDomainCommand).build() {
};
__name(_DeleteDomainCommand, "DeleteDomainCommand");
var DeleteDomainCommand = _DeleteDomainCommand;

// src/commands/DeleteEventStreamCommand.ts




var _DeleteEventStreamCommand = class _DeleteEventStreamCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("CustomerProfiles_20200815", "DeleteEventStream", {}).n("CustomerProfilesClient", "DeleteEventStreamCommand").f(void 0, void 0).ser(se_DeleteEventStreamCommand).de(de_DeleteEventStreamCommand).build() {
};
__name(_DeleteEventStreamCommand, "DeleteEventStreamCommand");
var DeleteEventStreamCommand = _DeleteEventStreamCommand;

// src/commands/DeleteIntegrationCommand.ts




var _DeleteIntegrationCommand = class _DeleteIntegrationCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("CustomerProfiles_20200815", "DeleteIntegration", {}).n("CustomerProfilesClient", "DeleteIntegrationCommand").f(void 0, void 0).ser(se_DeleteIntegrationCommand).de(de_DeleteIntegrationCommand).build() {
};
__name(_DeleteIntegrationCommand, "DeleteIntegrationCommand");
var DeleteIntegrationCommand = _DeleteIntegrationCommand;

// src/commands/DeleteProfileCommand.ts




var _DeleteProfileCommand = class _DeleteProfileCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("CustomerProfiles_20200815", "DeleteProfile", {}).n("CustomerProfilesClient", "DeleteProfileCommand").f(void 0, void 0).ser(se_DeleteProfileCommand).de(de_DeleteProfileCommand).build() {
};
__name(_DeleteProfileCommand, "DeleteProfileCommand");
var DeleteProfileCommand = _DeleteProfileCommand;

// src/commands/DeleteProfileKeyCommand.ts




var _DeleteProfileKeyCommand = class _DeleteProfileKeyCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("CustomerProfiles_20200815", "DeleteProfileKey", {}).n("CustomerProfilesClient", "DeleteProfileKeyCommand").f(void 0, void 0).ser(se_DeleteProfileKeyCommand).de(de_DeleteProfileKeyCommand).build() {
};
__name(_DeleteProfileKeyCommand, "DeleteProfileKeyCommand");
var DeleteProfileKeyCommand = _DeleteProfileKeyCommand;

// src/commands/DeleteProfileObjectCommand.ts




var _DeleteProfileObjectCommand = class _DeleteProfileObjectCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("CustomerProfiles_20200815", "DeleteProfileObject", {}).n("CustomerProfilesClient", "DeleteProfileObjectCommand").f(void 0, void 0).ser(se_DeleteProfileObjectCommand).de(de_DeleteProfileObjectCommand).build() {
};
__name(_DeleteProfileObjectCommand, "DeleteProfileObjectCommand");
var DeleteProfileObjectCommand = _DeleteProfileObjectCommand;

// src/commands/DeleteProfileObjectTypeCommand.ts




var _DeleteProfileObjectTypeCommand = class _DeleteProfileObjectTypeCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("CustomerProfiles_20200815", "DeleteProfileObjectType", {}).n("CustomerProfilesClient", "DeleteProfileObjectTypeCommand").f(void 0, void 0).ser(se_DeleteProfileObjectTypeCommand).de(de_DeleteProfileObjectTypeCommand).build() {
};
__name(_DeleteProfileObjectTypeCommand, "DeleteProfileObjectTypeCommand");
var DeleteProfileObjectTypeCommand = _DeleteProfileObjectTypeCommand;

// src/commands/DeleteWorkflowCommand.ts




var _DeleteWorkflowCommand = class _DeleteWorkflowCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("CustomerProfiles_20200815", "DeleteWorkflow", {}).n("CustomerProfilesClient", "DeleteWorkflowCommand").f(void 0, void 0).ser(se_DeleteWorkflowCommand).de(de_DeleteWorkflowCommand).build() {
};
__name(_DeleteWorkflowCommand, "DeleteWorkflowCommand");
var DeleteWorkflowCommand = _DeleteWorkflowCommand;

// src/commands/DetectProfileObjectTypeCommand.ts




var _DetectProfileObjectTypeCommand = class _DetectProfileObjectTypeCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("CustomerProfiles_20200815", "DetectProfileObjectType", {}).n("CustomerProfilesClient", "DetectProfileObjectTypeCommand").f(DetectProfileObjectTypeRequestFilterSensitiveLog, DetectProfileObjectTypeResponseFilterSensitiveLog).ser(se_DetectProfileObjectTypeCommand).de(de_DetectProfileObjectTypeCommand).build() {
};
__name(_DetectProfileObjectTypeCommand, "DetectProfileObjectTypeCommand");
var DetectProfileObjectTypeCommand = _DetectProfileObjectTypeCommand;

// src/commands/GetAutoMergingPreviewCommand.ts




var _GetAutoMergingPreviewCommand = class _GetAutoMergingPreviewCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("CustomerProfiles_20200815", "GetAutoMergingPreview", {}).n("CustomerProfilesClient", "GetAutoMergingPreviewCommand").f(void 0, void 0).ser(se_GetAutoMergingPreviewCommand).de(de_GetAutoMergingPreviewCommand).build() {
};
__name(_GetAutoMergingPreviewCommand, "GetAutoMergingPreviewCommand");
var GetAutoMergingPreviewCommand = _GetAutoMergingPreviewCommand;

// src/commands/GetCalculatedAttributeDefinitionCommand.ts




var _GetCalculatedAttributeDefinitionCommand = class _GetCalculatedAttributeDefinitionCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("CustomerProfiles_20200815", "GetCalculatedAttributeDefinition", {}).n("CustomerProfilesClient", "GetCalculatedAttributeDefinitionCommand").f(void 0, GetCalculatedAttributeDefinitionResponseFilterSensitiveLog).ser(se_GetCalculatedAttributeDefinitionCommand).de(de_GetCalculatedAttributeDefinitionCommand).build() {
};
__name(_GetCalculatedAttributeDefinitionCommand, "GetCalculatedAttributeDefinitionCommand");
var GetCalculatedAttributeDefinitionCommand = _GetCalculatedAttributeDefinitionCommand;

// src/commands/GetCalculatedAttributeForProfileCommand.ts




var _GetCalculatedAttributeForProfileCommand = class _GetCalculatedAttributeForProfileCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("CustomerProfiles_20200815", "GetCalculatedAttributeForProfile", {}).n("CustomerProfilesClient", "GetCalculatedAttributeForProfileCommand").f(void 0, void 0).ser(se_GetCalculatedAttributeForProfileCommand).de(de_GetCalculatedAttributeForProfileCommand).build() {
};
__name(_GetCalculatedAttributeForProfileCommand, "GetCalculatedAttributeForProfileCommand");
var GetCalculatedAttributeForProfileCommand = _GetCalculatedAttributeForProfileCommand;

// src/commands/GetDomainCommand.ts




var _GetDomainCommand = class _GetDomainCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("CustomerProfiles_20200815", "GetDomain", {}).n("CustomerProfilesClient", "GetDomainCommand").f(void 0, void 0).ser(se_GetDomainCommand).de(de_GetDomainCommand).build() {
};
__name(_GetDomainCommand, "GetDomainCommand");
var GetDomainCommand = _GetDomainCommand;

// src/commands/GetEventStreamCommand.ts




var _GetEventStreamCommand = class _GetEventStreamCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("CustomerProfiles_20200815", "GetEventStream", {}).n("CustomerProfilesClient", "GetEventStreamCommand").f(void 0, void 0).ser(se_GetEventStreamCommand).de(de_GetEventStreamCommand).build() {
};
__name(_GetEventStreamCommand, "GetEventStreamCommand");
var GetEventStreamCommand = _GetEventStreamCommand;

// src/commands/GetIdentityResolutionJobCommand.ts




var _GetIdentityResolutionJobCommand = class _GetIdentityResolutionJobCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("CustomerProfiles_20200815", "GetIdentityResolutionJob", {}).n("CustomerProfilesClient", "GetIdentityResolutionJobCommand").f(void 0, void 0).ser(se_GetIdentityResolutionJobCommand).de(de_GetIdentityResolutionJobCommand).build() {
};
__name(_GetIdentityResolutionJobCommand, "GetIdentityResolutionJobCommand");
var GetIdentityResolutionJobCommand = _GetIdentityResolutionJobCommand;

// src/commands/GetIntegrationCommand.ts




var _GetIntegrationCommand = class _GetIntegrationCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("CustomerProfiles_20200815", "GetIntegration", {}).n("CustomerProfilesClient", "GetIntegrationCommand").f(void 0, void 0).ser(se_GetIntegrationCommand).de(de_GetIntegrationCommand).build() {
};
__name(_GetIntegrationCommand, "GetIntegrationCommand");
var GetIntegrationCommand = _GetIntegrationCommand;

// src/commands/GetMatchesCommand.ts




var _GetMatchesCommand = class _GetMatchesCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("CustomerProfiles_20200815", "GetMatches", {}).n("CustomerProfilesClient", "GetMatchesCommand").f(void 0, void 0).ser(se_GetMatchesCommand).de(de_GetMatchesCommand).build() {
};
__name(_GetMatchesCommand, "GetMatchesCommand");
var GetMatchesCommand = _GetMatchesCommand;

// src/commands/GetProfileObjectTypeCommand.ts




var _GetProfileObjectTypeCommand = class _GetProfileObjectTypeCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("CustomerProfiles_20200815", "GetProfileObjectType", {}).n("CustomerProfilesClient", "GetProfileObjectTypeCommand").f(void 0, GetProfileObjectTypeResponseFilterSensitiveLog).ser(se_GetProfileObjectTypeCommand).de(de_GetProfileObjectTypeCommand).build() {
};
__name(_GetProfileObjectTypeCommand, "GetProfileObjectTypeCommand");
var GetProfileObjectTypeCommand = _GetProfileObjectTypeCommand;

// src/commands/GetProfileObjectTypeTemplateCommand.ts




var _GetProfileObjectTypeTemplateCommand = class _GetProfileObjectTypeTemplateCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("CustomerProfiles_20200815", "GetProfileObjectTypeTemplate", {}).n("CustomerProfilesClient", "GetProfileObjectTypeTemplateCommand").f(void 0, GetProfileObjectTypeTemplateResponseFilterSensitiveLog).ser(se_GetProfileObjectTypeTemplateCommand).de(de_GetProfileObjectTypeTemplateCommand).build() {
};
__name(_GetProfileObjectTypeTemplateCommand, "GetProfileObjectTypeTemplateCommand");
var GetProfileObjectTypeTemplateCommand = _GetProfileObjectTypeTemplateCommand;

// src/commands/GetSimilarProfilesCommand.ts




var _GetSimilarProfilesCommand = class _GetSimilarProfilesCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("CustomerProfiles_20200815", "GetSimilarProfiles", {}).n("CustomerProfilesClient", "GetSimilarProfilesCommand").f(void 0, void 0).ser(se_GetSimilarProfilesCommand).de(de_GetSimilarProfilesCommand).build() {
};
__name(_GetSimilarProfilesCommand, "GetSimilarProfilesCommand");
var GetSimilarProfilesCommand = _GetSimilarProfilesCommand;

// src/commands/GetWorkflowCommand.ts




var _GetWorkflowCommand = class _GetWorkflowCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("CustomerProfiles_20200815", "GetWorkflow", {}).n("CustomerProfilesClient", "GetWorkflowCommand").f(void 0, void 0).ser(se_GetWorkflowCommand).de(de_GetWorkflowCommand).build() {
};
__name(_GetWorkflowCommand, "GetWorkflowCommand");
var GetWorkflowCommand = _GetWorkflowCommand;

// src/commands/GetWorkflowStepsCommand.ts




var _GetWorkflowStepsCommand = class _GetWorkflowStepsCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("CustomerProfiles_20200815", "GetWorkflowSteps", {}).n("CustomerProfilesClient", "GetWorkflowStepsCommand").f(void 0, void 0).ser(se_GetWorkflowStepsCommand).de(de_GetWorkflowStepsCommand).build() {
};
__name(_GetWorkflowStepsCommand, "GetWorkflowStepsCommand");
var GetWorkflowStepsCommand = _GetWorkflowStepsCommand;

// src/commands/ListAccountIntegrationsCommand.ts




var _ListAccountIntegrationsCommand = class _ListAccountIntegrationsCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("CustomerProfiles_20200815", "ListAccountIntegrations", {}).n("CustomerProfilesClient", "ListAccountIntegrationsCommand").f(void 0, void 0).ser(se_ListAccountIntegrationsCommand).de(de_ListAccountIntegrationsCommand).build() {
};
__name(_ListAccountIntegrationsCommand, "ListAccountIntegrationsCommand");
var ListAccountIntegrationsCommand = _ListAccountIntegrationsCommand;

// src/commands/ListCalculatedAttributeDefinitionsCommand.ts




var _ListCalculatedAttributeDefinitionsCommand = class _ListCalculatedAttributeDefinitionsCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("CustomerProfiles_20200815", "ListCalculatedAttributeDefinitions", {}).n("CustomerProfilesClient", "ListCalculatedAttributeDefinitionsCommand").f(void 0, ListCalculatedAttributeDefinitionsResponseFilterSensitiveLog).ser(se_ListCalculatedAttributeDefinitionsCommand).de(de_ListCalculatedAttributeDefinitionsCommand).build() {
};
__name(_ListCalculatedAttributeDefinitionsCommand, "ListCalculatedAttributeDefinitionsCommand");
var ListCalculatedAttributeDefinitionsCommand = _ListCalculatedAttributeDefinitionsCommand;

// src/commands/ListCalculatedAttributesForProfileCommand.ts




var _ListCalculatedAttributesForProfileCommand = class _ListCalculatedAttributesForProfileCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("CustomerProfiles_20200815", "ListCalculatedAttributesForProfile", {}).n("CustomerProfilesClient", "ListCalculatedAttributesForProfileCommand").f(void 0, void 0).ser(se_ListCalculatedAttributesForProfileCommand).de(de_ListCalculatedAttributesForProfileCommand).build() {
};
__name(_ListCalculatedAttributesForProfileCommand, "ListCalculatedAttributesForProfileCommand");
var ListCalculatedAttributesForProfileCommand = _ListCalculatedAttributesForProfileCommand;

// src/commands/ListDomainsCommand.ts




var _ListDomainsCommand = class _ListDomainsCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("CustomerProfiles_20200815", "ListDomains", {}).n("CustomerProfilesClient", "ListDomainsCommand").f(void 0, void 0).ser(se_ListDomainsCommand).de(de_ListDomainsCommand).build() {
};
__name(_ListDomainsCommand, "ListDomainsCommand");
var ListDomainsCommand = _ListDomainsCommand;

// src/commands/ListEventStreamsCommand.ts




var _ListEventStreamsCommand = class _ListEventStreamsCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("CustomerProfiles_20200815", "ListEventStreams", {}).n("CustomerProfilesClient", "ListEventStreamsCommand").f(void 0, void 0).ser(se_ListEventStreamsCommand).de(de_ListEventStreamsCommand).build() {
};
__name(_ListEventStreamsCommand, "ListEventStreamsCommand");
var ListEventStreamsCommand = _ListEventStreamsCommand;

// src/commands/ListIdentityResolutionJobsCommand.ts




var _ListIdentityResolutionJobsCommand = class _ListIdentityResolutionJobsCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("CustomerProfiles_20200815", "ListIdentityResolutionJobs", {}).n("CustomerProfilesClient", "ListIdentityResolutionJobsCommand").f(void 0, void 0).ser(se_ListIdentityResolutionJobsCommand).de(de_ListIdentityResolutionJobsCommand).build() {
};
__name(_ListIdentityResolutionJobsCommand, "ListIdentityResolutionJobsCommand");
var ListIdentityResolutionJobsCommand = _ListIdentityResolutionJobsCommand;

// src/commands/ListIntegrationsCommand.ts




var _ListIntegrationsCommand = class _ListIntegrationsCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("CustomerProfiles_20200815", "ListIntegrations", {}).n("CustomerProfilesClient", "ListIntegrationsCommand").f(void 0, void 0).ser(se_ListIntegrationsCommand).de(de_ListIntegrationsCommand).build() {
};
__name(_ListIntegrationsCommand, "ListIntegrationsCommand");
var ListIntegrationsCommand = _ListIntegrationsCommand;

// src/commands/ListProfileObjectsCommand.ts




var _ListProfileObjectsCommand = class _ListProfileObjectsCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("CustomerProfiles_20200815", "ListProfileObjects", {}).n("CustomerProfilesClient", "ListProfileObjectsCommand").f(void 0, ListProfileObjectsResponseFilterSensitiveLog).ser(se_ListProfileObjectsCommand).de(de_ListProfileObjectsCommand).build() {
};
__name(_ListProfileObjectsCommand, "ListProfileObjectsCommand");
var ListProfileObjectsCommand = _ListProfileObjectsCommand;

// src/commands/ListProfileObjectTypesCommand.ts




var _ListProfileObjectTypesCommand = class _ListProfileObjectTypesCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("CustomerProfiles_20200815", "ListProfileObjectTypes", {}).n("CustomerProfilesClient", "ListProfileObjectTypesCommand").f(void 0, ListProfileObjectTypesResponseFilterSensitiveLog).ser(se_ListProfileObjectTypesCommand).de(de_ListProfileObjectTypesCommand).build() {
};
__name(_ListProfileObjectTypesCommand, "ListProfileObjectTypesCommand");
var ListProfileObjectTypesCommand = _ListProfileObjectTypesCommand;

// src/commands/ListProfileObjectTypeTemplatesCommand.ts




var _ListProfileObjectTypeTemplatesCommand = class _ListProfileObjectTypeTemplatesCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("CustomerProfiles_20200815", "ListProfileObjectTypeTemplates", {}).n("CustomerProfilesClient", "ListProfileObjectTypeTemplatesCommand").f(void 0, void 0).ser(se_ListProfileObjectTypeTemplatesCommand).de(de_ListProfileObjectTypeTemplatesCommand).build() {
};
__name(_ListProfileObjectTypeTemplatesCommand, "ListProfileObjectTypeTemplatesCommand");
var ListProfileObjectTypeTemplatesCommand = _ListProfileObjectTypeTemplatesCommand;

// src/commands/ListRuleBasedMatchesCommand.ts




var _ListRuleBasedMatchesCommand = class _ListRuleBasedMatchesCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("CustomerProfiles_20200815", "ListRuleBasedMatches", {}).n("CustomerProfilesClient", "ListRuleBasedMatchesCommand").f(void 0, void 0).ser(se_ListRuleBasedMatchesCommand).de(de_ListRuleBasedMatchesCommand).build() {
};
__name(_ListRuleBasedMatchesCommand, "ListRuleBasedMatchesCommand");
var ListRuleBasedMatchesCommand = _ListRuleBasedMatchesCommand;

// src/commands/ListTagsForResourceCommand.ts




var _ListTagsForResourceCommand = class _ListTagsForResourceCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("CustomerProfiles_20200815", "ListTagsForResource", {}).n("CustomerProfilesClient", "ListTagsForResourceCommand").f(void 0, void 0).ser(se_ListTagsForResourceCommand).de(de_ListTagsForResourceCommand).build() {
};
__name(_ListTagsForResourceCommand, "ListTagsForResourceCommand");
var ListTagsForResourceCommand = _ListTagsForResourceCommand;

// src/commands/ListWorkflowsCommand.ts




var _ListWorkflowsCommand = class _ListWorkflowsCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("CustomerProfiles_20200815", "ListWorkflows", {}).n("CustomerProfilesClient", "ListWorkflowsCommand").f(void 0, void 0).ser(se_ListWorkflowsCommand).de(de_ListWorkflowsCommand).build() {
};
__name(_ListWorkflowsCommand, "ListWorkflowsCommand");
var ListWorkflowsCommand = _ListWorkflowsCommand;

// src/commands/MergeProfilesCommand.ts




var _MergeProfilesCommand = class _MergeProfilesCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("CustomerProfiles_20200815", "MergeProfiles", {}).n("CustomerProfilesClient", "MergeProfilesCommand").f(void 0, void 0).ser(se_MergeProfilesCommand).de(de_MergeProfilesCommand).build() {
};
__name(_MergeProfilesCommand, "MergeProfilesCommand");
var MergeProfilesCommand = _MergeProfilesCommand;

// src/commands/PutIntegrationCommand.ts




var _PutIntegrationCommand = class _PutIntegrationCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("CustomerProfiles_20200815", "PutIntegration", {}).n("CustomerProfilesClient", "PutIntegrationCommand").f(PutIntegrationRequestFilterSensitiveLog, void 0).ser(se_PutIntegrationCommand).de(de_PutIntegrationCommand).build() {
};
__name(_PutIntegrationCommand, "PutIntegrationCommand");
var PutIntegrationCommand = _PutIntegrationCommand;

// src/commands/PutProfileObjectCommand.ts




var _PutProfileObjectCommand = class _PutProfileObjectCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("CustomerProfiles_20200815", "PutProfileObject", {}).n("CustomerProfilesClient", "PutProfileObjectCommand").f(PutProfileObjectRequestFilterSensitiveLog, void 0).ser(se_PutProfileObjectCommand).de(de_PutProfileObjectCommand).build() {
};
__name(_PutProfileObjectCommand, "PutProfileObjectCommand");
var PutProfileObjectCommand = _PutProfileObjectCommand;

// src/commands/PutProfileObjectTypeCommand.ts




var _PutProfileObjectTypeCommand = class _PutProfileObjectTypeCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("CustomerProfiles_20200815", "PutProfileObjectType", {}).n("CustomerProfilesClient", "PutProfileObjectTypeCommand").f(PutProfileObjectTypeRequestFilterSensitiveLog, PutProfileObjectTypeResponseFilterSensitiveLog).ser(se_PutProfileObjectTypeCommand).de(de_PutProfileObjectTypeCommand).build() {
};
__name(_PutProfileObjectTypeCommand, "PutProfileObjectTypeCommand");
var PutProfileObjectTypeCommand = _PutProfileObjectTypeCommand;

// src/commands/SearchProfilesCommand.ts




var _SearchProfilesCommand = class _SearchProfilesCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("CustomerProfiles_20200815", "SearchProfiles", {}).n("CustomerProfilesClient", "SearchProfilesCommand").f(void 0, SearchProfilesResponseFilterSensitiveLog).ser(se_SearchProfilesCommand).de(de_SearchProfilesCommand).build() {
};
__name(_SearchProfilesCommand, "SearchProfilesCommand");
var SearchProfilesCommand = _SearchProfilesCommand;

// src/commands/TagResourceCommand.ts




var _TagResourceCommand = class _TagResourceCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("CustomerProfiles_20200815", "TagResource", {}).n("CustomerProfilesClient", "TagResourceCommand").f(void 0, void 0).ser(se_TagResourceCommand).de(de_TagResourceCommand).build() {
};
__name(_TagResourceCommand, "TagResourceCommand");
var TagResourceCommand = _TagResourceCommand;

// src/commands/UntagResourceCommand.ts




var _UntagResourceCommand = class _UntagResourceCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("CustomerProfiles_20200815", "UntagResource", {}).n("CustomerProfilesClient", "UntagResourceCommand").f(void 0, void 0).ser(se_UntagResourceCommand).de(de_UntagResourceCommand).build() {
};
__name(_UntagResourceCommand, "UntagResourceCommand");
var UntagResourceCommand = _UntagResourceCommand;

// src/commands/UpdateCalculatedAttributeDefinitionCommand.ts




var _UpdateCalculatedAttributeDefinitionCommand = class _UpdateCalculatedAttributeDefinitionCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("CustomerProfiles_20200815", "UpdateCalculatedAttributeDefinition", {}).n("CustomerProfilesClient", "UpdateCalculatedAttributeDefinitionCommand").f(
  UpdateCalculatedAttributeDefinitionRequestFilterSensitiveLog,
  UpdateCalculatedAttributeDefinitionResponseFilterSensitiveLog
).ser(se_UpdateCalculatedAttributeDefinitionCommand).de(de_UpdateCalculatedAttributeDefinitionCommand).build() {
};
__name(_UpdateCalculatedAttributeDefinitionCommand, "UpdateCalculatedAttributeDefinitionCommand");
var UpdateCalculatedAttributeDefinitionCommand = _UpdateCalculatedAttributeDefinitionCommand;

// src/commands/UpdateDomainCommand.ts




var _UpdateDomainCommand = class _UpdateDomainCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("CustomerProfiles_20200815", "UpdateDomain", {}).n("CustomerProfilesClient", "UpdateDomainCommand").f(void 0, void 0).ser(se_UpdateDomainCommand).de(de_UpdateDomainCommand).build() {
};
__name(_UpdateDomainCommand, "UpdateDomainCommand");
var UpdateDomainCommand = _UpdateDomainCommand;

// src/commands/UpdateProfileCommand.ts




var _UpdateProfileCommand = class _UpdateProfileCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("CustomerProfiles_20200815", "UpdateProfile", {}).n("CustomerProfilesClient", "UpdateProfileCommand").f(UpdateProfileRequestFilterSensitiveLog, void 0).ser(se_UpdateProfileCommand).de(de_UpdateProfileCommand).build() {
};
__name(_UpdateProfileCommand, "UpdateProfileCommand");
var UpdateProfileCommand = _UpdateProfileCommand;

// src/CustomerProfiles.ts
var commands = {
  AddProfileKeyCommand,
  CreateCalculatedAttributeDefinitionCommand,
  CreateDomainCommand,
  CreateEventStreamCommand,
  CreateIntegrationWorkflowCommand,
  CreateProfileCommand,
  DeleteCalculatedAttributeDefinitionCommand,
  DeleteDomainCommand,
  DeleteEventStreamCommand,
  DeleteIntegrationCommand,
  DeleteProfileCommand,
  DeleteProfileKeyCommand,
  DeleteProfileObjectCommand,
  DeleteProfileObjectTypeCommand,
  DeleteWorkflowCommand,
  DetectProfileObjectTypeCommand,
  GetAutoMergingPreviewCommand,
  GetCalculatedAttributeDefinitionCommand,
  GetCalculatedAttributeForProfileCommand,
  GetDomainCommand,
  GetEventStreamCommand,
  GetIdentityResolutionJobCommand,
  GetIntegrationCommand,
  GetMatchesCommand,
  GetProfileObjectTypeCommand,
  GetProfileObjectTypeTemplateCommand,
  GetSimilarProfilesCommand,
  GetWorkflowCommand,
  GetWorkflowStepsCommand,
  ListAccountIntegrationsCommand,
  ListCalculatedAttributeDefinitionsCommand,
  ListCalculatedAttributesForProfileCommand,
  ListDomainsCommand,
  ListEventStreamsCommand,
  ListIdentityResolutionJobsCommand,
  ListIntegrationsCommand,
  ListProfileObjectsCommand,
  ListProfileObjectTypesCommand,
  ListProfileObjectTypeTemplatesCommand,
  ListRuleBasedMatchesCommand,
  ListTagsForResourceCommand,
  ListWorkflowsCommand,
  MergeProfilesCommand,
  PutIntegrationCommand,
  PutProfileObjectCommand,
  PutProfileObjectTypeCommand,
  SearchProfilesCommand,
  TagResourceCommand,
  UntagResourceCommand,
  UpdateCalculatedAttributeDefinitionCommand,
  UpdateDomainCommand,
  UpdateProfileCommand
};
var _CustomerProfiles = class _CustomerProfiles extends CustomerProfilesClient {
};
__name(_CustomerProfiles, "CustomerProfiles");
var CustomerProfiles = _CustomerProfiles;
(0, import_smithy_client.createAggregatedClient)(commands, CustomerProfiles);

// src/pagination/ListEventStreamsPaginator.ts

var paginateListEventStreams = (0, import_core.createPaginator)(CustomerProfilesClient, ListEventStreamsCommand, "NextToken", "NextToken", "MaxResults");

// src/index.ts
var import_util_endpoints = require("@aws-sdk/util-endpoints");
// Annotate the CommonJS export names for ESM import in node:

0 && (module.exports = {
  CustomerProfilesServiceException,
  __Client,
  CustomerProfilesClient,
  CustomerProfiles,
  $Command,
  AddProfileKeyCommand,
  CreateCalculatedAttributeDefinitionCommand,
  CreateDomainCommand,
  CreateEventStreamCommand,
  CreateIntegrationWorkflowCommand,
  CreateProfileCommand,
  DeleteCalculatedAttributeDefinitionCommand,
  DeleteDomainCommand,
  DeleteEventStreamCommand,
  DeleteIntegrationCommand,
  DeleteProfileCommand,
  DeleteProfileKeyCommand,
  DeleteProfileObjectCommand,
  DeleteProfileObjectTypeCommand,
  DeleteWorkflowCommand,
  DetectProfileObjectTypeCommand,
  GetAutoMergingPreviewCommand,
  GetCalculatedAttributeDefinitionCommand,
  GetCalculatedAttributeForProfileCommand,
  GetDomainCommand,
  GetEventStreamCommand,
  GetIdentityResolutionJobCommand,
  GetIntegrationCommand,
  GetMatchesCommand,
  GetProfileObjectTypeCommand,
  GetProfileObjectTypeTemplateCommand,
  GetSimilarProfilesCommand,
  GetWorkflowCommand,
  GetWorkflowStepsCommand,
  ListAccountIntegrationsCommand,
  ListCalculatedAttributeDefinitionsCommand,
  ListCalculatedAttributesForProfileCommand,
  ListDomainsCommand,
  ListEventStreamsCommand,
  ListIdentityResolutionJobsCommand,
  ListIntegrationsCommand,
  ListProfileObjectTypeTemplatesCommand,
  ListProfileObjectTypesCommand,
  ListProfileObjectsCommand,
  ListRuleBasedMatchesCommand,
  ListTagsForResourceCommand,
  ListWorkflowsCommand,
  MergeProfilesCommand,
  PutIntegrationCommand,
  PutProfileObjectCommand,
  PutProfileObjectTypeCommand,
  SearchProfilesCommand,
  TagResourceCommand,
  UntagResourceCommand,
  UpdateCalculatedAttributeDefinitionCommand,
  UpdateDomainCommand,
  UpdateProfileCommand,
  paginateListEventStreams,
  AccessDeniedException,
  BadRequestException,
  InternalServerException,
  ResourceNotFoundException,
  ThrottlingException,
  SourceConnectorType,
  MarketoConnectorOperator,
  S3ConnectorOperator,
  SalesforceConnectorOperator,
  ServiceNowConnectorOperator,
  ZendeskConnectorOperator,
  OperatorPropertiesKeys,
  TaskType,
  DataPullMode,
  TriggerType,
  Status,
  AttributeMatchingModel,
  ConflictResolvingModel,
  Unit,
  Operator,
  Statistic,
  JobScheduleDayOfTheWeek,
  RuleBasedMatchingStatus,
  WorkflowType,
  Gender,
  PartyType,
  FieldContentType,
  StandardIdentifier,
  EventStreamDestinationStatus,
  EventStreamState,
  IdentityResolutionJobStatus,
  MatchType,
  LogicalOperator,
  AddressFilterSensitiveLog,
  FlowDefinitionFilterSensitiveLog,
  AppflowIntegrationFilterSensitiveLog,
  AttributeDetailsFilterSensitiveLog,
  ListCalculatedAttributeDefinitionItemFilterSensitiveLog,
  ConditionsFilterSensitiveLog,
  CreateCalculatedAttributeDefinitionRequestFilterSensitiveLog,
  CreateCalculatedAttributeDefinitionResponseFilterSensitiveLog,
  IntegrationConfigFilterSensitiveLog,
  CreateIntegrationWorkflowRequestFilterSensitiveLog,
  CreateProfileRequestFilterSensitiveLog,
  DetectProfileObjectTypeRequestFilterSensitiveLog,
  DetectedProfileObjectTypeFilterSensitiveLog,
  DetectProfileObjectTypeResponseFilterSensitiveLog,
  GetCalculatedAttributeDefinitionResponseFilterSensitiveLog,
  GetProfileObjectTypeResponseFilterSensitiveLog,
  GetProfileObjectTypeTemplateResponseFilterSensitiveLog,
  ListCalculatedAttributeDefinitionsResponseFilterSensitiveLog,
  ListProfileObjectsItemFilterSensitiveLog,
  ListProfileObjectsResponseFilterSensitiveLog,
  ListProfileObjectTypesResponseFilterSensitiveLog,
  PutIntegrationRequestFilterSensitiveLog,
  PutProfileObjectRequestFilterSensitiveLog,
  PutProfileObjectTypeRequestFilterSensitiveLog,
  PutProfileObjectTypeResponseFilterSensitiveLog,
  ProfileFilterSensitiveLog,
  SearchProfilesResponseFilterSensitiveLog,
  UpdateCalculatedAttributeDefinitionRequestFilterSensitiveLog,
  UpdateCalculatedAttributeDefinitionResponseFilterSensitiveLog,
  UpdateAddressFilterSensitiveLog,
  UpdateProfileRequestFilterSensitiveLog
});
