// @azure/arm-appinsights/src/models/workbooksMappers.ts
/*
 * Copyright (c) Microsoft Corporation.
 * Licensed under the MIT License.
 *
 * Code generated by Microsoft (R) AutoRest Code Generator.
 * Changes may cause incorrect behavior and will be lost if the code is regenerated.
 */

export {
  ApplicationInsightsComponent,
  ApplicationInsightsComponentProactiveDetectionConfiguration,
  ApplicationInsightsComponentProactiveDetectionConfigurationRuleDefinitions,
  AzureEntityResource,
  BaseResource,
  ComponentLinkedStorageAccounts,
  ComponentsResource,
  ManagedServiceIdentity,
  MyWorkbook,
  MyWorkbookManagedIdentity,
  MyWorkbookResource,
  MyWorkbookUserAssignedIdentities,
  PrivateLinkScopedResource,
  ProxyResource,
  Resource,
  SystemData,
  TrackedResource,
  UserAssignedIdentity,
  WebTest,
  WebTestGeolocation,
  WebTestPropertiesConfiguration,
  WebtestsResource,
  Workbook,
  WorkbookError,
  WorkbookErrorDefinition,
  WorkbookResource,
  WorkbookResourceIdentity,
  WorkbooksListResult,
  WorkbookTemplate,
  WorkbookTemplateGallery,
  WorkbookTemplateLocalizedGallery,
  WorkbookTemplateResource,
  WorkbookUpdateParameters
} from "../models/mappers";
