// @azure/arm-digitaltwins/dist-esm/src/models/index.js
/*
 * Copyright (c) Microsoft Corporation.
 * Licensed under the MIT License.
 *
 * Code generated by Microsoft (R) AutoRest Code Generator.
 * Changes may cause incorrect behavior and will be lost if the code is regenerated.
 */
/** Known values of {@link ProvisioningState} that the service accepts. */
export var KnownProvisioningState;
(function (KnownProvisioningState) {
    /** Provisioning */
    KnownProvisioningState["Provisioning"] = "Provisioning";
    /** Deleting */
    KnownProvisioningState["Deleting"] = "Deleting";
    /** Updating */
    KnownProvisioningState["Updating"] = "Updating";
    /** Succeeded */
    KnownProvisioningState["Succeeded"] = "Succeeded";
    /** Failed */
    KnownProvisioningState["Failed"] = "Failed";
    /** Canceled */
    KnownProvisioningState["Canceled"] = "Canceled";
    /** Deleted */
    KnownProvisioningState["Deleted"] = "Deleted";
    /** Warning */
    KnownProvisioningState["Warning"] = "Warning";
    /** Suspending */
    KnownProvisioningState["Suspending"] = "Suspending";
    /** Restoring */
    KnownProvisioningState["Restoring"] = "Restoring";
    /** Moving */
    KnownProvisioningState["Moving"] = "Moving";
})(KnownProvisioningState || (KnownProvisioningState = {}));
/** Known values of {@link ConnectionPropertiesProvisioningState} that the service accepts. */
export var KnownConnectionPropertiesProvisioningState;
(function (KnownConnectionPropertiesProvisioningState) {
    /** Pending */
    KnownConnectionPropertiesProvisioningState["Pending"] = "Pending";
    /** Approved */
    KnownConnectionPropertiesProvisioningState["Approved"] = "Approved";
    /** Rejected */
    KnownConnectionPropertiesProvisioningState["Rejected"] = "Rejected";
    /** Disconnected */
    KnownConnectionPropertiesProvisioningState["Disconnected"] = "Disconnected";
})(KnownConnectionPropertiesProvisioningState || (KnownConnectionPropertiesProvisioningState = {}));
/** Known values of {@link PrivateLinkServiceConnectionStatus} that the service accepts. */
export var KnownPrivateLinkServiceConnectionStatus;
(function (KnownPrivateLinkServiceConnectionStatus) {
    /** Pending */
    KnownPrivateLinkServiceConnectionStatus["Pending"] = "Pending";
    /** Approved */
    KnownPrivateLinkServiceConnectionStatus["Approved"] = "Approved";
    /** Rejected */
    KnownPrivateLinkServiceConnectionStatus["Rejected"] = "Rejected";
    /** Disconnected */
    KnownPrivateLinkServiceConnectionStatus["Disconnected"] = "Disconnected";
})(KnownPrivateLinkServiceConnectionStatus || (KnownPrivateLinkServiceConnectionStatus = {}));
/** Known values of {@link CreatedByType} that the service accepts. */
export var KnownCreatedByType;
(function (KnownCreatedByType) {
    /** User */
    KnownCreatedByType["User"] = "User";
    /** Application */
    KnownCreatedByType["Application"] = "Application";
    /** ManagedIdentity */
    KnownCreatedByType["ManagedIdentity"] = "ManagedIdentity";
    /** Key */
    KnownCreatedByType["Key"] = "Key";
})(KnownCreatedByType || (KnownCreatedByType = {}));
/** Known values of {@link PublicNetworkAccess} that the service accepts. */
export var KnownPublicNetworkAccess;
(function (KnownPublicNetworkAccess) {
    /** Enabled */
    KnownPublicNetworkAccess["Enabled"] = "Enabled";
    /** Disabled */
    KnownPublicNetworkAccess["Disabled"] = "Disabled";
})(KnownPublicNetworkAccess || (KnownPublicNetworkAccess = {}));
/** Known values of {@link DigitalTwinsIdentityType} that the service accepts. */
export var KnownDigitalTwinsIdentityType;
(function (KnownDigitalTwinsIdentityType) {
    /** None */
    KnownDigitalTwinsIdentityType["None"] = "None";
    /** SystemAssigned */
    KnownDigitalTwinsIdentityType["SystemAssigned"] = "SystemAssigned";
    /** UserAssigned */
    KnownDigitalTwinsIdentityType["UserAssigned"] = "UserAssigned";
    /** SystemAssignedUserAssigned */
    KnownDigitalTwinsIdentityType["SystemAssignedUserAssigned"] = "SystemAssigned,UserAssigned";
})(KnownDigitalTwinsIdentityType || (KnownDigitalTwinsIdentityType = {}));
/** Known values of {@link EndpointType} that the service accepts. */
export var KnownEndpointType;
(function (KnownEndpointType) {
    /** EventHub */
    KnownEndpointType["EventHub"] = "EventHub";
    /** EventGrid */
    KnownEndpointType["EventGrid"] = "EventGrid";
    /** ServiceBus */
    KnownEndpointType["ServiceBus"] = "ServiceBus";
})(KnownEndpointType || (KnownEndpointType = {}));
/** Known values of {@link EndpointProvisioningState} that the service accepts. */
export var KnownEndpointProvisioningState;
(function (KnownEndpointProvisioningState) {
    /** Provisioning */
    KnownEndpointProvisioningState["Provisioning"] = "Provisioning";
    /** Deleting */
    KnownEndpointProvisioningState["Deleting"] = "Deleting";
    /** Updating */
    KnownEndpointProvisioningState["Updating"] = "Updating";
    /** Succeeded */
    KnownEndpointProvisioningState["Succeeded"] = "Succeeded";
    /** Failed */
    KnownEndpointProvisioningState["Failed"] = "Failed";
    /** Canceled */
    KnownEndpointProvisioningState["Canceled"] = "Canceled";
    /** Deleted */
    KnownEndpointProvisioningState["Deleted"] = "Deleted";
    /** Warning */
    KnownEndpointProvisioningState["Warning"] = "Warning";
    /** Suspending */
    KnownEndpointProvisioningState["Suspending"] = "Suspending";
    /** Restoring */
    KnownEndpointProvisioningState["Restoring"] = "Restoring";
    /** Moving */
    KnownEndpointProvisioningState["Moving"] = "Moving";
    /** Disabled */
    KnownEndpointProvisioningState["Disabled"] = "Disabled";
})(KnownEndpointProvisioningState || (KnownEndpointProvisioningState = {}));
/** Known values of {@link AuthenticationType} that the service accepts. */
export var KnownAuthenticationType;
(function (KnownAuthenticationType) {
    /** KeyBased */
    KnownAuthenticationType["KeyBased"] = "KeyBased";
    /** IdentityBased */
    KnownAuthenticationType["IdentityBased"] = "IdentityBased";
})(KnownAuthenticationType || (KnownAuthenticationType = {}));
/** Known values of {@link IdentityType} that the service accepts. */
export var KnownIdentityType;
(function (KnownIdentityType) {
    /** SystemAssigned */
    KnownIdentityType["SystemAssigned"] = "SystemAssigned";
    /** UserAssigned */
    KnownIdentityType["UserAssigned"] = "UserAssigned";
})(KnownIdentityType || (KnownIdentityType = {}));
/** Known values of {@link Reason} that the service accepts. */
export var KnownReason;
(function (KnownReason) {
    /** Invalid */
    KnownReason["Invalid"] = "Invalid";
    /** AlreadyExists */
    KnownReason["AlreadyExists"] = "AlreadyExists";
})(KnownReason || (KnownReason = {}));
/** Known values of {@link ConnectionType} that the service accepts. */
export var KnownConnectionType;
(function (KnownConnectionType) {
    /** AzureDataExplorer */
    KnownConnectionType["AzureDataExplorer"] = "AzureDataExplorer";
})(KnownConnectionType || (KnownConnectionType = {}));
/** Known values of {@link TimeSeriesDatabaseConnectionState} that the service accepts. */
export var KnownTimeSeriesDatabaseConnectionState;
(function (KnownTimeSeriesDatabaseConnectionState) {
    /** Provisioning */
    KnownTimeSeriesDatabaseConnectionState["Provisioning"] = "Provisioning";
    /** Deleting */
    KnownTimeSeriesDatabaseConnectionState["Deleting"] = "Deleting";
    /** Updating */
    KnownTimeSeriesDatabaseConnectionState["Updating"] = "Updating";
    /** Succeeded */
    KnownTimeSeriesDatabaseConnectionState["Succeeded"] = "Succeeded";
    /** Failed */
    KnownTimeSeriesDatabaseConnectionState["Failed"] = "Failed";
    /** Canceled */
    KnownTimeSeriesDatabaseConnectionState["Canceled"] = "Canceled";
    /** Deleted */
    KnownTimeSeriesDatabaseConnectionState["Deleted"] = "Deleted";
    /** Warning */
    KnownTimeSeriesDatabaseConnectionState["Warning"] = "Warning";
    /** Suspending */
    KnownTimeSeriesDatabaseConnectionState["Suspending"] = "Suspending";
    /** Restoring */
    KnownTimeSeriesDatabaseConnectionState["Restoring"] = "Restoring";
    /** Moving */
    KnownTimeSeriesDatabaseConnectionState["Moving"] = "Moving";
    /** Disabled */
    KnownTimeSeriesDatabaseConnectionState["Disabled"] = "Disabled";
})(KnownTimeSeriesDatabaseConnectionState || (KnownTimeSeriesDatabaseConnectionState = {}));
/** Known values of {@link CleanupConnectionArtifacts} that the service accepts. */
export var KnownCleanupConnectionArtifacts;
(function (KnownCleanupConnectionArtifacts) {
    /** True */
    KnownCleanupConnectionArtifacts["True"] = "true";
    /** False */
    KnownCleanupConnectionArtifacts["False"] = "false";
})(KnownCleanupConnectionArtifacts || (KnownCleanupConnectionArtifacts = {}));
/** Known values of {@link RecordPropertyAndItemRemovals} that the service accepts. */
export var KnownRecordPropertyAndItemRemovals;
(function (KnownRecordPropertyAndItemRemovals) {
    /** True */
    KnownRecordPropertyAndItemRemovals["True"] = "true";
    /** False */
    KnownRecordPropertyAndItemRemovals["False"] = "false";
})(KnownRecordPropertyAndItemRemovals || (KnownRecordPropertyAndItemRemovals = {}));
