// @azure/arm-postgresql-flexible/dist-esm/src/models/index.js
/*
 * Copyright (c) Microsoft Corporation.
 * Licensed under the MIT License.
 *
 * Code generated by Microsoft (R) AutoRest Code Generator.
 * Changes may cause incorrect behavior and will be lost if the code is regenerated.
 */
/** Known values of {@link PrincipalType} that the service accepts. */
export var KnownPrincipalType;
(function (KnownPrincipalType) {
    /** Unknown */
    KnownPrincipalType["Unknown"] = "Unknown";
    /** User */
    KnownPrincipalType["User"] = "User";
    /** Group */
    KnownPrincipalType["Group"] = "Group";
    /** ServicePrincipal */
    KnownPrincipalType["ServicePrincipal"] = "ServicePrincipal";
})(KnownPrincipalType || (KnownPrincipalType = {}));
/** Known values of {@link CreatedByType} that the service accepts. */
export var KnownCreatedByType;
(function (KnownCreatedByType) {
    /** User */
    KnownCreatedByType["User"] = "User";
    /** Application */
    KnownCreatedByType["Application"] = "Application";
    /** ManagedIdentity */
    KnownCreatedByType["ManagedIdentity"] = "ManagedIdentity";
    /** Key */
    KnownCreatedByType["Key"] = "Key";
})(KnownCreatedByType || (KnownCreatedByType = {}));
/** Known values of {@link Origin} that the service accepts. */
export var KnownOrigin;
(function (KnownOrigin) {
    /** Full */
    KnownOrigin["Full"] = "Full";
})(KnownOrigin || (KnownOrigin = {}));
/** Known values of {@link CheckNameAvailabilityReason} that the service accepts. */
export var KnownCheckNameAvailabilityReason;
(function (KnownCheckNameAvailabilityReason) {
    /** Invalid */
    KnownCheckNameAvailabilityReason["Invalid"] = "Invalid";
    /** AlreadyExists */
    KnownCheckNameAvailabilityReason["AlreadyExists"] = "AlreadyExists";
})(KnownCheckNameAvailabilityReason || (KnownCheckNameAvailabilityReason = {}));
/** Known values of {@link ConfigurationDataType} that the service accepts. */
export var KnownConfigurationDataType;
(function (KnownConfigurationDataType) {
    /** Boolean */
    KnownConfigurationDataType["Boolean"] = "Boolean";
    /** Numeric */
    KnownConfigurationDataType["Numeric"] = "Numeric";
    /** Integer */
    KnownConfigurationDataType["Integer"] = "Integer";
    /** Enumeration */
    KnownConfigurationDataType["Enumeration"] = "Enumeration";
})(KnownConfigurationDataType || (KnownConfigurationDataType = {}));
/** Known values of {@link SkuTier} that the service accepts. */
export var KnownSkuTier;
(function (KnownSkuTier) {
    /** Burstable */
    KnownSkuTier["Burstable"] = "Burstable";
    /** GeneralPurpose */
    KnownSkuTier["GeneralPurpose"] = "GeneralPurpose";
    /** MemoryOptimized */
    KnownSkuTier["MemoryOptimized"] = "MemoryOptimized";
})(KnownSkuTier || (KnownSkuTier = {}));
/** Known values of {@link IdentityType} that the service accepts. */
export var KnownIdentityType;
(function (KnownIdentityType) {
    /** None */
    KnownIdentityType["None"] = "None";
    /** SystemAssigned */
    KnownIdentityType["SystemAssigned"] = "SystemAssigned";
    /** UserAssigned */
    KnownIdentityType["UserAssigned"] = "UserAssigned";
})(KnownIdentityType || (KnownIdentityType = {}));
/** Known values of {@link ServerVersion} that the service accepts. */
export var KnownServerVersion;
(function (KnownServerVersion) {
    /** Fourteen */
    KnownServerVersion["Fourteen"] = "14";
    /** Thirteen */
    KnownServerVersion["Thirteen"] = "13";
    /** Twelve */
    KnownServerVersion["Twelve"] = "12";
    /** Eleven */
    KnownServerVersion["Eleven"] = "11";
})(KnownServerVersion || (KnownServerVersion = {}));
/** Known values of {@link ServerState} that the service accepts. */
export var KnownServerState;
(function (KnownServerState) {
    /** Ready */
    KnownServerState["Ready"] = "Ready";
    /** Dropping */
    KnownServerState["Dropping"] = "Dropping";
    /** Disabled */
    KnownServerState["Disabled"] = "Disabled";
    /** Starting */
    KnownServerState["Starting"] = "Starting";
    /** Stopping */
    KnownServerState["Stopping"] = "Stopping";
    /** Stopped */
    KnownServerState["Stopped"] = "Stopped";
    /** Updating */
    KnownServerState["Updating"] = "Updating";
})(KnownServerState || (KnownServerState = {}));
/** Known values of {@link ActiveDirectoryAuthEnum} that the service accepts. */
export var KnownActiveDirectoryAuthEnum;
(function (KnownActiveDirectoryAuthEnum) {
    /** Enabled */
    KnownActiveDirectoryAuthEnum["Enabled"] = "Enabled";
    /** Disabled */
    KnownActiveDirectoryAuthEnum["Disabled"] = "Disabled";
})(KnownActiveDirectoryAuthEnum || (KnownActiveDirectoryAuthEnum = {}));
/** Known values of {@link PasswordAuthEnum} that the service accepts. */
export var KnownPasswordAuthEnum;
(function (KnownPasswordAuthEnum) {
    /** Enabled */
    KnownPasswordAuthEnum["Enabled"] = "Enabled";
    /** Disabled */
    KnownPasswordAuthEnum["Disabled"] = "Disabled";
})(KnownPasswordAuthEnum || (KnownPasswordAuthEnum = {}));
/** Known values of {@link ArmServerKeyType} that the service accepts. */
export var KnownArmServerKeyType;
(function (KnownArmServerKeyType) {
    /** SystemManaged */
    KnownArmServerKeyType["SystemManaged"] = "SystemManaged";
    /** AzureKeyVault */
    KnownArmServerKeyType["AzureKeyVault"] = "AzureKeyVault";
})(KnownArmServerKeyType || (KnownArmServerKeyType = {}));
/** Known values of {@link GeoRedundantBackupEnum} that the service accepts. */
export var KnownGeoRedundantBackupEnum;
(function (KnownGeoRedundantBackupEnum) {
    /** Enabled */
    KnownGeoRedundantBackupEnum["Enabled"] = "Enabled";
    /** Disabled */
    KnownGeoRedundantBackupEnum["Disabled"] = "Disabled";
})(KnownGeoRedundantBackupEnum || (KnownGeoRedundantBackupEnum = {}));
/** Known values of {@link ServerPublicNetworkAccessState} that the service accepts. */
export var KnownServerPublicNetworkAccessState;
(function (KnownServerPublicNetworkAccessState) {
    /** Enabled */
    KnownServerPublicNetworkAccessState["Enabled"] = "Enabled";
    /** Disabled */
    KnownServerPublicNetworkAccessState["Disabled"] = "Disabled";
})(KnownServerPublicNetworkAccessState || (KnownServerPublicNetworkAccessState = {}));
/** Known values of {@link HighAvailabilityMode} that the service accepts. */
export var KnownHighAvailabilityMode;
(function (KnownHighAvailabilityMode) {
    /** Disabled */
    KnownHighAvailabilityMode["Disabled"] = "Disabled";
    /** ZoneRedundant */
    KnownHighAvailabilityMode["ZoneRedundant"] = "ZoneRedundant";
    /** SameZone */
    KnownHighAvailabilityMode["SameZone"] = "SameZone";
})(KnownHighAvailabilityMode || (KnownHighAvailabilityMode = {}));
/** Known values of {@link ServerHAState} that the service accepts. */
export var KnownServerHAState;
(function (KnownServerHAState) {
    /** NotEnabled */
    KnownServerHAState["NotEnabled"] = "NotEnabled";
    /** CreatingStandby */
    KnownServerHAState["CreatingStandby"] = "CreatingStandby";
    /** ReplicatingData */
    KnownServerHAState["ReplicatingData"] = "ReplicatingData";
    /** FailingOver */
    KnownServerHAState["FailingOver"] = "FailingOver";
    /** Healthy */
    KnownServerHAState["Healthy"] = "Healthy";
    /** RemovingStandby */
    KnownServerHAState["RemovingStandby"] = "RemovingStandby";
})(KnownServerHAState || (KnownServerHAState = {}));
/** Known values of {@link ReplicationRole} that the service accepts. */
export var KnownReplicationRole;
(function (KnownReplicationRole) {
    /** None */
    KnownReplicationRole["None"] = "None";
    /** Primary */
    KnownReplicationRole["Primary"] = "Primary";
    /** AsyncReplica */
    KnownReplicationRole["AsyncReplica"] = "AsyncReplica";
    /** GeoAsyncReplica */
    KnownReplicationRole["GeoAsyncReplica"] = "GeoAsyncReplica";
})(KnownReplicationRole || (KnownReplicationRole = {}));
/** Known values of {@link CreateMode} that the service accepts. */
export var KnownCreateMode;
(function (KnownCreateMode) {
    /** Default */
    KnownCreateMode["Default"] = "Default";
    /** Create */
    KnownCreateMode["Create"] = "Create";
    /** Update */
    KnownCreateMode["Update"] = "Update";
    /** PointInTimeRestore */
    KnownCreateMode["PointInTimeRestore"] = "PointInTimeRestore";
    /** GeoRestore */
    KnownCreateMode["GeoRestore"] = "GeoRestore";
    /** Replica */
    KnownCreateMode["Replica"] = "Replica";
})(KnownCreateMode || (KnownCreateMode = {}));
/** Known values of {@link CreateModeForUpdate} that the service accepts. */
export var KnownCreateModeForUpdate;
(function (KnownCreateModeForUpdate) {
    /** Default */
    KnownCreateModeForUpdate["Default"] = "Default";
    /** Update */
    KnownCreateModeForUpdate["Update"] = "Update";
})(KnownCreateModeForUpdate || (KnownCreateModeForUpdate = {}));
/** Known values of {@link OperationOrigin} that the service accepts. */
export var KnownOperationOrigin;
(function (KnownOperationOrigin) {
    /** NotSpecified */
    KnownOperationOrigin["NotSpecified"] = "NotSpecified";
    /** User */
    KnownOperationOrigin["User"] = "user";
    /** System */
    KnownOperationOrigin["System"] = "system";
})(KnownOperationOrigin || (KnownOperationOrigin = {}));
/** Known values of {@link FailoverMode} that the service accepts. */
export var KnownFailoverMode;
(function (KnownFailoverMode) {
    /** PlannedFailover */
    KnownFailoverMode["PlannedFailover"] = "PlannedFailover";
    /** ForcedFailover */
    KnownFailoverMode["ForcedFailover"] = "ForcedFailover";
    /** PlannedSwitchover */
    KnownFailoverMode["PlannedSwitchover"] = "PlannedSwitchover";
    /** ForcedSwitchover */
    KnownFailoverMode["ForcedSwitchover"] = "ForcedSwitchover";
})(KnownFailoverMode || (KnownFailoverMode = {}));
