// @azure/arm-postgresql-flexible/dist-esm/src/models/mappers.js
/*
 * Copyright (c) Microsoft Corporation.
 * Licensed under the MIT License.
 *
 * Code generated by Microsoft (R) AutoRest Code Generator.
 * Changes may cause incorrect behavior and will be lost if the code is regenerated.
 */
export const ActiveDirectoryAdministratorAdd = {
    type: {
        name: "Composite",
        className: "ActiveDirectoryAdministratorAdd",
        modelProperties: {
            principalType: {
                serializedName: "properties.principalType",
                type: {
                    name: "String"
                }
            },
            principalName: {
                serializedName: "properties.principalName",
                type: {
                    name: "String"
                }
            },
            tenantId: {
                serializedName: "properties.tenantId",
                type: {
                    name: "String"
                }
            }
        }
    }
};
export const Resource = {
    type: {
        name: "Composite",
        className: "Resource",
        modelProperties: {
            id: {
                serializedName: "id",
                readOnly: true,
                type: {
                    name: "String"
                }
            },
            name: {
                serializedName: "name",
                readOnly: true,
                type: {
                    name: "String"
                }
            },
            type: {
                serializedName: "type",
                readOnly: true,
                type: {
                    name: "String"
                }
            },
            systemData: {
                serializedName: "systemData",
                type: {
                    name: "Composite",
                    className: "SystemData"
                }
            }
        }
    }
};
export const SystemData = {
    type: {
        name: "Composite",
        className: "SystemData",
        modelProperties: {
            createdBy: {
                serializedName: "createdBy",
                type: {
                    name: "String"
                }
            },
            createdByType: {
                serializedName: "createdByType",
                type: {
                    name: "String"
                }
            },
            createdAt: {
                serializedName: "createdAt",
                type: {
                    name: "DateTime"
                }
            },
            lastModifiedBy: {
                serializedName: "lastModifiedBy",
                type: {
                    name: "String"
                }
            },
            lastModifiedByType: {
                serializedName: "lastModifiedByType",
                type: {
                    name: "String"
                }
            },
            lastModifiedAt: {
                serializedName: "lastModifiedAt",
                type: {
                    name: "DateTime"
                }
            }
        }
    }
};
export const ErrorResponse = {
    type: {
        name: "Composite",
        className: "ErrorResponse",
        modelProperties: {
            error: {
                serializedName: "error",
                type: {
                    name: "Composite",
                    className: "ErrorDetail"
                }
            }
        }
    }
};
export const ErrorDetail = {
    type: {
        name: "Composite",
        className: "ErrorDetail",
        modelProperties: {
            code: {
                serializedName: "code",
                readOnly: true,
                type: {
                    name: "String"
                }
            },
            message: {
                serializedName: "message",
                readOnly: true,
                type: {
                    name: "String"
                }
            },
            target: {
                serializedName: "target",
                readOnly: true,
                type: {
                    name: "String"
                }
            },
            details: {
                serializedName: "details",
                readOnly: true,
                type: {
                    name: "Sequence",
                    element: {
                        type: {
                            name: "Composite",
                            className: "ErrorDetail"
                        }
                    }
                }
            },
            additionalInfo: {
                serializedName: "additionalInfo",
                readOnly: true,
                type: {
                    name: "Sequence",
                    element: {
                        type: {
                            name: "Composite",
                            className: "ErrorAdditionalInfo"
                        }
                    }
                }
            }
        }
    }
};
export const ErrorAdditionalInfo = {
    type: {
        name: "Composite",
        className: "ErrorAdditionalInfo",
        modelProperties: {
            type: {
                serializedName: "type",
                readOnly: true,
                type: {
                    name: "String"
                }
            },
            info: {
                serializedName: "info",
                readOnly: true,
                type: {
                    name: "Dictionary",
                    value: { type: { name: "any" } }
                }
            }
        }
    }
};
export const AdministratorListResult = {
    type: {
        name: "Composite",
        className: "AdministratorListResult",
        modelProperties: {
            value: {
                serializedName: "value",
                type: {
                    name: "Sequence",
                    element: {
                        type: {
                            name: "Composite",
                            className: "ActiveDirectoryAdministrator"
                        }
                    }
                }
            },
            nextLink: {
                serializedName: "nextLink",
                type: {
                    name: "String"
                }
            }
        }
    }
};
export const ServerBackupListResult = {
    type: {
        name: "Composite",
        className: "ServerBackupListResult",
        modelProperties: {
            value: {
                serializedName: "value",
                type: {
                    name: "Sequence",
                    element: {
                        type: {
                            name: "Composite",
                            className: "ServerBackup"
                        }
                    }
                }
            },
            nextLink: {
                serializedName: "nextLink",
                type: {
                    name: "String"
                }
            }
        }
    }
};
export const CapabilitiesListResult = {
    type: {
        name: "Composite",
        className: "CapabilitiesListResult",
        modelProperties: {
            value: {
                serializedName: "value",
                readOnly: true,
                type: {
                    name: "Sequence",
                    element: {
                        type: {
                            name: "Composite",
                            className: "CapabilityProperties"
                        }
                    }
                }
            },
            nextLink: {
                serializedName: "nextLink",
                readOnly: true,
                type: {
                    name: "String"
                }
            }
        }
    }
};
export const CapabilityProperties = {
    type: {
        name: "Composite",
        className: "CapabilityProperties",
        modelProperties: {
            zone: {
                serializedName: "zone",
                readOnly: true,
                type: {
                    name: "String"
                }
            },
            supportedHAMode: {
                serializedName: "supportedHAMode",
                readOnly: true,
                type: {
                    name: "Sequence",
                    element: {
                        type: {
                            name: "String"
                        }
                    }
                }
            },
            geoBackupSupported: {
                serializedName: "geoBackupSupported",
                readOnly: true,
                type: {
                    name: "Boolean"
                }
            },
            zoneRedundantHaSupported: {
                serializedName: "zoneRedundantHaSupported",
                readOnly: true,
                type: {
                    name: "Boolean"
                }
            },
            zoneRedundantHaAndGeoBackupSupported: {
                serializedName: "zoneRedundantHaAndGeoBackupSupported",
                readOnly: true,
                type: {
                    name: "Boolean"
                }
            },
            supportedFlexibleServerEditions: {
                serializedName: "supportedFlexibleServerEditions",
                readOnly: true,
                type: {
                    name: "Sequence",
                    element: {
                        type: {
                            name: "Composite",
                            className: "FlexibleServerEditionCapability"
                        }
                    }
                }
            },
            supportedHyperscaleNodeEditions: {
                serializedName: "supportedHyperscaleNodeEditions",
                readOnly: true,
                type: {
                    name: "Sequence",
                    element: {
                        type: {
                            name: "Composite",
                            className: "HyperscaleNodeEditionCapability"
                        }
                    }
                }
            },
            fastProvisioningSupported: {
                serializedName: "fastProvisioningSupported",
                readOnly: true,
                type: {
                    name: "Boolean"
                }
            },
            supportedFastProvisioningEditions: {
                serializedName: "supportedFastProvisioningEditions",
                readOnly: true,
                type: {
                    name: "Sequence",
                    element: {
                        type: {
                            name: "Composite",
                            className: "FastProvisioningEditionCapability"
                        }
                    }
                }
            },
            status: {
                serializedName: "status",
                readOnly: true,
                type: {
                    name: "String"
                }
            }
        }
    }
};
export const FlexibleServerEditionCapability = {
    type: {
        name: "Composite",
        className: "FlexibleServerEditionCapability",
        modelProperties: {
            name: {
                serializedName: "name",
                readOnly: true,
                type: {
                    name: "String"
                }
            },
            supportedStorageEditions: {
                serializedName: "supportedStorageEditions",
                readOnly: true,
                type: {
                    name: "Sequence",
                    element: {
                        type: {
                            name: "Composite",
                            className: "StorageEditionCapability"
                        }
                    }
                }
            },
            supportedServerVersions: {
                serializedName: "supportedServerVersions",
                readOnly: true,
                type: {
                    name: "Sequence",
                    element: {
                        type: {
                            name: "Composite",
                            className: "ServerVersionCapability"
                        }
                    }
                }
            },
            status: {
                serializedName: "status",
                readOnly: true,
                type: {
                    name: "String"
                }
            }
        }
    }
};
export const StorageEditionCapability = {
    type: {
        name: "Composite",
        className: "StorageEditionCapability",
        modelProperties: {
            name: {
                serializedName: "name",
                readOnly: true,
                type: {
                    name: "String"
                }
            },
            supportedStorageMB: {
                serializedName: "supportedStorageMB",
                readOnly: true,
                type: {
                    name: "Sequence",
                    element: {
                        type: {
                            name: "Composite",
                            className: "StorageMBCapability"
                        }
                    }
                }
            },
            status: {
                serializedName: "status",
                readOnly: true,
                type: {
                    name: "String"
                }
            }
        }
    }
};
export const StorageMBCapability = {
    type: {
        name: "Composite",
        className: "StorageMBCapability",
        modelProperties: {
            name: {
                serializedName: "name",
                readOnly: true,
                type: {
                    name: "String"
                }
            },
            supportedIops: {
                serializedName: "supportedIops",
                readOnly: true,
                type: {
                    name: "Number"
                }
            },
            storageSizeMB: {
                serializedName: "storageSizeMB",
                readOnly: true,
                type: {
                    name: "Number"
                }
            },
            supportedUpgradableTierList: {
                serializedName: "supportedUpgradableTierList",
                readOnly: true,
                type: {
                    name: "Sequence",
                    element: {
                        type: {
                            name: "Composite",
                            className: "StorageTierCapability"
                        }
                    }
                }
            },
            status: {
                serializedName: "status",
                readOnly: true,
                type: {
                    name: "String"
                }
            }
        }
    }
};
export const StorageTierCapability = {
    type: {
        name: "Composite",
        className: "StorageTierCapability",
        modelProperties: {
            name: {
                serializedName: "name",
                readOnly: true,
                type: {
                    name: "String"
                }
            },
            tierName: {
                serializedName: "tierName",
                readOnly: true,
                type: {
                    name: "String"
                }
            },
            iops: {
                serializedName: "iops",
                readOnly: true,
                type: {
                    name: "Number"
                }
            },
            isBaseline: {
                serializedName: "isBaseline",
                readOnly: true,
                type: {
                    name: "Boolean"
                }
            },
            status: {
                serializedName: "status",
                readOnly: true,
                type: {
                    name: "String"
                }
            }
        }
    }
};
export const ServerVersionCapability = {
    type: {
        name: "Composite",
        className: "ServerVersionCapability",
        modelProperties: {
            name: {
                serializedName: "name",
                readOnly: true,
                type: {
                    name: "String"
                }
            },
            supportedVersionsToUpgrade: {
                serializedName: "supportedVersionsToUpgrade",
                readOnly: true,
                type: {
                    name: "Sequence",
                    element: {
                        type: {
                            name: "String"
                        }
                    }
                }
            },
            supportedVcores: {
                serializedName: "supportedVcores",
                readOnly: true,
                type: {
                    name: "Sequence",
                    element: {
                        type: {
                            name: "Composite",
                            className: "VcoreCapability"
                        }
                    }
                }
            },
            status: {
                serializedName: "status",
                readOnly: true,
                type: {
                    name: "String"
                }
            }
        }
    }
};
export const VcoreCapability = {
    type: {
        name: "Composite",
        className: "VcoreCapability",
        modelProperties: {
            name: {
                serializedName: "name",
                readOnly: true,
                type: {
                    name: "String"
                }
            },
            vCores: {
                serializedName: "vCores",
                readOnly: true,
                type: {
                    name: "Number"
                }
            },
            supportedIops: {
                serializedName: "supportedIops",
                readOnly: true,
                type: {
                    name: "Number"
                }
            },
            supportedMemoryPerVcoreMB: {
                serializedName: "supportedMemoryPerVcoreMB",
                readOnly: true,
                type: {
                    name: "Number"
                }
            },
            status: {
                serializedName: "status",
                readOnly: true,
                type: {
                    name: "String"
                }
            }
        }
    }
};
export const HyperscaleNodeEditionCapability = {
    type: {
        name: "Composite",
        className: "HyperscaleNodeEditionCapability",
        modelProperties: {
            name: {
                serializedName: "name",
                readOnly: true,
                type: {
                    name: "String"
                }
            },
            supportedStorageEditions: {
                serializedName: "supportedStorageEditions",
                readOnly: true,
                type: {
                    name: "Sequence",
                    element: {
                        type: {
                            name: "Composite",
                            className: "StorageEditionCapability"
                        }
                    }
                }
            },
            supportedServerVersions: {
                serializedName: "supportedServerVersions",
                readOnly: true,
                type: {
                    name: "Sequence",
                    element: {
                        type: {
                            name: "Composite",
                            className: "ServerVersionCapability"
                        }
                    }
                }
            },
            supportedNodeTypes: {
                serializedName: "supportedNodeTypes",
                readOnly: true,
                type: {
                    name: "Sequence",
                    element: {
                        type: {
                            name: "Composite",
                            className: "NodeTypeCapability"
                        }
                    }
                }
            },
            status: {
                serializedName: "status",
                readOnly: true,
                type: {
                    name: "String"
                }
            }
        }
    }
};
export const NodeTypeCapability = {
    type: {
        name: "Composite",
        className: "NodeTypeCapability",
        modelProperties: {
            name: {
                serializedName: "name",
                readOnly: true,
                type: {
                    name: "String"
                }
            },
            nodeType: {
                serializedName: "nodeType",
                readOnly: true,
                type: {
                    name: "String"
                }
            },
            status: {
                serializedName: "status",
                readOnly: true,
                type: {
                    name: "String"
                }
            }
        }
    }
};
export const FastProvisioningEditionCapability = {
    type: {
        name: "Composite",
        className: "FastProvisioningEditionCapability",
        modelProperties: {
            supportedSku: {
                serializedName: "supportedSku",
                readOnly: true,
                type: {
                    name: "String"
                }
            },
            supportedStorageGb: {
                serializedName: "supportedStorageGb",
                readOnly: true,
                type: {
                    name: "Number"
                }
            },
            supportedServerVersions: {
                serializedName: "supportedServerVersions",
                readOnly: true,
                type: {
                    name: "String"
                }
            }
        }
    }
};
export const CheckNameAvailabilityRequest = {
    type: {
        name: "Composite",
        className: "CheckNameAvailabilityRequest",
        modelProperties: {
            name: {
                serializedName: "name",
                type: {
                    name: "String"
                }
            },
            type: {
                serializedName: "type",
                type: {
                    name: "String"
                }
            }
        }
    }
};
export const CheckNameAvailabilityResponse = {
    type: {
        name: "Composite",
        className: "CheckNameAvailabilityResponse",
        modelProperties: {
            nameAvailable: {
                serializedName: "nameAvailable",
                type: {
                    name: "Boolean"
                }
            },
            reason: {
                serializedName: "reason",
                type: {
                    name: "String"
                }
            },
            message: {
                serializedName: "message",
                type: {
                    name: "String"
                }
            }
        }
    }
};
export const ConfigurationListResult = {
    type: {
        name: "Composite",
        className: "ConfigurationListResult",
        modelProperties: {
            value: {
                serializedName: "value",
                type: {
                    name: "Sequence",
                    element: {
                        type: {
                            name: "Composite",
                            className: "Configuration"
                        }
                    }
                }
            },
            nextLink: {
                serializedName: "nextLink",
                type: {
                    name: "String"
                }
            }
        }
    }
};
export const ConfigurationForUpdate = {
    type: {
        name: "Composite",
        className: "ConfigurationForUpdate",
        modelProperties: {
            value: {
                serializedName: "properties.value",
                type: {
                    name: "String"
                }
            },
            description: {
                serializedName: "properties.description",
                readOnly: true,
                type: {
                    name: "String"
                }
            },
            defaultValue: {
                serializedName: "properties.defaultValue",
                readOnly: true,
                type: {
                    name: "String"
                }
            },
            dataType: {
                serializedName: "properties.dataType",
                readOnly: true,
                type: {
                    name: "String"
                }
            },
            allowedValues: {
                serializedName: "properties.allowedValues",
                readOnly: true,
                type: {
                    name: "String"
                }
            },
            source: {
                serializedName: "properties.source",
                type: {
                    name: "String"
                }
            },
            isDynamicConfig: {
                serializedName: "properties.isDynamicConfig",
                readOnly: true,
                type: {
                    name: "Boolean"
                }
            },
            isReadOnly: {
                serializedName: "properties.isReadOnly",
                readOnly: true,
                type: {
                    name: "Boolean"
                }
            },
            isConfigPendingRestart: {
                serializedName: "properties.isConfigPendingRestart",
                readOnly: true,
                type: {
                    name: "Boolean"
                }
            },
            unit: {
                serializedName: "properties.unit",
                readOnly: true,
                type: {
                    name: "String"
                }
            },
            documentationLink: {
                serializedName: "properties.documentationLink",
                readOnly: true,
                type: {
                    name: "String"
                }
            }
        }
    }
};
export const DatabaseListResult = {
    type: {
        name: "Composite",
        className: "DatabaseListResult",
        modelProperties: {
            value: {
                serializedName: "value",
                type: {
                    name: "Sequence",
                    element: {
                        type: {
                            name: "Composite",
                            className: "Database"
                        }
                    }
                }
            },
            nextLink: {
                serializedName: "nextLink",
                type: {
                    name: "String"
                }
            }
        }
    }
};
export const FirewallRuleListResult = {
    type: {
        name: "Composite",
        className: "FirewallRuleListResult",
        modelProperties: {
            value: {
                serializedName: "value",
                type: {
                    name: "Sequence",
                    element: {
                        type: {
                            name: "Composite",
                            className: "FirewallRule"
                        }
                    }
                }
            },
            nextLink: {
                serializedName: "nextLink",
                type: {
                    name: "String"
                }
            }
        }
    }
};
export const Sku = {
    type: {
        name: "Composite",
        className: "Sku",
        modelProperties: {
            name: {
                serializedName: "name",
                required: true,
                type: {
                    name: "String"
                }
            },
            tier: {
                serializedName: "tier",
                required: true,
                type: {
                    name: "String"
                }
            }
        }
    }
};
export const UserAssignedIdentity = {
    type: {
        name: "Composite",
        className: "UserAssignedIdentity",
        modelProperties: {
            userAssignedIdentities: {
                serializedName: "userAssignedIdentities",
                type: {
                    name: "Dictionary",
                    value: { type: { name: "Composite", className: "UserIdentity" } }
                }
            },
            type: {
                serializedName: "type",
                required: true,
                type: {
                    name: "String"
                }
            }
        }
    }
};
export const UserIdentity = {
    type: {
        name: "Composite",
        className: "UserIdentity",
        modelProperties: {
            principalId: {
                serializedName: "principalId",
                type: {
                    name: "String"
                }
            },
            clientId: {
                serializedName: "clientId",
                type: {
                    name: "String"
                }
            }
        }
    }
};
export const Storage = {
    type: {
        name: "Composite",
        className: "Storage",
        modelProperties: {
            storageSizeGB: {
                serializedName: "storageSizeGB",
                type: {
                    name: "Number"
                }
            }
        }
    }
};
export const AuthConfig = {
    type: {
        name: "Composite",
        className: "AuthConfig",
        modelProperties: {
            activeDirectoryAuth: {
                serializedName: "activeDirectoryAuth",
                type: {
                    name: "String"
                }
            },
            passwordAuth: {
                defaultValue: "Enabled",
                serializedName: "passwordAuth",
                type: {
                    name: "String"
                }
            },
            tenantId: {
                defaultValue: "",
                serializedName: "tenantId",
                type: {
                    name: "String"
                }
            }
        }
    }
};
export const DataEncryption = {
    type: {
        name: "Composite",
        className: "DataEncryption",
        modelProperties: {
            primaryKeyURI: {
                serializedName: "primaryKeyURI",
                type: {
                    name: "String"
                }
            },
            primaryUserAssignedIdentityId: {
                serializedName: "primaryUserAssignedIdentityId",
                type: {
                    name: "String"
                }
            },
            type: {
                serializedName: "type",
                type: {
                    name: "String"
                }
            }
        }
    }
};
export const Backup = {
    type: {
        name: "Composite",
        className: "Backup",
        modelProperties: {
            backupRetentionDays: {
                defaultValue: 7,
                serializedName: "backupRetentionDays",
                type: {
                    name: "Number"
                }
            },
            geoRedundantBackup: {
                defaultValue: "Disabled",
                serializedName: "geoRedundantBackup",
                type: {
                    name: "String"
                }
            },
            earliestRestoreDate: {
                serializedName: "earliestRestoreDate",
                readOnly: true,
                type: {
                    name: "DateTime"
                }
            }
        }
    }
};
export const Network = {
    type: {
        name: "Composite",
        className: "Network",
        modelProperties: {
            publicNetworkAccess: {
                serializedName: "publicNetworkAccess",
                readOnly: true,
                type: {
                    name: "String"
                }
            },
            delegatedSubnetResourceId: {
                defaultValue: "",
                serializedName: "delegatedSubnetResourceId",
                type: {
                    name: "String"
                }
            },
            privateDnsZoneArmResourceId: {
                defaultValue: "",
                serializedName: "privateDnsZoneArmResourceId",
                type: {
                    name: "String"
                }
            }
        }
    }
};
export const HighAvailability = {
    type: {
        name: "Composite",
        className: "HighAvailability",
        modelProperties: {
            mode: {
                defaultValue: "Disabled",
                serializedName: "mode",
                type: {
                    name: "String"
                }
            },
            state: {
                serializedName: "state",
                readOnly: true,
                type: {
                    name: "String"
                }
            },
            standbyAvailabilityZone: {
                defaultValue: "",
                serializedName: "standbyAvailabilityZone",
                type: {
                    name: "String"
                }
            }
        }
    }
};
export const MaintenanceWindow = {
    type: {
        name: "Composite",
        className: "MaintenanceWindow",
        modelProperties: {
            customWindow: {
                defaultValue: "Disabled",
                serializedName: "customWindow",
                type: {
                    name: "String"
                }
            },
            startHour: {
                defaultValue: 0,
                serializedName: "startHour",
                type: {
                    name: "Number"
                }
            },
            startMinute: {
                defaultValue: 0,
                serializedName: "startMinute",
                type: {
                    name: "Number"
                }
            },
            dayOfWeek: {
                defaultValue: 0,
                serializedName: "dayOfWeek",
                type: {
                    name: "Number"
                }
            }
        }
    }
};
export const ServerForUpdate = {
    type: {
        name: "Composite",
        className: "ServerForUpdate",
        modelProperties: {
            sku: {
                serializedName: "sku",
                type: {
                    name: "Composite",
                    className: "Sku"
                }
            },
            identity: {
                serializedName: "identity",
                type: {
                    name: "Composite",
                    className: "UserAssignedIdentity"
                }
            },
            tags: {
                serializedName: "tags",
                type: {
                    name: "Dictionary",
                    value: { type: { name: "String" } }
                }
            },
            administratorLoginPassword: {
                serializedName: "properties.administratorLoginPassword",
                type: {
                    name: "String"
                }
            },
            version: {
                serializedName: "properties.version",
                type: {
                    name: "String"
                }
            },
            storage: {
                serializedName: "properties.storage",
                type: {
                    name: "Composite",
                    className: "Storage"
                }
            },
            backup: {
                serializedName: "properties.backup",
                type: {
                    name: "Composite",
                    className: "Backup"
                }
            },
            highAvailability: {
                serializedName: "properties.highAvailability",
                type: {
                    name: "Composite",
                    className: "HighAvailability"
                }
            },
            maintenanceWindow: {
                serializedName: "properties.maintenanceWindow",
                type: {
                    name: "Composite",
                    className: "MaintenanceWindow"
                }
            },
            authConfig: {
                serializedName: "properties.authConfig",
                type: {
                    name: "Composite",
                    className: "AuthConfig"
                }
            },
            dataEncryption: {
                serializedName: "properties.dataEncryption",
                type: {
                    name: "Composite",
                    className: "DataEncryption"
                }
            },
            createMode: {
                serializedName: "properties.createMode",
                type: {
                    name: "String"
                }
            },
            replicationRole: {
                serializedName: "properties.replicationRole",
                type: {
                    name: "String"
                }
            }
        }
    }
};
export const ServerListResult = {
    type: {
        name: "Composite",
        className: "ServerListResult",
        modelProperties: {
            value: {
                serializedName: "value",
                type: {
                    name: "Sequence",
                    element: {
                        type: {
                            name: "Composite",
                            className: "Server"
                        }
                    }
                }
            },
            nextLink: {
                serializedName: "nextLink",
                type: {
                    name: "String"
                }
            }
        }
    }
};
export const OperationListResult = {
    type: {
        name: "Composite",
        className: "OperationListResult",
        modelProperties: {
            value: {
                serializedName: "value",
                type: {
                    name: "Sequence",
                    element: {
                        type: {
                            name: "Composite",
                            className: "Operation"
                        }
                    }
                }
            },
            nextLink: {
                serializedName: "nextLink",
                type: {
                    name: "String"
                }
            }
        }
    }
};
export const Operation = {
    type: {
        name: "Composite",
        className: "Operation",
        modelProperties: {
            name: {
                serializedName: "name",
                readOnly: true,
                type: {
                    name: "String"
                }
            },
            display: {
                serializedName: "display",
                type: {
                    name: "Composite",
                    className: "OperationDisplay"
                }
            },
            isDataAction: {
                serializedName: "isDataAction",
                type: {
                    name: "Boolean"
                }
            },
            origin: {
                serializedName: "origin",
                readOnly: true,
                type: {
                    name: "String"
                }
            },
            properties: {
                serializedName: "properties",
                readOnly: true,
                type: {
                    name: "Dictionary",
                    value: {
                        type: { name: "Dictionary", value: { type: { name: "any" } } }
                    }
                }
            }
        }
    }
};
export const OperationDisplay = {
    type: {
        name: "Composite",
        className: "OperationDisplay",
        modelProperties: {
            provider: {
                serializedName: "provider",
                readOnly: true,
                type: {
                    name: "String"
                }
            },
            resource: {
                serializedName: "resource",
                readOnly: true,
                type: {
                    name: "String"
                }
            },
            operation: {
                serializedName: "operation",
                readOnly: true,
                type: {
                    name: "String"
                }
            },
            description: {
                serializedName: "description",
                readOnly: true,
                type: {
                    name: "String"
                }
            }
        }
    }
};
export const RestartParameter = {
    type: {
        name: "Composite",
        className: "RestartParameter",
        modelProperties: {
            restartWithFailover: {
                serializedName: "restartWithFailover",
                type: {
                    name: "Boolean"
                }
            },
            failoverMode: {
                serializedName: "failoverMode",
                type: {
                    name: "String"
                }
            }
        }
    }
};
export const VirtualNetworkSubnetUsageParameter = {
    type: {
        name: "Composite",
        className: "VirtualNetworkSubnetUsageParameter",
        modelProperties: {
            virtualNetworkArmResourceId: {
                serializedName: "virtualNetworkArmResourceId",
                type: {
                    name: "String"
                }
            }
        }
    }
};
export const VirtualNetworkSubnetUsageResult = {
    type: {
        name: "Composite",
        className: "VirtualNetworkSubnetUsageResult",
        modelProperties: {
            delegatedSubnetsUsage: {
                serializedName: "delegatedSubnetsUsage",
                readOnly: true,
                type: {
                    name: "Sequence",
                    element: {
                        type: {
                            name: "Composite",
                            className: "DelegatedSubnetUsage"
                        }
                    }
                }
            },
            location: {
                serializedName: "location",
                readOnly: true,
                type: {
                    name: "String"
                }
            },
            subscriptionId: {
                serializedName: "subscriptionId",
                readOnly: true,
                type: {
                    name: "String"
                }
            }
        }
    }
};
export const DelegatedSubnetUsage = {
    type: {
        name: "Composite",
        className: "DelegatedSubnetUsage",
        modelProperties: {
            subnetName: {
                serializedName: "subnetName",
                readOnly: true,
                type: {
                    name: "String"
                }
            },
            usage: {
                serializedName: "usage",
                readOnly: true,
                type: {
                    name: "Number"
                }
            }
        }
    }
};
export const ProxyResource = {
    type: {
        name: "Composite",
        className: "ProxyResource",
        modelProperties: Object.assign({}, Resource.type.modelProperties)
    }
};
export const TrackedResource = {
    type: {
        name: "Composite",
        className: "TrackedResource",
        modelProperties: Object.assign(Object.assign({}, Resource.type.modelProperties), { tags: {
                serializedName: "tags",
                type: {
                    name: "Dictionary",
                    value: { type: { name: "String" } }
                }
            }, location: {
                serializedName: "location",
                required: true,
                type: {
                    name: "String"
                }
            } })
    }
};
export const NameAvailability = {
    type: {
        name: "Composite",
        className: "NameAvailability",
        modelProperties: Object.assign(Object.assign({}, CheckNameAvailabilityResponse.type.modelProperties), { name: {
                serializedName: "name",
                readOnly: true,
                type: {
                    name: "String"
                }
            }, type: {
                serializedName: "type",
                readOnly: true,
                type: {
                    name: "String"
                }
            } })
    }
};
export const ActiveDirectoryAdministrator = {
    type: {
        name: "Composite",
        className: "ActiveDirectoryAdministrator",
        modelProperties: Object.assign(Object.assign({}, ProxyResource.type.modelProperties), { principalType: {
                serializedName: "properties.principalType",
                type: {
                    name: "String"
                }
            }, principalName: {
                serializedName: "properties.principalName",
                type: {
                    name: "String"
                }
            }, objectId: {
                serializedName: "properties.objectId",
                type: {
                    name: "String"
                }
            }, tenantId: {
                serializedName: "properties.tenantId",
                type: {
                    name: "String"
                }
            } })
    }
};
export const ServerBackup = {
    type: {
        name: "Composite",
        className: "ServerBackup",
        modelProperties: Object.assign(Object.assign({}, ProxyResource.type.modelProperties), { backupType: {
                serializedName: "properties.backupType",
                type: {
                    name: "String"
                }
            }, completedTime: {
                serializedName: "properties.completedTime",
                type: {
                    name: "DateTime"
                }
            }, source: {
                serializedName: "properties.source",
                type: {
                    name: "String"
                }
            } })
    }
};
export const Configuration = {
    type: {
        name: "Composite",
        className: "Configuration",
        modelProperties: Object.assign(Object.assign({}, ProxyResource.type.modelProperties), { value: {
                serializedName: "properties.value",
                type: {
                    name: "String"
                }
            }, description: {
                serializedName: "properties.description",
                readOnly: true,
                type: {
                    name: "String"
                }
            }, defaultValue: {
                serializedName: "properties.defaultValue",
                readOnly: true,
                type: {
                    name: "String"
                }
            }, dataType: {
                serializedName: "properties.dataType",
                readOnly: true,
                type: {
                    name: "String"
                }
            }, allowedValues: {
                serializedName: "properties.allowedValues",
                readOnly: true,
                type: {
                    name: "String"
                }
            }, source: {
                serializedName: "properties.source",
                type: {
                    name: "String"
                }
            }, isDynamicConfig: {
                serializedName: "properties.isDynamicConfig",
                readOnly: true,
                type: {
                    name: "Boolean"
                }
            }, isReadOnly: {
                serializedName: "properties.isReadOnly",
                readOnly: true,
                type: {
                    name: "Boolean"
                }
            }, isConfigPendingRestart: {
                serializedName: "properties.isConfigPendingRestart",
                readOnly: true,
                type: {
                    name: "Boolean"
                }
            }, unit: {
                serializedName: "properties.unit",
                readOnly: true,
                type: {
                    name: "String"
                }
            }, documentationLink: {
                serializedName: "properties.documentationLink",
                readOnly: true,
                type: {
                    name: "String"
                }
            } })
    }
};
export const Database = {
    type: {
        name: "Composite",
        className: "Database",
        modelProperties: Object.assign(Object.assign({}, ProxyResource.type.modelProperties), { charset: {
                constraints: {
                    Pattern: new RegExp("^[a-zA-Z]+\\w*$")
                },
                serializedName: "properties.charset",
                type: {
                    name: "String"
                }
            }, collation: {
                constraints: {
                    Pattern: new RegExp("^[a-zA-Z\\-]+([. ]|\\w)*$")
                },
                serializedName: "properties.collation",
                type: {
                    name: "String"
                }
            } })
    }
};
export const FirewallRule = {
    type: {
        name: "Composite",
        className: "FirewallRule",
        modelProperties: Object.assign(Object.assign({}, ProxyResource.type.modelProperties), { startIpAddress: {
                constraints: {
                    Pattern: new RegExp("^(([0-9]|[1-9][0-9]|1[0-9]{2}|2[0-4][0-9]|25[0-5])\\.){3}([0-9]|[1-9][0-9]|1[0-9]{2}|2[0-4][0-9]|25[0-5])$")
                },
                serializedName: "properties.startIpAddress",
                required: true,
                type: {
                    name: "String"
                }
            }, endIpAddress: {
                constraints: {
                    Pattern: new RegExp("^(([0-9]|[1-9][0-9]|1[0-9]{2}|2[0-4][0-9]|25[0-5])\\.){3}([0-9]|[1-9][0-9]|1[0-9]{2}|2[0-4][0-9]|25[0-5])$")
                },
                serializedName: "properties.endIpAddress",
                required: true,
                type: {
                    name: "String"
                }
            } })
    }
};
export const Server = {
    type: {
        name: "Composite",
        className: "Server",
        modelProperties: Object.assign(Object.assign({}, TrackedResource.type.modelProperties), { sku: {
                serializedName: "sku",
                type: {
                    name: "Composite",
                    className: "Sku"
                }
            }, identity: {
                serializedName: "identity",
                type: {
                    name: "Composite",
                    className: "UserAssignedIdentity"
                }
            }, administratorLogin: {
                serializedName: "properties.administratorLogin",
                type: {
                    name: "String"
                }
            }, administratorLoginPassword: {
                serializedName: "properties.administratorLoginPassword",
                type: {
                    name: "String"
                }
            }, version: {
                serializedName: "properties.version",
                type: {
                    name: "String"
                }
            }, minorVersion: {
                serializedName: "properties.minorVersion",
                readOnly: true,
                type: {
                    name: "String"
                }
            }, state: {
                serializedName: "properties.state",
                readOnly: true,
                type: {
                    name: "String"
                }
            }, fullyQualifiedDomainName: {
                serializedName: "properties.fullyQualifiedDomainName",
                readOnly: true,
                type: {
                    name: "String"
                }
            }, storage: {
                serializedName: "properties.storage",
                type: {
                    name: "Composite",
                    className: "Storage"
                }
            }, authConfig: {
                serializedName: "properties.authConfig",
                type: {
                    name: "Composite",
                    className: "AuthConfig"
                }
            }, dataEncryption: {
                serializedName: "properties.dataEncryption",
                type: {
                    name: "Composite",
                    className: "DataEncryption"
                }
            }, backup: {
                serializedName: "properties.backup",
                type: {
                    name: "Composite",
                    className: "Backup"
                }
            }, network: {
                serializedName: "properties.network",
                type: {
                    name: "Composite",
                    className: "Network"
                }
            }, highAvailability: {
                serializedName: "properties.highAvailability",
                type: {
                    name: "Composite",
                    className: "HighAvailability"
                }
            }, maintenanceWindow: {
                serializedName: "properties.maintenanceWindow",
                type: {
                    name: "Composite",
                    className: "MaintenanceWindow"
                }
            }, sourceServerResourceId: {
                serializedName: "properties.sourceServerResourceId",
                type: {
                    name: "String"
                }
            }, pointInTimeUTC: {
                serializedName: "properties.pointInTimeUTC",
                type: {
                    name: "DateTime"
                }
            }, availabilityZone: {
                defaultValue: "",
                serializedName: "properties.availabilityZone",
                type: {
                    name: "String"
                }
            }, replicationRole: {
                serializedName: "properties.replicationRole",
                type: {
                    name: "String"
                }
            }, replicaCapacity: {
                serializedName: "properties.replicaCapacity",
                type: {
                    name: "Number"
                }
            }, createMode: {
                serializedName: "properties.createMode",
                type: {
                    name: "String"
                }
            } })
    }
};
export const AdministratorsCreateHeaders = {
    type: {
        name: "Composite",
        className: "AdministratorsCreateHeaders",
        modelProperties: {
            location: {
                serializedName: "location",
                type: {
                    name: "String"
                }
            }
        }
    }
};
export const AdministratorsDeleteHeaders = {
    type: {
        name: "Composite",
        className: "AdministratorsDeleteHeaders",
        modelProperties: {
            location: {
                serializedName: "location",
                type: {
                    name: "String"
                }
            }
        }
    }
};
export const ConfigurationsUpdateHeaders = {
    type: {
        name: "Composite",
        className: "ConfigurationsUpdateHeaders",
        modelProperties: {
            location: {
                serializedName: "location",
                type: {
                    name: "String"
                }
            }
        }
    }
};
export const ConfigurationsPutHeaders = {
    type: {
        name: "Composite",
        className: "ConfigurationsPutHeaders",
        modelProperties: {
            location: {
                serializedName: "location",
                type: {
                    name: "String"
                }
            }
        }
    }
};
export const DatabasesCreateHeaders = {
    type: {
        name: "Composite",
        className: "DatabasesCreateHeaders",
        modelProperties: {
            location: {
                serializedName: "location",
                type: {
                    name: "String"
                }
            }
        }
    }
};
export const DatabasesDeleteHeaders = {
    type: {
        name: "Composite",
        className: "DatabasesDeleteHeaders",
        modelProperties: {
            location: {
                serializedName: "location",
                type: {
                    name: "String"
                }
            }
        }
    }
};
export const FirewallRulesCreateOrUpdateHeaders = {
    type: {
        name: "Composite",
        className: "FirewallRulesCreateOrUpdateHeaders",
        modelProperties: {
            location: {
                serializedName: "location",
                type: {
                    name: "String"
                }
            }
        }
    }
};
export const FirewallRulesDeleteHeaders = {
    type: {
        name: "Composite",
        className: "FirewallRulesDeleteHeaders",
        modelProperties: {
            location: {
                serializedName: "location",
                type: {
                    name: "String"
                }
            }
        }
    }
};
export const ServersCreateHeaders = {
    type: {
        name: "Composite",
        className: "ServersCreateHeaders",
        modelProperties: {
            location: {
                serializedName: "location",
                type: {
                    name: "String"
                }
            }
        }
    }
};
export const ServersUpdateHeaders = {
    type: {
        name: "Composite",
        className: "ServersUpdateHeaders",
        modelProperties: {
            location: {
                serializedName: "location",
                type: {
                    name: "String"
                }
            }
        }
    }
};
export const ServersDeleteHeaders = {
    type: {
        name: "Composite",
        className: "ServersDeleteHeaders",
        modelProperties: {
            location: {
                serializedName: "location",
                type: {
                    name: "String"
                }
            }
        }
    }
};
export const ServersRestartHeaders = {
    type: {
        name: "Composite",
        className: "ServersRestartHeaders",
        modelProperties: {
            location: {
                serializedName: "location",
                type: {
                    name: "String"
                }
            }
        }
    }
};
export const ServersStartHeaders = {
    type: {
        name: "Composite",
        className: "ServersStartHeaders",
        modelProperties: {
            location: {
                serializedName: "location",
                type: {
                    name: "String"
                }
            }
        }
    }
};
export const ServersStopHeaders = {
    type: {
        name: "Composite",
        className: "ServersStopHeaders",
        modelProperties: {
            location: {
                serializedName: "location",
                type: {
                    name: "String"
                }
            }
        }
    }
};
