// @azure/arm-postgresql-flexible/dist-esm/src/models/parameters.js
/*
 * Copyright (c) Microsoft Corporation.
 * Licensed under the MIT License.
 *
 * Code generated by Microsoft (R) AutoRest Code Generator.
 * Changes may cause incorrect behavior and will be lost if the code is regenerated.
 */
import { ActiveDirectoryAdministratorAdd as ActiveDirectoryAdministratorAddMapper, CheckNameAvailabilityRequest as CheckNameAvailabilityRequestMapper, ConfigurationForUpdate as ConfigurationForUpdateMapper, Configuration as ConfigurationMapper, Database as DatabaseMapper, FirewallRule as FirewallRuleMapper, Server as ServerMapper, ServerForUpdate as ServerForUpdateMapper, RestartParameter as RestartParameterMapper, VirtualNetworkSubnetUsageParameter as VirtualNetworkSubnetUsageParameterMapper } from "../models/mappers";
export const contentType = {
    parameterPath: ["options", "contentType"],
    mapper: {
        defaultValue: "application/json",
        isConstant: true,
        serializedName: "Content-Type",
        type: {
            name: "String"
        }
    }
};
export const parameters = {
    parameterPath: "parameters",
    mapper: ActiveDirectoryAdministratorAddMapper
};
export const accept = {
    parameterPath: "accept",
    mapper: {
        defaultValue: "application/json",
        isConstant: true,
        serializedName: "Accept",
        type: {
            name: "String"
        }
    }
};
export const $host = {
    parameterPath: "$host",
    mapper: {
        serializedName: "$host",
        required: true,
        type: {
            name: "String"
        }
    },
    skipEncoding: true
};
export const subscriptionId = {
    parameterPath: "subscriptionId",
    mapper: {
        constraints: {
            MinLength: 1
        },
        serializedName: "subscriptionId",
        required: true,
        type: {
            name: "String"
        }
    }
};
export const resourceGroupName = {
    parameterPath: "resourceGroupName",
    mapper: {
        constraints: {
            MaxLength: 90,
            MinLength: 1
        },
        serializedName: "resourceGroupName",
        required: true,
        type: {
            name: "String"
        }
    }
};
export const serverName = {
    parameterPath: "serverName",
    mapper: {
        constraints: {
            Pattern: new RegExp("^[a-zA-Z0-9]+(-[a-zA-Z0-9]+)*"),
            MaxLength: 63,
            MinLength: 3
        },
        serializedName: "serverName",
        required: true,
        type: {
            name: "String"
        }
    }
};
export const objectId = {
    parameterPath: "objectId",
    mapper: {
        serializedName: "objectId",
        required: true,
        type: {
            name: "String"
        }
    }
};
export const apiVersion = {
    parameterPath: "apiVersion",
    mapper: {
        defaultValue: "2022-12-01",
        isConstant: true,
        serializedName: "api-version",
        type: {
            name: "String"
        }
    }
};
export const nextLink = {
    parameterPath: "nextLink",
    mapper: {
        serializedName: "nextLink",
        required: true,
        type: {
            name: "String"
        }
    },
    skipEncoding: true
};
export const backupName = {
    parameterPath: "backupName",
    mapper: {
        constraints: {
            Pattern: new RegExp("^[-\\w\\._]+$"),
            MinLength: 1
        },
        serializedName: "backupName",
        required: true,
        type: {
            name: "String"
        }
    }
};
export const locationName = {
    parameterPath: "locationName",
    mapper: {
        constraints: {
            Pattern: new RegExp("^[-\\w\\._]+$"),
            MinLength: 1
        },
        serializedName: "locationName",
        required: true,
        type: {
            name: "String"
        }
    }
};
export const nameAvailabilityRequest = {
    parameterPath: "nameAvailabilityRequest",
    mapper: CheckNameAvailabilityRequestMapper
};
export const configurationName = {
    parameterPath: "configurationName",
    mapper: {
        constraints: {
            Pattern: new RegExp("^[-\\w\\._]+$"),
            MinLength: 1
        },
        serializedName: "configurationName",
        required: true,
        type: {
            name: "String"
        }
    }
};
export const parameters1 = {
    parameterPath: "parameters",
    mapper: ConfigurationForUpdateMapper
};
export const parameters2 = {
    parameterPath: "parameters",
    mapper: ConfigurationMapper
};
export const parameters3 = {
    parameterPath: "parameters",
    mapper: DatabaseMapper
};
export const databaseName = {
    parameterPath: "databaseName",
    mapper: {
        constraints: {
            Pattern: new RegExp("^[-\\w\\._]+$"),
            MinLength: 1
        },
        serializedName: "databaseName",
        required: true,
        type: {
            name: "String"
        }
    }
};
export const parameters4 = {
    parameterPath: "parameters",
    mapper: FirewallRuleMapper
};
export const firewallRuleName = {
    parameterPath: "firewallRuleName",
    mapper: {
        constraints: {
            Pattern: new RegExp("^[-\\w\\._]+$"),
            MinLength: 1
        },
        serializedName: "firewallRuleName",
        required: true,
        type: {
            name: "String"
        }
    }
};
export const parameters5 = {
    parameterPath: "parameters",
    mapper: ServerMapper
};
export const parameters6 = {
    parameterPath: "parameters",
    mapper: ServerForUpdateMapper
};
export const parameters7 = {
    parameterPath: ["options", "parameters"],
    mapper: RestartParameterMapper
};
export const parameters8 = {
    parameterPath: "parameters",
    mapper: VirtualNetworkSubnetUsageParameterMapper
};
