// @azure/arm-templatespecs/dist-esm/src/operations/templateSpecVersions.js
/*
 * Copyright (c) Microsoft Corporation.
 * Licensed under the MIT License.
 *
 * Code generated by Microsoft (R) AutoRest Code Generator.
 * Changes may cause incorrect behavior and will be lost if the code is regenerated.
 */
import { __asyncDelegator, __asyncGenerator, __asyncValues, __await } from "tslib";
import { setContinuationToken } from "../pagingHelper";
import * as coreClient from "@azure/core-client";
import * as Mappers from "../models/mappers";
import * as Parameters from "../models/parameters";
/// <reference lib="esnext.asynciterable" />
/** Class containing TemplateSpecVersions operations. */
export class TemplateSpecVersionsImpl {
    /**
     * Initialize a new instance of the class TemplateSpecVersions class.
     * @param client Reference to the service client
     */
    constructor(client) {
        this.client = client;
    }
    /**
     * Lists all the Template Spec versions in the specified Template Spec.
     * @param resourceGroupName The name of the resource group. The name is case insensitive.
     * @param templateSpecName Name of the Template Spec.
     * @param options The options parameters.
     */
    list(resourceGroupName, templateSpecName, options) {
        const iter = this.listPagingAll(resourceGroupName, templateSpecName, options);
        return {
            next() {
                return iter.next();
            },
            [Symbol.asyncIterator]() {
                return this;
            },
            byPage: (settings) => {
                if (settings === null || settings === void 0 ? void 0 : settings.maxPageSize) {
                    throw new Error("maxPageSize is not supported by this operation.");
                }
                return this.listPagingPage(resourceGroupName, templateSpecName, options, settings);
            }
        };
    }
    listPagingPage(resourceGroupName, templateSpecName, options, settings) {
        return __asyncGenerator(this, arguments, function* listPagingPage_1() {
            let result;
            let continuationToken = settings === null || settings === void 0 ? void 0 : settings.continuationToken;
            if (!continuationToken) {
                result = yield __await(this._list(resourceGroupName, templateSpecName, options));
                let page = result.value || [];
                continuationToken = result.nextLink;
                setContinuationToken(page, continuationToken);
                yield yield __await(page);
            }
            while (continuationToken) {
                result = yield __await(this._listNext(resourceGroupName, templateSpecName, continuationToken, options));
                continuationToken = result.nextLink;
                let page = result.value || [];
                setContinuationToken(page, continuationToken);
                yield yield __await(page);
            }
        });
    }
    listPagingAll(resourceGroupName, templateSpecName, options) {
        return __asyncGenerator(this, arguments, function* listPagingAll_1() {
            var e_1, _a;
            try {
                for (var _b = __asyncValues(this.listPagingPage(resourceGroupName, templateSpecName, options)), _c; _c = yield __await(_b.next()), !_c.done;) {
                    const page = _c.value;
                    yield __await(yield* __asyncDelegator(__asyncValues(page)));
                }
            }
            catch (e_1_1) { e_1 = { error: e_1_1 }; }
            finally {
                try {
                    if (_c && !_c.done && (_a = _b.return)) yield __await(_a.call(_b));
                }
                finally { if (e_1) throw e_1.error; }
            }
        });
    }
    /**
     * Creates or updates a Template Spec version.
     * @param resourceGroupName The name of the resource group. The name is case insensitive.
     * @param templateSpecName Name of the Template Spec.
     * @param templateSpecVersion The version of the Template Spec.
     * @param templateSpecVersionModel Template Spec Version supplied to the operation.
     * @param options The options parameters.
     */
    createOrUpdate(resourceGroupName, templateSpecName, templateSpecVersion, templateSpecVersionModel, options) {
        return this.client.sendOperationRequest({
            resourceGroupName,
            templateSpecName,
            templateSpecVersion,
            templateSpecVersionModel,
            options
        }, createOrUpdateOperationSpec);
    }
    /**
     * Updates Template Spec Version tags with specified values.
     * @param resourceGroupName The name of the resource group. The name is case insensitive.
     * @param templateSpecName Name of the Template Spec.
     * @param templateSpecVersion The version of the Template Spec.
     * @param options The options parameters.
     */
    update(resourceGroupName, templateSpecName, templateSpecVersion, options) {
        return this.client.sendOperationRequest({ resourceGroupName, templateSpecName, templateSpecVersion, options }, updateOperationSpec);
    }
    /**
     * Gets a Template Spec version from a specific Template Spec.
     * @param resourceGroupName The name of the resource group. The name is case insensitive.
     * @param templateSpecName Name of the Template Spec.
     * @param templateSpecVersion The version of the Template Spec.
     * @param options The options parameters.
     */
    get(resourceGroupName, templateSpecName, templateSpecVersion, options) {
        return this.client.sendOperationRequest({ resourceGroupName, templateSpecName, templateSpecVersion, options }, getOperationSpec);
    }
    /**
     * Deletes a specific version from a Template Spec. When operation completes, status code 200 returned
     * without content.
     * @param resourceGroupName The name of the resource group. The name is case insensitive.
     * @param templateSpecName Name of the Template Spec.
     * @param templateSpecVersion The version of the Template Spec.
     * @param options The options parameters.
     */
    delete(resourceGroupName, templateSpecName, templateSpecVersion, options) {
        return this.client.sendOperationRequest({ resourceGroupName, templateSpecName, templateSpecVersion, options }, deleteOperationSpec);
    }
    /**
     * Lists all the Template Spec versions in the specified Template Spec.
     * @param resourceGroupName The name of the resource group. The name is case insensitive.
     * @param templateSpecName Name of the Template Spec.
     * @param options The options parameters.
     */
    _list(resourceGroupName, templateSpecName, options) {
        return this.client.sendOperationRequest({ resourceGroupName, templateSpecName, options }, listOperationSpec);
    }
    /**
     * ListNext
     * @param resourceGroupName The name of the resource group. The name is case insensitive.
     * @param templateSpecName Name of the Template Spec.
     * @param nextLink The nextLink from the previous successful call to the List method.
     * @param options The options parameters.
     */
    _listNext(resourceGroupName, templateSpecName, nextLink, options) {
        return this.client.sendOperationRequest({ resourceGroupName, templateSpecName, nextLink, options }, listNextOperationSpec);
    }
}
// Operation Specifications
const serializer = coreClient.createSerializer(Mappers, /* isXml */ false);
const createOrUpdateOperationSpec = {
    path: "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Resources/templateSpecs/{templateSpecName}/versions/{templateSpecVersion}",
    httpMethod: "PUT",
    responses: {
        200: {
            bodyMapper: Mappers.TemplateSpecVersion
        },
        201: {
            bodyMapper: Mappers.TemplateSpecVersion
        },
        default: {
            bodyMapper: Mappers.TemplateSpecsError
        }
    },
    requestBody: Parameters.templateSpecVersionModel,
    queryParameters: [Parameters.apiVersion],
    urlParameters: [
        Parameters.$host,
        Parameters.subscriptionId,
        Parameters.resourceGroupName,
        Parameters.templateSpecName,
        Parameters.templateSpecVersion
    ],
    headerParameters: [Parameters.contentType, Parameters.accept],
    mediaType: "json",
    serializer
};
const updateOperationSpec = {
    path: "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Resources/templateSpecs/{templateSpecName}/versions/{templateSpecVersion}",
    httpMethod: "PATCH",
    responses: {
        200: {
            bodyMapper: Mappers.TemplateSpecVersion
        },
        default: {
            bodyMapper: Mappers.TemplateSpecsError
        }
    },
    requestBody: Parameters.templateSpecVersionUpdateModel,
    queryParameters: [Parameters.apiVersion],
    urlParameters: [
        Parameters.$host,
        Parameters.subscriptionId,
        Parameters.resourceGroupName,
        Parameters.templateSpecName,
        Parameters.templateSpecVersion
    ],
    headerParameters: [Parameters.contentType, Parameters.accept],
    mediaType: "json",
    serializer
};
const getOperationSpec = {
    path: "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Resources/templateSpecs/{templateSpecName}/versions/{templateSpecVersion}",
    httpMethod: "GET",
    responses: {
        200: {
            bodyMapper: Mappers.TemplateSpecVersion
        },
        default: {
            bodyMapper: Mappers.TemplateSpecsError
        }
    },
    queryParameters: [Parameters.apiVersion],
    urlParameters: [
        Parameters.$host,
        Parameters.subscriptionId,
        Parameters.resourceGroupName,
        Parameters.templateSpecName,
        Parameters.templateSpecVersion
    ],
    headerParameters: [Parameters.accept],
    serializer
};
const deleteOperationSpec = {
    path: "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Resources/templateSpecs/{templateSpecName}/versions/{templateSpecVersion}",
    httpMethod: "DELETE",
    responses: {
        200: {},
        204: {},
        default: {
            bodyMapper: Mappers.TemplateSpecsError
        }
    },
    queryParameters: [Parameters.apiVersion],
    urlParameters: [
        Parameters.$host,
        Parameters.subscriptionId,
        Parameters.resourceGroupName,
        Parameters.templateSpecName,
        Parameters.templateSpecVersion
    ],
    headerParameters: [Parameters.accept],
    serializer
};
const listOperationSpec = {
    path: "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Resources/templateSpecs/{templateSpecName}/versions",
    httpMethod: "GET",
    responses: {
        200: {
            bodyMapper: Mappers.TemplateSpecVersionsListResult
        },
        default: {
            bodyMapper: Mappers.TemplateSpecsError
        }
    },
    queryParameters: [Parameters.apiVersion],
    urlParameters: [
        Parameters.$host,
        Parameters.subscriptionId,
        Parameters.resourceGroupName,
        Parameters.templateSpecName
    ],
    headerParameters: [Parameters.accept],
    serializer
};
const listNextOperationSpec = {
    path: "{nextLink}",
    httpMethod: "GET",
    responses: {
        200: {
            bodyMapper: Mappers.TemplateSpecVersionsListResult
        },
        default: {
            bodyMapper: Mappers.TemplateSpecsError
        }
    },
    queryParameters: [Parameters.apiVersion],
    urlParameters: [
        Parameters.$host,
        Parameters.subscriptionId,
        Parameters.resourceGroupName,
        Parameters.templateSpecName,
        Parameters.nextLink
    ],
    headerParameters: [Parameters.accept],
    serializer
};
