// @aws-sdk/client-elasticache/dist-es/models/models_1.js
import { ElastiCacheServiceException as __BaseException } from "./ElastiCacheServiceException";
export class ReplicationGroupAlreadyUnderMigrationFault extends __BaseException {
    constructor(opts) {
        super({
            name: "ReplicationGroupAlreadyUnderMigrationFault",
            $fault: "client",
            ...opts,
        });
        this.name = "ReplicationGroupAlreadyUnderMigrationFault";
        this.$fault = "client";
        Object.setPrototypeOf(this, ReplicationGroupAlreadyUnderMigrationFault.prototype);
    }
}
export class APICallRateForCustomerExceededFault extends __BaseException {
    constructor(opts) {
        super({
            name: "APICallRateForCustomerExceededFault",
            $fault: "client",
            ...opts,
        });
        this.name = "APICallRateForCustomerExceededFault";
        this.$fault = "client";
        Object.setPrototypeOf(this, APICallRateForCustomerExceededFault.prototype);
    }
}
export class NodeGroupNotFoundFault extends __BaseException {
    constructor(opts) {
        super({
            name: "NodeGroupNotFoundFault",
            $fault: "client",
            ...opts,
        });
        this.name = "NodeGroupNotFoundFault";
        this.$fault = "client";
        Object.setPrototypeOf(this, NodeGroupNotFoundFault.prototype);
    }
}
export class TestFailoverNotAvailableFault extends __BaseException {
    constructor(opts) {
        super({
            name: "TestFailoverNotAvailableFault",
            $fault: "client",
            ...opts,
        });
        this.name = "TestFailoverNotAvailableFault";
        this.$fault = "client";
        Object.setPrototypeOf(this, TestFailoverNotAvailableFault.prototype);
    }
}
