// @azure/arm-cosmosdb/dist-esm/test/cosmosdb_examples.js
/*
 * Copyright (c) Microsoft Corporation.
 * Licensed under the MIT License.
 *
 * Code generated by Microsoft (R) AutoRest Code Generator.
 * Changes may cause incorrect behavior and will be lost if the code is regenerated.
 */
import { __asyncValues, __awaiter } from "tslib";
import { env, Recorder, isPlaybackMode, } from "@azure-tools/test-recorder";
import { createTestCredential } from "@azure-tools/test-credential";
import { assert } from "chai";
import { CosmosDBManagementClient } from "../src/cosmosDBManagementClient";
const replaceableVariables = {
    AZURE_CLIENT_ID: "azure_client_id",
    AZURE_CLIENT_SECRET: "azure_client_secret",
    AZURE_TENANT_ID: "88888888-8888-8888-8888-888888888888",
    SUBSCRIPTION_ID: "88888888-8888-8888-8888-888888888888"
};
const recorderOptions = {
    envSetupForPlayback: replaceableVariables
};
export const testPollingOptions = {
    updateIntervalInMs: isPlaybackMode() ? 0 : undefined,
};
describe("Cosmosdb test", () => {
    let recorder;
    let client;
    let subscriptionId;
    let location;
    let resourceGroupName;
    let accountName;
    beforeEach(function () {
        return __awaiter(this, void 0, void 0, function* () {
            recorder = new Recorder(this.currentTest);
            yield recorder.start(recorderOptions);
            subscriptionId = env.SUBSCRIPTION_ID || '';
            // This is an example of how the environment variables are used
            const credential = createTestCredential();
            client = new CosmosDBManagementClient(credential, subscriptionId, recorder.configureClientOptions({}));
            location = "eastus";
            resourceGroupName = "myjstest";
            accountName = "myaccountxxyy2";
        });
    });
    afterEach(function () {
        return __awaiter(this, void 0, void 0, function* () {
            yield recorder.stop();
        });
    });
    it("databaseAccounts create test", function () {
        return __awaiter(this, void 0, void 0, function* () {
            const res = yield client.databaseAccounts.beginCreateOrUpdateAndWait(resourceGroupName, accountName, {
                databaseAccountOfferType: "Standard",
                locations: [
                    {
                        failoverPriority: 2,
                        locationName: "southcentralus",
                        isZoneRedundant: false
                    },
                    {
                        locationName: "eastus",
                        failoverPriority: 1
                    },
                    {
                        locationName: "westus",
                        failoverPriority: 0
                    }
                ],
                location: location,
                createMode: "Default"
            }, testPollingOptions);
            assert.equal(res.name, accountName);
        });
    });
    it("databaseAccounts get test", function () {
        return __awaiter(this, void 0, void 0, function* () {
            const res = yield client.databaseAccounts.get(resourceGroupName, accountName);
            assert.equal(res.name, accountName);
        });
    });
    it("databaseAccounts list test", function () {
        var _a, e_1, _b, _c;
        return __awaiter(this, void 0, void 0, function* () {
            const resArray = new Array();
            try {
                for (var _d = true, _e = __asyncValues(client.databaseAccounts.listByResourceGroup(resourceGroupName)), _f; _f = yield _e.next(), _a = _f.done, !_a;) {
                    _c = _f.value;
                    _d = false;
                    try {
                        let item = _c;
                        resArray.push(item);
                    }
                    finally {
                        _d = true;
                    }
                }
            }
            catch (e_1_1) { e_1 = { error: e_1_1 }; }
            finally {
                try {
                    if (!_d && !_a && (_b = _e.return)) yield _b.call(_e);
                }
                finally { if (e_1) throw e_1.error; }
            }
            assert.equal(resArray.length, 1);
        });
    });
    it("databaseAccounts delete test", function () {
        var _a, e_2, _b, _c;
        return __awaiter(this, void 0, void 0, function* () {
            yield client.databaseAccounts.beginDeleteAndWait(resourceGroupName, accountName, testPollingOptions);
            const resArray = new Array();
            try {
                for (var _d = true, _e = __asyncValues(client.databaseAccounts.listByResourceGroup(resourceGroupName)), _f; _f = yield _e.next(), _a = _f.done, !_a;) {
                    _c = _f.value;
                    _d = false;
                    try {
                        let item = _c;
                        resArray.push(item);
                    }
                    finally {
                        _d = true;
                    }
                }
            }
            catch (e_2_1) { e_2 = { error: e_2_1 }; }
            finally {
                try {
                    if (!_d && !_a && (_b = _e.return)) yield _b.call(_e);
                }
                finally { if (e_2) throw e_2.error; }
            }
            assert.equal(resArray.length, 0);
        });
    }).timeout(3600000);
});
