// @azure/arm-mobilenetwork/dist-esm/src/operations/packetCoreDataPlanes.js
/*
 * Copyright (c) Microsoft Corporation.
 * Licensed under the MIT License.
 *
 * Code generated by Microsoft (R) AutoRest Code Generator.
 * Changes may cause incorrect behavior and will be lost if the code is regenerated.
 */
import { __asyncDelegator, __asyncGenerator, __asyncValues, __await, __awaiter } from "tslib";
import { setContinuationToken } from "../pagingHelper";
import * as coreClient from "@azure/core-client";
import * as Mappers from "../models/mappers";
import * as Parameters from "../models/parameters";
import { createHttpPoller } from "@azure/core-lro";
import { createLroSpec } from "../lroImpl";
/// <reference lib="esnext.asynciterable" />
/** Class containing PacketCoreDataPlanes operations. */
export class PacketCoreDataPlanesImpl {
    /**
     * Initialize a new instance of the class PacketCoreDataPlanes class.
     * @param client Reference to the service client
     */
    constructor(client) {
        this.client = client;
    }
    /**
     * Lists all the packet core data planes associated with a packet core control plane.
     * @param resourceGroupName The name of the resource group. The name is case insensitive.
     * @param packetCoreControlPlaneName The name of the packet core control plane.
     * @param options The options parameters.
     */
    listByPacketCoreControlPlane(resourceGroupName, packetCoreControlPlaneName, options) {
        const iter = this.listByPacketCoreControlPlanePagingAll(resourceGroupName, packetCoreControlPlaneName, options);
        return {
            next() {
                return iter.next();
            },
            [Symbol.asyncIterator]() {
                return this;
            },
            byPage: (settings) => {
                if (settings === null || settings === void 0 ? void 0 : settings.maxPageSize) {
                    throw new Error("maxPageSize is not supported by this operation.");
                }
                return this.listByPacketCoreControlPlanePagingPage(resourceGroupName, packetCoreControlPlaneName, options, settings);
            }
        };
    }
    listByPacketCoreControlPlanePagingPage(resourceGroupName, packetCoreControlPlaneName, options, settings) {
        return __asyncGenerator(this, arguments, function* listByPacketCoreControlPlanePagingPage_1() {
            let result;
            let continuationToken = settings === null || settings === void 0 ? void 0 : settings.continuationToken;
            if (!continuationToken) {
                result = yield __await(this._listByPacketCoreControlPlane(resourceGroupName, packetCoreControlPlaneName, options));
                let page = result.value || [];
                continuationToken = result.nextLink;
                setContinuationToken(page, continuationToken);
                yield yield __await(page);
            }
            while (continuationToken) {
                result = yield __await(this._listByPacketCoreControlPlaneNext(resourceGroupName, packetCoreControlPlaneName, continuationToken, options));
                continuationToken = result.nextLink;
                let page = result.value || [];
                setContinuationToken(page, continuationToken);
                yield yield __await(page);
            }
        });
    }
    listByPacketCoreControlPlanePagingAll(resourceGroupName, packetCoreControlPlaneName, options) {
        return __asyncGenerator(this, arguments, function* listByPacketCoreControlPlanePagingAll_1() {
            var _a, e_1, _b, _c;
            try {
                for (var _d = true, _e = __asyncValues(this.listByPacketCoreControlPlanePagingPage(resourceGroupName, packetCoreControlPlaneName, options)), _f; _f = yield __await(_e.next()), _a = _f.done, !_a;) {
                    _c = _f.value;
                    _d = false;
                    try {
                        const page = _c;
                        yield __await(yield* __asyncDelegator(__asyncValues(page)));
                    }
                    finally {
                        _d = true;
                    }
                }
            }
            catch (e_1_1) { e_1 = { error: e_1_1 }; }
            finally {
                try {
                    if (!_d && !_a && (_b = _e.return)) yield __await(_b.call(_e));
                }
                finally { if (e_1) throw e_1.error; }
            }
        });
    }
    /**
     * Deletes the specified packet core data plane.
     * @param resourceGroupName The name of the resource group. The name is case insensitive.
     * @param packetCoreControlPlaneName The name of the packet core control plane.
     * @param packetCoreDataPlaneName The name of the packet core data plane.
     * @param options The options parameters.
     */
    beginDelete(resourceGroupName, packetCoreControlPlaneName, packetCoreDataPlaneName, options) {
        return __awaiter(this, void 0, void 0, function* () {
            const directSendOperation = (args, spec) => __awaiter(this, void 0, void 0, function* () {
                return this.client.sendOperationRequest(args, spec);
            });
            const sendOperationFn = (args, spec) => __awaiter(this, void 0, void 0, function* () {
                var _a;
                let currentRawResponse = undefined;
                const providedCallback = (_a = args.options) === null || _a === void 0 ? void 0 : _a.onResponse;
                const callback = (rawResponse, flatResponse) => {
                    currentRawResponse = rawResponse;
                    providedCallback === null || providedCallback === void 0 ? void 0 : providedCallback(rawResponse, flatResponse);
                };
                const updatedArgs = Object.assign(Object.assign({}, args), { options: Object.assign(Object.assign({}, args.options), { onResponse: callback }) });
                const flatResponse = yield directSendOperation(updatedArgs, spec);
                return {
                    flatResponse,
                    rawResponse: {
                        statusCode: currentRawResponse.status,
                        body: currentRawResponse.parsedBody,
                        headers: currentRawResponse.headers.toJSON()
                    }
                };
            });
            const lro = createLroSpec({
                sendOperationFn,
                args: {
                    resourceGroupName,
                    packetCoreControlPlaneName,
                    packetCoreDataPlaneName,
                    options
                },
                spec: deleteOperationSpec
            });
            const poller = yield createHttpPoller(lro, {
                restoreFrom: options === null || options === void 0 ? void 0 : options.resumeFrom,
                intervalInMs: options === null || options === void 0 ? void 0 : options.updateIntervalInMs,
                resourceLocationConfig: "location"
            });
            yield poller.poll();
            return poller;
        });
    }
    /**
     * Deletes the specified packet core data plane.
     * @param resourceGroupName The name of the resource group. The name is case insensitive.
     * @param packetCoreControlPlaneName The name of the packet core control plane.
     * @param packetCoreDataPlaneName The name of the packet core data plane.
     * @param options The options parameters.
     */
    beginDeleteAndWait(resourceGroupName, packetCoreControlPlaneName, packetCoreDataPlaneName, options) {
        return __awaiter(this, void 0, void 0, function* () {
            const poller = yield this.beginDelete(resourceGroupName, packetCoreControlPlaneName, packetCoreDataPlaneName, options);
            return poller.pollUntilDone();
        });
    }
    /**
     * Gets information about the specified packet core data plane.
     * @param resourceGroupName The name of the resource group. The name is case insensitive.
     * @param packetCoreControlPlaneName The name of the packet core control plane.
     * @param packetCoreDataPlaneName The name of the packet core data plane.
     * @param options The options parameters.
     */
    get(resourceGroupName, packetCoreControlPlaneName, packetCoreDataPlaneName, options) {
        return this.client.sendOperationRequest({
            resourceGroupName,
            packetCoreControlPlaneName,
            packetCoreDataPlaneName,
            options
        }, getOperationSpec);
    }
    /**
     * Creates or updates a packet core data plane. Must be created in the same location as its parent
     * packet core control plane.
     * @param resourceGroupName The name of the resource group. The name is case insensitive.
     * @param packetCoreControlPlaneName The name of the packet core control plane.
     * @param packetCoreDataPlaneName The name of the packet core data plane.
     * @param parameters Parameters supplied to the create or update packet core data plane operation.
     * @param options The options parameters.
     */
    beginCreateOrUpdate(resourceGroupName, packetCoreControlPlaneName, packetCoreDataPlaneName, parameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            const directSendOperation = (args, spec) => __awaiter(this, void 0, void 0, function* () {
                return this.client.sendOperationRequest(args, spec);
            });
            const sendOperationFn = (args, spec) => __awaiter(this, void 0, void 0, function* () {
                var _a;
                let currentRawResponse = undefined;
                const providedCallback = (_a = args.options) === null || _a === void 0 ? void 0 : _a.onResponse;
                const callback = (rawResponse, flatResponse) => {
                    currentRawResponse = rawResponse;
                    providedCallback === null || providedCallback === void 0 ? void 0 : providedCallback(rawResponse, flatResponse);
                };
                const updatedArgs = Object.assign(Object.assign({}, args), { options: Object.assign(Object.assign({}, args.options), { onResponse: callback }) });
                const flatResponse = yield directSendOperation(updatedArgs, spec);
                return {
                    flatResponse,
                    rawResponse: {
                        statusCode: currentRawResponse.status,
                        body: currentRawResponse.parsedBody,
                        headers: currentRawResponse.headers.toJSON()
                    }
                };
            });
            const lro = createLroSpec({
                sendOperationFn,
                args: {
                    resourceGroupName,
                    packetCoreControlPlaneName,
                    packetCoreDataPlaneName,
                    parameters,
                    options
                },
                spec: createOrUpdateOperationSpec
            });
            const poller = yield createHttpPoller(lro, {
                restoreFrom: options === null || options === void 0 ? void 0 : options.resumeFrom,
                intervalInMs: options === null || options === void 0 ? void 0 : options.updateIntervalInMs,
                resourceLocationConfig: "azure-async-operation"
            });
            yield poller.poll();
            return poller;
        });
    }
    /**
     * Creates or updates a packet core data plane. Must be created in the same location as its parent
     * packet core control plane.
     * @param resourceGroupName The name of the resource group. The name is case insensitive.
     * @param packetCoreControlPlaneName The name of the packet core control plane.
     * @param packetCoreDataPlaneName The name of the packet core data plane.
     * @param parameters Parameters supplied to the create or update packet core data plane operation.
     * @param options The options parameters.
     */
    beginCreateOrUpdateAndWait(resourceGroupName, packetCoreControlPlaneName, packetCoreDataPlaneName, parameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            const poller = yield this.beginCreateOrUpdate(resourceGroupName, packetCoreControlPlaneName, packetCoreDataPlaneName, parameters, options);
            return poller.pollUntilDone();
        });
    }
    /**
     * Updates packet core data planes tags.
     * @param resourceGroupName The name of the resource group. The name is case insensitive.
     * @param packetCoreControlPlaneName The name of the packet core control plane.
     * @param packetCoreDataPlaneName The name of the packet core data plane.
     * @param parameters Parameters supplied to update packet core data plane tags.
     * @param options The options parameters.
     */
    updateTags(resourceGroupName, packetCoreControlPlaneName, packetCoreDataPlaneName, parameters, options) {
        return this.client.sendOperationRequest({
            resourceGroupName,
            packetCoreControlPlaneName,
            packetCoreDataPlaneName,
            parameters,
            options
        }, updateTagsOperationSpec);
    }
    /**
     * Lists all the packet core data planes associated with a packet core control plane.
     * @param resourceGroupName The name of the resource group. The name is case insensitive.
     * @param packetCoreControlPlaneName The name of the packet core control plane.
     * @param options The options parameters.
     */
    _listByPacketCoreControlPlane(resourceGroupName, packetCoreControlPlaneName, options) {
        return this.client.sendOperationRequest({ resourceGroupName, packetCoreControlPlaneName, options }, listByPacketCoreControlPlaneOperationSpec);
    }
    /**
     * ListByPacketCoreControlPlaneNext
     * @param resourceGroupName The name of the resource group. The name is case insensitive.
     * @param packetCoreControlPlaneName The name of the packet core control plane.
     * @param nextLink The nextLink from the previous successful call to the ListByPacketCoreControlPlane
     *                 method.
     * @param options The options parameters.
     */
    _listByPacketCoreControlPlaneNext(resourceGroupName, packetCoreControlPlaneName, nextLink, options) {
        return this.client.sendOperationRequest({ resourceGroupName, packetCoreControlPlaneName, nextLink, options }, listByPacketCoreControlPlaneNextOperationSpec);
    }
}
// Operation Specifications
const serializer = coreClient.createSerializer(Mappers, /* isXml */ false);
const deleteOperationSpec = {
    path: "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.MobileNetwork/packetCoreControlPlanes/{packetCoreControlPlaneName}/packetCoreDataPlanes/{packetCoreDataPlaneName}",
    httpMethod: "DELETE",
    responses: {
        200: {},
        201: {},
        202: {},
        204: {},
        default: {
            bodyMapper: Mappers.ErrorResponse
        }
    },
    queryParameters: [Parameters.apiVersion],
    urlParameters: [
        Parameters.$host,
        Parameters.subscriptionId,
        Parameters.resourceGroupName,
        Parameters.packetCoreControlPlaneName,
        Parameters.packetCoreDataPlaneName
    ],
    headerParameters: [Parameters.accept],
    serializer
};
const getOperationSpec = {
    path: "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.MobileNetwork/packetCoreControlPlanes/{packetCoreControlPlaneName}/packetCoreDataPlanes/{packetCoreDataPlaneName}",
    httpMethod: "GET",
    responses: {
        200: {
            bodyMapper: Mappers.PacketCoreDataPlane
        },
        default: {
            bodyMapper: Mappers.ErrorResponse
        }
    },
    queryParameters: [Parameters.apiVersion],
    urlParameters: [
        Parameters.$host,
        Parameters.subscriptionId,
        Parameters.resourceGroupName,
        Parameters.packetCoreControlPlaneName,
        Parameters.packetCoreDataPlaneName
    ],
    headerParameters: [Parameters.accept],
    serializer
};
const createOrUpdateOperationSpec = {
    path: "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.MobileNetwork/packetCoreControlPlanes/{packetCoreControlPlaneName}/packetCoreDataPlanes/{packetCoreDataPlaneName}",
    httpMethod: "PUT",
    responses: {
        200: {
            bodyMapper: Mappers.PacketCoreDataPlane
        },
        201: {
            bodyMapper: Mappers.PacketCoreDataPlane
        },
        202: {
            bodyMapper: Mappers.PacketCoreDataPlane
        },
        204: {
            bodyMapper: Mappers.PacketCoreDataPlane
        },
        default: {
            bodyMapper: Mappers.ErrorResponse
        }
    },
    requestBody: Parameters.parameters8,
    queryParameters: [Parameters.apiVersion],
    urlParameters: [
        Parameters.$host,
        Parameters.subscriptionId,
        Parameters.resourceGroupName,
        Parameters.packetCoreControlPlaneName,
        Parameters.packetCoreDataPlaneName
    ],
    headerParameters: [Parameters.accept, Parameters.contentType],
    mediaType: "json",
    serializer
};
const updateTagsOperationSpec = {
    path: "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.MobileNetwork/packetCoreControlPlanes/{packetCoreControlPlaneName}/packetCoreDataPlanes/{packetCoreDataPlaneName}",
    httpMethod: "PATCH",
    responses: {
        200: {
            bodyMapper: Mappers.PacketCoreDataPlane
        },
        default: {
            bodyMapper: Mappers.ErrorResponse
        }
    },
    requestBody: Parameters.parameters1,
    queryParameters: [Parameters.apiVersion],
    urlParameters: [
        Parameters.$host,
        Parameters.subscriptionId,
        Parameters.resourceGroupName,
        Parameters.packetCoreControlPlaneName,
        Parameters.packetCoreDataPlaneName
    ],
    headerParameters: [Parameters.accept, Parameters.contentType],
    mediaType: "json",
    serializer
};
const listByPacketCoreControlPlaneOperationSpec = {
    path: "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.MobileNetwork/packetCoreControlPlanes/{packetCoreControlPlaneName}/packetCoreDataPlanes",
    httpMethod: "GET",
    responses: {
        200: {
            bodyMapper: Mappers.PacketCoreDataPlaneListResult
        },
        default: {
            bodyMapper: Mappers.ErrorResponse
        }
    },
    queryParameters: [Parameters.apiVersion],
    urlParameters: [
        Parameters.$host,
        Parameters.subscriptionId,
        Parameters.resourceGroupName,
        Parameters.packetCoreControlPlaneName
    ],
    headerParameters: [Parameters.accept],
    serializer
};
const listByPacketCoreControlPlaneNextOperationSpec = {
    path: "{nextLink}",
    httpMethod: "GET",
    responses: {
        200: {
            bodyMapper: Mappers.PacketCoreDataPlaneListResult
        },
        default: {
            bodyMapper: Mappers.ErrorResponse
        }
    },
    urlParameters: [
        Parameters.$host,
        Parameters.subscriptionId,
        Parameters.resourceGroupName,
        Parameters.packetCoreControlPlaneName,
        Parameters.nextLink
    ],
    headerParameters: [Parameters.accept],
    serializer
};
