// @aws-sdk/client-redshiftserverless/dist-es/commands/ConvertRecoveryPointToSnapshotCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { ConvertRecoveryPointToSnapshotRequest, ConvertRecoveryPointToSnapshotResponse } from "../models/models_0";
import { deserializeAws_json1_1ConvertRecoveryPointToSnapshotCommand, serializeAws_json1_1ConvertRecoveryPointToSnapshotCommand, } from "../protocols/Aws_json1_1";
var ConvertRecoveryPointToSnapshotCommand = (function (_super) {
    __extends(ConvertRecoveryPointToSnapshotCommand, _super);
    function ConvertRecoveryPointToSnapshotCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    ConvertRecoveryPointToSnapshotCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "RedshiftServerlessClient";
        var commandName = "ConvertRecoveryPointToSnapshotCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: ConvertRecoveryPointToSnapshotRequest.filterSensitiveLog,
            outputFilterSensitiveLog: ConvertRecoveryPointToSnapshotResponse.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    ConvertRecoveryPointToSnapshotCommand.prototype.serialize = function (input, context) {
        return serializeAws_json1_1ConvertRecoveryPointToSnapshotCommand(input, context);
    };
    ConvertRecoveryPointToSnapshotCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_json1_1ConvertRecoveryPointToSnapshotCommand(output, context);
    };
    return ConvertRecoveryPointToSnapshotCommand;
}($Command));
export { ConvertRecoveryPointToSnapshotCommand };
