// @aws-sdk/client-redshiftserverless/dist-es/commands/CreateNamespaceCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { CreateNamespaceRequest, CreateNamespaceResponse } from "../models/models_0";
import { deserializeAws_json1_1CreateNamespaceCommand, serializeAws_json1_1CreateNamespaceCommand, } from "../protocols/Aws_json1_1";
var CreateNamespaceCommand = (function (_super) {
    __extends(CreateNamespaceCommand, _super);
    function CreateNamespaceCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    CreateNamespaceCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "RedshiftServerlessClient";
        var commandName = "CreateNamespaceCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: CreateNamespaceRequest.filterSensitiveLog,
            outputFilterSensitiveLog: CreateNamespaceResponse.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    CreateNamespaceCommand.prototype.serialize = function (input, context) {
        return serializeAws_json1_1CreateNamespaceCommand(input, context);
    };
    CreateNamespaceCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_json1_1CreateNamespaceCommand(output, context);
    };
    return CreateNamespaceCommand;
}($Command));
export { CreateNamespaceCommand };
