// @aws-sdk/client-redshiftserverless/dist-es/commands/CreateUsageLimitCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { CreateUsageLimitRequest, CreateUsageLimitResponse } from "../models/models_0";
import { deserializeAws_json1_1CreateUsageLimitCommand, serializeAws_json1_1CreateUsageLimitCommand, } from "../protocols/Aws_json1_1";
var CreateUsageLimitCommand = (function (_super) {
    __extends(CreateUsageLimitCommand, _super);
    function CreateUsageLimitCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    CreateUsageLimitCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "RedshiftServerlessClient";
        var commandName = "CreateUsageLimitCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: CreateUsageLimitRequest.filterSensitiveLog,
            outputFilterSensitiveLog: CreateUsageLimitResponse.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    CreateUsageLimitCommand.prototype.serialize = function (input, context) {
        return serializeAws_json1_1CreateUsageLimitCommand(input, context);
    };
    CreateUsageLimitCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_json1_1CreateUsageLimitCommand(output, context);
    };
    return CreateUsageLimitCommand;
}($Command));
export { CreateUsageLimitCommand };
