// @aws-sdk/client-redshiftserverless/dist-es/commands/DeleteNamespaceCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { DeleteNamespaceRequest, DeleteNamespaceResponse } from "../models/models_0";
import { deserializeAws_json1_1DeleteNamespaceCommand, serializeAws_json1_1DeleteNamespaceCommand, } from "../protocols/Aws_json1_1";
var DeleteNamespaceCommand = (function (_super) {
    __extends(DeleteNamespaceCommand, _super);
    function DeleteNamespaceCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    DeleteNamespaceCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "RedshiftServerlessClient";
        var commandName = "DeleteNamespaceCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: DeleteNamespaceRequest.filterSensitiveLog,
            outputFilterSensitiveLog: DeleteNamespaceResponse.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    DeleteNamespaceCommand.prototype.serialize = function (input, context) {
        return serializeAws_json1_1DeleteNamespaceCommand(input, context);
    };
    DeleteNamespaceCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_json1_1DeleteNamespaceCommand(output, context);
    };
    return DeleteNamespaceCommand;
}($Command));
export { DeleteNamespaceCommand };
