// @aws-sdk/client-redshiftserverless/dist-es/commands/GetSnapshotCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { GetSnapshotRequest, GetSnapshotResponse } from "../models/models_0";
import { deserializeAws_json1_1GetSnapshotCommand, serializeAws_json1_1GetSnapshotCommand, } from "../protocols/Aws_json1_1";
var GetSnapshotCommand = (function (_super) {
    __extends(GetSnapshotCommand, _super);
    function GetSnapshotCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    GetSnapshotCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "RedshiftServerlessClient";
        var commandName = "GetSnapshotCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: GetSnapshotRequest.filterSensitiveLog,
            outputFilterSensitiveLog: GetSnapshotResponse.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    GetSnapshotCommand.prototype.serialize = function (input, context) {
        return serializeAws_json1_1GetSnapshotCommand(input, context);
    };
    GetSnapshotCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_json1_1GetSnapshotCommand(output, context);
    };
    return GetSnapshotCommand;
}($Command));
export { GetSnapshotCommand };
