// @aws-sdk/client-redshiftserverless/dist-es/commands/GetWorkgroupCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { GetWorkgroupRequest, GetWorkgroupResponse } from "../models/models_0";
import { deserializeAws_json1_1GetWorkgroupCommand, serializeAws_json1_1GetWorkgroupCommand, } from "../protocols/Aws_json1_1";
var GetWorkgroupCommand = (function (_super) {
    __extends(GetWorkgroupCommand, _super);
    function GetWorkgroupCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    GetWorkgroupCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "RedshiftServerlessClient";
        var commandName = "GetWorkgroupCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: GetWorkgroupRequest.filterSensitiveLog,
            outputFilterSensitiveLog: GetWorkgroupResponse.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    GetWorkgroupCommand.prototype.serialize = function (input, context) {
        return serializeAws_json1_1GetWorkgroupCommand(input, context);
    };
    GetWorkgroupCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_json1_1GetWorkgroupCommand(output, context);
    };
    return GetWorkgroupCommand;
}($Command));
export { GetWorkgroupCommand };
