// @aws-sdk/client-redshiftserverless/dist-es/commands/ListUsageLimitsCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { ListUsageLimitsRequest, ListUsageLimitsResponse } from "../models/models_0";
import { deserializeAws_json1_1ListUsageLimitsCommand, serializeAws_json1_1ListUsageLimitsCommand, } from "../protocols/Aws_json1_1";
var ListUsageLimitsCommand = (function (_super) {
    __extends(ListUsageLimitsCommand, _super);
    function ListUsageLimitsCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    ListUsageLimitsCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "RedshiftServerlessClient";
        var commandName = "ListUsageLimitsCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: ListUsageLimitsRequest.filterSensitiveLog,
            outputFilterSensitiveLog: ListUsageLimitsResponse.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    ListUsageLimitsCommand.prototype.serialize = function (input, context) {
        return serializeAws_json1_1ListUsageLimitsCommand(input, context);
    };
    ListUsageLimitsCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_json1_1ListUsageLimitsCommand(output, context);
    };
    return ListUsageLimitsCommand;
}($Command));
export { ListUsageLimitsCommand };
