// @aws-sdk/client-redshiftserverless/dist-es/commands/ListNamespacesCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { ListNamespacesRequest, ListNamespacesResponse } from "../models/models_0";
import { deserializeAws_json1_1ListNamespacesCommand, serializeAws_json1_1ListNamespacesCommand, } from "../protocols/Aws_json1_1";
var ListNamespacesCommand = (function (_super) {
    __extends(ListNamespacesCommand, _super);
    function ListNamespacesCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    ListNamespacesCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "RedshiftServerlessClient";
        var commandName = "ListNamespacesCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: ListNamespacesRequest.filterSensitiveLog,
            outputFilterSensitiveLog: ListNamespacesResponse.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    ListNamespacesCommand.prototype.serialize = function (input, context) {
        return serializeAws_json1_1ListNamespacesCommand(input, context);
    };
    ListNamespacesCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_json1_1ListNamespacesCommand(output, context);
    };
    return ListNamespacesCommand;
}($Command));
export { ListNamespacesCommand };
