// @aws-sdk/client-redshiftserverless/dist-es/commands/ListWorkgroupsCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { ListWorkgroupsRequest, ListWorkgroupsResponse } from "../models/models_0";
import { deserializeAws_json1_1ListWorkgroupsCommand, serializeAws_json1_1ListWorkgroupsCommand, } from "../protocols/Aws_json1_1";
var ListWorkgroupsCommand = (function (_super) {
    __extends(ListWorkgroupsCommand, _super);
    function ListWorkgroupsCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    ListWorkgroupsCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "RedshiftServerlessClient";
        var commandName = "ListWorkgroupsCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: ListWorkgroupsRequest.filterSensitiveLog,
            outputFilterSensitiveLog: ListWorkgroupsResponse.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    ListWorkgroupsCommand.prototype.serialize = function (input, context) {
        return serializeAws_json1_1ListWorkgroupsCommand(input, context);
    };
    ListWorkgroupsCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_json1_1ListWorkgroupsCommand(output, context);
    };
    return ListWorkgroupsCommand;
}($Command));
export { ListWorkgroupsCommand };
