// @aws-sdk/client-redshiftserverless/dist-es/commands/RestoreFromSnapshotCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { RestoreFromSnapshotRequest, RestoreFromSnapshotResponse } from "../models/models_0";
import { deserializeAws_json1_1RestoreFromSnapshotCommand, serializeAws_json1_1RestoreFromSnapshotCommand, } from "../protocols/Aws_json1_1";
var RestoreFromSnapshotCommand = (function (_super) {
    __extends(RestoreFromSnapshotCommand, _super);
    function RestoreFromSnapshotCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    RestoreFromSnapshotCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "RedshiftServerlessClient";
        var commandName = "RestoreFromSnapshotCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: RestoreFromSnapshotRequest.filterSensitiveLog,
            outputFilterSensitiveLog: RestoreFromSnapshotResponse.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    RestoreFromSnapshotCommand.prototype.serialize = function (input, context) {
        return serializeAws_json1_1RestoreFromSnapshotCommand(input, context);
    };
    RestoreFromSnapshotCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_json1_1RestoreFromSnapshotCommand(output, context);
    };
    return RestoreFromSnapshotCommand;
}($Command));
export { RestoreFromSnapshotCommand };
