// @aws-sdk/client-redshiftserverless/dist-es/commands/UpdateSnapshotCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { UpdateSnapshotRequest, UpdateSnapshotResponse } from "../models/models_0";
import { deserializeAws_json1_1UpdateSnapshotCommand, serializeAws_json1_1UpdateSnapshotCommand, } from "../protocols/Aws_json1_1";
var UpdateSnapshotCommand = (function (_super) {
    __extends(UpdateSnapshotCommand, _super);
    function UpdateSnapshotCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    UpdateSnapshotCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "RedshiftServerlessClient";
        var commandName = "UpdateSnapshotCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: UpdateSnapshotRequest.filterSensitiveLog,
            outputFilterSensitiveLog: UpdateSnapshotResponse.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    UpdateSnapshotCommand.prototype.serialize = function (input, context) {
        return serializeAws_json1_1UpdateSnapshotCommand(input, context);
    };
    UpdateSnapshotCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_json1_1UpdateSnapshotCommand(output, context);
    };
    return UpdateSnapshotCommand;
}($Command));
export { UpdateSnapshotCommand };
