// @aws-sdk/client-redshiftserverless/dist-es/commands/UpdateNamespaceCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { UpdateNamespaceRequest, UpdateNamespaceResponse } from "../models/models_0";
import { deserializeAws_json1_1UpdateNamespaceCommand, serializeAws_json1_1UpdateNamespaceCommand, } from "../protocols/Aws_json1_1";
var UpdateNamespaceCommand = (function (_super) {
    __extends(UpdateNamespaceCommand, _super);
    function UpdateNamespaceCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    UpdateNamespaceCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "RedshiftServerlessClient";
        var commandName = "UpdateNamespaceCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: UpdateNamespaceRequest.filterSensitiveLog,
            outputFilterSensitiveLog: UpdateNamespaceResponse.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    UpdateNamespaceCommand.prototype.serialize = function (input, context) {
        return serializeAws_json1_1UpdateNamespaceCommand(input, context);
    };
    UpdateNamespaceCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_json1_1UpdateNamespaceCommand(output, context);
    };
    return UpdateNamespaceCommand;
}($Command));
export { UpdateNamespaceCommand };
