// @aws-sdk/client-redshiftserverless/dist-es/commands/UpdateUsageLimitCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { UpdateUsageLimitRequest, UpdateUsageLimitResponse } from "../models/models_0";
import { deserializeAws_json1_1UpdateUsageLimitCommand, serializeAws_json1_1UpdateUsageLimitCommand, } from "../protocols/Aws_json1_1";
var UpdateUsageLimitCommand = (function (_super) {
    __extends(UpdateUsageLimitCommand, _super);
    function UpdateUsageLimitCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    UpdateUsageLimitCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "RedshiftServerlessClient";
        var commandName = "UpdateUsageLimitCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: UpdateUsageLimitRequest.filterSensitiveLog,
            outputFilterSensitiveLog: UpdateUsageLimitResponse.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    UpdateUsageLimitCommand.prototype.serialize = function (input, context) {
        return serializeAws_json1_1UpdateUsageLimitCommand(input, context);
    };
    UpdateUsageLimitCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_json1_1UpdateUsageLimitCommand(output, context);
    };
    return UpdateUsageLimitCommand;
}($Command));
export { UpdateUsageLimitCommand };
