// @azure/arm-keyvault/dist-esm/test/keyvault_examples.js
/*
 * Copyright (c) Microsoft Corporation.
 * Licensed under the MIT License.
 *
 * Code generated by Microsoft (R) AutoRest Code Generator.
 * Changes may cause incorrect behavior and will be lost if the code is regenerated.
 */
import { __asyncValues, __awaiter } from "tslib";
import { env, Recorder, isPlaybackMode, } from "@azure-tools/test-recorder";
import { createTestCredential } from "@azure-tools/test-credential";
import { assert } from "chai";
import { KeyVaultManagementClient } from "../src/keyVaultManagementClient";
const replaceableVariables = {
    AZURE_CLIENT_ID: "azure_client_id",
    AZURE_CLIENT_SECRET: "azure_client_secret",
    AZURE_TENANT_ID: "88888888-8888-8888-8888-888888888888",
    SUBSCRIPTION_ID: "azure_subscription_id"
};
const recorderOptions = {
    envSetupForPlayback: replaceableVariables
};
export const testPollingOptions = {
    updateIntervalInMs: isPlaybackMode() ? 0 : undefined,
};
describe("Keyvault test", () => {
    let recorder;
    let subscriptionId;
    let client;
    let location;
    let resourceGroup;
    let vaultName;
    let tenantId;
    beforeEach(function () {
        return __awaiter(this, void 0, void 0, function* () {
            recorder = new Recorder(this.currentTest);
            yield recorder.start(recorderOptions);
            subscriptionId = env.SUBSCRIPTION_ID || '';
            tenantId = env.AZURE_TENANT_ID || '';
            // This is an example of how the environment variables are used
            const credential = createTestCredential();
            client = new KeyVaultManagementClient(credential, subscriptionId, recorder.configureClientOptions({}));
            location = "eastus";
            resourceGroup = "myjstest";
            vaultName = "myvaultzzzz" + "231019";
        });
    });
    afterEach(function () {
        return __awaiter(this, void 0, void 0, function* () {
            yield recorder.stop();
        });
    });
    it("vaults create test", function () {
        return __awaiter(this, void 0, void 0, function* () {
            const res = yield client.vaults.beginCreateOrUpdateAndWait(resourceGroup, vaultName, {
                location: "eastus",
                properties: {
                    tenantId: tenantId,
                    sku: {
                        family: "A",
                        name: "standard",
                    },
                    accessPolicies: [
                        {
                            tenantId: tenantId,
                            objectId: "00000000-0000-0000-0000-000000000000",
                            permissions: {
                                keys: [
                                    "encrypt",
                                    "decrypt",
                                    "wrapKey",
                                    "unwrapKey",
                                    "sign",
                                    "verify",
                                    "get",
                                    "list",
                                    "create",
                                    "update",
                                    "import",
                                    "delete",
                                    "backup",
                                    "restore",
                                    "recover",
                                    "purge",
                                ],
                                secrets: [
                                    "get",
                                    "list",
                                    "set",
                                    "delete",
                                    "backup",
                                    "restore",
                                    "recover",
                                    "purge",
                                ],
                                certificates: [
                                    "get",
                                    "list",
                                    "delete",
                                    "create",
                                    "import",
                                    "update",
                                    "managecontacts",
                                    "getissuers",
                                    "listissuers",
                                    "setissuers",
                                    "deleteissuers",
                                    "manageissuers",
                                    "recover",
                                    "purge",
                                ],
                            },
                        },
                    ],
                    enabledForDeployment: true,
                    enabledForDiskEncryption: true,
                    enabledForTemplateDeployment: true,
                }
            }, testPollingOptions);
            assert.equal(res.name, vaultName);
        });
    });
    it("vaults get test", function () {
        return __awaiter(this, void 0, void 0, function* () {
            if (isPlaybackMode()) {
                this.skip();
            }
            const res = yield client.vaults.get(resourceGroup, vaultName);
            assert.equal(res.name, vaultName);
        });
    });
    it("vaults list test", function () {
        var _a, e_1, _b, _c;
        return __awaiter(this, void 0, void 0, function* () {
            if (isPlaybackMode()) {
                this.skip();
            }
            const resArray = new Array();
            try {
                for (var _d = true, _e = __asyncValues(client.vaults.listByResourceGroup(resourceGroup)), _f; _f = yield _e.next(), _a = _f.done, !_a;) {
                    _c = _f.value;
                    _d = false;
                    try {
                        let item = _c;
                        resArray.push(item);
                    }
                    finally {
                        _d = true;
                    }
                }
            }
            catch (e_1_1) { e_1 = { error: e_1_1 }; }
            finally {
                try {
                    if (!_d && !_a && (_b = _e.return)) yield _b.call(_e);
                }
                finally { if (e_1) throw e_1.error; }
            }
            assert.equal(resArray.length, 1);
        });
    });
    it("vaults update test", function () {
        return __awaiter(this, void 0, void 0, function* () {
            const res = yield client.vaults.update(resourceGroup, vaultName, {
                properties: {
                    tenantId: tenantId,
                    sku: {
                        family: "A",
                        name: "standard",
                    },
                    accessPolicies: [
                        {
                            tenantId: tenantId,
                            objectId: "00000000-0000-0000-0000-000000000000",
                            permissions: {
                                keys: [
                                    "encrypt",
                                    "decrypt",
                                    "wrapKey",
                                    "unwrapKey",
                                    "sign",
                                    "verify",
                                    "get",
                                    "list",
                                    "create",
                                    "update",
                                    "import",
                                    "delete",
                                    "backup",
                                    "restore",
                                    "recover",
                                    "purge",
                                ],
                                secrets: [
                                    "get",
                                    "list",
                                    "set",
                                    "delete",
                                    "backup",
                                    "restore",
                                    "recover",
                                    "purge",
                                ],
                                certificates: [
                                    "get",
                                    "list",
                                    "delete",
                                    "create",
                                    "import",
                                    "update",
                                    "managecontacts",
                                    "getissuers",
                                    "listissuers",
                                    "setissuers",
                                    "deleteissuers",
                                    "manageissuers",
                                    "recover",
                                    "purge",
                                ],
                            },
                        },
                    ],
                    enabledForDeployment: true,
                    enabledForDiskEncryption: true,
                    enabledForTemplateDeployment: true,
                }
            });
            assert.equal(res.type, "Microsoft.KeyVault/vaults");
        });
    });
    it("vaults delete test", function () {
        var _a, e_2, _b, _c;
        return __awaiter(this, void 0, void 0, function* () {
            const res = yield client.vaults.delete(resourceGroup, vaultName);
            const resArray = new Array();
            try {
                for (var _d = true, _e = __asyncValues(client.vaults.listByResourceGroup(resourceGroup)), _f; _f = yield _e.next(), _a = _f.done, !_a;) {
                    _c = _f.value;
                    _d = false;
                    try {
                        let item = _c;
                        resArray.push(item);
                    }
                    finally {
                        _d = true;
                    }
                }
            }
            catch (e_2_1) { e_2 = { error: e_2_1 }; }
            finally {
                try {
                    if (!_d && !_a && (_b = _e.return)) yield _b.call(_e);
                }
                finally { if (e_2) throw e_2.error; }
            }
            assert.equal(resArray.length, 0);
        });
    });
});
