// @azure/arm-storageimportexport/dist-esm/samples-dev/jobsUpdateSample.js
/*
 * Copyright (c) Microsoft Corporation.
 * Licensed under the MIT License.
 *
 * Code generated by Microsoft (R) AutoRest Code Generator.
 * Changes may cause incorrect behavior and will be lost if the code is regenerated.
 */
import { __awaiter } from "tslib";
// Copyright (c) Microsoft Corporation.
// Licensed under the MIT License.
import { StorageImportExport } from "@azure/arm-storageimportexport";
import { DefaultAzureCredential } from "@azure/identity";
import * as dotenv from "dotenv";
dotenv.config();
/**
 * This sample demonstrates how to Updates specific properties of a job. You can call this operation to notify the Import/Export service that the hard drives comprising the import or export job have been shipped to the Microsoft data center. It can also be used to cancel an existing job.
 *
 * @summary Updates specific properties of a job. You can call this operation to notify the Import/Export service that the hard drives comprising the import or export job have been shipped to the Microsoft data center. It can also be used to cancel an existing job.
 * x-ms-original-file: specification/storageimportexport/resource-manager/Microsoft.ImportExport/preview/2021-01-01/examples/UpdateExportJob.json
 */
function updateExportJob() {
    return __awaiter(this, void 0, void 0, function* () {
        const subscriptionId = process.env["STORAGEIMPORTEXPORT_SUBSCRIPTION_ID"] ||
            "xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx";
        const jobName = "myExportJob";
        const resourceGroupName = process.env["STORAGEIMPORTEXPORT_RESOURCE_GROUP"] || "myResourceGroup";
        const body = {
            backupDriveManifest: true,
            logLevel: "Verbose",
            state: ""
        };
        const credential = new DefaultAzureCredential();
        const client = new StorageImportExport(credential, subscriptionId);
        const result = yield client.jobs.update(jobName, resourceGroupName, body);
        console.log(result);
    });
}
/**
 * This sample demonstrates how to Updates specific properties of a job. You can call this operation to notify the Import/Export service that the hard drives comprising the import or export job have been shipped to the Microsoft data center. It can also be used to cancel an existing job.
 *
 * @summary Updates specific properties of a job. You can call this operation to notify the Import/Export service that the hard drives comprising the import or export job have been shipped to the Microsoft data center. It can also be used to cancel an existing job.
 * x-ms-original-file: specification/storageimportexport/resource-manager/Microsoft.ImportExport/preview/2021-01-01/examples/UpdateJob.json
 */
function updateImportJob() {
    return __awaiter(this, void 0, void 0, function* () {
        const subscriptionId = process.env["STORAGEIMPORTEXPORT_SUBSCRIPTION_ID"] ||
            "xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx";
        const jobName = "myJob";
        const resourceGroupName = process.env["STORAGEIMPORTEXPORT_RESOURCE_GROUP"] || "myResourceGroup";
        const body = {
            backupDriveManifest: true,
            logLevel: "Verbose",
            state: ""
        };
        const credential = new DefaultAzureCredential();
        const client = new StorageImportExport(credential, subscriptionId);
        const result = yield client.jobs.update(jobName, resourceGroupName, body);
        console.log(result);
    });
}
function main() {
    return __awaiter(this, void 0, void 0, function* () {
        updateExportJob();
        updateImportJob();
    });
}
main().catch(console.error);
