// @azure/arm-managementgroups/dist-esm/src/operations/managementGroupSubscriptions.js
/*
 * Copyright (c) Microsoft Corporation.
 * Licensed under the MIT License.
 *
 * Code generated by Microsoft (R) AutoRest Code Generator.
 * Changes may cause incorrect behavior and will be lost if the code is regenerated.
 */
import { __asyncDelegator, __asyncGenerator, __asyncValues, __await } from "tslib";
import { setContinuationToken } from "../pagingHelper";
import * as coreClient from "@azure/core-client";
import * as Mappers from "../models/mappers";
import * as Parameters from "../models/parameters";
/// <reference lib="esnext.asynciterable" />
/** Class containing ManagementGroupSubscriptions operations. */
export class ManagementGroupSubscriptionsImpl {
    /**
     * Initialize a new instance of the class ManagementGroupSubscriptions class.
     * @param client Reference to the service client
     */
    constructor(client) {
        this.client = client;
    }
    /**
     * Retrieves details about all subscriptions which are associated with the management group.
     *
     * @param groupId Management Group ID.
     * @param options The options parameters.
     */
    listSubscriptionsUnderManagementGroup(groupId, options) {
        const iter = this.getSubscriptionsUnderManagementGroupPagingAll(groupId, options);
        return {
            next() {
                return iter.next();
            },
            [Symbol.asyncIterator]() {
                return this;
            },
            byPage: (settings) => {
                if (settings === null || settings === void 0 ? void 0 : settings.maxPageSize) {
                    throw new Error("maxPageSize is not supported by this operation.");
                }
                return this.getSubscriptionsUnderManagementGroupPagingPage(groupId, options, settings);
            }
        };
    }
    getSubscriptionsUnderManagementGroupPagingPage(groupId, options, settings) {
        return __asyncGenerator(this, arguments, function* getSubscriptionsUnderManagementGroupPagingPage_1() {
            let result;
            let continuationToken = settings === null || settings === void 0 ? void 0 : settings.continuationToken;
            if (!continuationToken) {
                result = yield __await(this._getSubscriptionsUnderManagementGroup(groupId, options));
                let page = result.value || [];
                continuationToken = result.nextLink;
                setContinuationToken(page, continuationToken);
                yield yield __await(page);
            }
            while (continuationToken) {
                result = yield __await(this._getSubscriptionsUnderManagementGroupNext(groupId, continuationToken, options));
                continuationToken = result.nextLink;
                let page = result.value || [];
                setContinuationToken(page, continuationToken);
                yield yield __await(page);
            }
        });
    }
    getSubscriptionsUnderManagementGroupPagingAll(groupId, options) {
        return __asyncGenerator(this, arguments, function* getSubscriptionsUnderManagementGroupPagingAll_1() {
            var e_1, _a;
            try {
                for (var _b = __asyncValues(this.getSubscriptionsUnderManagementGroupPagingPage(groupId, options)), _c; _c = yield __await(_b.next()), !_c.done;) {
                    const page = _c.value;
                    yield __await(yield* __asyncDelegator(__asyncValues(page)));
                }
            }
            catch (e_1_1) { e_1 = { error: e_1_1 }; }
            finally {
                try {
                    if (_c && !_c.done && (_a = _b.return)) yield __await(_a.call(_b));
                }
                finally { if (e_1) throw e_1.error; }
            }
        });
    }
    /**
     * Associates existing subscription with the management group.
     *
     * @param groupId Management Group ID.
     * @param subscriptionId Subscription ID.
     * @param options The options parameters.
     */
    create(groupId, subscriptionId, options) {
        return this.client.sendOperationRequest({ groupId, subscriptionId, options }, createOperationSpec);
    }
    /**
     * De-associates subscription from the management group.
     *
     * @param groupId Management Group ID.
     * @param subscriptionId Subscription ID.
     * @param options The options parameters.
     */
    delete(groupId, subscriptionId, options) {
        return this.client.sendOperationRequest({ groupId, subscriptionId, options }, deleteOperationSpec);
    }
    /**
     * Retrieves details about given subscription which is associated with the management group.
     *
     * @param groupId Management Group ID.
     * @param subscriptionId Subscription ID.
     * @param options The options parameters.
     */
    getSubscription(groupId, subscriptionId, options) {
        return this.client.sendOperationRequest({ groupId, subscriptionId, options }, getSubscriptionOperationSpec);
    }
    /**
     * Retrieves details about all subscriptions which are associated with the management group.
     *
     * @param groupId Management Group ID.
     * @param options The options parameters.
     */
    _getSubscriptionsUnderManagementGroup(groupId, options) {
        return this.client.sendOperationRequest({ groupId, options }, getSubscriptionsUnderManagementGroupOperationSpec);
    }
    /**
     * GetSubscriptionsUnderManagementGroupNext
     * @param groupId Management Group ID.
     * @param nextLink The nextLink from the previous successful call to the
     *                 GetSubscriptionsUnderManagementGroup method.
     * @param options The options parameters.
     */
    _getSubscriptionsUnderManagementGroupNext(groupId, nextLink, options) {
        return this.client.sendOperationRequest({ groupId, nextLink, options }, getSubscriptionsUnderManagementGroupNextOperationSpec);
    }
}
// Operation Specifications
const serializer = coreClient.createSerializer(Mappers, /* isXml */ false);
const createOperationSpec = {
    path: "/providers/Microsoft.Management/managementGroups/{groupId}/subscriptions/{subscriptionId}",
    httpMethod: "PUT",
    responses: {
        200: {
            bodyMapper: Mappers.SubscriptionUnderManagementGroup
        },
        default: {
            bodyMapper: Mappers.ErrorResponse
        }
    },
    queryParameters: [Parameters.apiVersion],
    urlParameters: [
        Parameters.$host,
        Parameters.groupId,
        Parameters.subscriptionId
    ],
    headerParameters: [Parameters.accept, Parameters.cacheControl],
    serializer
};
const deleteOperationSpec = {
    path: "/providers/Microsoft.Management/managementGroups/{groupId}/subscriptions/{subscriptionId}",
    httpMethod: "DELETE",
    responses: {
        200: {},
        204: {},
        default: {
            bodyMapper: Mappers.ErrorResponse
        }
    },
    queryParameters: [Parameters.apiVersion],
    urlParameters: [
        Parameters.$host,
        Parameters.groupId,
        Parameters.subscriptionId
    ],
    headerParameters: [Parameters.accept, Parameters.cacheControl],
    serializer
};
const getSubscriptionOperationSpec = {
    path: "/providers/Microsoft.Management/managementGroups/{groupId}/subscriptions/{subscriptionId}",
    httpMethod: "GET",
    responses: {
        200: {
            bodyMapper: Mappers.SubscriptionUnderManagementGroup
        },
        default: {
            bodyMapper: Mappers.ErrorResponse
        }
    },
    queryParameters: [Parameters.apiVersion],
    urlParameters: [
        Parameters.$host,
        Parameters.groupId,
        Parameters.subscriptionId
    ],
    headerParameters: [Parameters.accept, Parameters.cacheControl],
    serializer
};
const getSubscriptionsUnderManagementGroupOperationSpec = {
    path: "/providers/Microsoft.Management/managementGroups/{groupId}/subscriptions",
    httpMethod: "GET",
    responses: {
        200: {
            bodyMapper: Mappers.ListSubscriptionUnderManagementGroup
        },
        default: {
            bodyMapper: Mappers.ErrorResponse
        }
    },
    queryParameters: [Parameters.apiVersion, Parameters.skiptoken],
    urlParameters: [Parameters.$host, Parameters.groupId],
    headerParameters: [Parameters.accept],
    serializer
};
const getSubscriptionsUnderManagementGroupNextOperationSpec = {
    path: "{nextLink}",
    httpMethod: "GET",
    responses: {
        200: {
            bodyMapper: Mappers.ListSubscriptionUnderManagementGroup
        },
        default: {
            bodyMapper: Mappers.ErrorResponse
        }
    },
    queryParameters: [Parameters.apiVersion, Parameters.skiptoken],
    urlParameters: [Parameters.$host, Parameters.groupId, Parameters.nextLink],
    headerParameters: [Parameters.accept],
    serializer
};
