// @azure/arm-managementgroups/dist-esm/src/operations/managementGroups.js
/*
 * Copyright (c) Microsoft Corporation.
 * Licensed under the MIT License.
 *
 * Code generated by Microsoft (R) AutoRest Code Generator.
 * Changes may cause incorrect behavior and will be lost if the code is regenerated.
 */
import { __asyncDelegator, __asyncGenerator, __asyncValues, __await, __awaiter } from "tslib";
import { setContinuationToken } from "../pagingHelper";
import * as coreClient from "@azure/core-client";
import * as Mappers from "../models/mappers";
import * as Parameters from "../models/parameters";
import { LroEngine } from "@azure/core-lro";
import { LroImpl } from "../lroImpl";
/// <reference lib="esnext.asynciterable" />
/** Class containing ManagementGroups operations. */
export class ManagementGroupsImpl {
    /**
     * Initialize a new instance of the class ManagementGroups class.
     * @param client Reference to the service client
     */
    constructor(client) {
        this.client = client;
    }
    /**
     * List management groups for the authenticated user.
     *
     * @param options The options parameters.
     */
    list(options) {
        const iter = this.listPagingAll(options);
        return {
            next() {
                return iter.next();
            },
            [Symbol.asyncIterator]() {
                return this;
            },
            byPage: (settings) => {
                if (settings === null || settings === void 0 ? void 0 : settings.maxPageSize) {
                    throw new Error("maxPageSize is not supported by this operation.");
                }
                return this.listPagingPage(options, settings);
            }
        };
    }
    listPagingPage(options, settings) {
        return __asyncGenerator(this, arguments, function* listPagingPage_1() {
            let result;
            let continuationToken = settings === null || settings === void 0 ? void 0 : settings.continuationToken;
            if (!continuationToken) {
                result = yield __await(this._list(options));
                let page = result.value || [];
                continuationToken = result.nextLink;
                setContinuationToken(page, continuationToken);
                yield yield __await(page);
            }
            while (continuationToken) {
                result = yield __await(this._listNext(continuationToken, options));
                continuationToken = result.nextLink;
                let page = result.value || [];
                setContinuationToken(page, continuationToken);
                yield yield __await(page);
            }
        });
    }
    listPagingAll(options) {
        return __asyncGenerator(this, arguments, function* listPagingAll_1() {
            var e_1, _a;
            try {
                for (var _b = __asyncValues(this.listPagingPage(options)), _c; _c = yield __await(_b.next()), !_c.done;) {
                    const page = _c.value;
                    yield __await(yield* __asyncDelegator(__asyncValues(page)));
                }
            }
            catch (e_1_1) { e_1 = { error: e_1_1 }; }
            finally {
                try {
                    if (_c && !_c.done && (_a = _b.return)) yield __await(_a.call(_b));
                }
                finally { if (e_1) throw e_1.error; }
            }
        });
    }
    /**
     * List all entities that descend from a management group.
     *
     * @param groupId Management Group ID.
     * @param options The options parameters.
     */
    listDescendants(groupId, options) {
        const iter = this.getDescendantsPagingAll(groupId, options);
        return {
            next() {
                return iter.next();
            },
            [Symbol.asyncIterator]() {
                return this;
            },
            byPage: (settings) => {
                if (settings === null || settings === void 0 ? void 0 : settings.maxPageSize) {
                    throw new Error("maxPageSize is not supported by this operation.");
                }
                return this.getDescendantsPagingPage(groupId, options, settings);
            }
        };
    }
    getDescendantsPagingPage(groupId, options, settings) {
        return __asyncGenerator(this, arguments, function* getDescendantsPagingPage_1() {
            let result;
            let continuationToken = settings === null || settings === void 0 ? void 0 : settings.continuationToken;
            if (!continuationToken) {
                result = yield __await(this._getDescendants(groupId, options));
                let page = result.value || [];
                continuationToken = result.nextLink;
                setContinuationToken(page, continuationToken);
                yield yield __await(page);
            }
            while (continuationToken) {
                result = yield __await(this._getDescendantsNext(groupId, continuationToken, options));
                continuationToken = result.nextLink;
                let page = result.value || [];
                setContinuationToken(page, continuationToken);
                yield yield __await(page);
            }
        });
    }
    getDescendantsPagingAll(groupId, options) {
        return __asyncGenerator(this, arguments, function* getDescendantsPagingAll_1() {
            var e_2, _a;
            try {
                for (var _b = __asyncValues(this.getDescendantsPagingPage(groupId, options)), _c; _c = yield __await(_b.next()), !_c.done;) {
                    const page = _c.value;
                    yield __await(yield* __asyncDelegator(__asyncValues(page)));
                }
            }
            catch (e_2_1) { e_2 = { error: e_2_1 }; }
            finally {
                try {
                    if (_c && !_c.done && (_a = _b.return)) yield __await(_a.call(_b));
                }
                finally { if (e_2) throw e_2.error; }
            }
        });
    }
    /**
     * List management groups for the authenticated user.
     *
     * @param options The options parameters.
     */
    _list(options) {
        return this.client.sendOperationRequest({ options }, listOperationSpec);
    }
    /**
     * Get the details of the management group.
     *
     * @param groupId Management Group ID.
     * @param options The options parameters.
     */
    get(groupId, options) {
        return this.client.sendOperationRequest({ groupId, options }, getOperationSpec);
    }
    /**
     * Create or update a management group.
     * If a management group is already created and a subsequent create request is issued with different
     * properties, the management group properties will be updated.
     *
     * @param groupId Management Group ID.
     * @param createManagementGroupRequest Management group creation parameters.
     * @param options The options parameters.
     */
    beginCreateOrUpdate(groupId, createManagementGroupRequest, options) {
        return __awaiter(this, void 0, void 0, function* () {
            const directSendOperation = (args, spec) => __awaiter(this, void 0, void 0, function* () {
                return this.client.sendOperationRequest(args, spec);
            });
            const sendOperation = (args, spec) => __awaiter(this, void 0, void 0, function* () {
                var _a;
                let currentRawResponse = undefined;
                const providedCallback = (_a = args.options) === null || _a === void 0 ? void 0 : _a.onResponse;
                const callback = (rawResponse, flatResponse) => {
                    currentRawResponse = rawResponse;
                    providedCallback === null || providedCallback === void 0 ? void 0 : providedCallback(rawResponse, flatResponse);
                };
                const updatedArgs = Object.assign(Object.assign({}, args), { options: Object.assign(Object.assign({}, args.options), { onResponse: callback }) });
                const flatResponse = yield directSendOperation(updatedArgs, spec);
                return {
                    flatResponse,
                    rawResponse: {
                        statusCode: currentRawResponse.status,
                        body: currentRawResponse.parsedBody,
                        headers: currentRawResponse.headers.toJSON()
                    }
                };
            });
            const lro = new LroImpl(sendOperation, { groupId, createManagementGroupRequest, options }, createOrUpdateOperationSpec);
            const poller = new LroEngine(lro, {
                resumeFrom: options === null || options === void 0 ? void 0 : options.resumeFrom,
                intervalInMs: options === null || options === void 0 ? void 0 : options.updateIntervalInMs,
                lroResourceLocationConfig: "azure-async-operation"
            });
            yield poller.poll();
            return poller;
        });
    }
    /**
     * Create or update a management group.
     * If a management group is already created and a subsequent create request is issued with different
     * properties, the management group properties will be updated.
     *
     * @param groupId Management Group ID.
     * @param createManagementGroupRequest Management group creation parameters.
     * @param options The options parameters.
     */
    beginCreateOrUpdateAndWait(groupId, createManagementGroupRequest, options) {
        return __awaiter(this, void 0, void 0, function* () {
            const poller = yield this.beginCreateOrUpdate(groupId, createManagementGroupRequest, options);
            return poller.pollUntilDone();
        });
    }
    /**
     * Update a management group.
     *
     * @param groupId Management Group ID.
     * @param patchGroupRequest Management group patch parameters.
     * @param options The options parameters.
     */
    update(groupId, patchGroupRequest, options) {
        return this.client.sendOperationRequest({ groupId, patchGroupRequest, options }, updateOperationSpec);
    }
    /**
     * Delete management group.
     * If a management group contains child resources, the request will fail.
     *
     * @param groupId Management Group ID.
     * @param options The options parameters.
     */
    beginDelete(groupId, options) {
        return __awaiter(this, void 0, void 0, function* () {
            const directSendOperation = (args, spec) => __awaiter(this, void 0, void 0, function* () {
                return this.client.sendOperationRequest(args, spec);
            });
            const sendOperation = (args, spec) => __awaiter(this, void 0, void 0, function* () {
                var _a;
                let currentRawResponse = undefined;
                const providedCallback = (_a = args.options) === null || _a === void 0 ? void 0 : _a.onResponse;
                const callback = (rawResponse, flatResponse) => {
                    currentRawResponse = rawResponse;
                    providedCallback === null || providedCallback === void 0 ? void 0 : providedCallback(rawResponse, flatResponse);
                };
                const updatedArgs = Object.assign(Object.assign({}, args), { options: Object.assign(Object.assign({}, args.options), { onResponse: callback }) });
                const flatResponse = yield directSendOperation(updatedArgs, spec);
                return {
                    flatResponse,
                    rawResponse: {
                        statusCode: currentRawResponse.status,
                        body: currentRawResponse.parsedBody,
                        headers: currentRawResponse.headers.toJSON()
                    }
                };
            });
            const lro = new LroImpl(sendOperation, { groupId, options }, deleteOperationSpec);
            const poller = new LroEngine(lro, {
                resumeFrom: options === null || options === void 0 ? void 0 : options.resumeFrom,
                intervalInMs: options === null || options === void 0 ? void 0 : options.updateIntervalInMs,
                lroResourceLocationConfig: "azure-async-operation"
            });
            yield poller.poll();
            return poller;
        });
    }
    /**
     * Delete management group.
     * If a management group contains child resources, the request will fail.
     *
     * @param groupId Management Group ID.
     * @param options The options parameters.
     */
    beginDeleteAndWait(groupId, options) {
        return __awaiter(this, void 0, void 0, function* () {
            const poller = yield this.beginDelete(groupId, options);
            return poller.pollUntilDone();
        });
    }
    /**
     * List all entities that descend from a management group.
     *
     * @param groupId Management Group ID.
     * @param options The options parameters.
     */
    _getDescendants(groupId, options) {
        return this.client.sendOperationRequest({ groupId, options }, getDescendantsOperationSpec);
    }
    /**
     * ListNext
     * @param nextLink The nextLink from the previous successful call to the List method.
     * @param options The options parameters.
     */
    _listNext(nextLink, options) {
        return this.client.sendOperationRequest({ nextLink, options }, listNextOperationSpec);
    }
    /**
     * GetDescendantsNext
     * @param groupId Management Group ID.
     * @param nextLink The nextLink from the previous successful call to the GetDescendants method.
     * @param options The options parameters.
     */
    _getDescendantsNext(groupId, nextLink, options) {
        return this.client.sendOperationRequest({ groupId, nextLink, options }, getDescendantsNextOperationSpec);
    }
}
// Operation Specifications
const serializer = coreClient.createSerializer(Mappers, /* isXml */ false);
const listOperationSpec = {
    path: "/providers/Microsoft.Management/managementGroups",
    httpMethod: "GET",
    responses: {
        200: {
            bodyMapper: Mappers.ManagementGroupListResult
        },
        default: {
            bodyMapper: Mappers.ErrorResponse
        }
    },
    queryParameters: [Parameters.apiVersion, Parameters.skiptoken],
    urlParameters: [Parameters.$host],
    headerParameters: [Parameters.accept, Parameters.cacheControl],
    serializer
};
const getOperationSpec = {
    path: "/providers/Microsoft.Management/managementGroups/{groupId}",
    httpMethod: "GET",
    responses: {
        200: {
            bodyMapper: Mappers.ManagementGroup
        },
        default: {
            bodyMapper: Mappers.ErrorResponse
        }
    },
    queryParameters: [
        Parameters.apiVersion,
        Parameters.expand,
        Parameters.recurse,
        Parameters.filter
    ],
    urlParameters: [Parameters.$host, Parameters.groupId],
    headerParameters: [Parameters.accept, Parameters.cacheControl],
    serializer
};
const createOrUpdateOperationSpec = {
    path: "/providers/Microsoft.Management/managementGroups/{groupId}",
    httpMethod: "PUT",
    responses: {
        200: {
            bodyMapper: Mappers.ManagementGroup
        },
        201: {
            bodyMapper: Mappers.ManagementGroup
        },
        202: {
            bodyMapper: Mappers.ManagementGroup
        },
        204: {
            bodyMapper: Mappers.ManagementGroup
        },
        default: {
            bodyMapper: Mappers.ErrorResponse
        }
    },
    requestBody: Parameters.createManagementGroupRequest,
    queryParameters: [Parameters.apiVersion],
    urlParameters: [Parameters.$host, Parameters.groupId],
    headerParameters: [
        Parameters.accept,
        Parameters.cacheControl,
        Parameters.contentType
    ],
    mediaType: "json",
    serializer
};
const updateOperationSpec = {
    path: "/providers/Microsoft.Management/managementGroups/{groupId}",
    httpMethod: "PATCH",
    responses: {
        200: {
            bodyMapper: Mappers.ManagementGroup
        },
        default: {
            bodyMapper: Mappers.ErrorResponse
        }
    },
    requestBody: Parameters.patchGroupRequest,
    queryParameters: [Parameters.apiVersion],
    urlParameters: [Parameters.$host, Parameters.groupId],
    headerParameters: [
        Parameters.accept,
        Parameters.cacheControl,
        Parameters.contentType
    ],
    mediaType: "json",
    serializer
};
const deleteOperationSpec = {
    path: "/providers/Microsoft.Management/managementGroups/{groupId}",
    httpMethod: "DELETE",
    responses: {
        200: {
            bodyMapper: Mappers.AzureAsyncOperationResults,
            headersMapper: Mappers.ManagementGroupsDeleteHeaders
        },
        201: {
            bodyMapper: Mappers.AzureAsyncOperationResults,
            headersMapper: Mappers.ManagementGroupsDeleteHeaders
        },
        202: {
            bodyMapper: Mappers.AzureAsyncOperationResults,
            headersMapper: Mappers.ManagementGroupsDeleteHeaders
        },
        204: {
            bodyMapper: Mappers.AzureAsyncOperationResults,
            headersMapper: Mappers.ManagementGroupsDeleteHeaders
        },
        default: {
            bodyMapper: Mappers.ErrorResponse
        }
    },
    queryParameters: [Parameters.apiVersion],
    urlParameters: [Parameters.$host, Parameters.groupId],
    headerParameters: [Parameters.accept, Parameters.cacheControl],
    serializer
};
const getDescendantsOperationSpec = {
    path: "/providers/Microsoft.Management/managementGroups/{groupId}/descendants",
    httpMethod: "GET",
    responses: {
        200: {
            bodyMapper: Mappers.DescendantListResult
        },
        default: {
            bodyMapper: Mappers.ErrorResponse
        }
    },
    queryParameters: [
        Parameters.apiVersion,
        Parameters.skiptoken,
        Parameters.top
    ],
    urlParameters: [Parameters.$host, Parameters.groupId],
    headerParameters: [Parameters.accept],
    serializer
};
const listNextOperationSpec = {
    path: "{nextLink}",
    httpMethod: "GET",
    responses: {
        200: {
            bodyMapper: Mappers.ManagementGroupListResult
        },
        default: {
            bodyMapper: Mappers.ErrorResponse
        }
    },
    queryParameters: [Parameters.apiVersion, Parameters.skiptoken],
    urlParameters: [Parameters.$host, Parameters.nextLink],
    headerParameters: [Parameters.accept, Parameters.cacheControl],
    serializer
};
const getDescendantsNextOperationSpec = {
    path: "{nextLink}",
    httpMethod: "GET",
    responses: {
        200: {
            bodyMapper: Mappers.DescendantListResult
        },
        default: {
            bodyMapper: Mappers.ErrorResponse
        }
    },
    queryParameters: [
        Parameters.apiVersion,
        Parameters.skiptoken,
        Parameters.top
    ],
    urlParameters: [Parameters.$host, Parameters.groupId, Parameters.nextLink],
    headerParameters: [Parameters.accept],
    serializer
};
