// @azure/arm-frontdoor/dist-esm/src/operations/frontDoorNameAvailabilityWithSubscription.js
/*
 * Copyright (c) Microsoft Corporation.
 * Licensed under the MIT License.
 *
 * Code generated by Microsoft (R) AutoRest Code Generator.
 * Changes may cause incorrect behavior and will be lost if the code is regenerated.
 */
import * as coreClient from "@azure/core-client";
import * as Mappers from "../models/mappers";
import * as Parameters from "../models/parameters";
/** Class containing FrontDoorNameAvailabilityWithSubscription operations. */
export class FrontDoorNameAvailabilityWithSubscriptionImpl {
    /**
     * Initialize a new instance of the class FrontDoorNameAvailabilityWithSubscription class.
     * @param client Reference to the service client
     */
    constructor(client) {
        this.client = client;
    }
    /**
     * Check the availability of a Front Door subdomain.
     * @param checkFrontDoorNameAvailabilityInput Input to check.
     * @param options The options parameters.
     */
    check(checkFrontDoorNameAvailabilityInput, options) {
        return this.client.sendOperationRequest({ checkFrontDoorNameAvailabilityInput, options }, checkOperationSpec);
    }
}
// Operation Specifications
const serializer = coreClient.createSerializer(Mappers, /* isXml */ false);
const checkOperationSpec = {
    path: "/subscriptions/{subscriptionId}/providers/Microsoft.Network/checkFrontDoorNameAvailability",
    httpMethod: "POST",
    responses: {
        200: {
            bodyMapper: Mappers.CheckNameAvailabilityOutput,
        },
        default: {
            bodyMapper: Mappers.ErrorResponse,
        },
    },
    requestBody: Parameters.checkFrontDoorNameAvailabilityInput,
    queryParameters: [Parameters.apiVersion1],
    urlParameters: [Parameters.$host, Parameters.subscriptionId],
    headerParameters: [Parameters.accept, Parameters.contentType],
    mediaType: "json",
    serializer,
};
