// @azure/arm-servicemap/esm/operations/summaries.js
/*
 * Copyright (c) Microsoft Corporation. All rights reserved.
 * Licensed under the MIT License. See License.txt in the project root for
 * license information.
 *
 * Code generated by Microsoft (R) AutoRest Code Generator.
 * Changes may cause incorrect behavior and will be lost if the code is
 * regenerated.
 */
import * as msRest from "@azure/ms-rest-js";
import * as Mappers from "../models/summariesMappers";
import * as Parameters from "../models/parameters";
/** Class representing a Summaries. */
var Summaries = /** @class */ (function () {
    /**
     * Create a Summaries.
     * @param {ServicemapManagementClientContext} client Reference to the service client.
     */
    function Summaries(client) {
        this.client = client;
    }
    Summaries.prototype.getMachines = function (resourceGroupName, workspaceName, options, callback) {
        return this.client.sendOperationRequest({
            resourceGroupName: resourceGroupName,
            workspaceName: workspaceName,
            options: options
        }, getMachinesOperationSpec, callback);
    };
    return Summaries;
}());
export { Summaries };
// Operation Specifications
var serializer = new msRest.Serializer(Mappers);
var getMachinesOperationSpec = {
    httpMethod: "GET",
    path: "subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.OperationalInsights/workspaces/{workspaceName}/features/serviceMap/summaries/machines",
    urlParameters: [
        Parameters.subscriptionId,
        Parameters.resourceGroupName,
        Parameters.workspaceName
    ],
    queryParameters: [
        Parameters.apiVersion,
        Parameters.startTime,
        Parameters.endTime
    ],
    headerParameters: [
        Parameters.acceptLanguage
    ],
    responses: {
        200: {
            bodyMapper: Mappers.MachinesSummary
        },
        default: {
            bodyMapper: Mappers.ErrorResponse
        }
    },
    serializer: serializer
};
