// @azure/arm-storageimportexport/dist-esm/src/models/index.js
/*
 * Copyright (c) Microsoft Corporation.
 * Licensed under the MIT License.
 *
 * Code generated by Microsoft (R) AutoRest Code Generator.
 * Changes may cause incorrect behavior and will be lost if the code is regenerated.
 */
/** Known values of {@link CreatedByType} that the service accepts. */
export var KnownCreatedByType;
(function (KnownCreatedByType) {
    /** User */
    KnownCreatedByType["User"] = "User";
    /** Application */
    KnownCreatedByType["Application"] = "Application";
    /** ManagedIdentity */
    KnownCreatedByType["ManagedIdentity"] = "ManagedIdentity";
    /** Key */
    KnownCreatedByType["Key"] = "Key";
})(KnownCreatedByType || (KnownCreatedByType = {}));
/** Known values of {@link DriveState} that the service accepts. */
export var KnownDriveState;
(function (KnownDriveState) {
    /** Specified */
    KnownDriveState["Specified"] = "Specified";
    /** Received */
    KnownDriveState["Received"] = "Received";
    /** NeverReceived */
    KnownDriveState["NeverReceived"] = "NeverReceived";
    /** Transferring */
    KnownDriveState["Transferring"] = "Transferring";
    /** Completed */
    KnownDriveState["Completed"] = "Completed";
    /** CompletedMoreInfo */
    KnownDriveState["CompletedMoreInfo"] = "CompletedMoreInfo";
    /** ShippedBack */
    KnownDriveState["ShippedBack"] = "ShippedBack";
})(KnownDriveState || (KnownDriveState = {}));
/** Known values of {@link EncryptionKekType} that the service accepts. */
export var KnownEncryptionKekType;
(function (KnownEncryptionKekType) {
    /** MicrosoftManaged */
    KnownEncryptionKekType["MicrosoftManaged"] = "MicrosoftManaged";
    /** CustomerManaged */
    KnownEncryptionKekType["CustomerManaged"] = "CustomerManaged";
})(KnownEncryptionKekType || (KnownEncryptionKekType = {}));
/** Known values of {@link IdentityType} that the service accepts. */
export var KnownIdentityType;
(function (KnownIdentityType) {
    /** None */
    KnownIdentityType["None"] = "None";
    /** SystemAssigned */
    KnownIdentityType["SystemAssigned"] = "SystemAssigned";
    /** UserAssigned */
    KnownIdentityType["UserAssigned"] = "UserAssigned";
})(KnownIdentityType || (KnownIdentityType = {}));
