// @azure/arm-commerce/esm/operations/rateCard.js
/*
 * Copyright (c) Microsoft Corporation.
 * Licensed under the MIT License.
 *
 * Code generated by Microsoft (R) AutoRest Code Generator.
 * Changes may cause incorrect behavior and will be lost if the code is
 * regenerated.
 */
import * as msRest from "@azure/ms-rest-js";
import * as Mappers from "../models/rateCardMappers";
import * as Parameters from "../models/parameters";
/** Class representing a RateCard. */
var RateCard = /** @class */ (function () {
    /**
     * Create a RateCard.
     * @param {UsageManagementClientContext} client Reference to the service client.
     */
    function RateCard(client) {
        this.client = client;
    }
    RateCard.prototype.get = function (filter, options, callback) {
        return this.client.sendOperationRequest({
            filter: filter,
            options: options
        }, getOperationSpec, callback);
    };
    return RateCard;
}());
export { RateCard };
// Operation Specifications
var serializer = new msRest.Serializer(Mappers);
var getOperationSpec = {
    httpMethod: "GET",
    path: "subscriptions/{subscriptionId}/providers/Microsoft.Commerce/RateCard",
    urlParameters: [
        Parameters.subscriptionId
    ],
    queryParameters: [
        Parameters.filter,
        Parameters.apiVersion
    ],
    headerParameters: [
        Parameters.acceptLanguage
    ],
    responses: {
        200: {
            bodyMapper: Mappers.ResourceRateCardInfo
        },
        default: {
            bodyMapper: Mappers.ErrorResponse
        }
    },
    serializer: serializer
};
