// @azure/arm-commerce/esm/operations/usageAggregates.js
/*
 * Copyright (c) Microsoft Corporation.
 * Licensed under the MIT License.
 *
 * Code generated by Microsoft (R) AutoRest Code Generator.
 * Changes may cause incorrect behavior and will be lost if the code is
 * regenerated.
 */
import * as msRest from "@azure/ms-rest-js";
import * as Mappers from "../models/usageAggregatesMappers";
import * as Parameters from "../models/parameters";
/** Class representing a UsageAggregates. */
var UsageAggregates = /** @class */ (function () {
    /**
     * Create a UsageAggregates.
     * @param {UsageManagementClientContext} client Reference to the service client.
     */
    function UsageAggregates(client) {
        this.client = client;
    }
    UsageAggregates.prototype.list = function (reportedStartTime, reportedEndTime, options, callback) {
        return this.client.sendOperationRequest({
            reportedStartTime: reportedStartTime,
            reportedEndTime: reportedEndTime,
            options: options
        }, listOperationSpec, callback);
    };
    UsageAggregates.prototype.listNext = function (nextPageLink, reportedStartTime, reportedEndTime, options, callback) {
        return this.client.sendOperationRequest({
            nextPageLink: nextPageLink,
            reportedStartTime: reportedStartTime,
            reportedEndTime: reportedEndTime,
            options: options
        }, listNextOperationSpec, callback);
    };
    return UsageAggregates;
}());
export { UsageAggregates };
// Operation Specifications
var serializer = new msRest.Serializer(Mappers);
var listOperationSpec = {
    httpMethod: "GET",
    path: "subscriptions/{subscriptionId}/providers/Microsoft.Commerce/UsageAggregates",
    urlParameters: [
        Parameters.subscriptionId
    ],
    queryParameters: [
        Parameters.reportedStartTime,
        Parameters.reportedEndTime,
        Parameters.showDetails,
        Parameters.aggregationGranularity,
        Parameters.continuationToken,
        Parameters.apiVersion
    ],
    headerParameters: [
        Parameters.acceptLanguage
    ],
    responses: {
        200: {
            bodyMapper: Mappers.UsageAggregationListResult
        },
        default: {
            bodyMapper: Mappers.ErrorResponse
        }
    },
    serializer: serializer
};
var listNextOperationSpec = {
    httpMethod: "GET",
    baseUrl: "https://management.azure.com",
    path: "{nextLink}",
    urlParameters: [
        Parameters.nextPageLink
    ],
    queryParameters: [
        Parameters.reportedStartTime,
        Parameters.reportedEndTime,
        Parameters.showDetails,
        Parameters.aggregationGranularity,
        Parameters.continuationToken,
        Parameters.apiVersion
    ],
    headerParameters: [
        Parameters.acceptLanguage
    ],
    responses: {
        200: {
            bodyMapper: Mappers.UsageAggregationListResult
        },
        default: {
            bodyMapper: Mappers.ErrorResponse
        }
    },
    serializer: serializer
};
