// @azure/arm-containerservice/dist-esm/test/containerservice_examples.js
/*
 * Copyright (c) Microsoft Corporation.
 * Licensed under the MIT License.
 *
 * Code generated by Microsoft (R) AutoRest Code Generator.
 * Changes may cause incorrect behavior and will be lost if the code is regenerated.
 */
import { __asyncValues, __awaiter } from "tslib";
import { env, Recorder, isPlaybackMode, } from "@azure-tools/test-recorder";
import { createTestCredential } from "@azure-tools/test-credential";
import { assert } from "chai";
import { ContainerServiceClient } from "../src/containerServiceClient";
const replaceableVariables = {
    AZURE_CLIENT_ID: "azure_client_id",
    AZURE_CLIENT_SECRET: "azure_client_secret",
    AZURE_TENANT_ID: "88888888-8888-8888-8888-888888888888",
    SUBSCRIPTION_ID: "azure_subscription_id"
};
const recorderOptions = {
    envSetupForPlayback: replaceableVariables
};
export const testPollingOptions = {
    updateIntervalInMs: isPlaybackMode() ? 0 : undefined,
};
describe("ContainerService test", () => {
    let recorder;
    let subscriptionId;
    let clientId;
    let secret;
    let client;
    let location;
    let resourceGroupName;
    let resourceName;
    beforeEach(function () {
        return __awaiter(this, void 0, void 0, function* () {
            recorder = new Recorder(this.currentTest);
            yield recorder.start(recorderOptions);
            subscriptionId = env.SUBSCRIPTION_ID || '';
            clientId = env.AZURE_CLIENT_ID || '';
            secret = env.AZURE_CLIENT_SECRET || '';
            // This is an example of how the environment variables are used
            const credential = createTestCredential();
            client = new ContainerServiceClient(credential, subscriptionId, recorder.configureClientOptions({}));
            location = "eastus";
            resourceGroupName = "myjstest";
            resourceName = "myreourcexyz";
        });
    });
    afterEach(function () {
        return __awaiter(this, void 0, void 0, function* () {
            yield recorder.stop();
        });
    });
    it("managedClusters create test", function () {
        return __awaiter(this, void 0, void 0, function* () {
            const res = yield client.managedClusters.beginCreateOrUpdateAndWait(resourceGroupName, resourceName, {
                dnsPrefix: "aksjssdk",
                agentPoolProfiles: [
                    {
                        name: "aksagent",
                        count: 1,
                        vmSize: "Standard_DS2_v2",
                        maxPods: 110,
                        minCount: 1,
                        maxCount: 100,
                        osType: "Linux",
                        type: "VirtualMachineScaleSets",
                        enableAutoScaling: true,
                        mode: "System"
                    }
                ],
                servicePrincipalProfile: {
                    clientId: clientId,
                    secret: secret
                },
                location: location
            }, testPollingOptions);
            assert.equal(res.name, resourceName);
        });
    });
    it("managedClusters get test", function () {
        return __awaiter(this, void 0, void 0, function* () {
            const res = yield client.managedClusters.get(resourceGroupName, resourceName);
            assert.equal(res.name, resourceName);
        });
    });
    it("managedClusters getUpgradeProfile test", function () {
        return __awaiter(this, void 0, void 0, function* () {
            const res = yield client.managedClusters.getUpgradeProfile(resourceGroupName, resourceName);
            assert.equal(res.name, "default");
        });
    });
    it("managedClusters list test", function () {
        var _a, e_1, _b, _c;
        return __awaiter(this, void 0, void 0, function* () {
            const resArray = new Array();
            try {
                for (var _d = true, _e = __asyncValues(client.managedClusters.listByResourceGroup(resourceGroupName)), _f; _f = yield _e.next(), _a = _f.done, !_a;) {
                    _c = _f.value;
                    _d = false;
                    try {
                        let item = _c;
                        resArray.push(item);
                    }
                    finally {
                        _d = true;
                    }
                }
            }
            catch (e_1_1) { e_1 = { error: e_1_1 }; }
            finally {
                try {
                    if (!_d && !_a && (_b = _e.return)) yield _b.call(_e);
                }
                finally { if (e_1) throw e_1.error; }
            }
            assert.equal(resArray.length, 1);
        });
    });
    it("managedClusters update test", function () {
        return __awaiter(this, void 0, void 0, function* () {
            const res = yield client.managedClusters.beginUpdateTagsAndWait(resourceGroupName, resourceName, { tags: { tier: "testing", archv3: "" } }, testPollingOptions);
            assert.equal(res.type, "Microsoft.ContainerService/ManagedClusters");
        });
    });
    it("managedClusters delete test", function () {
        var _a, e_2, _b, _c;
        return __awaiter(this, void 0, void 0, function* () {
            const res = yield client.managedClusters.beginDeleteAndWait(resourceGroupName, resourceName, testPollingOptions);
            const resArray = new Array();
            try {
                for (var _d = true, _e = __asyncValues(client.managedClusters.listByResourceGroup(resourceGroupName)), _f; _f = yield _e.next(), _a = _f.done, !_a;) {
                    _c = _f.value;
                    _d = false;
                    try {
                        let item = _c;
                        resArray.push(item);
                    }
                    finally {
                        _d = true;
                    }
                }
            }
            catch (e_2_1) { e_2 = { error: e_2_1 }; }
            finally {
                try {
                    if (!_d && !_a && (_b = _e.return)) yield _b.call(_e);
                }
                finally { if (e_2) throw e_2.error; }
            }
            assert.equal(resArray.length, 0);
        });
    });
});
