// @azure/arm-security/dist-esm/src/operations/alertsSuppressionRules.js
/*
 * Copyright (c) Microsoft Corporation.
 * Licensed under the MIT License.
 *
 * Code generated by Microsoft (R) AutoRest Code Generator.
 * Changes may cause incorrect behavior and will be lost if the code is regenerated.
 */
import { __asyncDelegator, __asyncGenerator, __asyncValues, __await } from "tslib";
import * as coreClient from "@azure/core-client";
import * as Mappers from "../models/mappers";
import * as Parameters from "../models/parameters";
/// <reference lib="esnext.asynciterable" />
/** Class containing AlertsSuppressionRules operations. */
export class AlertsSuppressionRulesImpl {
    /**
     * Initialize a new instance of the class AlertsSuppressionRules class.
     * @param client Reference to the service client
     */
    constructor(client) {
        this.client = client;
    }
    /**
     * List of all the dismiss rules for the given subscription
     * @param options The options parameters.
     */
    list(options) {
        const iter = this.listPagingAll(options);
        return {
            next() {
                return iter.next();
            },
            [Symbol.asyncIterator]() {
                return this;
            },
            byPage: () => {
                return this.listPagingPage(options);
            }
        };
    }
    listPagingPage(options) {
        return __asyncGenerator(this, arguments, function* listPagingPage_1() {
            let result = yield __await(this._list(options));
            yield yield __await(result.value || []);
            let continuationToken = result.nextLink;
            while (continuationToken) {
                result = yield __await(this._listNext(continuationToken, options));
                continuationToken = result.nextLink;
                yield yield __await(result.value || []);
            }
        });
    }
    listPagingAll(options) {
        return __asyncGenerator(this, arguments, function* listPagingAll_1() {
            var e_1, _a;
            try {
                for (var _b = __asyncValues(this.listPagingPage(options)), _c; _c = yield __await(_b.next()), !_c.done;) {
                    const page = _c.value;
                    yield __await(yield* __asyncDelegator(__asyncValues(page)));
                }
            }
            catch (e_1_1) { e_1 = { error: e_1_1 }; }
            finally {
                try {
                    if (_c && !_c.done && (_a = _b.return)) yield __await(_a.call(_b));
                }
                finally { if (e_1) throw e_1.error; }
            }
        });
    }
    /**
     * List of all the dismiss rules for the given subscription
     * @param options The options parameters.
     */
    _list(options) {
        return this.client.sendOperationRequest({ options }, listOperationSpec);
    }
    /**
     * Get dismiss rule, with name: {alertsSuppressionRuleName}, for the given subscription
     * @param alertsSuppressionRuleName The unique name of the suppression alert rule
     * @param options The options parameters.
     */
    get(alertsSuppressionRuleName, options) {
        return this.client.sendOperationRequest({ alertsSuppressionRuleName, options }, getOperationSpec);
    }
    /**
     * Update existing rule or create new rule if it doesn't exist
     * @param alertsSuppressionRuleName The unique name of the suppression alert rule
     * @param alertsSuppressionRule Suppression rule object
     * @param options The options parameters.
     */
    update(alertsSuppressionRuleName, alertsSuppressionRule, options) {
        return this.client.sendOperationRequest({ alertsSuppressionRuleName, alertsSuppressionRule, options }, updateOperationSpec);
    }
    /**
     * Delete dismiss alert rule for this subscription.
     * @param alertsSuppressionRuleName The unique name of the suppression alert rule
     * @param options The options parameters.
     */
    delete(alertsSuppressionRuleName, options) {
        return this.client.sendOperationRequest({ alertsSuppressionRuleName, options }, deleteOperationSpec);
    }
    /**
     * ListNext
     * @param nextLink The nextLink from the previous successful call to the List method.
     * @param options The options parameters.
     */
    _listNext(nextLink, options) {
        return this.client.sendOperationRequest({ nextLink, options }, listNextOperationSpec);
    }
}
// Operation Specifications
const serializer = coreClient.createSerializer(Mappers, /* isXml */ false);
const listOperationSpec = {
    path: "/subscriptions/{subscriptionId}/providers/Microsoft.Security/alertsSuppressionRules",
    httpMethod: "GET",
    responses: {
        200: {
            bodyMapper: Mappers.AlertsSuppressionRulesList
        },
        default: {
            bodyMapper: Mappers.CloudError
        }
    },
    queryParameters: [Parameters.apiVersion8, Parameters.alertType],
    urlParameters: [Parameters.$host, Parameters.subscriptionId],
    headerParameters: [Parameters.accept],
    serializer
};
const getOperationSpec = {
    path: "/subscriptions/{subscriptionId}/providers/Microsoft.Security/alertsSuppressionRules/{alertsSuppressionRuleName}",
    httpMethod: "GET",
    responses: {
        200: {
            bodyMapper: Mappers.AlertsSuppressionRule
        },
        default: {
            bodyMapper: Mappers.CloudError
        }
    },
    queryParameters: [Parameters.apiVersion8],
    urlParameters: [
        Parameters.$host,
        Parameters.subscriptionId,
        Parameters.alertsSuppressionRuleName
    ],
    headerParameters: [Parameters.accept],
    serializer
};
const updateOperationSpec = {
    path: "/subscriptions/{subscriptionId}/providers/Microsoft.Security/alertsSuppressionRules/{alertsSuppressionRuleName}",
    httpMethod: "PUT",
    responses: {
        200: {
            bodyMapper: Mappers.AlertsSuppressionRule
        },
        default: {
            bodyMapper: Mappers.CloudError
        }
    },
    requestBody: Parameters.alertsSuppressionRule,
    queryParameters: [Parameters.apiVersion8],
    urlParameters: [
        Parameters.$host,
        Parameters.subscriptionId,
        Parameters.alertsSuppressionRuleName
    ],
    headerParameters: [Parameters.accept, Parameters.contentType],
    mediaType: "json",
    serializer
};
const deleteOperationSpec = {
    path: "/subscriptions/{subscriptionId}/providers/Microsoft.Security/alertsSuppressionRules/{alertsSuppressionRuleName}",
    httpMethod: "DELETE",
    responses: {
        204: {},
        default: {
            bodyMapper: Mappers.CloudError
        }
    },
    queryParameters: [Parameters.apiVersion8],
    urlParameters: [
        Parameters.$host,
        Parameters.subscriptionId,
        Parameters.alertsSuppressionRuleName
    ],
    headerParameters: [Parameters.accept],
    serializer
};
const listNextOperationSpec = {
    path: "{nextLink}",
    httpMethod: "GET",
    responses: {
        200: {
            bodyMapper: Mappers.AlertsSuppressionRulesList
        },
        default: {
            bodyMapper: Mappers.CloudError
        }
    },
    queryParameters: [Parameters.apiVersion8, Parameters.alertType],
    urlParameters: [
        Parameters.$host,
        Parameters.subscriptionId,
        Parameters.nextLink
    ],
    headerParameters: [Parameters.accept],
    serializer
};
