// @azure/arm-security/dist-esm/src/operations/connectors.js
/*
 * Copyright (c) Microsoft Corporation.
 * Licensed under the MIT License.
 *
 * Code generated by Microsoft (R) AutoRest Code Generator.
 * Changes may cause incorrect behavior and will be lost if the code is regenerated.
 */
import { __asyncDelegator, __asyncGenerator, __asyncValues, __await } from "tslib";
import * as coreClient from "@azure/core-client";
import * as Mappers from "../models/mappers";
import * as Parameters from "../models/parameters";
/// <reference lib="esnext.asynciterable" />
/** Class containing Connectors operations. */
export class ConnectorsImpl {
    /**
     * Initialize a new instance of the class Connectors class.
     * @param client Reference to the service client
     */
    constructor(client) {
        this.client = client;
    }
    /**
     * Cloud accounts connectors of a subscription
     * @param options The options parameters.
     */
    list(options) {
        const iter = this.listPagingAll(options);
        return {
            next() {
                return iter.next();
            },
            [Symbol.asyncIterator]() {
                return this;
            },
            byPage: () => {
                return this.listPagingPage(options);
            }
        };
    }
    listPagingPage(options) {
        return __asyncGenerator(this, arguments, function* listPagingPage_1() {
            let result = yield __await(this._list(options));
            yield yield __await(result.value || []);
            let continuationToken = result.nextLink;
            while (continuationToken) {
                result = yield __await(this._listNext(continuationToken, options));
                continuationToken = result.nextLink;
                yield yield __await(result.value || []);
            }
        });
    }
    listPagingAll(options) {
        return __asyncGenerator(this, arguments, function* listPagingAll_1() {
            var e_1, _a;
            try {
                for (var _b = __asyncValues(this.listPagingPage(options)), _c; _c = yield __await(_b.next()), !_c.done;) {
                    const page = _c.value;
                    yield __await(yield* __asyncDelegator(__asyncValues(page)));
                }
            }
            catch (e_1_1) { e_1 = { error: e_1_1 }; }
            finally {
                try {
                    if (_c && !_c.done && (_a = _b.return)) yield __await(_a.call(_b));
                }
                finally { if (e_1) throw e_1.error; }
            }
        });
    }
    /**
     * Cloud accounts connectors of a subscription
     * @param options The options parameters.
     */
    _list(options) {
        return this.client.sendOperationRequest({ options }, listOperationSpec);
    }
    /**
     * Details of a specific cloud account connector
     * @param connectorName Name of the cloud account connector
     * @param options The options parameters.
     */
    get(connectorName, options) {
        return this.client.sendOperationRequest({ connectorName, options }, getOperationSpec);
    }
    /**
     * Create a cloud account connector or update an existing one. Connect to your cloud account. For AWS,
     * use either account credentials or role-based authentication. For GCP, use account organization
     * credentials.
     * @param connectorName Name of the cloud account connector
     * @param connectorSetting Settings for the cloud account connector
     * @param options The options parameters.
     */
    createOrUpdate(connectorName, connectorSetting, options) {
        return this.client.sendOperationRequest({ connectorName, connectorSetting, options }, createOrUpdateOperationSpec);
    }
    /**
     * Delete a cloud account connector from a subscription
     * @param connectorName Name of the cloud account connector
     * @param options The options parameters.
     */
    delete(connectorName, options) {
        return this.client.sendOperationRequest({ connectorName, options }, deleteOperationSpec);
    }
    /**
     * ListNext
     * @param nextLink The nextLink from the previous successful call to the List method.
     * @param options The options parameters.
     */
    _listNext(nextLink, options) {
        return this.client.sendOperationRequest({ nextLink, options }, listNextOperationSpec);
    }
}
// Operation Specifications
const serializer = coreClient.createSerializer(Mappers, /* isXml */ false);
const listOperationSpec = {
    path: "/subscriptions/{subscriptionId}/providers/Microsoft.Security/connectors",
    httpMethod: "GET",
    responses: {
        200: {
            bodyMapper: Mappers.ConnectorSettingList
        },
        default: {
            bodyMapper: Mappers.CloudError
        }
    },
    queryParameters: [Parameters.apiVersion11],
    urlParameters: [Parameters.$host, Parameters.subscriptionId],
    headerParameters: [Parameters.accept],
    serializer
};
const getOperationSpec = {
    path: "/subscriptions/{subscriptionId}/providers/Microsoft.Security/connectors/{connectorName}",
    httpMethod: "GET",
    responses: {
        200: {
            bodyMapper: Mappers.ConnectorSetting
        },
        default: {
            bodyMapper: Mappers.CloudError
        }
    },
    queryParameters: [Parameters.apiVersion11],
    urlParameters: [
        Parameters.$host,
        Parameters.subscriptionId,
        Parameters.connectorName
    ],
    headerParameters: [Parameters.accept],
    serializer
};
const createOrUpdateOperationSpec = {
    path: "/subscriptions/{subscriptionId}/providers/Microsoft.Security/connectors/{connectorName}",
    httpMethod: "PUT",
    responses: {
        200: {
            bodyMapper: Mappers.ConnectorSetting
        },
        default: {
            bodyMapper: Mappers.CloudError
        }
    },
    requestBody: Parameters.connectorSetting,
    queryParameters: [Parameters.apiVersion11],
    urlParameters: [
        Parameters.$host,
        Parameters.subscriptionId,
        Parameters.connectorName
    ],
    headerParameters: [Parameters.accept, Parameters.contentType],
    mediaType: "json",
    serializer
};
const deleteOperationSpec = {
    path: "/subscriptions/{subscriptionId}/providers/Microsoft.Security/connectors/{connectorName}",
    httpMethod: "DELETE",
    responses: {
        200: {},
        204: {},
        default: {
            bodyMapper: Mappers.CloudError
        }
    },
    queryParameters: [Parameters.apiVersion11],
    urlParameters: [
        Parameters.$host,
        Parameters.subscriptionId,
        Parameters.connectorName
    ],
    headerParameters: [Parameters.accept],
    serializer
};
const listNextOperationSpec = {
    path: "{nextLink}",
    httpMethod: "GET",
    responses: {
        200: {
            bodyMapper: Mappers.ConnectorSettingList
        },
        default: {
            bodyMapper: Mappers.CloudError
        }
    },
    queryParameters: [Parameters.apiVersion11],
    urlParameters: [
        Parameters.$host,
        Parameters.subscriptionId,
        Parameters.nextLink
    ],
    headerParameters: [Parameters.accept],
    serializer
};
