// @azure/arm-security/dist-esm/src/operations/ingestionSettings.js
/*
 * Copyright (c) Microsoft Corporation.
 * Licensed under the MIT License.
 *
 * Code generated by Microsoft (R) AutoRest Code Generator.
 * Changes may cause incorrect behavior and will be lost if the code is regenerated.
 */
import { __asyncDelegator, __asyncGenerator, __asyncValues, __await } from "tslib";
import * as coreClient from "@azure/core-client";
import * as Mappers from "../models/mappers";
import * as Parameters from "../models/parameters";
/// <reference lib="esnext.asynciterable" />
/** Class containing IngestionSettings operations. */
export class IngestionSettingsImpl {
    /**
     * Initialize a new instance of the class IngestionSettings class.
     * @param client Reference to the service client
     */
    constructor(client) {
        this.client = client;
    }
    /**
     * Settings for ingesting security data and logs to correlate with resources associated with the
     * subscription.
     * @param options The options parameters.
     */
    list(options) {
        const iter = this.listPagingAll(options);
        return {
            next() {
                return iter.next();
            },
            [Symbol.asyncIterator]() {
                return this;
            },
            byPage: () => {
                return this.listPagingPage(options);
            }
        };
    }
    listPagingPage(options) {
        return __asyncGenerator(this, arguments, function* listPagingPage_1() {
            let result = yield __await(this._list(options));
            yield yield __await(result.value || []);
            let continuationToken = result.nextLink;
            while (continuationToken) {
                result = yield __await(this._listNext(continuationToken, options));
                continuationToken = result.nextLink;
                yield yield __await(result.value || []);
            }
        });
    }
    listPagingAll(options) {
        return __asyncGenerator(this, arguments, function* listPagingAll_1() {
            var e_1, _a;
            try {
                for (var _b = __asyncValues(this.listPagingPage(options)), _c; _c = yield __await(_b.next()), !_c.done;) {
                    const page = _c.value;
                    yield __await(yield* __asyncDelegator(__asyncValues(page)));
                }
            }
            catch (e_1_1) { e_1 = { error: e_1_1 }; }
            finally {
                try {
                    if (_c && !_c.done && (_a = _b.return)) yield __await(_a.call(_b));
                }
                finally { if (e_1) throw e_1.error; }
            }
        });
    }
    /**
     * Settings for ingesting security data and logs to correlate with resources associated with the
     * subscription.
     * @param options The options parameters.
     */
    _list(options) {
        return this.client.sendOperationRequest({ options }, listOperationSpec);
    }
    /**
     * Settings for ingesting security data and logs to correlate with resources associated with the
     * subscription.
     * @param ingestionSettingName Name of the ingestion setting
     * @param options The options parameters.
     */
    get(ingestionSettingName, options) {
        return this.client.sendOperationRequest({ ingestionSettingName, options }, getOperationSpec);
    }
    /**
     * Create setting for ingesting security data and logs to correlate with resources associated with the
     * subscription.
     * @param ingestionSettingName Name of the ingestion setting
     * @param ingestionSetting Ingestion setting object
     * @param options The options parameters.
     */
    create(ingestionSettingName, ingestionSetting, options) {
        return this.client.sendOperationRequest({ ingestionSettingName, ingestionSetting, options }, createOperationSpec);
    }
    /**
     * Deletes the ingestion settings for this subscription.
     * @param ingestionSettingName Name of the ingestion setting
     * @param options The options parameters.
     */
    delete(ingestionSettingName, options) {
        return this.client.sendOperationRequest({ ingestionSettingName, options }, deleteOperationSpec);
    }
    /**
     * Returns the token that is used for correlating ingested telemetry with the resources in the
     * subscription.
     * @param ingestionSettingName Name of the ingestion setting
     * @param options The options parameters.
     */
    listTokens(ingestionSettingName, options) {
        return this.client.sendOperationRequest({ ingestionSettingName, options }, listTokensOperationSpec);
    }
    /**
     * Connection strings for ingesting security scan logs and data.
     * @param ingestionSettingName Name of the ingestion setting
     * @param options The options parameters.
     */
    listConnectionStrings(ingestionSettingName, options) {
        return this.client.sendOperationRequest({ ingestionSettingName, options }, listConnectionStringsOperationSpec);
    }
    /**
     * ListNext
     * @param nextLink The nextLink from the previous successful call to the List method.
     * @param options The options parameters.
     */
    _listNext(nextLink, options) {
        return this.client.sendOperationRequest({ nextLink, options }, listNextOperationSpec);
    }
}
// Operation Specifications
const serializer = coreClient.createSerializer(Mappers, /* isXml */ false);
const listOperationSpec = {
    path: "/subscriptions/{subscriptionId}/providers/Microsoft.Security/ingestionSettings",
    httpMethod: "GET",
    responses: {
        200: {
            bodyMapper: Mappers.IngestionSettingList
        },
        default: {
            bodyMapper: Mappers.CloudError
        }
    },
    queryParameters: [Parameters.apiVersion15],
    urlParameters: [Parameters.$host, Parameters.subscriptionId],
    headerParameters: [Parameters.accept],
    serializer
};
const getOperationSpec = {
    path: "/subscriptions/{subscriptionId}/providers/Microsoft.Security/ingestionSettings/{ingestionSettingName}",
    httpMethod: "GET",
    responses: {
        200: {
            bodyMapper: Mappers.IngestionSetting
        },
        default: {
            bodyMapper: Mappers.CloudError
        }
    },
    queryParameters: [Parameters.apiVersion15],
    urlParameters: [
        Parameters.$host,
        Parameters.subscriptionId,
        Parameters.ingestionSettingName
    ],
    headerParameters: [Parameters.accept],
    serializer
};
const createOperationSpec = {
    path: "/subscriptions/{subscriptionId}/providers/Microsoft.Security/ingestionSettings/{ingestionSettingName}",
    httpMethod: "PUT",
    responses: {
        200: {
            bodyMapper: Mappers.IngestionSetting
        },
        default: {
            bodyMapper: Mappers.CloudError
        }
    },
    requestBody: Parameters.ingestionSetting,
    queryParameters: [Parameters.apiVersion15],
    urlParameters: [
        Parameters.$host,
        Parameters.subscriptionId,
        Parameters.ingestionSettingName
    ],
    headerParameters: [Parameters.accept, Parameters.contentType],
    mediaType: "json",
    serializer
};
const deleteOperationSpec = {
    path: "/subscriptions/{subscriptionId}/providers/Microsoft.Security/ingestionSettings/{ingestionSettingName}",
    httpMethod: "DELETE",
    responses: {
        200: {},
        204: {},
        default: {
            bodyMapper: Mappers.CloudError
        }
    },
    queryParameters: [Parameters.apiVersion15],
    urlParameters: [
        Parameters.$host,
        Parameters.subscriptionId,
        Parameters.ingestionSettingName
    ],
    headerParameters: [Parameters.accept],
    serializer
};
const listTokensOperationSpec = {
    path: "/subscriptions/{subscriptionId}/providers/Microsoft.Security/ingestionSettings/{ingestionSettingName}/listTokens",
    httpMethod: "POST",
    responses: {
        200: {
            bodyMapper: Mappers.IngestionSettingToken
        },
        default: {
            bodyMapper: Mappers.CloudError
        }
    },
    queryParameters: [Parameters.apiVersion15],
    urlParameters: [
        Parameters.$host,
        Parameters.subscriptionId,
        Parameters.ingestionSettingName
    ],
    headerParameters: [Parameters.accept],
    serializer
};
const listConnectionStringsOperationSpec = {
    path: "/subscriptions/{subscriptionId}/providers/Microsoft.Security/ingestionSettings/{ingestionSettingName}/listConnectionStrings",
    httpMethod: "POST",
    responses: {
        200: {
            bodyMapper: Mappers.ConnectionStrings
        },
        default: {
            bodyMapper: Mappers.CloudError
        }
    },
    queryParameters: [Parameters.apiVersion15],
    urlParameters: [
        Parameters.$host,
        Parameters.subscriptionId,
        Parameters.ingestionSettingName
    ],
    headerParameters: [Parameters.accept],
    serializer
};
const listNextOperationSpec = {
    path: "{nextLink}",
    httpMethod: "GET",
    responses: {
        200: {
            bodyMapper: Mappers.IngestionSettingList
        },
        default: {
            bodyMapper: Mappers.CloudError
        }
    },
    queryParameters: [Parameters.apiVersion15],
    urlParameters: [
        Parameters.$host,
        Parameters.subscriptionId,
        Parameters.nextLink
    ],
    headerParameters: [Parameters.accept],
    serializer
};
