// @azure/arm-security/dist-esm/src/operations/iotSecuritySolutionAnalytics.js
/*
 * Copyright (c) Microsoft Corporation.
 * Licensed under the MIT License.
 *
 * Code generated by Microsoft (R) AutoRest Code Generator.
 * Changes may cause incorrect behavior and will be lost if the code is regenerated.
 */
import * as coreClient from "@azure/core-client";
import * as Mappers from "../models/mappers";
import * as Parameters from "../models/parameters";
/** Class containing IotSecuritySolutionAnalytics operations. */
export class IotSecuritySolutionAnalyticsImpl {
    /**
     * Initialize a new instance of the class IotSecuritySolutionAnalytics class.
     * @param client Reference to the service client
     */
    constructor(client) {
        this.client = client;
    }
    /**
     * Use this method to get IoT security Analytics metrics in an array.
     * @param resourceGroupName The name of the resource group within the user's subscription. The name is
     *                          case insensitive.
     * @param solutionName The name of the IoT Security solution.
     * @param options The options parameters.
     */
    list(resourceGroupName, solutionName, options) {
        return this.client.sendOperationRequest({ resourceGroupName, solutionName, options }, listOperationSpec);
    }
    /**
     * Use this method to get IoT Security Analytics metrics.
     * @param resourceGroupName The name of the resource group within the user's subscription. The name is
     *                          case insensitive.
     * @param solutionName The name of the IoT Security solution.
     * @param options The options parameters.
     */
    get(resourceGroupName, solutionName, options) {
        return this.client.sendOperationRequest({ resourceGroupName, solutionName, options }, getOperationSpec);
    }
}
// Operation Specifications
const serializer = coreClient.createSerializer(Mappers, /* isXml */ false);
const listOperationSpec = {
    path: "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Security/iotSecuritySolutions/{solutionName}/analyticsModels",
    httpMethod: "GET",
    responses: {
        200: {
            bodyMapper: Mappers.IoTSecuritySolutionAnalyticsModelList
        },
        default: {
            bodyMapper: Mappers.CloudError
        }
    },
    queryParameters: [Parameters.apiVersion5],
    urlParameters: [
        Parameters.$host,
        Parameters.subscriptionId,
        Parameters.resourceGroupName,
        Parameters.solutionName
    ],
    headerParameters: [Parameters.accept],
    serializer
};
const getOperationSpec = {
    path: "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Security/iotSecuritySolutions/{solutionName}/analyticsModels/default",
    httpMethod: "GET",
    responses: {
        200: {
            bodyMapper: Mappers.IoTSecuritySolutionAnalyticsModel
        },
        default: {
            bodyMapper: Mappers.CloudError
        }
    },
    queryParameters: [Parameters.apiVersion5],
    urlParameters: [
        Parameters.$host,
        Parameters.subscriptionId,
        Parameters.resourceGroupName,
        Parameters.solutionName
    ],
    headerParameters: [Parameters.accept],
    serializer
};
