// @azure/arm-security/dist-esm/src/operations/regulatoryComplianceControls.js
/*
 * Copyright (c) Microsoft Corporation.
 * Licensed under the MIT License.
 *
 * Code generated by Microsoft (R) AutoRest Code Generator.
 * Changes may cause incorrect behavior and will be lost if the code is regenerated.
 */
import { __asyncDelegator, __asyncGenerator, __asyncValues, __await } from "tslib";
import * as coreClient from "@azure/core-client";
import * as Mappers from "../models/mappers";
import * as Parameters from "../models/parameters";
/// <reference lib="esnext.asynciterable" />
/** Class containing RegulatoryComplianceControls operations. */
export class RegulatoryComplianceControlsImpl {
    /**
     * Initialize a new instance of the class RegulatoryComplianceControls class.
     * @param client Reference to the service client
     */
    constructor(client) {
        this.client = client;
    }
    /**
     * All supported regulatory compliance controls details and state for selected standard
     * @param regulatoryComplianceStandardName Name of the regulatory compliance standard object
     * @param options The options parameters.
     */
    list(regulatoryComplianceStandardName, options) {
        const iter = this.listPagingAll(regulatoryComplianceStandardName, options);
        return {
            next() {
                return iter.next();
            },
            [Symbol.asyncIterator]() {
                return this;
            },
            byPage: () => {
                return this.listPagingPage(regulatoryComplianceStandardName, options);
            }
        };
    }
    listPagingPage(regulatoryComplianceStandardName, options) {
        return __asyncGenerator(this, arguments, function* listPagingPage_1() {
            let result = yield __await(this._list(regulatoryComplianceStandardName, options));
            yield yield __await(result.value || []);
            let continuationToken = result.nextLink;
            while (continuationToken) {
                result = yield __await(this._listNext(regulatoryComplianceStandardName, continuationToken, options));
                continuationToken = result.nextLink;
                yield yield __await(result.value || []);
            }
        });
    }
    listPagingAll(regulatoryComplianceStandardName, options) {
        return __asyncGenerator(this, arguments, function* listPagingAll_1() {
            var e_1, _a;
            try {
                for (var _b = __asyncValues(this.listPagingPage(regulatoryComplianceStandardName, options)), _c; _c = yield __await(_b.next()), !_c.done;) {
                    const page = _c.value;
                    yield __await(yield* __asyncDelegator(__asyncValues(page)));
                }
            }
            catch (e_1_1) { e_1 = { error: e_1_1 }; }
            finally {
                try {
                    if (_c && !_c.done && (_a = _b.return)) yield __await(_a.call(_b));
                }
                finally { if (e_1) throw e_1.error; }
            }
        });
    }
    /**
     * All supported regulatory compliance controls details and state for selected standard
     * @param regulatoryComplianceStandardName Name of the regulatory compliance standard object
     * @param options The options parameters.
     */
    _list(regulatoryComplianceStandardName, options) {
        return this.client.sendOperationRequest({ regulatoryComplianceStandardName, options }, listOperationSpec);
    }
    /**
     * Selected regulatory compliance control details and state
     * @param regulatoryComplianceStandardName Name of the regulatory compliance standard object
     * @param regulatoryComplianceControlName Name of the regulatory compliance control object
     * @param options The options parameters.
     */
    get(regulatoryComplianceStandardName, regulatoryComplianceControlName, options) {
        return this.client.sendOperationRequest({
            regulatoryComplianceStandardName,
            regulatoryComplianceControlName,
            options
        }, getOperationSpec);
    }
    /**
     * ListNext
     * @param regulatoryComplianceStandardName Name of the regulatory compliance standard object
     * @param nextLink The nextLink from the previous successful call to the List method.
     * @param options The options parameters.
     */
    _listNext(regulatoryComplianceStandardName, nextLink, options) {
        return this.client.sendOperationRequest({ regulatoryComplianceStandardName, nextLink, options }, listNextOperationSpec);
    }
}
// Operation Specifications
const serializer = coreClient.createSerializer(Mappers, /* isXml */ false);
const listOperationSpec = {
    path: "/subscriptions/{subscriptionId}/providers/Microsoft.Security/regulatoryComplianceStandards/{regulatoryComplianceStandardName}/regulatoryComplianceControls",
    httpMethod: "GET",
    responses: {
        200: {
            bodyMapper: Mappers.RegulatoryComplianceControlList
        },
        default: {
            bodyMapper: Mappers.CloudError
        }
    },
    queryParameters: [Parameters.filter, Parameters.apiVersion8],
    urlParameters: [
        Parameters.$host,
        Parameters.subscriptionId,
        Parameters.regulatoryComplianceStandardName
    ],
    headerParameters: [Parameters.accept],
    serializer
};
const getOperationSpec = {
    path: "/subscriptions/{subscriptionId}/providers/Microsoft.Security/regulatoryComplianceStandards/{regulatoryComplianceStandardName}/regulatoryComplianceControls/{regulatoryComplianceControlName}",
    httpMethod: "GET",
    responses: {
        200: {
            bodyMapper: Mappers.RegulatoryComplianceControl
        },
        default: {
            bodyMapper: Mappers.CloudError
        }
    },
    queryParameters: [Parameters.apiVersion8],
    urlParameters: [
        Parameters.$host,
        Parameters.subscriptionId,
        Parameters.regulatoryComplianceStandardName,
        Parameters.regulatoryComplianceControlName
    ],
    headerParameters: [Parameters.accept],
    serializer
};
const listNextOperationSpec = {
    path: "{nextLink}",
    httpMethod: "GET",
    responses: {
        200: {
            bodyMapper: Mappers.RegulatoryComplianceControlList
        },
        default: {
            bodyMapper: Mappers.CloudError
        }
    },
    queryParameters: [Parameters.filter, Parameters.apiVersion8],
    urlParameters: [
        Parameters.$host,
        Parameters.subscriptionId,
        Parameters.nextLink,
        Parameters.regulatoryComplianceStandardName
    ],
    headerParameters: [Parameters.accept],
    serializer
};
