// @azure/arm-security/dist-esm/src/operations/secureScoreControls.js
/*
 * Copyright (c) Microsoft Corporation.
 * Licensed under the MIT License.
 *
 * Code generated by Microsoft (R) AutoRest Code Generator.
 * Changes may cause incorrect behavior and will be lost if the code is regenerated.
 */
import { __asyncDelegator, __asyncGenerator, __asyncValues, __await } from "tslib";
import * as coreClient from "@azure/core-client";
import * as Mappers from "../models/mappers";
import * as Parameters from "../models/parameters";
/// <reference lib="esnext.asynciterable" />
/** Class containing SecureScoreControls operations. */
export class SecureScoreControlsImpl {
    /**
     * Initialize a new instance of the class SecureScoreControls class.
     * @param client Reference to the service client
     */
    constructor(client) {
        this.client = client;
    }
    /**
     * Get all security controls for a specific initiative within a scope
     * @param secureScoreName The initiative name. For the ASC Default initiative, use 'ascScore' as in the
     *                        sample request below.
     * @param options The options parameters.
     */
    listBySecureScore(secureScoreName, options) {
        const iter = this.listBySecureScorePagingAll(secureScoreName, options);
        return {
            next() {
                return iter.next();
            },
            [Symbol.asyncIterator]() {
                return this;
            },
            byPage: () => {
                return this.listBySecureScorePagingPage(secureScoreName, options);
            }
        };
    }
    listBySecureScorePagingPage(secureScoreName, options) {
        return __asyncGenerator(this, arguments, function* listBySecureScorePagingPage_1() {
            let result = yield __await(this._listBySecureScore(secureScoreName, options));
            yield yield __await(result.value || []);
            let continuationToken = result.nextLink;
            while (continuationToken) {
                result = yield __await(this._listBySecureScoreNext(secureScoreName, continuationToken, options));
                continuationToken = result.nextLink;
                yield yield __await(result.value || []);
            }
        });
    }
    listBySecureScorePagingAll(secureScoreName, options) {
        return __asyncGenerator(this, arguments, function* listBySecureScorePagingAll_1() {
            var e_1, _a;
            try {
                for (var _b = __asyncValues(this.listBySecureScorePagingPage(secureScoreName, options)), _c; _c = yield __await(_b.next()), !_c.done;) {
                    const page = _c.value;
                    yield __await(yield* __asyncDelegator(__asyncValues(page)));
                }
            }
            catch (e_1_1) { e_1 = { error: e_1_1 }; }
            finally {
                try {
                    if (_c && !_c.done && (_a = _b.return)) yield __await(_a.call(_b));
                }
                finally { if (e_1) throw e_1.error; }
            }
        });
    }
    /**
     * Get all security controls within a scope
     * @param options The options parameters.
     */
    list(options) {
        const iter = this.listPagingAll(options);
        return {
            next() {
                return iter.next();
            },
            [Symbol.asyncIterator]() {
                return this;
            },
            byPage: () => {
                return this.listPagingPage(options);
            }
        };
    }
    listPagingPage(options) {
        return __asyncGenerator(this, arguments, function* listPagingPage_1() {
            let result = yield __await(this._list(options));
            yield yield __await(result.value || []);
            let continuationToken = result.nextLink;
            while (continuationToken) {
                result = yield __await(this._listNext(continuationToken, options));
                continuationToken = result.nextLink;
                yield yield __await(result.value || []);
            }
        });
    }
    listPagingAll(options) {
        return __asyncGenerator(this, arguments, function* listPagingAll_1() {
            var e_2, _a;
            try {
                for (var _b = __asyncValues(this.listPagingPage(options)), _c; _c = yield __await(_b.next()), !_c.done;) {
                    const page = _c.value;
                    yield __await(yield* __asyncDelegator(__asyncValues(page)));
                }
            }
            catch (e_2_1) { e_2 = { error: e_2_1 }; }
            finally {
                try {
                    if (_c && !_c.done && (_a = _b.return)) yield __await(_a.call(_b));
                }
                finally { if (e_2) throw e_2.error; }
            }
        });
    }
    /**
     * Get all security controls for a specific initiative within a scope
     * @param secureScoreName The initiative name. For the ASC Default initiative, use 'ascScore' as in the
     *                        sample request below.
     * @param options The options parameters.
     */
    _listBySecureScore(secureScoreName, options) {
        return this.client.sendOperationRequest({ secureScoreName, options }, listBySecureScoreOperationSpec);
    }
    /**
     * Get all security controls within a scope
     * @param options The options parameters.
     */
    _list(options) {
        return this.client.sendOperationRequest({ options }, listOperationSpec);
    }
    /**
     * ListBySecureScoreNext
     * @param secureScoreName The initiative name. For the ASC Default initiative, use 'ascScore' as in the
     *                        sample request below.
     * @param nextLink The nextLink from the previous successful call to the ListBySecureScore method.
     * @param options The options parameters.
     */
    _listBySecureScoreNext(secureScoreName, nextLink, options) {
        return this.client.sendOperationRequest({ secureScoreName, nextLink, options }, listBySecureScoreNextOperationSpec);
    }
    /**
     * ListNext
     * @param nextLink The nextLink from the previous successful call to the List method.
     * @param options The options parameters.
     */
    _listNext(nextLink, options) {
        return this.client.sendOperationRequest({ nextLink, options }, listNextOperationSpec);
    }
}
// Operation Specifications
const serializer = coreClient.createSerializer(Mappers, /* isXml */ false);
const listBySecureScoreOperationSpec = {
    path: "/subscriptions/{subscriptionId}/providers/Microsoft.Security/secureScores/{secureScoreName}/secureScoreControls",
    httpMethod: "GET",
    responses: {
        200: {
            bodyMapper: Mappers.SecureScoreControlList
        },
        default: {
            bodyMapper: Mappers.CloudError
        }
    },
    queryParameters: [Parameters.apiVersion9, Parameters.expand1],
    urlParameters: [
        Parameters.$host,
        Parameters.subscriptionId,
        Parameters.secureScoreName
    ],
    headerParameters: [Parameters.accept],
    serializer
};
const listOperationSpec = {
    path: "/subscriptions/{subscriptionId}/providers/Microsoft.Security/secureScoreControls",
    httpMethod: "GET",
    responses: {
        200: {
            bodyMapper: Mappers.SecureScoreControlList
        },
        default: {
            bodyMapper: Mappers.CloudError
        }
    },
    queryParameters: [Parameters.apiVersion9, Parameters.expand1],
    urlParameters: [Parameters.$host, Parameters.subscriptionId],
    headerParameters: [Parameters.accept],
    serializer
};
const listBySecureScoreNextOperationSpec = {
    path: "{nextLink}",
    httpMethod: "GET",
    responses: {
        200: {
            bodyMapper: Mappers.SecureScoreControlList
        },
        default: {
            bodyMapper: Mappers.CloudError
        }
    },
    queryParameters: [Parameters.apiVersion9, Parameters.expand1],
    urlParameters: [
        Parameters.$host,
        Parameters.subscriptionId,
        Parameters.nextLink,
        Parameters.secureScoreName
    ],
    headerParameters: [Parameters.accept],
    serializer
};
const listNextOperationSpec = {
    path: "{nextLink}",
    httpMethod: "GET",
    responses: {
        200: {
            bodyMapper: Mappers.SecureScoreControlList
        },
        default: {
            bodyMapper: Mappers.CloudError
        }
    },
    queryParameters: [Parameters.apiVersion9, Parameters.expand1],
    urlParameters: [
        Parameters.$host,
        Parameters.subscriptionId,
        Parameters.nextLink
    ],
    headerParameters: [Parameters.accept],
    serializer
};
