// @azure/arm-desktopvirtualization/dist-esm/src/models/parameters.js
/*
 * Copyright (c) Microsoft Corporation.
 * Licensed under the MIT License.
 *
 * Code generated by Microsoft (R) AutoRest Code Generator.
 * Changes may cause incorrect behavior and will be lost if the code is regenerated.
 */
import { Workspace as WorkspaceMapper, WorkspacePatch as WorkspacePatchMapper, PrivateEndpointConnection as PrivateEndpointConnectionMapper, ScalingPlan as ScalingPlanMapper, ScalingPlanPatch as ScalingPlanPatchMapper, ScalingPlanPooledSchedule as ScalingPlanPooledScheduleMapper, ScalingPlanPooledSchedulePatch as ScalingPlanPooledSchedulePatchMapper, ScalingPlanPersonalSchedule as ScalingPlanPersonalScheduleMapper, ScalingPlanPersonalSchedulePatch as ScalingPlanPersonalSchedulePatchMapper, ApplicationGroup as ApplicationGroupMapper, ApplicationGroupPatch as ApplicationGroupPatchMapper, Application as ApplicationMapper, ApplicationPatch as ApplicationPatchMapper, DesktopPatch as DesktopPatchMapper, HostPool as HostPoolMapper, HostPoolPatch as HostPoolPatchMapper, SendMessage as SendMessageMapper, SessionHostPatch as SessionHostPatchMapper, MsixPackage as MsixPackageMapper, MsixPackagePatch as MsixPackagePatchMapper, MsixImageURI as MsixImageURIMapper } from "../models/mappers";
export const accept = {
    parameterPath: "accept",
    mapper: {
        defaultValue: "application/json",
        isConstant: true,
        serializedName: "Accept",
        type: {
            name: "String"
        }
    }
};
export const $host = {
    parameterPath: "$host",
    mapper: {
        serializedName: "$host",
        required: true,
        type: {
            name: "String"
        }
    },
    skipEncoding: true
};
export const apiVersion = {
    parameterPath: "apiVersion",
    mapper: {
        defaultValue: "2023-09-05",
        isConstant: true,
        serializedName: "api-version",
        type: {
            name: "String"
        }
    }
};
export const nextLink = {
    parameterPath: "nextLink",
    mapper: {
        serializedName: "nextLink",
        required: true,
        type: {
            name: "String"
        }
    },
    skipEncoding: true
};
export const subscriptionId = {
    parameterPath: "subscriptionId",
    mapper: {
        constraints: {
            MinLength: 1
        },
        serializedName: "subscriptionId",
        required: true,
        type: {
            name: "String"
        }
    }
};
export const resourceGroupName = {
    parameterPath: "resourceGroupName",
    mapper: {
        constraints: {
            MaxLength: 90,
            MinLength: 1
        },
        serializedName: "resourceGroupName",
        required: true,
        type: {
            name: "String"
        }
    }
};
export const workspaceName = {
    parameterPath: "workspaceName",
    mapper: {
        constraints: {
            MaxLength: 64,
            MinLength: 3
        },
        serializedName: "workspaceName",
        required: true,
        type: {
            name: "String"
        }
    }
};
export const contentType = {
    parameterPath: ["options", "contentType"],
    mapper: {
        defaultValue: "application/json",
        isConstant: true,
        serializedName: "Content-Type",
        type: {
            name: "String"
        }
    }
};
export const workspace = {
    parameterPath: "workspace",
    mapper: WorkspaceMapper
};
export const workspace1 = {
    parameterPath: ["options", "workspace"],
    mapper: WorkspacePatchMapper
};
export const pageSize = {
    parameterPath: ["options", "pageSize"],
    mapper: {
        serializedName: "pageSize",
        type: {
            name: "Number"
        }
    }
};
export const isDescending = {
    parameterPath: ["options", "isDescending"],
    mapper: {
        serializedName: "isDescending",
        type: {
            name: "Boolean"
        }
    }
};
export const initialSkip = {
    parameterPath: ["options", "initialSkip"],
    mapper: {
        serializedName: "initialSkip",
        type: {
            name: "Number"
        }
    }
};
export const privateEndpointConnectionName = {
    parameterPath: "privateEndpointConnectionName",
    mapper: {
        serializedName: "privateEndpointConnectionName",
        required: true,
        type: {
            name: "String"
        }
    }
};
export const connection = {
    parameterPath: "connection",
    mapper: PrivateEndpointConnectionMapper
};
export const hostPoolName = {
    parameterPath: "hostPoolName",
    mapper: {
        constraints: {
            MaxLength: 64,
            MinLength: 3
        },
        serializedName: "hostPoolName",
        required: true,
        type: {
            name: "String"
        }
    }
};
export const scalingPlanName = {
    parameterPath: "scalingPlanName",
    mapper: {
        constraints: {
            MaxLength: 64,
            MinLength: 3
        },
        serializedName: "scalingPlanName",
        required: true,
        type: {
            name: "String"
        }
    }
};
export const scalingPlan = {
    parameterPath: "scalingPlan",
    mapper: ScalingPlanMapper
};
export const scalingPlan1 = {
    parameterPath: ["options", "scalingPlan"],
    mapper: ScalingPlanPatchMapper
};
export const scalingPlanScheduleName = {
    parameterPath: "scalingPlanScheduleName",
    mapper: {
        constraints: {
            MaxLength: 64,
            MinLength: 1
        },
        serializedName: "scalingPlanScheduleName",
        required: true,
        type: {
            name: "String"
        }
    }
};
export const scalingPlanSchedule = {
    parameterPath: "scalingPlanSchedule",
    mapper: ScalingPlanPooledScheduleMapper
};
export const scalingPlanSchedule1 = {
    parameterPath: ["options", "scalingPlanSchedule"],
    mapper: ScalingPlanPooledSchedulePatchMapper
};
export const scalingPlanSchedule2 = {
    parameterPath: "scalingPlanSchedule",
    mapper: ScalingPlanPersonalScheduleMapper
};
export const scalingPlanSchedule3 = {
    parameterPath: ["options", "scalingPlanSchedule"],
    mapper: ScalingPlanPersonalSchedulePatchMapper
};
export const applicationGroupName = {
    parameterPath: "applicationGroupName",
    mapper: {
        constraints: {
            MaxLength: 64,
            MinLength: 3
        },
        serializedName: "applicationGroupName",
        required: true,
        type: {
            name: "String"
        }
    }
};
export const applicationGroup = {
    parameterPath: "applicationGroup",
    mapper: ApplicationGroupMapper
};
export const applicationGroup1 = {
    parameterPath: ["options", "applicationGroup"],
    mapper: ApplicationGroupPatchMapper
};
export const filter = {
    parameterPath: ["options", "filter"],
    mapper: {
        serializedName: "$filter",
        type: {
            name: "String"
        }
    }
};
export const applicationName = {
    parameterPath: "applicationName",
    mapper: {
        constraints: {
            MaxLength: 24,
            MinLength: 3
        },
        serializedName: "applicationName",
        required: true,
        type: {
            name: "String"
        }
    }
};
export const application = {
    parameterPath: "application",
    mapper: ApplicationMapper
};
export const application1 = {
    parameterPath: ["options", "application"],
    mapper: ApplicationPatchMapper
};
export const desktopName = {
    parameterPath: "desktopName",
    mapper: {
        constraints: {
            MaxLength: 24,
            MinLength: 3
        },
        serializedName: "desktopName",
        required: true,
        type: {
            name: "String"
        }
    }
};
export const desktop = {
    parameterPath: ["options", "desktop"],
    mapper: DesktopPatchMapper
};
export const hostPool = {
    parameterPath: "hostPool",
    mapper: HostPoolMapper
};
export const force = {
    parameterPath: ["options", "force"],
    mapper: {
        serializedName: "force",
        type: {
            name: "Boolean"
        }
    }
};
export const hostPool1 = {
    parameterPath: ["options", "hostPool"],
    mapper: HostPoolPatchMapper
};
export const sessionHostName = {
    parameterPath: "sessionHostName",
    mapper: {
        constraints: {
            MaxLength: 48,
            MinLength: 3
        },
        serializedName: "sessionHostName",
        required: true,
        type: {
            name: "String"
        }
    }
};
export const userSessionId = {
    parameterPath: "userSessionId",
    mapper: {
        constraints: {
            MaxLength: 24,
            MinLength: 1
        },
        serializedName: "userSessionId",
        required: true,
        type: {
            name: "String"
        }
    }
};
export const sendMessage = {
    parameterPath: ["options", "sendMessage"],
    mapper: SendMessageMapper
};
export const sessionHost = {
    parameterPath: ["options", "sessionHost"],
    mapper: SessionHostPatchMapper
};
export const msixPackageFullName = {
    parameterPath: "msixPackageFullName",
    mapper: {
        constraints: {
            MaxLength: 100,
            MinLength: 3
        },
        serializedName: "msixPackageFullName",
        required: true,
        type: {
            name: "String"
        }
    }
};
export const msixPackage = {
    parameterPath: "msixPackage",
    mapper: MsixPackageMapper
};
export const msixPackage1 = {
    parameterPath: ["options", "msixPackage"],
    mapper: MsixPackagePatchMapper
};
export const msixImageURI = {
    parameterPath: "msixImageURI",
    mapper: MsixImageURIMapper
};
