// @azure/arm-mediaservices/dist-esm/src/azureMediaServices.js
/*
 * Copyright (c) Microsoft Corporation.
 * Licensed under the MIT License.
 *
 * Code generated by Microsoft (R) AutoRest Code Generator.
 * Changes may cause incorrect behavior and will be lost if the code is regenerated.
 */
import * as coreClient from "@azure/core-client";
import * as coreRestPipeline from "@azure/core-rest-pipeline";
import { AccountFiltersImpl, OperationsImpl, MediaservicesImpl, PrivateLinkResourcesImpl, PrivateEndpointConnectionsImpl, LocationsImpl, MediaServicesOperationStatusesImpl, MediaServicesOperationResultsImpl, AssetsImpl, AssetFiltersImpl, TracksImpl, OperationStatusesImpl, OperationResultsImpl, ContentKeyPoliciesImpl, TransformsImpl, JobsImpl, StreamingPoliciesImpl, StreamingLocatorsImpl, LiveEventsImpl, LiveOutputsImpl, StreamingEndpointsImpl } from "./operations";
export class AzureMediaServices extends coreClient.ServiceClient {
    /**
     * Initializes a new instance of the AzureMediaServices class.
     * @param credentials Subscription credentials which uniquely identify client subscription.
     * @param subscriptionId The unique identifier for a Microsoft Azure subscription.
     * @param options The parameter options
     */
    constructor(credentials, subscriptionId, options) {
        var _a, _b, _c;
        if (credentials === undefined) {
            throw new Error("'credentials' cannot be null");
        }
        if (subscriptionId === undefined) {
            throw new Error("'subscriptionId' cannot be null");
        }
        // Initializing default values for options
        if (!options) {
            options = {};
        }
        const defaults = {
            requestContentType: "application/json; charset=utf-8",
            credential: credentials
        };
        const packageDetails = `azsdk-js-arm-mediaservices/13.1.0`;
        const userAgentPrefix = options.userAgentOptions && options.userAgentOptions.userAgentPrefix
            ? `${options.userAgentOptions.userAgentPrefix} ${packageDetails}`
            : `${packageDetails}`;
        const optionsWithDefaults = Object.assign(Object.assign(Object.assign({}, defaults), options), { userAgentOptions: {
                userAgentPrefix
            }, endpoint: (_b = (_a = options.endpoint) !== null && _a !== void 0 ? _a : options.baseUri) !== null && _b !== void 0 ? _b : "https://management.azure.com" });
        super(optionsWithDefaults);
        let bearerTokenAuthenticationPolicyFound = false;
        if ((options === null || options === void 0 ? void 0 : options.pipeline) && options.pipeline.getOrderedPolicies().length > 0) {
            const pipelinePolicies = options.pipeline.getOrderedPolicies();
            bearerTokenAuthenticationPolicyFound = pipelinePolicies.some((pipelinePolicy) => pipelinePolicy.name ===
                coreRestPipeline.bearerTokenAuthenticationPolicyName);
        }
        if (!options ||
            !options.pipeline ||
            options.pipeline.getOrderedPolicies().length == 0 ||
            !bearerTokenAuthenticationPolicyFound) {
            this.pipeline.removePolicy({
                name: coreRestPipeline.bearerTokenAuthenticationPolicyName
            });
            this.pipeline.addPolicy(coreRestPipeline.bearerTokenAuthenticationPolicy({
                credential: credentials,
                scopes: (_c = optionsWithDefaults.credentialScopes) !== null && _c !== void 0 ? _c : `${optionsWithDefaults.endpoint}/.default`,
                challengeCallbacks: {
                    authorizeRequestOnChallenge: coreClient.authorizeRequestOnClaimChallenge
                }
            }));
        }
        // Parameter assignments
        this.subscriptionId = subscriptionId;
        // Assigning values to Constant parameters
        this.$host = options.$host || "https://management.azure.com";
        this.accountFilters = new AccountFiltersImpl(this);
        this.operations = new OperationsImpl(this);
        this.mediaservices = new MediaservicesImpl(this);
        this.privateLinkResources = new PrivateLinkResourcesImpl(this);
        this.privateEndpointConnections = new PrivateEndpointConnectionsImpl(this);
        this.locations = new LocationsImpl(this);
        this.mediaServicesOperationStatuses = new MediaServicesOperationStatusesImpl(this);
        this.mediaServicesOperationResults = new MediaServicesOperationResultsImpl(this);
        this.assets = new AssetsImpl(this);
        this.assetFilters = new AssetFiltersImpl(this);
        this.tracks = new TracksImpl(this);
        this.operationStatuses = new OperationStatusesImpl(this);
        this.operationResults = new OperationResultsImpl(this);
        this.contentKeyPolicies = new ContentKeyPoliciesImpl(this);
        this.transforms = new TransformsImpl(this);
        this.jobs = new JobsImpl(this);
        this.streamingPolicies = new StreamingPoliciesImpl(this);
        this.streamingLocators = new StreamingLocatorsImpl(this);
        this.liveEvents = new LiveEventsImpl(this);
        this.liveOutputs = new LiveOutputsImpl(this);
        this.streamingEndpoints = new StreamingEndpointsImpl(this);
    }
}
