// @azure/arm-synapse/dist-esm/src/operations/extendedSqlPoolBlobAuditingPolicies.js
/*
 * Copyright (c) Microsoft Corporation.
 * Licensed under the MIT License.
 *
 * Code generated by Microsoft (R) AutoRest Code Generator.
 * Changes may cause incorrect behavior and will be lost if the code is regenerated.
 */
import { __asyncDelegator, __asyncGenerator, __asyncValues, __await } from "tslib";
import * as coreClient from "@azure/core-client";
import * as Mappers from "../models/mappers";
import * as Parameters from "../models/parameters";
/// <reference lib="esnext.asynciterable" />
/** Class containing ExtendedSqlPoolBlobAuditingPolicies operations. */
export class ExtendedSqlPoolBlobAuditingPoliciesImpl {
    /**
     * Initialize a new instance of the class ExtendedSqlPoolBlobAuditingPolicies class.
     * @param client Reference to the service client
     */
    constructor(client) {
        this.client = client;
    }
    /**
     * Lists extended auditing settings of a Sql pool.
     * @param resourceGroupName The name of the resource group. The name is case insensitive.
     * @param workspaceName The name of the workspace.
     * @param sqlPoolName SQL pool name
     * @param options The options parameters.
     */
    listBySqlPool(resourceGroupName, workspaceName, sqlPoolName, options) {
        const iter = this.listBySqlPoolPagingAll(resourceGroupName, workspaceName, sqlPoolName, options);
        return {
            next() {
                return iter.next();
            },
            [Symbol.asyncIterator]() {
                return this;
            },
            byPage: () => {
                return this.listBySqlPoolPagingPage(resourceGroupName, workspaceName, sqlPoolName, options);
            }
        };
    }
    listBySqlPoolPagingPage(resourceGroupName, workspaceName, sqlPoolName, options) {
        return __asyncGenerator(this, arguments, function* listBySqlPoolPagingPage_1() {
            let result = yield __await(this._listBySqlPool(resourceGroupName, workspaceName, sqlPoolName, options));
            yield yield __await(result.value || []);
            let continuationToken = result.nextLink;
            while (continuationToken) {
                result = yield __await(this._listBySqlPoolNext(resourceGroupName, workspaceName, sqlPoolName, continuationToken, options));
                continuationToken = result.nextLink;
                yield yield __await(result.value || []);
            }
        });
    }
    listBySqlPoolPagingAll(resourceGroupName, workspaceName, sqlPoolName, options) {
        return __asyncGenerator(this, arguments, function* listBySqlPoolPagingAll_1() {
            var e_1, _a;
            try {
                for (var _b = __asyncValues(this.listBySqlPoolPagingPage(resourceGroupName, workspaceName, sqlPoolName, options)), _c; _c = yield __await(_b.next()), !_c.done;) {
                    const page = _c.value;
                    yield __await(yield* __asyncDelegator(__asyncValues(page)));
                }
            }
            catch (e_1_1) { e_1 = { error: e_1_1 }; }
            finally {
                try {
                    if (_c && !_c.done && (_a = _b.return)) yield __await(_a.call(_b));
                }
                finally { if (e_1) throw e_1.error; }
            }
        });
    }
    /**
     * Gets an extended Sql pool's blob auditing policy.
     * @param resourceGroupName The name of the resource group. The name is case insensitive.
     * @param workspaceName The name of the workspace.
     * @param sqlPoolName SQL pool name
     * @param options The options parameters.
     */
    get(resourceGroupName, workspaceName, sqlPoolName, options) {
        return this.client.sendOperationRequest({ resourceGroupName, workspaceName, sqlPoolName, options }, getOperationSpec);
    }
    /**
     * Creates or updates an extended Sql pool's blob auditing policy.
     * @param resourceGroupName The name of the resource group. The name is case insensitive.
     * @param workspaceName The name of the workspace.
     * @param sqlPoolName SQL pool name
     * @param parameters The extended Sql pool blob auditing policy.
     * @param options The options parameters.
     */
    createOrUpdate(resourceGroupName, workspaceName, sqlPoolName, parameters, options) {
        return this.client.sendOperationRequest({ resourceGroupName, workspaceName, sqlPoolName, parameters, options }, createOrUpdateOperationSpec);
    }
    /**
     * Lists extended auditing settings of a Sql pool.
     * @param resourceGroupName The name of the resource group. The name is case insensitive.
     * @param workspaceName The name of the workspace.
     * @param sqlPoolName SQL pool name
     * @param options The options parameters.
     */
    _listBySqlPool(resourceGroupName, workspaceName, sqlPoolName, options) {
        return this.client.sendOperationRequest({ resourceGroupName, workspaceName, sqlPoolName, options }, listBySqlPoolOperationSpec);
    }
    /**
     * ListBySqlPoolNext
     * @param resourceGroupName The name of the resource group. The name is case insensitive.
     * @param workspaceName The name of the workspace.
     * @param sqlPoolName SQL pool name
     * @param nextLink The nextLink from the previous successful call to the ListBySqlPool method.
     * @param options The options parameters.
     */
    _listBySqlPoolNext(resourceGroupName, workspaceName, sqlPoolName, nextLink, options) {
        return this.client.sendOperationRequest({ resourceGroupName, workspaceName, sqlPoolName, nextLink, options }, listBySqlPoolNextOperationSpec);
    }
}
// Operation Specifications
const serializer = coreClient.createSerializer(Mappers, /* isXml */ false);
const getOperationSpec = {
    path: "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Synapse/workspaces/{workspaceName}/sqlPools/{sqlPoolName}/extendedAuditingSettings/{blobAuditingPolicyName}",
    httpMethod: "GET",
    responses: {
        200: {
            bodyMapper: Mappers.ExtendedSqlPoolBlobAuditingPolicy
        },
        default: {}
    },
    queryParameters: [Parameters.apiVersion],
    urlParameters: [
        Parameters.$host,
        Parameters.subscriptionId,
        Parameters.resourceGroupName,
        Parameters.workspaceName,
        Parameters.sqlPoolName,
        Parameters.blobAuditingPolicyName
    ],
    headerParameters: [Parameters.accept],
    serializer
};
const createOrUpdateOperationSpec = {
    path: "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Synapse/workspaces/{workspaceName}/sqlPools/{sqlPoolName}/extendedAuditingSettings/{blobAuditingPolicyName}",
    httpMethod: "PUT",
    responses: {
        200: {
            bodyMapper: Mappers.ExtendedSqlPoolBlobAuditingPolicy
        },
        201: {
            bodyMapper: Mappers.ExtendedSqlPoolBlobAuditingPolicy
        },
        default: {}
    },
    requestBody: Parameters.parameters12,
    queryParameters: [Parameters.apiVersion],
    urlParameters: [
        Parameters.$host,
        Parameters.subscriptionId,
        Parameters.resourceGroupName,
        Parameters.workspaceName,
        Parameters.sqlPoolName,
        Parameters.blobAuditingPolicyName
    ],
    headerParameters: [Parameters.accept, Parameters.contentType],
    mediaType: "json",
    serializer
};
const listBySqlPoolOperationSpec = {
    path: "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Synapse/workspaces/{workspaceName}/sqlPools/{sqlPoolName}/extendedAuditingSettings",
    httpMethod: "GET",
    responses: {
        200: {
            bodyMapper: Mappers.ExtendedSqlPoolBlobAuditingPolicyListResult
        },
        default: {}
    },
    queryParameters: [Parameters.apiVersion],
    urlParameters: [
        Parameters.$host,
        Parameters.subscriptionId,
        Parameters.resourceGroupName,
        Parameters.workspaceName,
        Parameters.sqlPoolName
    ],
    headerParameters: [Parameters.accept],
    serializer
};
const listBySqlPoolNextOperationSpec = {
    path: "{nextLink}",
    httpMethod: "GET",
    responses: {
        200: {
            bodyMapper: Mappers.ExtendedSqlPoolBlobAuditingPolicyListResult
        },
        default: {}
    },
    queryParameters: [Parameters.apiVersion],
    urlParameters: [
        Parameters.$host,
        Parameters.subscriptionId,
        Parameters.resourceGroupName,
        Parameters.workspaceName,
        Parameters.nextLink,
        Parameters.sqlPoolName
    ],
    headerParameters: [Parameters.accept],
    serializer
};
