// @azure/arm-synapse/dist-esm/src/operations/integrationRuntimeMonitoringDataOperations.js
/*
 * Copyright (c) Microsoft Corporation.
 * Licensed under the MIT License.
 *
 * Code generated by Microsoft (R) AutoRest Code Generator.
 * Changes may cause incorrect behavior and will be lost if the code is regenerated.
 */
import * as coreClient from "@azure/core-client";
import * as Mappers from "../models/mappers";
import * as Parameters from "../models/parameters";
/** Class containing IntegrationRuntimeMonitoringDataOperations operations. */
export class IntegrationRuntimeMonitoringDataOperationsImpl {
    /**
     * Initialize a new instance of the class IntegrationRuntimeMonitoringDataOperations class.
     * @param client Reference to the service client
     */
    constructor(client) {
        this.client = client;
    }
    /**
     * Get monitoring data for an integration runtime
     * @param resourceGroupName The name of the resource group. The name is case insensitive.
     * @param workspaceName The name of the workspace.
     * @param integrationRuntimeName Integration runtime name
     * @param options The options parameters.
     */
    list(resourceGroupName, workspaceName, integrationRuntimeName, options) {
        return this.client.sendOperationRequest({ resourceGroupName, workspaceName, integrationRuntimeName, options }, listOperationSpec);
    }
}
// Operation Specifications
const serializer = coreClient.createSerializer(Mappers, /* isXml */ false);
const listOperationSpec = {
    path: "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Synapse/workspaces/{workspaceName}/integrationRuntimes/{integrationRuntimeName}/monitoringData",
    httpMethod: "POST",
    responses: {
        200: {
            bodyMapper: Mappers.IntegrationRuntimeMonitoringData
        },
        default: {
            bodyMapper: Mappers.ErrorResponse
        }
    },
    queryParameters: [Parameters.apiVersion1],
    urlParameters: [
        Parameters.$host,
        Parameters.subscriptionId,
        Parameters.resourceGroupName,
        Parameters.workspaceName,
        Parameters.integrationRuntimeName
    ],
    headerParameters: [Parameters.accept],
    serializer
};
