// @azure/arm-synapse/dist-esm/src/operations/sparkConfiguration.js
/*
 * Copyright (c) Microsoft Corporation.
 * Licensed under the MIT License.
 *
 * Code generated by Microsoft (R) AutoRest Code Generator.
 * Changes may cause incorrect behavior and will be lost if the code is regenerated.
 */
import * as coreClient from "@azure/core-client";
import * as Mappers from "../models/mappers";
import * as Parameters from "../models/parameters";
/** Class containing SparkConfiguration operations. */
export class SparkConfigurationImpl {
    /**
     * Initialize a new instance of the class SparkConfiguration class.
     * @param client Reference to the service client
     */
    constructor(client) {
        this.client = client;
    }
    /**
     * Get SparkConfiguration by name in a workspace.
     * @param resourceGroupName The name of the resource group. The name is case insensitive.
     * @param sparkConfigurationName SparkConfiguration name
     * @param workspaceName The name of the workspace.
     * @param options The options parameters.
     */
    get(resourceGroupName, sparkConfigurationName, workspaceName, options) {
        return this.client.sendOperationRequest({ resourceGroupName, sparkConfigurationName, workspaceName, options }, getOperationSpec);
    }
}
// Operation Specifications
const serializer = coreClient.createSerializer(Mappers, /* isXml */ false);
const getOperationSpec = {
    path: "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Synapse/workspaces/{workspaceName}/sparkconfigurations/{sparkConfigurationName}",
    httpMethod: "GET",
    responses: {
        200: {
            bodyMapper: Mappers.SparkConfigurationResource
        },
        default: {
            bodyMapper: Mappers.ErrorResponse
        }
    },
    queryParameters: [Parameters.apiVersion1],
    urlParameters: [
        Parameters.$host,
        Parameters.subscriptionId,
        Parameters.resourceGroupName,
        Parameters.workspaceName,
        Parameters.sparkConfigurationName
    ],
    headerParameters: [Parameters.accept],
    serializer
};
