// @azure/arm-synapse/dist-esm/src/operations/sqlPoolTableColumns.js
/*
 * Copyright (c) Microsoft Corporation.
 * Licensed under the MIT License.
 *
 * Code generated by Microsoft (R) AutoRest Code Generator.
 * Changes may cause incorrect behavior and will be lost if the code is regenerated.
 */
import { __asyncDelegator, __asyncGenerator, __asyncValues, __await } from "tslib";
import * as coreClient from "@azure/core-client";
import * as Mappers from "../models/mappers";
import * as Parameters from "../models/parameters";
/// <reference lib="esnext.asynciterable" />
/** Class containing SqlPoolTableColumns operations. */
export class SqlPoolTableColumnsImpl {
    /**
     * Initialize a new instance of the class SqlPoolTableColumns class.
     * @param client Reference to the service client
     */
    constructor(client) {
        this.client = client;
    }
    /**
     * Gets columns in a given table in a SQL pool.
     * @param resourceGroupName The name of the resource group. The name is case insensitive.
     * @param workspaceName The name of the workspace.
     * @param sqlPoolName SQL pool name
     * @param schemaName The name of the schema.
     * @param tableName The name of the table.
     * @param options The options parameters.
     */
    listByTableName(resourceGroupName, workspaceName, sqlPoolName, schemaName, tableName, options) {
        const iter = this.listByTableNamePagingAll(resourceGroupName, workspaceName, sqlPoolName, schemaName, tableName, options);
        return {
            next() {
                return iter.next();
            },
            [Symbol.asyncIterator]() {
                return this;
            },
            byPage: () => {
                return this.listByTableNamePagingPage(resourceGroupName, workspaceName, sqlPoolName, schemaName, tableName, options);
            }
        };
    }
    listByTableNamePagingPage(resourceGroupName, workspaceName, sqlPoolName, schemaName, tableName, options) {
        return __asyncGenerator(this, arguments, function* listByTableNamePagingPage_1() {
            let result = yield __await(this._listByTableName(resourceGroupName, workspaceName, sqlPoolName, schemaName, tableName, options));
            yield yield __await(result.value || []);
            let continuationToken = result.nextLink;
            while (continuationToken) {
                result = yield __await(this._listByTableNameNext(resourceGroupName, workspaceName, sqlPoolName, schemaName, tableName, continuationToken, options));
                continuationToken = result.nextLink;
                yield yield __await(result.value || []);
            }
        });
    }
    listByTableNamePagingAll(resourceGroupName, workspaceName, sqlPoolName, schemaName, tableName, options) {
        return __asyncGenerator(this, arguments, function* listByTableNamePagingAll_1() {
            var e_1, _a;
            try {
                for (var _b = __asyncValues(this.listByTableNamePagingPage(resourceGroupName, workspaceName, sqlPoolName, schemaName, tableName, options)), _c; _c = yield __await(_b.next()), !_c.done;) {
                    const page = _c.value;
                    yield __await(yield* __asyncDelegator(__asyncValues(page)));
                }
            }
            catch (e_1_1) { e_1 = { error: e_1_1 }; }
            finally {
                try {
                    if (_c && !_c.done && (_a = _b.return)) yield __await(_a.call(_b));
                }
                finally { if (e_1) throw e_1.error; }
            }
        });
    }
    /**
     * Gets columns in a given table in a SQL pool.
     * @param resourceGroupName The name of the resource group. The name is case insensitive.
     * @param workspaceName The name of the workspace.
     * @param sqlPoolName SQL pool name
     * @param schemaName The name of the schema.
     * @param tableName The name of the table.
     * @param options The options parameters.
     */
    _listByTableName(resourceGroupName, workspaceName, sqlPoolName, schemaName, tableName, options) {
        return this.client.sendOperationRequest({
            resourceGroupName,
            workspaceName,
            sqlPoolName,
            schemaName,
            tableName,
            options
        }, listByTableNameOperationSpec);
    }
    /**
     * ListByTableNameNext
     * @param resourceGroupName The name of the resource group. The name is case insensitive.
     * @param workspaceName The name of the workspace.
     * @param sqlPoolName SQL pool name
     * @param schemaName The name of the schema.
     * @param tableName The name of the table.
     * @param nextLink The nextLink from the previous successful call to the ListByTableName method.
     * @param options The options parameters.
     */
    _listByTableNameNext(resourceGroupName, workspaceName, sqlPoolName, schemaName, tableName, nextLink, options) {
        return this.client.sendOperationRequest({
            resourceGroupName,
            workspaceName,
            sqlPoolName,
            schemaName,
            tableName,
            nextLink,
            options
        }, listByTableNameNextOperationSpec);
    }
}
// Operation Specifications
const serializer = coreClient.createSerializer(Mappers, /* isXml */ false);
const listByTableNameOperationSpec = {
    path: "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Synapse/workspaces/{workspaceName}/sqlPools/{sqlPoolName}/schemas/{schemaName}/tables/{tableName}/columns",
    httpMethod: "GET",
    responses: {
        200: {
            bodyMapper: Mappers.SqlPoolColumnListResult
        },
        default: {}
    },
    queryParameters: [Parameters.apiVersion, Parameters.filter],
    urlParameters: [
        Parameters.$host,
        Parameters.subscriptionId,
        Parameters.resourceGroupName,
        Parameters.workspaceName,
        Parameters.sqlPoolName,
        Parameters.schemaName,
        Parameters.tableName
    ],
    headerParameters: [Parameters.accept],
    serializer
};
const listByTableNameNextOperationSpec = {
    path: "{nextLink}",
    httpMethod: "GET",
    responses: {
        200: {
            bodyMapper: Mappers.SqlPoolColumnListResult
        },
        default: {}
    },
    queryParameters: [Parameters.apiVersion, Parameters.filter],
    urlParameters: [
        Parameters.$host,
        Parameters.subscriptionId,
        Parameters.resourceGroupName,
        Parameters.workspaceName,
        Parameters.nextLink,
        Parameters.sqlPoolName,
        Parameters.schemaName,
        Parameters.tableName
    ],
    headerParameters: [Parameters.accept],
    serializer
};
