// @google-cloud/service-usage/build/src/v1beta1/service_usage_client.js
"use strict";
// Copyright 2024 Google LLC
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     https://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.
//
// ** This file is automatically generated by gapic-generator-typescript. **
// ** https://github.com/googleapis/gapic-generator-typescript **
// ** All changes to this file may be overwritten. **
Object.defineProperty(exports, "__esModule", { value: true });
exports.ServiceUsageClient = void 0;
const jsonProtos = require("../../protos/protos.json");
/**
 * Client JSON configuration object, loaded from
 * `src/v1beta1/service_usage_client_config.json`.
 * This file defines retry strategy and timeouts for all API methods in this library.
 */
const gapicConfig = require("./service_usage_client_config.json");
const version = require('../../../package.json').version;
/**
 *  [Service Usage API](https://cloud.google.com/service-usage/docs/overview)
 * @class
 * @memberof v1beta1
 */
class ServiceUsageClient {
    /**
     * Construct an instance of ServiceUsageClient.
     *
     * @param {object} [options] - The configuration object.
     * The options accepted by the constructor are described in detail
     * in [this document](https://github.com/googleapis/gax-nodejs/blob/main/client-libraries.md#creating-the-client-instance).
     * The common options are:
     * @param {object} [options.credentials] - Credentials object.
     * @param {string} [options.credentials.client_email]
     * @param {string} [options.credentials.private_key]
     * @param {string} [options.email] - Account email address. Required when
     *     using a .pem or .p12 keyFilename.
     * @param {string} [options.keyFilename] - Full path to the a .json, .pem, or
     *     .p12 key downloaded from the Google Developers Console. If you provide
     *     a path to a JSON file, the projectId option below is not necessary.
     *     NOTE: .pem and .p12 require you to specify options.email as well.
     * @param {number} [options.port] - The port on which to connect to
     *     the remote host.
     * @param {string} [options.projectId] - The project ID from the Google
     *     Developer's Console, e.g. 'grape-spaceship-123'. We will also check
     *     the environment variable GCLOUD_PROJECT for your project ID. If your
     *     app is running in an environment which supports
     *     {@link https://developers.google.com/identity/protocols/application-default-credentials Application Default Credentials},
     *     your project ID will be detected automatically.
     * @param {string} [options.apiEndpoint] - The domain name of the
     *     API remote host.
     * @param {gax.ClientConfig} [options.clientConfig] - Client configuration override.
     *     Follows the structure of {@link gapicConfig}.
     * @param {boolean} [options.fallback] - Use HTTP/1.1 REST mode.
     *     For more information, please check the
     *     {@link https://github.com/googleapis/gax-nodejs/blob/main/client-libraries.md#http11-rest-api-mode documentation}.
     * @param {gax} [gaxInstance]: loaded instance of `google-gax`. Useful if you
     *     need to avoid loading the default gRPC version and want to use the fallback
     *     HTTP implementation. Load only fallback version and pass it to the constructor:
     *     ```
     *     const gax = require('google-gax/build/src/fallback'); // avoids loading google-gax with gRPC
     *     const client = new ServiceUsageClient({fallback: true}, gax);
     *     ```
     */
    constructor(opts, gaxInstance) {
        var _a, _b, _c, _d, _e;
        this._terminated = false;
        this.descriptors = {
            page: {},
            stream: {},
            longrunning: {},
            batching: {},
        };
        // Ensure that options include all the required fields.
        const staticMembers = this.constructor;
        if ((opts === null || opts === void 0 ? void 0 : opts.universe_domain) &&
            (opts === null || opts === void 0 ? void 0 : opts.universeDomain) &&
            (opts === null || opts === void 0 ? void 0 : opts.universe_domain) !== (opts === null || opts === void 0 ? void 0 : opts.universeDomain)) {
            throw new Error('Please set either universe_domain or universeDomain, but not both.');
        }
        const universeDomainEnvVar = typeof process === 'object' && typeof process.env === 'object'
            ? process.env['GOOGLE_CLOUD_UNIVERSE_DOMAIN']
            : undefined;
        this._universeDomain =
            (_c = (_b = (_a = opts === null || opts === void 0 ? void 0 : opts.universeDomain) !== null && _a !== void 0 ? _a : opts === null || opts === void 0 ? void 0 : opts.universe_domain) !== null && _b !== void 0 ? _b : universeDomainEnvVar) !== null && _c !== void 0 ? _c : 'googleapis.com';
        this._servicePath = 'serviceusage.' + this._universeDomain;
        const servicePath = (opts === null || opts === void 0 ? void 0 : opts.servicePath) || (opts === null || opts === void 0 ? void 0 : opts.apiEndpoint) || this._servicePath;
        this._providedCustomServicePath = !!((opts === null || opts === void 0 ? void 0 : opts.servicePath) || (opts === null || opts === void 0 ? void 0 : opts.apiEndpoint));
        const port = (opts === null || opts === void 0 ? void 0 : opts.port) || staticMembers.port;
        const clientConfig = (_d = opts === null || opts === void 0 ? void 0 : opts.clientConfig) !== null && _d !== void 0 ? _d : {};
        const fallback = (_e = opts === null || opts === void 0 ? void 0 : opts.fallback) !== null && _e !== void 0 ? _e : (typeof window !== 'undefined' && typeof (window === null || window === void 0 ? void 0 : window.fetch) === 'function');
        opts = Object.assign({ servicePath, port, clientConfig, fallback }, opts);
        // Request numeric enum values if REST transport is used.
        opts.numericEnums = true;
        // If scopes are unset in options and we're connecting to a non-default endpoint, set scopes just in case.
        if (servicePath !== this._servicePath && !('scopes' in opts)) {
            opts['scopes'] = staticMembers.scopes;
        }
        // Load google-gax module synchronously if needed
        if (!gaxInstance) {
            gaxInstance = require('google-gax');
        }
        // Choose either gRPC or proto-over-HTTP implementation of google-gax.
        this._gaxModule = opts.fallback ? gaxInstance.fallback : gaxInstance;
        // Create a `gaxGrpc` object, with any grpc-specific options sent to the client.
        this._gaxGrpc = new this._gaxModule.GrpcClient(opts);
        // Save options to use in initialize() method.
        this._opts = opts;
        // Save the auth object to the client, for use by other methods.
        this.auth = this._gaxGrpc.auth;
        // Set useJWTAccessWithScope on the auth object.
        this.auth.useJWTAccessWithScope = true;
        // Set defaultServicePath on the auth object.
        this.auth.defaultServicePath = this._servicePath;
        // Set the default scopes in auth client if needed.
        if (servicePath === this._servicePath) {
            this.auth.defaultScopes = staticMembers.scopes;
        }
        // Determine the client header string.
        const clientHeader = [`gax/${this._gaxModule.version}`, `gapic/${version}`];
        if (typeof process === 'object' && 'versions' in process) {
            clientHeader.push(`gl-node/${process.versions.node}`);
        }
        else {
            clientHeader.push(`gl-web/${this._gaxModule.version}`);
        }
        if (!opts.fallback) {
            clientHeader.push(`grpc/${this._gaxGrpc.grpcVersion}`);
        }
        else {
            clientHeader.push(`rest/${this._gaxGrpc.grpcVersion}`);
        }
        if (opts.libName && opts.libVersion) {
            clientHeader.push(`${opts.libName}/${opts.libVersion}`);
        }
        // Load the applicable protos.
        this._protos = this._gaxGrpc.loadProtoJSON(jsonProtos);
        // Some of the methods on this service return "paged" results,
        // (e.g. 50 results at a time, with tokens to get subsequent
        // pages). Denote the keys used for pagination and results.
        this.descriptors.page = {
            listServices: new this._gaxModule.PageDescriptor('pageToken', 'nextPageToken', 'services'),
            listConsumerQuotaMetrics: new this._gaxModule.PageDescriptor('pageToken', 'nextPageToken', 'metrics'),
            listAdminOverrides: new this._gaxModule.PageDescriptor('pageToken', 'nextPageToken', 'overrides'),
            listConsumerOverrides: new this._gaxModule.PageDescriptor('pageToken', 'nextPageToken', 'overrides'),
        };
        const protoFilesRoot = this._gaxModule.protobuf.Root.fromJSON(jsonProtos);
        // This API contains "long-running operations", which return a
        // an Operation object that allows for tracking of the operation,
        // rather than holding a request open.
        const lroOptions = {
            auth: this.auth,
            grpc: 'grpc' in this._gaxGrpc ? this._gaxGrpc.grpc : undefined,
        };
        if (opts.fallback) {
            lroOptions.protoJson = protoFilesRoot;
            lroOptions.httpRules = [
                {
                    selector: 'google.longrunning.Operations.GetOperation',
                    get: '/v1beta1/{name=operations/*}',
                },
                {
                    selector: 'google.longrunning.Operations.ListOperations',
                    get: '/v1beta1/operations',
                },
            ];
        }
        this.operationsClient = this._gaxModule
            .lro(lroOptions)
            .operationsClient(opts);
        const enableServiceResponse = protoFilesRoot.lookup('.google.protobuf.Empty');
        const enableServiceMetadata = protoFilesRoot.lookup('.google.api.serviceusage.v1beta1.OperationMetadata');
        const disableServiceResponse = protoFilesRoot.lookup('.google.protobuf.Empty');
        const disableServiceMetadata = protoFilesRoot.lookup('.google.api.serviceusage.v1beta1.OperationMetadata');
        const batchEnableServicesResponse = protoFilesRoot.lookup('.google.protobuf.Empty');
        const batchEnableServicesMetadata = protoFilesRoot.lookup('.google.api.serviceusage.v1beta1.OperationMetadata');
        const createAdminOverrideResponse = protoFilesRoot.lookup('.google.api.serviceusage.v1beta1.QuotaOverride');
        const createAdminOverrideMetadata = protoFilesRoot.lookup('.google.api.serviceusage.v1beta1.OperationMetadata');
        const updateAdminOverrideResponse = protoFilesRoot.lookup('.google.api.serviceusage.v1beta1.QuotaOverride');
        const updateAdminOverrideMetadata = protoFilesRoot.lookup('.google.api.serviceusage.v1beta1.OperationMetadata');
        const deleteAdminOverrideResponse = protoFilesRoot.lookup('.google.protobuf.Empty');
        const deleteAdminOverrideMetadata = protoFilesRoot.lookup('.google.api.serviceusage.v1beta1.OperationMetadata');
        const importAdminOverridesResponse = protoFilesRoot.lookup('.google.api.serviceusage.v1beta1.ImportAdminOverridesResponse');
        const importAdminOverridesMetadata = protoFilesRoot.lookup('.google.api.serviceusage.v1beta1.ImportAdminOverridesMetadata');
        const createConsumerOverrideResponse = protoFilesRoot.lookup('.google.api.serviceusage.v1beta1.QuotaOverride');
        const createConsumerOverrideMetadata = protoFilesRoot.lookup('.google.api.serviceusage.v1beta1.OperationMetadata');
        const updateConsumerOverrideResponse = protoFilesRoot.lookup('.google.api.serviceusage.v1beta1.QuotaOverride');
        const updateConsumerOverrideMetadata = protoFilesRoot.lookup('.google.api.serviceusage.v1beta1.OperationMetadata');
        const deleteConsumerOverrideResponse = protoFilesRoot.lookup('.google.protobuf.Empty');
        const deleteConsumerOverrideMetadata = protoFilesRoot.lookup('.google.api.serviceusage.v1beta1.OperationMetadata');
        const importConsumerOverridesResponse = protoFilesRoot.lookup('.google.api.serviceusage.v1beta1.ImportConsumerOverridesResponse');
        const importConsumerOverridesMetadata = protoFilesRoot.lookup('.google.api.serviceusage.v1beta1.ImportConsumerOverridesMetadata');
        const generateServiceIdentityResponse = protoFilesRoot.lookup('.google.api.serviceusage.v1beta1.ServiceIdentity');
        const generateServiceIdentityMetadata = protoFilesRoot.lookup('.google.protobuf.Empty');
        this.descriptors.longrunning = {
            enableService: new this._gaxModule.LongrunningDescriptor(this.operationsClient, enableServiceResponse.decode.bind(enableServiceResponse), enableServiceMetadata.decode.bind(enableServiceMetadata)),
            disableService: new this._gaxModule.LongrunningDescriptor(this.operationsClient, disableServiceResponse.decode.bind(disableServiceResponse), disableServiceMetadata.decode.bind(disableServiceMetadata)),
            batchEnableServices: new this._gaxModule.LongrunningDescriptor(this.operationsClient, batchEnableServicesResponse.decode.bind(batchEnableServicesResponse), batchEnableServicesMetadata.decode.bind(batchEnableServicesMetadata)),
            createAdminOverride: new this._gaxModule.LongrunningDescriptor(this.operationsClient, createAdminOverrideResponse.decode.bind(createAdminOverrideResponse), createAdminOverrideMetadata.decode.bind(createAdminOverrideMetadata)),
            updateAdminOverride: new this._gaxModule.LongrunningDescriptor(this.operationsClient, updateAdminOverrideResponse.decode.bind(updateAdminOverrideResponse), updateAdminOverrideMetadata.decode.bind(updateAdminOverrideMetadata)),
            deleteAdminOverride: new this._gaxModule.LongrunningDescriptor(this.operationsClient, deleteAdminOverrideResponse.decode.bind(deleteAdminOverrideResponse), deleteAdminOverrideMetadata.decode.bind(deleteAdminOverrideMetadata)),
            importAdminOverrides: new this._gaxModule.LongrunningDescriptor(this.operationsClient, importAdminOverridesResponse.decode.bind(importAdminOverridesResponse), importAdminOverridesMetadata.decode.bind(importAdminOverridesMetadata)),
            createConsumerOverride: new this._gaxModule.LongrunningDescriptor(this.operationsClient, createConsumerOverrideResponse.decode.bind(createConsumerOverrideResponse), createConsumerOverrideMetadata.decode.bind(createConsumerOverrideMetadata)),
            updateConsumerOverride: new this._gaxModule.LongrunningDescriptor(this.operationsClient, updateConsumerOverrideResponse.decode.bind(updateConsumerOverrideResponse), updateConsumerOverrideMetadata.decode.bind(updateConsumerOverrideMetadata)),
            deleteConsumerOverride: new this._gaxModule.LongrunningDescriptor(this.operationsClient, deleteConsumerOverrideResponse.decode.bind(deleteConsumerOverrideResponse), deleteConsumerOverrideMetadata.decode.bind(deleteConsumerOverrideMetadata)),
            importConsumerOverrides: new this._gaxModule.LongrunningDescriptor(this.operationsClient, importConsumerOverridesResponse.decode.bind(importConsumerOverridesResponse), importConsumerOverridesMetadata.decode.bind(importConsumerOverridesMetadata)),
            generateServiceIdentity: new this._gaxModule.LongrunningDescriptor(this.operationsClient, generateServiceIdentityResponse.decode.bind(generateServiceIdentityResponse), generateServiceIdentityMetadata.decode.bind(generateServiceIdentityMetadata)),
        };
        // Put together the default options sent with requests.
        this._defaults = this._gaxGrpc.constructSettings('google.api.serviceusage.v1beta1.ServiceUsage', gapicConfig, opts.clientConfig || {}, { 'x-goog-api-client': clientHeader.join(' ') });
        // Set up a dictionary of "inner API calls"; the core implementation
        // of calling the API is handled in `google-gax`, with this code
        // merely providing the destination and request information.
        this.innerApiCalls = {};
        // Add a warn function to the client constructor so it can be easily tested.
        this.warn = this._gaxModule.warn;
    }
    /**
     * Initialize the client.
     * Performs asynchronous operations (such as authentication) and prepares the client.
     * This function will be called automatically when any class method is called for the
     * first time, but if you need to initialize it before calling an actual method,
     * feel free to call initialize() directly.
     *
     * You can await on this method if you want to make sure the client is initialized.
     *
     * @returns {Promise} A promise that resolves to an authenticated service stub.
     */
    initialize() {
        // If the client stub promise is already initialized, return immediately.
        if (this.serviceUsageStub) {
            return this.serviceUsageStub;
        }
        // Put together the "service stub" for
        // google.api.serviceusage.v1beta1.ServiceUsage.
        this.serviceUsageStub = this._gaxGrpc.createStub(this._opts.fallback
            ? this._protos.lookupService('google.api.serviceusage.v1beta1.ServiceUsage')
            : // eslint-disable-next-line @typescript-eslint/no-explicit-any
                this._protos.google.api.serviceusage.v1beta1.ServiceUsage, this._opts, this._providedCustomServicePath);
        // Iterate over each of the methods that the service provides
        // and create an API call method for each.
        const serviceUsageStubMethods = [
            'enableService',
            'disableService',
            'getService',
            'listServices',
            'batchEnableServices',
            'listConsumerQuotaMetrics',
            'getConsumerQuotaMetric',
            'getConsumerQuotaLimit',
            'createAdminOverride',
            'updateAdminOverride',
            'deleteAdminOverride',
            'listAdminOverrides',
            'importAdminOverrides',
            'createConsumerOverride',
            'updateConsumerOverride',
            'deleteConsumerOverride',
            'listConsumerOverrides',
            'importConsumerOverrides',
            'generateServiceIdentity',
        ];
        for (const methodName of serviceUsageStubMethods) {
            const callPromise = this.serviceUsageStub.then(stub => (...args) => {
                if (this._terminated) {
                    return Promise.reject('The client has already been closed.');
                }
                const func = stub[methodName];
                return func.apply(stub, args);
            }, (err) => () => {
                throw err;
            });
            const descriptor = this.descriptors.page[methodName] ||
                this.descriptors.longrunning[methodName] ||
                undefined;
            const apiCall = this._gaxModule.createApiCall(callPromise, this._defaults[methodName], descriptor, this._opts.fallback);
            this.innerApiCalls[methodName] = apiCall;
        }
        return this.serviceUsageStub;
    }
    /**
     * The DNS address for this API service.
     * @deprecated Use the apiEndpoint method of the client instance.
     * @returns {string} The DNS address for this service.
     */
    static get servicePath() {
        if (typeof process === 'object' &&
            typeof process.emitWarning === 'function') {
            process.emitWarning('Static servicePath is deprecated, please use the instance method instead.', 'DeprecationWarning');
        }
        return 'serviceusage.googleapis.com';
    }
    /**
     * The DNS address for this API service - same as servicePath.
     * @deprecated Use the apiEndpoint method of the client instance.
     * @returns {string} The DNS address for this service.
     */
    static get apiEndpoint() {
        if (typeof process === 'object' &&
            typeof process.emitWarning === 'function') {
            process.emitWarning('Static apiEndpoint is deprecated, please use the instance method instead.', 'DeprecationWarning');
        }
        return 'serviceusage.googleapis.com';
    }
    /**
     * The DNS address for this API service.
     * @returns {string} The DNS address for this service.
     */
    get apiEndpoint() {
        return this._servicePath;
    }
    get universeDomain() {
        return this._universeDomain;
    }
    /**
     * The port for this API service.
     * @returns {number} The default port for this service.
     */
    static get port() {
        return 443;
    }
    /**
     * The scopes needed to make gRPC calls for every method defined
     * in this service.
     * @returns {string[]} List of default scopes.
     */
    static get scopes() {
        return [
            'https://www.googleapis.com/auth/cloud-platform',
            'https://www.googleapis.com/auth/cloud-platform.read-only',
            'https://www.googleapis.com/auth/service.management',
        ];
    }
    /**
     * Return the project ID used by this class.
     * @returns {Promise} A promise that resolves to string containing the project ID.
     */
    getProjectId(callback) {
        if (callback) {
            this.auth.getProjectId(callback);
            return;
        }
        return this.auth.getProjectId();
    }
    getService(request, optionsOrCallback, callback) {
        var _a;
        request = request || {};
        let options;
        if (typeof optionsOrCallback === 'function' && callback === undefined) {
            callback = optionsOrCallback;
            options = {};
        }
        else {
            options = optionsOrCallback;
        }
        options = options || {};
        options.otherArgs = options.otherArgs || {};
        options.otherArgs.headers = options.otherArgs.headers || {};
        options.otherArgs.headers['x-goog-request-params'] =
            this._gaxModule.routingHeader.fromParams({
                name: (_a = request.name) !== null && _a !== void 0 ? _a : '',
            });
        this.initialize();
        this.warn('DEP$ServiceUsage-$GetService', 'GetService is deprecated and may be removed in a future version.', 'DeprecationWarning');
        return this.innerApiCalls.getService(request, options, callback);
    }
    getConsumerQuotaMetric(request, optionsOrCallback, callback) {
        var _a;
        request = request || {};
        let options;
        if (typeof optionsOrCallback === 'function' && callback === undefined) {
            callback = optionsOrCallback;
            options = {};
        }
        else {
            options = optionsOrCallback;
        }
        options = options || {};
        options.otherArgs = options.otherArgs || {};
        options.otherArgs.headers = options.otherArgs.headers || {};
        options.otherArgs.headers['x-goog-request-params'] =
            this._gaxModule.routingHeader.fromParams({
                name: (_a = request.name) !== null && _a !== void 0 ? _a : '',
            });
        this.initialize();
        return this.innerApiCalls.getConsumerQuotaMetric(request, options, callback);
    }
    getConsumerQuotaLimit(request, optionsOrCallback, callback) {
        var _a;
        request = request || {};
        let options;
        if (typeof optionsOrCallback === 'function' && callback === undefined) {
            callback = optionsOrCallback;
            options = {};
        }
        else {
            options = optionsOrCallback;
        }
        options = options || {};
        options.otherArgs = options.otherArgs || {};
        options.otherArgs.headers = options.otherArgs.headers || {};
        options.otherArgs.headers['x-goog-request-params'] =
            this._gaxModule.routingHeader.fromParams({
                name: (_a = request.name) !== null && _a !== void 0 ? _a : '',
            });
        this.initialize();
        return this.innerApiCalls.getConsumerQuotaLimit(request, options, callback);
    }
    enableService(request, optionsOrCallback, callback) {
        var _a;
        request = request || {};
        let options;
        if (typeof optionsOrCallback === 'function' && callback === undefined) {
            callback = optionsOrCallback;
            options = {};
        }
        else {
            options = optionsOrCallback;
        }
        options = options || {};
        options.otherArgs = options.otherArgs || {};
        options.otherArgs.headers = options.otherArgs.headers || {};
        options.otherArgs.headers['x-goog-request-params'] =
            this._gaxModule.routingHeader.fromParams({
                name: (_a = request.name) !== null && _a !== void 0 ? _a : '',
            });
        this.initialize();
        this.warn('DEP$ServiceUsage-$EnableService', 'EnableService is deprecated and may be removed in a future version.', 'DeprecationWarning');
        return this.innerApiCalls.enableService(request, options, callback);
    }
    /**
     * Check the status of the long running operation returned by `enableService()`.
     * @param {String} name
     *   The operation name that will be passed.
     * @returns {Promise} - The promise which resolves to an object.
     *   The decoded operation object has result and metadata field to get information from.
     *   Please see the {@link https://github.com/googleapis/gax-nodejs/blob/master/client-libraries.md#long-running-operations | documentation }
     *   for more details and examples.
     * @example <caption>include:samples/generated/v1beta1/service_usage.enable_service.js</caption>
     * region_tag:serviceusage_v1beta1_generated_ServiceUsage_EnableService_async
     * @deprecated EnableService is deprecated and may be removed in a future version.
     */
    async checkEnableServiceProgress(name) {
        this.warn('DEP$ServiceUsage-$checkEnableServiceProgress', 'checkEnableServiceProgress is deprecated and may be removed in a future version.', 'DeprecationWarning');
        const request = new this._gaxModule.operationsProtos.google.longrunning.GetOperationRequest({ name });
        const [operation] = await this.operationsClient.getOperation(request);
        const decodeOperation = new this._gaxModule.Operation(operation, this.descriptors.longrunning.enableService, this._gaxModule.createDefaultBackoffSettings());
        return decodeOperation;
    }
    disableService(request, optionsOrCallback, callback) {
        var _a;
        request = request || {};
        let options;
        if (typeof optionsOrCallback === 'function' && callback === undefined) {
            callback = optionsOrCallback;
            options = {};
        }
        else {
            options = optionsOrCallback;
        }
        options = options || {};
        options.otherArgs = options.otherArgs || {};
        options.otherArgs.headers = options.otherArgs.headers || {};
        options.otherArgs.headers['x-goog-request-params'] =
            this._gaxModule.routingHeader.fromParams({
                name: (_a = request.name) !== null && _a !== void 0 ? _a : '',
            });
        this.initialize();
        this.warn('DEP$ServiceUsage-$DisableService', 'DisableService is deprecated and may be removed in a future version.', 'DeprecationWarning');
        return this.innerApiCalls.disableService(request, options, callback);
    }
    /**
     * Check the status of the long running operation returned by `disableService()`.
     * @param {String} name
     *   The operation name that will be passed.
     * @returns {Promise} - The promise which resolves to an object.
     *   The decoded operation object has result and metadata field to get information from.
     *   Please see the {@link https://github.com/googleapis/gax-nodejs/blob/master/client-libraries.md#long-running-operations | documentation }
     *   for more details and examples.
     * @example <caption>include:samples/generated/v1beta1/service_usage.disable_service.js</caption>
     * region_tag:serviceusage_v1beta1_generated_ServiceUsage_DisableService_async
     * @deprecated DisableService is deprecated and may be removed in a future version.
     */
    async checkDisableServiceProgress(name) {
        this.warn('DEP$ServiceUsage-$checkDisableServiceProgress', 'checkDisableServiceProgress is deprecated and may be removed in a future version.', 'DeprecationWarning');
        const request = new this._gaxModule.operationsProtos.google.longrunning.GetOperationRequest({ name });
        const [operation] = await this.operationsClient.getOperation(request);
        const decodeOperation = new this._gaxModule.Operation(operation, this.descriptors.longrunning.disableService, this._gaxModule.createDefaultBackoffSettings());
        return decodeOperation;
    }
    batchEnableServices(request, optionsOrCallback, callback) {
        var _a;
        request = request || {};
        let options;
        if (typeof optionsOrCallback === 'function' && callback === undefined) {
            callback = optionsOrCallback;
            options = {};
        }
        else {
            options = optionsOrCallback;
        }
        options = options || {};
        options.otherArgs = options.otherArgs || {};
        options.otherArgs.headers = options.otherArgs.headers || {};
        options.otherArgs.headers['x-goog-request-params'] =
            this._gaxModule.routingHeader.fromParams({
                parent: (_a = request.parent) !== null && _a !== void 0 ? _a : '',
            });
        this.initialize();
        this.warn('DEP$ServiceUsage-$BatchEnableServices', 'BatchEnableServices is deprecated and may be removed in a future version.', 'DeprecationWarning');
        return this.innerApiCalls.batchEnableServices(request, options, callback);
    }
    /**
     * Check the status of the long running operation returned by `batchEnableServices()`.
     * @param {String} name
     *   The operation name that will be passed.
     * @returns {Promise} - The promise which resolves to an object.
     *   The decoded operation object has result and metadata field to get information from.
     *   Please see the {@link https://github.com/googleapis/gax-nodejs/blob/master/client-libraries.md#long-running-operations | documentation }
     *   for more details and examples.
     * @example <caption>include:samples/generated/v1beta1/service_usage.batch_enable_services.js</caption>
     * region_tag:serviceusage_v1beta1_generated_ServiceUsage_BatchEnableServices_async
     * @deprecated BatchEnableServices is deprecated and may be removed in a future version.
     */
    async checkBatchEnableServicesProgress(name) {
        this.warn('DEP$ServiceUsage-$checkBatchEnableServicesProgress', 'checkBatchEnableServicesProgress is deprecated and may be removed in a future version.', 'DeprecationWarning');
        const request = new this._gaxModule.operationsProtos.google.longrunning.GetOperationRequest({ name });
        const [operation] = await this.operationsClient.getOperation(request);
        const decodeOperation = new this._gaxModule.Operation(operation, this.descriptors.longrunning.batchEnableServices, this._gaxModule.createDefaultBackoffSettings());
        return decodeOperation;
    }
    createAdminOverride(request, optionsOrCallback, callback) {
        var _a;
        request = request || {};
        let options;
        if (typeof optionsOrCallback === 'function' && callback === undefined) {
            callback = optionsOrCallback;
            options = {};
        }
        else {
            options = optionsOrCallback;
        }
        options = options || {};
        options.otherArgs = options.otherArgs || {};
        options.otherArgs.headers = options.otherArgs.headers || {};
        options.otherArgs.headers['x-goog-request-params'] =
            this._gaxModule.routingHeader.fromParams({
                parent: (_a = request.parent) !== null && _a !== void 0 ? _a : '',
            });
        this.initialize();
        return this.innerApiCalls.createAdminOverride(request, options, callback);
    }
    /**
     * Check the status of the long running operation returned by `createAdminOverride()`.
     * @param {String} name
     *   The operation name that will be passed.
     * @returns {Promise} - The promise which resolves to an object.
     *   The decoded operation object has result and metadata field to get information from.
     *   Please see the {@link https://github.com/googleapis/gax-nodejs/blob/master/client-libraries.md#long-running-operations | documentation }
     *   for more details and examples.
     * @example <caption>include:samples/generated/v1beta1/service_usage.create_admin_override.js</caption>
     * region_tag:serviceusage_v1beta1_generated_ServiceUsage_CreateAdminOverride_async
     */
    async checkCreateAdminOverrideProgress(name) {
        const request = new this._gaxModule.operationsProtos.google.longrunning.GetOperationRequest({ name });
        const [operation] = await this.operationsClient.getOperation(request);
        const decodeOperation = new this._gaxModule.Operation(operation, this.descriptors.longrunning.createAdminOverride, this._gaxModule.createDefaultBackoffSettings());
        return decodeOperation;
    }
    updateAdminOverride(request, optionsOrCallback, callback) {
        var _a;
        request = request || {};
        let options;
        if (typeof optionsOrCallback === 'function' && callback === undefined) {
            callback = optionsOrCallback;
            options = {};
        }
        else {
            options = optionsOrCallback;
        }
        options = options || {};
        options.otherArgs = options.otherArgs || {};
        options.otherArgs.headers = options.otherArgs.headers || {};
        options.otherArgs.headers['x-goog-request-params'] =
            this._gaxModule.routingHeader.fromParams({
                name: (_a = request.name) !== null && _a !== void 0 ? _a : '',
            });
        this.initialize();
        return this.innerApiCalls.updateAdminOverride(request, options, callback);
    }
    /**
     * Check the status of the long running operation returned by `updateAdminOverride()`.
     * @param {String} name
     *   The operation name that will be passed.
     * @returns {Promise} - The promise which resolves to an object.
     *   The decoded operation object has result and metadata field to get information from.
     *   Please see the {@link https://github.com/googleapis/gax-nodejs/blob/master/client-libraries.md#long-running-operations | documentation }
     *   for more details and examples.
     * @example <caption>include:samples/generated/v1beta1/service_usage.update_admin_override.js</caption>
     * region_tag:serviceusage_v1beta1_generated_ServiceUsage_UpdateAdminOverride_async
     */
    async checkUpdateAdminOverrideProgress(name) {
        const request = new this._gaxModule.operationsProtos.google.longrunning.GetOperationRequest({ name });
        const [operation] = await this.operationsClient.getOperation(request);
        const decodeOperation = new this._gaxModule.Operation(operation, this.descriptors.longrunning.updateAdminOverride, this._gaxModule.createDefaultBackoffSettings());
        return decodeOperation;
    }
    deleteAdminOverride(request, optionsOrCallback, callback) {
        var _a;
        request = request || {};
        let options;
        if (typeof optionsOrCallback === 'function' && callback === undefined) {
            callback = optionsOrCallback;
            options = {};
        }
        else {
            options = optionsOrCallback;
        }
        options = options || {};
        options.otherArgs = options.otherArgs || {};
        options.otherArgs.headers = options.otherArgs.headers || {};
        options.otherArgs.headers['x-goog-request-params'] =
            this._gaxModule.routingHeader.fromParams({
                name: (_a = request.name) !== null && _a !== void 0 ? _a : '',
            });
        this.initialize();
        return this.innerApiCalls.deleteAdminOverride(request, options, callback);
    }
    /**
     * Check the status of the long running operation returned by `deleteAdminOverride()`.
     * @param {String} name
     *   The operation name that will be passed.
     * @returns {Promise} - The promise which resolves to an object.
     *   The decoded operation object has result and metadata field to get information from.
     *   Please see the {@link https://github.com/googleapis/gax-nodejs/blob/master/client-libraries.md#long-running-operations | documentation }
     *   for more details and examples.
     * @example <caption>include:samples/generated/v1beta1/service_usage.delete_admin_override.js</caption>
     * region_tag:serviceusage_v1beta1_generated_ServiceUsage_DeleteAdminOverride_async
     */
    async checkDeleteAdminOverrideProgress(name) {
        const request = new this._gaxModule.operationsProtos.google.longrunning.GetOperationRequest({ name });
        const [operation] = await this.operationsClient.getOperation(request);
        const decodeOperation = new this._gaxModule.Operation(operation, this.descriptors.longrunning.deleteAdminOverride, this._gaxModule.createDefaultBackoffSettings());
        return decodeOperation;
    }
    importAdminOverrides(request, optionsOrCallback, callback) {
        var _a;
        request = request || {};
        let options;
        if (typeof optionsOrCallback === 'function' && callback === undefined) {
            callback = optionsOrCallback;
            options = {};
        }
        else {
            options = optionsOrCallback;
        }
        options = options || {};
        options.otherArgs = options.otherArgs || {};
        options.otherArgs.headers = options.otherArgs.headers || {};
        options.otherArgs.headers['x-goog-request-params'] =
            this._gaxModule.routingHeader.fromParams({
                parent: (_a = request.parent) !== null && _a !== void 0 ? _a : '',
            });
        this.initialize();
        return this.innerApiCalls.importAdminOverrides(request, options, callback);
    }
    /**
     * Check the status of the long running operation returned by `importAdminOverrides()`.
     * @param {String} name
     *   The operation name that will be passed.
     * @returns {Promise} - The promise which resolves to an object.
     *   The decoded operation object has result and metadata field to get information from.
     *   Please see the {@link https://github.com/googleapis/gax-nodejs/blob/master/client-libraries.md#long-running-operations | documentation }
     *   for more details and examples.
     * @example <caption>include:samples/generated/v1beta1/service_usage.import_admin_overrides.js</caption>
     * region_tag:serviceusage_v1beta1_generated_ServiceUsage_ImportAdminOverrides_async
     */
    async checkImportAdminOverridesProgress(name) {
        const request = new this._gaxModule.operationsProtos.google.longrunning.GetOperationRequest({ name });
        const [operation] = await this.operationsClient.getOperation(request);
        const decodeOperation = new this._gaxModule.Operation(operation, this.descriptors.longrunning.importAdminOverrides, this._gaxModule.createDefaultBackoffSettings());
        return decodeOperation;
    }
    createConsumerOverride(request, optionsOrCallback, callback) {
        var _a;
        request = request || {};
        let options;
        if (typeof optionsOrCallback === 'function' && callback === undefined) {
            callback = optionsOrCallback;
            options = {};
        }
        else {
            options = optionsOrCallback;
        }
        options = options || {};
        options.otherArgs = options.otherArgs || {};
        options.otherArgs.headers = options.otherArgs.headers || {};
        options.otherArgs.headers['x-goog-request-params'] =
            this._gaxModule.routingHeader.fromParams({
                parent: (_a = request.parent) !== null && _a !== void 0 ? _a : '',
            });
        this.initialize();
        return this.innerApiCalls.createConsumerOverride(request, options, callback);
    }
    /**
     * Check the status of the long running operation returned by `createConsumerOverride()`.
     * @param {String} name
     *   The operation name that will be passed.
     * @returns {Promise} - The promise which resolves to an object.
     *   The decoded operation object has result and metadata field to get information from.
     *   Please see the {@link https://github.com/googleapis/gax-nodejs/blob/master/client-libraries.md#long-running-operations | documentation }
     *   for more details and examples.
     * @example <caption>include:samples/generated/v1beta1/service_usage.create_consumer_override.js</caption>
     * region_tag:serviceusage_v1beta1_generated_ServiceUsage_CreateConsumerOverride_async
     */
    async checkCreateConsumerOverrideProgress(name) {
        const request = new this._gaxModule.operationsProtos.google.longrunning.GetOperationRequest({ name });
        const [operation] = await this.operationsClient.getOperation(request);
        const decodeOperation = new this._gaxModule.Operation(operation, this.descriptors.longrunning.createConsumerOverride, this._gaxModule.createDefaultBackoffSettings());
        return decodeOperation;
    }
    updateConsumerOverride(request, optionsOrCallback, callback) {
        var _a;
        request = request || {};
        let options;
        if (typeof optionsOrCallback === 'function' && callback === undefined) {
            callback = optionsOrCallback;
            options = {};
        }
        else {
            options = optionsOrCallback;
        }
        options = options || {};
        options.otherArgs = options.otherArgs || {};
        options.otherArgs.headers = options.otherArgs.headers || {};
        options.otherArgs.headers['x-goog-request-params'] =
            this._gaxModule.routingHeader.fromParams({
                name: (_a = request.name) !== null && _a !== void 0 ? _a : '',
            });
        this.initialize();
        return this.innerApiCalls.updateConsumerOverride(request, options, callback);
    }
    /**
     * Check the status of the long running operation returned by `updateConsumerOverride()`.
     * @param {String} name
     *   The operation name that will be passed.
     * @returns {Promise} - The promise which resolves to an object.
     *   The decoded operation object has result and metadata field to get information from.
     *   Please see the {@link https://github.com/googleapis/gax-nodejs/blob/master/client-libraries.md#long-running-operations | documentation }
     *   for more details and examples.
     * @example <caption>include:samples/generated/v1beta1/service_usage.update_consumer_override.js</caption>
     * region_tag:serviceusage_v1beta1_generated_ServiceUsage_UpdateConsumerOverride_async
     */
    async checkUpdateConsumerOverrideProgress(name) {
        const request = new this._gaxModule.operationsProtos.google.longrunning.GetOperationRequest({ name });
        const [operation] = await this.operationsClient.getOperation(request);
        const decodeOperation = new this._gaxModule.Operation(operation, this.descriptors.longrunning.updateConsumerOverride, this._gaxModule.createDefaultBackoffSettings());
        return decodeOperation;
    }
    deleteConsumerOverride(request, optionsOrCallback, callback) {
        var _a;
        request = request || {};
        let options;
        if (typeof optionsOrCallback === 'function' && callback === undefined) {
            callback = optionsOrCallback;
            options = {};
        }
        else {
            options = optionsOrCallback;
        }
        options = options || {};
        options.otherArgs = options.otherArgs || {};
        options.otherArgs.headers = options.otherArgs.headers || {};
        options.otherArgs.headers['x-goog-request-params'] =
            this._gaxModule.routingHeader.fromParams({
                name: (_a = request.name) !== null && _a !== void 0 ? _a : '',
            });
        this.initialize();
        return this.innerApiCalls.deleteConsumerOverride(request, options, callback);
    }
    /**
     * Check the status of the long running operation returned by `deleteConsumerOverride()`.
     * @param {String} name
     *   The operation name that will be passed.
     * @returns {Promise} - The promise which resolves to an object.
     *   The decoded operation object has result and metadata field to get information from.
     *   Please see the {@link https://github.com/googleapis/gax-nodejs/blob/master/client-libraries.md#long-running-operations | documentation }
     *   for more details and examples.
     * @example <caption>include:samples/generated/v1beta1/service_usage.delete_consumer_override.js</caption>
     * region_tag:serviceusage_v1beta1_generated_ServiceUsage_DeleteConsumerOverride_async
     */
    async checkDeleteConsumerOverrideProgress(name) {
        const request = new this._gaxModule.operationsProtos.google.longrunning.GetOperationRequest({ name });
        const [operation] = await this.operationsClient.getOperation(request);
        const decodeOperation = new this._gaxModule.Operation(operation, this.descriptors.longrunning.deleteConsumerOverride, this._gaxModule.createDefaultBackoffSettings());
        return decodeOperation;
    }
    importConsumerOverrides(request, optionsOrCallback, callback) {
        var _a;
        request = request || {};
        let options;
        if (typeof optionsOrCallback === 'function' && callback === undefined) {
            callback = optionsOrCallback;
            options = {};
        }
        else {
            options = optionsOrCallback;
        }
        options = options || {};
        options.otherArgs = options.otherArgs || {};
        options.otherArgs.headers = options.otherArgs.headers || {};
        options.otherArgs.headers['x-goog-request-params'] =
            this._gaxModule.routingHeader.fromParams({
                parent: (_a = request.parent) !== null && _a !== void 0 ? _a : '',
            });
        this.initialize();
        return this.innerApiCalls.importConsumerOverrides(request, options, callback);
    }
    /**
     * Check the status of the long running operation returned by `importConsumerOverrides()`.
     * @param {String} name
     *   The operation name that will be passed.
     * @returns {Promise} - The promise which resolves to an object.
     *   The decoded operation object has result and metadata field to get information from.
     *   Please see the {@link https://github.com/googleapis/gax-nodejs/blob/master/client-libraries.md#long-running-operations | documentation }
     *   for more details and examples.
     * @example <caption>include:samples/generated/v1beta1/service_usage.import_consumer_overrides.js</caption>
     * region_tag:serviceusage_v1beta1_generated_ServiceUsage_ImportConsumerOverrides_async
     */
    async checkImportConsumerOverridesProgress(name) {
        const request = new this._gaxModule.operationsProtos.google.longrunning.GetOperationRequest({ name });
        const [operation] = await this.operationsClient.getOperation(request);
        const decodeOperation = new this._gaxModule.Operation(operation, this.descriptors.longrunning.importConsumerOverrides, this._gaxModule.createDefaultBackoffSettings());
        return decodeOperation;
    }
    generateServiceIdentity(request, optionsOrCallback, callback) {
        var _a;
        request = request || {};
        let options;
        if (typeof optionsOrCallback === 'function' && callback === undefined) {
            callback = optionsOrCallback;
            options = {};
        }
        else {
            options = optionsOrCallback;
        }
        options = options || {};
        options.otherArgs = options.otherArgs || {};
        options.otherArgs.headers = options.otherArgs.headers || {};
        options.otherArgs.headers['x-goog-request-params'] =
            this._gaxModule.routingHeader.fromParams({
                parent: (_a = request.parent) !== null && _a !== void 0 ? _a : '',
            });
        this.initialize();
        return this.innerApiCalls.generateServiceIdentity(request, options, callback);
    }
    /**
     * Check the status of the long running operation returned by `generateServiceIdentity()`.
     * @param {String} name
     *   The operation name that will be passed.
     * @returns {Promise} - The promise which resolves to an object.
     *   The decoded operation object has result and metadata field to get information from.
     *   Please see the {@link https://github.com/googleapis/gax-nodejs/blob/master/client-libraries.md#long-running-operations | documentation }
     *   for more details and examples.
     * @example <caption>include:samples/generated/v1beta1/service_usage.generate_service_identity.js</caption>
     * region_tag:serviceusage_v1beta1_generated_ServiceUsage_GenerateServiceIdentity_async
     */
    async checkGenerateServiceIdentityProgress(name) {
        const request = new this._gaxModule.operationsProtos.google.longrunning.GetOperationRequest({ name });
        const [operation] = await this.operationsClient.getOperation(request);
        const decodeOperation = new this._gaxModule.Operation(operation, this.descriptors.longrunning.generateServiceIdentity, this._gaxModule.createDefaultBackoffSettings());
        return decodeOperation;
    }
    listServices(request, optionsOrCallback, callback) {
        var _a;
        request = request || {};
        let options;
        if (typeof optionsOrCallback === 'function' && callback === undefined) {
            callback = optionsOrCallback;
            options = {};
        }
        else {
            options = optionsOrCallback;
        }
        options = options || {};
        options.otherArgs = options.otherArgs || {};
        options.otherArgs.headers = options.otherArgs.headers || {};
        options.otherArgs.headers['x-goog-request-params'] =
            this._gaxModule.routingHeader.fromParams({
                parent: (_a = request.parent) !== null && _a !== void 0 ? _a : '',
            });
        this.initialize();
        this.warn('DEP$ServiceUsage-$ListServices', 'ListServices is deprecated and may be removed in a future version.', 'DeprecationWarning');
        return this.innerApiCalls.listServices(request, options, callback);
    }
    /**
     * Equivalent to `method.name.toCamelCase()`, but returns a NodeJS Stream object.
     * @param {Object} request
     *   The request object that will be sent.
     * @param {string} request.parent
     *   Parent to search for services on.
     *
     *   An example name would be:
     *   `projects/123`
     *   where `123` is the project number (not project ID).
     * @param {number} request.pageSize
     *   Requested size of the next page of data.
     *   Requested page size cannot exceed 200.
     *    If not set, the default page size is 50.
     * @param {string} request.pageToken
     *   Token identifying which result to start with, which is returned by a
     *   previous list call.
     * @param {string} request.filter
     *   Only list services that conform to the given filter.
     *   The allowed filter strings are `state:ENABLED` and `state:DISABLED`.
     * @param {object} [options]
     *   Call options. See {@link https://googleapis.dev/nodejs/google-gax/latest/interfaces/CallOptions.html|CallOptions} for more details.
     * @returns {Stream}
     *   An object stream which emits an object representing {@link protos.google.api.serviceusage.v1beta1.Service|Service} on 'data' event.
     *   The client library will perform auto-pagination by default: it will call the API as many
     *   times as needed. Note that it can affect your quota.
     *   We recommend using `listServicesAsync()`
     *   method described below for async iteration which you can stop as needed.
     *   Please see the {@link https://github.com/googleapis/gax-nodejs/blob/master/client-libraries.md#auto-pagination | documentation }
     *   for more details and examples.
     * @deprecated ListServices is deprecated and may be removed in a future version.
     */
    listServicesStream(request, options) {
        var _a;
        request = request || {};
        options = options || {};
        options.otherArgs = options.otherArgs || {};
        options.otherArgs.headers = options.otherArgs.headers || {};
        options.otherArgs.headers['x-goog-request-params'] =
            this._gaxModule.routingHeader.fromParams({
                parent: (_a = request.parent) !== null && _a !== void 0 ? _a : '',
            });
        const defaultCallSettings = this._defaults['listServices'];
        const callSettings = defaultCallSettings.merge(options);
        this.initialize();
        this.warn('DEP$ServiceUsage-$ListServices', 'ListServices is deprecated and may be removed in a future version.', 'DeprecationWarning');
        return this.descriptors.page.listServices.createStream(this.innerApiCalls.listServices, request, callSettings);
    }
    /**
     * Equivalent to `listServices`, but returns an iterable object.
     *
     * `for`-`await`-`of` syntax is used with the iterable to get response elements on-demand.
     * @param {Object} request
     *   The request object that will be sent.
     * @param {string} request.parent
     *   Parent to search for services on.
     *
     *   An example name would be:
     *   `projects/123`
     *   where `123` is the project number (not project ID).
     * @param {number} request.pageSize
     *   Requested size of the next page of data.
     *   Requested page size cannot exceed 200.
     *    If not set, the default page size is 50.
     * @param {string} request.pageToken
     *   Token identifying which result to start with, which is returned by a
     *   previous list call.
     * @param {string} request.filter
     *   Only list services that conform to the given filter.
     *   The allowed filter strings are `state:ENABLED` and `state:DISABLED`.
     * @param {object} [options]
     *   Call options. See {@link https://googleapis.dev/nodejs/google-gax/latest/interfaces/CallOptions.html|CallOptions} for more details.
     * @returns {Object}
     *   An iterable Object that allows {@link https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Iteration_protocols | async iteration }.
     *   When you iterate the returned iterable, each element will be an object representing
     *   {@link protos.google.api.serviceusage.v1beta1.Service|Service}. The API will be called under the hood as needed, once per the page,
     *   so you can stop the iteration when you don't need more results.
     *   Please see the {@link https://github.com/googleapis/gax-nodejs/blob/master/client-libraries.md#auto-pagination | documentation }
     *   for more details and examples.
     * @example <caption>include:samples/generated/v1beta1/service_usage.list_services.js</caption>
     * region_tag:serviceusage_v1beta1_generated_ServiceUsage_ListServices_async
     * @deprecated ListServices is deprecated and may be removed in a future version.
     */
    listServicesAsync(request, options) {
        var _a;
        request = request || {};
        options = options || {};
        options.otherArgs = options.otherArgs || {};
        options.otherArgs.headers = options.otherArgs.headers || {};
        options.otherArgs.headers['x-goog-request-params'] =
            this._gaxModule.routingHeader.fromParams({
                parent: (_a = request.parent) !== null && _a !== void 0 ? _a : '',
            });
        const defaultCallSettings = this._defaults['listServices'];
        const callSettings = defaultCallSettings.merge(options);
        this.initialize();
        this.warn('DEP$ServiceUsage-$ListServices', 'ListServices is deprecated and may be removed in a future version.', 'DeprecationWarning');
        return this.descriptors.page.listServices.asyncIterate(this.innerApiCalls['listServices'], request, callSettings);
    }
    listConsumerQuotaMetrics(request, optionsOrCallback, callback) {
        var _a;
        request = request || {};
        let options;
        if (typeof optionsOrCallback === 'function' && callback === undefined) {
            callback = optionsOrCallback;
            options = {};
        }
        else {
            options = optionsOrCallback;
        }
        options = options || {};
        options.otherArgs = options.otherArgs || {};
        options.otherArgs.headers = options.otherArgs.headers || {};
        options.otherArgs.headers['x-goog-request-params'] =
            this._gaxModule.routingHeader.fromParams({
                parent: (_a = request.parent) !== null && _a !== void 0 ? _a : '',
            });
        this.initialize();
        return this.innerApiCalls.listConsumerQuotaMetrics(request, options, callback);
    }
    /**
     * Equivalent to `method.name.toCamelCase()`, but returns a NodeJS Stream object.
     * @param {Object} request
     *   The request object that will be sent.
     * @param {string} request.parent
     *   Parent of the quotas resource.
     *
     *   Some example names would be:
     *   `projects/123/services/serviceconsumermanagement.googleapis.com`
     *   `folders/345/services/serviceconsumermanagement.googleapis.com`
     *   `organizations/456/services/serviceconsumermanagement.googleapis.com`
     * @param {number} request.pageSize
     *   Requested size of the next page of data.
     * @param {string} request.pageToken
     *   Token identifying which result to start with; returned by a previous list
     *   call.
     * @param {google.api.serviceusage.v1beta1.QuotaView} request.view
     *   Specifies the level of detail for quota information in the response.
     * @param {object} [options]
     *   Call options. See {@link https://googleapis.dev/nodejs/google-gax/latest/interfaces/CallOptions.html|CallOptions} for more details.
     * @returns {Stream}
     *   An object stream which emits an object representing {@link protos.google.api.serviceusage.v1beta1.ConsumerQuotaMetric|ConsumerQuotaMetric} on 'data' event.
     *   The client library will perform auto-pagination by default: it will call the API as many
     *   times as needed. Note that it can affect your quota.
     *   We recommend using `listConsumerQuotaMetricsAsync()`
     *   method described below for async iteration which you can stop as needed.
     *   Please see the {@link https://github.com/googleapis/gax-nodejs/blob/master/client-libraries.md#auto-pagination | documentation }
     *   for more details and examples.
     */
    listConsumerQuotaMetricsStream(request, options) {
        var _a;
        request = request || {};
        options = options || {};
        options.otherArgs = options.otherArgs || {};
        options.otherArgs.headers = options.otherArgs.headers || {};
        options.otherArgs.headers['x-goog-request-params'] =
            this._gaxModule.routingHeader.fromParams({
                parent: (_a = request.parent) !== null && _a !== void 0 ? _a : '',
            });
        const defaultCallSettings = this._defaults['listConsumerQuotaMetrics'];
        const callSettings = defaultCallSettings.merge(options);
        this.initialize();
        return this.descriptors.page.listConsumerQuotaMetrics.createStream(this.innerApiCalls.listConsumerQuotaMetrics, request, callSettings);
    }
    /**
     * Equivalent to `listConsumerQuotaMetrics`, but returns an iterable object.
     *
     * `for`-`await`-`of` syntax is used with the iterable to get response elements on-demand.
     * @param {Object} request
     *   The request object that will be sent.
     * @param {string} request.parent
     *   Parent of the quotas resource.
     *
     *   Some example names would be:
     *   `projects/123/services/serviceconsumermanagement.googleapis.com`
     *   `folders/345/services/serviceconsumermanagement.googleapis.com`
     *   `organizations/456/services/serviceconsumermanagement.googleapis.com`
     * @param {number} request.pageSize
     *   Requested size of the next page of data.
     * @param {string} request.pageToken
     *   Token identifying which result to start with; returned by a previous list
     *   call.
     * @param {google.api.serviceusage.v1beta1.QuotaView} request.view
     *   Specifies the level of detail for quota information in the response.
     * @param {object} [options]
     *   Call options. See {@link https://googleapis.dev/nodejs/google-gax/latest/interfaces/CallOptions.html|CallOptions} for more details.
     * @returns {Object}
     *   An iterable Object that allows {@link https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Iteration_protocols | async iteration }.
     *   When you iterate the returned iterable, each element will be an object representing
     *   {@link protos.google.api.serviceusage.v1beta1.ConsumerQuotaMetric|ConsumerQuotaMetric}. The API will be called under the hood as needed, once per the page,
     *   so you can stop the iteration when you don't need more results.
     *   Please see the {@link https://github.com/googleapis/gax-nodejs/blob/master/client-libraries.md#auto-pagination | documentation }
     *   for more details and examples.
     * @example <caption>include:samples/generated/v1beta1/service_usage.list_consumer_quota_metrics.js</caption>
     * region_tag:serviceusage_v1beta1_generated_ServiceUsage_ListConsumerQuotaMetrics_async
     */
    listConsumerQuotaMetricsAsync(request, options) {
        var _a;
        request = request || {};
        options = options || {};
        options.otherArgs = options.otherArgs || {};
        options.otherArgs.headers = options.otherArgs.headers || {};
        options.otherArgs.headers['x-goog-request-params'] =
            this._gaxModule.routingHeader.fromParams({
                parent: (_a = request.parent) !== null && _a !== void 0 ? _a : '',
            });
        const defaultCallSettings = this._defaults['listConsumerQuotaMetrics'];
        const callSettings = defaultCallSettings.merge(options);
        this.initialize();
        return this.descriptors.page.listConsumerQuotaMetrics.asyncIterate(this.innerApiCalls['listConsumerQuotaMetrics'], request, callSettings);
    }
    listAdminOverrides(request, optionsOrCallback, callback) {
        var _a;
        request = request || {};
        let options;
        if (typeof optionsOrCallback === 'function' && callback === undefined) {
            callback = optionsOrCallback;
            options = {};
        }
        else {
            options = optionsOrCallback;
        }
        options = options || {};
        options.otherArgs = options.otherArgs || {};
        options.otherArgs.headers = options.otherArgs.headers || {};
        options.otherArgs.headers['x-goog-request-params'] =
            this._gaxModule.routingHeader.fromParams({
                parent: (_a = request.parent) !== null && _a !== void 0 ? _a : '',
            });
        this.initialize();
        return this.innerApiCalls.listAdminOverrides(request, options, callback);
    }
    /**
     * Equivalent to `method.name.toCamelCase()`, but returns a NodeJS Stream object.
     * @param {Object} request
     *   The request object that will be sent.
     * @param {string} request.parent
     *   The resource name of the parent quota limit, returned by a
     *   ListConsumerQuotaMetrics or GetConsumerQuotaMetric call.
     *
     *   An example name would be:
     *   `projects/123/services/compute.googleapis.com/consumerQuotaMetrics/compute.googleapis.com%2Fcpus/limits/%2Fproject%2Fregion`
     * @param {number} request.pageSize
     *   Requested size of the next page of data.
     * @param {string} request.pageToken
     *   Token identifying which result to start with; returned by a previous list
     *   call.
     * @param {object} [options]
     *   Call options. See {@link https://googleapis.dev/nodejs/google-gax/latest/interfaces/CallOptions.html|CallOptions} for more details.
     * @returns {Stream}
     *   An object stream which emits an object representing {@link protos.google.api.serviceusage.v1beta1.QuotaOverride|QuotaOverride} on 'data' event.
     *   The client library will perform auto-pagination by default: it will call the API as many
     *   times as needed. Note that it can affect your quota.
     *   We recommend using `listAdminOverridesAsync()`
     *   method described below for async iteration which you can stop as needed.
     *   Please see the {@link https://github.com/googleapis/gax-nodejs/blob/master/client-libraries.md#auto-pagination | documentation }
     *   for more details and examples.
     */
    listAdminOverridesStream(request, options) {
        var _a;
        request = request || {};
        options = options || {};
        options.otherArgs = options.otherArgs || {};
        options.otherArgs.headers = options.otherArgs.headers || {};
        options.otherArgs.headers['x-goog-request-params'] =
            this._gaxModule.routingHeader.fromParams({
                parent: (_a = request.parent) !== null && _a !== void 0 ? _a : '',
            });
        const defaultCallSettings = this._defaults['listAdminOverrides'];
        const callSettings = defaultCallSettings.merge(options);
        this.initialize();
        return this.descriptors.page.listAdminOverrides.createStream(this.innerApiCalls.listAdminOverrides, request, callSettings);
    }
    /**
     * Equivalent to `listAdminOverrides`, but returns an iterable object.
     *
     * `for`-`await`-`of` syntax is used with the iterable to get response elements on-demand.
     * @param {Object} request
     *   The request object that will be sent.
     * @param {string} request.parent
     *   The resource name of the parent quota limit, returned by a
     *   ListConsumerQuotaMetrics or GetConsumerQuotaMetric call.
     *
     *   An example name would be:
     *   `projects/123/services/compute.googleapis.com/consumerQuotaMetrics/compute.googleapis.com%2Fcpus/limits/%2Fproject%2Fregion`
     * @param {number} request.pageSize
     *   Requested size of the next page of data.
     * @param {string} request.pageToken
     *   Token identifying which result to start with; returned by a previous list
     *   call.
     * @param {object} [options]
     *   Call options. See {@link https://googleapis.dev/nodejs/google-gax/latest/interfaces/CallOptions.html|CallOptions} for more details.
     * @returns {Object}
     *   An iterable Object that allows {@link https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Iteration_protocols | async iteration }.
     *   When you iterate the returned iterable, each element will be an object representing
     *   {@link protos.google.api.serviceusage.v1beta1.QuotaOverride|QuotaOverride}. The API will be called under the hood as needed, once per the page,
     *   so you can stop the iteration when you don't need more results.
     *   Please see the {@link https://github.com/googleapis/gax-nodejs/blob/master/client-libraries.md#auto-pagination | documentation }
     *   for more details and examples.
     * @example <caption>include:samples/generated/v1beta1/service_usage.list_admin_overrides.js</caption>
     * region_tag:serviceusage_v1beta1_generated_ServiceUsage_ListAdminOverrides_async
     */
    listAdminOverridesAsync(request, options) {
        var _a;
        request = request || {};
        options = options || {};
        options.otherArgs = options.otherArgs || {};
        options.otherArgs.headers = options.otherArgs.headers || {};
        options.otherArgs.headers['x-goog-request-params'] =
            this._gaxModule.routingHeader.fromParams({
                parent: (_a = request.parent) !== null && _a !== void 0 ? _a : '',
            });
        const defaultCallSettings = this._defaults['listAdminOverrides'];
        const callSettings = defaultCallSettings.merge(options);
        this.initialize();
        return this.descriptors.page.listAdminOverrides.asyncIterate(this.innerApiCalls['listAdminOverrides'], request, callSettings);
    }
    listConsumerOverrides(request, optionsOrCallback, callback) {
        var _a;
        request = request || {};
        let options;
        if (typeof optionsOrCallback === 'function' && callback === undefined) {
            callback = optionsOrCallback;
            options = {};
        }
        else {
            options = optionsOrCallback;
        }
        options = options || {};
        options.otherArgs = options.otherArgs || {};
        options.otherArgs.headers = options.otherArgs.headers || {};
        options.otherArgs.headers['x-goog-request-params'] =
            this._gaxModule.routingHeader.fromParams({
                parent: (_a = request.parent) !== null && _a !== void 0 ? _a : '',
            });
        this.initialize();
        return this.innerApiCalls.listConsumerOverrides(request, options, callback);
    }
    /**
     * Equivalent to `method.name.toCamelCase()`, but returns a NodeJS Stream object.
     * @param {Object} request
     *   The request object that will be sent.
     * @param {string} request.parent
     *   The resource name of the parent quota limit, returned by a
     *   ListConsumerQuotaMetrics or GetConsumerQuotaMetric call.
     *
     *   An example name would be:
     *   `projects/123/services/compute.googleapis.com/consumerQuotaMetrics/compute.googleapis.com%2Fcpus/limits/%2Fproject%2Fregion`
     * @param {number} request.pageSize
     *   Requested size of the next page of data.
     * @param {string} request.pageToken
     *   Token identifying which result to start with; returned by a previous list
     *   call.
     * @param {object} [options]
     *   Call options. See {@link https://googleapis.dev/nodejs/google-gax/latest/interfaces/CallOptions.html|CallOptions} for more details.
     * @returns {Stream}
     *   An object stream which emits an object representing {@link protos.google.api.serviceusage.v1beta1.QuotaOverride|QuotaOverride} on 'data' event.
     *   The client library will perform auto-pagination by default: it will call the API as many
     *   times as needed. Note that it can affect your quota.
     *   We recommend using `listConsumerOverridesAsync()`
     *   method described below for async iteration which you can stop as needed.
     *   Please see the {@link https://github.com/googleapis/gax-nodejs/blob/master/client-libraries.md#auto-pagination | documentation }
     *   for more details and examples.
     */
    listConsumerOverridesStream(request, options) {
        var _a;
        request = request || {};
        options = options || {};
        options.otherArgs = options.otherArgs || {};
        options.otherArgs.headers = options.otherArgs.headers || {};
        options.otherArgs.headers['x-goog-request-params'] =
            this._gaxModule.routingHeader.fromParams({
                parent: (_a = request.parent) !== null && _a !== void 0 ? _a : '',
            });
        const defaultCallSettings = this._defaults['listConsumerOverrides'];
        const callSettings = defaultCallSettings.merge(options);
        this.initialize();
        return this.descriptors.page.listConsumerOverrides.createStream(this.innerApiCalls.listConsumerOverrides, request, callSettings);
    }
    /**
     * Equivalent to `listConsumerOverrides`, but returns an iterable object.
     *
     * `for`-`await`-`of` syntax is used with the iterable to get response elements on-demand.
     * @param {Object} request
     *   The request object that will be sent.
     * @param {string} request.parent
     *   The resource name of the parent quota limit, returned by a
     *   ListConsumerQuotaMetrics or GetConsumerQuotaMetric call.
     *
     *   An example name would be:
     *   `projects/123/services/compute.googleapis.com/consumerQuotaMetrics/compute.googleapis.com%2Fcpus/limits/%2Fproject%2Fregion`
     * @param {number} request.pageSize
     *   Requested size of the next page of data.
     * @param {string} request.pageToken
     *   Token identifying which result to start with; returned by a previous list
     *   call.
     * @param {object} [options]
     *   Call options. See {@link https://googleapis.dev/nodejs/google-gax/latest/interfaces/CallOptions.html|CallOptions} for more details.
     * @returns {Object}
     *   An iterable Object that allows {@link https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Iteration_protocols | async iteration }.
     *   When you iterate the returned iterable, each element will be an object representing
     *   {@link protos.google.api.serviceusage.v1beta1.QuotaOverride|QuotaOverride}. The API will be called under the hood as needed, once per the page,
     *   so you can stop the iteration when you don't need more results.
     *   Please see the {@link https://github.com/googleapis/gax-nodejs/blob/master/client-libraries.md#auto-pagination | documentation }
     *   for more details and examples.
     * @example <caption>include:samples/generated/v1beta1/service_usage.list_consumer_overrides.js</caption>
     * region_tag:serviceusage_v1beta1_generated_ServiceUsage_ListConsumerOverrides_async
     */
    listConsumerOverridesAsync(request, options) {
        var _a;
        request = request || {};
        options = options || {};
        options.otherArgs = options.otherArgs || {};
        options.otherArgs.headers = options.otherArgs.headers || {};
        options.otherArgs.headers['x-goog-request-params'] =
            this._gaxModule.routingHeader.fromParams({
                parent: (_a = request.parent) !== null && _a !== void 0 ? _a : '',
            });
        const defaultCallSettings = this._defaults['listConsumerOverrides'];
        const callSettings = defaultCallSettings.merge(options);
        this.initialize();
        return this.descriptors.page.listConsumerOverrides.asyncIterate(this.innerApiCalls['listConsumerOverrides'], request, callSettings);
    }
    /**
     * Gets the latest state of a long-running operation.  Clients can use this
     * method to poll the operation result at intervals as recommended by the API
     * service.
     *
     * @param {Object} request - The request object that will be sent.
     * @param {string} request.name - The name of the operation resource.
     * @param {Object=} options
     *   Optional parameters. You can override the default settings for this call,
     *   e.g, timeout, retries, paginations, etc. See {@link
     *   https://googleapis.github.io/gax-nodejs/global.html#CallOptions | gax.CallOptions}
     *   for the details.
     * @param {function(?Error, ?Object)=} callback
     *   The function which will be called with the result of the API call.
     *
     *   The second parameter to the callback is an object representing
     *   {@link google.longrunning.Operation | google.longrunning.Operation}.
     * @return {Promise} - The promise which resolves to an array.
     *   The first element of the array is an object representing
     * {@link google.longrunning.Operation | google.longrunning.Operation}.
     * The promise has a method named "cancel" which cancels the ongoing API call.
     *
     * @example
     * ```
     * const client = longrunning.operationsClient();
     * const name = '';
     * const [response] = await client.getOperation({name});
     * // doThingsWith(response)
     * ```
     */
    getOperation(request, options, callback) {
        return this.operationsClient.getOperation(request, options, callback);
    }
    /**
     * Lists operations that match the specified filter in the request. If the
     * server doesn't support this method, it returns `UNIMPLEMENTED`. Returns an iterable object.
     *
     * For-await-of syntax is used with the iterable to recursively get response element on-demand.
     *
     * @param {Object} request - The request object that will be sent.
     * @param {string} request.name - The name of the operation collection.
     * @param {string} request.filter - The standard list filter.
     * @param {number=} request.pageSize -
     *   The maximum number of resources contained in the underlying API
     *   response. If page streaming is performed per-resource, this
     *   parameter does not affect the return value. If page streaming is
     *   performed per-page, this determines the maximum number of
     *   resources in a page.
     * @param {Object=} options
     *   Optional parameters. You can override the default settings for this call,
     *   e.g, timeout, retries, paginations, etc. See {@link
     *   https://googleapis.github.io/gax-nodejs/global.html#CallOptions | gax.CallOptions} for the
     *   details.
     * @returns {Object}
     *   An iterable Object that conforms to {@link https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Iteration_protocols | iteration protocols}.
     *
     * @example
     * ```
     * const client = longrunning.operationsClient();
     * for await (const response of client.listOperationsAsync(request));
     * // doThingsWith(response)
     * ```
     */
    listOperationsAsync(request, options) {
        return this.operationsClient.listOperationsAsync(request, options);
    }
    /**
     * Starts asynchronous cancellation on a long-running operation.  The server
     * makes a best effort to cancel the operation, but success is not
     * guaranteed.  If the server doesn't support this method, it returns
     * `google.rpc.Code.UNIMPLEMENTED`.  Clients can use
     * {@link Operations.GetOperation} or
     * other methods to check whether the cancellation succeeded or whether the
     * operation completed despite cancellation. On successful cancellation,
     * the operation is not deleted; instead, it becomes an operation with
     * an {@link Operation.error} value with a {@link google.rpc.Status.code} of
     * 1, corresponding to `Code.CANCELLED`.
     *
     * @param {Object} request - The request object that will be sent.
     * @param {string} request.name - The name of the operation resource to be cancelled.
     * @param {Object=} options
     *   Optional parameters. You can override the default settings for this call,
     * e.g, timeout, retries, paginations, etc. See {@link
     * https://googleapis.github.io/gax-nodejs/global.html#CallOptions | gax.CallOptions} for the
     * details.
     * @param {function(?Error)=} callback
     *   The function which will be called with the result of the API call.
     * @return {Promise} - The promise which resolves when API call finishes.
     *   The promise has a method named "cancel" which cancels the ongoing API
     * call.
     *
     * @example
     * ```
     * const client = longrunning.operationsClient();
     * await client.cancelOperation({name: ''});
     * ```
     */
    cancelOperation(request, options, callback) {
        return this.operationsClient.cancelOperation(request, options, callback);
    }
    /**
     * Deletes a long-running operation. This method indicates that the client is
     * no longer interested in the operation result. It does not cancel the
     * operation. If the server doesn't support this method, it returns
     * `google.rpc.Code.UNIMPLEMENTED`.
     *
     * @param {Object} request - The request object that will be sent.
     * @param {string} request.name - The name of the operation resource to be deleted.
     * @param {Object=} options
     *   Optional parameters. You can override the default settings for this call,
     * e.g, timeout, retries, paginations, etc. See {@link
     * https://googleapis.github.io/gax-nodejs/global.html#CallOptions | gax.CallOptions}
     * for the details.
     * @param {function(?Error)=} callback
     *   The function which will be called with the result of the API call.
     * @return {Promise} - The promise which resolves when API call finishes.
     *   The promise has a method named "cancel" which cancels the ongoing API
     * call.
     *
     * @example
     * ```
     * const client = longrunning.operationsClient();
     * await client.deleteOperation({name: ''});
     * ```
     */
    deleteOperation(request, options, callback) {
        return this.operationsClient.deleteOperation(request, options, callback);
    }
    /**
     * Terminate the gRPC channel and close the client.
     *
     * The client will no longer be usable and all future behavior is undefined.
     * @returns {Promise} A promise that resolves when the client is closed.
     */
    close() {
        if (this.serviceUsageStub && !this._terminated) {
            return this.serviceUsageStub.then(stub => {
                this._terminated = true;
                stub.close();
                this.operationsClient.close();
            });
        }
        return Promise.resolve();
    }
}
exports.ServiceUsageClient = ServiceUsageClient;
