// @azure/arm-hybridkubernetes/dist-esm/src/models/index.js
/*
 * Copyright (c) Microsoft Corporation.
 * Licensed under the MIT License.
 *
 * Code generated by Microsoft (R) AutoRest Code Generator.
 * Changes may cause incorrect behavior and will be lost if the code is regenerated.
 */
/** Known values of {@link ProvisioningState} that the service accepts. */
export var KnownProvisioningState;
(function (KnownProvisioningState) {
    /** Succeeded */
    KnownProvisioningState["Succeeded"] = "Succeeded";
    /** Failed */
    KnownProvisioningState["Failed"] = "Failed";
    /** Canceled */
    KnownProvisioningState["Canceled"] = "Canceled";
    /** Provisioning */
    KnownProvisioningState["Provisioning"] = "Provisioning";
    /** Updating */
    KnownProvisioningState["Updating"] = "Updating";
    /** Deleting */
    KnownProvisioningState["Deleting"] = "Deleting";
    /** Accepted */
    KnownProvisioningState["Accepted"] = "Accepted";
})(KnownProvisioningState || (KnownProvisioningState = {}));
/** Known values of {@link ConnectivityStatus} that the service accepts. */
export var KnownConnectivityStatus;
(function (KnownConnectivityStatus) {
    /** Connecting */
    KnownConnectivityStatus["Connecting"] = "Connecting";
    /** Connected */
    KnownConnectivityStatus["Connected"] = "Connected";
    /** Offline */
    KnownConnectivityStatus["Offline"] = "Offline";
    /** Expired */
    KnownConnectivityStatus["Expired"] = "Expired";
})(KnownConnectivityStatus || (KnownConnectivityStatus = {}));
/** Known values of {@link CreatedByType} that the service accepts. */
export var KnownCreatedByType;
(function (KnownCreatedByType) {
    /** User */
    KnownCreatedByType["User"] = "User";
    /** Application */
    KnownCreatedByType["Application"] = "Application";
    /** ManagedIdentity */
    KnownCreatedByType["ManagedIdentity"] = "ManagedIdentity";
    /** Key */
    KnownCreatedByType["Key"] = "Key";
})(KnownCreatedByType || (KnownCreatedByType = {}));
/** Known values of {@link LastModifiedByType} that the service accepts. */
export var KnownLastModifiedByType;
(function (KnownLastModifiedByType) {
    /** User */
    KnownLastModifiedByType["User"] = "User";
    /** Application */
    KnownLastModifiedByType["Application"] = "Application";
    /** ManagedIdentity */
    KnownLastModifiedByType["ManagedIdentity"] = "ManagedIdentity";
    /** Key */
    KnownLastModifiedByType["Key"] = "Key";
})(KnownLastModifiedByType || (KnownLastModifiedByType = {}));
/** Known values of {@link AuthenticationMethod} that the service accepts. */
export var KnownAuthenticationMethod;
(function (KnownAuthenticationMethod) {
    /** Token */
    KnownAuthenticationMethod["Token"] = "Token";
    /** AAD */
    KnownAuthenticationMethod["AAD"] = "AAD";
})(KnownAuthenticationMethod || (KnownAuthenticationMethod = {}));
