// @azure/arm-streamanalytics/dist-esm/src/models/index.js
/*
 * Copyright (c) Microsoft Corporation.
 * Licensed under the MIT License.
 *
 * Code generated by Microsoft (R) AutoRest Code Generator.
 * Changes may cause incorrect behavior and will be lost if the code is regenerated.
 */
/** Known values of {@link SkuName} that the service accepts. */
export var KnownSkuName;
(function (KnownSkuName) {
    /** Standard */
    KnownSkuName["Standard"] = "Standard";
})(KnownSkuName || (KnownSkuName = {}));
/** Known values of {@link JobType} that the service accepts. */
export var KnownJobType;
(function (KnownJobType) {
    /** Cloud */
    KnownJobType["Cloud"] = "Cloud";
    /** Edge */
    KnownJobType["Edge"] = "Edge";
})(KnownJobType || (KnownJobType = {}));
/** Known values of {@link OutputStartMode} that the service accepts. */
export var KnownOutputStartMode;
(function (KnownOutputStartMode) {
    /** JobStartTime */
    KnownOutputStartMode["JobStartTime"] = "JobStartTime";
    /** CustomTime */
    KnownOutputStartMode["CustomTime"] = "CustomTime";
    /** LastOutputEventTime */
    KnownOutputStartMode["LastOutputEventTime"] = "LastOutputEventTime";
})(KnownOutputStartMode || (KnownOutputStartMode = {}));
/** Known values of {@link EventsOutOfOrderPolicy} that the service accepts. */
export var KnownEventsOutOfOrderPolicy;
(function (KnownEventsOutOfOrderPolicy) {
    /** Adjust */
    KnownEventsOutOfOrderPolicy["Adjust"] = "Adjust";
    /** Drop */
    KnownEventsOutOfOrderPolicy["Drop"] = "Drop";
})(KnownEventsOutOfOrderPolicy || (KnownEventsOutOfOrderPolicy = {}));
/** Known values of {@link OutputErrorPolicy} that the service accepts. */
export var KnownOutputErrorPolicy;
(function (KnownOutputErrorPolicy) {
    /** Stop */
    KnownOutputErrorPolicy["Stop"] = "Stop";
    /** Drop */
    KnownOutputErrorPolicy["Drop"] = "Drop";
})(KnownOutputErrorPolicy || (KnownOutputErrorPolicy = {}));
/** Known values of {@link CompatibilityLevel} that the service accepts. */
export var KnownCompatibilityLevel;
(function (KnownCompatibilityLevel) {
    /** One0 */
    KnownCompatibilityLevel["One0"] = "1.0";
    /** One2 */
    KnownCompatibilityLevel["One2"] = "1.2";
})(KnownCompatibilityLevel || (KnownCompatibilityLevel = {}));
/** Known values of {@link EventSerializationType} that the service accepts. */
export var KnownEventSerializationType;
(function (KnownEventSerializationType) {
    /** Csv */
    KnownEventSerializationType["Csv"] = "Csv";
    /** Avro */
    KnownEventSerializationType["Avro"] = "Avro";
    /** Json */
    KnownEventSerializationType["Json"] = "Json";
    /** Parquet */
    KnownEventSerializationType["Parquet"] = "Parquet";
})(KnownEventSerializationType || (KnownEventSerializationType = {}));
/** Known values of {@link CompressionType} that the service accepts. */
export var KnownCompressionType;
(function (KnownCompressionType) {
    /** None */
    KnownCompressionType["None"] = "None";
    /** GZip */
    KnownCompressionType["GZip"] = "GZip";
    /** Deflate */
    KnownCompressionType["Deflate"] = "Deflate";
})(KnownCompressionType || (KnownCompressionType = {}));
/** Known values of {@link AuthenticationMode} that the service accepts. */
export var KnownAuthenticationMode;
(function (KnownAuthenticationMode) {
    /** Msi */
    KnownAuthenticationMode["Msi"] = "Msi";
    /** UserToken */
    KnownAuthenticationMode["UserToken"] = "UserToken";
    /** ConnectionString */
    KnownAuthenticationMode["ConnectionString"] = "ConnectionString";
})(KnownAuthenticationMode || (KnownAuthenticationMode = {}));
/** Known values of {@link ContentStoragePolicy} that the service accepts. */
export var KnownContentStoragePolicy;
(function (KnownContentStoragePolicy) {
    /** SystemAccount */
    KnownContentStoragePolicy["SystemAccount"] = "SystemAccount";
    /** JobStorageAccount */
    KnownContentStoragePolicy["JobStorageAccount"] = "JobStorageAccount";
})(KnownContentStoragePolicy || (KnownContentStoragePolicy = {}));
/** Known values of {@link ClusterSkuName} that the service accepts. */
export var KnownClusterSkuName;
(function (KnownClusterSkuName) {
    /** The default SKU. */
    KnownClusterSkuName["Default"] = "Default";
})(KnownClusterSkuName || (KnownClusterSkuName = {}));
/** Known values of {@link ClusterProvisioningState} that the service accepts. */
export var KnownClusterProvisioningState;
(function (KnownClusterProvisioningState) {
    /** The cluster provisioning succeeded. */
    KnownClusterProvisioningState["Succeeded"] = "Succeeded";
    /** The cluster provisioning failed. */
    KnownClusterProvisioningState["Failed"] = "Failed";
    /** The cluster provisioning was canceled. */
    KnownClusterProvisioningState["Canceled"] = "Canceled";
    /** The cluster provisioning was inprogress. */
    KnownClusterProvisioningState["InProgress"] = "InProgress";
})(KnownClusterProvisioningState || (KnownClusterProvisioningState = {}));
/** Known values of {@link JobState} that the service accepts. */
export var KnownJobState;
(function (KnownJobState) {
    /** The job is currently in the Created state. */
    KnownJobState["Created"] = "Created";
    /** The job is currently in the Starting state. */
    KnownJobState["Starting"] = "Starting";
    /** The job is currently in the Running state. */
    KnownJobState["Running"] = "Running";
    /** The job is currently in the Stopping state. */
    KnownJobState["Stopping"] = "Stopping";
    /** The job is currently in the Stopped state. */
    KnownJobState["Stopped"] = "Stopped";
    /** The job is currently in the Deleting state. */
    KnownJobState["Deleting"] = "Deleting";
    /** The job is currently in the Failed state. */
    KnownJobState["Failed"] = "Failed";
    /** The job is currently in the Degraded state. */
    KnownJobState["Degraded"] = "Degraded";
    /** The job is currently in the Restarting state. */
    KnownJobState["Restarting"] = "Restarting";
    /** The job is currently in the Scaling state. */
    KnownJobState["Scaling"] = "Scaling";
})(KnownJobState || (KnownJobState = {}));
/** Known values of {@link Encoding} that the service accepts. */
export var KnownEncoding;
(function (KnownEncoding) {
    /** UTF8 */
    KnownEncoding["UTF8"] = "UTF8";
})(KnownEncoding || (KnownEncoding = {}));
/** Known values of {@link JsonOutputSerializationFormat} that the service accepts. */
export var KnownJsonOutputSerializationFormat;
(function (KnownJsonOutputSerializationFormat) {
    /** LineSeparated */
    KnownJsonOutputSerializationFormat["LineSeparated"] = "LineSeparated";
    /** Array */
    KnownJsonOutputSerializationFormat["Array"] = "Array";
})(KnownJsonOutputSerializationFormat || (KnownJsonOutputSerializationFormat = {}));
/** Known values of {@link RefreshType} that the service accepts. */
export var KnownRefreshType;
(function (KnownRefreshType) {
    /** Static */
    KnownRefreshType["Static"] = "Static";
    /** RefreshPeriodicallyWithFull */
    KnownRefreshType["RefreshPeriodicallyWithFull"] = "RefreshPeriodicallyWithFull";
    /** RefreshPeriodicallyWithDelta */
    KnownRefreshType["RefreshPeriodicallyWithDelta"] = "RefreshPeriodicallyWithDelta";
})(KnownRefreshType || (KnownRefreshType = {}));
