// @azure/arm-storagesync/dist-esm/src/operations/operationStatusOperations.js
/*
 * Copyright (c) Microsoft Corporation.
 * Licensed under the MIT License.
 *
 * Code generated by Microsoft (R) AutoRest Code Generator.
 * Changes may cause incorrect behavior and will be lost if the code is regenerated.
 */
import * as coreClient from "@azure/core-client";
import * as Mappers from "../models/mappers";
import * as Parameters from "../models/parameters";
/** Class containing OperationStatusOperations operations. */
export class OperationStatusOperationsImpl {
    /**
     * Initialize a new instance of the class OperationStatusOperations class.
     * @param client Reference to the service client
     */
    constructor(client) {
        this.client = client;
    }
    /**
     * Get Operation status
     * @param resourceGroupName The name of the resource group. The name is case insensitive.
     * @param locationName The desired region to obtain information from.
     * @param workflowId workflow Id
     * @param operationId operation Id
     * @param options The options parameters.
     */
    get(resourceGroupName, locationName, workflowId, operationId, options) {
        return this.client.sendOperationRequest({ resourceGroupName, locationName, workflowId, operationId, options }, getOperationSpec);
    }
}
// Operation Specifications
const serializer = coreClient.createSerializer(Mappers, /* isXml */ false);
const getOperationSpec = {
    path: "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.StorageSync/locations/{locationName}/workflows/{workflowId}/operations/{operationId}",
    httpMethod: "GET",
    responses: {
        200: {
            bodyMapper: Mappers.OperationStatus,
            headersMapper: Mappers.OperationStatusGetHeaders
        },
        default: {
            bodyMapper: Mappers.StorageSyncError
        }
    },
    queryParameters: [Parameters.apiVersion],
    urlParameters: [
        Parameters.$host,
        Parameters.locationName,
        Parameters.subscriptionId,
        Parameters.resourceGroupName,
        Parameters.workflowId,
        Parameters.operationId
    ],
    headerParameters: [Parameters.accept],
    serializer
};
