// @azure/arm-storagesync/dist-esm/src/operations/syncGroups.js
/*
 * Copyright (c) Microsoft Corporation.
 * Licensed under the MIT License.
 *
 * Code generated by Microsoft (R) AutoRest Code Generator.
 * Changes may cause incorrect behavior and will be lost if the code is regenerated.
 */
import { __asyncDelegator, __asyncGenerator, __asyncValues, __await } from "tslib";
import * as coreClient from "@azure/core-client";
import * as Mappers from "../models/mappers";
import * as Parameters from "../models/parameters";
/// <reference lib="esnext.asynciterable" />
/** Class containing SyncGroups operations. */
export class SyncGroupsImpl {
    /**
     * Initialize a new instance of the class SyncGroups class.
     * @param client Reference to the service client
     */
    constructor(client) {
        this.client = client;
    }
    /**
     * Get a SyncGroup List.
     * @param resourceGroupName The name of the resource group. The name is case insensitive.
     * @param storageSyncServiceName Name of Storage Sync Service resource.
     * @param options The options parameters.
     */
    listByStorageSyncService(resourceGroupName, storageSyncServiceName, options) {
        const iter = this.listByStorageSyncServicePagingAll(resourceGroupName, storageSyncServiceName, options);
        return {
            next() {
                return iter.next();
            },
            [Symbol.asyncIterator]() {
                return this;
            },
            byPage: (settings) => {
                if (settings === null || settings === void 0 ? void 0 : settings.maxPageSize) {
                    throw new Error("maxPageSize is not supported by this operation.");
                }
                return this.listByStorageSyncServicePagingPage(resourceGroupName, storageSyncServiceName, options, settings);
            }
        };
    }
    listByStorageSyncServicePagingPage(resourceGroupName, storageSyncServiceName, options, _settings) {
        return __asyncGenerator(this, arguments, function* listByStorageSyncServicePagingPage_1() {
            let result;
            result = yield __await(this._listByStorageSyncService(resourceGroupName, storageSyncServiceName, options));
            yield yield __await(result.value || []);
        });
    }
    listByStorageSyncServicePagingAll(resourceGroupName, storageSyncServiceName, options) {
        return __asyncGenerator(this, arguments, function* listByStorageSyncServicePagingAll_1() {
            var e_1, _a;
            try {
                for (var _b = __asyncValues(this.listByStorageSyncServicePagingPage(resourceGroupName, storageSyncServiceName, options)), _c; _c = yield __await(_b.next()), !_c.done;) {
                    const page = _c.value;
                    yield __await(yield* __asyncDelegator(__asyncValues(page)));
                }
            }
            catch (e_1_1) { e_1 = { error: e_1_1 }; }
            finally {
                try {
                    if (_c && !_c.done && (_a = _b.return)) yield __await(_a.call(_b));
                }
                finally { if (e_1) throw e_1.error; }
            }
        });
    }
    /**
     * Get a SyncGroup List.
     * @param resourceGroupName The name of the resource group. The name is case insensitive.
     * @param storageSyncServiceName Name of Storage Sync Service resource.
     * @param options The options parameters.
     */
    _listByStorageSyncService(resourceGroupName, storageSyncServiceName, options) {
        return this.client.sendOperationRequest({ resourceGroupName, storageSyncServiceName, options }, listByStorageSyncServiceOperationSpec);
    }
    /**
     * Create a new SyncGroup.
     * @param resourceGroupName The name of the resource group. The name is case insensitive.
     * @param storageSyncServiceName Name of Storage Sync Service resource.
     * @param syncGroupName Name of Sync Group resource.
     * @param parameters Sync Group Body
     * @param options The options parameters.
     */
    create(resourceGroupName, storageSyncServiceName, syncGroupName, parameters, options) {
        return this.client.sendOperationRequest({
            resourceGroupName,
            storageSyncServiceName,
            syncGroupName,
            parameters,
            options
        }, createOperationSpec);
    }
    /**
     * Get a given SyncGroup.
     * @param resourceGroupName The name of the resource group. The name is case insensitive.
     * @param storageSyncServiceName Name of Storage Sync Service resource.
     * @param syncGroupName Name of Sync Group resource.
     * @param options The options parameters.
     */
    get(resourceGroupName, storageSyncServiceName, syncGroupName, options) {
        return this.client.sendOperationRequest({ resourceGroupName, storageSyncServiceName, syncGroupName, options }, getOperationSpec);
    }
    /**
     * Delete a given SyncGroup.
     * @param resourceGroupName The name of the resource group. The name is case insensitive.
     * @param storageSyncServiceName Name of Storage Sync Service resource.
     * @param syncGroupName Name of Sync Group resource.
     * @param options The options parameters.
     */
    delete(resourceGroupName, storageSyncServiceName, syncGroupName, options) {
        return this.client.sendOperationRequest({ resourceGroupName, storageSyncServiceName, syncGroupName, options }, deleteOperationSpec);
    }
}
// Operation Specifications
const serializer = coreClient.createSerializer(Mappers, /* isXml */ false);
const listByStorageSyncServiceOperationSpec = {
    path: "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.StorageSync/storageSyncServices/{storageSyncServiceName}/syncGroups",
    httpMethod: "GET",
    responses: {
        200: {
            bodyMapper: Mappers.SyncGroupArray,
            headersMapper: Mappers.SyncGroupsListByStorageSyncServiceHeaders
        },
        default: {
            bodyMapper: Mappers.StorageSyncError
        }
    },
    queryParameters: [Parameters.apiVersion],
    urlParameters: [
        Parameters.$host,
        Parameters.subscriptionId,
        Parameters.resourceGroupName,
        Parameters.storageSyncServiceName
    ],
    headerParameters: [Parameters.accept],
    serializer
};
const createOperationSpec = {
    path: "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.StorageSync/storageSyncServices/{storageSyncServiceName}/syncGroups/{syncGroupName}",
    httpMethod: "PUT",
    responses: {
        200: {
            bodyMapper: Mappers.SyncGroup,
            headersMapper: Mappers.SyncGroupsCreateHeaders
        },
        default: {
            bodyMapper: Mappers.StorageSyncError
        }
    },
    requestBody: Parameters.parameters3,
    queryParameters: [Parameters.apiVersion],
    urlParameters: [
        Parameters.$host,
        Parameters.subscriptionId,
        Parameters.resourceGroupName,
        Parameters.storageSyncServiceName,
        Parameters.syncGroupName
    ],
    headerParameters: [Parameters.accept, Parameters.contentType],
    mediaType: "json",
    serializer
};
const getOperationSpec = {
    path: "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.StorageSync/storageSyncServices/{storageSyncServiceName}/syncGroups/{syncGroupName}",
    httpMethod: "GET",
    responses: {
        200: {
            bodyMapper: Mappers.SyncGroup,
            headersMapper: Mappers.SyncGroupsGetHeaders
        },
        default: {
            bodyMapper: Mappers.StorageSyncError
        }
    },
    queryParameters: [Parameters.apiVersion],
    urlParameters: [
        Parameters.$host,
        Parameters.subscriptionId,
        Parameters.resourceGroupName,
        Parameters.storageSyncServiceName,
        Parameters.syncGroupName
    ],
    headerParameters: [Parameters.accept],
    serializer
};
const deleteOperationSpec = {
    path: "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.StorageSync/storageSyncServices/{storageSyncServiceName}/syncGroups/{syncGroupName}",
    httpMethod: "DELETE",
    responses: {
        200: {
            headersMapper: Mappers.SyncGroupsDeleteHeaders
        },
        204: {},
        default: {
            bodyMapper: Mappers.StorageSyncError
        }
    },
    queryParameters: [Parameters.apiVersion],
    urlParameters: [
        Parameters.$host,
        Parameters.subscriptionId,
        Parameters.resourceGroupName,
        Parameters.storageSyncServiceName,
        Parameters.syncGroupName
    ],
    headerParameters: [Parameters.accept],
    serializer
};
