// @aws-sdk/client-macie/dist-cjs/Macie.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.Macie = void 0;
const smithy_client_1 = require("@smithy/smithy-client");
const AssociateMemberAccountCommand_1 = require("./commands/AssociateMemberAccountCommand");
const AssociateS3ResourcesCommand_1 = require("./commands/AssociateS3ResourcesCommand");
const DisassociateMemberAccountCommand_1 = require("./commands/DisassociateMemberAccountCommand");
const DisassociateS3ResourcesCommand_1 = require("./commands/DisassociateS3ResourcesCommand");
const ListMemberAccountsCommand_1 = require("./commands/ListMemberAccountsCommand");
const ListS3ResourcesCommand_1 = require("./commands/ListS3ResourcesCommand");
const UpdateS3ResourcesCommand_1 = require("./commands/UpdateS3ResourcesCommand");
const MacieClient_1 = require("./MacieClient");
const commands = {
    AssociateMemberAccountCommand: AssociateMemberAccountCommand_1.AssociateMemberAccountCommand,
    AssociateS3ResourcesCommand: AssociateS3ResourcesCommand_1.AssociateS3ResourcesCommand,
    DisassociateMemberAccountCommand: DisassociateMemberAccountCommand_1.DisassociateMemberAccountCommand,
    DisassociateS3ResourcesCommand: DisassociateS3ResourcesCommand_1.DisassociateS3ResourcesCommand,
    ListMemberAccountsCommand: ListMemberAccountsCommand_1.ListMemberAccountsCommand,
    ListS3ResourcesCommand: ListS3ResourcesCommand_1.ListS3ResourcesCommand,
    UpdateS3ResourcesCommand: UpdateS3ResourcesCommand_1.UpdateS3ResourcesCommand,
};
class Macie extends MacieClient_1.MacieClient {
}
exports.Macie = Macie;
(0, smithy_client_1.createAggregatedClient)(commands, Macie);
