// @azure/arm-edgegateway/esm/operations/triggers.js
/*
 * Copyright (c) Microsoft Corporation. All rights reserved.
 * Licensed under the MIT License. See License.txt in the project root for
 * license information.
 *
 * Code generated by Microsoft (R) AutoRest Code Generator.
 * Changes may cause incorrect behavior and will be lost if the code is
 * regenerated.
 */
import { __assign } from "tslib";
import * as msRest from "@azure/ms-rest-js";
import * as Mappers from "../models/triggersMappers";
import * as Parameters from "../models/parameters";
/** Class representing a Triggers. */
var Triggers = /** @class */ (function () {
    /**
     * Create a Triggers.
     * @param {DataBoxEdgeManagementClientContext} client Reference to the service client.
     */
    function Triggers(client) {
        this.client = client;
    }
    Triggers.prototype.listByDataBoxEdgeDevice = function (deviceName, resourceGroupName, options, callback) {
        return this.client.sendOperationRequest({
            deviceName: deviceName,
            resourceGroupName: resourceGroupName,
            options: options
        }, listByDataBoxEdgeDeviceOperationSpec, callback);
    };
    Triggers.prototype.get = function (deviceName, name, resourceGroupName, options, callback) {
        return this.client.sendOperationRequest({
            deviceName: deviceName,
            name: name,
            resourceGroupName: resourceGroupName,
            options: options
        }, getOperationSpec, callback);
    };
    /**
     * Creates or updates a trigger.
     * @param deviceName Creates or updates a trigger
     * @param name The trigger name.
     * @param trigger The trigger.
     * @param resourceGroupName The resource group name.
     * @param [options] The optional parameters
     * @returns Promise<Models.TriggersCreateOrUpdateResponse>
     */
    Triggers.prototype.createOrUpdate = function (deviceName, name, trigger, resourceGroupName, options) {
        return this.beginCreateOrUpdate(deviceName, name, trigger, resourceGroupName, options)
            .then(function (lroPoller) { return lroPoller.pollUntilFinished(); });
    };
    /**
     * Deletes the trigger on the gateway device.
     * @param deviceName The device name.
     * @param name The trigger name.
     * @param resourceGroupName The resource group name.
     * @param [options] The optional parameters
     * @returns Promise<msRest.RestResponse>
     */
    Triggers.prototype.deleteMethod = function (deviceName, name, resourceGroupName, options) {
        return this.beginDeleteMethod(deviceName, name, resourceGroupName, options)
            .then(function (lroPoller) { return lroPoller.pollUntilFinished(); });
    };
    /**
     * Creates or updates a trigger.
     * @param deviceName Creates or updates a trigger
     * @param name The trigger name.
     * @param trigger The trigger.
     * @param resourceGroupName The resource group name.
     * @param [options] The optional parameters
     * @returns Promise<msRestAzure.LROPoller>
     */
    Triggers.prototype.beginCreateOrUpdate = function (deviceName, name, trigger, resourceGroupName, options) {
        return this.client.sendLRORequest({
            deviceName: deviceName,
            name: name,
            trigger: trigger,
            resourceGroupName: resourceGroupName,
            options: options
        }, beginCreateOrUpdateOperationSpec, options);
    };
    /**
     * Deletes the trigger on the gateway device.
     * @param deviceName The device name.
     * @param name The trigger name.
     * @param resourceGroupName The resource group name.
     * @param [options] The optional parameters
     * @returns Promise<msRestAzure.LROPoller>
     */
    Triggers.prototype.beginDeleteMethod = function (deviceName, name, resourceGroupName, options) {
        return this.client.sendLRORequest({
            deviceName: deviceName,
            name: name,
            resourceGroupName: resourceGroupName,
            options: options
        }, beginDeleteMethodOperationSpec, options);
    };
    Triggers.prototype.listByDataBoxEdgeDeviceNext = function (nextPageLink, options, callback) {
        return this.client.sendOperationRequest({
            nextPageLink: nextPageLink,
            options: options
        }, listByDataBoxEdgeDeviceNextOperationSpec, callback);
    };
    return Triggers;
}());
export { Triggers };
// Operation Specifications
var serializer = new msRest.Serializer(Mappers);
var listByDataBoxEdgeDeviceOperationSpec = {
    httpMethod: "GET",
    path: "subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.DataBoxEdge/dataBoxEdgeDevices/{deviceName}/triggers",
    urlParameters: [
        Parameters.deviceName,
        Parameters.subscriptionId,
        Parameters.resourceGroupName
    ],
    queryParameters: [
        Parameters.apiVersion,
        Parameters.expand
    ],
    headerParameters: [
        Parameters.acceptLanguage
    ],
    responses: {
        200: {
            bodyMapper: Mappers.TriggerList
        },
        default: {
            bodyMapper: Mappers.CloudError
        }
    },
    serializer: serializer
};
var getOperationSpec = {
    httpMethod: "GET",
    path: "subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.DataBoxEdge/dataBoxEdgeDevices/{deviceName}/triggers/{name}",
    urlParameters: [
        Parameters.deviceName,
        Parameters.name,
        Parameters.subscriptionId,
        Parameters.resourceGroupName
    ],
    queryParameters: [
        Parameters.apiVersion
    ],
    headerParameters: [
        Parameters.acceptLanguage
    ],
    responses: {
        200: {
            bodyMapper: Mappers.Trigger
        },
        default: {
            bodyMapper: Mappers.CloudError
        }
    },
    serializer: serializer
};
var beginCreateOrUpdateOperationSpec = {
    httpMethod: "PUT",
    path: "subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.DataBoxEdge/dataBoxEdgeDevices/{deviceName}/triggers/{name}",
    urlParameters: [
        Parameters.deviceName,
        Parameters.name,
        Parameters.subscriptionId,
        Parameters.resourceGroupName
    ],
    queryParameters: [
        Parameters.apiVersion
    ],
    headerParameters: [
        Parameters.acceptLanguage
    ],
    requestBody: {
        parameterPath: "trigger",
        mapper: __assign(__assign({}, Mappers.Trigger), { required: true })
    },
    responses: {
        200: {
            bodyMapper: Mappers.Trigger
        },
        202: {},
        default: {
            bodyMapper: Mappers.CloudError
        }
    },
    serializer: serializer
};
var beginDeleteMethodOperationSpec = {
    httpMethod: "DELETE",
    path: "subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.DataBoxEdge/dataBoxEdgeDevices/{deviceName}/triggers/{name}",
    urlParameters: [
        Parameters.deviceName,
        Parameters.name,
        Parameters.subscriptionId,
        Parameters.resourceGroupName
    ],
    queryParameters: [
        Parameters.apiVersion
    ],
    headerParameters: [
        Parameters.acceptLanguage
    ],
    responses: {
        200: {},
        202: {},
        204: {},
        default: {
            bodyMapper: Mappers.CloudError
        }
    },
    serializer: serializer
};
var listByDataBoxEdgeDeviceNextOperationSpec = {
    httpMethod: "GET",
    baseUrl: "https://management.azure.com",
    path: "{nextLink}",
    urlParameters: [
        Parameters.nextPageLink
    ],
    headerParameters: [
        Parameters.acceptLanguage
    ],
    responses: {
        200: {
            bodyMapper: Mappers.TriggerList
        },
        default: {
            bodyMapper: Mappers.CloudError
        }
    },
    serializer: serializer
};
