// @azure/arm-extendedlocation/dist-esm/src/models/mappers.js
/*
 * Copyright (c) Microsoft Corporation.
 * Licensed under the MIT License.
 *
 * Code generated by Microsoft (R) AutoRest Code Generator.
 * Changes may cause incorrect behavior and will be lost if the code is regenerated.
 */
export const CustomLocationOperationsList = {
    type: {
        name: "Composite",
        className: "CustomLocationOperationsList",
        modelProperties: {
            nextLink: {
                serializedName: "nextLink",
                type: {
                    name: "String"
                }
            },
            value: {
                serializedName: "value",
                required: true,
                type: {
                    name: "Sequence",
                    element: {
                        type: {
                            name: "Composite",
                            className: "CustomLocationOperation"
                        }
                    }
                }
            }
        }
    }
};
export const CustomLocationOperation = {
    type: {
        name: "Composite",
        className: "CustomLocationOperation",
        modelProperties: {
            isDataAction: {
                serializedName: "isDataAction",
                readOnly: true,
                type: {
                    name: "Boolean"
                }
            },
            name: {
                serializedName: "name",
                readOnly: true,
                type: {
                    name: "String"
                }
            },
            origin: {
                serializedName: "origin",
                readOnly: true,
                type: {
                    name: "String"
                }
            },
            description: {
                serializedName: "display.description",
                readOnly: true,
                type: {
                    name: "String"
                }
            },
            operation: {
                serializedName: "display.operation",
                readOnly: true,
                type: {
                    name: "String"
                }
            },
            provider: {
                serializedName: "display.provider",
                readOnly: true,
                type: {
                    name: "String"
                }
            },
            resource: {
                serializedName: "display.resource",
                readOnly: true,
                type: {
                    name: "String"
                }
            }
        }
    }
};
export const ErrorResponse = {
    type: {
        name: "Composite",
        className: "ErrorResponse",
        modelProperties: {
            error: {
                serializedName: "error",
                type: {
                    name: "Composite",
                    className: "ErrorDetail"
                }
            }
        }
    }
};
export const ErrorDetail = {
    type: {
        name: "Composite",
        className: "ErrorDetail",
        modelProperties: {
            code: {
                serializedName: "code",
                readOnly: true,
                type: {
                    name: "String"
                }
            },
            message: {
                serializedName: "message",
                readOnly: true,
                type: {
                    name: "String"
                }
            },
            target: {
                serializedName: "target",
                readOnly: true,
                type: {
                    name: "String"
                }
            },
            details: {
                serializedName: "details",
                readOnly: true,
                type: {
                    name: "Sequence",
                    element: {
                        type: {
                            name: "Composite",
                            className: "ErrorDetail"
                        }
                    }
                }
            },
            additionalInfo: {
                serializedName: "additionalInfo",
                readOnly: true,
                type: {
                    name: "Sequence",
                    element: {
                        type: {
                            name: "Composite",
                            className: "ErrorAdditionalInfo"
                        }
                    }
                }
            }
        }
    }
};
export const ErrorAdditionalInfo = {
    type: {
        name: "Composite",
        className: "ErrorAdditionalInfo",
        modelProperties: {
            type: {
                serializedName: "type",
                readOnly: true,
                type: {
                    name: "String"
                }
            },
            info: {
                serializedName: "info",
                readOnly: true,
                type: {
                    name: "Dictionary",
                    value: { type: { name: "any" } }
                }
            }
        }
    }
};
export const CustomLocationListResult = {
    type: {
        name: "Composite",
        className: "CustomLocationListResult",
        modelProperties: {
            nextLink: {
                serializedName: "nextLink",
                readOnly: true,
                type: {
                    name: "String"
                }
            },
            value: {
                serializedName: "value",
                readOnly: true,
                type: {
                    name: "Sequence",
                    element: {
                        type: {
                            name: "Composite",
                            className: "CustomLocation"
                        }
                    }
                }
            }
        }
    }
};
export const Identity = {
    type: {
        name: "Composite",
        className: "Identity",
        modelProperties: {
            principalId: {
                serializedName: "principalId",
                readOnly: true,
                type: {
                    name: "String"
                }
            },
            tenantId: {
                serializedName: "tenantId",
                readOnly: true,
                type: {
                    name: "String"
                }
            },
            type: {
                serializedName: "type",
                type: {
                    name: "String"
                }
            }
        }
    }
};
export const CustomLocationPropertiesAuthentication = {
    type: {
        name: "Composite",
        className: "CustomLocationPropertiesAuthentication",
        modelProperties: {
            type: {
                serializedName: "type",
                type: {
                    name: "String"
                }
            },
            value: {
                serializedName: "value",
                type: {
                    name: "String"
                }
            }
        }
    }
};
export const SystemData = {
    type: {
        name: "Composite",
        className: "SystemData",
        modelProperties: {
            createdBy: {
                serializedName: "createdBy",
                type: {
                    name: "String"
                }
            },
            createdByType: {
                serializedName: "createdByType",
                type: {
                    name: "String"
                }
            },
            createdAt: {
                serializedName: "createdAt",
                type: {
                    name: "DateTime"
                }
            },
            lastModifiedBy: {
                serializedName: "lastModifiedBy",
                type: {
                    name: "String"
                }
            },
            lastModifiedByType: {
                serializedName: "lastModifiedByType",
                type: {
                    name: "String"
                }
            },
            lastModifiedAt: {
                serializedName: "lastModifiedAt",
                type: {
                    name: "DateTime"
                }
            }
        }
    }
};
export const Resource = {
    type: {
        name: "Composite",
        className: "Resource",
        modelProperties: {
            id: {
                serializedName: "id",
                readOnly: true,
                type: {
                    name: "String"
                }
            },
            name: {
                serializedName: "name",
                readOnly: true,
                type: {
                    name: "String"
                }
            },
            type: {
                serializedName: "type",
                readOnly: true,
                type: {
                    name: "String"
                }
            }
        }
    }
};
export const PatchableCustomLocations = {
    type: {
        name: "Composite",
        className: "PatchableCustomLocations",
        modelProperties: {
            identity: {
                serializedName: "identity",
                type: {
                    name: "Composite",
                    className: "Identity"
                }
            },
            tags: {
                serializedName: "tags",
                type: {
                    name: "Dictionary",
                    value: { type: { name: "String" } }
                }
            },
            authentication: {
                serializedName: "properties.authentication",
                type: {
                    name: "Composite",
                    className: "CustomLocationPropertiesAuthentication"
                }
            },
            clusterExtensionIds: {
                serializedName: "properties.clusterExtensionIds",
                type: {
                    name: "Sequence",
                    element: {
                        type: {
                            name: "String"
                        }
                    }
                }
            },
            displayName: {
                serializedName: "properties.displayName",
                type: {
                    name: "String"
                }
            },
            hostResourceId: {
                serializedName: "properties.hostResourceId",
                type: {
                    name: "String"
                }
            },
            hostType: {
                serializedName: "properties.hostType",
                type: {
                    name: "String"
                }
            },
            namespace: {
                serializedName: "properties.namespace",
                type: {
                    name: "String"
                }
            },
            provisioningState: {
                serializedName: "properties.provisioningState",
                type: {
                    name: "String"
                }
            }
        }
    }
};
export const EnabledResourceTypesListResult = {
    type: {
        name: "Composite",
        className: "EnabledResourceTypesListResult",
        modelProperties: {
            nextLink: {
                serializedName: "nextLink",
                readOnly: true,
                type: {
                    name: "String"
                }
            },
            value: {
                serializedName: "value",
                readOnly: true,
                type: {
                    name: "Sequence",
                    element: {
                        type: {
                            name: "Composite",
                            className: "EnabledResourceType"
                        }
                    }
                }
            }
        }
    }
};
export const EnabledResourceTypePropertiesTypesMetadataItem = {
    type: {
        name: "Composite",
        className: "EnabledResourceTypePropertiesTypesMetadataItem",
        modelProperties: {
            apiVersion: {
                serializedName: "apiVersion",
                type: {
                    name: "String"
                }
            },
            resourceProviderNamespace: {
                serializedName: "resourceProviderNamespace",
                type: {
                    name: "String"
                }
            },
            resourceType: {
                serializedName: "resourceType",
                type: {
                    name: "String"
                }
            }
        }
    }
};
export const CustomLocationFindTargetResourceGroupProperties = {
    type: {
        name: "Composite",
        className: "CustomLocationFindTargetResourceGroupProperties",
        modelProperties: {
            labels: {
                serializedName: "labels",
                type: {
                    name: "Dictionary",
                    value: { type: { name: "String" } }
                }
            }
        }
    }
};
export const CustomLocationFindTargetResourceGroupResult = {
    type: {
        name: "Composite",
        className: "CustomLocationFindTargetResourceGroupResult",
        modelProperties: {
            matchedResourceSyncRule: {
                serializedName: "matchedResourceSyncRule",
                readOnly: true,
                type: {
                    name: "String"
                }
            },
            targetResourceGroup: {
                serializedName: "targetResourceGroup",
                readOnly: true,
                type: {
                    name: "String"
                }
            }
        }
    }
};
export const ResourceSyncRuleListResult = {
    type: {
        name: "Composite",
        className: "ResourceSyncRuleListResult",
        modelProperties: {
            nextLink: {
                serializedName: "nextLink",
                readOnly: true,
                type: {
                    name: "String"
                }
            },
            value: {
                serializedName: "value",
                readOnly: true,
                type: {
                    name: "Sequence",
                    element: {
                        type: {
                            name: "Composite",
                            className: "ResourceSyncRule"
                        }
                    }
                }
            }
        }
    }
};
export const ResourceSyncRulePropertiesSelector = {
    type: {
        name: "Composite",
        className: "ResourceSyncRulePropertiesSelector",
        modelProperties: {
            matchExpressions: {
                serializedName: "matchExpressions",
                type: {
                    name: "Sequence",
                    element: {
                        type: {
                            name: "Composite",
                            className: "MatchExpressionsProperties"
                        }
                    }
                }
            },
            matchLabels: {
                serializedName: "matchLabels",
                type: {
                    name: "Dictionary",
                    value: { type: { name: "String" } }
                }
            }
        }
    }
};
export const MatchExpressionsProperties = {
    type: {
        name: "Composite",
        className: "MatchExpressionsProperties",
        modelProperties: {
            key: {
                serializedName: "key",
                type: {
                    name: "String"
                }
            },
            operator: {
                serializedName: "operator",
                type: {
                    name: "String"
                }
            },
            values: {
                serializedName: "values",
                type: {
                    name: "Sequence",
                    element: {
                        type: {
                            name: "String"
                        }
                    }
                }
            }
        }
    }
};
export const PatchableResourceSyncRule = {
    type: {
        name: "Composite",
        className: "PatchableResourceSyncRule",
        modelProperties: {
            tags: {
                serializedName: "tags",
                type: {
                    name: "Dictionary",
                    value: { type: { name: "String" } }
                }
            },
            priority: {
                serializedName: "properties.priority",
                type: {
                    name: "Number"
                }
            },
            provisioningState: {
                serializedName: "properties.provisioningState",
                readOnly: true,
                type: {
                    name: "String"
                }
            },
            selector: {
                serializedName: "properties.selector",
                type: {
                    name: "Composite",
                    className: "ResourceSyncRulePropertiesSelector"
                }
            },
            targetResourceGroup: {
                serializedName: "properties.targetResourceGroup",
                type: {
                    name: "String"
                }
            }
        }
    }
};
export const TrackedResource = {
    type: {
        name: "Composite",
        className: "TrackedResource",
        modelProperties: Object.assign(Object.assign({}, Resource.type.modelProperties), { tags: {
                serializedName: "tags",
                type: {
                    name: "Dictionary",
                    value: { type: { name: "String" } }
                }
            }, location: {
                serializedName: "location",
                required: true,
                type: {
                    name: "String"
                }
            } })
    }
};
export const ProxyResource = {
    type: {
        name: "Composite",
        className: "ProxyResource",
        modelProperties: Object.assign({}, Resource.type.modelProperties)
    }
};
export const CustomLocation = {
    type: {
        name: "Composite",
        className: "CustomLocation",
        modelProperties: Object.assign(Object.assign({}, TrackedResource.type.modelProperties), { identity: {
                serializedName: "identity",
                type: {
                    name: "Composite",
                    className: "Identity"
                }
            }, systemData: {
                serializedName: "systemData",
                type: {
                    name: "Composite",
                    className: "SystemData"
                }
            }, authentication: {
                serializedName: "properties.authentication",
                type: {
                    name: "Composite",
                    className: "CustomLocationPropertiesAuthentication"
                }
            }, clusterExtensionIds: {
                serializedName: "properties.clusterExtensionIds",
                type: {
                    name: "Sequence",
                    element: {
                        type: {
                            name: "String"
                        }
                    }
                }
            }, displayName: {
                serializedName: "properties.displayName",
                type: {
                    name: "String"
                }
            }, hostResourceId: {
                serializedName: "properties.hostResourceId",
                type: {
                    name: "String"
                }
            }, hostType: {
                serializedName: "properties.hostType",
                type: {
                    name: "String"
                }
            }, namespace: {
                serializedName: "properties.namespace",
                type: {
                    name: "String"
                }
            }, provisioningState: {
                serializedName: "properties.provisioningState",
                type: {
                    name: "String"
                }
            } })
    }
};
export const ResourceSyncRule = {
    type: {
        name: "Composite",
        className: "ResourceSyncRule",
        modelProperties: Object.assign(Object.assign({}, TrackedResource.type.modelProperties), { systemData: {
                serializedName: "systemData",
                type: {
                    name: "Composite",
                    className: "SystemData"
                }
            }, priority: {
                serializedName: "properties.priority",
                type: {
                    name: "Number"
                }
            }, provisioningState: {
                serializedName: "properties.provisioningState",
                readOnly: true,
                type: {
                    name: "String"
                }
            }, selector: {
                serializedName: "properties.selector",
                type: {
                    name: "Composite",
                    className: "ResourceSyncRulePropertiesSelector"
                }
            }, targetResourceGroup: {
                serializedName: "properties.targetResourceGroup",
                type: {
                    name: "String"
                }
            } })
    }
};
export const EnabledResourceType = {
    type: {
        name: "Composite",
        className: "EnabledResourceType",
        modelProperties: Object.assign(Object.assign({}, ProxyResource.type.modelProperties), { systemData: {
                serializedName: "systemData",
                type: {
                    name: "Composite",
                    className: "SystemData"
                }
            }, clusterExtensionId: {
                serializedName: "properties.clusterExtensionId",
                type: {
                    name: "String"
                }
            }, extensionType: {
                serializedName: "properties.extensionType",
                type: {
                    name: "String"
                }
            }, typesMetadata: {
                serializedName: "properties.typesMetadata",
                type: {
                    name: "Sequence",
                    element: {
                        type: {
                            name: "Composite",
                            className: "EnabledResourceTypePropertiesTypesMetadataItem"
                        }
                    }
                }
            } })
    }
};
