// @aws-sdk/client-inspector/dist-cjs/index.js
"use strict";
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __name = (target, value) => __defProp(target, "name", { value, configurable: true });
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// src/index.ts
var src_exports = {};
__export(src_exports, {
  AccessDeniedErrorCode: () => AccessDeniedErrorCode,
  AccessDeniedException: () => AccessDeniedException,
  AddAttributesToFindingsCommand: () => AddAttributesToFindingsCommand,
  AgentHealth: () => AgentHealth,
  AgentHealthCode: () => AgentHealthCode,
  AgentsAlreadyRunningAssessmentException: () => AgentsAlreadyRunningAssessmentException,
  AssessmentRunInProgressException: () => AssessmentRunInProgressException,
  AssessmentRunNotificationSnsStatusCode: () => AssessmentRunNotificationSnsStatusCode,
  AssessmentRunState: () => AssessmentRunState,
  AssetType: () => AssetType,
  CreateAssessmentTargetCommand: () => CreateAssessmentTargetCommand,
  CreateAssessmentTemplateCommand: () => CreateAssessmentTemplateCommand,
  CreateExclusionsPreviewCommand: () => CreateExclusionsPreviewCommand,
  CreateResourceGroupCommand: () => CreateResourceGroupCommand,
  DeleteAssessmentRunCommand: () => DeleteAssessmentRunCommand,
  DeleteAssessmentTargetCommand: () => DeleteAssessmentTargetCommand,
  DeleteAssessmentTemplateCommand: () => DeleteAssessmentTemplateCommand,
  DescribeAssessmentRunsCommand: () => DescribeAssessmentRunsCommand,
  DescribeAssessmentTargetsCommand: () => DescribeAssessmentTargetsCommand,
  DescribeAssessmentTemplatesCommand: () => DescribeAssessmentTemplatesCommand,
  DescribeCrossAccountAccessRoleCommand: () => DescribeCrossAccountAccessRoleCommand,
  DescribeExclusionsCommand: () => DescribeExclusionsCommand,
  DescribeFindingsCommand: () => DescribeFindingsCommand,
  DescribeResourceGroupsCommand: () => DescribeResourceGroupsCommand,
  DescribeRulesPackagesCommand: () => DescribeRulesPackagesCommand,
  FailedItemErrorCode: () => FailedItemErrorCode,
  GetAssessmentReportCommand: () => GetAssessmentReportCommand,
  GetExclusionsPreviewCommand: () => GetExclusionsPreviewCommand,
  GetTelemetryMetadataCommand: () => GetTelemetryMetadataCommand,
  Inspector: () => Inspector,
  InspectorClient: () => InspectorClient,
  InspectorEvent: () => InspectorEvent,
  InspectorServiceException: () => InspectorServiceException,
  InternalException: () => InternalException,
  InvalidCrossAccountRoleErrorCode: () => InvalidCrossAccountRoleErrorCode,
  InvalidCrossAccountRoleException: () => InvalidCrossAccountRoleException,
  InvalidInputErrorCode: () => InvalidInputErrorCode,
  InvalidInputException: () => InvalidInputException,
  LimitExceededErrorCode: () => LimitExceededErrorCode,
  LimitExceededException: () => LimitExceededException,
  ListAssessmentRunAgentsCommand: () => ListAssessmentRunAgentsCommand,
  ListAssessmentRunsCommand: () => ListAssessmentRunsCommand,
  ListAssessmentTargetsCommand: () => ListAssessmentTargetsCommand,
  ListAssessmentTemplatesCommand: () => ListAssessmentTemplatesCommand,
  ListEventSubscriptionsCommand: () => ListEventSubscriptionsCommand,
  ListExclusionsCommand: () => ListExclusionsCommand,
  ListFindingsCommand: () => ListFindingsCommand,
  ListRulesPackagesCommand: () => ListRulesPackagesCommand,
  ListTagsForResourceCommand: () => ListTagsForResourceCommand,
  Locale: () => Locale,
  NoSuchEntityErrorCode: () => NoSuchEntityErrorCode,
  NoSuchEntityException: () => NoSuchEntityException,
  PreviewAgentsCommand: () => PreviewAgentsCommand,
  PreviewGenerationInProgressException: () => PreviewGenerationInProgressException,
  PreviewStatus: () => PreviewStatus,
  RegisterCrossAccountAccessRoleCommand: () => RegisterCrossAccountAccessRoleCommand,
  RemoveAttributesFromFindingsCommand: () => RemoveAttributesFromFindingsCommand,
  ReportFileFormat: () => ReportFileFormat,
  ReportStatus: () => ReportStatus,
  ReportType: () => ReportType,
  ScopeType: () => ScopeType,
  ServiceTemporarilyUnavailableException: () => ServiceTemporarilyUnavailableException,
  SetTagsForResourceCommand: () => SetTagsForResourceCommand,
  Severity: () => Severity,
  StartAssessmentRunCommand: () => StartAssessmentRunCommand,
  StopAction: () => StopAction,
  StopAssessmentRunCommand: () => StopAssessmentRunCommand,
  SubscribeToEventCommand: () => SubscribeToEventCommand,
  UnsubscribeFromEventCommand: () => UnsubscribeFromEventCommand,
  UnsupportedFeatureException: () => UnsupportedFeatureException,
  UpdateAssessmentTargetCommand: () => UpdateAssessmentTargetCommand,
  __Client: () => import_smithy_client.Client,
  paginateGetExclusionsPreview: () => paginateGetExclusionsPreview,
  paginateListAssessmentRunAgents: () => paginateListAssessmentRunAgents,
  paginateListAssessmentRuns: () => paginateListAssessmentRuns,
  paginateListAssessmentTargets: () => paginateListAssessmentTargets,
  paginateListAssessmentTemplates: () => paginateListAssessmentTemplates,
  paginateListEventSubscriptions: () => paginateListEventSubscriptions,
  paginateListExclusions: () => paginateListExclusions,
  paginateListFindings: () => paginateListFindings,
  paginateListRulesPackages: () => paginateListRulesPackages,
  paginatePreviewAgents: () => paginatePreviewAgents
});
module.exports = __toCommonJS(src_exports);

// src/InspectorClient.ts
var import_middleware_host_header = require("@aws-sdk/middleware-host-header");
var import_middleware_logger = require("@aws-sdk/middleware-logger");
var import_middleware_recursion_detection = require("@aws-sdk/middleware-recursion-detection");
var import_middleware_user_agent = require("@aws-sdk/middleware-user-agent");
var import_config_resolver = require("@smithy/config-resolver");
var import_core = require("@smithy/core");
var import_middleware_content_length = require("@smithy/middleware-content-length");
var import_middleware_endpoint = require("@smithy/middleware-endpoint");
var import_middleware_retry = require("@smithy/middleware-retry");

var import_httpAuthSchemeProvider = require("./auth/httpAuthSchemeProvider");

// src/endpoint/EndpointParameters.ts
var resolveClientEndpointParameters = /* @__PURE__ */ __name((options) => {
  return {
    ...options,
    useDualstackEndpoint: options.useDualstackEndpoint ?? false,
    useFipsEndpoint: options.useFipsEndpoint ?? false,
    defaultSigningName: "inspector"
  };
}, "resolveClientEndpointParameters");
var commonParams = {
  UseFIPS: { type: "builtInParams", name: "useFipsEndpoint" },
  Endpoint: { type: "builtInParams", name: "endpoint" },
  Region: { type: "builtInParams", name: "region" },
  UseDualStack: { type: "builtInParams", name: "useDualstackEndpoint" }
};

// src/InspectorClient.ts
var import_runtimeConfig = require("././runtimeConfig");

// src/runtimeExtensions.ts
var import_region_config_resolver = require("@aws-sdk/region-config-resolver");
var import_protocol_http = require("@smithy/protocol-http");
var import_smithy_client = require("@smithy/smithy-client");

// src/auth/httpAuthExtensionConfiguration.ts
var getHttpAuthExtensionConfiguration = /* @__PURE__ */ __name((runtimeConfig) => {
  const _httpAuthSchemes = runtimeConfig.httpAuthSchemes;
  let _httpAuthSchemeProvider = runtimeConfig.httpAuthSchemeProvider;
  let _credentials = runtimeConfig.credentials;
  return {
    setHttpAuthScheme(httpAuthScheme) {
      const index = _httpAuthSchemes.findIndex((scheme) => scheme.schemeId === httpAuthScheme.schemeId);
      if (index === -1) {
        _httpAuthSchemes.push(httpAuthScheme);
      } else {
        _httpAuthSchemes.splice(index, 1, httpAuthScheme);
      }
    },
    httpAuthSchemes() {
      return _httpAuthSchemes;
    },
    setHttpAuthSchemeProvider(httpAuthSchemeProvider) {
      _httpAuthSchemeProvider = httpAuthSchemeProvider;
    },
    httpAuthSchemeProvider() {
      return _httpAuthSchemeProvider;
    },
    setCredentials(credentials) {
      _credentials = credentials;
    },
    credentials() {
      return _credentials;
    }
  };
}, "getHttpAuthExtensionConfiguration");
var resolveHttpAuthRuntimeConfig = /* @__PURE__ */ __name((config) => {
  return {
    httpAuthSchemes: config.httpAuthSchemes(),
    httpAuthSchemeProvider: config.httpAuthSchemeProvider(),
    credentials: config.credentials()
  };
}, "resolveHttpAuthRuntimeConfig");

// src/runtimeExtensions.ts
var asPartial = /* @__PURE__ */ __name((t) => t, "asPartial");
var resolveRuntimeExtensions = /* @__PURE__ */ __name((runtimeConfig, extensions) => {
  const extensionConfiguration = {
    ...asPartial((0, import_region_config_resolver.getAwsRegionExtensionConfiguration)(runtimeConfig)),
    ...asPartial((0, import_smithy_client.getDefaultExtensionConfiguration)(runtimeConfig)),
    ...asPartial((0, import_protocol_http.getHttpHandlerExtensionConfiguration)(runtimeConfig)),
    ...asPartial(getHttpAuthExtensionConfiguration(runtimeConfig))
  };
  extensions.forEach((extension) => extension.configure(extensionConfiguration));
  return {
    ...runtimeConfig,
    ...(0, import_region_config_resolver.resolveAwsRegionExtensionConfiguration)(extensionConfiguration),
    ...(0, import_smithy_client.resolveDefaultRuntimeConfig)(extensionConfiguration),
    ...(0, import_protocol_http.resolveHttpHandlerRuntimeConfig)(extensionConfiguration),
    ...resolveHttpAuthRuntimeConfig(extensionConfiguration)
  };
}, "resolveRuntimeExtensions");

// src/InspectorClient.ts
var _InspectorClient = class _InspectorClient extends import_smithy_client.Client {
  constructor(...[configuration]) {
    const _config_0 = (0, import_runtimeConfig.getRuntimeConfig)(configuration || {});
    const _config_1 = resolveClientEndpointParameters(_config_0);
    const _config_2 = (0, import_config_resolver.resolveRegionConfig)(_config_1);
    const _config_3 = (0, import_middleware_endpoint.resolveEndpointConfig)(_config_2);
    const _config_4 = (0, import_middleware_retry.resolveRetryConfig)(_config_3);
    const _config_5 = (0, import_middleware_host_header.resolveHostHeaderConfig)(_config_4);
    const _config_6 = (0, import_middleware_user_agent.resolveUserAgentConfig)(_config_5);
    const _config_7 = (0, import_httpAuthSchemeProvider.resolveHttpAuthSchemeConfig)(_config_6);
    const _config_8 = resolveRuntimeExtensions(_config_7, (configuration == null ? void 0 : configuration.extensions) || []);
    super(_config_8);
    this.config = _config_8;
    this.middlewareStack.use((0, import_middleware_retry.getRetryPlugin)(this.config));
    this.middlewareStack.use((0, import_middleware_content_length.getContentLengthPlugin)(this.config));
    this.middlewareStack.use((0, import_middleware_host_header.getHostHeaderPlugin)(this.config));
    this.middlewareStack.use((0, import_middleware_logger.getLoggerPlugin)(this.config));
    this.middlewareStack.use((0, import_middleware_recursion_detection.getRecursionDetectionPlugin)(this.config));
    this.middlewareStack.use((0, import_middleware_user_agent.getUserAgentPlugin)(this.config));
    this.middlewareStack.use(
      (0, import_core.getHttpAuthSchemeEndpointRuleSetPlugin)(this.config, {
        httpAuthSchemeParametersProvider: this.getDefaultHttpAuthSchemeParametersProvider(),
        identityProviderConfigProvider: this.getIdentityProviderConfigProvider()
      })
    );
    this.middlewareStack.use((0, import_core.getHttpSigningPlugin)(this.config));
  }
  /**
   * Destroy underlying resources, like sockets. It's usually not necessary to do this.
   * However in Node.js, it's best to explicitly shut down the client's agent when it is no longer needed.
   * Otherwise, sockets might stay open for quite a long time before the server terminates them.
   */
  destroy() {
    super.destroy();
  }
  getDefaultHttpAuthSchemeParametersProvider() {
    return import_httpAuthSchemeProvider.defaultInspectorHttpAuthSchemeParametersProvider;
  }
  getIdentityProviderConfigProvider() {
    return async (config) => new import_core.DefaultIdentityProviderConfig({
      "aws.auth#sigv4": config.credentials
    });
  }
};
__name(_InspectorClient, "InspectorClient");
var InspectorClient = _InspectorClient;

// src/Inspector.ts


// src/commands/AddAttributesToFindingsCommand.ts

var import_middleware_serde = require("@smithy/middleware-serde");

var import_types = require("@smithy/types");

// src/protocols/Aws_json1_1.ts
var import_core2 = require("@aws-sdk/core");



// src/models/InspectorServiceException.ts

var _InspectorServiceException = class _InspectorServiceException extends import_smithy_client.ServiceException {
  /**
   * @internal
   */
  constructor(options) {
    super(options);
    Object.setPrototypeOf(this, _InspectorServiceException.prototype);
  }
};
__name(_InspectorServiceException, "InspectorServiceException");
var InspectorServiceException = _InspectorServiceException;

// src/models/models_0.ts
var AccessDeniedErrorCode = {
  ACCESS_DENIED_TO_ASSESSMENT_RUN: "ACCESS_DENIED_TO_ASSESSMENT_RUN",
  ACCESS_DENIED_TO_ASSESSMENT_TARGET: "ACCESS_DENIED_TO_ASSESSMENT_TARGET",
  ACCESS_DENIED_TO_ASSESSMENT_TEMPLATE: "ACCESS_DENIED_TO_ASSESSMENT_TEMPLATE",
  ACCESS_DENIED_TO_FINDING: "ACCESS_DENIED_TO_FINDING",
  ACCESS_DENIED_TO_IAM_ROLE: "ACCESS_DENIED_TO_IAM_ROLE",
  ACCESS_DENIED_TO_RESOURCE_GROUP: "ACCESS_DENIED_TO_RESOURCE_GROUP",
  ACCESS_DENIED_TO_RULES_PACKAGE: "ACCESS_DENIED_TO_RULES_PACKAGE",
  ACCESS_DENIED_TO_SNS_TOPIC: "ACCESS_DENIED_TO_SNS_TOPIC"
};
var _AccessDeniedException = class _AccessDeniedException extends InspectorServiceException {
  /**
   * @internal
   */
  constructor(opts) {
    super({
      name: "AccessDeniedException",
      $fault: "client",
      ...opts
    });
    this.name = "AccessDeniedException";
    this.$fault = "client";
    Object.setPrototypeOf(this, _AccessDeniedException.prototype);
    this.errorCode = opts.errorCode;
    this.canRetry = opts.canRetry;
  }
};
__name(_AccessDeniedException, "AccessDeniedException");
var AccessDeniedException = _AccessDeniedException;
var FailedItemErrorCode = {
  ACCESS_DENIED: "ACCESS_DENIED",
  DUPLICATE_ARN: "DUPLICATE_ARN",
  INTERNAL_ERROR: "INTERNAL_ERROR",
  INVALID_ARN: "INVALID_ARN",
  ITEM_DOES_NOT_EXIST: "ITEM_DOES_NOT_EXIST",
  LIMIT_EXCEEDED: "LIMIT_EXCEEDED"
};
var _InternalException = class _InternalException extends InspectorServiceException {
  /**
   * @internal
   */
  constructor(opts) {
    super({
      name: "InternalException",
      $fault: "server",
      ...opts
    });
    this.name = "InternalException";
    this.$fault = "server";
    Object.setPrototypeOf(this, _InternalException.prototype);
    this.canRetry = opts.canRetry;
  }
};
__name(_InternalException, "InternalException");
var InternalException = _InternalException;
var InvalidInputErrorCode = {
  ASSESSMENT_TARGET_NAME_ALREADY_TAKEN: "ASSESSMENT_TARGET_NAME_ALREADY_TAKEN",
  ASSESSMENT_TEMPLATE_NAME_ALREADY_TAKEN: "ASSESSMENT_TEMPLATE_NAME_ALREADY_TAKEN",
  INVALID_AGENT_ID: "INVALID_AGENT_ID",
  INVALID_ASSESSMENT_RUN_ARN: "INVALID_ASSESSMENT_RUN_ARN",
  INVALID_ASSESSMENT_RUN_COMPLETION_TIME_RANGE: "INVALID_ASSESSMENT_RUN_COMPLETION_TIME_RANGE",
  INVALID_ASSESSMENT_RUN_DURATION_RANGE: "INVALID_ASSESSMENT_RUN_DURATION_RANGE",
  INVALID_ASSESSMENT_RUN_START_TIME_RANGE: "INVALID_ASSESSMENT_RUN_START_TIME_RANGE",
  INVALID_ASSESSMENT_RUN_STATE: "INVALID_ASSESSMENT_RUN_STATE",
  INVALID_ASSESSMENT_RUN_STATE_CHANGE_TIME_RANGE: "INVALID_ASSESSMENT_RUN_STATE_CHANGE_TIME_RANGE",
  INVALID_ASSESSMENT_TARGET_ARN: "INVALID_ASSESSMENT_TARGET_ARN",
  INVALID_ASSESSMENT_TARGET_NAME: "INVALID_ASSESSMENT_TARGET_NAME",
  INVALID_ASSESSMENT_TARGET_NAME_PATTERN: "INVALID_ASSESSMENT_TARGET_NAME_PATTERN",
  INVALID_ASSESSMENT_TEMPLATE_ARN: "INVALID_ASSESSMENT_TEMPLATE_ARN",
  INVALID_ASSESSMENT_TEMPLATE_DURATION: "INVALID_ASSESSMENT_TEMPLATE_DURATION",
  INVALID_ASSESSMENT_TEMPLATE_DURATION_RANGE: "INVALID_ASSESSMENT_TEMPLATE_DURATION_RANGE",
  INVALID_ASSESSMENT_TEMPLATE_NAME: "INVALID_ASSESSMENT_TEMPLATE_NAME",
  INVALID_ASSESSMENT_TEMPLATE_NAME_PATTERN: "INVALID_ASSESSMENT_TEMPLATE_NAME_PATTERN",
  INVALID_ATTRIBUTE: "INVALID_ATTRIBUTE",
  INVALID_AUTO_SCALING_GROUP: "INVALID_AUTO_SCALING_GROUP",
  INVALID_EVENT: "INVALID_EVENT",
  INVALID_FINDING_ARN: "INVALID_FINDING_ARN",
  INVALID_IAM_ROLE_ARN: "INVALID_IAM_ROLE_ARN",
  INVALID_LOCALE: "INVALID_LOCALE",
  INVALID_MAX_RESULTS: "INVALID_MAX_RESULTS",
  INVALID_NUMBER_OF_AGENT_IDS: "INVALID_NUMBER_OF_AGENT_IDS",
  INVALID_NUMBER_OF_ASSESSMENT_RUN_ARNS: "INVALID_NUMBER_OF_ASSESSMENT_RUN_ARNS",
  INVALID_NUMBER_OF_ASSESSMENT_RUN_STATES: "INVALID_NUMBER_OF_ASSESSMENT_RUN_STATES",
  INVALID_NUMBER_OF_ASSESSMENT_TARGET_ARNS: "INVALID_NUMBER_OF_ASSESSMENT_TARGET_ARNS",
  INVALID_NUMBER_OF_ASSESSMENT_TEMPLATE_ARNS: "INVALID_NUMBER_OF_ASSESSMENT_TEMPLATE_ARNS",
  INVALID_NUMBER_OF_ATTRIBUTES: "INVALID_NUMBER_OF_ATTRIBUTES",
  INVALID_NUMBER_OF_AUTO_SCALING_GROUPS: "INVALID_NUMBER_OF_AUTO_SCALING_GROUPS",
  INVALID_NUMBER_OF_FINDING_ARNS: "INVALID_NUMBER_OF_FINDING_ARNS",
  INVALID_NUMBER_OF_RESOURCE_GROUP_ARNS: "INVALID_NUMBER_OF_RESOURCE_GROUP_ARNS",
  INVALID_NUMBER_OF_RESOURCE_GROUP_TAGS: "INVALID_NUMBER_OF_RESOURCE_GROUP_TAGS",
  INVALID_NUMBER_OF_RULES_PACKAGE_ARNS: "INVALID_NUMBER_OF_RULES_PACKAGE_ARNS",
  INVALID_NUMBER_OF_RULE_NAMES: "INVALID_NUMBER_OF_RULE_NAMES",
  INVALID_NUMBER_OF_SEVERITIES: "INVALID_NUMBER_OF_SEVERITIES",
  INVALID_NUMBER_OF_TAGS: "INVALID_NUMBER_OF_TAGS",
  INVALID_NUMBER_OF_USER_ATTRIBUTES: "INVALID_NUMBER_OF_USER_ATTRIBUTES",
  INVALID_PAGINATION_TOKEN: "INVALID_PAGINATION_TOKEN",
  INVALID_RESOURCE_ARN: "INVALID_RESOURCE_ARN",
  INVALID_RESOURCE_GROUP_ARN: "INVALID_RESOURCE_GROUP_ARN",
  INVALID_RESOURCE_GROUP_TAG_KEY: "INVALID_RESOURCE_GROUP_TAG_KEY",
  INVALID_RESOURCE_GROUP_TAG_VALUE: "INVALID_RESOURCE_GROUP_TAG_VALUE",
  INVALID_RULES_PACKAGE_ARN: "INVALID_RULES_PACKAGE_ARN",
  INVALID_RULE_NAME: "INVALID_RULE_NAME",
  INVALID_SEVERITY: "INVALID_SEVERITY",
  INVALID_SNS_TOPIC_ARN: "INVALID_SNS_TOPIC_ARN",
  INVALID_TAG: "INVALID_TAG",
  INVALID_TAG_KEY: "INVALID_TAG_KEY",
  INVALID_TAG_VALUE: "INVALID_TAG_VALUE",
  INVALID_USER_ATTRIBUTE: "INVALID_USER_ATTRIBUTE",
  INVALID_USER_ATTRIBUTE_KEY: "INVALID_USER_ATTRIBUTE_KEY",
  INVALID_USER_ATTRIBUTE_VALUE: "INVALID_USER_ATTRIBUTE_VALUE"
};
var _InvalidInputException = class _InvalidInputException extends InspectorServiceException {
  /**
   * @internal
   */
  constructor(opts) {
    super({
      name: "InvalidInputException",
      $fault: "client",
      ...opts
    });
    this.name = "InvalidInputException";
    this.$fault = "client";
    Object.setPrototypeOf(this, _InvalidInputException.prototype);
    this.errorCode = opts.errorCode;
    this.canRetry = opts.canRetry;
  }
};
__name(_InvalidInputException, "InvalidInputException");
var InvalidInputException = _InvalidInputException;
var NoSuchEntityErrorCode = {
  ASSESSMENT_RUN_DOES_NOT_EXIST: "ASSESSMENT_RUN_DOES_NOT_EXIST",
  ASSESSMENT_TARGET_DOES_NOT_EXIST: "ASSESSMENT_TARGET_DOES_NOT_EXIST",
  ASSESSMENT_TEMPLATE_DOES_NOT_EXIST: "ASSESSMENT_TEMPLATE_DOES_NOT_EXIST",
  FINDING_DOES_NOT_EXIST: "FINDING_DOES_NOT_EXIST",
  IAM_ROLE_DOES_NOT_EXIST: "IAM_ROLE_DOES_NOT_EXIST",
  RESOURCE_GROUP_DOES_NOT_EXIST: "RESOURCE_GROUP_DOES_NOT_EXIST",
  RULES_PACKAGE_DOES_NOT_EXIST: "RULES_PACKAGE_DOES_NOT_EXIST",
  SNS_TOPIC_DOES_NOT_EXIST: "SNS_TOPIC_DOES_NOT_EXIST"
};
var _NoSuchEntityException = class _NoSuchEntityException extends InspectorServiceException {
  /**
   * @internal
   */
  constructor(opts) {
    super({
      name: "NoSuchEntityException",
      $fault: "client",
      ...opts
    });
    this.name = "NoSuchEntityException";
    this.$fault = "client";
    Object.setPrototypeOf(this, _NoSuchEntityException.prototype);
    this.errorCode = opts.errorCode;
    this.canRetry = opts.canRetry;
  }
};
__name(_NoSuchEntityException, "NoSuchEntityException");
var NoSuchEntityException = _NoSuchEntityException;
var _ServiceTemporarilyUnavailableException = class _ServiceTemporarilyUnavailableException extends InspectorServiceException {
  /**
   * @internal
   */
  constructor(opts) {
    super({
      name: "ServiceTemporarilyUnavailableException",
      $fault: "server",
      ...opts
    });
    this.name = "ServiceTemporarilyUnavailableException";
    this.$fault = "server";
    Object.setPrototypeOf(this, _ServiceTemporarilyUnavailableException.prototype);
    this.canRetry = opts.canRetry;
  }
};
__name(_ServiceTemporarilyUnavailableException, "ServiceTemporarilyUnavailableException");
var ServiceTemporarilyUnavailableException = _ServiceTemporarilyUnavailableException;
var AgentHealthCode = {
  IDLE: "IDLE",
  RUNNING: "RUNNING",
  SHUTDOWN: "SHUTDOWN",
  THROTTLED: "THROTTLED",
  UNHEALTHY: "UNHEALTHY",
  UNKNOWN: "UNKNOWN"
};
var AgentHealth = {
  HEALTHY: "HEALTHY",
  UNHEALTHY: "UNHEALTHY",
  UNKNOWN: "UNKNOWN"
};
var _AgentsAlreadyRunningAssessmentException = class _AgentsAlreadyRunningAssessmentException extends InspectorServiceException {
  /**
   * @internal
   */
  constructor(opts) {
    super({
      name: "AgentsAlreadyRunningAssessmentException",
      $fault: "client",
      ...opts
    });
    this.name = "AgentsAlreadyRunningAssessmentException";
    this.$fault = "client";
    Object.setPrototypeOf(this, _AgentsAlreadyRunningAssessmentException.prototype);
    this.agents = opts.agents;
    this.agentsTruncated = opts.agentsTruncated;
    this.canRetry = opts.canRetry;
  }
};
__name(_AgentsAlreadyRunningAssessmentException, "AgentsAlreadyRunningAssessmentException");
var AgentsAlreadyRunningAssessmentException = _AgentsAlreadyRunningAssessmentException;
var Severity = {
  HIGH: "High",
  INFORMATIONAL: "Informational",
  LOW: "Low",
  MEDIUM: "Medium",
  UNDEFINED: "Undefined"
};
var InspectorEvent = {
  ASSESSMENT_RUN_COMPLETED: "ASSESSMENT_RUN_COMPLETED",
  ASSESSMENT_RUN_STARTED: "ASSESSMENT_RUN_STARTED",
  ASSESSMENT_RUN_STATE_CHANGED: "ASSESSMENT_RUN_STATE_CHANGED",
  FINDING_REPORTED: "FINDING_REPORTED",
  OTHER: "OTHER"
};
var AssessmentRunNotificationSnsStatusCode = {
  ACCESS_DENIED: "ACCESS_DENIED",
  INTERNAL_ERROR: "INTERNAL_ERROR",
  SUCCESS: "SUCCESS",
  TOPIC_DOES_NOT_EXIST: "TOPIC_DOES_NOT_EXIST"
};
var AssessmentRunState = {
  CANCELED: "CANCELED",
  COLLECTING_DATA: "COLLECTING_DATA",
  COMPLETED: "COMPLETED",
  COMPLETED_WITH_ERRORS: "COMPLETED_WITH_ERRORS",
  CREATED: "CREATED",
  DATA_COLLECTED: "DATA_COLLECTED",
  ERROR: "ERROR",
  EVALUATING_RULES: "EVALUATING_RULES",
  FAILED: "FAILED",
  START_DATA_COLLECTION_IN_PROGRESS: "START_DATA_COLLECTION_IN_PROGRESS",
  START_DATA_COLLECTION_PENDING: "START_DATA_COLLECTION_PENDING",
  START_EVALUATING_RULES_PENDING: "START_EVALUATING_RULES_PENDING",
  STOP_DATA_COLLECTION_PENDING: "STOP_DATA_COLLECTION_PENDING"
};
var _AssessmentRunInProgressException = class _AssessmentRunInProgressException extends InspectorServiceException {
  /**
   * @internal
   */
  constructor(opts) {
    super({
      name: "AssessmentRunInProgressException",
      $fault: "client",
      ...opts
    });
    this.name = "AssessmentRunInProgressException";
    this.$fault = "client";
    Object.setPrototypeOf(this, _AssessmentRunInProgressException.prototype);
    this.assessmentRunArns = opts.assessmentRunArns;
    this.assessmentRunArnsTruncated = opts.assessmentRunArnsTruncated;
    this.canRetry = opts.canRetry;
  }
};
__name(_AssessmentRunInProgressException, "AssessmentRunInProgressException");
var AssessmentRunInProgressException = _AssessmentRunInProgressException;
var AssetType = {
  EC2_INSTANCE: "ec2-instance"
};
var InvalidCrossAccountRoleErrorCode = {
  ROLE_DOES_NOT_EXIST_OR_INVALID_TRUST_RELATIONSHIP: "ROLE_DOES_NOT_EXIST_OR_INVALID_TRUST_RELATIONSHIP",
  ROLE_DOES_NOT_HAVE_CORRECT_POLICY: "ROLE_DOES_NOT_HAVE_CORRECT_POLICY"
};
var _InvalidCrossAccountRoleException = class _InvalidCrossAccountRoleException extends InspectorServiceException {
  /**
   * @internal
   */
  constructor(opts) {
    super({
      name: "InvalidCrossAccountRoleException",
      $fault: "client",
      ...opts
    });
    this.name = "InvalidCrossAccountRoleException";
    this.$fault = "client";
    Object.setPrototypeOf(this, _InvalidCrossAccountRoleException.prototype);
    this.errorCode = opts.errorCode;
    this.canRetry = opts.canRetry;
  }
};
__name(_InvalidCrossAccountRoleException, "InvalidCrossAccountRoleException");
var InvalidCrossAccountRoleException = _InvalidCrossAccountRoleException;
var LimitExceededErrorCode = {
  ASSESSMENT_RUN_LIMIT_EXCEEDED: "ASSESSMENT_RUN_LIMIT_EXCEEDED",
  ASSESSMENT_TARGET_LIMIT_EXCEEDED: "ASSESSMENT_TARGET_LIMIT_EXCEEDED",
  ASSESSMENT_TEMPLATE_LIMIT_EXCEEDED: "ASSESSMENT_TEMPLATE_LIMIT_EXCEEDED",
  EVENT_SUBSCRIPTION_LIMIT_EXCEEDED: "EVENT_SUBSCRIPTION_LIMIT_EXCEEDED",
  RESOURCE_GROUP_LIMIT_EXCEEDED: "RESOURCE_GROUP_LIMIT_EXCEEDED"
};
var _LimitExceededException = class _LimitExceededException extends InspectorServiceException {
  /**
   * @internal
   */
  constructor(opts) {
    super({
      name: "LimitExceededException",
      $fault: "client",
      ...opts
    });
    this.name = "LimitExceededException";
    this.$fault = "client";
    Object.setPrototypeOf(this, _LimitExceededException.prototype);
    this.errorCode = opts.errorCode;
    this.canRetry = opts.canRetry;
  }
};
__name(_LimitExceededException, "LimitExceededException");
var LimitExceededException = _LimitExceededException;
var _PreviewGenerationInProgressException = class _PreviewGenerationInProgressException extends InspectorServiceException {
  /**
   * @internal
   */
  constructor(opts) {
    super({
      name: "PreviewGenerationInProgressException",
      $fault: "client",
      ...opts
    });
    this.name = "PreviewGenerationInProgressException";
    this.$fault = "client";
    Object.setPrototypeOf(this, _PreviewGenerationInProgressException.prototype);
  }
};
__name(_PreviewGenerationInProgressException, "PreviewGenerationInProgressException");
var PreviewGenerationInProgressException = _PreviewGenerationInProgressException;
var Locale = {
  EN_US: "EN_US"
};
var ScopeType = {
  INSTANCE_ID: "INSTANCE_ID",
  RULES_PACKAGE_ARN: "RULES_PACKAGE_ARN"
};
var ReportFileFormat = {
  HTML: "HTML",
  PDF: "PDF"
};
var ReportType = {
  FINDING: "FINDING",
  FULL: "FULL"
};
var ReportStatus = {
  COMPLETED: "COMPLETED",
  FAILED: "FAILED",
  WORK_IN_PROGRESS: "WORK_IN_PROGRESS"
};
var _UnsupportedFeatureException = class _UnsupportedFeatureException extends InspectorServiceException {
  /**
   * @internal
   */
  constructor(opts) {
    super({
      name: "UnsupportedFeatureException",
      $fault: "client",
      ...opts
    });
    this.name = "UnsupportedFeatureException";
    this.$fault = "client";
    Object.setPrototypeOf(this, _UnsupportedFeatureException.prototype);
    this.canRetry = opts.canRetry;
  }
};
__name(_UnsupportedFeatureException, "UnsupportedFeatureException");
var UnsupportedFeatureException = _UnsupportedFeatureException;
var PreviewStatus = {
  COMPLETED: "COMPLETED",
  WORK_IN_PROGRESS: "WORK_IN_PROGRESS"
};
var StopAction = {
  SKIP_EVALUATION: "SKIP_EVALUATION",
  START_EVALUATION: "START_EVALUATION"
};

// src/protocols/Aws_json1_1.ts
var se_AddAttributesToFindingsCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("AddAttributesToFindings");
  let body;
  body = JSON.stringify((0, import_smithy_client._json)(input));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_AddAttributesToFindingsCommand");
var se_CreateAssessmentTargetCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("CreateAssessmentTarget");
  let body;
  body = JSON.stringify((0, import_smithy_client._json)(input));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_CreateAssessmentTargetCommand");
var se_CreateAssessmentTemplateCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("CreateAssessmentTemplate");
  let body;
  body = JSON.stringify((0, import_smithy_client._json)(input));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_CreateAssessmentTemplateCommand");
var se_CreateExclusionsPreviewCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("CreateExclusionsPreview");
  let body;
  body = JSON.stringify((0, import_smithy_client._json)(input));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_CreateExclusionsPreviewCommand");
var se_CreateResourceGroupCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("CreateResourceGroup");
  let body;
  body = JSON.stringify((0, import_smithy_client._json)(input));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_CreateResourceGroupCommand");
var se_DeleteAssessmentRunCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("DeleteAssessmentRun");
  let body;
  body = JSON.stringify((0, import_smithy_client._json)(input));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_DeleteAssessmentRunCommand");
var se_DeleteAssessmentTargetCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("DeleteAssessmentTarget");
  let body;
  body = JSON.stringify((0, import_smithy_client._json)(input));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_DeleteAssessmentTargetCommand");
var se_DeleteAssessmentTemplateCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("DeleteAssessmentTemplate");
  let body;
  body = JSON.stringify((0, import_smithy_client._json)(input));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_DeleteAssessmentTemplateCommand");
var se_DescribeAssessmentRunsCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("DescribeAssessmentRuns");
  let body;
  body = JSON.stringify((0, import_smithy_client._json)(input));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_DescribeAssessmentRunsCommand");
var se_DescribeAssessmentTargetsCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("DescribeAssessmentTargets");
  let body;
  body = JSON.stringify((0, import_smithy_client._json)(input));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_DescribeAssessmentTargetsCommand");
var se_DescribeAssessmentTemplatesCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("DescribeAssessmentTemplates");
  let body;
  body = JSON.stringify((0, import_smithy_client._json)(input));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_DescribeAssessmentTemplatesCommand");
var se_DescribeCrossAccountAccessRoleCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("DescribeCrossAccountAccessRole");
  const body = "{}";
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_DescribeCrossAccountAccessRoleCommand");
var se_DescribeExclusionsCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("DescribeExclusions");
  let body;
  body = JSON.stringify((0, import_smithy_client._json)(input));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_DescribeExclusionsCommand");
var se_DescribeFindingsCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("DescribeFindings");
  let body;
  body = JSON.stringify((0, import_smithy_client._json)(input));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_DescribeFindingsCommand");
var se_DescribeResourceGroupsCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("DescribeResourceGroups");
  let body;
  body = JSON.stringify((0, import_smithy_client._json)(input));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_DescribeResourceGroupsCommand");
var se_DescribeRulesPackagesCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("DescribeRulesPackages");
  let body;
  body = JSON.stringify((0, import_smithy_client._json)(input));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_DescribeRulesPackagesCommand");
var se_GetAssessmentReportCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("GetAssessmentReport");
  let body;
  body = JSON.stringify((0, import_smithy_client._json)(input));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_GetAssessmentReportCommand");
var se_GetExclusionsPreviewCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("GetExclusionsPreview");
  let body;
  body = JSON.stringify((0, import_smithy_client._json)(input));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_GetExclusionsPreviewCommand");
var se_GetTelemetryMetadataCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("GetTelemetryMetadata");
  let body;
  body = JSON.stringify((0, import_smithy_client._json)(input));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_GetTelemetryMetadataCommand");
var se_ListAssessmentRunAgentsCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("ListAssessmentRunAgents");
  let body;
  body = JSON.stringify((0, import_smithy_client._json)(input));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_ListAssessmentRunAgentsCommand");
var se_ListAssessmentRunsCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("ListAssessmentRuns");
  let body;
  body = JSON.stringify(se_ListAssessmentRunsRequest(input, context));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_ListAssessmentRunsCommand");
var se_ListAssessmentTargetsCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("ListAssessmentTargets");
  let body;
  body = JSON.stringify((0, import_smithy_client._json)(input));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_ListAssessmentTargetsCommand");
var se_ListAssessmentTemplatesCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("ListAssessmentTemplates");
  let body;
  body = JSON.stringify((0, import_smithy_client._json)(input));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_ListAssessmentTemplatesCommand");
var se_ListEventSubscriptionsCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("ListEventSubscriptions");
  let body;
  body = JSON.stringify((0, import_smithy_client._json)(input));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_ListEventSubscriptionsCommand");
var se_ListExclusionsCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("ListExclusions");
  let body;
  body = JSON.stringify((0, import_smithy_client._json)(input));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_ListExclusionsCommand");
var se_ListFindingsCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("ListFindings");
  let body;
  body = JSON.stringify(se_ListFindingsRequest(input, context));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_ListFindingsCommand");
var se_ListRulesPackagesCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("ListRulesPackages");
  let body;
  body = JSON.stringify((0, import_smithy_client._json)(input));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_ListRulesPackagesCommand");
var se_ListTagsForResourceCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("ListTagsForResource");
  let body;
  body = JSON.stringify((0, import_smithy_client._json)(input));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_ListTagsForResourceCommand");
var se_PreviewAgentsCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("PreviewAgents");
  let body;
  body = JSON.stringify((0, import_smithy_client._json)(input));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_PreviewAgentsCommand");
var se_RegisterCrossAccountAccessRoleCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("RegisterCrossAccountAccessRole");
  let body;
  body = JSON.stringify((0, import_smithy_client._json)(input));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_RegisterCrossAccountAccessRoleCommand");
var se_RemoveAttributesFromFindingsCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("RemoveAttributesFromFindings");
  let body;
  body = JSON.stringify((0, import_smithy_client._json)(input));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_RemoveAttributesFromFindingsCommand");
var se_SetTagsForResourceCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("SetTagsForResource");
  let body;
  body = JSON.stringify((0, import_smithy_client._json)(input));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_SetTagsForResourceCommand");
var se_StartAssessmentRunCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("StartAssessmentRun");
  let body;
  body = JSON.stringify((0, import_smithy_client._json)(input));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_StartAssessmentRunCommand");
var se_StopAssessmentRunCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("StopAssessmentRun");
  let body;
  body = JSON.stringify((0, import_smithy_client._json)(input));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_StopAssessmentRunCommand");
var se_SubscribeToEventCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("SubscribeToEvent");
  let body;
  body = JSON.stringify((0, import_smithy_client._json)(input));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_SubscribeToEventCommand");
var se_UnsubscribeFromEventCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("UnsubscribeFromEvent");
  let body;
  body = JSON.stringify((0, import_smithy_client._json)(input));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_UnsubscribeFromEventCommand");
var se_UpdateAssessmentTargetCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("UpdateAssessmentTarget");
  let body;
  body = JSON.stringify((0, import_smithy_client._json)(input));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_UpdateAssessmentTargetCommand");
var de_AddAttributesToFindingsCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const data = await (0, import_core2.parseJsonBody)(output.body, context);
  let contents = {};
  contents = (0, import_smithy_client._json)(data);
  const response = {
    $metadata: deserializeMetadata(output),
    ...contents
  };
  return response;
}, "de_AddAttributesToFindingsCommand");
var de_CreateAssessmentTargetCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const data = await (0, import_core2.parseJsonBody)(output.body, context);
  let contents = {};
  contents = (0, import_smithy_client._json)(data);
  const response = {
    $metadata: deserializeMetadata(output),
    ...contents
  };
  return response;
}, "de_CreateAssessmentTargetCommand");
var de_CreateAssessmentTemplateCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const data = await (0, import_core2.parseJsonBody)(output.body, context);
  let contents = {};
  contents = (0, import_smithy_client._json)(data);
  const response = {
    $metadata: deserializeMetadata(output),
    ...contents
  };
  return response;
}, "de_CreateAssessmentTemplateCommand");
var de_CreateExclusionsPreviewCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const data = await (0, import_core2.parseJsonBody)(output.body, context);
  let contents = {};
  contents = (0, import_smithy_client._json)(data);
  const response = {
    $metadata: deserializeMetadata(output),
    ...contents
  };
  return response;
}, "de_CreateExclusionsPreviewCommand");
var de_CreateResourceGroupCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const data = await (0, import_core2.parseJsonBody)(output.body, context);
  let contents = {};
  contents = (0, import_smithy_client._json)(data);
  const response = {
    $metadata: deserializeMetadata(output),
    ...contents
  };
  return response;
}, "de_CreateResourceGroupCommand");
var de_DeleteAssessmentRunCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  await (0, import_smithy_client.collectBody)(output.body, context);
  const response = {
    $metadata: deserializeMetadata(output)
  };
  return response;
}, "de_DeleteAssessmentRunCommand");
var de_DeleteAssessmentTargetCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  await (0, import_smithy_client.collectBody)(output.body, context);
  const response = {
    $metadata: deserializeMetadata(output)
  };
  return response;
}, "de_DeleteAssessmentTargetCommand");
var de_DeleteAssessmentTemplateCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  await (0, import_smithy_client.collectBody)(output.body, context);
  const response = {
    $metadata: deserializeMetadata(output)
  };
  return response;
}, "de_DeleteAssessmentTemplateCommand");
var de_DescribeAssessmentRunsCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const data = await (0, import_core2.parseJsonBody)(output.body, context);
  let contents = {};
  contents = de_DescribeAssessmentRunsResponse(data, context);
  const response = {
    $metadata: deserializeMetadata(output),
    ...contents
  };
  return response;
}, "de_DescribeAssessmentRunsCommand");
var de_DescribeAssessmentTargetsCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const data = await (0, import_core2.parseJsonBody)(output.body, context);
  let contents = {};
  contents = de_DescribeAssessmentTargetsResponse(data, context);
  const response = {
    $metadata: deserializeMetadata(output),
    ...contents
  };
  return response;
}, "de_DescribeAssessmentTargetsCommand");
var de_DescribeAssessmentTemplatesCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const data = await (0, import_core2.parseJsonBody)(output.body, context);
  let contents = {};
  contents = de_DescribeAssessmentTemplatesResponse(data, context);
  const response = {
    $metadata: deserializeMetadata(output),
    ...contents
  };
  return response;
}, "de_DescribeAssessmentTemplatesCommand");
var de_DescribeCrossAccountAccessRoleCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const data = await (0, import_core2.parseJsonBody)(output.body, context);
  let contents = {};
  contents = de_DescribeCrossAccountAccessRoleResponse(data, context);
  const response = {
    $metadata: deserializeMetadata(output),
    ...contents
  };
  return response;
}, "de_DescribeCrossAccountAccessRoleCommand");
var de_DescribeExclusionsCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const data = await (0, import_core2.parseJsonBody)(output.body, context);
  let contents = {};
  contents = (0, import_smithy_client._json)(data);
  const response = {
    $metadata: deserializeMetadata(output),
    ...contents
  };
  return response;
}, "de_DescribeExclusionsCommand");
var de_DescribeFindingsCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const data = await (0, import_core2.parseJsonBody)(output.body, context);
  let contents = {};
  contents = de_DescribeFindingsResponse(data, context);
  const response = {
    $metadata: deserializeMetadata(output),
    ...contents
  };
  return response;
}, "de_DescribeFindingsCommand");
var de_DescribeResourceGroupsCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const data = await (0, import_core2.parseJsonBody)(output.body, context);
  let contents = {};
  contents = de_DescribeResourceGroupsResponse(data, context);
  const response = {
    $metadata: deserializeMetadata(output),
    ...contents
  };
  return response;
}, "de_DescribeResourceGroupsCommand");
var de_DescribeRulesPackagesCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const data = await (0, import_core2.parseJsonBody)(output.body, context);
  let contents = {};
  contents = (0, import_smithy_client._json)(data);
  const response = {
    $metadata: deserializeMetadata(output),
    ...contents
  };
  return response;
}, "de_DescribeRulesPackagesCommand");
var de_GetAssessmentReportCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const data = await (0, import_core2.parseJsonBody)(output.body, context);
  let contents = {};
  contents = (0, import_smithy_client._json)(data);
  const response = {
    $metadata: deserializeMetadata(output),
    ...contents
  };
  return response;
}, "de_GetAssessmentReportCommand");
var de_GetExclusionsPreviewCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const data = await (0, import_core2.parseJsonBody)(output.body, context);
  let contents = {};
  contents = (0, import_smithy_client._json)(data);
  const response = {
    $metadata: deserializeMetadata(output),
    ...contents
  };
  return response;
}, "de_GetExclusionsPreviewCommand");
var de_GetTelemetryMetadataCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const data = await (0, import_core2.parseJsonBody)(output.body, context);
  let contents = {};
  contents = (0, import_smithy_client._json)(data);
  const response = {
    $metadata: deserializeMetadata(output),
    ...contents
  };
  return response;
}, "de_GetTelemetryMetadataCommand");
var de_ListAssessmentRunAgentsCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const data = await (0, import_core2.parseJsonBody)(output.body, context);
  let contents = {};
  contents = (0, import_smithy_client._json)(data);
  const response = {
    $metadata: deserializeMetadata(output),
    ...contents
  };
  return response;
}, "de_ListAssessmentRunAgentsCommand");
var de_ListAssessmentRunsCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const data = await (0, import_core2.parseJsonBody)(output.body, context);
  let contents = {};
  contents = (0, import_smithy_client._json)(data);
  const response = {
    $metadata: deserializeMetadata(output),
    ...contents
  };
  return response;
}, "de_ListAssessmentRunsCommand");
var de_ListAssessmentTargetsCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const data = await (0, import_core2.parseJsonBody)(output.body, context);
  let contents = {};
  contents = (0, import_smithy_client._json)(data);
  const response = {
    $metadata: deserializeMetadata(output),
    ...contents
  };
  return response;
}, "de_ListAssessmentTargetsCommand");
var de_ListAssessmentTemplatesCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const data = await (0, import_core2.parseJsonBody)(output.body, context);
  let contents = {};
  contents = (0, import_smithy_client._json)(data);
  const response = {
    $metadata: deserializeMetadata(output),
    ...contents
  };
  return response;
}, "de_ListAssessmentTemplatesCommand");
var de_ListEventSubscriptionsCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const data = await (0, import_core2.parseJsonBody)(output.body, context);
  let contents = {};
  contents = de_ListEventSubscriptionsResponse(data, context);
  const response = {
    $metadata: deserializeMetadata(output),
    ...contents
  };
  return response;
}, "de_ListEventSubscriptionsCommand");
var de_ListExclusionsCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const data = await (0, import_core2.parseJsonBody)(output.body, context);
  let contents = {};
  contents = (0, import_smithy_client._json)(data);
  const response = {
    $metadata: deserializeMetadata(output),
    ...contents
  };
  return response;
}, "de_ListExclusionsCommand");
var de_ListFindingsCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const data = await (0, import_core2.parseJsonBody)(output.body, context);
  let contents = {};
  contents = (0, import_smithy_client._json)(data);
  const response = {
    $metadata: deserializeMetadata(output),
    ...contents
  };
  return response;
}, "de_ListFindingsCommand");
var de_ListRulesPackagesCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const data = await (0, import_core2.parseJsonBody)(output.body, context);
  let contents = {};
  contents = (0, import_smithy_client._json)(data);
  const response = {
    $metadata: deserializeMetadata(output),
    ...contents
  };
  return response;
}, "de_ListRulesPackagesCommand");
var de_ListTagsForResourceCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const data = await (0, import_core2.parseJsonBody)(output.body, context);
  let contents = {};
  contents = (0, import_smithy_client._json)(data);
  const response = {
    $metadata: deserializeMetadata(output),
    ...contents
  };
  return response;
}, "de_ListTagsForResourceCommand");
var de_PreviewAgentsCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const data = await (0, import_core2.parseJsonBody)(output.body, context);
  let contents = {};
  contents = (0, import_smithy_client._json)(data);
  const response = {
    $metadata: deserializeMetadata(output),
    ...contents
  };
  return response;
}, "de_PreviewAgentsCommand");
var de_RegisterCrossAccountAccessRoleCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  await (0, import_smithy_client.collectBody)(output.body, context);
  const response = {
    $metadata: deserializeMetadata(output)
  };
  return response;
}, "de_RegisterCrossAccountAccessRoleCommand");
var de_RemoveAttributesFromFindingsCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const data = await (0, import_core2.parseJsonBody)(output.body, context);
  let contents = {};
  contents = (0, import_smithy_client._json)(data);
  const response = {
    $metadata: deserializeMetadata(output),
    ...contents
  };
  return response;
}, "de_RemoveAttributesFromFindingsCommand");
var de_SetTagsForResourceCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  await (0, import_smithy_client.collectBody)(output.body, context);
  const response = {
    $metadata: deserializeMetadata(output)
  };
  return response;
}, "de_SetTagsForResourceCommand");
var de_StartAssessmentRunCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const data = await (0, import_core2.parseJsonBody)(output.body, context);
  let contents = {};
  contents = (0, import_smithy_client._json)(data);
  const response = {
    $metadata: deserializeMetadata(output),
    ...contents
  };
  return response;
}, "de_StartAssessmentRunCommand");
var de_StopAssessmentRunCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  await (0, import_smithy_client.collectBody)(output.body, context);
  const response = {
    $metadata: deserializeMetadata(output)
  };
  return response;
}, "de_StopAssessmentRunCommand");
var de_SubscribeToEventCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  await (0, import_smithy_client.collectBody)(output.body, context);
  const response = {
    $metadata: deserializeMetadata(output)
  };
  return response;
}, "de_SubscribeToEventCommand");
var de_UnsubscribeFromEventCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  await (0, import_smithy_client.collectBody)(output.body, context);
  const response = {
    $metadata: deserializeMetadata(output)
  };
  return response;
}, "de_UnsubscribeFromEventCommand");
var de_UpdateAssessmentTargetCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  await (0, import_smithy_client.collectBody)(output.body, context);
  const response = {
    $metadata: deserializeMetadata(output)
  };
  return response;
}, "de_UpdateAssessmentTargetCommand");
var de_CommandError = /* @__PURE__ */ __name(async (output, context) => {
  const parsedOutput = {
    ...output,
    body: await (0, import_core2.parseJsonErrorBody)(output.body, context)
  };
  const errorCode = (0, import_core2.loadRestJsonErrorCode)(output, parsedOutput.body);
  switch (errorCode) {
    case "AccessDeniedException":
    case "com.amazonaws.inspector#AccessDeniedException":
      throw await de_AccessDeniedExceptionRes(parsedOutput, context);
    case "InternalException":
    case "com.amazonaws.inspector#InternalException":
      throw await de_InternalExceptionRes(parsedOutput, context);
    case "InvalidInputException":
    case "com.amazonaws.inspector#InvalidInputException":
      throw await de_InvalidInputExceptionRes(parsedOutput, context);
    case "NoSuchEntityException":
    case "com.amazonaws.inspector#NoSuchEntityException":
      throw await de_NoSuchEntityExceptionRes(parsedOutput, context);
    case "ServiceTemporarilyUnavailableException":
    case "com.amazonaws.inspector#ServiceTemporarilyUnavailableException":
      throw await de_ServiceTemporarilyUnavailableExceptionRes(parsedOutput, context);
    case "InvalidCrossAccountRoleException":
    case "com.amazonaws.inspector#InvalidCrossAccountRoleException":
      throw await de_InvalidCrossAccountRoleExceptionRes(parsedOutput, context);
    case "LimitExceededException":
    case "com.amazonaws.inspector#LimitExceededException":
      throw await de_LimitExceededExceptionRes(parsedOutput, context);
    case "PreviewGenerationInProgressException":
    case "com.amazonaws.inspector#PreviewGenerationInProgressException":
      throw await de_PreviewGenerationInProgressExceptionRes(parsedOutput, context);
    case "AssessmentRunInProgressException":
    case "com.amazonaws.inspector#AssessmentRunInProgressException":
      throw await de_AssessmentRunInProgressExceptionRes(parsedOutput, context);
    case "UnsupportedFeatureException":
    case "com.amazonaws.inspector#UnsupportedFeatureException":
      throw await de_UnsupportedFeatureExceptionRes(parsedOutput, context);
    case "AgentsAlreadyRunningAssessmentException":
    case "com.amazonaws.inspector#AgentsAlreadyRunningAssessmentException":
      throw await de_AgentsAlreadyRunningAssessmentExceptionRes(parsedOutput, context);
    default:
      const parsedBody = parsedOutput.body;
      return throwDefaultError({
        output,
        parsedBody,
        errorCode
      });
  }
}, "de_CommandError");
var de_AccessDeniedExceptionRes = /* @__PURE__ */ __name(async (parsedOutput, context) => {
  const body = parsedOutput.body;
  const deserialized = (0, import_smithy_client._json)(body);
  const exception = new AccessDeniedException({
    $metadata: deserializeMetadata(parsedOutput),
    ...deserialized
  });
  return (0, import_smithy_client.decorateServiceException)(exception, body);
}, "de_AccessDeniedExceptionRes");
var de_AgentsAlreadyRunningAssessmentExceptionRes = /* @__PURE__ */ __name(async (parsedOutput, context) => {
  const body = parsedOutput.body;
  const deserialized = (0, import_smithy_client._json)(body);
  const exception = new AgentsAlreadyRunningAssessmentException({
    $metadata: deserializeMetadata(parsedOutput),
    ...deserialized
  });
  return (0, import_smithy_client.decorateServiceException)(exception, body);
}, "de_AgentsAlreadyRunningAssessmentExceptionRes");
var de_AssessmentRunInProgressExceptionRes = /* @__PURE__ */ __name(async (parsedOutput, context) => {
  const body = parsedOutput.body;
  const deserialized = (0, import_smithy_client._json)(body);
  const exception = new AssessmentRunInProgressException({
    $metadata: deserializeMetadata(parsedOutput),
    ...deserialized
  });
  return (0, import_smithy_client.decorateServiceException)(exception, body);
}, "de_AssessmentRunInProgressExceptionRes");
var de_InternalExceptionRes = /* @__PURE__ */ __name(async (parsedOutput, context) => {
  const body = parsedOutput.body;
  const deserialized = (0, import_smithy_client._json)(body);
  const exception = new InternalException({
    $metadata: deserializeMetadata(parsedOutput),
    ...deserialized
  });
  return (0, import_smithy_client.decorateServiceException)(exception, body);
}, "de_InternalExceptionRes");
var de_InvalidCrossAccountRoleExceptionRes = /* @__PURE__ */ __name(async (parsedOutput, context) => {
  const body = parsedOutput.body;
  const deserialized = (0, import_smithy_client._json)(body);
  const exception = new InvalidCrossAccountRoleException({
    $metadata: deserializeMetadata(parsedOutput),
    ...deserialized
  });
  return (0, import_smithy_client.decorateServiceException)(exception, body);
}, "de_InvalidCrossAccountRoleExceptionRes");
var de_InvalidInputExceptionRes = /* @__PURE__ */ __name(async (parsedOutput, context) => {
  const body = parsedOutput.body;
  const deserialized = (0, import_smithy_client._json)(body);
  const exception = new InvalidInputException({
    $metadata: deserializeMetadata(parsedOutput),
    ...deserialized
  });
  return (0, import_smithy_client.decorateServiceException)(exception, body);
}, "de_InvalidInputExceptionRes");
var de_LimitExceededExceptionRes = /* @__PURE__ */ __name(async (parsedOutput, context) => {
  const body = parsedOutput.body;
  const deserialized = (0, import_smithy_client._json)(body);
  const exception = new LimitExceededException({
    $metadata: deserializeMetadata(parsedOutput),
    ...deserialized
  });
  return (0, import_smithy_client.decorateServiceException)(exception, body);
}, "de_LimitExceededExceptionRes");
var de_NoSuchEntityExceptionRes = /* @__PURE__ */ __name(async (parsedOutput, context) => {
  const body = parsedOutput.body;
  const deserialized = (0, import_smithy_client._json)(body);
  const exception = new NoSuchEntityException({
    $metadata: deserializeMetadata(parsedOutput),
    ...deserialized
  });
  return (0, import_smithy_client.decorateServiceException)(exception, body);
}, "de_NoSuchEntityExceptionRes");
var de_PreviewGenerationInProgressExceptionRes = /* @__PURE__ */ __name(async (parsedOutput, context) => {
  const body = parsedOutput.body;
  const deserialized = (0, import_smithy_client._json)(body);
  const exception = new PreviewGenerationInProgressException({
    $metadata: deserializeMetadata(parsedOutput),
    ...deserialized
  });
  return (0, import_smithy_client.decorateServiceException)(exception, body);
}, "de_PreviewGenerationInProgressExceptionRes");
var de_ServiceTemporarilyUnavailableExceptionRes = /* @__PURE__ */ __name(async (parsedOutput, context) => {
  const body = parsedOutput.body;
  const deserialized = (0, import_smithy_client._json)(body);
  const exception = new ServiceTemporarilyUnavailableException({
    $metadata: deserializeMetadata(parsedOutput),
    ...deserialized
  });
  return (0, import_smithy_client.decorateServiceException)(exception, body);
}, "de_ServiceTemporarilyUnavailableExceptionRes");
var de_UnsupportedFeatureExceptionRes = /* @__PURE__ */ __name(async (parsedOutput, context) => {
  const body = parsedOutput.body;
  const deserialized = (0, import_smithy_client._json)(body);
  const exception = new UnsupportedFeatureException({
    $metadata: deserializeMetadata(parsedOutput),
    ...deserialized
  });
  return (0, import_smithy_client.decorateServiceException)(exception, body);
}, "de_UnsupportedFeatureExceptionRes");
var se_AssessmentRunFilter = /* @__PURE__ */ __name((input, context) => {
  return (0, import_smithy_client.take)(input, {
    completionTimeRange: (_) => se_TimestampRange(_, context),
    durationRange: import_smithy_client._json,
    namePattern: [],
    rulesPackageArns: import_smithy_client._json,
    startTimeRange: (_) => se_TimestampRange(_, context),
    stateChangeTimeRange: (_) => se_TimestampRange(_, context),
    states: import_smithy_client._json
  });
}, "se_AssessmentRunFilter");
var se_FindingFilter = /* @__PURE__ */ __name((input, context) => {
  return (0, import_smithy_client.take)(input, {
    agentIds: import_smithy_client._json,
    attributes: import_smithy_client._json,
    autoScalingGroups: import_smithy_client._json,
    creationTimeRange: (_) => se_TimestampRange(_, context),
    ruleNames: import_smithy_client._json,
    rulesPackageArns: import_smithy_client._json,
    severities: import_smithy_client._json,
    userAttributes: import_smithy_client._json
  });
}, "se_FindingFilter");
var se_ListAssessmentRunsRequest = /* @__PURE__ */ __name((input, context) => {
  return (0, import_smithy_client.take)(input, {
    assessmentTemplateArns: import_smithy_client._json,
    filter: (_) => se_AssessmentRunFilter(_, context),
    maxResults: [],
    nextToken: []
  });
}, "se_ListAssessmentRunsRequest");
var se_ListFindingsRequest = /* @__PURE__ */ __name((input, context) => {
  return (0, import_smithy_client.take)(input, {
    assessmentRunArns: import_smithy_client._json,
    filter: (_) => se_FindingFilter(_, context),
    maxResults: [],
    nextToken: []
  });
}, "se_ListFindingsRequest");
var se_TimestampRange = /* @__PURE__ */ __name((input, context) => {
  return (0, import_smithy_client.take)(input, {
    beginDate: (_) => Math.round(_.getTime() / 1e3),
    endDate: (_) => Math.round(_.getTime() / 1e3)
  });
}, "se_TimestampRange");
var de_AssessmentRun = /* @__PURE__ */ __name((output, context) => {
  return (0, import_smithy_client.take)(output, {
    arn: import_smithy_client.expectString,
    assessmentTemplateArn: import_smithy_client.expectString,
    completedAt: (_) => (0, import_smithy_client.expectNonNull)((0, import_smithy_client.parseEpochTimestamp)((0, import_smithy_client.expectNumber)(_))),
    createdAt: (_) => (0, import_smithy_client.expectNonNull)((0, import_smithy_client.parseEpochTimestamp)((0, import_smithy_client.expectNumber)(_))),
    dataCollected: import_smithy_client.expectBoolean,
    durationInSeconds: import_smithy_client.expectInt32,
    findingCounts: import_smithy_client._json,
    name: import_smithy_client.expectString,
    notifications: (_) => de_AssessmentRunNotificationList(_, context),
    rulesPackageArns: import_smithy_client._json,
    startedAt: (_) => (0, import_smithy_client.expectNonNull)((0, import_smithy_client.parseEpochTimestamp)((0, import_smithy_client.expectNumber)(_))),
    state: import_smithy_client.expectString,
    stateChangedAt: (_) => (0, import_smithy_client.expectNonNull)((0, import_smithy_client.parseEpochTimestamp)((0, import_smithy_client.expectNumber)(_))),
    stateChanges: (_) => de_AssessmentRunStateChangeList(_, context),
    userAttributesForFindings: import_smithy_client._json
  });
}, "de_AssessmentRun");
var de_AssessmentRunList = /* @__PURE__ */ __name((output, context) => {
  const retVal = (output || []).filter((e) => e != null).map((entry) => {
    return de_AssessmentRun(entry, context);
  });
  return retVal;
}, "de_AssessmentRunList");
var de_AssessmentRunNotification = /* @__PURE__ */ __name((output, context) => {
  return (0, import_smithy_client.take)(output, {
    date: (_) => (0, import_smithy_client.expectNonNull)((0, import_smithy_client.parseEpochTimestamp)((0, import_smithy_client.expectNumber)(_))),
    error: import_smithy_client.expectBoolean,
    event: import_smithy_client.expectString,
    message: import_smithy_client.expectString,
    snsPublishStatusCode: import_smithy_client.expectString,
    snsTopicArn: import_smithy_client.expectString
  });
}, "de_AssessmentRunNotification");
var de_AssessmentRunNotificationList = /* @__PURE__ */ __name((output, context) => {
  const retVal = (output || []).filter((e) => e != null).map((entry) => {
    return de_AssessmentRunNotification(entry, context);
  });
  return retVal;
}, "de_AssessmentRunNotificationList");
var de_AssessmentRunStateChange = /* @__PURE__ */ __name((output, context) => {
  return (0, import_smithy_client.take)(output, {
    state: import_smithy_client.expectString,
    stateChangedAt: (_) => (0, import_smithy_client.expectNonNull)((0, import_smithy_client.parseEpochTimestamp)((0, import_smithy_client.expectNumber)(_)))
  });
}, "de_AssessmentRunStateChange");
var de_AssessmentRunStateChangeList = /* @__PURE__ */ __name((output, context) => {
  const retVal = (output || []).filter((e) => e != null).map((entry) => {
    return de_AssessmentRunStateChange(entry, context);
  });
  return retVal;
}, "de_AssessmentRunStateChangeList");
var de_AssessmentTarget = /* @__PURE__ */ __name((output, context) => {
  return (0, import_smithy_client.take)(output, {
    arn: import_smithy_client.expectString,
    createdAt: (_) => (0, import_smithy_client.expectNonNull)((0, import_smithy_client.parseEpochTimestamp)((0, import_smithy_client.expectNumber)(_))),
    name: import_smithy_client.expectString,
    resourceGroupArn: import_smithy_client.expectString,
    updatedAt: (_) => (0, import_smithy_client.expectNonNull)((0, import_smithy_client.parseEpochTimestamp)((0, import_smithy_client.expectNumber)(_)))
  });
}, "de_AssessmentTarget");
var de_AssessmentTargetList = /* @__PURE__ */ __name((output, context) => {
  const retVal = (output || []).filter((e) => e != null).map((entry) => {
    return de_AssessmentTarget(entry, context);
  });
  return retVal;
}, "de_AssessmentTargetList");
var de_AssessmentTemplate = /* @__PURE__ */ __name((output, context) => {
  return (0, import_smithy_client.take)(output, {
    arn: import_smithy_client.expectString,
    assessmentRunCount: import_smithy_client.expectInt32,
    assessmentTargetArn: import_smithy_client.expectString,
    createdAt: (_) => (0, import_smithy_client.expectNonNull)((0, import_smithy_client.parseEpochTimestamp)((0, import_smithy_client.expectNumber)(_))),
    durationInSeconds: import_smithy_client.expectInt32,
    lastAssessmentRunArn: import_smithy_client.expectString,
    name: import_smithy_client.expectString,
    rulesPackageArns: import_smithy_client._json,
    userAttributesForFindings: import_smithy_client._json
  });
}, "de_AssessmentTemplate");
var de_AssessmentTemplateList = /* @__PURE__ */ __name((output, context) => {
  const retVal = (output || []).filter((e) => e != null).map((entry) => {
    return de_AssessmentTemplate(entry, context);
  });
  return retVal;
}, "de_AssessmentTemplateList");
var de_DescribeAssessmentRunsResponse = /* @__PURE__ */ __name((output, context) => {
  return (0, import_smithy_client.take)(output, {
    assessmentRuns: (_) => de_AssessmentRunList(_, context),
    failedItems: import_smithy_client._json
  });
}, "de_DescribeAssessmentRunsResponse");
var de_DescribeAssessmentTargetsResponse = /* @__PURE__ */ __name((output, context) => {
  return (0, import_smithy_client.take)(output, {
    assessmentTargets: (_) => de_AssessmentTargetList(_, context),
    failedItems: import_smithy_client._json
  });
}, "de_DescribeAssessmentTargetsResponse");
var de_DescribeAssessmentTemplatesResponse = /* @__PURE__ */ __name((output, context) => {
  return (0, import_smithy_client.take)(output, {
    assessmentTemplates: (_) => de_AssessmentTemplateList(_, context),
    failedItems: import_smithy_client._json
  });
}, "de_DescribeAssessmentTemplatesResponse");
var de_DescribeCrossAccountAccessRoleResponse = /* @__PURE__ */ __name((output, context) => {
  return (0, import_smithy_client.take)(output, {
    registeredAt: (_) => (0, import_smithy_client.expectNonNull)((0, import_smithy_client.parseEpochTimestamp)((0, import_smithy_client.expectNumber)(_))),
    roleArn: import_smithy_client.expectString,
    valid: import_smithy_client.expectBoolean
  });
}, "de_DescribeCrossAccountAccessRoleResponse");
var de_DescribeFindingsResponse = /* @__PURE__ */ __name((output, context) => {
  return (0, import_smithy_client.take)(output, {
    failedItems: import_smithy_client._json,
    findings: (_) => de_FindingList(_, context)
  });
}, "de_DescribeFindingsResponse");
var de_DescribeResourceGroupsResponse = /* @__PURE__ */ __name((output, context) => {
  return (0, import_smithy_client.take)(output, {
    failedItems: import_smithy_client._json,
    resourceGroups: (_) => de_ResourceGroupList(_, context)
  });
}, "de_DescribeResourceGroupsResponse");
var de_EventSubscription = /* @__PURE__ */ __name((output, context) => {
  return (0, import_smithy_client.take)(output, {
    event: import_smithy_client.expectString,
    subscribedAt: (_) => (0, import_smithy_client.expectNonNull)((0, import_smithy_client.parseEpochTimestamp)((0, import_smithy_client.expectNumber)(_)))
  });
}, "de_EventSubscription");
var de_EventSubscriptionList = /* @__PURE__ */ __name((output, context) => {
  const retVal = (output || []).filter((e) => e != null).map((entry) => {
    return de_EventSubscription(entry, context);
  });
  return retVal;
}, "de_EventSubscriptionList");
var de_Finding = /* @__PURE__ */ __name((output, context) => {
  return (0, import_smithy_client.take)(output, {
    arn: import_smithy_client.expectString,
    assetAttributes: import_smithy_client._json,
    assetType: import_smithy_client.expectString,
    attributes: import_smithy_client._json,
    confidence: import_smithy_client.expectInt32,
    createdAt: (_) => (0, import_smithy_client.expectNonNull)((0, import_smithy_client.parseEpochTimestamp)((0, import_smithy_client.expectNumber)(_))),
    description: import_smithy_client.expectString,
    id: import_smithy_client.expectString,
    indicatorOfCompromise: import_smithy_client.expectBoolean,
    numericSeverity: import_smithy_client.limitedParseDouble,
    recommendation: import_smithy_client.expectString,
    schemaVersion: import_smithy_client.expectInt32,
    service: import_smithy_client.expectString,
    serviceAttributes: import_smithy_client._json,
    severity: import_smithy_client.expectString,
    title: import_smithy_client.expectString,
    updatedAt: (_) => (0, import_smithy_client.expectNonNull)((0, import_smithy_client.parseEpochTimestamp)((0, import_smithy_client.expectNumber)(_))),
    userAttributes: import_smithy_client._json
  });
}, "de_Finding");
var de_FindingList = /* @__PURE__ */ __name((output, context) => {
  const retVal = (output || []).filter((e) => e != null).map((entry) => {
    return de_Finding(entry, context);
  });
  return retVal;
}, "de_FindingList");
var de_ListEventSubscriptionsResponse = /* @__PURE__ */ __name((output, context) => {
  return (0, import_smithy_client.take)(output, {
    nextToken: import_smithy_client.expectString,
    subscriptions: (_) => de_SubscriptionList(_, context)
  });
}, "de_ListEventSubscriptionsResponse");
var de_ResourceGroup = /* @__PURE__ */ __name((output, context) => {
  return (0, import_smithy_client.take)(output, {
    arn: import_smithy_client.expectString,
    createdAt: (_) => (0, import_smithy_client.expectNonNull)((0, import_smithy_client.parseEpochTimestamp)((0, import_smithy_client.expectNumber)(_))),
    tags: import_smithy_client._json
  });
}, "de_ResourceGroup");
var de_ResourceGroupList = /* @__PURE__ */ __name((output, context) => {
  const retVal = (output || []).filter((e) => e != null).map((entry) => {
    return de_ResourceGroup(entry, context);
  });
  return retVal;
}, "de_ResourceGroupList");
var de_Subscription = /* @__PURE__ */ __name((output, context) => {
  return (0, import_smithy_client.take)(output, {
    eventSubscriptions: (_) => de_EventSubscriptionList(_, context),
    resourceArn: import_smithy_client.expectString,
    topicArn: import_smithy_client.expectString
  });
}, "de_Subscription");
var de_SubscriptionList = /* @__PURE__ */ __name((output, context) => {
  const retVal = (output || []).filter((e) => e != null).map((entry) => {
    return de_Subscription(entry, context);
  });
  return retVal;
}, "de_SubscriptionList");
var deserializeMetadata = /* @__PURE__ */ __name((output) => ({
  httpStatusCode: output.statusCode,
  requestId: output.headers["x-amzn-requestid"] ?? output.headers["x-amzn-request-id"] ?? output.headers["x-amz-request-id"],
  extendedRequestId: output.headers["x-amz-id-2"],
  cfId: output.headers["x-amz-cf-id"]
}), "deserializeMetadata");
var throwDefaultError = (0, import_smithy_client.withBaseException)(InspectorServiceException);
var buildHttpRpcRequest = /* @__PURE__ */ __name(async (context, headers, path, resolvedHostname, body) => {
  const { hostname, protocol = "https", port, path: basePath } = await context.endpoint();
  const contents = {
    protocol,
    hostname,
    port,
    method: "POST",
    path: basePath.endsWith("/") ? basePath.slice(0, -1) + path : basePath + path,
    headers
  };
  if (resolvedHostname !== void 0) {
    contents.hostname = resolvedHostname;
  }
  if (body !== void 0) {
    contents.body = body;
  }
  return new import_protocol_http.HttpRequest(contents);
}, "buildHttpRpcRequest");
function sharedHeaders(operation) {
  return {
    "content-type": "application/x-amz-json-1.1",
    "x-amz-target": `InspectorService.${operation}`
  };
}
__name(sharedHeaders, "sharedHeaders");

// src/commands/AddAttributesToFindingsCommand.ts
var _AddAttributesToFindingsCommand = class _AddAttributesToFindingsCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("InspectorService", "AddAttributesToFindings", {}).n("InspectorClient", "AddAttributesToFindingsCommand").f(void 0, void 0).ser(se_AddAttributesToFindingsCommand).de(de_AddAttributesToFindingsCommand).build() {
};
__name(_AddAttributesToFindingsCommand, "AddAttributesToFindingsCommand");
var AddAttributesToFindingsCommand = _AddAttributesToFindingsCommand;

// src/commands/CreateAssessmentTargetCommand.ts




var _CreateAssessmentTargetCommand = class _CreateAssessmentTargetCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("InspectorService", "CreateAssessmentTarget", {}).n("InspectorClient", "CreateAssessmentTargetCommand").f(void 0, void 0).ser(se_CreateAssessmentTargetCommand).de(de_CreateAssessmentTargetCommand).build() {
};
__name(_CreateAssessmentTargetCommand, "CreateAssessmentTargetCommand");
var CreateAssessmentTargetCommand = _CreateAssessmentTargetCommand;

// src/commands/CreateAssessmentTemplateCommand.ts




var _CreateAssessmentTemplateCommand = class _CreateAssessmentTemplateCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("InspectorService", "CreateAssessmentTemplate", {}).n("InspectorClient", "CreateAssessmentTemplateCommand").f(void 0, void 0).ser(se_CreateAssessmentTemplateCommand).de(de_CreateAssessmentTemplateCommand).build() {
};
__name(_CreateAssessmentTemplateCommand, "CreateAssessmentTemplateCommand");
var CreateAssessmentTemplateCommand = _CreateAssessmentTemplateCommand;

// src/commands/CreateExclusionsPreviewCommand.ts




var _CreateExclusionsPreviewCommand = class _CreateExclusionsPreviewCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("InspectorService", "CreateExclusionsPreview", {}).n("InspectorClient", "CreateExclusionsPreviewCommand").f(void 0, void 0).ser(se_CreateExclusionsPreviewCommand).de(de_CreateExclusionsPreviewCommand).build() {
};
__name(_CreateExclusionsPreviewCommand, "CreateExclusionsPreviewCommand");
var CreateExclusionsPreviewCommand = _CreateExclusionsPreviewCommand;

// src/commands/CreateResourceGroupCommand.ts




var _CreateResourceGroupCommand = class _CreateResourceGroupCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("InspectorService", "CreateResourceGroup", {}).n("InspectorClient", "CreateResourceGroupCommand").f(void 0, void 0).ser(se_CreateResourceGroupCommand).de(de_CreateResourceGroupCommand).build() {
};
__name(_CreateResourceGroupCommand, "CreateResourceGroupCommand");
var CreateResourceGroupCommand = _CreateResourceGroupCommand;

// src/commands/DeleteAssessmentRunCommand.ts




var _DeleteAssessmentRunCommand = class _DeleteAssessmentRunCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("InspectorService", "DeleteAssessmentRun", {}).n("InspectorClient", "DeleteAssessmentRunCommand").f(void 0, void 0).ser(se_DeleteAssessmentRunCommand).de(de_DeleteAssessmentRunCommand).build() {
};
__name(_DeleteAssessmentRunCommand, "DeleteAssessmentRunCommand");
var DeleteAssessmentRunCommand = _DeleteAssessmentRunCommand;

// src/commands/DeleteAssessmentTargetCommand.ts




var _DeleteAssessmentTargetCommand = class _DeleteAssessmentTargetCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("InspectorService", "DeleteAssessmentTarget", {}).n("InspectorClient", "DeleteAssessmentTargetCommand").f(void 0, void 0).ser(se_DeleteAssessmentTargetCommand).de(de_DeleteAssessmentTargetCommand).build() {
};
__name(_DeleteAssessmentTargetCommand, "DeleteAssessmentTargetCommand");
var DeleteAssessmentTargetCommand = _DeleteAssessmentTargetCommand;

// src/commands/DeleteAssessmentTemplateCommand.ts




var _DeleteAssessmentTemplateCommand = class _DeleteAssessmentTemplateCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("InspectorService", "DeleteAssessmentTemplate", {}).n("InspectorClient", "DeleteAssessmentTemplateCommand").f(void 0, void 0).ser(se_DeleteAssessmentTemplateCommand).de(de_DeleteAssessmentTemplateCommand).build() {
};
__name(_DeleteAssessmentTemplateCommand, "DeleteAssessmentTemplateCommand");
var DeleteAssessmentTemplateCommand = _DeleteAssessmentTemplateCommand;

// src/commands/DescribeAssessmentRunsCommand.ts




var _DescribeAssessmentRunsCommand = class _DescribeAssessmentRunsCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("InspectorService", "DescribeAssessmentRuns", {}).n("InspectorClient", "DescribeAssessmentRunsCommand").f(void 0, void 0).ser(se_DescribeAssessmentRunsCommand).de(de_DescribeAssessmentRunsCommand).build() {
};
__name(_DescribeAssessmentRunsCommand, "DescribeAssessmentRunsCommand");
var DescribeAssessmentRunsCommand = _DescribeAssessmentRunsCommand;

// src/commands/DescribeAssessmentTargetsCommand.ts




var _DescribeAssessmentTargetsCommand = class _DescribeAssessmentTargetsCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("InspectorService", "DescribeAssessmentTargets", {}).n("InspectorClient", "DescribeAssessmentTargetsCommand").f(void 0, void 0).ser(se_DescribeAssessmentTargetsCommand).de(de_DescribeAssessmentTargetsCommand).build() {
};
__name(_DescribeAssessmentTargetsCommand, "DescribeAssessmentTargetsCommand");
var DescribeAssessmentTargetsCommand = _DescribeAssessmentTargetsCommand;

// src/commands/DescribeAssessmentTemplatesCommand.ts




var _DescribeAssessmentTemplatesCommand = class _DescribeAssessmentTemplatesCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("InspectorService", "DescribeAssessmentTemplates", {}).n("InspectorClient", "DescribeAssessmentTemplatesCommand").f(void 0, void 0).ser(se_DescribeAssessmentTemplatesCommand).de(de_DescribeAssessmentTemplatesCommand).build() {
};
__name(_DescribeAssessmentTemplatesCommand, "DescribeAssessmentTemplatesCommand");
var DescribeAssessmentTemplatesCommand = _DescribeAssessmentTemplatesCommand;

// src/commands/DescribeCrossAccountAccessRoleCommand.ts




var _DescribeCrossAccountAccessRoleCommand = class _DescribeCrossAccountAccessRoleCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("InspectorService", "DescribeCrossAccountAccessRole", {}).n("InspectorClient", "DescribeCrossAccountAccessRoleCommand").f(void 0, void 0).ser(se_DescribeCrossAccountAccessRoleCommand).de(de_DescribeCrossAccountAccessRoleCommand).build() {
};
__name(_DescribeCrossAccountAccessRoleCommand, "DescribeCrossAccountAccessRoleCommand");
var DescribeCrossAccountAccessRoleCommand = _DescribeCrossAccountAccessRoleCommand;

// src/commands/DescribeExclusionsCommand.ts




var _DescribeExclusionsCommand = class _DescribeExclusionsCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("InspectorService", "DescribeExclusions", {}).n("InspectorClient", "DescribeExclusionsCommand").f(void 0, void 0).ser(se_DescribeExclusionsCommand).de(de_DescribeExclusionsCommand).build() {
};
__name(_DescribeExclusionsCommand, "DescribeExclusionsCommand");
var DescribeExclusionsCommand = _DescribeExclusionsCommand;

// src/commands/DescribeFindingsCommand.ts




var _DescribeFindingsCommand = class _DescribeFindingsCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("InspectorService", "DescribeFindings", {}).n("InspectorClient", "DescribeFindingsCommand").f(void 0, void 0).ser(se_DescribeFindingsCommand).de(de_DescribeFindingsCommand).build() {
};
__name(_DescribeFindingsCommand, "DescribeFindingsCommand");
var DescribeFindingsCommand = _DescribeFindingsCommand;

// src/commands/DescribeResourceGroupsCommand.ts




var _DescribeResourceGroupsCommand = class _DescribeResourceGroupsCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("InspectorService", "DescribeResourceGroups", {}).n("InspectorClient", "DescribeResourceGroupsCommand").f(void 0, void 0).ser(se_DescribeResourceGroupsCommand).de(de_DescribeResourceGroupsCommand).build() {
};
__name(_DescribeResourceGroupsCommand, "DescribeResourceGroupsCommand");
var DescribeResourceGroupsCommand = _DescribeResourceGroupsCommand;

// src/commands/DescribeRulesPackagesCommand.ts




var _DescribeRulesPackagesCommand = class _DescribeRulesPackagesCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("InspectorService", "DescribeRulesPackages", {}).n("InspectorClient", "DescribeRulesPackagesCommand").f(void 0, void 0).ser(se_DescribeRulesPackagesCommand).de(de_DescribeRulesPackagesCommand).build() {
};
__name(_DescribeRulesPackagesCommand, "DescribeRulesPackagesCommand");
var DescribeRulesPackagesCommand = _DescribeRulesPackagesCommand;

// src/commands/GetAssessmentReportCommand.ts




var _GetAssessmentReportCommand = class _GetAssessmentReportCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("InspectorService", "GetAssessmentReport", {}).n("InspectorClient", "GetAssessmentReportCommand").f(void 0, void 0).ser(se_GetAssessmentReportCommand).de(de_GetAssessmentReportCommand).build() {
};
__name(_GetAssessmentReportCommand, "GetAssessmentReportCommand");
var GetAssessmentReportCommand = _GetAssessmentReportCommand;

// src/commands/GetExclusionsPreviewCommand.ts




var _GetExclusionsPreviewCommand = class _GetExclusionsPreviewCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("InspectorService", "GetExclusionsPreview", {}).n("InspectorClient", "GetExclusionsPreviewCommand").f(void 0, void 0).ser(se_GetExclusionsPreviewCommand).de(de_GetExclusionsPreviewCommand).build() {
};
__name(_GetExclusionsPreviewCommand, "GetExclusionsPreviewCommand");
var GetExclusionsPreviewCommand = _GetExclusionsPreviewCommand;

// src/commands/GetTelemetryMetadataCommand.ts




var _GetTelemetryMetadataCommand = class _GetTelemetryMetadataCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("InspectorService", "GetTelemetryMetadata", {}).n("InspectorClient", "GetTelemetryMetadataCommand").f(void 0, void 0).ser(se_GetTelemetryMetadataCommand).de(de_GetTelemetryMetadataCommand).build() {
};
__name(_GetTelemetryMetadataCommand, "GetTelemetryMetadataCommand");
var GetTelemetryMetadataCommand = _GetTelemetryMetadataCommand;

// src/commands/ListAssessmentRunAgentsCommand.ts




var _ListAssessmentRunAgentsCommand = class _ListAssessmentRunAgentsCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("InspectorService", "ListAssessmentRunAgents", {}).n("InspectorClient", "ListAssessmentRunAgentsCommand").f(void 0, void 0).ser(se_ListAssessmentRunAgentsCommand).de(de_ListAssessmentRunAgentsCommand).build() {
};
__name(_ListAssessmentRunAgentsCommand, "ListAssessmentRunAgentsCommand");
var ListAssessmentRunAgentsCommand = _ListAssessmentRunAgentsCommand;

// src/commands/ListAssessmentRunsCommand.ts




var _ListAssessmentRunsCommand = class _ListAssessmentRunsCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("InspectorService", "ListAssessmentRuns", {}).n("InspectorClient", "ListAssessmentRunsCommand").f(void 0, void 0).ser(se_ListAssessmentRunsCommand).de(de_ListAssessmentRunsCommand).build() {
};
__name(_ListAssessmentRunsCommand, "ListAssessmentRunsCommand");
var ListAssessmentRunsCommand = _ListAssessmentRunsCommand;

// src/commands/ListAssessmentTargetsCommand.ts




var _ListAssessmentTargetsCommand = class _ListAssessmentTargetsCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("InspectorService", "ListAssessmentTargets", {}).n("InspectorClient", "ListAssessmentTargetsCommand").f(void 0, void 0).ser(se_ListAssessmentTargetsCommand).de(de_ListAssessmentTargetsCommand).build() {
};
__name(_ListAssessmentTargetsCommand, "ListAssessmentTargetsCommand");
var ListAssessmentTargetsCommand = _ListAssessmentTargetsCommand;

// src/commands/ListAssessmentTemplatesCommand.ts




var _ListAssessmentTemplatesCommand = class _ListAssessmentTemplatesCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("InspectorService", "ListAssessmentTemplates", {}).n("InspectorClient", "ListAssessmentTemplatesCommand").f(void 0, void 0).ser(se_ListAssessmentTemplatesCommand).de(de_ListAssessmentTemplatesCommand).build() {
};
__name(_ListAssessmentTemplatesCommand, "ListAssessmentTemplatesCommand");
var ListAssessmentTemplatesCommand = _ListAssessmentTemplatesCommand;

// src/commands/ListEventSubscriptionsCommand.ts




var _ListEventSubscriptionsCommand = class _ListEventSubscriptionsCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("InspectorService", "ListEventSubscriptions", {}).n("InspectorClient", "ListEventSubscriptionsCommand").f(void 0, void 0).ser(se_ListEventSubscriptionsCommand).de(de_ListEventSubscriptionsCommand).build() {
};
__name(_ListEventSubscriptionsCommand, "ListEventSubscriptionsCommand");
var ListEventSubscriptionsCommand = _ListEventSubscriptionsCommand;

// src/commands/ListExclusionsCommand.ts




var _ListExclusionsCommand = class _ListExclusionsCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("InspectorService", "ListExclusions", {}).n("InspectorClient", "ListExclusionsCommand").f(void 0, void 0).ser(se_ListExclusionsCommand).de(de_ListExclusionsCommand).build() {
};
__name(_ListExclusionsCommand, "ListExclusionsCommand");
var ListExclusionsCommand = _ListExclusionsCommand;

// src/commands/ListFindingsCommand.ts




var _ListFindingsCommand = class _ListFindingsCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("InspectorService", "ListFindings", {}).n("InspectorClient", "ListFindingsCommand").f(void 0, void 0).ser(se_ListFindingsCommand).de(de_ListFindingsCommand).build() {
};
__name(_ListFindingsCommand, "ListFindingsCommand");
var ListFindingsCommand = _ListFindingsCommand;

// src/commands/ListRulesPackagesCommand.ts




var _ListRulesPackagesCommand = class _ListRulesPackagesCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("InspectorService", "ListRulesPackages", {}).n("InspectorClient", "ListRulesPackagesCommand").f(void 0, void 0).ser(se_ListRulesPackagesCommand).de(de_ListRulesPackagesCommand).build() {
};
__name(_ListRulesPackagesCommand, "ListRulesPackagesCommand");
var ListRulesPackagesCommand = _ListRulesPackagesCommand;

// src/commands/ListTagsForResourceCommand.ts




var _ListTagsForResourceCommand = class _ListTagsForResourceCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("InspectorService", "ListTagsForResource", {}).n("InspectorClient", "ListTagsForResourceCommand").f(void 0, void 0).ser(se_ListTagsForResourceCommand).de(de_ListTagsForResourceCommand).build() {
};
__name(_ListTagsForResourceCommand, "ListTagsForResourceCommand");
var ListTagsForResourceCommand = _ListTagsForResourceCommand;

// src/commands/PreviewAgentsCommand.ts




var _PreviewAgentsCommand = class _PreviewAgentsCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("InspectorService", "PreviewAgents", {}).n("InspectorClient", "PreviewAgentsCommand").f(void 0, void 0).ser(se_PreviewAgentsCommand).de(de_PreviewAgentsCommand).build() {
};
__name(_PreviewAgentsCommand, "PreviewAgentsCommand");
var PreviewAgentsCommand = _PreviewAgentsCommand;

// src/commands/RegisterCrossAccountAccessRoleCommand.ts




var _RegisterCrossAccountAccessRoleCommand = class _RegisterCrossAccountAccessRoleCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("InspectorService", "RegisterCrossAccountAccessRole", {}).n("InspectorClient", "RegisterCrossAccountAccessRoleCommand").f(void 0, void 0).ser(se_RegisterCrossAccountAccessRoleCommand).de(de_RegisterCrossAccountAccessRoleCommand).build() {
};
__name(_RegisterCrossAccountAccessRoleCommand, "RegisterCrossAccountAccessRoleCommand");
var RegisterCrossAccountAccessRoleCommand = _RegisterCrossAccountAccessRoleCommand;

// src/commands/RemoveAttributesFromFindingsCommand.ts




var _RemoveAttributesFromFindingsCommand = class _RemoveAttributesFromFindingsCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("InspectorService", "RemoveAttributesFromFindings", {}).n("InspectorClient", "RemoveAttributesFromFindingsCommand").f(void 0, void 0).ser(se_RemoveAttributesFromFindingsCommand).de(de_RemoveAttributesFromFindingsCommand).build() {
};
__name(_RemoveAttributesFromFindingsCommand, "RemoveAttributesFromFindingsCommand");
var RemoveAttributesFromFindingsCommand = _RemoveAttributesFromFindingsCommand;

// src/commands/SetTagsForResourceCommand.ts




var _SetTagsForResourceCommand = class _SetTagsForResourceCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("InspectorService", "SetTagsForResource", {}).n("InspectorClient", "SetTagsForResourceCommand").f(void 0, void 0).ser(se_SetTagsForResourceCommand).de(de_SetTagsForResourceCommand).build() {
};
__name(_SetTagsForResourceCommand, "SetTagsForResourceCommand");
var SetTagsForResourceCommand = _SetTagsForResourceCommand;

// src/commands/StartAssessmentRunCommand.ts




var _StartAssessmentRunCommand = class _StartAssessmentRunCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("InspectorService", "StartAssessmentRun", {}).n("InspectorClient", "StartAssessmentRunCommand").f(void 0, void 0).ser(se_StartAssessmentRunCommand).de(de_StartAssessmentRunCommand).build() {
};
__name(_StartAssessmentRunCommand, "StartAssessmentRunCommand");
var StartAssessmentRunCommand = _StartAssessmentRunCommand;

// src/commands/StopAssessmentRunCommand.ts




var _StopAssessmentRunCommand = class _StopAssessmentRunCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("InspectorService", "StopAssessmentRun", {}).n("InspectorClient", "StopAssessmentRunCommand").f(void 0, void 0).ser(se_StopAssessmentRunCommand).de(de_StopAssessmentRunCommand).build() {
};
__name(_StopAssessmentRunCommand, "StopAssessmentRunCommand");
var StopAssessmentRunCommand = _StopAssessmentRunCommand;

// src/commands/SubscribeToEventCommand.ts




var _SubscribeToEventCommand = class _SubscribeToEventCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("InspectorService", "SubscribeToEvent", {}).n("InspectorClient", "SubscribeToEventCommand").f(void 0, void 0).ser(se_SubscribeToEventCommand).de(de_SubscribeToEventCommand).build() {
};
__name(_SubscribeToEventCommand, "SubscribeToEventCommand");
var SubscribeToEventCommand = _SubscribeToEventCommand;

// src/commands/UnsubscribeFromEventCommand.ts




var _UnsubscribeFromEventCommand = class _UnsubscribeFromEventCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("InspectorService", "UnsubscribeFromEvent", {}).n("InspectorClient", "UnsubscribeFromEventCommand").f(void 0, void 0).ser(se_UnsubscribeFromEventCommand).de(de_UnsubscribeFromEventCommand).build() {
};
__name(_UnsubscribeFromEventCommand, "UnsubscribeFromEventCommand");
var UnsubscribeFromEventCommand = _UnsubscribeFromEventCommand;

// src/commands/UpdateAssessmentTargetCommand.ts




var _UpdateAssessmentTargetCommand = class _UpdateAssessmentTargetCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("InspectorService", "UpdateAssessmentTarget", {}).n("InspectorClient", "UpdateAssessmentTargetCommand").f(void 0, void 0).ser(se_UpdateAssessmentTargetCommand).de(de_UpdateAssessmentTargetCommand).build() {
};
__name(_UpdateAssessmentTargetCommand, "UpdateAssessmentTargetCommand");
var UpdateAssessmentTargetCommand = _UpdateAssessmentTargetCommand;

// src/Inspector.ts
var commands = {
  AddAttributesToFindingsCommand,
  CreateAssessmentTargetCommand,
  CreateAssessmentTemplateCommand,
  CreateExclusionsPreviewCommand,
  CreateResourceGroupCommand,
  DeleteAssessmentRunCommand,
  DeleteAssessmentTargetCommand,
  DeleteAssessmentTemplateCommand,
  DescribeAssessmentRunsCommand,
  DescribeAssessmentTargetsCommand,
  DescribeAssessmentTemplatesCommand,
  DescribeCrossAccountAccessRoleCommand,
  DescribeExclusionsCommand,
  DescribeFindingsCommand,
  DescribeResourceGroupsCommand,
  DescribeRulesPackagesCommand,
  GetAssessmentReportCommand,
  GetExclusionsPreviewCommand,
  GetTelemetryMetadataCommand,
  ListAssessmentRunAgentsCommand,
  ListAssessmentRunsCommand,
  ListAssessmentTargetsCommand,
  ListAssessmentTemplatesCommand,
  ListEventSubscriptionsCommand,
  ListExclusionsCommand,
  ListFindingsCommand,
  ListRulesPackagesCommand,
  ListTagsForResourceCommand,
  PreviewAgentsCommand,
  RegisterCrossAccountAccessRoleCommand,
  RemoveAttributesFromFindingsCommand,
  SetTagsForResourceCommand,
  StartAssessmentRunCommand,
  StopAssessmentRunCommand,
  SubscribeToEventCommand,
  UnsubscribeFromEventCommand,
  UpdateAssessmentTargetCommand
};
var _Inspector = class _Inspector extends InspectorClient {
};
__name(_Inspector, "Inspector");
var Inspector = _Inspector;
(0, import_smithy_client.createAggregatedClient)(commands, Inspector);

// src/pagination/GetExclusionsPreviewPaginator.ts

var paginateGetExclusionsPreview = (0, import_core.createPaginator)(InspectorClient, GetExclusionsPreviewCommand, "nextToken", "nextToken", "maxResults");

// src/pagination/ListAssessmentRunAgentsPaginator.ts

var paginateListAssessmentRunAgents = (0, import_core.createPaginator)(InspectorClient, ListAssessmentRunAgentsCommand, "nextToken", "nextToken", "maxResults");

// src/pagination/ListAssessmentRunsPaginator.ts

var paginateListAssessmentRuns = (0, import_core.createPaginator)(InspectorClient, ListAssessmentRunsCommand, "nextToken", "nextToken", "maxResults");

// src/pagination/ListAssessmentTargetsPaginator.ts

var paginateListAssessmentTargets = (0, import_core.createPaginator)(InspectorClient, ListAssessmentTargetsCommand, "nextToken", "nextToken", "maxResults");

// src/pagination/ListAssessmentTemplatesPaginator.ts

var paginateListAssessmentTemplates = (0, import_core.createPaginator)(InspectorClient, ListAssessmentTemplatesCommand, "nextToken", "nextToken", "maxResults");

// src/pagination/ListEventSubscriptionsPaginator.ts

var paginateListEventSubscriptions = (0, import_core.createPaginator)(InspectorClient, ListEventSubscriptionsCommand, "nextToken", "nextToken", "maxResults");

// src/pagination/ListExclusionsPaginator.ts

var paginateListExclusions = (0, import_core.createPaginator)(InspectorClient, ListExclusionsCommand, "nextToken", "nextToken", "maxResults");

// src/pagination/ListFindingsPaginator.ts

var paginateListFindings = (0, import_core.createPaginator)(InspectorClient, ListFindingsCommand, "nextToken", "nextToken", "maxResults");

// src/pagination/ListRulesPackagesPaginator.ts

var paginateListRulesPackages = (0, import_core.createPaginator)(InspectorClient, ListRulesPackagesCommand, "nextToken", "nextToken", "maxResults");

// src/pagination/PreviewAgentsPaginator.ts

var paginatePreviewAgents = (0, import_core.createPaginator)(InspectorClient, PreviewAgentsCommand, "nextToken", "nextToken", "maxResults");
// Annotate the CommonJS export names for ESM import in node:

0 && (module.exports = {
  InspectorServiceException,
  __Client,
  InspectorClient,
  Inspector,
  $Command,
  AddAttributesToFindingsCommand,
  CreateAssessmentTargetCommand,
  CreateAssessmentTemplateCommand,
  CreateExclusionsPreviewCommand,
  CreateResourceGroupCommand,
  DeleteAssessmentRunCommand,
  DeleteAssessmentTargetCommand,
  DeleteAssessmentTemplateCommand,
  DescribeAssessmentRunsCommand,
  DescribeAssessmentTargetsCommand,
  DescribeAssessmentTemplatesCommand,
  DescribeCrossAccountAccessRoleCommand,
  DescribeExclusionsCommand,
  DescribeFindingsCommand,
  DescribeResourceGroupsCommand,
  DescribeRulesPackagesCommand,
  GetAssessmentReportCommand,
  GetExclusionsPreviewCommand,
  GetTelemetryMetadataCommand,
  ListAssessmentRunAgentsCommand,
  ListAssessmentRunsCommand,
  ListAssessmentTargetsCommand,
  ListAssessmentTemplatesCommand,
  ListEventSubscriptionsCommand,
  ListExclusionsCommand,
  ListFindingsCommand,
  ListRulesPackagesCommand,
  ListTagsForResourceCommand,
  PreviewAgentsCommand,
  RegisterCrossAccountAccessRoleCommand,
  RemoveAttributesFromFindingsCommand,
  SetTagsForResourceCommand,
  StartAssessmentRunCommand,
  StopAssessmentRunCommand,
  SubscribeToEventCommand,
  UnsubscribeFromEventCommand,
  UpdateAssessmentTargetCommand,
  paginateGetExclusionsPreview,
  paginateListAssessmentRunAgents,
  paginateListAssessmentRuns,
  paginateListAssessmentTargets,
  paginateListAssessmentTemplates,
  paginateListEventSubscriptions,
  paginateListExclusions,
  paginateListFindings,
  paginateListRulesPackages,
  paginatePreviewAgents,
  AccessDeniedErrorCode,
  AccessDeniedException,
  FailedItemErrorCode,
  InternalException,
  InvalidInputErrorCode,
  InvalidInputException,
  NoSuchEntityErrorCode,
  NoSuchEntityException,
  ServiceTemporarilyUnavailableException,
  AgentHealthCode,
  AgentHealth,
  AgentsAlreadyRunningAssessmentException,
  Severity,
  InspectorEvent,
  AssessmentRunNotificationSnsStatusCode,
  AssessmentRunState,
  AssessmentRunInProgressException,
  AssetType,
  InvalidCrossAccountRoleErrorCode,
  InvalidCrossAccountRoleException,
  LimitExceededErrorCode,
  LimitExceededException,
  PreviewGenerationInProgressException,
  Locale,
  ScopeType,
  ReportFileFormat,
  ReportType,
  ReportStatus,
  UnsupportedFeatureException,
  PreviewStatus,
  StopAction
});
