// @aws-sdk/client-marketplace-commerce-analytics/dist-es/protocols/Aws_json1_1.js
import { loadRestJsonErrorCode, parseJsonBody as parseBody, parseJsonErrorBody as parseErrorBody } from "@aws-sdk/core";
import { HttpRequest as __HttpRequest } from "@smithy/protocol-http";
import { _json, collectBody, decorateServiceException as __decorateServiceException, take, withBaseException, } from "@smithy/smithy-client";
import { MarketplaceCommerceAnalyticsServiceException as __BaseException } from "../models/MarketplaceCommerceAnalyticsServiceException";
import { MarketplaceCommerceAnalyticsException, } from "../models/models_0";
export const se_GenerateDataSetCommand = async (input, context) => {
    const headers = sharedHeaders("GenerateDataSet");
    let body;
    body = JSON.stringify(se_GenerateDataSetRequest(input, context));
    return buildHttpRpcRequest(context, headers, "/", undefined, body);
};
export const se_StartSupportDataExportCommand = async (input, context) => {
    const headers = sharedHeaders("StartSupportDataExport");
    let body;
    body = JSON.stringify(se_StartSupportDataExportRequest(input, context));
    return buildHttpRpcRequest(context, headers, "/", undefined, body);
};
export const de_GenerateDataSetCommand = async (output, context) => {
    if (output.statusCode >= 300) {
        return de_CommandError(output, context);
    }
    const data = await parseBody(output.body, context);
    let contents = {};
    contents = _json(data);
    const response = {
        $metadata: deserializeMetadata(output),
        ...contents,
    };
    return response;
};
export const de_StartSupportDataExportCommand = async (output, context) => {
    if (output.statusCode >= 300) {
        return de_CommandError(output, context);
    }
    const data = await parseBody(output.body, context);
    let contents = {};
    contents = _json(data);
    const response = {
        $metadata: deserializeMetadata(output),
        ...contents,
    };
    return response;
};
const de_CommandError = async (output, context) => {
    const parsedOutput = {
        ...output,
        body: await parseErrorBody(output.body, context),
    };
    const errorCode = loadRestJsonErrorCode(output, parsedOutput.body);
    switch (errorCode) {
        case "MarketplaceCommerceAnalyticsException":
        case "com.amazonaws.marketplacecommerceanalytics#MarketplaceCommerceAnalyticsException":
            throw await de_MarketplaceCommerceAnalyticsExceptionRes(parsedOutput, context);
        default:
            const parsedBody = parsedOutput.body;
            return throwDefaultError({
                output,
                parsedBody,
                errorCode,
            });
    }
};
const de_MarketplaceCommerceAnalyticsExceptionRes = async (parsedOutput, context) => {
    const body = parsedOutput.body;
    const deserialized = _json(body);
    const exception = new MarketplaceCommerceAnalyticsException({
        $metadata: deserializeMetadata(parsedOutput),
        ...deserialized,
    });
    return __decorateServiceException(exception, body);
};
const se_GenerateDataSetRequest = (input, context) => {
    return take(input, {
        customerDefinedValues: _json,
        dataSetPublicationDate: (_) => Math.round(_.getTime() / 1000),
        dataSetType: [],
        destinationS3BucketName: [],
        destinationS3Prefix: [],
        roleNameArn: [],
        snsTopicArn: [],
    });
};
const se_StartSupportDataExportRequest = (input, context) => {
    return take(input, {
        customerDefinedValues: _json,
        dataSetType: [],
        destinationS3BucketName: [],
        destinationS3Prefix: [],
        fromDate: (_) => Math.round(_.getTime() / 1000),
        roleNameArn: [],
        snsTopicArn: [],
    });
};
const deserializeMetadata = (output) => ({
    httpStatusCode: output.statusCode,
    requestId: output.headers["x-amzn-requestid"] ?? output.headers["x-amzn-request-id"] ?? output.headers["x-amz-request-id"],
    extendedRequestId: output.headers["x-amz-id-2"],
    cfId: output.headers["x-amz-cf-id"],
});
const collectBodyString = (streamBody, context) => collectBody(streamBody, context).then((body) => context.utf8Encoder(body));
const throwDefaultError = withBaseException(__BaseException);
const buildHttpRpcRequest = async (context, headers, path, resolvedHostname, body) => {
    const { hostname, protocol = "https", port, path: basePath } = await context.endpoint();
    const contents = {
        protocol,
        hostname,
        port,
        method: "POST",
        path: basePath.endsWith("/") ? basePath.slice(0, -1) + path : basePath + path,
        headers,
    };
    if (resolvedHostname !== undefined) {
        contents.hostname = resolvedHostname;
    }
    if (body !== undefined) {
        contents.body = body;
    }
    return new __HttpRequest(contents);
};
function sharedHeaders(operation) {
    return {
        "content-type": "application/x-amz-json-1.1",
        "x-amz-target": `MarketplaceCommerceAnalytics20150701.${operation}`,
    };
}
