// @azure/arm-dnsresolver/dist-esm/test/dnsresolver_examples.spec.js
/*
 * Copyright (c) Microsoft Corporation.
 * Licensed under the MIT License.
 *
 * Code generated by Microsoft (R) AutoRest Code Generator.
 * Changes may cause incorrect behavior and will be lost if the code is regenerated.
 */
import { __asyncValues, __awaiter } from "tslib";
import { env, Recorder, isPlaybackMode, } from "@azure-tools/test-recorder";
import { createTestCredential } from "@azure-tools/test-credential";
import { assert } from "chai";
import { DnsResolverManagementClient } from "../src/dnsResolverManagementClient";
const replaceableVariables = {
    AZURE_CLIENT_ID: "azure_client_id",
    AZURE_CLIENT_SECRET: "azure_client_secret",
    AZURE_TENANT_ID: "88888888-8888-8888-8888-888888888888",
    SUBSCRIPTION_ID: "azure_subscription_id"
};
const recorderOptions = {
    envSetupForPlayback: replaceableVariables
};
export const testPollingOptions = {
    updateIntervalInMs: isPlaybackMode() ? 0 : undefined,
};
describe("dnsresolve test", () => {
    let recorder;
    let subscriptionId;
    let client;
    let resourceGroup;
    let parameters;
    let dnsResolverName;
    beforeEach(function () {
        return __awaiter(this, void 0, void 0, function* () {
            recorder = new Recorder(this.currentTest);
            yield recorder.start(recorderOptions);
            subscriptionId = env.SUBSCRIPTION_ID || '';
            // This is an example of how the environment variables are used
            const credential = createTestCredential();
            client = new DnsResolverManagementClient(credential, subscriptionId, recorder.configureClientOptions({}));
            resourceGroup = "myjstest";
            dnsResolverName = "sampleDnsResolver";
            parameters = {
                location: "EastUS",
                tags: { key1: "value1" },
                virtualNetwork: {
                    id: "/subscriptions/" + subscriptionId + "/resourceGroups/" + resourceGroup + "/providers/Microsoft.Network/virtualNetworks/sampleVirtualNetwork"
                }
            };
        });
    });
    afterEach(function () {
        return __awaiter(this, void 0, void 0, function* () {
            yield recorder.stop();
        });
    });
    it("dnsResolvers create clusters", function () {
        return __awaiter(this, void 0, void 0, function* () {
            let res = yield client.dnsResolvers.beginCreateOrUpdateAndWait(resourceGroup, dnsResolverName, parameters, testPollingOptions);
            assert.equal(res.name, dnsResolverName);
        });
    });
    it("dnsResolvers get clusters", function () {
        return __awaiter(this, void 0, void 0, function* () {
            let res = yield client.dnsResolvers.get(resourceGroup, dnsResolverName);
            assert.equal(res.name, dnsResolverName);
        });
    });
    it("dnsResolvers list clusters", function () {
        var e_1, _a;
        return __awaiter(this, void 0, void 0, function* () {
            const resArray = new Array();
            try {
                for (var _b = __asyncValues(client.dnsResolvers.listByResourceGroup(resourceGroup)), _c; _c = yield _b.next(), !_c.done;) {
                    const item = _c.value;
                    resArray.push(item);
                }
            }
            catch (e_1_1) { e_1 = { error: e_1_1 }; }
            finally {
                try {
                    if (_c && !_c.done && (_a = _b.return)) yield _a.call(_b);
                }
                finally { if (e_1) throw e_1.error; }
            }
            assert.equal(resArray.length, 1);
        });
    });
    it("dnsResolvers delete clusters", function () {
        var e_2, _a;
        return __awaiter(this, void 0, void 0, function* () {
            const resArray = new Array();
            const res = yield client.dnsResolvers.beginDeleteAndWait(resourceGroup, dnsResolverName);
            try {
                for (var _b = __asyncValues(client.dnsResolvers.listByResourceGroup(resourceGroup)), _c; _c = yield _b.next(), !_c.done;) {
                    const item = _c.value;
                    resArray.push(item);
                }
            }
            catch (e_2_1) { e_2 = { error: e_2_1 }; }
            finally {
                try {
                    if (_c && !_c.done && (_a = _b.return)) yield _a.call(_b);
                }
                finally { if (e_2) throw e_2.error; }
            }
            assert.equal(resArray.length, 0);
        });
    });
});
