// @azure/arm-authorization/dist-esm/src/models/index.js
/*
 * Copyright (c) Microsoft Corporation.
 * Licensed under the MIT License.
 *
 * Code generated by Microsoft (R) AutoRest Code Generator.
 * Changes may cause incorrect behavior and will be lost if the code is regenerated.
 */
/** Known values of {@link PrincipalType} that the service accepts. */
export var KnownPrincipalType;
(function (KnownPrincipalType) {
    /** User */
    KnownPrincipalType["User"] = "User";
    /** Group */
    KnownPrincipalType["Group"] = "Group";
    /** ServicePrincipal */
    KnownPrincipalType["ServicePrincipal"] = "ServicePrincipal";
    /** ForeignGroup */
    KnownPrincipalType["ForeignGroup"] = "ForeignGroup";
    /** Device */
    KnownPrincipalType["Device"] = "Device";
})(KnownPrincipalType || (KnownPrincipalType = {}));
/** Known values of {@link AssignmentType} that the service accepts. */
export var KnownAssignmentType;
(function (KnownAssignmentType) {
    /** Activated */
    KnownAssignmentType["Activated"] = "Activated";
    /** Assigned */
    KnownAssignmentType["Assigned"] = "Assigned";
})(KnownAssignmentType || (KnownAssignmentType = {}));
/** Known values of {@link MemberType} that the service accepts. */
export var KnownMemberType;
(function (KnownMemberType) {
    /** Inherited */
    KnownMemberType["Inherited"] = "Inherited";
    /** Direct */
    KnownMemberType["Direct"] = "Direct";
    /** Group */
    KnownMemberType["Group"] = "Group";
})(KnownMemberType || (KnownMemberType = {}));
/** Known values of {@link Status} that the service accepts. */
export var KnownStatus;
(function (KnownStatus) {
    /** Accepted */
    KnownStatus["Accepted"] = "Accepted";
    /** PendingEvaluation */
    KnownStatus["PendingEvaluation"] = "PendingEvaluation";
    /** Granted */
    KnownStatus["Granted"] = "Granted";
    /** Denied */
    KnownStatus["Denied"] = "Denied";
    /** PendingProvisioning */
    KnownStatus["PendingProvisioning"] = "PendingProvisioning";
    /** Provisioned */
    KnownStatus["Provisioned"] = "Provisioned";
    /** PendingRevocation */
    KnownStatus["PendingRevocation"] = "PendingRevocation";
    /** Revoked */
    KnownStatus["Revoked"] = "Revoked";
    /** Canceled */
    KnownStatus["Canceled"] = "Canceled";
    /** Failed */
    KnownStatus["Failed"] = "Failed";
    /** PendingApprovalProvisioning */
    KnownStatus["PendingApprovalProvisioning"] = "PendingApprovalProvisioning";
    /** PendingApproval */
    KnownStatus["PendingApproval"] = "PendingApproval";
    /** FailedAsResourceIsLocked */
    KnownStatus["FailedAsResourceIsLocked"] = "FailedAsResourceIsLocked";
    /** PendingAdminDecision */
    KnownStatus["PendingAdminDecision"] = "PendingAdminDecision";
    /** AdminApproved */
    KnownStatus["AdminApproved"] = "AdminApproved";
    /** AdminDenied */
    KnownStatus["AdminDenied"] = "AdminDenied";
    /** TimedOut */
    KnownStatus["TimedOut"] = "TimedOut";
    /** ProvisioningStarted */
    KnownStatus["ProvisioningStarted"] = "ProvisioningStarted";
    /** Invalid */
    KnownStatus["Invalid"] = "Invalid";
    /** PendingScheduleCreation */
    KnownStatus["PendingScheduleCreation"] = "PendingScheduleCreation";
    /** ScheduleCreated */
    KnownStatus["ScheduleCreated"] = "ScheduleCreated";
    /** PendingExternalProvisioning */
    KnownStatus["PendingExternalProvisioning"] = "PendingExternalProvisioning";
})(KnownStatus || (KnownStatus = {}));
/** Known values of {@link RequestType} that the service accepts. */
export var KnownRequestType;
(function (KnownRequestType) {
    /** AdminAssign */
    KnownRequestType["AdminAssign"] = "AdminAssign";
    /** AdminRemove */
    KnownRequestType["AdminRemove"] = "AdminRemove";
    /** AdminUpdate */
    KnownRequestType["AdminUpdate"] = "AdminUpdate";
    /** AdminExtend */
    KnownRequestType["AdminExtend"] = "AdminExtend";
    /** AdminRenew */
    KnownRequestType["AdminRenew"] = "AdminRenew";
    /** SelfActivate */
    KnownRequestType["SelfActivate"] = "SelfActivate";
    /** SelfDeactivate */
    KnownRequestType["SelfDeactivate"] = "SelfDeactivate";
    /** SelfExtend */
    KnownRequestType["SelfExtend"] = "SelfExtend";
    /** SelfRenew */
    KnownRequestType["SelfRenew"] = "SelfRenew";
})(KnownRequestType || (KnownRequestType = {}));
/** Known values of {@link Type} that the service accepts. */
export var KnownType;
(function (KnownType) {
    /** AfterDuration */
    KnownType["AfterDuration"] = "AfterDuration";
    /** AfterDateTime */
    KnownType["AfterDateTime"] = "AfterDateTime";
    /** NoExpiration */
    KnownType["NoExpiration"] = "NoExpiration";
})(KnownType || (KnownType = {}));
/** Known values of {@link RoleManagementPolicyRuleType} that the service accepts. */
export var KnownRoleManagementPolicyRuleType;
(function (KnownRoleManagementPolicyRuleType) {
    /** RoleManagementPolicyApprovalRule */
    KnownRoleManagementPolicyRuleType["RoleManagementPolicyApprovalRule"] = "RoleManagementPolicyApprovalRule";
    /** RoleManagementPolicyAuthenticationContextRule */
    KnownRoleManagementPolicyRuleType["RoleManagementPolicyAuthenticationContextRule"] = "RoleManagementPolicyAuthenticationContextRule";
    /** RoleManagementPolicyEnablementRule */
    KnownRoleManagementPolicyRuleType["RoleManagementPolicyEnablementRule"] = "RoleManagementPolicyEnablementRule";
    /** RoleManagementPolicyExpirationRule */
    KnownRoleManagementPolicyRuleType["RoleManagementPolicyExpirationRule"] = "RoleManagementPolicyExpirationRule";
    /** RoleManagementPolicyNotificationRule */
    KnownRoleManagementPolicyRuleType["RoleManagementPolicyNotificationRule"] = "RoleManagementPolicyNotificationRule";
})(KnownRoleManagementPolicyRuleType || (KnownRoleManagementPolicyRuleType = {}));
/** Known values of {@link ApprovalMode} that the service accepts. */
export var KnownApprovalMode;
(function (KnownApprovalMode) {
    /** SingleStage */
    KnownApprovalMode["SingleStage"] = "SingleStage";
    /** Serial */
    KnownApprovalMode["Serial"] = "Serial";
    /** Parallel */
    KnownApprovalMode["Parallel"] = "Parallel";
    /** NoApproval */
    KnownApprovalMode["NoApproval"] = "NoApproval";
})(KnownApprovalMode || (KnownApprovalMode = {}));
/** Known values of {@link UserType} that the service accepts. */
export var KnownUserType;
(function (KnownUserType) {
    /** User */
    KnownUserType["User"] = "User";
    /** Group */
    KnownUserType["Group"] = "Group";
})(KnownUserType || (KnownUserType = {}));
/** Known values of {@link EnablementRules} that the service accepts. */
export var KnownEnablementRules;
(function (KnownEnablementRules) {
    /** MultiFactorAuthentication */
    KnownEnablementRules["MultiFactorAuthentication"] = "MultiFactorAuthentication";
    /** Justification */
    KnownEnablementRules["Justification"] = "Justification";
    /** Ticketing */
    KnownEnablementRules["Ticketing"] = "Ticketing";
})(KnownEnablementRules || (KnownEnablementRules = {}));
/** Known values of {@link NotificationDeliveryMechanism} that the service accepts. */
export var KnownNotificationDeliveryMechanism;
(function (KnownNotificationDeliveryMechanism) {
    /** Email */
    KnownNotificationDeliveryMechanism["Email"] = "Email";
})(KnownNotificationDeliveryMechanism || (KnownNotificationDeliveryMechanism = {}));
/** Known values of {@link NotificationLevel} that the service accepts. */
export var KnownNotificationLevel;
(function (KnownNotificationLevel) {
    /** None */
    KnownNotificationLevel["None"] = "None";
    /** Critical */
    KnownNotificationLevel["Critical"] = "Critical";
    /** All */
    KnownNotificationLevel["All"] = "All";
})(KnownNotificationLevel || (KnownNotificationLevel = {}));
/** Known values of {@link RecipientType} that the service accepts. */
export var KnownRecipientType;
(function (KnownRecipientType) {
    /** Requestor */
    KnownRecipientType["Requestor"] = "Requestor";
    /** Approver */
    KnownRecipientType["Approver"] = "Approver";
    /** Admin */
    KnownRecipientType["Admin"] = "Admin";
})(KnownRecipientType || (KnownRecipientType = {}));
