// @azure/arm-domainservices/dist-esm/src/models/index.js
/*
 * Copyright (c) Microsoft Corporation.
 * Licensed under the MIT License.
 *
 * Code generated by Microsoft (R) AutoRest Code Generator.
 * Changes may cause incorrect behavior and will be lost if the code is regenerated.
 */
/** Known values of {@link Ldaps} that the service accepts. */
export var KnownLdaps;
(function (KnownLdaps) {
    /** Enabled */
    KnownLdaps["Enabled"] = "Enabled";
    /** Disabled */
    KnownLdaps["Disabled"] = "Disabled";
})(KnownLdaps || (KnownLdaps = {}));
/** Known values of {@link ExternalAccess} that the service accepts. */
export var KnownExternalAccess;
(function (KnownExternalAccess) {
    /** Enabled */
    KnownExternalAccess["Enabled"] = "Enabled";
    /** Disabled */
    KnownExternalAccess["Disabled"] = "Disabled";
})(KnownExternalAccess || (KnownExternalAccess = {}));
/** Known values of {@link NtlmV1} that the service accepts. */
export var KnownNtlmV1;
(function (KnownNtlmV1) {
    /** Enabled */
    KnownNtlmV1["Enabled"] = "Enabled";
    /** Disabled */
    KnownNtlmV1["Disabled"] = "Disabled";
})(KnownNtlmV1 || (KnownNtlmV1 = {}));
/** Known values of {@link TlsV1} that the service accepts. */
export var KnownTlsV1;
(function (KnownTlsV1) {
    /** Enabled */
    KnownTlsV1["Enabled"] = "Enabled";
    /** Disabled */
    KnownTlsV1["Disabled"] = "Disabled";
})(KnownTlsV1 || (KnownTlsV1 = {}));
/** Known values of {@link SyncNtlmPasswords} that the service accepts. */
export var KnownSyncNtlmPasswords;
(function (KnownSyncNtlmPasswords) {
    /** Enabled */
    KnownSyncNtlmPasswords["Enabled"] = "Enabled";
    /** Disabled */
    KnownSyncNtlmPasswords["Disabled"] = "Disabled";
})(KnownSyncNtlmPasswords || (KnownSyncNtlmPasswords = {}));
/** Known values of {@link SyncKerberosPasswords} that the service accepts. */
export var KnownSyncKerberosPasswords;
(function (KnownSyncKerberosPasswords) {
    /** Enabled */
    KnownSyncKerberosPasswords["Enabled"] = "Enabled";
    /** Disabled */
    KnownSyncKerberosPasswords["Disabled"] = "Disabled";
})(KnownSyncKerberosPasswords || (KnownSyncKerberosPasswords = {}));
/** Known values of {@link SyncOnPremPasswords} that the service accepts. */
export var KnownSyncOnPremPasswords;
(function (KnownSyncOnPremPasswords) {
    /** Enabled */
    KnownSyncOnPremPasswords["Enabled"] = "Enabled";
    /** Disabled */
    KnownSyncOnPremPasswords["Disabled"] = "Disabled";
})(KnownSyncOnPremPasswords || (KnownSyncOnPremPasswords = {}));
/** Known values of {@link KerberosRc4Encryption} that the service accepts. */
export var KnownKerberosRc4Encryption;
(function (KnownKerberosRc4Encryption) {
    /** Enabled */
    KnownKerberosRc4Encryption["Enabled"] = "Enabled";
    /** Disabled */
    KnownKerberosRc4Encryption["Disabled"] = "Disabled";
})(KnownKerberosRc4Encryption || (KnownKerberosRc4Encryption = {}));
/** Known values of {@link KerberosArmoring} that the service accepts. */
export var KnownKerberosArmoring;
(function (KnownKerberosArmoring) {
    /** Enabled */
    KnownKerberosArmoring["Enabled"] = "Enabled";
    /** Disabled */
    KnownKerberosArmoring["Disabled"] = "Disabled";
})(KnownKerberosArmoring || (KnownKerberosArmoring = {}));
/** Known values of {@link FilteredSync} that the service accepts. */
export var KnownFilteredSync;
(function (KnownFilteredSync) {
    /** Enabled */
    KnownFilteredSync["Enabled"] = "Enabled";
    /** Disabled */
    KnownFilteredSync["Disabled"] = "Disabled";
})(KnownFilteredSync || (KnownFilteredSync = {}));
/** Known values of {@link NotifyGlobalAdmins} that the service accepts. */
export var KnownNotifyGlobalAdmins;
(function (KnownNotifyGlobalAdmins) {
    /** Enabled */
    KnownNotifyGlobalAdmins["Enabled"] = "Enabled";
    /** Disabled */
    KnownNotifyGlobalAdmins["Disabled"] = "Disabled";
})(KnownNotifyGlobalAdmins || (KnownNotifyGlobalAdmins = {}));
/** Known values of {@link NotifyDcAdmins} that the service accepts. */
export var KnownNotifyDcAdmins;
(function (KnownNotifyDcAdmins) {
    /** Enabled */
    KnownNotifyDcAdmins["Enabled"] = "Enabled";
    /** Disabled */
    KnownNotifyDcAdmins["Disabled"] = "Disabled";
})(KnownNotifyDcAdmins || (KnownNotifyDcAdmins = {}));
/** Known values of {@link Status} that the service accepts. */
export var KnownStatus;
(function (KnownStatus) {
    /** None */
    KnownStatus["None"] = "None";
    /** Running */
    KnownStatus["Running"] = "Running";
    /** OK */
    KnownStatus["OK"] = "OK";
    /** Failure */
    KnownStatus["Failure"] = "Failure";
    /** Warning */
    KnownStatus["Warning"] = "Warning";
    /** Skipped */
    KnownStatus["Skipped"] = "Skipped";
})(KnownStatus || (KnownStatus = {}));
/** Known values of {@link CreatedByType} that the service accepts. */
export var KnownCreatedByType;
(function (KnownCreatedByType) {
    /** User */
    KnownCreatedByType["User"] = "User";
    /** Application */
    KnownCreatedByType["Application"] = "Application";
    /** ManagedIdentity */
    KnownCreatedByType["ManagedIdentity"] = "ManagedIdentity";
    /** Key */
    KnownCreatedByType["Key"] = "Key";
})(KnownCreatedByType || (KnownCreatedByType = {}));
