// @aws-sdk/client-gamesparks/dist-cjs/GameSparks.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.GameSparks = void 0;
const smithy_client_1 = require("@smithy/smithy-client");
const CreateGameCommand_1 = require("./commands/CreateGameCommand");
const CreateSnapshotCommand_1 = require("./commands/CreateSnapshotCommand");
const CreateStageCommand_1 = require("./commands/CreateStageCommand");
const DeleteGameCommand_1 = require("./commands/DeleteGameCommand");
const DeleteStageCommand_1 = require("./commands/DeleteStageCommand");
const DisconnectPlayerCommand_1 = require("./commands/DisconnectPlayerCommand");
const ExportSnapshotCommand_1 = require("./commands/ExportSnapshotCommand");
const GetExtensionCommand_1 = require("./commands/GetExtensionCommand");
const GetExtensionVersionCommand_1 = require("./commands/GetExtensionVersionCommand");
const GetGameCommand_1 = require("./commands/GetGameCommand");
const GetGameConfigurationCommand_1 = require("./commands/GetGameConfigurationCommand");
const GetGeneratedCodeJobCommand_1 = require("./commands/GetGeneratedCodeJobCommand");
const GetPlayerConnectionStatusCommand_1 = require("./commands/GetPlayerConnectionStatusCommand");
const GetSnapshotCommand_1 = require("./commands/GetSnapshotCommand");
const GetStageCommand_1 = require("./commands/GetStageCommand");
const GetStageDeploymentCommand_1 = require("./commands/GetStageDeploymentCommand");
const ImportGameConfigurationCommand_1 = require("./commands/ImportGameConfigurationCommand");
const ListExtensionsCommand_1 = require("./commands/ListExtensionsCommand");
const ListExtensionVersionsCommand_1 = require("./commands/ListExtensionVersionsCommand");
const ListGamesCommand_1 = require("./commands/ListGamesCommand");
const ListGeneratedCodeJobsCommand_1 = require("./commands/ListGeneratedCodeJobsCommand");
const ListSnapshotsCommand_1 = require("./commands/ListSnapshotsCommand");
const ListStageDeploymentsCommand_1 = require("./commands/ListStageDeploymentsCommand");
const ListStagesCommand_1 = require("./commands/ListStagesCommand");
const ListTagsForResourceCommand_1 = require("./commands/ListTagsForResourceCommand");
const StartGeneratedCodeJobCommand_1 = require("./commands/StartGeneratedCodeJobCommand");
const StartStageDeploymentCommand_1 = require("./commands/StartStageDeploymentCommand");
const TagResourceCommand_1 = require("./commands/TagResourceCommand");
const UntagResourceCommand_1 = require("./commands/UntagResourceCommand");
const UpdateGameCommand_1 = require("./commands/UpdateGameCommand");
const UpdateGameConfigurationCommand_1 = require("./commands/UpdateGameConfigurationCommand");
const UpdateSnapshotCommand_1 = require("./commands/UpdateSnapshotCommand");
const UpdateStageCommand_1 = require("./commands/UpdateStageCommand");
const GameSparksClient_1 = require("./GameSparksClient");
const commands = {
    CreateGameCommand: CreateGameCommand_1.CreateGameCommand,
    CreateSnapshotCommand: CreateSnapshotCommand_1.CreateSnapshotCommand,
    CreateStageCommand: CreateStageCommand_1.CreateStageCommand,
    DeleteGameCommand: DeleteGameCommand_1.DeleteGameCommand,
    DeleteStageCommand: DeleteStageCommand_1.DeleteStageCommand,
    DisconnectPlayerCommand: DisconnectPlayerCommand_1.DisconnectPlayerCommand,
    ExportSnapshotCommand: ExportSnapshotCommand_1.ExportSnapshotCommand,
    GetExtensionCommand: GetExtensionCommand_1.GetExtensionCommand,
    GetExtensionVersionCommand: GetExtensionVersionCommand_1.GetExtensionVersionCommand,
    GetGameCommand: GetGameCommand_1.GetGameCommand,
    GetGameConfigurationCommand: GetGameConfigurationCommand_1.GetGameConfigurationCommand,
    GetGeneratedCodeJobCommand: GetGeneratedCodeJobCommand_1.GetGeneratedCodeJobCommand,
    GetPlayerConnectionStatusCommand: GetPlayerConnectionStatusCommand_1.GetPlayerConnectionStatusCommand,
    GetSnapshotCommand: GetSnapshotCommand_1.GetSnapshotCommand,
    GetStageCommand: GetStageCommand_1.GetStageCommand,
    GetStageDeploymentCommand: GetStageDeploymentCommand_1.GetStageDeploymentCommand,
    ImportGameConfigurationCommand: ImportGameConfigurationCommand_1.ImportGameConfigurationCommand,
    ListExtensionsCommand: ListExtensionsCommand_1.ListExtensionsCommand,
    ListExtensionVersionsCommand: ListExtensionVersionsCommand_1.ListExtensionVersionsCommand,
    ListGamesCommand: ListGamesCommand_1.ListGamesCommand,
    ListGeneratedCodeJobsCommand: ListGeneratedCodeJobsCommand_1.ListGeneratedCodeJobsCommand,
    ListSnapshotsCommand: ListSnapshotsCommand_1.ListSnapshotsCommand,
    ListStageDeploymentsCommand: ListStageDeploymentsCommand_1.ListStageDeploymentsCommand,
    ListStagesCommand: ListStagesCommand_1.ListStagesCommand,
    ListTagsForResourceCommand: ListTagsForResourceCommand_1.ListTagsForResourceCommand,
    StartGeneratedCodeJobCommand: StartGeneratedCodeJobCommand_1.StartGeneratedCodeJobCommand,
    StartStageDeploymentCommand: StartStageDeploymentCommand_1.StartStageDeploymentCommand,
    TagResourceCommand: TagResourceCommand_1.TagResourceCommand,
    UntagResourceCommand: UntagResourceCommand_1.UntagResourceCommand,
    UpdateGameCommand: UpdateGameCommand_1.UpdateGameCommand,
    UpdateGameConfigurationCommand: UpdateGameConfigurationCommand_1.UpdateGameConfigurationCommand,
    UpdateSnapshotCommand: UpdateSnapshotCommand_1.UpdateSnapshotCommand,
    UpdateStageCommand: UpdateStageCommand_1.UpdateStageCommand,
};
class GameSparks extends GameSparksClient_1.GameSparksClient {
}
exports.GameSparks = GameSparks;
(0, smithy_client_1.createAggregatedClient)(commands, GameSparks);
