// @aws-sdk/client-workdocs/dist-es/models/models_0.js
import { SENSITIVE_STRING } from "@smithy/smithy-client";
import { WorkDocsServiceException as __BaseException } from "./WorkDocsServiceException";
export class ConcurrentModificationException extends __BaseException {
    constructor(opts) {
        super({
            name: "ConcurrentModificationException",
            $fault: "client",
            ...opts,
        });
        this.name = "ConcurrentModificationException";
        this.$fault = "client";
        Object.setPrototypeOf(this, ConcurrentModificationException.prototype);
        this.Message = opts.Message;
    }
}
export class EntityNotExistsException extends __BaseException {
    constructor(opts) {
        super({
            name: "EntityNotExistsException",
            $fault: "client",
            ...opts,
        });
        this.name = "EntityNotExistsException";
        this.$fault = "client";
        Object.setPrototypeOf(this, EntityNotExistsException.prototype);
        this.Message = opts.Message;
        this.EntityIds = opts.EntityIds;
    }
}
export class FailedDependencyException extends __BaseException {
    constructor(opts) {
        super({
            name: "FailedDependencyException",
            $fault: "client",
            ...opts,
        });
        this.name = "FailedDependencyException";
        this.$fault = "client";
        Object.setPrototypeOf(this, FailedDependencyException.prototype);
        this.Message = opts.Message;
    }
}
export class ProhibitedStateException extends __BaseException {
    constructor(opts) {
        super({
            name: "ProhibitedStateException",
            $fault: "client",
            ...opts,
        });
        this.name = "ProhibitedStateException";
        this.$fault = "client";
        Object.setPrototypeOf(this, ProhibitedStateException.prototype);
        this.Message = opts.Message;
    }
}
export class ServiceUnavailableException extends __BaseException {
    constructor(opts) {
        super({
            name: "ServiceUnavailableException",
            $fault: "server",
            ...opts,
        });
        this.name = "ServiceUnavailableException";
        this.$fault = "server";
        Object.setPrototypeOf(this, ServiceUnavailableException.prototype);
        this.Message = opts.Message;
    }
}
export class UnauthorizedOperationException extends __BaseException {
    constructor(opts) {
        super({
            name: "UnauthorizedOperationException",
            $fault: "client",
            ...opts,
        });
        this.name = "UnauthorizedOperationException";
        this.$fault = "client";
        Object.setPrototypeOf(this, UnauthorizedOperationException.prototype);
        this.Message = opts.Message;
        this.Code = opts.Code;
    }
}
export class UnauthorizedResourceAccessException extends __BaseException {
    constructor(opts) {
        super({
            name: "UnauthorizedResourceAccessException",
            $fault: "client",
            ...opts,
        });
        this.name = "UnauthorizedResourceAccessException";
        this.$fault = "client";
        Object.setPrototypeOf(this, UnauthorizedResourceAccessException.prototype);
        this.Message = opts.Message;
    }
}
export const LocaleType = {
    DE: "de",
    DEFAULT: "default",
    EN: "en",
    ES: "es",
    FR: "fr",
    JA: "ja",
    KO: "ko",
    PT_BR: "pt_BR",
    RU: "ru",
    ZH_CN: "zh_CN",
    ZH_TW: "zh_TW",
};
export const UserStatusType = {
    ACTIVE: "ACTIVE",
    INACTIVE: "INACTIVE",
    PENDING: "PENDING",
};
export const StorageType = {
    QUOTA: "QUOTA",
    UNLIMITED: "UNLIMITED",
};
export const UserType = {
    ADMIN: "ADMIN",
    MINIMALUSER: "MINIMALUSER",
    POWERUSER: "POWERUSER",
    USER: "USER",
    WORKSPACESUSER: "WORKSPACESUSER",
};
export const CommentStatusType = {
    DELETED: "DELETED",
    DRAFT: "DRAFT",
    PUBLISHED: "PUBLISHED",
};
export const ResourceType = {
    DOCUMENT: "DOCUMENT",
    FOLDER: "FOLDER",
};
export const ActivityType = {
    DOCUMENT_ANNOTATION_ADDED: "DOCUMENT_ANNOTATION_ADDED",
    DOCUMENT_ANNOTATION_DELETED: "DOCUMENT_ANNOTATION_DELETED",
    DOCUMENT_CHECKED_IN: "DOCUMENT_CHECKED_IN",
    DOCUMENT_CHECKED_OUT: "DOCUMENT_CHECKED_OUT",
    DOCUMENT_COMMENT_ADDED: "DOCUMENT_COMMENT_ADDED",
    DOCUMENT_COMMENT_DELETED: "DOCUMENT_COMMENT_DELETED",
    DOCUMENT_MOVED: "DOCUMENT_MOVED",
    DOCUMENT_RECYCLED: "DOCUMENT_RECYCLED",
    DOCUMENT_RENAMED: "DOCUMENT_RENAMED",
    DOCUMENT_RESTORED: "DOCUMENT_RESTORED",
    DOCUMENT_REVERTED: "DOCUMENT_REVERTED",
    DOCUMENT_SHAREABLE_LINK_CREATED: "DOCUMENT_SHAREABLE_LINK_CREATED",
    DOCUMENT_SHAREABLE_LINK_PERMISSION_CHANGED: "DOCUMENT_SHAREABLE_LINK_PERMISSION_CHANGED",
    DOCUMENT_SHAREABLE_LINK_REMOVED: "DOCUMENT_SHAREABLE_LINK_REMOVED",
    DOCUMENT_SHARED: "DOCUMENT_SHARED",
    DOCUMENT_SHARE_PERMISSION_CHANGED: "DOCUMENT_SHARE_PERMISSION_CHANGED",
    DOCUMENT_UNSHARED: "DOCUMENT_UNSHARED",
    DOCUMENT_VERSION_DELETED: "DOCUMENT_VERSION_DELETED",
    DOCUMENT_VERSION_DOWNLOADED: "DOCUMENT_VERSION_DOWNLOADED",
    DOCUMENT_VERSION_UPLOADED: "DOCUMENT_VERSION_UPLOADED",
    DOCUMENT_VERSION_VIEWED: "DOCUMENT_VERSION_VIEWED",
    FOLDER_CREATED: "FOLDER_CREATED",
    FOLDER_DELETED: "FOLDER_DELETED",
    FOLDER_MOVED: "FOLDER_MOVED",
    FOLDER_RECYCLED: "FOLDER_RECYCLED",
    FOLDER_RENAMED: "FOLDER_RENAMED",
    FOLDER_RESTORED: "FOLDER_RESTORED",
    FOLDER_SHAREABLE_LINK_CREATED: "FOLDER_SHAREABLE_LINK_CREATED",
    FOLDER_SHAREABLE_LINK_PERMISSION_CHANGED: "FOLDER_SHAREABLE_LINK_PERMISSION_CHANGED",
    FOLDER_SHAREABLE_LINK_REMOVED: "FOLDER_SHAREABLE_LINK_REMOVED",
    FOLDER_SHARED: "FOLDER_SHARED",
    FOLDER_SHARE_PERMISSION_CHANGED: "FOLDER_SHARE_PERMISSION_CHANGED",
    FOLDER_UNSHARED: "FOLDER_UNSHARED",
};
export const AdditionalResponseFieldType = {
    WEBURL: "WEBURL",
};
export const RoleType = {
    CONTRIBUTOR: "CONTRIBUTOR",
    COOWNER: "COOWNER",
    OWNER: "OWNER",
    VIEWER: "VIEWER",
};
export const PrincipalType = {
    ANONYMOUS: "ANONYMOUS",
    GROUP: "GROUP",
    INVITE: "INVITE",
    ORGANIZATION: "ORGANIZATION",
    USER: "USER",
};
export const ShareStatusType = {
    FAILURE: "FAILURE",
    SUCCESS: "SUCCESS",
};
export const CommentVisibilityType = {
    PRIVATE: "PRIVATE",
    PUBLIC: "PUBLIC",
};
export class DocumentLockedForCommentsException extends __BaseException {
    constructor(opts) {
        super({
            name: "DocumentLockedForCommentsException",
            $fault: "client",
            ...opts,
        });
        this.name = "DocumentLockedForCommentsException";
        this.$fault = "client";
        Object.setPrototypeOf(this, DocumentLockedForCommentsException.prototype);
        this.Message = opts.Message;
    }
}
export class InvalidCommentOperationException extends __BaseException {
    constructor(opts) {
        super({
            name: "InvalidCommentOperationException",
            $fault: "client",
            ...opts,
        });
        this.name = "InvalidCommentOperationException";
        this.$fault = "client";
        Object.setPrototypeOf(this, InvalidCommentOperationException.prototype);
        this.Message = opts.Message;
    }
}
export class CustomMetadataLimitExceededException extends __BaseException {
    constructor(opts) {
        super({
            name: "CustomMetadataLimitExceededException",
            $fault: "client",
            ...opts,
        });
        this.name = "CustomMetadataLimitExceededException";
        this.$fault = "client";
        Object.setPrototypeOf(this, CustomMetadataLimitExceededException.prototype);
        this.Message = opts.Message;
    }
}
export class ConflictingOperationException extends __BaseException {
    constructor(opts) {
        super({
            name: "ConflictingOperationException",
            $fault: "client",
            ...opts,
        });
        this.name = "ConflictingOperationException";
        this.$fault = "client";
        Object.setPrototypeOf(this, ConflictingOperationException.prototype);
        this.Message = opts.Message;
    }
}
export const ResourceStateType = {
    ACTIVE: "ACTIVE",
    RECYCLED: "RECYCLED",
    RECYCLING: "RECYCLING",
    RESTORING: "RESTORING",
};
export class EntityAlreadyExistsException extends __BaseException {
    constructor(opts) {
        super({
            name: "EntityAlreadyExistsException",
            $fault: "client",
            ...opts,
        });
        this.name = "EntityAlreadyExistsException";
        this.$fault = "client";
        Object.setPrototypeOf(this, EntityAlreadyExistsException.prototype);
        this.Message = opts.Message;
    }
}
export class LimitExceededException extends __BaseException {
    constructor(opts) {
        super({
            name: "LimitExceededException",
            $fault: "client",
            ...opts,
        });
        this.name = "LimitExceededException";
        this.$fault = "client";
        Object.setPrototypeOf(this, LimitExceededException.prototype);
        this.Message = opts.Message;
    }
}
export class TooManyLabelsException extends __BaseException {
    constructor(opts) {
        super({
            name: "TooManyLabelsException",
            $fault: "client",
            ...opts,
        });
        this.name = "TooManyLabelsException";
        this.$fault = "client";
        Object.setPrototypeOf(this, TooManyLabelsException.prototype);
        this.Message = opts.Message;
    }
}
export const SubscriptionProtocolType = {
    HTTPS: "HTTPS",
    SQS: "SQS",
};
export const SubscriptionType = {
    ALL: "ALL",
};
export class InvalidArgumentException extends __BaseException {
    constructor(opts) {
        super({
            name: "InvalidArgumentException",
            $fault: "client",
            ...opts,
        });
        this.name = "InvalidArgumentException";
        this.$fault = "client";
        Object.setPrototypeOf(this, InvalidArgumentException.prototype);
        this.Message = opts.Message;
    }
}
export class TooManySubscriptionsException extends __BaseException {
    constructor(opts) {
        super({
            name: "TooManySubscriptionsException",
            $fault: "client",
            ...opts,
        });
        this.name = "TooManySubscriptionsException";
        this.$fault = "client";
        Object.setPrototypeOf(this, TooManySubscriptionsException.prototype);
        this.Message = opts.Message;
    }
}
export class InvalidOperationException extends __BaseException {
    constructor(opts) {
        super({
            name: "InvalidOperationException",
            $fault: "client",
            ...opts,
        });
        this.name = "InvalidOperationException";
        this.$fault = "client";
        Object.setPrototypeOf(this, InvalidOperationException.prototype);
        this.Message = opts.Message;
    }
}
export const DocumentSourceType = {
    ORIGINAL: "ORIGINAL",
    WITH_COMMENTS: "WITH_COMMENTS",
};
export const DocumentStatusType = {
    ACTIVE: "ACTIVE",
    INITIALIZED: "INITIALIZED",
};
export const DocumentThumbnailType = {
    LARGE: "LARGE",
    SMALL: "SMALL",
    SMALL_HQ: "SMALL_HQ",
};
export class InvalidPasswordException extends __BaseException {
    constructor(opts) {
        super({
            name: "InvalidPasswordException",
            $fault: "client",
            ...opts,
        });
        this.name = "InvalidPasswordException";
        this.$fault = "client";
        Object.setPrototypeOf(this, InvalidPasswordException.prototype);
        this.Message = opts.Message;
    }
}
export const OrderType = {
    ASCENDING: "ASCENDING",
    DESCENDING: "DESCENDING",
};
export const ResourceSortType = {
    DATE: "DATE",
    NAME: "NAME",
};
export const FolderContentType = {
    ALL: "ALL",
    DOCUMENT: "DOCUMENT",
    FOLDER: "FOLDER",
};
export const RolePermissionType = {
    DIRECT: "DIRECT",
    INHERITED: "INHERITED",
};
export const UserFilterType = {
    ACTIVE_PENDING: "ACTIVE_PENDING",
    ALL: "ALL",
};
export const UserSortType = {
    FULL_NAME: "FULL_NAME",
    STORAGE_LIMIT: "STORAGE_LIMIT",
    STORAGE_USED: "STORAGE_USED",
    USER_NAME: "USER_NAME",
    USER_STATUS: "USER_STATUS",
};
export class RequestedEntityTooLargeException extends __BaseException {
    constructor(opts) {
        super({
            name: "RequestedEntityTooLargeException",
            $fault: "client",
            ...opts,
        });
        this.name = "RequestedEntityTooLargeException";
        this.$fault = "client";
        Object.setPrototypeOf(this, RequestedEntityTooLargeException.prototype);
        this.Message = opts.Message;
    }
}
export const ResourceCollectionType = {
    SHARED_WITH_ME: "SHARED_WITH_ME",
};
export class DraftUploadOutOfSyncException extends __BaseException {
    constructor(opts) {
        super({
            name: "DraftUploadOutOfSyncException",
            $fault: "client",
            ...opts,
        });
        this.name = "DraftUploadOutOfSyncException";
        this.$fault = "client";
        Object.setPrototypeOf(this, DraftUploadOutOfSyncException.prototype);
        this.Message = opts.Message;
    }
}
export class ResourceAlreadyCheckedOutException extends __BaseException {
    constructor(opts) {
        super({
            name: "ResourceAlreadyCheckedOutException",
            $fault: "client",
            ...opts,
        });
        this.name = "ResourceAlreadyCheckedOutException";
        this.$fault = "client";
        Object.setPrototypeOf(this, ResourceAlreadyCheckedOutException.prototype);
        this.Message = opts.Message;
    }
}
export class StorageLimitExceededException extends __BaseException {
    constructor(opts) {
        super({
            name: "StorageLimitExceededException",
            $fault: "client",
            ...opts,
        });
        this.name = "StorageLimitExceededException";
        this.$fault = "client";
        Object.setPrototypeOf(this, StorageLimitExceededException.prototype);
        this.Message = opts.Message;
    }
}
export class StorageLimitWillExceedException extends __BaseException {
    constructor(opts) {
        super({
            name: "StorageLimitWillExceedException",
            $fault: "client",
            ...opts,
        });
        this.name = "StorageLimitWillExceedException";
        this.$fault = "client";
        Object.setPrototypeOf(this, StorageLimitWillExceedException.prototype);
        this.Message = opts.Message;
    }
}
export const ContentCategoryType = {
    AUDIO: "AUDIO",
    DOCUMENT: "DOCUMENT",
    IMAGE: "IMAGE",
    OTHER: "OTHER",
    PDF: "PDF",
    PRESENTATION: "PRESENTATION",
    SOURCE_CODE: "SOURCE_CODE",
    SPREADSHEET: "SPREADSHEET",
    VIDEO: "VIDEO",
};
export const PrincipalRoleType = {
    CONTRIBUTOR: "CONTRIBUTOR",
    COOWNER: "COOWNER",
    OWNER: "OWNER",
    VIEWER: "VIEWER",
};
export const SearchResourceType = {
    COMMENT: "COMMENT",
    DOCUMENT: "DOCUMENT",
    DOCUMENT_VERSION: "DOCUMENT_VERSION",
    FOLDER: "FOLDER",
};
export const SearchCollectionType = {
    OWNED: "OWNED",
    SHARED_WITH_ME: "SHARED_WITH_ME",
};
export const LanguageCodeType = {
    AR: "AR",
    BG: "BG",
    BN: "BN",
    CS: "CS",
    DA: "DA",
    DE: "DE",
    DEFAULT: "DEFAULT",
    EL: "EL",
    EN: "EN",
    ES: "ES",
    FA: "FA",
    FI: "FI",
    FR: "FR",
    HI: "HI",
    HU: "HU",
    ID: "ID",
    IT: "IT",
    JA: "JA",
    KO: "KO",
    LT: "LT",
    LV: "LV",
    NL: "NL",
    NO: "NO",
    PT: "PT",
    RO: "RO",
    RU: "RU",
    SV: "SV",
    SW: "SW",
    TH: "TH",
    TR: "TR",
    ZH: "ZH",
};
export const OrderByFieldType = {
    CREATED_TIMESTAMP: "CREATED_TIMESTAMP",
    MODIFIED_TIMESTAMP: "MODIFIED_TIMESTAMP",
    NAME: "NAME",
    RELEVANCE: "RELEVANCE",
    SIZE: "SIZE",
};
export const SortOrder = {
    ASC: "ASC",
    DESC: "DESC",
};
export const SearchQueryScopeType = {
    CONTENT: "CONTENT",
    NAME: "NAME",
};
export const ResponseItemType = {
    COMMENT: "COMMENT",
    DOCUMENT: "DOCUMENT",
    DOCUMENT_VERSION: "DOCUMENT_VERSION",
    FOLDER: "FOLDER",
};
export const DocumentVersionStatus = {
    ACTIVE: "ACTIVE",
};
export class DeactivatingLastSystemUserException extends __BaseException {
    constructor(opts) {
        super({
            name: "DeactivatingLastSystemUserException",
            $fault: "client",
            ...opts,
        });
        this.name = "DeactivatingLastSystemUserException";
        this.$fault = "client";
        Object.setPrototypeOf(this, DeactivatingLastSystemUserException.prototype);
        this.Message = opts.Message;
        this.Code = opts.Code;
    }
}
export class IllegalUserStateException extends __BaseException {
    constructor(opts) {
        super({
            name: "IllegalUserStateException",
            $fault: "client",
            ...opts,
        });
        this.name = "IllegalUserStateException";
        this.$fault = "client";
        Object.setPrototypeOf(this, IllegalUserStateException.prototype);
        this.Message = opts.Message;
    }
}
export const BooleanEnumType = {
    FALSE: "FALSE",
    TRUE: "TRUE",
};
export const AbortDocumentVersionUploadRequestFilterSensitiveLog = (obj) => ({
    ...obj,
    ...(obj.AuthenticationToken && { AuthenticationToken: SENSITIVE_STRING }),
});
export const ActivateUserRequestFilterSensitiveLog = (obj) => ({
    ...obj,
    ...(obj.AuthenticationToken && { AuthenticationToken: SENSITIVE_STRING }),
});
export const UserFilterSensitiveLog = (obj) => ({
    ...obj,
    ...(obj.Username && { Username: SENSITIVE_STRING }),
    ...(obj.EmailAddress && { EmailAddress: SENSITIVE_STRING }),
    ...(obj.GivenName && { GivenName: SENSITIVE_STRING }),
    ...(obj.Surname && { Surname: SENSITIVE_STRING }),
});
export const ActivateUserResponseFilterSensitiveLog = (obj) => ({
    ...obj,
    ...(obj.User && { User: UserFilterSensitiveLog(obj.User) }),
});
export const CommentMetadataFilterSensitiveLog = (obj) => ({
    ...obj,
    ...(obj.Contributor && { Contributor: UserFilterSensitiveLog(obj.Contributor) }),
});
export const UserMetadataFilterSensitiveLog = (obj) => ({
    ...obj,
    ...(obj.Username && { Username: SENSITIVE_STRING }),
    ...(obj.GivenName && { GivenName: SENSITIVE_STRING }),
    ...(obj.Surname && { Surname: SENSITIVE_STRING }),
    ...(obj.EmailAddress && { EmailAddress: SENSITIVE_STRING }),
});
export const ResourceMetadataFilterSensitiveLog = (obj) => ({
    ...obj,
    ...(obj.Name && { Name: SENSITIVE_STRING }),
    ...(obj.OriginalName && { OriginalName: SENSITIVE_STRING }),
    ...(obj.Owner && { Owner: UserMetadataFilterSensitiveLog(obj.Owner) }),
});
export const ParticipantsFilterSensitiveLog = (obj) => ({
    ...obj,
    ...(obj.Users && { Users: obj.Users.map((item) => UserMetadataFilterSensitiveLog(item)) }),
});
export const ActivityFilterSensitiveLog = (obj) => ({
    ...obj,
    ...(obj.Initiator && { Initiator: UserMetadataFilterSensitiveLog(obj.Initiator) }),
    ...(obj.Participants && { Participants: ParticipantsFilterSensitiveLog(obj.Participants) }),
    ...(obj.ResourceMetadata && { ResourceMetadata: ResourceMetadataFilterSensitiveLog(obj.ResourceMetadata) }),
    ...(obj.OriginalParent && { OriginalParent: ResourceMetadataFilterSensitiveLog(obj.OriginalParent) }),
    ...(obj.CommentMetadata && { CommentMetadata: CommentMetadataFilterSensitiveLog(obj.CommentMetadata) }),
});
export const NotificationOptionsFilterSensitiveLog = (obj) => ({
    ...obj,
    ...(obj.EmailMessage && { EmailMessage: SENSITIVE_STRING }),
});
export const AddResourcePermissionsRequestFilterSensitiveLog = (obj) => ({
    ...obj,
    ...(obj.AuthenticationToken && { AuthenticationToken: SENSITIVE_STRING }),
    ...(obj.NotificationOptions && {
        NotificationOptions: NotificationOptionsFilterSensitiveLog(obj.NotificationOptions),
    }),
});
export const ShareResultFilterSensitiveLog = (obj) => ({
    ...obj,
    ...(obj.StatusMessage && { StatusMessage: SENSITIVE_STRING }),
});
export const AddResourcePermissionsResponseFilterSensitiveLog = (obj) => ({
    ...obj,
    ...(obj.ShareResults && { ShareResults: obj.ShareResults.map((item) => ShareResultFilterSensitiveLog(item)) }),
});
export const CreateCommentRequestFilterSensitiveLog = (obj) => ({
    ...obj,
    ...(obj.AuthenticationToken && { AuthenticationToken: SENSITIVE_STRING }),
    ...(obj.Text && { Text: SENSITIVE_STRING }),
});
export const CommentFilterSensitiveLog = (obj) => ({
    ...obj,
    ...(obj.Text && { Text: SENSITIVE_STRING }),
    ...(obj.Contributor && { Contributor: UserFilterSensitiveLog(obj.Contributor) }),
});
export const CreateCommentResponseFilterSensitiveLog = (obj) => ({
    ...obj,
    ...(obj.Comment && { Comment: CommentFilterSensitiveLog(obj.Comment) }),
});
export const CreateCustomMetadataRequestFilterSensitiveLog = (obj) => ({
    ...obj,
    ...(obj.AuthenticationToken && { AuthenticationToken: SENSITIVE_STRING }),
});
export const CreateFolderRequestFilterSensitiveLog = (obj) => ({
    ...obj,
    ...(obj.AuthenticationToken && { AuthenticationToken: SENSITIVE_STRING }),
    ...(obj.Name && { Name: SENSITIVE_STRING }),
});
export const FolderMetadataFilterSensitiveLog = (obj) => ({
    ...obj,
    ...(obj.Name && { Name: SENSITIVE_STRING }),
});
export const CreateFolderResponseFilterSensitiveLog = (obj) => ({
    ...obj,
    ...(obj.Metadata && { Metadata: FolderMetadataFilterSensitiveLog(obj.Metadata) }),
});
export const CreateLabelsRequestFilterSensitiveLog = (obj) => ({
    ...obj,
    ...(obj.AuthenticationToken && { AuthenticationToken: SENSITIVE_STRING }),
});
export const CreateUserRequestFilterSensitiveLog = (obj) => ({
    ...obj,
    ...(obj.Username && { Username: SENSITIVE_STRING }),
    ...(obj.EmailAddress && { EmailAddress: SENSITIVE_STRING }),
    ...(obj.GivenName && { GivenName: SENSITIVE_STRING }),
    ...(obj.Surname && { Surname: SENSITIVE_STRING }),
    ...(obj.Password && { Password: SENSITIVE_STRING }),
    ...(obj.AuthenticationToken && { AuthenticationToken: SENSITIVE_STRING }),
});
export const CreateUserResponseFilterSensitiveLog = (obj) => ({
    ...obj,
    ...(obj.User && { User: UserFilterSensitiveLog(obj.User) }),
});
export const DeactivateUserRequestFilterSensitiveLog = (obj) => ({
    ...obj,
    ...(obj.AuthenticationToken && { AuthenticationToken: SENSITIVE_STRING }),
});
export const DeleteCommentRequestFilterSensitiveLog = (obj) => ({
    ...obj,
    ...(obj.AuthenticationToken && { AuthenticationToken: SENSITIVE_STRING }),
});
export const DeleteCustomMetadataRequestFilterSensitiveLog = (obj) => ({
    ...obj,
    ...(obj.AuthenticationToken && { AuthenticationToken: SENSITIVE_STRING }),
});
export const DeleteDocumentRequestFilterSensitiveLog = (obj) => ({
    ...obj,
    ...(obj.AuthenticationToken && { AuthenticationToken: SENSITIVE_STRING }),
});
export const DeleteDocumentVersionRequestFilterSensitiveLog = (obj) => ({
    ...obj,
    ...(obj.AuthenticationToken && { AuthenticationToken: SENSITIVE_STRING }),
});
export const DeleteFolderRequestFilterSensitiveLog = (obj) => ({
    ...obj,
    ...(obj.AuthenticationToken && { AuthenticationToken: SENSITIVE_STRING }),
});
export const DeleteFolderContentsRequestFilterSensitiveLog = (obj) => ({
    ...obj,
    ...(obj.AuthenticationToken && { AuthenticationToken: SENSITIVE_STRING }),
});
export const DeleteLabelsRequestFilterSensitiveLog = (obj) => ({
    ...obj,
    ...(obj.AuthenticationToken && { AuthenticationToken: SENSITIVE_STRING }),
});
export const DeleteUserRequestFilterSensitiveLog = (obj) => ({
    ...obj,
    ...(obj.AuthenticationToken && { AuthenticationToken: SENSITIVE_STRING }),
});
export const DescribeActivitiesRequestFilterSensitiveLog = (obj) => ({
    ...obj,
    ...(obj.AuthenticationToken && { AuthenticationToken: SENSITIVE_STRING }),
});
export const DescribeActivitiesResponseFilterSensitiveLog = (obj) => ({
    ...obj,
    ...(obj.UserActivities && { UserActivities: obj.UserActivities.map((item) => ActivityFilterSensitiveLog(item)) }),
});
export const DescribeCommentsRequestFilterSensitiveLog = (obj) => ({
    ...obj,
    ...(obj.AuthenticationToken && { AuthenticationToken: SENSITIVE_STRING }),
});
export const DescribeCommentsResponseFilterSensitiveLog = (obj) => ({
    ...obj,
    ...(obj.Comments && { Comments: obj.Comments.map((item) => CommentFilterSensitiveLog(item)) }),
});
export const DescribeDocumentVersionsRequestFilterSensitiveLog = (obj) => ({
    ...obj,
    ...(obj.AuthenticationToken && { AuthenticationToken: SENSITIVE_STRING }),
});
export const DocumentVersionMetadataFilterSensitiveLog = (obj) => ({
    ...obj,
    ...(obj.Name && { Name: SENSITIVE_STRING }),
    ...(obj.Thumbnail && { Thumbnail: SENSITIVE_STRING }),
    ...(obj.Source && { Source: SENSITIVE_STRING }),
});
export const DescribeDocumentVersionsResponseFilterSensitiveLog = (obj) => ({
    ...obj,
    ...(obj.DocumentVersions && {
        DocumentVersions: obj.DocumentVersions.map((item) => DocumentVersionMetadataFilterSensitiveLog(item)),
    }),
});
export const DescribeFolderContentsRequestFilterSensitiveLog = (obj) => ({
    ...obj,
    ...(obj.AuthenticationToken && { AuthenticationToken: SENSITIVE_STRING }),
});
export const DocumentMetadataFilterSensitiveLog = (obj) => ({
    ...obj,
    ...(obj.LatestVersionMetadata && {
        LatestVersionMetadata: DocumentVersionMetadataFilterSensitiveLog(obj.LatestVersionMetadata),
    }),
});
export const DescribeFolderContentsResponseFilterSensitiveLog = (obj) => ({
    ...obj,
    ...(obj.Folders && { Folders: obj.Folders.map((item) => FolderMetadataFilterSensitiveLog(item)) }),
    ...(obj.Documents && { Documents: obj.Documents.map((item) => DocumentMetadataFilterSensitiveLog(item)) }),
});
export const DescribeGroupsRequestFilterSensitiveLog = (obj) => ({
    ...obj,
    ...(obj.AuthenticationToken && { AuthenticationToken: SENSITIVE_STRING }),
    ...(obj.SearchQuery && { SearchQuery: SENSITIVE_STRING }),
});
export const DescribeResourcePermissionsRequestFilterSensitiveLog = (obj) => ({
    ...obj,
    ...(obj.AuthenticationToken && { AuthenticationToken: SENSITIVE_STRING }),
});
export const DescribeRootFoldersRequestFilterSensitiveLog = (obj) => ({
    ...obj,
    ...(obj.AuthenticationToken && { AuthenticationToken: SENSITIVE_STRING }),
});
export const DescribeRootFoldersResponseFilterSensitiveLog = (obj) => ({
    ...obj,
    ...(obj.Folders && { Folders: obj.Folders.map((item) => FolderMetadataFilterSensitiveLog(item)) }),
});
export const DescribeUsersRequestFilterSensitiveLog = (obj) => ({
    ...obj,
    ...(obj.AuthenticationToken && { AuthenticationToken: SENSITIVE_STRING }),
    ...(obj.Query && { Query: SENSITIVE_STRING }),
});
export const DescribeUsersResponseFilterSensitiveLog = (obj) => ({
    ...obj,
    ...(obj.Users && { Users: obj.Users.map((item) => UserFilterSensitiveLog(item)) }),
});
export const GetCurrentUserRequestFilterSensitiveLog = (obj) => ({
    ...obj,
    ...(obj.AuthenticationToken && { AuthenticationToken: SENSITIVE_STRING }),
});
export const GetCurrentUserResponseFilterSensitiveLog = (obj) => ({
    ...obj,
    ...(obj.User && { User: UserFilterSensitiveLog(obj.User) }),
});
export const GetDocumentRequestFilterSensitiveLog = (obj) => ({
    ...obj,
    ...(obj.AuthenticationToken && { AuthenticationToken: SENSITIVE_STRING }),
});
export const GetDocumentResponseFilterSensitiveLog = (obj) => ({
    ...obj,
    ...(obj.Metadata && { Metadata: DocumentMetadataFilterSensitiveLog(obj.Metadata) }),
});
export const GetDocumentPathRequestFilterSensitiveLog = (obj) => ({
    ...obj,
    ...(obj.AuthenticationToken && { AuthenticationToken: SENSITIVE_STRING }),
});
export const ResourcePathComponentFilterSensitiveLog = (obj) => ({
    ...obj,
    ...(obj.Name && { Name: SENSITIVE_STRING }),
});
export const ResourcePathFilterSensitiveLog = (obj) => ({
    ...obj,
    ...(obj.Components && { Components: obj.Components.map((item) => ResourcePathComponentFilterSensitiveLog(item)) }),
});
export const GetDocumentPathResponseFilterSensitiveLog = (obj) => ({
    ...obj,
    ...(obj.Path && { Path: ResourcePathFilterSensitiveLog(obj.Path) }),
});
export const GetDocumentVersionRequestFilterSensitiveLog = (obj) => ({
    ...obj,
    ...(obj.AuthenticationToken && { AuthenticationToken: SENSITIVE_STRING }),
});
export const GetDocumentVersionResponseFilterSensitiveLog = (obj) => ({
    ...obj,
    ...(obj.Metadata && { Metadata: DocumentVersionMetadataFilterSensitiveLog(obj.Metadata) }),
});
export const GetFolderRequestFilterSensitiveLog = (obj) => ({
    ...obj,
    ...(obj.AuthenticationToken && { AuthenticationToken: SENSITIVE_STRING }),
});
export const GetFolderResponseFilterSensitiveLog = (obj) => ({
    ...obj,
    ...(obj.Metadata && { Metadata: FolderMetadataFilterSensitiveLog(obj.Metadata) }),
});
export const GetFolderPathRequestFilterSensitiveLog = (obj) => ({
    ...obj,
    ...(obj.AuthenticationToken && { AuthenticationToken: SENSITIVE_STRING }),
});
export const GetFolderPathResponseFilterSensitiveLog = (obj) => ({
    ...obj,
    ...(obj.Path && { Path: ResourcePathFilterSensitiveLog(obj.Path) }),
});
export const GetResourcesRequestFilterSensitiveLog = (obj) => ({
    ...obj,
    ...(obj.AuthenticationToken && { AuthenticationToken: SENSITIVE_STRING }),
});
export const GetResourcesResponseFilterSensitiveLog = (obj) => ({
    ...obj,
    ...(obj.Folders && { Folders: obj.Folders.map((item) => FolderMetadataFilterSensitiveLog(item)) }),
    ...(obj.Documents && { Documents: obj.Documents.map((item) => DocumentMetadataFilterSensitiveLog(item)) }),
});
export const InitiateDocumentVersionUploadRequestFilterSensitiveLog = (obj) => ({
    ...obj,
    ...(obj.AuthenticationToken && { AuthenticationToken: SENSITIVE_STRING }),
    ...(obj.Name && { Name: SENSITIVE_STRING }),
});
export const UploadMetadataFilterSensitiveLog = (obj) => ({
    ...obj,
    ...(obj.UploadUrl && { UploadUrl: SENSITIVE_STRING }),
});
export const InitiateDocumentVersionUploadResponseFilterSensitiveLog = (obj) => ({
    ...obj,
    ...(obj.Metadata && { Metadata: DocumentMetadataFilterSensitiveLog(obj.Metadata) }),
    ...(obj.UploadMetadata && { UploadMetadata: UploadMetadataFilterSensitiveLog(obj.UploadMetadata) }),
});
export const RemoveAllResourcePermissionsRequestFilterSensitiveLog = (obj) => ({
    ...obj,
    ...(obj.AuthenticationToken && { AuthenticationToken: SENSITIVE_STRING }),
});
export const RemoveResourcePermissionRequestFilterSensitiveLog = (obj) => ({
    ...obj,
    ...(obj.AuthenticationToken && { AuthenticationToken: SENSITIVE_STRING }),
});
export const RestoreDocumentVersionsRequestFilterSensitiveLog = (obj) => ({
    ...obj,
    ...(obj.AuthenticationToken && { AuthenticationToken: SENSITIVE_STRING }),
});
export const SearchResourcesRequestFilterSensitiveLog = (obj) => ({
    ...obj,
    ...(obj.AuthenticationToken && { AuthenticationToken: SENSITIVE_STRING }),
    ...(obj.QueryText && { QueryText: SENSITIVE_STRING }),
});
export const ResponseItemFilterSensitiveLog = (obj) => ({
    ...obj,
    ...(obj.WebUrl && { WebUrl: SENSITIVE_STRING }),
    ...(obj.DocumentMetadata && { DocumentMetadata: DocumentMetadataFilterSensitiveLog(obj.DocumentMetadata) }),
    ...(obj.FolderMetadata && { FolderMetadata: FolderMetadataFilterSensitiveLog(obj.FolderMetadata) }),
    ...(obj.CommentMetadata && { CommentMetadata: CommentMetadataFilterSensitiveLog(obj.CommentMetadata) }),
    ...(obj.DocumentVersionMetadata && {
        DocumentVersionMetadata: DocumentVersionMetadataFilterSensitiveLog(obj.DocumentVersionMetadata),
    }),
});
export const SearchResourcesResponseFilterSensitiveLog = (obj) => ({
    ...obj,
    ...(obj.Items && { Items: obj.Items.map((item) => ResponseItemFilterSensitiveLog(item)) }),
});
export const UpdateDocumentRequestFilterSensitiveLog = (obj) => ({
    ...obj,
    ...(obj.AuthenticationToken && { AuthenticationToken: SENSITIVE_STRING }),
    ...(obj.Name && { Name: SENSITIVE_STRING }),
});
export const UpdateDocumentVersionRequestFilterSensitiveLog = (obj) => ({
    ...obj,
    ...(obj.AuthenticationToken && { AuthenticationToken: SENSITIVE_STRING }),
});
export const UpdateFolderRequestFilterSensitiveLog = (obj) => ({
    ...obj,
    ...(obj.AuthenticationToken && { AuthenticationToken: SENSITIVE_STRING }),
    ...(obj.Name && { Name: SENSITIVE_STRING }),
});
export const UpdateUserRequestFilterSensitiveLog = (obj) => ({
    ...obj,
    ...(obj.AuthenticationToken && { AuthenticationToken: SENSITIVE_STRING }),
    ...(obj.GivenName && { GivenName: SENSITIVE_STRING }),
    ...(obj.Surname && { Surname: SENSITIVE_STRING }),
});
export const UpdateUserResponseFilterSensitiveLog = (obj) => ({
    ...obj,
    ...(obj.User && { User: UserFilterSensitiveLog(obj.User) }),
});
