// @aws-sdk/client-compute-optimizer/dist-cjs/index.js
"use strict";
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __name = (target, value) => __defProp(target, "name", { value, configurable: true });
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// src/index.ts
var src_exports = {};
__export(src_exports, {
  AccessDeniedException: () => AccessDeniedException,
  AutoScalingConfiguration: () => AutoScalingConfiguration,
  ComputeOptimizer: () => ComputeOptimizer,
  ComputeOptimizerClient: () => ComputeOptimizerClient,
  ComputeOptimizerServiceException: () => ComputeOptimizerServiceException,
  CpuVendorArchitecture: () => CpuVendorArchitecture,
  Currency: () => Currency,
  CurrentPerformanceRisk: () => CurrentPerformanceRisk,
  CustomizableMetricHeadroom: () => CustomizableMetricHeadroom,
  CustomizableMetricName: () => CustomizableMetricName,
  CustomizableMetricThreshold: () => CustomizableMetricThreshold,
  DeleteRecommendationPreferencesCommand: () => DeleteRecommendationPreferencesCommand,
  DescribeRecommendationExportJobsCommand: () => DescribeRecommendationExportJobsCommand,
  EBSFilterName: () => EBSFilterName,
  EBSFinding: () => EBSFinding,
  EBSMetricName: () => EBSMetricName,
  EBSSavingsEstimationModeSource: () => EBSSavingsEstimationModeSource,
  ECSSavingsEstimationModeSource: () => ECSSavingsEstimationModeSource,
  ECSServiceLaunchType: () => ECSServiceLaunchType,
  ECSServiceMetricName: () => ECSServiceMetricName,
  ECSServiceMetricStatistic: () => ECSServiceMetricStatistic,
  ECSServiceRecommendationFilterName: () => ECSServiceRecommendationFilterName,
  ECSServiceRecommendationFinding: () => ECSServiceRecommendationFinding,
  ECSServiceRecommendationFindingReasonCode: () => ECSServiceRecommendationFindingReasonCode,
  EnhancedInfrastructureMetrics: () => EnhancedInfrastructureMetrics,
  EnrollmentFilterName: () => EnrollmentFilterName,
  ExportAutoScalingGroupRecommendationsCommand: () => ExportAutoScalingGroupRecommendationsCommand,
  ExportEBSVolumeRecommendationsCommand: () => ExportEBSVolumeRecommendationsCommand,
  ExportEC2InstanceRecommendationsCommand: () => ExportEC2InstanceRecommendationsCommand,
  ExportECSServiceRecommendationsCommand: () => ExportECSServiceRecommendationsCommand,
  ExportLambdaFunctionRecommendationsCommand: () => ExportLambdaFunctionRecommendationsCommand,
  ExportLicenseRecommendationsCommand: () => ExportLicenseRecommendationsCommand,
  ExportableAutoScalingGroupField: () => ExportableAutoScalingGroupField,
  ExportableECSServiceField: () => ExportableECSServiceField,
  ExportableInstanceField: () => ExportableInstanceField,
  ExportableLambdaFunctionField: () => ExportableLambdaFunctionField,
  ExportableLicenseField: () => ExportableLicenseField,
  ExportableVolumeField: () => ExportableVolumeField,
  ExternalMetricStatusCode: () => ExternalMetricStatusCode,
  ExternalMetricsSource: () => ExternalMetricsSource,
  FileFormat: () => FileFormat,
  FilterName: () => FilterName,
  Finding: () => Finding,
  FindingReasonCode: () => FindingReasonCode,
  GetAutoScalingGroupRecommendationsCommand: () => GetAutoScalingGroupRecommendationsCommand,
  GetEBSVolumeRecommendationsCommand: () => GetEBSVolumeRecommendationsCommand,
  GetEC2InstanceRecommendationsCommand: () => GetEC2InstanceRecommendationsCommand,
  GetEC2RecommendationProjectedMetricsCommand: () => GetEC2RecommendationProjectedMetricsCommand,
  GetECSServiceRecommendationProjectedMetricsCommand: () => GetECSServiceRecommendationProjectedMetricsCommand,
  GetECSServiceRecommendationsCommand: () => GetECSServiceRecommendationsCommand,
  GetEffectiveRecommendationPreferencesCommand: () => GetEffectiveRecommendationPreferencesCommand,
  GetEnrollmentStatusCommand: () => GetEnrollmentStatusCommand,
  GetEnrollmentStatusesForOrganizationCommand: () => GetEnrollmentStatusesForOrganizationCommand,
  GetLambdaFunctionRecommendationsCommand: () => GetLambdaFunctionRecommendationsCommand,
  GetLicenseRecommendationsCommand: () => GetLicenseRecommendationsCommand,
  GetRecommendationPreferencesCommand: () => GetRecommendationPreferencesCommand,
  GetRecommendationSummariesCommand: () => GetRecommendationSummariesCommand,
  InferredWorkloadType: () => InferredWorkloadType,
  InferredWorkloadTypesPreference: () => InferredWorkloadTypesPreference,
  InstanceIdle: () => InstanceIdle,
  InstanceRecommendationFindingReasonCode: () => InstanceRecommendationFindingReasonCode,
  InstanceSavingsEstimationModeSource: () => InstanceSavingsEstimationModeSource,
  InstanceState: () => InstanceState,
  InternalServerException: () => InternalServerException,
  InvalidParameterValueException: () => InvalidParameterValueException,
  JobFilterName: () => JobFilterName,
  JobStatus: () => JobStatus,
  LambdaFunctionMemoryMetricName: () => LambdaFunctionMemoryMetricName,
  LambdaFunctionMemoryMetricStatistic: () => LambdaFunctionMemoryMetricStatistic,
  LambdaFunctionMetricName: () => LambdaFunctionMetricName,
  LambdaFunctionMetricStatistic: () => LambdaFunctionMetricStatistic,
  LambdaFunctionRecommendationFilterName: () => LambdaFunctionRecommendationFilterName,
  LambdaFunctionRecommendationFinding: () => LambdaFunctionRecommendationFinding,
  LambdaFunctionRecommendationFindingReasonCode: () => LambdaFunctionRecommendationFindingReasonCode,
  LambdaSavingsEstimationModeSource: () => LambdaSavingsEstimationModeSource,
  LicenseEdition: () => LicenseEdition,
  LicenseFinding: () => LicenseFinding,
  LicenseFindingReasonCode: () => LicenseFindingReasonCode,
  LicenseModel: () => LicenseModel,
  LicenseName: () => LicenseName,
  LicenseRecommendationFilterName: () => LicenseRecommendationFilterName,
  LimitExceededException: () => LimitExceededException,
  LookBackPeriodPreference: () => LookBackPeriodPreference,
  MetricName: () => MetricName,
  MetricSourceProvider: () => MetricSourceProvider,
  MetricStatistic: () => MetricStatistic,
  MigrationEffort: () => MigrationEffort,
  MissingAuthenticationToken: () => MissingAuthenticationToken,
  OptInRequiredException: () => OptInRequiredException,
  PlatformDifference: () => PlatformDifference,
  PreferredResourceName: () => PreferredResourceName,
  PutRecommendationPreferencesCommand: () => PutRecommendationPreferencesCommand,
  RecommendationPreferenceName: () => RecommendationPreferenceName,
  RecommendationSourceType: () => RecommendationSourceType,
  ResourceNotFoundException: () => ResourceNotFoundException,
  ResourceType: () => ResourceType,
  SavingsEstimationMode: () => SavingsEstimationMode,
  ScopeName: () => ScopeName,
  ServiceUnavailableException: () => ServiceUnavailableException,
  Status: () => Status,
  ThrottlingException: () => ThrottlingException,
  UpdateEnrollmentStatusCommand: () => UpdateEnrollmentStatusCommand,
  __Client: () => import_smithy_client.Client,
  paginateDescribeRecommendationExportJobs: () => paginateDescribeRecommendationExportJobs,
  paginateGetEnrollmentStatusesForOrganization: () => paginateGetEnrollmentStatusesForOrganization,
  paginateGetLambdaFunctionRecommendations: () => paginateGetLambdaFunctionRecommendations,
  paginateGetRecommendationPreferences: () => paginateGetRecommendationPreferences,
  paginateGetRecommendationSummaries: () => paginateGetRecommendationSummaries
});
module.exports = __toCommonJS(src_exports);

// src/ComputeOptimizerClient.ts
var import_middleware_host_header = require("@aws-sdk/middleware-host-header");
var import_middleware_logger = require("@aws-sdk/middleware-logger");
var import_middleware_recursion_detection = require("@aws-sdk/middleware-recursion-detection");
var import_middleware_user_agent = require("@aws-sdk/middleware-user-agent");
var import_config_resolver = require("@smithy/config-resolver");
var import_core = require("@smithy/core");
var import_middleware_content_length = require("@smithy/middleware-content-length");
var import_middleware_endpoint = require("@smithy/middleware-endpoint");
var import_middleware_retry = require("@smithy/middleware-retry");

var import_httpAuthSchemeProvider = require("./auth/httpAuthSchemeProvider");

// src/endpoint/EndpointParameters.ts
var resolveClientEndpointParameters = /* @__PURE__ */ __name((options) => {
  return {
    ...options,
    useDualstackEndpoint: options.useDualstackEndpoint ?? false,
    useFipsEndpoint: options.useFipsEndpoint ?? false,
    defaultSigningName: "compute-optimizer"
  };
}, "resolveClientEndpointParameters");
var commonParams = {
  UseFIPS: { type: "builtInParams", name: "useFipsEndpoint" },
  Endpoint: { type: "builtInParams", name: "endpoint" },
  Region: { type: "builtInParams", name: "region" },
  UseDualStack: { type: "builtInParams", name: "useDualstackEndpoint" }
};

// src/ComputeOptimizerClient.ts
var import_runtimeConfig = require("././runtimeConfig");

// src/runtimeExtensions.ts
var import_region_config_resolver = require("@aws-sdk/region-config-resolver");
var import_protocol_http = require("@smithy/protocol-http");
var import_smithy_client = require("@smithy/smithy-client");

// src/auth/httpAuthExtensionConfiguration.ts
var getHttpAuthExtensionConfiguration = /* @__PURE__ */ __name((runtimeConfig) => {
  const _httpAuthSchemes = runtimeConfig.httpAuthSchemes;
  let _httpAuthSchemeProvider = runtimeConfig.httpAuthSchemeProvider;
  let _credentials = runtimeConfig.credentials;
  return {
    setHttpAuthScheme(httpAuthScheme) {
      const index = _httpAuthSchemes.findIndex((scheme) => scheme.schemeId === httpAuthScheme.schemeId);
      if (index === -1) {
        _httpAuthSchemes.push(httpAuthScheme);
      } else {
        _httpAuthSchemes.splice(index, 1, httpAuthScheme);
      }
    },
    httpAuthSchemes() {
      return _httpAuthSchemes;
    },
    setHttpAuthSchemeProvider(httpAuthSchemeProvider) {
      _httpAuthSchemeProvider = httpAuthSchemeProvider;
    },
    httpAuthSchemeProvider() {
      return _httpAuthSchemeProvider;
    },
    setCredentials(credentials) {
      _credentials = credentials;
    },
    credentials() {
      return _credentials;
    }
  };
}, "getHttpAuthExtensionConfiguration");
var resolveHttpAuthRuntimeConfig = /* @__PURE__ */ __name((config) => {
  return {
    httpAuthSchemes: config.httpAuthSchemes(),
    httpAuthSchemeProvider: config.httpAuthSchemeProvider(),
    credentials: config.credentials()
  };
}, "resolveHttpAuthRuntimeConfig");

// src/runtimeExtensions.ts
var asPartial = /* @__PURE__ */ __name((t) => t, "asPartial");
var resolveRuntimeExtensions = /* @__PURE__ */ __name((runtimeConfig, extensions) => {
  const extensionConfiguration = {
    ...asPartial((0, import_region_config_resolver.getAwsRegionExtensionConfiguration)(runtimeConfig)),
    ...asPartial((0, import_smithy_client.getDefaultExtensionConfiguration)(runtimeConfig)),
    ...asPartial((0, import_protocol_http.getHttpHandlerExtensionConfiguration)(runtimeConfig)),
    ...asPartial(getHttpAuthExtensionConfiguration(runtimeConfig))
  };
  extensions.forEach((extension) => extension.configure(extensionConfiguration));
  return {
    ...runtimeConfig,
    ...(0, import_region_config_resolver.resolveAwsRegionExtensionConfiguration)(extensionConfiguration),
    ...(0, import_smithy_client.resolveDefaultRuntimeConfig)(extensionConfiguration),
    ...(0, import_protocol_http.resolveHttpHandlerRuntimeConfig)(extensionConfiguration),
    ...resolveHttpAuthRuntimeConfig(extensionConfiguration)
  };
}, "resolveRuntimeExtensions");

// src/ComputeOptimizerClient.ts
var _ComputeOptimizerClient = class _ComputeOptimizerClient extends import_smithy_client.Client {
  constructor(...[configuration]) {
    const _config_0 = (0, import_runtimeConfig.getRuntimeConfig)(configuration || {});
    const _config_1 = resolveClientEndpointParameters(_config_0);
    const _config_2 = (0, import_config_resolver.resolveRegionConfig)(_config_1);
    const _config_3 = (0, import_middleware_endpoint.resolveEndpointConfig)(_config_2);
    const _config_4 = (0, import_middleware_retry.resolveRetryConfig)(_config_3);
    const _config_5 = (0, import_middleware_host_header.resolveHostHeaderConfig)(_config_4);
    const _config_6 = (0, import_middleware_user_agent.resolveUserAgentConfig)(_config_5);
    const _config_7 = (0, import_httpAuthSchemeProvider.resolveHttpAuthSchemeConfig)(_config_6);
    const _config_8 = resolveRuntimeExtensions(_config_7, (configuration == null ? void 0 : configuration.extensions) || []);
    super(_config_8);
    this.config = _config_8;
    this.middlewareStack.use((0, import_middleware_retry.getRetryPlugin)(this.config));
    this.middlewareStack.use((0, import_middleware_content_length.getContentLengthPlugin)(this.config));
    this.middlewareStack.use((0, import_middleware_host_header.getHostHeaderPlugin)(this.config));
    this.middlewareStack.use((0, import_middleware_logger.getLoggerPlugin)(this.config));
    this.middlewareStack.use((0, import_middleware_recursion_detection.getRecursionDetectionPlugin)(this.config));
    this.middlewareStack.use((0, import_middleware_user_agent.getUserAgentPlugin)(this.config));
    this.middlewareStack.use(
      (0, import_core.getHttpAuthSchemeEndpointRuleSetPlugin)(this.config, {
        httpAuthSchemeParametersProvider: this.getDefaultHttpAuthSchemeParametersProvider(),
        identityProviderConfigProvider: this.getIdentityProviderConfigProvider()
      })
    );
    this.middlewareStack.use((0, import_core.getHttpSigningPlugin)(this.config));
  }
  /**
   * Destroy underlying resources, like sockets. It's usually not necessary to do this.
   * However in Node.js, it's best to explicitly shut down the client's agent when it is no longer needed.
   * Otherwise, sockets might stay open for quite a long time before the server terminates them.
   */
  destroy() {
    super.destroy();
  }
  getDefaultHttpAuthSchemeParametersProvider() {
    return import_httpAuthSchemeProvider.defaultComputeOptimizerHttpAuthSchemeParametersProvider;
  }
  getIdentityProviderConfigProvider() {
    return async (config) => new import_core.DefaultIdentityProviderConfig({
      "aws.auth#sigv4": config.credentials
    });
  }
};
__name(_ComputeOptimizerClient, "ComputeOptimizerClient");
var ComputeOptimizerClient = _ComputeOptimizerClient;

// src/ComputeOptimizer.ts


// src/commands/DeleteRecommendationPreferencesCommand.ts

var import_middleware_serde = require("@smithy/middleware-serde");

var import_types = require("@smithy/types");

// src/protocols/Aws_json1_0.ts



// src/models/ComputeOptimizerServiceException.ts

var _ComputeOptimizerServiceException = class _ComputeOptimizerServiceException extends import_smithy_client.ServiceException {
  /**
   * @internal
   */
  constructor(options) {
    super(options);
    Object.setPrototypeOf(this, _ComputeOptimizerServiceException.prototype);
  }
};
__name(_ComputeOptimizerServiceException, "ComputeOptimizerServiceException");
var ComputeOptimizerServiceException = _ComputeOptimizerServiceException;

// src/models/models_0.ts
var _AccessDeniedException = class _AccessDeniedException extends ComputeOptimizerServiceException {
  /**
   * @internal
   */
  constructor(opts) {
    super({
      name: "AccessDeniedException",
      $fault: "client",
      ...opts
    });
    this.name = "AccessDeniedException";
    this.$fault = "client";
    Object.setPrototypeOf(this, _AccessDeniedException.prototype);
  }
};
__name(_AccessDeniedException, "AccessDeniedException");
var AccessDeniedException = _AccessDeniedException;
var Status = {
  ACTIVE: "Active",
  FAILED: "Failed",
  INACTIVE: "Inactive",
  PENDING: "Pending"
};
var AutoScalingConfiguration = {
  TARGET_TRACKING_SCALING_CPU: "TargetTrackingScalingCpu",
  TARGET_TRACKING_SCALING_MEMORY: "TargetTrackingScalingMemory"
};
var Currency = {
  CNY: "CNY",
  USD: "USD"
};
var CurrentPerformanceRisk = {
  HIGH: "High",
  LOW: "Low",
  MEDIUM: "Medium",
  VERY_LOW: "VeryLow"
};
var CpuVendorArchitecture = {
  AWS_ARM64: "AWS_ARM64",
  CURRENT: "CURRENT"
};
var EnhancedInfrastructureMetrics = {
  ACTIVE: "Active",
  INACTIVE: "Inactive"
};
var ExternalMetricsSource = {
  DATADOG: "Datadog",
  DYNATRACE: "Dynatrace",
  INSTANA: "Instana",
  NEWRELIC: "NewRelic"
};
var InferredWorkloadTypesPreference = {
  ACTIVE: "Active",
  INACTIVE: "Inactive"
};
var LookBackPeriodPreference = {
  DAYS_14: "DAYS_14",
  DAYS_32: "DAYS_32",
  DAYS_93: "DAYS_93"
};
var PreferredResourceName = {
  EC2_INSTANCE_TYPES: "Ec2InstanceTypes"
};
var InstanceSavingsEstimationModeSource = {
  COST_EXPLORER_RIGHTSIZING: "CostExplorerRightsizing",
  COST_OPTIMIZATION_HUB: "CostOptimizationHub",
  PUBLIC_PRICING: "PublicPricing"
};
var CustomizableMetricName = {
  CPU_UTILIZATION: "CpuUtilization"
};
var CustomizableMetricHeadroom = {
  PERCENT_0: "PERCENT_0",
  PERCENT_20: "PERCENT_20",
  PERCENT_30: "PERCENT_30"
};
var CustomizableMetricThreshold = {
  P90: "P90",
  P95: "P95",
  P99_5: "P99_5"
};
var Finding = {
  NOT_OPTIMIZED: "NotOptimized",
  OPTIMIZED: "Optimized",
  OVER_PROVISIONED: "Overprovisioned",
  UNDER_PROVISIONED: "Underprovisioned"
};
var InferredWorkloadType = {
  AMAZON_EMR: "AmazonEmr",
  APACHE_CASSANDRA: "ApacheCassandra",
  APACHE_HADOOP: "ApacheHadoop",
  KAFKA: "Kafka",
  MEMCACHED: "Memcached",
  NGINX: "Nginx",
  POSTGRE_SQL: "PostgreSql",
  REDIS: "Redis",
  SQLSERVER: "SQLServer"
};
var MigrationEffort = {
  HIGH: "High",
  LOW: "Low",
  MEDIUM: "Medium",
  VERY_LOW: "VeryLow"
};
var MetricName = {
  CPU: "Cpu",
  DISK_READ_BYTES_PER_SECOND: "DISK_READ_BYTES_PER_SECOND",
  DISK_READ_OPS_PER_SECOND: "DISK_READ_OPS_PER_SECOND",
  DISK_WRITE_BYTES_PER_SECOND: "DISK_WRITE_BYTES_PER_SECOND",
  DISK_WRITE_OPS_PER_SECOND: "DISK_WRITE_OPS_PER_SECOND",
  EBS_READ_BYTES_PER_SECOND: "EBS_READ_BYTES_PER_SECOND",
  EBS_READ_OPS_PER_SECOND: "EBS_READ_OPS_PER_SECOND",
  EBS_WRITE_BYTES_PER_SECOND: "EBS_WRITE_BYTES_PER_SECOND",
  EBS_WRITE_OPS_PER_SECOND: "EBS_WRITE_OPS_PER_SECOND",
  GPU_MEMORY_PERCENTAGE: "GPU_MEMORY_PERCENTAGE",
  GPU_PERCENTAGE: "GPU_PERCENTAGE",
  MEMORY: "Memory",
  NETWORK_IN_BYTES_PER_SECOND: "NETWORK_IN_BYTES_PER_SECOND",
  NETWORK_OUT_BYTES_PER_SECOND: "NETWORK_OUT_BYTES_PER_SECOND",
  NETWORK_PACKETS_IN_PER_SECOND: "NETWORK_PACKETS_IN_PER_SECOND",
  NETWORK_PACKETS_OUT_PER_SECOND: "NETWORK_PACKETS_OUT_PER_SECOND"
};
var MetricStatistic = {
  AVERAGE: "Average",
  MAXIMUM: "Maximum"
};
var RecommendationPreferenceName = {
  ENHANCED_INFRASTRUCTURE_METRICS: "EnhancedInfrastructureMetrics",
  EXTERNAL_METRICS_PREFERENCE: "ExternalMetricsPreference",
  INFERRED_WORKLOAD_TYPES: "InferredWorkloadTypes",
  LOOKBACK_PERIOD_PREFERENCE: "LookBackPeriodPreference",
  PREFERRED_RESOURCES: "PreferredResources",
  UTILIZATION_PREFERENCES: "UtilizationPreferences"
};
var ResourceType = {
  AUTO_SCALING_GROUP: "AutoScalingGroup",
  EBS_VOLUME: "EbsVolume",
  EC2_INSTANCE: "Ec2Instance",
  ECS_SERVICE: "EcsService",
  LAMBDA_FUNCTION: "LambdaFunction",
  LICENSE: "License",
  NOT_APPLICABLE: "NotApplicable"
};
var ScopeName = {
  ACCOUNT_ID: "AccountId",
  ORGANIZATION: "Organization",
  RESOURCE_ARN: "ResourceArn"
};
var _InternalServerException = class _InternalServerException extends ComputeOptimizerServiceException {
  /**
   * @internal
   */
  constructor(opts) {
    super({
      name: "InternalServerException",
      $fault: "server",
      ...opts
    });
    this.name = "InternalServerException";
    this.$fault = "server";
    Object.setPrototypeOf(this, _InternalServerException.prototype);
  }
};
__name(_InternalServerException, "InternalServerException");
var InternalServerException = _InternalServerException;
var _InvalidParameterValueException = class _InvalidParameterValueException extends ComputeOptimizerServiceException {
  /**
   * @internal
   */
  constructor(opts) {
    super({
      name: "InvalidParameterValueException",
      $fault: "client",
      ...opts
    });
    this.name = "InvalidParameterValueException";
    this.$fault = "client";
    Object.setPrototypeOf(this, _InvalidParameterValueException.prototype);
  }
};
__name(_InvalidParameterValueException, "InvalidParameterValueException");
var InvalidParameterValueException = _InvalidParameterValueException;
var _MissingAuthenticationToken = class _MissingAuthenticationToken extends ComputeOptimizerServiceException {
  /**
   * @internal
   */
  constructor(opts) {
    super({
      name: "MissingAuthenticationToken",
      $fault: "client",
      ...opts
    });
    this.name = "MissingAuthenticationToken";
    this.$fault = "client";
    Object.setPrototypeOf(this, _MissingAuthenticationToken.prototype);
  }
};
__name(_MissingAuthenticationToken, "MissingAuthenticationToken");
var MissingAuthenticationToken = _MissingAuthenticationToken;
var _OptInRequiredException = class _OptInRequiredException extends ComputeOptimizerServiceException {
  /**
   * @internal
   */
  constructor(opts) {
    super({
      name: "OptInRequiredException",
      $fault: "client",
      ...opts
    });
    this.name = "OptInRequiredException";
    this.$fault = "client";
    Object.setPrototypeOf(this, _OptInRequiredException.prototype);
  }
};
__name(_OptInRequiredException, "OptInRequiredException");
var OptInRequiredException = _OptInRequiredException;
var _ResourceNotFoundException = class _ResourceNotFoundException extends ComputeOptimizerServiceException {
  /**
   * @internal
   */
  constructor(opts) {
    super({
      name: "ResourceNotFoundException",
      $fault: "client",
      ...opts
    });
    this.name = "ResourceNotFoundException";
    this.$fault = "client";
    Object.setPrototypeOf(this, _ResourceNotFoundException.prototype);
  }
};
__name(_ResourceNotFoundException, "ResourceNotFoundException");
var ResourceNotFoundException = _ResourceNotFoundException;
var _ServiceUnavailableException = class _ServiceUnavailableException extends ComputeOptimizerServiceException {
  /**
   * @internal
   */
  constructor(opts) {
    super({
      name: "ServiceUnavailableException",
      $fault: "server",
      ...opts
    });
    this.name = "ServiceUnavailableException";
    this.$fault = "server";
    Object.setPrototypeOf(this, _ServiceUnavailableException.prototype);
  }
};
__name(_ServiceUnavailableException, "ServiceUnavailableException");
var ServiceUnavailableException = _ServiceUnavailableException;
var _ThrottlingException = class _ThrottlingException extends ComputeOptimizerServiceException {
  /**
   * @internal
   */
  constructor(opts) {
    super({
      name: "ThrottlingException",
      $fault: "client",
      ...opts
    });
    this.name = "ThrottlingException";
    this.$fault = "client";
    Object.setPrototypeOf(this, _ThrottlingException.prototype);
  }
};
__name(_ThrottlingException, "ThrottlingException");
var ThrottlingException = _ThrottlingException;
var JobFilterName = {
  JOB_STATUS: "JobStatus",
  RESOURCE_TYPE: "ResourceType"
};
var JobStatus = {
  COMPLETE: "Complete",
  FAILED: "Failed",
  IN_PROGRESS: "InProgress",
  QUEUED: "Queued"
};
var ExportableAutoScalingGroupField = {
  ACCOUNT_ID: "AccountId",
  AUTO_SCALING_GROUP_ARN: "AutoScalingGroupArn",
  AUTO_SCALING_GROUP_NAME: "AutoScalingGroupName",
  CURRENT_CONFIGURATION_DESIRED_CAPACITY: "CurrentConfigurationDesiredCapacity",
  CURRENT_CONFIGURATION_INSTANCE_TYPE: "CurrentConfigurationInstanceType",
  CURRENT_CONFIGURATION_MAX_SIZE: "CurrentConfigurationMaxSize",
  CURRENT_CONFIGURATION_MIN_SIZE: "CurrentConfigurationMinSize",
  CURRENT_INSTANCE_GPU_INFO: "CurrentInstanceGpuInfo",
  CURRENT_MEMORY: "CurrentMemory",
  CURRENT_NETWORK: "CurrentNetwork",
  CURRENT_ON_DEMAND_PRICE: "CurrentOnDemandPrice",
  CURRENT_PERFORMANCE_RISK: "CurrentPerformanceRisk",
  CURRENT_STANDARD_ONE_YEAR_NO_UPFRONT_RESERVED_PRICE: "CurrentStandardOneYearNoUpfrontReservedPrice",
  CURRENT_STANDARD_THREE_YEAR_NO_UPFRONT_RESERVED_PRICE: "CurrentStandardThreeYearNoUpfrontReservedPrice",
  CURRENT_STORAGE: "CurrentStorage",
  CURRENT_VCPUS: "CurrentVCpus",
  EFFECTIVE_RECOMMENDATION_PREFERENCES_CPU_VENDOR_ARCHITECTURES: "EffectiveRecommendationPreferencesCpuVendorArchitectures",
  EFFECTIVE_RECOMMENDATION_PREFERENCES_ENHANCED_INFRASTRUCTURE_METRICS: "EffectiveRecommendationPreferencesEnhancedInfrastructureMetrics",
  EFFECTIVE_RECOMMENDATION_PREFERENCES_INFERRED_WORKLOAD_TYPES: "EffectiveRecommendationPreferencesInferredWorkloadTypes",
  EFFECTIVE_RECOMMENDATION_PREFERENCES_LOOKBACK_PERIOD: "EffectiveRecommendationPreferencesLookBackPeriod",
  EFFECTIVE_RECOMMENDATION_PREFERENCES_PREFERRED_RESOURCES: "EffectiveRecommendationPreferencesPreferredResources",
  EFFECTIVE_RECOMMENDATION_PREFERENCES_SAVINGS_ESTIMATION_MODE: "EffectiveRecommendationPreferencesSavingsEstimationMode",
  FINDING: "Finding",
  INFERRED_WORKLOAD_TYPES: "InferredWorkloadTypes",
  LAST_REFRESH_TIMESTAMP: "LastRefreshTimestamp",
  LOOKBACK_PERIOD_IN_DAYS: "LookbackPeriodInDays",
  RECOMMENDATION_OPTIONS_CONFIGURATION_DESIRED_CAPACITY: "RecommendationOptionsConfigurationDesiredCapacity",
  RECOMMENDATION_OPTIONS_CONFIGURATION_INSTANCE_TYPE: "RecommendationOptionsConfigurationInstanceType",
  RECOMMENDATION_OPTIONS_CONFIGURATION_MAX_SIZE: "RecommendationOptionsConfigurationMaxSize",
  RECOMMENDATION_OPTIONS_CONFIGURATION_MIN_SIZE: "RecommendationOptionsConfigurationMinSize",
  RECOMMENDATION_OPTIONS_ESTIMATED_MONTHLY_SAVINGS_CURRENCY: "RecommendationOptionsEstimatedMonthlySavingsCurrency",
  RECOMMENDATION_OPTIONS_ESTIMATED_MONTHLY_SAVINGS_CURRENCY_AFTER_DISCOUNTS: "RecommendationOptionsEstimatedMonthlySavingsCurrencyAfterDiscounts",
  RECOMMENDATION_OPTIONS_ESTIMATED_MONTHLY_SAVINGS_VALUE: "RecommendationOptionsEstimatedMonthlySavingsValue",
  RECOMMENDATION_OPTIONS_ESTIMATED_MONTHLY_SAVINGS_VALUE_AFTER_DISCOUNTS: "RecommendationOptionsEstimatedMonthlySavingsValueAfterDiscounts",
  RECOMMENDATION_OPTIONS_INSTANCE_GPU_INFO: "RecommendationOptionsInstanceGpuInfo",
  RECOMMENDATION_OPTIONS_MEMORY: "RecommendationOptionsMemory",
  RECOMMENDATION_OPTIONS_MIGRATION_EFFORT: "RecommendationOptionsMigrationEffort",
  RECOMMENDATION_OPTIONS_NETWORK: "RecommendationOptionsNetwork",
  RECOMMENDATION_OPTIONS_ON_DEMAND_PRICE: "RecommendationOptionsOnDemandPrice",
  RECOMMENDATION_OPTIONS_PERFORMANCE_RISK: "RecommendationOptionsPerformanceRisk",
  RECOMMENDATION_OPTIONS_PROJECTED_UTILIZATION_METRICS_CPU_MAXIMUM: "RecommendationOptionsProjectedUtilizationMetricsCpuMaximum",
  RECOMMENDATION_OPTIONS_PROJECTED_UTILIZATION_METRICS_GPU_MAXIMUM: "RecommendationOptionsProjectedUtilizationMetricsGpuPercentageMaximum",
  RECOMMENDATION_OPTIONS_PROJECTED_UTILIZATION_METRICS_GPU_MEMORY_MAXIMUM: "RecommendationOptionsProjectedUtilizationMetricsGpuMemoryPercentageMaximum",
  RECOMMENDATION_OPTIONS_PROJECTED_UTILIZATION_METRICS_MEMORY_MAXIMUM: "RecommendationOptionsProjectedUtilizationMetricsMemoryMaximum",
  RECOMMENDATION_OPTIONS_SAVINGS_OPPORTUNITY_AFTER_DISCOUNTS_PERCENTAGE: "RecommendationOptionsSavingsOpportunityAfterDiscountsPercentage",
  RECOMMENDATION_OPTIONS_SAVINGS_OPPORTUNITY_PERCENTAGE: "RecommendationOptionsSavingsOpportunityPercentage",
  RECOMMENDATION_OPTIONS_STANDARD_ONE_YEAR_NO_UPFRONT_RESERVED_PRICE: "RecommendationOptionsStandardOneYearNoUpfrontReservedPrice",
  RECOMMENDATION_OPTIONS_STANDARD_THREE_YEAR_NO_UPFRONT_RESERVED_PRICE: "RecommendationOptionsStandardThreeYearNoUpfrontReservedPrice",
  RECOMMENDATION_OPTIONS_STORAGE: "RecommendationOptionsStorage",
  RECOMMENDATION_OPTIONS_VCPUS: "RecommendationOptionsVcpus",
  UTILIZATION_METRICS_CPU_MAXIMUM: "UtilizationMetricsCpuMaximum",
  UTILIZATION_METRICS_DISK_READ_BYTES_PER_SECOND_MAXIMUM: "UtilizationMetricsDiskReadBytesPerSecondMaximum",
  UTILIZATION_METRICS_DISK_READ_OPS_PER_SECOND_MAXIMUM: "UtilizationMetricsDiskReadOpsPerSecondMaximum",
  UTILIZATION_METRICS_DISK_WRITE_BYTES_PER_SECOND_MAXIMUM: "UtilizationMetricsDiskWriteBytesPerSecondMaximum",
  UTILIZATION_METRICS_DISK_WRITE_OPS_PER_SECOND_MAXIMUM: "UtilizationMetricsDiskWriteOpsPerSecondMaximum",
  UTILIZATION_METRICS_EBS_READ_BYTES_PER_SECOND_MAXIMUM: "UtilizationMetricsEbsReadBytesPerSecondMaximum",
  UTILIZATION_METRICS_EBS_READ_OPS_PER_SECOND_MAXIMUM: "UtilizationMetricsEbsReadOpsPerSecondMaximum",
  UTILIZATION_METRICS_EBS_WRITE_BYTES_PER_SECOND_MAXIMUM: "UtilizationMetricsEbsWriteBytesPerSecondMaximum",
  UTILIZATION_METRICS_EBS_WRITE_OPS_PER_SECOND_MAXIMUM: "UtilizationMetricsEbsWriteOpsPerSecondMaximum",
  UTILIZATION_METRICS_GPU_MEMORY_PERCENTAGE_MAXIMUM: "UtilizationMetricsGpuMemoryPercentageMaximum",
  UTILIZATION_METRICS_GPU_PERCENTAGE_MAXIMUM: "UtilizationMetricsGpuPercentageMaximum",
  UTILIZATION_METRICS_MEMORY_MAXIMUM: "UtilizationMetricsMemoryMaximum",
  UTILIZATION_METRICS_NETWORK_IN_BYTES_PER_SECOND_MAXIMUM: "UtilizationMetricsNetworkInBytesPerSecondMaximum",
  UTILIZATION_METRICS_NETWORK_OUT_BYTES_PER_SECOND_MAXIMUM: "UtilizationMetricsNetworkOutBytesPerSecondMaximum",
  UTILIZATION_METRICS_NETWORK_PACKETS_IN_PER_SECOND_MAXIMUM: "UtilizationMetricsNetworkPacketsInPerSecondMaximum",
  UTILIZATION_METRICS_NETWORK_PACKETS_OUT_PER_SECOND_MAXIMUM: "UtilizationMetricsNetworkPacketsOutPerSecondMaximum"
};
var FileFormat = {
  CSV: "Csv"
};
var FilterName = {
  FINDING: "Finding",
  FINDING_REASON_CODES: "FindingReasonCodes",
  INFERRED_WORKLOAD_TYPES: "InferredWorkloadTypes",
  RECOMMENDATION_SOURCE_TYPE: "RecommendationSourceType"
};
var _LimitExceededException = class _LimitExceededException extends ComputeOptimizerServiceException {
  /**
   * @internal
   */
  constructor(opts) {
    super({
      name: "LimitExceededException",
      $fault: "client",
      ...opts
    });
    this.name = "LimitExceededException";
    this.$fault = "client";
    Object.setPrototypeOf(this, _LimitExceededException.prototype);
  }
};
__name(_LimitExceededException, "LimitExceededException");
var LimitExceededException = _LimitExceededException;
var ExportableVolumeField = {
  ACCOUNT_ID: "AccountId",
  CURRENT_CONFIGURATION_ROOT_VOLUME: "CurrentConfigurationRootVolume",
  CURRENT_CONFIGURATION_VOLUME_BASELINE_IOPS: "CurrentConfigurationVolumeBaselineIOPS",
  CURRENT_CONFIGURATION_VOLUME_BASELINE_THROUGHPUT: "CurrentConfigurationVolumeBaselineThroughput",
  CURRENT_CONFIGURATION_VOLUME_BURST_IOPS: "CurrentConfigurationVolumeBurstIOPS",
  CURRENT_CONFIGURATION_VOLUME_BURST_THROUGHPUT: "CurrentConfigurationVolumeBurstThroughput",
  CURRENT_CONFIGURATION_VOLUME_SIZE: "CurrentConfigurationVolumeSize",
  CURRENT_CONFIGURATION_VOLUME_TYPE: "CurrentConfigurationVolumeType",
  CURRENT_MONTHLY_PRICE: "CurrentMonthlyPrice",
  CURRENT_PERFORMANCE_RISK: "CurrentPerformanceRisk",
  EFFECTIVE_RECOMMENDATION_PREFERENCES_SAVINGS_ESTIMATION_MODE: "EffectiveRecommendationPreferencesSavingsEstimationMode",
  FINDING: "Finding",
  LAST_REFRESH_TIMESTAMP: "LastRefreshTimestamp",
  LOOKBACK_PERIOD_IN_DAYS: "LookbackPeriodInDays",
  RECOMMENDATION_OPTIONS_CONFIGURATION_VOLUME_BASELINE_IOPS: "RecommendationOptionsConfigurationVolumeBaselineIOPS",
  RECOMMENDATION_OPTIONS_CONFIGURATION_VOLUME_BASELINE_THROUGHPUT: "RecommendationOptionsConfigurationVolumeBaselineThroughput",
  RECOMMENDATION_OPTIONS_CONFIGURATION_VOLUME_BURST_IOPS: "RecommendationOptionsConfigurationVolumeBurstIOPS",
  RECOMMENDATION_OPTIONS_CONFIGURATION_VOLUME_BURST_THROUGHPUT: "RecommendationOptionsConfigurationVolumeBurstThroughput",
  RECOMMENDATION_OPTIONS_CONFIGURATION_VOLUME_SIZE: "RecommendationOptionsConfigurationVolumeSize",
  RECOMMENDATION_OPTIONS_CONFIGURATION_VOLUME_TYPE: "RecommendationOptionsConfigurationVolumeType",
  RECOMMENDATION_OPTIONS_ESTIMATED_MONTHLY_SAVINGS_CURRENCY: "RecommendationOptionsEstimatedMonthlySavingsCurrency",
  RECOMMENDATION_OPTIONS_ESTIMATED_MONTHLY_SAVINGS_CURRENCY_AFTER_DISCOUNTS: "RecommendationOptionsEstimatedMonthlySavingsCurrencyAfterDiscounts",
  RECOMMENDATION_OPTIONS_ESTIMATED_MONTHLY_SAVINGS_VALUE: "RecommendationOptionsEstimatedMonthlySavingsValue",
  RECOMMENDATION_OPTIONS_ESTIMATED_MONTHLY_SAVINGS_VALUE_AFTER_DISCOUNTS: "RecommendationOptionsEstimatedMonthlySavingsValueAfterDiscounts",
  RECOMMENDATION_OPTIONS_MONTHLY_PRICE: "RecommendationOptionsMonthlyPrice",
  RECOMMENDATION_OPTIONS_PERFORMANCE_RISK: "RecommendationOptionsPerformanceRisk",
  RECOMMENDATION_OPTIONS_SAVINGS_OPPORTUNITY_AFTER_DISCOUNTS_PERCENTAGE: "RecommendationOptionsSavingsOpportunityAfterDiscountsPercentage",
  RECOMMENDATION_OPTIONS_SAVINGS_OPPORTUNITY_PERCENTAGE: "RecommendationOptionsSavingsOpportunityPercentage",
  ROOT_VOLUME: "RootVolume",
  TAGS: "Tags",
  UTILIZATION_METRICS_VOLUME_READ_BYTES_PER_SECOND_MAXIMUM: "UtilizationMetricsVolumeReadBytesPerSecondMaximum",
  UTILIZATION_METRICS_VOLUME_READ_OPS_PER_SECOND_MAXIMUM: "UtilizationMetricsVolumeReadOpsPerSecondMaximum",
  UTILIZATION_METRICS_VOLUME_WRITE_BYTES_PER_SECOND_MAXIMUM: "UtilizationMetricsVolumeWriteBytesPerSecondMaximum",
  UTILIZATION_METRICS_VOLUME_WRITE_OPS_PER_SECOND_MAXIMUM: "UtilizationMetricsVolumeWriteOpsPerSecondMaximum",
  VOLUME_ARN: "VolumeArn"
};
var EBSFilterName = {
  FINDING: "Finding"
};
var ExportableInstanceField = {
  ACCOUNT_ID: "AccountId",
  CURRENT_INSTANCE_GPU_INFO: "CurrentInstanceGpuInfo",
  CURRENT_INSTANCE_TYPE: "CurrentInstanceType",
  CURRENT_MEMORY: "CurrentMemory",
  CURRENT_NETWORK: "CurrentNetwork",
  CURRENT_ON_DEMAND_PRICE: "CurrentOnDemandPrice",
  CURRENT_PERFORMANCE_RISK: "CurrentPerformanceRisk",
  CURRENT_STANDARD_ONE_YEAR_NO_UPFRONT_RESERVED_PRICE: "CurrentStandardOneYearNoUpfrontReservedPrice",
  CURRENT_STANDARD_THREE_YEAR_NO_UPFRONT_RESERVED_PRICE: "CurrentStandardThreeYearNoUpfrontReservedPrice",
  CURRENT_STORAGE: "CurrentStorage",
  CURRENT_VCPUS: "CurrentVCpus",
  EFFECTIVE_RECOMMENDATION_PREFERENCES_CPU_VENDOR_ARCHITECTURES: "EffectiveRecommendationPreferencesCpuVendorArchitectures",
  EFFECTIVE_RECOMMENDATION_PREFERENCES_ENHANCED_INFRASTRUCTURE_METRICS: "EffectiveRecommendationPreferencesEnhancedInfrastructureMetrics",
  EFFECTIVE_RECOMMENDATION_PREFERENCES_EXTERNAL_METRICS_SOURCE: "EffectiveRecommendationPreferencesExternalMetricsSource",
  EFFECTIVE_RECOMMENDATION_PREFERENCES_INFERRED_WORKLOAD_TYPES: "EffectiveRecommendationPreferencesInferredWorkloadTypes",
  EFFECTIVE_RECOMMENDATION_PREFERENCES_LOOKBACK_PERIOD: "EffectiveRecommendationPreferencesLookBackPeriod",
  EFFECTIVE_RECOMMENDATION_PREFERENCES_PREFERRED_RESOURCES: "EffectiveRecommendationPreferencesPreferredResources",
  EFFECTIVE_RECOMMENDATION_PREFERENCES_SAVINGS_ESTIMATION_MODE: "EffectiveRecommendationPreferencesSavingsEstimationMode",
  EFFECTIVE_RECOMMENDATION_PREFERENCES_UTILIZATION_PREFERENCES: "EffectiveRecommendationPreferencesUtilizationPreferences",
  EXTERNAL_METRIC_STATUS_CODE: "ExternalMetricStatusCode",
  EXTERNAL_METRIC_STATUS_REASON: "ExternalMetricStatusReason",
  FINDING: "Finding",
  Finding_Reason_Codes: "FindingReasonCodes",
  IDLE: "Idle",
  INFERRED_WORKLOAD_TYPES: "InferredWorkloadTypes",
  INSTANCE_ARN: "InstanceArn",
  INSTANCE_NAME: "InstanceName",
  INSTANCE_STATE: "InstanceState",
  LAST_REFRESH_TIMESTAMP: "LastRefreshTimestamp",
  LOOKBACK_PERIOD_IN_DAYS: "LookbackPeriodInDays",
  RECOMMENDATIONS_SOURCES_RECOMMENDATION_SOURCE_ARN: "RecommendationsSourcesRecommendationSourceArn",
  RECOMMENDATIONS_SOURCES_RECOMMENDATION_SOURCE_TYPE: "RecommendationsSourcesRecommendationSourceType",
  RECOMMENDATION_OPTIONS_ESTIMATED_MONTHLY_SAVINGS_CURRENCY: "RecommendationOptionsEstimatedMonthlySavingsCurrency",
  RECOMMENDATION_OPTIONS_ESTIMATED_MONTHLY_SAVINGS_CURRENCY_AFTER_DISCOUNTS: "RecommendationOptionsEstimatedMonthlySavingsCurrencyAfterDiscounts",
  RECOMMENDATION_OPTIONS_ESTIMATED_MONTHLY_SAVINGS_VALUE: "RecommendationOptionsEstimatedMonthlySavingsValue",
  RECOMMENDATION_OPTIONS_ESTIMATED_MONTHLY_SAVINGS_VALUE_AFTER_DISCOUNTS: "RecommendationOptionsEstimatedMonthlySavingsValueAfterDiscounts",
  RECOMMENDATION_OPTIONS_INSTANCE_GPU_INFO: "RecommendationOptionsInstanceGpuInfo",
  RECOMMENDATION_OPTIONS_INSTANCE_TYPE: "RecommendationOptionsInstanceType",
  RECOMMENDATION_OPTIONS_MEMORY: "RecommendationOptionsMemory",
  RECOMMENDATION_OPTIONS_MIGRATION_EFFORT: "RecommendationOptionsMigrationEffort",
  RECOMMENDATION_OPTIONS_NETWORK: "RecommendationOptionsNetwork",
  RECOMMENDATION_OPTIONS_ON_DEMAND_PRICE: "RecommendationOptionsOnDemandPrice",
  RECOMMENDATION_OPTIONS_PERFORMANCE_RISK: "RecommendationOptionsPerformanceRisk",
  RECOMMENDATION_OPTIONS_PLATFORM_DIFFERENCES: "RecommendationOptionsPlatformDifferences",
  RECOMMENDATION_OPTIONS_PROJECTED_UTILIZATION_METRICS_CPU_MAXIMUM: "RecommendationOptionsProjectedUtilizationMetricsCpuMaximum",
  RECOMMENDATION_OPTIONS_PROJECTED_UTILIZATION_METRICS_GPU_MEMORY_PERCENTAGE_MAXIMUM: "RecommendationOptionsProjectedUtilizationMetricsGpuMemoryPercentageMaximum",
  RECOMMENDATION_OPTIONS_PROJECTED_UTILIZATION_METRICS_GPU_PERCENTAGE_MAXIMUM: "RecommendationOptionsProjectedUtilizationMetricsGpuPercentageMaximum",
  RECOMMENDATION_OPTIONS_PROJECTED_UTILIZATION_METRICS_MEMORY_MAXIMUM: "RecommendationOptionsProjectedUtilizationMetricsMemoryMaximum",
  RECOMMENDATION_OPTIONS_SAVINGS_OPPORTUNITY_AFTER_DISCOUNTS_PERCENTAGE: "RecommendationOptionsSavingsOpportunityAfterDiscountsPercentage",
  RECOMMENDATION_OPTIONS_SAVINGS_OPPORTUNITY_PERCENTAGE: "RecommendationOptionsSavingsOpportunityPercentage",
  RECOMMENDATION_OPTIONS_STANDARD_ONE_YEAR_NO_UPFRONT_RESERVED_PRICE: "RecommendationOptionsStandardOneYearNoUpfrontReservedPrice",
  RECOMMENDATION_OPTIONS_STANDARD_THREE_YEAR_NO_UPFRONT_RESERVED_PRICE: "RecommendationOptionsStandardThreeYearNoUpfrontReservedPrice",
  RECOMMENDATION_OPTIONS_STORAGE: "RecommendationOptionsStorage",
  RECOMMENDATION_OPTIONS_VCPUS: "RecommendationOptionsVcpus",
  TAGS: "Tags",
  UTILIZATION_METRICS_CPU_MAXIMUM: "UtilizationMetricsCpuMaximum",
  UTILIZATION_METRICS_DISK_READ_BYTES_PER_SECOND_MAXIMUM: "UtilizationMetricsDiskReadBytesPerSecondMaximum",
  UTILIZATION_METRICS_DISK_READ_OPS_PER_SECOND_MAXIMUM: "UtilizationMetricsDiskReadOpsPerSecondMaximum",
  UTILIZATION_METRICS_DISK_WRITE_BYTES_PER_SECOND_MAXIMUM: "UtilizationMetricsDiskWriteBytesPerSecondMaximum",
  UTILIZATION_METRICS_DISK_WRITE_OPS_PER_SECOND_MAXIMUM: "UtilizationMetricsDiskWriteOpsPerSecondMaximum",
  UTILIZATION_METRICS_EBS_READ_BYTES_PER_SECOND_MAXIMUM: "UtilizationMetricsEbsReadBytesPerSecondMaximum",
  UTILIZATION_METRICS_EBS_READ_OPS_PER_SECOND_MAXIMUM: "UtilizationMetricsEbsReadOpsPerSecondMaximum",
  UTILIZATION_METRICS_EBS_WRITE_BYTES_PER_SECOND_MAXIMUM: "UtilizationMetricsEbsWriteBytesPerSecondMaximum",
  UTILIZATION_METRICS_EBS_WRITE_OPS_PER_SECOND_MAXIMUM: "UtilizationMetricsEbsWriteOpsPerSecondMaximum",
  UTILIZATION_METRICS_GPU_MEMORY_PERCENTAGE_MAXIMUM: "UtilizationMetricsGpuMemoryPercentageMaximum",
  UTILIZATION_METRICS_GPU_PERCENTAGE_MAXIMUM: "UtilizationMetricsGpuPercentageMaximum",
  UTILIZATION_METRICS_MEMORY_MAXIMUM: "UtilizationMetricsMemoryMaximum",
  UTILIZATION_METRICS_NETWORK_IN_BYTES_PER_SECOND_MAXIMUM: "UtilizationMetricsNetworkInBytesPerSecondMaximum",
  UTILIZATION_METRICS_NETWORK_OUT_BYTES_PER_SECOND_MAXIMUM: "UtilizationMetricsNetworkOutBytesPerSecondMaximum",
  UTILIZATION_METRICS_NETWORK_PACKETS_IN_PER_SECOND_MAXIMUM: "UtilizationMetricsNetworkPacketsInPerSecondMaximum",
  UTILIZATION_METRICS_NETWORK_PACKETS_OUT_PER_SECOND_MAXIMUM: "UtilizationMetricsNetworkPacketsOutPerSecondMaximum"
};
var ExportableECSServiceField = {
  ACCOUNT_ID: "AccountId",
  CURRENT_PERFORMANCE_RISK: "CurrentPerformanceRisk",
  CURRENT_SERVICE_CONFIGURATION_AUTO_SCALING_CONFIGURATION: "CurrentServiceConfigurationAutoScalingConfiguration",
  CURRENT_SERVICE_CONFIGURATION_CPU: "CurrentServiceConfigurationCpu",
  CURRENT_SERVICE_CONFIGURATION_MEMORY: "CurrentServiceConfigurationMemory",
  CURRENT_SERVICE_CONFIGURATION_TASK_DEFINITION_ARN: "CurrentServiceConfigurationTaskDefinitionArn",
  CURRENT_SERVICE_CONTAINER_CONFIGURATIONS: "CurrentServiceContainerConfigurations",
  EFFECTIVE_RECOMMENDATION_PREFERENCES_SAVINGS_ESTIMATION_MODE: "EffectiveRecommendationPreferencesSavingsEstimationMode",
  FINDING: "Finding",
  FINDING_REASON_CODES: "FindingReasonCodes",
  LAST_REFRESH_TIMESTAMP: "LastRefreshTimestamp",
  LAUNCH_TYPE: "LaunchType",
  LOOKBACK_PERIOD_IN_DAYS: "LookbackPeriodInDays",
  RECOMMENDATION_OPTIONS_CONTAINER_RECOMMENDATIONS: "RecommendationOptionsContainerRecommendations",
  RECOMMENDATION_OPTIONS_CPU: "RecommendationOptionsCpu",
  RECOMMENDATION_OPTIONS_ESTIMATED_MONTHLY_SAVINGS_CURRENCY: "RecommendationOptionsEstimatedMonthlySavingsCurrency",
  RECOMMENDATION_OPTIONS_ESTIMATED_MONTHLY_SAVINGS_CURRENCY_AFTER_DISCOUNTS: "RecommendationOptionsEstimatedMonthlySavingsCurrencyAfterDiscounts",
  RECOMMENDATION_OPTIONS_ESTIMATED_MONTHLY_SAVINGS_VALUE: "RecommendationOptionsEstimatedMonthlySavingsValue",
  RECOMMENDATION_OPTIONS_ESTIMATED_MONTHLY_SAVINGS_VALUE_AFTER_DISCOUNTS: "RecommendationOptionsEstimatedMonthlySavingsValueAfterDiscounts",
  RECOMMENDATION_OPTIONS_MEMORY: "RecommendationOptionsMemory",
  RECOMMENDATION_OPTIONS_PROJECTED_UTILIZATION_METRICS_CPU_MAXIMUM: "RecommendationOptionsProjectedUtilizationMetricsCpuMaximum",
  RECOMMENDATION_OPTIONS_PROJECTED_UTILIZATION_METRICS_MEMORY_MAXIMUM: "RecommendationOptionsProjectedUtilizationMetricsMemoryMaximum",
  RECOMMENDATION_OPTIONS_SAVINGS_OPPORTUNITY_AFTER_DISCOUNTS_PERCENTAGE: "RecommendationOptionsSavingsOpportunityAfterDiscountsPercentage",
  RECOMMENDATION_OPTIONS_SAVINGS_OPPORTUNITY_PERCENTAGE: "RecommendationOptionsSavingsOpportunityPercentage",
  SERVICE_ARN: "ServiceArn",
  TAGS: "Tags",
  UTILIZATION_METRICS_CPU_MAXIMUM: "UtilizationMetricsCpuMaximum",
  UTILIZATION_METRICS_MEMORY_MAXIMUM: "UtilizationMetricsMemoryMaximum"
};
var ECSServiceRecommendationFilterName = {
  FINDING: "Finding",
  FINDING_REASON_CODE: "FindingReasonCode"
};
var ExportableLambdaFunctionField = {
  ACCOUNT_ID: "AccountId",
  CURRENT_CONFIGURATION_MEMORY_SIZE: "CurrentConfigurationMemorySize",
  CURRENT_CONFIGURATION_TIMEOUT: "CurrentConfigurationTimeout",
  CURRENT_COST_AVERAGE: "CurrentCostAverage",
  CURRENT_COST_TOTAL: "CurrentCostTotal",
  CURRENT_PERFORMANCE_RISK: "CurrentPerformanceRisk",
  EFFECTIVE_RECOMMENDATION_PREFERENCES_SAVINGS_ESTIMATION_MODE: "EffectiveRecommendationPreferencesSavingsEstimationMode",
  FINDING: "Finding",
  FINDING_REASON_CODES: "FindingReasonCodes",
  FUNCTION_ARN: "FunctionArn",
  FUNCTION_VERSION: "FunctionVersion",
  LAST_REFRESH_TIMESTAMP: "LastRefreshTimestamp",
  LOOKBACK_PERIOD_IN_DAYS: "LookbackPeriodInDays",
  NUMBER_OF_INVOCATIONS: "NumberOfInvocations",
  RECOMMENDATION_OPTIONS_CONFIGURATION_MEMORY_SIZE: "RecommendationOptionsConfigurationMemorySize",
  RECOMMENDATION_OPTIONS_COST_HIGH: "RecommendationOptionsCostHigh",
  RECOMMENDATION_OPTIONS_COST_LOW: "RecommendationOptionsCostLow",
  RECOMMENDATION_OPTIONS_ESTIMATED_MONTHLY_SAVINGS_CURRENCY: "RecommendationOptionsEstimatedMonthlySavingsCurrency",
  RECOMMENDATION_OPTIONS_ESTIMATED_MONTHLY_SAVINGS_CURRENCY_AFTER_DISCOUNTS: "RecommendationOptionsEstimatedMonthlySavingsCurrencyAfterDiscounts",
  RECOMMENDATION_OPTIONS_ESTIMATED_MONTHLY_SAVINGS_VALUE: "RecommendationOptionsEstimatedMonthlySavingsValue",
  RECOMMENDATION_OPTIONS_ESTIMATED_MONTHLY_SAVINGS_VALUE_AFTER_DISCOUNTS: "RecommendationOptionsEstimatedMonthlySavingsValueAfterDiscounts",
  RECOMMENDATION_OPTIONS_PROJECTED_UTILIZATION_METRICS_DURATION_EXPECTED: "RecommendationOptionsProjectedUtilizationMetricsDurationExpected",
  RECOMMENDATION_OPTIONS_PROJECTED_UTILIZATION_METRICS_DURATION_LOWER_BOUND: "RecommendationOptionsProjectedUtilizationMetricsDurationLowerBound",
  RECOMMENDATION_OPTIONS_PROJECTED_UTILIZATION_METRICS_DURATION_UPPER_BOUND: "RecommendationOptionsProjectedUtilizationMetricsDurationUpperBound",
  RECOMMENDATION_OPTIONS_SAVINGS_OPPORTUNITY_AFTER_DISCOUNTS_PERCENTAGE: "RecommendationOptionsSavingsOpportunityAfterDiscountsPercentage",
  RECOMMENDATION_OPTIONS_SAVINGS_OPPORTUNITY_PERCENTAGE: "RecommendationOptionsSavingsOpportunityPercentage",
  TAGS: "Tags",
  UTILIZATION_METRICS_DURATION_AVERAGE: "UtilizationMetricsDurationAverage",
  UTILIZATION_METRICS_DURATION_MAXIMUM: "UtilizationMetricsDurationMaximum",
  UTILIZATION_METRICS_MEMORY_AVERAGE: "UtilizationMetricsMemoryAverage",
  UTILIZATION_METRICS_MEMORY_MAXIMUM: "UtilizationMetricsMemoryMaximum"
};
var LambdaFunctionRecommendationFilterName = {
  FINDING: "Finding",
  FINDING_REASON_CODE: "FindingReasonCode"
};
var ExportableLicenseField = {
  ACCOUNT_ID: "AccountId",
  CURRENT_LICENSE_CONFIGURATION_INSTANCE_TYPE: "CurrentLicenseConfigurationInstanceType",
  CURRENT_LICENSE_CONFIGURATION_LICENSE_EDITION: "CurrentLicenseConfigurationLicenseEdition",
  CURRENT_LICENSE_CONFIGURATION_LICENSE_MODEL: "CurrentLicenseConfigurationLicenseModel",
  CURRENT_LICENSE_CONFIGURATION_LICENSE_NAME: "CurrentLicenseConfigurationLicenseName",
  CURRENT_LICENSE_CONFIGURATION_LICENSE_VERSION: "CurrentLicenseConfigurationLicenseVersion",
  CURRENT_LICENSE_CONFIGURATION_METRICS_SOURCE: "CurrentLicenseConfigurationMetricsSource",
  CURRENT_LICENSE_CONFIGURATION_NUMBER_OF_CORES: "CurrentLicenseConfigurationNumberOfCores",
  CURRENT_LICENSE_CONFIGURATION_OPERATING_SYSTEM: "CurrentLicenseConfigurationOperatingSystem",
  LAST_REFRESH_TIMESTAMP: "LastRefreshTimestamp",
  LICENSE_FINDING: "Finding",
  LICENSE_FINDING_REASON_CODES: "FindingReasonCodes",
  LOOKBACK_PERIOD_IN_DAYS: "LookbackPeriodInDays",
  RECOMMENDATION_OPTIONS_ESTIMATED_MONTHLY_SAVINGS_CURRENCY: "RecommendationOptionsEstimatedMonthlySavingsCurrency",
  RECOMMENDATION_OPTIONS_ESTIMATED_MONTHLY_SAVINGS_VALUE: "RecommendationOptionsEstimatedMonthlySavingsValue",
  RECOMMENDATION_OPTIONS_LICENSE_EDITION: "RecommendationOptionsLicenseEdition",
  RECOMMENDATION_OPTIONS_LICENSE_MODEL: "RecommendationOptionsLicenseModel",
  RECOMMENDATION_OPTIONS_OPERATING_SYSTEM: "RecommendationOptionsOperatingSystem",
  RECOMMENDATION_OPTIONS_SAVINGS_OPPORTUNITY_PERCENTAGE: "RecommendationOptionsSavingsOpportunityPercentage",
  RESOURCE_ARN: "ResourceArn",
  TAGS: "Tags"
};
var LicenseRecommendationFilterName = {
  LICENSE_FINDING: "Finding",
  LICENSE_FINDING_REASON_CODE: "FindingReasonCode",
  LICENSE_NAME: "LicenseName"
};
var EBSSavingsEstimationModeSource = {
  COST_EXPLORER_RIGHTSIZING: "CostExplorerRightsizing",
  COST_OPTIMIZATION_HUB: "CostOptimizationHub",
  PUBLIC_PRICING: "PublicPricing"
};
var EBSFinding = {
  NOT_OPTIMIZED: "NotOptimized",
  OPTIMIZED: "Optimized"
};
var EBSMetricName = {
  VOLUME_READ_BYTES_PER_SECOND: "VolumeReadBytesPerSecond",
  VOLUME_READ_OPS_PER_SECOND: "VolumeReadOpsPerSecond",
  VOLUME_WRITE_BYTES_PER_SECOND: "VolumeWriteBytesPerSecond",
  VOLUME_WRITE_OPS_PER_SECOND: "VolumeWriteOpsPerSecond"
};
var ExternalMetricStatusCode = {
  DATADOG_INTEGRATION_ERROR: "DATADOG_INTEGRATION_ERROR",
  DYNATRACE_INTEGRATION_ERROR: "DYNATRACE_INTEGRATION_ERROR",
  INSTANA_INTEGRATION_ERROR: "INSTANA_INTEGRATION_ERROR",
  INSUFFICIENT_DATADOG_METRICS: "INSUFFICIENT_DATADOG_METRICS",
  INSUFFICIENT_DYNATRACE_METRICS: "INSUFFICIENT_DYNATRACE_METRICS",
  INSUFFICIENT_INSTANA_METRICS: "INSUFFICIENT_INSTANA_METRICS",
  INSUFFICIENT_NEWRELIC_METRICS: "INSUFFICIENT_NEWRELIC_METRICS",
  INTEGRATION_SUCCESS: "INTEGRATION_SUCCESS",
  NEWRELIC_INTEGRATION_ERROR: "NEWRELIC_INTEGRATION_ERROR",
  NO_EXTERNAL_METRIC_SET: "NO_EXTERNAL_METRIC_SET"
};
var InstanceRecommendationFindingReasonCode = {
  CPU_OVER_PROVISIONED: "CPUOverprovisioned",
  CPU_UNDER_PROVISIONED: "CPUUnderprovisioned",
  DISK_IOPS_OVER_PROVISIONED: "DiskIOPSOverprovisioned",
  DISK_IOPS_UNDER_PROVISIONED: "DiskIOPSUnderprovisioned",
  DISK_THROUGHPUT_OVER_PROVISIONED: "DiskThroughputOverprovisioned",
  DISK_THROUGHPUT_UNDER_PROVISIONED: "DiskThroughputUnderprovisioned",
  EBS_IOPS_OVER_PROVISIONED: "EBSIOPSOverprovisioned",
  EBS_IOPS_UNDER_PROVISIONED: "EBSIOPSUnderprovisioned",
  EBS_THROUGHPUT_OVER_PROVISIONED: "EBSThroughputOverprovisioned",
  EBS_THROUGHPUT_UNDER_PROVISIONED: "EBSThroughputUnderprovisioned",
  GPU_MEMORY_OVER_PROVISIONED: "GPUMemoryOverprovisioned",
  GPU_MEMORY_UNDER_PROVISIONED: "GPUMemoryUnderprovisioned",
  GPU_OVER_PROVISIONED: "GPUOverprovisioned",
  GPU_UNDER_PROVISIONED: "GPUUnderprovisioned",
  MEMORY_OVER_PROVISIONED: "MemoryOverprovisioned",
  MEMORY_UNDER_PROVISIONED: "MemoryUnderprovisioned",
  NETWORK_BANDWIDTH_OVER_PROVISIONED: "NetworkBandwidthOverprovisioned",
  NETWORK_BANDWIDTH_UNDER_PROVISIONED: "NetworkBandwidthUnderprovisioned",
  NETWORK_PPS_OVER_PROVISIONED: "NetworkPPSOverprovisioned",
  NETWORK_PPS_UNDER_PROVISIONED: "NetworkPPSUnderprovisioned"
};
var InstanceIdle = {
  FALSE: "False",
  TRUE: "True"
};
var InstanceState = {
  PENDING: "pending",
  RUNNING: "running",
  SHUTTING_DOWN: "shutting-down",
  STOPPED: "stopped",
  STOPPING: "stopping",
  TERMINATED: "terminated"
};
var PlatformDifference = {
  ARCHITECTURE: "Architecture",
  HYPERVISOR: "Hypervisor",
  INSTANCE_STORE_AVAILABILITY: "InstanceStoreAvailability",
  NETWORK_INTERFACE: "NetworkInterface",
  STORAGE_INTERFACE: "StorageInterface",
  VIRTUALIZATION_TYPE: "VirtualizationType"
};
var RecommendationSourceType = {
  AUTO_SCALING_GROUP: "AutoScalingGroup",
  EBS_VOLUME: "EbsVolume",
  EC2_INSTANCE: "Ec2Instance",
  ECS_SERVICE: "EcsService",
  LAMBDA_FUNCTION: "LambdaFunction",
  LICENSE: "License"
};
var ECSServiceMetricName = {
  CPU: "Cpu",
  MEMORY: "Memory"
};
var ECSSavingsEstimationModeSource = {
  COST_EXPLORER_RIGHTSIZING: "CostExplorerRightsizing",
  COST_OPTIMIZATION_HUB: "CostOptimizationHub",
  PUBLIC_PRICING: "PublicPricing"
};
var ECSServiceRecommendationFinding = {
  OPTIMIZED: "Optimized",
  OVER_PROVISIONED: "Overprovisioned",
  UNDER_PROVISIONED: "Underprovisioned"
};
var ECSServiceRecommendationFindingReasonCode = {
  CPU_OVER_PROVISIONED: "CPUOverprovisioned",
  CPU_UNDER_PROVISIONED: "CPUUnderprovisioned",
  MEMORY_OVER_PROVISIONED: "MemoryOverprovisioned",
  MEMORY_UNDER_PROVISIONED: "MemoryUnderprovisioned"
};
var ECSServiceLaunchType = {
  EC2: "EC2",
  FARGATE: "Fargate"
};
var ECSServiceMetricStatistic = {
  AVERAGE: "Average",
  MAXIMUM: "Maximum"
};
var EnrollmentFilterName = {
  STATUS: "Status"
};
var LambdaSavingsEstimationModeSource = {
  COST_EXPLORER_RIGHTSIZING: "CostExplorerRightsizing",
  COST_OPTIMIZATION_HUB: "CostOptimizationHub",
  PUBLIC_PRICING: "PublicPricing"
};
var LambdaFunctionRecommendationFinding = {
  NOT_OPTIMIZED: "NotOptimized",
  OPTIMIZED: "Optimized",
  UNAVAILABLE: "Unavailable"
};
var LambdaFunctionRecommendationFindingReasonCode = {
  INCONCLUSIVE: "Inconclusive",
  INSUFFICIENT_DATA: "InsufficientData",
  MEMORY_OVER_PROVISIONED: "MemoryOverprovisioned",
  MEMORY_UNDER_PROVISIONED: "MemoryUnderprovisioned"
};
var LambdaFunctionMemoryMetricName = {
  DURATION: "Duration"
};
var LambdaFunctionMemoryMetricStatistic = {
  EXPECTED: "Expected",
  LOWER_BOUND: "LowerBound",
  UPPER_BOUND: "UpperBound"
};
var LambdaFunctionMetricName = {
  DURATION: "Duration",
  MEMORY: "Memory"
};
var LambdaFunctionMetricStatistic = {
  AVERAGE: "Average",
  MAXIMUM: "Maximum"
};
var LicenseEdition = {
  ENTERPRISE: "Enterprise",
  FREE: "Free",
  NO_LICENSE_EDITION_FOUND: "NoLicenseEditionFound",
  STANDARD: "Standard"
};
var LicenseModel = {
  BRING_YOUR_OWN_LICENSE: "BringYourOwnLicense",
  LICENSE_INCLUDED: "LicenseIncluded"
};
var LicenseName = {
  SQLSERVER: "SQLServer"
};
var MetricSourceProvider = {
  CloudWatchAppInsights: "CloudWatchApplicationInsights"
};
var LicenseFinding = {
  INSUFFICIENT_METRICS: "InsufficientMetrics",
  NOT_OPTIMIZED: "NotOptimized",
  OPTIMIZED: "Optimized"
};
var LicenseFindingReasonCode = {
  CW_APP_INSIGHTS_DISABLED: "InvalidCloudWatchApplicationInsightsSetup",
  CW_APP_INSIGHTS_ERROR: "CloudWatchApplicationInsightsError",
  LICENSE_OVER_PROVISIONED: "LicenseOverprovisioned",
  OPTIMIZED: "Optimized"
};
var SavingsEstimationMode = {
  AFTER_DISCOUNTS: "AfterDiscounts",
  BEFORE_DISCOUNTS: "BeforeDiscounts"
};
var FindingReasonCode = {
  MEMORY_OVER_PROVISIONED: "MemoryOverprovisioned",
  MEMORY_UNDER_PROVISIONED: "MemoryUnderprovisioned"
};

// src/protocols/Aws_json1_0.ts
var se_DeleteRecommendationPreferencesCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("DeleteRecommendationPreferences");
  let body;
  body = JSON.stringify((0, import_smithy_client._json)(input));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_DeleteRecommendationPreferencesCommand");
var se_DescribeRecommendationExportJobsCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("DescribeRecommendationExportJobs");
  let body;
  body = JSON.stringify((0, import_smithy_client._json)(input));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_DescribeRecommendationExportJobsCommand");
var se_ExportAutoScalingGroupRecommendationsCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("ExportAutoScalingGroupRecommendations");
  let body;
  body = JSON.stringify((0, import_smithy_client._json)(input));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_ExportAutoScalingGroupRecommendationsCommand");
var se_ExportEBSVolumeRecommendationsCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("ExportEBSVolumeRecommendations");
  let body;
  body = JSON.stringify((0, import_smithy_client._json)(input));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_ExportEBSVolumeRecommendationsCommand");
var se_ExportEC2InstanceRecommendationsCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("ExportEC2InstanceRecommendations");
  let body;
  body = JSON.stringify((0, import_smithy_client._json)(input));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_ExportEC2InstanceRecommendationsCommand");
var se_ExportECSServiceRecommendationsCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("ExportECSServiceRecommendations");
  let body;
  body = JSON.stringify((0, import_smithy_client._json)(input));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_ExportECSServiceRecommendationsCommand");
var se_ExportLambdaFunctionRecommendationsCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("ExportLambdaFunctionRecommendations");
  let body;
  body = JSON.stringify((0, import_smithy_client._json)(input));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_ExportLambdaFunctionRecommendationsCommand");
var se_ExportLicenseRecommendationsCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("ExportLicenseRecommendations");
  let body;
  body = JSON.stringify((0, import_smithy_client._json)(input));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_ExportLicenseRecommendationsCommand");
var se_GetAutoScalingGroupRecommendationsCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("GetAutoScalingGroupRecommendations");
  let body;
  body = JSON.stringify((0, import_smithy_client._json)(input));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_GetAutoScalingGroupRecommendationsCommand");
var se_GetEBSVolumeRecommendationsCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("GetEBSVolumeRecommendations");
  let body;
  body = JSON.stringify((0, import_smithy_client._json)(input));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_GetEBSVolumeRecommendationsCommand");
var se_GetEC2InstanceRecommendationsCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("GetEC2InstanceRecommendations");
  let body;
  body = JSON.stringify((0, import_smithy_client._json)(input));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_GetEC2InstanceRecommendationsCommand");
var se_GetEC2RecommendationProjectedMetricsCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("GetEC2RecommendationProjectedMetrics");
  let body;
  body = JSON.stringify(se_GetEC2RecommendationProjectedMetricsRequest(input, context));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_GetEC2RecommendationProjectedMetricsCommand");
var se_GetECSServiceRecommendationProjectedMetricsCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("GetECSServiceRecommendationProjectedMetrics");
  let body;
  body = JSON.stringify(se_GetECSServiceRecommendationProjectedMetricsRequest(input, context));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_GetECSServiceRecommendationProjectedMetricsCommand");
var se_GetECSServiceRecommendationsCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("GetECSServiceRecommendations");
  let body;
  body = JSON.stringify((0, import_smithy_client._json)(input));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_GetECSServiceRecommendationsCommand");
var se_GetEffectiveRecommendationPreferencesCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("GetEffectiveRecommendationPreferences");
  let body;
  body = JSON.stringify((0, import_smithy_client._json)(input));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_GetEffectiveRecommendationPreferencesCommand");
var se_GetEnrollmentStatusCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("GetEnrollmentStatus");
  let body;
  body = JSON.stringify((0, import_smithy_client._json)(input));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_GetEnrollmentStatusCommand");
var se_GetEnrollmentStatusesForOrganizationCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("GetEnrollmentStatusesForOrganization");
  let body;
  body = JSON.stringify((0, import_smithy_client._json)(input));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_GetEnrollmentStatusesForOrganizationCommand");
var se_GetLambdaFunctionRecommendationsCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("GetLambdaFunctionRecommendations");
  let body;
  body = JSON.stringify((0, import_smithy_client._json)(input));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_GetLambdaFunctionRecommendationsCommand");
var se_GetLicenseRecommendationsCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("GetLicenseRecommendations");
  let body;
  body = JSON.stringify((0, import_smithy_client._json)(input));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_GetLicenseRecommendationsCommand");
var se_GetRecommendationPreferencesCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("GetRecommendationPreferences");
  let body;
  body = JSON.stringify((0, import_smithy_client._json)(input));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_GetRecommendationPreferencesCommand");
var se_GetRecommendationSummariesCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("GetRecommendationSummaries");
  let body;
  body = JSON.stringify((0, import_smithy_client._json)(input));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_GetRecommendationSummariesCommand");
var se_PutRecommendationPreferencesCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("PutRecommendationPreferences");
  let body;
  body = JSON.stringify((0, import_smithy_client._json)(input));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_PutRecommendationPreferencesCommand");
var se_UpdateEnrollmentStatusCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("UpdateEnrollmentStatus");
  let body;
  body = JSON.stringify((0, import_smithy_client._json)(input));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_UpdateEnrollmentStatusCommand");
var de_DeleteRecommendationPreferencesCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const data = await parseBody(output.body, context);
  let contents = {};
  contents = (0, import_smithy_client._json)(data);
  const response = {
    $metadata: deserializeMetadata(output),
    ...contents
  };
  return response;
}, "de_DeleteRecommendationPreferencesCommand");
var de_DescribeRecommendationExportJobsCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const data = await parseBody(output.body, context);
  let contents = {};
  contents = de_DescribeRecommendationExportJobsResponse(data, context);
  const response = {
    $metadata: deserializeMetadata(output),
    ...contents
  };
  return response;
}, "de_DescribeRecommendationExportJobsCommand");
var de_ExportAutoScalingGroupRecommendationsCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const data = await parseBody(output.body, context);
  let contents = {};
  contents = (0, import_smithy_client._json)(data);
  const response = {
    $metadata: deserializeMetadata(output),
    ...contents
  };
  return response;
}, "de_ExportAutoScalingGroupRecommendationsCommand");
var de_ExportEBSVolumeRecommendationsCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const data = await parseBody(output.body, context);
  let contents = {};
  contents = (0, import_smithy_client._json)(data);
  const response = {
    $metadata: deserializeMetadata(output),
    ...contents
  };
  return response;
}, "de_ExportEBSVolumeRecommendationsCommand");
var de_ExportEC2InstanceRecommendationsCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const data = await parseBody(output.body, context);
  let contents = {};
  contents = (0, import_smithy_client._json)(data);
  const response = {
    $metadata: deserializeMetadata(output),
    ...contents
  };
  return response;
}, "de_ExportEC2InstanceRecommendationsCommand");
var de_ExportECSServiceRecommendationsCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const data = await parseBody(output.body, context);
  let contents = {};
  contents = (0, import_smithy_client._json)(data);
  const response = {
    $metadata: deserializeMetadata(output),
    ...contents
  };
  return response;
}, "de_ExportECSServiceRecommendationsCommand");
var de_ExportLambdaFunctionRecommendationsCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const data = await parseBody(output.body, context);
  let contents = {};
  contents = (0, import_smithy_client._json)(data);
  const response = {
    $metadata: deserializeMetadata(output),
    ...contents
  };
  return response;
}, "de_ExportLambdaFunctionRecommendationsCommand");
var de_ExportLicenseRecommendationsCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const data = await parseBody(output.body, context);
  let contents = {};
  contents = (0, import_smithy_client._json)(data);
  const response = {
    $metadata: deserializeMetadata(output),
    ...contents
  };
  return response;
}, "de_ExportLicenseRecommendationsCommand");
var de_GetAutoScalingGroupRecommendationsCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const data = await parseBody(output.body, context);
  let contents = {};
  contents = de_GetAutoScalingGroupRecommendationsResponse(data, context);
  const response = {
    $metadata: deserializeMetadata(output),
    ...contents
  };
  return response;
}, "de_GetAutoScalingGroupRecommendationsCommand");
var de_GetEBSVolumeRecommendationsCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const data = await parseBody(output.body, context);
  let contents = {};
  contents = de_GetEBSVolumeRecommendationsResponse(data, context);
  const response = {
    $metadata: deserializeMetadata(output),
    ...contents
  };
  return response;
}, "de_GetEBSVolumeRecommendationsCommand");
var de_GetEC2InstanceRecommendationsCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const data = await parseBody(output.body, context);
  let contents = {};
  contents = de_GetEC2InstanceRecommendationsResponse(data, context);
  const response = {
    $metadata: deserializeMetadata(output),
    ...contents
  };
  return response;
}, "de_GetEC2InstanceRecommendationsCommand");
var de_GetEC2RecommendationProjectedMetricsCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const data = await parseBody(output.body, context);
  let contents = {};
  contents = de_GetEC2RecommendationProjectedMetricsResponse(data, context);
  const response = {
    $metadata: deserializeMetadata(output),
    ...contents
  };
  return response;
}, "de_GetEC2RecommendationProjectedMetricsCommand");
var de_GetECSServiceRecommendationProjectedMetricsCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const data = await parseBody(output.body, context);
  let contents = {};
  contents = de_GetECSServiceRecommendationProjectedMetricsResponse(data, context);
  const response = {
    $metadata: deserializeMetadata(output),
    ...contents
  };
  return response;
}, "de_GetECSServiceRecommendationProjectedMetricsCommand");
var de_GetECSServiceRecommendationsCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const data = await parseBody(output.body, context);
  let contents = {};
  contents = de_GetECSServiceRecommendationsResponse(data, context);
  const response = {
    $metadata: deserializeMetadata(output),
    ...contents
  };
  return response;
}, "de_GetECSServiceRecommendationsCommand");
var de_GetEffectiveRecommendationPreferencesCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const data = await parseBody(output.body, context);
  let contents = {};
  contents = (0, import_smithy_client._json)(data);
  const response = {
    $metadata: deserializeMetadata(output),
    ...contents
  };
  return response;
}, "de_GetEffectiveRecommendationPreferencesCommand");
var de_GetEnrollmentStatusCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const data = await parseBody(output.body, context);
  let contents = {};
  contents = de_GetEnrollmentStatusResponse(data, context);
  const response = {
    $metadata: deserializeMetadata(output),
    ...contents
  };
  return response;
}, "de_GetEnrollmentStatusCommand");
var de_GetEnrollmentStatusesForOrganizationCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const data = await parseBody(output.body, context);
  let contents = {};
  contents = de_GetEnrollmentStatusesForOrganizationResponse(data, context);
  const response = {
    $metadata: deserializeMetadata(output),
    ...contents
  };
  return response;
}, "de_GetEnrollmentStatusesForOrganizationCommand");
var de_GetLambdaFunctionRecommendationsCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const data = await parseBody(output.body, context);
  let contents = {};
  contents = de_GetLambdaFunctionRecommendationsResponse(data, context);
  const response = {
    $metadata: deserializeMetadata(output),
    ...contents
  };
  return response;
}, "de_GetLambdaFunctionRecommendationsCommand");
var de_GetLicenseRecommendationsCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const data = await parseBody(output.body, context);
  let contents = {};
  contents = de_GetLicenseRecommendationsResponse(data, context);
  const response = {
    $metadata: deserializeMetadata(output),
    ...contents
  };
  return response;
}, "de_GetLicenseRecommendationsCommand");
var de_GetRecommendationPreferencesCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const data = await parseBody(output.body, context);
  let contents = {};
  contents = (0, import_smithy_client._json)(data);
  const response = {
    $metadata: deserializeMetadata(output),
    ...contents
  };
  return response;
}, "de_GetRecommendationPreferencesCommand");
var de_GetRecommendationSummariesCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const data = await parseBody(output.body, context);
  let contents = {};
  contents = de_GetRecommendationSummariesResponse(data, context);
  const response = {
    $metadata: deserializeMetadata(output),
    ...contents
  };
  return response;
}, "de_GetRecommendationSummariesCommand");
var de_PutRecommendationPreferencesCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const data = await parseBody(output.body, context);
  let contents = {};
  contents = (0, import_smithy_client._json)(data);
  const response = {
    $metadata: deserializeMetadata(output),
    ...contents
  };
  return response;
}, "de_PutRecommendationPreferencesCommand");
var de_UpdateEnrollmentStatusCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const data = await parseBody(output.body, context);
  let contents = {};
  contents = (0, import_smithy_client._json)(data);
  const response = {
    $metadata: deserializeMetadata(output),
    ...contents
  };
  return response;
}, "de_UpdateEnrollmentStatusCommand");
var de_CommandError = /* @__PURE__ */ __name(async (output, context) => {
  const parsedOutput = {
    ...output,
    body: await parseErrorBody(output.body, context)
  };
  const errorCode = loadRestJsonErrorCode(output, parsedOutput.body);
  switch (errorCode) {
    case "AccessDeniedException":
    case "com.amazonaws.computeoptimizer#AccessDeniedException":
      throw await de_AccessDeniedExceptionRes(parsedOutput, context);
    case "InternalServerException":
    case "com.amazonaws.computeoptimizer#InternalServerException":
      throw await de_InternalServerExceptionRes(parsedOutput, context);
    case "InvalidParameterValueException":
    case "com.amazonaws.computeoptimizer#InvalidParameterValueException":
      throw await de_InvalidParameterValueExceptionRes(parsedOutput, context);
    case "MissingAuthenticationToken":
    case "com.amazonaws.computeoptimizer#MissingAuthenticationToken":
      throw await de_MissingAuthenticationTokenRes(parsedOutput, context);
    case "OptInRequiredException":
    case "com.amazonaws.computeoptimizer#OptInRequiredException":
      throw await de_OptInRequiredExceptionRes(parsedOutput, context);
    case "ResourceNotFoundException":
    case "com.amazonaws.computeoptimizer#ResourceNotFoundException":
      throw await de_ResourceNotFoundExceptionRes(parsedOutput, context);
    case "ServiceUnavailableException":
    case "com.amazonaws.computeoptimizer#ServiceUnavailableException":
      throw await de_ServiceUnavailableExceptionRes(parsedOutput, context);
    case "ThrottlingException":
    case "com.amazonaws.computeoptimizer#ThrottlingException":
      throw await de_ThrottlingExceptionRes(parsedOutput, context);
    case "LimitExceededException":
    case "com.amazonaws.computeoptimizer#LimitExceededException":
      throw await de_LimitExceededExceptionRes(parsedOutput, context);
    default:
      const parsedBody = parsedOutput.body;
      return throwDefaultError({
        output,
        parsedBody,
        errorCode
      });
  }
}, "de_CommandError");
var de_AccessDeniedExceptionRes = /* @__PURE__ */ __name(async (parsedOutput, context) => {
  const body = parsedOutput.body;
  const deserialized = (0, import_smithy_client._json)(body);
  const exception = new AccessDeniedException({
    $metadata: deserializeMetadata(parsedOutput),
    ...deserialized
  });
  return (0, import_smithy_client.decorateServiceException)(exception, body);
}, "de_AccessDeniedExceptionRes");
var de_InternalServerExceptionRes = /* @__PURE__ */ __name(async (parsedOutput, context) => {
  const body = parsedOutput.body;
  const deserialized = (0, import_smithy_client._json)(body);
  const exception = new InternalServerException({
    $metadata: deserializeMetadata(parsedOutput),
    ...deserialized
  });
  return (0, import_smithy_client.decorateServiceException)(exception, body);
}, "de_InternalServerExceptionRes");
var de_InvalidParameterValueExceptionRes = /* @__PURE__ */ __name(async (parsedOutput, context) => {
  const body = parsedOutput.body;
  const deserialized = (0, import_smithy_client._json)(body);
  const exception = new InvalidParameterValueException({
    $metadata: deserializeMetadata(parsedOutput),
    ...deserialized
  });
  return (0, import_smithy_client.decorateServiceException)(exception, body);
}, "de_InvalidParameterValueExceptionRes");
var de_LimitExceededExceptionRes = /* @__PURE__ */ __name(async (parsedOutput, context) => {
  const body = parsedOutput.body;
  const deserialized = (0, import_smithy_client._json)(body);
  const exception = new LimitExceededException({
    $metadata: deserializeMetadata(parsedOutput),
    ...deserialized
  });
  return (0, import_smithy_client.decorateServiceException)(exception, body);
}, "de_LimitExceededExceptionRes");
var de_MissingAuthenticationTokenRes = /* @__PURE__ */ __name(async (parsedOutput, context) => {
  const body = parsedOutput.body;
  const deserialized = (0, import_smithy_client._json)(body);
  const exception = new MissingAuthenticationToken({
    $metadata: deserializeMetadata(parsedOutput),
    ...deserialized
  });
  return (0, import_smithy_client.decorateServiceException)(exception, body);
}, "de_MissingAuthenticationTokenRes");
var de_OptInRequiredExceptionRes = /* @__PURE__ */ __name(async (parsedOutput, context) => {
  const body = parsedOutput.body;
  const deserialized = (0, import_smithy_client._json)(body);
  const exception = new OptInRequiredException({
    $metadata: deserializeMetadata(parsedOutput),
    ...deserialized
  });
  return (0, import_smithy_client.decorateServiceException)(exception, body);
}, "de_OptInRequiredExceptionRes");
var de_ResourceNotFoundExceptionRes = /* @__PURE__ */ __name(async (parsedOutput, context) => {
  const body = parsedOutput.body;
  const deserialized = (0, import_smithy_client._json)(body);
  const exception = new ResourceNotFoundException({
    $metadata: deserializeMetadata(parsedOutput),
    ...deserialized
  });
  return (0, import_smithy_client.decorateServiceException)(exception, body);
}, "de_ResourceNotFoundExceptionRes");
var de_ServiceUnavailableExceptionRes = /* @__PURE__ */ __name(async (parsedOutput, context) => {
  const body = parsedOutput.body;
  const deserialized = (0, import_smithy_client._json)(body);
  const exception = new ServiceUnavailableException({
    $metadata: deserializeMetadata(parsedOutput),
    ...deserialized
  });
  return (0, import_smithy_client.decorateServiceException)(exception, body);
}, "de_ServiceUnavailableExceptionRes");
var de_ThrottlingExceptionRes = /* @__PURE__ */ __name(async (parsedOutput, context) => {
  const body = parsedOutput.body;
  const deserialized = (0, import_smithy_client._json)(body);
  const exception = new ThrottlingException({
    $metadata: deserializeMetadata(parsedOutput),
    ...deserialized
  });
  return (0, import_smithy_client.decorateServiceException)(exception, body);
}, "de_ThrottlingExceptionRes");
var se_GetEC2RecommendationProjectedMetricsRequest = /* @__PURE__ */ __name((input, context) => {
  return (0, import_smithy_client.take)(input, {
    endTime: (_) => Math.round(_.getTime() / 1e3),
    instanceArn: [],
    period: [],
    recommendationPreferences: import_smithy_client._json,
    startTime: (_) => Math.round(_.getTime() / 1e3),
    stat: []
  });
}, "se_GetEC2RecommendationProjectedMetricsRequest");
var se_GetECSServiceRecommendationProjectedMetricsRequest = /* @__PURE__ */ __name((input, context) => {
  return (0, import_smithy_client.take)(input, {
    endTime: (_) => Math.round(_.getTime() / 1e3),
    period: [],
    serviceArn: [],
    startTime: (_) => Math.round(_.getTime() / 1e3),
    stat: []
  });
}, "se_GetECSServiceRecommendationProjectedMetricsRequest");
var de_AccountEnrollmentStatus = /* @__PURE__ */ __name((output, context) => {
  return (0, import_smithy_client.take)(output, {
    accountId: import_smithy_client.expectString,
    lastUpdatedTimestamp: (_) => (0, import_smithy_client.expectNonNull)((0, import_smithy_client.parseEpochTimestamp)((0, import_smithy_client.expectNumber)(_))),
    status: import_smithy_client.expectString,
    statusReason: import_smithy_client.expectString
  });
}, "de_AccountEnrollmentStatus");
var de_AccountEnrollmentStatuses = /* @__PURE__ */ __name((output, context) => {
  const retVal = (output || []).filter((e) => e != null).map((entry) => {
    return de_AccountEnrollmentStatus(entry, context);
  });
  return retVal;
}, "de_AccountEnrollmentStatuses");
var de_AutoScalingGroupEstimatedMonthlySavings = /* @__PURE__ */ __name((output, context) => {
  return (0, import_smithy_client.take)(output, {
    currency: import_smithy_client.expectString,
    value: import_smithy_client.limitedParseDouble
  });
}, "de_AutoScalingGroupEstimatedMonthlySavings");
var de_AutoScalingGroupRecommendation = /* @__PURE__ */ __name((output, context) => {
  return (0, import_smithy_client.take)(output, {
    accountId: import_smithy_client.expectString,
    autoScalingGroupArn: import_smithy_client.expectString,
    autoScalingGroupName: import_smithy_client.expectString,
    currentConfiguration: import_smithy_client._json,
    currentInstanceGpuInfo: import_smithy_client._json,
    currentPerformanceRisk: import_smithy_client.expectString,
    effectiveRecommendationPreferences: import_smithy_client._json,
    finding: import_smithy_client.expectString,
    inferredWorkloadTypes: import_smithy_client._json,
    lastRefreshTimestamp: (_) => (0, import_smithy_client.expectNonNull)((0, import_smithy_client.parseEpochTimestamp)((0, import_smithy_client.expectNumber)(_))),
    lookBackPeriodInDays: import_smithy_client.limitedParseDouble,
    recommendationOptions: (_) => de_AutoScalingGroupRecommendationOptions(_, context),
    utilizationMetrics: (_) => de_UtilizationMetrics(_, context)
  });
}, "de_AutoScalingGroupRecommendation");
var de_AutoScalingGroupRecommendationOption = /* @__PURE__ */ __name((output, context) => {
  return (0, import_smithy_client.take)(output, {
    configuration: import_smithy_client._json,
    instanceGpuInfo: import_smithy_client._json,
    migrationEffort: import_smithy_client.expectString,
    performanceRisk: import_smithy_client.limitedParseDouble,
    projectedUtilizationMetrics: (_) => de_ProjectedUtilizationMetrics(_, context),
    rank: import_smithy_client.expectInt32,
    savingsOpportunity: (_) => de_SavingsOpportunity(_, context),
    savingsOpportunityAfterDiscounts: (_) => de_AutoScalingGroupSavingsOpportunityAfterDiscounts(_, context)
  });
}, "de_AutoScalingGroupRecommendationOption");
var de_AutoScalingGroupRecommendationOptions = /* @__PURE__ */ __name((output, context) => {
  const retVal = (output || []).filter((e) => e != null).map((entry) => {
    return de_AutoScalingGroupRecommendationOption(entry, context);
  });
  return retVal;
}, "de_AutoScalingGroupRecommendationOptions");
var de_AutoScalingGroupRecommendations = /* @__PURE__ */ __name((output, context) => {
  const retVal = (output || []).filter((e) => e != null).map((entry) => {
    return de_AutoScalingGroupRecommendation(entry, context);
  });
  return retVal;
}, "de_AutoScalingGroupRecommendations");
var de_AutoScalingGroupSavingsOpportunityAfterDiscounts = /* @__PURE__ */ __name((output, context) => {
  return (0, import_smithy_client.take)(output, {
    estimatedMonthlySavings: (_) => de_AutoScalingGroupEstimatedMonthlySavings(_, context),
    savingsOpportunityPercentage: import_smithy_client.limitedParseDouble
  });
}, "de_AutoScalingGroupSavingsOpportunityAfterDiscounts");
var de_DescribeRecommendationExportJobsResponse = /* @__PURE__ */ __name((output, context) => {
  return (0, import_smithy_client.take)(output, {
    nextToken: import_smithy_client.expectString,
    recommendationExportJobs: (_) => de_RecommendationExportJobs(_, context)
  });
}, "de_DescribeRecommendationExportJobsResponse");
var de_EBSEstimatedMonthlySavings = /* @__PURE__ */ __name((output, context) => {
  return (0, import_smithy_client.take)(output, {
    currency: import_smithy_client.expectString,
    value: import_smithy_client.limitedParseDouble
  });
}, "de_EBSEstimatedMonthlySavings");
var de_EBSSavingsOpportunityAfterDiscounts = /* @__PURE__ */ __name((output, context) => {
  return (0, import_smithy_client.take)(output, {
    estimatedMonthlySavings: (_) => de_EBSEstimatedMonthlySavings(_, context),
    savingsOpportunityPercentage: import_smithy_client.limitedParseDouble
  });
}, "de_EBSSavingsOpportunityAfterDiscounts");
var de_EBSUtilizationMetric = /* @__PURE__ */ __name((output, context) => {
  return (0, import_smithy_client.take)(output, {
    name: import_smithy_client.expectString,
    statistic: import_smithy_client.expectString,
    value: import_smithy_client.limitedParseDouble
  });
}, "de_EBSUtilizationMetric");
var de_EBSUtilizationMetrics = /* @__PURE__ */ __name((output, context) => {
  const retVal = (output || []).filter((e) => e != null).map((entry) => {
    return de_EBSUtilizationMetric(entry, context);
  });
  return retVal;
}, "de_EBSUtilizationMetrics");
var de_ECSEstimatedMonthlySavings = /* @__PURE__ */ __name((output, context) => {
  return (0, import_smithy_client.take)(output, {
    currency: import_smithy_client.expectString,
    value: import_smithy_client.limitedParseDouble
  });
}, "de_ECSEstimatedMonthlySavings");
var de_ECSSavingsOpportunityAfterDiscounts = /* @__PURE__ */ __name((output, context) => {
  return (0, import_smithy_client.take)(output, {
    estimatedMonthlySavings: (_) => de_ECSEstimatedMonthlySavings(_, context),
    savingsOpportunityPercentage: import_smithy_client.limitedParseDouble
  });
}, "de_ECSSavingsOpportunityAfterDiscounts");
var de_ECSServiceProjectedMetric = /* @__PURE__ */ __name((output, context) => {
  return (0, import_smithy_client.take)(output, {
    lowerBoundValues: (_) => de_MetricValues(_, context),
    name: import_smithy_client.expectString,
    timestamps: (_) => de_Timestamps(_, context),
    upperBoundValues: (_) => de_MetricValues(_, context)
  });
}, "de_ECSServiceProjectedMetric");
var de_ECSServiceProjectedMetrics = /* @__PURE__ */ __name((output, context) => {
  const retVal = (output || []).filter((e) => e != null).map((entry) => {
    return de_ECSServiceProjectedMetric(entry, context);
  });
  return retVal;
}, "de_ECSServiceProjectedMetrics");
var de_ECSServiceProjectedUtilizationMetric = /* @__PURE__ */ __name((output, context) => {
  return (0, import_smithy_client.take)(output, {
    lowerBoundValue: import_smithy_client.limitedParseDouble,
    name: import_smithy_client.expectString,
    statistic: import_smithy_client.expectString,
    upperBoundValue: import_smithy_client.limitedParseDouble
  });
}, "de_ECSServiceProjectedUtilizationMetric");
var de_ECSServiceProjectedUtilizationMetrics = /* @__PURE__ */ __name((output, context) => {
  const retVal = (output || []).filter((e) => e != null).map((entry) => {
    return de_ECSServiceProjectedUtilizationMetric(entry, context);
  });
  return retVal;
}, "de_ECSServiceProjectedUtilizationMetrics");
var de_ECSServiceRecommendation = /* @__PURE__ */ __name((output, context) => {
  return (0, import_smithy_client.take)(output, {
    accountId: import_smithy_client.expectString,
    currentPerformanceRisk: import_smithy_client.expectString,
    currentServiceConfiguration: import_smithy_client._json,
    effectiveRecommendationPreferences: import_smithy_client._json,
    finding: import_smithy_client.expectString,
    findingReasonCodes: import_smithy_client._json,
    lastRefreshTimestamp: (_) => (0, import_smithy_client.expectNonNull)((0, import_smithy_client.parseEpochTimestamp)((0, import_smithy_client.expectNumber)(_))),
    launchType: import_smithy_client.expectString,
    lookbackPeriodInDays: import_smithy_client.limitedParseDouble,
    serviceArn: import_smithy_client.expectString,
    serviceRecommendationOptions: (_) => de_ECSServiceRecommendationOptions(_, context),
    tags: import_smithy_client._json,
    utilizationMetrics: (_) => de_ECSServiceUtilizationMetrics(_, context)
  });
}, "de_ECSServiceRecommendation");
var de_ECSServiceRecommendationOption = /* @__PURE__ */ __name((output, context) => {
  return (0, import_smithy_client.take)(output, {
    containerRecommendations: import_smithy_client._json,
    cpu: import_smithy_client.expectInt32,
    memory: import_smithy_client.expectInt32,
    projectedUtilizationMetrics: (_) => de_ECSServiceProjectedUtilizationMetrics(_, context),
    savingsOpportunity: (_) => de_SavingsOpportunity(_, context),
    savingsOpportunityAfterDiscounts: (_) => de_ECSSavingsOpportunityAfterDiscounts(_, context)
  });
}, "de_ECSServiceRecommendationOption");
var de_ECSServiceRecommendationOptions = /* @__PURE__ */ __name((output, context) => {
  const retVal = (output || []).filter((e) => e != null).map((entry) => {
    return de_ECSServiceRecommendationOption(entry, context);
  });
  return retVal;
}, "de_ECSServiceRecommendationOptions");
var de_ECSServiceRecommendations = /* @__PURE__ */ __name((output, context) => {
  const retVal = (output || []).filter((e) => e != null).map((entry) => {
    return de_ECSServiceRecommendation(entry, context);
  });
  return retVal;
}, "de_ECSServiceRecommendations");
var de_ECSServiceRecommendedOptionProjectedMetric = /* @__PURE__ */ __name((output, context) => {
  return (0, import_smithy_client.take)(output, {
    projectedMetrics: (_) => de_ECSServiceProjectedMetrics(_, context),
    recommendedCpuUnits: import_smithy_client.expectInt32,
    recommendedMemorySize: import_smithy_client.expectInt32
  });
}, "de_ECSServiceRecommendedOptionProjectedMetric");
var de_ECSServiceRecommendedOptionProjectedMetrics = /* @__PURE__ */ __name((output, context) => {
  const retVal = (output || []).filter((e) => e != null).map((entry) => {
    return de_ECSServiceRecommendedOptionProjectedMetric(entry, context);
  });
  return retVal;
}, "de_ECSServiceRecommendedOptionProjectedMetrics");
var de_ECSServiceUtilizationMetric = /* @__PURE__ */ __name((output, context) => {
  return (0, import_smithy_client.take)(output, {
    name: import_smithy_client.expectString,
    statistic: import_smithy_client.expectString,
    value: import_smithy_client.limitedParseDouble
  });
}, "de_ECSServiceUtilizationMetric");
var de_ECSServiceUtilizationMetrics = /* @__PURE__ */ __name((output, context) => {
  const retVal = (output || []).filter((e) => e != null).map((entry) => {
    return de_ECSServiceUtilizationMetric(entry, context);
  });
  return retVal;
}, "de_ECSServiceUtilizationMetrics");
var de_EstimatedMonthlySavings = /* @__PURE__ */ __name((output, context) => {
  return (0, import_smithy_client.take)(output, {
    currency: import_smithy_client.expectString,
    value: import_smithy_client.limitedParseDouble
  });
}, "de_EstimatedMonthlySavings");
var de_GetAutoScalingGroupRecommendationsResponse = /* @__PURE__ */ __name((output, context) => {
  return (0, import_smithy_client.take)(output, {
    autoScalingGroupRecommendations: (_) => de_AutoScalingGroupRecommendations(_, context),
    errors: import_smithy_client._json,
    nextToken: import_smithy_client.expectString
  });
}, "de_GetAutoScalingGroupRecommendationsResponse");
var de_GetEBSVolumeRecommendationsResponse = /* @__PURE__ */ __name((output, context) => {
  return (0, import_smithy_client.take)(output, {
    errors: import_smithy_client._json,
    nextToken: import_smithy_client.expectString,
    volumeRecommendations: (_) => de_VolumeRecommendations(_, context)
  });
}, "de_GetEBSVolumeRecommendationsResponse");
var de_GetEC2InstanceRecommendationsResponse = /* @__PURE__ */ __name((output, context) => {
  return (0, import_smithy_client.take)(output, {
    errors: import_smithy_client._json,
    instanceRecommendations: (_) => de_InstanceRecommendations(_, context),
    nextToken: import_smithy_client.expectString
  });
}, "de_GetEC2InstanceRecommendationsResponse");
var de_GetEC2RecommendationProjectedMetricsResponse = /* @__PURE__ */ __name((output, context) => {
  return (0, import_smithy_client.take)(output, {
    recommendedOptionProjectedMetrics: (_) => de_RecommendedOptionProjectedMetrics(_, context)
  });
}, "de_GetEC2RecommendationProjectedMetricsResponse");
var de_GetECSServiceRecommendationProjectedMetricsResponse = /* @__PURE__ */ __name((output, context) => {
  return (0, import_smithy_client.take)(output, {
    recommendedOptionProjectedMetrics: (_) => de_ECSServiceRecommendedOptionProjectedMetrics(_, context)
  });
}, "de_GetECSServiceRecommendationProjectedMetricsResponse");
var de_GetECSServiceRecommendationsResponse = /* @__PURE__ */ __name((output, context) => {
  return (0, import_smithy_client.take)(output, {
    ecsServiceRecommendations: (_) => de_ECSServiceRecommendations(_, context),
    errors: import_smithy_client._json,
    nextToken: import_smithy_client.expectString
  });
}, "de_GetECSServiceRecommendationsResponse");
var de_GetEnrollmentStatusesForOrganizationResponse = /* @__PURE__ */ __name((output, context) => {
  return (0, import_smithy_client.take)(output, {
    accountEnrollmentStatuses: (_) => de_AccountEnrollmentStatuses(_, context),
    nextToken: import_smithy_client.expectString
  });
}, "de_GetEnrollmentStatusesForOrganizationResponse");
var de_GetEnrollmentStatusResponse = /* @__PURE__ */ __name((output, context) => {
  return (0, import_smithy_client.take)(output, {
    lastUpdatedTimestamp: (_) => (0, import_smithy_client.expectNonNull)((0, import_smithy_client.parseEpochTimestamp)((0, import_smithy_client.expectNumber)(_))),
    memberAccountsEnrolled: import_smithy_client.expectBoolean,
    numberOfMemberAccountsOptedIn: import_smithy_client.expectInt32,
    status: import_smithy_client.expectString,
    statusReason: import_smithy_client.expectString
  });
}, "de_GetEnrollmentStatusResponse");
var de_GetLambdaFunctionRecommendationsResponse = /* @__PURE__ */ __name((output, context) => {
  return (0, import_smithy_client.take)(output, {
    lambdaFunctionRecommendations: (_) => de_LambdaFunctionRecommendations(_, context),
    nextToken: import_smithy_client.expectString
  });
}, "de_GetLambdaFunctionRecommendationsResponse");
var de_GetLicenseRecommendationsResponse = /* @__PURE__ */ __name((output, context) => {
  return (0, import_smithy_client.take)(output, {
    errors: import_smithy_client._json,
    licenseRecommendations: (_) => de_LicenseRecommendations(_, context),
    nextToken: import_smithy_client.expectString
  });
}, "de_GetLicenseRecommendationsResponse");
var de_GetRecommendationSummariesResponse = /* @__PURE__ */ __name((output, context) => {
  return (0, import_smithy_client.take)(output, {
    nextToken: import_smithy_client.expectString,
    recommendationSummaries: (_) => de_RecommendationSummaries(_, context)
  });
}, "de_GetRecommendationSummariesResponse");
var de_InferredWorkloadSaving = /* @__PURE__ */ __name((output, context) => {
  return (0, import_smithy_client.take)(output, {
    estimatedMonthlySavings: (_) => de_EstimatedMonthlySavings(_, context),
    inferredWorkloadTypes: import_smithy_client._json
  });
}, "de_InferredWorkloadSaving");
var de_InferredWorkloadSavings = /* @__PURE__ */ __name((output, context) => {
  const retVal = (output || []).filter((e) => e != null).map((entry) => {
    return de_InferredWorkloadSaving(entry, context);
  });
  return retVal;
}, "de_InferredWorkloadSavings");
var de_InstanceEstimatedMonthlySavings = /* @__PURE__ */ __name((output, context) => {
  return (0, import_smithy_client.take)(output, {
    currency: import_smithy_client.expectString,
    value: import_smithy_client.limitedParseDouble
  });
}, "de_InstanceEstimatedMonthlySavings");
var de_InstanceRecommendation = /* @__PURE__ */ __name((output, context) => {
  return (0, import_smithy_client.take)(output, {
    accountId: import_smithy_client.expectString,
    currentInstanceGpuInfo: import_smithy_client._json,
    currentInstanceType: import_smithy_client.expectString,
    currentPerformanceRisk: import_smithy_client.expectString,
    effectiveRecommendationPreferences: import_smithy_client._json,
    externalMetricStatus: import_smithy_client._json,
    finding: import_smithy_client.expectString,
    findingReasonCodes: import_smithy_client._json,
    idle: import_smithy_client.expectString,
    inferredWorkloadTypes: import_smithy_client._json,
    instanceArn: import_smithy_client.expectString,
    instanceName: import_smithy_client.expectString,
    instanceState: import_smithy_client.expectString,
    lastRefreshTimestamp: (_) => (0, import_smithy_client.expectNonNull)((0, import_smithy_client.parseEpochTimestamp)((0, import_smithy_client.expectNumber)(_))),
    lookBackPeriodInDays: import_smithy_client.limitedParseDouble,
    recommendationOptions: (_) => de_RecommendationOptions(_, context),
    recommendationSources: import_smithy_client._json,
    tags: import_smithy_client._json,
    utilizationMetrics: (_) => de_UtilizationMetrics(_, context)
  });
}, "de_InstanceRecommendation");
var de_InstanceRecommendationOption = /* @__PURE__ */ __name((output, context) => {
  return (0, import_smithy_client.take)(output, {
    instanceGpuInfo: import_smithy_client._json,
    instanceType: import_smithy_client.expectString,
    migrationEffort: import_smithy_client.expectString,
    performanceRisk: import_smithy_client.limitedParseDouble,
    platformDifferences: import_smithy_client._json,
    projectedUtilizationMetrics: (_) => de_ProjectedUtilizationMetrics(_, context),
    rank: import_smithy_client.expectInt32,
    savingsOpportunity: (_) => de_SavingsOpportunity(_, context),
    savingsOpportunityAfterDiscounts: (_) => de_InstanceSavingsOpportunityAfterDiscounts(_, context)
  });
}, "de_InstanceRecommendationOption");
var de_InstanceRecommendations = /* @__PURE__ */ __name((output, context) => {
  const retVal = (output || []).filter((e) => e != null).map((entry) => {
    return de_InstanceRecommendation(entry, context);
  });
  return retVal;
}, "de_InstanceRecommendations");
var de_InstanceSavingsOpportunityAfterDiscounts = /* @__PURE__ */ __name((output, context) => {
  return (0, import_smithy_client.take)(output, {
    estimatedMonthlySavings: (_) => de_InstanceEstimatedMonthlySavings(_, context),
    savingsOpportunityPercentage: import_smithy_client.limitedParseDouble
  });
}, "de_InstanceSavingsOpportunityAfterDiscounts");
var de_LambdaEstimatedMonthlySavings = /* @__PURE__ */ __name((output, context) => {
  return (0, import_smithy_client.take)(output, {
    currency: import_smithy_client.expectString,
    value: import_smithy_client.limitedParseDouble
  });
}, "de_LambdaEstimatedMonthlySavings");
var de_LambdaFunctionMemoryProjectedMetric = /* @__PURE__ */ __name((output, context) => {
  return (0, import_smithy_client.take)(output, {
    name: import_smithy_client.expectString,
    statistic: import_smithy_client.expectString,
    value: import_smithy_client.limitedParseDouble
  });
}, "de_LambdaFunctionMemoryProjectedMetric");
var de_LambdaFunctionMemoryProjectedMetrics = /* @__PURE__ */ __name((output, context) => {
  const retVal = (output || []).filter((e) => e != null).map((entry) => {
    return de_LambdaFunctionMemoryProjectedMetric(entry, context);
  });
  return retVal;
}, "de_LambdaFunctionMemoryProjectedMetrics");
var de_LambdaFunctionMemoryRecommendationOption = /* @__PURE__ */ __name((output, context) => {
  return (0, import_smithy_client.take)(output, {
    memorySize: import_smithy_client.expectInt32,
    projectedUtilizationMetrics: (_) => de_LambdaFunctionMemoryProjectedMetrics(_, context),
    rank: import_smithy_client.expectInt32,
    savingsOpportunity: (_) => de_SavingsOpportunity(_, context),
    savingsOpportunityAfterDiscounts: (_) => de_LambdaSavingsOpportunityAfterDiscounts(_, context)
  });
}, "de_LambdaFunctionMemoryRecommendationOption");
var de_LambdaFunctionMemoryRecommendationOptions = /* @__PURE__ */ __name((output, context) => {
  const retVal = (output || []).filter((e) => e != null).map((entry) => {
    return de_LambdaFunctionMemoryRecommendationOption(entry, context);
  });
  return retVal;
}, "de_LambdaFunctionMemoryRecommendationOptions");
var de_LambdaFunctionRecommendation = /* @__PURE__ */ __name((output, context) => {
  return (0, import_smithy_client.take)(output, {
    accountId: import_smithy_client.expectString,
    currentMemorySize: import_smithy_client.expectInt32,
    currentPerformanceRisk: import_smithy_client.expectString,
    effectiveRecommendationPreferences: import_smithy_client._json,
    finding: import_smithy_client.expectString,
    findingReasonCodes: import_smithy_client._json,
    functionArn: import_smithy_client.expectString,
    functionVersion: import_smithy_client.expectString,
    lastRefreshTimestamp: (_) => (0, import_smithy_client.expectNonNull)((0, import_smithy_client.parseEpochTimestamp)((0, import_smithy_client.expectNumber)(_))),
    lookbackPeriodInDays: import_smithy_client.limitedParseDouble,
    memorySizeRecommendationOptions: (_) => de_LambdaFunctionMemoryRecommendationOptions(_, context),
    numberOfInvocations: import_smithy_client.expectLong,
    tags: import_smithy_client._json,
    utilizationMetrics: (_) => de_LambdaFunctionUtilizationMetrics(_, context)
  });
}, "de_LambdaFunctionRecommendation");
var de_LambdaFunctionRecommendations = /* @__PURE__ */ __name((output, context) => {
  const retVal = (output || []).filter((e) => e != null).map((entry) => {
    return de_LambdaFunctionRecommendation(entry, context);
  });
  return retVal;
}, "de_LambdaFunctionRecommendations");
var de_LambdaFunctionUtilizationMetric = /* @__PURE__ */ __name((output, context) => {
  return (0, import_smithy_client.take)(output, {
    name: import_smithy_client.expectString,
    statistic: import_smithy_client.expectString,
    value: import_smithy_client.limitedParseDouble
  });
}, "de_LambdaFunctionUtilizationMetric");
var de_LambdaFunctionUtilizationMetrics = /* @__PURE__ */ __name((output, context) => {
  const retVal = (output || []).filter((e) => e != null).map((entry) => {
    return de_LambdaFunctionUtilizationMetric(entry, context);
  });
  return retVal;
}, "de_LambdaFunctionUtilizationMetrics");
var de_LambdaSavingsOpportunityAfterDiscounts = /* @__PURE__ */ __name((output, context) => {
  return (0, import_smithy_client.take)(output, {
    estimatedMonthlySavings: (_) => de_LambdaEstimatedMonthlySavings(_, context),
    savingsOpportunityPercentage: import_smithy_client.limitedParseDouble
  });
}, "de_LambdaSavingsOpportunityAfterDiscounts");
var de_LicenseRecommendation = /* @__PURE__ */ __name((output, context) => {
  return (0, import_smithy_client.take)(output, {
    accountId: import_smithy_client.expectString,
    currentLicenseConfiguration: import_smithy_client._json,
    finding: import_smithy_client.expectString,
    findingReasonCodes: import_smithy_client._json,
    lastRefreshTimestamp: (_) => (0, import_smithy_client.expectNonNull)((0, import_smithy_client.parseEpochTimestamp)((0, import_smithy_client.expectNumber)(_))),
    licenseRecommendationOptions: (_) => de_LicenseRecommendationOptions(_, context),
    lookbackPeriodInDays: import_smithy_client.limitedParseDouble,
    resourceArn: import_smithy_client.expectString,
    tags: import_smithy_client._json
  });
}, "de_LicenseRecommendation");
var de_LicenseRecommendationOption = /* @__PURE__ */ __name((output, context) => {
  return (0, import_smithy_client.take)(output, {
    licenseEdition: import_smithy_client.expectString,
    licenseModel: import_smithy_client.expectString,
    operatingSystem: import_smithy_client.expectString,
    rank: import_smithy_client.expectInt32,
    savingsOpportunity: (_) => de_SavingsOpportunity(_, context)
  });
}, "de_LicenseRecommendationOption");
var de_LicenseRecommendationOptions = /* @__PURE__ */ __name((output, context) => {
  const retVal = (output || []).filter((e) => e != null).map((entry) => {
    return de_LicenseRecommendationOption(entry, context);
  });
  return retVal;
}, "de_LicenseRecommendationOptions");
var de_LicenseRecommendations = /* @__PURE__ */ __name((output, context) => {
  const retVal = (output || []).filter((e) => e != null).map((entry) => {
    return de_LicenseRecommendation(entry, context);
  });
  return retVal;
}, "de_LicenseRecommendations");
var de_MetricValues = /* @__PURE__ */ __name((output, context) => {
  const retVal = (output || []).filter((e) => e != null).map((entry) => {
    return (0, import_smithy_client.limitedParseDouble)(entry);
  });
  return retVal;
}, "de_MetricValues");
var de_ProjectedMetric = /* @__PURE__ */ __name((output, context) => {
  return (0, import_smithy_client.take)(output, {
    name: import_smithy_client.expectString,
    timestamps: (_) => de_Timestamps(_, context),
    values: (_) => de_MetricValues(_, context)
  });
}, "de_ProjectedMetric");
var de_ProjectedMetrics = /* @__PURE__ */ __name((output, context) => {
  const retVal = (output || []).filter((e) => e != null).map((entry) => {
    return de_ProjectedMetric(entry, context);
  });
  return retVal;
}, "de_ProjectedMetrics");
var de_ProjectedUtilizationMetrics = /* @__PURE__ */ __name((output, context) => {
  const retVal = (output || []).filter((e) => e != null).map((entry) => {
    return de_UtilizationMetric(entry, context);
  });
  return retVal;
}, "de_ProjectedUtilizationMetrics");
var de_ReasonCodeSummaries = /* @__PURE__ */ __name((output, context) => {
  const retVal = (output || []).filter((e) => e != null).map((entry) => {
    return de_ReasonCodeSummary(entry, context);
  });
  return retVal;
}, "de_ReasonCodeSummaries");
var de_ReasonCodeSummary = /* @__PURE__ */ __name((output, context) => {
  return (0, import_smithy_client.take)(output, {
    name: import_smithy_client.expectString,
    value: import_smithy_client.limitedParseDouble
  });
}, "de_ReasonCodeSummary");
var de_RecommendationExportJob = /* @__PURE__ */ __name((output, context) => {
  return (0, import_smithy_client.take)(output, {
    creationTimestamp: (_) => (0, import_smithy_client.expectNonNull)((0, import_smithy_client.parseEpochTimestamp)((0, import_smithy_client.expectNumber)(_))),
    destination: import_smithy_client._json,
    failureReason: import_smithy_client.expectString,
    jobId: import_smithy_client.expectString,
    lastUpdatedTimestamp: (_) => (0, import_smithy_client.expectNonNull)((0, import_smithy_client.parseEpochTimestamp)((0, import_smithy_client.expectNumber)(_))),
    resourceType: import_smithy_client.expectString,
    status: import_smithy_client.expectString
  });
}, "de_RecommendationExportJob");
var de_RecommendationExportJobs = /* @__PURE__ */ __name((output, context) => {
  const retVal = (output || []).filter((e) => e != null).map((entry) => {
    return de_RecommendationExportJob(entry, context);
  });
  return retVal;
}, "de_RecommendationExportJobs");
var de_RecommendationOptions = /* @__PURE__ */ __name((output, context) => {
  const retVal = (output || []).filter((e) => e != null).map((entry) => {
    return de_InstanceRecommendationOption(entry, context);
  });
  return retVal;
}, "de_RecommendationOptions");
var de_RecommendationSummaries = /* @__PURE__ */ __name((output, context) => {
  const retVal = (output || []).filter((e) => e != null).map((entry) => {
    return de_RecommendationSummary(entry, context);
  });
  return retVal;
}, "de_RecommendationSummaries");
var de_RecommendationSummary = /* @__PURE__ */ __name((output, context) => {
  return (0, import_smithy_client.take)(output, {
    accountId: import_smithy_client.expectString,
    currentPerformanceRiskRatings: import_smithy_client._json,
    inferredWorkloadSavings: (_) => de_InferredWorkloadSavings(_, context),
    recommendationResourceType: import_smithy_client.expectString,
    savingsOpportunity: (_) => de_SavingsOpportunity(_, context),
    summaries: (_) => de_Summaries(_, context)
  });
}, "de_RecommendationSummary");
var de_RecommendedOptionProjectedMetric = /* @__PURE__ */ __name((output, context) => {
  return (0, import_smithy_client.take)(output, {
    projectedMetrics: (_) => de_ProjectedMetrics(_, context),
    rank: import_smithy_client.expectInt32,
    recommendedInstanceType: import_smithy_client.expectString
  });
}, "de_RecommendedOptionProjectedMetric");
var de_RecommendedOptionProjectedMetrics = /* @__PURE__ */ __name((output, context) => {
  const retVal = (output || []).filter((e) => e != null).map((entry) => {
    return de_RecommendedOptionProjectedMetric(entry, context);
  });
  return retVal;
}, "de_RecommendedOptionProjectedMetrics");
var de_SavingsOpportunity = /* @__PURE__ */ __name((output, context) => {
  return (0, import_smithy_client.take)(output, {
    estimatedMonthlySavings: (_) => de_EstimatedMonthlySavings(_, context),
    savingsOpportunityPercentage: import_smithy_client.limitedParseDouble
  });
}, "de_SavingsOpportunity");
var de_Summaries = /* @__PURE__ */ __name((output, context) => {
  const retVal = (output || []).filter((e) => e != null).map((entry) => {
    return de_Summary(entry, context);
  });
  return retVal;
}, "de_Summaries");
var de_Summary = /* @__PURE__ */ __name((output, context) => {
  return (0, import_smithy_client.take)(output, {
    name: import_smithy_client.expectString,
    reasonCodeSummaries: (_) => de_ReasonCodeSummaries(_, context),
    value: import_smithy_client.limitedParseDouble
  });
}, "de_Summary");
var de_Timestamps = /* @__PURE__ */ __name((output, context) => {
  const retVal = (output || []).filter((e) => e != null).map((entry) => {
    return (0, import_smithy_client.expectNonNull)((0, import_smithy_client.parseEpochTimestamp)((0, import_smithy_client.expectNumber)(entry)));
  });
  return retVal;
}, "de_Timestamps");
var de_UtilizationMetric = /* @__PURE__ */ __name((output, context) => {
  return (0, import_smithy_client.take)(output, {
    name: import_smithy_client.expectString,
    statistic: import_smithy_client.expectString,
    value: import_smithy_client.limitedParseDouble
  });
}, "de_UtilizationMetric");
var de_UtilizationMetrics = /* @__PURE__ */ __name((output, context) => {
  const retVal = (output || []).filter((e) => e != null).map((entry) => {
    return de_UtilizationMetric(entry, context);
  });
  return retVal;
}, "de_UtilizationMetrics");
var de_VolumeRecommendation = /* @__PURE__ */ __name((output, context) => {
  return (0, import_smithy_client.take)(output, {
    accountId: import_smithy_client.expectString,
    currentConfiguration: import_smithy_client._json,
    currentPerformanceRisk: import_smithy_client.expectString,
    effectiveRecommendationPreferences: import_smithy_client._json,
    finding: import_smithy_client.expectString,
    lastRefreshTimestamp: (_) => (0, import_smithy_client.expectNonNull)((0, import_smithy_client.parseEpochTimestamp)((0, import_smithy_client.expectNumber)(_))),
    lookBackPeriodInDays: import_smithy_client.limitedParseDouble,
    tags: import_smithy_client._json,
    utilizationMetrics: (_) => de_EBSUtilizationMetrics(_, context),
    volumeArn: import_smithy_client.expectString,
    volumeRecommendationOptions: (_) => de_VolumeRecommendationOptions(_, context)
  });
}, "de_VolumeRecommendation");
var de_VolumeRecommendationOption = /* @__PURE__ */ __name((output, context) => {
  return (0, import_smithy_client.take)(output, {
    configuration: import_smithy_client._json,
    performanceRisk: import_smithy_client.limitedParseDouble,
    rank: import_smithy_client.expectInt32,
    savingsOpportunity: (_) => de_SavingsOpportunity(_, context),
    savingsOpportunityAfterDiscounts: (_) => de_EBSSavingsOpportunityAfterDiscounts(_, context)
  });
}, "de_VolumeRecommendationOption");
var de_VolumeRecommendationOptions = /* @__PURE__ */ __name((output, context) => {
  const retVal = (output || []).filter((e) => e != null).map((entry) => {
    return de_VolumeRecommendationOption(entry, context);
  });
  return retVal;
}, "de_VolumeRecommendationOptions");
var de_VolumeRecommendations = /* @__PURE__ */ __name((output, context) => {
  const retVal = (output || []).filter((e) => e != null).map((entry) => {
    return de_VolumeRecommendation(entry, context);
  });
  return retVal;
}, "de_VolumeRecommendations");
var deserializeMetadata = /* @__PURE__ */ __name((output) => ({
  httpStatusCode: output.statusCode,
  requestId: output.headers["x-amzn-requestid"] ?? output.headers["x-amzn-request-id"] ?? output.headers["x-amz-request-id"],
  extendedRequestId: output.headers["x-amz-id-2"],
  cfId: output.headers["x-amz-cf-id"]
}), "deserializeMetadata");
var collectBodyString = /* @__PURE__ */ __name((streamBody, context) => (0, import_smithy_client.collectBody)(streamBody, context).then((body) => context.utf8Encoder(body)), "collectBodyString");
var throwDefaultError = (0, import_smithy_client.withBaseException)(ComputeOptimizerServiceException);
var buildHttpRpcRequest = /* @__PURE__ */ __name(async (context, headers, path, resolvedHostname, body) => {
  const { hostname, protocol = "https", port, path: basePath } = await context.endpoint();
  const contents = {
    protocol,
    hostname,
    port,
    method: "POST",
    path: basePath.endsWith("/") ? basePath.slice(0, -1) + path : basePath + path,
    headers
  };
  if (resolvedHostname !== void 0) {
    contents.hostname = resolvedHostname;
  }
  if (body !== void 0) {
    contents.body = body;
  }
  return new import_protocol_http.HttpRequest(contents);
}, "buildHttpRpcRequest");
function sharedHeaders(operation) {
  return {
    "content-type": "application/x-amz-json-1.0",
    "x-amz-target": `ComputeOptimizerService.${operation}`
  };
}
__name(sharedHeaders, "sharedHeaders");
var parseBody = /* @__PURE__ */ __name((streamBody, context) => collectBodyString(streamBody, context).then((encoded) => {
  if (encoded.length) {
    return JSON.parse(encoded);
  }
  return {};
}), "parseBody");
var parseErrorBody = /* @__PURE__ */ __name(async (errorBody, context) => {
  const value = await parseBody(errorBody, context);
  value.message = value.message ?? value.Message;
  return value;
}, "parseErrorBody");
var loadRestJsonErrorCode = /* @__PURE__ */ __name((output, data) => {
  const findKey = /* @__PURE__ */ __name((object, key) => Object.keys(object).find((k) => k.toLowerCase() === key.toLowerCase()), "findKey");
  const sanitizeErrorCode = /* @__PURE__ */ __name((rawValue) => {
    let cleanValue = rawValue;
    if (typeof cleanValue === "number") {
      cleanValue = cleanValue.toString();
    }
    if (cleanValue.indexOf(",") >= 0) {
      cleanValue = cleanValue.split(",")[0];
    }
    if (cleanValue.indexOf(":") >= 0) {
      cleanValue = cleanValue.split(":")[0];
    }
    if (cleanValue.indexOf("#") >= 0) {
      cleanValue = cleanValue.split("#")[1];
    }
    return cleanValue;
  }, "sanitizeErrorCode");
  const headerKey = findKey(output.headers, "x-amzn-errortype");
  if (headerKey !== void 0) {
    return sanitizeErrorCode(output.headers[headerKey]);
  }
  if (data.code !== void 0) {
    return sanitizeErrorCode(data.code);
  }
  if (data["__type"] !== void 0) {
    return sanitizeErrorCode(data["__type"]);
  }
}, "loadRestJsonErrorCode");

// src/commands/DeleteRecommendationPreferencesCommand.ts
var _DeleteRecommendationPreferencesCommand = class _DeleteRecommendationPreferencesCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("ComputeOptimizerService", "DeleteRecommendationPreferences", {}).n("ComputeOptimizerClient", "DeleteRecommendationPreferencesCommand").f(void 0, void 0).ser(se_DeleteRecommendationPreferencesCommand).de(de_DeleteRecommendationPreferencesCommand).build() {
};
__name(_DeleteRecommendationPreferencesCommand, "DeleteRecommendationPreferencesCommand");
var DeleteRecommendationPreferencesCommand = _DeleteRecommendationPreferencesCommand;

// src/commands/DescribeRecommendationExportJobsCommand.ts




var _DescribeRecommendationExportJobsCommand = class _DescribeRecommendationExportJobsCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("ComputeOptimizerService", "DescribeRecommendationExportJobs", {}).n("ComputeOptimizerClient", "DescribeRecommendationExportJobsCommand").f(void 0, void 0).ser(se_DescribeRecommendationExportJobsCommand).de(de_DescribeRecommendationExportJobsCommand).build() {
};
__name(_DescribeRecommendationExportJobsCommand, "DescribeRecommendationExportJobsCommand");
var DescribeRecommendationExportJobsCommand = _DescribeRecommendationExportJobsCommand;

// src/commands/ExportAutoScalingGroupRecommendationsCommand.ts




var _ExportAutoScalingGroupRecommendationsCommand = class _ExportAutoScalingGroupRecommendationsCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("ComputeOptimizerService", "ExportAutoScalingGroupRecommendations", {}).n("ComputeOptimizerClient", "ExportAutoScalingGroupRecommendationsCommand").f(void 0, void 0).ser(se_ExportAutoScalingGroupRecommendationsCommand).de(de_ExportAutoScalingGroupRecommendationsCommand).build() {
};
__name(_ExportAutoScalingGroupRecommendationsCommand, "ExportAutoScalingGroupRecommendationsCommand");
var ExportAutoScalingGroupRecommendationsCommand = _ExportAutoScalingGroupRecommendationsCommand;

// src/commands/ExportEBSVolumeRecommendationsCommand.ts




var _ExportEBSVolumeRecommendationsCommand = class _ExportEBSVolumeRecommendationsCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("ComputeOptimizerService", "ExportEBSVolumeRecommendations", {}).n("ComputeOptimizerClient", "ExportEBSVolumeRecommendationsCommand").f(void 0, void 0).ser(se_ExportEBSVolumeRecommendationsCommand).de(de_ExportEBSVolumeRecommendationsCommand).build() {
};
__name(_ExportEBSVolumeRecommendationsCommand, "ExportEBSVolumeRecommendationsCommand");
var ExportEBSVolumeRecommendationsCommand = _ExportEBSVolumeRecommendationsCommand;

// src/commands/ExportEC2InstanceRecommendationsCommand.ts




var _ExportEC2InstanceRecommendationsCommand = class _ExportEC2InstanceRecommendationsCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("ComputeOptimizerService", "ExportEC2InstanceRecommendations", {}).n("ComputeOptimizerClient", "ExportEC2InstanceRecommendationsCommand").f(void 0, void 0).ser(se_ExportEC2InstanceRecommendationsCommand).de(de_ExportEC2InstanceRecommendationsCommand).build() {
};
__name(_ExportEC2InstanceRecommendationsCommand, "ExportEC2InstanceRecommendationsCommand");
var ExportEC2InstanceRecommendationsCommand = _ExportEC2InstanceRecommendationsCommand;

// src/commands/ExportECSServiceRecommendationsCommand.ts




var _ExportECSServiceRecommendationsCommand = class _ExportECSServiceRecommendationsCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("ComputeOptimizerService", "ExportECSServiceRecommendations", {}).n("ComputeOptimizerClient", "ExportECSServiceRecommendationsCommand").f(void 0, void 0).ser(se_ExportECSServiceRecommendationsCommand).de(de_ExportECSServiceRecommendationsCommand).build() {
};
__name(_ExportECSServiceRecommendationsCommand, "ExportECSServiceRecommendationsCommand");
var ExportECSServiceRecommendationsCommand = _ExportECSServiceRecommendationsCommand;

// src/commands/ExportLambdaFunctionRecommendationsCommand.ts




var _ExportLambdaFunctionRecommendationsCommand = class _ExportLambdaFunctionRecommendationsCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("ComputeOptimizerService", "ExportLambdaFunctionRecommendations", {}).n("ComputeOptimizerClient", "ExportLambdaFunctionRecommendationsCommand").f(void 0, void 0).ser(se_ExportLambdaFunctionRecommendationsCommand).de(de_ExportLambdaFunctionRecommendationsCommand).build() {
};
__name(_ExportLambdaFunctionRecommendationsCommand, "ExportLambdaFunctionRecommendationsCommand");
var ExportLambdaFunctionRecommendationsCommand = _ExportLambdaFunctionRecommendationsCommand;

// src/commands/ExportLicenseRecommendationsCommand.ts




var _ExportLicenseRecommendationsCommand = class _ExportLicenseRecommendationsCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("ComputeOptimizerService", "ExportLicenseRecommendations", {}).n("ComputeOptimizerClient", "ExportLicenseRecommendationsCommand").f(void 0, void 0).ser(se_ExportLicenseRecommendationsCommand).de(de_ExportLicenseRecommendationsCommand).build() {
};
__name(_ExportLicenseRecommendationsCommand, "ExportLicenseRecommendationsCommand");
var ExportLicenseRecommendationsCommand = _ExportLicenseRecommendationsCommand;

// src/commands/GetAutoScalingGroupRecommendationsCommand.ts




var _GetAutoScalingGroupRecommendationsCommand = class _GetAutoScalingGroupRecommendationsCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("ComputeOptimizerService", "GetAutoScalingGroupRecommendations", {}).n("ComputeOptimizerClient", "GetAutoScalingGroupRecommendationsCommand").f(void 0, void 0).ser(se_GetAutoScalingGroupRecommendationsCommand).de(de_GetAutoScalingGroupRecommendationsCommand).build() {
};
__name(_GetAutoScalingGroupRecommendationsCommand, "GetAutoScalingGroupRecommendationsCommand");
var GetAutoScalingGroupRecommendationsCommand = _GetAutoScalingGroupRecommendationsCommand;

// src/commands/GetEBSVolumeRecommendationsCommand.ts




var _GetEBSVolumeRecommendationsCommand = class _GetEBSVolumeRecommendationsCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("ComputeOptimizerService", "GetEBSVolumeRecommendations", {}).n("ComputeOptimizerClient", "GetEBSVolumeRecommendationsCommand").f(void 0, void 0).ser(se_GetEBSVolumeRecommendationsCommand).de(de_GetEBSVolumeRecommendationsCommand).build() {
};
__name(_GetEBSVolumeRecommendationsCommand, "GetEBSVolumeRecommendationsCommand");
var GetEBSVolumeRecommendationsCommand = _GetEBSVolumeRecommendationsCommand;

// src/commands/GetEC2InstanceRecommendationsCommand.ts




var _GetEC2InstanceRecommendationsCommand = class _GetEC2InstanceRecommendationsCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("ComputeOptimizerService", "GetEC2InstanceRecommendations", {}).n("ComputeOptimizerClient", "GetEC2InstanceRecommendationsCommand").f(void 0, void 0).ser(se_GetEC2InstanceRecommendationsCommand).de(de_GetEC2InstanceRecommendationsCommand).build() {
};
__name(_GetEC2InstanceRecommendationsCommand, "GetEC2InstanceRecommendationsCommand");
var GetEC2InstanceRecommendationsCommand = _GetEC2InstanceRecommendationsCommand;

// src/commands/GetEC2RecommendationProjectedMetricsCommand.ts




var _GetEC2RecommendationProjectedMetricsCommand = class _GetEC2RecommendationProjectedMetricsCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("ComputeOptimizerService", "GetEC2RecommendationProjectedMetrics", {}).n("ComputeOptimizerClient", "GetEC2RecommendationProjectedMetricsCommand").f(void 0, void 0).ser(se_GetEC2RecommendationProjectedMetricsCommand).de(de_GetEC2RecommendationProjectedMetricsCommand).build() {
};
__name(_GetEC2RecommendationProjectedMetricsCommand, "GetEC2RecommendationProjectedMetricsCommand");
var GetEC2RecommendationProjectedMetricsCommand = _GetEC2RecommendationProjectedMetricsCommand;

// src/commands/GetECSServiceRecommendationProjectedMetricsCommand.ts




var _GetECSServiceRecommendationProjectedMetricsCommand = class _GetECSServiceRecommendationProjectedMetricsCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("ComputeOptimizerService", "GetECSServiceRecommendationProjectedMetrics", {}).n("ComputeOptimizerClient", "GetECSServiceRecommendationProjectedMetricsCommand").f(void 0, void 0).ser(se_GetECSServiceRecommendationProjectedMetricsCommand).de(de_GetECSServiceRecommendationProjectedMetricsCommand).build() {
};
__name(_GetECSServiceRecommendationProjectedMetricsCommand, "GetECSServiceRecommendationProjectedMetricsCommand");
var GetECSServiceRecommendationProjectedMetricsCommand = _GetECSServiceRecommendationProjectedMetricsCommand;

// src/commands/GetECSServiceRecommendationsCommand.ts




var _GetECSServiceRecommendationsCommand = class _GetECSServiceRecommendationsCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("ComputeOptimizerService", "GetECSServiceRecommendations", {}).n("ComputeOptimizerClient", "GetECSServiceRecommendationsCommand").f(void 0, void 0).ser(se_GetECSServiceRecommendationsCommand).de(de_GetECSServiceRecommendationsCommand).build() {
};
__name(_GetECSServiceRecommendationsCommand, "GetECSServiceRecommendationsCommand");
var GetECSServiceRecommendationsCommand = _GetECSServiceRecommendationsCommand;

// src/commands/GetEffectiveRecommendationPreferencesCommand.ts




var _GetEffectiveRecommendationPreferencesCommand = class _GetEffectiveRecommendationPreferencesCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("ComputeOptimizerService", "GetEffectiveRecommendationPreferences", {}).n("ComputeOptimizerClient", "GetEffectiveRecommendationPreferencesCommand").f(void 0, void 0).ser(se_GetEffectiveRecommendationPreferencesCommand).de(de_GetEffectiveRecommendationPreferencesCommand).build() {
};
__name(_GetEffectiveRecommendationPreferencesCommand, "GetEffectiveRecommendationPreferencesCommand");
var GetEffectiveRecommendationPreferencesCommand = _GetEffectiveRecommendationPreferencesCommand;

// src/commands/GetEnrollmentStatusCommand.ts




var _GetEnrollmentStatusCommand = class _GetEnrollmentStatusCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("ComputeOptimizerService", "GetEnrollmentStatus", {}).n("ComputeOptimizerClient", "GetEnrollmentStatusCommand").f(void 0, void 0).ser(se_GetEnrollmentStatusCommand).de(de_GetEnrollmentStatusCommand).build() {
};
__name(_GetEnrollmentStatusCommand, "GetEnrollmentStatusCommand");
var GetEnrollmentStatusCommand = _GetEnrollmentStatusCommand;

// src/commands/GetEnrollmentStatusesForOrganizationCommand.ts




var _GetEnrollmentStatusesForOrganizationCommand = class _GetEnrollmentStatusesForOrganizationCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("ComputeOptimizerService", "GetEnrollmentStatusesForOrganization", {}).n("ComputeOptimizerClient", "GetEnrollmentStatusesForOrganizationCommand").f(void 0, void 0).ser(se_GetEnrollmentStatusesForOrganizationCommand).de(de_GetEnrollmentStatusesForOrganizationCommand).build() {
};
__name(_GetEnrollmentStatusesForOrganizationCommand, "GetEnrollmentStatusesForOrganizationCommand");
var GetEnrollmentStatusesForOrganizationCommand = _GetEnrollmentStatusesForOrganizationCommand;

// src/commands/GetLambdaFunctionRecommendationsCommand.ts




var _GetLambdaFunctionRecommendationsCommand = class _GetLambdaFunctionRecommendationsCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("ComputeOptimizerService", "GetLambdaFunctionRecommendations", {}).n("ComputeOptimizerClient", "GetLambdaFunctionRecommendationsCommand").f(void 0, void 0).ser(se_GetLambdaFunctionRecommendationsCommand).de(de_GetLambdaFunctionRecommendationsCommand).build() {
};
__name(_GetLambdaFunctionRecommendationsCommand, "GetLambdaFunctionRecommendationsCommand");
var GetLambdaFunctionRecommendationsCommand = _GetLambdaFunctionRecommendationsCommand;

// src/commands/GetLicenseRecommendationsCommand.ts




var _GetLicenseRecommendationsCommand = class _GetLicenseRecommendationsCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("ComputeOptimizerService", "GetLicenseRecommendations", {}).n("ComputeOptimizerClient", "GetLicenseRecommendationsCommand").f(void 0, void 0).ser(se_GetLicenseRecommendationsCommand).de(de_GetLicenseRecommendationsCommand).build() {
};
__name(_GetLicenseRecommendationsCommand, "GetLicenseRecommendationsCommand");
var GetLicenseRecommendationsCommand = _GetLicenseRecommendationsCommand;

// src/commands/GetRecommendationPreferencesCommand.ts




var _GetRecommendationPreferencesCommand = class _GetRecommendationPreferencesCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("ComputeOptimizerService", "GetRecommendationPreferences", {}).n("ComputeOptimizerClient", "GetRecommendationPreferencesCommand").f(void 0, void 0).ser(se_GetRecommendationPreferencesCommand).de(de_GetRecommendationPreferencesCommand).build() {
};
__name(_GetRecommendationPreferencesCommand, "GetRecommendationPreferencesCommand");
var GetRecommendationPreferencesCommand = _GetRecommendationPreferencesCommand;

// src/commands/GetRecommendationSummariesCommand.ts




var _GetRecommendationSummariesCommand = class _GetRecommendationSummariesCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("ComputeOptimizerService", "GetRecommendationSummaries", {}).n("ComputeOptimizerClient", "GetRecommendationSummariesCommand").f(void 0, void 0).ser(se_GetRecommendationSummariesCommand).de(de_GetRecommendationSummariesCommand).build() {
};
__name(_GetRecommendationSummariesCommand, "GetRecommendationSummariesCommand");
var GetRecommendationSummariesCommand = _GetRecommendationSummariesCommand;

// src/commands/PutRecommendationPreferencesCommand.ts




var _PutRecommendationPreferencesCommand = class _PutRecommendationPreferencesCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("ComputeOptimizerService", "PutRecommendationPreferences", {}).n("ComputeOptimizerClient", "PutRecommendationPreferencesCommand").f(void 0, void 0).ser(se_PutRecommendationPreferencesCommand).de(de_PutRecommendationPreferencesCommand).build() {
};
__name(_PutRecommendationPreferencesCommand, "PutRecommendationPreferencesCommand");
var PutRecommendationPreferencesCommand = _PutRecommendationPreferencesCommand;

// src/commands/UpdateEnrollmentStatusCommand.ts




var _UpdateEnrollmentStatusCommand = class _UpdateEnrollmentStatusCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("ComputeOptimizerService", "UpdateEnrollmentStatus", {}).n("ComputeOptimizerClient", "UpdateEnrollmentStatusCommand").f(void 0, void 0).ser(se_UpdateEnrollmentStatusCommand).de(de_UpdateEnrollmentStatusCommand).build() {
};
__name(_UpdateEnrollmentStatusCommand, "UpdateEnrollmentStatusCommand");
var UpdateEnrollmentStatusCommand = _UpdateEnrollmentStatusCommand;

// src/ComputeOptimizer.ts
var commands = {
  DeleteRecommendationPreferencesCommand,
  DescribeRecommendationExportJobsCommand,
  ExportAutoScalingGroupRecommendationsCommand,
  ExportEBSVolumeRecommendationsCommand,
  ExportEC2InstanceRecommendationsCommand,
  ExportECSServiceRecommendationsCommand,
  ExportLambdaFunctionRecommendationsCommand,
  ExportLicenseRecommendationsCommand,
  GetAutoScalingGroupRecommendationsCommand,
  GetEBSVolumeRecommendationsCommand,
  GetEC2InstanceRecommendationsCommand,
  GetEC2RecommendationProjectedMetricsCommand,
  GetECSServiceRecommendationProjectedMetricsCommand,
  GetECSServiceRecommendationsCommand,
  GetEffectiveRecommendationPreferencesCommand,
  GetEnrollmentStatusCommand,
  GetEnrollmentStatusesForOrganizationCommand,
  GetLambdaFunctionRecommendationsCommand,
  GetLicenseRecommendationsCommand,
  GetRecommendationPreferencesCommand,
  GetRecommendationSummariesCommand,
  PutRecommendationPreferencesCommand,
  UpdateEnrollmentStatusCommand
};
var _ComputeOptimizer = class _ComputeOptimizer extends ComputeOptimizerClient {
};
__name(_ComputeOptimizer, "ComputeOptimizer");
var ComputeOptimizer = _ComputeOptimizer;
(0, import_smithy_client.createAggregatedClient)(commands, ComputeOptimizer);

// src/pagination/DescribeRecommendationExportJobsPaginator.ts

var paginateDescribeRecommendationExportJobs = (0, import_core.createPaginator)(ComputeOptimizerClient, DescribeRecommendationExportJobsCommand, "nextToken", "nextToken", "maxResults");

// src/pagination/GetEnrollmentStatusesForOrganizationPaginator.ts

var paginateGetEnrollmentStatusesForOrganization = (0, import_core.createPaginator)(ComputeOptimizerClient, GetEnrollmentStatusesForOrganizationCommand, "nextToken", "nextToken", "maxResults");

// src/pagination/GetLambdaFunctionRecommendationsPaginator.ts

var paginateGetLambdaFunctionRecommendations = (0, import_core.createPaginator)(ComputeOptimizerClient, GetLambdaFunctionRecommendationsCommand, "nextToken", "nextToken", "maxResults");

// src/pagination/GetRecommendationPreferencesPaginator.ts

var paginateGetRecommendationPreferences = (0, import_core.createPaginator)(ComputeOptimizerClient, GetRecommendationPreferencesCommand, "nextToken", "nextToken", "maxResults");

// src/pagination/GetRecommendationSummariesPaginator.ts

var paginateGetRecommendationSummaries = (0, import_core.createPaginator)(ComputeOptimizerClient, GetRecommendationSummariesCommand, "nextToken", "nextToken", "maxResults");

// src/index.ts
var import_util_endpoints = require("@aws-sdk/util-endpoints");
// Annotate the CommonJS export names for ESM import in node:

0 && (module.exports = {
  ComputeOptimizerServiceException,
  __Client,
  ComputeOptimizerClient,
  ComputeOptimizer,
  $Command,
  DeleteRecommendationPreferencesCommand,
  DescribeRecommendationExportJobsCommand,
  ExportAutoScalingGroupRecommendationsCommand,
  ExportEBSVolumeRecommendationsCommand,
  ExportEC2InstanceRecommendationsCommand,
  ExportECSServiceRecommendationsCommand,
  ExportLambdaFunctionRecommendationsCommand,
  ExportLicenseRecommendationsCommand,
  GetAutoScalingGroupRecommendationsCommand,
  GetEBSVolumeRecommendationsCommand,
  GetEC2InstanceRecommendationsCommand,
  GetEC2RecommendationProjectedMetricsCommand,
  GetECSServiceRecommendationProjectedMetricsCommand,
  GetECSServiceRecommendationsCommand,
  GetEffectiveRecommendationPreferencesCommand,
  GetEnrollmentStatusCommand,
  GetEnrollmentStatusesForOrganizationCommand,
  GetLambdaFunctionRecommendationsCommand,
  GetLicenseRecommendationsCommand,
  GetRecommendationPreferencesCommand,
  GetRecommendationSummariesCommand,
  PutRecommendationPreferencesCommand,
  UpdateEnrollmentStatusCommand,
  paginateDescribeRecommendationExportJobs,
  paginateGetEnrollmentStatusesForOrganization,
  paginateGetLambdaFunctionRecommendations,
  paginateGetRecommendationPreferences,
  paginateGetRecommendationSummaries,
  AccessDeniedException,
  Status,
  AutoScalingConfiguration,
  Currency,
  CurrentPerformanceRisk,
  CpuVendorArchitecture,
  EnhancedInfrastructureMetrics,
  ExternalMetricsSource,
  InferredWorkloadTypesPreference,
  LookBackPeriodPreference,
  PreferredResourceName,
  InstanceSavingsEstimationModeSource,
  CustomizableMetricName,
  CustomizableMetricHeadroom,
  CustomizableMetricThreshold,
  Finding,
  InferredWorkloadType,
  MigrationEffort,
  MetricName,
  MetricStatistic,
  RecommendationPreferenceName,
  ResourceType,
  ScopeName,
  InternalServerException,
  InvalidParameterValueException,
  MissingAuthenticationToken,
  OptInRequiredException,
  ResourceNotFoundException,
  ServiceUnavailableException,
  ThrottlingException,
  JobFilterName,
  JobStatus,
  ExportableAutoScalingGroupField,
  FileFormat,
  FilterName,
  LimitExceededException,
  ExportableVolumeField,
  EBSFilterName,
  ExportableInstanceField,
  ExportableECSServiceField,
  ECSServiceRecommendationFilterName,
  ExportableLambdaFunctionField,
  LambdaFunctionRecommendationFilterName,
  ExportableLicenseField,
  LicenseRecommendationFilterName,
  EBSSavingsEstimationModeSource,
  EBSFinding,
  EBSMetricName,
  ExternalMetricStatusCode,
  InstanceRecommendationFindingReasonCode,
  InstanceIdle,
  InstanceState,
  PlatformDifference,
  RecommendationSourceType,
  ECSServiceMetricName,
  ECSSavingsEstimationModeSource,
  ECSServiceRecommendationFinding,
  ECSServiceRecommendationFindingReasonCode,
  ECSServiceLaunchType,
  ECSServiceMetricStatistic,
  EnrollmentFilterName,
  LambdaSavingsEstimationModeSource,
  LambdaFunctionRecommendationFinding,
  LambdaFunctionRecommendationFindingReasonCode,
  LambdaFunctionMemoryMetricName,
  LambdaFunctionMemoryMetricStatistic,
  LambdaFunctionMetricName,
  LambdaFunctionMetricStatistic,
  LicenseEdition,
  LicenseModel,
  LicenseName,
  MetricSourceProvider,
  LicenseFinding,
  LicenseFindingReasonCode,
  SavingsEstimationMode,
  FindingReasonCode
});
