// @azure/arm-devtestlabs/dist-esm/src/operations/costs.js
/*
 * Copyright (c) Microsoft Corporation.
 * Licensed under the MIT License.
 *
 * Code generated by Microsoft (R) AutoRest Code Generator.
 * Changes may cause incorrect behavior and will be lost if the code is regenerated.
 */
import * as coreClient from "@azure/core-client";
import * as Mappers from "../models/mappers";
import * as Parameters from "../models/parameters";
/** Class containing Costs operations. */
export class CostsImpl {
    /**
     * Initialize a new instance of the class Costs class.
     * @param client Reference to the service client
     */
    constructor(client) {
        this.client = client;
    }
    /**
     * Get cost.
     * @param resourceGroupName The name of the resource group.
     * @param labName The name of the lab.
     * @param name The name of the cost.
     * @param options The options parameters.
     */
    get(resourceGroupName, labName, name, options) {
        return this.client.sendOperationRequest({ resourceGroupName, labName, name, options }, getOperationSpec);
    }
    /**
     * Create or replace an existing cost.
     * @param resourceGroupName The name of the resource group.
     * @param labName The name of the lab.
     * @param name The name of the cost.
     * @param labCost A cost item.
     * @param options The options parameters.
     */
    createOrUpdate(resourceGroupName, labName, name, labCost, options) {
        return this.client.sendOperationRequest({ resourceGroupName, labName, name, labCost, options }, createOrUpdateOperationSpec);
    }
}
// Operation Specifications
const serializer = coreClient.createSerializer(Mappers, /* isXml */ false);
const getOperationSpec = {
    path: "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.DevTestLab/labs/{labName}/costs/{name}",
    httpMethod: "GET",
    responses: {
        200: {
            bodyMapper: Mappers.LabCost
        },
        default: {
            bodyMapper: Mappers.CloudError
        }
    },
    queryParameters: [Parameters.apiVersion, Parameters.expand],
    urlParameters: [
        Parameters.$host,
        Parameters.subscriptionId,
        Parameters.resourceGroupName,
        Parameters.name,
        Parameters.labName
    ],
    headerParameters: [Parameters.accept],
    serializer
};
const createOrUpdateOperationSpec = {
    path: "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.DevTestLab/labs/{labName}/costs/{name}",
    httpMethod: "PUT",
    responses: {
        200: {
            bodyMapper: Mappers.LabCost
        },
        201: {
            bodyMapper: Mappers.LabCost
        },
        default: {
            bodyMapper: Mappers.CloudError
        }
    },
    requestBody: Parameters.labCost,
    queryParameters: [Parameters.apiVersion],
    urlParameters: [
        Parameters.$host,
        Parameters.subscriptionId,
        Parameters.resourceGroupName,
        Parameters.name,
        Parameters.labName
    ],
    headerParameters: [Parameters.accept, Parameters.contentType],
    mediaType: "json",
    serializer
};
