// @aws-sdk/client-lightsail/dist-es/models/models_1.js
import { SENSITIVE_STRING } from "@smithy/smithy-client";
export const HttpEndpoint = {
    disabled: "disabled",
    enabled: "enabled",
};
export const HttpProtocolIpv6 = {
    disabled: "disabled",
    enabled: "enabled",
};
export const HttpTokens = {
    optional: "optional",
    required: "required",
};
export const InstanceMetadataState = {
    applied: "applied",
    pending: "pending",
};
export const PortAccessType = {
    Private: "Private",
    Public: "Public",
};
export const InstanceAccessProtocol = {
    rdp: "rdp",
    ssh: "ssh",
};
export const InstanceMetricName = {
    BurstCapacityPercentage: "BurstCapacityPercentage",
    BurstCapacityTime: "BurstCapacityTime",
    CPUUtilization: "CPUUtilization",
    MetadataNoToken: "MetadataNoToken",
    NetworkIn: "NetworkIn",
    NetworkOut: "NetworkOut",
    StatusCheckFailed: "StatusCheckFailed",
    StatusCheckFailed_Instance: "StatusCheckFailed_Instance",
    StatusCheckFailed_System: "StatusCheckFailed_System",
};
export const PortState = {
    Closed: "closed",
    Open: "open",
};
export const InstanceSnapshotState = {
    Available: "available",
    Error: "error",
    Pending: "pending",
};
export const LoadBalancerAttributeName = {
    HealthCheckPath: "HealthCheckPath",
    HttpsRedirectionEnabled: "HttpsRedirectionEnabled",
    SessionStickinessEnabled: "SessionStickinessEnabled",
    SessionStickiness_LB_CookieDurationSeconds: "SessionStickiness_LB_CookieDurationSeconds",
    TlsPolicyName: "TlsPolicyName",
};
export const InstanceHealthState = {
    Draining: "draining",
    Healthy: "healthy",
    Initial: "initial",
    Unavailable: "unavailable",
    Unhealthy: "unhealthy",
    Unused: "unused",
};
export const InstanceHealthReason = {
    InstanceDeregistrationInProgress: "Instance.DeregistrationInProgress",
    InstanceFailedHealthChecks: "Instance.FailedHealthChecks",
    InstanceInvalidState: "Instance.InvalidState",
    InstanceIpUnusable: "Instance.IpUnusable",
    InstanceNotInUse: "Instance.NotInUse",
    InstanceNotRegistered: "Instance.NotRegistered",
    InstanceResponseCodeMismatch: "Instance.ResponseCodeMismatch",
    InstanceTimeout: "Instance.Timeout",
    LbInitialHealthChecking: "Lb.InitialHealthChecking",
    LbInternalError: "Lb.InternalError",
    LbRegistrationInProgress: "Lb.RegistrationInProgress",
};
export const LoadBalancerProtocol = {
    HTTP: "HTTP",
    HTTP_HTTPS: "HTTP_HTTPS",
};
export const LoadBalancerState = {
    Active: "active",
    ActiveImpaired: "active_impaired",
    Failed: "failed",
    Provisioning: "provisioning",
    Unknown: "unknown",
};
export const LoadBalancerMetricName = {
    ClientTLSNegotiationErrorCount: "ClientTLSNegotiationErrorCount",
    HTTPCode_Instance_2XX_Count: "HTTPCode_Instance_2XX_Count",
    HTTPCode_Instance_3XX_Count: "HTTPCode_Instance_3XX_Count",
    HTTPCode_Instance_4XX_Count: "HTTPCode_Instance_4XX_Count",
    HTTPCode_Instance_5XX_Count: "HTTPCode_Instance_5XX_Count",
    HTTPCode_LB_4XX_Count: "HTTPCode_LB_4XX_Count",
    HTTPCode_LB_5XX_Count: "HTTPCode_LB_5XX_Count",
    HealthyHostCount: "HealthyHostCount",
    InstanceResponseTime: "InstanceResponseTime",
    RejectedConnectionCount: "RejectedConnectionCount",
    RequestCount: "RequestCount",
    UnhealthyHostCount: "UnhealthyHostCount",
};
export const LoadBalancerTlsCertificateDnsRecordCreationStateCode = {
    Failed: "FAILED",
    Started: "STARTED",
    Succeeded: "SUCCEEDED",
};
export const LoadBalancerTlsCertificateDomainStatus = {
    Failed: "FAILED",
    PendingValidation: "PENDING_VALIDATION",
    Success: "SUCCESS",
};
export const LoadBalancerTlsCertificateFailureReason = {
    AdditionalVerificationRequired: "ADDITIONAL_VERIFICATION_REQUIRED",
    DomainNotAllowed: "DOMAIN_NOT_ALLOWED",
    InvalidPublicDomain: "INVALID_PUBLIC_DOMAIN",
    NoAvailableContacts: "NO_AVAILABLE_CONTACTS",
    Other: "OTHER",
};
export const LoadBalancerTlsCertificateRenewalStatus = {
    Failed: "FAILED",
    PendingAutoRenewal: "PENDING_AUTO_RENEWAL",
    PendingValidation: "PENDING_VALIDATION",
    Success: "SUCCESS",
};
export const LoadBalancerTlsCertificateRevocationReason = {
    AACompromise: "A_A_COMPROMISE",
    AffiliationChanged: "AFFILIATION_CHANGED",
    CaCompromise: "CA_COMPROMISE",
    CertificateHold: "CERTIFICATE_HOLD",
    CessationOfOperation: "CESSATION_OF_OPERATION",
    KeyCompromise: "KEY_COMPROMISE",
    PrivilegeWithdrawn: "PRIVILEGE_WITHDRAWN",
    RemoveFromCrl: "REMOVE_FROM_CRL",
    Superceded: "SUPERCEDED",
    Unspecified: "UNSPECIFIED",
};
export const LoadBalancerTlsCertificateStatus = {
    Expired: "EXPIRED",
    Failed: "FAILED",
    Inactive: "INACTIVE",
    Issued: "ISSUED",
    PendingValidation: "PENDING_VALIDATION",
    Revoked: "REVOKED",
    Unknown: "UNKNOWN",
    ValidationTimedOut: "VALIDATION_TIMED_OUT",
};
export const RelationalDatabaseEngine = {
    MYSQL: "mysql",
};
export const RelationalDatabasePasswordVersion = {
    CURRENT: "CURRENT",
    PENDING: "PENDING",
    PREVIOUS: "PREVIOUS",
};
export const RelationalDatabaseMetricName = {
    CPUUtilization: "CPUUtilization",
    DatabaseConnections: "DatabaseConnections",
    DiskQueueDepth: "DiskQueueDepth",
    FreeStorageSpace: "FreeStorageSpace",
    NetworkReceiveThroughput: "NetworkReceiveThroughput",
    NetworkTransmitThroughput: "NetworkTransmitThroughput",
};
export const SetupStatus = {
    Failed: "failed",
    InProgress: "inProgress",
    Succeeded: "succeeded",
};
export const ResourceBucketAccess = {
    Allow: "allow",
    Deny: "deny",
};
export const GetRelationalDatabaseMasterUserPasswordResultFilterSensitiveLog = (obj) => ({
    ...obj,
    ...(obj.masterUserPassword && { masterUserPassword: SENSITIVE_STRING }),
});
export const SetupInstanceHttpsRequestFilterSensitiveLog = (obj) => ({
    ...obj,
    ...(obj.emailAddress && { emailAddress: SENSITIVE_STRING }),
});
export const UpdateRelationalDatabaseRequestFilterSensitiveLog = (obj) => ({
    ...obj,
    ...(obj.masterUserPassword && { masterUserPassword: SENSITIVE_STRING }),
});
