// @aws-sdk/client-personalize-runtime/dist-es/models/models_0.js
import { SENSITIVE_STRING } from "@smithy/smithy-client";
import { PersonalizeRuntimeServiceException as __BaseException } from "./PersonalizeRuntimeServiceException";
export class InvalidInputException extends __BaseException {
    constructor(opts) {
        super({
            name: "InvalidInputException",
            $fault: "client",
            ...opts,
        });
        this.name = "InvalidInputException";
        this.$fault = "client";
        Object.setPrototypeOf(this, InvalidInputException.prototype);
    }
}
export class ResourceNotFoundException extends __BaseException {
    constructor(opts) {
        super({
            name: "ResourceNotFoundException",
            $fault: "client",
            ...opts,
        });
        this.name = "ResourceNotFoundException";
        this.$fault = "client";
        Object.setPrototypeOf(this, ResourceNotFoundException.prototype);
    }
}
export const GetActionRecommendationsRequestFilterSensitiveLog = (obj) => ({
    ...obj,
    ...(obj.filterValues && { filterValues: SENSITIVE_STRING }),
});
export const GetPersonalizedRankingRequestFilterSensitiveLog = (obj) => ({
    ...obj,
    ...(obj.context && { context: SENSITIVE_STRING }),
    ...(obj.filterValues && { filterValues: SENSITIVE_STRING }),
});
export const PromotionFilterSensitiveLog = (obj) => ({
    ...obj,
    ...(obj.filterValues && { filterValues: SENSITIVE_STRING }),
});
export const GetRecommendationsRequestFilterSensitiveLog = (obj) => ({
    ...obj,
    ...(obj.context && { context: SENSITIVE_STRING }),
    ...(obj.filterValues && { filterValues: SENSITIVE_STRING }),
    ...(obj.promotions && { promotions: obj.promotions.map((item) => PromotionFilterSensitiveLog(item)) }),
});
