// @azure/arm-sqlvirtualmachine/esm/operations/sqlVirtualMachineGroups.js
/*
 * Copyright (c) Microsoft Corporation. All rights reserved.
 * Licensed under the MIT License. See License.txt in the project root for
 * license information.
 *
 * Code generated by Microsoft (R) AutoRest Code Generator.
 * Changes may cause incorrect behavior and will be lost if the code is
 * regenerated.
 */
import { __assign } from "tslib";
import * as msRest from "@azure/ms-rest-js";
import * as Mappers from "../models/sqlVirtualMachineGroupsMappers";
import * as Parameters from "../models/parameters";
/** Class representing a SqlVirtualMachineGroups. */
var SqlVirtualMachineGroups = /** @class */ (function () {
    /**
     * Create a SqlVirtualMachineGroups.
     * @param {SqlVirtualMachineManagementClientContext} client Reference to the service client.
     */
    function SqlVirtualMachineGroups(client) {
        this.client = client;
    }
    SqlVirtualMachineGroups.prototype.get = function (resourceGroupName, sqlVirtualMachineGroupName, options, callback) {
        return this.client.sendOperationRequest({
            resourceGroupName: resourceGroupName,
            sqlVirtualMachineGroupName: sqlVirtualMachineGroupName,
            options: options
        }, getOperationSpec, callback);
    };
    /**
     * Creates or updates a SQL virtual machine group.
     * @param resourceGroupName Name of the resource group that contains the resource. You can obtain
     * this value from the Azure Resource Manager API or the portal.
     * @param sqlVirtualMachineGroupName Name of the SQL virtual machine group.
     * @param parameters The SQL virtual machine group.
     * @param [options] The optional parameters
     * @returns Promise<Models.SqlVirtualMachineGroupsCreateOrUpdateResponse>
     */
    SqlVirtualMachineGroups.prototype.createOrUpdate = function (resourceGroupName, sqlVirtualMachineGroupName, parameters, options) {
        return this.beginCreateOrUpdate(resourceGroupName, sqlVirtualMachineGroupName, parameters, options)
            .then(function (lroPoller) { return lroPoller.pollUntilFinished(); });
    };
    /**
     * Deletes a SQL virtual machine group.
     * @param resourceGroupName Name of the resource group that contains the resource. You can obtain
     * this value from the Azure Resource Manager API or the portal.
     * @param sqlVirtualMachineGroupName Name of the SQL virtual machine group.
     * @param [options] The optional parameters
     * @returns Promise<msRest.RestResponse>
     */
    SqlVirtualMachineGroups.prototype.deleteMethod = function (resourceGroupName, sqlVirtualMachineGroupName, options) {
        return this.beginDeleteMethod(resourceGroupName, sqlVirtualMachineGroupName, options)
            .then(function (lroPoller) { return lroPoller.pollUntilFinished(); });
    };
    /**
     * Updates SQL virtual machine group tags.
     * @param resourceGroupName Name of the resource group that contains the resource. You can obtain
     * this value from the Azure Resource Manager API or the portal.
     * @param sqlVirtualMachineGroupName Name of the SQL virtual machine group.
     * @param parameters The SQL virtual machine group.
     * @param [options] The optional parameters
     * @returns Promise<Models.SqlVirtualMachineGroupsUpdateResponse>
     */
    SqlVirtualMachineGroups.prototype.update = function (resourceGroupName, sqlVirtualMachineGroupName, parameters, options) {
        return this.beginUpdate(resourceGroupName, sqlVirtualMachineGroupName, parameters, options)
            .then(function (lroPoller) { return lroPoller.pollUntilFinished(); });
    };
    SqlVirtualMachineGroups.prototype.listByResourceGroup = function (resourceGroupName, options, callback) {
        return this.client.sendOperationRequest({
            resourceGroupName: resourceGroupName,
            options: options
        }, listByResourceGroupOperationSpec, callback);
    };
    SqlVirtualMachineGroups.prototype.list = function (options, callback) {
        return this.client.sendOperationRequest({
            options: options
        }, listOperationSpec, callback);
    };
    /**
     * Creates or updates a SQL virtual machine group.
     * @param resourceGroupName Name of the resource group that contains the resource. You can obtain
     * this value from the Azure Resource Manager API or the portal.
     * @param sqlVirtualMachineGroupName Name of the SQL virtual machine group.
     * @param parameters The SQL virtual machine group.
     * @param [options] The optional parameters
     * @returns Promise<msRestAzure.LROPoller>
     */
    SqlVirtualMachineGroups.prototype.beginCreateOrUpdate = function (resourceGroupName, sqlVirtualMachineGroupName, parameters, options) {
        return this.client.sendLRORequest({
            resourceGroupName: resourceGroupName,
            sqlVirtualMachineGroupName: sqlVirtualMachineGroupName,
            parameters: parameters,
            options: options
        }, beginCreateOrUpdateOperationSpec, options);
    };
    /**
     * Deletes a SQL virtual machine group.
     * @param resourceGroupName Name of the resource group that contains the resource. You can obtain
     * this value from the Azure Resource Manager API or the portal.
     * @param sqlVirtualMachineGroupName Name of the SQL virtual machine group.
     * @param [options] The optional parameters
     * @returns Promise<msRestAzure.LROPoller>
     */
    SqlVirtualMachineGroups.prototype.beginDeleteMethod = function (resourceGroupName, sqlVirtualMachineGroupName, options) {
        return this.client.sendLRORequest({
            resourceGroupName: resourceGroupName,
            sqlVirtualMachineGroupName: sqlVirtualMachineGroupName,
            options: options
        }, beginDeleteMethodOperationSpec, options);
    };
    /**
     * Updates SQL virtual machine group tags.
     * @param resourceGroupName Name of the resource group that contains the resource. You can obtain
     * this value from the Azure Resource Manager API or the portal.
     * @param sqlVirtualMachineGroupName Name of the SQL virtual machine group.
     * @param parameters The SQL virtual machine group.
     * @param [options] The optional parameters
     * @returns Promise<msRestAzure.LROPoller>
     */
    SqlVirtualMachineGroups.prototype.beginUpdate = function (resourceGroupName, sqlVirtualMachineGroupName, parameters, options) {
        return this.client.sendLRORequest({
            resourceGroupName: resourceGroupName,
            sqlVirtualMachineGroupName: sqlVirtualMachineGroupName,
            parameters: parameters,
            options: options
        }, beginUpdateOperationSpec, options);
    };
    SqlVirtualMachineGroups.prototype.listByResourceGroupNext = function (nextPageLink, options, callback) {
        return this.client.sendOperationRequest({
            nextPageLink: nextPageLink,
            options: options
        }, listByResourceGroupNextOperationSpec, callback);
    };
    SqlVirtualMachineGroups.prototype.listNext = function (nextPageLink, options, callback) {
        return this.client.sendOperationRequest({
            nextPageLink: nextPageLink,
            options: options
        }, listNextOperationSpec, callback);
    };
    return SqlVirtualMachineGroups;
}());
export { SqlVirtualMachineGroups };
// Operation Specifications
var serializer = new msRest.Serializer(Mappers);
var getOperationSpec = {
    httpMethod: "GET",
    path: "subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.SqlVirtualMachine/sqlVirtualMachineGroups/{sqlVirtualMachineGroupName}",
    urlParameters: [
        Parameters.resourceGroupName,
        Parameters.sqlVirtualMachineGroupName,
        Parameters.subscriptionId
    ],
    queryParameters: [
        Parameters.apiVersion
    ],
    headerParameters: [
        Parameters.acceptLanguage
    ],
    responses: {
        200: {
            bodyMapper: Mappers.SqlVirtualMachineGroup
        },
        default: {
            bodyMapper: Mappers.CloudError
        }
    },
    serializer: serializer
};
var listByResourceGroupOperationSpec = {
    httpMethod: "GET",
    path: "subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.SqlVirtualMachine/sqlVirtualMachineGroups",
    urlParameters: [
        Parameters.resourceGroupName,
        Parameters.subscriptionId
    ],
    queryParameters: [
        Parameters.apiVersion
    ],
    headerParameters: [
        Parameters.acceptLanguage
    ],
    responses: {
        200: {
            bodyMapper: Mappers.SqlVirtualMachineGroupListResult
        },
        default: {
            bodyMapper: Mappers.CloudError
        }
    },
    serializer: serializer
};
var listOperationSpec = {
    httpMethod: "GET",
    path: "subscriptions/{subscriptionId}/providers/Microsoft.SqlVirtualMachine/sqlVirtualMachineGroups",
    urlParameters: [
        Parameters.subscriptionId
    ],
    queryParameters: [
        Parameters.apiVersion
    ],
    headerParameters: [
        Parameters.acceptLanguage
    ],
    responses: {
        200: {
            bodyMapper: Mappers.SqlVirtualMachineGroupListResult
        },
        default: {
            bodyMapper: Mappers.CloudError
        }
    },
    serializer: serializer
};
var beginCreateOrUpdateOperationSpec = {
    httpMethod: "PUT",
    path: "subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.SqlVirtualMachine/sqlVirtualMachineGroups/{sqlVirtualMachineGroupName}",
    urlParameters: [
        Parameters.resourceGroupName,
        Parameters.sqlVirtualMachineGroupName,
        Parameters.subscriptionId
    ],
    queryParameters: [
        Parameters.apiVersion
    ],
    headerParameters: [
        Parameters.acceptLanguage
    ],
    requestBody: {
        parameterPath: "parameters",
        mapper: __assign(__assign({}, Mappers.SqlVirtualMachineGroup), { required: true })
    },
    responses: {
        200: {
            bodyMapper: Mappers.SqlVirtualMachineGroup
        },
        201: {
            bodyMapper: Mappers.SqlVirtualMachineGroup
        },
        default: {
            bodyMapper: Mappers.CloudError
        }
    },
    serializer: serializer
};
var beginDeleteMethodOperationSpec = {
    httpMethod: "DELETE",
    path: "subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.SqlVirtualMachine/sqlVirtualMachineGroups/{sqlVirtualMachineGroupName}",
    urlParameters: [
        Parameters.resourceGroupName,
        Parameters.sqlVirtualMachineGroupName,
        Parameters.subscriptionId
    ],
    queryParameters: [
        Parameters.apiVersion
    ],
    headerParameters: [
        Parameters.acceptLanguage
    ],
    responses: {
        200: {},
        202: {},
        204: {},
        default: {
            bodyMapper: Mappers.CloudError
        }
    },
    serializer: serializer
};
var beginUpdateOperationSpec = {
    httpMethod: "PATCH",
    path: "subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.SqlVirtualMachine/sqlVirtualMachineGroups/{sqlVirtualMachineGroupName}",
    urlParameters: [
        Parameters.resourceGroupName,
        Parameters.sqlVirtualMachineGroupName,
        Parameters.subscriptionId
    ],
    queryParameters: [
        Parameters.apiVersion
    ],
    headerParameters: [
        Parameters.acceptLanguage
    ],
    requestBody: {
        parameterPath: "parameters",
        mapper: __assign(__assign({}, Mappers.SqlVirtualMachineGroupUpdate), { required: true })
    },
    responses: {
        200: {
            bodyMapper: Mappers.SqlVirtualMachineGroup
        },
        default: {
            bodyMapper: Mappers.CloudError
        }
    },
    serializer: serializer
};
var listByResourceGroupNextOperationSpec = {
    httpMethod: "GET",
    baseUrl: "https://management.azure.com",
    path: "{nextLink}",
    urlParameters: [
        Parameters.nextPageLink
    ],
    headerParameters: [
        Parameters.acceptLanguage
    ],
    responses: {
        200: {
            bodyMapper: Mappers.SqlVirtualMachineGroupListResult
        },
        default: {
            bodyMapper: Mappers.CloudError
        }
    },
    serializer: serializer
};
var listNextOperationSpec = {
    httpMethod: "GET",
    baseUrl: "https://management.azure.com",
    path: "{nextLink}",
    urlParameters: [
        Parameters.nextPageLink
    ],
    headerParameters: [
        Parameters.acceptLanguage
    ],
    responses: {
        200: {
            bodyMapper: Mappers.SqlVirtualMachineGroupListResult
        },
        default: {
            bodyMapper: Mappers.CloudError
        }
    },
    serializer: serializer
};
