// @azure/arm-powerbidedicated/dist-esm/src/models/index.js
/*
 * Copyright (c) Microsoft Corporation.
 * Licensed under the MIT License.
 *
 * Code generated by Microsoft (R) AutoRest Code Generator.
 * Changes may cause incorrect behavior and will be lost if the code is regenerated.
 */
/** Known values of {@link CapacitySkuTier} that the service accepts. */
export var KnownCapacitySkuTier;
(function (KnownCapacitySkuTier) {
    /** PbieAzure */
    KnownCapacitySkuTier["PbieAzure"] = "PBIE_Azure";
    /** Premium */
    KnownCapacitySkuTier["Premium"] = "Premium";
    /** AutoPremiumHost */
    KnownCapacitySkuTier["AutoPremiumHost"] = "AutoPremiumHost";
})(KnownCapacitySkuTier || (KnownCapacitySkuTier = {}));
/** Known values of {@link State} that the service accepts. */
export var KnownState;
(function (KnownState) {
    /** Deleting */
    KnownState["Deleting"] = "Deleting";
    /** Succeeded */
    KnownState["Succeeded"] = "Succeeded";
    /** Failed */
    KnownState["Failed"] = "Failed";
    /** Paused */
    KnownState["Paused"] = "Paused";
    /** Suspended */
    KnownState["Suspended"] = "Suspended";
    /** Provisioning */
    KnownState["Provisioning"] = "Provisioning";
    /** Updating */
    KnownState["Updating"] = "Updating";
    /** Suspending */
    KnownState["Suspending"] = "Suspending";
    /** Pausing */
    KnownState["Pausing"] = "Pausing";
    /** Resuming */
    KnownState["Resuming"] = "Resuming";
    /** Preparing */
    KnownState["Preparing"] = "Preparing";
    /** Scaling */
    KnownState["Scaling"] = "Scaling";
})(KnownState || (KnownState = {}));
/** Known values of {@link CapacityProvisioningState} that the service accepts. */
export var KnownCapacityProvisioningState;
(function (KnownCapacityProvisioningState) {
    /** Deleting */
    KnownCapacityProvisioningState["Deleting"] = "Deleting";
    /** Succeeded */
    KnownCapacityProvisioningState["Succeeded"] = "Succeeded";
    /** Failed */
    KnownCapacityProvisioningState["Failed"] = "Failed";
    /** Paused */
    KnownCapacityProvisioningState["Paused"] = "Paused";
    /** Suspended */
    KnownCapacityProvisioningState["Suspended"] = "Suspended";
    /** Provisioning */
    KnownCapacityProvisioningState["Provisioning"] = "Provisioning";
    /** Updating */
    KnownCapacityProvisioningState["Updating"] = "Updating";
    /** Suspending */
    KnownCapacityProvisioningState["Suspending"] = "Suspending";
    /** Pausing */
    KnownCapacityProvisioningState["Pausing"] = "Pausing";
    /** Resuming */
    KnownCapacityProvisioningState["Resuming"] = "Resuming";
    /** Preparing */
    KnownCapacityProvisioningState["Preparing"] = "Preparing";
    /** Scaling */
    KnownCapacityProvisioningState["Scaling"] = "Scaling";
})(KnownCapacityProvisioningState || (KnownCapacityProvisioningState = {}));
/** Known values of {@link Mode} that the service accepts. */
export var KnownMode;
(function (KnownMode) {
    /** Gen1 */
    KnownMode["Gen1"] = "Gen1";
    /** Gen2 */
    KnownMode["Gen2"] = "Gen2";
})(KnownMode || (KnownMode = {}));
/** Known values of {@link IdentityType} that the service accepts. */
export var KnownIdentityType;
(function (KnownIdentityType) {
    /** User */
    KnownIdentityType["User"] = "User";
    /** Application */
    KnownIdentityType["Application"] = "Application";
    /** ManagedIdentity */
    KnownIdentityType["ManagedIdentity"] = "ManagedIdentity";
    /** Key */
    KnownIdentityType["Key"] = "Key";
})(KnownIdentityType || (KnownIdentityType = {}));
/** Known values of {@link VCoreSkuTier} that the service accepts. */
export var KnownVCoreSkuTier;
(function (KnownVCoreSkuTier) {
    /** AutoScale */
    KnownVCoreSkuTier["AutoScale"] = "AutoScale";
})(KnownVCoreSkuTier || (KnownVCoreSkuTier = {}));
/** Known values of {@link VCoreProvisioningState} that the service accepts. */
export var KnownVCoreProvisioningState;
(function (KnownVCoreProvisioningState) {
    /** Succeeded */
    KnownVCoreProvisioningState["Succeeded"] = "Succeeded";
})(KnownVCoreProvisioningState || (KnownVCoreProvisioningState = {}));
