// @azure/arm-powerbidedicated/dist-esm/src/models/mappers.js
/*
 * Copyright (c) Microsoft Corporation.
 * Licensed under the MIT License.
 *
 * Code generated by Microsoft (R) AutoRest Code Generator.
 * Changes may cause incorrect behavior and will be lost if the code is regenerated.
 */
export const CapacitySku = {
    type: {
        name: "Composite",
        className: "CapacitySku",
        modelProperties: {
            name: {
                serializedName: "name",
                required: true,
                type: {
                    name: "String"
                }
            },
            tier: {
                serializedName: "tier",
                type: {
                    name: "String"
                }
            }
        }
    }
};
export const DedicatedCapacityMutableProperties = {
    type: {
        name: "Composite",
        className: "DedicatedCapacityMutableProperties",
        modelProperties: {
            administration: {
                serializedName: "administration",
                type: {
                    name: "Composite",
                    className: "DedicatedCapacityAdministrators"
                }
            },
            mode: {
                serializedName: "mode",
                type: {
                    name: "String"
                }
            }
        }
    }
};
export const DedicatedCapacityAdministrators = {
    type: {
        name: "Composite",
        className: "DedicatedCapacityAdministrators",
        modelProperties: {
            members: {
                serializedName: "members",
                type: {
                    name: "Sequence",
                    element: {
                        type: {
                            name: "String"
                        }
                    }
                }
            }
        }
    }
};
export const Resource = {
    type: {
        name: "Composite",
        className: "Resource",
        modelProperties: {
            id: {
                serializedName: "id",
                readOnly: true,
                type: {
                    name: "String"
                }
            },
            name: {
                serializedName: "name",
                readOnly: true,
                type: {
                    name: "String"
                }
            },
            type: {
                serializedName: "type",
                readOnly: true,
                type: {
                    name: "String"
                }
            },
            location: {
                serializedName: "location",
                required: true,
                type: {
                    name: "String"
                }
            },
            tags: {
                serializedName: "tags",
                type: {
                    name: "Dictionary",
                    value: { type: { name: "String" } }
                }
            },
            systemData: {
                serializedName: "systemData",
                type: {
                    name: "Composite",
                    className: "SystemData"
                }
            }
        }
    }
};
export const SystemData = {
    type: {
        name: "Composite",
        className: "SystemData",
        modelProperties: {
            createdBy: {
                serializedName: "createdBy",
                type: {
                    name: "String"
                }
            },
            createdByType: {
                serializedName: "createdByType",
                type: {
                    name: "String"
                }
            },
            createdAt: {
                serializedName: "createdAt",
                type: {
                    name: "DateTime"
                }
            },
            lastModifiedBy: {
                serializedName: "lastModifiedBy",
                type: {
                    name: "String"
                }
            },
            lastModifiedByType: {
                serializedName: "lastModifiedByType",
                type: {
                    name: "String"
                }
            },
            lastModifiedAt: {
                serializedName: "lastModifiedAt",
                type: {
                    name: "DateTime"
                }
            }
        }
    }
};
export const ErrorResponse = {
    type: {
        name: "Composite",
        className: "ErrorResponse",
        modelProperties: {
            error: {
                serializedName: "error",
                type: {
                    name: "Composite",
                    className: "ErrorResponseError"
                }
            }
        }
    }
};
export const ErrorResponseError = {
    type: {
        name: "Composite",
        className: "ErrorResponseError",
        modelProperties: {
            code: {
                serializedName: "code",
                type: {
                    name: "String"
                }
            },
            message: {
                serializedName: "message",
                type: {
                    name: "String"
                }
            }
        }
    }
};
export const DedicatedCapacityUpdateParameters = {
    type: {
        name: "Composite",
        className: "DedicatedCapacityUpdateParameters",
        modelProperties: {
            sku: {
                serializedName: "sku",
                type: {
                    name: "Composite",
                    className: "CapacitySku"
                }
            },
            tags: {
                serializedName: "tags",
                type: {
                    name: "Dictionary",
                    value: { type: { name: "String" } }
                }
            },
            administration: {
                serializedName: "properties.administration",
                type: {
                    name: "Composite",
                    className: "DedicatedCapacityAdministrators"
                }
            },
            mode: {
                serializedName: "properties.mode",
                type: {
                    name: "String"
                }
            }
        }
    }
};
export const DedicatedCapacities = {
    type: {
        name: "Composite",
        className: "DedicatedCapacities",
        modelProperties: {
            value: {
                serializedName: "value",
                required: true,
                type: {
                    name: "Sequence",
                    element: {
                        type: {
                            name: "Composite",
                            className: "DedicatedCapacity"
                        }
                    }
                }
            }
        }
    }
};
export const SkuEnumerationForNewResourceResult = {
    type: {
        name: "Composite",
        className: "SkuEnumerationForNewResourceResult",
        modelProperties: {
            value: {
                serializedName: "value",
                type: {
                    name: "Sequence",
                    element: {
                        type: {
                            name: "Composite",
                            className: "CapacitySku"
                        }
                    }
                }
            }
        }
    }
};
export const SkuEnumerationForExistingResourceResult = {
    type: {
        name: "Composite",
        className: "SkuEnumerationForExistingResourceResult",
        modelProperties: {
            value: {
                serializedName: "value",
                type: {
                    name: "Sequence",
                    element: {
                        type: {
                            name: "Composite",
                            className: "SkuDetailsForExistingResource"
                        }
                    }
                }
            }
        }
    }
};
export const SkuDetailsForExistingResource = {
    type: {
        name: "Composite",
        className: "SkuDetailsForExistingResource",
        modelProperties: {
            sku: {
                serializedName: "sku",
                type: {
                    name: "Composite",
                    className: "CapacitySku"
                }
            }
        }
    }
};
export const OperationListResult = {
    type: {
        name: "Composite",
        className: "OperationListResult",
        modelProperties: {
            value: {
                serializedName: "value",
                readOnly: true,
                type: {
                    name: "Sequence",
                    element: {
                        type: {
                            name: "Composite",
                            className: "Operation"
                        }
                    }
                }
            },
            nextLink: {
                serializedName: "nextLink",
                readOnly: true,
                type: {
                    name: "String"
                }
            }
        }
    }
};
export const Operation = {
    type: {
        name: "Composite",
        className: "Operation",
        modelProperties: {
            name: {
                serializedName: "name",
                readOnly: true,
                type: {
                    name: "String"
                }
            },
            display: {
                serializedName: "display",
                type: {
                    name: "Composite",
                    className: "OperationDisplay"
                }
            }
        }
    }
};
export const OperationDisplay = {
    type: {
        name: "Composite",
        className: "OperationDisplay",
        modelProperties: {
            provider: {
                serializedName: "provider",
                readOnly: true,
                type: {
                    name: "String"
                }
            },
            resource: {
                serializedName: "resource",
                readOnly: true,
                type: {
                    name: "String"
                }
            },
            operation: {
                serializedName: "operation",
                readOnly: true,
                type: {
                    name: "String"
                }
            }
        }
    }
};
export const CheckCapacityNameAvailabilityParameters = {
    type: {
        name: "Composite",
        className: "CheckCapacityNameAvailabilityParameters",
        modelProperties: {
            name: {
                constraints: {
                    Pattern: new RegExp("^[a-z][a-z0-9]*$"),
                    MaxLength: 63,
                    MinLength: 3
                },
                serializedName: "name",
                type: {
                    name: "String"
                }
            },
            type: {
                defaultValue: "Microsoft.PowerBIDedicated/capacities",
                serializedName: "type",
                type: {
                    name: "String"
                }
            }
        }
    }
};
export const CheckCapacityNameAvailabilityResult = {
    type: {
        name: "Composite",
        className: "CheckCapacityNameAvailabilityResult",
        modelProperties: {
            nameAvailable: {
                serializedName: "nameAvailable",
                type: {
                    name: "Boolean"
                }
            },
            reason: {
                serializedName: "reason",
                type: {
                    name: "String"
                }
            },
            message: {
                serializedName: "message",
                type: {
                    name: "String"
                }
            }
        }
    }
};
export const AutoScaleVCoreSku = {
    type: {
        name: "Composite",
        className: "AutoScaleVCoreSku",
        modelProperties: {
            name: {
                serializedName: "name",
                required: true,
                type: {
                    name: "String"
                }
            },
            tier: {
                serializedName: "tier",
                type: {
                    name: "String"
                }
            },
            capacity: {
                serializedName: "capacity",
                type: {
                    name: "Number"
                }
            }
        }
    }
};
export const AutoScaleVCoreMutableProperties = {
    type: {
        name: "Composite",
        className: "AutoScaleVCoreMutableProperties",
        modelProperties: {
            capacityLimit: {
                serializedName: "capacityLimit",
                type: {
                    name: "Number"
                }
            }
        }
    }
};
export const AutoScaleVCoreUpdateParameters = {
    type: {
        name: "Composite",
        className: "AutoScaleVCoreUpdateParameters",
        modelProperties: {
            sku: {
                serializedName: "sku",
                type: {
                    name: "Composite",
                    className: "AutoScaleVCoreSku"
                }
            },
            tags: {
                serializedName: "tags",
                type: {
                    name: "Dictionary",
                    value: { type: { name: "String" } }
                }
            },
            capacityLimit: {
                serializedName: "properties.capacityLimit",
                type: {
                    name: "Number"
                }
            }
        }
    }
};
export const AutoScaleVCoreListResult = {
    type: {
        name: "Composite",
        className: "AutoScaleVCoreListResult",
        modelProperties: {
            value: {
                serializedName: "value",
                required: true,
                type: {
                    name: "Sequence",
                    element: {
                        type: {
                            name: "Composite",
                            className: "AutoScaleVCore"
                        }
                    }
                }
            }
        }
    }
};
export const DedicatedCapacityProperties = {
    type: {
        name: "Composite",
        className: "DedicatedCapacityProperties",
        modelProperties: Object.assign(Object.assign({}, DedicatedCapacityMutableProperties.type.modelProperties), { state: {
                serializedName: "state",
                readOnly: true,
                type: {
                    name: "String"
                }
            }, provisioningState: {
                serializedName: "provisioningState",
                readOnly: true,
                type: {
                    name: "String"
                }
            } })
    }
};
export const DedicatedCapacity = {
    type: {
        name: "Composite",
        className: "DedicatedCapacity",
        modelProperties: Object.assign(Object.assign({}, Resource.type.modelProperties), { sku: {
                serializedName: "sku",
                type: {
                    name: "Composite",
                    className: "CapacitySku"
                }
            }, administration: {
                serializedName: "properties.administration",
                type: {
                    name: "Composite",
                    className: "DedicatedCapacityAdministrators"
                }
            }, mode: {
                serializedName: "properties.mode",
                type: {
                    name: "String"
                }
            }, state: {
                serializedName: "properties.state",
                readOnly: true,
                type: {
                    name: "String"
                }
            }, provisioningState: {
                serializedName: "properties.provisioningState",
                readOnly: true,
                type: {
                    name: "String"
                }
            } })
    }
};
export const AutoScaleVCore = {
    type: {
        name: "Composite",
        className: "AutoScaleVCore",
        modelProperties: Object.assign(Object.assign({}, Resource.type.modelProperties), { sku: {
                serializedName: "sku",
                type: {
                    name: "Composite",
                    className: "AutoScaleVCoreSku"
                }
            }, capacityLimit: {
                serializedName: "properties.capacityLimit",
                type: {
                    name: "Number"
                }
            }, capacityObjectId: {
                serializedName: "properties.capacityObjectId",
                type: {
                    name: "String"
                }
            }, provisioningState: {
                serializedName: "properties.provisioningState",
                readOnly: true,
                type: {
                    name: "String"
                }
            } })
    }
};
export const AutoScaleVCoreProperties = {
    type: {
        name: "Composite",
        className: "AutoScaleVCoreProperties",
        modelProperties: Object.assign(Object.assign({}, AutoScaleVCoreMutableProperties.type.modelProperties), { capacityObjectId: {
                serializedName: "capacityObjectId",
                type: {
                    name: "String"
                }
            }, provisioningState: {
                serializedName: "provisioningState",
                readOnly: true,
                type: {
                    name: "String"
                }
            } })
    }
};
