// @azure/arm-machinelearningcompute/esm/operations/machineLearningCompute.js
/*
 * Copyright (c) Microsoft Corporation.
 * Licensed under the MIT License.
 *
 * Code generated by Microsoft (R) AutoRest Code Generator.
 * Changes may cause incorrect behavior and will be lost if the code is
 * regenerated.
 */
import * as msRest from "@azure/ms-rest-js";
import * as Mappers from "../models/machineLearningComputeMappers";
import * as Parameters from "../models/parameters";
/** Class representing a MachineLearningCompute. */
var MachineLearningCompute = /** @class */ (function () {
    /**
     * Create a MachineLearningCompute.
     * @param {MachineLearningComputeManagementClientContext} client Reference to the service client.
     */
    function MachineLearningCompute(client) {
        this.client = client;
    }
    MachineLearningCompute.prototype.listAvailableOperations = function (options, callback) {
        return this.client.sendOperationRequest({
            options: options
        }, listAvailableOperationsOperationSpec, callback);
    };
    return MachineLearningCompute;
}());
export { MachineLearningCompute };
// Operation Specifications
var serializer = new msRest.Serializer(Mappers);
var listAvailableOperationsOperationSpec = {
    httpMethod: "GET",
    path: "providers/Microsoft.MachineLearningCompute/operations",
    queryParameters: [
        Parameters.apiVersion
    ],
    headerParameters: [
        Parameters.acceptLanguage
    ],
    responses: {
        200: {
            bodyMapper: Mappers.AvailableOperations
        },
        default: {
            bodyMapper: Mappers.CloudError
        }
    },
    serializer: serializer
};
