// @azure/arm-portal/dist-esm/src/models/mappers.js
/*
 * Copyright (c) Microsoft Corporation.
 * Licensed under the MIT License.
 *
 * Code generated by Microsoft (R) AutoRest Code Generator.
 * Changes may cause incorrect behavior and will be lost if the code is regenerated.
 */
export const ResourceProviderOperationList = {
    type: {
        name: "Composite",
        className: "ResourceProviderOperationList",
        modelProperties: {
            value: {
                serializedName: "value",
                type: {
                    name: "Sequence",
                    element: {
                        type: {
                            name: "Composite",
                            className: "ResourceProviderOperation"
                        }
                    }
                }
            },
            nextLink: {
                serializedName: "nextLink",
                type: {
                    name: "String"
                }
            }
        }
    }
};
export const ResourceProviderOperation = {
    type: {
        name: "Composite",
        className: "ResourceProviderOperation",
        modelProperties: {
            name: {
                serializedName: "name",
                type: {
                    name: "String"
                }
            },
            isDataAction: {
                serializedName: "isDataAction",
                type: {
                    name: "String"
                }
            },
            display: {
                serializedName: "display",
                type: {
                    name: "Composite",
                    className: "ResourceProviderOperationDisplay"
                }
            }
        }
    }
};
export const ResourceProviderOperationDisplay = {
    type: {
        name: "Composite",
        className: "ResourceProviderOperationDisplay",
        modelProperties: {
            provider: {
                serializedName: "provider",
                type: {
                    name: "String"
                }
            },
            resource: {
                serializedName: "resource",
                type: {
                    name: "String"
                }
            },
            operation: {
                serializedName: "operation",
                type: {
                    name: "String"
                }
            },
            description: {
                serializedName: "description",
                type: {
                    name: "String"
                }
            }
        }
    }
};
export const ErrorResponse = {
    type: {
        name: "Composite",
        className: "ErrorResponse",
        modelProperties: {
            error: {
                serializedName: "error",
                type: {
                    name: "Composite",
                    className: "ErrorDefinition"
                }
            }
        }
    }
};
export const ErrorDefinition = {
    type: {
        name: "Composite",
        className: "ErrorDefinition",
        modelProperties: {
            code: {
                serializedName: "code",
                readOnly: true,
                type: {
                    name: "Number"
                }
            },
            message: {
                serializedName: "message",
                readOnly: true,
                type: {
                    name: "String"
                }
            },
            details: {
                serializedName: "details",
                readOnly: true,
                type: {
                    name: "Sequence",
                    element: {
                        type: {
                            name: "Composite",
                            className: "ErrorDefinition"
                        }
                    }
                }
            }
        }
    }
};
export const Dashboard = {
    type: {
        name: "Composite",
        className: "Dashboard",
        modelProperties: {
            id: {
                serializedName: "id",
                readOnly: true,
                type: {
                    name: "String"
                }
            },
            name: {
                serializedName: "name",
                readOnly: true,
                type: {
                    name: "String"
                }
            },
            type: {
                serializedName: "type",
                readOnly: true,
                type: {
                    name: "String"
                }
            },
            location: {
                serializedName: "location",
                required: true,
                type: {
                    name: "String"
                }
            },
            tags: {
                serializedName: "tags",
                type: {
                    name: "Dictionary",
                    value: { type: { name: "String" } }
                }
            },
            lenses: {
                serializedName: "properties.lenses",
                type: {
                    name: "Sequence",
                    element: {
                        type: {
                            name: "Composite",
                            className: "DashboardLens"
                        }
                    }
                }
            },
            metadata: {
                serializedName: "properties.metadata",
                type: {
                    name: "Dictionary",
                    value: {
                        type: { name: "Dictionary", value: { type: { name: "any" } } }
                    }
                }
            }
        }
    }
};
export const DashboardLens = {
    type: {
        name: "Composite",
        className: "DashboardLens",
        modelProperties: {
            order: {
                serializedName: "order",
                required: true,
                type: {
                    name: "Number"
                }
            },
            parts: {
                serializedName: "parts",
                required: true,
                type: {
                    name: "Sequence",
                    element: {
                        type: {
                            name: "Composite",
                            className: "DashboardParts"
                        }
                    }
                }
            },
            metadata: {
                serializedName: "metadata",
                type: {
                    name: "Dictionary",
                    value: {
                        type: { name: "Dictionary", value: { type: { name: "any" } } }
                    }
                }
            }
        }
    }
};
export const DashboardParts = {
    type: {
        name: "Composite",
        className: "DashboardParts",
        modelProperties: {
            position: {
                serializedName: "position",
                type: {
                    name: "Composite",
                    className: "DashboardPartsPosition"
                }
            },
            metadata: {
                serializedName: "metadata",
                type: {
                    name: "Composite",
                    className: "DashboardPartMetadata"
                }
            }
        }
    }
};
export const DashboardPartsPosition = {
    type: {
        name: "Composite",
        className: "DashboardPartsPosition",
        modelProperties: {
            x: {
                serializedName: "x",
                required: true,
                type: {
                    name: "Number"
                }
            },
            y: {
                serializedName: "y",
                required: true,
                type: {
                    name: "Number"
                }
            },
            rowSpan: {
                serializedName: "rowSpan",
                required: true,
                type: {
                    name: "Number"
                }
            },
            colSpan: {
                serializedName: "colSpan",
                required: true,
                type: {
                    name: "Number"
                }
            },
            metadata: {
                serializedName: "metadata",
                type: {
                    name: "Dictionary",
                    value: {
                        type: { name: "Dictionary", value: { type: { name: "any" } } }
                    }
                }
            }
        }
    }
};
export const DashboardPartMetadata = {
    serializedName: "DashboardPartMetadata",
    type: {
        name: "Composite",
        className: "DashboardPartMetadata",
        uberParent: "DashboardPartMetadata",
        additionalProperties: { type: { name: "Object" } },
        polymorphicDiscriminator: {
            serializedName: "type",
            clientName: "type"
        },
        modelProperties: {
            type: {
                serializedName: "type",
                required: true,
                type: {
                    name: "String"
                }
            }
        }
    }
};
export const PatchableDashboard = {
    type: {
        name: "Composite",
        className: "PatchableDashboard",
        modelProperties: {
            tags: {
                serializedName: "tags",
                type: {
                    name: "Dictionary",
                    value: { type: { name: "String" } }
                }
            },
            lenses: {
                serializedName: "properties.lenses",
                type: {
                    name: "Sequence",
                    element: {
                        type: {
                            name: "Composite",
                            className: "DashboardLens"
                        }
                    }
                }
            },
            metadata: {
                serializedName: "properties.metadata",
                type: {
                    name: "Dictionary",
                    value: {
                        type: { name: "Dictionary", value: { type: { name: "any" } } }
                    }
                }
            }
        }
    }
};
export const DashboardListResult = {
    type: {
        name: "Composite",
        className: "DashboardListResult",
        modelProperties: {
            value: {
                serializedName: "value",
                type: {
                    name: "Sequence",
                    element: {
                        type: {
                            name: "Composite",
                            className: "Dashboard"
                        }
                    }
                }
            },
            nextLink: {
                serializedName: "nextLink",
                type: {
                    name: "String"
                }
            }
        }
    }
};
export const ConfigurationList = {
    type: {
        name: "Composite",
        className: "ConfigurationList",
        modelProperties: {
            value: {
                serializedName: "value",
                type: {
                    name: "Sequence",
                    element: {
                        type: {
                            name: "Composite",
                            className: "Configuration"
                        }
                    }
                }
            },
            nextLink: {
                serializedName: "nextLink",
                type: {
                    name: "String"
                }
            }
        }
    }
};
export const Resource = {
    type: {
        name: "Composite",
        className: "Resource",
        modelProperties: {
            id: {
                serializedName: "id",
                readOnly: true,
                type: {
                    name: "String"
                }
            },
            name: {
                serializedName: "name",
                readOnly: true,
                type: {
                    name: "String"
                }
            },
            type: {
                serializedName: "type",
                readOnly: true,
                type: {
                    name: "String"
                }
            }
        }
    }
};
export const ViolationsList = {
    type: {
        name: "Composite",
        className: "ViolationsList",
        modelProperties: {
            value: {
                serializedName: "value",
                type: {
                    name: "Sequence",
                    element: {
                        type: {
                            name: "Composite",
                            className: "Violation"
                        }
                    }
                }
            },
            nextLink: {
                serializedName: "nextLink",
                type: {
                    name: "String"
                }
            }
        }
    }
};
export const Violation = {
    type: {
        name: "Composite",
        className: "Violation",
        modelProperties: {
            id: {
                serializedName: "id",
                readOnly: true,
                type: {
                    name: "String"
                }
            },
            userId: {
                serializedName: "userId",
                readOnly: true,
                type: {
                    name: "String"
                }
            },
            errorMessage: {
                serializedName: "errorMessage",
                readOnly: true,
                type: {
                    name: "String"
                }
            }
        }
    }
};
export const MarkdownPartMetadataSettings = {
    type: {
        name: "Composite",
        className: "MarkdownPartMetadataSettings",
        modelProperties: {
            content: {
                serializedName: "content",
                type: {
                    name: "Composite",
                    className: "MarkdownPartMetadataSettingsContent"
                }
            }
        }
    }
};
export const MarkdownPartMetadataSettingsContent = {
    type: {
        name: "Composite",
        className: "MarkdownPartMetadataSettingsContent",
        modelProperties: {
            settings: {
                serializedName: "settings",
                type: {
                    name: "Composite",
                    className: "MarkdownPartMetadataSettingsContentSettings"
                }
            }
        }
    }
};
export const MarkdownPartMetadataSettingsContentSettings = {
    type: {
        name: "Composite",
        className: "MarkdownPartMetadataSettingsContentSettings",
        modelProperties: {
            content: {
                serializedName: "content",
                type: {
                    name: "String"
                }
            },
            title: {
                serializedName: "title",
                type: {
                    name: "String"
                }
            },
            subtitle: {
                serializedName: "subtitle",
                type: {
                    name: "String"
                }
            },
            markdownSource: {
                serializedName: "markdownSource",
                type: {
                    name: "Number"
                }
            },
            markdownUri: {
                serializedName: "markdownUri",
                type: {
                    name: "String"
                }
            }
        }
    }
};
export const MarkdownPartMetadata = {
    serializedName: "Extension/HubsExtension/PartType/MarkdownPart",
    type: {
        name: "Composite",
        className: "MarkdownPartMetadata",
        uberParent: "DashboardPartMetadata",
        additionalProperties: { type: { name: "Object" } },
        polymorphicDiscriminator: DashboardPartMetadata.type.polymorphicDiscriminator,
        modelProperties: Object.assign(Object.assign({}, DashboardPartMetadata.type.modelProperties), { inputs: {
                serializedName: "inputs",
                type: {
                    name: "Sequence",
                    element: {
                        type: {
                            name: "Dictionary",
                            value: { type: { name: "any" } }
                        }
                    }
                }
            }, settings: {
                serializedName: "settings",
                type: {
                    name: "Composite",
                    className: "MarkdownPartMetadataSettings"
                }
            } })
    }
};
export const ProxyResource = {
    type: {
        name: "Composite",
        className: "ProxyResource",
        modelProperties: Object.assign({}, Resource.type.modelProperties)
    }
};
export const Configuration = {
    type: {
        name: "Composite",
        className: "Configuration",
        modelProperties: Object.assign(Object.assign({}, ProxyResource.type.modelProperties), { enforcePrivateMarkdownStorage: {
                serializedName: "properties.enforcePrivateMarkdownStorage",
                type: {
                    name: "Boolean"
                }
            } })
    }
};
export let discriminators = {
    DashboardPartMetadata: DashboardPartMetadata,
    "DashboardPartMetadata.Extension/HubsExtension/PartType/MarkdownPart": MarkdownPartMetadata
};
