// @azure/arm-streamanalytics/dist-esm/src/operations/subscriptions.js
/*
 * Copyright (c) Microsoft Corporation.
 * Licensed under the MIT License.
 *
 * Code generated by Microsoft (R) AutoRest Code Generator.
 * Changes may cause incorrect behavior and will be lost if the code is regenerated.
 */
import * as coreClient from "@azure/core-client";
import * as Mappers from "../models/mappers";
import * as Parameters from "../models/parameters";
/** Class containing Subscriptions operations. */
export class SubscriptionsImpl {
    /**
     * Initialize a new instance of the class Subscriptions class.
     * @param client Reference to the service client
     */
    constructor(client) {
        this.client = client;
    }
    /**
     * Retrieves the subscription's current quota information in a particular region.
     * @param location The region in which to retrieve the subscription's quota information. You can find
     *                 out which regions Azure Stream Analytics is supported in here:
     *                 https://azure.microsoft.com/en-us/regions/
     * @param options The options parameters.
     */
    listQuotas(location, options) {
        return this.client.sendOperationRequest({ location, options }, listQuotasOperationSpec);
    }
}
// Operation Specifications
const serializer = coreClient.createSerializer(Mappers, /* isXml */ false);
const listQuotasOperationSpec = {
    path: "/subscriptions/{subscriptionId}/providers/Microsoft.StreamAnalytics/locations/{location}/quotas",
    httpMethod: "GET",
    responses: {
        200: {
            bodyMapper: Mappers.SubscriptionQuotasListResult
        },
        default: {
            bodyMapper: Mappers.ErrorModel
        }
    },
    queryParameters: [Parameters.apiVersion],
    urlParameters: [
        Parameters.$host,
        Parameters.subscriptionId,
        Parameters.location
    ],
    headerParameters: [Parameters.accept],
    serializer
};
