// @azure/arm-mediaservices/dist-esm/src/operations/mediaServicesOperationStatuses.js
/*
 * Copyright (c) Microsoft Corporation.
 * Licensed under the MIT License.
 *
 * Code generated by Microsoft (R) AutoRest Code Generator.
 * Changes may cause incorrect behavior and will be lost if the code is regenerated.
 */
import * as coreClient from "@azure/core-client";
import * as Mappers from "../models/mappers";
import * as Parameters from "../models/parameters";
/** Class containing MediaServicesOperationStatuses operations. */
export class MediaServicesOperationStatusesImpl {
    /**
     * Initialize a new instance of the class MediaServicesOperationStatuses class.
     * @param client Reference to the service client
     */
    constructor(client) {
        this.client = client;
    }
    /**
     * Get media service operation status.
     * @param locationName Location name.
     * @param operationId Operation ID.
     * @param options The options parameters.
     */
    get(locationName, operationId, options) {
        return this.client.sendOperationRequest({ locationName, operationId, options }, getOperationSpec);
    }
}
// Operation Specifications
const serializer = coreClient.createSerializer(Mappers, /* isXml */ false);
const getOperationSpec = {
    path: "/subscriptions/{subscriptionId}/providers/Microsoft.Media/locations/{locationName}/mediaServicesOperationStatuses/{operationId}",
    httpMethod: "GET",
    responses: {
        200: {
            bodyMapper: Mappers.MediaServiceOperationStatus
        },
        default: {
            bodyMapper: Mappers.ErrorResponse
        }
    },
    queryParameters: [Parameters.apiVersion1],
    urlParameters: [
        Parameters.$host,
        Parameters.subscriptionId,
        Parameters.locationName,
        Parameters.operationId
    ],
    headerParameters: [Parameters.accept],
    serializer
};
