// @azure/arm-billing/dist-esm/src/operations/address.js
/*
 * Copyright (c) Microsoft Corporation.
 * Licensed under the MIT License.
 *
 * Code generated by Microsoft (R) AutoRest Code Generator.
 * Changes may cause incorrect behavior and will be lost if the code is regenerated.
 */
import * as coreClient from "@azure/core-client";
import * as Mappers from "../models/mappers";
import * as Parameters from "../models/parameters";
/** Class containing Address operations. */
export class AddressImpl {
    /**
     * Initialize a new instance of the class Address class.
     * @param client Reference to the service client
     */
    constructor(client) {
        this.client = client;
    }
    /**
     * Validates an address. Use the operation to validate an address before using it as soldTo or a billTo
     * address.
     * @param address Address details.
     * @param options The options parameters.
     */
    validate(address, options) {
        return this.client.sendOperationRequest({ address, options }, validateOperationSpec);
    }
}
// Operation Specifications
const serializer = coreClient.createSerializer(Mappers, /* isXml */ false);
const validateOperationSpec = {
    path: "/providers/Microsoft.Billing/validateAddress",
    httpMethod: "POST",
    responses: {
        200: {
            bodyMapper: Mappers.ValidateAddressResponse
        },
        default: {
            bodyMapper: Mappers.ErrorResponse
        }
    },
    requestBody: Parameters.address,
    queryParameters: [Parameters.apiVersion],
    urlParameters: [Parameters.$host],
    headerParameters: [Parameters.accept, Parameters.contentType],
    mediaType: "json",
    serializer
};
