// @azure/arm-billing/dist-esm/src/operations/billingPropertyOperations.js
/*
 * Copyright (c) Microsoft Corporation.
 * Licensed under the MIT License.
 *
 * Code generated by Microsoft (R) AutoRest Code Generator.
 * Changes may cause incorrect behavior and will be lost if the code is regenerated.
 */
import * as coreClient from "@azure/core-client";
import * as Mappers from "../models/mappers";
import * as Parameters from "../models/parameters";
/** Class containing BillingPropertyOperations operations. */
export class BillingPropertyOperationsImpl {
    /**
     * Initialize a new instance of the class BillingPropertyOperations class.
     * @param client Reference to the service client
     */
    constructor(client) {
        this.client = client;
    }
    /**
     * Get the billing properties for a subscription. This operation is not supported for billing accounts
     * with agreement type Enterprise Agreement.
     * @param options The options parameters.
     */
    get(options) {
        return this.client.sendOperationRequest({ options }, getOperationSpec);
    }
    /**
     * Updates the billing property of a subscription. Currently, cost center can be updated. The operation
     * is supported only for billing accounts with agreement type Microsoft Customer Agreement.
     * @param parameters Request parameters that are provided to the update billing property operation.
     * @param options The options parameters.
     */
    update(parameters, options) {
        return this.client.sendOperationRequest({ parameters, options }, updateOperationSpec);
    }
}
// Operation Specifications
const serializer = coreClient.createSerializer(Mappers, /* isXml */ false);
const getOperationSpec = {
    path: "/subscriptions/{subscriptionId}/providers/Microsoft.Billing/billingProperty/default",
    httpMethod: "GET",
    responses: {
        200: {
            bodyMapper: Mappers.BillingProperty
        },
        default: {
            bodyMapper: Mappers.ErrorResponse
        }
    },
    queryParameters: [Parameters.apiVersion],
    urlParameters: [Parameters.$host, Parameters.subscriptionId],
    headerParameters: [Parameters.accept],
    serializer
};
const updateOperationSpec = {
    path: "/subscriptions/{subscriptionId}/providers/Microsoft.Billing/billingProperty/default",
    httpMethod: "PATCH",
    responses: {
        200: {
            bodyMapper: Mappers.BillingProperty
        },
        default: {
            bodyMapper: Mappers.ErrorResponse
        }
    },
    requestBody: Parameters.parameters10,
    queryParameters: [Parameters.apiVersion],
    urlParameters: [Parameters.$host, Parameters.subscriptionId],
    headerParameters: [Parameters.accept, Parameters.contentType],
    mediaType: "json",
    serializer
};
