// @azure/arm-customerinsights/dist-esm/src/operations/authorizationPolicies.js
/*
 * Copyright (c) Microsoft Corporation.
 * Licensed under the MIT License.
 *
 * Code generated by Microsoft (R) AutoRest Code Generator.
 * Changes may cause incorrect behavior and will be lost if the code is regenerated.
 */
import { __asyncDelegator, __asyncGenerator, __asyncValues, __await } from "tslib";
import { setContinuationToken } from "../pagingHelper";
import * as coreClient from "@azure/core-client";
import * as Mappers from "../models/mappers";
import * as Parameters from "../models/parameters";
/// <reference lib="esnext.asynciterable" />
/** Class containing AuthorizationPolicies operations. */
export class AuthorizationPoliciesImpl {
    /**
     * Initialize a new instance of the class AuthorizationPolicies class.
     * @param client Reference to the service client
     */
    constructor(client) {
        this.client = client;
    }
    /**
     * Gets all the authorization policies in a specified hub.
     * @param resourceGroupName The name of the resource group.
     * @param hubName The name of the hub.
     * @param options The options parameters.
     */
    listByHub(resourceGroupName, hubName, options) {
        const iter = this.listByHubPagingAll(resourceGroupName, hubName, options);
        return {
            next() {
                return iter.next();
            },
            [Symbol.asyncIterator]() {
                return this;
            },
            byPage: (settings) => {
                if (settings === null || settings === void 0 ? void 0 : settings.maxPageSize) {
                    throw new Error("maxPageSize is not supported by this operation.");
                }
                return this.listByHubPagingPage(resourceGroupName, hubName, options, settings);
            }
        };
    }
    listByHubPagingPage(resourceGroupName, hubName, options, settings) {
        return __asyncGenerator(this, arguments, function* listByHubPagingPage_1() {
            let result;
            let continuationToken = settings === null || settings === void 0 ? void 0 : settings.continuationToken;
            if (!continuationToken) {
                result = yield __await(this._listByHub(resourceGroupName, hubName, options));
                let page = result.value || [];
                continuationToken = result.nextLink;
                setContinuationToken(page, continuationToken);
                yield yield __await(page);
            }
            while (continuationToken) {
                result = yield __await(this._listByHubNext(resourceGroupName, hubName, continuationToken, options));
                continuationToken = result.nextLink;
                let page = result.value || [];
                setContinuationToken(page, continuationToken);
                yield yield __await(page);
            }
        });
    }
    listByHubPagingAll(resourceGroupName, hubName, options) {
        return __asyncGenerator(this, arguments, function* listByHubPagingAll_1() {
            var e_1, _a;
            try {
                for (var _b = __asyncValues(this.listByHubPagingPage(resourceGroupName, hubName, options)), _c; _c = yield __await(_b.next()), !_c.done;) {
                    const page = _c.value;
                    yield __await(yield* __asyncDelegator(__asyncValues(page)));
                }
            }
            catch (e_1_1) { e_1 = { error: e_1_1 }; }
            finally {
                try {
                    if (_c && !_c.done && (_a = _b.return)) yield __await(_a.call(_b));
                }
                finally { if (e_1) throw e_1.error; }
            }
        });
    }
    /**
     * Creates an authorization policy or updates an existing authorization policy.
     * @param resourceGroupName The name of the resource group.
     * @param hubName The name of the hub.
     * @param authorizationPolicyName The name of the policy.
     * @param parameters Parameters supplied to the CreateOrUpdate authorization policy operation.
     * @param options The options parameters.
     */
    createOrUpdate(resourceGroupName, hubName, authorizationPolicyName, parameters, options) {
        return this.client.sendOperationRequest({
            resourceGroupName,
            hubName,
            authorizationPolicyName,
            parameters,
            options
        }, createOrUpdateOperationSpec);
    }
    /**
     * Gets an authorization policy in the hub.
     * @param resourceGroupName The name of the resource group.
     * @param hubName The name of the hub.
     * @param authorizationPolicyName The name of the policy.
     * @param options The options parameters.
     */
    get(resourceGroupName, hubName, authorizationPolicyName, options) {
        return this.client.sendOperationRequest({ resourceGroupName, hubName, authorizationPolicyName, options }, getOperationSpec);
    }
    /**
     * Gets all the authorization policies in a specified hub.
     * @param resourceGroupName The name of the resource group.
     * @param hubName The name of the hub.
     * @param options The options parameters.
     */
    _listByHub(resourceGroupName, hubName, options) {
        return this.client.sendOperationRequest({ resourceGroupName, hubName, options }, listByHubOperationSpec);
    }
    /**
     * Regenerates the primary policy key of the specified authorization policy.
     * @param resourceGroupName The name of the resource group.
     * @param hubName The name of the hub.
     * @param authorizationPolicyName The name of the policy.
     * @param options The options parameters.
     */
    regeneratePrimaryKey(resourceGroupName, hubName, authorizationPolicyName, options) {
        return this.client.sendOperationRequest({ resourceGroupName, hubName, authorizationPolicyName, options }, regeneratePrimaryKeyOperationSpec);
    }
    /**
     * Regenerates the secondary policy key of the specified authorization policy.
     * @param resourceGroupName The name of the resource group.
     * @param hubName The name of the hub.
     * @param authorizationPolicyName The name of the policy.
     * @param options The options parameters.
     */
    regenerateSecondaryKey(resourceGroupName, hubName, authorizationPolicyName, options) {
        return this.client.sendOperationRequest({ resourceGroupName, hubName, authorizationPolicyName, options }, regenerateSecondaryKeyOperationSpec);
    }
    /**
     * ListByHubNext
     * @param resourceGroupName The name of the resource group.
     * @param hubName The name of the hub.
     * @param nextLink The nextLink from the previous successful call to the ListByHub method.
     * @param options The options parameters.
     */
    _listByHubNext(resourceGroupName, hubName, nextLink, options) {
        return this.client.sendOperationRequest({ resourceGroupName, hubName, nextLink, options }, listByHubNextOperationSpec);
    }
}
// Operation Specifications
const serializer = coreClient.createSerializer(Mappers, /* isXml */ false);
const createOrUpdateOperationSpec = {
    path: "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.CustomerInsights/hubs/{hubName}/authorizationPolicies/{authorizationPolicyName}",
    httpMethod: "PUT",
    responses: {
        200: {
            bodyMapper: Mappers.AuthorizationPolicyResourceFormat
        },
        201: {
            bodyMapper: Mappers.AuthorizationPolicyResourceFormat
        }
    },
    requestBody: Parameters.parameters5,
    queryParameters: [Parameters.apiVersion],
    urlParameters: [
        Parameters.$host,
        Parameters.resourceGroupName,
        Parameters.subscriptionId,
        Parameters.hubName1,
        Parameters.authorizationPolicyName
    ],
    headerParameters: [Parameters.accept, Parameters.contentType],
    mediaType: "json",
    serializer
};
const getOperationSpec = {
    path: "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.CustomerInsights/hubs/{hubName}/authorizationPolicies/{authorizationPolicyName}",
    httpMethod: "GET",
    responses: {
        200: {
            bodyMapper: Mappers.AuthorizationPolicyResourceFormat
        }
    },
    queryParameters: [Parameters.apiVersion],
    urlParameters: [
        Parameters.$host,
        Parameters.resourceGroupName,
        Parameters.subscriptionId,
        Parameters.hubName1,
        Parameters.authorizationPolicyName1
    ],
    headerParameters: [Parameters.accept],
    serializer
};
const listByHubOperationSpec = {
    path: "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.CustomerInsights/hubs/{hubName}/authorizationPolicies",
    httpMethod: "GET",
    responses: {
        200: {
            bodyMapper: Mappers.AuthorizationPolicyListResult
        }
    },
    queryParameters: [Parameters.apiVersion],
    urlParameters: [
        Parameters.$host,
        Parameters.resourceGroupName,
        Parameters.subscriptionId,
        Parameters.hubName1
    ],
    headerParameters: [Parameters.accept],
    serializer
};
const regeneratePrimaryKeyOperationSpec = {
    path: "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.CustomerInsights/hubs/{hubName}/authorizationPolicies/{authorizationPolicyName}/regeneratePrimaryKey",
    httpMethod: "POST",
    responses: {
        200: {
            bodyMapper: Mappers.AuthorizationPolicy
        }
    },
    queryParameters: [Parameters.apiVersion],
    urlParameters: [
        Parameters.$host,
        Parameters.resourceGroupName,
        Parameters.subscriptionId,
        Parameters.hubName1,
        Parameters.authorizationPolicyName1
    ],
    headerParameters: [Parameters.accept],
    serializer
};
const regenerateSecondaryKeyOperationSpec = {
    path: "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.CustomerInsights/hubs/{hubName}/authorizationPolicies/{authorizationPolicyName}/regenerateSecondaryKey",
    httpMethod: "POST",
    responses: {
        200: {
            bodyMapper: Mappers.AuthorizationPolicy
        }
    },
    queryParameters: [Parameters.apiVersion],
    urlParameters: [
        Parameters.$host,
        Parameters.resourceGroupName,
        Parameters.subscriptionId,
        Parameters.hubName1,
        Parameters.authorizationPolicyName1
    ],
    headerParameters: [Parameters.accept],
    serializer
};
const listByHubNextOperationSpec = {
    path: "{nextLink}",
    httpMethod: "GET",
    responses: {
        200: {
            bodyMapper: Mappers.AuthorizationPolicyListResult
        }
    },
    queryParameters: [Parameters.apiVersion],
    urlParameters: [
        Parameters.$host,
        Parameters.nextLink,
        Parameters.resourceGroupName,
        Parameters.subscriptionId,
        Parameters.hubName1
    ],
    headerParameters: [Parameters.accept],
    serializer
};
