// @azure/arm-customerinsights/dist-esm/src/operations/views.js
/*
 * Copyright (c) Microsoft Corporation.
 * Licensed under the MIT License.
 *
 * Code generated by Microsoft (R) AutoRest Code Generator.
 * Changes may cause incorrect behavior and will be lost if the code is regenerated.
 */
import { __asyncDelegator, __asyncGenerator, __asyncValues, __await } from "tslib";
import { setContinuationToken } from "../pagingHelper";
import * as coreClient from "@azure/core-client";
import * as Mappers from "../models/mappers";
import * as Parameters from "../models/parameters";
/// <reference lib="esnext.asynciterable" />
/** Class containing Views operations. */
export class ViewsImpl {
    /**
     * Initialize a new instance of the class Views class.
     * @param client Reference to the service client
     */
    constructor(client) {
        this.client = client;
    }
    /**
     * Gets all available views for given user in the specified hub.
     * @param resourceGroupName The name of the resource group.
     * @param hubName The name of the hub.
     * @param userId The user ID. Use * to retrieve hub level views.
     * @param options The options parameters.
     */
    listByHub(resourceGroupName, hubName, userId, options) {
        const iter = this.listByHubPagingAll(resourceGroupName, hubName, userId, options);
        return {
            next() {
                return iter.next();
            },
            [Symbol.asyncIterator]() {
                return this;
            },
            byPage: (settings) => {
                if (settings === null || settings === void 0 ? void 0 : settings.maxPageSize) {
                    throw new Error("maxPageSize is not supported by this operation.");
                }
                return this.listByHubPagingPage(resourceGroupName, hubName, userId, options, settings);
            }
        };
    }
    listByHubPagingPage(resourceGroupName, hubName, userId, options, settings) {
        return __asyncGenerator(this, arguments, function* listByHubPagingPage_1() {
            let result;
            let continuationToken = settings === null || settings === void 0 ? void 0 : settings.continuationToken;
            if (!continuationToken) {
                result = yield __await(this._listByHub(resourceGroupName, hubName, userId, options));
                let page = result.value || [];
                continuationToken = result.nextLink;
                setContinuationToken(page, continuationToken);
                yield yield __await(page);
            }
            while (continuationToken) {
                result = yield __await(this._listByHubNext(resourceGroupName, hubName, userId, continuationToken, options));
                continuationToken = result.nextLink;
                let page = result.value || [];
                setContinuationToken(page, continuationToken);
                yield yield __await(page);
            }
        });
    }
    listByHubPagingAll(resourceGroupName, hubName, userId, options) {
        return __asyncGenerator(this, arguments, function* listByHubPagingAll_1() {
            var e_1, _a;
            try {
                for (var _b = __asyncValues(this.listByHubPagingPage(resourceGroupName, hubName, userId, options)), _c; _c = yield __await(_b.next()), !_c.done;) {
                    const page = _c.value;
                    yield __await(yield* __asyncDelegator(__asyncValues(page)));
                }
            }
            catch (e_1_1) { e_1 = { error: e_1_1 }; }
            finally {
                try {
                    if (_c && !_c.done && (_a = _b.return)) yield __await(_a.call(_b));
                }
                finally { if (e_1) throw e_1.error; }
            }
        });
    }
    /**
     * Gets all available views for given user in the specified hub.
     * @param resourceGroupName The name of the resource group.
     * @param hubName The name of the hub.
     * @param userId The user ID. Use * to retrieve hub level views.
     * @param options The options parameters.
     */
    _listByHub(resourceGroupName, hubName, userId, options) {
        return this.client.sendOperationRequest({ resourceGroupName, hubName, userId, options }, listByHubOperationSpec);
    }
    /**
     * Creates a view or updates an existing view in the hub.
     * @param resourceGroupName The name of the resource group.
     * @param hubName The name of the hub.
     * @param viewName The name of the view.
     * @param parameters Parameters supplied to the CreateOrUpdate View operation.
     * @param options The options parameters.
     */
    createOrUpdate(resourceGroupName, hubName, viewName, parameters, options) {
        return this.client.sendOperationRequest({ resourceGroupName, hubName, viewName, parameters, options }, createOrUpdateOperationSpec);
    }
    /**
     * Gets a view in the hub.
     * @param resourceGroupName The name of the resource group.
     * @param hubName The name of the hub.
     * @param viewName The name of the view.
     * @param userId The user ID. Use * to retrieve hub level view.
     * @param options The options parameters.
     */
    get(resourceGroupName, hubName, viewName, userId, options) {
        return this.client.sendOperationRequest({ resourceGroupName, hubName, viewName, userId, options }, getOperationSpec);
    }
    /**
     * Deletes a view in the specified hub.
     * @param resourceGroupName The name of the resource group.
     * @param hubName The name of the hub.
     * @param viewName The name of the view.
     * @param userId The user ID. Use * to retrieve hub level view.
     * @param options The options parameters.
     */
    delete(resourceGroupName, hubName, viewName, userId, options) {
        return this.client.sendOperationRequest({ resourceGroupName, hubName, viewName, userId, options }, deleteOperationSpec);
    }
    /**
     * ListByHubNext
     * @param resourceGroupName The name of the resource group.
     * @param hubName The name of the hub.
     * @param userId The user ID. Use * to retrieve hub level views.
     * @param nextLink The nextLink from the previous successful call to the ListByHub method.
     * @param options The options parameters.
     */
    _listByHubNext(resourceGroupName, hubName, userId, nextLink, options) {
        return this.client.sendOperationRequest({ resourceGroupName, hubName, userId, nextLink, options }, listByHubNextOperationSpec);
    }
}
// Operation Specifications
const serializer = coreClient.createSerializer(Mappers, /* isXml */ false);
const listByHubOperationSpec = {
    path: "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.CustomerInsights/hubs/{hubName}/views",
    httpMethod: "GET",
    responses: {
        200: {
            bodyMapper: Mappers.ViewListResult
        }
    },
    queryParameters: [Parameters.apiVersion, Parameters.userId],
    urlParameters: [
        Parameters.$host,
        Parameters.resourceGroupName,
        Parameters.subscriptionId,
        Parameters.hubName1
    ],
    headerParameters: [Parameters.accept],
    serializer
};
const createOrUpdateOperationSpec = {
    path: "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.CustomerInsights/hubs/{hubName}/views/{viewName}",
    httpMethod: "PUT",
    responses: {
        200: {
            bodyMapper: Mappers.ViewResourceFormat
        }
    },
    requestBody: Parameters.parameters9,
    queryParameters: [Parameters.apiVersion],
    urlParameters: [
        Parameters.$host,
        Parameters.resourceGroupName,
        Parameters.subscriptionId,
        Parameters.hubName1,
        Parameters.viewName
    ],
    headerParameters: [Parameters.accept, Parameters.contentType],
    mediaType: "json",
    serializer
};
const getOperationSpec = {
    path: "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.CustomerInsights/hubs/{hubName}/views/{viewName}",
    httpMethod: "GET",
    responses: {
        200: {
            bodyMapper: Mappers.ViewResourceFormat
        }
    },
    queryParameters: [Parameters.apiVersion, Parameters.userId],
    urlParameters: [
        Parameters.$host,
        Parameters.resourceGroupName,
        Parameters.subscriptionId,
        Parameters.hubName1,
        Parameters.viewName1
    ],
    headerParameters: [Parameters.accept],
    serializer
};
const deleteOperationSpec = {
    path: "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.CustomerInsights/hubs/{hubName}/views/{viewName}",
    httpMethod: "DELETE",
    responses: { 200: {} },
    queryParameters: [Parameters.apiVersion, Parameters.userId],
    urlParameters: [
        Parameters.$host,
        Parameters.resourceGroupName,
        Parameters.subscriptionId,
        Parameters.hubName1,
        Parameters.viewName1
    ],
    serializer
};
const listByHubNextOperationSpec = {
    path: "{nextLink}",
    httpMethod: "GET",
    responses: {
        200: {
            bodyMapper: Mappers.ViewListResult
        }
    },
    queryParameters: [Parameters.apiVersion, Parameters.userId],
    urlParameters: [
        Parameters.$host,
        Parameters.nextLink,
        Parameters.resourceGroupName,
        Parameters.subscriptionId,
        Parameters.hubName1
    ],
    headerParameters: [Parameters.accept],
    serializer
};
