// @azure/arm-azurestack/esm/operations/customerSubscriptions.js
/*
 * Copyright (c) Microsoft Corporation. All rights reserved.
 * Licensed under the MIT License. See License.txt in the project root for
 * license information.
 *
 * Code generated by Microsoft (R) AutoRest Code Generator.
 * Changes may cause incorrect behavior and will be lost if the code is
 * regenerated.
 */
import { __assign } from "tslib";
import * as msRest from "@azure/ms-rest-js";
import * as Mappers from "../models/customerSubscriptionsMappers";
import * as Parameters from "../models/parameters";
/** Class representing a CustomerSubscriptions. */
var CustomerSubscriptions = /** @class */ (function () {
    /**
     * Create a CustomerSubscriptions.
     * @param {AzureStackManagementClientContext} client Reference to the service client.
     */
    function CustomerSubscriptions(client) {
        this.client = client;
    }
    CustomerSubscriptions.prototype.list = function (resourceGroup, registrationName, options, callback) {
        return this.client.sendOperationRequest({
            resourceGroup: resourceGroup,
            registrationName: registrationName,
            options: options
        }, listOperationSpec, callback);
    };
    CustomerSubscriptions.prototype.get = function (resourceGroup, registrationName, customerSubscriptionName, options, callback) {
        return this.client.sendOperationRequest({
            resourceGroup: resourceGroup,
            registrationName: registrationName,
            customerSubscriptionName: customerSubscriptionName,
            options: options
        }, getOperationSpec, callback);
    };
    CustomerSubscriptions.prototype.deleteMethod = function (resourceGroup, registrationName, customerSubscriptionName, options, callback) {
        return this.client.sendOperationRequest({
            resourceGroup: resourceGroup,
            registrationName: registrationName,
            customerSubscriptionName: customerSubscriptionName,
            options: options
        }, deleteMethodOperationSpec, callback);
    };
    CustomerSubscriptions.prototype.create = function (resourceGroup, registrationName, customerSubscriptionName, customerCreationParameters, options, callback) {
        return this.client.sendOperationRequest({
            resourceGroup: resourceGroup,
            registrationName: registrationName,
            customerSubscriptionName: customerSubscriptionName,
            customerCreationParameters: customerCreationParameters,
            options: options
        }, createOperationSpec, callback);
    };
    CustomerSubscriptions.prototype.listNext = function (nextPageLink, options, callback) {
        return this.client.sendOperationRequest({
            nextPageLink: nextPageLink,
            options: options
        }, listNextOperationSpec, callback);
    };
    return CustomerSubscriptions;
}());
export { CustomerSubscriptions };
// Operation Specifications
var serializer = new msRest.Serializer(Mappers);
var listOperationSpec = {
    httpMethod: "GET",
    path: "subscriptions/{subscriptionId}/resourceGroups/{resourceGroup}/providers/Microsoft.AzureStack/registrations/{registrationName}/customerSubscriptions",
    urlParameters: [
        Parameters.subscriptionId,
        Parameters.resourceGroup,
        Parameters.registrationName
    ],
    queryParameters: [
        Parameters.apiVersion
    ],
    headerParameters: [
        Parameters.acceptLanguage
    ],
    responses: {
        200: {
            bodyMapper: Mappers.CustomerSubscriptionList
        },
        default: {
            bodyMapper: Mappers.ErrorResponse
        }
    },
    serializer: serializer
};
var getOperationSpec = {
    httpMethod: "GET",
    path: "subscriptions/{subscriptionId}/resourceGroups/{resourceGroup}/providers/Microsoft.AzureStack/registrations/{registrationName}/customerSubscriptions/{customerSubscriptionName}",
    urlParameters: [
        Parameters.subscriptionId,
        Parameters.resourceGroup,
        Parameters.registrationName,
        Parameters.customerSubscriptionName
    ],
    queryParameters: [
        Parameters.apiVersion
    ],
    headerParameters: [
        Parameters.acceptLanguage
    ],
    responses: {
        200: {
            bodyMapper: Mappers.CustomerSubscription
        },
        default: {
            bodyMapper: Mappers.ErrorResponse
        }
    },
    serializer: serializer
};
var deleteMethodOperationSpec = {
    httpMethod: "DELETE",
    path: "subscriptions/{subscriptionId}/resourceGroups/{resourceGroup}/providers/Microsoft.AzureStack/registrations/{registrationName}/customerSubscriptions/{customerSubscriptionName}",
    urlParameters: [
        Parameters.subscriptionId,
        Parameters.resourceGroup,
        Parameters.registrationName,
        Parameters.customerSubscriptionName
    ],
    queryParameters: [
        Parameters.apiVersion
    ],
    headerParameters: [
        Parameters.acceptLanguage
    ],
    responses: {
        200: {},
        204: {},
        default: {
            bodyMapper: Mappers.CloudError
        }
    },
    serializer: serializer
};
var createOperationSpec = {
    httpMethod: "PUT",
    path: "subscriptions/{subscriptionId}/resourceGroups/{resourceGroup}/providers/Microsoft.AzureStack/registrations/{registrationName}/customerSubscriptions/{customerSubscriptionName}",
    urlParameters: [
        Parameters.subscriptionId,
        Parameters.resourceGroup,
        Parameters.registrationName,
        Parameters.customerSubscriptionName
    ],
    queryParameters: [
        Parameters.apiVersion
    ],
    headerParameters: [
        Parameters.acceptLanguage
    ],
    requestBody: {
        parameterPath: "customerCreationParameters",
        mapper: __assign(__assign({}, Mappers.CustomerSubscription), { required: true })
    },
    responses: {
        200: {
            bodyMapper: Mappers.CustomerSubscription
        },
        default: {
            bodyMapper: Mappers.CloudError
        }
    },
    serializer: serializer
};
var listNextOperationSpec = {
    httpMethod: "GET",
    baseUrl: "https://management.azure.com",
    path: "{nextLink}",
    urlParameters: [
        Parameters.nextPageLink
    ],
    headerParameters: [
        Parameters.acceptLanguage
    ],
    responses: {
        200: {
            bodyMapper: Mappers.CustomerSubscriptionList
        },
        default: {
            bodyMapper: Mappers.ErrorResponse
        }
    },
    serializer: serializer
};
