// @azure/arm-peering/dist-esm/src/peeringManagementClient.js
/*
 * Copyright (c) Microsoft Corporation.
 * Licensed under the MIT License.
 *
 * Code generated by Microsoft (R) AutoRest Code Generator.
 * Changes may cause incorrect behavior and will be lost if the code is regenerated.
 */
import { __awaiter } from "tslib";
import * as coreClient from "@azure/core-client";
import * as coreRestPipeline from "@azure/core-rest-pipeline";
import { CdnPeeringPrefixesImpl, LegacyPeeringsImpl, LookingGlassImpl, OperationsImpl, PeerAsnsImpl, PeeringLocationsImpl, RegisteredAsnsImpl, RegisteredPrefixesImpl, PeeringsImpl, ReceivedRoutesImpl, ConnectionMonitorTestsImpl, PeeringServiceCountriesImpl, PeeringServiceLocationsImpl, PrefixesImpl, PeeringServiceProvidersImpl, PeeringServicesImpl } from "./operations";
import * as Parameters from "./models/parameters";
import * as Mappers from "./models/mappers";
export class PeeringManagementClient extends coreClient.ServiceClient {
    /**
     * Initializes a new instance of the PeeringManagementClient class.
     * @param credentials Subscription credentials which uniquely identify client subscription.
     * @param subscriptionId The Azure subscription ID.
     * @param options The parameter options
     */
    constructor(credentials, subscriptionId, options) {
        var _a, _b, _c;
        if (credentials === undefined) {
            throw new Error("'credentials' cannot be null");
        }
        if (subscriptionId === undefined) {
            throw new Error("'subscriptionId' cannot be null");
        }
        // Initializing default values for options
        if (!options) {
            options = {};
        }
        const defaults = {
            requestContentType: "application/json; charset=utf-8",
            credential: credentials
        };
        const packageDetails = `azsdk-js-arm-peering/2.1.0`;
        const userAgentPrefix = options.userAgentOptions && options.userAgentOptions.userAgentPrefix
            ? `${options.userAgentOptions.userAgentPrefix} ${packageDetails}`
            : `${packageDetails}`;
        const optionsWithDefaults = Object.assign(Object.assign(Object.assign({}, defaults), options), { userAgentOptions: {
                userAgentPrefix
            }, endpoint: (_b = (_a = options.endpoint) !== null && _a !== void 0 ? _a : options.baseUri) !== null && _b !== void 0 ? _b : "https://management.azure.com" });
        super(optionsWithDefaults);
        let bearerTokenAuthenticationPolicyFound = false;
        if ((options === null || options === void 0 ? void 0 : options.pipeline) && options.pipeline.getOrderedPolicies().length > 0) {
            const pipelinePolicies = options.pipeline.getOrderedPolicies();
            bearerTokenAuthenticationPolicyFound = pipelinePolicies.some((pipelinePolicy) => pipelinePolicy.name ===
                coreRestPipeline.bearerTokenAuthenticationPolicyName);
        }
        if (!options ||
            !options.pipeline ||
            options.pipeline.getOrderedPolicies().length == 0 ||
            !bearerTokenAuthenticationPolicyFound) {
            this.pipeline.removePolicy({
                name: coreRestPipeline.bearerTokenAuthenticationPolicyName
            });
            this.pipeline.addPolicy(coreRestPipeline.bearerTokenAuthenticationPolicy({
                credential: credentials,
                scopes: (_c = optionsWithDefaults.credentialScopes) !== null && _c !== void 0 ? _c : `${optionsWithDefaults.endpoint}/.default`,
                challengeCallbacks: {
                    authorizeRequestOnChallenge: coreClient.authorizeRequestOnClaimChallenge
                }
            }));
        }
        // Parameter assignments
        this.subscriptionId = subscriptionId;
        // Assigning values to Constant parameters
        this.$host = options.$host || "https://management.azure.com";
        this.apiVersion = options.apiVersion || "2021-06-01";
        this.cdnPeeringPrefixes = new CdnPeeringPrefixesImpl(this);
        this.legacyPeerings = new LegacyPeeringsImpl(this);
        this.lookingGlass = new LookingGlassImpl(this);
        this.operations = new OperationsImpl(this);
        this.peerAsns = new PeerAsnsImpl(this);
        this.peeringLocations = new PeeringLocationsImpl(this);
        this.registeredAsns = new RegisteredAsnsImpl(this);
        this.registeredPrefixes = new RegisteredPrefixesImpl(this);
        this.peerings = new PeeringsImpl(this);
        this.receivedRoutes = new ReceivedRoutesImpl(this);
        this.connectionMonitorTests = new ConnectionMonitorTestsImpl(this);
        this.peeringServiceCountries = new PeeringServiceCountriesImpl(this);
        this.peeringServiceLocations = new PeeringServiceLocationsImpl(this);
        this.prefixes = new PrefixesImpl(this);
        this.peeringServiceProviders = new PeeringServiceProvidersImpl(this);
        this.peeringServices = new PeeringServicesImpl(this);
        this.addCustomApiVersionPolicy(options.apiVersion);
    }
    /** A function that adds a policy that sets the api-version (or equivalent) to reflect the library version. */
    addCustomApiVersionPolicy(apiVersion) {
        if (!apiVersion) {
            return;
        }
        const apiVersionPolicy = {
            name: "CustomApiVersionPolicy",
            sendRequest(request, next) {
                return __awaiter(this, void 0, void 0, function* () {
                    const param = request.url.split("?");
                    if (param.length > 1) {
                        const newParams = param[1].split("&").map((item) => {
                            if (item.indexOf("api-version") > -1) {
                                return "api-version=" + apiVersion;
                            }
                            else {
                                return item;
                            }
                        });
                        request.url = param[0] + "?" + newParams.join("&");
                    }
                    return next(request);
                });
            }
        };
        this.pipeline.addPolicy(apiVersionPolicy);
    }
    /**
     * Checks if the peering service provider is present within 1000 miles of customer's location
     * @param checkServiceProviderAvailabilityInput The CheckServiceProviderAvailabilityInput indicating
     *                                              customer location and service provider.
     * @param options The options parameters.
     */
    checkServiceProviderAvailability(checkServiceProviderAvailabilityInput, options) {
        return this.sendOperationRequest({ checkServiceProviderAvailabilityInput, options }, checkServiceProviderAvailabilityOperationSpec);
    }
}
// Operation Specifications
const serializer = coreClient.createSerializer(Mappers, /* isXml */ false);
const checkServiceProviderAvailabilityOperationSpec = {
    path: "/subscriptions/{subscriptionId}/providers/Microsoft.Peering/checkServiceProviderAvailability",
    httpMethod: "POST",
    responses: {
        200: {
            bodyMapper: { type: { name: "String" } }
        },
        default: {
            bodyMapper: Mappers.ErrorResponse
        }
    },
    requestBody: Parameters.checkServiceProviderAvailabilityInput,
    queryParameters: [Parameters.apiVersion],
    urlParameters: [Parameters.$host, Parameters.subscriptionId],
    headerParameters: [Parameters.accept, Parameters.contentType],
    mediaType: "json",
    serializer
};
