// @aws-sdk/client-budgets/dist-cjs/index.js
"use strict";
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __name = (target, value) => __defProp(target, "name", { value, configurable: true });
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// src/index.ts
var src_exports = {};
__export(src_exports, {
  AccessDeniedException: () => AccessDeniedException,
  ActionFilterSensitiveLog: () => ActionFilterSensitiveLog,
  ActionHistoryDetailsFilterSensitiveLog: () => ActionHistoryDetailsFilterSensitiveLog,
  ActionHistoryFilterSensitiveLog: () => ActionHistoryFilterSensitiveLog,
  ActionStatus: () => ActionStatus,
  ActionSubType: () => ActionSubType,
  ActionType: () => ActionType,
  ApprovalModel: () => ApprovalModel,
  AutoAdjustType: () => AutoAdjustType,
  BudgetType: () => BudgetType,
  Budgets: () => Budgets,
  BudgetsClient: () => BudgetsClient,
  BudgetsServiceException: () => BudgetsServiceException,
  ComparisonOperator: () => ComparisonOperator,
  CreateBudgetActionCommand: () => CreateBudgetActionCommand,
  CreateBudgetActionRequestFilterSensitiveLog: () => CreateBudgetActionRequestFilterSensitiveLog,
  CreateBudgetCommand: () => CreateBudgetCommand,
  CreateBudgetRequestFilterSensitiveLog: () => CreateBudgetRequestFilterSensitiveLog,
  CreateNotificationCommand: () => CreateNotificationCommand,
  CreateNotificationRequestFilterSensitiveLog: () => CreateNotificationRequestFilterSensitiveLog,
  CreateSubscriberCommand: () => CreateSubscriberCommand,
  CreateSubscriberRequestFilterSensitiveLog: () => CreateSubscriberRequestFilterSensitiveLog,
  CreationLimitExceededException: () => CreationLimitExceededException,
  DeleteBudgetActionCommand: () => DeleteBudgetActionCommand,
  DeleteBudgetActionResponseFilterSensitiveLog: () => DeleteBudgetActionResponseFilterSensitiveLog,
  DeleteBudgetCommand: () => DeleteBudgetCommand,
  DeleteNotificationCommand: () => DeleteNotificationCommand,
  DeleteSubscriberCommand: () => DeleteSubscriberCommand,
  DeleteSubscriberRequestFilterSensitiveLog: () => DeleteSubscriberRequestFilterSensitiveLog,
  DescribeBudgetActionCommand: () => DescribeBudgetActionCommand,
  DescribeBudgetActionHistoriesCommand: () => DescribeBudgetActionHistoriesCommand,
  DescribeBudgetActionHistoriesResponseFilterSensitiveLog: () => DescribeBudgetActionHistoriesResponseFilterSensitiveLog,
  DescribeBudgetActionResponseFilterSensitiveLog: () => DescribeBudgetActionResponseFilterSensitiveLog,
  DescribeBudgetActionsForAccountCommand: () => DescribeBudgetActionsForAccountCommand,
  DescribeBudgetActionsForAccountResponseFilterSensitiveLog: () => DescribeBudgetActionsForAccountResponseFilterSensitiveLog,
  DescribeBudgetActionsForBudgetCommand: () => DescribeBudgetActionsForBudgetCommand,
  DescribeBudgetActionsForBudgetResponseFilterSensitiveLog: () => DescribeBudgetActionsForBudgetResponseFilterSensitiveLog,
  DescribeBudgetCommand: () => DescribeBudgetCommand,
  DescribeBudgetNotificationsForAccountCommand: () => DescribeBudgetNotificationsForAccountCommand,
  DescribeBudgetPerformanceHistoryCommand: () => DescribeBudgetPerformanceHistoryCommand,
  DescribeBudgetsCommand: () => DescribeBudgetsCommand,
  DescribeNotificationsForBudgetCommand: () => DescribeNotificationsForBudgetCommand,
  DescribeSubscribersForNotificationCommand: () => DescribeSubscribersForNotificationCommand,
  DescribeSubscribersForNotificationResponseFilterSensitiveLog: () => DescribeSubscribersForNotificationResponseFilterSensitiveLog,
  DuplicateRecordException: () => DuplicateRecordException,
  EventType: () => EventType,
  ExecuteBudgetActionCommand: () => ExecuteBudgetActionCommand,
  ExecutionType: () => ExecutionType,
  ExpiredNextTokenException: () => ExpiredNextTokenException,
  InternalErrorException: () => InternalErrorException,
  InvalidNextTokenException: () => InvalidNextTokenException,
  InvalidParameterException: () => InvalidParameterException,
  NotFoundException: () => NotFoundException,
  NotificationState: () => NotificationState,
  NotificationType: () => NotificationType,
  NotificationWithSubscribersFilterSensitiveLog: () => NotificationWithSubscribersFilterSensitiveLog,
  ResourceLockedException: () => ResourceLockedException,
  SubscriberFilterSensitiveLog: () => SubscriberFilterSensitiveLog,
  SubscriptionType: () => SubscriptionType,
  ThresholdType: () => ThresholdType,
  ThrottlingException: () => ThrottlingException,
  TimeUnit: () => TimeUnit,
  UpdateBudgetActionCommand: () => UpdateBudgetActionCommand,
  UpdateBudgetActionRequestFilterSensitiveLog: () => UpdateBudgetActionRequestFilterSensitiveLog,
  UpdateBudgetActionResponseFilterSensitiveLog: () => UpdateBudgetActionResponseFilterSensitiveLog,
  UpdateBudgetCommand: () => UpdateBudgetCommand,
  UpdateNotificationCommand: () => UpdateNotificationCommand,
  UpdateSubscriberCommand: () => UpdateSubscriberCommand,
  UpdateSubscriberRequestFilterSensitiveLog: () => UpdateSubscriberRequestFilterSensitiveLog,
  __Client: () => import_smithy_client.Client,
  paginateDescribeBudgetActionHistories: () => paginateDescribeBudgetActionHistories,
  paginateDescribeBudgetActionsForAccount: () => paginateDescribeBudgetActionsForAccount,
  paginateDescribeBudgetActionsForBudget: () => paginateDescribeBudgetActionsForBudget,
  paginateDescribeBudgetNotificationsForAccount: () => paginateDescribeBudgetNotificationsForAccount,
  paginateDescribeBudgetPerformanceHistory: () => paginateDescribeBudgetPerformanceHistory,
  paginateDescribeBudgets: () => paginateDescribeBudgets,
  paginateDescribeNotificationsForBudget: () => paginateDescribeNotificationsForBudget,
  paginateDescribeSubscribersForNotification: () => paginateDescribeSubscribersForNotification
});
module.exports = __toCommonJS(src_exports);

// src/BudgetsClient.ts
var import_middleware_host_header = require("@aws-sdk/middleware-host-header");
var import_middleware_logger = require("@aws-sdk/middleware-logger");
var import_middleware_recursion_detection = require("@aws-sdk/middleware-recursion-detection");
var import_middleware_user_agent = require("@aws-sdk/middleware-user-agent");
var import_config_resolver = require("@smithy/config-resolver");
var import_core = require("@smithy/core");
var import_middleware_content_length = require("@smithy/middleware-content-length");
var import_middleware_endpoint = require("@smithy/middleware-endpoint");
var import_middleware_retry = require("@smithy/middleware-retry");

var import_httpAuthSchemeProvider = require("./auth/httpAuthSchemeProvider");

// src/endpoint/EndpointParameters.ts
var resolveClientEndpointParameters = /* @__PURE__ */ __name((options) => {
  return {
    ...options,
    useDualstackEndpoint: options.useDualstackEndpoint ?? false,
    useFipsEndpoint: options.useFipsEndpoint ?? false,
    defaultSigningName: "budgets"
  };
}, "resolveClientEndpointParameters");
var commonParams = {
  UseFIPS: { type: "builtInParams", name: "useFipsEndpoint" },
  Endpoint: { type: "builtInParams", name: "endpoint" },
  Region: { type: "builtInParams", name: "region" },
  UseDualStack: { type: "builtInParams", name: "useDualstackEndpoint" }
};

// src/BudgetsClient.ts
var import_runtimeConfig = require("././runtimeConfig");

// src/runtimeExtensions.ts
var import_region_config_resolver = require("@aws-sdk/region-config-resolver");
var import_protocol_http = require("@smithy/protocol-http");
var import_smithy_client = require("@smithy/smithy-client");

// src/auth/httpAuthExtensionConfiguration.ts
var getHttpAuthExtensionConfiguration = /* @__PURE__ */ __name((runtimeConfig) => {
  const _httpAuthSchemes = runtimeConfig.httpAuthSchemes;
  let _httpAuthSchemeProvider = runtimeConfig.httpAuthSchemeProvider;
  let _credentials = runtimeConfig.credentials;
  return {
    setHttpAuthScheme(httpAuthScheme) {
      const index = _httpAuthSchemes.findIndex((scheme) => scheme.schemeId === httpAuthScheme.schemeId);
      if (index === -1) {
        _httpAuthSchemes.push(httpAuthScheme);
      } else {
        _httpAuthSchemes.splice(index, 1, httpAuthScheme);
      }
    },
    httpAuthSchemes() {
      return _httpAuthSchemes;
    },
    setHttpAuthSchemeProvider(httpAuthSchemeProvider) {
      _httpAuthSchemeProvider = httpAuthSchemeProvider;
    },
    httpAuthSchemeProvider() {
      return _httpAuthSchemeProvider;
    },
    setCredentials(credentials) {
      _credentials = credentials;
    },
    credentials() {
      return _credentials;
    }
  };
}, "getHttpAuthExtensionConfiguration");
var resolveHttpAuthRuntimeConfig = /* @__PURE__ */ __name((config) => {
  return {
    httpAuthSchemes: config.httpAuthSchemes(),
    httpAuthSchemeProvider: config.httpAuthSchemeProvider(),
    credentials: config.credentials()
  };
}, "resolveHttpAuthRuntimeConfig");

// src/runtimeExtensions.ts
var asPartial = /* @__PURE__ */ __name((t) => t, "asPartial");
var resolveRuntimeExtensions = /* @__PURE__ */ __name((runtimeConfig, extensions) => {
  const extensionConfiguration = {
    ...asPartial((0, import_region_config_resolver.getAwsRegionExtensionConfiguration)(runtimeConfig)),
    ...asPartial((0, import_smithy_client.getDefaultExtensionConfiguration)(runtimeConfig)),
    ...asPartial((0, import_protocol_http.getHttpHandlerExtensionConfiguration)(runtimeConfig)),
    ...asPartial(getHttpAuthExtensionConfiguration(runtimeConfig))
  };
  extensions.forEach((extension) => extension.configure(extensionConfiguration));
  return {
    ...runtimeConfig,
    ...(0, import_region_config_resolver.resolveAwsRegionExtensionConfiguration)(extensionConfiguration),
    ...(0, import_smithy_client.resolveDefaultRuntimeConfig)(extensionConfiguration),
    ...(0, import_protocol_http.resolveHttpHandlerRuntimeConfig)(extensionConfiguration),
    ...resolveHttpAuthRuntimeConfig(extensionConfiguration)
  };
}, "resolveRuntimeExtensions");

// src/BudgetsClient.ts
var _BudgetsClient = class _BudgetsClient extends import_smithy_client.Client {
  constructor(...[configuration]) {
    const _config_0 = (0, import_runtimeConfig.getRuntimeConfig)(configuration || {});
    const _config_1 = resolveClientEndpointParameters(_config_0);
    const _config_2 = (0, import_config_resolver.resolveRegionConfig)(_config_1);
    const _config_3 = (0, import_middleware_endpoint.resolveEndpointConfig)(_config_2);
    const _config_4 = (0, import_middleware_retry.resolveRetryConfig)(_config_3);
    const _config_5 = (0, import_middleware_host_header.resolveHostHeaderConfig)(_config_4);
    const _config_6 = (0, import_middleware_user_agent.resolveUserAgentConfig)(_config_5);
    const _config_7 = (0, import_httpAuthSchemeProvider.resolveHttpAuthSchemeConfig)(_config_6);
    const _config_8 = resolveRuntimeExtensions(_config_7, (configuration == null ? void 0 : configuration.extensions) || []);
    super(_config_8);
    this.config = _config_8;
    this.middlewareStack.use((0, import_middleware_retry.getRetryPlugin)(this.config));
    this.middlewareStack.use((0, import_middleware_content_length.getContentLengthPlugin)(this.config));
    this.middlewareStack.use((0, import_middleware_host_header.getHostHeaderPlugin)(this.config));
    this.middlewareStack.use((0, import_middleware_logger.getLoggerPlugin)(this.config));
    this.middlewareStack.use((0, import_middleware_recursion_detection.getRecursionDetectionPlugin)(this.config));
    this.middlewareStack.use((0, import_middleware_user_agent.getUserAgentPlugin)(this.config));
    this.middlewareStack.use(
      (0, import_core.getHttpAuthSchemeEndpointRuleSetPlugin)(this.config, {
        httpAuthSchemeParametersProvider: this.getDefaultHttpAuthSchemeParametersProvider(),
        identityProviderConfigProvider: this.getIdentityProviderConfigProvider()
      })
    );
    this.middlewareStack.use((0, import_core.getHttpSigningPlugin)(this.config));
  }
  /**
   * Destroy underlying resources, like sockets. It's usually not necessary to do this.
   * However in Node.js, it's best to explicitly shut down the client's agent when it is no longer needed.
   * Otherwise, sockets might stay open for quite a long time before the server terminates them.
   */
  destroy() {
    super.destroy();
  }
  getDefaultHttpAuthSchemeParametersProvider() {
    return import_httpAuthSchemeProvider.defaultBudgetsHttpAuthSchemeParametersProvider;
  }
  getIdentityProviderConfigProvider() {
    return async (config) => new import_core.DefaultIdentityProviderConfig({
      "aws.auth#sigv4": config.credentials
    });
  }
};
__name(_BudgetsClient, "BudgetsClient");
var BudgetsClient = _BudgetsClient;

// src/Budgets.ts


// src/commands/CreateBudgetActionCommand.ts

var import_middleware_serde = require("@smithy/middleware-serde");

var import_types = require("@smithy/types");

// src/models/models_0.ts


// src/models/BudgetsServiceException.ts

var _BudgetsServiceException = class _BudgetsServiceException extends import_smithy_client.ServiceException {
  /**
   * @internal
   */
  constructor(options) {
    super(options);
    Object.setPrototypeOf(this, _BudgetsServiceException.prototype);
  }
};
__name(_BudgetsServiceException, "BudgetsServiceException");
var BudgetsServiceException = _BudgetsServiceException;

// src/models/models_0.ts
var _AccessDeniedException = class _AccessDeniedException extends BudgetsServiceException {
  /**
   * @internal
   */
  constructor(opts) {
    super({
      name: "AccessDeniedException",
      $fault: "client",
      ...opts
    });
    this.name = "AccessDeniedException";
    this.$fault = "client";
    Object.setPrototypeOf(this, _AccessDeniedException.prototype);
    this.Message = opts.Message;
  }
};
__name(_AccessDeniedException, "AccessDeniedException");
var AccessDeniedException = _AccessDeniedException;
var ThresholdType = {
  ABSOLUTE_VALUE: "ABSOLUTE_VALUE",
  PERCENTAGE: "PERCENTAGE"
};
var ActionType = {
  IAM: "APPLY_IAM_POLICY",
  SCP: "APPLY_SCP_POLICY",
  SSM: "RUN_SSM_DOCUMENTS"
};
var ApprovalModel = {
  AUTO: "AUTOMATIC",
  MANUAL: "MANUAL"
};
var ActionSubType = {
  STOP_EC2: "STOP_EC2_INSTANCES",
  STOP_RDS: "STOP_RDS_INSTANCES"
};
var NotificationType = {
  ACTUAL: "ACTUAL",
  FORECASTED: "FORECASTED"
};
var ActionStatus = {
  Execution_Failure: "EXECUTION_FAILURE",
  Execution_In_Progress: "EXECUTION_IN_PROGRESS",
  Execution_Success: "EXECUTION_SUCCESS",
  Pending: "PENDING",
  Reset_Failure: "RESET_FAILURE",
  Reset_In_Progress: "RESET_IN_PROGRESS",
  Reverse_Failure: "REVERSE_FAILURE",
  Reverse_In_Progress: "REVERSE_IN_PROGRESS",
  Reverse_Success: "REVERSE_SUCCESS",
  Standby: "STANDBY"
};
var SubscriptionType = {
  EMAIL: "EMAIL",
  SNS: "SNS"
};
var EventType = {
  CreateAction: "CREATE_ACTION",
  DeleteAction: "DELETE_ACTION",
  ExecuteAction: "EXECUTE_ACTION",
  System: "SYSTEM",
  UpdateAction: "UPDATE_ACTION"
};
var AutoAdjustType = {
  FORECAST: "FORECAST",
  HISTORICAL: "HISTORICAL"
};
var BudgetType = {
  Cost: "COST",
  RICoverage: "RI_COVERAGE",
  RIUtilization: "RI_UTILIZATION",
  SPCoverage: "SAVINGS_PLANS_COVERAGE",
  SPUtilization: "SAVINGS_PLANS_UTILIZATION",
  Usage: "USAGE"
};
var TimeUnit = {
  ANNUALLY: "ANNUALLY",
  DAILY: "DAILY",
  MONTHLY: "MONTHLY",
  QUARTERLY: "QUARTERLY"
};
var ComparisonOperator = {
  EQUAL_TO: "EQUAL_TO",
  GREATER_THAN: "GREATER_THAN",
  LESS_THAN: "LESS_THAN"
};
var NotificationState = {
  ALARM: "ALARM",
  OK: "OK"
};
var _CreationLimitExceededException = class _CreationLimitExceededException extends BudgetsServiceException {
  /**
   * @internal
   */
  constructor(opts) {
    super({
      name: "CreationLimitExceededException",
      $fault: "client",
      ...opts
    });
    this.name = "CreationLimitExceededException";
    this.$fault = "client";
    Object.setPrototypeOf(this, _CreationLimitExceededException.prototype);
    this.Message = opts.Message;
  }
};
__name(_CreationLimitExceededException, "CreationLimitExceededException");
var CreationLimitExceededException = _CreationLimitExceededException;
var _DuplicateRecordException = class _DuplicateRecordException extends BudgetsServiceException {
  /**
   * @internal
   */
  constructor(opts) {
    super({
      name: "DuplicateRecordException",
      $fault: "client",
      ...opts
    });
    this.name = "DuplicateRecordException";
    this.$fault = "client";
    Object.setPrototypeOf(this, _DuplicateRecordException.prototype);
    this.Message = opts.Message;
  }
};
__name(_DuplicateRecordException, "DuplicateRecordException");
var DuplicateRecordException = _DuplicateRecordException;
var _InternalErrorException = class _InternalErrorException extends BudgetsServiceException {
  /**
   * @internal
   */
  constructor(opts) {
    super({
      name: "InternalErrorException",
      $fault: "server",
      ...opts
    });
    this.name = "InternalErrorException";
    this.$fault = "server";
    Object.setPrototypeOf(this, _InternalErrorException.prototype);
    this.Message = opts.Message;
  }
};
__name(_InternalErrorException, "InternalErrorException");
var InternalErrorException = _InternalErrorException;
var _InvalidParameterException = class _InvalidParameterException extends BudgetsServiceException {
  /**
   * @internal
   */
  constructor(opts) {
    super({
      name: "InvalidParameterException",
      $fault: "client",
      ...opts
    });
    this.name = "InvalidParameterException";
    this.$fault = "client";
    Object.setPrototypeOf(this, _InvalidParameterException.prototype);
    this.Message = opts.Message;
  }
};
__name(_InvalidParameterException, "InvalidParameterException");
var InvalidParameterException = _InvalidParameterException;
var _ThrottlingException = class _ThrottlingException extends BudgetsServiceException {
  /**
   * @internal
   */
  constructor(opts) {
    super({
      name: "ThrottlingException",
      $fault: "client",
      ...opts
    });
    this.name = "ThrottlingException";
    this.$fault = "client";
    Object.setPrototypeOf(this, _ThrottlingException.prototype);
    this.Message = opts.Message;
  }
};
__name(_ThrottlingException, "ThrottlingException");
var ThrottlingException = _ThrottlingException;
var _NotFoundException = class _NotFoundException extends BudgetsServiceException {
  /**
   * @internal
   */
  constructor(opts) {
    super({
      name: "NotFoundException",
      $fault: "client",
      ...opts
    });
    this.name = "NotFoundException";
    this.$fault = "client";
    Object.setPrototypeOf(this, _NotFoundException.prototype);
    this.Message = opts.Message;
  }
};
__name(_NotFoundException, "NotFoundException");
var NotFoundException = _NotFoundException;
var _ResourceLockedException = class _ResourceLockedException extends BudgetsServiceException {
  /**
   * @internal
   */
  constructor(opts) {
    super({
      name: "ResourceLockedException",
      $fault: "client",
      ...opts
    });
    this.name = "ResourceLockedException";
    this.$fault = "client";
    Object.setPrototypeOf(this, _ResourceLockedException.prototype);
    this.Message = opts.Message;
  }
};
__name(_ResourceLockedException, "ResourceLockedException");
var ResourceLockedException = _ResourceLockedException;
var _InvalidNextTokenException = class _InvalidNextTokenException extends BudgetsServiceException {
  /**
   * @internal
   */
  constructor(opts) {
    super({
      name: "InvalidNextTokenException",
      $fault: "client",
      ...opts
    });
    this.name = "InvalidNextTokenException";
    this.$fault = "client";
    Object.setPrototypeOf(this, _InvalidNextTokenException.prototype);
    this.Message = opts.Message;
  }
};
__name(_InvalidNextTokenException, "InvalidNextTokenException");
var InvalidNextTokenException = _InvalidNextTokenException;
var _ExpiredNextTokenException = class _ExpiredNextTokenException extends BudgetsServiceException {
  /**
   * @internal
   */
  constructor(opts) {
    super({
      name: "ExpiredNextTokenException",
      $fault: "client",
      ...opts
    });
    this.name = "ExpiredNextTokenException";
    this.$fault = "client";
    Object.setPrototypeOf(this, _ExpiredNextTokenException.prototype);
    this.Message = opts.Message;
  }
};
__name(_ExpiredNextTokenException, "ExpiredNextTokenException");
var ExpiredNextTokenException = _ExpiredNextTokenException;
var ExecutionType = {
  ApproveBudgetAction: "APPROVE_BUDGET_ACTION",
  ResetBudgetAction: "RESET_BUDGET_ACTION",
  RetryBudgetAction: "RETRY_BUDGET_ACTION",
  ReverseBudgetAction: "REVERSE_BUDGET_ACTION"
};
var SubscriberFilterSensitiveLog = /* @__PURE__ */ __name((obj) => ({
  ...obj,
  ...obj.Address && { Address: import_smithy_client.SENSITIVE_STRING }
}), "SubscriberFilterSensitiveLog");
var ActionFilterSensitiveLog = /* @__PURE__ */ __name((obj) => ({
  ...obj,
  ...obj.Subscribers && { Subscribers: obj.Subscribers.map((item) => SubscriberFilterSensitiveLog(item)) }
}), "ActionFilterSensitiveLog");
var ActionHistoryDetailsFilterSensitiveLog = /* @__PURE__ */ __name((obj) => ({
  ...obj,
  ...obj.Action && { Action: ActionFilterSensitiveLog(obj.Action) }
}), "ActionHistoryDetailsFilterSensitiveLog");
var ActionHistoryFilterSensitiveLog = /* @__PURE__ */ __name((obj) => ({
  ...obj,
  ...obj.ActionHistoryDetails && {
    ActionHistoryDetails: ActionHistoryDetailsFilterSensitiveLog(obj.ActionHistoryDetails)
  }
}), "ActionHistoryFilterSensitiveLog");
var NotificationWithSubscribersFilterSensitiveLog = /* @__PURE__ */ __name((obj) => ({
  ...obj,
  ...obj.Subscribers && { Subscribers: obj.Subscribers.map((item) => SubscriberFilterSensitiveLog(item)) }
}), "NotificationWithSubscribersFilterSensitiveLog");
var CreateBudgetRequestFilterSensitiveLog = /* @__PURE__ */ __name((obj) => ({
  ...obj
}), "CreateBudgetRequestFilterSensitiveLog");
var CreateBudgetActionRequestFilterSensitiveLog = /* @__PURE__ */ __name((obj) => ({
  ...obj,
  ...obj.Subscribers && { Subscribers: obj.Subscribers.map((item) => SubscriberFilterSensitiveLog(item)) }
}), "CreateBudgetActionRequestFilterSensitiveLog");
var CreateNotificationRequestFilterSensitiveLog = /* @__PURE__ */ __name((obj) => ({
  ...obj,
  ...obj.Subscribers && { Subscribers: obj.Subscribers.map((item) => SubscriberFilterSensitiveLog(item)) }
}), "CreateNotificationRequestFilterSensitiveLog");
var CreateSubscriberRequestFilterSensitiveLog = /* @__PURE__ */ __name((obj) => ({
  ...obj,
  ...obj.Subscriber && { Subscriber: SubscriberFilterSensitiveLog(obj.Subscriber) }
}), "CreateSubscriberRequestFilterSensitiveLog");
var DeleteBudgetActionResponseFilterSensitiveLog = /* @__PURE__ */ __name((obj) => ({
  ...obj,
  ...obj.Action && { Action: ActionFilterSensitiveLog(obj.Action) }
}), "DeleteBudgetActionResponseFilterSensitiveLog");
var DeleteSubscriberRequestFilterSensitiveLog = /* @__PURE__ */ __name((obj) => ({
  ...obj,
  ...obj.Subscriber && { Subscriber: SubscriberFilterSensitiveLog(obj.Subscriber) }
}), "DeleteSubscriberRequestFilterSensitiveLog");
var DescribeBudgetActionResponseFilterSensitiveLog = /* @__PURE__ */ __name((obj) => ({
  ...obj,
  ...obj.Action && { Action: ActionFilterSensitiveLog(obj.Action) }
}), "DescribeBudgetActionResponseFilterSensitiveLog");
var DescribeBudgetActionHistoriesResponseFilterSensitiveLog = /* @__PURE__ */ __name((obj) => ({
  ...obj
}), "DescribeBudgetActionHistoriesResponseFilterSensitiveLog");
var DescribeBudgetActionsForAccountResponseFilterSensitiveLog = /* @__PURE__ */ __name((obj) => ({
  ...obj
}), "DescribeBudgetActionsForAccountResponseFilterSensitiveLog");
var DescribeBudgetActionsForBudgetResponseFilterSensitiveLog = /* @__PURE__ */ __name((obj) => ({
  ...obj
}), "DescribeBudgetActionsForBudgetResponseFilterSensitiveLog");
var DescribeSubscribersForNotificationResponseFilterSensitiveLog = /* @__PURE__ */ __name((obj) => ({
  ...obj,
  ...obj.Subscribers && { Subscribers: obj.Subscribers.map((item) => SubscriberFilterSensitiveLog(item)) }
}), "DescribeSubscribersForNotificationResponseFilterSensitiveLog");
var UpdateBudgetActionRequestFilterSensitiveLog = /* @__PURE__ */ __name((obj) => ({
  ...obj,
  ...obj.Subscribers && { Subscribers: obj.Subscribers.map((item) => SubscriberFilterSensitiveLog(item)) }
}), "UpdateBudgetActionRequestFilterSensitiveLog");
var UpdateBudgetActionResponseFilterSensitiveLog = /* @__PURE__ */ __name((obj) => ({
  ...obj,
  ...obj.OldAction && { OldAction: ActionFilterSensitiveLog(obj.OldAction) },
  ...obj.NewAction && { NewAction: ActionFilterSensitiveLog(obj.NewAction) }
}), "UpdateBudgetActionResponseFilterSensitiveLog");
var UpdateSubscriberRequestFilterSensitiveLog = /* @__PURE__ */ __name((obj) => ({
  ...obj,
  ...obj.OldSubscriber && { OldSubscriber: SubscriberFilterSensitiveLog(obj.OldSubscriber) },
  ...obj.NewSubscriber && { NewSubscriber: SubscriberFilterSensitiveLog(obj.NewSubscriber) }
}), "UpdateSubscriberRequestFilterSensitiveLog");

// src/protocols/Aws_json1_1.ts


var se_CreateBudgetCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("CreateBudget");
  let body;
  body = JSON.stringify(se_CreateBudgetRequest(input, context));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_CreateBudgetCommand");
var se_CreateBudgetActionCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("CreateBudgetAction");
  let body;
  body = JSON.stringify(se_CreateBudgetActionRequest(input, context));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_CreateBudgetActionCommand");
var se_CreateNotificationCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("CreateNotification");
  let body;
  body = JSON.stringify(se_CreateNotificationRequest(input, context));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_CreateNotificationCommand");
var se_CreateSubscriberCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("CreateSubscriber");
  let body;
  body = JSON.stringify(se_CreateSubscriberRequest(input, context));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_CreateSubscriberCommand");
var se_DeleteBudgetCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("DeleteBudget");
  let body;
  body = JSON.stringify((0, import_smithy_client._json)(input));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_DeleteBudgetCommand");
var se_DeleteBudgetActionCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("DeleteBudgetAction");
  let body;
  body = JSON.stringify((0, import_smithy_client._json)(input));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_DeleteBudgetActionCommand");
var se_DeleteNotificationCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("DeleteNotification");
  let body;
  body = JSON.stringify(se_DeleteNotificationRequest(input, context));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_DeleteNotificationCommand");
var se_DeleteSubscriberCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("DeleteSubscriber");
  let body;
  body = JSON.stringify(se_DeleteSubscriberRequest(input, context));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_DeleteSubscriberCommand");
var se_DescribeBudgetCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("DescribeBudget");
  let body;
  body = JSON.stringify((0, import_smithy_client._json)(input));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_DescribeBudgetCommand");
var se_DescribeBudgetActionCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("DescribeBudgetAction");
  let body;
  body = JSON.stringify((0, import_smithy_client._json)(input));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_DescribeBudgetActionCommand");
var se_DescribeBudgetActionHistoriesCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("DescribeBudgetActionHistories");
  let body;
  body = JSON.stringify(se_DescribeBudgetActionHistoriesRequest(input, context));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_DescribeBudgetActionHistoriesCommand");
var se_DescribeBudgetActionsForAccountCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("DescribeBudgetActionsForAccount");
  let body;
  body = JSON.stringify((0, import_smithy_client._json)(input));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_DescribeBudgetActionsForAccountCommand");
var se_DescribeBudgetActionsForBudgetCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("DescribeBudgetActionsForBudget");
  let body;
  body = JSON.stringify((0, import_smithy_client._json)(input));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_DescribeBudgetActionsForBudgetCommand");
var se_DescribeBudgetNotificationsForAccountCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("DescribeBudgetNotificationsForAccount");
  let body;
  body = JSON.stringify((0, import_smithy_client._json)(input));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_DescribeBudgetNotificationsForAccountCommand");
var se_DescribeBudgetPerformanceHistoryCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("DescribeBudgetPerformanceHistory");
  let body;
  body = JSON.stringify(se_DescribeBudgetPerformanceHistoryRequest(input, context));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_DescribeBudgetPerformanceHistoryCommand");
var se_DescribeBudgetsCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("DescribeBudgets");
  let body;
  body = JSON.stringify((0, import_smithy_client._json)(input));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_DescribeBudgetsCommand");
var se_DescribeNotificationsForBudgetCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("DescribeNotificationsForBudget");
  let body;
  body = JSON.stringify((0, import_smithy_client._json)(input));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_DescribeNotificationsForBudgetCommand");
var se_DescribeSubscribersForNotificationCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("DescribeSubscribersForNotification");
  let body;
  body = JSON.stringify(se_DescribeSubscribersForNotificationRequest(input, context));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_DescribeSubscribersForNotificationCommand");
var se_ExecuteBudgetActionCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("ExecuteBudgetAction");
  let body;
  body = JSON.stringify((0, import_smithy_client._json)(input));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_ExecuteBudgetActionCommand");
var se_UpdateBudgetCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("UpdateBudget");
  let body;
  body = JSON.stringify(se_UpdateBudgetRequest(input, context));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_UpdateBudgetCommand");
var se_UpdateBudgetActionCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("UpdateBudgetAction");
  let body;
  body = JSON.stringify(se_UpdateBudgetActionRequest(input, context));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_UpdateBudgetActionCommand");
var se_UpdateNotificationCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("UpdateNotification");
  let body;
  body = JSON.stringify(se_UpdateNotificationRequest(input, context));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_UpdateNotificationCommand");
var se_UpdateSubscriberCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("UpdateSubscriber");
  let body;
  body = JSON.stringify(se_UpdateSubscriberRequest(input, context));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_UpdateSubscriberCommand");
var de_CreateBudgetCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const data = await parseBody(output.body, context);
  let contents = {};
  contents = (0, import_smithy_client._json)(data);
  const response = {
    $metadata: deserializeMetadata(output),
    ...contents
  };
  return response;
}, "de_CreateBudgetCommand");
var de_CreateBudgetActionCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const data = await parseBody(output.body, context);
  let contents = {};
  contents = (0, import_smithy_client._json)(data);
  const response = {
    $metadata: deserializeMetadata(output),
    ...contents
  };
  return response;
}, "de_CreateBudgetActionCommand");
var de_CreateNotificationCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const data = await parseBody(output.body, context);
  let contents = {};
  contents = (0, import_smithy_client._json)(data);
  const response = {
    $metadata: deserializeMetadata(output),
    ...contents
  };
  return response;
}, "de_CreateNotificationCommand");
var de_CreateSubscriberCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const data = await parseBody(output.body, context);
  let contents = {};
  contents = (0, import_smithy_client._json)(data);
  const response = {
    $metadata: deserializeMetadata(output),
    ...contents
  };
  return response;
}, "de_CreateSubscriberCommand");
var de_DeleteBudgetCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const data = await parseBody(output.body, context);
  let contents = {};
  contents = (0, import_smithy_client._json)(data);
  const response = {
    $metadata: deserializeMetadata(output),
    ...contents
  };
  return response;
}, "de_DeleteBudgetCommand");
var de_DeleteBudgetActionCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const data = await parseBody(output.body, context);
  let contents = {};
  contents = de_DeleteBudgetActionResponse(data, context);
  const response = {
    $metadata: deserializeMetadata(output),
    ...contents
  };
  return response;
}, "de_DeleteBudgetActionCommand");
var de_DeleteNotificationCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const data = await parseBody(output.body, context);
  let contents = {};
  contents = (0, import_smithy_client._json)(data);
  const response = {
    $metadata: deserializeMetadata(output),
    ...contents
  };
  return response;
}, "de_DeleteNotificationCommand");
var de_DeleteSubscriberCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const data = await parseBody(output.body, context);
  let contents = {};
  contents = (0, import_smithy_client._json)(data);
  const response = {
    $metadata: deserializeMetadata(output),
    ...contents
  };
  return response;
}, "de_DeleteSubscriberCommand");
var de_DescribeBudgetCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const data = await parseBody(output.body, context);
  let contents = {};
  contents = de_DescribeBudgetResponse(data, context);
  const response = {
    $metadata: deserializeMetadata(output),
    ...contents
  };
  return response;
}, "de_DescribeBudgetCommand");
var de_DescribeBudgetActionCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const data = await parseBody(output.body, context);
  let contents = {};
  contents = de_DescribeBudgetActionResponse(data, context);
  const response = {
    $metadata: deserializeMetadata(output),
    ...contents
  };
  return response;
}, "de_DescribeBudgetActionCommand");
var de_DescribeBudgetActionHistoriesCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const data = await parseBody(output.body, context);
  let contents = {};
  contents = de_DescribeBudgetActionHistoriesResponse(data, context);
  const response = {
    $metadata: deserializeMetadata(output),
    ...contents
  };
  return response;
}, "de_DescribeBudgetActionHistoriesCommand");
var de_DescribeBudgetActionsForAccountCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const data = await parseBody(output.body, context);
  let contents = {};
  contents = de_DescribeBudgetActionsForAccountResponse(data, context);
  const response = {
    $metadata: deserializeMetadata(output),
    ...contents
  };
  return response;
}, "de_DescribeBudgetActionsForAccountCommand");
var de_DescribeBudgetActionsForBudgetCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const data = await parseBody(output.body, context);
  let contents = {};
  contents = de_DescribeBudgetActionsForBudgetResponse(data, context);
  const response = {
    $metadata: deserializeMetadata(output),
    ...contents
  };
  return response;
}, "de_DescribeBudgetActionsForBudgetCommand");
var de_DescribeBudgetNotificationsForAccountCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const data = await parseBody(output.body, context);
  let contents = {};
  contents = de_DescribeBudgetNotificationsForAccountResponse(data, context);
  const response = {
    $metadata: deserializeMetadata(output),
    ...contents
  };
  return response;
}, "de_DescribeBudgetNotificationsForAccountCommand");
var de_DescribeBudgetPerformanceHistoryCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const data = await parseBody(output.body, context);
  let contents = {};
  contents = de_DescribeBudgetPerformanceHistoryResponse(data, context);
  const response = {
    $metadata: deserializeMetadata(output),
    ...contents
  };
  return response;
}, "de_DescribeBudgetPerformanceHistoryCommand");
var de_DescribeBudgetsCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const data = await parseBody(output.body, context);
  let contents = {};
  contents = de_DescribeBudgetsResponse(data, context);
  const response = {
    $metadata: deserializeMetadata(output),
    ...contents
  };
  return response;
}, "de_DescribeBudgetsCommand");
var de_DescribeNotificationsForBudgetCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const data = await parseBody(output.body, context);
  let contents = {};
  contents = de_DescribeNotificationsForBudgetResponse(data, context);
  const response = {
    $metadata: deserializeMetadata(output),
    ...contents
  };
  return response;
}, "de_DescribeNotificationsForBudgetCommand");
var de_DescribeSubscribersForNotificationCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const data = await parseBody(output.body, context);
  let contents = {};
  contents = (0, import_smithy_client._json)(data);
  const response = {
    $metadata: deserializeMetadata(output),
    ...contents
  };
  return response;
}, "de_DescribeSubscribersForNotificationCommand");
var de_ExecuteBudgetActionCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const data = await parseBody(output.body, context);
  let contents = {};
  contents = (0, import_smithy_client._json)(data);
  const response = {
    $metadata: deserializeMetadata(output),
    ...contents
  };
  return response;
}, "de_ExecuteBudgetActionCommand");
var de_UpdateBudgetCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const data = await parseBody(output.body, context);
  let contents = {};
  contents = (0, import_smithy_client._json)(data);
  const response = {
    $metadata: deserializeMetadata(output),
    ...contents
  };
  return response;
}, "de_UpdateBudgetCommand");
var de_UpdateBudgetActionCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const data = await parseBody(output.body, context);
  let contents = {};
  contents = de_UpdateBudgetActionResponse(data, context);
  const response = {
    $metadata: deserializeMetadata(output),
    ...contents
  };
  return response;
}, "de_UpdateBudgetActionCommand");
var de_UpdateNotificationCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const data = await parseBody(output.body, context);
  let contents = {};
  contents = (0, import_smithy_client._json)(data);
  const response = {
    $metadata: deserializeMetadata(output),
    ...contents
  };
  return response;
}, "de_UpdateNotificationCommand");
var de_UpdateSubscriberCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const data = await parseBody(output.body, context);
  let contents = {};
  contents = (0, import_smithy_client._json)(data);
  const response = {
    $metadata: deserializeMetadata(output),
    ...contents
  };
  return response;
}, "de_UpdateSubscriberCommand");
var de_CommandError = /* @__PURE__ */ __name(async (output, context) => {
  const parsedOutput = {
    ...output,
    body: await parseErrorBody(output.body, context)
  };
  const errorCode = loadRestJsonErrorCode(output, parsedOutput.body);
  switch (errorCode) {
    case "AccessDeniedException":
    case "com.amazonaws.budgets#AccessDeniedException":
      throw await de_AccessDeniedExceptionRes(parsedOutput, context);
    case "CreationLimitExceededException":
    case "com.amazonaws.budgets#CreationLimitExceededException":
      throw await de_CreationLimitExceededExceptionRes(parsedOutput, context);
    case "DuplicateRecordException":
    case "com.amazonaws.budgets#DuplicateRecordException":
      throw await de_DuplicateRecordExceptionRes(parsedOutput, context);
    case "InternalErrorException":
    case "com.amazonaws.budgets#InternalErrorException":
      throw await de_InternalErrorExceptionRes(parsedOutput, context);
    case "InvalidParameterException":
    case "com.amazonaws.budgets#InvalidParameterException":
      throw await de_InvalidParameterExceptionRes(parsedOutput, context);
    case "ThrottlingException":
    case "com.amazonaws.budgets#ThrottlingException":
      throw await de_ThrottlingExceptionRes(parsedOutput, context);
    case "NotFoundException":
    case "com.amazonaws.budgets#NotFoundException":
      throw await de_NotFoundExceptionRes(parsedOutput, context);
    case "ResourceLockedException":
    case "com.amazonaws.budgets#ResourceLockedException":
      throw await de_ResourceLockedExceptionRes(parsedOutput, context);
    case "InvalidNextTokenException":
    case "com.amazonaws.budgets#InvalidNextTokenException":
      throw await de_InvalidNextTokenExceptionRes(parsedOutput, context);
    case "ExpiredNextTokenException":
    case "com.amazonaws.budgets#ExpiredNextTokenException":
      throw await de_ExpiredNextTokenExceptionRes(parsedOutput, context);
    default:
      const parsedBody = parsedOutput.body;
      return throwDefaultError({
        output,
        parsedBody,
        errorCode
      });
  }
}, "de_CommandError");
var de_AccessDeniedExceptionRes = /* @__PURE__ */ __name(async (parsedOutput, context) => {
  const body = parsedOutput.body;
  const deserialized = (0, import_smithy_client._json)(body);
  const exception = new AccessDeniedException({
    $metadata: deserializeMetadata(parsedOutput),
    ...deserialized
  });
  return (0, import_smithy_client.decorateServiceException)(exception, body);
}, "de_AccessDeniedExceptionRes");
var de_CreationLimitExceededExceptionRes = /* @__PURE__ */ __name(async (parsedOutput, context) => {
  const body = parsedOutput.body;
  const deserialized = (0, import_smithy_client._json)(body);
  const exception = new CreationLimitExceededException({
    $metadata: deserializeMetadata(parsedOutput),
    ...deserialized
  });
  return (0, import_smithy_client.decorateServiceException)(exception, body);
}, "de_CreationLimitExceededExceptionRes");
var de_DuplicateRecordExceptionRes = /* @__PURE__ */ __name(async (parsedOutput, context) => {
  const body = parsedOutput.body;
  const deserialized = (0, import_smithy_client._json)(body);
  const exception = new DuplicateRecordException({
    $metadata: deserializeMetadata(parsedOutput),
    ...deserialized
  });
  return (0, import_smithy_client.decorateServiceException)(exception, body);
}, "de_DuplicateRecordExceptionRes");
var de_ExpiredNextTokenExceptionRes = /* @__PURE__ */ __name(async (parsedOutput, context) => {
  const body = parsedOutput.body;
  const deserialized = (0, import_smithy_client._json)(body);
  const exception = new ExpiredNextTokenException({
    $metadata: deserializeMetadata(parsedOutput),
    ...deserialized
  });
  return (0, import_smithy_client.decorateServiceException)(exception, body);
}, "de_ExpiredNextTokenExceptionRes");
var de_InternalErrorExceptionRes = /* @__PURE__ */ __name(async (parsedOutput, context) => {
  const body = parsedOutput.body;
  const deserialized = (0, import_smithy_client._json)(body);
  const exception = new InternalErrorException({
    $metadata: deserializeMetadata(parsedOutput),
    ...deserialized
  });
  return (0, import_smithy_client.decorateServiceException)(exception, body);
}, "de_InternalErrorExceptionRes");
var de_InvalidNextTokenExceptionRes = /* @__PURE__ */ __name(async (parsedOutput, context) => {
  const body = parsedOutput.body;
  const deserialized = (0, import_smithy_client._json)(body);
  const exception = new InvalidNextTokenException({
    $metadata: deserializeMetadata(parsedOutput),
    ...deserialized
  });
  return (0, import_smithy_client.decorateServiceException)(exception, body);
}, "de_InvalidNextTokenExceptionRes");
var de_InvalidParameterExceptionRes = /* @__PURE__ */ __name(async (parsedOutput, context) => {
  const body = parsedOutput.body;
  const deserialized = (0, import_smithy_client._json)(body);
  const exception = new InvalidParameterException({
    $metadata: deserializeMetadata(parsedOutput),
    ...deserialized
  });
  return (0, import_smithy_client.decorateServiceException)(exception, body);
}, "de_InvalidParameterExceptionRes");
var de_NotFoundExceptionRes = /* @__PURE__ */ __name(async (parsedOutput, context) => {
  const body = parsedOutput.body;
  const deserialized = (0, import_smithy_client._json)(body);
  const exception = new NotFoundException({
    $metadata: deserializeMetadata(parsedOutput),
    ...deserialized
  });
  return (0, import_smithy_client.decorateServiceException)(exception, body);
}, "de_NotFoundExceptionRes");
var de_ResourceLockedExceptionRes = /* @__PURE__ */ __name(async (parsedOutput, context) => {
  const body = parsedOutput.body;
  const deserialized = (0, import_smithy_client._json)(body);
  const exception = new ResourceLockedException({
    $metadata: deserializeMetadata(parsedOutput),
    ...deserialized
  });
  return (0, import_smithy_client.decorateServiceException)(exception, body);
}, "de_ResourceLockedExceptionRes");
var de_ThrottlingExceptionRes = /* @__PURE__ */ __name(async (parsedOutput, context) => {
  const body = parsedOutput.body;
  const deserialized = (0, import_smithy_client._json)(body);
  const exception = new ThrottlingException({
    $metadata: deserializeMetadata(parsedOutput),
    ...deserialized
  });
  return (0, import_smithy_client.decorateServiceException)(exception, body);
}, "de_ThrottlingExceptionRes");
var se_ActionThreshold = /* @__PURE__ */ __name((input, context) => {
  return (0, import_smithy_client.take)(input, {
    ActionThresholdType: [],
    ActionThresholdValue: import_smithy_client.serializeFloat
  });
}, "se_ActionThreshold");
var se_AutoAdjustData = /* @__PURE__ */ __name((input, context) => {
  return (0, import_smithy_client.take)(input, {
    AutoAdjustType: [],
    HistoricalOptions: import_smithy_client._json,
    LastAutoAdjustTime: (_) => Math.round(_.getTime() / 1e3)
  });
}, "se_AutoAdjustData");
var se_Budget = /* @__PURE__ */ __name((input, context) => {
  return (0, import_smithy_client.take)(input, {
    AutoAdjustData: (_) => se_AutoAdjustData(_, context),
    BudgetLimit: import_smithy_client._json,
    BudgetName: [],
    BudgetType: [],
    CalculatedSpend: import_smithy_client._json,
    CostFilters: import_smithy_client._json,
    CostTypes: import_smithy_client._json,
    LastUpdatedTime: (_) => Math.round(_.getTime() / 1e3),
    PlannedBudgetLimits: import_smithy_client._json,
    TimePeriod: (_) => se_TimePeriod(_, context),
    TimeUnit: []
  });
}, "se_Budget");
var se_CreateBudgetActionRequest = /* @__PURE__ */ __name((input, context) => {
  return (0, import_smithy_client.take)(input, {
    AccountId: [],
    ActionThreshold: (_) => se_ActionThreshold(_, context),
    ActionType: [],
    ApprovalModel: [],
    BudgetName: [],
    Definition: import_smithy_client._json,
    ExecutionRoleArn: [],
    NotificationType: [],
    Subscribers: import_smithy_client._json
  });
}, "se_CreateBudgetActionRequest");
var se_CreateBudgetRequest = /* @__PURE__ */ __name((input, context) => {
  return (0, import_smithy_client.take)(input, {
    AccountId: [],
    Budget: (_) => se_Budget(_, context),
    NotificationsWithSubscribers: (_) => se_NotificationWithSubscribersList(_, context)
  });
}, "se_CreateBudgetRequest");
var se_CreateNotificationRequest = /* @__PURE__ */ __name((input, context) => {
  return (0, import_smithy_client.take)(input, {
    AccountId: [],
    BudgetName: [],
    Notification: (_) => se_Notification(_, context),
    Subscribers: import_smithy_client._json
  });
}, "se_CreateNotificationRequest");
var se_CreateSubscriberRequest = /* @__PURE__ */ __name((input, context) => {
  return (0, import_smithy_client.take)(input, {
    AccountId: [],
    BudgetName: [],
    Notification: (_) => se_Notification(_, context),
    Subscriber: import_smithy_client._json
  });
}, "se_CreateSubscriberRequest");
var se_DeleteNotificationRequest = /* @__PURE__ */ __name((input, context) => {
  return (0, import_smithy_client.take)(input, {
    AccountId: [],
    BudgetName: [],
    Notification: (_) => se_Notification(_, context)
  });
}, "se_DeleteNotificationRequest");
var se_DeleteSubscriberRequest = /* @__PURE__ */ __name((input, context) => {
  return (0, import_smithy_client.take)(input, {
    AccountId: [],
    BudgetName: [],
    Notification: (_) => se_Notification(_, context),
    Subscriber: import_smithy_client._json
  });
}, "se_DeleteSubscriberRequest");
var se_DescribeBudgetActionHistoriesRequest = /* @__PURE__ */ __name((input, context) => {
  return (0, import_smithy_client.take)(input, {
    AccountId: [],
    ActionId: [],
    BudgetName: [],
    MaxResults: [],
    NextToken: [],
    TimePeriod: (_) => se_TimePeriod(_, context)
  });
}, "se_DescribeBudgetActionHistoriesRequest");
var se_DescribeBudgetPerformanceHistoryRequest = /* @__PURE__ */ __name((input, context) => {
  return (0, import_smithy_client.take)(input, {
    AccountId: [],
    BudgetName: [],
    MaxResults: [],
    NextToken: [],
    TimePeriod: (_) => se_TimePeriod(_, context)
  });
}, "se_DescribeBudgetPerformanceHistoryRequest");
var se_DescribeSubscribersForNotificationRequest = /* @__PURE__ */ __name((input, context) => {
  return (0, import_smithy_client.take)(input, {
    AccountId: [],
    BudgetName: [],
    MaxResults: [],
    NextToken: [],
    Notification: (_) => se_Notification(_, context)
  });
}, "se_DescribeSubscribersForNotificationRequest");
var se_Notification = /* @__PURE__ */ __name((input, context) => {
  return (0, import_smithy_client.take)(input, {
    ComparisonOperator: [],
    NotificationState: [],
    NotificationType: [],
    Threshold: import_smithy_client.serializeFloat,
    ThresholdType: []
  });
}, "se_Notification");
var se_NotificationWithSubscribers = /* @__PURE__ */ __name((input, context) => {
  return (0, import_smithy_client.take)(input, {
    Notification: (_) => se_Notification(_, context),
    Subscribers: import_smithy_client._json
  });
}, "se_NotificationWithSubscribers");
var se_NotificationWithSubscribersList = /* @__PURE__ */ __name((input, context) => {
  return input.filter((e) => e != null).map((entry) => {
    return se_NotificationWithSubscribers(entry, context);
  });
}, "se_NotificationWithSubscribersList");
var se_TimePeriod = /* @__PURE__ */ __name((input, context) => {
  return (0, import_smithy_client.take)(input, {
    End: (_) => Math.round(_.getTime() / 1e3),
    Start: (_) => Math.round(_.getTime() / 1e3)
  });
}, "se_TimePeriod");
var se_UpdateBudgetActionRequest = /* @__PURE__ */ __name((input, context) => {
  return (0, import_smithy_client.take)(input, {
    AccountId: [],
    ActionId: [],
    ActionThreshold: (_) => se_ActionThreshold(_, context),
    ApprovalModel: [],
    BudgetName: [],
    Definition: import_smithy_client._json,
    ExecutionRoleArn: [],
    NotificationType: [],
    Subscribers: import_smithy_client._json
  });
}, "se_UpdateBudgetActionRequest");
var se_UpdateBudgetRequest = /* @__PURE__ */ __name((input, context) => {
  return (0, import_smithy_client.take)(input, {
    AccountId: [],
    NewBudget: (_) => se_Budget(_, context)
  });
}, "se_UpdateBudgetRequest");
var se_UpdateNotificationRequest = /* @__PURE__ */ __name((input, context) => {
  return (0, import_smithy_client.take)(input, {
    AccountId: [],
    BudgetName: [],
    NewNotification: (_) => se_Notification(_, context),
    OldNotification: (_) => se_Notification(_, context)
  });
}, "se_UpdateNotificationRequest");
var se_UpdateSubscriberRequest = /* @__PURE__ */ __name((input, context) => {
  return (0, import_smithy_client.take)(input, {
    AccountId: [],
    BudgetName: [],
    NewSubscriber: import_smithy_client._json,
    Notification: (_) => se_Notification(_, context),
    OldSubscriber: import_smithy_client._json
  });
}, "se_UpdateSubscriberRequest");
var de_Action = /* @__PURE__ */ __name((output, context) => {
  return (0, import_smithy_client.take)(output, {
    ActionId: import_smithy_client.expectString,
    ActionThreshold: (_) => de_ActionThreshold(_, context),
    ActionType: import_smithy_client.expectString,
    ApprovalModel: import_smithy_client.expectString,
    BudgetName: import_smithy_client.expectString,
    Definition: import_smithy_client._json,
    ExecutionRoleArn: import_smithy_client.expectString,
    NotificationType: import_smithy_client.expectString,
    Status: import_smithy_client.expectString,
    Subscribers: import_smithy_client._json
  });
}, "de_Action");
var de_ActionHistories = /* @__PURE__ */ __name((output, context) => {
  const retVal = (output || []).filter((e) => e != null).map((entry) => {
    return de_ActionHistory(entry, context);
  });
  return retVal;
}, "de_ActionHistories");
var de_ActionHistory = /* @__PURE__ */ __name((output, context) => {
  return (0, import_smithy_client.take)(output, {
    ActionHistoryDetails: (_) => de_ActionHistoryDetails(_, context),
    EventType: import_smithy_client.expectString,
    Status: import_smithy_client.expectString,
    Timestamp: (_) => (0, import_smithy_client.expectNonNull)((0, import_smithy_client.parseEpochTimestamp)((0, import_smithy_client.expectNumber)(_)))
  });
}, "de_ActionHistory");
var de_ActionHistoryDetails = /* @__PURE__ */ __name((output, context) => {
  return (0, import_smithy_client.take)(output, {
    Action: (_) => de_Action(_, context),
    Message: import_smithy_client.expectString
  });
}, "de_ActionHistoryDetails");
var de_Actions = /* @__PURE__ */ __name((output, context) => {
  const retVal = (output || []).filter((e) => e != null).map((entry) => {
    return de_Action(entry, context);
  });
  return retVal;
}, "de_Actions");
var de_ActionThreshold = /* @__PURE__ */ __name((output, context) => {
  return (0, import_smithy_client.take)(output, {
    ActionThresholdType: import_smithy_client.expectString,
    ActionThresholdValue: import_smithy_client.limitedParseDouble
  });
}, "de_ActionThreshold");
var de_AutoAdjustData = /* @__PURE__ */ __name((output, context) => {
  return (0, import_smithy_client.take)(output, {
    AutoAdjustType: import_smithy_client.expectString,
    HistoricalOptions: import_smithy_client._json,
    LastAutoAdjustTime: (_) => (0, import_smithy_client.expectNonNull)((0, import_smithy_client.parseEpochTimestamp)((0, import_smithy_client.expectNumber)(_)))
  });
}, "de_AutoAdjustData");
var de_Budget = /* @__PURE__ */ __name((output, context) => {
  return (0, import_smithy_client.take)(output, {
    AutoAdjustData: (_) => de_AutoAdjustData(_, context),
    BudgetLimit: import_smithy_client._json,
    BudgetName: import_smithy_client.expectString,
    BudgetType: import_smithy_client.expectString,
    CalculatedSpend: import_smithy_client._json,
    CostFilters: import_smithy_client._json,
    CostTypes: import_smithy_client._json,
    LastUpdatedTime: (_) => (0, import_smithy_client.expectNonNull)((0, import_smithy_client.parseEpochTimestamp)((0, import_smithy_client.expectNumber)(_))),
    PlannedBudgetLimits: import_smithy_client._json,
    TimePeriod: (_) => de_TimePeriod(_, context),
    TimeUnit: import_smithy_client.expectString
  });
}, "de_Budget");
var de_BudgetedAndActualAmounts = /* @__PURE__ */ __name((output, context) => {
  return (0, import_smithy_client.take)(output, {
    ActualAmount: import_smithy_client._json,
    BudgetedAmount: import_smithy_client._json,
    TimePeriod: (_) => de_TimePeriod(_, context)
  });
}, "de_BudgetedAndActualAmounts");
var de_BudgetedAndActualAmountsList = /* @__PURE__ */ __name((output, context) => {
  const retVal = (output || []).filter((e) => e != null).map((entry) => {
    return de_BudgetedAndActualAmounts(entry, context);
  });
  return retVal;
}, "de_BudgetedAndActualAmountsList");
var de_BudgetNotificationsForAccount = /* @__PURE__ */ __name((output, context) => {
  return (0, import_smithy_client.take)(output, {
    BudgetName: import_smithy_client.expectString,
    Notifications: (_) => de_Notifications(_, context)
  });
}, "de_BudgetNotificationsForAccount");
var de_BudgetNotificationsForAccountList = /* @__PURE__ */ __name((output, context) => {
  const retVal = (output || []).filter((e) => e != null).map((entry) => {
    return de_BudgetNotificationsForAccount(entry, context);
  });
  return retVal;
}, "de_BudgetNotificationsForAccountList");
var de_BudgetPerformanceHistory = /* @__PURE__ */ __name((output, context) => {
  return (0, import_smithy_client.take)(output, {
    BudgetName: import_smithy_client.expectString,
    BudgetType: import_smithy_client.expectString,
    BudgetedAndActualAmountsList: (_) => de_BudgetedAndActualAmountsList(_, context),
    CostFilters: import_smithy_client._json,
    CostTypes: import_smithy_client._json,
    TimeUnit: import_smithy_client.expectString
  });
}, "de_BudgetPerformanceHistory");
var de_Budgets = /* @__PURE__ */ __name((output, context) => {
  const retVal = (output || []).filter((e) => e != null).map((entry) => {
    return de_Budget(entry, context);
  });
  return retVal;
}, "de_Budgets");
var de_DeleteBudgetActionResponse = /* @__PURE__ */ __name((output, context) => {
  return (0, import_smithy_client.take)(output, {
    AccountId: import_smithy_client.expectString,
    Action: (_) => de_Action(_, context),
    BudgetName: import_smithy_client.expectString
  });
}, "de_DeleteBudgetActionResponse");
var de_DescribeBudgetActionHistoriesResponse = /* @__PURE__ */ __name((output, context) => {
  return (0, import_smithy_client.take)(output, {
    ActionHistories: (_) => de_ActionHistories(_, context),
    NextToken: import_smithy_client.expectString
  });
}, "de_DescribeBudgetActionHistoriesResponse");
var de_DescribeBudgetActionResponse = /* @__PURE__ */ __name((output, context) => {
  return (0, import_smithy_client.take)(output, {
    AccountId: import_smithy_client.expectString,
    Action: (_) => de_Action(_, context),
    BudgetName: import_smithy_client.expectString
  });
}, "de_DescribeBudgetActionResponse");
var de_DescribeBudgetActionsForAccountResponse = /* @__PURE__ */ __name((output, context) => {
  return (0, import_smithy_client.take)(output, {
    Actions: (_) => de_Actions(_, context),
    NextToken: import_smithy_client.expectString
  });
}, "de_DescribeBudgetActionsForAccountResponse");
var de_DescribeBudgetActionsForBudgetResponse = /* @__PURE__ */ __name((output, context) => {
  return (0, import_smithy_client.take)(output, {
    Actions: (_) => de_Actions(_, context),
    NextToken: import_smithy_client.expectString
  });
}, "de_DescribeBudgetActionsForBudgetResponse");
var de_DescribeBudgetNotificationsForAccountResponse = /* @__PURE__ */ __name((output, context) => {
  return (0, import_smithy_client.take)(output, {
    BudgetNotificationsForAccount: (_) => de_BudgetNotificationsForAccountList(_, context),
    NextToken: import_smithy_client.expectString
  });
}, "de_DescribeBudgetNotificationsForAccountResponse");
var de_DescribeBudgetPerformanceHistoryResponse = /* @__PURE__ */ __name((output, context) => {
  return (0, import_smithy_client.take)(output, {
    BudgetPerformanceHistory: (_) => de_BudgetPerformanceHistory(_, context),
    NextToken: import_smithy_client.expectString
  });
}, "de_DescribeBudgetPerformanceHistoryResponse");
var de_DescribeBudgetResponse = /* @__PURE__ */ __name((output, context) => {
  return (0, import_smithy_client.take)(output, {
    Budget: (_) => de_Budget(_, context)
  });
}, "de_DescribeBudgetResponse");
var de_DescribeBudgetsResponse = /* @__PURE__ */ __name((output, context) => {
  return (0, import_smithy_client.take)(output, {
    Budgets: (_) => de_Budgets(_, context),
    NextToken: import_smithy_client.expectString
  });
}, "de_DescribeBudgetsResponse");
var de_DescribeNotificationsForBudgetResponse = /* @__PURE__ */ __name((output, context) => {
  return (0, import_smithy_client.take)(output, {
    NextToken: import_smithy_client.expectString,
    Notifications: (_) => de_Notifications(_, context)
  });
}, "de_DescribeNotificationsForBudgetResponse");
var de_Notification = /* @__PURE__ */ __name((output, context) => {
  return (0, import_smithy_client.take)(output, {
    ComparisonOperator: import_smithy_client.expectString,
    NotificationState: import_smithy_client.expectString,
    NotificationType: import_smithy_client.expectString,
    Threshold: import_smithy_client.limitedParseDouble,
    ThresholdType: import_smithy_client.expectString
  });
}, "de_Notification");
var de_Notifications = /* @__PURE__ */ __name((output, context) => {
  const retVal = (output || []).filter((e) => e != null).map((entry) => {
    return de_Notification(entry, context);
  });
  return retVal;
}, "de_Notifications");
var de_TimePeriod = /* @__PURE__ */ __name((output, context) => {
  return (0, import_smithy_client.take)(output, {
    End: (_) => (0, import_smithy_client.expectNonNull)((0, import_smithy_client.parseEpochTimestamp)((0, import_smithy_client.expectNumber)(_))),
    Start: (_) => (0, import_smithy_client.expectNonNull)((0, import_smithy_client.parseEpochTimestamp)((0, import_smithy_client.expectNumber)(_)))
  });
}, "de_TimePeriod");
var de_UpdateBudgetActionResponse = /* @__PURE__ */ __name((output, context) => {
  return (0, import_smithy_client.take)(output, {
    AccountId: import_smithy_client.expectString,
    BudgetName: import_smithy_client.expectString,
    NewAction: (_) => de_Action(_, context),
    OldAction: (_) => de_Action(_, context)
  });
}, "de_UpdateBudgetActionResponse");
var deserializeMetadata = /* @__PURE__ */ __name((output) => ({
  httpStatusCode: output.statusCode,
  requestId: output.headers["x-amzn-requestid"] ?? output.headers["x-amzn-request-id"] ?? output.headers["x-amz-request-id"],
  extendedRequestId: output.headers["x-amz-id-2"],
  cfId: output.headers["x-amz-cf-id"]
}), "deserializeMetadata");
var collectBodyString = /* @__PURE__ */ __name((streamBody, context) => (0, import_smithy_client.collectBody)(streamBody, context).then((body) => context.utf8Encoder(body)), "collectBodyString");
var throwDefaultError = (0, import_smithy_client.withBaseException)(BudgetsServiceException);
var buildHttpRpcRequest = /* @__PURE__ */ __name(async (context, headers, path, resolvedHostname, body) => {
  const { hostname, protocol = "https", port, path: basePath } = await context.endpoint();
  const contents = {
    protocol,
    hostname,
    port,
    method: "POST",
    path: basePath.endsWith("/") ? basePath.slice(0, -1) + path : basePath + path,
    headers
  };
  if (resolvedHostname !== void 0) {
    contents.hostname = resolvedHostname;
  }
  if (body !== void 0) {
    contents.body = body;
  }
  return new import_protocol_http.HttpRequest(contents);
}, "buildHttpRpcRequest");
function sharedHeaders(operation) {
  return {
    "content-type": "application/x-amz-json-1.1",
    "x-amz-target": `AWSBudgetServiceGateway.${operation}`
  };
}
__name(sharedHeaders, "sharedHeaders");
var parseBody = /* @__PURE__ */ __name((streamBody, context) => collectBodyString(streamBody, context).then((encoded) => {
  if (encoded.length) {
    return JSON.parse(encoded);
  }
  return {};
}), "parseBody");
var parseErrorBody = /* @__PURE__ */ __name(async (errorBody, context) => {
  const value = await parseBody(errorBody, context);
  value.message = value.message ?? value.Message;
  return value;
}, "parseErrorBody");
var loadRestJsonErrorCode = /* @__PURE__ */ __name((output, data) => {
  const findKey = /* @__PURE__ */ __name((object, key) => Object.keys(object).find((k) => k.toLowerCase() === key.toLowerCase()), "findKey");
  const sanitizeErrorCode = /* @__PURE__ */ __name((rawValue) => {
    let cleanValue = rawValue;
    if (typeof cleanValue === "number") {
      cleanValue = cleanValue.toString();
    }
    if (cleanValue.indexOf(",") >= 0) {
      cleanValue = cleanValue.split(",")[0];
    }
    if (cleanValue.indexOf(":") >= 0) {
      cleanValue = cleanValue.split(":")[0];
    }
    if (cleanValue.indexOf("#") >= 0) {
      cleanValue = cleanValue.split("#")[1];
    }
    return cleanValue;
  }, "sanitizeErrorCode");
  const headerKey = findKey(output.headers, "x-amzn-errortype");
  if (headerKey !== void 0) {
    return sanitizeErrorCode(output.headers[headerKey]);
  }
  if (data.code !== void 0) {
    return sanitizeErrorCode(data.code);
  }
  if (data["__type"] !== void 0) {
    return sanitizeErrorCode(data["__type"]);
  }
}, "loadRestJsonErrorCode");

// src/commands/CreateBudgetActionCommand.ts
var _CreateBudgetActionCommand = class _CreateBudgetActionCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AWSBudgetServiceGateway", "CreateBudgetAction", {}).n("BudgetsClient", "CreateBudgetActionCommand").f(CreateBudgetActionRequestFilterSensitiveLog, void 0).ser(se_CreateBudgetActionCommand).de(de_CreateBudgetActionCommand).build() {
};
__name(_CreateBudgetActionCommand, "CreateBudgetActionCommand");
var CreateBudgetActionCommand = _CreateBudgetActionCommand;

// src/commands/CreateBudgetCommand.ts




var _CreateBudgetCommand = class _CreateBudgetCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AWSBudgetServiceGateway", "CreateBudget", {}).n("BudgetsClient", "CreateBudgetCommand").f(CreateBudgetRequestFilterSensitiveLog, void 0).ser(se_CreateBudgetCommand).de(de_CreateBudgetCommand).build() {
};
__name(_CreateBudgetCommand, "CreateBudgetCommand");
var CreateBudgetCommand = _CreateBudgetCommand;

// src/commands/CreateNotificationCommand.ts




var _CreateNotificationCommand = class _CreateNotificationCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AWSBudgetServiceGateway", "CreateNotification", {}).n("BudgetsClient", "CreateNotificationCommand").f(CreateNotificationRequestFilterSensitiveLog, void 0).ser(se_CreateNotificationCommand).de(de_CreateNotificationCommand).build() {
};
__name(_CreateNotificationCommand, "CreateNotificationCommand");
var CreateNotificationCommand = _CreateNotificationCommand;

// src/commands/CreateSubscriberCommand.ts




var _CreateSubscriberCommand = class _CreateSubscriberCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AWSBudgetServiceGateway", "CreateSubscriber", {}).n("BudgetsClient", "CreateSubscriberCommand").f(CreateSubscriberRequestFilterSensitiveLog, void 0).ser(se_CreateSubscriberCommand).de(de_CreateSubscriberCommand).build() {
};
__name(_CreateSubscriberCommand, "CreateSubscriberCommand");
var CreateSubscriberCommand = _CreateSubscriberCommand;

// src/commands/DeleteBudgetActionCommand.ts




var _DeleteBudgetActionCommand = class _DeleteBudgetActionCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AWSBudgetServiceGateway", "DeleteBudgetAction", {}).n("BudgetsClient", "DeleteBudgetActionCommand").f(void 0, DeleteBudgetActionResponseFilterSensitiveLog).ser(se_DeleteBudgetActionCommand).de(de_DeleteBudgetActionCommand).build() {
};
__name(_DeleteBudgetActionCommand, "DeleteBudgetActionCommand");
var DeleteBudgetActionCommand = _DeleteBudgetActionCommand;

// src/commands/DeleteBudgetCommand.ts




var _DeleteBudgetCommand = class _DeleteBudgetCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AWSBudgetServiceGateway", "DeleteBudget", {}).n("BudgetsClient", "DeleteBudgetCommand").f(void 0, void 0).ser(se_DeleteBudgetCommand).de(de_DeleteBudgetCommand).build() {
};
__name(_DeleteBudgetCommand, "DeleteBudgetCommand");
var DeleteBudgetCommand = _DeleteBudgetCommand;

// src/commands/DeleteNotificationCommand.ts




var _DeleteNotificationCommand = class _DeleteNotificationCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AWSBudgetServiceGateway", "DeleteNotification", {}).n("BudgetsClient", "DeleteNotificationCommand").f(void 0, void 0).ser(se_DeleteNotificationCommand).de(de_DeleteNotificationCommand).build() {
};
__name(_DeleteNotificationCommand, "DeleteNotificationCommand");
var DeleteNotificationCommand = _DeleteNotificationCommand;

// src/commands/DeleteSubscriberCommand.ts




var _DeleteSubscriberCommand = class _DeleteSubscriberCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AWSBudgetServiceGateway", "DeleteSubscriber", {}).n("BudgetsClient", "DeleteSubscriberCommand").f(DeleteSubscriberRequestFilterSensitiveLog, void 0).ser(se_DeleteSubscriberCommand).de(de_DeleteSubscriberCommand).build() {
};
__name(_DeleteSubscriberCommand, "DeleteSubscriberCommand");
var DeleteSubscriberCommand = _DeleteSubscriberCommand;

// src/commands/DescribeBudgetActionCommand.ts




var _DescribeBudgetActionCommand = class _DescribeBudgetActionCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AWSBudgetServiceGateway", "DescribeBudgetAction", {}).n("BudgetsClient", "DescribeBudgetActionCommand").f(void 0, DescribeBudgetActionResponseFilterSensitiveLog).ser(se_DescribeBudgetActionCommand).de(de_DescribeBudgetActionCommand).build() {
};
__name(_DescribeBudgetActionCommand, "DescribeBudgetActionCommand");
var DescribeBudgetActionCommand = _DescribeBudgetActionCommand;

// src/commands/DescribeBudgetActionHistoriesCommand.ts




var _DescribeBudgetActionHistoriesCommand = class _DescribeBudgetActionHistoriesCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AWSBudgetServiceGateway", "DescribeBudgetActionHistories", {}).n("BudgetsClient", "DescribeBudgetActionHistoriesCommand").f(void 0, DescribeBudgetActionHistoriesResponseFilterSensitiveLog).ser(se_DescribeBudgetActionHistoriesCommand).de(de_DescribeBudgetActionHistoriesCommand).build() {
};
__name(_DescribeBudgetActionHistoriesCommand, "DescribeBudgetActionHistoriesCommand");
var DescribeBudgetActionHistoriesCommand = _DescribeBudgetActionHistoriesCommand;

// src/commands/DescribeBudgetActionsForAccountCommand.ts




var _DescribeBudgetActionsForAccountCommand = class _DescribeBudgetActionsForAccountCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AWSBudgetServiceGateway", "DescribeBudgetActionsForAccount", {}).n("BudgetsClient", "DescribeBudgetActionsForAccountCommand").f(void 0, DescribeBudgetActionsForAccountResponseFilterSensitiveLog).ser(se_DescribeBudgetActionsForAccountCommand).de(de_DescribeBudgetActionsForAccountCommand).build() {
};
__name(_DescribeBudgetActionsForAccountCommand, "DescribeBudgetActionsForAccountCommand");
var DescribeBudgetActionsForAccountCommand = _DescribeBudgetActionsForAccountCommand;

// src/commands/DescribeBudgetActionsForBudgetCommand.ts




var _DescribeBudgetActionsForBudgetCommand = class _DescribeBudgetActionsForBudgetCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AWSBudgetServiceGateway", "DescribeBudgetActionsForBudget", {}).n("BudgetsClient", "DescribeBudgetActionsForBudgetCommand").f(void 0, DescribeBudgetActionsForBudgetResponseFilterSensitiveLog).ser(se_DescribeBudgetActionsForBudgetCommand).de(de_DescribeBudgetActionsForBudgetCommand).build() {
};
__name(_DescribeBudgetActionsForBudgetCommand, "DescribeBudgetActionsForBudgetCommand");
var DescribeBudgetActionsForBudgetCommand = _DescribeBudgetActionsForBudgetCommand;

// src/commands/DescribeBudgetCommand.ts




var _DescribeBudgetCommand = class _DescribeBudgetCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AWSBudgetServiceGateway", "DescribeBudget", {}).n("BudgetsClient", "DescribeBudgetCommand").f(void 0, void 0).ser(se_DescribeBudgetCommand).de(de_DescribeBudgetCommand).build() {
};
__name(_DescribeBudgetCommand, "DescribeBudgetCommand");
var DescribeBudgetCommand = _DescribeBudgetCommand;

// src/commands/DescribeBudgetNotificationsForAccountCommand.ts




var _DescribeBudgetNotificationsForAccountCommand = class _DescribeBudgetNotificationsForAccountCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AWSBudgetServiceGateway", "DescribeBudgetNotificationsForAccount", {}).n("BudgetsClient", "DescribeBudgetNotificationsForAccountCommand").f(void 0, void 0).ser(se_DescribeBudgetNotificationsForAccountCommand).de(de_DescribeBudgetNotificationsForAccountCommand).build() {
};
__name(_DescribeBudgetNotificationsForAccountCommand, "DescribeBudgetNotificationsForAccountCommand");
var DescribeBudgetNotificationsForAccountCommand = _DescribeBudgetNotificationsForAccountCommand;

// src/commands/DescribeBudgetPerformanceHistoryCommand.ts




var _DescribeBudgetPerformanceHistoryCommand = class _DescribeBudgetPerformanceHistoryCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AWSBudgetServiceGateway", "DescribeBudgetPerformanceHistory", {}).n("BudgetsClient", "DescribeBudgetPerformanceHistoryCommand").f(void 0, void 0).ser(se_DescribeBudgetPerformanceHistoryCommand).de(de_DescribeBudgetPerformanceHistoryCommand).build() {
};
__name(_DescribeBudgetPerformanceHistoryCommand, "DescribeBudgetPerformanceHistoryCommand");
var DescribeBudgetPerformanceHistoryCommand = _DescribeBudgetPerformanceHistoryCommand;

// src/commands/DescribeBudgetsCommand.ts




var _DescribeBudgetsCommand = class _DescribeBudgetsCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AWSBudgetServiceGateway", "DescribeBudgets", {}).n("BudgetsClient", "DescribeBudgetsCommand").f(void 0, void 0).ser(se_DescribeBudgetsCommand).de(de_DescribeBudgetsCommand).build() {
};
__name(_DescribeBudgetsCommand, "DescribeBudgetsCommand");
var DescribeBudgetsCommand = _DescribeBudgetsCommand;

// src/commands/DescribeNotificationsForBudgetCommand.ts




var _DescribeNotificationsForBudgetCommand = class _DescribeNotificationsForBudgetCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AWSBudgetServiceGateway", "DescribeNotificationsForBudget", {}).n("BudgetsClient", "DescribeNotificationsForBudgetCommand").f(void 0, void 0).ser(se_DescribeNotificationsForBudgetCommand).de(de_DescribeNotificationsForBudgetCommand).build() {
};
__name(_DescribeNotificationsForBudgetCommand, "DescribeNotificationsForBudgetCommand");
var DescribeNotificationsForBudgetCommand = _DescribeNotificationsForBudgetCommand;

// src/commands/DescribeSubscribersForNotificationCommand.ts




var _DescribeSubscribersForNotificationCommand = class _DescribeSubscribersForNotificationCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AWSBudgetServiceGateway", "DescribeSubscribersForNotification", {}).n("BudgetsClient", "DescribeSubscribersForNotificationCommand").f(void 0, DescribeSubscribersForNotificationResponseFilterSensitiveLog).ser(se_DescribeSubscribersForNotificationCommand).de(de_DescribeSubscribersForNotificationCommand).build() {
};
__name(_DescribeSubscribersForNotificationCommand, "DescribeSubscribersForNotificationCommand");
var DescribeSubscribersForNotificationCommand = _DescribeSubscribersForNotificationCommand;

// src/commands/ExecuteBudgetActionCommand.ts




var _ExecuteBudgetActionCommand = class _ExecuteBudgetActionCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AWSBudgetServiceGateway", "ExecuteBudgetAction", {}).n("BudgetsClient", "ExecuteBudgetActionCommand").f(void 0, void 0).ser(se_ExecuteBudgetActionCommand).de(de_ExecuteBudgetActionCommand).build() {
};
__name(_ExecuteBudgetActionCommand, "ExecuteBudgetActionCommand");
var ExecuteBudgetActionCommand = _ExecuteBudgetActionCommand;

// src/commands/UpdateBudgetActionCommand.ts




var _UpdateBudgetActionCommand = class _UpdateBudgetActionCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AWSBudgetServiceGateway", "UpdateBudgetAction", {}).n("BudgetsClient", "UpdateBudgetActionCommand").f(UpdateBudgetActionRequestFilterSensitiveLog, UpdateBudgetActionResponseFilterSensitiveLog).ser(se_UpdateBudgetActionCommand).de(de_UpdateBudgetActionCommand).build() {
};
__name(_UpdateBudgetActionCommand, "UpdateBudgetActionCommand");
var UpdateBudgetActionCommand = _UpdateBudgetActionCommand;

// src/commands/UpdateBudgetCommand.ts




var _UpdateBudgetCommand = class _UpdateBudgetCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AWSBudgetServiceGateway", "UpdateBudget", {}).n("BudgetsClient", "UpdateBudgetCommand").f(void 0, void 0).ser(se_UpdateBudgetCommand).de(de_UpdateBudgetCommand).build() {
};
__name(_UpdateBudgetCommand, "UpdateBudgetCommand");
var UpdateBudgetCommand = _UpdateBudgetCommand;

// src/commands/UpdateNotificationCommand.ts




var _UpdateNotificationCommand = class _UpdateNotificationCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AWSBudgetServiceGateway", "UpdateNotification", {}).n("BudgetsClient", "UpdateNotificationCommand").f(void 0, void 0).ser(se_UpdateNotificationCommand).de(de_UpdateNotificationCommand).build() {
};
__name(_UpdateNotificationCommand, "UpdateNotificationCommand");
var UpdateNotificationCommand = _UpdateNotificationCommand;

// src/commands/UpdateSubscriberCommand.ts




var _UpdateSubscriberCommand = class _UpdateSubscriberCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AWSBudgetServiceGateway", "UpdateSubscriber", {}).n("BudgetsClient", "UpdateSubscriberCommand").f(UpdateSubscriberRequestFilterSensitiveLog, void 0).ser(se_UpdateSubscriberCommand).de(de_UpdateSubscriberCommand).build() {
};
__name(_UpdateSubscriberCommand, "UpdateSubscriberCommand");
var UpdateSubscriberCommand = _UpdateSubscriberCommand;

// src/Budgets.ts
var commands = {
  CreateBudgetCommand,
  CreateBudgetActionCommand,
  CreateNotificationCommand,
  CreateSubscriberCommand,
  DeleteBudgetCommand,
  DeleteBudgetActionCommand,
  DeleteNotificationCommand,
  DeleteSubscriberCommand,
  DescribeBudgetCommand,
  DescribeBudgetActionCommand,
  DescribeBudgetActionHistoriesCommand,
  DescribeBudgetActionsForAccountCommand,
  DescribeBudgetActionsForBudgetCommand,
  DescribeBudgetNotificationsForAccountCommand,
  DescribeBudgetPerformanceHistoryCommand,
  DescribeBudgetsCommand,
  DescribeNotificationsForBudgetCommand,
  DescribeSubscribersForNotificationCommand,
  ExecuteBudgetActionCommand,
  UpdateBudgetCommand,
  UpdateBudgetActionCommand,
  UpdateNotificationCommand,
  UpdateSubscriberCommand
};
var _Budgets = class _Budgets extends BudgetsClient {
};
__name(_Budgets, "Budgets");
var Budgets = _Budgets;
(0, import_smithy_client.createAggregatedClient)(commands, Budgets);

// src/pagination/DescribeBudgetActionHistoriesPaginator.ts

var paginateDescribeBudgetActionHistories = (0, import_core.createPaginator)(BudgetsClient, DescribeBudgetActionHistoriesCommand, "NextToken", "NextToken", "MaxResults");

// src/pagination/DescribeBudgetActionsForAccountPaginator.ts

var paginateDescribeBudgetActionsForAccount = (0, import_core.createPaginator)(BudgetsClient, DescribeBudgetActionsForAccountCommand, "NextToken", "NextToken", "MaxResults");

// src/pagination/DescribeBudgetActionsForBudgetPaginator.ts

var paginateDescribeBudgetActionsForBudget = (0, import_core.createPaginator)(BudgetsClient, DescribeBudgetActionsForBudgetCommand, "NextToken", "NextToken", "MaxResults");

// src/pagination/DescribeBudgetNotificationsForAccountPaginator.ts

var paginateDescribeBudgetNotificationsForAccount = (0, import_core.createPaginator)(BudgetsClient, DescribeBudgetNotificationsForAccountCommand, "NextToken", "NextToken", "MaxResults");

// src/pagination/DescribeBudgetPerformanceHistoryPaginator.ts

var paginateDescribeBudgetPerformanceHistory = (0, import_core.createPaginator)(BudgetsClient, DescribeBudgetPerformanceHistoryCommand, "NextToken", "NextToken", "MaxResults");

// src/pagination/DescribeBudgetsPaginator.ts

var paginateDescribeBudgets = (0, import_core.createPaginator)(BudgetsClient, DescribeBudgetsCommand, "NextToken", "NextToken", "MaxResults");

// src/pagination/DescribeNotificationsForBudgetPaginator.ts

var paginateDescribeNotificationsForBudget = (0, import_core.createPaginator)(BudgetsClient, DescribeNotificationsForBudgetCommand, "NextToken", "NextToken", "MaxResults");

// src/pagination/DescribeSubscribersForNotificationPaginator.ts

var paginateDescribeSubscribersForNotification = (0, import_core.createPaginator)(BudgetsClient, DescribeSubscribersForNotificationCommand, "NextToken", "NextToken", "MaxResults");

// src/index.ts
var import_util_endpoints = require("@aws-sdk/util-endpoints");
// Annotate the CommonJS export names for ESM import in node:

0 && (module.exports = {
  BudgetsServiceException,
  __Client,
  BudgetsClient,
  Budgets,
  $Command,
  CreateBudgetActionCommand,
  CreateBudgetCommand,
  CreateNotificationCommand,
  CreateSubscriberCommand,
  DeleteBudgetActionCommand,
  DeleteBudgetCommand,
  DeleteNotificationCommand,
  DeleteSubscriberCommand,
  DescribeBudgetActionCommand,
  DescribeBudgetActionHistoriesCommand,
  DescribeBudgetActionsForAccountCommand,
  DescribeBudgetActionsForBudgetCommand,
  DescribeBudgetCommand,
  DescribeBudgetNotificationsForAccountCommand,
  DescribeBudgetPerformanceHistoryCommand,
  DescribeBudgetsCommand,
  DescribeNotificationsForBudgetCommand,
  DescribeSubscribersForNotificationCommand,
  ExecuteBudgetActionCommand,
  UpdateBudgetActionCommand,
  UpdateBudgetCommand,
  UpdateNotificationCommand,
  UpdateSubscriberCommand,
  paginateDescribeBudgetActionHistories,
  paginateDescribeBudgetActionsForAccount,
  paginateDescribeBudgetActionsForBudget,
  paginateDescribeBudgetNotificationsForAccount,
  paginateDescribeBudgetPerformanceHistory,
  paginateDescribeBudgets,
  paginateDescribeNotificationsForBudget,
  paginateDescribeSubscribersForNotification,
  AccessDeniedException,
  ThresholdType,
  ActionType,
  ApprovalModel,
  ActionSubType,
  NotificationType,
  ActionStatus,
  SubscriptionType,
  EventType,
  AutoAdjustType,
  BudgetType,
  TimeUnit,
  ComparisonOperator,
  NotificationState,
  CreationLimitExceededException,
  DuplicateRecordException,
  InternalErrorException,
  InvalidParameterException,
  ThrottlingException,
  NotFoundException,
  ResourceLockedException,
  InvalidNextTokenException,
  ExpiredNextTokenException,
  ExecutionType,
  SubscriberFilterSensitiveLog,
  ActionFilterSensitiveLog,
  ActionHistoryDetailsFilterSensitiveLog,
  ActionHistoryFilterSensitiveLog,
  NotificationWithSubscribersFilterSensitiveLog,
  CreateBudgetRequestFilterSensitiveLog,
  CreateBudgetActionRequestFilterSensitiveLog,
  CreateNotificationRequestFilterSensitiveLog,
  CreateSubscriberRequestFilterSensitiveLog,
  DeleteBudgetActionResponseFilterSensitiveLog,
  DeleteSubscriberRequestFilterSensitiveLog,
  DescribeBudgetActionResponseFilterSensitiveLog,
  DescribeBudgetActionHistoriesResponseFilterSensitiveLog,
  DescribeBudgetActionsForAccountResponseFilterSensitiveLog,
  DescribeBudgetActionsForBudgetResponseFilterSensitiveLog,
  DescribeSubscribersForNotificationResponseFilterSensitiveLog,
  UpdateBudgetActionRequestFilterSensitiveLog,
  UpdateBudgetActionResponseFilterSensitiveLog,
  UpdateSubscriberRequestFilterSensitiveLog
});
