// @azure/arm-appcontainers/dist-esm/test/appcontainers_examples.spec.js
/*
 * Copyright (c) Microsoft Corporation.
 * Licensed under the MIT License.
 *
 * Code generated by Microsoft (R) AutoRest Code Generator.
 * Changes may cause incorrect behavior and will be lost if the code is regenerated.
 */
import { __asyncValues, __awaiter } from "tslib";
import { env, Recorder, isPlaybackMode, } from "@azure-tools/test-recorder";
import { createTestCredential } from "@azure-tools/test-credential";
import { assert } from "chai";
import { ContainerAppsAPIClient } from "../src/containerAppsAPIClient";
const replaceableVariables = {
    AZURE_CLIENT_ID: "azure_client_id",
    AZURE_CLIENT_SECRET: "azure_client_secret",
    AZURE_TENANT_ID: "88888888-8888-8888-8888-888888888888",
    SUBSCRIPTION_ID: "azure_subscription_id"
};
const recorderOptions = {
    envSetupForPlayback: replaceableVariables
};
export const testPollingOptions = {
    updateIntervalInMs: isPlaybackMode() ? 0 : undefined,
};
describe("AppContainer test", () => {
    let recorder;
    let subscriptionId;
    let client;
    let location;
    let resourceGroup;
    let containerAppName;
    let environmentEnvelope;
    let containerAppEnvelope;
    let environmentName;
    beforeEach(function () {
        return __awaiter(this, void 0, void 0, function* () {
            recorder = new Recorder(this.currentTest);
            yield recorder.start(recorderOptions);
            subscriptionId = env.SUBSCRIPTION_ID || '';
            // This is an example of how the environment variables are used
            const credential = createTestCredential();
            client = new ContainerAppsAPIClient(credential, subscriptionId, recorder.configureClientOptions({}));
            location = "eastus";
            resourceGroup = "myjstest";
            containerAppName = "mycontainerappxxx";
            environmentName = "testcontainerenv12";
        });
    });
    afterEach(function () {
        return __awaiter(this, void 0, void 0, function* () {
            yield recorder.stop();
        });
    });
    it("managedEnvironments create test", function () {
        return __awaiter(this, void 0, void 0, function* () {
            environmentEnvelope = {
                location: "East US",
                zoneRedundant: false
            };
            const res = yield client.managedEnvironments.beginCreateOrUpdateAndWait(resourceGroup, environmentName, environmentEnvelope, testPollingOptions);
            assert.equal(res.name, environmentName);
        });
    });
    it("containerApp create test", function () {
        return __awaiter(this, void 0, void 0, function* () {
            containerAppEnvelope = {
                "location": location,
                "managedEnvironmentId": "/subscriptions/" + subscriptionId + "/resourceGroups/" + resourceGroup + "/providers/Microsoft.App/managedEnvironments/" + environmentName,
                template: {
                    containers: [
                        {
                            name: "simple-hello-world-container",
                            image: "mcr.microsoft.com/azuredocs/containerapps-helloworld",
                            resources: {
                                cpu: 0.25,
                                memory: "0.5Gi"
                            }
                        }
                    ]
                }
            };
            const res = yield client.containerApps.beginCreateOrUpdateAndWait(resourceGroup, containerAppName, containerAppEnvelope, testPollingOptions);
            assert.equal(res.name, containerAppName);
        });
    });
    it("containerapp list Secrets test", function () {
        return __awaiter(this, void 0, void 0, function* () {
            const res = yield client.containerApps.listSecrets(resourceGroup, containerAppName);
        });
    });
    it("containerapp delete test", function () {
        var _a, e_1, _b, _c;
        return __awaiter(this, void 0, void 0, function* () {
            const res = yield client.containerApps.beginDeleteAndWait(resourceGroup, containerAppName);
            const resArray = new Array();
            try {
                for (var _d = true, _e = __asyncValues(client.containerApps.listByResourceGroup(resourceGroup)), _f; _f = yield _e.next(), _a = _f.done, !_a;) {
                    _c = _f.value;
                    _d = false;
                    try {
                        let item = _c;
                        resArray.push(item);
                    }
                    finally {
                        _d = true;
                    }
                }
            }
            catch (e_1_1) { e_1 = { error: e_1_1 }; }
            finally {
                try {
                    if (!_d && !_a && (_b = _e.return)) yield _b.call(_e);
                }
                finally { if (e_1) throw e_1.error; }
            }
            assert.equal(resArray.length, 0);
        });
    });
    it("managedEnvironments delete test", function () {
        var _a, e_2, _b, _c;
        return __awaiter(this, void 0, void 0, function* () {
            const res = yield client.managedEnvironments.beginDeleteAndWait(resourceGroup, environmentName);
            const resArray = new Array();
            try {
                for (var _d = true, _e = __asyncValues(client.managedEnvironments.listByResourceGroup(resourceGroup)), _f; _f = yield _e.next(), _a = _f.done, !_a;) {
                    _c = _f.value;
                    _d = false;
                    try {
                        let item = _c;
                        resArray.push(item);
                    }
                    finally {
                        _d = true;
                    }
                }
            }
            catch (e_2_1) { e_2 = { error: e_2_1 }; }
            finally {
                try {
                    if (!_d && !_a && (_b = _e.return)) yield _b.call(_e);
                }
                finally { if (e_2) throw e_2.error; }
            }
            assert.equal(resArray.length, 0);
        });
    });
});
