// @aws-sdk/client-grafana/dist-cjs/index.js
"use strict";
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __name = (target, value) => __defProp(target, "name", { value, configurable: true });
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// src/index.ts
var src_exports = {};
__export(src_exports, {
  AccessDeniedException: () => AccessDeniedException,
  AccountAccessType: () => AccountAccessType,
  AssociateLicenseCommand: () => AssociateLicenseCommand,
  AssociateLicenseResponseFilterSensitiveLog: () => AssociateLicenseResponseFilterSensitiveLog,
  AuthenticationDescriptionFilterSensitiveLog: () => AuthenticationDescriptionFilterSensitiveLog,
  AuthenticationProviderTypes: () => AuthenticationProviderTypes,
  ConflictException: () => ConflictException,
  CreateWorkspaceApiKeyCommand: () => CreateWorkspaceApiKeyCommand,
  CreateWorkspaceApiKeyResponseFilterSensitiveLog: () => CreateWorkspaceApiKeyResponseFilterSensitiveLog,
  CreateWorkspaceCommand: () => CreateWorkspaceCommand,
  CreateWorkspaceRequestFilterSensitiveLog: () => CreateWorkspaceRequestFilterSensitiveLog,
  CreateWorkspaceResponseFilterSensitiveLog: () => CreateWorkspaceResponseFilterSensitiveLog,
  DataSourceType: () => DataSourceType,
  DeleteWorkspaceApiKeyCommand: () => DeleteWorkspaceApiKeyCommand,
  DeleteWorkspaceCommand: () => DeleteWorkspaceCommand,
  DeleteWorkspaceResponseFilterSensitiveLog: () => DeleteWorkspaceResponseFilterSensitiveLog,
  DescribeWorkspaceAuthenticationCommand: () => DescribeWorkspaceAuthenticationCommand,
  DescribeWorkspaceAuthenticationResponseFilterSensitiveLog: () => DescribeWorkspaceAuthenticationResponseFilterSensitiveLog,
  DescribeWorkspaceCommand: () => DescribeWorkspaceCommand,
  DescribeWorkspaceConfigurationCommand: () => DescribeWorkspaceConfigurationCommand,
  DescribeWorkspaceResponseFilterSensitiveLog: () => DescribeWorkspaceResponseFilterSensitiveLog,
  DisassociateLicenseCommand: () => DisassociateLicenseCommand,
  DisassociateLicenseResponseFilterSensitiveLog: () => DisassociateLicenseResponseFilterSensitiveLog,
  Grafana: () => Grafana,
  GrafanaClient: () => GrafanaClient,
  GrafanaServiceException: () => GrafanaServiceException,
  IdpMetadata: () => IdpMetadata,
  InternalServerException: () => InternalServerException,
  LicenseType: () => LicenseType,
  ListPermissionsCommand: () => ListPermissionsCommand,
  ListTagsForResourceCommand: () => ListTagsForResourceCommand,
  ListVersionsCommand: () => ListVersionsCommand,
  ListWorkspacesCommand: () => ListWorkspacesCommand,
  ListWorkspacesResponseFilterSensitiveLog: () => ListWorkspacesResponseFilterSensitiveLog,
  NotificationDestinationType: () => NotificationDestinationType,
  PermissionType: () => PermissionType,
  ResourceNotFoundException: () => ResourceNotFoundException,
  Role: () => Role,
  RoleValuesFilterSensitiveLog: () => RoleValuesFilterSensitiveLog,
  SamlAuthenticationFilterSensitiveLog: () => SamlAuthenticationFilterSensitiveLog,
  SamlConfigurationFilterSensitiveLog: () => SamlConfigurationFilterSensitiveLog,
  SamlConfigurationStatus: () => SamlConfigurationStatus,
  ServiceQuotaExceededException: () => ServiceQuotaExceededException,
  TagResourceCommand: () => TagResourceCommand,
  ThrottlingException: () => ThrottlingException,
  UntagResourceCommand: () => UntagResourceCommand,
  UpdateAction: () => UpdateAction,
  UpdatePermissionsCommand: () => UpdatePermissionsCommand,
  UpdateWorkspaceAuthenticationCommand: () => UpdateWorkspaceAuthenticationCommand,
  UpdateWorkspaceAuthenticationRequestFilterSensitiveLog: () => UpdateWorkspaceAuthenticationRequestFilterSensitiveLog,
  UpdateWorkspaceAuthenticationResponseFilterSensitiveLog: () => UpdateWorkspaceAuthenticationResponseFilterSensitiveLog,
  UpdateWorkspaceCommand: () => UpdateWorkspaceCommand,
  UpdateWorkspaceConfigurationCommand: () => UpdateWorkspaceConfigurationCommand,
  UpdateWorkspaceRequestFilterSensitiveLog: () => UpdateWorkspaceRequestFilterSensitiveLog,
  UpdateWorkspaceResponseFilterSensitiveLog: () => UpdateWorkspaceResponseFilterSensitiveLog,
  UserType: () => UserType,
  ValidationException: () => ValidationException,
  ValidationExceptionReason: () => ValidationExceptionReason,
  WorkspaceDescriptionFilterSensitiveLog: () => WorkspaceDescriptionFilterSensitiveLog,
  WorkspaceStatus: () => WorkspaceStatus,
  WorkspaceSummaryFilterSensitiveLog: () => WorkspaceSummaryFilterSensitiveLog,
  __Client: () => import_smithy_client.Client,
  paginateListPermissions: () => paginateListPermissions,
  paginateListVersions: () => paginateListVersions,
  paginateListWorkspaces: () => paginateListWorkspaces
});
module.exports = __toCommonJS(src_exports);

// src/GrafanaClient.ts
var import_middleware_host_header = require("@aws-sdk/middleware-host-header");
var import_middleware_logger = require("@aws-sdk/middleware-logger");
var import_middleware_recursion_detection = require("@aws-sdk/middleware-recursion-detection");
var import_middleware_user_agent = require("@aws-sdk/middleware-user-agent");
var import_config_resolver = require("@smithy/config-resolver");
var import_core = require("@smithy/core");
var import_middleware_content_length = require("@smithy/middleware-content-length");
var import_middleware_endpoint = require("@smithy/middleware-endpoint");
var import_middleware_retry = require("@smithy/middleware-retry");

var import_httpAuthSchemeProvider = require("./auth/httpAuthSchemeProvider");

// src/endpoint/EndpointParameters.ts
var resolveClientEndpointParameters = /* @__PURE__ */ __name((options) => {
  return {
    ...options,
    useDualstackEndpoint: options.useDualstackEndpoint ?? false,
    useFipsEndpoint: options.useFipsEndpoint ?? false,
    defaultSigningName: "grafana"
  };
}, "resolveClientEndpointParameters");
var commonParams = {
  UseFIPS: { type: "builtInParams", name: "useFipsEndpoint" },
  Endpoint: { type: "builtInParams", name: "endpoint" },
  Region: { type: "builtInParams", name: "region" },
  UseDualStack: { type: "builtInParams", name: "useDualstackEndpoint" }
};

// src/GrafanaClient.ts
var import_runtimeConfig = require("././runtimeConfig");

// src/runtimeExtensions.ts
var import_region_config_resolver = require("@aws-sdk/region-config-resolver");
var import_protocol_http = require("@smithy/protocol-http");
var import_smithy_client = require("@smithy/smithy-client");

// src/auth/httpAuthExtensionConfiguration.ts
var getHttpAuthExtensionConfiguration = /* @__PURE__ */ __name((runtimeConfig) => {
  const _httpAuthSchemes = runtimeConfig.httpAuthSchemes;
  let _httpAuthSchemeProvider = runtimeConfig.httpAuthSchemeProvider;
  let _credentials = runtimeConfig.credentials;
  return {
    setHttpAuthScheme(httpAuthScheme) {
      const index = _httpAuthSchemes.findIndex((scheme) => scheme.schemeId === httpAuthScheme.schemeId);
      if (index === -1) {
        _httpAuthSchemes.push(httpAuthScheme);
      } else {
        _httpAuthSchemes.splice(index, 1, httpAuthScheme);
      }
    },
    httpAuthSchemes() {
      return _httpAuthSchemes;
    },
    setHttpAuthSchemeProvider(httpAuthSchemeProvider) {
      _httpAuthSchemeProvider = httpAuthSchemeProvider;
    },
    httpAuthSchemeProvider() {
      return _httpAuthSchemeProvider;
    },
    setCredentials(credentials) {
      _credentials = credentials;
    },
    credentials() {
      return _credentials;
    }
  };
}, "getHttpAuthExtensionConfiguration");
var resolveHttpAuthRuntimeConfig = /* @__PURE__ */ __name((config) => {
  return {
    httpAuthSchemes: config.httpAuthSchemes(),
    httpAuthSchemeProvider: config.httpAuthSchemeProvider(),
    credentials: config.credentials()
  };
}, "resolveHttpAuthRuntimeConfig");

// src/runtimeExtensions.ts
var asPartial = /* @__PURE__ */ __name((t) => t, "asPartial");
var resolveRuntimeExtensions = /* @__PURE__ */ __name((runtimeConfig, extensions) => {
  const extensionConfiguration = {
    ...asPartial((0, import_region_config_resolver.getAwsRegionExtensionConfiguration)(runtimeConfig)),
    ...asPartial((0, import_smithy_client.getDefaultExtensionConfiguration)(runtimeConfig)),
    ...asPartial((0, import_protocol_http.getHttpHandlerExtensionConfiguration)(runtimeConfig)),
    ...asPartial(getHttpAuthExtensionConfiguration(runtimeConfig))
  };
  extensions.forEach((extension) => extension.configure(extensionConfiguration));
  return {
    ...runtimeConfig,
    ...(0, import_region_config_resolver.resolveAwsRegionExtensionConfiguration)(extensionConfiguration),
    ...(0, import_smithy_client.resolveDefaultRuntimeConfig)(extensionConfiguration),
    ...(0, import_protocol_http.resolveHttpHandlerRuntimeConfig)(extensionConfiguration),
    ...resolveHttpAuthRuntimeConfig(extensionConfiguration)
  };
}, "resolveRuntimeExtensions");

// src/GrafanaClient.ts
var _GrafanaClient = class _GrafanaClient extends import_smithy_client.Client {
  constructor(...[configuration]) {
    const _config_0 = (0, import_runtimeConfig.getRuntimeConfig)(configuration || {});
    const _config_1 = resolveClientEndpointParameters(_config_0);
    const _config_2 = (0, import_config_resolver.resolveRegionConfig)(_config_1);
    const _config_3 = (0, import_middleware_endpoint.resolveEndpointConfig)(_config_2);
    const _config_4 = (0, import_middleware_retry.resolveRetryConfig)(_config_3);
    const _config_5 = (0, import_middleware_host_header.resolveHostHeaderConfig)(_config_4);
    const _config_6 = (0, import_middleware_user_agent.resolveUserAgentConfig)(_config_5);
    const _config_7 = (0, import_httpAuthSchemeProvider.resolveHttpAuthSchemeConfig)(_config_6);
    const _config_8 = resolveRuntimeExtensions(_config_7, (configuration == null ? void 0 : configuration.extensions) || []);
    super(_config_8);
    this.config = _config_8;
    this.middlewareStack.use((0, import_middleware_retry.getRetryPlugin)(this.config));
    this.middlewareStack.use((0, import_middleware_content_length.getContentLengthPlugin)(this.config));
    this.middlewareStack.use((0, import_middleware_host_header.getHostHeaderPlugin)(this.config));
    this.middlewareStack.use((0, import_middleware_logger.getLoggerPlugin)(this.config));
    this.middlewareStack.use((0, import_middleware_recursion_detection.getRecursionDetectionPlugin)(this.config));
    this.middlewareStack.use((0, import_middleware_user_agent.getUserAgentPlugin)(this.config));
    this.middlewareStack.use(
      (0, import_core.getHttpAuthSchemeEndpointRuleSetPlugin)(this.config, {
        httpAuthSchemeParametersProvider: this.getDefaultHttpAuthSchemeParametersProvider(),
        identityProviderConfigProvider: this.getIdentityProviderConfigProvider()
      })
    );
    this.middlewareStack.use((0, import_core.getHttpSigningPlugin)(this.config));
  }
  /**
   * Destroy underlying resources, like sockets. It's usually not necessary to do this.
   * However in Node.js, it's best to explicitly shut down the client's agent when it is no longer needed.
   * Otherwise, sockets might stay open for quite a long time before the server terminates them.
   */
  destroy() {
    super.destroy();
  }
  getDefaultHttpAuthSchemeParametersProvider() {
    return import_httpAuthSchemeProvider.defaultGrafanaHttpAuthSchemeParametersProvider;
  }
  getIdentityProviderConfigProvider() {
    return async (config) => new import_core.DefaultIdentityProviderConfig({
      "aws.auth#sigv4": config.credentials
    });
  }
};
__name(_GrafanaClient, "GrafanaClient");
var GrafanaClient = _GrafanaClient;

// src/Grafana.ts


// src/commands/AssociateLicenseCommand.ts

var import_middleware_serde = require("@smithy/middleware-serde");

var import_types = require("@smithy/types");

// src/models/models_0.ts


// src/models/GrafanaServiceException.ts

var _GrafanaServiceException = class _GrafanaServiceException extends import_smithy_client.ServiceException {
  /**
   * @internal
   */
  constructor(options) {
    super(options);
    Object.setPrototypeOf(this, _GrafanaServiceException.prototype);
  }
};
__name(_GrafanaServiceException, "GrafanaServiceException");
var GrafanaServiceException = _GrafanaServiceException;

// src/models/models_0.ts
var _AccessDeniedException = class _AccessDeniedException extends GrafanaServiceException {
  /**
   * @internal
   */
  constructor(opts) {
    super({
      name: "AccessDeniedException",
      $fault: "client",
      ...opts
    });
    this.name = "AccessDeniedException";
    this.$fault = "client";
    Object.setPrototypeOf(this, _AccessDeniedException.prototype);
  }
};
__name(_AccessDeniedException, "AccessDeniedException");
var AccessDeniedException = _AccessDeniedException;
var AccountAccessType = {
  /**
   * Indicates that the customer is using Grafana to monitor resources in their current account.
   */
  CURRENT_ACCOUNT: "CURRENT_ACCOUNT",
  /**
   * Indicates that the customer is using Grafana to monitor resources in organizational units.
   */
  ORGANIZATION: "ORGANIZATION"
};
var _ConflictException = class _ConflictException extends GrafanaServiceException {
  /**
   * @internal
   */
  constructor(opts) {
    super({
      name: "ConflictException",
      $fault: "client",
      ...opts
    });
    this.name = "ConflictException";
    this.$fault = "client";
    Object.setPrototypeOf(this, _ConflictException.prototype);
    this.resourceId = opts.resourceId;
    this.resourceType = opts.resourceType;
  }
};
__name(_ConflictException, "ConflictException");
var ConflictException = _ConflictException;
var _InternalServerException = class _InternalServerException extends GrafanaServiceException {
  /**
   * @internal
   */
  constructor(opts) {
    super({
      name: "InternalServerException",
      $fault: "server",
      ...opts
    });
    this.name = "InternalServerException";
    this.$fault = "server";
    this.$retryable = {};
    Object.setPrototypeOf(this, _InternalServerException.prototype);
    this.retryAfterSeconds = opts.retryAfterSeconds;
  }
};
__name(_InternalServerException, "InternalServerException");
var InternalServerException = _InternalServerException;
var _ResourceNotFoundException = class _ResourceNotFoundException extends GrafanaServiceException {
  /**
   * @internal
   */
  constructor(opts) {
    super({
      name: "ResourceNotFoundException",
      $fault: "client",
      ...opts
    });
    this.name = "ResourceNotFoundException";
    this.$fault = "client";
    Object.setPrototypeOf(this, _ResourceNotFoundException.prototype);
    this.resourceId = opts.resourceId;
    this.resourceType = opts.resourceType;
  }
};
__name(_ResourceNotFoundException, "ResourceNotFoundException");
var ResourceNotFoundException = _ResourceNotFoundException;
var _ServiceQuotaExceededException = class _ServiceQuotaExceededException extends GrafanaServiceException {
  /**
   * @internal
   */
  constructor(opts) {
    super({
      name: "ServiceQuotaExceededException",
      $fault: "client",
      ...opts
    });
    this.name = "ServiceQuotaExceededException";
    this.$fault = "client";
    Object.setPrototypeOf(this, _ServiceQuotaExceededException.prototype);
    this.resourceId = opts.resourceId;
    this.resourceType = opts.resourceType;
    this.serviceCode = opts.serviceCode;
    this.quotaCode = opts.quotaCode;
  }
};
__name(_ServiceQuotaExceededException, "ServiceQuotaExceededException");
var ServiceQuotaExceededException = _ServiceQuotaExceededException;
var _ThrottlingException = class _ThrottlingException extends GrafanaServiceException {
  /**
   * @internal
   */
  constructor(opts) {
    super({
      name: "ThrottlingException",
      $fault: "client",
      ...opts
    });
    this.name = "ThrottlingException";
    this.$fault = "client";
    this.$retryable = {};
    Object.setPrototypeOf(this, _ThrottlingException.prototype);
    this.serviceCode = opts.serviceCode;
    this.quotaCode = opts.quotaCode;
    this.retryAfterSeconds = opts.retryAfterSeconds;
  }
};
__name(_ThrottlingException, "ThrottlingException");
var ThrottlingException = _ThrottlingException;
var ValidationExceptionReason = {
  CANNOT_PARSE: "CANNOT_PARSE",
  FIELD_VALIDATION_FAILED: "FIELD_VALIDATION_FAILED",
  OTHER: "OTHER",
  UNKNOWN_OPERATION: "UNKNOWN_OPERATION"
};
var _ValidationException = class _ValidationException extends GrafanaServiceException {
  /**
   * @internal
   */
  constructor(opts) {
    super({
      name: "ValidationException",
      $fault: "client",
      ...opts
    });
    this.name = "ValidationException";
    this.$fault = "client";
    Object.setPrototypeOf(this, _ValidationException.prototype);
    this.reason = opts.reason;
    this.fieldList = opts.fieldList;
  }
};
__name(_ValidationException, "ValidationException");
var ValidationException = _ValidationException;
var LicenseType = {
  /**
   * Grafana Enterprise License.
   */
  ENTERPRISE: "ENTERPRISE",
  /**
   * Grafana Enterprise Free Trial License.
   */
  ENTERPRISE_FREE_TRIAL: "ENTERPRISE_FREE_TRIAL"
};
var AuthenticationProviderTypes = {
  /**
   * Indicates that AMG workspace has AWS SSO enabled as its authentication provider.
   */
  AWS_SSO: "AWS_SSO",
  /**
   * Indicates that the AMG workspace has SAML enabled as its authentication provider.
   */
  SAML: "SAML"
};
var SamlConfigurationStatus = {
  /**
   * Indicates that SAML on an AMG workspace is enabled and has been configured.
   */
  CONFIGURED: "CONFIGURED",
  /**
   * Indicates that SAML on an AMG workspace is enabled but has not been configured.
   */
  NOT_CONFIGURED: "NOT_CONFIGURED"
};
var DataSourceType = {
  /**
   * Amazon OpenSearch Service
   */
  AMAZON_OPENSEARCH_SERVICE: "AMAZON_OPENSEARCH_SERVICE",
  /**
   * Amazon Athena
   */
  ATHENA: "ATHENA",
  /**
   * CloudWatch Logs
   */
  CLOUDWATCH: "CLOUDWATCH",
  /**
   * Managed Prometheus
   */
  PROMETHEUS: "PROMETHEUS",
  /**
   * Redshift
   */
  REDSHIFT: "REDSHIFT",
  /**
   * IoT SiteWise
   */
  SITEWISE: "SITEWISE",
  /**
   * Timestream
   */
  TIMESTREAM: "TIMESTREAM",
  /**
   * IoT TwinMaker
   */
  TWINMAKER: "TWINMAKER",
  /**
   * X-Ray
   */
  XRAY: "XRAY"
};
var NotificationDestinationType = {
  /**
   * AWS Simple Notification Service
   */
  SNS: "SNS"
};
var PermissionType = {
  /**
   * Customer Managed
   */
  CUSTOMER_MANAGED: "CUSTOMER_MANAGED",
  /**
   * Service Managed
   */
  SERVICE_MANAGED: "SERVICE_MANAGED"
};
var WorkspaceStatus = {
  /**
   * Workspace is active.
   */
  ACTIVE: "ACTIVE",
  /**
   * Workspace is being created.
   */
  CREATING: "CREATING",
  /**
   * Workspace creation failed.
   */
  CREATION_FAILED: "CREATION_FAILED",
  /**
   * Workspace is being deleted.
   */
  DELETING: "DELETING",
  /**
   * Workspace deletion failed.
   */
  DELETION_FAILED: "DELETION_FAILED",
  /**
   * Workspace is in an invalid state, it can only and should be deleted.
   */
  FAILED: "FAILED",
  /**
   * Failed to remove enterprise license from workspace.
   */
  LICENSE_REMOVAL_FAILED: "LICENSE_REMOVAL_FAILED",
  /**
   * Workspace update failed.
   */
  UPDATE_FAILED: "UPDATE_FAILED",
  /**
   * Workspace is being updated.
   */
  UPDATING: "UPDATING",
  /**
   * Workspace upgrade failed.
   */
  UPGRADE_FAILED: "UPGRADE_FAILED",
  /**
   * Workspace is being upgraded to enterprise.
   */
  UPGRADING: "UPGRADING",
  /**
   * Workspace version update failed.
   */
  VERSION_UPDATE_FAILED: "VERSION_UPDATE_FAILED",
  /**
   * Workspace version is being updated.
   */
  VERSION_UPDATING: "VERSION_UPDATING"
};
var IdpMetadata;
((IdpMetadata3) => {
  IdpMetadata3.visit = /* @__PURE__ */ __name((value, visitor) => {
    if (value.url !== void 0)
      return visitor.url(value.url);
    if (value.xml !== void 0)
      return visitor.xml(value.xml);
    return visitor._(value.$unknown[0], value.$unknown[1]);
  }, "visit");
})(IdpMetadata || (IdpMetadata = {}));
var UserType = {
  /**
   * SSO group.
   */
  SSO_GROUP: "SSO_GROUP",
  /**
   * SSO user.
   */
  SSO_USER: "SSO_USER"
};
var Role = {
  /**
   * Role Admin.
   */
  ADMIN: "ADMIN",
  /**
   * Role Editor.
   */
  EDITOR: "EDITOR",
  /**
   * Role Viewer.
   */
  VIEWER: "VIEWER"
};
var UpdateAction = {
  /**
   * Add permissions.
   */
  ADD: "ADD",
  /**
   * Revoke permissions.
   */
  REVOKE: "REVOKE"
};
var CreateWorkspaceApiKeyResponseFilterSensitiveLog = /* @__PURE__ */ __name((obj) => ({
  ...obj,
  ...obj.key && { key: import_smithy_client.SENSITIVE_STRING }
}), "CreateWorkspaceApiKeyResponseFilterSensitiveLog");
var WorkspaceDescriptionFilterSensitiveLog = /* @__PURE__ */ __name((obj) => ({
  ...obj,
  ...obj.description && { description: import_smithy_client.SENSITIVE_STRING },
  ...obj.name && { name: import_smithy_client.SENSITIVE_STRING },
  ...obj.organizationRoleName && { organizationRoleName: import_smithy_client.SENSITIVE_STRING },
  ...obj.organizationalUnits && { organizationalUnits: import_smithy_client.SENSITIVE_STRING },
  ...obj.workspaceRoleArn && { workspaceRoleArn: import_smithy_client.SENSITIVE_STRING }
}), "WorkspaceDescriptionFilterSensitiveLog");
var AssociateLicenseResponseFilterSensitiveLog = /* @__PURE__ */ __name((obj) => ({
  ...obj,
  ...obj.workspace && { workspace: WorkspaceDescriptionFilterSensitiveLog(obj.workspace) }
}), "AssociateLicenseResponseFilterSensitiveLog");
var RoleValuesFilterSensitiveLog = /* @__PURE__ */ __name((obj) => ({
  ...obj,
  ...obj.editor && { editor: import_smithy_client.SENSITIVE_STRING },
  ...obj.admin && { admin: import_smithy_client.SENSITIVE_STRING }
}), "RoleValuesFilterSensitiveLog");
var SamlConfigurationFilterSensitiveLog = /* @__PURE__ */ __name((obj) => ({
  ...obj,
  ...obj.idpMetadata && { idpMetadata: obj.idpMetadata },
  ...obj.roleValues && { roleValues: RoleValuesFilterSensitiveLog(obj.roleValues) }
}), "SamlConfigurationFilterSensitiveLog");
var SamlAuthenticationFilterSensitiveLog = /* @__PURE__ */ __name((obj) => ({
  ...obj,
  ...obj.configuration && { configuration: SamlConfigurationFilterSensitiveLog(obj.configuration) }
}), "SamlAuthenticationFilterSensitiveLog");
var AuthenticationDescriptionFilterSensitiveLog = /* @__PURE__ */ __name((obj) => ({
  ...obj,
  ...obj.saml && { saml: SamlAuthenticationFilterSensitiveLog(obj.saml) }
}), "AuthenticationDescriptionFilterSensitiveLog");
var DescribeWorkspaceAuthenticationResponseFilterSensitiveLog = /* @__PURE__ */ __name((obj) => ({
  ...obj,
  ...obj.authentication && { authentication: AuthenticationDescriptionFilterSensitiveLog(obj.authentication) }
}), "DescribeWorkspaceAuthenticationResponseFilterSensitiveLog");
var UpdateWorkspaceAuthenticationRequestFilterSensitiveLog = /* @__PURE__ */ __name((obj) => ({
  ...obj,
  ...obj.samlConfiguration && { samlConfiguration: SamlConfigurationFilterSensitiveLog(obj.samlConfiguration) }
}), "UpdateWorkspaceAuthenticationRequestFilterSensitiveLog");
var UpdateWorkspaceAuthenticationResponseFilterSensitiveLog = /* @__PURE__ */ __name((obj) => ({
  ...obj,
  ...obj.authentication && { authentication: AuthenticationDescriptionFilterSensitiveLog(obj.authentication) }
}), "UpdateWorkspaceAuthenticationResponseFilterSensitiveLog");
var DisassociateLicenseResponseFilterSensitiveLog = /* @__PURE__ */ __name((obj) => ({
  ...obj,
  ...obj.workspace && { workspace: WorkspaceDescriptionFilterSensitiveLog(obj.workspace) }
}), "DisassociateLicenseResponseFilterSensitiveLog");
var CreateWorkspaceRequestFilterSensitiveLog = /* @__PURE__ */ __name((obj) => ({
  ...obj,
  ...obj.organizationRoleName && { organizationRoleName: import_smithy_client.SENSITIVE_STRING },
  ...obj.workspaceDescription && { workspaceDescription: import_smithy_client.SENSITIVE_STRING },
  ...obj.workspaceName && { workspaceName: import_smithy_client.SENSITIVE_STRING },
  ...obj.workspaceOrganizationalUnits && { workspaceOrganizationalUnits: import_smithy_client.SENSITIVE_STRING },
  ...obj.workspaceRoleArn && { workspaceRoleArn: import_smithy_client.SENSITIVE_STRING }
}), "CreateWorkspaceRequestFilterSensitiveLog");
var CreateWorkspaceResponseFilterSensitiveLog = /* @__PURE__ */ __name((obj) => ({
  ...obj,
  ...obj.workspace && { workspace: WorkspaceDescriptionFilterSensitiveLog(obj.workspace) }
}), "CreateWorkspaceResponseFilterSensitiveLog");
var DeleteWorkspaceResponseFilterSensitiveLog = /* @__PURE__ */ __name((obj) => ({
  ...obj,
  ...obj.workspace && { workspace: WorkspaceDescriptionFilterSensitiveLog(obj.workspace) }
}), "DeleteWorkspaceResponseFilterSensitiveLog");
var DescribeWorkspaceResponseFilterSensitiveLog = /* @__PURE__ */ __name((obj) => ({
  ...obj,
  ...obj.workspace && { workspace: WorkspaceDescriptionFilterSensitiveLog(obj.workspace) }
}), "DescribeWorkspaceResponseFilterSensitiveLog");
var WorkspaceSummaryFilterSensitiveLog = /* @__PURE__ */ __name((obj) => ({
  ...obj,
  ...obj.description && { description: import_smithy_client.SENSITIVE_STRING },
  ...obj.name && { name: import_smithy_client.SENSITIVE_STRING }
}), "WorkspaceSummaryFilterSensitiveLog");
var ListWorkspacesResponseFilterSensitiveLog = /* @__PURE__ */ __name((obj) => ({
  ...obj,
  ...obj.workspaces && { workspaces: obj.workspaces.map((item) => WorkspaceSummaryFilterSensitiveLog(item)) }
}), "ListWorkspacesResponseFilterSensitiveLog");
var UpdateWorkspaceRequestFilterSensitiveLog = /* @__PURE__ */ __name((obj) => ({
  ...obj,
  ...obj.organizationRoleName && { organizationRoleName: import_smithy_client.SENSITIVE_STRING },
  ...obj.workspaceDescription && { workspaceDescription: import_smithy_client.SENSITIVE_STRING },
  ...obj.workspaceName && { workspaceName: import_smithy_client.SENSITIVE_STRING },
  ...obj.workspaceOrganizationalUnits && { workspaceOrganizationalUnits: import_smithy_client.SENSITIVE_STRING },
  ...obj.workspaceRoleArn && { workspaceRoleArn: import_smithy_client.SENSITIVE_STRING }
}), "UpdateWorkspaceRequestFilterSensitiveLog");
var UpdateWorkspaceResponseFilterSensitiveLog = /* @__PURE__ */ __name((obj) => ({
  ...obj,
  ...obj.workspace && { workspace: WorkspaceDescriptionFilterSensitiveLog(obj.workspace) }
}), "UpdateWorkspaceResponseFilterSensitiveLog");

// src/protocols/Aws_restJson1.ts
var import_core2 = require("@aws-sdk/core");


var import_uuid = require("uuid");
var se_AssociateLicenseCommand = /* @__PURE__ */ __name(async (input, context) => {
  const b = (0, import_core.requestBuilder)(input, context);
  const headers = (0, import_smithy_client.map)({}, isSerializableHeaderValue, {
    [_gt]: input[_gT]
  });
  b.bp("/workspaces/{workspaceId}/licenses/{licenseType}");
  b.p("workspaceId", () => input.workspaceId, "{workspaceId}", false);
  b.p("licenseType", () => input.licenseType, "{licenseType}", false);
  let body;
  b.m("POST").h(headers).b(body);
  return b.build();
}, "se_AssociateLicenseCommand");
var se_CreateWorkspaceCommand = /* @__PURE__ */ __name(async (input, context) => {
  const b = (0, import_core.requestBuilder)(input, context);
  const headers = {
    "content-type": "application/json"
  };
  b.bp("/workspaces");
  let body;
  body = JSON.stringify(
    (0, import_smithy_client.take)(input, {
      accountAccessType: [],
      authenticationProviders: (_) => (0, import_smithy_client._json)(_),
      clientToken: [true, (_) => _ ?? (0, import_uuid.v4)()],
      configuration: (_) => import_smithy_client.LazyJsonString.fromObject(_),
      grafanaVersion: [],
      networkAccessControl: (_) => (0, import_smithy_client._json)(_),
      organizationRoleName: [],
      permissionType: [],
      stackSetName: [],
      tags: (_) => (0, import_smithy_client._json)(_),
      vpcConfiguration: (_) => (0, import_smithy_client._json)(_),
      workspaceDataSources: (_) => (0, import_smithy_client._json)(_),
      workspaceDescription: [],
      workspaceName: [],
      workspaceNotificationDestinations: (_) => (0, import_smithy_client._json)(_),
      workspaceOrganizationalUnits: (_) => (0, import_smithy_client._json)(_),
      workspaceRoleArn: []
    })
  );
  b.m("POST").h(headers).b(body);
  return b.build();
}, "se_CreateWorkspaceCommand");
var se_CreateWorkspaceApiKeyCommand = /* @__PURE__ */ __name(async (input, context) => {
  const b = (0, import_core.requestBuilder)(input, context);
  const headers = {
    "content-type": "application/json"
  };
  b.bp("/workspaces/{workspaceId}/apikeys");
  b.p("workspaceId", () => input.workspaceId, "{workspaceId}", false);
  let body;
  body = JSON.stringify(
    (0, import_smithy_client.take)(input, {
      keyName: [],
      keyRole: [],
      secondsToLive: []
    })
  );
  b.m("POST").h(headers).b(body);
  return b.build();
}, "se_CreateWorkspaceApiKeyCommand");
var se_DeleteWorkspaceCommand = /* @__PURE__ */ __name(async (input, context) => {
  const b = (0, import_core.requestBuilder)(input, context);
  const headers = {};
  b.bp("/workspaces/{workspaceId}");
  b.p("workspaceId", () => input.workspaceId, "{workspaceId}", false);
  let body;
  b.m("DELETE").h(headers).b(body);
  return b.build();
}, "se_DeleteWorkspaceCommand");
var se_DeleteWorkspaceApiKeyCommand = /* @__PURE__ */ __name(async (input, context) => {
  const b = (0, import_core.requestBuilder)(input, context);
  const headers = {};
  b.bp("/workspaces/{workspaceId}/apikeys/{keyName}");
  b.p("keyName", () => input.keyName, "{keyName}", false);
  b.p("workspaceId", () => input.workspaceId, "{workspaceId}", false);
  let body;
  b.m("DELETE").h(headers).b(body);
  return b.build();
}, "se_DeleteWorkspaceApiKeyCommand");
var se_DescribeWorkspaceCommand = /* @__PURE__ */ __name(async (input, context) => {
  const b = (0, import_core.requestBuilder)(input, context);
  const headers = {};
  b.bp("/workspaces/{workspaceId}");
  b.p("workspaceId", () => input.workspaceId, "{workspaceId}", false);
  let body;
  b.m("GET").h(headers).b(body);
  return b.build();
}, "se_DescribeWorkspaceCommand");
var se_DescribeWorkspaceAuthenticationCommand = /* @__PURE__ */ __name(async (input, context) => {
  const b = (0, import_core.requestBuilder)(input, context);
  const headers = {};
  b.bp("/workspaces/{workspaceId}/authentication");
  b.p("workspaceId", () => input.workspaceId, "{workspaceId}", false);
  let body;
  b.m("GET").h(headers).b(body);
  return b.build();
}, "se_DescribeWorkspaceAuthenticationCommand");
var se_DescribeWorkspaceConfigurationCommand = /* @__PURE__ */ __name(async (input, context) => {
  const b = (0, import_core.requestBuilder)(input, context);
  const headers = {};
  b.bp("/workspaces/{workspaceId}/configuration");
  b.p("workspaceId", () => input.workspaceId, "{workspaceId}", false);
  let body;
  b.m("GET").h(headers).b(body);
  return b.build();
}, "se_DescribeWorkspaceConfigurationCommand");
var se_DisassociateLicenseCommand = /* @__PURE__ */ __name(async (input, context) => {
  const b = (0, import_core.requestBuilder)(input, context);
  const headers = {};
  b.bp("/workspaces/{workspaceId}/licenses/{licenseType}");
  b.p("workspaceId", () => input.workspaceId, "{workspaceId}", false);
  b.p("licenseType", () => input.licenseType, "{licenseType}", false);
  let body;
  b.m("DELETE").h(headers).b(body);
  return b.build();
}, "se_DisassociateLicenseCommand");
var se_ListPermissionsCommand = /* @__PURE__ */ __name(async (input, context) => {
  const b = (0, import_core.requestBuilder)(input, context);
  const headers = {};
  b.bp("/workspaces/{workspaceId}/permissions");
  b.p("workspaceId", () => input.workspaceId, "{workspaceId}", false);
  const query = (0, import_smithy_client.map)({
    [_mR]: [() => input.maxResults !== void 0, () => input[_mR].toString()],
    [_nT]: [, input[_nT]],
    [_uT]: [, input[_uT]],
    [_uI]: [, input[_uI]],
    [_gI]: [, input[_gI]]
  });
  let body;
  b.m("GET").h(headers).q(query).b(body);
  return b.build();
}, "se_ListPermissionsCommand");
var se_ListTagsForResourceCommand = /* @__PURE__ */ __name(async (input, context) => {
  const b = (0, import_core.requestBuilder)(input, context);
  const headers = {};
  b.bp("/tags/{resourceArn}");
  b.p("resourceArn", () => input.resourceArn, "{resourceArn}", false);
  let body;
  b.m("GET").h(headers).b(body);
  return b.build();
}, "se_ListTagsForResourceCommand");
var se_ListVersionsCommand = /* @__PURE__ */ __name(async (input, context) => {
  const b = (0, import_core.requestBuilder)(input, context);
  const headers = {};
  b.bp("/versions");
  const query = (0, import_smithy_client.map)({
    [_mR]: [() => input.maxResults !== void 0, () => input[_mR].toString()],
    [_nT]: [, input[_nT]],
    [_wi]: [, input[_wI]]
  });
  let body;
  b.m("GET").h(headers).q(query).b(body);
  return b.build();
}, "se_ListVersionsCommand");
var se_ListWorkspacesCommand = /* @__PURE__ */ __name(async (input, context) => {
  const b = (0, import_core.requestBuilder)(input, context);
  const headers = {};
  b.bp("/workspaces");
  const query = (0, import_smithy_client.map)({
    [_mR]: [() => input.maxResults !== void 0, () => input[_mR].toString()],
    [_nT]: [, input[_nT]]
  });
  let body;
  b.m("GET").h(headers).q(query).b(body);
  return b.build();
}, "se_ListWorkspacesCommand");
var se_TagResourceCommand = /* @__PURE__ */ __name(async (input, context) => {
  const b = (0, import_core.requestBuilder)(input, context);
  const headers = {
    "content-type": "application/json"
  };
  b.bp("/tags/{resourceArn}");
  b.p("resourceArn", () => input.resourceArn, "{resourceArn}", false);
  let body;
  body = JSON.stringify(
    (0, import_smithy_client.take)(input, {
      tags: (_) => (0, import_smithy_client._json)(_)
    })
  );
  b.m("POST").h(headers).b(body);
  return b.build();
}, "se_TagResourceCommand");
var se_UntagResourceCommand = /* @__PURE__ */ __name(async (input, context) => {
  const b = (0, import_core.requestBuilder)(input, context);
  const headers = {};
  b.bp("/tags/{resourceArn}");
  b.p("resourceArn", () => input.resourceArn, "{resourceArn}", false);
  const query = (0, import_smithy_client.map)({
    [_tK]: [
      (0, import_smithy_client.expectNonNull)(input.tagKeys, `tagKeys`) != null,
      () => (input[_tK] || []).map((_entry) => _entry)
    ]
  });
  let body;
  b.m("DELETE").h(headers).q(query).b(body);
  return b.build();
}, "se_UntagResourceCommand");
var se_UpdatePermissionsCommand = /* @__PURE__ */ __name(async (input, context) => {
  const b = (0, import_core.requestBuilder)(input, context);
  const headers = {
    "content-type": "application/json"
  };
  b.bp("/workspaces/{workspaceId}/permissions");
  b.p("workspaceId", () => input.workspaceId, "{workspaceId}", false);
  let body;
  body = JSON.stringify(
    (0, import_smithy_client.take)(input, {
      updateInstructionBatch: (_) => (0, import_smithy_client._json)(_)
    })
  );
  b.m("PATCH").h(headers).b(body);
  return b.build();
}, "se_UpdatePermissionsCommand");
var se_UpdateWorkspaceCommand = /* @__PURE__ */ __name(async (input, context) => {
  const b = (0, import_core.requestBuilder)(input, context);
  const headers = {
    "content-type": "application/json"
  };
  b.bp("/workspaces/{workspaceId}");
  b.p("workspaceId", () => input.workspaceId, "{workspaceId}", false);
  let body;
  body = JSON.stringify(
    (0, import_smithy_client.take)(input, {
      accountAccessType: [],
      networkAccessControl: (_) => (0, import_smithy_client._json)(_),
      organizationRoleName: [],
      permissionType: [],
      removeNetworkAccessConfiguration: [],
      removeVpcConfiguration: [],
      stackSetName: [],
      vpcConfiguration: (_) => (0, import_smithy_client._json)(_),
      workspaceDataSources: (_) => (0, import_smithy_client._json)(_),
      workspaceDescription: [],
      workspaceName: [],
      workspaceNotificationDestinations: (_) => (0, import_smithy_client._json)(_),
      workspaceOrganizationalUnits: (_) => (0, import_smithy_client._json)(_),
      workspaceRoleArn: []
    })
  );
  b.m("PUT").h(headers).b(body);
  return b.build();
}, "se_UpdateWorkspaceCommand");
var se_UpdateWorkspaceAuthenticationCommand = /* @__PURE__ */ __name(async (input, context) => {
  const b = (0, import_core.requestBuilder)(input, context);
  const headers = {
    "content-type": "application/json"
  };
  b.bp("/workspaces/{workspaceId}/authentication");
  b.p("workspaceId", () => input.workspaceId, "{workspaceId}", false);
  let body;
  body = JSON.stringify(
    (0, import_smithy_client.take)(input, {
      authenticationProviders: (_) => (0, import_smithy_client._json)(_),
      samlConfiguration: (_) => (0, import_smithy_client._json)(_)
    })
  );
  b.m("POST").h(headers).b(body);
  return b.build();
}, "se_UpdateWorkspaceAuthenticationCommand");
var se_UpdateWorkspaceConfigurationCommand = /* @__PURE__ */ __name(async (input, context) => {
  const b = (0, import_core.requestBuilder)(input, context);
  const headers = {
    "content-type": "application/json"
  };
  b.bp("/workspaces/{workspaceId}/configuration");
  b.p("workspaceId", () => input.workspaceId, "{workspaceId}", false);
  let body;
  body = JSON.stringify(
    (0, import_smithy_client.take)(input, {
      configuration: (_) => import_smithy_client.LazyJsonString.fromObject(_),
      grafanaVersion: []
    })
  );
  b.m("PUT").h(headers).b(body);
  return b.build();
}, "se_UpdateWorkspaceConfigurationCommand");
var de_AssociateLicenseCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode !== 202 && output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const contents = (0, import_smithy_client.map)({
    $metadata: deserializeMetadata(output)
  });
  const data = (0, import_smithy_client.expectNonNull)((0, import_smithy_client.expectObject)(await (0, import_core2.parseJsonBody)(output.body, context)), "body");
  const doc = (0, import_smithy_client.take)(data, {
    workspace: (_) => de_WorkspaceDescription(_, context)
  });
  Object.assign(contents, doc);
  return contents;
}, "de_AssociateLicenseCommand");
var de_CreateWorkspaceCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode !== 202 && output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const contents = (0, import_smithy_client.map)({
    $metadata: deserializeMetadata(output)
  });
  const data = (0, import_smithy_client.expectNonNull)((0, import_smithy_client.expectObject)(await (0, import_core2.parseJsonBody)(output.body, context)), "body");
  const doc = (0, import_smithy_client.take)(data, {
    workspace: (_) => de_WorkspaceDescription(_, context)
  });
  Object.assign(contents, doc);
  return contents;
}, "de_CreateWorkspaceCommand");
var de_CreateWorkspaceApiKeyCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode !== 200 && output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const contents = (0, import_smithy_client.map)({
    $metadata: deserializeMetadata(output)
  });
  const data = (0, import_smithy_client.expectNonNull)((0, import_smithy_client.expectObject)(await (0, import_core2.parseJsonBody)(output.body, context)), "body");
  const doc = (0, import_smithy_client.take)(data, {
    key: import_smithy_client.expectString,
    keyName: import_smithy_client.expectString,
    workspaceId: import_smithy_client.expectString
  });
  Object.assign(contents, doc);
  return contents;
}, "de_CreateWorkspaceApiKeyCommand");
var de_DeleteWorkspaceCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode !== 202 && output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const contents = (0, import_smithy_client.map)({
    $metadata: deserializeMetadata(output)
  });
  const data = (0, import_smithy_client.expectNonNull)((0, import_smithy_client.expectObject)(await (0, import_core2.parseJsonBody)(output.body, context)), "body");
  const doc = (0, import_smithy_client.take)(data, {
    workspace: (_) => de_WorkspaceDescription(_, context)
  });
  Object.assign(contents, doc);
  return contents;
}, "de_DeleteWorkspaceCommand");
var de_DeleteWorkspaceApiKeyCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode !== 200 && output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const contents = (0, import_smithy_client.map)({
    $metadata: deserializeMetadata(output)
  });
  const data = (0, import_smithy_client.expectNonNull)((0, import_smithy_client.expectObject)(await (0, import_core2.parseJsonBody)(output.body, context)), "body");
  const doc = (0, import_smithy_client.take)(data, {
    keyName: import_smithy_client.expectString,
    workspaceId: import_smithy_client.expectString
  });
  Object.assign(contents, doc);
  return contents;
}, "de_DeleteWorkspaceApiKeyCommand");
var de_DescribeWorkspaceCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode !== 200 && output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const contents = (0, import_smithy_client.map)({
    $metadata: deserializeMetadata(output)
  });
  const data = (0, import_smithy_client.expectNonNull)((0, import_smithy_client.expectObject)(await (0, import_core2.parseJsonBody)(output.body, context)), "body");
  const doc = (0, import_smithy_client.take)(data, {
    workspace: (_) => de_WorkspaceDescription(_, context)
  });
  Object.assign(contents, doc);
  return contents;
}, "de_DescribeWorkspaceCommand");
var de_DescribeWorkspaceAuthenticationCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode !== 200 && output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const contents = (0, import_smithy_client.map)({
    $metadata: deserializeMetadata(output)
  });
  const data = (0, import_smithy_client.expectNonNull)((0, import_smithy_client.expectObject)(await (0, import_core2.parseJsonBody)(output.body, context)), "body");
  const doc = (0, import_smithy_client.take)(data, {
    authentication: import_smithy_client._json
  });
  Object.assign(contents, doc);
  return contents;
}, "de_DescribeWorkspaceAuthenticationCommand");
var de_DescribeWorkspaceConfigurationCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode !== 200 && output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const contents = (0, import_smithy_client.map)({
    $metadata: deserializeMetadata(output)
  });
  const data = (0, import_smithy_client.expectNonNull)((0, import_smithy_client.expectObject)(await (0, import_core2.parseJsonBody)(output.body, context)), "body");
  const doc = (0, import_smithy_client.take)(data, {
    configuration: (_) => new import_smithy_client.LazyJsonString(_),
    grafanaVersion: import_smithy_client.expectString
  });
  Object.assign(contents, doc);
  return contents;
}, "de_DescribeWorkspaceConfigurationCommand");
var de_DisassociateLicenseCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode !== 202 && output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const contents = (0, import_smithy_client.map)({
    $metadata: deserializeMetadata(output)
  });
  const data = (0, import_smithy_client.expectNonNull)((0, import_smithy_client.expectObject)(await (0, import_core2.parseJsonBody)(output.body, context)), "body");
  const doc = (0, import_smithy_client.take)(data, {
    workspace: (_) => de_WorkspaceDescription(_, context)
  });
  Object.assign(contents, doc);
  return contents;
}, "de_DisassociateLicenseCommand");
var de_ListPermissionsCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode !== 200 && output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const contents = (0, import_smithy_client.map)({
    $metadata: deserializeMetadata(output)
  });
  const data = (0, import_smithy_client.expectNonNull)((0, import_smithy_client.expectObject)(await (0, import_core2.parseJsonBody)(output.body, context)), "body");
  const doc = (0, import_smithy_client.take)(data, {
    nextToken: import_smithy_client.expectString,
    permissions: import_smithy_client._json
  });
  Object.assign(contents, doc);
  return contents;
}, "de_ListPermissionsCommand");
var de_ListTagsForResourceCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode !== 200 && output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const contents = (0, import_smithy_client.map)({
    $metadata: deserializeMetadata(output)
  });
  const data = (0, import_smithy_client.expectNonNull)((0, import_smithy_client.expectObject)(await (0, import_core2.parseJsonBody)(output.body, context)), "body");
  const doc = (0, import_smithy_client.take)(data, {
    tags: import_smithy_client._json
  });
  Object.assign(contents, doc);
  return contents;
}, "de_ListTagsForResourceCommand");
var de_ListVersionsCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode !== 200 && output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const contents = (0, import_smithy_client.map)({
    $metadata: deserializeMetadata(output)
  });
  const data = (0, import_smithy_client.expectNonNull)((0, import_smithy_client.expectObject)(await (0, import_core2.parseJsonBody)(output.body, context)), "body");
  const doc = (0, import_smithy_client.take)(data, {
    grafanaVersions: import_smithy_client._json,
    nextToken: import_smithy_client.expectString
  });
  Object.assign(contents, doc);
  return contents;
}, "de_ListVersionsCommand");
var de_ListWorkspacesCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode !== 200 && output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const contents = (0, import_smithy_client.map)({
    $metadata: deserializeMetadata(output)
  });
  const data = (0, import_smithy_client.expectNonNull)((0, import_smithy_client.expectObject)(await (0, import_core2.parseJsonBody)(output.body, context)), "body");
  const doc = (0, import_smithy_client.take)(data, {
    nextToken: import_smithy_client.expectString,
    workspaces: (_) => de_WorkspaceList(_, context)
  });
  Object.assign(contents, doc);
  return contents;
}, "de_ListWorkspacesCommand");
var de_TagResourceCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode !== 200 && output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const contents = (0, import_smithy_client.map)({
    $metadata: deserializeMetadata(output)
  });
  await (0, import_smithy_client.collectBody)(output.body, context);
  return contents;
}, "de_TagResourceCommand");
var de_UntagResourceCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode !== 200 && output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const contents = (0, import_smithy_client.map)({
    $metadata: deserializeMetadata(output)
  });
  await (0, import_smithy_client.collectBody)(output.body, context);
  return contents;
}, "de_UntagResourceCommand");
var de_UpdatePermissionsCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode !== 200 && output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const contents = (0, import_smithy_client.map)({
    $metadata: deserializeMetadata(output)
  });
  const data = (0, import_smithy_client.expectNonNull)((0, import_smithy_client.expectObject)(await (0, import_core2.parseJsonBody)(output.body, context)), "body");
  const doc = (0, import_smithy_client.take)(data, {
    errors: import_smithy_client._json
  });
  Object.assign(contents, doc);
  return contents;
}, "de_UpdatePermissionsCommand");
var de_UpdateWorkspaceCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode !== 202 && output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const contents = (0, import_smithy_client.map)({
    $metadata: deserializeMetadata(output)
  });
  const data = (0, import_smithy_client.expectNonNull)((0, import_smithy_client.expectObject)(await (0, import_core2.parseJsonBody)(output.body, context)), "body");
  const doc = (0, import_smithy_client.take)(data, {
    workspace: (_) => de_WorkspaceDescription(_, context)
  });
  Object.assign(contents, doc);
  return contents;
}, "de_UpdateWorkspaceCommand");
var de_UpdateWorkspaceAuthenticationCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode !== 200 && output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const contents = (0, import_smithy_client.map)({
    $metadata: deserializeMetadata(output)
  });
  const data = (0, import_smithy_client.expectNonNull)((0, import_smithy_client.expectObject)(await (0, import_core2.parseJsonBody)(output.body, context)), "body");
  const doc = (0, import_smithy_client.take)(data, {
    authentication: import_smithy_client._json
  });
  Object.assign(contents, doc);
  return contents;
}, "de_UpdateWorkspaceAuthenticationCommand");
var de_UpdateWorkspaceConfigurationCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode !== 202 && output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const contents = (0, import_smithy_client.map)({
    $metadata: deserializeMetadata(output)
  });
  await (0, import_smithy_client.collectBody)(output.body, context);
  return contents;
}, "de_UpdateWorkspaceConfigurationCommand");
var de_CommandError = /* @__PURE__ */ __name(async (output, context) => {
  const parsedOutput = {
    ...output,
    body: await (0, import_core2.parseJsonErrorBody)(output.body, context)
  };
  const errorCode = (0, import_core2.loadRestJsonErrorCode)(output, parsedOutput.body);
  switch (errorCode) {
    case "AccessDeniedException":
    case "com.amazonaws.grafana#AccessDeniedException":
      throw await de_AccessDeniedExceptionRes(parsedOutput, context);
    case "InternalServerException":
    case "com.amazonaws.grafana#InternalServerException":
      throw await de_InternalServerExceptionRes(parsedOutput, context);
    case "ResourceNotFoundException":
    case "com.amazonaws.grafana#ResourceNotFoundException":
      throw await de_ResourceNotFoundExceptionRes(parsedOutput, context);
    case "ThrottlingException":
    case "com.amazonaws.grafana#ThrottlingException":
      throw await de_ThrottlingExceptionRes(parsedOutput, context);
    case "ValidationException":
    case "com.amazonaws.grafana#ValidationException":
      throw await de_ValidationExceptionRes(parsedOutput, context);
    case "ConflictException":
    case "com.amazonaws.grafana#ConflictException":
      throw await de_ConflictExceptionRes(parsedOutput, context);
    case "ServiceQuotaExceededException":
    case "com.amazonaws.grafana#ServiceQuotaExceededException":
      throw await de_ServiceQuotaExceededExceptionRes(parsedOutput, context);
    default:
      const parsedBody = parsedOutput.body;
      return throwDefaultError({
        output,
        parsedBody,
        errorCode
      });
  }
}, "de_CommandError");
var throwDefaultError = (0, import_smithy_client.withBaseException)(GrafanaServiceException);
var de_AccessDeniedExceptionRes = /* @__PURE__ */ __name(async (parsedOutput, context) => {
  const contents = (0, import_smithy_client.map)({});
  const data = parsedOutput.body;
  const doc = (0, import_smithy_client.take)(data, {
    message: import_smithy_client.expectString
  });
  Object.assign(contents, doc);
  const exception = new AccessDeniedException({
    $metadata: deserializeMetadata(parsedOutput),
    ...contents
  });
  return (0, import_smithy_client.decorateServiceException)(exception, parsedOutput.body);
}, "de_AccessDeniedExceptionRes");
var de_ConflictExceptionRes = /* @__PURE__ */ __name(async (parsedOutput, context) => {
  const contents = (0, import_smithy_client.map)({});
  const data = parsedOutput.body;
  const doc = (0, import_smithy_client.take)(data, {
    message: import_smithy_client.expectString,
    resourceId: import_smithy_client.expectString,
    resourceType: import_smithy_client.expectString
  });
  Object.assign(contents, doc);
  const exception = new ConflictException({
    $metadata: deserializeMetadata(parsedOutput),
    ...contents
  });
  return (0, import_smithy_client.decorateServiceException)(exception, parsedOutput.body);
}, "de_ConflictExceptionRes");
var de_InternalServerExceptionRes = /* @__PURE__ */ __name(async (parsedOutput, context) => {
  const contents = (0, import_smithy_client.map)({
    [_rAS]: [() => void 0 !== parsedOutput.headers[_ra], () => (0, import_smithy_client.strictParseInt32)(parsedOutput.headers[_ra])]
  });
  const data = parsedOutput.body;
  const doc = (0, import_smithy_client.take)(data, {
    message: import_smithy_client.expectString
  });
  Object.assign(contents, doc);
  const exception = new InternalServerException({
    $metadata: deserializeMetadata(parsedOutput),
    ...contents
  });
  return (0, import_smithy_client.decorateServiceException)(exception, parsedOutput.body);
}, "de_InternalServerExceptionRes");
var de_ResourceNotFoundExceptionRes = /* @__PURE__ */ __name(async (parsedOutput, context) => {
  const contents = (0, import_smithy_client.map)({});
  const data = parsedOutput.body;
  const doc = (0, import_smithy_client.take)(data, {
    message: import_smithy_client.expectString,
    resourceId: import_smithy_client.expectString,
    resourceType: import_smithy_client.expectString
  });
  Object.assign(contents, doc);
  const exception = new ResourceNotFoundException({
    $metadata: deserializeMetadata(parsedOutput),
    ...contents
  });
  return (0, import_smithy_client.decorateServiceException)(exception, parsedOutput.body);
}, "de_ResourceNotFoundExceptionRes");
var de_ServiceQuotaExceededExceptionRes = /* @__PURE__ */ __name(async (parsedOutput, context) => {
  const contents = (0, import_smithy_client.map)({});
  const data = parsedOutput.body;
  const doc = (0, import_smithy_client.take)(data, {
    message: import_smithy_client.expectString,
    quotaCode: import_smithy_client.expectString,
    resourceId: import_smithy_client.expectString,
    resourceType: import_smithy_client.expectString,
    serviceCode: import_smithy_client.expectString
  });
  Object.assign(contents, doc);
  const exception = new ServiceQuotaExceededException({
    $metadata: deserializeMetadata(parsedOutput),
    ...contents
  });
  return (0, import_smithy_client.decorateServiceException)(exception, parsedOutput.body);
}, "de_ServiceQuotaExceededExceptionRes");
var de_ThrottlingExceptionRes = /* @__PURE__ */ __name(async (parsedOutput, context) => {
  const contents = (0, import_smithy_client.map)({
    [_rAS]: [() => void 0 !== parsedOutput.headers[_ra], () => (0, import_smithy_client.strictParseInt32)(parsedOutput.headers[_ra])]
  });
  const data = parsedOutput.body;
  const doc = (0, import_smithy_client.take)(data, {
    message: import_smithy_client.expectString,
    quotaCode: import_smithy_client.expectString,
    serviceCode: import_smithy_client.expectString
  });
  Object.assign(contents, doc);
  const exception = new ThrottlingException({
    $metadata: deserializeMetadata(parsedOutput),
    ...contents
  });
  return (0, import_smithy_client.decorateServiceException)(exception, parsedOutput.body);
}, "de_ThrottlingExceptionRes");
var de_ValidationExceptionRes = /* @__PURE__ */ __name(async (parsedOutput, context) => {
  const contents = (0, import_smithy_client.map)({});
  const data = parsedOutput.body;
  const doc = (0, import_smithy_client.take)(data, {
    fieldList: import_smithy_client._json,
    message: import_smithy_client.expectString,
    reason: import_smithy_client.expectString
  });
  Object.assign(contents, doc);
  const exception = new ValidationException({
    $metadata: deserializeMetadata(parsedOutput),
    ...contents
  });
  return (0, import_smithy_client.decorateServiceException)(exception, parsedOutput.body);
}, "de_ValidationExceptionRes");
var de_WorkspaceDescription = /* @__PURE__ */ __name((output, context) => {
  return (0, import_smithy_client.take)(output, {
    accountAccessType: import_smithy_client.expectString,
    authentication: import_smithy_client._json,
    created: (_) => (0, import_smithy_client.expectNonNull)((0, import_smithy_client.parseEpochTimestamp)((0, import_smithy_client.expectNumber)(_))),
    dataSources: import_smithy_client._json,
    description: import_smithy_client.expectString,
    endpoint: import_smithy_client.expectString,
    freeTrialConsumed: import_smithy_client.expectBoolean,
    freeTrialExpiration: (_) => (0, import_smithy_client.expectNonNull)((0, import_smithy_client.parseEpochTimestamp)((0, import_smithy_client.expectNumber)(_))),
    grafanaToken: import_smithy_client.expectString,
    grafanaVersion: import_smithy_client.expectString,
    id: import_smithy_client.expectString,
    licenseExpiration: (_) => (0, import_smithy_client.expectNonNull)((0, import_smithy_client.parseEpochTimestamp)((0, import_smithy_client.expectNumber)(_))),
    licenseType: import_smithy_client.expectString,
    modified: (_) => (0, import_smithy_client.expectNonNull)((0, import_smithy_client.parseEpochTimestamp)((0, import_smithy_client.expectNumber)(_))),
    name: import_smithy_client.expectString,
    networkAccessControl: import_smithy_client._json,
    notificationDestinations: import_smithy_client._json,
    organizationRoleName: import_smithy_client.expectString,
    organizationalUnits: import_smithy_client._json,
    permissionType: import_smithy_client.expectString,
    stackSetName: import_smithy_client.expectString,
    status: import_smithy_client.expectString,
    tags: import_smithy_client._json,
    vpcConfiguration: import_smithy_client._json,
    workspaceRoleArn: import_smithy_client.expectString
  });
}, "de_WorkspaceDescription");
var de_WorkspaceList = /* @__PURE__ */ __name((output, context) => {
  const retVal = (output || []).filter((e) => e != null).map((entry) => {
    return de_WorkspaceSummary(entry, context);
  });
  return retVal;
}, "de_WorkspaceList");
var de_WorkspaceSummary = /* @__PURE__ */ __name((output, context) => {
  return (0, import_smithy_client.take)(output, {
    authentication: import_smithy_client._json,
    created: (_) => (0, import_smithy_client.expectNonNull)((0, import_smithy_client.parseEpochTimestamp)((0, import_smithy_client.expectNumber)(_))),
    description: import_smithy_client.expectString,
    endpoint: import_smithy_client.expectString,
    grafanaToken: import_smithy_client.expectString,
    grafanaVersion: import_smithy_client.expectString,
    id: import_smithy_client.expectString,
    licenseType: import_smithy_client.expectString,
    modified: (_) => (0, import_smithy_client.expectNonNull)((0, import_smithy_client.parseEpochTimestamp)((0, import_smithy_client.expectNumber)(_))),
    name: import_smithy_client.expectString,
    notificationDestinations: import_smithy_client._json,
    status: import_smithy_client.expectString,
    tags: import_smithy_client._json
  });
}, "de_WorkspaceSummary");
var deserializeMetadata = /* @__PURE__ */ __name((output) => ({
  httpStatusCode: output.statusCode,
  requestId: output.headers["x-amzn-requestid"] ?? output.headers["x-amzn-request-id"] ?? output.headers["x-amz-request-id"],
  extendedRequestId: output.headers["x-amz-id-2"],
  cfId: output.headers["x-amz-cf-id"]
}), "deserializeMetadata");
var isSerializableHeaderValue = /* @__PURE__ */ __name((value) => value !== void 0 && value !== null && value !== "" && (!Object.getOwnPropertyNames(value).includes("length") || value.length != 0) && (!Object.getOwnPropertyNames(value).includes("size") || value.size != 0), "isSerializableHeaderValue");
var _gI = "groupId";
var _gT = "grafanaToken";
var _gt = "grafana-token";
var _mR = "maxResults";
var _nT = "nextToken";
var _rAS = "retryAfterSeconds";
var _ra = "retry-after";
var _tK = "tagKeys";
var _uI = "userId";
var _uT = "userType";
var _wI = "workspaceId";
var _wi = "workspace-id";

// src/commands/AssociateLicenseCommand.ts
var _AssociateLicenseCommand = class _AssociateLicenseCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AWSGrafanaControlPlane", "AssociateLicense", {}).n("GrafanaClient", "AssociateLicenseCommand").f(void 0, AssociateLicenseResponseFilterSensitiveLog).ser(se_AssociateLicenseCommand).de(de_AssociateLicenseCommand).build() {
};
__name(_AssociateLicenseCommand, "AssociateLicenseCommand");
var AssociateLicenseCommand = _AssociateLicenseCommand;

// src/commands/CreateWorkspaceApiKeyCommand.ts




var _CreateWorkspaceApiKeyCommand = class _CreateWorkspaceApiKeyCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AWSGrafanaControlPlane", "CreateWorkspaceApiKey", {}).n("GrafanaClient", "CreateWorkspaceApiKeyCommand").f(void 0, CreateWorkspaceApiKeyResponseFilterSensitiveLog).ser(se_CreateWorkspaceApiKeyCommand).de(de_CreateWorkspaceApiKeyCommand).build() {
};
__name(_CreateWorkspaceApiKeyCommand, "CreateWorkspaceApiKeyCommand");
var CreateWorkspaceApiKeyCommand = _CreateWorkspaceApiKeyCommand;

// src/commands/CreateWorkspaceCommand.ts




var _CreateWorkspaceCommand = class _CreateWorkspaceCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AWSGrafanaControlPlane", "CreateWorkspace", {}).n("GrafanaClient", "CreateWorkspaceCommand").f(CreateWorkspaceRequestFilterSensitiveLog, CreateWorkspaceResponseFilterSensitiveLog).ser(se_CreateWorkspaceCommand).de(de_CreateWorkspaceCommand).build() {
};
__name(_CreateWorkspaceCommand, "CreateWorkspaceCommand");
var CreateWorkspaceCommand = _CreateWorkspaceCommand;

// src/commands/DeleteWorkspaceApiKeyCommand.ts




var _DeleteWorkspaceApiKeyCommand = class _DeleteWorkspaceApiKeyCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AWSGrafanaControlPlane", "DeleteWorkspaceApiKey", {}).n("GrafanaClient", "DeleteWorkspaceApiKeyCommand").f(void 0, void 0).ser(se_DeleteWorkspaceApiKeyCommand).de(de_DeleteWorkspaceApiKeyCommand).build() {
};
__name(_DeleteWorkspaceApiKeyCommand, "DeleteWorkspaceApiKeyCommand");
var DeleteWorkspaceApiKeyCommand = _DeleteWorkspaceApiKeyCommand;

// src/commands/DeleteWorkspaceCommand.ts




var _DeleteWorkspaceCommand = class _DeleteWorkspaceCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AWSGrafanaControlPlane", "DeleteWorkspace", {}).n("GrafanaClient", "DeleteWorkspaceCommand").f(void 0, DeleteWorkspaceResponseFilterSensitiveLog).ser(se_DeleteWorkspaceCommand).de(de_DeleteWorkspaceCommand).build() {
};
__name(_DeleteWorkspaceCommand, "DeleteWorkspaceCommand");
var DeleteWorkspaceCommand = _DeleteWorkspaceCommand;

// src/commands/DescribeWorkspaceAuthenticationCommand.ts




var _DescribeWorkspaceAuthenticationCommand = class _DescribeWorkspaceAuthenticationCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AWSGrafanaControlPlane", "DescribeWorkspaceAuthentication", {}).n("GrafanaClient", "DescribeWorkspaceAuthenticationCommand").f(void 0, DescribeWorkspaceAuthenticationResponseFilterSensitiveLog).ser(se_DescribeWorkspaceAuthenticationCommand).de(de_DescribeWorkspaceAuthenticationCommand).build() {
};
__name(_DescribeWorkspaceAuthenticationCommand, "DescribeWorkspaceAuthenticationCommand");
var DescribeWorkspaceAuthenticationCommand = _DescribeWorkspaceAuthenticationCommand;

// src/commands/DescribeWorkspaceCommand.ts




var _DescribeWorkspaceCommand = class _DescribeWorkspaceCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AWSGrafanaControlPlane", "DescribeWorkspace", {}).n("GrafanaClient", "DescribeWorkspaceCommand").f(void 0, DescribeWorkspaceResponseFilterSensitiveLog).ser(se_DescribeWorkspaceCommand).de(de_DescribeWorkspaceCommand).build() {
};
__name(_DescribeWorkspaceCommand, "DescribeWorkspaceCommand");
var DescribeWorkspaceCommand = _DescribeWorkspaceCommand;

// src/commands/DescribeWorkspaceConfigurationCommand.ts




var _DescribeWorkspaceConfigurationCommand = class _DescribeWorkspaceConfigurationCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AWSGrafanaControlPlane", "DescribeWorkspaceConfiguration", {}).n("GrafanaClient", "DescribeWorkspaceConfigurationCommand").f(void 0, void 0).ser(se_DescribeWorkspaceConfigurationCommand).de(de_DescribeWorkspaceConfigurationCommand).build() {
};
__name(_DescribeWorkspaceConfigurationCommand, "DescribeWorkspaceConfigurationCommand");
var DescribeWorkspaceConfigurationCommand = _DescribeWorkspaceConfigurationCommand;

// src/commands/DisassociateLicenseCommand.ts




var _DisassociateLicenseCommand = class _DisassociateLicenseCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AWSGrafanaControlPlane", "DisassociateLicense", {}).n("GrafanaClient", "DisassociateLicenseCommand").f(void 0, DisassociateLicenseResponseFilterSensitiveLog).ser(se_DisassociateLicenseCommand).de(de_DisassociateLicenseCommand).build() {
};
__name(_DisassociateLicenseCommand, "DisassociateLicenseCommand");
var DisassociateLicenseCommand = _DisassociateLicenseCommand;

// src/commands/ListPermissionsCommand.ts




var _ListPermissionsCommand = class _ListPermissionsCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AWSGrafanaControlPlane", "ListPermissions", {}).n("GrafanaClient", "ListPermissionsCommand").f(void 0, void 0).ser(se_ListPermissionsCommand).de(de_ListPermissionsCommand).build() {
};
__name(_ListPermissionsCommand, "ListPermissionsCommand");
var ListPermissionsCommand = _ListPermissionsCommand;

// src/commands/ListTagsForResourceCommand.ts




var _ListTagsForResourceCommand = class _ListTagsForResourceCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AWSGrafanaControlPlane", "ListTagsForResource", {}).n("GrafanaClient", "ListTagsForResourceCommand").f(void 0, void 0).ser(se_ListTagsForResourceCommand).de(de_ListTagsForResourceCommand).build() {
};
__name(_ListTagsForResourceCommand, "ListTagsForResourceCommand");
var ListTagsForResourceCommand = _ListTagsForResourceCommand;

// src/commands/ListVersionsCommand.ts




var _ListVersionsCommand = class _ListVersionsCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AWSGrafanaControlPlane", "ListVersions", {}).n("GrafanaClient", "ListVersionsCommand").f(void 0, void 0).ser(se_ListVersionsCommand).de(de_ListVersionsCommand).build() {
};
__name(_ListVersionsCommand, "ListVersionsCommand");
var ListVersionsCommand = _ListVersionsCommand;

// src/commands/ListWorkspacesCommand.ts




var _ListWorkspacesCommand = class _ListWorkspacesCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AWSGrafanaControlPlane", "ListWorkspaces", {}).n("GrafanaClient", "ListWorkspacesCommand").f(void 0, ListWorkspacesResponseFilterSensitiveLog).ser(se_ListWorkspacesCommand).de(de_ListWorkspacesCommand).build() {
};
__name(_ListWorkspacesCommand, "ListWorkspacesCommand");
var ListWorkspacesCommand = _ListWorkspacesCommand;

// src/commands/TagResourceCommand.ts




var _TagResourceCommand = class _TagResourceCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AWSGrafanaControlPlane", "TagResource", {}).n("GrafanaClient", "TagResourceCommand").f(void 0, void 0).ser(se_TagResourceCommand).de(de_TagResourceCommand).build() {
};
__name(_TagResourceCommand, "TagResourceCommand");
var TagResourceCommand = _TagResourceCommand;

// src/commands/UntagResourceCommand.ts




var _UntagResourceCommand = class _UntagResourceCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AWSGrafanaControlPlane", "UntagResource", {}).n("GrafanaClient", "UntagResourceCommand").f(void 0, void 0).ser(se_UntagResourceCommand).de(de_UntagResourceCommand).build() {
};
__name(_UntagResourceCommand, "UntagResourceCommand");
var UntagResourceCommand = _UntagResourceCommand;

// src/commands/UpdatePermissionsCommand.ts




var _UpdatePermissionsCommand = class _UpdatePermissionsCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AWSGrafanaControlPlane", "UpdatePermissions", {}).n("GrafanaClient", "UpdatePermissionsCommand").f(void 0, void 0).ser(se_UpdatePermissionsCommand).de(de_UpdatePermissionsCommand).build() {
};
__name(_UpdatePermissionsCommand, "UpdatePermissionsCommand");
var UpdatePermissionsCommand = _UpdatePermissionsCommand;

// src/commands/UpdateWorkspaceAuthenticationCommand.ts




var _UpdateWorkspaceAuthenticationCommand = class _UpdateWorkspaceAuthenticationCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AWSGrafanaControlPlane", "UpdateWorkspaceAuthentication", {}).n("GrafanaClient", "UpdateWorkspaceAuthenticationCommand").f(UpdateWorkspaceAuthenticationRequestFilterSensitiveLog, UpdateWorkspaceAuthenticationResponseFilterSensitiveLog).ser(se_UpdateWorkspaceAuthenticationCommand).de(de_UpdateWorkspaceAuthenticationCommand).build() {
};
__name(_UpdateWorkspaceAuthenticationCommand, "UpdateWorkspaceAuthenticationCommand");
var UpdateWorkspaceAuthenticationCommand = _UpdateWorkspaceAuthenticationCommand;

// src/commands/UpdateWorkspaceCommand.ts




var _UpdateWorkspaceCommand = class _UpdateWorkspaceCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AWSGrafanaControlPlane", "UpdateWorkspace", {}).n("GrafanaClient", "UpdateWorkspaceCommand").f(UpdateWorkspaceRequestFilterSensitiveLog, UpdateWorkspaceResponseFilterSensitiveLog).ser(se_UpdateWorkspaceCommand).de(de_UpdateWorkspaceCommand).build() {
};
__name(_UpdateWorkspaceCommand, "UpdateWorkspaceCommand");
var UpdateWorkspaceCommand = _UpdateWorkspaceCommand;

// src/commands/UpdateWorkspaceConfigurationCommand.ts




var _UpdateWorkspaceConfigurationCommand = class _UpdateWorkspaceConfigurationCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AWSGrafanaControlPlane", "UpdateWorkspaceConfiguration", {}).n("GrafanaClient", "UpdateWorkspaceConfigurationCommand").f(void 0, void 0).ser(se_UpdateWorkspaceConfigurationCommand).de(de_UpdateWorkspaceConfigurationCommand).build() {
};
__name(_UpdateWorkspaceConfigurationCommand, "UpdateWorkspaceConfigurationCommand");
var UpdateWorkspaceConfigurationCommand = _UpdateWorkspaceConfigurationCommand;

// src/Grafana.ts
var commands = {
  AssociateLicenseCommand,
  CreateWorkspaceCommand,
  CreateWorkspaceApiKeyCommand,
  DeleteWorkspaceCommand,
  DeleteWorkspaceApiKeyCommand,
  DescribeWorkspaceCommand,
  DescribeWorkspaceAuthenticationCommand,
  DescribeWorkspaceConfigurationCommand,
  DisassociateLicenseCommand,
  ListPermissionsCommand,
  ListTagsForResourceCommand,
  ListVersionsCommand,
  ListWorkspacesCommand,
  TagResourceCommand,
  UntagResourceCommand,
  UpdatePermissionsCommand,
  UpdateWorkspaceCommand,
  UpdateWorkspaceAuthenticationCommand,
  UpdateWorkspaceConfigurationCommand
};
var _Grafana = class _Grafana extends GrafanaClient {
};
__name(_Grafana, "Grafana");
var Grafana = _Grafana;
(0, import_smithy_client.createAggregatedClient)(commands, Grafana);

// src/pagination/ListPermissionsPaginator.ts

var paginateListPermissions = (0, import_core.createPaginator)(GrafanaClient, ListPermissionsCommand, "nextToken", "nextToken", "maxResults");

// src/pagination/ListVersionsPaginator.ts

var paginateListVersions = (0, import_core.createPaginator)(GrafanaClient, ListVersionsCommand, "nextToken", "nextToken", "maxResults");

// src/pagination/ListWorkspacesPaginator.ts

var paginateListWorkspaces = (0, import_core.createPaginator)(GrafanaClient, ListWorkspacesCommand, "nextToken", "nextToken", "maxResults");
// Annotate the CommonJS export names for ESM import in node:

0 && (module.exports = {
  GrafanaServiceException,
  __Client,
  GrafanaClient,
  Grafana,
  $Command,
  AssociateLicenseCommand,
  CreateWorkspaceApiKeyCommand,
  CreateWorkspaceCommand,
  DeleteWorkspaceApiKeyCommand,
  DeleteWorkspaceCommand,
  DescribeWorkspaceAuthenticationCommand,
  DescribeWorkspaceCommand,
  DescribeWorkspaceConfigurationCommand,
  DisassociateLicenseCommand,
  ListPermissionsCommand,
  ListTagsForResourceCommand,
  ListVersionsCommand,
  ListWorkspacesCommand,
  TagResourceCommand,
  UntagResourceCommand,
  UpdatePermissionsCommand,
  UpdateWorkspaceAuthenticationCommand,
  UpdateWorkspaceCommand,
  UpdateWorkspaceConfigurationCommand,
  paginateListPermissions,
  paginateListVersions,
  paginateListWorkspaces,
  AccessDeniedException,
  AccountAccessType,
  ConflictException,
  InternalServerException,
  ResourceNotFoundException,
  ServiceQuotaExceededException,
  ThrottlingException,
  ValidationExceptionReason,
  ValidationException,
  LicenseType,
  AuthenticationProviderTypes,
  SamlConfigurationStatus,
  DataSourceType,
  NotificationDestinationType,
  PermissionType,
  WorkspaceStatus,
  IdpMetadata,
  UserType,
  Role,
  UpdateAction,
  CreateWorkspaceApiKeyResponseFilterSensitiveLog,
  WorkspaceDescriptionFilterSensitiveLog,
  AssociateLicenseResponseFilterSensitiveLog,
  RoleValuesFilterSensitiveLog,
  SamlConfigurationFilterSensitiveLog,
  SamlAuthenticationFilterSensitiveLog,
  AuthenticationDescriptionFilterSensitiveLog,
  DescribeWorkspaceAuthenticationResponseFilterSensitiveLog,
  UpdateWorkspaceAuthenticationRequestFilterSensitiveLog,
  UpdateWorkspaceAuthenticationResponseFilterSensitiveLog,
  DisassociateLicenseResponseFilterSensitiveLog,
  CreateWorkspaceRequestFilterSensitiveLog,
  CreateWorkspaceResponseFilterSensitiveLog,
  DeleteWorkspaceResponseFilterSensitiveLog,
  DescribeWorkspaceResponseFilterSensitiveLog,
  WorkspaceSummaryFilterSensitiveLog,
  ListWorkspacesResponseFilterSensitiveLog,
  UpdateWorkspaceRequestFilterSensitiveLog,
  UpdateWorkspaceResponseFilterSensitiveLog
});
