// @aws-sdk/client-appconfig/dist-cjs/index.js
"use strict";
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __name = (target, value) => __defProp(target, "name", { value, configurable: true });
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// src/index.ts
var src_exports = {};
__export(src_exports, {
  ActionPoint: () => ActionPoint,
  AppConfig: () => AppConfig,
  AppConfigClient: () => AppConfigClient,
  AppConfigServiceException: () => AppConfigServiceException,
  BadRequestDetails: () => BadRequestDetails,
  BadRequestException: () => BadRequestException,
  BadRequestReason: () => BadRequestReason,
  BytesMeasure: () => BytesMeasure,
  ConfigurationFilterSensitiveLog: () => ConfigurationFilterSensitiveLog,
  ConfigurationProfileFilterSensitiveLog: () => ConfigurationProfileFilterSensitiveLog,
  ConflictException: () => ConflictException,
  CreateApplicationCommand: () => CreateApplicationCommand,
  CreateConfigurationProfileCommand: () => CreateConfigurationProfileCommand,
  CreateConfigurationProfileRequestFilterSensitiveLog: () => CreateConfigurationProfileRequestFilterSensitiveLog,
  CreateDeploymentStrategyCommand: () => CreateDeploymentStrategyCommand,
  CreateEnvironmentCommand: () => CreateEnvironmentCommand,
  CreateExtensionAssociationCommand: () => CreateExtensionAssociationCommand,
  CreateExtensionCommand: () => CreateExtensionCommand,
  CreateHostedConfigurationVersionCommand: () => CreateHostedConfigurationVersionCommand,
  CreateHostedConfigurationVersionRequestFilterSensitiveLog: () => CreateHostedConfigurationVersionRequestFilterSensitiveLog,
  DeleteApplicationCommand: () => DeleteApplicationCommand,
  DeleteConfigurationProfileCommand: () => DeleteConfigurationProfileCommand,
  DeleteDeploymentStrategyCommand: () => DeleteDeploymentStrategyCommand,
  DeleteEnvironmentCommand: () => DeleteEnvironmentCommand,
  DeleteExtensionAssociationCommand: () => DeleteExtensionAssociationCommand,
  DeleteExtensionCommand: () => DeleteExtensionCommand,
  DeleteHostedConfigurationVersionCommand: () => DeleteHostedConfigurationVersionCommand,
  DeploymentEventType: () => DeploymentEventType,
  DeploymentState: () => DeploymentState,
  EnvironmentState: () => EnvironmentState,
  GetApplicationCommand: () => GetApplicationCommand,
  GetConfigurationCommand: () => GetConfigurationCommand,
  GetConfigurationProfileCommand: () => GetConfigurationProfileCommand,
  GetDeploymentCommand: () => GetDeploymentCommand,
  GetDeploymentStrategyCommand: () => GetDeploymentStrategyCommand,
  GetEnvironmentCommand: () => GetEnvironmentCommand,
  GetExtensionAssociationCommand: () => GetExtensionAssociationCommand,
  GetExtensionCommand: () => GetExtensionCommand,
  GetHostedConfigurationVersionCommand: () => GetHostedConfigurationVersionCommand,
  GrowthType: () => GrowthType,
  HostedConfigurationVersionFilterSensitiveLog: () => HostedConfigurationVersionFilterSensitiveLog,
  InternalServerException: () => InternalServerException,
  ListApplicationsCommand: () => ListApplicationsCommand,
  ListConfigurationProfilesCommand: () => ListConfigurationProfilesCommand,
  ListDeploymentStrategiesCommand: () => ListDeploymentStrategiesCommand,
  ListDeploymentsCommand: () => ListDeploymentsCommand,
  ListEnvironmentsCommand: () => ListEnvironmentsCommand,
  ListExtensionAssociationsCommand: () => ListExtensionAssociationsCommand,
  ListExtensionsCommand: () => ListExtensionsCommand,
  ListHostedConfigurationVersionsCommand: () => ListHostedConfigurationVersionsCommand,
  ListTagsForResourceCommand: () => ListTagsForResourceCommand,
  PayloadTooLargeException: () => PayloadTooLargeException,
  ReplicateTo: () => ReplicateTo,
  ResourceNotFoundException: () => ResourceNotFoundException,
  ServiceQuotaExceededException: () => ServiceQuotaExceededException,
  StartDeploymentCommand: () => StartDeploymentCommand,
  StopDeploymentCommand: () => StopDeploymentCommand,
  TagResourceCommand: () => TagResourceCommand,
  TriggeredBy: () => TriggeredBy,
  UntagResourceCommand: () => UntagResourceCommand,
  UpdateApplicationCommand: () => UpdateApplicationCommand,
  UpdateConfigurationProfileCommand: () => UpdateConfigurationProfileCommand,
  UpdateConfigurationProfileRequestFilterSensitiveLog: () => UpdateConfigurationProfileRequestFilterSensitiveLog,
  UpdateDeploymentStrategyCommand: () => UpdateDeploymentStrategyCommand,
  UpdateEnvironmentCommand: () => UpdateEnvironmentCommand,
  UpdateExtensionAssociationCommand: () => UpdateExtensionAssociationCommand,
  UpdateExtensionCommand: () => UpdateExtensionCommand,
  ValidateConfigurationCommand: () => ValidateConfigurationCommand,
  ValidatorFilterSensitiveLog: () => ValidatorFilterSensitiveLog,
  ValidatorType: () => ValidatorType,
  __Client: () => import_smithy_client.Client,
  paginateListApplications: () => paginateListApplications,
  paginateListConfigurationProfiles: () => paginateListConfigurationProfiles,
  paginateListDeploymentStrategies: () => paginateListDeploymentStrategies,
  paginateListDeployments: () => paginateListDeployments,
  paginateListEnvironments: () => paginateListEnvironments,
  paginateListExtensionAssociations: () => paginateListExtensionAssociations,
  paginateListExtensions: () => paginateListExtensions,
  paginateListHostedConfigurationVersions: () => paginateListHostedConfigurationVersions
});
module.exports = __toCommonJS(src_exports);

// src/AppConfigClient.ts
var import_middleware_host_header = require("@aws-sdk/middleware-host-header");
var import_middleware_logger = require("@aws-sdk/middleware-logger");
var import_middleware_recursion_detection = require("@aws-sdk/middleware-recursion-detection");
var import_middleware_user_agent = require("@aws-sdk/middleware-user-agent");
var import_config_resolver = require("@smithy/config-resolver");
var import_core = require("@smithy/core");
var import_middleware_content_length = require("@smithy/middleware-content-length");
var import_middleware_endpoint = require("@smithy/middleware-endpoint");
var import_middleware_retry = require("@smithy/middleware-retry");

var import_httpAuthSchemeProvider = require("./auth/httpAuthSchemeProvider");

// src/endpoint/EndpointParameters.ts
var resolveClientEndpointParameters = /* @__PURE__ */ __name((options) => {
  return {
    ...options,
    useDualstackEndpoint: options.useDualstackEndpoint ?? false,
    useFipsEndpoint: options.useFipsEndpoint ?? false,
    defaultSigningName: "appconfig"
  };
}, "resolveClientEndpointParameters");
var commonParams = {
  UseFIPS: { type: "builtInParams", name: "useFipsEndpoint" },
  Endpoint: { type: "builtInParams", name: "endpoint" },
  Region: { type: "builtInParams", name: "region" },
  UseDualStack: { type: "builtInParams", name: "useDualstackEndpoint" }
};

// src/AppConfigClient.ts
var import_runtimeConfig = require("././runtimeConfig");

// src/runtimeExtensions.ts
var import_region_config_resolver = require("@aws-sdk/region-config-resolver");
var import_protocol_http = require("@smithy/protocol-http");
var import_smithy_client = require("@smithy/smithy-client");

// src/auth/httpAuthExtensionConfiguration.ts
var getHttpAuthExtensionConfiguration = /* @__PURE__ */ __name((runtimeConfig) => {
  const _httpAuthSchemes = runtimeConfig.httpAuthSchemes;
  let _httpAuthSchemeProvider = runtimeConfig.httpAuthSchemeProvider;
  let _credentials = runtimeConfig.credentials;
  return {
    setHttpAuthScheme(httpAuthScheme) {
      const index = _httpAuthSchemes.findIndex((scheme) => scheme.schemeId === httpAuthScheme.schemeId);
      if (index === -1) {
        _httpAuthSchemes.push(httpAuthScheme);
      } else {
        _httpAuthSchemes.splice(index, 1, httpAuthScheme);
      }
    },
    httpAuthSchemes() {
      return _httpAuthSchemes;
    },
    setHttpAuthSchemeProvider(httpAuthSchemeProvider) {
      _httpAuthSchemeProvider = httpAuthSchemeProvider;
    },
    httpAuthSchemeProvider() {
      return _httpAuthSchemeProvider;
    },
    setCredentials(credentials) {
      _credentials = credentials;
    },
    credentials() {
      return _credentials;
    }
  };
}, "getHttpAuthExtensionConfiguration");
var resolveHttpAuthRuntimeConfig = /* @__PURE__ */ __name((config) => {
  return {
    httpAuthSchemes: config.httpAuthSchemes(),
    httpAuthSchemeProvider: config.httpAuthSchemeProvider(),
    credentials: config.credentials()
  };
}, "resolveHttpAuthRuntimeConfig");

// src/runtimeExtensions.ts
var asPartial = /* @__PURE__ */ __name((t) => t, "asPartial");
var resolveRuntimeExtensions = /* @__PURE__ */ __name((runtimeConfig, extensions) => {
  const extensionConfiguration = {
    ...asPartial((0, import_region_config_resolver.getAwsRegionExtensionConfiguration)(runtimeConfig)),
    ...asPartial((0, import_smithy_client.getDefaultExtensionConfiguration)(runtimeConfig)),
    ...asPartial((0, import_protocol_http.getHttpHandlerExtensionConfiguration)(runtimeConfig)),
    ...asPartial(getHttpAuthExtensionConfiguration(runtimeConfig))
  };
  extensions.forEach((extension) => extension.configure(extensionConfiguration));
  return {
    ...runtimeConfig,
    ...(0, import_region_config_resolver.resolveAwsRegionExtensionConfiguration)(extensionConfiguration),
    ...(0, import_smithy_client.resolveDefaultRuntimeConfig)(extensionConfiguration),
    ...(0, import_protocol_http.resolveHttpHandlerRuntimeConfig)(extensionConfiguration),
    ...resolveHttpAuthRuntimeConfig(extensionConfiguration)
  };
}, "resolveRuntimeExtensions");

// src/AppConfigClient.ts
var _AppConfigClient = class _AppConfigClient extends import_smithy_client.Client {
  constructor(...[configuration]) {
    const _config_0 = (0, import_runtimeConfig.getRuntimeConfig)(configuration || {});
    const _config_1 = resolveClientEndpointParameters(_config_0);
    const _config_2 = (0, import_config_resolver.resolveRegionConfig)(_config_1);
    const _config_3 = (0, import_middleware_endpoint.resolveEndpointConfig)(_config_2);
    const _config_4 = (0, import_middleware_retry.resolveRetryConfig)(_config_3);
    const _config_5 = (0, import_middleware_host_header.resolveHostHeaderConfig)(_config_4);
    const _config_6 = (0, import_middleware_user_agent.resolveUserAgentConfig)(_config_5);
    const _config_7 = (0, import_httpAuthSchemeProvider.resolveHttpAuthSchemeConfig)(_config_6);
    const _config_8 = resolveRuntimeExtensions(_config_7, (configuration == null ? void 0 : configuration.extensions) || []);
    super(_config_8);
    this.config = _config_8;
    this.middlewareStack.use((0, import_middleware_retry.getRetryPlugin)(this.config));
    this.middlewareStack.use((0, import_middleware_content_length.getContentLengthPlugin)(this.config));
    this.middlewareStack.use((0, import_middleware_host_header.getHostHeaderPlugin)(this.config));
    this.middlewareStack.use((0, import_middleware_logger.getLoggerPlugin)(this.config));
    this.middlewareStack.use((0, import_middleware_recursion_detection.getRecursionDetectionPlugin)(this.config));
    this.middlewareStack.use((0, import_middleware_user_agent.getUserAgentPlugin)(this.config));
    this.middlewareStack.use(
      (0, import_core.getHttpAuthSchemeEndpointRuleSetPlugin)(this.config, {
        httpAuthSchemeParametersProvider: this.getDefaultHttpAuthSchemeParametersProvider(),
        identityProviderConfigProvider: this.getIdentityProviderConfigProvider()
      })
    );
    this.middlewareStack.use((0, import_core.getHttpSigningPlugin)(this.config));
  }
  /**
   * Destroy underlying resources, like sockets. It's usually not necessary to do this.
   * However in Node.js, it's best to explicitly shut down the client's agent when it is no longer needed.
   * Otherwise, sockets might stay open for quite a long time before the server terminates them.
   */
  destroy() {
    super.destroy();
  }
  getDefaultHttpAuthSchemeParametersProvider() {
    return import_httpAuthSchemeProvider.defaultAppConfigHttpAuthSchemeParametersProvider;
  }
  getIdentityProviderConfigProvider() {
    return async (config) => new import_core.DefaultIdentityProviderConfig({
      "aws.auth#sigv4": config.credentials
    });
  }
};
__name(_AppConfigClient, "AppConfigClient");
var AppConfigClient = _AppConfigClient;

// src/AppConfig.ts


// src/commands/CreateApplicationCommand.ts

var import_middleware_serde = require("@smithy/middleware-serde");

var import_types = require("@smithy/types");

// src/protocols/Aws_restJson1.ts
var import_core2 = require("@aws-sdk/core");



// src/models/AppConfigServiceException.ts

var _AppConfigServiceException = class _AppConfigServiceException extends import_smithy_client.ServiceException {
  /**
   * @internal
   */
  constructor(options) {
    super(options);
    Object.setPrototypeOf(this, _AppConfigServiceException.prototype);
  }
};
__name(_AppConfigServiceException, "AppConfigServiceException");
var AppConfigServiceException = _AppConfigServiceException;

// src/models/models_0.ts

var ActionPoint = {
  ON_DEPLOYMENT_BAKING: "ON_DEPLOYMENT_BAKING",
  ON_DEPLOYMENT_COMPLETE: "ON_DEPLOYMENT_COMPLETE",
  ON_DEPLOYMENT_ROLLED_BACK: "ON_DEPLOYMENT_ROLLED_BACK",
  ON_DEPLOYMENT_START: "ON_DEPLOYMENT_START",
  ON_DEPLOYMENT_STEP: "ON_DEPLOYMENT_STEP",
  PRE_CREATE_HOSTED_CONFIGURATION_VERSION: "PRE_CREATE_HOSTED_CONFIGURATION_VERSION",
  PRE_START_DEPLOYMENT: "PRE_START_DEPLOYMENT"
};
var BadRequestDetails;
((BadRequestDetails2) => {
  BadRequestDetails2.visit = /* @__PURE__ */ __name((value, visitor) => {
    if (value.InvalidConfiguration !== void 0)
      return visitor.InvalidConfiguration(value.InvalidConfiguration);
    return visitor._(value.$unknown[0], value.$unknown[1]);
  }, "visit");
})(BadRequestDetails || (BadRequestDetails = {}));
var BadRequestReason = {
  INVALID_CONFIGURATION: "InvalidConfiguration"
};
var _BadRequestException = class _BadRequestException extends AppConfigServiceException {
  /**
   * @internal
   */
  constructor(opts) {
    super({
      name: "BadRequestException",
      $fault: "client",
      ...opts
    });
    this.name = "BadRequestException";
    this.$fault = "client";
    Object.setPrototypeOf(this, _BadRequestException.prototype);
    this.Message = opts.Message;
    this.Reason = opts.Reason;
    this.Details = opts.Details;
  }
};
__name(_BadRequestException, "BadRequestException");
var BadRequestException = _BadRequestException;
var _InternalServerException = class _InternalServerException extends AppConfigServiceException {
  /**
   * @internal
   */
  constructor(opts) {
    super({
      name: "InternalServerException",
      $fault: "server",
      ...opts
    });
    this.name = "InternalServerException";
    this.$fault = "server";
    Object.setPrototypeOf(this, _InternalServerException.prototype);
    this.Message = opts.Message;
  }
};
__name(_InternalServerException, "InternalServerException");
var InternalServerException = _InternalServerException;
var _ServiceQuotaExceededException = class _ServiceQuotaExceededException extends AppConfigServiceException {
  /**
   * @internal
   */
  constructor(opts) {
    super({
      name: "ServiceQuotaExceededException",
      $fault: "client",
      ...opts
    });
    this.name = "ServiceQuotaExceededException";
    this.$fault = "client";
    Object.setPrototypeOf(this, _ServiceQuotaExceededException.prototype);
    this.Message = opts.Message;
  }
};
__name(_ServiceQuotaExceededException, "ServiceQuotaExceededException");
var ServiceQuotaExceededException = _ServiceQuotaExceededException;
var ValidatorType = {
  JSON_SCHEMA: "JSON_SCHEMA",
  LAMBDA: "LAMBDA"
};
var _ResourceNotFoundException = class _ResourceNotFoundException extends AppConfigServiceException {
  /**
   * @internal
   */
  constructor(opts) {
    super({
      name: "ResourceNotFoundException",
      $fault: "client",
      ...opts
    });
    this.name = "ResourceNotFoundException";
    this.$fault = "client";
    Object.setPrototypeOf(this, _ResourceNotFoundException.prototype);
    this.Message = opts.Message;
    this.ResourceName = opts.ResourceName;
  }
};
__name(_ResourceNotFoundException, "ResourceNotFoundException");
var ResourceNotFoundException = _ResourceNotFoundException;
var GrowthType = {
  EXPONENTIAL: "EXPONENTIAL",
  LINEAR: "LINEAR"
};
var ReplicateTo = {
  NONE: "NONE",
  SSM_DOCUMENT: "SSM_DOCUMENT"
};
var EnvironmentState = {
  DEPLOYING: "DEPLOYING",
  READY_FOR_DEPLOYMENT: "READY_FOR_DEPLOYMENT",
  ROLLED_BACK: "ROLLED_BACK",
  ROLLING_BACK: "ROLLING_BACK"
};
var _ConflictException = class _ConflictException extends AppConfigServiceException {
  /**
   * @internal
   */
  constructor(opts) {
    super({
      name: "ConflictException",
      $fault: "client",
      ...opts
    });
    this.name = "ConflictException";
    this.$fault = "client";
    Object.setPrototypeOf(this, _ConflictException.prototype);
    this.Message = opts.Message;
  }
};
__name(_ConflictException, "ConflictException");
var ConflictException = _ConflictException;
var BytesMeasure = {
  KILOBYTES: "KILOBYTES"
};
var _PayloadTooLargeException = class _PayloadTooLargeException extends AppConfigServiceException {
  /**
   * @internal
   */
  constructor(opts) {
    super({
      name: "PayloadTooLargeException",
      $fault: "client",
      ...opts
    });
    this.name = "PayloadTooLargeException";
    this.$fault = "client";
    Object.setPrototypeOf(this, _PayloadTooLargeException.prototype);
    this.Message = opts.Message;
    this.Measure = opts.Measure;
    this.Limit = opts.Limit;
    this.Size = opts.Size;
  }
};
__name(_PayloadTooLargeException, "PayloadTooLargeException");
var PayloadTooLargeException = _PayloadTooLargeException;
var DeploymentEventType = {
  BAKE_TIME_STARTED: "BAKE_TIME_STARTED",
  DEPLOYMENT_COMPLETED: "DEPLOYMENT_COMPLETED",
  DEPLOYMENT_STARTED: "DEPLOYMENT_STARTED",
  PERCENTAGE_UPDATED: "PERCENTAGE_UPDATED",
  ROLLBACK_COMPLETED: "ROLLBACK_COMPLETED",
  ROLLBACK_STARTED: "ROLLBACK_STARTED"
};
var TriggeredBy = {
  APPCONFIG: "APPCONFIG",
  CLOUDWATCH_ALARM: "CLOUDWATCH_ALARM",
  INTERNAL_ERROR: "INTERNAL_ERROR",
  USER: "USER"
};
var DeploymentState = {
  BAKING: "BAKING",
  COMPLETE: "COMPLETE",
  DEPLOYING: "DEPLOYING",
  ROLLED_BACK: "ROLLED_BACK",
  ROLLING_BACK: "ROLLING_BACK",
  VALIDATING: "VALIDATING"
};
var ValidatorFilterSensitiveLog = /* @__PURE__ */ __name((obj) => ({
  ...obj,
  ...obj.Content && { Content: import_smithy_client.SENSITIVE_STRING }
}), "ValidatorFilterSensitiveLog");
var ConfigurationProfileFilterSensitiveLog = /* @__PURE__ */ __name((obj) => ({
  ...obj,
  ...obj.Validators && { Validators: obj.Validators.map((item) => ValidatorFilterSensitiveLog(item)) }
}), "ConfigurationProfileFilterSensitiveLog");
var CreateConfigurationProfileRequestFilterSensitiveLog = /* @__PURE__ */ __name((obj) => ({
  ...obj,
  ...obj.Validators && { Validators: obj.Validators.map((item) => ValidatorFilterSensitiveLog(item)) }
}), "CreateConfigurationProfileRequestFilterSensitiveLog");
var CreateHostedConfigurationVersionRequestFilterSensitiveLog = /* @__PURE__ */ __name((obj) => ({
  ...obj,
  ...obj.Content && { Content: import_smithy_client.SENSITIVE_STRING }
}), "CreateHostedConfigurationVersionRequestFilterSensitiveLog");
var HostedConfigurationVersionFilterSensitiveLog = /* @__PURE__ */ __name((obj) => ({
  ...obj,
  ...obj.Content && { Content: import_smithy_client.SENSITIVE_STRING }
}), "HostedConfigurationVersionFilterSensitiveLog");
var ConfigurationFilterSensitiveLog = /* @__PURE__ */ __name((obj) => ({
  ...obj,
  ...obj.Content && { Content: import_smithy_client.SENSITIVE_STRING }
}), "ConfigurationFilterSensitiveLog");
var UpdateConfigurationProfileRequestFilterSensitiveLog = /* @__PURE__ */ __name((obj) => ({
  ...obj,
  ...obj.Validators && { Validators: obj.Validators.map((item) => ValidatorFilterSensitiveLog(item)) }
}), "UpdateConfigurationProfileRequestFilterSensitiveLog");

// src/protocols/Aws_restJson1.ts
var se_CreateApplicationCommand = /* @__PURE__ */ __name(async (input, context) => {
  const b = (0, import_core.requestBuilder)(input, context);
  const headers = {
    "content-type": "application/json"
  };
  b.bp("/applications");
  let body;
  body = JSON.stringify(
    (0, import_smithy_client.take)(input, {
      Description: [],
      Name: [],
      Tags: (_) => (0, import_smithy_client._json)(_)
    })
  );
  b.m("POST").h(headers).b(body);
  return b.build();
}, "se_CreateApplicationCommand");
var se_CreateConfigurationProfileCommand = /* @__PURE__ */ __name(async (input, context) => {
  const b = (0, import_core.requestBuilder)(input, context);
  const headers = {
    "content-type": "application/json"
  };
  b.bp("/applications/{ApplicationId}/configurationprofiles");
  b.p("ApplicationId", () => input.ApplicationId, "{ApplicationId}", false);
  let body;
  body = JSON.stringify(
    (0, import_smithy_client.take)(input, {
      Description: [],
      KmsKeyIdentifier: [],
      LocationUri: [],
      Name: [],
      RetrievalRoleArn: [],
      Tags: (_) => (0, import_smithy_client._json)(_),
      Type: [],
      Validators: (_) => (0, import_smithy_client._json)(_)
    })
  );
  b.m("POST").h(headers).b(body);
  return b.build();
}, "se_CreateConfigurationProfileCommand");
var se_CreateDeploymentStrategyCommand = /* @__PURE__ */ __name(async (input, context) => {
  const b = (0, import_core.requestBuilder)(input, context);
  const headers = {
    "content-type": "application/json"
  };
  b.bp("/deploymentstrategies");
  let body;
  body = JSON.stringify(
    (0, import_smithy_client.take)(input, {
      DeploymentDurationInMinutes: [],
      Description: [],
      FinalBakeTimeInMinutes: [],
      GrowthFactor: (_) => (0, import_smithy_client.serializeFloat)(_),
      GrowthType: [],
      Name: [],
      ReplicateTo: [],
      Tags: (_) => (0, import_smithy_client._json)(_)
    })
  );
  b.m("POST").h(headers).b(body);
  return b.build();
}, "se_CreateDeploymentStrategyCommand");
var se_CreateEnvironmentCommand = /* @__PURE__ */ __name(async (input, context) => {
  const b = (0, import_core.requestBuilder)(input, context);
  const headers = {
    "content-type": "application/json"
  };
  b.bp("/applications/{ApplicationId}/environments");
  b.p("ApplicationId", () => input.ApplicationId, "{ApplicationId}", false);
  let body;
  body = JSON.stringify(
    (0, import_smithy_client.take)(input, {
      Description: [],
      Monitors: (_) => (0, import_smithy_client._json)(_),
      Name: [],
      Tags: (_) => (0, import_smithy_client._json)(_)
    })
  );
  b.m("POST").h(headers).b(body);
  return b.build();
}, "se_CreateEnvironmentCommand");
var se_CreateExtensionCommand = /* @__PURE__ */ __name(async (input, context) => {
  const b = (0, import_core.requestBuilder)(input, context);
  const headers = (0, import_smithy_client.map)({}, isSerializableHeaderValue, {
    "content-type": "application/json",
    [_lvn]: [() => isSerializableHeaderValue(input[_LVN]), () => input[_LVN].toString()]
  });
  b.bp("/extensions");
  let body;
  body = JSON.stringify(
    (0, import_smithy_client.take)(input, {
      Actions: (_) => (0, import_smithy_client._json)(_),
      Description: [],
      Name: [],
      Parameters: (_) => (0, import_smithy_client._json)(_),
      Tags: (_) => (0, import_smithy_client._json)(_)
    })
  );
  b.m("POST").h(headers).b(body);
  return b.build();
}, "se_CreateExtensionCommand");
var se_CreateExtensionAssociationCommand = /* @__PURE__ */ __name(async (input, context) => {
  const b = (0, import_core.requestBuilder)(input, context);
  const headers = {
    "content-type": "application/json"
  };
  b.bp("/extensionassociations");
  let body;
  body = JSON.stringify(
    (0, import_smithy_client.take)(input, {
      ExtensionIdentifier: [],
      ExtensionVersionNumber: [],
      Parameters: (_) => (0, import_smithy_client._json)(_),
      ResourceIdentifier: [],
      Tags: (_) => (0, import_smithy_client._json)(_)
    })
  );
  b.m("POST").h(headers).b(body);
  return b.build();
}, "se_CreateExtensionAssociationCommand");
var se_CreateHostedConfigurationVersionCommand = /* @__PURE__ */ __name(async (input, context) => {
  const b = (0, import_core.requestBuilder)(input, context);
  const headers = (0, import_smithy_client.map)({}, isSerializableHeaderValue, {
    [_ct]: input[_CT] || "application/octet-stream",
    [_d]: input[_D],
    [_lvn]: [() => isSerializableHeaderValue(input[_LVN]), () => input[_LVN].toString()],
    [_v]: input[_VL]
  });
  b.bp("/applications/{ApplicationId}/configurationprofiles/{ConfigurationProfileId}/hostedconfigurationversions");
  b.p("ApplicationId", () => input.ApplicationId, "{ApplicationId}", false);
  b.p("ConfigurationProfileId", () => input.ConfigurationProfileId, "{ConfigurationProfileId}", false);
  let body;
  if (input.Content !== void 0) {
    body = input.Content;
  }
  b.m("POST").h(headers).b(body);
  return b.build();
}, "se_CreateHostedConfigurationVersionCommand");
var se_DeleteApplicationCommand = /* @__PURE__ */ __name(async (input, context) => {
  const b = (0, import_core.requestBuilder)(input, context);
  const headers = {};
  b.bp("/applications/{ApplicationId}");
  b.p("ApplicationId", () => input.ApplicationId, "{ApplicationId}", false);
  let body;
  b.m("DELETE").h(headers).b(body);
  return b.build();
}, "se_DeleteApplicationCommand");
var se_DeleteConfigurationProfileCommand = /* @__PURE__ */ __name(async (input, context) => {
  const b = (0, import_core.requestBuilder)(input, context);
  const headers = {};
  b.bp("/applications/{ApplicationId}/configurationprofiles/{ConfigurationProfileId}");
  b.p("ApplicationId", () => input.ApplicationId, "{ApplicationId}", false);
  b.p("ConfigurationProfileId", () => input.ConfigurationProfileId, "{ConfigurationProfileId}", false);
  let body;
  b.m("DELETE").h(headers).b(body);
  return b.build();
}, "se_DeleteConfigurationProfileCommand");
var se_DeleteDeploymentStrategyCommand = /* @__PURE__ */ __name(async (input, context) => {
  const b = (0, import_core.requestBuilder)(input, context);
  const headers = {};
  b.bp("/deployementstrategies/{DeploymentStrategyId}");
  b.p("DeploymentStrategyId", () => input.DeploymentStrategyId, "{DeploymentStrategyId}", false);
  let body;
  b.m("DELETE").h(headers).b(body);
  return b.build();
}, "se_DeleteDeploymentStrategyCommand");
var se_DeleteEnvironmentCommand = /* @__PURE__ */ __name(async (input, context) => {
  const b = (0, import_core.requestBuilder)(input, context);
  const headers = {};
  b.bp("/applications/{ApplicationId}/environments/{EnvironmentId}");
  b.p("ApplicationId", () => input.ApplicationId, "{ApplicationId}", false);
  b.p("EnvironmentId", () => input.EnvironmentId, "{EnvironmentId}", false);
  let body;
  b.m("DELETE").h(headers).b(body);
  return b.build();
}, "se_DeleteEnvironmentCommand");
var se_DeleteExtensionCommand = /* @__PURE__ */ __name(async (input, context) => {
  const b = (0, import_core.requestBuilder)(input, context);
  const headers = {};
  b.bp("/extensions/{ExtensionIdentifier}");
  b.p("ExtensionIdentifier", () => input.ExtensionIdentifier, "{ExtensionIdentifier}", false);
  const query = (0, import_smithy_client.map)({
    [_ve]: [() => input.VersionNumber !== void 0, () => input[_VN].toString()]
  });
  let body;
  b.m("DELETE").h(headers).q(query).b(body);
  return b.build();
}, "se_DeleteExtensionCommand");
var se_DeleteExtensionAssociationCommand = /* @__PURE__ */ __name(async (input, context) => {
  const b = (0, import_core.requestBuilder)(input, context);
  const headers = {};
  b.bp("/extensionassociations/{ExtensionAssociationId}");
  b.p("ExtensionAssociationId", () => input.ExtensionAssociationId, "{ExtensionAssociationId}", false);
  let body;
  b.m("DELETE").h(headers).b(body);
  return b.build();
}, "se_DeleteExtensionAssociationCommand");
var se_DeleteHostedConfigurationVersionCommand = /* @__PURE__ */ __name(async (input, context) => {
  const b = (0, import_core.requestBuilder)(input, context);
  const headers = {};
  b.bp(
    "/applications/{ApplicationId}/configurationprofiles/{ConfigurationProfileId}/hostedconfigurationversions/{VersionNumber}"
  );
  b.p("ApplicationId", () => input.ApplicationId, "{ApplicationId}", false);
  b.p("ConfigurationProfileId", () => input.ConfigurationProfileId, "{ConfigurationProfileId}", false);
  b.p("VersionNumber", () => input.VersionNumber.toString(), "{VersionNumber}", false);
  let body;
  b.m("DELETE").h(headers).b(body);
  return b.build();
}, "se_DeleteHostedConfigurationVersionCommand");
var se_GetApplicationCommand = /* @__PURE__ */ __name(async (input, context) => {
  const b = (0, import_core.requestBuilder)(input, context);
  const headers = {};
  b.bp("/applications/{ApplicationId}");
  b.p("ApplicationId", () => input.ApplicationId, "{ApplicationId}", false);
  let body;
  b.m("GET").h(headers).b(body);
  return b.build();
}, "se_GetApplicationCommand");
var se_GetConfigurationCommand = /* @__PURE__ */ __name(async (input, context) => {
  const b = (0, import_core.requestBuilder)(input, context);
  const headers = {};
  b.bp("/applications/{Application}/environments/{Environment}/configurations/{Configuration}");
  b.p("Application", () => input.Application, "{Application}", false);
  b.p("Environment", () => input.Environment, "{Environment}", false);
  b.p("Configuration", () => input.Configuration, "{Configuration}", false);
  const query = (0, import_smithy_client.map)({
    [_ci]: [, (0, import_smithy_client.expectNonNull)(input[_CI], `ClientId`)],
    [_ccv]: [, input[_CCV]]
  });
  let body;
  b.m("GET").h(headers).q(query).b(body);
  return b.build();
}, "se_GetConfigurationCommand");
var se_GetConfigurationProfileCommand = /* @__PURE__ */ __name(async (input, context) => {
  const b = (0, import_core.requestBuilder)(input, context);
  const headers = {};
  b.bp("/applications/{ApplicationId}/configurationprofiles/{ConfigurationProfileId}");
  b.p("ApplicationId", () => input.ApplicationId, "{ApplicationId}", false);
  b.p("ConfigurationProfileId", () => input.ConfigurationProfileId, "{ConfigurationProfileId}", false);
  let body;
  b.m("GET").h(headers).b(body);
  return b.build();
}, "se_GetConfigurationProfileCommand");
var se_GetDeploymentCommand = /* @__PURE__ */ __name(async (input, context) => {
  const b = (0, import_core.requestBuilder)(input, context);
  const headers = {};
  b.bp("/applications/{ApplicationId}/environments/{EnvironmentId}/deployments/{DeploymentNumber}");
  b.p("ApplicationId", () => input.ApplicationId, "{ApplicationId}", false);
  b.p("EnvironmentId", () => input.EnvironmentId, "{EnvironmentId}", false);
  b.p("DeploymentNumber", () => input.DeploymentNumber.toString(), "{DeploymentNumber}", false);
  let body;
  b.m("GET").h(headers).b(body);
  return b.build();
}, "se_GetDeploymentCommand");
var se_GetDeploymentStrategyCommand = /* @__PURE__ */ __name(async (input, context) => {
  const b = (0, import_core.requestBuilder)(input, context);
  const headers = {};
  b.bp("/deploymentstrategies/{DeploymentStrategyId}");
  b.p("DeploymentStrategyId", () => input.DeploymentStrategyId, "{DeploymentStrategyId}", false);
  let body;
  b.m("GET").h(headers).b(body);
  return b.build();
}, "se_GetDeploymentStrategyCommand");
var se_GetEnvironmentCommand = /* @__PURE__ */ __name(async (input, context) => {
  const b = (0, import_core.requestBuilder)(input, context);
  const headers = {};
  b.bp("/applications/{ApplicationId}/environments/{EnvironmentId}");
  b.p("ApplicationId", () => input.ApplicationId, "{ApplicationId}", false);
  b.p("EnvironmentId", () => input.EnvironmentId, "{EnvironmentId}", false);
  let body;
  b.m("GET").h(headers).b(body);
  return b.build();
}, "se_GetEnvironmentCommand");
var se_GetExtensionCommand = /* @__PURE__ */ __name(async (input, context) => {
  const b = (0, import_core.requestBuilder)(input, context);
  const headers = {};
  b.bp("/extensions/{ExtensionIdentifier}");
  b.p("ExtensionIdentifier", () => input.ExtensionIdentifier, "{ExtensionIdentifier}", false);
  const query = (0, import_smithy_client.map)({
    [_vn]: [() => input.VersionNumber !== void 0, () => input[_VN].toString()]
  });
  let body;
  b.m("GET").h(headers).q(query).b(body);
  return b.build();
}, "se_GetExtensionCommand");
var se_GetExtensionAssociationCommand = /* @__PURE__ */ __name(async (input, context) => {
  const b = (0, import_core.requestBuilder)(input, context);
  const headers = {};
  b.bp("/extensionassociations/{ExtensionAssociationId}");
  b.p("ExtensionAssociationId", () => input.ExtensionAssociationId, "{ExtensionAssociationId}", false);
  let body;
  b.m("GET").h(headers).b(body);
  return b.build();
}, "se_GetExtensionAssociationCommand");
var se_GetHostedConfigurationVersionCommand = /* @__PURE__ */ __name(async (input, context) => {
  const b = (0, import_core.requestBuilder)(input, context);
  const headers = {};
  b.bp(
    "/applications/{ApplicationId}/configurationprofiles/{ConfigurationProfileId}/hostedconfigurationversions/{VersionNumber}"
  );
  b.p("ApplicationId", () => input.ApplicationId, "{ApplicationId}", false);
  b.p("ConfigurationProfileId", () => input.ConfigurationProfileId, "{ConfigurationProfileId}", false);
  b.p("VersionNumber", () => input.VersionNumber.toString(), "{VersionNumber}", false);
  let body;
  b.m("GET").h(headers).b(body);
  return b.build();
}, "se_GetHostedConfigurationVersionCommand");
var se_ListApplicationsCommand = /* @__PURE__ */ __name(async (input, context) => {
  const b = (0, import_core.requestBuilder)(input, context);
  const headers = {};
  b.bp("/applications");
  const query = (0, import_smithy_client.map)({
    [_mr]: [() => input.MaxResults !== void 0, () => input[_MR].toString()],
    [_nt]: [, input[_NT]]
  });
  let body;
  b.m("GET").h(headers).q(query).b(body);
  return b.build();
}, "se_ListApplicationsCommand");
var se_ListConfigurationProfilesCommand = /* @__PURE__ */ __name(async (input, context) => {
  const b = (0, import_core.requestBuilder)(input, context);
  const headers = {};
  b.bp("/applications/{ApplicationId}/configurationprofiles");
  b.p("ApplicationId", () => input.ApplicationId, "{ApplicationId}", false);
  const query = (0, import_smithy_client.map)({
    [_mr]: [() => input.MaxResults !== void 0, () => input[_MR].toString()],
    [_nt]: [, input[_NT]],
    [_t]: [, input[_T]]
  });
  let body;
  b.m("GET").h(headers).q(query).b(body);
  return b.build();
}, "se_ListConfigurationProfilesCommand");
var se_ListDeploymentsCommand = /* @__PURE__ */ __name(async (input, context) => {
  const b = (0, import_core.requestBuilder)(input, context);
  const headers = {};
  b.bp("/applications/{ApplicationId}/environments/{EnvironmentId}/deployments");
  b.p("ApplicationId", () => input.ApplicationId, "{ApplicationId}", false);
  b.p("EnvironmentId", () => input.EnvironmentId, "{EnvironmentId}", false);
  const query = (0, import_smithy_client.map)({
    [_mr]: [() => input.MaxResults !== void 0, () => input[_MR].toString()],
    [_nt]: [, input[_NT]]
  });
  let body;
  b.m("GET").h(headers).q(query).b(body);
  return b.build();
}, "se_ListDeploymentsCommand");
var se_ListDeploymentStrategiesCommand = /* @__PURE__ */ __name(async (input, context) => {
  const b = (0, import_core.requestBuilder)(input, context);
  const headers = {};
  b.bp("/deploymentstrategies");
  const query = (0, import_smithy_client.map)({
    [_mr]: [() => input.MaxResults !== void 0, () => input[_MR].toString()],
    [_nt]: [, input[_NT]]
  });
  let body;
  b.m("GET").h(headers).q(query).b(body);
  return b.build();
}, "se_ListDeploymentStrategiesCommand");
var se_ListEnvironmentsCommand = /* @__PURE__ */ __name(async (input, context) => {
  const b = (0, import_core.requestBuilder)(input, context);
  const headers = {};
  b.bp("/applications/{ApplicationId}/environments");
  b.p("ApplicationId", () => input.ApplicationId, "{ApplicationId}", false);
  const query = (0, import_smithy_client.map)({
    [_mr]: [() => input.MaxResults !== void 0, () => input[_MR].toString()],
    [_nt]: [, input[_NT]]
  });
  let body;
  b.m("GET").h(headers).q(query).b(body);
  return b.build();
}, "se_ListEnvironmentsCommand");
var se_ListExtensionAssociationsCommand = /* @__PURE__ */ __name(async (input, context) => {
  const b = (0, import_core.requestBuilder)(input, context);
  const headers = {};
  b.bp("/extensionassociations");
  const query = (0, import_smithy_client.map)({
    [_ri]: [, input[_RI]],
    [_ei]: [, input[_EI]],
    [_evn]: [() => input.ExtensionVersionNumber !== void 0, () => input[_EVN].toString()],
    [_mr]: [() => input.MaxResults !== void 0, () => input[_MR].toString()],
    [_nt]: [, input[_NT]]
  });
  let body;
  b.m("GET").h(headers).q(query).b(body);
  return b.build();
}, "se_ListExtensionAssociationsCommand");
var se_ListExtensionsCommand = /* @__PURE__ */ __name(async (input, context) => {
  const b = (0, import_core.requestBuilder)(input, context);
  const headers = {};
  b.bp("/extensions");
  const query = (0, import_smithy_client.map)({
    [_mr]: [() => input.MaxResults !== void 0, () => input[_MR].toString()],
    [_nt]: [, input[_NT]],
    [_n]: [, input[_N]]
  });
  let body;
  b.m("GET").h(headers).q(query).b(body);
  return b.build();
}, "se_ListExtensionsCommand");
var se_ListHostedConfigurationVersionsCommand = /* @__PURE__ */ __name(async (input, context) => {
  const b = (0, import_core.requestBuilder)(input, context);
  const headers = {};
  b.bp("/applications/{ApplicationId}/configurationprofiles/{ConfigurationProfileId}/hostedconfigurationversions");
  b.p("ApplicationId", () => input.ApplicationId, "{ApplicationId}", false);
  b.p("ConfigurationProfileId", () => input.ConfigurationProfileId, "{ConfigurationProfileId}", false);
  const query = (0, import_smithy_client.map)({
    [_mr]: [() => input.MaxResults !== void 0, () => input[_MR].toString()],
    [_nt]: [, input[_NT]],
    [_vl]: [, input[_VL]]
  });
  let body;
  b.m("GET").h(headers).q(query).b(body);
  return b.build();
}, "se_ListHostedConfigurationVersionsCommand");
var se_ListTagsForResourceCommand = /* @__PURE__ */ __name(async (input, context) => {
  const b = (0, import_core.requestBuilder)(input, context);
  const headers = {};
  b.bp("/tags/{ResourceArn}");
  b.p("ResourceArn", () => input.ResourceArn, "{ResourceArn}", false);
  let body;
  b.m("GET").h(headers).b(body);
  return b.build();
}, "se_ListTagsForResourceCommand");
var se_StartDeploymentCommand = /* @__PURE__ */ __name(async (input, context) => {
  const b = (0, import_core.requestBuilder)(input, context);
  const headers = {
    "content-type": "application/json"
  };
  b.bp("/applications/{ApplicationId}/environments/{EnvironmentId}/deployments");
  b.p("ApplicationId", () => input.ApplicationId, "{ApplicationId}", false);
  b.p("EnvironmentId", () => input.EnvironmentId, "{EnvironmentId}", false);
  let body;
  body = JSON.stringify(
    (0, import_smithy_client.take)(input, {
      ConfigurationProfileId: [],
      ConfigurationVersion: [],
      DeploymentStrategyId: [],
      Description: [],
      KmsKeyIdentifier: [],
      Tags: (_) => (0, import_smithy_client._json)(_)
    })
  );
  b.m("POST").h(headers).b(body);
  return b.build();
}, "se_StartDeploymentCommand");
var se_StopDeploymentCommand = /* @__PURE__ */ __name(async (input, context) => {
  const b = (0, import_core.requestBuilder)(input, context);
  const headers = {};
  b.bp("/applications/{ApplicationId}/environments/{EnvironmentId}/deployments/{DeploymentNumber}");
  b.p("ApplicationId", () => input.ApplicationId, "{ApplicationId}", false);
  b.p("EnvironmentId", () => input.EnvironmentId, "{EnvironmentId}", false);
  b.p("DeploymentNumber", () => input.DeploymentNumber.toString(), "{DeploymentNumber}", false);
  let body;
  b.m("DELETE").h(headers).b(body);
  return b.build();
}, "se_StopDeploymentCommand");
var se_TagResourceCommand = /* @__PURE__ */ __name(async (input, context) => {
  const b = (0, import_core.requestBuilder)(input, context);
  const headers = {
    "content-type": "application/json"
  };
  b.bp("/tags/{ResourceArn}");
  b.p("ResourceArn", () => input.ResourceArn, "{ResourceArn}", false);
  let body;
  body = JSON.stringify(
    (0, import_smithy_client.take)(input, {
      Tags: (_) => (0, import_smithy_client._json)(_)
    })
  );
  b.m("POST").h(headers).b(body);
  return b.build();
}, "se_TagResourceCommand");
var se_UntagResourceCommand = /* @__PURE__ */ __name(async (input, context) => {
  const b = (0, import_core.requestBuilder)(input, context);
  const headers = {};
  b.bp("/tags/{ResourceArn}");
  b.p("ResourceArn", () => input.ResourceArn, "{ResourceArn}", false);
  const query = (0, import_smithy_client.map)({
    [_tK]: [
      (0, import_smithy_client.expectNonNull)(input.TagKeys, `TagKeys`) != null,
      () => (input[_TK] || []).map((_entry) => _entry)
    ]
  });
  let body;
  b.m("DELETE").h(headers).q(query).b(body);
  return b.build();
}, "se_UntagResourceCommand");
var se_UpdateApplicationCommand = /* @__PURE__ */ __name(async (input, context) => {
  const b = (0, import_core.requestBuilder)(input, context);
  const headers = {
    "content-type": "application/json"
  };
  b.bp("/applications/{ApplicationId}");
  b.p("ApplicationId", () => input.ApplicationId, "{ApplicationId}", false);
  let body;
  body = JSON.stringify(
    (0, import_smithy_client.take)(input, {
      Description: [],
      Name: []
    })
  );
  b.m("PATCH").h(headers).b(body);
  return b.build();
}, "se_UpdateApplicationCommand");
var se_UpdateConfigurationProfileCommand = /* @__PURE__ */ __name(async (input, context) => {
  const b = (0, import_core.requestBuilder)(input, context);
  const headers = {
    "content-type": "application/json"
  };
  b.bp("/applications/{ApplicationId}/configurationprofiles/{ConfigurationProfileId}");
  b.p("ApplicationId", () => input.ApplicationId, "{ApplicationId}", false);
  b.p("ConfigurationProfileId", () => input.ConfigurationProfileId, "{ConfigurationProfileId}", false);
  let body;
  body = JSON.stringify(
    (0, import_smithy_client.take)(input, {
      Description: [],
      KmsKeyIdentifier: [],
      Name: [],
      RetrievalRoleArn: [],
      Validators: (_) => (0, import_smithy_client._json)(_)
    })
  );
  b.m("PATCH").h(headers).b(body);
  return b.build();
}, "se_UpdateConfigurationProfileCommand");
var se_UpdateDeploymentStrategyCommand = /* @__PURE__ */ __name(async (input, context) => {
  const b = (0, import_core.requestBuilder)(input, context);
  const headers = {
    "content-type": "application/json"
  };
  b.bp("/deploymentstrategies/{DeploymentStrategyId}");
  b.p("DeploymentStrategyId", () => input.DeploymentStrategyId, "{DeploymentStrategyId}", false);
  let body;
  body = JSON.stringify(
    (0, import_smithy_client.take)(input, {
      DeploymentDurationInMinutes: [],
      Description: [],
      FinalBakeTimeInMinutes: [],
      GrowthFactor: (_) => (0, import_smithy_client.serializeFloat)(_),
      GrowthType: []
    })
  );
  b.m("PATCH").h(headers).b(body);
  return b.build();
}, "se_UpdateDeploymentStrategyCommand");
var se_UpdateEnvironmentCommand = /* @__PURE__ */ __name(async (input, context) => {
  const b = (0, import_core.requestBuilder)(input, context);
  const headers = {
    "content-type": "application/json"
  };
  b.bp("/applications/{ApplicationId}/environments/{EnvironmentId}");
  b.p("ApplicationId", () => input.ApplicationId, "{ApplicationId}", false);
  b.p("EnvironmentId", () => input.EnvironmentId, "{EnvironmentId}", false);
  let body;
  body = JSON.stringify(
    (0, import_smithy_client.take)(input, {
      Description: [],
      Monitors: (_) => (0, import_smithy_client._json)(_),
      Name: []
    })
  );
  b.m("PATCH").h(headers).b(body);
  return b.build();
}, "se_UpdateEnvironmentCommand");
var se_UpdateExtensionCommand = /* @__PURE__ */ __name(async (input, context) => {
  const b = (0, import_core.requestBuilder)(input, context);
  const headers = {
    "content-type": "application/json"
  };
  b.bp("/extensions/{ExtensionIdentifier}");
  b.p("ExtensionIdentifier", () => input.ExtensionIdentifier, "{ExtensionIdentifier}", false);
  let body;
  body = JSON.stringify(
    (0, import_smithy_client.take)(input, {
      Actions: (_) => (0, import_smithy_client._json)(_),
      Description: [],
      Parameters: (_) => (0, import_smithy_client._json)(_),
      VersionNumber: []
    })
  );
  b.m("PATCH").h(headers).b(body);
  return b.build();
}, "se_UpdateExtensionCommand");
var se_UpdateExtensionAssociationCommand = /* @__PURE__ */ __name(async (input, context) => {
  const b = (0, import_core.requestBuilder)(input, context);
  const headers = {
    "content-type": "application/json"
  };
  b.bp("/extensionassociations/{ExtensionAssociationId}");
  b.p("ExtensionAssociationId", () => input.ExtensionAssociationId, "{ExtensionAssociationId}", false);
  let body;
  body = JSON.stringify(
    (0, import_smithy_client.take)(input, {
      Parameters: (_) => (0, import_smithy_client._json)(_)
    })
  );
  b.m("PATCH").h(headers).b(body);
  return b.build();
}, "se_UpdateExtensionAssociationCommand");
var se_ValidateConfigurationCommand = /* @__PURE__ */ __name(async (input, context) => {
  const b = (0, import_core.requestBuilder)(input, context);
  const headers = {};
  b.bp("/applications/{ApplicationId}/configurationprofiles/{ConfigurationProfileId}/validators");
  b.p("ApplicationId", () => input.ApplicationId, "{ApplicationId}", false);
  b.p("ConfigurationProfileId", () => input.ConfigurationProfileId, "{ConfigurationProfileId}", false);
  const query = (0, import_smithy_client.map)({
    [_cv]: [, (0, import_smithy_client.expectNonNull)(input[_CV], `ConfigurationVersion`)]
  });
  let body;
  b.m("POST").h(headers).q(query).b(body);
  return b.build();
}, "se_ValidateConfigurationCommand");
var de_CreateApplicationCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode !== 201 && output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const contents = (0, import_smithy_client.map)({
    $metadata: deserializeMetadata(output)
  });
  const data = (0, import_smithy_client.expectNonNull)((0, import_smithy_client.expectObject)(await parseBody(output.body, context)), "body");
  const doc = (0, import_smithy_client.take)(data, {
    Description: import_smithy_client.expectString,
    Id: import_smithy_client.expectString,
    Name: import_smithy_client.expectString
  });
  Object.assign(contents, doc);
  return contents;
}, "de_CreateApplicationCommand");
var de_CreateConfigurationProfileCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode !== 201 && output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const contents = (0, import_smithy_client.map)({
    $metadata: deserializeMetadata(output)
  });
  const data = (0, import_smithy_client.expectNonNull)((0, import_smithy_client.expectObject)(await parseBody(output.body, context)), "body");
  const doc = (0, import_smithy_client.take)(data, {
    ApplicationId: import_smithy_client.expectString,
    Description: import_smithy_client.expectString,
    Id: import_smithy_client.expectString,
    KmsKeyArn: import_smithy_client.expectString,
    KmsKeyIdentifier: import_smithy_client.expectString,
    LocationUri: import_smithy_client.expectString,
    Name: import_smithy_client.expectString,
    RetrievalRoleArn: import_smithy_client.expectString,
    Type: import_smithy_client.expectString,
    Validators: import_smithy_client._json
  });
  Object.assign(contents, doc);
  return contents;
}, "de_CreateConfigurationProfileCommand");
var de_CreateDeploymentStrategyCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode !== 201 && output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const contents = (0, import_smithy_client.map)({
    $metadata: deserializeMetadata(output)
  });
  const data = (0, import_smithy_client.expectNonNull)((0, import_smithy_client.expectObject)(await parseBody(output.body, context)), "body");
  const doc = (0, import_smithy_client.take)(data, {
    DeploymentDurationInMinutes: import_smithy_client.expectInt32,
    Description: import_smithy_client.expectString,
    FinalBakeTimeInMinutes: import_smithy_client.expectInt32,
    GrowthFactor: import_smithy_client.limitedParseFloat32,
    GrowthType: import_smithy_client.expectString,
    Id: import_smithy_client.expectString,
    Name: import_smithy_client.expectString,
    ReplicateTo: import_smithy_client.expectString
  });
  Object.assign(contents, doc);
  return contents;
}, "de_CreateDeploymentStrategyCommand");
var de_CreateEnvironmentCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode !== 201 && output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const contents = (0, import_smithy_client.map)({
    $metadata: deserializeMetadata(output)
  });
  const data = (0, import_smithy_client.expectNonNull)((0, import_smithy_client.expectObject)(await parseBody(output.body, context)), "body");
  const doc = (0, import_smithy_client.take)(data, {
    ApplicationId: import_smithy_client.expectString,
    Description: import_smithy_client.expectString,
    Id: import_smithy_client.expectString,
    Monitors: import_smithy_client._json,
    Name: import_smithy_client.expectString,
    State: import_smithy_client.expectString
  });
  Object.assign(contents, doc);
  return contents;
}, "de_CreateEnvironmentCommand");
var de_CreateExtensionCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode !== 201 && output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const contents = (0, import_smithy_client.map)({
    $metadata: deserializeMetadata(output)
  });
  const data = (0, import_smithy_client.expectNonNull)((0, import_smithy_client.expectObject)(await parseBody(output.body, context)), "body");
  const doc = (0, import_smithy_client.take)(data, {
    Actions: import_smithy_client._json,
    Arn: import_smithy_client.expectString,
    Description: import_smithy_client.expectString,
    Id: import_smithy_client.expectString,
    Name: import_smithy_client.expectString,
    Parameters: import_smithy_client._json,
    VersionNumber: import_smithy_client.expectInt32
  });
  Object.assign(contents, doc);
  return contents;
}, "de_CreateExtensionCommand");
var de_CreateExtensionAssociationCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode !== 201 && output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const contents = (0, import_smithy_client.map)({
    $metadata: deserializeMetadata(output)
  });
  const data = (0, import_smithy_client.expectNonNull)((0, import_smithy_client.expectObject)(await parseBody(output.body, context)), "body");
  const doc = (0, import_smithy_client.take)(data, {
    Arn: import_smithy_client.expectString,
    ExtensionArn: import_smithy_client.expectString,
    ExtensionVersionNumber: import_smithy_client.expectInt32,
    Id: import_smithy_client.expectString,
    Parameters: import_smithy_client._json,
    ResourceArn: import_smithy_client.expectString
  });
  Object.assign(contents, doc);
  return contents;
}, "de_CreateExtensionAssociationCommand");
var de_CreateHostedConfigurationVersionCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode !== 201 && output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const contents = (0, import_smithy_client.map)({
    $metadata: deserializeMetadata(output),
    [_AI]: [, output.headers[_ai]],
    [_CPI]: [, output.headers[_cpi]],
    [_VN]: [() => void 0 !== output.headers[_vn_], () => (0, import_smithy_client.strictParseInt32)(output.headers[_vn_])],
    [_D]: [, output.headers[_d]],
    [_CT]: [, output.headers[_ct]],
    [_VL]: [, output.headers[_v]],
    [_KKA]: [, output.headers[_k]]
  });
  const data = await (0, import_smithy_client.collectBody)(output.body, context);
  contents.Content = data;
  return contents;
}, "de_CreateHostedConfigurationVersionCommand");
var de_DeleteApplicationCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode !== 204 && output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const contents = (0, import_smithy_client.map)({
    $metadata: deserializeMetadata(output)
  });
  await (0, import_smithy_client.collectBody)(output.body, context);
  return contents;
}, "de_DeleteApplicationCommand");
var de_DeleteConfigurationProfileCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode !== 204 && output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const contents = (0, import_smithy_client.map)({
    $metadata: deserializeMetadata(output)
  });
  await (0, import_smithy_client.collectBody)(output.body, context);
  return contents;
}, "de_DeleteConfigurationProfileCommand");
var de_DeleteDeploymentStrategyCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode !== 204 && output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const contents = (0, import_smithy_client.map)({
    $metadata: deserializeMetadata(output)
  });
  await (0, import_smithy_client.collectBody)(output.body, context);
  return contents;
}, "de_DeleteDeploymentStrategyCommand");
var de_DeleteEnvironmentCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode !== 204 && output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const contents = (0, import_smithy_client.map)({
    $metadata: deserializeMetadata(output)
  });
  await (0, import_smithy_client.collectBody)(output.body, context);
  return contents;
}, "de_DeleteEnvironmentCommand");
var de_DeleteExtensionCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode !== 204 && output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const contents = (0, import_smithy_client.map)({
    $metadata: deserializeMetadata(output)
  });
  await (0, import_smithy_client.collectBody)(output.body, context);
  return contents;
}, "de_DeleteExtensionCommand");
var de_DeleteExtensionAssociationCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode !== 204 && output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const contents = (0, import_smithy_client.map)({
    $metadata: deserializeMetadata(output)
  });
  await (0, import_smithy_client.collectBody)(output.body, context);
  return contents;
}, "de_DeleteExtensionAssociationCommand");
var de_DeleteHostedConfigurationVersionCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode !== 204 && output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const contents = (0, import_smithy_client.map)({
    $metadata: deserializeMetadata(output)
  });
  await (0, import_smithy_client.collectBody)(output.body, context);
  return contents;
}, "de_DeleteHostedConfigurationVersionCommand");
var de_GetApplicationCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode !== 200 && output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const contents = (0, import_smithy_client.map)({
    $metadata: deserializeMetadata(output)
  });
  const data = (0, import_smithy_client.expectNonNull)((0, import_smithy_client.expectObject)(await parseBody(output.body, context)), "body");
  const doc = (0, import_smithy_client.take)(data, {
    Description: import_smithy_client.expectString,
    Id: import_smithy_client.expectString,
    Name: import_smithy_client.expectString
  });
  Object.assign(contents, doc);
  return contents;
}, "de_GetApplicationCommand");
var de_GetConfigurationCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode !== 200 && output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const contents = (0, import_smithy_client.map)({
    $metadata: deserializeMetadata(output),
    [_CV]: [, output.headers[_cv_]],
    [_CT]: [, output.headers[_ct]]
  });
  const data = await (0, import_smithy_client.collectBody)(output.body, context);
  contents.Content = data;
  return contents;
}, "de_GetConfigurationCommand");
var de_GetConfigurationProfileCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode !== 200 && output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const contents = (0, import_smithy_client.map)({
    $metadata: deserializeMetadata(output)
  });
  const data = (0, import_smithy_client.expectNonNull)((0, import_smithy_client.expectObject)(await parseBody(output.body, context)), "body");
  const doc = (0, import_smithy_client.take)(data, {
    ApplicationId: import_smithy_client.expectString,
    Description: import_smithy_client.expectString,
    Id: import_smithy_client.expectString,
    KmsKeyArn: import_smithy_client.expectString,
    KmsKeyIdentifier: import_smithy_client.expectString,
    LocationUri: import_smithy_client.expectString,
    Name: import_smithy_client.expectString,
    RetrievalRoleArn: import_smithy_client.expectString,
    Type: import_smithy_client.expectString,
    Validators: import_smithy_client._json
  });
  Object.assign(contents, doc);
  return contents;
}, "de_GetConfigurationProfileCommand");
var de_GetDeploymentCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode !== 200 && output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const contents = (0, import_smithy_client.map)({
    $metadata: deserializeMetadata(output)
  });
  const data = (0, import_smithy_client.expectNonNull)((0, import_smithy_client.expectObject)(await parseBody(output.body, context)), "body");
  const doc = (0, import_smithy_client.take)(data, {
    ApplicationId: import_smithy_client.expectString,
    AppliedExtensions: import_smithy_client._json,
    CompletedAt: (_) => (0, import_smithy_client.expectNonNull)((0, import_smithy_client.parseRfc3339DateTimeWithOffset)(_)),
    ConfigurationLocationUri: import_smithy_client.expectString,
    ConfigurationName: import_smithy_client.expectString,
    ConfigurationProfileId: import_smithy_client.expectString,
    ConfigurationVersion: import_smithy_client.expectString,
    DeploymentDurationInMinutes: import_smithy_client.expectInt32,
    DeploymentNumber: import_smithy_client.expectInt32,
    DeploymentStrategyId: import_smithy_client.expectString,
    Description: import_smithy_client.expectString,
    EnvironmentId: import_smithy_client.expectString,
    EventLog: (_) => de_DeploymentEvents(_, context),
    FinalBakeTimeInMinutes: import_smithy_client.expectInt32,
    GrowthFactor: import_smithy_client.limitedParseFloat32,
    GrowthType: import_smithy_client.expectString,
    KmsKeyArn: import_smithy_client.expectString,
    KmsKeyIdentifier: import_smithy_client.expectString,
    PercentageComplete: import_smithy_client.limitedParseFloat32,
    StartedAt: (_) => (0, import_smithy_client.expectNonNull)((0, import_smithy_client.parseRfc3339DateTimeWithOffset)(_)),
    State: import_smithy_client.expectString,
    VersionLabel: import_smithy_client.expectString
  });
  Object.assign(contents, doc);
  return contents;
}, "de_GetDeploymentCommand");
var de_GetDeploymentStrategyCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode !== 200 && output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const contents = (0, import_smithy_client.map)({
    $metadata: deserializeMetadata(output)
  });
  const data = (0, import_smithy_client.expectNonNull)((0, import_smithy_client.expectObject)(await parseBody(output.body, context)), "body");
  const doc = (0, import_smithy_client.take)(data, {
    DeploymentDurationInMinutes: import_smithy_client.expectInt32,
    Description: import_smithy_client.expectString,
    FinalBakeTimeInMinutes: import_smithy_client.expectInt32,
    GrowthFactor: import_smithy_client.limitedParseFloat32,
    GrowthType: import_smithy_client.expectString,
    Id: import_smithy_client.expectString,
    Name: import_smithy_client.expectString,
    ReplicateTo: import_smithy_client.expectString
  });
  Object.assign(contents, doc);
  return contents;
}, "de_GetDeploymentStrategyCommand");
var de_GetEnvironmentCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode !== 200 && output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const contents = (0, import_smithy_client.map)({
    $metadata: deserializeMetadata(output)
  });
  const data = (0, import_smithy_client.expectNonNull)((0, import_smithy_client.expectObject)(await parseBody(output.body, context)), "body");
  const doc = (0, import_smithy_client.take)(data, {
    ApplicationId: import_smithy_client.expectString,
    Description: import_smithy_client.expectString,
    Id: import_smithy_client.expectString,
    Monitors: import_smithy_client._json,
    Name: import_smithy_client.expectString,
    State: import_smithy_client.expectString
  });
  Object.assign(contents, doc);
  return contents;
}, "de_GetEnvironmentCommand");
var de_GetExtensionCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode !== 200 && output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const contents = (0, import_smithy_client.map)({
    $metadata: deserializeMetadata(output)
  });
  const data = (0, import_smithy_client.expectNonNull)((0, import_smithy_client.expectObject)(await parseBody(output.body, context)), "body");
  const doc = (0, import_smithy_client.take)(data, {
    Actions: import_smithy_client._json,
    Arn: import_smithy_client.expectString,
    Description: import_smithy_client.expectString,
    Id: import_smithy_client.expectString,
    Name: import_smithy_client.expectString,
    Parameters: import_smithy_client._json,
    VersionNumber: import_smithy_client.expectInt32
  });
  Object.assign(contents, doc);
  return contents;
}, "de_GetExtensionCommand");
var de_GetExtensionAssociationCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode !== 200 && output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const contents = (0, import_smithy_client.map)({
    $metadata: deserializeMetadata(output)
  });
  const data = (0, import_smithy_client.expectNonNull)((0, import_smithy_client.expectObject)(await parseBody(output.body, context)), "body");
  const doc = (0, import_smithy_client.take)(data, {
    Arn: import_smithy_client.expectString,
    ExtensionArn: import_smithy_client.expectString,
    ExtensionVersionNumber: import_smithy_client.expectInt32,
    Id: import_smithy_client.expectString,
    Parameters: import_smithy_client._json,
    ResourceArn: import_smithy_client.expectString
  });
  Object.assign(contents, doc);
  return contents;
}, "de_GetExtensionAssociationCommand");
var de_GetHostedConfigurationVersionCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode !== 200 && output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const contents = (0, import_smithy_client.map)({
    $metadata: deserializeMetadata(output),
    [_AI]: [, output.headers[_ai]],
    [_CPI]: [, output.headers[_cpi]],
    [_VN]: [() => void 0 !== output.headers[_vn_], () => (0, import_smithy_client.strictParseInt32)(output.headers[_vn_])],
    [_D]: [, output.headers[_d]],
    [_CT]: [, output.headers[_ct]],
    [_VL]: [, output.headers[_v]],
    [_KKA]: [, output.headers[_k]]
  });
  const data = await (0, import_smithy_client.collectBody)(output.body, context);
  contents.Content = data;
  return contents;
}, "de_GetHostedConfigurationVersionCommand");
var de_ListApplicationsCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode !== 200 && output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const contents = (0, import_smithy_client.map)({
    $metadata: deserializeMetadata(output)
  });
  const data = (0, import_smithy_client.expectNonNull)((0, import_smithy_client.expectObject)(await parseBody(output.body, context)), "body");
  const doc = (0, import_smithy_client.take)(data, {
    Items: import_smithy_client._json,
    NextToken: import_smithy_client.expectString
  });
  Object.assign(contents, doc);
  return contents;
}, "de_ListApplicationsCommand");
var de_ListConfigurationProfilesCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode !== 200 && output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const contents = (0, import_smithy_client.map)({
    $metadata: deserializeMetadata(output)
  });
  const data = (0, import_smithy_client.expectNonNull)((0, import_smithy_client.expectObject)(await parseBody(output.body, context)), "body");
  const doc = (0, import_smithy_client.take)(data, {
    Items: import_smithy_client._json,
    NextToken: import_smithy_client.expectString
  });
  Object.assign(contents, doc);
  return contents;
}, "de_ListConfigurationProfilesCommand");
var de_ListDeploymentsCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode !== 200 && output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const contents = (0, import_smithy_client.map)({
    $metadata: deserializeMetadata(output)
  });
  const data = (0, import_smithy_client.expectNonNull)((0, import_smithy_client.expectObject)(await parseBody(output.body, context)), "body");
  const doc = (0, import_smithy_client.take)(data, {
    Items: (_) => de_DeploymentList(_, context),
    NextToken: import_smithy_client.expectString
  });
  Object.assign(contents, doc);
  return contents;
}, "de_ListDeploymentsCommand");
var de_ListDeploymentStrategiesCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode !== 200 && output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const contents = (0, import_smithy_client.map)({
    $metadata: deserializeMetadata(output)
  });
  const data = (0, import_smithy_client.expectNonNull)((0, import_smithy_client.expectObject)(await parseBody(output.body, context)), "body");
  const doc = (0, import_smithy_client.take)(data, {
    Items: (_) => de_DeploymentStrategyList(_, context),
    NextToken: import_smithy_client.expectString
  });
  Object.assign(contents, doc);
  return contents;
}, "de_ListDeploymentStrategiesCommand");
var de_ListEnvironmentsCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode !== 200 && output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const contents = (0, import_smithy_client.map)({
    $metadata: deserializeMetadata(output)
  });
  const data = (0, import_smithy_client.expectNonNull)((0, import_smithy_client.expectObject)(await parseBody(output.body, context)), "body");
  const doc = (0, import_smithy_client.take)(data, {
    Items: import_smithy_client._json,
    NextToken: import_smithy_client.expectString
  });
  Object.assign(contents, doc);
  return contents;
}, "de_ListEnvironmentsCommand");
var de_ListExtensionAssociationsCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode !== 200 && output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const contents = (0, import_smithy_client.map)({
    $metadata: deserializeMetadata(output)
  });
  const data = (0, import_smithy_client.expectNonNull)((0, import_smithy_client.expectObject)(await parseBody(output.body, context)), "body");
  const doc = (0, import_smithy_client.take)(data, {
    Items: import_smithy_client._json,
    NextToken: import_smithy_client.expectString
  });
  Object.assign(contents, doc);
  return contents;
}, "de_ListExtensionAssociationsCommand");
var de_ListExtensionsCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode !== 200 && output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const contents = (0, import_smithy_client.map)({
    $metadata: deserializeMetadata(output)
  });
  const data = (0, import_smithy_client.expectNonNull)((0, import_smithy_client.expectObject)(await parseBody(output.body, context)), "body");
  const doc = (0, import_smithy_client.take)(data, {
    Items: import_smithy_client._json,
    NextToken: import_smithy_client.expectString
  });
  Object.assign(contents, doc);
  return contents;
}, "de_ListExtensionsCommand");
var de_ListHostedConfigurationVersionsCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode !== 200 && output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const contents = (0, import_smithy_client.map)({
    $metadata: deserializeMetadata(output)
  });
  const data = (0, import_smithy_client.expectNonNull)((0, import_smithy_client.expectObject)(await parseBody(output.body, context)), "body");
  const doc = (0, import_smithy_client.take)(data, {
    Items: import_smithy_client._json,
    NextToken: import_smithy_client.expectString
  });
  Object.assign(contents, doc);
  return contents;
}, "de_ListHostedConfigurationVersionsCommand");
var de_ListTagsForResourceCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode !== 200 && output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const contents = (0, import_smithy_client.map)({
    $metadata: deserializeMetadata(output)
  });
  const data = (0, import_smithy_client.expectNonNull)((0, import_smithy_client.expectObject)(await parseBody(output.body, context)), "body");
  const doc = (0, import_smithy_client.take)(data, {
    Tags: import_smithy_client._json
  });
  Object.assign(contents, doc);
  return contents;
}, "de_ListTagsForResourceCommand");
var de_StartDeploymentCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode !== 201 && output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const contents = (0, import_smithy_client.map)({
    $metadata: deserializeMetadata(output)
  });
  const data = (0, import_smithy_client.expectNonNull)((0, import_smithy_client.expectObject)(await parseBody(output.body, context)), "body");
  const doc = (0, import_smithy_client.take)(data, {
    ApplicationId: import_smithy_client.expectString,
    AppliedExtensions: import_smithy_client._json,
    CompletedAt: (_) => (0, import_smithy_client.expectNonNull)((0, import_smithy_client.parseRfc3339DateTimeWithOffset)(_)),
    ConfigurationLocationUri: import_smithy_client.expectString,
    ConfigurationName: import_smithy_client.expectString,
    ConfigurationProfileId: import_smithy_client.expectString,
    ConfigurationVersion: import_smithy_client.expectString,
    DeploymentDurationInMinutes: import_smithy_client.expectInt32,
    DeploymentNumber: import_smithy_client.expectInt32,
    DeploymentStrategyId: import_smithy_client.expectString,
    Description: import_smithy_client.expectString,
    EnvironmentId: import_smithy_client.expectString,
    EventLog: (_) => de_DeploymentEvents(_, context),
    FinalBakeTimeInMinutes: import_smithy_client.expectInt32,
    GrowthFactor: import_smithy_client.limitedParseFloat32,
    GrowthType: import_smithy_client.expectString,
    KmsKeyArn: import_smithy_client.expectString,
    KmsKeyIdentifier: import_smithy_client.expectString,
    PercentageComplete: import_smithy_client.limitedParseFloat32,
    StartedAt: (_) => (0, import_smithy_client.expectNonNull)((0, import_smithy_client.parseRfc3339DateTimeWithOffset)(_)),
    State: import_smithy_client.expectString,
    VersionLabel: import_smithy_client.expectString
  });
  Object.assign(contents, doc);
  return contents;
}, "de_StartDeploymentCommand");
var de_StopDeploymentCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode !== 202 && output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const contents = (0, import_smithy_client.map)({
    $metadata: deserializeMetadata(output)
  });
  const data = (0, import_smithy_client.expectNonNull)((0, import_smithy_client.expectObject)(await parseBody(output.body, context)), "body");
  const doc = (0, import_smithy_client.take)(data, {
    ApplicationId: import_smithy_client.expectString,
    AppliedExtensions: import_smithy_client._json,
    CompletedAt: (_) => (0, import_smithy_client.expectNonNull)((0, import_smithy_client.parseRfc3339DateTimeWithOffset)(_)),
    ConfigurationLocationUri: import_smithy_client.expectString,
    ConfigurationName: import_smithy_client.expectString,
    ConfigurationProfileId: import_smithy_client.expectString,
    ConfigurationVersion: import_smithy_client.expectString,
    DeploymentDurationInMinutes: import_smithy_client.expectInt32,
    DeploymentNumber: import_smithy_client.expectInt32,
    DeploymentStrategyId: import_smithy_client.expectString,
    Description: import_smithy_client.expectString,
    EnvironmentId: import_smithy_client.expectString,
    EventLog: (_) => de_DeploymentEvents(_, context),
    FinalBakeTimeInMinutes: import_smithy_client.expectInt32,
    GrowthFactor: import_smithy_client.limitedParseFloat32,
    GrowthType: import_smithy_client.expectString,
    KmsKeyArn: import_smithy_client.expectString,
    KmsKeyIdentifier: import_smithy_client.expectString,
    PercentageComplete: import_smithy_client.limitedParseFloat32,
    StartedAt: (_) => (0, import_smithy_client.expectNonNull)((0, import_smithy_client.parseRfc3339DateTimeWithOffset)(_)),
    State: import_smithy_client.expectString,
    VersionLabel: import_smithy_client.expectString
  });
  Object.assign(contents, doc);
  return contents;
}, "de_StopDeploymentCommand");
var de_TagResourceCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode !== 204 && output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const contents = (0, import_smithy_client.map)({
    $metadata: deserializeMetadata(output)
  });
  await (0, import_smithy_client.collectBody)(output.body, context);
  return contents;
}, "de_TagResourceCommand");
var de_UntagResourceCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode !== 204 && output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const contents = (0, import_smithy_client.map)({
    $metadata: deserializeMetadata(output)
  });
  await (0, import_smithy_client.collectBody)(output.body, context);
  return contents;
}, "de_UntagResourceCommand");
var de_UpdateApplicationCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode !== 200 && output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const contents = (0, import_smithy_client.map)({
    $metadata: deserializeMetadata(output)
  });
  const data = (0, import_smithy_client.expectNonNull)((0, import_smithy_client.expectObject)(await parseBody(output.body, context)), "body");
  const doc = (0, import_smithy_client.take)(data, {
    Description: import_smithy_client.expectString,
    Id: import_smithy_client.expectString,
    Name: import_smithy_client.expectString
  });
  Object.assign(contents, doc);
  return contents;
}, "de_UpdateApplicationCommand");
var de_UpdateConfigurationProfileCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode !== 200 && output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const contents = (0, import_smithy_client.map)({
    $metadata: deserializeMetadata(output)
  });
  const data = (0, import_smithy_client.expectNonNull)((0, import_smithy_client.expectObject)(await parseBody(output.body, context)), "body");
  const doc = (0, import_smithy_client.take)(data, {
    ApplicationId: import_smithy_client.expectString,
    Description: import_smithy_client.expectString,
    Id: import_smithy_client.expectString,
    KmsKeyArn: import_smithy_client.expectString,
    KmsKeyIdentifier: import_smithy_client.expectString,
    LocationUri: import_smithy_client.expectString,
    Name: import_smithy_client.expectString,
    RetrievalRoleArn: import_smithy_client.expectString,
    Type: import_smithy_client.expectString,
    Validators: import_smithy_client._json
  });
  Object.assign(contents, doc);
  return contents;
}, "de_UpdateConfigurationProfileCommand");
var de_UpdateDeploymentStrategyCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode !== 200 && output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const contents = (0, import_smithy_client.map)({
    $metadata: deserializeMetadata(output)
  });
  const data = (0, import_smithy_client.expectNonNull)((0, import_smithy_client.expectObject)(await parseBody(output.body, context)), "body");
  const doc = (0, import_smithy_client.take)(data, {
    DeploymentDurationInMinutes: import_smithy_client.expectInt32,
    Description: import_smithy_client.expectString,
    FinalBakeTimeInMinutes: import_smithy_client.expectInt32,
    GrowthFactor: import_smithy_client.limitedParseFloat32,
    GrowthType: import_smithy_client.expectString,
    Id: import_smithy_client.expectString,
    Name: import_smithy_client.expectString,
    ReplicateTo: import_smithy_client.expectString
  });
  Object.assign(contents, doc);
  return contents;
}, "de_UpdateDeploymentStrategyCommand");
var de_UpdateEnvironmentCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode !== 200 && output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const contents = (0, import_smithy_client.map)({
    $metadata: deserializeMetadata(output)
  });
  const data = (0, import_smithy_client.expectNonNull)((0, import_smithy_client.expectObject)(await parseBody(output.body, context)), "body");
  const doc = (0, import_smithy_client.take)(data, {
    ApplicationId: import_smithy_client.expectString,
    Description: import_smithy_client.expectString,
    Id: import_smithy_client.expectString,
    Monitors: import_smithy_client._json,
    Name: import_smithy_client.expectString,
    State: import_smithy_client.expectString
  });
  Object.assign(contents, doc);
  return contents;
}, "de_UpdateEnvironmentCommand");
var de_UpdateExtensionCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode !== 200 && output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const contents = (0, import_smithy_client.map)({
    $metadata: deserializeMetadata(output)
  });
  const data = (0, import_smithy_client.expectNonNull)((0, import_smithy_client.expectObject)(await parseBody(output.body, context)), "body");
  const doc = (0, import_smithy_client.take)(data, {
    Actions: import_smithy_client._json,
    Arn: import_smithy_client.expectString,
    Description: import_smithy_client.expectString,
    Id: import_smithy_client.expectString,
    Name: import_smithy_client.expectString,
    Parameters: import_smithy_client._json,
    VersionNumber: import_smithy_client.expectInt32
  });
  Object.assign(contents, doc);
  return contents;
}, "de_UpdateExtensionCommand");
var de_UpdateExtensionAssociationCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode !== 200 && output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const contents = (0, import_smithy_client.map)({
    $metadata: deserializeMetadata(output)
  });
  const data = (0, import_smithy_client.expectNonNull)((0, import_smithy_client.expectObject)(await parseBody(output.body, context)), "body");
  const doc = (0, import_smithy_client.take)(data, {
    Arn: import_smithy_client.expectString,
    ExtensionArn: import_smithy_client.expectString,
    ExtensionVersionNumber: import_smithy_client.expectInt32,
    Id: import_smithy_client.expectString,
    Parameters: import_smithy_client._json,
    ResourceArn: import_smithy_client.expectString
  });
  Object.assign(contents, doc);
  return contents;
}, "de_UpdateExtensionAssociationCommand");
var de_ValidateConfigurationCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode !== 204 && output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const contents = (0, import_smithy_client.map)({
    $metadata: deserializeMetadata(output)
  });
  await (0, import_smithy_client.collectBody)(output.body, context);
  return contents;
}, "de_ValidateConfigurationCommand");
var de_CommandError = /* @__PURE__ */ __name(async (output, context) => {
  const parsedOutput = {
    ...output,
    body: await parseErrorBody(output.body, context)
  };
  const errorCode = loadRestJsonErrorCode(output, parsedOutput.body);
  switch (errorCode) {
    case "BadRequestException":
    case "com.amazonaws.appconfig#BadRequestException":
      throw await de_BadRequestExceptionRes(parsedOutput, context);
    case "InternalServerException":
    case "com.amazonaws.appconfig#InternalServerException":
      throw await de_InternalServerExceptionRes(parsedOutput, context);
    case "ServiceQuotaExceededException":
    case "com.amazonaws.appconfig#ServiceQuotaExceededException":
      throw await de_ServiceQuotaExceededExceptionRes(parsedOutput, context);
    case "ResourceNotFoundException":
    case "com.amazonaws.appconfig#ResourceNotFoundException":
      throw await de_ResourceNotFoundExceptionRes(parsedOutput, context);
    case "ConflictException":
    case "com.amazonaws.appconfig#ConflictException":
      throw await de_ConflictExceptionRes(parsedOutput, context);
    case "PayloadTooLargeException":
    case "com.amazonaws.appconfig#PayloadTooLargeException":
      throw await de_PayloadTooLargeExceptionRes(parsedOutput, context);
    default:
      const parsedBody = parsedOutput.body;
      return throwDefaultError({
        output,
        parsedBody,
        errorCode
      });
  }
}, "de_CommandError");
var throwDefaultError = (0, import_smithy_client.withBaseException)(AppConfigServiceException);
var de_BadRequestExceptionRes = /* @__PURE__ */ __name(async (parsedOutput, context) => {
  const contents = (0, import_smithy_client.map)({});
  const data = parsedOutput.body;
  const doc = (0, import_smithy_client.take)(data, {
    Details: (_) => (0, import_smithy_client._json)((0, import_core2.awsExpectUnion)(_)),
    Message: import_smithy_client.expectString,
    Reason: import_smithy_client.expectString
  });
  Object.assign(contents, doc);
  const exception = new BadRequestException({
    $metadata: deserializeMetadata(parsedOutput),
    ...contents
  });
  return (0, import_smithy_client.decorateServiceException)(exception, parsedOutput.body);
}, "de_BadRequestExceptionRes");
var de_ConflictExceptionRes = /* @__PURE__ */ __name(async (parsedOutput, context) => {
  const contents = (0, import_smithy_client.map)({});
  const data = parsedOutput.body;
  const doc = (0, import_smithy_client.take)(data, {
    Message: import_smithy_client.expectString
  });
  Object.assign(contents, doc);
  const exception = new ConflictException({
    $metadata: deserializeMetadata(parsedOutput),
    ...contents
  });
  return (0, import_smithy_client.decorateServiceException)(exception, parsedOutput.body);
}, "de_ConflictExceptionRes");
var de_InternalServerExceptionRes = /* @__PURE__ */ __name(async (parsedOutput, context) => {
  const contents = (0, import_smithy_client.map)({});
  const data = parsedOutput.body;
  const doc = (0, import_smithy_client.take)(data, {
    Message: import_smithy_client.expectString
  });
  Object.assign(contents, doc);
  const exception = new InternalServerException({
    $metadata: deserializeMetadata(parsedOutput),
    ...contents
  });
  return (0, import_smithy_client.decorateServiceException)(exception, parsedOutput.body);
}, "de_InternalServerExceptionRes");
var de_PayloadTooLargeExceptionRes = /* @__PURE__ */ __name(async (parsedOutput, context) => {
  const contents = (0, import_smithy_client.map)({});
  const data = parsedOutput.body;
  const doc = (0, import_smithy_client.take)(data, {
    Limit: import_smithy_client.limitedParseFloat32,
    Measure: import_smithy_client.expectString,
    Message: import_smithy_client.expectString,
    Size: import_smithy_client.limitedParseFloat32
  });
  Object.assign(contents, doc);
  const exception = new PayloadTooLargeException({
    $metadata: deserializeMetadata(parsedOutput),
    ...contents
  });
  return (0, import_smithy_client.decorateServiceException)(exception, parsedOutput.body);
}, "de_PayloadTooLargeExceptionRes");
var de_ResourceNotFoundExceptionRes = /* @__PURE__ */ __name(async (parsedOutput, context) => {
  const contents = (0, import_smithy_client.map)({});
  const data = parsedOutput.body;
  const doc = (0, import_smithy_client.take)(data, {
    Message: import_smithy_client.expectString,
    ResourceName: import_smithy_client.expectString
  });
  Object.assign(contents, doc);
  const exception = new ResourceNotFoundException({
    $metadata: deserializeMetadata(parsedOutput),
    ...contents
  });
  return (0, import_smithy_client.decorateServiceException)(exception, parsedOutput.body);
}, "de_ResourceNotFoundExceptionRes");
var de_ServiceQuotaExceededExceptionRes = /* @__PURE__ */ __name(async (parsedOutput, context) => {
  const contents = (0, import_smithy_client.map)({});
  const data = parsedOutput.body;
  const doc = (0, import_smithy_client.take)(data, {
    Message: import_smithy_client.expectString
  });
  Object.assign(contents, doc);
  const exception = new ServiceQuotaExceededException({
    $metadata: deserializeMetadata(parsedOutput),
    ...contents
  });
  return (0, import_smithy_client.decorateServiceException)(exception, parsedOutput.body);
}, "de_ServiceQuotaExceededExceptionRes");
var de_DeploymentEvent = /* @__PURE__ */ __name((output, context) => {
  return (0, import_smithy_client.take)(output, {
    ActionInvocations: import_smithy_client._json,
    Description: import_smithy_client.expectString,
    EventType: import_smithy_client.expectString,
    OccurredAt: (_) => (0, import_smithy_client.expectNonNull)((0, import_smithy_client.parseRfc3339DateTimeWithOffset)(_)),
    TriggeredBy: import_smithy_client.expectString
  });
}, "de_DeploymentEvent");
var de_DeploymentEvents = /* @__PURE__ */ __name((output, context) => {
  const retVal = (output || []).filter((e) => e != null).map((entry) => {
    return de_DeploymentEvent(entry, context);
  });
  return retVal;
}, "de_DeploymentEvents");
var de_DeploymentList = /* @__PURE__ */ __name((output, context) => {
  const retVal = (output || []).filter((e) => e != null).map((entry) => {
    return de_DeploymentSummary(entry, context);
  });
  return retVal;
}, "de_DeploymentList");
var de_DeploymentStrategy = /* @__PURE__ */ __name((output, context) => {
  return (0, import_smithy_client.take)(output, {
    DeploymentDurationInMinutes: import_smithy_client.expectInt32,
    Description: import_smithy_client.expectString,
    FinalBakeTimeInMinutes: import_smithy_client.expectInt32,
    GrowthFactor: import_smithy_client.limitedParseFloat32,
    GrowthType: import_smithy_client.expectString,
    Id: import_smithy_client.expectString,
    Name: import_smithy_client.expectString,
    ReplicateTo: import_smithy_client.expectString
  });
}, "de_DeploymentStrategy");
var de_DeploymentStrategyList = /* @__PURE__ */ __name((output, context) => {
  const retVal = (output || []).filter((e) => e != null).map((entry) => {
    return de_DeploymentStrategy(entry, context);
  });
  return retVal;
}, "de_DeploymentStrategyList");
var de_DeploymentSummary = /* @__PURE__ */ __name((output, context) => {
  return (0, import_smithy_client.take)(output, {
    CompletedAt: (_) => (0, import_smithy_client.expectNonNull)((0, import_smithy_client.parseRfc3339DateTimeWithOffset)(_)),
    ConfigurationName: import_smithy_client.expectString,
    ConfigurationVersion: import_smithy_client.expectString,
    DeploymentDurationInMinutes: import_smithy_client.expectInt32,
    DeploymentNumber: import_smithy_client.expectInt32,
    FinalBakeTimeInMinutes: import_smithy_client.expectInt32,
    GrowthFactor: import_smithy_client.limitedParseFloat32,
    GrowthType: import_smithy_client.expectString,
    PercentageComplete: import_smithy_client.limitedParseFloat32,
    StartedAt: (_) => (0, import_smithy_client.expectNonNull)((0, import_smithy_client.parseRfc3339DateTimeWithOffset)(_)),
    State: import_smithy_client.expectString,
    VersionLabel: import_smithy_client.expectString
  });
}, "de_DeploymentSummary");
var deserializeMetadata = /* @__PURE__ */ __name((output) => ({
  httpStatusCode: output.statusCode,
  requestId: output.headers["x-amzn-requestid"] ?? output.headers["x-amzn-request-id"] ?? output.headers["x-amz-request-id"],
  extendedRequestId: output.headers["x-amz-id-2"],
  cfId: output.headers["x-amz-cf-id"]
}), "deserializeMetadata");
var collectBodyString = /* @__PURE__ */ __name((streamBody, context) => (0, import_smithy_client.collectBody)(streamBody, context).then((body) => context.utf8Encoder(body)), "collectBodyString");
var isSerializableHeaderValue = /* @__PURE__ */ __name((value) => value !== void 0 && value !== null && value !== "" && (!Object.getOwnPropertyNames(value).includes("length") || value.length != 0) && (!Object.getOwnPropertyNames(value).includes("size") || value.size != 0), "isSerializableHeaderValue");
var _AI = "ApplicationId";
var _CCV = "ClientConfigurationVersion";
var _CI = "ClientId";
var _CPI = "ConfigurationProfileId";
var _CT = "ContentType";
var _CV = "ConfigurationVersion";
var _D = "Description";
var _EI = "ExtensionIdentifier";
var _EVN = "ExtensionVersionNumber";
var _KKA = "KmsKeyArn";
var _LVN = "LatestVersionNumber";
var _MR = "MaxResults";
var _N = "Name";
var _NT = "NextToken";
var _RI = "ResourceIdentifier";
var _T = "Type";
var _TK = "TagKeys";
var _VL = "VersionLabel";
var _VN = "VersionNumber";
var _ai = "application-id";
var _ccv = "client_configuration_version";
var _ci = "client_id";
var _cpi = "configuration-profile-id";
var _ct = "content-type";
var _cv = "configuration_version";
var _cv_ = "configuration-version";
var _d = "description";
var _ei = "extension_identifier";
var _evn = "extension_version_number";
var _k = "kmskeyarn";
var _lvn = "latest-version-number";
var _mr = "max_results";
var _n = "name";
var _nt = "next_token";
var _ri = "resource_identifier";
var _t = "type";
var _tK = "tagKeys";
var _v = "versionlabel";
var _ve = "version";
var _vl = "version_label";
var _vn = "version_number";
var _vn_ = "version-number";
var parseBody = /* @__PURE__ */ __name((streamBody, context) => collectBodyString(streamBody, context).then((encoded) => {
  if (encoded.length) {
    return JSON.parse(encoded);
  }
  return {};
}), "parseBody");
var parseErrorBody = /* @__PURE__ */ __name(async (errorBody, context) => {
  const value = await parseBody(errorBody, context);
  value.message = value.message ?? value.Message;
  return value;
}, "parseErrorBody");
var loadRestJsonErrorCode = /* @__PURE__ */ __name((output, data) => {
  const findKey = /* @__PURE__ */ __name((object, key) => Object.keys(object).find((k) => k.toLowerCase() === key.toLowerCase()), "findKey");
  const sanitizeErrorCode = /* @__PURE__ */ __name((rawValue) => {
    let cleanValue = rawValue;
    if (typeof cleanValue === "number") {
      cleanValue = cleanValue.toString();
    }
    if (cleanValue.indexOf(",") >= 0) {
      cleanValue = cleanValue.split(",")[0];
    }
    if (cleanValue.indexOf(":") >= 0) {
      cleanValue = cleanValue.split(":")[0];
    }
    if (cleanValue.indexOf("#") >= 0) {
      cleanValue = cleanValue.split("#")[1];
    }
    return cleanValue;
  }, "sanitizeErrorCode");
  const headerKey = findKey(output.headers, "x-amzn-errortype");
  if (headerKey !== void 0) {
    return sanitizeErrorCode(output.headers[headerKey]);
  }
  if (data.code !== void 0) {
    return sanitizeErrorCode(data.code);
  }
  if (data["__type"] !== void 0) {
    return sanitizeErrorCode(data["__type"]);
  }
}, "loadRestJsonErrorCode");

// src/commands/CreateApplicationCommand.ts
var _CreateApplicationCommand = class _CreateApplicationCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AmazonAppConfig", "CreateApplication", {}).n("AppConfigClient", "CreateApplicationCommand").f(void 0, void 0).ser(se_CreateApplicationCommand).de(de_CreateApplicationCommand).build() {
};
__name(_CreateApplicationCommand, "CreateApplicationCommand");
var CreateApplicationCommand = _CreateApplicationCommand;

// src/commands/CreateConfigurationProfileCommand.ts




var _CreateConfigurationProfileCommand = class _CreateConfigurationProfileCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AmazonAppConfig", "CreateConfigurationProfile", {}).n("AppConfigClient", "CreateConfigurationProfileCommand").f(CreateConfigurationProfileRequestFilterSensitiveLog, ConfigurationProfileFilterSensitiveLog).ser(se_CreateConfigurationProfileCommand).de(de_CreateConfigurationProfileCommand).build() {
};
__name(_CreateConfigurationProfileCommand, "CreateConfigurationProfileCommand");
var CreateConfigurationProfileCommand = _CreateConfigurationProfileCommand;

// src/commands/CreateDeploymentStrategyCommand.ts




var _CreateDeploymentStrategyCommand = class _CreateDeploymentStrategyCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AmazonAppConfig", "CreateDeploymentStrategy", {}).n("AppConfigClient", "CreateDeploymentStrategyCommand").f(void 0, void 0).ser(se_CreateDeploymentStrategyCommand).de(de_CreateDeploymentStrategyCommand).build() {
};
__name(_CreateDeploymentStrategyCommand, "CreateDeploymentStrategyCommand");
var CreateDeploymentStrategyCommand = _CreateDeploymentStrategyCommand;

// src/commands/CreateEnvironmentCommand.ts




var _CreateEnvironmentCommand = class _CreateEnvironmentCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AmazonAppConfig", "CreateEnvironment", {}).n("AppConfigClient", "CreateEnvironmentCommand").f(void 0, void 0).ser(se_CreateEnvironmentCommand).de(de_CreateEnvironmentCommand).build() {
};
__name(_CreateEnvironmentCommand, "CreateEnvironmentCommand");
var CreateEnvironmentCommand = _CreateEnvironmentCommand;

// src/commands/CreateExtensionAssociationCommand.ts




var _CreateExtensionAssociationCommand = class _CreateExtensionAssociationCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AmazonAppConfig", "CreateExtensionAssociation", {}).n("AppConfigClient", "CreateExtensionAssociationCommand").f(void 0, void 0).ser(se_CreateExtensionAssociationCommand).de(de_CreateExtensionAssociationCommand).build() {
};
__name(_CreateExtensionAssociationCommand, "CreateExtensionAssociationCommand");
var CreateExtensionAssociationCommand = _CreateExtensionAssociationCommand;

// src/commands/CreateExtensionCommand.ts




var _CreateExtensionCommand = class _CreateExtensionCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AmazonAppConfig", "CreateExtension", {}).n("AppConfigClient", "CreateExtensionCommand").f(void 0, void 0).ser(se_CreateExtensionCommand).de(de_CreateExtensionCommand).build() {
};
__name(_CreateExtensionCommand, "CreateExtensionCommand");
var CreateExtensionCommand = _CreateExtensionCommand;

// src/commands/CreateHostedConfigurationVersionCommand.ts




var _CreateHostedConfigurationVersionCommand = class _CreateHostedConfigurationVersionCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AmazonAppConfig", "CreateHostedConfigurationVersion", {}).n("AppConfigClient", "CreateHostedConfigurationVersionCommand").f(CreateHostedConfigurationVersionRequestFilterSensitiveLog, HostedConfigurationVersionFilterSensitiveLog).ser(se_CreateHostedConfigurationVersionCommand).de(de_CreateHostedConfigurationVersionCommand).build() {
};
__name(_CreateHostedConfigurationVersionCommand, "CreateHostedConfigurationVersionCommand");
var CreateHostedConfigurationVersionCommand = _CreateHostedConfigurationVersionCommand;

// src/commands/DeleteApplicationCommand.ts




var _DeleteApplicationCommand = class _DeleteApplicationCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AmazonAppConfig", "DeleteApplication", {}).n("AppConfigClient", "DeleteApplicationCommand").f(void 0, void 0).ser(se_DeleteApplicationCommand).de(de_DeleteApplicationCommand).build() {
};
__name(_DeleteApplicationCommand, "DeleteApplicationCommand");
var DeleteApplicationCommand = _DeleteApplicationCommand;

// src/commands/DeleteConfigurationProfileCommand.ts




var _DeleteConfigurationProfileCommand = class _DeleteConfigurationProfileCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AmazonAppConfig", "DeleteConfigurationProfile", {}).n("AppConfigClient", "DeleteConfigurationProfileCommand").f(void 0, void 0).ser(se_DeleteConfigurationProfileCommand).de(de_DeleteConfigurationProfileCommand).build() {
};
__name(_DeleteConfigurationProfileCommand, "DeleteConfigurationProfileCommand");
var DeleteConfigurationProfileCommand = _DeleteConfigurationProfileCommand;

// src/commands/DeleteDeploymentStrategyCommand.ts




var _DeleteDeploymentStrategyCommand = class _DeleteDeploymentStrategyCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AmazonAppConfig", "DeleteDeploymentStrategy", {}).n("AppConfigClient", "DeleteDeploymentStrategyCommand").f(void 0, void 0).ser(se_DeleteDeploymentStrategyCommand).de(de_DeleteDeploymentStrategyCommand).build() {
};
__name(_DeleteDeploymentStrategyCommand, "DeleteDeploymentStrategyCommand");
var DeleteDeploymentStrategyCommand = _DeleteDeploymentStrategyCommand;

// src/commands/DeleteEnvironmentCommand.ts




var _DeleteEnvironmentCommand = class _DeleteEnvironmentCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AmazonAppConfig", "DeleteEnvironment", {}).n("AppConfigClient", "DeleteEnvironmentCommand").f(void 0, void 0).ser(se_DeleteEnvironmentCommand).de(de_DeleteEnvironmentCommand).build() {
};
__name(_DeleteEnvironmentCommand, "DeleteEnvironmentCommand");
var DeleteEnvironmentCommand = _DeleteEnvironmentCommand;

// src/commands/DeleteExtensionAssociationCommand.ts




var _DeleteExtensionAssociationCommand = class _DeleteExtensionAssociationCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AmazonAppConfig", "DeleteExtensionAssociation", {}).n("AppConfigClient", "DeleteExtensionAssociationCommand").f(void 0, void 0).ser(se_DeleteExtensionAssociationCommand).de(de_DeleteExtensionAssociationCommand).build() {
};
__name(_DeleteExtensionAssociationCommand, "DeleteExtensionAssociationCommand");
var DeleteExtensionAssociationCommand = _DeleteExtensionAssociationCommand;

// src/commands/DeleteExtensionCommand.ts




var _DeleteExtensionCommand = class _DeleteExtensionCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AmazonAppConfig", "DeleteExtension", {}).n("AppConfigClient", "DeleteExtensionCommand").f(void 0, void 0).ser(se_DeleteExtensionCommand).de(de_DeleteExtensionCommand).build() {
};
__name(_DeleteExtensionCommand, "DeleteExtensionCommand");
var DeleteExtensionCommand = _DeleteExtensionCommand;

// src/commands/DeleteHostedConfigurationVersionCommand.ts




var _DeleteHostedConfigurationVersionCommand = class _DeleteHostedConfigurationVersionCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AmazonAppConfig", "DeleteHostedConfigurationVersion", {}).n("AppConfigClient", "DeleteHostedConfigurationVersionCommand").f(void 0, void 0).ser(se_DeleteHostedConfigurationVersionCommand).de(de_DeleteHostedConfigurationVersionCommand).build() {
};
__name(_DeleteHostedConfigurationVersionCommand, "DeleteHostedConfigurationVersionCommand");
var DeleteHostedConfigurationVersionCommand = _DeleteHostedConfigurationVersionCommand;

// src/commands/GetApplicationCommand.ts




var _GetApplicationCommand = class _GetApplicationCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AmazonAppConfig", "GetApplication", {}).n("AppConfigClient", "GetApplicationCommand").f(void 0, void 0).ser(se_GetApplicationCommand).de(de_GetApplicationCommand).build() {
};
__name(_GetApplicationCommand, "GetApplicationCommand");
var GetApplicationCommand = _GetApplicationCommand;

// src/commands/GetConfigurationCommand.ts




var _GetConfigurationCommand = class _GetConfigurationCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AmazonAppConfig", "GetConfiguration", {}).n("AppConfigClient", "GetConfigurationCommand").f(void 0, ConfigurationFilterSensitiveLog).ser(se_GetConfigurationCommand).de(de_GetConfigurationCommand).build() {
};
__name(_GetConfigurationCommand, "GetConfigurationCommand");
var GetConfigurationCommand = _GetConfigurationCommand;

// src/commands/GetConfigurationProfileCommand.ts




var _GetConfigurationProfileCommand = class _GetConfigurationProfileCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AmazonAppConfig", "GetConfigurationProfile", {}).n("AppConfigClient", "GetConfigurationProfileCommand").f(void 0, ConfigurationProfileFilterSensitiveLog).ser(se_GetConfigurationProfileCommand).de(de_GetConfigurationProfileCommand).build() {
};
__name(_GetConfigurationProfileCommand, "GetConfigurationProfileCommand");
var GetConfigurationProfileCommand = _GetConfigurationProfileCommand;

// src/commands/GetDeploymentCommand.ts




var _GetDeploymentCommand = class _GetDeploymentCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AmazonAppConfig", "GetDeployment", {}).n("AppConfigClient", "GetDeploymentCommand").f(void 0, void 0).ser(se_GetDeploymentCommand).de(de_GetDeploymentCommand).build() {
};
__name(_GetDeploymentCommand, "GetDeploymentCommand");
var GetDeploymentCommand = _GetDeploymentCommand;

// src/commands/GetDeploymentStrategyCommand.ts




var _GetDeploymentStrategyCommand = class _GetDeploymentStrategyCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AmazonAppConfig", "GetDeploymentStrategy", {}).n("AppConfigClient", "GetDeploymentStrategyCommand").f(void 0, void 0).ser(se_GetDeploymentStrategyCommand).de(de_GetDeploymentStrategyCommand).build() {
};
__name(_GetDeploymentStrategyCommand, "GetDeploymentStrategyCommand");
var GetDeploymentStrategyCommand = _GetDeploymentStrategyCommand;

// src/commands/GetEnvironmentCommand.ts




var _GetEnvironmentCommand = class _GetEnvironmentCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AmazonAppConfig", "GetEnvironment", {}).n("AppConfigClient", "GetEnvironmentCommand").f(void 0, void 0).ser(se_GetEnvironmentCommand).de(de_GetEnvironmentCommand).build() {
};
__name(_GetEnvironmentCommand, "GetEnvironmentCommand");
var GetEnvironmentCommand = _GetEnvironmentCommand;

// src/commands/GetExtensionAssociationCommand.ts




var _GetExtensionAssociationCommand = class _GetExtensionAssociationCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AmazonAppConfig", "GetExtensionAssociation", {}).n("AppConfigClient", "GetExtensionAssociationCommand").f(void 0, void 0).ser(se_GetExtensionAssociationCommand).de(de_GetExtensionAssociationCommand).build() {
};
__name(_GetExtensionAssociationCommand, "GetExtensionAssociationCommand");
var GetExtensionAssociationCommand = _GetExtensionAssociationCommand;

// src/commands/GetExtensionCommand.ts




var _GetExtensionCommand = class _GetExtensionCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AmazonAppConfig", "GetExtension", {}).n("AppConfigClient", "GetExtensionCommand").f(void 0, void 0).ser(se_GetExtensionCommand).de(de_GetExtensionCommand).build() {
};
__name(_GetExtensionCommand, "GetExtensionCommand");
var GetExtensionCommand = _GetExtensionCommand;

// src/commands/GetHostedConfigurationVersionCommand.ts




var _GetHostedConfigurationVersionCommand = class _GetHostedConfigurationVersionCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AmazonAppConfig", "GetHostedConfigurationVersion", {}).n("AppConfigClient", "GetHostedConfigurationVersionCommand").f(void 0, HostedConfigurationVersionFilterSensitiveLog).ser(se_GetHostedConfigurationVersionCommand).de(de_GetHostedConfigurationVersionCommand).build() {
};
__name(_GetHostedConfigurationVersionCommand, "GetHostedConfigurationVersionCommand");
var GetHostedConfigurationVersionCommand = _GetHostedConfigurationVersionCommand;

// src/commands/ListApplicationsCommand.ts




var _ListApplicationsCommand = class _ListApplicationsCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AmazonAppConfig", "ListApplications", {}).n("AppConfigClient", "ListApplicationsCommand").f(void 0, void 0).ser(se_ListApplicationsCommand).de(de_ListApplicationsCommand).build() {
};
__name(_ListApplicationsCommand, "ListApplicationsCommand");
var ListApplicationsCommand = _ListApplicationsCommand;

// src/commands/ListConfigurationProfilesCommand.ts




var _ListConfigurationProfilesCommand = class _ListConfigurationProfilesCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AmazonAppConfig", "ListConfigurationProfiles", {}).n("AppConfigClient", "ListConfigurationProfilesCommand").f(void 0, void 0).ser(se_ListConfigurationProfilesCommand).de(de_ListConfigurationProfilesCommand).build() {
};
__name(_ListConfigurationProfilesCommand, "ListConfigurationProfilesCommand");
var ListConfigurationProfilesCommand = _ListConfigurationProfilesCommand;

// src/commands/ListDeploymentsCommand.ts




var _ListDeploymentsCommand = class _ListDeploymentsCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AmazonAppConfig", "ListDeployments", {}).n("AppConfigClient", "ListDeploymentsCommand").f(void 0, void 0).ser(se_ListDeploymentsCommand).de(de_ListDeploymentsCommand).build() {
};
__name(_ListDeploymentsCommand, "ListDeploymentsCommand");
var ListDeploymentsCommand = _ListDeploymentsCommand;

// src/commands/ListDeploymentStrategiesCommand.ts




var _ListDeploymentStrategiesCommand = class _ListDeploymentStrategiesCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AmazonAppConfig", "ListDeploymentStrategies", {}).n("AppConfigClient", "ListDeploymentStrategiesCommand").f(void 0, void 0).ser(se_ListDeploymentStrategiesCommand).de(de_ListDeploymentStrategiesCommand).build() {
};
__name(_ListDeploymentStrategiesCommand, "ListDeploymentStrategiesCommand");
var ListDeploymentStrategiesCommand = _ListDeploymentStrategiesCommand;

// src/commands/ListEnvironmentsCommand.ts




var _ListEnvironmentsCommand = class _ListEnvironmentsCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AmazonAppConfig", "ListEnvironments", {}).n("AppConfigClient", "ListEnvironmentsCommand").f(void 0, void 0).ser(se_ListEnvironmentsCommand).de(de_ListEnvironmentsCommand).build() {
};
__name(_ListEnvironmentsCommand, "ListEnvironmentsCommand");
var ListEnvironmentsCommand = _ListEnvironmentsCommand;

// src/commands/ListExtensionAssociationsCommand.ts




var _ListExtensionAssociationsCommand = class _ListExtensionAssociationsCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AmazonAppConfig", "ListExtensionAssociations", {}).n("AppConfigClient", "ListExtensionAssociationsCommand").f(void 0, void 0).ser(se_ListExtensionAssociationsCommand).de(de_ListExtensionAssociationsCommand).build() {
};
__name(_ListExtensionAssociationsCommand, "ListExtensionAssociationsCommand");
var ListExtensionAssociationsCommand = _ListExtensionAssociationsCommand;

// src/commands/ListExtensionsCommand.ts




var _ListExtensionsCommand = class _ListExtensionsCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AmazonAppConfig", "ListExtensions", {}).n("AppConfigClient", "ListExtensionsCommand").f(void 0, void 0).ser(se_ListExtensionsCommand).de(de_ListExtensionsCommand).build() {
};
__name(_ListExtensionsCommand, "ListExtensionsCommand");
var ListExtensionsCommand = _ListExtensionsCommand;

// src/commands/ListHostedConfigurationVersionsCommand.ts




var _ListHostedConfigurationVersionsCommand = class _ListHostedConfigurationVersionsCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AmazonAppConfig", "ListHostedConfigurationVersions", {}).n("AppConfigClient", "ListHostedConfigurationVersionsCommand").f(void 0, void 0).ser(se_ListHostedConfigurationVersionsCommand).de(de_ListHostedConfigurationVersionsCommand).build() {
};
__name(_ListHostedConfigurationVersionsCommand, "ListHostedConfigurationVersionsCommand");
var ListHostedConfigurationVersionsCommand = _ListHostedConfigurationVersionsCommand;

// src/commands/ListTagsForResourceCommand.ts




var _ListTagsForResourceCommand = class _ListTagsForResourceCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AmazonAppConfig", "ListTagsForResource", {}).n("AppConfigClient", "ListTagsForResourceCommand").f(void 0, void 0).ser(se_ListTagsForResourceCommand).de(de_ListTagsForResourceCommand).build() {
};
__name(_ListTagsForResourceCommand, "ListTagsForResourceCommand");
var ListTagsForResourceCommand = _ListTagsForResourceCommand;

// src/commands/StartDeploymentCommand.ts




var _StartDeploymentCommand = class _StartDeploymentCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AmazonAppConfig", "StartDeployment", {}).n("AppConfigClient", "StartDeploymentCommand").f(void 0, void 0).ser(se_StartDeploymentCommand).de(de_StartDeploymentCommand).build() {
};
__name(_StartDeploymentCommand, "StartDeploymentCommand");
var StartDeploymentCommand = _StartDeploymentCommand;

// src/commands/StopDeploymentCommand.ts




var _StopDeploymentCommand = class _StopDeploymentCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AmazonAppConfig", "StopDeployment", {}).n("AppConfigClient", "StopDeploymentCommand").f(void 0, void 0).ser(se_StopDeploymentCommand).de(de_StopDeploymentCommand).build() {
};
__name(_StopDeploymentCommand, "StopDeploymentCommand");
var StopDeploymentCommand = _StopDeploymentCommand;

// src/commands/TagResourceCommand.ts




var _TagResourceCommand = class _TagResourceCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AmazonAppConfig", "TagResource", {}).n("AppConfigClient", "TagResourceCommand").f(void 0, void 0).ser(se_TagResourceCommand).de(de_TagResourceCommand).build() {
};
__name(_TagResourceCommand, "TagResourceCommand");
var TagResourceCommand = _TagResourceCommand;

// src/commands/UntagResourceCommand.ts




var _UntagResourceCommand = class _UntagResourceCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AmazonAppConfig", "UntagResource", {}).n("AppConfigClient", "UntagResourceCommand").f(void 0, void 0).ser(se_UntagResourceCommand).de(de_UntagResourceCommand).build() {
};
__name(_UntagResourceCommand, "UntagResourceCommand");
var UntagResourceCommand = _UntagResourceCommand;

// src/commands/UpdateApplicationCommand.ts




var _UpdateApplicationCommand = class _UpdateApplicationCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AmazonAppConfig", "UpdateApplication", {}).n("AppConfigClient", "UpdateApplicationCommand").f(void 0, void 0).ser(se_UpdateApplicationCommand).de(de_UpdateApplicationCommand).build() {
};
__name(_UpdateApplicationCommand, "UpdateApplicationCommand");
var UpdateApplicationCommand = _UpdateApplicationCommand;

// src/commands/UpdateConfigurationProfileCommand.ts




var _UpdateConfigurationProfileCommand = class _UpdateConfigurationProfileCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AmazonAppConfig", "UpdateConfigurationProfile", {}).n("AppConfigClient", "UpdateConfigurationProfileCommand").f(UpdateConfigurationProfileRequestFilterSensitiveLog, ConfigurationProfileFilterSensitiveLog).ser(se_UpdateConfigurationProfileCommand).de(de_UpdateConfigurationProfileCommand).build() {
};
__name(_UpdateConfigurationProfileCommand, "UpdateConfigurationProfileCommand");
var UpdateConfigurationProfileCommand = _UpdateConfigurationProfileCommand;

// src/commands/UpdateDeploymentStrategyCommand.ts




var _UpdateDeploymentStrategyCommand = class _UpdateDeploymentStrategyCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AmazonAppConfig", "UpdateDeploymentStrategy", {}).n("AppConfigClient", "UpdateDeploymentStrategyCommand").f(void 0, void 0).ser(se_UpdateDeploymentStrategyCommand).de(de_UpdateDeploymentStrategyCommand).build() {
};
__name(_UpdateDeploymentStrategyCommand, "UpdateDeploymentStrategyCommand");
var UpdateDeploymentStrategyCommand = _UpdateDeploymentStrategyCommand;

// src/commands/UpdateEnvironmentCommand.ts




var _UpdateEnvironmentCommand = class _UpdateEnvironmentCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AmazonAppConfig", "UpdateEnvironment", {}).n("AppConfigClient", "UpdateEnvironmentCommand").f(void 0, void 0).ser(se_UpdateEnvironmentCommand).de(de_UpdateEnvironmentCommand).build() {
};
__name(_UpdateEnvironmentCommand, "UpdateEnvironmentCommand");
var UpdateEnvironmentCommand = _UpdateEnvironmentCommand;

// src/commands/UpdateExtensionAssociationCommand.ts




var _UpdateExtensionAssociationCommand = class _UpdateExtensionAssociationCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AmazonAppConfig", "UpdateExtensionAssociation", {}).n("AppConfigClient", "UpdateExtensionAssociationCommand").f(void 0, void 0).ser(se_UpdateExtensionAssociationCommand).de(de_UpdateExtensionAssociationCommand).build() {
};
__name(_UpdateExtensionAssociationCommand, "UpdateExtensionAssociationCommand");
var UpdateExtensionAssociationCommand = _UpdateExtensionAssociationCommand;

// src/commands/UpdateExtensionCommand.ts




var _UpdateExtensionCommand = class _UpdateExtensionCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AmazonAppConfig", "UpdateExtension", {}).n("AppConfigClient", "UpdateExtensionCommand").f(void 0, void 0).ser(se_UpdateExtensionCommand).de(de_UpdateExtensionCommand).build() {
};
__name(_UpdateExtensionCommand, "UpdateExtensionCommand");
var UpdateExtensionCommand = _UpdateExtensionCommand;

// src/commands/ValidateConfigurationCommand.ts




var _ValidateConfigurationCommand = class _ValidateConfigurationCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AmazonAppConfig", "ValidateConfiguration", {}).n("AppConfigClient", "ValidateConfigurationCommand").f(void 0, void 0).ser(se_ValidateConfigurationCommand).de(de_ValidateConfigurationCommand).build() {
};
__name(_ValidateConfigurationCommand, "ValidateConfigurationCommand");
var ValidateConfigurationCommand = _ValidateConfigurationCommand;

// src/AppConfig.ts
var commands = {
  CreateApplicationCommand,
  CreateConfigurationProfileCommand,
  CreateDeploymentStrategyCommand,
  CreateEnvironmentCommand,
  CreateExtensionCommand,
  CreateExtensionAssociationCommand,
  CreateHostedConfigurationVersionCommand,
  DeleteApplicationCommand,
  DeleteConfigurationProfileCommand,
  DeleteDeploymentStrategyCommand,
  DeleteEnvironmentCommand,
  DeleteExtensionCommand,
  DeleteExtensionAssociationCommand,
  DeleteHostedConfigurationVersionCommand,
  GetApplicationCommand,
  GetConfigurationCommand,
  GetConfigurationProfileCommand,
  GetDeploymentCommand,
  GetDeploymentStrategyCommand,
  GetEnvironmentCommand,
  GetExtensionCommand,
  GetExtensionAssociationCommand,
  GetHostedConfigurationVersionCommand,
  ListApplicationsCommand,
  ListConfigurationProfilesCommand,
  ListDeploymentsCommand,
  ListDeploymentStrategiesCommand,
  ListEnvironmentsCommand,
  ListExtensionAssociationsCommand,
  ListExtensionsCommand,
  ListHostedConfigurationVersionsCommand,
  ListTagsForResourceCommand,
  StartDeploymentCommand,
  StopDeploymentCommand,
  TagResourceCommand,
  UntagResourceCommand,
  UpdateApplicationCommand,
  UpdateConfigurationProfileCommand,
  UpdateDeploymentStrategyCommand,
  UpdateEnvironmentCommand,
  UpdateExtensionCommand,
  UpdateExtensionAssociationCommand,
  ValidateConfigurationCommand
};
var _AppConfig = class _AppConfig extends AppConfigClient {
};
__name(_AppConfig, "AppConfig");
var AppConfig = _AppConfig;
(0, import_smithy_client.createAggregatedClient)(commands, AppConfig);

// src/pagination/ListApplicationsPaginator.ts

var paginateListApplications = (0, import_core.createPaginator)(AppConfigClient, ListApplicationsCommand, "NextToken", "NextToken", "MaxResults");

// src/pagination/ListConfigurationProfilesPaginator.ts

var paginateListConfigurationProfiles = (0, import_core.createPaginator)(AppConfigClient, ListConfigurationProfilesCommand, "NextToken", "NextToken", "MaxResults");

// src/pagination/ListDeploymentStrategiesPaginator.ts

var paginateListDeploymentStrategies = (0, import_core.createPaginator)(AppConfigClient, ListDeploymentStrategiesCommand, "NextToken", "NextToken", "MaxResults");

// src/pagination/ListDeploymentsPaginator.ts

var paginateListDeployments = (0, import_core.createPaginator)(AppConfigClient, ListDeploymentsCommand, "NextToken", "NextToken", "MaxResults");

// src/pagination/ListEnvironmentsPaginator.ts

var paginateListEnvironments = (0, import_core.createPaginator)(AppConfigClient, ListEnvironmentsCommand, "NextToken", "NextToken", "MaxResults");

// src/pagination/ListExtensionAssociationsPaginator.ts

var paginateListExtensionAssociations = (0, import_core.createPaginator)(AppConfigClient, ListExtensionAssociationsCommand, "NextToken", "NextToken", "MaxResults");

// src/pagination/ListExtensionsPaginator.ts

var paginateListExtensions = (0, import_core.createPaginator)(AppConfigClient, ListExtensionsCommand, "NextToken", "NextToken", "MaxResults");

// src/pagination/ListHostedConfigurationVersionsPaginator.ts

var paginateListHostedConfigurationVersions = (0, import_core.createPaginator)(AppConfigClient, ListHostedConfigurationVersionsCommand, "NextToken", "NextToken", "MaxResults");

// src/index.ts
var import_util_endpoints = require("@aws-sdk/util-endpoints");
// Annotate the CommonJS export names for ESM import in node:

0 && (module.exports = {
  AppConfigServiceException,
  __Client,
  AppConfigClient,
  AppConfig,
  $Command,
  CreateApplicationCommand,
  CreateConfigurationProfileCommand,
  CreateDeploymentStrategyCommand,
  CreateEnvironmentCommand,
  CreateExtensionAssociationCommand,
  CreateExtensionCommand,
  CreateHostedConfigurationVersionCommand,
  DeleteApplicationCommand,
  DeleteConfigurationProfileCommand,
  DeleteDeploymentStrategyCommand,
  DeleteEnvironmentCommand,
  DeleteExtensionAssociationCommand,
  DeleteExtensionCommand,
  DeleteHostedConfigurationVersionCommand,
  GetApplicationCommand,
  GetConfigurationCommand,
  GetConfigurationProfileCommand,
  GetDeploymentCommand,
  GetDeploymentStrategyCommand,
  GetEnvironmentCommand,
  GetExtensionAssociationCommand,
  GetExtensionCommand,
  GetHostedConfigurationVersionCommand,
  ListApplicationsCommand,
  ListConfigurationProfilesCommand,
  ListDeploymentStrategiesCommand,
  ListDeploymentsCommand,
  ListEnvironmentsCommand,
  ListExtensionAssociationsCommand,
  ListExtensionsCommand,
  ListHostedConfigurationVersionsCommand,
  ListTagsForResourceCommand,
  StartDeploymentCommand,
  StopDeploymentCommand,
  TagResourceCommand,
  UntagResourceCommand,
  UpdateApplicationCommand,
  UpdateConfigurationProfileCommand,
  UpdateDeploymentStrategyCommand,
  UpdateEnvironmentCommand,
  UpdateExtensionAssociationCommand,
  UpdateExtensionCommand,
  ValidateConfigurationCommand,
  paginateListApplications,
  paginateListConfigurationProfiles,
  paginateListDeploymentStrategies,
  paginateListDeployments,
  paginateListEnvironments,
  paginateListExtensionAssociations,
  paginateListExtensions,
  paginateListHostedConfigurationVersions,
  ActionPoint,
  BadRequestDetails,
  BadRequestReason,
  BadRequestException,
  InternalServerException,
  ServiceQuotaExceededException,
  ValidatorType,
  ResourceNotFoundException,
  GrowthType,
  ReplicateTo,
  EnvironmentState,
  ConflictException,
  BytesMeasure,
  PayloadTooLargeException,
  DeploymentEventType,
  TriggeredBy,
  DeploymentState,
  ValidatorFilterSensitiveLog,
  ConfigurationProfileFilterSensitiveLog,
  CreateConfigurationProfileRequestFilterSensitiveLog,
  CreateHostedConfigurationVersionRequestFilterSensitiveLog,
  HostedConfigurationVersionFilterSensitiveLog,
  ConfigurationFilterSensitiveLog,
  UpdateConfigurationProfileRequestFilterSensitiveLog
});
