// @azure/arm-commitmentplans/esm/operations/usageHistory.js
/*
 * Copyright (c) Microsoft Corporation. All rights reserved.
 * Licensed under the MIT License. See License.txt in the project root for
 * license information.
 *
 * Code generated by Microsoft (R) AutoRest Code Generator.
 * Changes may cause incorrect behavior and will be lost if the code is
 * regenerated.
 */
import * as msRest from "@azure/ms-rest-js";
import * as Mappers from "../models/usageHistoryMappers";
import * as Parameters from "../models/parameters";
/** Class representing a UsageHistory. */
var UsageHistory = /** @class */ (function () {
    /**
     * Create a UsageHistory.
     * @param {AzureMLCommitmentPlansManagementClientContext} client Reference to the service client.
     */
    function UsageHistory(client) {
        this.client = client;
    }
    UsageHistory.prototype.list = function (resourceGroupName, commitmentPlanName, options, callback) {
        return this.client.sendOperationRequest({
            resourceGroupName: resourceGroupName,
            commitmentPlanName: commitmentPlanName,
            options: options
        }, listOperationSpec, callback);
    };
    UsageHistory.prototype.listNext = function (nextPageLink, options, callback) {
        return this.client.sendOperationRequest({
            nextPageLink: nextPageLink,
            options: options
        }, listNextOperationSpec, callback);
    };
    return UsageHistory;
}());
export { UsageHistory };
// Operation Specifications
var serializer = new msRest.Serializer(Mappers);
var listOperationSpec = {
    httpMethod: "GET",
    path: "subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.MachineLearning/commitmentPlans/{commitmentPlanName}/usageHistory",
    urlParameters: [
        Parameters.subscriptionId,
        Parameters.resourceGroupName,
        Parameters.commitmentPlanName
    ],
    queryParameters: [
        Parameters.skipToken,
        Parameters.apiVersion
    ],
    headerParameters: [
        Parameters.acceptLanguage
    ],
    responses: {
        200: {
            bodyMapper: Mappers.PlanUsageHistoryListResult
        },
        default: {
            bodyMapper: Mappers.CloudError
        }
    },
    serializer: serializer
};
var listNextOperationSpec = {
    httpMethod: "GET",
    baseUrl: "https://management.azure.com",
    path: "{nextLink}",
    urlParameters: [
        Parameters.nextPageLink
    ],
    headerParameters: [
        Parameters.acceptLanguage
    ],
    responses: {
        200: {
            bodyMapper: Mappers.PlanUsageHistoryListResult
        },
        default: {
            bodyMapper: Mappers.CloudError
        }
    },
    serializer: serializer
};
