// @azure/arm-serialconsole/dist-esm/src/models/mappers.js
/*
 * Copyright (c) Microsoft Corporation.
 * Licensed under the MIT License.
 *
 * Code generated by Microsoft (R) AutoRest Code Generator.
 * Changes may cause incorrect behavior and will be lost if the code is regenerated.
 */
export const SerialConsoleOperations = {
    type: {
        name: "Composite",
        className: "SerialConsoleOperations",
        modelProperties: {
            value: {
                serializedName: "value",
                type: {
                    name: "Sequence",
                    element: {
                        type: {
                            name: "Composite",
                            className: "SerialConsoleOperationsValueItem"
                        }
                    }
                }
            }
        }
    }
};
export const SerialConsoleOperationsValueItem = {
    type: {
        name: "Composite",
        className: "SerialConsoleOperationsValueItem",
        modelProperties: {
            name: {
                serializedName: "name",
                type: {
                    name: "String"
                }
            },
            isDataAction: {
                serializedName: "isDataAction",
                type: {
                    name: "String"
                }
            },
            display: {
                serializedName: "display",
                type: {
                    name: "Composite",
                    className: "SerialConsoleOperationsValueItemDisplay"
                }
            }
        }
    }
};
export const SerialConsoleOperationsValueItemDisplay = {
    type: {
        name: "Composite",
        className: "SerialConsoleOperationsValueItemDisplay",
        modelProperties: {
            provider: {
                serializedName: "provider",
                type: {
                    name: "String"
                }
            },
            resource: {
                serializedName: "resource",
                type: {
                    name: "String"
                }
            },
            operation: {
                serializedName: "operation",
                type: {
                    name: "String"
                }
            },
            description: {
                serializedName: "description",
                type: {
                    name: "String"
                }
            }
        }
    }
};
export const SerialConsoleStatus = {
    type: {
        name: "Composite",
        className: "SerialConsoleStatus",
        modelProperties: {
            disabled: {
                serializedName: "disabled",
                type: {
                    name: "Boolean"
                }
            }
        }
    }
};
export const GetSerialConsoleSubscriptionNotFound = {
    type: {
        name: "Composite",
        className: "GetSerialConsoleSubscriptionNotFound",
        modelProperties: {
            code: {
                serializedName: "code",
                type: {
                    name: "String"
                }
            },
            message: {
                serializedName: "message",
                type: {
                    name: "String"
                }
            }
        }
    }
};
export const DisableSerialConsoleResult = {
    type: {
        name: "Composite",
        className: "DisableSerialConsoleResult",
        modelProperties: {
            disabled: {
                serializedName: "disabled",
                type: {
                    name: "Boolean"
                }
            }
        }
    }
};
export const EnableSerialConsoleResult = {
    type: {
        name: "Composite",
        className: "EnableSerialConsoleResult",
        modelProperties: {
            disabled: {
                serializedName: "disabled",
                type: {
                    name: "Boolean"
                }
            }
        }
    }
};
export const SerialPortListResult = {
    type: {
        name: "Composite",
        className: "SerialPortListResult",
        modelProperties: {
            value: {
                serializedName: "value",
                type: {
                    name: "Sequence",
                    element: {
                        type: {
                            name: "Composite",
                            className: "SerialPort"
                        }
                    }
                }
            }
        }
    }
};
export const Resource = {
    type: {
        name: "Composite",
        className: "Resource",
        modelProperties: {
            id: {
                serializedName: "id",
                readOnly: true,
                type: {
                    name: "String"
                }
            },
            name: {
                serializedName: "name",
                readOnly: true,
                type: {
                    name: "String"
                }
            },
            type: {
                serializedName: "type",
                readOnly: true,
                type: {
                    name: "String"
                }
            }
        }
    }
};
export const CloudError = {
    type: {
        name: "Composite",
        className: "CloudError",
        modelProperties: {
            error: {
                serializedName: "error",
                type: {
                    name: "Composite",
                    className: "CloudErrorBody"
                }
            }
        }
    }
};
export const CloudErrorBody = {
    type: {
        name: "Composite",
        className: "CloudErrorBody",
        modelProperties: {
            code: {
                serializedName: "code",
                type: {
                    name: "String"
                }
            },
            message: {
                serializedName: "message",
                type: {
                    name: "String"
                }
            },
            target: {
                serializedName: "target",
                type: {
                    name: "String"
                }
            },
            details: {
                serializedName: "details",
                type: {
                    name: "Sequence",
                    element: {
                        type: {
                            name: "Composite",
                            className: "CloudErrorBody"
                        }
                    }
                }
            }
        }
    }
};
export const SerialPortConnectResult = {
    type: {
        name: "Composite",
        className: "SerialPortConnectResult",
        modelProperties: {
            connectionString: {
                serializedName: "connectionString",
                type: {
                    name: "String"
                }
            }
        }
    }
};
export const ProxyResource = {
    type: {
        name: "Composite",
        className: "ProxyResource",
        modelProperties: Object.assign({}, Resource.type.modelProperties)
    }
};
export const SerialPort = {
    type: {
        name: "Composite",
        className: "SerialPort",
        modelProperties: Object.assign(Object.assign({}, ProxyResource.type.modelProperties), { state: {
                serializedName: "properties.state",
                type: {
                    name: "Enum",
                    allowedValues: ["enabled", "disabled"]
                }
            } })
    }
};
