// @azure/arm-visualstudio/esm/models/parameters.js
/*
 * Copyright (c) Microsoft Corporation.
 * Licensed under the MIT License.
 *
 * Code generated by Microsoft (R) AutoRest Code Generator.
 * Changes may cause incorrect behavior and will be lost if the code is
 * regenerated.
 */
export var acceptLanguage = {
    parameterPath: "acceptLanguage",
    mapper: {
        serializedName: "accept-language",
        defaultValue: 'en-US',
        type: {
            name: "String"
        }
    }
};
export var accountResourceName = {
    parameterPath: "accountResourceName",
    mapper: {
        required: true,
        serializedName: "accountResourceName",
        type: {
            name: "String"
        }
    }
};
export var apiVersion = {
    parameterPath: "apiVersion",
    mapper: {
        required: true,
        serializedName: "api-version",
        type: {
            name: "String"
        }
    }
};
export var extensionResourceName = {
    parameterPath: "extensionResourceName",
    mapper: {
        required: true,
        serializedName: "extensionResourceName",
        type: {
            name: "String"
        }
    }
};
export var jobId = {
    parameterPath: [
        "options",
        "jobId"
    ],
    mapper: {
        serializedName: "jobId",
        type: {
            name: "Uuid"
        }
    }
};
export var operation = {
    parameterPath: "operation",
    mapper: {
        required: true,
        serializedName: "operation",
        type: {
            name: "String"
        }
    }
};
export var resourceGroupName = {
    parameterPath: "resourceGroupName",
    mapper: {
        required: true,
        serializedName: "resourceGroupName",
        type: {
            name: "String"
        }
    }
};
export var resourceName = {
    parameterPath: "resourceName",
    mapper: {
        required: true,
        serializedName: "resourceName",
        type: {
            name: "String"
        }
    }
};
export var rootResourceName = {
    parameterPath: "rootResourceName",
    mapper: {
        required: true,
        serializedName: "rootResourceName",
        type: {
            name: "String"
        }
    }
};
export var subContainerName = {
    parameterPath: "subContainerName",
    mapper: {
        required: true,
        serializedName: "subContainerName",
        type: {
            name: "String"
        }
    }
};
export var subscriptionId = {
    parameterPath: "subscriptionId",
    mapper: {
        required: true,
        serializedName: "subscriptionId",
        type: {
            name: "String"
        }
    }
};
export var validating = {
    parameterPath: [
        "options",
        "validating"
    ],
    mapper: {
        serializedName: "validating",
        type: {
            name: "String"
        }
    }
};
