// @azure/arm-automanage/dist-esm/src/operations/bestPracticesVersions.js
/*
 * Copyright (c) Microsoft Corporation.
 * Licensed under the MIT License.
 *
 * Code generated by Microsoft (R) AutoRest Code Generator.
 * Changes may cause incorrect behavior and will be lost if the code is regenerated.
 */
import { __asyncDelegator, __asyncGenerator, __asyncValues, __await } from "tslib";
import * as coreClient from "@azure/core-client";
import * as Mappers from "../models/mappers";
import * as Parameters from "../models/parameters";
/// <reference lib="esnext.asynciterable" />
/** Class containing BestPracticesVersions operations. */
export class BestPracticesVersionsImpl {
    /**
     * Initialize a new instance of the class BestPracticesVersions class.
     * @param client Reference to the service client
     */
    constructor(client) {
        this.client = client;
    }
    /**
     * Retrieve a list of Automanage best practices versions
     * @param bestPracticeName The Automanage best practice name.
     * @param options The options parameters.
     */
    listByTenant(bestPracticeName, options) {
        const iter = this.listByTenantPagingAll(bestPracticeName, options);
        return {
            next() {
                return iter.next();
            },
            [Symbol.asyncIterator]() {
                return this;
            },
            byPage: () => {
                return this.listByTenantPagingPage(bestPracticeName, options);
            }
        };
    }
    listByTenantPagingPage(bestPracticeName, options) {
        return __asyncGenerator(this, arguments, function* listByTenantPagingPage_1() {
            let result = yield __await(this._listByTenant(bestPracticeName, options));
            yield yield __await(result.value || []);
        });
    }
    listByTenantPagingAll(bestPracticeName, options) {
        return __asyncGenerator(this, arguments, function* listByTenantPagingAll_1() {
            var e_1, _a;
            try {
                for (var _b = __asyncValues(this.listByTenantPagingPage(bestPracticeName, options)), _c; _c = yield __await(_b.next()), !_c.done;) {
                    const page = _c.value;
                    yield __await(yield* __asyncDelegator(__asyncValues(page)));
                }
            }
            catch (e_1_1) { e_1 = { error: e_1_1 }; }
            finally {
                try {
                    if (_c && !_c.done && (_a = _b.return)) yield __await(_a.call(_b));
                }
                finally { if (e_1) throw e_1.error; }
            }
        });
    }
    /**
     * Get information about a Automanage best practice version
     * @param bestPracticeName The Automanage best practice name.
     * @param versionName The Automanage best practice version name.
     * @param options The options parameters.
     */
    get(bestPracticeName, versionName, options) {
        return this.client.sendOperationRequest({ bestPracticeName, versionName, options }, getOperationSpec);
    }
    /**
     * Retrieve a list of Automanage best practices versions
     * @param bestPracticeName The Automanage best practice name.
     * @param options The options parameters.
     */
    _listByTenant(bestPracticeName, options) {
        return this.client.sendOperationRequest({ bestPracticeName, options }, listByTenantOperationSpec);
    }
}
// Operation Specifications
const serializer = coreClient.createSerializer(Mappers, /* isXml */ false);
const getOperationSpec = {
    path: "/providers/Microsoft.Automanage/bestPractices/{bestPracticeName}/versions/{versionName}",
    httpMethod: "GET",
    responses: {
        200: {
            bodyMapper: Mappers.BestPractice
        },
        default: {
            bodyMapper: Mappers.ErrorResponse
        }
    },
    queryParameters: [Parameters.apiVersion],
    urlParameters: [
        Parameters.$host,
        Parameters.bestPracticeName,
        Parameters.versionName
    ],
    headerParameters: [Parameters.accept],
    serializer
};
const listByTenantOperationSpec = {
    path: "/providers/Microsoft.Automanage/bestPractices/{bestPracticeName}/versions",
    httpMethod: "GET",
    responses: {
        200: {
            bodyMapper: Mappers.BestPracticeList
        },
        default: {
            bodyMapper: Mappers.ErrorResponse
        }
    },
    queryParameters: [Parameters.apiVersion],
    urlParameters: [Parameters.$host, Parameters.bestPracticeName],
    headerParameters: [Parameters.accept],
    serializer
};
