// @azure/arm-automanage/dist-esm/src/operations/hCIReports.js
/*
 * Copyright (c) Microsoft Corporation.
 * Licensed under the MIT License.
 *
 * Code generated by Microsoft (R) AutoRest Code Generator.
 * Changes may cause incorrect behavior and will be lost if the code is regenerated.
 */
import { __asyncDelegator, __asyncGenerator, __asyncValues, __await } from "tslib";
import * as coreClient from "@azure/core-client";
import * as Mappers from "../models/mappers";
import * as Parameters from "../models/parameters";
/// <reference lib="esnext.asynciterable" />
/** Class containing HCIReports operations. */
export class HCIReportsImpl {
    /**
     * Initialize a new instance of the class HCIReports class.
     * @param client Reference to the service client
     */
    constructor(client) {
        this.client = client;
    }
    /**
     * Retrieve a list of reports within a given configuration profile assignment
     * @param resourceGroupName The name of the resource group. The name is case insensitive.
     * @param clusterName The name of the Arc machine.
     * @param configurationProfileAssignmentName The configuration profile assignment name.
     * @param options The options parameters.
     */
    listByConfigurationProfileAssignments(resourceGroupName, clusterName, configurationProfileAssignmentName, options) {
        const iter = this.listByConfigurationProfileAssignmentsPagingAll(resourceGroupName, clusterName, configurationProfileAssignmentName, options);
        return {
            next() {
                return iter.next();
            },
            [Symbol.asyncIterator]() {
                return this;
            },
            byPage: () => {
                return this.listByConfigurationProfileAssignmentsPagingPage(resourceGroupName, clusterName, configurationProfileAssignmentName, options);
            }
        };
    }
    listByConfigurationProfileAssignmentsPagingPage(resourceGroupName, clusterName, configurationProfileAssignmentName, options) {
        return __asyncGenerator(this, arguments, function* listByConfigurationProfileAssignmentsPagingPage_1() {
            let result = yield __await(this._listByConfigurationProfileAssignments(resourceGroupName, clusterName, configurationProfileAssignmentName, options));
            yield yield __await(result.value || []);
        });
    }
    listByConfigurationProfileAssignmentsPagingAll(resourceGroupName, clusterName, configurationProfileAssignmentName, options) {
        return __asyncGenerator(this, arguments, function* listByConfigurationProfileAssignmentsPagingAll_1() {
            var e_1, _a;
            try {
                for (var _b = __asyncValues(this.listByConfigurationProfileAssignmentsPagingPage(resourceGroupName, clusterName, configurationProfileAssignmentName, options)), _c; _c = yield __await(_b.next()), !_c.done;) {
                    const page = _c.value;
                    yield __await(yield* __asyncDelegator(__asyncValues(page)));
                }
            }
            catch (e_1_1) { e_1 = { error: e_1_1 }; }
            finally {
                try {
                    if (_c && !_c.done && (_a = _b.return)) yield __await(_a.call(_b));
                }
                finally { if (e_1) throw e_1.error; }
            }
        });
    }
    /**
     * Get information about a report associated with a configuration profile assignment run
     * @param resourceGroupName The name of the resource group. The name is case insensitive.
     * @param clusterName The name of the Arc machine.
     * @param configurationProfileAssignmentName The configuration profile assignment name.
     * @param reportName The report name.
     * @param options The options parameters.
     */
    get(resourceGroupName, clusterName, configurationProfileAssignmentName, reportName, options) {
        return this.client.sendOperationRequest({
            resourceGroupName,
            clusterName,
            configurationProfileAssignmentName,
            reportName,
            options
        }, getOperationSpec);
    }
    /**
     * Retrieve a list of reports within a given configuration profile assignment
     * @param resourceGroupName The name of the resource group. The name is case insensitive.
     * @param clusterName The name of the Arc machine.
     * @param configurationProfileAssignmentName The configuration profile assignment name.
     * @param options The options parameters.
     */
    _listByConfigurationProfileAssignments(resourceGroupName, clusterName, configurationProfileAssignmentName, options) {
        return this.client.sendOperationRequest({
            resourceGroupName,
            clusterName,
            configurationProfileAssignmentName,
            options
        }, listByConfigurationProfileAssignmentsOperationSpec);
    }
}
// Operation Specifications
const serializer = coreClient.createSerializer(Mappers, /* isXml */ false);
const getOperationSpec = {
    path: "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.AzureStackHci/clusters/{clusterName}/providers/Microsoft.Automanage/configurationProfileAssignments/{configurationProfileAssignmentName}/reports/{reportName}",
    httpMethod: "GET",
    responses: {
        200: {
            bodyMapper: Mappers.Report
        },
        default: {
            bodyMapper: Mappers.ErrorResponse
        }
    },
    queryParameters: [Parameters.apiVersion],
    urlParameters: [
        Parameters.$host,
        Parameters.subscriptionId,
        Parameters.resourceGroupName,
        Parameters.configurationProfileAssignmentName,
        Parameters.clusterName,
        Parameters.reportName
    ],
    headerParameters: [Parameters.accept],
    serializer
};
const listByConfigurationProfileAssignmentsOperationSpec = {
    path: "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.AzureStackHci/clusters/{clusterName}/providers/Microsoft.Automanage/configurationProfileAssignments/{configurationProfileAssignmentName}/reports",
    httpMethod: "GET",
    responses: {
        200: {
            bodyMapper: Mappers.ReportList
        },
        default: {
            bodyMapper: Mappers.ErrorResponse
        }
    },
    queryParameters: [Parameters.apiVersion],
    urlParameters: [
        Parameters.$host,
        Parameters.subscriptionId,
        Parameters.resourceGroupName,
        Parameters.configurationProfileAssignmentName,
        Parameters.clusterName
    ],
    headerParameters: [Parameters.accept],
    serializer
};
