// @azure/arm-eventhub/dist-esm/src/models/index.js
/*
 * Copyright (c) Microsoft Corporation.
 * Licensed under the MIT License.
 *
 * Code generated by Microsoft (R) AutoRest Code Generator.
 * Changes may cause incorrect behavior and will be lost if the code is regenerated.
 */
/** Known values of {@link ClusterSkuName} that the service accepts. */
export var KnownClusterSkuName;
(function (KnownClusterSkuName) {
    /** Dedicated */
    KnownClusterSkuName["Dedicated"] = "Dedicated";
})(KnownClusterSkuName || (KnownClusterSkuName = {}));
/** Known values of {@link CreatedByType} that the service accepts. */
export var KnownCreatedByType;
(function (KnownCreatedByType) {
    /** User */
    KnownCreatedByType["User"] = "User";
    /** Application */
    KnownCreatedByType["Application"] = "Application";
    /** ManagedIdentity */
    KnownCreatedByType["ManagedIdentity"] = "ManagedIdentity";
    /** Key */
    KnownCreatedByType["Key"] = "Key";
})(KnownCreatedByType || (KnownCreatedByType = {}));
/** Known values of {@link SkuName} that the service accepts. */
export var KnownSkuName;
(function (KnownSkuName) {
    /** Basic */
    KnownSkuName["Basic"] = "Basic";
    /** Standard */
    KnownSkuName["Standard"] = "Standard";
    /** Premium */
    KnownSkuName["Premium"] = "Premium";
})(KnownSkuName || (KnownSkuName = {}));
/** Known values of {@link SkuTier} that the service accepts. */
export var KnownSkuTier;
(function (KnownSkuTier) {
    /** Basic */
    KnownSkuTier["Basic"] = "Basic";
    /** Standard */
    KnownSkuTier["Standard"] = "Standard";
    /** Premium */
    KnownSkuTier["Premium"] = "Premium";
})(KnownSkuTier || (KnownSkuTier = {}));
/** Known values of {@link PrivateLinkConnectionStatus} that the service accepts. */
export var KnownPrivateLinkConnectionStatus;
(function (KnownPrivateLinkConnectionStatus) {
    /** Pending */
    KnownPrivateLinkConnectionStatus["Pending"] = "Pending";
    /** Approved */
    KnownPrivateLinkConnectionStatus["Approved"] = "Approved";
    /** Rejected */
    KnownPrivateLinkConnectionStatus["Rejected"] = "Rejected";
    /** Disconnected */
    KnownPrivateLinkConnectionStatus["Disconnected"] = "Disconnected";
})(KnownPrivateLinkConnectionStatus || (KnownPrivateLinkConnectionStatus = {}));
/** Known values of {@link EndPointProvisioningState} that the service accepts. */
export var KnownEndPointProvisioningState;
(function (KnownEndPointProvisioningState) {
    /** Creating */
    KnownEndPointProvisioningState["Creating"] = "Creating";
    /** Updating */
    KnownEndPointProvisioningState["Updating"] = "Updating";
    /** Deleting */
    KnownEndPointProvisioningState["Deleting"] = "Deleting";
    /** Succeeded */
    KnownEndPointProvisioningState["Succeeded"] = "Succeeded";
    /** Canceled */
    KnownEndPointProvisioningState["Canceled"] = "Canceled";
    /** Failed */
    KnownEndPointProvisioningState["Failed"] = "Failed";
})(KnownEndPointProvisioningState || (KnownEndPointProvisioningState = {}));
/** Known values of {@link DefaultAction} that the service accepts. */
export var KnownDefaultAction;
(function (KnownDefaultAction) {
    /** Allow */
    KnownDefaultAction["Allow"] = "Allow";
    /** Deny */
    KnownDefaultAction["Deny"] = "Deny";
})(KnownDefaultAction || (KnownDefaultAction = {}));
/** Known values of {@link NetworkRuleIPAction} that the service accepts. */
export var KnownNetworkRuleIPAction;
(function (KnownNetworkRuleIPAction) {
    /** Allow */
    KnownNetworkRuleIPAction["Allow"] = "Allow";
})(KnownNetworkRuleIPAction || (KnownNetworkRuleIPAction = {}));
/** Known values of {@link PublicNetworkAccessFlag} that the service accepts. */
export var KnownPublicNetworkAccessFlag;
(function (KnownPublicNetworkAccessFlag) {
    /** Enabled */
    KnownPublicNetworkAccessFlag["Enabled"] = "Enabled";
    /** Disabled */
    KnownPublicNetworkAccessFlag["Disabled"] = "Disabled";
})(KnownPublicNetworkAccessFlag || (KnownPublicNetworkAccessFlag = {}));
/** Known values of {@link AccessRights} that the service accepts. */
export var KnownAccessRights;
(function (KnownAccessRights) {
    /** Manage */
    KnownAccessRights["Manage"] = "Manage";
    /** Send */
    KnownAccessRights["Send"] = "Send";
    /** Listen */
    KnownAccessRights["Listen"] = "Listen";
})(KnownAccessRights || (KnownAccessRights = {}));
/** Known values of {@link KeyType} that the service accepts. */
export var KnownKeyType;
(function (KnownKeyType) {
    /** PrimaryKey */
    KnownKeyType["PrimaryKey"] = "PrimaryKey";
    /** SecondaryKey */
    KnownKeyType["SecondaryKey"] = "SecondaryKey";
})(KnownKeyType || (KnownKeyType = {}));
/** Known values of {@link SchemaCompatibility} that the service accepts. */
export var KnownSchemaCompatibility;
(function (KnownSchemaCompatibility) {
    /** None */
    KnownSchemaCompatibility["None"] = "None";
    /** Backward */
    KnownSchemaCompatibility["Backward"] = "Backward";
    /** Forward */
    KnownSchemaCompatibility["Forward"] = "Forward";
})(KnownSchemaCompatibility || (KnownSchemaCompatibility = {}));
/** Known values of {@link SchemaType} that the service accepts. */
export var KnownSchemaType;
(function (KnownSchemaType) {
    /** Unknown */
    KnownSchemaType["Unknown"] = "Unknown";
    /** Avro */
    KnownSchemaType["Avro"] = "Avro";
})(KnownSchemaType || (KnownSchemaType = {}));
