// @azure/arm-app/dist-esm/src/operations/certificates.js
/*
 * Copyright (c) Microsoft Corporation.
 * Licensed under the MIT License.
 *
 * Code generated by Microsoft (R) AutoRest Code Generator.
 * Changes may cause incorrect behavior and will be lost if the code is regenerated.
 */
import { __asyncDelegator, __asyncGenerator, __asyncValues, __await } from "tslib";
import * as coreClient from "@azure/core-client";
import * as Mappers from "../models/mappers";
import * as Parameters from "../models/parameters";
/// <reference lib="esnext.asynciterable" />
/** Class containing Certificates operations. */
export class CertificatesImpl {
    /**
     * Initialize a new instance of the class Certificates class.
     * @param client Reference to the service client
     */
    constructor(client) {
        this.client = client;
    }
    /**
     * Get the Certificates in a given managed environment.
     * @param resourceGroupName The name of the resource group. The name is case insensitive.
     * @param managedEnvironmentName Name of the Managed Environment.
     * @param options The options parameters.
     */
    list(resourceGroupName, managedEnvironmentName, options) {
        const iter = this.listPagingAll(resourceGroupName, managedEnvironmentName, options);
        return {
            next() {
                return iter.next();
            },
            [Symbol.asyncIterator]() {
                return this;
            },
            byPage: () => {
                return this.listPagingPage(resourceGroupName, managedEnvironmentName, options);
            }
        };
    }
    listPagingPage(resourceGroupName, managedEnvironmentName, options) {
        return __asyncGenerator(this, arguments, function* listPagingPage_1() {
            let result = yield __await(this._list(resourceGroupName, managedEnvironmentName, options));
            yield yield __await(result.value || []);
            let continuationToken = result.nextLink;
            while (continuationToken) {
                result = yield __await(this._listNext(resourceGroupName, managedEnvironmentName, continuationToken, options));
                continuationToken = result.nextLink;
                yield yield __await(result.value || []);
            }
        });
    }
    listPagingAll(resourceGroupName, managedEnvironmentName, options) {
        return __asyncGenerator(this, arguments, function* listPagingAll_1() {
            var e_1, _a;
            try {
                for (var _b = __asyncValues(this.listPagingPage(resourceGroupName, managedEnvironmentName, options)), _c; _c = yield __await(_b.next()), !_c.done;) {
                    const page = _c.value;
                    yield __await(yield* __asyncDelegator(__asyncValues(page)));
                }
            }
            catch (e_1_1) { e_1 = { error: e_1_1 }; }
            finally {
                try {
                    if (_c && !_c.done && (_a = _b.return)) yield __await(_a.call(_b));
                }
                finally { if (e_1) throw e_1.error; }
            }
        });
    }
    /**
     * Get the Certificates in a given managed environment.
     * @param resourceGroupName The name of the resource group. The name is case insensitive.
     * @param managedEnvironmentName Name of the Managed Environment.
     * @param options The options parameters.
     */
    _list(resourceGroupName, managedEnvironmentName, options) {
        return this.client.sendOperationRequest({ resourceGroupName, managedEnvironmentName, options }, listOperationSpec);
    }
    /**
     * Get the specified Certificate.
     * @param resourceGroupName The name of the resource group. The name is case insensitive.
     * @param managedEnvironmentName Name of the Managed Environment.
     * @param name Name of the Certificate.
     * @param options The options parameters.
     */
    get(resourceGroupName, managedEnvironmentName, name, options) {
        return this.client.sendOperationRequest({ resourceGroupName, managedEnvironmentName, name, options }, getOperationSpec);
    }
    /**
     * Create or Update a Certificate.
     * @param resourceGroupName The name of the resource group. The name is case insensitive.
     * @param managedEnvironmentName Name of the Managed Environment.
     * @param name Name of the Certificate.
     * @param options The options parameters.
     */
    createOrUpdate(resourceGroupName, managedEnvironmentName, name, options) {
        return this.client.sendOperationRequest({ resourceGroupName, managedEnvironmentName, name, options }, createOrUpdateOperationSpec);
    }
    /**
     * Deletes the specified Certificate.
     * @param resourceGroupName The name of the resource group. The name is case insensitive.
     * @param managedEnvironmentName Name of the Managed Environment.
     * @param name Name of the Certificate.
     * @param options The options parameters.
     */
    delete(resourceGroupName, managedEnvironmentName, name, options) {
        return this.client.sendOperationRequest({ resourceGroupName, managedEnvironmentName, name, options }, deleteOperationSpec);
    }
    /**
     * Patches a certificate. Currently only patching of tags is supported
     * @param resourceGroupName The name of the resource group. The name is case insensitive.
     * @param managedEnvironmentName Name of the Managed Environment.
     * @param name Name of the Certificate.
     * @param certificateEnvelope Properties of a certificate that need to be updated
     * @param options The options parameters.
     */
    update(resourceGroupName, managedEnvironmentName, name, certificateEnvelope, options) {
        return this.client.sendOperationRequest({
            resourceGroupName,
            managedEnvironmentName,
            name,
            certificateEnvelope,
            options
        }, updateOperationSpec);
    }
    /**
     * ListNext
     * @param resourceGroupName The name of the resource group. The name is case insensitive.
     * @param managedEnvironmentName Name of the Managed Environment.
     * @param nextLink The nextLink from the previous successful call to the List method.
     * @param options The options parameters.
     */
    _listNext(resourceGroupName, managedEnvironmentName, nextLink, options) {
        return this.client.sendOperationRequest({ resourceGroupName, managedEnvironmentName, nextLink, options }, listNextOperationSpec);
    }
}
// Operation Specifications
const serializer = coreClient.createSerializer(Mappers, /* isXml */ false);
const listOperationSpec = {
    path: "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.App/managedEnvironments/{managedEnvironmentName}/certificates",
    httpMethod: "GET",
    responses: {
        200: {
            bodyMapper: Mappers.CertificateCollection
        },
        default: {
            bodyMapper: Mappers.DefaultErrorResponse
        }
    },
    queryParameters: [Parameters.apiVersion],
    urlParameters: [
        Parameters.$host,
        Parameters.subscriptionId,
        Parameters.resourceGroupName,
        Parameters.managedEnvironmentName
    ],
    headerParameters: [Parameters.accept],
    serializer
};
const getOperationSpec = {
    path: "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.App/managedEnvironments/{managedEnvironmentName}/certificates/{name}",
    httpMethod: "GET",
    responses: {
        200: {
            bodyMapper: Mappers.Certificate
        },
        default: {
            bodyMapper: Mappers.DefaultErrorResponse
        }
    },
    queryParameters: [Parameters.apiVersion],
    urlParameters: [
        Parameters.$host,
        Parameters.subscriptionId,
        Parameters.resourceGroupName,
        Parameters.name,
        Parameters.managedEnvironmentName
    ],
    headerParameters: [Parameters.accept],
    serializer
};
const createOrUpdateOperationSpec = {
    path: "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.App/managedEnvironments/{managedEnvironmentName}/certificates/{name}",
    httpMethod: "PUT",
    responses: {
        200: {
            bodyMapper: Mappers.Certificate
        },
        default: {
            bodyMapper: Mappers.DefaultErrorResponse
        }
    },
    requestBody: Parameters.certificateEnvelope,
    queryParameters: [Parameters.apiVersion],
    urlParameters: [
        Parameters.$host,
        Parameters.subscriptionId,
        Parameters.resourceGroupName,
        Parameters.name,
        Parameters.managedEnvironmentName
    ],
    headerParameters: [Parameters.accept, Parameters.contentType],
    mediaType: "json",
    serializer
};
const deleteOperationSpec = {
    path: "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.App/managedEnvironments/{managedEnvironmentName}/certificates/{name}",
    httpMethod: "DELETE",
    responses: {
        200: {},
        204: {},
        default: {
            bodyMapper: Mappers.DefaultErrorResponse
        }
    },
    queryParameters: [Parameters.apiVersion],
    urlParameters: [
        Parameters.$host,
        Parameters.subscriptionId,
        Parameters.resourceGroupName,
        Parameters.name,
        Parameters.managedEnvironmentName
    ],
    headerParameters: [Parameters.accept],
    serializer
};
const updateOperationSpec = {
    path: "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.App/managedEnvironments/{managedEnvironmentName}/certificates/{name}",
    httpMethod: "PATCH",
    responses: {
        200: {
            bodyMapper: Mappers.Certificate
        },
        default: {
            bodyMapper: Mappers.DefaultErrorResponse
        }
    },
    requestBody: Parameters.certificateEnvelope1,
    queryParameters: [Parameters.apiVersion],
    urlParameters: [
        Parameters.$host,
        Parameters.subscriptionId,
        Parameters.resourceGroupName,
        Parameters.name,
        Parameters.managedEnvironmentName
    ],
    headerParameters: [Parameters.accept, Parameters.contentType],
    mediaType: "json",
    serializer
};
const listNextOperationSpec = {
    path: "{nextLink}",
    httpMethod: "GET",
    responses: {
        200: {
            bodyMapper: Mappers.CertificateCollection
        },
        default: {
            bodyMapper: Mappers.DefaultErrorResponse
        }
    },
    queryParameters: [Parameters.apiVersion],
    urlParameters: [
        Parameters.$host,
        Parameters.subscriptionId,
        Parameters.resourceGroupName,
        Parameters.nextLink,
        Parameters.managedEnvironmentName
    ],
    headerParameters: [Parameters.accept],
    serializer
};
