// @azure/arm-migrate/dist-esm/src/models/index.js
/*
 * Copyright (c) Microsoft Corporation.
 * Licensed under the MIT License.
 *
 * Code generated by Microsoft (R) AutoRest Code Generator.
 * Changes may cause incorrect behavior and will be lost if the code is regenerated.
 */
/** Known values of {@link ProjectStatus} that the service accepts. */
export var KnownProjectStatus;
(function (KnownProjectStatus) {
    /** Active */
    KnownProjectStatus["Active"] = "Active";
    /** Inactive */
    KnownProjectStatus["Inactive"] = "Inactive";
})(KnownProjectStatus || (KnownProjectStatus = {}));
/** Known values of {@link PrivateEndpointConnectionPropertiesProvisioningState} that the service accepts. */
export var KnownPrivateEndpointConnectionPropertiesProvisioningState;
(function (KnownPrivateEndpointConnectionPropertiesProvisioningState) {
    /** Accepted */
    KnownPrivateEndpointConnectionPropertiesProvisioningState["Accepted"] = "Accepted";
    /** InProgress */
    KnownPrivateEndpointConnectionPropertiesProvisioningState["InProgress"] = "InProgress";
    /** Succeeded */
    KnownPrivateEndpointConnectionPropertiesProvisioningState["Succeeded"] = "Succeeded";
    /** Failed */
    KnownPrivateEndpointConnectionPropertiesProvisioningState["Failed"] = "Failed";
})(KnownPrivateEndpointConnectionPropertiesProvisioningState || (KnownPrivateEndpointConnectionPropertiesProvisioningState = {}));
/** Known values of {@link PrivateLinkServiceConnectionStateStatus} that the service accepts. */
export var KnownPrivateLinkServiceConnectionStateStatus;
(function (KnownPrivateLinkServiceConnectionStateStatus) {
    /** Approved */
    KnownPrivateLinkServiceConnectionStateStatus["Approved"] = "Approved";
    /** Pending */
    KnownPrivateLinkServiceConnectionStateStatus["Pending"] = "Pending";
    /** Rejected */
    KnownPrivateLinkServiceConnectionStateStatus["Rejected"] = "Rejected";
    /** Disconnected */
    KnownPrivateLinkServiceConnectionStateStatus["Disconnected"] = "Disconnected";
})(KnownPrivateLinkServiceConnectionStateStatus || (KnownPrivateLinkServiceConnectionStateStatus = {}));
/** Known values of {@link ProvisioningState} that the service accepts. */
export var KnownProvisioningState;
(function (KnownProvisioningState) {
    /** Accepted */
    KnownProvisioningState["Accepted"] = "Accepted";
    /** Creating */
    KnownProvisioningState["Creating"] = "Creating";
    /** Deleting */
    KnownProvisioningState["Deleting"] = "Deleting";
    /** Failed */
    KnownProvisioningState["Failed"] = "Failed";
    /** Moving */
    KnownProvisioningState["Moving"] = "Moving";
    /** Succeeded */
    KnownProvisioningState["Succeeded"] = "Succeeded";
})(KnownProvisioningState || (KnownProvisioningState = {}));
/** Known values of {@link MachineBootType} that the service accepts. */
export var KnownMachineBootType;
(function (KnownMachineBootType) {
    /** Unknown */
    KnownMachineBootType["Unknown"] = "Unknown";
    /** EFI */
    KnownMachineBootType["EFI"] = "EFI";
    /** Bios */
    KnownMachineBootType["Bios"] = "BIOS";
})(KnownMachineBootType || (KnownMachineBootType = {}));
/** Known values of {@link GroupStatus} that the service accepts. */
export var KnownGroupStatus;
(function (KnownGroupStatus) {
    /** Created */
    KnownGroupStatus["Created"] = "Created";
    /** Updated */
    KnownGroupStatus["Updated"] = "Updated";
    /** Running */
    KnownGroupStatus["Running"] = "Running";
    /** Completed */
    KnownGroupStatus["Completed"] = "Completed";
    /** Invalid */
    KnownGroupStatus["Invalid"] = "Invalid";
})(KnownGroupStatus || (KnownGroupStatus = {}));
/** Known values of {@link GroupUpdateOperation} that the service accepts. */
export var KnownGroupUpdateOperation;
(function (KnownGroupUpdateOperation) {
    /** Add */
    KnownGroupUpdateOperation["Add"] = "Add";
    /** Remove */
    KnownGroupUpdateOperation["Remove"] = "Remove";
})(KnownGroupUpdateOperation || (KnownGroupUpdateOperation = {}));
/** Known values of {@link AzureLocation} that the service accepts. */
export var KnownAzureLocation;
(function (KnownAzureLocation) {
    /** Unknown */
    KnownAzureLocation["Unknown"] = "Unknown";
    /** EastAsia */
    KnownAzureLocation["EastAsia"] = "EastAsia";
    /** SoutheastAsia */
    KnownAzureLocation["SoutheastAsia"] = "SoutheastAsia";
    /** AustraliaEast */
    KnownAzureLocation["AustraliaEast"] = "AustraliaEast";
    /** AustraliaSoutheast */
    KnownAzureLocation["AustraliaSoutheast"] = "AustraliaSoutheast";
    /** BrazilSouth */
    KnownAzureLocation["BrazilSouth"] = "BrazilSouth";
    /** CanadaCentral */
    KnownAzureLocation["CanadaCentral"] = "CanadaCentral";
    /** CanadaEast */
    KnownAzureLocation["CanadaEast"] = "CanadaEast";
    /** WestEurope */
    KnownAzureLocation["WestEurope"] = "WestEurope";
    /** NorthEurope */
    KnownAzureLocation["NorthEurope"] = "NorthEurope";
    /** CentralIndia */
    KnownAzureLocation["CentralIndia"] = "CentralIndia";
    /** SouthIndia */
    KnownAzureLocation["SouthIndia"] = "SouthIndia";
    /** WestIndia */
    KnownAzureLocation["WestIndia"] = "WestIndia";
    /** JapanEast */
    KnownAzureLocation["JapanEast"] = "JapanEast";
    /** JapanWest */
    KnownAzureLocation["JapanWest"] = "JapanWest";
    /** KoreaCentral */
    KnownAzureLocation["KoreaCentral"] = "KoreaCentral";
    /** KoreaSouth */
    KnownAzureLocation["KoreaSouth"] = "KoreaSouth";
    /** UkWest */
    KnownAzureLocation["UkWest"] = "UkWest";
    /** UkSouth */
    KnownAzureLocation["UkSouth"] = "UkSouth";
    /** NorthCentralUs */
    KnownAzureLocation["NorthCentralUs"] = "NorthCentralUs";
    /** EastUs */
    KnownAzureLocation["EastUs"] = "EastUs";
    /** WestUs2 */
    KnownAzureLocation["WestUs2"] = "WestUs2";
    /** SouthCentralUs */
    KnownAzureLocation["SouthCentralUs"] = "SouthCentralUs";
    /** CentralUs */
    KnownAzureLocation["CentralUs"] = "CentralUs";
    /** EastUs2 */
    KnownAzureLocation["EastUs2"] = "EastUs2";
    /** WestUs */
    KnownAzureLocation["WestUs"] = "WestUs";
    /** WestCentralUs */
    KnownAzureLocation["WestCentralUs"] = "WestCentralUs";
    /** GermanyCentral */
    KnownAzureLocation["GermanyCentral"] = "GermanyCentral";
    /** GermanyNortheast */
    KnownAzureLocation["GermanyNortheast"] = "GermanyNortheast";
    /** ChinaNorth */
    KnownAzureLocation["ChinaNorth"] = "ChinaNorth";
    /** ChinaEast */
    KnownAzureLocation["ChinaEast"] = "ChinaEast";
    /** USGovArizona */
    KnownAzureLocation["USGovArizona"] = "USGovArizona";
    /** USGovTexas */
    KnownAzureLocation["USGovTexas"] = "USGovTexas";
    /** USGovIowa */
    KnownAzureLocation["USGovIowa"] = "USGovIowa";
    /** USGovVirginia */
    KnownAzureLocation["USGovVirginia"] = "USGovVirginia";
    /** USDoDCentral */
    KnownAzureLocation["USDoDCentral"] = "USDoDCentral";
    /** USDoDEast */
    KnownAzureLocation["USDoDEast"] = "USDoDEast";
})(KnownAzureLocation || (KnownAzureLocation = {}));
/** Known values of {@link AzureOfferCode} that the service accepts. */
export var KnownAzureOfferCode;
(function (KnownAzureOfferCode) {
    /** Unknown */
    KnownAzureOfferCode["Unknown"] = "Unknown";
    /** Msazr0003P */
    KnownAzureOfferCode["Msazr0003P"] = "MSAZR0003P";
    /** Msazr0044P */
    KnownAzureOfferCode["Msazr0044P"] = "MSAZR0044P";
    /** Msazr0059P */
    KnownAzureOfferCode["Msazr0059P"] = "MSAZR0059P";
    /** Msazr0060P */
    KnownAzureOfferCode["Msazr0060P"] = "MSAZR0060P";
    /** Msazr0062P */
    KnownAzureOfferCode["Msazr0062P"] = "MSAZR0062P";
    /** Msazr0063P */
    KnownAzureOfferCode["Msazr0063P"] = "MSAZR0063P";
    /** Msazr0064P */
    KnownAzureOfferCode["Msazr0064P"] = "MSAZR0064P";
    /** Msazr0029P */
    KnownAzureOfferCode["Msazr0029P"] = "MSAZR0029P";
    /** Msazr0022P */
    KnownAzureOfferCode["Msazr0022P"] = "MSAZR0022P";
    /** Msazr0023P */
    KnownAzureOfferCode["Msazr0023P"] = "MSAZR0023P";
    /** Msazr0148P */
    KnownAzureOfferCode["Msazr0148P"] = "MSAZR0148P";
    /** Msazr0025P */
    KnownAzureOfferCode["Msazr0025P"] = "MSAZR0025P";
    /** Msazr0036P */
    KnownAzureOfferCode["Msazr0036P"] = "MSAZR0036P";
    /** Msazr0120P */
    KnownAzureOfferCode["Msazr0120P"] = "MSAZR0120P";
    /** Msazr0121P */
    KnownAzureOfferCode["Msazr0121P"] = "MSAZR0121P";
    /** Msazr0122P */
    KnownAzureOfferCode["Msazr0122P"] = "MSAZR0122P";
    /** Msazr0123P */
    KnownAzureOfferCode["Msazr0123P"] = "MSAZR0123P";
    /** Msazr0124P */
    KnownAzureOfferCode["Msazr0124P"] = "MSAZR0124P";
    /** Msazr0125P */
    KnownAzureOfferCode["Msazr0125P"] = "MSAZR0125P";
    /** Msazr0126P */
    KnownAzureOfferCode["Msazr0126P"] = "MSAZR0126P";
    /** Msazr0127P */
    KnownAzureOfferCode["Msazr0127P"] = "MSAZR0127P";
    /** Msazr0128P */
    KnownAzureOfferCode["Msazr0128P"] = "MSAZR0128P";
    /** Msazr0129P */
    KnownAzureOfferCode["Msazr0129P"] = "MSAZR0129P";
    /** Msazr0130P */
    KnownAzureOfferCode["Msazr0130P"] = "MSAZR0130P";
    /** Msazr0111P */
    KnownAzureOfferCode["Msazr0111P"] = "MSAZR0111P";
    /** Msazr0144P */
    KnownAzureOfferCode["Msazr0144P"] = "MSAZR0144P";
    /** Msazr0149P */
    KnownAzureOfferCode["Msazr0149P"] = "MSAZR0149P";
    /** Msmcazr0044P */
    KnownAzureOfferCode["Msmcazr0044P"] = "MSMCAZR0044P";
    /** Msmcazr0059P */
    KnownAzureOfferCode["Msmcazr0059P"] = "MSMCAZR0059P";
    /** Msmcazr0060P */
    KnownAzureOfferCode["Msmcazr0060P"] = "MSMCAZR0060P";
    /** Msmcazr0063P */
    KnownAzureOfferCode["Msmcazr0063P"] = "MSMCAZR0063P";
    /** Msmcazr0120P */
    KnownAzureOfferCode["Msmcazr0120P"] = "MSMCAZR0120P";
    /** Msmcazr0121P */
    KnownAzureOfferCode["Msmcazr0121P"] = "MSMCAZR0121P";
    /** Msmcazr0125P */
    KnownAzureOfferCode["Msmcazr0125P"] = "MSMCAZR0125P";
    /** Msmcazr0128P */
    KnownAzureOfferCode["Msmcazr0128P"] = "MSMCAZR0128P";
    /** Msazrde0003P */
    KnownAzureOfferCode["Msazrde0003P"] = "MSAZRDE0003P";
    /** Msazrde0044P */
    KnownAzureOfferCode["Msazrde0044P"] = "MSAZRDE0044P";
    /** Msazrusgov0003P */
    KnownAzureOfferCode["Msazrusgov0003P"] = "MSAZRUSGOV0003P";
    /** EA */
    KnownAzureOfferCode["EA"] = "EA";
})(KnownAzureOfferCode || (KnownAzureOfferCode = {}));
/** Known values of {@link AzurePricingTier} that the service accepts. */
export var KnownAzurePricingTier;
(function (KnownAzurePricingTier) {
    /** Standard */
    KnownAzurePricingTier["Standard"] = "Standard";
    /** Basic */
    KnownAzurePricingTier["Basic"] = "Basic";
})(KnownAzurePricingTier || (KnownAzurePricingTier = {}));
/** Known values of {@link AzureStorageRedundancy} that the service accepts. */
export var KnownAzureStorageRedundancy;
(function (KnownAzureStorageRedundancy) {
    /** Unknown */
    KnownAzureStorageRedundancy["Unknown"] = "Unknown";
    /** LocallyRedundant */
    KnownAzureStorageRedundancy["LocallyRedundant"] = "LocallyRedundant";
    /** ZoneRedundant */
    KnownAzureStorageRedundancy["ZoneRedundant"] = "ZoneRedundant";
    /** GeoRedundant */
    KnownAzureStorageRedundancy["GeoRedundant"] = "GeoRedundant";
    /** ReadAccessGeoRedundant */
    KnownAzureStorageRedundancy["ReadAccessGeoRedundant"] = "ReadAccessGeoRedundant";
})(KnownAzureStorageRedundancy || (KnownAzureStorageRedundancy = {}));
/** Known values of {@link Percentile} that the service accepts. */
export var KnownPercentile;
(function (KnownPercentile) {
    /** Percentile50 */
    KnownPercentile["Percentile50"] = "Percentile50";
    /** Percentile90 */
    KnownPercentile["Percentile90"] = "Percentile90";
    /** Percentile95 */
    KnownPercentile["Percentile95"] = "Percentile95";
    /** Percentile99 */
    KnownPercentile["Percentile99"] = "Percentile99";
})(KnownPercentile || (KnownPercentile = {}));
/** Known values of {@link TimeRange} that the service accepts. */
export var KnownTimeRange;
(function (KnownTimeRange) {
    /** Day */
    KnownTimeRange["Day"] = "Day";
    /** Week */
    KnownTimeRange["Week"] = "Week";
    /** Month */
    KnownTimeRange["Month"] = "Month";
    /** Custom */
    KnownTimeRange["Custom"] = "Custom";
})(KnownTimeRange || (KnownTimeRange = {}));
/** Known values of {@link AssessmentStage} that the service accepts. */
export var KnownAssessmentStage;
(function (KnownAssessmentStage) {
    /** InProgress */
    KnownAssessmentStage["InProgress"] = "InProgress";
    /** UnderReview */
    KnownAssessmentStage["UnderReview"] = "UnderReview";
    /** Approved */
    KnownAssessmentStage["Approved"] = "Approved";
})(KnownAssessmentStage || (KnownAssessmentStage = {}));
/** Known values of {@link Currency} that the service accepts. */
export var KnownCurrency;
(function (KnownCurrency) {
    /** Unknown */
    KnownCurrency["Unknown"] = "Unknown";
    /** USD */
    KnownCurrency["USD"] = "USD";
    /** DKK */
    KnownCurrency["DKK"] = "DKK";
    /** CAD */
    KnownCurrency["CAD"] = "CAD";
    /** IDR */
    KnownCurrency["IDR"] = "IDR";
    /** JPY */
    KnownCurrency["JPY"] = "JPY";
    /** KRW */
    KnownCurrency["KRW"] = "KRW";
    /** NZD */
    KnownCurrency["NZD"] = "NZD";
    /** NOK */
    KnownCurrency["NOK"] = "NOK";
    /** RUB */
    KnownCurrency["RUB"] = "RUB";
    /** SAR */
    KnownCurrency["SAR"] = "SAR";
    /** ZAR */
    KnownCurrency["ZAR"] = "ZAR";
    /** SEK */
    KnownCurrency["SEK"] = "SEK";
    /** TRY */
    KnownCurrency["TRY"] = "TRY";
    /** GBP */
    KnownCurrency["GBP"] = "GBP";
    /** MXN */
    KnownCurrency["MXN"] = "MXN";
    /** MYR */
    KnownCurrency["MYR"] = "MYR";
    /** INR */
    KnownCurrency["INR"] = "INR";
    /** HKD */
    KnownCurrency["HKD"] = "HKD";
    /** BRL */
    KnownCurrency["BRL"] = "BRL";
    /** TWD */
    KnownCurrency["TWD"] = "TWD";
    /** EUR */
    KnownCurrency["EUR"] = "EUR";
    /** CHF */
    KnownCurrency["CHF"] = "CHF";
    /** ARS */
    KnownCurrency["ARS"] = "ARS";
    /** AUD */
    KnownCurrency["AUD"] = "AUD";
    /** CNY */
    KnownCurrency["CNY"] = "CNY";
})(KnownCurrency || (KnownCurrency = {}));
/** Known values of {@link AzureHybridUseBenefit} that the service accepts. */
export var KnownAzureHybridUseBenefit;
(function (KnownAzureHybridUseBenefit) {
    /** Unknown */
    KnownAzureHybridUseBenefit["Unknown"] = "Unknown";
    /** Yes */
    KnownAzureHybridUseBenefit["Yes"] = "Yes";
    /** No */
    KnownAzureHybridUseBenefit["No"] = "No";
})(KnownAzureHybridUseBenefit || (KnownAzureHybridUseBenefit = {}));
/** Known values of {@link AssessmentSizingCriterion} that the service accepts. */
export var KnownAssessmentSizingCriterion;
(function (KnownAssessmentSizingCriterion) {
    /** PerformanceBased */
    KnownAssessmentSizingCriterion["PerformanceBased"] = "PerformanceBased";
    /** AsOnPremises */
    KnownAssessmentSizingCriterion["AsOnPremises"] = "AsOnPremises";
})(KnownAssessmentSizingCriterion || (KnownAssessmentSizingCriterion = {}));
/** Known values of {@link ReservedInstance} that the service accepts. */
export var KnownReservedInstance;
(function (KnownReservedInstance) {
    /** None */
    KnownReservedInstance["None"] = "None";
    /** RI1Year */
    KnownReservedInstance["RI1Year"] = "RI1Year";
    /** RI3Year */
    KnownReservedInstance["RI3Year"] = "RI3Year";
})(KnownReservedInstance || (KnownReservedInstance = {}));
/** Known values of {@link AzureVmFamily} that the service accepts. */
export var KnownAzureVmFamily;
(function (KnownAzureVmFamily) {
    /** Unknown */
    KnownAzureVmFamily["Unknown"] = "Unknown";
    /** BasicA0A4 */
    KnownAzureVmFamily["BasicA0A4"] = "Basic_A0_A4";
    /** StandardA0A7 */
    KnownAzureVmFamily["StandardA0A7"] = "Standard_A0_A7";
    /** StandardA8A11 */
    KnownAzureVmFamily["StandardA8A11"] = "Standard_A8_A11";
    /** Av2Series */
    KnownAzureVmFamily["Av2Series"] = "Av2_series";
    /** DSeries */
    KnownAzureVmFamily["DSeries"] = "D_series";
    /** Dv2Series */
    KnownAzureVmFamily["Dv2Series"] = "Dv2_series";
    /** DSSeries */
    KnownAzureVmFamily["DSSeries"] = "DS_series";
    /** DSv2Series */
    KnownAzureVmFamily["DSv2Series"] = "DSv2_series";
    /** FSeries */
    KnownAzureVmFamily["FSeries"] = "F_series";
    /** FsSeries */
    KnownAzureVmFamily["FsSeries"] = "Fs_series";
    /** GSeries */
    KnownAzureVmFamily["GSeries"] = "G_series";
    /** GSSeries */
    KnownAzureVmFamily["GSSeries"] = "GS_series";
    /** HSeries */
    KnownAzureVmFamily["HSeries"] = "H_series";
    /** LsSeries */
    KnownAzureVmFamily["LsSeries"] = "Ls_series";
    /** Dsv3Series */
    KnownAzureVmFamily["Dsv3Series"] = "Dsv3_series";
    /** Dv3Series */
    KnownAzureVmFamily["Dv3Series"] = "Dv3_series";
    /** Fsv2Series */
    KnownAzureVmFamily["Fsv2Series"] = "Fsv2_series";
    /** Ev3Series */
    KnownAzureVmFamily["Ev3Series"] = "Ev3_series";
    /** Esv3Series */
    KnownAzureVmFamily["Esv3Series"] = "Esv3_series";
    /** MSeries */
    KnownAzureVmFamily["MSeries"] = "M_series";
    /** DCSeries */
    KnownAzureVmFamily["DCSeries"] = "DC_Series";
})(KnownAzureVmFamily || (KnownAzureVmFamily = {}));
/** Known values of {@link AzureDiskType} that the service accepts. */
export var KnownAzureDiskType;
(function (KnownAzureDiskType) {
    /** Unknown */
    KnownAzureDiskType["Unknown"] = "Unknown";
    /** Standard */
    KnownAzureDiskType["Standard"] = "Standard";
    /** Premium */
    KnownAzureDiskType["Premium"] = "Premium";
    /** StandardSSD */
    KnownAzureDiskType["StandardSSD"] = "StandardSSD";
    /** StandardOrPremium */
    KnownAzureDiskType["StandardOrPremium"] = "StandardOrPremium";
})(KnownAzureDiskType || (KnownAzureDiskType = {}));
/** Known values of {@link AssessmentStatus} that the service accepts. */
export var KnownAssessmentStatus;
(function (KnownAssessmentStatus) {
    /** Created */
    KnownAssessmentStatus["Created"] = "Created";
    /** Updated */
    KnownAssessmentStatus["Updated"] = "Updated";
    /** Running */
    KnownAssessmentStatus["Running"] = "Running";
    /** Completed */
    KnownAssessmentStatus["Completed"] = "Completed";
    /** Invalid */
    KnownAssessmentStatus["Invalid"] = "Invalid";
    /** OutOfSync */
    KnownAssessmentStatus["OutOfSync"] = "OutOfSync";
    /** OutDated */
    KnownAssessmentStatus["OutDated"] = "OutDated";
})(KnownAssessmentStatus || (KnownAssessmentStatus = {}));
/** Known values of {@link AzureDiskSize} that the service accepts. */
export var KnownAzureDiskSize;
(function (KnownAzureDiskSize) {
    /** Unknown */
    KnownAzureDiskSize["Unknown"] = "Unknown";
    /** StandardS4 */
    KnownAzureDiskSize["StandardS4"] = "Standard_S4";
    /** StandardS6 */
    KnownAzureDiskSize["StandardS6"] = "Standard_S6";
    /** StandardS10 */
    KnownAzureDiskSize["StandardS10"] = "Standard_S10";
    /** StandardS15 */
    KnownAzureDiskSize["StandardS15"] = "Standard_S15";
    /** StandardS20 */
    KnownAzureDiskSize["StandardS20"] = "Standard_S20";
    /** StandardS30 */
    KnownAzureDiskSize["StandardS30"] = "Standard_S30";
    /** StandardS40 */
    KnownAzureDiskSize["StandardS40"] = "Standard_S40";
    /** StandardS50 */
    KnownAzureDiskSize["StandardS50"] = "Standard_S50";
    /** PremiumP4 */
    KnownAzureDiskSize["PremiumP4"] = "Premium_P4";
    /** PremiumP6 */
    KnownAzureDiskSize["PremiumP6"] = "Premium_P6";
    /** PremiumP10 */
    KnownAzureDiskSize["PremiumP10"] = "Premium_P10";
    /** PremiumP15 */
    KnownAzureDiskSize["PremiumP15"] = "Premium_P15";
    /** PremiumP20 */
    KnownAzureDiskSize["PremiumP20"] = "Premium_P20";
    /** PremiumP30 */
    KnownAzureDiskSize["PremiumP30"] = "Premium_P30";
    /** PremiumP40 */
    KnownAzureDiskSize["PremiumP40"] = "Premium_P40";
    /** PremiumP50 */
    KnownAzureDiskSize["PremiumP50"] = "Premium_P50";
    /** StandardS60 */
    KnownAzureDiskSize["StandardS60"] = "Standard_S60";
    /** StandardS70 */
    KnownAzureDiskSize["StandardS70"] = "Standard_S70";
    /** StandardS80 */
    KnownAzureDiskSize["StandardS80"] = "Standard_S80";
    /** PremiumP60 */
    KnownAzureDiskSize["PremiumP60"] = "Premium_P60";
    /** PremiumP70 */
    KnownAzureDiskSize["PremiumP70"] = "Premium_P70";
    /** PremiumP80 */
    KnownAzureDiskSize["PremiumP80"] = "Premium_P80";
    /** StandardSSDE10 */
    KnownAzureDiskSize["StandardSSDE10"] = "StandardSSD_E10";
    /** StandardSSDE15 */
    KnownAzureDiskSize["StandardSSDE15"] = "StandardSSD_E15";
    /** StandardSSDE20 */
    KnownAzureDiskSize["StandardSSDE20"] = "StandardSSD_E20";
    /** StandardSSDE30 */
    KnownAzureDiskSize["StandardSSDE30"] = "StandardSSD_E30";
    /** StandardSSDE40 */
    KnownAzureDiskSize["StandardSSDE40"] = "StandardSSD_E40";
    /** StandardSSDE50 */
    KnownAzureDiskSize["StandardSSDE50"] = "StandardSSD_E50";
    /** StandardSSDE60 */
    KnownAzureDiskSize["StandardSSDE60"] = "StandardSSD_E60";
    /** StandardSSDE70 */
    KnownAzureDiskSize["StandardSSDE70"] = "StandardSSD_E70";
    /** StandardSSDE80 */
    KnownAzureDiskSize["StandardSSDE80"] = "StandardSSD_E80";
    /** StandardSSDE4 */
    KnownAzureDiskSize["StandardSSDE4"] = "StandardSSD_E4";
    /** StandardSSDE6 */
    KnownAzureDiskSize["StandardSSDE6"] = "StandardSSD_E6";
})(KnownAzureDiskSize || (KnownAzureDiskSize = {}));
/** Known values of {@link CloudSuitability} that the service accepts. */
export var KnownCloudSuitability;
(function (KnownCloudSuitability) {
    /** Unknown */
    KnownCloudSuitability["Unknown"] = "Unknown";
    /** NotSuitable */
    KnownCloudSuitability["NotSuitable"] = "NotSuitable";
    /** Suitable */
    KnownCloudSuitability["Suitable"] = "Suitable";
    /** ConditionallySuitable */
    KnownCloudSuitability["ConditionallySuitable"] = "ConditionallySuitable";
    /** ReadinessUnknown */
    KnownCloudSuitability["ReadinessUnknown"] = "ReadinessUnknown";
})(KnownCloudSuitability || (KnownCloudSuitability = {}));
/** Known values of {@link AzureDiskSuitabilityExplanation} that the service accepts. */
export var KnownAzureDiskSuitabilityExplanation;
(function (KnownAzureDiskSuitabilityExplanation) {
    /** Unknown */
    KnownAzureDiskSuitabilityExplanation["Unknown"] = "Unknown";
    /** NotApplicable */
    KnownAzureDiskSuitabilityExplanation["NotApplicable"] = "NotApplicable";
    /** DiskSizeGreaterThanSupported */
    KnownAzureDiskSuitabilityExplanation["DiskSizeGreaterThanSupported"] = "DiskSizeGreaterThanSupported";
    /** NoSuitableDiskSizeForIops */
    KnownAzureDiskSuitabilityExplanation["NoSuitableDiskSizeForIops"] = "NoSuitableDiskSizeForIops";
    /** NoSuitableDiskSizeForThroughput */
    KnownAzureDiskSuitabilityExplanation["NoSuitableDiskSizeForThroughput"] = "NoSuitableDiskSizeForThroughput";
    /** NoDiskSizeFoundInSelectedLocation */
    KnownAzureDiskSuitabilityExplanation["NoDiskSizeFoundInSelectedLocation"] = "NoDiskSizeFoundInSelectedLocation";
    /** NoDiskSizeFoundForSelectedRedundancy */
    KnownAzureDiskSuitabilityExplanation["NoDiskSizeFoundForSelectedRedundancy"] = "NoDiskSizeFoundForSelectedRedundancy";
    /** InternalErrorOccurredForDiskEvaluation */
    KnownAzureDiskSuitabilityExplanation["InternalErrorOccurredForDiskEvaluation"] = "InternalErrorOccurredForDiskEvaluation";
    /** NoEaPriceFoundForDiskSize */
    KnownAzureDiskSuitabilityExplanation["NoEaPriceFoundForDiskSize"] = "NoEaPriceFoundForDiskSize";
})(KnownAzureDiskSuitabilityExplanation || (KnownAzureDiskSuitabilityExplanation = {}));
/** Known values of {@link AzureDiskSuitabilityDetail} that the service accepts. */
export var KnownAzureDiskSuitabilityDetail;
(function (KnownAzureDiskSuitabilityDetail) {
    /** None */
    KnownAzureDiskSuitabilityDetail["None"] = "None";
    /** NumberOfReadOperationsPerSecondMissing */
    KnownAzureDiskSuitabilityDetail["NumberOfReadOperationsPerSecondMissing"] = "NumberOfReadOperationsPerSecondMissing";
    /** NumberOfWriteOperationsPerSecondMissing */
    KnownAzureDiskSuitabilityDetail["NumberOfWriteOperationsPerSecondMissing"] = "NumberOfWriteOperationsPerSecondMissing";
    /** MegabytesPerSecondOfReadMissing */
    KnownAzureDiskSuitabilityDetail["MegabytesPerSecondOfReadMissing"] = "MegabytesPerSecondOfReadMissing";
    /** MegabytesPerSecondOfWriteMissing */
    KnownAzureDiskSuitabilityDetail["MegabytesPerSecondOfWriteMissing"] = "MegabytesPerSecondOfWriteMissing";
    /** DiskGigabytesConsumedMissing */
    KnownAzureDiskSuitabilityDetail["DiskGigabytesConsumedMissing"] = "DiskGigabytesConsumedMissing";
    /** DiskGigabytesProvisionedMissing */
    KnownAzureDiskSuitabilityDetail["DiskGigabytesProvisionedMissing"] = "DiskGigabytesProvisionedMissing";
    /** NumberOfReadOperationsPerSecondOutOfRange */
    KnownAzureDiskSuitabilityDetail["NumberOfReadOperationsPerSecondOutOfRange"] = "NumberOfReadOperationsPerSecondOutOfRange";
    /** NumberOfWriteOperationsPerSecondOutOfRange */
    KnownAzureDiskSuitabilityDetail["NumberOfWriteOperationsPerSecondOutOfRange"] = "NumberOfWriteOperationsPerSecondOutOfRange";
    /** MegabytesPerSecondOfReadOutOfRange */
    KnownAzureDiskSuitabilityDetail["MegabytesPerSecondOfReadOutOfRange"] = "MegabytesPerSecondOfReadOutOfRange";
    /** MegabytesPerSecondOfWriteOutOfRange */
    KnownAzureDiskSuitabilityDetail["MegabytesPerSecondOfWriteOutOfRange"] = "MegabytesPerSecondOfWriteOutOfRange";
    /** DiskGigabytesConsumedOutOfRange */
    KnownAzureDiskSuitabilityDetail["DiskGigabytesConsumedOutOfRange"] = "DiskGigabytesConsumedOutOfRange";
    /** DiskGigabytesProvisionedOutOfRange */
    KnownAzureDiskSuitabilityDetail["DiskGigabytesProvisionedOutOfRange"] = "DiskGigabytesProvisionedOutOfRange";
})(KnownAzureDiskSuitabilityDetail || (KnownAzureDiskSuitabilityDetail = {}));
/** Known values of {@link AzureNetworkAdapterSuitabilityExplanation} that the service accepts. */
export var KnownAzureNetworkAdapterSuitabilityExplanation;
(function (KnownAzureNetworkAdapterSuitabilityExplanation) {
    /** Unknown */
    KnownAzureNetworkAdapterSuitabilityExplanation["Unknown"] = "Unknown";
    /** NotApplicable */
    KnownAzureNetworkAdapterSuitabilityExplanation["NotApplicable"] = "NotApplicable";
    /** InternalErrorOccurred */
    KnownAzureNetworkAdapterSuitabilityExplanation["InternalErrorOccurred"] = "InternalErrorOccurred";
})(KnownAzureNetworkAdapterSuitabilityExplanation || (KnownAzureNetworkAdapterSuitabilityExplanation = {}));
/** Known values of {@link AzureNetworkAdapterSuitabilityDetail} that the service accepts. */
export var KnownAzureNetworkAdapterSuitabilityDetail;
(function (KnownAzureNetworkAdapterSuitabilityDetail) {
    /** None */
    KnownAzureNetworkAdapterSuitabilityDetail["None"] = "None";
    /** MegabytesOfDataTransmittedMissing */
    KnownAzureNetworkAdapterSuitabilityDetail["MegabytesOfDataTransmittedMissing"] = "MegabytesOfDataTransmittedMissing";
    /** MegabytesOfDataTransmittedOutOfRange */
    KnownAzureNetworkAdapterSuitabilityDetail["MegabytesOfDataTransmittedOutOfRange"] = "MegabytesOfDataTransmittedOutOfRange";
})(KnownAzureNetworkAdapterSuitabilityDetail || (KnownAzureNetworkAdapterSuitabilityDetail = {}));
/** Known values of {@link AzureVmSize} that the service accepts. */
export var KnownAzureVmSize;
(function (KnownAzureVmSize) {
    /** Unknown */
    KnownAzureVmSize["Unknown"] = "Unknown";
    /** BasicA0 */
    KnownAzureVmSize["BasicA0"] = "Basic_A0";
    /** BasicA1 */
    KnownAzureVmSize["BasicA1"] = "Basic_A1";
    /** BasicA2 */
    KnownAzureVmSize["BasicA2"] = "Basic_A2";
    /** BasicA3 */
    KnownAzureVmSize["BasicA3"] = "Basic_A3";
    /** BasicA4 */
    KnownAzureVmSize["BasicA4"] = "Basic_A4";
    /** StandardA0 */
    KnownAzureVmSize["StandardA0"] = "Standard_A0";
    /** StandardA1 */
    KnownAzureVmSize["StandardA1"] = "Standard_A1";
    /** StandardA2 */
    KnownAzureVmSize["StandardA2"] = "Standard_A2";
    /** StandardA3 */
    KnownAzureVmSize["StandardA3"] = "Standard_A3";
    /** StandardA4 */
    KnownAzureVmSize["StandardA4"] = "Standard_A4";
    /** StandardA5 */
    KnownAzureVmSize["StandardA5"] = "Standard_A5";
    /** StandardA6 */
    KnownAzureVmSize["StandardA6"] = "Standard_A6";
    /** StandardA7 */
    KnownAzureVmSize["StandardA7"] = "Standard_A7";
    /** StandardA8 */
    KnownAzureVmSize["StandardA8"] = "Standard_A8";
    /** StandardA9 */
    KnownAzureVmSize["StandardA9"] = "Standard_A9";
    /** StandardA10 */
    KnownAzureVmSize["StandardA10"] = "Standard_A10";
    /** StandardA11 */
    KnownAzureVmSize["StandardA11"] = "Standard_A11";
    /** StandardA1V2 */
    KnownAzureVmSize["StandardA1V2"] = "Standard_A1_v2";
    /** StandardA2V2 */
    KnownAzureVmSize["StandardA2V2"] = "Standard_A2_v2";
    /** StandardA4V2 */
    KnownAzureVmSize["StandardA4V2"] = "Standard_A4_v2";
    /** StandardA8V2 */
    KnownAzureVmSize["StandardA8V2"] = "Standard_A8_v2";
    /** StandardA2MV2 */
    KnownAzureVmSize["StandardA2MV2"] = "Standard_A2m_v2";
    /** StandardA4MV2 */
    KnownAzureVmSize["StandardA4MV2"] = "Standard_A4m_v2";
    /** StandardA8MV2 */
    KnownAzureVmSize["StandardA8MV2"] = "Standard_A8m_v2";
    /** StandardD1 */
    KnownAzureVmSize["StandardD1"] = "Standard_D1";
    /** StandardD2 */
    KnownAzureVmSize["StandardD2"] = "Standard_D2";
    /** StandardD3 */
    KnownAzureVmSize["StandardD3"] = "Standard_D3";
    /** StandardD4 */
    KnownAzureVmSize["StandardD4"] = "Standard_D4";
    /** StandardD11 */
    KnownAzureVmSize["StandardD11"] = "Standard_D11";
    /** StandardD12 */
    KnownAzureVmSize["StandardD12"] = "Standard_D12";
    /** StandardD13 */
    KnownAzureVmSize["StandardD13"] = "Standard_D13";
    /** StandardD14 */
    KnownAzureVmSize["StandardD14"] = "Standard_D14";
    /** StandardD1V2 */
    KnownAzureVmSize["StandardD1V2"] = "Standard_D1_v2";
    /** StandardD2V2 */
    KnownAzureVmSize["StandardD2V2"] = "Standard_D2_v2";
    /** StandardD3V2 */
    KnownAzureVmSize["StandardD3V2"] = "Standard_D3_v2";
    /** StandardD4V2 */
    KnownAzureVmSize["StandardD4V2"] = "Standard_D4_v2";
    /** StandardD5V2 */
    KnownAzureVmSize["StandardD5V2"] = "Standard_D5_v2";
    /** StandardD11V2 */
    KnownAzureVmSize["StandardD11V2"] = "Standard_D11_v2";
    /** StandardD12V2 */
    KnownAzureVmSize["StandardD12V2"] = "Standard_D12_v2";
    /** StandardD13V2 */
    KnownAzureVmSize["StandardD13V2"] = "Standard_D13_v2";
    /** StandardD14V2 */
    KnownAzureVmSize["StandardD14V2"] = "Standard_D14_v2";
    /** StandardD15V2 */
    KnownAzureVmSize["StandardD15V2"] = "Standard_D15_v2";
    /** StandardDS1 */
    KnownAzureVmSize["StandardDS1"] = "Standard_DS1";
    /** StandardDS2 */
    KnownAzureVmSize["StandardDS2"] = "Standard_DS2";
    /** StandardDS3 */
    KnownAzureVmSize["StandardDS3"] = "Standard_DS3";
    /** StandardDS4 */
    KnownAzureVmSize["StandardDS4"] = "Standard_DS4";
    /** StandardDS11 */
    KnownAzureVmSize["StandardDS11"] = "Standard_DS11";
    /** StandardDS12 */
    KnownAzureVmSize["StandardDS12"] = "Standard_DS12";
    /** StandardDS13 */
    KnownAzureVmSize["StandardDS13"] = "Standard_DS13";
    /** StandardDS14 */
    KnownAzureVmSize["StandardDS14"] = "Standard_DS14";
    /** StandardDS1V2 */
    KnownAzureVmSize["StandardDS1V2"] = "Standard_DS1_v2";
    /** StandardDS2V2 */
    KnownAzureVmSize["StandardDS2V2"] = "Standard_DS2_v2";
    /** StandardDS3V2 */
    KnownAzureVmSize["StandardDS3V2"] = "Standard_DS3_v2";
    /** StandardDS4V2 */
    KnownAzureVmSize["StandardDS4V2"] = "Standard_DS4_v2";
    /** StandardDS5V2 */
    KnownAzureVmSize["StandardDS5V2"] = "Standard_DS5_v2";
    /** StandardDS11V2 */
    KnownAzureVmSize["StandardDS11V2"] = "Standard_DS11_v2";
    /** StandardDS12V2 */
    KnownAzureVmSize["StandardDS12V2"] = "Standard_DS12_v2";
    /** StandardDS13V2 */
    KnownAzureVmSize["StandardDS13V2"] = "Standard_DS13_v2";
    /** StandardDS14V2 */
    KnownAzureVmSize["StandardDS14V2"] = "Standard_DS14_v2";
    /** StandardDS15V2 */
    KnownAzureVmSize["StandardDS15V2"] = "Standard_DS15_v2";
    /** StandardF1 */
    KnownAzureVmSize["StandardF1"] = "Standard_F1";
    /** StandardF2 */
    KnownAzureVmSize["StandardF2"] = "Standard_F2";
    /** StandardF4 */
    KnownAzureVmSize["StandardF4"] = "Standard_F4";
    /** StandardF8 */
    KnownAzureVmSize["StandardF8"] = "Standard_F8";
    /** StandardF16 */
    KnownAzureVmSize["StandardF16"] = "Standard_F16";
    /** StandardF1S */
    KnownAzureVmSize["StandardF1S"] = "Standard_F1s";
    /** StandardF2S */
    KnownAzureVmSize["StandardF2S"] = "Standard_F2s";
    /** StandardF4S */
    KnownAzureVmSize["StandardF4S"] = "Standard_F4s";
    /** StandardF8S */
    KnownAzureVmSize["StandardF8S"] = "Standard_F8s";
    /** StandardF16S */
    KnownAzureVmSize["StandardF16S"] = "Standard_F16s";
    /** StandardG1 */
    KnownAzureVmSize["StandardG1"] = "Standard_G1";
    /** StandardG2 */
    KnownAzureVmSize["StandardG2"] = "Standard_G2";
    /** StandardG3 */
    KnownAzureVmSize["StandardG3"] = "Standard_G3";
    /** StandardG4 */
    KnownAzureVmSize["StandardG4"] = "Standard_G4";
    /** StandardG5 */
    KnownAzureVmSize["StandardG5"] = "Standard_G5";
    /** StandardGS1 */
    KnownAzureVmSize["StandardGS1"] = "Standard_GS1";
    /** StandardGS2 */
    KnownAzureVmSize["StandardGS2"] = "Standard_GS2";
    /** StandardGS3 */
    KnownAzureVmSize["StandardGS3"] = "Standard_GS3";
    /** StandardGS4 */
    KnownAzureVmSize["StandardGS4"] = "Standard_GS4";
    /** StandardGS5 */
    KnownAzureVmSize["StandardGS5"] = "Standard_GS5";
    /** StandardH8 */
    KnownAzureVmSize["StandardH8"] = "Standard_H8";
    /** StandardH16 */
    KnownAzureVmSize["StandardH16"] = "Standard_H16";
    /** StandardH8M */
    KnownAzureVmSize["StandardH8M"] = "Standard_H8m";
    /** StandardH16M */
    KnownAzureVmSize["StandardH16M"] = "Standard_H16m";
    /** StandardH16R */
    KnownAzureVmSize["StandardH16R"] = "Standard_H16r";
    /** StandardH16Mr */
    KnownAzureVmSize["StandardH16Mr"] = "Standard_H16mr";
    /** StandardL4S */
    KnownAzureVmSize["StandardL4S"] = "Standard_L4s";
    /** StandardL8S */
    KnownAzureVmSize["StandardL8S"] = "Standard_L8s";
    /** StandardL16S */
    KnownAzureVmSize["StandardL16S"] = "Standard_L16s";
    /** StandardL32S */
    KnownAzureVmSize["StandardL32S"] = "Standard_L32s";
    /** StandardD2SV3 */
    KnownAzureVmSize["StandardD2SV3"] = "Standard_D2s_v3";
    /** StandardD4SV3 */
    KnownAzureVmSize["StandardD4SV3"] = "Standard_D4s_v3";
    /** StandardD8SV3 */
    KnownAzureVmSize["StandardD8SV3"] = "Standard_D8s_v3";
    /** StandardD16SV3 */
    KnownAzureVmSize["StandardD16SV3"] = "Standard_D16s_v3";
    /** StandardD32SV3 */
    KnownAzureVmSize["StandardD32SV3"] = "Standard_D32s_v3";
    /** StandardD64SV3 */
    KnownAzureVmSize["StandardD64SV3"] = "Standard_D64s_v3";
    /** StandardD2V3 */
    KnownAzureVmSize["StandardD2V3"] = "Standard_D2_v3";
    /** StandardD4V3 */
    KnownAzureVmSize["StandardD4V3"] = "Standard_D4_v3";
    /** StandardD8V3 */
    KnownAzureVmSize["StandardD8V3"] = "Standard_D8_v3";
    /** StandardD16V3 */
    KnownAzureVmSize["StandardD16V3"] = "Standard_D16_v3";
    /** StandardD32V3 */
    KnownAzureVmSize["StandardD32V3"] = "Standard_D32_v3";
    /** StandardD64V3 */
    KnownAzureVmSize["StandardD64V3"] = "Standard_D64_v3";
    /** StandardF2SV2 */
    KnownAzureVmSize["StandardF2SV2"] = "Standard_F2s_v2";
    /** StandardF4SV2 */
    KnownAzureVmSize["StandardF4SV2"] = "Standard_F4s_v2";
    /** StandardF8SV2 */
    KnownAzureVmSize["StandardF8SV2"] = "Standard_F8s_v2";
    /** StandardF16SV2 */
    KnownAzureVmSize["StandardF16SV2"] = "Standard_F16s_v2";
    /** StandardF32SV2 */
    KnownAzureVmSize["StandardF32SV2"] = "Standard_F32s_v2";
    /** StandardF64SV2 */
    KnownAzureVmSize["StandardF64SV2"] = "Standard_F64s_v2";
    /** StandardF72SV2 */
    KnownAzureVmSize["StandardF72SV2"] = "Standard_F72s_v2";
    /** StandardE2V3 */
    KnownAzureVmSize["StandardE2V3"] = "Standard_E2_v3";
    /** StandardE4V3 */
    KnownAzureVmSize["StandardE4V3"] = "Standard_E4_v3";
    /** StandardE8V3 */
    KnownAzureVmSize["StandardE8V3"] = "Standard_E8_v3";
    /** StandardE16V3 */
    KnownAzureVmSize["StandardE16V3"] = "Standard_E16_v3";
    /** StandardE32V3 */
    KnownAzureVmSize["StandardE32V3"] = "Standard_E32_v3";
    /** StandardE64V3 */
    KnownAzureVmSize["StandardE64V3"] = "Standard_E64_v3";
    /** StandardE2SV3 */
    KnownAzureVmSize["StandardE2SV3"] = "Standard_E2s_v3";
    /** StandardE4SV3 */
    KnownAzureVmSize["StandardE4SV3"] = "Standard_E4s_v3";
    /** StandardE8SV3 */
    KnownAzureVmSize["StandardE8SV3"] = "Standard_E8s_v3";
    /** StandardE16SV3 */
    KnownAzureVmSize["StandardE16SV3"] = "Standard_E16s_v3";
    /** StandardE32SV3 */
    KnownAzureVmSize["StandardE32SV3"] = "Standard_E32s_v3";
    /** StandardE64SV3 */
    KnownAzureVmSize["StandardE64SV3"] = "Standard_E64s_v3";
    /** StandardM64S */
    KnownAzureVmSize["StandardM64S"] = "Standard_M64s";
    /** StandardM64Ms */
    KnownAzureVmSize["StandardM64Ms"] = "Standard_M64ms";
    /** StandardM128S */
    KnownAzureVmSize["StandardM128S"] = "Standard_M128s";
    /** StandardM128Ms */
    KnownAzureVmSize["StandardM128Ms"] = "Standard_M128ms";
})(KnownAzureVmSize || (KnownAzureVmSize = {}));
/** Known values of {@link AzureVmSuitabilityExplanation} that the service accepts. */
export var KnownAzureVmSuitabilityExplanation;
(function (KnownAzureVmSuitabilityExplanation) {
    /** Unknown */
    KnownAzureVmSuitabilityExplanation["Unknown"] = "Unknown";
    /** NotApplicable */
    KnownAzureVmSuitabilityExplanation["NotApplicable"] = "NotApplicable";
    /** GuestOperatingSystemArchitectureNotSupported */
    KnownAzureVmSuitabilityExplanation["GuestOperatingSystemArchitectureNotSupported"] = "GuestOperatingSystemArchitectureNotSupported";
    /** GuestOperatingSystemNotSupported */
    KnownAzureVmSuitabilityExplanation["GuestOperatingSystemNotSupported"] = "GuestOperatingSystemNotSupported";
    /** BootTypeNotSupported */
    KnownAzureVmSuitabilityExplanation["BootTypeNotSupported"] = "BootTypeNotSupported";
    /** MoreDisksThanSupported */
    KnownAzureVmSuitabilityExplanation["MoreDisksThanSupported"] = "MoreDisksThanSupported";
    /** NoSuitableVmSizeFound */
    KnownAzureVmSuitabilityExplanation["NoSuitableVmSizeFound"] = "NoSuitableVmSizeFound";
    /** OneOrMoreDisksNotSuitable */
    KnownAzureVmSuitabilityExplanation["OneOrMoreDisksNotSuitable"] = "OneOrMoreDisksNotSuitable";
    /** OneOrMoreAdaptersNotSuitable */
    KnownAzureVmSuitabilityExplanation["OneOrMoreAdaptersNotSuitable"] = "OneOrMoreAdaptersNotSuitable";
    /** InternalErrorOccurredDuringComputeEvaluation */
    KnownAzureVmSuitabilityExplanation["InternalErrorOccurredDuringComputeEvaluation"] = "InternalErrorOccurredDuringComputeEvaluation";
    /** InternalErrorOccurredDuringStorageEvaluation */
    KnownAzureVmSuitabilityExplanation["InternalErrorOccurredDuringStorageEvaluation"] = "InternalErrorOccurredDuringStorageEvaluation";
    /** InternalErrorOccurredDuringNetworkEvaluation */
    KnownAzureVmSuitabilityExplanation["InternalErrorOccurredDuringNetworkEvaluation"] = "InternalErrorOccurredDuringNetworkEvaluation";
    /** NoVmSizeSupportsStoragePerformance */
    KnownAzureVmSuitabilityExplanation["NoVmSizeSupportsStoragePerformance"] = "NoVmSizeSupportsStoragePerformance";
    /** NoVmSizeSupportsNetworkPerformance */
    KnownAzureVmSuitabilityExplanation["NoVmSizeSupportsNetworkPerformance"] = "NoVmSizeSupportsNetworkPerformance";
    /** NoVmSizeForSelectedPricingTier */
    KnownAzureVmSuitabilityExplanation["NoVmSizeForSelectedPricingTier"] = "NoVmSizeForSelectedPricingTier";
    /** NoVmSizeForSelectedAzureLocation */
    KnownAzureVmSuitabilityExplanation["NoVmSizeForSelectedAzureLocation"] = "NoVmSizeForSelectedAzureLocation";
    /** CheckRedHatLinuxVersion */
    KnownAzureVmSuitabilityExplanation["CheckRedHatLinuxVersion"] = "CheckRedHatLinuxVersion";
    /** CheckOpenSuseLinuxVersion */
    KnownAzureVmSuitabilityExplanation["CheckOpenSuseLinuxVersion"] = "CheckOpenSuseLinuxVersion";
    /** CheckWindowsServer2008R2Version */
    KnownAzureVmSuitabilityExplanation["CheckWindowsServer2008R2Version"] = "CheckWindowsServer2008R2Version";
    /** CheckCentOsVersion */
    KnownAzureVmSuitabilityExplanation["CheckCentOsVersion"] = "CheckCentOsVersion";
    /** CheckDebianLinuxVersion */
    KnownAzureVmSuitabilityExplanation["CheckDebianLinuxVersion"] = "CheckDebianLinuxVersion";
    /** CheckSuseLinuxVersion */
    KnownAzureVmSuitabilityExplanation["CheckSuseLinuxVersion"] = "CheckSuseLinuxVersion";
    /** CheckOracleLinuxVersion */
    KnownAzureVmSuitabilityExplanation["CheckOracleLinuxVersion"] = "CheckOracleLinuxVersion";
    /** CheckUbuntuLinuxVersion */
    KnownAzureVmSuitabilityExplanation["CheckUbuntuLinuxVersion"] = "CheckUbuntuLinuxVersion";
    /** CheckCoreOsLinuxVersion */
    KnownAzureVmSuitabilityExplanation["CheckCoreOsLinuxVersion"] = "CheckCoreOsLinuxVersion";
    /** WindowsServerVersionConditionallySupported */
    KnownAzureVmSuitabilityExplanation["WindowsServerVersionConditionallySupported"] = "WindowsServerVersionConditionallySupported";
    /** NoGuestOperatingSystemConditionallySupported */
    KnownAzureVmSuitabilityExplanation["NoGuestOperatingSystemConditionallySupported"] = "NoGuestOperatingSystemConditionallySupported";
    /** WindowsClientVersionsConditionallySupported */
    KnownAzureVmSuitabilityExplanation["WindowsClientVersionsConditionallySupported"] = "WindowsClientVersionsConditionallySupported";
    /** BootTypeUnknown */
    KnownAzureVmSuitabilityExplanation["BootTypeUnknown"] = "BootTypeUnknown";
    /** GuestOperatingSystemUnknown */
    KnownAzureVmSuitabilityExplanation["GuestOperatingSystemUnknown"] = "GuestOperatingSystemUnknown";
    /** WindowsServerVersionsSupportedWithCaveat */
    KnownAzureVmSuitabilityExplanation["WindowsServerVersionsSupportedWithCaveat"] = "WindowsServerVersionsSupportedWithCaveat";
    /** WindowsOSNoLongerUnderMSSupport */
    KnownAzureVmSuitabilityExplanation["WindowsOSNoLongerUnderMSSupport"] = "WindowsOSNoLongerUnderMSSupport";
    /** EndorsedWithConditionsLinuxDistributions */
    KnownAzureVmSuitabilityExplanation["EndorsedWithConditionsLinuxDistributions"] = "EndorsedWithConditionsLinuxDistributions";
    /** UnendorsedLinuxDistributions */
    KnownAzureVmSuitabilityExplanation["UnendorsedLinuxDistributions"] = "UnendorsedLinuxDistributions";
    /** NoVmSizeForStandardPricingTier */
    KnownAzureVmSuitabilityExplanation["NoVmSizeForStandardPricingTier"] = "NoVmSizeForStandardPricingTier";
    /** NoVmSizeForBasicPricingTier */
    KnownAzureVmSuitabilityExplanation["NoVmSizeForBasicPricingTier"] = "NoVmSizeForBasicPricingTier";
})(KnownAzureVmSuitabilityExplanation || (KnownAzureVmSuitabilityExplanation = {}));
/** Known values of {@link AzureVmSuitabilityDetail} that the service accepts. */
export var KnownAzureVmSuitabilityDetail;
(function (KnownAzureVmSuitabilityDetail) {
    /** None */
    KnownAzureVmSuitabilityDetail["None"] = "None";
    /** RecommendedSizeHasLessNetworkAdapters */
    KnownAzureVmSuitabilityDetail["RecommendedSizeHasLessNetworkAdapters"] = "RecommendedSizeHasLessNetworkAdapters";
    /** CannotReportComputeCost */
    KnownAzureVmSuitabilityDetail["CannotReportComputeCost"] = "CannotReportComputeCost";
    /** CannotReportStorageCost */
    KnownAzureVmSuitabilityDetail["CannotReportStorageCost"] = "CannotReportStorageCost";
    /** CannotReportBandwidthCosts */
    KnownAzureVmSuitabilityDetail["CannotReportBandwidthCosts"] = "CannotReportBandwidthCosts";
    /** PercentageOfCoresUtilizedMissing */
    KnownAzureVmSuitabilityDetail["PercentageOfCoresUtilizedMissing"] = "PercentageOfCoresUtilizedMissing";
    /** PercentageOfMemoryUtilizedMissing */
    KnownAzureVmSuitabilityDetail["PercentageOfMemoryUtilizedMissing"] = "PercentageOfMemoryUtilizedMissing";
    /** PercentageOfCoresUtilizedOutOfRange */
    KnownAzureVmSuitabilityDetail["PercentageOfCoresUtilizedOutOfRange"] = "PercentageOfCoresUtilizedOutOfRange";
    /** PercentageOfMemoryUtilizedOutOfRange */
    KnownAzureVmSuitabilityDetail["PercentageOfMemoryUtilizedOutOfRange"] = "PercentageOfMemoryUtilizedOutOfRange";
})(KnownAzureVmSuitabilityDetail || (KnownAzureVmSuitabilityDetail = {}));
