// @azure/arm-deviceprovisioningservices/dist-esm/src/models/index.js
/*
 * Copyright (c) Microsoft Corporation.
 * Licensed under the MIT License.
 *
 * Code generated by Microsoft (R) AutoRest Code Generator.
 * Changes may cause incorrect behavior and will be lost if the code is regenerated.
 */
/** Known values of {@link CreatedByType} that the service accepts. */
export var KnownCreatedByType;
(function (KnownCreatedByType) {
    /** User */
    KnownCreatedByType["User"] = "User";
    /** Application */
    KnownCreatedByType["Application"] = "Application";
    /** ManagedIdentity */
    KnownCreatedByType["ManagedIdentity"] = "ManagedIdentity";
    /** Key */
    KnownCreatedByType["Key"] = "Key";
})(KnownCreatedByType || (KnownCreatedByType = {}));
/** Known values of {@link CertificatePurpose} that the service accepts. */
export var KnownCertificatePurpose;
(function (KnownCertificatePurpose) {
    /** ClientAuthentication */
    KnownCertificatePurpose["ClientAuthentication"] = "clientAuthentication";
    /** ServerAuthentication */
    KnownCertificatePurpose["ServerAuthentication"] = "serverAuthentication";
})(KnownCertificatePurpose || (KnownCertificatePurpose = {}));
/** Known values of {@link State} that the service accepts. */
export var KnownState;
(function (KnownState) {
    /** Activating */
    KnownState["Activating"] = "Activating";
    /** Active */
    KnownState["Active"] = "Active";
    /** Deleting */
    KnownState["Deleting"] = "Deleting";
    /** Deleted */
    KnownState["Deleted"] = "Deleted";
    /** ActivationFailed */
    KnownState["ActivationFailed"] = "ActivationFailed";
    /** DeletionFailed */
    KnownState["DeletionFailed"] = "DeletionFailed";
    /** Transitioning */
    KnownState["Transitioning"] = "Transitioning";
    /** Suspending */
    KnownState["Suspending"] = "Suspending";
    /** Suspended */
    KnownState["Suspended"] = "Suspended";
    /** Resuming */
    KnownState["Resuming"] = "Resuming";
    /** FailingOver */
    KnownState["FailingOver"] = "FailingOver";
    /** FailoverFailed */
    KnownState["FailoverFailed"] = "FailoverFailed";
})(KnownState || (KnownState = {}));
/** Known values of {@link PublicNetworkAccess} that the service accepts. */
export var KnownPublicNetworkAccess;
(function (KnownPublicNetworkAccess) {
    /** Enabled */
    KnownPublicNetworkAccess["Enabled"] = "Enabled";
    /** Disabled */
    KnownPublicNetworkAccess["Disabled"] = "Disabled";
})(KnownPublicNetworkAccess || (KnownPublicNetworkAccess = {}));
/** Known values of {@link PrivateLinkServiceConnectionStatus} that the service accepts. */
export var KnownPrivateLinkServiceConnectionStatus;
(function (KnownPrivateLinkServiceConnectionStatus) {
    /** Pending */
    KnownPrivateLinkServiceConnectionStatus["Pending"] = "Pending";
    /** Approved */
    KnownPrivateLinkServiceConnectionStatus["Approved"] = "Approved";
    /** Rejected */
    KnownPrivateLinkServiceConnectionStatus["Rejected"] = "Rejected";
    /** Disconnected */
    KnownPrivateLinkServiceConnectionStatus["Disconnected"] = "Disconnected";
})(KnownPrivateLinkServiceConnectionStatus || (KnownPrivateLinkServiceConnectionStatus = {}));
/** Known values of {@link AllocationPolicy} that the service accepts. */
export var KnownAllocationPolicy;
(function (KnownAllocationPolicy) {
    /** Hashed */
    KnownAllocationPolicy["Hashed"] = "Hashed";
    /** GeoLatency */
    KnownAllocationPolicy["GeoLatency"] = "GeoLatency";
    /** Static */
    KnownAllocationPolicy["Static"] = "Static";
})(KnownAllocationPolicy || (KnownAllocationPolicy = {}));
/** Known values of {@link AccessRightsDescription} that the service accepts. */
export var KnownAccessRightsDescription;
(function (KnownAccessRightsDescription) {
    /** ServiceConfig */
    KnownAccessRightsDescription["ServiceConfig"] = "ServiceConfig";
    /** EnrollmentRead */
    KnownAccessRightsDescription["EnrollmentRead"] = "EnrollmentRead";
    /** EnrollmentWrite */
    KnownAccessRightsDescription["EnrollmentWrite"] = "EnrollmentWrite";
    /** DeviceConnect */
    KnownAccessRightsDescription["DeviceConnect"] = "DeviceConnect";
    /** RegistrationStatusRead */
    KnownAccessRightsDescription["RegistrationStatusRead"] = "RegistrationStatusRead";
    /** RegistrationStatusWrite */
    KnownAccessRightsDescription["RegistrationStatusWrite"] = "RegistrationStatusWrite";
})(KnownAccessRightsDescription || (KnownAccessRightsDescription = {}));
/** Known values of {@link IotDpsSku} that the service accepts. */
export var KnownIotDpsSku;
(function (KnownIotDpsSku) {
    /** S1 */
    KnownIotDpsSku["S1"] = "S1";
})(KnownIotDpsSku || (KnownIotDpsSku = {}));
/** Known values of {@link NameUnavailabilityReason} that the service accepts. */
export var KnownNameUnavailabilityReason;
(function (KnownNameUnavailabilityReason) {
    /** Invalid */
    KnownNameUnavailabilityReason["Invalid"] = "Invalid";
    /** AlreadyExists */
    KnownNameUnavailabilityReason["AlreadyExists"] = "AlreadyExists";
})(KnownNameUnavailabilityReason || (KnownNameUnavailabilityReason = {}));
