// @azure/arm-labservices/dist-esm/src/models/index.js
/*
 * Copyright (c) Microsoft Corporation.
 * Licensed under the MIT License.
 *
 * Code generated by Microsoft (R) AutoRest Code Generator.
 * Changes may cause incorrect behavior and will be lost if the code is regenerated.
 */
/** Known values of {@link CreatedByType} that the service accepts. */
export var KnownCreatedByType;
(function (KnownCreatedByType) {
    /** User */
    KnownCreatedByType["User"] = "User";
    /** Application */
    KnownCreatedByType["Application"] = "Application";
    /** ManagedIdentity */
    KnownCreatedByType["ManagedIdentity"] = "ManagedIdentity";
    /** Key */
    KnownCreatedByType["Key"] = "Key";
})(KnownCreatedByType || (KnownCreatedByType = {}));
/** Known values of {@link Origin} that the service accepts. */
export var KnownOrigin;
(function (KnownOrigin) {
    /** User */
    KnownOrigin["User"] = "user";
    /** System */
    KnownOrigin["System"] = "system";
    /** UserSystem */
    KnownOrigin["UserSystem"] = "user,system";
})(KnownOrigin || (KnownOrigin = {}));
/** Known values of {@link ActionType} that the service accepts. */
export var KnownActionType;
(function (KnownActionType) {
    /** Internal */
    KnownActionType["Internal"] = "Internal";
})(KnownActionType || (KnownActionType = {}));
/** Known values of {@link LabServicesSkuTier} that the service accepts. */
export var KnownLabServicesSkuTier;
(function (KnownLabServicesSkuTier) {
    /** Standard */
    KnownLabServicesSkuTier["Standard"] = "Standard";
    /** Premium */
    KnownLabServicesSkuTier["Premium"] = "Premium";
})(KnownLabServicesSkuTier || (KnownLabServicesSkuTier = {}));
/** Known values of {@link ScaleType} that the service accepts. */
export var KnownScaleType;
(function (KnownScaleType) {
    /** The capacity is not adjustable in any way. */
    KnownScaleType["None"] = "None";
    /** The user must manually scale this SKU in and out. */
    KnownScaleType["Manual"] = "Manual";
    /** The user is permitted to scale this SKU in and out. */
    KnownScaleType["Automatic"] = "Automatic";
})(KnownScaleType || (KnownScaleType = {}));
/** Known values of {@link RestrictionType} that the service accepts. */
export var KnownRestrictionType;
(function (KnownRestrictionType) {
    /** Location */
    KnownRestrictionType["Location"] = "Location";
})(KnownRestrictionType || (KnownRestrictionType = {}));
/** Known values of {@link RestrictionReasonCode} that the service accepts. */
export var KnownRestrictionReasonCode;
(function (KnownRestrictionReasonCode) {
    /** QuotaId */
    KnownRestrictionReasonCode["QuotaId"] = "QuotaId";
    /** NotAvailableForSubscription */
    KnownRestrictionReasonCode["NotAvailableForSubscription"] = "NotAvailableForSubscription";
})(KnownRestrictionReasonCode || (KnownRestrictionReasonCode = {}));
/** Known values of {@link UsageUnit} that the service accepts. */
export var KnownUsageUnit;
(function (KnownUsageUnit) {
    /** Count */
    KnownUsageUnit["Count"] = "Count";
})(KnownUsageUnit || (KnownUsageUnit = {}));
