// @azure/arm-kusto/dist-esm/src/models/index.js
/*
 * Copyright (c) Microsoft Corporation.
 * Licensed under the MIT License.
 *
 * Code generated by Microsoft (R) AutoRest Code Generator.
 * Changes may cause incorrect behavior and will be lost if the code is regenerated.
 */
/** Known values of {@link AzureSkuName} that the service accepts. */
export var KnownAzureSkuName;
(function (KnownAzureSkuName) {
    /** DevNoSLAStandardD11V2 */
    KnownAzureSkuName["DevNoSLAStandardD11V2"] = "Dev(No SLA)_Standard_D11_v2";
    /** DevNoSLAStandardE2AV4 */
    KnownAzureSkuName["DevNoSLAStandardE2AV4"] = "Dev(No SLA)_Standard_E2a_v4";
    /** StandardD11V2 */
    KnownAzureSkuName["StandardD11V2"] = "Standard_D11_v2";
    /** StandardD12V2 */
    KnownAzureSkuName["StandardD12V2"] = "Standard_D12_v2";
    /** StandardD13V2 */
    KnownAzureSkuName["StandardD13V2"] = "Standard_D13_v2";
    /** StandardD14V2 */
    KnownAzureSkuName["StandardD14V2"] = "Standard_D14_v2";
    /** StandardD32DV4 */
    KnownAzureSkuName["StandardD32DV4"] = "Standard_D32d_v4";
    /** StandardD16DV5 */
    KnownAzureSkuName["StandardD16DV5"] = "Standard_D16d_v5";
    /** StandardD32DV5 */
    KnownAzureSkuName["StandardD32DV5"] = "Standard_D32d_v5";
    /** StandardDS13V21TBPS */
    KnownAzureSkuName["StandardDS13V21TBPS"] = "Standard_DS13_v2+1TB_PS";
    /** StandardDS13V22TBPS */
    KnownAzureSkuName["StandardDS13V22TBPS"] = "Standard_DS13_v2+2TB_PS";
    /** StandardDS14V23TBPS */
    KnownAzureSkuName["StandardDS14V23TBPS"] = "Standard_DS14_v2+3TB_PS";
    /** StandardDS14V24TBPS */
    KnownAzureSkuName["StandardDS14V24TBPS"] = "Standard_DS14_v2+4TB_PS";
    /** StandardL4S */
    KnownAzureSkuName["StandardL4S"] = "Standard_L4s";
    /** StandardL8S */
    KnownAzureSkuName["StandardL8S"] = "Standard_L8s";
    /** StandardL16S */
    KnownAzureSkuName["StandardL16S"] = "Standard_L16s";
    /** StandardL8SV2 */
    KnownAzureSkuName["StandardL8SV2"] = "Standard_L8s_v2";
    /** StandardL16SV2 */
    KnownAzureSkuName["StandardL16SV2"] = "Standard_L16s_v2";
    /** StandardL8SV3 */
    KnownAzureSkuName["StandardL8SV3"] = "Standard_L8s_v3";
    /** StandardL16SV3 */
    KnownAzureSkuName["StandardL16SV3"] = "Standard_L16s_v3";
    /** StandardL32SV3 */
    KnownAzureSkuName["StandardL32SV3"] = "Standard_L32s_v3";
    /** StandardL8AsV3 */
    KnownAzureSkuName["StandardL8AsV3"] = "Standard_L8as_v3";
    /** StandardL16AsV3 */
    KnownAzureSkuName["StandardL16AsV3"] = "Standard_L16as_v3";
    /** StandardL32AsV3 */
    KnownAzureSkuName["StandardL32AsV3"] = "Standard_L32as_v3";
    /** StandardE64IV3 */
    KnownAzureSkuName["StandardE64IV3"] = "Standard_E64i_v3";
    /** StandardE80IdsV4 */
    KnownAzureSkuName["StandardE80IdsV4"] = "Standard_E80ids_v4";
    /** StandardE2AV4 */
    KnownAzureSkuName["StandardE2AV4"] = "Standard_E2a_v4";
    /** StandardE4AV4 */
    KnownAzureSkuName["StandardE4AV4"] = "Standard_E4a_v4";
    /** StandardE8AV4 */
    KnownAzureSkuName["StandardE8AV4"] = "Standard_E8a_v4";
    /** StandardE16AV4 */
    KnownAzureSkuName["StandardE16AV4"] = "Standard_E16a_v4";
    /** StandardE8AsV41TBPS */
    KnownAzureSkuName["StandardE8AsV41TBPS"] = "Standard_E8as_v4+1TB_PS";
    /** StandardE8AsV42TBPS */
    KnownAzureSkuName["StandardE8AsV42TBPS"] = "Standard_E8as_v4+2TB_PS";
    /** StandardE16AsV43TBPS */
    KnownAzureSkuName["StandardE16AsV43TBPS"] = "Standard_E16as_v4+3TB_PS";
    /** StandardE16AsV44TBPS */
    KnownAzureSkuName["StandardE16AsV44TBPS"] = "Standard_E16as_v4+4TB_PS";
    /** StandardE8AsV51TBPS */
    KnownAzureSkuName["StandardE8AsV51TBPS"] = "Standard_E8as_v5+1TB_PS";
    /** StandardE8AsV52TBPS */
    KnownAzureSkuName["StandardE8AsV52TBPS"] = "Standard_E8as_v5+2TB_PS";
    /** StandardE16AsV53TBPS */
    KnownAzureSkuName["StandardE16AsV53TBPS"] = "Standard_E16as_v5+3TB_PS";
    /** StandardE16AsV54TBPS */
    KnownAzureSkuName["StandardE16AsV54TBPS"] = "Standard_E16as_v5+4TB_PS";
    /** StandardE2AdsV5 */
    KnownAzureSkuName["StandardE2AdsV5"] = "Standard_E2ads_v5";
    /** StandardE4AdsV5 */
    KnownAzureSkuName["StandardE4AdsV5"] = "Standard_E4ads_v5";
    /** StandardE8AdsV5 */
    KnownAzureSkuName["StandardE8AdsV5"] = "Standard_E8ads_v5";
    /** StandardE16AdsV5 */
    KnownAzureSkuName["StandardE16AdsV5"] = "Standard_E16ads_v5";
    /** StandardEC8AsV51TBPS */
    KnownAzureSkuName["StandardEC8AsV51TBPS"] = "Standard_EC8as_v5+1TB_PS";
    /** StandardEC8AsV52TBPS */
    KnownAzureSkuName["StandardEC8AsV52TBPS"] = "Standard_EC8as_v5+2TB_PS";
    /** StandardEC16AsV53TBPS */
    KnownAzureSkuName["StandardEC16AsV53TBPS"] = "Standard_EC16as_v5+3TB_PS";
    /** StandardEC16AsV54TBPS */
    KnownAzureSkuName["StandardEC16AsV54TBPS"] = "Standard_EC16as_v5+4TB_PS";
    /** StandardEC8AdsV5 */
    KnownAzureSkuName["StandardEC8AdsV5"] = "Standard_EC8ads_v5";
    /** StandardEC16AdsV5 */
    KnownAzureSkuName["StandardEC16AdsV5"] = "Standard_EC16ads_v5";
    /** StandardE8SV41TBPS */
    KnownAzureSkuName["StandardE8SV41TBPS"] = "Standard_E8s_v4+1TB_PS";
    /** StandardE8SV42TBPS */
    KnownAzureSkuName["StandardE8SV42TBPS"] = "Standard_E8s_v4+2TB_PS";
    /** StandardE16SV43TBPS */
    KnownAzureSkuName["StandardE16SV43TBPS"] = "Standard_E16s_v4+3TB_PS";
    /** StandardE16SV44TBPS */
    KnownAzureSkuName["StandardE16SV44TBPS"] = "Standard_E16s_v4+4TB_PS";
    /** StandardE8SV51TBPS */
    KnownAzureSkuName["StandardE8SV51TBPS"] = "Standard_E8s_v5+1TB_PS";
    /** StandardE8SV52TBPS */
    KnownAzureSkuName["StandardE8SV52TBPS"] = "Standard_E8s_v5+2TB_PS";
    /** StandardE16SV53TBPS */
    KnownAzureSkuName["StandardE16SV53TBPS"] = "Standard_E16s_v5+3TB_PS";
    /** StandardE16SV54TBPS */
    KnownAzureSkuName["StandardE16SV54TBPS"] = "Standard_E16s_v5+4TB_PS";
    /** StandardE2DV4 */
    KnownAzureSkuName["StandardE2DV4"] = "Standard_E2d_v4";
    /** StandardE4DV4 */
    KnownAzureSkuName["StandardE4DV4"] = "Standard_E4d_v4";
    /** StandardE8DV4 */
    KnownAzureSkuName["StandardE8DV4"] = "Standard_E8d_v4";
    /** StandardE16DV4 */
    KnownAzureSkuName["StandardE16DV4"] = "Standard_E16d_v4";
    /** StandardE2DV5 */
    KnownAzureSkuName["StandardE2DV5"] = "Standard_E2d_v5";
    /** StandardE4DV5 */
    KnownAzureSkuName["StandardE4DV5"] = "Standard_E4d_v5";
    /** StandardE8DV5 */
    KnownAzureSkuName["StandardE8DV5"] = "Standard_E8d_v5";
    /** StandardE16DV5 */
    KnownAzureSkuName["StandardE16DV5"] = "Standard_E16d_v5";
})(KnownAzureSkuName || (KnownAzureSkuName = {}));
/** Known values of {@link AzureSkuTier} that the service accepts. */
export var KnownAzureSkuTier;
(function (KnownAzureSkuTier) {
    /** Basic */
    KnownAzureSkuTier["Basic"] = "Basic";
    /** Standard */
    KnownAzureSkuTier["Standard"] = "Standard";
})(KnownAzureSkuTier || (KnownAzureSkuTier = {}));
/** Known values of {@link CreatedByType} that the service accepts. */
export var KnownCreatedByType;
(function (KnownCreatedByType) {
    /** User */
    KnownCreatedByType["User"] = "User";
    /** Application */
    KnownCreatedByType["Application"] = "Application";
    /** ManagedIdentity */
    KnownCreatedByType["ManagedIdentity"] = "ManagedIdentity";
    /** Key */
    KnownCreatedByType["Key"] = "Key";
})(KnownCreatedByType || (KnownCreatedByType = {}));
/** Known values of {@link IdentityType} that the service accepts. */
export var KnownIdentityType;
(function (KnownIdentityType) {
    /** None */
    KnownIdentityType["None"] = "None";
    /** SystemAssigned */
    KnownIdentityType["SystemAssigned"] = "SystemAssigned";
    /** UserAssigned */
    KnownIdentityType["UserAssigned"] = "UserAssigned";
    /** SystemAssignedUserAssigned */
    KnownIdentityType["SystemAssignedUserAssigned"] = "SystemAssigned, UserAssigned";
})(KnownIdentityType || (KnownIdentityType = {}));
/** Known values of {@link State} that the service accepts. */
export var KnownState;
(function (KnownState) {
    /** Creating */
    KnownState["Creating"] = "Creating";
    /** Unavailable */
    KnownState["Unavailable"] = "Unavailable";
    /** Running */
    KnownState["Running"] = "Running";
    /** Deleting */
    KnownState["Deleting"] = "Deleting";
    /** Deleted */
    KnownState["Deleted"] = "Deleted";
    /** Stopping */
    KnownState["Stopping"] = "Stopping";
    /** Stopped */
    KnownState["Stopped"] = "Stopped";
    /** Starting */
    KnownState["Starting"] = "Starting";
    /** Updating */
    KnownState["Updating"] = "Updating";
    /** Migrated */
    KnownState["Migrated"] = "Migrated";
})(KnownState || (KnownState = {}));
/** Known values of {@link ProvisioningState} that the service accepts. */
export var KnownProvisioningState;
(function (KnownProvisioningState) {
    /** Running */
    KnownProvisioningState["Running"] = "Running";
    /** Creating */
    KnownProvisioningState["Creating"] = "Creating";
    /** Deleting */
    KnownProvisioningState["Deleting"] = "Deleting";
    /** Succeeded */
    KnownProvisioningState["Succeeded"] = "Succeeded";
    /** Failed */
    KnownProvisioningState["Failed"] = "Failed";
    /** Moving */
    KnownProvisioningState["Moving"] = "Moving";
    /** Canceled */
    KnownProvisioningState["Canceled"] = "Canceled";
})(KnownProvisioningState || (KnownProvisioningState = {}));
/** Known values of {@link VnetState} that the service accepts. */
export var KnownVnetState;
(function (KnownVnetState) {
    /** Enabled */
    KnownVnetState["Enabled"] = "Enabled";
    /** Disabled */
    KnownVnetState["Disabled"] = "Disabled";
})(KnownVnetState || (KnownVnetState = {}));
/** Known values of {@link LanguageExtensionName} that the service accepts. */
export var KnownLanguageExtensionName;
(function (KnownLanguageExtensionName) {
    /** Python */
    KnownLanguageExtensionName["Python"] = "PYTHON";
    /** R */
    KnownLanguageExtensionName["R"] = "R";
})(KnownLanguageExtensionName || (KnownLanguageExtensionName = {}));
/** Known values of {@link LanguageExtensionImageName} that the service accepts. */
export var KnownLanguageExtensionImageName;
(function (KnownLanguageExtensionImageName) {
    /** R */
    KnownLanguageExtensionImageName["R"] = "R";
    /** Python365 */
    KnownLanguageExtensionImageName["Python365"] = "Python3_6_5";
    /** Python3108 */
    KnownLanguageExtensionImageName["Python3108"] = "Python3_10_8";
    /** Python3108DL */
    KnownLanguageExtensionImageName["Python3108DL"] = "Python3_10_8_DL";
    /** PythonCustomImage */
    KnownLanguageExtensionImageName["PythonCustomImage"] = "PythonCustomImage";
})(KnownLanguageExtensionImageName || (KnownLanguageExtensionImageName = {}));
/** Known values of {@link PublicNetworkAccess} that the service accepts. */
export var KnownPublicNetworkAccess;
(function (KnownPublicNetworkAccess) {
    /** Enabled */
    KnownPublicNetworkAccess["Enabled"] = "Enabled";
    /** Disabled */
    KnownPublicNetworkAccess["Disabled"] = "Disabled";
})(KnownPublicNetworkAccess || (KnownPublicNetworkAccess = {}));
/** Known values of {@link EngineType} that the service accepts. */
export var KnownEngineType;
(function (KnownEngineType) {
    /** V2 */
    KnownEngineType["V2"] = "V2";
    /** V3 */
    KnownEngineType["V3"] = "V3";
})(KnownEngineType || (KnownEngineType = {}));
/** Known values of {@link ClusterNetworkAccessFlag} that the service accepts. */
export var KnownClusterNetworkAccessFlag;
(function (KnownClusterNetworkAccessFlag) {
    /** Enabled */
    KnownClusterNetworkAccessFlag["Enabled"] = "Enabled";
    /** Disabled */
    KnownClusterNetworkAccessFlag["Disabled"] = "Disabled";
})(KnownClusterNetworkAccessFlag || (KnownClusterNetworkAccessFlag = {}));
/** Known values of {@link PublicIPType} that the service accepts. */
export var KnownPublicIPType;
(function (KnownPublicIPType) {
    /** IPv4 */
    KnownPublicIPType["IPv4"] = "IPv4";
    /** DualStack */
    KnownPublicIPType["DualStack"] = "DualStack";
})(KnownPublicIPType || (KnownPublicIPType = {}));
/** Known values of {@link MigrationClusterRole} that the service accepts. */
export var KnownMigrationClusterRole;
(function (KnownMigrationClusterRole) {
    /** Source */
    KnownMigrationClusterRole["Source"] = "Source";
    /** Destination */
    KnownMigrationClusterRole["Destination"] = "Destination";
})(KnownMigrationClusterRole || (KnownMigrationClusterRole = {}));
/** Known values of {@link Reason} that the service accepts. */
export var KnownReason;
(function (KnownReason) {
    /** Invalid */
    KnownReason["Invalid"] = "Invalid";
    /** AlreadyExists */
    KnownReason["AlreadyExists"] = "AlreadyExists";
})(KnownReason || (KnownReason = {}));
/** Known values of {@link ClusterPrincipalRole} that the service accepts. */
export var KnownClusterPrincipalRole;
(function (KnownClusterPrincipalRole) {
    /** AllDatabasesAdmin */
    KnownClusterPrincipalRole["AllDatabasesAdmin"] = "AllDatabasesAdmin";
    /** AllDatabasesViewer */
    KnownClusterPrincipalRole["AllDatabasesViewer"] = "AllDatabasesViewer";
})(KnownClusterPrincipalRole || (KnownClusterPrincipalRole = {}));
/** Known values of {@link PrincipalType} that the service accepts. */
export var KnownPrincipalType;
(function (KnownPrincipalType) {
    /** App */
    KnownPrincipalType["App"] = "App";
    /** Group */
    KnownPrincipalType["Group"] = "Group";
    /** User */
    KnownPrincipalType["User"] = "User";
})(KnownPrincipalType || (KnownPrincipalType = {}));
/** Known values of {@link DatabaseShareOrigin} that the service accepts. */
export var KnownDatabaseShareOrigin;
(function (KnownDatabaseShareOrigin) {
    /** Direct */
    KnownDatabaseShareOrigin["Direct"] = "Direct";
    /** DataShare */
    KnownDatabaseShareOrigin["DataShare"] = "DataShare";
    /** Other */
    KnownDatabaseShareOrigin["Other"] = "Other";
})(KnownDatabaseShareOrigin || (KnownDatabaseShareOrigin = {}));
/** Known values of {@link AzureScaleType} that the service accepts. */
export var KnownAzureScaleType;
(function (KnownAzureScaleType) {
    /** Automatic */
    KnownAzureScaleType["Automatic"] = "automatic";
    /** Manual */
    KnownAzureScaleType["Manual"] = "manual";
    /** None */
    KnownAzureScaleType["None"] = "none";
})(KnownAzureScaleType || (KnownAzureScaleType = {}));
/** Known values of {@link Kind} that the service accepts. */
export var KnownKind;
(function (KnownKind) {
    /** ReadWrite */
    KnownKind["ReadWrite"] = "ReadWrite";
    /** ReadOnlyFollowing */
    KnownKind["ReadOnlyFollowing"] = "ReadOnlyFollowing";
})(KnownKind || (KnownKind = {}));
/** Known values of {@link CallerRole} that the service accepts. */
export var KnownCallerRole;
(function (KnownCallerRole) {
    /** Admin */
    KnownCallerRole["Admin"] = "Admin";
    /** None */
    KnownCallerRole["None"] = "None";
})(KnownCallerRole || (KnownCallerRole = {}));
/** Known values of {@link DatabasePrincipalRole} that the service accepts. */
export var KnownDatabasePrincipalRole;
(function (KnownDatabasePrincipalRole) {
    /** Admin */
    KnownDatabasePrincipalRole["Admin"] = "Admin";
    /** Ingestor */
    KnownDatabasePrincipalRole["Ingestor"] = "Ingestor";
    /** Monitor */
    KnownDatabasePrincipalRole["Monitor"] = "Monitor";
    /** User */
    KnownDatabasePrincipalRole["User"] = "User";
    /** UnrestrictedViewer */
    KnownDatabasePrincipalRole["UnrestrictedViewer"] = "UnrestrictedViewer";
    /** Viewer */
    KnownDatabasePrincipalRole["Viewer"] = "Viewer";
})(KnownDatabasePrincipalRole || (KnownDatabasePrincipalRole = {}));
/** Known values of {@link DatabasePrincipalType} that the service accepts. */
export var KnownDatabasePrincipalType;
(function (KnownDatabasePrincipalType) {
    /** App */
    KnownDatabasePrincipalType["App"] = "App";
    /** Group */
    KnownDatabasePrincipalType["Group"] = "Group";
    /** User */
    KnownDatabasePrincipalType["User"] = "User";
})(KnownDatabasePrincipalType || (KnownDatabasePrincipalType = {}));
/** Known values of {@link Language} that the service accepts. */
export var KnownLanguage;
(function (KnownLanguage) {
    /** Python */
    KnownLanguage["Python"] = "Python";
})(KnownLanguage || (KnownLanguage = {}));
/** Known values of {@link DefaultPrincipalsModificationKind} that the service accepts. */
export var KnownDefaultPrincipalsModificationKind;
(function (KnownDefaultPrincipalsModificationKind) {
    /** Union */
    KnownDefaultPrincipalsModificationKind["Union"] = "Union";
    /** Replace */
    KnownDefaultPrincipalsModificationKind["Replace"] = "Replace";
    /** None */
    KnownDefaultPrincipalsModificationKind["None"] = "None";
})(KnownDefaultPrincipalsModificationKind || (KnownDefaultPrincipalsModificationKind = {}));
/** Known values of {@link DataConnectionKind} that the service accepts. */
export var KnownDataConnectionKind;
(function (KnownDataConnectionKind) {
    /** EventHub */
    KnownDataConnectionKind["EventHub"] = "EventHub";
    /** EventGrid */
    KnownDataConnectionKind["EventGrid"] = "EventGrid";
    /** IotHub */
    KnownDataConnectionKind["IotHub"] = "IotHub";
    /** CosmosDb */
    KnownDataConnectionKind["CosmosDb"] = "CosmosDb";
})(KnownDataConnectionKind || (KnownDataConnectionKind = {}));
/** Known values of {@link Status} that the service accepts. */
export var KnownStatus;
(function (KnownStatus) {
    /** Succeeded */
    KnownStatus["Succeeded"] = "Succeeded";
    /** Canceled */
    KnownStatus["Canceled"] = "Canceled";
    /** Failed */
    KnownStatus["Failed"] = "Failed";
    /** Running */
    KnownStatus["Running"] = "Running";
})(KnownStatus || (KnownStatus = {}));
/** Known values of {@link PrincipalsModificationKind} that the service accepts. */
export var KnownPrincipalsModificationKind;
(function (KnownPrincipalsModificationKind) {
    /** Union */
    KnownPrincipalsModificationKind["Union"] = "Union";
    /** Replace */
    KnownPrincipalsModificationKind["Replace"] = "Replace";
    /** None */
    KnownPrincipalsModificationKind["None"] = "None";
})(KnownPrincipalsModificationKind || (KnownPrincipalsModificationKind = {}));
/** Known values of {@link EventHubDataFormat} that the service accepts. */
export var KnownEventHubDataFormat;
(function (KnownEventHubDataFormat) {
    /** Multijson */
    KnownEventHubDataFormat["Multijson"] = "MULTIJSON";
    /** Json */
    KnownEventHubDataFormat["Json"] = "JSON";
    /** CSV */
    KnownEventHubDataFormat["CSV"] = "CSV";
    /** TSV */
    KnownEventHubDataFormat["TSV"] = "TSV";
    /** Scsv */
    KnownEventHubDataFormat["Scsv"] = "SCSV";
    /** Sohsv */
    KnownEventHubDataFormat["Sohsv"] = "SOHSV";
    /** PSV */
    KnownEventHubDataFormat["PSV"] = "PSV";
    /** TXT */
    KnownEventHubDataFormat["TXT"] = "TXT";
    /** RAW */
    KnownEventHubDataFormat["RAW"] = "RAW";
    /** Singlejson */
    KnownEventHubDataFormat["Singlejson"] = "SINGLEJSON";
    /** Avro */
    KnownEventHubDataFormat["Avro"] = "AVRO";
    /** Tsve */
    KnownEventHubDataFormat["Tsve"] = "TSVE";
    /** Parquet */
    KnownEventHubDataFormat["Parquet"] = "PARQUET";
    /** ORC */
    KnownEventHubDataFormat["ORC"] = "ORC";
    /** Apacheavro */
    KnownEventHubDataFormat["Apacheavro"] = "APACHEAVRO";
    /** W3Clogfile */
    KnownEventHubDataFormat["W3Clogfile"] = "W3CLOGFILE";
})(KnownEventHubDataFormat || (KnownEventHubDataFormat = {}));
/** Known values of {@link Compression} that the service accepts. */
export var KnownCompression;
(function (KnownCompression) {
    /** None */
    KnownCompression["None"] = "None";
    /** GZip */
    KnownCompression["GZip"] = "GZip";
})(KnownCompression || (KnownCompression = {}));
/** Known values of {@link DatabaseRouting} that the service accepts. */
export var KnownDatabaseRouting;
(function (KnownDatabaseRouting) {
    /** Single */
    KnownDatabaseRouting["Single"] = "Single";
    /** Multi */
    KnownDatabaseRouting["Multi"] = "Multi";
})(KnownDatabaseRouting || (KnownDatabaseRouting = {}));
/** Known values of {@link IotHubDataFormat} that the service accepts. */
export var KnownIotHubDataFormat;
(function (KnownIotHubDataFormat) {
    /** Multijson */
    KnownIotHubDataFormat["Multijson"] = "MULTIJSON";
    /** Json */
    KnownIotHubDataFormat["Json"] = "JSON";
    /** CSV */
    KnownIotHubDataFormat["CSV"] = "CSV";
    /** TSV */
    KnownIotHubDataFormat["TSV"] = "TSV";
    /** Scsv */
    KnownIotHubDataFormat["Scsv"] = "SCSV";
    /** Sohsv */
    KnownIotHubDataFormat["Sohsv"] = "SOHSV";
    /** PSV */
    KnownIotHubDataFormat["PSV"] = "PSV";
    /** TXT */
    KnownIotHubDataFormat["TXT"] = "TXT";
    /** RAW */
    KnownIotHubDataFormat["RAW"] = "RAW";
    /** Singlejson */
    KnownIotHubDataFormat["Singlejson"] = "SINGLEJSON";
    /** Avro */
    KnownIotHubDataFormat["Avro"] = "AVRO";
    /** Tsve */
    KnownIotHubDataFormat["Tsve"] = "TSVE";
    /** Parquet */
    KnownIotHubDataFormat["Parquet"] = "PARQUET";
    /** ORC */
    KnownIotHubDataFormat["ORC"] = "ORC";
    /** Apacheavro */
    KnownIotHubDataFormat["Apacheavro"] = "APACHEAVRO";
    /** W3Clogfile */
    KnownIotHubDataFormat["W3Clogfile"] = "W3CLOGFILE";
})(KnownIotHubDataFormat || (KnownIotHubDataFormat = {}));
/** Known values of {@link EventGridDataFormat} that the service accepts. */
export var KnownEventGridDataFormat;
(function (KnownEventGridDataFormat) {
    /** Multijson */
    KnownEventGridDataFormat["Multijson"] = "MULTIJSON";
    /** Json */
    KnownEventGridDataFormat["Json"] = "JSON";
    /** CSV */
    KnownEventGridDataFormat["CSV"] = "CSV";
    /** TSV */
    KnownEventGridDataFormat["TSV"] = "TSV";
    /** Scsv */
    KnownEventGridDataFormat["Scsv"] = "SCSV";
    /** Sohsv */
    KnownEventGridDataFormat["Sohsv"] = "SOHSV";
    /** PSV */
    KnownEventGridDataFormat["PSV"] = "PSV";
    /** TXT */
    KnownEventGridDataFormat["TXT"] = "TXT";
    /** RAW */
    KnownEventGridDataFormat["RAW"] = "RAW";
    /** Singlejson */
    KnownEventGridDataFormat["Singlejson"] = "SINGLEJSON";
    /** Avro */
    KnownEventGridDataFormat["Avro"] = "AVRO";
    /** Tsve */
    KnownEventGridDataFormat["Tsve"] = "TSVE";
    /** Parquet */
    KnownEventGridDataFormat["Parquet"] = "PARQUET";
    /** ORC */
    KnownEventGridDataFormat["ORC"] = "ORC";
    /** Apacheavro */
    KnownEventGridDataFormat["Apacheavro"] = "APACHEAVRO";
    /** W3Clogfile */
    KnownEventGridDataFormat["W3Clogfile"] = "W3CLOGFILE";
})(KnownEventGridDataFormat || (KnownEventGridDataFormat = {}));
/** Known values of {@link BlobStorageEventType} that the service accepts. */
export var KnownBlobStorageEventType;
(function (KnownBlobStorageEventType) {
    /** MicrosoftStorageBlobCreated */
    KnownBlobStorageEventType["MicrosoftStorageBlobCreated"] = "Microsoft.Storage.BlobCreated";
    /** MicrosoftStorageBlobRenamed */
    KnownBlobStorageEventType["MicrosoftStorageBlobRenamed"] = "Microsoft.Storage.BlobRenamed";
})(KnownBlobStorageEventType || (KnownBlobStorageEventType = {}));
