// @aws-sdk/client-glue/dist-es/models/models_1.js
import { GlueServiceException as __BaseException } from "./GlueServiceException";
export class IdempotentParameterMismatchException extends __BaseException {
    constructor(opts) {
        super({
            name: "IdempotentParameterMismatchException",
            $fault: "client",
            ...opts,
        });
        this.name = "IdempotentParameterMismatchException";
        this.$fault = "client";
        Object.setPrototypeOf(this, IdempotentParameterMismatchException.prototype);
        this.Message = opts.Message;
    }
}
export class ConcurrentModificationException extends __BaseException {
    constructor(opts) {
        super({
            name: "ConcurrentModificationException",
            $fault: "client",
            ...opts,
        });
        this.name = "ConcurrentModificationException";
        this.$fault = "client";
        Object.setPrototypeOf(this, ConcurrentModificationException.prototype);
        this.Message = opts.Message;
    }
}
export const Permission = {
    ALL: "ALL",
    ALTER: "ALTER",
    CREATE_DATABASE: "CREATE_DATABASE",
    CREATE_TABLE: "CREATE_TABLE",
    DATA_LOCATION_ACCESS: "DATA_LOCATION_ACCESS",
    DELETE: "DELETE",
    DROP: "DROP",
    INSERT: "INSERT",
    SELECT: "SELECT",
};
export class FederatedResourceAlreadyExistsException extends __BaseException {
    constructor(opts) {
        super({
            name: "FederatedResourceAlreadyExistsException",
            $fault: "client",
            ...opts,
        });
        this.name = "FederatedResourceAlreadyExistsException";
        this.$fault = "client";
        Object.setPrototypeOf(this, FederatedResourceAlreadyExistsException.prototype);
        this.Message = opts.Message;
        this.AssociatedGlueResource = opts.AssociatedGlueResource;
    }
}
export class ValidationException extends __BaseException {
    constructor(opts) {
        super({
            name: "ValidationException",
            $fault: "client",
            ...opts,
        });
        this.name = "ValidationException";
        this.$fault = "client";
        Object.setPrototypeOf(this, ValidationException.prototype);
        this.Message = opts.Message;
    }
}
export const TransformType = {
    FIND_MATCHES: "FIND_MATCHES",
};
export const MLUserDataEncryptionModeString = {
    DISABLED: "DISABLED",
    SSEKMS: "SSE-KMS",
};
export const Compatibility = {
    BACKWARD: "BACKWARD",
    BACKWARD_ALL: "BACKWARD_ALL",
    DISABLED: "DISABLED",
    FORWARD: "FORWARD",
    FORWARD_ALL: "FORWARD_ALL",
    FULL: "FULL",
    FULL_ALL: "FULL_ALL",
    NONE: "NONE",
};
export const SchemaStatus = {
    AVAILABLE: "AVAILABLE",
    DELETING: "DELETING",
    PENDING: "PENDING",
};
export const SchemaVersionStatus = {
    AVAILABLE: "AVAILABLE",
    DELETING: "DELETING",
    FAILURE: "FAILURE",
    PENDING: "PENDING",
};
export const Language = {
    PYTHON: "PYTHON",
    SCALA: "SCALA",
};
export const CloudWatchEncryptionMode = {
    DISABLED: "DISABLED",
    SSEKMS: "SSE-KMS",
};
export const JobBookmarksEncryptionMode = {
    CSEKMS: "CSE-KMS",
    DISABLED: "DISABLED",
};
export const S3EncryptionMode = {
    DISABLED: "DISABLED",
    SSEKMS: "SSE-KMS",
    SSES3: "SSE-S3",
};
export const SessionStatus = {
    FAILED: "FAILED",
    PROVISIONING: "PROVISIONING",
    READY: "READY",
    STOPPED: "STOPPED",
    STOPPING: "STOPPING",
    TIMEOUT: "TIMEOUT",
};
export const MetadataOperation = {
    CREATE: "CREATE",
};
export const PrincipalType = {
    GROUP: "GROUP",
    ROLE: "ROLE",
    USER: "USER",
};
export const ResourceType = {
    ARCHIVE: "ARCHIVE",
    FILE: "FILE",
    JAR: "JAR",
};
export class CrawlerRunningException extends __BaseException {
    constructor(opts) {
        super({
            name: "CrawlerRunningException",
            $fault: "client",
            ...opts,
        });
        this.name = "CrawlerRunningException";
        this.$fault = "client";
        Object.setPrototypeOf(this, CrawlerRunningException.prototype);
        this.Message = opts.Message;
    }
}
export class SchedulerTransitioningException extends __BaseException {
    constructor(opts) {
        super({
            name: "SchedulerTransitioningException",
            $fault: "client",
            ...opts,
        });
        this.name = "SchedulerTransitioningException";
        this.$fault = "client";
        Object.setPrototypeOf(this, SchedulerTransitioningException.prototype);
        this.Message = opts.Message;
    }
}
export class ConflictException extends __BaseException {
    constructor(opts) {
        super({
            name: "ConflictException",
            $fault: "client",
            ...opts,
        });
        this.name = "ConflictException";
        this.$fault = "client";
        Object.setPrototypeOf(this, ConflictException.prototype);
        this.Message = opts.Message;
    }
}
export const RegistryStatus = {
    AVAILABLE: "AVAILABLE",
    DELETING: "DELETING",
};
export class ConditionCheckFailureException extends __BaseException {
    constructor(opts) {
        super({
            name: "ConditionCheckFailureException",
            $fault: "client",
            ...opts,
        });
        this.name = "ConditionCheckFailureException";
        this.$fault = "client";
        Object.setPrototypeOf(this, ConditionCheckFailureException.prototype);
        this.Message = opts.Message;
    }
}
export const BlueprintRunState = {
    FAILED: "FAILED",
    ROLLING_BACK: "ROLLING_BACK",
    RUNNING: "RUNNING",
    SUCCEEDED: "SUCCEEDED",
};
export const ColumnStatisticsType = {
    BINARY: "BINARY",
    BOOLEAN: "BOOLEAN",
    DATE: "DATE",
    DECIMAL: "DECIMAL",
    DOUBLE: "DOUBLE",
    LONG: "LONG",
    STRING: "STRING",
};
export const ColumnStatisticsState = {
    FAILED: "FAILED",
    RUNNING: "RUNNING",
    STARTING: "STARTING",
    STOPPED: "STOPPED",
    SUCCEEDED: "SUCCEEDED",
};
export const ResourceShareType = {
    ALL: "ALL",
    FEDERATED: "FEDERATED",
    FOREIGN: "FOREIGN",
};
export const CatalogEncryptionMode = {
    DISABLED: "DISABLED",
    SSEKMS: "SSE-KMS",
    SSEKMSWITHSERVICEROLE: "SSE-KMS-WITH-SERVICE-ROLE",
};
export const TaskType = {
    EVALUATION: "EVALUATION",
    EXPORT_LABELS: "EXPORT_LABELS",
    FIND_MATCHES: "FIND_MATCHES",
    IMPORT_LABELS: "IMPORT_LABELS",
    LABELING_SET_GENERATION: "LABELING_SET_GENERATION",
};
export const TaskRunSortColumnType = {
    STARTED: "STARTED",
    STATUS: "STATUS",
    TASK_RUN_TYPE: "TASK_RUN_TYPE",
};
export const SortDirectionType = {
    ASCENDING: "ASCENDING",
    DESCENDING: "DESCENDING",
};
export const TransformStatusType = {
    DELETING: "DELETING",
    NOT_READY: "NOT_READY",
    READY: "READY",
};
export const TransformSortColumnType = {
    CREATED: "CREATED",
    LAST_MODIFIED: "LAST_MODIFIED",
    NAME: "NAME",
    STATUS: "STATUS",
    TRANSFORM_TYPE: "TRANSFORM_TYPE",
};
export const BackfillErrorCode = {
    ENCRYPTED_PARTITION_ERROR: "ENCRYPTED_PARTITION_ERROR",
    INTERNAL_ERROR: "INTERNAL_ERROR",
    INVALID_PARTITION_TYPE_DATA_ERROR: "INVALID_PARTITION_TYPE_DATA_ERROR",
    MISSING_PARTITION_VALUE_ERROR: "MISSING_PARTITION_VALUE_ERROR",
    UNSUPPORTED_PARTITION_CHARACTER_ERROR: "UNSUPPORTED_PARTITION_CHARACTER_ERROR",
};
export const PartitionIndexStatus = {
    ACTIVE: "ACTIVE",
    CREATING: "CREATING",
    DELETING: "DELETING",
    FAILED: "FAILED",
};
export const SchemaDiffType = {
    SYNTAX_DIFF: "SYNTAX_DIFF",
};
