// @azure/arm-consumption/dist-esm/src/models/index.js
/*
 * Copyright (c) Microsoft Corporation.
 * Licensed under the MIT License.
 *
 * Code generated by Microsoft (R) AutoRest Code Generator.
 * Changes may cause incorrect behavior and will be lost if the code is regenerated.
 */
/** Known values of {@link Metrictype} that the service accepts. */
export var KnownMetrictype;
(function (KnownMetrictype) {
    /** Actual cost data. */
    KnownMetrictype["ActualCostMetricType"] = "actualcost";
    /** Amortized cost data. */
    KnownMetrictype["AmortizedCostMetricType"] = "amortizedcost";
    /** Usage data. */
    KnownMetrictype["UsageMetricType"] = "usage";
})(KnownMetrictype || (KnownMetrictype = {}));
/** Known values of {@link UsageDetailsKind} that the service accepts. */
export var KnownUsageDetailsKind;
(function (KnownUsageDetailsKind) {
    /** Legacy */
    KnownUsageDetailsKind["Legacy"] = "legacy";
    /** Modern */
    KnownUsageDetailsKind["Modern"] = "modern";
})(KnownUsageDetailsKind || (KnownUsageDetailsKind = {}));
/** Known values of {@link CategoryType} that the service accepts. */
export var KnownCategoryType;
(function (KnownCategoryType) {
    /** Cost */
    KnownCategoryType["Cost"] = "Cost";
})(KnownCategoryType || (KnownCategoryType = {}));
/** Known values of {@link TimeGrainType} that the service accepts. */
export var KnownTimeGrainType;
(function (KnownTimeGrainType) {
    /** Monthly */
    KnownTimeGrainType["Monthly"] = "Monthly";
    /** Quarterly */
    KnownTimeGrainType["Quarterly"] = "Quarterly";
    /** Annually */
    KnownTimeGrainType["Annually"] = "Annually";
    /** BillingMonth */
    KnownTimeGrainType["BillingMonth"] = "BillingMonth";
    /** BillingQuarter */
    KnownTimeGrainType["BillingQuarter"] = "BillingQuarter";
    /** BillingAnnual */
    KnownTimeGrainType["BillingAnnual"] = "BillingAnnual";
})(KnownTimeGrainType || (KnownTimeGrainType = {}));
/** Known values of {@link BudgetOperatorType} that the service accepts. */
export var KnownBudgetOperatorType;
(function (KnownBudgetOperatorType) {
    /** In */
    KnownBudgetOperatorType["In"] = "In";
})(KnownBudgetOperatorType || (KnownBudgetOperatorType = {}));
/** Known values of {@link OperatorType} that the service accepts. */
export var KnownOperatorType;
(function (KnownOperatorType) {
    /** EqualTo */
    KnownOperatorType["EqualTo"] = "EqualTo";
    /** GreaterThan */
    KnownOperatorType["GreaterThan"] = "GreaterThan";
    /** GreaterThanOrEqualTo */
    KnownOperatorType["GreaterThanOrEqualTo"] = "GreaterThanOrEqualTo";
})(KnownOperatorType || (KnownOperatorType = {}));
/** Known values of {@link ThresholdType} that the service accepts. */
export var KnownThresholdType;
(function (KnownThresholdType) {
    /** Actual */
    KnownThresholdType["Actual"] = "Actual";
    /** Forecasted */
    KnownThresholdType["Forecasted"] = "Forecasted";
})(KnownThresholdType || (KnownThresholdType = {}));
/** Known values of {@link CultureCode} that the service accepts. */
export var KnownCultureCode;
(function (KnownCultureCode) {
    /** EnUs */
    KnownCultureCode["EnUs"] = "en-us";
    /** JaJp */
    KnownCultureCode["JaJp"] = "ja-jp";
    /** ZhCn */
    KnownCultureCode["ZhCn"] = "zh-cn";
    /** DeDe */
    KnownCultureCode["DeDe"] = "de-de";
    /** EsEs */
    KnownCultureCode["EsEs"] = "es-es";
    /** FrFr */
    KnownCultureCode["FrFr"] = "fr-fr";
    /** ItIt */
    KnownCultureCode["ItIt"] = "it-it";
    /** KoKr */
    KnownCultureCode["KoKr"] = "ko-kr";
    /** PtBr */
    KnownCultureCode["PtBr"] = "pt-br";
    /** RuRu */
    KnownCultureCode["RuRu"] = "ru-ru";
    /** ZhTw */
    KnownCultureCode["ZhTw"] = "zh-tw";
    /** CsCz */
    KnownCultureCode["CsCz"] = "cs-cz";
    /** PlPl */
    KnownCultureCode["PlPl"] = "pl-pl";
    /** TrTr */
    KnownCultureCode["TrTr"] = "tr-tr";
    /** DaDk */
    KnownCultureCode["DaDk"] = "da-dk";
    /** EnGb */
    KnownCultureCode["EnGb"] = "en-gb";
    /** HuHu */
    KnownCultureCode["HuHu"] = "hu-hu";
    /** NbNo */
    KnownCultureCode["NbNo"] = "nb-no";
    /** NlNl */
    KnownCultureCode["NlNl"] = "nl-nl";
    /** PtPt */
    KnownCultureCode["PtPt"] = "pt-pt";
    /** SvSe */
    KnownCultureCode["SvSe"] = "sv-se";
})(KnownCultureCode || (KnownCultureCode = {}));
/** Known values of {@link ChargeSummaryKind} that the service accepts. */
export var KnownChargeSummaryKind;
(function (KnownChargeSummaryKind) {
    /** Legacy */
    KnownChargeSummaryKind["Legacy"] = "legacy";
    /** Modern */
    KnownChargeSummaryKind["Modern"] = "modern";
})(KnownChargeSummaryKind || (KnownChargeSummaryKind = {}));
/** Known values of {@link BillingFrequency} that the service accepts. */
export var KnownBillingFrequency;
(function (KnownBillingFrequency) {
    /** Month */
    KnownBillingFrequency["Month"] = "Month";
    /** Quarter */
    KnownBillingFrequency["Quarter"] = "Quarter";
    /** Year */
    KnownBillingFrequency["Year"] = "Year";
})(KnownBillingFrequency || (KnownBillingFrequency = {}));
/** Known values of {@link Datagrain} that the service accepts. */
export var KnownDatagrain;
(function (KnownDatagrain) {
    /** Daily grain of data */
    KnownDatagrain["DailyGrain"] = "daily";
    /** Monthly grain of data */
    KnownDatagrain["MonthlyGrain"] = "monthly";
})(KnownDatagrain || (KnownDatagrain = {}));
/** Known values of {@link ReservationRecommendationKind} that the service accepts. */
export var KnownReservationRecommendationKind;
(function (KnownReservationRecommendationKind) {
    /** Legacy */
    KnownReservationRecommendationKind["Legacy"] = "legacy";
    /** Modern */
    KnownReservationRecommendationKind["Modern"] = "modern";
})(KnownReservationRecommendationKind || (KnownReservationRecommendationKind = {}));
/** Known values of {@link Term} that the service accepts. */
export var KnownTerm;
(function (KnownTerm) {
    /** 1 year reservation term */
    KnownTerm["P1Y"] = "P1Y";
    /** 3 year reservation term */
    KnownTerm["P3Y"] = "P3Y";
})(KnownTerm || (KnownTerm = {}));
/** Known values of {@link LookBackPeriod} that the service accepts. */
export var KnownLookBackPeriod;
(function (KnownLookBackPeriod) {
    /** Use 7 days of data for recommendations */
    KnownLookBackPeriod["Last07Days"] = "Last7Days";
    /** Use 30 days of data for recommendations */
    KnownLookBackPeriod["Last30Days"] = "Last30Days";
    /** Use 60 days of data for recommendations */
    KnownLookBackPeriod["Last60Days"] = "Last60Days";
})(KnownLookBackPeriod || (KnownLookBackPeriod = {}));
/** Known values of {@link EventType} that the service accepts. */
export var KnownEventType;
(function (KnownEventType) {
    /** SettledCharges */
    KnownEventType["SettledCharges"] = "SettledCharges";
    /** PendingCharges */
    KnownEventType["PendingCharges"] = "PendingCharges";
    /** PendingAdjustments */
    KnownEventType["PendingAdjustments"] = "PendingAdjustments";
    /** PendingNewCredit */
    KnownEventType["PendingNewCredit"] = "PendingNewCredit";
    /** PendingExpiredCredit */
    KnownEventType["PendingExpiredCredit"] = "PendingExpiredCredit";
    /** UnKnown */
    KnownEventType["UnKnown"] = "UnKnown";
    /** NewCredit */
    KnownEventType["NewCredit"] = "NewCredit";
})(KnownEventType || (KnownEventType = {}));
/** Known values of {@link LotSource} that the service accepts. */
export var KnownLotSource;
(function (KnownLotSource) {
    /** PurchasedCredit */
    KnownLotSource["PurchasedCredit"] = "PurchasedCredit";
    /** PromotionalCredit */
    KnownLotSource["PromotionalCredit"] = "PromotionalCredit";
    /** ConsumptionCommitment */
    KnownLotSource["ConsumptionCommitment"] = "ConsumptionCommitment";
})(KnownLotSource || (KnownLotSource = {}));
/** Known values of {@link Status} that the service accepts. */
export var KnownStatus;
(function (KnownStatus) {
    /** None */
    KnownStatus["None"] = "None";
    /** Active */
    KnownStatus["Active"] = "Active";
    /** Inactive */
    KnownStatus["Inactive"] = "Inactive";
    /** Expired */
    KnownStatus["Expired"] = "Expired";
    /** Complete */
    KnownStatus["Complete"] = "Complete";
    /** Canceled */
    KnownStatus["Canceled"] = "Canceled";
})(KnownStatus || (KnownStatus = {}));
/** Known values of {@link PricingModelType} that the service accepts. */
export var KnownPricingModelType;
(function (KnownPricingModelType) {
    /** OnDemand */
    KnownPricingModelType["OnDemand"] = "On Demand";
    /** Reservation */
    KnownPricingModelType["Reservation"] = "Reservation";
    /** Spot */
    KnownPricingModelType["Spot"] = "Spot";
})(KnownPricingModelType || (KnownPricingModelType = {}));
/** Known values of {@link Scope} that the service accepts. */
export var KnownScope;
(function (KnownScope) {
    /** Single */
    KnownScope["Single"] = "Single";
    /** Shared */
    KnownScope["Shared"] = "Shared";
})(KnownScope || (KnownScope = {}));
