// @azure/arm-communication/dist-esm/src/operations/senderUsernames.js
/*
 * Copyright (c) Microsoft Corporation.
 * Licensed under the MIT License.
 *
 * Code generated by Microsoft (R) AutoRest Code Generator.
 * Changes may cause incorrect behavior and will be lost if the code is regenerated.
 */
import { __asyncDelegator, __asyncGenerator, __asyncValues, __await } from "tslib";
import { setContinuationToken } from "../pagingHelper";
import * as coreClient from "@azure/core-client";
import * as Mappers from "../models/mappers";
import * as Parameters from "../models/parameters";
/// <reference lib="esnext.asynciterable" />
/** Class containing SenderUsernames operations. */
export class SenderUsernamesImpl {
    /**
     * Initialize a new instance of the class SenderUsernames class.
     * @param client Reference to the service client
     */
    constructor(client) {
        this.client = client;
    }
    /**
     * List all valid sender usernames for a domains resource.
     * @param resourceGroupName The name of the resource group. The name is case insensitive.
     * @param emailServiceName The name of the EmailService resource.
     * @param domainName The name of the Domains resource.
     * @param options The options parameters.
     */
    listByDomains(resourceGroupName, emailServiceName, domainName, options) {
        const iter = this.listByDomainsPagingAll(resourceGroupName, emailServiceName, domainName, options);
        return {
            next() {
                return iter.next();
            },
            [Symbol.asyncIterator]() {
                return this;
            },
            byPage: (settings) => {
                if (settings === null || settings === void 0 ? void 0 : settings.maxPageSize) {
                    throw new Error("maxPageSize is not supported by this operation.");
                }
                return this.listByDomainsPagingPage(resourceGroupName, emailServiceName, domainName, options, settings);
            },
        };
    }
    listByDomainsPagingPage(resourceGroupName, emailServiceName, domainName, options, settings) {
        return __asyncGenerator(this, arguments, function* listByDomainsPagingPage_1() {
            let result;
            let continuationToken = settings === null || settings === void 0 ? void 0 : settings.continuationToken;
            if (!continuationToken) {
                result = yield __await(this._listByDomains(resourceGroupName, emailServiceName, domainName, options));
                let page = result.value || [];
                continuationToken = result.nextLink;
                setContinuationToken(page, continuationToken);
                yield yield __await(page);
            }
            while (continuationToken) {
                result = yield __await(this._listByDomainsNext(resourceGroupName, emailServiceName, domainName, continuationToken, options));
                continuationToken = result.nextLink;
                let page = result.value || [];
                setContinuationToken(page, continuationToken);
                yield yield __await(page);
            }
        });
    }
    listByDomainsPagingAll(resourceGroupName, emailServiceName, domainName, options) {
        return __asyncGenerator(this, arguments, function* listByDomainsPagingAll_1() {
            var _a, e_1, _b, _c;
            try {
                for (var _d = true, _e = __asyncValues(this.listByDomainsPagingPage(resourceGroupName, emailServiceName, domainName, options)), _f; _f = yield __await(_e.next()), _a = _f.done, !_a; _d = true) {
                    _c = _f.value;
                    _d = false;
                    const page = _c;
                    yield __await(yield* __asyncDelegator(__asyncValues(page)));
                }
            }
            catch (e_1_1) { e_1 = { error: e_1_1 }; }
            finally {
                try {
                    if (!_d && !_a && (_b = _e.return)) yield __await(_b.call(_e));
                }
                finally { if (e_1) throw e_1.error; }
            }
        });
    }
    /**
     * List all valid sender usernames for a domains resource.
     * @param resourceGroupName The name of the resource group. The name is case insensitive.
     * @param emailServiceName The name of the EmailService resource.
     * @param domainName The name of the Domains resource.
     * @param options The options parameters.
     */
    _listByDomains(resourceGroupName, emailServiceName, domainName, options) {
        return this.client.sendOperationRequest({ resourceGroupName, emailServiceName, domainName, options }, listByDomainsOperationSpec);
    }
    /**
     * Get a valid sender username for a domains resource.
     * @param resourceGroupName The name of the resource group. The name is case insensitive.
     * @param emailServiceName The name of the EmailService resource.
     * @param domainName The name of the Domains resource.
     * @param senderUsername The valid sender Username.
     * @param options The options parameters.
     */
    get(resourceGroupName, emailServiceName, domainName, senderUsername, options) {
        return this.client.sendOperationRequest({
            resourceGroupName,
            emailServiceName,
            domainName,
            senderUsername,
            options,
        }, getOperationSpec);
    }
    /**
     * Add a new SenderUsername resource under the parent Domains resource or update an existing
     * SenderUsername resource.
     * @param resourceGroupName The name of the resource group. The name is case insensitive.
     * @param emailServiceName The name of the EmailService resource.
     * @param domainName The name of the Domains resource.
     * @param senderUsername The valid sender Username.
     * @param parameters Parameters for the create or update operation
     * @param options The options parameters.
     */
    createOrUpdate(resourceGroupName, emailServiceName, domainName, senderUsername, parameters, options) {
        return this.client.sendOperationRequest({
            resourceGroupName,
            emailServiceName,
            domainName,
            senderUsername,
            parameters,
            options,
        }, createOrUpdateOperationSpec);
    }
    /**
     * Operation to delete a SenderUsernames resource.
     * @param resourceGroupName The name of the resource group. The name is case insensitive.
     * @param emailServiceName The name of the EmailService resource.
     * @param domainName The name of the Domains resource.
     * @param senderUsername The valid sender Username.
     * @param options The options parameters.
     */
    delete(resourceGroupName, emailServiceName, domainName, senderUsername, options) {
        return this.client.sendOperationRequest({
            resourceGroupName,
            emailServiceName,
            domainName,
            senderUsername,
            options,
        }, deleteOperationSpec);
    }
    /**
     * ListByDomainsNext
     * @param resourceGroupName The name of the resource group. The name is case insensitive.
     * @param emailServiceName The name of the EmailService resource.
     * @param domainName The name of the Domains resource.
     * @param nextLink The nextLink from the previous successful call to the ListByDomains method.
     * @param options The options parameters.
     */
    _listByDomainsNext(resourceGroupName, emailServiceName, domainName, nextLink, options) {
        return this.client.sendOperationRequest({ resourceGroupName, emailServiceName, domainName, nextLink, options }, listByDomainsNextOperationSpec);
    }
}
// Operation Specifications
const serializer = coreClient.createSerializer(Mappers, /* isXml */ false);
const listByDomainsOperationSpec = {
    path: "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Communication/emailServices/{emailServiceName}/domains/{domainName}/senderUsernames",
    httpMethod: "GET",
    responses: {
        200: {
            bodyMapper: Mappers.SenderUsernameResourceCollection,
        },
        default: {
            bodyMapper: Mappers.ErrorResponse,
        },
    },
    queryParameters: [Parameters.apiVersion],
    urlParameters: [
        Parameters.$host,
        Parameters.subscriptionId,
        Parameters.resourceGroupName,
        Parameters.emailServiceName,
        Parameters.domainName,
    ],
    headerParameters: [Parameters.accept],
    serializer,
};
const getOperationSpec = {
    path: "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Communication/emailServices/{emailServiceName}/domains/{domainName}/senderUsernames/{senderUsername}",
    httpMethod: "GET",
    responses: {
        200: {
            bodyMapper: Mappers.SenderUsernameResource,
        },
        default: {
            bodyMapper: Mappers.ErrorResponse,
        },
    },
    queryParameters: [Parameters.apiVersion],
    urlParameters: [
        Parameters.$host,
        Parameters.subscriptionId,
        Parameters.resourceGroupName,
        Parameters.emailServiceName,
        Parameters.domainName,
        Parameters.senderUsername,
    ],
    headerParameters: [Parameters.accept],
    serializer,
};
const createOrUpdateOperationSpec = {
    path: "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Communication/emailServices/{emailServiceName}/domains/{domainName}/senderUsernames/{senderUsername}",
    httpMethod: "PUT",
    responses: {
        200: {
            bodyMapper: Mappers.SenderUsernameResource,
        },
        201: {
            bodyMapper: Mappers.SenderUsernameResource,
        },
        default: {
            bodyMapper: Mappers.ErrorResponse,
        },
    },
    requestBody: Parameters.parameters8,
    queryParameters: [Parameters.apiVersion],
    urlParameters: [
        Parameters.$host,
        Parameters.subscriptionId,
        Parameters.resourceGroupName,
        Parameters.emailServiceName,
        Parameters.domainName,
        Parameters.senderUsername,
    ],
    headerParameters: [Parameters.accept, Parameters.contentType],
    mediaType: "json",
    serializer,
};
const deleteOperationSpec = {
    path: "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Communication/emailServices/{emailServiceName}/domains/{domainName}/senderUsernames/{senderUsername}",
    httpMethod: "DELETE",
    responses: {
        200: {},
        204: {},
        default: {
            bodyMapper: Mappers.ErrorResponse,
        },
    },
    queryParameters: [Parameters.apiVersion],
    urlParameters: [
        Parameters.$host,
        Parameters.subscriptionId,
        Parameters.resourceGroupName,
        Parameters.emailServiceName,
        Parameters.domainName,
        Parameters.senderUsername,
    ],
    headerParameters: [Parameters.accept],
    serializer,
};
const listByDomainsNextOperationSpec = {
    path: "{nextLink}",
    httpMethod: "GET",
    responses: {
        200: {
            bodyMapper: Mappers.SenderUsernameResourceCollection,
        },
        default: {
            bodyMapper: Mappers.ErrorResponse,
        },
    },
    urlParameters: [
        Parameters.$host,
        Parameters.nextLink,
        Parameters.subscriptionId,
        Parameters.resourceGroupName,
        Parameters.emailServiceName,
        Parameters.domainName,
    ],
    headerParameters: [Parameters.accept],
    serializer,
};
