// @azure/arm-chaos/dist-esm/src/operations/capabilities.js
/*
 * Copyright (c) Microsoft Corporation.
 * Licensed under the MIT License.
 *
 * Code generated by Microsoft (R) AutoRest Code Generator.
 * Changes may cause incorrect behavior and will be lost if the code is regenerated.
 */
import { __asyncDelegator, __asyncGenerator, __asyncValues, __await } from "tslib";
import { setContinuationToken } from "../pagingHelper";
import * as coreClient from "@azure/core-client";
import * as Mappers from "../models/mappers";
import * as Parameters from "../models/parameters";
/// <reference lib="esnext.asynciterable" />
/** Class containing Capabilities operations. */
export class CapabilitiesImpl {
    /**
     * Initialize a new instance of the class Capabilities class.
     * @param client Reference to the service client
     */
    constructor(client) {
        this.client = client;
    }
    /**
     * Get a list of Capability resources that extend a Target resource..
     * @param resourceGroupName String that represents an Azure resource group.
     * @param parentProviderNamespace String that represents a resource provider namespace.
     * @param parentResourceType String that represents a resource type.
     * @param parentResourceName String that represents a resource name.
     * @param targetName String that represents a Target resource name.
     * @param options The options parameters.
     */
    list(resourceGroupName, parentProviderNamespace, parentResourceType, parentResourceName, targetName, options) {
        const iter = this.listPagingAll(resourceGroupName, parentProviderNamespace, parentResourceType, parentResourceName, targetName, options);
        return {
            next() {
                return iter.next();
            },
            [Symbol.asyncIterator]() {
                return this;
            },
            byPage: (settings) => {
                if (settings === null || settings === void 0 ? void 0 : settings.maxPageSize) {
                    throw new Error("maxPageSize is not supported by this operation.");
                }
                return this.listPagingPage(resourceGroupName, parentProviderNamespace, parentResourceType, parentResourceName, targetName, options, settings);
            }
        };
    }
    listPagingPage(resourceGroupName, parentProviderNamespace, parentResourceType, parentResourceName, targetName, options, settings) {
        return __asyncGenerator(this, arguments, function* listPagingPage_1() {
            let result;
            let continuationToken = settings === null || settings === void 0 ? void 0 : settings.continuationToken;
            if (!continuationToken) {
                result = yield __await(this._list(resourceGroupName, parentProviderNamespace, parentResourceType, parentResourceName, targetName, options));
                let page = result.value || [];
                continuationToken = result.nextLink;
                setContinuationToken(page, continuationToken);
                yield yield __await(page);
            }
            while (continuationToken) {
                result = yield __await(this._listNext(resourceGroupName, parentProviderNamespace, parentResourceType, parentResourceName, targetName, continuationToken, options));
                continuationToken = result.nextLink;
                let page = result.value || [];
                setContinuationToken(page, continuationToken);
                yield yield __await(page);
            }
        });
    }
    listPagingAll(resourceGroupName, parentProviderNamespace, parentResourceType, parentResourceName, targetName, options) {
        return __asyncGenerator(this, arguments, function* listPagingAll_1() {
            var _a, e_1, _b, _c;
            try {
                for (var _d = true, _e = __asyncValues(this.listPagingPage(resourceGroupName, parentProviderNamespace, parentResourceType, parentResourceName, targetName, options)), _f; _f = yield __await(_e.next()), _a = _f.done, !_a; _d = true) {
                    _c = _f.value;
                    _d = false;
                    const page = _c;
                    yield __await(yield* __asyncDelegator(__asyncValues(page)));
                }
            }
            catch (e_1_1) { e_1 = { error: e_1_1 }; }
            finally {
                try {
                    if (!_d && !_a && (_b = _e.return)) yield __await(_b.call(_e));
                }
                finally { if (e_1) throw e_1.error; }
            }
        });
    }
    /**
     * Get a list of Capability resources that extend a Target resource..
     * @param resourceGroupName String that represents an Azure resource group.
     * @param parentProviderNamespace String that represents a resource provider namespace.
     * @param parentResourceType String that represents a resource type.
     * @param parentResourceName String that represents a resource name.
     * @param targetName String that represents a Target resource name.
     * @param options The options parameters.
     */
    _list(resourceGroupName, parentProviderNamespace, parentResourceType, parentResourceName, targetName, options) {
        return this.client.sendOperationRequest({
            resourceGroupName,
            parentProviderNamespace,
            parentResourceType,
            parentResourceName,
            targetName,
            options
        }, listOperationSpec);
    }
    /**
     * Get a Capability resource that extends a Target resource.
     * @param resourceGroupName String that represents an Azure resource group.
     * @param parentProviderNamespace String that represents a resource provider namespace.
     * @param parentResourceType String that represents a resource type.
     * @param parentResourceName String that represents a resource name.
     * @param targetName String that represents a Target resource name.
     * @param capabilityName String that represents a Capability resource name.
     * @param options The options parameters.
     */
    get(resourceGroupName, parentProviderNamespace, parentResourceType, parentResourceName, targetName, capabilityName, options) {
        return this.client.sendOperationRequest({
            resourceGroupName,
            parentProviderNamespace,
            parentResourceType,
            parentResourceName,
            targetName,
            capabilityName,
            options
        }, getOperationSpec);
    }
    /**
     * Delete a Capability that extends a Target resource.
     * @param resourceGroupName String that represents an Azure resource group.
     * @param parentProviderNamespace String that represents a resource provider namespace.
     * @param parentResourceType String that represents a resource type.
     * @param parentResourceName String that represents a resource name.
     * @param targetName String that represents a Target resource name.
     * @param capabilityName String that represents a Capability resource name.
     * @param options The options parameters.
     */
    delete(resourceGroupName, parentProviderNamespace, parentResourceType, parentResourceName, targetName, capabilityName, options) {
        return this.client.sendOperationRequest({
            resourceGroupName,
            parentProviderNamespace,
            parentResourceType,
            parentResourceName,
            targetName,
            capabilityName,
            options
        }, deleteOperationSpec);
    }
    /**
     * Create or update a Capability resource that extends a Target resource.
     * @param resourceGroupName String that represents an Azure resource group.
     * @param parentProviderNamespace String that represents a resource provider namespace.
     * @param parentResourceType String that represents a resource type.
     * @param parentResourceName String that represents a resource name.
     * @param targetName String that represents a Target resource name.
     * @param capabilityName String that represents a Capability resource name.
     * @param capability Capability resource to be created or updated.
     * @param options The options parameters.
     */
    createOrUpdate(resourceGroupName, parentProviderNamespace, parentResourceType, parentResourceName, targetName, capabilityName, capability, options) {
        return this.client.sendOperationRequest({
            resourceGroupName,
            parentProviderNamespace,
            parentResourceType,
            parentResourceName,
            targetName,
            capabilityName,
            capability,
            options
        }, createOrUpdateOperationSpec);
    }
    /**
     * ListNext
     * @param resourceGroupName String that represents an Azure resource group.
     * @param parentProviderNamespace String that represents a resource provider namespace.
     * @param parentResourceType String that represents a resource type.
     * @param parentResourceName String that represents a resource name.
     * @param targetName String that represents a Target resource name.
     * @param nextLink The nextLink from the previous successful call to the List method.
     * @param options The options parameters.
     */
    _listNext(resourceGroupName, parentProviderNamespace, parentResourceType, parentResourceName, targetName, nextLink, options) {
        return this.client.sendOperationRequest({
            resourceGroupName,
            parentProviderNamespace,
            parentResourceType,
            parentResourceName,
            targetName,
            nextLink,
            options
        }, listNextOperationSpec);
    }
}
// Operation Specifications
const serializer = coreClient.createSerializer(Mappers, /* isXml */ false);
const listOperationSpec = {
    path: "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/{parentProviderNamespace}/{parentResourceType}/{parentResourceName}/providers/Microsoft.Chaos/targets/{targetName}/capabilities",
    httpMethod: "GET",
    responses: {
        200: {
            bodyMapper: Mappers.CapabilityListResult
        },
        default: {
            bodyMapper: Mappers.ErrorResponse
        }
    },
    queryParameters: [Parameters.apiVersion, Parameters.continuationToken],
    urlParameters: [
        Parameters.$host,
        Parameters.subscriptionId,
        Parameters.resourceGroupName,
        Parameters.parentProviderNamespace,
        Parameters.parentResourceType,
        Parameters.parentResourceName,
        Parameters.targetName
    ],
    headerParameters: [Parameters.accept],
    serializer
};
const getOperationSpec = {
    path: "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/{parentProviderNamespace}/{parentResourceType}/{parentResourceName}/providers/Microsoft.Chaos/targets/{targetName}/capabilities/{capabilityName}",
    httpMethod: "GET",
    responses: {
        200: {
            bodyMapper: Mappers.Capability
        },
        default: {
            bodyMapper: Mappers.ErrorResponse
        }
    },
    queryParameters: [Parameters.apiVersion],
    urlParameters: [
        Parameters.$host,
        Parameters.subscriptionId,
        Parameters.resourceGroupName,
        Parameters.parentProviderNamespace,
        Parameters.parentResourceType,
        Parameters.parentResourceName,
        Parameters.targetName,
        Parameters.capabilityName
    ],
    headerParameters: [Parameters.accept],
    serializer
};
const deleteOperationSpec = {
    path: "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/{parentProviderNamespace}/{parentResourceType}/{parentResourceName}/providers/Microsoft.Chaos/targets/{targetName}/capabilities/{capabilityName}",
    httpMethod: "DELETE",
    responses: {
        200: {},
        204: {},
        default: {
            bodyMapper: Mappers.ErrorResponse
        }
    },
    queryParameters: [Parameters.apiVersion],
    urlParameters: [
        Parameters.$host,
        Parameters.subscriptionId,
        Parameters.resourceGroupName,
        Parameters.parentProviderNamespace,
        Parameters.parentResourceType,
        Parameters.parentResourceName,
        Parameters.targetName,
        Parameters.capabilityName
    ],
    headerParameters: [Parameters.accept],
    serializer
};
const createOrUpdateOperationSpec = {
    path: "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/{parentProviderNamespace}/{parentResourceType}/{parentResourceName}/providers/Microsoft.Chaos/targets/{targetName}/capabilities/{capabilityName}",
    httpMethod: "PUT",
    responses: {
        200: {
            bodyMapper: Mappers.Capability
        },
        default: {
            bodyMapper: Mappers.ErrorResponse
        }
    },
    requestBody: Parameters.capability,
    queryParameters: [Parameters.apiVersion],
    urlParameters: [
        Parameters.$host,
        Parameters.subscriptionId,
        Parameters.resourceGroupName,
        Parameters.parentProviderNamespace,
        Parameters.parentResourceType,
        Parameters.parentResourceName,
        Parameters.targetName,
        Parameters.capabilityName
    ],
    headerParameters: [Parameters.accept, Parameters.contentType],
    mediaType: "json",
    serializer
};
const listNextOperationSpec = {
    path: "{nextLink}",
    httpMethod: "GET",
    responses: {
        200: {
            bodyMapper: Mappers.CapabilityListResult
        },
        default: {
            bodyMapper: Mappers.ErrorResponse
        }
    },
    urlParameters: [
        Parameters.$host,
        Parameters.subscriptionId,
        Parameters.resourceGroupName,
        Parameters.parentProviderNamespace,
        Parameters.parentResourceType,
        Parameters.parentResourceName,
        Parameters.targetName,
        Parameters.nextLink
    ],
    headerParameters: [Parameters.accept],
    serializer
};
