// @azure/arm-datafactory/dist-esm/src/models/parameters.js
/*
 * Copyright (c) Microsoft Corporation.
 * Licensed under the MIT License.
 *
 * Code generated by Microsoft (R) AutoRest Code Generator.
 * Changes may cause incorrect behavior and will be lost if the code is regenerated.
 */
import { FactoryRepoUpdate as FactoryRepoUpdateMapper, Factory as FactoryMapper, FactoryUpdateParameters as FactoryUpdateParametersMapper, GitHubAccessTokenRequest as GitHubAccessTokenRequestMapper, UserAccessPolicy as UserAccessPolicyMapper, ExposureControlRequest as ExposureControlRequestMapper, ExposureControlBatchRequest as ExposureControlBatchRequestMapper, IntegrationRuntimeResource as IntegrationRuntimeResourceMapper, UpdateIntegrationRuntimeRequest as UpdateIntegrationRuntimeRequestMapper, IntegrationRuntimeRegenerateKeyParameters as IntegrationRuntimeRegenerateKeyParametersMapper, LinkedIntegrationRuntimeRequest as LinkedIntegrationRuntimeRequestMapper, CreateLinkedIntegrationRuntimeRequest as CreateLinkedIntegrationRuntimeRequestMapper, GetSsisObjectMetadataRequest as GetSsisObjectMetadataRequestMapper, UpdateIntegrationRuntimeNodeRequest as UpdateIntegrationRuntimeNodeRequestMapper, LinkedServiceResource as LinkedServiceResourceMapper, DatasetResource as DatasetResourceMapper, PipelineResource as PipelineResourceMapper, RunFilterParameters as RunFilterParametersMapper, TriggerFilterParameters as TriggerFilterParametersMapper, TriggerResource as TriggerResourceMapper, DataFlowResource as DataFlowResourceMapper, CreateDataFlowDebugSessionRequest as CreateDataFlowDebugSessionRequestMapper, DataFlowDebugPackage as DataFlowDebugPackageMapper, DeleteDataFlowDebugSessionRequest as DeleteDataFlowDebugSessionRequestMapper, DataFlowDebugCommandRequest as DataFlowDebugCommandRequestMapper, ManagedVirtualNetworkResource as ManagedVirtualNetworkResourceMapper, ManagedPrivateEndpointResource as ManagedPrivateEndpointResourceMapper, ManagedIdentityCredentialResource as ManagedIdentityCredentialResourceMapper, PrivateLinkConnectionApprovalRequestResource as PrivateLinkConnectionApprovalRequestResourceMapper, GlobalParameterResource as GlobalParameterResourceMapper, ChangeDataCaptureResource as ChangeDataCaptureResourceMapper } from "../models/mappers";
export const accept = {
    parameterPath: "accept",
    mapper: {
        defaultValue: "application/json",
        isConstant: true,
        serializedName: "Accept",
        type: {
            name: "String"
        }
    }
};
export const $host = {
    parameterPath: "$host",
    mapper: {
        serializedName: "$host",
        required: true,
        type: {
            name: "String"
        }
    },
    skipEncoding: true
};
export const apiVersion = {
    parameterPath: "apiVersion",
    mapper: {
        defaultValue: "2018-06-01",
        isConstant: true,
        serializedName: "api-version",
        type: {
            name: "String"
        }
    }
};
export const nextLink = {
    parameterPath: "nextLink",
    mapper: {
        serializedName: "nextLink",
        required: true,
        type: {
            name: "String"
        }
    },
    skipEncoding: true
};
export const subscriptionId = {
    parameterPath: "subscriptionId",
    mapper: {
        serializedName: "subscriptionId",
        required: true,
        type: {
            name: "String"
        }
    }
};
export const contentType = {
    parameterPath: ["options", "contentType"],
    mapper: {
        defaultValue: "application/json",
        isConstant: true,
        serializedName: "Content-Type",
        type: {
            name: "String"
        }
    }
};
export const factoryRepoUpdate = {
    parameterPath: "factoryRepoUpdate",
    mapper: FactoryRepoUpdateMapper
};
export const locationId = {
    parameterPath: "locationId",
    mapper: {
        serializedName: "locationId",
        required: true,
        type: {
            name: "String"
        }
    }
};
export const resourceGroupName = {
    parameterPath: "resourceGroupName",
    mapper: {
        constraints: {
            Pattern: new RegExp("^[-\\w\\._\\(\\)]+$"),
            MaxLength: 90,
            MinLength: 1
        },
        serializedName: "resourceGroupName",
        required: true,
        type: {
            name: "String"
        }
    }
};
export const factory = {
    parameterPath: "factory",
    mapper: FactoryMapper
};
export const factoryName = {
    parameterPath: "factoryName",
    mapper: {
        constraints: {
            Pattern: new RegExp("^[A-Za-z0-9]+(?:-[A-Za-z0-9]+)*$"),
            MaxLength: 63,
            MinLength: 3
        },
        serializedName: "factoryName",
        required: true,
        type: {
            name: "String"
        }
    }
};
export const ifMatch = {
    parameterPath: ["options", "ifMatch"],
    mapper: {
        serializedName: "If-Match",
        type: {
            name: "String"
        }
    }
};
export const factoryUpdateParameters = {
    parameterPath: "factoryUpdateParameters",
    mapper: FactoryUpdateParametersMapper
};
export const ifNoneMatch = {
    parameterPath: ["options", "ifNoneMatch"],
    mapper: {
        serializedName: "If-None-Match",
        type: {
            name: "String"
        }
    }
};
export const gitHubAccessTokenRequest = {
    parameterPath: "gitHubAccessTokenRequest",
    mapper: GitHubAccessTokenRequestMapper
};
export const policy = {
    parameterPath: "policy",
    mapper: UserAccessPolicyMapper
};
export const exposureControlRequest = {
    parameterPath: "exposureControlRequest",
    mapper: ExposureControlRequestMapper
};
export const exposureControlBatchRequest = {
    parameterPath: "exposureControlBatchRequest",
    mapper: ExposureControlBatchRequestMapper
};
export const integrationRuntime = {
    parameterPath: "integrationRuntime",
    mapper: IntegrationRuntimeResourceMapper
};
export const integrationRuntimeName = {
    parameterPath: "integrationRuntimeName",
    mapper: {
        constraints: {
            Pattern: new RegExp("^[A-Za-z0-9]+(?:-[A-Za-z0-9]+)*$"),
            MaxLength: 63,
            MinLength: 3
        },
        serializedName: "integrationRuntimeName",
        required: true,
        type: {
            name: "String"
        }
    }
};
export const updateIntegrationRuntimeRequest = {
    parameterPath: "updateIntegrationRuntimeRequest",
    mapper: UpdateIntegrationRuntimeRequestMapper
};
export const regenerateKeyParameters = {
    parameterPath: "regenerateKeyParameters",
    mapper: IntegrationRuntimeRegenerateKeyParametersMapper
};
export const linkedIntegrationRuntimeRequest = {
    parameterPath: "linkedIntegrationRuntimeRequest",
    mapper: LinkedIntegrationRuntimeRequestMapper
};
export const createLinkedIntegrationRuntimeRequest = {
    parameterPath: "createLinkedIntegrationRuntimeRequest",
    mapper: CreateLinkedIntegrationRuntimeRequestMapper
};
export const getMetadataRequest = {
    parameterPath: ["options", "getMetadataRequest"],
    mapper: GetSsisObjectMetadataRequestMapper
};
export const nodeName = {
    parameterPath: "nodeName",
    mapper: {
        constraints: {
            Pattern: new RegExp("^[a-z0-9A-Z][a-z0-9A-Z_-]{0,149}$"),
            MaxLength: 150,
            MinLength: 1
        },
        serializedName: "nodeName",
        required: true,
        type: {
            name: "String"
        }
    }
};
export const updateIntegrationRuntimeNodeRequest = {
    parameterPath: "updateIntegrationRuntimeNodeRequest",
    mapper: UpdateIntegrationRuntimeNodeRequestMapper
};
export const linkedService = {
    parameterPath: "linkedService",
    mapper: LinkedServiceResourceMapper
};
export const linkedServiceName = {
    parameterPath: "linkedServiceName",
    mapper: {
        constraints: {
            Pattern: new RegExp("^[A-Za-z0-9_][^<>*#.%&:\\\\+?/]*$"),
            MaxLength: 260,
            MinLength: 1
        },
        serializedName: "linkedServiceName",
        required: true,
        type: {
            name: "String"
        }
    }
};
export const dataset = {
    parameterPath: "dataset",
    mapper: DatasetResourceMapper
};
export const datasetName = {
    parameterPath: "datasetName",
    mapper: {
        constraints: {
            Pattern: new RegExp("^[A-Za-z0-9_][^<>*#.%&:\\\\+?/]*$"),
            MaxLength: 260,
            MinLength: 1
        },
        serializedName: "datasetName",
        required: true,
        type: {
            name: "String"
        }
    }
};
export const pipeline = {
    parameterPath: "pipeline",
    mapper: PipelineResourceMapper
};
export const pipelineName = {
    parameterPath: "pipelineName",
    mapper: {
        constraints: {
            Pattern: new RegExp("^[A-Za-z0-9_][^<>*#.%&:\\\\+?/]*$"),
            MaxLength: 260,
            MinLength: 1
        },
        serializedName: "pipelineName",
        required: true,
        type: {
            name: "String"
        }
    }
};
export const parameters = {
    parameterPath: ["options", "parameters"],
    mapper: {
        serializedName: "parameters",
        type: {
            name: "Dictionary",
            value: { type: { name: "any" } }
        }
    }
};
export const referencePipelineRunId = {
    parameterPath: ["options", "referencePipelineRunId"],
    mapper: {
        serializedName: "referencePipelineRunId",
        type: {
            name: "String"
        }
    }
};
export const isRecovery = {
    parameterPath: ["options", "isRecovery"],
    mapper: {
        serializedName: "isRecovery",
        type: {
            name: "Boolean"
        }
    }
};
export const startActivityName = {
    parameterPath: ["options", "startActivityName"],
    mapper: {
        serializedName: "startActivityName",
        type: {
            name: "String"
        }
    }
};
export const startFromFailure = {
    parameterPath: ["options", "startFromFailure"],
    mapper: {
        serializedName: "startFromFailure",
        type: {
            name: "Boolean"
        }
    }
};
export const filterParameters = {
    parameterPath: "filterParameters",
    mapper: RunFilterParametersMapper
};
export const runId = {
    parameterPath: "runId",
    mapper: {
        serializedName: "runId",
        required: true,
        type: {
            name: "String"
        }
    }
};
export const isRecursive = {
    parameterPath: ["options", "isRecursive"],
    mapper: {
        serializedName: "isRecursive",
        type: {
            name: "Boolean"
        }
    }
};
export const filterParameters1 = {
    parameterPath: "filterParameters",
    mapper: TriggerFilterParametersMapper
};
export const trigger = {
    parameterPath: "trigger",
    mapper: TriggerResourceMapper
};
export const triggerName = {
    parameterPath: "triggerName",
    mapper: {
        constraints: {
            Pattern: new RegExp("^[A-Za-z0-9_][^<>*#.%&:\\\\+?/]*$"),
            MaxLength: 260,
            MinLength: 1
        },
        serializedName: "triggerName",
        required: true,
        type: {
            name: "String"
        }
    }
};
export const dataFlow = {
    parameterPath: "dataFlow",
    mapper: DataFlowResourceMapper
};
export const dataFlowName = {
    parameterPath: "dataFlowName",
    mapper: {
        constraints: {
            Pattern: new RegExp("^[A-Za-z0-9_][^<>*#.%&:\\\\+?/]*$"),
            MaxLength: 260,
            MinLength: 1
        },
        serializedName: "dataFlowName",
        required: true,
        type: {
            name: "String"
        }
    }
};
export const request = {
    parameterPath: "request",
    mapper: CreateDataFlowDebugSessionRequestMapper
};
export const request1 = {
    parameterPath: "request",
    mapper: DataFlowDebugPackageMapper
};
export const request2 = {
    parameterPath: "request",
    mapper: DeleteDataFlowDebugSessionRequestMapper
};
export const request3 = {
    parameterPath: "request",
    mapper: DataFlowDebugCommandRequestMapper
};
export const managedVirtualNetwork = {
    parameterPath: "managedVirtualNetwork",
    mapper: ManagedVirtualNetworkResourceMapper
};
export const managedVirtualNetworkName = {
    parameterPath: "managedVirtualNetworkName",
    mapper: {
        constraints: {
            Pattern: new RegExp("^([_A-Za-z0-9]|([_A-Za-z0-9][-_A-Za-z0-9]{0,125}[_A-Za-z0-9]))$"),
            MaxLength: 127,
            MinLength: 1
        },
        serializedName: "managedVirtualNetworkName",
        required: true,
        type: {
            name: "String"
        }
    }
};
export const managedPrivateEndpoint = {
    parameterPath: "managedPrivateEndpoint",
    mapper: ManagedPrivateEndpointResourceMapper
};
export const managedPrivateEndpointName = {
    parameterPath: "managedPrivateEndpointName",
    mapper: {
        constraints: {
            Pattern: new RegExp("^([_A-Za-z0-9]|([_A-Za-z0-9][-_A-Za-z0-9]{0,125}[_A-Za-z0-9]))$"),
            MaxLength: 127,
            MinLength: 1
        },
        serializedName: "managedPrivateEndpointName",
        required: true,
        type: {
            name: "String"
        }
    }
};
export const credential = {
    parameterPath: "credential",
    mapper: ManagedIdentityCredentialResourceMapper
};
export const credentialName = {
    parameterPath: "credentialName",
    mapper: {
        constraints: {
            Pattern: new RegExp("^([_A-Za-z0-9]|([_A-Za-z0-9][-_A-Za-z0-9]{0,125}[_A-Za-z0-9]))$"),
            MaxLength: 127,
            MinLength: 1
        },
        serializedName: "credentialName",
        required: true,
        type: {
            name: "String"
        }
    }
};
export const privateEndpointWrapper = {
    parameterPath: "privateEndpointWrapper",
    mapper: PrivateLinkConnectionApprovalRequestResourceMapper
};
export const privateEndpointConnectionName = {
    parameterPath: "privateEndpointConnectionName",
    mapper: {
        serializedName: "privateEndpointConnectionName",
        required: true,
        type: {
            name: "String"
        }
    }
};
export const globalParameterName = {
    parameterPath: "globalParameterName",
    mapper: {
        constraints: {
            Pattern: new RegExp("^[A-Za-z0-9_][^<>*#.%&:\\\\+?/]*$"),
            MaxLength: 260,
            MinLength: 1
        },
        serializedName: "globalParameterName",
        required: true,
        type: {
            name: "String"
        }
    }
};
export const defaultParam = {
    parameterPath: "defaultParam",
    mapper: GlobalParameterResourceMapper
};
export const changeDataCapture = {
    parameterPath: "changeDataCapture",
    mapper: ChangeDataCaptureResourceMapper
};
export const changeDataCaptureName = {
    parameterPath: "changeDataCaptureName",
    mapper: {
        constraints: {
            Pattern: new RegExp("^[A-Za-z0-9_][^<>*#.%&:\\\\+?/]*$"),
            MaxLength: 260,
            MinLength: 1
        },
        serializedName: "changeDataCaptureName",
        required: true,
        type: {
            name: "String"
        }
    }
};
