// @azure/arm-datamigration/esm/models/parameters.js
/*
 * Copyright (c) Microsoft Corporation. All rights reserved.
 * Licensed under the MIT License. See License.txt in the project root for
 * license information.
 *
 * Code generated by Microsoft (R) AutoRest Code Generator.
 * Changes may cause incorrect behavior and will be lost if the code is
 * regenerated.
 */
export var acceptLanguage = {
    parameterPath: "acceptLanguage",
    mapper: {
        serializedName: "accept-language",
        defaultValue: 'en-US',
        type: {
            name: "String"
        }
    }
};
export var apiVersion = {
    parameterPath: "apiVersion",
    mapper: {
        required: true,
        serializedName: "api-version",
        type: {
            name: "String"
        }
    }
};
export var deleteRunningTasks = {
    parameterPath: [
        "options",
        "deleteRunningTasks"
    ],
    mapper: {
        serializedName: "deleteRunningTasks",
        type: {
            name: "Boolean"
        }
    }
};
export var expand = {
    parameterPath: [
        "options",
        "expand"
    ],
    mapper: {
        serializedName: "$expand",
        type: {
            name: "String"
        }
    }
};
export var fileName = {
    parameterPath: "fileName",
    mapper: {
        required: true,
        serializedName: "fileName",
        type: {
            name: "String"
        }
    }
};
export var groupName = {
    parameterPath: "groupName",
    mapper: {
        required: true,
        serializedName: "groupName",
        type: {
            name: "String"
        }
    }
};
export var location = {
    parameterPath: "location",
    mapper: {
        required: true,
        serializedName: "location",
        type: {
            name: "String"
        }
    }
};
export var nextPageLink = {
    parameterPath: "nextPageLink",
    mapper: {
        required: true,
        serializedName: "nextLink",
        type: {
            name: "String"
        }
    },
    skipEncoding: true
};
export var projectName = {
    parameterPath: "projectName",
    mapper: {
        required: true,
        serializedName: "projectName",
        type: {
            name: "String"
        }
    }
};
export var serviceName = {
    parameterPath: "serviceName",
    mapper: {
        required: true,
        serializedName: "serviceName",
        type: {
            name: "String"
        }
    }
};
export var subscriptionId = {
    parameterPath: "subscriptionId",
    mapper: {
        required: true,
        serializedName: "subscriptionId",
        type: {
            name: "String"
        }
    }
};
export var taskName = {
    parameterPath: "taskName",
    mapper: {
        required: true,
        serializedName: "taskName",
        type: {
            name: "String"
        }
    }
};
export var taskType = {
    parameterPath: [
        "options",
        "taskType"
    ],
    mapper: {
        serializedName: "taskType",
        type: {
            name: "String"
        }
    }
};
