// @azure/arm-search/dist-esm/src/models/index.js
/*
 * Copyright (c) Microsoft Corporation.
 * Licensed under the MIT License.
 *
 * Code generated by Microsoft (R) AutoRest Code Generator.
 * Changes may cause incorrect behavior and will be lost if the code is regenerated.
 */
/** Known values of {@link PrivateLinkServiceConnectionProvisioningState} that the service accepts. */
export var KnownPrivateLinkServiceConnectionProvisioningState;
(function (KnownPrivateLinkServiceConnectionProvisioningState) {
    /** The private link service connection is in the process of being created along with other resources for it to be fully functional. */
    KnownPrivateLinkServiceConnectionProvisioningState["Updating"] = "Updating";
    /** The private link service connection is in the process of being deleted. */
    KnownPrivateLinkServiceConnectionProvisioningState["Deleting"] = "Deleting";
    /** The private link service connection has failed to be provisioned or deleted. */
    KnownPrivateLinkServiceConnectionProvisioningState["Failed"] = "Failed";
    /** The private link service connection has finished provisioning and is ready for approval. */
    KnownPrivateLinkServiceConnectionProvisioningState["Succeeded"] = "Succeeded";
    /** Provisioning request for the private link service connection resource has been accepted but the process of creation has not commenced yet. */
    KnownPrivateLinkServiceConnectionProvisioningState["Incomplete"] = "Incomplete";
    /** Provisioning request for the private link service connection resource has been canceled */
    KnownPrivateLinkServiceConnectionProvisioningState["Canceled"] = "Canceled";
})(KnownPrivateLinkServiceConnectionProvisioningState || (KnownPrivateLinkServiceConnectionProvisioningState = {}));
/** Known values of {@link SearchSemanticSearch} that the service accepts. */
export var KnownSearchSemanticSearch;
(function (KnownSearchSemanticSearch) {
    /** Indicates that semantic search is disabled for the search service. */
    KnownSearchSemanticSearch["Disabled"] = "disabled";
    /** Enables semantic search on a search service and indicates that it is to be used within the limits of the free tier. This would cap the volume of semantic search requests and is offered at no extra charge. This is the default for newly provisioned search services. */
    KnownSearchSemanticSearch["Free"] = "free";
    /** Enables semantic search on a search service as a billable feature, with higher throughput and volume of semantic search queries. */
    KnownSearchSemanticSearch["Standard"] = "standard";
})(KnownSearchSemanticSearch || (KnownSearchSemanticSearch = {}));
/** Known values of {@link UnavailableNameReason} that the service accepts. */
export var KnownUnavailableNameReason;
(function (KnownUnavailableNameReason) {
    /** The search service name does not match naming requirements. */
    KnownUnavailableNameReason["Invalid"] = "Invalid";
    /** The search service name is already assigned to a different search service. */
    KnownUnavailableNameReason["AlreadyExists"] = "AlreadyExists";
})(KnownUnavailableNameReason || (KnownUnavailableNameReason = {}));
/** Known values of {@link SharedPrivateLinkResourceAsyncOperationResult} that the service accepts. */
export var KnownSharedPrivateLinkResourceAsyncOperationResult;
(function (KnownSharedPrivateLinkResourceAsyncOperationResult) {
    /** Running */
    KnownSharedPrivateLinkResourceAsyncOperationResult["Running"] = "Running";
    /** Succeeded */
    KnownSharedPrivateLinkResourceAsyncOperationResult["Succeeded"] = "Succeeded";
    /** Failed */
    KnownSharedPrivateLinkResourceAsyncOperationResult["Failed"] = "Failed";
})(KnownSharedPrivateLinkResourceAsyncOperationResult || (KnownSharedPrivateLinkResourceAsyncOperationResult = {}));
