// @aws-sdk/client-devops-guru/dist-cjs/index.js
"use strict";
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __name = (target, value) => __defProp(target, "name", { value, configurable: true });
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// src/index.ts
var src_exports = {};
__export(src_exports, {
  AccessDeniedException: () => AccessDeniedException,
  AddNotificationChannelCommand: () => AddNotificationChannelCommand,
  AnomalySeverity: () => AnomalySeverity,
  AnomalyStatus: () => AnomalyStatus,
  AnomalyType: () => AnomalyType,
  CloudWatchMetricDataStatusCode: () => CloudWatchMetricDataStatusCode,
  CloudWatchMetricsStat: () => CloudWatchMetricsStat,
  ConflictException: () => ConflictException,
  CostEstimationServiceResourceState: () => CostEstimationServiceResourceState,
  CostEstimationStatus: () => CostEstimationStatus,
  DeleteInsightCommand: () => DeleteInsightCommand,
  DescribeAccountHealthCommand: () => DescribeAccountHealthCommand,
  DescribeAccountOverviewCommand: () => DescribeAccountOverviewCommand,
  DescribeAnomalyCommand: () => DescribeAnomalyCommand,
  DescribeEventSourcesConfigCommand: () => DescribeEventSourcesConfigCommand,
  DescribeFeedbackCommand: () => DescribeFeedbackCommand,
  DescribeInsightCommand: () => DescribeInsightCommand,
  DescribeOrganizationHealthCommand: () => DescribeOrganizationHealthCommand,
  DescribeOrganizationOverviewCommand: () => DescribeOrganizationOverviewCommand,
  DescribeOrganizationResourceCollectionHealthCommand: () => DescribeOrganizationResourceCollectionHealthCommand,
  DescribeResourceCollectionHealthCommand: () => DescribeResourceCollectionHealthCommand,
  DescribeServiceIntegrationCommand: () => DescribeServiceIntegrationCommand,
  DevOpsGuru: () => DevOpsGuru,
  DevOpsGuruClient: () => DevOpsGuruClient,
  DevOpsGuruServiceException: () => DevOpsGuruServiceException,
  EventClass: () => EventClass,
  EventDataSource: () => EventDataSource,
  EventSourceOptInStatus: () => EventSourceOptInStatus,
  GetCostEstimationCommand: () => GetCostEstimationCommand,
  GetResourceCollectionCommand: () => GetResourceCollectionCommand,
  InsightFeedbackOption: () => InsightFeedbackOption,
  InsightSeverity: () => InsightSeverity,
  InsightStatus: () => InsightStatus,
  InsightType: () => InsightType,
  InternalServerException: () => InternalServerException,
  ListAnomaliesForInsightCommand: () => ListAnomaliesForInsightCommand,
  ListAnomalousLogGroupsCommand: () => ListAnomalousLogGroupsCommand,
  ListEventsCommand: () => ListEventsCommand,
  ListInsightsCommand: () => ListInsightsCommand,
  ListMonitoredResourcesCommand: () => ListMonitoredResourcesCommand,
  ListNotificationChannelsCommand: () => ListNotificationChannelsCommand,
  ListOrganizationInsightsCommand: () => ListOrganizationInsightsCommand,
  ListRecommendationsCommand: () => ListRecommendationsCommand,
  Locale: () => Locale,
  LogAnomalyType: () => LogAnomalyType,
  NotificationMessageType: () => NotificationMessageType,
  OptInStatus: () => OptInStatus,
  OrganizationResourceCollectionType: () => OrganizationResourceCollectionType,
  PutFeedbackCommand: () => PutFeedbackCommand,
  RemoveNotificationChannelCommand: () => RemoveNotificationChannelCommand,
  ResourceCollectionType: () => ResourceCollectionType,
  ResourceNotFoundException: () => ResourceNotFoundException,
  ResourcePermission: () => ResourcePermission,
  ResourceTypeFilter: () => ResourceTypeFilter,
  SearchInsightsCommand: () => SearchInsightsCommand,
  SearchOrganizationInsightsCommand: () => SearchOrganizationInsightsCommand,
  ServerSideEncryptionType: () => ServerSideEncryptionType,
  ServiceName: () => ServiceName,
  ServiceQuotaExceededException: () => ServiceQuotaExceededException,
  StartCostEstimationCommand: () => StartCostEstimationCommand,
  ThrottlingException: () => ThrottlingException,
  UpdateEventSourcesConfigCommand: () => UpdateEventSourcesConfigCommand,
  UpdateResourceCollectionAction: () => UpdateResourceCollectionAction,
  UpdateResourceCollectionCommand: () => UpdateResourceCollectionCommand,
  UpdateServiceIntegrationCommand: () => UpdateServiceIntegrationCommand,
  ValidationException: () => ValidationException,
  ValidationExceptionReason: () => ValidationExceptionReason,
  __Client: () => import_smithy_client.Client,
  paginateDescribeOrganizationResourceCollectionHealth: () => paginateDescribeOrganizationResourceCollectionHealth,
  paginateDescribeResourceCollectionHealth: () => paginateDescribeResourceCollectionHealth,
  paginateGetCostEstimation: () => paginateGetCostEstimation,
  paginateGetResourceCollection: () => paginateGetResourceCollection,
  paginateListAnomaliesForInsight: () => paginateListAnomaliesForInsight,
  paginateListAnomalousLogGroups: () => paginateListAnomalousLogGroups,
  paginateListEvents: () => paginateListEvents,
  paginateListInsights: () => paginateListInsights,
  paginateListMonitoredResources: () => paginateListMonitoredResources,
  paginateListNotificationChannels: () => paginateListNotificationChannels,
  paginateListOrganizationInsights: () => paginateListOrganizationInsights,
  paginateListRecommendations: () => paginateListRecommendations,
  paginateSearchInsights: () => paginateSearchInsights,
  paginateSearchOrganizationInsights: () => paginateSearchOrganizationInsights
});
module.exports = __toCommonJS(src_exports);

// src/DevOpsGuruClient.ts
var import_middleware_host_header = require("@aws-sdk/middleware-host-header");
var import_middleware_logger = require("@aws-sdk/middleware-logger");
var import_middleware_recursion_detection = require("@aws-sdk/middleware-recursion-detection");
var import_middleware_user_agent = require("@aws-sdk/middleware-user-agent");
var import_config_resolver = require("@smithy/config-resolver");
var import_core = require("@smithy/core");
var import_middleware_content_length = require("@smithy/middleware-content-length");
var import_middleware_endpoint = require("@smithy/middleware-endpoint");
var import_middleware_retry = require("@smithy/middleware-retry");

var import_httpAuthSchemeProvider = require("./auth/httpAuthSchemeProvider");

// src/endpoint/EndpointParameters.ts
var resolveClientEndpointParameters = /* @__PURE__ */ __name((options) => {
  return {
    ...options,
    useDualstackEndpoint: options.useDualstackEndpoint ?? false,
    useFipsEndpoint: options.useFipsEndpoint ?? false,
    defaultSigningName: "devops-guru"
  };
}, "resolveClientEndpointParameters");
var commonParams = {
  UseFIPS: { type: "builtInParams", name: "useFipsEndpoint" },
  Endpoint: { type: "builtInParams", name: "endpoint" },
  Region: { type: "builtInParams", name: "region" },
  UseDualStack: { type: "builtInParams", name: "useDualstackEndpoint" }
};

// src/DevOpsGuruClient.ts
var import_runtimeConfig = require("././runtimeConfig");

// src/runtimeExtensions.ts
var import_region_config_resolver = require("@aws-sdk/region-config-resolver");
var import_protocol_http = require("@smithy/protocol-http");
var import_smithy_client = require("@smithy/smithy-client");

// src/auth/httpAuthExtensionConfiguration.ts
var getHttpAuthExtensionConfiguration = /* @__PURE__ */ __name((runtimeConfig) => {
  const _httpAuthSchemes = runtimeConfig.httpAuthSchemes;
  let _httpAuthSchemeProvider = runtimeConfig.httpAuthSchemeProvider;
  let _credentials = runtimeConfig.credentials;
  return {
    setHttpAuthScheme(httpAuthScheme) {
      const index = _httpAuthSchemes.findIndex((scheme) => scheme.schemeId === httpAuthScheme.schemeId);
      if (index === -1) {
        _httpAuthSchemes.push(httpAuthScheme);
      } else {
        _httpAuthSchemes.splice(index, 1, httpAuthScheme);
      }
    },
    httpAuthSchemes() {
      return _httpAuthSchemes;
    },
    setHttpAuthSchemeProvider(httpAuthSchemeProvider) {
      _httpAuthSchemeProvider = httpAuthSchemeProvider;
    },
    httpAuthSchemeProvider() {
      return _httpAuthSchemeProvider;
    },
    setCredentials(credentials) {
      _credentials = credentials;
    },
    credentials() {
      return _credentials;
    }
  };
}, "getHttpAuthExtensionConfiguration");
var resolveHttpAuthRuntimeConfig = /* @__PURE__ */ __name((config) => {
  return {
    httpAuthSchemes: config.httpAuthSchemes(),
    httpAuthSchemeProvider: config.httpAuthSchemeProvider(),
    credentials: config.credentials()
  };
}, "resolveHttpAuthRuntimeConfig");

// src/runtimeExtensions.ts
var asPartial = /* @__PURE__ */ __name((t) => t, "asPartial");
var resolveRuntimeExtensions = /* @__PURE__ */ __name((runtimeConfig, extensions) => {
  const extensionConfiguration = {
    ...asPartial((0, import_region_config_resolver.getAwsRegionExtensionConfiguration)(runtimeConfig)),
    ...asPartial((0, import_smithy_client.getDefaultExtensionConfiguration)(runtimeConfig)),
    ...asPartial((0, import_protocol_http.getHttpHandlerExtensionConfiguration)(runtimeConfig)),
    ...asPartial(getHttpAuthExtensionConfiguration(runtimeConfig))
  };
  extensions.forEach((extension) => extension.configure(extensionConfiguration));
  return {
    ...runtimeConfig,
    ...(0, import_region_config_resolver.resolveAwsRegionExtensionConfiguration)(extensionConfiguration),
    ...(0, import_smithy_client.resolveDefaultRuntimeConfig)(extensionConfiguration),
    ...(0, import_protocol_http.resolveHttpHandlerRuntimeConfig)(extensionConfiguration),
    ...resolveHttpAuthRuntimeConfig(extensionConfiguration)
  };
}, "resolveRuntimeExtensions");

// src/DevOpsGuruClient.ts
var _DevOpsGuruClient = class _DevOpsGuruClient extends import_smithy_client.Client {
  constructor(...[configuration]) {
    const _config_0 = (0, import_runtimeConfig.getRuntimeConfig)(configuration || {});
    const _config_1 = resolveClientEndpointParameters(_config_0);
    const _config_2 = (0, import_config_resolver.resolveRegionConfig)(_config_1);
    const _config_3 = (0, import_middleware_endpoint.resolveEndpointConfig)(_config_2);
    const _config_4 = (0, import_middleware_retry.resolveRetryConfig)(_config_3);
    const _config_5 = (0, import_middleware_host_header.resolveHostHeaderConfig)(_config_4);
    const _config_6 = (0, import_middleware_user_agent.resolveUserAgentConfig)(_config_5);
    const _config_7 = (0, import_httpAuthSchemeProvider.resolveHttpAuthSchemeConfig)(_config_6);
    const _config_8 = resolveRuntimeExtensions(_config_7, (configuration == null ? void 0 : configuration.extensions) || []);
    super(_config_8);
    this.config = _config_8;
    this.middlewareStack.use((0, import_middleware_retry.getRetryPlugin)(this.config));
    this.middlewareStack.use((0, import_middleware_content_length.getContentLengthPlugin)(this.config));
    this.middlewareStack.use((0, import_middleware_host_header.getHostHeaderPlugin)(this.config));
    this.middlewareStack.use((0, import_middleware_logger.getLoggerPlugin)(this.config));
    this.middlewareStack.use((0, import_middleware_recursion_detection.getRecursionDetectionPlugin)(this.config));
    this.middlewareStack.use((0, import_middleware_user_agent.getUserAgentPlugin)(this.config));
    this.middlewareStack.use(
      (0, import_core.getHttpAuthSchemeEndpointRuleSetPlugin)(this.config, {
        httpAuthSchemeParametersProvider: this.getDefaultHttpAuthSchemeParametersProvider(),
        identityProviderConfigProvider: this.getIdentityProviderConfigProvider()
      })
    );
    this.middlewareStack.use((0, import_core.getHttpSigningPlugin)(this.config));
  }
  /**
   * Destroy underlying resources, like sockets. It's usually not necessary to do this.
   * However in Node.js, it's best to explicitly shut down the client's agent when it is no longer needed.
   * Otherwise, sockets might stay open for quite a long time before the server terminates them.
   */
  destroy() {
    super.destroy();
  }
  getDefaultHttpAuthSchemeParametersProvider() {
    return import_httpAuthSchemeProvider.defaultDevOpsGuruHttpAuthSchemeParametersProvider;
  }
  getIdentityProviderConfigProvider() {
    return async (config) => new import_core.DefaultIdentityProviderConfig({
      "aws.auth#sigv4": config.credentials
    });
  }
};
__name(_DevOpsGuruClient, "DevOpsGuruClient");
var DevOpsGuruClient = _DevOpsGuruClient;

// src/DevOpsGuru.ts


// src/commands/AddNotificationChannelCommand.ts

var import_middleware_serde = require("@smithy/middleware-serde");

var import_types = require("@smithy/types");

// src/protocols/Aws_restJson1.ts
var import_core2 = require("@aws-sdk/core");


var import_uuid = require("uuid");

// src/models/DevOpsGuruServiceException.ts

var _DevOpsGuruServiceException = class _DevOpsGuruServiceException extends import_smithy_client.ServiceException {
  /**
   * @internal
   */
  constructor(options) {
    super(options);
    Object.setPrototypeOf(this, _DevOpsGuruServiceException.prototype);
  }
};
__name(_DevOpsGuruServiceException, "DevOpsGuruServiceException");
var DevOpsGuruServiceException = _DevOpsGuruServiceException;

// src/models/models_0.ts
var _AccessDeniedException = class _AccessDeniedException extends DevOpsGuruServiceException {
  /**
   * @internal
   */
  constructor(opts) {
    super({
      name: "AccessDeniedException",
      $fault: "client",
      ...opts
    });
    this.name = "AccessDeniedException";
    this.$fault = "client";
    Object.setPrototypeOf(this, _AccessDeniedException.prototype);
    this.Message = opts.Message;
  }
};
__name(_AccessDeniedException, "AccessDeniedException");
var AccessDeniedException = _AccessDeniedException;
var NotificationMessageType = {
  CLOSED_INSIGHT: "CLOSED_INSIGHT",
  NEW_ASSOCIATION: "NEW_ASSOCIATION",
  NEW_INSIGHT: "NEW_INSIGHT",
  NEW_RECOMMENDATION: "NEW_RECOMMENDATION",
  SEVERITY_UPGRADED: "SEVERITY_UPGRADED"
};
var InsightSeverity = {
  HIGH: "HIGH",
  LOW: "LOW",
  MEDIUM: "MEDIUM"
};
var _ConflictException = class _ConflictException extends DevOpsGuruServiceException {
  /**
   * @internal
   */
  constructor(opts) {
    super({
      name: "ConflictException",
      $fault: "client",
      ...opts
    });
    this.name = "ConflictException";
    this.$fault = "client";
    Object.setPrototypeOf(this, _ConflictException.prototype);
    this.Message = opts.Message;
    this.ResourceId = opts.ResourceId;
    this.ResourceType = opts.ResourceType;
  }
};
__name(_ConflictException, "ConflictException");
var ConflictException = _ConflictException;
var _InternalServerException = class _InternalServerException extends DevOpsGuruServiceException {
  /**
   * @internal
   */
  constructor(opts) {
    super({
      name: "InternalServerException",
      $fault: "server",
      ...opts
    });
    this.name = "InternalServerException";
    this.$fault = "server";
    Object.setPrototypeOf(this, _InternalServerException.prototype);
    this.Message = opts.Message;
    this.RetryAfterSeconds = opts.RetryAfterSeconds;
  }
};
__name(_InternalServerException, "InternalServerException");
var InternalServerException = _InternalServerException;
var _ResourceNotFoundException = class _ResourceNotFoundException extends DevOpsGuruServiceException {
  /**
   * @internal
   */
  constructor(opts) {
    super({
      name: "ResourceNotFoundException",
      $fault: "client",
      ...opts
    });
    this.name = "ResourceNotFoundException";
    this.$fault = "client";
    Object.setPrototypeOf(this, _ResourceNotFoundException.prototype);
    this.Message = opts.Message;
    this.ResourceId = opts.ResourceId;
    this.ResourceType = opts.ResourceType;
  }
};
__name(_ResourceNotFoundException, "ResourceNotFoundException");
var ResourceNotFoundException = _ResourceNotFoundException;
var _ServiceQuotaExceededException = class _ServiceQuotaExceededException extends DevOpsGuruServiceException {
  /**
   * @internal
   */
  constructor(opts) {
    super({
      name: "ServiceQuotaExceededException",
      $fault: "client",
      ...opts
    });
    this.name = "ServiceQuotaExceededException";
    this.$fault = "client";
    Object.setPrototypeOf(this, _ServiceQuotaExceededException.prototype);
    this.Message = opts.Message;
  }
};
__name(_ServiceQuotaExceededException, "ServiceQuotaExceededException");
var ServiceQuotaExceededException = _ServiceQuotaExceededException;
var _ThrottlingException = class _ThrottlingException extends DevOpsGuruServiceException {
  /**
   * @internal
   */
  constructor(opts) {
    super({
      name: "ThrottlingException",
      $fault: "client",
      ...opts
    });
    this.name = "ThrottlingException";
    this.$fault = "client";
    Object.setPrototypeOf(this, _ThrottlingException.prototype);
    this.Message = opts.Message;
    this.QuotaCode = opts.QuotaCode;
    this.ServiceCode = opts.ServiceCode;
    this.RetryAfterSeconds = opts.RetryAfterSeconds;
  }
};
__name(_ThrottlingException, "ThrottlingException");
var ThrottlingException = _ThrottlingException;
var ValidationExceptionReason = {
  CANNOT_PARSE: "CANNOT_PARSE",
  FIELD_VALIDATION_FAILED: "FIELD_VALIDATION_FAILED",
  INVALID_PARAMETER_COMBINATION: "INVALID_PARAMETER_COMBINATION",
  OTHER: "OTHER",
  PARAMETER_INCONSISTENT_WITH_SERVICE_STATE: "PARAMETER_INCONSISTENT_WITH_SERVICE_STATE",
  UNKNOWN_OPERATION: "UNKNOWN_OPERATION"
};
var _ValidationException = class _ValidationException extends DevOpsGuruServiceException {
  /**
   * @internal
   */
  constructor(opts) {
    super({
      name: "ValidationException",
      $fault: "client",
      ...opts
    });
    this.name = "ValidationException";
    this.$fault = "client";
    Object.setPrototypeOf(this, _ValidationException.prototype);
    this.Message = opts.Message;
    this.Reason = opts.Reason;
    this.Fields = opts.Fields;
  }
};
__name(_ValidationException, "ValidationException");
var ValidationException = _ValidationException;
var EventSourceOptInStatus = {
  DISABLED: "DISABLED",
  ENABLED: "ENABLED"
};
var LogAnomalyType = {
  BLOCK_FORMAT: "BLOCK_FORMAT",
  FORMAT: "FORMAT",
  HTTP_CODE: "HTTP_CODE",
  KEYWORD: "KEYWORD",
  KEYWORD_TOKEN: "KEYWORD_TOKEN",
  NEW_FIELD_NAME: "NEW_FIELD_NAME",
  NUMERICAL_NAN: "NUMERICAL_NAN",
  NUMERICAL_POINT: "NUMERICAL_POINT"
};
var AnomalySeverity = {
  HIGH: "HIGH",
  LOW: "LOW",
  MEDIUM: "MEDIUM"
};
var CloudWatchMetricDataStatusCode = {
  COMPLETE: "Complete",
  INTERNAL_ERROR: "InternalError",
  PARTIAL_DATA: "PartialData"
};
var CloudWatchMetricsStat = {
  AVERAGE: "Average",
  MAXIMUM: "Maximum",
  MINIMUM: "Minimum",
  P50: "p50",
  P90: "p90",
  P99: "p99",
  SAMPLE_COUNT: "SampleCount",
  SUM: "Sum"
};
var AnomalyStatus = {
  CLOSED: "CLOSED",
  ONGOING: "ONGOING"
};
var AnomalyType = {
  CAUSAL: "CAUSAL",
  CONTEXTUAL: "CONTEXTUAL"
};
var InsightFeedbackOption = {
  ALERT_TOO_SENSITIVE: "ALERT_TOO_SENSITIVE",
  DATA_INCORRECT: "DATA_INCORRECT",
  DATA_NOISY_ANOMALY: "DATA_NOISY_ANOMALY",
  RECOMMENDATION_USEFUL: "RECOMMENDATION_USEFUL",
  VALID_COLLECTION: "VALID_COLLECTION"
};
var InsightStatus = {
  CLOSED: "CLOSED",
  ONGOING: "ONGOING"
};
var OrganizationResourceCollectionType = {
  AWS_ACCOUNT: "AWS_ACCOUNT",
  AWS_CLOUD_FORMATION: "AWS_CLOUD_FORMATION",
  AWS_SERVICE: "AWS_SERVICE",
  AWS_TAGS: "AWS_TAGS"
};
var ServiceName = {
  API_GATEWAY: "API_GATEWAY",
  APPLICATION_ELB: "APPLICATION_ELB",
  AUTO_SCALING_GROUP: "AUTO_SCALING_GROUP",
  CLOUD_FRONT: "CLOUD_FRONT",
  DYNAMO_DB: "DYNAMO_DB",
  EC2: "EC2",
  ECS: "ECS",
  EKS: "EKS",
  ELASTIC_BEANSTALK: "ELASTIC_BEANSTALK",
  ELASTI_CACHE: "ELASTI_CACHE",
  ELB: "ELB",
  ES: "ES",
  KINESIS: "KINESIS",
  LAMBDA: "LAMBDA",
  NAT_GATEWAY: "NAT_GATEWAY",
  NETWORK_ELB: "NETWORK_ELB",
  RDS: "RDS",
  REDSHIFT: "REDSHIFT",
  ROUTE_53: "ROUTE_53",
  S3: "S3",
  SAGE_MAKER: "SAGE_MAKER",
  SNS: "SNS",
  SQS: "SQS",
  STEP_FUNCTIONS: "STEP_FUNCTIONS",
  SWF: "SWF"
};
var ResourceCollectionType = {
  AWS_CLOUD_FORMATION: "AWS_CLOUD_FORMATION",
  AWS_SERVICE: "AWS_SERVICE",
  AWS_TAGS: "AWS_TAGS"
};
var OptInStatus = {
  DISABLED: "DISABLED",
  ENABLED: "ENABLED"
};
var ServerSideEncryptionType = {
  AWS_OWNED_KMS_KEY: "AWS_OWNED_KMS_KEY",
  CUSTOMER_MANAGED_KEY: "CUSTOMER_MANAGED_KEY"
};
var CostEstimationServiceResourceState = {
  ACTIVE: "ACTIVE",
  INACTIVE: "INACTIVE"
};
var CostEstimationStatus = {
  COMPLETED: "COMPLETED",
  ONGOING: "ONGOING"
};
var EventDataSource = {
  AWS_CLOUD_TRAIL: "AWS_CLOUD_TRAIL",
  AWS_CODE_DEPLOY: "AWS_CODE_DEPLOY"
};
var EventClass = {
  CONFIG_CHANGE: "CONFIG_CHANGE",
  DEPLOYMENT: "DEPLOYMENT",
  INFRASTRUCTURE: "INFRASTRUCTURE",
  SCHEMA_CHANGE: "SCHEMA_CHANGE",
  SECURITY_CHANGE: "SECURITY_CHANGE"
};
var InsightType = {
  PROACTIVE: "PROACTIVE",
  REACTIVE: "REACTIVE"
};
var ResourcePermission = {
  FULL_PERMISSION: "FULL_PERMISSION",
  MISSING_PERMISSION: "MISSING_PERMISSION"
};
var ResourceTypeFilter = {
  CLOUDFRONT_DISTRIBUTION: "CLOUDFRONT_DISTRIBUTION",
  DYNAMODB_TABLE: "DYNAMODB_TABLE",
  EC2_NAT_GATEWAY: "EC2_NAT_GATEWAY",
  ECS_CLUSTER: "ECS_CLUSTER",
  ECS_SERVICE: "ECS_SERVICE",
  EKS_CLUSTER: "EKS_CLUSTER",
  ELASTICACHE_CACHE_CLUSTER: "ELASTICACHE_CACHE_CLUSTER",
  ELASTICSEARCH_DOMAIN: "ELASTICSEARCH_DOMAIN",
  ELASTIC_BEANSTALK_ENVIRONMENT: "ELASTIC_BEANSTALK_ENVIRONMENT",
  ELASTIC_LOAD_BALANCER_LOAD_BALANCER: "ELASTIC_LOAD_BALANCER_LOAD_BALANCER",
  ELASTIC_LOAD_BALANCING_V2_LOAD_BALANCER: "ELASTIC_LOAD_BALANCING_V2_LOAD_BALANCER",
  ELASTIC_LOAD_BALANCING_V2_TARGET_GROUP: "ELASTIC_LOAD_BALANCING_V2_TARGET_GROUP",
  KINESIS_STREAM: "KINESIS_STREAM",
  LAMBDA_FUNCTION: "LAMBDA_FUNCTION",
  LOG_GROUPS: "LOG_GROUPS",
  OPEN_SEARCH_SERVICE_DOMAIN: "OPEN_SEARCH_SERVICE_DOMAIN",
  RDS_DB_CLUSTER: "RDS_DB_CLUSTER",
  RDS_DB_INSTANCE: "RDS_DB_INSTANCE",
  REDSHIFT_CLUSTER: "REDSHIFT_CLUSTER",
  ROUTE53_HEALTH_CHECK: "ROUTE53_HEALTH_CHECK",
  ROUTE53_HOSTED_ZONE: "ROUTE53_HOSTED_ZONE",
  S3_BUCKET: "S3_BUCKET",
  SAGEMAKER_ENDPOINT: "SAGEMAKER_ENDPOINT",
  SNS_TOPIC: "SNS_TOPIC",
  SQS_QUEUE: "SQS_QUEUE",
  STEP_FUNCTIONS_ACTIVITY: "STEP_FUNCTIONS_ACTIVITY",
  STEP_FUNCTIONS_STATE_MACHINE: "STEP_FUNCTIONS_STATE_MACHINE"
};
var Locale = {
  DE_DE: "DE_DE",
  EN_GB: "EN_GB",
  EN_US: "EN_US",
  ES_ES: "ES_ES",
  FR_FR: "FR_FR",
  IT_IT: "IT_IT",
  JA_JP: "JA_JP",
  KO_KR: "KO_KR",
  PT_BR: "PT_BR",
  ZH_CN: "ZH_CN",
  ZH_TW: "ZH_TW"
};
var UpdateResourceCollectionAction = {
  ADD: "ADD",
  REMOVE: "REMOVE"
};

// src/protocols/Aws_restJson1.ts
var se_AddNotificationChannelCommand = /* @__PURE__ */ __name(async (input, context) => {
  const b = (0, import_core.requestBuilder)(input, context);
  const headers = {
    "content-type": "application/json"
  };
  b.bp("/channels");
  let body;
  body = JSON.stringify(
    (0, import_smithy_client.take)(input, {
      Config: (_) => (0, import_smithy_client._json)(_)
    })
  );
  b.m("PUT").h(headers).b(body);
  return b.build();
}, "se_AddNotificationChannelCommand");
var se_DeleteInsightCommand = /* @__PURE__ */ __name(async (input, context) => {
  const b = (0, import_core.requestBuilder)(input, context);
  const headers = {};
  b.bp("/insights/{Id}");
  b.p("Id", () => input.Id, "{Id}", false);
  let body;
  b.m("DELETE").h(headers).b(body);
  return b.build();
}, "se_DeleteInsightCommand");
var se_DescribeAccountHealthCommand = /* @__PURE__ */ __name(async (input, context) => {
  const b = (0, import_core.requestBuilder)(input, context);
  const headers = {
    "content-type": "application/json"
  };
  b.bp("/accounts/health");
  let body;
  body = "";
  b.m("GET").h(headers).b(body);
  return b.build();
}, "se_DescribeAccountHealthCommand");
var se_DescribeAccountOverviewCommand = /* @__PURE__ */ __name(async (input, context) => {
  const b = (0, import_core.requestBuilder)(input, context);
  const headers = {
    "content-type": "application/json"
  };
  b.bp("/accounts/overview");
  let body;
  body = JSON.stringify(
    (0, import_smithy_client.take)(input, {
      FromTime: (_) => Math.round(_.getTime() / 1e3),
      ToTime: (_) => Math.round(_.getTime() / 1e3)
    })
  );
  b.m("POST").h(headers).b(body);
  return b.build();
}, "se_DescribeAccountOverviewCommand");
var se_DescribeAnomalyCommand = /* @__PURE__ */ __name(async (input, context) => {
  const b = (0, import_core.requestBuilder)(input, context);
  const headers = {};
  b.bp("/anomalies/{Id}");
  b.p("Id", () => input.Id, "{Id}", false);
  const query = (0, import_smithy_client.map)({
    [_AI]: [, input[_AI]]
  });
  let body;
  b.m("GET").h(headers).q(query).b(body);
  return b.build();
}, "se_DescribeAnomalyCommand");
var se_DescribeEventSourcesConfigCommand = /* @__PURE__ */ __name(async (input, context) => {
  const b = (0, import_core.requestBuilder)(input, context);
  const headers = {
    "content-type": "application/json"
  };
  b.bp("/event-sources");
  let body;
  body = "";
  b.m("POST").h(headers).b(body);
  return b.build();
}, "se_DescribeEventSourcesConfigCommand");
var se_DescribeFeedbackCommand = /* @__PURE__ */ __name(async (input, context) => {
  const b = (0, import_core.requestBuilder)(input, context);
  const headers = {
    "content-type": "application/json"
  };
  b.bp("/feedback");
  let body;
  body = JSON.stringify(
    (0, import_smithy_client.take)(input, {
      InsightId: []
    })
  );
  b.m("POST").h(headers).b(body);
  return b.build();
}, "se_DescribeFeedbackCommand");
var se_DescribeInsightCommand = /* @__PURE__ */ __name(async (input, context) => {
  const b = (0, import_core.requestBuilder)(input, context);
  const headers = {};
  b.bp("/insights/{Id}");
  b.p("Id", () => input.Id, "{Id}", false);
  const query = (0, import_smithy_client.map)({
    [_AI]: [, input[_AI]]
  });
  let body;
  b.m("GET").h(headers).q(query).b(body);
  return b.build();
}, "se_DescribeInsightCommand");
var se_DescribeOrganizationHealthCommand = /* @__PURE__ */ __name(async (input, context) => {
  const b = (0, import_core.requestBuilder)(input, context);
  const headers = {
    "content-type": "application/json"
  };
  b.bp("/organization/health");
  let body;
  body = JSON.stringify(
    (0, import_smithy_client.take)(input, {
      AccountIds: (_) => (0, import_smithy_client._json)(_),
      OrganizationalUnitIds: (_) => (0, import_smithy_client._json)(_)
    })
  );
  b.m("POST").h(headers).b(body);
  return b.build();
}, "se_DescribeOrganizationHealthCommand");
var se_DescribeOrganizationOverviewCommand = /* @__PURE__ */ __name(async (input, context) => {
  const b = (0, import_core.requestBuilder)(input, context);
  const headers = {
    "content-type": "application/json"
  };
  b.bp("/organization/overview");
  let body;
  body = JSON.stringify(
    (0, import_smithy_client.take)(input, {
      AccountIds: (_) => (0, import_smithy_client._json)(_),
      FromTime: (_) => Math.round(_.getTime() / 1e3),
      OrganizationalUnitIds: (_) => (0, import_smithy_client._json)(_),
      ToTime: (_) => Math.round(_.getTime() / 1e3)
    })
  );
  b.m("POST").h(headers).b(body);
  return b.build();
}, "se_DescribeOrganizationOverviewCommand");
var se_DescribeOrganizationResourceCollectionHealthCommand = /* @__PURE__ */ __name(async (input, context) => {
  const b = (0, import_core.requestBuilder)(input, context);
  const headers = {
    "content-type": "application/json"
  };
  b.bp("/organization/health/resource-collection");
  let body;
  body = JSON.stringify(
    (0, import_smithy_client.take)(input, {
      AccountIds: (_) => (0, import_smithy_client._json)(_),
      MaxResults: [],
      NextToken: [],
      OrganizationResourceCollectionType: [],
      OrganizationalUnitIds: (_) => (0, import_smithy_client._json)(_)
    })
  );
  b.m("POST").h(headers).b(body);
  return b.build();
}, "se_DescribeOrganizationResourceCollectionHealthCommand");
var se_DescribeResourceCollectionHealthCommand = /* @__PURE__ */ __name(async (input, context) => {
  const b = (0, import_core.requestBuilder)(input, context);
  const headers = {};
  b.bp("/accounts/health/resource-collection/{ResourceCollectionType}");
  b.p("ResourceCollectionType", () => input.ResourceCollectionType, "{ResourceCollectionType}", false);
  const query = (0, import_smithy_client.map)({
    [_NT]: [, input[_NT]]
  });
  let body;
  b.m("GET").h(headers).q(query).b(body);
  return b.build();
}, "se_DescribeResourceCollectionHealthCommand");
var se_DescribeServiceIntegrationCommand = /* @__PURE__ */ __name(async (input, context) => {
  const b = (0, import_core.requestBuilder)(input, context);
  const headers = {
    "content-type": "application/json"
  };
  b.bp("/service-integrations");
  let body;
  body = "";
  b.m("GET").h(headers).b(body);
  return b.build();
}, "se_DescribeServiceIntegrationCommand");
var se_GetCostEstimationCommand = /* @__PURE__ */ __name(async (input, context) => {
  const b = (0, import_core.requestBuilder)(input, context);
  const headers = {};
  b.bp("/cost-estimation");
  const query = (0, import_smithy_client.map)({
    [_NT]: [, input[_NT]]
  });
  let body;
  b.m("GET").h(headers).q(query).b(body);
  return b.build();
}, "se_GetCostEstimationCommand");
var se_GetResourceCollectionCommand = /* @__PURE__ */ __name(async (input, context) => {
  const b = (0, import_core.requestBuilder)(input, context);
  const headers = {};
  b.bp("/resource-collections/{ResourceCollectionType}");
  b.p("ResourceCollectionType", () => input.ResourceCollectionType, "{ResourceCollectionType}", false);
  const query = (0, import_smithy_client.map)({
    [_NT]: [, input[_NT]]
  });
  let body;
  b.m("GET").h(headers).q(query).b(body);
  return b.build();
}, "se_GetResourceCollectionCommand");
var se_ListAnomaliesForInsightCommand = /* @__PURE__ */ __name(async (input, context) => {
  const b = (0, import_core.requestBuilder)(input, context);
  const headers = {
    "content-type": "application/json"
  };
  b.bp("/anomalies/insight/{InsightId}");
  b.p("InsightId", () => input.InsightId, "{InsightId}", false);
  let body;
  body = JSON.stringify(
    (0, import_smithy_client.take)(input, {
      AccountId: [],
      Filters: (_) => (0, import_smithy_client._json)(_),
      MaxResults: [],
      NextToken: [],
      StartTimeRange: (_) => se_StartTimeRange(_, context)
    })
  );
  b.m("POST").h(headers).b(body);
  return b.build();
}, "se_ListAnomaliesForInsightCommand");
var se_ListAnomalousLogGroupsCommand = /* @__PURE__ */ __name(async (input, context) => {
  const b = (0, import_core.requestBuilder)(input, context);
  const headers = {
    "content-type": "application/json"
  };
  b.bp("/list-log-anomalies");
  let body;
  body = JSON.stringify(
    (0, import_smithy_client.take)(input, {
      InsightId: [],
      MaxResults: [],
      NextToken: []
    })
  );
  b.m("POST").h(headers).b(body);
  return b.build();
}, "se_ListAnomalousLogGroupsCommand");
var se_ListEventsCommand = /* @__PURE__ */ __name(async (input, context) => {
  const b = (0, import_core.requestBuilder)(input, context);
  const headers = {
    "content-type": "application/json"
  };
  b.bp("/events");
  let body;
  body = JSON.stringify(
    (0, import_smithy_client.take)(input, {
      AccountId: [],
      Filters: (_) => se_ListEventsFilters(_, context),
      MaxResults: [],
      NextToken: []
    })
  );
  b.m("POST").h(headers).b(body);
  return b.build();
}, "se_ListEventsCommand");
var se_ListInsightsCommand = /* @__PURE__ */ __name(async (input, context) => {
  const b = (0, import_core.requestBuilder)(input, context);
  const headers = {
    "content-type": "application/json"
  };
  b.bp("/insights");
  let body;
  body = JSON.stringify(
    (0, import_smithy_client.take)(input, {
      MaxResults: [],
      NextToken: [],
      StatusFilter: (_) => se_ListInsightsStatusFilter(_, context)
    })
  );
  b.m("POST").h(headers).b(body);
  return b.build();
}, "se_ListInsightsCommand");
var se_ListMonitoredResourcesCommand = /* @__PURE__ */ __name(async (input, context) => {
  const b = (0, import_core.requestBuilder)(input, context);
  const headers = {
    "content-type": "application/json"
  };
  b.bp("/monitoredResources");
  let body;
  body = JSON.stringify(
    (0, import_smithy_client.take)(input, {
      Filters: (_) => (0, import_smithy_client._json)(_),
      MaxResults: [],
      NextToken: []
    })
  );
  b.m("POST").h(headers).b(body);
  return b.build();
}, "se_ListMonitoredResourcesCommand");
var se_ListNotificationChannelsCommand = /* @__PURE__ */ __name(async (input, context) => {
  const b = (0, import_core.requestBuilder)(input, context);
  const headers = {
    "content-type": "application/json"
  };
  b.bp("/channels");
  let body;
  body = JSON.stringify(
    (0, import_smithy_client.take)(input, {
      NextToken: []
    })
  );
  b.m("POST").h(headers).b(body);
  return b.build();
}, "se_ListNotificationChannelsCommand");
var se_ListOrganizationInsightsCommand = /* @__PURE__ */ __name(async (input, context) => {
  const b = (0, import_core.requestBuilder)(input, context);
  const headers = {
    "content-type": "application/json"
  };
  b.bp("/organization/insights");
  let body;
  body = JSON.stringify(
    (0, import_smithy_client.take)(input, {
      AccountIds: (_) => (0, import_smithy_client._json)(_),
      MaxResults: [],
      NextToken: [],
      OrganizationalUnitIds: (_) => (0, import_smithy_client._json)(_),
      StatusFilter: (_) => se_ListInsightsStatusFilter(_, context)
    })
  );
  b.m("POST").h(headers).b(body);
  return b.build();
}, "se_ListOrganizationInsightsCommand");
var se_ListRecommendationsCommand = /* @__PURE__ */ __name(async (input, context) => {
  const b = (0, import_core.requestBuilder)(input, context);
  const headers = {
    "content-type": "application/json"
  };
  b.bp("/recommendations");
  let body;
  body = JSON.stringify(
    (0, import_smithy_client.take)(input, {
      AccountId: [],
      InsightId: [],
      Locale: [],
      NextToken: []
    })
  );
  b.m("POST").h(headers).b(body);
  return b.build();
}, "se_ListRecommendationsCommand");
var se_PutFeedbackCommand = /* @__PURE__ */ __name(async (input, context) => {
  const b = (0, import_core.requestBuilder)(input, context);
  const headers = {
    "content-type": "application/json"
  };
  b.bp("/feedback");
  let body;
  body = JSON.stringify(
    (0, import_smithy_client.take)(input, {
      InsightFeedback: (_) => (0, import_smithy_client._json)(_)
    })
  );
  b.m("PUT").h(headers).b(body);
  return b.build();
}, "se_PutFeedbackCommand");
var se_RemoveNotificationChannelCommand = /* @__PURE__ */ __name(async (input, context) => {
  const b = (0, import_core.requestBuilder)(input, context);
  const headers = {};
  b.bp("/channels/{Id}");
  b.p("Id", () => input.Id, "{Id}", false);
  let body;
  b.m("DELETE").h(headers).b(body);
  return b.build();
}, "se_RemoveNotificationChannelCommand");
var se_SearchInsightsCommand = /* @__PURE__ */ __name(async (input, context) => {
  const b = (0, import_core.requestBuilder)(input, context);
  const headers = {
    "content-type": "application/json"
  };
  b.bp("/insights/search");
  let body;
  body = JSON.stringify(
    (0, import_smithy_client.take)(input, {
      Filters: (_) => (0, import_smithy_client._json)(_),
      MaxResults: [],
      NextToken: [],
      StartTimeRange: (_) => se_StartTimeRange(_, context),
      Type: []
    })
  );
  b.m("POST").h(headers).b(body);
  return b.build();
}, "se_SearchInsightsCommand");
var se_SearchOrganizationInsightsCommand = /* @__PURE__ */ __name(async (input, context) => {
  const b = (0, import_core.requestBuilder)(input, context);
  const headers = {
    "content-type": "application/json"
  };
  b.bp("/organization/insights/search");
  let body;
  body = JSON.stringify(
    (0, import_smithy_client.take)(input, {
      AccountIds: (_) => (0, import_smithy_client._json)(_),
      Filters: (_) => (0, import_smithy_client._json)(_),
      MaxResults: [],
      NextToken: [],
      StartTimeRange: (_) => se_StartTimeRange(_, context),
      Type: []
    })
  );
  b.m("POST").h(headers).b(body);
  return b.build();
}, "se_SearchOrganizationInsightsCommand");
var se_StartCostEstimationCommand = /* @__PURE__ */ __name(async (input, context) => {
  const b = (0, import_core.requestBuilder)(input, context);
  const headers = {
    "content-type": "application/json"
  };
  b.bp("/cost-estimation");
  let body;
  body = JSON.stringify(
    (0, import_smithy_client.take)(input, {
      ClientToken: [true, (_) => _ ?? (0, import_uuid.v4)()],
      ResourceCollection: (_) => (0, import_smithy_client._json)(_)
    })
  );
  b.m("PUT").h(headers).b(body);
  return b.build();
}, "se_StartCostEstimationCommand");
var se_UpdateEventSourcesConfigCommand = /* @__PURE__ */ __name(async (input, context) => {
  const b = (0, import_core.requestBuilder)(input, context);
  const headers = {
    "content-type": "application/json"
  };
  b.bp("/event-sources");
  let body;
  body = JSON.stringify(
    (0, import_smithy_client.take)(input, {
      EventSources: (_) => (0, import_smithy_client._json)(_)
    })
  );
  b.m("PUT").h(headers).b(body);
  return b.build();
}, "se_UpdateEventSourcesConfigCommand");
var se_UpdateResourceCollectionCommand = /* @__PURE__ */ __name(async (input, context) => {
  const b = (0, import_core.requestBuilder)(input, context);
  const headers = {
    "content-type": "application/json"
  };
  b.bp("/resource-collections");
  let body;
  body = JSON.stringify(
    (0, import_smithy_client.take)(input, {
      Action: [],
      ResourceCollection: (_) => (0, import_smithy_client._json)(_)
    })
  );
  b.m("PUT").h(headers).b(body);
  return b.build();
}, "se_UpdateResourceCollectionCommand");
var se_UpdateServiceIntegrationCommand = /* @__PURE__ */ __name(async (input, context) => {
  const b = (0, import_core.requestBuilder)(input, context);
  const headers = {
    "content-type": "application/json"
  };
  b.bp("/service-integrations");
  let body;
  body = JSON.stringify(
    (0, import_smithy_client.take)(input, {
      ServiceIntegration: (_) => (0, import_smithy_client._json)(_)
    })
  );
  b.m("PUT").h(headers).b(body);
  return b.build();
}, "se_UpdateServiceIntegrationCommand");
var de_AddNotificationChannelCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode !== 200 && output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const contents = (0, import_smithy_client.map)({
    $metadata: deserializeMetadata(output)
  });
  const data = (0, import_smithy_client.expectNonNull)((0, import_smithy_client.expectObject)(await (0, import_core2.parseJsonBody)(output.body, context)), "body");
  const doc = (0, import_smithy_client.take)(data, {
    Id: import_smithy_client.expectString
  });
  Object.assign(contents, doc);
  return contents;
}, "de_AddNotificationChannelCommand");
var de_DeleteInsightCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode !== 200 && output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const contents = (0, import_smithy_client.map)({
    $metadata: deserializeMetadata(output)
  });
  await (0, import_smithy_client.collectBody)(output.body, context);
  return contents;
}, "de_DeleteInsightCommand");
var de_DescribeAccountHealthCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode !== 200 && output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const contents = (0, import_smithy_client.map)({
    $metadata: deserializeMetadata(output)
  });
  const data = (0, import_smithy_client.expectNonNull)((0, import_smithy_client.expectObject)(await (0, import_core2.parseJsonBody)(output.body, context)), "body");
  const doc = (0, import_smithy_client.take)(data, {
    AnalyzedResourceCount: import_smithy_client.expectLong,
    MetricsAnalyzed: import_smithy_client.expectInt32,
    OpenProactiveInsights: import_smithy_client.expectInt32,
    OpenReactiveInsights: import_smithy_client.expectInt32,
    ResourceHours: import_smithy_client.expectLong
  });
  Object.assign(contents, doc);
  return contents;
}, "de_DescribeAccountHealthCommand");
var de_DescribeAccountOverviewCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode !== 200 && output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const contents = (0, import_smithy_client.map)({
    $metadata: deserializeMetadata(output)
  });
  const data = (0, import_smithy_client.expectNonNull)((0, import_smithy_client.expectObject)(await (0, import_core2.parseJsonBody)(output.body, context)), "body");
  const doc = (0, import_smithy_client.take)(data, {
    MeanTimeToRecoverInMilliseconds: import_smithy_client.expectLong,
    ProactiveInsights: import_smithy_client.expectInt32,
    ReactiveInsights: import_smithy_client.expectInt32
  });
  Object.assign(contents, doc);
  return contents;
}, "de_DescribeAccountOverviewCommand");
var de_DescribeAnomalyCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode !== 200 && output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const contents = (0, import_smithy_client.map)({
    $metadata: deserializeMetadata(output)
  });
  const data = (0, import_smithy_client.expectNonNull)((0, import_smithy_client.expectObject)(await (0, import_core2.parseJsonBody)(output.body, context)), "body");
  const doc = (0, import_smithy_client.take)(data, {
    ProactiveAnomaly: (_) => de_ProactiveAnomaly(_, context),
    ReactiveAnomaly: (_) => de_ReactiveAnomaly(_, context)
  });
  Object.assign(contents, doc);
  return contents;
}, "de_DescribeAnomalyCommand");
var de_DescribeEventSourcesConfigCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode !== 200 && output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const contents = (0, import_smithy_client.map)({
    $metadata: deserializeMetadata(output)
  });
  const data = (0, import_smithy_client.expectNonNull)((0, import_smithy_client.expectObject)(await (0, import_core2.parseJsonBody)(output.body, context)), "body");
  const doc = (0, import_smithy_client.take)(data, {
    EventSources: import_smithy_client._json
  });
  Object.assign(contents, doc);
  return contents;
}, "de_DescribeEventSourcesConfigCommand");
var de_DescribeFeedbackCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode !== 200 && output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const contents = (0, import_smithy_client.map)({
    $metadata: deserializeMetadata(output)
  });
  const data = (0, import_smithy_client.expectNonNull)((0, import_smithy_client.expectObject)(await (0, import_core2.parseJsonBody)(output.body, context)), "body");
  const doc = (0, import_smithy_client.take)(data, {
    InsightFeedback: import_smithy_client._json
  });
  Object.assign(contents, doc);
  return contents;
}, "de_DescribeFeedbackCommand");
var de_DescribeInsightCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode !== 200 && output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const contents = (0, import_smithy_client.map)({
    $metadata: deserializeMetadata(output)
  });
  const data = (0, import_smithy_client.expectNonNull)((0, import_smithy_client.expectObject)(await (0, import_core2.parseJsonBody)(output.body, context)), "body");
  const doc = (0, import_smithy_client.take)(data, {
    ProactiveInsight: (_) => de_ProactiveInsight(_, context),
    ReactiveInsight: (_) => de_ReactiveInsight(_, context)
  });
  Object.assign(contents, doc);
  return contents;
}, "de_DescribeInsightCommand");
var de_DescribeOrganizationHealthCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode !== 200 && output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const contents = (0, import_smithy_client.map)({
    $metadata: deserializeMetadata(output)
  });
  const data = (0, import_smithy_client.expectNonNull)((0, import_smithy_client.expectObject)(await (0, import_core2.parseJsonBody)(output.body, context)), "body");
  const doc = (0, import_smithy_client.take)(data, {
    MetricsAnalyzed: import_smithy_client.expectInt32,
    OpenProactiveInsights: import_smithy_client.expectInt32,
    OpenReactiveInsights: import_smithy_client.expectInt32,
    ResourceHours: import_smithy_client.expectLong
  });
  Object.assign(contents, doc);
  return contents;
}, "de_DescribeOrganizationHealthCommand");
var de_DescribeOrganizationOverviewCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode !== 200 && output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const contents = (0, import_smithy_client.map)({
    $metadata: deserializeMetadata(output)
  });
  const data = (0, import_smithy_client.expectNonNull)((0, import_smithy_client.expectObject)(await (0, import_core2.parseJsonBody)(output.body, context)), "body");
  const doc = (0, import_smithy_client.take)(data, {
    ProactiveInsights: import_smithy_client.expectInt32,
    ReactiveInsights: import_smithy_client.expectInt32
  });
  Object.assign(contents, doc);
  return contents;
}, "de_DescribeOrganizationOverviewCommand");
var de_DescribeOrganizationResourceCollectionHealthCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode !== 200 && output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const contents = (0, import_smithy_client.map)({
    $metadata: deserializeMetadata(output)
  });
  const data = (0, import_smithy_client.expectNonNull)((0, import_smithy_client.expectObject)(await (0, import_core2.parseJsonBody)(output.body, context)), "body");
  const doc = (0, import_smithy_client.take)(data, {
    Account: import_smithy_client._json,
    CloudFormation: import_smithy_client._json,
    NextToken: import_smithy_client.expectString,
    Service: import_smithy_client._json,
    Tags: import_smithy_client._json
  });
  Object.assign(contents, doc);
  return contents;
}, "de_DescribeOrganizationResourceCollectionHealthCommand");
var de_DescribeResourceCollectionHealthCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode !== 200 && output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const contents = (0, import_smithy_client.map)({
    $metadata: deserializeMetadata(output)
  });
  const data = (0, import_smithy_client.expectNonNull)((0, import_smithy_client.expectObject)(await (0, import_core2.parseJsonBody)(output.body, context)), "body");
  const doc = (0, import_smithy_client.take)(data, {
    CloudFormation: import_smithy_client._json,
    NextToken: import_smithy_client.expectString,
    Service: import_smithy_client._json,
    Tags: import_smithy_client._json
  });
  Object.assign(contents, doc);
  return contents;
}, "de_DescribeResourceCollectionHealthCommand");
var de_DescribeServiceIntegrationCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode !== 200 && output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const contents = (0, import_smithy_client.map)({
    $metadata: deserializeMetadata(output)
  });
  const data = (0, import_smithy_client.expectNonNull)((0, import_smithy_client.expectObject)(await (0, import_core2.parseJsonBody)(output.body, context)), "body");
  const doc = (0, import_smithy_client.take)(data, {
    ServiceIntegration: import_smithy_client._json
  });
  Object.assign(contents, doc);
  return contents;
}, "de_DescribeServiceIntegrationCommand");
var de_GetCostEstimationCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode !== 200 && output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const contents = (0, import_smithy_client.map)({
    $metadata: deserializeMetadata(output)
  });
  const data = (0, import_smithy_client.expectNonNull)((0, import_smithy_client.expectObject)(await (0, import_core2.parseJsonBody)(output.body, context)), "body");
  const doc = (0, import_smithy_client.take)(data, {
    Costs: (_) => de_ServiceResourceCosts(_, context),
    NextToken: import_smithy_client.expectString,
    ResourceCollection: import_smithy_client._json,
    Status: import_smithy_client.expectString,
    TimeRange: (_) => de_CostEstimationTimeRange(_, context),
    TotalCost: import_smithy_client.limitedParseDouble
  });
  Object.assign(contents, doc);
  return contents;
}, "de_GetCostEstimationCommand");
var de_GetResourceCollectionCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode !== 200 && output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const contents = (0, import_smithy_client.map)({
    $metadata: deserializeMetadata(output)
  });
  const data = (0, import_smithy_client.expectNonNull)((0, import_smithy_client.expectObject)(await (0, import_core2.parseJsonBody)(output.body, context)), "body");
  const doc = (0, import_smithy_client.take)(data, {
    NextToken: import_smithy_client.expectString,
    ResourceCollection: import_smithy_client._json
  });
  Object.assign(contents, doc);
  return contents;
}, "de_GetResourceCollectionCommand");
var de_ListAnomaliesForInsightCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode !== 200 && output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const contents = (0, import_smithy_client.map)({
    $metadata: deserializeMetadata(output)
  });
  const data = (0, import_smithy_client.expectNonNull)((0, import_smithy_client.expectObject)(await (0, import_core2.parseJsonBody)(output.body, context)), "body");
  const doc = (0, import_smithy_client.take)(data, {
    NextToken: import_smithy_client.expectString,
    ProactiveAnomalies: (_) => de_ProactiveAnomalies(_, context),
    ReactiveAnomalies: (_) => de_ReactiveAnomalies(_, context)
  });
  Object.assign(contents, doc);
  return contents;
}, "de_ListAnomaliesForInsightCommand");
var de_ListAnomalousLogGroupsCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode !== 200 && output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const contents = (0, import_smithy_client.map)({
    $metadata: deserializeMetadata(output)
  });
  const data = (0, import_smithy_client.expectNonNull)((0, import_smithy_client.expectObject)(await (0, import_core2.parseJsonBody)(output.body, context)), "body");
  const doc = (0, import_smithy_client.take)(data, {
    AnomalousLogGroups: (_) => de_AnomalousLogGroups(_, context),
    InsightId: import_smithy_client.expectString,
    NextToken: import_smithy_client.expectString
  });
  Object.assign(contents, doc);
  return contents;
}, "de_ListAnomalousLogGroupsCommand");
var de_ListEventsCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode !== 200 && output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const contents = (0, import_smithy_client.map)({
    $metadata: deserializeMetadata(output)
  });
  const data = (0, import_smithy_client.expectNonNull)((0, import_smithy_client.expectObject)(await (0, import_core2.parseJsonBody)(output.body, context)), "body");
  const doc = (0, import_smithy_client.take)(data, {
    Events: (_) => de_Events(_, context),
    NextToken: import_smithy_client.expectString
  });
  Object.assign(contents, doc);
  return contents;
}, "de_ListEventsCommand");
var de_ListInsightsCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode !== 200 && output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const contents = (0, import_smithy_client.map)({
    $metadata: deserializeMetadata(output)
  });
  const data = (0, import_smithy_client.expectNonNull)((0, import_smithy_client.expectObject)(await (0, import_core2.parseJsonBody)(output.body, context)), "body");
  const doc = (0, import_smithy_client.take)(data, {
    NextToken: import_smithy_client.expectString,
    ProactiveInsights: (_) => de_ProactiveInsights(_, context),
    ReactiveInsights: (_) => de_ReactiveInsights(_, context)
  });
  Object.assign(contents, doc);
  return contents;
}, "de_ListInsightsCommand");
var de_ListMonitoredResourcesCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode !== 200 && output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const contents = (0, import_smithy_client.map)({
    $metadata: deserializeMetadata(output)
  });
  const data = (0, import_smithy_client.expectNonNull)((0, import_smithy_client.expectObject)(await (0, import_core2.parseJsonBody)(output.body, context)), "body");
  const doc = (0, import_smithy_client.take)(data, {
    MonitoredResourceIdentifiers: (_) => de_MonitoredResourceIdentifiers(_, context),
    NextToken: import_smithy_client.expectString
  });
  Object.assign(contents, doc);
  return contents;
}, "de_ListMonitoredResourcesCommand");
var de_ListNotificationChannelsCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode !== 200 && output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const contents = (0, import_smithy_client.map)({
    $metadata: deserializeMetadata(output)
  });
  const data = (0, import_smithy_client.expectNonNull)((0, import_smithy_client.expectObject)(await (0, import_core2.parseJsonBody)(output.body, context)), "body");
  const doc = (0, import_smithy_client.take)(data, {
    Channels: import_smithy_client._json,
    NextToken: import_smithy_client.expectString
  });
  Object.assign(contents, doc);
  return contents;
}, "de_ListNotificationChannelsCommand");
var de_ListOrganizationInsightsCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode !== 200 && output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const contents = (0, import_smithy_client.map)({
    $metadata: deserializeMetadata(output)
  });
  const data = (0, import_smithy_client.expectNonNull)((0, import_smithy_client.expectObject)(await (0, import_core2.parseJsonBody)(output.body, context)), "body");
  const doc = (0, import_smithy_client.take)(data, {
    NextToken: import_smithy_client.expectString,
    ProactiveInsights: (_) => de_ProactiveOrganizationInsights(_, context),
    ReactiveInsights: (_) => de_ReactiveOrganizationInsights(_, context)
  });
  Object.assign(contents, doc);
  return contents;
}, "de_ListOrganizationInsightsCommand");
var de_ListRecommendationsCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode !== 200 && output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const contents = (0, import_smithy_client.map)({
    $metadata: deserializeMetadata(output)
  });
  const data = (0, import_smithy_client.expectNonNull)((0, import_smithy_client.expectObject)(await (0, import_core2.parseJsonBody)(output.body, context)), "body");
  const doc = (0, import_smithy_client.take)(data, {
    NextToken: import_smithy_client.expectString,
    Recommendations: import_smithy_client._json
  });
  Object.assign(contents, doc);
  return contents;
}, "de_ListRecommendationsCommand");
var de_PutFeedbackCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode !== 200 && output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const contents = (0, import_smithy_client.map)({
    $metadata: deserializeMetadata(output)
  });
  await (0, import_smithy_client.collectBody)(output.body, context);
  return contents;
}, "de_PutFeedbackCommand");
var de_RemoveNotificationChannelCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode !== 200 && output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const contents = (0, import_smithy_client.map)({
    $metadata: deserializeMetadata(output)
  });
  await (0, import_smithy_client.collectBody)(output.body, context);
  return contents;
}, "de_RemoveNotificationChannelCommand");
var de_SearchInsightsCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode !== 200 && output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const contents = (0, import_smithy_client.map)({
    $metadata: deserializeMetadata(output)
  });
  const data = (0, import_smithy_client.expectNonNull)((0, import_smithy_client.expectObject)(await (0, import_core2.parseJsonBody)(output.body, context)), "body");
  const doc = (0, import_smithy_client.take)(data, {
    NextToken: import_smithy_client.expectString,
    ProactiveInsights: (_) => de_ProactiveInsights(_, context),
    ReactiveInsights: (_) => de_ReactiveInsights(_, context)
  });
  Object.assign(contents, doc);
  return contents;
}, "de_SearchInsightsCommand");
var de_SearchOrganizationInsightsCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode !== 200 && output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const contents = (0, import_smithy_client.map)({
    $metadata: deserializeMetadata(output)
  });
  const data = (0, import_smithy_client.expectNonNull)((0, import_smithy_client.expectObject)(await (0, import_core2.parseJsonBody)(output.body, context)), "body");
  const doc = (0, import_smithy_client.take)(data, {
    NextToken: import_smithy_client.expectString,
    ProactiveInsights: (_) => de_ProactiveInsights(_, context),
    ReactiveInsights: (_) => de_ReactiveInsights(_, context)
  });
  Object.assign(contents, doc);
  return contents;
}, "de_SearchOrganizationInsightsCommand");
var de_StartCostEstimationCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode !== 200 && output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const contents = (0, import_smithy_client.map)({
    $metadata: deserializeMetadata(output)
  });
  await (0, import_smithy_client.collectBody)(output.body, context);
  return contents;
}, "de_StartCostEstimationCommand");
var de_UpdateEventSourcesConfigCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode !== 200 && output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const contents = (0, import_smithy_client.map)({
    $metadata: deserializeMetadata(output)
  });
  await (0, import_smithy_client.collectBody)(output.body, context);
  return contents;
}, "de_UpdateEventSourcesConfigCommand");
var de_UpdateResourceCollectionCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode !== 200 && output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const contents = (0, import_smithy_client.map)({
    $metadata: deserializeMetadata(output)
  });
  await (0, import_smithy_client.collectBody)(output.body, context);
  return contents;
}, "de_UpdateResourceCollectionCommand");
var de_UpdateServiceIntegrationCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode !== 200 && output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const contents = (0, import_smithy_client.map)({
    $metadata: deserializeMetadata(output)
  });
  await (0, import_smithy_client.collectBody)(output.body, context);
  return contents;
}, "de_UpdateServiceIntegrationCommand");
var de_CommandError = /* @__PURE__ */ __name(async (output, context) => {
  const parsedOutput = {
    ...output,
    body: await (0, import_core2.parseJsonErrorBody)(output.body, context)
  };
  const errorCode = (0, import_core2.loadRestJsonErrorCode)(output, parsedOutput.body);
  switch (errorCode) {
    case "AccessDeniedException":
    case "com.amazonaws.devopsguru#AccessDeniedException":
      throw await de_AccessDeniedExceptionRes(parsedOutput, context);
    case "ConflictException":
    case "com.amazonaws.devopsguru#ConflictException":
      throw await de_ConflictExceptionRes(parsedOutput, context);
    case "InternalServerException":
    case "com.amazonaws.devopsguru#InternalServerException":
      throw await de_InternalServerExceptionRes(parsedOutput, context);
    case "ResourceNotFoundException":
    case "com.amazonaws.devopsguru#ResourceNotFoundException":
      throw await de_ResourceNotFoundExceptionRes(parsedOutput, context);
    case "ServiceQuotaExceededException":
    case "com.amazonaws.devopsguru#ServiceQuotaExceededException":
      throw await de_ServiceQuotaExceededExceptionRes(parsedOutput, context);
    case "ThrottlingException":
    case "com.amazonaws.devopsguru#ThrottlingException":
      throw await de_ThrottlingExceptionRes(parsedOutput, context);
    case "ValidationException":
    case "com.amazonaws.devopsguru#ValidationException":
      throw await de_ValidationExceptionRes(parsedOutput, context);
    default:
      const parsedBody = parsedOutput.body;
      return throwDefaultError({
        output,
        parsedBody,
        errorCode
      });
  }
}, "de_CommandError");
var throwDefaultError = (0, import_smithy_client.withBaseException)(DevOpsGuruServiceException);
var de_AccessDeniedExceptionRes = /* @__PURE__ */ __name(async (parsedOutput, context) => {
  const contents = (0, import_smithy_client.map)({});
  const data = parsedOutput.body;
  const doc = (0, import_smithy_client.take)(data, {
    Message: import_smithy_client.expectString
  });
  Object.assign(contents, doc);
  const exception = new AccessDeniedException({
    $metadata: deserializeMetadata(parsedOutput),
    ...contents
  });
  return (0, import_smithy_client.decorateServiceException)(exception, parsedOutput.body);
}, "de_AccessDeniedExceptionRes");
var de_ConflictExceptionRes = /* @__PURE__ */ __name(async (parsedOutput, context) => {
  const contents = (0, import_smithy_client.map)({});
  const data = parsedOutput.body;
  const doc = (0, import_smithy_client.take)(data, {
    Message: import_smithy_client.expectString,
    ResourceId: import_smithy_client.expectString,
    ResourceType: import_smithy_client.expectString
  });
  Object.assign(contents, doc);
  const exception = new ConflictException({
    $metadata: deserializeMetadata(parsedOutput),
    ...contents
  });
  return (0, import_smithy_client.decorateServiceException)(exception, parsedOutput.body);
}, "de_ConflictExceptionRes");
var de_InternalServerExceptionRes = /* @__PURE__ */ __name(async (parsedOutput, context) => {
  const contents = (0, import_smithy_client.map)({
    [_RAS]: [() => void 0 !== parsedOutput.headers[_ra], () => (0, import_smithy_client.strictParseInt32)(parsedOutput.headers[_ra])]
  });
  const data = parsedOutput.body;
  const doc = (0, import_smithy_client.take)(data, {
    Message: import_smithy_client.expectString
  });
  Object.assign(contents, doc);
  const exception = new InternalServerException({
    $metadata: deserializeMetadata(parsedOutput),
    ...contents
  });
  return (0, import_smithy_client.decorateServiceException)(exception, parsedOutput.body);
}, "de_InternalServerExceptionRes");
var de_ResourceNotFoundExceptionRes = /* @__PURE__ */ __name(async (parsedOutput, context) => {
  const contents = (0, import_smithy_client.map)({});
  const data = parsedOutput.body;
  const doc = (0, import_smithy_client.take)(data, {
    Message: import_smithy_client.expectString,
    ResourceId: import_smithy_client.expectString,
    ResourceType: import_smithy_client.expectString
  });
  Object.assign(contents, doc);
  const exception = new ResourceNotFoundException({
    $metadata: deserializeMetadata(parsedOutput),
    ...contents
  });
  return (0, import_smithy_client.decorateServiceException)(exception, parsedOutput.body);
}, "de_ResourceNotFoundExceptionRes");
var de_ServiceQuotaExceededExceptionRes = /* @__PURE__ */ __name(async (parsedOutput, context) => {
  const contents = (0, import_smithy_client.map)({});
  const data = parsedOutput.body;
  const doc = (0, import_smithy_client.take)(data, {
    Message: import_smithy_client.expectString
  });
  Object.assign(contents, doc);
  const exception = new ServiceQuotaExceededException({
    $metadata: deserializeMetadata(parsedOutput),
    ...contents
  });
  return (0, import_smithy_client.decorateServiceException)(exception, parsedOutput.body);
}, "de_ServiceQuotaExceededExceptionRes");
var de_ThrottlingExceptionRes = /* @__PURE__ */ __name(async (parsedOutput, context) => {
  const contents = (0, import_smithy_client.map)({
    [_RAS]: [() => void 0 !== parsedOutput.headers[_ra], () => (0, import_smithy_client.strictParseInt32)(parsedOutput.headers[_ra])]
  });
  const data = parsedOutput.body;
  const doc = (0, import_smithy_client.take)(data, {
    Message: import_smithy_client.expectString,
    QuotaCode: import_smithy_client.expectString,
    ServiceCode: import_smithy_client.expectString
  });
  Object.assign(contents, doc);
  const exception = new ThrottlingException({
    $metadata: deserializeMetadata(parsedOutput),
    ...contents
  });
  return (0, import_smithy_client.decorateServiceException)(exception, parsedOutput.body);
}, "de_ThrottlingExceptionRes");
var de_ValidationExceptionRes = /* @__PURE__ */ __name(async (parsedOutput, context) => {
  const contents = (0, import_smithy_client.map)({});
  const data = parsedOutput.body;
  const doc = (0, import_smithy_client.take)(data, {
    Fields: import_smithy_client._json,
    Message: import_smithy_client.expectString,
    Reason: import_smithy_client.expectString
  });
  Object.assign(contents, doc);
  const exception = new ValidationException({
    $metadata: deserializeMetadata(parsedOutput),
    ...contents
  });
  return (0, import_smithy_client.decorateServiceException)(exception, parsedOutput.body);
}, "de_ValidationExceptionRes");
var se_EndTimeRange = /* @__PURE__ */ __name((input, context) => {
  return (0, import_smithy_client.take)(input, {
    FromTime: (_) => Math.round(_.getTime() / 1e3),
    ToTime: (_) => Math.round(_.getTime() / 1e3)
  });
}, "se_EndTimeRange");
var se_EventTimeRange = /* @__PURE__ */ __name((input, context) => {
  return (0, import_smithy_client.take)(input, {
    FromTime: (_) => Math.round(_.getTime() / 1e3),
    ToTime: (_) => Math.round(_.getTime() / 1e3)
  });
}, "se_EventTimeRange");
var se_ListEventsFilters = /* @__PURE__ */ __name((input, context) => {
  return (0, import_smithy_client.take)(input, {
    DataSource: [],
    EventClass: [],
    EventSource: [],
    EventTimeRange: (_) => se_EventTimeRange(_, context),
    InsightId: [],
    ResourceCollection: import_smithy_client._json
  });
}, "se_ListEventsFilters");
var se_ListInsightsAnyStatusFilter = /* @__PURE__ */ __name((input, context) => {
  return (0, import_smithy_client.take)(input, {
    StartTimeRange: (_) => se_StartTimeRange(_, context),
    Type: []
  });
}, "se_ListInsightsAnyStatusFilter");
var se_ListInsightsClosedStatusFilter = /* @__PURE__ */ __name((input, context) => {
  return (0, import_smithy_client.take)(input, {
    EndTimeRange: (_) => se_EndTimeRange(_, context),
    Type: []
  });
}, "se_ListInsightsClosedStatusFilter");
var se_ListInsightsStatusFilter = /* @__PURE__ */ __name((input, context) => {
  return (0, import_smithy_client.take)(input, {
    Any: (_) => se_ListInsightsAnyStatusFilter(_, context),
    Closed: (_) => se_ListInsightsClosedStatusFilter(_, context),
    Ongoing: import_smithy_client._json
  });
}, "se_ListInsightsStatusFilter");
var se_StartTimeRange = /* @__PURE__ */ __name((input, context) => {
  return (0, import_smithy_client.take)(input, {
    FromTime: (_) => Math.round(_.getTime() / 1e3),
    ToTime: (_) => Math.round(_.getTime() / 1e3)
  });
}, "se_StartTimeRange");
var de_AnomalousLogGroup = /* @__PURE__ */ __name((output, context) => {
  return (0, import_smithy_client.take)(output, {
    ImpactEndTime: (_) => (0, import_smithy_client.expectNonNull)((0, import_smithy_client.parseEpochTimestamp)((0, import_smithy_client.expectNumber)(_))),
    ImpactStartTime: (_) => (0, import_smithy_client.expectNonNull)((0, import_smithy_client.parseEpochTimestamp)((0, import_smithy_client.expectNumber)(_))),
    LogAnomalyShowcases: (_) => de_LogAnomalyShowcases(_, context),
    LogGroupName: import_smithy_client.expectString,
    NumberOfLogLinesScanned: import_smithy_client.expectInt32
  });
}, "de_AnomalousLogGroup");
var de_AnomalousLogGroups = /* @__PURE__ */ __name((output, context) => {
  const retVal = (output || []).filter((e) => e != null).map((entry) => {
    return de_AnomalousLogGroup(entry, context);
  });
  return retVal;
}, "de_AnomalousLogGroups");
var de_AnomalyReportedTimeRange = /* @__PURE__ */ __name((output, context) => {
  return (0, import_smithy_client.take)(output, {
    CloseTime: (_) => (0, import_smithy_client.expectNonNull)((0, import_smithy_client.parseEpochTimestamp)((0, import_smithy_client.expectNumber)(_))),
    OpenTime: (_) => (0, import_smithy_client.expectNonNull)((0, import_smithy_client.parseEpochTimestamp)((0, import_smithy_client.expectNumber)(_)))
  });
}, "de_AnomalyReportedTimeRange");
var de_AnomalySourceDetails = /* @__PURE__ */ __name((output, context) => {
  return (0, import_smithy_client.take)(output, {
    CloudWatchMetrics: (_) => de_CloudWatchMetricsDetails(_, context),
    PerformanceInsightsMetrics: (_) => de_PerformanceInsightsMetricsDetails(_, context)
  });
}, "de_AnomalySourceDetails");
var de_AnomalyTimeRange = /* @__PURE__ */ __name((output, context) => {
  return (0, import_smithy_client.take)(output, {
    EndTime: (_) => (0, import_smithy_client.expectNonNull)((0, import_smithy_client.parseEpochTimestamp)((0, import_smithy_client.expectNumber)(_))),
    StartTime: (_) => (0, import_smithy_client.expectNonNull)((0, import_smithy_client.parseEpochTimestamp)((0, import_smithy_client.expectNumber)(_)))
  });
}, "de_AnomalyTimeRange");
var de_CloudWatchMetricsDataSummary = /* @__PURE__ */ __name((output, context) => {
  return (0, import_smithy_client.take)(output, {
    StatusCode: import_smithy_client.expectString,
    TimestampMetricValuePairList: (_) => de_TimestampMetricValuePairList(_, context)
  });
}, "de_CloudWatchMetricsDataSummary");
var de_CloudWatchMetricsDetail = /* @__PURE__ */ __name((output, context) => {
  return (0, import_smithy_client.take)(output, {
    Dimensions: import_smithy_client._json,
    MetricDataSummary: (_) => de_CloudWatchMetricsDataSummary(_, context),
    MetricName: import_smithy_client.expectString,
    Namespace: import_smithy_client.expectString,
    Period: import_smithy_client.expectInt32,
    Stat: import_smithy_client.expectString,
    Unit: import_smithy_client.expectString
  });
}, "de_CloudWatchMetricsDetail");
var de_CloudWatchMetricsDetails = /* @__PURE__ */ __name((output, context) => {
  const retVal = (output || []).filter((e) => e != null).map((entry) => {
    return de_CloudWatchMetricsDetail(entry, context);
  });
  return retVal;
}, "de_CloudWatchMetricsDetails");
var de_CostEstimationTimeRange = /* @__PURE__ */ __name((output, context) => {
  return (0, import_smithy_client.take)(output, {
    EndTime: (_) => (0, import_smithy_client.expectNonNull)((0, import_smithy_client.parseEpochTimestamp)((0, import_smithy_client.expectNumber)(_))),
    StartTime: (_) => (0, import_smithy_client.expectNonNull)((0, import_smithy_client.parseEpochTimestamp)((0, import_smithy_client.expectNumber)(_)))
  });
}, "de_CostEstimationTimeRange");
var de_Event = /* @__PURE__ */ __name((output, context) => {
  return (0, import_smithy_client.take)(output, {
    DataSource: import_smithy_client.expectString,
    EventClass: import_smithy_client.expectString,
    EventSource: import_smithy_client.expectString,
    Id: import_smithy_client.expectString,
    Name: import_smithy_client.expectString,
    ResourceCollection: import_smithy_client._json,
    Resources: import_smithy_client._json,
    Time: (_) => (0, import_smithy_client.expectNonNull)((0, import_smithy_client.parseEpochTimestamp)((0, import_smithy_client.expectNumber)(_)))
  });
}, "de_Event");
var de_Events = /* @__PURE__ */ __name((output, context) => {
  const retVal = (output || []).filter((e) => e != null).map((entry) => {
    return de_Event(entry, context);
  });
  return retVal;
}, "de_Events");
var de_InsightTimeRange = /* @__PURE__ */ __name((output, context) => {
  return (0, import_smithy_client.take)(output, {
    EndTime: (_) => (0, import_smithy_client.expectNonNull)((0, import_smithy_client.parseEpochTimestamp)((0, import_smithy_client.expectNumber)(_))),
    StartTime: (_) => (0, import_smithy_client.expectNonNull)((0, import_smithy_client.parseEpochTimestamp)((0, import_smithy_client.expectNumber)(_)))
  });
}, "de_InsightTimeRange");
var de_LogAnomalyClass = /* @__PURE__ */ __name((output, context) => {
  return (0, import_smithy_client.take)(output, {
    Explanation: import_smithy_client.expectString,
    LogAnomalyToken: import_smithy_client.expectString,
    LogAnomalyType: import_smithy_client.expectString,
    LogEventId: import_smithy_client.expectString,
    LogEventTimestamp: (_) => (0, import_smithy_client.expectNonNull)((0, import_smithy_client.parseEpochTimestamp)((0, import_smithy_client.expectNumber)(_))),
    LogStreamName: import_smithy_client.expectString,
    NumberOfLogLinesOccurrences: import_smithy_client.expectInt32
  });
}, "de_LogAnomalyClass");
var de_LogAnomalyClasses = /* @__PURE__ */ __name((output, context) => {
  const retVal = (output || []).filter((e) => e != null).map((entry) => {
    return de_LogAnomalyClass(entry, context);
  });
  return retVal;
}, "de_LogAnomalyClasses");
var de_LogAnomalyShowcase = /* @__PURE__ */ __name((output, context) => {
  return (0, import_smithy_client.take)(output, {
    LogAnomalyClasses: (_) => de_LogAnomalyClasses(_, context)
  });
}, "de_LogAnomalyShowcase");
var de_LogAnomalyShowcases = /* @__PURE__ */ __name((output, context) => {
  const retVal = (output || []).filter((e) => e != null).map((entry) => {
    return de_LogAnomalyShowcase(entry, context);
  });
  return retVal;
}, "de_LogAnomalyShowcases");
var de_MonitoredResourceIdentifier = /* @__PURE__ */ __name((output, context) => {
  return (0, import_smithy_client.take)(output, {
    LastUpdated: (_) => (0, import_smithy_client.expectNonNull)((0, import_smithy_client.parseEpochTimestamp)((0, import_smithy_client.expectNumber)(_))),
    MonitoredResourceName: import_smithy_client.expectString,
    ResourceCollection: import_smithy_client._json,
    ResourcePermission: import_smithy_client.expectString,
    Type: import_smithy_client.expectString
  });
}, "de_MonitoredResourceIdentifier");
var de_MonitoredResourceIdentifiers = /* @__PURE__ */ __name((output, context) => {
  const retVal = (output || []).filter((e) => e != null).map((entry) => {
    return de_MonitoredResourceIdentifier(entry, context);
  });
  return retVal;
}, "de_MonitoredResourceIdentifiers");
var de_PerformanceInsightsMetricsDetail = /* @__PURE__ */ __name((output, context) => {
  return (0, import_smithy_client.take)(output, {
    MetricDisplayName: import_smithy_client.expectString,
    MetricQuery: import_smithy_client._json,
    ReferenceData: (_) => de_PerformanceInsightsReferenceDataList(_, context),
    StatsAtAnomaly: (_) => de_PerformanceInsightsStats(_, context),
    StatsAtBaseline: (_) => de_PerformanceInsightsStats(_, context),
    Unit: import_smithy_client.expectString
  });
}, "de_PerformanceInsightsMetricsDetail");
var de_PerformanceInsightsMetricsDetails = /* @__PURE__ */ __name((output, context) => {
  const retVal = (output || []).filter((e) => e != null).map((entry) => {
    return de_PerformanceInsightsMetricsDetail(entry, context);
  });
  return retVal;
}, "de_PerformanceInsightsMetricsDetails");
var de_PerformanceInsightsReferenceComparisonValues = /* @__PURE__ */ __name((output, context) => {
  return (0, import_smithy_client.take)(output, {
    ReferenceMetric: import_smithy_client._json,
    ReferenceScalar: (_) => de_PerformanceInsightsReferenceScalar(_, context)
  });
}, "de_PerformanceInsightsReferenceComparisonValues");
var de_PerformanceInsightsReferenceData = /* @__PURE__ */ __name((output, context) => {
  return (0, import_smithy_client.take)(output, {
    ComparisonValues: (_) => de_PerformanceInsightsReferenceComparisonValues(_, context),
    Name: import_smithy_client.expectString
  });
}, "de_PerformanceInsightsReferenceData");
var de_PerformanceInsightsReferenceDataList = /* @__PURE__ */ __name((output, context) => {
  const retVal = (output || []).filter((e) => e != null).map((entry) => {
    return de_PerformanceInsightsReferenceData(entry, context);
  });
  return retVal;
}, "de_PerformanceInsightsReferenceDataList");
var de_PerformanceInsightsReferenceScalar = /* @__PURE__ */ __name((output, context) => {
  return (0, import_smithy_client.take)(output, {
    Value: import_smithy_client.limitedParseDouble
  });
}, "de_PerformanceInsightsReferenceScalar");
var de_PerformanceInsightsStat = /* @__PURE__ */ __name((output, context) => {
  return (0, import_smithy_client.take)(output, {
    Type: import_smithy_client.expectString,
    Value: import_smithy_client.limitedParseDouble
  });
}, "de_PerformanceInsightsStat");
var de_PerformanceInsightsStats = /* @__PURE__ */ __name((output, context) => {
  const retVal = (output || []).filter((e) => e != null).map((entry) => {
    return de_PerformanceInsightsStat(entry, context);
  });
  return retVal;
}, "de_PerformanceInsightsStats");
var de_PredictionTimeRange = /* @__PURE__ */ __name((output, context) => {
  return (0, import_smithy_client.take)(output, {
    EndTime: (_) => (0, import_smithy_client.expectNonNull)((0, import_smithy_client.parseEpochTimestamp)((0, import_smithy_client.expectNumber)(_))),
    StartTime: (_) => (0, import_smithy_client.expectNonNull)((0, import_smithy_client.parseEpochTimestamp)((0, import_smithy_client.expectNumber)(_)))
  });
}, "de_PredictionTimeRange");
var de_ProactiveAnomalies = /* @__PURE__ */ __name((output, context) => {
  const retVal = (output || []).filter((e) => e != null).map((entry) => {
    return de_ProactiveAnomalySummary(entry, context);
  });
  return retVal;
}, "de_ProactiveAnomalies");
var de_ProactiveAnomaly = /* @__PURE__ */ __name((output, context) => {
  return (0, import_smithy_client.take)(output, {
    AnomalyReportedTimeRange: (_) => de_AnomalyReportedTimeRange(_, context),
    AnomalyResources: import_smithy_client._json,
    AnomalyTimeRange: (_) => de_AnomalyTimeRange(_, context),
    AssociatedInsightId: import_smithy_client.expectString,
    Description: import_smithy_client.expectString,
    Id: import_smithy_client.expectString,
    Limit: import_smithy_client.limitedParseDouble,
    PredictionTimeRange: (_) => de_PredictionTimeRange(_, context),
    ResourceCollection: import_smithy_client._json,
    Severity: import_smithy_client.expectString,
    SourceDetails: (_) => de_AnomalySourceDetails(_, context),
    SourceMetadata: import_smithy_client._json,
    Status: import_smithy_client.expectString,
    UpdateTime: (_) => (0, import_smithy_client.expectNonNull)((0, import_smithy_client.parseEpochTimestamp)((0, import_smithy_client.expectNumber)(_)))
  });
}, "de_ProactiveAnomaly");
var de_ProactiveAnomalySummary = /* @__PURE__ */ __name((output, context) => {
  return (0, import_smithy_client.take)(output, {
    AnomalyReportedTimeRange: (_) => de_AnomalyReportedTimeRange(_, context),
    AnomalyResources: import_smithy_client._json,
    AnomalyTimeRange: (_) => de_AnomalyTimeRange(_, context),
    AssociatedInsightId: import_smithy_client.expectString,
    Description: import_smithy_client.expectString,
    Id: import_smithy_client.expectString,
    Limit: import_smithy_client.limitedParseDouble,
    PredictionTimeRange: (_) => de_PredictionTimeRange(_, context),
    ResourceCollection: import_smithy_client._json,
    Severity: import_smithy_client.expectString,
    SourceDetails: (_) => de_AnomalySourceDetails(_, context),
    SourceMetadata: import_smithy_client._json,
    Status: import_smithy_client.expectString,
    UpdateTime: (_) => (0, import_smithy_client.expectNonNull)((0, import_smithy_client.parseEpochTimestamp)((0, import_smithy_client.expectNumber)(_)))
  });
}, "de_ProactiveAnomalySummary");
var de_ProactiveInsight = /* @__PURE__ */ __name((output, context) => {
  return (0, import_smithy_client.take)(output, {
    Description: import_smithy_client.expectString,
    Id: import_smithy_client.expectString,
    InsightTimeRange: (_) => de_InsightTimeRange(_, context),
    Name: import_smithy_client.expectString,
    PredictionTimeRange: (_) => de_PredictionTimeRange(_, context),
    ResourceCollection: import_smithy_client._json,
    Severity: import_smithy_client.expectString,
    SsmOpsItemId: import_smithy_client.expectString,
    Status: import_smithy_client.expectString
  });
}, "de_ProactiveInsight");
var de_ProactiveInsights = /* @__PURE__ */ __name((output, context) => {
  const retVal = (output || []).filter((e) => e != null).map((entry) => {
    return de_ProactiveInsightSummary(entry, context);
  });
  return retVal;
}, "de_ProactiveInsights");
var de_ProactiveInsightSummary = /* @__PURE__ */ __name((output, context) => {
  return (0, import_smithy_client.take)(output, {
    AssociatedResourceArns: import_smithy_client._json,
    Id: import_smithy_client.expectString,
    InsightTimeRange: (_) => de_InsightTimeRange(_, context),
    Name: import_smithy_client.expectString,
    PredictionTimeRange: (_) => de_PredictionTimeRange(_, context),
    ResourceCollection: import_smithy_client._json,
    ServiceCollection: import_smithy_client._json,
    Severity: import_smithy_client.expectString,
    Status: import_smithy_client.expectString
  });
}, "de_ProactiveInsightSummary");
var de_ProactiveOrganizationInsights = /* @__PURE__ */ __name((output, context) => {
  const retVal = (output || []).filter((e) => e != null).map((entry) => {
    return de_ProactiveOrganizationInsightSummary(entry, context);
  });
  return retVal;
}, "de_ProactiveOrganizationInsights");
var de_ProactiveOrganizationInsightSummary = /* @__PURE__ */ __name((output, context) => {
  return (0, import_smithy_client.take)(output, {
    AccountId: import_smithy_client.expectString,
    Id: import_smithy_client.expectString,
    InsightTimeRange: (_) => de_InsightTimeRange(_, context),
    Name: import_smithy_client.expectString,
    OrganizationalUnitId: import_smithy_client.expectString,
    PredictionTimeRange: (_) => de_PredictionTimeRange(_, context),
    ResourceCollection: import_smithy_client._json,
    ServiceCollection: import_smithy_client._json,
    Severity: import_smithy_client.expectString,
    Status: import_smithy_client.expectString
  });
}, "de_ProactiveOrganizationInsightSummary");
var de_ReactiveAnomalies = /* @__PURE__ */ __name((output, context) => {
  const retVal = (output || []).filter((e) => e != null).map((entry) => {
    return de_ReactiveAnomalySummary(entry, context);
  });
  return retVal;
}, "de_ReactiveAnomalies");
var de_ReactiveAnomaly = /* @__PURE__ */ __name((output, context) => {
  return (0, import_smithy_client.take)(output, {
    AnomalyReportedTimeRange: (_) => de_AnomalyReportedTimeRange(_, context),
    AnomalyResources: import_smithy_client._json,
    AnomalyTimeRange: (_) => de_AnomalyTimeRange(_, context),
    AssociatedInsightId: import_smithy_client.expectString,
    CausalAnomalyId: import_smithy_client.expectString,
    Description: import_smithy_client.expectString,
    Id: import_smithy_client.expectString,
    Name: import_smithy_client.expectString,
    ResourceCollection: import_smithy_client._json,
    Severity: import_smithy_client.expectString,
    SourceDetails: (_) => de_AnomalySourceDetails(_, context),
    Status: import_smithy_client.expectString,
    Type: import_smithy_client.expectString
  });
}, "de_ReactiveAnomaly");
var de_ReactiveAnomalySummary = /* @__PURE__ */ __name((output, context) => {
  return (0, import_smithy_client.take)(output, {
    AnomalyReportedTimeRange: (_) => de_AnomalyReportedTimeRange(_, context),
    AnomalyResources: import_smithy_client._json,
    AnomalyTimeRange: (_) => de_AnomalyTimeRange(_, context),
    AssociatedInsightId: import_smithy_client.expectString,
    CausalAnomalyId: import_smithy_client.expectString,
    Description: import_smithy_client.expectString,
    Id: import_smithy_client.expectString,
    Name: import_smithy_client.expectString,
    ResourceCollection: import_smithy_client._json,
    Severity: import_smithy_client.expectString,
    SourceDetails: (_) => de_AnomalySourceDetails(_, context),
    Status: import_smithy_client.expectString,
    Type: import_smithy_client.expectString
  });
}, "de_ReactiveAnomalySummary");
var de_ReactiveInsight = /* @__PURE__ */ __name((output, context) => {
  return (0, import_smithy_client.take)(output, {
    Description: import_smithy_client.expectString,
    Id: import_smithy_client.expectString,
    InsightTimeRange: (_) => de_InsightTimeRange(_, context),
    Name: import_smithy_client.expectString,
    ResourceCollection: import_smithy_client._json,
    Severity: import_smithy_client.expectString,
    SsmOpsItemId: import_smithy_client.expectString,
    Status: import_smithy_client.expectString
  });
}, "de_ReactiveInsight");
var de_ReactiveInsights = /* @__PURE__ */ __name((output, context) => {
  const retVal = (output || []).filter((e) => e != null).map((entry) => {
    return de_ReactiveInsightSummary(entry, context);
  });
  return retVal;
}, "de_ReactiveInsights");
var de_ReactiveInsightSummary = /* @__PURE__ */ __name((output, context) => {
  return (0, import_smithy_client.take)(output, {
    AssociatedResourceArns: import_smithy_client._json,
    Id: import_smithy_client.expectString,
    InsightTimeRange: (_) => de_InsightTimeRange(_, context),
    Name: import_smithy_client.expectString,
    ResourceCollection: import_smithy_client._json,
    ServiceCollection: import_smithy_client._json,
    Severity: import_smithy_client.expectString,
    Status: import_smithy_client.expectString
  });
}, "de_ReactiveInsightSummary");
var de_ReactiveOrganizationInsights = /* @__PURE__ */ __name((output, context) => {
  const retVal = (output || []).filter((e) => e != null).map((entry) => {
    return de_ReactiveOrganizationInsightSummary(entry, context);
  });
  return retVal;
}, "de_ReactiveOrganizationInsights");
var de_ReactiveOrganizationInsightSummary = /* @__PURE__ */ __name((output, context) => {
  return (0, import_smithy_client.take)(output, {
    AccountId: import_smithy_client.expectString,
    Id: import_smithy_client.expectString,
    InsightTimeRange: (_) => de_InsightTimeRange(_, context),
    Name: import_smithy_client.expectString,
    OrganizationalUnitId: import_smithy_client.expectString,
    ResourceCollection: import_smithy_client._json,
    ServiceCollection: import_smithy_client._json,
    Severity: import_smithy_client.expectString,
    Status: import_smithy_client.expectString
  });
}, "de_ReactiveOrganizationInsightSummary");
var de_ServiceResourceCost = /* @__PURE__ */ __name((output, context) => {
  return (0, import_smithy_client.take)(output, {
    Cost: import_smithy_client.limitedParseDouble,
    Count: import_smithy_client.expectInt32,
    State: import_smithy_client.expectString,
    Type: import_smithy_client.expectString,
    UnitCost: import_smithy_client.limitedParseDouble
  });
}, "de_ServiceResourceCost");
var de_ServiceResourceCosts = /* @__PURE__ */ __name((output, context) => {
  const retVal = (output || []).filter((e) => e != null).map((entry) => {
    return de_ServiceResourceCost(entry, context);
  });
  return retVal;
}, "de_ServiceResourceCosts");
var de_TimestampMetricValuePair = /* @__PURE__ */ __name((output, context) => {
  return (0, import_smithy_client.take)(output, {
    MetricValue: import_smithy_client.limitedParseDouble,
    Timestamp: (_) => (0, import_smithy_client.expectNonNull)((0, import_smithy_client.parseEpochTimestamp)((0, import_smithy_client.expectNumber)(_)))
  });
}, "de_TimestampMetricValuePair");
var de_TimestampMetricValuePairList = /* @__PURE__ */ __name((output, context) => {
  const retVal = (output || []).filter((e) => e != null).map((entry) => {
    return de_TimestampMetricValuePair(entry, context);
  });
  return retVal;
}, "de_TimestampMetricValuePairList");
var deserializeMetadata = /* @__PURE__ */ __name((output) => ({
  httpStatusCode: output.statusCode,
  requestId: output.headers["x-amzn-requestid"] ?? output.headers["x-amzn-request-id"] ?? output.headers["x-amz-request-id"],
  extendedRequestId: output.headers["x-amz-id-2"],
  cfId: output.headers["x-amz-cf-id"]
}), "deserializeMetadata");
var _AI = "AccountId";
var _NT = "NextToken";
var _RAS = "RetryAfterSeconds";
var _ra = "retry-after";

// src/commands/AddNotificationChannelCommand.ts
var _AddNotificationChannelCommand = class _AddNotificationChannelCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("CapstoneControlPlaneService", "AddNotificationChannel", {}).n("DevOpsGuruClient", "AddNotificationChannelCommand").f(void 0, void 0).ser(se_AddNotificationChannelCommand).de(de_AddNotificationChannelCommand).build() {
};
__name(_AddNotificationChannelCommand, "AddNotificationChannelCommand");
var AddNotificationChannelCommand = _AddNotificationChannelCommand;

// src/commands/DeleteInsightCommand.ts




var _DeleteInsightCommand = class _DeleteInsightCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("CapstoneControlPlaneService", "DeleteInsight", {}).n("DevOpsGuruClient", "DeleteInsightCommand").f(void 0, void 0).ser(se_DeleteInsightCommand).de(de_DeleteInsightCommand).build() {
};
__name(_DeleteInsightCommand, "DeleteInsightCommand");
var DeleteInsightCommand = _DeleteInsightCommand;

// src/commands/DescribeAccountHealthCommand.ts




var _DescribeAccountHealthCommand = class _DescribeAccountHealthCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("CapstoneControlPlaneService", "DescribeAccountHealth", {}).n("DevOpsGuruClient", "DescribeAccountHealthCommand").f(void 0, void 0).ser(se_DescribeAccountHealthCommand).de(de_DescribeAccountHealthCommand).build() {
};
__name(_DescribeAccountHealthCommand, "DescribeAccountHealthCommand");
var DescribeAccountHealthCommand = _DescribeAccountHealthCommand;

// src/commands/DescribeAccountOverviewCommand.ts




var _DescribeAccountOverviewCommand = class _DescribeAccountOverviewCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("CapstoneControlPlaneService", "DescribeAccountOverview", {}).n("DevOpsGuruClient", "DescribeAccountOverviewCommand").f(void 0, void 0).ser(se_DescribeAccountOverviewCommand).de(de_DescribeAccountOverviewCommand).build() {
};
__name(_DescribeAccountOverviewCommand, "DescribeAccountOverviewCommand");
var DescribeAccountOverviewCommand = _DescribeAccountOverviewCommand;

// src/commands/DescribeAnomalyCommand.ts




var _DescribeAnomalyCommand = class _DescribeAnomalyCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("CapstoneControlPlaneService", "DescribeAnomaly", {}).n("DevOpsGuruClient", "DescribeAnomalyCommand").f(void 0, void 0).ser(se_DescribeAnomalyCommand).de(de_DescribeAnomalyCommand).build() {
};
__name(_DescribeAnomalyCommand, "DescribeAnomalyCommand");
var DescribeAnomalyCommand = _DescribeAnomalyCommand;

// src/commands/DescribeEventSourcesConfigCommand.ts




var _DescribeEventSourcesConfigCommand = class _DescribeEventSourcesConfigCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("CapstoneControlPlaneService", "DescribeEventSourcesConfig", {}).n("DevOpsGuruClient", "DescribeEventSourcesConfigCommand").f(void 0, void 0).ser(se_DescribeEventSourcesConfigCommand).de(de_DescribeEventSourcesConfigCommand).build() {
};
__name(_DescribeEventSourcesConfigCommand, "DescribeEventSourcesConfigCommand");
var DescribeEventSourcesConfigCommand = _DescribeEventSourcesConfigCommand;

// src/commands/DescribeFeedbackCommand.ts




var _DescribeFeedbackCommand = class _DescribeFeedbackCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("CapstoneControlPlaneService", "DescribeFeedback", {}).n("DevOpsGuruClient", "DescribeFeedbackCommand").f(void 0, void 0).ser(se_DescribeFeedbackCommand).de(de_DescribeFeedbackCommand).build() {
};
__name(_DescribeFeedbackCommand, "DescribeFeedbackCommand");
var DescribeFeedbackCommand = _DescribeFeedbackCommand;

// src/commands/DescribeInsightCommand.ts




var _DescribeInsightCommand = class _DescribeInsightCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("CapstoneControlPlaneService", "DescribeInsight", {}).n("DevOpsGuruClient", "DescribeInsightCommand").f(void 0, void 0).ser(se_DescribeInsightCommand).de(de_DescribeInsightCommand).build() {
};
__name(_DescribeInsightCommand, "DescribeInsightCommand");
var DescribeInsightCommand = _DescribeInsightCommand;

// src/commands/DescribeOrganizationHealthCommand.ts




var _DescribeOrganizationHealthCommand = class _DescribeOrganizationHealthCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("CapstoneControlPlaneService", "DescribeOrganizationHealth", {}).n("DevOpsGuruClient", "DescribeOrganizationHealthCommand").f(void 0, void 0).ser(se_DescribeOrganizationHealthCommand).de(de_DescribeOrganizationHealthCommand).build() {
};
__name(_DescribeOrganizationHealthCommand, "DescribeOrganizationHealthCommand");
var DescribeOrganizationHealthCommand = _DescribeOrganizationHealthCommand;

// src/commands/DescribeOrganizationOverviewCommand.ts




var _DescribeOrganizationOverviewCommand = class _DescribeOrganizationOverviewCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("CapstoneControlPlaneService", "DescribeOrganizationOverview", {}).n("DevOpsGuruClient", "DescribeOrganizationOverviewCommand").f(void 0, void 0).ser(se_DescribeOrganizationOverviewCommand).de(de_DescribeOrganizationOverviewCommand).build() {
};
__name(_DescribeOrganizationOverviewCommand, "DescribeOrganizationOverviewCommand");
var DescribeOrganizationOverviewCommand = _DescribeOrganizationOverviewCommand;

// src/commands/DescribeOrganizationResourceCollectionHealthCommand.ts




var _DescribeOrganizationResourceCollectionHealthCommand = class _DescribeOrganizationResourceCollectionHealthCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("CapstoneControlPlaneService", "DescribeOrganizationResourceCollectionHealth", {}).n("DevOpsGuruClient", "DescribeOrganizationResourceCollectionHealthCommand").f(void 0, void 0).ser(se_DescribeOrganizationResourceCollectionHealthCommand).de(de_DescribeOrganizationResourceCollectionHealthCommand).build() {
};
__name(_DescribeOrganizationResourceCollectionHealthCommand, "DescribeOrganizationResourceCollectionHealthCommand");
var DescribeOrganizationResourceCollectionHealthCommand = _DescribeOrganizationResourceCollectionHealthCommand;

// src/commands/DescribeResourceCollectionHealthCommand.ts




var _DescribeResourceCollectionHealthCommand = class _DescribeResourceCollectionHealthCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("CapstoneControlPlaneService", "DescribeResourceCollectionHealth", {}).n("DevOpsGuruClient", "DescribeResourceCollectionHealthCommand").f(void 0, void 0).ser(se_DescribeResourceCollectionHealthCommand).de(de_DescribeResourceCollectionHealthCommand).build() {
};
__name(_DescribeResourceCollectionHealthCommand, "DescribeResourceCollectionHealthCommand");
var DescribeResourceCollectionHealthCommand = _DescribeResourceCollectionHealthCommand;

// src/commands/DescribeServiceIntegrationCommand.ts




var _DescribeServiceIntegrationCommand = class _DescribeServiceIntegrationCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("CapstoneControlPlaneService", "DescribeServiceIntegration", {}).n("DevOpsGuruClient", "DescribeServiceIntegrationCommand").f(void 0, void 0).ser(se_DescribeServiceIntegrationCommand).de(de_DescribeServiceIntegrationCommand).build() {
};
__name(_DescribeServiceIntegrationCommand, "DescribeServiceIntegrationCommand");
var DescribeServiceIntegrationCommand = _DescribeServiceIntegrationCommand;

// src/commands/GetCostEstimationCommand.ts




var _GetCostEstimationCommand = class _GetCostEstimationCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("CapstoneControlPlaneService", "GetCostEstimation", {}).n("DevOpsGuruClient", "GetCostEstimationCommand").f(void 0, void 0).ser(se_GetCostEstimationCommand).de(de_GetCostEstimationCommand).build() {
};
__name(_GetCostEstimationCommand, "GetCostEstimationCommand");
var GetCostEstimationCommand = _GetCostEstimationCommand;

// src/commands/GetResourceCollectionCommand.ts




var _GetResourceCollectionCommand = class _GetResourceCollectionCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("CapstoneControlPlaneService", "GetResourceCollection", {}).n("DevOpsGuruClient", "GetResourceCollectionCommand").f(void 0, void 0).ser(se_GetResourceCollectionCommand).de(de_GetResourceCollectionCommand).build() {
};
__name(_GetResourceCollectionCommand, "GetResourceCollectionCommand");
var GetResourceCollectionCommand = _GetResourceCollectionCommand;

// src/commands/ListAnomaliesForInsightCommand.ts




var _ListAnomaliesForInsightCommand = class _ListAnomaliesForInsightCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("CapstoneControlPlaneService", "ListAnomaliesForInsight", {}).n("DevOpsGuruClient", "ListAnomaliesForInsightCommand").f(void 0, void 0).ser(se_ListAnomaliesForInsightCommand).de(de_ListAnomaliesForInsightCommand).build() {
};
__name(_ListAnomaliesForInsightCommand, "ListAnomaliesForInsightCommand");
var ListAnomaliesForInsightCommand = _ListAnomaliesForInsightCommand;

// src/commands/ListAnomalousLogGroupsCommand.ts




var _ListAnomalousLogGroupsCommand = class _ListAnomalousLogGroupsCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("CapstoneControlPlaneService", "ListAnomalousLogGroups", {}).n("DevOpsGuruClient", "ListAnomalousLogGroupsCommand").f(void 0, void 0).ser(se_ListAnomalousLogGroupsCommand).de(de_ListAnomalousLogGroupsCommand).build() {
};
__name(_ListAnomalousLogGroupsCommand, "ListAnomalousLogGroupsCommand");
var ListAnomalousLogGroupsCommand = _ListAnomalousLogGroupsCommand;

// src/commands/ListEventsCommand.ts




var _ListEventsCommand = class _ListEventsCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("CapstoneControlPlaneService", "ListEvents", {}).n("DevOpsGuruClient", "ListEventsCommand").f(void 0, void 0).ser(se_ListEventsCommand).de(de_ListEventsCommand).build() {
};
__name(_ListEventsCommand, "ListEventsCommand");
var ListEventsCommand = _ListEventsCommand;

// src/commands/ListInsightsCommand.ts




var _ListInsightsCommand = class _ListInsightsCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("CapstoneControlPlaneService", "ListInsights", {}).n("DevOpsGuruClient", "ListInsightsCommand").f(void 0, void 0).ser(se_ListInsightsCommand).de(de_ListInsightsCommand).build() {
};
__name(_ListInsightsCommand, "ListInsightsCommand");
var ListInsightsCommand = _ListInsightsCommand;

// src/commands/ListMonitoredResourcesCommand.ts




var _ListMonitoredResourcesCommand = class _ListMonitoredResourcesCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("CapstoneControlPlaneService", "ListMonitoredResources", {}).n("DevOpsGuruClient", "ListMonitoredResourcesCommand").f(void 0, void 0).ser(se_ListMonitoredResourcesCommand).de(de_ListMonitoredResourcesCommand).build() {
};
__name(_ListMonitoredResourcesCommand, "ListMonitoredResourcesCommand");
var ListMonitoredResourcesCommand = _ListMonitoredResourcesCommand;

// src/commands/ListNotificationChannelsCommand.ts




var _ListNotificationChannelsCommand = class _ListNotificationChannelsCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("CapstoneControlPlaneService", "ListNotificationChannels", {}).n("DevOpsGuruClient", "ListNotificationChannelsCommand").f(void 0, void 0).ser(se_ListNotificationChannelsCommand).de(de_ListNotificationChannelsCommand).build() {
};
__name(_ListNotificationChannelsCommand, "ListNotificationChannelsCommand");
var ListNotificationChannelsCommand = _ListNotificationChannelsCommand;

// src/commands/ListOrganizationInsightsCommand.ts




var _ListOrganizationInsightsCommand = class _ListOrganizationInsightsCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("CapstoneControlPlaneService", "ListOrganizationInsights", {}).n("DevOpsGuruClient", "ListOrganizationInsightsCommand").f(void 0, void 0).ser(se_ListOrganizationInsightsCommand).de(de_ListOrganizationInsightsCommand).build() {
};
__name(_ListOrganizationInsightsCommand, "ListOrganizationInsightsCommand");
var ListOrganizationInsightsCommand = _ListOrganizationInsightsCommand;

// src/commands/ListRecommendationsCommand.ts




var _ListRecommendationsCommand = class _ListRecommendationsCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("CapstoneControlPlaneService", "ListRecommendations", {}).n("DevOpsGuruClient", "ListRecommendationsCommand").f(void 0, void 0).ser(se_ListRecommendationsCommand).de(de_ListRecommendationsCommand).build() {
};
__name(_ListRecommendationsCommand, "ListRecommendationsCommand");
var ListRecommendationsCommand = _ListRecommendationsCommand;

// src/commands/PutFeedbackCommand.ts




var _PutFeedbackCommand = class _PutFeedbackCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("CapstoneControlPlaneService", "PutFeedback", {}).n("DevOpsGuruClient", "PutFeedbackCommand").f(void 0, void 0).ser(se_PutFeedbackCommand).de(de_PutFeedbackCommand).build() {
};
__name(_PutFeedbackCommand, "PutFeedbackCommand");
var PutFeedbackCommand = _PutFeedbackCommand;

// src/commands/RemoveNotificationChannelCommand.ts




var _RemoveNotificationChannelCommand = class _RemoveNotificationChannelCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("CapstoneControlPlaneService", "RemoveNotificationChannel", {}).n("DevOpsGuruClient", "RemoveNotificationChannelCommand").f(void 0, void 0).ser(se_RemoveNotificationChannelCommand).de(de_RemoveNotificationChannelCommand).build() {
};
__name(_RemoveNotificationChannelCommand, "RemoveNotificationChannelCommand");
var RemoveNotificationChannelCommand = _RemoveNotificationChannelCommand;

// src/commands/SearchInsightsCommand.ts




var _SearchInsightsCommand = class _SearchInsightsCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("CapstoneControlPlaneService", "SearchInsights", {}).n("DevOpsGuruClient", "SearchInsightsCommand").f(void 0, void 0).ser(se_SearchInsightsCommand).de(de_SearchInsightsCommand).build() {
};
__name(_SearchInsightsCommand, "SearchInsightsCommand");
var SearchInsightsCommand = _SearchInsightsCommand;

// src/commands/SearchOrganizationInsightsCommand.ts




var _SearchOrganizationInsightsCommand = class _SearchOrganizationInsightsCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("CapstoneControlPlaneService", "SearchOrganizationInsights", {}).n("DevOpsGuruClient", "SearchOrganizationInsightsCommand").f(void 0, void 0).ser(se_SearchOrganizationInsightsCommand).de(de_SearchOrganizationInsightsCommand).build() {
};
__name(_SearchOrganizationInsightsCommand, "SearchOrganizationInsightsCommand");
var SearchOrganizationInsightsCommand = _SearchOrganizationInsightsCommand;

// src/commands/StartCostEstimationCommand.ts




var _StartCostEstimationCommand = class _StartCostEstimationCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("CapstoneControlPlaneService", "StartCostEstimation", {}).n("DevOpsGuruClient", "StartCostEstimationCommand").f(void 0, void 0).ser(se_StartCostEstimationCommand).de(de_StartCostEstimationCommand).build() {
};
__name(_StartCostEstimationCommand, "StartCostEstimationCommand");
var StartCostEstimationCommand = _StartCostEstimationCommand;

// src/commands/UpdateEventSourcesConfigCommand.ts




var _UpdateEventSourcesConfigCommand = class _UpdateEventSourcesConfigCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("CapstoneControlPlaneService", "UpdateEventSourcesConfig", {}).n("DevOpsGuruClient", "UpdateEventSourcesConfigCommand").f(void 0, void 0).ser(se_UpdateEventSourcesConfigCommand).de(de_UpdateEventSourcesConfigCommand).build() {
};
__name(_UpdateEventSourcesConfigCommand, "UpdateEventSourcesConfigCommand");
var UpdateEventSourcesConfigCommand = _UpdateEventSourcesConfigCommand;

// src/commands/UpdateResourceCollectionCommand.ts




var _UpdateResourceCollectionCommand = class _UpdateResourceCollectionCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("CapstoneControlPlaneService", "UpdateResourceCollection", {}).n("DevOpsGuruClient", "UpdateResourceCollectionCommand").f(void 0, void 0).ser(se_UpdateResourceCollectionCommand).de(de_UpdateResourceCollectionCommand).build() {
};
__name(_UpdateResourceCollectionCommand, "UpdateResourceCollectionCommand");
var UpdateResourceCollectionCommand = _UpdateResourceCollectionCommand;

// src/commands/UpdateServiceIntegrationCommand.ts




var _UpdateServiceIntegrationCommand = class _UpdateServiceIntegrationCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("CapstoneControlPlaneService", "UpdateServiceIntegration", {}).n("DevOpsGuruClient", "UpdateServiceIntegrationCommand").f(void 0, void 0).ser(se_UpdateServiceIntegrationCommand).de(de_UpdateServiceIntegrationCommand).build() {
};
__name(_UpdateServiceIntegrationCommand, "UpdateServiceIntegrationCommand");
var UpdateServiceIntegrationCommand = _UpdateServiceIntegrationCommand;

// src/DevOpsGuru.ts
var commands = {
  AddNotificationChannelCommand,
  DeleteInsightCommand,
  DescribeAccountHealthCommand,
  DescribeAccountOverviewCommand,
  DescribeAnomalyCommand,
  DescribeEventSourcesConfigCommand,
  DescribeFeedbackCommand,
  DescribeInsightCommand,
  DescribeOrganizationHealthCommand,
  DescribeOrganizationOverviewCommand,
  DescribeOrganizationResourceCollectionHealthCommand,
  DescribeResourceCollectionHealthCommand,
  DescribeServiceIntegrationCommand,
  GetCostEstimationCommand,
  GetResourceCollectionCommand,
  ListAnomaliesForInsightCommand,
  ListAnomalousLogGroupsCommand,
  ListEventsCommand,
  ListInsightsCommand,
  ListMonitoredResourcesCommand,
  ListNotificationChannelsCommand,
  ListOrganizationInsightsCommand,
  ListRecommendationsCommand,
  PutFeedbackCommand,
  RemoveNotificationChannelCommand,
  SearchInsightsCommand,
  SearchOrganizationInsightsCommand,
  StartCostEstimationCommand,
  UpdateEventSourcesConfigCommand,
  UpdateResourceCollectionCommand,
  UpdateServiceIntegrationCommand
};
var _DevOpsGuru = class _DevOpsGuru extends DevOpsGuruClient {
};
__name(_DevOpsGuru, "DevOpsGuru");
var DevOpsGuru = _DevOpsGuru;
(0, import_smithy_client.createAggregatedClient)(commands, DevOpsGuru);

// src/pagination/DescribeOrganizationResourceCollectionHealthPaginator.ts

var paginateDescribeOrganizationResourceCollectionHealth = (0, import_core.createPaginator)(DevOpsGuruClient, DescribeOrganizationResourceCollectionHealthCommand, "NextToken", "NextToken", "");

// src/pagination/DescribeResourceCollectionHealthPaginator.ts

var paginateDescribeResourceCollectionHealth = (0, import_core.createPaginator)(DevOpsGuruClient, DescribeResourceCollectionHealthCommand, "NextToken", "NextToken", "");

// src/pagination/GetCostEstimationPaginator.ts

var paginateGetCostEstimation = (0, import_core.createPaginator)(DevOpsGuruClient, GetCostEstimationCommand, "NextToken", "NextToken", "");

// src/pagination/GetResourceCollectionPaginator.ts

var paginateGetResourceCollection = (0, import_core.createPaginator)(DevOpsGuruClient, GetResourceCollectionCommand, "NextToken", "NextToken", "");

// src/pagination/ListAnomaliesForInsightPaginator.ts

var paginateListAnomaliesForInsight = (0, import_core.createPaginator)(DevOpsGuruClient, ListAnomaliesForInsightCommand, "NextToken", "NextToken", "MaxResults");

// src/pagination/ListAnomalousLogGroupsPaginator.ts

var paginateListAnomalousLogGroups = (0, import_core.createPaginator)(DevOpsGuruClient, ListAnomalousLogGroupsCommand, "NextToken", "NextToken", "MaxResults");

// src/pagination/ListEventsPaginator.ts

var paginateListEvents = (0, import_core.createPaginator)(DevOpsGuruClient, ListEventsCommand, "NextToken", "NextToken", "MaxResults");

// src/pagination/ListInsightsPaginator.ts

var paginateListInsights = (0, import_core.createPaginator)(DevOpsGuruClient, ListInsightsCommand, "NextToken", "NextToken", "MaxResults");

// src/pagination/ListMonitoredResourcesPaginator.ts

var paginateListMonitoredResources = (0, import_core.createPaginator)(DevOpsGuruClient, ListMonitoredResourcesCommand, "NextToken", "NextToken", "MaxResults");

// src/pagination/ListNotificationChannelsPaginator.ts

var paginateListNotificationChannels = (0, import_core.createPaginator)(DevOpsGuruClient, ListNotificationChannelsCommand, "NextToken", "NextToken", "");

// src/pagination/ListOrganizationInsightsPaginator.ts

var paginateListOrganizationInsights = (0, import_core.createPaginator)(DevOpsGuruClient, ListOrganizationInsightsCommand, "NextToken", "NextToken", "MaxResults");

// src/pagination/ListRecommendationsPaginator.ts

var paginateListRecommendations = (0, import_core.createPaginator)(DevOpsGuruClient, ListRecommendationsCommand, "NextToken", "NextToken", "");

// src/pagination/SearchInsightsPaginator.ts

var paginateSearchInsights = (0, import_core.createPaginator)(DevOpsGuruClient, SearchInsightsCommand, "NextToken", "NextToken", "MaxResults");

// src/pagination/SearchOrganizationInsightsPaginator.ts

var paginateSearchOrganizationInsights = (0, import_core.createPaginator)(DevOpsGuruClient, SearchOrganizationInsightsCommand, "NextToken", "NextToken", "MaxResults");
// Annotate the CommonJS export names for ESM import in node:

0 && (module.exports = {
  DevOpsGuruServiceException,
  __Client,
  DevOpsGuruClient,
  DevOpsGuru,
  $Command,
  AddNotificationChannelCommand,
  DeleteInsightCommand,
  DescribeAccountHealthCommand,
  DescribeAccountOverviewCommand,
  DescribeAnomalyCommand,
  DescribeEventSourcesConfigCommand,
  DescribeFeedbackCommand,
  DescribeInsightCommand,
  DescribeOrganizationHealthCommand,
  DescribeOrganizationOverviewCommand,
  DescribeOrganizationResourceCollectionHealthCommand,
  DescribeResourceCollectionHealthCommand,
  DescribeServiceIntegrationCommand,
  GetCostEstimationCommand,
  GetResourceCollectionCommand,
  ListAnomaliesForInsightCommand,
  ListAnomalousLogGroupsCommand,
  ListEventsCommand,
  ListInsightsCommand,
  ListMonitoredResourcesCommand,
  ListNotificationChannelsCommand,
  ListOrganizationInsightsCommand,
  ListRecommendationsCommand,
  PutFeedbackCommand,
  RemoveNotificationChannelCommand,
  SearchInsightsCommand,
  SearchOrganizationInsightsCommand,
  StartCostEstimationCommand,
  UpdateEventSourcesConfigCommand,
  UpdateResourceCollectionCommand,
  UpdateServiceIntegrationCommand,
  paginateDescribeOrganizationResourceCollectionHealth,
  paginateDescribeResourceCollectionHealth,
  paginateGetCostEstimation,
  paginateGetResourceCollection,
  paginateListAnomaliesForInsight,
  paginateListAnomalousLogGroups,
  paginateListEvents,
  paginateListInsights,
  paginateListMonitoredResources,
  paginateListNotificationChannels,
  paginateListOrganizationInsights,
  paginateListRecommendations,
  paginateSearchInsights,
  paginateSearchOrganizationInsights,
  AccessDeniedException,
  NotificationMessageType,
  InsightSeverity,
  ConflictException,
  InternalServerException,
  ResourceNotFoundException,
  ServiceQuotaExceededException,
  ThrottlingException,
  ValidationExceptionReason,
  ValidationException,
  EventSourceOptInStatus,
  LogAnomalyType,
  AnomalySeverity,
  CloudWatchMetricDataStatusCode,
  CloudWatchMetricsStat,
  AnomalyStatus,
  AnomalyType,
  InsightFeedbackOption,
  InsightStatus,
  OrganizationResourceCollectionType,
  ServiceName,
  ResourceCollectionType,
  OptInStatus,
  ServerSideEncryptionType,
  CostEstimationServiceResourceState,
  CostEstimationStatus,
  EventDataSource,
  EventClass,
  InsightType,
  ResourcePermission,
  ResourceTypeFilter,
  Locale,
  UpdateResourceCollectionAction
});
