// @azure/arm-apimanagement/dist-esm/test/apimanagement_examples.js
/*
 * Copyright (c) Microsoft Corporation.
 * Licensed under the MIT License.
 *
 * Code generated by Microsoft (R) AutoRest Code Generator.
 * Changes may cause incorrect behavior and will be lost if the code is regenerated.
 */
import { __asyncValues, __awaiter } from "tslib";
import { env, Recorder, delay, isPlaybackMode, } from "@azure-tools/test-recorder";
import { createTestCredential } from "@azure-tools/test-credential";
import { assert } from "chai";
import { ApiManagementClient } from "../src/apiManagementClient";
const replaceableVariables = {
    AZURE_CLIENT_ID: "azure_client_id",
    AZURE_CLIENT_SECRET: "azure_client_secret",
    AZURE_TENANT_ID: "88888888-8888-8888-8888-888888888888",
    SUBSCRIPTION_ID: "azure_subscription_id"
};
const recorderOptions = {
    envSetupForPlayback: replaceableVariables
};
export const testPollingOptions = {
    updateIntervalInMs: isPlaybackMode() ? 0 : undefined,
};
describe("Apimanagement test", () => {
    let recorder;
    let subscriptionId;
    let client;
    let location;
    let resourceGroupName;
    let serviceName;
    beforeEach(function () {
        return __awaiter(this, void 0, void 0, function* () {
            recorder = new Recorder(this.currentTest);
            yield recorder.start(recorderOptions);
            subscriptionId = env.SUBSCRIPTION_ID || '';
            // This is an example of how the environment variables are used
            const credential = createTestCredential();
            client = new ApiManagementClient(credential, subscriptionId, recorder.configureClientOptions({}));
            location = "eastus";
            resourceGroupName = "myjstest";
            serviceName = "myserviceyyy1";
        });
    });
    afterEach(function () {
        return __awaiter(this, void 0, void 0, function* () {
            yield recorder.stop();
        });
    });
    function sleep(ms) {
        return new Promise(resolve => setTimeout(resolve, ms));
    }
    it("apiManagementService create test", function () {
        return __awaiter(this, void 0, void 0, function* () {
            const res = yield client.apiManagementService.beginCreateOrUpdateAndWait(resourceGroupName, serviceName, {
                location: location,
                sku: {
                    name: "Standard",
                    capacity: 1
                },
                publisherEmail: "foo@contoso.com",
                publisherName: "foo"
            }, testPollingOptions);
            assert.equal(res.name, serviceName);
        });
    }).timeout(3600000);
    it("apiManagementService get test", function () {
        return __awaiter(this, void 0, void 0, function* () {
            const res = yield client.apiManagementService.get(resourceGroupName, serviceName);
            assert.equal(res.name, serviceName);
        });
    });
    it("apiManagementService listByResourceGroup test", function () {
        var _a, e_1, _b, _c;
        return __awaiter(this, void 0, void 0, function* () {
            const resArray = new Array();
            try {
                for (var _d = true, _e = __asyncValues(client.apiManagementService.listByResourceGroup(resourceGroupName)), _f; _f = yield _e.next(), _a = _f.done, !_a;) {
                    _c = _f.value;
                    _d = false;
                    try {
                        let item = _c;
                        resArray.push(item);
                    }
                    finally {
                        _d = true;
                    }
                }
            }
            catch (e_1_1) { e_1 = { error: e_1_1 }; }
            finally {
                try {
                    if (!_d && !_a && (_b = _e.return)) yield _b.call(_e);
                }
                finally { if (e_1) throw e_1.error; }
            }
            assert.equal(resArray.length, 1);
        });
    });
    it("apiManagementService update test", function () {
        return __awaiter(this, void 0, void 0, function* () {
            // this.timeout(3600000);
            let count = 0;
            while (count < 20) {
                count++;
                const res = yield client.apiManagementService.get(resourceGroupName, serviceName);
                if (res.provisioningState == "Succeeded") {
                    const res = yield client.apiManagementService.beginUpdateAndWait(resourceGroupName, serviceName, {
                        customProperties: {
                            "Microsoft.WindowsAzure.ApiManagement.Gateway.Security.Protocols.Tls10": "false"
                        }
                    }, testPollingOptions);
                    assert.equal(res.type, "Microsoft.ApiManagement/service");
                    break;
                }
                else {
                    // The resource is activating
                    yield delay(300000);
                }
            }
        });
    }).timeout(3600000);
    it("backend create test1", function () {
        return __awaiter(this, void 0, void 0, function* () {
            const result = yield client.backend.createOrUpdate(resourceGroupName, serviceName, "sfbackend1", {
                description: "Service Fabric Test App 1",
                url: "https://backendname26441",
                protocol: "http"
            });
        });
    });
    it("backend create test2", function () {
        return __awaiter(this, void 0, void 0, function* () {
            const result = yield client.backend.createOrUpdate(resourceGroupName, serviceName, "sfbackend2", {
                description: "Service Fabric Test App 1",
                url: "https://backendname26442",
                protocol: "http"
            });
        });
    });
    it("backend list test", function () {
        var _a, e_2, _b, _c;
        return __awaiter(this, void 0, void 0, function* () {
            const resArray = new Array();
            try {
                for (var _d = true, _e = __asyncValues(client.backend.listByService(resourceGroupName, serviceName, { top: 1 })), _f; _f = yield _e.next(), _a = _f.done, !_a;) {
                    _c = _f.value;
                    _d = false;
                    try {
                        let item = _c;
                        resArray.push(item);
                    }
                    finally {
                        _d = true;
                    }
                }
            }
            catch (e_2_1) { e_2 = { error: e_2_1 }; }
            finally {
                try {
                    if (!_d && !_a && (_b = _e.return)) yield _b.call(_e);
                }
                finally { if (e_2) throw e_2.error; }
            }
            assert.equal(resArray.length, 2);
        });
    });
    it("backend delete test", function () {
        var _a, e_3, _b, _c;
        return __awaiter(this, void 0, void 0, function* () {
            const res1 = yield client.backend.delete(resourceGroupName, serviceName, "sfbackend1", "*");
            const res2 = yield client.backend.delete(resourceGroupName, serviceName, "sfbackend2", "*");
            const resArray = new Array();
            try {
                for (var _d = true, _e = __asyncValues(client.backend.listByService(resourceGroupName, serviceName)), _f; _f = yield _e.next(), _a = _f.done, !_a;) {
                    _c = _f.value;
                    _d = false;
                    try {
                        let item = _c;
                        resArray.push(item);
                    }
                    finally {
                        _d = true;
                    }
                }
            }
            catch (e_3_1) { e_3 = { error: e_3_1 }; }
            finally {
                try {
                    if (!_d && !_a && (_b = _e.return)) yield _b.call(_e);
                }
                finally { if (e_3) throw e_3.error; }
            }
            assert.equal(resArray.length, 0);
        });
    });
    it("apiManagementService delete test", function () {
        var _a, e_4, _b, _c;
        return __awaiter(this, void 0, void 0, function* () {
            let count = 0;
            while (count < 20) {
                count++;
                const res = yield client.apiManagementService.get(resourceGroupName, serviceName);
                if (res.provisioningState == "Succeeded") {
                    const res = yield client.apiManagementService.beginDeleteAndWait(resourceGroupName, serviceName, testPollingOptions);
                    const purge_resource = yield client.deletedServices.beginPurgeAndWait(serviceName, location, testPollingOptions);
                    const resArray = new Array();
                    try {
                        for (var _d = true, _e = (e_4 = void 0, __asyncValues(client.apiManagementService.listByResourceGroup(resourceGroupName))), _f; _f = yield _e.next(), _a = _f.done, !_a;) {
                            _c = _f.value;
                            _d = false;
                            try {
                                let item = _c;
                                resArray.push(item);
                            }
                            finally {
                                _d = true;
                            }
                        }
                    }
                    catch (e_4_1) { e_4 = { error: e_4_1 }; }
                    finally {
                        try {
                            if (!_d && !_a && (_b = _e.return)) yield _b.call(_e);
                        }
                        finally { if (e_4) throw e_4.error; }
                    }
                    assert.equal(resArray.length, 0);
                    break;
                }
                else {
                    // The resource is activating
                    yield delay(300000);
                }
            }
        });
    }).timeout(3600000);
});
