// @azure/arm-chaos/dist-esm/src/models/index.js
/*
 * Copyright (c) Microsoft Corporation.
 * Licensed under the MIT License.
 *
 * Code generated by Microsoft (R) AutoRest Code Generator.
 * Changes may cause incorrect behavior and will be lost if the code is regenerated.
 */
/** Known values of {@link CreatedByType} that the service accepts. */
export var KnownCreatedByType;
(function (KnownCreatedByType) {
    /** User */
    KnownCreatedByType["User"] = "User";
    /** Application */
    KnownCreatedByType["Application"] = "Application";
    /** ManagedIdentity */
    KnownCreatedByType["ManagedIdentity"] = "ManagedIdentity";
    /** Key */
    KnownCreatedByType["Key"] = "Key";
})(KnownCreatedByType || (KnownCreatedByType = {}));
/** Known values of {@link ProvisioningState} that the service accepts. */
export var KnownProvisioningState;
(function (KnownProvisioningState) {
    /** Succeeded */
    KnownProvisioningState["Succeeded"] = "Succeeded";
    /** Failed */
    KnownProvisioningState["Failed"] = "Failed";
    /** Canceled */
    KnownProvisioningState["Canceled"] = "Canceled";
    /** Creating */
    KnownProvisioningState["Creating"] = "Creating";
    /** Updating */
    KnownProvisioningState["Updating"] = "Updating";
    /** Deleting */
    KnownProvisioningState["Deleting"] = "Deleting";
})(KnownProvisioningState || (KnownProvisioningState = {}));
/** Known values of {@link SelectorType} that the service accepts. */
export var KnownSelectorType;
(function (KnownSelectorType) {
    /** List */
    KnownSelectorType["List"] = "List";
    /** Query */
    KnownSelectorType["Query"] = "Query";
})(KnownSelectorType || (KnownSelectorType = {}));
/** Known values of {@link FilterType} that the service accepts. */
export var KnownFilterType;
(function (KnownFilterType) {
    /** Simple */
    KnownFilterType["Simple"] = "Simple";
})(KnownFilterType || (KnownFilterType = {}));
/** Known values of {@link Origin} that the service accepts. */
export var KnownOrigin;
(function (KnownOrigin) {
    /** User */
    KnownOrigin["User"] = "user";
    /** System */
    KnownOrigin["System"] = "system";
    /** UserSystem */
    KnownOrigin["UserSystem"] = "user,system";
})(KnownOrigin || (KnownOrigin = {}));
/** Known values of {@link ActionType} that the service accepts. */
export var KnownActionType;
(function (KnownActionType) {
    /** Internal */
    KnownActionType["Internal"] = "Internal";
})(KnownActionType || (KnownActionType = {}));
/** Known values of {@link TargetReferenceType} that the service accepts. */
export var KnownTargetReferenceType;
(function (KnownTargetReferenceType) {
    /** ChaosTarget */
    KnownTargetReferenceType["ChaosTarget"] = "ChaosTarget";
})(KnownTargetReferenceType || (KnownTargetReferenceType = {}));
