// @azure/arm-trafficmanager/dist-esm/src/operations/trafficManagerUserMetricsKeys.js
/*
 * Copyright (c) Microsoft Corporation.
 * Licensed under the MIT License.
 *
 * Code generated by Microsoft (R) AutoRest Code Generator.
 * Changes may cause incorrect behavior and will be lost if the code is regenerated.
 */
import * as coreClient from "@azure/core-client";
import * as Mappers from "../models/mappers";
import * as Parameters from "../models/parameters";
/** Class containing TrafficManagerUserMetricsKeys operations. */
export class TrafficManagerUserMetricsKeysImpl {
    /**
     * Initialize a new instance of the class TrafficManagerUserMetricsKeys class.
     * @param client Reference to the service client
     */
    constructor(client) {
        this.client = client;
    }
    /**
     * Get the subscription-level key used for Real User Metrics collection.
     * @param options The options parameters.
     */
    get(options) {
        return this.client.sendOperationRequest({ options }, getOperationSpec);
    }
    /**
     * Create or update a subscription-level key used for Real User Metrics collection.
     * @param options The options parameters.
     */
    createOrUpdate(options) {
        return this.client.sendOperationRequest({ options }, createOrUpdateOperationSpec);
    }
    /**
     * Delete a subscription-level key used for Real User Metrics collection.
     * @param options The options parameters.
     */
    delete(options) {
        return this.client.sendOperationRequest({ options }, deleteOperationSpec);
    }
}
// Operation Specifications
const serializer = coreClient.createSerializer(Mappers, /* isXml */ false);
const getOperationSpec = {
    path: "/subscriptions/{subscriptionId}/providers/Microsoft.Network/trafficManagerUserMetricsKeys/default",
    httpMethod: "GET",
    responses: {
        200: {
            bodyMapper: Mappers.UserMetricsModel
        },
        default: {
            bodyMapper: Mappers.CloudError
        }
    },
    queryParameters: [Parameters.apiVersion],
    urlParameters: [Parameters.$host, Parameters.subscriptionId],
    headerParameters: [Parameters.accept],
    serializer
};
const createOrUpdateOperationSpec = {
    path: "/subscriptions/{subscriptionId}/providers/Microsoft.Network/trafficManagerUserMetricsKeys/default",
    httpMethod: "PUT",
    responses: {
        201: {
            bodyMapper: Mappers.UserMetricsModel
        },
        default: {
            bodyMapper: Mappers.CloudError
        }
    },
    queryParameters: [Parameters.apiVersion],
    urlParameters: [Parameters.$host, Parameters.subscriptionId],
    headerParameters: [Parameters.accept],
    serializer
};
const deleteOperationSpec = {
    path: "/subscriptions/{subscriptionId}/providers/Microsoft.Network/trafficManagerUserMetricsKeys/default",
    httpMethod: "DELETE",
    responses: {
        200: {
            bodyMapper: Mappers.DeleteOperationResult
        },
        default: {
            bodyMapper: Mappers.CloudError
        }
    },
    queryParameters: [Parameters.apiVersion],
    urlParameters: [Parameters.$host, Parameters.subscriptionId],
    headerParameters: [Parameters.accept],
    serializer
};
