// @azure/arm-rediscache/dist-esm/src/models/index.js
/*
 * Copyright (c) Microsoft Corporation.
 * Licensed under the MIT License.
 *
 * Code generated by Microsoft (R) AutoRest Code Generator.
 * Changes may cause incorrect behavior and will be lost if the code is regenerated.
 */
/** Known values of {@link SkuName} that the service accepts. */
export var KnownSkuName;
(function (KnownSkuName) {
    /** Basic */
    KnownSkuName["Basic"] = "Basic";
    /** Standard */
    KnownSkuName["Standard"] = "Standard";
    /** Premium */
    KnownSkuName["Premium"] = "Premium";
})(KnownSkuName || (KnownSkuName = {}));
/** Known values of {@link SkuFamily} that the service accepts. */
export var KnownSkuFamily;
(function (KnownSkuFamily) {
    /** C */
    KnownSkuFamily["C"] = "C";
    /** P */
    KnownSkuFamily["P"] = "P";
})(KnownSkuFamily || (KnownSkuFamily = {}));
/** Known values of {@link TlsVersion} that the service accepts. */
export var KnownTlsVersion;
(function (KnownTlsVersion) {
    /** One0 */
    KnownTlsVersion["One0"] = "1.0";
    /** One1 */
    KnownTlsVersion["One1"] = "1.1";
    /** One2 */
    KnownTlsVersion["One2"] = "1.2";
})(KnownTlsVersion || (KnownTlsVersion = {}));
/** Known values of {@link PublicNetworkAccess} that the service accepts. */
export var KnownPublicNetworkAccess;
(function (KnownPublicNetworkAccess) {
    /** Enabled */
    KnownPublicNetworkAccess["Enabled"] = "Enabled";
    /** Disabled */
    KnownPublicNetworkAccess["Disabled"] = "Disabled";
})(KnownPublicNetworkAccess || (KnownPublicNetworkAccess = {}));
/** Known values of {@link UpdateChannel} that the service accepts. */
export var KnownUpdateChannel;
(function (KnownUpdateChannel) {
    /** Stable */
    KnownUpdateChannel["Stable"] = "Stable";
    /** Preview */
    KnownUpdateChannel["Preview"] = "Preview";
})(KnownUpdateChannel || (KnownUpdateChannel = {}));
/** Known values of {@link ManagedServiceIdentityType} that the service accepts. */
export var KnownManagedServiceIdentityType;
(function (KnownManagedServiceIdentityType) {
    /** None */
    KnownManagedServiceIdentityType["None"] = "None";
    /** SystemAssigned */
    KnownManagedServiceIdentityType["SystemAssigned"] = "SystemAssigned";
    /** UserAssigned */
    KnownManagedServiceIdentityType["UserAssigned"] = "UserAssigned";
    /** SystemAssignedUserAssigned */
    KnownManagedServiceIdentityType["SystemAssignedUserAssigned"] = "SystemAssigned, UserAssigned";
})(KnownManagedServiceIdentityType || (KnownManagedServiceIdentityType = {}));
/** Known values of {@link ProvisioningState} that the service accepts. */
export var KnownProvisioningState;
(function (KnownProvisioningState) {
    /** Creating */
    KnownProvisioningState["Creating"] = "Creating";
    /** Deleting */
    KnownProvisioningState["Deleting"] = "Deleting";
    /** Disabled */
    KnownProvisioningState["Disabled"] = "Disabled";
    /** Failed */
    KnownProvisioningState["Failed"] = "Failed";
    /** Linking */
    KnownProvisioningState["Linking"] = "Linking";
    /** Provisioning */
    KnownProvisioningState["Provisioning"] = "Provisioning";
    /** RecoveringScaleFailure */
    KnownProvisioningState["RecoveringScaleFailure"] = "RecoveringScaleFailure";
    /** Scaling */
    KnownProvisioningState["Scaling"] = "Scaling";
    /** Succeeded */
    KnownProvisioningState["Succeeded"] = "Succeeded";
    /** Unlinking */
    KnownProvisioningState["Unlinking"] = "Unlinking";
    /** Unprovisioning */
    KnownProvisioningState["Unprovisioning"] = "Unprovisioning";
    /** Updating */
    KnownProvisioningState["Updating"] = "Updating";
    /** ConfiguringAAD */
    KnownProvisioningState["ConfiguringAAD"] = "ConfiguringAAD";
})(KnownProvisioningState || (KnownProvisioningState = {}));
/** Known values of {@link PrivateEndpointServiceConnectionStatus} that the service accepts. */
export var KnownPrivateEndpointServiceConnectionStatus;
(function (KnownPrivateEndpointServiceConnectionStatus) {
    /** Pending */
    KnownPrivateEndpointServiceConnectionStatus["Pending"] = "Pending";
    /** Approved */
    KnownPrivateEndpointServiceConnectionStatus["Approved"] = "Approved";
    /** Rejected */
    KnownPrivateEndpointServiceConnectionStatus["Rejected"] = "Rejected";
})(KnownPrivateEndpointServiceConnectionStatus || (KnownPrivateEndpointServiceConnectionStatus = {}));
/** Known values of {@link PrivateEndpointConnectionProvisioningState} that the service accepts. */
export var KnownPrivateEndpointConnectionProvisioningState;
(function (KnownPrivateEndpointConnectionProvisioningState) {
    /** Succeeded */
    KnownPrivateEndpointConnectionProvisioningState["Succeeded"] = "Succeeded";
    /** Creating */
    KnownPrivateEndpointConnectionProvisioningState["Creating"] = "Creating";
    /** Deleting */
    KnownPrivateEndpointConnectionProvisioningState["Deleting"] = "Deleting";
    /** Failed */
    KnownPrivateEndpointConnectionProvisioningState["Failed"] = "Failed";
})(KnownPrivateEndpointConnectionProvisioningState || (KnownPrivateEndpointConnectionProvisioningState = {}));
/** Known values of {@link RebootType} that the service accepts. */
export var KnownRebootType;
(function (KnownRebootType) {
    /** PrimaryNode */
    KnownRebootType["PrimaryNode"] = "PrimaryNode";
    /** SecondaryNode */
    KnownRebootType["SecondaryNode"] = "SecondaryNode";
    /** AllNodes */
    KnownRebootType["AllNodes"] = "AllNodes";
})(KnownRebootType || (KnownRebootType = {}));
/** Known values of {@link DefaultName} that the service accepts. */
export var KnownDefaultName;
(function (KnownDefaultName) {
    /** Default */
    KnownDefaultName["Default"] = "default";
})(KnownDefaultName || (KnownDefaultName = {}));
/** Known values of {@link AccessPolicyProvisioningState} that the service accepts. */
export var KnownAccessPolicyProvisioningState;
(function (KnownAccessPolicyProvisioningState) {
    /** Updating */
    KnownAccessPolicyProvisioningState["Updating"] = "Updating";
    /** Succeeded */
    KnownAccessPolicyProvisioningState["Succeeded"] = "Succeeded";
    /** Deleting */
    KnownAccessPolicyProvisioningState["Deleting"] = "Deleting";
    /** Deleted */
    KnownAccessPolicyProvisioningState["Deleted"] = "Deleted";
    /** Canceled */
    KnownAccessPolicyProvisioningState["Canceled"] = "Canceled";
    /** Failed */
    KnownAccessPolicyProvisioningState["Failed"] = "Failed";
})(KnownAccessPolicyProvisioningState || (KnownAccessPolicyProvisioningState = {}));
/** Known values of {@link AccessPolicyType} that the service accepts. */
export var KnownAccessPolicyType;
(function (KnownAccessPolicyType) {
    /** Custom */
    KnownAccessPolicyType["Custom"] = "Custom";
    /** BuiltIn */
    KnownAccessPolicyType["BuiltIn"] = "BuiltIn";
})(KnownAccessPolicyType || (KnownAccessPolicyType = {}));
/** Known values of {@link AccessPolicyAssignmentProvisioningState} that the service accepts. */
export var KnownAccessPolicyAssignmentProvisioningState;
(function (KnownAccessPolicyAssignmentProvisioningState) {
    /** Updating */
    KnownAccessPolicyAssignmentProvisioningState["Updating"] = "Updating";
    /** Succeeded */
    KnownAccessPolicyAssignmentProvisioningState["Succeeded"] = "Succeeded";
    /** Deleting */
    KnownAccessPolicyAssignmentProvisioningState["Deleting"] = "Deleting";
    /** Deleted */
    KnownAccessPolicyAssignmentProvisioningState["Deleted"] = "Deleted";
    /** Canceled */
    KnownAccessPolicyAssignmentProvisioningState["Canceled"] = "Canceled";
    /** Failed */
    KnownAccessPolicyAssignmentProvisioningState["Failed"] = "Failed";
})(KnownAccessPolicyAssignmentProvisioningState || (KnownAccessPolicyAssignmentProvisioningState = {}));
