// @azure/arm-hdinsight/dist-esm/src/models/parameters.js
/*
 * Copyright (c) Microsoft Corporation.
 * Licensed under the MIT License.
 *
 * Code generated by Microsoft (R) AutoRest Code Generator.
 * Changes may cause incorrect behavior and will be lost if the code is regenerated.
 */
import { ClusterCreateParametersExtended as ClusterCreateParametersExtendedMapper, ClusterPatchParameters as ClusterPatchParametersMapper, ClusterResizeParameters as ClusterResizeParametersMapper, AutoscaleConfigurationUpdateParameter as AutoscaleConfigurationUpdateParameterMapper, ClusterDiskEncryptionParameters as ClusterDiskEncryptionParametersMapper, UpdateGatewaySettingsParameters as UpdateGatewaySettingsParametersMapper, UpdateClusterIdentityCertificateParameters as UpdateClusterIdentityCertificateParametersMapper, ExecuteScriptActionParameters as ExecuteScriptActionParametersMapper, Application as ApplicationMapper, NameAvailabilityCheckRequestParameters as NameAvailabilityCheckRequestParametersMapper, ClusterCreateRequestValidationParameters as ClusterCreateRequestValidationParametersMapper, ClusterMonitoringRequest as ClusterMonitoringRequestMapper, AzureMonitorRequest as AzureMonitorRequestMapper, Extension as ExtensionMapper, PrivateEndpointConnection as PrivateEndpointConnectionMapper } from "../models/mappers";
export const contentType = {
    parameterPath: ["options", "contentType"],
    mapper: {
        defaultValue: "application/json",
        isConstant: true,
        serializedName: "Content-Type",
        type: {
            name: "String"
        }
    }
};
export const parameters = {
    parameterPath: "parameters",
    mapper: ClusterCreateParametersExtendedMapper
};
export const accept = {
    parameterPath: "accept",
    mapper: {
        defaultValue: "application/json",
        isConstant: true,
        serializedName: "Accept",
        type: {
            name: "String"
        }
    }
};
export const $host = {
    parameterPath: "$host",
    mapper: {
        serializedName: "$host",
        required: true,
        type: {
            name: "String"
        }
    },
    skipEncoding: true
};
export const subscriptionId = {
    parameterPath: "subscriptionId",
    mapper: {
        serializedName: "subscriptionId",
        required: true,
        type: {
            name: "String"
        }
    }
};
export const resourceGroupName = {
    parameterPath: "resourceGroupName",
    mapper: {
        serializedName: "resourceGroupName",
        required: true,
        type: {
            name: "String"
        }
    }
};
export const clusterName = {
    parameterPath: "clusterName",
    mapper: {
        serializedName: "clusterName",
        required: true,
        type: {
            name: "String"
        }
    }
};
export const apiVersion = {
    parameterPath: "apiVersion",
    mapper: {
        defaultValue: "2021-06-01",
        isConstant: true,
        serializedName: "api-version",
        type: {
            name: "String"
        }
    }
};
export const parameters1 = {
    parameterPath: "parameters",
    mapper: ClusterPatchParametersMapper
};
export const parameters2 = {
    parameterPath: "parameters",
    mapper: ClusterResizeParametersMapper
};
export const roleName = {
    parameterPath: "roleName",
    mapper: {
        serializedName: "roleName",
        required: true,
        type: {
            name: "String"
        }
    }
};
export const parameters3 = {
    parameterPath: "parameters",
    mapper: AutoscaleConfigurationUpdateParameterMapper
};
export const parameters4 = {
    parameterPath: "parameters",
    mapper: ClusterDiskEncryptionParametersMapper
};
export const parameters5 = {
    parameterPath: "parameters",
    mapper: UpdateGatewaySettingsParametersMapper
};
export const operationId = {
    parameterPath: "operationId",
    mapper: {
        serializedName: "operationId",
        required: true,
        type: {
            name: "String"
        }
    }
};
export const parameters6 = {
    parameterPath: "parameters",
    mapper: UpdateClusterIdentityCertificateParametersMapper
};
export const parameters7 = {
    parameterPath: "parameters",
    mapper: ExecuteScriptActionParametersMapper
};
export const nextLink = {
    parameterPath: "nextLink",
    mapper: {
        serializedName: "nextLink",
        required: true,
        type: {
            name: "String"
        }
    },
    skipEncoding: true
};
export const applicationName = {
    parameterPath: "applicationName",
    mapper: {
        serializedName: "applicationName",
        required: true,
        type: {
            name: "String"
        }
    }
};
export const parameters8 = {
    parameterPath: "parameters",
    mapper: ApplicationMapper
};
export const location = {
    parameterPath: "location",
    mapper: {
        serializedName: "location",
        required: true,
        type: {
            name: "String"
        }
    }
};
export const parameters9 = {
    parameterPath: "parameters",
    mapper: NameAvailabilityCheckRequestParametersMapper
};
export const parameters10 = {
    parameterPath: "parameters",
    mapper: ClusterCreateRequestValidationParametersMapper
};
export const parameters11 = {
    parameterPath: "parameters",
    mapper: {
        serializedName: "parameters",
        required: true,
        type: {
            name: "Dictionary",
            value: { type: { name: "String" } }
        }
    }
};
export const configurationName = {
    parameterPath: "configurationName",
    mapper: {
        serializedName: "configurationName",
        required: true,
        type: {
            name: "String"
        }
    }
};
export const parameters12 = {
    parameterPath: "parameters",
    mapper: ClusterMonitoringRequestMapper
};
export const parameters13 = {
    parameterPath: "parameters",
    mapper: AzureMonitorRequestMapper
};
export const parameters14 = {
    parameterPath: "parameters",
    mapper: ExtensionMapper
};
export const extensionName = {
    parameterPath: "extensionName",
    mapper: {
        serializedName: "extensionName",
        required: true,
        type: {
            name: "String"
        }
    }
};
export const scriptName = {
    parameterPath: "scriptName",
    mapper: {
        serializedName: "scriptName",
        required: true,
        type: {
            name: "String"
        }
    }
};
export const scriptExecutionId = {
    parameterPath: "scriptExecutionId",
    mapper: {
        serializedName: "scriptExecutionId",
        required: true,
        type: {
            name: "String"
        }
    }
};
export const hosts = {
    parameterPath: "hosts",
    mapper: {
        serializedName: "hosts",
        required: true,
        type: {
            name: "Sequence",
            element: {
                type: {
                    name: "String"
                }
            }
        }
    }
};
export const parameters15 = {
    parameterPath: "parameters",
    mapper: PrivateEndpointConnectionMapper
};
export const privateEndpointConnectionName = {
    parameterPath: "privateEndpointConnectionName",
    mapper: {
        serializedName: "privateEndpointConnectionName",
        required: true,
        type: {
            name: "String"
        }
    }
};
export const privateLinkResourceName = {
    parameterPath: "privateLinkResourceName",
    mapper: {
        serializedName: "privateLinkResourceName",
        required: true,
        type: {
            name: "String"
        }
    }
};
