// @azure/arm-securityinsight/dist-esm/src/operations/actions.js
/*
 * Copyright (c) Microsoft Corporation.
 * Licensed under the MIT License.
 *
 * Code generated by Microsoft (R) AutoRest Code Generator.
 * Changes may cause incorrect behavior and will be lost if the code is regenerated.
 */
import { __asyncDelegator, __asyncGenerator, __asyncValues, __await } from "tslib";
import { setContinuationToken } from "../pagingHelper";
import * as coreClient from "@azure/core-client";
import * as Mappers from "../models/mappers";
import * as Parameters from "../models/parameters";
/// <reference lib="esnext.asynciterable" />
/** Class containing Actions operations. */
export class ActionsImpl {
    /**
     * Initialize a new instance of the class Actions class.
     * @param client Reference to the service client
     */
    constructor(client) {
        this.client = client;
    }
    /**
     * Gets all actions of alert rule.
     * @param resourceGroupName The name of the resource group. The name is case insensitive.
     * @param workspaceName The name of the workspace.
     * @param ruleId Alert rule ID
     * @param options The options parameters.
     */
    listByAlertRule(resourceGroupName, workspaceName, ruleId, options) {
        const iter = this.listByAlertRulePagingAll(resourceGroupName, workspaceName, ruleId, options);
        return {
            next() {
                return iter.next();
            },
            [Symbol.asyncIterator]() {
                return this;
            },
            byPage: (settings) => {
                if (settings === null || settings === void 0 ? void 0 : settings.maxPageSize) {
                    throw new Error("maxPageSize is not supported by this operation.");
                }
                return this.listByAlertRulePagingPage(resourceGroupName, workspaceName, ruleId, options, settings);
            }
        };
    }
    listByAlertRulePagingPage(resourceGroupName, workspaceName, ruleId, options, settings) {
        return __asyncGenerator(this, arguments, function* listByAlertRulePagingPage_1() {
            let result;
            let continuationToken = settings === null || settings === void 0 ? void 0 : settings.continuationToken;
            if (!continuationToken) {
                result = yield __await(this._listByAlertRule(resourceGroupName, workspaceName, ruleId, options));
                let page = result.value || [];
                continuationToken = result.nextLink;
                setContinuationToken(page, continuationToken);
                yield yield __await(page);
            }
            while (continuationToken) {
                result = yield __await(this._listByAlertRuleNext(resourceGroupName, workspaceName, ruleId, continuationToken, options));
                continuationToken = result.nextLink;
                let page = result.value || [];
                setContinuationToken(page, continuationToken);
                yield yield __await(page);
            }
        });
    }
    listByAlertRulePagingAll(resourceGroupName, workspaceName, ruleId, options) {
        return __asyncGenerator(this, arguments, function* listByAlertRulePagingAll_1() {
            var e_1, _a;
            try {
                for (var _b = __asyncValues(this.listByAlertRulePagingPage(resourceGroupName, workspaceName, ruleId, options)), _c; _c = yield __await(_b.next()), !_c.done;) {
                    const page = _c.value;
                    yield __await(yield* __asyncDelegator(__asyncValues(page)));
                }
            }
            catch (e_1_1) { e_1 = { error: e_1_1 }; }
            finally {
                try {
                    if (_c && !_c.done && (_a = _b.return)) yield __await(_a.call(_b));
                }
                finally { if (e_1) throw e_1.error; }
            }
        });
    }
    /**
     * Gets all actions of alert rule.
     * @param resourceGroupName The name of the resource group. The name is case insensitive.
     * @param workspaceName The name of the workspace.
     * @param ruleId Alert rule ID
     * @param options The options parameters.
     */
    _listByAlertRule(resourceGroupName, workspaceName, ruleId, options) {
        return this.client.sendOperationRequest({ resourceGroupName, workspaceName, ruleId, options }, listByAlertRuleOperationSpec);
    }
    /**
     * Gets the action of alert rule.
     * @param resourceGroupName The name of the resource group. The name is case insensitive.
     * @param workspaceName The name of the workspace.
     * @param ruleId Alert rule ID
     * @param actionId Action ID
     * @param options The options parameters.
     */
    get(resourceGroupName, workspaceName, ruleId, actionId, options) {
        return this.client.sendOperationRequest({ resourceGroupName, workspaceName, ruleId, actionId, options }, getOperationSpec);
    }
    /**
     * Creates or updates the action of alert rule.
     * @param resourceGroupName The name of the resource group. The name is case insensitive.
     * @param workspaceName The name of the workspace.
     * @param ruleId Alert rule ID
     * @param actionId Action ID
     * @param action The action
     * @param options The options parameters.
     */
    createOrUpdate(resourceGroupName, workspaceName, ruleId, actionId, action, options) {
        return this.client.sendOperationRequest({ resourceGroupName, workspaceName, ruleId, actionId, action, options }, createOrUpdateOperationSpec);
    }
    /**
     * Delete the action of alert rule.
     * @param resourceGroupName The name of the resource group. The name is case insensitive.
     * @param workspaceName The name of the workspace.
     * @param ruleId Alert rule ID
     * @param actionId Action ID
     * @param options The options parameters.
     */
    delete(resourceGroupName, workspaceName, ruleId, actionId, options) {
        return this.client.sendOperationRequest({ resourceGroupName, workspaceName, ruleId, actionId, options }, deleteOperationSpec);
    }
    /**
     * ListByAlertRuleNext
     * @param resourceGroupName The name of the resource group. The name is case insensitive.
     * @param workspaceName The name of the workspace.
     * @param ruleId Alert rule ID
     * @param nextLink The nextLink from the previous successful call to the ListByAlertRule method.
     * @param options The options parameters.
     */
    _listByAlertRuleNext(resourceGroupName, workspaceName, ruleId, nextLink, options) {
        return this.client.sendOperationRequest({ resourceGroupName, workspaceName, ruleId, nextLink, options }, listByAlertRuleNextOperationSpec);
    }
}
// Operation Specifications
const serializer = coreClient.createSerializer(Mappers, /* isXml */ false);
const listByAlertRuleOperationSpec = {
    path: "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.OperationalInsights/workspaces/{workspaceName}/providers/Microsoft.SecurityInsights/alertRules/{ruleId}/actions",
    httpMethod: "GET",
    responses: {
        200: {
            bodyMapper: Mappers.ActionsList
        },
        default: {
            bodyMapper: Mappers.CloudError
        }
    },
    queryParameters: [Parameters.apiVersion],
    urlParameters: [
        Parameters.$host,
        Parameters.subscriptionId,
        Parameters.resourceGroupName,
        Parameters.workspaceName,
        Parameters.ruleId
    ],
    headerParameters: [Parameters.accept],
    serializer
};
const getOperationSpec = {
    path: "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.OperationalInsights/workspaces/{workspaceName}/providers/Microsoft.SecurityInsights/alertRules/{ruleId}/actions/{actionId}",
    httpMethod: "GET",
    responses: {
        200: {
            bodyMapper: Mappers.ActionResponse
        },
        default: {
            bodyMapper: Mappers.CloudError
        }
    },
    queryParameters: [Parameters.apiVersion],
    urlParameters: [
        Parameters.$host,
        Parameters.subscriptionId,
        Parameters.resourceGroupName,
        Parameters.workspaceName,
        Parameters.ruleId,
        Parameters.actionId
    ],
    headerParameters: [Parameters.accept],
    serializer
};
const createOrUpdateOperationSpec = {
    path: "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.OperationalInsights/workspaces/{workspaceName}/providers/Microsoft.SecurityInsights/alertRules/{ruleId}/actions/{actionId}",
    httpMethod: "PUT",
    responses: {
        200: {
            bodyMapper: Mappers.ActionResponse
        },
        201: {
            bodyMapper: Mappers.ActionResponse
        },
        default: {
            bodyMapper: Mappers.CloudError
        }
    },
    requestBody: Parameters.action,
    queryParameters: [Parameters.apiVersion],
    urlParameters: [
        Parameters.$host,
        Parameters.subscriptionId,
        Parameters.resourceGroupName,
        Parameters.workspaceName,
        Parameters.ruleId,
        Parameters.actionId
    ],
    headerParameters: [Parameters.accept, Parameters.contentType],
    mediaType: "json",
    serializer
};
const deleteOperationSpec = {
    path: "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.OperationalInsights/workspaces/{workspaceName}/providers/Microsoft.SecurityInsights/alertRules/{ruleId}/actions/{actionId}",
    httpMethod: "DELETE",
    responses: {
        200: {},
        204: {},
        default: {
            bodyMapper: Mappers.CloudError
        }
    },
    queryParameters: [Parameters.apiVersion],
    urlParameters: [
        Parameters.$host,
        Parameters.subscriptionId,
        Parameters.resourceGroupName,
        Parameters.workspaceName,
        Parameters.ruleId,
        Parameters.actionId
    ],
    headerParameters: [Parameters.accept],
    serializer
};
const listByAlertRuleNextOperationSpec = {
    path: "{nextLink}",
    httpMethod: "GET",
    responses: {
        200: {
            bodyMapper: Mappers.ActionsList
        },
        default: {
            bodyMapper: Mappers.CloudError
        }
    },
    urlParameters: [
        Parameters.$host,
        Parameters.subscriptionId,
        Parameters.resourceGroupName,
        Parameters.workspaceName,
        Parameters.ruleId,
        Parameters.nextLink
    ],
    headerParameters: [Parameters.accept],
    serializer
};
