// @azure/arm-securityinsight/dist-esm/src/operations/domainWhois.js
/*
 * Copyright (c) Microsoft Corporation.
 * Licensed under the MIT License.
 *
 * Code generated by Microsoft (R) AutoRest Code Generator.
 * Changes may cause incorrect behavior and will be lost if the code is regenerated.
 */
import * as coreClient from "@azure/core-client";
import * as Mappers from "../models/mappers";
import * as Parameters from "../models/parameters";
/** Class containing DomainWhois operations. */
export class DomainWhoisImpl {
    /**
     * Initialize a new instance of the class DomainWhois class.
     * @param client Reference to the service client
     */
    constructor(client) {
        this.client = client;
    }
    /**
     * Get whois information for a single domain name
     * @param resourceGroupName The name of the resource group. The name is case insensitive.
     * @param domain Domain name to be enriched
     * @param options The options parameters.
     */
    get(resourceGroupName, domain, options) {
        return this.client.sendOperationRequest({ resourceGroupName, domain, options }, getOperationSpec);
    }
}
// Operation Specifications
const serializer = coreClient.createSerializer(Mappers, /* isXml */ false);
const getOperationSpec = {
    path: "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.SecurityInsights/enrichment/domain/whois/",
    httpMethod: "GET",
    responses: {
        200: {
            bodyMapper: Mappers.EnrichmentDomainWhois
        },
        default: {
            bodyMapper: Mappers.CloudError
        }
    },
    queryParameters: [Parameters.apiVersion, Parameters.domain],
    urlParameters: [
        Parameters.$host,
        Parameters.subscriptionId,
        Parameters.resourceGroupName
    ],
    headerParameters: [Parameters.accept],
    serializer
};
