// @azure/arm-securityinsight/dist-esm/src/operations/sentinelOnboardingStates.js
/*
 * Copyright (c) Microsoft Corporation.
 * Licensed under the MIT License.
 *
 * Code generated by Microsoft (R) AutoRest Code Generator.
 * Changes may cause incorrect behavior and will be lost if the code is regenerated.
 */
import * as coreClient from "@azure/core-client";
import * as Mappers from "../models/mappers";
import * as Parameters from "../models/parameters";
/** Class containing SentinelOnboardingStates operations. */
export class SentinelOnboardingStatesImpl {
    /**
     * Initialize a new instance of the class SentinelOnboardingStates class.
     * @param client Reference to the service client
     */
    constructor(client) {
        this.client = client;
    }
    /**
     * Get Sentinel onboarding state
     * @param resourceGroupName The name of the resource group. The name is case insensitive.
     * @param workspaceName The name of the workspace.
     * @param sentinelOnboardingStateName The Sentinel onboarding state name. Supports - default
     * @param options The options parameters.
     */
    get(resourceGroupName, workspaceName, sentinelOnboardingStateName, options) {
        return this.client.sendOperationRequest({
            resourceGroupName,
            workspaceName,
            sentinelOnboardingStateName,
            options
        }, getOperationSpec);
    }
    /**
     * Create Sentinel onboarding state
     * @param resourceGroupName The name of the resource group. The name is case insensitive.
     * @param workspaceName The name of the workspace.
     * @param sentinelOnboardingStateName The Sentinel onboarding state name. Supports - default
     * @param options The options parameters.
     */
    create(resourceGroupName, workspaceName, sentinelOnboardingStateName, options) {
        return this.client.sendOperationRequest({
            resourceGroupName,
            workspaceName,
            sentinelOnboardingStateName,
            options
        }, createOperationSpec);
    }
    /**
     * Delete Sentinel onboarding state
     * @param resourceGroupName The name of the resource group. The name is case insensitive.
     * @param workspaceName The name of the workspace.
     * @param sentinelOnboardingStateName The Sentinel onboarding state name. Supports - default
     * @param options The options parameters.
     */
    delete(resourceGroupName, workspaceName, sentinelOnboardingStateName, options) {
        return this.client.sendOperationRequest({
            resourceGroupName,
            workspaceName,
            sentinelOnboardingStateName,
            options
        }, deleteOperationSpec);
    }
    /**
     * Gets all Sentinel onboarding states
     * @param resourceGroupName The name of the resource group. The name is case insensitive.
     * @param workspaceName The name of the workspace.
     * @param options The options parameters.
     */
    list(resourceGroupName, workspaceName, options) {
        return this.client.sendOperationRequest({ resourceGroupName, workspaceName, options }, listOperationSpec);
    }
}
// Operation Specifications
const serializer = coreClient.createSerializer(Mappers, /* isXml */ false);
const getOperationSpec = {
    path: "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.OperationalInsights/workspaces/{workspaceName}/providers/Microsoft.SecurityInsights/onboardingStates/{sentinelOnboardingStateName}",
    httpMethod: "GET",
    responses: {
        200: {
            bodyMapper: Mappers.SentinelOnboardingState
        },
        default: {
            bodyMapper: Mappers.CloudError
        }
    },
    queryParameters: [Parameters.apiVersion],
    urlParameters: [
        Parameters.$host,
        Parameters.subscriptionId,
        Parameters.resourceGroupName,
        Parameters.workspaceName,
        Parameters.sentinelOnboardingStateName
    ],
    headerParameters: [Parameters.accept],
    serializer
};
const createOperationSpec = {
    path: "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.OperationalInsights/workspaces/{workspaceName}/providers/Microsoft.SecurityInsights/onboardingStates/{sentinelOnboardingStateName}",
    httpMethod: "PUT",
    responses: {
        200: {
            bodyMapper: Mappers.SentinelOnboardingState
        },
        201: {
            bodyMapper: Mappers.SentinelOnboardingState
        },
        default: {
            bodyMapper: Mappers.CloudError
        }
    },
    requestBody: Parameters.sentinelOnboardingStateParameter,
    queryParameters: [Parameters.apiVersion],
    urlParameters: [
        Parameters.$host,
        Parameters.subscriptionId,
        Parameters.resourceGroupName,
        Parameters.workspaceName,
        Parameters.sentinelOnboardingStateName
    ],
    headerParameters: [Parameters.accept, Parameters.contentType],
    mediaType: "json",
    serializer
};
const deleteOperationSpec = {
    path: "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.OperationalInsights/workspaces/{workspaceName}/providers/Microsoft.SecurityInsights/onboardingStates/{sentinelOnboardingStateName}",
    httpMethod: "DELETE",
    responses: {
        200: {},
        204: {},
        default: {
            bodyMapper: Mappers.CloudError
        }
    },
    queryParameters: [Parameters.apiVersion],
    urlParameters: [
        Parameters.$host,
        Parameters.subscriptionId,
        Parameters.resourceGroupName,
        Parameters.workspaceName,
        Parameters.sentinelOnboardingStateName
    ],
    headerParameters: [Parameters.accept],
    serializer
};
const listOperationSpec = {
    path: "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.OperationalInsights/workspaces/{workspaceName}/providers/Microsoft.SecurityInsights/onboardingStates",
    httpMethod: "GET",
    responses: {
        200: {
            bodyMapper: Mappers.SentinelOnboardingStatesList
        },
        default: {
            bodyMapper: Mappers.CloudError
        }
    },
    queryParameters: [Parameters.apiVersion],
    urlParameters: [
        Parameters.$host,
        Parameters.subscriptionId,
        Parameters.resourceGroupName,
        Parameters.workspaceName
    ],
    headerParameters: [Parameters.accept],
    serializer
};
