// @azure/arm-securityinsight/dist-esm/src/operations/sourceControlOperations.js
/*
 * Copyright (c) Microsoft Corporation.
 * Licensed under the MIT License.
 *
 * Code generated by Microsoft (R) AutoRest Code Generator.
 * Changes may cause incorrect behavior and will be lost if the code is regenerated.
 */
import { __asyncDelegator, __asyncGenerator, __asyncValues, __await } from "tslib";
import { setContinuationToken } from "../pagingHelper";
import * as coreClient from "@azure/core-client";
import * as Mappers from "../models/mappers";
import * as Parameters from "../models/parameters";
/// <reference lib="esnext.asynciterable" />
/** Class containing SourceControlOperations operations. */
export class SourceControlOperationsImpl {
    /**
     * Initialize a new instance of the class SourceControlOperations class.
     * @param client Reference to the service client
     */
    constructor(client) {
        this.client = client;
    }
    /**
     * Gets a list of repositories metadata.
     * @param resourceGroupName The name of the resource group. The name is case insensitive.
     * @param workspaceName The name of the workspace.
     * @param repoType The repo type.
     * @param options The options parameters.
     */
    listRepositories(resourceGroupName, workspaceName, repoType, options) {
        const iter = this.listRepositoriesPagingAll(resourceGroupName, workspaceName, repoType, options);
        return {
            next() {
                return iter.next();
            },
            [Symbol.asyncIterator]() {
                return this;
            },
            byPage: (settings) => {
                if (settings === null || settings === void 0 ? void 0 : settings.maxPageSize) {
                    throw new Error("maxPageSize is not supported by this operation.");
                }
                return this.listRepositoriesPagingPage(resourceGroupName, workspaceName, repoType, options, settings);
            }
        };
    }
    listRepositoriesPagingPage(resourceGroupName, workspaceName, repoType, options, settings) {
        return __asyncGenerator(this, arguments, function* listRepositoriesPagingPage_1() {
            let result;
            let continuationToken = settings === null || settings === void 0 ? void 0 : settings.continuationToken;
            if (!continuationToken) {
                result = yield __await(this._listRepositories(resourceGroupName, workspaceName, repoType, options));
                let page = result.value || [];
                continuationToken = result.nextLink;
                setContinuationToken(page, continuationToken);
                yield yield __await(page);
            }
            while (continuationToken) {
                result = yield __await(this._listRepositoriesNext(resourceGroupName, workspaceName, repoType, continuationToken, options));
                continuationToken = result.nextLink;
                let page = result.value || [];
                setContinuationToken(page, continuationToken);
                yield yield __await(page);
            }
        });
    }
    listRepositoriesPagingAll(resourceGroupName, workspaceName, repoType, options) {
        return __asyncGenerator(this, arguments, function* listRepositoriesPagingAll_1() {
            var e_1, _a;
            try {
                for (var _b = __asyncValues(this.listRepositoriesPagingPage(resourceGroupName, workspaceName, repoType, options)), _c; _c = yield __await(_b.next()), !_c.done;) {
                    const page = _c.value;
                    yield __await(yield* __asyncDelegator(__asyncValues(page)));
                }
            }
            catch (e_1_1) { e_1 = { error: e_1_1 }; }
            finally {
                try {
                    if (_c && !_c.done && (_a = _b.return)) yield __await(_a.call(_b));
                }
                finally { if (e_1) throw e_1.error; }
            }
        });
    }
    /**
     * Gets a list of repositories metadata.
     * @param resourceGroupName The name of the resource group. The name is case insensitive.
     * @param workspaceName The name of the workspace.
     * @param repoType The repo type.
     * @param options The options parameters.
     */
    _listRepositories(resourceGroupName, workspaceName, repoType, options) {
        return this.client.sendOperationRequest({ resourceGroupName, workspaceName, repoType, options }, listRepositoriesOperationSpec);
    }
    /**
     * ListRepositoriesNext
     * @param resourceGroupName The name of the resource group. The name is case insensitive.
     * @param workspaceName The name of the workspace.
     * @param repoType The repo type.
     * @param nextLink The nextLink from the previous successful call to the ListRepositories method.
     * @param options The options parameters.
     */
    _listRepositoriesNext(resourceGroupName, workspaceName, repoType, nextLink, options) {
        return this.client.sendOperationRequest({ resourceGroupName, workspaceName, repoType, nextLink, options }, listRepositoriesNextOperationSpec);
    }
}
// Operation Specifications
const serializer = coreClient.createSerializer(Mappers, /* isXml */ false);
const listRepositoriesOperationSpec = {
    path: "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.OperationalInsights/workspaces/{workspaceName}/providers/Microsoft.SecurityInsights/listRepositories",
    httpMethod: "POST",
    responses: {
        200: {
            bodyMapper: Mappers.RepoList
        },
        default: {
            bodyMapper: Mappers.CloudError
        }
    },
    requestBody: Parameters.repoType,
    queryParameters: [Parameters.apiVersion],
    urlParameters: [
        Parameters.$host,
        Parameters.subscriptionId,
        Parameters.resourceGroupName,
        Parameters.workspaceName
    ],
    headerParameters: [Parameters.accept, Parameters.contentType],
    mediaType: "json",
    serializer
};
const listRepositoriesNextOperationSpec = {
    path: "{nextLink}",
    httpMethod: "GET",
    responses: {
        200: {
            bodyMapper: Mappers.RepoList
        },
        default: {
            bodyMapper: Mappers.CloudError
        }
    },
    urlParameters: [
        Parameters.$host,
        Parameters.subscriptionId,
        Parameters.resourceGroupName,
        Parameters.workspaceName,
        Parameters.nextLink
    ],
    headerParameters: [Parameters.accept, Parameters.contentType],
    mediaType: "json",
    serializer
};
