// @google-cloud/retail/build/src/index.js
"use strict";
// Copyright 2023 Google LLC
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     https://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.
//
// ** This file is automatically generated by synthtool. **
// ** https://github.com/googleapis/synthtool **
// ** All changes to this file may be overwritten. **
Object.defineProperty(exports, "__esModule", { value: true });
exports.protos = exports.UserEventServiceClient = exports.ServingConfigServiceClient = exports.SearchServiceClient = exports.ProductServiceClient = exports.PredictionServiceClient = exports.ModelServiceClient = exports.ControlServiceClient = exports.CompletionServiceClient = exports.CatalogServiceClient = exports.v2beta = exports.v2alpha = exports.v2 = void 0;
const v2 = require("./v2");
exports.v2 = v2;
const v2alpha = require("./v2alpha");
exports.v2alpha = v2alpha;
const v2beta = require("./v2beta");
exports.v2beta = v2beta;
const CatalogServiceClient = v2beta.CatalogServiceClient;
exports.CatalogServiceClient = CatalogServiceClient;
const CompletionServiceClient = v2beta.CompletionServiceClient;
exports.CompletionServiceClient = CompletionServiceClient;
const ControlServiceClient = v2beta.ControlServiceClient;
exports.ControlServiceClient = ControlServiceClient;
const ModelServiceClient = v2beta.ModelServiceClient;
exports.ModelServiceClient = ModelServiceClient;
const PredictionServiceClient = v2beta.PredictionServiceClient;
exports.PredictionServiceClient = PredictionServiceClient;
const ProductServiceClient = v2beta.ProductServiceClient;
exports.ProductServiceClient = ProductServiceClient;
const SearchServiceClient = v2beta.SearchServiceClient;
exports.SearchServiceClient = SearchServiceClient;
const ServingConfigServiceClient = v2beta.ServingConfigServiceClient;
exports.ServingConfigServiceClient = ServingConfigServiceClient;
const UserEventServiceClient = v2beta.UserEventServiceClient;
exports.UserEventServiceClient = UserEventServiceClient;
exports.default = {
    v2,
    v2alpha,
    v2beta,
    CatalogServiceClient,
    CompletionServiceClient,
    ControlServiceClient,
    ModelServiceClient,
    PredictionServiceClient,
    ProductServiceClient,
    SearchServiceClient,
    ServingConfigServiceClient,
    UserEventServiceClient,
};
const protos = require("../protos/protos");
exports.protos = protos;
