// @azure/arm-dns/dist-esm/src/models/parameters.js
/*
 * Copyright (c) Microsoft Corporation.
 * Licensed under the MIT License.
 *
 * Code generated by Microsoft (R) AutoRest Code Generator.
 * Changes may cause incorrect behavior and will be lost if the code is regenerated.
 */
import { RecordSet as RecordSetMapper, Zone as ZoneMapper, ZoneUpdate as ZoneUpdateMapper, DnsResourceReferenceRequest as DnsResourceReferenceRequestMapper } from "../models/mappers";
export const contentType = {
    parameterPath: ["options", "contentType"],
    mapper: {
        defaultValue: "application/json",
        isConstant: true,
        serializedName: "Content-Type",
        type: {
            name: "String"
        }
    }
};
export const parameters = {
    parameterPath: "parameters",
    mapper: RecordSetMapper
};
export const accept = {
    parameterPath: "accept",
    mapper: {
        defaultValue: "application/json",
        isConstant: true,
        serializedName: "Accept",
        type: {
            name: "String"
        }
    }
};
export const $host = {
    parameterPath: "$host",
    mapper: {
        serializedName: "$host",
        required: true,
        type: {
            name: "String"
        }
    },
    skipEncoding: true
};
export const resourceGroupName = {
    parameterPath: "resourceGroupName",
    mapper: {
        serializedName: "resourceGroupName",
        required: true,
        type: {
            name: "String"
        }
    }
};
export const zoneName = {
    parameterPath: "zoneName",
    mapper: {
        serializedName: "zoneName",
        required: true,
        type: {
            name: "String"
        }
    }
};
export const relativeRecordSetName = {
    parameterPath: "relativeRecordSetName",
    mapper: {
        serializedName: "relativeRecordSetName",
        required: true,
        type: {
            name: "String"
        }
    },
    skipEncoding: true
};
export const recordType = {
    parameterPath: "recordType",
    mapper: {
        serializedName: "recordType",
        required: true,
        type: {
            name: "Enum",
            allowedValues: [
                "A",
                "AAAA",
                "CAA",
                "CNAME",
                "MX",
                "NS",
                "PTR",
                "SOA",
                "SRV",
                "TXT"
            ]
        }
    }
};
export const ifMatch = {
    parameterPath: ["options", "ifMatch"],
    mapper: {
        serializedName: "If-Match",
        type: {
            name: "String"
        }
    }
};
export const apiVersion = {
    parameterPath: "apiVersion",
    mapper: {
        defaultValue: "2018-05-01",
        isConstant: true,
        serializedName: "api-version",
        type: {
            name: "String"
        }
    }
};
export const subscriptionId = {
    parameterPath: "subscriptionId",
    mapper: {
        serializedName: "subscriptionId",
        required: true,
        type: {
            name: "String"
        }
    }
};
export const ifNoneMatch = {
    parameterPath: ["options", "ifNoneMatch"],
    mapper: {
        serializedName: "If-None-Match",
        type: {
            name: "String"
        }
    }
};
export const top = {
    parameterPath: ["options", "top"],
    mapper: {
        serializedName: "$top",
        type: {
            name: "Number"
        }
    }
};
export const recordsetnamesuffix = {
    parameterPath: ["options", "recordsetnamesuffix"],
    mapper: {
        serializedName: "$recordsetnamesuffix",
        type: {
            name: "String"
        }
    }
};
export const recordSetNameSuffix = {
    parameterPath: ["options", "recordSetNameSuffix"],
    mapper: {
        serializedName: "$recordsetnamesuffix",
        type: {
            name: "String"
        }
    }
};
export const nextLink = {
    parameterPath: "nextLink",
    mapper: {
        serializedName: "nextLink",
        required: true,
        type: {
            name: "String"
        }
    },
    skipEncoding: true
};
export const parameters1 = {
    parameterPath: "parameters",
    mapper: ZoneMapper
};
export const parameters2 = {
    parameterPath: "parameters",
    mapper: ZoneUpdateMapper
};
export const parameters3 = {
    parameterPath: "parameters",
    mapper: DnsResourceReferenceRequestMapper
};
