// @azure/arm-datadog/dist-esm/src/operations/creationSupported.js
/*
 * Copyright (c) Microsoft Corporation.
 * Licensed under the MIT License.
 *
 * Code generated by Microsoft (R) AutoRest Code Generator.
 * Changes may cause incorrect behavior and will be lost if the code is regenerated.
 */
import { __asyncDelegator, __asyncGenerator, __asyncValues, __await } from "tslib";
import * as coreClient from "@azure/core-client";
import * as Mappers from "../models/mappers";
import * as Parameters from "../models/parameters";
/// <reference lib="esnext.asynciterable" />
/** Class containing CreationSupported operations. */
export class CreationSupportedImpl {
    /**
     * Initialize a new instance of the class CreationSupported class.
     * @param client Reference to the service client
     */
    constructor(client) {
        this.client = client;
    }
    /**
     * Informs if the current subscription is being already monitored for selected Datadog organization.
     * @param datadogOrganizationId Datadog Organization Id
     * @param options The options parameters.
     */
    list(datadogOrganizationId, options) {
        const iter = this.listPagingAll(datadogOrganizationId, options);
        return {
            next() {
                return iter.next();
            },
            [Symbol.asyncIterator]() {
                return this;
            },
            byPage: (settings) => {
                if (settings === null || settings === void 0 ? void 0 : settings.maxPageSize) {
                    throw new Error("maxPageSize is not supported by this operation.");
                }
                return this.listPagingPage(datadogOrganizationId, options, settings);
            }
        };
    }
    listPagingPage(datadogOrganizationId, options, _settings) {
        return __asyncGenerator(this, arguments, function* listPagingPage_1() {
            let result;
            result = yield __await(this._list(datadogOrganizationId, options));
            yield yield __await(result.value || []);
        });
    }
    listPagingAll(datadogOrganizationId, options) {
        return __asyncGenerator(this, arguments, function* listPagingAll_1() {
            var _a, e_1, _b, _c;
            try {
                for (var _d = true, _e = __asyncValues(this.listPagingPage(datadogOrganizationId, options)), _f; _f = yield __await(_e.next()), _a = _f.done, !_a; _d = true) {
                    _c = _f.value;
                    _d = false;
                    const page = _c;
                    yield __await(yield* __asyncDelegator(__asyncValues(page)));
                }
            }
            catch (e_1_1) { e_1 = { error: e_1_1 }; }
            finally {
                try {
                    if (!_d && !_a && (_b = _e.return)) yield __await(_b.call(_e));
                }
                finally { if (e_1) throw e_1.error; }
            }
        });
    }
    /**
     * Informs if the current subscription is being already monitored for selected Datadog organization.
     * @param datadogOrganizationId Datadog Organization Id
     * @param options The options parameters.
     */
    _list(datadogOrganizationId, options) {
        return this.client.sendOperationRequest({ datadogOrganizationId, options }, listOperationSpec);
    }
    /**
     * Informs if the current subscription is being already monitored for selected Datadog organization.
     * @param datadogOrganizationId Datadog Organization Id
     * @param options The options parameters.
     */
    get(datadogOrganizationId, options) {
        return this.client.sendOperationRequest({ datadogOrganizationId, options }, getOperationSpec);
    }
}
// Operation Specifications
const serializer = coreClient.createSerializer(Mappers, /* isXml */ false);
const listOperationSpec = {
    path: "/subscriptions/{subscriptionId}/providers/Microsoft.Datadog/subscriptionStatuses",
    httpMethod: "GET",
    responses: {
        200: {
            bodyMapper: Mappers.CreateResourceSupportedResponseList
        },
        default: {
            bodyMapper: Mappers.ErrorResponse
        }
    },
    queryParameters: [Parameters.apiVersion, Parameters.datadogOrganizationId],
    urlParameters: [Parameters.$host, Parameters.subscriptionId],
    headerParameters: [Parameters.accept],
    serializer
};
const getOperationSpec = {
    path: "/subscriptions/{subscriptionId}/providers/Microsoft.Datadog/subscriptionStatuses/default",
    httpMethod: "GET",
    responses: {
        200: {
            bodyMapper: Mappers.CreateResourceSupportedResponse
        },
        default: {
            bodyMapper: Mappers.ErrorResponse
        }
    },
    queryParameters: [Parameters.apiVersion, Parameters.datadogOrganizationId],
    urlParameters: [Parameters.$host, Parameters.subscriptionId],
    headerParameters: [Parameters.accept],
    serializer
};
