// @azure/arm-appinsights/esm/operations/componentLinkedStorageAccountsOperations.js
/*
 * Copyright (c) Microsoft Corporation.
 * Licensed under the MIT License.
 *
 * Code generated by Microsoft (R) AutoRest Code Generator.
 * Changes may cause incorrect behavior and will be lost if the code is
 * regenerated.
 */
import { __assign } from "tslib";
import * as msRest from "@azure/ms-rest-js";
import * as Mappers from "../models/componentLinkedStorageAccountsOperationsMappers";
import * as Parameters from "../models/parameters";
/** Class representing a ComponentLinkedStorageAccountsOperations. */
var ComponentLinkedStorageAccountsOperations = /** @class */ (function () {
    /**
     * Create a ComponentLinkedStorageAccountsOperations.
     * @param {ApplicationInsightsManagementClientContext} client Reference to the service client.
     */
    function ComponentLinkedStorageAccountsOperations(client) {
        this.client = client;
    }
    ComponentLinkedStorageAccountsOperations.prototype.get = function (resourceGroupName, resourceName, options, callback) {
        return this.client.sendOperationRequest({
            resourceGroupName: resourceGroupName,
            resourceName: resourceName,
            options: options
        }, getOperationSpec, callback);
    };
    ComponentLinkedStorageAccountsOperations.prototype.createAndUpdate = function (resourceGroupName, resourceName, linkedStorageAccountsProperties, options, callback) {
        return this.client.sendOperationRequest({
            resourceGroupName: resourceGroupName,
            resourceName: resourceName,
            linkedStorageAccountsProperties: linkedStorageAccountsProperties,
            options: options
        }, createAndUpdateOperationSpec, callback);
    };
    ComponentLinkedStorageAccountsOperations.prototype.update = function (resourceGroupName, resourceName, linkedStorageAccountsProperties, options, callback) {
        return this.client.sendOperationRequest({
            resourceGroupName: resourceGroupName,
            resourceName: resourceName,
            linkedStorageAccountsProperties: linkedStorageAccountsProperties,
            options: options
        }, updateOperationSpec, callback);
    };
    ComponentLinkedStorageAccountsOperations.prototype.deleteMethod = function (resourceGroupName, resourceName, options, callback) {
        return this.client.sendOperationRequest({
            resourceGroupName: resourceGroupName,
            resourceName: resourceName,
            options: options
        }, deleteMethodOperationSpec, callback);
    };
    return ComponentLinkedStorageAccountsOperations;
}());
export { ComponentLinkedStorageAccountsOperations };
// Operation Specifications
var serializer = new msRest.Serializer(Mappers);
var getOperationSpec = {
    httpMethod: "GET",
    path: "subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/microsoft.insights/components/{resourceName}/linkedStorageAccounts/{storageType}",
    urlParameters: [
        Parameters.resourceGroupName,
        Parameters.subscriptionId,
        Parameters.resourceName,
        Parameters.storageType
    ],
    queryParameters: [
        Parameters.apiVersion6
    ],
    headerParameters: [
        Parameters.acceptLanguage
    ],
    responses: {
        200: {
            bodyMapper: Mappers.ComponentLinkedStorageAccounts
        },
        default: {
            bodyMapper: Mappers.ErrorResponseLinkedStorage
        }
    },
    serializer: serializer
};
var createAndUpdateOperationSpec = {
    httpMethod: "PUT",
    path: "subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/microsoft.insights/components/{resourceName}/linkedStorageAccounts/{storageType}",
    urlParameters: [
        Parameters.resourceGroupName,
        Parameters.subscriptionId,
        Parameters.resourceName,
        Parameters.storageType
    ],
    queryParameters: [
        Parameters.apiVersion6
    ],
    headerParameters: [
        Parameters.acceptLanguage
    ],
    requestBody: {
        parameterPath: "linkedStorageAccountsProperties",
        mapper: __assign(__assign({}, Mappers.ComponentLinkedStorageAccounts), { required: true })
    },
    responses: {
        200: {
            bodyMapper: Mappers.ComponentLinkedStorageAccounts
        },
        default: {
            bodyMapper: Mappers.ErrorResponseLinkedStorage
        }
    },
    serializer: serializer
};
var updateOperationSpec = {
    httpMethod: "PATCH",
    path: "subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/microsoft.insights/components/{resourceName}/linkedStorageAccounts/{storageType}",
    urlParameters: [
        Parameters.resourceGroupName,
        Parameters.subscriptionId,
        Parameters.resourceName,
        Parameters.storageType
    ],
    queryParameters: [
        Parameters.apiVersion6
    ],
    headerParameters: [
        Parameters.acceptLanguage
    ],
    requestBody: {
        parameterPath: "linkedStorageAccountsProperties",
        mapper: __assign(__assign({}, Mappers.ComponentLinkedStorageAccountsPatch), { required: true })
    },
    responses: {
        200: {
            bodyMapper: Mappers.ComponentLinkedStorageAccounts
        },
        default: {
            bodyMapper: Mappers.ErrorResponseLinkedStorage
        }
    },
    serializer: serializer
};
var deleteMethodOperationSpec = {
    httpMethod: "DELETE",
    path: "subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/microsoft.insights/components/{resourceName}/linkedStorageAccounts/{storageType}",
    urlParameters: [
        Parameters.resourceGroupName,
        Parameters.subscriptionId,
        Parameters.resourceName,
        Parameters.storageType
    ],
    queryParameters: [
        Parameters.apiVersion6
    ],
    headerParameters: [
        Parameters.acceptLanguage
    ],
    responses: {
        200: {},
        204: {},
        default: {
            bodyMapper: Mappers.ErrorResponseLinkedStorage
        }
    },
    serializer: serializer
};
