// @azure/arm-appinsights/esm/operations/liveToken.js
/*
 * Copyright (c) Microsoft Corporation.
 * Licensed under the MIT License.
 *
 * Code generated by Microsoft (R) AutoRest Code Generator.
 * Changes may cause incorrect behavior and will be lost if the code is
 * regenerated.
 */
import * as msRest from "@azure/ms-rest-js";
import * as Mappers from "../models/liveTokenMappers";
import * as Parameters from "../models/parameters";
/** Class representing a LiveToken. */
var LiveToken = /** @class */ (function () {
    /**
     * Create a LiveToken.
     * @param {ApplicationInsightsManagementClientContext} client Reference to the service client.
     */
    function LiveToken(client) {
        this.client = client;
    }
    LiveToken.prototype.get = function (resourceUri, options, callback) {
        return this.client.sendOperationRequest({
            resourceUri: resourceUri,
            options: options
        }, getOperationSpec, callback);
    };
    return LiveToken;
}());
export { LiveToken };
// Operation Specifications
var serializer = new msRest.Serializer(Mappers);
var getOperationSpec = {
    httpMethod: "POST",
    path: "{resourceUri}/providers/microsoft.insights/generatelivetoken",
    urlParameters: [
        Parameters.resourceUri
    ],
    queryParameters: [
        Parameters.apiVersion1
    ],
    headerParameters: [
        Parameters.acceptLanguage
    ],
    responses: {
        200: {
            bodyMapper: Mappers.LiveTokenResponse
        },
        default: {
            bodyMapper: Mappers.ErrorResponseLinkedStorage
        }
    },
    serializer: serializer
};
