// @azure/arm-appinsights/esm/operations/myWorkbooks.js
/*
 * Copyright (c) Microsoft Corporation.
 * Licensed under the MIT License.
 *
 * Code generated by Microsoft (R) AutoRest Code Generator.
 * Changes may cause incorrect behavior and will be lost if the code is
 * regenerated.
 */
import { __assign } from "tslib";
import * as msRest from "@azure/ms-rest-js";
import * as Mappers from "../models/myWorkbooksMappers";
import * as Parameters from "../models/parameters";
/** Class representing a MyWorkbooks. */
var MyWorkbooks = /** @class */ (function () {
    /**
     * Create a MyWorkbooks.
     * @param {ApplicationInsightsManagementClientContext} client Reference to the service client.
     */
    function MyWorkbooks(client) {
        this.client = client;
    }
    MyWorkbooks.prototype.listByResourceGroup = function (resourceGroupName, category, options, callback) {
        return this.client.sendOperationRequest({
            resourceGroupName: resourceGroupName,
            category: category,
            options: options
        }, listByResourceGroupOperationSpec, callback);
    };
    MyWorkbooks.prototype.listBySubscription = function (category, options, callback) {
        return this.client.sendOperationRequest({
            category: category,
            options: options
        }, listBySubscriptionOperationSpec, callback);
    };
    MyWorkbooks.prototype.get = function (resourceGroupName, resourceName, options, callback) {
        return this.client.sendOperationRequest({
            resourceGroupName: resourceGroupName,
            resourceName: resourceName,
            options: options
        }, getOperationSpec, callback);
    };
    MyWorkbooks.prototype.deleteMethod = function (resourceGroupName, resourceName, options, callback) {
        return this.client.sendOperationRequest({
            resourceGroupName: resourceGroupName,
            resourceName: resourceName,
            options: options
        }, deleteMethodOperationSpec, callback);
    };
    MyWorkbooks.prototype.createOrUpdate = function (resourceGroupName, resourceName, workbookProperties, options, callback) {
        return this.client.sendOperationRequest({
            resourceGroupName: resourceGroupName,
            resourceName: resourceName,
            workbookProperties: workbookProperties,
            options: options
        }, createOrUpdateOperationSpec, callback);
    };
    MyWorkbooks.prototype.update = function (resourceGroupName, resourceName, workbookProperties, options, callback) {
        return this.client.sendOperationRequest({
            resourceGroupName: resourceGroupName,
            resourceName: resourceName,
            workbookProperties: workbookProperties,
            options: options
        }, updateOperationSpec, callback);
    };
    MyWorkbooks.prototype.listByResourceGroupNext = function (nextPageLink, category, options, callback) {
        return this.client.sendOperationRequest({
            nextPageLink: nextPageLink,
            category: category,
            options: options
        }, listByResourceGroupNextOperationSpec, callback);
    };
    MyWorkbooks.prototype.listBySubscriptionNext = function (nextPageLink, category, options, callback) {
        return this.client.sendOperationRequest({
            nextPageLink: nextPageLink,
            category: category,
            options: options
        }, listBySubscriptionNextOperationSpec, callback);
    };
    return MyWorkbooks;
}());
export { MyWorkbooks };
// Operation Specifications
var serializer = new msRest.Serializer(Mappers);
var listByResourceGroupOperationSpec = {
    httpMethod: "GET",
    path: "subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Insights/myWorkbooks",
    urlParameters: [
        Parameters.subscriptionId,
        Parameters.resourceGroupName
    ],
    queryParameters: [
        Parameters.category,
        Parameters.tags,
        Parameters.sourceId,
        Parameters.canFetchContent,
        Parameters.apiVersion3
    ],
    headerParameters: [
        Parameters.acceptLanguage
    ],
    responses: {
        200: {
            bodyMapper: Mappers.MyWorkbooksListResult
        },
        default: {
            bodyMapper: Mappers.MyWorkbookError
        }
    },
    serializer: serializer
};
var listBySubscriptionOperationSpec = {
    httpMethod: "GET",
    path: "subscriptions/{subscriptionId}/providers/Microsoft.Insights/myWorkbooks",
    urlParameters: [
        Parameters.subscriptionId
    ],
    queryParameters: [
        Parameters.category,
        Parameters.tags,
        Parameters.canFetchContent,
        Parameters.apiVersion3
    ],
    headerParameters: [
        Parameters.acceptLanguage
    ],
    responses: {
        200: {
            bodyMapper: Mappers.MyWorkbooksListResult
        },
        default: {
            bodyMapper: Mappers.MyWorkbookError
        }
    },
    serializer: serializer
};
var getOperationSpec = {
    httpMethod: "GET",
    path: "subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Insights/myWorkbooks/{resourceName}",
    urlParameters: [
        Parameters.subscriptionId,
        Parameters.resourceGroupName,
        Parameters.resourceName
    ],
    queryParameters: [
        Parameters.apiVersion3
    ],
    headerParameters: [
        Parameters.acceptLanguage
    ],
    responses: {
        200: {
            bodyMapper: Mappers.MyWorkbook
        },
        default: {
            bodyMapper: Mappers.MyWorkbookError
        }
    },
    serializer: serializer
};
var deleteMethodOperationSpec = {
    httpMethod: "DELETE",
    path: "subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Insights/myWorkbooks/{resourceName}",
    urlParameters: [
        Parameters.subscriptionId,
        Parameters.resourceGroupName,
        Parameters.resourceName
    ],
    queryParameters: [
        Parameters.apiVersion3
    ],
    headerParameters: [
        Parameters.acceptLanguage
    ],
    responses: {
        200: {},
        204: {},
        default: {
            bodyMapper: Mappers.MyWorkbookError
        }
    },
    serializer: serializer
};
var createOrUpdateOperationSpec = {
    httpMethod: "PUT",
    path: "subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Insights/myWorkbooks/{resourceName}",
    urlParameters: [
        Parameters.subscriptionId,
        Parameters.resourceGroupName,
        Parameters.resourceName
    ],
    queryParameters: [
        Parameters.sourceId,
        Parameters.apiVersion3
    ],
    headerParameters: [
        Parameters.acceptLanguage
    ],
    requestBody: {
        parameterPath: "workbookProperties",
        mapper: __assign(__assign({}, Mappers.MyWorkbook), { required: true })
    },
    responses: {
        200: {
            bodyMapper: Mappers.MyWorkbook
        },
        201: {
            bodyMapper: Mappers.MyWorkbook
        },
        default: {
            bodyMapper: Mappers.MyWorkbookError
        }
    },
    serializer: serializer
};
var updateOperationSpec = {
    httpMethod: "PATCH",
    path: "subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Insights/myWorkbooks/{resourceName}",
    urlParameters: [
        Parameters.subscriptionId,
        Parameters.resourceGroupName,
        Parameters.resourceName
    ],
    queryParameters: [
        Parameters.sourceId,
        Parameters.apiVersion3
    ],
    headerParameters: [
        Parameters.acceptLanguage
    ],
    requestBody: {
        parameterPath: "workbookProperties",
        mapper: __assign(__assign({}, Mappers.MyWorkbook), { required: true })
    },
    responses: {
        201: {
            bodyMapper: Mappers.MyWorkbook
        },
        default: {
            bodyMapper: Mappers.MyWorkbookError
        }
    },
    serializer: serializer
};
var listByResourceGroupNextOperationSpec = {
    httpMethod: "GET",
    baseUrl: "https://management.azure.com",
    path: "{nextLink}",
    urlParameters: [
        Parameters.nextPageLink
    ],
    queryParameters: [
        Parameters.category,
        Parameters.tags,
        Parameters.sourceId,
        Parameters.canFetchContent,
        Parameters.apiVersion3
    ],
    headerParameters: [
        Parameters.acceptLanguage
    ],
    responses: {
        200: {
            bodyMapper: Mappers.MyWorkbooksListResult
        },
        default: {
            bodyMapper: Mappers.MyWorkbookError
        }
    },
    serializer: serializer
};
var listBySubscriptionNextOperationSpec = {
    httpMethod: "GET",
    baseUrl: "https://management.azure.com",
    path: "{nextLink}",
    urlParameters: [
        Parameters.nextPageLink
    ],
    queryParameters: [
        Parameters.category,
        Parameters.tags,
        Parameters.canFetchContent,
        Parameters.apiVersion3
    ],
    headerParameters: [
        Parameters.acceptLanguage
    ],
    responses: {
        200: {
            bodyMapper: Mappers.MyWorkbooksListResult
        },
        default: {
            bodyMapper: Mappers.MyWorkbookError
        }
    },
    serializer: serializer
};
