// @azure/arm-appinsights/esm/operations/workbooks.js
/*
 * Copyright (c) Microsoft Corporation.
 * Licensed under the MIT License.
 *
 * Code generated by Microsoft (R) AutoRest Code Generator.
 * Changes may cause incorrect behavior and will be lost if the code is
 * regenerated.
 */
import { __assign } from "tslib";
import * as msRest from "@azure/ms-rest-js";
import * as Mappers from "../models/workbooksMappers";
import * as Parameters from "../models/parameters";
/** Class representing a Workbooks. */
var Workbooks = /** @class */ (function () {
    /**
     * Create a Workbooks.
     * @param {ApplicationInsightsManagementClientContext} client Reference to the service client.
     */
    function Workbooks(client) {
        this.client = client;
    }
    Workbooks.prototype.listBySubscription = function (category, options, callback) {
        return this.client.sendOperationRequest({
            category: category,
            options: options
        }, listBySubscriptionOperationSpec, callback);
    };
    Workbooks.prototype.listByResourceGroup = function (resourceGroupName, category, options, callback) {
        return this.client.sendOperationRequest({
            resourceGroupName: resourceGroupName,
            category: category,
            options: options
        }, listByResourceGroupOperationSpec, callback);
    };
    Workbooks.prototype.get = function (resourceGroupName, resourceName, options, callback) {
        return this.client.sendOperationRequest({
            resourceGroupName: resourceGroupName,
            resourceName: resourceName,
            options: options
        }, getOperationSpec, callback);
    };
    Workbooks.prototype.deleteMethod = function (resourceGroupName, resourceName, options, callback) {
        return this.client.sendOperationRequest({
            resourceGroupName: resourceGroupName,
            resourceName: resourceName,
            options: options
        }, deleteMethodOperationSpec, callback);
    };
    Workbooks.prototype.createOrUpdate = function (resourceGroupName, resourceName, workbookProperties, options, callback) {
        return this.client.sendOperationRequest({
            resourceGroupName: resourceGroupName,
            resourceName: resourceName,
            workbookProperties: workbookProperties,
            options: options
        }, createOrUpdateOperationSpec, callback);
    };
    Workbooks.prototype.update = function (resourceGroupName, resourceName, options, callback) {
        return this.client.sendOperationRequest({
            resourceGroupName: resourceGroupName,
            resourceName: resourceName,
            options: options
        }, updateOperationSpec, callback);
    };
    Workbooks.prototype.revisionsList = function (resourceGroupName, resourceName, options, callback) {
        return this.client.sendOperationRequest({
            resourceGroupName: resourceGroupName,
            resourceName: resourceName,
            options: options
        }, revisionsListOperationSpec, callback);
    };
    Workbooks.prototype.revisionGet = function (resourceGroupName, resourceName, revisionId, options, callback) {
        return this.client.sendOperationRequest({
            resourceGroupName: resourceGroupName,
            resourceName: resourceName,
            revisionId: revisionId,
            options: options
        }, revisionGetOperationSpec, callback);
    };
    Workbooks.prototype.listBySubscriptionNext = function (nextPageLink, category, options, callback) {
        return this.client.sendOperationRequest({
            nextPageLink: nextPageLink,
            category: category,
            options: options
        }, listBySubscriptionNextOperationSpec, callback);
    };
    Workbooks.prototype.listByResourceGroupNext = function (nextPageLink, category, options, callback) {
        return this.client.sendOperationRequest({
            nextPageLink: nextPageLink,
            category: category,
            options: options
        }, listByResourceGroupNextOperationSpec, callback);
    };
    Workbooks.prototype.revisionsListNext = function (nextPageLink, options, callback) {
        return this.client.sendOperationRequest({
            nextPageLink: nextPageLink,
            options: options
        }, revisionsListNextOperationSpec, callback);
    };
    return Workbooks;
}());
export { Workbooks };
// Operation Specifications
var serializer = new msRest.Serializer(Mappers);
var listBySubscriptionOperationSpec = {
    httpMethod: "GET",
    path: "subscriptions/{subscriptionId}/providers/Microsoft.Insights/workbooks",
    urlParameters: [
        Parameters.subscriptionId
    ],
    queryParameters: [
        Parameters.category,
        Parameters.tags,
        Parameters.canFetchContent,
        Parameters.apiVersion4
    ],
    headerParameters: [
        Parameters.acceptLanguage
    ],
    responses: {
        200: {
            bodyMapper: Mappers.WorkbooksListResult
        },
        default: {
            bodyMapper: Mappers.WorkbookError
        }
    },
    serializer: serializer
};
var listByResourceGroupOperationSpec = {
    httpMethod: "GET",
    path: "subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Insights/workbooks",
    urlParameters: [
        Parameters.subscriptionId,
        Parameters.resourceGroupName
    ],
    queryParameters: [
        Parameters.category,
        Parameters.tags,
        Parameters.sourceId,
        Parameters.canFetchContent,
        Parameters.apiVersion4
    ],
    headerParameters: [
        Parameters.acceptLanguage
    ],
    responses: {
        200: {
            bodyMapper: Mappers.WorkbooksListResult
        },
        default: {
            bodyMapper: Mappers.WorkbookError
        }
    },
    serializer: serializer
};
var getOperationSpec = {
    httpMethod: "GET",
    path: "subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Insights/workbooks/{resourceName}",
    urlParameters: [
        Parameters.subscriptionId,
        Parameters.resourceGroupName,
        Parameters.resourceName
    ],
    queryParameters: [
        Parameters.apiVersion4
    ],
    headerParameters: [
        Parameters.acceptLanguage
    ],
    responses: {
        200: {
            bodyMapper: Mappers.Workbook
        },
        default: {
            bodyMapper: Mappers.WorkbookError
        }
    },
    serializer: serializer
};
var deleteMethodOperationSpec = {
    httpMethod: "DELETE",
    path: "subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Insights/workbooks/{resourceName}",
    urlParameters: [
        Parameters.subscriptionId,
        Parameters.resourceGroupName,
        Parameters.resourceName
    ],
    queryParameters: [
        Parameters.apiVersion4
    ],
    headerParameters: [
        Parameters.acceptLanguage
    ],
    responses: {
        200: {},
        204: {},
        default: {
            bodyMapper: Mappers.WorkbookError
        }
    },
    serializer: serializer
};
var createOrUpdateOperationSpec = {
    httpMethod: "PUT",
    path: "subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Insights/workbooks/{resourceName}",
    urlParameters: [
        Parameters.subscriptionId,
        Parameters.resourceGroupName,
        Parameters.resourceName
    ],
    queryParameters: [
        Parameters.sourceId,
        Parameters.apiVersion4
    ],
    headerParameters: [
        Parameters.acceptLanguage
    ],
    requestBody: {
        parameterPath: "workbookProperties",
        mapper: __assign(__assign({}, Mappers.Workbook), { required: true })
    },
    responses: {
        200: {
            bodyMapper: Mappers.Workbook
        },
        201: {
            bodyMapper: Mappers.Workbook
        },
        default: {
            bodyMapper: Mappers.WorkbookError
        }
    },
    serializer: serializer
};
var updateOperationSpec = {
    httpMethod: "PATCH",
    path: "subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Insights/workbooks/{resourceName}",
    urlParameters: [
        Parameters.subscriptionId,
        Parameters.resourceGroupName,
        Parameters.resourceName
    ],
    queryParameters: [
        Parameters.sourceId,
        Parameters.apiVersion4
    ],
    headerParameters: [
        Parameters.acceptLanguage
    ],
    requestBody: {
        parameterPath: [
            "options",
            "workbookUpdateParameters"
        ],
        mapper: Mappers.WorkbookUpdateParameters
    },
    responses: {
        201: {
            bodyMapper: Mappers.Workbook
        },
        default: {
            bodyMapper: Mappers.WorkbookError
        }
    },
    serializer: serializer
};
var revisionsListOperationSpec = {
    httpMethod: "GET",
    path: "subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Insights/workbooks/{resourceName}/revisions",
    urlParameters: [
        Parameters.subscriptionId,
        Parameters.resourceGroupName,
        Parameters.resourceName
    ],
    queryParameters: [
        Parameters.apiVersion4
    ],
    headerParameters: [
        Parameters.acceptLanguage
    ],
    responses: {
        200: {
            bodyMapper: Mappers.WorkbooksListResult
        },
        default: {
            bodyMapper: Mappers.WorkbookError
        }
    },
    serializer: serializer
};
var revisionGetOperationSpec = {
    httpMethod: "GET",
    path: "subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Insights/workbooks/{resourceName}/revisions/{revisionId}",
    urlParameters: [
        Parameters.subscriptionId,
        Parameters.resourceGroupName,
        Parameters.resourceName,
        Parameters.revisionId
    ],
    queryParameters: [
        Parameters.apiVersion4
    ],
    headerParameters: [
        Parameters.acceptLanguage
    ],
    responses: {
        200: {
            bodyMapper: Mappers.Workbook
        },
        default: {
            bodyMapper: Mappers.WorkbookError
        }
    },
    serializer: serializer
};
var listBySubscriptionNextOperationSpec = {
    httpMethod: "GET",
    baseUrl: "https://management.azure.com",
    path: "{nextLink}",
    urlParameters: [
        Parameters.nextPageLink
    ],
    queryParameters: [
        Parameters.category,
        Parameters.tags,
        Parameters.canFetchContent,
        Parameters.apiVersion4
    ],
    headerParameters: [
        Parameters.acceptLanguage
    ],
    responses: {
        200: {
            bodyMapper: Mappers.WorkbooksListResult
        },
        default: {
            bodyMapper: Mappers.WorkbookError
        }
    },
    serializer: serializer
};
var listByResourceGroupNextOperationSpec = {
    httpMethod: "GET",
    baseUrl: "https://management.azure.com",
    path: "{nextLink}",
    urlParameters: [
        Parameters.nextPageLink
    ],
    queryParameters: [
        Parameters.category,
        Parameters.tags,
        Parameters.sourceId,
        Parameters.canFetchContent,
        Parameters.apiVersion4
    ],
    headerParameters: [
        Parameters.acceptLanguage
    ],
    responses: {
        200: {
            bodyMapper: Mappers.WorkbooksListResult
        },
        default: {
            bodyMapper: Mappers.WorkbookError
        }
    },
    serializer: serializer
};
var revisionsListNextOperationSpec = {
    httpMethod: "GET",
    baseUrl: "https://management.azure.com",
    path: "{nextLink}",
    urlParameters: [
        Parameters.nextPageLink
    ],
    queryParameters: [
        Parameters.apiVersion4
    ],
    headerParameters: [
        Parameters.acceptLanguage
    ],
    responses: {
        200: {
            bodyMapper: Mappers.WorkbooksListResult
        },
        default: {
            bodyMapper: Mappers.WorkbookError
        }
    },
    serializer: serializer
};
