// @azure/arm-deviceupdate/dist-esm/src/operations/privateEndpointConnectionProxies.js
/*
 * Copyright (c) Microsoft Corporation.
 * Licensed under the MIT License.
 *
 * Code generated by Microsoft (R) AutoRest Code Generator.
 * Changes may cause incorrect behavior and will be lost if the code is regenerated.
 */
import { __asyncDelegator, __asyncGenerator, __asyncValues, __await, __awaiter } from "tslib";
import * as coreClient from "@azure/core-client";
import * as Mappers from "../models/mappers";
import * as Parameters from "../models/parameters";
import { createHttpPoller } from "@azure/core-lro";
import { createLroSpec } from "../lroImpl";
/// <reference lib="esnext.asynciterable" />
/** Class containing PrivateEndpointConnectionProxies operations. */
export class PrivateEndpointConnectionProxiesImpl {
    /**
     * Initialize a new instance of the class PrivateEndpointConnectionProxies class.
     * @param client Reference to the service client
     */
    constructor(client) {
        this.client = client;
    }
    /**
     * (INTERNAL - DO NOT USE) List all private endpoint connection proxies in a device update account.
     * @param resourceGroupName The resource group name.
     * @param accountName Account name.
     * @param options The options parameters.
     */
    listByAccount(resourceGroupName, accountName, options) {
        const iter = this.listByAccountPagingAll(resourceGroupName, accountName, options);
        return {
            next() {
                return iter.next();
            },
            [Symbol.asyncIterator]() {
                return this;
            },
            byPage: (settings) => {
                if (settings === null || settings === void 0 ? void 0 : settings.maxPageSize) {
                    throw new Error("maxPageSize is not supported by this operation.");
                }
                return this.listByAccountPagingPage(resourceGroupName, accountName, options, settings);
            }
        };
    }
    listByAccountPagingPage(resourceGroupName, accountName, options, _settings) {
        return __asyncGenerator(this, arguments, function* listByAccountPagingPage_1() {
            let result;
            result = yield __await(this._listByAccount(resourceGroupName, accountName, options));
            yield yield __await(result.value || []);
        });
    }
    listByAccountPagingAll(resourceGroupName, accountName, options) {
        return __asyncGenerator(this, arguments, function* listByAccountPagingAll_1() {
            var _a, e_1, _b, _c;
            try {
                for (var _d = true, _e = __asyncValues(this.listByAccountPagingPage(resourceGroupName, accountName, options)), _f; _f = yield __await(_e.next()), _a = _f.done, !_a; _d = true) {
                    _c = _f.value;
                    _d = false;
                    const page = _c;
                    yield __await(yield* __asyncDelegator(__asyncValues(page)));
                }
            }
            catch (e_1_1) { e_1 = { error: e_1_1 }; }
            finally {
                try {
                    if (!_d && !_a && (_b = _e.return)) yield __await(_b.call(_e));
                }
                finally { if (e_1) throw e_1.error; }
            }
        });
    }
    /**
     * (INTERNAL - DO NOT USE) List all private endpoint connection proxies in a device update account.
     * @param resourceGroupName The resource group name.
     * @param accountName Account name.
     * @param options The options parameters.
     */
    _listByAccount(resourceGroupName, accountName, options) {
        return this.client.sendOperationRequest({ resourceGroupName, accountName, options }, listByAccountOperationSpec);
    }
    /**
     * (INTERNAL - DO NOT USE) Validates a private endpoint connection proxy object.
     * @param resourceGroupName The resource group name.
     * @param accountName Account name.
     * @param privateEndpointConnectionProxyId The ID of the private endpoint connection proxy object.
     * @param privateEndpointConnectionProxy The parameters for creating a private endpoint connection
     *                                       proxy.
     * @param options The options parameters.
     */
    validate(resourceGroupName, accountName, privateEndpointConnectionProxyId, privateEndpointConnectionProxy, options) {
        return this.client.sendOperationRequest({
            resourceGroupName,
            accountName,
            privateEndpointConnectionProxyId,
            privateEndpointConnectionProxy,
            options
        }, validateOperationSpec);
    }
    /**
     * (INTERNAL - DO NOT USE) Updates a private endpoint inside the private endpoint connection proxy
     * object.
     * @param resourceGroupName The resource group name.
     * @param accountName Account name.
     * @param privateEndpointConnectionProxyId The ID of the private endpoint connection proxy object.
     * @param privateEndpointUpdate The parameters for updating a private endpoint connection proxy.
     * @param options The options parameters.
     */
    updatePrivateEndpointProperties(resourceGroupName, accountName, privateEndpointConnectionProxyId, privateEndpointUpdate, options) {
        return this.client.sendOperationRequest({
            resourceGroupName,
            accountName,
            privateEndpointConnectionProxyId,
            privateEndpointUpdate,
            options
        }, updatePrivateEndpointPropertiesOperationSpec);
    }
    /**
     * (INTERNAL - DO NOT USE) Get the specified private endpoint connection proxy associated with the
     * device update account.
     * @param resourceGroupName The resource group name.
     * @param accountName Account name.
     * @param privateEndpointConnectionProxyId The ID of the private endpoint connection proxy object.
     * @param options The options parameters.
     */
    get(resourceGroupName, accountName, privateEndpointConnectionProxyId, options) {
        return this.client.sendOperationRequest({
            resourceGroupName,
            accountName,
            privateEndpointConnectionProxyId,
            options
        }, getOperationSpec);
    }
    /**
     * (INTERNAL - DO NOT USE) Creates or updates the specified private endpoint connection proxy resource
     * associated with the device update account.
     * @param resourceGroupName The resource group name.
     * @param accountName Account name.
     * @param privateEndpointConnectionProxyId The ID of the private endpoint connection proxy object.
     * @param privateEndpointConnectionProxy The parameters for creating a private endpoint connection
     *                                       proxy.
     * @param options The options parameters.
     */
    beginCreateOrUpdate(resourceGroupName, accountName, privateEndpointConnectionProxyId, privateEndpointConnectionProxy, options) {
        return __awaiter(this, void 0, void 0, function* () {
            const directSendOperation = (args, spec) => __awaiter(this, void 0, void 0, function* () {
                return this.client.sendOperationRequest(args, spec);
            });
            const sendOperationFn = (args, spec) => __awaiter(this, void 0, void 0, function* () {
                var _a;
                let currentRawResponse = undefined;
                const providedCallback = (_a = args.options) === null || _a === void 0 ? void 0 : _a.onResponse;
                const callback = (rawResponse, flatResponse) => {
                    currentRawResponse = rawResponse;
                    providedCallback === null || providedCallback === void 0 ? void 0 : providedCallback(rawResponse, flatResponse);
                };
                const updatedArgs = Object.assign(Object.assign({}, args), { options: Object.assign(Object.assign({}, args.options), { onResponse: callback }) });
                const flatResponse = yield directSendOperation(updatedArgs, spec);
                return {
                    flatResponse,
                    rawResponse: {
                        statusCode: currentRawResponse.status,
                        body: currentRawResponse.parsedBody,
                        headers: currentRawResponse.headers.toJSON()
                    }
                };
            });
            const lro = createLroSpec({
                sendOperationFn,
                args: {
                    resourceGroupName,
                    accountName,
                    privateEndpointConnectionProxyId,
                    privateEndpointConnectionProxy,
                    options
                },
                spec: createOrUpdateOperationSpec
            });
            const poller = yield createHttpPoller(lro, {
                restoreFrom: options === null || options === void 0 ? void 0 : options.resumeFrom,
                intervalInMs: options === null || options === void 0 ? void 0 : options.updateIntervalInMs,
                resourceLocationConfig: "azure-async-operation"
            });
            yield poller.poll();
            return poller;
        });
    }
    /**
     * (INTERNAL - DO NOT USE) Creates or updates the specified private endpoint connection proxy resource
     * associated with the device update account.
     * @param resourceGroupName The resource group name.
     * @param accountName Account name.
     * @param privateEndpointConnectionProxyId The ID of the private endpoint connection proxy object.
     * @param privateEndpointConnectionProxy The parameters for creating a private endpoint connection
     *                                       proxy.
     * @param options The options parameters.
     */
    beginCreateOrUpdateAndWait(resourceGroupName, accountName, privateEndpointConnectionProxyId, privateEndpointConnectionProxy, options) {
        return __awaiter(this, void 0, void 0, function* () {
            const poller = yield this.beginCreateOrUpdate(resourceGroupName, accountName, privateEndpointConnectionProxyId, privateEndpointConnectionProxy, options);
            return poller.pollUntilDone();
        });
    }
    /**
     * (INTERNAL - DO NOT USE) Deletes the specified private endpoint connection proxy associated with the
     * device update account.
     * @param resourceGroupName The resource group name.
     * @param accountName Account name.
     * @param privateEndpointConnectionProxyId The ID of the private endpoint connection proxy object.
     * @param options The options parameters.
     */
    beginDelete(resourceGroupName, accountName, privateEndpointConnectionProxyId, options) {
        return __awaiter(this, void 0, void 0, function* () {
            const directSendOperation = (args, spec) => __awaiter(this, void 0, void 0, function* () {
                return this.client.sendOperationRequest(args, spec);
            });
            const sendOperationFn = (args, spec) => __awaiter(this, void 0, void 0, function* () {
                var _a;
                let currentRawResponse = undefined;
                const providedCallback = (_a = args.options) === null || _a === void 0 ? void 0 : _a.onResponse;
                const callback = (rawResponse, flatResponse) => {
                    currentRawResponse = rawResponse;
                    providedCallback === null || providedCallback === void 0 ? void 0 : providedCallback(rawResponse, flatResponse);
                };
                const updatedArgs = Object.assign(Object.assign({}, args), { options: Object.assign(Object.assign({}, args.options), { onResponse: callback }) });
                const flatResponse = yield directSendOperation(updatedArgs, spec);
                return {
                    flatResponse,
                    rawResponse: {
                        statusCode: currentRawResponse.status,
                        body: currentRawResponse.parsedBody,
                        headers: currentRawResponse.headers.toJSON()
                    }
                };
            });
            const lro = createLroSpec({
                sendOperationFn,
                args: {
                    resourceGroupName,
                    accountName,
                    privateEndpointConnectionProxyId,
                    options
                },
                spec: deleteOperationSpec
            });
            const poller = yield createHttpPoller(lro, {
                restoreFrom: options === null || options === void 0 ? void 0 : options.resumeFrom,
                intervalInMs: options === null || options === void 0 ? void 0 : options.updateIntervalInMs,
                resourceLocationConfig: "location"
            });
            yield poller.poll();
            return poller;
        });
    }
    /**
     * (INTERNAL - DO NOT USE) Deletes the specified private endpoint connection proxy associated with the
     * device update account.
     * @param resourceGroupName The resource group name.
     * @param accountName Account name.
     * @param privateEndpointConnectionProxyId The ID of the private endpoint connection proxy object.
     * @param options The options parameters.
     */
    beginDeleteAndWait(resourceGroupName, accountName, privateEndpointConnectionProxyId, options) {
        return __awaiter(this, void 0, void 0, function* () {
            const poller = yield this.beginDelete(resourceGroupName, accountName, privateEndpointConnectionProxyId, options);
            return poller.pollUntilDone();
        });
    }
}
// Operation Specifications
const serializer = coreClient.createSerializer(Mappers, /* isXml */ false);
const listByAccountOperationSpec = {
    path: "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.DeviceUpdate/accounts/{accountName}/privateEndpointConnectionProxies",
    httpMethod: "GET",
    responses: {
        200: {
            bodyMapper: Mappers.PrivateEndpointConnectionProxyListResult
        },
        default: {
            bodyMapper: Mappers.ErrorResponse
        }
    },
    queryParameters: [Parameters.apiVersion],
    urlParameters: [
        Parameters.$host,
        Parameters.subscriptionId,
        Parameters.resourceGroupName,
        Parameters.accountName
    ],
    headerParameters: [Parameters.accept],
    serializer
};
const validateOperationSpec = {
    path: "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.DeviceUpdate/accounts/{accountName}/privateEndpointConnectionProxies/{privateEndpointConnectionProxyId}/validate",
    httpMethod: "POST",
    responses: {
        200: {},
        default: {
            bodyMapper: Mappers.ErrorResponse
        }
    },
    requestBody: Parameters.privateEndpointConnectionProxy,
    queryParameters: [Parameters.apiVersion],
    urlParameters: [
        Parameters.$host,
        Parameters.subscriptionId,
        Parameters.resourceGroupName,
        Parameters.accountName,
        Parameters.privateEndpointConnectionProxyId
    ],
    headerParameters: [Parameters.contentType, Parameters.accept],
    mediaType: "json",
    serializer
};
const updatePrivateEndpointPropertiesOperationSpec = {
    path: "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.DeviceUpdate/accounts/{accountName}/privateEndpointConnectionProxies/{privateEndpointConnectionProxyId}/updatePrivateEndpointProperties",
    httpMethod: "POST",
    responses: {
        200: {},
        default: {
            bodyMapper: Mappers.ErrorResponse
        }
    },
    requestBody: Parameters.privateEndpointUpdate,
    queryParameters: [Parameters.apiVersion],
    urlParameters: [
        Parameters.$host,
        Parameters.subscriptionId,
        Parameters.resourceGroupName,
        Parameters.accountName,
        Parameters.privateEndpointConnectionProxyId
    ],
    headerParameters: [Parameters.contentType, Parameters.accept],
    mediaType: "json",
    serializer
};
const getOperationSpec = {
    path: "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.DeviceUpdate/accounts/{accountName}/privateEndpointConnectionProxies/{privateEndpointConnectionProxyId}",
    httpMethod: "GET",
    responses: {
        200: {
            bodyMapper: Mappers.PrivateEndpointConnectionProxy
        },
        default: {
            bodyMapper: Mappers.ErrorResponse
        }
    },
    queryParameters: [Parameters.apiVersion],
    urlParameters: [
        Parameters.$host,
        Parameters.subscriptionId,
        Parameters.resourceGroupName,
        Parameters.accountName,
        Parameters.privateEndpointConnectionProxyId
    ],
    headerParameters: [Parameters.accept],
    serializer
};
const createOrUpdateOperationSpec = {
    path: "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.DeviceUpdate/accounts/{accountName}/privateEndpointConnectionProxies/{privateEndpointConnectionProxyId}",
    httpMethod: "PUT",
    responses: {
        200: {
            bodyMapper: Mappers.PrivateEndpointConnectionProxy
        },
        201: {
            bodyMapper: Mappers.PrivateEndpointConnectionProxy
        },
        202: {
            bodyMapper: Mappers.PrivateEndpointConnectionProxy
        },
        204: {
            bodyMapper: Mappers.PrivateEndpointConnectionProxy
        },
        default: {
            bodyMapper: Mappers.ErrorResponse
        }
    },
    requestBody: Parameters.privateEndpointConnectionProxy,
    queryParameters: [Parameters.apiVersion],
    urlParameters: [
        Parameters.$host,
        Parameters.subscriptionId,
        Parameters.resourceGroupName,
        Parameters.accountName,
        Parameters.privateEndpointConnectionProxyId
    ],
    headerParameters: [Parameters.contentType, Parameters.accept],
    mediaType: "json",
    serializer
};
const deleteOperationSpec = {
    path: "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.DeviceUpdate/accounts/{accountName}/privateEndpointConnectionProxies/{privateEndpointConnectionProxyId}",
    httpMethod: "DELETE",
    responses: {
        200: {},
        201: {},
        202: {},
        204: {},
        default: {
            bodyMapper: Mappers.ErrorResponse
        }
    },
    queryParameters: [Parameters.apiVersion],
    urlParameters: [
        Parameters.$host,
        Parameters.subscriptionId,
        Parameters.resourceGroupName,
        Parameters.accountName,
        Parameters.privateEndpointConnectionProxyId
    ],
    headerParameters: [Parameters.accept],
    serializer
};
