// @aws-sdk/client-mediaconvert/dist-es/models/models_0.js
export const RequiredFlag = {
    DISABLED: "DISABLED",
    ENABLED: "ENABLED",
};
export const AudioChannelTag = {
    C: "C",
    CS: "CS",
    HI: "HI",
    L: "L",
    LC: "LC",
    LFE: "LFE",
    LFE2: "LFE2",
    LS: "LS",
    LSD: "LSD",
    LT: "LT",
    LW: "LW",
    M: "M",
    NAR: "NAR",
    R: "R",
    RC: "RC",
    RS: "RS",
    RSD: "RSD",
    RSL: "RSL",
    RSR: "RSR",
    RT: "RT",
    RW: "RW",
    TBC: "TBC",
    TBL: "TBL",
    TBR: "TBR",
    TCS: "TCS",
    VHC: "VHC",
    VHL: "VHL",
    VHR: "VHR",
};
export const AudioNormalizationAlgorithm = {
    ITU_BS_1770_1: "ITU_BS_1770_1",
    ITU_BS_1770_2: "ITU_BS_1770_2",
    ITU_BS_1770_3: "ITU_BS_1770_3",
    ITU_BS_1770_4: "ITU_BS_1770_4",
};
export const AudioNormalizationAlgorithmControl = {
    CORRECT_AUDIO: "CORRECT_AUDIO",
    MEASURE_ONLY: "MEASURE_ONLY",
};
export const AudioNormalizationLoudnessLogging = {
    DONT_LOG: "DONT_LOG",
    LOG: "LOG",
};
export const AudioNormalizationPeakCalculation = {
    NONE: "NONE",
    TRUE_PEAK: "TRUE_PEAK",
};
export const AudioTypeControl = {
    FOLLOW_INPUT: "FOLLOW_INPUT",
    USE_CONFIGURED: "USE_CONFIGURED",
};
export const AacAudioDescriptionBroadcasterMix = {
    BROADCASTER_MIXED_AD: "BROADCASTER_MIXED_AD",
    NORMAL: "NORMAL",
};
export const AacCodecProfile = {
    HEV1: "HEV1",
    HEV2: "HEV2",
    LC: "LC",
};
export const AacCodingMode = {
    AD_RECEIVER_MIX: "AD_RECEIVER_MIX",
    CODING_MODE_1_0: "CODING_MODE_1_0",
    CODING_MODE_1_1: "CODING_MODE_1_1",
    CODING_MODE_2_0: "CODING_MODE_2_0",
    CODING_MODE_5_1: "CODING_MODE_5_1",
};
export const AacRateControlMode = {
    CBR: "CBR",
    VBR: "VBR",
};
export const AacRawFormat = {
    LATM_LOAS: "LATM_LOAS",
    NONE: "NONE",
};
export const AacSpecification = {
    MPEG2: "MPEG2",
    MPEG4: "MPEG4",
};
export const AacVbrQuality = {
    HIGH: "HIGH",
    LOW: "LOW",
    MEDIUM_HIGH: "MEDIUM_HIGH",
    MEDIUM_LOW: "MEDIUM_LOW",
};
export const Ac3BitstreamMode = {
    COMMENTARY: "COMMENTARY",
    COMPLETE_MAIN: "COMPLETE_MAIN",
    DIALOGUE: "DIALOGUE",
    EMERGENCY: "EMERGENCY",
    HEARING_IMPAIRED: "HEARING_IMPAIRED",
    MUSIC_AND_EFFECTS: "MUSIC_AND_EFFECTS",
    VISUALLY_IMPAIRED: "VISUALLY_IMPAIRED",
    VOICE_OVER: "VOICE_OVER",
};
export const Ac3CodingMode = {
    CODING_MODE_1_0: "CODING_MODE_1_0",
    CODING_MODE_1_1: "CODING_MODE_1_1",
    CODING_MODE_2_0: "CODING_MODE_2_0",
    CODING_MODE_3_2_LFE: "CODING_MODE_3_2_LFE",
};
export const Ac3DynamicRangeCompressionLine = {
    FILM_LIGHT: "FILM_LIGHT",
    FILM_STANDARD: "FILM_STANDARD",
    MUSIC_LIGHT: "MUSIC_LIGHT",
    MUSIC_STANDARD: "MUSIC_STANDARD",
    NONE: "NONE",
    SPEECH: "SPEECH",
};
export const Ac3DynamicRangeCompressionProfile = {
    FILM_STANDARD: "FILM_STANDARD",
    NONE: "NONE",
};
export const Ac3DynamicRangeCompressionRf = {
    FILM_LIGHT: "FILM_LIGHT",
    FILM_STANDARD: "FILM_STANDARD",
    MUSIC_LIGHT: "MUSIC_LIGHT",
    MUSIC_STANDARD: "MUSIC_STANDARD",
    NONE: "NONE",
    SPEECH: "SPEECH",
};
export const Ac3LfeFilter = {
    DISABLED: "DISABLED",
    ENABLED: "ENABLED",
};
export const Ac3MetadataControl = {
    FOLLOW_INPUT: "FOLLOW_INPUT",
    USE_CONFIGURED: "USE_CONFIGURED",
};
export const AudioCodec = {
    AAC: "AAC",
    AC3: "AC3",
    AIFF: "AIFF",
    EAC3: "EAC3",
    EAC3_ATMOS: "EAC3_ATMOS",
    FLAC: "FLAC",
    MP2: "MP2",
    MP3: "MP3",
    OPUS: "OPUS",
    PASSTHROUGH: "PASSTHROUGH",
    VORBIS: "VORBIS",
    WAV: "WAV",
};
export const Eac3AtmosBitstreamMode = {
    COMPLETE_MAIN: "COMPLETE_MAIN",
};
export const Eac3AtmosCodingMode = {
    CODING_MODE_5_1_4: "CODING_MODE_5_1_4",
    CODING_MODE_7_1_4: "CODING_MODE_7_1_4",
    CODING_MODE_9_1_6: "CODING_MODE_9_1_6",
    CODING_MODE_AUTO: "CODING_MODE_AUTO",
};
export const Eac3AtmosDialogueIntelligence = {
    DISABLED: "DISABLED",
    ENABLED: "ENABLED",
};
export const Eac3AtmosDownmixControl = {
    INITIALIZE_FROM_SOURCE: "INITIALIZE_FROM_SOURCE",
    SPECIFIED: "SPECIFIED",
};
export const Eac3AtmosDynamicRangeCompressionLine = {
    FILM_LIGHT: "FILM_LIGHT",
    FILM_STANDARD: "FILM_STANDARD",
    MUSIC_LIGHT: "MUSIC_LIGHT",
    MUSIC_STANDARD: "MUSIC_STANDARD",
    NONE: "NONE",
    SPEECH: "SPEECH",
};
export const Eac3AtmosDynamicRangeCompressionRf = {
    FILM_LIGHT: "FILM_LIGHT",
    FILM_STANDARD: "FILM_STANDARD",
    MUSIC_LIGHT: "MUSIC_LIGHT",
    MUSIC_STANDARD: "MUSIC_STANDARD",
    NONE: "NONE",
    SPEECH: "SPEECH",
};
export const Eac3AtmosDynamicRangeControl = {
    INITIALIZE_FROM_SOURCE: "INITIALIZE_FROM_SOURCE",
    SPECIFIED: "SPECIFIED",
};
export const Eac3AtmosMeteringMode = {
    ITU_BS_1770_1: "ITU_BS_1770_1",
    ITU_BS_1770_2: "ITU_BS_1770_2",
    ITU_BS_1770_3: "ITU_BS_1770_3",
    ITU_BS_1770_4: "ITU_BS_1770_4",
    LEQ_A: "LEQ_A",
};
export const Eac3AtmosStereoDownmix = {
    DPL2: "DPL2",
    NOT_INDICATED: "NOT_INDICATED",
    STEREO: "STEREO",
    SURROUND: "SURROUND",
};
export const Eac3AtmosSurroundExMode = {
    DISABLED: "DISABLED",
    ENABLED: "ENABLED",
    NOT_INDICATED: "NOT_INDICATED",
};
export const Eac3AttenuationControl = {
    ATTENUATE_3_DB: "ATTENUATE_3_DB",
    NONE: "NONE",
};
export const Eac3BitstreamMode = {
    COMMENTARY: "COMMENTARY",
    COMPLETE_MAIN: "COMPLETE_MAIN",
    EMERGENCY: "EMERGENCY",
    HEARING_IMPAIRED: "HEARING_IMPAIRED",
    VISUALLY_IMPAIRED: "VISUALLY_IMPAIRED",
};
export const Eac3CodingMode = {
    CODING_MODE_1_0: "CODING_MODE_1_0",
    CODING_MODE_2_0: "CODING_MODE_2_0",
    CODING_MODE_3_2: "CODING_MODE_3_2",
};
export const Eac3DcFilter = {
    DISABLED: "DISABLED",
    ENABLED: "ENABLED",
};
export const Eac3DynamicRangeCompressionLine = {
    FILM_LIGHT: "FILM_LIGHT",
    FILM_STANDARD: "FILM_STANDARD",
    MUSIC_LIGHT: "MUSIC_LIGHT",
    MUSIC_STANDARD: "MUSIC_STANDARD",
    NONE: "NONE",
    SPEECH: "SPEECH",
};
export const Eac3DynamicRangeCompressionRf = {
    FILM_LIGHT: "FILM_LIGHT",
    FILM_STANDARD: "FILM_STANDARD",
    MUSIC_LIGHT: "MUSIC_LIGHT",
    MUSIC_STANDARD: "MUSIC_STANDARD",
    NONE: "NONE",
    SPEECH: "SPEECH",
};
export const Eac3LfeControl = {
    LFE: "LFE",
    NO_LFE: "NO_LFE",
};
export const Eac3LfeFilter = {
    DISABLED: "DISABLED",
    ENABLED: "ENABLED",
};
export const Eac3MetadataControl = {
    FOLLOW_INPUT: "FOLLOW_INPUT",
    USE_CONFIGURED: "USE_CONFIGURED",
};
export const Eac3PassthroughControl = {
    NO_PASSTHROUGH: "NO_PASSTHROUGH",
    WHEN_POSSIBLE: "WHEN_POSSIBLE",
};
export const Eac3PhaseControl = {
    NO_SHIFT: "NO_SHIFT",
    SHIFT_90_DEGREES: "SHIFT_90_DEGREES",
};
export const Eac3StereoDownmix = {
    DPL2: "DPL2",
    LO_RO: "LO_RO",
    LT_RT: "LT_RT",
    NOT_INDICATED: "NOT_INDICATED",
};
export const Eac3SurroundExMode = {
    DISABLED: "DISABLED",
    ENABLED: "ENABLED",
    NOT_INDICATED: "NOT_INDICATED",
};
export const Eac3SurroundMode = {
    DISABLED: "DISABLED",
    ENABLED: "ENABLED",
    NOT_INDICATED: "NOT_INDICATED",
};
export const Mp3RateControlMode = {
    CBR: "CBR",
    VBR: "VBR",
};
export const WavFormat = {
    RF64: "RF64",
    RIFF: "RIFF",
};
export const LanguageCode = {
    AAR: "AAR",
    ABK: "ABK",
    AFR: "AFR",
    AKA: "AKA",
    AMH: "AMH",
    ARA: "ARA",
    ARG: "ARG",
    ASM: "ASM",
    AVA: "AVA",
    AVE: "AVE",
    AYM: "AYM",
    AZE: "AZE",
    BAK: "BAK",
    BAM: "BAM",
    BEL: "BEL",
    BEN: "BEN",
    BIH: "BIH",
    BIS: "BIS",
    BOD: "BOD",
    BOS: "BOS",
    BRE: "BRE",
    BUL: "BUL",
    CAT: "CAT",
    CES: "CES",
    CHA: "CHA",
    CHE: "CHE",
    CHU: "CHU",
    CHV: "CHV",
    COR: "COR",
    COS: "COS",
    CRE: "CRE",
    CYM: "CYM",
    DAN: "DAN",
    DEU: "DEU",
    DIV: "DIV",
    DZO: "DZO",
    ELL: "ELL",
    ENG: "ENG",
    ENM: "ENM",
    EPO: "EPO",
    EST: "EST",
    EUS: "EUS",
    EWE: "EWE",
    FAO: "FAO",
    FAS: "FAS",
    FIJ: "FIJ",
    FIN: "FIN",
    FRA: "FRA",
    FRM: "FRM",
    FRY: "FRY",
    FUL: "FUL",
    GER: "GER",
    GLA: "GLA",
    GLE: "GLE",
    GLG: "GLG",
    GLV: "GLV",
    GRN: "GRN",
    GUJ: "GUJ",
    HAT: "HAT",
    HAU: "HAU",
    HEB: "HEB",
    HER: "HER",
    HIN: "HIN",
    HMO: "HMO",
    HRV: "HRV",
    HUN: "HUN",
    HYE: "HYE",
    IBO: "IBO",
    IDO: "IDO",
    III: "III",
    IKU: "IKU",
    ILE: "ILE",
    INA: "INA",
    IND: "IND",
    IPK: "IPK",
    ISL: "ISL",
    ITA: "ITA",
    JAV: "JAV",
    JPN: "JPN",
    KAL: "KAL",
    KAN: "KAN",
    KAS: "KAS",
    KAT: "KAT",
    KAU: "KAU",
    KAZ: "KAZ",
    KHM: "KHM",
    KIK: "KIK",
    KIN: "KIN",
    KIR: "KIR",
    KOM: "KOM",
    KON: "KON",
    KOR: "KOR",
    KUA: "KUA",
    KUR: "KUR",
    LAO: "LAO",
    LAT: "LAT",
    LAV: "LAV",
    LIM: "LIM",
    LIN: "LIN",
    LIT: "LIT",
    LTZ: "LTZ",
    LUB: "LUB",
    LUG: "LUG",
    MAH: "MAH",
    MAL: "MAL",
    MAR: "MAR",
    MKD: "MKD",
    MLG: "MLG",
    MLT: "MLT",
    MON: "MON",
    MRI: "MRI",
    MSA: "MSA",
    MYA: "MYA",
    NAU: "NAU",
    NAV: "NAV",
    NBL: "NBL",
    NDE: "NDE",
    NDO: "NDO",
    NEP: "NEP",
    NLD: "NLD",
    NNO: "NNO",
    NOB: "NOB",
    NOR: "NOR",
    NYA: "NYA",
    OCI: "OCI",
    OJI: "OJI",
    ORI: "ORI",
    ORJ: "ORJ",
    ORM: "ORM",
    OSS: "OSS",
    PAN: "PAN",
    PLI: "PLI",
    POL: "POL",
    POR: "POR",
    PUS: "PUS",
    QAA: "QAA",
    QPC: "QPC",
    QUE: "QUE",
    ROH: "ROH",
    RON: "RON",
    RUN: "RUN",
    RUS: "RUS",
    SAG: "SAG",
    SAN: "SAN",
    SIN: "SIN",
    SLK: "SLK",
    SLV: "SLV",
    SME: "SME",
    SMO: "SMO",
    SNA: "SNA",
    SND: "SND",
    SOM: "SOM",
    SOT: "SOT",
    SPA: "SPA",
    SQI: "SQI",
    SRB: "SRB",
    SRD: "SRD",
    SRP: "SRP",
    SSW: "SSW",
    SUN: "SUN",
    SWA: "SWA",
    SWE: "SWE",
    TAH: "TAH",
    TAM: "TAM",
    TAT: "TAT",
    TEL: "TEL",
    TGK: "TGK",
    TGL: "TGL",
    THA: "THA",
    TIR: "TIR",
    TNG: "TNG",
    TON: "TON",
    TSN: "TSN",
    TSO: "TSO",
    TUK: "TUK",
    TUR: "TUR",
    TWI: "TWI",
    UIG: "UIG",
    UKR: "UKR",
    URD: "URD",
    UZB: "UZB",
    VEN: "VEN",
    VIE: "VIE",
    VOL: "VOL",
    WLN: "WLN",
    WOL: "WOL",
    XHO: "XHO",
    YID: "YID",
    YOR: "YOR",
    ZHA: "ZHA",
    ZHO: "ZHO",
    ZUL: "ZUL",
};
export const AudioLanguageCodeControl = {
    FOLLOW_INPUT: "FOLLOW_INPUT",
    USE_CONFIGURED: "USE_CONFIGURED",
};
export const RuleType = {
    ALLOWED_RENDITIONS: "ALLOWED_RENDITIONS",
    FORCE_INCLUDE_RENDITIONS: "FORCE_INCLUDE_RENDITIONS",
    MIN_BOTTOM_RENDITION_SIZE: "MIN_BOTTOM_RENDITION_SIZE",
    MIN_TOP_RENDITION_SIZE: "MIN_TOP_RENDITION_SIZE",
};
export const BurninSubtitleAlignment = {
    AUTO: "AUTO",
    CENTERED: "CENTERED",
    LEFT: "LEFT",
};
export const BurninSubtitleApplyFontColor = {
    ALL_TEXT: "ALL_TEXT",
    WHITE_TEXT_ONLY: "WHITE_TEXT_ONLY",
};
export const BurninSubtitleBackgroundColor = {
    AUTO: "AUTO",
    BLACK: "BLACK",
    NONE: "NONE",
    WHITE: "WHITE",
};
export const BurninSubtitleFallbackFont = {
    BEST_MATCH: "BEST_MATCH",
    MONOSPACED_SANSSERIF: "MONOSPACED_SANSSERIF",
    MONOSPACED_SERIF: "MONOSPACED_SERIF",
    PROPORTIONAL_SANSSERIF: "PROPORTIONAL_SANSSERIF",
    PROPORTIONAL_SERIF: "PROPORTIONAL_SERIF",
};
export const BurninSubtitleFontColor = {
    AUTO: "AUTO",
    BLACK: "BLACK",
    BLUE: "BLUE",
    GREEN: "GREEN",
    HEX: "HEX",
    RED: "RED",
    WHITE: "WHITE",
    YELLOW: "YELLOW",
};
export const FontScript = {
    AUTOMATIC: "AUTOMATIC",
    HANS: "HANS",
    HANT: "HANT",
};
export const BurninSubtitleOutlineColor = {
    AUTO: "AUTO",
    BLACK: "BLACK",
    BLUE: "BLUE",
    GREEN: "GREEN",
    RED: "RED",
    WHITE: "WHITE",
    YELLOW: "YELLOW",
};
export const BurninSubtitleShadowColor = {
    AUTO: "AUTO",
    BLACK: "BLACK",
    NONE: "NONE",
    WHITE: "WHITE",
};
export const BurnInSubtitleStylePassthrough = {
    DISABLED: "DISABLED",
    ENABLED: "ENABLED",
};
export const BurninSubtitleTeletextSpacing = {
    AUTO: "AUTO",
    FIXED_GRID: "FIXED_GRID",
    PROPORTIONAL: "PROPORTIONAL",
};
export const CaptionDestinationType = {
    BURN_IN: "BURN_IN",
    DVB_SUB: "DVB_SUB",
    EMBEDDED: "EMBEDDED",
    EMBEDDED_PLUS_SCTE20: "EMBEDDED_PLUS_SCTE20",
    IMSC: "IMSC",
    SCC: "SCC",
    SCTE20_PLUS_EMBEDDED: "SCTE20_PLUS_EMBEDDED",
    SMI: "SMI",
    SRT: "SRT",
    TELETEXT: "TELETEXT",
    TTML: "TTML",
    WEBVTT: "WEBVTT",
};
export const DvbSubtitleAlignment = {
    AUTO: "AUTO",
    CENTERED: "CENTERED",
    LEFT: "LEFT",
};
export const DvbSubtitleApplyFontColor = {
    ALL_TEXT: "ALL_TEXT",
    WHITE_TEXT_ONLY: "WHITE_TEXT_ONLY",
};
export const DvbSubtitleBackgroundColor = {
    AUTO: "AUTO",
    BLACK: "BLACK",
    NONE: "NONE",
    WHITE: "WHITE",
};
export const DvbddsHandling = {
    NONE: "NONE",
    NO_DISPLAY_WINDOW: "NO_DISPLAY_WINDOW",
    SPECIFIED: "SPECIFIED",
};
export const DvbSubSubtitleFallbackFont = {
    BEST_MATCH: "BEST_MATCH",
    MONOSPACED_SANSSERIF: "MONOSPACED_SANSSERIF",
    MONOSPACED_SERIF: "MONOSPACED_SERIF",
    PROPORTIONAL_SANSSERIF: "PROPORTIONAL_SANSSERIF",
    PROPORTIONAL_SERIF: "PROPORTIONAL_SERIF",
};
export const DvbSubtitleFontColor = {
    AUTO: "AUTO",
    BLACK: "BLACK",
    BLUE: "BLUE",
    GREEN: "GREEN",
    HEX: "HEX",
    RED: "RED",
    WHITE: "WHITE",
    YELLOW: "YELLOW",
};
export const DvbSubtitleOutlineColor = {
    AUTO: "AUTO",
    BLACK: "BLACK",
    BLUE: "BLUE",
    GREEN: "GREEN",
    RED: "RED",
    WHITE: "WHITE",
    YELLOW: "YELLOW",
};
export const DvbSubtitleShadowColor = {
    AUTO: "AUTO",
    BLACK: "BLACK",
    NONE: "NONE",
    WHITE: "WHITE",
};
export const DvbSubtitleStylePassthrough = {
    DISABLED: "DISABLED",
    ENABLED: "ENABLED",
};
export const DvbSubtitlingType = {
    HEARING_IMPAIRED: "HEARING_IMPAIRED",
    STANDARD: "STANDARD",
};
export const DvbSubtitleTeletextSpacing = {
    AUTO: "AUTO",
    FIXED_GRID: "FIXED_GRID",
    PROPORTIONAL: "PROPORTIONAL",
};
export const ImscAccessibilitySubs = {
    DISABLED: "DISABLED",
    ENABLED: "ENABLED",
};
export const ImscStylePassthrough = {
    DISABLED: "DISABLED",
    ENABLED: "ENABLED",
};
export const SccDestinationFramerate = {
    FRAMERATE_23_97: "FRAMERATE_23_97",
    FRAMERATE_24: "FRAMERATE_24",
    FRAMERATE_25: "FRAMERATE_25",
    FRAMERATE_29_97_DROPFRAME: "FRAMERATE_29_97_DROPFRAME",
    FRAMERATE_29_97_NON_DROPFRAME: "FRAMERATE_29_97_NON_DROPFRAME",
};
export const SrtStylePassthrough = {
    DISABLED: "DISABLED",
    ENABLED: "ENABLED",
};
export const TeletextPageType = {
    PAGE_TYPE_ADDL_INFO: "PAGE_TYPE_ADDL_INFO",
    PAGE_TYPE_HEARING_IMPAIRED_SUBTITLE: "PAGE_TYPE_HEARING_IMPAIRED_SUBTITLE",
    PAGE_TYPE_INITIAL: "PAGE_TYPE_INITIAL",
    PAGE_TYPE_PROGRAM_SCHEDULE: "PAGE_TYPE_PROGRAM_SCHEDULE",
    PAGE_TYPE_SUBTITLE: "PAGE_TYPE_SUBTITLE",
};
export const TtmlStylePassthrough = {
    DISABLED: "DISABLED",
    ENABLED: "ENABLED",
};
export const WebvttAccessibilitySubs = {
    DISABLED: "DISABLED",
    ENABLED: "ENABLED",
};
export const WebvttStylePassthrough = {
    DISABLED: "DISABLED",
    ENABLED: "ENABLED",
    STRICT: "STRICT",
};
export const ColorSpace = {
    FOLLOW: "FOLLOW",
    HDR10: "HDR10",
    HLG_2020: "HLG_2020",
    P3D65_HDR: "P3D65_HDR",
    P3D65_SDR: "P3D65_SDR",
    P3DCI: "P3DCI",
    REC_601: "REC_601",
    REC_709: "REC_709",
};
export const HlsAdMarkers = {
    ELEMENTAL: "ELEMENTAL",
    ELEMENTAL_SCTE35: "ELEMENTAL_SCTE35",
};
export const AdvancedInputFilter = {
    DISABLED: "DISABLED",
    ENABLED: "ENABLED",
};
export const AdvancedInputFilterAddTexture = {
    DISABLED: "DISABLED",
    ENABLED: "ENABLED",
};
export const AdvancedInputFilterSharpen = {
    HIGH: "HIGH",
    LOW: "LOW",
    OFF: "OFF",
};
export const AudioDurationCorrection = {
    AUTO: "AUTO",
    DISABLED: "DISABLED",
    FRAME: "FRAME",
    TRACK: "TRACK",
};
export const AudioDefaultSelection = {
    DEFAULT: "DEFAULT",
    NOT_DEFAULT: "NOT_DEFAULT",
};
export const AudioSelectorType = {
    HLS_RENDITION_GROUP: "HLS_RENDITION_GROUP",
    LANGUAGE_CODE: "LANGUAGE_CODE",
    PID: "PID",
    TRACK: "TRACK",
};
export const AncillaryConvert608To708 = {
    DISABLED: "DISABLED",
    UPCONVERT: "UPCONVERT",
};
export const AncillaryTerminateCaptions = {
    DISABLED: "DISABLED",
    END_OF_INPUT: "END_OF_INPUT",
};
export const EmbeddedConvert608To708 = {
    DISABLED: "DISABLED",
    UPCONVERT: "UPCONVERT",
};
export const EmbeddedTerminateCaptions = {
    DISABLED: "DISABLED",
    END_OF_INPUT: "END_OF_INPUT",
};
export const FileSourceConvert608To708 = {
    DISABLED: "DISABLED",
    UPCONVERT: "UPCONVERT",
};
export const CaptionSourceConvertPaintOnToPopOn = {
    DISABLED: "DISABLED",
    ENABLED: "ENABLED",
};
export const FileSourceTimeDeltaUnits = {
    MILLISECONDS: "MILLISECONDS",
    SECONDS: "SECONDS",
};
export const CaptionSourceType = {
    ANCILLARY: "ANCILLARY",
    DVB_SUB: "DVB_SUB",
    EMBEDDED: "EMBEDDED",
    IMSC: "IMSC",
    NULL_SOURCE: "NULL_SOURCE",
    SCC: "SCC",
    SCTE20: "SCTE20",
    SMI: "SMI",
    SMPTE_TT: "SMPTE_TT",
    SRT: "SRT",
    STL: "STL",
    TELETEXT: "TELETEXT",
    TTML: "TTML",
    WEBVTT: "WEBVTT",
};
export const InputDeblockFilter = {
    DISABLED: "DISABLED",
    ENABLED: "ENABLED",
};
export const DecryptionMode = {
    AES_CBC: "AES_CBC",
    AES_CTR: "AES_CTR",
    AES_GCM: "AES_GCM",
};
export const InputDenoiseFilter = {
    DISABLED: "DISABLED",
    ENABLED: "ENABLED",
};
export const InputFilterEnable = {
    AUTO: "AUTO",
    DISABLE: "DISABLE",
    FORCE: "FORCE",
};
export const InputScanType = {
    AUTO: "AUTO",
    PSF: "PSF",
};
export const InputPsiControl = {
    IGNORE_PSI: "IGNORE_PSI",
    USE_PSI: "USE_PSI",
};
export const InputTimecodeSource = {
    EMBEDDED: "EMBEDDED",
    SPECIFIEDSTART: "SPECIFIEDSTART",
    ZEROBASED: "ZEROBASED",
};
export const AlphaBehavior = {
    DISCARD: "DISCARD",
    REMAP_TO_LUMA: "REMAP_TO_LUMA",
};
export const ColorSpaceUsage = {
    FALLBACK: "FALLBACK",
    FORCE: "FORCE",
};
export const EmbeddedTimecodeOverride = {
    NONE: "NONE",
    USE_MDPM: "USE_MDPM",
};
export const PadVideo = {
    BLACK: "BLACK",
    DISABLED: "DISABLED",
};
export const InputRotate = {
    AUTO: "AUTO",
    DEGREES_180: "DEGREES_180",
    DEGREES_270: "DEGREES_270",
    DEGREES_90: "DEGREES_90",
    DEGREE_0: "DEGREE_0",
};
export const InputSampleRange = {
    FOLLOW: "FOLLOW",
    FULL_RANGE: "FULL_RANGE",
    LIMITED_RANGE: "LIMITED_RANGE",
};
export const AccelerationMode = {
    DISABLED: "DISABLED",
    ENABLED: "ENABLED",
    PREFERRED: "PREFERRED",
};
export const AccelerationStatus = {
    ACCELERATED: "ACCELERATED",
    IN_PROGRESS: "IN_PROGRESS",
    NOT_ACCELERATED: "NOT_ACCELERATED",
    NOT_APPLICABLE: "NOT_APPLICABLE",
};
export const BillingTagsSource = {
    JOB: "JOB",
    JOB_TEMPLATE: "JOB_TEMPLATE",
    PRESET: "PRESET",
    QUEUE: "QUEUE",
};
export const JobPhase = {
    PROBING: "PROBING",
    TRANSCODING: "TRANSCODING",
    UPLOADING: "UPLOADING",
};
export const CopyProtectionAction = {
    PASSTHROUGH: "PASSTHROUGH",
    STRIP: "STRIP",
};
export const VchipAction = {
    PASSTHROUGH: "PASSTHROUGH",
    STRIP: "STRIP",
};
export const MotionImageInsertionMode = {
    MOV: "MOV",
    PNG: "PNG",
};
export const MotionImagePlayback = {
    ONCE: "ONCE",
    REPEAT: "REPEAT",
};
export const NielsenActiveWatermarkProcessType = {
    CBET: "CBET",
    NAES2_AND_NW: "NAES2_AND_NW",
    NAES2_AND_NW_AND_CBET: "NAES2_AND_NW_AND_CBET",
};
export const NielsenSourceWatermarkStatusType = {
    CLEAN: "CLEAN",
    WATERMARKED: "WATERMARKED",
};
export const NielsenUniqueTicPerAudioTrackType = {
    RESERVE_UNIQUE_TICS_PER_TRACK: "RESERVE_UNIQUE_TICS_PER_TRACK",
    SAME_TICS_PER_TRACK: "SAME_TICS_PER_TRACK",
};
export const CmafClientCache = {
    DISABLED: "DISABLED",
    ENABLED: "ENABLED",
};
export const CmafCodecSpecification = {
    RFC_4281: "RFC_4281",
    RFC_6381: "RFC_6381",
};
export const DashManifestStyle = {
    BASIC: "BASIC",
    COMPACT: "COMPACT",
    DISTINCT: "DISTINCT",
};
export const S3ObjectCannedAcl = {
    AUTHENTICATED_READ: "AUTHENTICATED_READ",
    BUCKET_OWNER_FULL_CONTROL: "BUCKET_OWNER_FULL_CONTROL",
    BUCKET_OWNER_READ: "BUCKET_OWNER_READ",
    PUBLIC_READ: "PUBLIC_READ",
};
export const S3ServerSideEncryptionType = {
    SERVER_SIDE_ENCRYPTION_KMS: "SERVER_SIDE_ENCRYPTION_KMS",
    SERVER_SIDE_ENCRYPTION_S3: "SERVER_SIDE_ENCRYPTION_S3",
};
export const S3StorageClass = {
    DEEP_ARCHIVE: "DEEP_ARCHIVE",
    GLACIER: "GLACIER",
    INTELLIGENT_TIERING: "INTELLIGENT_TIERING",
    ONEZONE_IA: "ONEZONE_IA",
    REDUCED_REDUNDANCY: "REDUCED_REDUNDANCY",
    STANDARD: "STANDARD",
    STANDARD_IA: "STANDARD_IA",
};
export const CmafEncryptionType = {
    AES_CTR: "AES_CTR",
    SAMPLE_AES: "SAMPLE_AES",
};
export const CmafInitializationVectorInManifest = {
    EXCLUDE: "EXCLUDE",
    INCLUDE: "INCLUDE",
};
export const CmafKeyProviderType = {
    SPEKE: "SPEKE",
    STATIC_KEY: "STATIC_KEY",
};
export const CmafImageBasedTrickPlay = {
    ADVANCED: "ADVANCED",
    NONE: "NONE",
    THUMBNAIL: "THUMBNAIL",
    THUMBNAIL_AND_FULLFRAME: "THUMBNAIL_AND_FULLFRAME",
};
export const CmafIntervalCadence = {
    FOLLOW_CUSTOM: "FOLLOW_CUSTOM",
    FOLLOW_IFRAME: "FOLLOW_IFRAME",
};
export const CmafManifestCompression = {
    GZIP: "GZIP",
    NONE: "NONE",
};
export const CmafManifestDurationFormat = {
    FLOATING_POINT: "FLOATING_POINT",
    INTEGER: "INTEGER",
};
export const CmafMpdManifestBandwidthType = {
    AVERAGE: "AVERAGE",
    MAX: "MAX",
};
export const CmafMpdProfile = {
    MAIN_PROFILE: "MAIN_PROFILE",
    ON_DEMAND_PROFILE: "ON_DEMAND_PROFILE",
};
export const CmafPtsOffsetHandlingForBFrames = {
    MATCH_INITIAL_PTS: "MATCH_INITIAL_PTS",
    ZERO_BASED: "ZERO_BASED",
};
export const CmafSegmentControl = {
    SEGMENTED_FILES: "SEGMENTED_FILES",
    SINGLE_FILE: "SINGLE_FILE",
};
export const CmafSegmentLengthControl = {
    EXACT: "EXACT",
    GOP_MULTIPLE: "GOP_MULTIPLE",
};
export const CmafStreamInfResolution = {
    EXCLUDE: "EXCLUDE",
    INCLUDE: "INCLUDE",
};
export const CmafTargetDurationCompatibilityMode = {
    LEGACY: "LEGACY",
    SPEC_COMPLIANT: "SPEC_COMPLIANT",
};
export const CmafVideoCompositionOffsets = {
    SIGNED: "SIGNED",
    UNSIGNED: "UNSIGNED",
};
export const CmafWriteDASHManifest = {
    DISABLED: "DISABLED",
    ENABLED: "ENABLED",
};
export const CmafWriteHLSManifest = {
    DISABLED: "DISABLED",
    ENABLED: "ENABLED",
};
export const CmafWriteSegmentTimelineInRepresentation = {
    DISABLED: "DISABLED",
    ENABLED: "ENABLED",
};
export const DashIsoGroupAudioChannelConfigSchemeIdUri = {
    DOLBY_CHANNEL_CONFIGURATION: "DOLBY_CHANNEL_CONFIGURATION",
    MPEG_CHANNEL_CONFIGURATION: "MPEG_CHANNEL_CONFIGURATION",
};
export const DashIsoPlaybackDeviceCompatibility = {
    CENC_V1: "CENC_V1",
    UNENCRYPTED_SEI: "UNENCRYPTED_SEI",
};
export const DashIsoHbbtvCompliance = {
    HBBTV_1_5: "HBBTV_1_5",
    NONE: "NONE",
};
export const DashIsoImageBasedTrickPlay = {
    ADVANCED: "ADVANCED",
    NONE: "NONE",
    THUMBNAIL: "THUMBNAIL",
    THUMBNAIL_AND_FULLFRAME: "THUMBNAIL_AND_FULLFRAME",
};
export const DashIsoIntervalCadence = {
    FOLLOW_CUSTOM: "FOLLOW_CUSTOM",
    FOLLOW_IFRAME: "FOLLOW_IFRAME",
};
export const DashIsoMpdManifestBandwidthType = {
    AVERAGE: "AVERAGE",
    MAX: "MAX",
};
export const DashIsoMpdProfile = {
    MAIN_PROFILE: "MAIN_PROFILE",
    ON_DEMAND_PROFILE: "ON_DEMAND_PROFILE",
};
export const DashIsoPtsOffsetHandlingForBFrames = {
    MATCH_INITIAL_PTS: "MATCH_INITIAL_PTS",
    ZERO_BASED: "ZERO_BASED",
};
export const DashIsoSegmentControl = {
    SEGMENTED_FILES: "SEGMENTED_FILES",
    SINGLE_FILE: "SINGLE_FILE",
};
export const DashIsoSegmentLengthControl = {
    EXACT: "EXACT",
    GOP_MULTIPLE: "GOP_MULTIPLE",
};
export const DashIsoVideoCompositionOffsets = {
    SIGNED: "SIGNED",
    UNSIGNED: "UNSIGNED",
};
export const DashIsoWriteSegmentTimelineInRepresentation = {
    DISABLED: "DISABLED",
    ENABLED: "ENABLED",
};
export const HlsAudioOnlyHeader = {
    EXCLUDE: "EXCLUDE",
    INCLUDE: "INCLUDE",
};
export const HlsCaptionLanguageSetting = {
    INSERT: "INSERT",
    NONE: "NONE",
    OMIT: "OMIT",
};
export const HlsCaptionSegmentLengthControl = {
    LARGE_SEGMENTS: "LARGE_SEGMENTS",
    MATCH_VIDEO: "MATCH_VIDEO",
};
export const HlsClientCache = {
    DISABLED: "DISABLED",
    ENABLED: "ENABLED",
};
export const HlsCodecSpecification = {
    RFC_4281: "RFC_4281",
    RFC_6381: "RFC_6381",
};
export const HlsDirectoryStructure = {
    SINGLE_DIRECTORY: "SINGLE_DIRECTORY",
    SUBDIRECTORY_PER_STREAM: "SUBDIRECTORY_PER_STREAM",
};
export const HlsEncryptionType = {
    AES128: "AES128",
    SAMPLE_AES: "SAMPLE_AES",
};
export const HlsInitializationVectorInManifest = {
    EXCLUDE: "EXCLUDE",
    INCLUDE: "INCLUDE",
};
export const HlsOfflineEncrypted = {
    DISABLED: "DISABLED",
    ENABLED: "ENABLED",
};
export const HlsKeyProviderType = {
    SPEKE: "SPEKE",
    STATIC_KEY: "STATIC_KEY",
};
export const HlsImageBasedTrickPlay = {
    ADVANCED: "ADVANCED",
    NONE: "NONE",
    THUMBNAIL: "THUMBNAIL",
    THUMBNAIL_AND_FULLFRAME: "THUMBNAIL_AND_FULLFRAME",
};
export const HlsIntervalCadence = {
    FOLLOW_CUSTOM: "FOLLOW_CUSTOM",
    FOLLOW_IFRAME: "FOLLOW_IFRAME",
};
export const HlsManifestCompression = {
    GZIP: "GZIP",
    NONE: "NONE",
};
export const HlsManifestDurationFormat = {
    FLOATING_POINT: "FLOATING_POINT",
    INTEGER: "INTEGER",
};
export const HlsOutputSelection = {
    MANIFESTS_AND_SEGMENTS: "MANIFESTS_AND_SEGMENTS",
    SEGMENTS_ONLY: "SEGMENTS_ONLY",
};
export const HlsProgramDateTime = {
    EXCLUDE: "EXCLUDE",
    INCLUDE: "INCLUDE",
};
export const HlsProgressiveWriteHlsManifest = {
    DISABLED: "DISABLED",
    ENABLED: "ENABLED",
};
export const HlsSegmentControl = {
    SEGMENTED_FILES: "SEGMENTED_FILES",
    SINGLE_FILE: "SINGLE_FILE",
};
export const HlsSegmentLengthControl = {
    EXACT: "EXACT",
    GOP_MULTIPLE: "GOP_MULTIPLE",
};
export const HlsStreamInfResolution = {
    EXCLUDE: "EXCLUDE",
    INCLUDE: "INCLUDE",
};
export const HlsTargetDurationCompatibilityMode = {
    LEGACY: "LEGACY",
    SPEC_COMPLIANT: "SPEC_COMPLIANT",
};
export const HlsTimedMetadataId3Frame = {
    NONE: "NONE",
    PRIV: "PRIV",
    TDRL: "TDRL",
};
export const MsSmoothAudioDeduplication = {
    COMBINE_DUPLICATE_STREAMS: "COMBINE_DUPLICATE_STREAMS",
    NONE: "NONE",
};
export const MsSmoothFragmentLengthControl = {
    EXACT: "EXACT",
    GOP_MULTIPLE: "GOP_MULTIPLE",
};
export const MsSmoothManifestEncoding = {
    UTF16: "UTF16",
    UTF8: "UTF8",
};
export const OutputGroupType = {
    CMAF_GROUP_SETTINGS: "CMAF_GROUP_SETTINGS",
    DASH_ISO_GROUP_SETTINGS: "DASH_ISO_GROUP_SETTINGS",
    FILE_GROUP_SETTINGS: "FILE_GROUP_SETTINGS",
    HLS_GROUP_SETTINGS: "HLS_GROUP_SETTINGS",
    MS_SMOOTH_GROUP_SETTINGS: "MS_SMOOTH_GROUP_SETTINGS",
};
export const CmfcAudioDuration = {
    DEFAULT_CODEC_DURATION: "DEFAULT_CODEC_DURATION",
    MATCH_VIDEO_DURATION: "MATCH_VIDEO_DURATION",
};
export const CmfcAudioTrackType = {
    ALTERNATE_AUDIO_AUTO_SELECT: "ALTERNATE_AUDIO_AUTO_SELECT",
    ALTERNATE_AUDIO_AUTO_SELECT_DEFAULT: "ALTERNATE_AUDIO_AUTO_SELECT_DEFAULT",
    ALTERNATE_AUDIO_NOT_AUTO_SELECT: "ALTERNATE_AUDIO_NOT_AUTO_SELECT",
    AUDIO_ONLY_VARIANT_STREAM: "AUDIO_ONLY_VARIANT_STREAM",
};
export const CmfcDescriptiveVideoServiceFlag = {
    DONT_FLAG: "DONT_FLAG",
    FLAG: "FLAG",
};
export const CmfcIFrameOnlyManifest = {
    EXCLUDE: "EXCLUDE",
    INCLUDE: "INCLUDE",
};
export const CmfcKlvMetadata = {
    NONE: "NONE",
    PASSTHROUGH: "PASSTHROUGH",
};
export const CmfcManifestMetadataSignaling = {
    DISABLED: "DISABLED",
    ENABLED: "ENABLED",
};
export const CmfcScte35Esam = {
    INSERT: "INSERT",
    NONE: "NONE",
};
