// @azure/arm-hardwaresecuritymodules/dist-esm/src/models/index.js
/*
 * Copyright (c) Microsoft Corporation.
 * Licensed under the MIT License.
 *
 * Code generated by Microsoft (R) AutoRest Code Generator.
 * Changes may cause incorrect behavior and will be lost if the code is regenerated.
 */
/** Known values of {@link IdentityType} that the service accepts. */
export var KnownIdentityType;
(function (KnownIdentityType) {
    KnownIdentityType["User"] = "User";
    KnownIdentityType["Application"] = "Application";
    KnownIdentityType["ManagedIdentity"] = "ManagedIdentity";
    KnownIdentityType["Key"] = "Key";
})(KnownIdentityType || (KnownIdentityType = {}));
/** Known values of {@link JsonWebKeyType} that the service accepts. */
export var KnownJsonWebKeyType;
(function (KnownJsonWebKeyType) {
    /** The dedicated HSM has been full provisioned. */
    KnownJsonWebKeyType["Succeeded"] = "Succeeded";
    /** The dedicated HSM is currently being provisioned. */
    KnownJsonWebKeyType["Provisioning"] = "Provisioning";
    /** A device is currently being allocated for the dedicated HSM resource. */
    KnownJsonWebKeyType["Allocating"] = "Allocating";
    /** The dedicated HSM is being connected to the virtual network. */
    KnownJsonWebKeyType["Connecting"] = "Connecting";
    /** Provisioning of the dedicated HSM has failed. */
    KnownJsonWebKeyType["Failed"] = "Failed";
    /** Validating the subscription has sufficient quota to allocate a dedicated HSM device. */
    KnownJsonWebKeyType["CheckingQuota"] = "CheckingQuota";
    /** The dedicated HSM is currently being deleted. */
    KnownJsonWebKeyType["Deleting"] = "Deleting";
})(KnownJsonWebKeyType || (KnownJsonWebKeyType = {}));
/** Known values of {@link SkuName} that the service accepts. */
export var KnownSkuName;
(function (KnownSkuName) {
    /** The dedicated HSM is a Safenet Luna Network HSM A790 device. */
    KnownSkuName["SafeNetLunaNetworkHSMA790"] = "SafeNet Luna Network HSM A790";
    /** The dedicated HSM is a payShield 10K, model PS10-D, 10Gb Ethernet Hardware Platform device with 1 local master key which supports up to 60 calls per second. */
    KnownSkuName["PayShield10KLMK1CPS60"] = "payShield10K_LMK1_CPS60";
    /** The dedicated HSM is a payShield 10K, model PS10-D, 10Gb Ethernet Hardware Platform device with 1 local master key which supports up to 250 calls per second. */
    KnownSkuName["PayShield10KLMK1CPS250"] = "payShield10K_LMK1_CPS250";
    /** The dedicated HSM is a payShield 10K, model PS10-D, 10Gb Ethernet Hardware Platform device with 1 local master key which supports up to 2500 calls per second. */
    KnownSkuName["PayShield10KLMK1CPS2500"] = "payShield10K_LMK1_CPS2500";
    /** The dedicated HSM is a payShield 10K, model PS10-D, 10Gb Ethernet Hardware Platform device with 2 local master keys which supports up to 60 calls per second. */
    KnownSkuName["PayShield10KLMK2CPS60"] = "payShield10K_LMK2_CPS60";
    /** The dedicated HSM is a payShield 10K, model PS10-D, 10Gb Ethernet Hardware Platform device with 2 local master keys which supports up to 250 calls per second. */
    KnownSkuName["PayShield10KLMK2CPS250"] = "payShield10K_LMK2_CPS250";
    /** The dedicated HSM is a payShield 10K, model PS10-D, 10Gb Ethernet Hardware Platform device with 2 local master keys which supports up to 2500 calls per second. */
    KnownSkuName["PayShield10KLMK2CPS2500"] = "payShield10K_LMK2_CPS2500";
})(KnownSkuName || (KnownSkuName = {}));
