// @azure/arm-authorization/dist-esm/src/authorizationManagementClient.js
/*
 * Copyright (c) Microsoft Corporation.
 * Licensed under the MIT License.
 *
 * Code generated by Microsoft (R) AutoRest Code Generator.
 * Changes may cause incorrect behavior and will be lost if the code is regenerated.
 */
import * as coreClient from "@azure/core-client";
import * as coreRestPipeline from "@azure/core-rest-pipeline";
import { ClassicAdministratorsImpl, GlobalAdministratorImpl, DenyAssignmentsImpl, ProviderOperationsMetadataOperationsImpl, RoleAssignmentsImpl, PermissionsImpl, RoleDefinitionsImpl, EligibleChildResourcesImpl, RoleAssignmentSchedulesImpl, RoleAssignmentScheduleInstancesImpl, RoleAssignmentScheduleRequestsImpl, RoleEligibilitySchedulesImpl, RoleEligibilityScheduleInstancesImpl, RoleEligibilityScheduleRequestsImpl, RoleManagementPoliciesImpl, RoleManagementPolicyAssignmentsImpl } from "./operations";
export class AuthorizationManagementClient extends coreClient.ServiceClient {
    /**
     * Initializes a new instance of the AuthorizationManagementClient class.
     * @param credentials Subscription credentials which uniquely identify client subscription.
     * @param subscriptionId The ID of the target subscription.
     * @param options The parameter options
     */
    constructor(credentials, subscriptionId, options) {
        var _a, _b, _c;
        if (credentials === undefined) {
            throw new Error("'credentials' cannot be null");
        }
        if (subscriptionId === undefined) {
            throw new Error("'subscriptionId' cannot be null");
        }
        // Initializing default values for options
        if (!options) {
            options = {};
        }
        const defaults = {
            requestContentType: "application/json; charset=utf-8",
            credential: credentials
        };
        const packageDetails = `azsdk-js-arm-authorization/9.0.0`;
        const userAgentPrefix = options.userAgentOptions && options.userAgentOptions.userAgentPrefix
            ? `${options.userAgentOptions.userAgentPrefix} ${packageDetails}`
            : `${packageDetails}`;
        const optionsWithDefaults = Object.assign(Object.assign(Object.assign({}, defaults), options), { userAgentOptions: {
                userAgentPrefix
            }, endpoint: (_b = (_a = options.endpoint) !== null && _a !== void 0 ? _a : options.baseUri) !== null && _b !== void 0 ? _b : "https://management.azure.com" });
        super(optionsWithDefaults);
        let bearerTokenAuthenticationPolicyFound = false;
        if ((options === null || options === void 0 ? void 0 : options.pipeline) && options.pipeline.getOrderedPolicies().length > 0) {
            const pipelinePolicies = options.pipeline.getOrderedPolicies();
            bearerTokenAuthenticationPolicyFound = pipelinePolicies.some((pipelinePolicy) => pipelinePolicy.name ===
                coreRestPipeline.bearerTokenAuthenticationPolicyName);
        }
        if (!options ||
            !options.pipeline ||
            options.pipeline.getOrderedPolicies().length == 0 ||
            !bearerTokenAuthenticationPolicyFound) {
            this.pipeline.removePolicy({
                name: coreRestPipeline.bearerTokenAuthenticationPolicyName
            });
            this.pipeline.addPolicy(coreRestPipeline.bearerTokenAuthenticationPolicy({
                credential: credentials,
                scopes: (_c = optionsWithDefaults.credentialScopes) !== null && _c !== void 0 ? _c : `${optionsWithDefaults.endpoint}/.default`,
                challengeCallbacks: {
                    authorizeRequestOnChallenge: coreClient.authorizeRequestOnClaimChallenge
                }
            }));
        }
        // Parameter assignments
        this.subscriptionId = subscriptionId;
        // Assigning values to Constant parameters
        this.$host = options.$host || "https://management.azure.com";
        this.classicAdministrators = new ClassicAdministratorsImpl(this);
        this.globalAdministrator = new GlobalAdministratorImpl(this);
        this.denyAssignments = new DenyAssignmentsImpl(this);
        this.providerOperationsMetadataOperations = new ProviderOperationsMetadataOperationsImpl(this);
        this.roleAssignments = new RoleAssignmentsImpl(this);
        this.permissions = new PermissionsImpl(this);
        this.roleDefinitions = new RoleDefinitionsImpl(this);
        this.eligibleChildResources = new EligibleChildResourcesImpl(this);
        this.roleAssignmentSchedules = new RoleAssignmentSchedulesImpl(this);
        this.roleAssignmentScheduleInstances = new RoleAssignmentScheduleInstancesImpl(this);
        this.roleAssignmentScheduleRequests = new RoleAssignmentScheduleRequestsImpl(this);
        this.roleEligibilitySchedules = new RoleEligibilitySchedulesImpl(this);
        this.roleEligibilityScheduleInstances = new RoleEligibilityScheduleInstancesImpl(this);
        this.roleEligibilityScheduleRequests = new RoleEligibilityScheduleRequestsImpl(this);
        this.roleManagementPolicies = new RoleManagementPoliciesImpl(this);
        this.roleManagementPolicyAssignments = new RoleManagementPolicyAssignmentsImpl(this);
    }
}
