// @azure/arm-loadtestservice/dist-esm/src/models/index.js
/*
 * Copyright (c) Microsoft Corporation.
 * Licensed under the MIT License.
 *
 * Code generated by Microsoft (R) AutoRest Code Generator.
 * Changes may cause incorrect behavior and will be lost if the code is regenerated.
 */
/** Known values of {@link Origin} that the service accepts. */
export var KnownOrigin;
(function (KnownOrigin) {
    KnownOrigin["User"] = "user";
    KnownOrigin["System"] = "system";
    KnownOrigin["UserSystem"] = "user,system";
})(KnownOrigin || (KnownOrigin = {}));
/** Known values of {@link ActionType} that the service accepts. */
export var KnownActionType;
(function (KnownActionType) {
    KnownActionType["Internal"] = "Internal";
})(KnownActionType || (KnownActionType = {}));
/** Known values of {@link ResourceState} that the service accepts. */
export var KnownResourceState;
(function (KnownResourceState) {
    KnownResourceState["Succeeded"] = "Succeeded";
    KnownResourceState["Failed"] = "Failed";
    KnownResourceState["Canceled"] = "Canceled";
    KnownResourceState["Deleted"] = "Deleted";
})(KnownResourceState || (KnownResourceState = {}));
/** Known values of {@link SystemAssignedServiceIdentityType} that the service accepts. */
export var KnownSystemAssignedServiceIdentityType;
(function (KnownSystemAssignedServiceIdentityType) {
    KnownSystemAssignedServiceIdentityType["None"] = "None";
    KnownSystemAssignedServiceIdentityType["SystemAssigned"] = "SystemAssigned";
})(KnownSystemAssignedServiceIdentityType || (KnownSystemAssignedServiceIdentityType = {}));
/** Known values of {@link CreatedByType} that the service accepts. */
export var KnownCreatedByType;
(function (KnownCreatedByType) {
    KnownCreatedByType["User"] = "User";
    KnownCreatedByType["Application"] = "Application";
    KnownCreatedByType["ManagedIdentity"] = "ManagedIdentity";
    KnownCreatedByType["Key"] = "Key";
})(KnownCreatedByType || (KnownCreatedByType = {}));
