// @azure/arm-servicebus/dist-esm/src/operations/rules.js
/*
 * Copyright (c) Microsoft Corporation.
 * Licensed under the MIT License.
 *
 * Code generated by Microsoft (R) AutoRest Code Generator.
 * Changes may cause incorrect behavior and will be lost if the code is regenerated.
 */
import { __asyncDelegator, __asyncGenerator, __asyncValues, __await } from "tslib";
import { setContinuationToken } from "../pagingHelper";
import * as coreClient from "@azure/core-client";
import * as Mappers from "../models/mappers";
import * as Parameters from "../models/parameters";
/// <reference lib="esnext.asynciterable" />
/** Class containing Rules operations. */
export class RulesImpl {
    /**
     * Initialize a new instance of the class Rules class.
     * @param client Reference to the service client
     */
    constructor(client) {
        this.client = client;
    }
    /**
     * List all the rules within given topic-subscription
     * @param resourceGroupName Name of the Resource group within the Azure subscription.
     * @param namespaceName The namespace name
     * @param topicName The topic name.
     * @param subscriptionName The subscription name.
     * @param options The options parameters.
     */
    listBySubscriptions(resourceGroupName, namespaceName, topicName, subscriptionName, options) {
        const iter = this.listBySubscriptionsPagingAll(resourceGroupName, namespaceName, topicName, subscriptionName, options);
        return {
            next() {
                return iter.next();
            },
            [Symbol.asyncIterator]() {
                return this;
            },
            byPage: (settings) => {
                if (settings === null || settings === void 0 ? void 0 : settings.maxPageSize) {
                    throw new Error("maxPageSize is not supported by this operation.");
                }
                return this.listBySubscriptionsPagingPage(resourceGroupName, namespaceName, topicName, subscriptionName, options, settings);
            }
        };
    }
    listBySubscriptionsPagingPage(resourceGroupName, namespaceName, topicName, subscriptionName, options, settings) {
        return __asyncGenerator(this, arguments, function* listBySubscriptionsPagingPage_1() {
            let result;
            let continuationToken = settings === null || settings === void 0 ? void 0 : settings.continuationToken;
            if (!continuationToken) {
                result = yield __await(this._listBySubscriptions(resourceGroupName, namespaceName, topicName, subscriptionName, options));
                let page = result.value || [];
                continuationToken = result.nextLink;
                setContinuationToken(page, continuationToken);
                yield yield __await(page);
            }
            while (continuationToken) {
                result = yield __await(this._listBySubscriptionsNext(resourceGroupName, namespaceName, topicName, subscriptionName, continuationToken, options));
                continuationToken = result.nextLink;
                let page = result.value || [];
                setContinuationToken(page, continuationToken);
                yield yield __await(page);
            }
        });
    }
    listBySubscriptionsPagingAll(resourceGroupName, namespaceName, topicName, subscriptionName, options) {
        return __asyncGenerator(this, arguments, function* listBySubscriptionsPagingAll_1() {
            var e_1, _a;
            try {
                for (var _b = __asyncValues(this.listBySubscriptionsPagingPage(resourceGroupName, namespaceName, topicName, subscriptionName, options)), _c; _c = yield __await(_b.next()), !_c.done;) {
                    const page = _c.value;
                    yield __await(yield* __asyncDelegator(__asyncValues(page)));
                }
            }
            catch (e_1_1) { e_1 = { error: e_1_1 }; }
            finally {
                try {
                    if (_c && !_c.done && (_a = _b.return)) yield __await(_a.call(_b));
                }
                finally { if (e_1) throw e_1.error; }
            }
        });
    }
    /**
     * List all the rules within given topic-subscription
     * @param resourceGroupName Name of the Resource group within the Azure subscription.
     * @param namespaceName The namespace name
     * @param topicName The topic name.
     * @param subscriptionName The subscription name.
     * @param options The options parameters.
     */
    _listBySubscriptions(resourceGroupName, namespaceName, topicName, subscriptionName, options) {
        return this.client.sendOperationRequest({
            resourceGroupName,
            namespaceName,
            topicName,
            subscriptionName,
            options
        }, listBySubscriptionsOperationSpec);
    }
    /**
     * Creates a new rule and updates an existing rule
     * @param resourceGroupName Name of the Resource group within the Azure subscription.
     * @param namespaceName The namespace name
     * @param topicName The topic name.
     * @param subscriptionName The subscription name.
     * @param ruleName The rule name.
     * @param parameters Parameters supplied to create a rule.
     * @param options The options parameters.
     */
    createOrUpdate(resourceGroupName, namespaceName, topicName, subscriptionName, ruleName, parameters, options) {
        return this.client.sendOperationRequest({
            resourceGroupName,
            namespaceName,
            topicName,
            subscriptionName,
            ruleName,
            parameters,
            options
        }, createOrUpdateOperationSpec);
    }
    /**
     * Deletes an existing rule.
     * @param resourceGroupName Name of the Resource group within the Azure subscription.
     * @param namespaceName The namespace name
     * @param topicName The topic name.
     * @param subscriptionName The subscription name.
     * @param ruleName The rule name.
     * @param options The options parameters.
     */
    delete(resourceGroupName, namespaceName, topicName, subscriptionName, ruleName, options) {
        return this.client.sendOperationRequest({
            resourceGroupName,
            namespaceName,
            topicName,
            subscriptionName,
            ruleName,
            options
        }, deleteOperationSpec);
    }
    /**
     * Retrieves the description for the specified rule.
     * @param resourceGroupName Name of the Resource group within the Azure subscription.
     * @param namespaceName The namespace name
     * @param topicName The topic name.
     * @param subscriptionName The subscription name.
     * @param ruleName The rule name.
     * @param options The options parameters.
     */
    get(resourceGroupName, namespaceName, topicName, subscriptionName, ruleName, options) {
        return this.client.sendOperationRequest({
            resourceGroupName,
            namespaceName,
            topicName,
            subscriptionName,
            ruleName,
            options
        }, getOperationSpec);
    }
    /**
     * ListBySubscriptionsNext
     * @param resourceGroupName Name of the Resource group within the Azure subscription.
     * @param namespaceName The namespace name
     * @param topicName The topic name.
     * @param subscriptionName The subscription name.
     * @param nextLink The nextLink from the previous successful call to the ListBySubscriptions method.
     * @param options The options parameters.
     */
    _listBySubscriptionsNext(resourceGroupName, namespaceName, topicName, subscriptionName, nextLink, options) {
        return this.client.sendOperationRequest({
            resourceGroupName,
            namespaceName,
            topicName,
            subscriptionName,
            nextLink,
            options
        }, listBySubscriptionsNextOperationSpec);
    }
}
// Operation Specifications
const serializer = coreClient.createSerializer(Mappers, /* isXml */ false);
const listBySubscriptionsOperationSpec = {
    path: "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.ServiceBus/namespaces/{namespaceName}/topics/{topicName}/subscriptions/{subscriptionName}/rules",
    httpMethod: "GET",
    responses: {
        200: {
            bodyMapper: Mappers.RuleListResult
        },
        default: {
            bodyMapper: Mappers.ErrorResponse
        }
    },
    queryParameters: [Parameters.apiVersion, Parameters.skip, Parameters.top],
    urlParameters: [
        Parameters.$host,
        Parameters.subscriptionId,
        Parameters.resourceGroupName,
        Parameters.namespaceName1,
        Parameters.topicName,
        Parameters.subscriptionName
    ],
    headerParameters: [Parameters.accept],
    serializer
};
const createOrUpdateOperationSpec = {
    path: "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.ServiceBus/namespaces/{namespaceName}/topics/{topicName}/subscriptions/{subscriptionName}/rules/{ruleName}",
    httpMethod: "PUT",
    responses: {
        200: {
            bodyMapper: Mappers.Rule
        },
        default: {
            bodyMapper: Mappers.ErrorResponse
        }
    },
    requestBody: Parameters.parameters12,
    queryParameters: [Parameters.apiVersion],
    urlParameters: [
        Parameters.$host,
        Parameters.subscriptionId,
        Parameters.resourceGroupName,
        Parameters.namespaceName1,
        Parameters.topicName,
        Parameters.subscriptionName,
        Parameters.ruleName
    ],
    headerParameters: [Parameters.accept, Parameters.contentType],
    mediaType: "json",
    serializer
};
const deleteOperationSpec = {
    path: "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.ServiceBus/namespaces/{namespaceName}/topics/{topicName}/subscriptions/{subscriptionName}/rules/{ruleName}",
    httpMethod: "DELETE",
    responses: {
        200: {},
        204: {},
        default: {
            bodyMapper: Mappers.ErrorResponse
        }
    },
    queryParameters: [Parameters.apiVersion],
    urlParameters: [
        Parameters.$host,
        Parameters.subscriptionId,
        Parameters.resourceGroupName,
        Parameters.namespaceName1,
        Parameters.topicName,
        Parameters.subscriptionName,
        Parameters.ruleName
    ],
    headerParameters: [Parameters.accept],
    serializer
};
const getOperationSpec = {
    path: "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.ServiceBus/namespaces/{namespaceName}/topics/{topicName}/subscriptions/{subscriptionName}/rules/{ruleName}",
    httpMethod: "GET",
    responses: {
        200: {
            bodyMapper: Mappers.Rule
        },
        default: {
            bodyMapper: Mappers.ErrorResponse
        }
    },
    queryParameters: [Parameters.apiVersion],
    urlParameters: [
        Parameters.$host,
        Parameters.subscriptionId,
        Parameters.resourceGroupName,
        Parameters.namespaceName1,
        Parameters.topicName,
        Parameters.subscriptionName,
        Parameters.ruleName
    ],
    headerParameters: [Parameters.accept],
    serializer
};
const listBySubscriptionsNextOperationSpec = {
    path: "{nextLink}",
    httpMethod: "GET",
    responses: {
        200: {
            bodyMapper: Mappers.RuleListResult
        },
        default: {
            bodyMapper: Mappers.ErrorResponse
        }
    },
    urlParameters: [
        Parameters.$host,
        Parameters.subscriptionId,
        Parameters.resourceGroupName,
        Parameters.namespaceName1,
        Parameters.nextLink,
        Parameters.topicName,
        Parameters.subscriptionName
    ],
    headerParameters: [Parameters.accept],
    serializer
};
