// @azure/arm-servicebus/dist-esm/src/operations/subscriptions.js
/*
 * Copyright (c) Microsoft Corporation.
 * Licensed under the MIT License.
 *
 * Code generated by Microsoft (R) AutoRest Code Generator.
 * Changes may cause incorrect behavior and will be lost if the code is regenerated.
 */
import { __asyncDelegator, __asyncGenerator, __asyncValues, __await } from "tslib";
import { setContinuationToken } from "../pagingHelper";
import * as coreClient from "@azure/core-client";
import * as Mappers from "../models/mappers";
import * as Parameters from "../models/parameters";
/// <reference lib="esnext.asynciterable" />
/** Class containing Subscriptions operations. */
export class SubscriptionsImpl {
    /**
     * Initialize a new instance of the class Subscriptions class.
     * @param client Reference to the service client
     */
    constructor(client) {
        this.client = client;
    }
    /**
     * List all the subscriptions under a specified topic.
     * @param resourceGroupName Name of the Resource group within the Azure subscription.
     * @param namespaceName The namespace name
     * @param topicName The topic name.
     * @param options The options parameters.
     */
    listByTopic(resourceGroupName, namespaceName, topicName, options) {
        const iter = this.listByTopicPagingAll(resourceGroupName, namespaceName, topicName, options);
        return {
            next() {
                return iter.next();
            },
            [Symbol.asyncIterator]() {
                return this;
            },
            byPage: (settings) => {
                if (settings === null || settings === void 0 ? void 0 : settings.maxPageSize) {
                    throw new Error("maxPageSize is not supported by this operation.");
                }
                return this.listByTopicPagingPage(resourceGroupName, namespaceName, topicName, options, settings);
            }
        };
    }
    listByTopicPagingPage(resourceGroupName, namespaceName, topicName, options, settings) {
        return __asyncGenerator(this, arguments, function* listByTopicPagingPage_1() {
            let result;
            let continuationToken = settings === null || settings === void 0 ? void 0 : settings.continuationToken;
            if (!continuationToken) {
                result = yield __await(this._listByTopic(resourceGroupName, namespaceName, topicName, options));
                let page = result.value || [];
                continuationToken = result.nextLink;
                setContinuationToken(page, continuationToken);
                yield yield __await(page);
            }
            while (continuationToken) {
                result = yield __await(this._listByTopicNext(resourceGroupName, namespaceName, topicName, continuationToken, options));
                continuationToken = result.nextLink;
                let page = result.value || [];
                setContinuationToken(page, continuationToken);
                yield yield __await(page);
            }
        });
    }
    listByTopicPagingAll(resourceGroupName, namespaceName, topicName, options) {
        return __asyncGenerator(this, arguments, function* listByTopicPagingAll_1() {
            var e_1, _a;
            try {
                for (var _b = __asyncValues(this.listByTopicPagingPage(resourceGroupName, namespaceName, topicName, options)), _c; _c = yield __await(_b.next()), !_c.done;) {
                    const page = _c.value;
                    yield __await(yield* __asyncDelegator(__asyncValues(page)));
                }
            }
            catch (e_1_1) { e_1 = { error: e_1_1 }; }
            finally {
                try {
                    if (_c && !_c.done && (_a = _b.return)) yield __await(_a.call(_b));
                }
                finally { if (e_1) throw e_1.error; }
            }
        });
    }
    /**
     * List all the subscriptions under a specified topic.
     * @param resourceGroupName Name of the Resource group within the Azure subscription.
     * @param namespaceName The namespace name
     * @param topicName The topic name.
     * @param options The options parameters.
     */
    _listByTopic(resourceGroupName, namespaceName, topicName, options) {
        return this.client.sendOperationRequest({ resourceGroupName, namespaceName, topicName, options }, listByTopicOperationSpec);
    }
    /**
     * Creates a topic subscription.
     * @param resourceGroupName Name of the Resource group within the Azure subscription.
     * @param namespaceName The namespace name
     * @param topicName The topic name.
     * @param subscriptionName The subscription name.
     * @param parameters Parameters supplied to create a subscription resource.
     * @param options The options parameters.
     */
    createOrUpdate(resourceGroupName, namespaceName, topicName, subscriptionName, parameters, options) {
        return this.client.sendOperationRequest({
            resourceGroupName,
            namespaceName,
            topicName,
            subscriptionName,
            parameters,
            options
        }, createOrUpdateOperationSpec);
    }
    /**
     * Deletes a subscription from the specified topic.
     * @param resourceGroupName Name of the Resource group within the Azure subscription.
     * @param namespaceName The namespace name
     * @param topicName The topic name.
     * @param subscriptionName The subscription name.
     * @param options The options parameters.
     */
    delete(resourceGroupName, namespaceName, topicName, subscriptionName, options) {
        return this.client.sendOperationRequest({
            resourceGroupName,
            namespaceName,
            topicName,
            subscriptionName,
            options
        }, deleteOperationSpec);
    }
    /**
     * Returns a subscription description for the specified topic.
     * @param resourceGroupName Name of the Resource group within the Azure subscription.
     * @param namespaceName The namespace name
     * @param topicName The topic name.
     * @param subscriptionName The subscription name.
     * @param options The options parameters.
     */
    get(resourceGroupName, namespaceName, topicName, subscriptionName, options) {
        return this.client.sendOperationRequest({
            resourceGroupName,
            namespaceName,
            topicName,
            subscriptionName,
            options
        }, getOperationSpec);
    }
    /**
     * ListByTopicNext
     * @param resourceGroupName Name of the Resource group within the Azure subscription.
     * @param namespaceName The namespace name
     * @param topicName The topic name.
     * @param nextLink The nextLink from the previous successful call to the ListByTopic method.
     * @param options The options parameters.
     */
    _listByTopicNext(resourceGroupName, namespaceName, topicName, nextLink, options) {
        return this.client.sendOperationRequest({ resourceGroupName, namespaceName, topicName, nextLink, options }, listByTopicNextOperationSpec);
    }
}
// Operation Specifications
const serializer = coreClient.createSerializer(Mappers, /* isXml */ false);
const listByTopicOperationSpec = {
    path: "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.ServiceBus/namespaces/{namespaceName}/topics/{topicName}/subscriptions",
    httpMethod: "GET",
    responses: {
        200: {
            bodyMapper: Mappers.SBSubscriptionListResult
        },
        default: {
            bodyMapper: Mappers.ErrorResponse
        }
    },
    queryParameters: [Parameters.apiVersion, Parameters.skip, Parameters.top],
    urlParameters: [
        Parameters.$host,
        Parameters.subscriptionId,
        Parameters.resourceGroupName,
        Parameters.namespaceName1,
        Parameters.topicName
    ],
    headerParameters: [Parameters.accept],
    serializer
};
const createOrUpdateOperationSpec = {
    path: "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.ServiceBus/namespaces/{namespaceName}/topics/{topicName}/subscriptions/{subscriptionName}",
    httpMethod: "PUT",
    responses: {
        200: {
            bodyMapper: Mappers.SBSubscription
        },
        default: {
            bodyMapper: Mappers.ErrorResponse
        }
    },
    requestBody: Parameters.parameters13,
    queryParameters: [Parameters.apiVersion],
    urlParameters: [
        Parameters.$host,
        Parameters.subscriptionId,
        Parameters.resourceGroupName,
        Parameters.namespaceName1,
        Parameters.topicName,
        Parameters.subscriptionName
    ],
    headerParameters: [Parameters.accept, Parameters.contentType],
    mediaType: "json",
    serializer
};
const deleteOperationSpec = {
    path: "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.ServiceBus/namespaces/{namespaceName}/topics/{topicName}/subscriptions/{subscriptionName}",
    httpMethod: "DELETE",
    responses: {
        200: {},
        204: {},
        default: {
            bodyMapper: Mappers.ErrorResponse
        }
    },
    queryParameters: [Parameters.apiVersion],
    urlParameters: [
        Parameters.$host,
        Parameters.subscriptionId,
        Parameters.resourceGroupName,
        Parameters.namespaceName1,
        Parameters.topicName,
        Parameters.subscriptionName
    ],
    headerParameters: [Parameters.accept],
    serializer
};
const getOperationSpec = {
    path: "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.ServiceBus/namespaces/{namespaceName}/topics/{topicName}/subscriptions/{subscriptionName}",
    httpMethod: "GET",
    responses: {
        200: {
            bodyMapper: Mappers.SBSubscription
        },
        default: {
            bodyMapper: Mappers.ErrorResponse
        }
    },
    queryParameters: [Parameters.apiVersion],
    urlParameters: [
        Parameters.$host,
        Parameters.subscriptionId,
        Parameters.resourceGroupName,
        Parameters.namespaceName1,
        Parameters.topicName,
        Parameters.subscriptionName
    ],
    headerParameters: [Parameters.accept],
    serializer
};
const listByTopicNextOperationSpec = {
    path: "{nextLink}",
    httpMethod: "GET",
    responses: {
        200: {
            bodyMapper: Mappers.SBSubscriptionListResult
        },
        default: {
            bodyMapper: Mappers.ErrorResponse
        }
    },
    urlParameters: [
        Parameters.$host,
        Parameters.subscriptionId,
        Parameters.resourceGroupName,
        Parameters.namespaceName1,
        Parameters.nextLink,
        Parameters.topicName
    ],
    headerParameters: [Parameters.accept],
    serializer
};
