// @aws-sdk/client-gamelift/dist-es/models/models_1.js
import { DesiredPlayerSessionFilterSensitiveLog, GameSessionFilterSensitiveLog, GameSessionPlacementFilterSensitiveLog, IpPermissionFilterSensitiveLog, MatchmakingTicketFilterSensitiveLog, PlayerFilterSensitiveLog, PlayerLatencyFilterSensitiveLog, } from "./models_0";
export const GameServerHealthCheck = {
    HEALTHY: "HEALTHY",
};
export const SearchGameSessionsOutputFilterSensitiveLog = (obj) => ({
    ...obj,
    ...(obj.GameSessions && { GameSessions: obj.GameSessions.map((item) => GameSessionFilterSensitiveLog(item)) }),
});
export const StartGameSessionPlacementInputFilterSensitiveLog = (obj) => ({
    ...obj,
    ...(obj.PlayerLatencies && {
        PlayerLatencies: obj.PlayerLatencies.map((item) => PlayerLatencyFilterSensitiveLog(item)),
    }),
    ...(obj.DesiredPlayerSessions && {
        DesiredPlayerSessions: obj.DesiredPlayerSessions.map((item) => DesiredPlayerSessionFilterSensitiveLog(item)),
    }),
});
export const StartGameSessionPlacementOutputFilterSensitiveLog = (obj) => ({
    ...obj,
    ...(obj.GameSessionPlacement && {
        GameSessionPlacement: GameSessionPlacementFilterSensitiveLog(obj.GameSessionPlacement),
    }),
});
export const StartMatchBackfillInputFilterSensitiveLog = (obj) => ({
    ...obj,
    ...(obj.Players && { Players: obj.Players.map((item) => PlayerFilterSensitiveLog(item)) }),
});
export const StartMatchBackfillOutputFilterSensitiveLog = (obj) => ({
    ...obj,
    ...(obj.MatchmakingTicket && { MatchmakingTicket: MatchmakingTicketFilterSensitiveLog(obj.MatchmakingTicket) }),
});
export const StartMatchmakingInputFilterSensitiveLog = (obj) => ({
    ...obj,
    ...(obj.Players && { Players: obj.Players.map((item) => PlayerFilterSensitiveLog(item)) }),
});
export const StartMatchmakingOutputFilterSensitiveLog = (obj) => ({
    ...obj,
    ...(obj.MatchmakingTicket && { MatchmakingTicket: MatchmakingTicketFilterSensitiveLog(obj.MatchmakingTicket) }),
});
export const StopGameSessionPlacementOutputFilterSensitiveLog = (obj) => ({
    ...obj,
    ...(obj.GameSessionPlacement && {
        GameSessionPlacement: GameSessionPlacementFilterSensitiveLog(obj.GameSessionPlacement),
    }),
});
export const UpdateFleetPortSettingsInputFilterSensitiveLog = (obj) => ({
    ...obj,
    ...(obj.InboundPermissionAuthorizations && {
        InboundPermissionAuthorizations: obj.InboundPermissionAuthorizations.map((item) => IpPermissionFilterSensitiveLog(item)),
    }),
    ...(obj.InboundPermissionRevocations && {
        InboundPermissionRevocations: obj.InboundPermissionRevocations.map((item) => IpPermissionFilterSensitiveLog(item)),
    }),
});
export const UpdateGameSessionOutputFilterSensitiveLog = (obj) => ({
    ...obj,
    ...(obj.GameSession && { GameSession: GameSessionFilterSensitiveLog(obj.GameSession) }),
});
