// @aws-sdk/client-healthlake/dist-cjs/index.js
"use strict";
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __name = (target, value) => __defProp(target, "name", { value, configurable: true });
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// src/index.ts
var src_exports = {};
__export(src_exports, {
  AccessDeniedException: () => AccessDeniedException,
  AuthorizationStrategy: () => AuthorizationStrategy,
  CmkType: () => CmkType,
  ConflictException: () => ConflictException,
  CreateFHIRDatastoreCommand: () => CreateFHIRDatastoreCommand,
  DatastoreStatus: () => DatastoreStatus,
  DeleteFHIRDatastoreCommand: () => DeleteFHIRDatastoreCommand,
  DescribeFHIRDatastoreCommand: () => DescribeFHIRDatastoreCommand,
  DescribeFHIRExportJobCommand: () => DescribeFHIRExportJobCommand,
  DescribeFHIRImportJobCommand: () => DescribeFHIRImportJobCommand,
  ErrorCategory: () => ErrorCategory,
  FHIRVersion: () => FHIRVersion,
  HealthLake: () => HealthLake,
  HealthLakeClient: () => HealthLakeClient,
  HealthLakeServiceException: () => HealthLakeServiceException,
  InputDataConfig: () => InputDataConfig,
  InternalServerException: () => InternalServerException,
  JobStatus: () => JobStatus,
  ListFHIRDatastoresCommand: () => ListFHIRDatastoresCommand,
  ListFHIRExportJobsCommand: () => ListFHIRExportJobsCommand,
  ListFHIRImportJobsCommand: () => ListFHIRImportJobsCommand,
  ListTagsForResourceCommand: () => ListTagsForResourceCommand,
  OutputDataConfig: () => OutputDataConfig,
  PreloadDataType: () => PreloadDataType,
  ResourceNotFoundException: () => ResourceNotFoundException,
  StartFHIRExportJobCommand: () => StartFHIRExportJobCommand,
  StartFHIRImportJobCommand: () => StartFHIRImportJobCommand,
  TagResourceCommand: () => TagResourceCommand,
  ThrottlingException: () => ThrottlingException,
  UntagResourceCommand: () => UntagResourceCommand,
  ValidationException: () => ValidationException,
  __Client: () => import_smithy_client.Client,
  paginateListFHIRDatastores: () => paginateListFHIRDatastores,
  paginateListFHIRExportJobs: () => paginateListFHIRExportJobs,
  paginateListFHIRImportJobs: () => paginateListFHIRImportJobs
});
module.exports = __toCommonJS(src_exports);

// src/HealthLakeClient.ts
var import_middleware_host_header = require("@aws-sdk/middleware-host-header");
var import_middleware_logger = require("@aws-sdk/middleware-logger");
var import_middleware_recursion_detection = require("@aws-sdk/middleware-recursion-detection");
var import_middleware_user_agent = require("@aws-sdk/middleware-user-agent");
var import_config_resolver = require("@smithy/config-resolver");
var import_core = require("@smithy/core");
var import_middleware_content_length = require("@smithy/middleware-content-length");
var import_middleware_endpoint = require("@smithy/middleware-endpoint");
var import_middleware_retry = require("@smithy/middleware-retry");

var import_httpAuthSchemeProvider = require("./auth/httpAuthSchemeProvider");

// src/endpoint/EndpointParameters.ts
var resolveClientEndpointParameters = /* @__PURE__ */ __name((options) => {
  return {
    ...options,
    useDualstackEndpoint: options.useDualstackEndpoint ?? false,
    useFipsEndpoint: options.useFipsEndpoint ?? false,
    defaultSigningName: "healthlake"
  };
}, "resolveClientEndpointParameters");
var commonParams = {
  UseFIPS: { type: "builtInParams", name: "useFipsEndpoint" },
  Endpoint: { type: "builtInParams", name: "endpoint" },
  Region: { type: "builtInParams", name: "region" },
  UseDualStack: { type: "builtInParams", name: "useDualstackEndpoint" }
};

// src/HealthLakeClient.ts
var import_runtimeConfig = require("././runtimeConfig");

// src/runtimeExtensions.ts
var import_region_config_resolver = require("@aws-sdk/region-config-resolver");
var import_protocol_http = require("@smithy/protocol-http");
var import_smithy_client = require("@smithy/smithy-client");

// src/auth/httpAuthExtensionConfiguration.ts
var getHttpAuthExtensionConfiguration = /* @__PURE__ */ __name((runtimeConfig) => {
  const _httpAuthSchemes = runtimeConfig.httpAuthSchemes;
  let _httpAuthSchemeProvider = runtimeConfig.httpAuthSchemeProvider;
  let _credentials = runtimeConfig.credentials;
  return {
    setHttpAuthScheme(httpAuthScheme) {
      const index = _httpAuthSchemes.findIndex((scheme) => scheme.schemeId === httpAuthScheme.schemeId);
      if (index === -1) {
        _httpAuthSchemes.push(httpAuthScheme);
      } else {
        _httpAuthSchemes.splice(index, 1, httpAuthScheme);
      }
    },
    httpAuthSchemes() {
      return _httpAuthSchemes;
    },
    setHttpAuthSchemeProvider(httpAuthSchemeProvider) {
      _httpAuthSchemeProvider = httpAuthSchemeProvider;
    },
    httpAuthSchemeProvider() {
      return _httpAuthSchemeProvider;
    },
    setCredentials(credentials) {
      _credentials = credentials;
    },
    credentials() {
      return _credentials;
    }
  };
}, "getHttpAuthExtensionConfiguration");
var resolveHttpAuthRuntimeConfig = /* @__PURE__ */ __name((config) => {
  return {
    httpAuthSchemes: config.httpAuthSchemes(),
    httpAuthSchemeProvider: config.httpAuthSchemeProvider(),
    credentials: config.credentials()
  };
}, "resolveHttpAuthRuntimeConfig");

// src/runtimeExtensions.ts
var asPartial = /* @__PURE__ */ __name((t) => t, "asPartial");
var resolveRuntimeExtensions = /* @__PURE__ */ __name((runtimeConfig, extensions) => {
  const extensionConfiguration = {
    ...asPartial((0, import_region_config_resolver.getAwsRegionExtensionConfiguration)(runtimeConfig)),
    ...asPartial((0, import_smithy_client.getDefaultExtensionConfiguration)(runtimeConfig)),
    ...asPartial((0, import_protocol_http.getHttpHandlerExtensionConfiguration)(runtimeConfig)),
    ...asPartial(getHttpAuthExtensionConfiguration(runtimeConfig))
  };
  extensions.forEach((extension) => extension.configure(extensionConfiguration));
  return {
    ...runtimeConfig,
    ...(0, import_region_config_resolver.resolveAwsRegionExtensionConfiguration)(extensionConfiguration),
    ...(0, import_smithy_client.resolveDefaultRuntimeConfig)(extensionConfiguration),
    ...(0, import_protocol_http.resolveHttpHandlerRuntimeConfig)(extensionConfiguration),
    ...resolveHttpAuthRuntimeConfig(extensionConfiguration)
  };
}, "resolveRuntimeExtensions");

// src/HealthLakeClient.ts
var _HealthLakeClient = class _HealthLakeClient extends import_smithy_client.Client {
  constructor(...[configuration]) {
    const _config_0 = (0, import_runtimeConfig.getRuntimeConfig)(configuration || {});
    const _config_1 = resolveClientEndpointParameters(_config_0);
    const _config_2 = (0, import_config_resolver.resolveRegionConfig)(_config_1);
    const _config_3 = (0, import_middleware_endpoint.resolveEndpointConfig)(_config_2);
    const _config_4 = (0, import_middleware_retry.resolveRetryConfig)(_config_3);
    const _config_5 = (0, import_middleware_host_header.resolveHostHeaderConfig)(_config_4);
    const _config_6 = (0, import_middleware_user_agent.resolveUserAgentConfig)(_config_5);
    const _config_7 = (0, import_httpAuthSchemeProvider.resolveHttpAuthSchemeConfig)(_config_6);
    const _config_8 = resolveRuntimeExtensions(_config_7, (configuration == null ? void 0 : configuration.extensions) || []);
    super(_config_8);
    this.config = _config_8;
    this.middlewareStack.use((0, import_middleware_retry.getRetryPlugin)(this.config));
    this.middlewareStack.use((0, import_middleware_content_length.getContentLengthPlugin)(this.config));
    this.middlewareStack.use((0, import_middleware_host_header.getHostHeaderPlugin)(this.config));
    this.middlewareStack.use((0, import_middleware_logger.getLoggerPlugin)(this.config));
    this.middlewareStack.use((0, import_middleware_recursion_detection.getRecursionDetectionPlugin)(this.config));
    this.middlewareStack.use((0, import_middleware_user_agent.getUserAgentPlugin)(this.config));
    this.middlewareStack.use(
      (0, import_core.getHttpAuthSchemeEndpointRuleSetPlugin)(this.config, {
        httpAuthSchemeParametersProvider: this.getDefaultHttpAuthSchemeParametersProvider(),
        identityProviderConfigProvider: this.getIdentityProviderConfigProvider()
      })
    );
    this.middlewareStack.use((0, import_core.getHttpSigningPlugin)(this.config));
  }
  /**
   * Destroy underlying resources, like sockets. It's usually not necessary to do this.
   * However in Node.js, it's best to explicitly shut down the client's agent when it is no longer needed.
   * Otherwise, sockets might stay open for quite a long time before the server terminates them.
   */
  destroy() {
    super.destroy();
  }
  getDefaultHttpAuthSchemeParametersProvider() {
    return import_httpAuthSchemeProvider.defaultHealthLakeHttpAuthSchemeParametersProvider;
  }
  getIdentityProviderConfigProvider() {
    return async (config) => new import_core.DefaultIdentityProviderConfig({
      "aws.auth#sigv4": config.credentials
    });
  }
};
__name(_HealthLakeClient, "HealthLakeClient");
var HealthLakeClient = _HealthLakeClient;

// src/HealthLake.ts


// src/commands/CreateFHIRDatastoreCommand.ts

var import_middleware_serde = require("@smithy/middleware-serde");

var import_types = require("@smithy/types");

// src/protocols/Aws_json1_0.ts
var import_core2 = require("@aws-sdk/core");


var import_uuid = require("uuid");

// src/models/HealthLakeServiceException.ts

var _HealthLakeServiceException = class _HealthLakeServiceException extends import_smithy_client.ServiceException {
  /**
   * @internal
   */
  constructor(options) {
    super(options);
    Object.setPrototypeOf(this, _HealthLakeServiceException.prototype);
  }
};
__name(_HealthLakeServiceException, "HealthLakeServiceException");
var HealthLakeServiceException = _HealthLakeServiceException;

// src/models/models_0.ts
var _AccessDeniedException = class _AccessDeniedException extends HealthLakeServiceException {
  /**
   * @internal
   */
  constructor(opts) {
    super({
      name: "AccessDeniedException",
      $fault: "client",
      ...opts
    });
    this.name = "AccessDeniedException";
    this.$fault = "client";
    Object.setPrototypeOf(this, _AccessDeniedException.prototype);
    this.Message = opts.Message;
  }
};
__name(_AccessDeniedException, "AccessDeniedException");
var AccessDeniedException = _AccessDeniedException;
var AuthorizationStrategy = {
  AWS_AUTH: "AWS_AUTH",
  SMARTV1: "SMART_ON_FHIR_V1"
};
var CmkType = {
  AO_CMK: "AWS_OWNED_KMS_KEY",
  CM_CMK: "CUSTOMER_MANAGED_KMS_KEY"
};
var _ConflictException = class _ConflictException extends HealthLakeServiceException {
  /**
   * @internal
   */
  constructor(opts) {
    super({
      name: "ConflictException",
      $fault: "client",
      ...opts
    });
    this.name = "ConflictException";
    this.$fault = "client";
    Object.setPrototypeOf(this, _ConflictException.prototype);
    this.Message = opts.Message;
  }
};
__name(_ConflictException, "ConflictException");
var ConflictException = _ConflictException;
var FHIRVersion = {
  R4: "R4"
};
var PreloadDataType = {
  SYNTHEA: "SYNTHEA"
};
var DatastoreStatus = {
  ACTIVE: "ACTIVE",
  CREATE_FAILED: "CREATE_FAILED",
  CREATING: "CREATING",
  DELETED: "DELETED",
  DELETING: "DELETING"
};
var _InternalServerException = class _InternalServerException extends HealthLakeServiceException {
  /**
   * @internal
   */
  constructor(opts) {
    super({
      name: "InternalServerException",
      $fault: "server",
      ...opts
    });
    this.name = "InternalServerException";
    this.$fault = "server";
    Object.setPrototypeOf(this, _InternalServerException.prototype);
    this.Message = opts.Message;
  }
};
__name(_InternalServerException, "InternalServerException");
var InternalServerException = _InternalServerException;
var _ThrottlingException = class _ThrottlingException extends HealthLakeServiceException {
  /**
   * @internal
   */
  constructor(opts) {
    super({
      name: "ThrottlingException",
      $fault: "client",
      ...opts
    });
    this.name = "ThrottlingException";
    this.$fault = "client";
    Object.setPrototypeOf(this, _ThrottlingException.prototype);
    this.Message = opts.Message;
  }
};
__name(_ThrottlingException, "ThrottlingException");
var ThrottlingException = _ThrottlingException;
var _ValidationException = class _ValidationException extends HealthLakeServiceException {
  /**
   * @internal
   */
  constructor(opts) {
    super({
      name: "ValidationException",
      $fault: "client",
      ...opts
    });
    this.name = "ValidationException";
    this.$fault = "client";
    Object.setPrototypeOf(this, _ValidationException.prototype);
    this.Message = opts.Message;
  }
};
__name(_ValidationException, "ValidationException");
var ValidationException = _ValidationException;
var ErrorCategory = {
  NON_RETRYABLE_ERROR: "NON_RETRYABLE_ERROR",
  RETRYABLE_ERROR: "RETRYABLE_ERROR"
};
var _ResourceNotFoundException = class _ResourceNotFoundException extends HealthLakeServiceException {
  /**
   * @internal
   */
  constructor(opts) {
    super({
      name: "ResourceNotFoundException",
      $fault: "client",
      ...opts
    });
    this.name = "ResourceNotFoundException";
    this.$fault = "client";
    Object.setPrototypeOf(this, _ResourceNotFoundException.prototype);
    this.Message = opts.Message;
  }
};
__name(_ResourceNotFoundException, "ResourceNotFoundException");
var ResourceNotFoundException = _ResourceNotFoundException;
var JobStatus = {
  CANCEL_COMPLETED: "CANCEL_COMPLETED",
  CANCEL_FAILED: "CANCEL_FAILED",
  CANCEL_IN_PROGRESS: "CANCEL_IN_PROGRESS",
  CANCEL_SUBMITTED: "CANCEL_SUBMITTED",
  COMPLETED: "COMPLETED",
  COMPLETED_WITH_ERRORS: "COMPLETED_WITH_ERRORS",
  FAILED: "FAILED",
  IN_PROGRESS: "IN_PROGRESS",
  SUBMITTED: "SUBMITTED"
};
var OutputDataConfig;
((OutputDataConfig3) => {
  OutputDataConfig3.visit = /* @__PURE__ */ __name((value, visitor) => {
    if (value.S3Configuration !== void 0)
      return visitor.S3Configuration(value.S3Configuration);
    return visitor._(value.$unknown[0], value.$unknown[1]);
  }, "visit");
})(OutputDataConfig || (OutputDataConfig = {}));
var InputDataConfig;
((InputDataConfig3) => {
  InputDataConfig3.visit = /* @__PURE__ */ __name((value, visitor) => {
    if (value.S3Uri !== void 0)
      return visitor.S3Uri(value.S3Uri);
    return visitor._(value.$unknown[0], value.$unknown[1]);
  }, "visit");
})(InputDataConfig || (InputDataConfig = {}));

// src/protocols/Aws_json1_0.ts
var se_CreateFHIRDatastoreCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("CreateFHIRDatastore");
  let body;
  body = JSON.stringify(se_CreateFHIRDatastoreRequest(input, context));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_CreateFHIRDatastoreCommand");
var se_DeleteFHIRDatastoreCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("DeleteFHIRDatastore");
  let body;
  body = JSON.stringify((0, import_smithy_client._json)(input));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_DeleteFHIRDatastoreCommand");
var se_DescribeFHIRDatastoreCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("DescribeFHIRDatastore");
  let body;
  body = JSON.stringify((0, import_smithy_client._json)(input));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_DescribeFHIRDatastoreCommand");
var se_DescribeFHIRExportJobCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("DescribeFHIRExportJob");
  let body;
  body = JSON.stringify((0, import_smithy_client._json)(input));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_DescribeFHIRExportJobCommand");
var se_DescribeFHIRImportJobCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("DescribeFHIRImportJob");
  let body;
  body = JSON.stringify((0, import_smithy_client._json)(input));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_DescribeFHIRImportJobCommand");
var se_ListFHIRDatastoresCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("ListFHIRDatastores");
  let body;
  body = JSON.stringify(se_ListFHIRDatastoresRequest(input, context));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_ListFHIRDatastoresCommand");
var se_ListFHIRExportJobsCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("ListFHIRExportJobs");
  let body;
  body = JSON.stringify(se_ListFHIRExportJobsRequest(input, context));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_ListFHIRExportJobsCommand");
var se_ListFHIRImportJobsCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("ListFHIRImportJobs");
  let body;
  body = JSON.stringify(se_ListFHIRImportJobsRequest(input, context));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_ListFHIRImportJobsCommand");
var se_ListTagsForResourceCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("ListTagsForResource");
  let body;
  body = JSON.stringify((0, import_smithy_client._json)(input));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_ListTagsForResourceCommand");
var se_StartFHIRExportJobCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("StartFHIRExportJob");
  let body;
  body = JSON.stringify(se_StartFHIRExportJobRequest(input, context));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_StartFHIRExportJobCommand");
var se_StartFHIRImportJobCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("StartFHIRImportJob");
  let body;
  body = JSON.stringify(se_StartFHIRImportJobRequest(input, context));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_StartFHIRImportJobCommand");
var se_TagResourceCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("TagResource");
  let body;
  body = JSON.stringify((0, import_smithy_client._json)(input));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_TagResourceCommand");
var se_UntagResourceCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("UntagResource");
  let body;
  body = JSON.stringify((0, import_smithy_client._json)(input));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_UntagResourceCommand");
var de_CreateFHIRDatastoreCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const data = await (0, import_core2.parseJsonBody)(output.body, context);
  let contents = {};
  contents = (0, import_smithy_client._json)(data);
  const response = {
    $metadata: deserializeMetadata(output),
    ...contents
  };
  return response;
}, "de_CreateFHIRDatastoreCommand");
var de_DeleteFHIRDatastoreCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const data = await (0, import_core2.parseJsonBody)(output.body, context);
  let contents = {};
  contents = (0, import_smithy_client._json)(data);
  const response = {
    $metadata: deserializeMetadata(output),
    ...contents
  };
  return response;
}, "de_DeleteFHIRDatastoreCommand");
var de_DescribeFHIRDatastoreCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const data = await (0, import_core2.parseJsonBody)(output.body, context);
  let contents = {};
  contents = de_DescribeFHIRDatastoreResponse(data, context);
  const response = {
    $metadata: deserializeMetadata(output),
    ...contents
  };
  return response;
}, "de_DescribeFHIRDatastoreCommand");
var de_DescribeFHIRExportJobCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const data = await (0, import_core2.parseJsonBody)(output.body, context);
  let contents = {};
  contents = de_DescribeFHIRExportJobResponse(data, context);
  const response = {
    $metadata: deserializeMetadata(output),
    ...contents
  };
  return response;
}, "de_DescribeFHIRExportJobCommand");
var de_DescribeFHIRImportJobCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const data = await (0, import_core2.parseJsonBody)(output.body, context);
  let contents = {};
  contents = de_DescribeFHIRImportJobResponse(data, context);
  const response = {
    $metadata: deserializeMetadata(output),
    ...contents
  };
  return response;
}, "de_DescribeFHIRImportJobCommand");
var de_ListFHIRDatastoresCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const data = await (0, import_core2.parseJsonBody)(output.body, context);
  let contents = {};
  contents = de_ListFHIRDatastoresResponse(data, context);
  const response = {
    $metadata: deserializeMetadata(output),
    ...contents
  };
  return response;
}, "de_ListFHIRDatastoresCommand");
var de_ListFHIRExportJobsCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const data = await (0, import_core2.parseJsonBody)(output.body, context);
  let contents = {};
  contents = de_ListFHIRExportJobsResponse(data, context);
  const response = {
    $metadata: deserializeMetadata(output),
    ...contents
  };
  return response;
}, "de_ListFHIRExportJobsCommand");
var de_ListFHIRImportJobsCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const data = await (0, import_core2.parseJsonBody)(output.body, context);
  let contents = {};
  contents = de_ListFHIRImportJobsResponse(data, context);
  const response = {
    $metadata: deserializeMetadata(output),
    ...contents
  };
  return response;
}, "de_ListFHIRImportJobsCommand");
var de_ListTagsForResourceCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const data = await (0, import_core2.parseJsonBody)(output.body, context);
  let contents = {};
  contents = (0, import_smithy_client._json)(data);
  const response = {
    $metadata: deserializeMetadata(output),
    ...contents
  };
  return response;
}, "de_ListTagsForResourceCommand");
var de_StartFHIRExportJobCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const data = await (0, import_core2.parseJsonBody)(output.body, context);
  let contents = {};
  contents = (0, import_smithy_client._json)(data);
  const response = {
    $metadata: deserializeMetadata(output),
    ...contents
  };
  return response;
}, "de_StartFHIRExportJobCommand");
var de_StartFHIRImportJobCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const data = await (0, import_core2.parseJsonBody)(output.body, context);
  let contents = {};
  contents = (0, import_smithy_client._json)(data);
  const response = {
    $metadata: deserializeMetadata(output),
    ...contents
  };
  return response;
}, "de_StartFHIRImportJobCommand");
var de_TagResourceCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const data = await (0, import_core2.parseJsonBody)(output.body, context);
  let contents = {};
  contents = (0, import_smithy_client._json)(data);
  const response = {
    $metadata: deserializeMetadata(output),
    ...contents
  };
  return response;
}, "de_TagResourceCommand");
var de_UntagResourceCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const data = await (0, import_core2.parseJsonBody)(output.body, context);
  let contents = {};
  contents = (0, import_smithy_client._json)(data);
  const response = {
    $metadata: deserializeMetadata(output),
    ...contents
  };
  return response;
}, "de_UntagResourceCommand");
var de_CommandError = /* @__PURE__ */ __name(async (output, context) => {
  const parsedOutput = {
    ...output,
    body: await (0, import_core2.parseJsonErrorBody)(output.body, context)
  };
  const errorCode = (0, import_core2.loadRestJsonErrorCode)(output, parsedOutput.body);
  switch (errorCode) {
    case "AccessDeniedException":
    case "com.amazonaws.healthlake#AccessDeniedException":
      throw await de_AccessDeniedExceptionRes(parsedOutput, context);
    case "InternalServerException":
    case "com.amazonaws.healthlake#InternalServerException":
      throw await de_InternalServerExceptionRes(parsedOutput, context);
    case "ThrottlingException":
    case "com.amazonaws.healthlake#ThrottlingException":
      throw await de_ThrottlingExceptionRes(parsedOutput, context);
    case "ValidationException":
    case "com.amazonaws.healthlake#ValidationException":
      throw await de_ValidationExceptionRes(parsedOutput, context);
    case "ConflictException":
    case "com.amazonaws.healthlake#ConflictException":
      throw await de_ConflictExceptionRes(parsedOutput, context);
    case "ResourceNotFoundException":
    case "com.amazonaws.healthlake#ResourceNotFoundException":
      throw await de_ResourceNotFoundExceptionRes(parsedOutput, context);
    default:
      const parsedBody = parsedOutput.body;
      return throwDefaultError({
        output,
        parsedBody,
        errorCode
      });
  }
}, "de_CommandError");
var de_AccessDeniedExceptionRes = /* @__PURE__ */ __name(async (parsedOutput, context) => {
  const body = parsedOutput.body;
  const deserialized = (0, import_smithy_client._json)(body);
  const exception = new AccessDeniedException({
    $metadata: deserializeMetadata(parsedOutput),
    ...deserialized
  });
  return (0, import_smithy_client.decorateServiceException)(exception, body);
}, "de_AccessDeniedExceptionRes");
var de_ConflictExceptionRes = /* @__PURE__ */ __name(async (parsedOutput, context) => {
  const body = parsedOutput.body;
  const deserialized = (0, import_smithy_client._json)(body);
  const exception = new ConflictException({
    $metadata: deserializeMetadata(parsedOutput),
    ...deserialized
  });
  return (0, import_smithy_client.decorateServiceException)(exception, body);
}, "de_ConflictExceptionRes");
var de_InternalServerExceptionRes = /* @__PURE__ */ __name(async (parsedOutput, context) => {
  const body = parsedOutput.body;
  const deserialized = (0, import_smithy_client._json)(body);
  const exception = new InternalServerException({
    $metadata: deserializeMetadata(parsedOutput),
    ...deserialized
  });
  return (0, import_smithy_client.decorateServiceException)(exception, body);
}, "de_InternalServerExceptionRes");
var de_ResourceNotFoundExceptionRes = /* @__PURE__ */ __name(async (parsedOutput, context) => {
  const body = parsedOutput.body;
  const deserialized = (0, import_smithy_client._json)(body);
  const exception = new ResourceNotFoundException({
    $metadata: deserializeMetadata(parsedOutput),
    ...deserialized
  });
  return (0, import_smithy_client.decorateServiceException)(exception, body);
}, "de_ResourceNotFoundExceptionRes");
var de_ThrottlingExceptionRes = /* @__PURE__ */ __name(async (parsedOutput, context) => {
  const body = parsedOutput.body;
  const deserialized = (0, import_smithy_client._json)(body);
  const exception = new ThrottlingException({
    $metadata: deserializeMetadata(parsedOutput),
    ...deserialized
  });
  return (0, import_smithy_client.decorateServiceException)(exception, body);
}, "de_ThrottlingExceptionRes");
var de_ValidationExceptionRes = /* @__PURE__ */ __name(async (parsedOutput, context) => {
  const body = parsedOutput.body;
  const deserialized = (0, import_smithy_client._json)(body);
  const exception = new ValidationException({
    $metadata: deserializeMetadata(parsedOutput),
    ...deserialized
  });
  return (0, import_smithy_client.decorateServiceException)(exception, body);
}, "de_ValidationExceptionRes");
var se_CreateFHIRDatastoreRequest = /* @__PURE__ */ __name((input, context) => {
  return (0, import_smithy_client.take)(input, {
    ClientToken: [true, (_) => _ ?? (0, import_uuid.v4)()],
    DatastoreName: [],
    DatastoreTypeVersion: [],
    IdentityProviderConfiguration: import_smithy_client._json,
    PreloadDataConfig: import_smithy_client._json,
    SseConfiguration: import_smithy_client._json,
    Tags: import_smithy_client._json
  });
}, "se_CreateFHIRDatastoreRequest");
var se_DatastoreFilter = /* @__PURE__ */ __name((input, context) => {
  return (0, import_smithy_client.take)(input, {
    CreatedAfter: (_) => Math.round(_.getTime() / 1e3),
    CreatedBefore: (_) => Math.round(_.getTime() / 1e3),
    DatastoreName: [],
    DatastoreStatus: []
  });
}, "se_DatastoreFilter");
var se_ListFHIRDatastoresRequest = /* @__PURE__ */ __name((input, context) => {
  return (0, import_smithy_client.take)(input, {
    Filter: (_) => se_DatastoreFilter(_, context),
    MaxResults: [],
    NextToken: []
  });
}, "se_ListFHIRDatastoresRequest");
var se_ListFHIRExportJobsRequest = /* @__PURE__ */ __name((input, context) => {
  return (0, import_smithy_client.take)(input, {
    DatastoreId: [],
    JobName: [],
    JobStatus: [],
    MaxResults: [],
    NextToken: [],
    SubmittedAfter: (_) => Math.round(_.getTime() / 1e3),
    SubmittedBefore: (_) => Math.round(_.getTime() / 1e3)
  });
}, "se_ListFHIRExportJobsRequest");
var se_ListFHIRImportJobsRequest = /* @__PURE__ */ __name((input, context) => {
  return (0, import_smithy_client.take)(input, {
    DatastoreId: [],
    JobName: [],
    JobStatus: [],
    MaxResults: [],
    NextToken: [],
    SubmittedAfter: (_) => Math.round(_.getTime() / 1e3),
    SubmittedBefore: (_) => Math.round(_.getTime() / 1e3)
  });
}, "se_ListFHIRImportJobsRequest");
var se_StartFHIRExportJobRequest = /* @__PURE__ */ __name((input, context) => {
  return (0, import_smithy_client.take)(input, {
    ClientToken: [true, (_) => _ ?? (0, import_uuid.v4)()],
    DataAccessRoleArn: [],
    DatastoreId: [],
    JobName: [],
    OutputDataConfig: import_smithy_client._json
  });
}, "se_StartFHIRExportJobRequest");
var se_StartFHIRImportJobRequest = /* @__PURE__ */ __name((input, context) => {
  return (0, import_smithy_client.take)(input, {
    ClientToken: [true, (_) => _ ?? (0, import_uuid.v4)()],
    DataAccessRoleArn: [],
    DatastoreId: [],
    InputDataConfig: import_smithy_client._json,
    JobName: [],
    JobOutputDataConfig: import_smithy_client._json
  });
}, "se_StartFHIRImportJobRequest");
var de_DatastoreProperties = /* @__PURE__ */ __name((output, context) => {
  return (0, import_smithy_client.take)(output, {
    CreatedAt: (_) => (0, import_smithy_client.expectNonNull)((0, import_smithy_client.parseEpochTimestamp)((0, import_smithy_client.expectNumber)(_))),
    DatastoreArn: import_smithy_client.expectString,
    DatastoreEndpoint: import_smithy_client.expectString,
    DatastoreId: import_smithy_client.expectString,
    DatastoreName: import_smithy_client.expectString,
    DatastoreStatus: import_smithy_client.expectString,
    DatastoreTypeVersion: import_smithy_client.expectString,
    ErrorCause: import_smithy_client._json,
    IdentityProviderConfiguration: import_smithy_client._json,
    PreloadDataConfig: import_smithy_client._json,
    SseConfiguration: import_smithy_client._json
  });
}, "de_DatastoreProperties");
var de_DatastorePropertiesList = /* @__PURE__ */ __name((output, context) => {
  const retVal = (output || []).filter((e) => e != null).map((entry) => {
    return de_DatastoreProperties(entry, context);
  });
  return retVal;
}, "de_DatastorePropertiesList");
var de_DescribeFHIRDatastoreResponse = /* @__PURE__ */ __name((output, context) => {
  return (0, import_smithy_client.take)(output, {
    DatastoreProperties: (_) => de_DatastoreProperties(_, context)
  });
}, "de_DescribeFHIRDatastoreResponse");
var de_DescribeFHIRExportJobResponse = /* @__PURE__ */ __name((output, context) => {
  return (0, import_smithy_client.take)(output, {
    ExportJobProperties: (_) => de_ExportJobProperties(_, context)
  });
}, "de_DescribeFHIRExportJobResponse");
var de_DescribeFHIRImportJobResponse = /* @__PURE__ */ __name((output, context) => {
  return (0, import_smithy_client.take)(output, {
    ImportJobProperties: (_) => de_ImportJobProperties(_, context)
  });
}, "de_DescribeFHIRImportJobResponse");
var de_ExportJobProperties = /* @__PURE__ */ __name((output, context) => {
  return (0, import_smithy_client.take)(output, {
    DataAccessRoleArn: import_smithy_client.expectString,
    DatastoreId: import_smithy_client.expectString,
    EndTime: (_) => (0, import_smithy_client.expectNonNull)((0, import_smithy_client.parseEpochTimestamp)((0, import_smithy_client.expectNumber)(_))),
    JobId: import_smithy_client.expectString,
    JobName: import_smithy_client.expectString,
    JobStatus: import_smithy_client.expectString,
    Message: import_smithy_client.expectString,
    OutputDataConfig: (_) => (0, import_smithy_client._json)((0, import_core2.awsExpectUnion)(_)),
    SubmitTime: (_) => (0, import_smithy_client.expectNonNull)((0, import_smithy_client.parseEpochTimestamp)((0, import_smithy_client.expectNumber)(_)))
  });
}, "de_ExportJobProperties");
var de_ExportJobPropertiesList = /* @__PURE__ */ __name((output, context) => {
  const retVal = (output || []).filter((e) => e != null).map((entry) => {
    return de_ExportJobProperties(entry, context);
  });
  return retVal;
}, "de_ExportJobPropertiesList");
var de_ImportJobProperties = /* @__PURE__ */ __name((output, context) => {
  return (0, import_smithy_client.take)(output, {
    DataAccessRoleArn: import_smithy_client.expectString,
    DatastoreId: import_smithy_client.expectString,
    EndTime: (_) => (0, import_smithy_client.expectNonNull)((0, import_smithy_client.parseEpochTimestamp)((0, import_smithy_client.expectNumber)(_))),
    InputDataConfig: (_) => (0, import_smithy_client._json)((0, import_core2.awsExpectUnion)(_)),
    JobId: import_smithy_client.expectString,
    JobName: import_smithy_client.expectString,
    JobOutputDataConfig: (_) => (0, import_smithy_client._json)((0, import_core2.awsExpectUnion)(_)),
    JobProgressReport: (_) => de_JobProgressReport(_, context),
    JobStatus: import_smithy_client.expectString,
    Message: import_smithy_client.expectString,
    SubmitTime: (_) => (0, import_smithy_client.expectNonNull)((0, import_smithy_client.parseEpochTimestamp)((0, import_smithy_client.expectNumber)(_)))
  });
}, "de_ImportJobProperties");
var de_ImportJobPropertiesList = /* @__PURE__ */ __name((output, context) => {
  const retVal = (output || []).filter((e) => e != null).map((entry) => {
    return de_ImportJobProperties(entry, context);
  });
  return retVal;
}, "de_ImportJobPropertiesList");
var de_JobProgressReport = /* @__PURE__ */ __name((output, context) => {
  return (0, import_smithy_client.take)(output, {
    Throughput: import_smithy_client.limitedParseDouble,
    TotalNumberOfFilesReadWithCustomerError: import_smithy_client.expectLong,
    TotalNumberOfImportedFiles: import_smithy_client.expectLong,
    TotalNumberOfResourcesImported: import_smithy_client.expectLong,
    TotalNumberOfResourcesScanned: import_smithy_client.expectLong,
    TotalNumberOfResourcesWithCustomerError: import_smithy_client.expectLong,
    TotalNumberOfScannedFiles: import_smithy_client.expectLong,
    TotalSizeOfScannedFilesInMB: import_smithy_client.limitedParseDouble
  });
}, "de_JobProgressReport");
var de_ListFHIRDatastoresResponse = /* @__PURE__ */ __name((output, context) => {
  return (0, import_smithy_client.take)(output, {
    DatastorePropertiesList: (_) => de_DatastorePropertiesList(_, context),
    NextToken: import_smithy_client.expectString
  });
}, "de_ListFHIRDatastoresResponse");
var de_ListFHIRExportJobsResponse = /* @__PURE__ */ __name((output, context) => {
  return (0, import_smithy_client.take)(output, {
    ExportJobPropertiesList: (_) => de_ExportJobPropertiesList(_, context),
    NextToken: import_smithy_client.expectString
  });
}, "de_ListFHIRExportJobsResponse");
var de_ListFHIRImportJobsResponse = /* @__PURE__ */ __name((output, context) => {
  return (0, import_smithy_client.take)(output, {
    ImportJobPropertiesList: (_) => de_ImportJobPropertiesList(_, context),
    NextToken: import_smithy_client.expectString
  });
}, "de_ListFHIRImportJobsResponse");
var deserializeMetadata = /* @__PURE__ */ __name((output) => ({
  httpStatusCode: output.statusCode,
  requestId: output.headers["x-amzn-requestid"] ?? output.headers["x-amzn-request-id"] ?? output.headers["x-amz-request-id"],
  extendedRequestId: output.headers["x-amz-id-2"],
  cfId: output.headers["x-amz-cf-id"]
}), "deserializeMetadata");
var throwDefaultError = (0, import_smithy_client.withBaseException)(HealthLakeServiceException);
var buildHttpRpcRequest = /* @__PURE__ */ __name(async (context, headers, path, resolvedHostname, body) => {
  const { hostname, protocol = "https", port, path: basePath } = await context.endpoint();
  const contents = {
    protocol,
    hostname,
    port,
    method: "POST",
    path: basePath.endsWith("/") ? basePath.slice(0, -1) + path : basePath + path,
    headers
  };
  if (resolvedHostname !== void 0) {
    contents.hostname = resolvedHostname;
  }
  if (body !== void 0) {
    contents.body = body;
  }
  return new import_protocol_http.HttpRequest(contents);
}, "buildHttpRpcRequest");
function sharedHeaders(operation) {
  return {
    "content-type": "application/x-amz-json-1.0",
    "x-amz-target": `HealthLake.${operation}`
  };
}
__name(sharedHeaders, "sharedHeaders");

// src/commands/CreateFHIRDatastoreCommand.ts
var _CreateFHIRDatastoreCommand = class _CreateFHIRDatastoreCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("HealthLake", "CreateFHIRDatastore", {}).n("HealthLakeClient", "CreateFHIRDatastoreCommand").f(void 0, void 0).ser(se_CreateFHIRDatastoreCommand).de(de_CreateFHIRDatastoreCommand).build() {
};
__name(_CreateFHIRDatastoreCommand, "CreateFHIRDatastoreCommand");
var CreateFHIRDatastoreCommand = _CreateFHIRDatastoreCommand;

// src/commands/DeleteFHIRDatastoreCommand.ts




var _DeleteFHIRDatastoreCommand = class _DeleteFHIRDatastoreCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("HealthLake", "DeleteFHIRDatastore", {}).n("HealthLakeClient", "DeleteFHIRDatastoreCommand").f(void 0, void 0).ser(se_DeleteFHIRDatastoreCommand).de(de_DeleteFHIRDatastoreCommand).build() {
};
__name(_DeleteFHIRDatastoreCommand, "DeleteFHIRDatastoreCommand");
var DeleteFHIRDatastoreCommand = _DeleteFHIRDatastoreCommand;

// src/commands/DescribeFHIRDatastoreCommand.ts




var _DescribeFHIRDatastoreCommand = class _DescribeFHIRDatastoreCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("HealthLake", "DescribeFHIRDatastore", {}).n("HealthLakeClient", "DescribeFHIRDatastoreCommand").f(void 0, void 0).ser(se_DescribeFHIRDatastoreCommand).de(de_DescribeFHIRDatastoreCommand).build() {
};
__name(_DescribeFHIRDatastoreCommand, "DescribeFHIRDatastoreCommand");
var DescribeFHIRDatastoreCommand = _DescribeFHIRDatastoreCommand;

// src/commands/DescribeFHIRExportJobCommand.ts




var _DescribeFHIRExportJobCommand = class _DescribeFHIRExportJobCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("HealthLake", "DescribeFHIRExportJob", {}).n("HealthLakeClient", "DescribeFHIRExportJobCommand").f(void 0, void 0).ser(se_DescribeFHIRExportJobCommand).de(de_DescribeFHIRExportJobCommand).build() {
};
__name(_DescribeFHIRExportJobCommand, "DescribeFHIRExportJobCommand");
var DescribeFHIRExportJobCommand = _DescribeFHIRExportJobCommand;

// src/commands/DescribeFHIRImportJobCommand.ts




var _DescribeFHIRImportJobCommand = class _DescribeFHIRImportJobCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("HealthLake", "DescribeFHIRImportJob", {}).n("HealthLakeClient", "DescribeFHIRImportJobCommand").f(void 0, void 0).ser(se_DescribeFHIRImportJobCommand).de(de_DescribeFHIRImportJobCommand).build() {
};
__name(_DescribeFHIRImportJobCommand, "DescribeFHIRImportJobCommand");
var DescribeFHIRImportJobCommand = _DescribeFHIRImportJobCommand;

// src/commands/ListFHIRDatastoresCommand.ts




var _ListFHIRDatastoresCommand = class _ListFHIRDatastoresCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("HealthLake", "ListFHIRDatastores", {}).n("HealthLakeClient", "ListFHIRDatastoresCommand").f(void 0, void 0).ser(se_ListFHIRDatastoresCommand).de(de_ListFHIRDatastoresCommand).build() {
};
__name(_ListFHIRDatastoresCommand, "ListFHIRDatastoresCommand");
var ListFHIRDatastoresCommand = _ListFHIRDatastoresCommand;

// src/commands/ListFHIRExportJobsCommand.ts




var _ListFHIRExportJobsCommand = class _ListFHIRExportJobsCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("HealthLake", "ListFHIRExportJobs", {}).n("HealthLakeClient", "ListFHIRExportJobsCommand").f(void 0, void 0).ser(se_ListFHIRExportJobsCommand).de(de_ListFHIRExportJobsCommand).build() {
};
__name(_ListFHIRExportJobsCommand, "ListFHIRExportJobsCommand");
var ListFHIRExportJobsCommand = _ListFHIRExportJobsCommand;

// src/commands/ListFHIRImportJobsCommand.ts




var _ListFHIRImportJobsCommand = class _ListFHIRImportJobsCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("HealthLake", "ListFHIRImportJobs", {}).n("HealthLakeClient", "ListFHIRImportJobsCommand").f(void 0, void 0).ser(se_ListFHIRImportJobsCommand).de(de_ListFHIRImportJobsCommand).build() {
};
__name(_ListFHIRImportJobsCommand, "ListFHIRImportJobsCommand");
var ListFHIRImportJobsCommand = _ListFHIRImportJobsCommand;

// src/commands/ListTagsForResourceCommand.ts




var _ListTagsForResourceCommand = class _ListTagsForResourceCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("HealthLake", "ListTagsForResource", {}).n("HealthLakeClient", "ListTagsForResourceCommand").f(void 0, void 0).ser(se_ListTagsForResourceCommand).de(de_ListTagsForResourceCommand).build() {
};
__name(_ListTagsForResourceCommand, "ListTagsForResourceCommand");
var ListTagsForResourceCommand = _ListTagsForResourceCommand;

// src/commands/StartFHIRExportJobCommand.ts




var _StartFHIRExportJobCommand = class _StartFHIRExportJobCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("HealthLake", "StartFHIRExportJob", {}).n("HealthLakeClient", "StartFHIRExportJobCommand").f(void 0, void 0).ser(se_StartFHIRExportJobCommand).de(de_StartFHIRExportJobCommand).build() {
};
__name(_StartFHIRExportJobCommand, "StartFHIRExportJobCommand");
var StartFHIRExportJobCommand = _StartFHIRExportJobCommand;

// src/commands/StartFHIRImportJobCommand.ts




var _StartFHIRImportJobCommand = class _StartFHIRImportJobCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("HealthLake", "StartFHIRImportJob", {}).n("HealthLakeClient", "StartFHIRImportJobCommand").f(void 0, void 0).ser(se_StartFHIRImportJobCommand).de(de_StartFHIRImportJobCommand).build() {
};
__name(_StartFHIRImportJobCommand, "StartFHIRImportJobCommand");
var StartFHIRImportJobCommand = _StartFHIRImportJobCommand;

// src/commands/TagResourceCommand.ts




var _TagResourceCommand = class _TagResourceCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("HealthLake", "TagResource", {}).n("HealthLakeClient", "TagResourceCommand").f(void 0, void 0).ser(se_TagResourceCommand).de(de_TagResourceCommand).build() {
};
__name(_TagResourceCommand, "TagResourceCommand");
var TagResourceCommand = _TagResourceCommand;

// src/commands/UntagResourceCommand.ts




var _UntagResourceCommand = class _UntagResourceCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("HealthLake", "UntagResource", {}).n("HealthLakeClient", "UntagResourceCommand").f(void 0, void 0).ser(se_UntagResourceCommand).de(de_UntagResourceCommand).build() {
};
__name(_UntagResourceCommand, "UntagResourceCommand");
var UntagResourceCommand = _UntagResourceCommand;

// src/HealthLake.ts
var commands = {
  CreateFHIRDatastoreCommand,
  DeleteFHIRDatastoreCommand,
  DescribeFHIRDatastoreCommand,
  DescribeFHIRExportJobCommand,
  DescribeFHIRImportJobCommand,
  ListFHIRDatastoresCommand,
  ListFHIRExportJobsCommand,
  ListFHIRImportJobsCommand,
  ListTagsForResourceCommand,
  StartFHIRExportJobCommand,
  StartFHIRImportJobCommand,
  TagResourceCommand,
  UntagResourceCommand
};
var _HealthLake = class _HealthLake extends HealthLakeClient {
};
__name(_HealthLake, "HealthLake");
var HealthLake = _HealthLake;
(0, import_smithy_client.createAggregatedClient)(commands, HealthLake);

// src/pagination/ListFHIRDatastoresPaginator.ts

var paginateListFHIRDatastores = (0, import_core.createPaginator)(HealthLakeClient, ListFHIRDatastoresCommand, "NextToken", "NextToken", "MaxResults");

// src/pagination/ListFHIRExportJobsPaginator.ts

var paginateListFHIRExportJobs = (0, import_core.createPaginator)(HealthLakeClient, ListFHIRExportJobsCommand, "NextToken", "NextToken", "MaxResults");

// src/pagination/ListFHIRImportJobsPaginator.ts

var paginateListFHIRImportJobs = (0, import_core.createPaginator)(HealthLakeClient, ListFHIRImportJobsCommand, "NextToken", "NextToken", "MaxResults");
// Annotate the CommonJS export names for ESM import in node:

0 && (module.exports = {
  HealthLakeServiceException,
  __Client,
  HealthLakeClient,
  HealthLake,
  $Command,
  CreateFHIRDatastoreCommand,
  DeleteFHIRDatastoreCommand,
  DescribeFHIRDatastoreCommand,
  DescribeFHIRExportJobCommand,
  DescribeFHIRImportJobCommand,
  ListFHIRDatastoresCommand,
  ListFHIRExportJobsCommand,
  ListFHIRImportJobsCommand,
  ListTagsForResourceCommand,
  StartFHIRExportJobCommand,
  StartFHIRImportJobCommand,
  TagResourceCommand,
  UntagResourceCommand,
  paginateListFHIRDatastores,
  paginateListFHIRExportJobs,
  paginateListFHIRImportJobs,
  AccessDeniedException,
  AuthorizationStrategy,
  CmkType,
  ConflictException,
  FHIRVersion,
  PreloadDataType,
  DatastoreStatus,
  InternalServerException,
  ThrottlingException,
  ValidationException,
  ErrorCategory,
  ResourceNotFoundException,
  JobStatus,
  OutputDataConfig,
  InputDataConfig
});
