// @azure/arm-operations/esm/operations/managementAssociations.js
/*
 * Copyright (c) Microsoft Corporation.
 * Licensed under the MIT License.
 *
 * Code generated by Microsoft (R) AutoRest Code Generator.
 * Changes may cause incorrect behavior and will be lost if the code is
 * regenerated.
 */
import { __assign } from "tslib";
import * as msRest from "@azure/ms-rest-js";
import * as Mappers from "../models/managementAssociationsMappers";
import * as Parameters from "../models/parameters";
/** Class representing a ManagementAssociations. */
var ManagementAssociations = /** @class */ (function () {
    /**
     * Create a ManagementAssociations.
     * @param {OperationsManagementClientContext} client Reference to the service client.
     */
    function ManagementAssociations(client) {
        this.client = client;
    }
    ManagementAssociations.prototype.listBySubscription = function (options, callback) {
        return this.client.sendOperationRequest({
            options: options
        }, listBySubscriptionOperationSpec, callback);
    };
    ManagementAssociations.prototype.createOrUpdate = function (resourceGroupName, managementAssociationName, parameters, options, callback) {
        return this.client.sendOperationRequest({
            resourceGroupName: resourceGroupName,
            managementAssociationName: managementAssociationName,
            parameters: parameters,
            options: options
        }, createOrUpdateOperationSpec, callback);
    };
    ManagementAssociations.prototype.deleteMethod = function (resourceGroupName, managementAssociationName, options, callback) {
        return this.client.sendOperationRequest({
            resourceGroupName: resourceGroupName,
            managementAssociationName: managementAssociationName,
            options: options
        }, deleteMethodOperationSpec, callback);
    };
    ManagementAssociations.prototype.get = function (resourceGroupName, managementAssociationName, options, callback) {
        return this.client.sendOperationRequest({
            resourceGroupName: resourceGroupName,
            managementAssociationName: managementAssociationName,
            options: options
        }, getOperationSpec, callback);
    };
    return ManagementAssociations;
}());
export { ManagementAssociations };
// Operation Specifications
var serializer = new msRest.Serializer(Mappers);
var listBySubscriptionOperationSpec = {
    httpMethod: "GET",
    path: "subscriptions/{subscriptionId}/providers/Microsoft.OperationsManagement/ManagementAssociations",
    urlParameters: [
        Parameters.subscriptionId
    ],
    queryParameters: [
        Parameters.apiVersion
    ],
    headerParameters: [
        Parameters.acceptLanguage
    ],
    responses: {
        200: {
            bodyMapper: Mappers.ManagementAssociationPropertiesList
        },
        default: {
            bodyMapper: Mappers.CodeMessageError
        }
    },
    serializer: serializer
};
var createOrUpdateOperationSpec = {
    httpMethod: "PUT",
    path: "subscriptions/{subscriptionId}/resourcegroups/{resourceGroupName}/providers/{providerName}/{resourceType}/{resourceName}/providers/Microsoft.OperationsManagement/ManagementAssociations/{managementAssociationName}",
    urlParameters: [
        Parameters.subscriptionId,
        Parameters.resourceGroupName,
        Parameters.providerName,
        Parameters.resourceType,
        Parameters.resourceName,
        Parameters.managementAssociationName
    ],
    queryParameters: [
        Parameters.apiVersion
    ],
    headerParameters: [
        Parameters.acceptLanguage
    ],
    requestBody: {
        parameterPath: "parameters",
        mapper: __assign(__assign({}, Mappers.ManagementAssociation), { required: true })
    },
    responses: {
        200: {
            bodyMapper: Mappers.ManagementAssociation
        },
        default: {
            bodyMapper: Mappers.CodeMessageError
        }
    },
    serializer: serializer
};
var deleteMethodOperationSpec = {
    httpMethod: "DELETE",
    path: "subscriptions/{subscriptionId}/resourcegroups/{resourceGroupName}/providers/{providerName}/{resourceType}/{resourceName}/providers/Microsoft.OperationsManagement/ManagementAssociations/{managementAssociationName}",
    urlParameters: [
        Parameters.subscriptionId,
        Parameters.resourceGroupName,
        Parameters.providerName,
        Parameters.resourceType,
        Parameters.resourceName,
        Parameters.managementAssociationName
    ],
    queryParameters: [
        Parameters.apiVersion
    ],
    headerParameters: [
        Parameters.acceptLanguage
    ],
    responses: {
        200: {},
        default: {
            bodyMapper: Mappers.CodeMessageError
        }
    },
    serializer: serializer
};
var getOperationSpec = {
    httpMethod: "GET",
    path: "subscriptions/{subscriptionId}/resourcegroups/{resourceGroupName}/providers/{providerName}/{resourceType}/{resourceName}/providers/Microsoft.OperationsManagement/ManagementAssociations/{managementAssociationName}",
    urlParameters: [
        Parameters.subscriptionId,
        Parameters.resourceGroupName,
        Parameters.providerName,
        Parameters.resourceType,
        Parameters.resourceName,
        Parameters.managementAssociationName
    ],
    queryParameters: [
        Parameters.apiVersion
    ],
    headerParameters: [
        Parameters.acceptLanguage
    ],
    responses: {
        200: {
            bodyMapper: Mappers.ManagementAssociation
        },
        default: {
            bodyMapper: Mappers.CodeMessageError
        }
    },
    serializer: serializer
};
