// @aws-sdk/client-global-accelerator/dist-cjs/index.js
"use strict";
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __name = (target, value) => __defProp(target, "name", { value, configurable: true });
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// src/index.ts
var src_exports = {};
__export(src_exports, {
  AcceleratorNotDisabledException: () => AcceleratorNotDisabledException,
  AcceleratorNotFoundException: () => AcceleratorNotFoundException,
  AcceleratorStatus: () => AcceleratorStatus,
  AccessDeniedException: () => AccessDeniedException,
  AddCustomRoutingEndpointsCommand: () => AddCustomRoutingEndpointsCommand,
  AddEndpointsCommand: () => AddEndpointsCommand,
  AdvertiseByoipCidrCommand: () => AdvertiseByoipCidrCommand,
  AllowCustomRoutingTrafficCommand: () => AllowCustomRoutingTrafficCommand,
  AssociatedEndpointGroupFoundException: () => AssociatedEndpointGroupFoundException,
  AssociatedListenerFoundException: () => AssociatedListenerFoundException,
  AttachmentNotFoundException: () => AttachmentNotFoundException,
  ByoipCidrNotFoundException: () => ByoipCidrNotFoundException,
  ByoipCidrState: () => ByoipCidrState,
  ClientAffinity: () => ClientAffinity,
  ConflictException: () => ConflictException,
  CreateAcceleratorCommand: () => CreateAcceleratorCommand,
  CreateCrossAccountAttachmentCommand: () => CreateCrossAccountAttachmentCommand,
  CreateCustomRoutingAcceleratorCommand: () => CreateCustomRoutingAcceleratorCommand,
  CreateCustomRoutingEndpointGroupCommand: () => CreateCustomRoutingEndpointGroupCommand,
  CreateCustomRoutingListenerCommand: () => CreateCustomRoutingListenerCommand,
  CreateEndpointGroupCommand: () => CreateEndpointGroupCommand,
  CreateListenerCommand: () => CreateListenerCommand,
  CustomRoutingAcceleratorStatus: () => CustomRoutingAcceleratorStatus,
  CustomRoutingDestinationTrafficState: () => CustomRoutingDestinationTrafficState,
  CustomRoutingProtocol: () => CustomRoutingProtocol,
  DeleteAcceleratorCommand: () => DeleteAcceleratorCommand,
  DeleteCrossAccountAttachmentCommand: () => DeleteCrossAccountAttachmentCommand,
  DeleteCustomRoutingAcceleratorCommand: () => DeleteCustomRoutingAcceleratorCommand,
  DeleteCustomRoutingEndpointGroupCommand: () => DeleteCustomRoutingEndpointGroupCommand,
  DeleteCustomRoutingListenerCommand: () => DeleteCustomRoutingListenerCommand,
  DeleteEndpointGroupCommand: () => DeleteEndpointGroupCommand,
  DeleteListenerCommand: () => DeleteListenerCommand,
  DenyCustomRoutingTrafficCommand: () => DenyCustomRoutingTrafficCommand,
  DeprovisionByoipCidrCommand: () => DeprovisionByoipCidrCommand,
  DescribeAcceleratorAttributesCommand: () => DescribeAcceleratorAttributesCommand,
  DescribeAcceleratorCommand: () => DescribeAcceleratorCommand,
  DescribeCrossAccountAttachmentCommand: () => DescribeCrossAccountAttachmentCommand,
  DescribeCustomRoutingAcceleratorAttributesCommand: () => DescribeCustomRoutingAcceleratorAttributesCommand,
  DescribeCustomRoutingAcceleratorCommand: () => DescribeCustomRoutingAcceleratorCommand,
  DescribeCustomRoutingEndpointGroupCommand: () => DescribeCustomRoutingEndpointGroupCommand,
  DescribeCustomRoutingListenerCommand: () => DescribeCustomRoutingListenerCommand,
  DescribeEndpointGroupCommand: () => DescribeEndpointGroupCommand,
  DescribeListenerCommand: () => DescribeListenerCommand,
  EndpointAlreadyExistsException: () => EndpointAlreadyExistsException,
  EndpointGroupAlreadyExistsException: () => EndpointGroupAlreadyExistsException,
  EndpointGroupNotFoundException: () => EndpointGroupNotFoundException,
  EndpointNotFoundException: () => EndpointNotFoundException,
  GlobalAccelerator: () => GlobalAccelerator,
  GlobalAcceleratorClient: () => GlobalAcceleratorClient,
  GlobalAcceleratorServiceException: () => GlobalAcceleratorServiceException,
  HealthCheckProtocol: () => HealthCheckProtocol,
  HealthState: () => HealthState,
  IncorrectCidrStateException: () => IncorrectCidrStateException,
  InternalServiceErrorException: () => InternalServiceErrorException,
  InvalidArgumentException: () => InvalidArgumentException,
  InvalidNextTokenException: () => InvalidNextTokenException,
  InvalidPortRangeException: () => InvalidPortRangeException,
  IpAddressFamily: () => IpAddressFamily,
  IpAddressType: () => IpAddressType,
  LimitExceededException: () => LimitExceededException,
  ListAcceleratorsCommand: () => ListAcceleratorsCommand,
  ListByoipCidrsCommand: () => ListByoipCidrsCommand,
  ListCrossAccountAttachmentsCommand: () => ListCrossAccountAttachmentsCommand,
  ListCrossAccountResourceAccountsCommand: () => ListCrossAccountResourceAccountsCommand,
  ListCrossAccountResourcesCommand: () => ListCrossAccountResourcesCommand,
  ListCustomRoutingAcceleratorsCommand: () => ListCustomRoutingAcceleratorsCommand,
  ListCustomRoutingEndpointGroupsCommand: () => ListCustomRoutingEndpointGroupsCommand,
  ListCustomRoutingListenersCommand: () => ListCustomRoutingListenersCommand,
  ListCustomRoutingPortMappingsByDestinationCommand: () => ListCustomRoutingPortMappingsByDestinationCommand,
  ListCustomRoutingPortMappingsCommand: () => ListCustomRoutingPortMappingsCommand,
  ListEndpointGroupsCommand: () => ListEndpointGroupsCommand,
  ListListenersCommand: () => ListListenersCommand,
  ListTagsForResourceCommand: () => ListTagsForResourceCommand,
  ListenerNotFoundException: () => ListenerNotFoundException,
  Protocol: () => Protocol,
  ProvisionByoipCidrCommand: () => ProvisionByoipCidrCommand,
  RemoveCustomRoutingEndpointsCommand: () => RemoveCustomRoutingEndpointsCommand,
  RemoveEndpointsCommand: () => RemoveEndpointsCommand,
  TagResourceCommand: () => TagResourceCommand,
  TransactionInProgressException: () => TransactionInProgressException,
  UntagResourceCommand: () => UntagResourceCommand,
  UpdateAcceleratorAttributesCommand: () => UpdateAcceleratorAttributesCommand,
  UpdateAcceleratorCommand: () => UpdateAcceleratorCommand,
  UpdateCrossAccountAttachmentCommand: () => UpdateCrossAccountAttachmentCommand,
  UpdateCustomRoutingAcceleratorAttributesCommand: () => UpdateCustomRoutingAcceleratorAttributesCommand,
  UpdateCustomRoutingAcceleratorCommand: () => UpdateCustomRoutingAcceleratorCommand,
  UpdateCustomRoutingListenerCommand: () => UpdateCustomRoutingListenerCommand,
  UpdateEndpointGroupCommand: () => UpdateEndpointGroupCommand,
  UpdateListenerCommand: () => UpdateListenerCommand,
  WithdrawByoipCidrCommand: () => WithdrawByoipCidrCommand,
  __Client: () => import_smithy_client.Client,
  paginateListAccelerators: () => paginateListAccelerators,
  paginateListByoipCidrs: () => paginateListByoipCidrs,
  paginateListCrossAccountAttachments: () => paginateListCrossAccountAttachments,
  paginateListCrossAccountResources: () => paginateListCrossAccountResources,
  paginateListCustomRoutingAccelerators: () => paginateListCustomRoutingAccelerators,
  paginateListCustomRoutingEndpointGroups: () => paginateListCustomRoutingEndpointGroups,
  paginateListCustomRoutingListeners: () => paginateListCustomRoutingListeners,
  paginateListCustomRoutingPortMappings: () => paginateListCustomRoutingPortMappings,
  paginateListCustomRoutingPortMappingsByDestination: () => paginateListCustomRoutingPortMappingsByDestination,
  paginateListEndpointGroups: () => paginateListEndpointGroups,
  paginateListListeners: () => paginateListListeners
});
module.exports = __toCommonJS(src_exports);

// src/GlobalAcceleratorClient.ts
var import_middleware_host_header = require("@aws-sdk/middleware-host-header");
var import_middleware_logger = require("@aws-sdk/middleware-logger");
var import_middleware_recursion_detection = require("@aws-sdk/middleware-recursion-detection");
var import_middleware_user_agent = require("@aws-sdk/middleware-user-agent");
var import_config_resolver = require("@smithy/config-resolver");
var import_core = require("@smithy/core");
var import_middleware_content_length = require("@smithy/middleware-content-length");
var import_middleware_endpoint = require("@smithy/middleware-endpoint");
var import_middleware_retry = require("@smithy/middleware-retry");

var import_httpAuthSchemeProvider = require("./auth/httpAuthSchemeProvider");

// src/endpoint/EndpointParameters.ts
var resolveClientEndpointParameters = /* @__PURE__ */ __name((options) => {
  return {
    ...options,
    useDualstackEndpoint: options.useDualstackEndpoint ?? false,
    useFipsEndpoint: options.useFipsEndpoint ?? false,
    defaultSigningName: "globalaccelerator"
  };
}, "resolveClientEndpointParameters");
var commonParams = {
  UseFIPS: { type: "builtInParams", name: "useFipsEndpoint" },
  Endpoint: { type: "builtInParams", name: "endpoint" },
  Region: { type: "builtInParams", name: "region" },
  UseDualStack: { type: "builtInParams", name: "useDualstackEndpoint" }
};

// src/GlobalAcceleratorClient.ts
var import_runtimeConfig = require("././runtimeConfig");

// src/runtimeExtensions.ts
var import_region_config_resolver = require("@aws-sdk/region-config-resolver");
var import_protocol_http = require("@smithy/protocol-http");
var import_smithy_client = require("@smithy/smithy-client");

// src/auth/httpAuthExtensionConfiguration.ts
var getHttpAuthExtensionConfiguration = /* @__PURE__ */ __name((runtimeConfig) => {
  const _httpAuthSchemes = runtimeConfig.httpAuthSchemes;
  let _httpAuthSchemeProvider = runtimeConfig.httpAuthSchemeProvider;
  let _credentials = runtimeConfig.credentials;
  return {
    setHttpAuthScheme(httpAuthScheme) {
      const index = _httpAuthSchemes.findIndex((scheme) => scheme.schemeId === httpAuthScheme.schemeId);
      if (index === -1) {
        _httpAuthSchemes.push(httpAuthScheme);
      } else {
        _httpAuthSchemes.splice(index, 1, httpAuthScheme);
      }
    },
    httpAuthSchemes() {
      return _httpAuthSchemes;
    },
    setHttpAuthSchemeProvider(httpAuthSchemeProvider) {
      _httpAuthSchemeProvider = httpAuthSchemeProvider;
    },
    httpAuthSchemeProvider() {
      return _httpAuthSchemeProvider;
    },
    setCredentials(credentials) {
      _credentials = credentials;
    },
    credentials() {
      return _credentials;
    }
  };
}, "getHttpAuthExtensionConfiguration");
var resolveHttpAuthRuntimeConfig = /* @__PURE__ */ __name((config) => {
  return {
    httpAuthSchemes: config.httpAuthSchemes(),
    httpAuthSchemeProvider: config.httpAuthSchemeProvider(),
    credentials: config.credentials()
  };
}, "resolveHttpAuthRuntimeConfig");

// src/runtimeExtensions.ts
var asPartial = /* @__PURE__ */ __name((t) => t, "asPartial");
var resolveRuntimeExtensions = /* @__PURE__ */ __name((runtimeConfig, extensions) => {
  const extensionConfiguration = {
    ...asPartial((0, import_region_config_resolver.getAwsRegionExtensionConfiguration)(runtimeConfig)),
    ...asPartial((0, import_smithy_client.getDefaultExtensionConfiguration)(runtimeConfig)),
    ...asPartial((0, import_protocol_http.getHttpHandlerExtensionConfiguration)(runtimeConfig)),
    ...asPartial(getHttpAuthExtensionConfiguration(runtimeConfig))
  };
  extensions.forEach((extension) => extension.configure(extensionConfiguration));
  return {
    ...runtimeConfig,
    ...(0, import_region_config_resolver.resolveAwsRegionExtensionConfiguration)(extensionConfiguration),
    ...(0, import_smithy_client.resolveDefaultRuntimeConfig)(extensionConfiguration),
    ...(0, import_protocol_http.resolveHttpHandlerRuntimeConfig)(extensionConfiguration),
    ...resolveHttpAuthRuntimeConfig(extensionConfiguration)
  };
}, "resolveRuntimeExtensions");

// src/GlobalAcceleratorClient.ts
var _GlobalAcceleratorClient = class _GlobalAcceleratorClient extends import_smithy_client.Client {
  constructor(...[configuration]) {
    const _config_0 = (0, import_runtimeConfig.getRuntimeConfig)(configuration || {});
    const _config_1 = resolveClientEndpointParameters(_config_0);
    const _config_2 = (0, import_config_resolver.resolveRegionConfig)(_config_1);
    const _config_3 = (0, import_middleware_endpoint.resolveEndpointConfig)(_config_2);
    const _config_4 = (0, import_middleware_retry.resolveRetryConfig)(_config_3);
    const _config_5 = (0, import_middleware_host_header.resolveHostHeaderConfig)(_config_4);
    const _config_6 = (0, import_middleware_user_agent.resolveUserAgentConfig)(_config_5);
    const _config_7 = (0, import_httpAuthSchemeProvider.resolveHttpAuthSchemeConfig)(_config_6);
    const _config_8 = resolveRuntimeExtensions(_config_7, (configuration == null ? void 0 : configuration.extensions) || []);
    super(_config_8);
    this.config = _config_8;
    this.middlewareStack.use((0, import_middleware_retry.getRetryPlugin)(this.config));
    this.middlewareStack.use((0, import_middleware_content_length.getContentLengthPlugin)(this.config));
    this.middlewareStack.use((0, import_middleware_host_header.getHostHeaderPlugin)(this.config));
    this.middlewareStack.use((0, import_middleware_logger.getLoggerPlugin)(this.config));
    this.middlewareStack.use((0, import_middleware_recursion_detection.getRecursionDetectionPlugin)(this.config));
    this.middlewareStack.use((0, import_middleware_user_agent.getUserAgentPlugin)(this.config));
    this.middlewareStack.use(
      (0, import_core.getHttpAuthSchemeEndpointRuleSetPlugin)(this.config, {
        httpAuthSchemeParametersProvider: this.getDefaultHttpAuthSchemeParametersProvider(),
        identityProviderConfigProvider: this.getIdentityProviderConfigProvider()
      })
    );
    this.middlewareStack.use((0, import_core.getHttpSigningPlugin)(this.config));
  }
  /**
   * Destroy underlying resources, like sockets. It's usually not necessary to do this.
   * However in Node.js, it's best to explicitly shut down the client's agent when it is no longer needed.
   * Otherwise, sockets might stay open for quite a long time before the server terminates them.
   */
  destroy() {
    super.destroy();
  }
  getDefaultHttpAuthSchemeParametersProvider() {
    return import_httpAuthSchemeProvider.defaultGlobalAcceleratorHttpAuthSchemeParametersProvider;
  }
  getIdentityProviderConfigProvider() {
    return async (config) => new import_core.DefaultIdentityProviderConfig({
      "aws.auth#sigv4": config.credentials
    });
  }
};
__name(_GlobalAcceleratorClient, "GlobalAcceleratorClient");
var GlobalAcceleratorClient = _GlobalAcceleratorClient;

// src/GlobalAccelerator.ts


// src/commands/AddCustomRoutingEndpointsCommand.ts

var import_middleware_serde = require("@smithy/middleware-serde");

var import_types = require("@smithy/types");

// src/protocols/Aws_json1_1.ts
var import_core2 = require("@aws-sdk/core");


var import_uuid = require("uuid");

// src/models/GlobalAcceleratorServiceException.ts

var _GlobalAcceleratorServiceException = class _GlobalAcceleratorServiceException extends import_smithy_client.ServiceException {
  /**
   * @internal
   */
  constructor(options) {
    super(options);
    Object.setPrototypeOf(this, _GlobalAcceleratorServiceException.prototype);
  }
};
__name(_GlobalAcceleratorServiceException, "GlobalAcceleratorServiceException");
var GlobalAcceleratorServiceException = _GlobalAcceleratorServiceException;

// src/models/models_0.ts
var IpAddressType = {
  DUAL_STACK: "DUAL_STACK",
  IPV4: "IPV4"
};
var IpAddressFamily = {
  IPv4: "IPv4",
  IPv6: "IPv6"
};
var AcceleratorStatus = {
  DEPLOYED: "DEPLOYED",
  IN_PROGRESS: "IN_PROGRESS"
};
var _AcceleratorNotDisabledException = class _AcceleratorNotDisabledException extends GlobalAcceleratorServiceException {
  /**
   * @internal
   */
  constructor(opts) {
    super({
      name: "AcceleratorNotDisabledException",
      $fault: "client",
      ...opts
    });
    this.name = "AcceleratorNotDisabledException";
    this.$fault = "client";
    Object.setPrototypeOf(this, _AcceleratorNotDisabledException.prototype);
    this.Message = opts.Message;
  }
};
__name(_AcceleratorNotDisabledException, "AcceleratorNotDisabledException");
var AcceleratorNotDisabledException = _AcceleratorNotDisabledException;
var _AcceleratorNotFoundException = class _AcceleratorNotFoundException extends GlobalAcceleratorServiceException {
  /**
   * @internal
   */
  constructor(opts) {
    super({
      name: "AcceleratorNotFoundException",
      $fault: "client",
      ...opts
    });
    this.name = "AcceleratorNotFoundException";
    this.$fault = "client";
    Object.setPrototypeOf(this, _AcceleratorNotFoundException.prototype);
    this.Message = opts.Message;
  }
};
__name(_AcceleratorNotFoundException, "AcceleratorNotFoundException");
var AcceleratorNotFoundException = _AcceleratorNotFoundException;
var _AccessDeniedException = class _AccessDeniedException extends GlobalAcceleratorServiceException {
  /**
   * @internal
   */
  constructor(opts) {
    super({
      name: "AccessDeniedException",
      $fault: "client",
      ...opts
    });
    this.name = "AccessDeniedException";
    this.$fault = "client";
    Object.setPrototypeOf(this, _AccessDeniedException.prototype);
    this.Message = opts.Message;
  }
};
__name(_AccessDeniedException, "AccessDeniedException");
var AccessDeniedException = _AccessDeniedException;
var _ConflictException = class _ConflictException extends GlobalAcceleratorServiceException {
  /**
   * @internal
   */
  constructor(opts) {
    super({
      name: "ConflictException",
      $fault: "client",
      ...opts
    });
    this.name = "ConflictException";
    this.$fault = "client";
    Object.setPrototypeOf(this, _ConflictException.prototype);
    this.Message = opts.Message;
  }
};
__name(_ConflictException, "ConflictException");
var ConflictException = _ConflictException;
var _EndpointAlreadyExistsException = class _EndpointAlreadyExistsException extends GlobalAcceleratorServiceException {
  /**
   * @internal
   */
  constructor(opts) {
    super({
      name: "EndpointAlreadyExistsException",
      $fault: "client",
      ...opts
    });
    this.name = "EndpointAlreadyExistsException";
    this.$fault = "client";
    Object.setPrototypeOf(this, _EndpointAlreadyExistsException.prototype);
    this.Message = opts.Message;
  }
};
__name(_EndpointAlreadyExistsException, "EndpointAlreadyExistsException");
var EndpointAlreadyExistsException = _EndpointAlreadyExistsException;
var _EndpointGroupNotFoundException = class _EndpointGroupNotFoundException extends GlobalAcceleratorServiceException {
  /**
   * @internal
   */
  constructor(opts) {
    super({
      name: "EndpointGroupNotFoundException",
      $fault: "client",
      ...opts
    });
    this.name = "EndpointGroupNotFoundException";
    this.$fault = "client";
    Object.setPrototypeOf(this, _EndpointGroupNotFoundException.prototype);
    this.Message = opts.Message;
  }
};
__name(_EndpointGroupNotFoundException, "EndpointGroupNotFoundException");
var EndpointGroupNotFoundException = _EndpointGroupNotFoundException;
var _InternalServiceErrorException = class _InternalServiceErrorException extends GlobalAcceleratorServiceException {
  /**
   * @internal
   */
  constructor(opts) {
    super({
      name: "InternalServiceErrorException",
      $fault: "server",
      ...opts
    });
    this.name = "InternalServiceErrorException";
    this.$fault = "server";
    Object.setPrototypeOf(this, _InternalServiceErrorException.prototype);
    this.Message = opts.Message;
  }
};
__name(_InternalServiceErrorException, "InternalServiceErrorException");
var InternalServiceErrorException = _InternalServiceErrorException;
var _InvalidArgumentException = class _InvalidArgumentException extends GlobalAcceleratorServiceException {
  /**
   * @internal
   */
  constructor(opts) {
    super({
      name: "InvalidArgumentException",
      $fault: "client",
      ...opts
    });
    this.name = "InvalidArgumentException";
    this.$fault = "client";
    Object.setPrototypeOf(this, _InvalidArgumentException.prototype);
    this.Message = opts.Message;
  }
};
__name(_InvalidArgumentException, "InvalidArgumentException");
var InvalidArgumentException = _InvalidArgumentException;
var _LimitExceededException = class _LimitExceededException extends GlobalAcceleratorServiceException {
  /**
   * @internal
   */
  constructor(opts) {
    super({
      name: "LimitExceededException",
      $fault: "client",
      ...opts
    });
    this.name = "LimitExceededException";
    this.$fault = "client";
    Object.setPrototypeOf(this, _LimitExceededException.prototype);
    this.Message = opts.Message;
  }
};
__name(_LimitExceededException, "LimitExceededException");
var LimitExceededException = _LimitExceededException;
var HealthState = {
  HEALTHY: "HEALTHY",
  INITIAL: "INITIAL",
  UNHEALTHY: "UNHEALTHY"
};
var _TransactionInProgressException = class _TransactionInProgressException extends GlobalAcceleratorServiceException {
  /**
   * @internal
   */
  constructor(opts) {
    super({
      name: "TransactionInProgressException",
      $fault: "client",
      ...opts
    });
    this.name = "TransactionInProgressException";
    this.$fault = "client";
    Object.setPrototypeOf(this, _TransactionInProgressException.prototype);
    this.Message = opts.Message;
  }
};
__name(_TransactionInProgressException, "TransactionInProgressException");
var TransactionInProgressException = _TransactionInProgressException;
var ByoipCidrState = {
  ADVERTISING: "ADVERTISING",
  DEPROVISIONED: "DEPROVISIONED",
  FAILED_ADVERTISING: "FAILED_ADVERTISING",
  FAILED_DEPROVISION: "FAILED_DEPROVISION",
  FAILED_PROVISION: "FAILED_PROVISION",
  FAILED_WITHDRAW: "FAILED_WITHDRAW",
  PENDING_ADVERTISING: "PENDING_ADVERTISING",
  PENDING_DEPROVISIONING: "PENDING_DEPROVISIONING",
  PENDING_PROVISIONING: "PENDING_PROVISIONING",
  PENDING_WITHDRAWING: "PENDING_WITHDRAWING",
  READY: "READY"
};
var _ByoipCidrNotFoundException = class _ByoipCidrNotFoundException extends GlobalAcceleratorServiceException {
  /**
   * @internal
   */
  constructor(opts) {
    super({
      name: "ByoipCidrNotFoundException",
      $fault: "client",
      ...opts
    });
    this.name = "ByoipCidrNotFoundException";
    this.$fault = "client";
    Object.setPrototypeOf(this, _ByoipCidrNotFoundException.prototype);
    this.Message = opts.Message;
  }
};
__name(_ByoipCidrNotFoundException, "ByoipCidrNotFoundException");
var ByoipCidrNotFoundException = _ByoipCidrNotFoundException;
var _IncorrectCidrStateException = class _IncorrectCidrStateException extends GlobalAcceleratorServiceException {
  /**
   * @internal
   */
  constructor(opts) {
    super({
      name: "IncorrectCidrStateException",
      $fault: "client",
      ...opts
    });
    this.name = "IncorrectCidrStateException";
    this.$fault = "client";
    Object.setPrototypeOf(this, _IncorrectCidrStateException.prototype);
    this.Message = opts.Message;
  }
};
__name(_IncorrectCidrStateException, "IncorrectCidrStateException");
var IncorrectCidrStateException = _IncorrectCidrStateException;
var _AssociatedEndpointGroupFoundException = class _AssociatedEndpointGroupFoundException extends GlobalAcceleratorServiceException {
  /**
   * @internal
   */
  constructor(opts) {
    super({
      name: "AssociatedEndpointGroupFoundException",
      $fault: "client",
      ...opts
    });
    this.name = "AssociatedEndpointGroupFoundException";
    this.$fault = "client";
    Object.setPrototypeOf(this, _AssociatedEndpointGroupFoundException.prototype);
    this.Message = opts.Message;
  }
};
__name(_AssociatedEndpointGroupFoundException, "AssociatedEndpointGroupFoundException");
var AssociatedEndpointGroupFoundException = _AssociatedEndpointGroupFoundException;
var _AssociatedListenerFoundException = class _AssociatedListenerFoundException extends GlobalAcceleratorServiceException {
  /**
   * @internal
   */
  constructor(opts) {
    super({
      name: "AssociatedListenerFoundException",
      $fault: "client",
      ...opts
    });
    this.name = "AssociatedListenerFoundException";
    this.$fault = "client";
    Object.setPrototypeOf(this, _AssociatedListenerFoundException.prototype);
    this.Message = opts.Message;
  }
};
__name(_AssociatedListenerFoundException, "AssociatedListenerFoundException");
var AssociatedListenerFoundException = _AssociatedListenerFoundException;
var _AttachmentNotFoundException = class _AttachmentNotFoundException extends GlobalAcceleratorServiceException {
  /**
   * @internal
   */
  constructor(opts) {
    super({
      name: "AttachmentNotFoundException",
      $fault: "client",
      ...opts
    });
    this.name = "AttachmentNotFoundException";
    this.$fault = "client";
    Object.setPrototypeOf(this, _AttachmentNotFoundException.prototype);
    this.Message = opts.Message;
  }
};
__name(_AttachmentNotFoundException, "AttachmentNotFoundException");
var AttachmentNotFoundException = _AttachmentNotFoundException;
var ClientAffinity = {
  NONE: "NONE",
  SOURCE_IP: "SOURCE_IP"
};
var CustomRoutingAcceleratorStatus = {
  DEPLOYED: "DEPLOYED",
  IN_PROGRESS: "IN_PROGRESS"
};
var CustomRoutingProtocol = {
  TCP: "TCP",
  UDP: "UDP"
};
var Protocol = {
  TCP: "TCP",
  UDP: "UDP"
};
var _EndpointGroupAlreadyExistsException = class _EndpointGroupAlreadyExistsException extends GlobalAcceleratorServiceException {
  /**
   * @internal
   */
  constructor(opts) {
    super({
      name: "EndpointGroupAlreadyExistsException",
      $fault: "client",
      ...opts
    });
    this.name = "EndpointGroupAlreadyExistsException";
    this.$fault = "client";
    Object.setPrototypeOf(this, _EndpointGroupAlreadyExistsException.prototype);
    this.Message = opts.Message;
  }
};
__name(_EndpointGroupAlreadyExistsException, "EndpointGroupAlreadyExistsException");
var EndpointGroupAlreadyExistsException = _EndpointGroupAlreadyExistsException;
var _InvalidPortRangeException = class _InvalidPortRangeException extends GlobalAcceleratorServiceException {
  /**
   * @internal
   */
  constructor(opts) {
    super({
      name: "InvalidPortRangeException",
      $fault: "client",
      ...opts
    });
    this.name = "InvalidPortRangeException";
    this.$fault = "client";
    Object.setPrototypeOf(this, _InvalidPortRangeException.prototype);
    this.Message = opts.Message;
  }
};
__name(_InvalidPortRangeException, "InvalidPortRangeException");
var InvalidPortRangeException = _InvalidPortRangeException;
var _ListenerNotFoundException = class _ListenerNotFoundException extends GlobalAcceleratorServiceException {
  /**
   * @internal
   */
  constructor(opts) {
    super({
      name: "ListenerNotFoundException",
      $fault: "client",
      ...opts
    });
    this.name = "ListenerNotFoundException";
    this.$fault = "client";
    Object.setPrototypeOf(this, _ListenerNotFoundException.prototype);
    this.Message = opts.Message;
  }
};
__name(_ListenerNotFoundException, "ListenerNotFoundException");
var ListenerNotFoundException = _ListenerNotFoundException;
var HealthCheckProtocol = {
  HTTP: "HTTP",
  HTTPS: "HTTPS",
  TCP: "TCP"
};
var CustomRoutingDestinationTrafficState = {
  ALLOW: "ALLOW",
  DENY: "DENY"
};
var _EndpointNotFoundException = class _EndpointNotFoundException extends GlobalAcceleratorServiceException {
  /**
   * @internal
   */
  constructor(opts) {
    super({
      name: "EndpointNotFoundException",
      $fault: "client",
      ...opts
    });
    this.name = "EndpointNotFoundException";
    this.$fault = "client";
    Object.setPrototypeOf(this, _EndpointNotFoundException.prototype);
    this.Message = opts.Message;
  }
};
__name(_EndpointNotFoundException, "EndpointNotFoundException");
var EndpointNotFoundException = _EndpointNotFoundException;
var _InvalidNextTokenException = class _InvalidNextTokenException extends GlobalAcceleratorServiceException {
  /**
   * @internal
   */
  constructor(opts) {
    super({
      name: "InvalidNextTokenException",
      $fault: "client",
      ...opts
    });
    this.name = "InvalidNextTokenException";
    this.$fault = "client";
    Object.setPrototypeOf(this, _InvalidNextTokenException.prototype);
    this.Message = opts.Message;
  }
};
__name(_InvalidNextTokenException, "InvalidNextTokenException");
var InvalidNextTokenException = _InvalidNextTokenException;

// src/protocols/Aws_json1_1.ts
var se_AddCustomRoutingEndpointsCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("AddCustomRoutingEndpoints");
  let body;
  body = JSON.stringify((0, import_smithy_client._json)(input));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_AddCustomRoutingEndpointsCommand");
var se_AddEndpointsCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("AddEndpoints");
  let body;
  body = JSON.stringify((0, import_smithy_client._json)(input));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_AddEndpointsCommand");
var se_AdvertiseByoipCidrCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("AdvertiseByoipCidr");
  let body;
  body = JSON.stringify((0, import_smithy_client._json)(input));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_AdvertiseByoipCidrCommand");
var se_AllowCustomRoutingTrafficCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("AllowCustomRoutingTraffic");
  let body;
  body = JSON.stringify((0, import_smithy_client._json)(input));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_AllowCustomRoutingTrafficCommand");
var se_CreateAcceleratorCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("CreateAccelerator");
  let body;
  body = JSON.stringify(se_CreateAcceleratorRequest(input, context));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_CreateAcceleratorCommand");
var se_CreateCrossAccountAttachmentCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("CreateCrossAccountAttachment");
  let body;
  body = JSON.stringify(se_CreateCrossAccountAttachmentRequest(input, context));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_CreateCrossAccountAttachmentCommand");
var se_CreateCustomRoutingAcceleratorCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("CreateCustomRoutingAccelerator");
  let body;
  body = JSON.stringify(se_CreateCustomRoutingAcceleratorRequest(input, context));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_CreateCustomRoutingAcceleratorCommand");
var se_CreateCustomRoutingEndpointGroupCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("CreateCustomRoutingEndpointGroup");
  let body;
  body = JSON.stringify(se_CreateCustomRoutingEndpointGroupRequest(input, context));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_CreateCustomRoutingEndpointGroupCommand");
var se_CreateCustomRoutingListenerCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("CreateCustomRoutingListener");
  let body;
  body = JSON.stringify(se_CreateCustomRoutingListenerRequest(input, context));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_CreateCustomRoutingListenerCommand");
var se_CreateEndpointGroupCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("CreateEndpointGroup");
  let body;
  body = JSON.stringify(se_CreateEndpointGroupRequest(input, context));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_CreateEndpointGroupCommand");
var se_CreateListenerCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("CreateListener");
  let body;
  body = JSON.stringify(se_CreateListenerRequest(input, context));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_CreateListenerCommand");
var se_DeleteAcceleratorCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("DeleteAccelerator");
  let body;
  body = JSON.stringify((0, import_smithy_client._json)(input));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_DeleteAcceleratorCommand");
var se_DeleteCrossAccountAttachmentCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("DeleteCrossAccountAttachment");
  let body;
  body = JSON.stringify((0, import_smithy_client._json)(input));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_DeleteCrossAccountAttachmentCommand");
var se_DeleteCustomRoutingAcceleratorCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("DeleteCustomRoutingAccelerator");
  let body;
  body = JSON.stringify((0, import_smithy_client._json)(input));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_DeleteCustomRoutingAcceleratorCommand");
var se_DeleteCustomRoutingEndpointGroupCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("DeleteCustomRoutingEndpointGroup");
  let body;
  body = JSON.stringify((0, import_smithy_client._json)(input));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_DeleteCustomRoutingEndpointGroupCommand");
var se_DeleteCustomRoutingListenerCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("DeleteCustomRoutingListener");
  let body;
  body = JSON.stringify((0, import_smithy_client._json)(input));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_DeleteCustomRoutingListenerCommand");
var se_DeleteEndpointGroupCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("DeleteEndpointGroup");
  let body;
  body = JSON.stringify((0, import_smithy_client._json)(input));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_DeleteEndpointGroupCommand");
var se_DeleteListenerCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("DeleteListener");
  let body;
  body = JSON.stringify((0, import_smithy_client._json)(input));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_DeleteListenerCommand");
var se_DenyCustomRoutingTrafficCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("DenyCustomRoutingTraffic");
  let body;
  body = JSON.stringify((0, import_smithy_client._json)(input));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_DenyCustomRoutingTrafficCommand");
var se_DeprovisionByoipCidrCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("DeprovisionByoipCidr");
  let body;
  body = JSON.stringify((0, import_smithy_client._json)(input));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_DeprovisionByoipCidrCommand");
var se_DescribeAcceleratorCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("DescribeAccelerator");
  let body;
  body = JSON.stringify((0, import_smithy_client._json)(input));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_DescribeAcceleratorCommand");
var se_DescribeAcceleratorAttributesCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("DescribeAcceleratorAttributes");
  let body;
  body = JSON.stringify((0, import_smithy_client._json)(input));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_DescribeAcceleratorAttributesCommand");
var se_DescribeCrossAccountAttachmentCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("DescribeCrossAccountAttachment");
  let body;
  body = JSON.stringify((0, import_smithy_client._json)(input));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_DescribeCrossAccountAttachmentCommand");
var se_DescribeCustomRoutingAcceleratorCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("DescribeCustomRoutingAccelerator");
  let body;
  body = JSON.stringify((0, import_smithy_client._json)(input));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_DescribeCustomRoutingAcceleratorCommand");
var se_DescribeCustomRoutingAcceleratorAttributesCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("DescribeCustomRoutingAcceleratorAttributes");
  let body;
  body = JSON.stringify((0, import_smithy_client._json)(input));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_DescribeCustomRoutingAcceleratorAttributesCommand");
var se_DescribeCustomRoutingEndpointGroupCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("DescribeCustomRoutingEndpointGroup");
  let body;
  body = JSON.stringify((0, import_smithy_client._json)(input));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_DescribeCustomRoutingEndpointGroupCommand");
var se_DescribeCustomRoutingListenerCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("DescribeCustomRoutingListener");
  let body;
  body = JSON.stringify((0, import_smithy_client._json)(input));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_DescribeCustomRoutingListenerCommand");
var se_DescribeEndpointGroupCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("DescribeEndpointGroup");
  let body;
  body = JSON.stringify((0, import_smithy_client._json)(input));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_DescribeEndpointGroupCommand");
var se_DescribeListenerCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("DescribeListener");
  let body;
  body = JSON.stringify((0, import_smithy_client._json)(input));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_DescribeListenerCommand");
var se_ListAcceleratorsCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("ListAccelerators");
  let body;
  body = JSON.stringify((0, import_smithy_client._json)(input));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_ListAcceleratorsCommand");
var se_ListByoipCidrsCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("ListByoipCidrs");
  let body;
  body = JSON.stringify((0, import_smithy_client._json)(input));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_ListByoipCidrsCommand");
var se_ListCrossAccountAttachmentsCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("ListCrossAccountAttachments");
  let body;
  body = JSON.stringify((0, import_smithy_client._json)(input));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_ListCrossAccountAttachmentsCommand");
var se_ListCrossAccountResourceAccountsCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("ListCrossAccountResourceAccounts");
  let body;
  body = JSON.stringify((0, import_smithy_client._json)(input));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_ListCrossAccountResourceAccountsCommand");
var se_ListCrossAccountResourcesCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("ListCrossAccountResources");
  let body;
  body = JSON.stringify((0, import_smithy_client._json)(input));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_ListCrossAccountResourcesCommand");
var se_ListCustomRoutingAcceleratorsCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("ListCustomRoutingAccelerators");
  let body;
  body = JSON.stringify((0, import_smithy_client._json)(input));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_ListCustomRoutingAcceleratorsCommand");
var se_ListCustomRoutingEndpointGroupsCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("ListCustomRoutingEndpointGroups");
  let body;
  body = JSON.stringify((0, import_smithy_client._json)(input));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_ListCustomRoutingEndpointGroupsCommand");
var se_ListCustomRoutingListenersCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("ListCustomRoutingListeners");
  let body;
  body = JSON.stringify((0, import_smithy_client._json)(input));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_ListCustomRoutingListenersCommand");
var se_ListCustomRoutingPortMappingsCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("ListCustomRoutingPortMappings");
  let body;
  body = JSON.stringify((0, import_smithy_client._json)(input));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_ListCustomRoutingPortMappingsCommand");
var se_ListCustomRoutingPortMappingsByDestinationCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("ListCustomRoutingPortMappingsByDestination");
  let body;
  body = JSON.stringify((0, import_smithy_client._json)(input));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_ListCustomRoutingPortMappingsByDestinationCommand");
var se_ListEndpointGroupsCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("ListEndpointGroups");
  let body;
  body = JSON.stringify((0, import_smithy_client._json)(input));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_ListEndpointGroupsCommand");
var se_ListListenersCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("ListListeners");
  let body;
  body = JSON.stringify((0, import_smithy_client._json)(input));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_ListListenersCommand");
var se_ListTagsForResourceCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("ListTagsForResource");
  let body;
  body = JSON.stringify((0, import_smithy_client._json)(input));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_ListTagsForResourceCommand");
var se_ProvisionByoipCidrCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("ProvisionByoipCidr");
  let body;
  body = JSON.stringify((0, import_smithy_client._json)(input));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_ProvisionByoipCidrCommand");
var se_RemoveCustomRoutingEndpointsCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("RemoveCustomRoutingEndpoints");
  let body;
  body = JSON.stringify((0, import_smithy_client._json)(input));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_RemoveCustomRoutingEndpointsCommand");
var se_RemoveEndpointsCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("RemoveEndpoints");
  let body;
  body = JSON.stringify((0, import_smithy_client._json)(input));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_RemoveEndpointsCommand");
var se_TagResourceCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("TagResource");
  let body;
  body = JSON.stringify((0, import_smithy_client._json)(input));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_TagResourceCommand");
var se_UntagResourceCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("UntagResource");
  let body;
  body = JSON.stringify((0, import_smithy_client._json)(input));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_UntagResourceCommand");
var se_UpdateAcceleratorCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("UpdateAccelerator");
  let body;
  body = JSON.stringify((0, import_smithy_client._json)(input));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_UpdateAcceleratorCommand");
var se_UpdateAcceleratorAttributesCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("UpdateAcceleratorAttributes");
  let body;
  body = JSON.stringify((0, import_smithy_client._json)(input));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_UpdateAcceleratorAttributesCommand");
var se_UpdateCrossAccountAttachmentCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("UpdateCrossAccountAttachment");
  let body;
  body = JSON.stringify((0, import_smithy_client._json)(input));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_UpdateCrossAccountAttachmentCommand");
var se_UpdateCustomRoutingAcceleratorCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("UpdateCustomRoutingAccelerator");
  let body;
  body = JSON.stringify((0, import_smithy_client._json)(input));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_UpdateCustomRoutingAcceleratorCommand");
var se_UpdateCustomRoutingAcceleratorAttributesCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("UpdateCustomRoutingAcceleratorAttributes");
  let body;
  body = JSON.stringify((0, import_smithy_client._json)(input));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_UpdateCustomRoutingAcceleratorAttributesCommand");
var se_UpdateCustomRoutingListenerCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("UpdateCustomRoutingListener");
  let body;
  body = JSON.stringify((0, import_smithy_client._json)(input));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_UpdateCustomRoutingListenerCommand");
var se_UpdateEndpointGroupCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("UpdateEndpointGroup");
  let body;
  body = JSON.stringify(se_UpdateEndpointGroupRequest(input, context));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_UpdateEndpointGroupCommand");
var se_UpdateListenerCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("UpdateListener");
  let body;
  body = JSON.stringify((0, import_smithy_client._json)(input));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_UpdateListenerCommand");
var se_WithdrawByoipCidrCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("WithdrawByoipCidr");
  let body;
  body = JSON.stringify((0, import_smithy_client._json)(input));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_WithdrawByoipCidrCommand");
var de_AddCustomRoutingEndpointsCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const data = await (0, import_core2.parseJsonBody)(output.body, context);
  let contents = {};
  contents = (0, import_smithy_client._json)(data);
  const response = {
    $metadata: deserializeMetadata(output),
    ...contents
  };
  return response;
}, "de_AddCustomRoutingEndpointsCommand");
var de_AddEndpointsCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const data = await (0, import_core2.parseJsonBody)(output.body, context);
  let contents = {};
  contents = (0, import_smithy_client._json)(data);
  const response = {
    $metadata: deserializeMetadata(output),
    ...contents
  };
  return response;
}, "de_AddEndpointsCommand");
var de_AdvertiseByoipCidrCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const data = await (0, import_core2.parseJsonBody)(output.body, context);
  let contents = {};
  contents = de_AdvertiseByoipCidrResponse(data, context);
  const response = {
    $metadata: deserializeMetadata(output),
    ...contents
  };
  return response;
}, "de_AdvertiseByoipCidrCommand");
var de_AllowCustomRoutingTrafficCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  await (0, import_smithy_client.collectBody)(output.body, context);
  const response = {
    $metadata: deserializeMetadata(output)
  };
  return response;
}, "de_AllowCustomRoutingTrafficCommand");
var de_CreateAcceleratorCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const data = await (0, import_core2.parseJsonBody)(output.body, context);
  let contents = {};
  contents = de_CreateAcceleratorResponse(data, context);
  const response = {
    $metadata: deserializeMetadata(output),
    ...contents
  };
  return response;
}, "de_CreateAcceleratorCommand");
var de_CreateCrossAccountAttachmentCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const data = await (0, import_core2.parseJsonBody)(output.body, context);
  let contents = {};
  contents = de_CreateCrossAccountAttachmentResponse(data, context);
  const response = {
    $metadata: deserializeMetadata(output),
    ...contents
  };
  return response;
}, "de_CreateCrossAccountAttachmentCommand");
var de_CreateCustomRoutingAcceleratorCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const data = await (0, import_core2.parseJsonBody)(output.body, context);
  let contents = {};
  contents = de_CreateCustomRoutingAcceleratorResponse(data, context);
  const response = {
    $metadata: deserializeMetadata(output),
    ...contents
  };
  return response;
}, "de_CreateCustomRoutingAcceleratorCommand");
var de_CreateCustomRoutingEndpointGroupCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const data = await (0, import_core2.parseJsonBody)(output.body, context);
  let contents = {};
  contents = (0, import_smithy_client._json)(data);
  const response = {
    $metadata: deserializeMetadata(output),
    ...contents
  };
  return response;
}, "de_CreateCustomRoutingEndpointGroupCommand");
var de_CreateCustomRoutingListenerCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const data = await (0, import_core2.parseJsonBody)(output.body, context);
  let contents = {};
  contents = (0, import_smithy_client._json)(data);
  const response = {
    $metadata: deserializeMetadata(output),
    ...contents
  };
  return response;
}, "de_CreateCustomRoutingListenerCommand");
var de_CreateEndpointGroupCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const data = await (0, import_core2.parseJsonBody)(output.body, context);
  let contents = {};
  contents = de_CreateEndpointGroupResponse(data, context);
  const response = {
    $metadata: deserializeMetadata(output),
    ...contents
  };
  return response;
}, "de_CreateEndpointGroupCommand");
var de_CreateListenerCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const data = await (0, import_core2.parseJsonBody)(output.body, context);
  let contents = {};
  contents = (0, import_smithy_client._json)(data);
  const response = {
    $metadata: deserializeMetadata(output),
    ...contents
  };
  return response;
}, "de_CreateListenerCommand");
var de_DeleteAcceleratorCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  await (0, import_smithy_client.collectBody)(output.body, context);
  const response = {
    $metadata: deserializeMetadata(output)
  };
  return response;
}, "de_DeleteAcceleratorCommand");
var de_DeleteCrossAccountAttachmentCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  await (0, import_smithy_client.collectBody)(output.body, context);
  const response = {
    $metadata: deserializeMetadata(output)
  };
  return response;
}, "de_DeleteCrossAccountAttachmentCommand");
var de_DeleteCustomRoutingAcceleratorCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  await (0, import_smithy_client.collectBody)(output.body, context);
  const response = {
    $metadata: deserializeMetadata(output)
  };
  return response;
}, "de_DeleteCustomRoutingAcceleratorCommand");
var de_DeleteCustomRoutingEndpointGroupCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  await (0, import_smithy_client.collectBody)(output.body, context);
  const response = {
    $metadata: deserializeMetadata(output)
  };
  return response;
}, "de_DeleteCustomRoutingEndpointGroupCommand");
var de_DeleteCustomRoutingListenerCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  await (0, import_smithy_client.collectBody)(output.body, context);
  const response = {
    $metadata: deserializeMetadata(output)
  };
  return response;
}, "de_DeleteCustomRoutingListenerCommand");
var de_DeleteEndpointGroupCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  await (0, import_smithy_client.collectBody)(output.body, context);
  const response = {
    $metadata: deserializeMetadata(output)
  };
  return response;
}, "de_DeleteEndpointGroupCommand");
var de_DeleteListenerCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  await (0, import_smithy_client.collectBody)(output.body, context);
  const response = {
    $metadata: deserializeMetadata(output)
  };
  return response;
}, "de_DeleteListenerCommand");
var de_DenyCustomRoutingTrafficCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  await (0, import_smithy_client.collectBody)(output.body, context);
  const response = {
    $metadata: deserializeMetadata(output)
  };
  return response;
}, "de_DenyCustomRoutingTrafficCommand");
var de_DeprovisionByoipCidrCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const data = await (0, import_core2.parseJsonBody)(output.body, context);
  let contents = {};
  contents = de_DeprovisionByoipCidrResponse(data, context);
  const response = {
    $metadata: deserializeMetadata(output),
    ...contents
  };
  return response;
}, "de_DeprovisionByoipCidrCommand");
var de_DescribeAcceleratorCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const data = await (0, import_core2.parseJsonBody)(output.body, context);
  let contents = {};
  contents = de_DescribeAcceleratorResponse(data, context);
  const response = {
    $metadata: deserializeMetadata(output),
    ...contents
  };
  return response;
}, "de_DescribeAcceleratorCommand");
var de_DescribeAcceleratorAttributesCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const data = await (0, import_core2.parseJsonBody)(output.body, context);
  let contents = {};
  contents = (0, import_smithy_client._json)(data);
  const response = {
    $metadata: deserializeMetadata(output),
    ...contents
  };
  return response;
}, "de_DescribeAcceleratorAttributesCommand");
var de_DescribeCrossAccountAttachmentCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const data = await (0, import_core2.parseJsonBody)(output.body, context);
  let contents = {};
  contents = de_DescribeCrossAccountAttachmentResponse(data, context);
  const response = {
    $metadata: deserializeMetadata(output),
    ...contents
  };
  return response;
}, "de_DescribeCrossAccountAttachmentCommand");
var de_DescribeCustomRoutingAcceleratorCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const data = await (0, import_core2.parseJsonBody)(output.body, context);
  let contents = {};
  contents = de_DescribeCustomRoutingAcceleratorResponse(data, context);
  const response = {
    $metadata: deserializeMetadata(output),
    ...contents
  };
  return response;
}, "de_DescribeCustomRoutingAcceleratorCommand");
var de_DescribeCustomRoutingAcceleratorAttributesCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const data = await (0, import_core2.parseJsonBody)(output.body, context);
  let contents = {};
  contents = (0, import_smithy_client._json)(data);
  const response = {
    $metadata: deserializeMetadata(output),
    ...contents
  };
  return response;
}, "de_DescribeCustomRoutingAcceleratorAttributesCommand");
var de_DescribeCustomRoutingEndpointGroupCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const data = await (0, import_core2.parseJsonBody)(output.body, context);
  let contents = {};
  contents = (0, import_smithy_client._json)(data);
  const response = {
    $metadata: deserializeMetadata(output),
    ...contents
  };
  return response;
}, "de_DescribeCustomRoutingEndpointGroupCommand");
var de_DescribeCustomRoutingListenerCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const data = await (0, import_core2.parseJsonBody)(output.body, context);
  let contents = {};
  contents = (0, import_smithy_client._json)(data);
  const response = {
    $metadata: deserializeMetadata(output),
    ...contents
  };
  return response;
}, "de_DescribeCustomRoutingListenerCommand");
var de_DescribeEndpointGroupCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const data = await (0, import_core2.parseJsonBody)(output.body, context);
  let contents = {};
  contents = de_DescribeEndpointGroupResponse(data, context);
  const response = {
    $metadata: deserializeMetadata(output),
    ...contents
  };
  return response;
}, "de_DescribeEndpointGroupCommand");
var de_DescribeListenerCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const data = await (0, import_core2.parseJsonBody)(output.body, context);
  let contents = {};
  contents = (0, import_smithy_client._json)(data);
  const response = {
    $metadata: deserializeMetadata(output),
    ...contents
  };
  return response;
}, "de_DescribeListenerCommand");
var de_ListAcceleratorsCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const data = await (0, import_core2.parseJsonBody)(output.body, context);
  let contents = {};
  contents = de_ListAcceleratorsResponse(data, context);
  const response = {
    $metadata: deserializeMetadata(output),
    ...contents
  };
  return response;
}, "de_ListAcceleratorsCommand");
var de_ListByoipCidrsCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const data = await (0, import_core2.parseJsonBody)(output.body, context);
  let contents = {};
  contents = de_ListByoipCidrsResponse(data, context);
  const response = {
    $metadata: deserializeMetadata(output),
    ...contents
  };
  return response;
}, "de_ListByoipCidrsCommand");
var de_ListCrossAccountAttachmentsCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const data = await (0, import_core2.parseJsonBody)(output.body, context);
  let contents = {};
  contents = de_ListCrossAccountAttachmentsResponse(data, context);
  const response = {
    $metadata: deserializeMetadata(output),
    ...contents
  };
  return response;
}, "de_ListCrossAccountAttachmentsCommand");
var de_ListCrossAccountResourceAccountsCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const data = await (0, import_core2.parseJsonBody)(output.body, context);
  let contents = {};
  contents = (0, import_smithy_client._json)(data);
  const response = {
    $metadata: deserializeMetadata(output),
    ...contents
  };
  return response;
}, "de_ListCrossAccountResourceAccountsCommand");
var de_ListCrossAccountResourcesCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const data = await (0, import_core2.parseJsonBody)(output.body, context);
  let contents = {};
  contents = (0, import_smithy_client._json)(data);
  const response = {
    $metadata: deserializeMetadata(output),
    ...contents
  };
  return response;
}, "de_ListCrossAccountResourcesCommand");
var de_ListCustomRoutingAcceleratorsCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const data = await (0, import_core2.parseJsonBody)(output.body, context);
  let contents = {};
  contents = de_ListCustomRoutingAcceleratorsResponse(data, context);
  const response = {
    $metadata: deserializeMetadata(output),
    ...contents
  };
  return response;
}, "de_ListCustomRoutingAcceleratorsCommand");
var de_ListCustomRoutingEndpointGroupsCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const data = await (0, import_core2.parseJsonBody)(output.body, context);
  let contents = {};
  contents = (0, import_smithy_client._json)(data);
  const response = {
    $metadata: deserializeMetadata(output),
    ...contents
  };
  return response;
}, "de_ListCustomRoutingEndpointGroupsCommand");
var de_ListCustomRoutingListenersCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const data = await (0, import_core2.parseJsonBody)(output.body, context);
  let contents = {};
  contents = (0, import_smithy_client._json)(data);
  const response = {
    $metadata: deserializeMetadata(output),
    ...contents
  };
  return response;
}, "de_ListCustomRoutingListenersCommand");
var de_ListCustomRoutingPortMappingsCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const data = await (0, import_core2.parseJsonBody)(output.body, context);
  let contents = {};
  contents = (0, import_smithy_client._json)(data);
  const response = {
    $metadata: deserializeMetadata(output),
    ...contents
  };
  return response;
}, "de_ListCustomRoutingPortMappingsCommand");
var de_ListCustomRoutingPortMappingsByDestinationCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const data = await (0, import_core2.parseJsonBody)(output.body, context);
  let contents = {};
  contents = (0, import_smithy_client._json)(data);
  const response = {
    $metadata: deserializeMetadata(output),
    ...contents
  };
  return response;
}, "de_ListCustomRoutingPortMappingsByDestinationCommand");
var de_ListEndpointGroupsCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const data = await (0, import_core2.parseJsonBody)(output.body, context);
  let contents = {};
  contents = de_ListEndpointGroupsResponse(data, context);
  const response = {
    $metadata: deserializeMetadata(output),
    ...contents
  };
  return response;
}, "de_ListEndpointGroupsCommand");
var de_ListListenersCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const data = await (0, import_core2.parseJsonBody)(output.body, context);
  let contents = {};
  contents = (0, import_smithy_client._json)(data);
  const response = {
    $metadata: deserializeMetadata(output),
    ...contents
  };
  return response;
}, "de_ListListenersCommand");
var de_ListTagsForResourceCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const data = await (0, import_core2.parseJsonBody)(output.body, context);
  let contents = {};
  contents = (0, import_smithy_client._json)(data);
  const response = {
    $metadata: deserializeMetadata(output),
    ...contents
  };
  return response;
}, "de_ListTagsForResourceCommand");
var de_ProvisionByoipCidrCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const data = await (0, import_core2.parseJsonBody)(output.body, context);
  let contents = {};
  contents = de_ProvisionByoipCidrResponse(data, context);
  const response = {
    $metadata: deserializeMetadata(output),
    ...contents
  };
  return response;
}, "de_ProvisionByoipCidrCommand");
var de_RemoveCustomRoutingEndpointsCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  await (0, import_smithy_client.collectBody)(output.body, context);
  const response = {
    $metadata: deserializeMetadata(output)
  };
  return response;
}, "de_RemoveCustomRoutingEndpointsCommand");
var de_RemoveEndpointsCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  await (0, import_smithy_client.collectBody)(output.body, context);
  const response = {
    $metadata: deserializeMetadata(output)
  };
  return response;
}, "de_RemoveEndpointsCommand");
var de_TagResourceCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const data = await (0, import_core2.parseJsonBody)(output.body, context);
  let contents = {};
  contents = (0, import_smithy_client._json)(data);
  const response = {
    $metadata: deserializeMetadata(output),
    ...contents
  };
  return response;
}, "de_TagResourceCommand");
var de_UntagResourceCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const data = await (0, import_core2.parseJsonBody)(output.body, context);
  let contents = {};
  contents = (0, import_smithy_client._json)(data);
  const response = {
    $metadata: deserializeMetadata(output),
    ...contents
  };
  return response;
}, "de_UntagResourceCommand");
var de_UpdateAcceleratorCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const data = await (0, import_core2.parseJsonBody)(output.body, context);
  let contents = {};
  contents = de_UpdateAcceleratorResponse(data, context);
  const response = {
    $metadata: deserializeMetadata(output),
    ...contents
  };
  return response;
}, "de_UpdateAcceleratorCommand");
var de_UpdateAcceleratorAttributesCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const data = await (0, import_core2.parseJsonBody)(output.body, context);
  let contents = {};
  contents = (0, import_smithy_client._json)(data);
  const response = {
    $metadata: deserializeMetadata(output),
    ...contents
  };
  return response;
}, "de_UpdateAcceleratorAttributesCommand");
var de_UpdateCrossAccountAttachmentCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const data = await (0, import_core2.parseJsonBody)(output.body, context);
  let contents = {};
  contents = de_UpdateCrossAccountAttachmentResponse(data, context);
  const response = {
    $metadata: deserializeMetadata(output),
    ...contents
  };
  return response;
}, "de_UpdateCrossAccountAttachmentCommand");
var de_UpdateCustomRoutingAcceleratorCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const data = await (0, import_core2.parseJsonBody)(output.body, context);
  let contents = {};
  contents = de_UpdateCustomRoutingAcceleratorResponse(data, context);
  const response = {
    $metadata: deserializeMetadata(output),
    ...contents
  };
  return response;
}, "de_UpdateCustomRoutingAcceleratorCommand");
var de_UpdateCustomRoutingAcceleratorAttributesCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const data = await (0, import_core2.parseJsonBody)(output.body, context);
  let contents = {};
  contents = (0, import_smithy_client._json)(data);
  const response = {
    $metadata: deserializeMetadata(output),
    ...contents
  };
  return response;
}, "de_UpdateCustomRoutingAcceleratorAttributesCommand");
var de_UpdateCustomRoutingListenerCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const data = await (0, import_core2.parseJsonBody)(output.body, context);
  let contents = {};
  contents = (0, import_smithy_client._json)(data);
  const response = {
    $metadata: deserializeMetadata(output),
    ...contents
  };
  return response;
}, "de_UpdateCustomRoutingListenerCommand");
var de_UpdateEndpointGroupCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const data = await (0, import_core2.parseJsonBody)(output.body, context);
  let contents = {};
  contents = de_UpdateEndpointGroupResponse(data, context);
  const response = {
    $metadata: deserializeMetadata(output),
    ...contents
  };
  return response;
}, "de_UpdateEndpointGroupCommand");
var de_UpdateListenerCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const data = await (0, import_core2.parseJsonBody)(output.body, context);
  let contents = {};
  contents = (0, import_smithy_client._json)(data);
  const response = {
    $metadata: deserializeMetadata(output),
    ...contents
  };
  return response;
}, "de_UpdateListenerCommand");
var de_WithdrawByoipCidrCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const data = await (0, import_core2.parseJsonBody)(output.body, context);
  let contents = {};
  contents = de_WithdrawByoipCidrResponse(data, context);
  const response = {
    $metadata: deserializeMetadata(output),
    ...contents
  };
  return response;
}, "de_WithdrawByoipCidrCommand");
var de_CommandError = /* @__PURE__ */ __name(async (output, context) => {
  const parsedOutput = {
    ...output,
    body: await (0, import_core2.parseJsonErrorBody)(output.body, context)
  };
  const errorCode = (0, import_core2.loadRestJsonErrorCode)(output, parsedOutput.body);
  switch (errorCode) {
    case "AccessDeniedException":
    case "com.amazonaws.globalaccelerator#AccessDeniedException":
      throw await de_AccessDeniedExceptionRes(parsedOutput, context);
    case "ConflictException":
    case "com.amazonaws.globalaccelerator#ConflictException":
      throw await de_ConflictExceptionRes(parsedOutput, context);
    case "EndpointAlreadyExistsException":
    case "com.amazonaws.globalaccelerator#EndpointAlreadyExistsException":
      throw await de_EndpointAlreadyExistsExceptionRes(parsedOutput, context);
    case "EndpointGroupNotFoundException":
    case "com.amazonaws.globalaccelerator#EndpointGroupNotFoundException":
      throw await de_EndpointGroupNotFoundExceptionRes(parsedOutput, context);
    case "InternalServiceErrorException":
    case "com.amazonaws.globalaccelerator#InternalServiceErrorException":
      throw await de_InternalServiceErrorExceptionRes(parsedOutput, context);
    case "InvalidArgumentException":
    case "com.amazonaws.globalaccelerator#InvalidArgumentException":
      throw await de_InvalidArgumentExceptionRes(parsedOutput, context);
    case "LimitExceededException":
    case "com.amazonaws.globalaccelerator#LimitExceededException":
      throw await de_LimitExceededExceptionRes(parsedOutput, context);
    case "TransactionInProgressException":
    case "com.amazonaws.globalaccelerator#TransactionInProgressException":
      throw await de_TransactionInProgressExceptionRes(parsedOutput, context);
    case "ByoipCidrNotFoundException":
    case "com.amazonaws.globalaccelerator#ByoipCidrNotFoundException":
      throw await de_ByoipCidrNotFoundExceptionRes(parsedOutput, context);
    case "IncorrectCidrStateException":
    case "com.amazonaws.globalaccelerator#IncorrectCidrStateException":
      throw await de_IncorrectCidrStateExceptionRes(parsedOutput, context);
    case "AcceleratorNotFoundException":
    case "com.amazonaws.globalaccelerator#AcceleratorNotFoundException":
      throw await de_AcceleratorNotFoundExceptionRes(parsedOutput, context);
    case "EndpointGroupAlreadyExistsException":
    case "com.amazonaws.globalaccelerator#EndpointGroupAlreadyExistsException":
      throw await de_EndpointGroupAlreadyExistsExceptionRes(parsedOutput, context);
    case "InvalidPortRangeException":
    case "com.amazonaws.globalaccelerator#InvalidPortRangeException":
      throw await de_InvalidPortRangeExceptionRes(parsedOutput, context);
    case "ListenerNotFoundException":
    case "com.amazonaws.globalaccelerator#ListenerNotFoundException":
      throw await de_ListenerNotFoundExceptionRes(parsedOutput, context);
    case "AcceleratorNotDisabledException":
    case "com.amazonaws.globalaccelerator#AcceleratorNotDisabledException":
      throw await de_AcceleratorNotDisabledExceptionRes(parsedOutput, context);
    case "AssociatedListenerFoundException":
    case "com.amazonaws.globalaccelerator#AssociatedListenerFoundException":
      throw await de_AssociatedListenerFoundExceptionRes(parsedOutput, context);
    case "AttachmentNotFoundException":
    case "com.amazonaws.globalaccelerator#AttachmentNotFoundException":
      throw await de_AttachmentNotFoundExceptionRes(parsedOutput, context);
    case "AssociatedEndpointGroupFoundException":
    case "com.amazonaws.globalaccelerator#AssociatedEndpointGroupFoundException":
      throw await de_AssociatedEndpointGroupFoundExceptionRes(parsedOutput, context);
    case "InvalidNextTokenException":
    case "com.amazonaws.globalaccelerator#InvalidNextTokenException":
      throw await de_InvalidNextTokenExceptionRes(parsedOutput, context);
    case "EndpointNotFoundException":
    case "com.amazonaws.globalaccelerator#EndpointNotFoundException":
      throw await de_EndpointNotFoundExceptionRes(parsedOutput, context);
    default:
      const parsedBody = parsedOutput.body;
      return throwDefaultError({
        output,
        parsedBody,
        errorCode
      });
  }
}, "de_CommandError");
var de_AcceleratorNotDisabledExceptionRes = /* @__PURE__ */ __name(async (parsedOutput, context) => {
  const body = parsedOutput.body;
  const deserialized = (0, import_smithy_client._json)(body);
  const exception = new AcceleratorNotDisabledException({
    $metadata: deserializeMetadata(parsedOutput),
    ...deserialized
  });
  return (0, import_smithy_client.decorateServiceException)(exception, body);
}, "de_AcceleratorNotDisabledExceptionRes");
var de_AcceleratorNotFoundExceptionRes = /* @__PURE__ */ __name(async (parsedOutput, context) => {
  const body = parsedOutput.body;
  const deserialized = (0, import_smithy_client._json)(body);
  const exception = new AcceleratorNotFoundException({
    $metadata: deserializeMetadata(parsedOutput),
    ...deserialized
  });
  return (0, import_smithy_client.decorateServiceException)(exception, body);
}, "de_AcceleratorNotFoundExceptionRes");
var de_AccessDeniedExceptionRes = /* @__PURE__ */ __name(async (parsedOutput, context) => {
  const body = parsedOutput.body;
  const deserialized = (0, import_smithy_client._json)(body);
  const exception = new AccessDeniedException({
    $metadata: deserializeMetadata(parsedOutput),
    ...deserialized
  });
  return (0, import_smithy_client.decorateServiceException)(exception, body);
}, "de_AccessDeniedExceptionRes");
var de_AssociatedEndpointGroupFoundExceptionRes = /* @__PURE__ */ __name(async (parsedOutput, context) => {
  const body = parsedOutput.body;
  const deserialized = (0, import_smithy_client._json)(body);
  const exception = new AssociatedEndpointGroupFoundException({
    $metadata: deserializeMetadata(parsedOutput),
    ...deserialized
  });
  return (0, import_smithy_client.decorateServiceException)(exception, body);
}, "de_AssociatedEndpointGroupFoundExceptionRes");
var de_AssociatedListenerFoundExceptionRes = /* @__PURE__ */ __name(async (parsedOutput, context) => {
  const body = parsedOutput.body;
  const deserialized = (0, import_smithy_client._json)(body);
  const exception = new AssociatedListenerFoundException({
    $metadata: deserializeMetadata(parsedOutput),
    ...deserialized
  });
  return (0, import_smithy_client.decorateServiceException)(exception, body);
}, "de_AssociatedListenerFoundExceptionRes");
var de_AttachmentNotFoundExceptionRes = /* @__PURE__ */ __name(async (parsedOutput, context) => {
  const body = parsedOutput.body;
  const deserialized = (0, import_smithy_client._json)(body);
  const exception = new AttachmentNotFoundException({
    $metadata: deserializeMetadata(parsedOutput),
    ...deserialized
  });
  return (0, import_smithy_client.decorateServiceException)(exception, body);
}, "de_AttachmentNotFoundExceptionRes");
var de_ByoipCidrNotFoundExceptionRes = /* @__PURE__ */ __name(async (parsedOutput, context) => {
  const body = parsedOutput.body;
  const deserialized = (0, import_smithy_client._json)(body);
  const exception = new ByoipCidrNotFoundException({
    $metadata: deserializeMetadata(parsedOutput),
    ...deserialized
  });
  return (0, import_smithy_client.decorateServiceException)(exception, body);
}, "de_ByoipCidrNotFoundExceptionRes");
var de_ConflictExceptionRes = /* @__PURE__ */ __name(async (parsedOutput, context) => {
  const body = parsedOutput.body;
  const deserialized = (0, import_smithy_client._json)(body);
  const exception = new ConflictException({
    $metadata: deserializeMetadata(parsedOutput),
    ...deserialized
  });
  return (0, import_smithy_client.decorateServiceException)(exception, body);
}, "de_ConflictExceptionRes");
var de_EndpointAlreadyExistsExceptionRes = /* @__PURE__ */ __name(async (parsedOutput, context) => {
  const body = parsedOutput.body;
  const deserialized = (0, import_smithy_client._json)(body);
  const exception = new EndpointAlreadyExistsException({
    $metadata: deserializeMetadata(parsedOutput),
    ...deserialized
  });
  return (0, import_smithy_client.decorateServiceException)(exception, body);
}, "de_EndpointAlreadyExistsExceptionRes");
var de_EndpointGroupAlreadyExistsExceptionRes = /* @__PURE__ */ __name(async (parsedOutput, context) => {
  const body = parsedOutput.body;
  const deserialized = (0, import_smithy_client._json)(body);
  const exception = new EndpointGroupAlreadyExistsException({
    $metadata: deserializeMetadata(parsedOutput),
    ...deserialized
  });
  return (0, import_smithy_client.decorateServiceException)(exception, body);
}, "de_EndpointGroupAlreadyExistsExceptionRes");
var de_EndpointGroupNotFoundExceptionRes = /* @__PURE__ */ __name(async (parsedOutput, context) => {
  const body = parsedOutput.body;
  const deserialized = (0, import_smithy_client._json)(body);
  const exception = new EndpointGroupNotFoundException({
    $metadata: deserializeMetadata(parsedOutput),
    ...deserialized
  });
  return (0, import_smithy_client.decorateServiceException)(exception, body);
}, "de_EndpointGroupNotFoundExceptionRes");
var de_EndpointNotFoundExceptionRes = /* @__PURE__ */ __name(async (parsedOutput, context) => {
  const body = parsedOutput.body;
  const deserialized = (0, import_smithy_client._json)(body);
  const exception = new EndpointNotFoundException({
    $metadata: deserializeMetadata(parsedOutput),
    ...deserialized
  });
  return (0, import_smithy_client.decorateServiceException)(exception, body);
}, "de_EndpointNotFoundExceptionRes");
var de_IncorrectCidrStateExceptionRes = /* @__PURE__ */ __name(async (parsedOutput, context) => {
  const body = parsedOutput.body;
  const deserialized = (0, import_smithy_client._json)(body);
  const exception = new IncorrectCidrStateException({
    $metadata: deserializeMetadata(parsedOutput),
    ...deserialized
  });
  return (0, import_smithy_client.decorateServiceException)(exception, body);
}, "de_IncorrectCidrStateExceptionRes");
var de_InternalServiceErrorExceptionRes = /* @__PURE__ */ __name(async (parsedOutput, context) => {
  const body = parsedOutput.body;
  const deserialized = (0, import_smithy_client._json)(body);
  const exception = new InternalServiceErrorException({
    $metadata: deserializeMetadata(parsedOutput),
    ...deserialized
  });
  return (0, import_smithy_client.decorateServiceException)(exception, body);
}, "de_InternalServiceErrorExceptionRes");
var de_InvalidArgumentExceptionRes = /* @__PURE__ */ __name(async (parsedOutput, context) => {
  const body = parsedOutput.body;
  const deserialized = (0, import_smithy_client._json)(body);
  const exception = new InvalidArgumentException({
    $metadata: deserializeMetadata(parsedOutput),
    ...deserialized
  });
  return (0, import_smithy_client.decorateServiceException)(exception, body);
}, "de_InvalidArgumentExceptionRes");
var de_InvalidNextTokenExceptionRes = /* @__PURE__ */ __name(async (parsedOutput, context) => {
  const body = parsedOutput.body;
  const deserialized = (0, import_smithy_client._json)(body);
  const exception = new InvalidNextTokenException({
    $metadata: deserializeMetadata(parsedOutput),
    ...deserialized
  });
  return (0, import_smithy_client.decorateServiceException)(exception, body);
}, "de_InvalidNextTokenExceptionRes");
var de_InvalidPortRangeExceptionRes = /* @__PURE__ */ __name(async (parsedOutput, context) => {
  const body = parsedOutput.body;
  const deserialized = (0, import_smithy_client._json)(body);
  const exception = new InvalidPortRangeException({
    $metadata: deserializeMetadata(parsedOutput),
    ...deserialized
  });
  return (0, import_smithy_client.decorateServiceException)(exception, body);
}, "de_InvalidPortRangeExceptionRes");
var de_LimitExceededExceptionRes = /* @__PURE__ */ __name(async (parsedOutput, context) => {
  const body = parsedOutput.body;
  const deserialized = (0, import_smithy_client._json)(body);
  const exception = new LimitExceededException({
    $metadata: deserializeMetadata(parsedOutput),
    ...deserialized
  });
  return (0, import_smithy_client.decorateServiceException)(exception, body);
}, "de_LimitExceededExceptionRes");
var de_ListenerNotFoundExceptionRes = /* @__PURE__ */ __name(async (parsedOutput, context) => {
  const body = parsedOutput.body;
  const deserialized = (0, import_smithy_client._json)(body);
  const exception = new ListenerNotFoundException({
    $metadata: deserializeMetadata(parsedOutput),
    ...deserialized
  });
  return (0, import_smithy_client.decorateServiceException)(exception, body);
}, "de_ListenerNotFoundExceptionRes");
var de_TransactionInProgressExceptionRes = /* @__PURE__ */ __name(async (parsedOutput, context) => {
  const body = parsedOutput.body;
  const deserialized = (0, import_smithy_client._json)(body);
  const exception = new TransactionInProgressException({
    $metadata: deserializeMetadata(parsedOutput),
    ...deserialized
  });
  return (0, import_smithy_client.decorateServiceException)(exception, body);
}, "de_TransactionInProgressExceptionRes");
var se_CreateAcceleratorRequest = /* @__PURE__ */ __name((input, context) => {
  return (0, import_smithy_client.take)(input, {
    Enabled: [],
    IdempotencyToken: [true, (_) => _ ?? (0, import_uuid.v4)()],
    IpAddressType: [],
    IpAddresses: import_smithy_client._json,
    Name: [],
    Tags: import_smithy_client._json
  });
}, "se_CreateAcceleratorRequest");
var se_CreateCrossAccountAttachmentRequest = /* @__PURE__ */ __name((input, context) => {
  return (0, import_smithy_client.take)(input, {
    IdempotencyToken: [true, (_) => _ ?? (0, import_uuid.v4)()],
    Name: [],
    Principals: import_smithy_client._json,
    Resources: import_smithy_client._json,
    Tags: import_smithy_client._json
  });
}, "se_CreateCrossAccountAttachmentRequest");
var se_CreateCustomRoutingAcceleratorRequest = /* @__PURE__ */ __name((input, context) => {
  return (0, import_smithy_client.take)(input, {
    Enabled: [],
    IdempotencyToken: [true, (_) => _ ?? (0, import_uuid.v4)()],
    IpAddressType: [],
    IpAddresses: import_smithy_client._json,
    Name: [],
    Tags: import_smithy_client._json
  });
}, "se_CreateCustomRoutingAcceleratorRequest");
var se_CreateCustomRoutingEndpointGroupRequest = /* @__PURE__ */ __name((input, context) => {
  return (0, import_smithy_client.take)(input, {
    DestinationConfigurations: import_smithy_client._json,
    EndpointGroupRegion: [],
    IdempotencyToken: [true, (_) => _ ?? (0, import_uuid.v4)()],
    ListenerArn: []
  });
}, "se_CreateCustomRoutingEndpointGroupRequest");
var se_CreateCustomRoutingListenerRequest = /* @__PURE__ */ __name((input, context) => {
  return (0, import_smithy_client.take)(input, {
    AcceleratorArn: [],
    IdempotencyToken: [true, (_) => _ ?? (0, import_uuid.v4)()],
    PortRanges: import_smithy_client._json
  });
}, "se_CreateCustomRoutingListenerRequest");
var se_CreateEndpointGroupRequest = /* @__PURE__ */ __name((input, context) => {
  return (0, import_smithy_client.take)(input, {
    EndpointConfigurations: import_smithy_client._json,
    EndpointGroupRegion: [],
    HealthCheckIntervalSeconds: [],
    HealthCheckPath: [],
    HealthCheckPort: [],
    HealthCheckProtocol: [],
    IdempotencyToken: [true, (_) => _ ?? (0, import_uuid.v4)()],
    ListenerArn: [],
    PortOverrides: import_smithy_client._json,
    ThresholdCount: [],
    TrafficDialPercentage: import_smithy_client.serializeFloat
  });
}, "se_CreateEndpointGroupRequest");
var se_CreateListenerRequest = /* @__PURE__ */ __name((input, context) => {
  return (0, import_smithy_client.take)(input, {
    AcceleratorArn: [],
    ClientAffinity: [],
    IdempotencyToken: [true, (_) => _ ?? (0, import_uuid.v4)()],
    PortRanges: import_smithy_client._json,
    Protocol: []
  });
}, "se_CreateListenerRequest");
var se_UpdateEndpointGroupRequest = /* @__PURE__ */ __name((input, context) => {
  return (0, import_smithy_client.take)(input, {
    EndpointConfigurations: import_smithy_client._json,
    EndpointGroupArn: [],
    HealthCheckIntervalSeconds: [],
    HealthCheckPath: [],
    HealthCheckPort: [],
    HealthCheckProtocol: [],
    PortOverrides: import_smithy_client._json,
    ThresholdCount: [],
    TrafficDialPercentage: import_smithy_client.serializeFloat
  });
}, "se_UpdateEndpointGroupRequest");
var de_Accelerator = /* @__PURE__ */ __name((output, context) => {
  return (0, import_smithy_client.take)(output, {
    AcceleratorArn: import_smithy_client.expectString,
    CreatedTime: (_) => (0, import_smithy_client.expectNonNull)((0, import_smithy_client.parseEpochTimestamp)((0, import_smithy_client.expectNumber)(_))),
    DnsName: import_smithy_client.expectString,
    DualStackDnsName: import_smithy_client.expectString,
    Enabled: import_smithy_client.expectBoolean,
    Events: (_) => de_AcceleratorEvents(_, context),
    IpAddressType: import_smithy_client.expectString,
    IpSets: import_smithy_client._json,
    LastModifiedTime: (_) => (0, import_smithy_client.expectNonNull)((0, import_smithy_client.parseEpochTimestamp)((0, import_smithy_client.expectNumber)(_))),
    Name: import_smithy_client.expectString,
    Status: import_smithy_client.expectString
  });
}, "de_Accelerator");
var de_AcceleratorEvent = /* @__PURE__ */ __name((output, context) => {
  return (0, import_smithy_client.take)(output, {
    Message: import_smithy_client.expectString,
    Timestamp: (_) => (0, import_smithy_client.expectNonNull)((0, import_smithy_client.parseEpochTimestamp)((0, import_smithy_client.expectNumber)(_)))
  });
}, "de_AcceleratorEvent");
var de_AcceleratorEvents = /* @__PURE__ */ __name((output, context) => {
  const retVal = (output || []).filter((e) => e != null).map((entry) => {
    return de_AcceleratorEvent(entry, context);
  });
  return retVal;
}, "de_AcceleratorEvents");
var de_Accelerators = /* @__PURE__ */ __name((output, context) => {
  const retVal = (output || []).filter((e) => e != null).map((entry) => {
    return de_Accelerator(entry, context);
  });
  return retVal;
}, "de_Accelerators");
var de_AdvertiseByoipCidrResponse = /* @__PURE__ */ __name((output, context) => {
  return (0, import_smithy_client.take)(output, {
    ByoipCidr: (_) => de_ByoipCidr(_, context)
  });
}, "de_AdvertiseByoipCidrResponse");
var de_Attachment = /* @__PURE__ */ __name((output, context) => {
  return (0, import_smithy_client.take)(output, {
    AttachmentArn: import_smithy_client.expectString,
    CreatedTime: (_) => (0, import_smithy_client.expectNonNull)((0, import_smithy_client.parseEpochTimestamp)((0, import_smithy_client.expectNumber)(_))),
    LastModifiedTime: (_) => (0, import_smithy_client.expectNonNull)((0, import_smithy_client.parseEpochTimestamp)((0, import_smithy_client.expectNumber)(_))),
    Name: import_smithy_client.expectString,
    Principals: import_smithy_client._json,
    Resources: import_smithy_client._json
  });
}, "de_Attachment");
var de_Attachments = /* @__PURE__ */ __name((output, context) => {
  const retVal = (output || []).filter((e) => e != null).map((entry) => {
    return de_Attachment(entry, context);
  });
  return retVal;
}, "de_Attachments");
var de_ByoipCidr = /* @__PURE__ */ __name((output, context) => {
  return (0, import_smithy_client.take)(output, {
    Cidr: import_smithy_client.expectString,
    Events: (_) => de_ByoipCidrEvents(_, context),
    State: import_smithy_client.expectString
  });
}, "de_ByoipCidr");
var de_ByoipCidrEvent = /* @__PURE__ */ __name((output, context) => {
  return (0, import_smithy_client.take)(output, {
    Message: import_smithy_client.expectString,
    Timestamp: (_) => (0, import_smithy_client.expectNonNull)((0, import_smithy_client.parseEpochTimestamp)((0, import_smithy_client.expectNumber)(_)))
  });
}, "de_ByoipCidrEvent");
var de_ByoipCidrEvents = /* @__PURE__ */ __name((output, context) => {
  const retVal = (output || []).filter((e) => e != null).map((entry) => {
    return de_ByoipCidrEvent(entry, context);
  });
  return retVal;
}, "de_ByoipCidrEvents");
var de_ByoipCidrs = /* @__PURE__ */ __name((output, context) => {
  const retVal = (output || []).filter((e) => e != null).map((entry) => {
    return de_ByoipCidr(entry, context);
  });
  return retVal;
}, "de_ByoipCidrs");
var de_CreateAcceleratorResponse = /* @__PURE__ */ __name((output, context) => {
  return (0, import_smithy_client.take)(output, {
    Accelerator: (_) => de_Accelerator(_, context)
  });
}, "de_CreateAcceleratorResponse");
var de_CreateCrossAccountAttachmentResponse = /* @__PURE__ */ __name((output, context) => {
  return (0, import_smithy_client.take)(output, {
    CrossAccountAttachment: (_) => de_Attachment(_, context)
  });
}, "de_CreateCrossAccountAttachmentResponse");
var de_CreateCustomRoutingAcceleratorResponse = /* @__PURE__ */ __name((output, context) => {
  return (0, import_smithy_client.take)(output, {
    Accelerator: (_) => de_CustomRoutingAccelerator(_, context)
  });
}, "de_CreateCustomRoutingAcceleratorResponse");
var de_CreateEndpointGroupResponse = /* @__PURE__ */ __name((output, context) => {
  return (0, import_smithy_client.take)(output, {
    EndpointGroup: (_) => de_EndpointGroup(_, context)
  });
}, "de_CreateEndpointGroupResponse");
var de_CustomRoutingAccelerator = /* @__PURE__ */ __name((output, context) => {
  return (0, import_smithy_client.take)(output, {
    AcceleratorArn: import_smithy_client.expectString,
    CreatedTime: (_) => (0, import_smithy_client.expectNonNull)((0, import_smithy_client.parseEpochTimestamp)((0, import_smithy_client.expectNumber)(_))),
    DnsName: import_smithy_client.expectString,
    Enabled: import_smithy_client.expectBoolean,
    IpAddressType: import_smithy_client.expectString,
    IpSets: import_smithy_client._json,
    LastModifiedTime: (_) => (0, import_smithy_client.expectNonNull)((0, import_smithy_client.parseEpochTimestamp)((0, import_smithy_client.expectNumber)(_))),
    Name: import_smithy_client.expectString,
    Status: import_smithy_client.expectString
  });
}, "de_CustomRoutingAccelerator");
var de_CustomRoutingAccelerators = /* @__PURE__ */ __name((output, context) => {
  const retVal = (output || []).filter((e) => e != null).map((entry) => {
    return de_CustomRoutingAccelerator(entry, context);
  });
  return retVal;
}, "de_CustomRoutingAccelerators");
var de_DeprovisionByoipCidrResponse = /* @__PURE__ */ __name((output, context) => {
  return (0, import_smithy_client.take)(output, {
    ByoipCidr: (_) => de_ByoipCidr(_, context)
  });
}, "de_DeprovisionByoipCidrResponse");
var de_DescribeAcceleratorResponse = /* @__PURE__ */ __name((output, context) => {
  return (0, import_smithy_client.take)(output, {
    Accelerator: (_) => de_Accelerator(_, context)
  });
}, "de_DescribeAcceleratorResponse");
var de_DescribeCrossAccountAttachmentResponse = /* @__PURE__ */ __name((output, context) => {
  return (0, import_smithy_client.take)(output, {
    CrossAccountAttachment: (_) => de_Attachment(_, context)
  });
}, "de_DescribeCrossAccountAttachmentResponse");
var de_DescribeCustomRoutingAcceleratorResponse = /* @__PURE__ */ __name((output, context) => {
  return (0, import_smithy_client.take)(output, {
    Accelerator: (_) => de_CustomRoutingAccelerator(_, context)
  });
}, "de_DescribeCustomRoutingAcceleratorResponse");
var de_DescribeEndpointGroupResponse = /* @__PURE__ */ __name((output, context) => {
  return (0, import_smithy_client.take)(output, {
    EndpointGroup: (_) => de_EndpointGroup(_, context)
  });
}, "de_DescribeEndpointGroupResponse");
var de_EndpointGroup = /* @__PURE__ */ __name((output, context) => {
  return (0, import_smithy_client.take)(output, {
    EndpointDescriptions: import_smithy_client._json,
    EndpointGroupArn: import_smithy_client.expectString,
    EndpointGroupRegion: import_smithy_client.expectString,
    HealthCheckIntervalSeconds: import_smithy_client.expectInt32,
    HealthCheckPath: import_smithy_client.expectString,
    HealthCheckPort: import_smithy_client.expectInt32,
    HealthCheckProtocol: import_smithy_client.expectString,
    PortOverrides: import_smithy_client._json,
    ThresholdCount: import_smithy_client.expectInt32,
    TrafficDialPercentage: import_smithy_client.limitedParseFloat32
  });
}, "de_EndpointGroup");
var de_EndpointGroups = /* @__PURE__ */ __name((output, context) => {
  const retVal = (output || []).filter((e) => e != null).map((entry) => {
    return de_EndpointGroup(entry, context);
  });
  return retVal;
}, "de_EndpointGroups");
var de_ListAcceleratorsResponse = /* @__PURE__ */ __name((output, context) => {
  return (0, import_smithy_client.take)(output, {
    Accelerators: (_) => de_Accelerators(_, context),
    NextToken: import_smithy_client.expectString
  });
}, "de_ListAcceleratorsResponse");
var de_ListByoipCidrsResponse = /* @__PURE__ */ __name((output, context) => {
  return (0, import_smithy_client.take)(output, {
    ByoipCidrs: (_) => de_ByoipCidrs(_, context),
    NextToken: import_smithy_client.expectString
  });
}, "de_ListByoipCidrsResponse");
var de_ListCrossAccountAttachmentsResponse = /* @__PURE__ */ __name((output, context) => {
  return (0, import_smithy_client.take)(output, {
    CrossAccountAttachments: (_) => de_Attachments(_, context),
    NextToken: import_smithy_client.expectString
  });
}, "de_ListCrossAccountAttachmentsResponse");
var de_ListCustomRoutingAcceleratorsResponse = /* @__PURE__ */ __name((output, context) => {
  return (0, import_smithy_client.take)(output, {
    Accelerators: (_) => de_CustomRoutingAccelerators(_, context),
    NextToken: import_smithy_client.expectString
  });
}, "de_ListCustomRoutingAcceleratorsResponse");
var de_ListEndpointGroupsResponse = /* @__PURE__ */ __name((output, context) => {
  return (0, import_smithy_client.take)(output, {
    EndpointGroups: (_) => de_EndpointGroups(_, context),
    NextToken: import_smithy_client.expectString
  });
}, "de_ListEndpointGroupsResponse");
var de_ProvisionByoipCidrResponse = /* @__PURE__ */ __name((output, context) => {
  return (0, import_smithy_client.take)(output, {
    ByoipCidr: (_) => de_ByoipCidr(_, context)
  });
}, "de_ProvisionByoipCidrResponse");
var de_UpdateAcceleratorResponse = /* @__PURE__ */ __name((output, context) => {
  return (0, import_smithy_client.take)(output, {
    Accelerator: (_) => de_Accelerator(_, context)
  });
}, "de_UpdateAcceleratorResponse");
var de_UpdateCrossAccountAttachmentResponse = /* @__PURE__ */ __name((output, context) => {
  return (0, import_smithy_client.take)(output, {
    CrossAccountAttachment: (_) => de_Attachment(_, context)
  });
}, "de_UpdateCrossAccountAttachmentResponse");
var de_UpdateCustomRoutingAcceleratorResponse = /* @__PURE__ */ __name((output, context) => {
  return (0, import_smithy_client.take)(output, {
    Accelerator: (_) => de_CustomRoutingAccelerator(_, context)
  });
}, "de_UpdateCustomRoutingAcceleratorResponse");
var de_UpdateEndpointGroupResponse = /* @__PURE__ */ __name((output, context) => {
  return (0, import_smithy_client.take)(output, {
    EndpointGroup: (_) => de_EndpointGroup(_, context)
  });
}, "de_UpdateEndpointGroupResponse");
var de_WithdrawByoipCidrResponse = /* @__PURE__ */ __name((output, context) => {
  return (0, import_smithy_client.take)(output, {
    ByoipCidr: (_) => de_ByoipCidr(_, context)
  });
}, "de_WithdrawByoipCidrResponse");
var deserializeMetadata = /* @__PURE__ */ __name((output) => ({
  httpStatusCode: output.statusCode,
  requestId: output.headers["x-amzn-requestid"] ?? output.headers["x-amzn-request-id"] ?? output.headers["x-amz-request-id"],
  extendedRequestId: output.headers["x-amz-id-2"],
  cfId: output.headers["x-amz-cf-id"]
}), "deserializeMetadata");
var throwDefaultError = (0, import_smithy_client.withBaseException)(GlobalAcceleratorServiceException);
var buildHttpRpcRequest = /* @__PURE__ */ __name(async (context, headers, path, resolvedHostname, body) => {
  const { hostname, protocol = "https", port, path: basePath } = await context.endpoint();
  const contents = {
    protocol,
    hostname,
    port,
    method: "POST",
    path: basePath.endsWith("/") ? basePath.slice(0, -1) + path : basePath + path,
    headers
  };
  if (resolvedHostname !== void 0) {
    contents.hostname = resolvedHostname;
  }
  if (body !== void 0) {
    contents.body = body;
  }
  return new import_protocol_http.HttpRequest(contents);
}, "buildHttpRpcRequest");
function sharedHeaders(operation) {
  return {
    "content-type": "application/x-amz-json-1.1",
    "x-amz-target": `GlobalAccelerator_V20180706.${operation}`
  };
}
__name(sharedHeaders, "sharedHeaders");

// src/commands/AddCustomRoutingEndpointsCommand.ts
var _AddCustomRoutingEndpointsCommand = class _AddCustomRoutingEndpointsCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("GlobalAccelerator_V20180706", "AddCustomRoutingEndpoints", {}).n("GlobalAcceleratorClient", "AddCustomRoutingEndpointsCommand").f(void 0, void 0).ser(se_AddCustomRoutingEndpointsCommand).de(de_AddCustomRoutingEndpointsCommand).build() {
};
__name(_AddCustomRoutingEndpointsCommand, "AddCustomRoutingEndpointsCommand");
var AddCustomRoutingEndpointsCommand = _AddCustomRoutingEndpointsCommand;

// src/commands/AddEndpointsCommand.ts




var _AddEndpointsCommand = class _AddEndpointsCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("GlobalAccelerator_V20180706", "AddEndpoints", {}).n("GlobalAcceleratorClient", "AddEndpointsCommand").f(void 0, void 0).ser(se_AddEndpointsCommand).de(de_AddEndpointsCommand).build() {
};
__name(_AddEndpointsCommand, "AddEndpointsCommand");
var AddEndpointsCommand = _AddEndpointsCommand;

// src/commands/AdvertiseByoipCidrCommand.ts




var _AdvertiseByoipCidrCommand = class _AdvertiseByoipCidrCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("GlobalAccelerator_V20180706", "AdvertiseByoipCidr", {}).n("GlobalAcceleratorClient", "AdvertiseByoipCidrCommand").f(void 0, void 0).ser(se_AdvertiseByoipCidrCommand).de(de_AdvertiseByoipCidrCommand).build() {
};
__name(_AdvertiseByoipCidrCommand, "AdvertiseByoipCidrCommand");
var AdvertiseByoipCidrCommand = _AdvertiseByoipCidrCommand;

// src/commands/AllowCustomRoutingTrafficCommand.ts




var _AllowCustomRoutingTrafficCommand = class _AllowCustomRoutingTrafficCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("GlobalAccelerator_V20180706", "AllowCustomRoutingTraffic", {}).n("GlobalAcceleratorClient", "AllowCustomRoutingTrafficCommand").f(void 0, void 0).ser(se_AllowCustomRoutingTrafficCommand).de(de_AllowCustomRoutingTrafficCommand).build() {
};
__name(_AllowCustomRoutingTrafficCommand, "AllowCustomRoutingTrafficCommand");
var AllowCustomRoutingTrafficCommand = _AllowCustomRoutingTrafficCommand;

// src/commands/CreateAcceleratorCommand.ts




var _CreateAcceleratorCommand = class _CreateAcceleratorCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("GlobalAccelerator_V20180706", "CreateAccelerator", {}).n("GlobalAcceleratorClient", "CreateAcceleratorCommand").f(void 0, void 0).ser(se_CreateAcceleratorCommand).de(de_CreateAcceleratorCommand).build() {
};
__name(_CreateAcceleratorCommand, "CreateAcceleratorCommand");
var CreateAcceleratorCommand = _CreateAcceleratorCommand;

// src/commands/CreateCrossAccountAttachmentCommand.ts




var _CreateCrossAccountAttachmentCommand = class _CreateCrossAccountAttachmentCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("GlobalAccelerator_V20180706", "CreateCrossAccountAttachment", {}).n("GlobalAcceleratorClient", "CreateCrossAccountAttachmentCommand").f(void 0, void 0).ser(se_CreateCrossAccountAttachmentCommand).de(de_CreateCrossAccountAttachmentCommand).build() {
};
__name(_CreateCrossAccountAttachmentCommand, "CreateCrossAccountAttachmentCommand");
var CreateCrossAccountAttachmentCommand = _CreateCrossAccountAttachmentCommand;

// src/commands/CreateCustomRoutingAcceleratorCommand.ts




var _CreateCustomRoutingAcceleratorCommand = class _CreateCustomRoutingAcceleratorCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("GlobalAccelerator_V20180706", "CreateCustomRoutingAccelerator", {}).n("GlobalAcceleratorClient", "CreateCustomRoutingAcceleratorCommand").f(void 0, void 0).ser(se_CreateCustomRoutingAcceleratorCommand).de(de_CreateCustomRoutingAcceleratorCommand).build() {
};
__name(_CreateCustomRoutingAcceleratorCommand, "CreateCustomRoutingAcceleratorCommand");
var CreateCustomRoutingAcceleratorCommand = _CreateCustomRoutingAcceleratorCommand;

// src/commands/CreateCustomRoutingEndpointGroupCommand.ts




var _CreateCustomRoutingEndpointGroupCommand = class _CreateCustomRoutingEndpointGroupCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("GlobalAccelerator_V20180706", "CreateCustomRoutingEndpointGroup", {}).n("GlobalAcceleratorClient", "CreateCustomRoutingEndpointGroupCommand").f(void 0, void 0).ser(se_CreateCustomRoutingEndpointGroupCommand).de(de_CreateCustomRoutingEndpointGroupCommand).build() {
};
__name(_CreateCustomRoutingEndpointGroupCommand, "CreateCustomRoutingEndpointGroupCommand");
var CreateCustomRoutingEndpointGroupCommand = _CreateCustomRoutingEndpointGroupCommand;

// src/commands/CreateCustomRoutingListenerCommand.ts




var _CreateCustomRoutingListenerCommand = class _CreateCustomRoutingListenerCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("GlobalAccelerator_V20180706", "CreateCustomRoutingListener", {}).n("GlobalAcceleratorClient", "CreateCustomRoutingListenerCommand").f(void 0, void 0).ser(se_CreateCustomRoutingListenerCommand).de(de_CreateCustomRoutingListenerCommand).build() {
};
__name(_CreateCustomRoutingListenerCommand, "CreateCustomRoutingListenerCommand");
var CreateCustomRoutingListenerCommand = _CreateCustomRoutingListenerCommand;

// src/commands/CreateEndpointGroupCommand.ts




var _CreateEndpointGroupCommand = class _CreateEndpointGroupCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("GlobalAccelerator_V20180706", "CreateEndpointGroup", {}).n("GlobalAcceleratorClient", "CreateEndpointGroupCommand").f(void 0, void 0).ser(se_CreateEndpointGroupCommand).de(de_CreateEndpointGroupCommand).build() {
};
__name(_CreateEndpointGroupCommand, "CreateEndpointGroupCommand");
var CreateEndpointGroupCommand = _CreateEndpointGroupCommand;

// src/commands/CreateListenerCommand.ts




var _CreateListenerCommand = class _CreateListenerCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("GlobalAccelerator_V20180706", "CreateListener", {}).n("GlobalAcceleratorClient", "CreateListenerCommand").f(void 0, void 0).ser(se_CreateListenerCommand).de(de_CreateListenerCommand).build() {
};
__name(_CreateListenerCommand, "CreateListenerCommand");
var CreateListenerCommand = _CreateListenerCommand;

// src/commands/DeleteAcceleratorCommand.ts




var _DeleteAcceleratorCommand = class _DeleteAcceleratorCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("GlobalAccelerator_V20180706", "DeleteAccelerator", {}).n("GlobalAcceleratorClient", "DeleteAcceleratorCommand").f(void 0, void 0).ser(se_DeleteAcceleratorCommand).de(de_DeleteAcceleratorCommand).build() {
};
__name(_DeleteAcceleratorCommand, "DeleteAcceleratorCommand");
var DeleteAcceleratorCommand = _DeleteAcceleratorCommand;

// src/commands/DeleteCrossAccountAttachmentCommand.ts




var _DeleteCrossAccountAttachmentCommand = class _DeleteCrossAccountAttachmentCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("GlobalAccelerator_V20180706", "DeleteCrossAccountAttachment", {}).n("GlobalAcceleratorClient", "DeleteCrossAccountAttachmentCommand").f(void 0, void 0).ser(se_DeleteCrossAccountAttachmentCommand).de(de_DeleteCrossAccountAttachmentCommand).build() {
};
__name(_DeleteCrossAccountAttachmentCommand, "DeleteCrossAccountAttachmentCommand");
var DeleteCrossAccountAttachmentCommand = _DeleteCrossAccountAttachmentCommand;

// src/commands/DeleteCustomRoutingAcceleratorCommand.ts




var _DeleteCustomRoutingAcceleratorCommand = class _DeleteCustomRoutingAcceleratorCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("GlobalAccelerator_V20180706", "DeleteCustomRoutingAccelerator", {}).n("GlobalAcceleratorClient", "DeleteCustomRoutingAcceleratorCommand").f(void 0, void 0).ser(se_DeleteCustomRoutingAcceleratorCommand).de(de_DeleteCustomRoutingAcceleratorCommand).build() {
};
__name(_DeleteCustomRoutingAcceleratorCommand, "DeleteCustomRoutingAcceleratorCommand");
var DeleteCustomRoutingAcceleratorCommand = _DeleteCustomRoutingAcceleratorCommand;

// src/commands/DeleteCustomRoutingEndpointGroupCommand.ts




var _DeleteCustomRoutingEndpointGroupCommand = class _DeleteCustomRoutingEndpointGroupCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("GlobalAccelerator_V20180706", "DeleteCustomRoutingEndpointGroup", {}).n("GlobalAcceleratorClient", "DeleteCustomRoutingEndpointGroupCommand").f(void 0, void 0).ser(se_DeleteCustomRoutingEndpointGroupCommand).de(de_DeleteCustomRoutingEndpointGroupCommand).build() {
};
__name(_DeleteCustomRoutingEndpointGroupCommand, "DeleteCustomRoutingEndpointGroupCommand");
var DeleteCustomRoutingEndpointGroupCommand = _DeleteCustomRoutingEndpointGroupCommand;

// src/commands/DeleteCustomRoutingListenerCommand.ts




var _DeleteCustomRoutingListenerCommand = class _DeleteCustomRoutingListenerCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("GlobalAccelerator_V20180706", "DeleteCustomRoutingListener", {}).n("GlobalAcceleratorClient", "DeleteCustomRoutingListenerCommand").f(void 0, void 0).ser(se_DeleteCustomRoutingListenerCommand).de(de_DeleteCustomRoutingListenerCommand).build() {
};
__name(_DeleteCustomRoutingListenerCommand, "DeleteCustomRoutingListenerCommand");
var DeleteCustomRoutingListenerCommand = _DeleteCustomRoutingListenerCommand;

// src/commands/DeleteEndpointGroupCommand.ts




var _DeleteEndpointGroupCommand = class _DeleteEndpointGroupCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("GlobalAccelerator_V20180706", "DeleteEndpointGroup", {}).n("GlobalAcceleratorClient", "DeleteEndpointGroupCommand").f(void 0, void 0).ser(se_DeleteEndpointGroupCommand).de(de_DeleteEndpointGroupCommand).build() {
};
__name(_DeleteEndpointGroupCommand, "DeleteEndpointGroupCommand");
var DeleteEndpointGroupCommand = _DeleteEndpointGroupCommand;

// src/commands/DeleteListenerCommand.ts




var _DeleteListenerCommand = class _DeleteListenerCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("GlobalAccelerator_V20180706", "DeleteListener", {}).n("GlobalAcceleratorClient", "DeleteListenerCommand").f(void 0, void 0).ser(se_DeleteListenerCommand).de(de_DeleteListenerCommand).build() {
};
__name(_DeleteListenerCommand, "DeleteListenerCommand");
var DeleteListenerCommand = _DeleteListenerCommand;

// src/commands/DenyCustomRoutingTrafficCommand.ts




var _DenyCustomRoutingTrafficCommand = class _DenyCustomRoutingTrafficCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("GlobalAccelerator_V20180706", "DenyCustomRoutingTraffic", {}).n("GlobalAcceleratorClient", "DenyCustomRoutingTrafficCommand").f(void 0, void 0).ser(se_DenyCustomRoutingTrafficCommand).de(de_DenyCustomRoutingTrafficCommand).build() {
};
__name(_DenyCustomRoutingTrafficCommand, "DenyCustomRoutingTrafficCommand");
var DenyCustomRoutingTrafficCommand = _DenyCustomRoutingTrafficCommand;

// src/commands/DeprovisionByoipCidrCommand.ts




var _DeprovisionByoipCidrCommand = class _DeprovisionByoipCidrCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("GlobalAccelerator_V20180706", "DeprovisionByoipCidr", {}).n("GlobalAcceleratorClient", "DeprovisionByoipCidrCommand").f(void 0, void 0).ser(se_DeprovisionByoipCidrCommand).de(de_DeprovisionByoipCidrCommand).build() {
};
__name(_DeprovisionByoipCidrCommand, "DeprovisionByoipCidrCommand");
var DeprovisionByoipCidrCommand = _DeprovisionByoipCidrCommand;

// src/commands/DescribeAcceleratorAttributesCommand.ts




var _DescribeAcceleratorAttributesCommand = class _DescribeAcceleratorAttributesCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("GlobalAccelerator_V20180706", "DescribeAcceleratorAttributes", {}).n("GlobalAcceleratorClient", "DescribeAcceleratorAttributesCommand").f(void 0, void 0).ser(se_DescribeAcceleratorAttributesCommand).de(de_DescribeAcceleratorAttributesCommand).build() {
};
__name(_DescribeAcceleratorAttributesCommand, "DescribeAcceleratorAttributesCommand");
var DescribeAcceleratorAttributesCommand = _DescribeAcceleratorAttributesCommand;

// src/commands/DescribeAcceleratorCommand.ts




var _DescribeAcceleratorCommand = class _DescribeAcceleratorCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("GlobalAccelerator_V20180706", "DescribeAccelerator", {}).n("GlobalAcceleratorClient", "DescribeAcceleratorCommand").f(void 0, void 0).ser(se_DescribeAcceleratorCommand).de(de_DescribeAcceleratorCommand).build() {
};
__name(_DescribeAcceleratorCommand, "DescribeAcceleratorCommand");
var DescribeAcceleratorCommand = _DescribeAcceleratorCommand;

// src/commands/DescribeCrossAccountAttachmentCommand.ts




var _DescribeCrossAccountAttachmentCommand = class _DescribeCrossAccountAttachmentCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("GlobalAccelerator_V20180706", "DescribeCrossAccountAttachment", {}).n("GlobalAcceleratorClient", "DescribeCrossAccountAttachmentCommand").f(void 0, void 0).ser(se_DescribeCrossAccountAttachmentCommand).de(de_DescribeCrossAccountAttachmentCommand).build() {
};
__name(_DescribeCrossAccountAttachmentCommand, "DescribeCrossAccountAttachmentCommand");
var DescribeCrossAccountAttachmentCommand = _DescribeCrossAccountAttachmentCommand;

// src/commands/DescribeCustomRoutingAcceleratorAttributesCommand.ts




var _DescribeCustomRoutingAcceleratorAttributesCommand = class _DescribeCustomRoutingAcceleratorAttributesCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("GlobalAccelerator_V20180706", "DescribeCustomRoutingAcceleratorAttributes", {}).n("GlobalAcceleratorClient", "DescribeCustomRoutingAcceleratorAttributesCommand").f(void 0, void 0).ser(se_DescribeCustomRoutingAcceleratorAttributesCommand).de(de_DescribeCustomRoutingAcceleratorAttributesCommand).build() {
};
__name(_DescribeCustomRoutingAcceleratorAttributesCommand, "DescribeCustomRoutingAcceleratorAttributesCommand");
var DescribeCustomRoutingAcceleratorAttributesCommand = _DescribeCustomRoutingAcceleratorAttributesCommand;

// src/commands/DescribeCustomRoutingAcceleratorCommand.ts




var _DescribeCustomRoutingAcceleratorCommand = class _DescribeCustomRoutingAcceleratorCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("GlobalAccelerator_V20180706", "DescribeCustomRoutingAccelerator", {}).n("GlobalAcceleratorClient", "DescribeCustomRoutingAcceleratorCommand").f(void 0, void 0).ser(se_DescribeCustomRoutingAcceleratorCommand).de(de_DescribeCustomRoutingAcceleratorCommand).build() {
};
__name(_DescribeCustomRoutingAcceleratorCommand, "DescribeCustomRoutingAcceleratorCommand");
var DescribeCustomRoutingAcceleratorCommand = _DescribeCustomRoutingAcceleratorCommand;

// src/commands/DescribeCustomRoutingEndpointGroupCommand.ts




var _DescribeCustomRoutingEndpointGroupCommand = class _DescribeCustomRoutingEndpointGroupCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("GlobalAccelerator_V20180706", "DescribeCustomRoutingEndpointGroup", {}).n("GlobalAcceleratorClient", "DescribeCustomRoutingEndpointGroupCommand").f(void 0, void 0).ser(se_DescribeCustomRoutingEndpointGroupCommand).de(de_DescribeCustomRoutingEndpointGroupCommand).build() {
};
__name(_DescribeCustomRoutingEndpointGroupCommand, "DescribeCustomRoutingEndpointGroupCommand");
var DescribeCustomRoutingEndpointGroupCommand = _DescribeCustomRoutingEndpointGroupCommand;

// src/commands/DescribeCustomRoutingListenerCommand.ts




var _DescribeCustomRoutingListenerCommand = class _DescribeCustomRoutingListenerCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("GlobalAccelerator_V20180706", "DescribeCustomRoutingListener", {}).n("GlobalAcceleratorClient", "DescribeCustomRoutingListenerCommand").f(void 0, void 0).ser(se_DescribeCustomRoutingListenerCommand).de(de_DescribeCustomRoutingListenerCommand).build() {
};
__name(_DescribeCustomRoutingListenerCommand, "DescribeCustomRoutingListenerCommand");
var DescribeCustomRoutingListenerCommand = _DescribeCustomRoutingListenerCommand;

// src/commands/DescribeEndpointGroupCommand.ts




var _DescribeEndpointGroupCommand = class _DescribeEndpointGroupCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("GlobalAccelerator_V20180706", "DescribeEndpointGroup", {}).n("GlobalAcceleratorClient", "DescribeEndpointGroupCommand").f(void 0, void 0).ser(se_DescribeEndpointGroupCommand).de(de_DescribeEndpointGroupCommand).build() {
};
__name(_DescribeEndpointGroupCommand, "DescribeEndpointGroupCommand");
var DescribeEndpointGroupCommand = _DescribeEndpointGroupCommand;

// src/commands/DescribeListenerCommand.ts




var _DescribeListenerCommand = class _DescribeListenerCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("GlobalAccelerator_V20180706", "DescribeListener", {}).n("GlobalAcceleratorClient", "DescribeListenerCommand").f(void 0, void 0).ser(se_DescribeListenerCommand).de(de_DescribeListenerCommand).build() {
};
__name(_DescribeListenerCommand, "DescribeListenerCommand");
var DescribeListenerCommand = _DescribeListenerCommand;

// src/commands/ListAcceleratorsCommand.ts




var _ListAcceleratorsCommand = class _ListAcceleratorsCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("GlobalAccelerator_V20180706", "ListAccelerators", {}).n("GlobalAcceleratorClient", "ListAcceleratorsCommand").f(void 0, void 0).ser(se_ListAcceleratorsCommand).de(de_ListAcceleratorsCommand).build() {
};
__name(_ListAcceleratorsCommand, "ListAcceleratorsCommand");
var ListAcceleratorsCommand = _ListAcceleratorsCommand;

// src/commands/ListByoipCidrsCommand.ts




var _ListByoipCidrsCommand = class _ListByoipCidrsCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("GlobalAccelerator_V20180706", "ListByoipCidrs", {}).n("GlobalAcceleratorClient", "ListByoipCidrsCommand").f(void 0, void 0).ser(se_ListByoipCidrsCommand).de(de_ListByoipCidrsCommand).build() {
};
__name(_ListByoipCidrsCommand, "ListByoipCidrsCommand");
var ListByoipCidrsCommand = _ListByoipCidrsCommand;

// src/commands/ListCrossAccountAttachmentsCommand.ts




var _ListCrossAccountAttachmentsCommand = class _ListCrossAccountAttachmentsCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("GlobalAccelerator_V20180706", "ListCrossAccountAttachments", {}).n("GlobalAcceleratorClient", "ListCrossAccountAttachmentsCommand").f(void 0, void 0).ser(se_ListCrossAccountAttachmentsCommand).de(de_ListCrossAccountAttachmentsCommand).build() {
};
__name(_ListCrossAccountAttachmentsCommand, "ListCrossAccountAttachmentsCommand");
var ListCrossAccountAttachmentsCommand = _ListCrossAccountAttachmentsCommand;

// src/commands/ListCrossAccountResourceAccountsCommand.ts




var _ListCrossAccountResourceAccountsCommand = class _ListCrossAccountResourceAccountsCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("GlobalAccelerator_V20180706", "ListCrossAccountResourceAccounts", {}).n("GlobalAcceleratorClient", "ListCrossAccountResourceAccountsCommand").f(void 0, void 0).ser(se_ListCrossAccountResourceAccountsCommand).de(de_ListCrossAccountResourceAccountsCommand).build() {
};
__name(_ListCrossAccountResourceAccountsCommand, "ListCrossAccountResourceAccountsCommand");
var ListCrossAccountResourceAccountsCommand = _ListCrossAccountResourceAccountsCommand;

// src/commands/ListCrossAccountResourcesCommand.ts




var _ListCrossAccountResourcesCommand = class _ListCrossAccountResourcesCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("GlobalAccelerator_V20180706", "ListCrossAccountResources", {}).n("GlobalAcceleratorClient", "ListCrossAccountResourcesCommand").f(void 0, void 0).ser(se_ListCrossAccountResourcesCommand).de(de_ListCrossAccountResourcesCommand).build() {
};
__name(_ListCrossAccountResourcesCommand, "ListCrossAccountResourcesCommand");
var ListCrossAccountResourcesCommand = _ListCrossAccountResourcesCommand;

// src/commands/ListCustomRoutingAcceleratorsCommand.ts




var _ListCustomRoutingAcceleratorsCommand = class _ListCustomRoutingAcceleratorsCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("GlobalAccelerator_V20180706", "ListCustomRoutingAccelerators", {}).n("GlobalAcceleratorClient", "ListCustomRoutingAcceleratorsCommand").f(void 0, void 0).ser(se_ListCustomRoutingAcceleratorsCommand).de(de_ListCustomRoutingAcceleratorsCommand).build() {
};
__name(_ListCustomRoutingAcceleratorsCommand, "ListCustomRoutingAcceleratorsCommand");
var ListCustomRoutingAcceleratorsCommand = _ListCustomRoutingAcceleratorsCommand;

// src/commands/ListCustomRoutingEndpointGroupsCommand.ts




var _ListCustomRoutingEndpointGroupsCommand = class _ListCustomRoutingEndpointGroupsCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("GlobalAccelerator_V20180706", "ListCustomRoutingEndpointGroups", {}).n("GlobalAcceleratorClient", "ListCustomRoutingEndpointGroupsCommand").f(void 0, void 0).ser(se_ListCustomRoutingEndpointGroupsCommand).de(de_ListCustomRoutingEndpointGroupsCommand).build() {
};
__name(_ListCustomRoutingEndpointGroupsCommand, "ListCustomRoutingEndpointGroupsCommand");
var ListCustomRoutingEndpointGroupsCommand = _ListCustomRoutingEndpointGroupsCommand;

// src/commands/ListCustomRoutingListenersCommand.ts




var _ListCustomRoutingListenersCommand = class _ListCustomRoutingListenersCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("GlobalAccelerator_V20180706", "ListCustomRoutingListeners", {}).n("GlobalAcceleratorClient", "ListCustomRoutingListenersCommand").f(void 0, void 0).ser(se_ListCustomRoutingListenersCommand).de(de_ListCustomRoutingListenersCommand).build() {
};
__name(_ListCustomRoutingListenersCommand, "ListCustomRoutingListenersCommand");
var ListCustomRoutingListenersCommand = _ListCustomRoutingListenersCommand;

// src/commands/ListCustomRoutingPortMappingsByDestinationCommand.ts




var _ListCustomRoutingPortMappingsByDestinationCommand = class _ListCustomRoutingPortMappingsByDestinationCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("GlobalAccelerator_V20180706", "ListCustomRoutingPortMappingsByDestination", {}).n("GlobalAcceleratorClient", "ListCustomRoutingPortMappingsByDestinationCommand").f(void 0, void 0).ser(se_ListCustomRoutingPortMappingsByDestinationCommand).de(de_ListCustomRoutingPortMappingsByDestinationCommand).build() {
};
__name(_ListCustomRoutingPortMappingsByDestinationCommand, "ListCustomRoutingPortMappingsByDestinationCommand");
var ListCustomRoutingPortMappingsByDestinationCommand = _ListCustomRoutingPortMappingsByDestinationCommand;

// src/commands/ListCustomRoutingPortMappingsCommand.ts




var _ListCustomRoutingPortMappingsCommand = class _ListCustomRoutingPortMappingsCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("GlobalAccelerator_V20180706", "ListCustomRoutingPortMappings", {}).n("GlobalAcceleratorClient", "ListCustomRoutingPortMappingsCommand").f(void 0, void 0).ser(se_ListCustomRoutingPortMappingsCommand).de(de_ListCustomRoutingPortMappingsCommand).build() {
};
__name(_ListCustomRoutingPortMappingsCommand, "ListCustomRoutingPortMappingsCommand");
var ListCustomRoutingPortMappingsCommand = _ListCustomRoutingPortMappingsCommand;

// src/commands/ListEndpointGroupsCommand.ts




var _ListEndpointGroupsCommand = class _ListEndpointGroupsCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("GlobalAccelerator_V20180706", "ListEndpointGroups", {}).n("GlobalAcceleratorClient", "ListEndpointGroupsCommand").f(void 0, void 0).ser(se_ListEndpointGroupsCommand).de(de_ListEndpointGroupsCommand).build() {
};
__name(_ListEndpointGroupsCommand, "ListEndpointGroupsCommand");
var ListEndpointGroupsCommand = _ListEndpointGroupsCommand;

// src/commands/ListListenersCommand.ts




var _ListListenersCommand = class _ListListenersCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("GlobalAccelerator_V20180706", "ListListeners", {}).n("GlobalAcceleratorClient", "ListListenersCommand").f(void 0, void 0).ser(se_ListListenersCommand).de(de_ListListenersCommand).build() {
};
__name(_ListListenersCommand, "ListListenersCommand");
var ListListenersCommand = _ListListenersCommand;

// src/commands/ListTagsForResourceCommand.ts




var _ListTagsForResourceCommand = class _ListTagsForResourceCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("GlobalAccelerator_V20180706", "ListTagsForResource", {}).n("GlobalAcceleratorClient", "ListTagsForResourceCommand").f(void 0, void 0).ser(se_ListTagsForResourceCommand).de(de_ListTagsForResourceCommand).build() {
};
__name(_ListTagsForResourceCommand, "ListTagsForResourceCommand");
var ListTagsForResourceCommand = _ListTagsForResourceCommand;

// src/commands/ProvisionByoipCidrCommand.ts




var _ProvisionByoipCidrCommand = class _ProvisionByoipCidrCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("GlobalAccelerator_V20180706", "ProvisionByoipCidr", {}).n("GlobalAcceleratorClient", "ProvisionByoipCidrCommand").f(void 0, void 0).ser(se_ProvisionByoipCidrCommand).de(de_ProvisionByoipCidrCommand).build() {
};
__name(_ProvisionByoipCidrCommand, "ProvisionByoipCidrCommand");
var ProvisionByoipCidrCommand = _ProvisionByoipCidrCommand;

// src/commands/RemoveCustomRoutingEndpointsCommand.ts




var _RemoveCustomRoutingEndpointsCommand = class _RemoveCustomRoutingEndpointsCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("GlobalAccelerator_V20180706", "RemoveCustomRoutingEndpoints", {}).n("GlobalAcceleratorClient", "RemoveCustomRoutingEndpointsCommand").f(void 0, void 0).ser(se_RemoveCustomRoutingEndpointsCommand).de(de_RemoveCustomRoutingEndpointsCommand).build() {
};
__name(_RemoveCustomRoutingEndpointsCommand, "RemoveCustomRoutingEndpointsCommand");
var RemoveCustomRoutingEndpointsCommand = _RemoveCustomRoutingEndpointsCommand;

// src/commands/RemoveEndpointsCommand.ts




var _RemoveEndpointsCommand = class _RemoveEndpointsCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("GlobalAccelerator_V20180706", "RemoveEndpoints", {}).n("GlobalAcceleratorClient", "RemoveEndpointsCommand").f(void 0, void 0).ser(se_RemoveEndpointsCommand).de(de_RemoveEndpointsCommand).build() {
};
__name(_RemoveEndpointsCommand, "RemoveEndpointsCommand");
var RemoveEndpointsCommand = _RemoveEndpointsCommand;

// src/commands/TagResourceCommand.ts




var _TagResourceCommand = class _TagResourceCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("GlobalAccelerator_V20180706", "TagResource", {}).n("GlobalAcceleratorClient", "TagResourceCommand").f(void 0, void 0).ser(se_TagResourceCommand).de(de_TagResourceCommand).build() {
};
__name(_TagResourceCommand, "TagResourceCommand");
var TagResourceCommand = _TagResourceCommand;

// src/commands/UntagResourceCommand.ts




var _UntagResourceCommand = class _UntagResourceCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("GlobalAccelerator_V20180706", "UntagResource", {}).n("GlobalAcceleratorClient", "UntagResourceCommand").f(void 0, void 0).ser(se_UntagResourceCommand).de(de_UntagResourceCommand).build() {
};
__name(_UntagResourceCommand, "UntagResourceCommand");
var UntagResourceCommand = _UntagResourceCommand;

// src/commands/UpdateAcceleratorAttributesCommand.ts




var _UpdateAcceleratorAttributesCommand = class _UpdateAcceleratorAttributesCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("GlobalAccelerator_V20180706", "UpdateAcceleratorAttributes", {}).n("GlobalAcceleratorClient", "UpdateAcceleratorAttributesCommand").f(void 0, void 0).ser(se_UpdateAcceleratorAttributesCommand).de(de_UpdateAcceleratorAttributesCommand).build() {
};
__name(_UpdateAcceleratorAttributesCommand, "UpdateAcceleratorAttributesCommand");
var UpdateAcceleratorAttributesCommand = _UpdateAcceleratorAttributesCommand;

// src/commands/UpdateAcceleratorCommand.ts




var _UpdateAcceleratorCommand = class _UpdateAcceleratorCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("GlobalAccelerator_V20180706", "UpdateAccelerator", {}).n("GlobalAcceleratorClient", "UpdateAcceleratorCommand").f(void 0, void 0).ser(se_UpdateAcceleratorCommand).de(de_UpdateAcceleratorCommand).build() {
};
__name(_UpdateAcceleratorCommand, "UpdateAcceleratorCommand");
var UpdateAcceleratorCommand = _UpdateAcceleratorCommand;

// src/commands/UpdateCrossAccountAttachmentCommand.ts




var _UpdateCrossAccountAttachmentCommand = class _UpdateCrossAccountAttachmentCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("GlobalAccelerator_V20180706", "UpdateCrossAccountAttachment", {}).n("GlobalAcceleratorClient", "UpdateCrossAccountAttachmentCommand").f(void 0, void 0).ser(se_UpdateCrossAccountAttachmentCommand).de(de_UpdateCrossAccountAttachmentCommand).build() {
};
__name(_UpdateCrossAccountAttachmentCommand, "UpdateCrossAccountAttachmentCommand");
var UpdateCrossAccountAttachmentCommand = _UpdateCrossAccountAttachmentCommand;

// src/commands/UpdateCustomRoutingAcceleratorAttributesCommand.ts




var _UpdateCustomRoutingAcceleratorAttributesCommand = class _UpdateCustomRoutingAcceleratorAttributesCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("GlobalAccelerator_V20180706", "UpdateCustomRoutingAcceleratorAttributes", {}).n("GlobalAcceleratorClient", "UpdateCustomRoutingAcceleratorAttributesCommand").f(void 0, void 0).ser(se_UpdateCustomRoutingAcceleratorAttributesCommand).de(de_UpdateCustomRoutingAcceleratorAttributesCommand).build() {
};
__name(_UpdateCustomRoutingAcceleratorAttributesCommand, "UpdateCustomRoutingAcceleratorAttributesCommand");
var UpdateCustomRoutingAcceleratorAttributesCommand = _UpdateCustomRoutingAcceleratorAttributesCommand;

// src/commands/UpdateCustomRoutingAcceleratorCommand.ts




var _UpdateCustomRoutingAcceleratorCommand = class _UpdateCustomRoutingAcceleratorCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("GlobalAccelerator_V20180706", "UpdateCustomRoutingAccelerator", {}).n("GlobalAcceleratorClient", "UpdateCustomRoutingAcceleratorCommand").f(void 0, void 0).ser(se_UpdateCustomRoutingAcceleratorCommand).de(de_UpdateCustomRoutingAcceleratorCommand).build() {
};
__name(_UpdateCustomRoutingAcceleratorCommand, "UpdateCustomRoutingAcceleratorCommand");
var UpdateCustomRoutingAcceleratorCommand = _UpdateCustomRoutingAcceleratorCommand;

// src/commands/UpdateCustomRoutingListenerCommand.ts




var _UpdateCustomRoutingListenerCommand = class _UpdateCustomRoutingListenerCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("GlobalAccelerator_V20180706", "UpdateCustomRoutingListener", {}).n("GlobalAcceleratorClient", "UpdateCustomRoutingListenerCommand").f(void 0, void 0).ser(se_UpdateCustomRoutingListenerCommand).de(de_UpdateCustomRoutingListenerCommand).build() {
};
__name(_UpdateCustomRoutingListenerCommand, "UpdateCustomRoutingListenerCommand");
var UpdateCustomRoutingListenerCommand = _UpdateCustomRoutingListenerCommand;

// src/commands/UpdateEndpointGroupCommand.ts




var _UpdateEndpointGroupCommand = class _UpdateEndpointGroupCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("GlobalAccelerator_V20180706", "UpdateEndpointGroup", {}).n("GlobalAcceleratorClient", "UpdateEndpointGroupCommand").f(void 0, void 0).ser(se_UpdateEndpointGroupCommand).de(de_UpdateEndpointGroupCommand).build() {
};
__name(_UpdateEndpointGroupCommand, "UpdateEndpointGroupCommand");
var UpdateEndpointGroupCommand = _UpdateEndpointGroupCommand;

// src/commands/UpdateListenerCommand.ts




var _UpdateListenerCommand = class _UpdateListenerCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("GlobalAccelerator_V20180706", "UpdateListener", {}).n("GlobalAcceleratorClient", "UpdateListenerCommand").f(void 0, void 0).ser(se_UpdateListenerCommand).de(de_UpdateListenerCommand).build() {
};
__name(_UpdateListenerCommand, "UpdateListenerCommand");
var UpdateListenerCommand = _UpdateListenerCommand;

// src/commands/WithdrawByoipCidrCommand.ts




var _WithdrawByoipCidrCommand = class _WithdrawByoipCidrCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("GlobalAccelerator_V20180706", "WithdrawByoipCidr", {}).n("GlobalAcceleratorClient", "WithdrawByoipCidrCommand").f(void 0, void 0).ser(se_WithdrawByoipCidrCommand).de(de_WithdrawByoipCidrCommand).build() {
};
__name(_WithdrawByoipCidrCommand, "WithdrawByoipCidrCommand");
var WithdrawByoipCidrCommand = _WithdrawByoipCidrCommand;

// src/GlobalAccelerator.ts
var commands = {
  AddCustomRoutingEndpointsCommand,
  AddEndpointsCommand,
  AdvertiseByoipCidrCommand,
  AllowCustomRoutingTrafficCommand,
  CreateAcceleratorCommand,
  CreateCrossAccountAttachmentCommand,
  CreateCustomRoutingAcceleratorCommand,
  CreateCustomRoutingEndpointGroupCommand,
  CreateCustomRoutingListenerCommand,
  CreateEndpointGroupCommand,
  CreateListenerCommand,
  DeleteAcceleratorCommand,
  DeleteCrossAccountAttachmentCommand,
  DeleteCustomRoutingAcceleratorCommand,
  DeleteCustomRoutingEndpointGroupCommand,
  DeleteCustomRoutingListenerCommand,
  DeleteEndpointGroupCommand,
  DeleteListenerCommand,
  DenyCustomRoutingTrafficCommand,
  DeprovisionByoipCidrCommand,
  DescribeAcceleratorCommand,
  DescribeAcceleratorAttributesCommand,
  DescribeCrossAccountAttachmentCommand,
  DescribeCustomRoutingAcceleratorCommand,
  DescribeCustomRoutingAcceleratorAttributesCommand,
  DescribeCustomRoutingEndpointGroupCommand,
  DescribeCustomRoutingListenerCommand,
  DescribeEndpointGroupCommand,
  DescribeListenerCommand,
  ListAcceleratorsCommand,
  ListByoipCidrsCommand,
  ListCrossAccountAttachmentsCommand,
  ListCrossAccountResourceAccountsCommand,
  ListCrossAccountResourcesCommand,
  ListCustomRoutingAcceleratorsCommand,
  ListCustomRoutingEndpointGroupsCommand,
  ListCustomRoutingListenersCommand,
  ListCustomRoutingPortMappingsCommand,
  ListCustomRoutingPortMappingsByDestinationCommand,
  ListEndpointGroupsCommand,
  ListListenersCommand,
  ListTagsForResourceCommand,
  ProvisionByoipCidrCommand,
  RemoveCustomRoutingEndpointsCommand,
  RemoveEndpointsCommand,
  TagResourceCommand,
  UntagResourceCommand,
  UpdateAcceleratorCommand,
  UpdateAcceleratorAttributesCommand,
  UpdateCrossAccountAttachmentCommand,
  UpdateCustomRoutingAcceleratorCommand,
  UpdateCustomRoutingAcceleratorAttributesCommand,
  UpdateCustomRoutingListenerCommand,
  UpdateEndpointGroupCommand,
  UpdateListenerCommand,
  WithdrawByoipCidrCommand
};
var _GlobalAccelerator = class _GlobalAccelerator extends GlobalAcceleratorClient {
};
__name(_GlobalAccelerator, "GlobalAccelerator");
var GlobalAccelerator = _GlobalAccelerator;
(0, import_smithy_client.createAggregatedClient)(commands, GlobalAccelerator);

// src/pagination/ListAcceleratorsPaginator.ts

var paginateListAccelerators = (0, import_core.createPaginator)(GlobalAcceleratorClient, ListAcceleratorsCommand, "NextToken", "NextToken", "MaxResults");

// src/pagination/ListByoipCidrsPaginator.ts

var paginateListByoipCidrs = (0, import_core.createPaginator)(GlobalAcceleratorClient, ListByoipCidrsCommand, "NextToken", "NextToken", "MaxResults");

// src/pagination/ListCrossAccountAttachmentsPaginator.ts

var paginateListCrossAccountAttachments = (0, import_core.createPaginator)(GlobalAcceleratorClient, ListCrossAccountAttachmentsCommand, "NextToken", "NextToken", "MaxResults");

// src/pagination/ListCrossAccountResourcesPaginator.ts

var paginateListCrossAccountResources = (0, import_core.createPaginator)(GlobalAcceleratorClient, ListCrossAccountResourcesCommand, "NextToken", "NextToken", "MaxResults");

// src/pagination/ListCustomRoutingAcceleratorsPaginator.ts

var paginateListCustomRoutingAccelerators = (0, import_core.createPaginator)(GlobalAcceleratorClient, ListCustomRoutingAcceleratorsCommand, "NextToken", "NextToken", "MaxResults");

// src/pagination/ListCustomRoutingEndpointGroupsPaginator.ts

var paginateListCustomRoutingEndpointGroups = (0, import_core.createPaginator)(GlobalAcceleratorClient, ListCustomRoutingEndpointGroupsCommand, "NextToken", "NextToken", "MaxResults");

// src/pagination/ListCustomRoutingListenersPaginator.ts

var paginateListCustomRoutingListeners = (0, import_core.createPaginator)(GlobalAcceleratorClient, ListCustomRoutingListenersCommand, "NextToken", "NextToken", "MaxResults");

// src/pagination/ListCustomRoutingPortMappingsByDestinationPaginator.ts

var paginateListCustomRoutingPortMappingsByDestination = (0, import_core.createPaginator)(GlobalAcceleratorClient, ListCustomRoutingPortMappingsByDestinationCommand, "NextToken", "NextToken", "MaxResults");

// src/pagination/ListCustomRoutingPortMappingsPaginator.ts

var paginateListCustomRoutingPortMappings = (0, import_core.createPaginator)(GlobalAcceleratorClient, ListCustomRoutingPortMappingsCommand, "NextToken", "NextToken", "MaxResults");

// src/pagination/ListEndpointGroupsPaginator.ts

var paginateListEndpointGroups = (0, import_core.createPaginator)(GlobalAcceleratorClient, ListEndpointGroupsCommand, "NextToken", "NextToken", "MaxResults");

// src/pagination/ListListenersPaginator.ts

var paginateListListeners = (0, import_core.createPaginator)(GlobalAcceleratorClient, ListListenersCommand, "NextToken", "NextToken", "MaxResults");
// Annotate the CommonJS export names for ESM import in node:

0 && (module.exports = {
  GlobalAcceleratorServiceException,
  __Client,
  GlobalAcceleratorClient,
  GlobalAccelerator,
  $Command,
  AddCustomRoutingEndpointsCommand,
  AddEndpointsCommand,
  AdvertiseByoipCidrCommand,
  AllowCustomRoutingTrafficCommand,
  CreateAcceleratorCommand,
  CreateCrossAccountAttachmentCommand,
  CreateCustomRoutingAcceleratorCommand,
  CreateCustomRoutingEndpointGroupCommand,
  CreateCustomRoutingListenerCommand,
  CreateEndpointGroupCommand,
  CreateListenerCommand,
  DeleteAcceleratorCommand,
  DeleteCrossAccountAttachmentCommand,
  DeleteCustomRoutingAcceleratorCommand,
  DeleteCustomRoutingEndpointGroupCommand,
  DeleteCustomRoutingListenerCommand,
  DeleteEndpointGroupCommand,
  DeleteListenerCommand,
  DenyCustomRoutingTrafficCommand,
  DeprovisionByoipCidrCommand,
  DescribeAcceleratorAttributesCommand,
  DescribeAcceleratorCommand,
  DescribeCrossAccountAttachmentCommand,
  DescribeCustomRoutingAcceleratorAttributesCommand,
  DescribeCustomRoutingAcceleratorCommand,
  DescribeCustomRoutingEndpointGroupCommand,
  DescribeCustomRoutingListenerCommand,
  DescribeEndpointGroupCommand,
  DescribeListenerCommand,
  ListAcceleratorsCommand,
  ListByoipCidrsCommand,
  ListCrossAccountAttachmentsCommand,
  ListCrossAccountResourceAccountsCommand,
  ListCrossAccountResourcesCommand,
  ListCustomRoutingAcceleratorsCommand,
  ListCustomRoutingEndpointGroupsCommand,
  ListCustomRoutingListenersCommand,
  ListCustomRoutingPortMappingsByDestinationCommand,
  ListCustomRoutingPortMappingsCommand,
  ListEndpointGroupsCommand,
  ListListenersCommand,
  ListTagsForResourceCommand,
  ProvisionByoipCidrCommand,
  RemoveCustomRoutingEndpointsCommand,
  RemoveEndpointsCommand,
  TagResourceCommand,
  UntagResourceCommand,
  UpdateAcceleratorAttributesCommand,
  UpdateAcceleratorCommand,
  UpdateCrossAccountAttachmentCommand,
  UpdateCustomRoutingAcceleratorAttributesCommand,
  UpdateCustomRoutingAcceleratorCommand,
  UpdateCustomRoutingListenerCommand,
  UpdateEndpointGroupCommand,
  UpdateListenerCommand,
  WithdrawByoipCidrCommand,
  paginateListAccelerators,
  paginateListByoipCidrs,
  paginateListCrossAccountAttachments,
  paginateListCrossAccountResources,
  paginateListCustomRoutingAccelerators,
  paginateListCustomRoutingEndpointGroups,
  paginateListCustomRoutingListeners,
  paginateListCustomRoutingPortMappingsByDestination,
  paginateListCustomRoutingPortMappings,
  paginateListEndpointGroups,
  paginateListListeners,
  IpAddressType,
  IpAddressFamily,
  AcceleratorStatus,
  AcceleratorNotDisabledException,
  AcceleratorNotFoundException,
  AccessDeniedException,
  ConflictException,
  EndpointAlreadyExistsException,
  EndpointGroupNotFoundException,
  InternalServiceErrorException,
  InvalidArgumentException,
  LimitExceededException,
  HealthState,
  TransactionInProgressException,
  ByoipCidrState,
  ByoipCidrNotFoundException,
  IncorrectCidrStateException,
  AssociatedEndpointGroupFoundException,
  AssociatedListenerFoundException,
  AttachmentNotFoundException,
  ClientAffinity,
  CustomRoutingAcceleratorStatus,
  CustomRoutingProtocol,
  Protocol,
  EndpointGroupAlreadyExistsException,
  InvalidPortRangeException,
  ListenerNotFoundException,
  HealthCheckProtocol,
  CustomRoutingDestinationTrafficState,
  EndpointNotFoundException,
  InvalidNextTokenException
});
