// @aws-sdk/client-groundstation/dist-cjs/index.js
"use strict";
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __name = (target, value) => __defProp(target, "name", { value, configurable: true });
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// src/index.ts
var src_exports = {};
__export(src_exports, {
  AgentStatus: () => AgentStatus,
  AngleUnits: () => AngleUnits,
  AuditResults: () => AuditResults,
  BandwidthUnits: () => BandwidthUnits,
  CancelContactCommand: () => CancelContactCommand,
  CapabilityHealth: () => CapabilityHealth,
  CapabilityHealthReason: () => CapabilityHealthReason,
  ConfigCapabilityType: () => ConfigCapabilityType,
  ConfigDetails: () => ConfigDetails,
  ConfigTypeData: () => ConfigTypeData,
  ContactStatus: () => ContactStatus,
  CreateConfigCommand: () => CreateConfigCommand,
  CreateDataflowEndpointGroupCommand: () => CreateDataflowEndpointGroupCommand,
  CreateEphemerisCommand: () => CreateEphemerisCommand,
  CreateMissionProfileCommand: () => CreateMissionProfileCommand,
  Criticality: () => Criticality,
  DeleteConfigCommand: () => DeleteConfigCommand,
  DeleteDataflowEndpointGroupCommand: () => DeleteDataflowEndpointGroupCommand,
  DeleteEphemerisCommand: () => DeleteEphemerisCommand,
  DeleteMissionProfileCommand: () => DeleteMissionProfileCommand,
  DependencyException: () => DependencyException,
  DescribeContactCommand: () => DescribeContactCommand,
  DescribeEphemerisCommand: () => DescribeEphemerisCommand,
  EirpUnits: () => EirpUnits,
  EndpointStatus: () => EndpointStatus,
  EphemerisData: () => EphemerisData,
  EphemerisInvalidReason: () => EphemerisInvalidReason,
  EphemerisSource: () => EphemerisSource,
  EphemerisStatus: () => EphemerisStatus,
  EphemerisTypeDescription: () => EphemerisTypeDescription,
  FrequencyUnits: () => FrequencyUnits,
  GetAgentConfigurationCommand: () => GetAgentConfigurationCommand,
  GetConfigCommand: () => GetConfigCommand,
  GetDataflowEndpointGroupCommand: () => GetDataflowEndpointGroupCommand,
  GetMinuteUsageCommand: () => GetMinuteUsageCommand,
  GetMissionProfileCommand: () => GetMissionProfileCommand,
  GetSatelliteCommand: () => GetSatelliteCommand,
  GroundStation: () => GroundStation,
  GroundStationClient: () => GroundStationClient,
  GroundStationServiceException: () => GroundStationServiceException,
  InvalidParameterException: () => InvalidParameterException,
  KmsKey: () => KmsKey,
  ListConfigsCommand: () => ListConfigsCommand,
  ListContactsCommand: () => ListContactsCommand,
  ListDataflowEndpointGroupsCommand: () => ListDataflowEndpointGroupsCommand,
  ListEphemeridesCommand: () => ListEphemeridesCommand,
  ListGroundStationsCommand: () => ListGroundStationsCommand,
  ListMissionProfilesCommand: () => ListMissionProfilesCommand,
  ListSatellitesCommand: () => ListSatellitesCommand,
  ListTagsForResourceCommand: () => ListTagsForResourceCommand,
  Polarization: () => Polarization,
  RegisterAgentCommand: () => RegisterAgentCommand,
  ReserveContactCommand: () => ReserveContactCommand,
  ResourceLimitExceededException: () => ResourceLimitExceededException,
  ResourceNotFoundException: () => ResourceNotFoundException,
  TagResourceCommand: () => TagResourceCommand,
  UntagResourceCommand: () => UntagResourceCommand,
  UpdateAgentStatusCommand: () => UpdateAgentStatusCommand,
  UpdateConfigCommand: () => UpdateConfigCommand,
  UpdateEphemerisCommand: () => UpdateEphemerisCommand,
  UpdateMissionProfileCommand: () => UpdateMissionProfileCommand,
  __Client: () => import_smithy_client.Client,
  paginateListConfigs: () => paginateListConfigs,
  paginateListContacts: () => paginateListContacts,
  paginateListDataflowEndpointGroups: () => paginateListDataflowEndpointGroups,
  paginateListEphemerides: () => paginateListEphemerides,
  paginateListGroundStations: () => paginateListGroundStations,
  paginateListMissionProfiles: () => paginateListMissionProfiles,
  paginateListSatellites: () => paginateListSatellites,
  waitForContactScheduled: () => waitForContactScheduled,
  waitUntilContactScheduled: () => waitUntilContactScheduled
});
module.exports = __toCommonJS(src_exports);

// src/GroundStationClient.ts
var import_middleware_host_header = require("@aws-sdk/middleware-host-header");
var import_middleware_logger = require("@aws-sdk/middleware-logger");
var import_middleware_recursion_detection = require("@aws-sdk/middleware-recursion-detection");
var import_middleware_user_agent = require("@aws-sdk/middleware-user-agent");
var import_config_resolver = require("@smithy/config-resolver");
var import_core = require("@smithy/core");
var import_middleware_content_length = require("@smithy/middleware-content-length");
var import_middleware_endpoint = require("@smithy/middleware-endpoint");
var import_middleware_retry = require("@smithy/middleware-retry");

var import_httpAuthSchemeProvider = require("./auth/httpAuthSchemeProvider");

// src/endpoint/EndpointParameters.ts
var resolveClientEndpointParameters = /* @__PURE__ */ __name((options) => {
  return {
    ...options,
    useDualstackEndpoint: options.useDualstackEndpoint ?? false,
    useFipsEndpoint: options.useFipsEndpoint ?? false,
    defaultSigningName: "groundstation"
  };
}, "resolveClientEndpointParameters");
var commonParams = {
  UseFIPS: { type: "builtInParams", name: "useFipsEndpoint" },
  Endpoint: { type: "builtInParams", name: "endpoint" },
  Region: { type: "builtInParams", name: "region" },
  UseDualStack: { type: "builtInParams", name: "useDualstackEndpoint" }
};

// src/GroundStationClient.ts
var import_runtimeConfig = require("././runtimeConfig");

// src/runtimeExtensions.ts
var import_region_config_resolver = require("@aws-sdk/region-config-resolver");
var import_protocol_http = require("@smithy/protocol-http");
var import_smithy_client = require("@smithy/smithy-client");

// src/auth/httpAuthExtensionConfiguration.ts
var getHttpAuthExtensionConfiguration = /* @__PURE__ */ __name((runtimeConfig) => {
  const _httpAuthSchemes = runtimeConfig.httpAuthSchemes;
  let _httpAuthSchemeProvider = runtimeConfig.httpAuthSchemeProvider;
  let _credentials = runtimeConfig.credentials;
  return {
    setHttpAuthScheme(httpAuthScheme) {
      const index = _httpAuthSchemes.findIndex((scheme) => scheme.schemeId === httpAuthScheme.schemeId);
      if (index === -1) {
        _httpAuthSchemes.push(httpAuthScheme);
      } else {
        _httpAuthSchemes.splice(index, 1, httpAuthScheme);
      }
    },
    httpAuthSchemes() {
      return _httpAuthSchemes;
    },
    setHttpAuthSchemeProvider(httpAuthSchemeProvider) {
      _httpAuthSchemeProvider = httpAuthSchemeProvider;
    },
    httpAuthSchemeProvider() {
      return _httpAuthSchemeProvider;
    },
    setCredentials(credentials) {
      _credentials = credentials;
    },
    credentials() {
      return _credentials;
    }
  };
}, "getHttpAuthExtensionConfiguration");
var resolveHttpAuthRuntimeConfig = /* @__PURE__ */ __name((config) => {
  return {
    httpAuthSchemes: config.httpAuthSchemes(),
    httpAuthSchemeProvider: config.httpAuthSchemeProvider(),
    credentials: config.credentials()
  };
}, "resolveHttpAuthRuntimeConfig");

// src/runtimeExtensions.ts
var asPartial = /* @__PURE__ */ __name((t) => t, "asPartial");
var resolveRuntimeExtensions = /* @__PURE__ */ __name((runtimeConfig, extensions) => {
  const extensionConfiguration = {
    ...asPartial((0, import_region_config_resolver.getAwsRegionExtensionConfiguration)(runtimeConfig)),
    ...asPartial((0, import_smithy_client.getDefaultExtensionConfiguration)(runtimeConfig)),
    ...asPartial((0, import_protocol_http.getHttpHandlerExtensionConfiguration)(runtimeConfig)),
    ...asPartial(getHttpAuthExtensionConfiguration(runtimeConfig))
  };
  extensions.forEach((extension) => extension.configure(extensionConfiguration));
  return {
    ...runtimeConfig,
    ...(0, import_region_config_resolver.resolveAwsRegionExtensionConfiguration)(extensionConfiguration),
    ...(0, import_smithy_client.resolveDefaultRuntimeConfig)(extensionConfiguration),
    ...(0, import_protocol_http.resolveHttpHandlerRuntimeConfig)(extensionConfiguration),
    ...resolveHttpAuthRuntimeConfig(extensionConfiguration)
  };
}, "resolveRuntimeExtensions");

// src/GroundStationClient.ts
var _GroundStationClient = class _GroundStationClient extends import_smithy_client.Client {
  constructor(...[configuration]) {
    const _config_0 = (0, import_runtimeConfig.getRuntimeConfig)(configuration || {});
    const _config_1 = resolveClientEndpointParameters(_config_0);
    const _config_2 = (0, import_config_resolver.resolveRegionConfig)(_config_1);
    const _config_3 = (0, import_middleware_endpoint.resolveEndpointConfig)(_config_2);
    const _config_4 = (0, import_middleware_retry.resolveRetryConfig)(_config_3);
    const _config_5 = (0, import_middleware_host_header.resolveHostHeaderConfig)(_config_4);
    const _config_6 = (0, import_middleware_user_agent.resolveUserAgentConfig)(_config_5);
    const _config_7 = (0, import_httpAuthSchemeProvider.resolveHttpAuthSchemeConfig)(_config_6);
    const _config_8 = resolveRuntimeExtensions(_config_7, (configuration == null ? void 0 : configuration.extensions) || []);
    super(_config_8);
    this.config = _config_8;
    this.middlewareStack.use((0, import_middleware_retry.getRetryPlugin)(this.config));
    this.middlewareStack.use((0, import_middleware_content_length.getContentLengthPlugin)(this.config));
    this.middlewareStack.use((0, import_middleware_host_header.getHostHeaderPlugin)(this.config));
    this.middlewareStack.use((0, import_middleware_logger.getLoggerPlugin)(this.config));
    this.middlewareStack.use((0, import_middleware_recursion_detection.getRecursionDetectionPlugin)(this.config));
    this.middlewareStack.use((0, import_middleware_user_agent.getUserAgentPlugin)(this.config));
    this.middlewareStack.use(
      (0, import_core.getHttpAuthSchemeEndpointRuleSetPlugin)(this.config, {
        httpAuthSchemeParametersProvider: this.getDefaultHttpAuthSchemeParametersProvider(),
        identityProviderConfigProvider: this.getIdentityProviderConfigProvider()
      })
    );
    this.middlewareStack.use((0, import_core.getHttpSigningPlugin)(this.config));
  }
  /**
   * Destroy underlying resources, like sockets. It's usually not necessary to do this.
   * However in Node.js, it's best to explicitly shut down the client's agent when it is no longer needed.
   * Otherwise, sockets might stay open for quite a long time before the server terminates them.
   */
  destroy() {
    super.destroy();
  }
  getDefaultHttpAuthSchemeParametersProvider() {
    return import_httpAuthSchemeProvider.defaultGroundStationHttpAuthSchemeParametersProvider;
  }
  getIdentityProviderConfigProvider() {
    return async (config) => new import_core.DefaultIdentityProviderConfig({
      "aws.auth#sigv4": config.credentials
    });
  }
};
__name(_GroundStationClient, "GroundStationClient");
var GroundStationClient = _GroundStationClient;

// src/GroundStation.ts


// src/commands/CancelContactCommand.ts

var import_middleware_serde = require("@smithy/middleware-serde");

var import_types = require("@smithy/types");

// src/protocols/Aws_restJson1.ts
var import_core2 = require("@aws-sdk/core");



// src/models/GroundStationServiceException.ts

var _GroundStationServiceException = class _GroundStationServiceException extends import_smithy_client.ServiceException {
  /**
   * @internal
   */
  constructor(options) {
    super(options);
    Object.setPrototypeOf(this, _GroundStationServiceException.prototype);
  }
};
__name(_GroundStationServiceException, "GroundStationServiceException");
var GroundStationServiceException = _GroundStationServiceException;

// src/models/models_0.ts
var _DependencyException = class _DependencyException extends GroundStationServiceException {
  /**
   * @internal
   */
  constructor(opts) {
    super({
      name: "DependencyException",
      $fault: "server",
      ...opts
    });
    this.name = "DependencyException";
    this.$fault = "server";
    Object.setPrototypeOf(this, _DependencyException.prototype);
    this.parameterName = opts.parameterName;
  }
};
__name(_DependencyException, "DependencyException");
var DependencyException = _DependencyException;
var _InvalidParameterException = class _InvalidParameterException extends GroundStationServiceException {
  /**
   * @internal
   */
  constructor(opts) {
    super({
      name: "InvalidParameterException",
      $fault: "client",
      ...opts
    });
    this.name = "InvalidParameterException";
    this.$fault = "client";
    Object.setPrototypeOf(this, _InvalidParameterException.prototype);
    this.parameterName = opts.parameterName;
  }
};
__name(_InvalidParameterException, "InvalidParameterException");
var InvalidParameterException = _InvalidParameterException;
var _ResourceNotFoundException = class _ResourceNotFoundException extends GroundStationServiceException {
  /**
   * @internal
   */
  constructor(opts) {
    super({
      name: "ResourceNotFoundException",
      $fault: "client",
      ...opts
    });
    this.name = "ResourceNotFoundException";
    this.$fault = "client";
    Object.setPrototypeOf(this, _ResourceNotFoundException.prototype);
  }
};
__name(_ResourceNotFoundException, "ResourceNotFoundException");
var ResourceNotFoundException = _ResourceNotFoundException;
var AgentStatus = {
  ACTIVE: "ACTIVE",
  FAILED: "FAILED",
  INACTIVE: "INACTIVE",
  SUCCESS: "SUCCESS"
};
var AngleUnits = {
  DEGREE_ANGLE: "DEGREE_ANGLE",
  RADIAN: "RADIAN"
};
var BandwidthUnits = {
  GHZ: "GHz",
  KHZ: "kHz",
  MHZ: "MHz"
};
var FrequencyUnits = {
  GHZ: "GHz",
  KHZ: "kHz",
  MHZ: "MHz"
};
var Polarization = {
  LEFT_HAND: "LEFT_HAND",
  NONE: "NONE",
  RIGHT_HAND: "RIGHT_HAND"
};
var EirpUnits = {
  DBW: "dBW"
};
var AuditResults = {
  HEALTHY: "HEALTHY",
  UNHEALTHY: "UNHEALTHY"
};
var CapabilityHealth = {
  HEALTHY: "HEALTHY",
  UNHEALTHY: "UNHEALTHY"
};
var CapabilityHealthReason = {
  DATAPLANE_FAILURE: "DATAPLANE_FAILURE",
  HEALTHY: "HEALTHY",
  INITIALIZING_DATAPLANE: "INITIALIZING_DATAPLANE",
  INVALID_IP_OWNERSHIP: "INVALID_IP_OWNERSHIP",
  NOT_AUTHORIZED_TO_CREATE_SLR: "NOT_AUTHORIZED_TO_CREATE_SLR",
  NO_REGISTERED_AGENT: "NO_REGISTERED_AGENT",
  UNVERIFIED_IP_OWNERSHIP: "UNVERIFIED_IP_OWNERSHIP"
};
var ConfigCapabilityType = {
  ANTENNA_DOWNLINK: "antenna-downlink",
  ANTENNA_DOWNLINK_DEMOD_DECODE: "antenna-downlink-demod-decode",
  ANTENNA_UPLINK: "antenna-uplink",
  DATAFLOW_ENDPOINT: "dataflow-endpoint",
  S3_RECORDING: "s3-recording",
  TRACKING: "tracking",
  UPLINK_ECHO: "uplink-echo"
};
var Criticality = {
  PREFERRED: "PREFERRED",
  REMOVED: "REMOVED",
  REQUIRED: "REQUIRED"
};
var ConfigTypeData;
((ConfigTypeData2) => {
  ConfigTypeData2.visit = /* @__PURE__ */ __name((value, visitor) => {
    if (value.antennaDownlinkConfig !== void 0)
      return visitor.antennaDownlinkConfig(value.antennaDownlinkConfig);
    if (value.trackingConfig !== void 0)
      return visitor.trackingConfig(value.trackingConfig);
    if (value.dataflowEndpointConfig !== void 0)
      return visitor.dataflowEndpointConfig(value.dataflowEndpointConfig);
    if (value.antennaDownlinkDemodDecodeConfig !== void 0)
      return visitor.antennaDownlinkDemodDecodeConfig(value.antennaDownlinkDemodDecodeConfig);
    if (value.antennaUplinkConfig !== void 0)
      return visitor.antennaUplinkConfig(value.antennaUplinkConfig);
    if (value.uplinkEchoConfig !== void 0)
      return visitor.uplinkEchoConfig(value.uplinkEchoConfig);
    if (value.s3RecordingConfig !== void 0)
      return visitor.s3RecordingConfig(value.s3RecordingConfig);
    return visitor._(value.$unknown[0], value.$unknown[1]);
  }, "visit");
})(ConfigTypeData || (ConfigTypeData = {}));
var _ResourceLimitExceededException = class _ResourceLimitExceededException extends GroundStationServiceException {
  /**
   * @internal
   */
  constructor(opts) {
    super({
      name: "ResourceLimitExceededException",
      $fault: "client",
      ...opts
    });
    this.name = "ResourceLimitExceededException";
    this.$fault = "client";
    Object.setPrototypeOf(this, _ResourceLimitExceededException.prototype);
    this.parameterName = opts.parameterName;
  }
};
__name(_ResourceLimitExceededException, "ResourceLimitExceededException");
var ResourceLimitExceededException = _ResourceLimitExceededException;
var EndpointStatus = {
  created: "created",
  creating: "creating",
  deleted: "deleted",
  deleting: "deleting",
  failed: "failed"
};
var ConfigDetails;
((ConfigDetails2) => {
  ConfigDetails2.visit = /* @__PURE__ */ __name((value, visitor) => {
    if (value.endpointDetails !== void 0)
      return visitor.endpointDetails(value.endpointDetails);
    if (value.antennaDemodDecodeDetails !== void 0)
      return visitor.antennaDemodDecodeDetails(value.antennaDemodDecodeDetails);
    if (value.s3RecordingDetails !== void 0)
      return visitor.s3RecordingDetails(value.s3RecordingDetails);
    return visitor._(value.$unknown[0], value.$unknown[1]);
  }, "visit");
})(ConfigDetails || (ConfigDetails = {}));
var ContactStatus = {
  AVAILABLE: "AVAILABLE",
  AWS_CANCELLED: "AWS_CANCELLED",
  AWS_FAILED: "AWS_FAILED",
  CANCELLED: "CANCELLED",
  CANCELLING: "CANCELLING",
  COMPLETED: "COMPLETED",
  FAILED: "FAILED",
  FAILED_TO_SCHEDULE: "FAILED_TO_SCHEDULE",
  PASS: "PASS",
  POSTPASS: "POSTPASS",
  PREPASS: "PREPASS",
  SCHEDULED: "SCHEDULED",
  SCHEDULING: "SCHEDULING"
};
var EphemerisData;
((EphemerisData2) => {
  EphemerisData2.visit = /* @__PURE__ */ __name((value, visitor) => {
    if (value.tle !== void 0)
      return visitor.tle(value.tle);
    if (value.oem !== void 0)
      return visitor.oem(value.oem);
    return visitor._(value.$unknown[0], value.$unknown[1]);
  }, "visit");
})(EphemerisData || (EphemerisData = {}));
var KmsKey;
((KmsKey3) => {
  KmsKey3.visit = /* @__PURE__ */ __name((value, visitor) => {
    if (value.kmsKeyArn !== void 0)
      return visitor.kmsKeyArn(value.kmsKeyArn);
    if (value.kmsAliasArn !== void 0)
      return visitor.kmsAliasArn(value.kmsAliasArn);
    if (value.kmsAliasName !== void 0)
      return visitor.kmsAliasName(value.kmsAliasName);
    return visitor._(value.$unknown[0], value.$unknown[1]);
  }, "visit");
})(KmsKey || (KmsKey = {}));
var EphemerisInvalidReason = {
  /**
   * Provided KMS key is invalid
   */
  KMS_KEY_INVALID: "KMS_KEY_INVALID",
  /**
   * Provided spacecraft identifiers such as spacecraft NORAD Id are invalid
   */
  METADATA_INVALID: "METADATA_INVALID",
  /**
   * Start, end, or expiration time(s) are invalid for the provided ephemeris
   */
  TIME_RANGE_INVALID: "TIME_RANGE_INVALID",
  /**
   * Provided ephemeris defines invalid spacecraft trajectory
   */
  TRAJECTORY_INVALID: "TRAJECTORY_INVALID",
  /**
   * Internal Service Error occurred while processing ephemeris
   */
  VALIDATION_ERROR: "VALIDATION_ERROR"
};
var EphemerisStatus = {
  DISABLED: "DISABLED",
  ENABLED: "ENABLED",
  ERROR: "ERROR",
  EXPIRED: "EXPIRED",
  INVALID: "INVALID",
  VALIDATING: "VALIDATING"
};
var EphemerisTypeDescription;
((EphemerisTypeDescription2) => {
  EphemerisTypeDescription2.visit = /* @__PURE__ */ __name((value, visitor) => {
    if (value.tle !== void 0)
      return visitor.tle(value.tle);
    if (value.oem !== void 0)
      return visitor.oem(value.oem);
    return visitor._(value.$unknown[0], value.$unknown[1]);
  }, "visit");
})(EphemerisTypeDescription || (EphemerisTypeDescription = {}));
var EphemerisSource = {
  CUSTOMER_PROVIDED: "CUSTOMER_PROVIDED",
  SPACE_TRACK: "SPACE_TRACK"
};

// src/protocols/Aws_restJson1.ts
var se_CancelContactCommand = /* @__PURE__ */ __name(async (input, context) => {
  const b = (0, import_core.requestBuilder)(input, context);
  const headers = {};
  b.bp("/contact/{contactId}");
  b.p("contactId", () => input.contactId, "{contactId}", false);
  let body;
  b.m("DELETE").h(headers).b(body);
  return b.build();
}, "se_CancelContactCommand");
var se_CreateConfigCommand = /* @__PURE__ */ __name(async (input, context) => {
  const b = (0, import_core.requestBuilder)(input, context);
  const headers = {
    "content-type": "application/json"
  };
  b.bp("/config");
  let body;
  body = JSON.stringify(
    (0, import_smithy_client.take)(input, {
      configData: (_) => se_ConfigTypeData(_, context),
      name: [],
      tags: (_) => (0, import_smithy_client._json)(_)
    })
  );
  b.m("POST").h(headers).b(body);
  return b.build();
}, "se_CreateConfigCommand");
var se_CreateDataflowEndpointGroupCommand = /* @__PURE__ */ __name(async (input, context) => {
  const b = (0, import_core.requestBuilder)(input, context);
  const headers = {
    "content-type": "application/json"
  };
  b.bp("/dataflowEndpointGroup");
  let body;
  body = JSON.stringify(
    (0, import_smithy_client.take)(input, {
      contactPostPassDurationSeconds: [],
      contactPrePassDurationSeconds: [],
      endpointDetails: (_) => (0, import_smithy_client._json)(_),
      tags: (_) => (0, import_smithy_client._json)(_)
    })
  );
  b.m("POST").h(headers).b(body);
  return b.build();
}, "se_CreateDataflowEndpointGroupCommand");
var se_CreateEphemerisCommand = /* @__PURE__ */ __name(async (input, context) => {
  const b = (0, import_core.requestBuilder)(input, context);
  const headers = {
    "content-type": "application/json"
  };
  b.bp("/ephemeris");
  let body;
  body = JSON.stringify(
    (0, import_smithy_client.take)(input, {
      enabled: [],
      ephemeris: (_) => se_EphemerisData(_, context),
      expirationTime: (_) => Math.round(_.getTime() / 1e3),
      kmsKeyArn: [],
      name: [],
      priority: [],
      satelliteId: [],
      tags: (_) => (0, import_smithy_client._json)(_)
    })
  );
  b.m("POST").h(headers).b(body);
  return b.build();
}, "se_CreateEphemerisCommand");
var se_CreateMissionProfileCommand = /* @__PURE__ */ __name(async (input, context) => {
  const b = (0, import_core.requestBuilder)(input, context);
  const headers = {
    "content-type": "application/json"
  };
  b.bp("/missionprofile");
  let body;
  body = JSON.stringify(
    (0, import_smithy_client.take)(input, {
      contactPostPassDurationSeconds: [],
      contactPrePassDurationSeconds: [],
      dataflowEdges: (_) => (0, import_smithy_client._json)(_),
      minimumViableContactDurationSeconds: [],
      name: [],
      streamsKmsKey: (_) => (0, import_smithy_client._json)(_),
      streamsKmsRole: [],
      tags: (_) => (0, import_smithy_client._json)(_),
      trackingConfigArn: []
    })
  );
  b.m("POST").h(headers).b(body);
  return b.build();
}, "se_CreateMissionProfileCommand");
var se_DeleteConfigCommand = /* @__PURE__ */ __name(async (input, context) => {
  const b = (0, import_core.requestBuilder)(input, context);
  const headers = {};
  b.bp("/config/{configType}/{configId}");
  b.p("configId", () => input.configId, "{configId}", false);
  b.p("configType", () => input.configType, "{configType}", false);
  let body;
  b.m("DELETE").h(headers).b(body);
  return b.build();
}, "se_DeleteConfigCommand");
var se_DeleteDataflowEndpointGroupCommand = /* @__PURE__ */ __name(async (input, context) => {
  const b = (0, import_core.requestBuilder)(input, context);
  const headers = {};
  b.bp("/dataflowEndpointGroup/{dataflowEndpointGroupId}");
  b.p("dataflowEndpointGroupId", () => input.dataflowEndpointGroupId, "{dataflowEndpointGroupId}", false);
  let body;
  b.m("DELETE").h(headers).b(body);
  return b.build();
}, "se_DeleteDataflowEndpointGroupCommand");
var se_DeleteEphemerisCommand = /* @__PURE__ */ __name(async (input, context) => {
  const b = (0, import_core.requestBuilder)(input, context);
  const headers = {};
  b.bp("/ephemeris/{ephemerisId}");
  b.p("ephemerisId", () => input.ephemerisId, "{ephemerisId}", false);
  let body;
  b.m("DELETE").h(headers).b(body);
  return b.build();
}, "se_DeleteEphemerisCommand");
var se_DeleteMissionProfileCommand = /* @__PURE__ */ __name(async (input, context) => {
  const b = (0, import_core.requestBuilder)(input, context);
  const headers = {};
  b.bp("/missionprofile/{missionProfileId}");
  b.p("missionProfileId", () => input.missionProfileId, "{missionProfileId}", false);
  let body;
  b.m("DELETE").h(headers).b(body);
  return b.build();
}, "se_DeleteMissionProfileCommand");
var se_DescribeContactCommand = /* @__PURE__ */ __name(async (input, context) => {
  const b = (0, import_core.requestBuilder)(input, context);
  const headers = {};
  b.bp("/contact/{contactId}");
  b.p("contactId", () => input.contactId, "{contactId}", false);
  let body;
  b.m("GET").h(headers).b(body);
  return b.build();
}, "se_DescribeContactCommand");
var se_DescribeEphemerisCommand = /* @__PURE__ */ __name(async (input, context) => {
  const b = (0, import_core.requestBuilder)(input, context);
  const headers = {};
  b.bp("/ephemeris/{ephemerisId}");
  b.p("ephemerisId", () => input.ephemerisId, "{ephemerisId}", false);
  let body;
  b.m("GET").h(headers).b(body);
  return b.build();
}, "se_DescribeEphemerisCommand");
var se_GetAgentConfigurationCommand = /* @__PURE__ */ __name(async (input, context) => {
  const b = (0, import_core.requestBuilder)(input, context);
  const headers = {};
  b.bp("/agent/{agentId}/configuration");
  b.p("agentId", () => input.agentId, "{agentId}", false);
  let body;
  b.m("GET").h(headers).b(body);
  return b.build();
}, "se_GetAgentConfigurationCommand");
var se_GetConfigCommand = /* @__PURE__ */ __name(async (input, context) => {
  const b = (0, import_core.requestBuilder)(input, context);
  const headers = {};
  b.bp("/config/{configType}/{configId}");
  b.p("configId", () => input.configId, "{configId}", false);
  b.p("configType", () => input.configType, "{configType}", false);
  let body;
  b.m("GET").h(headers).b(body);
  return b.build();
}, "se_GetConfigCommand");
var se_GetDataflowEndpointGroupCommand = /* @__PURE__ */ __name(async (input, context) => {
  const b = (0, import_core.requestBuilder)(input, context);
  const headers = {};
  b.bp("/dataflowEndpointGroup/{dataflowEndpointGroupId}");
  b.p("dataflowEndpointGroupId", () => input.dataflowEndpointGroupId, "{dataflowEndpointGroupId}", false);
  let body;
  b.m("GET").h(headers).b(body);
  return b.build();
}, "se_GetDataflowEndpointGroupCommand");
var se_GetMinuteUsageCommand = /* @__PURE__ */ __name(async (input, context) => {
  const b = (0, import_core.requestBuilder)(input, context);
  const headers = {
    "content-type": "application/json"
  };
  b.bp("/minute-usage");
  let body;
  body = JSON.stringify(
    (0, import_smithy_client.take)(input, {
      month: [],
      year: []
    })
  );
  b.m("POST").h(headers).b(body);
  return b.build();
}, "se_GetMinuteUsageCommand");
var se_GetMissionProfileCommand = /* @__PURE__ */ __name(async (input, context) => {
  const b = (0, import_core.requestBuilder)(input, context);
  const headers = {};
  b.bp("/missionprofile/{missionProfileId}");
  b.p("missionProfileId", () => input.missionProfileId, "{missionProfileId}", false);
  let body;
  b.m("GET").h(headers).b(body);
  return b.build();
}, "se_GetMissionProfileCommand");
var se_GetSatelliteCommand = /* @__PURE__ */ __name(async (input, context) => {
  const b = (0, import_core.requestBuilder)(input, context);
  const headers = {};
  b.bp("/satellite/{satelliteId}");
  b.p("satelliteId", () => input.satelliteId, "{satelliteId}", false);
  let body;
  b.m("GET").h(headers).b(body);
  return b.build();
}, "se_GetSatelliteCommand");
var se_ListConfigsCommand = /* @__PURE__ */ __name(async (input, context) => {
  const b = (0, import_core.requestBuilder)(input, context);
  const headers = {};
  b.bp("/config");
  const query = (0, import_smithy_client.map)({
    [_mR]: [() => input.maxResults !== void 0, () => input[_mR].toString()],
    [_nT]: [, input[_nT]]
  });
  let body;
  b.m("GET").h(headers).q(query).b(body);
  return b.build();
}, "se_ListConfigsCommand");
var se_ListContactsCommand = /* @__PURE__ */ __name(async (input, context) => {
  const b = (0, import_core.requestBuilder)(input, context);
  const headers = {
    "content-type": "application/json"
  };
  b.bp("/contacts");
  let body;
  body = JSON.stringify(
    (0, import_smithy_client.take)(input, {
      endTime: (_) => Math.round(_.getTime() / 1e3),
      groundStation: [],
      maxResults: [],
      missionProfileArn: [],
      nextToken: [],
      satelliteArn: [],
      startTime: (_) => Math.round(_.getTime() / 1e3),
      statusList: (_) => (0, import_smithy_client._json)(_)
    })
  );
  b.m("POST").h(headers).b(body);
  return b.build();
}, "se_ListContactsCommand");
var se_ListDataflowEndpointGroupsCommand = /* @__PURE__ */ __name(async (input, context) => {
  const b = (0, import_core.requestBuilder)(input, context);
  const headers = {};
  b.bp("/dataflowEndpointGroup");
  const query = (0, import_smithy_client.map)({
    [_mR]: [() => input.maxResults !== void 0, () => input[_mR].toString()],
    [_nT]: [, input[_nT]]
  });
  let body;
  b.m("GET").h(headers).q(query).b(body);
  return b.build();
}, "se_ListDataflowEndpointGroupsCommand");
var se_ListEphemeridesCommand = /* @__PURE__ */ __name(async (input, context) => {
  const b = (0, import_core.requestBuilder)(input, context);
  const headers = {
    "content-type": "application/json"
  };
  b.bp("/ephemerides");
  const query = (0, import_smithy_client.map)({
    [_mR]: [() => input.maxResults !== void 0, () => input[_mR].toString()],
    [_nT]: [, input[_nT]]
  });
  let body;
  body = JSON.stringify(
    (0, import_smithy_client.take)(input, {
      endTime: (_) => Math.round(_.getTime() / 1e3),
      satelliteId: [],
      startTime: (_) => Math.round(_.getTime() / 1e3),
      statusList: (_) => (0, import_smithy_client._json)(_)
    })
  );
  b.m("POST").h(headers).q(query).b(body);
  return b.build();
}, "se_ListEphemeridesCommand");
var se_ListGroundStationsCommand = /* @__PURE__ */ __name(async (input, context) => {
  const b = (0, import_core.requestBuilder)(input, context);
  const headers = {};
  b.bp("/groundstation");
  const query = (0, import_smithy_client.map)({
    [_sI]: [, input[_sI]],
    [_mR]: [() => input.maxResults !== void 0, () => input[_mR].toString()],
    [_nT]: [, input[_nT]]
  });
  let body;
  b.m("GET").h(headers).q(query).b(body);
  return b.build();
}, "se_ListGroundStationsCommand");
var se_ListMissionProfilesCommand = /* @__PURE__ */ __name(async (input, context) => {
  const b = (0, import_core.requestBuilder)(input, context);
  const headers = {};
  b.bp("/missionprofile");
  const query = (0, import_smithy_client.map)({
    [_mR]: [() => input.maxResults !== void 0, () => input[_mR].toString()],
    [_nT]: [, input[_nT]]
  });
  let body;
  b.m("GET").h(headers).q(query).b(body);
  return b.build();
}, "se_ListMissionProfilesCommand");
var se_ListSatellitesCommand = /* @__PURE__ */ __name(async (input, context) => {
  const b = (0, import_core.requestBuilder)(input, context);
  const headers = {};
  b.bp("/satellite");
  const query = (0, import_smithy_client.map)({
    [_mR]: [() => input.maxResults !== void 0, () => input[_mR].toString()],
    [_nT]: [, input[_nT]]
  });
  let body;
  b.m("GET").h(headers).q(query).b(body);
  return b.build();
}, "se_ListSatellitesCommand");
var se_ListTagsForResourceCommand = /* @__PURE__ */ __name(async (input, context) => {
  const b = (0, import_core.requestBuilder)(input, context);
  const headers = {};
  b.bp("/tags/{resourceArn}");
  b.p("resourceArn", () => input.resourceArn, "{resourceArn}", false);
  let body;
  b.m("GET").h(headers).b(body);
  return b.build();
}, "se_ListTagsForResourceCommand");
var se_RegisterAgentCommand = /* @__PURE__ */ __name(async (input, context) => {
  const b = (0, import_core.requestBuilder)(input, context);
  const headers = {
    "content-type": "application/json"
  };
  b.bp("/agent");
  let body;
  body = JSON.stringify(
    (0, import_smithy_client.take)(input, {
      agentDetails: (_) => (0, import_smithy_client._json)(_),
      discoveryData: (_) => (0, import_smithy_client._json)(_)
    })
  );
  b.m("POST").h(headers).b(body);
  return b.build();
}, "se_RegisterAgentCommand");
var se_ReserveContactCommand = /* @__PURE__ */ __name(async (input, context) => {
  const b = (0, import_core.requestBuilder)(input, context);
  const headers = {
    "content-type": "application/json"
  };
  b.bp("/contact");
  let body;
  body = JSON.stringify(
    (0, import_smithy_client.take)(input, {
      endTime: (_) => Math.round(_.getTime() / 1e3),
      groundStation: [],
      missionProfileArn: [],
      satelliteArn: [],
      startTime: (_) => Math.round(_.getTime() / 1e3),
      tags: (_) => (0, import_smithy_client._json)(_)
    })
  );
  b.m("POST").h(headers).b(body);
  return b.build();
}, "se_ReserveContactCommand");
var se_TagResourceCommand = /* @__PURE__ */ __name(async (input, context) => {
  const b = (0, import_core.requestBuilder)(input, context);
  const headers = {
    "content-type": "application/json"
  };
  b.bp("/tags/{resourceArn}");
  b.p("resourceArn", () => input.resourceArn, "{resourceArn}", false);
  let body;
  body = JSON.stringify(
    (0, import_smithy_client.take)(input, {
      tags: (_) => (0, import_smithy_client._json)(_)
    })
  );
  b.m("POST").h(headers).b(body);
  return b.build();
}, "se_TagResourceCommand");
var se_UntagResourceCommand = /* @__PURE__ */ __name(async (input, context) => {
  const b = (0, import_core.requestBuilder)(input, context);
  const headers = {};
  b.bp("/tags/{resourceArn}");
  b.p("resourceArn", () => input.resourceArn, "{resourceArn}", false);
  const query = (0, import_smithy_client.map)({
    [_tK]: [
      (0, import_smithy_client.expectNonNull)(input.tagKeys, `tagKeys`) != null,
      () => (input[_tK] || []).map((_entry) => _entry)
    ]
  });
  let body;
  b.m("DELETE").h(headers).q(query).b(body);
  return b.build();
}, "se_UntagResourceCommand");
var se_UpdateAgentStatusCommand = /* @__PURE__ */ __name(async (input, context) => {
  const b = (0, import_core.requestBuilder)(input, context);
  const headers = {
    "content-type": "application/json"
  };
  b.bp("/agent/{agentId}");
  b.p("agentId", () => input.agentId, "{agentId}", false);
  let body;
  body = JSON.stringify(
    (0, import_smithy_client.take)(input, {
      aggregateStatus: (_) => (0, import_smithy_client._json)(_),
      componentStatuses: (_) => (0, import_smithy_client._json)(_),
      taskId: []
    })
  );
  b.m("PUT").h(headers).b(body);
  return b.build();
}, "se_UpdateAgentStatusCommand");
var se_UpdateConfigCommand = /* @__PURE__ */ __name(async (input, context) => {
  const b = (0, import_core.requestBuilder)(input, context);
  const headers = {
    "content-type": "application/json"
  };
  b.bp("/config/{configType}/{configId}");
  b.p("configId", () => input.configId, "{configId}", false);
  b.p("configType", () => input.configType, "{configType}", false);
  let body;
  body = JSON.stringify(
    (0, import_smithy_client.take)(input, {
      configData: (_) => se_ConfigTypeData(_, context),
      name: []
    })
  );
  b.m("PUT").h(headers).b(body);
  return b.build();
}, "se_UpdateConfigCommand");
var se_UpdateEphemerisCommand = /* @__PURE__ */ __name(async (input, context) => {
  const b = (0, import_core.requestBuilder)(input, context);
  const headers = {
    "content-type": "application/json"
  };
  b.bp("/ephemeris/{ephemerisId}");
  b.p("ephemerisId", () => input.ephemerisId, "{ephemerisId}", false);
  let body;
  body = JSON.stringify(
    (0, import_smithy_client.take)(input, {
      enabled: [],
      name: [],
      priority: []
    })
  );
  b.m("PUT").h(headers).b(body);
  return b.build();
}, "se_UpdateEphemerisCommand");
var se_UpdateMissionProfileCommand = /* @__PURE__ */ __name(async (input, context) => {
  const b = (0, import_core.requestBuilder)(input, context);
  const headers = {
    "content-type": "application/json"
  };
  b.bp("/missionprofile/{missionProfileId}");
  b.p("missionProfileId", () => input.missionProfileId, "{missionProfileId}", false);
  let body;
  body = JSON.stringify(
    (0, import_smithy_client.take)(input, {
      contactPostPassDurationSeconds: [],
      contactPrePassDurationSeconds: [],
      dataflowEdges: (_) => (0, import_smithy_client._json)(_),
      minimumViableContactDurationSeconds: [],
      name: [],
      streamsKmsKey: (_) => (0, import_smithy_client._json)(_),
      streamsKmsRole: [],
      trackingConfigArn: []
    })
  );
  b.m("PUT").h(headers).b(body);
  return b.build();
}, "se_UpdateMissionProfileCommand");
var de_CancelContactCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode !== 200 && output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const contents = (0, import_smithy_client.map)({
    $metadata: deserializeMetadata(output)
  });
  const data = (0, import_smithy_client.expectNonNull)((0, import_smithy_client.expectObject)(await (0, import_core2.parseJsonBody)(output.body, context)), "body");
  const doc = (0, import_smithy_client.take)(data, {
    contactId: import_smithy_client.expectString
  });
  Object.assign(contents, doc);
  return contents;
}, "de_CancelContactCommand");
var de_CreateConfigCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode !== 200 && output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const contents = (0, import_smithy_client.map)({
    $metadata: deserializeMetadata(output)
  });
  const data = (0, import_smithy_client.expectNonNull)((0, import_smithy_client.expectObject)(await (0, import_core2.parseJsonBody)(output.body, context)), "body");
  const doc = (0, import_smithy_client.take)(data, {
    configArn: import_smithy_client.expectString,
    configId: import_smithy_client.expectString,
    configType: import_smithy_client.expectString
  });
  Object.assign(contents, doc);
  return contents;
}, "de_CreateConfigCommand");
var de_CreateDataflowEndpointGroupCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode !== 200 && output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const contents = (0, import_smithy_client.map)({
    $metadata: deserializeMetadata(output)
  });
  const data = (0, import_smithy_client.expectNonNull)((0, import_smithy_client.expectObject)(await (0, import_core2.parseJsonBody)(output.body, context)), "body");
  const doc = (0, import_smithy_client.take)(data, {
    dataflowEndpointGroupId: import_smithy_client.expectString
  });
  Object.assign(contents, doc);
  return contents;
}, "de_CreateDataflowEndpointGroupCommand");
var de_CreateEphemerisCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode !== 200 && output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const contents = (0, import_smithy_client.map)({
    $metadata: deserializeMetadata(output)
  });
  const data = (0, import_smithy_client.expectNonNull)((0, import_smithy_client.expectObject)(await (0, import_core2.parseJsonBody)(output.body, context)), "body");
  const doc = (0, import_smithy_client.take)(data, {
    ephemerisId: import_smithy_client.expectString
  });
  Object.assign(contents, doc);
  return contents;
}, "de_CreateEphemerisCommand");
var de_CreateMissionProfileCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode !== 200 && output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const contents = (0, import_smithy_client.map)({
    $metadata: deserializeMetadata(output)
  });
  const data = (0, import_smithy_client.expectNonNull)((0, import_smithy_client.expectObject)(await (0, import_core2.parseJsonBody)(output.body, context)), "body");
  const doc = (0, import_smithy_client.take)(data, {
    missionProfileId: import_smithy_client.expectString
  });
  Object.assign(contents, doc);
  return contents;
}, "de_CreateMissionProfileCommand");
var de_DeleteConfigCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode !== 200 && output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const contents = (0, import_smithy_client.map)({
    $metadata: deserializeMetadata(output)
  });
  const data = (0, import_smithy_client.expectNonNull)((0, import_smithy_client.expectObject)(await (0, import_core2.parseJsonBody)(output.body, context)), "body");
  const doc = (0, import_smithy_client.take)(data, {
    configArn: import_smithy_client.expectString,
    configId: import_smithy_client.expectString,
    configType: import_smithy_client.expectString
  });
  Object.assign(contents, doc);
  return contents;
}, "de_DeleteConfigCommand");
var de_DeleteDataflowEndpointGroupCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode !== 200 && output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const contents = (0, import_smithy_client.map)({
    $metadata: deserializeMetadata(output)
  });
  const data = (0, import_smithy_client.expectNonNull)((0, import_smithy_client.expectObject)(await (0, import_core2.parseJsonBody)(output.body, context)), "body");
  const doc = (0, import_smithy_client.take)(data, {
    dataflowEndpointGroupId: import_smithy_client.expectString
  });
  Object.assign(contents, doc);
  return contents;
}, "de_DeleteDataflowEndpointGroupCommand");
var de_DeleteEphemerisCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode !== 200 && output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const contents = (0, import_smithy_client.map)({
    $metadata: deserializeMetadata(output)
  });
  const data = (0, import_smithy_client.expectNonNull)((0, import_smithy_client.expectObject)(await (0, import_core2.parseJsonBody)(output.body, context)), "body");
  const doc = (0, import_smithy_client.take)(data, {
    ephemerisId: import_smithy_client.expectString
  });
  Object.assign(contents, doc);
  return contents;
}, "de_DeleteEphemerisCommand");
var de_DeleteMissionProfileCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode !== 200 && output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const contents = (0, import_smithy_client.map)({
    $metadata: deserializeMetadata(output)
  });
  const data = (0, import_smithy_client.expectNonNull)((0, import_smithy_client.expectObject)(await (0, import_core2.parseJsonBody)(output.body, context)), "body");
  const doc = (0, import_smithy_client.take)(data, {
    missionProfileId: import_smithy_client.expectString
  });
  Object.assign(contents, doc);
  return contents;
}, "de_DeleteMissionProfileCommand");
var de_DescribeContactCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode !== 200 && output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const contents = (0, import_smithy_client.map)({
    $metadata: deserializeMetadata(output)
  });
  const data = (0, import_smithy_client.expectNonNull)((0, import_smithy_client.expectObject)(await (0, import_core2.parseJsonBody)(output.body, context)), "body");
  const doc = (0, import_smithy_client.take)(data, {
    contactId: import_smithy_client.expectString,
    contactStatus: import_smithy_client.expectString,
    dataflowList: import_smithy_client._json,
    endTime: (_) => (0, import_smithy_client.expectNonNull)((0, import_smithy_client.parseEpochTimestamp)((0, import_smithy_client.expectNumber)(_))),
    errorMessage: import_smithy_client.expectString,
    groundStation: import_smithy_client.expectString,
    maximumElevation: (_) => de_Elevation(_, context),
    missionProfileArn: import_smithy_client.expectString,
    postPassEndTime: (_) => (0, import_smithy_client.expectNonNull)((0, import_smithy_client.parseEpochTimestamp)((0, import_smithy_client.expectNumber)(_))),
    prePassStartTime: (_) => (0, import_smithy_client.expectNonNull)((0, import_smithy_client.parseEpochTimestamp)((0, import_smithy_client.expectNumber)(_))),
    region: import_smithy_client.expectString,
    satelliteArn: import_smithy_client.expectString,
    startTime: (_) => (0, import_smithy_client.expectNonNull)((0, import_smithy_client.parseEpochTimestamp)((0, import_smithy_client.expectNumber)(_))),
    tags: import_smithy_client._json,
    visibilityEndTime: (_) => (0, import_smithy_client.expectNonNull)((0, import_smithy_client.parseEpochTimestamp)((0, import_smithy_client.expectNumber)(_))),
    visibilityStartTime: (_) => (0, import_smithy_client.expectNonNull)((0, import_smithy_client.parseEpochTimestamp)((0, import_smithy_client.expectNumber)(_)))
  });
  Object.assign(contents, doc);
  return contents;
}, "de_DescribeContactCommand");
var de_DescribeEphemerisCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode !== 200 && output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const contents = (0, import_smithy_client.map)({
    $metadata: deserializeMetadata(output)
  });
  const data = (0, import_smithy_client.expectNonNull)((0, import_smithy_client.expectObject)(await (0, import_core2.parseJsonBody)(output.body, context)), "body");
  const doc = (0, import_smithy_client.take)(data, {
    creationTime: (_) => (0, import_smithy_client.expectNonNull)((0, import_smithy_client.parseEpochTimestamp)((0, import_smithy_client.expectNumber)(_))),
    enabled: import_smithy_client.expectBoolean,
    ephemerisId: import_smithy_client.expectString,
    invalidReason: import_smithy_client.expectString,
    name: import_smithy_client.expectString,
    priority: import_smithy_client.expectInt32,
    satelliteId: import_smithy_client.expectString,
    status: import_smithy_client.expectString,
    suppliedData: (_) => (0, import_smithy_client._json)((0, import_core2.awsExpectUnion)(_)),
    tags: import_smithy_client._json
  });
  Object.assign(contents, doc);
  return contents;
}, "de_DescribeEphemerisCommand");
var de_GetAgentConfigurationCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode !== 200 && output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const contents = (0, import_smithy_client.map)({
    $metadata: deserializeMetadata(output)
  });
  const data = (0, import_smithy_client.expectNonNull)((0, import_smithy_client.expectObject)(await (0, import_core2.parseJsonBody)(output.body, context)), "body");
  const doc = (0, import_smithy_client.take)(data, {
    agentId: import_smithy_client.expectString,
    taskingDocument: import_smithy_client.expectString
  });
  Object.assign(contents, doc);
  return contents;
}, "de_GetAgentConfigurationCommand");
var de_GetConfigCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode !== 200 && output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const contents = (0, import_smithy_client.map)({
    $metadata: deserializeMetadata(output)
  });
  const data = (0, import_smithy_client.expectNonNull)((0, import_smithy_client.expectObject)(await (0, import_core2.parseJsonBody)(output.body, context)), "body");
  const doc = (0, import_smithy_client.take)(data, {
    configArn: import_smithy_client.expectString,
    configData: (_) => de_ConfigTypeData((0, import_core2.awsExpectUnion)(_), context),
    configId: import_smithy_client.expectString,
    configType: import_smithy_client.expectString,
    name: import_smithy_client.expectString,
    tags: import_smithy_client._json
  });
  Object.assign(contents, doc);
  return contents;
}, "de_GetConfigCommand");
var de_GetDataflowEndpointGroupCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode !== 200 && output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const contents = (0, import_smithy_client.map)({
    $metadata: deserializeMetadata(output)
  });
  const data = (0, import_smithy_client.expectNonNull)((0, import_smithy_client.expectObject)(await (0, import_core2.parseJsonBody)(output.body, context)), "body");
  const doc = (0, import_smithy_client.take)(data, {
    contactPostPassDurationSeconds: import_smithy_client.expectInt32,
    contactPrePassDurationSeconds: import_smithy_client.expectInt32,
    dataflowEndpointGroupArn: import_smithy_client.expectString,
    dataflowEndpointGroupId: import_smithy_client.expectString,
    endpointsDetails: import_smithy_client._json,
    tags: import_smithy_client._json
  });
  Object.assign(contents, doc);
  return contents;
}, "de_GetDataflowEndpointGroupCommand");
var de_GetMinuteUsageCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode !== 200 && output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const contents = (0, import_smithy_client.map)({
    $metadata: deserializeMetadata(output)
  });
  const data = (0, import_smithy_client.expectNonNull)((0, import_smithy_client.expectObject)(await (0, import_core2.parseJsonBody)(output.body, context)), "body");
  const doc = (0, import_smithy_client.take)(data, {
    estimatedMinutesRemaining: import_smithy_client.expectInt32,
    isReservedMinutesCustomer: import_smithy_client.expectBoolean,
    totalReservedMinuteAllocation: import_smithy_client.expectInt32,
    totalScheduledMinutes: import_smithy_client.expectInt32,
    upcomingMinutesScheduled: import_smithy_client.expectInt32
  });
  Object.assign(contents, doc);
  return contents;
}, "de_GetMinuteUsageCommand");
var de_GetMissionProfileCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode !== 200 && output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const contents = (0, import_smithy_client.map)({
    $metadata: deserializeMetadata(output)
  });
  const data = (0, import_smithy_client.expectNonNull)((0, import_smithy_client.expectObject)(await (0, import_core2.parseJsonBody)(output.body, context)), "body");
  const doc = (0, import_smithy_client.take)(data, {
    contactPostPassDurationSeconds: import_smithy_client.expectInt32,
    contactPrePassDurationSeconds: import_smithy_client.expectInt32,
    dataflowEdges: import_smithy_client._json,
    minimumViableContactDurationSeconds: import_smithy_client.expectInt32,
    missionProfileArn: import_smithy_client.expectString,
    missionProfileId: import_smithy_client.expectString,
    name: import_smithy_client.expectString,
    region: import_smithy_client.expectString,
    streamsKmsKey: (_) => (0, import_smithy_client._json)((0, import_core2.awsExpectUnion)(_)),
    streamsKmsRole: import_smithy_client.expectString,
    tags: import_smithy_client._json,
    trackingConfigArn: import_smithy_client.expectString
  });
  Object.assign(contents, doc);
  return contents;
}, "de_GetMissionProfileCommand");
var de_GetSatelliteCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode !== 200 && output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const contents = (0, import_smithy_client.map)({
    $metadata: deserializeMetadata(output)
  });
  const data = (0, import_smithy_client.expectNonNull)((0, import_smithy_client.expectObject)(await (0, import_core2.parseJsonBody)(output.body, context)), "body");
  const doc = (0, import_smithy_client.take)(data, {
    currentEphemeris: (_) => de_EphemerisMetaData(_, context),
    groundStations: import_smithy_client._json,
    noradSatelliteID: import_smithy_client.expectInt32,
    satelliteArn: import_smithy_client.expectString,
    satelliteId: import_smithy_client.expectString
  });
  Object.assign(contents, doc);
  return contents;
}, "de_GetSatelliteCommand");
var de_ListConfigsCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode !== 200 && output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const contents = (0, import_smithy_client.map)({
    $metadata: deserializeMetadata(output)
  });
  const data = (0, import_smithy_client.expectNonNull)((0, import_smithy_client.expectObject)(await (0, import_core2.parseJsonBody)(output.body, context)), "body");
  const doc = (0, import_smithy_client.take)(data, {
    configList: import_smithy_client._json,
    nextToken: import_smithy_client.expectString
  });
  Object.assign(contents, doc);
  return contents;
}, "de_ListConfigsCommand");
var de_ListContactsCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode !== 200 && output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const contents = (0, import_smithy_client.map)({
    $metadata: deserializeMetadata(output)
  });
  const data = (0, import_smithy_client.expectNonNull)((0, import_smithy_client.expectObject)(await (0, import_core2.parseJsonBody)(output.body, context)), "body");
  const doc = (0, import_smithy_client.take)(data, {
    contactList: (_) => de_ContactList(_, context),
    nextToken: import_smithy_client.expectString
  });
  Object.assign(contents, doc);
  return contents;
}, "de_ListContactsCommand");
var de_ListDataflowEndpointGroupsCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode !== 200 && output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const contents = (0, import_smithy_client.map)({
    $metadata: deserializeMetadata(output)
  });
  const data = (0, import_smithy_client.expectNonNull)((0, import_smithy_client.expectObject)(await (0, import_core2.parseJsonBody)(output.body, context)), "body");
  const doc = (0, import_smithy_client.take)(data, {
    dataflowEndpointGroupList: import_smithy_client._json,
    nextToken: import_smithy_client.expectString
  });
  Object.assign(contents, doc);
  return contents;
}, "de_ListDataflowEndpointGroupsCommand");
var de_ListEphemeridesCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode !== 200 && output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const contents = (0, import_smithy_client.map)({
    $metadata: deserializeMetadata(output)
  });
  const data = (0, import_smithy_client.expectNonNull)((0, import_smithy_client.expectObject)(await (0, import_core2.parseJsonBody)(output.body, context)), "body");
  const doc = (0, import_smithy_client.take)(data, {
    ephemerides: (_) => de_EphemeridesList(_, context),
    nextToken: import_smithy_client.expectString
  });
  Object.assign(contents, doc);
  return contents;
}, "de_ListEphemeridesCommand");
var de_ListGroundStationsCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode !== 200 && output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const contents = (0, import_smithy_client.map)({
    $metadata: deserializeMetadata(output)
  });
  const data = (0, import_smithy_client.expectNonNull)((0, import_smithy_client.expectObject)(await (0, import_core2.parseJsonBody)(output.body, context)), "body");
  const doc = (0, import_smithy_client.take)(data, {
    groundStationList: import_smithy_client._json,
    nextToken: import_smithy_client.expectString
  });
  Object.assign(contents, doc);
  return contents;
}, "de_ListGroundStationsCommand");
var de_ListMissionProfilesCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode !== 200 && output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const contents = (0, import_smithy_client.map)({
    $metadata: deserializeMetadata(output)
  });
  const data = (0, import_smithy_client.expectNonNull)((0, import_smithy_client.expectObject)(await (0, import_core2.parseJsonBody)(output.body, context)), "body");
  const doc = (0, import_smithy_client.take)(data, {
    missionProfileList: import_smithy_client._json,
    nextToken: import_smithy_client.expectString
  });
  Object.assign(contents, doc);
  return contents;
}, "de_ListMissionProfilesCommand");
var de_ListSatellitesCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode !== 200 && output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const contents = (0, import_smithy_client.map)({
    $metadata: deserializeMetadata(output)
  });
  const data = (0, import_smithy_client.expectNonNull)((0, import_smithy_client.expectObject)(await (0, import_core2.parseJsonBody)(output.body, context)), "body");
  const doc = (0, import_smithy_client.take)(data, {
    nextToken: import_smithy_client.expectString,
    satellites: (_) => de_SatelliteList(_, context)
  });
  Object.assign(contents, doc);
  return contents;
}, "de_ListSatellitesCommand");
var de_ListTagsForResourceCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode !== 200 && output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const contents = (0, import_smithy_client.map)({
    $metadata: deserializeMetadata(output)
  });
  const data = (0, import_smithy_client.expectNonNull)((0, import_smithy_client.expectObject)(await (0, import_core2.parseJsonBody)(output.body, context)), "body");
  const doc = (0, import_smithy_client.take)(data, {
    tags: import_smithy_client._json
  });
  Object.assign(contents, doc);
  return contents;
}, "de_ListTagsForResourceCommand");
var de_RegisterAgentCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode !== 200 && output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const contents = (0, import_smithy_client.map)({
    $metadata: deserializeMetadata(output)
  });
  const data = (0, import_smithy_client.expectNonNull)((0, import_smithy_client.expectObject)(await (0, import_core2.parseJsonBody)(output.body, context)), "body");
  const doc = (0, import_smithy_client.take)(data, {
    agentId: import_smithy_client.expectString
  });
  Object.assign(contents, doc);
  return contents;
}, "de_RegisterAgentCommand");
var de_ReserveContactCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode !== 200 && output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const contents = (0, import_smithy_client.map)({
    $metadata: deserializeMetadata(output)
  });
  const data = (0, import_smithy_client.expectNonNull)((0, import_smithy_client.expectObject)(await (0, import_core2.parseJsonBody)(output.body, context)), "body");
  const doc = (0, import_smithy_client.take)(data, {
    contactId: import_smithy_client.expectString
  });
  Object.assign(contents, doc);
  return contents;
}, "de_ReserveContactCommand");
var de_TagResourceCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode !== 200 && output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const contents = (0, import_smithy_client.map)({
    $metadata: deserializeMetadata(output)
  });
  await (0, import_smithy_client.collectBody)(output.body, context);
  return contents;
}, "de_TagResourceCommand");
var de_UntagResourceCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode !== 200 && output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const contents = (0, import_smithy_client.map)({
    $metadata: deserializeMetadata(output)
  });
  await (0, import_smithy_client.collectBody)(output.body, context);
  return contents;
}, "de_UntagResourceCommand");
var de_UpdateAgentStatusCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode !== 200 && output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const contents = (0, import_smithy_client.map)({
    $metadata: deserializeMetadata(output)
  });
  const data = (0, import_smithy_client.expectNonNull)((0, import_smithy_client.expectObject)(await (0, import_core2.parseJsonBody)(output.body, context)), "body");
  const doc = (0, import_smithy_client.take)(data, {
    agentId: import_smithy_client.expectString
  });
  Object.assign(contents, doc);
  return contents;
}, "de_UpdateAgentStatusCommand");
var de_UpdateConfigCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode !== 200 && output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const contents = (0, import_smithy_client.map)({
    $metadata: deserializeMetadata(output)
  });
  const data = (0, import_smithy_client.expectNonNull)((0, import_smithy_client.expectObject)(await (0, import_core2.parseJsonBody)(output.body, context)), "body");
  const doc = (0, import_smithy_client.take)(data, {
    configArn: import_smithy_client.expectString,
    configId: import_smithy_client.expectString,
    configType: import_smithy_client.expectString
  });
  Object.assign(contents, doc);
  return contents;
}, "de_UpdateConfigCommand");
var de_UpdateEphemerisCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode !== 200 && output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const contents = (0, import_smithy_client.map)({
    $metadata: deserializeMetadata(output)
  });
  const data = (0, import_smithy_client.expectNonNull)((0, import_smithy_client.expectObject)(await (0, import_core2.parseJsonBody)(output.body, context)), "body");
  const doc = (0, import_smithy_client.take)(data, {
    ephemerisId: import_smithy_client.expectString
  });
  Object.assign(contents, doc);
  return contents;
}, "de_UpdateEphemerisCommand");
var de_UpdateMissionProfileCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode !== 200 && output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const contents = (0, import_smithy_client.map)({
    $metadata: deserializeMetadata(output)
  });
  const data = (0, import_smithy_client.expectNonNull)((0, import_smithy_client.expectObject)(await (0, import_core2.parseJsonBody)(output.body, context)), "body");
  const doc = (0, import_smithy_client.take)(data, {
    missionProfileId: import_smithy_client.expectString
  });
  Object.assign(contents, doc);
  return contents;
}, "de_UpdateMissionProfileCommand");
var de_CommandError = /* @__PURE__ */ __name(async (output, context) => {
  const parsedOutput = {
    ...output,
    body: await (0, import_core2.parseJsonErrorBody)(output.body, context)
  };
  const errorCode = (0, import_core2.loadRestJsonErrorCode)(output, parsedOutput.body);
  switch (errorCode) {
    case "DependencyException":
    case "com.amazonaws.groundstation#DependencyException":
      throw await de_DependencyExceptionRes(parsedOutput, context);
    case "InvalidParameterException":
    case "com.amazonaws.groundstation#InvalidParameterException":
      throw await de_InvalidParameterExceptionRes(parsedOutput, context);
    case "ResourceNotFoundException":
    case "com.amazonaws.groundstation#ResourceNotFoundException":
      throw await de_ResourceNotFoundExceptionRes(parsedOutput, context);
    case "ResourceLimitExceededException":
    case "com.amazonaws.groundstation#ResourceLimitExceededException":
      throw await de_ResourceLimitExceededExceptionRes(parsedOutput, context);
    default:
      const parsedBody = parsedOutput.body;
      return throwDefaultError({
        output,
        parsedBody,
        errorCode
      });
  }
}, "de_CommandError");
var throwDefaultError = (0, import_smithy_client.withBaseException)(GroundStationServiceException);
var de_DependencyExceptionRes = /* @__PURE__ */ __name(async (parsedOutput, context) => {
  const contents = (0, import_smithy_client.map)({});
  const data = parsedOutput.body;
  const doc = (0, import_smithy_client.take)(data, {
    message: import_smithy_client.expectString,
    parameterName: import_smithy_client.expectString
  });
  Object.assign(contents, doc);
  const exception = new DependencyException({
    $metadata: deserializeMetadata(parsedOutput),
    ...contents
  });
  return (0, import_smithy_client.decorateServiceException)(exception, parsedOutput.body);
}, "de_DependencyExceptionRes");
var de_InvalidParameterExceptionRes = /* @__PURE__ */ __name(async (parsedOutput, context) => {
  const contents = (0, import_smithy_client.map)({});
  const data = parsedOutput.body;
  const doc = (0, import_smithy_client.take)(data, {
    message: import_smithy_client.expectString,
    parameterName: import_smithy_client.expectString
  });
  Object.assign(contents, doc);
  const exception = new InvalidParameterException({
    $metadata: deserializeMetadata(parsedOutput),
    ...contents
  });
  return (0, import_smithy_client.decorateServiceException)(exception, parsedOutput.body);
}, "de_InvalidParameterExceptionRes");
var de_ResourceLimitExceededExceptionRes = /* @__PURE__ */ __name(async (parsedOutput, context) => {
  const contents = (0, import_smithy_client.map)({});
  const data = parsedOutput.body;
  const doc = (0, import_smithy_client.take)(data, {
    message: import_smithy_client.expectString,
    parameterName: import_smithy_client.expectString
  });
  Object.assign(contents, doc);
  const exception = new ResourceLimitExceededException({
    $metadata: deserializeMetadata(parsedOutput),
    ...contents
  });
  return (0, import_smithy_client.decorateServiceException)(exception, parsedOutput.body);
}, "de_ResourceLimitExceededExceptionRes");
var de_ResourceNotFoundExceptionRes = /* @__PURE__ */ __name(async (parsedOutput, context) => {
  const contents = (0, import_smithy_client.map)({});
  const data = parsedOutput.body;
  const doc = (0, import_smithy_client.take)(data, {
    message: import_smithy_client.expectString
  });
  Object.assign(contents, doc);
  const exception = new ResourceNotFoundException({
    $metadata: deserializeMetadata(parsedOutput),
    ...contents
  });
  return (0, import_smithy_client.decorateServiceException)(exception, parsedOutput.body);
}, "de_ResourceNotFoundExceptionRes");
var se_AntennaDownlinkConfig = /* @__PURE__ */ __name((input, context) => {
  return (0, import_smithy_client.take)(input, {
    spectrumConfig: (_) => se_SpectrumConfig(_, context)
  });
}, "se_AntennaDownlinkConfig");
var se_AntennaDownlinkDemodDecodeConfig = /* @__PURE__ */ __name((input, context) => {
  return (0, import_smithy_client.take)(input, {
    decodeConfig: import_smithy_client._json,
    demodulationConfig: import_smithy_client._json,
    spectrumConfig: (_) => se_SpectrumConfig(_, context)
  });
}, "se_AntennaDownlinkDemodDecodeConfig");
var se_AntennaUplinkConfig = /* @__PURE__ */ __name((input, context) => {
  return (0, import_smithy_client.take)(input, {
    spectrumConfig: (_) => se_UplinkSpectrumConfig(_, context),
    targetEirp: (_) => se_Eirp(_, context),
    transmitDisabled: []
  });
}, "se_AntennaUplinkConfig");
var se_ConfigTypeData = /* @__PURE__ */ __name((input, context) => {
  return ConfigTypeData.visit(input, {
    antennaDownlinkConfig: (value) => ({ antennaDownlinkConfig: se_AntennaDownlinkConfig(value, context) }),
    antennaDownlinkDemodDecodeConfig: (value) => ({
      antennaDownlinkDemodDecodeConfig: se_AntennaDownlinkDemodDecodeConfig(value, context)
    }),
    antennaUplinkConfig: (value) => ({ antennaUplinkConfig: se_AntennaUplinkConfig(value, context) }),
    dataflowEndpointConfig: (value) => ({ dataflowEndpointConfig: (0, import_smithy_client._json)(value) }),
    s3RecordingConfig: (value) => ({ s3RecordingConfig: (0, import_smithy_client._json)(value) }),
    trackingConfig: (value) => ({ trackingConfig: (0, import_smithy_client._json)(value) }),
    uplinkEchoConfig: (value) => ({ uplinkEchoConfig: (0, import_smithy_client._json)(value) }),
    _: (name, value) => ({ name: value })
  });
}, "se_ConfigTypeData");
var se_Eirp = /* @__PURE__ */ __name((input, context) => {
  return (0, import_smithy_client.take)(input, {
    units: [],
    value: import_smithy_client.serializeFloat
  });
}, "se_Eirp");
var se_EphemerisData = /* @__PURE__ */ __name((input, context) => {
  return EphemerisData.visit(input, {
    oem: (value) => ({ oem: (0, import_smithy_client._json)(value) }),
    tle: (value) => ({ tle: se_TLEEphemeris(value, context) }),
    _: (name, value) => ({ name: value })
  });
}, "se_EphemerisData");
var se_Frequency = /* @__PURE__ */ __name((input, context) => {
  return (0, import_smithy_client.take)(input, {
    units: [],
    value: import_smithy_client.serializeFloat
  });
}, "se_Frequency");
var se_FrequencyBandwidth = /* @__PURE__ */ __name((input, context) => {
  return (0, import_smithy_client.take)(input, {
    units: [],
    value: import_smithy_client.serializeFloat
  });
}, "se_FrequencyBandwidth");
var se_SpectrumConfig = /* @__PURE__ */ __name((input, context) => {
  return (0, import_smithy_client.take)(input, {
    bandwidth: (_) => se_FrequencyBandwidth(_, context),
    centerFrequency: (_) => se_Frequency(_, context),
    polarization: []
  });
}, "se_SpectrumConfig");
var se_TimeRange = /* @__PURE__ */ __name((input, context) => {
  return (0, import_smithy_client.take)(input, {
    endTime: (_) => Math.round(_.getTime() / 1e3),
    startTime: (_) => Math.round(_.getTime() / 1e3)
  });
}, "se_TimeRange");
var se_TLEData = /* @__PURE__ */ __name((input, context) => {
  return (0, import_smithy_client.take)(input, {
    tleLine1: [],
    tleLine2: [],
    validTimeRange: (_) => se_TimeRange(_, context)
  });
}, "se_TLEData");
var se_TLEDataList = /* @__PURE__ */ __name((input, context) => {
  return input.filter((e) => e != null).map((entry) => {
    return se_TLEData(entry, context);
  });
}, "se_TLEDataList");
var se_TLEEphemeris = /* @__PURE__ */ __name((input, context) => {
  return (0, import_smithy_client.take)(input, {
    s3Object: import_smithy_client._json,
    tleData: (_) => se_TLEDataList(_, context)
  });
}, "se_TLEEphemeris");
var se_UplinkSpectrumConfig = /* @__PURE__ */ __name((input, context) => {
  return (0, import_smithy_client.take)(input, {
    centerFrequency: (_) => se_Frequency(_, context),
    polarization: []
  });
}, "se_UplinkSpectrumConfig");
var de_AntennaDownlinkConfig = /* @__PURE__ */ __name((output, context) => {
  return (0, import_smithy_client.take)(output, {
    spectrumConfig: (_) => de_SpectrumConfig(_, context)
  });
}, "de_AntennaDownlinkConfig");
var de_AntennaDownlinkDemodDecodeConfig = /* @__PURE__ */ __name((output, context) => {
  return (0, import_smithy_client.take)(output, {
    decodeConfig: import_smithy_client._json,
    demodulationConfig: import_smithy_client._json,
    spectrumConfig: (_) => de_SpectrumConfig(_, context)
  });
}, "de_AntennaDownlinkDemodDecodeConfig");
var de_AntennaUplinkConfig = /* @__PURE__ */ __name((output, context) => {
  return (0, import_smithy_client.take)(output, {
    spectrumConfig: (_) => de_UplinkSpectrumConfig(_, context),
    targetEirp: (_) => de_Eirp(_, context),
    transmitDisabled: import_smithy_client.expectBoolean
  });
}, "de_AntennaUplinkConfig");
var de_ConfigTypeData = /* @__PURE__ */ __name((output, context) => {
  if (output.antennaDownlinkConfig != null) {
    return {
      antennaDownlinkConfig: de_AntennaDownlinkConfig(output.antennaDownlinkConfig, context)
    };
  }
  if (output.antennaDownlinkDemodDecodeConfig != null) {
    return {
      antennaDownlinkDemodDecodeConfig: de_AntennaDownlinkDemodDecodeConfig(
        output.antennaDownlinkDemodDecodeConfig,
        context
      )
    };
  }
  if (output.antennaUplinkConfig != null) {
    return {
      antennaUplinkConfig: de_AntennaUplinkConfig(output.antennaUplinkConfig, context)
    };
  }
  if (output.dataflowEndpointConfig != null) {
    return {
      dataflowEndpointConfig: (0, import_smithy_client._json)(output.dataflowEndpointConfig)
    };
  }
  if (output.s3RecordingConfig != null) {
    return {
      s3RecordingConfig: (0, import_smithy_client._json)(output.s3RecordingConfig)
    };
  }
  if (output.trackingConfig != null) {
    return {
      trackingConfig: (0, import_smithy_client._json)(output.trackingConfig)
    };
  }
  if (output.uplinkEchoConfig != null) {
    return {
      uplinkEchoConfig: (0, import_smithy_client._json)(output.uplinkEchoConfig)
    };
  }
  return { $unknown: Object.entries(output)[0] };
}, "de_ConfigTypeData");
var de_ContactData = /* @__PURE__ */ __name((output, context) => {
  return (0, import_smithy_client.take)(output, {
    contactId: import_smithy_client.expectString,
    contactStatus: import_smithy_client.expectString,
    endTime: (_) => (0, import_smithy_client.expectNonNull)((0, import_smithy_client.parseEpochTimestamp)((0, import_smithy_client.expectNumber)(_))),
    errorMessage: import_smithy_client.expectString,
    groundStation: import_smithy_client.expectString,
    maximumElevation: (_) => de_Elevation(_, context),
    missionProfileArn: import_smithy_client.expectString,
    postPassEndTime: (_) => (0, import_smithy_client.expectNonNull)((0, import_smithy_client.parseEpochTimestamp)((0, import_smithy_client.expectNumber)(_))),
    prePassStartTime: (_) => (0, import_smithy_client.expectNonNull)((0, import_smithy_client.parseEpochTimestamp)((0, import_smithy_client.expectNumber)(_))),
    region: import_smithy_client.expectString,
    satelliteArn: import_smithy_client.expectString,
    startTime: (_) => (0, import_smithy_client.expectNonNull)((0, import_smithy_client.parseEpochTimestamp)((0, import_smithy_client.expectNumber)(_))),
    tags: import_smithy_client._json,
    visibilityEndTime: (_) => (0, import_smithy_client.expectNonNull)((0, import_smithy_client.parseEpochTimestamp)((0, import_smithy_client.expectNumber)(_))),
    visibilityStartTime: (_) => (0, import_smithy_client.expectNonNull)((0, import_smithy_client.parseEpochTimestamp)((0, import_smithy_client.expectNumber)(_)))
  });
}, "de_ContactData");
var de_ContactList = /* @__PURE__ */ __name((output, context) => {
  const retVal = (output || []).filter((e) => e != null).map((entry) => {
    return de_ContactData(entry, context);
  });
  return retVal;
}, "de_ContactList");
var de_Eirp = /* @__PURE__ */ __name((output, context) => {
  return (0, import_smithy_client.take)(output, {
    units: import_smithy_client.expectString,
    value: import_smithy_client.limitedParseDouble
  });
}, "de_Eirp");
var de_Elevation = /* @__PURE__ */ __name((output, context) => {
  return (0, import_smithy_client.take)(output, {
    unit: import_smithy_client.expectString,
    value: import_smithy_client.limitedParseDouble
  });
}, "de_Elevation");
var de_EphemeridesList = /* @__PURE__ */ __name((output, context) => {
  const retVal = (output || []).filter((e) => e != null).map((entry) => {
    return de_EphemerisItem(entry, context);
  });
  return retVal;
}, "de_EphemeridesList");
var de_EphemerisItem = /* @__PURE__ */ __name((output, context) => {
  return (0, import_smithy_client.take)(output, {
    creationTime: (_) => (0, import_smithy_client.expectNonNull)((0, import_smithy_client.parseEpochTimestamp)((0, import_smithy_client.expectNumber)(_))),
    enabled: import_smithy_client.expectBoolean,
    ephemerisId: import_smithy_client.expectString,
    name: import_smithy_client.expectString,
    priority: import_smithy_client.expectInt32,
    sourceS3Object: import_smithy_client._json,
    status: import_smithy_client.expectString
  });
}, "de_EphemerisItem");
var de_EphemerisMetaData = /* @__PURE__ */ __name((output, context) => {
  return (0, import_smithy_client.take)(output, {
    ephemerisId: import_smithy_client.expectString,
    epoch: (_) => (0, import_smithy_client.expectNonNull)((0, import_smithy_client.parseEpochTimestamp)((0, import_smithy_client.expectNumber)(_))),
    name: import_smithy_client.expectString,
    source: import_smithy_client.expectString
  });
}, "de_EphemerisMetaData");
var de_Frequency = /* @__PURE__ */ __name((output, context) => {
  return (0, import_smithy_client.take)(output, {
    units: import_smithy_client.expectString,
    value: import_smithy_client.limitedParseDouble
  });
}, "de_Frequency");
var de_FrequencyBandwidth = /* @__PURE__ */ __name((output, context) => {
  return (0, import_smithy_client.take)(output, {
    units: import_smithy_client.expectString,
    value: import_smithy_client.limitedParseDouble
  });
}, "de_FrequencyBandwidth");
var de_SatelliteList = /* @__PURE__ */ __name((output, context) => {
  const retVal = (output || []).filter((e) => e != null).map((entry) => {
    return de_SatelliteListItem(entry, context);
  });
  return retVal;
}, "de_SatelliteList");
var de_SatelliteListItem = /* @__PURE__ */ __name((output, context) => {
  return (0, import_smithy_client.take)(output, {
    currentEphemeris: (_) => de_EphemerisMetaData(_, context),
    groundStations: import_smithy_client._json,
    noradSatelliteID: import_smithy_client.expectInt32,
    satelliteArn: import_smithy_client.expectString,
    satelliteId: import_smithy_client.expectString
  });
}, "de_SatelliteListItem");
var de_SpectrumConfig = /* @__PURE__ */ __name((output, context) => {
  return (0, import_smithy_client.take)(output, {
    bandwidth: (_) => de_FrequencyBandwidth(_, context),
    centerFrequency: (_) => de_Frequency(_, context),
    polarization: import_smithy_client.expectString
  });
}, "de_SpectrumConfig");
var de_UplinkSpectrumConfig = /* @__PURE__ */ __name((output, context) => {
  return (0, import_smithy_client.take)(output, {
    centerFrequency: (_) => de_Frequency(_, context),
    polarization: import_smithy_client.expectString
  });
}, "de_UplinkSpectrumConfig");
var deserializeMetadata = /* @__PURE__ */ __name((output) => ({
  httpStatusCode: output.statusCode,
  requestId: output.headers["x-amzn-requestid"] ?? output.headers["x-amzn-request-id"] ?? output.headers["x-amz-request-id"],
  extendedRequestId: output.headers["x-amz-id-2"],
  cfId: output.headers["x-amz-cf-id"]
}), "deserializeMetadata");
var _mR = "maxResults";
var _nT = "nextToken";
var _sI = "satelliteId";
var _tK = "tagKeys";

// src/commands/CancelContactCommand.ts
var _CancelContactCommand = class _CancelContactCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("GroundStation", "CancelContact", {}).n("GroundStationClient", "CancelContactCommand").f(void 0, void 0).ser(se_CancelContactCommand).de(de_CancelContactCommand).build() {
};
__name(_CancelContactCommand, "CancelContactCommand");
var CancelContactCommand = _CancelContactCommand;

// src/commands/CreateConfigCommand.ts




var _CreateConfigCommand = class _CreateConfigCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("GroundStation", "CreateConfig", {}).n("GroundStationClient", "CreateConfigCommand").f(void 0, void 0).ser(se_CreateConfigCommand).de(de_CreateConfigCommand).build() {
};
__name(_CreateConfigCommand, "CreateConfigCommand");
var CreateConfigCommand = _CreateConfigCommand;

// src/commands/CreateDataflowEndpointGroupCommand.ts




var _CreateDataflowEndpointGroupCommand = class _CreateDataflowEndpointGroupCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("GroundStation", "CreateDataflowEndpointGroup", {}).n("GroundStationClient", "CreateDataflowEndpointGroupCommand").f(void 0, void 0).ser(se_CreateDataflowEndpointGroupCommand).de(de_CreateDataflowEndpointGroupCommand).build() {
};
__name(_CreateDataflowEndpointGroupCommand, "CreateDataflowEndpointGroupCommand");
var CreateDataflowEndpointGroupCommand = _CreateDataflowEndpointGroupCommand;

// src/commands/CreateEphemerisCommand.ts




var _CreateEphemerisCommand = class _CreateEphemerisCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("GroundStation", "CreateEphemeris", {}).n("GroundStationClient", "CreateEphemerisCommand").f(void 0, void 0).ser(se_CreateEphemerisCommand).de(de_CreateEphemerisCommand).build() {
};
__name(_CreateEphemerisCommand, "CreateEphemerisCommand");
var CreateEphemerisCommand = _CreateEphemerisCommand;

// src/commands/CreateMissionProfileCommand.ts




var _CreateMissionProfileCommand = class _CreateMissionProfileCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("GroundStation", "CreateMissionProfile", {}).n("GroundStationClient", "CreateMissionProfileCommand").f(void 0, void 0).ser(se_CreateMissionProfileCommand).de(de_CreateMissionProfileCommand).build() {
};
__name(_CreateMissionProfileCommand, "CreateMissionProfileCommand");
var CreateMissionProfileCommand = _CreateMissionProfileCommand;

// src/commands/DeleteConfigCommand.ts




var _DeleteConfigCommand = class _DeleteConfigCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("GroundStation", "DeleteConfig", {}).n("GroundStationClient", "DeleteConfigCommand").f(void 0, void 0).ser(se_DeleteConfigCommand).de(de_DeleteConfigCommand).build() {
};
__name(_DeleteConfigCommand, "DeleteConfigCommand");
var DeleteConfigCommand = _DeleteConfigCommand;

// src/commands/DeleteDataflowEndpointGroupCommand.ts




var _DeleteDataflowEndpointGroupCommand = class _DeleteDataflowEndpointGroupCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("GroundStation", "DeleteDataflowEndpointGroup", {}).n("GroundStationClient", "DeleteDataflowEndpointGroupCommand").f(void 0, void 0).ser(se_DeleteDataflowEndpointGroupCommand).de(de_DeleteDataflowEndpointGroupCommand).build() {
};
__name(_DeleteDataflowEndpointGroupCommand, "DeleteDataflowEndpointGroupCommand");
var DeleteDataflowEndpointGroupCommand = _DeleteDataflowEndpointGroupCommand;

// src/commands/DeleteEphemerisCommand.ts




var _DeleteEphemerisCommand = class _DeleteEphemerisCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("GroundStation", "DeleteEphemeris", {}).n("GroundStationClient", "DeleteEphemerisCommand").f(void 0, void 0).ser(se_DeleteEphemerisCommand).de(de_DeleteEphemerisCommand).build() {
};
__name(_DeleteEphemerisCommand, "DeleteEphemerisCommand");
var DeleteEphemerisCommand = _DeleteEphemerisCommand;

// src/commands/DeleteMissionProfileCommand.ts




var _DeleteMissionProfileCommand = class _DeleteMissionProfileCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("GroundStation", "DeleteMissionProfile", {}).n("GroundStationClient", "DeleteMissionProfileCommand").f(void 0, void 0).ser(se_DeleteMissionProfileCommand).de(de_DeleteMissionProfileCommand).build() {
};
__name(_DeleteMissionProfileCommand, "DeleteMissionProfileCommand");
var DeleteMissionProfileCommand = _DeleteMissionProfileCommand;

// src/commands/DescribeContactCommand.ts




var _DescribeContactCommand = class _DescribeContactCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("GroundStation", "DescribeContact", {}).n("GroundStationClient", "DescribeContactCommand").f(void 0, void 0).ser(se_DescribeContactCommand).de(de_DescribeContactCommand).build() {
};
__name(_DescribeContactCommand, "DescribeContactCommand");
var DescribeContactCommand = _DescribeContactCommand;

// src/commands/DescribeEphemerisCommand.ts




var _DescribeEphemerisCommand = class _DescribeEphemerisCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("GroundStation", "DescribeEphemeris", {}).n("GroundStationClient", "DescribeEphemerisCommand").f(void 0, void 0).ser(se_DescribeEphemerisCommand).de(de_DescribeEphemerisCommand).build() {
};
__name(_DescribeEphemerisCommand, "DescribeEphemerisCommand");
var DescribeEphemerisCommand = _DescribeEphemerisCommand;

// src/commands/GetAgentConfigurationCommand.ts




var _GetAgentConfigurationCommand = class _GetAgentConfigurationCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("GroundStation", "GetAgentConfiguration", {}).n("GroundStationClient", "GetAgentConfigurationCommand").f(void 0, void 0).ser(se_GetAgentConfigurationCommand).de(de_GetAgentConfigurationCommand).build() {
};
__name(_GetAgentConfigurationCommand, "GetAgentConfigurationCommand");
var GetAgentConfigurationCommand = _GetAgentConfigurationCommand;

// src/commands/GetConfigCommand.ts




var _GetConfigCommand = class _GetConfigCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("GroundStation", "GetConfig", {}).n("GroundStationClient", "GetConfigCommand").f(void 0, void 0).ser(se_GetConfigCommand).de(de_GetConfigCommand).build() {
};
__name(_GetConfigCommand, "GetConfigCommand");
var GetConfigCommand = _GetConfigCommand;

// src/commands/GetDataflowEndpointGroupCommand.ts




var _GetDataflowEndpointGroupCommand = class _GetDataflowEndpointGroupCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("GroundStation", "GetDataflowEndpointGroup", {}).n("GroundStationClient", "GetDataflowEndpointGroupCommand").f(void 0, void 0).ser(se_GetDataflowEndpointGroupCommand).de(de_GetDataflowEndpointGroupCommand).build() {
};
__name(_GetDataflowEndpointGroupCommand, "GetDataflowEndpointGroupCommand");
var GetDataflowEndpointGroupCommand = _GetDataflowEndpointGroupCommand;

// src/commands/GetMinuteUsageCommand.ts




var _GetMinuteUsageCommand = class _GetMinuteUsageCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("GroundStation", "GetMinuteUsage", {}).n("GroundStationClient", "GetMinuteUsageCommand").f(void 0, void 0).ser(se_GetMinuteUsageCommand).de(de_GetMinuteUsageCommand).build() {
};
__name(_GetMinuteUsageCommand, "GetMinuteUsageCommand");
var GetMinuteUsageCommand = _GetMinuteUsageCommand;

// src/commands/GetMissionProfileCommand.ts




var _GetMissionProfileCommand = class _GetMissionProfileCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("GroundStation", "GetMissionProfile", {}).n("GroundStationClient", "GetMissionProfileCommand").f(void 0, void 0).ser(se_GetMissionProfileCommand).de(de_GetMissionProfileCommand).build() {
};
__name(_GetMissionProfileCommand, "GetMissionProfileCommand");
var GetMissionProfileCommand = _GetMissionProfileCommand;

// src/commands/GetSatelliteCommand.ts




var _GetSatelliteCommand = class _GetSatelliteCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("GroundStation", "GetSatellite", {}).n("GroundStationClient", "GetSatelliteCommand").f(void 0, void 0).ser(se_GetSatelliteCommand).de(de_GetSatelliteCommand).build() {
};
__name(_GetSatelliteCommand, "GetSatelliteCommand");
var GetSatelliteCommand = _GetSatelliteCommand;

// src/commands/ListConfigsCommand.ts




var _ListConfigsCommand = class _ListConfigsCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("GroundStation", "ListConfigs", {}).n("GroundStationClient", "ListConfigsCommand").f(void 0, void 0).ser(se_ListConfigsCommand).de(de_ListConfigsCommand).build() {
};
__name(_ListConfigsCommand, "ListConfigsCommand");
var ListConfigsCommand = _ListConfigsCommand;

// src/commands/ListContactsCommand.ts




var _ListContactsCommand = class _ListContactsCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("GroundStation", "ListContacts", {}).n("GroundStationClient", "ListContactsCommand").f(void 0, void 0).ser(se_ListContactsCommand).de(de_ListContactsCommand).build() {
};
__name(_ListContactsCommand, "ListContactsCommand");
var ListContactsCommand = _ListContactsCommand;

// src/commands/ListDataflowEndpointGroupsCommand.ts




var _ListDataflowEndpointGroupsCommand = class _ListDataflowEndpointGroupsCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("GroundStation", "ListDataflowEndpointGroups", {}).n("GroundStationClient", "ListDataflowEndpointGroupsCommand").f(void 0, void 0).ser(se_ListDataflowEndpointGroupsCommand).de(de_ListDataflowEndpointGroupsCommand).build() {
};
__name(_ListDataflowEndpointGroupsCommand, "ListDataflowEndpointGroupsCommand");
var ListDataflowEndpointGroupsCommand = _ListDataflowEndpointGroupsCommand;

// src/commands/ListEphemeridesCommand.ts




var _ListEphemeridesCommand = class _ListEphemeridesCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("GroundStation", "ListEphemerides", {}).n("GroundStationClient", "ListEphemeridesCommand").f(void 0, void 0).ser(se_ListEphemeridesCommand).de(de_ListEphemeridesCommand).build() {
};
__name(_ListEphemeridesCommand, "ListEphemeridesCommand");
var ListEphemeridesCommand = _ListEphemeridesCommand;

// src/commands/ListGroundStationsCommand.ts




var _ListGroundStationsCommand = class _ListGroundStationsCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("GroundStation", "ListGroundStations", {}).n("GroundStationClient", "ListGroundStationsCommand").f(void 0, void 0).ser(se_ListGroundStationsCommand).de(de_ListGroundStationsCommand).build() {
};
__name(_ListGroundStationsCommand, "ListGroundStationsCommand");
var ListGroundStationsCommand = _ListGroundStationsCommand;

// src/commands/ListMissionProfilesCommand.ts




var _ListMissionProfilesCommand = class _ListMissionProfilesCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("GroundStation", "ListMissionProfiles", {}).n("GroundStationClient", "ListMissionProfilesCommand").f(void 0, void 0).ser(se_ListMissionProfilesCommand).de(de_ListMissionProfilesCommand).build() {
};
__name(_ListMissionProfilesCommand, "ListMissionProfilesCommand");
var ListMissionProfilesCommand = _ListMissionProfilesCommand;

// src/commands/ListSatellitesCommand.ts




var _ListSatellitesCommand = class _ListSatellitesCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("GroundStation", "ListSatellites", {}).n("GroundStationClient", "ListSatellitesCommand").f(void 0, void 0).ser(se_ListSatellitesCommand).de(de_ListSatellitesCommand).build() {
};
__name(_ListSatellitesCommand, "ListSatellitesCommand");
var ListSatellitesCommand = _ListSatellitesCommand;

// src/commands/ListTagsForResourceCommand.ts




var _ListTagsForResourceCommand = class _ListTagsForResourceCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("GroundStation", "ListTagsForResource", {}).n("GroundStationClient", "ListTagsForResourceCommand").f(void 0, void 0).ser(se_ListTagsForResourceCommand).de(de_ListTagsForResourceCommand).build() {
};
__name(_ListTagsForResourceCommand, "ListTagsForResourceCommand");
var ListTagsForResourceCommand = _ListTagsForResourceCommand;

// src/commands/RegisterAgentCommand.ts




var _RegisterAgentCommand = class _RegisterAgentCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("GroundStation", "RegisterAgent", {}).n("GroundStationClient", "RegisterAgentCommand").f(void 0, void 0).ser(se_RegisterAgentCommand).de(de_RegisterAgentCommand).build() {
};
__name(_RegisterAgentCommand, "RegisterAgentCommand");
var RegisterAgentCommand = _RegisterAgentCommand;

// src/commands/ReserveContactCommand.ts




var _ReserveContactCommand = class _ReserveContactCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("GroundStation", "ReserveContact", {}).n("GroundStationClient", "ReserveContactCommand").f(void 0, void 0).ser(se_ReserveContactCommand).de(de_ReserveContactCommand).build() {
};
__name(_ReserveContactCommand, "ReserveContactCommand");
var ReserveContactCommand = _ReserveContactCommand;

// src/commands/TagResourceCommand.ts




var _TagResourceCommand = class _TagResourceCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("GroundStation", "TagResource", {}).n("GroundStationClient", "TagResourceCommand").f(void 0, void 0).ser(se_TagResourceCommand).de(de_TagResourceCommand).build() {
};
__name(_TagResourceCommand, "TagResourceCommand");
var TagResourceCommand = _TagResourceCommand;

// src/commands/UntagResourceCommand.ts




var _UntagResourceCommand = class _UntagResourceCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("GroundStation", "UntagResource", {}).n("GroundStationClient", "UntagResourceCommand").f(void 0, void 0).ser(se_UntagResourceCommand).de(de_UntagResourceCommand).build() {
};
__name(_UntagResourceCommand, "UntagResourceCommand");
var UntagResourceCommand = _UntagResourceCommand;

// src/commands/UpdateAgentStatusCommand.ts




var _UpdateAgentStatusCommand = class _UpdateAgentStatusCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("GroundStation", "UpdateAgentStatus", {}).n("GroundStationClient", "UpdateAgentStatusCommand").f(void 0, void 0).ser(se_UpdateAgentStatusCommand).de(de_UpdateAgentStatusCommand).build() {
};
__name(_UpdateAgentStatusCommand, "UpdateAgentStatusCommand");
var UpdateAgentStatusCommand = _UpdateAgentStatusCommand;

// src/commands/UpdateConfigCommand.ts




var _UpdateConfigCommand = class _UpdateConfigCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("GroundStation", "UpdateConfig", {}).n("GroundStationClient", "UpdateConfigCommand").f(void 0, void 0).ser(se_UpdateConfigCommand).de(de_UpdateConfigCommand).build() {
};
__name(_UpdateConfigCommand, "UpdateConfigCommand");
var UpdateConfigCommand = _UpdateConfigCommand;

// src/commands/UpdateEphemerisCommand.ts




var _UpdateEphemerisCommand = class _UpdateEphemerisCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("GroundStation", "UpdateEphemeris", {}).n("GroundStationClient", "UpdateEphemerisCommand").f(void 0, void 0).ser(se_UpdateEphemerisCommand).de(de_UpdateEphemerisCommand).build() {
};
__name(_UpdateEphemerisCommand, "UpdateEphemerisCommand");
var UpdateEphemerisCommand = _UpdateEphemerisCommand;

// src/commands/UpdateMissionProfileCommand.ts




var _UpdateMissionProfileCommand = class _UpdateMissionProfileCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("GroundStation", "UpdateMissionProfile", {}).n("GroundStationClient", "UpdateMissionProfileCommand").f(void 0, void 0).ser(se_UpdateMissionProfileCommand).de(de_UpdateMissionProfileCommand).build() {
};
__name(_UpdateMissionProfileCommand, "UpdateMissionProfileCommand");
var UpdateMissionProfileCommand = _UpdateMissionProfileCommand;

// src/GroundStation.ts
var commands = {
  CancelContactCommand,
  CreateConfigCommand,
  CreateDataflowEndpointGroupCommand,
  CreateEphemerisCommand,
  CreateMissionProfileCommand,
  DeleteConfigCommand,
  DeleteDataflowEndpointGroupCommand,
  DeleteEphemerisCommand,
  DeleteMissionProfileCommand,
  DescribeContactCommand,
  DescribeEphemerisCommand,
  GetAgentConfigurationCommand,
  GetConfigCommand,
  GetDataflowEndpointGroupCommand,
  GetMinuteUsageCommand,
  GetMissionProfileCommand,
  GetSatelliteCommand,
  ListConfigsCommand,
  ListContactsCommand,
  ListDataflowEndpointGroupsCommand,
  ListEphemeridesCommand,
  ListGroundStationsCommand,
  ListMissionProfilesCommand,
  ListSatellitesCommand,
  ListTagsForResourceCommand,
  RegisterAgentCommand,
  ReserveContactCommand,
  TagResourceCommand,
  UntagResourceCommand,
  UpdateAgentStatusCommand,
  UpdateConfigCommand,
  UpdateEphemerisCommand,
  UpdateMissionProfileCommand
};
var _GroundStation = class _GroundStation extends GroundStationClient {
};
__name(_GroundStation, "GroundStation");
var GroundStation = _GroundStation;
(0, import_smithy_client.createAggregatedClient)(commands, GroundStation);

// src/pagination/ListConfigsPaginator.ts

var paginateListConfigs = (0, import_core.createPaginator)(GroundStationClient, ListConfigsCommand, "nextToken", "nextToken", "maxResults");

// src/pagination/ListContactsPaginator.ts

var paginateListContacts = (0, import_core.createPaginator)(GroundStationClient, ListContactsCommand, "nextToken", "nextToken", "maxResults");

// src/pagination/ListDataflowEndpointGroupsPaginator.ts

var paginateListDataflowEndpointGroups = (0, import_core.createPaginator)(GroundStationClient, ListDataflowEndpointGroupsCommand, "nextToken", "nextToken", "maxResults");

// src/pagination/ListEphemeridesPaginator.ts

var paginateListEphemerides = (0, import_core.createPaginator)(GroundStationClient, ListEphemeridesCommand, "nextToken", "nextToken", "maxResults");

// src/pagination/ListGroundStationsPaginator.ts

var paginateListGroundStations = (0, import_core.createPaginator)(GroundStationClient, ListGroundStationsCommand, "nextToken", "nextToken", "maxResults");

// src/pagination/ListMissionProfilesPaginator.ts

var paginateListMissionProfiles = (0, import_core.createPaginator)(GroundStationClient, ListMissionProfilesCommand, "nextToken", "nextToken", "maxResults");

// src/pagination/ListSatellitesPaginator.ts

var paginateListSatellites = (0, import_core.createPaginator)(GroundStationClient, ListSatellitesCommand, "nextToken", "nextToken", "maxResults");

// src/waiters/waitForContactScheduled.ts
var import_util_waiter = require("@smithy/util-waiter");
var checkState = /* @__PURE__ */ __name(async (client, input) => {
  let reason;
  try {
    const result = await client.send(new DescribeContactCommand(input));
    reason = result;
    try {
      const returnComparator = /* @__PURE__ */ __name(() => {
        return result.contactStatus;
      }, "returnComparator");
      if (returnComparator() === "FAILED_TO_SCHEDULE") {
        return { state: import_util_waiter.WaiterState.FAILURE, reason };
      }
    } catch (e) {
    }
    try {
      const returnComparator = /* @__PURE__ */ __name(() => {
        return result.contactStatus;
      }, "returnComparator");
      if (returnComparator() === "SCHEDULED") {
        return { state: import_util_waiter.WaiterState.SUCCESS, reason };
      }
    } catch (e) {
    }
  } catch (exception) {
    reason = exception;
  }
  return { state: import_util_waiter.WaiterState.RETRY, reason };
}, "checkState");
var waitForContactScheduled = /* @__PURE__ */ __name(async (params, input) => {
  const serviceDefaults = { minDelay: 5, maxDelay: 900 };
  return (0, import_util_waiter.createWaiter)({ ...serviceDefaults, ...params }, input, checkState);
}, "waitForContactScheduled");
var waitUntilContactScheduled = /* @__PURE__ */ __name(async (params, input) => {
  const serviceDefaults = { minDelay: 5, maxDelay: 900 };
  const result = await (0, import_util_waiter.createWaiter)({ ...serviceDefaults, ...params }, input, checkState);
  return (0, import_util_waiter.checkExceptions)(result);
}, "waitUntilContactScheduled");
// Annotate the CommonJS export names for ESM import in node:

0 && (module.exports = {
  GroundStationServiceException,
  __Client,
  GroundStationClient,
  GroundStation,
  $Command,
  CancelContactCommand,
  CreateConfigCommand,
  CreateDataflowEndpointGroupCommand,
  CreateEphemerisCommand,
  CreateMissionProfileCommand,
  DeleteConfigCommand,
  DeleteDataflowEndpointGroupCommand,
  DeleteEphemerisCommand,
  DeleteMissionProfileCommand,
  DescribeContactCommand,
  DescribeEphemerisCommand,
  GetAgentConfigurationCommand,
  GetConfigCommand,
  GetDataflowEndpointGroupCommand,
  GetMinuteUsageCommand,
  GetMissionProfileCommand,
  GetSatelliteCommand,
  ListConfigsCommand,
  ListContactsCommand,
  ListDataflowEndpointGroupsCommand,
  ListEphemeridesCommand,
  ListGroundStationsCommand,
  ListMissionProfilesCommand,
  ListSatellitesCommand,
  ListTagsForResourceCommand,
  RegisterAgentCommand,
  ReserveContactCommand,
  TagResourceCommand,
  UntagResourceCommand,
  UpdateAgentStatusCommand,
  UpdateConfigCommand,
  UpdateEphemerisCommand,
  UpdateMissionProfileCommand,
  paginateListConfigs,
  paginateListContacts,
  paginateListDataflowEndpointGroups,
  paginateListEphemerides,
  paginateListGroundStations,
  paginateListMissionProfiles,
  paginateListSatellites,
  waitForContactScheduled,
  waitUntilContactScheduled,
  DependencyException,
  InvalidParameterException,
  ResourceNotFoundException,
  AgentStatus,
  AngleUnits,
  BandwidthUnits,
  FrequencyUnits,
  Polarization,
  EirpUnits,
  AuditResults,
  CapabilityHealth,
  CapabilityHealthReason,
  ConfigCapabilityType,
  Criticality,
  ConfigTypeData,
  ResourceLimitExceededException,
  EndpointStatus,
  ConfigDetails,
  ContactStatus,
  EphemerisData,
  KmsKey,
  EphemerisInvalidReason,
  EphemerisStatus,
  EphemerisTypeDescription,
  EphemerisSource
});
