// @aws-sdk/client-workspaces-web/dist-cjs/index.js
"use strict";
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __name = (target, value) => __defProp(target, "name", { value, configurable: true });
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// src/index.ts
var src_exports = {};
__export(src_exports, {
  AccessDeniedException: () => AccessDeniedException,
  AssociateBrowserSettingsCommand: () => AssociateBrowserSettingsCommand,
  AssociateIpAccessSettingsCommand: () => AssociateIpAccessSettingsCommand,
  AssociateNetworkSettingsCommand: () => AssociateNetworkSettingsCommand,
  AssociateTrustStoreCommand: () => AssociateTrustStoreCommand,
  AssociateUserAccessLoggingSettingsCommand: () => AssociateUserAccessLoggingSettingsCommand,
  AssociateUserSettingsCommand: () => AssociateUserSettingsCommand,
  AuthenticationType: () => AuthenticationType,
  BrowserSettingsFilterSensitiveLog: () => BrowserSettingsFilterSensitiveLog,
  BrowserType: () => BrowserType,
  ConflictException: () => ConflictException,
  CookieSpecificationFilterSensitiveLog: () => CookieSpecificationFilterSensitiveLog,
  CookieSynchronizationConfigurationFilterSensitiveLog: () => CookieSynchronizationConfigurationFilterSensitiveLog,
  CreateBrowserSettingsCommand: () => CreateBrowserSettingsCommand,
  CreateBrowserSettingsRequestFilterSensitiveLog: () => CreateBrowserSettingsRequestFilterSensitiveLog,
  CreateIdentityProviderCommand: () => CreateIdentityProviderCommand,
  CreateIdentityProviderRequestFilterSensitiveLog: () => CreateIdentityProviderRequestFilterSensitiveLog,
  CreateIpAccessSettingsCommand: () => CreateIpAccessSettingsCommand,
  CreateIpAccessSettingsRequestFilterSensitiveLog: () => CreateIpAccessSettingsRequestFilterSensitiveLog,
  CreateNetworkSettingsCommand: () => CreateNetworkSettingsCommand,
  CreateNetworkSettingsRequestFilterSensitiveLog: () => CreateNetworkSettingsRequestFilterSensitiveLog,
  CreatePortalCommand: () => CreatePortalCommand,
  CreatePortalRequestFilterSensitiveLog: () => CreatePortalRequestFilterSensitiveLog,
  CreateTrustStoreCommand: () => CreateTrustStoreCommand,
  CreateTrustStoreRequestFilterSensitiveLog: () => CreateTrustStoreRequestFilterSensitiveLog,
  CreateUserAccessLoggingSettingsCommand: () => CreateUserAccessLoggingSettingsCommand,
  CreateUserAccessLoggingSettingsRequestFilterSensitiveLog: () => CreateUserAccessLoggingSettingsRequestFilterSensitiveLog,
  CreateUserSettingsCommand: () => CreateUserSettingsCommand,
  CreateUserSettingsRequestFilterSensitiveLog: () => CreateUserSettingsRequestFilterSensitiveLog,
  DeleteBrowserSettingsCommand: () => DeleteBrowserSettingsCommand,
  DeleteIdentityProviderCommand: () => DeleteIdentityProviderCommand,
  DeleteIpAccessSettingsCommand: () => DeleteIpAccessSettingsCommand,
  DeleteNetworkSettingsCommand: () => DeleteNetworkSettingsCommand,
  DeletePortalCommand: () => DeletePortalCommand,
  DeleteTrustStoreCommand: () => DeleteTrustStoreCommand,
  DeleteUserAccessLoggingSettingsCommand: () => DeleteUserAccessLoggingSettingsCommand,
  DeleteUserSettingsCommand: () => DeleteUserSettingsCommand,
  DisassociateBrowserSettingsCommand: () => DisassociateBrowserSettingsCommand,
  DisassociateIpAccessSettingsCommand: () => DisassociateIpAccessSettingsCommand,
  DisassociateNetworkSettingsCommand: () => DisassociateNetworkSettingsCommand,
  DisassociateTrustStoreCommand: () => DisassociateTrustStoreCommand,
  DisassociateUserAccessLoggingSettingsCommand: () => DisassociateUserAccessLoggingSettingsCommand,
  DisassociateUserSettingsCommand: () => DisassociateUserSettingsCommand,
  EnabledType: () => EnabledType,
  GetBrowserSettingsCommand: () => GetBrowserSettingsCommand,
  GetBrowserSettingsResponseFilterSensitiveLog: () => GetBrowserSettingsResponseFilterSensitiveLog,
  GetIdentityProviderCommand: () => GetIdentityProviderCommand,
  GetIdentityProviderResponseFilterSensitiveLog: () => GetIdentityProviderResponseFilterSensitiveLog,
  GetIpAccessSettingsCommand: () => GetIpAccessSettingsCommand,
  GetIpAccessSettingsResponseFilterSensitiveLog: () => GetIpAccessSettingsResponseFilterSensitiveLog,
  GetNetworkSettingsCommand: () => GetNetworkSettingsCommand,
  GetPortalCommand: () => GetPortalCommand,
  GetPortalResponseFilterSensitiveLog: () => GetPortalResponseFilterSensitiveLog,
  GetPortalServiceProviderMetadataCommand: () => GetPortalServiceProviderMetadataCommand,
  GetTrustStoreCertificateCommand: () => GetTrustStoreCertificateCommand,
  GetTrustStoreCommand: () => GetTrustStoreCommand,
  GetUserAccessLoggingSettingsCommand: () => GetUserAccessLoggingSettingsCommand,
  GetUserSettingsCommand: () => GetUserSettingsCommand,
  GetUserSettingsResponseFilterSensitiveLog: () => GetUserSettingsResponseFilterSensitiveLog,
  IdentityProviderFilterSensitiveLog: () => IdentityProviderFilterSensitiveLog,
  IdentityProviderSummaryFilterSensitiveLog: () => IdentityProviderSummaryFilterSensitiveLog,
  IdentityProviderType: () => IdentityProviderType,
  InternalServerException: () => InternalServerException,
  IpAccessSettingsFilterSensitiveLog: () => IpAccessSettingsFilterSensitiveLog,
  IpAccessSettingsSummaryFilterSensitiveLog: () => IpAccessSettingsSummaryFilterSensitiveLog,
  IpRuleFilterSensitiveLog: () => IpRuleFilterSensitiveLog,
  ListBrowserSettingsCommand: () => ListBrowserSettingsCommand,
  ListIdentityProvidersCommand: () => ListIdentityProvidersCommand,
  ListIdentityProvidersResponseFilterSensitiveLog: () => ListIdentityProvidersResponseFilterSensitiveLog,
  ListIpAccessSettingsCommand: () => ListIpAccessSettingsCommand,
  ListIpAccessSettingsResponseFilterSensitiveLog: () => ListIpAccessSettingsResponseFilterSensitiveLog,
  ListNetworkSettingsCommand: () => ListNetworkSettingsCommand,
  ListPortalsCommand: () => ListPortalsCommand,
  ListPortalsResponseFilterSensitiveLog: () => ListPortalsResponseFilterSensitiveLog,
  ListTagsForResourceCommand: () => ListTagsForResourceCommand,
  ListTagsForResourceResponseFilterSensitiveLog: () => ListTagsForResourceResponseFilterSensitiveLog,
  ListTrustStoreCertificatesCommand: () => ListTrustStoreCertificatesCommand,
  ListTrustStoresCommand: () => ListTrustStoresCommand,
  ListUserAccessLoggingSettingsCommand: () => ListUserAccessLoggingSettingsCommand,
  ListUserSettingsCommand: () => ListUserSettingsCommand,
  ListUserSettingsResponseFilterSensitiveLog: () => ListUserSettingsResponseFilterSensitiveLog,
  PortalFilterSensitiveLog: () => PortalFilterSensitiveLog,
  PortalStatus: () => PortalStatus,
  PortalSummaryFilterSensitiveLog: () => PortalSummaryFilterSensitiveLog,
  RendererType: () => RendererType,
  ResourceNotFoundException: () => ResourceNotFoundException,
  ServiceQuotaExceededException: () => ServiceQuotaExceededException,
  TagFilterSensitiveLog: () => TagFilterSensitiveLog,
  TagResourceCommand: () => TagResourceCommand,
  TagResourceRequestFilterSensitiveLog: () => TagResourceRequestFilterSensitiveLog,
  ThrottlingException: () => ThrottlingException,
  TooManyTagsException: () => TooManyTagsException,
  UntagResourceCommand: () => UntagResourceCommand,
  UntagResourceRequestFilterSensitiveLog: () => UntagResourceRequestFilterSensitiveLog,
  UpdateBrowserSettingsCommand: () => UpdateBrowserSettingsCommand,
  UpdateBrowserSettingsRequestFilterSensitiveLog: () => UpdateBrowserSettingsRequestFilterSensitiveLog,
  UpdateBrowserSettingsResponseFilterSensitiveLog: () => UpdateBrowserSettingsResponseFilterSensitiveLog,
  UpdateIdentityProviderCommand: () => UpdateIdentityProviderCommand,
  UpdateIdentityProviderRequestFilterSensitiveLog: () => UpdateIdentityProviderRequestFilterSensitiveLog,
  UpdateIdentityProviderResponseFilterSensitiveLog: () => UpdateIdentityProviderResponseFilterSensitiveLog,
  UpdateIpAccessSettingsCommand: () => UpdateIpAccessSettingsCommand,
  UpdateIpAccessSettingsRequestFilterSensitiveLog: () => UpdateIpAccessSettingsRequestFilterSensitiveLog,
  UpdateIpAccessSettingsResponseFilterSensitiveLog: () => UpdateIpAccessSettingsResponseFilterSensitiveLog,
  UpdateNetworkSettingsCommand: () => UpdateNetworkSettingsCommand,
  UpdatePortalCommand: () => UpdatePortalCommand,
  UpdatePortalRequestFilterSensitiveLog: () => UpdatePortalRequestFilterSensitiveLog,
  UpdatePortalResponseFilterSensitiveLog: () => UpdatePortalResponseFilterSensitiveLog,
  UpdateTrustStoreCommand: () => UpdateTrustStoreCommand,
  UpdateUserAccessLoggingSettingsCommand: () => UpdateUserAccessLoggingSettingsCommand,
  UpdateUserSettingsCommand: () => UpdateUserSettingsCommand,
  UpdateUserSettingsRequestFilterSensitiveLog: () => UpdateUserSettingsRequestFilterSensitiveLog,
  UpdateUserSettingsResponseFilterSensitiveLog: () => UpdateUserSettingsResponseFilterSensitiveLog,
  UserSettingsFilterSensitiveLog: () => UserSettingsFilterSensitiveLog,
  UserSettingsSummaryFilterSensitiveLog: () => UserSettingsSummaryFilterSensitiveLog,
  ValidationException: () => ValidationException,
  ValidationExceptionReason: () => ValidationExceptionReason,
  WorkSpacesWeb: () => WorkSpacesWeb,
  WorkSpacesWebClient: () => WorkSpacesWebClient,
  WorkSpacesWebServiceException: () => WorkSpacesWebServiceException,
  _InstanceType: () => _InstanceType,
  __Client: () => import_smithy_client.Client,
  paginateListBrowserSettings: () => paginateListBrowserSettings,
  paginateListIdentityProviders: () => paginateListIdentityProviders,
  paginateListIpAccessSettings: () => paginateListIpAccessSettings,
  paginateListNetworkSettings: () => paginateListNetworkSettings,
  paginateListPortals: () => paginateListPortals,
  paginateListTrustStoreCertificates: () => paginateListTrustStoreCertificates,
  paginateListTrustStores: () => paginateListTrustStores,
  paginateListUserAccessLoggingSettings: () => paginateListUserAccessLoggingSettings,
  paginateListUserSettings: () => paginateListUserSettings
});
module.exports = __toCommonJS(src_exports);

// src/WorkSpacesWebClient.ts
var import_middleware_host_header = require("@aws-sdk/middleware-host-header");
var import_middleware_logger = require("@aws-sdk/middleware-logger");
var import_middleware_recursion_detection = require("@aws-sdk/middleware-recursion-detection");
var import_middleware_user_agent = require("@aws-sdk/middleware-user-agent");
var import_config_resolver = require("@smithy/config-resolver");
var import_core = require("@smithy/core");
var import_middleware_content_length = require("@smithy/middleware-content-length");
var import_middleware_endpoint = require("@smithy/middleware-endpoint");
var import_middleware_retry = require("@smithy/middleware-retry");

var import_httpAuthSchemeProvider = require("./auth/httpAuthSchemeProvider");

// src/endpoint/EndpointParameters.ts
var resolveClientEndpointParameters = /* @__PURE__ */ __name((options) => {
  return {
    ...options,
    useDualstackEndpoint: options.useDualstackEndpoint ?? false,
    useFipsEndpoint: options.useFipsEndpoint ?? false,
    defaultSigningName: "workspaces-web"
  };
}, "resolveClientEndpointParameters");
var commonParams = {
  UseFIPS: { type: "builtInParams", name: "useFipsEndpoint" },
  Endpoint: { type: "builtInParams", name: "endpoint" },
  Region: { type: "builtInParams", name: "region" },
  UseDualStack: { type: "builtInParams", name: "useDualstackEndpoint" }
};

// src/WorkSpacesWebClient.ts
var import_runtimeConfig = require("././runtimeConfig");

// src/runtimeExtensions.ts
var import_region_config_resolver = require("@aws-sdk/region-config-resolver");
var import_protocol_http = require("@smithy/protocol-http");
var import_smithy_client = require("@smithy/smithy-client");

// src/auth/httpAuthExtensionConfiguration.ts
var getHttpAuthExtensionConfiguration = /* @__PURE__ */ __name((runtimeConfig) => {
  const _httpAuthSchemes = runtimeConfig.httpAuthSchemes;
  let _httpAuthSchemeProvider = runtimeConfig.httpAuthSchemeProvider;
  let _credentials = runtimeConfig.credentials;
  return {
    setHttpAuthScheme(httpAuthScheme) {
      const index = _httpAuthSchemes.findIndex((scheme) => scheme.schemeId === httpAuthScheme.schemeId);
      if (index === -1) {
        _httpAuthSchemes.push(httpAuthScheme);
      } else {
        _httpAuthSchemes.splice(index, 1, httpAuthScheme);
      }
    },
    httpAuthSchemes() {
      return _httpAuthSchemes;
    },
    setHttpAuthSchemeProvider(httpAuthSchemeProvider) {
      _httpAuthSchemeProvider = httpAuthSchemeProvider;
    },
    httpAuthSchemeProvider() {
      return _httpAuthSchemeProvider;
    },
    setCredentials(credentials) {
      _credentials = credentials;
    },
    credentials() {
      return _credentials;
    }
  };
}, "getHttpAuthExtensionConfiguration");
var resolveHttpAuthRuntimeConfig = /* @__PURE__ */ __name((config) => {
  return {
    httpAuthSchemes: config.httpAuthSchemes(),
    httpAuthSchemeProvider: config.httpAuthSchemeProvider(),
    credentials: config.credentials()
  };
}, "resolveHttpAuthRuntimeConfig");

// src/runtimeExtensions.ts
var asPartial = /* @__PURE__ */ __name((t) => t, "asPartial");
var resolveRuntimeExtensions = /* @__PURE__ */ __name((runtimeConfig, extensions) => {
  const extensionConfiguration = {
    ...asPartial((0, import_region_config_resolver.getAwsRegionExtensionConfiguration)(runtimeConfig)),
    ...asPartial((0, import_smithy_client.getDefaultExtensionConfiguration)(runtimeConfig)),
    ...asPartial((0, import_protocol_http.getHttpHandlerExtensionConfiguration)(runtimeConfig)),
    ...asPartial(getHttpAuthExtensionConfiguration(runtimeConfig))
  };
  extensions.forEach((extension) => extension.configure(extensionConfiguration));
  return {
    ...runtimeConfig,
    ...(0, import_region_config_resolver.resolveAwsRegionExtensionConfiguration)(extensionConfiguration),
    ...(0, import_smithy_client.resolveDefaultRuntimeConfig)(extensionConfiguration),
    ...(0, import_protocol_http.resolveHttpHandlerRuntimeConfig)(extensionConfiguration),
    ...resolveHttpAuthRuntimeConfig(extensionConfiguration)
  };
}, "resolveRuntimeExtensions");

// src/WorkSpacesWebClient.ts
var _WorkSpacesWebClient = class _WorkSpacesWebClient extends import_smithy_client.Client {
  constructor(...[configuration]) {
    const _config_0 = (0, import_runtimeConfig.getRuntimeConfig)(configuration || {});
    const _config_1 = resolveClientEndpointParameters(_config_0);
    const _config_2 = (0, import_config_resolver.resolveRegionConfig)(_config_1);
    const _config_3 = (0, import_middleware_endpoint.resolveEndpointConfig)(_config_2);
    const _config_4 = (0, import_middleware_retry.resolveRetryConfig)(_config_3);
    const _config_5 = (0, import_middleware_host_header.resolveHostHeaderConfig)(_config_4);
    const _config_6 = (0, import_middleware_user_agent.resolveUserAgentConfig)(_config_5);
    const _config_7 = (0, import_httpAuthSchemeProvider.resolveHttpAuthSchemeConfig)(_config_6);
    const _config_8 = resolveRuntimeExtensions(_config_7, (configuration == null ? void 0 : configuration.extensions) || []);
    super(_config_8);
    this.config = _config_8;
    this.middlewareStack.use((0, import_middleware_retry.getRetryPlugin)(this.config));
    this.middlewareStack.use((0, import_middleware_content_length.getContentLengthPlugin)(this.config));
    this.middlewareStack.use((0, import_middleware_host_header.getHostHeaderPlugin)(this.config));
    this.middlewareStack.use((0, import_middleware_logger.getLoggerPlugin)(this.config));
    this.middlewareStack.use((0, import_middleware_recursion_detection.getRecursionDetectionPlugin)(this.config));
    this.middlewareStack.use((0, import_middleware_user_agent.getUserAgentPlugin)(this.config));
    this.middlewareStack.use(
      (0, import_core.getHttpAuthSchemeEndpointRuleSetPlugin)(this.config, {
        httpAuthSchemeParametersProvider: this.getDefaultHttpAuthSchemeParametersProvider(),
        identityProviderConfigProvider: this.getIdentityProviderConfigProvider()
      })
    );
    this.middlewareStack.use((0, import_core.getHttpSigningPlugin)(this.config));
  }
  /**
   * Destroy underlying resources, like sockets. It's usually not necessary to do this.
   * However in Node.js, it's best to explicitly shut down the client's agent when it is no longer needed.
   * Otherwise, sockets might stay open for quite a long time before the server terminates them.
   */
  destroy() {
    super.destroy();
  }
  getDefaultHttpAuthSchemeParametersProvider() {
    return import_httpAuthSchemeProvider.defaultWorkSpacesWebHttpAuthSchemeParametersProvider;
  }
  getIdentityProviderConfigProvider() {
    return async (config) => new import_core.DefaultIdentityProviderConfig({
      "aws.auth#sigv4": config.credentials
    });
  }
};
__name(_WorkSpacesWebClient, "WorkSpacesWebClient");
var WorkSpacesWebClient = _WorkSpacesWebClient;

// src/WorkSpacesWeb.ts


// src/commands/AssociateBrowserSettingsCommand.ts

var import_middleware_serde = require("@smithy/middleware-serde");

var import_types = require("@smithy/types");

// src/protocols/Aws_restJson1.ts
var import_core2 = require("@aws-sdk/core");


var import_uuid = require("uuid");

// src/models/models_0.ts


// src/models/WorkSpacesWebServiceException.ts

var _WorkSpacesWebServiceException = class _WorkSpacesWebServiceException extends import_smithy_client.ServiceException {
  /**
   * @internal
   */
  constructor(options) {
    super(options);
    Object.setPrototypeOf(this, _WorkSpacesWebServiceException.prototype);
  }
};
__name(_WorkSpacesWebServiceException, "WorkSpacesWebServiceException");
var WorkSpacesWebServiceException = _WorkSpacesWebServiceException;

// src/models/models_0.ts
var _AccessDeniedException = class _AccessDeniedException extends WorkSpacesWebServiceException {
  /**
   * @internal
   */
  constructor(opts) {
    super({
      name: "AccessDeniedException",
      $fault: "client",
      ...opts
    });
    this.name = "AccessDeniedException";
    this.$fault = "client";
    Object.setPrototypeOf(this, _AccessDeniedException.prototype);
  }
};
__name(_AccessDeniedException, "AccessDeniedException");
var AccessDeniedException = _AccessDeniedException;
var _ConflictException = class _ConflictException extends WorkSpacesWebServiceException {
  /**
   * @internal
   */
  constructor(opts) {
    super({
      name: "ConflictException",
      $fault: "client",
      ...opts
    });
    this.name = "ConflictException";
    this.$fault = "client";
    Object.setPrototypeOf(this, _ConflictException.prototype);
    this.resourceId = opts.resourceId;
    this.resourceType = opts.resourceType;
  }
};
__name(_ConflictException, "ConflictException");
var ConflictException = _ConflictException;
var _InternalServerException = class _InternalServerException extends WorkSpacesWebServiceException {
  /**
   * @internal
   */
  constructor(opts) {
    super({
      name: "InternalServerException",
      $fault: "server",
      ...opts
    });
    this.name = "InternalServerException";
    this.$fault = "server";
    Object.setPrototypeOf(this, _InternalServerException.prototype);
    this.retryAfterSeconds = opts.retryAfterSeconds;
  }
};
__name(_InternalServerException, "InternalServerException");
var InternalServerException = _InternalServerException;
var _ResourceNotFoundException = class _ResourceNotFoundException extends WorkSpacesWebServiceException {
  /**
   * @internal
   */
  constructor(opts) {
    super({
      name: "ResourceNotFoundException",
      $fault: "client",
      ...opts
    });
    this.name = "ResourceNotFoundException";
    this.$fault = "client";
    Object.setPrototypeOf(this, _ResourceNotFoundException.prototype);
    this.resourceId = opts.resourceId;
    this.resourceType = opts.resourceType;
  }
};
__name(_ResourceNotFoundException, "ResourceNotFoundException");
var ResourceNotFoundException = _ResourceNotFoundException;
var _ThrottlingException = class _ThrottlingException extends WorkSpacesWebServiceException {
  /**
   * @internal
   */
  constructor(opts) {
    super({
      name: "ThrottlingException",
      $fault: "client",
      ...opts
    });
    this.name = "ThrottlingException";
    this.$fault = "client";
    Object.setPrototypeOf(this, _ThrottlingException.prototype);
    this.serviceCode = opts.serviceCode;
    this.quotaCode = opts.quotaCode;
    this.retryAfterSeconds = opts.retryAfterSeconds;
  }
};
__name(_ThrottlingException, "ThrottlingException");
var ThrottlingException = _ThrottlingException;
var ValidationExceptionReason = {
  CANNOT_PARSE: "cannotParse",
  FIELD_VALIDATION_FAILED: "fieldValidationFailed",
  OTHER: "other",
  UNKNOWN_OPERATION: "unknownOperation"
};
var _ValidationException = class _ValidationException extends WorkSpacesWebServiceException {
  /**
   * @internal
   */
  constructor(opts) {
    super({
      name: "ValidationException",
      $fault: "client",
      ...opts
    });
    this.name = "ValidationException";
    this.$fault = "client";
    Object.setPrototypeOf(this, _ValidationException.prototype);
    this.reason = opts.reason;
    this.fieldList = opts.fieldList;
  }
};
__name(_ValidationException, "ValidationException");
var ValidationException = _ValidationException;
var AuthenticationType = {
  IAM_IDENTITY_CENTER: "IAM_Identity_Center",
  STANDARD: "Standard"
};
var _ServiceQuotaExceededException = class _ServiceQuotaExceededException extends WorkSpacesWebServiceException {
  /**
   * @internal
   */
  constructor(opts) {
    super({
      name: "ServiceQuotaExceededException",
      $fault: "client",
      ...opts
    });
    this.name = "ServiceQuotaExceededException";
    this.$fault = "client";
    Object.setPrototypeOf(this, _ServiceQuotaExceededException.prototype);
    this.resourceId = opts.resourceId;
    this.resourceType = opts.resourceType;
    this.serviceCode = opts.serviceCode;
    this.quotaCode = opts.quotaCode;
  }
};
__name(_ServiceQuotaExceededException, "ServiceQuotaExceededException");
var ServiceQuotaExceededException = _ServiceQuotaExceededException;
var IdentityProviderType = {
  Facebook: "Facebook",
  Google: "Google",
  LoginWithAmazon: "LoginWithAmazon",
  OIDC: "OIDC",
  SAML: "SAML",
  SignInWithApple: "SignInWithApple"
};
var BrowserType = {
  CHROME: "Chrome"
};
var _InstanceType = {
  STANDARD_LARGE: "standard.large",
  STANDARD_REGULAR: "standard.regular",
  STANDARD_XLARGE: "standard.xlarge"
};
var PortalStatus = {
  ACTIVE: "Active",
  INCOMPLETE: "Incomplete",
  PENDING: "Pending"
};
var RendererType = {
  APPSTREAM: "AppStream"
};
var _TooManyTagsException = class _TooManyTagsException extends WorkSpacesWebServiceException {
  /**
   * @internal
   */
  constructor(opts) {
    super({
      name: "TooManyTagsException",
      $fault: "client",
      ...opts
    });
    this.name = "TooManyTagsException";
    this.$fault = "client";
    Object.setPrototypeOf(this, _TooManyTagsException.prototype);
    this.resourceName = opts.resourceName;
  }
};
__name(_TooManyTagsException, "TooManyTagsException");
var TooManyTagsException = _TooManyTagsException;
var EnabledType = {
  DISABLED: "Disabled",
  ENABLED: "Enabled"
};
var TagFilterSensitiveLog = /* @__PURE__ */ __name((obj) => ({
  ...obj,
  ...obj.Key && { Key: import_smithy_client.SENSITIVE_STRING },
  ...obj.Value && { Value: import_smithy_client.SENSITIVE_STRING }
}), "TagFilterSensitiveLog");
var CreateBrowserSettingsRequestFilterSensitiveLog = /* @__PURE__ */ __name((obj) => ({
  ...obj,
  ...obj.tags && { tags: import_smithy_client.SENSITIVE_STRING },
  ...obj.browserPolicy && { browserPolicy: import_smithy_client.SENSITIVE_STRING }
}), "CreateBrowserSettingsRequestFilterSensitiveLog");
var BrowserSettingsFilterSensitiveLog = /* @__PURE__ */ __name((obj) => ({
  ...obj,
  ...obj.browserPolicy && { browserPolicy: import_smithy_client.SENSITIVE_STRING }
}), "BrowserSettingsFilterSensitiveLog");
var GetBrowserSettingsResponseFilterSensitiveLog = /* @__PURE__ */ __name((obj) => ({
  ...obj,
  ...obj.browserSettings && { browserSettings: BrowserSettingsFilterSensitiveLog(obj.browserSettings) }
}), "GetBrowserSettingsResponseFilterSensitiveLog");
var UpdateBrowserSettingsRequestFilterSensitiveLog = /* @__PURE__ */ __name((obj) => ({
  ...obj,
  ...obj.browserPolicy && { browserPolicy: import_smithy_client.SENSITIVE_STRING }
}), "UpdateBrowserSettingsRequestFilterSensitiveLog");
var UpdateBrowserSettingsResponseFilterSensitiveLog = /* @__PURE__ */ __name((obj) => ({
  ...obj,
  ...obj.browserSettings && { browserSettings: BrowserSettingsFilterSensitiveLog(obj.browserSettings) }
}), "UpdateBrowserSettingsResponseFilterSensitiveLog");
var CreateIdentityProviderRequestFilterSensitiveLog = /* @__PURE__ */ __name((obj) => ({
  ...obj,
  ...obj.identityProviderName && { identityProviderName: import_smithy_client.SENSITIVE_STRING },
  ...obj.identityProviderDetails && { identityProviderDetails: import_smithy_client.SENSITIVE_STRING }
}), "CreateIdentityProviderRequestFilterSensitiveLog");
var IdentityProviderFilterSensitiveLog = /* @__PURE__ */ __name((obj) => ({
  ...obj,
  ...obj.identityProviderName && { identityProviderName: import_smithy_client.SENSITIVE_STRING },
  ...obj.identityProviderDetails && { identityProviderDetails: import_smithy_client.SENSITIVE_STRING }
}), "IdentityProviderFilterSensitiveLog");
var GetIdentityProviderResponseFilterSensitiveLog = /* @__PURE__ */ __name((obj) => ({
  ...obj,
  ...obj.identityProvider && { identityProvider: IdentityProviderFilterSensitiveLog(obj.identityProvider) }
}), "GetIdentityProviderResponseFilterSensitiveLog");
var IdentityProviderSummaryFilterSensitiveLog = /* @__PURE__ */ __name((obj) => ({
  ...obj,
  ...obj.identityProviderName && { identityProviderName: import_smithy_client.SENSITIVE_STRING }
}), "IdentityProviderSummaryFilterSensitiveLog");
var ListIdentityProvidersResponseFilterSensitiveLog = /* @__PURE__ */ __name((obj) => ({
  ...obj,
  ...obj.identityProviders && {
    identityProviders: obj.identityProviders.map((item) => IdentityProviderSummaryFilterSensitiveLog(item))
  }
}), "ListIdentityProvidersResponseFilterSensitiveLog");
var UpdateIdentityProviderRequestFilterSensitiveLog = /* @__PURE__ */ __name((obj) => ({
  ...obj,
  ...obj.identityProviderName && { identityProviderName: import_smithy_client.SENSITIVE_STRING },
  ...obj.identityProviderDetails && { identityProviderDetails: import_smithy_client.SENSITIVE_STRING }
}), "UpdateIdentityProviderRequestFilterSensitiveLog");
var UpdateIdentityProviderResponseFilterSensitiveLog = /* @__PURE__ */ __name((obj) => ({
  ...obj,
  ...obj.identityProvider && { identityProvider: IdentityProviderFilterSensitiveLog(obj.identityProvider) }
}), "UpdateIdentityProviderResponseFilterSensitiveLog");
var IpRuleFilterSensitiveLog = /* @__PURE__ */ __name((obj) => ({
  ...obj,
  ...obj.ipRange && { ipRange: import_smithy_client.SENSITIVE_STRING },
  ...obj.description && { description: import_smithy_client.SENSITIVE_STRING }
}), "IpRuleFilterSensitiveLog");
var CreateIpAccessSettingsRequestFilterSensitiveLog = /* @__PURE__ */ __name((obj) => ({
  ...obj,
  ...obj.displayName && { displayName: import_smithy_client.SENSITIVE_STRING },
  ...obj.description && { description: import_smithy_client.SENSITIVE_STRING },
  ...obj.tags && { tags: import_smithy_client.SENSITIVE_STRING },
  ...obj.ipRules && { ipRules: import_smithy_client.SENSITIVE_STRING }
}), "CreateIpAccessSettingsRequestFilterSensitiveLog");
var IpAccessSettingsFilterSensitiveLog = /* @__PURE__ */ __name((obj) => ({
  ...obj,
  ...obj.ipRules && { ipRules: import_smithy_client.SENSITIVE_STRING },
  ...obj.displayName && { displayName: import_smithy_client.SENSITIVE_STRING },
  ...obj.description && { description: import_smithy_client.SENSITIVE_STRING }
}), "IpAccessSettingsFilterSensitiveLog");
var GetIpAccessSettingsResponseFilterSensitiveLog = /* @__PURE__ */ __name((obj) => ({
  ...obj,
  ...obj.ipAccessSettings && { ipAccessSettings: IpAccessSettingsFilterSensitiveLog(obj.ipAccessSettings) }
}), "GetIpAccessSettingsResponseFilterSensitiveLog");
var IpAccessSettingsSummaryFilterSensitiveLog = /* @__PURE__ */ __name((obj) => ({
  ...obj,
  ...obj.displayName && { displayName: import_smithy_client.SENSITIVE_STRING },
  ...obj.description && { description: import_smithy_client.SENSITIVE_STRING }
}), "IpAccessSettingsSummaryFilterSensitiveLog");
var ListIpAccessSettingsResponseFilterSensitiveLog = /* @__PURE__ */ __name((obj) => ({
  ...obj,
  ...obj.ipAccessSettings && {
    ipAccessSettings: obj.ipAccessSettings.map((item) => IpAccessSettingsSummaryFilterSensitiveLog(item))
  }
}), "ListIpAccessSettingsResponseFilterSensitiveLog");
var UpdateIpAccessSettingsRequestFilterSensitiveLog = /* @__PURE__ */ __name((obj) => ({
  ...obj,
  ...obj.displayName && { displayName: import_smithy_client.SENSITIVE_STRING },
  ...obj.description && { description: import_smithy_client.SENSITIVE_STRING },
  ...obj.ipRules && { ipRules: import_smithy_client.SENSITIVE_STRING }
}), "UpdateIpAccessSettingsRequestFilterSensitiveLog");
var UpdateIpAccessSettingsResponseFilterSensitiveLog = /* @__PURE__ */ __name((obj) => ({
  ...obj,
  ...obj.ipAccessSettings && { ipAccessSettings: IpAccessSettingsFilterSensitiveLog(obj.ipAccessSettings) }
}), "UpdateIpAccessSettingsResponseFilterSensitiveLog");
var ListTagsForResourceResponseFilterSensitiveLog = /* @__PURE__ */ __name((obj) => ({
  ...obj,
  ...obj.tags && { tags: import_smithy_client.SENSITIVE_STRING }
}), "ListTagsForResourceResponseFilterSensitiveLog");
var CreateNetworkSettingsRequestFilterSensitiveLog = /* @__PURE__ */ __name((obj) => ({
  ...obj,
  ...obj.tags && { tags: import_smithy_client.SENSITIVE_STRING }
}), "CreateNetworkSettingsRequestFilterSensitiveLog");
var CreatePortalRequestFilterSensitiveLog = /* @__PURE__ */ __name((obj) => ({
  ...obj,
  ...obj.displayName && { displayName: import_smithy_client.SENSITIVE_STRING },
  ...obj.tags && { tags: import_smithy_client.SENSITIVE_STRING }
}), "CreatePortalRequestFilterSensitiveLog");
var PortalFilterSensitiveLog = /* @__PURE__ */ __name((obj) => ({
  ...obj,
  ...obj.displayName && { displayName: import_smithy_client.SENSITIVE_STRING }
}), "PortalFilterSensitiveLog");
var GetPortalResponseFilterSensitiveLog = /* @__PURE__ */ __name((obj) => ({
  ...obj,
  ...obj.portal && { portal: PortalFilterSensitiveLog(obj.portal) }
}), "GetPortalResponseFilterSensitiveLog");
var PortalSummaryFilterSensitiveLog = /* @__PURE__ */ __name((obj) => ({
  ...obj,
  ...obj.displayName && { displayName: import_smithy_client.SENSITIVE_STRING }
}), "PortalSummaryFilterSensitiveLog");
var ListPortalsResponseFilterSensitiveLog = /* @__PURE__ */ __name((obj) => ({
  ...obj,
  ...obj.portals && { portals: obj.portals.map((item) => PortalSummaryFilterSensitiveLog(item)) }
}), "ListPortalsResponseFilterSensitiveLog");
var UpdatePortalRequestFilterSensitiveLog = /* @__PURE__ */ __name((obj) => ({
  ...obj,
  ...obj.displayName && { displayName: import_smithy_client.SENSITIVE_STRING }
}), "UpdatePortalRequestFilterSensitiveLog");
var UpdatePortalResponseFilterSensitiveLog = /* @__PURE__ */ __name((obj) => ({
  ...obj,
  ...obj.portal && { portal: PortalFilterSensitiveLog(obj.portal) }
}), "UpdatePortalResponseFilterSensitiveLog");
var TagResourceRequestFilterSensitiveLog = /* @__PURE__ */ __name((obj) => ({
  ...obj,
  ...obj.tags && { tags: import_smithy_client.SENSITIVE_STRING }
}), "TagResourceRequestFilterSensitiveLog");
var CreateTrustStoreRequestFilterSensitiveLog = /* @__PURE__ */ __name((obj) => ({
  ...obj,
  ...obj.tags && { tags: import_smithy_client.SENSITIVE_STRING }
}), "CreateTrustStoreRequestFilterSensitiveLog");
var UntagResourceRequestFilterSensitiveLog = /* @__PURE__ */ __name((obj) => ({
  ...obj,
  ...obj.tagKeys && { tagKeys: import_smithy_client.SENSITIVE_STRING }
}), "UntagResourceRequestFilterSensitiveLog");
var CreateUserAccessLoggingSettingsRequestFilterSensitiveLog = /* @__PURE__ */ __name((obj) => ({
  ...obj,
  ...obj.tags && { tags: import_smithy_client.SENSITIVE_STRING }
}), "CreateUserAccessLoggingSettingsRequestFilterSensitiveLog");
var CookieSpecificationFilterSensitiveLog = /* @__PURE__ */ __name((obj) => ({
  ...obj,
  ...obj.domain && { domain: import_smithy_client.SENSITIVE_STRING },
  ...obj.name && { name: import_smithy_client.SENSITIVE_STRING },
  ...obj.path && { path: import_smithy_client.SENSITIVE_STRING }
}), "CookieSpecificationFilterSensitiveLog");
var CookieSynchronizationConfigurationFilterSensitiveLog = /* @__PURE__ */ __name((obj) => ({
  ...obj,
  ...obj.allowlist && { allowlist: obj.allowlist.map((item) => CookieSpecificationFilterSensitiveLog(item)) },
  ...obj.blocklist && { blocklist: obj.blocklist.map((item) => CookieSpecificationFilterSensitiveLog(item)) }
}), "CookieSynchronizationConfigurationFilterSensitiveLog");
var CreateUserSettingsRequestFilterSensitiveLog = /* @__PURE__ */ __name((obj) => ({
  ...obj,
  ...obj.tags && { tags: import_smithy_client.SENSITIVE_STRING },
  ...obj.cookieSynchronizationConfiguration && { cookieSynchronizationConfiguration: import_smithy_client.SENSITIVE_STRING }
}), "CreateUserSettingsRequestFilterSensitiveLog");
var UserSettingsFilterSensitiveLog = /* @__PURE__ */ __name((obj) => ({
  ...obj,
  ...obj.cookieSynchronizationConfiguration && { cookieSynchronizationConfiguration: import_smithy_client.SENSITIVE_STRING }
}), "UserSettingsFilterSensitiveLog");
var GetUserSettingsResponseFilterSensitiveLog = /* @__PURE__ */ __name((obj) => ({
  ...obj,
  ...obj.userSettings && { userSettings: UserSettingsFilterSensitiveLog(obj.userSettings) }
}), "GetUserSettingsResponseFilterSensitiveLog");
var UserSettingsSummaryFilterSensitiveLog = /* @__PURE__ */ __name((obj) => ({
  ...obj,
  ...obj.cookieSynchronizationConfiguration && { cookieSynchronizationConfiguration: import_smithy_client.SENSITIVE_STRING }
}), "UserSettingsSummaryFilterSensitiveLog");
var ListUserSettingsResponseFilterSensitiveLog = /* @__PURE__ */ __name((obj) => ({
  ...obj,
  ...obj.userSettings && {
    userSettings: obj.userSettings.map((item) => UserSettingsSummaryFilterSensitiveLog(item))
  }
}), "ListUserSettingsResponseFilterSensitiveLog");
var UpdateUserSettingsRequestFilterSensitiveLog = /* @__PURE__ */ __name((obj) => ({
  ...obj,
  ...obj.cookieSynchronizationConfiguration && { cookieSynchronizationConfiguration: import_smithy_client.SENSITIVE_STRING }
}), "UpdateUserSettingsRequestFilterSensitiveLog");
var UpdateUserSettingsResponseFilterSensitiveLog = /* @__PURE__ */ __name((obj) => ({
  ...obj,
  ...obj.userSettings && { userSettings: UserSettingsFilterSensitiveLog(obj.userSettings) }
}), "UpdateUserSettingsResponseFilterSensitiveLog");

// src/protocols/Aws_restJson1.ts
var se_AssociateBrowserSettingsCommand = /* @__PURE__ */ __name(async (input, context) => {
  const b = (0, import_core.requestBuilder)(input, context);
  const headers = {};
  b.bp("/portals/{portalArn+}/browserSettings");
  b.p("portalArn", () => input.portalArn, "{portalArn+}", true);
  const query = (0, import_smithy_client.map)({
    [_bSA]: [, (0, import_smithy_client.expectNonNull)(input[_bSA], `browserSettingsArn`)]
  });
  let body;
  b.m("PUT").h(headers).q(query).b(body);
  return b.build();
}, "se_AssociateBrowserSettingsCommand");
var se_AssociateIpAccessSettingsCommand = /* @__PURE__ */ __name(async (input, context) => {
  const b = (0, import_core.requestBuilder)(input, context);
  const headers = {};
  b.bp("/portals/{portalArn+}/ipAccessSettings");
  b.p("portalArn", () => input.portalArn, "{portalArn+}", true);
  const query = (0, import_smithy_client.map)({
    [_iASA]: [, (0, import_smithy_client.expectNonNull)(input[_iASA], `ipAccessSettingsArn`)]
  });
  let body;
  b.m("PUT").h(headers).q(query).b(body);
  return b.build();
}, "se_AssociateIpAccessSettingsCommand");
var se_AssociateNetworkSettingsCommand = /* @__PURE__ */ __name(async (input, context) => {
  const b = (0, import_core.requestBuilder)(input, context);
  const headers = {};
  b.bp("/portals/{portalArn+}/networkSettings");
  b.p("portalArn", () => input.portalArn, "{portalArn+}", true);
  const query = (0, import_smithy_client.map)({
    [_nSA]: [, (0, import_smithy_client.expectNonNull)(input[_nSA], `networkSettingsArn`)]
  });
  let body;
  b.m("PUT").h(headers).q(query).b(body);
  return b.build();
}, "se_AssociateNetworkSettingsCommand");
var se_AssociateTrustStoreCommand = /* @__PURE__ */ __name(async (input, context) => {
  const b = (0, import_core.requestBuilder)(input, context);
  const headers = {};
  b.bp("/portals/{portalArn+}/trustStores");
  b.p("portalArn", () => input.portalArn, "{portalArn+}", true);
  const query = (0, import_smithy_client.map)({
    [_tSA]: [, (0, import_smithy_client.expectNonNull)(input[_tSA], `trustStoreArn`)]
  });
  let body;
  b.m("PUT").h(headers).q(query).b(body);
  return b.build();
}, "se_AssociateTrustStoreCommand");
var se_AssociateUserAccessLoggingSettingsCommand = /* @__PURE__ */ __name(async (input, context) => {
  const b = (0, import_core.requestBuilder)(input, context);
  const headers = {};
  b.bp("/portals/{portalArn+}/userAccessLoggingSettings");
  b.p("portalArn", () => input.portalArn, "{portalArn+}", true);
  const query = (0, import_smithy_client.map)({
    [_uALSA]: [, (0, import_smithy_client.expectNonNull)(input[_uALSA], `userAccessLoggingSettingsArn`)]
  });
  let body;
  b.m("PUT").h(headers).q(query).b(body);
  return b.build();
}, "se_AssociateUserAccessLoggingSettingsCommand");
var se_AssociateUserSettingsCommand = /* @__PURE__ */ __name(async (input, context) => {
  const b = (0, import_core.requestBuilder)(input, context);
  const headers = {};
  b.bp("/portals/{portalArn+}/userSettings");
  b.p("portalArn", () => input.portalArn, "{portalArn+}", true);
  const query = (0, import_smithy_client.map)({
    [_uSA]: [, (0, import_smithy_client.expectNonNull)(input[_uSA], `userSettingsArn`)]
  });
  let body;
  b.m("PUT").h(headers).q(query).b(body);
  return b.build();
}, "se_AssociateUserSettingsCommand");
var se_CreateBrowserSettingsCommand = /* @__PURE__ */ __name(async (input, context) => {
  const b = (0, import_core.requestBuilder)(input, context);
  const headers = {
    "content-type": "application/json"
  };
  b.bp("/browserSettings");
  let body;
  body = JSON.stringify(
    (0, import_smithy_client.take)(input, {
      additionalEncryptionContext: (_) => (0, import_smithy_client._json)(_),
      browserPolicy: [],
      clientToken: [true, (_) => _ ?? (0, import_uuid.v4)()],
      customerManagedKey: [],
      tags: (_) => (0, import_smithy_client._json)(_)
    })
  );
  b.m("POST").h(headers).b(body);
  return b.build();
}, "se_CreateBrowserSettingsCommand");
var se_CreateIdentityProviderCommand = /* @__PURE__ */ __name(async (input, context) => {
  const b = (0, import_core.requestBuilder)(input, context);
  const headers = {
    "content-type": "application/json"
  };
  b.bp("/identityProviders");
  let body;
  body = JSON.stringify(
    (0, import_smithy_client.take)(input, {
      clientToken: [true, (_) => _ ?? (0, import_uuid.v4)()],
      identityProviderDetails: (_) => (0, import_smithy_client._json)(_),
      identityProviderName: [],
      identityProviderType: [],
      portalArn: []
    })
  );
  b.m("POST").h(headers).b(body);
  return b.build();
}, "se_CreateIdentityProviderCommand");
var se_CreateIpAccessSettingsCommand = /* @__PURE__ */ __name(async (input, context) => {
  const b = (0, import_core.requestBuilder)(input, context);
  const headers = {
    "content-type": "application/json"
  };
  b.bp("/ipAccessSettings");
  let body;
  body = JSON.stringify(
    (0, import_smithy_client.take)(input, {
      additionalEncryptionContext: (_) => (0, import_smithy_client._json)(_),
      clientToken: [true, (_) => _ ?? (0, import_uuid.v4)()],
      customerManagedKey: [],
      description: [],
      displayName: [],
      ipRules: (_) => (0, import_smithy_client._json)(_),
      tags: (_) => (0, import_smithy_client._json)(_)
    })
  );
  b.m("POST").h(headers).b(body);
  return b.build();
}, "se_CreateIpAccessSettingsCommand");
var se_CreateNetworkSettingsCommand = /* @__PURE__ */ __name(async (input, context) => {
  const b = (0, import_core.requestBuilder)(input, context);
  const headers = {
    "content-type": "application/json"
  };
  b.bp("/networkSettings");
  let body;
  body = JSON.stringify(
    (0, import_smithy_client.take)(input, {
      clientToken: [true, (_) => _ ?? (0, import_uuid.v4)()],
      securityGroupIds: (_) => (0, import_smithy_client._json)(_),
      subnetIds: (_) => (0, import_smithy_client._json)(_),
      tags: (_) => (0, import_smithy_client._json)(_),
      vpcId: []
    })
  );
  b.m("POST").h(headers).b(body);
  return b.build();
}, "se_CreateNetworkSettingsCommand");
var se_CreatePortalCommand = /* @__PURE__ */ __name(async (input, context) => {
  const b = (0, import_core.requestBuilder)(input, context);
  const headers = {
    "content-type": "application/json"
  };
  b.bp("/portals");
  let body;
  body = JSON.stringify(
    (0, import_smithy_client.take)(input, {
      additionalEncryptionContext: (_) => (0, import_smithy_client._json)(_),
      authenticationType: [],
      clientToken: [true, (_) => _ ?? (0, import_uuid.v4)()],
      customerManagedKey: [],
      displayName: [],
      instanceType: [],
      maxConcurrentSessions: [],
      tags: (_) => (0, import_smithy_client._json)(_)
    })
  );
  b.m("POST").h(headers).b(body);
  return b.build();
}, "se_CreatePortalCommand");
var se_CreateTrustStoreCommand = /* @__PURE__ */ __name(async (input, context) => {
  const b = (0, import_core.requestBuilder)(input, context);
  const headers = {
    "content-type": "application/json"
  };
  b.bp("/trustStores");
  let body;
  body = JSON.stringify(
    (0, import_smithy_client.take)(input, {
      certificateList: (_) => se_CertificateList(_, context),
      clientToken: [true, (_) => _ ?? (0, import_uuid.v4)()],
      tags: (_) => (0, import_smithy_client._json)(_)
    })
  );
  b.m("POST").h(headers).b(body);
  return b.build();
}, "se_CreateTrustStoreCommand");
var se_CreateUserAccessLoggingSettingsCommand = /* @__PURE__ */ __name(async (input, context) => {
  const b = (0, import_core.requestBuilder)(input, context);
  const headers = {
    "content-type": "application/json"
  };
  b.bp("/userAccessLoggingSettings");
  let body;
  body = JSON.stringify(
    (0, import_smithy_client.take)(input, {
      clientToken: [true, (_) => _ ?? (0, import_uuid.v4)()],
      kinesisStreamArn: [],
      tags: (_) => (0, import_smithy_client._json)(_)
    })
  );
  b.m("POST").h(headers).b(body);
  return b.build();
}, "se_CreateUserAccessLoggingSettingsCommand");
var se_CreateUserSettingsCommand = /* @__PURE__ */ __name(async (input, context) => {
  const b = (0, import_core.requestBuilder)(input, context);
  const headers = {
    "content-type": "application/json"
  };
  b.bp("/userSettings");
  let body;
  body = JSON.stringify(
    (0, import_smithy_client.take)(input, {
      additionalEncryptionContext: (_) => (0, import_smithy_client._json)(_),
      clientToken: [true, (_) => _ ?? (0, import_uuid.v4)()],
      cookieSynchronizationConfiguration: (_) => (0, import_smithy_client._json)(_),
      copyAllowed: [],
      customerManagedKey: [],
      disconnectTimeoutInMinutes: [],
      downloadAllowed: [],
      idleDisconnectTimeoutInMinutes: [],
      pasteAllowed: [],
      printAllowed: [],
      tags: (_) => (0, import_smithy_client._json)(_),
      uploadAllowed: []
    })
  );
  b.m("POST").h(headers).b(body);
  return b.build();
}, "se_CreateUserSettingsCommand");
var se_DeleteBrowserSettingsCommand = /* @__PURE__ */ __name(async (input, context) => {
  const b = (0, import_core.requestBuilder)(input, context);
  const headers = {};
  b.bp("/browserSettings/{browserSettingsArn+}");
  b.p("browserSettingsArn", () => input.browserSettingsArn, "{browserSettingsArn+}", true);
  let body;
  b.m("DELETE").h(headers).b(body);
  return b.build();
}, "se_DeleteBrowserSettingsCommand");
var se_DeleteIdentityProviderCommand = /* @__PURE__ */ __name(async (input, context) => {
  const b = (0, import_core.requestBuilder)(input, context);
  const headers = {};
  b.bp("/identityProviders/{identityProviderArn+}");
  b.p("identityProviderArn", () => input.identityProviderArn, "{identityProviderArn+}", true);
  let body;
  b.m("DELETE").h(headers).b(body);
  return b.build();
}, "se_DeleteIdentityProviderCommand");
var se_DeleteIpAccessSettingsCommand = /* @__PURE__ */ __name(async (input, context) => {
  const b = (0, import_core.requestBuilder)(input, context);
  const headers = {};
  b.bp("/ipAccessSettings/{ipAccessSettingsArn+}");
  b.p("ipAccessSettingsArn", () => input.ipAccessSettingsArn, "{ipAccessSettingsArn+}", true);
  let body;
  b.m("DELETE").h(headers).b(body);
  return b.build();
}, "se_DeleteIpAccessSettingsCommand");
var se_DeleteNetworkSettingsCommand = /* @__PURE__ */ __name(async (input, context) => {
  const b = (0, import_core.requestBuilder)(input, context);
  const headers = {};
  b.bp("/networkSettings/{networkSettingsArn+}");
  b.p("networkSettingsArn", () => input.networkSettingsArn, "{networkSettingsArn+}", true);
  let body;
  b.m("DELETE").h(headers).b(body);
  return b.build();
}, "se_DeleteNetworkSettingsCommand");
var se_DeletePortalCommand = /* @__PURE__ */ __name(async (input, context) => {
  const b = (0, import_core.requestBuilder)(input, context);
  const headers = {};
  b.bp("/portals/{portalArn+}");
  b.p("portalArn", () => input.portalArn, "{portalArn+}", true);
  let body;
  b.m("DELETE").h(headers).b(body);
  return b.build();
}, "se_DeletePortalCommand");
var se_DeleteTrustStoreCommand = /* @__PURE__ */ __name(async (input, context) => {
  const b = (0, import_core.requestBuilder)(input, context);
  const headers = {};
  b.bp("/trustStores/{trustStoreArn+}");
  b.p("trustStoreArn", () => input.trustStoreArn, "{trustStoreArn+}", true);
  let body;
  b.m("DELETE").h(headers).b(body);
  return b.build();
}, "se_DeleteTrustStoreCommand");
var se_DeleteUserAccessLoggingSettingsCommand = /* @__PURE__ */ __name(async (input, context) => {
  const b = (0, import_core.requestBuilder)(input, context);
  const headers = {};
  b.bp("/userAccessLoggingSettings/{userAccessLoggingSettingsArn+}");
  b.p(
    "userAccessLoggingSettingsArn",
    () => input.userAccessLoggingSettingsArn,
    "{userAccessLoggingSettingsArn+}",
    true
  );
  let body;
  b.m("DELETE").h(headers).b(body);
  return b.build();
}, "se_DeleteUserAccessLoggingSettingsCommand");
var se_DeleteUserSettingsCommand = /* @__PURE__ */ __name(async (input, context) => {
  const b = (0, import_core.requestBuilder)(input, context);
  const headers = {};
  b.bp("/userSettings/{userSettingsArn+}");
  b.p("userSettingsArn", () => input.userSettingsArn, "{userSettingsArn+}", true);
  let body;
  b.m("DELETE").h(headers).b(body);
  return b.build();
}, "se_DeleteUserSettingsCommand");
var se_DisassociateBrowserSettingsCommand = /* @__PURE__ */ __name(async (input, context) => {
  const b = (0, import_core.requestBuilder)(input, context);
  const headers = {};
  b.bp("/portals/{portalArn+}/browserSettings");
  b.p("portalArn", () => input.portalArn, "{portalArn+}", true);
  let body;
  b.m("DELETE").h(headers).b(body);
  return b.build();
}, "se_DisassociateBrowserSettingsCommand");
var se_DisassociateIpAccessSettingsCommand = /* @__PURE__ */ __name(async (input, context) => {
  const b = (0, import_core.requestBuilder)(input, context);
  const headers = {};
  b.bp("/portals/{portalArn+}/ipAccessSettings");
  b.p("portalArn", () => input.portalArn, "{portalArn+}", true);
  let body;
  b.m("DELETE").h(headers).b(body);
  return b.build();
}, "se_DisassociateIpAccessSettingsCommand");
var se_DisassociateNetworkSettingsCommand = /* @__PURE__ */ __name(async (input, context) => {
  const b = (0, import_core.requestBuilder)(input, context);
  const headers = {};
  b.bp("/portals/{portalArn+}/networkSettings");
  b.p("portalArn", () => input.portalArn, "{portalArn+}", true);
  let body;
  b.m("DELETE").h(headers).b(body);
  return b.build();
}, "se_DisassociateNetworkSettingsCommand");
var se_DisassociateTrustStoreCommand = /* @__PURE__ */ __name(async (input, context) => {
  const b = (0, import_core.requestBuilder)(input, context);
  const headers = {};
  b.bp("/portals/{portalArn+}/trustStores");
  b.p("portalArn", () => input.portalArn, "{portalArn+}", true);
  let body;
  b.m("DELETE").h(headers).b(body);
  return b.build();
}, "se_DisassociateTrustStoreCommand");
var se_DisassociateUserAccessLoggingSettingsCommand = /* @__PURE__ */ __name(async (input, context) => {
  const b = (0, import_core.requestBuilder)(input, context);
  const headers = {};
  b.bp("/portals/{portalArn+}/userAccessLoggingSettings");
  b.p("portalArn", () => input.portalArn, "{portalArn+}", true);
  let body;
  b.m("DELETE").h(headers).b(body);
  return b.build();
}, "se_DisassociateUserAccessLoggingSettingsCommand");
var se_DisassociateUserSettingsCommand = /* @__PURE__ */ __name(async (input, context) => {
  const b = (0, import_core.requestBuilder)(input, context);
  const headers = {};
  b.bp("/portals/{portalArn+}/userSettings");
  b.p("portalArn", () => input.portalArn, "{portalArn+}", true);
  let body;
  b.m("DELETE").h(headers).b(body);
  return b.build();
}, "se_DisassociateUserSettingsCommand");
var se_GetBrowserSettingsCommand = /* @__PURE__ */ __name(async (input, context) => {
  const b = (0, import_core.requestBuilder)(input, context);
  const headers = {};
  b.bp("/browserSettings/{browserSettingsArn+}");
  b.p("browserSettingsArn", () => input.browserSettingsArn, "{browserSettingsArn+}", true);
  let body;
  b.m("GET").h(headers).b(body);
  return b.build();
}, "se_GetBrowserSettingsCommand");
var se_GetIdentityProviderCommand = /* @__PURE__ */ __name(async (input, context) => {
  const b = (0, import_core.requestBuilder)(input, context);
  const headers = {};
  b.bp("/identityProviders/{identityProviderArn+}");
  b.p("identityProviderArn", () => input.identityProviderArn, "{identityProviderArn+}", true);
  let body;
  b.m("GET").h(headers).b(body);
  return b.build();
}, "se_GetIdentityProviderCommand");
var se_GetIpAccessSettingsCommand = /* @__PURE__ */ __name(async (input, context) => {
  const b = (0, import_core.requestBuilder)(input, context);
  const headers = {};
  b.bp("/ipAccessSettings/{ipAccessSettingsArn+}");
  b.p("ipAccessSettingsArn", () => input.ipAccessSettingsArn, "{ipAccessSettingsArn+}", true);
  let body;
  b.m("GET").h(headers).b(body);
  return b.build();
}, "se_GetIpAccessSettingsCommand");
var se_GetNetworkSettingsCommand = /* @__PURE__ */ __name(async (input, context) => {
  const b = (0, import_core.requestBuilder)(input, context);
  const headers = {};
  b.bp("/networkSettings/{networkSettingsArn+}");
  b.p("networkSettingsArn", () => input.networkSettingsArn, "{networkSettingsArn+}", true);
  let body;
  b.m("GET").h(headers).b(body);
  return b.build();
}, "se_GetNetworkSettingsCommand");
var se_GetPortalCommand = /* @__PURE__ */ __name(async (input, context) => {
  const b = (0, import_core.requestBuilder)(input, context);
  const headers = {};
  b.bp("/portals/{portalArn+}");
  b.p("portalArn", () => input.portalArn, "{portalArn+}", true);
  let body;
  b.m("GET").h(headers).b(body);
  return b.build();
}, "se_GetPortalCommand");
var se_GetPortalServiceProviderMetadataCommand = /* @__PURE__ */ __name(async (input, context) => {
  const b = (0, import_core.requestBuilder)(input, context);
  const headers = {};
  b.bp("/portalIdp/{portalArn+}");
  b.p("portalArn", () => input.portalArn, "{portalArn+}", true);
  let body;
  b.m("GET").h(headers).b(body);
  return b.build();
}, "se_GetPortalServiceProviderMetadataCommand");
var se_GetTrustStoreCommand = /* @__PURE__ */ __name(async (input, context) => {
  const b = (0, import_core.requestBuilder)(input, context);
  const headers = {};
  b.bp("/trustStores/{trustStoreArn+}");
  b.p("trustStoreArn", () => input.trustStoreArn, "{trustStoreArn+}", true);
  let body;
  b.m("GET").h(headers).b(body);
  return b.build();
}, "se_GetTrustStoreCommand");
var se_GetTrustStoreCertificateCommand = /* @__PURE__ */ __name(async (input, context) => {
  const b = (0, import_core.requestBuilder)(input, context);
  const headers = {};
  b.bp("/trustStores/{trustStoreArn+}/certificate");
  b.p("trustStoreArn", () => input.trustStoreArn, "{trustStoreArn+}", true);
  const query = (0, import_smithy_client.map)({
    [_t]: [, (0, import_smithy_client.expectNonNull)(input[_t], `thumbprint`)]
  });
  let body;
  b.m("GET").h(headers).q(query).b(body);
  return b.build();
}, "se_GetTrustStoreCertificateCommand");
var se_GetUserAccessLoggingSettingsCommand = /* @__PURE__ */ __name(async (input, context) => {
  const b = (0, import_core.requestBuilder)(input, context);
  const headers = {};
  b.bp("/userAccessLoggingSettings/{userAccessLoggingSettingsArn+}");
  b.p(
    "userAccessLoggingSettingsArn",
    () => input.userAccessLoggingSettingsArn,
    "{userAccessLoggingSettingsArn+}",
    true
  );
  let body;
  b.m("GET").h(headers).b(body);
  return b.build();
}, "se_GetUserAccessLoggingSettingsCommand");
var se_GetUserSettingsCommand = /* @__PURE__ */ __name(async (input, context) => {
  const b = (0, import_core.requestBuilder)(input, context);
  const headers = {};
  b.bp("/userSettings/{userSettingsArn+}");
  b.p("userSettingsArn", () => input.userSettingsArn, "{userSettingsArn+}", true);
  let body;
  b.m("GET").h(headers).b(body);
  return b.build();
}, "se_GetUserSettingsCommand");
var se_ListBrowserSettingsCommand = /* @__PURE__ */ __name(async (input, context) => {
  const b = (0, import_core.requestBuilder)(input, context);
  const headers = {};
  b.bp("/browserSettings");
  const query = (0, import_smithy_client.map)({
    [_nT]: [, input[_nT]],
    [_mR]: [() => input.maxResults !== void 0, () => input[_mR].toString()]
  });
  let body;
  b.m("GET").h(headers).q(query).b(body);
  return b.build();
}, "se_ListBrowserSettingsCommand");
var se_ListIdentityProvidersCommand = /* @__PURE__ */ __name(async (input, context) => {
  const b = (0, import_core.requestBuilder)(input, context);
  const headers = {};
  b.bp("/portals/{portalArn+}/identityProviders");
  b.p("portalArn", () => input.portalArn, "{portalArn+}", true);
  const query = (0, import_smithy_client.map)({
    [_nT]: [, input[_nT]],
    [_mR]: [() => input.maxResults !== void 0, () => input[_mR].toString()]
  });
  let body;
  b.m("GET").h(headers).q(query).b(body);
  return b.build();
}, "se_ListIdentityProvidersCommand");
var se_ListIpAccessSettingsCommand = /* @__PURE__ */ __name(async (input, context) => {
  const b = (0, import_core.requestBuilder)(input, context);
  const headers = {};
  b.bp("/ipAccessSettings");
  const query = (0, import_smithy_client.map)({
    [_nT]: [, input[_nT]],
    [_mR]: [() => input.maxResults !== void 0, () => input[_mR].toString()]
  });
  let body;
  b.m("GET").h(headers).q(query).b(body);
  return b.build();
}, "se_ListIpAccessSettingsCommand");
var se_ListNetworkSettingsCommand = /* @__PURE__ */ __name(async (input, context) => {
  const b = (0, import_core.requestBuilder)(input, context);
  const headers = {};
  b.bp("/networkSettings");
  const query = (0, import_smithy_client.map)({
    [_nT]: [, input[_nT]],
    [_mR]: [() => input.maxResults !== void 0, () => input[_mR].toString()]
  });
  let body;
  b.m("GET").h(headers).q(query).b(body);
  return b.build();
}, "se_ListNetworkSettingsCommand");
var se_ListPortalsCommand = /* @__PURE__ */ __name(async (input, context) => {
  const b = (0, import_core.requestBuilder)(input, context);
  const headers = {};
  b.bp("/portals");
  const query = (0, import_smithy_client.map)({
    [_nT]: [, input[_nT]],
    [_mR]: [() => input.maxResults !== void 0, () => input[_mR].toString()]
  });
  let body;
  b.m("GET").h(headers).q(query).b(body);
  return b.build();
}, "se_ListPortalsCommand");
var se_ListTagsForResourceCommand = /* @__PURE__ */ __name(async (input, context) => {
  const b = (0, import_core.requestBuilder)(input, context);
  const headers = {};
  b.bp("/tags/{resourceArn+}");
  b.p("resourceArn", () => input.resourceArn, "{resourceArn+}", true);
  let body;
  b.m("GET").h(headers).b(body);
  return b.build();
}, "se_ListTagsForResourceCommand");
var se_ListTrustStoreCertificatesCommand = /* @__PURE__ */ __name(async (input, context) => {
  const b = (0, import_core.requestBuilder)(input, context);
  const headers = {};
  b.bp("/trustStores/{trustStoreArn+}/certificates");
  b.p("trustStoreArn", () => input.trustStoreArn, "{trustStoreArn+}", true);
  const query = (0, import_smithy_client.map)({
    [_nT]: [, input[_nT]],
    [_mR]: [() => input.maxResults !== void 0, () => input[_mR].toString()]
  });
  let body;
  b.m("GET").h(headers).q(query).b(body);
  return b.build();
}, "se_ListTrustStoreCertificatesCommand");
var se_ListTrustStoresCommand = /* @__PURE__ */ __name(async (input, context) => {
  const b = (0, import_core.requestBuilder)(input, context);
  const headers = {};
  b.bp("/trustStores");
  const query = (0, import_smithy_client.map)({
    [_nT]: [, input[_nT]],
    [_mR]: [() => input.maxResults !== void 0, () => input[_mR].toString()]
  });
  let body;
  b.m("GET").h(headers).q(query).b(body);
  return b.build();
}, "se_ListTrustStoresCommand");
var se_ListUserAccessLoggingSettingsCommand = /* @__PURE__ */ __name(async (input, context) => {
  const b = (0, import_core.requestBuilder)(input, context);
  const headers = {};
  b.bp("/userAccessLoggingSettings");
  const query = (0, import_smithy_client.map)({
    [_nT]: [, input[_nT]],
    [_mR]: [() => input.maxResults !== void 0, () => input[_mR].toString()]
  });
  let body;
  b.m("GET").h(headers).q(query).b(body);
  return b.build();
}, "se_ListUserAccessLoggingSettingsCommand");
var se_ListUserSettingsCommand = /* @__PURE__ */ __name(async (input, context) => {
  const b = (0, import_core.requestBuilder)(input, context);
  const headers = {};
  b.bp("/userSettings");
  const query = (0, import_smithy_client.map)({
    [_nT]: [, input[_nT]],
    [_mR]: [() => input.maxResults !== void 0, () => input[_mR].toString()]
  });
  let body;
  b.m("GET").h(headers).q(query).b(body);
  return b.build();
}, "se_ListUserSettingsCommand");
var se_TagResourceCommand = /* @__PURE__ */ __name(async (input, context) => {
  const b = (0, import_core.requestBuilder)(input, context);
  const headers = {
    "content-type": "application/json"
  };
  b.bp("/tags/{resourceArn+}");
  b.p("resourceArn", () => input.resourceArn, "{resourceArn+}", true);
  let body;
  body = JSON.stringify(
    (0, import_smithy_client.take)(input, {
      clientToken: [true, (_) => _ ?? (0, import_uuid.v4)()],
      tags: (_) => (0, import_smithy_client._json)(_)
    })
  );
  b.m("POST").h(headers).b(body);
  return b.build();
}, "se_TagResourceCommand");
var se_UntagResourceCommand = /* @__PURE__ */ __name(async (input, context) => {
  const b = (0, import_core.requestBuilder)(input, context);
  const headers = {};
  b.bp("/tags/{resourceArn+}");
  b.p("resourceArn", () => input.resourceArn, "{resourceArn+}", true);
  const query = (0, import_smithy_client.map)({
    [_tK]: [
      (0, import_smithy_client.expectNonNull)(input.tagKeys, `tagKeys`) != null,
      () => (input[_tK] || []).map((_entry) => _entry)
    ]
  });
  let body;
  b.m("DELETE").h(headers).q(query).b(body);
  return b.build();
}, "se_UntagResourceCommand");
var se_UpdateBrowserSettingsCommand = /* @__PURE__ */ __name(async (input, context) => {
  const b = (0, import_core.requestBuilder)(input, context);
  const headers = {
    "content-type": "application/json"
  };
  b.bp("/browserSettings/{browserSettingsArn+}");
  b.p("browserSettingsArn", () => input.browserSettingsArn, "{browserSettingsArn+}", true);
  let body;
  body = JSON.stringify(
    (0, import_smithy_client.take)(input, {
      browserPolicy: [],
      clientToken: [true, (_) => _ ?? (0, import_uuid.v4)()]
    })
  );
  b.m("PATCH").h(headers).b(body);
  return b.build();
}, "se_UpdateBrowserSettingsCommand");
var se_UpdateIdentityProviderCommand = /* @__PURE__ */ __name(async (input, context) => {
  const b = (0, import_core.requestBuilder)(input, context);
  const headers = {
    "content-type": "application/json"
  };
  b.bp("/identityProviders/{identityProviderArn+}");
  b.p("identityProviderArn", () => input.identityProviderArn, "{identityProviderArn+}", true);
  let body;
  body = JSON.stringify(
    (0, import_smithy_client.take)(input, {
      clientToken: [true, (_) => _ ?? (0, import_uuid.v4)()],
      identityProviderDetails: (_) => (0, import_smithy_client._json)(_),
      identityProviderName: [],
      identityProviderType: []
    })
  );
  b.m("PATCH").h(headers).b(body);
  return b.build();
}, "se_UpdateIdentityProviderCommand");
var se_UpdateIpAccessSettingsCommand = /* @__PURE__ */ __name(async (input, context) => {
  const b = (0, import_core.requestBuilder)(input, context);
  const headers = {
    "content-type": "application/json"
  };
  b.bp("/ipAccessSettings/{ipAccessSettingsArn+}");
  b.p("ipAccessSettingsArn", () => input.ipAccessSettingsArn, "{ipAccessSettingsArn+}", true);
  let body;
  body = JSON.stringify(
    (0, import_smithy_client.take)(input, {
      clientToken: [true, (_) => _ ?? (0, import_uuid.v4)()],
      description: [],
      displayName: [],
      ipRules: (_) => (0, import_smithy_client._json)(_)
    })
  );
  b.m("PATCH").h(headers).b(body);
  return b.build();
}, "se_UpdateIpAccessSettingsCommand");
var se_UpdateNetworkSettingsCommand = /* @__PURE__ */ __name(async (input, context) => {
  const b = (0, import_core.requestBuilder)(input, context);
  const headers = {
    "content-type": "application/json"
  };
  b.bp("/networkSettings/{networkSettingsArn+}");
  b.p("networkSettingsArn", () => input.networkSettingsArn, "{networkSettingsArn+}", true);
  let body;
  body = JSON.stringify(
    (0, import_smithy_client.take)(input, {
      clientToken: [true, (_) => _ ?? (0, import_uuid.v4)()],
      securityGroupIds: (_) => (0, import_smithy_client._json)(_),
      subnetIds: (_) => (0, import_smithy_client._json)(_),
      vpcId: []
    })
  );
  b.m("PATCH").h(headers).b(body);
  return b.build();
}, "se_UpdateNetworkSettingsCommand");
var se_UpdatePortalCommand = /* @__PURE__ */ __name(async (input, context) => {
  const b = (0, import_core.requestBuilder)(input, context);
  const headers = {
    "content-type": "application/json"
  };
  b.bp("/portals/{portalArn+}");
  b.p("portalArn", () => input.portalArn, "{portalArn+}", true);
  let body;
  body = JSON.stringify(
    (0, import_smithy_client.take)(input, {
      authenticationType: [],
      displayName: [],
      instanceType: [],
      maxConcurrentSessions: []
    })
  );
  b.m("PUT").h(headers).b(body);
  return b.build();
}, "se_UpdatePortalCommand");
var se_UpdateTrustStoreCommand = /* @__PURE__ */ __name(async (input, context) => {
  const b = (0, import_core.requestBuilder)(input, context);
  const headers = {
    "content-type": "application/json"
  };
  b.bp("/trustStores/{trustStoreArn+}");
  b.p("trustStoreArn", () => input.trustStoreArn, "{trustStoreArn+}", true);
  let body;
  body = JSON.stringify(
    (0, import_smithy_client.take)(input, {
      certificatesToAdd: (_) => se_CertificateList(_, context),
      certificatesToDelete: (_) => (0, import_smithy_client._json)(_),
      clientToken: [true, (_) => _ ?? (0, import_uuid.v4)()]
    })
  );
  b.m("PATCH").h(headers).b(body);
  return b.build();
}, "se_UpdateTrustStoreCommand");
var se_UpdateUserAccessLoggingSettingsCommand = /* @__PURE__ */ __name(async (input, context) => {
  const b = (0, import_core.requestBuilder)(input, context);
  const headers = {
    "content-type": "application/json"
  };
  b.bp("/userAccessLoggingSettings/{userAccessLoggingSettingsArn+}");
  b.p(
    "userAccessLoggingSettingsArn",
    () => input.userAccessLoggingSettingsArn,
    "{userAccessLoggingSettingsArn+}",
    true
  );
  let body;
  body = JSON.stringify(
    (0, import_smithy_client.take)(input, {
      clientToken: [true, (_) => _ ?? (0, import_uuid.v4)()],
      kinesisStreamArn: []
    })
  );
  b.m("PATCH").h(headers).b(body);
  return b.build();
}, "se_UpdateUserAccessLoggingSettingsCommand");
var se_UpdateUserSettingsCommand = /* @__PURE__ */ __name(async (input, context) => {
  const b = (0, import_core.requestBuilder)(input, context);
  const headers = {
    "content-type": "application/json"
  };
  b.bp("/userSettings/{userSettingsArn+}");
  b.p("userSettingsArn", () => input.userSettingsArn, "{userSettingsArn+}", true);
  let body;
  body = JSON.stringify(
    (0, import_smithy_client.take)(input, {
      clientToken: [true, (_) => _ ?? (0, import_uuid.v4)()],
      cookieSynchronizationConfiguration: (_) => (0, import_smithy_client._json)(_),
      copyAllowed: [],
      disconnectTimeoutInMinutes: [],
      downloadAllowed: [],
      idleDisconnectTimeoutInMinutes: [],
      pasteAllowed: [],
      printAllowed: [],
      uploadAllowed: []
    })
  );
  b.m("PATCH").h(headers).b(body);
  return b.build();
}, "se_UpdateUserSettingsCommand");
var de_AssociateBrowserSettingsCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode !== 200 && output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const contents = (0, import_smithy_client.map)({
    $metadata: deserializeMetadata(output)
  });
  const data = (0, import_smithy_client.expectNonNull)((0, import_smithy_client.expectObject)(await (0, import_core2.parseJsonBody)(output.body, context)), "body");
  const doc = (0, import_smithy_client.take)(data, {
    browserSettingsArn: import_smithy_client.expectString,
    portalArn: import_smithy_client.expectString
  });
  Object.assign(contents, doc);
  return contents;
}, "de_AssociateBrowserSettingsCommand");
var de_AssociateIpAccessSettingsCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode !== 200 && output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const contents = (0, import_smithy_client.map)({
    $metadata: deserializeMetadata(output)
  });
  const data = (0, import_smithy_client.expectNonNull)((0, import_smithy_client.expectObject)(await (0, import_core2.parseJsonBody)(output.body, context)), "body");
  const doc = (0, import_smithy_client.take)(data, {
    ipAccessSettingsArn: import_smithy_client.expectString,
    portalArn: import_smithy_client.expectString
  });
  Object.assign(contents, doc);
  return contents;
}, "de_AssociateIpAccessSettingsCommand");
var de_AssociateNetworkSettingsCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode !== 200 && output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const contents = (0, import_smithy_client.map)({
    $metadata: deserializeMetadata(output)
  });
  const data = (0, import_smithy_client.expectNonNull)((0, import_smithy_client.expectObject)(await (0, import_core2.parseJsonBody)(output.body, context)), "body");
  const doc = (0, import_smithy_client.take)(data, {
    networkSettingsArn: import_smithy_client.expectString,
    portalArn: import_smithy_client.expectString
  });
  Object.assign(contents, doc);
  return contents;
}, "de_AssociateNetworkSettingsCommand");
var de_AssociateTrustStoreCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode !== 200 && output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const contents = (0, import_smithy_client.map)({
    $metadata: deserializeMetadata(output)
  });
  const data = (0, import_smithy_client.expectNonNull)((0, import_smithy_client.expectObject)(await (0, import_core2.parseJsonBody)(output.body, context)), "body");
  const doc = (0, import_smithy_client.take)(data, {
    portalArn: import_smithy_client.expectString,
    trustStoreArn: import_smithy_client.expectString
  });
  Object.assign(contents, doc);
  return contents;
}, "de_AssociateTrustStoreCommand");
var de_AssociateUserAccessLoggingSettingsCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode !== 200 && output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const contents = (0, import_smithy_client.map)({
    $metadata: deserializeMetadata(output)
  });
  const data = (0, import_smithy_client.expectNonNull)((0, import_smithy_client.expectObject)(await (0, import_core2.parseJsonBody)(output.body, context)), "body");
  const doc = (0, import_smithy_client.take)(data, {
    portalArn: import_smithy_client.expectString,
    userAccessLoggingSettingsArn: import_smithy_client.expectString
  });
  Object.assign(contents, doc);
  return contents;
}, "de_AssociateUserAccessLoggingSettingsCommand");
var de_AssociateUserSettingsCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode !== 200 && output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const contents = (0, import_smithy_client.map)({
    $metadata: deserializeMetadata(output)
  });
  const data = (0, import_smithy_client.expectNonNull)((0, import_smithy_client.expectObject)(await (0, import_core2.parseJsonBody)(output.body, context)), "body");
  const doc = (0, import_smithy_client.take)(data, {
    portalArn: import_smithy_client.expectString,
    userSettingsArn: import_smithy_client.expectString
  });
  Object.assign(contents, doc);
  return contents;
}, "de_AssociateUserSettingsCommand");
var de_CreateBrowserSettingsCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode !== 200 && output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const contents = (0, import_smithy_client.map)({
    $metadata: deserializeMetadata(output)
  });
  const data = (0, import_smithy_client.expectNonNull)((0, import_smithy_client.expectObject)(await (0, import_core2.parseJsonBody)(output.body, context)), "body");
  const doc = (0, import_smithy_client.take)(data, {
    browserSettingsArn: import_smithy_client.expectString
  });
  Object.assign(contents, doc);
  return contents;
}, "de_CreateBrowserSettingsCommand");
var de_CreateIdentityProviderCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode !== 200 && output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const contents = (0, import_smithy_client.map)({
    $metadata: deserializeMetadata(output)
  });
  const data = (0, import_smithy_client.expectNonNull)((0, import_smithy_client.expectObject)(await (0, import_core2.parseJsonBody)(output.body, context)), "body");
  const doc = (0, import_smithy_client.take)(data, {
    identityProviderArn: import_smithy_client.expectString
  });
  Object.assign(contents, doc);
  return contents;
}, "de_CreateIdentityProviderCommand");
var de_CreateIpAccessSettingsCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode !== 200 && output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const contents = (0, import_smithy_client.map)({
    $metadata: deserializeMetadata(output)
  });
  const data = (0, import_smithy_client.expectNonNull)((0, import_smithy_client.expectObject)(await (0, import_core2.parseJsonBody)(output.body, context)), "body");
  const doc = (0, import_smithy_client.take)(data, {
    ipAccessSettingsArn: import_smithy_client.expectString
  });
  Object.assign(contents, doc);
  return contents;
}, "de_CreateIpAccessSettingsCommand");
var de_CreateNetworkSettingsCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode !== 200 && output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const contents = (0, import_smithy_client.map)({
    $metadata: deserializeMetadata(output)
  });
  const data = (0, import_smithy_client.expectNonNull)((0, import_smithy_client.expectObject)(await (0, import_core2.parseJsonBody)(output.body, context)), "body");
  const doc = (0, import_smithy_client.take)(data, {
    networkSettingsArn: import_smithy_client.expectString
  });
  Object.assign(contents, doc);
  return contents;
}, "de_CreateNetworkSettingsCommand");
var de_CreatePortalCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode !== 200 && output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const contents = (0, import_smithy_client.map)({
    $metadata: deserializeMetadata(output)
  });
  const data = (0, import_smithy_client.expectNonNull)((0, import_smithy_client.expectObject)(await (0, import_core2.parseJsonBody)(output.body, context)), "body");
  const doc = (0, import_smithy_client.take)(data, {
    portalArn: import_smithy_client.expectString,
    portalEndpoint: import_smithy_client.expectString
  });
  Object.assign(contents, doc);
  return contents;
}, "de_CreatePortalCommand");
var de_CreateTrustStoreCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode !== 200 && output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const contents = (0, import_smithy_client.map)({
    $metadata: deserializeMetadata(output)
  });
  const data = (0, import_smithy_client.expectNonNull)((0, import_smithy_client.expectObject)(await (0, import_core2.parseJsonBody)(output.body, context)), "body");
  const doc = (0, import_smithy_client.take)(data, {
    trustStoreArn: import_smithy_client.expectString
  });
  Object.assign(contents, doc);
  return contents;
}, "de_CreateTrustStoreCommand");
var de_CreateUserAccessLoggingSettingsCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode !== 200 && output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const contents = (0, import_smithy_client.map)({
    $metadata: deserializeMetadata(output)
  });
  const data = (0, import_smithy_client.expectNonNull)((0, import_smithy_client.expectObject)(await (0, import_core2.parseJsonBody)(output.body, context)), "body");
  const doc = (0, import_smithy_client.take)(data, {
    userAccessLoggingSettingsArn: import_smithy_client.expectString
  });
  Object.assign(contents, doc);
  return contents;
}, "de_CreateUserAccessLoggingSettingsCommand");
var de_CreateUserSettingsCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode !== 200 && output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const contents = (0, import_smithy_client.map)({
    $metadata: deserializeMetadata(output)
  });
  const data = (0, import_smithy_client.expectNonNull)((0, import_smithy_client.expectObject)(await (0, import_core2.parseJsonBody)(output.body, context)), "body");
  const doc = (0, import_smithy_client.take)(data, {
    userSettingsArn: import_smithy_client.expectString
  });
  Object.assign(contents, doc);
  return contents;
}, "de_CreateUserSettingsCommand");
var de_DeleteBrowserSettingsCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode !== 200 && output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const contents = (0, import_smithy_client.map)({
    $metadata: deserializeMetadata(output)
  });
  await (0, import_smithy_client.collectBody)(output.body, context);
  return contents;
}, "de_DeleteBrowserSettingsCommand");
var de_DeleteIdentityProviderCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode !== 200 && output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const contents = (0, import_smithy_client.map)({
    $metadata: deserializeMetadata(output)
  });
  await (0, import_smithy_client.collectBody)(output.body, context);
  return contents;
}, "de_DeleteIdentityProviderCommand");
var de_DeleteIpAccessSettingsCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode !== 200 && output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const contents = (0, import_smithy_client.map)({
    $metadata: deserializeMetadata(output)
  });
  await (0, import_smithy_client.collectBody)(output.body, context);
  return contents;
}, "de_DeleteIpAccessSettingsCommand");
var de_DeleteNetworkSettingsCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode !== 200 && output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const contents = (0, import_smithy_client.map)({
    $metadata: deserializeMetadata(output)
  });
  await (0, import_smithy_client.collectBody)(output.body, context);
  return contents;
}, "de_DeleteNetworkSettingsCommand");
var de_DeletePortalCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode !== 200 && output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const contents = (0, import_smithy_client.map)({
    $metadata: deserializeMetadata(output)
  });
  await (0, import_smithy_client.collectBody)(output.body, context);
  return contents;
}, "de_DeletePortalCommand");
var de_DeleteTrustStoreCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode !== 200 && output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const contents = (0, import_smithy_client.map)({
    $metadata: deserializeMetadata(output)
  });
  await (0, import_smithy_client.collectBody)(output.body, context);
  return contents;
}, "de_DeleteTrustStoreCommand");
var de_DeleteUserAccessLoggingSettingsCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode !== 200 && output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const contents = (0, import_smithy_client.map)({
    $metadata: deserializeMetadata(output)
  });
  await (0, import_smithy_client.collectBody)(output.body, context);
  return contents;
}, "de_DeleteUserAccessLoggingSettingsCommand");
var de_DeleteUserSettingsCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode !== 200 && output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const contents = (0, import_smithy_client.map)({
    $metadata: deserializeMetadata(output)
  });
  await (0, import_smithy_client.collectBody)(output.body, context);
  return contents;
}, "de_DeleteUserSettingsCommand");
var de_DisassociateBrowserSettingsCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode !== 200 && output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const contents = (0, import_smithy_client.map)({
    $metadata: deserializeMetadata(output)
  });
  await (0, import_smithy_client.collectBody)(output.body, context);
  return contents;
}, "de_DisassociateBrowserSettingsCommand");
var de_DisassociateIpAccessSettingsCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode !== 200 && output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const contents = (0, import_smithy_client.map)({
    $metadata: deserializeMetadata(output)
  });
  await (0, import_smithy_client.collectBody)(output.body, context);
  return contents;
}, "de_DisassociateIpAccessSettingsCommand");
var de_DisassociateNetworkSettingsCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode !== 200 && output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const contents = (0, import_smithy_client.map)({
    $metadata: deserializeMetadata(output)
  });
  await (0, import_smithy_client.collectBody)(output.body, context);
  return contents;
}, "de_DisassociateNetworkSettingsCommand");
var de_DisassociateTrustStoreCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode !== 200 && output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const contents = (0, import_smithy_client.map)({
    $metadata: deserializeMetadata(output)
  });
  await (0, import_smithy_client.collectBody)(output.body, context);
  return contents;
}, "de_DisassociateTrustStoreCommand");
var de_DisassociateUserAccessLoggingSettingsCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode !== 200 && output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const contents = (0, import_smithy_client.map)({
    $metadata: deserializeMetadata(output)
  });
  await (0, import_smithy_client.collectBody)(output.body, context);
  return contents;
}, "de_DisassociateUserAccessLoggingSettingsCommand");
var de_DisassociateUserSettingsCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode !== 200 && output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const contents = (0, import_smithy_client.map)({
    $metadata: deserializeMetadata(output)
  });
  await (0, import_smithy_client.collectBody)(output.body, context);
  return contents;
}, "de_DisassociateUserSettingsCommand");
var de_GetBrowserSettingsCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode !== 200 && output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const contents = (0, import_smithy_client.map)({
    $metadata: deserializeMetadata(output)
  });
  const data = (0, import_smithy_client.expectNonNull)((0, import_smithy_client.expectObject)(await (0, import_core2.parseJsonBody)(output.body, context)), "body");
  const doc = (0, import_smithy_client.take)(data, {
    browserSettings: import_smithy_client._json
  });
  Object.assign(contents, doc);
  return contents;
}, "de_GetBrowserSettingsCommand");
var de_GetIdentityProviderCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode !== 200 && output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const contents = (0, import_smithy_client.map)({
    $metadata: deserializeMetadata(output)
  });
  const data = (0, import_smithy_client.expectNonNull)((0, import_smithy_client.expectObject)(await (0, import_core2.parseJsonBody)(output.body, context)), "body");
  const doc = (0, import_smithy_client.take)(data, {
    identityProvider: import_smithy_client._json
  });
  Object.assign(contents, doc);
  return contents;
}, "de_GetIdentityProviderCommand");
var de_GetIpAccessSettingsCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode !== 200 && output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const contents = (0, import_smithy_client.map)({
    $metadata: deserializeMetadata(output)
  });
  const data = (0, import_smithy_client.expectNonNull)((0, import_smithy_client.expectObject)(await (0, import_core2.parseJsonBody)(output.body, context)), "body");
  const doc = (0, import_smithy_client.take)(data, {
    ipAccessSettings: (_) => de_IpAccessSettings(_, context)
  });
  Object.assign(contents, doc);
  return contents;
}, "de_GetIpAccessSettingsCommand");
var de_GetNetworkSettingsCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode !== 200 && output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const contents = (0, import_smithy_client.map)({
    $metadata: deserializeMetadata(output)
  });
  const data = (0, import_smithy_client.expectNonNull)((0, import_smithy_client.expectObject)(await (0, import_core2.parseJsonBody)(output.body, context)), "body");
  const doc = (0, import_smithy_client.take)(data, {
    networkSettings: import_smithy_client._json
  });
  Object.assign(contents, doc);
  return contents;
}, "de_GetNetworkSettingsCommand");
var de_GetPortalCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode !== 200 && output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const contents = (0, import_smithy_client.map)({
    $metadata: deserializeMetadata(output)
  });
  const data = (0, import_smithy_client.expectNonNull)((0, import_smithy_client.expectObject)(await (0, import_core2.parseJsonBody)(output.body, context)), "body");
  const doc = (0, import_smithy_client.take)(data, {
    portal: (_) => de_Portal(_, context)
  });
  Object.assign(contents, doc);
  return contents;
}, "de_GetPortalCommand");
var de_GetPortalServiceProviderMetadataCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode !== 200 && output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const contents = (0, import_smithy_client.map)({
    $metadata: deserializeMetadata(output)
  });
  const data = (0, import_smithy_client.expectNonNull)((0, import_smithy_client.expectObject)(await (0, import_core2.parseJsonBody)(output.body, context)), "body");
  const doc = (0, import_smithy_client.take)(data, {
    portalArn: import_smithy_client.expectString,
    serviceProviderSamlMetadata: import_smithy_client.expectString
  });
  Object.assign(contents, doc);
  return contents;
}, "de_GetPortalServiceProviderMetadataCommand");
var de_GetTrustStoreCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode !== 200 && output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const contents = (0, import_smithy_client.map)({
    $metadata: deserializeMetadata(output)
  });
  const data = (0, import_smithy_client.expectNonNull)((0, import_smithy_client.expectObject)(await (0, import_core2.parseJsonBody)(output.body, context)), "body");
  const doc = (0, import_smithy_client.take)(data, {
    trustStore: import_smithy_client._json
  });
  Object.assign(contents, doc);
  return contents;
}, "de_GetTrustStoreCommand");
var de_GetTrustStoreCertificateCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode !== 200 && output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const contents = (0, import_smithy_client.map)({
    $metadata: deserializeMetadata(output)
  });
  const data = (0, import_smithy_client.expectNonNull)((0, import_smithy_client.expectObject)(await (0, import_core2.parseJsonBody)(output.body, context)), "body");
  const doc = (0, import_smithy_client.take)(data, {
    certificate: (_) => de_Certificate(_, context),
    trustStoreArn: import_smithy_client.expectString
  });
  Object.assign(contents, doc);
  return contents;
}, "de_GetTrustStoreCertificateCommand");
var de_GetUserAccessLoggingSettingsCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode !== 200 && output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const contents = (0, import_smithy_client.map)({
    $metadata: deserializeMetadata(output)
  });
  const data = (0, import_smithy_client.expectNonNull)((0, import_smithy_client.expectObject)(await (0, import_core2.parseJsonBody)(output.body, context)), "body");
  const doc = (0, import_smithy_client.take)(data, {
    userAccessLoggingSettings: import_smithy_client._json
  });
  Object.assign(contents, doc);
  return contents;
}, "de_GetUserAccessLoggingSettingsCommand");
var de_GetUserSettingsCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode !== 200 && output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const contents = (0, import_smithy_client.map)({
    $metadata: deserializeMetadata(output)
  });
  const data = (0, import_smithy_client.expectNonNull)((0, import_smithy_client.expectObject)(await (0, import_core2.parseJsonBody)(output.body, context)), "body");
  const doc = (0, import_smithy_client.take)(data, {
    userSettings: import_smithy_client._json
  });
  Object.assign(contents, doc);
  return contents;
}, "de_GetUserSettingsCommand");
var de_ListBrowserSettingsCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode !== 200 && output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const contents = (0, import_smithy_client.map)({
    $metadata: deserializeMetadata(output)
  });
  const data = (0, import_smithy_client.expectNonNull)((0, import_smithy_client.expectObject)(await (0, import_core2.parseJsonBody)(output.body, context)), "body");
  const doc = (0, import_smithy_client.take)(data, {
    browserSettings: import_smithy_client._json,
    nextToken: import_smithy_client.expectString
  });
  Object.assign(contents, doc);
  return contents;
}, "de_ListBrowserSettingsCommand");
var de_ListIdentityProvidersCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode !== 200 && output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const contents = (0, import_smithy_client.map)({
    $metadata: deserializeMetadata(output)
  });
  const data = (0, import_smithy_client.expectNonNull)((0, import_smithy_client.expectObject)(await (0, import_core2.parseJsonBody)(output.body, context)), "body");
  const doc = (0, import_smithy_client.take)(data, {
    identityProviders: import_smithy_client._json,
    nextToken: import_smithy_client.expectString
  });
  Object.assign(contents, doc);
  return contents;
}, "de_ListIdentityProvidersCommand");
var de_ListIpAccessSettingsCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode !== 200 && output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const contents = (0, import_smithy_client.map)({
    $metadata: deserializeMetadata(output)
  });
  const data = (0, import_smithy_client.expectNonNull)((0, import_smithy_client.expectObject)(await (0, import_core2.parseJsonBody)(output.body, context)), "body");
  const doc = (0, import_smithy_client.take)(data, {
    ipAccessSettings: (_) => de_IpAccessSettingsList(_, context),
    nextToken: import_smithy_client.expectString
  });
  Object.assign(contents, doc);
  return contents;
}, "de_ListIpAccessSettingsCommand");
var de_ListNetworkSettingsCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode !== 200 && output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const contents = (0, import_smithy_client.map)({
    $metadata: deserializeMetadata(output)
  });
  const data = (0, import_smithy_client.expectNonNull)((0, import_smithy_client.expectObject)(await (0, import_core2.parseJsonBody)(output.body, context)), "body");
  const doc = (0, import_smithy_client.take)(data, {
    networkSettings: import_smithy_client._json,
    nextToken: import_smithy_client.expectString
  });
  Object.assign(contents, doc);
  return contents;
}, "de_ListNetworkSettingsCommand");
var de_ListPortalsCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode !== 200 && output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const contents = (0, import_smithy_client.map)({
    $metadata: deserializeMetadata(output)
  });
  const data = (0, import_smithy_client.expectNonNull)((0, import_smithy_client.expectObject)(await (0, import_core2.parseJsonBody)(output.body, context)), "body");
  const doc = (0, import_smithy_client.take)(data, {
    nextToken: import_smithy_client.expectString,
    portals: (_) => de_PortalList(_, context)
  });
  Object.assign(contents, doc);
  return contents;
}, "de_ListPortalsCommand");
var de_ListTagsForResourceCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode !== 200 && output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const contents = (0, import_smithy_client.map)({
    $metadata: deserializeMetadata(output)
  });
  const data = (0, import_smithy_client.expectNonNull)((0, import_smithy_client.expectObject)(await (0, import_core2.parseJsonBody)(output.body, context)), "body");
  const doc = (0, import_smithy_client.take)(data, {
    tags: import_smithy_client._json
  });
  Object.assign(contents, doc);
  return contents;
}, "de_ListTagsForResourceCommand");
var de_ListTrustStoreCertificatesCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode !== 200 && output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const contents = (0, import_smithy_client.map)({
    $metadata: deserializeMetadata(output)
  });
  const data = (0, import_smithy_client.expectNonNull)((0, import_smithy_client.expectObject)(await (0, import_core2.parseJsonBody)(output.body, context)), "body");
  const doc = (0, import_smithy_client.take)(data, {
    certificateList: (_) => de_CertificateSummaryList(_, context),
    nextToken: import_smithy_client.expectString,
    trustStoreArn: import_smithy_client.expectString
  });
  Object.assign(contents, doc);
  return contents;
}, "de_ListTrustStoreCertificatesCommand");
var de_ListTrustStoresCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode !== 200 && output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const contents = (0, import_smithy_client.map)({
    $metadata: deserializeMetadata(output)
  });
  const data = (0, import_smithy_client.expectNonNull)((0, import_smithy_client.expectObject)(await (0, import_core2.parseJsonBody)(output.body, context)), "body");
  const doc = (0, import_smithy_client.take)(data, {
    nextToken: import_smithy_client.expectString,
    trustStores: import_smithy_client._json
  });
  Object.assign(contents, doc);
  return contents;
}, "de_ListTrustStoresCommand");
var de_ListUserAccessLoggingSettingsCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode !== 200 && output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const contents = (0, import_smithy_client.map)({
    $metadata: deserializeMetadata(output)
  });
  const data = (0, import_smithy_client.expectNonNull)((0, import_smithy_client.expectObject)(await (0, import_core2.parseJsonBody)(output.body, context)), "body");
  const doc = (0, import_smithy_client.take)(data, {
    nextToken: import_smithy_client.expectString,
    userAccessLoggingSettings: import_smithy_client._json
  });
  Object.assign(contents, doc);
  return contents;
}, "de_ListUserAccessLoggingSettingsCommand");
var de_ListUserSettingsCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode !== 200 && output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const contents = (0, import_smithy_client.map)({
    $metadata: deserializeMetadata(output)
  });
  const data = (0, import_smithy_client.expectNonNull)((0, import_smithy_client.expectObject)(await (0, import_core2.parseJsonBody)(output.body, context)), "body");
  const doc = (0, import_smithy_client.take)(data, {
    nextToken: import_smithy_client.expectString,
    userSettings: import_smithy_client._json
  });
  Object.assign(contents, doc);
  return contents;
}, "de_ListUserSettingsCommand");
var de_TagResourceCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode !== 200 && output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const contents = (0, import_smithy_client.map)({
    $metadata: deserializeMetadata(output)
  });
  await (0, import_smithy_client.collectBody)(output.body, context);
  return contents;
}, "de_TagResourceCommand");
var de_UntagResourceCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode !== 200 && output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const contents = (0, import_smithy_client.map)({
    $metadata: deserializeMetadata(output)
  });
  await (0, import_smithy_client.collectBody)(output.body, context);
  return contents;
}, "de_UntagResourceCommand");
var de_UpdateBrowserSettingsCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode !== 200 && output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const contents = (0, import_smithy_client.map)({
    $metadata: deserializeMetadata(output)
  });
  const data = (0, import_smithy_client.expectNonNull)((0, import_smithy_client.expectObject)(await (0, import_core2.parseJsonBody)(output.body, context)), "body");
  const doc = (0, import_smithy_client.take)(data, {
    browserSettings: import_smithy_client._json
  });
  Object.assign(contents, doc);
  return contents;
}, "de_UpdateBrowserSettingsCommand");
var de_UpdateIdentityProviderCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode !== 200 && output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const contents = (0, import_smithy_client.map)({
    $metadata: deserializeMetadata(output)
  });
  const data = (0, import_smithy_client.expectNonNull)((0, import_smithy_client.expectObject)(await (0, import_core2.parseJsonBody)(output.body, context)), "body");
  const doc = (0, import_smithy_client.take)(data, {
    identityProvider: import_smithy_client._json
  });
  Object.assign(contents, doc);
  return contents;
}, "de_UpdateIdentityProviderCommand");
var de_UpdateIpAccessSettingsCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode !== 200 && output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const contents = (0, import_smithy_client.map)({
    $metadata: deserializeMetadata(output)
  });
  const data = (0, import_smithy_client.expectNonNull)((0, import_smithy_client.expectObject)(await (0, import_core2.parseJsonBody)(output.body, context)), "body");
  const doc = (0, import_smithy_client.take)(data, {
    ipAccessSettings: (_) => de_IpAccessSettings(_, context)
  });
  Object.assign(contents, doc);
  return contents;
}, "de_UpdateIpAccessSettingsCommand");
var de_UpdateNetworkSettingsCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode !== 200 && output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const contents = (0, import_smithy_client.map)({
    $metadata: deserializeMetadata(output)
  });
  const data = (0, import_smithy_client.expectNonNull)((0, import_smithy_client.expectObject)(await (0, import_core2.parseJsonBody)(output.body, context)), "body");
  const doc = (0, import_smithy_client.take)(data, {
    networkSettings: import_smithy_client._json
  });
  Object.assign(contents, doc);
  return contents;
}, "de_UpdateNetworkSettingsCommand");
var de_UpdatePortalCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode !== 200 && output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const contents = (0, import_smithy_client.map)({
    $metadata: deserializeMetadata(output)
  });
  const data = (0, import_smithy_client.expectNonNull)((0, import_smithy_client.expectObject)(await (0, import_core2.parseJsonBody)(output.body, context)), "body");
  const doc = (0, import_smithy_client.take)(data, {
    portal: (_) => de_Portal(_, context)
  });
  Object.assign(contents, doc);
  return contents;
}, "de_UpdatePortalCommand");
var de_UpdateTrustStoreCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode !== 200 && output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const contents = (0, import_smithy_client.map)({
    $metadata: deserializeMetadata(output)
  });
  const data = (0, import_smithy_client.expectNonNull)((0, import_smithy_client.expectObject)(await (0, import_core2.parseJsonBody)(output.body, context)), "body");
  const doc = (0, import_smithy_client.take)(data, {
    trustStoreArn: import_smithy_client.expectString
  });
  Object.assign(contents, doc);
  return contents;
}, "de_UpdateTrustStoreCommand");
var de_UpdateUserAccessLoggingSettingsCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode !== 200 && output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const contents = (0, import_smithy_client.map)({
    $metadata: deserializeMetadata(output)
  });
  const data = (0, import_smithy_client.expectNonNull)((0, import_smithy_client.expectObject)(await (0, import_core2.parseJsonBody)(output.body, context)), "body");
  const doc = (0, import_smithy_client.take)(data, {
    userAccessLoggingSettings: import_smithy_client._json
  });
  Object.assign(contents, doc);
  return contents;
}, "de_UpdateUserAccessLoggingSettingsCommand");
var de_UpdateUserSettingsCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode !== 200 && output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const contents = (0, import_smithy_client.map)({
    $metadata: deserializeMetadata(output)
  });
  const data = (0, import_smithy_client.expectNonNull)((0, import_smithy_client.expectObject)(await (0, import_core2.parseJsonBody)(output.body, context)), "body");
  const doc = (0, import_smithy_client.take)(data, {
    userSettings: import_smithy_client._json
  });
  Object.assign(contents, doc);
  return contents;
}, "de_UpdateUserSettingsCommand");
var de_CommandError = /* @__PURE__ */ __name(async (output, context) => {
  const parsedOutput = {
    ...output,
    body: await (0, import_core2.parseJsonErrorBody)(output.body, context)
  };
  const errorCode = (0, import_core2.loadRestJsonErrorCode)(output, parsedOutput.body);
  switch (errorCode) {
    case "AccessDeniedException":
    case "com.amazonaws.workspacesweb#AccessDeniedException":
      throw await de_AccessDeniedExceptionRes(parsedOutput, context);
    case "ConflictException":
    case "com.amazonaws.workspacesweb#ConflictException":
      throw await de_ConflictExceptionRes(parsedOutput, context);
    case "InternalServerException":
    case "com.amazonaws.workspacesweb#InternalServerException":
      throw await de_InternalServerExceptionRes(parsedOutput, context);
    case "ResourceNotFoundException":
    case "com.amazonaws.workspacesweb#ResourceNotFoundException":
      throw await de_ResourceNotFoundExceptionRes(parsedOutput, context);
    case "ThrottlingException":
    case "com.amazonaws.workspacesweb#ThrottlingException":
      throw await de_ThrottlingExceptionRes(parsedOutput, context);
    case "ValidationException":
    case "com.amazonaws.workspacesweb#ValidationException":
      throw await de_ValidationExceptionRes(parsedOutput, context);
    case "ServiceQuotaExceededException":
    case "com.amazonaws.workspacesweb#ServiceQuotaExceededException":
      throw await de_ServiceQuotaExceededExceptionRes(parsedOutput, context);
    case "TooManyTagsException":
    case "com.amazonaws.workspacesweb#TooManyTagsException":
      throw await de_TooManyTagsExceptionRes(parsedOutput, context);
    default:
      const parsedBody = parsedOutput.body;
      return throwDefaultError({
        output,
        parsedBody,
        errorCode
      });
  }
}, "de_CommandError");
var throwDefaultError = (0, import_smithy_client.withBaseException)(WorkSpacesWebServiceException);
var de_AccessDeniedExceptionRes = /* @__PURE__ */ __name(async (parsedOutput, context) => {
  const contents = (0, import_smithy_client.map)({});
  const data = parsedOutput.body;
  const doc = (0, import_smithy_client.take)(data, {
    message: import_smithy_client.expectString
  });
  Object.assign(contents, doc);
  const exception = new AccessDeniedException({
    $metadata: deserializeMetadata(parsedOutput),
    ...contents
  });
  return (0, import_smithy_client.decorateServiceException)(exception, parsedOutput.body);
}, "de_AccessDeniedExceptionRes");
var de_ConflictExceptionRes = /* @__PURE__ */ __name(async (parsedOutput, context) => {
  const contents = (0, import_smithy_client.map)({});
  const data = parsedOutput.body;
  const doc = (0, import_smithy_client.take)(data, {
    message: import_smithy_client.expectString,
    resourceId: import_smithy_client.expectString,
    resourceType: import_smithy_client.expectString
  });
  Object.assign(contents, doc);
  const exception = new ConflictException({
    $metadata: deserializeMetadata(parsedOutput),
    ...contents
  });
  return (0, import_smithy_client.decorateServiceException)(exception, parsedOutput.body);
}, "de_ConflictExceptionRes");
var de_InternalServerExceptionRes = /* @__PURE__ */ __name(async (parsedOutput, context) => {
  const contents = (0, import_smithy_client.map)({
    [_rAS]: [() => void 0 !== parsedOutput.headers[_ra], () => (0, import_smithy_client.strictParseInt32)(parsedOutput.headers[_ra])]
  });
  const data = parsedOutput.body;
  const doc = (0, import_smithy_client.take)(data, {
    message: import_smithy_client.expectString
  });
  Object.assign(contents, doc);
  const exception = new InternalServerException({
    $metadata: deserializeMetadata(parsedOutput),
    ...contents
  });
  return (0, import_smithy_client.decorateServiceException)(exception, parsedOutput.body);
}, "de_InternalServerExceptionRes");
var de_ResourceNotFoundExceptionRes = /* @__PURE__ */ __name(async (parsedOutput, context) => {
  const contents = (0, import_smithy_client.map)({});
  const data = parsedOutput.body;
  const doc = (0, import_smithy_client.take)(data, {
    message: import_smithy_client.expectString,
    resourceId: import_smithy_client.expectString,
    resourceType: import_smithy_client.expectString
  });
  Object.assign(contents, doc);
  const exception = new ResourceNotFoundException({
    $metadata: deserializeMetadata(parsedOutput),
    ...contents
  });
  return (0, import_smithy_client.decorateServiceException)(exception, parsedOutput.body);
}, "de_ResourceNotFoundExceptionRes");
var de_ServiceQuotaExceededExceptionRes = /* @__PURE__ */ __name(async (parsedOutput, context) => {
  const contents = (0, import_smithy_client.map)({});
  const data = parsedOutput.body;
  const doc = (0, import_smithy_client.take)(data, {
    message: import_smithy_client.expectString,
    quotaCode: import_smithy_client.expectString,
    resourceId: import_smithy_client.expectString,
    resourceType: import_smithy_client.expectString,
    serviceCode: import_smithy_client.expectString
  });
  Object.assign(contents, doc);
  const exception = new ServiceQuotaExceededException({
    $metadata: deserializeMetadata(parsedOutput),
    ...contents
  });
  return (0, import_smithy_client.decorateServiceException)(exception, parsedOutput.body);
}, "de_ServiceQuotaExceededExceptionRes");
var de_ThrottlingExceptionRes = /* @__PURE__ */ __name(async (parsedOutput, context) => {
  const contents = (0, import_smithy_client.map)({
    [_rAS]: [() => void 0 !== parsedOutput.headers[_ra], () => (0, import_smithy_client.strictParseInt32)(parsedOutput.headers[_ra])]
  });
  const data = parsedOutput.body;
  const doc = (0, import_smithy_client.take)(data, {
    message: import_smithy_client.expectString,
    quotaCode: import_smithy_client.expectString,
    serviceCode: import_smithy_client.expectString
  });
  Object.assign(contents, doc);
  const exception = new ThrottlingException({
    $metadata: deserializeMetadata(parsedOutput),
    ...contents
  });
  return (0, import_smithy_client.decorateServiceException)(exception, parsedOutput.body);
}, "de_ThrottlingExceptionRes");
var de_TooManyTagsExceptionRes = /* @__PURE__ */ __name(async (parsedOutput, context) => {
  const contents = (0, import_smithy_client.map)({});
  const data = parsedOutput.body;
  const doc = (0, import_smithy_client.take)(data, {
    message: import_smithy_client.expectString,
    resourceName: import_smithy_client.expectString
  });
  Object.assign(contents, doc);
  const exception = new TooManyTagsException({
    $metadata: deserializeMetadata(parsedOutput),
    ...contents
  });
  return (0, import_smithy_client.decorateServiceException)(exception, parsedOutput.body);
}, "de_TooManyTagsExceptionRes");
var de_ValidationExceptionRes = /* @__PURE__ */ __name(async (parsedOutput, context) => {
  const contents = (0, import_smithy_client.map)({});
  const data = parsedOutput.body;
  const doc = (0, import_smithy_client.take)(data, {
    fieldList: import_smithy_client._json,
    message: import_smithy_client.expectString,
    reason: import_smithy_client.expectString
  });
  Object.assign(contents, doc);
  const exception = new ValidationException({
    $metadata: deserializeMetadata(parsedOutput),
    ...contents
  });
  return (0, import_smithy_client.decorateServiceException)(exception, parsedOutput.body);
}, "de_ValidationExceptionRes");
var se_CertificateList = /* @__PURE__ */ __name((input, context) => {
  return input.filter((e) => e != null).map((entry) => {
    return context.base64Encoder(entry);
  });
}, "se_CertificateList");
var de_Certificate = /* @__PURE__ */ __name((output, context) => {
  return (0, import_smithy_client.take)(output, {
    body: context.base64Decoder,
    issuer: import_smithy_client.expectString,
    notValidAfter: (_) => (0, import_smithy_client.expectNonNull)((0, import_smithy_client.parseEpochTimestamp)((0, import_smithy_client.expectNumber)(_))),
    notValidBefore: (_) => (0, import_smithy_client.expectNonNull)((0, import_smithy_client.parseEpochTimestamp)((0, import_smithy_client.expectNumber)(_))),
    subject: import_smithy_client.expectString,
    thumbprint: import_smithy_client.expectString
  });
}, "de_Certificate");
var de_CertificateSummary = /* @__PURE__ */ __name((output, context) => {
  return (0, import_smithy_client.take)(output, {
    issuer: import_smithy_client.expectString,
    notValidAfter: (_) => (0, import_smithy_client.expectNonNull)((0, import_smithy_client.parseEpochTimestamp)((0, import_smithy_client.expectNumber)(_))),
    notValidBefore: (_) => (0, import_smithy_client.expectNonNull)((0, import_smithy_client.parseEpochTimestamp)((0, import_smithy_client.expectNumber)(_))),
    subject: import_smithy_client.expectString,
    thumbprint: import_smithy_client.expectString
  });
}, "de_CertificateSummary");
var de_CertificateSummaryList = /* @__PURE__ */ __name((output, context) => {
  const retVal = (output || []).filter((e) => e != null).map((entry) => {
    return de_CertificateSummary(entry, context);
  });
  return retVal;
}, "de_CertificateSummaryList");
var de_IpAccessSettings = /* @__PURE__ */ __name((output, context) => {
  return (0, import_smithy_client.take)(output, {
    additionalEncryptionContext: import_smithy_client._json,
    associatedPortalArns: import_smithy_client._json,
    creationDate: (_) => (0, import_smithy_client.expectNonNull)((0, import_smithy_client.parseEpochTimestamp)((0, import_smithy_client.expectNumber)(_))),
    customerManagedKey: import_smithy_client.expectString,
    description: import_smithy_client.expectString,
    displayName: import_smithy_client.expectString,
    ipAccessSettingsArn: import_smithy_client.expectString,
    ipRules: import_smithy_client._json
  });
}, "de_IpAccessSettings");
var de_IpAccessSettingsList = /* @__PURE__ */ __name((output, context) => {
  const retVal = (output || []).filter((e) => e != null).map((entry) => {
    return de_IpAccessSettingsSummary(entry, context);
  });
  return retVal;
}, "de_IpAccessSettingsList");
var de_IpAccessSettingsSummary = /* @__PURE__ */ __name((output, context) => {
  return (0, import_smithy_client.take)(output, {
    creationDate: (_) => (0, import_smithy_client.expectNonNull)((0, import_smithy_client.parseEpochTimestamp)((0, import_smithy_client.expectNumber)(_))),
    description: import_smithy_client.expectString,
    displayName: import_smithy_client.expectString,
    ipAccessSettingsArn: import_smithy_client.expectString
  });
}, "de_IpAccessSettingsSummary");
var de_Portal = /* @__PURE__ */ __name((output, context) => {
  return (0, import_smithy_client.take)(output, {
    additionalEncryptionContext: import_smithy_client._json,
    authenticationType: import_smithy_client.expectString,
    browserSettingsArn: import_smithy_client.expectString,
    browserType: import_smithy_client.expectString,
    creationDate: (_) => (0, import_smithy_client.expectNonNull)((0, import_smithy_client.parseEpochTimestamp)((0, import_smithy_client.expectNumber)(_))),
    customerManagedKey: import_smithy_client.expectString,
    displayName: import_smithy_client.expectString,
    instanceType: import_smithy_client.expectString,
    ipAccessSettingsArn: import_smithy_client.expectString,
    maxConcurrentSessions: import_smithy_client.expectInt32,
    networkSettingsArn: import_smithy_client.expectString,
    portalArn: import_smithy_client.expectString,
    portalEndpoint: import_smithy_client.expectString,
    portalStatus: import_smithy_client.expectString,
    rendererType: import_smithy_client.expectString,
    statusReason: import_smithy_client.expectString,
    trustStoreArn: import_smithy_client.expectString,
    userAccessLoggingSettingsArn: import_smithy_client.expectString,
    userSettingsArn: import_smithy_client.expectString
  });
}, "de_Portal");
var de_PortalList = /* @__PURE__ */ __name((output, context) => {
  const retVal = (output || []).filter((e) => e != null).map((entry) => {
    return de_PortalSummary(entry, context);
  });
  return retVal;
}, "de_PortalList");
var de_PortalSummary = /* @__PURE__ */ __name((output, context) => {
  return (0, import_smithy_client.take)(output, {
    authenticationType: import_smithy_client.expectString,
    browserSettingsArn: import_smithy_client.expectString,
    browserType: import_smithy_client.expectString,
    creationDate: (_) => (0, import_smithy_client.expectNonNull)((0, import_smithy_client.parseEpochTimestamp)((0, import_smithy_client.expectNumber)(_))),
    displayName: import_smithy_client.expectString,
    instanceType: import_smithy_client.expectString,
    ipAccessSettingsArn: import_smithy_client.expectString,
    maxConcurrentSessions: import_smithy_client.expectInt32,
    networkSettingsArn: import_smithy_client.expectString,
    portalArn: import_smithy_client.expectString,
    portalEndpoint: import_smithy_client.expectString,
    portalStatus: import_smithy_client.expectString,
    rendererType: import_smithy_client.expectString,
    trustStoreArn: import_smithy_client.expectString,
    userAccessLoggingSettingsArn: import_smithy_client.expectString,
    userSettingsArn: import_smithy_client.expectString
  });
}, "de_PortalSummary");
var deserializeMetadata = /* @__PURE__ */ __name((output) => ({
  httpStatusCode: output.statusCode,
  requestId: output.headers["x-amzn-requestid"] ?? output.headers["x-amzn-request-id"] ?? output.headers["x-amz-request-id"],
  extendedRequestId: output.headers["x-amz-id-2"],
  cfId: output.headers["x-amz-cf-id"]
}), "deserializeMetadata");
var _bSA = "browserSettingsArn";
var _iASA = "ipAccessSettingsArn";
var _mR = "maxResults";
var _nSA = "networkSettingsArn";
var _nT = "nextToken";
var _rAS = "retryAfterSeconds";
var _ra = "retry-after";
var _t = "thumbprint";
var _tK = "tagKeys";
var _tSA = "trustStoreArn";
var _uALSA = "userAccessLoggingSettingsArn";
var _uSA = "userSettingsArn";

// src/commands/AssociateBrowserSettingsCommand.ts
var _AssociateBrowserSettingsCommand = class _AssociateBrowserSettingsCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AWSErmineControlPlaneService", "AssociateBrowserSettings", {}).n("WorkSpacesWebClient", "AssociateBrowserSettingsCommand").f(void 0, void 0).ser(se_AssociateBrowserSettingsCommand).de(de_AssociateBrowserSettingsCommand).build() {
};
__name(_AssociateBrowserSettingsCommand, "AssociateBrowserSettingsCommand");
var AssociateBrowserSettingsCommand = _AssociateBrowserSettingsCommand;

// src/commands/AssociateIpAccessSettingsCommand.ts




var _AssociateIpAccessSettingsCommand = class _AssociateIpAccessSettingsCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AWSErmineControlPlaneService", "AssociateIpAccessSettings", {}).n("WorkSpacesWebClient", "AssociateIpAccessSettingsCommand").f(void 0, void 0).ser(se_AssociateIpAccessSettingsCommand).de(de_AssociateIpAccessSettingsCommand).build() {
};
__name(_AssociateIpAccessSettingsCommand, "AssociateIpAccessSettingsCommand");
var AssociateIpAccessSettingsCommand = _AssociateIpAccessSettingsCommand;

// src/commands/AssociateNetworkSettingsCommand.ts




var _AssociateNetworkSettingsCommand = class _AssociateNetworkSettingsCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AWSErmineControlPlaneService", "AssociateNetworkSettings", {}).n("WorkSpacesWebClient", "AssociateNetworkSettingsCommand").f(void 0, void 0).ser(se_AssociateNetworkSettingsCommand).de(de_AssociateNetworkSettingsCommand).build() {
};
__name(_AssociateNetworkSettingsCommand, "AssociateNetworkSettingsCommand");
var AssociateNetworkSettingsCommand = _AssociateNetworkSettingsCommand;

// src/commands/AssociateTrustStoreCommand.ts




var _AssociateTrustStoreCommand = class _AssociateTrustStoreCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AWSErmineControlPlaneService", "AssociateTrustStore", {}).n("WorkSpacesWebClient", "AssociateTrustStoreCommand").f(void 0, void 0).ser(se_AssociateTrustStoreCommand).de(de_AssociateTrustStoreCommand).build() {
};
__name(_AssociateTrustStoreCommand, "AssociateTrustStoreCommand");
var AssociateTrustStoreCommand = _AssociateTrustStoreCommand;

// src/commands/AssociateUserAccessLoggingSettingsCommand.ts




var _AssociateUserAccessLoggingSettingsCommand = class _AssociateUserAccessLoggingSettingsCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AWSErmineControlPlaneService", "AssociateUserAccessLoggingSettings", {}).n("WorkSpacesWebClient", "AssociateUserAccessLoggingSettingsCommand").f(void 0, void 0).ser(se_AssociateUserAccessLoggingSettingsCommand).de(de_AssociateUserAccessLoggingSettingsCommand).build() {
};
__name(_AssociateUserAccessLoggingSettingsCommand, "AssociateUserAccessLoggingSettingsCommand");
var AssociateUserAccessLoggingSettingsCommand = _AssociateUserAccessLoggingSettingsCommand;

// src/commands/AssociateUserSettingsCommand.ts




var _AssociateUserSettingsCommand = class _AssociateUserSettingsCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AWSErmineControlPlaneService", "AssociateUserSettings", {}).n("WorkSpacesWebClient", "AssociateUserSettingsCommand").f(void 0, void 0).ser(se_AssociateUserSettingsCommand).de(de_AssociateUserSettingsCommand).build() {
};
__name(_AssociateUserSettingsCommand, "AssociateUserSettingsCommand");
var AssociateUserSettingsCommand = _AssociateUserSettingsCommand;

// src/commands/CreateBrowserSettingsCommand.ts




var _CreateBrowserSettingsCommand = class _CreateBrowserSettingsCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AWSErmineControlPlaneService", "CreateBrowserSettings", {}).n("WorkSpacesWebClient", "CreateBrowserSettingsCommand").f(CreateBrowserSettingsRequestFilterSensitiveLog, void 0).ser(se_CreateBrowserSettingsCommand).de(de_CreateBrowserSettingsCommand).build() {
};
__name(_CreateBrowserSettingsCommand, "CreateBrowserSettingsCommand");
var CreateBrowserSettingsCommand = _CreateBrowserSettingsCommand;

// src/commands/CreateIdentityProviderCommand.ts




var _CreateIdentityProviderCommand = class _CreateIdentityProviderCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AWSErmineControlPlaneService", "CreateIdentityProvider", {}).n("WorkSpacesWebClient", "CreateIdentityProviderCommand").f(CreateIdentityProviderRequestFilterSensitiveLog, void 0).ser(se_CreateIdentityProviderCommand).de(de_CreateIdentityProviderCommand).build() {
};
__name(_CreateIdentityProviderCommand, "CreateIdentityProviderCommand");
var CreateIdentityProviderCommand = _CreateIdentityProviderCommand;

// src/commands/CreateIpAccessSettingsCommand.ts




var _CreateIpAccessSettingsCommand = class _CreateIpAccessSettingsCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AWSErmineControlPlaneService", "CreateIpAccessSettings", {}).n("WorkSpacesWebClient", "CreateIpAccessSettingsCommand").f(CreateIpAccessSettingsRequestFilterSensitiveLog, void 0).ser(se_CreateIpAccessSettingsCommand).de(de_CreateIpAccessSettingsCommand).build() {
};
__name(_CreateIpAccessSettingsCommand, "CreateIpAccessSettingsCommand");
var CreateIpAccessSettingsCommand = _CreateIpAccessSettingsCommand;

// src/commands/CreateNetworkSettingsCommand.ts




var _CreateNetworkSettingsCommand = class _CreateNetworkSettingsCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AWSErmineControlPlaneService", "CreateNetworkSettings", {}).n("WorkSpacesWebClient", "CreateNetworkSettingsCommand").f(CreateNetworkSettingsRequestFilterSensitiveLog, void 0).ser(se_CreateNetworkSettingsCommand).de(de_CreateNetworkSettingsCommand).build() {
};
__name(_CreateNetworkSettingsCommand, "CreateNetworkSettingsCommand");
var CreateNetworkSettingsCommand = _CreateNetworkSettingsCommand;

// src/commands/CreatePortalCommand.ts




var _CreatePortalCommand = class _CreatePortalCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AWSErmineControlPlaneService", "CreatePortal", {}).n("WorkSpacesWebClient", "CreatePortalCommand").f(CreatePortalRequestFilterSensitiveLog, void 0).ser(se_CreatePortalCommand).de(de_CreatePortalCommand).build() {
};
__name(_CreatePortalCommand, "CreatePortalCommand");
var CreatePortalCommand = _CreatePortalCommand;

// src/commands/CreateTrustStoreCommand.ts




var _CreateTrustStoreCommand = class _CreateTrustStoreCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AWSErmineControlPlaneService", "CreateTrustStore", {}).n("WorkSpacesWebClient", "CreateTrustStoreCommand").f(CreateTrustStoreRequestFilterSensitiveLog, void 0).ser(se_CreateTrustStoreCommand).de(de_CreateTrustStoreCommand).build() {
};
__name(_CreateTrustStoreCommand, "CreateTrustStoreCommand");
var CreateTrustStoreCommand = _CreateTrustStoreCommand;

// src/commands/CreateUserAccessLoggingSettingsCommand.ts




var _CreateUserAccessLoggingSettingsCommand = class _CreateUserAccessLoggingSettingsCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AWSErmineControlPlaneService", "CreateUserAccessLoggingSettings", {}).n("WorkSpacesWebClient", "CreateUserAccessLoggingSettingsCommand").f(CreateUserAccessLoggingSettingsRequestFilterSensitiveLog, void 0).ser(se_CreateUserAccessLoggingSettingsCommand).de(de_CreateUserAccessLoggingSettingsCommand).build() {
};
__name(_CreateUserAccessLoggingSettingsCommand, "CreateUserAccessLoggingSettingsCommand");
var CreateUserAccessLoggingSettingsCommand = _CreateUserAccessLoggingSettingsCommand;

// src/commands/CreateUserSettingsCommand.ts




var _CreateUserSettingsCommand = class _CreateUserSettingsCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AWSErmineControlPlaneService", "CreateUserSettings", {}).n("WorkSpacesWebClient", "CreateUserSettingsCommand").f(CreateUserSettingsRequestFilterSensitiveLog, void 0).ser(se_CreateUserSettingsCommand).de(de_CreateUserSettingsCommand).build() {
};
__name(_CreateUserSettingsCommand, "CreateUserSettingsCommand");
var CreateUserSettingsCommand = _CreateUserSettingsCommand;

// src/commands/DeleteBrowserSettingsCommand.ts




var _DeleteBrowserSettingsCommand = class _DeleteBrowserSettingsCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AWSErmineControlPlaneService", "DeleteBrowserSettings", {}).n("WorkSpacesWebClient", "DeleteBrowserSettingsCommand").f(void 0, void 0).ser(se_DeleteBrowserSettingsCommand).de(de_DeleteBrowserSettingsCommand).build() {
};
__name(_DeleteBrowserSettingsCommand, "DeleteBrowserSettingsCommand");
var DeleteBrowserSettingsCommand = _DeleteBrowserSettingsCommand;

// src/commands/DeleteIdentityProviderCommand.ts




var _DeleteIdentityProviderCommand = class _DeleteIdentityProviderCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AWSErmineControlPlaneService", "DeleteIdentityProvider", {}).n("WorkSpacesWebClient", "DeleteIdentityProviderCommand").f(void 0, void 0).ser(se_DeleteIdentityProviderCommand).de(de_DeleteIdentityProviderCommand).build() {
};
__name(_DeleteIdentityProviderCommand, "DeleteIdentityProviderCommand");
var DeleteIdentityProviderCommand = _DeleteIdentityProviderCommand;

// src/commands/DeleteIpAccessSettingsCommand.ts




var _DeleteIpAccessSettingsCommand = class _DeleteIpAccessSettingsCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AWSErmineControlPlaneService", "DeleteIpAccessSettings", {}).n("WorkSpacesWebClient", "DeleteIpAccessSettingsCommand").f(void 0, void 0).ser(se_DeleteIpAccessSettingsCommand).de(de_DeleteIpAccessSettingsCommand).build() {
};
__name(_DeleteIpAccessSettingsCommand, "DeleteIpAccessSettingsCommand");
var DeleteIpAccessSettingsCommand = _DeleteIpAccessSettingsCommand;

// src/commands/DeleteNetworkSettingsCommand.ts




var _DeleteNetworkSettingsCommand = class _DeleteNetworkSettingsCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AWSErmineControlPlaneService", "DeleteNetworkSettings", {}).n("WorkSpacesWebClient", "DeleteNetworkSettingsCommand").f(void 0, void 0).ser(se_DeleteNetworkSettingsCommand).de(de_DeleteNetworkSettingsCommand).build() {
};
__name(_DeleteNetworkSettingsCommand, "DeleteNetworkSettingsCommand");
var DeleteNetworkSettingsCommand = _DeleteNetworkSettingsCommand;

// src/commands/DeletePortalCommand.ts




var _DeletePortalCommand = class _DeletePortalCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AWSErmineControlPlaneService", "DeletePortal", {}).n("WorkSpacesWebClient", "DeletePortalCommand").f(void 0, void 0).ser(se_DeletePortalCommand).de(de_DeletePortalCommand).build() {
};
__name(_DeletePortalCommand, "DeletePortalCommand");
var DeletePortalCommand = _DeletePortalCommand;

// src/commands/DeleteTrustStoreCommand.ts




var _DeleteTrustStoreCommand = class _DeleteTrustStoreCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AWSErmineControlPlaneService", "DeleteTrustStore", {}).n("WorkSpacesWebClient", "DeleteTrustStoreCommand").f(void 0, void 0).ser(se_DeleteTrustStoreCommand).de(de_DeleteTrustStoreCommand).build() {
};
__name(_DeleteTrustStoreCommand, "DeleteTrustStoreCommand");
var DeleteTrustStoreCommand = _DeleteTrustStoreCommand;

// src/commands/DeleteUserAccessLoggingSettingsCommand.ts




var _DeleteUserAccessLoggingSettingsCommand = class _DeleteUserAccessLoggingSettingsCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AWSErmineControlPlaneService", "DeleteUserAccessLoggingSettings", {}).n("WorkSpacesWebClient", "DeleteUserAccessLoggingSettingsCommand").f(void 0, void 0).ser(se_DeleteUserAccessLoggingSettingsCommand).de(de_DeleteUserAccessLoggingSettingsCommand).build() {
};
__name(_DeleteUserAccessLoggingSettingsCommand, "DeleteUserAccessLoggingSettingsCommand");
var DeleteUserAccessLoggingSettingsCommand = _DeleteUserAccessLoggingSettingsCommand;

// src/commands/DeleteUserSettingsCommand.ts




var _DeleteUserSettingsCommand = class _DeleteUserSettingsCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AWSErmineControlPlaneService", "DeleteUserSettings", {}).n("WorkSpacesWebClient", "DeleteUserSettingsCommand").f(void 0, void 0).ser(se_DeleteUserSettingsCommand).de(de_DeleteUserSettingsCommand).build() {
};
__name(_DeleteUserSettingsCommand, "DeleteUserSettingsCommand");
var DeleteUserSettingsCommand = _DeleteUserSettingsCommand;

// src/commands/DisassociateBrowserSettingsCommand.ts




var _DisassociateBrowserSettingsCommand = class _DisassociateBrowserSettingsCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AWSErmineControlPlaneService", "DisassociateBrowserSettings", {}).n("WorkSpacesWebClient", "DisassociateBrowserSettingsCommand").f(void 0, void 0).ser(se_DisassociateBrowserSettingsCommand).de(de_DisassociateBrowserSettingsCommand).build() {
};
__name(_DisassociateBrowserSettingsCommand, "DisassociateBrowserSettingsCommand");
var DisassociateBrowserSettingsCommand = _DisassociateBrowserSettingsCommand;

// src/commands/DisassociateIpAccessSettingsCommand.ts




var _DisassociateIpAccessSettingsCommand = class _DisassociateIpAccessSettingsCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AWSErmineControlPlaneService", "DisassociateIpAccessSettings", {}).n("WorkSpacesWebClient", "DisassociateIpAccessSettingsCommand").f(void 0, void 0).ser(se_DisassociateIpAccessSettingsCommand).de(de_DisassociateIpAccessSettingsCommand).build() {
};
__name(_DisassociateIpAccessSettingsCommand, "DisassociateIpAccessSettingsCommand");
var DisassociateIpAccessSettingsCommand = _DisassociateIpAccessSettingsCommand;

// src/commands/DisassociateNetworkSettingsCommand.ts




var _DisassociateNetworkSettingsCommand = class _DisassociateNetworkSettingsCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AWSErmineControlPlaneService", "DisassociateNetworkSettings", {}).n("WorkSpacesWebClient", "DisassociateNetworkSettingsCommand").f(void 0, void 0).ser(se_DisassociateNetworkSettingsCommand).de(de_DisassociateNetworkSettingsCommand).build() {
};
__name(_DisassociateNetworkSettingsCommand, "DisassociateNetworkSettingsCommand");
var DisassociateNetworkSettingsCommand = _DisassociateNetworkSettingsCommand;

// src/commands/DisassociateTrustStoreCommand.ts




var _DisassociateTrustStoreCommand = class _DisassociateTrustStoreCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AWSErmineControlPlaneService", "DisassociateTrustStore", {}).n("WorkSpacesWebClient", "DisassociateTrustStoreCommand").f(void 0, void 0).ser(se_DisassociateTrustStoreCommand).de(de_DisassociateTrustStoreCommand).build() {
};
__name(_DisassociateTrustStoreCommand, "DisassociateTrustStoreCommand");
var DisassociateTrustStoreCommand = _DisassociateTrustStoreCommand;

// src/commands/DisassociateUserAccessLoggingSettingsCommand.ts




var _DisassociateUserAccessLoggingSettingsCommand = class _DisassociateUserAccessLoggingSettingsCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AWSErmineControlPlaneService", "DisassociateUserAccessLoggingSettings", {}).n("WorkSpacesWebClient", "DisassociateUserAccessLoggingSettingsCommand").f(void 0, void 0).ser(se_DisassociateUserAccessLoggingSettingsCommand).de(de_DisassociateUserAccessLoggingSettingsCommand).build() {
};
__name(_DisassociateUserAccessLoggingSettingsCommand, "DisassociateUserAccessLoggingSettingsCommand");
var DisassociateUserAccessLoggingSettingsCommand = _DisassociateUserAccessLoggingSettingsCommand;

// src/commands/DisassociateUserSettingsCommand.ts




var _DisassociateUserSettingsCommand = class _DisassociateUserSettingsCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AWSErmineControlPlaneService", "DisassociateUserSettings", {}).n("WorkSpacesWebClient", "DisassociateUserSettingsCommand").f(void 0, void 0).ser(se_DisassociateUserSettingsCommand).de(de_DisassociateUserSettingsCommand).build() {
};
__name(_DisassociateUserSettingsCommand, "DisassociateUserSettingsCommand");
var DisassociateUserSettingsCommand = _DisassociateUserSettingsCommand;

// src/commands/GetBrowserSettingsCommand.ts




var _GetBrowserSettingsCommand = class _GetBrowserSettingsCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AWSErmineControlPlaneService", "GetBrowserSettings", {}).n("WorkSpacesWebClient", "GetBrowserSettingsCommand").f(void 0, GetBrowserSettingsResponseFilterSensitiveLog).ser(se_GetBrowserSettingsCommand).de(de_GetBrowserSettingsCommand).build() {
};
__name(_GetBrowserSettingsCommand, "GetBrowserSettingsCommand");
var GetBrowserSettingsCommand = _GetBrowserSettingsCommand;

// src/commands/GetIdentityProviderCommand.ts




var _GetIdentityProviderCommand = class _GetIdentityProviderCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AWSErmineControlPlaneService", "GetIdentityProvider", {}).n("WorkSpacesWebClient", "GetIdentityProviderCommand").f(void 0, GetIdentityProviderResponseFilterSensitiveLog).ser(se_GetIdentityProviderCommand).de(de_GetIdentityProviderCommand).build() {
};
__name(_GetIdentityProviderCommand, "GetIdentityProviderCommand");
var GetIdentityProviderCommand = _GetIdentityProviderCommand;

// src/commands/GetIpAccessSettingsCommand.ts




var _GetIpAccessSettingsCommand = class _GetIpAccessSettingsCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AWSErmineControlPlaneService", "GetIpAccessSettings", {}).n("WorkSpacesWebClient", "GetIpAccessSettingsCommand").f(void 0, GetIpAccessSettingsResponseFilterSensitiveLog).ser(se_GetIpAccessSettingsCommand).de(de_GetIpAccessSettingsCommand).build() {
};
__name(_GetIpAccessSettingsCommand, "GetIpAccessSettingsCommand");
var GetIpAccessSettingsCommand = _GetIpAccessSettingsCommand;

// src/commands/GetNetworkSettingsCommand.ts




var _GetNetworkSettingsCommand = class _GetNetworkSettingsCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AWSErmineControlPlaneService", "GetNetworkSettings", {}).n("WorkSpacesWebClient", "GetNetworkSettingsCommand").f(void 0, void 0).ser(se_GetNetworkSettingsCommand).de(de_GetNetworkSettingsCommand).build() {
};
__name(_GetNetworkSettingsCommand, "GetNetworkSettingsCommand");
var GetNetworkSettingsCommand = _GetNetworkSettingsCommand;

// src/commands/GetPortalCommand.ts




var _GetPortalCommand = class _GetPortalCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AWSErmineControlPlaneService", "GetPortal", {}).n("WorkSpacesWebClient", "GetPortalCommand").f(void 0, GetPortalResponseFilterSensitiveLog).ser(se_GetPortalCommand).de(de_GetPortalCommand).build() {
};
__name(_GetPortalCommand, "GetPortalCommand");
var GetPortalCommand = _GetPortalCommand;

// src/commands/GetPortalServiceProviderMetadataCommand.ts




var _GetPortalServiceProviderMetadataCommand = class _GetPortalServiceProviderMetadataCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AWSErmineControlPlaneService", "GetPortalServiceProviderMetadata", {}).n("WorkSpacesWebClient", "GetPortalServiceProviderMetadataCommand").f(void 0, void 0).ser(se_GetPortalServiceProviderMetadataCommand).de(de_GetPortalServiceProviderMetadataCommand).build() {
};
__name(_GetPortalServiceProviderMetadataCommand, "GetPortalServiceProviderMetadataCommand");
var GetPortalServiceProviderMetadataCommand = _GetPortalServiceProviderMetadataCommand;

// src/commands/GetTrustStoreCertificateCommand.ts




var _GetTrustStoreCertificateCommand = class _GetTrustStoreCertificateCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AWSErmineControlPlaneService", "GetTrustStoreCertificate", {}).n("WorkSpacesWebClient", "GetTrustStoreCertificateCommand").f(void 0, void 0).ser(se_GetTrustStoreCertificateCommand).de(de_GetTrustStoreCertificateCommand).build() {
};
__name(_GetTrustStoreCertificateCommand, "GetTrustStoreCertificateCommand");
var GetTrustStoreCertificateCommand = _GetTrustStoreCertificateCommand;

// src/commands/GetTrustStoreCommand.ts




var _GetTrustStoreCommand = class _GetTrustStoreCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AWSErmineControlPlaneService", "GetTrustStore", {}).n("WorkSpacesWebClient", "GetTrustStoreCommand").f(void 0, void 0).ser(se_GetTrustStoreCommand).de(de_GetTrustStoreCommand).build() {
};
__name(_GetTrustStoreCommand, "GetTrustStoreCommand");
var GetTrustStoreCommand = _GetTrustStoreCommand;

// src/commands/GetUserAccessLoggingSettingsCommand.ts




var _GetUserAccessLoggingSettingsCommand = class _GetUserAccessLoggingSettingsCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AWSErmineControlPlaneService", "GetUserAccessLoggingSettings", {}).n("WorkSpacesWebClient", "GetUserAccessLoggingSettingsCommand").f(void 0, void 0).ser(se_GetUserAccessLoggingSettingsCommand).de(de_GetUserAccessLoggingSettingsCommand).build() {
};
__name(_GetUserAccessLoggingSettingsCommand, "GetUserAccessLoggingSettingsCommand");
var GetUserAccessLoggingSettingsCommand = _GetUserAccessLoggingSettingsCommand;

// src/commands/GetUserSettingsCommand.ts




var _GetUserSettingsCommand = class _GetUserSettingsCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AWSErmineControlPlaneService", "GetUserSettings", {}).n("WorkSpacesWebClient", "GetUserSettingsCommand").f(void 0, GetUserSettingsResponseFilterSensitiveLog).ser(se_GetUserSettingsCommand).de(de_GetUserSettingsCommand).build() {
};
__name(_GetUserSettingsCommand, "GetUserSettingsCommand");
var GetUserSettingsCommand = _GetUserSettingsCommand;

// src/commands/ListBrowserSettingsCommand.ts




var _ListBrowserSettingsCommand = class _ListBrowserSettingsCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AWSErmineControlPlaneService", "ListBrowserSettings", {}).n("WorkSpacesWebClient", "ListBrowserSettingsCommand").f(void 0, void 0).ser(se_ListBrowserSettingsCommand).de(de_ListBrowserSettingsCommand).build() {
};
__name(_ListBrowserSettingsCommand, "ListBrowserSettingsCommand");
var ListBrowserSettingsCommand = _ListBrowserSettingsCommand;

// src/commands/ListIdentityProvidersCommand.ts




var _ListIdentityProvidersCommand = class _ListIdentityProvidersCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AWSErmineControlPlaneService", "ListIdentityProviders", {}).n("WorkSpacesWebClient", "ListIdentityProvidersCommand").f(void 0, ListIdentityProvidersResponseFilterSensitiveLog).ser(se_ListIdentityProvidersCommand).de(de_ListIdentityProvidersCommand).build() {
};
__name(_ListIdentityProvidersCommand, "ListIdentityProvidersCommand");
var ListIdentityProvidersCommand = _ListIdentityProvidersCommand;

// src/commands/ListIpAccessSettingsCommand.ts




var _ListIpAccessSettingsCommand = class _ListIpAccessSettingsCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AWSErmineControlPlaneService", "ListIpAccessSettings", {}).n("WorkSpacesWebClient", "ListIpAccessSettingsCommand").f(void 0, ListIpAccessSettingsResponseFilterSensitiveLog).ser(se_ListIpAccessSettingsCommand).de(de_ListIpAccessSettingsCommand).build() {
};
__name(_ListIpAccessSettingsCommand, "ListIpAccessSettingsCommand");
var ListIpAccessSettingsCommand = _ListIpAccessSettingsCommand;

// src/commands/ListNetworkSettingsCommand.ts




var _ListNetworkSettingsCommand = class _ListNetworkSettingsCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AWSErmineControlPlaneService", "ListNetworkSettings", {}).n("WorkSpacesWebClient", "ListNetworkSettingsCommand").f(void 0, void 0).ser(se_ListNetworkSettingsCommand).de(de_ListNetworkSettingsCommand).build() {
};
__name(_ListNetworkSettingsCommand, "ListNetworkSettingsCommand");
var ListNetworkSettingsCommand = _ListNetworkSettingsCommand;

// src/commands/ListPortalsCommand.ts




var _ListPortalsCommand = class _ListPortalsCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AWSErmineControlPlaneService", "ListPortals", {}).n("WorkSpacesWebClient", "ListPortalsCommand").f(void 0, ListPortalsResponseFilterSensitiveLog).ser(se_ListPortalsCommand).de(de_ListPortalsCommand).build() {
};
__name(_ListPortalsCommand, "ListPortalsCommand");
var ListPortalsCommand = _ListPortalsCommand;

// src/commands/ListTagsForResourceCommand.ts




var _ListTagsForResourceCommand = class _ListTagsForResourceCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AWSErmineControlPlaneService", "ListTagsForResource", {}).n("WorkSpacesWebClient", "ListTagsForResourceCommand").f(void 0, ListTagsForResourceResponseFilterSensitiveLog).ser(se_ListTagsForResourceCommand).de(de_ListTagsForResourceCommand).build() {
};
__name(_ListTagsForResourceCommand, "ListTagsForResourceCommand");
var ListTagsForResourceCommand = _ListTagsForResourceCommand;

// src/commands/ListTrustStoreCertificatesCommand.ts




var _ListTrustStoreCertificatesCommand = class _ListTrustStoreCertificatesCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AWSErmineControlPlaneService", "ListTrustStoreCertificates", {}).n("WorkSpacesWebClient", "ListTrustStoreCertificatesCommand").f(void 0, void 0).ser(se_ListTrustStoreCertificatesCommand).de(de_ListTrustStoreCertificatesCommand).build() {
};
__name(_ListTrustStoreCertificatesCommand, "ListTrustStoreCertificatesCommand");
var ListTrustStoreCertificatesCommand = _ListTrustStoreCertificatesCommand;

// src/commands/ListTrustStoresCommand.ts




var _ListTrustStoresCommand = class _ListTrustStoresCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AWSErmineControlPlaneService", "ListTrustStores", {}).n("WorkSpacesWebClient", "ListTrustStoresCommand").f(void 0, void 0).ser(se_ListTrustStoresCommand).de(de_ListTrustStoresCommand).build() {
};
__name(_ListTrustStoresCommand, "ListTrustStoresCommand");
var ListTrustStoresCommand = _ListTrustStoresCommand;

// src/commands/ListUserAccessLoggingSettingsCommand.ts




var _ListUserAccessLoggingSettingsCommand = class _ListUserAccessLoggingSettingsCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AWSErmineControlPlaneService", "ListUserAccessLoggingSettings", {}).n("WorkSpacesWebClient", "ListUserAccessLoggingSettingsCommand").f(void 0, void 0).ser(se_ListUserAccessLoggingSettingsCommand).de(de_ListUserAccessLoggingSettingsCommand).build() {
};
__name(_ListUserAccessLoggingSettingsCommand, "ListUserAccessLoggingSettingsCommand");
var ListUserAccessLoggingSettingsCommand = _ListUserAccessLoggingSettingsCommand;

// src/commands/ListUserSettingsCommand.ts




var _ListUserSettingsCommand = class _ListUserSettingsCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AWSErmineControlPlaneService", "ListUserSettings", {}).n("WorkSpacesWebClient", "ListUserSettingsCommand").f(void 0, ListUserSettingsResponseFilterSensitiveLog).ser(se_ListUserSettingsCommand).de(de_ListUserSettingsCommand).build() {
};
__name(_ListUserSettingsCommand, "ListUserSettingsCommand");
var ListUserSettingsCommand = _ListUserSettingsCommand;

// src/commands/TagResourceCommand.ts




var _TagResourceCommand = class _TagResourceCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AWSErmineControlPlaneService", "TagResource", {}).n("WorkSpacesWebClient", "TagResourceCommand").f(TagResourceRequestFilterSensitiveLog, void 0).ser(se_TagResourceCommand).de(de_TagResourceCommand).build() {
};
__name(_TagResourceCommand, "TagResourceCommand");
var TagResourceCommand = _TagResourceCommand;

// src/commands/UntagResourceCommand.ts




var _UntagResourceCommand = class _UntagResourceCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AWSErmineControlPlaneService", "UntagResource", {}).n("WorkSpacesWebClient", "UntagResourceCommand").f(UntagResourceRequestFilterSensitiveLog, void 0).ser(se_UntagResourceCommand).de(de_UntagResourceCommand).build() {
};
__name(_UntagResourceCommand, "UntagResourceCommand");
var UntagResourceCommand = _UntagResourceCommand;

// src/commands/UpdateBrowserSettingsCommand.ts




var _UpdateBrowserSettingsCommand = class _UpdateBrowserSettingsCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AWSErmineControlPlaneService", "UpdateBrowserSettings", {}).n("WorkSpacesWebClient", "UpdateBrowserSettingsCommand").f(UpdateBrowserSettingsRequestFilterSensitiveLog, UpdateBrowserSettingsResponseFilterSensitiveLog).ser(se_UpdateBrowserSettingsCommand).de(de_UpdateBrowserSettingsCommand).build() {
};
__name(_UpdateBrowserSettingsCommand, "UpdateBrowserSettingsCommand");
var UpdateBrowserSettingsCommand = _UpdateBrowserSettingsCommand;

// src/commands/UpdateIdentityProviderCommand.ts




var _UpdateIdentityProviderCommand = class _UpdateIdentityProviderCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AWSErmineControlPlaneService", "UpdateIdentityProvider", {}).n("WorkSpacesWebClient", "UpdateIdentityProviderCommand").f(UpdateIdentityProviderRequestFilterSensitiveLog, UpdateIdentityProviderResponseFilterSensitiveLog).ser(se_UpdateIdentityProviderCommand).de(de_UpdateIdentityProviderCommand).build() {
};
__name(_UpdateIdentityProviderCommand, "UpdateIdentityProviderCommand");
var UpdateIdentityProviderCommand = _UpdateIdentityProviderCommand;

// src/commands/UpdateIpAccessSettingsCommand.ts




var _UpdateIpAccessSettingsCommand = class _UpdateIpAccessSettingsCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AWSErmineControlPlaneService", "UpdateIpAccessSettings", {}).n("WorkSpacesWebClient", "UpdateIpAccessSettingsCommand").f(UpdateIpAccessSettingsRequestFilterSensitiveLog, UpdateIpAccessSettingsResponseFilterSensitiveLog).ser(se_UpdateIpAccessSettingsCommand).de(de_UpdateIpAccessSettingsCommand).build() {
};
__name(_UpdateIpAccessSettingsCommand, "UpdateIpAccessSettingsCommand");
var UpdateIpAccessSettingsCommand = _UpdateIpAccessSettingsCommand;

// src/commands/UpdateNetworkSettingsCommand.ts




var _UpdateNetworkSettingsCommand = class _UpdateNetworkSettingsCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AWSErmineControlPlaneService", "UpdateNetworkSettings", {}).n("WorkSpacesWebClient", "UpdateNetworkSettingsCommand").f(void 0, void 0).ser(se_UpdateNetworkSettingsCommand).de(de_UpdateNetworkSettingsCommand).build() {
};
__name(_UpdateNetworkSettingsCommand, "UpdateNetworkSettingsCommand");
var UpdateNetworkSettingsCommand = _UpdateNetworkSettingsCommand;

// src/commands/UpdatePortalCommand.ts




var _UpdatePortalCommand = class _UpdatePortalCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AWSErmineControlPlaneService", "UpdatePortal", {}).n("WorkSpacesWebClient", "UpdatePortalCommand").f(UpdatePortalRequestFilterSensitiveLog, UpdatePortalResponseFilterSensitiveLog).ser(se_UpdatePortalCommand).de(de_UpdatePortalCommand).build() {
};
__name(_UpdatePortalCommand, "UpdatePortalCommand");
var UpdatePortalCommand = _UpdatePortalCommand;

// src/commands/UpdateTrustStoreCommand.ts




var _UpdateTrustStoreCommand = class _UpdateTrustStoreCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AWSErmineControlPlaneService", "UpdateTrustStore", {}).n("WorkSpacesWebClient", "UpdateTrustStoreCommand").f(void 0, void 0).ser(se_UpdateTrustStoreCommand).de(de_UpdateTrustStoreCommand).build() {
};
__name(_UpdateTrustStoreCommand, "UpdateTrustStoreCommand");
var UpdateTrustStoreCommand = _UpdateTrustStoreCommand;

// src/commands/UpdateUserAccessLoggingSettingsCommand.ts




var _UpdateUserAccessLoggingSettingsCommand = class _UpdateUserAccessLoggingSettingsCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AWSErmineControlPlaneService", "UpdateUserAccessLoggingSettings", {}).n("WorkSpacesWebClient", "UpdateUserAccessLoggingSettingsCommand").f(void 0, void 0).ser(se_UpdateUserAccessLoggingSettingsCommand).de(de_UpdateUserAccessLoggingSettingsCommand).build() {
};
__name(_UpdateUserAccessLoggingSettingsCommand, "UpdateUserAccessLoggingSettingsCommand");
var UpdateUserAccessLoggingSettingsCommand = _UpdateUserAccessLoggingSettingsCommand;

// src/commands/UpdateUserSettingsCommand.ts




var _UpdateUserSettingsCommand = class _UpdateUserSettingsCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AWSErmineControlPlaneService", "UpdateUserSettings", {}).n("WorkSpacesWebClient", "UpdateUserSettingsCommand").f(UpdateUserSettingsRequestFilterSensitiveLog, UpdateUserSettingsResponseFilterSensitiveLog).ser(se_UpdateUserSettingsCommand).de(de_UpdateUserSettingsCommand).build() {
};
__name(_UpdateUserSettingsCommand, "UpdateUserSettingsCommand");
var UpdateUserSettingsCommand = _UpdateUserSettingsCommand;

// src/WorkSpacesWeb.ts
var commands = {
  AssociateBrowserSettingsCommand,
  AssociateIpAccessSettingsCommand,
  AssociateNetworkSettingsCommand,
  AssociateTrustStoreCommand,
  AssociateUserAccessLoggingSettingsCommand,
  AssociateUserSettingsCommand,
  CreateBrowserSettingsCommand,
  CreateIdentityProviderCommand,
  CreateIpAccessSettingsCommand,
  CreateNetworkSettingsCommand,
  CreatePortalCommand,
  CreateTrustStoreCommand,
  CreateUserAccessLoggingSettingsCommand,
  CreateUserSettingsCommand,
  DeleteBrowserSettingsCommand,
  DeleteIdentityProviderCommand,
  DeleteIpAccessSettingsCommand,
  DeleteNetworkSettingsCommand,
  DeletePortalCommand,
  DeleteTrustStoreCommand,
  DeleteUserAccessLoggingSettingsCommand,
  DeleteUserSettingsCommand,
  DisassociateBrowserSettingsCommand,
  DisassociateIpAccessSettingsCommand,
  DisassociateNetworkSettingsCommand,
  DisassociateTrustStoreCommand,
  DisassociateUserAccessLoggingSettingsCommand,
  DisassociateUserSettingsCommand,
  GetBrowserSettingsCommand,
  GetIdentityProviderCommand,
  GetIpAccessSettingsCommand,
  GetNetworkSettingsCommand,
  GetPortalCommand,
  GetPortalServiceProviderMetadataCommand,
  GetTrustStoreCommand,
  GetTrustStoreCertificateCommand,
  GetUserAccessLoggingSettingsCommand,
  GetUserSettingsCommand,
  ListBrowserSettingsCommand,
  ListIdentityProvidersCommand,
  ListIpAccessSettingsCommand,
  ListNetworkSettingsCommand,
  ListPortalsCommand,
  ListTagsForResourceCommand,
  ListTrustStoreCertificatesCommand,
  ListTrustStoresCommand,
  ListUserAccessLoggingSettingsCommand,
  ListUserSettingsCommand,
  TagResourceCommand,
  UntagResourceCommand,
  UpdateBrowserSettingsCommand,
  UpdateIdentityProviderCommand,
  UpdateIpAccessSettingsCommand,
  UpdateNetworkSettingsCommand,
  UpdatePortalCommand,
  UpdateTrustStoreCommand,
  UpdateUserAccessLoggingSettingsCommand,
  UpdateUserSettingsCommand
};
var _WorkSpacesWeb = class _WorkSpacesWeb extends WorkSpacesWebClient {
};
__name(_WorkSpacesWeb, "WorkSpacesWeb");
var WorkSpacesWeb = _WorkSpacesWeb;
(0, import_smithy_client.createAggregatedClient)(commands, WorkSpacesWeb);

// src/pagination/ListBrowserSettingsPaginator.ts

var paginateListBrowserSettings = (0, import_core.createPaginator)(WorkSpacesWebClient, ListBrowserSettingsCommand, "nextToken", "nextToken", "maxResults");

// src/pagination/ListIdentityProvidersPaginator.ts

var paginateListIdentityProviders = (0, import_core.createPaginator)(WorkSpacesWebClient, ListIdentityProvidersCommand, "nextToken", "nextToken", "maxResults");

// src/pagination/ListIpAccessSettingsPaginator.ts

var paginateListIpAccessSettings = (0, import_core.createPaginator)(WorkSpacesWebClient, ListIpAccessSettingsCommand, "nextToken", "nextToken", "maxResults");

// src/pagination/ListNetworkSettingsPaginator.ts

var paginateListNetworkSettings = (0, import_core.createPaginator)(WorkSpacesWebClient, ListNetworkSettingsCommand, "nextToken", "nextToken", "maxResults");

// src/pagination/ListPortalsPaginator.ts

var paginateListPortals = (0, import_core.createPaginator)(WorkSpacesWebClient, ListPortalsCommand, "nextToken", "nextToken", "maxResults");

// src/pagination/ListTrustStoreCertificatesPaginator.ts

var paginateListTrustStoreCertificates = (0, import_core.createPaginator)(WorkSpacesWebClient, ListTrustStoreCertificatesCommand, "nextToken", "nextToken", "maxResults");

// src/pagination/ListTrustStoresPaginator.ts

var paginateListTrustStores = (0, import_core.createPaginator)(WorkSpacesWebClient, ListTrustStoresCommand, "nextToken", "nextToken", "maxResults");

// src/pagination/ListUserAccessLoggingSettingsPaginator.ts

var paginateListUserAccessLoggingSettings = (0, import_core.createPaginator)(WorkSpacesWebClient, ListUserAccessLoggingSettingsCommand, "nextToken", "nextToken", "maxResults");

// src/pagination/ListUserSettingsPaginator.ts

var paginateListUserSettings = (0, import_core.createPaginator)(WorkSpacesWebClient, ListUserSettingsCommand, "nextToken", "nextToken", "maxResults");
// Annotate the CommonJS export names for ESM import in node:

0 && (module.exports = {
  WorkSpacesWebServiceException,
  __Client,
  WorkSpacesWebClient,
  WorkSpacesWeb,
  $Command,
  AssociateBrowserSettingsCommand,
  AssociateIpAccessSettingsCommand,
  AssociateNetworkSettingsCommand,
  AssociateTrustStoreCommand,
  AssociateUserAccessLoggingSettingsCommand,
  AssociateUserSettingsCommand,
  CreateBrowserSettingsCommand,
  CreateIdentityProviderCommand,
  CreateIpAccessSettingsCommand,
  CreateNetworkSettingsCommand,
  CreatePortalCommand,
  CreateTrustStoreCommand,
  CreateUserAccessLoggingSettingsCommand,
  CreateUserSettingsCommand,
  DeleteBrowserSettingsCommand,
  DeleteIdentityProviderCommand,
  DeleteIpAccessSettingsCommand,
  DeleteNetworkSettingsCommand,
  DeletePortalCommand,
  DeleteTrustStoreCommand,
  DeleteUserAccessLoggingSettingsCommand,
  DeleteUserSettingsCommand,
  DisassociateBrowserSettingsCommand,
  DisassociateIpAccessSettingsCommand,
  DisassociateNetworkSettingsCommand,
  DisassociateTrustStoreCommand,
  DisassociateUserAccessLoggingSettingsCommand,
  DisassociateUserSettingsCommand,
  GetBrowserSettingsCommand,
  GetIdentityProviderCommand,
  GetIpAccessSettingsCommand,
  GetNetworkSettingsCommand,
  GetPortalCommand,
  GetPortalServiceProviderMetadataCommand,
  GetTrustStoreCertificateCommand,
  GetTrustStoreCommand,
  GetUserAccessLoggingSettingsCommand,
  GetUserSettingsCommand,
  ListBrowserSettingsCommand,
  ListIdentityProvidersCommand,
  ListIpAccessSettingsCommand,
  ListNetworkSettingsCommand,
  ListPortalsCommand,
  ListTagsForResourceCommand,
  ListTrustStoreCertificatesCommand,
  ListTrustStoresCommand,
  ListUserAccessLoggingSettingsCommand,
  ListUserSettingsCommand,
  TagResourceCommand,
  UntagResourceCommand,
  UpdateBrowserSettingsCommand,
  UpdateIdentityProviderCommand,
  UpdateIpAccessSettingsCommand,
  UpdateNetworkSettingsCommand,
  UpdatePortalCommand,
  UpdateTrustStoreCommand,
  UpdateUserAccessLoggingSettingsCommand,
  UpdateUserSettingsCommand,
  paginateListBrowserSettings,
  paginateListIdentityProviders,
  paginateListIpAccessSettings,
  paginateListNetworkSettings,
  paginateListPortals,
  paginateListTrustStoreCertificates,
  paginateListTrustStores,
  paginateListUserAccessLoggingSettings,
  paginateListUserSettings,
  AccessDeniedException,
  ConflictException,
  InternalServerException,
  ResourceNotFoundException,
  ThrottlingException,
  ValidationExceptionReason,
  ValidationException,
  AuthenticationType,
  ServiceQuotaExceededException,
  IdentityProviderType,
  BrowserType,
  _InstanceType,
  PortalStatus,
  RendererType,
  TooManyTagsException,
  EnabledType,
  TagFilterSensitiveLog,
  CreateBrowserSettingsRequestFilterSensitiveLog,
  BrowserSettingsFilterSensitiveLog,
  GetBrowserSettingsResponseFilterSensitiveLog,
  UpdateBrowserSettingsRequestFilterSensitiveLog,
  UpdateBrowserSettingsResponseFilterSensitiveLog,
  CreateIdentityProviderRequestFilterSensitiveLog,
  IdentityProviderFilterSensitiveLog,
  GetIdentityProviderResponseFilterSensitiveLog,
  IdentityProviderSummaryFilterSensitiveLog,
  ListIdentityProvidersResponseFilterSensitiveLog,
  UpdateIdentityProviderRequestFilterSensitiveLog,
  UpdateIdentityProviderResponseFilterSensitiveLog,
  IpRuleFilterSensitiveLog,
  CreateIpAccessSettingsRequestFilterSensitiveLog,
  IpAccessSettingsFilterSensitiveLog,
  GetIpAccessSettingsResponseFilterSensitiveLog,
  IpAccessSettingsSummaryFilterSensitiveLog,
  ListIpAccessSettingsResponseFilterSensitiveLog,
  UpdateIpAccessSettingsRequestFilterSensitiveLog,
  UpdateIpAccessSettingsResponseFilterSensitiveLog,
  ListTagsForResourceResponseFilterSensitiveLog,
  CreateNetworkSettingsRequestFilterSensitiveLog,
  CreatePortalRequestFilterSensitiveLog,
  PortalFilterSensitiveLog,
  GetPortalResponseFilterSensitiveLog,
  PortalSummaryFilterSensitiveLog,
  ListPortalsResponseFilterSensitiveLog,
  UpdatePortalRequestFilterSensitiveLog,
  UpdatePortalResponseFilterSensitiveLog,
  TagResourceRequestFilterSensitiveLog,
  CreateTrustStoreRequestFilterSensitiveLog,
  UntagResourceRequestFilterSensitiveLog,
  CreateUserAccessLoggingSettingsRequestFilterSensitiveLog,
  CookieSpecificationFilterSensitiveLog,
  CookieSynchronizationConfigurationFilterSensitiveLog,
  CreateUserSettingsRequestFilterSensitiveLog,
  UserSettingsFilterSensitiveLog,
  GetUserSettingsResponseFilterSensitiveLog,
  UserSettingsSummaryFilterSensitiveLog,
  ListUserSettingsResponseFilterSensitiveLog,
  UpdateUserSettingsRequestFilterSensitiveLog,
  UpdateUserSettingsResponseFilterSensitiveLog
});
