// @azure/arm-authorization/dist-esm/src/operations/globalAdministrator.js
/*
 * Copyright (c) Microsoft Corporation.
 * Licensed under the MIT License.
 *
 * Code generated by Microsoft (R) AutoRest Code Generator.
 * Changes may cause incorrect behavior and will be lost if the code is regenerated.
 */
import * as coreClient from "@azure/core-client";
import * as Mappers from "../models/mappers";
import * as Parameters from "../models/parameters";
/** Class containing GlobalAdministrator operations. */
export class GlobalAdministratorImpl {
    /**
     * Initialize a new instance of the class GlobalAdministrator class.
     * @param client Reference to the service client
     */
    constructor(client) {
        this.client = client;
    }
    /**
     * Elevates access for a Global Administrator.
     * @param options The options parameters.
     */
    elevateAccess(options) {
        return this.client.sendOperationRequest({ options }, elevateAccessOperationSpec);
    }
}
// Operation Specifications
const serializer = coreClient.createSerializer(Mappers, /* isXml */ false);
const elevateAccessOperationSpec = {
    path: "/providers/Microsoft.Authorization/elevateAccess",
    httpMethod: "POST",
    responses: {
        200: {},
        default: {
            bodyMapper: Mappers.ErrorResponse
        }
    },
    queryParameters: [Parameters.apiVersion],
    urlParameters: [Parameters.$host],
    headerParameters: [Parameters.accept],
    serializer
};
