// @azure/arm-authorization/dist-esm/src/operations/roleEligibilityScheduleRequests.js
/*
 * Copyright (c) Microsoft Corporation.
 * Licensed under the MIT License.
 *
 * Code generated by Microsoft (R) AutoRest Code Generator.
 * Changes may cause incorrect behavior and will be lost if the code is regenerated.
 */
import { __asyncDelegator, __asyncGenerator, __asyncValues, __await } from "tslib";
import { setContinuationToken } from "../pagingHelper";
import * as coreClient from "@azure/core-client";
import * as Mappers from "../models/mappers";
import * as Parameters from "../models/parameters";
/// <reference lib="esnext.asynciterable" />
/** Class containing RoleEligibilityScheduleRequests operations. */
export class RoleEligibilityScheduleRequestsImpl {
    /**
     * Initialize a new instance of the class RoleEligibilityScheduleRequests class.
     * @param client Reference to the service client
     */
    constructor(client) {
        this.client = client;
    }
    /**
     * Gets role eligibility schedule requests for a scope.
     * @param scope The scope of the role eligibility schedule requests.
     * @param options The options parameters.
     */
    listForScope(scope, options) {
        const iter = this.listForScopePagingAll(scope, options);
        return {
            next() {
                return iter.next();
            },
            [Symbol.asyncIterator]() {
                return this;
            },
            byPage: (settings) => {
                if (settings === null || settings === void 0 ? void 0 : settings.maxPageSize) {
                    throw new Error("maxPageSize is not supported by this operation.");
                }
                return this.listForScopePagingPage(scope, options, settings);
            }
        };
    }
    listForScopePagingPage(scope, options, settings) {
        return __asyncGenerator(this, arguments, function* listForScopePagingPage_1() {
            let result;
            let continuationToken = settings === null || settings === void 0 ? void 0 : settings.continuationToken;
            if (!continuationToken) {
                result = yield __await(this._listForScope(scope, options));
                let page = result.value || [];
                continuationToken = result.nextLink;
                setContinuationToken(page, continuationToken);
                yield yield __await(page);
            }
            while (continuationToken) {
                result = yield __await(this._listForScopeNext(scope, continuationToken, options));
                continuationToken = result.nextLink;
                let page = result.value || [];
                setContinuationToken(page, continuationToken);
                yield yield __await(page);
            }
        });
    }
    listForScopePagingAll(scope, options) {
        return __asyncGenerator(this, arguments, function* listForScopePagingAll_1() {
            var e_1, _a;
            try {
                for (var _b = __asyncValues(this.listForScopePagingPage(scope, options)), _c; _c = yield __await(_b.next()), !_c.done;) {
                    const page = _c.value;
                    yield __await(yield* __asyncDelegator(__asyncValues(page)));
                }
            }
            catch (e_1_1) { e_1 = { error: e_1_1 }; }
            finally {
                try {
                    if (_c && !_c.done && (_a = _b.return)) yield __await(_a.call(_b));
                }
                finally { if (e_1) throw e_1.error; }
            }
        });
    }
    /**
     * Creates a role eligibility schedule request.
     * @param scope The scope of the role eligibility schedule request to create. The scope can be any REST
     *              resource instance. For example, use '/subscriptions/{subscription-id}/' for a subscription,
     *              '/subscriptions/{subscription-id}/resourceGroups/{resource-group-name}' for a resource group, and
     *              '/subscriptions/{subscription-id}/resourceGroups/{resource-group-name}/providers/{resource-provider}/{resource-type}/{resource-name}'
     *              for a resource.
     * @param roleEligibilityScheduleRequestName The name of the role eligibility to create. It can be any
     *                                           valid GUID.
     * @param parameters Parameters for the role eligibility schedule request.
     * @param options The options parameters.
     */
    create(scope, roleEligibilityScheduleRequestName, parameters, options) {
        return this.client.sendOperationRequest({ scope, roleEligibilityScheduleRequestName, parameters, options }, createOperationSpec);
    }
    /**
     * Get the specified role eligibility schedule request.
     * @param scope The scope of the role eligibility schedule request.
     * @param roleEligibilityScheduleRequestName The name (guid) of the role eligibility schedule request
     *                                           to get.
     * @param options The options parameters.
     */
    get(scope, roleEligibilityScheduleRequestName, options) {
        return this.client.sendOperationRequest({ scope, roleEligibilityScheduleRequestName, options }, getOperationSpec);
    }
    /**
     * Gets role eligibility schedule requests for a scope.
     * @param scope The scope of the role eligibility schedule requests.
     * @param options The options parameters.
     */
    _listForScope(scope, options) {
        return this.client.sendOperationRequest({ scope, options }, listForScopeOperationSpec);
    }
    /**
     * Cancels a pending role eligibility schedule request.
     * @param scope The scope of the role eligibility request to cancel.
     * @param roleEligibilityScheduleRequestName The name of the role eligibility request to cancel.
     * @param options The options parameters.
     */
    cancel(scope, roleEligibilityScheduleRequestName, options) {
        return this.client.sendOperationRequest({ scope, roleEligibilityScheduleRequestName, options }, cancelOperationSpec);
    }
    /**
     * Validates a new role eligibility schedule request.
     * @param scope The scope of the role eligibility request to validate.
     * @param roleEligibilityScheduleRequestName The name of the role eligibility request to validate.
     * @param parameters Parameters for the role eligibility schedule request.
     * @param options The options parameters.
     */
    validate(scope, roleEligibilityScheduleRequestName, parameters, options) {
        return this.client.sendOperationRequest({ scope, roleEligibilityScheduleRequestName, parameters, options }, validateOperationSpec);
    }
    /**
     * ListForScopeNext
     * @param scope The scope of the role eligibility schedule requests.
     * @param nextLink The nextLink from the previous successful call to the ListForScope method.
     * @param options The options parameters.
     */
    _listForScopeNext(scope, nextLink, options) {
        return this.client.sendOperationRequest({ scope, nextLink, options }, listForScopeNextOperationSpec);
    }
}
// Operation Specifications
const serializer = coreClient.createSerializer(Mappers, /* isXml */ false);
const createOperationSpec = {
    path: "/{scope}/providers/Microsoft.Authorization/roleEligibilityScheduleRequests/{roleEligibilityScheduleRequestName}",
    httpMethod: "PUT",
    responses: {
        201: {
            bodyMapper: Mappers.RoleEligibilityScheduleRequest
        },
        default: {
            bodyMapper: Mappers.CloudError
        }
    },
    requestBody: Parameters.parameters2,
    queryParameters: [Parameters.apiVersion2],
    urlParameters: [
        Parameters.$host,
        Parameters.scope,
        Parameters.roleEligibilityScheduleRequestName
    ],
    headerParameters: [Parameters.accept, Parameters.contentType],
    mediaType: "json",
    serializer
};
const getOperationSpec = {
    path: "/{scope}/providers/Microsoft.Authorization/roleEligibilityScheduleRequests/{roleEligibilityScheduleRequestName}",
    httpMethod: "GET",
    responses: {
        200: {
            bodyMapper: Mappers.RoleEligibilityScheduleRequest
        },
        default: {
            bodyMapper: Mappers.CloudError
        }
    },
    queryParameters: [Parameters.apiVersion2],
    urlParameters: [
        Parameters.$host,
        Parameters.scope,
        Parameters.roleEligibilityScheduleRequestName
    ],
    headerParameters: [Parameters.accept],
    serializer
};
const listForScopeOperationSpec = {
    path: "/{scope}/providers/Microsoft.Authorization/roleEligibilityScheduleRequests",
    httpMethod: "GET",
    responses: {
        200: {
            bodyMapper: Mappers.RoleEligibilityScheduleRequestListResult
        },
        default: {
            bodyMapper: Mappers.CloudError
        }
    },
    queryParameters: [Parameters.filter, Parameters.apiVersion2],
    urlParameters: [Parameters.$host, Parameters.scope],
    headerParameters: [Parameters.accept],
    serializer
};
const cancelOperationSpec = {
    path: "/{scope}/providers/Microsoft.Authorization/roleEligibilityScheduleRequests/{roleEligibilityScheduleRequestName}/cancel",
    httpMethod: "POST",
    responses: {
        200: {},
        default: {
            bodyMapper: Mappers.CloudError
        }
    },
    queryParameters: [Parameters.apiVersion2],
    urlParameters: [
        Parameters.$host,
        Parameters.scope,
        Parameters.roleEligibilityScheduleRequestName
    ],
    headerParameters: [Parameters.accept],
    serializer
};
const validateOperationSpec = {
    path: "/{scope}/providers/Microsoft.Authorization/roleEligibilityScheduleRequests/{roleEligibilityScheduleRequestName}/validate",
    httpMethod: "POST",
    responses: {
        200: {
            bodyMapper: Mappers.RoleEligibilityScheduleRequest
        },
        default: {
            bodyMapper: Mappers.CloudError
        }
    },
    requestBody: Parameters.parameters2,
    queryParameters: [Parameters.apiVersion2],
    urlParameters: [
        Parameters.$host,
        Parameters.scope,
        Parameters.roleEligibilityScheduleRequestName
    ],
    headerParameters: [Parameters.accept, Parameters.contentType],
    mediaType: "json",
    serializer
};
const listForScopeNextOperationSpec = {
    path: "{nextLink}",
    httpMethod: "GET",
    responses: {
        200: {
            bodyMapper: Mappers.RoleEligibilityScheduleRequestListResult
        },
        default: {
            bodyMapper: Mappers.CloudError
        }
    },
    urlParameters: [Parameters.$host, Parameters.nextLink, Parameters.scope],
    headerParameters: [Parameters.accept],
    serializer
};
