// @azure/arm-databox/dist-esm/src/models/index.js
/*
 * Copyright (c) Microsoft Corporation.
 * Licensed under the MIT License.
 *
 * Code generated by Microsoft (R) AutoRest Code Generator.
 * Changes may cause incorrect behavior and will be lost if the code is regenerated.
 */
/** Known values of {@link StageName} that the service accepts. */
export var KnownStageName;
(function (KnownStageName) {
    /** An order has been created. */
    KnownStageName["DeviceOrdered"] = "DeviceOrdered";
    /** A device has been prepared for the order. */
    KnownStageName["DevicePrepared"] = "DevicePrepared";
    /** Device has been dispatched to the user of the order. */
    KnownStageName["Dispatched"] = "Dispatched";
    /** Device has been delivered to the user of the order. */
    KnownStageName["Delivered"] = "Delivered";
    /** Device has been picked up from user and in transit to Azure datacenter. */
    KnownStageName["PickedUp"] = "PickedUp";
    /** Device has been received at Azure datacenter from the user. */
    KnownStageName["AtAzureDC"] = "AtAzureDC";
    /** Data copy from the device at Azure datacenter. */
    KnownStageName["DataCopy"] = "DataCopy";
    /** Order has completed. */
    KnownStageName["Completed"] = "Completed";
    /** Order has completed with errors. */
    KnownStageName["CompletedWithErrors"] = "CompletedWithErrors";
    /** Order has been cancelled. */
    KnownStageName["Cancelled"] = "Cancelled";
    /** Order has failed due to issue reported by user. */
    KnownStageName["FailedIssueReportedAtCustomer"] = "Failed_IssueReportedAtCustomer";
    /** Order has failed due to issue detected at Azure datacenter. */
    KnownStageName["FailedIssueDetectedAtAzureDC"] = "Failed_IssueDetectedAtAzureDC";
    /** Order has been aborted. */
    KnownStageName["Aborted"] = "Aborted";
    /** Order has completed with warnings. */
    KnownStageName["CompletedWithWarnings"] = "CompletedWithWarnings";
    /** Device is ready to be handed to customer from Azure DC. */
    KnownStageName["ReadyToDispatchFromAzureDC"] = "ReadyToDispatchFromAzureDC";
    /** Device can be dropped off at Azure DC. */
    KnownStageName["ReadyToReceiveAtAzureDC"] = "ReadyToReceiveAtAzureDC";
    /** Job created by the customer. */
    KnownStageName["Created"] = "Created";
    /** User shipped the device to AzureDC. */
    KnownStageName["ShippedToAzureDC"] = "ShippedToAzureDC";
    /** Awaiting shipment details of device from customer. */
    KnownStageName["AwaitingShipmentDetails"] = "AwaitingShipmentDetails";
    /** Preparing the device to ship to customer. */
    KnownStageName["PreparingToShipFromAzureDC"] = "PreparingToShipFromAzureDC";
    /** Shipped the device to customer. */
    KnownStageName["ShippedToCustomer"] = "ShippedToCustomer";
})(KnownStageName || (KnownStageName = {}));
/** Known values of {@link NotificationStageName} that the service accepts. */
export var KnownNotificationStageName;
(function (KnownNotificationStageName) {
    /** Notification at device prepared stage. */
    KnownNotificationStageName["DevicePrepared"] = "DevicePrepared";
    /** Notification at device dispatched stage. */
    KnownNotificationStageName["Dispatched"] = "Dispatched";
    /** Notification at device delivered stage. */
    KnownNotificationStageName["Delivered"] = "Delivered";
    /** Notification at device picked up from user stage. */
    KnownNotificationStageName["PickedUp"] = "PickedUp";
    /** Notification at device received at Azure datacenter stage. */
    KnownNotificationStageName["AtAzureDC"] = "AtAzureDC";
    /** Notification at data copy started stage. */
    KnownNotificationStageName["DataCopy"] = "DataCopy";
    /** Notification at job created stage. */
    KnownNotificationStageName["Created"] = "Created";
    /** Notification at shipped devices to customer stage. */
    KnownNotificationStageName["ShippedToCustomer"] = "ShippedToCustomer";
})(KnownNotificationStageName || (KnownNotificationStageName = {}));
/** Known values of {@link DataCenterCode} that the service accepts. */
export var KnownDataCenterCode;
(function (KnownDataCenterCode) {
    /** Invalid */
    KnownDataCenterCode["Invalid"] = "Invalid";
    /** BY2 */
    KnownDataCenterCode["BY2"] = "BY2";
    /** BY1 */
    KnownDataCenterCode["BY1"] = "BY1";
    /** ORK70 */
    KnownDataCenterCode["ORK70"] = "ORK70";
    /** AM2 */
    KnownDataCenterCode["AM2"] = "AM2";
    /** AMS20 */
    KnownDataCenterCode["AMS20"] = "AMS20";
    /** BY21 */
    KnownDataCenterCode["BY21"] = "BY21";
    /** BY24 */
    KnownDataCenterCode["BY24"] = "BY24";
    /** MWH01 */
    KnownDataCenterCode["MWH01"] = "MWH01";
    /** AMS06 */
    KnownDataCenterCode["AMS06"] = "AMS06";
    /** SSE90 */
    KnownDataCenterCode["SSE90"] = "SSE90";
    /** SYD03 */
    KnownDataCenterCode["SYD03"] = "SYD03";
    /** SYD23 */
    KnownDataCenterCode["SYD23"] = "SYD23";
    /** CBR20 */
    KnownDataCenterCode["CBR20"] = "CBR20";
    /** YTO20 */
    KnownDataCenterCode["YTO20"] = "YTO20";
    /** CWL20 */
    KnownDataCenterCode["CWL20"] = "CWL20";
    /** LON24 */
    KnownDataCenterCode["LON24"] = "LON24";
    /** BOM01 */
    KnownDataCenterCode["BOM01"] = "BOM01";
    /** BL20 */
    KnownDataCenterCode["BL20"] = "BL20";
    /** BL7 */
    KnownDataCenterCode["BL7"] = "BL7";
    /** SEL20 */
    KnownDataCenterCode["SEL20"] = "SEL20";
    /** TYO01 */
    KnownDataCenterCode["TYO01"] = "TYO01";
    /** BN1 */
    KnownDataCenterCode["BN1"] = "BN1";
    /** SN5 */
    KnownDataCenterCode["SN5"] = "SN5";
    /** CYS04 */
    KnownDataCenterCode["CYS04"] = "CYS04";
    /** TYO22 */
    KnownDataCenterCode["TYO22"] = "TYO22";
    /** YTO21 */
    KnownDataCenterCode["YTO21"] = "YTO21";
    /** YQB20 */
    KnownDataCenterCode["YQB20"] = "YQB20";
    /** FRA22 */
    KnownDataCenterCode["FRA22"] = "FRA22";
    /** MAA01 */
    KnownDataCenterCode["MAA01"] = "MAA01";
    /** CPQ02 */
    KnownDataCenterCode["CPQ02"] = "CPQ02";
    /** CPQ20 */
    KnownDataCenterCode["CPQ20"] = "CPQ20";
    /** SIN20 */
    KnownDataCenterCode["SIN20"] = "SIN20";
    /** HKG20 */
    KnownDataCenterCode["HKG20"] = "HKG20";
    /** SG2 */
    KnownDataCenterCode["SG2"] = "SG2";
    /** MEL23 */
    KnownDataCenterCode["MEL23"] = "MEL23";
    /** SEL21 */
    KnownDataCenterCode["SEL21"] = "SEL21";
    /** OSA20 */
    KnownDataCenterCode["OSA20"] = "OSA20";
    /** SHA03 */
    KnownDataCenterCode["SHA03"] = "SHA03";
    /** BJB */
    KnownDataCenterCode["BJB"] = "BJB";
    /** JNB22 */
    KnownDataCenterCode["JNB22"] = "JNB22";
    /** JNB21 */
    KnownDataCenterCode["JNB21"] = "JNB21";
    /** MNZ21 */
    KnownDataCenterCode["MNZ21"] = "MNZ21";
    /** SN8 */
    KnownDataCenterCode["SN8"] = "SN8";
    /** AUH20 */
    KnownDataCenterCode["AUH20"] = "AUH20";
    /** ZRH20 */
    KnownDataCenterCode["ZRH20"] = "ZRH20";
    /** PUS20 */
    KnownDataCenterCode["PUS20"] = "PUS20";
    /** AdHoc */
    KnownDataCenterCode["AdHoc"] = "AdHoc";
    /** CH1 */
    KnownDataCenterCode["CH1"] = "CH1";
    /** DSM05 */
    KnownDataCenterCode["DSM05"] = "DSM05";
    /** DUB07 */
    KnownDataCenterCode["DUB07"] = "DUB07";
    /** PNQ01 */
    KnownDataCenterCode["PNQ01"] = "PNQ01";
    /** SVG20 */
    KnownDataCenterCode["SVG20"] = "SVG20";
    /** OSA02 */
    KnownDataCenterCode["OSA02"] = "OSA02";
    /** OSA22 */
    KnownDataCenterCode["OSA22"] = "OSA22";
    /** PAR22 */
    KnownDataCenterCode["PAR22"] = "PAR22";
    /** BN7 */
    KnownDataCenterCode["BN7"] = "BN7";
    /** SN6 */
    KnownDataCenterCode["SN6"] = "SN6";
    /** BJS20 */
    KnownDataCenterCode["BJS20"] = "BJS20";
})(KnownDataCenterCode || (KnownDataCenterCode = {}));
/** Known values of {@link CopyStatus} that the service accepts. */
export var KnownCopyStatus;
(function (KnownCopyStatus) {
    /** Data copy hasn't started yet. */
    KnownCopyStatus["NotStarted"] = "NotStarted";
    /** Data copy is in progress. */
    KnownCopyStatus["InProgress"] = "InProgress";
    /** Data copy completed. */
    KnownCopyStatus["Completed"] = "Completed";
    /** Data copy completed with errors. */
    KnownCopyStatus["CompletedWithErrors"] = "CompletedWithErrors";
    /** Data copy failed. No data was copied. */
    KnownCopyStatus["Failed"] = "Failed";
    /** No copy triggered as device was not returned. */
    KnownCopyStatus["NotReturned"] = "NotReturned";
    /** The Device has hit hardware issues. */
    KnownCopyStatus["HardwareError"] = "HardwareError";
    /** Data copy failed. The Device was formatted by user. */
    KnownCopyStatus["DeviceFormatted"] = "DeviceFormatted";
    /** Data copy failed. Device metadata was modified by user. */
    KnownCopyStatus["DeviceMetadataModified"] = "DeviceMetadataModified";
    /** Data copy failed. Storage Account was not accessible during copy. */
    KnownCopyStatus["StorageAccountNotAccessible"] = "StorageAccountNotAccessible";
    /** Data copy failed. The Device data content is not supported. */
    KnownCopyStatus["UnsupportedData"] = "UnsupportedData";
    /** No copy triggered as device was not received. */
    KnownCopyStatus["DriveNotReceived"] = "DriveNotReceived";
    /** No copy triggered as device type is not supported. */
    KnownCopyStatus["UnsupportedDrive"] = "UnsupportedDrive";
    /** Copy failed due to service error. */
    KnownCopyStatus["OtherServiceError"] = "OtherServiceError";
    /** Copy failed due to user error. */
    KnownCopyStatus["OtherUserError"] = "OtherUserError";
    /** Copy failed due to disk detection error. */
    KnownCopyStatus["DriveNotDetected"] = "DriveNotDetected";
    /** Copy failed due to corrupted drive. */
    KnownCopyStatus["DriveCorrupted"] = "DriveCorrupted";
    /** Copy failed due to modified or removed metadata files. */
    KnownCopyStatus["MetadataFilesModifiedOrRemoved"] = "MetadataFilesModifiedOrRemoved";
})(KnownCopyStatus || (KnownCopyStatus = {}));
