// @azure/arm-resourcehealth/dist-esm/src/models/index.js
/*
 * Copyright (c) Microsoft Corporation.
 * Licensed under the MIT License.
 *
 * Code generated by Microsoft (R) AutoRest Code Generator.
 * Changes may cause incorrect behavior and will be lost if the code is regenerated.
 */
/** Known values of {@link AvailabilityStateValues} that the service accepts. */
export var KnownAvailabilityStateValues;
(function (KnownAvailabilityStateValues) {
    /** Available */
    KnownAvailabilityStateValues["Available"] = "Available";
    /** Unavailable */
    KnownAvailabilityStateValues["Unavailable"] = "Unavailable";
    /** Degraded */
    KnownAvailabilityStateValues["Degraded"] = "Degraded";
    /** Unknown */
    KnownAvailabilityStateValues["Unknown"] = "Unknown";
})(KnownAvailabilityStateValues || (KnownAvailabilityStateValues = {}));
/** Known values of {@link ReasonChronicityTypes} that the service accepts. */
export var KnownReasonChronicityTypes;
(function (KnownReasonChronicityTypes) {
    /** Transient */
    KnownReasonChronicityTypes["Transient"] = "Transient";
    /** Persistent */
    KnownReasonChronicityTypes["Persistent"] = "Persistent";
})(KnownReasonChronicityTypes || (KnownReasonChronicityTypes = {}));
/** Known values of {@link Scenario} that the service accepts. */
export var KnownScenario;
(function (KnownScenario) {
    /** Alerts */
    KnownScenario["Alerts"] = "Alerts";
})(KnownScenario || (KnownScenario = {}));
/** Known values of {@link CreatedByType} that the service accepts. */
export var KnownCreatedByType;
(function (KnownCreatedByType) {
    /** User */
    KnownCreatedByType["User"] = "User";
    /** Application */
    KnownCreatedByType["Application"] = "Application";
    /** ManagedIdentity */
    KnownCreatedByType["ManagedIdentity"] = "ManagedIdentity";
    /** Key */
    KnownCreatedByType["Key"] = "Key";
})(KnownCreatedByType || (KnownCreatedByType = {}));
/** Known values of {@link EventTypeValues} that the service accepts. */
export var KnownEventTypeValues;
(function (KnownEventTypeValues) {
    /** ServiceIssue */
    KnownEventTypeValues["ServiceIssue"] = "ServiceIssue";
    /** PlannedMaintenance */
    KnownEventTypeValues["PlannedMaintenance"] = "PlannedMaintenance";
    /** HealthAdvisory */
    KnownEventTypeValues["HealthAdvisory"] = "HealthAdvisory";
    /** RCA */
    KnownEventTypeValues["RCA"] = "RCA";
    /** EmergingIssues */
    KnownEventTypeValues["EmergingIssues"] = "EmergingIssues";
    /** SecurityAdvisory */
    KnownEventTypeValues["SecurityAdvisory"] = "SecurityAdvisory";
})(KnownEventTypeValues || (KnownEventTypeValues = {}));
/** Known values of {@link EventSourceValues} that the service accepts. */
export var KnownEventSourceValues;
(function (KnownEventSourceValues) {
    /** ResourceHealth */
    KnownEventSourceValues["ResourceHealth"] = "ResourceHealth";
    /** ServiceHealth */
    KnownEventSourceValues["ServiceHealth"] = "ServiceHealth";
})(KnownEventSourceValues || (KnownEventSourceValues = {}));
/** Known values of {@link EventStatusValues} that the service accepts. */
export var KnownEventStatusValues;
(function (KnownEventStatusValues) {
    /** Active */
    KnownEventStatusValues["Active"] = "Active";
    /** Resolved */
    KnownEventStatusValues["Resolved"] = "Resolved";
})(KnownEventStatusValues || (KnownEventStatusValues = {}));
/** Known values of {@link LevelValues} that the service accepts. */
export var KnownLevelValues;
(function (KnownLevelValues) {
    /** Critical */
    KnownLevelValues["Critical"] = "Critical";
    /** Warning */
    KnownLevelValues["Warning"] = "Warning";
})(KnownLevelValues || (KnownLevelValues = {}));
/** Known values of {@link EventLevelValues} that the service accepts. */
export var KnownEventLevelValues;
(function (KnownEventLevelValues) {
    /** Critical */
    KnownEventLevelValues["Critical"] = "Critical";
    /** Error */
    KnownEventLevelValues["Error"] = "Error";
    /** Warning */
    KnownEventLevelValues["Warning"] = "Warning";
    /** Informational */
    KnownEventLevelValues["Informational"] = "Informational";
})(KnownEventLevelValues || (KnownEventLevelValues = {}));
/** Known values of {@link LinkTypeValues} that the service accepts. */
export var KnownLinkTypeValues;
(function (KnownLinkTypeValues) {
    /** Button */
    KnownLinkTypeValues["Button"] = "Button";
    /** Hyperlink */
    KnownLinkTypeValues["Hyperlink"] = "Hyperlink";
})(KnownLinkTypeValues || (KnownLinkTypeValues = {}));
/** Known values of {@link SeverityValues} that the service accepts. */
export var KnownSeverityValues;
(function (KnownSeverityValues) {
    /** Information */
    KnownSeverityValues["Information"] = "Information";
    /** Warning */
    KnownSeverityValues["Warning"] = "Warning";
    /** Error */
    KnownSeverityValues["Error"] = "Error";
})(KnownSeverityValues || (KnownSeverityValues = {}));
/** Known values of {@link StageValues} that the service accepts. */
export var KnownStageValues;
(function (KnownStageValues) {
    /** Active */
    KnownStageValues["Active"] = "Active";
    /** Resolve */
    KnownStageValues["Resolve"] = "Resolve";
    /** Archived */
    KnownStageValues["Archived"] = "Archived";
})(KnownStageValues || (KnownStageValues = {}));
/** Known values of {@link IssueNameParameter} that the service accepts. */
export var KnownIssueNameParameter;
(function (KnownIssueNameParameter) {
    /** Default */
    KnownIssueNameParameter["Default"] = "default";
})(KnownIssueNameParameter || (KnownIssueNameParameter = {}));
/** Known values of {@link ReasonTypeValues} that the service accepts. */
export var KnownReasonTypeValues;
(function (KnownReasonTypeValues) {
    /** Unplanned */
    KnownReasonTypeValues["Unplanned"] = "Unplanned";
    /** Planned */
    KnownReasonTypeValues["Planned"] = "Planned";
    /** UserInitiated */
    KnownReasonTypeValues["UserInitiated"] = "UserInitiated";
})(KnownReasonTypeValues || (KnownReasonTypeValues = {}));
