// @azure/arm-mediaservices/dist-esm/test/mediaservice_examples.js
/*
 * Copyright (c) Microsoft Corporation.
 * Licensed under the MIT License.
 *
 * Code generated by Microsoft (R) AutoRest Code Generator.
 * Changes may cause incorrect behavior and will be lost if the code is regenerated.
 */
import { __asyncValues, __awaiter } from "tslib";
import { env, Recorder, isPlaybackMode, } from "@azure-tools/test-recorder";
import { createTestCredential } from "@azure-tools/test-credential";
import { assert } from "chai";
import { AzureMediaServices } from "../src/azureMediaServices";
import { StorageManagementClient } from "@azure/arm-storage";
const replaceableVariables = {
    AZURE_CLIENT_ID: "azure_client_id",
    AZURE_CLIENT_SECRET: "azure_client_secret",
    AZURE_TENANT_ID: "88888888-8888-8888-8888-888888888888",
    SUBSCRIPTION_ID: "azure_subscription_id"
};
const recorderOptions = {
    envSetupForPlayback: replaceableVariables
};
export const testPollingOptions = {
    updateIntervalInMs: isPlaybackMode() ? 0 : undefined,
};
describe("MediaServices test", () => {
    let recorder;
    let subscriptionId;
    let client;
    let location;
    let resourceGroup;
    let mediaName;
    let storageAccountName;
    let storage_client;
    beforeEach(function () {
        return __awaiter(this, void 0, void 0, function* () {
            recorder = new Recorder(this.currentTest);
            yield recorder.start(recorderOptions);
            subscriptionId = env.SUBSCRIPTION_ID || '';
            // This is an example of how the environment variables are used
            const credential = createTestCredential();
            client = new AzureMediaServices(credential, subscriptionId, recorder.configureClientOptions({}));
            storage_client = new StorageManagementClient(credential, subscriptionId, recorder.configureClientOptions({}));
            location = "eastus";
            resourceGroup = "myjstest";
            mediaName = "mymediaxxx";
            storageAccountName = "mystorageaccountxxx111";
        });
    });
    afterEach(function () {
        return __awaiter(this, void 0, void 0, function* () {
            yield recorder.stop();
        });
    });
    it("mediaservices create test", function () {
        return __awaiter(this, void 0, void 0, function* () {
            const storage_res = yield storage_client.storageAccounts.beginCreateAndWait(resourceGroup, storageAccountName, {
                sku: {
                    name: "Standard_GRS",
                },
                kind: "StorageV2",
                location: "westeurope",
                encryption: {
                    services: {
                        file: {
                            keyType: "Account",
                            enabled: true,
                        },
                        blob: {
                            keyType: "Account",
                            enabled: true,
                        },
                    },
                    keySource: "Microsoft.Storage",
                },
                tags: {
                    key1: "value1",
                    key2: "value2",
                }
            }, testPollingOptions);
            const res = yield client.mediaservices.beginCreateOrUpdateAndWait(resourceGroup, mediaName, {
                location: location,
                storageAccounts: [
                    {
                        id: "/subscriptions/" + subscriptionId + "/resourceGroups/" + resourceGroup + "/providers/Microsoft.Storage/storageAccounts/" + storageAccountName,
                        type: "Primary"
                    }
                ]
            }, testPollingOptions);
            assert.equal(res.name, mediaName);
        });
    });
    it("mediaservices get test", function () {
        return __awaiter(this, void 0, void 0, function* () {
            const res = yield client.mediaservices.get(resourceGroup, mediaName);
            assert.equal(res.name, mediaName);
        });
    });
    it("mediaservices list test", function () {
        var e_1, _a;
        return __awaiter(this, void 0, void 0, function* () {
            const resArray = new Array();
            try {
                for (var _b = __asyncValues(client.mediaservices.list(resourceGroup)), _c; _c = yield _b.next(), !_c.done;) {
                    let item = _c.value;
                    resArray.push(item);
                }
            }
            catch (e_1_1) { e_1 = { error: e_1_1 }; }
            finally {
                try {
                    if (_c && !_c.done && (_a = _b.return)) yield _a.call(_b);
                }
                finally { if (e_1) throw e_1.error; }
            }
            assert.equal(resArray.length, 1);
        });
    });
    it("mediaservices delete test", function () {
        var e_2, _a;
        return __awaiter(this, void 0, void 0, function* () {
            const res = yield client.mediaservices.delete(resourceGroup, mediaName);
            const resArray = new Array();
            try {
                for (var _b = __asyncValues(client.mediaservices.list(resourceGroup)), _c; _c = yield _b.next(), !_c.done;) {
                    let item = _c.value;
                    resArray.push(item);
                }
            }
            catch (e_2_1) { e_2 = { error: e_2_1 }; }
            finally {
                try {
                    if (_c && !_c.done && (_a = _b.return)) yield _a.call(_b);
                }
                finally { if (e_2) throw e_2.error; }
            }
            assert.equal(resArray.length, 0);
        });
    });
});
