// @azure/arm-keyvault/dist-esm/src/operations/managedHsmKeys.js
/*
 * Copyright (c) Microsoft Corporation.
 * Licensed under the MIT License.
 *
 * Code generated by Microsoft (R) AutoRest Code Generator.
 * Changes may cause incorrect behavior and will be lost if the code is regenerated.
 */
import { __asyncDelegator, __asyncGenerator, __asyncValues, __await } from "tslib";
import { setContinuationToken } from "../pagingHelper";
import * as coreClient from "@azure/core-client";
import * as Mappers from "../models/mappers";
import * as Parameters from "../models/parameters";
/// <reference lib="esnext.asynciterable" />
/** Class containing ManagedHsmKeys operations. */
export class ManagedHsmKeysImpl {
    /**
     * Initialize a new instance of the class ManagedHsmKeys class.
     * @param client Reference to the service client
     */
    constructor(client) {
        this.client = client;
    }
    /**
     * Lists the keys in the specified managed HSM.
     * @param resourceGroupName The name of the resource group. The name is case insensitive.
     * @param name The name of the Managed HSM Pool within the specified resource group.
     * @param options The options parameters.
     */
    list(resourceGroupName, name, options) {
        const iter = this.listPagingAll(resourceGroupName, name, options);
        return {
            next() {
                return iter.next();
            },
            [Symbol.asyncIterator]() {
                return this;
            },
            byPage: (settings) => {
                if (settings === null || settings === void 0 ? void 0 : settings.maxPageSize) {
                    throw new Error("maxPageSize is not supported by this operation.");
                }
                return this.listPagingPage(resourceGroupName, name, options, settings);
            }
        };
    }
    listPagingPage(resourceGroupName, name, options, settings) {
        return __asyncGenerator(this, arguments, function* listPagingPage_1() {
            let result;
            let continuationToken = settings === null || settings === void 0 ? void 0 : settings.continuationToken;
            if (!continuationToken) {
                result = yield __await(this._list(resourceGroupName, name, options));
                let page = result.value || [];
                continuationToken = result.nextLink;
                setContinuationToken(page, continuationToken);
                yield yield __await(page);
            }
            while (continuationToken) {
                result = yield __await(this._listNext(resourceGroupName, name, continuationToken, options));
                continuationToken = result.nextLink;
                let page = result.value || [];
                setContinuationToken(page, continuationToken);
                yield yield __await(page);
            }
        });
    }
    listPagingAll(resourceGroupName, name, options) {
        return __asyncGenerator(this, arguments, function* listPagingAll_1() {
            var _a, e_1, _b, _c;
            try {
                for (var _d = true, _e = __asyncValues(this.listPagingPage(resourceGroupName, name, options)), _f; _f = yield __await(_e.next()), _a = _f.done, !_a;) {
                    _c = _f.value;
                    _d = false;
                    try {
                        const page = _c;
                        yield __await(yield* __asyncDelegator(__asyncValues(page)));
                    }
                    finally {
                        _d = true;
                    }
                }
            }
            catch (e_1_1) { e_1 = { error: e_1_1 }; }
            finally {
                try {
                    if (!_d && !_a && (_b = _e.return)) yield __await(_b.call(_e));
                }
                finally { if (e_1) throw e_1.error; }
            }
        });
    }
    /**
     * Lists the versions of the specified key in the specified managed HSM.
     * @param resourceGroupName The name of the resource group. The name is case insensitive.
     * @param name The name of the Managed HSM Pool within the specified resource group.
     * @param keyName The name of the key to be created. The value you provide may be copied globally for
     *                the purpose of running the service. The value provided should not include personally identifiable or
     *                sensitive information.
     * @param options The options parameters.
     */
    listVersions(resourceGroupName, name, keyName, options) {
        const iter = this.listVersionsPagingAll(resourceGroupName, name, keyName, options);
        return {
            next() {
                return iter.next();
            },
            [Symbol.asyncIterator]() {
                return this;
            },
            byPage: (settings) => {
                if (settings === null || settings === void 0 ? void 0 : settings.maxPageSize) {
                    throw new Error("maxPageSize is not supported by this operation.");
                }
                return this.listVersionsPagingPage(resourceGroupName, name, keyName, options, settings);
            }
        };
    }
    listVersionsPagingPage(resourceGroupName, name, keyName, options, settings) {
        return __asyncGenerator(this, arguments, function* listVersionsPagingPage_1() {
            let result;
            let continuationToken = settings === null || settings === void 0 ? void 0 : settings.continuationToken;
            if (!continuationToken) {
                result = yield __await(this._listVersions(resourceGroupName, name, keyName, options));
                let page = result.value || [];
                continuationToken = result.nextLink;
                setContinuationToken(page, continuationToken);
                yield yield __await(page);
            }
            while (continuationToken) {
                result = yield __await(this._listVersionsNext(resourceGroupName, name, keyName, continuationToken, options));
                continuationToken = result.nextLink;
                let page = result.value || [];
                setContinuationToken(page, continuationToken);
                yield yield __await(page);
            }
        });
    }
    listVersionsPagingAll(resourceGroupName, name, keyName, options) {
        return __asyncGenerator(this, arguments, function* listVersionsPagingAll_1() {
            var _a, e_2, _b, _c;
            try {
                for (var _d = true, _e = __asyncValues(this.listVersionsPagingPage(resourceGroupName, name, keyName, options)), _f; _f = yield __await(_e.next()), _a = _f.done, !_a;) {
                    _c = _f.value;
                    _d = false;
                    try {
                        const page = _c;
                        yield __await(yield* __asyncDelegator(__asyncValues(page)));
                    }
                    finally {
                        _d = true;
                    }
                }
            }
            catch (e_2_1) { e_2 = { error: e_2_1 }; }
            finally {
                try {
                    if (!_d && !_a && (_b = _e.return)) yield __await(_b.call(_e));
                }
                finally { if (e_2) throw e_2.error; }
            }
        });
    }
    /**
     * Creates the first version of a new key if it does not exist. If it already exists, then the existing
     * key is returned without any write operations being performed. This API does not create subsequent
     * versions, and does not update existing keys.
     * @param resourceGroupName The name of the resource group. The name is case insensitive.
     * @param name The name of the Managed HSM Pool within the specified resource group.
     * @param keyName The name of the key to be created. The value you provide may be copied globally for
     *                the purpose of running the service. The value provided should not include personally identifiable or
     *                sensitive information.
     * @param parameters The parameters used to create the specified key.
     * @param options The options parameters.
     */
    createIfNotExist(resourceGroupName, name, keyName, parameters, options) {
        return this.client.sendOperationRequest({ resourceGroupName, name, keyName, parameters, options }, createIfNotExistOperationSpec);
    }
    /**
     * Gets the current version of the specified key from the specified managed HSM.
     * @param resourceGroupName The name of the resource group. The name is case insensitive.
     * @param name The name of the Managed HSM Pool within the specified resource group.
     * @param keyName The name of the key to be created. The value you provide may be copied globally for
     *                the purpose of running the service. The value provided should not include personally identifiable or
     *                sensitive information.
     * @param options The options parameters.
     */
    get(resourceGroupName, name, keyName, options) {
        return this.client.sendOperationRequest({ resourceGroupName, name, keyName, options }, getOperationSpec);
    }
    /**
     * Lists the keys in the specified managed HSM.
     * @param resourceGroupName The name of the resource group. The name is case insensitive.
     * @param name The name of the Managed HSM Pool within the specified resource group.
     * @param options The options parameters.
     */
    _list(resourceGroupName, name, options) {
        return this.client.sendOperationRequest({ resourceGroupName, name, options }, listOperationSpec);
    }
    /**
     * Gets the specified version of the specified key in the specified managed HSM.
     * @param resourceGroupName The name of the resource group. The name is case insensitive.
     * @param name The name of the Managed HSM Pool within the specified resource group.
     * @param keyName The name of the key to be created. The value you provide may be copied globally for
     *                the purpose of running the service. The value provided should not include personally identifiable or
     *                sensitive information.
     * @param keyVersion The version of the key to be retrieved.
     * @param options The options parameters.
     */
    getVersion(resourceGroupName, name, keyName, keyVersion, options) {
        return this.client.sendOperationRequest({ resourceGroupName, name, keyName, keyVersion, options }, getVersionOperationSpec);
    }
    /**
     * Lists the versions of the specified key in the specified managed HSM.
     * @param resourceGroupName The name of the resource group. The name is case insensitive.
     * @param name The name of the Managed HSM Pool within the specified resource group.
     * @param keyName The name of the key to be created. The value you provide may be copied globally for
     *                the purpose of running the service. The value provided should not include personally identifiable or
     *                sensitive information.
     * @param options The options parameters.
     */
    _listVersions(resourceGroupName, name, keyName, options) {
        return this.client.sendOperationRequest({ resourceGroupName, name, keyName, options }, listVersionsOperationSpec);
    }
    /**
     * ListNext
     * @param resourceGroupName The name of the resource group. The name is case insensitive.
     * @param name The name of the Managed HSM Pool within the specified resource group.
     * @param nextLink The nextLink from the previous successful call to the List method.
     * @param options The options parameters.
     */
    _listNext(resourceGroupName, name, nextLink, options) {
        return this.client.sendOperationRequest({ resourceGroupName, name, nextLink, options }, listNextOperationSpec);
    }
    /**
     * ListVersionsNext
     * @param resourceGroupName The name of the resource group. The name is case insensitive.
     * @param name The name of the Managed HSM Pool within the specified resource group.
     * @param keyName The name of the key to be created. The value you provide may be copied globally for
     *                the purpose of running the service. The value provided should not include personally identifiable or
     *                sensitive information.
     * @param nextLink The nextLink from the previous successful call to the ListVersions method.
     * @param options The options parameters.
     */
    _listVersionsNext(resourceGroupName, name, keyName, nextLink, options) {
        return this.client.sendOperationRequest({ resourceGroupName, name, keyName, nextLink, options }, listVersionsNextOperationSpec);
    }
}
// Operation Specifications
const serializer = coreClient.createSerializer(Mappers, /* isXml */ false);
const createIfNotExistOperationSpec = {
    path: "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.KeyVault/managedHSMs/{name}/keys/{keyName}",
    httpMethod: "PUT",
    responses: {
        200: {
            bodyMapper: Mappers.ManagedHsmKey
        },
        default: {
            bodyMapper: Mappers.CloudError
        }
    },
    requestBody: Parameters.parameters1,
    queryParameters: [Parameters.apiVersion],
    urlParameters: [
        Parameters.$host,
        Parameters.subscriptionId,
        Parameters.keyName,
        Parameters.resourceGroupName1,
        Parameters.name
    ],
    headerParameters: [Parameters.contentType, Parameters.accept],
    mediaType: "json",
    serializer
};
const getOperationSpec = {
    path: "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.KeyVault/managedHSMs/{name}/keys/{keyName}",
    httpMethod: "GET",
    responses: {
        200: {
            bodyMapper: Mappers.ManagedHsmKey
        },
        default: {
            bodyMapper: Mappers.CloudError
        }
    },
    queryParameters: [Parameters.apiVersion],
    urlParameters: [
        Parameters.$host,
        Parameters.subscriptionId,
        Parameters.keyName,
        Parameters.resourceGroupName1,
        Parameters.name
    ],
    headerParameters: [Parameters.accept],
    serializer
};
const listOperationSpec = {
    path: "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.KeyVault/managedHSMs/{name}/keys",
    httpMethod: "GET",
    responses: {
        200: {
            bodyMapper: Mappers.ManagedHsmKeyListResult
        },
        default: {
            bodyMapper: Mappers.CloudError
        }
    },
    queryParameters: [Parameters.apiVersion],
    urlParameters: [
        Parameters.$host,
        Parameters.subscriptionId,
        Parameters.resourceGroupName1,
        Parameters.name
    ],
    headerParameters: [Parameters.accept],
    serializer
};
const getVersionOperationSpec = {
    path: "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.KeyVault/managedHSMs/{name}/keys/{keyName}/versions/{keyVersion}",
    httpMethod: "GET",
    responses: {
        200: {
            bodyMapper: Mappers.ManagedHsmKey
        },
        default: {
            bodyMapper: Mappers.CloudError
        }
    },
    queryParameters: [Parameters.apiVersion],
    urlParameters: [
        Parameters.$host,
        Parameters.subscriptionId,
        Parameters.keyName,
        Parameters.keyVersion,
        Parameters.resourceGroupName1,
        Parameters.name
    ],
    headerParameters: [Parameters.accept],
    serializer
};
const listVersionsOperationSpec = {
    path: "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.KeyVault/managedHSMs/{name}/keys/{keyName}/versions",
    httpMethod: "GET",
    responses: {
        200: {
            bodyMapper: Mappers.ManagedHsmKeyListResult
        },
        default: {
            bodyMapper: Mappers.CloudError
        }
    },
    queryParameters: [Parameters.apiVersion],
    urlParameters: [
        Parameters.$host,
        Parameters.subscriptionId,
        Parameters.keyName,
        Parameters.resourceGroupName1,
        Parameters.name
    ],
    headerParameters: [Parameters.accept],
    serializer
};
const listNextOperationSpec = {
    path: "{nextLink}",
    httpMethod: "GET",
    responses: {
        200: {
            bodyMapper: Mappers.ManagedHsmKeyListResult
        },
        default: {
            bodyMapper: Mappers.CloudError
        }
    },
    urlParameters: [
        Parameters.$host,
        Parameters.subscriptionId,
        Parameters.nextLink,
        Parameters.resourceGroupName1,
        Parameters.name
    ],
    headerParameters: [Parameters.accept],
    serializer
};
const listVersionsNextOperationSpec = {
    path: "{nextLink}",
    httpMethod: "GET",
    responses: {
        200: {
            bodyMapper: Mappers.ManagedHsmKeyListResult
        },
        default: {
            bodyMapper: Mappers.CloudError
        }
    },
    urlParameters: [
        Parameters.$host,
        Parameters.subscriptionId,
        Parameters.keyName,
        Parameters.nextLink,
        Parameters.resourceGroupName1,
        Parameters.name
    ],
    headerParameters: [Parameters.accept],
    serializer
};
