// @azure/arm-confluent/dist/index.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var tslib = require('tslib');
var coreClient = require('@azure/core-client');
var coreRestPipeline = require('@azure/core-rest-pipeline');
var coreLro = require('@azure/core-lro');

function _interopNamespaceDefault(e) {
    var n = Object.create(null);
    if (e) {
        Object.keys(e).forEach(function (k) {
            if (k !== 'default') {
                var d = Object.getOwnPropertyDescriptor(e, k);
                Object.defineProperty(n, k, d.get ? d : {
                    enumerable: true,
                    get: function () { return e[k]; }
                });
            }
        });
    }
    n.default = e;
    return Object.freeze(n);
}

var coreClient__namespace = /*#__PURE__*/_interopNamespaceDefault(coreClient);
var coreRestPipeline__namespace = /*#__PURE__*/_interopNamespaceDefault(coreRestPipeline);

/*
 * Copyright (c) Microsoft Corporation.
 * Licensed under the MIT License.
 *
 * Code generated by Microsoft (R) AutoRest Code Generator.
 * Changes may cause incorrect behavior and will be lost if the code is regenerated.
 */
const pageMap = new WeakMap();
/**
 * Given the last `.value` produced by the `byPage` iterator,
 * returns a continuation token that can be used to begin paging from
 * that point later.
 * @param page An object from accessing `value` on the IteratorResult from a `byPage` iterator.
 * @returns The continuation token that can be passed into byPage() during future calls.
 */
function getContinuationToken(page) {
    var _a;
    if (typeof page !== "object" || page === null) {
        return undefined;
    }
    return (_a = pageMap.get(page)) === null || _a === void 0 ? void 0 : _a.continuationToken;
}
function setContinuationToken(page, continuationToken) {
    var _a;
    if (typeof page !== "object" || page === null || !continuationToken) {
        return;
    }
    const pageInfo = (_a = pageMap.get(page)) !== null && _a !== void 0 ? _a : {};
    pageInfo.continuationToken = continuationToken;
    pageMap.set(page, pageInfo);
}

/*
 * Copyright (c) Microsoft Corporation.
 * Licensed under the MIT License.
 *
 * Code generated by Microsoft (R) AutoRest Code Generator.
 * Changes may cause incorrect behavior and will be lost if the code is regenerated.
 */
/** Known values of {@link CreatedByType} that the service accepts. */
exports.KnownCreatedByType = void 0;
(function (KnownCreatedByType) {
    /** User */
    KnownCreatedByType["User"] = "User";
    /** Application */
    KnownCreatedByType["Application"] = "Application";
    /** ManagedIdentity */
    KnownCreatedByType["ManagedIdentity"] = "ManagedIdentity";
    /** Key */
    KnownCreatedByType["Key"] = "Key";
})(exports.KnownCreatedByType || (exports.KnownCreatedByType = {}));
/** Known values of {@link ProvisionState} that the service accepts. */
exports.KnownProvisionState = void 0;
(function (KnownProvisionState) {
    /** Accepted */
    KnownProvisionState["Accepted"] = "Accepted";
    /** Creating */
    KnownProvisionState["Creating"] = "Creating";
    /** Updating */
    KnownProvisionState["Updating"] = "Updating";
    /** Deleting */
    KnownProvisionState["Deleting"] = "Deleting";
    /** Succeeded */
    KnownProvisionState["Succeeded"] = "Succeeded";
    /** Failed */
    KnownProvisionState["Failed"] = "Failed";
    /** Canceled */
    KnownProvisionState["Canceled"] = "Canceled";
    /** Deleted */
    KnownProvisionState["Deleted"] = "Deleted";
    /** NotSpecified */
    KnownProvisionState["NotSpecified"] = "NotSpecified";
})(exports.KnownProvisionState || (exports.KnownProvisionState = {}));
/** Known values of {@link SaaSOfferStatus} that the service accepts. */
exports.KnownSaaSOfferStatus = void 0;
(function (KnownSaaSOfferStatus) {
    /** Started */
    KnownSaaSOfferStatus["Started"] = "Started";
    /** PendingFulfillmentStart */
    KnownSaaSOfferStatus["PendingFulfillmentStart"] = "PendingFulfillmentStart";
    /** InProgress */
    KnownSaaSOfferStatus["InProgress"] = "InProgress";
    /** Subscribed */
    KnownSaaSOfferStatus["Subscribed"] = "Subscribed";
    /** Suspended */
    KnownSaaSOfferStatus["Suspended"] = "Suspended";
    /** Reinstated */
    KnownSaaSOfferStatus["Reinstated"] = "Reinstated";
    /** Succeeded */
    KnownSaaSOfferStatus["Succeeded"] = "Succeeded";
    /** Failed */
    KnownSaaSOfferStatus["Failed"] = "Failed";
    /** Unsubscribed */
    KnownSaaSOfferStatus["Unsubscribed"] = "Unsubscribed";
    /** Updating */
    KnownSaaSOfferStatus["Updating"] = "Updating";
})(exports.KnownSaaSOfferStatus || (exports.KnownSaaSOfferStatus = {}));

/*
 * Copyright (c) Microsoft Corporation.
 * Licensed under the MIT License.
 *
 * Code generated by Microsoft (R) AutoRest Code Generator.
 * Changes may cause incorrect behavior and will be lost if the code is regenerated.
 */
const ConfluentAgreementResourceListResponse = {
    type: {
        name: "Composite",
        className: "ConfluentAgreementResourceListResponse",
        modelProperties: {
            value: {
                serializedName: "value",
                type: {
                    name: "Sequence",
                    element: {
                        type: {
                            name: "Composite",
                            className: "ConfluentAgreementResource",
                        },
                    },
                },
            },
            nextLink: {
                serializedName: "nextLink",
                type: {
                    name: "String",
                },
            },
        },
    },
};
const ConfluentAgreementResource = {
    type: {
        name: "Composite",
        className: "ConfluentAgreementResource",
        modelProperties: {
            id: {
                serializedName: "id",
                readOnly: true,
                type: {
                    name: "String",
                },
            },
            name: {
                serializedName: "name",
                readOnly: true,
                type: {
                    name: "String",
                },
            },
            type: {
                serializedName: "type",
                readOnly: true,
                type: {
                    name: "String",
                },
            },
            systemData: {
                serializedName: "systemData",
                type: {
                    name: "Composite",
                    className: "SystemData",
                },
            },
            publisher: {
                serializedName: "properties.publisher",
                type: {
                    name: "String",
                },
            },
            product: {
                serializedName: "properties.product",
                type: {
                    name: "String",
                },
            },
            plan: {
                serializedName: "properties.plan",
                type: {
                    name: "String",
                },
            },
            licenseTextLink: {
                serializedName: "properties.licenseTextLink",
                type: {
                    name: "String",
                },
            },
            privacyPolicyLink: {
                serializedName: "properties.privacyPolicyLink",
                type: {
                    name: "String",
                },
            },
            retrieveDatetime: {
                serializedName: "properties.retrieveDatetime",
                type: {
                    name: "DateTime",
                },
            },
            signature: {
                serializedName: "properties.signature",
                type: {
                    name: "String",
                },
            },
            accepted: {
                serializedName: "properties.accepted",
                type: {
                    name: "Boolean",
                },
            },
        },
    },
};
const SystemData = {
    type: {
        name: "Composite",
        className: "SystemData",
        modelProperties: {
            createdBy: {
                serializedName: "createdBy",
                type: {
                    name: "String",
                },
            },
            createdByType: {
                serializedName: "createdByType",
                type: {
                    name: "String",
                },
            },
            createdAt: {
                serializedName: "createdAt",
                type: {
                    name: "DateTime",
                },
            },
            lastModifiedBy: {
                serializedName: "lastModifiedBy",
                type: {
                    name: "String",
                },
            },
            lastModifiedByType: {
                serializedName: "lastModifiedByType",
                type: {
                    name: "String",
                },
            },
            lastModifiedAt: {
                serializedName: "lastModifiedAt",
                type: {
                    name: "DateTime",
                },
            },
        },
    },
};
const ResourceProviderDefaultErrorResponse = {
    type: {
        name: "Composite",
        className: "ResourceProviderDefaultErrorResponse",
        modelProperties: {
            error: {
                serializedName: "error",
                type: {
                    name: "Composite",
                    className: "ErrorResponseBody",
                },
            },
        },
    },
};
const ErrorResponseBody = {
    type: {
        name: "Composite",
        className: "ErrorResponseBody",
        modelProperties: {
            code: {
                serializedName: "code",
                readOnly: true,
                type: {
                    name: "String",
                },
            },
            message: {
                serializedName: "message",
                readOnly: true,
                type: {
                    name: "String",
                },
            },
            target: {
                serializedName: "target",
                readOnly: true,
                type: {
                    name: "String",
                },
            },
            details: {
                serializedName: "details",
                readOnly: true,
                type: {
                    name: "Sequence",
                    element: {
                        type: {
                            name: "Composite",
                            className: "ErrorResponseBody",
                        },
                    },
                },
            },
        },
    },
};
const OperationListResult = {
    type: {
        name: "Composite",
        className: "OperationListResult",
        modelProperties: {
            value: {
                serializedName: "value",
                type: {
                    name: "Sequence",
                    element: {
                        type: {
                            name: "Composite",
                            className: "OperationResult",
                        },
                    },
                },
            },
            nextLink: {
                serializedName: "nextLink",
                type: {
                    name: "String",
                },
            },
        },
    },
};
const OperationResult = {
    type: {
        name: "Composite",
        className: "OperationResult",
        modelProperties: {
            name: {
                serializedName: "name",
                type: {
                    name: "String",
                },
            },
            display: {
                serializedName: "display",
                type: {
                    name: "Composite",
                    className: "OperationDisplay",
                },
            },
            isDataAction: {
                serializedName: "isDataAction",
                type: {
                    name: "Boolean",
                },
            },
        },
    },
};
const OperationDisplay = {
    type: {
        name: "Composite",
        className: "OperationDisplay",
        modelProperties: {
            provider: {
                serializedName: "provider",
                type: {
                    name: "String",
                },
            },
            resource: {
                serializedName: "resource",
                type: {
                    name: "String",
                },
            },
            operation: {
                serializedName: "operation",
                type: {
                    name: "String",
                },
            },
            description: {
                serializedName: "description",
                type: {
                    name: "String",
                },
            },
        },
    },
};
const OrganizationResourceListResult = {
    type: {
        name: "Composite",
        className: "OrganizationResourceListResult",
        modelProperties: {
            value: {
                serializedName: "value",
                type: {
                    name: "Sequence",
                    element: {
                        type: {
                            name: "Composite",
                            className: "OrganizationResource",
                        },
                    },
                },
            },
            nextLink: {
                serializedName: "nextLink",
                type: {
                    name: "String",
                },
            },
        },
    },
};
const OrganizationResource = {
    type: {
        name: "Composite",
        className: "OrganizationResource",
        modelProperties: {
            id: {
                serializedName: "id",
                readOnly: true,
                type: {
                    name: "String",
                },
            },
            name: {
                serializedName: "name",
                readOnly: true,
                type: {
                    name: "String",
                },
            },
            type: {
                serializedName: "type",
                readOnly: true,
                type: {
                    name: "String",
                },
            },
            systemData: {
                serializedName: "systemData",
                type: {
                    name: "Composite",
                    className: "SystemData",
                },
            },
            tags: {
                serializedName: "tags",
                type: {
                    name: "Dictionary",
                    value: { type: { name: "String" } },
                },
            },
            location: {
                serializedName: "location",
                type: {
                    name: "String",
                },
            },
            createdTime: {
                serializedName: "properties.createdTime",
                readOnly: true,
                type: {
                    name: "DateTime",
                },
            },
            provisioningState: {
                serializedName: "properties.provisioningState",
                readOnly: true,
                type: {
                    name: "String",
                },
            },
            organizationId: {
                serializedName: "properties.organizationId",
                readOnly: true,
                type: {
                    name: "String",
                },
            },
            ssoUrl: {
                serializedName: "properties.ssoUrl",
                readOnly: true,
                type: {
                    name: "String",
                },
            },
            offerDetail: {
                serializedName: "properties.offerDetail",
                type: {
                    name: "Composite",
                    className: "OfferDetail",
                },
            },
            userDetail: {
                serializedName: "properties.userDetail",
                type: {
                    name: "Composite",
                    className: "UserDetail",
                },
            },
            linkOrganization: {
                serializedName: "properties.linkOrganization",
                type: {
                    name: "Composite",
                    className: "LinkOrganization",
                },
            },
        },
    },
};
const OfferDetail = {
    type: {
        name: "Composite",
        className: "OfferDetail",
        modelProperties: {
            publisherId: {
                constraints: {
                    MaxLength: 50,
                },
                serializedName: "publisherId",
                required: true,
                type: {
                    name: "String",
                },
            },
            id: {
                constraints: {
                    MaxLength: 50,
                },
                serializedName: "id",
                required: true,
                type: {
                    name: "String",
                },
            },
            planId: {
                constraints: {
                    MaxLength: 200,
                },
                serializedName: "planId",
                required: true,
                type: {
                    name: "String",
                },
            },
            planName: {
                constraints: {
                    MaxLength: 200,
                },
                serializedName: "planName",
                required: true,
                type: {
                    name: "String",
                },
            },
            termUnit: {
                constraints: {
                    MaxLength: 25,
                },
                serializedName: "termUnit",
                required: true,
                type: {
                    name: "String",
                },
            },
            termId: {
                constraints: {
                    MaxLength: 50,
                },
                serializedName: "termId",
                type: {
                    name: "String",
                },
            },
            privateOfferId: {
                constraints: {
                    MaxLength: 255,
                },
                serializedName: "privateOfferId",
                type: {
                    name: "String",
                },
            },
            privateOfferIds: {
                serializedName: "privateOfferIds",
                type: {
                    name: "Sequence",
                    element: {
                        type: {
                            name: "String",
                        },
                    },
                },
            },
            status: {
                serializedName: "status",
                type: {
                    name: "String",
                },
            },
        },
    },
};
const UserDetail = {
    type: {
        name: "Composite",
        className: "UserDetail",
        modelProperties: {
            firstName: {
                constraints: {
                    MaxLength: 50,
                },
                serializedName: "firstName",
                type: {
                    name: "String",
                },
            },
            lastName: {
                constraints: {
                    MaxLength: 50,
                },
                serializedName: "lastName",
                type: {
                    name: "String",
                },
            },
            emailAddress: {
                constraints: {
                    Pattern: new RegExp("^\\S+@\\S+\\.\\S+$"),
                },
                serializedName: "emailAddress",
                required: true,
                type: {
                    name: "String",
                },
            },
            userPrincipalName: {
                serializedName: "userPrincipalName",
                type: {
                    name: "String",
                },
            },
            aadEmail: {
                serializedName: "aadEmail",
                type: {
                    name: "String",
                },
            },
        },
    },
};
const LinkOrganization = {
    type: {
        name: "Composite",
        className: "LinkOrganization",
        modelProperties: {
            token: {
                serializedName: "token",
                required: true,
                type: {
                    name: "String",
                },
            },
        },
    },
};
const OrganizationResourceUpdate = {
    type: {
        name: "Composite",
        className: "OrganizationResourceUpdate",
        modelProperties: {
            tags: {
                serializedName: "tags",
                type: {
                    name: "Dictionary",
                    value: { type: { name: "String" } },
                },
            },
        },
    },
};
const ValidationResponse = {
    type: {
        name: "Composite",
        className: "ValidationResponse",
        modelProperties: {
            info: {
                serializedName: "info",
                type: {
                    name: "Dictionary",
                    value: { type: { name: "String" } },
                },
            },
        },
    },
};
const ListAccessRequestModel = {
    type: {
        name: "Composite",
        className: "ListAccessRequestModel",
        modelProperties: {
            searchFilters: {
                serializedName: "searchFilters",
                type: {
                    name: "Dictionary",
                    value: { type: { name: "String" } },
                },
            },
        },
    },
};
const AccessListUsersSuccessResponse = {
    type: {
        name: "Composite",
        className: "AccessListUsersSuccessResponse",
        modelProperties: {
            kind: {
                serializedName: "kind",
                type: {
                    name: "String",
                },
            },
            metadata: {
                serializedName: "metadata",
                type: {
                    name: "Composite",
                    className: "ConfluentListMetadata",
                },
            },
            data: {
                serializedName: "data",
                type: {
                    name: "Sequence",
                    element: {
                        type: {
                            name: "Composite",
                            className: "UserRecord",
                        },
                    },
                },
            },
        },
    },
};
const ConfluentListMetadata = {
    type: {
        name: "Composite",
        className: "ConfluentListMetadata",
        modelProperties: {
            first: {
                serializedName: "first",
                type: {
                    name: "String",
                },
            },
            last: {
                serializedName: "last",
                type: {
                    name: "String",
                },
            },
            prev: {
                serializedName: "prev",
                type: {
                    name: "String",
                },
            },
            next: {
                serializedName: "next",
                type: {
                    name: "String",
                },
            },
            totalSize: {
                serializedName: "total_size",
                type: {
                    name: "Number",
                },
            },
        },
    },
};
const UserRecord = {
    type: {
        name: "Composite",
        className: "UserRecord",
        modelProperties: {
            kind: {
                serializedName: "kind",
                type: {
                    name: "String",
                },
            },
            id: {
                serializedName: "id",
                type: {
                    name: "String",
                },
            },
            metadata: {
                serializedName: "metadata",
                type: {
                    name: "Composite",
                    className: "MetadataEntity",
                },
            },
            email: {
                serializedName: "email",
                type: {
                    name: "String",
                },
            },
            fullName: {
                serializedName: "full_name",
                type: {
                    name: "String",
                },
            },
            authType: {
                serializedName: "auth_type",
                type: {
                    name: "String",
                },
            },
        },
    },
};
const MetadataEntity = {
    type: {
        name: "Composite",
        className: "MetadataEntity",
        modelProperties: {
            self: {
                serializedName: "self",
                type: {
                    name: "String",
                },
            },
            resourceName: {
                serializedName: "resource_name",
                type: {
                    name: "String",
                },
            },
            createdAt: {
                serializedName: "created_at",
                type: {
                    name: "String",
                },
            },
            updatedAt: {
                serializedName: "updated_at",
                type: {
                    name: "String",
                },
            },
            deletedAt: {
                serializedName: "deleted_at",
                type: {
                    name: "String",
                },
            },
        },
    },
};
const AccessListServiceAccountsSuccessResponse = {
    type: {
        name: "Composite",
        className: "AccessListServiceAccountsSuccessResponse",
        modelProperties: {
            kind: {
                serializedName: "kind",
                type: {
                    name: "String",
                },
            },
            metadata: {
                serializedName: "metadata",
                type: {
                    name: "Composite",
                    className: "ConfluentListMetadata",
                },
            },
            data: {
                serializedName: "data",
                type: {
                    name: "Sequence",
                    element: {
                        type: {
                            name: "Composite",
                            className: "ServiceAccountRecord",
                        },
                    },
                },
            },
        },
    },
};
const ServiceAccountRecord = {
    type: {
        name: "Composite",
        className: "ServiceAccountRecord",
        modelProperties: {
            kind: {
                serializedName: "kind",
                type: {
                    name: "String",
                },
            },
            id: {
                serializedName: "id",
                type: {
                    name: "String",
                },
            },
            metadata: {
                serializedName: "metadata",
                type: {
                    name: "Composite",
                    className: "MetadataEntity",
                },
            },
            displayName: {
                serializedName: "display_name",
                type: {
                    name: "String",
                },
            },
            description: {
                serializedName: "description",
                type: {
                    name: "String",
                },
            },
        },
    },
};
const AccessListInvitationsSuccessResponse = {
    type: {
        name: "Composite",
        className: "AccessListInvitationsSuccessResponse",
        modelProperties: {
            kind: {
                serializedName: "kind",
                type: {
                    name: "String",
                },
            },
            metadata: {
                serializedName: "metadata",
                type: {
                    name: "Composite",
                    className: "ConfluentListMetadata",
                },
            },
            data: {
                serializedName: "data",
                type: {
                    name: "Sequence",
                    element: {
                        type: {
                            name: "Composite",
                            className: "InvitationRecord",
                        },
                    },
                },
            },
        },
    },
};
const InvitationRecord = {
    type: {
        name: "Composite",
        className: "InvitationRecord",
        modelProperties: {
            kind: {
                serializedName: "kind",
                type: {
                    name: "String",
                },
            },
            id: {
                serializedName: "id",
                type: {
                    name: "String",
                },
            },
            metadata: {
                serializedName: "metadata",
                type: {
                    name: "Composite",
                    className: "MetadataEntity",
                },
            },
            email: {
                serializedName: "email",
                type: {
                    name: "String",
                },
            },
            authType: {
                serializedName: "auth_type",
                type: {
                    name: "String",
                },
            },
            status: {
                serializedName: "status",
                type: {
                    name: "String",
                },
            },
            acceptedAt: {
                serializedName: "accepted_at",
                type: {
                    name: "String",
                },
            },
            expiresAt: {
                serializedName: "expires_at",
                type: {
                    name: "String",
                },
            },
        },
    },
};
const AccessInviteUserAccountModel = {
    type: {
        name: "Composite",
        className: "AccessInviteUserAccountModel",
        modelProperties: {
            organizationId: {
                serializedName: "organizationId",
                type: {
                    name: "String",
                },
            },
            email: {
                serializedName: "email",
                type: {
                    name: "String",
                },
            },
            upn: {
                serializedName: "upn",
                type: {
                    name: "String",
                },
            },
            invitedUserDetails: {
                serializedName: "invitedUserDetails",
                type: {
                    name: "Composite",
                    className: "AccessInvitedUserDetails",
                },
            },
        },
    },
};
const AccessInvitedUserDetails = {
    type: {
        name: "Composite",
        className: "AccessInvitedUserDetails",
        modelProperties: {
            invitedEmail: {
                serializedName: "invitedEmail",
                type: {
                    name: "String",
                },
            },
            authType: {
                serializedName: "auth_type",
                type: {
                    name: "String",
                },
            },
        },
    },
};
const AccessListEnvironmentsSuccessResponse = {
    type: {
        name: "Composite",
        className: "AccessListEnvironmentsSuccessResponse",
        modelProperties: {
            kind: {
                serializedName: "kind",
                type: {
                    name: "String",
                },
            },
            metadata: {
                serializedName: "metadata",
                type: {
                    name: "Composite",
                    className: "ConfluentListMetadata",
                },
            },
            data: {
                serializedName: "data",
                type: {
                    name: "Sequence",
                    element: {
                        type: {
                            name: "Composite",
                            className: "EnvironmentRecord",
                        },
                    },
                },
            },
        },
    },
};
const EnvironmentRecord = {
    type: {
        name: "Composite",
        className: "EnvironmentRecord",
        modelProperties: {
            kind: {
                serializedName: "kind",
                type: {
                    name: "String",
                },
            },
            id: {
                serializedName: "id",
                type: {
                    name: "String",
                },
            },
            metadata: {
                serializedName: "metadata",
                type: {
                    name: "Composite",
                    className: "MetadataEntity",
                },
            },
            displayName: {
                serializedName: "display_name",
                type: {
                    name: "String",
                },
            },
        },
    },
};
const AccessListClusterSuccessResponse = {
    type: {
        name: "Composite",
        className: "AccessListClusterSuccessResponse",
        modelProperties: {
            kind: {
                serializedName: "kind",
                type: {
                    name: "String",
                },
            },
            metadata: {
                serializedName: "metadata",
                type: {
                    name: "Composite",
                    className: "ConfluentListMetadata",
                },
            },
            data: {
                serializedName: "data",
                type: {
                    name: "Sequence",
                    element: {
                        type: {
                            name: "Composite",
                            className: "ClusterRecord",
                        },
                    },
                },
            },
        },
    },
};
const ClusterRecord = {
    type: {
        name: "Composite",
        className: "ClusterRecord",
        modelProperties: {
            kind: {
                serializedName: "kind",
                type: {
                    name: "String",
                },
            },
            id: {
                serializedName: "id",
                type: {
                    name: "String",
                },
            },
            metadata: {
                serializedName: "metadata",
                type: {
                    name: "Composite",
                    className: "MetadataEntity",
                },
            },
            displayName: {
                serializedName: "display_name",
                type: {
                    name: "String",
                },
            },
            spec: {
                serializedName: "spec",
                type: {
                    name: "Composite",
                    className: "ClusterSpecEntity",
                },
            },
            status: {
                serializedName: "status",
                type: {
                    name: "Composite",
                    className: "ClusterStatusEntity",
                },
            },
        },
    },
};
const ClusterSpecEntity = {
    type: {
        name: "Composite",
        className: "ClusterSpecEntity",
        modelProperties: {
            displayName: {
                serializedName: "display_name",
                type: {
                    name: "String",
                },
            },
            availability: {
                serializedName: "availability",
                type: {
                    name: "String",
                },
            },
            cloud: {
                serializedName: "cloud",
                type: {
                    name: "String",
                },
            },
            zone: {
                serializedName: "zone",
                type: {
                    name: "String",
                },
            },
            region: {
                serializedName: "region",
                type: {
                    name: "String",
                },
            },
            kafkaBootstrapEndpoint: {
                serializedName: "kafka_bootstrap_endpoint",
                type: {
                    name: "String",
                },
            },
            httpEndpoint: {
                serializedName: "http_endpoint",
                type: {
                    name: "String",
                },
            },
            apiEndpoint: {
                serializedName: "api_endpoint",
                type: {
                    name: "String",
                },
            },
            config: {
                serializedName: "config",
                type: {
                    name: "Composite",
                    className: "ClusterConfigEntity",
                },
            },
            environment: {
                serializedName: "environment",
                type: {
                    name: "Composite",
                    className: "ClusterEnvironmentEntity",
                },
            },
            network: {
                serializedName: "network",
                type: {
                    name: "Composite",
                    className: "ClusterNetworkEntity",
                },
            },
            byok: {
                serializedName: "byok",
                type: {
                    name: "Composite",
                    className: "ClusterByokEntity",
                },
            },
        },
    },
};
const ClusterConfigEntity = {
    type: {
        name: "Composite",
        className: "ClusterConfigEntity",
        modelProperties: {
            kind: {
                serializedName: "kind",
                type: {
                    name: "String",
                },
            },
        },
    },
};
const ClusterEnvironmentEntity = {
    type: {
        name: "Composite",
        className: "ClusterEnvironmentEntity",
        modelProperties: {
            id: {
                serializedName: "id",
                type: {
                    name: "String",
                },
            },
            environment: {
                serializedName: "environment",
                type: {
                    name: "String",
                },
            },
            related: {
                serializedName: "related",
                type: {
                    name: "String",
                },
            },
            resourceName: {
                serializedName: "resource_name",
                type: {
                    name: "String",
                },
            },
        },
    },
};
const ClusterNetworkEntity = {
    type: {
        name: "Composite",
        className: "ClusterNetworkEntity",
        modelProperties: {
            id: {
                serializedName: "id",
                type: {
                    name: "String",
                },
            },
            environment: {
                serializedName: "environment",
                type: {
                    name: "String",
                },
            },
            related: {
                serializedName: "related",
                type: {
                    name: "String",
                },
            },
            resourceName: {
                serializedName: "resource_name",
                type: {
                    name: "String",
                },
            },
        },
    },
};
const ClusterByokEntity = {
    type: {
        name: "Composite",
        className: "ClusterByokEntity",
        modelProperties: {
            id: {
                serializedName: "id",
                type: {
                    name: "String",
                },
            },
            related: {
                serializedName: "related",
                type: {
                    name: "String",
                },
            },
            resourceName: {
                serializedName: "resource_name",
                type: {
                    name: "String",
                },
            },
        },
    },
};
const ClusterStatusEntity = {
    type: {
        name: "Composite",
        className: "ClusterStatusEntity",
        modelProperties: {
            phase: {
                serializedName: "phase",
                type: {
                    name: "String",
                },
            },
            cku: {
                serializedName: "cku",
                type: {
                    name: "Number",
                },
            },
        },
    },
};
const AccessListRoleBindingsSuccessResponse = {
    type: {
        name: "Composite",
        className: "AccessListRoleBindingsSuccessResponse",
        modelProperties: {
            kind: {
                serializedName: "kind",
                type: {
                    name: "String",
                },
            },
            metadata: {
                serializedName: "metadata",
                type: {
                    name: "Composite",
                    className: "ConfluentListMetadata",
                },
            },
            data: {
                serializedName: "data",
                type: {
                    name: "Sequence",
                    element: {
                        type: {
                            name: "Composite",
                            className: "RoleBindingRecord",
                        },
                    },
                },
            },
        },
    },
};
const RoleBindingRecord = {
    type: {
        name: "Composite",
        className: "RoleBindingRecord",
        modelProperties: {
            kind: {
                serializedName: "kind",
                type: {
                    name: "String",
                },
            },
            id: {
                serializedName: "id",
                type: {
                    name: "String",
                },
            },
            metadata: {
                serializedName: "metadata",
                type: {
                    name: "Composite",
                    className: "MetadataEntity",
                },
            },
            principal: {
                serializedName: "principal",
                type: {
                    name: "String",
                },
            },
            roleName: {
                serializedName: "role_name",
                type: {
                    name: "String",
                },
            },
            crnPattern: {
                serializedName: "crn_pattern",
                type: {
                    name: "String",
                },
            },
        },
    },
};
const AccessCreateRoleBindingRequestModel = {
    type: {
        name: "Composite",
        className: "AccessCreateRoleBindingRequestModel",
        modelProperties: {
            principal: {
                serializedName: "principal",
                type: {
                    name: "String",
                },
            },
            roleName: {
                serializedName: "role_name",
                type: {
                    name: "String",
                },
            },
            crnPattern: {
                serializedName: "crn_pattern",
                type: {
                    name: "String",
                },
            },
        },
    },
};
const AccessRoleBindingNameListSuccessResponse = {
    type: {
        name: "Composite",
        className: "AccessRoleBindingNameListSuccessResponse",
        modelProperties: {
            kind: {
                serializedName: "kind",
                type: {
                    name: "String",
                },
            },
            metadata: {
                serializedName: "metadata",
                type: {
                    name: "Composite",
                    className: "ConfluentListMetadata",
                },
            },
            data: {
                serializedName: "data",
                type: {
                    name: "Sequence",
                    element: {
                        type: {
                            name: "String",
                        },
                    },
                },
            },
        },
    },
};
const GetEnvironmentsResponse = {
    type: {
        name: "Composite",
        className: "GetEnvironmentsResponse",
        modelProperties: {
            value: {
                serializedName: "value",
                type: {
                    name: "Sequence",
                    element: {
                        type: {
                            name: "Composite",
                            className: "SCEnvironmentRecord",
                        },
                    },
                },
            },
            nextLink: {
                serializedName: "nextLink",
                type: {
                    name: "String",
                },
            },
        },
    },
};
const SCEnvironmentRecord = {
    type: {
        name: "Composite",
        className: "SCEnvironmentRecord",
        modelProperties: {
            kind: {
                serializedName: "kind",
                type: {
                    name: "String",
                },
            },
            id: {
                serializedName: "id",
                type: {
                    name: "String",
                },
            },
            name: {
                serializedName: "name",
                type: {
                    name: "String",
                },
            },
            metadata: {
                serializedName: "properties.metadata",
                type: {
                    name: "Composite",
                    className: "SCMetadataEntity",
                },
            },
        },
    },
};
const SCMetadataEntity = {
    type: {
        name: "Composite",
        className: "SCMetadataEntity",
        modelProperties: {
            self: {
                serializedName: "self",
                type: {
                    name: "String",
                },
            },
            resourceName: {
                serializedName: "resourceName",
                type: {
                    name: "String",
                },
            },
            createdTimestamp: {
                serializedName: "createdTimestamp",
                type: {
                    name: "String",
                },
            },
            updatedTimestamp: {
                serializedName: "updatedTimestamp",
                type: {
                    name: "String",
                },
            },
            deletedTimestamp: {
                serializedName: "deletedTimestamp",
                type: {
                    name: "String",
                },
            },
        },
    },
};
const ListClustersSuccessResponse = {
    type: {
        name: "Composite",
        className: "ListClustersSuccessResponse",
        modelProperties: {
            value: {
                serializedName: "value",
                type: {
                    name: "Sequence",
                    element: {
                        type: {
                            name: "Composite",
                            className: "SCClusterRecord",
                        },
                    },
                },
            },
            nextLink: {
                serializedName: "nextLink",
                type: {
                    name: "String",
                },
            },
        },
    },
};
const SCClusterRecord = {
    type: {
        name: "Composite",
        className: "SCClusterRecord",
        modelProperties: {
            kind: {
                serializedName: "kind",
                type: {
                    name: "String",
                },
            },
            id: {
                serializedName: "id",
                type: {
                    name: "String",
                },
            },
            name: {
                serializedName: "name",
                type: {
                    name: "String",
                },
            },
            metadata: {
                serializedName: "properties.metadata",
                type: {
                    name: "Composite",
                    className: "SCMetadataEntity",
                },
            },
            spec: {
                serializedName: "properties.spec",
                type: {
                    name: "Composite",
                    className: "SCClusterSpecEntity",
                },
            },
            status: {
                serializedName: "properties.status",
                type: {
                    name: "Composite",
                    className: "ClusterStatusEntity",
                },
            },
        },
    },
};
const SCClusterSpecEntity = {
    type: {
        name: "Composite",
        className: "SCClusterSpecEntity",
        modelProperties: {
            name: {
                serializedName: "name",
                type: {
                    name: "String",
                },
            },
            availability: {
                serializedName: "availability",
                type: {
                    name: "String",
                },
            },
            cloud: {
                serializedName: "cloud",
                type: {
                    name: "String",
                },
            },
            zone: {
                serializedName: "zone",
                type: {
                    name: "String",
                },
            },
            region: {
                serializedName: "region",
                type: {
                    name: "String",
                },
            },
            kafkaBootstrapEndpoint: {
                serializedName: "kafkaBootstrapEndpoint",
                type: {
                    name: "String",
                },
            },
            httpEndpoint: {
                serializedName: "httpEndpoint",
                type: {
                    name: "String",
                },
            },
            apiEndpoint: {
                serializedName: "apiEndpoint",
                type: {
                    name: "String",
                },
            },
            config: {
                serializedName: "config",
                type: {
                    name: "Composite",
                    className: "ClusterConfigEntity",
                },
            },
            environment: {
                serializedName: "environment",
                type: {
                    name: "Composite",
                    className: "SCClusterNetworkEnvironmentEntity",
                },
            },
            network: {
                serializedName: "network",
                type: {
                    name: "Composite",
                    className: "SCClusterNetworkEnvironmentEntity",
                },
            },
            byok: {
                serializedName: "byok",
                type: {
                    name: "Composite",
                    className: "SCClusterByokEntity",
                },
            },
        },
    },
};
const SCClusterNetworkEnvironmentEntity = {
    type: {
        name: "Composite",
        className: "SCClusterNetworkEnvironmentEntity",
        modelProperties: {
            id: {
                serializedName: "id",
                type: {
                    name: "String",
                },
            },
            environment: {
                serializedName: "environment",
                type: {
                    name: "String",
                },
            },
            related: {
                serializedName: "related",
                type: {
                    name: "String",
                },
            },
            resourceName: {
                serializedName: "resourceName",
                type: {
                    name: "String",
                },
            },
        },
    },
};
const SCClusterByokEntity = {
    type: {
        name: "Composite",
        className: "SCClusterByokEntity",
        modelProperties: {
            id: {
                serializedName: "id",
                type: {
                    name: "String",
                },
            },
            related: {
                serializedName: "related",
                type: {
                    name: "String",
                },
            },
            resourceName: {
                serializedName: "resourceName",
                type: {
                    name: "String",
                },
            },
        },
    },
};
const ListSchemaRegistryClustersResponse = {
    type: {
        name: "Composite",
        className: "ListSchemaRegistryClustersResponse",
        modelProperties: {
            value: {
                serializedName: "value",
                type: {
                    name: "Sequence",
                    element: {
                        type: {
                            name: "Composite",
                            className: "SchemaRegistryClusterRecord",
                        },
                    },
                },
            },
            nextLink: {
                serializedName: "nextLink",
                type: {
                    name: "String",
                },
            },
        },
    },
};
const SchemaRegistryClusterRecord = {
    type: {
        name: "Composite",
        className: "SchemaRegistryClusterRecord",
        modelProperties: {
            kind: {
                serializedName: "kind",
                type: {
                    name: "String",
                },
            },
            id: {
                serializedName: "id",
                type: {
                    name: "String",
                },
            },
            metadata: {
                serializedName: "properties.metadata",
                type: {
                    name: "Composite",
                    className: "SCMetadataEntity",
                },
            },
            spec: {
                serializedName: "properties.spec",
                type: {
                    name: "Composite",
                    className: "SchemaRegistryClusterSpecEntity",
                },
            },
            status: {
                serializedName: "properties.status",
                type: {
                    name: "Composite",
                    className: "SchemaRegistryClusterStatusEntity",
                },
            },
        },
    },
};
const SchemaRegistryClusterSpecEntity = {
    type: {
        name: "Composite",
        className: "SchemaRegistryClusterSpecEntity",
        modelProperties: {
            name: {
                serializedName: "name",
                type: {
                    name: "String",
                },
            },
            httpEndpoint: {
                serializedName: "httpEndpoint",
                type: {
                    name: "String",
                },
            },
            package: {
                serializedName: "package",
                type: {
                    name: "String",
                },
            },
            region: {
                serializedName: "region",
                type: {
                    name: "Composite",
                    className: "SchemaRegistryClusterEnvironmentRegionEntity",
                },
            },
            environment: {
                serializedName: "environment",
                type: {
                    name: "Composite",
                    className: "SchemaRegistryClusterEnvironmentRegionEntity",
                },
            },
            cloud: {
                serializedName: "cloud",
                type: {
                    name: "String",
                },
            },
        },
    },
};
const SchemaRegistryClusterEnvironmentRegionEntity = {
    type: {
        name: "Composite",
        className: "SchemaRegistryClusterEnvironmentRegionEntity",
        modelProperties: {
            id: {
                serializedName: "id",
                type: {
                    name: "String",
                },
            },
            related: {
                serializedName: "related",
                type: {
                    name: "String",
                },
            },
            resourceName: {
                serializedName: "resourceName",
                type: {
                    name: "String",
                },
            },
        },
    },
};
const SchemaRegistryClusterStatusEntity = {
    type: {
        name: "Composite",
        className: "SchemaRegistryClusterStatusEntity",
        modelProperties: {
            phase: {
                serializedName: "phase",
                type: {
                    name: "String",
                },
            },
        },
    },
};
const ListRegionsSuccessResponse = {
    type: {
        name: "Composite",
        className: "ListRegionsSuccessResponse",
        modelProperties: {
            data: {
                serializedName: "data",
                type: {
                    name: "Sequence",
                    element: {
                        type: {
                            name: "Composite",
                            className: "RegionRecord",
                        },
                    },
                },
            },
        },
    },
};
const RegionRecord = {
    type: {
        name: "Composite",
        className: "RegionRecord",
        modelProperties: {
            kind: {
                serializedName: "kind",
                type: {
                    name: "String",
                },
            },
            id: {
                serializedName: "id",
                type: {
                    name: "String",
                },
            },
            metadata: {
                serializedName: "properties.metadata",
                type: {
                    name: "Composite",
                    className: "SCMetadataEntity",
                },
            },
            spec: {
                serializedName: "properties.spec",
                type: {
                    name: "Composite",
                    className: "RegionSpecEntity",
                },
            },
        },
    },
};
const RegionSpecEntity = {
    type: {
        name: "Composite",
        className: "RegionSpecEntity",
        modelProperties: {
            name: {
                serializedName: "name",
                type: {
                    name: "String",
                },
            },
            cloud: {
                serializedName: "cloud",
                type: {
                    name: "String",
                },
            },
            regionName: {
                serializedName: "regionName",
                type: {
                    name: "String",
                },
            },
            packages: {
                serializedName: "packages",
                type: {
                    name: "Sequence",
                    element: {
                        type: {
                            name: "String",
                        },
                    },
                },
            },
        },
    },
};
const CreateAPIKeyModel = {
    type: {
        name: "Composite",
        className: "CreateAPIKeyModel",
        modelProperties: {
            name: {
                serializedName: "name",
                type: {
                    name: "String",
                },
            },
            description: {
                serializedName: "description",
                type: {
                    name: "String",
                },
            },
        },
    },
};
const APIKeyRecord = {
    type: {
        name: "Composite",
        className: "APIKeyRecord",
        modelProperties: {
            kind: {
                serializedName: "kind",
                type: {
                    name: "String",
                },
            },
            id: {
                serializedName: "id",
                type: {
                    name: "String",
                },
            },
            metadata: {
                serializedName: "properties.metadata",
                type: {
                    name: "Composite",
                    className: "SCMetadataEntity",
                },
            },
            spec: {
                serializedName: "properties.spec",
                type: {
                    name: "Composite",
                    className: "APIKeySpecEntity",
                },
            },
        },
    },
};
const APIKeySpecEntity = {
    type: {
        name: "Composite",
        className: "APIKeySpecEntity",
        modelProperties: {
            description: {
                serializedName: "description",
                type: {
                    name: "String",
                },
            },
            name: {
                serializedName: "name",
                type: {
                    name: "String",
                },
            },
            secret: {
                serializedName: "secret",
                type: {
                    name: "String",
                },
            },
            resource: {
                serializedName: "resource",
                type: {
                    name: "Composite",
                    className: "APIKeyResourceEntity",
                },
            },
            owner: {
                serializedName: "owner",
                type: {
                    name: "Composite",
                    className: "APIKeyOwnerEntity",
                },
            },
        },
    },
};
const APIKeyResourceEntity = {
    type: {
        name: "Composite",
        className: "APIKeyResourceEntity",
        modelProperties: {
            id: {
                serializedName: "id",
                type: {
                    name: "String",
                },
            },
            environment: {
                serializedName: "environment",
                type: {
                    name: "String",
                },
            },
            related: {
                serializedName: "related",
                type: {
                    name: "String",
                },
            },
            resourceName: {
                serializedName: "resourceName",
                type: {
                    name: "String",
                },
            },
            kind: {
                serializedName: "kind",
                type: {
                    name: "String",
                },
            },
        },
    },
};
const APIKeyOwnerEntity = {
    type: {
        name: "Composite",
        className: "APIKeyOwnerEntity",
        modelProperties: {
            id: {
                serializedName: "id",
                type: {
                    name: "String",
                },
            },
            related: {
                serializedName: "related",
                type: {
                    name: "String",
                },
            },
            resourceName: {
                serializedName: "resourceName",
                type: {
                    name: "String",
                },
            },
            kind: {
                serializedName: "kind",
                type: {
                    name: "String",
                },
            },
        },
    },
};
const SCConfluentListMetadata = {
    type: {
        name: "Composite",
        className: "SCConfluentListMetadata",
        modelProperties: {
            first: {
                serializedName: "first",
                type: {
                    name: "String",
                },
            },
            last: {
                serializedName: "last",
                type: {
                    name: "String",
                },
            },
            prev: {
                serializedName: "prev",
                type: {
                    name: "String",
                },
            },
            next: {
                serializedName: "next",
                type: {
                    name: "String",
                },
            },
            totalSize: {
                serializedName: "totalSize",
                type: {
                    name: "Number",
                },
            },
        },
    },
};

var Mappers = /*#__PURE__*/Object.freeze({
    __proto__: null,
    APIKeyOwnerEntity: APIKeyOwnerEntity,
    APIKeyRecord: APIKeyRecord,
    APIKeyResourceEntity: APIKeyResourceEntity,
    APIKeySpecEntity: APIKeySpecEntity,
    AccessCreateRoleBindingRequestModel: AccessCreateRoleBindingRequestModel,
    AccessInviteUserAccountModel: AccessInviteUserAccountModel,
    AccessInvitedUserDetails: AccessInvitedUserDetails,
    AccessListClusterSuccessResponse: AccessListClusterSuccessResponse,
    AccessListEnvironmentsSuccessResponse: AccessListEnvironmentsSuccessResponse,
    AccessListInvitationsSuccessResponse: AccessListInvitationsSuccessResponse,
    AccessListRoleBindingsSuccessResponse: AccessListRoleBindingsSuccessResponse,
    AccessListServiceAccountsSuccessResponse: AccessListServiceAccountsSuccessResponse,
    AccessListUsersSuccessResponse: AccessListUsersSuccessResponse,
    AccessRoleBindingNameListSuccessResponse: AccessRoleBindingNameListSuccessResponse,
    ClusterByokEntity: ClusterByokEntity,
    ClusterConfigEntity: ClusterConfigEntity,
    ClusterEnvironmentEntity: ClusterEnvironmentEntity,
    ClusterNetworkEntity: ClusterNetworkEntity,
    ClusterRecord: ClusterRecord,
    ClusterSpecEntity: ClusterSpecEntity,
    ClusterStatusEntity: ClusterStatusEntity,
    ConfluentAgreementResource: ConfluentAgreementResource,
    ConfluentAgreementResourceListResponse: ConfluentAgreementResourceListResponse,
    ConfluentListMetadata: ConfluentListMetadata,
    CreateAPIKeyModel: CreateAPIKeyModel,
    EnvironmentRecord: EnvironmentRecord,
    ErrorResponseBody: ErrorResponseBody,
    GetEnvironmentsResponse: GetEnvironmentsResponse,
    InvitationRecord: InvitationRecord,
    LinkOrganization: LinkOrganization,
    ListAccessRequestModel: ListAccessRequestModel,
    ListClustersSuccessResponse: ListClustersSuccessResponse,
    ListRegionsSuccessResponse: ListRegionsSuccessResponse,
    ListSchemaRegistryClustersResponse: ListSchemaRegistryClustersResponse,
    MetadataEntity: MetadataEntity,
    OfferDetail: OfferDetail,
    OperationDisplay: OperationDisplay,
    OperationListResult: OperationListResult,
    OperationResult: OperationResult,
    OrganizationResource: OrganizationResource,
    OrganizationResourceListResult: OrganizationResourceListResult,
    OrganizationResourceUpdate: OrganizationResourceUpdate,
    RegionRecord: RegionRecord,
    RegionSpecEntity: RegionSpecEntity,
    ResourceProviderDefaultErrorResponse: ResourceProviderDefaultErrorResponse,
    RoleBindingRecord: RoleBindingRecord,
    SCClusterByokEntity: SCClusterByokEntity,
    SCClusterNetworkEnvironmentEntity: SCClusterNetworkEnvironmentEntity,
    SCClusterRecord: SCClusterRecord,
    SCClusterSpecEntity: SCClusterSpecEntity,
    SCConfluentListMetadata: SCConfluentListMetadata,
    SCEnvironmentRecord: SCEnvironmentRecord,
    SCMetadataEntity: SCMetadataEntity,
    SchemaRegistryClusterEnvironmentRegionEntity: SchemaRegistryClusterEnvironmentRegionEntity,
    SchemaRegistryClusterRecord: SchemaRegistryClusterRecord,
    SchemaRegistryClusterSpecEntity: SchemaRegistryClusterSpecEntity,
    SchemaRegistryClusterStatusEntity: SchemaRegistryClusterStatusEntity,
    ServiceAccountRecord: ServiceAccountRecord,
    SystemData: SystemData,
    UserDetail: UserDetail,
    UserRecord: UserRecord,
    ValidationResponse: ValidationResponse
});

/*
 * Copyright (c) Microsoft Corporation.
 * Licensed under the MIT License.
 *
 * Code generated by Microsoft (R) AutoRest Code Generator.
 * Changes may cause incorrect behavior and will be lost if the code is regenerated.
 */
const accept = {
    parameterPath: "accept",
    mapper: {
        defaultValue: "application/json",
        isConstant: true,
        serializedName: "Accept",
        type: {
            name: "String",
        },
    },
};
const $host = {
    parameterPath: "$host",
    mapper: {
        serializedName: "$host",
        required: true,
        type: {
            name: "String",
        },
    },
    skipEncoding: true,
};
const apiVersion = {
    parameterPath: "apiVersion",
    mapper: {
        defaultValue: "2024-02-13",
        isConstant: true,
        serializedName: "api-version",
        type: {
            name: "String",
        },
    },
};
const subscriptionId = {
    parameterPath: "subscriptionId",
    mapper: {
        serializedName: "subscriptionId",
        required: true,
        type: {
            name: "String",
        },
    },
};
const contentType = {
    parameterPath: ["options", "contentType"],
    mapper: {
        defaultValue: "application/json",
        isConstant: true,
        serializedName: "Content-Type",
        type: {
            name: "String",
        },
    },
};
const body = {
    parameterPath: ["options", "body"],
    mapper: ConfluentAgreementResource,
};
const nextLink = {
    parameterPath: "nextLink",
    mapper: {
        serializedName: "nextLink",
        required: true,
        type: {
            name: "String",
        },
    },
    skipEncoding: true,
};
const resourceGroupName = {
    parameterPath: "resourceGroupName",
    mapper: {
        serializedName: "resourceGroupName",
        required: true,
        type: {
            name: "String",
        },
    },
};
const organizationName = {
    parameterPath: "organizationName",
    mapper: {
        serializedName: "organizationName",
        required: true,
        type: {
            name: "String",
        },
    },
};
const body1 = {
    parameterPath: ["options", "body"],
    mapper: OrganizationResource,
};
const body2 = {
    parameterPath: ["options", "body"],
    mapper: OrganizationResourceUpdate,
};
const resourceGroupName1 = {
    parameterPath: "resourceGroupName",
    mapper: {
        constraints: {
            MaxLength: 90,
            MinLength: 1,
        },
        serializedName: "resourceGroupName",
        required: true,
        type: {
            name: "String",
        },
    },
};
const pageSize = {
    parameterPath: ["options", "pageSize"],
    mapper: {
        serializedName: "pageSize",
        type: {
            name: "Number",
        },
    },
};
const pageToken = {
    parameterPath: ["options", "pageToken"],
    mapper: {
        serializedName: "pageToken",
        type: {
            name: "String",
        },
    },
};
const environmentId = {
    parameterPath: "environmentId",
    mapper: {
        serializedName: "environmentId",
        required: true,
        type: {
            name: "String",
        },
    },
};
const body3 = {
    parameterPath: "body",
    mapper: ListAccessRequestModel,
};
const body4 = {
    parameterPath: "body",
    mapper: CreateAPIKeyModel,
};
const clusterId = {
    parameterPath: "clusterId",
    mapper: {
        serializedName: "clusterId",
        required: true,
        type: {
            name: "String",
        },
    },
};
const apiKeyId = {
    parameterPath: "apiKeyId",
    mapper: {
        serializedName: "apiKeyId",
        required: true,
        type: {
            name: "String",
        },
    },
};
const body5 = {
    parameterPath: "body",
    mapper: OrganizationResource,
};
const body6 = {
    parameterPath: "body",
    mapper: AccessInviteUserAccountModel,
};
const body7 = {
    parameterPath: "body",
    mapper: AccessCreateRoleBindingRequestModel,
};
const roleBindingId = {
    parameterPath: "roleBindingId",
    mapper: {
        serializedName: "roleBindingId",
        required: true,
        type: {
            name: "String",
        },
    },
};

/*
 * Copyright (c) Microsoft Corporation.
 * Licensed under the MIT License.
 *
 * Code generated by Microsoft (R) AutoRest Code Generator.
 * Changes may cause incorrect behavior and will be lost if the code is regenerated.
 */
/// <reference lib="esnext.asynciterable" />
/** Class containing MarketplaceAgreements operations. */
class MarketplaceAgreementsImpl {
    /**
     * Initialize a new instance of the class MarketplaceAgreements class.
     * @param client Reference to the service client
     */
    constructor(client) {
        this.client = client;
    }
    /**
     * List Confluent marketplace agreements in the subscription.
     * @param options The options parameters.
     */
    list(options) {
        const iter = this.listPagingAll(options);
        return {
            next() {
                return iter.next();
            },
            [Symbol.asyncIterator]() {
                return this;
            },
            byPage: (settings) => {
                if (settings === null || settings === void 0 ? void 0 : settings.maxPageSize) {
                    throw new Error("maxPageSize is not supported by this operation.");
                }
                return this.listPagingPage(options, settings);
            },
        };
    }
    listPagingPage(options, settings) {
        return tslib.__asyncGenerator(this, arguments, function* listPagingPage_1() {
            let result;
            let continuationToken = settings === null || settings === void 0 ? void 0 : settings.continuationToken;
            if (!continuationToken) {
                result = yield tslib.__await(this._list(options));
                let page = result.value || [];
                continuationToken = result.nextLink;
                setContinuationToken(page, continuationToken);
                yield yield tslib.__await(page);
            }
            while (continuationToken) {
                result = yield tslib.__await(this._listNext(continuationToken, options));
                continuationToken = result.nextLink;
                let page = result.value || [];
                setContinuationToken(page, continuationToken);
                yield yield tslib.__await(page);
            }
        });
    }
    listPagingAll(options) {
        return tslib.__asyncGenerator(this, arguments, function* listPagingAll_1() {
            var _a, e_1, _b, _c;
            try {
                for (var _d = true, _e = tslib.__asyncValues(this.listPagingPage(options)), _f; _f = yield tslib.__await(_e.next()), _a = _f.done, !_a; _d = true) {
                    _c = _f.value;
                    _d = false;
                    const page = _c;
                    yield tslib.__await(yield* tslib.__asyncDelegator(tslib.__asyncValues(page)));
                }
            }
            catch (e_1_1) { e_1 = { error: e_1_1 }; }
            finally {
                try {
                    if (!_d && !_a && (_b = _e.return)) yield tslib.__await(_b.call(_e));
                }
                finally { if (e_1) throw e_1.error; }
            }
        });
    }
    /**
     * List Confluent marketplace agreements in the subscription.
     * @param options The options parameters.
     */
    _list(options) {
        return this.client.sendOperationRequest({ options }, listOperationSpec$1);
    }
    /**
     * Create Confluent Marketplace agreement in the subscription.
     * @param options The options parameters.
     */
    create(options) {
        return this.client.sendOperationRequest({ options }, createOperationSpec$1);
    }
    /**
     * ListNext
     * @param nextLink The nextLink from the previous successful call to the List method.
     * @param options The options parameters.
     */
    _listNext(nextLink, options) {
        return this.client.sendOperationRequest({ nextLink, options }, listNextOperationSpec$1);
    }
}
// Operation Specifications
const serializer$4 = coreClient__namespace.createSerializer(Mappers, /* isXml */ false);
const listOperationSpec$1 = {
    path: "/subscriptions/{subscriptionId}/providers/Microsoft.Confluent/agreements",
    httpMethod: "GET",
    responses: {
        200: {
            bodyMapper: ConfluentAgreementResourceListResponse,
        },
        default: {
            bodyMapper: ResourceProviderDefaultErrorResponse,
        },
    },
    queryParameters: [apiVersion],
    urlParameters: [$host, subscriptionId],
    headerParameters: [accept],
    serializer: serializer$4,
};
const createOperationSpec$1 = {
    path: "/subscriptions/{subscriptionId}/providers/Microsoft.Confluent/agreements/default",
    httpMethod: "PUT",
    responses: {
        200: {
            bodyMapper: ConfluentAgreementResource,
        },
        default: {
            bodyMapper: ResourceProviderDefaultErrorResponse,
        },
    },
    requestBody: body,
    queryParameters: [apiVersion],
    urlParameters: [$host, subscriptionId],
    headerParameters: [accept, contentType],
    mediaType: "json",
    serializer: serializer$4,
};
const listNextOperationSpec$1 = {
    path: "{nextLink}",
    httpMethod: "GET",
    responses: {
        200: {
            bodyMapper: ConfluentAgreementResourceListResponse,
        },
        default: {
            bodyMapper: ResourceProviderDefaultErrorResponse,
        },
    },
    urlParameters: [
        $host,
        subscriptionId,
        nextLink,
    ],
    headerParameters: [accept],
    serializer: serializer$4,
};

/*
 * Copyright (c) Microsoft Corporation.
 * Licensed under the MIT License.
 *
 * Code generated by Microsoft (R) AutoRest Code Generator.
 * Changes may cause incorrect behavior and will be lost if the code is regenerated.
 */
/// <reference lib="esnext.asynciterable" />
/** Class containing OrganizationOperations operations. */
class OrganizationOperationsImpl {
    /**
     * Initialize a new instance of the class OrganizationOperations class.
     * @param client Reference to the service client
     */
    constructor(client) {
        this.client = client;
    }
    /**
     * List all operations provided by Microsoft.Confluent.
     * @param options The options parameters.
     */
    list(options) {
        const iter = this.listPagingAll(options);
        return {
            next() {
                return iter.next();
            },
            [Symbol.asyncIterator]() {
                return this;
            },
            byPage: (settings) => {
                if (settings === null || settings === void 0 ? void 0 : settings.maxPageSize) {
                    throw new Error("maxPageSize is not supported by this operation.");
                }
                return this.listPagingPage(options, settings);
            },
        };
    }
    listPagingPage(options, settings) {
        return tslib.__asyncGenerator(this, arguments, function* listPagingPage_1() {
            let result;
            let continuationToken = settings === null || settings === void 0 ? void 0 : settings.continuationToken;
            if (!continuationToken) {
                result = yield tslib.__await(this._list(options));
                let page = result.value || [];
                continuationToken = result.nextLink;
                setContinuationToken(page, continuationToken);
                yield yield tslib.__await(page);
            }
            while (continuationToken) {
                result = yield tslib.__await(this._listNext(continuationToken, options));
                continuationToken = result.nextLink;
                let page = result.value || [];
                setContinuationToken(page, continuationToken);
                yield yield tslib.__await(page);
            }
        });
    }
    listPagingAll(options) {
        return tslib.__asyncGenerator(this, arguments, function* listPagingAll_1() {
            var _a, e_1, _b, _c;
            try {
                for (var _d = true, _e = tslib.__asyncValues(this.listPagingPage(options)), _f; _f = yield tslib.__await(_e.next()), _a = _f.done, !_a; _d = true) {
                    _c = _f.value;
                    _d = false;
                    const page = _c;
                    yield tslib.__await(yield* tslib.__asyncDelegator(tslib.__asyncValues(page)));
                }
            }
            catch (e_1_1) { e_1 = { error: e_1_1 }; }
            finally {
                try {
                    if (!_d && !_a && (_b = _e.return)) yield tslib.__await(_b.call(_e));
                }
                finally { if (e_1) throw e_1.error; }
            }
        });
    }
    /**
     * List all operations provided by Microsoft.Confluent.
     * @param options The options parameters.
     */
    _list(options) {
        return this.client.sendOperationRequest({ options }, listOperationSpec);
    }
    /**
     * ListNext
     * @param nextLink The nextLink from the previous successful call to the List method.
     * @param options The options parameters.
     */
    _listNext(nextLink, options) {
        return this.client.sendOperationRequest({ nextLink, options }, listNextOperationSpec);
    }
}
// Operation Specifications
const serializer$3 = coreClient__namespace.createSerializer(Mappers, /* isXml */ false);
const listOperationSpec = {
    path: "/providers/Microsoft.Confluent/operations",
    httpMethod: "GET",
    responses: {
        200: {
            bodyMapper: OperationListResult,
        },
        default: {
            bodyMapper: ResourceProviderDefaultErrorResponse,
        },
    },
    queryParameters: [apiVersion],
    urlParameters: [$host],
    headerParameters: [accept],
    serializer: serializer$3,
};
const listNextOperationSpec = {
    path: "{nextLink}",
    httpMethod: "GET",
    responses: {
        200: {
            bodyMapper: OperationListResult,
        },
        default: {
            bodyMapper: ResourceProviderDefaultErrorResponse,
        },
    },
    urlParameters: [$host, nextLink],
    headerParameters: [accept],
    serializer: serializer$3,
};

/*
 * Copyright (c) Microsoft Corporation.
 * Licensed under the MIT License.
 *
 * Code generated by Microsoft (R) AutoRest Code Generator.
 * Changes may cause incorrect behavior and will be lost if the code is regenerated.
 */
function createLroSpec(inputs) {
    const { args, spec, sendOperationFn } = inputs;
    return {
        requestMethod: spec.httpMethod,
        requestPath: spec.path,
        sendInitialRequest: () => sendOperationFn(args, spec),
        sendPollRequest: (path, options) => {
            const restSpec = tslib.__rest(spec, ["requestBody"]);
            return sendOperationFn(args, Object.assign(Object.assign({}, restSpec), { httpMethod: "GET", path, abortSignal: options === null || options === void 0 ? void 0 : options.abortSignal }));
        },
    };
}

/*
 * Copyright (c) Microsoft Corporation.
 * Licensed under the MIT License.
 *
 * Code generated by Microsoft (R) AutoRest Code Generator.
 * Changes may cause incorrect behavior and will be lost if the code is regenerated.
 */
/// <reference lib="esnext.asynciterable" />
/** Class containing Organization operations. */
class OrganizationImpl {
    /**
     * Initialize a new instance of the class Organization class.
     * @param client Reference to the service client
     */
    constructor(client) {
        this.client = client;
    }
    /**
     * List all organizations under the specified subscription.
     * @param options The options parameters.
     */
    listBySubscription(options) {
        const iter = this.listBySubscriptionPagingAll(options);
        return {
            next() {
                return iter.next();
            },
            [Symbol.asyncIterator]() {
                return this;
            },
            byPage: (settings) => {
                if (settings === null || settings === void 0 ? void 0 : settings.maxPageSize) {
                    throw new Error("maxPageSize is not supported by this operation.");
                }
                return this.listBySubscriptionPagingPage(options, settings);
            },
        };
    }
    listBySubscriptionPagingPage(options, settings) {
        return tslib.__asyncGenerator(this, arguments, function* listBySubscriptionPagingPage_1() {
            let result;
            let continuationToken = settings === null || settings === void 0 ? void 0 : settings.continuationToken;
            if (!continuationToken) {
                result = yield tslib.__await(this._listBySubscription(options));
                let page = result.value || [];
                continuationToken = result.nextLink;
                setContinuationToken(page, continuationToken);
                yield yield tslib.__await(page);
            }
            while (continuationToken) {
                result = yield tslib.__await(this._listBySubscriptionNext(continuationToken, options));
                continuationToken = result.nextLink;
                let page = result.value || [];
                setContinuationToken(page, continuationToken);
                yield yield tslib.__await(page);
            }
        });
    }
    listBySubscriptionPagingAll(options) {
        return tslib.__asyncGenerator(this, arguments, function* listBySubscriptionPagingAll_1() {
            var _a, e_1, _b, _c;
            try {
                for (var _d = true, _e = tslib.__asyncValues(this.listBySubscriptionPagingPage(options)), _f; _f = yield tslib.__await(_e.next()), _a = _f.done, !_a; _d = true) {
                    _c = _f.value;
                    _d = false;
                    const page = _c;
                    yield tslib.__await(yield* tslib.__asyncDelegator(tslib.__asyncValues(page)));
                }
            }
            catch (e_1_1) { e_1 = { error: e_1_1 }; }
            finally {
                try {
                    if (!_d && !_a && (_b = _e.return)) yield tslib.__await(_b.call(_e));
                }
                finally { if (e_1) throw e_1.error; }
            }
        });
    }
    /**
     * List all Organizations under the specified resource group.
     * @param resourceGroupName Resource group name
     * @param options The options parameters.
     */
    listByResourceGroup(resourceGroupName, options) {
        const iter = this.listByResourceGroupPagingAll(resourceGroupName, options);
        return {
            next() {
                return iter.next();
            },
            [Symbol.asyncIterator]() {
                return this;
            },
            byPage: (settings) => {
                if (settings === null || settings === void 0 ? void 0 : settings.maxPageSize) {
                    throw new Error("maxPageSize is not supported by this operation.");
                }
                return this.listByResourceGroupPagingPage(resourceGroupName, options, settings);
            },
        };
    }
    listByResourceGroupPagingPage(resourceGroupName, options, settings) {
        return tslib.__asyncGenerator(this, arguments, function* listByResourceGroupPagingPage_1() {
            let result;
            let continuationToken = settings === null || settings === void 0 ? void 0 : settings.continuationToken;
            if (!continuationToken) {
                result = yield tslib.__await(this._listByResourceGroup(resourceGroupName, options));
                let page = result.value || [];
                continuationToken = result.nextLink;
                setContinuationToken(page, continuationToken);
                yield yield tslib.__await(page);
            }
            while (continuationToken) {
                result = yield tslib.__await(this._listByResourceGroupNext(resourceGroupName, continuationToken, options));
                continuationToken = result.nextLink;
                let page = result.value || [];
                setContinuationToken(page, continuationToken);
                yield yield tslib.__await(page);
            }
        });
    }
    listByResourceGroupPagingAll(resourceGroupName, options) {
        return tslib.__asyncGenerator(this, arguments, function* listByResourceGroupPagingAll_1() {
            var _a, e_2, _b, _c;
            try {
                for (var _d = true, _e = tslib.__asyncValues(this.listByResourceGroupPagingPage(resourceGroupName, options)), _f; _f = yield tslib.__await(_e.next()), _a = _f.done, !_a; _d = true) {
                    _c = _f.value;
                    _d = false;
                    const page = _c;
                    yield tslib.__await(yield* tslib.__asyncDelegator(tslib.__asyncValues(page)));
                }
            }
            catch (e_2_1) { e_2 = { error: e_2_1 }; }
            finally {
                try {
                    if (!_d && !_a && (_b = _e.return)) yield tslib.__await(_b.call(_e));
                }
                finally { if (e_2) throw e_2.error; }
            }
        });
    }
    /**
     * Lists of all the environments in a organization
     * @param resourceGroupName The name of the resource group. The name is case insensitive.
     * @param organizationName Organization resource name
     * @param options The options parameters.
     */
    listEnvironments(resourceGroupName, organizationName, options) {
        const iter = this.listEnvironmentsPagingAll(resourceGroupName, organizationName, options);
        return {
            next() {
                return iter.next();
            },
            [Symbol.asyncIterator]() {
                return this;
            },
            byPage: (settings) => {
                if (settings === null || settings === void 0 ? void 0 : settings.maxPageSize) {
                    throw new Error("maxPageSize is not supported by this operation.");
                }
                return this.listEnvironmentsPagingPage(resourceGroupName, organizationName, options, settings);
            },
        };
    }
    listEnvironmentsPagingPage(resourceGroupName, organizationName, options, settings) {
        return tslib.__asyncGenerator(this, arguments, function* listEnvironmentsPagingPage_1() {
            let result;
            let continuationToken = settings === null || settings === void 0 ? void 0 : settings.continuationToken;
            if (!continuationToken) {
                result = yield tslib.__await(this._listEnvironments(resourceGroupName, organizationName, options));
                let page = result.value || [];
                continuationToken = result.nextLink;
                setContinuationToken(page, continuationToken);
                yield yield tslib.__await(page);
            }
            while (continuationToken) {
                result = yield tslib.__await(this._listEnvironmentsNext(resourceGroupName, organizationName, continuationToken, options));
                continuationToken = result.nextLink;
                let page = result.value || [];
                setContinuationToken(page, continuationToken);
                yield yield tslib.__await(page);
            }
        });
    }
    listEnvironmentsPagingAll(resourceGroupName, organizationName, options) {
        return tslib.__asyncGenerator(this, arguments, function* listEnvironmentsPagingAll_1() {
            var _a, e_3, _b, _c;
            try {
                for (var _d = true, _e = tslib.__asyncValues(this.listEnvironmentsPagingPage(resourceGroupName, organizationName, options)), _f; _f = yield tslib.__await(_e.next()), _a = _f.done, !_a; _d = true) {
                    _c = _f.value;
                    _d = false;
                    const page = _c;
                    yield tslib.__await(yield* tslib.__asyncDelegator(tslib.__asyncValues(page)));
                }
            }
            catch (e_3_1) { e_3 = { error: e_3_1 }; }
            finally {
                try {
                    if (!_d && !_a && (_b = _e.return)) yield tslib.__await(_b.call(_e));
                }
                finally { if (e_3) throw e_3.error; }
            }
        });
    }
    /**
     * Lists of all the clusters in a environment
     * @param resourceGroupName The name of the resource group. The name is case insensitive.
     * @param organizationName Organization resource name
     * @param environmentId Confluent environment id
     * @param options The options parameters.
     */
    listClusters(resourceGroupName, organizationName, environmentId, options) {
        const iter = this.listClustersPagingAll(resourceGroupName, organizationName, environmentId, options);
        return {
            next() {
                return iter.next();
            },
            [Symbol.asyncIterator]() {
                return this;
            },
            byPage: (settings) => {
                if (settings === null || settings === void 0 ? void 0 : settings.maxPageSize) {
                    throw new Error("maxPageSize is not supported by this operation.");
                }
                return this.listClustersPagingPage(resourceGroupName, organizationName, environmentId, options, settings);
            },
        };
    }
    listClustersPagingPage(resourceGroupName, organizationName, environmentId, options, settings) {
        return tslib.__asyncGenerator(this, arguments, function* listClustersPagingPage_1() {
            let result;
            let continuationToken = settings === null || settings === void 0 ? void 0 : settings.continuationToken;
            if (!continuationToken) {
                result = yield tslib.__await(this._listClusters(resourceGroupName, organizationName, environmentId, options));
                let page = result.value || [];
                continuationToken = result.nextLink;
                setContinuationToken(page, continuationToken);
                yield yield tslib.__await(page);
            }
            while (continuationToken) {
                result = yield tslib.__await(this._listClustersNext(resourceGroupName, organizationName, environmentId, continuationToken, options));
                continuationToken = result.nextLink;
                let page = result.value || [];
                setContinuationToken(page, continuationToken);
                yield yield tslib.__await(page);
            }
        });
    }
    listClustersPagingAll(resourceGroupName, organizationName, environmentId, options) {
        return tslib.__asyncGenerator(this, arguments, function* listClustersPagingAll_1() {
            var _a, e_4, _b, _c;
            try {
                for (var _d = true, _e = tslib.__asyncValues(this.listClustersPagingPage(resourceGroupName, organizationName, environmentId, options)), _f; _f = yield tslib.__await(_e.next()), _a = _f.done, !_a; _d = true) {
                    _c = _f.value;
                    _d = false;
                    const page = _c;
                    yield tslib.__await(yield* tslib.__asyncDelegator(tslib.__asyncValues(page)));
                }
            }
            catch (e_4_1) { e_4 = { error: e_4_1 }; }
            finally {
                try {
                    if (!_d && !_a && (_b = _e.return)) yield tslib.__await(_b.call(_e));
                }
                finally { if (e_4) throw e_4.error; }
            }
        });
    }
    /**
     * Get schema registry clusters
     * @param resourceGroupName The name of the resource group. The name is case insensitive.
     * @param organizationName Organization resource name
     * @param environmentId Confluent environment id
     * @param options The options parameters.
     */
    listSchemaRegistryClusters(resourceGroupName, organizationName, environmentId, options) {
        const iter = this.listSchemaRegistryClustersPagingAll(resourceGroupName, organizationName, environmentId, options);
        return {
            next() {
                return iter.next();
            },
            [Symbol.asyncIterator]() {
                return this;
            },
            byPage: (settings) => {
                if (settings === null || settings === void 0 ? void 0 : settings.maxPageSize) {
                    throw new Error("maxPageSize is not supported by this operation.");
                }
                return this.listSchemaRegistryClustersPagingPage(resourceGroupName, organizationName, environmentId, options, settings);
            },
        };
    }
    listSchemaRegistryClustersPagingPage(resourceGroupName, organizationName, environmentId, options, settings) {
        return tslib.__asyncGenerator(this, arguments, function* listSchemaRegistryClustersPagingPage_1() {
            let result;
            let continuationToken = settings === null || settings === void 0 ? void 0 : settings.continuationToken;
            if (!continuationToken) {
                result = yield tslib.__await(this._listSchemaRegistryClusters(resourceGroupName, organizationName, environmentId, options));
                let page = result.value || [];
                continuationToken = result.nextLink;
                setContinuationToken(page, continuationToken);
                yield yield tslib.__await(page);
            }
            while (continuationToken) {
                result = yield tslib.__await(this._listSchemaRegistryClustersNext(resourceGroupName, organizationName, environmentId, continuationToken, options));
                continuationToken = result.nextLink;
                let page = result.value || [];
                setContinuationToken(page, continuationToken);
                yield yield tslib.__await(page);
            }
        });
    }
    listSchemaRegistryClustersPagingAll(resourceGroupName, organizationName, environmentId, options) {
        return tslib.__asyncGenerator(this, arguments, function* listSchemaRegistryClustersPagingAll_1() {
            var _a, e_5, _b, _c;
            try {
                for (var _d = true, _e = tslib.__asyncValues(this.listSchemaRegistryClustersPagingPage(resourceGroupName, organizationName, environmentId, options)), _f; _f = yield tslib.__await(_e.next()), _a = _f.done, !_a; _d = true) {
                    _c = _f.value;
                    _d = false;
                    const page = _c;
                    yield tslib.__await(yield* tslib.__asyncDelegator(tslib.__asyncValues(page)));
                }
            }
            catch (e_5_1) { e_5 = { error: e_5_1 }; }
            finally {
                try {
                    if (!_d && !_a && (_b = _e.return)) yield tslib.__await(_b.call(_e));
                }
                finally { if (e_5) throw e_5.error; }
            }
        });
    }
    /**
     * List all organizations under the specified subscription.
     * @param options The options parameters.
     */
    _listBySubscription(options) {
        return this.client.sendOperationRequest({ options }, listBySubscriptionOperationSpec);
    }
    /**
     * List all Organizations under the specified resource group.
     * @param resourceGroupName Resource group name
     * @param options The options parameters.
     */
    _listByResourceGroup(resourceGroupName, options) {
        return this.client.sendOperationRequest({ resourceGroupName, options }, listByResourceGroupOperationSpec);
    }
    /**
     * Get the properties of a specific Organization resource.
     * @param resourceGroupName Resource group name
     * @param organizationName Organization resource name
     * @param options The options parameters.
     */
    get(resourceGroupName, organizationName, options) {
        return this.client.sendOperationRequest({ resourceGroupName, organizationName, options }, getOperationSpec);
    }
    /**
     * Create Organization resource
     * @param resourceGroupName Resource group name
     * @param organizationName Organization resource name
     * @param options The options parameters.
     */
    beginCreate(resourceGroupName, organizationName, options) {
        return tslib.__awaiter(this, void 0, void 0, function* () {
            const directSendOperation = (args, spec) => tslib.__awaiter(this, void 0, void 0, function* () {
                return this.client.sendOperationRequest(args, spec);
            });
            const sendOperationFn = (args, spec) => tslib.__awaiter(this, void 0, void 0, function* () {
                var _a;
                let currentRawResponse = undefined;
                const providedCallback = (_a = args.options) === null || _a === void 0 ? void 0 : _a.onResponse;
                const callback = (rawResponse, flatResponse) => {
                    currentRawResponse = rawResponse;
                    providedCallback === null || providedCallback === void 0 ? void 0 : providedCallback(rawResponse, flatResponse);
                };
                const updatedArgs = Object.assign(Object.assign({}, args), { options: Object.assign(Object.assign({}, args.options), { onResponse: callback }) });
                const flatResponse = yield directSendOperation(updatedArgs, spec);
                return {
                    flatResponse,
                    rawResponse: {
                        statusCode: currentRawResponse.status,
                        body: currentRawResponse.parsedBody,
                        headers: currentRawResponse.headers.toJSON(),
                    },
                };
            });
            const lro = createLroSpec({
                sendOperationFn,
                args: { resourceGroupName, organizationName, options },
                spec: createOperationSpec,
            });
            const poller = yield coreLro.createHttpPoller(lro, {
                restoreFrom: options === null || options === void 0 ? void 0 : options.resumeFrom,
                intervalInMs: options === null || options === void 0 ? void 0 : options.updateIntervalInMs,
                resourceLocationConfig: "azure-async-operation",
            });
            yield poller.poll();
            return poller;
        });
    }
    /**
     * Create Organization resource
     * @param resourceGroupName Resource group name
     * @param organizationName Organization resource name
     * @param options The options parameters.
     */
    beginCreateAndWait(resourceGroupName, organizationName, options) {
        return tslib.__awaiter(this, void 0, void 0, function* () {
            const poller = yield this.beginCreate(resourceGroupName, organizationName, options);
            return poller.pollUntilDone();
        });
    }
    /**
     * Update Organization resource
     * @param resourceGroupName Resource group name
     * @param organizationName Organization resource name
     * @param options The options parameters.
     */
    update(resourceGroupName, organizationName, options) {
        return this.client.sendOperationRequest({ resourceGroupName, organizationName, options }, updateOperationSpec);
    }
    /**
     * Delete Organization resource
     * @param resourceGroupName Resource group name
     * @param organizationName Organization resource name
     * @param options The options parameters.
     */
    beginDelete(resourceGroupName, organizationName, options) {
        return tslib.__awaiter(this, void 0, void 0, function* () {
            const directSendOperation = (args, spec) => tslib.__awaiter(this, void 0, void 0, function* () {
                return this.client.sendOperationRequest(args, spec);
            });
            const sendOperationFn = (args, spec) => tslib.__awaiter(this, void 0, void 0, function* () {
                var _a;
                let currentRawResponse = undefined;
                const providedCallback = (_a = args.options) === null || _a === void 0 ? void 0 : _a.onResponse;
                const callback = (rawResponse, flatResponse) => {
                    currentRawResponse = rawResponse;
                    providedCallback === null || providedCallback === void 0 ? void 0 : providedCallback(rawResponse, flatResponse);
                };
                const updatedArgs = Object.assign(Object.assign({}, args), { options: Object.assign(Object.assign({}, args.options), { onResponse: callback }) });
                const flatResponse = yield directSendOperation(updatedArgs, spec);
                return {
                    flatResponse,
                    rawResponse: {
                        statusCode: currentRawResponse.status,
                        body: currentRawResponse.parsedBody,
                        headers: currentRawResponse.headers.toJSON(),
                    },
                };
            });
            const lro = createLroSpec({
                sendOperationFn,
                args: { resourceGroupName, organizationName, options },
                spec: deleteOperationSpec,
            });
            const poller = yield coreLro.createHttpPoller(lro, {
                restoreFrom: options === null || options === void 0 ? void 0 : options.resumeFrom,
                intervalInMs: options === null || options === void 0 ? void 0 : options.updateIntervalInMs,
                resourceLocationConfig: "location",
            });
            yield poller.poll();
            return poller;
        });
    }
    /**
     * Delete Organization resource
     * @param resourceGroupName Resource group name
     * @param organizationName Organization resource name
     * @param options The options parameters.
     */
    beginDeleteAndWait(resourceGroupName, organizationName, options) {
        return tslib.__awaiter(this, void 0, void 0, function* () {
            const poller = yield this.beginDelete(resourceGroupName, organizationName, options);
            return poller.pollUntilDone();
        });
    }
    /**
     * Lists of all the environments in a organization
     * @param resourceGroupName The name of the resource group. The name is case insensitive.
     * @param organizationName Organization resource name
     * @param options The options parameters.
     */
    _listEnvironments(resourceGroupName, organizationName, options) {
        return this.client.sendOperationRequest({ resourceGroupName, organizationName, options }, listEnvironmentsOperationSpec$1);
    }
    /**
     * Get Environment details by environment Id
     * @param resourceGroupName The name of the resource group. The name is case insensitive.
     * @param organizationName Organization resource name
     * @param environmentId Confluent environment id
     * @param options The options parameters.
     */
    getEnvironmentById(resourceGroupName, organizationName, environmentId, options) {
        return this.client.sendOperationRequest({ resourceGroupName, organizationName, environmentId, options }, getEnvironmentByIdOperationSpec);
    }
    /**
     * Lists of all the clusters in a environment
     * @param resourceGroupName The name of the resource group. The name is case insensitive.
     * @param organizationName Organization resource name
     * @param environmentId Confluent environment id
     * @param options The options parameters.
     */
    _listClusters(resourceGroupName, organizationName, environmentId, options) {
        return this.client.sendOperationRequest({ resourceGroupName, organizationName, environmentId, options }, listClustersOperationSpec$1);
    }
    /**
     * Get schema registry clusters
     * @param resourceGroupName The name of the resource group. The name is case insensitive.
     * @param organizationName Organization resource name
     * @param environmentId Confluent environment id
     * @param options The options parameters.
     */
    _listSchemaRegistryClusters(resourceGroupName, organizationName, environmentId, options) {
        return this.client.sendOperationRequest({ resourceGroupName, organizationName, environmentId, options }, listSchemaRegistryClustersOperationSpec);
    }
    /**
     * cloud provider regions available for creating Schema Registry clusters.
     * @param resourceGroupName The name of the resource group. The name is case insensitive.
     * @param organizationName Organization resource name
     * @param body List Access Request Model
     * @param options The options parameters.
     */
    listRegions(resourceGroupName, organizationName, body, options) {
        return this.client.sendOperationRequest({ resourceGroupName, organizationName, body, options }, listRegionsOperationSpec);
    }
    /**
     * Creates API key for a schema registry Cluster ID or Kafka Cluster ID under a environment
     * @param resourceGroupName The name of the resource group. The name is case insensitive.
     * @param organizationName Organization resource name
     * @param environmentId Confluent environment id
     * @param clusterId Confluent kafka or schema registry cluster id
     * @param body Request payload for get creating API Key for schema registry Cluster ID or Kafka Cluster
     *             ID under a environment
     * @param options The options parameters.
     */
    createAPIKey(resourceGroupName, organizationName, environmentId, clusterId, body, options) {
        return this.client.sendOperationRequest({
            resourceGroupName,
            organizationName,
            environmentId,
            clusterId,
            body,
            options,
        }, createAPIKeyOperationSpec);
    }
    /**
     * Deletes API key of a kafka or schema registry cluster
     * @param resourceGroupName The name of the resource group. The name is case insensitive.
     * @param organizationName Organization resource name
     * @param apiKeyId Confluent API Key id
     * @param options The options parameters.
     */
    deleteClusterAPIKey(resourceGroupName, organizationName, apiKeyId, options) {
        return this.client.sendOperationRequest({ resourceGroupName, organizationName, apiKeyId, options }, deleteClusterAPIKeyOperationSpec);
    }
    /**
     * Get API key details of a kafka or schema registry cluster
     * @param resourceGroupName The name of the resource group. The name is case insensitive.
     * @param organizationName Organization resource name
     * @param apiKeyId Confluent API Key id
     * @param options The options parameters.
     */
    getClusterAPIKey(resourceGroupName, organizationName, apiKeyId, options) {
        return this.client.sendOperationRequest({ resourceGroupName, organizationName, apiKeyId, options }, getClusterAPIKeyOperationSpec);
    }
    /**
     * Get schema registry cluster by Id
     * @param resourceGroupName The name of the resource group. The name is case insensitive.
     * @param organizationName Organization resource name
     * @param environmentId Confluent environment id
     * @param clusterId Confluent kafka or schema registry cluster id
     * @param options The options parameters.
     */
    getSchemaRegistryClusterById(resourceGroupName, organizationName, environmentId, clusterId, options) {
        return this.client.sendOperationRequest({
            resourceGroupName,
            organizationName,
            environmentId,
            clusterId,
            options,
        }, getSchemaRegistryClusterByIdOperationSpec);
    }
    /**
     * Get cluster by Id
     * @param resourceGroupName The name of the resource group. The name is case insensitive.
     * @param organizationName Organization resource name
     * @param environmentId Confluent environment id
     * @param clusterId Confluent kafka or schema registry cluster id
     * @param options The options parameters.
     */
    getClusterById(resourceGroupName, organizationName, environmentId, clusterId, options) {
        return this.client.sendOperationRequest({
            resourceGroupName,
            organizationName,
            environmentId,
            clusterId,
            options,
        }, getClusterByIdOperationSpec);
    }
    /**
     * ListBySubscriptionNext
     * @param nextLink The nextLink from the previous successful call to the ListBySubscription method.
     * @param options The options parameters.
     */
    _listBySubscriptionNext(nextLink, options) {
        return this.client.sendOperationRequest({ nextLink, options }, listBySubscriptionNextOperationSpec);
    }
    /**
     * ListByResourceGroupNext
     * @param resourceGroupName Resource group name
     * @param nextLink The nextLink from the previous successful call to the ListByResourceGroup method.
     * @param options The options parameters.
     */
    _listByResourceGroupNext(resourceGroupName, nextLink, options) {
        return this.client.sendOperationRequest({ resourceGroupName, nextLink, options }, listByResourceGroupNextOperationSpec);
    }
    /**
     * ListEnvironmentsNext
     * @param resourceGroupName The name of the resource group. The name is case insensitive.
     * @param organizationName Organization resource name
     * @param nextLink The nextLink from the previous successful call to the ListEnvironments method.
     * @param options The options parameters.
     */
    _listEnvironmentsNext(resourceGroupName, organizationName, nextLink, options) {
        return this.client.sendOperationRequest({ resourceGroupName, organizationName, nextLink, options }, listEnvironmentsNextOperationSpec);
    }
    /**
     * ListClustersNext
     * @param resourceGroupName The name of the resource group. The name is case insensitive.
     * @param organizationName Organization resource name
     * @param environmentId Confluent environment id
     * @param nextLink The nextLink from the previous successful call to the ListClusters method.
     * @param options The options parameters.
     */
    _listClustersNext(resourceGroupName, organizationName, environmentId, nextLink, options) {
        return this.client.sendOperationRequest({ resourceGroupName, organizationName, environmentId, nextLink, options }, listClustersNextOperationSpec);
    }
    /**
     * ListSchemaRegistryClustersNext
     * @param resourceGroupName The name of the resource group. The name is case insensitive.
     * @param organizationName Organization resource name
     * @param environmentId Confluent environment id
     * @param nextLink The nextLink from the previous successful call to the ListSchemaRegistryClusters
     *                 method.
     * @param options The options parameters.
     */
    _listSchemaRegistryClustersNext(resourceGroupName, organizationName, environmentId, nextLink, options) {
        return this.client.sendOperationRequest({ resourceGroupName, organizationName, environmentId, nextLink, options }, listSchemaRegistryClustersNextOperationSpec);
    }
}
// Operation Specifications
const serializer$2 = coreClient__namespace.createSerializer(Mappers, /* isXml */ false);
const listBySubscriptionOperationSpec = {
    path: "/subscriptions/{subscriptionId}/providers/Microsoft.Confluent/organizations",
    httpMethod: "GET",
    responses: {
        200: {
            bodyMapper: OrganizationResourceListResult,
        },
        default: {
            bodyMapper: ResourceProviderDefaultErrorResponse,
        },
    },
    queryParameters: [apiVersion],
    urlParameters: [$host, subscriptionId],
    headerParameters: [accept],
    serializer: serializer$2,
};
const listByResourceGroupOperationSpec = {
    path: "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Confluent/organizations",
    httpMethod: "GET",
    responses: {
        200: {
            bodyMapper: OrganizationResourceListResult,
        },
        default: {
            bodyMapper: ResourceProviderDefaultErrorResponse,
        },
    },
    queryParameters: [apiVersion],
    urlParameters: [
        $host,
        subscriptionId,
        resourceGroupName,
    ],
    headerParameters: [accept],
    serializer: serializer$2,
};
const getOperationSpec = {
    path: "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Confluent/organizations/{organizationName}",
    httpMethod: "GET",
    responses: {
        200: {
            bodyMapper: OrganizationResource,
        },
        default: {
            bodyMapper: ResourceProviderDefaultErrorResponse,
        },
    },
    queryParameters: [apiVersion],
    urlParameters: [
        $host,
        subscriptionId,
        resourceGroupName,
        organizationName,
    ],
    headerParameters: [accept],
    serializer: serializer$2,
};
const createOperationSpec = {
    path: "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Confluent/organizations/{organizationName}",
    httpMethod: "PUT",
    responses: {
        200: {
            bodyMapper: OrganizationResource,
        },
        201: {
            bodyMapper: OrganizationResource,
        },
        202: {
            bodyMapper: OrganizationResource,
        },
        204: {
            bodyMapper: OrganizationResource,
        },
        default: {
            bodyMapper: ResourceProviderDefaultErrorResponse,
        },
    },
    requestBody: body1,
    queryParameters: [apiVersion],
    urlParameters: [
        $host,
        subscriptionId,
        resourceGroupName,
        organizationName,
    ],
    headerParameters: [accept, contentType],
    mediaType: "json",
    serializer: serializer$2,
};
const updateOperationSpec = {
    path: "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Confluent/organizations/{organizationName}",
    httpMethod: "PATCH",
    responses: {
        200: {
            bodyMapper: OrganizationResource,
        },
        default: {
            bodyMapper: ResourceProviderDefaultErrorResponse,
        },
    },
    requestBody: body2,
    queryParameters: [apiVersion],
    urlParameters: [
        $host,
        subscriptionId,
        resourceGroupName,
        organizationName,
    ],
    headerParameters: [accept, contentType],
    mediaType: "json",
    serializer: serializer$2,
};
const deleteOperationSpec = {
    path: "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Confluent/organizations/{organizationName}",
    httpMethod: "DELETE",
    responses: {
        200: {},
        201: {},
        202: {},
        204: {},
        default: {
            bodyMapper: ResourceProviderDefaultErrorResponse,
        },
    },
    queryParameters: [apiVersion],
    urlParameters: [
        $host,
        subscriptionId,
        resourceGroupName,
        organizationName,
    ],
    headerParameters: [accept],
    serializer: serializer$2,
};
const listEnvironmentsOperationSpec$1 = {
    path: "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Confluent/organizations/{organizationName}/environments",
    httpMethod: "GET",
    responses: {
        200: {
            bodyMapper: GetEnvironmentsResponse,
        },
        default: {
            bodyMapper: ResourceProviderDefaultErrorResponse,
        },
    },
    queryParameters: [
        apiVersion,
        pageSize,
        pageToken,
    ],
    urlParameters: [
        $host,
        subscriptionId,
        organizationName,
        resourceGroupName1,
    ],
    headerParameters: [accept],
    serializer: serializer$2,
};
const getEnvironmentByIdOperationSpec = {
    path: "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Confluent/organizations/{organizationName}/environments/{environmentId}",
    httpMethod: "GET",
    responses: {
        200: {
            bodyMapper: SCEnvironmentRecord,
        },
        default: {
            bodyMapper: ResourceProviderDefaultErrorResponse,
        },
    },
    queryParameters: [apiVersion],
    urlParameters: [
        $host,
        subscriptionId,
        organizationName,
        resourceGroupName1,
        environmentId,
    ],
    headerParameters: [accept],
    serializer: serializer$2,
};
const listClustersOperationSpec$1 = {
    path: "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Confluent/organizations/{organizationName}/environments/{environmentId}/clusters",
    httpMethod: "GET",
    responses: {
        200: {
            bodyMapper: ListClustersSuccessResponse,
        },
        default: {
            bodyMapper: ResourceProviderDefaultErrorResponse,
        },
    },
    queryParameters: [
        apiVersion,
        pageSize,
        pageToken,
    ],
    urlParameters: [
        $host,
        subscriptionId,
        organizationName,
        resourceGroupName1,
        environmentId,
    ],
    headerParameters: [accept],
    serializer: serializer$2,
};
const listSchemaRegistryClustersOperationSpec = {
    path: "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Confluent/organizations/{organizationName}/environments/{environmentId}/schemaRegistryClusters",
    httpMethod: "GET",
    responses: {
        200: {
            bodyMapper: ListSchemaRegistryClustersResponse,
        },
        default: {
            bodyMapper: ResourceProviderDefaultErrorResponse,
        },
    },
    queryParameters: [
        apiVersion,
        pageSize,
        pageToken,
    ],
    urlParameters: [
        $host,
        subscriptionId,
        organizationName,
        resourceGroupName1,
        environmentId,
    ],
    headerParameters: [accept],
    serializer: serializer$2,
};
const listRegionsOperationSpec = {
    path: "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Confluent/organizations/{organizationName}/listRegions",
    httpMethod: "POST",
    responses: {
        200: {
            bodyMapper: ListRegionsSuccessResponse,
        },
        default: {
            bodyMapper: ResourceProviderDefaultErrorResponse,
        },
    },
    requestBody: body3,
    queryParameters: [apiVersion],
    urlParameters: [
        $host,
        subscriptionId,
        organizationName,
        resourceGroupName1,
    ],
    headerParameters: [accept, contentType],
    mediaType: "json",
    serializer: serializer$2,
};
const createAPIKeyOperationSpec = {
    path: "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Confluent/organizations/{organizationName}/environments/{environmentId}/clusters/{clusterId}/createAPIKey",
    httpMethod: "POST",
    responses: {
        200: {
            bodyMapper: APIKeyRecord,
        },
        default: {
            bodyMapper: ResourceProviderDefaultErrorResponse,
        },
    },
    requestBody: body4,
    queryParameters: [apiVersion],
    urlParameters: [
        $host,
        subscriptionId,
        organizationName,
        resourceGroupName1,
        environmentId,
        clusterId,
    ],
    headerParameters: [accept, contentType],
    mediaType: "json",
    serializer: serializer$2,
};
const deleteClusterAPIKeyOperationSpec = {
    path: "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Confluent/organizations/{organizationName}/apiKeys/{apiKeyId}",
    httpMethod: "DELETE",
    responses: {
        200: {},
        204: {},
        default: {
            bodyMapper: ResourceProviderDefaultErrorResponse,
        },
    },
    queryParameters: [apiVersion],
    urlParameters: [
        $host,
        subscriptionId,
        organizationName,
        resourceGroupName1,
        apiKeyId,
    ],
    headerParameters: [accept],
    serializer: serializer$2,
};
const getClusterAPIKeyOperationSpec = {
    path: "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Confluent/organizations/{organizationName}/apiKeys/{apiKeyId}",
    httpMethod: "GET",
    responses: {
        200: {
            bodyMapper: APIKeyRecord,
        },
        default: {
            bodyMapper: ResourceProviderDefaultErrorResponse,
        },
    },
    queryParameters: [apiVersion],
    urlParameters: [
        $host,
        subscriptionId,
        organizationName,
        resourceGroupName1,
        apiKeyId,
    ],
    headerParameters: [accept],
    serializer: serializer$2,
};
const getSchemaRegistryClusterByIdOperationSpec = {
    path: "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Confluent/organizations/{organizationName}/environments/{environmentId}/schemaRegistryClusters/{clusterId}",
    httpMethod: "GET",
    responses: {
        200: {
            bodyMapper: SchemaRegistryClusterRecord,
        },
        default: {
            bodyMapper: ResourceProviderDefaultErrorResponse,
        },
    },
    queryParameters: [apiVersion],
    urlParameters: [
        $host,
        subscriptionId,
        organizationName,
        resourceGroupName1,
        environmentId,
        clusterId,
    ],
    headerParameters: [accept],
    serializer: serializer$2,
};
const getClusterByIdOperationSpec = {
    path: "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Confluent/organizations/{organizationName}/environments/{environmentId}/clusters/{clusterId}",
    httpMethod: "GET",
    responses: {
        200: {
            bodyMapper: SCClusterRecord,
        },
        default: {
            bodyMapper: ResourceProviderDefaultErrorResponse,
        },
    },
    queryParameters: [apiVersion],
    urlParameters: [
        $host,
        subscriptionId,
        organizationName,
        resourceGroupName1,
        environmentId,
        clusterId,
    ],
    headerParameters: [accept],
    serializer: serializer$2,
};
const listBySubscriptionNextOperationSpec = {
    path: "{nextLink}",
    httpMethod: "GET",
    responses: {
        200: {
            bodyMapper: OrganizationResourceListResult,
        },
        default: {
            bodyMapper: ResourceProviderDefaultErrorResponse,
        },
    },
    urlParameters: [
        $host,
        subscriptionId,
        nextLink,
    ],
    headerParameters: [accept],
    serializer: serializer$2,
};
const listByResourceGroupNextOperationSpec = {
    path: "{nextLink}",
    httpMethod: "GET",
    responses: {
        200: {
            bodyMapper: OrganizationResourceListResult,
        },
        default: {
            bodyMapper: ResourceProviderDefaultErrorResponse,
        },
    },
    urlParameters: [
        $host,
        subscriptionId,
        nextLink,
        resourceGroupName,
    ],
    headerParameters: [accept],
    serializer: serializer$2,
};
const listEnvironmentsNextOperationSpec = {
    path: "{nextLink}",
    httpMethod: "GET",
    responses: {
        200: {
            bodyMapper: GetEnvironmentsResponse,
        },
        default: {
            bodyMapper: ResourceProviderDefaultErrorResponse,
        },
    },
    urlParameters: [
        $host,
        subscriptionId,
        nextLink,
        organizationName,
        resourceGroupName1,
    ],
    headerParameters: [accept],
    serializer: serializer$2,
};
const listClustersNextOperationSpec = {
    path: "{nextLink}",
    httpMethod: "GET",
    responses: {
        200: {
            bodyMapper: ListClustersSuccessResponse,
        },
        default: {
            bodyMapper: ResourceProviderDefaultErrorResponse,
        },
    },
    urlParameters: [
        $host,
        subscriptionId,
        nextLink,
        organizationName,
        resourceGroupName1,
        environmentId,
    ],
    headerParameters: [accept],
    serializer: serializer$2,
};
const listSchemaRegistryClustersNextOperationSpec = {
    path: "{nextLink}",
    httpMethod: "GET",
    responses: {
        200: {
            bodyMapper: ListSchemaRegistryClustersResponse,
        },
        default: {
            bodyMapper: ResourceProviderDefaultErrorResponse,
        },
    },
    urlParameters: [
        $host,
        subscriptionId,
        nextLink,
        organizationName,
        resourceGroupName1,
        environmentId,
    ],
    headerParameters: [accept],
    serializer: serializer$2,
};

/*
 * Copyright (c) Microsoft Corporation.
 * Licensed under the MIT License.
 *
 * Code generated by Microsoft (R) AutoRest Code Generator.
 * Changes may cause incorrect behavior and will be lost if the code is regenerated.
 */
/** Class containing Validations operations. */
class ValidationsImpl {
    /**
     * Initialize a new instance of the class Validations class.
     * @param client Reference to the service client
     */
    constructor(client) {
        this.client = client;
    }
    /**
     * Organization Validate proxy resource
     * @param resourceGroupName Resource group name
     * @param organizationName Organization resource name
     * @param body Organization resource model
     * @param options The options parameters.
     */
    validateOrganization(resourceGroupName, organizationName, body, options) {
        return this.client.sendOperationRequest({ resourceGroupName, organizationName, body, options }, validateOrganizationOperationSpec);
    }
    /**
     * Organization Validate proxy resource
     * @param resourceGroupName Resource group name
     * @param organizationName Organization resource name
     * @param body Organization resource model
     * @param options The options parameters.
     */
    validateOrganizationV2(resourceGroupName, organizationName, body, options) {
        return this.client.sendOperationRequest({ resourceGroupName, organizationName, body, options }, validateOrganizationV2OperationSpec);
    }
}
// Operation Specifications
const serializer$1 = coreClient__namespace.createSerializer(Mappers, /* isXml */ false);
const validateOrganizationOperationSpec = {
    path: "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Confluent/validations/{organizationName}/orgvalidate",
    httpMethod: "POST",
    responses: {
        200: {
            bodyMapper: OrganizationResource,
        },
        default: {
            bodyMapper: ResourceProviderDefaultErrorResponse,
        },
    },
    requestBody: body5,
    queryParameters: [apiVersion],
    urlParameters: [
        $host,
        subscriptionId,
        resourceGroupName,
        organizationName,
    ],
    headerParameters: [accept, contentType],
    mediaType: "json",
    serializer: serializer$1,
};
const validateOrganizationV2OperationSpec = {
    path: "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Confluent/validations/{organizationName}/orgvalidateV2",
    httpMethod: "POST",
    responses: {
        200: {
            bodyMapper: ValidationResponse,
        },
        default: {
            bodyMapper: ResourceProviderDefaultErrorResponse,
        },
    },
    requestBody: body5,
    queryParameters: [apiVersion],
    urlParameters: [
        $host,
        subscriptionId,
        resourceGroupName,
        organizationName,
    ],
    headerParameters: [accept, contentType],
    mediaType: "json",
    serializer: serializer$1,
};

/*
 * Copyright (c) Microsoft Corporation.
 * Licensed under the MIT License.
 *
 * Code generated by Microsoft (R) AutoRest Code Generator.
 * Changes may cause incorrect behavior and will be lost if the code is regenerated.
 */
/** Class containing Access operations. */
class AccessImpl {
    /**
     * Initialize a new instance of the class Access class.
     * @param client Reference to the service client
     */
    constructor(client) {
        this.client = client;
    }
    /**
     * Organization users details
     * @param resourceGroupName Resource group name
     * @param organizationName Organization resource name
     * @param body List Access Request Model
     * @param options The options parameters.
     */
    listUsers(resourceGroupName, organizationName, body, options) {
        return this.client.sendOperationRequest({ resourceGroupName, organizationName, body, options }, listUsersOperationSpec);
    }
    /**
     * Organization service accounts details
     * @param resourceGroupName Resource group name
     * @param organizationName Organization resource name
     * @param body List Access Request Model
     * @param options The options parameters.
     */
    listServiceAccounts(resourceGroupName, organizationName, body, options) {
        return this.client.sendOperationRequest({ resourceGroupName, organizationName, body, options }, listServiceAccountsOperationSpec);
    }
    /**
     * Organization accounts invitation details
     * @param resourceGroupName Resource group name
     * @param organizationName Organization resource name
     * @param body List Access Request Model
     * @param options The options parameters.
     */
    listInvitations(resourceGroupName, organizationName, body, options) {
        return this.client.sendOperationRequest({ resourceGroupName, organizationName, body, options }, listInvitationsOperationSpec);
    }
    /**
     * Invite user to the organization
     * @param resourceGroupName Resource group name
     * @param organizationName Organization resource name
     * @param body Invite user account model
     * @param options The options parameters.
     */
    inviteUser(resourceGroupName, organizationName, body, options) {
        return this.client.sendOperationRequest({ resourceGroupName, organizationName, body, options }, inviteUserOperationSpec);
    }
    /**
     * Environment list of an organization
     * @param resourceGroupName The name of the resource group. The name is case insensitive.
     * @param organizationName Organization resource name
     * @param body List Access Request Model
     * @param options The options parameters.
     */
    listEnvironments(resourceGroupName, organizationName, body, options) {
        return this.client.sendOperationRequest({ resourceGroupName, organizationName, body, options }, listEnvironmentsOperationSpec);
    }
    /**
     * Cluster details
     * @param resourceGroupName The name of the resource group. The name is case insensitive.
     * @param organizationName Organization resource name
     * @param body List Access Request Model
     * @param options The options parameters.
     */
    listClusters(resourceGroupName, organizationName, body, options) {
        return this.client.sendOperationRequest({ resourceGroupName, organizationName, body, options }, listClustersOperationSpec);
    }
    /**
     * Organization role bindings
     * @param resourceGroupName The name of the resource group. The name is case insensitive.
     * @param organizationName Organization resource name
     * @param body List Access Request Model
     * @param options The options parameters.
     */
    listRoleBindings(resourceGroupName, organizationName, body, options) {
        return this.client.sendOperationRequest({ resourceGroupName, organizationName, body, options }, listRoleBindingsOperationSpec);
    }
    /**
     * Organization role bindings
     * @param resourceGroupName The name of the resource group. The name is case insensitive.
     * @param organizationName Organization resource name
     * @param body Create role binding Request Model
     * @param options The options parameters.
     */
    createRoleBinding(resourceGroupName, organizationName, body, options) {
        return this.client.sendOperationRequest({ resourceGroupName, organizationName, body, options }, createRoleBindingOperationSpec);
    }
    /**
     * Organization role bindings
     * @param resourceGroupName The name of the resource group. The name is case insensitive.
     * @param organizationName Organization resource name
     * @param roleBindingId Confluent Role binding id
     * @param options The options parameters.
     */
    deleteRoleBinding(resourceGroupName, organizationName, roleBindingId, options) {
        return this.client.sendOperationRequest({ resourceGroupName, organizationName, roleBindingId, options }, deleteRoleBindingOperationSpec);
    }
    /**
     * Organization role bindings
     * @param resourceGroupName The name of the resource group. The name is case insensitive.
     * @param organizationName Organization resource name
     * @param body List Access Request Model
     * @param options The options parameters.
     */
    listRoleBindingNameList(resourceGroupName, organizationName, body, options) {
        return this.client.sendOperationRequest({ resourceGroupName, organizationName, body, options }, listRoleBindingNameListOperationSpec);
    }
}
// Operation Specifications
const serializer = coreClient__namespace.createSerializer(Mappers, /* isXml */ false);
const listUsersOperationSpec = {
    path: "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Confluent/organizations/{organizationName}/access/default/listUsers",
    httpMethod: "POST",
    responses: {
        200: {
            bodyMapper: AccessListUsersSuccessResponse,
        },
        default: {
            bodyMapper: ResourceProviderDefaultErrorResponse,
        },
    },
    requestBody: body3,
    queryParameters: [apiVersion],
    urlParameters: [
        $host,
        subscriptionId,
        resourceGroupName,
        organizationName,
    ],
    headerParameters: [accept, contentType],
    mediaType: "json",
    serializer,
};
const listServiceAccountsOperationSpec = {
    path: "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Confluent/organizations/{organizationName}/access/default/listServiceAccounts",
    httpMethod: "POST",
    responses: {
        200: {
            bodyMapper: AccessListServiceAccountsSuccessResponse,
        },
        default: {
            bodyMapper: ResourceProviderDefaultErrorResponse,
        },
    },
    requestBody: body3,
    queryParameters: [apiVersion],
    urlParameters: [
        $host,
        subscriptionId,
        resourceGroupName,
        organizationName,
    ],
    headerParameters: [accept, contentType],
    mediaType: "json",
    serializer,
};
const listInvitationsOperationSpec = {
    path: "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Confluent/organizations/{organizationName}/access/default/listInvitations",
    httpMethod: "POST",
    responses: {
        200: {
            bodyMapper: AccessListInvitationsSuccessResponse,
        },
        default: {
            bodyMapper: ResourceProviderDefaultErrorResponse,
        },
    },
    requestBody: body3,
    queryParameters: [apiVersion],
    urlParameters: [
        $host,
        subscriptionId,
        resourceGroupName,
        organizationName,
    ],
    headerParameters: [accept, contentType],
    mediaType: "json",
    serializer,
};
const inviteUserOperationSpec = {
    path: "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Confluent/organizations/{organizationName}/access/default/createInvitation",
    httpMethod: "POST",
    responses: {
        200: {
            bodyMapper: InvitationRecord,
        },
        default: {
            bodyMapper: ResourceProviderDefaultErrorResponse,
        },
    },
    requestBody: body6,
    queryParameters: [apiVersion],
    urlParameters: [
        $host,
        subscriptionId,
        resourceGroupName,
        organizationName,
    ],
    headerParameters: [accept, contentType],
    mediaType: "json",
    serializer,
};
const listEnvironmentsOperationSpec = {
    path: "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Confluent/organizations/{organizationName}/access/default/listEnvironments",
    httpMethod: "POST",
    responses: {
        200: {
            bodyMapper: AccessListEnvironmentsSuccessResponse,
        },
        default: {
            bodyMapper: ResourceProviderDefaultErrorResponse,
        },
    },
    requestBody: body3,
    queryParameters: [apiVersion],
    urlParameters: [
        $host,
        subscriptionId,
        organizationName,
        resourceGroupName1,
    ],
    headerParameters: [accept, contentType],
    mediaType: "json",
    serializer,
};
const listClustersOperationSpec = {
    path: "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Confluent/organizations/{organizationName}/access/default/listClusters",
    httpMethod: "POST",
    responses: {
        200: {
            bodyMapper: AccessListClusterSuccessResponse,
        },
        default: {
            bodyMapper: ResourceProviderDefaultErrorResponse,
        },
    },
    requestBody: body3,
    queryParameters: [apiVersion],
    urlParameters: [
        $host,
        subscriptionId,
        organizationName,
        resourceGroupName1,
    ],
    headerParameters: [accept, contentType],
    mediaType: "json",
    serializer,
};
const listRoleBindingsOperationSpec = {
    path: "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Confluent/organizations/{organizationName}/access/default/listRoleBindings",
    httpMethod: "POST",
    responses: {
        200: {
            bodyMapper: AccessListRoleBindingsSuccessResponse,
        },
        default: {
            bodyMapper: ResourceProviderDefaultErrorResponse,
        },
    },
    requestBody: body3,
    queryParameters: [apiVersion],
    urlParameters: [
        $host,
        subscriptionId,
        organizationName,
        resourceGroupName1,
    ],
    headerParameters: [accept, contentType],
    mediaType: "json",
    serializer,
};
const createRoleBindingOperationSpec = {
    path: "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Confluent/organizations/{organizationName}/access/default/createRoleBinding",
    httpMethod: "POST",
    responses: {
        200: {
            bodyMapper: RoleBindingRecord,
        },
        default: {
            bodyMapper: ResourceProviderDefaultErrorResponse,
        },
    },
    requestBody: body7,
    queryParameters: [apiVersion],
    urlParameters: [
        $host,
        subscriptionId,
        organizationName,
        resourceGroupName1,
    ],
    headerParameters: [accept, contentType],
    mediaType: "json",
    serializer,
};
const deleteRoleBindingOperationSpec = {
    path: "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Confluent/organizations/{organizationName}/access/default/deleteRoleBinding/{roleBindingId}",
    httpMethod: "DELETE",
    responses: {
        200: {},
        204: {},
        default: {
            bodyMapper: ResourceProviderDefaultErrorResponse,
        },
    },
    queryParameters: [apiVersion],
    urlParameters: [
        $host,
        subscriptionId,
        organizationName,
        resourceGroupName1,
        roleBindingId,
    ],
    headerParameters: [accept],
    serializer,
};
const listRoleBindingNameListOperationSpec = {
    path: "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Confluent/organizations/{organizationName}/access/default/listRoleBindingNameList",
    httpMethod: "POST",
    responses: {
        200: {
            bodyMapper: AccessRoleBindingNameListSuccessResponse,
        },
        default: {
            bodyMapper: ResourceProviderDefaultErrorResponse,
        },
    },
    requestBody: body3,
    queryParameters: [apiVersion],
    urlParameters: [
        $host,
        subscriptionId,
        organizationName,
        resourceGroupName1,
    ],
    headerParameters: [accept, contentType],
    mediaType: "json",
    serializer,
};

/*
 * Copyright (c) Microsoft Corporation.
 * Licensed under the MIT License.
 *
 * Code generated by Microsoft (R) AutoRest Code Generator.
 * Changes may cause incorrect behavior and will be lost if the code is regenerated.
 */
class ConfluentManagementClient extends coreClient__namespace.ServiceClient {
    constructor(credentials, subscriptionIdOrOptions, options) {
        var _a, _b, _c;
        if (credentials === undefined) {
            throw new Error("'credentials' cannot be null");
        }
        let subscriptionId;
        if (typeof subscriptionIdOrOptions === "string") {
            subscriptionId = subscriptionIdOrOptions;
        }
        else if (typeof subscriptionIdOrOptions === "object") {
            options = subscriptionIdOrOptions;
        }
        // Initializing default values for options
        if (!options) {
            options = {};
        }
        const defaults = {
            requestContentType: "application/json; charset=utf-8",
            credential: credentials,
        };
        const packageDetails = `azsdk-js-arm-confluent/3.1.0`;
        const userAgentPrefix = options.userAgentOptions && options.userAgentOptions.userAgentPrefix
            ? `${options.userAgentOptions.userAgentPrefix} ${packageDetails}`
            : `${packageDetails}`;
        const optionsWithDefaults = Object.assign(Object.assign(Object.assign({}, defaults), options), { userAgentOptions: {
                userAgentPrefix,
            }, endpoint: (_b = (_a = options.endpoint) !== null && _a !== void 0 ? _a : options.baseUri) !== null && _b !== void 0 ? _b : "https://management.azure.com" });
        super(optionsWithDefaults);
        let bearerTokenAuthenticationPolicyFound = false;
        if ((options === null || options === void 0 ? void 0 : options.pipeline) && options.pipeline.getOrderedPolicies().length > 0) {
            const pipelinePolicies = options.pipeline.getOrderedPolicies();
            bearerTokenAuthenticationPolicyFound = pipelinePolicies.some((pipelinePolicy) => pipelinePolicy.name ===
                coreRestPipeline__namespace.bearerTokenAuthenticationPolicyName);
        }
        if (!options ||
            !options.pipeline ||
            options.pipeline.getOrderedPolicies().length == 0 ||
            !bearerTokenAuthenticationPolicyFound) {
            this.pipeline.removePolicy({
                name: coreRestPipeline__namespace.bearerTokenAuthenticationPolicyName,
            });
            this.pipeline.addPolicy(coreRestPipeline__namespace.bearerTokenAuthenticationPolicy({
                credential: credentials,
                scopes: (_c = optionsWithDefaults.credentialScopes) !== null && _c !== void 0 ? _c : `${optionsWithDefaults.endpoint}/.default`,
                challengeCallbacks: {
                    authorizeRequestOnChallenge: coreClient__namespace.authorizeRequestOnClaimChallenge,
                },
            }));
        }
        // Parameter assignments
        this.subscriptionId = subscriptionId;
        // Assigning values to Constant parameters
        this.$host = options.$host || "https://management.azure.com";
        this.apiVersion = options.apiVersion || "2024-02-13";
        this.marketplaceAgreements = new MarketplaceAgreementsImpl(this);
        this.organizationOperations = new OrganizationOperationsImpl(this);
        this.organization = new OrganizationImpl(this);
        this.validations = new ValidationsImpl(this);
        this.access = new AccessImpl(this);
        this.addCustomApiVersionPolicy(options.apiVersion);
    }
    /** A function that adds a policy that sets the api-version (or equivalent) to reflect the library version. */
    addCustomApiVersionPolicy(apiVersion) {
        if (!apiVersion) {
            return;
        }
        const apiVersionPolicy = {
            name: "CustomApiVersionPolicy",
            sendRequest(request, next) {
                return tslib.__awaiter(this, void 0, void 0, function* () {
                    const param = request.url.split("?");
                    if (param.length > 1) {
                        const newParams = param[1].split("&").map((item) => {
                            if (item.indexOf("api-version") > -1) {
                                return "api-version=" + apiVersion;
                            }
                            else {
                                return item;
                            }
                        });
                        request.url = param[0] + "?" + newParams.join("&");
                    }
                    return next(request);
                });
            },
        };
        this.pipeline.addPolicy(apiVersionPolicy);
    }
}

exports.ConfluentManagementClient = ConfluentManagementClient;
exports.getContinuationToken = getContinuationToken;
