// @azure/arm-servicebus/dist-esm/test/servicebus_example.spec.js
/*
 * Copyright (c) Microsoft Corporation.
 * Licensed under the MIT License.
 *
 * Code generated by Microsoft (R) AutoRest Code Generator.
 * Changes may cause incorrect behavior and will be lost if the code is regenerated.
 */
import { __asyncValues, __awaiter } from "tslib";
import { env, Recorder, isPlaybackMode, } from "@azure-tools/test-recorder";
import { createTestCredential } from "@azure-tools/test-credential";
import { assert } from "chai";
import { ServiceBusManagementClient } from "../src/serviceBusManagementClient";
const replaceableVariables = {
    AZURE_CLIENT_ID: "azure_client_id",
    AZURE_CLIENT_SECRET: "azure_client_secret",
    AZURE_TENANT_ID: "88888888-8888-8888-8888-888888888888",
    SUBSCRIPTION_ID: "azure_subscription_id"
};
const recorderOptions = {
    envSetupForPlayback: replaceableVariables
};
export const testPollingOptions = {
    updateIntervalInMs: isPlaybackMode() ? 0 : undefined,
};
describe("ServiceBus test", () => {
    let recorder;
    let subscriptionId;
    let client;
    let location;
    let resourceGroup;
    let namespacesName;
    let authorizationRuleName;
    let queueName;
    let topicName;
    beforeEach(function () {
        return __awaiter(this, void 0, void 0, function* () {
            recorder = new Recorder(this.currentTest);
            yield recorder.start(recorderOptions);
            subscriptionId = env.SUBSCRIPTION_ID || '';
            // This is an example of how the environment variables are used
            const credential = createTestCredential();
            client = new ServiceBusManagementClient(credential, subscriptionId, recorder.configureClientOptions({}));
            location = "eastus";
            resourceGroup = "myjstest";
            namespacesName = "mynamespacexxx";
            authorizationRuleName = "myAuthoriztionRule";
            queueName = "myQueue";
            topicName = "mytopic";
        });
    });
    afterEach(function () {
        return __awaiter(this, void 0, void 0, function* () {
            yield recorder.stop();
        });
    });
    it("namespaces create test", function () {
        return __awaiter(this, void 0, void 0, function* () {
            const res = yield client.namespaces.beginCreateOrUpdateAndWait(resourceGroup, namespacesName, {
                sku: {
                    name: "Standard",
                    tier: "Standard",
                },
                location: location,
                tags: {
                    tag1: "value1",
                    tag2: "value2",
                }
            }, testPollingOptions);
            assert.equal(res.name, namespacesName);
        });
    });
    it("queues create test", function () {
        return __awaiter(this, void 0, void 0, function* () {
            const res = yield client.queues.createOrUpdate(resourceGroup, namespacesName, queueName, { enablePartitioning: true });
            assert.equal(res.name, queueName);
        });
    });
    it("queues get test", function () {
        return __awaiter(this, void 0, void 0, function* () {
            const res = yield client.queues.get(resourceGroup, namespacesName, queueName);
            assert.equal(res.name, queueName);
        });
    });
    it("queues list test", function () {
        var e_1, _a;
        return __awaiter(this, void 0, void 0, function* () {
            const resArray = new Array();
            try {
                for (var _b = __asyncValues(client.queues.listByNamespace(resourceGroup, namespacesName)), _c; _c = yield _b.next(), !_c.done;) {
                    let item = _c.value;
                    resArray.push(item);
                }
            }
            catch (e_1_1) { e_1 = { error: e_1_1 }; }
            finally {
                try {
                    if (_c && !_c.done && (_a = _b.return)) yield _a.call(_b);
                }
                finally { if (e_1) throw e_1.error; }
            }
            assert.equal(resArray.length, 1);
        });
    });
    it("topics create test", function () {
        return __awaiter(this, void 0, void 0, function* () {
            const res = yield client.topics.createOrUpdate(resourceGroup, namespacesName, topicName, {
                enableExpress: true
            });
            assert.equal(res.name, topicName);
        });
    });
    it("topics get test", function () {
        return __awaiter(this, void 0, void 0, function* () {
            const res = yield client.topics.get(resourceGroup, namespacesName, topicName);
            assert.equal(res.name, topicName);
        });
    });
    it("topics list test", function () {
        var e_2, _a;
        return __awaiter(this, void 0, void 0, function* () {
            const resArray = new Array();
            try {
                for (var _b = __asyncValues(client.topics.listByNamespace(resourceGroup, namespacesName)), _c; _c = yield _b.next(), !_c.done;) {
                    let item = _c.value;
                    resArray.push(item);
                }
            }
            catch (e_2_1) { e_2 = { error: e_2_1 }; }
            finally {
                try {
                    if (_c && !_c.done && (_a = _b.return)) yield _a.call(_b);
                }
                finally { if (e_2) throw e_2.error; }
            }
            assert.equal(resArray.length, 1);
        });
    });
    it("queues delete test", function () {
        var e_3, _a;
        return __awaiter(this, void 0, void 0, function* () {
            const res = yield client.queues.delete(resourceGroup, namespacesName, queueName);
            const resArray = new Array();
            try {
                for (var _b = __asyncValues(client.queues.listByNamespace(resourceGroup, namespacesName)), _c; _c = yield _b.next(), !_c.done;) {
                    let item = _c.value;
                    resArray.push(item);
                }
            }
            catch (e_3_1) { e_3 = { error: e_3_1 }; }
            finally {
                try {
                    if (_c && !_c.done && (_a = _b.return)) yield _a.call(_b);
                }
                finally { if (e_3) throw e_3.error; }
            }
            assert.equal(resArray.length, 0);
        });
    });
    it("topics delete test", function () {
        var e_4, _a;
        return __awaiter(this, void 0, void 0, function* () {
            const res = yield client.topics.delete(resourceGroup, namespacesName, topicName);
            const resArray = new Array();
            try {
                for (var _b = __asyncValues(client.topics.listByNamespace(resourceGroup, namespacesName)), _c; _c = yield _b.next(), !_c.done;) {
                    let item = _c.value;
                    resArray.push(item);
                }
            }
            catch (e_4_1) { e_4 = { error: e_4_1 }; }
            finally {
                try {
                    if (_c && !_c.done && (_a = _b.return)) yield _a.call(_b);
                }
                finally { if (e_4) throw e_4.error; }
            }
            assert.equal(resArray.length, 0);
        });
    });
    it("namespaces delete test", function () {
        return __awaiter(this, void 0, void 0, function* () {
            const res = yield client.namespaces.beginDeleteAndWait(resourceGroup, namespacesName, testPollingOptions);
        });
    });
});
