// @azure/arm-notificationhubs/dist-esm/test/notificationhubs_examples.js
/*
 * Copyright (c) Microsoft Corporation.
 * Licensed under the MIT License.
 *
 * Code generated by Microsoft (R) AutoRest Code Generator.
 * Changes may cause incorrect behavior and will be lost if the code is regenerated.
 */
import { __asyncValues, __awaiter } from "tslib";
import { env, Recorder, isPlaybackMode, } from "@azure-tools/test-recorder";
import { createTestCredential } from "@azure-tools/test-credential";
import { assert } from "chai";
import { NotificationHubsManagementClient } from "../src/notificationHubsManagementClient";
const replaceableVariables = {
    AZURE_CLIENT_ID: "azure_client_id",
    AZURE_CLIENT_SECRET: "azure_client_secret",
    AZURE_TENANT_ID: "88888888-8888-8888-8888-888888888888",
    SUBSCRIPTION_ID: "azure_subscription_id"
};
const recorderOptions = {
    envSetupForPlayback: replaceableVariables
};
export const testPollingOptions = {
    updateIntervalInMs: isPlaybackMode() ? 0 : undefined,
};
describe("NotificationHubs test", () => {
    let recorder;
    let subscriptionId;
    let client;
    let location;
    let resourceGroup;
    let nameSpaceName;
    let notificationhubsName;
    beforeEach(function () {
        return __awaiter(this, void 0, void 0, function* () {
            recorder = new Recorder(this.currentTest);
            yield recorder.start(recorderOptions);
            subscriptionId = env.SUBSCRIPTION_ID || '';
            // This is an example of how the environment variables are used
            const credential = createTestCredential();
            client = new NotificationHubsManagementClient(credential, subscriptionId, recorder.configureClientOptions({}));
            location = "eastus";
            resourceGroup = "myjstest";
            nameSpaceName = "mynamespacexxx";
            notificationhubsName = "mynotificationhubsxxx";
        });
    });
    afterEach(function () {
        return __awaiter(this, void 0, void 0, function* () {
            yield recorder.stop();
        });
    });
    it("namespaces create test", function () {
        return __awaiter(this, void 0, void 0, function* () {
            const res = yield client.namespaces.createOrUpdate(resourceGroup, nameSpaceName, { location: location });
            assert.equal(res.name, nameSpaceName);
        });
    });
    it("namespaces get test", function () {
        return __awaiter(this, void 0, void 0, function* () {
            const res = yield client.namespaces.get(resourceGroup, nameSpaceName);
            assert.equal(res.name, nameSpaceName);
        });
    });
    it("notificationHubs create test", function () {
        return __awaiter(this, void 0, void 0, function* () {
            const res = yield client.notificationHubs.createOrUpdate(resourceGroup, nameSpaceName, notificationhubsName, { location: location });
            assert.equal(res.name, notificationhubsName);
        });
    });
    it("notificationHubs get test", function () {
        return __awaiter(this, void 0, void 0, function* () {
            const res = yield client.notificationHubs.get(resourceGroup, nameSpaceName, notificationhubsName);
            assert.equal(res.name, notificationhubsName);
        });
    });
    it("notificationHubs list test", function () {
        var e_1, _a;
        return __awaiter(this, void 0, void 0, function* () {
            const resArray = new Array();
            try {
                for (var _b = __asyncValues(client.notificationHubs.list(resourceGroup, nameSpaceName)), _c; _c = yield _b.next(), !_c.done;) {
                    let item = _c.value;
                    resArray.push(item);
                }
            }
            catch (e_1_1) { e_1 = { error: e_1_1 }; }
            finally {
                try {
                    if (_c && !_c.done && (_a = _b.return)) yield _a.call(_b);
                }
                finally { if (e_1) throw e_1.error; }
            }
            assert.equal(resArray.length, 1);
        });
    });
    it("notificationHubs delete test", function () {
        var e_2, _a;
        return __awaiter(this, void 0, void 0, function* () {
            const res = yield client.notificationHubs.delete(resourceGroup, nameSpaceName, notificationhubsName);
            const resArray = new Array();
            try {
                for (var _b = __asyncValues(client.notificationHubs.list(resourceGroup, nameSpaceName)), _c; _c = yield _b.next(), !_c.done;) {
                    let item = _c.value;
                    resArray.push(item);
                }
            }
            catch (e_2_1) { e_2 = { error: e_2_1 }; }
            finally {
                try {
                    if (_c && !_c.done && (_a = _b.return)) yield _a.call(_b);
                }
                finally { if (e_2) throw e_2.error; }
            }
            assert.equal(resArray.length, 0);
        });
    });
    it("namespaces delete test", function () {
        return __awaiter(this, void 0, void 0, function* () {
            const res = yield client.namespaces.beginDeleteAndWait(resourceGroup, nameSpaceName);
        });
    });
});
