// @aws-sdk/client-redshiftserverless/dist-cjs/models/models_0.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.DeleteResourcePolicyResponse = exports.DeleteResourcePolicyRequest = exports.DeleteNamespaceResponse = exports.DeleteNamespaceRequest = exports.UpdateEndpointAccessResponse = exports.UpdateEndpointAccessRequest = exports.ListEndpointAccessResponse = exports.ListEndpointAccessRequest = exports.GetEndpointAccessResponse = exports.GetEndpointAccessRequest = exports.DeleteEndpointAccessResponse = exports.DeleteEndpointAccessRequest = exports.InsufficientCapacityException = exports.CreateWorkgroupResponse = exports.Workgroup = exports.WorkgroupStatus = exports.Endpoint = exports.CreateWorkgroupRequest = exports.CreateUsageLimitResponse = exports.UsageLimit = exports.CreateUsageLimitRequest = exports.UsageLimitUsageType = exports.UsageLimitPeriod = exports.UsageLimitBreachAction = exports.CreateSnapshotResponse = exports.CreateSnapshotRequest = exports.TooManyTagsException = exports.CreateNamespaceResponse = exports.Namespace = exports.NamespaceStatus = exports.CreateNamespaceRequest = exports.Tag = exports.LogExport = exports.CreateEndpointAccessResponse = exports.EndpointAccess = exports.VpcSecurityGroupMembership = exports.VpcEndpoint = exports.NetworkInterface = exports.CreateEndpointAccessRequest = exports.ValidationException = exports.ServiceQuotaExceededException = exports.ResourceNotFoundException = exports.InternalServerException = exports.ConvertRecoveryPointToSnapshotResponse = exports.Snapshot = exports.SnapshotStatus = exports.ConvertRecoveryPointToSnapshotRequest = exports.ConflictException = exports.ConfigParameter = exports.AccessDeniedException = void 0;
exports.UntagResourceResponse = exports.UntagResourceRequest = exports.TagResourceResponse = exports.TagResourceRequest = exports.UpdateSnapshotResponse = exports.UpdateSnapshotRequest = exports.RestoreFromSnapshotResponse = exports.RestoreFromSnapshotRequest = exports.RestoreFromRecoveryPointResponse = exports.RestoreFromRecoveryPointRequest = exports.PutResourcePolicyResponse = exports.PutResourcePolicyRequest = exports.UpdateNamespaceResponse = exports.UpdateNamespaceRequest = exports.ListWorkgroupsResponse = exports.ListWorkgroupsRequest = exports.ListUsageLimitsResponse = exports.ListUsageLimitsRequest = exports.ThrottlingException = exports.ListTagsForResourceResponse = exports.ListTagsForResourceRequest = exports.ListSnapshotsResponse = exports.ListSnapshotsRequest = exports.ListRecoveryPointsResponse = exports.ListRecoveryPointsRequest = exports.ListNamespacesResponse = exports.ListNamespacesRequest = exports.InvalidPaginationException = exports.GetWorkgroupResponse = exports.GetWorkgroupRequest = exports.GetUsageLimitResponse = exports.GetUsageLimitRequest = exports.GetSnapshotResponse = exports.GetSnapshotRequest = exports.GetResourcePolicyResponse = exports.ResourcePolicy = exports.GetResourcePolicyRequest = exports.GetRecoveryPointResponse = exports.RecoveryPoint = exports.GetRecoveryPointRequest = exports.GetNamespaceResponse = exports.GetNamespaceRequest = exports.GetCredentialsResponse = exports.GetCredentialsRequest = exports.DeleteWorkgroupResponse = exports.DeleteWorkgroupRequest = exports.DeleteUsageLimitResponse = exports.DeleteUsageLimitRequest = exports.DeleteSnapshotResponse = exports.DeleteSnapshotRequest = void 0;
exports.UpdateWorkgroupResponse = exports.UpdateWorkgroupRequest = exports.UpdateUsageLimitResponse = exports.UpdateUsageLimitRequest = void 0;
const smithy_client_1 = require("@aws-sdk/smithy-client");
const RedshiftServerlessServiceException_1 = require("./RedshiftServerlessServiceException");
class AccessDeniedException extends RedshiftServerlessServiceException_1.RedshiftServerlessServiceException {
    constructor(opts) {
        super({
            name: "AccessDeniedException",
            $fault: "client",
            ...opts,
        });
        this.name = "AccessDeniedException";
        this.$fault = "client";
        Object.setPrototypeOf(this, AccessDeniedException.prototype);
        this.code = opts.code;
    }
}
exports.AccessDeniedException = AccessDeniedException;
var ConfigParameter;
(function (ConfigParameter) {
    ConfigParameter.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(ConfigParameter = exports.ConfigParameter || (exports.ConfigParameter = {}));
class ConflictException extends RedshiftServerlessServiceException_1.RedshiftServerlessServiceException {
    constructor(opts) {
        super({
            name: "ConflictException",
            $fault: "client",
            ...opts,
        });
        this.name = "ConflictException";
        this.$fault = "client";
        Object.setPrototypeOf(this, ConflictException.prototype);
    }
}
exports.ConflictException = ConflictException;
var ConvertRecoveryPointToSnapshotRequest;
(function (ConvertRecoveryPointToSnapshotRequest) {
    ConvertRecoveryPointToSnapshotRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(ConvertRecoveryPointToSnapshotRequest = exports.ConvertRecoveryPointToSnapshotRequest || (exports.ConvertRecoveryPointToSnapshotRequest = {}));
var SnapshotStatus;
(function (SnapshotStatus) {
    SnapshotStatus["AVAILABLE"] = "AVAILABLE";
    SnapshotStatus["CANCELLED"] = "CANCELLED";
    SnapshotStatus["COPYING"] = "COPYING";
    SnapshotStatus["CREATING"] = "CREATING";
    SnapshotStatus["DELETED"] = "DELETED";
    SnapshotStatus["FAILED"] = "FAILED";
})(SnapshotStatus = exports.SnapshotStatus || (exports.SnapshotStatus = {}));
var Snapshot;
(function (Snapshot) {
    Snapshot.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(Snapshot = exports.Snapshot || (exports.Snapshot = {}));
var ConvertRecoveryPointToSnapshotResponse;
(function (ConvertRecoveryPointToSnapshotResponse) {
    ConvertRecoveryPointToSnapshotResponse.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(ConvertRecoveryPointToSnapshotResponse = exports.ConvertRecoveryPointToSnapshotResponse || (exports.ConvertRecoveryPointToSnapshotResponse = {}));
class InternalServerException extends RedshiftServerlessServiceException_1.RedshiftServerlessServiceException {
    constructor(opts) {
        super({
            name: "InternalServerException",
            $fault: "server",
            ...opts,
        });
        this.name = "InternalServerException";
        this.$fault = "server";
        this.$retryable = {};
        Object.setPrototypeOf(this, InternalServerException.prototype);
    }
}
exports.InternalServerException = InternalServerException;
class ResourceNotFoundException extends RedshiftServerlessServiceException_1.RedshiftServerlessServiceException {
    constructor(opts) {
        super({
            name: "ResourceNotFoundException",
            $fault: "client",
            ...opts,
        });
        this.name = "ResourceNotFoundException";
        this.$fault = "client";
        Object.setPrototypeOf(this, ResourceNotFoundException.prototype);
        this.resourceName = opts.resourceName;
    }
}
exports.ResourceNotFoundException = ResourceNotFoundException;
class ServiceQuotaExceededException extends RedshiftServerlessServiceException_1.RedshiftServerlessServiceException {
    constructor(opts) {
        super({
            name: "ServiceQuotaExceededException",
            $fault: "client",
            ...opts,
        });
        this.name = "ServiceQuotaExceededException";
        this.$fault = "client";
        Object.setPrototypeOf(this, ServiceQuotaExceededException.prototype);
    }
}
exports.ServiceQuotaExceededException = ServiceQuotaExceededException;
class ValidationException extends RedshiftServerlessServiceException_1.RedshiftServerlessServiceException {
    constructor(opts) {
        super({
            name: "ValidationException",
            $fault: "client",
            ...opts,
        });
        this.name = "ValidationException";
        this.$fault = "client";
        Object.setPrototypeOf(this, ValidationException.prototype);
    }
}
exports.ValidationException = ValidationException;
var CreateEndpointAccessRequest;
(function (CreateEndpointAccessRequest) {
    CreateEndpointAccessRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(CreateEndpointAccessRequest = exports.CreateEndpointAccessRequest || (exports.CreateEndpointAccessRequest = {}));
var NetworkInterface;
(function (NetworkInterface) {
    NetworkInterface.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(NetworkInterface = exports.NetworkInterface || (exports.NetworkInterface = {}));
var VpcEndpoint;
(function (VpcEndpoint) {
    VpcEndpoint.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(VpcEndpoint = exports.VpcEndpoint || (exports.VpcEndpoint = {}));
var VpcSecurityGroupMembership;
(function (VpcSecurityGroupMembership) {
    VpcSecurityGroupMembership.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(VpcSecurityGroupMembership = exports.VpcSecurityGroupMembership || (exports.VpcSecurityGroupMembership = {}));
var EndpointAccess;
(function (EndpointAccess) {
    EndpointAccess.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(EndpointAccess = exports.EndpointAccess || (exports.EndpointAccess = {}));
var CreateEndpointAccessResponse;
(function (CreateEndpointAccessResponse) {
    CreateEndpointAccessResponse.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(CreateEndpointAccessResponse = exports.CreateEndpointAccessResponse || (exports.CreateEndpointAccessResponse = {}));
var LogExport;
(function (LogExport) {
    LogExport["CONNECTION_LOG"] = "connectionlog";
    LogExport["USER_ACTIVITY_LOG"] = "useractivitylog";
    LogExport["USER_LOG"] = "userlog";
})(LogExport = exports.LogExport || (exports.LogExport = {}));
var Tag;
(function (Tag) {
    Tag.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(Tag = exports.Tag || (exports.Tag = {}));
var CreateNamespaceRequest;
(function (CreateNamespaceRequest) {
    CreateNamespaceRequest.filterSensitiveLog = (obj) => ({
        ...obj,
        ...(obj.adminUsername && { adminUsername: smithy_client_1.SENSITIVE_STRING }),
        ...(obj.adminUserPassword && { adminUserPassword: smithy_client_1.SENSITIVE_STRING }),
    });
})(CreateNamespaceRequest = exports.CreateNamespaceRequest || (exports.CreateNamespaceRequest = {}));
var NamespaceStatus;
(function (NamespaceStatus) {
    NamespaceStatus["AVAILABLE"] = "AVAILABLE";
    NamespaceStatus["DELETING"] = "DELETING";
    NamespaceStatus["MODIFYING"] = "MODIFYING";
})(NamespaceStatus = exports.NamespaceStatus || (exports.NamespaceStatus = {}));
var Namespace;
(function (Namespace) {
    Namespace.filterSensitiveLog = (obj) => ({
        ...obj,
        ...(obj.adminUsername && { adminUsername: smithy_client_1.SENSITIVE_STRING }),
    });
})(Namespace = exports.Namespace || (exports.Namespace = {}));
var CreateNamespaceResponse;
(function (CreateNamespaceResponse) {
    CreateNamespaceResponse.filterSensitiveLog = (obj) => ({
        ...obj,
        ...(obj.namespace && { namespace: Namespace.filterSensitiveLog(obj.namespace) }),
    });
})(CreateNamespaceResponse = exports.CreateNamespaceResponse || (exports.CreateNamespaceResponse = {}));
class TooManyTagsException extends RedshiftServerlessServiceException_1.RedshiftServerlessServiceException {
    constructor(opts) {
        super({
            name: "TooManyTagsException",
            $fault: "client",
            ...opts,
        });
        this.name = "TooManyTagsException";
        this.$fault = "client";
        Object.setPrototypeOf(this, TooManyTagsException.prototype);
        this.resourceName = opts.resourceName;
    }
}
exports.TooManyTagsException = TooManyTagsException;
var CreateSnapshotRequest;
(function (CreateSnapshotRequest) {
    CreateSnapshotRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(CreateSnapshotRequest = exports.CreateSnapshotRequest || (exports.CreateSnapshotRequest = {}));
var CreateSnapshotResponse;
(function (CreateSnapshotResponse) {
    CreateSnapshotResponse.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(CreateSnapshotResponse = exports.CreateSnapshotResponse || (exports.CreateSnapshotResponse = {}));
var UsageLimitBreachAction;
(function (UsageLimitBreachAction) {
    UsageLimitBreachAction["DEACTIVATE"] = "deactivate";
    UsageLimitBreachAction["EMIT_METRIC"] = "emit-metric";
    UsageLimitBreachAction["LOG"] = "log";
})(UsageLimitBreachAction = exports.UsageLimitBreachAction || (exports.UsageLimitBreachAction = {}));
var UsageLimitPeriod;
(function (UsageLimitPeriod) {
    UsageLimitPeriod["DAILY"] = "daily";
    UsageLimitPeriod["MONTHLY"] = "monthly";
    UsageLimitPeriod["WEEKLY"] = "weekly";
})(UsageLimitPeriod = exports.UsageLimitPeriod || (exports.UsageLimitPeriod = {}));
var UsageLimitUsageType;
(function (UsageLimitUsageType) {
    UsageLimitUsageType["CROSS_REGION_DATASHARING"] = "cross-region-datasharing";
    UsageLimitUsageType["SERVERLESS_COMPUTE"] = "serverless-compute";
})(UsageLimitUsageType = exports.UsageLimitUsageType || (exports.UsageLimitUsageType = {}));
var CreateUsageLimitRequest;
(function (CreateUsageLimitRequest) {
    CreateUsageLimitRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(CreateUsageLimitRequest = exports.CreateUsageLimitRequest || (exports.CreateUsageLimitRequest = {}));
var UsageLimit;
(function (UsageLimit) {
    UsageLimit.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(UsageLimit = exports.UsageLimit || (exports.UsageLimit = {}));
var CreateUsageLimitResponse;
(function (CreateUsageLimitResponse) {
    CreateUsageLimitResponse.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(CreateUsageLimitResponse = exports.CreateUsageLimitResponse || (exports.CreateUsageLimitResponse = {}));
var CreateWorkgroupRequest;
(function (CreateWorkgroupRequest) {
    CreateWorkgroupRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(CreateWorkgroupRequest = exports.CreateWorkgroupRequest || (exports.CreateWorkgroupRequest = {}));
var Endpoint;
(function (Endpoint) {
    Endpoint.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(Endpoint = exports.Endpoint || (exports.Endpoint = {}));
var WorkgroupStatus;
(function (WorkgroupStatus) {
    WorkgroupStatus["AVAILABLE"] = "AVAILABLE";
    WorkgroupStatus["CREATING"] = "CREATING";
    WorkgroupStatus["DELETING"] = "DELETING";
    WorkgroupStatus["MODIFYING"] = "MODIFYING";
})(WorkgroupStatus = exports.WorkgroupStatus || (exports.WorkgroupStatus = {}));
var Workgroup;
(function (Workgroup) {
    Workgroup.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(Workgroup = exports.Workgroup || (exports.Workgroup = {}));
var CreateWorkgroupResponse;
(function (CreateWorkgroupResponse) {
    CreateWorkgroupResponse.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(CreateWorkgroupResponse = exports.CreateWorkgroupResponse || (exports.CreateWorkgroupResponse = {}));
class InsufficientCapacityException extends RedshiftServerlessServiceException_1.RedshiftServerlessServiceException {
    constructor(opts) {
        super({
            name: "InsufficientCapacityException",
            $fault: "client",
            ...opts,
        });
        this.name = "InsufficientCapacityException";
        this.$fault = "client";
        this.$retryable = {};
        Object.setPrototypeOf(this, InsufficientCapacityException.prototype);
    }
}
exports.InsufficientCapacityException = InsufficientCapacityException;
var DeleteEndpointAccessRequest;
(function (DeleteEndpointAccessRequest) {
    DeleteEndpointAccessRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(DeleteEndpointAccessRequest = exports.DeleteEndpointAccessRequest || (exports.DeleteEndpointAccessRequest = {}));
var DeleteEndpointAccessResponse;
(function (DeleteEndpointAccessResponse) {
    DeleteEndpointAccessResponse.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(DeleteEndpointAccessResponse = exports.DeleteEndpointAccessResponse || (exports.DeleteEndpointAccessResponse = {}));
var GetEndpointAccessRequest;
(function (GetEndpointAccessRequest) {
    GetEndpointAccessRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(GetEndpointAccessRequest = exports.GetEndpointAccessRequest || (exports.GetEndpointAccessRequest = {}));
var GetEndpointAccessResponse;
(function (GetEndpointAccessResponse) {
    GetEndpointAccessResponse.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(GetEndpointAccessResponse = exports.GetEndpointAccessResponse || (exports.GetEndpointAccessResponse = {}));
var ListEndpointAccessRequest;
(function (ListEndpointAccessRequest) {
    ListEndpointAccessRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(ListEndpointAccessRequest = exports.ListEndpointAccessRequest || (exports.ListEndpointAccessRequest = {}));
var ListEndpointAccessResponse;
(function (ListEndpointAccessResponse) {
    ListEndpointAccessResponse.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(ListEndpointAccessResponse = exports.ListEndpointAccessResponse || (exports.ListEndpointAccessResponse = {}));
var UpdateEndpointAccessRequest;
(function (UpdateEndpointAccessRequest) {
    UpdateEndpointAccessRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(UpdateEndpointAccessRequest = exports.UpdateEndpointAccessRequest || (exports.UpdateEndpointAccessRequest = {}));
var UpdateEndpointAccessResponse;
(function (UpdateEndpointAccessResponse) {
    UpdateEndpointAccessResponse.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(UpdateEndpointAccessResponse = exports.UpdateEndpointAccessResponse || (exports.UpdateEndpointAccessResponse = {}));
var DeleteNamespaceRequest;
(function (DeleteNamespaceRequest) {
    DeleteNamespaceRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(DeleteNamespaceRequest = exports.DeleteNamespaceRequest || (exports.DeleteNamespaceRequest = {}));
var DeleteNamespaceResponse;
(function (DeleteNamespaceResponse) {
    DeleteNamespaceResponse.filterSensitiveLog = (obj) => ({
        ...obj,
        ...(obj.namespace && { namespace: Namespace.filterSensitiveLog(obj.namespace) }),
    });
})(DeleteNamespaceResponse = exports.DeleteNamespaceResponse || (exports.DeleteNamespaceResponse = {}));
var DeleteResourcePolicyRequest;
(function (DeleteResourcePolicyRequest) {
    DeleteResourcePolicyRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(DeleteResourcePolicyRequest = exports.DeleteResourcePolicyRequest || (exports.DeleteResourcePolicyRequest = {}));
var DeleteResourcePolicyResponse;
(function (DeleteResourcePolicyResponse) {
    DeleteResourcePolicyResponse.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(DeleteResourcePolicyResponse = exports.DeleteResourcePolicyResponse || (exports.DeleteResourcePolicyResponse = {}));
var DeleteSnapshotRequest;
(function (DeleteSnapshotRequest) {
    DeleteSnapshotRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(DeleteSnapshotRequest = exports.DeleteSnapshotRequest || (exports.DeleteSnapshotRequest = {}));
var DeleteSnapshotResponse;
(function (DeleteSnapshotResponse) {
    DeleteSnapshotResponse.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(DeleteSnapshotResponse = exports.DeleteSnapshotResponse || (exports.DeleteSnapshotResponse = {}));
var DeleteUsageLimitRequest;
(function (DeleteUsageLimitRequest) {
    DeleteUsageLimitRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(DeleteUsageLimitRequest = exports.DeleteUsageLimitRequest || (exports.DeleteUsageLimitRequest = {}));
var DeleteUsageLimitResponse;
(function (DeleteUsageLimitResponse) {
    DeleteUsageLimitResponse.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(DeleteUsageLimitResponse = exports.DeleteUsageLimitResponse || (exports.DeleteUsageLimitResponse = {}));
var DeleteWorkgroupRequest;
(function (DeleteWorkgroupRequest) {
    DeleteWorkgroupRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(DeleteWorkgroupRequest = exports.DeleteWorkgroupRequest || (exports.DeleteWorkgroupRequest = {}));
var DeleteWorkgroupResponse;
(function (DeleteWorkgroupResponse) {
    DeleteWorkgroupResponse.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(DeleteWorkgroupResponse = exports.DeleteWorkgroupResponse || (exports.DeleteWorkgroupResponse = {}));
var GetCredentialsRequest;
(function (GetCredentialsRequest) {
    GetCredentialsRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(GetCredentialsRequest = exports.GetCredentialsRequest || (exports.GetCredentialsRequest = {}));
var GetCredentialsResponse;
(function (GetCredentialsResponse) {
    GetCredentialsResponse.filterSensitiveLog = (obj) => ({
        ...obj,
        ...(obj.dbUser && { dbUser: smithy_client_1.SENSITIVE_STRING }),
        ...(obj.dbPassword && { dbPassword: smithy_client_1.SENSITIVE_STRING }),
    });
})(GetCredentialsResponse = exports.GetCredentialsResponse || (exports.GetCredentialsResponse = {}));
var GetNamespaceRequest;
(function (GetNamespaceRequest) {
    GetNamespaceRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(GetNamespaceRequest = exports.GetNamespaceRequest || (exports.GetNamespaceRequest = {}));
var GetNamespaceResponse;
(function (GetNamespaceResponse) {
    GetNamespaceResponse.filterSensitiveLog = (obj) => ({
        ...obj,
        ...(obj.namespace && { namespace: Namespace.filterSensitiveLog(obj.namespace) }),
    });
})(GetNamespaceResponse = exports.GetNamespaceResponse || (exports.GetNamespaceResponse = {}));
var GetRecoveryPointRequest;
(function (GetRecoveryPointRequest) {
    GetRecoveryPointRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(GetRecoveryPointRequest = exports.GetRecoveryPointRequest || (exports.GetRecoveryPointRequest = {}));
var RecoveryPoint;
(function (RecoveryPoint) {
    RecoveryPoint.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(RecoveryPoint = exports.RecoveryPoint || (exports.RecoveryPoint = {}));
var GetRecoveryPointResponse;
(function (GetRecoveryPointResponse) {
    GetRecoveryPointResponse.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(GetRecoveryPointResponse = exports.GetRecoveryPointResponse || (exports.GetRecoveryPointResponse = {}));
var GetResourcePolicyRequest;
(function (GetResourcePolicyRequest) {
    GetResourcePolicyRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(GetResourcePolicyRequest = exports.GetResourcePolicyRequest || (exports.GetResourcePolicyRequest = {}));
var ResourcePolicy;
(function (ResourcePolicy) {
    ResourcePolicy.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(ResourcePolicy = exports.ResourcePolicy || (exports.ResourcePolicy = {}));
var GetResourcePolicyResponse;
(function (GetResourcePolicyResponse) {
    GetResourcePolicyResponse.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(GetResourcePolicyResponse = exports.GetResourcePolicyResponse || (exports.GetResourcePolicyResponse = {}));
var GetSnapshotRequest;
(function (GetSnapshotRequest) {
    GetSnapshotRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(GetSnapshotRequest = exports.GetSnapshotRequest || (exports.GetSnapshotRequest = {}));
var GetSnapshotResponse;
(function (GetSnapshotResponse) {
    GetSnapshotResponse.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(GetSnapshotResponse = exports.GetSnapshotResponse || (exports.GetSnapshotResponse = {}));
var GetUsageLimitRequest;
(function (GetUsageLimitRequest) {
    GetUsageLimitRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(GetUsageLimitRequest = exports.GetUsageLimitRequest || (exports.GetUsageLimitRequest = {}));
var GetUsageLimitResponse;
(function (GetUsageLimitResponse) {
    GetUsageLimitResponse.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(GetUsageLimitResponse = exports.GetUsageLimitResponse || (exports.GetUsageLimitResponse = {}));
var GetWorkgroupRequest;
(function (GetWorkgroupRequest) {
    GetWorkgroupRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(GetWorkgroupRequest = exports.GetWorkgroupRequest || (exports.GetWorkgroupRequest = {}));
var GetWorkgroupResponse;
(function (GetWorkgroupResponse) {
    GetWorkgroupResponse.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(GetWorkgroupResponse = exports.GetWorkgroupResponse || (exports.GetWorkgroupResponse = {}));
class InvalidPaginationException extends RedshiftServerlessServiceException_1.RedshiftServerlessServiceException {
    constructor(opts) {
        super({
            name: "InvalidPaginationException",
            $fault: "client",
            ...opts,
        });
        this.name = "InvalidPaginationException";
        this.$fault = "client";
        Object.setPrototypeOf(this, InvalidPaginationException.prototype);
    }
}
exports.InvalidPaginationException = InvalidPaginationException;
var ListNamespacesRequest;
(function (ListNamespacesRequest) {
    ListNamespacesRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(ListNamespacesRequest = exports.ListNamespacesRequest || (exports.ListNamespacesRequest = {}));
var ListNamespacesResponse;
(function (ListNamespacesResponse) {
    ListNamespacesResponse.filterSensitiveLog = (obj) => ({
        ...obj,
        ...(obj.namespaces && { namespaces: obj.namespaces.map((item) => Namespace.filterSensitiveLog(item)) }),
    });
})(ListNamespacesResponse = exports.ListNamespacesResponse || (exports.ListNamespacesResponse = {}));
var ListRecoveryPointsRequest;
(function (ListRecoveryPointsRequest) {
    ListRecoveryPointsRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(ListRecoveryPointsRequest = exports.ListRecoveryPointsRequest || (exports.ListRecoveryPointsRequest = {}));
var ListRecoveryPointsResponse;
(function (ListRecoveryPointsResponse) {
    ListRecoveryPointsResponse.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(ListRecoveryPointsResponse = exports.ListRecoveryPointsResponse || (exports.ListRecoveryPointsResponse = {}));
var ListSnapshotsRequest;
(function (ListSnapshotsRequest) {
    ListSnapshotsRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(ListSnapshotsRequest = exports.ListSnapshotsRequest || (exports.ListSnapshotsRequest = {}));
var ListSnapshotsResponse;
(function (ListSnapshotsResponse) {
    ListSnapshotsResponse.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(ListSnapshotsResponse = exports.ListSnapshotsResponse || (exports.ListSnapshotsResponse = {}));
var ListTagsForResourceRequest;
(function (ListTagsForResourceRequest) {
    ListTagsForResourceRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(ListTagsForResourceRequest = exports.ListTagsForResourceRequest || (exports.ListTagsForResourceRequest = {}));
var ListTagsForResourceResponse;
(function (ListTagsForResourceResponse) {
    ListTagsForResourceResponse.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(ListTagsForResourceResponse = exports.ListTagsForResourceResponse || (exports.ListTagsForResourceResponse = {}));
class ThrottlingException extends RedshiftServerlessServiceException_1.RedshiftServerlessServiceException {
    constructor(opts) {
        super({
            name: "ThrottlingException",
            $fault: "client",
            ...opts,
        });
        this.name = "ThrottlingException";
        this.$fault = "client";
        this.$retryable = {};
        Object.setPrototypeOf(this, ThrottlingException.prototype);
        this.code = opts.code;
    }
}
exports.ThrottlingException = ThrottlingException;
var ListUsageLimitsRequest;
(function (ListUsageLimitsRequest) {
    ListUsageLimitsRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(ListUsageLimitsRequest = exports.ListUsageLimitsRequest || (exports.ListUsageLimitsRequest = {}));
var ListUsageLimitsResponse;
(function (ListUsageLimitsResponse) {
    ListUsageLimitsResponse.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(ListUsageLimitsResponse = exports.ListUsageLimitsResponse || (exports.ListUsageLimitsResponse = {}));
var ListWorkgroupsRequest;
(function (ListWorkgroupsRequest) {
    ListWorkgroupsRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(ListWorkgroupsRequest = exports.ListWorkgroupsRequest || (exports.ListWorkgroupsRequest = {}));
var ListWorkgroupsResponse;
(function (ListWorkgroupsResponse) {
    ListWorkgroupsResponse.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(ListWorkgroupsResponse = exports.ListWorkgroupsResponse || (exports.ListWorkgroupsResponse = {}));
var UpdateNamespaceRequest;
(function (UpdateNamespaceRequest) {
    UpdateNamespaceRequest.filterSensitiveLog = (obj) => ({
        ...obj,
        ...(obj.adminUserPassword && { adminUserPassword: smithy_client_1.SENSITIVE_STRING }),
        ...(obj.adminUsername && { adminUsername: smithy_client_1.SENSITIVE_STRING }),
    });
})(UpdateNamespaceRequest = exports.UpdateNamespaceRequest || (exports.UpdateNamespaceRequest = {}));
var UpdateNamespaceResponse;
(function (UpdateNamespaceResponse) {
    UpdateNamespaceResponse.filterSensitiveLog = (obj) => ({
        ...obj,
        ...(obj.namespace && { namespace: Namespace.filterSensitiveLog(obj.namespace) }),
    });
})(UpdateNamespaceResponse = exports.UpdateNamespaceResponse || (exports.UpdateNamespaceResponse = {}));
var PutResourcePolicyRequest;
(function (PutResourcePolicyRequest) {
    PutResourcePolicyRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(PutResourcePolicyRequest = exports.PutResourcePolicyRequest || (exports.PutResourcePolicyRequest = {}));
var PutResourcePolicyResponse;
(function (PutResourcePolicyResponse) {
    PutResourcePolicyResponse.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(PutResourcePolicyResponse = exports.PutResourcePolicyResponse || (exports.PutResourcePolicyResponse = {}));
var RestoreFromRecoveryPointRequest;
(function (RestoreFromRecoveryPointRequest) {
    RestoreFromRecoveryPointRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(RestoreFromRecoveryPointRequest = exports.RestoreFromRecoveryPointRequest || (exports.RestoreFromRecoveryPointRequest = {}));
var RestoreFromRecoveryPointResponse;
(function (RestoreFromRecoveryPointResponse) {
    RestoreFromRecoveryPointResponse.filterSensitiveLog = (obj) => ({
        ...obj,
        ...(obj.namespace && { namespace: Namespace.filterSensitiveLog(obj.namespace) }),
    });
})(RestoreFromRecoveryPointResponse = exports.RestoreFromRecoveryPointResponse || (exports.RestoreFromRecoveryPointResponse = {}));
var RestoreFromSnapshotRequest;
(function (RestoreFromSnapshotRequest) {
    RestoreFromSnapshotRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(RestoreFromSnapshotRequest = exports.RestoreFromSnapshotRequest || (exports.RestoreFromSnapshotRequest = {}));
var RestoreFromSnapshotResponse;
(function (RestoreFromSnapshotResponse) {
    RestoreFromSnapshotResponse.filterSensitiveLog = (obj) => ({
        ...obj,
        ...(obj.namespace && { namespace: Namespace.filterSensitiveLog(obj.namespace) }),
    });
})(RestoreFromSnapshotResponse = exports.RestoreFromSnapshotResponse || (exports.RestoreFromSnapshotResponse = {}));
var UpdateSnapshotRequest;
(function (UpdateSnapshotRequest) {
    UpdateSnapshotRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(UpdateSnapshotRequest = exports.UpdateSnapshotRequest || (exports.UpdateSnapshotRequest = {}));
var UpdateSnapshotResponse;
(function (UpdateSnapshotResponse) {
    UpdateSnapshotResponse.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(UpdateSnapshotResponse = exports.UpdateSnapshotResponse || (exports.UpdateSnapshotResponse = {}));
var TagResourceRequest;
(function (TagResourceRequest) {
    TagResourceRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(TagResourceRequest = exports.TagResourceRequest || (exports.TagResourceRequest = {}));
var TagResourceResponse;
(function (TagResourceResponse) {
    TagResourceResponse.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(TagResourceResponse = exports.TagResourceResponse || (exports.TagResourceResponse = {}));
var UntagResourceRequest;
(function (UntagResourceRequest) {
    UntagResourceRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(UntagResourceRequest = exports.UntagResourceRequest || (exports.UntagResourceRequest = {}));
var UntagResourceResponse;
(function (UntagResourceResponse) {
    UntagResourceResponse.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(UntagResourceResponse = exports.UntagResourceResponse || (exports.UntagResourceResponse = {}));
var UpdateUsageLimitRequest;
(function (UpdateUsageLimitRequest) {
    UpdateUsageLimitRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(UpdateUsageLimitRequest = exports.UpdateUsageLimitRequest || (exports.UpdateUsageLimitRequest = {}));
var UpdateUsageLimitResponse;
(function (UpdateUsageLimitResponse) {
    UpdateUsageLimitResponse.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(UpdateUsageLimitResponse = exports.UpdateUsageLimitResponse || (exports.UpdateUsageLimitResponse = {}));
var UpdateWorkgroupRequest;
(function (UpdateWorkgroupRequest) {
    UpdateWorkgroupRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(UpdateWorkgroupRequest = exports.UpdateWorkgroupRequest || (exports.UpdateWorkgroupRequest = {}));
var UpdateWorkgroupResponse;
(function (UpdateWorkgroupResponse) {
    UpdateWorkgroupResponse.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(UpdateWorkgroupResponse = exports.UpdateWorkgroupResponse || (exports.UpdateWorkgroupResponse = {}));
