// @azure/arm-datamigration/esm/operations/projects.js
/*
 * Copyright (c) Microsoft Corporation. All rights reserved.
 * Licensed under the MIT License. See License.txt in the project root for
 * license information.
 *
 * Code generated by Microsoft (R) AutoRest Code Generator.
 * Changes may cause incorrect behavior and will be lost if the code is
 * regenerated.
 */
import { __assign } from "tslib";
import * as msRest from "@azure/ms-rest-js";
import * as Mappers from "../models/projectsMappers";
import * as Parameters from "../models/parameters";
/** Class representing a Projects. */
var Projects = /** @class */ (function () {
    /**
     * Create a Projects.
     * @param {DataMigrationServiceClientContext} client Reference to the service client.
     */
    function Projects(client) {
        this.client = client;
    }
    Projects.prototype.list = function (groupName, serviceName, options, callback) {
        return this.client.sendOperationRequest({
            groupName: groupName,
            serviceName: serviceName,
            options: options
        }, listOperationSpec, callback);
    };
    Projects.prototype.createOrUpdate = function (parameters, groupName, serviceName, projectName, options, callback) {
        return this.client.sendOperationRequest({
            parameters: parameters,
            groupName: groupName,
            serviceName: serviceName,
            projectName: projectName,
            options: options
        }, createOrUpdateOperationSpec, callback);
    };
    Projects.prototype.get = function (groupName, serviceName, projectName, options, callback) {
        return this.client.sendOperationRequest({
            groupName: groupName,
            serviceName: serviceName,
            projectName: projectName,
            options: options
        }, getOperationSpec, callback);
    };
    Projects.prototype.deleteMethod = function (groupName, serviceName, projectName, options, callback) {
        return this.client.sendOperationRequest({
            groupName: groupName,
            serviceName: serviceName,
            projectName: projectName,
            options: options
        }, deleteMethodOperationSpec, callback);
    };
    Projects.prototype.update = function (parameters, groupName, serviceName, projectName, options, callback) {
        return this.client.sendOperationRequest({
            parameters: parameters,
            groupName: groupName,
            serviceName: serviceName,
            projectName: projectName,
            options: options
        }, updateOperationSpec, callback);
    };
    Projects.prototype.listNext = function (nextPageLink, options, callback) {
        return this.client.sendOperationRequest({
            nextPageLink: nextPageLink,
            options: options
        }, listNextOperationSpec, callback);
    };
    return Projects;
}());
export { Projects };
// Operation Specifications
var serializer = new msRest.Serializer(Mappers);
var listOperationSpec = {
    httpMethod: "GET",
    path: "subscriptions/{subscriptionId}/resourceGroups/{groupName}/providers/Microsoft.DataMigration/services/{serviceName}/projects",
    urlParameters: [
        Parameters.subscriptionId,
        Parameters.groupName,
        Parameters.serviceName
    ],
    queryParameters: [
        Parameters.apiVersion
    ],
    headerParameters: [
        Parameters.acceptLanguage
    ],
    responses: {
        200: {
            bodyMapper: Mappers.ProjectList
        },
        default: {
            bodyMapper: Mappers.ApiError
        }
    },
    serializer: serializer
};
var createOrUpdateOperationSpec = {
    httpMethod: "PUT",
    path: "subscriptions/{subscriptionId}/resourceGroups/{groupName}/providers/Microsoft.DataMigration/services/{serviceName}/projects/{projectName}",
    urlParameters: [
        Parameters.subscriptionId,
        Parameters.groupName,
        Parameters.serviceName,
        Parameters.projectName
    ],
    queryParameters: [
        Parameters.apiVersion
    ],
    headerParameters: [
        Parameters.acceptLanguage
    ],
    requestBody: {
        parameterPath: "parameters",
        mapper: __assign(__assign({}, Mappers.Project), { required: true })
    },
    responses: {
        200: {
            bodyMapper: Mappers.Project
        },
        201: {
            bodyMapper: Mappers.Project
        },
        default: {
            bodyMapper: Mappers.ApiError
        }
    },
    serializer: serializer
};
var getOperationSpec = {
    httpMethod: "GET",
    path: "subscriptions/{subscriptionId}/resourceGroups/{groupName}/providers/Microsoft.DataMigration/services/{serviceName}/projects/{projectName}",
    urlParameters: [
        Parameters.subscriptionId,
        Parameters.groupName,
        Parameters.serviceName,
        Parameters.projectName
    ],
    queryParameters: [
        Parameters.apiVersion
    ],
    headerParameters: [
        Parameters.acceptLanguage
    ],
    responses: {
        200: {
            bodyMapper: Mappers.Project
        },
        default: {
            bodyMapper: Mappers.ApiError
        }
    },
    serializer: serializer
};
var deleteMethodOperationSpec = {
    httpMethod: "DELETE",
    path: "subscriptions/{subscriptionId}/resourceGroups/{groupName}/providers/Microsoft.DataMigration/services/{serviceName}/projects/{projectName}",
    urlParameters: [
        Parameters.subscriptionId,
        Parameters.groupName,
        Parameters.serviceName,
        Parameters.projectName
    ],
    queryParameters: [
        Parameters.deleteRunningTasks,
        Parameters.apiVersion
    ],
    headerParameters: [
        Parameters.acceptLanguage
    ],
    responses: {
        200: {},
        204: {},
        default: {
            bodyMapper: Mappers.ApiError
        }
    },
    serializer: serializer
};
var updateOperationSpec = {
    httpMethod: "PATCH",
    path: "subscriptions/{subscriptionId}/resourceGroups/{groupName}/providers/Microsoft.DataMigration/services/{serviceName}/projects/{projectName}",
    urlParameters: [
        Parameters.subscriptionId,
        Parameters.groupName,
        Parameters.serviceName,
        Parameters.projectName
    ],
    queryParameters: [
        Parameters.apiVersion
    ],
    headerParameters: [
        Parameters.acceptLanguage
    ],
    requestBody: {
        parameterPath: "parameters",
        mapper: __assign(__assign({}, Mappers.Project), { required: true })
    },
    responses: {
        200: {
            bodyMapper: Mappers.Project
        },
        default: {
            bodyMapper: Mappers.ApiError
        }
    },
    serializer: serializer
};
var listNextOperationSpec = {
    httpMethod: "GET",
    baseUrl: "https://management.azure.com",
    path: "{nextLink}",
    urlParameters: [
        Parameters.nextPageLink
    ],
    headerParameters: [
        Parameters.acceptLanguage
    ],
    responses: {
        200: {
            bodyMapper: Mappers.ProjectList
        },
        default: {
            bodyMapper: Mappers.ApiError
        }
    },
    serializer: serializer
};
