// @azure/arm-datamigration/esm/operations/resourceSkus.js
/*
 * Copyright (c) Microsoft Corporation. All rights reserved.
 * Licensed under the MIT License. See License.txt in the project root for
 * license information.
 *
 * Code generated by Microsoft (R) AutoRest Code Generator.
 * Changes may cause incorrect behavior and will be lost if the code is
 * regenerated.
 */
import * as msRest from "@azure/ms-rest-js";
import * as Mappers from "../models/resourceSkusMappers";
import * as Parameters from "../models/parameters";
/** Class representing a ResourceSkus. */
var ResourceSkus = /** @class */ (function () {
    /**
     * Create a ResourceSkus.
     * @param {DataMigrationServiceClientContext} client Reference to the service client.
     */
    function ResourceSkus(client) {
        this.client = client;
    }
    ResourceSkus.prototype.listSkus = function (options, callback) {
        return this.client.sendOperationRequest({
            options: options
        }, listSkusOperationSpec, callback);
    };
    ResourceSkus.prototype.listSkusNext = function (nextPageLink, options, callback) {
        return this.client.sendOperationRequest({
            nextPageLink: nextPageLink,
            options: options
        }, listSkusNextOperationSpec, callback);
    };
    return ResourceSkus;
}());
export { ResourceSkus };
// Operation Specifications
var serializer = new msRest.Serializer(Mappers);
var listSkusOperationSpec = {
    httpMethod: "GET",
    path: "subscriptions/{subscriptionId}/providers/Microsoft.DataMigration/skus",
    urlParameters: [
        Parameters.subscriptionId
    ],
    queryParameters: [
        Parameters.apiVersion
    ],
    headerParameters: [
        Parameters.acceptLanguage
    ],
    responses: {
        200: {
            bodyMapper: Mappers.ResourceSkusResult
        },
        default: {
            bodyMapper: Mappers.ApiError
        }
    },
    serializer: serializer
};
var listSkusNextOperationSpec = {
    httpMethod: "GET",
    baseUrl: "https://management.azure.com",
    path: "{nextLink}",
    urlParameters: [
        Parameters.nextPageLink
    ],
    headerParameters: [
        Parameters.acceptLanguage
    ],
    responses: {
        200: {
            bodyMapper: Mappers.ResourceSkusResult
        },
        default: {
            bodyMapper: Mappers.ApiError
        }
    },
    serializer: serializer
};
