// @aws-sdk/client-snowball/dist-cjs/index.js
"use strict";
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __name = (target, value) => __defProp(target, "name", { value, configurable: true });
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// src/index.ts
var src_exports = {};
__export(src_exports, {
  AddressType: () => AddressType,
  CancelClusterCommand: () => CancelClusterCommand,
  CancelJobCommand: () => CancelJobCommand,
  ClusterLimitExceededException: () => ClusterLimitExceededException,
  ClusterState: () => ClusterState,
  ConflictException: () => ConflictException,
  CreateAddressCommand: () => CreateAddressCommand,
  CreateClusterCommand: () => CreateClusterCommand,
  CreateJobCommand: () => CreateJobCommand,
  CreateJobRequestFilterSensitiveLog: () => CreateJobRequestFilterSensitiveLog,
  CreateLongTermPricingCommand: () => CreateLongTermPricingCommand,
  CreateReturnShippingLabelCommand: () => CreateReturnShippingLabelCommand,
  DescribeAddressCommand: () => DescribeAddressCommand,
  DescribeAddressesCommand: () => DescribeAddressesCommand,
  DescribeClusterCommand: () => DescribeClusterCommand,
  DescribeJobCommand: () => DescribeJobCommand,
  DescribeJobResultFilterSensitiveLog: () => DescribeJobResultFilterSensitiveLog,
  DescribeReturnShippingLabelCommand: () => DescribeReturnShippingLabelCommand,
  DeviceServiceName: () => DeviceServiceName,
  Ec2RequestFailedException: () => Ec2RequestFailedException,
  GetJobManifestCommand: () => GetJobManifestCommand,
  GetJobUnlockCodeCommand: () => GetJobUnlockCodeCommand,
  GetSnowballUsageCommand: () => GetSnowballUsageCommand,
  GetSoftwareUpdatesCommand: () => GetSoftwareUpdatesCommand,
  ImpactLevel: () => ImpactLevel,
  InvalidAddressException: () => InvalidAddressException,
  InvalidInputCombinationException: () => InvalidInputCombinationException,
  InvalidJobStateException: () => InvalidJobStateException,
  InvalidNextTokenException: () => InvalidNextTokenException,
  InvalidResourceException: () => InvalidResourceException,
  JobMetadataFilterSensitiveLog: () => JobMetadataFilterSensitiveLog,
  JobState: () => JobState,
  JobType: () => JobType,
  KMSRequestFailedException: () => KMSRequestFailedException,
  ListClusterJobsCommand: () => ListClusterJobsCommand,
  ListClustersCommand: () => ListClustersCommand,
  ListCompatibleImagesCommand: () => ListCompatibleImagesCommand,
  ListJobsCommand: () => ListJobsCommand,
  ListLongTermPricingCommand: () => ListLongTermPricingCommand,
  ListPickupLocationsCommand: () => ListPickupLocationsCommand,
  ListServiceVersionsCommand: () => ListServiceVersionsCommand,
  LongTermPricingType: () => LongTermPricingType,
  PickupDetailsFilterSensitiveLog: () => PickupDetailsFilterSensitiveLog,
  RemoteManagement: () => RemoteManagement,
  ReturnShippingLabelAlreadyExistsException: () => ReturnShippingLabelAlreadyExistsException,
  ServiceName: () => ServiceName,
  ShipmentState: () => ShipmentState,
  ShippingLabelStatus: () => ShippingLabelStatus,
  ShippingOption: () => ShippingOption,
  Snowball: () => Snowball,
  SnowballCapacity: () => SnowballCapacity,
  SnowballClient: () => SnowballClient,
  SnowballServiceException: () => SnowballServiceException,
  SnowballType: () => SnowballType,
  StorageUnit: () => StorageUnit,
  TransferOption: () => TransferOption,
  UnsupportedAddressException: () => UnsupportedAddressException,
  UpdateClusterCommand: () => UpdateClusterCommand,
  UpdateJobCommand: () => UpdateJobCommand,
  UpdateJobRequestFilterSensitiveLog: () => UpdateJobRequestFilterSensitiveLog,
  UpdateJobShipmentStateCommand: () => UpdateJobShipmentStateCommand,
  UpdateLongTermPricingCommand: () => UpdateLongTermPricingCommand,
  __Client: () => import_smithy_client.Client,
  paginateDescribeAddresses: () => paginateDescribeAddresses,
  paginateListClusterJobs: () => paginateListClusterJobs,
  paginateListClusters: () => paginateListClusters,
  paginateListCompatibleImages: () => paginateListCompatibleImages,
  paginateListJobs: () => paginateListJobs,
  paginateListLongTermPricing: () => paginateListLongTermPricing,
  paginateListPickupLocations: () => paginateListPickupLocations
});
module.exports = __toCommonJS(src_exports);

// src/SnowballClient.ts
var import_middleware_host_header = require("@aws-sdk/middleware-host-header");
var import_middleware_logger = require("@aws-sdk/middleware-logger");
var import_middleware_recursion_detection = require("@aws-sdk/middleware-recursion-detection");
var import_middleware_user_agent = require("@aws-sdk/middleware-user-agent");
var import_config_resolver = require("@smithy/config-resolver");
var import_core = require("@smithy/core");
var import_middleware_content_length = require("@smithy/middleware-content-length");
var import_middleware_endpoint = require("@smithy/middleware-endpoint");
var import_middleware_retry = require("@smithy/middleware-retry");

var import_httpAuthSchemeProvider = require("./auth/httpAuthSchemeProvider");

// src/endpoint/EndpointParameters.ts
var resolveClientEndpointParameters = /* @__PURE__ */ __name((options) => {
  return {
    ...options,
    useDualstackEndpoint: options.useDualstackEndpoint ?? false,
    useFipsEndpoint: options.useFipsEndpoint ?? false,
    defaultSigningName: "snowball"
  };
}, "resolveClientEndpointParameters");
var commonParams = {
  UseFIPS: { type: "builtInParams", name: "useFipsEndpoint" },
  Endpoint: { type: "builtInParams", name: "endpoint" },
  Region: { type: "builtInParams", name: "region" },
  UseDualStack: { type: "builtInParams", name: "useDualstackEndpoint" }
};

// src/SnowballClient.ts
var import_runtimeConfig = require("././runtimeConfig");

// src/runtimeExtensions.ts
var import_region_config_resolver = require("@aws-sdk/region-config-resolver");
var import_protocol_http = require("@smithy/protocol-http");
var import_smithy_client = require("@smithy/smithy-client");

// src/auth/httpAuthExtensionConfiguration.ts
var getHttpAuthExtensionConfiguration = /* @__PURE__ */ __name((runtimeConfig) => {
  const _httpAuthSchemes = runtimeConfig.httpAuthSchemes;
  let _httpAuthSchemeProvider = runtimeConfig.httpAuthSchemeProvider;
  let _credentials = runtimeConfig.credentials;
  return {
    setHttpAuthScheme(httpAuthScheme) {
      const index = _httpAuthSchemes.findIndex((scheme) => scheme.schemeId === httpAuthScheme.schemeId);
      if (index === -1) {
        _httpAuthSchemes.push(httpAuthScheme);
      } else {
        _httpAuthSchemes.splice(index, 1, httpAuthScheme);
      }
    },
    httpAuthSchemes() {
      return _httpAuthSchemes;
    },
    setHttpAuthSchemeProvider(httpAuthSchemeProvider) {
      _httpAuthSchemeProvider = httpAuthSchemeProvider;
    },
    httpAuthSchemeProvider() {
      return _httpAuthSchemeProvider;
    },
    setCredentials(credentials) {
      _credentials = credentials;
    },
    credentials() {
      return _credentials;
    }
  };
}, "getHttpAuthExtensionConfiguration");
var resolveHttpAuthRuntimeConfig = /* @__PURE__ */ __name((config) => {
  return {
    httpAuthSchemes: config.httpAuthSchemes(),
    httpAuthSchemeProvider: config.httpAuthSchemeProvider(),
    credentials: config.credentials()
  };
}, "resolveHttpAuthRuntimeConfig");

// src/runtimeExtensions.ts
var asPartial = /* @__PURE__ */ __name((t) => t, "asPartial");
var resolveRuntimeExtensions = /* @__PURE__ */ __name((runtimeConfig, extensions) => {
  const extensionConfiguration = {
    ...asPartial((0, import_region_config_resolver.getAwsRegionExtensionConfiguration)(runtimeConfig)),
    ...asPartial((0, import_smithy_client.getDefaultExtensionConfiguration)(runtimeConfig)),
    ...asPartial((0, import_protocol_http.getHttpHandlerExtensionConfiguration)(runtimeConfig)),
    ...asPartial(getHttpAuthExtensionConfiguration(runtimeConfig))
  };
  extensions.forEach((extension) => extension.configure(extensionConfiguration));
  return {
    ...runtimeConfig,
    ...(0, import_region_config_resolver.resolveAwsRegionExtensionConfiguration)(extensionConfiguration),
    ...(0, import_smithy_client.resolveDefaultRuntimeConfig)(extensionConfiguration),
    ...(0, import_protocol_http.resolveHttpHandlerRuntimeConfig)(extensionConfiguration),
    ...resolveHttpAuthRuntimeConfig(extensionConfiguration)
  };
}, "resolveRuntimeExtensions");

// src/SnowballClient.ts
var _SnowballClient = class _SnowballClient extends import_smithy_client.Client {
  constructor(...[configuration]) {
    const _config_0 = (0, import_runtimeConfig.getRuntimeConfig)(configuration || {});
    const _config_1 = resolveClientEndpointParameters(_config_0);
    const _config_2 = (0, import_config_resolver.resolveRegionConfig)(_config_1);
    const _config_3 = (0, import_middleware_endpoint.resolveEndpointConfig)(_config_2);
    const _config_4 = (0, import_middleware_retry.resolveRetryConfig)(_config_3);
    const _config_5 = (0, import_middleware_host_header.resolveHostHeaderConfig)(_config_4);
    const _config_6 = (0, import_middleware_user_agent.resolveUserAgentConfig)(_config_5);
    const _config_7 = (0, import_httpAuthSchemeProvider.resolveHttpAuthSchemeConfig)(_config_6);
    const _config_8 = resolveRuntimeExtensions(_config_7, (configuration == null ? void 0 : configuration.extensions) || []);
    super(_config_8);
    this.config = _config_8;
    this.middlewareStack.use((0, import_middleware_retry.getRetryPlugin)(this.config));
    this.middlewareStack.use((0, import_middleware_content_length.getContentLengthPlugin)(this.config));
    this.middlewareStack.use((0, import_middleware_host_header.getHostHeaderPlugin)(this.config));
    this.middlewareStack.use((0, import_middleware_logger.getLoggerPlugin)(this.config));
    this.middlewareStack.use((0, import_middleware_recursion_detection.getRecursionDetectionPlugin)(this.config));
    this.middlewareStack.use((0, import_middleware_user_agent.getUserAgentPlugin)(this.config));
    this.middlewareStack.use(
      (0, import_core.getHttpAuthSchemeEndpointRuleSetPlugin)(this.config, {
        httpAuthSchemeParametersProvider: this.getDefaultHttpAuthSchemeParametersProvider(),
        identityProviderConfigProvider: this.getIdentityProviderConfigProvider()
      })
    );
    this.middlewareStack.use((0, import_core.getHttpSigningPlugin)(this.config));
  }
  /**
   * Destroy underlying resources, like sockets. It's usually not necessary to do this.
   * However in Node.js, it's best to explicitly shut down the client's agent when it is no longer needed.
   * Otherwise, sockets might stay open for quite a long time before the server terminates them.
   */
  destroy() {
    super.destroy();
  }
  getDefaultHttpAuthSchemeParametersProvider() {
    return import_httpAuthSchemeProvider.defaultSnowballHttpAuthSchemeParametersProvider;
  }
  getIdentityProviderConfigProvider() {
    return async (config) => new import_core.DefaultIdentityProviderConfig({
      "aws.auth#sigv4": config.credentials
    });
  }
};
__name(_SnowballClient, "SnowballClient");
var SnowballClient = _SnowballClient;

// src/Snowball.ts


// src/commands/CancelClusterCommand.ts

var import_middleware_serde = require("@smithy/middleware-serde");

var import_types = require("@smithy/types");

// src/protocols/Aws_json1_1.ts



// src/models/models_0.ts


// src/models/SnowballServiceException.ts

var _SnowballServiceException = class _SnowballServiceException extends import_smithy_client.ServiceException {
  /**
   * @internal
   */
  constructor(options) {
    super(options);
    Object.setPrototypeOf(this, _SnowballServiceException.prototype);
  }
};
__name(_SnowballServiceException, "SnowballServiceException");
var SnowballServiceException = _SnowballServiceException;

// src/models/models_0.ts
var AddressType = {
  AWS_SHIP: "AWS_SHIP",
  CUST_PICKUP: "CUST_PICKUP"
};
var _InvalidJobStateException = class _InvalidJobStateException extends SnowballServiceException {
  /**
   * @internal
   */
  constructor(opts) {
    super({
      name: "InvalidJobStateException",
      $fault: "client",
      ...opts
    });
    this.name = "InvalidJobStateException";
    this.$fault = "client";
    Object.setPrototypeOf(this, _InvalidJobStateException.prototype);
    this.Message = opts.Message;
  }
};
__name(_InvalidJobStateException, "InvalidJobStateException");
var InvalidJobStateException = _InvalidJobStateException;
var _InvalidResourceException = class _InvalidResourceException extends SnowballServiceException {
  /**
   * @internal
   */
  constructor(opts) {
    super({
      name: "InvalidResourceException",
      $fault: "client",
      ...opts
    });
    this.name = "InvalidResourceException";
    this.$fault = "client";
    Object.setPrototypeOf(this, _InvalidResourceException.prototype);
    this.Message = opts.Message;
    this.ResourceType = opts.ResourceType;
  }
};
__name(_InvalidResourceException, "InvalidResourceException");
var InvalidResourceException = _InvalidResourceException;
var _KMSRequestFailedException = class _KMSRequestFailedException extends SnowballServiceException {
  /**
   * @internal
   */
  constructor(opts) {
    super({
      name: "KMSRequestFailedException",
      $fault: "client",
      ...opts
    });
    this.name = "KMSRequestFailedException";
    this.$fault = "client";
    Object.setPrototypeOf(this, _KMSRequestFailedException.prototype);
    this.Message = opts.Message;
  }
};
__name(_KMSRequestFailedException, "KMSRequestFailedException");
var KMSRequestFailedException = _KMSRequestFailedException;
var _InvalidAddressException = class _InvalidAddressException extends SnowballServiceException {
  /**
   * @internal
   */
  constructor(opts) {
    super({
      name: "InvalidAddressException",
      $fault: "client",
      ...opts
    });
    this.name = "InvalidAddressException";
    this.$fault = "client";
    Object.setPrototypeOf(this, _InvalidAddressException.prototype);
    this.Message = opts.Message;
  }
};
__name(_InvalidAddressException, "InvalidAddressException");
var InvalidAddressException = _InvalidAddressException;
var _UnsupportedAddressException = class _UnsupportedAddressException extends SnowballServiceException {
  /**
   * @internal
   */
  constructor(opts) {
    super({
      name: "UnsupportedAddressException",
      $fault: "client",
      ...opts
    });
    this.name = "UnsupportedAddressException";
    this.$fault = "client";
    Object.setPrototypeOf(this, _UnsupportedAddressException.prototype);
    this.Message = opts.Message;
  }
};
__name(_UnsupportedAddressException, "UnsupportedAddressException");
var UnsupportedAddressException = _UnsupportedAddressException;
var JobType = {
  EXPORT: "EXPORT",
  IMPORT: "IMPORT",
  LOCAL_USE: "LOCAL_USE"
};
var JobState = {
  CANCELLED: "Cancelled",
  COMPLETE: "Complete",
  IN_PROGRESS: "InProgress",
  IN_TRANSIT_TO_AWS: "InTransitToAWS",
  IN_TRANSIT_TO_CUSTOMER: "InTransitToCustomer",
  LISTING: "Listing",
  NEW: "New",
  PENDING: "Pending",
  PREPARING_APPLIANCE: "PreparingAppliance",
  PREPARING_SHIPMENT: "PreparingShipment",
  WITH_AWS: "WithAWS",
  WITH_AWS_SORTING_FACILITY: "WithAWSSortingFacility",
  WITH_CUSTOMER: "WithCustomer"
};
var StorageUnit = {
  TB: "TB"
};
var RemoteManagement = {
  INSTALLED_AUTOSTART: "INSTALLED_AUTOSTART",
  INSTALLED_ONLY: "INSTALLED_ONLY",
  NOT_INSTALLED: "NOT_INSTALLED"
};
var DeviceServiceName = {
  NFS_ON_DEVICE_SERVICE: "NFS_ON_DEVICE_SERVICE",
  S3_ON_DEVICE_SERVICE: "S3_ON_DEVICE_SERVICE"
};
var TransferOption = {
  EXPORT: "EXPORT",
  IMPORT: "IMPORT",
  LOCAL_USE: "LOCAL_USE"
};
var ShippingOption = {
  EXPRESS: "EXPRESS",
  NEXT_DAY: "NEXT_DAY",
  SECOND_DAY: "SECOND_DAY",
  STANDARD: "STANDARD"
};
var SnowballCapacity = {
  NO_PREFERENCE: "NoPreference",
  T100: "T100",
  T13: "T13",
  T14: "T14",
  T240: "T240",
  T32: "T32",
  T42: "T42",
  T50: "T50",
  T8: "T8",
  T80: "T80",
  T98: "T98"
};
var SnowballType = {
  EDGE: "EDGE",
  EDGE_C: "EDGE_C",
  EDGE_CG: "EDGE_CG",
  EDGE_S: "EDGE_S",
  RACK_5U_C: "RACK_5U_C",
  SNC1_HDD: "SNC1_HDD",
  SNC1_SSD: "SNC1_SSD",
  STANDARD: "STANDARD",
  V3_5C: "V3_5C",
  V3_5S: "V3_5S"
};
var _Ec2RequestFailedException = class _Ec2RequestFailedException extends SnowballServiceException {
  /**
   * @internal
   */
  constructor(opts) {
    super({
      name: "Ec2RequestFailedException",
      $fault: "client",
      ...opts
    });
    this.name = "Ec2RequestFailedException";
    this.$fault = "client";
    Object.setPrototypeOf(this, _Ec2RequestFailedException.prototype);
    this.Message = opts.Message;
  }
};
__name(_Ec2RequestFailedException, "Ec2RequestFailedException");
var Ec2RequestFailedException = _Ec2RequestFailedException;
var _InvalidInputCombinationException = class _InvalidInputCombinationException extends SnowballServiceException {
  /**
   * @internal
   */
  constructor(opts) {
    super({
      name: "InvalidInputCombinationException",
      $fault: "client",
      ...opts
    });
    this.name = "InvalidInputCombinationException";
    this.$fault = "client";
    Object.setPrototypeOf(this, _InvalidInputCombinationException.prototype);
    this.Message = opts.Message;
  }
};
__name(_InvalidInputCombinationException, "InvalidInputCombinationException");
var InvalidInputCombinationException = _InvalidInputCombinationException;
var _ClusterLimitExceededException = class _ClusterLimitExceededException extends SnowballServiceException {
  /**
   * @internal
   */
  constructor(opts) {
    super({
      name: "ClusterLimitExceededException",
      $fault: "client",
      ...opts
    });
    this.name = "ClusterLimitExceededException";
    this.$fault = "client";
    Object.setPrototypeOf(this, _ClusterLimitExceededException.prototype);
    this.Message = opts.Message;
  }
};
__name(_ClusterLimitExceededException, "ClusterLimitExceededException");
var ClusterLimitExceededException = _ClusterLimitExceededException;
var ImpactLevel = {
  IL2: "IL2",
  IL4: "IL4",
  IL5: "IL5",
  IL6: "IL6",
  IL99: "IL99"
};
var LongTermPricingType = {
  ONE_MONTH: "OneMonth",
  ONE_YEAR: "OneYear",
  THREE_YEAR: "ThreeYear"
};
var _ConflictException = class _ConflictException extends SnowballServiceException {
  /**
   * @internal
   */
  constructor(opts) {
    super({
      name: "ConflictException",
      $fault: "client",
      ...opts
    });
    this.name = "ConflictException";
    this.$fault = "client";
    Object.setPrototypeOf(this, _ConflictException.prototype);
    this.ConflictResource = opts.ConflictResource;
    this.Message = opts.Message;
  }
};
__name(_ConflictException, "ConflictException");
var ConflictException = _ConflictException;
var ShippingLabelStatus = {
  FAILED: "Failed",
  IN_PROGRESS: "InProgress",
  SUCCEEDED: "Succeeded",
  TIMED_OUT: "TimedOut"
};
var _ReturnShippingLabelAlreadyExistsException = class _ReturnShippingLabelAlreadyExistsException extends SnowballServiceException {
  /**
   * @internal
   */
  constructor(opts) {
    super({
      name: "ReturnShippingLabelAlreadyExistsException",
      $fault: "client",
      ...opts
    });
    this.name = "ReturnShippingLabelAlreadyExistsException";
    this.$fault = "client";
    Object.setPrototypeOf(this, _ReturnShippingLabelAlreadyExistsException.prototype);
    this.Message = opts.Message;
  }
};
__name(_ReturnShippingLabelAlreadyExistsException, "ReturnShippingLabelAlreadyExistsException");
var ReturnShippingLabelAlreadyExistsException = _ReturnShippingLabelAlreadyExistsException;
var _InvalidNextTokenException = class _InvalidNextTokenException extends SnowballServiceException {
  /**
   * @internal
   */
  constructor(opts) {
    super({
      name: "InvalidNextTokenException",
      $fault: "client",
      ...opts
    });
    this.name = "InvalidNextTokenException";
    this.$fault = "client";
    Object.setPrototypeOf(this, _InvalidNextTokenException.prototype);
    this.Message = opts.Message;
  }
};
__name(_InvalidNextTokenException, "InvalidNextTokenException");
var InvalidNextTokenException = _InvalidNextTokenException;
var ClusterState = {
  AWAITING_QUORUM: "AwaitingQuorum",
  CANCELLED: "Cancelled",
  COMPLETE: "Complete",
  IN_USE: "InUse",
  PENDING: "Pending"
};
var ServiceName = {
  EKS_ANYWHERE: "EKS_ANYWHERE",
  KUBERNETES: "KUBERNETES"
};
var ShipmentState = {
  RECEIVED: "RECEIVED",
  RETURNED: "RETURNED"
};
var PickupDetailsFilterSensitiveLog = /* @__PURE__ */ __name((obj) => ({
  ...obj,
  ...obj.PhoneNumber && { PhoneNumber: import_smithy_client.SENSITIVE_STRING },
  ...obj.Email && { Email: import_smithy_client.SENSITIVE_STRING }
}), "PickupDetailsFilterSensitiveLog");
var CreateJobRequestFilterSensitiveLog = /* @__PURE__ */ __name((obj) => ({
  ...obj,
  ...obj.PickupDetails && { PickupDetails: PickupDetailsFilterSensitiveLog(obj.PickupDetails) }
}), "CreateJobRequestFilterSensitiveLog");
var JobMetadataFilterSensitiveLog = /* @__PURE__ */ __name((obj) => ({
  ...obj,
  ...obj.PickupDetails && { PickupDetails: PickupDetailsFilterSensitiveLog(obj.PickupDetails) }
}), "JobMetadataFilterSensitiveLog");
var DescribeJobResultFilterSensitiveLog = /* @__PURE__ */ __name((obj) => ({
  ...obj,
  ...obj.JobMetadata && { JobMetadata: JobMetadataFilterSensitiveLog(obj.JobMetadata) },
  ...obj.SubJobMetadata && { SubJobMetadata: obj.SubJobMetadata.map((item) => JobMetadataFilterSensitiveLog(item)) }
}), "DescribeJobResultFilterSensitiveLog");
var UpdateJobRequestFilterSensitiveLog = /* @__PURE__ */ __name((obj) => ({
  ...obj,
  ...obj.PickupDetails && { PickupDetails: PickupDetailsFilterSensitiveLog(obj.PickupDetails) }
}), "UpdateJobRequestFilterSensitiveLog");

// src/protocols/Aws_json1_1.ts
var se_CancelClusterCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("CancelCluster");
  let body;
  body = JSON.stringify((0, import_smithy_client._json)(input));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_CancelClusterCommand");
var se_CancelJobCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("CancelJob");
  let body;
  body = JSON.stringify((0, import_smithy_client._json)(input));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_CancelJobCommand");
var se_CreateAddressCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("CreateAddress");
  let body;
  body = JSON.stringify((0, import_smithy_client._json)(input));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_CreateAddressCommand");
var se_CreateClusterCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("CreateCluster");
  let body;
  body = JSON.stringify(se_CreateClusterRequest(input, context));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_CreateClusterCommand");
var se_CreateJobCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("CreateJob");
  let body;
  body = JSON.stringify(se_CreateJobRequest(input, context));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_CreateJobCommand");
var se_CreateLongTermPricingCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("CreateLongTermPricing");
  let body;
  body = JSON.stringify((0, import_smithy_client._json)(input));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_CreateLongTermPricingCommand");
var se_CreateReturnShippingLabelCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("CreateReturnShippingLabel");
  let body;
  body = JSON.stringify((0, import_smithy_client._json)(input));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_CreateReturnShippingLabelCommand");
var se_DescribeAddressCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("DescribeAddress");
  let body;
  body = JSON.stringify((0, import_smithy_client._json)(input));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_DescribeAddressCommand");
var se_DescribeAddressesCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("DescribeAddresses");
  let body;
  body = JSON.stringify((0, import_smithy_client._json)(input));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_DescribeAddressesCommand");
var se_DescribeClusterCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("DescribeCluster");
  let body;
  body = JSON.stringify((0, import_smithy_client._json)(input));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_DescribeClusterCommand");
var se_DescribeJobCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("DescribeJob");
  let body;
  body = JSON.stringify((0, import_smithy_client._json)(input));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_DescribeJobCommand");
var se_DescribeReturnShippingLabelCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("DescribeReturnShippingLabel");
  let body;
  body = JSON.stringify((0, import_smithy_client._json)(input));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_DescribeReturnShippingLabelCommand");
var se_GetJobManifestCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("GetJobManifest");
  let body;
  body = JSON.stringify((0, import_smithy_client._json)(input));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_GetJobManifestCommand");
var se_GetJobUnlockCodeCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("GetJobUnlockCode");
  let body;
  body = JSON.stringify((0, import_smithy_client._json)(input));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_GetJobUnlockCodeCommand");
var se_GetSnowballUsageCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("GetSnowballUsage");
  let body;
  body = JSON.stringify((0, import_smithy_client._json)(input));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_GetSnowballUsageCommand");
var se_GetSoftwareUpdatesCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("GetSoftwareUpdates");
  let body;
  body = JSON.stringify((0, import_smithy_client._json)(input));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_GetSoftwareUpdatesCommand");
var se_ListClusterJobsCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("ListClusterJobs");
  let body;
  body = JSON.stringify((0, import_smithy_client._json)(input));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_ListClusterJobsCommand");
var se_ListClustersCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("ListClusters");
  let body;
  body = JSON.stringify((0, import_smithy_client._json)(input));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_ListClustersCommand");
var se_ListCompatibleImagesCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("ListCompatibleImages");
  let body;
  body = JSON.stringify((0, import_smithy_client._json)(input));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_ListCompatibleImagesCommand");
var se_ListJobsCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("ListJobs");
  let body;
  body = JSON.stringify((0, import_smithy_client._json)(input));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_ListJobsCommand");
var se_ListLongTermPricingCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("ListLongTermPricing");
  let body;
  body = JSON.stringify((0, import_smithy_client._json)(input));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_ListLongTermPricingCommand");
var se_ListPickupLocationsCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("ListPickupLocations");
  let body;
  body = JSON.stringify((0, import_smithy_client._json)(input));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_ListPickupLocationsCommand");
var se_ListServiceVersionsCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("ListServiceVersions");
  let body;
  body = JSON.stringify((0, import_smithy_client._json)(input));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_ListServiceVersionsCommand");
var se_UpdateClusterCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("UpdateCluster");
  let body;
  body = JSON.stringify(se_UpdateClusterRequest(input, context));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_UpdateClusterCommand");
var se_UpdateJobCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("UpdateJob");
  let body;
  body = JSON.stringify(se_UpdateJobRequest(input, context));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_UpdateJobCommand");
var se_UpdateJobShipmentStateCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("UpdateJobShipmentState");
  let body;
  body = JSON.stringify((0, import_smithy_client._json)(input));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_UpdateJobShipmentStateCommand");
var se_UpdateLongTermPricingCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("UpdateLongTermPricing");
  let body;
  body = JSON.stringify((0, import_smithy_client._json)(input));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_UpdateLongTermPricingCommand");
var de_CancelClusterCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const data = await parseBody(output.body, context);
  let contents = {};
  contents = (0, import_smithy_client._json)(data);
  const response = {
    $metadata: deserializeMetadata(output),
    ...contents
  };
  return response;
}, "de_CancelClusterCommand");
var de_CancelJobCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const data = await parseBody(output.body, context);
  let contents = {};
  contents = (0, import_smithy_client._json)(data);
  const response = {
    $metadata: deserializeMetadata(output),
    ...contents
  };
  return response;
}, "de_CancelJobCommand");
var de_CreateAddressCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const data = await parseBody(output.body, context);
  let contents = {};
  contents = (0, import_smithy_client._json)(data);
  const response = {
    $metadata: deserializeMetadata(output),
    ...contents
  };
  return response;
}, "de_CreateAddressCommand");
var de_CreateClusterCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const data = await parseBody(output.body, context);
  let contents = {};
  contents = de_CreateClusterResult(data, context);
  const response = {
    $metadata: deserializeMetadata(output),
    ...contents
  };
  return response;
}, "de_CreateClusterCommand");
var de_CreateJobCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const data = await parseBody(output.body, context);
  let contents = {};
  contents = (0, import_smithy_client._json)(data);
  const response = {
    $metadata: deserializeMetadata(output),
    ...contents
  };
  return response;
}, "de_CreateJobCommand");
var de_CreateLongTermPricingCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const data = await parseBody(output.body, context);
  let contents = {};
  contents = (0, import_smithy_client._json)(data);
  const response = {
    $metadata: deserializeMetadata(output),
    ...contents
  };
  return response;
}, "de_CreateLongTermPricingCommand");
var de_CreateReturnShippingLabelCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const data = await parseBody(output.body, context);
  let contents = {};
  contents = (0, import_smithy_client._json)(data);
  const response = {
    $metadata: deserializeMetadata(output),
    ...contents
  };
  return response;
}, "de_CreateReturnShippingLabelCommand");
var de_DescribeAddressCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const data = await parseBody(output.body, context);
  let contents = {};
  contents = (0, import_smithy_client._json)(data);
  const response = {
    $metadata: deserializeMetadata(output),
    ...contents
  };
  return response;
}, "de_DescribeAddressCommand");
var de_DescribeAddressesCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const data = await parseBody(output.body, context);
  let contents = {};
  contents = (0, import_smithy_client._json)(data);
  const response = {
    $metadata: deserializeMetadata(output),
    ...contents
  };
  return response;
}, "de_DescribeAddressesCommand");
var de_DescribeClusterCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const data = await parseBody(output.body, context);
  let contents = {};
  contents = de_DescribeClusterResult(data, context);
  const response = {
    $metadata: deserializeMetadata(output),
    ...contents
  };
  return response;
}, "de_DescribeClusterCommand");
var de_DescribeJobCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const data = await parseBody(output.body, context);
  let contents = {};
  contents = de_DescribeJobResult(data, context);
  const response = {
    $metadata: deserializeMetadata(output),
    ...contents
  };
  return response;
}, "de_DescribeJobCommand");
var de_DescribeReturnShippingLabelCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const data = await parseBody(output.body, context);
  let contents = {};
  contents = de_DescribeReturnShippingLabelResult(data, context);
  const response = {
    $metadata: deserializeMetadata(output),
    ...contents
  };
  return response;
}, "de_DescribeReturnShippingLabelCommand");
var de_GetJobManifestCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const data = await parseBody(output.body, context);
  let contents = {};
  contents = (0, import_smithy_client._json)(data);
  const response = {
    $metadata: deserializeMetadata(output),
    ...contents
  };
  return response;
}, "de_GetJobManifestCommand");
var de_GetJobUnlockCodeCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const data = await parseBody(output.body, context);
  let contents = {};
  contents = (0, import_smithy_client._json)(data);
  const response = {
    $metadata: deserializeMetadata(output),
    ...contents
  };
  return response;
}, "de_GetJobUnlockCodeCommand");
var de_GetSnowballUsageCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const data = await parseBody(output.body, context);
  let contents = {};
  contents = (0, import_smithy_client._json)(data);
  const response = {
    $metadata: deserializeMetadata(output),
    ...contents
  };
  return response;
}, "de_GetSnowballUsageCommand");
var de_GetSoftwareUpdatesCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const data = await parseBody(output.body, context);
  let contents = {};
  contents = (0, import_smithy_client._json)(data);
  const response = {
    $metadata: deserializeMetadata(output),
    ...contents
  };
  return response;
}, "de_GetSoftwareUpdatesCommand");
var de_ListClusterJobsCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const data = await parseBody(output.body, context);
  let contents = {};
  contents = de_ListClusterJobsResult(data, context);
  const response = {
    $metadata: deserializeMetadata(output),
    ...contents
  };
  return response;
}, "de_ListClusterJobsCommand");
var de_ListClustersCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const data = await parseBody(output.body, context);
  let contents = {};
  contents = de_ListClustersResult(data, context);
  const response = {
    $metadata: deserializeMetadata(output),
    ...contents
  };
  return response;
}, "de_ListClustersCommand");
var de_ListCompatibleImagesCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const data = await parseBody(output.body, context);
  let contents = {};
  contents = (0, import_smithy_client._json)(data);
  const response = {
    $metadata: deserializeMetadata(output),
    ...contents
  };
  return response;
}, "de_ListCompatibleImagesCommand");
var de_ListJobsCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const data = await parseBody(output.body, context);
  let contents = {};
  contents = de_ListJobsResult(data, context);
  const response = {
    $metadata: deserializeMetadata(output),
    ...contents
  };
  return response;
}, "de_ListJobsCommand");
var de_ListLongTermPricingCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const data = await parseBody(output.body, context);
  let contents = {};
  contents = de_ListLongTermPricingResult(data, context);
  const response = {
    $metadata: deserializeMetadata(output),
    ...contents
  };
  return response;
}, "de_ListLongTermPricingCommand");
var de_ListPickupLocationsCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const data = await parseBody(output.body, context);
  let contents = {};
  contents = (0, import_smithy_client._json)(data);
  const response = {
    $metadata: deserializeMetadata(output),
    ...contents
  };
  return response;
}, "de_ListPickupLocationsCommand");
var de_ListServiceVersionsCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const data = await parseBody(output.body, context);
  let contents = {};
  contents = (0, import_smithy_client._json)(data);
  const response = {
    $metadata: deserializeMetadata(output),
    ...contents
  };
  return response;
}, "de_ListServiceVersionsCommand");
var de_UpdateClusterCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const data = await parseBody(output.body, context);
  let contents = {};
  contents = (0, import_smithy_client._json)(data);
  const response = {
    $metadata: deserializeMetadata(output),
    ...contents
  };
  return response;
}, "de_UpdateClusterCommand");
var de_UpdateJobCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const data = await parseBody(output.body, context);
  let contents = {};
  contents = (0, import_smithy_client._json)(data);
  const response = {
    $metadata: deserializeMetadata(output),
    ...contents
  };
  return response;
}, "de_UpdateJobCommand");
var de_UpdateJobShipmentStateCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const data = await parseBody(output.body, context);
  let contents = {};
  contents = (0, import_smithy_client._json)(data);
  const response = {
    $metadata: deserializeMetadata(output),
    ...contents
  };
  return response;
}, "de_UpdateJobShipmentStateCommand");
var de_UpdateLongTermPricingCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const data = await parseBody(output.body, context);
  let contents = {};
  contents = (0, import_smithy_client._json)(data);
  const response = {
    $metadata: deserializeMetadata(output),
    ...contents
  };
  return response;
}, "de_UpdateLongTermPricingCommand");
var de_CommandError = /* @__PURE__ */ __name(async (output, context) => {
  const parsedOutput = {
    ...output,
    body: await parseErrorBody(output.body, context)
  };
  const errorCode = loadRestJsonErrorCode(output, parsedOutput.body);
  switch (errorCode) {
    case "InvalidJobStateException":
    case "com.amazonaws.snowball#InvalidJobStateException":
      throw await de_InvalidJobStateExceptionRes(parsedOutput, context);
    case "InvalidResourceException":
    case "com.amazonaws.snowball#InvalidResourceException":
      throw await de_InvalidResourceExceptionRes(parsedOutput, context);
    case "KMSRequestFailedException":
    case "com.amazonaws.snowball#KMSRequestFailedException":
      throw await de_KMSRequestFailedExceptionRes(parsedOutput, context);
    case "InvalidAddressException":
    case "com.amazonaws.snowball#InvalidAddressException":
      throw await de_InvalidAddressExceptionRes(parsedOutput, context);
    case "UnsupportedAddressException":
    case "com.amazonaws.snowball#UnsupportedAddressException":
      throw await de_UnsupportedAddressExceptionRes(parsedOutput, context);
    case "Ec2RequestFailedException":
    case "com.amazonaws.snowball#Ec2RequestFailedException":
      throw await de_Ec2RequestFailedExceptionRes(parsedOutput, context);
    case "InvalidInputCombinationException":
    case "com.amazonaws.snowball#InvalidInputCombinationException":
      throw await de_InvalidInputCombinationExceptionRes(parsedOutput, context);
    case "ClusterLimitExceededException":
    case "com.amazonaws.snowball#ClusterLimitExceededException":
      throw await de_ClusterLimitExceededExceptionRes(parsedOutput, context);
    case "ConflictException":
    case "com.amazonaws.snowball#ConflictException":
      throw await de_ConflictExceptionRes(parsedOutput, context);
    case "ReturnShippingLabelAlreadyExistsException":
    case "com.amazonaws.snowball#ReturnShippingLabelAlreadyExistsException":
      throw await de_ReturnShippingLabelAlreadyExistsExceptionRes(parsedOutput, context);
    case "InvalidNextTokenException":
    case "com.amazonaws.snowball#InvalidNextTokenException":
      throw await de_InvalidNextTokenExceptionRes(parsedOutput, context);
    default:
      const parsedBody = parsedOutput.body;
      return throwDefaultError({
        output,
        parsedBody,
        errorCode
      });
  }
}, "de_CommandError");
var de_ClusterLimitExceededExceptionRes = /* @__PURE__ */ __name(async (parsedOutput, context) => {
  const body = parsedOutput.body;
  const deserialized = (0, import_smithy_client._json)(body);
  const exception = new ClusterLimitExceededException({
    $metadata: deserializeMetadata(parsedOutput),
    ...deserialized
  });
  return (0, import_smithy_client.decorateServiceException)(exception, body);
}, "de_ClusterLimitExceededExceptionRes");
var de_ConflictExceptionRes = /* @__PURE__ */ __name(async (parsedOutput, context) => {
  const body = parsedOutput.body;
  const deserialized = (0, import_smithy_client._json)(body);
  const exception = new ConflictException({
    $metadata: deserializeMetadata(parsedOutput),
    ...deserialized
  });
  return (0, import_smithy_client.decorateServiceException)(exception, body);
}, "de_ConflictExceptionRes");
var de_Ec2RequestFailedExceptionRes = /* @__PURE__ */ __name(async (parsedOutput, context) => {
  const body = parsedOutput.body;
  const deserialized = (0, import_smithy_client._json)(body);
  const exception = new Ec2RequestFailedException({
    $metadata: deserializeMetadata(parsedOutput),
    ...deserialized
  });
  return (0, import_smithy_client.decorateServiceException)(exception, body);
}, "de_Ec2RequestFailedExceptionRes");
var de_InvalidAddressExceptionRes = /* @__PURE__ */ __name(async (parsedOutput, context) => {
  const body = parsedOutput.body;
  const deserialized = (0, import_smithy_client._json)(body);
  const exception = new InvalidAddressException({
    $metadata: deserializeMetadata(parsedOutput),
    ...deserialized
  });
  return (0, import_smithy_client.decorateServiceException)(exception, body);
}, "de_InvalidAddressExceptionRes");
var de_InvalidInputCombinationExceptionRes = /* @__PURE__ */ __name(async (parsedOutput, context) => {
  const body = parsedOutput.body;
  const deserialized = (0, import_smithy_client._json)(body);
  const exception = new InvalidInputCombinationException({
    $metadata: deserializeMetadata(parsedOutput),
    ...deserialized
  });
  return (0, import_smithy_client.decorateServiceException)(exception, body);
}, "de_InvalidInputCombinationExceptionRes");
var de_InvalidJobStateExceptionRes = /* @__PURE__ */ __name(async (parsedOutput, context) => {
  const body = parsedOutput.body;
  const deserialized = (0, import_smithy_client._json)(body);
  const exception = new InvalidJobStateException({
    $metadata: deserializeMetadata(parsedOutput),
    ...deserialized
  });
  return (0, import_smithy_client.decorateServiceException)(exception, body);
}, "de_InvalidJobStateExceptionRes");
var de_InvalidNextTokenExceptionRes = /* @__PURE__ */ __name(async (parsedOutput, context) => {
  const body = parsedOutput.body;
  const deserialized = (0, import_smithy_client._json)(body);
  const exception = new InvalidNextTokenException({
    $metadata: deserializeMetadata(parsedOutput),
    ...deserialized
  });
  return (0, import_smithy_client.decorateServiceException)(exception, body);
}, "de_InvalidNextTokenExceptionRes");
var de_InvalidResourceExceptionRes = /* @__PURE__ */ __name(async (parsedOutput, context) => {
  const body = parsedOutput.body;
  const deserialized = (0, import_smithy_client._json)(body);
  const exception = new InvalidResourceException({
    $metadata: deserializeMetadata(parsedOutput),
    ...deserialized
  });
  return (0, import_smithy_client.decorateServiceException)(exception, body);
}, "de_InvalidResourceExceptionRes");
var de_KMSRequestFailedExceptionRes = /* @__PURE__ */ __name(async (parsedOutput, context) => {
  const body = parsedOutput.body;
  const deserialized = (0, import_smithy_client._json)(body);
  const exception = new KMSRequestFailedException({
    $metadata: deserializeMetadata(parsedOutput),
    ...deserialized
  });
  return (0, import_smithy_client.decorateServiceException)(exception, body);
}, "de_KMSRequestFailedExceptionRes");
var de_ReturnShippingLabelAlreadyExistsExceptionRes = /* @__PURE__ */ __name(async (parsedOutput, context) => {
  const body = parsedOutput.body;
  const deserialized = (0, import_smithy_client._json)(body);
  const exception = new ReturnShippingLabelAlreadyExistsException({
    $metadata: deserializeMetadata(parsedOutput),
    ...deserialized
  });
  return (0, import_smithy_client.decorateServiceException)(exception, body);
}, "de_ReturnShippingLabelAlreadyExistsExceptionRes");
var de_UnsupportedAddressExceptionRes = /* @__PURE__ */ __name(async (parsedOutput, context) => {
  const body = parsedOutput.body;
  const deserialized = (0, import_smithy_client._json)(body);
  const exception = new UnsupportedAddressException({
    $metadata: deserializeMetadata(parsedOutput),
    ...deserialized
  });
  return (0, import_smithy_client.decorateServiceException)(exception, body);
}, "de_UnsupportedAddressExceptionRes");
var se_CreateClusterRequest = /* @__PURE__ */ __name((input, context) => {
  return (0, import_smithy_client.take)(input, {
    AddressId: [],
    Description: [],
    ForceCreateJobs: [],
    ForwardingAddressId: [],
    InitialClusterSize: [],
    JobType: [],
    KmsKeyARN: [],
    LongTermPricingIds: import_smithy_client._json,
    Notification: import_smithy_client._json,
    OnDeviceServiceConfiguration: (_) => se_OnDeviceServiceConfiguration(_, context),
    RemoteManagement: [],
    Resources: import_smithy_client._json,
    RoleARN: [],
    ShippingOption: [],
    SnowballCapacityPreference: [],
    SnowballType: [],
    TaxDocuments: import_smithy_client._json
  });
}, "se_CreateClusterRequest");
var se_CreateJobRequest = /* @__PURE__ */ __name((input, context) => {
  return (0, import_smithy_client.take)(input, {
    AddressId: [],
    ClusterId: [],
    Description: [],
    DeviceConfiguration: import_smithy_client._json,
    ForwardingAddressId: [],
    ImpactLevel: [],
    JobType: [],
    KmsKeyARN: [],
    LongTermPricingId: [],
    Notification: import_smithy_client._json,
    OnDeviceServiceConfiguration: (_) => se_OnDeviceServiceConfiguration(_, context),
    PickupDetails: (_) => se_PickupDetails(_, context),
    RemoteManagement: [],
    Resources: import_smithy_client._json,
    RoleARN: [],
    ShippingOption: [],
    SnowballCapacityPreference: [],
    SnowballType: [],
    TaxDocuments: import_smithy_client._json
  });
}, "se_CreateJobRequest");
var se_OnDeviceServiceConfiguration = /* @__PURE__ */ __name((input, context) => {
  return (0, import_smithy_client.take)(input, {
    EKSOnDeviceService: import_smithy_client._json,
    NFSOnDeviceService: import_smithy_client._json,
    S3OnDeviceService: (_) => se_S3OnDeviceServiceConfiguration(_, context),
    TGWOnDeviceService: import_smithy_client._json
  });
}, "se_OnDeviceServiceConfiguration");
var se_PickupDetails = /* @__PURE__ */ __name((input, context) => {
  return (0, import_smithy_client.take)(input, {
    DevicePickupId: [],
    Email: [],
    IdentificationExpirationDate: (_) => Math.round(_.getTime() / 1e3),
    IdentificationIssuingOrg: [],
    IdentificationNumber: [],
    Name: [],
    PhoneNumber: []
  });
}, "se_PickupDetails");
var se_S3OnDeviceServiceConfiguration = /* @__PURE__ */ __name((input, context) => {
  return (0, import_smithy_client.take)(input, {
    FaultTolerance: [],
    ServiceSize: [],
    StorageLimit: import_smithy_client.serializeFloat,
    StorageUnit: []
  });
}, "se_S3OnDeviceServiceConfiguration");
var se_UpdateClusterRequest = /* @__PURE__ */ __name((input, context) => {
  return (0, import_smithy_client.take)(input, {
    AddressId: [],
    ClusterId: [],
    Description: [],
    ForwardingAddressId: [],
    Notification: import_smithy_client._json,
    OnDeviceServiceConfiguration: (_) => se_OnDeviceServiceConfiguration(_, context),
    Resources: import_smithy_client._json,
    RoleARN: [],
    ShippingOption: []
  });
}, "se_UpdateClusterRequest");
var se_UpdateJobRequest = /* @__PURE__ */ __name((input, context) => {
  return (0, import_smithy_client.take)(input, {
    AddressId: [],
    Description: [],
    ForwardingAddressId: [],
    JobId: [],
    Notification: import_smithy_client._json,
    OnDeviceServiceConfiguration: (_) => se_OnDeviceServiceConfiguration(_, context),
    PickupDetails: (_) => se_PickupDetails(_, context),
    Resources: import_smithy_client._json,
    RoleARN: [],
    ShippingOption: [],
    SnowballCapacityPreference: []
  });
}, "se_UpdateJobRequest");
var de_ClusterListEntry = /* @__PURE__ */ __name((output, context) => {
  return (0, import_smithy_client.take)(output, {
    ClusterId: import_smithy_client.expectString,
    ClusterState: import_smithy_client.expectString,
    CreationDate: (_) => (0, import_smithy_client.expectNonNull)((0, import_smithy_client.parseEpochTimestamp)((0, import_smithy_client.expectNumber)(_))),
    Description: import_smithy_client.expectString
  });
}, "de_ClusterListEntry");
var de_ClusterListEntryList = /* @__PURE__ */ __name((output, context) => {
  const retVal = (output || []).filter((e) => e != null).map((entry) => {
    return de_ClusterListEntry(entry, context);
  });
  return retVal;
}, "de_ClusterListEntryList");
var de_ClusterMetadata = /* @__PURE__ */ __name((output, context) => {
  return (0, import_smithy_client.take)(output, {
    AddressId: import_smithy_client.expectString,
    ClusterId: import_smithy_client.expectString,
    ClusterState: import_smithy_client.expectString,
    CreationDate: (_) => (0, import_smithy_client.expectNonNull)((0, import_smithy_client.parseEpochTimestamp)((0, import_smithy_client.expectNumber)(_))),
    Description: import_smithy_client.expectString,
    ForwardingAddressId: import_smithy_client.expectString,
    JobType: import_smithy_client.expectString,
    KmsKeyARN: import_smithy_client.expectString,
    Notification: import_smithy_client._json,
    OnDeviceServiceConfiguration: (_) => de_OnDeviceServiceConfiguration(_, context),
    Resources: import_smithy_client._json,
    RoleARN: import_smithy_client.expectString,
    ShippingOption: import_smithy_client.expectString,
    SnowballType: import_smithy_client.expectString,
    TaxDocuments: import_smithy_client._json
  });
}, "de_ClusterMetadata");
var de_CreateClusterResult = /* @__PURE__ */ __name((output, context) => {
  return (0, import_smithy_client.take)(output, {
    ClusterId: import_smithy_client.expectString,
    JobListEntries: (_) => de_JobListEntryList(_, context)
  });
}, "de_CreateClusterResult");
var de_DescribeClusterResult = /* @__PURE__ */ __name((output, context) => {
  return (0, import_smithy_client.take)(output, {
    ClusterMetadata: (_) => de_ClusterMetadata(_, context)
  });
}, "de_DescribeClusterResult");
var de_DescribeJobResult = /* @__PURE__ */ __name((output, context) => {
  return (0, import_smithy_client.take)(output, {
    JobMetadata: (_) => de_JobMetadata(_, context),
    SubJobMetadata: (_) => de_JobMetadataList(_, context)
  });
}, "de_DescribeJobResult");
var de_DescribeReturnShippingLabelResult = /* @__PURE__ */ __name((output, context) => {
  return (0, import_smithy_client.take)(output, {
    ExpirationDate: (_) => (0, import_smithy_client.expectNonNull)((0, import_smithy_client.parseEpochTimestamp)((0, import_smithy_client.expectNumber)(_))),
    ReturnShippingLabelURI: import_smithy_client.expectString,
    Status: import_smithy_client.expectString
  });
}, "de_DescribeReturnShippingLabelResult");
var de_JobListEntry = /* @__PURE__ */ __name((output, context) => {
  return (0, import_smithy_client.take)(output, {
    CreationDate: (_) => (0, import_smithy_client.expectNonNull)((0, import_smithy_client.parseEpochTimestamp)((0, import_smithy_client.expectNumber)(_))),
    Description: import_smithy_client.expectString,
    IsMaster: import_smithy_client.expectBoolean,
    JobId: import_smithy_client.expectString,
    JobState: import_smithy_client.expectString,
    JobType: import_smithy_client.expectString,
    SnowballType: import_smithy_client.expectString
  });
}, "de_JobListEntry");
var de_JobListEntryList = /* @__PURE__ */ __name((output, context) => {
  const retVal = (output || []).filter((e) => e != null).map((entry) => {
    return de_JobListEntry(entry, context);
  });
  return retVal;
}, "de_JobListEntryList");
var de_JobMetadata = /* @__PURE__ */ __name((output, context) => {
  return (0, import_smithy_client.take)(output, {
    AddressId: import_smithy_client.expectString,
    ClusterId: import_smithy_client.expectString,
    CreationDate: (_) => (0, import_smithy_client.expectNonNull)((0, import_smithy_client.parseEpochTimestamp)((0, import_smithy_client.expectNumber)(_))),
    DataTransferProgress: import_smithy_client._json,
    Description: import_smithy_client.expectString,
    DeviceConfiguration: import_smithy_client._json,
    ForwardingAddressId: import_smithy_client.expectString,
    ImpactLevel: import_smithy_client.expectString,
    JobId: import_smithy_client.expectString,
    JobLogInfo: import_smithy_client._json,
    JobState: import_smithy_client.expectString,
    JobType: import_smithy_client.expectString,
    KmsKeyARN: import_smithy_client.expectString,
    LongTermPricingId: import_smithy_client.expectString,
    Notification: import_smithy_client._json,
    OnDeviceServiceConfiguration: (_) => de_OnDeviceServiceConfiguration(_, context),
    PickupDetails: (_) => de_PickupDetails(_, context),
    RemoteManagement: import_smithy_client.expectString,
    Resources: import_smithy_client._json,
    RoleARN: import_smithy_client.expectString,
    ShippingDetails: import_smithy_client._json,
    SnowballCapacityPreference: import_smithy_client.expectString,
    SnowballId: import_smithy_client.expectString,
    SnowballType: import_smithy_client.expectString,
    TaxDocuments: import_smithy_client._json
  });
}, "de_JobMetadata");
var de_JobMetadataList = /* @__PURE__ */ __name((output, context) => {
  const retVal = (output || []).filter((e) => e != null).map((entry) => {
    return de_JobMetadata(entry, context);
  });
  return retVal;
}, "de_JobMetadataList");
var de_ListClusterJobsResult = /* @__PURE__ */ __name((output, context) => {
  return (0, import_smithy_client.take)(output, {
    JobListEntries: (_) => de_JobListEntryList(_, context),
    NextToken: import_smithy_client.expectString
  });
}, "de_ListClusterJobsResult");
var de_ListClustersResult = /* @__PURE__ */ __name((output, context) => {
  return (0, import_smithy_client.take)(output, {
    ClusterListEntries: (_) => de_ClusterListEntryList(_, context),
    NextToken: import_smithy_client.expectString
  });
}, "de_ListClustersResult");
var de_ListJobsResult = /* @__PURE__ */ __name((output, context) => {
  return (0, import_smithy_client.take)(output, {
    JobListEntries: (_) => de_JobListEntryList(_, context),
    NextToken: import_smithy_client.expectString
  });
}, "de_ListJobsResult");
var de_ListLongTermPricingResult = /* @__PURE__ */ __name((output, context) => {
  return (0, import_smithy_client.take)(output, {
    LongTermPricingEntries: (_) => de_LongTermPricingEntryList(_, context),
    NextToken: import_smithy_client.expectString
  });
}, "de_ListLongTermPricingResult");
var de_LongTermPricingEntryList = /* @__PURE__ */ __name((output, context) => {
  const retVal = (output || []).filter((e) => e != null).map((entry) => {
    return de_LongTermPricingListEntry(entry, context);
  });
  return retVal;
}, "de_LongTermPricingEntryList");
var de_LongTermPricingListEntry = /* @__PURE__ */ __name((output, context) => {
  return (0, import_smithy_client.take)(output, {
    CurrentActiveJob: import_smithy_client.expectString,
    IsLongTermPricingAutoRenew: import_smithy_client.expectBoolean,
    JobIds: import_smithy_client._json,
    LongTermPricingEndDate: (_) => (0, import_smithy_client.expectNonNull)((0, import_smithy_client.parseEpochTimestamp)((0, import_smithy_client.expectNumber)(_))),
    LongTermPricingId: import_smithy_client.expectString,
    LongTermPricingStartDate: (_) => (0, import_smithy_client.expectNonNull)((0, import_smithy_client.parseEpochTimestamp)((0, import_smithy_client.expectNumber)(_))),
    LongTermPricingStatus: import_smithy_client.expectString,
    LongTermPricingType: import_smithy_client.expectString,
    ReplacementJob: import_smithy_client.expectString,
    SnowballType: import_smithy_client.expectString
  });
}, "de_LongTermPricingListEntry");
var de_OnDeviceServiceConfiguration = /* @__PURE__ */ __name((output, context) => {
  return (0, import_smithy_client.take)(output, {
    EKSOnDeviceService: import_smithy_client._json,
    NFSOnDeviceService: import_smithy_client._json,
    S3OnDeviceService: (_) => de_S3OnDeviceServiceConfiguration(_, context),
    TGWOnDeviceService: import_smithy_client._json
  });
}, "de_OnDeviceServiceConfiguration");
var de_PickupDetails = /* @__PURE__ */ __name((output, context) => {
  return (0, import_smithy_client.take)(output, {
    DevicePickupId: import_smithy_client.expectString,
    Email: import_smithy_client.expectString,
    IdentificationExpirationDate: (_) => (0, import_smithy_client.expectNonNull)((0, import_smithy_client.parseEpochTimestamp)((0, import_smithy_client.expectNumber)(_))),
    IdentificationIssuingOrg: import_smithy_client.expectString,
    IdentificationNumber: import_smithy_client.expectString,
    Name: import_smithy_client.expectString,
    PhoneNumber: import_smithy_client.expectString
  });
}, "de_PickupDetails");
var de_S3OnDeviceServiceConfiguration = /* @__PURE__ */ __name((output, context) => {
  return (0, import_smithy_client.take)(output, {
    FaultTolerance: import_smithy_client.expectInt32,
    ServiceSize: import_smithy_client.expectInt32,
    StorageLimit: import_smithy_client.limitedParseDouble,
    StorageUnit: import_smithy_client.expectString
  });
}, "de_S3OnDeviceServiceConfiguration");
var deserializeMetadata = /* @__PURE__ */ __name((output) => ({
  httpStatusCode: output.statusCode,
  requestId: output.headers["x-amzn-requestid"] ?? output.headers["x-amzn-request-id"] ?? output.headers["x-amz-request-id"],
  extendedRequestId: output.headers["x-amz-id-2"],
  cfId: output.headers["x-amz-cf-id"]
}), "deserializeMetadata");
var collectBodyString = /* @__PURE__ */ __name((streamBody, context) => (0, import_smithy_client.collectBody)(streamBody, context).then((body) => context.utf8Encoder(body)), "collectBodyString");
var throwDefaultError = (0, import_smithy_client.withBaseException)(SnowballServiceException);
var buildHttpRpcRequest = /* @__PURE__ */ __name(async (context, headers, path, resolvedHostname, body) => {
  const { hostname, protocol = "https", port, path: basePath } = await context.endpoint();
  const contents = {
    protocol,
    hostname,
    port,
    method: "POST",
    path: basePath.endsWith("/") ? basePath.slice(0, -1) + path : basePath + path,
    headers
  };
  if (resolvedHostname !== void 0) {
    contents.hostname = resolvedHostname;
  }
  if (body !== void 0) {
    contents.body = body;
  }
  return new import_protocol_http.HttpRequest(contents);
}, "buildHttpRpcRequest");
function sharedHeaders(operation) {
  return {
    "content-type": "application/x-amz-json-1.1",
    "x-amz-target": `AWSIESnowballJobManagementService.${operation}`
  };
}
__name(sharedHeaders, "sharedHeaders");
var parseBody = /* @__PURE__ */ __name((streamBody, context) => collectBodyString(streamBody, context).then((encoded) => {
  if (encoded.length) {
    return JSON.parse(encoded);
  }
  return {};
}), "parseBody");
var parseErrorBody = /* @__PURE__ */ __name(async (errorBody, context) => {
  const value = await parseBody(errorBody, context);
  value.message = value.message ?? value.Message;
  return value;
}, "parseErrorBody");
var loadRestJsonErrorCode = /* @__PURE__ */ __name((output, data) => {
  const findKey = /* @__PURE__ */ __name((object, key) => Object.keys(object).find((k) => k.toLowerCase() === key.toLowerCase()), "findKey");
  const sanitizeErrorCode = /* @__PURE__ */ __name((rawValue) => {
    let cleanValue = rawValue;
    if (typeof cleanValue === "number") {
      cleanValue = cleanValue.toString();
    }
    if (cleanValue.indexOf(",") >= 0) {
      cleanValue = cleanValue.split(",")[0];
    }
    if (cleanValue.indexOf(":") >= 0) {
      cleanValue = cleanValue.split(":")[0];
    }
    if (cleanValue.indexOf("#") >= 0) {
      cleanValue = cleanValue.split("#")[1];
    }
    return cleanValue;
  }, "sanitizeErrorCode");
  const headerKey = findKey(output.headers, "x-amzn-errortype");
  if (headerKey !== void 0) {
    return sanitizeErrorCode(output.headers[headerKey]);
  }
  if (data.code !== void 0) {
    return sanitizeErrorCode(data.code);
  }
  if (data["__type"] !== void 0) {
    return sanitizeErrorCode(data["__type"]);
  }
}, "loadRestJsonErrorCode");

// src/commands/CancelClusterCommand.ts
var _CancelClusterCommand = class _CancelClusterCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AWSIESnowballJobManagementService", "CancelCluster", {}).n("SnowballClient", "CancelClusterCommand").f(void 0, void 0).ser(se_CancelClusterCommand).de(de_CancelClusterCommand).build() {
};
__name(_CancelClusterCommand, "CancelClusterCommand");
var CancelClusterCommand = _CancelClusterCommand;

// src/commands/CancelJobCommand.ts




var _CancelJobCommand = class _CancelJobCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AWSIESnowballJobManagementService", "CancelJob", {}).n("SnowballClient", "CancelJobCommand").f(void 0, void 0).ser(se_CancelJobCommand).de(de_CancelJobCommand).build() {
};
__name(_CancelJobCommand, "CancelJobCommand");
var CancelJobCommand = _CancelJobCommand;

// src/commands/CreateAddressCommand.ts




var _CreateAddressCommand = class _CreateAddressCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AWSIESnowballJobManagementService", "CreateAddress", {}).n("SnowballClient", "CreateAddressCommand").f(void 0, void 0).ser(se_CreateAddressCommand).de(de_CreateAddressCommand).build() {
};
__name(_CreateAddressCommand, "CreateAddressCommand");
var CreateAddressCommand = _CreateAddressCommand;

// src/commands/CreateClusterCommand.ts




var _CreateClusterCommand = class _CreateClusterCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AWSIESnowballJobManagementService", "CreateCluster", {}).n("SnowballClient", "CreateClusterCommand").f(void 0, void 0).ser(se_CreateClusterCommand).de(de_CreateClusterCommand).build() {
};
__name(_CreateClusterCommand, "CreateClusterCommand");
var CreateClusterCommand = _CreateClusterCommand;

// src/commands/CreateJobCommand.ts




var _CreateJobCommand = class _CreateJobCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AWSIESnowballJobManagementService", "CreateJob", {}).n("SnowballClient", "CreateJobCommand").f(CreateJobRequestFilterSensitiveLog, void 0).ser(se_CreateJobCommand).de(de_CreateJobCommand).build() {
};
__name(_CreateJobCommand, "CreateJobCommand");
var CreateJobCommand = _CreateJobCommand;

// src/commands/CreateLongTermPricingCommand.ts




var _CreateLongTermPricingCommand = class _CreateLongTermPricingCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AWSIESnowballJobManagementService", "CreateLongTermPricing", {}).n("SnowballClient", "CreateLongTermPricingCommand").f(void 0, void 0).ser(se_CreateLongTermPricingCommand).de(de_CreateLongTermPricingCommand).build() {
};
__name(_CreateLongTermPricingCommand, "CreateLongTermPricingCommand");
var CreateLongTermPricingCommand = _CreateLongTermPricingCommand;

// src/commands/CreateReturnShippingLabelCommand.ts




var _CreateReturnShippingLabelCommand = class _CreateReturnShippingLabelCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AWSIESnowballJobManagementService", "CreateReturnShippingLabel", {}).n("SnowballClient", "CreateReturnShippingLabelCommand").f(void 0, void 0).ser(se_CreateReturnShippingLabelCommand).de(de_CreateReturnShippingLabelCommand).build() {
};
__name(_CreateReturnShippingLabelCommand, "CreateReturnShippingLabelCommand");
var CreateReturnShippingLabelCommand = _CreateReturnShippingLabelCommand;

// src/commands/DescribeAddressCommand.ts




var _DescribeAddressCommand = class _DescribeAddressCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AWSIESnowballJobManagementService", "DescribeAddress", {}).n("SnowballClient", "DescribeAddressCommand").f(void 0, void 0).ser(se_DescribeAddressCommand).de(de_DescribeAddressCommand).build() {
};
__name(_DescribeAddressCommand, "DescribeAddressCommand");
var DescribeAddressCommand = _DescribeAddressCommand;

// src/commands/DescribeAddressesCommand.ts




var _DescribeAddressesCommand = class _DescribeAddressesCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AWSIESnowballJobManagementService", "DescribeAddresses", {}).n("SnowballClient", "DescribeAddressesCommand").f(void 0, void 0).ser(se_DescribeAddressesCommand).de(de_DescribeAddressesCommand).build() {
};
__name(_DescribeAddressesCommand, "DescribeAddressesCommand");
var DescribeAddressesCommand = _DescribeAddressesCommand;

// src/commands/DescribeClusterCommand.ts




var _DescribeClusterCommand = class _DescribeClusterCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AWSIESnowballJobManagementService", "DescribeCluster", {}).n("SnowballClient", "DescribeClusterCommand").f(void 0, void 0).ser(se_DescribeClusterCommand).de(de_DescribeClusterCommand).build() {
};
__name(_DescribeClusterCommand, "DescribeClusterCommand");
var DescribeClusterCommand = _DescribeClusterCommand;

// src/commands/DescribeJobCommand.ts




var _DescribeJobCommand = class _DescribeJobCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AWSIESnowballJobManagementService", "DescribeJob", {}).n("SnowballClient", "DescribeJobCommand").f(void 0, DescribeJobResultFilterSensitiveLog).ser(se_DescribeJobCommand).de(de_DescribeJobCommand).build() {
};
__name(_DescribeJobCommand, "DescribeJobCommand");
var DescribeJobCommand = _DescribeJobCommand;

// src/commands/DescribeReturnShippingLabelCommand.ts




var _DescribeReturnShippingLabelCommand = class _DescribeReturnShippingLabelCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AWSIESnowballJobManagementService", "DescribeReturnShippingLabel", {}).n("SnowballClient", "DescribeReturnShippingLabelCommand").f(void 0, void 0).ser(se_DescribeReturnShippingLabelCommand).de(de_DescribeReturnShippingLabelCommand).build() {
};
__name(_DescribeReturnShippingLabelCommand, "DescribeReturnShippingLabelCommand");
var DescribeReturnShippingLabelCommand = _DescribeReturnShippingLabelCommand;

// src/commands/GetJobManifestCommand.ts




var _GetJobManifestCommand = class _GetJobManifestCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AWSIESnowballJobManagementService", "GetJobManifest", {}).n("SnowballClient", "GetJobManifestCommand").f(void 0, void 0).ser(se_GetJobManifestCommand).de(de_GetJobManifestCommand).build() {
};
__name(_GetJobManifestCommand, "GetJobManifestCommand");
var GetJobManifestCommand = _GetJobManifestCommand;

// src/commands/GetJobUnlockCodeCommand.ts




var _GetJobUnlockCodeCommand = class _GetJobUnlockCodeCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AWSIESnowballJobManagementService", "GetJobUnlockCode", {}).n("SnowballClient", "GetJobUnlockCodeCommand").f(void 0, void 0).ser(se_GetJobUnlockCodeCommand).de(de_GetJobUnlockCodeCommand).build() {
};
__name(_GetJobUnlockCodeCommand, "GetJobUnlockCodeCommand");
var GetJobUnlockCodeCommand = _GetJobUnlockCodeCommand;

// src/commands/GetSnowballUsageCommand.ts




var _GetSnowballUsageCommand = class _GetSnowballUsageCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AWSIESnowballJobManagementService", "GetSnowballUsage", {}).n("SnowballClient", "GetSnowballUsageCommand").f(void 0, void 0).ser(se_GetSnowballUsageCommand).de(de_GetSnowballUsageCommand).build() {
};
__name(_GetSnowballUsageCommand, "GetSnowballUsageCommand");
var GetSnowballUsageCommand = _GetSnowballUsageCommand;

// src/commands/GetSoftwareUpdatesCommand.ts




var _GetSoftwareUpdatesCommand = class _GetSoftwareUpdatesCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AWSIESnowballJobManagementService", "GetSoftwareUpdates", {}).n("SnowballClient", "GetSoftwareUpdatesCommand").f(void 0, void 0).ser(se_GetSoftwareUpdatesCommand).de(de_GetSoftwareUpdatesCommand).build() {
};
__name(_GetSoftwareUpdatesCommand, "GetSoftwareUpdatesCommand");
var GetSoftwareUpdatesCommand = _GetSoftwareUpdatesCommand;

// src/commands/ListClusterJobsCommand.ts




var _ListClusterJobsCommand = class _ListClusterJobsCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AWSIESnowballJobManagementService", "ListClusterJobs", {}).n("SnowballClient", "ListClusterJobsCommand").f(void 0, void 0).ser(se_ListClusterJobsCommand).de(de_ListClusterJobsCommand).build() {
};
__name(_ListClusterJobsCommand, "ListClusterJobsCommand");
var ListClusterJobsCommand = _ListClusterJobsCommand;

// src/commands/ListClustersCommand.ts




var _ListClustersCommand = class _ListClustersCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AWSIESnowballJobManagementService", "ListClusters", {}).n("SnowballClient", "ListClustersCommand").f(void 0, void 0).ser(se_ListClustersCommand).de(de_ListClustersCommand).build() {
};
__name(_ListClustersCommand, "ListClustersCommand");
var ListClustersCommand = _ListClustersCommand;

// src/commands/ListCompatibleImagesCommand.ts




var _ListCompatibleImagesCommand = class _ListCompatibleImagesCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AWSIESnowballJobManagementService", "ListCompatibleImages", {}).n("SnowballClient", "ListCompatibleImagesCommand").f(void 0, void 0).ser(se_ListCompatibleImagesCommand).de(de_ListCompatibleImagesCommand).build() {
};
__name(_ListCompatibleImagesCommand, "ListCompatibleImagesCommand");
var ListCompatibleImagesCommand = _ListCompatibleImagesCommand;

// src/commands/ListJobsCommand.ts




var _ListJobsCommand = class _ListJobsCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AWSIESnowballJobManagementService", "ListJobs", {}).n("SnowballClient", "ListJobsCommand").f(void 0, void 0).ser(se_ListJobsCommand).de(de_ListJobsCommand).build() {
};
__name(_ListJobsCommand, "ListJobsCommand");
var ListJobsCommand = _ListJobsCommand;

// src/commands/ListLongTermPricingCommand.ts




var _ListLongTermPricingCommand = class _ListLongTermPricingCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AWSIESnowballJobManagementService", "ListLongTermPricing", {}).n("SnowballClient", "ListLongTermPricingCommand").f(void 0, void 0).ser(se_ListLongTermPricingCommand).de(de_ListLongTermPricingCommand).build() {
};
__name(_ListLongTermPricingCommand, "ListLongTermPricingCommand");
var ListLongTermPricingCommand = _ListLongTermPricingCommand;

// src/commands/ListPickupLocationsCommand.ts




var _ListPickupLocationsCommand = class _ListPickupLocationsCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AWSIESnowballJobManagementService", "ListPickupLocations", {}).n("SnowballClient", "ListPickupLocationsCommand").f(void 0, void 0).ser(se_ListPickupLocationsCommand).de(de_ListPickupLocationsCommand).build() {
};
__name(_ListPickupLocationsCommand, "ListPickupLocationsCommand");
var ListPickupLocationsCommand = _ListPickupLocationsCommand;

// src/commands/ListServiceVersionsCommand.ts




var _ListServiceVersionsCommand = class _ListServiceVersionsCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AWSIESnowballJobManagementService", "ListServiceVersions", {}).n("SnowballClient", "ListServiceVersionsCommand").f(void 0, void 0).ser(se_ListServiceVersionsCommand).de(de_ListServiceVersionsCommand).build() {
};
__name(_ListServiceVersionsCommand, "ListServiceVersionsCommand");
var ListServiceVersionsCommand = _ListServiceVersionsCommand;

// src/commands/UpdateClusterCommand.ts




var _UpdateClusterCommand = class _UpdateClusterCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AWSIESnowballJobManagementService", "UpdateCluster", {}).n("SnowballClient", "UpdateClusterCommand").f(void 0, void 0).ser(se_UpdateClusterCommand).de(de_UpdateClusterCommand).build() {
};
__name(_UpdateClusterCommand, "UpdateClusterCommand");
var UpdateClusterCommand = _UpdateClusterCommand;

// src/commands/UpdateJobCommand.ts




var _UpdateJobCommand = class _UpdateJobCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AWSIESnowballJobManagementService", "UpdateJob", {}).n("SnowballClient", "UpdateJobCommand").f(UpdateJobRequestFilterSensitiveLog, void 0).ser(se_UpdateJobCommand).de(de_UpdateJobCommand).build() {
};
__name(_UpdateJobCommand, "UpdateJobCommand");
var UpdateJobCommand = _UpdateJobCommand;

// src/commands/UpdateJobShipmentStateCommand.ts




var _UpdateJobShipmentStateCommand = class _UpdateJobShipmentStateCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AWSIESnowballJobManagementService", "UpdateJobShipmentState", {}).n("SnowballClient", "UpdateJobShipmentStateCommand").f(void 0, void 0).ser(se_UpdateJobShipmentStateCommand).de(de_UpdateJobShipmentStateCommand).build() {
};
__name(_UpdateJobShipmentStateCommand, "UpdateJobShipmentStateCommand");
var UpdateJobShipmentStateCommand = _UpdateJobShipmentStateCommand;

// src/commands/UpdateLongTermPricingCommand.ts




var _UpdateLongTermPricingCommand = class _UpdateLongTermPricingCommand extends import_smithy_client.Command.classBuilder().ep({
  ...commonParams
}).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AWSIESnowballJobManagementService", "UpdateLongTermPricing", {}).n("SnowballClient", "UpdateLongTermPricingCommand").f(void 0, void 0).ser(se_UpdateLongTermPricingCommand).de(de_UpdateLongTermPricingCommand).build() {
};
__name(_UpdateLongTermPricingCommand, "UpdateLongTermPricingCommand");
var UpdateLongTermPricingCommand = _UpdateLongTermPricingCommand;

// src/Snowball.ts
var commands = {
  CancelClusterCommand,
  CancelJobCommand,
  CreateAddressCommand,
  CreateClusterCommand,
  CreateJobCommand,
  CreateLongTermPricingCommand,
  CreateReturnShippingLabelCommand,
  DescribeAddressCommand,
  DescribeAddressesCommand,
  DescribeClusterCommand,
  DescribeJobCommand,
  DescribeReturnShippingLabelCommand,
  GetJobManifestCommand,
  GetJobUnlockCodeCommand,
  GetSnowballUsageCommand,
  GetSoftwareUpdatesCommand,
  ListClusterJobsCommand,
  ListClustersCommand,
  ListCompatibleImagesCommand,
  ListJobsCommand,
  ListLongTermPricingCommand,
  ListPickupLocationsCommand,
  ListServiceVersionsCommand,
  UpdateClusterCommand,
  UpdateJobCommand,
  UpdateJobShipmentStateCommand,
  UpdateLongTermPricingCommand
};
var _Snowball = class _Snowball extends SnowballClient {
};
__name(_Snowball, "Snowball");
var Snowball = _Snowball;
(0, import_smithy_client.createAggregatedClient)(commands, Snowball);

// src/pagination/DescribeAddressesPaginator.ts

var paginateDescribeAddresses = (0, import_core.createPaginator)(SnowballClient, DescribeAddressesCommand, "NextToken", "NextToken", "MaxResults");

// src/pagination/ListClusterJobsPaginator.ts

var paginateListClusterJobs = (0, import_core.createPaginator)(SnowballClient, ListClusterJobsCommand, "NextToken", "NextToken", "MaxResults");

// src/pagination/ListClustersPaginator.ts

var paginateListClusters = (0, import_core.createPaginator)(SnowballClient, ListClustersCommand, "NextToken", "NextToken", "MaxResults");

// src/pagination/ListCompatibleImagesPaginator.ts

var paginateListCompatibleImages = (0, import_core.createPaginator)(SnowballClient, ListCompatibleImagesCommand, "NextToken", "NextToken", "MaxResults");

// src/pagination/ListJobsPaginator.ts

var paginateListJobs = (0, import_core.createPaginator)(SnowballClient, ListJobsCommand, "NextToken", "NextToken", "MaxResults");

// src/pagination/ListLongTermPricingPaginator.ts

var paginateListLongTermPricing = (0, import_core.createPaginator)(SnowballClient, ListLongTermPricingCommand, "NextToken", "NextToken", "MaxResults");

// src/pagination/ListPickupLocationsPaginator.ts

var paginateListPickupLocations = (0, import_core.createPaginator)(SnowballClient, ListPickupLocationsCommand, "NextToken", "NextToken", "MaxResults");

// src/index.ts
var import_util_endpoints = require("@aws-sdk/util-endpoints");
// Annotate the CommonJS export names for ESM import in node:

0 && (module.exports = {
  SnowballServiceException,
  __Client,
  SnowballClient,
  Snowball,
  $Command,
  CancelClusterCommand,
  CancelJobCommand,
  CreateAddressCommand,
  CreateClusterCommand,
  CreateJobCommand,
  CreateLongTermPricingCommand,
  CreateReturnShippingLabelCommand,
  DescribeAddressCommand,
  DescribeAddressesCommand,
  DescribeClusterCommand,
  DescribeJobCommand,
  DescribeReturnShippingLabelCommand,
  GetJobManifestCommand,
  GetJobUnlockCodeCommand,
  GetSnowballUsageCommand,
  GetSoftwareUpdatesCommand,
  ListClusterJobsCommand,
  ListClustersCommand,
  ListCompatibleImagesCommand,
  ListJobsCommand,
  ListLongTermPricingCommand,
  ListPickupLocationsCommand,
  ListServiceVersionsCommand,
  UpdateClusterCommand,
  UpdateJobCommand,
  UpdateJobShipmentStateCommand,
  UpdateLongTermPricingCommand,
  paginateDescribeAddresses,
  paginateListClusterJobs,
  paginateListClusters,
  paginateListCompatibleImages,
  paginateListJobs,
  paginateListLongTermPricing,
  paginateListPickupLocations,
  AddressType,
  InvalidJobStateException,
  InvalidResourceException,
  KMSRequestFailedException,
  InvalidAddressException,
  UnsupportedAddressException,
  JobType,
  JobState,
  StorageUnit,
  RemoteManagement,
  DeviceServiceName,
  TransferOption,
  ShippingOption,
  SnowballCapacity,
  SnowballType,
  Ec2RequestFailedException,
  InvalidInputCombinationException,
  ClusterLimitExceededException,
  ImpactLevel,
  LongTermPricingType,
  ConflictException,
  ShippingLabelStatus,
  ReturnShippingLabelAlreadyExistsException,
  InvalidNextTokenException,
  ClusterState,
  ServiceName,
  ShipmentState,
  PickupDetailsFilterSensitiveLog,
  CreateJobRequestFilterSensitiveLog,
  JobMetadataFilterSensitiveLog,
  DescribeJobResultFilterSensitiveLog,
  UpdateJobRequestFilterSensitiveLog
});
