// @azure/arm-dynatrace/dist-esm/src/models/index.js
/*
 * Copyright (c) Microsoft Corporation.
 * Licensed under the MIT License.
 *
 * Code generated by Microsoft (R) AutoRest Code Generator.
 * Changes may cause incorrect behavior and will be lost if the code is regenerated.
 */
/** Known values of {@link SendingMetricsStatus} that the service accepts. */
export var KnownSendingMetricsStatus;
(function (KnownSendingMetricsStatus) {
    /** Enabled */
    KnownSendingMetricsStatus["Enabled"] = "Enabled";
    /** Disabled */
    KnownSendingMetricsStatus["Disabled"] = "Disabled";
})(KnownSendingMetricsStatus || (KnownSendingMetricsStatus = {}));
/** Known values of {@link SendingLogsStatus} that the service accepts. */
export var KnownSendingLogsStatus;
(function (KnownSendingLogsStatus) {
    /** Enabled */
    KnownSendingLogsStatus["Enabled"] = "Enabled";
    /** Disabled */
    KnownSendingLogsStatus["Disabled"] = "Disabled";
})(KnownSendingLogsStatus || (KnownSendingLogsStatus = {}));
/** Known values of {@link MonitoringStatus} that the service accepts. */
export var KnownMonitoringStatus;
(function (KnownMonitoringStatus) {
    /** Enabled */
    KnownMonitoringStatus["Enabled"] = "Enabled";
    /** Disabled */
    KnownMonitoringStatus["Disabled"] = "Disabled";
})(KnownMonitoringStatus || (KnownMonitoringStatus = {}));
/** Known values of {@link MarketplaceSubscriptionStatus} that the service accepts. */
export var KnownMarketplaceSubscriptionStatus;
(function (KnownMarketplaceSubscriptionStatus) {
    /** Active */
    KnownMarketplaceSubscriptionStatus["Active"] = "Active";
    /** Suspended */
    KnownMarketplaceSubscriptionStatus["Suspended"] = "Suspended";
})(KnownMarketplaceSubscriptionStatus || (KnownMarketplaceSubscriptionStatus = {}));
/** Known values of {@link SingleSignOnStates} that the service accepts. */
export var KnownSingleSignOnStates;
(function (KnownSingleSignOnStates) {
    /** Initial */
    KnownSingleSignOnStates["Initial"] = "Initial";
    /** Enable */
    KnownSingleSignOnStates["Enable"] = "Enable";
    /** Disable */
    KnownSingleSignOnStates["Disable"] = "Disable";
    /** Existing */
    KnownSingleSignOnStates["Existing"] = "Existing";
})(KnownSingleSignOnStates || (KnownSingleSignOnStates = {}));
/** Known values of {@link ProvisioningState} that the service accepts. */
export var KnownProvisioningState;
(function (KnownProvisioningState) {
    /** Accepted */
    KnownProvisioningState["Accepted"] = "Accepted";
    /** Creating */
    KnownProvisioningState["Creating"] = "Creating";
    /** Updating */
    KnownProvisioningState["Updating"] = "Updating";
    /** Deleting */
    KnownProvisioningState["Deleting"] = "Deleting";
    /** Succeeded */
    KnownProvisioningState["Succeeded"] = "Succeeded";
    /** Failed */
    KnownProvisioningState["Failed"] = "Failed";
    /** Canceled */
    KnownProvisioningState["Canceled"] = "Canceled";
    /** Deleted */
    KnownProvisioningState["Deleted"] = "Deleted";
    /** NotSpecified */
    KnownProvisioningState["NotSpecified"] = "NotSpecified";
})(KnownProvisioningState || (KnownProvisioningState = {}));
/** Known values of {@link LiftrResourceCategories} that the service accepts. */
export var KnownLiftrResourceCategories;
(function (KnownLiftrResourceCategories) {
    /** Unknown */
    KnownLiftrResourceCategories["Unknown"] = "Unknown";
    /** MonitorLogs */
    KnownLiftrResourceCategories["MonitorLogs"] = "MonitorLogs";
})(KnownLiftrResourceCategories || (KnownLiftrResourceCategories = {}));
/** Known values of {@link CreatedByType} that the service accepts. */
export var KnownCreatedByType;
(function (KnownCreatedByType) {
    /** User */
    KnownCreatedByType["User"] = "User";
    /** Application */
    KnownCreatedByType["Application"] = "Application";
    /** ManagedIdentity */
    KnownCreatedByType["ManagedIdentity"] = "ManagedIdentity";
    /** Key */
    KnownCreatedByType["Key"] = "Key";
})(KnownCreatedByType || (KnownCreatedByType = {}));
/** Known values of {@link ManagedIdentityType} that the service accepts. */
export var KnownManagedIdentityType;
(function (KnownManagedIdentityType) {
    /** SystemAssigned */
    KnownManagedIdentityType["SystemAssigned"] = "SystemAssigned";
    /** UserAssigned */
    KnownManagedIdentityType["UserAssigned"] = "UserAssigned";
    /** SystemAndUserAssigned */
    KnownManagedIdentityType["SystemAndUserAssigned"] = "SystemAndUserAssigned";
})(KnownManagedIdentityType || (KnownManagedIdentityType = {}));
/** Known values of {@link Origin} that the service accepts. */
export var KnownOrigin;
(function (KnownOrigin) {
    /** User */
    KnownOrigin["User"] = "user";
    /** System */
    KnownOrigin["System"] = "system";
    /** UserSystem */
    KnownOrigin["UserSystem"] = "user,system";
})(KnownOrigin || (KnownOrigin = {}));
/** Known values of {@link ActionType} that the service accepts. */
export var KnownActionType;
(function (KnownActionType) {
    /** Internal */
    KnownActionType["Internal"] = "Internal";
})(KnownActionType || (KnownActionType = {}));
/** Known values of {@link SendAadLogsStatus} that the service accepts. */
export var KnownSendAadLogsStatus;
(function (KnownSendAadLogsStatus) {
    /** Enabled */
    KnownSendAadLogsStatus["Enabled"] = "Enabled";
    /** Disabled */
    KnownSendAadLogsStatus["Disabled"] = "Disabled";
})(KnownSendAadLogsStatus || (KnownSendAadLogsStatus = {}));
/** Known values of {@link SendSubscriptionLogsStatus} that the service accepts. */
export var KnownSendSubscriptionLogsStatus;
(function (KnownSendSubscriptionLogsStatus) {
    /** Enabled */
    KnownSendSubscriptionLogsStatus["Enabled"] = "Enabled";
    /** Disabled */
    KnownSendSubscriptionLogsStatus["Disabled"] = "Disabled";
})(KnownSendSubscriptionLogsStatus || (KnownSendSubscriptionLogsStatus = {}));
/** Known values of {@link SendActivityLogsStatus} that the service accepts. */
export var KnownSendActivityLogsStatus;
(function (KnownSendActivityLogsStatus) {
    /** Enabled */
    KnownSendActivityLogsStatus["Enabled"] = "Enabled";
    /** Disabled */
    KnownSendActivityLogsStatus["Disabled"] = "Disabled";
})(KnownSendActivityLogsStatus || (KnownSendActivityLogsStatus = {}));
/** Known values of {@link TagAction} that the service accepts. */
export var KnownTagAction;
(function (KnownTagAction) {
    /** Include */
    KnownTagAction["Include"] = "Include";
    /** Exclude */
    KnownTagAction["Exclude"] = "Exclude";
})(KnownTagAction || (KnownTagAction = {}));
/** Known values of {@link MonitoringType} that the service accepts. */
export var KnownMonitoringType;
(function (KnownMonitoringType) {
    /** CloudInfrastructure */
    KnownMonitoringType["CloudInfrastructure"] = "CLOUD_INFRASTRUCTURE";
    /** FullStack */
    KnownMonitoringType["FullStack"] = "FULL_STACK";
})(KnownMonitoringType || (KnownMonitoringType = {}));
/** Known values of {@link AutoUpdateSetting} that the service accepts. */
export var KnownAutoUpdateSetting;
(function (KnownAutoUpdateSetting) {
    /** Enabled */
    KnownAutoUpdateSetting["Enabled"] = "ENABLED";
    /** Disabled */
    KnownAutoUpdateSetting["Disabled"] = "DISABLED";
})(KnownAutoUpdateSetting || (KnownAutoUpdateSetting = {}));
/** Known values of {@link UpdateStatus} that the service accepts. */
export var KnownUpdateStatus;
(function (KnownUpdateStatus) {
    /** Incompatible */
    KnownUpdateStatus["Incompatible"] = "INCOMPATIBLE";
    /** Outdated */
    KnownUpdateStatus["Outdated"] = "OUTDATED";
    /** Scheduled */
    KnownUpdateStatus["Scheduled"] = "SCHEDULED";
    /** Suppressed */
    KnownUpdateStatus["Suppressed"] = "SUPPRESSED";
    /** Unknown */
    KnownUpdateStatus["Unknown"] = "UNKNOWN";
    /** UP2Date */
    KnownUpdateStatus["UP2Date"] = "UP2DATE";
    /** UpdateINProgress */
    KnownUpdateStatus["UpdateINProgress"] = "UPDATE_IN_PROGRESS";
    /** UpdatePending */
    KnownUpdateStatus["UpdatePending"] = "UPDATE_PENDING";
    /** UpdateProblem */
    KnownUpdateStatus["UpdateProblem"] = "UPDATE_PROBLEM";
})(KnownUpdateStatus || (KnownUpdateStatus = {}));
/** Known values of {@link AvailabilityState} that the service accepts. */
export var KnownAvailabilityState;
(function (KnownAvailabilityState) {
    /** Crashed */
    KnownAvailabilityState["Crashed"] = "CRASHED";
    /** Lost */
    KnownAvailabilityState["Lost"] = "LOST";
    /** Monitored */
    KnownAvailabilityState["Monitored"] = "MONITORED";
    /** PREMonitored */
    KnownAvailabilityState["PREMonitored"] = "PRE_MONITORED";
    /** Shutdown */
    KnownAvailabilityState["Shutdown"] = "SHUTDOWN";
    /** UnexpectedShutdown */
    KnownAvailabilityState["UnexpectedShutdown"] = "UNEXPECTED_SHUTDOWN";
    /** Unknown */
    KnownAvailabilityState["Unknown"] = "UNKNOWN";
    /** Unmonitored */
    KnownAvailabilityState["Unmonitored"] = "UNMONITORED";
})(KnownAvailabilityState || (KnownAvailabilityState = {}));
/** Known values of {@link LogModule} that the service accepts. */
export var KnownLogModule;
(function (KnownLogModule) {
    /** Enabled */
    KnownLogModule["Enabled"] = "ENABLED";
    /** Disabled */
    KnownLogModule["Disabled"] = "DISABLED";
})(KnownLogModule || (KnownLogModule = {}));
/** Known values of {@link SSOStatus} that the service accepts. */
export var KnownSSOStatus;
(function (KnownSSOStatus) {
    /** Enabled */
    KnownSSOStatus["Enabled"] = "Enabled";
    /** Disabled */
    KnownSSOStatus["Disabled"] = "Disabled";
})(KnownSSOStatus || (KnownSSOStatus = {}));
