// @azure/arm-dynatrace/dist-esm/src/models/parameters.js
/*
 * Copyright (c) Microsoft Corporation.
 * Licensed under the MIT License.
 *
 * Code generated by Microsoft (R) AutoRest Code Generator.
 * Changes may cause incorrect behavior and will be lost if the code is regenerated.
 */
import { MonitorResource as MonitorResourceMapper, MonitorResourceUpdate as MonitorResourceUpdateMapper, MarketplaceSaaSResourceDetailsRequest as MarketplaceSaaSResourceDetailsRequestMapper, SSODetailsRequest as SSODetailsRequestMapper, LinkableEnvironmentRequest as LinkableEnvironmentRequestMapper, TagRule as TagRuleMapper, DynatraceSingleSignOnResource as DynatraceSingleSignOnResourceMapper } from "../models/mappers";
export const accept = {
    parameterPath: "accept",
    mapper: {
        defaultValue: "application/json",
        isConstant: true,
        serializedName: "Accept",
        type: {
            name: "String"
        }
    }
};
export const $host = {
    parameterPath: "$host",
    mapper: {
        serializedName: "$host",
        required: true,
        type: {
            name: "String"
        }
    },
    skipEncoding: true
};
export const apiVersion = {
    parameterPath: "apiVersion",
    mapper: {
        defaultValue: "2023-04-27",
        isConstant: true,
        serializedName: "api-version",
        type: {
            name: "String"
        }
    }
};
export const subscriptionId = {
    parameterPath: "subscriptionId",
    mapper: {
        constraints: {
            MinLength: 1
        },
        serializedName: "subscriptionId",
        required: true,
        type: {
            name: "String"
        }
    }
};
export const resourceGroupName = {
    parameterPath: "resourceGroupName",
    mapper: {
        constraints: {
            MaxLength: 90,
            MinLength: 1
        },
        serializedName: "resourceGroupName",
        required: true,
        type: {
            name: "String"
        }
    }
};
export const monitorName = {
    parameterPath: "monitorName",
    mapper: {
        serializedName: "monitorName",
        required: true,
        type: {
            name: "String"
        }
    }
};
export const contentType = {
    parameterPath: ["options", "contentType"],
    mapper: {
        defaultValue: "application/json",
        isConstant: true,
        serializedName: "Content-Type",
        type: {
            name: "String"
        }
    }
};
export const resource = {
    parameterPath: "resource",
    mapper: MonitorResourceMapper
};
export const resource1 = {
    parameterPath: "resource",
    mapper: MonitorResourceUpdateMapper
};
export const request = {
    parameterPath: "request",
    mapper: MarketplaceSaaSResourceDetailsRequestMapper
};
export const request1 = {
    parameterPath: ["options", "request"],
    mapper: SSODetailsRequestMapper
};
export const request2 = {
    parameterPath: "request",
    mapper: LinkableEnvironmentRequestMapper
};
export const nextLink = {
    parameterPath: "nextLink",
    mapper: {
        serializedName: "nextLink",
        required: true,
        type: {
            name: "String"
        }
    },
    skipEncoding: true
};
export const ruleSetName = {
    parameterPath: "ruleSetName",
    mapper: {
        serializedName: "ruleSetName",
        required: true,
        type: {
            name: "String"
        }
    }
};
export const resource2 = {
    parameterPath: "resource",
    mapper: TagRuleMapper
};
export const resource3 = {
    parameterPath: "resource",
    mapper: DynatraceSingleSignOnResourceMapper
};
export const configurationName = {
    parameterPath: "configurationName",
    mapper: {
        serializedName: "configurationName",
        required: true,
        type: {
            name: "String"
        }
    }
};
