// @azure/arm-resourcehealth/dist-esm/src/operations/securityAdvisoryImpactedResources.js
/*
 * Copyright (c) Microsoft Corporation.
 * Licensed under the MIT License.
 *
 * Code generated by Microsoft (R) AutoRest Code Generator.
 * Changes may cause incorrect behavior and will be lost if the code is regenerated.
 */
import { __asyncDelegator, __asyncGenerator, __asyncValues, __await } from "tslib";
import { setContinuationToken } from "../pagingHelper";
import * as coreClient from "@azure/core-client";
import * as Mappers from "../models/mappers";
import * as Parameters from "../models/parameters";
/// <reference lib="esnext.asynciterable" />
/** Class containing SecurityAdvisoryImpactedResources operations. */
export class SecurityAdvisoryImpactedResourcesImpl {
    /**
     * Initialize a new instance of the class SecurityAdvisoryImpactedResources class.
     * @param client Reference to the service client
     */
    constructor(client) {
        this.client = client;
    }
    /**
     * Lists impacted resources in the subscription by an event (Security Advisory).
     * @param eventTrackingId Event Id which uniquely identifies ServiceHealth event.
     * @param options The options parameters.
     */
    listBySubscriptionIdAndEventId(eventTrackingId, options) {
        const iter = this.listBySubscriptionIdAndEventIdPagingAll(eventTrackingId, options);
        return {
            next() {
                return iter.next();
            },
            [Symbol.asyncIterator]() {
                return this;
            },
            byPage: (settings) => {
                if (settings === null || settings === void 0 ? void 0 : settings.maxPageSize) {
                    throw new Error("maxPageSize is not supported by this operation.");
                }
                return this.listBySubscriptionIdAndEventIdPagingPage(eventTrackingId, options, settings);
            }
        };
    }
    listBySubscriptionIdAndEventIdPagingPage(eventTrackingId, options, settings) {
        return __asyncGenerator(this, arguments, function* listBySubscriptionIdAndEventIdPagingPage_1() {
            let result;
            let continuationToken = settings === null || settings === void 0 ? void 0 : settings.continuationToken;
            if (!continuationToken) {
                result = yield __await(this._listBySubscriptionIdAndEventId(eventTrackingId, options));
                let page = result.value || [];
                continuationToken = result.nextLink;
                setContinuationToken(page, continuationToken);
                yield yield __await(page);
            }
            while (continuationToken) {
                result = yield __await(this._listBySubscriptionIdAndEventIdNext(eventTrackingId, continuationToken, options));
                continuationToken = result.nextLink;
                let page = result.value || [];
                setContinuationToken(page, continuationToken);
                yield yield __await(page);
            }
        });
    }
    listBySubscriptionIdAndEventIdPagingAll(eventTrackingId, options) {
        return __asyncGenerator(this, arguments, function* listBySubscriptionIdAndEventIdPagingAll_1() {
            var _a, e_1, _b, _c;
            try {
                for (var _d = true, _e = __asyncValues(this.listBySubscriptionIdAndEventIdPagingPage(eventTrackingId, options)), _f; _f = yield __await(_e.next()), _a = _f.done, !_a;) {
                    _c = _f.value;
                    _d = false;
                    try {
                        const page = _c;
                        yield __await(yield* __asyncDelegator(__asyncValues(page)));
                    }
                    finally {
                        _d = true;
                    }
                }
            }
            catch (e_1_1) { e_1 = { error: e_1_1 }; }
            finally {
                try {
                    if (!_d && !_a && (_b = _e.return)) yield __await(_b.call(_e));
                }
                finally { if (e_1) throw e_1.error; }
            }
        });
    }
    /**
     * Lists impacted resources in the tenant by an event (Security Advisory).
     * @param eventTrackingId Event Id which uniquely identifies ServiceHealth event.
     * @param options The options parameters.
     */
    listByTenantIdAndEventId(eventTrackingId, options) {
        const iter = this.listByTenantIdAndEventIdPagingAll(eventTrackingId, options);
        return {
            next() {
                return iter.next();
            },
            [Symbol.asyncIterator]() {
                return this;
            },
            byPage: (settings) => {
                if (settings === null || settings === void 0 ? void 0 : settings.maxPageSize) {
                    throw new Error("maxPageSize is not supported by this operation.");
                }
                return this.listByTenantIdAndEventIdPagingPage(eventTrackingId, options, settings);
            }
        };
    }
    listByTenantIdAndEventIdPagingPage(eventTrackingId, options, settings) {
        return __asyncGenerator(this, arguments, function* listByTenantIdAndEventIdPagingPage_1() {
            let result;
            let continuationToken = settings === null || settings === void 0 ? void 0 : settings.continuationToken;
            if (!continuationToken) {
                result = yield __await(this._listByTenantIdAndEventId(eventTrackingId, options));
                let page = result.value || [];
                continuationToken = result.nextLink;
                setContinuationToken(page, continuationToken);
                yield yield __await(page);
            }
            while (continuationToken) {
                result = yield __await(this._listByTenantIdAndEventIdNext(eventTrackingId, continuationToken, options));
                continuationToken = result.nextLink;
                let page = result.value || [];
                setContinuationToken(page, continuationToken);
                yield yield __await(page);
            }
        });
    }
    listByTenantIdAndEventIdPagingAll(eventTrackingId, options) {
        return __asyncGenerator(this, arguments, function* listByTenantIdAndEventIdPagingAll_1() {
            var _a, e_2, _b, _c;
            try {
                for (var _d = true, _e = __asyncValues(this.listByTenantIdAndEventIdPagingPage(eventTrackingId, options)), _f; _f = yield __await(_e.next()), _a = _f.done, !_a;) {
                    _c = _f.value;
                    _d = false;
                    try {
                        const page = _c;
                        yield __await(yield* __asyncDelegator(__asyncValues(page)));
                    }
                    finally {
                        _d = true;
                    }
                }
            }
            catch (e_2_1) { e_2 = { error: e_2_1 }; }
            finally {
                try {
                    if (!_d && !_a && (_b = _e.return)) yield __await(_b.call(_e));
                }
                finally { if (e_2) throw e_2.error; }
            }
        });
    }
    /**
     * Lists impacted resources in the subscription by an event (Security Advisory).
     * @param eventTrackingId Event Id which uniquely identifies ServiceHealth event.
     * @param options The options parameters.
     */
    _listBySubscriptionIdAndEventId(eventTrackingId, options) {
        return this.client.sendOperationRequest({ eventTrackingId, options }, listBySubscriptionIdAndEventIdOperationSpec);
    }
    /**
     * Lists impacted resources in the tenant by an event (Security Advisory).
     * @param eventTrackingId Event Id which uniquely identifies ServiceHealth event.
     * @param options The options parameters.
     */
    _listByTenantIdAndEventId(eventTrackingId, options) {
        return this.client.sendOperationRequest({ eventTrackingId, options }, listByTenantIdAndEventIdOperationSpec);
    }
    /**
     * ListBySubscriptionIdAndEventIdNext
     * @param eventTrackingId Event Id which uniquely identifies ServiceHealth event.
     * @param nextLink The nextLink from the previous successful call to the ListBySubscriptionIdAndEventId
     *                 method.
     * @param options The options parameters.
     */
    _listBySubscriptionIdAndEventIdNext(eventTrackingId, nextLink, options) {
        return this.client.sendOperationRequest({ eventTrackingId, nextLink, options }, listBySubscriptionIdAndEventIdNextOperationSpec);
    }
    /**
     * ListByTenantIdAndEventIdNext
     * @param eventTrackingId Event Id which uniquely identifies ServiceHealth event.
     * @param nextLink The nextLink from the previous successful call to the ListByTenantIdAndEventId
     *                 method.
     * @param options The options parameters.
     */
    _listByTenantIdAndEventIdNext(eventTrackingId, nextLink, options) {
        return this.client.sendOperationRequest({ eventTrackingId, nextLink, options }, listByTenantIdAndEventIdNextOperationSpec);
    }
}
// Operation Specifications
const serializer = coreClient.createSerializer(Mappers, /* isXml */ false);
const listBySubscriptionIdAndEventIdOperationSpec = {
    path: "/subscriptions/{subscriptionId}/providers/Microsoft.ResourceHealth/events/{eventTrackingId}/listSecurityAdvisoryImpactedResources",
    httpMethod: "POST",
    responses: {
        200: {
            bodyMapper: Mappers.EventImpactedResourceListResult
        },
        default: {
            bodyMapper: Mappers.ErrorResponse
        }
    },
    queryParameters: [Parameters.apiVersion, Parameters.filter],
    urlParameters: [
        Parameters.$host,
        Parameters.subscriptionId,
        Parameters.eventTrackingId
    ],
    headerParameters: [Parameters.accept],
    serializer
};
const listByTenantIdAndEventIdOperationSpec = {
    path: "/providers/Microsoft.ResourceHealth/events/{eventTrackingId}/listSecurityAdvisoryImpactedResources",
    httpMethod: "POST",
    responses: {
        200: {
            bodyMapper: Mappers.EventImpactedResourceListResult
        },
        default: {
            bodyMapper: Mappers.ErrorResponse
        }
    },
    queryParameters: [Parameters.apiVersion, Parameters.filter],
    urlParameters: [Parameters.$host, Parameters.eventTrackingId],
    headerParameters: [Parameters.accept],
    serializer
};
const listBySubscriptionIdAndEventIdNextOperationSpec = {
    path: "{nextLink}",
    httpMethod: "GET",
    responses: {
        200: {
            bodyMapper: Mappers.EventImpactedResourceListResult
        },
        default: {
            bodyMapper: Mappers.ErrorResponse
        }
    },
    urlParameters: [
        Parameters.$host,
        Parameters.subscriptionId,
        Parameters.nextLink,
        Parameters.eventTrackingId
    ],
    headerParameters: [Parameters.accept],
    serializer
};
const listByTenantIdAndEventIdNextOperationSpec = {
    path: "{nextLink}",
    httpMethod: "GET",
    responses: {
        200: {
            bodyMapper: Mappers.EventImpactedResourceListResult
        },
        default: {
            bodyMapper: Mappers.ErrorResponse
        }
    },
    urlParameters: [
        Parameters.$host,
        Parameters.nextLink,
        Parameters.eventTrackingId
    ],
    headerParameters: [Parameters.accept],
    serializer
};
