// @azure/arm-migrate/dist-esm/src/operations/assessedMachines.js
/*
 * Copyright (c) Microsoft Corporation.
 * Licensed under the MIT License.
 *
 * Code generated by Microsoft (R) AutoRest Code Generator.
 * Changes may cause incorrect behavior and will be lost if the code is regenerated.
 */
import { __asyncDelegator, __asyncGenerator, __asyncValues, __await } from "tslib";
import { setContinuationToken } from "../pagingHelper";
import * as coreClient from "@azure/core-client";
import * as Mappers from "../models/mappers";
import * as Parameters from "../models/parameters";
/// <reference lib="esnext.asynciterable" />
/** Class containing AssessedMachines operations. */
export class AssessedMachinesImpl {
    /**
     * Initialize a new instance of the class AssessedMachines class.
     * @param client Reference to the service client
     */
    constructor(client) {
        this.client = client;
    }
    /**
     * Get list of machines that assessed as part of the specified assessment. Returns a json array of
     * objects of type 'assessedMachine' as specified in the Models section.
     *
     * Whenever an assessment is created or updated, it goes under computation. During this phase, the
     * 'status' field of Assessment object reports 'Computing'.
     * During the period when the assessment is under computation, the list of assessed machines is empty
     * and no assessed machines are returned by this call.
     *
     * @param resourceGroupName Name of the Azure Resource Group that project is part of.
     * @param projectName Name of the Azure Migrate project.
     * @param groupName Unique name of a group within a project.
     * @param assessmentName Unique name of an assessment within a project.
     * @param options The options parameters.
     */
    listByAssessment(resourceGroupName, projectName, groupName, assessmentName, options) {
        const iter = this.listByAssessmentPagingAll(resourceGroupName, projectName, groupName, assessmentName, options);
        return {
            next() {
                return iter.next();
            },
            [Symbol.asyncIterator]() {
                return this;
            },
            byPage: (settings) => {
                if (settings === null || settings === void 0 ? void 0 : settings.maxPageSize) {
                    throw new Error("maxPageSize is not supported by this operation.");
                }
                return this.listByAssessmentPagingPage(resourceGroupName, projectName, groupName, assessmentName, options, settings);
            }
        };
    }
    listByAssessmentPagingPage(resourceGroupName, projectName, groupName, assessmentName, options, settings) {
        return __asyncGenerator(this, arguments, function* listByAssessmentPagingPage_1() {
            let result;
            let continuationToken = settings === null || settings === void 0 ? void 0 : settings.continuationToken;
            if (!continuationToken) {
                result = yield __await(this._listByAssessment(resourceGroupName, projectName, groupName, assessmentName, options));
                let page = result.value || [];
                continuationToken = result.nextLink;
                setContinuationToken(page, continuationToken);
                yield yield __await(page);
            }
            while (continuationToken) {
                result = yield __await(this._listByAssessmentNext(resourceGroupName, projectName, groupName, assessmentName, continuationToken, options));
                continuationToken = result.nextLink;
                let page = result.value || [];
                setContinuationToken(page, continuationToken);
                yield yield __await(page);
            }
        });
    }
    listByAssessmentPagingAll(resourceGroupName, projectName, groupName, assessmentName, options) {
        return __asyncGenerator(this, arguments, function* listByAssessmentPagingAll_1() {
            var e_1, _a;
            try {
                for (var _b = __asyncValues(this.listByAssessmentPagingPage(resourceGroupName, projectName, groupName, assessmentName, options)), _c; _c = yield __await(_b.next()), !_c.done;) {
                    const page = _c.value;
                    yield __await(yield* __asyncDelegator(__asyncValues(page)));
                }
            }
            catch (e_1_1) { e_1 = { error: e_1_1 }; }
            finally {
                try {
                    if (_c && !_c.done && (_a = _b.return)) yield __await(_a.call(_b));
                }
                finally { if (e_1) throw e_1.error; }
            }
        });
    }
    /**
     * Get list of machines that assessed as part of the specified assessment. Returns a json array of
     * objects of type 'assessedMachine' as specified in the Models section.
     *
     * Whenever an assessment is created or updated, it goes under computation. During this phase, the
     * 'status' field of Assessment object reports 'Computing'.
     * During the period when the assessment is under computation, the list of assessed machines is empty
     * and no assessed machines are returned by this call.
     *
     * @param resourceGroupName Name of the Azure Resource Group that project is part of.
     * @param projectName Name of the Azure Migrate project.
     * @param groupName Unique name of a group within a project.
     * @param assessmentName Unique name of an assessment within a project.
     * @param options The options parameters.
     */
    _listByAssessment(resourceGroupName, projectName, groupName, assessmentName, options) {
        return this.client.sendOperationRequest({ resourceGroupName, projectName, groupName, assessmentName, options }, listByAssessmentOperationSpec);
    }
    /**
     * Get an assessed machine with its size & cost estimate that was evaluated in the specified
     * assessment.
     * @param resourceGroupName Name of the Azure Resource Group that project is part of.
     * @param projectName Name of the Azure Migrate project.
     * @param groupName Unique name of a group within a project.
     * @param assessmentName Unique name of an assessment within a project.
     * @param assessedMachineName Unique name of an assessed machine evaluated as part of an assessment.
     * @param options The options parameters.
     */
    get(resourceGroupName, projectName, groupName, assessmentName, assessedMachineName, options) {
        return this.client.sendOperationRequest({
            resourceGroupName,
            projectName,
            groupName,
            assessmentName,
            assessedMachineName,
            options
        }, getOperationSpec);
    }
    /**
     * ListByAssessmentNext
     * @param resourceGroupName Name of the Azure Resource Group that project is part of.
     * @param projectName Name of the Azure Migrate project.
     * @param groupName Unique name of a group within a project.
     * @param assessmentName Unique name of an assessment within a project.
     * @param nextLink The nextLink from the previous successful call to the ListByAssessment method.
     * @param options The options parameters.
     */
    _listByAssessmentNext(resourceGroupName, projectName, groupName, assessmentName, nextLink, options) {
        return this.client.sendOperationRequest({
            resourceGroupName,
            projectName,
            groupName,
            assessmentName,
            nextLink,
            options
        }, listByAssessmentNextOperationSpec);
    }
}
// Operation Specifications
const serializer = coreClient.createSerializer(Mappers, /* isXml */ false);
const listByAssessmentOperationSpec = {
    path: "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Migrate/assessmentProjects/{projectName}/groups/{groupName}/assessments/{assessmentName}/assessedMachines",
    httpMethod: "GET",
    responses: {
        200: {
            bodyMapper: Mappers.AssessedMachineResultList,
            headersMapper: Mappers.AssessedMachinesListByAssessmentHeaders
        },
        default: {
            bodyMapper: Mappers.CloudError
        }
    },
    queryParameters: [Parameters.apiVersion],
    urlParameters: [
        Parameters.$host,
        Parameters.subscriptionId,
        Parameters.resourceGroupName,
        Parameters.projectName,
        Parameters.groupName,
        Parameters.assessmentName
    ],
    headerParameters: [Parameters.accept],
    serializer
};
const getOperationSpec = {
    path: "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Migrate/assessmentProjects/{projectName}/groups/{groupName}/assessments/{assessmentName}/assessedMachines/{assessedMachineName}",
    httpMethod: "GET",
    responses: {
        200: {
            bodyMapper: Mappers.AssessedMachine,
            headersMapper: Mappers.AssessedMachinesGetHeaders
        },
        default: {
            bodyMapper: Mappers.CloudError
        }
    },
    queryParameters: [Parameters.apiVersion],
    urlParameters: [
        Parameters.$host,
        Parameters.subscriptionId,
        Parameters.resourceGroupName,
        Parameters.projectName,
        Parameters.groupName,
        Parameters.assessmentName,
        Parameters.assessedMachineName
    ],
    headerParameters: [Parameters.accept],
    serializer
};
const listByAssessmentNextOperationSpec = {
    path: "{nextLink}",
    httpMethod: "GET",
    responses: {
        200: {
            bodyMapper: Mappers.AssessedMachineResultList,
            headersMapper: Mappers.AssessedMachinesListByAssessmentNextHeaders
        },
        default: {
            bodyMapper: Mappers.CloudError
        }
    },
    urlParameters: [
        Parameters.$host,
        Parameters.subscriptionId,
        Parameters.resourceGroupName,
        Parameters.projectName,
        Parameters.nextLink,
        Parameters.groupName,
        Parameters.assessmentName
    ],
    headerParameters: [Parameters.accept],
    serializer
};
