// @azure/arm-migrate/dist-esm/src/operations/assessments.js
/*
 * Copyright (c) Microsoft Corporation.
 * Licensed under the MIT License.
 *
 * Code generated by Microsoft (R) AutoRest Code Generator.
 * Changes may cause incorrect behavior and will be lost if the code is regenerated.
 */
import { __asyncDelegator, __asyncGenerator, __asyncValues, __await } from "tslib";
import * as coreClient from "@azure/core-client";
import * as Mappers from "../models/mappers";
import * as Parameters from "../models/parameters";
/// <reference lib="esnext.asynciterable" />
/** Class containing Assessments operations. */
export class AssessmentsImpl {
    /**
     * Initialize a new instance of the class Assessments class.
     * @param client Reference to the service client
     */
    constructor(client) {
        this.client = client;
    }
    /**
     * Get all assessments created for the specified group.
     *
     * Returns a json array of objects of type 'assessment' as specified in Models section.
     *
     * @param resourceGroupName Name of the Azure Resource Group that project is part of.
     * @param projectName Name of the Azure Migrate project.
     * @param groupName Unique name of a group within a project.
     * @param options The options parameters.
     */
    listByGroup(resourceGroupName, projectName, groupName, options) {
        const iter = this.listByGroupPagingAll(resourceGroupName, projectName, groupName, options);
        return {
            next() {
                return iter.next();
            },
            [Symbol.asyncIterator]() {
                return this;
            },
            byPage: (settings) => {
                if (settings === null || settings === void 0 ? void 0 : settings.maxPageSize) {
                    throw new Error("maxPageSize is not supported by this operation.");
                }
                return this.listByGroupPagingPage(resourceGroupName, projectName, groupName, options, settings);
            }
        };
    }
    listByGroupPagingPage(resourceGroupName, projectName, groupName, options, _settings) {
        return __asyncGenerator(this, arguments, function* listByGroupPagingPage_1() {
            let result;
            result = yield __await(this._listByGroup(resourceGroupName, projectName, groupName, options));
            yield yield __await(result.value || []);
        });
    }
    listByGroupPagingAll(resourceGroupName, projectName, groupName, options) {
        return __asyncGenerator(this, arguments, function* listByGroupPagingAll_1() {
            var e_1, _a;
            try {
                for (var _b = __asyncValues(this.listByGroupPagingPage(resourceGroupName, projectName, groupName, options)), _c; _c = yield __await(_b.next()), !_c.done;) {
                    const page = _c.value;
                    yield __await(yield* __asyncDelegator(__asyncValues(page)));
                }
            }
            catch (e_1_1) { e_1 = { error: e_1_1 }; }
            finally {
                try {
                    if (_c && !_c.done && (_a = _b.return)) yield __await(_a.call(_b));
                }
                finally { if (e_1) throw e_1.error; }
            }
        });
    }
    /**
     * Get all assessments created in the project.
     *
     * Returns a json array of objects of type 'assessment' as specified in Models section.
     *
     * @param resourceGroupName Name of the Azure Resource Group that project is part of.
     * @param projectName Name of the Azure Migrate project.
     * @param options The options parameters.
     */
    listByProject(resourceGroupName, projectName, options) {
        const iter = this.listByProjectPagingAll(resourceGroupName, projectName, options);
        return {
            next() {
                return iter.next();
            },
            [Symbol.asyncIterator]() {
                return this;
            },
            byPage: (settings) => {
                if (settings === null || settings === void 0 ? void 0 : settings.maxPageSize) {
                    throw new Error("maxPageSize is not supported by this operation.");
                }
                return this.listByProjectPagingPage(resourceGroupName, projectName, options, settings);
            }
        };
    }
    listByProjectPagingPage(resourceGroupName, projectName, options, _settings) {
        return __asyncGenerator(this, arguments, function* listByProjectPagingPage_1() {
            let result;
            result = yield __await(this._listByProject(resourceGroupName, projectName, options));
            yield yield __await(result.value || []);
        });
    }
    listByProjectPagingAll(resourceGroupName, projectName, options) {
        return __asyncGenerator(this, arguments, function* listByProjectPagingAll_1() {
            var e_2, _a;
            try {
                for (var _b = __asyncValues(this.listByProjectPagingPage(resourceGroupName, projectName, options)), _c; _c = yield __await(_b.next()), !_c.done;) {
                    const page = _c.value;
                    yield __await(yield* __asyncDelegator(__asyncValues(page)));
                }
            }
            catch (e_2_1) { e_2 = { error: e_2_1 }; }
            finally {
                try {
                    if (_c && !_c.done && (_a = _b.return)) yield __await(_a.call(_b));
                }
                finally { if (e_2) throw e_2.error; }
            }
        });
    }
    /**
     * Get all assessments created for the specified group.
     *
     * Returns a json array of objects of type 'assessment' as specified in Models section.
     *
     * @param resourceGroupName Name of the Azure Resource Group that project is part of.
     * @param projectName Name of the Azure Migrate project.
     * @param groupName Unique name of a group within a project.
     * @param options The options parameters.
     */
    _listByGroup(resourceGroupName, projectName, groupName, options) {
        return this.client.sendOperationRequest({ resourceGroupName, projectName, groupName, options }, listByGroupOperationSpec);
    }
    /**
     * Get all assessments created in the project.
     *
     * Returns a json array of objects of type 'assessment' as specified in Models section.
     *
     * @param resourceGroupName Name of the Azure Resource Group that project is part of.
     * @param projectName Name of the Azure Migrate project.
     * @param options The options parameters.
     */
    _listByProject(resourceGroupName, projectName, options) {
        return this.client.sendOperationRequest({ resourceGroupName, projectName, options }, listByProjectOperationSpec);
    }
    /**
     * Get an existing assessment with the specified name. Returns a json object of type 'assessment' as
     * specified in Models section.
     * @param resourceGroupName Name of the Azure Resource Group that project is part of.
     * @param projectName Name of the Azure Migrate project.
     * @param groupName Unique name of a group within a project.
     * @param assessmentName Unique name of an assessment within a project.
     * @param options The options parameters.
     */
    get(resourceGroupName, projectName, groupName, assessmentName, options) {
        return this.client.sendOperationRequest({ resourceGroupName, projectName, groupName, assessmentName, options }, getOperationSpec);
    }
    /**
     * Create a new assessment with the given name and the specified settings. Since name of an assessment
     * in a project is a unique identifier, if an assessment with the name provided already exists, then
     * the existing assessment is updated.
     *
     * Any PUT operation, resulting in either create or update on an assessment, will cause the assessment
     * to go in a "InProgress" state. This will be indicated by the field 'computationState' on the
     * Assessment object. During this time no other PUT operation will be allowed on that assessment
     * object, nor will a Delete operation. Once the computation for the assessment is complete, the field
     * 'computationState' will be updated to 'Ready', and then other PUT or DELETE operations can happen on
     * the assessment.
     *
     * When assessment is under computation, any PUT will lead to a 400 - Bad Request error.
     *
     * @param resourceGroupName Name of the Azure Resource Group that project is part of.
     * @param projectName Name of the Azure Migrate project.
     * @param groupName Unique name of a group within a project.
     * @param assessmentName Unique name of an assessment within a project.
     * @param options The options parameters.
     */
    create(resourceGroupName, projectName, groupName, assessmentName, options) {
        return this.client.sendOperationRequest({ resourceGroupName, projectName, groupName, assessmentName, options }, createOperationSpec);
    }
    /**
     * Delete an assessment from the project. The machines remain in the assessment. Deleting a
     * non-existent assessment results in a no-operation.
     *
     * When an assessment is under computation, as indicated by the 'computationState' field, it cannot be
     * deleted. Any such attempt will return a 400 - Bad Request.
     *
     * @param resourceGroupName Name of the Azure Resource Group that project is part of.
     * @param projectName Name of the Azure Migrate project.
     * @param groupName Unique name of a group within a project.
     * @param assessmentName Unique name of an assessment within a project.
     * @param options The options parameters.
     */
    delete(resourceGroupName, projectName, groupName, assessmentName, options) {
        return this.client.sendOperationRequest({ resourceGroupName, projectName, groupName, assessmentName, options }, deleteOperationSpec);
    }
    /**
     * Get the URL for downloading the assessment in a report format.
     * @param resourceGroupName Name of the Azure Resource Group that project is part of.
     * @param projectName Name of the Azure Migrate project.
     * @param groupName Unique name of a group within a project.
     * @param assessmentName Unique name of an assessment within a project.
     * @param options The options parameters.
     */
    getReportDownloadUrl(resourceGroupName, projectName, groupName, assessmentName, options) {
        return this.client.sendOperationRequest({ resourceGroupName, projectName, groupName, assessmentName, options }, getReportDownloadUrlOperationSpec);
    }
}
// Operation Specifications
const serializer = coreClient.createSerializer(Mappers, /* isXml */ false);
const listByGroupOperationSpec = {
    path: "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Migrate/assessmentProjects/{projectName}/groups/{groupName}/assessments",
    httpMethod: "GET",
    responses: {
        200: {
            bodyMapper: Mappers.AssessmentResultList,
            headersMapper: Mappers.AssessmentsListByGroupHeaders
        },
        default: {
            bodyMapper: Mappers.CloudError
        }
    },
    queryParameters: [Parameters.apiVersion],
    urlParameters: [
        Parameters.$host,
        Parameters.subscriptionId,
        Parameters.resourceGroupName,
        Parameters.projectName,
        Parameters.groupName
    ],
    headerParameters: [Parameters.accept],
    serializer
};
const listByProjectOperationSpec = {
    path: "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Migrate/assessmentProjects/{projectName}/assessments",
    httpMethod: "GET",
    responses: {
        200: {
            bodyMapper: Mappers.AssessmentResultList,
            headersMapper: Mappers.AssessmentsListByProjectHeaders
        },
        default: {
            bodyMapper: Mappers.CloudError
        }
    },
    queryParameters: [Parameters.apiVersion],
    urlParameters: [
        Parameters.$host,
        Parameters.subscriptionId,
        Parameters.resourceGroupName,
        Parameters.projectName
    ],
    headerParameters: [Parameters.accept],
    serializer
};
const getOperationSpec = {
    path: "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Migrate/assessmentProjects/{projectName}/groups/{groupName}/assessments/{assessmentName}",
    httpMethod: "GET",
    responses: {
        200: {
            bodyMapper: Mappers.Assessment,
            headersMapper: Mappers.AssessmentsGetHeaders
        },
        default: {
            bodyMapper: Mappers.CloudError
        }
    },
    queryParameters: [Parameters.apiVersion],
    urlParameters: [
        Parameters.$host,
        Parameters.subscriptionId,
        Parameters.resourceGroupName,
        Parameters.projectName,
        Parameters.groupName,
        Parameters.assessmentName
    ],
    headerParameters: [Parameters.accept],
    serializer
};
const createOperationSpec = {
    path: "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Migrate/assessmentProjects/{projectName}/groups/{groupName}/assessments/{assessmentName}",
    httpMethod: "PUT",
    responses: {
        200: {
            bodyMapper: Mappers.Assessment,
            headersMapper: Mappers.AssessmentsCreateHeaders
        },
        201: {
            bodyMapper: Mappers.Assessment,
            headersMapper: Mappers.AssessmentsCreateHeaders
        },
        default: {
            bodyMapper: Mappers.CloudError
        }
    },
    requestBody: Parameters.assessment,
    queryParameters: [Parameters.apiVersion],
    urlParameters: [
        Parameters.$host,
        Parameters.subscriptionId,
        Parameters.resourceGroupName,
        Parameters.projectName,
        Parameters.groupName,
        Parameters.assessmentName
    ],
    headerParameters: [Parameters.accept, Parameters.contentType],
    mediaType: "json",
    serializer
};
const deleteOperationSpec = {
    path: "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Migrate/assessmentProjects/{projectName}/groups/{groupName}/assessments/{assessmentName}",
    httpMethod: "DELETE",
    responses: {
        200: {
            headersMapper: Mappers.AssessmentsDeleteHeaders
        },
        204: {},
        default: {
            bodyMapper: Mappers.CloudError
        }
    },
    queryParameters: [Parameters.apiVersion],
    urlParameters: [
        Parameters.$host,
        Parameters.subscriptionId,
        Parameters.resourceGroupName,
        Parameters.projectName,
        Parameters.groupName,
        Parameters.assessmentName
    ],
    headerParameters: [Parameters.accept],
    serializer
};
const getReportDownloadUrlOperationSpec = {
    path: "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Migrate/assessmentProjects/{projectName}/groups/{groupName}/assessments/{assessmentName}/downloadUrl",
    httpMethod: "POST",
    responses: {
        200: {
            bodyMapper: Mappers.DownloadUrl,
            headersMapper: Mappers.AssessmentsGetReportDownloadUrlHeaders
        },
        default: {
            bodyMapper: Mappers.CloudError
        }
    },
    queryParameters: [Parameters.apiVersion],
    urlParameters: [
        Parameters.$host,
        Parameters.subscriptionId,
        Parameters.resourceGroupName,
        Parameters.projectName,
        Parameters.groupName,
        Parameters.assessmentName
    ],
    headerParameters: [Parameters.accept],
    serializer
};
