// @azure/arm-migrate/dist-esm/src/operations/groups.js
/*
 * Copyright (c) Microsoft Corporation.
 * Licensed under the MIT License.
 *
 * Code generated by Microsoft (R) AutoRest Code Generator.
 * Changes may cause incorrect behavior and will be lost if the code is regenerated.
 */
import { __asyncDelegator, __asyncGenerator, __asyncValues, __await } from "tslib";
import * as coreClient from "@azure/core-client";
import * as Mappers from "../models/mappers";
import * as Parameters from "../models/parameters";
/// <reference lib="esnext.asynciterable" />
/** Class containing Groups operations. */
export class GroupsImpl {
    /**
     * Initialize a new instance of the class Groups class.
     * @param client Reference to the service client
     */
    constructor(client) {
        this.client = client;
    }
    /**
     * Get all groups created in the project. Returns a json array of objects of type 'group' as specified
     * in the Models section.
     * @param resourceGroupName Name of the Azure Resource Group that project is part of.
     * @param projectName Name of the Azure Migrate project.
     * @param options The options parameters.
     */
    listByProject(resourceGroupName, projectName, options) {
        const iter = this.listByProjectPagingAll(resourceGroupName, projectName, options);
        return {
            next() {
                return iter.next();
            },
            [Symbol.asyncIterator]() {
                return this;
            },
            byPage: (settings) => {
                if (settings === null || settings === void 0 ? void 0 : settings.maxPageSize) {
                    throw new Error("maxPageSize is not supported by this operation.");
                }
                return this.listByProjectPagingPage(resourceGroupName, projectName, options, settings);
            }
        };
    }
    listByProjectPagingPage(resourceGroupName, projectName, options, _settings) {
        return __asyncGenerator(this, arguments, function* listByProjectPagingPage_1() {
            let result;
            result = yield __await(this._listByProject(resourceGroupName, projectName, options));
            yield yield __await(result.value || []);
        });
    }
    listByProjectPagingAll(resourceGroupName, projectName, options) {
        return __asyncGenerator(this, arguments, function* listByProjectPagingAll_1() {
            var e_1, _a;
            try {
                for (var _b = __asyncValues(this.listByProjectPagingPage(resourceGroupName, projectName, options)), _c; _c = yield __await(_b.next()), !_c.done;) {
                    const page = _c.value;
                    yield __await(yield* __asyncDelegator(__asyncValues(page)));
                }
            }
            catch (e_1_1) { e_1 = { error: e_1_1 }; }
            finally {
                try {
                    if (_c && !_c.done && (_a = _b.return)) yield __await(_a.call(_b));
                }
                finally { if (e_1) throw e_1.error; }
            }
        });
    }
    /**
     * Get all groups created in the project. Returns a json array of objects of type 'group' as specified
     * in the Models section.
     * @param resourceGroupName Name of the Azure Resource Group that project is part of.
     * @param projectName Name of the Azure Migrate project.
     * @param options The options parameters.
     */
    _listByProject(resourceGroupName, projectName, options) {
        return this.client.sendOperationRequest({ resourceGroupName, projectName, options }, listByProjectOperationSpec);
    }
    /**
     * Get information related to a specific group in the project. Returns a json object of type 'group' as
     * specified in the models section.
     * @param resourceGroupName Name of the Azure Resource Group that project is part of.
     * @param projectName Name of the Azure Migrate project.
     * @param groupName Unique name of a group within a project.
     * @param options The options parameters.
     */
    get(resourceGroupName, projectName, groupName, options) {
        return this.client.sendOperationRequest({ resourceGroupName, projectName, groupName, options }, getOperationSpec);
    }
    /**
     * Create a new group by sending a json object of type 'group' as given in Models section as part of
     * the Request Body. The group name in a project is unique.
     *
     * This operation is Idempotent.
     *
     * @param resourceGroupName Name of the Azure Resource Group that project is part of.
     * @param projectName Name of the Azure Migrate project.
     * @param groupName Unique name of a group within a project.
     * @param options The options parameters.
     */
    create(resourceGroupName, projectName, groupName, options) {
        return this.client.sendOperationRequest({ resourceGroupName, projectName, groupName, options }, createOperationSpec);
    }
    /**
     * Delete the group from the project. The machines remain in the project. Deleting a non-existent group
     * results in a no-operation.
     *
     * A group is an aggregation mechanism for machines in a project. Therefore, deleting group does not
     * delete machines in it.
     *
     * @param resourceGroupName Name of the Azure Resource Group that project is part of.
     * @param projectName Name of the Azure Migrate project.
     * @param groupName Unique name of a group within a project.
     * @param options The options parameters.
     */
    delete(resourceGroupName, projectName, groupName, options) {
        return this.client.sendOperationRequest({ resourceGroupName, projectName, groupName, options }, deleteOperationSpec);
    }
    /**
     * Update machines in group by adding or removing machines.
     * @param resourceGroupName Name of the Azure Resource Group that project is part of.
     * @param projectName Name of the Azure Migrate project.
     * @param groupName Unique name of a group within a project.
     * @param options The options parameters.
     */
    updateMachines(resourceGroupName, projectName, groupName, options) {
        return this.client.sendOperationRequest({ resourceGroupName, projectName, groupName, options }, updateMachinesOperationSpec);
    }
}
// Operation Specifications
const serializer = coreClient.createSerializer(Mappers, /* isXml */ false);
const listByProjectOperationSpec = {
    path: "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Migrate/assessmentProjects/{projectName}/groups",
    httpMethod: "GET",
    responses: {
        200: {
            bodyMapper: Mappers.GroupResultList,
            headersMapper: Mappers.GroupsListByProjectHeaders
        },
        default: {
            bodyMapper: Mappers.CloudError
        }
    },
    queryParameters: [Parameters.apiVersion],
    urlParameters: [
        Parameters.$host,
        Parameters.subscriptionId,
        Parameters.resourceGroupName,
        Parameters.projectName
    ],
    headerParameters: [Parameters.accept],
    serializer
};
const getOperationSpec = {
    path: "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Migrate/assessmentProjects/{projectName}/groups/{groupName}",
    httpMethod: "GET",
    responses: {
        200: {
            bodyMapper: Mappers.Group,
            headersMapper: Mappers.GroupsGetHeaders
        },
        default: {
            bodyMapper: Mappers.CloudError
        }
    },
    queryParameters: [Parameters.apiVersion],
    urlParameters: [
        Parameters.$host,
        Parameters.subscriptionId,
        Parameters.resourceGroupName,
        Parameters.projectName,
        Parameters.groupName
    ],
    headerParameters: [Parameters.accept],
    serializer
};
const createOperationSpec = {
    path: "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Migrate/assessmentProjects/{projectName}/groups/{groupName}",
    httpMethod: "PUT",
    responses: {
        200: {
            bodyMapper: Mappers.Group,
            headersMapper: Mappers.GroupsCreateHeaders
        },
        201: {
            bodyMapper: Mappers.Group,
            headersMapper: Mappers.GroupsCreateHeaders
        },
        default: {
            bodyMapper: Mappers.CloudError
        }
    },
    requestBody: Parameters.group,
    queryParameters: [Parameters.apiVersion],
    urlParameters: [
        Parameters.$host,
        Parameters.subscriptionId,
        Parameters.resourceGroupName,
        Parameters.projectName,
        Parameters.groupName
    ],
    headerParameters: [Parameters.accept, Parameters.contentType],
    mediaType: "json",
    serializer
};
const deleteOperationSpec = {
    path: "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Migrate/assessmentProjects/{projectName}/groups/{groupName}",
    httpMethod: "DELETE",
    responses: {
        200: {
            headersMapper: Mappers.GroupsDeleteHeaders
        },
        204: {},
        default: {
            bodyMapper: Mappers.CloudError
        }
    },
    queryParameters: [Parameters.apiVersion],
    urlParameters: [
        Parameters.$host,
        Parameters.subscriptionId,
        Parameters.resourceGroupName,
        Parameters.projectName,
        Parameters.groupName
    ],
    headerParameters: [Parameters.accept],
    serializer
};
const updateMachinesOperationSpec = {
    path: "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Migrate/assessmentProjects/{projectName}/groups/{groupName}/updateMachines",
    httpMethod: "POST",
    responses: {
        200: {
            bodyMapper: Mappers.Group,
            headersMapper: Mappers.GroupsUpdateMachinesHeaders
        },
        default: {
            bodyMapper: Mappers.CloudError
        }
    },
    requestBody: Parameters.groupUpdateProperties,
    queryParameters: [Parameters.apiVersion],
    urlParameters: [
        Parameters.$host,
        Parameters.subscriptionId,
        Parameters.resourceGroupName,
        Parameters.projectName,
        Parameters.groupName
    ],
    headerParameters: [Parameters.accept, Parameters.contentType],
    mediaType: "json",
    serializer
};
