// @azure/arm-migrate/dist-esm/src/operations/importCollectors.js
/*
 * Copyright (c) Microsoft Corporation.
 * Licensed under the MIT License.
 *
 * Code generated by Microsoft (R) AutoRest Code Generator.
 * Changes may cause incorrect behavior and will be lost if the code is regenerated.
 */
import { __asyncDelegator, __asyncGenerator, __asyncValues, __await } from "tslib";
import * as coreClient from "@azure/core-client";
import * as Mappers from "../models/mappers";
import * as Parameters from "../models/parameters";
/// <reference lib="esnext.asynciterable" />
/** Class containing ImportCollectors operations. */
export class ImportCollectorsImpl {
    /**
     * Initialize a new instance of the class ImportCollectors class.
     * @param client Reference to the service client
     */
    constructor(client) {
        this.client = client;
    }
    /**
     * Get a list of Import collector.
     * @param resourceGroupName Name of the Azure Resource Group that project is part of.
     * @param projectName Name of the Azure Migrate project.
     * @param options The options parameters.
     */
    listByProject(resourceGroupName, projectName, options) {
        const iter = this.listByProjectPagingAll(resourceGroupName, projectName, options);
        return {
            next() {
                return iter.next();
            },
            [Symbol.asyncIterator]() {
                return this;
            },
            byPage: (settings) => {
                if (settings === null || settings === void 0 ? void 0 : settings.maxPageSize) {
                    throw new Error("maxPageSize is not supported by this operation.");
                }
                return this.listByProjectPagingPage(resourceGroupName, projectName, options, settings);
            }
        };
    }
    listByProjectPagingPage(resourceGroupName, projectName, options, _settings) {
        return __asyncGenerator(this, arguments, function* listByProjectPagingPage_1() {
            let result;
            result = yield __await(this._listByProject(resourceGroupName, projectName, options));
            yield yield __await(result.value || []);
        });
    }
    listByProjectPagingAll(resourceGroupName, projectName, options) {
        return __asyncGenerator(this, arguments, function* listByProjectPagingAll_1() {
            var e_1, _a;
            try {
                for (var _b = __asyncValues(this.listByProjectPagingPage(resourceGroupName, projectName, options)), _c; _c = yield __await(_b.next()), !_c.done;) {
                    const page = _c.value;
                    yield __await(yield* __asyncDelegator(__asyncValues(page)));
                }
            }
            catch (e_1_1) { e_1 = { error: e_1_1 }; }
            finally {
                try {
                    if (_c && !_c.done && (_a = _b.return)) yield __await(_a.call(_b));
                }
                finally { if (e_1) throw e_1.error; }
            }
        });
    }
    /**
     * Get a list of Import collector.
     * @param resourceGroupName Name of the Azure Resource Group that project is part of.
     * @param projectName Name of the Azure Migrate project.
     * @param options The options parameters.
     */
    _listByProject(resourceGroupName, projectName, options) {
        return this.client.sendOperationRequest({ resourceGroupName, projectName, options }, listByProjectOperationSpec);
    }
    /**
     * Get a Import collector.
     * @param resourceGroupName Name of the Azure Resource Group that project is part of.
     * @param projectName Name of the Azure Migrate project.
     * @param importCollectorName Unique name of a Import collector within a project.
     * @param options The options parameters.
     */
    get(resourceGroupName, projectName, importCollectorName, options) {
        return this.client.sendOperationRequest({ resourceGroupName, projectName, importCollectorName, options }, getOperationSpec);
    }
    /**
     * Create or Update Import collector
     * @param resourceGroupName Name of the Azure Resource Group that project is part of.
     * @param projectName Name of the Azure Migrate project.
     * @param importCollectorName Unique name of a Import collector within a project.
     * @param options The options parameters.
     */
    create(resourceGroupName, projectName, importCollectorName, options) {
        return this.client.sendOperationRequest({ resourceGroupName, projectName, importCollectorName, options }, createOperationSpec);
    }
    /**
     * Delete a Import collector from the project.
     * @param resourceGroupName Name of the Azure Resource Group that project is part of.
     * @param projectName Name of the Azure Migrate project.
     * @param importCollectorName Unique name of a Import collector within a project.
     * @param options The options parameters.
     */
    delete(resourceGroupName, projectName, importCollectorName, options) {
        return this.client.sendOperationRequest({ resourceGroupName, projectName, importCollectorName, options }, deleteOperationSpec);
    }
}
// Operation Specifications
const serializer = coreClient.createSerializer(Mappers, /* isXml */ false);
const listByProjectOperationSpec = {
    path: "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Migrate/assessmentProjects/{projectName}/importcollectors",
    httpMethod: "GET",
    responses: {
        200: {
            bodyMapper: Mappers.ImportCollectorList,
            headersMapper: Mappers.ImportCollectorsListByProjectHeaders
        },
        default: {
            bodyMapper: Mappers.CloudError
        }
    },
    queryParameters: [Parameters.apiVersion],
    urlParameters: [
        Parameters.$host,
        Parameters.subscriptionId,
        Parameters.resourceGroupName,
        Parameters.projectName
    ],
    headerParameters: [Parameters.accept],
    serializer
};
const getOperationSpec = {
    path: "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Migrate/assessmentProjects/{projectName}/importcollectors/{importCollectorName}",
    httpMethod: "GET",
    responses: {
        200: {
            bodyMapper: Mappers.ImportCollector,
            headersMapper: Mappers.ImportCollectorsGetHeaders
        },
        default: {
            bodyMapper: Mappers.CloudError
        }
    },
    queryParameters: [Parameters.apiVersion],
    urlParameters: [
        Parameters.$host,
        Parameters.subscriptionId,
        Parameters.resourceGroupName,
        Parameters.projectName,
        Parameters.importCollectorName
    ],
    headerParameters: [Parameters.accept],
    serializer
};
const createOperationSpec = {
    path: "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Migrate/assessmentProjects/{projectName}/importcollectors/{importCollectorName}",
    httpMethod: "PUT",
    responses: {
        200: {
            bodyMapper: Mappers.ImportCollector,
            headersMapper: Mappers.ImportCollectorsCreateHeaders
        },
        201: {
            bodyMapper: Mappers.ImportCollector,
            headersMapper: Mappers.ImportCollectorsCreateHeaders
        },
        default: {
            bodyMapper: Mappers.CloudError
        }
    },
    requestBody: Parameters.collectorBody3,
    queryParameters: [Parameters.apiVersion],
    urlParameters: [
        Parameters.$host,
        Parameters.subscriptionId,
        Parameters.resourceGroupName,
        Parameters.projectName,
        Parameters.importCollectorName
    ],
    headerParameters: [Parameters.accept, Parameters.contentType],
    mediaType: "json",
    serializer
};
const deleteOperationSpec = {
    path: "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Migrate/assessmentProjects/{projectName}/importcollectors/{importCollectorName}",
    httpMethod: "DELETE",
    responses: {
        200: {
            headersMapper: Mappers.ImportCollectorsDeleteHeaders
        },
        204: {},
        default: {
            bodyMapper: Mappers.CloudError
        }
    },
    queryParameters: [Parameters.apiVersion],
    urlParameters: [
        Parameters.$host,
        Parameters.subscriptionId,
        Parameters.resourceGroupName,
        Parameters.projectName,
        Parameters.importCollectorName
    ],
    headerParameters: [Parameters.accept],
    serializer
};
