// @azure/arm-migrate/dist-esm/src/operations/machines.js
/*
 * Copyright (c) Microsoft Corporation.
 * Licensed under the MIT License.
 *
 * Code generated by Microsoft (R) AutoRest Code Generator.
 * Changes may cause incorrect behavior and will be lost if the code is regenerated.
 */
import { __asyncDelegator, __asyncGenerator, __asyncValues, __await } from "tslib";
import { setContinuationToken } from "../pagingHelper";
import * as coreClient from "@azure/core-client";
import * as Mappers from "../models/mappers";
import * as Parameters from "../models/parameters";
/// <reference lib="esnext.asynciterable" />
/** Class containing Machines operations. */
export class MachinesImpl {
    /**
     * Initialize a new instance of the class Machines class.
     * @param client Reference to the service client
     */
    constructor(client) {
        this.client = client;
    }
    /**
     * Get data of all the machines available in the project. Returns a json array of objects of type
     * 'machine' defined in Models section.
     * @param resourceGroupName Name of the Azure Resource Group that project is part of.
     * @param projectName Name of the Azure Migrate project.
     * @param options The options parameters.
     */
    listByProject(resourceGroupName, projectName, options) {
        const iter = this.listByProjectPagingAll(resourceGroupName, projectName, options);
        return {
            next() {
                return iter.next();
            },
            [Symbol.asyncIterator]() {
                return this;
            },
            byPage: (settings) => {
                if (settings === null || settings === void 0 ? void 0 : settings.maxPageSize) {
                    throw new Error("maxPageSize is not supported by this operation.");
                }
                return this.listByProjectPagingPage(resourceGroupName, projectName, options, settings);
            }
        };
    }
    listByProjectPagingPage(resourceGroupName, projectName, options, settings) {
        return __asyncGenerator(this, arguments, function* listByProjectPagingPage_1() {
            let result;
            let continuationToken = settings === null || settings === void 0 ? void 0 : settings.continuationToken;
            if (!continuationToken) {
                result = yield __await(this._listByProject(resourceGroupName, projectName, options));
                let page = result.value || [];
                continuationToken = result.nextLink;
                setContinuationToken(page, continuationToken);
                yield yield __await(page);
            }
            while (continuationToken) {
                result = yield __await(this._listByProjectNext(resourceGroupName, projectName, continuationToken, options));
                continuationToken = result.nextLink;
                let page = result.value || [];
                setContinuationToken(page, continuationToken);
                yield yield __await(page);
            }
        });
    }
    listByProjectPagingAll(resourceGroupName, projectName, options) {
        return __asyncGenerator(this, arguments, function* listByProjectPagingAll_1() {
            var e_1, _a;
            try {
                for (var _b = __asyncValues(this.listByProjectPagingPage(resourceGroupName, projectName, options)), _c; _c = yield __await(_b.next()), !_c.done;) {
                    const page = _c.value;
                    yield __await(yield* __asyncDelegator(__asyncValues(page)));
                }
            }
            catch (e_1_1) { e_1 = { error: e_1_1 }; }
            finally {
                try {
                    if (_c && !_c.done && (_a = _b.return)) yield __await(_a.call(_b));
                }
                finally { if (e_1) throw e_1.error; }
            }
        });
    }
    /**
     * Get data of all the machines available in the project. Returns a json array of objects of type
     * 'machine' defined in Models section.
     * @param resourceGroupName Name of the Azure Resource Group that project is part of.
     * @param projectName Name of the Azure Migrate project.
     * @param options The options parameters.
     */
    _listByProject(resourceGroupName, projectName, options) {
        return this.client.sendOperationRequest({ resourceGroupName, projectName, options }, listByProjectOperationSpec);
    }
    /**
     * Get the machine with the specified name. Returns a json object of type 'machine' defined in Models
     * section.
     * @param resourceGroupName Name of the Azure Resource Group that project is part of.
     * @param projectName Name of the Azure Migrate project.
     * @param machineName Unique name of a machine in private datacenter.
     * @param options The options parameters.
     */
    get(resourceGroupName, projectName, machineName, options) {
        return this.client.sendOperationRequest({ resourceGroupName, projectName, machineName, options }, getOperationSpec);
    }
    /**
     * ListByProjectNext
     * @param resourceGroupName Name of the Azure Resource Group that project is part of.
     * @param projectName Name of the Azure Migrate project.
     * @param nextLink The nextLink from the previous successful call to the ListByProject method.
     * @param options The options parameters.
     */
    _listByProjectNext(resourceGroupName, projectName, nextLink, options) {
        return this.client.sendOperationRequest({ resourceGroupName, projectName, nextLink, options }, listByProjectNextOperationSpec);
    }
}
// Operation Specifications
const serializer = coreClient.createSerializer(Mappers, /* isXml */ false);
const listByProjectOperationSpec = {
    path: "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Migrate/assessmentProjects/{projectName}/machines",
    httpMethod: "GET",
    responses: {
        200: {
            bodyMapper: Mappers.MachineResultList,
            headersMapper: Mappers.MachinesListByProjectHeaders
        },
        default: {
            bodyMapper: Mappers.CloudError
        }
    },
    queryParameters: [Parameters.apiVersion],
    urlParameters: [
        Parameters.$host,
        Parameters.subscriptionId,
        Parameters.resourceGroupName,
        Parameters.projectName
    ],
    headerParameters: [Parameters.accept],
    serializer
};
const getOperationSpec = {
    path: "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Migrate/assessmentProjects/{projectName}/machines/{machineName}",
    httpMethod: "GET",
    responses: {
        200: {
            bodyMapper: Mappers.Machine,
            headersMapper: Mappers.MachinesGetHeaders
        },
        default: {
            bodyMapper: Mappers.CloudError
        }
    },
    queryParameters: [Parameters.apiVersion],
    urlParameters: [
        Parameters.$host,
        Parameters.subscriptionId,
        Parameters.resourceGroupName,
        Parameters.projectName,
        Parameters.machineName
    ],
    headerParameters: [Parameters.accept],
    serializer
};
const listByProjectNextOperationSpec = {
    path: "{nextLink}",
    httpMethod: "GET",
    responses: {
        200: {
            bodyMapper: Mappers.MachineResultList,
            headersMapper: Mappers.MachinesListByProjectNextHeaders
        },
        default: {
            bodyMapper: Mappers.CloudError
        }
    },
    urlParameters: [
        Parameters.$host,
        Parameters.subscriptionId,
        Parameters.resourceGroupName,
        Parameters.projectName,
        Parameters.nextLink
    ],
    headerParameters: [Parameters.accept],
    serializer
};
